--- src/generic/grid.cpp.orig	Tue Oct  2 12:12:10 2001
+++ src/generic/grid.cpp	Tue Oct  2 22:19:51 2001
@@ -6396,7 +6421,6 @@
 {
     wxArrayString lines;
 
-    dc.SetClippingRegion( rect );
     StringToLines( value, lines );
 
 
@@ -6418,6 +6442,7 @@
     long textWidth, textHeight;
     long lineWidth, lineHeight;
 
+    dc.SetClippingRegion( rect );
     if ( lines.GetCount() )
     {
         GetTextBoxSize( dc, lines, &textWidth, &textHeight );

 	  	 
