diff -r -u -p a/samples/widgets/spinbtn.cpp b/samples/widgets/spinbtn.cpp
--- a/samples/widgets/spinbtn.cpp	2021-04-19 11:47:36.127255700 +0200
+++ b/samples/widgets/spinbtn.cpp	2021-04-19 11:48:32.174130700 +0200
@@ -483,6 +483,14 @@ void SpinBtnWidgetsPage::OnButtonSetBase
 
 void SpinBtnWidgetsPage::OnButtonSetValue(wxCommandEvent& WXUNUSED(event))
 {
+    if ( m_textValue->IsEmpty() )
+    {
+        m_spinctrl->SetValue( wxEmptyString );
+        m_spinctrldbl->SetValue( wxEmptyString );
+
+        return;
+    }
+
     long val;
     if ( !m_textValue->GetValue().ToLong(&val) || !IsValidValue(val) )
     {
@@ -499,7 +507,8 @@ void SpinBtnWidgetsPage::OnButtonSetValu
 void SpinBtnWidgetsPage::OnUpdateUIValueButton(wxUpdateUIEvent& event)
 {
     long val;
-    event.Enable( m_textValue->GetValue().ToLong(&val) && IsValidValue(val) );
+    event.Enable( m_textValue->IsEmpty() ||
+                  ( m_textValue->GetValue().ToLong(&val) && IsValidValue(val) ) );
 }
 
 void SpinBtnWidgetsPage::OnUpdateUIMinMaxButton(wxUpdateUIEvent& event)
