diff -bru wxWidgets/samples/mdi/mdi.cpp /mnt/winxp/wxWidgets/samples/mdi/mdi.cpp
--- wxWidgets/samples/mdi/mdi.cpp	2013-04-05 14:52:38.000000000 -0700
+++ /mnt/winxp/wxWidgets/samples/mdi/mdi.cpp	2013-04-18 13:16:16.125000000 -0700
@@ -221,6 +221,8 @@
 
     wxMenuBar *mbar = new wxMenuBar;
     mbar->Append(menuFile, "&File");
+    wxMenu *edit = new wxMenu;
+    mbar->Append( edit, "&Edit" );
     mbar->Append(menuHelp, "&Help");
 
     return mbar;
@@ -256,6 +258,8 @@
     (void)wxMessageBox("wxWidgets 2.0 MDI Demo\n"
                        "Author: Julian Smart (c) 1997\n"
                        "Usage: mdi.exe", "About MDI Demo");
+    wxMenu *menu = new wxMenu();
+    GetMenuBar()->Insert( 2, menu, "Foo" );
 }
 
 void MyFrame::OnNewWindow(wxCommandEvent& WXUNUSED(event) )
@@ -425,7 +429,7 @@
 
     // create our menu bar: it will be shown instead of the main frame one when
     // we're active
-#if wxUSE_MENUS
+/*#if wxUSE_MENUS
     wxMenuBar *mbar = MyFrame::CreateMainMenubar();
     mbar->GetMenu(0)->Insert(1, wxID_CLOSE, "&Close child\tCtrl-W",
                              "Close this window");
@@ -449,7 +453,7 @@
 
     // Associate the menu bar with the frame
     SetMenuBar(mbar);
-#endif // wxUSE_MENUS
+#endif // wxUSE_MENUS*/
 
     // this should work for MDI frames as well as for normal ones, provided
     // they can be resized at all
diff -bru wxWidgets/src/common/menucmn.cpp /mnt/winxp/wxWidgets/src/common/menucmn.cpp
--- wxWidgets/src/common/menucmn.cpp	2013-04-05 14:40:50.000000000 -0700
+++ /mnt/winxp/wxWidgets/src/common/menucmn.cpp	2013-04-08 17:29:39.578125000 -0700
@@ -423,7 +423,7 @@
 wxMenuItem *wxMenuBase::Remove(wxMenuItem *item)
 {
     wxCHECK_MSG( item, NULL, wxT("invalid item in wxMenu::Remove") );
-
+    m_items.DeleteObject( item );
     return DoRemove(item);
 }
 
diff -bru wxWidgets/src/msw/mdi.cpp /mnt/winxp/wxWidgets/src/msw/mdi.cpp
--- wxWidgets/src/msw/mdi.cpp	2013-04-05 14:42:47.000000000 -0700
+++ /mnt/winxp/wxWidgets/src/msw/mdi.cpp	2013-04-19 19:27:58.453125000 -0700
@@ -326,9 +326,6 @@
 {
     if ( m_windowMenu )
     {
-        // For correct handling of the events from this menu we also must
-        // attach it to the menu bar.
-        m_windowMenu->Attach(GetMenuBar());
 
         MDIInsertWindowMenu(GetClientWindow(), m_hMenu, GetMDIWindowMenu(this));
     }
@@ -340,7 +337,6 @@
     {
         MDIRemoveWindowMenu(GetClientWindow(), m_hMenu);
 
-        m_windowMenu->Detach();
     }
 }
 
@@ -1483,18 +1479,18 @@
             if ( label == wxGetStockLabel(wxID_HELP, wxSTOCK_NOFLAGS) )
             {
                 inserted = true;
-                ::InsertMenu(hmenu, i, MF_BYPOSITION | MF_POPUP | MF_STRING,
-                             (UINT_PTR)menuWin,
-                             wxString(wxGetTranslation(WINDOW_MENU_LABEL)).t_str());
+                wxMDIParentFrame *frame = (wxMDIParentFrame *) win->GetParent();
+                wxMenu *windowMenu = frame->GetWindowMenu();
+                frame->GetMenuBar()->Insert( i, windowMenu, wxString( wxGetTranslation( WINDOW_MENU_LABEL ) ).t_str() );
                 break;
             }
         }
 
         if ( !inserted )
         {
-            ::AppendMenu(hmenu, MF_POPUP,
-                         (UINT_PTR)menuWin,
-                         wxString(wxGetTranslation(WINDOW_MENU_LABEL)).t_str());
+            wxMDIParentFrame *frame = (wxMDIParentFrame *) win->GetParent();
+            wxMenu *windowMenu = frame->GetWindowMenu();
+            frame->GetMenuBar()->Append( windowMenu, wxString( wxGetTranslation( WINDOW_MENU_LABEL ) ).t_str() );
         }
     }
 
@@ -1526,10 +1522,8 @@
 
             if ( wxStrcmp(buf, wxGetTranslation(WINDOW_MENU_LABEL)) == 0 )
             {
-                if ( !::RemoveMenu(hmenu, i, MF_BYPOSITION) )
-                {
-                    wxLogLastError(wxT("RemoveMenu"));
-                }
+                wxMDIParentFrame *frame = (wxMDIParentFrame *) win->GetParent();
+                frame->GetMenuBar()->Remove( i );
 
                 break;
             }
