Index: src/aui/auibook.cpp
===================================================================
--- src/aui/auibook.cpp	(revision 63602)
+++ src/aui/auibook.cpp	(working copy)
@@ -728,7 +728,7 @@
 
 int wxAuiDefaultTabArt::ShowDropDown(wxWindow* wnd,
                                      const wxAuiNotebookPageArray& pages,
-                                     int active_idx)
+                                     int /*active_idx*/)
 {
     wxMenu menuPopup;
 
@@ -743,14 +743,12 @@
         if (caption.IsEmpty())
             caption = wxT(" ");
 
-        menuPopup.AppendCheckItem(1000+i, caption);
+        wxMenuItem* item = new wxMenuItem(0, 1000+i, caption);
+        if (page.bitmap.IsOk())
+            item->SetBitmap(page.bitmap);
+        menuPopup.Append(item);
     }
 
-    if (active_idx != -1)
-    {
-        menuPopup.Check(1000+active_idx, true);
-    }
-
     // find out where to put the popup menu of window items
     wxPoint pt = ::wxGetMousePosition();
     pt = wnd->ScreenToClient(pt);
@@ -4187,6 +4185,8 @@
     {
         SetSelection(idx);
     }
+
+    evt.Skip();
 }
 
 void wxAuiNotebook::OnNavigationKeyNotebook(wxNavigationKeyEvent& event)
