diff --git a/src/stc/ScintillaWX.cpp b/src/stc/ScintillaWX.cpp
index 153329c..b2e4134 100644
--- a/src/stc/ScintillaWX.cpp
+++ b/src/stc/ScintillaWX.cpp
@@ -749,6 +749,7 @@ void ScintillaWX::DoPaint(wxDC* dc, wxRect rect) {
     if (paintState == paintAbandoned) {
         // Painting area was insufficient to cover new styling or brace
         // highlight positions
+        stc->Refresh(false); // Refresh required to clear the clip region on Windows/GTK. May not be need on MAC.
         FullPaintDC(dc);
     }
     paintState = notPainting;
