Index: src/common/utilscmn.cpp
===================================================================
--- src/common/utilscmn.cpp	(revision 65461)
+++ src/common/utilscmn.cpp	(working copy)
@@ -1285,7 +1285,7 @@
 
 wxColour wxGetColourFromUser(wxWindow *parent, const wxColour& colInit, const wxString& caption)
 {
-    wxColourData data;
+    static wxColourData data;
     data.SetChooseFull(true);
     if ( colInit.Ok() )
     {
