 samples/minimal/minimal.cpp | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/samples/minimal/minimal.cpp b/samples/minimal/minimal.cpp
index a78e462..025c29e 100644
--- a/samples/minimal/minimal.cpp
+++ b/samples/minimal/minimal.cpp
@@ -127,10 +127,25 @@ bool MyApp::OnInit()
     // create the main application window
     MyFrame *frame = new MyFrame("Minimal wxWidgets App");
 
+    wxTextCtrl* txtCtrl1 = new wxTextCtrl(frame, -1);
+    txtCtrl1->SetFocus();
+
+    wxDialog dlg(frame, -1, "title");
+
+    wxTextCtrl* txtCtrl2 = new wxTextCtrl(frame, -1);
+    txtCtrl2->SetFocus();
+
+    wxBoxSizer* bs = new wxBoxSizer(wxVERTICAL);
+    bs->Add(txtCtrl1);
+    bs->Add(txtCtrl2);
+    frame->SetSizer(bs);
+
     // and show it (the frames, unlike simple controls, are not shown when
     // created initially)
     frame->Show(true);
 
+    dlg.ShowModal();
+
     // success: wxApp::OnRun() will be called which will enter the main message
     // loop and the application will run. If we returned false here, the
     // application would exit immediately.
