Index: src/msw/printwin.cpp
===================================================================
--- src/msw/printwin.cpp	(revision 60612)
+++ src/msw/printwin.cpp	(working copy)
@@ -50,6 +50,11 @@
 #include "wx/msw/private.h"
 #include "wx/msw/dcprint.h"
 
+#if wxUSE_HIGH_QUALITY_PREVIEW && wxUSE_PROGRESSDLG
+#include "wx/progdlg.h"
+#endif // wxUSE_HIGH_QUALITY_PREVIEW && wxUSE_PROGRESSDLG
+
+
 #include <stdlib.h>
 
 #ifndef __WIN32__
@@ -656,6 +661,10 @@
     wxRect todo(0, 0, m_pageWidth, initialStep); // rect to render
     int nextFinalLine = 0; // first not-yet-rendered output line
 
+#if wxUSE_PROGRESSDLG
+    wxProgressDialog progressDlg("Rendering high quality preview", "High quality preview is rendering", m_pageWidth, m_previewFrame);
+#endif // wxUSE_PROGRESSDLG
+
     while ( todo.y < m_pageHeight )
     {
         todo.SetBottom(wxMin(todo.GetBottom(), m_pageHeight - 1));
@@ -667,6 +676,10 @@
                                  todo,
                                  pageNum) )
         {
+#if wxUSE_PROGRESSDLG
+            progressDlg.Pulse();
+#endif // wxUSE_PROGRESSDLG
+
             if ( todo.height < 20 )
             {
                 // something is very wrong if we can't render even at this
@@ -691,6 +704,9 @@
 
         // move to the next segment
         todo.Offset(0, todo.height);
+#if wxUSE_PROGRESSDLG
+        progressDlg.Update(wxMin(todo.GetBottom(), m_pageHeight - 1));
+#endif // wxUSE_PROGRESSDLG
     }
 
     return true;
