diff --git "a/C:\\Users\\Rudresh Pandey\\Downloads\\wxWidgets-3.1.0\\samples\\dataview\\mymodels.cpp" "b/G:\\wxWidgets-3.1.0\\samples\\dataview\\mymodels.cpp"
index 590d3f2..f95a1be 100644
--- "a/C:\\Users\\Rudresh Pandey\\Downloads\\wxWidgets-3.1.0\\samples\\dataview\\mymodels.cpp"	
+++ "b/G:\\wxWidgets-3.1.0\\samples\\dataview\\mymodels.cpp"
@@ -337,7 +337,7 @@ MyListModel::MyListModel() :
     static const unsigned NUMBER_REAL_ITEMS = 100;
 
     m_textColValues.reserve(NUMBER_REAL_ITEMS);
-    m_textColValues.push_back("first row with long label to test ellipsization");
+    m_textColValues.push_back("first row with long label to test ellipsization.\nThis should be in next line!");
     for (unsigned int i = 1; i < NUMBER_REAL_ITEMS; i++)
     {
         m_textColValues.push_back(wxString::Format("real row %d", i));
