--- demo_ORIG.cpp	2012-03-04 01:30:21.000000000 +0000
+++ demo.cpp	2012-08-22 09:36:50.000000000 +0100
@@ -446,6 +446,12 @@
     wxMenuBar *menuBar = new wxMenuBar;
     menuBar->Append(menuFile, wxT("&File"));
 
+    wxMenu *editMenu = new wxMenu;
+    editMenu->Append(wxID_CUT, _("Cu&t\tCtrl+X"));
+    editMenu->Append(wxID_COPY, _("&Copy\tCtrl+C"));
+    editMenu->Append(wxID_PASTE, _("&Paste\tCtrl+V"));
+    menuBar->Append(editMenu, wxT("&Edit"));
+
     // ... and attach this menu bar to the frame
     SetMenuBar(menuBar);
 
