--- htmlwin_ORIG.h	2012-04-29 01:30:27.000000000 +0100
+++ htmlwin.h	2012-08-22 10:14:28.000000000 +0100
@@ -412,6 +412,8 @@
     void OnKeyUp(wxKeyEvent& event);
     void OnDoubleClick(wxMouseEvent& event);
     void OnCopy(wxCommandEvent& event);
+    void OnUpdateCopy(wxUpdateUIEvent& event);
+    void OnUpdateCutOrPaste(wxUpdateUIEvent& event);
     void OnClipboardEvent(wxClipboardTextEvent& event);
     void OnMouseEnter(wxMouseEvent& event);
     void OnMouseLeave(wxMouseEvent& event);
--- htmlwin_ORIG.cpp	2012-04-29 01:30:27.000000000 +0100
+++ htmlwin.cpp	2012-08-22 10:14:06.000000000 +0100
@@ -1555,6 +1555,16 @@
     (void) CopySelection();
 }
 
+void wxHtmlWindow::OnUpdateCopy(wxUpdateUIEvent& event)
+{
+    event.Enable(m_selection);
+}
+
+void wxHtmlWindow::OnUpdateCutOrPaste(wxUpdateUIEvent& event)
+{
+    event.Enable(false);
+}
+
 void wxHtmlWindow::OnClipboardEvent(wxClipboardTextEvent& WXUNUSED(event))
 {
     (void) CopySelection();
@@ -1696,6 +1706,9 @@
     EVT_MOUSE_CAPTURE_LOST(wxHtmlWindow::OnMouseCaptureLost)
     EVT_KEY_UP(wxHtmlWindow::OnKeyUp)
     EVT_MENU(wxID_COPY, wxHtmlWindow::OnCopy)
+    EVT_UPDATE_UI(wxID_COPY, wxHtmlWindow::OnUpdateCopy)
+    EVT_UPDATE_UI(wxID_CUT, wxHtmlWindow::OnUpdateCutOrPaste)
+    EVT_UPDATE_UI(wxID_PASTE, wxHtmlWindow::OnUpdateCutOrPaste)
     EVT_TEXT_COPY(wxID_ANY, wxHtmlWindow::OnClipboardEvent)
 #endif // wxUSE_CLIPBOARD
 END_EVENT_TABLE()
