diff -ru wxWidgets.orig/samples/aui/auidemo.cpp wxWidgets/samples/aui/auidemo.cpp
--- wxWidgets.orig/samples/aui/auidemo.cpp	2007-05-02 15:23:40.000000000 +0200
+++ wxWidgets/samples/aui/auidemo.cpp	2007-06-14 21:10:32.000000000 +0200
@@ -845,6 +845,16 @@
                   Left().Layer(1).Position(1).
                   CloseButton(true).MaximizeButton(true));
 
+
+    m_mgr.AddPane(CreateTreeCtrl(), wxAuiPaneInfo().
+                  Name(wxT("test8-1")).Caption(wxT("Tree Pane")).
+                  Right().Layer(1).Position(1).
+                  CloseButton(true).MaximizeButton(true));
+    m_mgr.AddPane(CreateTreeCtrl(), wxAuiPaneInfo().
+                  Name(wxT("test8-2")).Caption(wxT("Tree Pane")).
+                  Right().Layer(1).Position(1).
+                  CloseButton(true).MaximizeButton(true));
+    
     m_mgr.AddPane(CreateSizeReportCtrl(), wxAuiPaneInfo().
                   Name(wxT("test9")).Caption(wxT("Min Size 200x100")).
                   BestSize(wxSize(200,100)).MinSize(wxSize(200,100)).
@@ -929,7 +939,9 @@
     m_mgr.GetPane(wxT("tb1")).Hide();
     m_mgr.GetPane(wxT("tb6")).Hide();
     m_mgr.GetPane(wxT("test8")).Show().Left().Layer(0).Row(0).Position(0);
-    m_mgr.GetPane(wxT("test10")).Show().Bottom().Layer(0).Row(0).Position(0);
+    m_mgr.GetPane(wxT("test8-1")).Show().Right().Layer(0).Row(0).Position(0);
+    m_mgr.GetPane(wxT("test8-2")).Show().Right().Layer(0).Row(0).Position(0);
+    m_mgr.GetPane(wxT("test10")).Show(false).Bottom().Layer(0).Row(0).Position(0);
     m_mgr.GetPane(wxT("notebook_content")).Show();
     wxString perspective_default = m_mgr.SavePerspective();
 

 	  	 
