From c57330971d2e6d70eee2b0f0977e339c81bb5428 Mon Sep 17 00:00:00 2001
From: "Spencer T. Parkin" <spencerparkin@outlook.com>
Date: Mon, 16 Sep 2013 00:08:19 -0600
Subject: [PATCH] expose bug in auto-resize column header calculator

---
 samples/dataview/dataview.cpp | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/samples/dataview/dataview.cpp b/samples/dataview/dataview.cpp
index 5deed63..4564bc6 100644
--- a/samples/dataview/dataview.cpp
+++ b/samples/dataview/dataview.cpp
@@ -601,6 +601,19 @@ void MyFrame::BuildDataViewCtrl(wxPanel* parent, unsigned int nPanel, unsigned l
 
             // select initially the ninth symphony:
             m_ctrl[0]->Select(m_music_model->GetNinthItem());
+
+			unsigned int wantedModelColID = 4;	// popularity column ID
+			wxDataViewColumn* foundColumn = 0;
+			for( unsigned int i = 0; i < m_ctrl[0]->GetColumnCount() && !foundColumn; i++ )
+			{
+				wxDataViewColumn* column = m_ctrl[0]->GetColumn(i);
+				if( column->GetModelColumn() == wantedModelColID )
+					foundColumn = column;
+			}
+			if( foundColumn )
+				m_ctrl[0]->DeleteColumn( foundColumn );
+
+			// Now double-click the "custom" column header to auto-size it to fit, and you'll get the bug to happen.
         }
         break;
 
-- 
1.8.0.msysgit.0

