--- C:\dev\sdk\wxWidgets-2.6.3\src\msw\thread.1.95.cpp	2006-07-24 16:59:29.001 +0200
+++ C:\dev\sdk\wxWidgets-2.6.3\src\msw\thread.cpp	2006-07-24 17:02:59.000 +0200
@@ -693,6 +693,12 @@
     // (note that even in console applications we might have to process
     // messages if we use wxExecute() or timers or ...)
     DWORD result wxDUMMY_INITIALIZE(0);
+
+	// we need to try to handle the messages but if it turns out that
+	//	the message loop is down, we absolutely need to ignore them else
+	//	it will always finish killing abruptely the thread
+    bool actuallyWaitLoopMsgs = true;
+
     do
     {
         if ( wxThread::IsMain() )
@@ -715,7 +721,7 @@
                    &m_hThread,     // the objects
                    false,          // don't wait for all objects
                    INFINITE,       // no timeout
-                   QS_ALLINPUT|QS_ALLPOSTMESSAGE   // return as soon as there are any events
+                   actuallyWaitLoopMsgs ? QS_ALLINPUT|QS_ALLPOSTMESSAGE :0  // return as soon as there are any events
                  );
 
         switch ( result )
@@ -752,10 +758,17 @@
 
                     if ( traits && !traits->DoMessageFromThreadWait() )
                     {
-                        // WM_QUIT received: kill the thread
-                        Kill();
+                        if (actuallyWaitLoopMsgs)
+                        {
+                            actuallyWaitLoopMsgs = false;
+                        }
+                        else
+                        {
+                            // WM_QUIT received: kill the thread
+                            Kill();
 
-                        return wxTHREAD_KILLED;
+                            return wxTHREAD_KILLED;
+                        }
                     }
                 }
                 break;


 	  	 
