diff -uNr wxWidgets/src/common/url.cpp wxWidgets.l/src/common/url.cpp
--- wxWidgets/src/common/url.cpp	2021-03-21 19:16:50.396633043 +0300
+++ wxWidgets.l/src/common/url.cpp	2021-03-26 13:27:01.876056425 +0300
@@ -349,6 +349,19 @@
         m_error = wxURL_PROTOERR;
         return NULL;
     }
+#if wxUSE_PROTOCOL_HTTP
+    // check for HTTP/Redirect case:
+    if( m_protocol->IsKindOf( wxCLASSINFO(wxHTTP)) )
+    {
+        wxHTTP* p_http = (wxHTTP*) m_protocol;
+        int ecode = p_http->GetResponse();
+        if( (ecode>=300)&&(ecode<=399) )
+        {
+            *this = p_http->GetHeader( "location" );
+            return GetInputStream();
+        }
+    }
+#endif
 
     return the_i_stream;
 }
