Index: socket.cpp
===================================================================
--- socket.cpp	(revision 62501)
+++ socket.cpp	(working copy)
@@ -1372,7 +1372,7 @@
     // we're never going to become ready in a client if we're not connected any
     // more (OTOH a server can call this to precisely wait for a connection so
     // do wait for it in this case)
-    if ( !m_impl->IsServer() && !m_connected && !m_establishing )
+    if ( !m_impl->IsServer() && m_impl->m_stream && !m_connected && !m_establishing )
         return -1;
 
     // This can be set to true from Interrupt() to exit this function a.s.a.p.
