Index: docs/doxygen/groups/class_grid.h
===================================================================
--- docs/doxygen/groups/class_grid.h	(revision 0)
+++ docs/doxygen/groups/class_grid.h	(revision 0)
@@ -0,0 +1,15 @@
+/////////////////////////////////////////////////////////////////////////////
+// Name:        class_grid.h
+// Purpose:     wxGrid classes group docs
+// Author:      wxWidgets team
+// RCS-ID:      $Id: $
+// Licence:     wxWindows license
+/////////////////////////////////////////////////////////////////////////////
+
+/**
+
+@defgroup group_class_grid wxGrid
+@ingroup group_class
+
+*/
+
