Index: textctrl_osx.cpp
===================================================================
--- textctrl_osx.cpp	(revision 72060)
+++ textctrl_osx.cpp	(working copy)
@@ -166,6 +166,10 @@
 
 bool wxTextCtrl::SetStyle(long start, long end, const wxTextAttr& style)
 {
+    const long contentLength = GetValue().Length();
+    wxASSERT_MSG( (start <= contentLength) && (end <= contentLength), wxT("Range is not valid for the current content of the wxTextCtrl") );
+    wxUnusedVar( contentLength );
+
     if (GetTextPeer())
         GetTextPeer()->SetStyle( start , end , style ) ;
 
