diff --git a/src/richtext/richtextfontpage.cpp b/src/richtext/richtextfontpage.cpp
index 1a95a47..cf11798 100644
--- a/src/richtext/richtextfontpage.cpp
+++ b/src/richtext/richtextfontpage.cpp
@@ -686,7 +686,7 @@ bool wxRichTextFontPage::TransferDataToWindow()
                 m_smallCapitalsCtrl->Set3StateValue(wxCHK_UNCHECKED);
         }
         else
-            m_capitalsCtrl->Set3StateValue(wxCHK_UNDETERMINED);
+            m_smallCapitalsCtrl->Set3StateValue(wxCHK_UNDETERMINED);
 
         if ( attr->GetTextEffectFlags() & (wxTEXT_ATTR_EFFECT_SUPERSCRIPT | wxTEXT_ATTR_EFFECT_SUBSCRIPT) )
         {
