--- \src\aui\auibook.cpp    Mon Dec 04 08:59:10 2006
+++ auibook.cpp    Tue Mar 06 18:28:51 2007
@@ -1977,6 +1977,64 @@
 {
 }
 
+void wxAuiTabCtrl::MakeTabVisible(size_t tab_idx)
+{
+    if (tab_idx < m_tab_offset)
+    {
+        m_tab_offset = tab_idx;
+        Refresh();
+        Update();
+    }
+    else
+    {
+        unsigned int tab_offset = m_tab_offset;
+        
+        // TODO: Take width of right side buttons into account 
+        
+        // Keep increasing tab_offset until tab is visible
+        while (tab_offset < tab_idx)
+        {
+            int tab_width = 0; // total width from offset
+            wxClientDC dc(this);
+            for (unsigned int i = tab_offset; i <= tab_idx; ++i) {
+                wxAuiNotebookPage& page = m_pages.Item(i);
+
+                // determine if a close button is on this tab
+                bool close_button = false;
+                if ((m_flags & wxAUI_NB_CLOSE_ON_ALL_TABS) != 0 ||
+                    ((m_flags & wxAUI_NB_CLOSE_ON_ACTIVE_TAB) != 0 && page.active))
+                {
+                    close_button = true;
+                }
+                
+                // figure out the size of the tab
+                int x_extent;
+                const wxSize tab_size = m_art->GetTabSize(dc,
+                                                         this,
+                                                         page.caption,
+                                                         page.bitmap,
+                                                         page.active,
+                                                         close_button ?
+                                                          wxAUI_BUTTON_STATE_NORMAL :
+                                                          wxAUI_BUTTON_STATE_HIDDEN,
+                                                         &x_extent);
+                tab_width += tab_size.x;
+            }
+
+            if (tab_width < m_rect.GetRight()) break;
+
+            ++tab_offset;
+        }
+
+        if (tab_offset > m_tab_offset)
+        {
+            m_tab_offset = tab_offset;
+            Refresh();
+            Update();
+        }
+    }
+}
+
 void wxAuiTabCtrl::OnPaint(wxPaintEvent&)
 {
     wxPaintDC dc(this);
@@ -2827,6 +2885,7 @@
 
             ctrl->SetActivePage(ctrl_idx);
             DoSizing();
+            ctrl->MakeTabVisible(ctrl_idx);
             ctrl->DoShowHide();
 
 

 	  	 
