Index: docs/latex/wx/category.tex
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/docs/latex/wx/category.tex,v
retrieving revision 1.105
diff -u -2 -r1.105 category.tex
--- docs/latex/wx/category.tex	2005/02/20 00:07:32	1.105
+++ docs/latex/wx/category.tex	2005/03/13 02:24:31
@@ -494,4 +494,5 @@
 \twocolitem{\helpref{wxFFileInputStream}{wxffileinputstream}}{Another file input stream class}
 \twocolitem{\helpref{wxFFileOutputStream}{wxffileoutputstream}}{Another file output stream class}
+\twocolitem{\helpref{wxTempFileOutputStream}{wxtempfileoutputstream}}{Stream to safely replace an existing file}
 \twocolitem{\helpref{wxStringInputStream}{wxstringinputstream}}{String input stream class}
 \twocolitem{\helpref{wxStringOutputStream}{wxstringoutputstream}}{String output stream class}
Index: docs/latex/wx/classes.tex
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/docs/latex/wx/classes.tex,v
retrieving revision 1.180
diff -u -2 -r1.180 classes.tex
--- docs/latex/wx/classes.tex	2005/02/14 15:40:05	1.180
+++ docs/latex/wx/classes.tex	2005/03/13 02:24:34
@@ -324,4 +324,5 @@
 \input tcpservr.tex
 \input tempfile.tex
+\input tempfilestrm.tex
 \input text.tex
 \input txtdatob.tex
Index: docs/latex/wx/tempfile.tex
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/docs/latex/wx/tempfile.tex,v
retrieving revision 1.10
diff -u -2 -r1.10 tempfile.tex
--- docs/latex/wx/tempfile.tex	2004/05/04 08:25:49	1.10
+++ docs/latex/wx/tempfile.tex	2005/03/13 02:24:46
@@ -44,5 +44,6 @@
 \wxheading{See also:}
 
-\helpref{wxFile}{wxfile}
+\helpref{wxFile}{wxfile}\\
+\helpref{wxTempFileOutputStream}{wxtempfileoutputstream}
 
 \latexignore{\rtfignore{\wxheading{Members}}}
@@ -79,4 +80,23 @@
 
 Returns {\tt true} if the file was successfully opened.
+
+\membersection{wxTempFile::Length}\label{wxtempfilelength}
+
+\constfunc{wxFileOffset}{Length}{\void}
+
+Returns the length of the file.
+
+\membersection{wxTempFile::Seek}\label{wxtempfileseek}
+
+\func{wxFileOffset}{Seek}{\param{wxFileOffset }{ofs}, \param{wxSeekMode }{mode = wxFromStart}}
+
+Seeks to the specified position.
+
+\membersection{wxTempFile::Tell}\label{wxtempfiletell}
+
+\constfunc{wxFileOffset}{Tell}{\void}
+
+Returns the current position or wxInvalidOffset if file is not opened or if another
+error occurred.
 
 \membersection{wxTempFile::Write}\label{wxtempfilewrite}
Index: include/wx/file.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/file.h,v
retrieving revision 1.43
diff -u -2 -r1.43 file.h
--- include/wx/file.h	2005/02/12 21:53:43	1.43
+++ include/wx/file.h	2005/03/13 02:24:54
@@ -165,4 +165,11 @@
   // is the file opened?
   bool IsOpened() const { return m_file.IsOpened(); }
+    // get current file length
+  wxFileOffset Length() const { return m_file.Length(); }
+    // move ptr ofs bytes related to start/current offset/end of file
+  wxFileOffset Seek(wxFileOffset ofs, wxSeekMode mode = wxFromStart)
+    { return m_file.Seek(ofs, mode); }
+    // get current offset
+  wxFileOffset Tell() const { return m_file.Tell(); }
 
   // I/O (both functions return true on success, false on failure)
Index: include/wx/wfstream.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/wfstream.h,v
retrieving revision 1.25
diff -u -2 -r1.25 wfstream.h
--- include/wx/wfstream.h	2005/02/17 10:42:16	1.25
+++ include/wx/wfstream.h	2005/03/13 02:25:04
@@ -89,4 +89,29 @@
 };
 
+class WXDLLIMPEXP_BASE wxTempFileOutputStream : public wxOutputStream
+{
+public:
+    wxTempFileOutputStream(const wxString& fileName);
+    virtual ~wxTempFileOutputStream();
+
+    bool Close() { return Commit(); }
+    virtual bool Commit() { return m_file->Commit(); }
+    virtual void Discard() { m_file->Discard(); }
+
+    wxFileOffset GetLength() const { return m_file->Length(); }
+    bool IsSeekable() const { return true; }
+
+protected:
+    size_t OnSysWrite(const void *buffer, size_t size);
+    wxFileOffset OnSysSeek(wxFileOffset pos, wxSeekMode mode)
+        { return m_file->Seek(pos, mode); }
+    wxFileOffset OnSysTell() const { return m_file->Tell(); }
+
+private:
+    wxTempFile *m_file;
+
+    DECLARE_NO_COPY_CLASS(wxTempFileOutputStream)
+};
+
 class WXDLLIMPEXP_BASE wxFileStream : public wxFileInputStream,
                                       public wxFileOutputStream
Index: src/common/wfstream.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/common/wfstream.cpp,v
retrieving revision 1.38
diff -u -2 -r1.38 wfstream.cpp
--- src/common/wfstream.cpp	2005/02/17 10:42:22	1.38
+++ src/common/wfstream.cpp	2005/03/13 02:26:10
@@ -185,4 +185,31 @@
 
 // ----------------------------------------------------------------------------
+// wxTempFileOutputStream
+// ----------------------------------------------------------------------------
+
+wxTempFileOutputStream::wxTempFileOutputStream(const wxString& fileName)
+{
+    m_file = new wxTempFile(fileName);
+
+    if (!m_file->IsOpened())
+        m_lasterror = wxSTREAM_WRITE_ERROR;
+}
+
+wxTempFileOutputStream::~wxTempFileOutputStream()
+{
+    if (m_file->IsOpened())
+        Discard();
+    delete m_file;
+}
+
+size_t wxTempFileOutputStream::OnSysWrite(const void *buffer, size_t size)
+{
+    if (IsOk() && m_file->Write(buffer, size))
+        return size;
+    m_lasterror = wxSTREAM_WRITE_ERROR;
+    return 0;
+}
+
+// ----------------------------------------------------------------------------
 // wxFileStream
 // ----------------------------------------------------------------------------
Index: tests/Makefile.in
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/tests/Makefile.in,v
retrieving revision 1.36
diff -u -2 -r1.36 Makefile.in
--- tests/Makefile.in	2005/02/19 03:55:39	1.36
+++ tests/Makefile.in	2005/03/13 02:27:06
@@ -72,4 +72,5 @@
 	test_memstream.o \
 	test_sstream.o \
+	test_tempfile.o \
 	test_textstreamtest.o \
 	test_zlibstream.o \
@@ -321,4 +322,7 @@
 test_sstream.o: $(srcdir)/streams/sstream.cpp $(TEST_ODEP)
 	$(CXXC) -c -o $@ $(TEST_CXXFLAGS) $(srcdir)/streams/sstream.cpp
+
+test_tempfile.o: $(srcdir)/streams/tempfile.cpp $(TEST_ODEP)
+	$(CXXC) -c -o $@ $(TEST_CXXFLAGS) $(srcdir)/streams/tempfile.cpp
 
 test_textstreamtest.o: $(srcdir)/streams/textstreamtest.cpp $(TEST_ODEP)
Index: tests/makefile.bcc
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/tests/makefile.bcc,v
retrieving revision 1.29
diff -u -2 -r1.29 makefile.bcc
--- tests/makefile.bcc	2005/02/12 21:53:46	1.29
+++ tests/makefile.bcc	2005/03/13 02:27:06
@@ -1,5 +1,5 @@
 # =========================================================================
 #     This makefile was generated by
-#     Bakefile 0.1.6 (http://bakefile.sourceforge.net)
+#     Bakefile 0.1.7 (http://bakefile.sourceforge.net)
 #     Do not modify, all changes will be overwritten!
 # =========================================================================
@@ -29,8 +29,8 @@
 	$(LIBDIRNAME)\$(PORTNAME)$(WXUNIVNAME)$(WXUNICODEFLAG)$(WXDEBUGFLAG)
 TEST_CXXFLAGS = $(__RUNTIME_LIBS) -I$(BCCDIR)\include $(__DEBUGINFO) \
-	$(__OPTIMIZEFLAG) -tWM -D__WXMSW__ $(__WXUNIV_DEFINE_p) $(__DEBUG_DEFINE_p) \
-	$(__UNICODE_DEFINE_p) -I.\..\include -I$(SETUPHDIR) -I. $(__DLLFLAG_p) \
-	-DwxUSE_GUI=0 $(CPPUNIT_CFLAGS) -Hu -H=$(OBJS)\testprec_test.csm $(CPPFLAGS) \
-	$(CXXFLAGS)
+	$(__OPTIMIZEFLAG) $(__THREADSFLAG_0) -D__WXMSW__ $(__WXUNIV_DEFINE_p) \
+	$(__DEBUG_DEFINE_p) $(__UNICODE_DEFINE_p) -I.\..\include -I$(SETUPHDIR) -I. \
+	$(__DLLFLAG_p) -DwxUSE_GUI=0 $(CPPUNIT_CFLAGS) -Hu \
+	-H=$(OBJS)\testprec_test.csm $(CPPFLAGS) $(CXXFLAGS)
 TEST_OBJECTS =  \
 	$(OBJS)\test_dummy.obj \
@@ -63,11 +63,12 @@
 	$(OBJS)\test_memstream.obj \
 	$(OBJS)\test_sstream.obj \
+	$(OBJS)\test_tempfile.obj \
 	$(OBJS)\test_textstreamtest.obj \
 	$(OBJS)\test_zlibstream.obj \
 	$(OBJS)\test_uris.obj
 TEST_GUI_CXXFLAGS = $(__RUNTIME_LIBS) -I$(BCCDIR)\include $(__DEBUGINFO) \
-	$(__OPTIMIZEFLAG) -tWM -D__WXMSW__ $(__WXUNIV_DEFINE_p) $(__DEBUG_DEFINE_p) \
-	$(__UNICODE_DEFINE_p) -I.\..\include -I$(SETUPHDIR) -I. $(__DLLFLAG_p) \
-	-I.\..\samples -DNOPCH $(CPPUNIT_CFLAGS) -Hu \
+	$(__OPTIMIZEFLAG) $(__THREADSFLAG_0) -D__WXMSW__ $(__WXUNIV_DEFINE_p) \
+	$(__DEBUG_DEFINE_p) $(__UNICODE_DEFINE_p) -I.\..\include -I$(SETUPHDIR) -I. \
+	$(__DLLFLAG_p) -I.\..\samples -DNOPCH $(CPPUNIT_CFLAGS) -Hu \
 	-H=$(OBJS)\testprec_test_gui.csm $(CPPFLAGS) $(CXXFLAGS)
 TEST_GUI_OBJECTS =  \
@@ -145,4 +146,16 @@
 __OPTIMIZEFLAG = -O2
 !endif
+!if "$(USE_THREADS)" == "0"
+__THREADSFLAG = 
+!endif
+!if "$(USE_THREADS)" == "1"
+__THREADSFLAG = mt
+!endif
+!if "$(USE_THREADS)" == "0"
+__THREADSFLAG_0 = 
+!endif
+!if "$(USE_THREADS)" == "1"
+__THREADSFLAG_0 = -tWM
+!endif
 !if "$(RUNTIME_LIBS)" == "dynamic"
 __RUNTIME_LIBS = -tWR
@@ -233,5 +246,5 @@
 $(OBJS)\test.exe: $(OBJS)\test_dummy.obj  $(TEST_OBJECTS)
 	ilink32 -Tpe -q $(LDFLAGS) -L$(BCCDIR)\lib -L$(BCCDIR)\lib\psdk $(__DEBUGINFO)  -L$(LIBDIRNAME) -ap $(CPPUNIT_LIBS) @&&|
-	c0x32.obj $(TEST_OBJECTS),$@,, $(__WXLIB_NET_p)  $(__WXLIB_BASE_p)  $(__WXLIB_MONO_p) $(__LIB_TIFF_p) $(__LIB_JPEG_p) $(__LIB_PNG_p) wxzlib$(WXDEBUGFLAG).lib  wxregex$(WXUNICODEFLAG)$(WXDEBUGFLAG).lib wxexpat$(WXDEBUGFLAG).lib $(EXTRALIBS_FOR_BASE) $(__UNICOWS_LIB_p) ole2w32.lib oleacc.lib odbc32.lib import32.lib cw32mt$(__RUNTIME_LIBS_0).lib,,
+	c0x32.obj $(TEST_OBJECTS),$@,, $(__WXLIB_NET_p)  $(__WXLIB_BASE_p)  $(__WXLIB_MONO_p) $(__LIB_TIFF_p) $(__LIB_JPEG_p) $(__LIB_PNG_p) wxzlib$(WXDEBUGFLAG).lib  wxregex$(WXUNICODEFLAG)$(WXDEBUGFLAG).lib wxexpat$(WXDEBUGFLAG).lib $(EXTRALIBS_FOR_BASE) $(__UNICOWS_LIB_p) ole2w32.lib oleacc.lib odbc32.lib import32.lib cw32$(__THREADSFLAG)$(__RUNTIME_LIBS_0).lib,,
 |
 
@@ -239,5 +252,5 @@
 $(OBJS)\test_gui.exe: $(OBJS)\test_gui_dummy.obj  $(TEST_GUI_OBJECTS)  $(OBJS)\test_gui_sample.res
 	ilink32 -Tpe -q $(LDFLAGS) -L$(BCCDIR)\lib -L$(BCCDIR)\lib\psdk $(__DEBUGINFO)  -L$(LIBDIRNAME) $(CPPUNIT_LIBS) -ap @&&|
-	c0x32.obj $(TEST_GUI_OBJECTS),$@,, $(__WXLIB_CORE_p)  $(__WXLIB_BASE_p)  $(__WXLIB_MONO_p) $(__LIB_TIFF_p) $(__LIB_JPEG_p) $(__LIB_PNG_p) wxzlib$(WXDEBUGFLAG).lib  wxregex$(WXUNICODEFLAG)$(WXDEBUGFLAG).lib wxexpat$(WXDEBUGFLAG).lib $(EXTRALIBS_FOR_BASE) $(__UNICOWS_LIB_p) ole2w32.lib oleacc.lib odbc32.lib  import32.lib cw32mt$(__RUNTIME_LIBS_0).lib,, $(OBJS)\test_gui_sample.res
+	c0x32.obj $(TEST_GUI_OBJECTS),$@,, $(__WXLIB_CORE_p)  $(__WXLIB_BASE_p)  $(__WXLIB_MONO_p) $(__LIB_TIFF_p) $(__LIB_JPEG_p) $(__LIB_PNG_p) wxzlib$(WXDEBUGFLAG).lib  wxregex$(WXUNICODEFLAG)$(WXDEBUGFLAG).lib wxexpat$(WXDEBUGFLAG).lib $(EXTRALIBS_FOR_BASE) $(__UNICOWS_LIB_p) ole2w32.lib oleacc.lib odbc32.lib  import32.lib cw32$(__THREADSFLAG)$(__RUNTIME_LIBS_0).lib,, $(OBJS)\test_gui_sample.res
 |
 !endif
@@ -332,4 +345,7 @@
 
 $(OBJS)\test_sstream.obj: .\streams\sstream.cpp
+	$(CXX) -q -c -P -o$@ $(TEST_CXXFLAGS) $**
+
+$(OBJS)\test_tempfile.obj: .\streams\tempfile.cpp
 	$(CXX) -q -c -P -o$@ $(TEST_CXXFLAGS) $**
 
Index: tests/makefile.gcc
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/tests/makefile.gcc,v
retrieving revision 1.29
diff -u -2 -r1.29 makefile.gcc
--- tests/makefile.gcc	2005/02/12 21:53:47	1.29
+++ tests/makefile.gcc	2005/03/13 02:27:07
@@ -1,5 +1,5 @@
 # =========================================================================
 #     This makefile was generated by
-#     Bakefile 0.1.6 (http://bakefile.sourceforge.net)
+#     Bakefile 0.1.7 (http://bakefile.sourceforge.net)
 #     Do not modify, all changes will be overwritten!
 # =========================================================================
@@ -22,6 +22,6 @@
 	-D__WXMSW__ $(__WXUNIV_DEFINE_p) $(__DEBUG_DEFINE_p) $(__UNICODE_DEFINE_p) \
 	-I.\..\include -I$(SETUPHDIR) -W -Wall -I. $(__DLLFLAG_p) -DwxUSE_GUI=0 \
-	$(CPPUNIT_CFLAGS) $(__EXCEPTIONSFLAG) $(__EXCEPTIONSFLAG_0) \
-	-Wno-ctor-dtor-privacy $(CPPFLAGS) $(CXXFLAGS)
+	$(CPPUNIT_CFLAGS) $(__RTTIFLAG) $(__EXCEPTIONSFLAG) -Wno-ctor-dtor-privacy \
+	$(CPPFLAGS) $(CXXFLAGS)
 TEST_OBJECTS =  \
 	$(OBJS)\test_dummy.o \
@@ -54,4 +54,5 @@
 	$(OBJS)\test_memstream.o \
 	$(OBJS)\test_sstream.o \
+	$(OBJS)\test_tempfile.o \
 	$(OBJS)\test_textstreamtest.o \
 	$(OBJS)\test_zlibstream.o \
@@ -60,6 +61,6 @@
 	-DHAVE_W32API_H -D__WXMSW__ $(__WXUNIV_DEFINE_p) $(__DEBUG_DEFINE_p) \
 	$(__UNICODE_DEFINE_p) -I.\..\include -I$(SETUPHDIR) -W -Wall -I. \
-	$(__DLLFLAG_p) -I.\..\samples -DNOPCH $(CPPUNIT_CFLAGS) $(__EXCEPTIONSFLAG) \
-	$(__EXCEPTIONSFLAG_0) -Wno-ctor-dtor-privacy $(CPPFLAGS) $(CXXFLAGS)
+	$(__DLLFLAG_p) -I.\..\samples -DNOPCH $(CPPUNIT_CFLAGS) $(__RTTIFLAG) \
+	$(__EXCEPTIONSFLAG) -Wno-ctor-dtor-privacy $(CPPFLAGS) $(CXXFLAGS)
 TEST_GUI_OBJECTS =  \
 	$(OBJS)\test_gui_sample_rc.o \
@@ -144,15 +145,21 @@
 __OPTIMIZEFLAG = -O2
 endif
-ifeq ($(USE_EXCEPTIONS),0)
-__EXCEPTIONSFLAG = -fno-rtti
+ifeq ($(USE_THREADS),0)
+__THREADSFLAG = 
 endif
-ifeq ($(USE_EXCEPTIONS),1)
-__EXCEPTIONSFLAG = 
+ifeq ($(USE_THREADS),1)
+__THREADSFLAG = -mthreads
+endif
+ifeq ($(USE_RTTI),0)
+__RTTIFLAG = -fno-rtti
 endif
+ifeq ($(USE_RTTI),1)
+__RTTIFLAG = 
+endif
 ifeq ($(USE_EXCEPTIONS),0)
-__EXCEPTIONSFLAG_0 = -fno-exceptions
+__EXCEPTIONSFLAG = -fno-exceptions
 endif
 ifeq ($(USE_EXCEPTIONS),1)
-__EXCEPTIONSFLAG_0 = 
+__EXCEPTIONSFLAG = 
 endif
 ifeq ($(WXUNIV),1)
@@ -225,9 +232,9 @@
 
 $(OBJS)\test.exe: $(TEST_OBJECTS)
-	$(CXX) -o $@ $(TEST_OBJECTS) $(LDFLAGS) $(__DEBUGINFO) -mthreads -L$(LIBDIRNAME)  $(CPPUNIT_LIBS)  $(__WXLIB_NET_p)  $(__WXLIB_BASE_p)  $(__WXLIB_MONO_p) $(__LIB_TIFF_p) $(__LIB_JPEG_p) $(__LIB_PNG_p) -lwxzlib$(WXDEBUGFLAG)  -lwxregex$(WXUNICODEFLAG)$(WXDEBUGFLAG) -lwxexpat$(WXDEBUGFLAG) $(EXTRALIBS_FOR_BASE) $(__UNICOWS_LIB_p) -lkernel32 -luser32 -lgdi32 -lcomdlg32 -lwinspool -lwinmm -lshell32 -lcomctl32 -lole32 -loleaut32 -luuid -lrpcrt4 -ladvapi32 -lwsock32 -lodbc32
+	$(CXX) -o $@ $(TEST_OBJECTS) $(LDFLAGS) $(__DEBUGINFO) $(__THREADSFLAG) -L$(LIBDIRNAME)  $(CPPUNIT_LIBS)  $(__WXLIB_NET_p)  $(__WXLIB_BASE_p)  $(__WXLIB_MONO_p) $(__LIB_TIFF_p) $(__LIB_JPEG_p) $(__LIB_PNG_p) -lwxzlib$(WXDEBUGFLAG)  -lwxregex$(WXUNICODEFLAG)$(WXDEBUGFLAG) -lwxexpat$(WXDEBUGFLAG) $(EXTRALIBS_FOR_BASE) $(__UNICOWS_LIB_p) -lkernel32 -luser32 -lgdi32 -lcomdlg32 -lwinspool -lwinmm -lshell32 -lcomctl32 -lole32 -loleaut32 -luuid -lrpcrt4 -ladvapi32 -lwsock32 -lodbc32
 
 ifeq ($(USE_GUI),1)
 $(OBJS)\test_gui.exe: $(TEST_GUI_OBJECTS) $(OBJS)\test_gui_sample_rc.o
-	$(CXX) -o $@ $(TEST_GUI_OBJECTS) $(LDFLAGS) $(__DEBUGINFO) -mthreads -L$(LIBDIRNAME) $(CPPUNIT_LIBS)   $(__WXLIB_CORE_p)  $(__WXLIB_BASE_p)  $(__WXLIB_MONO_p) $(__LIB_TIFF_p) $(__LIB_JPEG_p) $(__LIB_PNG_p) -lwxzlib$(WXDEBUGFLAG)  -lwxregex$(WXUNICODEFLAG)$(WXDEBUGFLAG) -lwxexpat$(WXDEBUGFLAG) $(EXTRALIBS_FOR_BASE) $(__UNICOWS_LIB_p) -lkernel32 -luser32 -lgdi32 -lcomdlg32 -lwinspool -lwinmm -lshell32 -lcomctl32 -lole32 -loleaut32 -luuid -lrpcrt4 -ladvapi32 -lwsock32 -lodbc32
+	$(CXX) -o $@ $(TEST_GUI_OBJECTS) $(LDFLAGS) $(__DEBUGINFO) $(__THREADSFLAG) -L$(LIBDIRNAME) $(CPPUNIT_LIBS)   $(__WXLIB_CORE_p)  $(__WXLIB_BASE_p)  $(__WXLIB_MONO_p) $(__LIB_TIFF_p) $(__LIB_JPEG_p) $(__LIB_PNG_p) -lwxzlib$(WXDEBUGFLAG)  -lwxregex$(WXUNICODEFLAG)$(WXDEBUGFLAG) -lwxexpat$(WXDEBUGFLAG) $(EXTRALIBS_FOR_BASE) $(__UNICOWS_LIB_p) -lkernel32 -luser32 -lgdi32 -lcomdlg32 -lwinspool -lwinmm -lshell32 -lcomctl32 -lole32 -loleaut32 -luuid -lrpcrt4 -ladvapi32 -lwsock32 -lodbc32
 endif
 
@@ -321,4 +328,7 @@
 
 $(OBJS)\test_sstream.o: ./streams/sstream.cpp
+	$(CXX) -c -o $@ $(TEST_CXXFLAGS) $<
+
+$(OBJS)\test_tempfile.o: ./streams/tempfile.cpp
 	$(CXX) -c -o $@ $(TEST_CXXFLAGS) $<
 
Index: tests/makefile.vc
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/tests/makefile.vc,v
retrieving revision 1.29
diff -u -2 -r1.29 makefile.vc
--- tests/makefile.vc	2005/02/12 21:53:47	1.29
+++ tests/makefile.vc	2005/03/13 02:27:09
@@ -1,5 +1,5 @@
 # =========================================================================
 #     This makefile was generated by
-#     Bakefile 0.1.6 (http://bakefile.sourceforge.net)
+#     Bakefile 0.1.7 (http://bakefile.sourceforge.net)
 #     Do not modify, all changes will be overwritten!
 # =========================================================================
@@ -19,10 +19,10 @@
 SETUPHDIR = \
 	$(LIBDIRNAME)\$(PORTNAME)$(WXUNIVNAME)$(WXUNICODEFLAG)$(WXDEBUGFLAG)
-TEST_CXXFLAGS = /M$(__RUNTIME_LIBS)$(__DEBUGRUNTIME_3) /DWIN32 $(__DEBUGINFO) \
-	/Fd$(OBJS)\test.pdb $(____DEBUGRUNTIME_2_p) $(__OPTIMIZEFLAG) \
-	$(__NO_VC_CRTDBG_p) /D__WXMSW__ $(__WXUNIV_DEFINE_p) $(__DEBUG_DEFINE_p) \
-	$(__UNICODE_DEFINE_p) /I.\..\include /I$(SETUPHDIR) /W4 /I. $(__DLLFLAG_p) \
-	/D_CONSOLE /DwxUSE_GUI=0 $(CPPUNIT_CFLAGS) $(__EXCEPTIONSFLAG) \
-	$(__EXCEPTIONSFLAG_0) /Yu"testprec.h" /Fp"$(OBJS)\testprec_test.pch" \
+TEST_CXXFLAGS = /M$(__RUNTIME_LIBS_7)$(__DEBUGRUNTIME_3) /DWIN32 \
+	$(__DEBUGINFO) /Fd$(OBJS)\test.pdb $(____DEBUGRUNTIME_2_p) \
+	$(__OPTIMIZEFLAG) $(__NO_VC_CRTDBG_p) /D__WXMSW__ $(__WXUNIV_DEFINE_p) \
+	$(__DEBUG_DEFINE_p) $(__UNICODE_DEFINE_p) /I.\..\include /I$(SETUPHDIR) /W4 \
+	/I. $(__DLLFLAG_p) /D_CONSOLE /DwxUSE_GUI=0 $(CPPUNIT_CFLAGS) $(__RTTIFLAG) \
+	$(__EXCEPTIONSFLAG) /Yu"testprec.h" /Fp"$(OBJS)\testprec_test.pch" \
 	$(CPPFLAGS) $(CXXFLAGS)
 TEST_OBJECTS =  \
@@ -56,13 +56,14 @@
 	$(OBJS)\test_memstream.obj \
 	$(OBJS)\test_sstream.obj \
+	$(OBJS)\test_tempfile.obj \
 	$(OBJS)\test_textstreamtest.obj \
 	$(OBJS)\test_zlibstream.obj \
 	$(OBJS)\test_uris.obj
-TEST_GUI_CXXFLAGS = /M$(__RUNTIME_LIBS)$(__DEBUGRUNTIME_17) /DWIN32 \
+TEST_GUI_CXXFLAGS = /M$(__RUNTIME_LIBS_21)$(__DEBUGRUNTIME_17) /DWIN32 \
 	$(__DEBUGINFO) /Fd$(OBJS)\test_gui.pdb $(____DEBUGRUNTIME_16_p) \
 	$(__OPTIMIZEFLAG) $(__NO_VC_CRTDBG_p) /D__WXMSW__ $(__WXUNIV_DEFINE_p) \
 	$(__DEBUG_DEFINE_p) $(__UNICODE_DEFINE_p) /I.\..\include /I$(SETUPHDIR) /W4 \
 	/I. $(__DLLFLAG_p) /I.\..\samples /DNOPCH $(CPPUNIT_CFLAGS) /D_CONSOLE \
-	$(__EXCEPTIONSFLAG) $(__EXCEPTIONSFLAG_0) /Yu"testprec.h" \
+	$(__RTTIFLAG) $(__EXCEPTIONSFLAG) /Yu"testprec.h" \
 	/Fp"$(OBJS)\testprec_test_gui.pch" $(CPPFLAGS) $(CXXFLAGS)
 TEST_GUI_OBJECTS =  \
@@ -127,4 +128,10 @@
 __DEBUGRUNTIME_3 = $(__DEBUGINFO_2)
 !endif
+!if "$(RUNTIME_LIBS)" == "dynamic"
+__RUNTIME_LIBS_7 = D
+!endif
+!if "$(RUNTIME_LIBS)" == "static"
+__RUNTIME_LIBS_7 = $(__THREADSFLAG)
+!endif
 !if "$(MONOLITHIC)" == "0"
 __WXLIB_NET_p = \
@@ -173,4 +180,10 @@
 __DEBUGRUNTIME_17 = $(__DEBUGINFO_2)
 !endif
+!if "$(RUNTIME_LIBS)" == "dynamic"
+__RUNTIME_LIBS_21 = D
+!endif
+!if "$(RUNTIME_LIBS)" == "static"
+__RUNTIME_LIBS_21 = $(__THREADSFLAG)
+!endif
 !if "$(BUILD)" == "debug" && "$(DEBUG_RUNTIME_LIBS)" == "0"
 __NO_VC_CRTDBG_p_3 = /d __NO_VC_CRTDBG__
@@ -252,21 +265,21 @@
 __OPTIMIZEFLAG = /O2
 !endif
-!if "$(RUNTIME_LIBS)" == "dynamic"
-__RUNTIME_LIBS = D
+!if "$(USE_THREADS)" == "0"
+__THREADSFLAG = L
 !endif
-!if "$(RUNTIME_LIBS)" == "static"
-__RUNTIME_LIBS = T
+!if "$(USE_THREADS)" == "1"
+__THREADSFLAG = T
 !endif
-!if "$(USE_EXCEPTIONS)" == "0"
-__EXCEPTIONSFLAG = 
+!if "$(USE_RTTI)" == "0"
+__RTTIFLAG = 
 !endif
-!if "$(USE_EXCEPTIONS)" == "1"
-__EXCEPTIONSFLAG = /GR
+!if "$(USE_RTTI)" == "1"
+__RTTIFLAG = /GR
 !endif
 !if "$(USE_EXCEPTIONS)" == "0"
-__EXCEPTIONSFLAG_0 = 
+__EXCEPTIONSFLAG = 
 !endif
 !if "$(USE_EXCEPTIONS)" == "1"
-__EXCEPTIONSFLAG_0 = /GX
+__EXCEPTIONSFLAG = /GX
 !endif
 !if "$(BUILD)" == "debug" && "$(DEBUG_RUNTIME_LIBS)" == "0"
@@ -433,4 +446,7 @@
 
 $(OBJS)\test_sstream.obj: .\streams\sstream.cpp
+	$(CXX) /c /nologo /TP /Fo$@ $(TEST_CXXFLAGS) $**
+
+$(OBJS)\test_tempfile.obj: .\streams\tempfile.cpp
 	$(CXX) /c /nologo /TP /Fo$@ $(TEST_CXXFLAGS) $**
 
Index: tests/makefile.wat
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/tests/makefile.wat,v
retrieving revision 1.29
diff -u -2 -r1.29 makefile.wat
--- tests/makefile.wat	2005/02/12 21:53:47	1.29
+++ tests/makefile.wat	2005/03/13 02:27:09
@@ -1,5 +1,5 @@
 # =========================================================================
 #     This makefile was generated by
-#     Bakefile 0.1.6 (http://bakefile.sourceforge.net)
+#     Bakefile 0.1.7 (http://bakefile.sourceforge.net)
 #     Do not modify, all changes will be overwritten!
 # =========================================================================
@@ -125,4 +125,11 @@
 __OPTIMIZEFLAG = -ot -ox
 !endif
+__THREADSFLAG =
+!ifeq USE_THREADS 0
+__THREADSFLAG = 
+!endif
+!ifeq USE_THREADS 1
+__THREADSFLAG = -bm
+!endif
 __RUNTIME_LIBS =
 !ifeq RUNTIME_LIBS dynamic
@@ -132,17 +139,17 @@
 __RUNTIME_LIBS = 
 !endif
-__EXCEPTIONSFLAG =
-!ifeq USE_EXCEPTIONS 0
-__EXCEPTIONSFLAG = 
+__RTTIFLAG =
+!ifeq USE_RTTI 0
+__RTTIFLAG = 
 !endif
-!ifeq USE_EXCEPTIONS 1
-__EXCEPTIONSFLAG = -xr
+!ifeq USE_RTTI 1
+__RTTIFLAG = -xr
 !endif
-__EXCEPTIONSFLAG_0 =
+__EXCEPTIONSFLAG =
 !ifeq USE_EXCEPTIONS 0
-__EXCEPTIONSFLAG_0 = 
+__EXCEPTIONSFLAG = 
 !endif
 !ifeq USE_EXCEPTIONS 1
-__EXCEPTIONSFLAG_0 = -xs
+__EXCEPTIONSFLAG = -xs
 !endif
 __WXLIB_BASE_p =
@@ -198,10 +205,10 @@
 SETUPHDIR = &
 	$(LIBDIRNAME)\$(PORTNAME)$(WXUNIVNAME)$(WXUNICODEFLAG)$(WXDEBUGFLAG)
-TEST_CXXFLAGS = $(__DEBUGINFO) $(__OPTIMIZEFLAG) -bm $(__RUNTIME_LIBS) &
-	-d__WXMSW__ $(__WXUNIV_DEFINE_p) $(__DEBUG_DEFINE_p) $(__UNICODE_DEFINE_p) &
-	-i=.\..\include -i=$(SETUPHDIR) -wx -wcd=549 -wcd=656 -wcd=657 -wcd=667 -i=. &
-	$(__DLLFLAG_p) -dwxUSE_GUI=0 $(CPPUNIT_CFLAGS) &
-	/fh=$(OBJS)\testprec_test.pch $(__EXCEPTIONSFLAG) $(__EXCEPTIONSFLAG_0) &
-	$(CPPFLAGS) $(CXXFLAGS)
+TEST_CXXFLAGS = $(__DEBUGINFO) $(__OPTIMIZEFLAG) $(__THREADSFLAG) &
+	$(__RUNTIME_LIBS) -d__WXMSW__ $(__WXUNIV_DEFINE_p) $(__DEBUG_DEFINE_p) &
+	$(__UNICODE_DEFINE_p) -i=.\..\include -i=$(SETUPHDIR) -wx -wcd=549 -wcd=656 &
+	-wcd=657 -wcd=667 -i=. $(__DLLFLAG_p) -dwxUSE_GUI=0 $(CPPUNIT_CFLAGS) &
+	/fh=$(OBJS)\testprec_test.pch $(__RTTIFLAG) $(__EXCEPTIONSFLAG) $(CPPFLAGS) &
+	$(CXXFLAGS)
 TEST_OBJECTS =  &
 	$(OBJS)\test_dummy.obj &
@@ -234,13 +241,14 @@
 	$(OBJS)\test_memstream.obj &
 	$(OBJS)\test_sstream.obj &
+	$(OBJS)\test_tempfile.obj &
 	$(OBJS)\test_textstreamtest.obj &
 	$(OBJS)\test_zlibstream.obj &
 	$(OBJS)\test_uris.obj
-TEST_GUI_CXXFLAGS = $(__DEBUGINFO) $(__OPTIMIZEFLAG) -bm $(__RUNTIME_LIBS) &
-	-d__WXMSW__ $(__WXUNIV_DEFINE_p) $(__DEBUG_DEFINE_p) $(__UNICODE_DEFINE_p) &
-	-i=.\..\include -i=$(SETUPHDIR) -wx -wcd=549 -wcd=656 -wcd=657 -wcd=667 -i=. &
-	$(__DLLFLAG_p) -i=.\..\samples -dNOPCH $(CPPUNIT_CFLAGS) &
-	/fh=$(OBJS)\testprec_test_gui.pch $(__EXCEPTIONSFLAG) &
-	$(__EXCEPTIONSFLAG_0) $(CPPFLAGS) $(CXXFLAGS)
+TEST_GUI_CXXFLAGS = $(__DEBUGINFO) $(__OPTIMIZEFLAG) $(__THREADSFLAG) &
+	$(__RUNTIME_LIBS) -d__WXMSW__ $(__WXUNIV_DEFINE_p) $(__DEBUG_DEFINE_p) &
+	$(__UNICODE_DEFINE_p) -i=.\..\include -i=$(SETUPHDIR) -wx -wcd=549 -wcd=656 &
+	-wcd=657 -wcd=667 -i=. $(__DLLFLAG_p) -i=.\..\samples -dNOPCH &
+	$(CPPUNIT_CFLAGS) /fh=$(OBJS)\testprec_test_gui.pch $(__RTTIFLAG) &
+	$(__EXCEPTIONSFLAG) $(CPPFLAGS) $(CXXFLAGS)
 TEST_GUI_OBJECTS =  &
 	$(OBJS)\test_gui_dummy.obj &
@@ -381,4 +389,7 @@
 
 $(OBJS)\test_sstream.obj :  .AUTODEPEND .\streams\sstream.cpp
+	$(CXX) -zq -fo=$^@ $(TEST_CXXFLAGS) $<
+
+$(OBJS)\test_tempfile.obj :  .AUTODEPEND .\streams\tempfile.cpp
 	$(CXX) -zq -fo=$^@ $(TEST_CXXFLAGS) $<
 
Index: tests/test.bkl
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/tests/test.bkl,v
retrieving revision 1.35
diff -u -2 -r1.35 test.bkl
--- tests/test.bkl	2005/02/12 21:53:47	1.35
+++ tests/test.bkl	2005/03/13 02:27:10
@@ -52,4 +52,5 @@
             streams/memstream.cpp
             streams/sstream.cpp
+            streams/tempfile.cpp
             streams/textstreamtest.cpp
             streams/zlibstream.cpp
Index: tests/test_test.dsp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/tests/test_test.dsp,v
retrieving revision 1.5
diff -u -2 -r1.5 test_test.dsp
--- tests/test_test.dsp	2005/02/12 21:53:47	1.5
+++ tests/test_test.dsp	2005/03/13 02:27:12
@@ -538,4 +538,8 @@
 # Begin Source File
 
+SOURCE=.\streams\tempfile.cpp
+# End Source File
+# Begin Source File
+
 SOURCE=.\test.cpp
 # End Source File
Index: tests/streams/bstream.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/tests/streams/bstream.cpp,v
retrieving revision 1.6
diff -u -2 -r1.6 bstream.cpp
--- tests/streams/bstream.cpp	2005/02/04 09:43:44	1.6
+++ tests/streams/bstream.cpp	2005/03/13 02:27:13
@@ -51,4 +51,5 @@
     STREAM_REGISTER_SUB_SUITE(fileStream);
     STREAM_REGISTER_SUB_SUITE(ffileStream);
+    STREAM_REGISTER_SUB_SUITE(tempStream);
     STREAM_REGISTER_SUB_SUITE(zlibStream);
 
--- /dev/null	2004-12-12 21:49:49.000000000 +0000
+++ docs/latex/wx/tempfilestrm.tex	2005-03-13 02:21:51.690525232 +0000
@@ -0,0 +1,55 @@
+%
+% automatically generated by HelpGen $Revision: 1.35 $ from
+% wx/wfstream.h at 07/Mar/05 20:45:33
+%
+
+\section{\class{wxTempFileOutputStream}}\label{wxtempfileoutputstream}
+
+wxTempFileOutputStream is an output stream based on \helpref{wxTempFile}{wxtempfile}. It
+provides a relatively safe way to replace the contents of the
+existing file.
+
+\wxheading{Derived from}
+
+\helpref{wxOutputStream}{wxoutputstream}
+
+\wxheading{Include files}
+
+<wx/wfstream.h>
+
+\wxheading{See also}
+
+\helpref{wxTempFile}{wxtempfile}
+
+\latexignore{\rtfignore{\wxheading{Members}}}
+
+
+\membersection{wxTempFileOutputStream::wxTempFileOutputStream}\label{wxtempfileoutputstreamwxtempfileoutputstream}
+
+\func{}{wxTempFileOutputStream}{\param{const wxString\& }{fileName}}
+
+Associates wxTempFileOutputStream with the file to be replaced and opens it. You should use 
+\helpref{IsOk}{wxstreambaseisok} to verify if the constructor succeeded.
+
+Call \helpref{Commit()}{wxtempfileoutputstreamcommit} or \helpref{Close()}{wxoutputstreamclose} to
+replace the old file and close this one. Calling \helpref{Discard()}{wxtempfileoutputstreamdiscard} 
+(or allowing the destructor to do it) will discard the changes.
+
+
+\membersection{wxTempFileOutputStream::Commit}\label{wxtempfileoutputstreamcommit}
+
+\func{bool}{Commit}{\void}
+
+Validate changes: deletes the old file of the given name and renames the new
+file to the old name. Returns {\tt true} if both actions succeeded. If {\tt false} is
+returned it may unfortunately mean two quite different things: either that
+either the old file couldn't be deleted or that the new file couldn't be renamed
+to the old name.
+
+
+\membersection{wxTempFileOutputStream::Discard}\label{wxtempfileoutputstreamdiscard}
+
+\func{void}{Discard}{\void}
+
+Discard changes: the old file contents are not changed, the temporary file is
+deleted.
--- /dev/null	2004-12-12 21:49:49.000000000 +0000
+++ tests/streams/tempfile.cpp	2005-03-13 02:04:58.947485576 +0000
@@ -0,0 +1,107 @@
+///////////////////////////////////////////////////////////////////////////////
+// Name:        tests/streams/tempfile.cpp
+// Purpose:     Test wxTempFileOutputStream
+// Author:      Mike Wetherell
+// RCS-ID:      $Id$
+// Copyright:   (c) 2005 Mike Wetherell
+// Licence:     wxWidgets licence
+///////////////////////////////////////////////////////////////////////////////
+
+#include "testprec.h"
+
+#ifdef __BORLANDC__
+    #pragma hdrstop
+#endif
+
+// for all others, include the necessary headers
+#ifndef WX_PRECOMP
+    #include "wx/wx.h"
+#endif
+
+#include "wx/wfstream.h"
+#include "wx/filename.h"
+#include "bstream.h"
+
+#if wxUSE_STREAMS && wxUSE_FILE
+
+
+///////////////////////////////////////////////////////////////////////////////
+// Self deleting test file
+
+class TestFile
+{
+public:
+    TestFile();
+    ~TestFile() { if (wxFileExists(m_name)) wxRemoveFile(m_name); }
+    wxString GetName() const { return m_name; }
+private:
+    wxString m_name;
+};
+
+// Initialise with a test pattern so we can see if the file is replaced
+//
+TestFile::TestFile()
+{
+    wxFile file;
+    m_name = wxFileName::CreateTempFileName(_T("wxtest"), &file);
+    file.Write("Before", 6);
+}
+
+
+///////////////////////////////////////////////////////////////////////////////
+// The test case
+
+class tempStream : public CppUnit::TestCase
+{
+    CPPUNIT_TEST_SUITE(tempStream);
+        CPPUNIT_TEST(DoNothing);
+        CPPUNIT_TEST(Close);
+        CPPUNIT_TEST(Commit);
+        CPPUNIT_TEST(Discard);
+    CPPUNIT_TEST_SUITE_END();
+
+    void DoNothing() { DoTest(DONOTHING, false); }
+    void Close()     { DoTest(CLOSE, true); }
+    void Commit()    { DoTest(COMMIT, true); }
+    void Discard()   { DoTest(DISCARD, false); }
+
+    enum Action { DONOTHING, CLOSE, COMMIT, DISCARD };
+    void DoTest(Action action, bool shouldHaveCommited);
+};
+
+// the common test code
+//
+void tempStream::DoTest(Action action, bool shouldHaveCommited)
+{
+    TestFile temp;
+
+    {
+        wxTempFileOutputStream out(temp.GetName());
+        out.Write("Affer", 5);
+        CPPUNIT_ASSERT(out.SeekO(2) == 2);
+        out.Write("t", 1);
+        CPPUNIT_ASSERT(out.IsSeekable());
+        CPPUNIT_ASSERT(out.GetLength() == 5);
+        CPPUNIT_ASSERT(out.TellO() == 3);
+
+        switch (action) {
+            case DONOTHING: break;
+            case COMMIT:    out.Commit(); break;
+            case DISCARD:   out.Discard(); break;
+            case CLOSE:     out.Close();
+        }
+    }
+
+    wxFileInputStream in(temp.GetName());
+    char buf[32];
+    in.Read(buf, sizeof(buf));
+    buf[in.LastRead()] = 0;
+    CPPUNIT_ASSERT(strcmp(buf, shouldHaveCommited ? "After" : "Before") == 0);
+}
+
+
+// Register the stream sub suite, by using some stream helper macro.
+// Note: Don't forget to connect it to the base suite (See: bstream.cpp => StreamCase::suite())
+STREAM_TEST_SUBSUITE_NAMED_REGISTRATION(tempStream)
+
+#endif // wxUSE_STREAMS && wxUSE_FILE

 	  	 
