Index: wxWidgets-2.8/include/wx/app.h
===================================================================
--- wxWidgets-2.8/include/wx/app.h	(revision 55078)
+++ wxWidgets-2.8/include/wx/app.h	(working copy)
@@ -505,11 +505,11 @@
     // perform standard OnIdle behaviour, ensure that this is always called
     void OnIdle(wxIdleEvent& event);
 
-
-protected:
     // delete all objects in wxPendingDelete list
     void DeletePendingObjects();
 
+
+protected:
     // override base class method to use GUI traits
     virtual wxAppTraits *CreateTraits();
