Index: interface/wx/dataobj.h
===================================================================
--- interface/wx/dataobj.h	(revision 70852)
+++ interface/wx/dataobj.h	(working copy)
@@ -506,7 +506,7 @@
         @since 2.9.1
     */
     wxDataObjectSimple *GetObject(const wxDataFormat& format,
-                                  wxDataObjectBase::Direction dir = Get) const;
+                                  wxDataObject::Direction dir = Get) const;
 };
 
 
@@ -747,7 +747,7 @@
         Returns 1 under other platforms (e.g. wxMSW) or when building in ANSI mode
         (@c wxUSE_UNICODE==0).
     */
-    virtual size_t GetFormatCount(Direction dir = Get);
+    virtual size_t GetFormatCount(wxDataObject::Direction dir = Get);
 
     /**
         Returns the preferred format supported by this object.
@@ -764,7 +764,7 @@
         under other ports returns only one of the two, depending on the build mode.
     */
     virtual void GetAllFormats(wxDataFormat* formats,
-                               Direction dir = Get) const = 0;
+                               wxDataObject::Direction dir = Get) const = 0;
 
     /**
         Sets the text associated with the data object. This method is called
