--- G:\cyberboard\cb4\wxWidgets\src\msw\statusbar - Copy.cpp	Wed Mar 18 12:02:28 2009
+++ G:\cyberboard\cb4\wxWidgets\src\msw\statusbar.cpp	Mon Mar 23 23:36:23 2009
@@ -131,6 +131,17 @@
     m_pDC = new wxClientDC(this);
 
     InheritAttributes();
+    if (!m_hasFont) {
+        HFONT hFont = HFONT(SendMessage(GetHwnd(), WM_GETFONT, 0, 0));
+        LOGFONT lf;
+        BOOL b = GetObject(hFont, sizeof(lf), &lf) == sizeof(lf) &&
+                    SetFont(wxFont(wxCreateFontFromLogFont(&lf)));
+#ifdef __WXDEBUG__
+        wxASSERT(b);
+#else
+        wxUnusedVar(b);
+#endif
+    }
 
     SetBackgroundColour(wxSystemSettings::GetColour(wxSYS_COLOUR_MENUBAR));
 
