Index: listbox.cpp
===================================================================
--- listbox.cpp	(revision 63484)
+++ listbox.cpp	(working copy)
@@ -287,8 +287,11 @@
                  wxT("invalid index in wxListBox::Delete") );
 
 #if wxUSE_OWNER_DRAWN
-    delete m_aItems[n];
-    m_aItems.RemoveAt(n);
+	if ( HasFlag(wxLB_OWNERDRAW) )
+	{
+		delete m_aItems[n];
+		m_aItems.RemoveAt(n);
+	}
 #endif // wxUSE_OWNER_DRAWN
 
     SendMessage(GetHwnd(), LB_DELETESTRING, n, 0);
@@ -694,15 +697,18 @@
     wxListBoxBase::Delete(n);
 
     // free memory
-    delete m_aItems[n];
-    m_aItems.RemoveAt(n);
+//    delete m_aItems[n];
+//    m_aItems.RemoveAt(n);
 }
 
 bool wxListBox::SetFont(const wxFont &font)
 {
-    unsigned int i;
-    for ( i = 0; i < m_aItems.GetCount(); i++ )
-        m_aItems[i]->SetFont(font);
+	if ( HasFlag(wxLB_OWNERDRAW) )
+	{
+		unsigned int i;
+		for ( i = 0; i < m_aItems.GetCount(); i++ )
+			m_aItems[i]->SetFont(font);
+	}
 
     wxListBoxBase::SetFont(font);
 
@@ -790,7 +796,7 @@
 bool wxListBox::MSWOnDraw(WXDRAWITEMSTRUCT *item)
 {
     // only owner-drawn control should receive this message
-    wxCHECK( ((m_windowStyle & wxLB_OWNERDRAW) == wxLB_OWNERDRAW), false );
+    wxCHECK( HasFlag(wxLB_OWNERDRAW), false );
 
     DRAWITEMSTRUCT *pStruct = (DRAWITEMSTRUCT *)item;
 
     if ( pStruct->itemID == (UINT)-1 )
         return false;
 
     wxListBoxItem *pItem = (wxListBoxItem *)m_aItems[pStruct->itemID];

     wxDCTemp dc((WXHDC)pStruct->hDC);

     return pItem->OnDrawItem(dc, wxRectFromRECT(pStruct->rcItem),
                             (wxOwnerDrawn::wxODAction)pStruct->itemAction,
                             (wxOwnerDrawn::wxODStatus)(pStruct->itemState | wxOwnerDrawn::wxODHidePrefix));
 }
 
 #endif // wxUSE_OWNER_DRAWN
