Index: interface/wx/filedlg.h
===================================================================
--- interface/wx/filedlg.h	(revision 58270)
+++ interface/wx/filedlg.h	(working copy)
@@ -11,8 +11,6 @@
 
     This class represents the file chooser dialog.
 
-    It pops up a file selector box (native for Windows and GTK2.4+).
-
     The path and filename are distinct elements of a full file pathname.
     If path is wxEmptyString, the current directory will be used.
     If filename is wxEmptyString, no default filename will be supplied.
@@ -55,7 +53,7 @@
            file(s) chosen by the user are.
     @style{wxFD_PREVIEW}
            Show the preview of the selected files (currently only supported by
-           wxGTK using GTK+ 2.4 or later).
+           wxGTK).
     @endStyleTable
 
     @library{wxcore}
@@ -113,6 +111,8 @@
     /**
         If functions SetExtraControlCreator() and ShowModal() were called,
         returns the extra window. Otherwise returns @NULL.
+
+        @since 2.9.0
     */
     wxWindow* GetExtraControl() const;
 
@@ -174,6 +174,13 @@
     virtual void SetDirectory(const wxString& directory);
 
     /**
+        The type of function used as an argument for SetExtraControlCreator().
+
+        @since 2.9.0
+    */
+    typedef wxWindow *(*ExtraControlCreatorFunction)(wxWindow*);
+
+    /**
         Customize file dialog by adding extra window, which is typically placed
         below the list of files and above the buttons.
 
@@ -182,9 +189,11 @@
         The @c creator function should take pointer to parent window (file dialog)
         and should return a window allocated with operator new.
 
-        Supported platforms: wxGTK, wxUniv.
+        Supported platforms: wxGTK, wxMSW, wxUniv.
+
+        @since 2.9.0
     */
-    bool SetExtraControlCreator(ExtraControlCreatorFunction);
+    bool SetExtraControlCreator(ExtraControlCreatorFunction creator);
 
     /**
         Sets the default filename.
Index: include/wx/filedlg.h
===================================================================
--- include/wx/filedlg.h	(revision 58270)
+++ include/wx/filedlg.h	(working copy)
@@ -122,7 +122,7 @@
 
     virtual bool SupportsExtraControl() const { return false; }
 
-    bool SetExtraControlCreator(ExtraControlCreatorFunction WXUNUSED(c));
+    bool SetExtraControlCreator(ExtraControlCreatorFunction creator);
     wxWindow *GetExtraControl() const { return m_extraControl; }
 
     // Utility functions
Index: include/wx/msw/filedlg.h
===================================================================
--- include/wx/msw/filedlg.h	(revision 58270)
+++ include/wx/msw/filedlg.h	(working copy)
@@ -34,8 +34,8 @@
     virtual void GetFilenames(wxArrayString& files) const;
 #ifndef __WXWINCE__
     virtual bool SupportsExtraControl() const { return true; }
+    void MSWOnInitDialogHook(WXHWND hwnd);
 #endif // __WXWINCE__
-    void MSWOnInitDialogHook(WXHWND hwnd);
 
     virtual int ShowModal();
 
Index: src/common/fldlgcmn.cpp
===================================================================
--- src/common/fldlgcmn.cpp	(revision 58270)
+++ src/common/fldlgcmn.cpp	(working copy)
@@ -151,12 +151,12 @@
     return filePath + ext;
 }
 
-bool wxFileDialogBase::SetExtraControlCreator(ExtraControlCreatorFunction c)
+bool wxFileDialogBase::SetExtraControlCreator(ExtraControlCreatorFunction creator)
 {
     wxCHECK_MSG( !m_extraControlCreator, false,
                  "wxFileDialog::SetExtraControl() called second time" );
 
-    m_extraControlCreator = c;
+    m_extraControlCreator = creator;
     return SupportsExtraControl();
 }
 
Index: src/msw/filedlg.cpp
===================================================================
--- src/msw/filedlg.cpp	(revision 58270)
+++ src/msw/filedlg.cpp	(working copy)
@@ -86,6 +86,7 @@
 {
     switch ( iMsg )
     {
+#ifndef __WXWINCE__
         case WM_INITDIALOG:
             {
                 OPENFILENAME* ofn = reinterpret_cast<OPENFILENAME *>(lParam);
@@ -93,6 +94,7 @@
                     ->MSWOnInitDialogHook((WXHWND)hDlg);
             }
             break;
+#endif // __WXWINCE__
 
         case WM_NOTIFY:
             {
@@ -368,6 +370,7 @@
     return true;
 }
 
+#ifndef __WXWINCE__
 void wxFileDialog::MSWOnInitDialogHook(WXHWND hwnd)
 {
    SetHWND(hwnd);
@@ -376,6 +379,7 @@
 
    SetHWND(NULL);
 }
+#endif // __WXWINCE__
 
 int wxFileDialog::ShowModal()
 {
