diff -ruX wx-exclude wxWidgets-20060411/samples/image/image.cpp wxWidgets/samples/image/image.cpp
--- wxWidgets-20060411/samples/image/image.cpp	2006-01-29 16:44:21.000000000 +0100
+++ wxWidgets/samples/image/image.cpp	2006-04-13 14:59:16.531250000 +0200
@@ -86,6 +86,12 @@
 
     wxBitmap  *colorized_horse_jpeg;
 
+    wxBitmap my_toucan;
+    wxBitmap my_toucan_flipped_horiz;
+    wxBitmap my_toucan_flipped_vert;
+    wxBitmap my_toucan_flipped_both;
+    wxBitmap my_toucan_head;
+
     int xH, yH ;
     int m_ani_images ;
 
@@ -305,7 +311,7 @@
         // setting alpha in the loop below then)
         typedef wxAlphaPixelData Data;
         // typedef wxNativePixelData Data;
-        
+
         // First, clear the whole bitmap by making it alpha
         {
             Data data( m_bitmap, wxPoint(0,0), wxSize(SIZE, SIZE) );
@@ -487,6 +493,14 @@
         wxLogError(wxT("Can't load PNG image"));
     else
         my_horse_png = new wxBitmap( image );
+
+    image = wxImage(wxT("../toucan.png"));
+    my_toucan = wxBitmap(image);
+    my_toucan_flipped_horiz = wxBitmap(image.Mirror(true));
+    my_toucan_flipped_vert = wxBitmap(image.Mirror(false));
+    my_toucan_flipped_both = wxBitmap(image.Mirror(true).Mirror(false));
+    my_toucan_head = wxBitmap(image.GetSubImage(wxRect(40, 7, 80, 60)));
+
 #endif // wxUSE_LIBPNG
 
 #if wxUSE_LIBJPEG
@@ -752,6 +766,28 @@
     if (my_horse_xpm && my_horse_xpm->Ok())
         dc.DrawBitmap( *my_horse_xpm, 30, 1760 );
 
+    {
+        wxColour fg = dc.GetTextForeground();
+        dc.SetTextForeground(wxT("yellow"));
+        int x = 100, y = 150, yy = 170;;
+
+        dc.DrawText(wxT("original toucan"), x, y);
+        dc.DrawBitmap(my_toucan, x, y+15);
+        y += yy;
+        dc.DrawText(wxT("flipped horizontally"), x, y);
+        dc.DrawBitmap(my_toucan_flipped_horiz, x, y+15);
+        y += yy;
+        dc.DrawText(wxT("flipped vertically"), x, y);
+        dc.DrawBitmap(my_toucan_flipped_vert, x, y+15);
+        y += yy;
+        dc.DrawText(wxT("flipped both h&v"), x, y);
+        dc.DrawBitmap(my_toucan_flipped_both, x, y+15);
+
+        dc.DrawText(wxT("toucan's head"), 285, 220);
+        dc.DrawBitmap(my_toucan_head, 290, 235);
+
+        dc.SetTextForeground(fg);
+    }
 
     if (my_smile_xbm && my_smile_xbm->Ok())
     {

 	  	 
