Index: src/msw/listctrl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/msw/listctrl.cpp,v
retrieving revision 1.238
diff -b -u -2 -r1.238 listctrl.cpp
--- src/msw/listctrl.cpp	2005/10/05 15:42:16	1.238
+++ src/msw/listctrl.cpp	2005/11/10 04:27:28
@@ -1215,5 +1215,5 @@
         m_ownsImageListState = false;
     }
-    ListView_SetImageList(GetHwnd(), (HIMAGELIST) imageList ? imageList->GetHIMAGELIST() : 0, flags);
+    (void) ListView_SetImageList(GetHwnd(), (HIMAGELIST) imageList ? imageList->GetHIMAGELIST() : 0, flags);
 }
 
Index: src/msw/notebook.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/msw/notebook.cpp,v
retrieving revision 1.164
diff -b -u -2 -r1.164 notebook.cpp
--- src/msw/notebook.cpp	2005/11/09 20:14:32	1.164
+++ src/msw/notebook.cpp	2005/11/11 03:44:12
@@ -498,5 +498,5 @@
   if ( imageList )
   {
-    TabCtrl_SetImageList(GetHwnd(), (HIMAGELIST)imageList->GetHIMAGELIST());
+    (void) TabCtrl_SetImageList(GetHwnd(), (HIMAGELIST)imageList->GetHIMAGELIST());
   }
 }
Index: src/msw/tabctrl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/msw/tabctrl.cpp,v
retrieving revision 1.39
diff -b -u -2 -r1.39 tabctrl.cpp
--- src/msw/tabctrl.cpp	2005/09/23 12:55:12	1.39
+++ src/msw/tabctrl.cpp	2005/11/10 04:28:06
@@ -316,5 +316,5 @@
 {
     m_imageList = imageList;
-    TabCtrl_SetImageList( (HWND) GetHWND(), (HIMAGELIST) imageList->GetHIMAGELIST() );
+    (void) TabCtrl_SetImageList( (HWND) GetHWND(), (HIMAGELIST) imageList->GetHIMAGELIST() );
 }
 
Index: src/msw/treectrl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/msw/treectrl.cpp,v
retrieving revision 1.219
diff -b -u -2 -r1.219 treectrl.cpp
--- src/msw/treectrl.cpp	2005/10/28 14:05:45	1.219
+++ src/msw/treectrl.cpp	2005/11/11 03:40:16
@@ -831,5 +834,5 @@
 {
     // no error return
-    TreeView_SetImageList(GetHwnd(),
+    (void) TreeView_SetImageList(GetHwnd(),
                           imageList ? imageList->GetHIMAGELIST() : 0,
                           which);
@@ -2016,5 +2019,5 @@
     hitTestInfo.pt.y = (int)point.y;
 
-    TreeView_HitTest(GetHwnd(), &hitTestInfo);
+    (void) TreeView_HitTest(GetHwnd(), &hitTestInfo);
 
     flags = 0;
@@ -2211,5 +2214,5 @@
         tvht.pt.y = y;
     
-        TreeView_HitTest(GetHwnd(), &tvht);
+        (void) TreeView_HitTest(GetHwnd(), &tvht);
 
         switch ( nMsg )

 	  	 
