 src/common/toplvcmn.cpp | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/src/common/toplvcmn.cpp b/src/common/toplvcmn.cpp
index 03504cc..47dd090 100644
--- a/src/common/toplvcmn.cpp
+++ b/src/common/toplvcmn.cpp
@@ -101,10 +101,17 @@ wxTopLevelWindowBase::~wxTopLevelWindowBase()
 
 bool wxTopLevelWindowBase::Destroy()
 {
-    // delayed destruction: the frame will be deleted during the next idle
+    // delayed destruction: the TLW will be deleted during the next idle
     // loop iteration
-    if ( !wxPendingDelete.Member(this) )
+    // but it can only be done if the parent is not being deleted which would
+    // lead to an invalid pointer being added to wxPendingDelete list for this
+    // TLW and finally to a crash.
+    // (or maybe calling "delete this;" is better in this case?)
+    if ( !wxPendingDelete.Member(this) &&
+         !(GetParent() && GetParent()->IsBeingDeleted()) )
+    {
         wxPendingDelete.Append(this);
+    }
 
     // normally we want to hide the window immediately so that it doesn't get
     // stuck on the screen while it's being destroyed, however we shouldn't
