 samples/minimal/minimal.cpp | 22 ++++++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/samples/minimal/minimal.cpp b/samples/minimal/minimal.cpp
index a78e462..abd1d25 100644
--- a/samples/minimal/minimal.cpp
+++ b/samples/minimal/minimal.cpp
@@ -40,6 +40,8 @@
     #include "../sample.xpm"
 #endif
 
+#include <wx/colordlg.h>
+
 // ----------------------------------------------------------------------------
 // private classes
 // ----------------------------------------------------------------------------
@@ -137,6 +139,22 @@ bool MyApp::OnInit()
     return true;
 }
 
+class MyPanel : public wxPanel
+{
+public:
+    MyPanel(wxWindow* parent) : wxPanel(parent)
+    {
+        wxWindow* t = wxGetTopLevelParent(this);
+        m_clrDlg = new wxColourDialog(t);
+    }
+
+    ~MyPanel()
+    { m_clrDlg->Destroy(); }
+
+private:
+    wxColourDialog* m_clrDlg;
+};
+
 // ----------------------------------------------------------------------------
 // main frame
 // ----------------------------------------------------------------------------
@@ -172,6 +190,10 @@ MyFrame::MyFrame(const wxString& title)
     CreateStatusBar(2);
     SetStatusText("Welcome to wxWidgets!");
 #endif // wxUSE_STATUSBAR
+
+    // A wxColourDialog must be explicitly deleted, otherwise it leaks memory.
+    wxPanel* p1 = new wxPanel(this);
+    new MyPanel(p1);
 }
 
 
