Index: src/msw/msgdlg.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/msgdlg.cpp,v
retrieving revision 1.11
diff -b -u -2 -r1.11 msgdlg.cpp
--- src/msw/msgdlg.cpp	2003/03/17 11:21:54	1.11
+++ src/msw/msgdlg.cpp	2003/06/21 12:33:29
@@ -43,4 +43,11 @@
     m_dialogStyle = style;
     m_parent = parent;
+#ifdef __WXDEBUG__
+    // check for common programming errors
+    if ((style&wxID_OK)==wxID_OK) {
+        // programmer probably confused wxID_OK with wxOK. Correct is wxOK.
+        wxFAIL_MSG("Invalid value for parameter 'style'. Likely cause: you passed wxID_OK rather than wxOK to wxMessageBox");
+    }
+#endif
 }
 

 	  	 
