diff --git a/include/wx/geometry.h b/include/wx/geometry.h
index 8e41705..7a87af7 100644
--- a/include/wx/geometry.h
+++ b/include/wx/geometry.h
@@ -796,24 +796,6 @@ public :
     virtual wxRect2DInt        InverseTransform( const wxRect2DInt &r ) const ;
 };
 
-inline void    wxTransform2D::Transform( wxRect2DInt* r ) const
-    { wxPoint2DInt a = r->GetLeftTop() , b = r->GetRightBottom(); Transform( &a ); Transform( &b ); *r = wxRect2DInt( a , b ); }
-
-inline wxPoint2DInt    wxTransform2D::Transform( const wxPoint2DInt &pt ) const
-    { wxPoint2DInt res = pt; Transform( &res ); return res; }
-
-inline wxRect2DInt     wxTransform2D::Transform( const wxRect2DInt &r ) const
-    { wxRect2DInt res = r; Transform( &res ); return res; }
-
-inline void    wxTransform2D::InverseTransform( wxRect2DInt* r ) const
-    { wxPoint2DInt a = r->GetLeftTop() , b = r->GetRightBottom(); InverseTransform( &a ); InverseTransform( &b ); *r = wxRect2DInt( a , b ); }
-
-inline wxPoint2DInt    wxTransform2D::InverseTransform( const wxPoint2DInt &pt ) const
-    { wxPoint2DInt res = pt; InverseTransform( &res ); return res; }
-
-inline wxRect2DInt     wxTransform2D::InverseTransform( const wxRect2DInt &r ) const
-    { wxRect2DInt res = r; InverseTransform( &res ); return res; }
-
 
 #endif // wxUSE_GEOMETRY
 
diff --git a/src/common/geometry.cpp b/src/common/geometry.cpp
index d979e79..65dabf7 100644
--- a/src/common/geometry.cpp
+++ b/src/common/geometry.cpp
@@ -361,4 +361,48 @@ void wxRect2DInt::ReadFrom( wxDataInputStream &stream )
 }
 #endif // wxUSE_STREAMS
 
+void wxTransform2D::Transform( wxRect2DInt* r ) const
+{
+    wxPoint2DInt a = r->GetLeftTop() , b = r->GetRightBottom();
+    Transform( &a );
+    Transform( &b );
+    *r = wxRect2DInt( a , b );
+}
+
+wxPoint2DInt wxTransform2D::Transform( const wxPoint2DInt &pt ) const
+{
+    wxPoint2DInt res = pt;
+    Transform( &res );
+    return res;
+}
+
+wxRect2DInt wxTransform2D::Transform( const wxRect2DInt &r ) const
+{
+    wxRect2DInt res = r;
+    Transform( &res );
+    return res;
+}
+
+void wxTransform2D::InverseTransform( wxRect2DInt* r ) const
+{
+    wxPoint2DInt a = r->GetLeftTop() , b = r->GetRightBottom();
+    InverseTransform( &a );
+    InverseTransform( &b );
+    *r = wxRect2DInt( a , b );
+}
+
+wxPoint2DInt wxTransform2D::InverseTransform( const wxPoint2DInt &pt ) const
+{
+    wxPoint2DInt res = pt;
+    InverseTransform( &res );
+    return res;
+}
+
+wxRect2DInt wxTransform2D::InverseTransform( const wxRect2DInt &r ) const
+{
+    wxRect2DInt res = r;
+    InverseTransform( &res );
+    return res;
+}
+
 #endif // wxUSE_GEOMETRY
