diff -Naur aui/auidemo.cpp aui-modified/auidemo.cpp
--- aui/auidemo.cpp	2018-02-18 22:30:31.000000000 +0200
+++ aui-modified/auidemo.cpp	2018-03-01 20:10:53.664653162 +0200
@@ -33,6 +33,7 @@
 #include "wx/statusbr.h"
 #include "wx/msgdlg.h"
 #include "wx/textdlg.h"
+#include "wx/progdlg.h"
 
 #include "wx/aui/aui.h"
 #include "../sample.xpm"
@@ -165,6 +166,8 @@
 
     void OnPaneClose(wxAuiManagerEvent& evt);
 
+    void TabsChanged(wxAuiNotebookEvent& event);
+
 private:
 
     wxAuiManager m_mgr;
@@ -1690,9 +1693,31 @@
                 wxDefaultPosition, wxDefaultSize, wxTE_MULTILINE|wxNO_BORDER) , wxT("wxTextCtrl 8") );
 
    ctrl->Thaw();
+
+   ctrl->Connect(wxEVT_AUINOTEBOOK_PAGE_CHANGED, 
+            wxAuiNotebookEventHandler(MyFrame::TabsChanged),
+            NULL,
+            this);
+
    return ctrl;
 }
 
+void MyFrame::TabsChanged(wxAuiNotebookEvent& event)
+{
+    wxProgressDialog progressDialog(_T("Please Wait"),
+                    _T("Loading Test Result List"),
+                    100,    // range
+                    this,   // parent
+                    wxPD_CAN_ABORT | wxPD_APP_MODAL | wxPD_AUTO_HIDE | wxPD_SMOOTH
+                    );
+
+    for (int i = 0; i < 5; i++)
+    {
+        progressDialog.Pulse();
+        usleep(1000 * 100);
+    }
+}
+
 wxString MyFrame::GetIntroText()
 {
     const char* text =
