Index: src/unix/gsocket.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/unix/gsocket.cpp,v
retrieving revision 1.45
diff -u -r1.45 gsocket.cpp
--- src/unix/gsocket.cpp        2005/07/26 15:38:55     1.45
+++ src/unix/gsocket.cpp        2005/08/31 02:01:15
@@ -335,6 +335,8 @@

   assert(this);

+  gs_gui_functions->Disable_Events(this);
+
   /* If socket has been created, shutdown it */
   if (m_fd != INVALID_SOCKET)
   {


 	  	 
