--- src/stc/scintilla/src/LexVHDL.cxx	2012-07-08 05:28:27.000000000 -0500
+++ src/stc/scintilla/src/LexVHDL.new.cxx	2013-06-22 09:55:41.953214131 -0500
@@ -67,7 +67,6 @@
   WordList &Packages   = *keywordlists[4];
   WordList &Types      = *keywordlists[5];
   WordList &User       = *keywordlists[6];
-
   StyleContext sc(startPos, length, initStyle, styler);
 
   for (; sc.More(); sc.Forward())
@@ -101,7 +100,7 @@
         }
         sc.SetState(SCE_VHDL_DEFAULT);
       }
-    } else if (sc.state == SCE_VHDL_COMMENT || sc.state == SCE_V_COMMENTLINEBANG) {
+    } else if (sc.state == SCE_VHDL_COMMENT || sc.state == SCE_VHDL_COMMENTLINEBANG) {
       if (sc.atLineEnd) {
         sc.SetState(SCE_VHDL_DEFAULT);
       }
@@ -124,14 +123,12 @@
         sc.SetState(SCE_VHDL_NUMBER);
       } else if (IsAWordStart(sc.ch)) {
         sc.SetState(SCE_VHDL_IDENTIFIER);
-      } else if (sc.Match('-', '-')) {
-        sc.SetState(SCE_VHDL_COMMENT);
-        sc.Forward(); 
-      } else if (sc.Match('-', '-')) {
-        if (sc.Match("--!"))  // Nice to have a different comment style
-          sc.SetState(SCE_VHDL_COMMENTLINEBANG);
-        else
-          sc.SetState(SCE_VHDL_COMMENT);
+      } 
+    else if (sc.Match('-', '-')) {
+      if (sc.Match("--!"))	// Nice to have a different comment style
+        sc.SetState(SCE_VHDL_COMMENTLINEBANG);
+      else
+        sc.SetState(SCE_VHDL_COMMENT);	
       } else if (sc.ch == '\"') {
         sc.SetState(SCE_VHDL_STRING);
       } else if (isoperator(static_cast<char>(sc.ch))) {
