Index: contrib/include/wx/animate/animate.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/contrib/include/wx/animate/animate.h,v
retrieving revision 1.6
diff -u -r1.6 animate.h
--- contrib/include/wx/animate/animate.h	2005/09/23 12:46:49	1.6
+++ contrib/include/wx/animate/animate.h	2005/12/28 19:44:34
@@ -20,6 +20,7 @@
 #include <wx/bitmap.h>
 #include <wx/colour.h>
 #include <wx/control.h>
+#include <wx/stream.h>
 
 #ifdef WXMAKINGDLL_ANIMATE
     #define WXDLLIMPEXP_ANIMATE WXEXPORT
@@ -210,7 +211,7 @@
 
 //// Operations
 
-    virtual bool LoadFile(const wxString& WXUNUSED(filename)) { return FALSE; }
+    virtual bool LoadFile(wxInputStream& WXUNUSED(stream)) { return FALSE; }
 };
 
 /* wxGIFAnimation
@@ -243,7 +244,7 @@
 
 //// Operations
 
-    virtual bool LoadFile(const wxString& filename);
+    virtual bool LoadFile(wxInputStream& stream);
 
 protected:
 
@@ -281,7 +282,8 @@
             const wxString& name = wxT("animationControl"));
 
     //// Operations
-    virtual bool LoadFile(const wxString& filename = wxEmptyString);
+    bool LoadFile(const wxString& filename = wxEmptyString);
+    bool LoadFile(wxInputStream& stream);
     virtual bool Play(bool looped = TRUE) ;
     virtual void Stop() { m_animationPlayer.Stop(); }
     virtual void FitToAnimation();
@@ -302,6 +304,8 @@
 
     // Override this in derived classes
     virtual wxAnimationBase* DoCreateAnimation(const wxString& filename) = 0;
+
+    virtual bool DoLoadFile(wxInputStream& stream);
 
     wxAnimationPlayer   m_animationPlayer;
     wxAnimationBase*    m_animation;
Index: contrib/src/animate/animate.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/contrib/src/animate/animate.cpp,v
retrieving revision 1.13
diff -u -r1.13 animate.cpp
--- contrib/src/animate/animate.cpp	2005/12/30 18:40:20	1.13
+++ contrib/src/animate/animate.cpp	2005/12/31 00:19:38
@@ -494,11 +494,8 @@
     return ((m_decoder != NULL) && (m_decoder->IsAnimation()));
 }
 
-bool wxGIFAnimation::LoadFile(const wxString& filename)
+bool wxGIFAnimation::LoadFile(wxInputStream& stream)
 {
-    if (!wxFileExists(filename))
-        return false;
-
     bool result = true;
 
     if (m_decoder)
@@ -508,9 +505,7 @@
     }
 
     {
-        wxFileInputStream stream(filename);
-
-        if (stream.GetLength() > 0)
+         if (stream.GetLength() > 0)
             m_decoder = new wxGIFDecoder(&stream, true);
 
         result = ((m_decoder != NULL) && (m_decoder->ReadGIF() == wxGIF_OK));
@@ -579,9 +574,6 @@
 
 bool wxAnimationCtrlBase::LoadFile(const wxString& filename)
 {
-    if (m_animationPlayer.IsPlaying())
-        m_animationPlayer.Stop();
-
     wxString filename1(filename);
 
     if (filename1.IsEmpty())
@@ -591,17 +583,36 @@
             return false;
     }
 
+    if (wxFileExists(filename1))
+    {
+        wxFileInputStream stream(filename1);
+        return DoLoadFile(stream);
+    }
+    else
+        return false;
+}
+
+bool wxAnimationCtrlBase::LoadFile(wxInputStream& stream)
+{
+    return DoLoadFile(stream);
+}
+
+bool wxAnimationCtrlBase::DoLoadFile(wxInputStream& stream)
+{
+    if (m_animationPlayer.IsPlaying())
+        m_animationPlayer.Stop();
+
     if (m_animation)
     {
         delete m_animation;
         m_animation = NULL;
     }
 
-    m_animation = DoCreateAnimation(filename1);
+    m_animation = DoCreateAnimation(wxEmptyString);
     if (!m_animation)
         return false;
 
-    if (!m_animation->LoadFile(filename) || !m_animation->IsValid())
+    if (!m_animation->LoadFile(stream) || !m_animation->IsValid())
     {
         delete m_animation;
         m_animation = NULL;

 	  	 
