--- src/stc/ScintillaWX.cpp	2011-12-14 20:42:11.000000000 +0200
+++ src/stc/ScintillaWX.cpp	2012-04-11 19:37:07.469655082 +0300
@@ -969,7 +969,7 @@
     case WXK_CONTROL:           key = 0; break;
     case WXK_ALT:               key = 0; break;
     case WXK_SHIFT:             key = 0; break;
-    case WXK_MENU:              key = 0; break;
+    case WXK_MENU:              key = SCK_MENU; break;
     }
 
 #ifdef __WXMAC__
