diff --git a/samples/listctrl/listtest.cpp b/samples/listctrl/listtest.cpp
index 5cc6cf6dd4..883b7a315c 100644
--- a/samples/listctrl/listtest.cpp
+++ b/samples/listctrl/listtest.cpp
@@ -540,7 +540,7 @@ void MyFrame::InitWithReportItems()
     // items with images initially even if we specify dummy image id
     wxListItem itemCol;
     itemCol.SetText("Column 1");
-    itemCol.SetImage(-1);
+    //itemCol.SetImage(-1);
     m_listCtrl->InsertColumn(0, itemCol);
 
     itemCol.SetText("Column 2 (auto size excluding header)");
