Index: configure.in
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/configure.in,v
retrieving revision 1.876
diff -p -u -b -u -2 -r1.876 configure.in
--- configure.in	2004/10/11 02:02:22	1.876
+++ configure.in	2004/10/15 02:33:36
@@ -5734,5 +5734,5 @@ fi
 
 dnl ---------------------------------------------------------------------------
-dnl get the string with OS info - used by wxGetOsDescription()
+dnl get the string with OS info - used by wxGetOsDescription() on MacOS X
 dnl ---------------------------------------------------------------------------
 
Index: src/msw/utils.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/utils.cpp,v
retrieving revision 1.149
diff -p -u -b -u -2 -r1.149 utils.cpp
--- src/msw/utils.cpp	2004/09/27 10:58:26	1.149
+++ src/msw/utils.cpp	2004/10/15 02:33:42
@@ -938,6 +938,37 @@ wxString wxGetOsDescription()
 
             case VER_PLATFORM_WIN32_WINDOWS:
-                str.Printf(_("Windows 9%c"),
-                           info.dwMinorVersion == 0 ? _T('5') : _T('8'));
+                switch (info.dwMinorVersion)
+                {
+                    case 0:
+                        if ( info.szCSDVersion[1] == 'B' ||
+                             info.szCSDVersion[1] == 'C' )
+                        {
+                            str = _("Windows 95 OSR2");
+                        }
+                        else
+                        {
+                            str = _("Windows 95");
+                        }
+                        break;
+                    case 10:
+                        if ( info.szCSDVersion[1] == 'B' ||
+                             info.szCSDVersion[1] == 'C' )
+                        {
+                            str = _("Windows 98 SE");
+                        }
+                        else
+                        {
+                            str = _("Windows 98");
+                        }
+                        break;
+                    case 90:
+                        str = _("Windows ME");
+                        break;
+                    default:
+                        str.Printf(_("Windows 9x (%d.%d)"),
+                                   info.dwMajorVersion,
+                                   info.dwMinorVersion);
+                        break;
+                }
                 if ( !wxIsEmpty(info.szCSDVersion) )
                 {
@@ -947,8 +978,29 @@ wxString wxGetOsDescription()
 
             case VER_PLATFORM_WIN32_NT:
-                str.Printf(_T("Windows NT %lu.%lu (build %lu"),
+                if ( info.dwMajorVersion == 5 )
+                {
+                    switch ( info.dwMinorVersion )
+                    {
+                        case 0:
+                            str.Printf(_("Windows 2000 (build %lu"),
+                                       info.dwBuildNumber);
+                            break;
+                        case 1:
+                            str.Printf(_("Windows XP (build %lu"),
+                                       info.dwBuildNumber);
+                            break;
+                        case 2:
+                            str.Printf(_("Windows Server 2003 (build %lu"),
+                                       info.dwBuildNumber);
+                            break;
+                    }
+                }
+                if ( wxIsEmpty(str) )
+                {
+                    str.Printf(_("Windows NT %lu.%lu (build %lu"),
                            info.dwMajorVersion,
                            info.dwMinorVersion,
                            info.dwBuildNumber);
+                }
                 if ( !wxIsEmpty(info.szCSDVersion) )
                 {
Index: src/unix/baseunix.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/unix/baseunix.cpp,v
retrieving revision 1.10
diff -p -u -b -u -2 -r1.10 baseunix.cpp
--- src/unix/baseunix.cpp	2004/05/23 20:53:26	1.10
+++ src/unix/baseunix.cpp	2004/10/15 02:33:42
@@ -93,9 +93,22 @@ wxToolkitInfo& wxConsoleAppTraits::GetTo
     static wxToolkitInfo info;
     int major, minor;
-    char name[256];
 
-    if ( sscanf(WXWIN_OS_DESCRIPTION, "%255s %d.%d", name, &major, &minor) != 3 )
+    FILE *f = popen("uname -r", "r");
+    if (f)
+    {
+        char buf[32];
+        size_t c = fread(buf, 1, sizeof(buf) - 1, f);
+        pclose(f);
+        buf[c] = '\0';
+        if ( sscanf(buf, "%d.%d", &major, &minor) != 2 )
+        {
+            // unrecognized uname string format
+            major =
+            minor = -1;
+        }
+    }
+    else
     {
-        // unreckognized uname string format
+        // failed to run uname
         major =
         minor = -1;
Index: src/unix/utilsunx.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/unix/utilsunx.cpp,v
retrieving revision 1.113
diff -p -u -b -u -2 -r1.113 utilsunx.cpp
--- src/unix/utilsunx.cpp	2004/10/08 00:08:33	1.113
+++ src/unix/utilsunx.cpp	2004/10/15 02:33:42
@@ -838,9 +838,18 @@ bool wxGetUserName(wxChar *buf, int sz)
 wxString wxGetOsDescription()
 {
-#ifndef WXWIN_OS_DESCRIPTION
-    #error WXWIN_OS_DESCRIPTION should be defined in config.h by configure
-#else
-    return wxString::FromAscii( WXWIN_OS_DESCRIPTION );
-#endif
+    FILE *f = popen("uname -s -r -m", "r");
+    if (f)
+    {
+        char buf[256];
+        size_t c = fread(buf, 1, sizeof(buf) - 1, f);
+        pclose(f);
+        // Trim newline from output.
+        if (c && buf[c - 1] == '\n')
+            --c;
+        buf[c] = '\0';
+        return wxString::FromAscii( buf );
+    }
+    wxFAIL_MSG( _T("uname failed") );
+    return _T("");
 }
 

 	  	 
