 src/msw/listctrl.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/msw/listctrl.cpp b/src/msw/listctrl.cpp
index 73b51d7..2983887 100644
--- a/src/msw/listctrl.cpp
+++ b/src/msw/listctrl.cpp
@@ -301,6 +301,9 @@ void wxListCtrl::MSWSetExListStyles()
 #endif
         LVS_EX_FULLROWSELECT |
         LVS_EX_SUBITEMIMAGES |
+#ifdef LVS_EX_DOUBLEBUFFER
+        LVS_EX_DOUBLEBUFFER |
+#endif
         // normally this should be governed by a style as it's probably not
         // always appropriate, but we don't have any free styles left and
         // it seems better to enable it by default than disable
