diff -ur ../wx0/src/mac/carbon/mimetmac.cpp ./src/mac/carbon/mimetmac.cpp
--- ../wx0/src/mac/carbon/mimetmac.cpp	2005-05-31 19:44:02.000000000 -0700
+++ ./src/mac/carbon/mimetmac.cpp	2005-05-31 23:58:31.000000000 -0700
@@ -1462,7 +1416,7 @@
             wxCFDictionary cfdInfo;
             bool bInfoOpenSuccess = false;
             wxFile indictfile;
-            if(indictfile.Open(sInfoPath, wxFile::read));
+            if(indictfile.Open(sInfoPath, wxFile::read))
             {
                 CFIndex cfiBufLen = (CFIndex) indictfile.Length();
                 const UInt8* pBuffer = new UInt8[cfiBufLen];
@@ -1805,7 +1759,7 @@
             wxCFDictionary cfdInfo;
             bool bInfoOpenSuccess = false;
             wxFile indictfile;
-            if(indictfile.Open(sInfoPath, wxFile::read));
+            if(indictfile.Open(sInfoPath, wxFile::read))
             {
                 CFIndex cfiBufLen = (CFIndex) indictfile.Length();
                 const UInt8* pBuffer = new UInt8[cfiBufLen];

 	  	 
