Index: include/wx/valgen.h
===================================================================
--- include/wx/valgen.h	(révision 78392)
+++ include/wx/valgen.h	(copie de travail)
@@ -34,7 +34,8 @@
     wxGenericValidator(bool* val);
         // wxChoice, wxGauge, wxRadioBox, wxScrollBar, wxSlider, wxSpinButton
     wxGenericValidator(int* val);
-        // wxComboBox, wxTextCtrl, wxButton, wxStaticText (read-only)
+        // wxComboBox, wxTextCtrl, wxButton, wxStaticText (read-only),
+        // wxFilePickerCtrl, wxDirPickerCtrl
     wxGenericValidator(wxString* val);
         // wxListBox, wxCheckListBox
     wxGenericValidator(wxArrayInt* val);
@@ -42,7 +43,7 @@
         // wxDatePickerCtrl
     wxGenericValidator(wxDateTime* val);
 #endif // wxUSE_DATETIME
-        // wxTextCtrl
+        // wxTextCtrl, wxFilePickerCtrl, wxDirPickerCtrl
     wxGenericValidator(wxFileName* val);
         // wxTextCtrl
     wxGenericValidator(float* val);
Index: interface/wx/valgen.h
===================================================================
--- interface/wx/valgen.h	(révision 78392)
+++ interface/wx/valgen.h	(copie de travail)
@@ -15,7 +15,7 @@
     - wxButton, wxRadioButton, wxToggleButton, wxBitmapToggleButton, wxSpinButton
     - wxCheckBox, wxRadioBox, wxComboBox, wxListBox, wxCheckListBox
     - wxGauge, wxSlider, wxScrollBar, wxChoice, wxStaticText
-    - wxSpinCtrl, wxTextCtrl
+    - wxSpinCtrl, wxFilePickerCtrl, wxDirPickerCtrl, wxTextCtrl
 
     It checks the type of the window and uses an appropriate type for it.
     For example, wxButton and wxTextCtrl transfer data to and from a
@@ -52,7 +52,8 @@
     wxGenericValidator(bool* valPtr);
     /**
         Constructor taking a wxString pointer. This will be used for wxButton,
-        wxComboBox, wxStaticText, wxTextCtrl.
+        wxComboBox, wxFilePickerCtrl, wxDirPickerCtrl, wxStaticText
+        and wxTextCtrl.
 
         @param valPtr
             A pointer to a variable that contains the value. This variable
@@ -97,7 +98,7 @@
     wxGenericValidator(wxDateTime* valPtr);
     /**
         Constructor taking a wxFileName pointer. This will be used for
-        wxTextCtrl.
+        wxFilePickerCtrl, wxDirPickerCtrl and wxTextCtrl.
 
         @param valPtr
             A pointer to a variable that contains the value. This variable
Index: src/common/valgen.cpp
===================================================================
--- src/common/valgen.cpp	(révision 78392)
+++ src/common/valgen.cpp	(copie de travail)
@@ -45,6 +45,9 @@
 #if wxUSE_TOGGLEBTN
     #include "wx/tglbtn.h"
 #endif
+#if wxUSE_FILEPICKERCTRL
+    #include "wx/filepicker.h"
+#endif
 #include "wx/filename.h"
 
 #include "wx/valgen.h"
@@ -315,6 +318,38 @@
         }
     } else
 #endif
+#if wxUSE_FILEPICKERCTRL
+    if (wxDynamicCast(m_validatorWindow, wxFilePickerCtrl))
+    {
+        wxFilePickerCtrl* pControl = (wxFilePickerCtrl*) m_validatorWindow;
+        if (m_pString)
+        {
+            pControl->SetPath(*m_pString) ;
+            return true;
+        }
+        else if (m_pFileName)
+        {
+            pControl->SetFileName(*m_pFileName);
+            return true;
+        }
+    } else
+#endif
+#if wxUSE_DIRPICKERCTRL
+    if (wxDynamicCast(m_validatorWindow, wxDirPickerCtrl))
+    {
+        wxDirPickerCtrl* pControl = (wxDirPickerCtrl*) m_validatorWindow;
+        if (m_pString)
+        {
+            pControl->SetPath(*m_pString) ;
+            return true;
+        }
+        else if (m_pFileName)
+        {
+            pControl->SetDirName(*m_pFileName);
+            return true;
+        }
+    } else
+#endif
 #if wxUSE_STATTEXT
     if (wxDynamicCast(m_validatorWindow, wxStaticText))
     {
@@ -593,6 +628,38 @@
         }
     } else
 #endif
+#if wxUSE_FILEPICKERCTRL
+    if (wxDynamicCast(m_validatorWindow, wxFilePickerCtrl))
+    {
+        wxFilePickerCtrl* pControl = (wxFilePickerCtrl*) m_validatorWindow;
+        if (m_pString)
+        {
+            *m_pString = pControl->GetPath() ;
+            return true;
+        }
+        else if (m_pFileName)
+        {
+            *m_pFileName = pControl->GetFileName();
+            return true;
+        }
+    } else
+#endif
+#if wxUSE_DIRPICKERCTRL
+    if (wxDynamicCast(m_validatorWindow, wxDirPickerCtrl))
+    {
+        wxDirPickerCtrl* pControl = (wxDirPickerCtrl*) m_validatorWindow;
+        if (m_pString)
+        {
+            *m_pString = pControl->GetPath() ;
+            return true;
+        }
+        else if (m_pFileName)
+        {
+            *m_pFileName = pControl->GetDirName();
+            return true;
+        }
+    } else
+#endif
 #if wxUSE_STATTEXT
     if (wxDynamicCast(m_validatorWindow, wxStaticText))
     {
