Index: string.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/common/string.cpp,v
retrieving revision 1.174
diff -c -r1.174 string.cpp
*** string.cpp	2003/06/15 19:33:04	1.174
--- string.cpp	2003/06/28 08:28:27
***************
*** 264,272 ****
          }
  
          // MB2WC wants the buffer size, not the string length
!         if ( conv.MB2WC(m_pchData, psz, nLen + 1) != (size_t)-1 )
          {
              // initialized ok
              m_pchData[nLen] = 0;
              return;
          }
--- 264,275 ----
          }
  
          // MB2WC wants the buffer size, not the string length
!         nLen = conv.MB2WC(m_pchData, psz, nLen);
! 
!         if (nLen != (size_t)-1 )
          {
              // initialized ok
+             GetStringData()->nDataLength = nLen;
              m_pchData[nLen] = 0;
              return;
          }

 	  	 
