Index: C:/dev-fix/2.3.x/dev/sdk/wxWidgets-2.8.10/include/wx/generic/listctrl.h
===================================================================
--- C:/dev-fix/2.3.x/dev/sdk/wxWidgets-2.8.10/include/wx/generic/listctrl.h	(revision 722)
+++ C:/dev-fix/2.3.x/dev/sdk/wxWidgets-2.8.10/include/wx/generic/listctrl.h	(revision 723)
@@ -136,8 +136,7 @@
     bool ScrollList( int dx, int dy );
     bool SortItems( wxListCtrlCompare fn, long data );
     bool Update( long item );
-    // Must provide overload to avoid hiding it (and warnings about it)
-    virtual void Update() { wxControl::Update(); }
+    virtual void Update();
 
     // are we in report mode?
     bool InReportView() const { return HasFlag(wxLC_REPORT); }
Index: C:/dev-fix/2.3.x/dev/sdk/wxWidgets-2.8.10/src/generic/listctrl.cpp
===================================================================
--- C:/dev-fix/2.3.x/dev/sdk/wxWidgets-2.8.10/src/generic/listctrl.cpp	(revision 722)
+++ C:/dev-fix/2.3.x/dev/sdk/wxWidgets-2.8.10/src/generic/listctrl.cpp	(revision 723)
@@ -2732,7 +2732,10 @@
     wxCHECK_RET( m_freezeCount > 0, _T("thawing unfrozen list control?") );
 
     if ( --m_freezeCount == 0 )
-        Refresh();
+        if (m_dirty)
+            RecalculatePositions();
+        else
+            Refresh();
 }
 
 void wxListMainWindow::OnPaint( wxPaintEvent &WXUNUSED(event) )
@@ -5945,6 +5948,14 @@
     }
 }
 
+void wxGenericListCtrl::Update()
+{
+    if (m_mainWin && m_mainWin->m_dirty)
+        m_mainWin->RecalculatePositions();
+
+    wxControl::Update();
+}
+
 void wxGenericListCtrl::Freeze()
 {
     m_mainWin->Freeze();
