Index: setup.py
===================================================================
--- setup.py	(revision 67812)
+++ setup.py	(working copy)
@@ -923,6 +923,11 @@
 DATA_FILES += find_data_files('wx/tools/Editra/tests/syntax', '*')
 DATA_FILES += find_data_files('wx/tools/Editra', '[A-Z]*', recursive=False)
 
+# if on msw need the wx DLLs copied into the wxPython install dir
+if os.name == 'nt':
+    DATA_FILES += find_data_files('wx', 'wxbase%s*%s_*.dll' % (WXDLLVER, libFlag()),
+                                        'wxmsw%s*%s_*.dll' % (WXDLLVER, libFlag()),
+                                        recursive=False)
 
 ## import pprint
 ## pprint.pprint(DATA_FILES)
