Index: src/aui/auibook.cpp
===================================================================
--- src/aui/auibook.cpp	(revision 71932)
+++ src/aui/auibook.cpp	(working copy)
@@ -2544,6 +2544,10 @@
             dest_tabs->Refresh();
             m_lastDragX = pt.x;
 
+            // the move is on the fly and is not cancelable, we need to update
+            // 'm_tabs' on the fly as well
+            m_tabs.MovePage(src_tab, dest_idx);
+            m_curPage = dest_idx;
         }
 
         return;
