Index: include/wx/stc/stc.h
===================================================================
--- include/wx/stc/stc.h	(revision 53094)
+++ include/wx/stc/stc.h	(working copy)
@@ -3501,7 +3501,7 @@
 
     // Set style size, face, bold, italic, and underline attributes from
     // a wxFont's attributes.
-    void StyleSetFont(int styleNum, wxFont& font);
+    void StyleSetFont(int styleNum, const wxFont&);
 
 
 
@@ -3625,6 +3625,37 @@
     // Append a string to the end of the document without changing the selection.
     void AppendTextRaw(const char* text);
 
+    // wxTextCtrl compatibility section begin
+    
+    void SetValue(const wxString& str, bool discardedits = true)
+    { 
+        SetText(str);
+        if (discardedits) DiscardEdits();
+    }
+    
+    wxString GetValue()
+    { 
+        return GetText();
+    }
+    
+    void DiscardEdits()
+    { 
+        SetSavePoint();
+    }
+    
+    bool IsModified()
+    { 
+        return GetModify();
+    }
+    
+    virtual bool SetFont(const wxFont& font)
+    {
+        StyleSetFont(wxSTC_STYLE_DEFAULT, font);
+        return wxControl::SetFont(font);
+    }
+    
+    // wxTextCtrl compatibility section end
+
 #ifdef SWIG
     %pythoncode "_stc_utf8_methods.py"
 #endif
Index: src/stc/stc.cpp
===================================================================
--- src/stc/stc.cpp	(revision 53094)
+++ src/stc/stc.cpp	(working copy)
@@ -50,37 +50,26 @@
 
 #define MAKELONG(a, b) ((a) | ((b) << 16))
 
-
-static long wxColourAsLong(const wxColour& co) {
-    return (((long)co.Blue()  << 16) |
-            ((long)co.Green() <<  8) |
-            ((long)co.Red()));
+static long wxColourAsLong(const wxColour& clr)
+{
+    // No wxColour::ToLong()
+    return (((long)clr.Blue()  << 16) |
+            ((long)clr.Green() <<  8) |
+            ((long)clr.Red()));
 }
 
-static wxColour wxColourFromLong(long c) {
+static wxColour wxColourFromLong(long c)
+{
     wxColour clr;
-    clr.Set((unsigned char)(c & 0xff),
-            (unsigned char)((c >> 8) & 0xff),
-            (unsigned char)((c >> 16) & 0xff));
+    clr.Set((unsigned long)c);
     return clr;
 }
 
-
-static wxColour wxColourFromSpec(const wxString& spec) {
-    // spec should be a colour name or "#RRGGBB"
-    if (spec.GetChar(0) == wxT('#')) {
-
-        long red, green, blue;
-        red = green = blue = 0;
-        spec.Mid(1,2).ToLong(&red,   16);
-        spec.Mid(3,2).ToLong(&green, 16);
-        spec.Mid(5,2).ToLong(&blue,  16);
-        return wxColour((unsigned char)red,
-                        (unsigned char)green,
-                        (unsigned char)blue);
-    }
-    else
-        return wxColour(spec);
+static wxColour wxColourFromSpec(const wxString& spec)
+{
+    wxColour clr;
+    clr.Set(spec);
+    return clr;
 }
 
 //----------------------------------------------------------------------
@@ -2892,7 +2881,9 @@
 
 // Set style size, face, bold, italic, and underline attributes from
 // a wxFont's attributes.
-void wxStyledTextCtrl::StyleSetFont(int styleNum, wxFont& font) {
+void wxStyledTextCtrl::StyleSetFont(int styleNum, const wxFont& font_ref)
+{
+    wxFont font = font_ref;
 #ifdef __WXGTK__
     // Ensure that the native font is initialized
     int x, y;

 	  	 
