diff --git a/demos/forty/scorefil.cpp b/demos/forty/scorefil.cpp
index 96f66d3..9d60a32 100644
--- a/demos/forty/scorefil.cpp
+++ b/demos/forty/scorefil.cpp
@@ -20,11 +20,6 @@
 #include "wx/wx.h"
 #endif
 
-#ifdef __WXGTK__
-#include <unistd.h>
-#include <sys/stat.h>
-#include <fcntl.h>
-#endif
 #include "wx/textfile.h"
 #include "wx/config.h"
 #include "wx/fileconf.h"
diff --git a/src/gtk/dirdlg.cpp b/src/gtk/dirdlg.cpp
index c6655fd..cc5b68e 100644
--- a/src/gtk/dirdlg.cpp
+++ b/src/gtk/dirdlg.cpp
@@ -31,7 +31,9 @@
 
 #include "wx/gtk/private.h"
 
+#ifdef __UNIX__
 #include <unistd.h> // chdir
+#endif
 
 //-----------------------------------------------------------------------------
 // "clicked" for OK-button
diff --git a/src/gtk/filedlg.cpp b/src/gtk/filedlg.cpp
index a9359fa..2844d0b 100644
--- a/src/gtk/filedlg.cpp
+++ b/src/gtk/filedlg.cpp
@@ -22,7 +22,9 @@
 #include <gtk/gtk.h>
 #include "wx/gtk/private.h"
 
+#ifdef __UNIX__
 #include <unistd.h> // chdir
+#endif
 
 #include "wx/filename.h" // wxFilename
 #include "wx/tokenzr.h" // wxStringTokenizer
diff --git a/src/gtk/filepicker.cpp b/src/gtk/filepicker.cpp
index f457717..b3d1678 100644
--- a/src/gtk/filepicker.cpp
+++ b/src/gtk/filepicker.cpp
@@ -144,7 +144,9 @@ void wxFileButton::SetInitialDirectory(const wxString& dir)
 
 #if wxUSE_DIRPICKERCTRL && defined(__WXGTK26__)
 
+#ifdef __UNIX__
 #include <unistd.h> // chdir
+#endif
 
 //-----------------------------------------------------------------------------
 // "current-folder-changed"
diff --git a/src/gtk/utilsgtk.cpp b/src/gtk/utilsgtk.cpp
index abe0d04..7c34250 100644
--- a/src/gtk/utilsgtk.cpp
+++ b/src/gtk/utilsgtk.cpp
@@ -45,8 +47,9 @@
 #include <string.h>
 #include <sys/stat.h>
 #include <sys/types.h>
-#include <sys/wait.h>   // for WNOHANG
+#ifdef __UNIX__
 #include <unistd.h>
+#endif
 
 #if wxUSE_DETECT_SM
     #include <X11/SM/SMlib.h>
