diff --git a/include/wx/event.h b/include/wx/event.h
index 44d378a..403f854 100644
--- a/include/wx/event.h
+++ b/include/wx/event.h
@@ -332,7 +332,7 @@ class wxEventFunctorMethod
               <
                 Class,
                 EventArg,
-                wxConvertibleTo<Class, wxEvtHandler>::value != 0
+                wxIsPublicDerivedFromwxEvtHandler<Class, wxEvtHandler>::value != 0
               >
 {
 private:
diff --git a/include/wx/meta/convertible.h b/include/wx/meta/convertible.h
index 4599937..2f93886 100644
--- a/include/wx/meta/convertible.h
+++ b/include/wx/meta/convertible.h
@@ -20,8 +20,8 @@ struct wxConvertibleTo_SizeHelper
     static int  Match(...);
 };
 
-// Helper to decide if an object of type D is convertible to type B (the test
-// succeeds in particular when D derives from B)
+// Helper to decide if an object of type D is convertible to type B statically
+// (the test succeeds in particular when D derives from B)
 template <class D, class B>
 struct wxConvertibleTo
 {
@@ -34,5 +34,28 @@ struct wxConvertibleTo
     };
 };
 
+// Checks if the class is publicly derived from wxEvtHandler.
+template <class D, class B>
+struct wxIsPublicDerivedFromwxEvtHandler
+{
+    // void wxSpecialMethodForEventHandlerDetection() is a special method for this
+    typedef void (B::*pTowxSpecialMethodForEventHandlerDetection)();
+
+    template <typename T, T>
+    struct stTest;
+
+    template <typename U>
+    static char Match(stTest<pTowxSpecialMethodForEventHandlerDetection, &U::wxSpecialMethodForEventHandlerDetection>*);
+
+    // This function also needs to be a template
+    template <typename S>
+    static int Match(...);
+
+    enum
+    {
+        value = sizeof(Match<D>(0)) == sizeof(char)
+    };
+};
+
 #endif // _WX_META_CONVERTIBLE_H_
 
diff --git a/interface/wx/event.h b/interface/wx/event.h
index 3d92c2d..c882543 100644
--- a/interface/wx/event.h
+++ b/interface/wx/event.h
@@ -1273,6 +1273,13 @@ protected:
         @see ProcessEvent()
      */
     virtual bool TryAfter(wxEvent& event);
+
+private:
+    /**
+        This method is used by wxIsPublicDerivedFromwxEvtHandler to determine
+        if a class derives from wxEvtHandler.
+    */
+    void wxSpecialMethodForEventHandlerDetection();
 };
 
 #endif // wxUSE_BASE
