Index: src/common/strconv.cpp
===================================================================
--- src/common/strconv.cpp	(revision 65218)
+++ src/common/strconv.cpp	(working copy)
@@ -2496,6 +2496,13 @@
 #if wxUSE_UNICODE_UTF8
 bool wxMBConv_iconv::IsUTF8() const
 {
+#if wxUSE_THREADS
+    // protect m_name with a mutex since it can be used in multiple threads
+    // and wxString is not thread safe:
+    static wxMutex mutex;
+    wxMutexLocker lock(mutex);
+#endif
+
     return wxStricmp(m_name, "UTF-8") == 0 ||
            wxStricmp(m_name, "UTF8") == 0;
 }
@@ -3227,22 +3234,29 @@
 {
     if ( m_deferred )
     {
-        wxCSConv *self = (wxCSConv *)this; // const_cast
+#if wxUSE_THREADS
+        static wxMutex mutex;
+        wxMutexLocker lock(mutex);
+#endif
+        if(m_deferred)
+        {
+            wxCSConv *self = (wxCSConv *)this; // const_cast
 
-        // if we don't have neither the name nor the encoding, use the default
-        // encoding for this system
-        if ( !m_name && m_encoding == wxFONTENCODING_SYSTEM )
-        {
-#if wxUSE_INTL
-            self->m_encoding = wxLocale::GetSystemEncoding();
-#else
-            // fallback to some reasonable default:
-            self->m_encoding = wxFONTENCODING_ISO8859_1;
-#endif // wxUSE_INTL
+            // if we don't have neither the name nor the encoding, use the default
+            // encoding for this system
+            if ( !m_name && m_encoding == wxFONTENCODING_SYSTEM )
+            {
+    #if wxUSE_INTL
+                self->m_encoding = wxLocale::GetSystemEncoding();
+    #else
+                // fallback to some reasonable default:
+                self->m_encoding = wxFONTENCODING_ISO8859_1;
+    #endif // wxUSE_INTL
+            }
+
+            self->m_convReal = DoCreate();
+            self->m_deferred = false;
         }
-
-        self->m_convReal = DoCreate();
-        self->m_deferred = false;
     }
 }
 
