Index: interface/wx/textctrl.h
===================================================================
--- interface/wx/textctrl.h	(revision 54240)
+++ interface/wx/textctrl.h	(working copy)
@@ -1568,27 +1568,20 @@
     specify @c --enable-std_iostreams switch when running configure for this.
 
     Example of usage:
-
     @code
     using namespace std;
-
-      wxTextCtrl *text = new wxTextCtrl(...);
-
+      wxTextCtrl* text = new wxTextCtrl(...);
       {
         wxStreamToTextRedirector redirect(text);
 
         // this goes to the text control
-        cout  "Hello, text!"  endl;
+        cout << "Hello, text!" << endl;
       }
-
       // this goes somewhere else, presumably to stdout
-      cout  "Hello, console!"  endl;
+      cout << "Hello, console!" << endl;
     @endcode
-
-
     @library{wxcore}
     @category{logging}
-
     @see wxTextCtrl
 */
 class wxStreamToTextRedirector
