Index: src/common/docview.cpp
===================================================================
--- src/common/docview.cpp	(revision 61045)
+++ src/common/docview.cpp	(working copy)
@@ -899,8 +899,6 @@
 
 wxDocManager::wxDocManager(long WXUNUSED(flags), bool initialize)
 {
-    wxASSERT_MSG( !sm_docManager, "multiple wxDocManagers not allowed" );
-
     sm_docManager = this;
 
     m_defaultDocumentNameCounter = 1;
