diff -ru2 -xCVS -xDebug -xRelease -xlib -x'.#*' -x'*.orig' -x'*.rej' -x'*~' -x'*.dsp' -x'*.dsw' -x'*.ncb' -x'*.opt' -x'*.plg' -x'*.bak' wxWindows.orig/include/wx/event.h wxWindows/include/wx/event.h
--- wxWindows.orig/include/wx/event.h	2004-02-15 13:59:49.996406400 +0100
+++ wxWindows/include/wx/event.h	2004-03-16 15:33:08.563447300 +0100
@@ -497,4 +497,6 @@
         { }
 
+    ~wxCommandEvent();
+
     // Set/Get client data from controls
     void SetClientData(void* clientData) { m_clientData = clientData; }
diff -ru2 -xCVS -xDebug -xRelease -xlib -x'.#*' -x'*.orig' -x'*.rej' -x'*~' -x'*.dsp' -x'*.dsw' -x'*.ncb' -x'*.opt' -x'*.plg' -x'*.bak' wxWindows.orig/src/common/event.cpp wxWindows/src/common/event.cpp
--- wxWindows.orig/src/common/event.cpp	2004-02-07 13:54:29.000000000 +0100
+++ wxWindows/src/common/event.cpp	2004-03-16 17:10:44.847113500 +0100
@@ -394,4 +394,14 @@
 }
 
+wxCommandEvent::~wxCommandEvent()
+{
+    if( m_clientObject )
+    {
+        delete m_clientObject;
+    }
+
+}
+
+
 /*
  * UI update events

 	  	 
