Index: src/common/memory.cpp
===================================================================
--- src/common/memory.cpp	(revision 50921)
+++ src/common/memory.cpp	(working copy)
@@ -39,7 +39,7 @@
 #include "wx/ioswrap.h"
 
 #if !defined(__WATCOMC__) && !(defined(__VMS__) && ( __VMS_VER < 70000000 ) )\
-     && !defined( __MWERKS__ ) && !defined(__SALFORDC__)
+     && !defined( __MWERKS__ )
 #include <memory.h>
 #endif
 
Index: src/common/file.cpp
===================================================================
--- src/common/file.cpp	(revision 50921)
+++ src/common/file.cpp	(working copy)
@@ -1,5 +1,5 @@
 /////////////////////////////////////////////////////////////////////////////
-// Name:        file.cpp
+// Name:        src/common/file.cpp
 // Purpose:     wxFile - encapsulates low-level "file descriptor"
 //              wxTempFile
 // Author:      Vadim Zeitlin
@@ -26,7 +26,7 @@
 // standard
 #if defined(__WXMSW__) && !defined(__GNUWIN32__) && !defined(__WXMICROWIN__) && !defined(__WXWINCE__)
 
-#ifndef __SALFORDC__
+#if 1
     #define   WIN32_LEAN_AND_MEAN
     #define   NOSERVICE
     #define   NOIME
@@ -99,18 +99,6 @@
     };
 #endif // W_OK
 
-#ifdef __SALFORDC__
-    #include <unix.h>
-#endif
-
-// some broken compilers don't have 3rd argument in open() and creat()
-#ifdef __SALFORDC__
-    #define ACCESS(access)
-    #define stat    _stat
-#else // normal compiler
-    #define ACCESS(access)  , (access)
-#endif // Salford C
-
 // wxWidgets
 #ifndef WX_PRECOMP
     #include  "wx/string.h"
@@ -200,8 +188,8 @@
     // otherwise we only create the new file and fail if it already exists
     int fd = wxOpen( fileName,
                      O_BINARY | O_WRONLY | O_CREAT |
-                     (bOverwrite ? O_TRUNC : O_EXCL)
-                     ACCESS(accessMode) );
+                     (bOverwrite ? O_TRUNC : O_EXCL),
+                     accessMode );
     if ( fd == -1 )
     {
         wxLogSysError(_("can't create file '%s'"), fileName);
@@ -252,7 +240,7 @@
     accessMode &= wxS_IRUSR | wxS_IWUSR;
 #endif // __WINDOWS__
 
-    int fd = wxOpen( fileName, flags ACCESS(accessMode));
+    int fd = wxOpen( fileName, flags, accessMode);
 
     if ( fd == -1 )
     {
@@ -435,7 +423,7 @@
 
     wxFileOffset iRc;
 
-#if defined(__DOS__) || defined(__UNIX__) || defined(__GNUWIN32__) || defined( __MWERKS__ ) || defined(__SALFORDC__)
+#if defined(__DOS__) || defined(__UNIX__) || defined(__GNUWIN32__) || defined( __MWERKS__ )
     // @@ this doesn't work, of course, on unseekable file descriptors
     wxFileOffset ofsCur = Tell(),
     ofsMax = Length();
Index: src/common/string.cpp
===================================================================
--- src/common/string.cpp	(revision 50921)
+++ src/common/string.cpp	(working copy)
@@ -35,10 +35,6 @@
 #include <string.h>
 #include <stdlib.h>
 
-#ifdef __SALFORDC__
-    #include <clib.h>
-#endif
-
 #include "wx/hashmap.h"
 
 // string handling functions used by wxString:
Index: src/common/valtext.cpp
===================================================================
--- src/common/valtext.cpp	(revision 50921)
+++ src/common/valtext.cpp	(working copy)
@@ -33,10 +33,6 @@
 #include <string.h>
 #include <stdlib.h>
 
-#ifdef __SALFORDC__
-    #include <clib.h>
-#endif
-
 IMPLEMENT_DYNAMIC_CLASS(wxTextValidator, wxValidator)
 
 BEGIN_EVENT_TABLE(wxTextValidator, wxValidator)
Index: src/common/strconv.cpp
===================================================================
--- src/common/strconv.cpp	(revision 50921)
+++ src/common/strconv.cpp	(working copy)
@@ -44,10 +44,6 @@
     #define wxHAVE_WIN32_MB2WC
 #endif
 
-#ifdef __SALFORDC__
-    #include <clib.h>
-#endif
-
 #ifdef HAVE_ICONV
     #include <iconv.h>
     #include "wx/thread.h"
Index: src/common/imagpng.cpp
===================================================================
--- src/common/imagpng.cpp	(revision 50921)
+++ src/common/imagpng.cpp	(working copy)
@@ -1,5 +1,5 @@
 /////////////////////////////////////////////////////////////////////////////
-// Name:        src/common/imagepng.cpp
+// Name:        src/common/imagpng.cpp
 // Purpose:     wxImage PNG handler
 // Author:      Robert Roebling
 // RCS-ID:      $Id$
@@ -42,12 +42,6 @@
 // For memcpy
 #include <string.h>
 
-#ifdef __SALFORDC__
-#ifdef FAR
-#undef FAR
-#endif
-#endif
-
 // ----------------------------------------------------------------------------
 // constants
 // ----------------------------------------------------------------------------
Index: src/common/sckaddr.cpp
===================================================================
--- src/common/sckaddr.cpp	(revision 50921)
+++ src/common/sckaddr.cpp	(working copy)
@@ -27,7 +27,7 @@
     #include <stdlib.h>
     #include <ctype.h>
 
-    #if !defined(__MWERKS__) && !defined(__SALFORDC__)
+    #if !defined(__MWERKS__)
         #include <memory.h>
     #endif
 #endif // !WX_PRECOMP
Index: src/common/imagjpeg.cpp
===================================================================
--- src/common/imagjpeg.cpp	(revision 50921)
+++ src/common/imagjpeg.cpp	(working copy)
@@ -57,10 +57,6 @@
 // For JPEG library error handling
 #include <setjmp.h>
 
-#ifdef __SALFORDC__
-#undef FAR
-#endif
-
 // ----------------------------------------------------------------------------
 // types
 // ----------------------------------------------------------------------------
Index: src/common/imagbmp.cpp
===================================================================
--- src/common/imagbmp.cpp	(revision 50921)
+++ src/common/imagbmp.cpp	(working copy)
@@ -38,12 +38,6 @@
 // For memcpy
 #include <string.h>
 
-#ifdef __SALFORDC__
-#ifdef FAR
-#undef FAR
-#endif
-#endif
-
 //-----------------------------------------------------------------------------
 // wxBMPHandler
 //-----------------------------------------------------------------------------
Index: src/common/stringimpl.cpp
===================================================================
--- src/common/stringimpl.cpp	(revision 50921)
+++ src/common/stringimpl.cpp	(working copy)
@@ -42,10 +42,6 @@
 #include <string.h>
 #include <stdlib.h>
 
-#ifdef __SALFORDC__
-    #include <clib.h>
-#endif
-
 // allocating extra space for each string consumes more memory but speeds up
 // the concatenation operations (nLen is the current string's length)
 // NB: EXTRA_ALLOC must be >= 0!
Index: src/msw/dib.cpp
===================================================================
--- src/msw/dib.cpp	(revision 50921)
+++ src/msw/dib.cpp	(working copy)
@@ -45,7 +45,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 
-#if !defined(__MWERKS__) && !defined(__SALFORDC__)
+#if !defined(__MWERKS__)
     #include <memory.h>
 #endif
 
Index: src/msw/app.cpp
===================================================================
--- src/msw/app.cpp	(revision 50921)
+++ src/msw/app.cpp	(working copy)
@@ -61,7 +61,7 @@
 
 // OLE is used for drag-and-drop, clipboard, OLE Automation..., but some
 // compilers don't support it (missing headers, libs, ...)
-#if defined(__GNUWIN32_OLD__) || defined(__SYMANTEC__) || defined(__SALFORDC__)
+#if defined(__GNUWIN32_OLD__) || defined(__SYMANTEC__)
     #undef wxUSE_OLE
 
     #define  wxUSE_OLE 0
Index: src/msw/utilsexc.cpp
===================================================================
--- src/msw/utilsexc.cpp	(revision 50921)
+++ src/msw/utilsexc.cpp	(working copy)
@@ -44,7 +44,7 @@
 
 #include <ctype.h>
 
-#if !defined(__GNUWIN32__) && !defined(__SALFORDC__) && !defined(__WXMICROWIN__) && !defined(__WXWINCE__)
+#if !defined(__GNUWIN32__) && !defined(__WXMICROWIN__) && !defined(__WXWINCE__)
     #include <direct.h>
 #ifndef __MWERKS__
     #include <dos.h>
Index: src/msw/wince/tbarwce.cpp
===================================================================
--- src/msw/wince/tbarwce.cpp	(revision 50921)
+++ src/msw/wince/tbarwce.cpp	(working copy)
@@ -42,7 +42,7 @@
     #include "wx/control.h"
 #endif
 
-#if !defined(__GNUWIN32__) && !defined(__SALFORDC__)
+#if !defined(__GNUWIN32__)
     #include "malloc.h"
 #endif
 
Index: src/msw/utils.cpp
===================================================================
--- src/msw/utils.cpp	(revision 50921)
+++ src/msw/utils.cpp	(working copy)
@@ -55,7 +55,7 @@
     #include <winsock.h>
 #endif
 
-#if !defined(__GNUWIN32__) && !defined(__SALFORDC__) && !defined(__WXMICROWIN__) && !defined(__WXWINCE__)
+#if !defined(__GNUWIN32__) && !defined(__WXMICROWIN__) && !defined(__WXWINCE__)
     #include <direct.h>
 
     #ifndef __MWERKS__
Index: src/msw/printwin.cpp
===================================================================
--- src/msw/printwin.cpp	(revision 50921)
+++ src/msw/printwin.cpp	(working copy)
@@ -180,7 +180,7 @@
     wxWindow *win = CreateAbortWindow(parent, printout);
     wxYield();
 
-#if defined(__WATCOMC__) || defined(__BORLANDC__) || defined(__GNUWIN32__) || defined(__SALFORDC__) || !defined(__WIN32__)
+#if defined(__WATCOMC__) || defined(__BORLANDC__) || defined(__GNUWIN32__) || !defined(__WIN32__)
 #ifdef STRICT
     ::SetAbortProc((HDC) impl->GetHDC(), (ABORTPROC) m_lpAbortProc);
 #else
Index: include/wx/list.h
===================================================================
--- include/wx/list.h	(revision 50921)
+++ include/wx/list.h	(working copy)
@@ -548,10 +548,6 @@
                                    void *data,
                                    const wxListKey& key = wxDefaultListKey) = 0;
 
-// Can't access these from derived classes otherwise (bug in Salford C++?)
-#ifdef __SALFORDC__
-public:
-#endif
 
     // ctors
         // from an array
Index: include/wx/filefn.h
===================================================================
--- include/wx/filefn.h	(revision 50921)
+++ include/wx/filefn.h	(working copy)
@@ -60,7 +60,7 @@
 #endif
 
 #if defined(__WINDOWS__) && !defined(__WXMICROWIN__)
-#if !defined( __GNUWIN32__ ) && !defined( __MWERKS__ ) && !defined(__SALFORDC__) && !defined(__WXWINCE__) && !defined(__CYGWIN__)
+#if !defined( __GNUWIN32__ ) && !defined( __MWERKS__ ) && !defined(__WXWINCE__) && !defined(__CYGWIN__)
     #include <direct.h>
     #include <dos.h>
     #include <io.h>
@@ -85,11 +85,6 @@
     #include <dir.h>
 #endif
 
-#ifdef __SALFORDC__
-    #include <dir.h>
-    #include <unix.h>
-#endif
-
 #ifndef __WXPALMOS5__
 #ifndef __WXWINCE__
     #include  <fcntl.h>       // O_RDONLY &c
Index: include/wx/dirdlg.h
===================================================================
--- include/wx/dirdlg.h	(revision 50921)
+++ include/wx/dirdlg.h	(working copy)
@@ -96,8 +96,7 @@
 #if defined(__WXUNIVERSAL__)
     #include "wx/generic/dirdlgg.h"
     #define wxDirDialog wxGenericDirDialog
-#elif defined(__WXMSW__) && (defined(__SALFORDC__)    || \
-                             !wxUSE_OLE               || \
+#elif defined(__WXMSW__) && (!wxUSE_OLE               || \
                              (defined (__GNUWIN32__) && !wxUSE_NORLANDER_HEADERS))
     #include "wx/generic/dirdlgg.h"
     #define wxDirDialog wxGenericDirDialog
Index: include/wx/string.h
===================================================================
--- include/wx/string.h	(revision 50921)
+++ include/wx/string.h	(working copy)
@@ -133,8 +133,6 @@
   return _stricmp(psz1, psz2);
 #elif defined(__SC__)
   return _stricmp(psz1, psz2);
-#elif defined(__SALFORDC__)
-  return stricmp(psz1, psz2);
 #elif defined(__BORLANDC__)
   return stricmp(psz1, psz2);
 #elif defined(__WATCOMC__)
Index: include/wx/wxcrtbase.h
===================================================================
--- include/wx/wxcrtbase.h	(revision 50921)
+++ include/wx/wxcrtbase.h	(working copy)
@@ -229,7 +229,7 @@
  * versions */
 #if !defined(wxCRT_StricmpA)
     #if defined(__BORLANDC__) || defined(__WATCOMC__) || \
-            defined(__SALFORDC__) || defined(__VISAGECPP__) || \
+            defined(__VISAGECPP__) || \
             defined(__EMX__) || defined(__DJGPP__)
         #define wxCRT_StricmpA stricmp
         #define wxCRT_StrnicmpA strnicmp
Index: include/wx/defs.h
===================================================================
--- include/wx/defs.h	(revision 50921)
+++ include/wx/defs.h	(working copy)
@@ -103,15 +103,6 @@
     #endif /* VC++ 8 */
 #endif /*  __VISUALC__ */
 
-/*  suppress some Salford C++ warnings */
-#ifdef __SALFORDC__
-#   pragma suppress 353             /*  Possible nested comments */
-#   pragma suppress 593             /*  Define not used */
-#   pragma suppress 61              /*  enum has no name (doesn't suppress!) */
-#   pragma suppress 106             /*  unnamed, unused parameter */
-#   pragma suppress 571             /*  Virtual function hiding */
-#endif /*  __SALFORDC__ */
-
 /*  suppress some Borland C++ warnings */
 #ifdef __BORLANDC__
 #   pragma warn -inl                /*  Functions containing reserved words and certain constructs are not expanded inline */
Index: distrib/msw/msw_minimal.rsp
===================================================================
--- distrib/msw/msw_minimal.rsp	(revision 50921)
+++ distrib/msw/msw_minimal.rsp	(working copy)
@@ -195,10 +195,6 @@
 src/makeg95.env
 src/makeprog.g95
 src/makelib.g95
-src/makesl.env
-src/makeprog.sl
-src/makelib.sl
-src/salford.lnk
 src/maketwin.env
 src/makefile.bcc
 src/makefile.vc
Index: distrib/scripts/manifests/msw_minimal.rsp
===================================================================
--- distrib/scripts/manifests/msw_minimal.rsp	(revision 50921)
+++ distrib/scripts/manifests/msw_minimal.rsp	(working copy)
@@ -193,10 +193,6 @@
 src/makeg95.env
 src/makeprog.g95
 src/makelib.g95
-src/makesl.env
-src/makeprog.sl
-src/makelib.sl
-src/salford.lnk
 src/maketwin.env
 src/makefile.bcc
 src/makefile.vc

 	  	 
