Index: wxWindows/src/msw/slider95.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/slider95.cpp,v
retrieving revision 1.51
diff -b -u -2 -r1.51 slider95.cpp
--- wxWindows/src/msw/slider95.cpp	2003/09/15 19:54:51	1.51
+++ wxWindows/src/msw/slider95.cpp	2003/09/22 09:14:05
@@ -519,5 +519,7 @@
             x_offset += (int)(min_len + cx);
 
-            int slider_length = (int)(w1 - x_offset - max_len - cx);
+            // slider_length = (total width available) - (width used so far)
+            //                   - (width of max label) - (border)
+            int slider_length = (int)(w1 - (x_offset-x) - max_len - cx);
 
             int slider_height = h1;
@@ -596,5 +598,7 @@
             y_offset += cy;
 
-            int slider_length = (int)(h1 - y_offset - cy - cy);
+            //  slider_length = (total height available) - (height used so far) 
+            //                              - (height of max label) - (border)
+            int slider_length = (int)(h1 - (y_offset-y) - cy - cy);
 
             int slider_width = w1;

 	  	 
