--- src\common\string.cpp	Mon Jan 08 01:21:46 2007
+++ string.cpp	Wed Jun 27 16:36:23 2007
@@ -1570,16 +1570,24 @@
 
 wxString& wxString::MakeUpper()
 {
+#ifdef __WXMSW__
+  ::CharUpperBuff(begin(), length());
+#else
   for ( iterator it = begin(), en = end(); it != en; ++it )
     *it = (wxChar)wxToupper(*it);
+#endif
 
   return *this;
 }
 
 wxString& wxString::MakeLower()
 {
+#ifdef __WXMSW__
+  ::CharLowerBuff(begin(), length());
+#else
   for ( iterator it = begin(), en = end(); it != en; ++it )
     *it = (wxChar)wxTolower(*it);
+#endif
 
   return *this;
 }

 	  	 
