Index: auibook.py
===================================================================
--- auibook.py	(revision 66304)
+++ auibook.py	(working copy)
@@ -1868,6 +1868,7 @@
 
             return
 
+        self.OnLeftUpHack()
         if self.HasCapture():
             self.ReleaseMouse()
     
@@ -1901,6 +1902,14 @@
         self._click_tab = None
 
 
+    def OnLeftUpHack(self):
+        """
+        A hack to force the reseting of the mouse pointer and to hide the hint window
+        """
+        self.GetParent()._mgr.HideHint()
+        self.SetCursor(wx.StockCursor(wx.CURSOR_DEFAULT))
+        
+
     def OnMiddleUp(self, event):
         """
         Handles the ``wx.EVT_MIDDLE_UP`` event for L{AuiTabCtrl}.
