Index: src/generic/choicdgg.cpp
===================================================================
--- src/generic/choicdgg.cpp	(revision 70508)
+++ src/generic/choicdgg.cpp	(working copy)
@@ -222,7 +222,7 @@
                              int initialSelection)
 {
     wxSingleChoiceDialog dialog(parent, message, caption, n, choices,
-                                (char **)client_data);
+                                client_data);
 
     dialog.SetSelection(initialSelection);
 
@@ -472,7 +472,7 @@
                                            const wxString& caption,
                                            int n,
                                            const wxString *choices,
-                                           char **clientData,
+                                           void **clientData,
                                            long style,
                                            const wxPoint& WXUNUSED(pos))
 {
@@ -483,7 +483,7 @@
                                            const wxString& message,
                                            const wxString& caption,
                                            const wxArrayString& choices,
-                                           char **clientData,
+                                           void **clientData,
                                            long style,
                                            const wxPoint& WXUNUSED(pos))
 {
@@ -495,7 +495,7 @@
                                    const wxString& caption,
                                    int n,
                                    const wxString *choices,
-                                   char **clientData,
+                                   void **clientData,
                                    long style,
                                    const wxPoint& pos )
 {
@@ -519,7 +519,7 @@
                                    const wxString& message,
                                    const wxString& caption,
                                    const wxArrayString& choices,
-                                   char **clientData,
+                                   void **clientData,
                                    long style,
                                    const wxPoint& pos )
 {
Index: include/wx/generic/choicdgg.h
===================================================================
--- include/wx/generic/choicdgg.h	(revision 70508)
+++ include/wx/generic/choicdgg.h	(working copy)
@@ -106,14 +106,14 @@
                          const wxString& caption,
                          int n,
                          const wxString *choices,
-                         char **clientData = (char **)NULL,
+                         void **clientData = (void **)NULL,
                          long style = wxCHOICEDLG_STYLE,
                          const wxPoint& pos = wxDefaultPosition);
     wxSingleChoiceDialog(wxWindow *parent,
                          const wxString& message,
                          const wxString& caption,
                          const wxArrayString& choices,
-                         char **clientData = (char **)NULL,
+                         void **clientData = (void **)NULL,
                          long style = wxCHOICEDLG_STYLE,
                          const wxPoint& pos = wxDefaultPosition);
 
@@ -122,14 +122,14 @@
                 const wxString& caption,
                 int n,
                 const wxString *choices,
-                char **clientData = (char **)NULL,
+                void **clientData = (void **)NULL,
                 long style = wxCHOICEDLG_STYLE,
                 const wxPoint& pos = wxDefaultPosition);
     bool Create(wxWindow *parent,
                 const wxString& message,
                 const wxString& caption,
                 const wxArrayString& choices,
-                char **clientData = (char **)NULL,
+                void **clientData = (void **)NULL,
                 long style = wxCHOICEDLG_STYLE,
                 const wxPoint& pos = wxDefaultPosition);
 
