--- wxWidgets-3.1.0_orig/src/ribbon/page.cpp	2016-02-29 07:33:37.000000000 +1000
+++ wxWidgets-3.1.0/src/ribbon/page.cpp	2016-03-15 10:48:14.956331100 +1000
@@ -46,6 +46,7 @@ public:
                  const wxSize& size = wxDefaultSize,
                  long style = 0);
 
+	void DelayDestroy();
     virtual ~wxRibbonPageScrollButton();
 
 protected:
@@ -57,6 +58,7 @@ protected:
     void OnMouseLeave(wxMouseEvent& evt);
     void OnMouseDown(wxMouseEvent& evt);
     void OnMouseUp(wxMouseEvent& evt);
+	void OnClose( wxCloseEvent& evt );
 
     wxRibbonPage* m_sibling;
     long m_flags;
@@ -74,6 +76,7 @@ wxBEGIN_EVENT_TABLE(wxRibbonPageScrollBu
     EVT_LEFT_DOWN(wxRibbonPageScrollButton::OnMouseDown)
     EVT_LEFT_UP(wxRibbonPageScrollButton::OnMouseUp)
     EVT_PAINT(wxRibbonPageScrollButton::OnPaint)
+	EVT_CLOSE( wxRibbonPageScrollButton::OnClose )
 wxEND_EVENT_TABLE()
 
 wxRibbonPageScrollButton::wxRibbonPageScrollButton(wxRibbonPage* sibling,
@@ -91,6 +94,16 @@ wxRibbonPageScrollButton::~wxRibbonPageS
 {
 }
 
+void wxRibbonPageScrollButton::DelayDestroy()
+{
+	AddPendingEvent( wxCloseEvent( wxEVT_CLOSE_WINDOW, m_windowId ) );
+}
+
+void wxRibbonPageScrollButton::OnClose( wxCloseEvent& WXUNUSED(evt) )
+{
+	Destroy();
+}
+
 void wxRibbonPageScrollButton::OnEraseBackground(wxEraseEvent& WXUNUSED(evt))
 {
     // Do nothing - all painting done in main paint handler
@@ -831,7 +849,7 @@ void wxRibbonPage::ShowScrollButtons()
     {
         if(m_scroll_left_btn != NULL)
         {
-            m_scroll_left_btn->Destroy();
+			m_scroll_left_btn->DelayDestroy();
             m_scroll_left_btn = NULL;
             reposition = true;
         }
@@ -873,7 +891,7 @@ void wxRibbonPage::ShowScrollButtons()
     {
         if(m_scroll_right_btn != NULL)
         {
-            m_scroll_right_btn->Destroy();
+			m_scroll_right_btn->DelayDestroy();
             m_scroll_right_btn = NULL;
             reposition = true;
         }
