Index: src/msw/toplevel.cpp
===================================================================
--- src/msw/toplevel.cpp	(revision 54593)
+++ src/msw/toplevel.cpp	(working copy)
@@ -545,8 +545,16 @@
 
         // reuse the code in MSWGetStyle() but correct the results slightly for
         // the dialog
-        dlgTemplate->style = MSWGetStyle(style, &dlgTemplate->dwExtendedStyle);
 
+        // dwExtendedStyle is not aligned correctly for x64 architecture
+        // this gives a warning C4328 on Visual Studio 2005 and may, on some
+        // systems cause a runtime exception
+        // copy the value to a temporary 'extendedStyle', pass the pointer
+        WXDWORD extendedStyle = dlgTemplate->dwExtendedStyle;
+        dlgTemplate->style = MSWGetStyle(style, &extendedStyle);
+        // then copy the result back to dwExtendedStyle
+        dlgTemplate->dwExtendedStyle = extendedStyle;
+
         // all dialogs are popups
         dlgTemplate->style |= WS_POPUP;
 
