Index: src/msw/ole/dataobj.cpp
===================================================================
--- src/msw/ole/dataobj.cpp	(revision 61826)
+++ src/msw/ole/dataobj.cpp	(working copy)
@@ -1245,7 +1245,7 @@
         wxCharBuffer buf(len);
 
         if ( m_dataObjectLast->GetDataHere(buf.data()) )
-            url = buf;
+            url = wxString(buf, wxConvLibc);
 #else // !wxUSE_UNICODE
         // in ANSI build no conversion is necessary
         m_dataObjectLast->GetDataHere(wxStringBuffer(url, len));
