Index: interface/wx/sizer.h
===================================================================
--- interface/wx/sizer.h	(revision 61629)
+++ interface/wx/sizer.h	(working copy)
@@ -568,7 +568,7 @@
 
     /**
         Inserts non-stretchable space to the sizer.
-        More readable way of calling wxSizer::Insert(size, size, 0).
+        More readable way of calling wxSizer::Insert(index, size, size).
     */
     wxSizerItem* InsertSpacer(size_t index, int size);
 
