 samples/splitter/splitter.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/samples/splitter/splitter.cpp b/samples/splitter/splitter.cpp
index 1eae2c9..5c202eb 100644
--- a/samples/splitter/splitter.cpp
+++ b/samples/splitter/splitter.cpp
@@ -291,6 +291,7 @@ MyFrame::MyFrame()
     // zero gravity were used (although it would do no harm neither).
     m_splitter->SetSize(GetClientSize());
     m_splitter->SetSashGravity(1.0);
+    m_splitter->SetMinimumPaneSize(50);
 
 #if 1
     m_left = new MyCanvas(m_splitter, true);
@@ -550,7 +551,7 @@ void MySplitterWindow::OnUnsplitEvent(wxSplitterEvent& event)
 // ----------------------------------------------------------------------------
 
 MyCanvas::MyCanvas(wxWindow* parent, bool mirror)
-        : wxScrolledWindow(parent, wxID_ANY, wxDefaultPosition, wxDefaultSize,
+        : wxScrolledWindow(parent, wxID_ANY, wxDefaultPosition, wxSize(150,400),
                            wxHSCROLL | wxVSCROLL | wxNO_FULL_REPAINT_ON_RESIZE)
 {
     m_mirror = mirror;
