Index: src/msw/treectrl.cpp
===================================================================
--- src/msw/treectrl.cpp	(revision 66360)
+++ src/msw/treectrl.cpp	(working copy)
@@ -1770,15 +1770,6 @@
     tvItem.state = 0;
     DoSetItem(&tvItem);
 
-    if ( IsExpanded(item) )
-    {
-        wxTreeEvent event(wxEVT_COMMAND_TREE_ITEM_COLLAPSING,
-                          this, wxTreeItemId(item));
-
-        if ( !IsTreeEventAllowed(event) )
-            return;
-    }
-
     if ( TreeView_Expand(GetHwnd(), HITEM(item), flag) )
     {
         if ( IsExpanded(item) )
