Index: common/fldlgcmn.cpp
===================================================================
--- common/fldlgcmn.cpp	(revision 64376)
+++ common/fldlgcmn.cpp	(working copy)
@@ -75,13 +75,13 @@
     wxASSERT_MSG( !HasFdFlag(wxFD_OPEN) || !HasFdFlag(wxFD_OVERWRITE_PROMPT),
                   wxT("wxFD_OVERWRITE_PROMPT can't be used with wxFD_OPEN") );
 
-    if ( wildCard.empty() || wildCard == wxFileSelectorDefaultWildcardStr )
+    if ( wildCard == wxFileSelectorDefaultWildcardStr )
     {
         m_wildCard = wxString::Format(_("All files (%s)|%s"),
                                       wxFileSelectorDefaultWildcardStr,
                                       wxFileSelectorDefaultWildcardStr);
     }
-    else // have wild card
+    else if ( !wildCard.empty() ) // have wild card
     {
         // convert m_wildCard from "*.bar" to "bar files (*.bar)|*.bar"
         if ( m_wildCard.Find(wxT('|')) == wxNOT_FOUND )
