Index: include/wx/docview.h
===================================================================
--- include/wx/docview.h	(revision 51921)
+++ include/wx/docview.h	(working copy)
@@ -423,7 +423,7 @@
     virtual void FileHistoryUseMenu(wxMenu *menu);
     virtual void FileHistoryRemoveMenu(wxMenu *menu);
 #if wxUSE_CONFIG
-    virtual void FileHistoryLoad(wxConfigBase& config);
+    virtual void FileHistoryLoad(const wxConfigBase& config);
     virtual void FileHistorySave(wxConfigBase& config);
 #endif // wxUSE_CONFIG
 
@@ -601,7 +601,7 @@
     virtual void RemoveMenu(wxMenu *menu);
 
 #if wxUSE_CONFIG
-    virtual void Load(wxConfigBase& config);
+    virtual void Load(const wxConfigBase& config);
     virtual void Save(wxConfigBase& config);
 #endif // wxUSE_CONFIG
 
Index: src/common/docview.cpp
===================================================================
--- src/common/docview.cpp	(revision 51921)
+++ src/common/docview.cpp	(working copy)
@@ -1509,7 +1509,7 @@
 }
 
 #if wxUSE_CONFIG
-void wxDocManager::FileHistoryLoad(wxConfigBase& config)
+void wxDocManager::FileHistoryLoad(const wxConfigBase& config)
 {
     if (m_fileHistory)
         m_fileHistory->Load(config);
@@ -2275,7 +2275,7 @@
 }
 
 #if wxUSE_CONFIG
-void wxFileHistory::Load(wxConfigBase& config)
+void wxFileHistory::Load(const wxConfigBase& config)
 {
     m_fileHistory.Clear();
 

 	  	 
