Index: colordlg.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/mac/carbon/colordlg.cpp,v
retrieving revision 1.19
diff -u -r1.19 colordlg.cpp
--- colordlg.cpp	2006/04/29 19:19:58	1.19
+++ colordlg.cpp	2006/05/14 16:33:48
@@ -80,7 +80,8 @@
 
     if (GetColor( where, "\pSelect a new palette color.", &currentColor, &newColor ))
     {
-        m_colourData.m_dataColour.Set( (WXCOLORREF*) &newColor ) ;
+        //m_colourData.m_dataColour.Set( (WXCOLORREF*) &newColor ) ;
+        m_colourData.m_dataColour.Set(newColor.red,newColor.green,newColor.blue);
         return wxID_OK;
     }
 #endif

 	  	 
