Index: src/common/docview.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/common/docview.cpp,v
retrieving revision 1.127
diff -u -r1.127 docview.cpp
--- src/common/docview.cpp	2004/10/31 23:43:35	1.127
+++ src/common/docview.cpp	2005/01/21 16:51:17
@@ -908,6 +908,7 @@
     if (!CloseDocuments(force))
         return false;
 
+		m_currentView = NULL;
     wxList::compatibility_iterator node = m_templates.GetFirst();
     while (node)
     {

 	  	 
