Index: minimal.cpp
===================================================================
--- minimal.cpp	(revision 71959)
+++ minimal.cpp	(working copy)
@@ -30,6 +30,8 @@
     #include "wx/wx.h"
 #endif
 
+#include "wx/listbook.h"
+
 // ----------------------------------------------------------------------------
 // resources
 // ----------------------------------------------------------------------------
@@ -172,6 +174,29 @@
     CreateStatusBar(2);
     SetStatusText("Welcome to wxWidgets!");
 #endif // wxUSE_STATUSBAR
+    
+    wxListbook* listBook = new wxListbook(this, wxID_ANY);
+    
+    // page 1
+    wxPanel* panel1 = new wxPanel(listBook, wxID_ANY);
+    wxTextCtrl* textCtrl1 = new wxTextCtrl(panel1, wxID_ANY);
+    wxBoxSizer* boxSizer1 = new wxBoxSizer(wxVERTICAL);
+    boxSizer1->Add(textCtrl1, 1, wxEXPAND | wxALL, 10);
+    panel1->SetSizer(boxSizer1);
+
+    // page 2
+    wxPanel* panel2 = new wxPanel(listBook, wxID_ANY);
+    wxTextCtrl* textCtrl2 = new wxTextCtrl(panel2, wxID_ANY);
+    wxBoxSizer* boxSizer2 = new wxBoxSizer(wxVERTICAL);
+    boxSizer2->Add(textCtrl2, 1, wxEXPAND | wxALL, 10);
+    panel2->SetSizer(boxSizer2);
+
+    listBook->AddPage(panel1, wxT("Page 1"));
+    listBook->AddPage(panel2, wxT("Page 2"));
+
+    wxBoxSizer* mainSizer = new wxBoxSizer(wxVERTICAL);
+    mainSizer->Add(listBook, 1, wxEXPAND);
+    SetSizer(mainSizer);
 }
 
 
