diff -U 3 -H -d -r -N -a -- wxGTK-2.5.2/src/generic/choicdgg.cpp wxGTK-2.5.2_dbg/src/generic/choicdgg.cpp
--- wxGTK-2.5.2/src/generic/choicdgg.cpp	2004-05-31 13:11:59.000000000 +0200
+++ wxGTK-2.5.2_dbg/src/generic/choicdgg.cpp	2004-06-17 01:03:47.000000000 +0200
@@ -434,6 +434,10 @@
 
 void wxMultiChoiceDialog::SetSelections(const wxArrayInt& selections)
 {
+    for ( int n = 0; n < m_listbox->GetCount(); ++n )
+    {//first clear all currently selected items
+        m_listbox->Deselect(n);
+    }
     size_t count = selections.GetCount();
     for ( size_t n = 0; n < count; n++ )
     {

 	  	 
