 samples/text/text.cpp | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/samples/text/text.cpp b/samples/text/text.cpp
index 70c3cfe..eca3e7e 100644
--- a/samples/text/text.cpp
+++ b/samples/text/text.cpp
@@ -946,6 +946,13 @@ void MyTextCtrl::OnKeyDown(wxKeyEvent& event)
 {
     switch ( event.GetKeyCode() )
     {
+        case WXK_RETURN:
+        case WXK_NUMPAD_ENTER:
+            {
+                wxMessageDialog dlg(this, "Newline unwanted but still added...");
+                dlg.ShowModal();
+                return;
+            }
         case WXK_F1:
             // show current position and text length
             {
