### Eclipse Workspace Patch 1.0
#P wxWidgets
Index: include/wx/cocoa/checkbox.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/cocoa/checkbox.h,v
retrieving revision 1.6
diff -u -r1.6 checkbox.h
--- include/wx/cocoa/checkbox.h	2005/11/08 16:02:04	1.6
+++ include/wx/cocoa/checkbox.h	2007/03/12 20:30:14
@@ -2,7 +2,7 @@
 // Name:        wx/cocoa/checkbox.h
 // Purpose:     wxCheckBox class
 // Author:      David Elliott
-// Modified by:
+// Modified by:	Brookes Computing Society
 // Created:     2003/03/16
 // RCS-ID:      $Id: checkbox.h,v 1.6 2005/11/08 16:02:04 ABX Exp $
 // Copyright:   (c) 2003 David Elliott
@@ -59,6 +59,7 @@
     virtual void SetValue(bool);
     virtual bool GetValue() const;
     virtual void SetLabel(const wxString& label);
+	virtual wxString GetLabel() const;
 protected:
     virtual void DoSet3StateValue(wxCheckBoxState state);
     virtual wxCheckBoxState DoGet3StateValue() const;

 	  	 
