### Eclipse Workspace Patch 1.0
#P wxWidgets
Index: include/wx/cocoa/stattext.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/cocoa/stattext.h,v
retrieving revision 1.4
diff -u -r1.4 stattext.h
--- include/wx/cocoa/stattext.h	2006/10/17 14:44:18	1.4
+++ include/wx/cocoa/stattext.h	2007/03/12 21:58:13
@@ -2,7 +2,7 @@
 // Name:        wx/cocoa/stattext.h
 // Purpose:     wxStaticText class
 // Author:      David Elliott
-// Modified by:
+// Modified by:	Brookes Computing Society
 // Created:     2003/02/15
 // RCS-ID:      $Id: stattext.h,v 1.4 2006/10/17 14:44:18 ABX Exp $
 // Copyright:   (c) 2003 David Elliott
@@ -53,6 +53,7 @@
 // ------------------------------------------------------------------------
 public:
     void SetLabel(const wxString& label);
+	wxString GetLabel() const;
 };
 
 #endif

 	  	 
