diff -r -u -p a/samples/propgrid/propgrid.cpp b/samples/propgrid/propgrid.cpp
--- a/samples/propgrid/propgrid.cpp	2018-12-09 16:40:08.000000000 +0100
+++ b/samples/propgrid/propgrid.cpp	2019-01-22 07:31:21.526858500 +0100
@@ -2571,7 +2571,7 @@ FormMain::OnSetBackgroundColour( wxComma
 
     if ( col.IsOk() )
     {
-        bool recursively = (event.GetId()==ID_SETBGCOLOURRECUR) ? true : false;
+        int recursively = (event.GetId()==ID_SETBGCOLOURRECUR) ? wxPG_RECURSE : 0;
         pg->SetPropertyBackgroundColour(prop, col, recursively);
     }
 }
