Index: tags/WX_3_0_1/src/msw/window.cpp
===================================================================
--- tags/WX_3_0_1/src/msw/window.cpp	(revision 76913)
+++ tags/WX_3_0_1/src/msw/window.cpp	(working copy)
@@ -4854,6 +4854,22 @@
     eventNc.SetEventObject(this);
     HandleWindowEvent(eventNc);
 
+    if( !processed && wxDidCreatePaintDC )
+    {
+        // We get here in the case when ::BeginPaint() has been called
+        // (though accessing DC) and event.Skip() has been set.
+        // We are to call default Windows painting but we cannot do it by
+        // returning false from this function because after ::EndPaint()
+        // which is called at the end of this function the call to default
+        // painting will be useless.
+        // We need to call it now and pass current HDC as a parameter
+        // so that ::BeginPaint()/::EndPaint() would not be called from there
+        // but the current HDC would be used.
+        // It is exactly what OnPaint() does.
+        OnPaint(event);
+        processed = true;
+    }
+
     // don't keep an HRGN we don't need any longer (GetUpdateRegion() can only
     // be called from inside the event handlers called above)
     m_updateRegion.Clear();
