Index: ../wxWidgets/src/common/utilscmn.cpp
===================================================================
--- ../wxWidgets/src/common/utilscmn.cpp	(revision 66880)
+++ ../wxWidgets/src/common/utilscmn.cpp	(working copy)
@@ -580,6 +580,7 @@
 #else // non-MSVC non-Mac
     // Not sure if other compilers have _tenviron so use the (more standard)
     // ANSI version only for them.
+    extern char **environ;
     char **env = environ;
 #endif
 
