Index: src/common/datavcmn.cpp
===================================================================
--- src/common/datavcmn.cpp	(revision 63555)
+++ src/common/datavcmn.cpp	(working copy)
@@ -1486,6 +1486,7 @@
 void wxDataViewListStore::DeleteItem( unsigned int row )
 {
     wxVector<wxDataViewListStoreLine*>::iterator it = m_data.begin() + row;
+    delete *it;
     m_data.erase( it );
 
     RowDeleted( row );
