Index: include/wx/dir.h
===================================================================
--- include/wx/dir.h	(revision 72185)
+++ include/wx/dir.h	(working copy)
@@ -100,6 +100,9 @@
     // open the directory for enumerating
     bool Open(const wxString& dir);
 
+    // close the directory
+    void Close();
+
     // returns true if the directory was successfully opened
     bool IsOpened() const;
 
Index: interface/wx/dir.h
===================================================================
--- interface/wx/dir.h	(revision 72185)
+++ interface/wx/dir.h	(working copy)
@@ -166,6 +166,16 @@
     ~wxDir();
 
     /**
+        Close the directory.
+
+        Although the directory is closed after calling this function it can be
+        opened again using Open().
+
+        @since 2.9.5
+    */
+    void Close();
+
+    /**
         Test for existence of a directory with the given name.
     */
     static bool Exists(const wxString& dir);
Index: src/msw/dir.cpp
===================================================================
--- src/msw/dir.cpp	(revision 72185)
+++ src/msw/dir.cpp	(working copy)
@@ -316,6 +316,15 @@
     }
 }
 
+void wxDir::Close()
+{
+    if (m_data != NULL) 
+    {
+        delete m_data;
+        m_data = NULL;
+    }
+}
+
 bool wxDir::IsOpened() const
 {
     return m_data != NULL;
@@ -345,7 +354,7 @@
 
 wxDir::~wxDir()
 {
-    delete M_DIR;
+    Close();
 }
 
 // ----------------------------------------------------------------------------
Index: src/unix/dir.cpp
===================================================================
--- src/unix/dir.cpp	(revision 72185)
+++ src/unix/dir.cpp	(working copy)
@@ -224,6 +224,15 @@
     return true;
 }
 
+void wxDir::Close()
+{
+    if (m_data != NULL) 
+    {
+        delete m_data;
+        m_data = NULL;
+    }
+}
+
 bool wxDir::IsOpened() const
 {
     return m_data != NULL;
@@ -250,7 +259,7 @@
 
 wxDir::~wxDir()
 {
-    delete M_DIR;
+    Close();
 }
 
 // ----------------------------------------------------------------------------
