--- xmlres.cpp.org	2005-08-12 15:09:24.000000000 +0200
+++ xmlres.cpp	2006-01-27 16:32:08.000000000 +0100
@@ -1327,7 +1327,10 @@
 
 
     wxString facename;
+
+#if wxUSE_FONTMAP
     bool hasFacename = HasParam(wxT("face"));
+
     if (hasFacename)
     {
         wxString faces = GetParamValue(wxT("face"));
@@ -1344,11 +1347,15 @@
             }
         }
     }
+#endif
 
     // encoding
     wxFontEncoding enc = wxFONTENCODING_DEFAULT;
+
+#if wxUSE_FONTMAP  
     bool hasEncoding = HasParam(wxT("encoding"));
-    if (hasEncoding)
+
+  if (hasEncoding)
     {
         wxString encoding = GetParamValue(wxT("encoding"));
         wxFontMapper mapper;
@@ -1358,6 +1365,8 @@
             enc = wxFONTENCODING_DEFAULT;
     }
 
+#endif
+
     // is this font based on a system font?
     wxFont sysfont = GetSystemFont(GetParamValue(wxT("sysfont")));
 
@@ -1377,11 +1386,12 @@
             sysfont.SetUnderlined(underlined);
         if (hasFamily)
             sysfont.SetFamily(ifamily);
+#if wxUSE_FONTMAP
         if (hasFacename)
             sysfont.SetFaceName(facename);
         if (hasEncoding)
             sysfont.SetDefaultEncoding(enc);
-
+#endif
         m_node = oldnode;
         return sysfont;
     }

 	  	 
