Index: uris.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/tests/uris/uris.cpp,v
retrieving revision 1.13
diff -b -u -2 -r1.13 uris.cpp
--- uris.cpp	2005/06/17 11:27:43	1.13
+++ uris.cpp	2005/10/27 21:24:59
@@ -50,4 +50,5 @@
         CPPUNIT_TEST( Comparison );
         CPPUNIT_TEST( Unescaping );
+        CPPUNIT_TEST( FileBug );
 #if TEST_URL
         CPPUNIT_TEST( URLCompat );
@@ -69,4 +70,5 @@
     void Comparison();
     void Unescaping();
+    void FileBug();
 
 #if TEST_URL
@@ -385,4 +387,10 @@
 }
 #endif // wxUSE_PROTOCOL_HTTP
+
+void URITestCase::FileBug()
+{
+    wxURI uri(wxT("file://e:/wxcode/components/webupdate/website/script1.xml"));
+    CPPUNIT_ASSERT(uri.GetPath() == wxT("e:/wxcode/components/webupdate/website/script1.xml"));
+}
 
 #endif // TEST_URL

 	  	 
