Index: samples/dataview/dataview.cpp
===================================================================
--- samples/dataview/dataview.cpp	(revision 75726)
+++ samples/dataview/dataview.cpp	(working copy)
@@ -686,13 +686,19 @@
             tc->AppendItem( parent, "Child 1", 0 );
             tc->AppendItem( parent, "Child 2", 0 );
             tc->AppendItem( parent, "Child 3, very long, long, long, long", 0 );
+            tc->Freeze();
+            for (unsigned idx=0; idx<2000; ++idx)
+                tc->AppendItem( parent, wxString::Format("Child 4, very repeated %d", idx), 0 );
 
             wxDataViewItem cont =
                 tc->AppendContainer( parent, "Container child", 0 );
             tc->AppendItem( cont, "Child 4", 0 );
             tc->AppendItem( cont, "Child 5", 0 );
+            for (unsigned idx=0; idx<2000; ++idx)
+                tc->AppendItem( cont, wxString::Format("Child 5, very repeated %d", idx), 0 );
 
             tc->Expand(cont);
+            tc->Thaw();
         }
         break;
     }
