Index: ScintillaWX.cpp
===================================================================
--- ScintillaWX.cpp	(revision 72481)
+++ ScintillaWX.cpp	(working copy)
@@ -749,7 +749,8 @@
     if (paintState == paintAbandoned) {
         // Painting area was insufficient to cover new styling or brace
         // highlight positions
-        FullPaintDC(dc);
+        //FullPaintDC(dc);
+		FullPaint();
     }
     paintState = notPainting;
 }
@@ -757,8 +758,10 @@
 
 // Force the whole window to be repainted
 void ScintillaWX::FullPaint() {
-    wxClientDC dc(stc);
-    FullPaintDC(&dc);
+    //wxClientDC dc(stc);
+    //FullPaintDC(&dc);
+    stc->Refresh(false);
+    stc->Update();
 }
 
 
