Index: src/mac/carbon/app.cpp
===================================================================
--- src/mac/carbon/app.cpp	(revision 48581)
+++ src/mac/carbon/app.cpp	(working copy)
@@ -1731,7 +1731,30 @@
             break;
         } // end switch
     }
-
+    else
+    {
+        switch (keycode)
+        {
+            case 24:    // + key
+                keyval = WXK_ADD;
+                break;
+            case 27:    // - key
+                keyval = WXK_SUBTRACT;
+                break;
+            case 28:    // * key
+                keyval = WXK_MULTIPLY;
+                break;
+            case 44:    // / key
+                keyval = WXK_DIVIDE;
+                break;
+            case 47:    //. key
+                keyval = WXK_DECIMAL;
+                break;
+            default:
+                break;
+        }
+    }
+    
     event.m_shiftDown = modifiers & shiftKey;
     event.m_controlDown = modifiers & controlKey;
     event.m_altDown = modifiers & optionKey;

 	  	 
