--- wxWidgets-2.8.2-orig/src/mac/carbon/cursor.cpp	2007-05-09 12:20:47.000000000 +0300
+++ wxWidgets-2.8.2/src/mac/carbon/cursor.cpp	2007-05-09 12:19:36.000000000 +0300
@@ -520,4 +520,8 @@
         break;
 
+    case wxCURSOR_OPEN_HAND:
+        M_CURSORDATA->m_themeCursor = kThemeOpenHandCursor;
+        break;
+
     case wxCURSOR_WAIT:
         M_CURSORDATA->m_themeCursor = kThemeWatchCursor;

 	  	 
