Index: src/msw/listctrl.cpp
===================================================================
--- src/msw/listctrl.cpp	(revision 65496)
+++ src/msw/listctrl.cpp	(working copy)
@@ -2237,22 +2237,6 @@
                 eventType = wxEVT_COMMAND_LIST_DELETE_ITEM;
                 event.m_itemIndex = iItem;
 
-                // delete the associated internal data
-                if ( wxMSWListItemData *data = MSWGetItemData(iItem) )
-                {
-                    const unsigned count = m_internalData.size();
-                    for ( unsigned n = 0; n < count; n++ )
-                    {
-                        if ( m_internalData[n] == data )
-                        {
-                            m_internalData.erase(m_internalData.begin() + n);
-                            wxDELETE(data);
-                            break;
-                        }
-                    }
-
-                    wxASSERT_MSG( !data, "invalid internal data pointer?" );
-                }
                 break;
 
             case LVN_INSERTITEM:
@@ -2603,6 +2587,26 @@
             m_count = 0;
             return true;
 
+		case LVN_DELETEITEM:
+			// delete the associated internal data after the event has been handled
+			if ( wxMSWListItemData *data = MSWGetItemData(event.m_itemIndex) )
+			{
+				const unsigned count = m_internalData.size();
+				for ( unsigned n = 0; n < count; n++ )
+				{
+					if ( m_internalData[n] == data )
+					{
+						m_internalData.erase(m_internalData.begin() + n);
+						wxDELETE(data);
+						break;
+					}
+				}
+
+				wxASSERT_MSG( !data, "invalid internal data pointer?" );
+			}
+			break;
+
+
         case LVN_ENDLABELEDITA:
         case LVN_ENDLABELEDITW:
             // logic here is inverted compared to all the other messages
