diff -bru wxWidgets/samples/mdi/mdi.cpp /mnt/winxp/wxWidgets/samples/mdi/mdi.cpp
--- wxWidgets/samples/mdi/mdi.cpp	2012-11-27 11:39:13.000000000 -0800
+++ /mnt/winxp/wxWidgets/samples/mdi/mdi.cpp	2012-12-06 12:40:05.656250000 -0800
@@ -262,6 +262,7 @@
 {
     // create and show another child frame
     MyChild *subframe = new MyChild(this);
+    subframe->SetSizeHints( 100, 100, 500, 500 );
     subframe->Show(true);
 }
 
diff -bru wxWidgets/src/msw/mdi.cpp /mnt/winxp/wxWidgets/src/msw/mdi.cpp
--- wxWidgets/src/msw/mdi.cpp	2012-11-27 11:34:43.000000000 -0800
+++ /mnt/winxp/wxWidgets/src/msw/mdi.cpp	2012-12-06 15:03:39.625000000 -0800
@@ -1179,25 +1179,7 @@
     // frames because it is based on the size of the MDI client window,
     // not on the values specified in wxWindow m_max variables
     bool processed = MSWDefWindowProc(WM_GETMINMAXINFO, 0, (LPARAM)mmInfo) != 0;
-
-    int minWidth = GetMinWidth(),
-        minHeight = GetMinHeight();
-
-    // but allow GetSizeHints() to set the min size
-    if ( minWidth != wxDefaultCoord )
-    {
-        info->ptMinTrackSize.x = minWidth;
-
-        processed = true;
-    }
-
-    if ( minHeight != wxDefaultCoord )
-    {
-        info->ptMinTrackSize.y = minHeight;
-
-        processed = true;
-    }
-
+    processed = wxWindowMSW::HandleGetMinMaxInfo( mmInfo );
     return processed;
 }
 
