Index: src/xrc/xh_propdlg.cpp
===================================================================
--- src/xrc/xh_propdlg.cpp	(revision 48077)
+++ src/xrc/xh_propdlg.cpp	(working copy)
@@ -26,6 +26,8 @@
 
 #include "wx/bookctrl.h"
 #include "wx/propdlg.h"
+#include "wx/imaglist.h"
+#include "wx/frame.h"
 
 IMPLEMENT_DYNAMIC_CLASS(wxPropertySheetDialogXmlHandler, wxXmlResourceHandler)
 

 	  	 
