--- C:/Users/kev/Desktop/minimal.cpp	Sun Jul 19 19:37:48 2015
+++ C:/Users/kev/Downloads/wxWidgets-master/samples/minimal/minimal.cpp	Sun Jul 19 19:28:44 2015
@@ -18,6 +18,8 @@
 
 // For compilers that support precompilation, includes "wx/wx.h".
 #include "wx/wxprec.h"
+#include <wx/filesys.h>
+#include <wx/fs_zip.h>
 
 #ifdef __BORLANDC__
     #pragma hdrstop
@@ -129,6 +131,32 @@
     // and show it (the frames, unlike simple controls, are not shown when
     // created initially)
     frame->Show(true);
+
+	wxFileSystem::AddHandler(new wxZipFSHandler);
+
+	const wxChar *sTests[] =
+	{
+		wxT("c:\\archive\\test%231.txt"),					// success
+		wxT("c:\\archive\\test%231.zip#zip:test.txt"),		// success
+		wxT("c:\\archive\\test%231.zip#zip:test%231.txt")	// fails here
+	};
+
+	wxFileSystem* fileSystem = new wxFileSystem;
+
+	for (int i = 0; i < sizeof(sTests) / sizeof(sTests[0]); i++)
+	{
+		// will fail if there is no extension
+		wxFSFile* file = fileSystem->OpenFile(sTests[i]);
+		if (!file)
+		{
+			printf("Failed to open! ");
+		}
+		else
+		{
+			delete file;
+		}
+	}
+
 
     // success: wxApp::OnRun() will be called which will enter the main message
     // loop and the application will run. If we returned false here, the
