diff -bu1 -r wxWindows/contrib/samples/stc/edit.cpp wxWin/contrib/samples/stc/edit.cpp
--- wxWindows/contrib/samples/stc/edit.cpp	2003-09-19 09:25:54.000000000 +0200
+++ wxWin/contrib/samples/stc/edit.cpp	2003-09-19 13:18:58.000000000 +0200
@@ -363,3 +363,3 @@
 
-    LanguageInfo const* curInfo = NULL;
+    LanguageInfo const* curInfo;
 
diff -bu1 -r wxWindows/contrib/samples/stc/stctest.bkl wxWin/contrib/samples/stc/stctest.bkl
--- wxWindows/contrib/samples/stc/stctest.bkl	2003-09-18 09:53:36.000000000 +0200
+++ wxWin/contrib/samples/stc/stctest.bkl	2003-09-19 13:32:22.000000000 +0200
@@ -17,2 +17,8 @@
 
+    <wx-data id="data">   
+        <files>   
+             stctest.cpp   
+        </files>   
+    </wx-data>   
+  
 </makefile>
diff -bu1 -r wxWindows/contrib/samples/stc/stctest.cpp wxWin/contrib/samples/stc/stctest.cpp
--- wxWindows/contrib/samples/stc/stctest.cpp	2003-09-19 09:25:54.000000000 +0200
+++ wxWin/contrib/samples/stc/stctest.cpp	2003-09-19 13:35:17.000000000 +0200
@@ -56,3 +56,3 @@
 #define APP_NAME _T("STC-Test")
-#define APP_DESCR _("See http://wxguide.sourceforge.net/indexedit.html")
+#define APP_DESCR _("See http://wxguide.sourceforge.net/")
 
@@ -291,3 +291,3 @@
 AppFrame::AppFrame (const wxString &title)
-        : wxFrame ((wxFrame *)NULL, -1, title, wxDefaultPosition, wxSize(600,400),
+        : wxFrame ((wxFrame *)NULL, -1, title, wxDefaultPosition, wxDefaultSize,
                     wxDEFAULT_FRAME_STYLE | wxNO_FULL_REPAINT_ON_RESIZE) {
@@ -313,2 +313,3 @@
 
+    FileOpen (_T("stctest.cpp"));
 }
diff -bu1 -r wxWindows/contrib/src/stc/PlatWX.cpp wxWin/contrib/src/stc/PlatWX.cpp
--- wxWindows/contrib/src/stc/PlatWX.cpp	2003-09-18 09:53:36.000000000 +0200
+++ wxWin/contrib/src/stc/PlatWX.cpp	2003-09-19 13:24:58.000000000 +0200
@@ -677,2 +677,3 @@
         cursorId = wxCURSOR_HAND;
+        break;
     default:
@@ -890,3 +891,3 @@
     int maxw = maxStrWidth;
-    int maxh = 0;
+    int maxh ; // not initialized because always value is always assigned
 
diff -bu1 -r wxWindows/contrib/src/stc/stc.cpp wxWin/contrib/src/stc/stc.cpp
--- wxWindows/contrib/src/stc/stc.cpp	2003-09-19 09:26:04.000000000 +0200
+++ wxWin/contrib/src/stc/stc.cpp	2003-09-19 10:44:14.000000000 +0200
@@ -2144,3 +2144,3 @@
             buffer.UngetWriteBuf(len);
-#if #wxUSE_UNICODE
+#if wxUSE_UNICODE
             contents = wxString(buffer, *wxConvCurrent);

 	  	 
