Index: samples/image/image.cpp
===================================================================
--- samples/image/image.cpp	(revision 54693)
+++ samples/image/image.cpp	(working copy)
@@ -192,6 +192,7 @@
     {
         wxPaintDC dc(this);
         const wxSize size = GetClientSize();
+        dc.Clear();
         dc.DrawBitmap(m_bitmap,
                       (size.x - m_bitmap.GetWidth())/2,
                       (size.y - m_bitmap.GetHeight())/2,
@@ -1202,12 +1203,66 @@
     return filename;
 }
 
+static const char * back_xpm[] = {
+"16 15 4 1",
+" 	c None",
+".	c #000000",
+"+	c #C0E4CB",
+"@	c #77C490",
+"                ",
+"                ",
+"       .        ",
+"      ..        ",
+"     .+.        ",
+"    .++........ ",
+"   .++@+++++++. ",
+"  .++@@@@@@@@@. ",
+"   .+@@@@@@@@@. ",
+"    .+@........ ",
+"     .+.        ",
+"      ..        ",
+"       .        ",
+"                ",
+"                "};
+
 void MyFrame::OnNewFrame( wxCommandEvent &WXUNUSED(event) )
 {
-    wxImage image;
-    wxString filename = LoadUserImage(image);
-    if ( !filename.empty() )
-        (new MyImageFrame(this, filename, wxBitmap(image)))->Show();
+    wxImage image = wxBitmap(back_xpm).ConvertToImage();
+    wxASSERT(image.HasMask()); // mask from xpm
+
+    unsigned char mask_r,mask_g,mask_b;
+    image.GetOrFindMaskColour(&mask_r, &mask_g, &mask_b);
+
+    wxColour colorsample;
+
+    const wxSize topleft(0, 0);
+    colorsample = wxColour(image.GetRed(topleft.x, topleft.y), image.GetGreen(topleft.x, topleft.y), image.GetBlue(topleft.x, topleft.y));
+    wxASSERT(mask_r == colorsample.Red());
+    wxASSERT(mask_g == colorsample.Green());
+    wxASSERT(mask_b == colorsample.Blue());
+
+    if (wxYES == wxMessageBox(wxT("Quantize ?"), wxMessageBoxCaptionStr, wxYES_NO | wxICON_QUESTION))
+    {
+       wxImage copy;
+       wxPalette* pal;
+       if (wxQuantize::Quantize(image, copy, &pal, 256, 0, wxQUANTIZE_FILL_DESTINATION_IMAGE))
+       {
+           wxASSERT(!copy.HasMask()); // wxQuantize do not preserve our mask or produce a new one
+           wxASSERT(wxNOT_FOUND != pal->GetPixel(mask_r,mask_g,mask_b)); // but our mask colour is in the new palette
+           
+           colorsample = wxColour(copy.GetRed(topleft.x, topleft.y), copy.GetGreen(topleft.x, topleft.y), copy.GetBlue(topleft.x, topleft.y));
+           wxASSERT(mask_r == colorsample.Red());   // fails - wxQuantize changed this color slightly
+           wxASSERT(mask_g == colorsample.Green()); // fails - wxQuantize changed this color slightly
+           wxASSERT(mask_b == colorsample.Blue());  // fails - wxQuantize changed this color slightly
+
+           image = copy;
+           image.SetMaskColour(mask_r,mask_g,mask_b); // restore the mask - to no avail - (255, 0, 255) no longer in image :(
+           image.SetPalette(*pal);   // if you want to hang on to the palette
+           delete pal;
+       }
+    }
+    wxASSERT(image.HasMask());
+    (new MyImageFrame(this, wxT("back.xpm"), wxBitmap(image)))->Show();
 }
 
 void MyFrame::OnImageInfo( wxCommandEvent &WXUNUSED(event) )
