--- f:\wx280\src\common\intl.cpp	Sun Jan 21 19:35:05 2007
+++ intl.cpp	Sun Jan 21 19:34:50 2007
@@ -2506,5 +2506,6 @@
         if ( ms_languagesDB->Item(i).Language == lang )
         {
-            return &ms_languagesDB->Item(i);
+            // We need to create a temporary here in order to make this work with BCC in final build mode
+            wxLanguageInfo *ptr = &ms_languagesDB->Item(i);
+            return ptr;
         }
     }

 	  	 
