Index: dir.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/msw/dir.cpp,v
retrieving revision 1.19
diff -u -2 -r1.19 dir.cpp
--- dir.cpp	2005/09/23 12:54:56	1.19
+++ dir.cpp	2006/05/04 14:43:36
@@ -197,5 +197,6 @@
         DWORD err = ::GetLastError();
 
-        if ( err != ERROR_FILE_NOT_FOUND )
+        if ( err != ERROR_FILE_NOT_FOUND &&
+             err != ERROR_NO_MORE_FILES )
         {
             wxLogSysError(err, _("Can not enumerate files in directory '%s'"),

 	  	 
