Index: interface/wx/textctrl.h
===================================================================
--- interface/wx/textctrl.h	(revision 68142)
+++ interface/wx/textctrl.h	(working copy)
@@ -220,10 +220,6 @@
     bool Apply(const wxTextAttr& style,
                const wxTextAttr* compareWith = NULL);
 
-    /**
-        Creates a font from the font attributes.
-    */
-    wxFont CreateFont() const;
 
     /**
         Copies all defined/valid properties from overlay to current object.
