Index: include/wx/bookctrl.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/bookctrl.h,v
retrieving revision 1.40
diff -bu2 -r1.40 include/wx/bookctrl.h
--- include/wx/bookctrl.h
+++ include/wx/bookctrl.h
@@ -232,8 +232,10 @@

     // if using DoSetSelection() for implementing [Set|Change]Selection,
-    // then override UpdateSelectedPage() and MakeChangedEvent()
-    virtual int DoSetSelection(size_t nPage, int flags, wxBookCtrlBaseEvent &event);
+    // then override UpdateSelectedPage(), MakeChangingEvent() and MakeChangedEvent()
+    virtual int DoSetSelection(size_t nPage, int flags = 0);
     virtual void UpdateSelectedPage(size_t WXUNUSED(newsel))
         { wxFAIL_MSG(wxT("Override this function!")); }
+    virtual wxBookCtrlBaseEvent* MakeChangingEvent()
+        { wxFAIL_MSG(wxT("Override this function!")); return NULL; }
     virtual void MakeChangedEvent(wxBookCtrlBaseEvent &WXUNUSED(event))
         { wxFAIL_MSG(wxT("Override this function!")); }
Index: include/wx/choicebk.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/choicebk.h,v
retrieving revision 1.11
diff -bu2 -r1.11 include/wx/choicebk.h
--- include/wx/choicebk.h
+++ include/wx/choicebk.h
@@ -85,6 +85,4 @@
     virtual wxSize GetControllerSize() const;

-    int DoSetSelection(size_t nPage, int flags = 0);
-
     void UpdateSelectedPage(size_t newsel)
     {
@@ -93,8 +91,6 @@
     }

-    void MakeChangedEvent(wxBookCtrlBaseEvent &event)
-    {
-        event.SetEventType(wxEVT_COMMAND_CHOICEBOOK_PAGE_CHANGED);
-    }
+    wxBookCtrlBaseEvent* MakeChangingEvent();
+    void MakeChangedEvent(wxBookCtrlBaseEvent &event);

     // event handlers
Index: include/wx/listbook.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/listbook.h,v
retrieving revision 1.17
diff -bu2 -r1.17 include/wx/listbook.h
--- include/wx/listbook.h
+++ include/wx/listbook.h
@@ -86,12 +86,8 @@
     virtual int HitTest(const wxPoint& pt, long *flags = NULL) const;

-    int DoSetSelection(size_t nPage, int flags = 0);
-
     void UpdateSelectedPage(size_t newsel);

-    void MakeChangedEvent(wxBookCtrlBaseEvent &event)
-    {
-        event.SetEventType(wxEVT_COMMAND_LISTBOOK_PAGE_CHANGED);
-    }
+    wxBookCtrlBaseEvent* MakeChangingEvent();
+    void MakeChangedEvent(wxBookCtrlBaseEvent &event);

     // event handlers
Index: include/wx/toolbook.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/toolbook.h,v
retrieving revision 1.5
diff -bu2 -r1.5 include/wx/toolbook.h
--- include/wx/toolbook.h
+++ include/wx/toolbook.h
@@ -102,11 +102,8 @@
     void OnIdle(wxIdleEvent& event);

-    int DoSetSelection(size_t nPage, int flags = 0);
     void UpdateSelectedPage(size_t newsel);

-    void MakeChangedEvent(wxBookCtrlBaseEvent &event)
-    {
-        event.SetEventType(wxEVT_COMMAND_TOOLBOOK_PAGE_CHANGED);
-    }
+    wxBookCtrlBaseEvent* MakeChangingEvent();
+    void MakeChangedEvent(wxBookCtrlBaseEvent &event);

     // the currently selected page or wxNOT_FOUND if none
Index: src/generic/choicbkg.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/generic/choicbkg.cpp,v
retrieving revision 1.20
diff -bu2 -r1.20 src/generic/choicbkg.cpp
--- src/generic/choicbkg.cpp
+++ src/generic/choicbkg.cpp
@@ -210,8 +210,12 @@
 }

-int wxChoicebook::DoSetSelection(size_t n, int flags)
+wxBookCtrlBaseEvent* wxChoicebook::MakeChangingEvent()
 {
-    wxChoicebookEvent event(wxEVT_COMMAND_CHOICEBOOK_PAGE_CHANGING, m_windowId);
-    return wxBookCtrlBase::DoSetSelection(n, flags, event);
+    return new wxChoicebookEvent(wxEVT_COMMAND_CHOICEBOOK_PAGE_CHANGING, m_windowId);
+}
+
+void wxChoicebook::MakeChangedEvent(wxBookCtrlBaseEvent &event)
+{
+    event.SetEventType(wxEVT_COMMAND_CHOICEBOOK_PAGE_CHANGED);
 }

Index: src/generic/listbkg.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/generic/listbkg.cpp,v
retrieving revision 1.41
diff -bu2 -r1.41 src/generic/listbkg.cpp
--- src/generic/listbkg.cpp
+++ src/generic/listbkg.cpp
@@ -275,10 +275,15 @@
 }

-int wxListbook::DoSetSelection(size_t n, int flags)
+wxBookCtrlBaseEvent* wxListbook::MakeChangingEvent()
 {
-    wxListbookEvent event(wxEVT_COMMAND_LISTBOOK_PAGE_CHANGING, m_windowId);
-    return wxBookCtrlBase::DoSetSelection(n, flags, event);
+    return new wxListbookEvent(wxEVT_COMMAND_LISTBOOK_PAGE_CHANGING, m_windowId);
 }

+void wxListbook::MakeChangedEvent(wxBookCtrlBaseEvent &event)
+{
+    event.SetEventType(wxEVT_COMMAND_LISTBOOK_PAGE_CHANGED);
+}
+
+
 // ----------------------------------------------------------------------------
 // adding/removing the pages
Index: src/generic/toolbkg.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/generic/toolbkg.cpp,v
retrieving revision 1.18
diff -bu2 -r1.18 src/generic/toolbkg.cpp
--- src/generic/toolbkg.cpp
+++ src/generic/toolbkg.cpp
@@ -246,8 +246,12 @@
 }

-int wxToolbook::DoSetSelection(size_t n, int flags)
+wxBookCtrlBaseEvent* wxToolbook::MakeChangingEvent()
 {
-    wxToolbookEvent event(wxEVT_COMMAND_TOOLBOOK_PAGE_CHANGING, m_windowId);
-    return wxBookCtrlBase::DoSetSelection(n, flags, event);
+    return new wxToolbookEvent(wxEVT_COMMAND_TOOLBOOK_PAGE_CHANGING, m_windowId);
+}
+
+void wxToolbook::MakeChangedEvent(wxBookCtrlBaseEvent &event)
+{
+    event.SetEventType(wxEVT_COMMAND_TOOLBOOK_PAGE_CHANGED);
 }

Index: src/common/bookctrl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/common/bookctrl.cpp,v
retrieving revision 1.32
diff -bu2 -r1.32 src/common/bookctrl.cpp
--- src/common/bookctrl.cpp
+++ src/common/bookctrl.cpp
@@ -424,5 +424,5 @@
 }

-int wxBookCtrlBase::DoSetSelection(size_t n, int flags, wxBookCtrlBaseEvent &event)
+int wxBookCtrlBase::DoSetSelection(size_t n, int flags)
 {
     wxCHECK_MSG( n < GetPageCount(), wxNOT_FOUND,
@@ -433,13 +433,14 @@
     if ( oldSel != wxNOT_FOUND && n != (size_t)oldSel )
     {
+        wxBookCtrlBaseEvent *event = MakeChangingEvent();
         bool allowed = false;

         if ( flags & SetSelection_SendEvent )
         {
-            event.SetSelection(n);
-            event.SetOldSelection(oldSel);
-            event.SetEventObject(this);
+            event->SetSelection(n);
+            event->SetOldSelection(oldSel);
+            event->SetEventObject(this);

-            allowed = !GetEventHandler()->ProcessEvent(event) || event.IsAllowed();
+            allowed = !GetEventHandler()->ProcessEvent(*event) || event->IsAllowed();
         }

@@ -459,8 +460,10 @@
             {
                 // program allows the page change
-                MakeChangedEvent(event);
-                (void)GetEventHandler()->ProcessEvent(event);
+                MakeChangedEvent(*event);
+                (void)GetEventHandler()->ProcessEvent(*event);
             }
         }
+
+        delete event;
     }


 	  	 
