Index: samples/calendar/calendar.bkl
===================================================================
--- samples/calendar/calendar.bkl	(revision 55030)
+++ samples/calendar/calendar.bkl	(working copy)
@@ -6,8 +6,6 @@
     <exe id="calendar" template="wx_sample" template_append="wx_append">
         <sources>calendar.cpp</sources>
         <wx-lib>adv</wx-lib>
-        <wx-lib>html</wx-lib>
-        <wx-lib>xml</wx-lib>
         <wx-lib>core</wx-lib>
         <wx-lib>base</wx-lib>
         <win32-res>calendar.rc</win32-res>
Index: samples/dialogs/dialogs.bkl
===================================================================
--- samples/dialogs/dialogs.bkl	(revision 55030)
+++ samples/dialogs/dialogs.bkl	(working copy)
@@ -33,8 +33,6 @@
             $(GENERIC_DIALOGS_IN_NATIVE_BUILDS)
         </sources>
         <wx-lib>adv</wx-lib>
-        <wx-lib>html</wx-lib>
-        <wx-lib>xml</wx-lib>
         <wx-lib>core</wx-lib>
         <wx-lib>base</wx-lib>
         <win32-res>dialogs.rc</win32-res>
Index: samples/joytest/joytest.bkl
===================================================================
--- samples/joytest/joytest.bkl	(revision 55030)
+++ samples/joytest/joytest.bkl	(working copy)
@@ -6,8 +6,6 @@
     <exe id="joytest" template="wx_sample" template_append="wx_append">
         <sources>joytest.cpp</sources>
         <wx-lib>adv</wx-lib>
-        <wx-lib>html</wx-lib>
-        <wx-lib>xml</wx-lib>
         <wx-lib>core</wx-lib>
         <wx-lib>base</wx-lib>
         <win32-res>joytest.rc</win32-res>
Index: samples/sashtest/sashtest.bkl
===================================================================
--- samples/sashtest/sashtest.bkl	(revision 55030)
+++ samples/sashtest/sashtest.bkl	(working copy)
@@ -6,8 +6,6 @@
     <exe id="sashtest" template="wx_sample" template_append="wx_append">
         <sources>sashtest.cpp</sources>
         <wx-lib>adv</wx-lib>
-        <wx-lib>html</wx-lib>
-        <wx-lib>xml</wx-lib>
         <wx-lib>core</wx-lib>
         <wx-lib>base</wx-lib>
         <win32-res>sashtest.rc</win32-res>
Index: samples/sound/sound.bkl
===================================================================
--- samples/sound/sound.bkl	(revision 55030)
+++ samples/sound/sound.bkl	(working copy)
@@ -8,8 +8,6 @@
     <exe id="sound" template="wx_sample" template_append="wx_append">
         <sources>sound.cpp</sources>
         <wx-lib>adv</wx-lib>
-        <wx-lib>html</wx-lib>
-        <wx-lib>xml</wx-lib>
         <wx-lib>core</wx-lib>
         <wx-lib>base</wx-lib>
         <win32-res>sound.rc</win32-res>
Index: samples/splash/splash.bkl
===================================================================
--- samples/splash/splash.bkl	(revision 55030)
+++ samples/splash/splash.bkl	(working copy)
@@ -9,8 +9,6 @@
         <sources>splash.cpp</sources>
         <wx-lib>media</wx-lib>
         <wx-lib>adv</wx-lib>
-        <wx-lib>html</wx-lib>
-        <wx-lib>xml</wx-lib>
         <wx-lib>core</wx-lib>
         <wx-lib>base</wx-lib>
         <win32-res>../sample.rc</win32-res>
Index: samples/taskbar/taskbar.bkl
===================================================================
--- samples/taskbar/taskbar.bkl	(revision 55030)
+++ samples/taskbar/taskbar.bkl	(working copy)
@@ -6,8 +6,6 @@
     <exe id="taskbar" template="wx_sample" template_append="wx_append">
         <sources>tbtest.cpp</sources>
         <wx-lib>adv</wx-lib>
-        <wx-lib>html</wx-lib>
-        <wx-lib>xml</wx-lib>
         <wx-lib>core</wx-lib>
         <wx-lib>base</wx-lib>
     </exe>
Index: samples/widgets/widgets.bkl
===================================================================
--- samples/widgets/widgets.bkl	(revision 55030)
+++ samples/widgets/widgets.bkl	(working copy)
@@ -34,9 +34,7 @@
             toggle.cpp
             widgets.cpp
         </sources>
-        <wx-lib>xml</wx-lib>
         <wx-lib>adv</wx-lib>
-        <wx-lib>html</wx-lib>
         <wx-lib>core</wx-lib>
         <wx-lib>base</wx-lib>
         <win32-res>../sample.rc</win32-res>
Index: samples/wizard/wizard.bkl
===================================================================
--- samples/wizard/wizard.bkl	(revision 55030)
+++ samples/wizard/wizard.bkl	(working copy)
@@ -6,8 +6,6 @@
     <exe id="wizard" template="wx_sample" template_append="wx_append">
         <sources>wizard.cpp</sources>
         <wx-lib>adv</wx-lib>
-        <wx-lib>html</wx-lib>
-        <wx-lib>xml</wx-lib>
         <wx-lib>core</wx-lib>
         <wx-lib>base</wx-lib>
         <win32-res>wizard.rc</win32-res>
