--- wxWindows/include/wx/wxchar.h	2004-01-08 10:37:40.000000000 +0100
+++ wxWin/include/wx/wxchar.h	2004-01-21 20:36:19.000000000 +0100
@@ -299,3 +299,11 @@
     #define  wxPutc      _puttc
+    #ifdef __MINGW32__
+        #ifdef wxUSE_UNICODE
+            #define  wxPutchar   putwchar
+        #else
+            #define  wxPutchar   puttchar
+        #endif
+    #else
     #define  wxPutchar   _puttchar
+    #endif
     #define  wxPuts      _putts

 	  	 
