Index: scriptbinder.cpp
===================================================================
--- scriptbinder.cpp	(Revision 62759)
+++ scriptbinder.cpp	(Arbeitskopie)
@@ -593,7 +593,11 @@
     ScriptStream stm;
 
     // check if derived class agrees about saving it
-    if ( !OnSaveDocument( stm ) ) return 0;
+    if ( !OnSaveDocument( stm ) ) 
+	{
+    	fclose( fp );		
+		return 0;
+	}
 
     if ( pFromSection )
 
