diff -rbu2 wx-2.4.1-orig/include/wx/ffile.h wx-2.4.1-patch/include/wx/ffile.h
--- wx-2.4.1-orig/include/wx/ffile.h	Sat Aug 31 13:29:10 2002
+++ wx-2.4.1-patch/include/wx/ffile.h	Tue Sep 23 00:48:02 2003
@@ -91,7 +91,7 @@
   bool IsOpened() const { return m_fp != NULL; }
     // is end of file reached?
-  bool Eof() const { return feof(m_fp) != 0; }
+  bool Eof() const { return ( !IsOpened() || feof(m_fp) != 0 ); }
     // is an error occured?
-  bool Error() const { return ferror(m_fp) != 0; }
+  bool Error() const { return ( !IsOpened() || ferror(m_fp) != 0 ); }
     // get the file name
   const wxString& GetName() const { return m_name; }
diff -rbu2 wx-2.4.1-orig/src/common/ffile.cpp wx-2.4.1-patch/src/common/ffile.cpp
--- wx-2.4.1-orig/src/common/ffile.cpp	Tue Jun 26 22:59:10 2001
+++ wx-2.4.1-patch/src/common/ffile.cpp	Tue Sep 23 00:49:30 2003
@@ -221,4 +221,9 @@
 size_t wxFFile::Tell() const
 {
+    if ( !IsOpened() )
+    {
+      return (size_t)-1;
+    }
+    
     long rc = ftell(m_fp);
     if ( rc == -1 )
@@ -233,4 +238,9 @@
 size_t wxFFile::Length() const
 {
+    if ( !IsOpened() )
+    {
+      return (size_t)-1;
+    }
+    
     wxFFile& self = *(wxFFile *)this;   // const_cast
 

 	  	 
