Index: include/wx/aui/floatpane.h
===================================================================
--- include/wx/aui/floatpane.h	(revision 74279)
+++ include/wx/aui/floatpane.h	(working copy)
@@ -43,6 +43,9 @@
     void SetPaneWindow(const wxAuiPaneInfo& pane);
     wxAuiManager* GetOwnerManager() const;
 
+    // Allow processing accelerators to the parent frame
+    virtual bool IsTopNavigationDomain() const { return false; }
+
 protected:
     virtual void OnMoveStart();
     virtual void OnMoving(const wxRect& windowRect, wxDirection dir);
Index: src/msw/evtloop.cpp
===================================================================
--- src/msw/evtloop.cpp	(revision 74279)
+++ src/msw/evtloop.cpp	(working copy)
@@ -137,7 +137,7 @@
         // stop at first top level window, i.e. don't try to process the key
         // strokes originating in a dialog using the accelerators of the parent
         // frame - this doesn't make much sense
-        if ( wnd->IsTopLevel() )
+        if ( wnd->IsTopNavigationDomain() )
             break;
     }
 
@@ -151,7 +151,7 @@
         // if we don't do this, pressing ESC on a modal dialog shown as child
         // of a modal dialog with wxID_CANCEL will cause the parent dialog to
         // be closed, for example
-        if ( wnd->IsTopLevel() )
+        if ( wnd->IsTopNavigationDomain() )
             break;
     }
 
