--- listbox.cpp	2010-04-15 11:27:17.000000000 +0200
+++ listboxnew.cpp	2010-11-14 14:10:18.000000000 +0100
@@ -204,10 +204,21 @@
     DoSetItemClientData(n, clientData);
 }
 
+static int wxCMPFUNC_CONV IntSort(int* first, int* second)
+{ // sort function for integer array (used in GetSelections() below
+	if (*first == *second) return 0;
+	
+  return *first> *second;
+}
+
 // Return number of selections and an array of selected integers
 int wxListBox::GetSelections(wxArrayInt& aSelections) const
 {
-    return GetPeer()->MacGetSelections( aSelections );
+    int n = GetPeer()->MacGetSelections( aSelections );
+    
+    aSelections.Sort(IntSort);
+
+    return n;
 }
 
 // Get single selection, for single choice list items
