Index: include/wx/tbarbase.h
===================================================================
--- include/wx/tbarbase.h	(revision 59577)
+++ include/wx/tbarbase.h	(working copy)
@@ -359,6 +359,9 @@
     // delete all tools
     virtual void ClearTools();
 
+	// get tool by index
+	virtual wxToolBarToolBase *GetToolByPos(size_t pos);
+
     // must be called after all buttons have been created to finish toolbar
     // initialisation
     virtual bool Realize();
Index: interface/wx/toolbar.h
===================================================================
--- interface/wx/toolbar.h	(revision 59577)
+++ interface/wx/toolbar.h	(working copy)
@@ -373,6 +373,12 @@
     */
     wxToolBarToolBase* FindById(int id) const;
 
+	/**
+	    Returns a pointer to the tool at the specified position or @NULL if
+		given an invalid (out of range) position.
+	*/
+	virtual wxToolBarToolBase* GetToolByPos(size_t pos);
+
     /**
         Returns a pointer to the control identified by @a id or @NULL if no
         corresponding control is found.
Index: src/common/tbarbase.cpp
===================================================================
--- src/common/tbarbase.cpp	(revision 59577)
+++ src/common/tbarbase.cpp	(working copy)
@@ -351,6 +351,24 @@
     return true;
 }
 
+wxToolBarToolBase *wxToolBarBase::GetToolByPos(size_t pos)
+{
+
+	wxToolBarToolBase *tool = NULL;
+
+    wxCHECK_MSG( pos < GetToolsCount(), false,
+                 _T("invalid position in wxToolBar::GetToolByPos()") );
+
+    wxToolBarToolsList::compatibility_iterator node = m_tools.Item(pos);
+
+	if (node != NULL)
+	{
+		tool = node->GetData();
+	}
+
+	return tool;
+}
+
 bool wxToolBarBase::DeleteTool(int id)
 {
     size_t pos = 0;
