Index: src/gtk/tbargtk.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/gtk/tbargtk.cpp,v
retrieving revision 1.122.2.1
diff -u -r1.122.2.1 tbargtk.cpp
--- src/gtk/tbargtk.cpp	2007/01/25 23:34:57	1.122.2.1
+++ src/gtk/tbargtk.cpp	2007/05/13 21:52:42
@@ -191,6 +191,30 @@
 }
 
 //-----------------------------------------------------------------------------
+// "right-click"
+//-----------------------------------------------------------------------------
+extern "C" {
+static gboolean gtk_toolbar_tool_rclick_callback(GtkWidget *WXUNUSED(widget),
+                                                 GdkEventButton *event,
+                                                 wxToolBarToolBase *tool)
+{
+    if (event->button != 3)
+        return FALSE;
+
+    wxToolBar *tbar = (wxToolBar *)tool->GetToolBar();
+
+    if (tbar->m_blockEvent) return TRUE;
+
+    if (g_blockEventsOnDrag) return TRUE;
+    if (!tool->IsEnabled()) return TRUE;
+
+    tbar->OnRightClick( tool->GetId(), (int)event->x, (int)event->y );
+
+    return TRUE;
+}
+}
+
+//-----------------------------------------------------------------------------
 // "enter_notify_event" / "leave_notify_event"
 //-----------------------------------------------------------------------------
 
@@ -456,6 +480,9 @@
                                   tool);
                 g_signal_connect (tool->m_item, "leave_notify_event",
                                   G_CALLBACK (gtk_toolbar_tool_callback),
+                                  tool);
+                g_signal_connect(tool->m_item, "button-press-event",
+                                  G_CALLBACK (gtk_toolbar_tool_rclick_callback),
                                   tool);
             }
             break;

 	  	 
