Index: src/msw/spinctrl.cpp
===================================================================
--- src/msw/spinctrl.cpp	(revision 69503)
+++ src/msw/spinctrl.cpp	(working copy)
@@ -393,14 +393,9 @@
     // Set the range in the native control
     SetRange(min, max);
 
-    if ( !value.empty() )
+    if ( wxAtol(value) != GetValue() )
     {
         SetValue(value);
-        m_oldValue = (int) wxAtol(value);
-    }
-    else
-    {
-        SetValue(wxString::Format(wxT("%d"), initial));
         m_oldValue = initial;
     }
 
