Index: slider.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/mac/carbon/slider.cpp,v
retrieving revision 1.32
diff -u -2 -r1.32 slider.cpp
--- slider.cpp	2004/08/31 13:58:50	1.32
+++ slider.cpp	2004/09/17 18:53:42
@@ -84,5 +84,5 @@
     UInt16 tickMarks = 0 ;
     if ( style & wxSL_AUTOTICKS )
-        tickMarks = (maxValue - minValue);
+        tickMarks = (maxValue - minValue) + 1;
 
     if (tickMarks > 20)

 	  	 
