diff --git a/samples/collpane/collpane.cpp b/samples/collpane/collpane.cpp
index c80e18fd41..508acb980d 100644
--- a/samples/collpane/collpane.cpp
+++ b/samples/collpane/collpane.cpp
@@ -200,7 +200,7 @@ MyFrame::MyFrame()
     menuBar->Append(helpMenu, "&Help");
     SetMenuBar(menuBar);
 
-    m_collPane = new wxCollapsiblePane(this, -1, "This is a wxCollapsiblePane");
+    m_collPane = new wxCollapsiblePane(this, -1, "This is a wxCollapsiblePane", wxDefaultPosition, wxDefaultSize, wxCP_NO_TLW_RESIZE);
     wxWindow *win = m_collPane->GetPane();
 
     m_paneSizer = new wxBoxSizer( wxHORIZONTAL );
