diff -r -u -p a/src/common/dcbase.cpp b/src/common/dcbase.cpp
--- a/src/common/dcbase.cpp	2018-12-09 16:40:08.000000000 +0100
+++ b/src/common/dcbase.cpp	2019-06-28 11:22:20.588562300 +0200
@@ -1451,7 +1451,7 @@ double wxDCImpl::GetMMToPXx() const
 
 double wxDCImpl::GetMMToPXy() const
 {
-    if ( wxIsNullDouble(m_mm_to_pix_x) )
+    if ( wxIsNullDouble(m_mm_to_pix_y) )
     {
         m_mm_to_pix_y = (double)wxGetDisplaySize().GetHeight() /
                         (double)wxGetDisplaySizeMM().GetHeight();
