diff -ur wxWidgets-2.7.0-1.old/src/common/combocmn.cpp wxWidgets-2.7.0-1/src/common/combocmn.cpp
--- wxWidgets-2.7.0-1.old/src/common/combocmn.cpp	2006-08-08 10:28:45.000000000 -0300
+++ wxWidgets-2.7.0-1/src/common/combocmn.cpp	2006-10-11 23:31:35.000000000 -0300
@@ -468,6 +468,7 @@
                      (!event.ShiftDown()?wxNavigationKeyEvent::IsForward:
                                          wxNavigationKeyEvent::IsBackward));
         evt.SetEventObject(m_combo);
+	evt.SetCurrentFocus(m_combo);
         m_combo->GetParent()->GetEventHandler()->AddPendingEvent(evt);
         return;
     }

 	  	 
