Index: src/common/wxchar.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/common/wxchar.cpp,v
retrieving revision 1.134
diff -u -r1.134 wxchar.cpp
--- src/common/wxchar.cpp	2006/12/03 18:54:48	1.134
+++ src/common/wxchar.cpp	2007/01/03 11:14:17
@@ -213,7 +213,7 @@
 
     wxPAT_INT,          // %d, %i, %o, %u, %x, %X
     wxPAT_LONGINT,      // %ld, etc
-#if SIZEOF_LONG_LONG
+#ifdef wxLongLong_t
     wxPAT_LONGLONGINT,  // %Ld, etc
 #endif
     wxPAT_SIZET,        // %Zd, etc
@@ -238,7 +238,7 @@
 typedef union {
     int pad_int;                        //  %d, %i, %o, %u, %x, %X
     long int pad_longint;               // %ld, etc
-#if SIZEOF_LONG_LONG
+#ifdef wxLongLong_t
     long long int pad_longlongint;      // %Ld, etc
 #endif
     size_t pad_sizet;                   // %Zd, etc
@@ -347,7 +347,8 @@
 
     // temporary parse data
     size_t flagofs = 1;
-    bool in_prec, prec_dot;
+    bool in_prec,       // true if we found the dot in some previous iteration
+         prec_dot;      // true if the dot has been already added to m_szFlags
     int ilen = 0;
 
     m_bAlignLeft = in_prec = prec_dot = false;
@@ -417,7 +418,26 @@
                 CHECK_PREC
                 m_szFlags[flagofs++] = char(ch);
                 break;
+#ifdef __WXMSW__
+            // under Windows we support the special '%I64' notation
+            // as long-long integer conversion specifier
+            // (it behaves exactly as '%lli' or '%Li' or '%qi')
+            case wxT('I'):
+                if (*(m_pArgEnd+1) != wxT('6') ||
+                    *(m_pArgEnd+2) != wxT('4'))
+                    return false;       // bad format
 
+                m_pArgEnd++;
+                m_pArgEnd++;
+
+                ilen = 2;
+                CHECK_PREC
+                m_szFlags[flagofs++] = char(ch);
+                m_szFlags[flagofs++] = '6';
+                m_szFlags[flagofs++] = '4';
+                break;
+#endif      // __WXMSW__
+
             case wxT('Z'):
                 ilen = 3;
                 CHECK_PREC
@@ -508,7 +528,7 @@
                 else if (ilen == 1)
                     m_type = wxPAT_LONGINT;
                 else if (ilen == 2)
-#if SIZEOF_LONG_LONG
+#ifdef wxLongLong_t
                     m_type = wxPAT_LONGLONGINT;
 #else // !long long
                     m_type = wxPAT_LONGINT;
@@ -678,7 +698,7 @@
         case wxPAT_LONGINT:
             p->pad_longint = va_arg(argptr, long int);
             break;
-#if SIZEOF_LONG_LONG
+#ifdef wxLongLong_t
         case wxPAT_LONGLONGINT:
             p->pad_longlongint = va_arg(argptr, long long int);
             break;
@@ -762,7 +782,7 @@
             lenScratch = system_sprintf(szScratch, wxMAX_SVNPRINTF_SCRATCHBUFFER_LEN, m_szFlags, p->pad_longint);
             break;
 
-#if SIZEOF_LONG_LONG
+#ifdef wxLongLong_t
         case wxPAT_LONGLONGINT:
             lenScratch = system_sprintf(szScratch, wxMAX_SVNPRINTF_SCRATCHBUFFER_LEN, m_szFlags, p->pad_longlongint);
             break;
@@ -928,7 +948,7 @@
     {
         case wxPAT_INT:
         case wxPAT_LONGINT:
-#if SIZEOF_LONG_LONG
+#ifdef wxLongLong_t
         case wxPAT_LONGLONGINT:
 #endif
         case wxPAT_SIZET:
Index: tests/strings/vsnprintf.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/tests/strings/vsnprintf.cpp,v
retrieving revision 1.12
diff -bu2 -r1.12 tests/strings/vsnprintf.cpp
--- tests/strings/vsnprintf.cpp
+++ tests/strings/vsnprintf.cpp
@@ -311,4 +311,8 @@
 
     CMP3("123456789", "%llu", (wxULongLong_t)123456789);
+
+#ifdef __WXMSW__
+    CMP3("123456789", "%I64d", (wxLongLong_t)123456789);
+#endif
 }
 #endif

 	  	 

 	  	 
