--- wxWidgets/src/gtk/notebook.cpp	Mon Aug 16 08:12:32 2004
+++ wxW/src/gtk/notebook.cpp	Fri Sep 24 05:46:26 2004
@@ -382,21 +382,41 @@
 
 int wxNotebook::SetSelection( size_t page )
 {
-    wxCHECK_MSG( m_widget != NULL, -1, wxT("invalid notebook") );
+    wxCHECK_MSG( m_widget != NULL, wxNOT_FOUND, wxT("invalid notebook") );
 
-    wxCHECK_MSG( page < m_pagesData.GetCount(), -1, wxT("invalid notebook index") );
+    wxCHECK_MSG( page < m_pagesData.GetCount(), wxNOT_FOUND, wxT("invalid notebook index") );
 
-    int selOld = GetSelection();
+    const int oldSel = m_selection;
 
-    // cache the selection
-    m_selection = page;
-    gtk_notebook_set_page( GTK_NOTEBOOK(m_widget), page );
+    if ( int(n) != m_selection )
+    {
+        wxNotebookEvent event(wxEVT_COMMAND_NOTEBOOK_PAGE_CHANGING, m_windowId);
+        event.SetSelection(n);
+        event.SetOldSelection(m_selection);
+        event.SetEventObject(this);
+        if ( !GetEventHandler()->ProcessEvent(event) || event.IsAllowed() )
+        {
+            if ( m_selection != wxNOT_FOUND )
+                m_pages[m_selection]->Hide();
 
-    wxNotebookPage *client = GetPage(page);
-    if ( client )
-        client->SetFocus();
+            wxWindow *page = m_pages[n];
+            if(page)
+            {
+                page->SetSize(GetPageRect());
+                page->Show();
+                page->Focus();
+            }
 
-    return selOld;
+            m_selection = n;
+            gtk_notebook_set_page( GTK_NOTEBOOK(m_widget), page );
+
+            // program allows the page change
+            event.SetEventType(wxEVT_COMMAND_NOTEBOOK_PAGE_CHANGED);
+            (void)GetEventHandler()->ProcessEvent(event);
+        }
+    }
+
+    return oldSel;
 }
 
 bool wxNotebook::SetPageText( size_t page, const wxString &text )

 	  	 
