diff -U 3 -H -d -r -N -- wxGTK-2.6.0/utils/wxrc/wxrc.cpp wxGTK-2.6.0.new/utils/wxrc/wxrc.cpp
--- wxGTK-2.6.0/utils/wxrc/wxrc.cpp	2005-04-20 12:33:35.000000000 -0500
+++ wxGTK-2.6.0.new/utils/wxrc/wxrc.cpp	2005-05-31 11:00:12.000000000 -0500
@@ -179,7 +179,7 @@
     wxArrayString FindStrings();
     wxArrayString FindStrings(wxXmlNode *node);
 
-    bool flagVerbose, flagCPP, flagPython, flagGettext;
+    bool flagVerbose, flagCPP, flagPython, flagGettext, flagShortPath;
     wxString parOutput, parFuncname, parOutputPath;
     wxArrayString parFiles;
     int retCode;
@@ -199,6 +199,7 @@
               wxCMD_LINE_VAL_NONE, wxCMD_LINE_OPTION_HELP },
         { wxCMD_LINE_SWITCH, _T("v"), _T("verbose"), _T("be verbose"), (wxCmdLineParamType)0, 0 },
         { wxCMD_LINE_SWITCH, _T("e"), _T("extra-cpp-code"),  _T("output C++ header file with XRC derived classes"), (wxCmdLineParamType)0, 0 },
+        { wxCMD_LINE_SWITCH, _T("s"), _T("short-path"),  _T("Don't add into the zip file the XRCs full path information"), (wxCmdLineParamType)0, 0 },
         { wxCMD_LINE_SWITCH, _T("c"), _T("cpp-code"),  _T("output C++ source rather than .rsc file"), (wxCmdLineParamType)0, 0 },
         { wxCMD_LINE_SWITCH, _T("p"), _T("python-code"),  _T("output wxPython source rather than .rsc file"), (wxCmdLineParamType)0, 0 },
         { wxCMD_LINE_SWITCH, _T("g"), _T("gettext"),  _T("output list of translatable strings (to stdout or file if -o used)"), (wxCmdLineParamType)0, 0 },
@@ -243,6 +244,11 @@
     flagCPP = cmdline.Found(_T("c"));
     flagPython = cmdline.Found(_T("p"));
     flagH = flagCPP && cmdline.Found(_T("e"));
+    flagShortPath = cmdline.Found(_T("s"));
+    if(flagShortPath && flagVerbose)
+        wxPrintf(_T("Found the --short-path switch.\n"));
+    else if(flagVerbose)
+        wxPrintf(_T("No short paths are being used.\n"));
 
 
     if (!cmdline.Found(_T("o"), &parOutput))
@@ -315,12 +321,21 @@
 wxString XmlResApp::GetInternalFileName(const wxString& name, const wxArrayString& flist)
 {
     wxString name2 = name;
-    name2.Replace(_T(":"), _T("_"));
-    name2.Replace(_T("/"), _T("_"));
-    name2.Replace(_T("\\"), _T("_"));
-    name2.Replace(_T("*"), _T("_"));
-    name2.Replace(_T("?"), _T("_"));
 
+    if(flagShortPath)
+    {
+        wxFileName fn(name);
+        name2 = ::wxFileNameFromPath(name);
+    }
+    else
+    {
+        name2.Replace(_T(":"), _T("_"));
+        name2.Replace(_T("/"), _T("_"));
+        name2.Replace(_T("\\"), _T("_"));
+        name2.Replace(_T("*"), _T("_"));
+        name2.Replace(_T("?"), _T("_"));
+    }
+    
     wxString s = wxFileNameFromPath(parOutput) + _T("$") + name2;
 
     if (wxFileExists(s) && flist.Index(s) == wxNOT_FOUND)
diff -U 3 -H -d -r -N -- wxGTK-2.6.0/wxwin.m4 wxGTK-2.6.0.new/wxwin.m4
--- wxGTK-2.6.0/wxwin.m4	2005-04-20 12:32:34.000000000 -0500
+++ wxGTK-2.6.0.new/wxwin.m4	2005-05-31 15:06:12.000000000 -0500
@@ -240,3 +240,33 @@
   AC_SUBST(WX_LIBS_STATIC)
   AC_SUBST(WX_VERSION)
 ])
+
+dnl ---------------------------------------------------------------------------
+dnl AM_PATH_WXRC
+dnl
+dnl Test for wxWidgets' wxrc program for creating either C++, Python or XRS
+dnl resources.  The variable WXRC will be set and substituted in the configure
+dnl script and Makefiles.
+dnl
+dnl Example use:
+dnl   AM_PATH_WXRC
+dnl ---------------------------------------------------------------------------
+
+dnl
+dnl wxrc program from the wx-config script
+dnl
+AC_DEFUN([AM_PATH_WXRC],
+[
+	AC_MSG_CHECKING([for wxrc])
+	WXRC=`$WX_CONFIG_WITH_ARGS --utility=wxrc`
+	if test "x$WXRC" = x ; then
+		AC_MSG_ERROR([
+			The wxWidgets XML resource system isn't installed.
+			
+			Please check your wxWidgets installation.
+		])
+	fi
+	AC_MSG_RESULT($WXRC)
+	
+	AC_SUBST(WXRC)
+])

 	  	 
