diff -ru2 wxWindows-cvs-orig/src/common/textcmn.cpp wxWindows-cvs-patch/src/common/textcmn.cpp
--- wxWindows-cvs-orig/src/common/textcmn.cpp	Sun Oct 05 23:57:36 2003
+++ wxWindows-cvs-patch/src/common/textcmn.cpp	Thu Oct 09 22:51:32 2003
@@ -240,5 +240,5 @@
 
 #if wxUSE_FFILE
-    wxFFile file(filename, _T("w"));
+    wxFFile file(filenameToUse, _T("w"));
     if ( file.IsOpened() && file.Write(GetValue()) )
     {
@@ -246,5 +246,5 @@
         DiscardEdits();
 
-        m_filename = filename;
+        m_filename = filenameToUse;
 
         return TRUE;

 	  	 
