diff --git a/samples/minimal/minimal.cpp b/samples/minimal/minimal.cpp
--- a/samples/minimal/minimal.cpp
+++ b/samples/minimal/minimal.cpp
@@ -106,7 +106,8 @@ wxEND_EVENT_TABLE()
 // static object for many reasons) and also implements the accessor function
 // wxGetApp() which will return the reference of the right type (i.e. MyApp and
 // not wxApp)
-IMPLEMENT_APP(MyApp)
+//IMPLEMENT_APP(MyApp)
+IMPLEMENT_APP_NO_MAIN(MyApp)
 
 // ============================================================================
 // implementation
@@ -198,3 +199,20 @@ void MyFrame::OnAbout(wxCommandEvent& WXUNUSED(event))
                  wxOK | wxICON_INFORMATION,
                  this);
 }
+
+int CALLBACK WinMain(HINSTANCE hInst, HINSTANCE hPrevInst, LPSTR cmdLine, int cmdShow)
+{
+    if (!wxEntryStart(hInst, hPrevInst, cmdLine, cmdShow))
+        return 1;
+
+    if (!wxTheApp || !wxTheApp->CallOnInit())
+        return 1;
+
+    wxMessageBox(wxGetApp().GetAppName());
+
+    if (wxTheApp)
+        wxTheApp->OnExit();
+
+    wxEntryCleanup();
+    return 0;
+}
