diff -bu1 -P -r wxWidgets/./contrib/samples/deprecated/proplist/proplist.cpp wxW/./contrib/samples/deprecated/proplist/proplist.cpp
--- wxWidgets/./contrib/samples/deprecated/proplist/proplist.cpp	2004-02-28 21:12:58.000000000 +0100
+++ wxW/./contrib/samples/deprecated/proplist/proplist.cpp	2004-04-14 19:17:43.000000000 +0200
@@ -184,3 +184,3 @@
         propDialog = new PropListDialog(view, NULL, _T("Property Sheet Test"),
-            wxDefaultPosition, wxSize(400, 500), wxDEFAULT_DIALOG_STYLE|wxDIALOG_MODELESS);
+            wxDefaultPosition, wxSize(400, 500));
         m_childWindow = propDialog;
@@ -232,3 +232,3 @@
         propDialog = new PropFormDialog(view, NULL, _T("Property Form Test"),
-            wxDefaultPosition, wxSize(380, 250), wxDEFAULT_DIALOG_STYLE|wxDIALOG_MODAL);
+            wxDefaultPosition, wxSize(380, 250));
         m_childWindow = propDialog;
diff -bu1 -P -r wxWidgets/./contrib/samples/fl/fl_demo2.cpp wxW/./contrib/samples/fl/fl_demo2.cpp
--- wxWidgets/./contrib/samples/fl/fl_demo2.cpp	2003-11-09 21:40:23.000000000 +0100
+++ wxW/./contrib/samples/fl/fl_demo2.cpp	2004-04-14 19:00:25.000000000 +0200
@@ -175,3 +175,3 @@
                       wxSize( 385,220),
-                      wxDIALOG_MODAL | wxDEFAULT_DIALOG_STYLE | wxTAB_TRAVERSAL );
+                      wxDEFAULT_DIALOG_STYLE | wxTAB_TRAVERSAL );
     
diff -bu1 -P -r wxWidgets/./contrib/src/deprecated/proplist.cpp wxW/./contrib/src/deprecated/proplist.cpp
--- wxWidgets/./contrib/src/deprecated/proplist.cpp	2004-02-28 21:13:00.000000000 +0100
+++ wxW/./contrib/src/deprecated/proplist.cpp	2004-04-14 19:00:50.000000000 +0200
@@ -1645,3 +1645,3 @@
   wxPropertyStringListEditorDialog *dialog = new wxPropertyStringListEditorDialog(parent,
-      title, wxPoint(10, 10), wxSize(400, 400), wxDEFAULT_DIALOG_STYLE|wxDIALOG_MODAL);
+      title, wxPoint(10, 10), wxSize(400, 400));
 
diff -bu1 -P -r wxWidgets/./demos/forty/playerdg.cpp wxW/./demos/forty/playerdg.cpp
--- wxWidgets/./demos/forty/playerdg.cpp	2004-03-23 21:23:16.000000000 +0100
+++ wxW/./demos/forty/playerdg.cpp	2004-04-14 19:11:30.000000000 +0200
@@ -46,5 +46,3 @@
 							) :
-	wxDialog(parent, wxID_ANY, _T("Player Selection"),
-			wxDefaultPosition, wxDefaultSize,
-			wxDIALOG_MODAL | wxDEFAULT_DIALOG_STYLE),
+	wxDialog(parent, wxID_ANY, _T("Player Selection")),
 	m_scoreFile(file)
diff -bu1 -P -r wxWidgets/./demos/forty/scoredg.cpp wxW/./demos/forty/scoredg.cpp
--- wxWidgets/./demos/forty/scoredg.cpp	2004-03-23 21:23:17.000000000 +0100
+++ wxW/./demos/forty/scoredg.cpp	2004-04-14 19:11:54.000000000 +0200
@@ -139,4 +139,3 @@
 	wxDialog(parent, wxID_ANY, _("Scores"),
-			wxDefaultPosition, wxSize(400, 300),
-			wxDIALOG_MODAL | wxDEFAULT_DIALOG_STYLE),
+			wxDefaultPosition, wxSize(400, 300)),
 	m_scoreFile(file)
diff -bu1 -P -r wxWidgets/./demos/life/dialogs.cpp wxW/./demos/life/dialogs.cpp
--- wxWidgets/./demos/life/dialogs.cpp	2003-08-25 19:24:20.000000000 +0200
+++ wxW/./demos/life/dialogs.cpp	2004-04-14 19:13:11.000000000 +0200
@@ -80,7 +80,3 @@
 LifeSamplesDialog::LifeSamplesDialog(wxWindow *parent)
-                 : wxDialog(parent, -1,
-                            _("Sample games"),
-                            wxDefaultPosition,
-                            wxDefaultSize,
-                            wxDEFAULT_DIALOG_STYLE | wxDIALOG_MODAL)
+                 : wxDialog(parent, -1, _("Sample games"))
 {
@@ -167,6 +163,3 @@
                : wxDialog(parent, -1,
-                          _("About Life!"),
-                          wxDefaultPosition,
-                          wxDefaultSize,
-                          wxDEFAULT_DIALOG_STYLE | wxDIALOG_MODAL)
+                          _("About Life!"))
 {
diff -bu1 -P -r wxWidgets/./samples/tab/tab.cpp wxW/./samples/tab/tab.cpp
--- wxWidgets/./samples/tab/tab.cpp	2002-03-17 14:15:58.000000000 +0100
+++ wxW/./samples/tab/tab.cpp	2004-04-14 19:07:56.000000000 +0200
@@ -41,3 +41,3 @@
 #if USE_TABBED_DIALOG
-  dialog = new MyDialog((wxFrame *) NULL, -1, (char *) "Tabbed Dialog", wxPoint(-1, -1), wxSize(365, 390), wxDIALOG_MODAL|wxDEFAULT_DIALOG_STYLE);
+  dialog = new MyDialog((wxFrame *) NULL, -1, (char *) "Tabbed Dialog", wxPoint(-1, -1), wxSize(365, 390));
 
diff -bu1 -P -r wxWidgets/./samples/taskbar/tbtest.cpp wxW/./samples/taskbar/tbtest.cpp
--- wxWidgets/./samples/taskbar/tbtest.cpp	2004-03-23 23:20:16.000000000 +0100
+++ wxW/./samples/taskbar/tbtest.cpp	2004-04-14 19:19:23.000000000 +0200
@@ -38,3 +38,3 @@
     // Create the main frame window
-    dialog = new MyDialog(NULL, -1, wxT("wxTaskBarIcon Test Dialog"), wxPoint(-1, -1), wxSize(365, 290), wxDIALOG_MODELESS|wxDEFAULT_DIALOG_STYLE);
+    dialog = new MyDialog(NULL, -1, wxT("wxTaskBarIcon Test Dialog"), wxPoint(-1, -1), wxSize(365, 290));
 
diff -bu1 -P -r wxWidgets/./samples/validate/validate.cpp wxW/./samples/validate/validate.cpp
--- wxWidgets/./samples/validate/validate.cpp	2003-07-28 19:51:33.000000000 +0200
+++ wxW/./samples/validate/validate.cpp	2004-04-14 19:07:35.000000000 +0200
@@ -173,3 +173,3 @@
                     const wxPoint& pos, const wxSize& size, const long WXUNUSED(style) ) :
-    wxDialog(parent, VALIDATE_DIALOG_ID, title, pos, size, wxDEFAULT_DIALOG_STYLE|wxDIALOG_MODAL|wxRESIZE_BORDER)
+    wxDialog(parent, VALIDATE_DIALOG_ID, title, pos, size, wxDEFAULT_DIALOG_STYLE|wxRESIZE_BORDER)
 {
diff -bu1 -P -r wxWidgets/./src/generic/colrdlgg.cpp wxW/./src/generic/colrdlgg.cpp
--- wxWidgets/./src/generic/colrdlgg.cpp	2003-10-19 01:50:52.000000000 +0200
+++ wxW/./src/generic/colrdlgg.cpp	2004-04-14 18:54:35.000000000 +0200
@@ -146,4 +146,3 @@
     if ( !wxDialog::Create(parent, -1, wxT("Colour"),
-                           wxPoint(0, 0), wxSize(900, 900),
-                           wxDEFAULT_DIALOG_STYLE | wxDIALOG_MODAL) )
+                           wxPoint(0, 0), wxSize(900, 900)) )
         return FALSE;
diff -bu1 -P -r wxWidgets/./src/generic/numdlgg.cpp wxW/./src/generic/numdlgg.cpp
--- wxWidgets/./src/generic/numdlgg.cpp	2004-03-28 15:10:02.000000000 +0200
+++ wxW/./src/generic/numdlgg.cpp	2004-04-14 18:54:57.000000000 +0200
@@ -84,4 +84,3 @@
                    : wxDialog(parent, -1, caption,
-                              pos, wxDefaultSize,
-                              wxDEFAULT_DIALOG_STYLE | wxDIALOG_MODAL )
+                              pos, wxDefaultSize)
 {
diff -bu1 -P -r wxWidgets/./src/generic/prntdlgg.cpp wxW/./src/generic/prntdlgg.cpp
--- wxWidgets/./src/generic/prntdlgg.cpp	2004-03-29 20:02:48.000000000 +0200
+++ wxW/./src/generic/prntdlgg.cpp	2004-04-14 19:15:43.000000000 +0200
@@ -109,3 +109,2 @@
                                wxDEFAULT_DIALOG_STYLE |
-                               wxDIALOG_MODAL |
                                wxTAB_TRAVERSAL)
@@ -123,3 +122,2 @@
                                wxDEFAULT_DIALOG_STYLE |
-                               wxDIALOG_MODAL |
                                wxTAB_TRAVERSAL)
@@ -135,3 +133,3 @@
   //    wxDialog::Create(parent, -1, _("Print"), wxPoint(0, 0), wxSize(600, 600),
-  //                     wxDEFAULT_DIALOG_STYLE | wxDIALOG_MODAL | wxTAB_TRAVERSAL);
+  //                     wxDEFAULT_DIALOG_STYLE | wxTAB_TRAVERSAL);
 
@@ -407,3 +405,3 @@
 wxGenericPrintSetupDialog::wxGenericPrintSetupDialog(wxWindow *parent, wxPrintData* data):
-wxDialog(parent, -1, _("Print Setup"), wxPoint(0, 0), wxSize(600, 600), wxDEFAULT_DIALOG_STYLE|wxDIALOG_MODAL|wxTAB_TRAVERSAL)
+wxDialog(parent, -1, _("Print Setup"), wxPoint(0, 0), wxSize(600, 600), wxDEFAULT_DIALOG_STYLE|wxTAB_TRAVERSAL)
 {
@@ -600,3 +598,3 @@
                 wxSize(600, 600),
-                wxDIALOG_MODAL|wxDEFAULT_DIALOG_STYLE|wxTAB_TRAVERSAL )
+                wxDEFAULT_DIALOG_STYLE|wxTAB_TRAVERSAL )
 {
diff -bu1 -P -r wxWidgets/./src/generic/progdlgg.cpp wxW/./src/generic/progdlgg.cpp
--- wxWidgets/./src/generic/progdlgg.cpp	2004-03-27 08:51:10.000000000 +0100
+++ wxW/./src/generic/progdlgg.cpp	2004-04-14 18:56:34.000000000 +0200
@@ -146,4 +146,4 @@
     {
-        // note that we can't use wxGA_SMOOTH because it happens to also mean
-        // wxDIALOG_MODAL and will cause the dialog to be modal. Have an extra
+        // note that we can't use wxGA_SMOOTH because it happens to
+        // cause the dialog to be modal. Have an extra
         // style argument to wxProgressDialog, perhaps.
diff -bu1 -P -r wxWidgets/./src/generic/textdlgg.cpp wxW/./src/generic/textdlgg.cpp
--- wxWidgets/./src/generic/textdlgg.cpp	2003-08-09 14:46:03.000000000 +0200
+++ wxW/./src/generic/textdlgg.cpp	2004-04-14 19:16:40.000000000 +0200
@@ -74,4 +74,3 @@
                                      const wxPoint& pos)
-                 : wxDialog(parent, -1, caption, pos, wxDefaultSize,
-                            wxDEFAULT_DIALOG_STYLE | wxDIALOG_MODAL),
+                 : wxDialog(parent, -1, caption, pos),
                    m_value(value)
diff -bu1 -P -r wxWidgets/./src/gtk/dialog.cpp wxW/./src/gtk/dialog.cpp
--- wxWidgets/./src/gtk/dialog.cpp	2003-08-09 14:46:09.000000000 +0200
+++ wxW/./src/gtk/dialog.cpp	2004-04-14 18:58:34.000000000 +0200
@@ -179,8 +179,2 @@
 {
-/*
-  if (flag)
-    m_windowStyle |= wxDIALOG_MODAL;
-  else
-    if (m_windowStyle & wxDIALOG_MODAL) m_windowStyle -= wxDIALOG_MODAL;
-*/
     wxFAIL_MSG( wxT("wxDialog:SetModal obsolete now") );
diff -bu1 -P -r wxWidgets/./src/gtk/filedlg.cpp wxW/./src/gtk/filedlg.cpp
--- wxWidgets/./src/gtk/filedlg.cpp	2003-09-21 01:59:55.000000000 +0200
+++ wxW/./src/gtk/filedlg.cpp	2004-04-14 18:57:32.000000000 +0200
@@ -135,3 +135,3 @@
     if (!PreCreation( parent, pos, wxDefaultSize ) ||
-        !CreateBase( parent, -1, pos, wxDefaultSize, style | wxDIALOG_MODAL, wxDefaultValidator, wxT("filedialog") ))
+        !CreateBase( parent, -1, pos, wxDefaultSize, style, wxDefaultValidator, wxT("filedialog") ))
     {

 	  	 
