Index: interface/wx/display.h
===================================================================
--- interface/wx/display.h	(revision 66819)
+++ interface/wx/display.h	(working copy)
@@ -44,11 +44,6 @@
         there are differences on other platforms. On Unix variations using X11
         extensions it should behave as defined, but some irregularities may
         occur.
-
-        On wxMac passing in wxDefaultVideoMode as the mode parameter does
-        nothing. This happens because carbon no longer has access to
-        @c DMUseScreenPrefs(), an undocumented function that changed the video
-        mode to the system default by using the system's "scrn" resource.
     */
     bool ChangeMode(const wxVideoMode& mode = wxDefaultVideoMode);
 
Index: src/osx/core/display.cpp
===================================================================
--- src/osx/core/display.cpp	(revision 66819)
+++ src/osx/core/display.cpp	(working copy)
@@ -267,10 +267,14 @@
 
 bool wxDisplayImplMacOSX::ChangeMode( const wxVideoMode& mode )
 {
-    // Changing to default mode (wxDefaultVideoMode) doesn't
-    // work because we don't have access to the system's 'scrn'
-    // resource which holds the user's mode which the system
-    // will return to after this app is done
+#ifndef __WXOSX_IPHONE__
+    if (mode == wxDefaultVideoMode)
+    {
+        CGRestorePermanentDisplayConfiguration();
+        return true;
+    }
+#endif
+        
     boolean_t bExactMatch;
     CFDictionaryRef theCGMode = CGDisplayBestModeForParametersAndRefreshRate(
         m_id,
