Index: include/wx/dataview.h
===================================================================
--- include/wx/dataview.h	(Revision 71222)
+++ include/wx/dataview.h	(Arbeitskopie)
@@ -228,7 +228,7 @@
         return true;
     }
 
-    // define hierachy
+    // define hierarchy
     virtual wxDataViewItem GetParent( const wxDataViewItem &item ) const = 0;
     virtual bool IsContainer( const wxDataViewItem &item ) const = 0;
     // Is the container just a header or an item with all columns
Index: include/wx/datetime.h
===================================================================
--- include/wx/datetime.h	(Revision 71222)
+++ include/wx/datetime.h	(Arbeitskopie)
@@ -408,7 +408,7 @@
     // helper classes
     // ------------------------------------------------------------------------
 
-        // a class representing a time zone: basicly, this is just an offset
+        // a class representing a time zone: basically, this is just an offset
         // (in seconds) from GMT
     class WXDLLIMPEXP_BASE TimeZone
     {
@@ -693,7 +693,7 @@
         // default assignment operator is ok
 
     // calendar calculations (functions which set the date only leave the time
-    // unchanged, e.g. don't explictly zero it): SetXXX() functions modify the
+    // unchanged, e.g. don't explicitly zero it): SetXXX() functions modify the
     // object itself, GetXXX() ones return a new object.
     // ------------------------------------------------------------------------
 
Index: include/wx/sizer.h
===================================================================
--- include/wx/sizer.h	(Revision 71222)
+++ include/wx/sizer.h	(Arbeitskopie)
@@ -601,7 +601,7 @@
     virtual void DeleteWindows();
 
     // Inform sizer about the first direction that has been decided (by parent item)
-    // Returns true if it made use of the informtion (and recalculated min size)
+    // Returns true if it made use of the information (and recalculated min size)
     virtual bool InformFirstDirection( int WXUNUSED(direction), int WXUNUSED(size), int WXUNUSED(availableOtherDir) )
         { return false; }
 
Index: include/wx/confbase.h
===================================================================
--- include/wx/confbase.h	(Revision 71222)
+++ include/wx/confbase.h	(Arbeitskopie)
@@ -73,7 +73,7 @@
 // abstract base class wxConfigBase which defines the interface for derived
 // classes
 //
-// wxConfig organizes the items in a tree-like structure (modeled after the
+// wxConfig organizes the items in a tree-like structure (modelled after the
 // Unix/Dos filesystem). There are groups (directories) and keys (files).
 // There is always one current group given by the current path.
 //
Index: include/wx/textctrl.h
===================================================================
--- include/wx/textctrl.h	(Revision 71222)
+++ include/wx/textctrl.h	(Arbeitskopie)
@@ -806,7 +806,7 @@
           m_start(event.m_start),
           m_end(event.m_end) { }
 
-    // get the mouse event which happend over the URL
+    // get the mouse event which happened over the URL
     const wxMouseEvent& GetMouseEvent() const { return m_evtMouse; }
 
     // get the start of the URL
Index: include/wx/buffer.h
===================================================================
--- include/wx/buffer.h	(Revision 71222)
+++ include/wx/buffer.h	(Arbeitskopie)
@@ -438,7 +438,7 @@
 
     friend class wxMemoryBuffer;
 
-    // everyting is private as it can only be used by wxMemoryBuffer
+    // everything is private as it can only be used by wxMemoryBuffer
 private:
     wxMemoryBufferData(size_t size = wxMemoryBufferData::DefBufSize)
         : m_data(size ? malloc(size) : NULL), m_size(size), m_len(0), m_ref(0)
