Index: msgdlg.cpp
===================================================================
--- msgdlg.cpp	(revision 68330)
+++ msgdlg.cpp	(working copy)
@@ -715,6 +715,9 @@
     // custom label button array that can hold all buttons in use
     tdc.pButtons = buttons.get();
 
+	if ( style == 0 )
+		return;
+
     if ( style & wxYES_NO )
     {
         AddTaskDialogButton(tdc, IDYES, TDCBF_YES_BUTTON, btnYesLabel);
