--- C:/Dev/exacq/wx-2.8.9/src/msw/treectrl.cpp	Fri Sep  2 14:43:54 2011
+++ C:/Dev/exacq/wx-2.8.9/src/msw/treectrl - Copy.cpp	Tue Jun 12 16:31:21 2012
@@ -86,13 +86,13 @@
 {
 public:
     // unlock a single item
-    TreeItemUnlocker(HTREEITEM item) { ms_unlockedItem = item; }
+    TreeItemUnlocker(HTREEITEM item) { m_prevItem = ms_unlockedItem; ms_unlockedItem = item; }
 
     // unlock all items, don't use unless absolutely necessary
-    TreeItemUnlocker() { ms_unlockedItem = (HTREEITEM)-1; }
+    TreeItemUnlocker() { m_prevItem = ms_unlockedItem; ms_unlockedItem = (HTREEITEM)-1; }
 
     // lock everything back
-    ~TreeItemUnlocker() { ms_unlockedItem = NULL; }
+    ~TreeItemUnlocker() { ms_unlockedItem = m_prevItem; }
 
 
     // check if the item state is currently locked
@@ -101,6 +101,7 @@
 
 private:
     static HTREEITEM ms_unlockedItem;
+    HTREEITEM m_prevItem;
 };
 
 HTREEITEM TreeItemUnlocker::ms_unlockedItem = NULL;
