diff --git a/include/wx/generic/textdlgg.h b/include/wx/generic/textdlgg.h
index 023d497..b0cd9d0 100644
--- a/include/wx/generic/textdlgg.h
+++ b/include/wx/generic/textdlgg.h
@@ -96,12 +96,25 @@ private:
 class WXDLLIMPEXP_CORE wxPasswordEntryDialog : public wxTextEntryDialog
 {
 public:
+    wxPasswordEntryDialog();
     wxPasswordEntryDialog(wxWindow *parent,
                       const wxString& message,
                       const wxString& caption = wxGetPasswordFromUserPromptStr,
                       const wxString& value = wxEmptyString,
                       long style = wxTextEntryDialogStyle,
-                      const wxPoint& pos = wxDefaultPosition);
+                      const wxPoint& pos = wxDefaultPosition)
+    {
+        Create(parent, message, caption, value, style, pos);
+    }
+
+    bool Create(wxWindow *parent,
+                const wxString& message,
+                const wxString& caption = wxGetPasswordFromUserPromptStr,
+                const wxString& value = wxEmptyString,
+                long style = wxTextEntryDialogStyle,
+                const wxPoint& pos = wxDefaultPosition);
+                      
+                      
 private:
     DECLARE_DYNAMIC_CLASS(wxPasswordEntryDialog)
     wxDECLARE_NO_COPY_CLASS(wxPasswordEntryDialog);
diff --git a/src/generic/textdlgg.cpp b/src/generic/textdlgg.cpp
index f3d697f..78c6093 100644
--- a/src/generic/textdlgg.cpp
+++ b/src/generic/textdlgg.cpp
@@ -189,16 +189,16 @@ void wxTextEntryDialog::SetTextValidator( const wxTextValidator& validator )
 
 IMPLEMENT_CLASS(wxPasswordEntryDialog, wxTextEntryDialog)
 
-wxPasswordEntryDialog::wxPasswordEntryDialog(wxWindow *parent,
-                                     const wxString& message,
-                                     const wxString& caption,
-                                     const wxString& value,
-                                     long style,
-                                     const wxPoint& pos)
-                 : wxTextEntryDialog(parent, message, caption, value,
-                                     style | wxTE_PASSWORD, pos)
+bool wxPasswordEntryDialog::Create(wxWindow *parent,
+                                   const wxString& message,
+                                   const wxString& caption,
+                                   const wxString& value,
+                                   long style,
+                                   const wxPoint& pos)
 {
     // Only change from wxTextEntryDialog is the password style
+    return wxTextEntryDialog::Create(parent, message, caption, value,
+                                     style | wxTE_PASSWORD, pos);
 }
 
 #endif // wxUSE_TEXTDLG
