Index: tests/test.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/tests/test.cpp,v
retrieving revision 1.6
diff -u -2 -r1.6 test.cpp
--- tests/test.cpp	2004/07/20 17:18:49	1.6
+++ tests/test.cpp	2004/10/23 22:34:50
@@ -29,6 +29,13 @@
 #include <iostream>
 
-using namespace std;
-using namespace CppUnit;
+using CppUnit::Test;
+using CppUnit::TestSuite;
+using CppUnit::TestFactoryRegistry;
+using CppUnit::TextUi::TestRunner;
+
+using std::string;
+using std::vector;
+using std::auto_ptr;
+using std::cout;
 
 // The application class
@@ -67,5 +74,5 @@
 {
     cout << "Test program for wxWidgets\n"
-         << "build: " << WX_BUILD_OPTIONS_SIGNATURE << endl;
+         << "build: " << WX_BUILD_OPTIONS_SIGNATURE << std::endl;
     return wxAppConsole::OnInit();
 };
@@ -112,5 +119,5 @@
 int TestApp::OnRun()
 {
-    TextUi::TestRunner runner;
+    TestRunner runner;
 
     for (size_t i = 0; i < m_registries.size(); i++) {
@@ -132,14 +139,12 @@
 #if wxUSE_LOG
     // Switch off logging unless --verbose
-    wxLog::EnableLogging(wxLog::GetVerbose());
-#endif // wxUSE_LOG
-
-    return m_list || runner.run("", false, true, 
-#if wxUSE_LOG
-        !wxLog::GetVerbose()
+    bool verbose = wxLog::GetVerbose();
+    wxLog::EnableLogging(verbose);
 #else
-        true
-#endif // wxUSE_LOG
-    ) ? EXIT_SUCCESS : EXIT_FAILURE;
+    bool verbose = false;
+#endif
+
+    return m_list || runner.run("", false, true, !verbose) ?
+            EXIT_SUCCESS : EXIT_FAILURE;
 }
 
Index: tests/formatconverter/formatconvertertest.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/tests/formatconverter/formatconvertertest.cpp,v
retrieving revision 1.2
diff -u -2 -r1.2 formatconvertertest.cpp
--- tests/formatconverter/formatconvertertest.cpp	2004/04/18 20:04:30	1.2
+++ tests/formatconverter/formatconvertertest.cpp	2004/10/23 22:34:51
@@ -57,6 +57,6 @@
 #endif
 
-using namespace std;
-using namespace CppUnit;
+using CppUnit::TestCase;
+using std::string;
 
 
Index: tests/regex/regextest.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/tests/regex/regextest.cpp,v
retrieving revision 1.6
diff -u -2 -r1.6 regextest.cpp
--- tests/regex/regextest.cpp	2004/10/08 05:00:28	1.6
+++ tests/regex/regextest.cpp	2004/10/23 22:34:52
@@ -44,8 +44,14 @@
 #include "wx/regex.h"
 #include "wx/cppunit.h"
-#include <stdexcept>
+#include <string>
+#include <vector>
 
-using namespace std;
-using namespace CppUnit;
+using CppUnit::Test;
+using CppUnit::TestCase;
+using CppUnit::TestSuite;
+using CppUnit::Exception;
+
+using std::vector;
+using std::string;
 
 // many of the tests are specific to the builtin regex lib, so only attempts
Index: tests/regex/wxregextest.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/tests/regex/wxregextest.cpp,v
retrieving revision 1.1
diff -u -2 -r1.1 wxregextest.cpp
--- tests/regex/wxregextest.cpp	2004/04/21 20:16:37	1.1
+++ tests/regex/wxregextest.cpp	2004/10/23 22:34:53
@@ -25,6 +25,8 @@
 #include <string>
 
-using namespace std;
-using namespace CppUnit;
+using CppUnit::Test;
+using CppUnit::TestCase;
+using CppUnit::TestSuite;
+using std::string;
 
 
Index: tests/streams/bstream.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/tests/streams/bstream.cpp,v
retrieving revision 1.3
diff -u -2 -r1.3 bstream.cpp
--- tests/streams/bstream.cpp	2004/09/19 22:00:13	1.3
+++ tests/streams/bstream.cpp	2004/10/23 22:34:53
@@ -28,5 +28,6 @@
 #include "bstream.h"
 
-using namespace CppUnit;
+using CppUnit::TestSuite;
+using CppUnit::Test;
 
 ///////////////////////////////////////////////////////////////////////////////
Index: tests/streams/bstream.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/tests/streams/bstream.h,v
retrieving revision 1.3
diff -u -2 -r1.3 bstream.h
--- tests/streams/bstream.h	2004/04/18 19:26:20	1.3
+++ tests/streams/bstream.h	2004/10/23 22:34:54
@@ -12,5 +12,4 @@
 
 #include "wx/cppunit.h"
-using namespace CppUnit;
 
 ///////////////////////////////////////////////////////////////////////////////
@@ -22,8 +21,8 @@
     STREAM_TEST_NAME "." #Name
 #define STREAM_REGISTER_SUB_SUITE(Name) \
-    extern Test* Get##Name##Suite(); \
+    extern CppUnit::Test* Get##Name##Suite(); \
     suite->addTest(Get##Name##Suite())
 #define STREAM_IMPLEMENT_SUB_REGISTRATION_ROUTINE(Name) \
-    Test* Get##Name##Suite() { return Name::suite(); }
+    CppUnit::Test* Get##Name##Suite() { return Name::suite(); }
 #define STREAM_TEST_SUBSUITE_NAMED_REGISTRATION(Name) \
     CPPUNIT_TEST_SUITE_NAMED_REGISTRATION( Name, COMPOSE_TEST_NAME(Name) ); \
@@ -35,5 +34,5 @@
 //
 
-template <class TStreamIn, class TStreamOut> class BaseStreamTestCase : public TestCase
+template <class TStreamIn, class TStreamOut> class BaseStreamTestCase : public CppUnit::TestCase
 {
 protected:
Index: tests/streams/ffilestream.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/tests/streams/ffilestream.cpp,v
retrieving revision 1.3
diff -u -2 -r1.3 ffilestream.cpp
--- tests/streams/ffilestream.cpp	2004/04/18 19:26:20	1.3
+++ tests/streams/ffilestream.cpp	2004/10/23 22:34:54
@@ -30,7 +30,4 @@
 #include "bstream.h"
 
-using namespace std;
-using namespace CppUnit;
-
 #define DATABUFFER_SIZE     1024
 
Index: tests/streams/filestream.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/tests/streams/filestream.cpp,v
retrieving revision 1.2
diff -u -2 -r1.2 filestream.cpp
--- tests/streams/filestream.cpp	2004/04/11 20:59:33	1.2
+++ tests/streams/filestream.cpp	2004/10/23 22:34:54
@@ -30,7 +30,4 @@
 #include "bstream.h"
 
-using namespace std;
-using namespace CppUnit;
-
 #define DATABUFFER_SIZE     1024
 
Index: tests/streams/memstream.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/tests/streams/memstream.cpp,v
retrieving revision 1.1
diff -u -2 -r1.1 memstream.cpp
--- tests/streams/memstream.cpp	2004/04/01 10:13:47	1.1
+++ tests/streams/memstream.cpp	2004/10/23 22:34:54
@@ -30,7 +30,4 @@
 #include "bstream.h"
 
-using namespace std;
-using namespace CppUnit;
-
 #define DATABUFFER_SIZE     256
 
Index: tests/streams/sstream.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/tests/streams/sstream.cpp,v
retrieving revision 1.1
diff -u -2 -r1.1 sstream.cpp
--- tests/streams/sstream.cpp	2004/09/19 22:00:13	1.1
+++ tests/streams/sstream.cpp	2004/10/23 22:34:55
@@ -24,7 +24,4 @@
 #include "bstream.h"
 
-using namespace std;
-using namespace CppUnit;
-
 ///////////////////////////////////////////////////////////////////////////////
 // The test case
Index: tests/streams/zlibstream.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/tests/streams/zlibstream.cpp,v
retrieving revision 1.6
diff -u -2 -r1.6 zlibstream.cpp
--- tests/streams/zlibstream.cpp	2004/04/26 09:22:03	1.6
+++ tests/streams/zlibstream.cpp	2004/10/23 22:34:56
@@ -34,6 +34,5 @@
 #include "bstream.h"
 
-using namespace std;
-using namespace CppUnit;
+using std::string;
 
 #define WXTEST_WITH_GZIP_CONDITION(testMethod) \

 	  	 
