Index: src/osx/carbon/font.cpp
===================================================================
--- src/osx/carbon/font.cpp	(revision 55565)
+++ src/osx/carbon/font.cpp	(working copy)
@@ -415,7 +415,7 @@
                 traits |= kCTFontItalicTrait;
      
 // use font descriptor caching
-#if 0
+#if 1
             wxString lookupname = wxString::Format( "%s_%ld", m_faceName.c_str(), traits );
             
             static std::map< std::wstring , wxCFRef< CTFontDescriptorRef > > fontdescriptorcache ;
@@ -433,19 +433,22 @@
 #endif
             
 // use font caching
-#if 0
+#if 1
             wxString lookupnameWithSize = wxString::Format( "%s_%ld_%ld", m_faceName.c_str(), traits, m_pointSize );
             
             static std::map< std::wstring , wxCFRef< CTFontRef > > fontcache ;
             m_ctFont = fontcache[ std::wstring(lookupnameWithSize.wc_str()) ];
             if ( !m_ctFont )
             {
-                m_ctFont.reset( CTFontCreateWithFontDescriptor( m_ctFontDescriptor, m_pointSize, NULL ) );
+                CTFontRef font = CTFontCreateWithFontDescriptor( m_ctFontDescriptor, m_pointSize, NULL );
+                m_ctFont.reset( CTFontCreateCopyWithSymbolicTraits( font, m_pointSize, NULL, traits, 0x03 ) );
+                CFRelease(font);
                 fontcache[ std::wstring(lookupnameWithSize.wc_str()) ] = m_ctFont;
             }
 #else
             m_ctFont.reset( CTFontCreateWithFontDescriptor( m_ctFontDescriptor, m_pointSize, NULL ) );
 #endif
+#if 0
             if ( /* (CTFontGetSymbolicTraits( m_ctFont ) & 0x03) !=*/ traits )
             {
                 CTFontRef font = CTFontCreateWithName( cf, m_pointSize,  NULL );
@@ -459,6 +462,7 @@
                 }
 #endif
             }
+#endif
         }
 #if wxOSX_USE_ATSU_TEXT
         OSStatus status = noErr;
