diff --git a/include/wx/mdi.h b/include/wx/mdi.h
index 8bf2cd698c..01ccc8db5a 100644
--- a/include/wx/mdi.h
+++ b/include/wx/mdi.h
@@ -320,6 +320,12 @@ protected:
     {
         wxWindow::DoMoveWindow(x, y, width, height);
     }
+	
+	virtual void DoGetScreenPosition(int *x, int *y) const wxOVERRIDE
+    {
+        wxWindow::DoGetScreenPosition(x, y);
+    }
+
 
     // no size hints
     virtual void DoSetSizeHints(int WXUNUSED(minW), int WXUNUSED(minH),
