Index: src/aui/auibar.cpp
===================================================================
--- src/aui/auibar.cpp	(revision 61959)
+++ src/aui/auibar.cpp	(working copy)
@@ -38,7 +38,7 @@
 
 #ifdef __WXMAC__
 #include "wx/osx/private.h"
-// for themeing support
+// for theming support
 #include <Carbon/Carbon.h>
 #endif
 
Index: src/aui/dockart.cpp
===================================================================
--- src/aui/dockart.cpp	(revision 61959)
+++ src/aui/dockart.cpp	(working copy)
@@ -38,7 +38,7 @@
 #include "wx/osx/private.h"
 #include "wx/graphics.h"
 #include "wx/dcgraph.h"
-// for themeing support
+// for theming support
 #include <Carbon/Carbon.h>
 #endif
 
Index: src/aui/auibook.cpp
===================================================================
--- src/aui/auibook.cpp	(revision 61959)
+++ src/aui/auibook.cpp	(working copy)
@@ -35,7 +35,7 @@
 
 #ifdef __WXMAC__
 #include "wx/osx/private.h"
-// for themeing support
+// for theming support
 #include <Carbon/Carbon.h>
 #endif
 
Index: src/osx/window_osx.cpp
===================================================================
--- src/osx/window_osx.cpp	(revision 61959)
+++ src/osx/window_osx.cpp	(working copy)
@@ -68,7 +68,7 @@
 #else
 #include "wx/osx/private.h"
 #if wxOSX_USE_COCOA
-// bring in themeing
+// bring in theming
 #include <Carbon/Carbon.h>
 #endif
 #endif
Index: src/osx/utils_osx.cpp
===================================================================
--- src/osx/utils_osx.cpp	(revision 61959)
+++ src/osx/utils_osx.cpp	(working copy)
@@ -42,7 +42,7 @@
 #include "wx/osx/private.h"
 
 #if wxOSX_USE_COCOA
-// to get the themeing APIs
+// to get the theming APIs
 #include <Carbon/Carbon.h>
 #endif
 
Index: src/osx/carbon/window.cpp
===================================================================
--- src/osx/carbon/window.cpp	(revision 61959)
+++ src/osx/carbon/window.cpp	(working copy)
@@ -65,7 +65,7 @@
 #include "wx/osx/uma.h"
 #else
 #include "wx/osx/private.h"
-// bring in themeing
+// bring in theming
 #include <Carbon/Carbon.h>
 #endif
 
Index: src/osx/carbon/font.cpp
===================================================================
--- src/osx/carbon/font.cpp	(revision 61959)
+++ src/osx/carbon/font.cpp	(working copy)
@@ -29,7 +29,7 @@
 #include "wx/osx/private.h"
 
 #if wxOSX_USE_ATSU_TEXT && !wxOSX_USE_CARBON
-// include themeing support
+// include theming support
 #include <Carbon/Carbon.h>
 #endif
 
@@ -163,7 +163,7 @@
 public:
     bool            m_fontValid;
 #if wxOSX_USE_CARBON && wxOSX_USE_ATSU_TEXT
-    // for true themeing support we must store the correct font
+    // for true theming support we must store the correct font
     // information here, as this speeds up and optimizes rendering
     ThemeFontID     m_macThemeFontID ;
 #endif
@@ -346,7 +346,7 @@
                 m_macThemeFontID = kThemeMiniSystemFont;
                 break;
            case wxOSX_SYSTEM_FONT_MINI_BOLD:
-                // bold not available under themeing
+                // bold not available under theming
                 m_macThemeFontID = kThemeMiniSystemFont;
                 break;
             case wxOSX_SYSTEM_FONT_LABELS:
Index: src/generic/listctrl.cpp
===================================================================
--- src/generic/listctrl.cpp	(revision 61959)
+++ src/generic/listctrl.cpp	(working copy)
@@ -52,7 +52,7 @@
 
 #ifdef __WXMAC__
     #include "wx/osx/private.h"
-    // for themeing support
+    // for theming support
     #include <Carbon/Carbon.h>
 #endif
 
Index: include/wx/osx/cocoa/private.h
===================================================================
--- include/wx/osx/cocoa/private.h	(revision 61959)
+++ include/wx/osx/cocoa/private.h	(working copy)
@@ -33,7 +33,7 @@
 // shared between Cocoa and Carbon
 //
 
-// bring in themeing types without pulling in the headers
+// bring in theming types without pulling in the headers
 
 #if wxUSE_GUI
 typedef SInt16 ThemeBrush;
