--- wxWidgets-2.6.2/src/common/object.cpp.orig	2006-03-24 13:33:38.545899250 +0100
+++ wxWidgets-2.6.2/src/common/object.cpp	2006-03-24 13:35:12.367762750 +0100
@@ -214,11 +214,7 @@
 
 void wxClassInfo::CleanUp()
 {
-    if ( sm_classTable )
-    {
-        delete sm_classTable;
-        sm_classTable = NULL;
-    }
+    /* Removed */
 }
 
 void wxClassInfo::Register()
--- wxWidgets-2.6.2/src/common/init.cpp~	2005-08-13 02:24:13.000000000 +0200
+++ wxWidgets-2.6.2/src/common/init.cpp	2006-03-24 13:34:03.755474750 +0100
@@ -356,8 +356,6 @@
 {
     wxModule::CleanUpModules();
 
-    wxClassInfo::CleanUp();
-
     // we can't do this in wxApp itself because it doesn't know if argv had
     // been allocated
 #if wxUSE_UNICODE

 	  	 
