Index: src/common/pickerbase.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/common/pickerbase.cpp,v
retrieving revision 1.12
diff -u -r1.12 pickerbase.cpp
--- src/common/pickerbase.cpp	2006/10/30 18:07:37	1.12
+++ src/common/pickerbase.cpp	2006/12/26 17:39:18
@@ -30,11 +30,13 @@
     wxUSE_FONTPICKERCTRL
 
 #include "wx/pickerbase.h"
+#include "wx/tooltip.h"
 
 #ifndef WX_PRECOMP
     #include "wx/textctrl.h"
 #endif
 
+
 // ============================================================================
 // implementation
 // ============================================================================
@@ -121,6 +123,23 @@
     SetSizer(m_sizer);
     SetMinSize( m_sizer->GetMinSize() );
 }
+
+void wxPickerBase::DoSetToolTip( wxToolTip *tip )
+{
+    // don't set the tooltip on us but rather on our two child windows
+    // as otherwise it would appear only when the cursor is placed on the
+    // small area around the child windows which belong to wxPickerBase
+    m_picker->SetToolTip(tip);
+
+    // do a copy as wxWindow will own the pointer we pass
+    m_text->SetToolTip(tip ? new wxToolTip(tip->GetTip()) : NULL);
+}
+
+
+
+// ----------------------------------------------------------------------------
+// wxPickerBase - event handlers
+// ----------------------------------------------------------------------------
 
 void wxPickerBase::OnTextCtrlKillFocus(wxFocusEvent &)
 {
Index: include/wx/pickerbase.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/pickerbase.h,v
retrieving revision 1.10
diff -u -r1.10 pickerbase.h
--- include/wx/pickerbase.h	2006/09/04 16:39:56	1.10
+++ include/wx/pickerbase.h	2006/12/26 17:38:47
@@ -17,6 +17,7 @@
 #include "wx/containr.h"
 
 class WXDLLIMPEXP_CORE wxTextCtrl;
+class WXDLLEXPORT wxToolTip;
 
 extern WXDLLEXPORT_DATA(const wxChar) wxButtonNameStr[];
 
@@ -51,6 +52,7 @@
                     const wxValidator& validator = wxDefaultValidator,
                     const wxString& name = wxButtonNameStr);
 
+    virtual void DoSetToolTip( wxToolTip *tip );
 
 public:     // public API
 

 	  	 
