diff -aburN ../original/docs/latex/wx/dc.tex ./docs/latex/wx/dc.tex
--- ../original/docs/latex/wx/dc.tex	2007-09-06 10:58:06.000000000 +0200
+++ ./docs/latex/wx/dc.tex	2007-09-06 13:34:49.000000000 +0200
@@ -484,6 +484,24 @@
 the rectangle.
 
 
+docs/latex/wx/dc.tex
+
+\membersection{wxDC::DrawScaledBitmap}\label{wxdcdrawscaledbitmap}
+
+\func{void}{DrawScaledBitmap}{\param{const wxBitmap\&}{ bitmap}, \param{wxCoord}{ x}, \param{wxCoord}{ y}, \param{wxCoord}{ w}, \param{wxCoord}{ h}, \param{bool}{ transparent}, \param{int}{ quality = wxIMAGE_QUALITY_NORMAL}}
+
+Draw a bitmap on the device context at the specified point and resize it to the specified size. If {\it transparent} is true and the bitmap has a transparency mask, the bitmap will be drawn transparently.
+
+Under wxGTK with Gtk Printing support, you should use this method with a wxPrinterDC if you want to print an image with a resolution higher than 72 dpi. On every other DC and platform, resizing is done using \helpref{wxImage::Rescale}{wximagerescale}.
+
+For a description of the {\it quality} parameter, see the \helpref{Scale}{wximagescale} function.
+
+When drawing a mono-bitmap, the current text foreground colour will be used to draw the foreground
+of the bitmap (all bits set to 1), and the current text background colour to draw the background
+(all bits set to 0). See also \helpref{SetTextForeground}{wxdcsettextforeground},
+\helpref{SetTextBackground}{wxdcsettextbackground} and \helpref{wxMemoryDC}{wxmemorydc}.
+
+
 \membersection{wxDC::DrawSpline}\label{wxdcdrawspline}
 
 \func{void}{DrawSpline}{\param{int }{n}, \param{wxPoint }{points[]}}
diff -aburN ../original/include/wx/dc.h ./include/wx/dc.h
--- ../original/include/wx/dc.h	2007-09-06 10:59:46.000000000 +0200
+++ ./include/wx/dc.h	2007-09-06 13:36:40.000000000 +0200
@@ -873,6 +873,49 @@
                     bool useMask = false)
         { m_pimpl->DoDrawBitmap(bmp, pt.x, pt.y, useMask); }
 
+    virtual void DrawScaledBitmap(const wxBitmap &bmp, wxCoord x, wxCoord y, wxCoord w, wxCoord h, bool useMask = false, int quality = wxIMAGE_QUALITY_NORMAL)
+        {
+            if (bmp.GetWidth() != w || bmp.GetHeight() != h)
+            {
+                if (quality != wxIMAGE_QUALITY_HIGH)
+                    quality = wxIMAGE_QUALITY_NORMAL;
+                wxImage tmpImg = bmp.ConvertToImage();
+                tmpImg.Rescale( w, h, quality );
+                wxBitmap scaledBmp(tmpImg);
+                m_pimpl->DoDrawBitmap(scaledBmp, x, y, useMask);
+            }
+            else
+                m_pimpl->DoDrawBitmap(bmp, x, y, useMask);
+        }
+    virtual void DrawScaledBitmap(const wxBitmap &bmp, const wxPoint& pt, const wxSize& sz, bool useMask = false, int quality = wxIMAGE_QUALITY_NORMAL)
+        {
+            if (bmp.GetWidth() != sz.x || bmp.GetHeight() != sz.y)
+            {
+                if (quality != wxIMAGE_QUALITY_HIGH)
+                    quality = wxIMAGE_QUALITY_NORMAL;
+                wxImage tmpImg = bmp.ConvertToImage();
+                tmpImg.Rescale( sz.x, sz.y, quality );
+                wxBitmap scaledBmp(tmpImg);
+                m_pimpl->DoDrawBitmap(scaledBmp, pt.x, pt.y, useMask);
+            }
+            else
+                m_pimpl->DoDrawBitmap(bmp, pt.x, pt.y, useMask);
+        }
+    virtual void DrawScaledBitmap(const wxBitmap &bmp, const wxRect& rect, bool useMask = false, int quality = wxIMAGE_QUALITY_NORMAL)
+        {
+            if (bmp.GetWidth() != rect.width || bmp.GetHeight() != rect.height)
+            {
+                if (quality != wxIMAGE_QUALITY_HIGH)
+                    quality = wxIMAGE_QUALITY_NORMAL;
+                wxImage tmpImg = bmp.ConvertToImage();
+                tmpImg.Rescale( rect.width, rect.height, quality );
+                wxBitmap scaledBmp(tmpImg);
+                m_pimpl->DoDrawBitmap(scaledBmp, rect.x, rect.y, useMask);
+            }
+            else
+                m_pimpl->DoDrawBitmap(bmp, rect.x, rect.y, useMask);
+        }
+
     void DrawText(const wxString& text, wxCoord x, wxCoord y)
         { m_pimpl->DoDrawText(text, x, y); }
     void DrawText(const wxString& text, const wxPoint& pt)
@@ -1189,6 +1232,49 @@
                     bool useMask = false)
         { DoDrawBitmap(bmp, pt.x, pt.y, useMask); }
 
+    virtual void DrawScaledBitmap(const wxBitmap &bmp, wxCoord x, wxCoord y, wxCoord w, wxCoord h, bool useMask = false, int quality = wxIMAGE_QUALITY_NORMAL)
+        {
+            if (bmp.GetWidth() != w || bmp.GetHeight() != h)
+            {
+                if (quality != wxIMAGE_QUALITY_HIGH)
+                    quality = wxIMAGE_QUALITY_NORMAL;
+                wxImage tmpImg = bmp.ConvertToImage();
+                tmpImg.Rescale( w, h, quality );
+                wxBitmap scaledBmp(tmpImg);
+                DoDrawBitmap(scaledBmp, x, y, useMask);
+            }
+            else
+                DoDrawBitmap(bmp, x, y, useMask);
+        }
+    virtual void DrawScaledBitmap(const wxBitmap &bmp, const wxPoint& pt, const wxSize& sz, bool useMask = false, int quality = wxIMAGE_QUALITY_NORMAL)
+        {
+            if (bmp.GetWidth() != sz.x || bmp.GetHeight() != sz.y)
+            {
+                if (quality != wxIMAGE_QUALITY_HIGH)
+                    quality = wxIMAGE_QUALITY_NORMAL;
+                wxImage tmpImg = bmp.ConvertToImage();
+                tmpImg.Rescale( sz.x, sz.y, quality );
+                wxBitmap scaledBmp(tmpImg);
+                DoDrawBitmap(scaledBmp, pt.x, pt.y, useMask);
+            }
+            else
+                DoDrawBitmap(bmp, pt.x, pt.y, useMask);
+        }
+    virtual void DrawScaledBitmap(const wxBitmap &bmp, const wxRect& rect, bool useMask = false, int quality = wxIMAGE_QUALITY_NORMAL)
+        {
+            if (bmp.GetWidth() != rect.width || bmp.GetHeight() != rect.height)
+            {
+                if (quality != wxIMAGE_QUALITY_HIGH)
+                    quality = wxIMAGE_QUALITY_NORMAL;
+                wxImage tmpImg = bmp.ConvertToImage();
+                tmpImg.Rescale( rect.width, rect.height, quality );
+                wxBitmap scaledBmp(tmpImg);
+                DoDrawBitmap(scaledBmp, rect.x, rect.y, useMask);
+            }
+            else
+                DoDrawBitmap(bmp, rect.x, rect.y, useMask);
+        }
+
     void DrawText(const wxString& text, wxCoord x, wxCoord y)
         { DoDrawText(text, x, y); }
     void DrawText(const wxString& text, const wxPoint& pt)

 	  	 
