diff -aburN ../original/src/common/cairo.cpp ./src/common/cairo.cpp
--- ../original/src/common/cairo.cpp	2007-09-06 20:43:38.000000000 +0200
+++ ./src/common/cairo.cpp	2007-09-06 20:46:16.000000000 +0200
@@ -44,7 +44,7 @@
     if (!m_ok) return;
 
     m_pango_cairo_lib = new wxDynamicLibrary( wxT("libpangocairo-1.0.so.0") );
-    m_ok = m_cairo_lib->IsLoaded();
+    m_ok = m_pango_cairo_lib->IsLoaded();
     if (!m_ok) return;
 
     InitializeMethods();
@@ -150,7 +150,7 @@
     void OnExit();
 
 private:
-    DECLARE_DYNAMIC_CLASS(wxGtkPrintModule)
+    DECLARE_DYNAMIC_CLASS(wxCairoModule)
 };
 
 bool wxCairoModule::OnInit()

 	  	 
