diff -rbu6 wx-2.4.0-cvs030301/filedlgg.cpp wx-2.4.0-cvs030301-patch/filedlgg.cpp
--- wx-2.4.0-cvs030301/filedlgg.cpp	Sun Mar 02 20:12:54 2003
+++ wx-2.4.0-cvs030301-patch/filedlgg.cpp	Sun Mar 02 20:34:48 2003
@@ -1219,12 +1219,21 @@
     {
         wxConfig::Get()->Write(wxT("/wxWindows/wxFileDialog/ViewStyle"),
                                ms_lastViewStyle);
         wxConfig::Get()->Write(wxT("/wxWindows/wxFileDialog/ShowHidden"),
                                ms_lastShowHidden);
     }
+
+    for ( int i = 0; i < m_choice->GetCount(); i++ )
+    {
+        wxString *str = (wxString*) m_choice->GetClientData(i);
+        if ( str )
+        {
+            delete str;
+        }
+      }
 }
 
 int wxFileDialog::ShowModal()
 {
     m_list->GoToDir(m_dir);
     m_text->SetValue(m_fileName);

 	  	 
