Index: samples/calendar/calendar.cpp
===================================================================
--- samples/calendar/calendar.cpp	(revision 61605)
+++ samples/calendar/calendar.cpp	(working copy)
@@ -799,6 +799,11 @@
         if ( m_calendar->SetDateRange(today - diff, today + diff) )
         {
             wxLogStatus("Date range limited to 3 months around today.");
+            wxDateTime firstValidDate;
+            wxDateTime lastValidDate;
+            m_calendar->GetDateRange(&firstValidDate, &lastValidDate);
+            wxLogMessage(wxT("First valid date: %s, Last valid date: %s"), 
+               firstValidDate.FormatISODate().c_str(), lastValidDate.FormatISODate().c_str());
         }
         else
         {
