Index: interface/wx/msw/registry.h
===================================================================
--- interface/wx/msw/registry.h	(revision 70611)
+++ interface/wx/msw/registry.h	(working copy)
@@ -288,7 +288,7 @@
     /**
         Returns @true if any subkeys exist.
     */
-    bool HasSubKeys() const;
+    bool HasSubkeys() const;
 
     /**
         Returns @true if the value exists.
