Index: clipbrd.cpp
===================================================================
--- clipbrd.cpp	(revision 69626)
+++ clipbrd.cpp	(working copy)
@@ -157,10 +157,10 @@
 
     wxCHECK_MSG( m_open, false, wxT("clipboard not open") );
 
-    size_t formatcount = data.GetFormatCount() + 1;
+    size_t formatcount = data.GetFormatCount(wxDataObject::Set) + 1;
     wxDataFormat *array = new wxDataFormat[ formatcount ];
-    array[0] = data.GetPreferredFormat();
-    data.GetAllFormats( &array[1] );
+    array[0] = data.GetPreferredFormat(wxDataObject::Set);
+    data.GetAllFormats( &array[1], wxDataObject::Set );
 
     bool transferred = false;
 
