Index: samples/controls/controls.cpp
===================================================================
RCS file: /home/wxcvs/wxWindows/samples/controls/controls.cpp,v
retrieving revision 1.175
diff -u -2 -r1.175 controls.cpp
--- samples/controls/controls.cpp	2001/06/30 11:07:41	1.175
+++ samples/controls/controls.cpp	2001/07/30 16:07:22
@@ -285,5 +285,5 @@
 //----------------------------------------------------------------------
 
-static void SetControlClientData(const char *name,
+static void SetControlClientData(const wxChar *name,
                                  wxControlWithItems *control);
 
@@ -316,6 +316,6 @@
     if ( argc == 3 )
     {
-        wxSscanf(argv[1], "%d", &x);
-        wxSscanf(argv[2], "%d", &y);
+        //wxSscanf(argv[1], "%d", &x);
+        //wxSscanf(argv[2], "%d", &y);
     }
 
@@ -570,5 +570,5 @@
         if ( !bmp.Ok() || (imagelist->Add(bmp) == -1) )
         {
-            wxLogWarning("Couldn't load the image '%s' for the notebook page %d.",
+            wxLogWarning(wxT("Couldn't load the image '%s' for the notebook page %d."),
                     s_iconNames[n], n);
         }
@@ -597,6 +597,6 @@
                                      5, choices, wxLB_SORT );
 
-    SetControlClientData("listbox", m_listbox);
-    SetControlClientData("listbox", m_listboxSorted);
+    SetControlClientData(wxT("listbox"), m_listbox);
+    SetControlClientData(wxT("listbox"), m_listboxSorted);
 
     m_listbox->SetCursor(*wxCROSS_CURSOR);
@@ -635,6 +635,6 @@
                                    5, choices, wxCB_SORT );
 
-    SetControlClientData("choice", m_choice);
-    SetControlClientData("choice", m_choiceSorted);
+    SetControlClientData(wxT("choice"), m_choice);
+    SetControlClientData(wxT("choice"), m_choiceSorted);
 
     m_choice->SetSelection(2);
@@ -1505,5 +1505,5 @@
 }
 
-static void SetControlClientData(const char *name,
+static void SetControlClientData(const wxChar *name,
                                  wxControlWithItems *control)
 {
@@ -1512,5 +1512,5 @@
     {
         wxString s;
-        s.Printf("%s client data for '%s'",
+        s.Printf(wxT("%s client data for '%s'"),
                  name, control->GetString(n).c_str());
 
Index: samples/listctrl/listtest.cpp
===================================================================
RCS file: /home/wxcvs/wxWindows/samples/listctrl/listtest.cpp,v
retrieving revision 1.48
diff -u -2 -r1.48 listtest.cpp
--- samples/listctrl/listtest.cpp	2001/07/23 14:57:01	1.48
+++ samples/listctrl/listtest.cpp	2001/07/30 16:07:26
@@ -114,5 +114,5 @@
 {
   // Create the main frame window
-  MyFrame *frame = new MyFrame("wxListCtrl Test", 50, 50, 450, 340);
+  MyFrame *frame = new MyFrame(wxT("wxListCtrl Test"), 50, 50, 450, 340);
 
   // Show the frame
@@ -484,8 +484,8 @@
 {
     int count = m_listCtrl->GetColumnCount();
-    wxLogMessage("%d columns:", count);
+    wxLogMessage(wxT("%d columns:"), count);
     for ( int c = 0; c < count; c++ )
     {
-        wxLogMessage("\tcolumn %d has width %d", c,
+        wxLogMessage(wxT("\tcolumn %d has width %d"), c,
                      m_listCtrl->GetColumnWidth(c));
     }
@@ -505,5 +505,5 @@
         flags |= wxLC_SINGLE_SEL;
 
-    m_logWindow->WriteText(wxString::Format("Current selection mode: %sle\n",
+    m_logWindow->WriteText(wxString::Format(wxT("Current selection mode: %sle\n"),
                            (flags & wxLC_SINGLE_SEL) ? "sing" : "multip"));
 
@@ -555,5 +555,5 @@
 void MyListCtrl::OnCacheHint(wxListEvent& event)
 {
-    wxLogMessage( "OnCacheHint: cache items %ld..%ld",
+    wxLogMessage( wxT("OnCacheHint: cache items %ld..%ld"),
                   event.GetCacheFrom(), event.GetCacheTo() );
 }
@@ -561,10 +561,10 @@
 void MyListCtrl::OnColClick(wxListEvent& event)
 {
-    wxLogMessage( "OnColumnClick at %d.", event.GetColumn() );
+    wxLogMessage( wxT("OnColumnClick at %d."), event.GetColumn() );
 }
 
 void MyListCtrl::OnBeginDrag(wxListEvent& event)
 {
-    wxLogMessage( "OnBeginDrag at %d,%d.",
+    wxLogMessage( wxT("OnBeginDrag at %d,%d."),
                   event.m_pointDrag.x, event.m_pointDrag.y );
 }
@@ -572,5 +572,5 @@
 void MyListCtrl::OnBeginRDrag(wxListEvent& event)
 {
-    wxLogMessage( "OnBeginRDrag at %d,%d.",
+    wxLogMessage( wxT("OnBeginRDrag at %d,%d."),
                   event.m_pointDrag.x, event.m_pointDrag.y );
 }
@@ -578,10 +578,10 @@
 void MyListCtrl::OnBeginLabelEdit(wxListEvent& event)
 {
-    wxLogMessage("OnBeginLabelEdit: %s", event.m_item.m_text.c_str());
+    wxLogMessage( wxT("OnBeginLabelEdit: %s"), event.m_item.m_text.c_str());
 }
 
 void MyListCtrl::OnEndLabelEdit(wxListEvent& event)
 {
-    wxLogMessage("OnEndLabelEdit: %s", event.m_item.m_text.c_str());
+    wxLogMessage( wxT("OnEndLabelEdit: %s"), event.m_item.m_text.c_str());
 }
 
@@ -641,10 +641,10 @@
         if ( GetItem(info) )
         {
-            wxLogMessage("Value of the 2nd field of the selected item: %s",
+            wxLogMessage(wxT("Value of the 2nd field of the selected item: %s"),
                          info.m_text.c_str());
         }
         else
         {
-            wxFAIL_MSG("wxListCtrl::GetItem() failed");
+            wxFAIL_MSG(wxT("wxListCtrl::GetItem() failed"));
         }
     }


 	  	 
