Index: src/gtk/dialog.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/gtk/dialog.cpp,v
retrieving revision 1.117
diff -u -r1.117 dialog.cpp
--- src/gtk/dialog.cpp	2006/09/07 16:06:45	1.117
+++ src/gtk/dialog.cpp	2006/12/29 16:19:06
@@ -118,7 +118,7 @@
         wxWindow *parent = wxTheApp->GetTopWindow();
         if ( parent &&
                 parent != this &&
-                    parent->IsBeingDeleted() &&
+                    !parent->IsBeingDeleted() &&
                         !(parent->GetExtraStyle() & wxWS_EX_TRANSIENT) )
         {
             m_parent = parent;
Index: src/gtk1/dialog.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/gtk1/dialog.cpp,v
retrieving revision 1.115
diff -u -r1.115 dialog.cpp
--- src/gtk1/dialog.cpp	2006/05/22 07:11:13	1.115
+++ src/gtk1/dialog.cpp	2006/12/29 16:19:19
@@ -194,7 +194,7 @@
         wxWindow *parent = wxTheApp->GetTopWindow();
         if ( parent &&
                 parent != this &&
-                    parent->IsBeingDeleted() &&
+                    !parent->IsBeingDeleted() &&
                         !(parent->GetExtraStyle() & wxWS_EX_TRANSIENT) )
         {
             m_parent = parent;

 	  	 
