diff -bu1 -P -r wxWidgets/.\include\wx\features.h wxWsmart/.\include\wx\features.h
--- wxWidgets/.\include\wx\features.h	Thu Mar 04 15:33:20 2004
+++ wxWsmart/.\include\wx\features.h	Sat Apr 17 23:34:50 2004
@@ -31,10 +31,2 @@
 
-/*  detect SmartPhone */
-#ifdef WIN32_PLATFORM_WFSP
-    #ifdef wxUSE_SMARTPHONE
-    #undef wxUSE_SMARTPHONE
-    #endif
-    #define wxUSE_SMARTPHONE 1
-#endif
-
 /*  taskbar is only implemented in wxMSW and X11 ports */
diff -bu1 -P -r wxWidgets/.\include\wx\msw\chkconf.h wxWsmart/.\include\wx\msw\chkconf.h
--- wxWidgets/.\include\wx\msw\chkconf.h	Wed Dec 17 20:30:26 2003
+++ wxWsmart/.\include\wx\msw\chkconf.h	Sun Apr 18 19:39:46 2004
@@ -115,2 +115,10 @@
 
+/* Hopefully we can emulate these dialogs in due course */
+#ifdef __SMARTPHONE__
+#   ifdef wxUSE_COLOURDLG
+#       undef wxUSE_COLOURDLG
+#       define wxUSE_COLOURDLG 0
+#   endif
+#endif /* __SMARTPHONE__ */
+
 #endif
diff -bu1 -P -r wxWidgets/.\include\wx\msw\setup0.h wxWsmart/.\include\wx\msw\setup0.h
--- wxWidgets/.\include\wx\msw\setup0.h	Tue Mar 09 23:30:08 2004
+++ wxWsmart/.\include\wx\msw\setup0.h	Sat Apr 17 23:36:04 2004
@@ -1072,5 +1072,2 @@
 
-// Set to 1 to if you're developing for MS SmartPhone
-#define wxUSE_SMARTPHONE        0
-
 // ----------------------------------------------------------------------------
diff -bu1 -P -r wxWidgets/.\include\wx\msw\wince\setup.h wxWsmart/.\include\wx\msw\wince\setup.h
--- wxWidgets/.\include\wx\msw\wince\setup.h	Thu Apr 08 16:32:48 2004
+++ wxWsmart/.\include\wx\msw\wince\setup.h	Sat Apr 17 23:30:54 2004
@@ -1063,5 +1063,2 @@
 
-// Set to 1 if you're developing for MS SmartPhone
-#define wxUSE_SMARTPHONE        0
-
 // Set to 1 if you want to maintain a PocketPC-style UI on
diff -bu1 -P -r wxWidgets/.\include\wx\platform.h wxWsmart/.\include\wx\platform.h
--- wxWidgets/.\include\wx\platform.h	Fri Apr 16 08:43:46 2004
+++ wxWsmart/.\include\wx\platform.h	Sun Apr 18 19:38:32 2004
@@ -119,2 +119,9 @@
 
+/* detect SmartPhone */
+#if defined( WIN32_PLATFORM_WFSP )
+#    ifndef __SMARTPHONE__
+#        define __SMARTPHONE__
+#    endif
+#endif
+
 /*
diff -bu1 -P -r wxWidgets/.\src\common\cmndata.cpp wxWsmart/.\src\common\cmndata.cpp
--- wxWidgets/.\src\common\cmndata.cpp	Sun Mar 28 15:09:58 2004
+++ wxWsmart/.\src\common\cmndata.cpp	Sun Apr 18 19:44:56 2004
@@ -53,3 +53,3 @@
 
-    #if defined(__WXWINCE__) && !wxUSE_SMARTPHONE
+    #ifndef __SMARTPHONE__
         #include <commdlg.h>
diff -bu1 -P -r wxWidgets/.\src\msw\colordlg.cpp wxWsmart/.\src\msw\colordlg.cpp
--- wxWidgets/.\src\msw\colordlg.cpp	Tue Feb 10 14:02:50 2004
+++ wxWsmart/.\src\msw\colordlg.cpp	Sun Apr 18 19:22:38 2004
@@ -44,3 +44,3 @@
 
-#if wxUSE_COLOURDLG && !wxUSE_SMARTPHONE
+#if wxUSE_COLOURDLG
 
diff -bu1 -P -r wxWidgets/.\src\msw\filedlg.cpp wxWsmart/.\src\msw\filedlg.cpp
--- wxWidgets/.\src\msw\filedlg.cpp	Wed Jan 21 14:41:06 2004
+++ wxWsmart/.\src\msw\filedlg.cpp	Sat Apr 17 23:44:00 2004
@@ -30,3 +30,3 @@
 
-#if wxUSE_FILEDLG && !wxUSE_SMARTPHONE
+#if wxUSE_FILEDLG && !defined(__SMARTPHONE__)
 
diff -bu1 -P -r wxWidgets/.\src\msw\wince\filedlgwce.cpp wxWsmart/.\src\msw\wince\filedlgwce.cpp
--- wxWidgets/.\src\msw\wince\filedlgwce.cpp	Fri Apr 16 08:44:38 2004
+++ wxWsmart/.\src\msw\wince\filedlgwce.cpp	Sat Apr 17 23:43:34 2004
@@ -33,3 +33,3 @@
 
-#if wxUSE_FILEDLG && wxUSE_SMARTPHONE
+#if wxUSE_FILEDLG && defined(__SMARTPHONE__)
 
diff -bu1 -P -r wxWidgets/.\src\msw\wince\tbarwce.cpp wxWsmart/.\src\msw\wince\tbarwce.cpp
--- wxWidgets/.\src\msw\wince\tbarwce.cpp	Mon Apr 05 01:17:40 2004
+++ wxWsmart/.\src\msw\wince\tbarwce.cpp	Sat Apr 17 23:47:20 2004
@@ -261,3 +261,3 @@
     mbi.hwndParent = (HWND) GetParent()->GetHWND();
-#if wxUSE_SMARTPHONE
+#ifdef __SMARTPHONE__
     mbi.nToolBarId = 5002;

 	  	 
