diff --recursive '--unified=2' wxWidgets-3.0.0.original/src/gtk/sockgtk.cpp wxWidgets-3.0.0/src/gtk/sockgtk.cpp
--- wxWidgets-3.0.0.original/src/gtk/sockgtk.cpp	2013-11-11 21:10:39.000000000 +0800
+++ wxWidgets-3.0.0/src/gtk/sockgtk.cpp	2013-12-29 20:22:50.000000000 +0800
@@ -47,6 +47,7 @@
     virtual int AddInput(wxFDIOHandler *handler, int fd, Direction d)
     {
+        m_giochannel = g_io_channel_unix_new(fd);
         return g_io_add_watch(
-            g_io_channel_unix_new(fd),
+            m_giochannel,
             d == OUTPUT ? G_IO_OUT : G_IO_IN,
             wxSocket_Input,
@@ -57,6 +58,9 @@
     RemoveInput(wxFDIOHandler* WXUNUSED(handler), int fd, Direction WXUNUSED(d))
     {
+        g_io_channel_unref(m_giochannel);
         g_source_remove(fd);
     }
+private:
+    GIOChannel * m_giochannel;
 };
 
