diff --git a/src/gtk/animate.cpp b/src/gtk/animate.cpp
index a36a16f..ef4a16d 100644
--- a/src/gtk/animate.cpp
+++ b/src/gtk/animate.cpp
@@ -137,6 +137,7 @@ bool wxAnimation::Load(wxInputStream &stream, wxAnimationType type)
         {
             // gdk_pixbuf_loader_close wants the GError == NULL
             gdk_pixbuf_loader_close(loader, NULL);
+            g_object_unref(loader);
             return false;
         }
 
@@ -147,6 +148,7 @@ bool wxAnimation::Load(wxInputStream &stream, wxAnimationType type)
 
             // gdk_pixbuf_loader_close wants the GError == NULL
             gdk_pixbuf_loader_close(loader, NULL);
+            g_object_unref(loader);
             return false;
         }
 
@@ -156,6 +158,8 @@ bool wxAnimation::Load(wxInputStream &stream, wxAnimationType type)
     if (!data_written)
     {
         wxLogDebug("Could not read data from the stream...");
+        gdk_pixbuf_loader_close(loader, NULL);
+        g_object_unref(loader);
         return false;
     }
 
@@ -165,9 +169,12 @@ bool wxAnimation::Load(wxInputStream &stream, wxAnimationType type)
     if (!gdk_pixbuf_loader_close(loader, &error))
     {
         wxLogDebug(wxT("Could not close the loader: %s"), error->message);
+        g_object_unref(loader);
         return false;
     }
 
+    g_object_unref(loader);
+
     // wait until we get the last area_updated signal
     return data_written;
 }
