--- include/wx/stream.h.old	2007-02-02 21:56:46.000000000 +0300
+++ include/wx/stream.h	2007-03-23 17:57:42.000000000 +0300
@@ -240,6 +240,9 @@
 
     virtual size_t LastWrite() const { return wxStreamBase::m_lastcount; }
 
+    virtual void Truncate(void) {}
+    virtual void Reset(void);
+
     virtual void Sync();
     virtual bool Close() { return true; }
 
--- include/wx/mstream.h.old1	2007-03-16 15:34:32.000000000 +0300
+++ include/wx/mstream.h	2007-03-23 18:15:31.000000000 +0300
@@ -64,6 +64,8 @@
     virtual wxFileOffset GetLength() const { return m_o_streambuf->GetLastAccess(); }
     virtual bool IsSeekable() const { return true; }
 
+    virtual void Truncate(void);
+
     size_t CopyTo(void *buffer, size_t len) const;
 
     wxStreamBuffer *GetOutputStreamBuffer() const { return m_o_streambuf; }
--- src/common/mstream.cpp.old1	2007-03-16 15:34:32.000000000 +0300
+++ src/common/mstream.cpp	2007-03-23 18:15:52.000000000 +0300
@@ -205,4 +205,9 @@
     return len;
 }
 
+void wxMemoryOutputStream::Truncate(void)
+{
+    m_o_streambuf->Truncate();
+}
+
 #endif // wxUSE_STREAMS
--- src/common/stream.cpp.old	2007-02-12 21:45:13.000000000 +0300
+++ src/common/stream.cpp	2007-03-23 18:06:17.000000000 +0300
@@ -1005,6 +1005,12 @@
 {
 }
 
+void wxOutputStream::Reset(void)
+{
+    SeekO(0);
+    Truncate();
+}
+
 
 // ----------------------------------------------------------------------------
 // wxCountingOutputStream
--- docs/latex/wx/outptstr.tex.old	2005-01-20 21:47:03.000000000 +0300
+++ docs/latex/wx/outptstr.tex	2007-03-23 18:51:38.000000000 +0300
@@ -46,6 +46,20 @@
 by this stream, or left open otherwise.
 
 
+\membersection{wxOutputStream::Truncate}\label{wxoutputstreamtruncate}
+
+\func{void}{Truncate}{\void}
+
+Truncates the stream to the current position.
+
+\membersection{wxOutputStream::Reset}\label{wxoutputstreamreset}
+
+\func{void}{Reset}{\void}
+
+Resets the stream by seeking it to the start position and destroying its
+content.
+
+
 \membersection{wxOutputStream::LastWrite}\label{wxoutputstreamlastwrite}
 
 \constfunc{size\_t}{LastWrite}{\void}

 	  	 
