Index: Makefile.in
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/Makefile.in,v
retrieving revision 1.220
diff -b -u -2 -r1.220 Makefile.in
--- Makefile.in	2003/02/05 00:54:04	1.220
+++ Makefile.in	2003/03/03 13:16:54
@@ -152,6 +152,8 @@
 
 # for the objects and depfiles, we might be bulding only part of them
-# depending on configure arguments, so select a subset of ALL
-OBJECTS = @ALL_OBJECTS@
+# depending on configure arguments, so select a subset of ALL, and
+# remove duplicates which could arise with some builds.
+
+OBJECTS = $(sort @ALL_OBJECTS@)
 
 # the object files of sublibraries (we assume that they don't change [often],

 	  	 
