diff -ru wxWidgets.orig/src/msw/stattext.cpp wxWidgets/src/msw/stattext.cpp
--- wxWidgets.orig/src/msw/stattext.cpp	2005-09-25 19:15:37.000000000 +1000
+++ wxWidgets/src/msw/stattext.cpp	2005-11-04 00:23:53.000000000 +1100
@@ -135,7 +135,7 @@
     dc.SetFont(font);
 
     wxCoord widthTextMax, heightTextTotal;
-    dc.GetMultiLineTextExtent(GetLabel(), &widthTextMax, &heightTextTotal);
+    dc.GetMultiLineTextExtent(wxStripMenuCodes(GetLabel()), &widthTextMax, &heightTextTotal);
 
 #ifdef __WXWINCE__
     if ( widthTextMax )

 	  	 
