Index: bitmap.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWidgets/src/gtk/bitmap.cpp,v
retrieving revision 1.94
diff -b -u -2 -r1.94 bitmap.cpp
--- bitmap.cpp	2005/10/01 15:27:30	1.94
+++ bitmap.cpp	2005/10/20 15:29:07
@@ -366,4 +366,7 @@
     wxCHECK_MSG( Ok(), wxNullBitmap, wxT("invalid bitmap") );
 
+    //FIXME:
+    //wouldn't it be much less confusing to use either the GetWidth() method
+    //or the M_BMPDATA->width makro inside of one method?
     if (newy==M_BMPDATA->m_width && newy==M_BMPDATA->m_height)
         return *this;
@@ -434,6 +437,6 @@
 
         // be careful to use the right scaling factor
-        float scx = (float)M_BMPDATA->m_width/(float)newx;
-        float scy = (float)M_BMPDATA->m_height/(float)newy;
+        float scx = static_cast<float>(M_BMPDATA->m_width-1)/newx;
+        float scy = static_cast<float>(M_BMPDATA->m_height-1)/newy;
         // prepare accel-tables
         int *tablex = (int *)calloc(width,sizeof(int));
@@ -441,11 +444,14 @@
 
         // accel table filled with clipped values
-        for (int x = 0; x < width; x++)
-            tablex[x] = (int) (scx * (x+clipx));
-        for (int y = 0; y < height; y++)
-            tabley[y] = (int) (scy * (y+clipy));
+        for (int x = 0; x < width; ++x){
+            tablex[x] = static_cast<int>(scx*(x+clipx));
+        }
+
+        for (int y = 0; y < height; ++y){
+            tabley[y] = static_cast<int> (scy*(y+clipy));
+        }
 
         // Main rescaling routine starts here
-        for (int h = 0; h < height; h++)
+        for (int h = 0; h < height; ++h)
         {
             char outbyte = 0;
@@ -453,5 +459,5 @@
             guint32 old_pixval = 0;
 
-            for (int w = 0; w < width; w++)
+            for (int w = 0; w < width; ++w)
             {
                 guint32 pixval;
@@ -512,5 +518,5 @@
             img = gdk_image_get( GetMask()->GetBitmap(), 0, 0, GetWidth(), GetHeight() );
 
-            for (int h = 0; h < height; h++)
+            for (int h = 0; h < height; ++h)
             {
                 char outbyte = 0;
@@ -518,5 +524,5 @@
                 guint32 old_pixval = 0;
 
-                for (int w = 0; w < width; w++)
+                for (int w = 0; w < width; ++w)
                 {
                     guint32 pixval;

 	  	 
