Index: src/common/wxchar.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/common/wxchar.cpp,v
retrieving revision 1.71
diff -u -2 -r1.71 wxchar.cpp
--- src/common/wxchar.cpp	2003/11/18 21:25:28	1.71
+++ src/common/wxchar.cpp	2003/12/04 04:47:01
@@ -753,5 +753,9 @@
             if ( *format == _T('.') )
             {
-                SkipDigits(&format);
+                CopyFmtChar(*format++);
+                if ( *format == _T('*') )
+                    CopyFmtChar(*format++);
+                else
+                    SkipDigits(&format);
             }
 
@@ -800,21 +804,12 @@
                 case _T('s'):
                     // %c -> %lc but %hc stays %hc and %lc is still %lc
-                    switch ( size )
-                    {
-                        case Default:
-                            InsertFmtChar(_T('l'));
-                            break;
-
-                        case Short:
-                            CopyFmtChar(_T('h'));
-                            break;
-
-                        case Long:
-                            ;
-                    }
+                    if ( size == Default)
+                        InsertFmtChar(_T('l'));
                     // fall through
 
                 default:
                     // nothing special to do
+                    if ( size != Default )
+                        CopyFmtChar(*(format - 1));
                     CopyFmtChar(*format++);
             }

 	  	 
