Index: configure.in
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/configure.in,v
retrieving revision 1.762
diff -b -u -2 -r1.762 configure.in
--- configure.in	2003/09/26 20:24:22	1.762
+++ configure.in	2003/09/28 17:13:32
@@ -2981,4 +2981,26 @@
 
 dnl ---------------------------------------------------------------------------
+dnl Xinerama (for unix wxDisplay)
+dnl ---------------------------------------------------------------------------
+if test "$wxUSE_DISPLAY" = "yes"; then
+    if test "$wxUSE_UNIX" = "yes" -a "$wxUSE_MAC" != 1; then
+        AC_MSG_CHECKING([for Xinerama])
+        WX_PATH_FIND_LIBRARIES([$SEARCH_LIB],Xinerama)
+        if test "$ac_find_libraries" != "" ; then
+            WX_LINK_PATH_EXIST([$ac_find_libraries],[$LDFLAGS])
+            if test "$ac_path_to_link" != " -L/usr/lib" ; then
+                LDFLAGS="$LDFLAGS $ac_path_to_link"
+            fi
+            GUI_TK_LIBRARY="$GUI_TK_LIBRARY -lXinerama"
+            AC_MSG_RESULT([yes])
+        else
+            AC_MSG_RESULT([no])
+            AC_MSG_WARN("*** Xinerama not found; disabling wxDisplay")
+            wxUSE_DISPLAY="no"
+        fi
+    fi
+fi
+
+dnl ---------------------------------------------------------------------------
 dnl OpenGL libraries
 dnl ---------------------------------------------------------------------------

 	  	 
