--- wxWidgets.orig/include/wx/generic/combo.h	Sun Aug 13 14:24:35 2006
+++ wxWidgets/include/wx/generic/combo.h	Tue Sep 12 19:58:33 2006
@@ -10,6 +10,6 @@
 /////////////////////////////////////////////////////////////////////////////
 
-#ifndef _WX_GENERIC_COMBOCONTROL_H_
-#define _WX_GENERIC_COMBOCONTROL_H_
+#ifndef _WX_GENERIC_COMBOCTRL_H_
+#define _WX_GENERIC_COMBOCTRL_H_
 
 #if wxUSE_COMBOCTRL
@@ -34,18 +34,18 @@
 extern WXDLLIMPEXP_DATA_CORE(const wxChar) wxComboBoxNameStr[];
 
-class WXDLLEXPORT wxGenericComboControl : public wxComboCtrlBase
+class WXDLLEXPORT wxGenericComboCtrl : public wxComboCtrlBase
 {
 public:
     // ctors and such
-    wxGenericComboControl() : wxComboCtrlBase() { Init(); }
+    wxGenericComboCtrl() : wxComboCtrlBase() { Init(); }
 
-    wxGenericComboControl(wxWindow *parent,
-                          wxWindowID id = wxID_ANY,
-                          const wxString& value = wxEmptyString,
-                          const wxPoint& pos = wxDefaultPosition,
-                          const wxSize& size = wxDefaultSize,
-                          long style = 0,
-                          const wxValidator& validator = wxDefaultValidator,
-                          const wxString& name = wxComboBoxNameStr)
+    wxGenericComboCtrl(wxWindow *parent,
+                       wxWindowID id = wxID_ANY,
+                       const wxString& value = wxEmptyString,
+                       const wxPoint& pos = wxDefaultPosition,
+                       const wxSize& size = wxDefaultSize,
+                       long style = 0,
+                       const wxValidator& validator = wxDefaultValidator,
+                       const wxString& name = wxComboBoxNameStr)
         : wxComboCtrlBase()
     {
@@ -64,5 +64,5 @@
                 const wxString& name = wxComboBoxNameStr);
 
-    virtual ~wxGenericComboControl();
+    virtual ~wxGenericComboCtrl();
 
     virtual bool IsKeyPopupToggle(const wxKeyEvent& event) const;
@@ -91,5 +91,5 @@
     DECLARE_EVENT_TABLE()
 
-    DECLARE_DYNAMIC_CLASS(wxGenericComboControl)
+    DECLARE_DYNAMIC_CLASS(wxGenericComboCtrl)
 };
 
@@ -100,18 +100,18 @@
 // front-end so that wxRTTI works as expected.
 
-class WXDLLEXPORT wxComboCtrl : public wxGenericComboControl
+class WXDLLEXPORT wxComboCtrl : public wxGenericComboCtrl
 {
 public:
-    wxComboCtrl() : wxGenericComboControl() {}
+    wxComboCtrl() : wxGenericComboCtrl() {}
 
     wxComboCtrl(wxWindow *parent,
-                   wxWindowID id = wxID_ANY,
-                   const wxString& value = wxEmptyString,
-                   const wxPoint& pos = wxDefaultPosition,
-                   const wxSize& size = wxDefaultSize,
-                   long style = 0,
-                   const wxValidator& validator = wxDefaultValidator,
-                   const wxString& name = wxComboBoxNameStr)
-        : wxGenericComboControl()
+                wxWindowID id = wxID_ANY,
+                const wxString& value = wxEmptyString,
+                const wxPoint& pos = wxDefaultPosition,
+                const wxSize& size = wxDefaultSize,
+                long style = 0,
+                const wxValidator& validator = wxDefaultValidator,
+                const wxString& name = wxComboBoxNameStr)
+        : wxGenericComboCtrl()
     {
         (void)Create(parent, id, value, pos, size, style, validator, name);
@@ -130,5 +130,5 @@
 #else
 
-#define wxGenericComboControl   wxComboCtrl
+#define wxGenericComboCtrl   wxComboCtrl
 
 #endif // !defined(wxCOMBOCONTROL_FULLY_FEATURED)
@@ -136,3 +136,3 @@
 #endif // wxUSE_COMBOCTRL
 #endif
-    // _WX_GENERIC_COMBOCONTROL_H_
+    // _WX_GENERIC_COMBOCTRL_H_
--- wxWidgets.orig/samples/combo/combo.cpp	Thu Jul 20 18:31:01 2006
+++ wxWidgets/samples/combo/combo.cpp	Tue Sep 12 19:58:33 2006
@@ -629,5 +629,5 @@
 
     wxComboCtrl* cc;
-    wxGenericComboControl* gcc;
+    wxGenericComboCtrl* gcc;
     wxOwnerDrawnComboBox* odc;
 
@@ -747,7 +747,7 @@
     //
 
-    // Note that we test that wxGenericComboControl works
-    gcc = new wxGenericComboControl(panel,wxID_ANY,wxEmptyString,
-                                    wxDefaultPosition, wxDefaultSize);
+    // Note that we test that wxGenericComboCtrl works
+    gcc = new wxGenericComboCtrl(panel,wxID_ANY,wxEmptyString,
+                                 wxDefaultPosition, wxDefaultSize);
 
     // Set popup interface right away, otherwise some of the calls
--- wxWidgets.orig/src/generic/combog.cpp	Thu Aug 17 19:19:59 2006
+++ wxWidgets/src/generic/combog.cpp	Tue Sep 12 19:58:33 2006
@@ -87,27 +87,27 @@
 
 // ----------------------------------------------------------------------------
-// wxGenericComboControl
+// wxGenericComboCtrl
 // ----------------------------------------------------------------------------
 
-BEGIN_EVENT_TABLE(wxGenericComboControl, wxComboCtrlBase)
-    EVT_PAINT(wxGenericComboControl::OnPaintEvent)
-    EVT_MOUSE_EVENTS(wxGenericComboControl::OnMouseEvent)
+BEGIN_EVENT_TABLE(wxGenericComboCtrl, wxComboCtrlBase)
+    EVT_PAINT(wxGenericComboCtrl::OnPaintEvent)
+    EVT_MOUSE_EVENTS(wxGenericComboCtrl::OnMouseEvent)
 END_EVENT_TABLE()
 
 
-IMPLEMENT_DYNAMIC_CLASS(wxGenericComboControl, wxComboCtrlBase)
+IMPLEMENT_DYNAMIC_CLASS(wxGenericComboCtrl, wxComboCtrlBase)
 
-void wxGenericComboControl::Init()
+void wxGenericComboCtrl::Init()
 {
 }
 
-bool wxGenericComboControl::Create(wxWindow *parent,
-                                   wxWindowID id,
-                                   const wxString& value,
-                                   const wxPoint& pos,
-                                   const wxSize& size,
-                                   long style,
-                                   const wxValidator& validator,
-                                   const wxString& name)
+bool wxGenericComboCtrl::Create(wxWindow *parent,
+                                wxWindowID id,
+                                const wxString& value,
+                                const wxPoint& pos,
+                                const wxSize& size,
+                                long style,
+                                const wxValidator& validator,
+                                const wxString& name)
 {
 
@@ -121,10 +121,10 @@
 #elif defined(__WXMSW__)
         // For XP, have 1-width custom border, for older version use sunken
-        if ( wxUxThemeEngine::GetIfActive() )
+        /*if ( wxUxThemeEngine::GetIfActive() )
         {
             border = wxBORDER_NONE;
             m_widthCustomBorder = 1;
         }
-        else
+        else*/
             border = wxBORDER_SUNKEN;
 #elif defined(__WXGTK__)
@@ -173,9 +173,9 @@
 }
 
-wxGenericComboControl::~wxGenericComboControl()
+wxGenericComboCtrl::~wxGenericComboCtrl()
 {
 }
 
-void wxGenericComboControl::OnResize()
+void wxGenericComboCtrl::OnResize()
 {
 
@@ -199,5 +199,5 @@
 }
 
-void wxGenericComboControl::OnPaintEvent( wxPaintEvent& WXUNUSED(event) )
+void wxGenericComboCtrl::OnPaintEvent( wxPaintEvent& WXUNUSED(event) )
 {
     wxSize sz = GetClientSize();
@@ -279,5 +279,5 @@
 }
 
-void wxGenericComboControl::OnMouseEvent( wxMouseEvent& event )
+void wxGenericComboCtrl::OnMouseEvent( wxMouseEvent& event )
 {
     bool isOnButtonArea = m_btnArea.Inside(event.m_x,event.m_y);
@@ -322,5 +322,5 @@
 }
 
-bool wxGenericComboControl::IsKeyPopupToggle(const wxKeyEvent& event) const
+bool wxGenericComboCtrl::IsKeyPopupToggle(const wxKeyEvent& event) const
 {
     int keycode = event.GetKeyCode();
@@ -346,7 +346,7 @@
 #ifdef __WXUNIVERSAL__
 
-bool wxGenericComboControl::PerformAction(const wxControlAction& action,
-                                          long numArg,
-                                          const wxString& strArg)
+bool wxGenericComboCtrl::PerformAction(const wxControlAction& action,
+                                       long numArg,
+                                       const wxString& strArg)
 {
     bool processed = false;
@@ -384,5 +384,5 @@
 // front-end so that wxRTTI works as expected.
 #ifndef _WX_COMBOCONTROL_H_
-IMPLEMENT_DYNAMIC_CLASS(wxComboCtrl, wxGenericComboControl)
+IMPLEMENT_DYNAMIC_CLASS(wxComboCtrl, wxGenericComboCtrl)
 #endif
 

 	  	 
