Index: wx/lib/agw/hypertreelist.py
===================================================================
--- wx/lib/agw/hypertreelist.py	(revision 63904)
+++ wx/lib/agw/hypertreelist.py	(working copy)
@@ -2923,7 +2923,10 @@
 
             wnd = item.GetWindow(i)            
             if wnd:
-                wndx = text_x + text_w + 2*_MARGIN
+                if text_w == 0:
+                    wndx = text_x
+                else:
+                    wndx = text_x + text_w + 2*_MARGIN
                 xa, ya = self.CalcScrolledPosition((0, item.GetY()))
                 wndx += xa
                 if item.GetHeight() > item.GetWindowSize(i)[1]:
