Index: e:/dev/wxWindows/src/common/event.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/common/event.cpp,v
retrieving revision 1.116
diff -u -2 -r1.116 event.cpp
--- e:/dev/wxWindows/src/common/event.cpp	2003/04/02 11:10:32	1.116
+++ e:/dev/wxWindows/src/common/event.cpp	2003/05/25 09:21:43
@@ -289,4 +289,8 @@
 // ----------------------------------------------------------------------------
 
+#if wxUSE_GUI
+
+#endif
+
 /*
  * General wxWindows events, covering
@@ -306,5 +310,8 @@
     m_skipped = FALSE;
     m_callbackUserData = (wxObject *) NULL;
+#if wxUSE_GUI
     m_isCommandEvent = FALSE;
+    m_shouldPropagateLevel = m_pGetPropagationValue(*this);
+#endif
 }
 
@@ -317,11 +324,39 @@
     , m_callbackUserData(src.m_callbackUserData)
     , m_skipped(src.m_skipped)
+#if wxUSE_GUI
     , m_isCommandEvent(src.m_isCommandEvent)
+    , m_shouldPropagateLevel(src.m_shouldPropagateLevel)
+#endif
 {
 }
 
 #if wxUSE_GUI
+/* static */ PropagationCallback wxEvent::m_pGetPropagationValue = wxEvent::GetDefaultPropagationValue;
+/* static */ wxUint32 wxEvent::GetDefaultPropagationValue(wxEvent &event)
+{
+    if (/*event.IsCommandEvent()*/ event.IsKindOf(CLASSINFO(wxCommandEvent)))
+        return wxMAX_EVENT_PROPAGATION;
+    else
+        return wxNO_EVENT_PROPAGATION;
+}
 
 /*
+ * wxDisablePropagationTemporarily helper class.
+ */
+
+wxDisablePropagationTemporarily::wxDisablePropagationTemporarily(wxEvent &event)
+{
+    m_oldPropagationValue = event.m_shouldPropagateLevel;
+    m_pEvent = &event;
+    event.StopPropagate(); 
+}
+
+wxDisablePropagationTemporarily::~wxDisablePropagationTemporarily()
+{
+    m_pEvent->m_shouldPropagateLevel = m_oldPropagationValue;
+}
+
+
+/*
  * Command events
  *
@@ -337,4 +372,5 @@
     m_commandString = wxEmptyString;
     m_isCommandEvent = TRUE;
+    m_shouldPropagateLevel = m_pGetPropagationValue(*this);
 }
 
@@ -899,5 +935,5 @@
     // event: it wouldn't make sense for a parent to receive a child's size
     // event, for example
-    if ( m_isWindow && event.IsCommandEvent() )
+    if ( m_isWindow && event.ShouldPropagate() )
     {
         wxWindow *win = (wxWindow *)this;
@@ -911,5 +947,13 @@
             wxWindow *parent = win->GetParent();
             if ( parent && !parent->IsBeingDeleted() )
-                return parent->GetEventHandler()->ProcessEvent(event);
+            {
+                event.m_shouldPropagateLevel--;
+                if( parent->GetEventHandler()->ProcessEvent(event) )
+                {
+                    event.m_shouldPropagateLevel++;
+                    return TRUE;
+                }
+                event.m_shouldPropagateLevel++;
+            }
         }
     }
Index: e:/dev/wxWindows/include/wx/event.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/event.h,v
retrieving revision 1.162
diff -u -2 -r1.162 event.h
--- e:/dev/wxWindows/include/wx/event.h	2003/03/21 02:58:51	1.162
+++ e:/dev/wxWindows/include/wx/event.h	2003/05/25 09:21:57
@@ -324,4 +324,8 @@
 #endif // WXWIN_COMPATIBILITY
 
+#define wxMAX_EVENT_PROPAGATION     0xFFFFFF
+#define wxNO_EVENT_PROPAGATION      0x000000
+typedef wxUint32 (*PropagationCallback)(wxEvent &event);
+
 /*
  * wxWindows events, covering all interesting things that might happen
@@ -363,8 +367,4 @@
     bool GetSkipped() const { return m_skipped; };
 
-    // Implementation only: this test is explicitlty anti OO and this functions
-    // exists only for optimization purposes.
-    bool IsCommandEvent() const { return m_isCommandEvent; }
-
     // this function is used to create a copy of the event polymorphically and
     // all derived classes must implement it because otherwise wxPostEvent()
@@ -372,4 +372,21 @@
     virtual wxEvent *Clone() const = 0;
 
+#if wxUSE_GUI
+    // Implementation only: this test is explicitlty anti OO and this functions
+    // exists only for optimization purposes.
+    bool IsCommandEvent() const { return m_isCommandEvent; }
+
+    // Determine if this event should be propagating to its parent or not.
+    bool ShouldPropagate() const { return m_shouldPropagateLevel != wxNO_EVENT_PROPAGATION; }
+    // Stop an event from propagating to its parent window.
+    void StopPropagate() { m_shouldPropagateLevel = wxNO_EVENT_PROPAGATION; }
+
+    // Default PropagationCallback, 
+    // function decides what event should get what propagation behavior.
+    static wxUint32 GetDefaultPropagationValue(wxEvent &event);
+    // Set a new PropagationCallback function.
+    static void SetPropagationCallback(PropagationCallback func) { wxEvent::m_pGetPropagationValue = func; }
+#endif
+
 public:
     wxObject*         m_eventObject;
@@ -379,6 +396,11 @@
     wxObject*         m_callbackUserData;
     bool              m_skipped;
-    bool              m_isCommandEvent;
 
+#if wxUSE_GUI
+    bool              m_isCommandEvent;
+    wxUint32          m_shouldPropagateLevel;
+    static PropagationCallback m_pGetPropagationValue;
+#endif
+    
 private:
     DECLARE_ABSTRACT_CLASS(wxEvent)
@@ -386,4 +408,15 @@
 
 #if wxUSE_GUI
+
+class wxDisablePropagationTemporarily
+{
+public:
+    wxDisablePropagationTemporarily(wxEvent &event);
+    ~wxDisablePropagationTemporarily();
+private:
+    wxUint32    m_oldPropagationValue;
+    wxEvent*    m_pEvent;
+};
+
 
 // Item or menu event class

 	  	 
