diff -ur wxDFB-2.8.8.org/src/dfb/app.cpp wxDFB-2.8.8/src/dfb/app.cpp
--- wxDFB-2.8.8.org/src/dfb/app.cpp	2008-06-20 11:06:11.000000000 +0200
+++ wxDFB-2.8.8/src/dfb/app.cpp	2008-07-16 18:57:17.000000000 +0200
@@ -135,7 +135,9 @@
 
     s_inYield = true;
 
+#if wxUSE_LOG
     wxLog::Suspend();
+#endif // wxUSE_LOG
 
     // A guarentee that there will be an active event loop:
     wxEventLoopGuarantor dummyLoopIfNeeded;
@@ -146,7 +148,9 @@
     // OnUpdateUI() which is a nice (and desired) side effect)
     while ( ProcessIdle() ) {}
 
+#if wxUSE_LOG
     wxLog::Resume();
+#endif // wxUSE_LOG
 
     s_inYield = false;
 
