Index: src/common/popupcmn.cpp
===================================================================
--- src/common/popupcmn.cpp	(revision 49358)
+++ src/common/popupcmn.cpp	(working copy)
@@ -346,6 +346,9 @@
 #if defined( __WXMSW__ ) || defined( __WMAC__ )
     if (show && m_child)
     {
+        // Set it as the foreground window so the mouse can be captured
+        ::SetForegroundWindow(GetHwnd());
+        
         // Assume that the mouse is outside the popup to begin with
         m_child->CaptureMouse();
     }
Index: src/msw/popupwin.cpp
===================================================================
--- src/msw/popupwin.cpp	(revision 49358)
+++ src/msw/popupwin.cpp	(working copy)
@@ -102,9 +98,6 @@
         {
             wxLogLastError(_T("SetWindowPos"));
         }
-
-        // and set it as the foreground window so the mouse can be captured
-        ::SetForegroundWindow(GetHwnd());
     }
 
     return true;

 	  	 
