--- a/src/msw/toolbar.cpp
+++ b/src/msw/toolbar.cpp
@@ -1324,7 +1324,7 @@ bool wxToolBar::MSWCommand(WXUINT WXUNUSED(cmd), WXWORD id_)
     // "negative" (i.e. > 32767) WORD
     const int id = (signed short)id_;
 
-    wxToolBarToolBase *tool = FindById(id);
+    wxToolBarTool *tool = wxStaticCast(FindById(id), wxToolBarTool);
     if ( !tool )
         return false;
 
@@ -1361,6 +1361,12 @@ bool wxToolBar::MSWCommand(WXUINT WXUNUSED(cmd), WXWORD id_)
     Update();
 
     bool allowLeftClick = OnLeftClick(id, toggled);
+    // Check if tool has been deleted in the event handler.
+    // If so, the rest of state handling can be skipped.
+    if (tool->IsToBeDeleted())
+    {
+        return true;
+    }
 
     // Restore the unpressed state. Enabled/toggled state might have been
     // changed since so take care of it.
