Index: samples/minimal/minimal.cpp
===================================================================
--- samples/minimal/minimal.cpp	(revision 65792)
+++ samples/minimal/minimal.cpp	(working copy)
@@ -13,6 +13,9 @@
 // declarations
 // ============================================================================
 
+//#define WITHGRAPH
+#define BTID (wxID_HIGHEST+1)
+
 // ----------------------------------------------------------------------------
 // headers
 // ----------------------------------------------------------------------------
@@ -67,6 +70,10 @@
     // event handlers (these functions should _not_ be virtual)
     void OnQuit(wxCommandEvent& event);
     void OnAbout(wxCommandEvent& event);
+    void OnBt(wxCommandEvent& ev);
+#ifdef WITHGRAPH
+	void OnPaint(wxPaintEvent& ev);
+#endif
 
 private:
     // any class wishing to process wxWidgets events must use this macro
@@ -99,6 +106,10 @@
 BEGIN_EVENT_TABLE(MyFrame, wxFrame)
     EVT_MENU(Minimal_Quit,  MyFrame::OnQuit)
     EVT_MENU(Minimal_About, MyFrame::OnAbout)
+    EVT_BUTTON(BTID, MyFrame::OnBt)
+#ifdef WITHGRAPH
+	EVT_PAINT(MyFrame::OnPaint)
+#endif
 END_EVENT_TABLE()
 
 // Create a new application object: this macro will allow wxWidgets to create
@@ -147,6 +158,11 @@
 {
     // set the frame icon
     SetIcon(wxICON(sample));
+	
+    wxButton* bt = new wxButton(this, BTID, wxT("Scroll -10 vertivally"));
+	wxBoxSizer* sz = new wxBoxSizer(wxVERTICAL);
+	sz->Add(bt, 0, wxEXPAND|wxALL, 5);
+	SetSizer(sz);
 
 #if wxUSE_MENUS
     // create a menu bar
@@ -198,3 +214,15 @@
                  wxOK | wxICON_INFORMATION,
                  this);
 }
+
+void MyFrame::OnBt(wxCommandEvent&)
+{
+	ScrollWindow(0, -10);
+}
+#ifdef WITHGRAPH
+void MyFrame::OnPaint(wxPaintEvent& ev)
+{
+	wxPaintDC dc(this);
+	dc.DrawText(wxT("DrawText pos 50, 50"), 50, 50);
+}
+#endif
