--- a/samples/propgrid/propgrid.cpp
+++ b/samples/propgrid/propgrid.cpp
@@ -1168,6 +1168,16 @@ void FormMain::PopulateWithStandardItems ()
         pg->SetPropertyCell( p, 3, wxT("Cell 3"), bmp );
         pg->SetPropertyCell( p, 4, wxT("Cell 4"), wxNullBitmap, *wxWHITE, *wxBLACK );
     }
+
+    srand(time(NULL));
+    pg->Append( new wxPropertyCategory(wxT("Points"),wxT("PointsCategory")) );
+    for(int i = 0; i < 2000; i++)
+    {
+        wxString name = wxString::Format("Point %04i", i);
+        wxPoint pt(rand(), rand());
+        pg->Append( new wxPointProperty(name, wxPG_LABEL, pt) );
+    }
+    pg->ExpandAll();
 }
 
 // -----------------------------------------------------------------------
