--- wxWidgets-3.0.2/src/msw/toolbar.cpp	Mon Dec 15 18:38:34 2014
+++ wxWidgets-3.0.2/src/msw/toolbar.cpp	Mon Dec 15 18:25:07 2014
@@ -235,7 +235,7 @@ class wxToolBarTool : public wxToolBarToolBase
         }
     }
 
-    void ToBeDeleted() { m_toBeDeleted = true; }
+    void ToBeDeleted(bool value = true) { m_toBeDeleted = value; }
     bool IsToBeDeleted() const { return m_toBeDeleted; }
 
 private:
@@ -561,10 +561,11 @@ WXDWORD wxToolBar::MSWGetStyle(long style, WXDWORD
 // ----------------------------------------------------------------------------
 
 bool wxToolBar::DoInsertTool(size_t WXUNUSED(pos),
-                             wxToolBarToolBase * WXUNUSED(tool))
+                             wxToolBarToolBase * tool)
 {
     // nothing special to do here - we really create the toolbar buttons in
     // Realize() later
+    static_cast<wxToolBarTool*>(tool)->ToBeDeleted(false);
     InvalidateBestSize();
     return true;
 }
