Index: src/msw/gsockmsw.c
===================================================================
RCS file: /usr/home/njak/eng/wxw/src/msw/gsockmsw.c,v
retrieving revision 1.1
diff -b -u -2 -r1.1 gsockmsw.c
--- src/msw/gsockmsw.c	22 Oct 2004 13:59:21 -0000	1.1
+++ src/msw/gsockmsw.c	15 Dec 2004 01:55:54 -0000
@@ -76,5 +76,5 @@
 /* Global variables */

-extern HINSTANCE INSTANCE;
+extern "C" HINSTANCE INSTANCE;
 static HWND hWin;
 static CRITICAL_SECTION critical;
Index: src/msw/main.cpp
===================================================================
RCS file: /usr/home/njak/eng/wxw/src/msw/main.cpp,v
retrieving revision 1.1
diff -b -u -2 -r1.1 main.cpp
--- src/msw/main.cpp	22 Oct 2004 13:59:22 -0000	1.1
+++ src/msw/main.cpp	15 Dec 2004 01:54:59 -0000
@@ -141,9 +141,10 @@
 // ----------------------------------------------------------------------------

+extern "C" {
 HINSTANCE wxGetInstance()
 {
     return wxhInstance;
 }
-
+}
 void wxSetInstance(HINSTANCE hInst)
 {
 	  	 
