Index: src/univ/colschem.cpp
===================================================================
--- src/univ/colschem.cpp	(revision 67001)
+++ src/univ/colschem.cpp	(working copy)
@@ -79,6 +79,12 @@
         wxColourScheme::MAX /* wxSYS_COLOUR_INFOTEXT */,
         wxColourScheme::MAX /* wxSYS_COLOUR_INFOBK */,
         wxColourScheme::WINDOW /* wxSYS_COLOUR_LISTBOX */,
+        wxColourScheme::MAX /* wxSYS_COLOUR_HOTLIGHT */,
+        wxColourScheme::TITLEBAR_ACTIVE_TEXT /* wxSYS_COLOUR_GRADIENTACTIVECAPTION */,
+        wxColourScheme::TITLEBAR_TEXT /* wxSYS_COLOUR_GRADIENTINACTIVECAPTION */,
+        wxColourScheme::MAX /* wxSYS_COLOUR_MENUHILIGHT */,
+        wxColourScheme::MAX /* wxSYS_COLOUR_MENUBAR */,
+        wxColourScheme::CONTROL_TEXT /* wxSYS_COLOUR_LISTBOXTEXT */
     };
 
     wxCHECK_MSG( index < (int)WXSIZEOF(s_mapSysToThemeCol), wxNullColour,
