Index: include/wx/graphics.h
===================================================================
--- include/wx/graphics.h	(revision 71058)
+++ include/wx/graphics.h	(working copy)
@@ -635,7 +635,7 @@
 
     void DrawText( const wxString &str, wxDouble x, wxDouble y,
                    const wxGraphicsBrush& backgroundBrush )
-        { DoDrawFilledText(str, x, y, backgroundBrush); }
+        { wxGraphicsContext::DoDrawFilledText(str, x, y, backgroundBrush); }
 
     void DrawText( const wxString &str, wxDouble x, wxDouble y,
                    wxDouble angle, const wxGraphicsBrush& backgroundBrush )
