diff -rbu2 ./cvs/wx2/wxwindows/contrib/utils/wxrcedit/preview.cpp ./wx2/contrib/utils/wxrcedit/preview.cpp
--- ./cvs/wx2/wxwindows/contrib/utils/wxrcedit/preview.cpp	Fri May 18 20:50:24 2001
+++ ./wx2/contrib/utils/wxrcedit/preview.cpp	Sun May 20 18:16:32 2001
@@ -221,7 +221,16 @@
 
 
+#ifdef __WXMSW__
+// avoid Problems with setting the focus to a no longer existing child
+void PreviewFrame::OnActivate(wxActivateEvent &event)
+{
+}
+#endif
 
 BEGIN_EVENT_TABLE(PreviewFrame, wxFrame)
     EVT_ENTER_WINDOW(PreviewFrame::OnMouseEnter)
+#ifdef __WXMSW__
+    EVT_ACTIVATE(PreviewFrame::OnActivate)
+#endif
 END_EVENT_TABLE()
 
diff -rbu2 ./cvs/wx2/wxwindows/contrib/utils/wxrcedit/preview.h ./wx2/contrib/utils/wxrcedit/preview.h
--- ./cvs/wx2/wxwindows/contrib/utils/wxrcedit/preview.h	Fri Nov 24 17:19:48 2000
+++ ./wx2/contrib/utils/wxrcedit/preview.h	Sun May 20 18:15:52 2001
@@ -58,4 +58,7 @@
         DECLARE_EVENT_TABLE()
         void OnMouseEnter(wxMouseEvent& event);
+#ifdef __WXMSW__
+        void OnActivate(wxActivateEvent &event);
+#endif
 };
 

 	  	 
