Index: src/unix/utilsunx.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/unix/utilsunx.cpp,v
retrieving revision 1.141
diff -u2 -r1.141 src/unix/utilsunx.cpp
--- src/unix/utilsunx.cpp
+++ src/unix/utilsunx.cpp
@@ -961,6 +961,31 @@
         if ( fgets(buf, WXSIZEOF(buf), fp) && fgets(buf, WXSIZEOF(buf), fp) )
         {
-            long memTotal, memUsed;
-            sscanf(buf, "Mem: %ld %ld %ld", &memTotal, &memUsed, &memFree);
+            // /proc/meminfo changed its format in kernel 2.6
+            // wxPlatformInfo::Get()->GetOSMinorVersion returns the minor version
+            // of the runtime-detected kernel version and thus it's what we need
+            if (wxPlatformInfo::Get().GetOSMinorVersion() >= 6)
+            {
+                unsigned long cached, buffers;
+                sscanf(buf, "MemFree: %ld", &memFree);
+
+                fgets(buf, WXSIZEOF(buf), fp);
+                sscanf(buf, "Buffers: %lu", &buffers);
+
+                fgets(buf, WXSIZEOF(buf), fp);
+                sscanf(buf, "Cached: %lu", &cached);
+
+                // add to "MemFree" also the "Buffers" and "Cached" values as free(1)
+                // does as otherwise the value almost never makes sense: for kernerl
+                // 2.6 it's always almost 0
+                memFree += buffers + cached;
+
+                // values are always expressed in kB !
+                memFree *= 1024;        // we want bytes
+            }
+            else
+            {
+                long memTotal, memUsed;
+                sscanf(buf, "Mem: %ld %ld %ld", &memTotal, &memUsed, &memFree);
+            }
         }
 

 	  	 
