diff -bu1 -r h:/wxcvshead/wxWindows/contrib/src/fl/barhintspl.cpp h:/wxcvshd/wxWin/contrib/src/fl/barhintspl.cpp
--- h:/wxcvshead/wxWindows/contrib/src/fl/barhintspl.cpp	2003-04-02 13:10:26.000000000 +0200
+++ h:/wxcvshd/wxWin/contrib/src/fl/barhintspl.cpp	2003-08-19 12:56:40.000000000 +0200
@@ -123,3 +123,3 @@
 
-void cbBarHintsPlugin::Draw3DBox( wxDC& dc, const wxPoint& pos, bool pressed )
+void cbBarHintsPlugin::Draw3DBox( wxDC& WXUNUSED(dc), const wxPoint& WXUNUSED(pos), bool WXUNUSED(pressed) )
 {
@@ -127,3 +127,3 @@
 
-void cbBarHintsPlugin::DrawCloseBox( wxDC& dc, const wxPoint& pos, bool pressed )
+void cbBarHintsPlugin::DrawCloseBox( wxDC& WXUNUSED(dc), const wxPoint& WXUNUSED(pos), bool WXUNUSED(pressed) )
 {
@@ -131,4 +131,4 @@
 
-void cbBarHintsPlugin::DrawCollapseBox( wxDC& dc, const wxPoint& pos, 
-                                        bool atLeft, bool disabled, bool pressed )
+void cbBarHintsPlugin::DrawCollapseBox( wxDC& WXUNUSED(dc), const wxPoint& WXUNUSED(pos), 
+                                        bool WXUNUSED(atLeft), bool WXUNUSED(disabled), bool WXUNUSED(pressed) )
 {
@@ -202,3 +202,3 @@
 void cbBarHintsPlugin::DoDrawHint( wxDC& dc, wxRect& rect, 
-                                   int pos, int boxOfs, int grooveOfs,
+                                   int pos, int WXUNUSED(boxOfs), int grooveOfs,
                                    bool isFixed )
@@ -502,3 +502,3 @@
 
-        int result = HitTestHints( *mpClickedBar, event.mPos );
+        HitTestHints( *mpClickedBar, event.mPos );
 
diff -bu1 -r h:/wxcvshead/wxWindows/contrib/src/fl/controlbar.cpp h:/wxcvshd/wxWin/contrib/src/fl/controlbar.cpp
--- h:/wxcvshead/wxWindows/contrib/src/fl/controlbar.cpp	2003-07-04 15:49:39.000000000 +0200
+++ h:/wxcvshd/wxWin/contrib/src/fl/controlbar.cpp	2003-08-19 12:58:54.000000000 +0200
@@ -1475,3 +1475,3 @@
 
-void wxFrameLayout::OnEraseBackground( wxEraseEvent& event )
+void wxFrameLayout::OnEraseBackground( wxEraseEvent& WXUNUSED(event) )
 {
@@ -1498,3 +1498,3 @@
 
-void wxFrameLayout::OnKillFocus( wxFocusEvent& event )
+void wxFrameLayout::OnKillFocus( wxFocusEvent& WXUNUSED(event) )
 {
@@ -1504,3 +1504,3 @@
 
-void wxFrameLayout::OnSetFocus( wxFocusEvent& event )
+void wxFrameLayout::OnSetFocus( wxFocusEvent& WXUNUSED(event) )
 {
@@ -1509,3 +1509,3 @@
 
-void wxFrameLayout::OnActivate( wxActivateEvent& event )
+void wxFrameLayout::OnActivate( wxActivateEvent& WXUNUSED(event) )
 {
@@ -1624,3 +1624,3 @@
 
-void wxFrameLayout::ReleaseEventsFromPlugin( cbPluginBase* pPlugin )
+void wxFrameLayout::ReleaseEventsFromPlugin( cbPluginBase* WXUNUSED(pPlugin) )
 {
@@ -1642,3 +1642,3 @@
 
-void wxFrameLayout::ReleaseEventsFromPane( cbDockPane* fromPane )
+void wxFrameLayout::ReleaseEventsFromPane( cbDockPane* WXUNUSED(fromPane) )
 {
diff -bu1 -r h:/wxcvshead/wxWindows/contrib/src/fl/dyntbar.cpp h:/wxcvshd/wxWin/contrib/src/fl/dyntbar.cpp
--- h:/wxcvshead/wxWindows/contrib/src/fl/dyntbar.cpp	2003-08-07 10:52:08.000000000 +0200
+++ h:/wxcvshd/wxWin/contrib/src/fl/dyntbar.cpp	2003-08-19 13:04:33.000000000 +0200
@@ -78,3 +78,3 @@
                               const long style, 
-                              const int orientation, const int RowsOrColumns, 
+                              const int WXUNUSED(orientation), const int WXUNUSED(RowsOrColumns), 
                               const wxString& name)
@@ -111,3 +111,3 @@
                                 wxWindow* pToolWindow, 
-                                const wxSize& size
+                                const wxSize& WXUNUSED(size)
                               )
@@ -176,6 +176,6 @@
       wxDynamicToolBar::AddTool(const int toolIndex, const wxBitmap& bitmap, 
-                                const wxBitmap& pushedBitmap,
-                                const bool toggle, const long xPos, 
-                                const long yPos, wxObject *clientData,
-                                const wxString& helpString1, const wxString& helpString2)
+                                const wxBitmap& WXUNUSED(pushedBitmap),
+                                const bool WXUNUSED(toggle), const long WXUNUSED(xPos), 
+                                const long WXUNUSED(yPos), wxObject *WXUNUSED(clientData),
+                                const wxString& helpString1, const wxString& WXUNUSED(helpString2))
 {
@@ -271,3 +271,3 @@
 
-void wxDynamicToolBar::OnEraseBackground( wxEraseEvent& event )
+void wxDynamicToolBar::OnEraseBackground( wxEraseEvent& WXUNUSED(event) )
 {
@@ -276,3 +276,3 @@
 
-void wxDynamicToolBar::OnSize( wxSizeEvent& event )
+void wxDynamicToolBar::OnSize( wxSizeEvent& WXUNUSED(event) )
 {
@@ -312,3 +312,3 @@
 
-void wxDynamicToolBar::OnPaint( wxPaintEvent& event )
+void wxDynamicToolBar::OnPaint( wxPaintEvent& WXUNUSED(event) )
 {
@@ -571,3 +571,3 @@
 
-wxToolBarToolBase* wxDynamicToolBar::FindToolForPosition( wxCoord x, wxCoord y ) const
+wxToolBarToolBase* wxDynamicToolBar::FindToolForPosition( wxCoord WXUNUSED(x), wxCoord WXUNUSED(y) ) const
 {
@@ -576,3 +576,3 @@
 
-bool wxDynamicToolBar::DoInsertTool( size_t pos, wxToolBarToolBase* tool )
+bool wxDynamicToolBar::DoInsertTool( size_t WXUNUSED(pos), wxToolBarToolBase* WXUNUSED(tool) )
 {
@@ -581,3 +581,3 @@
 
-bool wxDynamicToolBar::DoDeleteTool( size_t pos, wxToolBarToolBase* tool )
+bool wxDynamicToolBar::DoDeleteTool( size_t WXUNUSED(pos), wxToolBarToolBase* WXUNUSED(tool) )
 {
@@ -586,3 +586,3 @@
 
-void wxDynamicToolBar::DoEnableTool( wxToolBarToolBase* tool, bool enable )
+void wxDynamicToolBar::DoEnableTool( wxToolBarToolBase* WXUNUSED(tool), bool WXUNUSED(enable) )
 {
@@ -590,3 +590,3 @@
 
-void wxDynamicToolBar::DoToggleTool( wxToolBarToolBase* tool, bool toggle )
+void wxDynamicToolBar::DoToggleTool( wxToolBarToolBase* WXUNUSED(tool), bool WXUNUSED(toggle) )
 {
@@ -594,3 +594,3 @@
 
-void wxDynamicToolBar::DoSetToggle( wxToolBarToolBase* tool, bool toggle )
+void wxDynamicToolBar::DoSetToggle( wxToolBarToolBase* WXUNUSED(tool), bool WXUNUSED(toggle) )
 {
@@ -598,3 +598,11 @@
 
-wxToolBarToolBase* wxDynamicToolBar::CreateTool( int id, const wxString& label, const wxBitmap& bmpNormal, const wxBitmap& bmpDisabled, wxItemKind kind, wxObject *clientData, const wxString& shortHelp, const wxString& longHelp)
+wxToolBarToolBase* wxDynamicToolBar::CreateTool( int WXUNUSED(id), 
+                                                 const wxString& WXUNUSED(label), 
+                                                 const wxBitmap& WXUNUSED(bmpNormal), 
+                                                 const wxBitmap& WXUNUSED(bmpDisabled), 
+                                                 wxItemKind WXUNUSED(kind), 
+                                                 wxObject *WXUNUSED(clientData), 
+                                                 const wxString& WXUNUSED(shortHelp), 
+                                                 const wxString& WXUNUSED(longHelp)
+                                                )
 {
@@ -603,3 +611,3 @@
 
-wxToolBarToolBase* wxDynamicToolBar::CreateTool( wxControl* control )
+wxToolBarToolBase* wxDynamicToolBar::CreateTool( wxControl* WXUNUSED(control) )
 {
diff -bu1 -r h:/wxcvshead/wxWindows/contrib/src/fl/dyntbarhnd.cpp h:/wxcvshd/wxWin/contrib/src/fl/dyntbarhnd.cpp
--- h:/wxcvshead/wxWindows/contrib/src/fl/dyntbarhnd.cpp	2002-01-21 22:34:41.000000000 +0100
+++ h:/wxcvshd/wxWin/contrib/src/fl/dyntbarhnd.cpp	2003-08-19 13:04:56.000000000 +0200
@@ -31,3 +31,3 @@
 
-void cbDynToolBarDimHandler::OnChangeBarState(cbBarInfo* pBar, int newState )
+void cbDynToolBarDimHandler::OnChangeBarState(cbBarInfo* WXUNUSED(pBar), int WXUNUSED(newState) )
 {
diff -bu1 -r h:/wxcvshead/wxWindows/contrib/src/fl/frmview.cpp h:/wxcvshd/wxWin/contrib/src/fl/frmview.cpp
--- h:/wxcvshead/wxWindows/contrib/src/fl/frmview.cpp	2003-07-28 12:14:12.000000000 +0200
+++ h:/wxcvshd/wxWin/contrib/src/fl/frmview.cpp	2003-08-19 13:06:41.000000000 +0200
@@ -155,3 +155,3 @@
 
-void wxFrameManager::DoSerialize( wxObjectStorage& store )
+void wxFrameManager::DoSerialize( wxObjectStorage& WXUNUSED(store) )
 {
@@ -295,3 +295,3 @@
 
-    if ( mActiveViewNo >= mViews.GetCount() )
+    if ( mActiveViewNo >= (int)mViews.GetCount() )
         mActiveViewNo = -1;
@@ -310,3 +310,3 @@
 
-void wxFrameManager::RemoveView( wxFrameView* pFrmView )
+void wxFrameManager::RemoveView( wxFrameView* WXUNUSED(pFrmView) )
 {
diff -bu1 -r h:/wxcvshead/wxWindows/contrib/src/fl/garbagec.cpp h:/wxcvshd/wxWin/contrib/src/fl/garbagec.cpp
--- h:/wxcvshead/wxWindows/contrib/src/fl/garbagec.cpp	2003-07-04 15:49:39.000000000 +0200
+++ h:/wxcvshd/wxWin/contrib/src/fl/garbagec.cpp	2003-08-19 13:07:03.000000000 +0200
@@ -142,3 +142,3 @@
 
-void GarbageCollector::AddObject( void* pObj, int refCnt )
+void GarbageCollector::AddObject( void* pObj, int WXUNUSED(refCnt) )
 {
diff -bu1 -r h:/wxcvshead/wxWindows/contrib/src/fl/newbmpbtn.cpp h:/wxcvshd/wxWin/contrib/src/fl/newbmpbtn.cpp
--- h:/wxcvshead/wxWindows/contrib/src/fl/newbmpbtn.cpp	2003-04-02 16:11:27.000000000 +0200
+++ h:/wxcvshd/wxWin/contrib/src/fl/newbmpbtn.cpp	2003-08-19 13:10:08.000000000 +0200
@@ -273,7 +273,7 @@
                                       bool isFlat,
-                                      int  firedEventType, 
-                                      int  marginX,
-                                      int  marginY,
-                                      int  textToLabelGap,
-                                      bool isSticky)
+                                      int  WXUNUSED(firedEventType), 
+                                      int  WXUNUSED(marginX),
+                                      int  WXUNUSED(marginY),
+                                      int  WXUNUSED(textToLabelGap),
+                                      bool WXUNUSED(isSticky))
 
@@ -654,3 +654,3 @@
 
-void wxNewBitmapButton::OnLButtonDown( wxMouseEvent& event )
+void wxNewBitmapButton::OnLButtonDown( wxMouseEvent& WXUNUSED(event) )
 {
@@ -689,3 +689,3 @@
 
-void wxNewBitmapButton::OnMouseEnter( wxMouseEvent& event )
+void wxNewBitmapButton::OnMouseEnter( wxMouseEvent& WXUNUSED(event) )
 {
@@ -703,3 +703,3 @@
 
-void wxNewBitmapButton::OnMouseLeave( wxMouseEvent& event )
+void wxNewBitmapButton::OnMouseLeave( wxMouseEvent& WXUNUSED(event) )
 {
@@ -718,3 +718,3 @@
 
-void wxNewBitmapButton::OnSize( wxSizeEvent& event )
+void wxNewBitmapButton::OnSize( wxSizeEvent& WXUNUSED(event) )
 {
@@ -777,3 +777,3 @@
 
-void wxNewBitmapButton::OnPaint( wxPaintEvent& event )
+void wxNewBitmapButton::OnPaint( wxPaintEvent& WXUNUSED(event) )
 {
@@ -789,3 +789,3 @@
 
-void wxNewBitmapButton::OnEraseBackground( wxEraseEvent& event )
+void wxNewBitmapButton::OnEraseBackground( wxEraseEvent& WXUNUSED(event) )
 {
@@ -794,3 +794,3 @@
 
-void wxNewBitmapButton::OnKillFocus( wxFocusEvent& event )
+void wxNewBitmapButton::OnKillFocus( wxFocusEvent& WXUNUSED(event) )
 {
diff -bu1 -r h:/wxcvshead/wxWindows/contrib/src/fl/panedrawpl.cpp h:/wxcvshd/wxWin/contrib/src/fl/panedrawpl.cpp
--- h:/wxcvshead/wxWindows/contrib/src/fl/panedrawpl.cpp	2003-02-21 16:02:33.000000000 +0100
+++ h:/wxcvshd/wxWin/contrib/src/fl/panedrawpl.cpp	2003-08-19 13:10:25.000000000 +0200
@@ -1278,3 +1278,3 @@
 
-void cbPaneDrawPlugin::OnFinishDrawInArea( cbFinishDrawInAreaEvent& event )
+void cbPaneDrawPlugin::OnFinishDrawInArea( cbFinishDrawInAreaEvent& WXUNUSED(event) )
 {
diff -bu1 -r h:/wxcvshead/wxWindows/contrib/src/fl/rowdragpl.cpp h:/wxcvshd/wxWin/contrib/src/fl/rowdragpl.cpp
--- h:/wxcvshead/wxWindows/contrib/src/fl/rowdragpl.cpp	2003-07-04 15:49:40.000000000 +0200
+++ h:/wxcvshd/wxWin/contrib/src/fl/rowdragpl.cpp	2003-08-19 13:10:44.000000000 +0200
@@ -1411,3 +1411,3 @@
 
-void cbRowDragPlugin::DrawRowsDragHintsBorder( wxDC& dc )
+void cbRowDragPlugin::DrawRowsDragHintsBorder( wxDC& WXUNUSED(dc) )
 {
diff -bu1 -r h:/wxcvshead/wxWindows/contrib/src/fl/rowlayoutpl.cpp h:/wxcvshd/wxWin/contrib/src/fl/rowlayoutpl.cpp
--- h:/wxcvshead/wxWindows/contrib/src/fl/rowlayoutpl.cpp	2003-06-05 11:38:34.000000000 +0200
+++ h:/wxcvshd/wxWin/contrib/src/fl/rowlayoutpl.cpp	2003-08-19 13:24:08.000000000 +0200
@@ -122,2 +122,4 @@
 
+    #if 1
+
    // FIXME:: something's wrong?
@@ -125,2 +127,4 @@
 
+    #else
+
     double freeSpc = (double)GetRowFreeSpace( pRow );
@@ -158,10 +162,14 @@
     }
+    #endif
 }
 
-void cbRowLayoutPlugin::AdjustLengthOfInserted( cbRowInfo* pRow, cbBarInfo* pTheBar )
+void cbRowLayoutPlugin::AdjustLengthOfInserted( cbRowInfo* WXUNUSED(pRow), cbBarInfo* WXUNUSED(pTheBar) )
 {
-    return;  // TBD: Makes following code unreachable
+    return;
 
-    // pTheBar is not-fixed
+#if 0
 
+    // TBD: Makes following code unreachable
+
+    // pTheBar is not-fixed
 
@@ -205,2 +213,5 @@
 #endif
+
+#endif
+
 }
@@ -669,3 +680,3 @@
 
-void cbRowLayoutPlugin::ShiftLeftTrashold( cbBarInfo* pTheBar, cbRowInfo& row )
+void cbRowLayoutPlugin::ShiftLeftTrashold( cbBarInfo* WXUNUSED(pTheBar), cbRowInfo& row )
 {
diff -bu1 -r h:/wxcvshead/wxWindows/contrib/src/fl/toolwnd.cpp h:/wxcvshd/wxWin/contrib/src/fl/toolwnd.cpp
--- h:/wxcvshead/wxWindows/contrib/src/fl/toolwnd.cpp	2003-04-02 16:11:27.000000000 +0200
+++ h:/wxcvshd/wxWin/contrib/src/fl/toolwnd.cpp	2003-08-19 13:27:38.000000000 +0200
@@ -159,3 +159,3 @@
 
-void wxToolWindow::OnPaint( wxPaintEvent& event )
+void wxToolWindow::OnPaint( wxPaintEvent& WXUNUSED(event) )
 {
@@ -674,3 +674,3 @@
 
-void wxToolWindow::OnSize( wxSizeEvent& event )
+void wxToolWindow::OnSize( wxSizeEvent& WXUNUSED(event) )
 {
@@ -699,3 +699,3 @@
 
-void wxToolWindow::OnEraseBackground( wxEraseEvent& event )
+void wxToolWindow::OnEraseBackground( wxEraseEvent& WXUNUSED(event) )
 {
@@ -757,3 +757,3 @@
 
-void cbMiniButton::OnLeftUp( const wxPoint& pos )
+void cbMiniButton::OnLeftUp( const wxPoint& WXUNUSED(pos) )
 {
@@ -1139,3 +1139,3 @@
 
-void cbFloatedBarWindow::OnDblClick( wxMouseEvent& event )
+void cbFloatedBarWindow::OnDblClick( wxMouseEvent& WXUNUSED(event) )
 {
diff -bu1 -r h:/wxcvshead/wxWindows/contrib/src/fl/updatesmgr.cpp h:/wxcvshd/wxWin/contrib/src/fl/updatesmgr.cpp
--- h:/wxcvshead/wxWindows/contrib/src/fl/updatesmgr.cpp	2003-07-04 15:49:40.000000000 +0200
+++ h:/wxcvshd/wxWin/contrib/src/fl/updatesmgr.cpp	2003-08-19 13:26:14.000000000 +0200
@@ -104,3 +104,3 @@
 
-void cbSimpleUpdatesMgr::OnRowWillChange( cbRowInfo* pRow, cbDockPane* pInPane )
+void cbSimpleUpdatesMgr::OnRowWillChange( cbRowInfo* WXUNUSED(pRow), cbDockPane* WXUNUSED(pInPane) )
 {
@@ -109,4 +109,4 @@
 
-void cbSimpleUpdatesMgr::OnBarWillChange( cbBarInfo* pBar, 
-                                          cbRowInfo* pInRow, cbDockPane* pInPane )
+void cbSimpleUpdatesMgr::OnBarWillChange( cbBarInfo* WXUNUSED(pBar), 
+                                          cbRowInfo* WXUNUSED(pInRow), cbDockPane* WXUNUSED(pInPane) )
 {
@@ -115,3 +115,3 @@
 
-void cbSimpleUpdatesMgr::OnPaneMarginsWillChange( cbDockPane* pPane )
+void cbSimpleUpdatesMgr::OnPaneMarginsWillChange( cbDockPane* WXUNUSED(pPane) )
 {
@@ -120,3 +120,3 @@
 
-void cbSimpleUpdatesMgr::OnPaneWillChange( cbDockPane* pPane )
+void cbSimpleUpdatesMgr::OnPaneWillChange( cbDockPane* WXUNUSED(pPane) )
 {

 	  	 
