--- ./cvs/wx2/wxwindows/src/msw/window.cpp	Fri Jun  1 06:28:00 2001
+++ ./wx2/src/msw/window.cpp	Sat Jun  2 06:56:40 2001
@@ -298,12 +298,15 @@
             if ( frame->GetLastFocus() == this )
                 frame->SetLastFocus((wxWindow*)NULL);
-            //break;
+            break;
         }
     }
 
+    DestroyChildren();
+
+    // Remove window from parent _after_ deleting its children,
+    // so that parent window chain does not get broken for the
+    // children (causes problems with wxFrame::m_winLastFocus)
     if ( m_parent )
         m_parent->RemoveChild(this);
-
-    DestroyChildren();
 
     if ( m_hWnd )

 	  	 
