diff -rbu2 ./cvs/wx2/wxwindows/include/wx/msw/frame.h ./wx2/include/wx/msw/frame.h
--- ./cvs/wx2/wxwindows/include/wx/msw/frame.h	Wed May 30 06:13:48 2001
+++ ./wx2/include/wx/msw/frame.h	Wed May 30 20:11:50 2001
@@ -117,8 +117,4 @@
     void SendSizeEvent();
 
-    // called by wxWindow whenever it gets focus
-    void SetLastFocus(wxWindow *win) { m_winLastFocused = win; }
-    wxWindow *GetLastFocus() const { return m_winLastFocused; }
-
 protected:
     // common part of all ctors
@@ -167,5 +163,5 @@
 
     // the last focused child: we restore focus to it on activation
-    wxWindow             *m_winLastFocused;
+    WXHWND                m_winLastFocused;
 
     // Data to save/restore when calling ShowFullScreen
diff -rbu2 ./cvs/wx2/wxwindows/src/msw/frame.cpp ./wx2/src/msw/frame.cpp
--- ./cvs/wx2/wxwindows/src/msw/frame.cpp	Wed May 30 06:13:50 2001
+++ ./wx2/src/msw/frame.cpp	Wed May 30 20:15:04 2001
@@ -113,5 +113,5 @@
     m_fsIsShowing = FALSE;
 
-    m_winLastFocused = (wxWindow *)NULL;
+    m_winLastFocused = 0;
 
     // unlike (almost?) all other windows, frames are created hidden
@@ -759,40 +759,24 @@
         wxLogTrace(_T("focus"), _T("wxFrame %08x activated."), m_hWnd);
 
-        wxWindow *parent = m_winLastFocused ? m_winLastFocused->GetParent()
-                                            : NULL;
-        if ( !parent )
-        {
-            parent = this;
-        }
-
-        wxSetFocusToChild(parent, &m_winLastFocused);
+        if (m_winLastFocused &&
+            (::GetFocus() != (HWND)m_winLastFocused) &&     // Avoid flicker by multiply setting the focus
+            ::IsWindow((HWND)m_winLastFocused) &&           // make sure the window still exists
+            ::IsChild((HWND)m_hWnd,(HWND)m_winLastFocused)) // make sure it is not reparented
+             ::SetFocus((HWND)m_winLastFocused);            // Set the focus (this will create a focus event)
     }
     else // deactivating
     {
         // remember the last focused child if it is our child
-        m_winLastFocused = FindFocus();
+        m_winLastFocused = (WXHWND)::GetFocus();
 
         // so we NULL it out if it's a child from some other frame
-        wxWindow *win = m_winLastFocused;
-        while ( win )
-        {
-            if ( win->IsTopLevel() )
-            {
-                if ( win != this )
-                {
-                    m_winLastFocused = NULL;
-                }
-
-                break;
-            }
-
-            win = win->GetParent();
-        }
+        if (m_winLastFocused &&
+            !::IsChild((HWND)m_hWnd,(HWND)m_winLastFocused))
+            m_winLastFocused = 0;
 
         wxLogTrace(_T("focus"),
                    _T("wxFrame %08x deactivated, last focused: %08x."),
                    m_hWnd,
-                   m_winLastFocused ? GetHwndOf(m_winLastFocused)
-                                    : NULL);
+                    m_winLastFocused ? m_winLastFocused : NULL);
 
         event.Skip();
diff -rbu2 ./cvs/wx2/wxwindows/src/msw/window.cpp ./wx2/src/msw/window.cpp
--- ./cvs/wx2/wxwindows/src/msw/window.cpp	Wed May 30 06:13:52 2001
+++ ./wx2/src/msw/window.cpp	Wed May 30 20:17:04 2001
@@ -290,16 +290,4 @@
     MSWDetachWindowMenu();
     
-    // VS: make sure there's no wxFrame with last focus set to us:
-    for (wxWindow *win = GetParent(); win; win = win->GetParent())
-    {
-        wxFrame *frame = wxDynamicCast(win, wxFrame);
-        if ( frame )
-        {
-            if ( frame->GetLastFocus() == this )
-                frame->SetLastFocus((wxWindow*)NULL);
-            break;
-        }
-    }
-
     if ( m_parent )
         m_parent->RemoveChild(this);

 	  	 
