Index: src/unix/gsocket.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/unix/gsocket.cpp,v
retrieving revision 1.47
diff -u -r1.47 gsocket.cpp
--- src/unix/gsocket.cpp	2005/09/03 23:16:50	1.47
+++ src/unix/gsocket.cpp	2005/09/07 13:03:34
@@ -965,6 +965,8 @@
     struct timeval tv;
 
     assert(this);
+    if (m_fd == -1)
+      return (GSOCK_LOST_FLAG & flags);
 
     /* Do not use a static struct, Linux can garble it */
     tv.tv_sec = m_timeout / 1000;

 	  	 
