--- src/common/sizer.cpp.old	2005-04-24 18:00:18.000000000 +0400
+++ src/common/sizer.cpp	2006-04-14 14:26:22.000000000 +0400
@@ -534,8 +534,10 @@
 
     wxSizerItem *item = node->GetData();
 
-    if ( item->IsWindow() )
+    if ( item->IsWindow() ) {
         item->GetWindow()->SetContainingSizer( NULL );
+        delete item->GetWindow();
+    }
 
     delete item;
     m_children.Erase( node );

 	  	 
