Index: src/html/htmlwin.cpp
===================================================================
--- src/html/htmlwin.cpp	(revision 73539)
+++ src/html/htmlwin.cpp	(working copy)
@@ -1090,7 +1090,7 @@
     if ( m_bmpBg.IsOk() )
     {
         // draw the background bitmap tiling it over the entire window area
-        const wxSize sz = GetClientSize();
+        const wxSize sz = GetVirtualSize();
         const wxSize sizeBmp(m_bmpBg.GetWidth(), m_bmpBg.GetHeight());
         for ( wxCoord x = 0; x < sz.x; x += sizeBmp.x )
         {
