Index: src/osx/carbon/renderer.cpp
===================================================================
--- src/osx/carbon/renderer.cpp	(revision 66840)
+++ src/osx/carbon/renderer.cpp	(working copy)
@@ -26,6 +26,7 @@
     #include "wx/dcclient.h"
     #include "wx/dcmemory.h"
     #include "wx/toplevel.h"
+	#include "wx/splitter.h"
 #endif
 
 #include "wx/renderer.h"
@@ -125,6 +126,8 @@
                             int flags,
                             int kind,
                             int adornment);
+						
+	wxSplitterRenderParams GetSplitterParams(const wxWindow *win);
 
     // the tree buttons
     wxBitmap m_bmpTreeExpanded;
@@ -274,6 +277,37 @@
     }
 }
 
+wxSplitterRenderParams 
+wxRendererMac::GetSplitterParams(const wxWindow *win)
+{
+	// see below
+    SInt32 sashWidth,
+            border;
+#if defined(wxOSX_USE_COCOA)
+    if ( win->HasFlag(wxSP_3DSASH) )
+		GetThemeMetric( kThemeMetricPaneSplitterHeight, &sashWidth ); // Cocoa == Carbon == 7
+    else if ( win->HasFlag(wxSP_NOSASH) ) // actually Cocoa doesn't allow 0
+        sashWidth = 0;
+    else // no 3D effect - Cocoa [NSSplitView dividerThickNess] for NSSplitViewDividerStyleThin
+        sashWidth = 1;
+#else
+    if ( win->HasFlag(wxSP_3DSASH) )
+        GetThemeMetric( kThemeMetricPaneSplitterHeight, &sashWidth );
+    else if ( win->HasFlag(wxSP_NOSASH) )
+        sashWidth = 0;
+    else // no 3D effect
+        GetThemeMetric( kThemeMetricSmallPaneSplitterHeight, &sashWidth );
+#endif
+
+    if ( win->HasFlag(wxSP_3DBORDER) )
+        border = 2;
+    else // no 3D effect
+        border = 0;
+  
+	return wxSplitterRenderParams(sashWidth, border, false);
+}
+
+
 void wxRendererMac::DrawSplitterSash( wxWindow *win,
     wxDC& dc,
     const wxSize& size,
@@ -283,7 +317,9 @@
 {
     bool hasMetal = win->MacGetTopLevelWindow()->GetExtraStyle() & wxFRAME_EX_METAL;
     SInt32 height;
-    GetThemeMetric( kThemeMetricSmallPaneSplitterHeight, &height );
+	
+	height = wxRendererNative::Get().GetSplitterParams( win ).widthSash;
+
     HIRect splitterRect;
     if (orient == wxVERTICAL)
         splitterRect = CGRectMake( position, 0, height, size.y );
@@ -317,12 +353,13 @@
             CGContextSetFillColorWithColor(cgContext,win->GetBackgroundColour().GetCGColor());
             CGContextFillRect(cgContext,splitterRect);
         }
-
-        HIThemeSplitterDrawInfo drawInfo;
-        drawInfo.version = 0;
-        drawInfo.state = kThemeStateActive;
-        drawInfo.adornment = hasMetal ? kHIThemeSplitterAdornmentMetal : kHIThemeSplitterAdornmentNone;
-        HIThemeDrawPaneSplitter( &splitterRect, &drawInfo, cgContext, kHIThemeOrientationNormal );
+  		if ( win->HasFlag(wxSP_3DSASH) ) {
+	        HIThemeSplitterDrawInfo drawInfo;
+	        drawInfo.version = 0;
+	        drawInfo.state = kThemeStateActive;
+	        drawInfo.adornment = hasMetal ? kHIThemeSplitterAdornmentMetal : kHIThemeSplitterAdornmentNone;
+	        HIThemeDrawPaneSplitter( &splitterRect, &drawInfo, cgContext, kHIThemeOrientationNormal );
+		}
     }
 }
 
