Index: src/common/dynlib.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/common/dynlib.cpp,v
retrieving revision 1.117
diff -u -r1.117 dynlib.cpp
--- src/common/dynlib.cpp	2006/10/09 15:58:55	1.117
+++ src/common/dynlib.cpp	2007/04/06 07:35:29
@@ -120,7 +120,7 @@

 #elif defined(__WXPM__) || defined(__EMX__)
     char err[256] = "";
-    DosLoadModule(err, sizeof(err), (PSZ)libname.c_str(), &m_handle);
+    DosLoadModule(err, sizeof(err), libname.c_str(), &m_handle);
 #else // this should be the only remaining branch eventually
     m_handle = RawLoad(libname, flags);
 #endif
@@ -177,7 +177,7 @@
     if( FindSymbol( m_handle, symName, &symAddress, &symClass ) == noErr )
         symbol = (void *)symAddress;
 #elif defined(__WXPM__) || defined(__EMX__)
-    DosQueryProcAddr( m_handle, 1L, (PSZ)name.c_str(), (PFN*)symbol );
+    DosQueryProcAddr( m_handle, 1L, name.c_str(), (PFN*)symbol );
 #else
     symbol = RawGetSymbol(m_handle, name);
 #endif
Index: src/common/filefn.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/common/filefn.cpp,v
retrieving revision 1.290
diff -u -r1.290 filefn.cpp
--- src/common/filefn.cpp	2007/03/30 16:27:52	1.290
+++ src/common/filefn.cpp	2007/04/06 07:35:33
@@ -1075,7 +1075,7 @@
         return false;
     }
 #elif defined(__OS2__)
-    if ( ::DosCopy((PSZ)file1.c_str(), (PSZ)file2.c_str(), overwrite ? DCPY_EXISTING : 0) != 0 )
+    if ( ::DosCopy(file1.c_str(), file2.c_str(), overwrite ? DCPY_EXISTING : 0) != 0 )
         return false;
 #elif defined(__PALMOS__)
     // TODO with http://www.palmos.com/dev/support/docs/protein_books/Memory_Databases_Files/
@@ -1267,7 +1267,7 @@
 #if defined(__VMS__)
     return false; //to be changed since rmdir exists in VMS7.x
 #elif defined(__OS2__)
-    return (::DosDeleteDir((PSZ)dir.c_str()) == 0);
+    return (::DosDeleteDir(dir.c_str()) == 0);
 #elif defined(__WXWINCE__)
     return (RemoveDirectory(dir) != 0);
 #elif defined(__WXPALMOS__)
@@ -1582,7 +1582,7 @@
 	if (d.length() == 2)
 	    return true;
     }
-    return (::DosSetCurrentDir((PSZ)d.c_str()) == 0);
+    return (::DosSetCurrentDir(d.c_str()) == 0);
 #elif defined(__UNIX__) || defined(__WXMAC__) || defined(__DOS__)
     return (chdir(wxFNSTRINGCAST d.fn_str()) == 0);
 #elif defined(__WINDOWS__)
Index: src/os2/accel.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/os2/accel.cpp,v
retrieving revision 1.24
diff -u -r1.24 accel.cpp
--- src/os2/accel.cpp	2006/10/08 21:56:47	1.24
+++ src/os2/accel.cpp	2007/04/06 07:35:34
@@ -75,7 +75,7 @@

     m_refData = new wxAcceleratorRefData;

-    ulId = atol((char*)rResource.c_str());
+    ulId = atol(rResource.c_str());
     hAccel = ::WinLoadAccelTable( vHabmain
                                  ,NULL // resources always in .exe
                                  ,(ULONG)ulId
Index: src/os2/button.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/os2/button.cpp,v
retrieving revision 1.33
diff -u -r1.33 button.cpp
--- src/os2/button.cpp	2006/07/26 05:23:40	1.33
+++ src/os2/button.cpp	2007/04/06 07:35:35
@@ -80,7 +80,7 @@

     m_hWnd = (WXHWND)::WinCreateWindow( GetHwndOf(pParent)   // Parent handle
                                        ,WC_BUTTON            // A Button class window
-                                       ,(PSZ)sLabel.c_str()  // Button text
+                                       ,sLabel.c_str()       // Button text
                                        ,lStyle               // Button style
                                        ,0, 0, 0, 0           // Location and size
                                        ,GetHwndOf(pParent)   // Owner handle
Index: src/os2/checkbox.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/os2/checkbox.cpp,v
retrieving revision 1.27
diff -u -r1.27 checkbox.cpp
--- src/os2/checkbox.cpp	2006/05/29 08:13:13	1.27
+++ src/os2/checkbox.cpp	2007/04/06 07:35:36
@@ -96,7 +96,7 @@
 void wxCheckBox::SetLabel( const wxString& rsLabel )
 {
     wxString  sLabel=::wxPMTextToLabel(rsLabel);
-    ::WinSetWindowText(GetHwnd(), (PSZ)sLabel.c_str());
+    ::WinSetWindowText(GetHwnd(), sLabel.c_str());
 } // end of wxCheckBox::SetLabel

 wxSize wxCheckBox::DoGetBestSize() const
Index: src/os2/choice.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/os2/choice.cpp,v
retrieving revision 1.41
diff -u -r1.41 choice.cpp
--- src/os2/choice.cpp	2006/10/31 08:49:37	1.41
+++ src/os2/choice.cpp	2007/04/06 07:35:36
@@ -134,7 +134,7 @@
     nIndex = (int)::WinSendMsg( GetHwnd()
                                ,LM_INSERTITEM
                                ,(MPARAM)nIndexType
-                               ,(MPARAM)rsItem.c_str()
+                               ,(MPARAM)(const char*)rsItem.c_str()
                               );
     return nIndex;
 } // end of wxChoice::DoAppend
@@ -157,7 +157,7 @@
     nIndex = (int)::WinSendMsg( GetHwnd()
                                ,LM_INSERTITEM
                                ,(MPARAM)nIndexType
-                               ,(MPARAM)rsItem.c_str()
+                               ,(MPARAM)(const char*)rsItem.c_str()
                               );
     return nIndex;
 } // end of wxChoice::DoInsert
@@ -232,7 +232,7 @@
     ::WinSendMsg( GetHwnd()
                  ,LM_INSERTITEM
                  ,(MPARAM)nIndexType
-                 ,(MPARAM)rsStr.c_str()
+                 ,(MPARAM)(const char*)rsStr.c_str()
                 );

     if (pData)
Index: src/os2/combobox.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/os2/combobox.cpp,v
retrieving revision 1.34
diff -u -r1.34 combobox.cpp
--- src/os2/combobox.cpp	2006/10/31 20:46:35	1.34
+++ src/os2/combobox.cpp	2007/04/06 07:35:37
@@ -193,7 +193,7 @@
     if ( HasFlag(wxCB_READONLY) )
         SetStringSelection(rsValue);
     else
-        ::WinSetWindowText(GetHwnd(), (PSZ)rsValue.c_str());
+        ::WinSetWindowText(GetHwnd(), rsValue.c_str());
 } // end of wxComboBox::SetValue

 //
@@ -299,7 +299,7 @@
     // Now replace with 'value', by pasting.
     //
     wxSetClipboardData( wxDF_TEXT
-                       ,(wxObject *)rsValue.c_str()
+                       ,(wxObject *)(const char*)rsValue.c_str()
                        ,0
                        ,0
                       );
Index: src/os2/control.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/os2/control.cpp,v
retrieving revision 1.44
diff -u -r1.44 control.cpp
--- src/os2/control.cpp	2007/03/25 22:31:27	1.44
+++ src/os2/control.cpp	2007/04/06 07:35:38
@@ -136,7 +136,7 @@

     m_hWnd = (WXHWND)::WinCreateWindow( (HWND)GetHwndOf(pParent) // Parent window handle
                                        ,(PSZ)zClass              // Window class
-                                       ,(PSZ)label.c_str()       // Initial Text
+                                       ,label.c_str()            // Initial Text
                                        ,(ULONG)dwStyle           // Style flags
                                        ,(LONG)0                  // X pos of origin
                                        ,(LONG)0                  // Y pos of origin
@@ -244,7 +244,7 @@
             label = ::wxPMTextToLabel(m_label);
         else
             label = m_label;
-        ::WinSetWindowText(GetHwnd(), (PSZ)label.c_str());
+        ::WinSetWindowText(GetHwnd(), label.c_str());
     }
 } // end of wxControl::SetLabel

Index: src/os2/dc.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/os2/dc.cpp,v
retrieving revision 1.83
diff -u -r1.83 dc.cpp
--- src/os2/dc.cpp	2006/10/31 23:50:32	1.83
+++ src/os2/dc.cpp	2007/04/06 07:35:43
@@ -1753,7 +1753,7 @@
             vPtlStart.y = vY;
     }

-    PCH pzStr = (PCH)rsText.c_str();
+    PCH pzStr = (char *)(const char*)rsText.c_str();

     ::GpiMove(m_hPS, &vPtlStart);
     lHits = ::GpiCharString( m_hPS
@@ -2178,7 +2178,7 @@
     //
     bRc = ::GpiQueryTextBox( m_hPS
                             ,l
-                            ,(PCH)rsString.c_str()
+                            ,(char *)(const char*)rsString.c_str()
                             ,TXTBOX_COUNT // return maximum information
                             ,avPoint      // array of coordinates points
                            );
Index: src/os2/dcprint.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/os2/dcprint.cpp,v
retrieving revision 1.19
diff -u -r1.19 dcprint.cpp
--- src/os2/dcprint.cpp	2006/10/27 13:07:33	1.19
+++ src/os2/dcprint.cpp	2007/04/06 07:35:44
@@ -41,8 +41,8 @@
                           bool bInteractive,
                           int nOrientation )
 {
-    DEVOPENSTRUC    vDevOpen = { (char*)rsDeviceName.c_str()
-                                ,(char*)rsDriverName.c_str()
+    DEVOPENSTRUC    vDevOpen = { (char*)(const char*)rsDeviceName.c_str()
+                                ,(char*)(const char*)rsDriverName.c_str()
                                 ,NULL
                                 ,NULL
                                 ,NULL
Index: src/os2/dialog.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/os2/dialog.cpp,v
retrieving revision 1.31
diff -u -r1.31 dialog.cpp
--- src/os2/dialog.cpp	2006/08/20 13:29:42	1.31
+++ src/os2/dialog.cpp	2007/04/06 07:35:44
@@ -222,7 +222,7 @@

     wxString title = GetTitle();
     if (!title.empty())
-        ::WinSetWindowText((HWND)GetHwnd(), (PSZ)title.c_str());
+        ::WinSetWindowText((HWND)GetHwnd(), title.c_str());

     if ( bShow )
     {
Index: src/os2/dir.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/os2/dir.cpp,v
retrieving revision 1.13
diff -u -r1.13 dir.cpp
--- src/os2/dir.cpp	2005/09/23 12:55:29	1.13
+++ src/os2/dir.cpp	2007/04/06 07:35:45
@@ -75,7 +75,7 @@
     FIND_DATA                   hDir = HDIR_CREATE;
     FIND_ATTR                   rc;

-    rc = ::DosFindFirst( (PSZ)rsSpec.c_str()
+    rc = ::DosFindFirst( rsSpec.c_str()
                         ,&hDir
                         ,0x37 // was: FILE_NORMAL
                         ,pFinddata
Index: src/os2/dnd.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/os2/dnd.cpp,v
retrieving revision 1.23
diff -u -r1.23 dnd.cpp
--- src/os2/dnd.cpp	2006/06/19 20:18:41	1.23
+++ src/os2/dnd.cpp	2007/04/06 07:35:46
@@ -387,7 +387,7 @@
         for (i = 0; i < ulItems; i++)
         {
             pDragItem = ::DrgQueryDragitemPtr(pDataSource, i);
-            if (::DrgVerifyRMF(pDragItem, (PSZ)sMechanism.c_str(), (PSZ)sFormat.c_str()))
+            if (::DrgVerifyRMF(pDragItem, sMechanism.c_str(), sFormat.c_str()))
             {
                 bValid = true;
                 break;
Index: src/os2/filedlg.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/os2/filedlg.cpp,v
retrieving revision 1.22
diff -u -r1.22 filedlg.cpp
--- src/os2/filedlg.cpp	2007/03/07 11:48:53	1.22
+++ src/os2/filedlg.cpp	2007/04/06 07:35:47
@@ -193,7 +193,7 @@
     else
         sTheFilter = m_wildCard;

-    wxStrtok((wxChar*)sTheFilter.c_str(), wxT("|"), &pzFilterBuffer);
+    wxStrtok((char*)(const char*)sTheFilter.c_str(), wxT("|"), &pzFilterBuffer);
     while(pzFilterBuffer != NULL)
     {
         if (nCount > 0 && !(nCount % 2))
Index: src/os2/gdiimage.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/os2/gdiimage.cpp,v
retrieving revision 1.23
diff -u -r1.23 gdiimage.cpp
--- src/os2/gdiimage.cpp	2006/10/23 11:58:28	1.23
+++ src/os2/gdiimage.cpp	2007/04/06 07:35:48
@@ -469,7 +469,7 @@
 {
     HPOINTER                        hIcon;

-    hIcon = ::WinLoadFileIcon( (PSZ)rName.c_str()
+    hIcon = ::WinLoadFileIcon( rName.c_str()
                               ,TRUE // load for private use
                              );

Index: src/os2/listbox.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/os2/listbox.cpp,v
retrieving revision 1.44
diff -u -r1.44 listbox.cpp
--- src/os2/listbox.cpp	2006/05/04 07:56:52	1.44
+++ src/os2/listbox.cpp	2007/04/06 07:35:49
@@ -281,7 +281,10 @@
     else
         lIndexType = LIT_END;

-    lIndex = (long)::WinSendMsg(GetHwnd(), LM_INSERTITEM, (MPARAM)lIndexType, (MPARAM)rsItem.c_str());
+    lIndex = (long)::WinSendMsg(  GetHwnd(),
+                                  LM_INSERTITEM,
+                                  (MPARAM)lIndexType,
+                                  (MPARAM)(const char*)rsItem.c_str());
     m_nNumItems++;

 #if wxUSE_OWNER_DRAWN
@@ -318,7 +321,10 @@
             lIndexType = LIT_SORTASCENDING;
         else
             lIndexType = LIT_END;
-        ::WinSendMsg(GetHwnd(), LM_INSERTITEM, (MPARAM)lIndexType, (MPARAM)raChoices[i].c_str());
+        ::WinSendMsg( GetHwnd(),
+                      LM_INSERTITEM,
+                      (MPARAM)lIndexType,
+                      (MPARAM)(const char*)raChoices[i].c_str());

         if (ppClientData)
         {
@@ -563,7 +569,7 @@
         int nIndex = (int)::WinSendMsg( GetHwnd(),
                                         LM_INSERTITEM,
                                         MPFROMLONG((LONG)(i + nPos)),
-                                        (MPARAM)asItems[i].c_str() );
+                                        (MPARAM)(const char*)asItems[i].c_str() );

         wxOwnerDrawn* pNewItem = CreateItem(nIndex);

@@ -613,7 +619,7 @@
     ::WinSendMsg( GetHwnd()
                  ,LM_INSERTITEM
                  ,(MPARAM)nNewN
-                 ,(MPARAM)rsString.c_str()
+                 ,(MPARAM)(const char*)rsString.c_str()
                 );

     //
Index: src/os2/menu.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/os2/menu.cpp,v
retrieving revision 1.57
diff -u -r1.57 menu.cpp
--- src/os2/menu.cpp	2006/09/04 23:56:47	1.57
+++ src/os2/menu.cpp	2007/04/06 07:35:51
@@ -287,7 +287,7 @@
         rItem.id = (USHORT)pItem->GetId();
     }

-    BYTE*                           pData=NULL;
+    char* pData=NULL;

 #if wxUSE_OWNER_DRAWN
     if (pItem->IsOwnerDrawn())
@@ -323,7 +323,7 @@
             //
             rItem.afStyle |= MIS_TEXT;
         }
-        pData = (char*)pItem->GetText().c_str();
+        pData = (char*)(const char*)pItem->GetText().c_str();
     }

     if (nPos == (size_t)-1)
@@ -562,7 +562,7 @@
     {
         if (!rLabel.empty())
         {
-            if (!::WinSetWindowText(hMenu, (PSZ)rLabel.c_str()))
+            if (!::WinSetWindowText(hMenu, rLabel.c_str()))
             {
                 wxLogLastError(wxT("SetMenuTitle"));
             }
@@ -583,7 +583,7 @@
             //
             // Modify the title
             //
-            if (!::WinSetWindowText(hMenu, (PSZ)rLabel.c_str()))
+            if (!::WinSetWindowText(hMenu, rLabel.c_str()))
             {
                 wxLogLastError(wxT("SetMenuTitle"));
             }
@@ -804,7 +804,10 @@

             (*it)->m_vMenuData.iPosition = (SHORT)i;

-            rc = (APIRET)::WinSendMsg(m_hMenu, MM_INSERTITEM, (MPARAM)&(*it)->m_vMenuData, (MPARAM)m_titles[i].c_str());
+            rc = (APIRET)::WinSendMsg(  m_hMenu,
+                                        MM_INSERTITEM,
+                                        (MPARAM)&(*it)->m_vMenuData,
+                                        (MPARAM)(const char*)m_titles[i].c_str());
             if (rc == (APIRET)MIT_MEMERROR || rc == (APIRET)MIT_ERROR)
             {
                 vError = ::WinGetLastError(vHabmain);
@@ -879,7 +882,10 @@
     }
     nId = vItem.id;

-    if (::WinSendMsg(GetHmenu(), MM_SETITEMTEXT, MPFROMSHORT(nId), (MPARAM)rLabel.c_str()));
+    if (::WinSendMsg( GetHmenu(),
+                      MM_SETITEMTEXT,
+                      MPFROMSHORT(nId),
+                      (MPARAM)(const char*)rLabel.c_str()));
     {
         wxLogLastError(wxT("ModifyMenu"));
     }
@@ -924,8 +930,14 @@
     m_titles[nPos] = sTitle;
     if (IsAttached())
     {
-        ::WinSendMsg((HWND)m_hMenu, MM_REMOVEITEM, MPFROM2SHORT(nId, TRUE), (MPARAM)0);
-        ::WinSendMsg((HWND)m_hMenu, MM_INSERTITEM, (MPARAM)&pMenu->m_vMenuData, (MPARAM)sTitle.c_str());
+        ::WinSendMsg( (HWND)m_hMenu,
+                      MM_REMOVEITEM,
+                      MPFROM2SHORT(nId, TRUE),
+                      (MPARAM)0);
+        ::WinSendMsg( (HWND)m_hMenu,
+                      MM_INSERTITEM,
+                      (MPARAM)&pMenu->m_vMenuData,
+                      (MPARAM)(const char*)sTitle.c_str());

 #if wxUSE_ACCEL
         if (pMenuOld->HasAccels() || pMenu->HasAccels())
@@ -958,7 +970,7 @@
         ::WinSendMsg( (HWND)m_hMenu
                      ,MM_INSERTITEM
                      ,(MPARAM)&pMenu->m_vMenuData
-                     ,(MPARAM)sTitle.c_str()
+                     ,(MPARAM)(const char*)sTitle.c_str()
                     );
 #if wxUSE_ACCEL
         if (pMenu->HasAccels())
@@ -990,7 +1002,10 @@
     if ( IsAttached() )
     {
         pMenu->m_vMenuData.iPosition = MIT_END;
-        ::WinSendMsg((HWND)m_hMenu, MM_INSERTITEM, (MPARAM)&pMenu->m_vMenuData, (MPARAM)sTitle.c_str());
+        ::WinSendMsg( (HWND)m_hMenu,
+                      MM_INSERTITEM,
+                      (MPARAM)&pMenu->m_vMenuData,
+                      (MPARAM)(const char*)sTitle.c_str());
 #if wxUSE_ACCEL
         if (pMenu->HasAccels())
         {
Index: src/os2/menuitem.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/os2/menuitem.cpp,v
retrieving revision 1.39
diff -u -r1.39 menuitem.cpp
--- src/os2/menuitem.cpp	2006/09/05 21:00:51	1.39
+++ src/os2/menuitem.cpp	2007/04/06 07:35:52
@@ -415,7 +415,7 @@
             uFlagsOld |= MIS_SUBMENU;
         }

-        BYTE*                       pData;
+        char* pData;

 #if wxUSE_OWNER_DRAWN
         if (IsOwnerDrawn())
@@ -427,7 +427,7 @@
 #endif  //owner drawn
         {
             uFlagsOld |= MIS_TEXT;
-            pData = (BYTE*)m_text.c_str();
+            pData = (char*)(const char*)m_text.c_str();
         }

         //
Index: src/os2/msgdlg.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/os2/msgdlg.cpp,v
retrieving revision 1.13
diff -u -r1.13 msgdlg.cpp
--- src/os2/msgdlg.cpp	2006/05/25 07:07:20	1.13
+++ src/os2/msgdlg.cpp	2007/04/06 07:35:52
@@ -107,12 +107,12 @@
     //
     hWnd = HWND_DESKTOP;

-    ULONG                           ulAns = ::WinMessageBox( hWnd
-                                                            ,hWnd
-                                                            ,(PSZ)m_sMessage.c_str()
-                                                            ,(PSZ)m_sCaption.c_str()
-                                                            ,0L
-                                                            ,ulStyle);
+    ULONG   ulAns = ::WinMessageBox(  hWnd,
+                                      hWnd,
+                                      m_sMessage.c_str(),
+                                      m_sCaption.c_str(),
+                                      0L,
+                                      ulStyle);
     switch (ulAns)
     {
         case MBID_CANCEL:
Index: src/os2/notebook.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/os2/notebook.cpp,v
retrieving revision 1.31
diff -u -r1.31 notebook.cpp
--- src/os2/notebook.cpp	2006/11/04 18:52:56	1.31
+++ src/os2/notebook.cpp	2007/04/06 07:35:54
@@ -266,7 +266,7 @@
     return (bool)::WinSendMsg( m_hWnd
                               ,BKM_SETTABTEXT
                               ,MPFROMLONG((ULONG)m_alPageId[nPage])
-                              ,MPFROMP((PSZ)rsStrText.c_str())
+                              ,MPFROMP((const char*)rsStrText.c_str())
                              );
 } // end of wxNotebook::SetPageText

Index: src/os2/ownerdrw.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/os2/ownerdrw.cpp,v
retrieving revision 1.30
diff -u -r1.30 ownerdrw.cpp
--- src/os2/ownerdrw.cpp	2006/10/08 21:28:44	1.30
+++ src/os2/ownerdrw.cpp	2007/04/06 07:35:55
@@ -351,7 +351,7 @@
     ::GpiCharStringAt( rDC.GetHPS()
                       ,&vPntStart
                       ,sFullString.length()
-                      ,(PCH)sFullString.c_str()
+                      ,(const char*)sFullString.c_str()
                      );
     if (bFoundMnemonic)
     {
@@ -389,7 +389,7 @@
         ::GpiCharStringAt( rDC.GetHPS()
                           ,&vPntStart
                           ,sAccel.length()
-                          ,(PCH)sAccel.c_str()
+                          ,(const char*)sAccel.c_str()
                          );
     }

Index: src/os2/radiobox.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/os2/radiobox.cpp,v
retrieving revision 1.44
diff -u -r1.44 radiobox.cpp
--- src/os2/radiobox.cpp	2006/06/07 13:25:59	1.44
+++ src/os2/radiobox.cpp	2007/04/06 07:35:58
@@ -781,7 +781,8 @@
     wxCHECK_RET( IsValid(nItem), wxT("invalid radiobox index") );

     m_pnRadioWidth[nItem] = m_pnRadioHeight[nItem] = -1;
-    ::WinSetWindowText((HWND)m_ahRadioButtons[nItem], (PSZ)rsLabel.c_str());
+    ::WinSetWindowText( (HWND)m_ahRadioButtons[nItem],
+                        rsLabel.c_str());
 } // end of wxRadioBox::SetString

 bool wxRadioBox::SetStringSelection(const wxString& rsStr)
Index: src/os2/spinctrl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/os2/spinctrl.cpp,v
retrieving revision 1.26
diff -u -r1.26 spinctrl.cpp
--- src/os2/spinctrl.cpp	2006/07/20 17:26:52	1.26
+++ src/os2/spinctrl.cpp	2007/04/06 07:35:59
@@ -456,7 +456,7 @@
 {
     long                            lVal;

-    lVal = atol((char*)rsText.c_str());
+    lVal = atol(rsText.c_str());
     wxSpinButton::SetValue(lVal);
 } // end of wxSpinCtrl::SetValue

Index: src/os2/stattext.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/os2/stattext.cpp,v
retrieving revision 1.26
diff -u -r1.26 stattext.cpp
--- src/os2/stattext.cpp	2007/04/01 14:13:13	1.26
+++ src/os2/stattext.cpp	2007/04/06 07:35:59
@@ -236,7 +236,7 @@
     m_labelOrig = rsLabel;       // save original label

     // OS/2 does not support neither ellipsize nor markup in static text:
-    DoSetLabel(GetEllipsizedLabelWithoutMarkup(label));
+    DoSetLabel(GetEllipsizedLabelWithoutMarkup());

     //
     // Adjust the size of the window to fit to the label unless autoresizing is
@@ -277,12 +277,13 @@
 void wxStaticText::DoSetLabel(const wxString& str)
 {
     wxString sLabel = ::wxPMTextToLabel(str);
-    ::WinSetWindowText(GetHwnd(), (PSZ)sLabel.c_str());
+    ::WinSetWindowText(GetHwnd(), sLabel.c_str());
 }

 wxString wxStaticText::DoGetLabel() const
 {
     // FIXME: how to retrieve the text?
     //return wxGetWindowText(GetHWND());
+    return wxString("");
 }

Index: src/os2/textctrl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/os2/textctrl.cpp,v
retrieving revision 1.52
diff -u -r1.52 textctrl.cpp
--- src/os2/textctrl.cpp	2006/10/08 17:46:07	1.52
+++ src/os2/textctrl.cpp	2007/04/06 07:36:02
@@ -160,7 +160,7 @@
     {
         m_hWnd = (WXHWND)::WinCreateWindow( (HWND)GetHwndOf(pParent) // Parent window handle
                                            ,WC_MLE                   // Window class
-                                           ,(PSZ)rsValue.c_str()     // Initial Text
+                                           ,rsValue.c_str()          // Initial Text
                                            ,(ULONG)lSstyle           // Style flags
                                            ,(LONG)0                  // X pos of origin
                                            ,(LONG)0                  // Y pos of origin
@@ -177,7 +177,7 @@
     {
         m_hWnd = (WXHWND)::WinCreateWindow( (HWND)GetHwndOf(pParent) // Parent window handle
                                            ,WC_ENTRYFIELD            // Window class
-                                           ,(PSZ)rsValue.c_str()     // Initial Text
+                                           ,rsValue.c_str()          // Initial Text
                                            ,(ULONG)lSstyle           // Style flags
                                            ,(LONG)0                  // X pos of origin
                                            ,(LONG)0                  // Y pos of origin
@@ -328,8 +328,8 @@

 wxString wxTextCtrl::GetValue() const
 {
-    wxString                        sStr = wxGetWindowText(GetHWND());
-    char*                           zStr = (char*)sStr.c_str();
+    wxString  sStr = wxGetWindowText(GetHWND());
+    char*     zStr = (char*)(const char*)sStr.c_str();

     for ( ; *zStr; zStr++ )
     {
@@ -360,7 +360,7 @@
         if ( flags & SetValue_SendEvent )
             m_bSkipUpdate = true;

-        ::WinSetWindowText(GetHwnd(), (PSZ)rsValue.c_str());
+        ::WinSetWindowText(GetHwnd(), rsValue.c_str());
         AdjustSpaceLimit();
     }
 } // end of wxTextCtrl::SetValue
@@ -370,9 +370,12 @@
 )
 {
     if (m_bIsMLE)
-        ::WinSendMsg(GetHwnd(), MLM_INSERT, MPARAM((PCHAR)rsValue.c_str()), MPARAM(0));
+        ::WinSendMsg( GetHwnd(),
+                      MLM_INSERT,
+                      MPARAM((const char*)rsValue.c_str()),
+                      MPARAM(0));
     else
-        ::WinSetWindowText(GetHwnd(), (PSZ)rsValue.c_str());
+        ::WinSetWindowText(GetHwnd(), rsValue.c_str());
     AdjustSpaceLimit();
 } // end of wxTextCtrl::WriteText

Index: src/os2/tooltip.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/os2/tooltip.cpp,v
retrieving revision 1.9
diff -u -r1.9 tooltip.cpp
--- src/os2/tooltip.cpp	2005/04/27 01:19:43	1.9
+++ src/os2/tooltip.cpp	2007/04/06 07:36:03
@@ -61,7 +61,7 @@

     m_hWnd = ::WinCreateWindow( HWND_DESKTOP
                                ,WC_ENTRYFIELD
-                               ,(PSZ)rsTip.c_str()
+                               ,rsTip.c_str()
                                ,lStyle
                                ,0, 0, 0, 0
                                ,NULLHANDLE
Index: src/os2/toplevel.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/os2/toplevel.cpp,v
retrieving revision 1.46
diff -u -r1.46 toplevel.cpp
--- src/os2/toplevel.cpp	2006/08/31 19:31:16	1.46
+++ src/os2/toplevel.cpp	2007/04/06 07:36:05
@@ -485,7 +485,7 @@
                                    ,ulStyleFlags          // frame-window style
                                    ,(PULONG)&lFlags       // window style
                                    ,(PSZ)wxFrameClassName // class name
-                                   ,(PSZ)rsTitle.c_str()  // window title
+                                   ,rsTitle.c_str()       // window title
                                    ,0L                    // default client style
                                    ,NULLHANDLE            // resource in executable file
                                    ,0                     // resource id
Index: src/os2/utils.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/os2/utils.cpp,v
retrieving revision 1.82
diff -u -r1.82 utils.cpp
--- src/os2/utils.cpp	2007/01/07 14:07:02	1.82
+++ src/os2/utils.cpp	2007/04/06 07:36:06
@@ -157,7 +157,7 @@
     SData.PgmName  = (char*)zShell;

     sInputs = _T("/C ") + rCommand;
-    SData.PgmInputs     = (BYTE*)sInputs.c_str();
+    SData.PgmInputs     = (BYTE*)(const char*)sInputs.c_str();
     SData.TermQ         = 0;
     SData.Environment   = 0;
     SData.InheritOpt    = SSF_INHERTOPT_SHELL;
Index: src/os2/utilsexc.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/os2/utilsexc.cpp,v
retrieving revision 1.25
diff -u -r1.25 utilsexc.cpp
--- src/os2/utilsexc.cpp	2006/04/26 08:21:14	1.25
+++ src/os2/utilsexc.cpp	2007/04/06 07:36:06
@@ -156,7 +156,7 @@
                       ,zArgs
                       ,zEnvs
                       ,&vResultCodes
-                      ,(PSZ)rCommand.c_str()
+                      ,rCommand.c_str()
                      );
     if (rc != NO_ERROR)
     {
Index: src/os2/window.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/os2/window.cpp,v
retrieving revision 1.182
diff -u -r1.182 window.cpp
--- src/os2/window.cpp	2007/03/25 22:31:27	1.182
+++ src/os2/window.cpp	2007/04/06 07:36:17
@@ -512,7 +512,7 @@

 void wxWindowOS2::SetLabel( const wxString& label )
 {
-    ::WinSetWindowText(GetHwnd(), (PSZ)label.c_str());
+    ::WinSetWindowText(GetHwnd(), label.c_str());
 } // end of wxWindowOS2::SetLabel

 wxString wxWindowOS2::GetLabel() const
@@ -1681,7 +1681,7 @@
     l = rString.length();
     if (l > 0L)
     {
-        pStr = (PCH)rString.c_str();
+        pStr = (char*)(const char*)rString.c_str();

         //
         // In world coordinates.
@@ -2971,7 +2971,7 @@
         sClassName += wxT("NR");
     }
     m_hWnd = (WXHWND)::WinCreateWindow( (HWND)OS2GetParent()
-                                       ,(PSZ)sClassName.c_str()
+                                       ,sClassName.c_str()
                                        ,(PSZ)(zTitle ? zTitle : wxEmptyString)
                                        ,(ULONG)dwStyle
                                        ,(LONG)0L

 	  	 
