diff -uNr wxWidgets-orig/src/common/datavcmn.cpp wxWidgets-mine/src/common/datavcmn.cpp
--- wxWidgets-orig/src/common/datavcmn.cpp	2016-02-28 22:33:38.000000000 +0100
+++ wxWidgets-mine/src/common/datavcmn.cpp	2016-03-17 20:22:10.000000000 +0100
@@ -1139,6 +1139,11 @@
     if (m_model)
     {
         m_model->IncRef();
+
+#if defined(__WXOSX_COCOA__)
+            // do not indent first column if there are no expandable items
+            if(model->IsListModel() || model->IsVirtualListModel()) SetIndent(0);
+#endif
     }
 
     return true;
