Index: include/wx/txtstrm.h
===================================================================
--- include/wx/txtstrm.h	(revision 60631)
+++ include/wx/txtstrm.h	(working copy)
@@ -45,6 +45,10 @@
 #endif
     ~wxTextInputStream();
 
+#if wxABI_VERSION >= 20811
+    wxInputStream* GetInputStream() { return &m_input; }
+#endif
+
     wxUint32 Read32(int base = 10); // base may be between 2 and 36, inclusive, or the special 0 (= C format)
     wxUint16 Read16(int base = 10);
     wxUint8  Read8(int base = 10);
@@ -116,6 +120,10 @@
 #endif
     virtual ~wxTextOutputStream();
 
+#if wxABI_VERSION >= 20811
+    wxOutputStream* GetOutputStream() { return &m_output; }
+#endif
+
     void SetMode( wxEOL mode = wxEOL_NATIVE );
     wxEOL GetMode() { return m_mode; }
 
Index: samples/console/console.cpp
===================================================================
--- samples/console/console.cpp	(revision 60631)
+++ samples/console/console.cpp	(working copy)
@@ -3700,13 +3700,13 @@
 
 static void TestTextInputStream()
 {
-    wxPuts(_T("\n*** wxTextInputStream test ***"));
+    wxPuts(wxT("\n*** wxTextInputStream test ***"));
 
-    wxString filename = _T("testdata.fc");
+    wxString filename = wxT("testdata.fc");
     wxFileInputStream fsIn(filename);
-    if ( !fsIn.Ok() )
+    if ( !fsIn.IsOk() )
     {
-        wxPuts(_T("ERROR: couldn't open file."));
+        wxPuts(wxT("ERROR: couldn't open file."));
     }
     else
     {
@@ -3719,10 +3719,10 @@
 
             // line could be non empty if the last line of the file isn't
             // terminated with EOL
-            if ( fsIn.Eof() && s.empty() )
+            if ( tis.GetInputStream()->Eof() && s.empty() )
                 break;
 
-            wxPrintf(_T("Line %d: %s\n"), line++, s.c_str());
+            wxPrintf(wxT("Line %d: %s\n"), line++, s.c_str());
         }
     }
 }
