Index: samples/mfc/mfctest.cpp
===================================================================
--- samples/mfc/mfctest.cpp	(revision 53010)
+++ samples/mfc/mfctest.cpp	(working copy)
@@ -39,7 +39,7 @@
 //     nmake -f makefile.vc BUILD=debug SHARED=0 DEBUG_RUNTIME_LIBS=0 RUNTIME_LIBS=static all
 //
 //     Unless the run-time library settings match for wxWidgets and MFC, you
-//     will get link errors for symbols such as __mbctype, __argc, and __argv 
+//     will get link errors for symbols such as __mbctype, __argc, and __argv
 //
 // (3) If you see bogus memory leaks within the MSVC IDE on exit, in this
 //     sample or in your own project, you must be using __WXDEBUG__ +
@@ -49,6 +49,10 @@
 //     first: mfc[version][u]d.lib
 //     -  [version] -> 42 or 70 or 80 etc
 //     -  u if using Unicode
+//
+// (4) Unicode builds may produce the linker error "unresolved external symbol _WinMain@16".
+//     MFC requires you to manually add the Unicode entry point to the linker settings,
+//     Entry point symbol -> wWinMainCRTStartup
 
 #include "stdafx.h"
 

 	  	 
