Index: include/wx/generic/splitter.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/generic/splitter.h,v
retrieving revision 1.48
diff -u -3 -r1.48 splitter.h
--- include/wx/generic/splitter.h	2004/09/26 14:04:32	1.48
+++ include/wx/generic/splitter.h	2004/10/13 10:23:51
@@ -155,6 +155,12 @@
     // Gets the sash position
     int GetSashPosition() const { return m_sashPosition; }
 
+    // Set the sash gravity
+    void SetSashGravity(double gravity);
+    
+    // Gets the sash gravity
+    double GetSashGravity() const { return m_sashGravity; }
+    
     // If this is zero, we can remove panes by dragging the sash.
     void SetMinimumPaneSize(int min);
     int GetMinimumPaneSize() const { return m_minimumPaneSize; }
@@ -271,6 +277,8 @@
     int         m_oldX;
     int         m_oldY;
     int         m_sashPosition; // Number of pixels from left or top
+    double      m_sashGravity;
+    wxSize      m_lastSize;
     int         m_requestedSashPosition;
     int         m_sashPositionCurrent; // while dragging
     int         m_firstX;
Index: src/generic/splitter.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/generic/splitter.cpp,v
retrieving revision 1.101
diff -u -3 -r1.101 splitter.cpp
--- src/generic/splitter.cpp	2004/09/16 20:15:17	1.101
+++ src/generic/splitter.cpp	2004/10/13 10:23:53
@@ -113,6 +113,8 @@
     m_firstX = 0;
     m_firstY = 0;
     m_sashPosition = m_requestedSashPosition = 0;
+    m_sashGravity = 0.0;
+    m_lastSize = wxSize(0,0);
     m_checkRequestedSashPosition = false;
     m_minimumPaneSize = 0;
     m_sashCursorWE = wxCursor(wxCURSOR_SIZEWE);
@@ -404,6 +406,8 @@
 
     if ( iconized )
     {
+	m_lastSize = wxSize(0,0);
+	
         event.Skip();
 
         return;
@@ -415,11 +419,33 @@
         GetClientSize(&w, &h);
 
         int size = m_splitMode == wxSPLIT_VERTICAL ? w : h;
+	
+	int old_size = m_splitMode == wxSPLIT_VERTICAL ? m_lastSize.x : m_lastSize.y;
+	if( old_size != 0 ) {
+	    int delta = (int) ( (size - old_size)*m_sashGravity );
+	    if( delta != 0 ) {
+		int newPosition = m_sashPosition + delta;
+		if( newPosition < m_minimumPaneSize )
+		    newPosition = m_minimumPaneSize;
+		SetSashPositionAndNotify(newPosition);
+	    }
+	}
+	
         if ( m_sashPosition >= size - 5 )
             SetSashPositionAndNotify(wxMax(10, size - 40));
+	m_lastSize = wxSize(w,h);
     }
 
     SizeWindows();
+}
+
+void wxSplitterWindow::SetSashGravity(double gravity)
+{
+    m_sashGravity = gravity;
+    if( m_sashGravity < 0.0 )
+	m_sashGravity = 0.0;
+    if( m_sashGravity > 1.0 )
+	m_sashGravity = 1.0;
 }
 
 bool wxSplitterWindow::SashHitTest(int x, int y, int tolerance)
Index: samples/splitter/splitter.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/samples/splitter/splitter.cpp,v
retrieving revision 1.21
diff -u -3 -r1.21 splitter.cpp
--- samples/splitter/splitter.cpp	2004/10/06 20:54:37	1.21
+++ samples/splitter/splitter.cpp	2004/10/13 10:23:54
@@ -52,7 +52,8 @@
     SPLIT_UNSPLIT,
     SPLIT_LIVE,
     SPLIT_SETPOSITION,
-    SPLIT_SETMINSIZE
+    SPLIT_SETMINSIZE,
+    SPLIT_SETGRAVITY
 };
 
 // ----------------------------------------------------------------------------
@@ -82,6 +83,7 @@
     void ToggleLive(wxCommandEvent& event);
     void SetPosition(wxCommandEvent& event);
     void SetMinSize(wxCommandEvent& event);
+    void SetGravity(wxCommandEvent& event);
 
     void Quit(wxCommandEvent& event);
 
@@ -162,6 +164,7 @@
     EVT_MENU(SPLIT_LIVE, MyFrame::ToggleLive)
     EVT_MENU(SPLIT_SETPOSITION, MyFrame::SetPosition)
     EVT_MENU(SPLIT_SETMINSIZE, MyFrame::SetMinSize)
+    EVT_MENU(SPLIT_SETGRAVITY, MyFrame::SetGravity)
 
     EVT_MENU(SPLIT_QUIT, MyFrame::Quit)
 
@@ -202,6 +205,9 @@
     splitMenu->Append(SPLIT_SETMINSIZE,
                       _T("Set &min size\tCtrl-M"),
                       _T("Set minimum pane size"));
+    splitMenu->Append(SPLIT_SETGRAVITY,
+                      _T("Set &gravity\tCtrl-G"),
+                      _T("Set gravity of sash"));
     splitMenu->AppendSeparator();
 
     splitMenu->Append(SPLIT_QUIT, _T("E&xit\tAlt-X"), _T("Exit"));
@@ -213,6 +219,8 @@
 
     menuBar->Check(SPLIT_LIVE, true);
     m_splitter = new MySplitterWindow(this);
+    
+    m_splitter->SetSashGravity(1.0);
 
 #if 1
     m_left = new MyCanvas(m_splitter, true);
@@ -327,6 +335,21 @@
     m_splitter->SetMinimumPaneSize(minsize);
 #if wxUSE_STATUSBAR
     str.Printf( wxT("Min pane size = %d"), minsize);
+    SetStatusText(str, 1);
+#endif // wxUSE_STATUSBAR
+}
+void MyFrame::SetGravity(wxCommandEvent& WXUNUSED(event) )
+{
+    wxString str;
+    str.Printf( wxT("%g"), m_splitter->GetSashGravity());
+    str = wxGetTextFromUser(_T("Enter sash gravity (0,1):"), _T(""), str, this);
+    if ( str.empty() )
+        return;
+
+    double gravity = wxStrtod( str, (wxChar**)NULL);
+    m_splitter->SetSashGravity(gravity);
+#if wxUSE_STATUSBAR
+    str.Printf( wxT("Gravity = %g"), gravity);
     SetStatusText(str, 1);
 #endif // wxUSE_STATUSBAR
 }
Index: docs/latex/wx/splitter.tex
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/docs/latex/wx/splitter.tex,v
retrieving revision 1.19
diff -u -3 -r1.19 splitter.tex
--- docs/latex/wx/splitter.tex	2004/09/22 10:59:54	1.19
+++ docs/latex/wx/splitter.tex	2004/10/13 10:23:55
@@ -131,6 +131,16 @@
 
 \helpref{wxSplitterWindow::SetMinimumPaneSize}{wxsplitterwindowsetminimumpanesize}
 
+\membersection{wxSplitterWindow::GetSashGravity}\label{wxsplitterwindowgetsashgravity}
+
+\func{double}{GetSashGravity}{\void}
+
+Returns the current sash gravity.
+
+\wxheading{See also}
+
+\helpref{wxSplitterWindow::SetSashGravity}{wxsplitterwindowsetsashgravity}
+
 \membersection{wxSplitterWindow::GetSashPosition}\label{wxsplitterwindowgetsashposition}
 
 \func{int}{GetSashPosition}{\void}
@@ -268,6 +278,37 @@
 \helpref{wxSplitterWindow::Unsplit}{wxsplitterwindowunsplit}\\
 \helpref{wxSplitterWindow::SplitVertically}{wxsplitterwindowsplitvertically}\\
 \helpref{wxSplitterWindow::SplitHorizontally}{wxsplitterwindowsplithorizontally}
+
+\membersection{wxSplitterWindow::SetSashGravity}\label{wxsplitterwindowsetsashgravity}
+
+\func{void}{SetSashGravity}{\param{double }{gravity}}
+
+Sets the sash gravity. 
+
+\wxheading{Parameters}
+
+\docparam{gravity}{The sash gravity. Value between 0.0 and 1.0.}
+
+
+\wxheading{Remarks}
+Gravity is real factor which controls position of sash while resizing wxSplitterWindow.
+Gravity tells wxSplitterWindow how much will left/top window grow while resizing.
+
+Example values:
+\begin{itemize}\itemsep=0pt
+\item{ 0.0  - only the bottom/right window is automaticaly resized}
+\item{ 0.5  - both windows grow by equal size}
+\item{ 1.0  - only left/top window grows}
+\end{itemize}
+
+Gravity should be real value betwwen 0.0 and 1.0.
+
+Default value of sash gravity is 0.0. That value is compatible with previous 
+(before gravity was introduced) behaviour of wxSplitterWindow.
+
+\wxheading{See also}
+
+\helpref{wxSplitterWindow::GetSashGravity}{wxsplitterwindowgetsashgravity}
 
 \membersection{wxSplitterWindow::SetSashPosition}\label{wxsplitterwindowsetsashposition}
 

 	  	 
