diff --git a/src/propgrid/advprops.cpp b/src/propgrid/advprops.cpp
index 7adc312..da110b8 100644
--- a/src/propgrid/advprops.cpp
+++ b/src/propgrid/advprops.cpp
@@ -1208,7 +1208,7 @@ wxSize wxSystemColourProperty::OnMeasureImage( int ) const
 
 int wxSystemColourProperty::GetCustomColourIndex() const
 {
-    return m_choices.GetCount() - 1;
+    return m_choices.GetCount() == 0 ? 0 : m_choices.GetCount() - 1;
 }
 
 
