--- a/include/wx/propgrid/propgrid.h
+++ b/include/wx/propgrid/propgrid.h
@@ -2092,6 +2092,7 @@ protected:
 
     void OnTLPClose( wxCloseEvent& event );
 
+    void OnSelFocus( wxCommandEvent& evt );
 protected:
 
     bool AddToSelectionFromInputEvent( wxPGProperty* prop,
--- a/src/propgrid/propgrid.cpp
+++ b/src/propgrid/propgrid.cpp
@@ -250,6 +250,8 @@ void wxPropertyGridInitGlobalsIfNeeded()
 
 IMPLEMENT_DYNAMIC_CLASS(wxPropertyGrid, wxControl)
 
+wxDEFINE_EVENT( wxEVT_PG_SEL_FOCUS, wxCommandEvent );
+
 BEGIN_EVENT_TABLE(wxPropertyGrid, wxControl)
   EVT_IDLE(wxPropertyGrid::OnIdle)
   EVT_PAINT(wxPropertyGrid::OnPaint)
@@ -268,8 +270,16 @@ BEGIN_EVENT_TABLE(wxPropertyGrid, wxControl)
   EVT_RIGHT_UP(wxPropertyGrid::OnMouseRightClick)
   EVT_LEFT_DCLICK(wxPropertyGrid::OnMouseDoubleClick)
   EVT_KEY_DOWN(wxPropertyGrid::OnKey)
+  EVT_COMMAND(wxID_ANY, wxEVT_PG_SEL_FOCUS, wxPropertyGrid::OnSelFocus)
 END_EVENT_TABLE()
 
+void wxPropertyGrid::OnSelFocus( wxCommandEvent& evt )
+{
+    // Focus editable control associated with selected property.
+    wxWindow *control = static_cast<wxWindow*>(evt.GetClientData());
+    control->SetFocus();
+}
+
 // -----------------------------------------------------------------------
 
 wxPropertyGrid::wxPropertyGrid()
@@ -4188,9 +4198,13 @@ bool wxPropertyGrid::DoSelectProperty( wxPGProperty* p, unsigned int flags )
                     // Focus and select all (wxTextCtrl, wxComboBox etc)
                     if ( flags & wxPG_SEL_FOCUS )
                     {
-                        primaryCtrl->SetFocus();
-
                         p->GetEditorClass()->OnFocus(p, primaryCtrl);
+                        // New editable control will be focused asynchronously with some delay
+                        // because for GTK+ widgets immediate focusing just after creation
+                        // can lead to unexpected side effects.
+                        wxCommandEvent *evt = new wxCommandEvent(wxEVT_PG_SEL_FOCUS, GetId());
+                        wxQueueEvent(this, evt);
+                        evt->SetClientData(primaryCtrl);
                     }
                     else
                     {
