Index: src/msw/utilsexc.cpp
===================================================================
--- src/msw/utilsexc.cpp	(revision 54419)
+++ src/msw/utilsexc.cpp	(working copy)
@@ -1018,9 +1018,16 @@
     wxString command;
     command.reserve(1024);
 
+    wxString temp;
     for ( ;; )
     {
-        command += *argv++;
+        temp = *argv++;
+        // Escape any quotes already in the string
+        temp.Replace("\"", "\\\"", true /* replace all */);
+
+        if(temp.Find(' ') == wxNOT_FOUND)
+            command += temp;
+        else
+            command += '\"' + temp + '\"';
+
         if ( !*argv )
             break;

