Index: notebook.cpp
===================================================================
--- notebook.cpp	(revision 63004)
+++ notebook.cpp	(working copy)
@@ -492,6 +492,8 @@
 {
     wxCHECK_MSG( IS_VALID_PAGE(nPage), wxNOT_FOUND, wxT("notebook page out of range") );
 
+    nOldPage = m_nSelection;
+
     if ( m_nSelection == wxNOT_FOUND || nPage != (size_t)m_nSelection )
     {
         TabCtrl_SetCurSel(GetHwnd(), nPage);
@@ -499,7 +501,7 @@
         UpdateSelection(nPage);
     }
 
-    return m_nSelection;
+    return nOldPage;
 }
 
 bool wxNotebook::SetPageText(size_t nPage, const wxString& strText)
