Index: src/mac/carbon/glcanvas.cpp
===================================================================
--- src/mac/carbon/glcanvas.cpp	(revision 47612)
+++ src/mac/carbon/glcanvas.cpp	(working copy)
@@ -63,10 +63,6 @@
                                     other ? other->m_aglContext : NULL);
     if ( !m_aglContext )
         wxLogAGLError("aglCreateContext");
-
-    return;
-    if ( !aglEnable(m_aglContext, AGL_BUFFER_RECT) )
-        wxLogAGLError("aglEnable(AGL_BUFFER_RECT)");
 }
 
 wxGLContext::~wxGLContext()
@@ -365,6 +361,9 @@
     if ( !aglSetInteger(context, AGL_BUFFER_RECT, parms) )
         wxLogAGLError("aglSetInteger(AGL_BUFFER_RECT)");
 
+    if ( !aglEnable(context, AGL_BUFFER_RECT) )
+        wxLogAGLError("aglEnable(AGL_BUFFER_RECT)");
+
     if ( !aglUpdateContext(context) )
         wxLogAGLError("aglUpdateContext");
 }

 	  	 
