Index: src/gtk/window.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/gtk/window.cpp,v
retrieving revision 1.441
diff -u -r1.441 window.cpp
--- src/gtk/window.cpp	2003/09/24 00:26:24	1.441
+++ src/gtk/window.cpp	2003/10/01 17:42:03
@@ -4284,9 +4284,7 @@
 
 bool wxWindowGTK::SetFont( const wxFont &font )
 {
-    wxCHECK_MSG( m_widget != NULL, FALSE, wxT("invalid window") );
-
-    if (!wxWindowBase::SetFont(font))
+    if (!wxWindowBase::SetFont(font) || !m_widget)
     {
         return FALSE;
     }

 	  	 
