diff -ru2 -xCVS -x.#* -x*.orig -x*.rej -x*~ -x*.obj -xsetup.h -x*.lib -x*.c -x*.lbc d:\Download\wxWindows-2.4.1\distrib\msw\tmake\wat.t d:\System\Apps\OpenWatcom\wxWindows-2.4.1\distrib\msw\tmake\wat.t
--- d:\Download\wxWindows-2.4.1\distrib\msw\tmake\wat.t	Wed Apr 02 09:22:08 2003
+++ d:\System\Apps\OpenWatcom\wxWindows-2.4.1\distrib\msw\tmake\wat.t	Tue Jul 15 21:41:38 2003
@@ -229,50 +229,50 @@
 png:   .SYMBOLIC
     cd $(WXDIR)\src\png
-    wmake -f makefile.wat all
+    wmake $(__MAKEOPTS__) -f makefile.wat all
     cd $(WXDIR)\src\msw
 
 clean_png:   .SYMBOLIC
     cd $(WXDIR)\src\png
-    wmake -f makefile.wat clean
+    wmake $(__MAKEOPTS__) -f makefile.wat clean
     cd $(WXDIR)\src\msw
 
 zlib:   .SYMBOLIC
     cd $(WXDIR)\src\zlib
-    wmake -f makefile.wat all
+    wmake $(__MAKEOPTS__) -f makefile.wat all
     cd $(WXDIR)\src\msw
 
 clean_zlib:   .SYMBOLIC
     cd $(WXDIR)\src\zlib
-    wmake -f makefile.wat clean
+    wmake $(__MAKEOPTS__) -f makefile.wat clean
     cd $(WXDIR)\src\msw
 
 jpeg:    .SYMBOLIC
     cd $(WXDIR)\src\jpeg
-    wmake -f makefile.wat all
+    wmake $(__MAKEOPTS__) -f makefile.wat all
     cd $(WXDIR)\src\msw
 
 clean_jpeg:   .SYMBOLIC
     cd $(WXDIR)\src\jpeg
-    wmake -f makefile.wat clean
+    wmake $(__MAKEOPTS__) -f makefile.wat clean
     cd $(WXDIR)\src\msw
 
 tiff:    .SYMBOLIC
     cd $(WXDIR)\src\tiff
-    wmake -f makefile.wat all
+    wmake $(__MAKEOPTS__) -f makefile.wat all
     cd $(WXDIR)\src\msw
 
 clean_tiff:   .SYMBOLIC
     cd $(WXDIR)\src\tiff
-    wmake -f makefile.wat clean
+    wmake $(__MAKEOPTS__) -f makefile.wat clean
     cd $(WXDIR)\src\msw
 
 regex:    .SYMBOLIC
     cd $(WXDIR)\src\regex
-    wmake -f makefile.wat all
+    wmake $(__MAKEOPTS__) -f makefile.wat all
     cd $(WXDIR)\src\msw
 
 clean_regex:   .SYMBOLIC
     cd $(WXDIR)\src\regex
-    wmake -f makefile.wat clean
+    wmake $(__MAKEOPTS__) -f makefile.wat clean
     cd $(WXDIR)\src\msw
 
diff -ru2 -xCVS -x.#* -x*.orig -x*.rej -x*~ -x*.obj -xsetup.h -x*.lib -x*.c -x*.lbc d:\Download\wxWindows-2.4.1\distrib\msw\tmake\watbase.t d:\System\Apps\OpenWatcom\wxWindows-2.4.1\distrib\msw\tmake\watbase.t
--- d:\Download\wxWindows-2.4.1\distrib\msw\tmake\watbase.t	Thu Feb 27 11:32:16 2003
+++ d:\System\Apps\OpenWatcom\wxWindows-2.4.1\distrib\msw\tmake\watbase.t	Tue Jul 15 21:42:40 2003
@@ -159,20 +159,20 @@
 zlib:   .SYMBOLIC
     cd $(WXDIR)\src\zlib
-    wmake -f makefile.wat all
+    wmake $(__MAKEOPTS__) -f makefile.wat all
     cd $(WXDIR)\src\msw
 
 clean_zlib:   .SYMBOLIC
     cd $(WXDIR)\src\zlib
-    wmake -f makefile.wat clean
+    wmake $(__MAKEOPTS__) -f makefile.wat clean
     cd $(WXDIR)\src\msw
 
 regex:    .SYMBOLIC
     cd $(WXDIR)\src\regex
-    wmake -f makefile.wat all
+    wmake $(__MAKEOPTS__) -f makefile.wat all
     cd $(WXDIR)\src\msw
 
 clean_regex:   .SYMBOLIC
     cd $(WXDIR)\src\regex
-    wmake -f makefile.wat clean
+    wmake $(__MAKEOPTS__) -f makefile.wat clean
     cd $(WXDIR)\src\msw
 
diff -ru2 -xCVS -x.#* -x*.orig -x*.rej -x*~ -x*.obj -xsetup.h -x*.lib -x*.c -x*.lbc d:\Download\wxWindows-2.4.1\distrib\msw\tmake\watmgl.t d:\System\Apps\OpenWatcom\wxWindows-2.4.1\distrib\msw\tmake\watmgl.t
--- d:\Download\wxWindows-2.4.1\distrib\msw\tmake\watmgl.t	Thu Sep 05 00:07:20 2002
+++ d:\System\Apps\OpenWatcom\wxWindows-2.4.1\distrib\msw\tmake\watmgl.t	Tue Jul 15 21:44:00 2003
@@ -263,50 +263,50 @@
 png:   .SYMBOLIC
     cd $(WXDIR)\src\png
-    wmake -f makefile.wat all
+    wmake $(__MAKEOPTS__) -f makefile.wat all
     cd $(WXDIR)\src\mgl
 
 clean_png:   .SYMBOLIC
     cd $(WXDIR)\src\png
-    wmake -f makefile.wat clean
+    wmake $(__MAKEOPTS__) -f makefile.wat clean
     cd $(WXDIR)\src\mgl
 
 zlib:   .SYMBOLIC
     cd $(WXDIR)\src\zlib
-    wmake -f makefile.wat all
+    wmake $(__MAKEOPTS__) -f makefile.wat all
     cd $(WXDIR)\src\mgl
 
 clean_zlib:   .SYMBOLIC
     cd $(WXDIR)\src\zlib
-    wmake -f makefile.wat clean
+    wmake $(__MAKEOPTS__) -f makefile.wat clean
     cd $(WXDIR)\src\mgl
 
 jpeg:    .SYMBOLIC
     cd $(WXDIR)\src\jpeg
-    wmake -f makefile.wat all
+    wmake $(__MAKEOPTS__) -f makefile.wat all
     cd $(WXDIR)\src\mgl
 
 clean_jpeg:   .SYMBOLIC
     cd $(WXDIR)\src\jpeg
-    wmake -f makefile.wat clean
+    wmake $(__MAKEOPTS__) -f makefile.wat clean
     cd $(WXDIR)\src\mgl
 
 tiff:    .SYMBOLIC
     cd $(WXDIR)\src\tiff
-    wmake -f makefile.wat all
+    wmake $(__MAKEOPTS__) -f makefile.wat all
     cd $(WXDIR)\src\mgl
 
 clean_tiff:   .SYMBOLIC
     cd $(WXDIR)\src\tiff
-    wmake -f makefile.wat clean
+    wmake $(__MAKEOPTS__) -f makefile.wat clean
     cd $(WXDIR)\src\mgl
 
 regex:    .SYMBOLIC
     cd $(WXDIR)\src\regex
-    wmake -f makefile.wat all
+    wmake $(__MAKEOPTS__) -f makefile.wat all
     cd $(WXDIR)\src\mgl
 
 clean_regex:   .SYMBOLIC
     cd $(WXDIR)\src\regex
-    wmake -f makefile.wat clean
+    wmake $(__MAKEOPTS__) -f makefile.wat clean
     cd $(WXDIR)\src\mgl
 
Only in d:\System\Apps\OpenWatcom\wxWindows-2.4.1\lib: msw
Only in d:\System\Apps\OpenWatcom\wxWindows-2.4.1: ReleaseWatcom
diff -ru2 -xCVS -x.#* -x*.orig -x*.rej -x*~ -x*.obj -xsetup.h -x*.lib -x*.c -x*.lbc d:\Download\wxWindows-2.4.1\src\makewat.env d:\System\Apps\OpenWatcom\wxWindows-2.4.1\src\makewat.env
--- d:\Download\wxWindows-2.4.1\src\makewat.env	Fri Feb 28 12:29:16 2003
+++ d:\System\Apps\OpenWatcom\wxWindows-2.4.1\src\makewat.env	Tue Jul 15 21:32:44 2003
@@ -19,4 +19,6 @@
 wxUSE_GUI=1
 !endif
+# CF: Recursive calls to wmake will get $(__MAKEOPTS__) passed.
+__MAKEOPTS__ += wxUSE_GUI=$(wxUSE_GUI)
 
 # set FINAL=1 to disable debug and enable optimizations
@@ -24,4 +26,6 @@
 FINAL=0
 !endif
+# CF: Recursive calls to wmake will get $(__MAKEOPTS__) passed.
+__MAKEOPTS__ += FINAL=$(FINAL)
 
 # set WATCOM_SUFFIX to nothing if you use only Watcom compiler, set it to
@@ -35,8 +39,12 @@
 
 # optimize for space/time: /os or /ot
+
+# CF: I'm entirely replacing the old flags with new, identical ones,
+# favouring time optimizations (faster code) over space (smaller code) now.
+# Note that this is *not* the reason for the resulting very large wx libs files.
 !ifeq FINAL 1
-OPTFLAGS=/oails /5r
+OPTFLAGS= -5r -zp8 -otexan
 !else
-OPTFLAGS=/od
+OPTFLAGS= -5r -od -d2
 !endif
 
@@ -154,10 +162,10 @@
 # NB: /bm switch is needed for thread-safe runtime, if you don't use
 #     wxThread you may remove it
-CPPFLAGS = /dWIN32 /bm /fo=$(OUTPUTDIR)\ /fr /zq $(IFLAGS) $(OPTFLAGS) $(WINVERFLAGS) $(EXTRACPPFLAGS)
+CPPFLAGS = /dWIN32 /bm /fo=$(OUTPUTDIR)\ /fr -zq $(IFLAGS) $(OPTFLAGS) $(WINVERFLAGS) $(EXTRACPPFLAGS)
 !ifeq wxUSE_GUI 0
 CPPFLAGS += /dwxUSE_GUI=0
 !endif
 
-# zm and zv as well as the liker options below are used to make the resulting
+# zm and zv as well as the linker options below are used to make the resulting
 # .exe smaller
 CFLAGS = $(CPPFLAGS) $(DEBUGCFLAGS) $(EXTRACFLAGS) /zm
diff -ru2 -xCVS -x.#* -x*.orig -x*.rej -x*~ -x*.obj -xsetup.h -x*.lib -x*.c -x*.lbc d:\Download\wxWindows-2.4.1\src\msw\makefile.wat d:\System\Apps\OpenWatcom\wxWindows-2.4.1\src\msw\makefile.wat
--- d:\Download\wxWindows-2.4.1\src\msw\makefile.wat	Mon Jun 02 14:26:14 2003
+++ d:\System\Apps\OpenWatcom\wxWindows-2.4.1\src\msw\makefile.wat	Tue Jul 15 21:35:08 2003
@@ -1291,50 +1291,50 @@
 png:   .SYMBOLIC
     cd $(WXDIR)\src\png
-    wmake -f makefile.wat all
+    wmake $(__MAKEOPTS__) -f makefile.wat all
     cd $(WXDIR)\src\msw
 
 clean_png:   .SYMBOLIC
     cd $(WXDIR)\src\png
-    wmake -f makefile.wat clean
+    wmake $(__MAKEOPTS__) -f makefile.wat clean
     cd $(WXDIR)\src\msw
 
 zlib:   .SYMBOLIC
     cd $(WXDIR)\src\zlib
-    wmake -f makefile.wat all
+    wmake $(__MAKEOPTS__) -f makefile.wat all
     cd $(WXDIR)\src\msw
 
 clean_zlib:   .SYMBOLIC
     cd $(WXDIR)\src\zlib
-    wmake -f makefile.wat clean
+    wmake $(__MAKEOPTS__) -f makefile.wat clean
     cd $(WXDIR)\src\msw
 
 jpeg:    .SYMBOLIC
     cd $(WXDIR)\src\jpeg
-    wmake -f makefile.wat all
+    wmake $(__MAKEOPTS__) -f makefile.wat all
     cd $(WXDIR)\src\msw
 
 clean_jpeg:   .SYMBOLIC
     cd $(WXDIR)\src\jpeg
-    wmake -f makefile.wat clean
+    wmake $(__MAKEOPTS__) -f makefile.wat clean
     cd $(WXDIR)\src\msw
 
 tiff:    .SYMBOLIC
     cd $(WXDIR)\src\tiff
-    wmake -f makefile.wat all
+    wmake $(__MAKEOPTS__) -f makefile.wat all
     cd $(WXDIR)\src\msw
 
 clean_tiff:   .SYMBOLIC
     cd $(WXDIR)\src\tiff
-    wmake -f makefile.wat clean
+    wmake $(__MAKEOPTS__) -f makefile.wat clean
     cd $(WXDIR)\src\msw
 
 regex:    .SYMBOLIC
     cd $(WXDIR)\src\regex
-    wmake -f makefile.wat all
+    wmake $(__MAKEOPTS__) -f makefile.wat all
     cd $(WXDIR)\src\msw
 
 clean_regex:   .SYMBOLIC
     cd $(WXDIR)\src\regex
-    wmake -f makefile.wat clean
+    wmake $(__MAKEOPTS__) -f makefile.wat clean
     cd $(WXDIR)\src\msw
 
Only in d:\System\Apps\OpenWatcom\wxWindows-2.4.1\src\zlib: watcom

 	  	 
