Index: src/gtk/slider.cpp
===================================================================
--- src/gtk/slider.cpp	(revision 49329)
+++ src/gtk/slider.cpp	(working copy)
@@ -277,7 +277,7 @@
 wxSlider::wxSlider()
 {
     m_pos = 0;
-    m_scrollEventType = 0;
+    m_scrollEventType = GTK_SCROLL_NONE;
     m_needThumbRelease = false;
     m_blockScrollEvent = false;
 }
@@ -293,6 +293,9 @@
                       const wxValidator& validator,
                       const wxString& name)
 {
+    m_pos = value;
+    m_scrollEventType = GTK_SCROLL_NONE;
+
     if (!PreCreation( parent, pos, size ) ||
         !CreateBase( parent, id, pos, size, style, validator, name ))
     {
@@ -323,8 +326,12 @@
     g_signal_handler_block(m_widget, handler_id);
 
     SetRange( minValue, maxValue );
-    SetValue( value );
 
+    // SetValue() is not used, to avoid comparing value with GetValue() 
+    m_blockScrollEvent = true;
+    gtk_range_set_value(GTK_RANGE (m_widget), value);
+    m_blockScrollEvent = false;
+
     m_parent->DoAddChild( this );
 
     PostCreation(size);

 	  	 
