diff -u -r1.127 notebook.cpp
--- notebook.cpp	2005/01/07 17:33:33	1.127
+++ notebook.cpp	2005/01/22 15:49:25
@@ -251,12 +253,43 @@
             style &= ~(wxNB_BOTTOM | wxNB_LEFT | wxNB_RIGHT);
         }
     }
+    LPCTSTR className = WC_TABCONTROL;
+    // TODO: what condition should it be ??? maybe some flag
+    //if( (style & wxNO_FULL_REPAINT_ON_RESIZE ) == wxNO_FULL_REPAINT_ON_RESIZE ) 
+    if( 1 )
+    {
+        // SysTabCtl32 class has natively CS_HREDRAW and CS_VREDRAW enabled
+        // and it causes horrible flicker when resizing notebook, so ... disable redraw.
+        static bool ourClassRegistered = false;
+	static LPCTSTR WX_NOTEBOOK_NOREDRAW = wxT("wxNotebookWrapperNoRedraw");
+	static WNDCLASSEX ourNotebookClass;
+
+        style |= wxNO_FULL_REPAINT_ON_RESIZE;
 
+        if( !ourClassRegistered ) 
+        {
+            ourNotebookClass.cbSize = sizeof(WNDCLASSEX);
+            
+            if( ::GetClassInfoEx(GetModuleHandle(NULL), WC_TABCONTROL, &ourNotebookClass) )
+            {
+                ourNotebookClass.lpszClassName = WX_NOTEBOOK_NOREDRAW;
+                ourNotebookClass.style &= ~(CS_HREDRAW | CS_VREDRAW);
+                ourNotebookClass.cbSize = sizeof(WNDCLASSEX);
+                
+                if( ! ::RegisterClassEx(&ourNotebookClass) ) 
+                    wxLogLastError(wxT("RegisterClassEx"));
+                else
+                    ourClassRegistered = true;
+            }
+        }
+        if( ourClassRegistered ) 
+            className = ourNotebookClass.lpszClassName;
+    }
     if ( !CreateControl(parent, id, pos, size, style | wxTAB_TRAVERSAL,
                         wxDefaultValidator, name) )
         return false;
 
-    if ( !MSWCreateControl(WC_TABCONTROL, wxEmptyString, pos, size) )
+    if ( !MSWCreateControl(className, wxEmptyString, pos, size) )
         return false;
 
     return true;
@@ -687,7 +720,8 @@
   RECT rc;
   rc.left = rc.top = 0;
   GetSize((int *)&rc.right, (int *)&rc.bottom);
-
+  RECT nrc;
+  nrc = rc;
   // there seems to be a bug in the implementation of TabCtrl_AdjustRect(): it
   // returns completely false values for multiline tab controls after the tabs
   // are added but before getting the first WM_SIZE (off by ~50 pixels, see
@@ -721,6 +755,21 @@
     pPage->SetSize(rc.left, rc.top, width, height);
   }
 
+  if( (GetWindowStyle() & wxNO_FULL_REPAINT_ON_RESIZE) == wxNO_FULL_REPAINT_ON_RESIZE)
+  {
+    // invalidate area not covered by pages
+    int nwidth = nrc.right - nrc.left,
+	nheight = nrc.bottom - nrc.top;
+    wxRect r;
+    r = wxRect(0,0,nwidth, rc.top);
+    Refresh(false, &r);
+    r = wxRect(0,rc.top, rc.left, height);
+    Refresh(false, &r);
+    r = wxRect(0,rc.bottom, nwidth, nheight-rc.bottom);
+    Refresh(false, &r);
+    r = wxRect(rc.right, rc.top, nwidth-rc.bottom, height);
+    Refresh(false, &r);
+  }
   event.Skip();
 }

 	  	 
