Index: src/mac/carbon/combobox.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/mac/carbon/combobox.cpp,v
retrieving revision 1.56
diff -b -u -2 -r1.56 combobox.cpp
--- src/mac/carbon/combobox.cpp	2005/02/18 04:43:45	1.56
+++ src/mac/carbon/combobox.cpp	2005/03/31 14:59:47
@@ -131,5 +131,17 @@
             m_cb->GetEventHandler()->ProcessEvent(event);
         }
+        wxPostEvent(GetParent(), event);
     }
+
+    void OnKeyDown( wxKeyEvent& event )
+    {
+        wxPostEvent(GetParent(), event);
+    }
+
+    void OnText( wxCommandEvent& event )
+    {
+        wxPostEvent(GetParent(), event);
+    }
+
 private:
     wxComboBox *m_cb;
@@ -141,4 +153,6 @@
     EVT_CHAR( wxComboBoxText::OnChar)
     EVT_KEY_UP( wxComboBoxText::OnKeyUp)
+    EVT_KEY_DOWN( wxComboBoxText::OnKeyDown)
+    EVT_TEXT( -1, wxComboBoxText::OnText)
 END_EVENT_TABLE()
 

 	  	 
