diff -uNr wxWidgets-orig/src/msw/textctrl.cpp wxWidgets-mine/src/msw/textctrl.cpp
--- wxWidgets-orig/src/msw/textctrl.cpp	2016-02-28 21:33:38.000000000 +0100
+++ wxWidgets-mine/src/msw/textctrl.cpp	2017-04-10 15:09:26.876655400 +0200
@@ -55,6 +55,8 @@
 #include "wx/msw/private.h"
 #include "wx/msw/winundef.h"
 
+#include "wx/msw/private/comptr.h"
+
 #include <string.h>
 #include <stdlib.h>
 
@@ -63,6 +65,7 @@
 #if wxUSE_RICHEDIT
     #include <richedit.h>
     #include <richole.h>
+    #include <tom.h>
     #include "wx/msw/ole/oleutils.h"
 #endif // wxUSE_RICHEDIT
 
@@ -96,6 +99,9 @@
 // defining it ourselves.
 DEFINE_GUID(wxIID_IRichEditOleCallback,
     0x00020d03, 0x0000, 0x0000, 0xc0, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x46);
+    
+DEFINE_GUID(wxIID_ITextDocument,
+    0x8CC497C0, 0xA1DF, 0x11CE, 0x80, 0x98, 0x00, 0xAA, 0x00, 0x47, 0xBE, 0x5D);    
 
 } // anonymous namespace
 #endif // wxUSE_OLE
@@ -2662,6 +2668,20 @@
 
     if ( GetRichVersion() >= 4 )
     {
+        LPRICHEDITOLE pRichEditOle = NULL;
+        wxCOMPtr<ITextDocument> pDoc;        
+        bool didsuspend = false;
+        
+    	if ( SendMessage(GetHwnd(), EM_GETOLEINTERFACE, 0, (LPARAM)&pRichEditOle) && pRichEditOle ) 
+    	{
+            HRESULT hr = pRichEditOle->QueryInterface(wxIID_ITextDocument, reinterpret_cast<void **> (&(pDoc)));    		
+            if( SUCCEEDED(hr) )
+            {
+                pDoc->Undo(tomSuspend, NULL); 
+                didsuspend = true;
+            }
+        }
+            	
         // Using WM_SETFONT is not enough with RichEdit 4.1: it does work but
         // for ASCII characters only and inserting a non-ASCII one into it
         // later reverts to the default font so use EM_SETCHARFORMAT to change
@@ -2669,6 +2689,12 @@
         wxTextAttr attr;
         attr.SetFont(font);
         SetStyle(-1, -1, attr);
+        
+        if ( didsuspend )
+            pDoc->Undo(tomResume, NULL);        		
+            	
+        if ( pRichEditOle )    	
+            pRichEditOle->Release();        	
     }
 
     return true;
