diff --git a/src/common/filefn.cpp b/src/common/filefn.cpp
index a0e5b27..6b24bfb 100644
--- a/src/common/filefn.cpp
+++ b/src/common/filefn.cpp
@@ -1042,7 +1042,8 @@ wxCopyFile (const wxString& file1, const wxString& file2, bool overwrite)
         return false;
     }
 
-    wxDoCopyFile(fileIn, fbuf, file2, overwrite);
+    if ( !wxDoCopyFile(fileIn, fbuf, file2, overwrite))
+        return false;
 
 #if defined(__WXMAC__)
     // copy the resource fork of the file too if it's present
