 src/msw/mdi.cpp | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/msw/mdi.cpp b/src/msw/mdi.cpp
index 2588499..2c0d9b2 100644
--- a/src/msw/mdi.cpp
+++ b/src/msw/mdi.cpp
@@ -911,7 +911,9 @@ wxMDIChildFrame::~wxMDIChildFrame()
     if ( !m_hWnd )
         return;
 
-    GetMDIParent()->RemoveMDIChild(this);
+    wxMDIParentFrame * const parent = GetMDIParent();
+
+    parent->RemoveMDIChild(this);
 
     // will be destroyed by DestroyChildren() but reset them before calling it
     // to avoid using dangling pointers if a callback comes in the meanwhile
@@ -926,6 +928,8 @@ wxMDIChildFrame::~wxMDIChildFrame()
 
     MDIRemoveWindowMenu(NULL, m_hMenu);
 
+    MDISetMenu(parent->GetClientWindow(), (HMENU)parent->MSWGetActiveMenu(), GetMDIWindowMenu(parent));
+
     MSWDestroyWindow();
 }
 
