Index: include/wx/docview.h
===================================================================
--- include/wx/docview.h	(revision 63748)
+++ include/wx/docview.h	(working copy)
@@ -478,6 +478,10 @@
 #endif // WXWIN_COMPATIBILITY_2_6
 
 protected:
+#if wxUSE_PRINTING_ARCHITECTURE
+    virtual wxPreviewFrame* CreatePreviewFrame(wxPrintPreviewBase* preview, wxWindow *parent, const wxString& title);
+#endif // wxUSE_PRINTING_ARCHITECTURE
+
     // hook the currently active view into event handlers chain here
     virtual bool TryBefore(wxEvent& event);
 
Index: interface/wx/docview.h
===================================================================
--- interface/wx/docview.h	(revision 63748)
+++ interface/wx/docview.h	(working copy)
@@ -517,6 +517,14 @@
     wxList& GetTemplates();
 
     /**
+        Creates an wxPreviewFrame instance. 
+        Override if you want to change the behaviour or appearance of the preview window.
+        
+        @since 2.9.1
+    */
+    virtual wxPreviewFrame* CreatePreviewFrame(wxPrintPreviewBase* preview, wxWindow* parent, const wxString& title);
+
+    /**
         Initializes data; currently just calls OnCreateFileHistory().
 
         Some data cannot always be initialized in the constructor because the
Index: src/common/docview.cpp
===================================================================
--- src/common/docview.cpp	(revision 63748)
+++ src/common/docview.cpp	(working copy)
@@ -1114,6 +1114,13 @@
 #endif // wxUSE_PRINTING_ARCHITECTURE
 }
 
+#if wxUSE_PRINTING_ARCHITECTURE
+wxPreviewFrame* wxDocManager::CreatePreviewFrame(wxPrintPreviewBase* preview, wxWindow *parent, const wxString& title)
+{
+    return new wxPreviewFrame(preview, parent, title);
+}
+#endif // wxUSE_PRINTING_ARCHITECTURE
+
 void wxDocManager::OnPreview(wxCommandEvent& WXUNUSED(event))
 {
 #if wxUSE_PRINTING_ARCHITECTURE
@@ -1135,9 +1142,7 @@
             return;
         }
 
-        wxPreviewFrame *
-            frame = new wxPreviewFrame(preview, wxTheApp->GetTopWindow(),
-                                       _("Print Preview"));
+        wxPreviewFrame* frame = CreatePreviewFrame(preview, wxTheApp->GetTopWindow(), _("Print Preview"));
         frame->Centre(wxBOTH);
         frame->Initialize();
         frame->Show(true);
