Index: samples/menu/menu.cpp
===================================================================
--- samples/menu/menu.cpp	(revision 63729)
+++ samples/menu/menu.cpp	(working copy)
@@ -1053,8 +1053,12 @@
     }
     else // normal case, shift not pressed
     {
+        wxMenu submenu;
+        submenu.Append(wxID_ANY, "sub menu item");
+
         menu.Append(Menu_Help_About, wxT("&About"));
-        menu.Append(Menu_Popup_Submenu, wxT("&Submenu"), CreateDummyMenu(NULL));
+        menu.AppendSubMenu(&submenu, "sub men");
+        //menu.Append(Menu_Popup_Submenu, wxT("&Submenu"), CreateDummyMenu(NULL));
         menu.Append(Menu_Popup_ToBeDeleted, wxT("To be &deleted"));
         menu.AppendCheckItem(Menu_Popup_ToBeChecked, wxT("To be &checked"));
         menu.Append(Menu_Popup_ToBeGreyed, wxT("To be &greyed"),
