diff -Naur wxGTK-2.4.2.orig/src/gtk/notebook.cpp wxGTK-2.4.2/src/gtk/notebook.cpp
--- wxGTK-2.4.2.orig/src/gtk/notebook.cpp	2003-09-21 13:31:57.000000000 +0200
+++ wxGTK-2.4.2/src/gtk/notebook.cpp	2004-03-05 19:07:18.000000000 +0100
@@ -643,7 +643,7 @@
     nb_page->m_text = text;
     if (nb_page->m_text.IsEmpty()) nb_page->m_text = wxT("");
   
-    nb_page->m_label = GTK_LABEL( gtk_label_new(nb_page->m_text.mbc_str()) );
+    nb_page->m_label = GTK_LABEL( gtk_label_new(wxGTK_CONV(nb_page->m_text)) );
     gtk_box_pack_end( GTK_BOX(nb_page->m_box), GTK_WIDGET(nb_page->m_label), FALSE, FALSE, m_padding );
   
     /* show the label */

 	  	 
