Index: src/generic/grid.cpp
===================================================================
--- src/generic/grid.cpp	(revision 69822)
+++ src/generic/grid.cpp	(working copy)
@@ -5199,7 +5199,7 @@
 void wxGrid::DrawCellHighlight( wxDC& dc, const wxGridCellAttr *attr )
 {
     // don't show highlight when the grid doesn't have focus
-    if ( !HasFocus() )
+    if ( !GetGridWindow()->HasFocus() )
         return;
 
     int row = m_currentCellCoords.GetRow();
