--- samples/toolbar/toolbar.cpp	2007-11-21 12:42:20.000000000 +0000
+++ samples/toolbar/toolbar.cpp	2008-08-14 15:23:04.000000000 +0100
@@ -396,9 +396,13 @@
 
     toolBar->SetToolBitmapSize(wxSize(w, h));
 
-    toolBar->AddTool(wxID_NEW, _T("New"),
+    wxToolBarToolBase* a_tool = toolBar->AddTool(wxID_NEW, _T("New"),
                      toolBarBitmaps[Tool_new], wxNullBitmap, wxITEM_NORMAL,
-                     _T("New file"), _T("This is help for new file tool"));
+                     _T("New file"), _T("This is help for new file tool")); 
+
+    a_tool->Enable(false);
+    a_tool->SetNormalBitmap(toolBarBitmaps[Tool_open]);
+
     toolBar->AddTool(wxID_OPEN, _T("Open"),
                      toolBarBitmaps[Tool_open], wxNullBitmap, wxITEM_NORMAL,
                      _T("Open file"), _T("This is help for open file tool"));
