diff --git a/src/msw/textentry.cpp b/src/msw/textentry.cpp
index d352b42..4a7ce62 100644
--- a/src/msw/textentry.cpp
+++ b/src/msw/textentry.cpp
@@ -40,7 +40,12 @@
 #endif
 
 #include "wx/msw/wrapwin.h"
-#include <Shlwapi.h>
+
+#ifdef __MINGW32__
+	#include <shlwapi.h>
+#else
+	#include <Shlwapi.h>
+#endif
 
 #define GetEditHwnd() ((HWND)(GetEditHWND()))
 
diff --git a/src/msw/utilsgui.cpp b/src/msw/utilsgui.cpp
index fec0847..6179eb9 100644
--- a/src/msw/utilsgui.cpp
+++ b/src/msw/utilsgui.cpp
@@ -32,7 +32,12 @@
 #include "wx/msw/private.h"     // includes <windows.h>
 
 #include "wx/msw/wrapwin.h"
-#include <Shlwapi.h>
+
+#ifdef __MINGW32__
+	#include <shlwapi.h>
+#else
+	#include <Shlwapi.h>
+#endif
 
 // ============================================================================
 // implementation
