Index: src/unix/gsocket.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/unix/gsocket.cpp,v
retrieving revision 1.28
diff -b -u -2 -r1.28 gsocket.cpp
--- src/unix/gsocket.cpp	2004/12/28 03:17:50	1.28
+++ src/unix/gsocket.cpp	2004/12/31 17:53:55
@@ -581,5 +581,4 @@
   if (err != GSOCK_NOERROR)
   {
-    GAddress_destroy(connection->m_peer);
     delete connection;
     m_error = err;

 	  	 
