--- wxWindows-2.4.0/src/common/regex.cpp.orig	2003-05-23 11:36:04.000000000 +0200
+++ wxWindows-2.4.0/src/common/regex.cpp	2003-05-23 11:56:41.000000000 +0200
@@ -81,6 +81,11 @@
             regfree(&m_RegEx);
 
             m_isCompiled = FALSE;
+            if ( m_Matches ) {
+              delete[] m_Matches;
+              m_Matches = NULL;
+            }
+            m_nMatches=0;
         }
     }
 
@@ -113,8 +118,6 @@
 wxRegExImpl::~wxRegExImpl()
 {
     Free();
-
-    delete [] m_Matches;
 }
 
 wxString wxRegExImpl::GetErrorMsg(int errorcode) const

 	  	 
