Index: src/aui/floatpane.cpp
===================================================================
--- src/aui/floatpane.cpp	(revision 67224)
+++ src/aui/floatpane.cpp	(working copy)
@@ -125,30 +125,37 @@
 
     if (pane.floating_size != wxDefaultSize)
     {
-        SetSize(pane.floating_size);
-    }
-    else
-    {
-        wxSize size = pane.best_size;
-        if (size == wxDefaultSize)
-            size = pane.min_size;
-        if (size == wxDefaultSize)
-            size = m_pane_window->GetSize();
-        if (m_owner_mgr && pane.HasGripper())
+        if (pane.IsFixed())
         {
-            if (pane.HasGripperTop())
-                size.y += m_owner_mgr->m_art->GetMetric(wxAUI_DOCKART_GRIPPER_SIZE);
-            else
-                size.x += m_owner_mgr->m_art->GetMetric(wxAUI_DOCKART_GRIPPER_SIZE);
+            SetWindowStyleFlag(GetWindowStyleFlag() & ~wxRESIZE_BORDER);
         }
 
-        SetClientSize(size);
+        SetSize(pane.floating_size);
+
+        return;
     }
 
+    wxSize size = pane.best_size;
+    if (size == wxDefaultSize)
+        size = pane.min_size;
+    if (size == wxDefaultSize)
+        size = m_pane_window->GetSize();
+    if (m_owner_mgr && pane.HasGripper())
+    {
+        if (pane.HasGripperTop())
+            size.y += m_owner_mgr->m_art->GetMetric(wxAUI_DOCKART_GRIPPER_SIZE);
+        else
+            size.x += m_owner_mgr->m_art->GetMetric(wxAUI_DOCKART_GRIPPER_SIZE);
+    }
+
+    // window style flag wxRESIZE_BORDER must be removed before SetClientSize()
+    // because resize border affects client size (at least in MSWin)
     if (pane.IsFixed())
     {
         SetWindowStyleFlag(GetWindowStyleFlag() & ~wxRESIZE_BORDER);
     }
+
+    SetClientSize(size);
 }
 
 wxAuiManager* wxAuiFloatingFrame::GetOwnerManager() const
