Index: src/unix/timerunx.cpp
===================================================================
--- src/unix/timerunx.cpp	(révision 56471)
+++ src/unix/timerunx.cpp	(copie de travail)
@@ -39,6 +39,7 @@
 
 #include "wx/listimpl.cpp"
 WX_DEFINE_LIST(wxTimerList)
+WX_DEFINE_LIST(wxImplTimerList)
 
 // trace mask for the debugging messages used here
 #define wxTrace_Timer wxT("timer")
@@ -139,6 +140,9 @@
     if ( m_timers.empty() )
       return;
 
+    // list of timers to notify
+    wxImplTimerList expirate_timers;
+
     const wxUsecClock_t now = wxGetLocalTimeUsec();
 
     wxTimerList::iterator cur,
@@ -180,9 +184,20 @@
             DoAddTimer(s);
         }
 
-        // and finally notify the timer
-        timer->Notify();
+        // notify expirate timer out of timers loop to avoid timers list corruption
+        expirate_timers.Insert(timer);
     }
+
+    // finally notify the expirate timers
+    if ( !expirate_timers.empty() )
+    {
+        wxImplTimerList::iterator cur_expirate_timer;
+        for ( cur_expirate_timer = expirate_timers.begin(); cur_expirate_timer != expirate_timers.end(); cur_expirate_timer++)
+        {
+            wxUnixTimerImpl * const timer = *cur_expirate_timer;
+            timer->Notify();
+        }
+    }
 }
 
 // ============================================================================
Index: src/unix/evtloopunix.cpp
===================================================================
--- src/unix/evtloopunix.cpp	(révision 56471)
+++ src/unix/evtloopunix.cpp	(copie de travail)
@@ -149,7 +149,6 @@
 
 bool wxConsoleEventLoop::Dispatch()
 {
-    m_dispatcher->Dispatch();
     wxTheApp->ProcessPendingEvents();
     return true;
 }
Index: include/wx/unix/private/timer.h
===================================================================
--- include/wx/unix/private/timer.h	(révision 56471)
+++ include/wx/unix/private/timer.h	(copie de travail)
@@ -68,6 +68,9 @@
 // the linked list of all active timers, we keep it sorted by expiration time
 WX_DECLARE_LIST(wxTimerSchedule, wxTimerList);
 
+// the list of timers to notify
+WX_DECLARE_LIST(wxUnixTimerImpl, wxImplTimerList);
+
 // ----------------------------------------------------------------------------
 // wxTimerScheduler: class responsible for updating all timers
 // ----------------------------------------------------------------------------
