Index: propgrid.cpp
===================================================================
--- propgrid.cpp	(revision 75875)
+++ propgrid.cpp	(working copy)
@@ -1035,6 +1035,15 @@
     wxPropertyGridManager* pgman = m_pPropGridManager;
     wxPropertyGridPage* pg = pgman->GetPage(wxT("Standard Items"));
 
+    wxPGProperty* p = new wxFloatProperty("Test");
+    p->SetAttribute("Min",0.1f);
+    p->SetAttribute("Max",1);
+    p->SetAttribute("Step",1);
+    p->SetAttribute("Wrap",false);
+    p->SetAttribute("MotionSpin",true);
+    p->SetEditor("SpinCtrl");
+    pg->Append( p );
+
     // Append is ideal way to add items to wxPropertyGrid.
     pg->Append( new wxPropertyCategory(wxT("Appearance"),wxPG_LABEL) );
 
