diff -uNr wxWidgets-3.0.0-orig\samples\propgrid/sampleprops.cpp wxWidgets-3.0.0-15625\samples\propgrid/sampleprops.cpp
--- wxWidgets-3.0.0-orig\samples\propgrid/sampleprops.cpp	2013-11-11 14:18:37.000000000 +0100
+++ wxWidgets-3.0.0-15625\samples\propgrid/sampleprops.cpp	2014-01-01 09:47:13.517478100 +0100
@@ -22,6 +22,9 @@
 #endif
 
 #include "wx/fontdlg.h"
+#if !WXWIN_COMPATIBILITY_3_0
+#include "wx/numformatter.h"
+#endif
 
 // -----------------------------------------------------------------------
 
@@ -377,7 +380,11 @@
 wxString wxArrayDoubleEditorDialog::ArrayGet( size_t index )
 {
     wxString str;
+#if WXWIN_COMPATIBILITY_3_0
     wxPropertyGrid::DoubleToString(str,m_array[index],m_precision,true,&m_dtoaTemplate);
+#else
+    str = wxNumberFormatter::ToString(m_array[index],m_precision,wxNumberFormatter::Style_NoTrailingZeroes);
+#endif
     return str;
 }
 
@@ -552,7 +559,9 @@
 void wxArrayDoubleProperty::GenerateValueAsString( wxString& target, int prec, bool removeZeroes ) const
 {
     wxString s;
+#if WXWIN_COMPATIBILITY_3_0
     wxString template_str;
+#endif
     wxChar between[3] = wxT(", ");
     size_t i;
 
@@ -565,7 +574,11 @@
     for ( i=0; i<value.GetCount(); i++ )
     {
 
+#if WXWIN_COMPATIBILITY_3_0
         wxPropertyGrid::DoubleToString(s,value[i],prec,removeZeroes,&template_str);
+#else
+        s = wxNumberFormatter::ToString(value[i],prec,removeZeroes? wxNumberFormatter::Style_NoTrailingZeroes: wxNumberFormatter::Style_None);
+#endif
 
         target += s;
 
diff -uNr wxWidgets-3.0.0-orig\samples\propgrid/tests.cpp wxWidgets-3.0.0-15625\samples\propgrid/tests.cpp
--- wxWidgets-3.0.0-orig\samples\propgrid/tests.cpp	2013-11-11 14:18:37.000000000 +0100
+++ wxWidgets-3.0.0-15625\samples\propgrid/tests.cpp	2014-01-01 10:14:43.752753100 +0100
@@ -21,6 +21,7 @@
 #include <wx/propgrid/propgrid.h>
 #include <wx/propgrid/advprops.h>
 #include <wx/propgrid/manager.h>
+#include "wx/numformatter.h"
 
 #include "propgrid.h"
 #include "sampleprops.h"
@@ -899,6 +900,7 @@
 
         wxString s;
 
+#if WXWIN_COMPATIBILITY_3_0
         if ( wxPropertyGrid::DoubleToString(s, 123.123, 2, true) !=
                 wxString::Format("123%s12", sep.c_str()) )
             RT_FAILURE();
@@ -910,6 +912,19 @@
             RT_FAILURE();
         if ( wxPropertyGrid::DoubleToString(s, -0.000123, 3, true) != "0" )
             RT_FAILURE();
+
+#endif
+        if ( wxNumberFormatter::ToString(123.123, 2, wxNumberFormatter::Style_NoTrailingZeroes) !=
+                wxString::Format("123%s12", sep.c_str()) )
+            RT_FAILURE();
+        if ( wxNumberFormatter::ToString(-123.123, 4, wxNumberFormatter::Style_None) !=
+                wxString::Format("-123%s1230", sep.c_str()) )
+            RT_FAILURE();
+        if ( wxNumberFormatter::ToString(-0.02, 1, wxNumberFormatter::Style_None) !=
+                wxString::Format("0%s0", sep) )
+            RT_FAILURE();
+        if ( wxNumberFormatter::ToString(-0.000123, 3, wxNumberFormatter::Style_NoTrailingZeroes) != "0" )
+            RT_FAILURE();
     }
 
     {
