Index: src/msw/mediactrl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/msw/mediactrl.cpp,v
retrieving revision 1.56.2.3
diff -u -r1.56.2.3 mediactrl.cpp
--- src/msw/mediactrl.cpp	2006/01/25 20:29:10	1.56.2.3
+++ src/msw/mediactrl.cpp	2006/03/05 21:25:42
@@ -69,6 +69,15 @@
 LRESULT WXDLLIMPEXP_CORE APIENTRY _EXPORT wxWndProc(HWND hWnd, UINT message,
                                    WPARAM wParam, LPARAM lParam);
 
+//---------------------------------------------------------------------------
+// Killed MSVC warnings
+//---------------------------------------------------------------------------
+//disable "cast truncates constant value" for VARIANT_BOOL values
+//passed as parameters in VC5 and up
+#ifdef _MSC_VER
+#pragma warning (disable:4310)
+#endif
+
 //===========================================================================
 //  BACKEND DECLARATIONS
 //===========================================================================

 	  	 
