diff -ru wxWidgets.orig/src/msw/listctrl.cpp wxWidgets/src/msw/listctrl.cpp
--- wxWidgets.orig/src/msw/listctrl.cpp	2007-05-02 15:23:28.000000000 +0200
+++ wxWidgets/src/msw/listctrl.cpp	2007-06-04 22:56:56.000000000 +0200
@@ -2418,7 +2418,7 @@
     return rc;
 }
 
-static bool HandleSubItemPrepaint(LPNMLVCUSTOMDRAW pLVCD, HFONT hfont)
+static bool HandleSubItemPrepaint(LPNMLVCUSTOMDRAW pLVCD, HFONT hfont, int colCount)
 {
     NMCUSTOMDRAW& nmcd = pLVCD->nmcd;
 
@@ -2433,13 +2433,13 @@
     // get the rectangle to paint
     RECT rc;
     ListView_GetSubItemRect(hwndList, item, col, LVIR_BOUNDS, &rc);
-    if ( !col )
+    if ( (col == 0) && (colCount > 1) )
     {
         // broken ListView_GetSubItemRect() returns the entire item rect for
         // 0th subitem while we really need just the part for this column
         RECT rc2;
         ListView_GetSubItemRect(hwndList, item, 1, LVIR_BOUNDS, &rc2);
-
+        // Note: ListView_GetSubItemRect returns TRUE even if col 1 doesn't exist
         rc.right = rc2.left;
         rc.left += 4;
     }
@@ -2603,7 +2603,7 @@
     for ( int col = 0; col < colCount; col++ )
     {
         pLVCD->iSubItem = col;
-        HandleSubItemPrepaint(pLVCD, hfont);
+        HandleSubItemPrepaint(pLVCD, hfont, colCount);
     }
 
     HandleItemPostpaint(nmcd);

 	  	 
