Index: e:/dev/wxWindows/src/common/appcmn.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/common/appcmn.cpp,v
retrieving revision 1.43.2.1
diff -u -2 -r1.43.2.1 appcmn.cpp
--- e:/dev/wxWindows/src/common/appcmn.cpp	2002/09/24 13:40:18	1.43.2.1
+++ e:/dev/wxWindows/src/common/appcmn.cpp	2002/12/01 14:31:03
@@ -386,4 +386,6 @@
         }
 
+        // Delete the defaultly created theme and set the new theme.
+        delete wxTheme::Get();
         wxTheme::Set(theme);
     }

 	  	 
