diff -uNr wxorig/src/osx/utils_osx.cpp wxnew/src/osx/utils_osx.cpp
--- wxorig/src/osx/utils_osx.cpp	2009-11-05 23:30:10.000000000 -0500
+++ wxnew/src/osx/utils_osx.cpp	2009-12-02 23:15:03.000000000 -0500
@@ -95,7 +95,7 @@
 
     static const char * const OPEN_CMD = "/usr/bin/open";
     if ( wxFileExists(OPEN_CMD) &&
-            wxExecute(wxString(OPEN_CMD) + " " + document) )
+            wxExecute(wxString(OPEN_CMD) + " \"" + document + "\"") )
         return true;
 
     return false;
