Index: contrib/include/wx/fl/dyntbar.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/contrib/include/wx/fl/dyntbar.h,v
retrieving revision 1.14
diff -u -r1.14 dyntbar.h
--- contrib/include/wx/fl/dyntbar.h	2003/11/09 21:43:34	1.14
+++ contrib/include/wx/fl/dyntbar.h	2004/02/04 13:12:04
@@ -187,6 +187,9 @@
 
     void RemveTool( int toolIndex );
 
+		// Removes All Tools in the toolbar
+	void ClearAllTools(void);
+
         // Draws a separator. The default implementation draws a shaded line.
 
     virtual void DrawSeparator( wxDynToolInfo& info, wxDC& dc );
Index: contrib/src/fl/dyntbar.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/contrib/src/fl/dyntbar.cpp,v
retrieving revision 1.12
diff -u -r1.12 dyntbar.cpp
--- contrib/src/fl/dyntbar.cpp	2003/11/09 21:40:31	1.12
+++ contrib/src/fl/dyntbar.cpp	2004/02/04 13:12:05
@@ -232,6 +232,16 @@
     // TODO:: if not found, should it be an assertion?
 }
 
+void wxDynamicToolBar::ClearAllTools( void )
+{
+	size_t i,index;
+	index=mTools.Count();
+	for( i = 0; i != index; ++i )
+	{
+		RemveTool(mTools[0]->mIndex);
+	}
+}
+
 void wxDynamicToolBar::AddSeparator( wxWindow* pSepartorWnd )
 {
     wxDynToolInfo* pInfo = new wxDynToolInfo();

 	  	 
