Index: src/common/string.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/common/string.cpp,v
retrieving revision 1.234
diff -u -r1.234 string.cpp
--- src/common/string.cpp	2005/02/15 19:12:36	1.234
+++ src/common/string.cpp	2005/02/18 03:21:50
@@ -1116,7 +1116,7 @@
 // get the pointer to writable buffer of (at least) nLen bytes
 wxChar *wxString::GetWriteBuf(size_t nLen)
 {
-  if ( !AllocBeforeWrite(nLen) ) {
+  if ( !Alloc(nLen) || !CopyBeforeWrite() ) {
     // allocation failure handled by caller
     return NULL;
   }

 	  	 
