--- wxWidgets/src/common/prntbase.cpp	Fri Apr  1 16:31:44 2005
+++ wxWidgets-2.6.1/src/common/prntbase.cpp	Fri Jul  1 12:42:44 2005
@@ -567,6 +567,9 @@
     EVT_PAINT(wxPreviewCanvas::OnPaint)
     EVT_CHAR(wxPreviewCanvas::OnChar)
     EVT_SYS_COLOUR_CHANGED(wxPreviewCanvas::OnSysColourChanged)
+#if wxUSE_MOUSEWHEEL
+    EVT_MOUSEWHEEL(wxPreviewCanvas::OnMouseWheel)
+#endif
 END_EVENT_TABLE()
 
 // VZ: the current code doesn't refresh properly without
@@ -669,6 +672,54 @@
     }
 }
 
+void wxPreviewCanvas::OnMouseWheel(wxMouseEvent& event)
+{
+    wxPreviewControlBar* controlBar = ((wxPreviewFrame*) GetParent())->GetControlBar();
+
+    if( controlBar != NULL ) {
+#if wxUSE_MOUSEWHEEL
+        if (event.ControlDown() && event.GetWheelRotation() != 0) 
+        {
+            int currentZoom = controlBar->GetZoomControl();
+
+            bool forward = event.GetWheelRotation() >= 0;
+            int delta = 0;
+            if( forward )
+            {
+                if( currentZoom <= 100 )
+                    delta = -5;
+                else if( currentZoom <= 120 )
+                    delta = -10;
+                else 
+                    delta = -50;
+            }
+            else 
+            {
+                if( currentZoom < 100 )
+                    delta = 5;
+                else if( currentZoom <= 120 )
+                    delta = 10;
+                else 
+                    delta = 50;
+            }
+            int newZoom = currentZoom + delta;
+            if( newZoom < 10 )
+                newZoom = 10;
+            if( newZoom > 200 )
+                newZoom = 200;
+            if( newZoom != currentZoom )
+            {
+                controlBar->SetZoomControl(newZoom);
+                m_printPreview->SetZoom(newZoom);
+                Refresh();
+            }
+            return;
+        }           
+#endif
+    }
+    event.Skip();
+    
+}
 //----------------------------------------------------------------------------
 // wxPreviewControlBar
 //----------------------------------------------------------------------------
--- wxWidgets/include/wx/prntbase.h	Wed Jun 15 11:19:28 2005
+++ wxWidgets-2.6.1/include/wx/prntbase.h	Fri Jul  1 12:44:02 2005
@@ -313,7 +313,7 @@
 
     void OnPaint(wxPaintEvent& event);
     void OnChar(wxKeyEvent &event);
-
+    void OnMouseWheel(wxMouseEvent& event);
     // Responds to colour changes
     void OnSysColourChanged(wxSysColourChangedEvent& event);
 


 	  	 
