--- C:/Users/T29D0~1.TIS/AppData/Local/Temp/inspection.py-revBASE.svn000.tmp.py	mar giu  4 18:10:17 2013
+++ C:/Users/t.tissino/git/Phoenix/wx/lib/inspection.py	lun set  9 11:26:51 2013
@@ -732,8 +732,6 @@ class InspectionInfoPanel(wx.stc.StyledTextCtrl):
             st.append(self.Fmt('vgap', obj.GetVGap()))
             st.append(self.Fmt('hgap', obj.GetHGap()))
         if isinstance(obj, wx.FlexGridSizer):
-            st.append(self.Fmt('rowheights', obj.GetRowHeights()))
-            st.append(self.Fmt('colwidths', obj.GetColWidths()))
             st.append(self.Fmt('flexdir',
                                FlagsFormatter(orientFlags, obj.GetFlexibleDirection())))
             st.append(self.Fmt('nonflexmode',
