Index: wincmn.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/common/wincmn.cpp,v
retrieving revision 1.129
diff -b -u -2 -r1.129 wincmn.cpp
--- wincmn.cpp	2003/01/27 09:02:51	1.129
+++ wincmn.cpp	2003/02/01 20:38:18
@@ -232,5 +232,5 @@
     if ( parent && (parent->GetExtraStyle() & wxWS_EX_VALIDATE_RECURSIVELY) )
     {
-        SetExtraStyle(wxWS_EX_VALIDATE_RECURSIVELY);
+        SetExtraStyle(GetExtraStyle() | wxWS_EX_VALIDATE_RECURSIVELY);
     }
 

 	  	 
