Index: samples/listctrl/listtest.cpp
===================================================================
--- samples/listctrl/listtest.cpp	(revision 53704)
+++ samples/listctrl/listtest.cpp	(working copy)
@@ -416,7 +416,8 @@
                                     wxDefaultPosition, wxDefaultSize,
                                     flags |
                                     wxSUNKEN_BORDER | wxLC_EDIT_LABELS);
-
+        const long style = (m_listCtrl->GetWindowStyleFlag() & ~ wxBORDER_MASK) | wxBORDER_NONE;
+        m_listCtrl->SetWindowStyleFlag(style);
         switch ( flags & wxLC_MASK_TYPE )
         {
             case wxLC_LIST:
