Index: common/log.cpp
===================================================================
--- common/log.cpp	(revision 67267)
+++ common/log.cpp	(working copy)
@@ -676,16 +676,9 @@
 #if wxUSE_DATETIME
     if ( !ms_timestamp.empty() )
     {
-        wxChar buf[256];
-        time_t timeNow;
-        (void)time(&timeNow);
-
-        struct tm tm;
-        wxStrftime(buf, WXSIZEOF(buf),
-                    ms_timestamp, wxLocaltime_r(&timeNow, &tm));
-
+        wxDateTime t = wxDateTime::UNow();
         str->Empty();
-        *str << buf << wxS(": ");
+        *str << t.Format(ms_timestamp) << wxT(": ");
     }
 #endif // wxUSE_DATETIME
 }
