Index: include/wx/msw/textctrl.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/msw/textctrl.h,v
retrieving revision 1.65
diff -u -r1.65 textctrl.h
--- include/wx/msw/textctrl.h	2005/02/15 14:02:18	1.65
+++ include/wx/msw/textctrl.h	2005/02/18 13:49:36
@@ -188,7 +188,7 @@
 
     // Show a context menu for Rich Edit controls (the standard
     // EDIT control has one already)
-    void OnRightClick(wxMouseEvent& event);
+    void OnContextMenu(wxContextMenuEvent& event);
 
     // be sure the caret remains invisible if the user
     // called HideNativeCaret() before
Index: src/msw/textctrl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/msw/textctrl.cpp,v
retrieving revision 1.239
diff -u -r1.239 textctrl.cpp
--- src/msw/textctrl.cpp	2005/02/15 14:02:25	1.239
+++ src/msw/textctrl.cpp	2005/02/18 14:26:46
@@ -212,7 +212,7 @@
     EVT_DROP_FILES(wxTextCtrl::OnDropFiles)
 
 #if wxUSE_RICHEDIT
-    EVT_RIGHT_UP(wxTextCtrl::OnRightClick)
+    EVT_CONTEXT_MENU(wxTextCtrl::OnContextMenu)
 #endif
 
     EVT_MENU(wxID_CUT, wxTextCtrl::OnCut)
@@ -2030,7 +2030,7 @@
     event.Enable(GetLastPosition() > 0);
 }
 
-void wxTextCtrl::OnRightClick(wxMouseEvent& event)
+void wxTextCtrl::OnContextMenu(wxContextMenuEvent& event)
 {
 #if wxUSE_RICHEDIT
     if (IsRich())
@@ -2048,7 +2048,7 @@
             m_privateContextMenu->AppendSeparator();
             m_privateContextMenu->Append(wxID_SELECTALL, _("Select &All"));
         }
-        PopupMenu(m_privateContextMenu, event.GetPosition());
+        PopupMenu(m_privateContextMenu);
         return;
     }
     else

 	  	 
