diff --git a/src/richtext/richtextbuffer.cpp b/src/richtext/richtextbuffer.cpp
index 41a4625..969cb5a 100644
--- a/src/richtext/richtextbuffer.cpp
+++ b/src/richtext/richtextbuffer.cpp
@@ -200,7 +200,7 @@ bool wxRichTextFloatCollector::GetFloatingObjects(wxRichTextObjectList& objects)
 
 /*
  * Binary search helper function
- * The argument point is the Y coordinate, and this fuction
+ * The argument point is the Y coordinate, and this function
  * always return the floating rect that contain this coordinate
  * or under this coordinate.
  */
@@ -1435,7 +1435,7 @@ bool wxRichTextCompositeObject::DeleteRange(const wxRichTextRange& range)
         // now match the ranges.
         // However, so long as deletion is not done on the same object twice this is OK.
         // If you may delete content from the same object twice, recalculate
-        // the ranges inbetween DeleteRange calls by calling CalculateRanges, and
+        // the ranges between DeleteRange calls by calling CalculateRanges, and
         // adjust the range you're deleting accordingly.
 
         if (!obj->GetRange().IsOutside(range))
@@ -5208,7 +5208,7 @@ bool wxRichTextParagraph::Layout(wxDC& dc, wxRichTextDrawingContext& context, co
     //wxASSERT(!(lastCompletedEndPos != -1 && lastCompletedEndPos < GetRange().GetEnd()-1));
 
     // Add the last line - it's the current pos -> last para pos
-    // Substract -1 because the last position is always the end-paragraph position.
+    // Subtract -1 because the last position is always the end-paragraph position.
     if (lastCompletedEndPos <= GetRange().GetEnd()-1)
     {
         int startOffset = (lineCount == 0 ? startPositionFirstLine : startPositionSubsequentLines);
@@ -9497,7 +9497,7 @@ bool wxRichTextCell::AdjustAttributes(wxRichTextAttr& attr, wxRichTextDrawingCon
 
             // Compute right border
 
-            // We need to explicity look at the spans, not just whether
+            // We need to explicitly look at the spans, not just whether
             // the cell is visible, because that doesn't tell us which
             // cell to look at for border information.
             wxRichTextCell* adjacentCellRight = NULL;
diff --git a/src/unix/dialup.cpp b/src/unix/dialup.cpp
index 95d37b2..c5f8edd 100644
--- a/src/unix/dialup.cpp
+++ b/src/unix/dialup.cpp
@@ -85,7 +85,7 @@ public:
        ISP (exact meaning of the parameter depends on the platform), returns
        true on success or false on failure and logs the appropriate error
        message in the latter case.
-       @param nameOfISP optional paramater for dial program
+       @param nameOfISP optional parameter for dial program
        @param username unused
        @param password unused
    */
diff --git a/src/unix/net.cpp b/src/unix/net.cpp
index 3125aad..318297d 100644
--- a/src/unix/net.cpp
+++ b/src/unix/net.cpp
@@ -84,7 +84,7 @@ public:
        ISP (exact meaning of the parameter depends on the platform), returns
        TRUE on success or FALSE on failure and logs the appropriate error
        message in the latter case.
-       @param nameOfISP optional paramater for dial program
+       @param nameOfISP optional parameter for dial program
        @param username unused
        @param password unused
    */
