--- xh_chckl.cpp	Sun Sep 01 18:11:38 2002
+++ contrib/src/xrc/xh_chckl.cpp	Sun Mar 09 12:28:06 2003
@@ -20,4 +20,6 @@
 #endif
 
+#if wxUSE_OWNER_DRAWN
+
 #include "wx/xrc/xh_chckl.h"
 #include "wx/checklst.h"
@@ -106,3 +108,3 @@
 }
 
-
+#endif
\ No newline at end of file
--- xh_menu.cpp	Tue Aug 20 23:28:16 2002
+++ contrib/src/xrc/xh_menu.cpp	Sun Mar 09 12:27:16 2003
@@ -87,7 +87,9 @@
                                                
 #if wxCHECK_VERSION(2,3,0) || defined(__WXMSW__)
+#if wxUSE_OWNER_DRAWN
                 if (HasParam(wxT("bitmap")))
                     mitem->SetBitmap(GetBitmap(wxT("bitmap"), wxART_MENU));
 #endif
+#endif
             p_menu->Append(mitem);
             mitem->Enable(GetBool(wxT("enabled"), TRUE));

 	  	 
