Index: src/gtk/toplevel.cpp
===================================================================
--- src/gtk/toplevel.cpp	(revision 56036)
+++ src/gtk/toplevel.cpp	(working copy)
@@ -597,7 +597,15 @@
     if ((style & wxSIMPLE_BORDER) || (style & wxNO_BORDER))
     {
         m_gdkDecor = 0;
-        m_gdkFunc = 0;
+        m_gdkFunc = GDK_FUNC_MOVE;
+
+	if ( ( style & wxMINIMIZE_BOX ) == wxMINIMIZE_BOX ) {
+	    m_gdkFunc |= GDK_FUNC_MINIMIZE;
+	}
+	
+	if ( ( style & wxCLOSE_BOX ) == wxCLOSE_BOX ) {
+	    m_gdkFunc |= GDK_FUNC_CLOSE;
+	}
     }
     else
     {
Index: samples/shaped/shaped.cpp
===================================================================
--- samples/shaped/shaped.cpp	(revision 56036)
+++ samples/shaped/shaped.cpp	(working copy)
@@ -365,7 +365,8 @@
                   0
                   | wxFRAME_SHAPED
                   | wxSIMPLE_BORDER
-                  | wxFRAME_NO_TASKBAR
+                  | wxCLOSE_BOX
+                  | wxMINIMIZE_BOX
                   | wxSTAY_ON_TOP
             )
 {
