--- src/osx/core/dcmemory.cpp	Sun Feb  8 16:04:46 2015
+++ src/osx/core/dcmemory.cpp	Sun Feb  8 16:43:12 2015
@@ -85,7 +85,10 @@ void wxMemoryDCImpl::DoSelect( const wxBitmap& bit
         {
             CGContextSetFillColorSpace( bmCtx, genericColorSpace );
             CGContextSetStrokeColorSpace( bmCtx, genericColorSpace );
-            SetGraphicsContext( wxGraphicsContext::CreateFromNative( bmCtx ) );
+            
+            wxGraphicsContext* context = wxGraphicsContext::CreateFromNative( bmCtx );
+            context->EnableOffset(true);
+            SetGraphicsContext( context );
         }
         m_ok = (m_graphicContext != NULL) ;
     }
