Index: contrib/gizmos/wxCode/src/treelistctrl.cpp
===================================================================
--- contrib/gizmos/wxCode/src/treelistctrl.cpp	(revision 63457)
+++ contrib/gizmos/wxCode/src/treelistctrl.cpp	(working copy)
@@ -3086,8 +3086,9 @@
 #endif // !__WXMAC__
             dc.SetTextForeground (colTextHilight);
         }else if (item->IsSelected()) {
-#if defined(__WXGTK2__) || defined(__WXMAC__)
+#if defined(__WXGTK20__) || defined(__WXMAC__)
         int flags = wxCONTROL_SELECTED;
+        dc.SetTextForeground (colTextHilight);
         if (m_hasFocus)
         {
             flags |= wxCONTROL_FOCUSED;
@@ -3095,7 +3096,10 @@
             dc.SetTextForeground( *wxWHITE );
 #endif
         }
-        wxRendererNative::GetDefault().DrawItemSelectionRect( m_owner, dc, wxRect( 0, item->GetY() + off_h, total_w, total_h - off_h ), flags); 
+        if ( item == m_curItem) { 
+            flags |= wxCONTROL_CURRENT;
+        }
+        wxRendererNative::GetDefault().DrawItemSelectionRect( this, dc, wxRect( 0, item->GetY() + off_h, total_w, total_h - off_h ), flags); 
 #else
             if (!m_isDragging && m_hasFocus) {
                 dc.SetBrush (*m_hilightBrush);
@@ -3105,13 +3109,13 @@
                 dc.SetPen (*wxTRANSPARENT_PEN);
             }
             dc.SetTextForeground (colTextHilight);
-#endif // defined(__WXGTK2__) || defined(__WXMAC__)
+#endif // defined(__WXGTK20__) || defined(__WXMAC__)
         }else if (item == m_curItem) {
             dc.SetPen (m_hasFocus? *wxBLACK_PEN: *wxTRANSPARENT_PEN);
         }else{
             dc.SetTextForeground (colText);
         }
-#if !defined(__WXGTK2__) && !defined(__WXMAC__)
+#if !defined(__WXGTK20__) && !defined(__WXMAC__)
         dc.DrawRectangle (0, item->GetY() + off_h, total_w, total_h - off_h);
 #endif 
     }else{
@@ -3174,8 +3178,9 @@
 #endif // !__WXMAC__
                     dc.SetTextForeground (colTextHilight);
                 }else if (item->IsSelected()) {
-#if defined(__WXGTK2__) || defined(__WXMAC__)
+#if defined(__WXGTK20__) || defined(__WXMAC__)
                     int flags = wxCONTROL_SELECTED;
+                    dc.SetTextForeground (colTextHilight);
                     if (m_hasFocus)
                     {
                         flags |= wxCONTROL_FOCUSED;
@@ -3183,7 +3188,20 @@
                         dc.SetTextForeground( *wxWHITE );
 #endif
                     }
-                    wxRendererNative::GetDefault().DrawItemSelectionRect( m_owner, dc, wxRect( 0, item->GetY() + off_h, total_w, total_h - off_h ), flags); 
+                    if ( item == m_curItem) { 
+                        flags |= wxCONTROL_CURRENT;
+                    }
+                    wxRendererNative::GetDefault().DrawItemSelectionRect( this, dc, wxRect( text_x-1, item->GetY() + off_h, text_w+2, total_h - off_h ), flags); 
+#ifdef __WXGTK20__
+                    // Work around ugly selection rectangle on wxGTK: there
+                    // are some artifacts at tle corners, outside the rectangle.
+                    // dc.SetClippingRegion doesn't seem to work for DrawItemSelectionRect,
+                    // so draw a rectangle using the background colour.
+                    wxColour bg_colour = wxSystemSettings::GetColour (wxSYS_COLOUR_WINDOW);
+                    dc.SetPen (wxPen(bg_colour));
+                    dc.SetBrush (*wxTRANSPARENT_BRUSH);
+                    dc.DrawRectangle( text_x-2, item->GetY() + off_h-1, text_w+4, total_h - off_h + 2 );
+#endif // __WXGTK20__
 #else 
                     if (!m_isDragging && m_hasFocus) {
                         dc.SetBrush (*m_hilightBrush);
@@ -3193,13 +3211,13 @@
                         dc.SetPen (*wxTRANSPARENT_PEN);
                     }
                     dc.SetTextForeground (colTextHilight);
-#endif // defined(__WXGTK2__) || defined(__WXMAC__)
+#endif // defined(__WXGTK20__) || defined(__WXMAC__)
                 }else if (item == m_curItem) {
                     dc.SetPen (m_hasFocus? *wxBLACK_PEN: *wxTRANSPARENT_PEN);
                 }else{
                     dc.SetTextForeground (colText);
                 }
-#if !defined(__WXGTK2__) && !defined(__WXMAC__)
+#if !defined(__WXGTK20__) && !defined(__WXMAC__)
                 dc.DrawRectangle (text_x, item->GetY() + off_h, text_w, total_h - off_h);
 #endif
             }else{
