Index: src/gtk/settings.cpp
===================================================================
--- src/gtk/settings.cpp	(revision 56422)
+++ src/gtk/settings.cpp	(working copy)
@@ -185,6 +185,10 @@
             color = wxColor(ListStyle()->base[GTK_STATE_NORMAL]);
             break;
 
+        case wxSYS_COLOUR_LISTBOXTEXT:
+            color = wxColor(ListStyle()->text[GTK_STATE_NORMAL]);
+            break;
+
         case wxSYS_COLOUR_MENUTEXT:
         case wxSYS_COLOUR_WINDOWTEXT:
         case wxSYS_COLOUR_CAPTIONTEXT:
Index: src/gtk1/settings.cpp
===================================================================
--- src/gtk1/settings.cpp	(revision 56422)
+++ src/gtk1/settings.cpp	(working copy)
@@ -250,6 +250,7 @@
         case wxSYS_COLOUR_CAPTIONTEXT:
         case wxSYS_COLOUR_INACTIVECAPTIONTEXT:
         case wxSYS_COLOUR_BTNTEXT:
+        case wxSYS_COLOUR_LISTBOXTEXT:
             if (!gs_objects.m_colBtnText.Ok())
             {
                 int red, green, blue;
Index: src/msw/settings.cpp
===================================================================
--- src/msw/settings.cpp	(revision 56422)
+++ src/msw/settings.cpp	(working copy)
@@ -127,9 +127,14 @@
         0,          // MENUBAR (unused)
     };
 
-    if ( index == wxSYS_COLOUR_LISTBOX )
+    if ( index == wxSYS_COLOUR_LISTBOXTEXT)
     {
         // there is no standard colour with this index, map to another one
+        index = wxSYS_COLOUR_WINDOWTEXT;
+    }
+    else if ( index == wxSYS_COLOUR_LISTBOX )
+    {
+        // there is no standard colour with this index, map to another one
         index = wxSYS_COLOUR_WINDOW;
     }
     else if ( index > wxSYS_COLOUR_BTNHIGHLIGHT )
Index: src/osx/carbon/settings.cpp
===================================================================
--- src/osx/carbon/settings.cpp	(revision 56422)
+++ src/osx/carbon/settings.cpp	(working copy)
@@ -75,6 +75,7 @@
         case wxSYS_COLOUR_CAPTIONTEXT:
         case wxSYS_COLOUR_INFOTEXT:
         case wxSYS_COLOUR_INACTIVECAPTIONTEXT:
+        case wxSYS_COLOUR_LISTBOXTEXT:
             resultColor = *wxBLACK;
             break ;
 
Index: src/generic/treectlg.cpp
===================================================================
--- src/generic/treectlg.cpp	(revision 56422)
+++ src/generic/treectlg.cpp	(working copy)
@@ -3945,7 +3945,7 @@
     return wxListBox::GetClassDefaultAttributes(variant);
 #else
     wxVisualAttributes attr;
-    attr.colFg = wxSystemSettings::GetColour(wxSYS_COLOUR_WINDOWTEXT);
+    attr.colFg = wxSystemSettings::GetColour(wxSYS_COLOUR_LISTBOXTEXT);
     attr.colBg = wxSystemSettings::GetColour(wxSYS_COLOUR_LISTBOX);
     attr.font  = wxSystemSettings::GetFont(wxSYS_DEFAULT_GUI_FONT);
     return attr;
Index: src/generic/listctrl.cpp
===================================================================
--- src/generic/listctrl.cpp	(revision 56422)
+++ src/generic/listctrl.cpp	(working copy)
@@ -5768,7 +5768,7 @@
 #else
     wxUnusedVar(variant);
     wxVisualAttributes attr;
-    attr.colFg = wxSystemSettings::GetColour(wxSYS_COLOUR_WINDOWTEXT);
+    attr.colFg = wxSystemSettings::GetColour(wxSYS_COLOUR_LISTBOXTEXT);
     attr.colBg = wxSystemSettings::GetColour(wxSYS_COLOUR_LISTBOX);
     attr.font  = wxSystemSettings::GetFont(wxSYS_DEFAULT_GUI_FONT);
     return attr;
Index: include/wx/settings.h
===================================================================
--- include/wx/settings.h	(revision 56422)
+++ include/wx/settings.h	(working copy)
@@ -79,6 +79,7 @@
     wxSYS_COLOUR_GRADIENTINACTIVECAPTION,
     wxSYS_COLOUR_MENUHILIGHT,
     wxSYS_COLOUR_MENUBAR,
+    wxSYS_COLOUR_LISTBOXTEXT,
 
     wxSYS_COLOUR_MAX
 };
