diff -ru wxWidgets.orig/src/msw/menu.cpp wxWidgets/src/msw/menu.cpp
--- wxWidgets.orig/src/msw/menu.cpp	2007-02-13 12:01:04.000000000 +0100
+++ wxWidgets/src/msw/menu.cpp	2007-03-01 15:54:19.000000000 +0100
@@ -59,6 +59,25 @@
 // other standard headers
 #include <string.h>
 
+#if _WIN32_WINNT < 0x0500
+#undef _WIN32_WINNT
+#define _WIN32_WINNT 0x0500
+#define MIIM_STRING      0x00000040
+#define MIIM_BITMAP      0x00000080
+#define MIIM_FTYPE       0x00000100
+#define HBMMENU_CALLBACK            ((HBITMAP) -1)
+typedef struct tagMENUINFO
+{
+    DWORD   cbSize;
+    DWORD   fMask;
+    DWORD   dwStyle;
+    UINT    cyMax;
+    HBRUSH  hbrBack;
+    DWORD   dwContextHelpID;
+    DWORD   dwMenuData;
+}   MENUINFO, FAR *LPMENUINFO;
+#endif
+
 #if wxUSE_OWNER_DRAWN && defined(MIIM_BITMAP)
     #include "wx/dynlib.h"
 #endif

 	  	 
