diff -bu1 -r wxWidgets/src/common/filefn.cpp wxW/src/common/filefn.cpp
--- wxWidgets/src/common/filefn.cpp	2004-03-05 18:40:31.000000000 +0100
+++ wxW/src/common/filefn.cpp	2004-03-25 12:01:16.000000000 +0100
@@ -753,6 +753,11 @@
   // Handle User's home (ignore root homes!)
-  size_t len = 0;
-  if ((val = wxGetUserHome (user)) != NULL &&
-      (len = wxStrlen(val)) > 2 &&
-      wxStrncmp(dest, val, len) == 0)
+  val = wxGetUserHome (user);
+  if (!val)
+    return dest;
+
+  size_t len = wxStrlen(val);
+  if (len <= 2)
+    return dest;
+
+  if (wxStrncmp(dest, val, len) == 0)
     {
diff -bu1 -r wxWidgets/src/generic/filedlgg.cpp wxW/src/generic/filedlgg.cpp
--- wxWidgets/src/generic/filedlgg.cpp	2004-03-01 23:54:15.000000000 +0100
+++ wxW/src/generic/filedlgg.cpp	2004-03-25 12:02:24.000000000 +0100
@@ -706,3 +706,3 @@
 
-void wxFileCtrl::OnListDeleteAllItems( wxListEvent &event )
+void wxFileCtrl::OnListDeleteAllItems( wxListEvent &WXUNUSED(event) )
 {

 	  	 
