Index: grid.cpp
===================================================================
RCS file: /home/wxcvs/wxWindows/src/generic/grid.cpp,v
retrieving revision 1.231
diff -b -u -2 -r1.231 grid.cpp
--- grid.cpp	15 Sep 2002 00:29:49 -0000	1.231
+++ grid.cpp	18 Sep 2002 17:46:41 -0000
@@ -2990,6 +2990,8 @@
     wxArrayString sa;
     sa.Alloc( numCols );
+    if(numCols)
     sa.Add( wxEmptyString, numCols );
 
+    if(numRows)
     m_data.Add( sa, numRows );
 }

 	  	 
