Index: src/gtk/window.cpp
===================================================================
RCS file: /home/wxcvs/wxWindows/src/gtk/window.cpp,v
retrieving revision 1.382
diff -c -r1.382 window.cpp
*** src/gtk/window.cpp	23 Apr 2002 20:16:40 -0000	1.382
--- src/gtk/window.cpp	10 May 2002 16:50:27 -0000
***************
*** 639,644 ****
--- 639,654 ----
  
  #endif
  
+ 
+ //-----------------------------------------------------------------------------
+ // "destroy" event
+ //-----------------------------------------------------------------------------
+ static void gtk_window_destroy_callback(GtkWidget * widget, wxWindow *win)
+ {
+     wxWindowDestroyEvent event(win);
+     win->GetEventHandler()->ProcessEvent(event);
+ }
+ 
  //-----------------------------------------------------------------------------
  // "key_press_event" from any window
  //-----------------------------------------------------------------------------
***************
*** 2714,2719 ****
--- 2724,2732 ----
  
      gtk_signal_connect( GTK_OBJECT(widget), "leave_notify_event",
        GTK_SIGNAL_FUNC(gtk_window_leave_callback), (gpointer)this );
+ 
+     gtk_signal_connect( GTK_OBJECT(widget), "destroy",
+       GTK_SIGNAL_FUNC(gtk_window_destroy_callback), (gpointer)this );
  }
  
  bool wxWindowGTK::Destroy()

 	  	 
