diff -rbu2 E:\temp\patch\modif/app.cpp E:\temp\patch\orig/app.cpp
--- E:\temp\patch\modif/app.cpp	Thu Aug  2 19:22:33 2001
+++ E:\temp\patch\orig/app.cpp	Thu Aug  2 18:26:02 2001
@@ -457,93 +457,4 @@
 
 // ---------------------------------------------------------------------------
-// UnregisterWindowClasses
-// ---------------------------------------------------------------------------
-
-bool wxApp::UnregisterWindowClasses()
-{
-	bool retval = TRUE;
-    // frame window class.
-    if ( !UnregisterClass(wxFrameClassName, wxhInstance) )
-    {
-        wxLogLastError(wxT("UnregisterClass(frame)"));
-
-        retval = FALSE;
-    }
-
-    // "no redraw" frame
-    if ( !UnregisterClass(wxFrameClassNameNoRedraw, wxhInstance) )
-    {
-        wxLogLastError(wxT("UnregisterClass(no redraw frame)"));
-
-        return FALSE;
-    }
-
-    // MDI frame window class.
-    if ( !UnregisterClass(wxMDIFrameClassName, wxhInstance) )
-    {
-        wxLogLastError(wxT("UnregisterClass(MDI parent)"));
-
-        retval = FALSE;
-    }
-
-    // "no redraw" MDI frame
-    if ( !UnregisterClass(wxMDIFrameClassNameNoRedraw, wxhInstance) )
-    {
-        wxLogLastError(wxT("UnregisterClass(no redraw MDI parent frame)"));
-
-        retval = FALSE;
-    }
-
-    // MDI child frame window class.
-    if ( !UnregisterClass(wxMDIChildFrameClassName, wxhInstance) )
-    {
-        wxLogLastError(wxT("UnregisterClass(MDI child)"));
-
-        retval = FALSE;
-    }
-
-    // "no redraw" MDI child frame
-    if ( !UnregisterClass(wxMDIChildFrameClassNameNoRedraw, wxhInstance) )
-    {
-        wxLogLastError(wxT("UnregisterClass(no redraw MDI child)"));
-
-        retval = FALSE;
-    }
-
-    // panel window class.
-    if ( !UnregisterClass(wxPanelClassName, wxhInstance) )
-    {
-        wxLogLastError(wxT("UnregisterClass(panel)"));
-
-        retval = FALSE;
-    }
-
-    // no redraw panel window class.
-    if ( !UnregisterClass(wxPanelClassNameNR, wxhInstance) )
-    {
-        wxLogLastError(wxT("UnregisterClass(no redraw panel)"));
-
-        retval = FALSE;
-    }
-
-    // canvas and textsubwindow class name
-    if ( !UnregisterClass(wxCanvasClassName, wxhInstance) )
-    {
-        wxLogLastError(wxT("UnregisterClass(canvas)"));
-
-        retval = FALSE;
-    }
-
-    if ( !UnregisterClass(wxCanvasClassNameNR, wxhInstance) )
-    {
-        wxLogLastError(wxT("UnregisterClass(no redraw canvas)"));
-
-        retval = FALSE;
-    }
-
-    return retval;
-}
-
-// ---------------------------------------------------------------------------
 // Convert Windows to argc, argv style
 // ---------------------------------------------------------------------------
@@ -689,6 +600,4 @@
     ::OleUninitialize();
 #endif
-
-	UnregisterWindowClasses();
 
 #if wxUSE_CTL3D
diff -rbu2 E:\temp\patch\modif/app.h E:\temp\patch\orig/app.h
--- E:\temp\patch\modif/app.h	Thu Aug  2 19:20:42 2001
+++ E:\temp\patch\orig/app.h	Thu Aug  2 19:19:23 2001
@@ -78,5 +78,4 @@
 
     static bool RegisterWindowClasses();
-	static bool UnregisterWindowClasses();
 
     // Convert Windows to argc, argv style

 	  	 
