Index: include/wx/msw/icon.h
===================================================================
--- include/wx/msw/icon.h	(revision 73445)
+++ include/wx/msw/icon.h	(working copy)
@@ -65,6 +65,8 @@
                           wxBitmapType type = wxICON_DEFAULT_TYPE,
                           int desiredWidth = -1, int desiredHeight = -1);
 
+    bool CreateFromHICON(WXHICON icon);
+
     // implementation only from now on
     wxIconRefData *GetIconData() const { return (wxIconRefData *)m_refData; }
 
Index: interface/wx/icon.h
===================================================================
--- interface/wx/icon.h	(revision 73445)
+++ interface/wx/icon.h	(working copy)
@@ -180,6 +180,15 @@
     virtual ~wxIcon();
 
     /**
+        Attach a Windows icon handle. Returns @true if successful.
+
+        @onlyfor{wxmsw}
+
+        @since 2.9.5
+    */
+    bool CreateFromHICON(WXHICON icon);
+
+    /**
         Returns disabled (dimmed) version of the icon.
 
         This method is available in wxIcon only under wxMSW, other ports only
Index: src/msw/gdiimage.cpp
===================================================================
--- src/msw/gdiimage.cpp	(revision 73445)
+++ src/msw/gdiimage.cpp	(working copy)
@@ -530,10 +530,7 @@
         return false;
     }
 
-    icon->SetHICON((WXHICON)hicon);
-    icon->SetSize(size.x, size.y);
-
-    return icon->IsOk();
+    return icon->CreateFromHICON((WXHICON)hicon);
 }
 
 bool wxICOResourceHandler::LoadIcon(wxIcon *icon,
@@ -593,12 +590,7 @@
     }
 #endif
 
-    wxSize size = wxGetHiconSize(hicon);
-    icon->SetSize(size.x, size.y);
-
-    icon->SetHICON((WXHICON)hicon);
-
-    return icon->IsOk();
+    return icon->CreateFromHICON((WXHICON)hicon);
 }
 
 #if wxUSE_PNG_RESOURCE_HANDLER
Index: src/msw/icon.cpp
===================================================================
--- src/msw/icon.cpp	(revision 73445)
+++ src/msw/icon.cpp	(working copy)
@@ -156,3 +156,11 @@
 
     return handler->Load(this, filename, type, desiredWidth, desiredHeight);
 }
+
+bool wxIcon::CreateFromHICON(WXHICON icon)
+{
+    SetHICON(icon);
+    if (IsOk())
+        SetSize(wxGetHiconSize(icon));
+    return IsOk();
+}
