--- griddemo.cpp	Wed Dec 14 18:44:01 2011
+++ griddemo.cpp	Mon Jul  2 11:19:40 2012
@@ -453,8 +453,8 @@
     grid->SetCellValue(3, 7, wxString::Format(wxT("%g"), 12345.67890));
 
     grid->SetColFormatCustom(8,
-            wxString::Format("%s:%i,%i,%s", wxGRID_VALUE_FLOAT, -1, 4, "g"));
-    grid->SetCellValue(0, 8, "Compact\nformat");
+            wxString::Format("%s:%i,%i,%s", wxGRID_VALUE_FLOAT, -1, 4, "e"));
+    grid->SetCellValue(0, 8, "Scientific\nformat");
     grid->SetCellValue(1, 8, wxT("31415e-4"));
     grid->SetCellValue(2, 8, wxT("1415"));
     grid->SetCellValue(3, 8, wxT("123456789e-4"));
