Index: src/html/m_image.cpp
===================================================================
--- src/html/m_image.cpp	(revision 72568)
+++ src/html/m_image.cpp	(working copy)
@@ -702,6 +702,7 @@
                                           m_WParser->GetPixelScale(),
                                           al, mn);
                 m_WParser->ApplyStateToCell(cel);
+                m_WParser->SetLastWasSpace(false);
                 cel->SetId(tag.GetParam(wxT("id"))); // may be empty
                 m_WParser->GetContainer()->InsertCell(cel);
                 if (str)
Index: src/html/winpars.cpp
===================================================================
--- src/html/winpars.cpp	(revision 72568)
+++ src/html/winpars.cpp	(working copy)
@@ -664,6 +664,10 @@
     cell->SetScriptMode(GetScriptMode(), GetScriptBaseline());
 }
 
+void wxHtmlWinParser::SetLastWasSpace(bool wasSpace)
+{
+   m_tmpLastWasSpace = wasSpace;
+}
 
 #if !wxUSE_UNICODE
 void wxHtmlWinParser::SetInputEncoding(wxFontEncoding enc)
Index: include/wx/html/winpars.h
===================================================================
--- include/wx/html/winpars.h	(revision 72568)
+++ include/wx/html/winpars.h	(working copy)
@@ -138,6 +138,8 @@
     // applies current parser state (link, sub/supscript, ...) to given cell
     void ApplyStateToCell(wxHtmlCell *cell);
 
+    void SetLastWasSpace(bool wasSpace);
+
 #if !wxUSE_UNICODE
     void SetInputEncoding(wxFontEncoding enc);
     wxFontEncoding GetInputEncoding() const { return m_InputEnc; }
