diff -ru -xCVS -x'.#*' -x'*.orig' -x'*.rej' -x'*~' wxWidgets.orig/src/univ/themes/win32.cpp wxWidgets/src/univ/themes/win32.cpp
--- wxWidgets.orig/src/univ/themes/win32.cpp	2005-06-01 00:17:57.000000000 +0800
+++ wxWidgets/src/univ/themes/win32.cpp	2006-04-15 12:51:38.224452800 +0800
@@ -2499,9 +2499,17 @@
     {
         // leave a small gap aroudn the line, also account for the toolbar
         // border itself
-        DrawVerticalLine(dc, rectOrig.x + rectOrig.width/2,
+        if(rectOrig.height > rectOrig.width) { // horizontal
+            DrawVerticalLine(dc, rectOrig.x + rectOrig.width/2,
                          rectOrig.y + 2*BORDER_THICKNESS,
                          rectOrig.GetBottom() - BORDER_THICKNESS);
+        }
+        else
+        {
+            DrawHorizontalLine(dc, rectOrig.y + rectOrig.height/2,
+                         rectOrig.x + 2*BORDER_THICKNESS,
+                         rectOrig.GetRight() - BORDER_THICKNESS);
+        }
     }
     // don't draw wxTOOL_STYLE_CONTROL
 }

 	  	 
