diff -bru wxWidgets/src/msw/frame.cpp /mnt/win7/wxWidgets/src/msw/frame.cpp
--- wxWidgets/src/msw/frame.cpp	2012-01-18 13:59:02.000000000 -0800
+++ /mnt/win7/wxWidgets/src/msw/frame.cpp	2012-01-18 21:00:58.367741700 -0800
@@ -873,6 +873,8 @@
 
 bool wxFrame::HandleMenuPopup(wxEventType evtType, WXHMENU hMenu)
 {
+    bool sendEvent = true;
+    wxMenuBar *bar = GetMenuBar();
     bool isPopup = false;
     wxMenu* menu = NULL;
     if ( wxCurrentPopupMenu && wxCurrentPopupMenu->GetHMenu() == hMenu )
@@ -880,13 +882,17 @@
         menu = wxCurrentPopupMenu;
         isPopup = true;
     }
-    else if ( GetMenuBar() )
+    else if ( bar )
     {
-        menu = GetMenuBar()->MSWGetMenu(hMenu);
+        menu = bar->MSWGetMenu(hMenu);
+        if( menu && !bar->IsEnabledTop( bar->FindMenu( menu->GetTitle() ) ))
+            sendEvent = false;
     }
 
-
+    if( sendEvent )
         return DoSendMenuOpenCloseEvent(evtType, menu, isPopup);
+    else
+        return false;
 }
 
 #endif // wxUSE_MENUS
