diff -bru wxWidgets/include/wx/msw/menu.h /mnt/winxp/wxWidgets/include/wx/msw/menu.h
--- wxWidgets/include/wx/msw/menu.h	2014-08-11 13:00:05.000000000 -0700
+++ /mnt/winxp/wxWidgets/include/wx/msw/menu.h	2014-08-25 10:43:04.625000000 -0700
@@ -226,7 +226,7 @@
 
         // get the menu handle
     WXHMENU GetHMenu() const { return m_hMenu; }
-
+    int MSWGetTopMenuPos( WXHMENU hMenu ) const;
     // if the menubar is modified, the display is not updated automatically,
     // call this function to update it (m_menuBarFrame should be !NULL)
     void Refresh();
diff -bru wxWidgets/src/msw/menu.cpp /mnt/winxp/wxWidgets/src/msw/menu.cpp
--- wxWidgets/src/msw/menu.cpp	2014-08-11 12:59:25.000000000 -0700
+++ /mnt/winxp/wxWidgets/src/msw/menu.cpp	2014-08-25 10:43:05.015625000 -0700
@@ -1307,6 +1307,17 @@
     return m_menus[pos]->GetTitle();
 }
 
+int wxMenuBar::MSWGetTopMenuPos( WXHMENU hMenu ) const
+{
+    int res = wxNOT_FOUND;
+    for( int i = 0; i < GetMenuCount(); i++ )
+    {
+        if( GetMenu( i )->GetHMenu() == hMenu )
+            res = i;
+    }
+    return res;
+}
+
 // ---------------------------------------------------------------------------
 // wxMenuBar construction
 // ---------------------------------------------------------------------------
diff -bru wxWidgets/src/msw/window.cpp /mnt/winxp/wxWidgets/src/msw/window.cpp
--- wxWidgets/src/msw/window.cpp	2014-08-11 12:59:25.000000000 -0700
+++ /mnt/winxp/wxWidgets/src/msw/window.cpp	2014-08-25 10:44:28.625000000 -0700
@@ -2808,7 +2808,7 @@
 {
     // did we process the message?
     bool processed = false;
-
+    wxMenuBar *bar = NULL;
     // the return value
     union
     {
@@ -3559,7 +3559,11 @@
 
 #if !defined(__WXMICROWIN__) && !defined(__WXWINCE__)
         case WM_INITMENUPOPUP:
+            bar = wxDynamicCastThis( wxFrame )->GetMenuBar();
+            if( bar && bar->IsEnabledTop( LOWORD( lParam ) ) )
                 processed = HandleMenuPopup(wxEVT_MENU_OPEN, (WXHMENU)wParam);
+            else
+                return 0;
             break;
 
         case WM_MENUSELECT:
@@ -3573,7 +3577,11 @@
             break;
 
         case WM_UNINITMENUPOPUP:
+            bar = wxDynamicCastThis( wxFrame )->GetMenuBar();
+            if( bar && bar->IsEnabledTop( bar->MSWGetTopMenuPos( (WXHMENU) wParam ) ) )
                 processed = HandleMenuPopup(wxEVT_MENU_CLOSE, (WXHMENU)wParam);
+            else
+                return 0;
             break;
 #endif // !__WXMICROWIN__
 #endif // wxUSE_MENUS && !defined(__WXUNIVERSAL__)

