Index: samples/controls/controls.cpp
===================================================================
--- samples/controls/controls.cpp	(revision 76806)
+++ samples/controls/controls.cpp	(working copy)
@@ -738,9 +738,19 @@
     m_listbox = new wxListBox( panel, ID_LISTBOX,
                                wxPoint(10,10), wxSize(120,70),
                                5, choices, wxLB_MULTIPLE | wxLB_ALWAYS_SB | wxHSCROLL );
+	m_listbox->Append("test1");
+	m_listbox->Append("test2");
+	m_listbox->Append("test3");
+	m_listbox->Append("test4");
+	m_listbox->Append("test5");
     m_listboxSorted = new wxListBox( panel, ID_LISTBOX_SORTED,
                                      wxPoint(10,90), wxSize(120,70),
-                                     3, choices, wxLB_SORT );
+                                     5, choices, /*wxLB_SORT*/0 );
+	m_listboxSorted->Append("test1");
+	m_listboxSorted->Append("test2");
+	m_listboxSorted->Append("test3");
+	m_listboxSorted->Append("test4");
+	m_listboxSorted->Append("test5");
 
     SetListboxClientData(wxT("listbox"), m_listbox);
     SetListboxClientData(wxT("listbox"), m_listboxSorted);
@@ -1301,6 +1311,12 @@
     else
         m_text->AppendText( wxString(wxT("none")) );
     m_text->AppendText( wxT("'\n") );
+	
+	if (event.GetId() != ID_LISTBOX) {
+		if (0 <= event.GetInt() && event.GetInt() < m_listbox->GetCount())
+			m_listbox->EnsureVisible(event.GetInt());
+	}
+
 }
 
 void MyPanel::OnListBoxDoubleClick( wxCommandEvent &event )
