Index: src/msw/listbox.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/listbox.cpp,v
retrieving revision 1.91
diff -b -u -2 -r1.91 listbox.cpp
--- src/msw/listbox.cpp	2003/09/24 00:41:05	1.91
+++ src/msw/listbox.cpp	2003/10/10 13:58:57
@@ -401,5 +401,6 @@
 void wxListBox::SetSelection(int N, bool select)
 {
-    wxCHECK_RET( N >= 0 && N < m_noItems,
+    wxCHECK_RET( N == wxNOT_FOUND || 
+                    (N >= 0 && N < m_noItems),
                  wxT("invalid index in wxListBox::SetSelection") );
 

 	  	 
