diff --git a/samples/typetest/typetest.cpp b/samples/typetest/typetest.cpp
index 4708584..49bde0e 100644
--- a/samples/typetest/typetest.cpp
+++ b/samples/typetest/typetest.cpp
@@ -909,7 +909,7 @@ void MyApp::DoMIMEDemo(wxCommandEvent& WXUNUSED(event))
     if ( !!ext )
     {
         s_defaultExt = ext;
-
+wxStopWatch sw;
         // init MIME database if not done yet
         if ( !m_mimeDatabase )
         {
@@ -962,6 +962,7 @@ void MyApp::DoMIMEDemo(wxCommandEvent& WXUNUSED(event))
 
             delete filetype;
         }
+        wxLogDebug("Time taken: %li", sw.Time());
     }
     //else: cancelled by user
 }
