Index: src/msw/listbox.cpp
===================================================================
--- src/msw/listbox.cpp	(revision 52902)
+++ src/msw/listbox.cpp	(working copy)
@@ -306,6 +306,11 @@
     wxCHECK_RET( N == wxNOT_FOUND || IsValid(N),
                  wxT("invalid index in wxListBox::SetSelection") );
 
+    if (N < 0 || IsValid(N))
+    {
+        return;
+    }
+
     if ( HasMultipleSelection() )
     {
         SendMessage(GetHwnd(), LB_SETSEL, select, N);

 	  	 
