diff -r -u -p a/include/wx/dialog.h b/include/wx/dialog.h
--- a/include/wx/dialog.h	2020-05-04 07:09:52.769889100 +0200
+++ b/include/wx/dialog.h	2020-05-04 12:37:43.536456100 +0200
@@ -73,8 +73,8 @@ public:
     virtual bool IsModal() const = 0;
     // show the dialog frame-modally (needs a parent), using app-modal
     // dialogs on platforms that don't support it
-    virtual void ShowWindowModal () ;
-    virtual void SendWindowModalDialogEvent ( wxEventType type );
+    virtual void ShowWindowModal();
+    virtual void SendWindowModalDialogEvent( wxEventType type );
 
     template<typename Functor>
     void ShowWindowModalThenDo(const Functor& onEndModal);
diff -r -u -p a/interface/wx/dialog.h b/interface/wx/dialog.h
--- a/interface/wx/dialog.h	2020-05-04 07:09:53.129304100 +0200
+++ b/interface/wx/dialog.h	2020-05-04 12:37:43.524743900 +0200
@@ -624,7 +624,7 @@ public:
 
         @since 2.9.0
      */
-    void ShowWindowModal();
+    virtual void ShowWindowModal();
 
     /**
         Shows a dialog modal to the parent top level window only and call a
diff -r -u -p a/src/common/dlgcmn.cpp b/src/common/dlgcmn.cpp
--- a/src/common/dlgcmn.cpp	2020-05-04 07:09:53.716348600 +0200
+++ b/src/common/dlgcmn.cpp	2020-05-04 13:55:05.307425900 +0200
@@ -474,7 +474,7 @@ wxDEFINE_EVENT( wxEVT_WINDOW_MODAL_DIALO
 
 wxIMPLEMENT_DYNAMIC_CLASS(wxWindowModalDialogEvent, wxCommandEvent);
 
-void wxDialogBase::ShowWindowModal ()
+void wxDialogBase::ShowWindowModal()
 {
     int retval = ShowModal();
     // wxWindowModalDialogEvent relies on GetReturnCode() returning correct
@@ -483,12 +483,12 @@ void wxDialogBase::ShowWindowModal ()
     // The worst that can happen is that it will be set twice to the same
     // value.
     SetReturnCode(retval);
-    SendWindowModalDialogEvent ( wxEVT_WINDOW_MODAL_DIALOG_CLOSED  );
+    SendWindowModalDialogEvent( wxEVT_WINDOW_MODAL_DIALOG_CLOSED );
 }
 
-void wxDialogBase::SendWindowModalDialogEvent ( wxEventType type )
+void wxDialogBase::SendWindowModalDialogEvent( wxEventType type )
 {
-    wxWindowModalDialogEvent event ( type, GetId());
+    wxWindowModalDialogEvent event( type, GetId() );
     event.SetEventObject(this);
 
     if ( !GetEventHandler()->ProcessEvent(event) )
