Index: src/common/dbgrid.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/common/dbgrid.cpp,v
retrieving revision 1.25
diff -u -r1.25 dbgrid.cpp
--- src/common/dbgrid.cpp	2007/03/15 17:25:08	1.25
+++ src/common/dbgrid.cpp	2007/05/31 05:16:14
@@ -663,7 +663,11 @@
         for (trow = m_keys.GetCount(); trow <= row; trow++)
         {
             wxLogDebug(wxT("Fetching row %i.."), trow);
-            bool ret = m_data->GetNext();
+#ifdef __WXDEBUG__
+            bool ret =
+#else
+            m_data->GetNext();
+#endif
 
             wxLogDebug(wxT(" ...success=(%i)"),ret);
             GenericKey k = m_data->GetKey();

 	  	 
