diff -uNr wxorig/src/generic/hyperlinkg.cpp wxnew/src/generic/hyperlinkg.cpp
--- wxorig/src/generic/hyperlinkg.cpp	2010-07-18 18:53:58.000000000 -0400
+++ wxnew/src/generic/hyperlinkg.cpp	2010-07-23 13:18:06.000000000 -0400
@@ -68,6 +68,8 @@
     if (!wxControl::Create(parent, id, pos, size, style, wxDefaultValidator, name))
         return false;
 
+    SetBackgroundColour(parent->GetBackgroundColour());
+
     // set to non empty strings both the url and the label
     SetURL(url.empty() ? label : url);
     SetLabel(label.empty() ? url : label);
