Index: src/osx/core/fontenum.cpp
===================================================================
--- src/osx/core/fontenum.cpp	(revision 65218)
+++ src/osx/core/fontenum.cpp	(working copy)
@@ -36,8 +36,8 @@
 {
     if ( fixedWidthOnly )
     {
-        wxFAIL_MSG( "enumerating only fixed width fonts not supported" );
-        return false;
+//        wxFAIL_MSG( "enumerating only fixed width fonts not supported" );
+//        return false;
     }
 
     wxArrayString fontFamilies ;
