Index: src/msw/radiobox.cpp
===================================================================
--- src/msw/radiobox.cpp	(revision 51611)
+++ src/msw/radiobox.cpp	(working copy)
@@ -461,6 +461,25 @@
 
 #endif // wxUSE_TOOLTIPS
 
+bool wxRadioBox::Reparent(wxWindowBase *newParent)
+{
+    if ( !wxStaticBox::Reparent( newParent ) )
+    {
+        return false;
+    }
+
+    HWND hwndParent = GetHwndOf(GetParent());
+    for ( size_t item = 0; item < m_radioButtons->GetCount(); item++ )
+    {
+        ::SetParent((*m_radioButtons)[item], hwndParent );
+    }
+#ifdef __WXWINCE__
+    // Set the z-order correctly
+    SetWindowPos(GetHwnd(), HWND_BOTTOM, 0, 0, 0, 0, SWP_NOMOVE|SWP_NOSIZE);
+#endif
+    return true;
+}
+
 WX_FORWARD_STD_METHODS_TO_SUBWINDOWS(wxRadioBox, wxStaticBox, m_radioButtons)
 
 // ----------------------------------------------------------------------------
Index: include/wx/msw/radiobox.h
===================================================================
--- include/wx/msw/radiobox.h	(revision 51611)
+++ include/wx/msw/radiobox.h	(working copy)
@@ -111,6 +111,8 @@
     }
 #endif // wxUSE_HELP
 
+    virtual bool Reparent(wxWindowBase *newParent);
+
     // we inherit a version always returning false from wxStaticBox, override
     // it to behave normally
     virtual bool AcceptsFocus() const { return wxControl::AcceptsFocus(); }

 	  	 
