Index: src/generic/msgdlgg.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/generic/msgdlgg.cpp,v
retrieving revision 1.52
diff -u -r1.52 msgdlgg.cpp
--- src/generic/msgdlgg.cpp	2005/03/12 19:35:44	1.52
+++ src/generic/msgdlgg.cpp	2005/07/06 17:09:54
@@ -85,19 +85,19 @@
                 // fall through
 
             case wxICON_ERROR:
-                bitmap = wxArtProvider::GetIcon(wxART_ERROR, wxART_MESSAGE_BOX);
+                bitmap = wxArtProvider::GetBitmap(wxART_ERROR, wxART_MESSAGE_BOX);
                 break;
 
             case wxICON_INFORMATION:
-                bitmap = wxArtProvider::GetIcon(wxART_INFORMATION, wxART_MESSAGE_BOX);
+                bitmap = wxArtProvider::GetBitmap(wxART_INFORMATION, wxART_MESSAGE_BOX);
                 break;
 
             case wxICON_WARNING:
-                bitmap = wxArtProvider::GetIcon(wxART_WARNING, wxART_MESSAGE_BOX);
+                bitmap = wxArtProvider::GetBitmap(wxART_WARNING, wxART_MESSAGE_BOX);
                 break;
 
             case wxICON_QUESTION:
-                bitmap = wxArtProvider::GetIcon(wxART_QUESTION, wxART_MESSAGE_BOX);
+                bitmap = wxArtProvider::GetBitmap(wxART_QUESTION, wxART_MESSAGE_BOX);
                 break;
         }
         wxStaticBitmap *icon = new wxStaticBitmap(this, wxID_ANY, bitmap);

 	  	 
