--- src/gtk/window.cpp	Sun Jan 18 07:47:05 2015
+++ src/gtk/window.cpp	Sun Jan 18 07:45:33 2015
@@ -2066,6 +2066,15 @@ static void frame_clock_layout(GdkFrameClock*, wxW
 
 void wxWindowGTK::GTKHandleRealized()
 {
+    // Need to ensure there is a native window for this BEFORE freezing
+    // If we don't then the impl_window could change underneath us later on
+    if (m_widget && m_widget != m_wxwindow)
+    {
+        GdkWindow *ensure = gtk_widget_get_window(m_widget);
+        if (ensure)
+            gdk_window_ensure_native(ensure);
+    }
+
     if (IsFrozen())
         DoFreeze();
 
