Index: include/wx/msw/private.h
===================================================================
--- include/wx/msw/private.h	(revision 76598)
+++ include/wx/msw/private.h	(working copy)
@@ -924,7 +924,10 @@
     wxWinVersion_Vista = wxWinVersion_6,
     wxWinVersion_NT6 = wxWinVersion_6,
 
-    wxWinVersion_7 = 0x601
+    wxWinVersion_7 = 0x601,
+
+    wxWinVersion_8 = 0x602,
+    wxWinVersion_8_1 = 0x603
 };
 
 WXDLLIMPEXP_BASE wxWinVersion wxGetWinVersion();
Index: src/msw/utils.cpp
===================================================================
--- src/msw/utils.cpp	(revision 76598)
+++ src/msw/utils.cpp	(working copy)
@@ -1351,10 +1351,25 @@
                     break;
 
                 case 6:
-                    return wxWinVersion_NT6;
+                    switch ( verMin )
+                    {
+                        case 0:
+                            return wxWinVersion_Vista;
+                        
+                        case 1:
+                            return wxWinVersion_7;
+                        
+                        case 2:
+                            return wxWinVersion_8;
+                        
+                        case 3:
+                            return wxWinVersion_8_1;
+                        
+                        default:
+                            return wxWinVersion_8_1;
+                     }                    
+                    break;
             }
-            break;
-
         default:
             // Do nothing just to silence GCC warning
             break;
