--- 1/samples/popup/popup.cpp	2018-10-05 18:54:45.000000000 +0200
+++ 2/samples/popup/popup.cpp	2018-10-08 20:16:39.298179788 +0200
@@ -135,6 +135,7 @@
     topSizer->Add( text, 0, wxALL, 5 );
     topSizer->Add( m_button, 0, wxALL, 5 );
     topSizer->Add( m_spinCtrl, 0, wxALL, 5 );
+    topSizer->Add( new wxTextCtrl{m_panel, wxID_ANY, "try to type here"}, 5 );
     topSizer->Add( m_mouseText, 0, wxCENTRE|wxALL, 5 );
 
     if ( scrolled )
