Index: samples/font/font.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/samples/font/font.cpp,v
retrieving revision 1.55
diff -u -r1.55 font.cpp
--- samples/font/font.cpp	2006/08/22 11:19:06	1.55
+++ samples/font/font.cpp	2006/08/26 00:11:16
@@ -40,7 +40,7 @@
 #endif
 
 // used as title for several dialog boxes
-static const wxChar SAMPLE_TITLE[] = "wxWidgets Font Sample";
+static const wxChar SAMPLE_TITLE[] = _T("wxWidgets Font Sample");
 
 // ----------------------------------------------------------------------------
 // private classes
Index: samples/menu/menu.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/samples/menu/menu.cpp,v
retrieving revision 1.56
diff -u -r1.56 menu.cpp
--- samples/menu/menu.cpp	2006/08/20 22:24:57	1.56
+++ samples/menu/menu.cpp	2006/08/26 00:15:25
@@ -379,56 +379,56 @@
     wxMenu *fileMenu = new wxMenu;
 
     wxMenu *stockSubMenu = new wxMenu;
-    stockSubMenu->Append(wxID_ADD,              "wxID_ADD");
-    stockSubMenu->Append(wxID_APPLY,            "wxID_APPLY");
-    stockSubMenu->Append(wxID_BOLD,             "wxID_BOLD");
-    stockSubMenu->Append(wxID_CANCEL,           "wxID_CANCEL");
-    stockSubMenu->Append(wxID_CLEAR,            "wxID_CLEAR");
-    stockSubMenu->Append(wxID_CLOSE,            "wxID_CLOSE");
-    stockSubMenu->Append(wxID_COPY,             "wxID_COPY");
-    stockSubMenu->Append(wxID_CUT,              "wxID_CUT");
-    stockSubMenu->Append(wxID_DELETE,           "wxID_DELETE");
-    stockSubMenu->Append(wxID_FIND,             "wxID_FIND");
-    stockSubMenu->Append(wxID_REPLACE,          "wxID_REPLACE");
-    stockSubMenu->Append(wxID_BACKWARD,         "wxID_BACKWARD");
-    stockSubMenu->Append(wxID_DOWN,             "wxID_DOWN");
-    stockSubMenu->Append(wxID_FORWARD,          "wxID_FORWARD");
-    stockSubMenu->Append(wxID_UP,               "wxID_UP");
-    stockSubMenu->Append(wxID_HELP,             "wxID_HELP");
-    stockSubMenu->Append(wxID_HOME,             "wxID_HOME");
-    stockSubMenu->Append(wxID_INDENT,           "wxID_INDENT");
-    stockSubMenu->Append(wxID_INDEX,            "wxID_INDEX");
-    stockSubMenu->Append(wxID_ITALIC,           "wxID_ITALIC");
-    stockSubMenu->Append(wxID_JUSTIFY_CENTER,   "wxID_JUSTIFY_CENTER");
-    stockSubMenu->Append(wxID_JUSTIFY_FILL,     "wxID_JUSTIFY_FILL");
-    stockSubMenu->Append(wxID_JUSTIFY_LEFT,     "wxID_JUSTIFY_LEFT");
-    stockSubMenu->Append(wxID_JUSTIFY_RIGHT,    "wxID_JUSTIFY_RIGHT");
-    stockSubMenu->Append(wxID_NEW,              "wxID_NEW");
-    stockSubMenu->Append(wxID_NO,               "wxID_NO");
-    stockSubMenu->Append(wxID_OK,               "wxID_OK");
-    stockSubMenu->Append(wxID_OPEN,             "wxID_OPEN");
-    stockSubMenu->Append(wxID_PASTE,            "wxID_PASTE");
-    stockSubMenu->Append(wxID_PREFERENCES,      "wxID_PREFERENCES");
-    stockSubMenu->Append(wxID_PRINT,            "wxID_PRINT");
-    stockSubMenu->Append(wxID_PREVIEW,          "wxID_PREVIEW");
-    stockSubMenu->Append(wxID_PROPERTIES,       "wxID_PROPERTIES");
-    stockSubMenu->Append(wxID_EXIT,             "wxID_EXIT");
-    stockSubMenu->Append(wxID_REDO,             "wxID_REDO");
-    stockSubMenu->Append(wxID_REFRESH,          "wxID_REFRESH");
-    stockSubMenu->Append(wxID_REMOVE,           "wxID_REMOVE");
-    stockSubMenu->Append(wxID_REVERT_TO_SAVED,  "wxID_REVERT_TO_SAVED");
-    stockSubMenu->Append(wxID_SAVE,             "wxID_SAVE");
-    stockSubMenu->Append(wxID_SAVEAS,           "wxID_SAVEAS");
-    stockSubMenu->Append(wxID_STOP,             "wxID_STOP");
-    stockSubMenu->Append(wxID_UNDELETE,         "wxID_UNDELETE");
-    stockSubMenu->Append(wxID_UNDERLINE,        "wxID_UNDERLINE");
-    stockSubMenu->Append(wxID_UNDO,             "wxID_UNDO");
-    stockSubMenu->Append(wxID_UNINDENT,         "wxID_UNINDENT");
-    stockSubMenu->Append(wxID_YES,              "wxID_YES");
-    stockSubMenu->Append(wxID_ZOOM_100,         "wxID_ZOOM_100");
-    stockSubMenu->Append(wxID_ZOOM_FIT,         "wxID_ZOOM_FIT");
-    stockSubMenu->Append(wxID_ZOOM_IN,          "wxID_ZOOM_IN");
-    stockSubMenu->Append(wxID_ZOOM_OUT,         "wxID_ZOOM_OUT");
+    stockSubMenu->Append(wxID_ADD,             _T("wxID_ADD"));
+    stockSubMenu->Append(wxID_APPLY,           _T("wxID_APPLY"));
+    stockSubMenu->Append(wxID_BOLD,            _T("wxID_BOLD"));
+    stockSubMenu->Append(wxID_CANCEL,          _T("wxID_CANCEL"));
+    stockSubMenu->Append(wxID_CLEAR,           _T("wxID_CLEAR"));
+    stockSubMenu->Append(wxID_CLOSE,           _T("wxID_CLOSE"));
+    stockSubMenu->Append(wxID_COPY,            _T("wxID_COPY"));
+    stockSubMenu->Append(wxID_CUT,             _T("wxID_CUT"));
+    stockSubMenu->Append(wxID_DELETE,          _T("wxID_DELETE"));
+    stockSubMenu->Append(wxID_FIND,            _T("wxID_FIND"));
+    stockSubMenu->Append(wxID_REPLACE,         _T("wxID_REPLACE"));
+    stockSubMenu->Append(wxID_BACKWARD,        _T("wxID_BACKWARD"));
+    stockSubMenu->Append(wxID_DOWN,            _T("wxID_DOWN"));
+    stockSubMenu->Append(wxID_FORWARD,         _T("wxID_FORWARD"));
+    stockSubMenu->Append(wxID_UP,              _T("wxID_UP"));
+    stockSubMenu->Append(wxID_HELP,            _T("wxID_HELP"));
+    stockSubMenu->Append(wxID_HOME,            _T("wxID_HOME"));
+    stockSubMenu->Append(wxID_INDENT,          _T("wxID_INDENT"));
+    stockSubMenu->Append(wxID_INDEX,           _T("wxID_INDEX"));
+    stockSubMenu->Append(wxID_ITALIC,          _T("wxID_ITALIC"));
+    stockSubMenu->Append(wxID_JUSTIFY_CENTER,  _T("wxID_JUSTIFY_CENTER"));
+    stockSubMenu->Append(wxID_JUSTIFY_FILL,    _T("wxID_JUSTIFY_FILL"));
+    stockSubMenu->Append(wxID_JUSTIFY_LEFT,    _T("wxID_JUSTIFY_LEFT"));
+    stockSubMenu->Append(wxID_JUSTIFY_RIGHT,   _T("wxID_JUSTIFY_RIGHT"));
+    stockSubMenu->Append(wxID_NEW,             _T("wxID_NEW"));
+    stockSubMenu->Append(wxID_NO,              _T("wxID_NO"));
+    stockSubMenu->Append(wxID_OK,              _T("wxID_OK"));
+    stockSubMenu->Append(wxID_OPEN,            _T("wxID_OPEN"));
+    stockSubMenu->Append(wxID_PASTE,           _T("wxID_PASTE"));
+    stockSubMenu->Append(wxID_PREFERENCES,     _T("wxID_PREFERENCES"));
+    stockSubMenu->Append(wxID_PRINT,           _T("wxID_PRINT"));
+    stockSubMenu->Append(wxID_PREVIEW,         _T("wxID_PREVIEW"));
+    stockSubMenu->Append(wxID_PROPERTIES,      _T("wxID_PROPERTIES"));
+    stockSubMenu->Append(wxID_EXIT,            _T("wxID_EXIT"));
+    stockSubMenu->Append(wxID_REDO,            _T("wxID_REDO"));
+    stockSubMenu->Append(wxID_REFRESH,         _T("wxID_REFRESH"));
+    stockSubMenu->Append(wxID_REMOVE,          _T("wxID_REMOVE"));
+    stockSubMenu->Append(wxID_REVERT_TO_SAVED, _T("wxID_REVERT_TO_SAVED"));
+    stockSubMenu->Append(wxID_SAVE,            _T("wxID_SAVE"));
+    stockSubMenu->Append(wxID_SAVEAS,          _T("wxID_SAVEAS"));
+    stockSubMenu->Append(wxID_STOP,            _T("wxID_STOP"));
+    stockSubMenu->Append(wxID_UNDELETE,        _T("wxID_UNDELETE"));
+    stockSubMenu->Append(wxID_UNDERLINE,       _T("wxID_UNDERLINE"));
+    stockSubMenu->Append(wxID_UNDO,            _T("wxID_UNDO"));
+    stockSubMenu->Append(wxID_UNINDENT,        _T("wxID_UNINDENT"));
+    stockSubMenu->Append(wxID_YES,             _T("wxID_YES"));
+    stockSubMenu->Append(wxID_ZOOM_100,        _T("wxID_ZOOM_100"));
+    stockSubMenu->Append(wxID_ZOOM_FIT,        _T("wxID_ZOOM_FIT"));
+    stockSubMenu->Append(wxID_ZOOM_IN,         _T("wxID_ZOOM_IN"));
+    stockSubMenu->Append(wxID_ZOOM_OUT,        _T("wxID_ZOOM_OUT"));
     fileMenu->AppendSubMenu(stockSubMenu, _T("&Standard items demo"));
 
 #if USE_LOG_WINDOW

 	  	 
