Index: include/wx/mac/setup0.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/mac/setup0.h,v
retrieving revision 1.21
diff -u -r1.21 setup0.h
--- include/wx/mac/setup0.h	2003/06/16 12:31:37	1.21
+++ include/wx/mac/setup0.h	2003/07/23 15:43:03
@@ -433,6 +433,14 @@
 // wxWave class
 #define wxUSE_WAVE      1
 
+// XML parsing classes. Note that their API will change in the future, so
+// using wxXmlDocument and wxXmlNode in your app is not recommended.
+//
+// Default is 1
+//
+// Recommended setting: 1 (needed by XRC)
+#define wxUSE_XML       1
+
 // ----------------------------------------------------------------------------
 // Individual GUI controls
 // ----------------------------------------------------------------------------

 	  	 
