diff -ru wxWidgets.orig/src/common/wincmn.cpp wxWidgets/src/common/wincmn.cpp
--- wxWidgets.orig/src/common/wincmn.cpp	2005-07-02 21:19:51.000000000 +1000
+++ wxWidgets/src/common/wincmn.cpp	2005-08-11 14:47:43.000000000 +1000
@@ -495,8 +495,8 @@
     yNew += posParent.y;
 
     // FIXME:  This needs to get the client display rect of the display
-    // the window is (via wxDisplay::GetFromWindow). 
-    
+    // the window is (via wxDisplay::GetFromWindow).
+
     // Base size of the visible dimensions of the display
     // to take into account the taskbar. And the Mac menu bar at top.
     wxRect clientrect = wxGetClientDisplayRect();
@@ -2251,26 +2251,30 @@
 
         switch ( wxGetOsVersion() )
         {
-            case wxMOTIF_X:     port += _T("Motif"); break;
+            case wxMOTIF_X:            port += _T("Motif"); break;
             case wxMAC:
-            case wxMAC_DARWIN:  port += _T("Mac"); break;
-            case wxBEOS:        port += _T("BeOS"); break;
+            case wxMAC_DARWIN:         port += _T("Mac"); break;
+            case wxBEOS:               port += _T("BeOS"); break;
             case wxGTK:
             case wxGTK_WIN32:
             case wxGTK_OS2:
-            case wxGTK_BEOS:    port += _T("GTK"); break;
+            case wxGTK_BEOS:           port += _T("GTK"); break;
             case wxWINDOWS:
             case wxPENWINDOWS:
             case wxWINDOWS_NT:
             case wxWIN32S:
             case wxWIN95:
-            case wxWIN386:      port += _T("MS Windows"); break;
+            case wxWIN386:             port += _T("MS Windows"); break;
             case wxMGL_UNIX:
             case wxMGL_X:
             case wxMGL_WIN32:
-            case wxMGL_OS2:     port += _T("MGL"); break;
+            case wxMGL_OS2:            port += _T("MGL"); break;
             case wxWINDOWS_OS2:
-            case wxOS2_PM:      port += _T("OS/2"); break;
+            case wxOS2_PM:             port += _T("OS/2"); break;
+            case wxPALMOS:             port += _T("Palm OS"); break;
+            case wxWINDOWS_CE:         port += _T("Windows CE (generic)"); break;
+            case wxWINDOWS_POCKETPC:   port += _T("Windows CE PocketPC"); break;
+            case wxWINDOWS_SMARTPHONE: port += _T("Windows CE Smartphone"); break;
             default:            port += _T("unknown"); break;
         }
 
diff -ru wxWidgets.orig/src/palmos/utils.cpp wxWidgets/src/palmos/utils.cpp
--- wxWidgets.orig/src/palmos/utils.cpp	2005-07-02 05:36:58.000000000 +1000
+++ wxWidgets/src/palmos/utils.cpp	2005-08-11 14:44:48.000000000 +1000
@@ -213,6 +213,7 @@
 wxToolkitInfo& wxAppTraits::GetToolkitInfo()
 {
     static wxToolkitInfo info;
+    info.os = wxPALMOS;
     info.name = _T("wxBase");
     return info;
 }

 	  	 
