 samples/minimal/minimal.cpp | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/samples/minimal/minimal.cpp b/samples/minimal/minimal.cpp
index a78e462..c0263d7 100644
--- a/samples/minimal/minimal.cpp
+++ b/samples/minimal/minimal.cpp
@@ -143,7 +143,7 @@ bool MyApp::OnInit()
 
 // frame constructor
 MyFrame::MyFrame(const wxString& title)
-       : wxFrame(NULL, wxID_ANY, title)
+       : wxFrame(NULL, wxID_ANY, title, wxDefaultPosition, wxSize(300, 100), wxRESIZE_BORDER|wxCLIP_CHILDREN)
 {
     // set the frame icon
     SetIcon(wxICON(sample));
@@ -167,11 +167,7 @@ MyFrame::MyFrame(const wxString& title)
     SetMenuBar(menuBar);
 #endif // wxUSE_MENUS
 
-#if wxUSE_STATUSBAR
-    // create a status bar just for fun (by default with 1 pane only)
-    CreateStatusBar(2);
-    SetStatusText("Welcome to wxWidgets!");
-#endif // wxUSE_STATUSBAR
+    wxPanel* panel = new wxPanel(this);
 }
 
 
