diff -rbu2 wxWindows-cvs-orig/src/msw/ownerdrw.cpp wxWindows-cvs-patch/src/msw/ownerdrw.cpp
--- wxWindows-cvs-orig/src/msw/ownerdrw.cpp	Mon Sep 29 21:18:44 2003
+++ wxWindows-cvs-patch/src/msw/ownerdrw.cpp	Mon Sep 29 22:15:26 2003
@@ -262,9 +262,12 @@
 
 
-  // if background is white, don't draw an edge around the bitmap
+  // don't draw an edge around the bitmap, if background is white ...
   DWORD menu_bg_color = GetSysColor(COLOR_MENU);
-  if (GetRValue(menu_bg_color) >= 0xf0 &&
-      GetGValue(menu_bg_color) >= 0xf0 &&
-      GetBValue(menu_bg_color) >= 0xf0)
+  if (    ( GetRValue( menu_bg_color ) >= 0xf0 &&
+            GetGValue( menu_bg_color ) >= 0xf0 &&
+            GetBValue( menu_bg_color ) >= 0xf0 )
+       // ... or if the menu item is disabled
+       || ( st & wxODDisabled )
+    )
   {
       draw_bitmap_edge = false;
@@ -403,5 +406,7 @@
   }
   else {
-    wxBitmap bmp( // for disabled items we use m_bmpDisabled if it exists
+    wxBitmap bmp( // for selected but disabled items: no Bitmap
+                  ( st & wxODSelected ) && ( st & wxODDisabled ) ? wxNullBitmap :
+                  // for unselected and disabled items we use m_bmpDisabled if it exists
                   ( GetDisabledBitmap() != wxNullBitmap && ( st & wxODDisabled ) ) ? GetDisabledBitmap() :
     // for uncheckable item we use only the 'checked' bitmap

 	  	 
