Index: include/wx/sizer.h
===================================================================
--- include/wx/sizer.h	(revision 53976)
+++ include/wx/sizer.h	(working copy)
@@ -58,6 +58,7 @@
         return *this;
     }
 
+    // it removes current alignment flags and sets the given alignment
     wxSizerFlags& Align(int alignment) // combination of wxAlignment values
     {
         m_flags &= ~wxALIGN_MASK;
@@ -73,13 +74,34 @@
     }
 
     // some shortcuts for Align()
+    // don't remove vertical flags when setting a horizontal one and vice versa
     wxSizerFlags& Centre() { return Align(wxALIGN_CENTRE); }
     wxSizerFlags& Center() { return Centre(); }
-    wxSizerFlags& Top() { return Align(wxALIGN_TOP); }
-    wxSizerFlags& Left() { return Align(wxALIGN_LEFT); }
-    wxSizerFlags& Right() { return Align(wxALIGN_RIGHT); }
-    wxSizerFlags& Bottom() { return Align(wxALIGN_BOTTOM); }
 
+    wxSizerFlags& Top() 
+    { 
+        m_flags &= ~(wxALIGN_BOTTOM | wxALIGN_CENTRE_VERTICAL); 
+        return *this; 
+    }
+
+    wxSizerFlags& Left() 
+    { 
+        m_flags &= ~(wxALIGN_RIGHT | wxALIGN_CENTRE_HORIZONTAL); 
+        return *this; 
+    }
+
+    wxSizerFlags& Right() 
+    { 
+        m_flags = (m_flags & ~wxALIGN_CENTRE_HORIZONTAL) | wxALIGN_RIGHT; 
+        return *this; 
+    }
+
+    wxSizerFlags& Bottom() 
+    { 
+        m_flags = (m_flags & ~wxALIGN_CENTRE_VERTICAL) | wxALIGN_BOTTOM; 
+        return *this; 
+    }
+
     // default border size used by Border() below
     static int GetDefaultBorder()
     {
