diff --git a/src/aui/barartmsw.cpp b/src/aui/barartmsw.cpp
index 87eb536ce7..63e0cd2196 100644
--- a/src/aui/barartmsw.cpp
+++ b/src/aui/barartmsw.cpp
@@ -29,6 +29,7 @@
 
 wxAuiMSWToolBarArt::wxAuiMSWToolBarArt()
 {
+#if wxUSE_UXTHEME
     if ( wxUxThemeIsActive() )
     {
         m_themed = true;
@@ -61,6 +62,9 @@ wxAuiMSWToolBarArt::wxAuiMSWToolBarArt()
     }
     else
         m_themed = false;
+#else
+    m_themed = false;
+#endif
 }
 
 wxAuiToolBarArt* wxAuiMSWToolBarArt::Clone()
diff --git a/src/aui/tabartmsw.cpp b/src/aui/tabartmsw.cpp
index 9276070124..bef0fc2d52 100644
--- a/src/aui/tabartmsw.cpp
+++ b/src/aui/tabartmsw.cpp
@@ -29,8 +29,11 @@ wxAuiMSWTabArt::wxAuiMSWTabArt()
 {
     m_closeBtnSize = wxDefaultSize;
     m_maxTabHeight = 0;
-
+#if wxUSE_UXTHEME
     m_themed = wxUxThemeIsActive();
+#else
+    m_themed = false;
+#endif
 }
 
 wxAuiMSWTabArt::~wxAuiMSWTabArt()
diff --git a/src/generic/richtooltipg.cpp b/src/generic/richtooltipg.cpp
index 574f36b672..8a97ac701d 100644
--- a/src/generic/richtooltipg.cpp
+++ b/src/generic/richtooltipg.cpp
@@ -278,7 +278,11 @@ private:
         if ( wxGetWinVersion() <= wxWinVersion_XP )
             return false;
 		else
+#if wxUSE_UXTHEME
 			return wxUxThemeIsActive();
+#else
+            return false;
+#endif
     }
 #endif // __WXMSW__
 
diff --git a/src/msw/combo.cpp b/src/msw/combo.cpp
index 2923f575c7..f94cdaa848 100644
--- a/src/msw/combo.cpp
+++ b/src/msw/combo.cpp
@@ -709,8 +709,10 @@ bool wxComboCtrl::AnimateShow( const wxRect& rect, int flags )
 
 wxCoord wxComboCtrl::GetNativeTextIndent() const
 {
+#if wxUSE_UXTHEME
     if ( wxUxThemeIsActive() )
         return NATIVE_TEXT_INDENT_XP;
+#endif
     return NATIVE_TEXT_INDENT_CLASSIC;
 }
 
diff --git a/src/msw/richtooltip.cpp b/src/msw/richtooltip.cpp
index 2dacd70e84..edb7c7c8f4 100644
--- a/src/msw/richtooltip.cpp
+++ b/src/msw/richtooltip.cpp
@@ -192,9 +192,10 @@ wxRichToolTipImpl::Create(const wxString& title, const wxString& message)
 {
     // EM_SHOWBALLOONTIP is only implemented by comctl32.dll v6 so don't even
     // bother using the native implementation if we're not using themes.
+#if wxUSE_UXTHEME
     if ( wxUxThemeIsActive() )
         return new wxRichToolTipMSWImpl(title, message);
-
+#endif
     return new wxRichToolTipGenericImpl(title, message);
 }
 
