--- wxWidgets\src\generic\listctrl.cpp	2005-05-31 10:35:08.000000000 +0200
+++ wxWidgetsCVS\src\generic\listctrl.cpp	2005-07-12 12:58:50.921875000 +0200
@@ -4,5 +4,5 @@
 // Author:      Robert Roebling
 //              Vadim Zeitlin (virtual list control support)
-// Id:          $Id: listctrl.cpp,v 1.374 2005/05/31 09:35:07 JS Exp $
+// Id:          $Id: listctrl.cpp,v 1.375 2005/06/07 19:01:44 ABX Exp $
 // Copyright:   (c) 1998 Robert Roebling
 // Licence:     wxWindows licence
@@ -2244,7 +2244,4 @@
                                     );
 
-    wxSize sz = size;
-    sz.y = 25;
-
     SetScrollbars( 0, 0, 0, 0, 0, 0 );
 
@@ -5098,4 +5095,20 @@
 }
 
+void wxGenericListCtrl::SetItemFont( long item, const wxFont &f )
+{
+    wxListItem info;
+    info.m_itemId = item;
+    info.SetFont( f );
+    m_mainWin->SetItem( info );
+}
+
+wxFont wxGenericListCtrl::GetItemFont( long item ) const
+{
+    wxListItem info;
+    info.m_itemId = item;
+    m_mainWin->GetItem( info );
+    return info.GetFont();
+}
+
 int wxGenericListCtrl::GetSelectedItemCount() const
 {

 	  	 
