--- wxWidgets\src\msw\listctrl.cpp	2005-05-31 10:20:32.000000000 +0200
+++ wxWidgetsCVS\src\msw\listctrl.cpp	2005-07-12 12:57:01.765625000 +0200
@@ -1093,4 +1093,22 @@
 }
 
+void wxListCtrl::SetItemFont( long item, const wxFont &f )
+{
+    wxListItem info;
+    info.m_itemId = item;
+    info.SetFont( f );
+    SetItem( info );
+}
+
+wxFont wxListCtrl::GetItemFont( long item ) const
+{
+    wxFont f;
+    wxListItemInternalData *data = wxGetInternalData(this, item);
+    if ( data && data->attr )
+        f = data->attr->GetFont();
+
+    return f;
+}
+
 // Gets the number of selected items in the list control
 int wxListCtrl::GetSelectedItemCount() const

 	  	 
