diff -ru wxWidgets.orig/include/wx/msw/listbox.h wxWidgets/include/wx/msw/listbox.h
--- wxWidgets.orig/include/wx/msw/listbox.h	2005-04-11 16:41:38.000000000 +1000
+++ wxWidgets/include/wx/msw/listbox.h	2005-06-12 02:33:51.000000000 +1000
@@ -151,6 +151,13 @@
 
     virtual wxSize DoGetBestSize() const;
 
+    // under XP when using "transition effect for menus and tooltips" if we
+    // return true for WM_PRINTCLIENT here then it causes noticable slowdown
+    virtual bool MSWShouldPropagatePrintChild()
+    {
+        return false;
+    }
+
 #if wxUSE_OWNER_DRAWN
     // control items
     wxListBoxItemsArray m_aItems;
diff -ru wxWidgets.orig/include/wx/msw/window.h wxWidgets/include/wx/msw/window.h
--- wxWidgets.orig/include/wx/msw/window.h	2005-05-23 14:49:45.000000000 +1000
+++ wxWidgets/include/wx/msw/window.h	2005-06-12 02:25:58.000000000 +1000
@@ -391,6 +391,14 @@
         return false;
     }
 
+    // some controls (e.g. wxListBox) need to set the return value themselves
+    //
+    // return true to let parent handle it if we don't, false otherwise
+    virtual bool MSWShouldPropagatePrintChild()
+    {
+        return true;
+    }
+
 
     // Responds to colour changes: passes event on to children.
     void OnSysColourChanged(wxSysColourChangedEvent& event);
diff -ru wxWidgets.orig/src/msw/window.cpp wxWidgets/src/msw/window.cpp
--- wxWidgets.orig/src/msw/window.cpp	2005-06-01 02:29:09.000000000 +1000
+++ wxWidgets/src/msw/window.cpp	2005-06-12 02:30:12.000000000 +1000
@@ -4150,6 +4150,11 @@
     if ( IsTopLevel() || InheritsBackgroundColour() )
         return false;
 
+    // sometimes we don't want the parent to handle it at all, instead
+    // return whatever value this window wants
+    if ( !MSWShouldPropagatePrintChild() )
+        return MSWPrintChild(hDC, (wxWindow *)this);
+
     for ( wxWindow *win = GetParent(); win; win = win->GetParent() )
     {
         if ( win->MSWPrintChild(hDC, (wxWindow *)this) )

 	  	 
