--- fdrepdlg.cpp	Sat Apr 05 23:01:16 2003
+++ src/msw/fdrepdlg.cpp	Sun Apr 13 08:43:40 2003
@@ -229,7 +229,25 @@
                                          WPARAM wParam, LPARAM lParam)
 {
+    static unsigned long lastmsgflags = 0;
+    // This flag helps us to identify the bogus ANSI message
+    // sent by UNICOWS.DLL (see below)
+    // while we're sending our message to the dialog
+    // we ignore possible messages sent in between
+    bool active = false;
+
     if ( nMsg == wxFindReplaceDialogImpl::GetFindDialogMessage() )
     {
         FINDREPLACE *pFR = (FINDREPLACE *)lParam;
+        // This is a hack for a MSLU problem: Versions up to 1.0.4011
+        // of UNICOWS.DLL send the correct UNICODE item after button press
+        // and a bogus ANSI mode item right after this, so lets ignore
+        // the second bogus message
+        if (lastmsgflags == pFR->Flags)
+        {
+            lastmsgflags = 0;
+            return 0;
+        }
+
+        lastmsgflags = pFR->Flags;
         wxFindReplaceDialog *dialog = (wxFindReplaceDialog *)pFR->lCustData;
 
@@ -287,6 +305,11 @@
         }
 
+        active = true;
         dialog->Send(event);
+        active = false;
     }
+    else
+    if (!active)
+        lastmsgflags = 0;
 
     WNDPROC wndProc = (WNDPROC)::GetWindowLong(hwnd, GWL_USERDATA);

 	  	 
