Index: include/wx/docview.h
===================================================================
--- include/wx/docview.h	(revision 70377)
+++ include/wx/docview.h	(working copy)
@@ -865,12 +865,12 @@
 
 protected:
     // hook the document manager into event handling chain here
-    virtual bool TryBefore(wxEvent& event)
+    virtual bool TryAfter(wxEvent& event)
     {
         if ( m_docManager && m_docManager->ProcessEventLocally(event) )
             return true;
 
-        return BaseFrame::TryBefore(event);
+        return BaseFrame::TryAfter(event);
     }
 
 private:
