--- wxWindows/contrib/samples/svg/svgtest.cpp	2002-06-21 11:46:32.000000000 +0200
+++ wxWin/contrib/samples/svg/svgtest.cpp	2003-08-21 21:10:24.000000000 +0200
@@ -384,3 +384,3 @@
     m_child = (MyChild *) parent ;
-    SetBackgroundColour(wxColour("WHITE"));
+    SetBackgroundColour(wxColour(_T("WHITE")));
     m_index = m_child->m_frame->nWinCreated % 7 ;
@@ -427,6 +427,6 @@
             dc.DrawRectangle(10, 10, 100, 70);
-            wB = wxBrush ("DARK ORCHID", wxTRANSPARENT);
+            wB = wxBrush (_T("DARK ORCHID"), wxTRANSPARENT);
             dc.SetBrush (wB);
             dc.DrawRoundedRectangle(50, 50, 100, 70, 20);
-            dc.SetBrush (wxBrush("GOLDENROD", wxSOLID) );
+            dc.SetBrush (wxBrush(_T("GOLDENROD"), wxSOLID) );
             dc.DrawEllipse(100, 100, 100, 50);
@@ -451,6 +451,6 @@
             wC = dc.GetTextForeground() ;
-            dc.SetTextForeground ("FIREBRICK");
+            dc.SetTextForeground (_T("FIREBRICK"));
                                  
             // no effect in msw ??
-            dc.SetTextBackground ("WHEAT");
+            dc.SetTextBackground (_T("WHEAT"));
             dc.DrawText(wxT("This is a Red string"), 50, 200);
@@ -475,3 +475,3 @@
 
-            wP.SetColour ("CADET BLUE");
+            wP.SetColour (_T("CADET BLUE"));
             dc.SetPen(wP);
@@ -479,3 +479,3 @@
 
-            wP.SetColour ("SALMON");
+            wP.SetColour (_T("SALMON"));
             dc.SetPen(wP);
@@ -490,3 +490,3 @@
                                  //wxTRANSPARENT));
-            dc.SetBrush (wxBrush ("SALMON",wxSOLID)) ;
+            dc.SetBrush (wxBrush (_T("SALMON"),wxSOLID)) ;
             dc.DrawEllipticArc(300,  0,200,100, 0.0,145.0) ;
@@ -501,3 +501,3 @@
             dc.DrawCheckMark ( 30,30,25,25);
-            dc.SetBrush (wxBrush ("SALMON",wxTRANSPARENT));
+            dc.SetBrush (wxBrush (_T("SALMON"),wxTRANSPARENT));
             dc.DrawCheckMark ( 80,50,75,75);

 	  	 
