Index: wxWidgets/src/common/image.cpp
===================================================================
--- wxWidgets/src/common/image.cpp	(révision 52833)
+++ wxWidgets/src/common/image.cpp	(copie de travail)
@@ -2203,17 +2203,20 @@
     if ( type == wxBITMAP_TYPE_ANY )
     {
         wxList &list=GetHandlers();
-
-        for ( wxList::compatibility_iterator node = list.GetFirst(); node; node = node->GetNext() )
+        bool found = false;
+        
+        for ( wxList::compatibility_iterator node = list.GetFirst(); !found && node; node = node->GetNext() )
         {
              handler=(wxImageHandler*)node->GetData();
              if ( handler->CanRead(stream) )
-                 return handler->LoadFile(this, stream, true/*verbose*/, index);
+                 found = handler->LoadFile(this, stream, true/*verbose*/, index);
 
         }
 
-        wxLogWarning( _("No handler found for image type.") );
-        return false;
+        if ( !found)
+          wxLogWarning( _("No handler found for image type.") );
+          
+        return found;
     }
 
     handler = FindHandler(type);

 	  	 
