Index: include/wx/compositewin.h
===================================================================
--- include/wx/compositewin.h	(revision 69214)
+++ include/wx/compositewin.h	(working copy)
@@ -3,7 +3,7 @@
 // Purpose:     wxCompositeWindow<> declaration
 // Author:      Vadim Zeitlin
 // Created:     2011-01-02
-// RCS-ID:      $Id: wxhead.h,v 1.12 2010-04-22 12:44:51 zeitlin Exp $
+// RCS-ID:      $Id$
 // Copyright:   (c) 2011 Vadim Zeitlin <vadim@wxwidgets.org>
 // Licence:     wxWindows licence
 ///////////////////////////////////////////////////////////////////////////////
@@ -12,6 +12,7 @@
 #define _WX_COMPOSITEWIN_H_
 
 #include "wx/window.h"
+class wxToolTip;
 
 // NB: This is an experimental and, as for now, undocumented class used only by
 //     wxWidgets itself internally. Don't use it in your code until its API is
@@ -59,7 +60,7 @@
         if ( !BaseWindowClass::SetForegroundColour(colour) )
             return false;
 
-        DoSetForAllParts(&wxWindowBase::SetForegroundColour, colour);
+        SetForAllParts<bool>(&wxWindowBase::SetForegroundColour, colour);
 
         return true;
     }
@@ -69,7 +70,7 @@
         if ( !BaseWindowClass::SetBackgroundColour(colour) )
             return false;
 
-        DoSetForAllParts(&wxWindowBase::SetBackgroundColour, colour);
+        SetForAllParts<bool>(&wxWindowBase::SetBackgroundColour, colour);
 
         return true;
     }
@@ -79,7 +80,7 @@
         if ( !BaseWindowClass::SetFont(font) )
             return false;
 
-        DoSetForAllParts(&wxWindowBase::SetFont, font);
+        SetForAllParts<bool>(&wxWindowBase::SetFont, font);
 
         return true;
     }
@@ -89,19 +90,40 @@
         if ( !BaseWindowClass::SetCursor(cursor) )
             return false;
 
-        DoSetForAllParts(&wxWindowBase::SetCursor, cursor);
+        SetForAllParts<bool>(&wxWindowBase::SetCursor, cursor);
 
         return true;
     }
 
+#if wxUSE_TOOLTIPS
+    virtual void DoSetToolTip(wxToolTip *tip)
+    {
+        BaseWindowClass::DoSetToolTip(tip);
+
+        SetForAllParts<>(&wxWindowBase::SetToolTipCopy, tip);
+    }
+#endif // wxUSE_TOOLTIPS
+
 private:
     // Must be implemented by the derived class to return all children to which
     // the public methods we override should forward to.
     virtual wxWindowList GetCompositeWindowParts() const = 0;
 
-    template <class T>
-    void DoSetForAllParts(bool (wxWindowBase::*func)(const T&), const T& arg)
+    template <typename R, class T>
+    void SetForAllParts(R (wxWindowBase::*func)(const T&), const T& arg)
     {
+        DoSetForAllParts<R, const T&>(func, arg);
+    }
+
+    template <typename R, class T>
+    void SetForAllParts(R (wxWindowBase::*func)(T*), T* arg)
+    {
+        DoSetForAllParts<R, T*>(func, arg);
+    }
+
+    template <typename R, class T>
+    void DoSetForAllParts(R (wxWindowBase::*func)(T), T& arg)
+    {
         // Simply call the setters for all parts of this composite window.
         const wxWindowList parts = GetCompositeWindowParts();
         for ( wxWindowList::const_iterator i = parts.begin();
Index: include/wx/window.h
===================================================================
--- include/wx/window.h	(revision 69214)
+++ include/wx/window.h	(working copy)
@@ -1707,6 +1707,8 @@
 
 #if wxUSE_TOOLTIPS
     virtual void DoSetToolTip( wxToolTip *tip );
+        // attach a new tooltip to the window with a copy of the message
+    void SetToolTipCopy( wxToolTip *tip );
 #endif // wxUSE_TOOLTIPS
 
 #if wxUSE_MENUS
Index: samples/widgets/searchctrl.cpp
===================================================================
--- samples/widgets/searchctrl.cpp	(revision 69214)
+++ samples/widgets/searchctrl.cpp	(working copy)
@@ -174,6 +174,8 @@
 
     m_srchCtrl = new wxSearchCtrl(this, -1, wxEmptyString, wxDefaultPosition,
                                   wxSize(150, -1), style);
+
+    m_srchCtrl->SetToolTip("Same tooltip for entire wxSearchCtrl.");
 }
 
 void SearchCtrlWidgetsPage::Reset()
Index: src/common/wincmn.cpp
===================================================================
--- src/common/wincmn.cpp	(revision 69214)
+++ src/common/wincmn.cpp	(working copy)
@@ -2121,6 +2121,11 @@
     }
 }
 
+void wxWindowBase::SetToolTipCopy( wxToolTip *tip )
+{
+    SetToolTip(tip ? new wxToolTip(tip->GetTip()) : NULL);
+}
+
 #endif // wxUSE_TOOLTIPS
 
 // ----------------------------------------------------------------------------
