Index: include/wx/valtext.h
===================================================================
--- include/wx/valtext.h	(revision 57950)
+++ include/wx/valtext.h	(working copy)
@@ -89,14 +89,15 @@
     wxArrayString        m_excludes;
 
 private:
-    DECLARE_DYNAMIC_CLASS(wxTextValidator)
-    DECLARE_EVENT_TABLE()
 // Cannot use
 //  DECLARE_NO_COPY_CLASS(wxTextValidator)
 // because copy constructor is explicitly declared above;
 // but no copy assignment operator is defined, so declare
 // it private to prevent the compiler from defining it:
     wxTextValidator& operator=(const wxTextValidator&);
+
+    DECLARE_DYNAMIC_CLASS(wxTextValidator)
+    DECLARE_EVENT_TABLE()
 };
 
 #endif
