Index: src/generic/datavgen.cpp
===================================================================
--- src/generic/datavgen.cpp	(revision 67106)
+++ src/generic/datavgen.cpp	(working copy)
@@ -214,6 +214,9 @@
             model->Resort();
 
         owner->OnColumnChange(idx);
+        // Send column sorted event if needed
+        if(col->IsSortable())
+                SendEvent(wxEVT_COMMAND_DATAVIEW_COLUMN_SORTED, idx);
     }
 
     void OnRClick(wxHeaderCtrlEvent& event)
