--- editors.cpp.orig	2012-12-06 19:15:53.348839212 +0200
+++ editors.cpp	2012-12-06 19:32:37.924858066 +0200
@@ -1416,58 +1416,13 @@
 const int wxSCB_SETVALUE_CYCLE = 2;
 
 
-static void DrawSimpleCheckBox( wxDC& dc, const wxRect& rect, int box_hei,
+static void DrawSimpleCheckBox( wxWindow* win, wxDC& dc, const wxRect& rect, int box_hei,
                                 int state )
 {
     // Box rectangle.
     wxRect r(rect.x+wxPG_XBEFORETEXT,rect.y+((rect.height-box_hei)/2),
              box_hei,box_hei);
-    wxColour useCol = dc.GetTextForeground();
-
-    if ( state & wxSCB_STATE_UNSPECIFIED )
-    {
-        useCol = wxColour(220, 220, 220);
-    }
-
-    // Draw check mark first because it is likely to overdraw the
-    // surrounding rectangle.
-    if ( state & wxSCB_STATE_CHECKED )
-    {
-        wxRect r2(r.x+wxPG_CHECKMARK_XADJ,
-                  r.y+wxPG_CHECKMARK_YADJ,
-                  r.width+wxPG_CHECKMARK_WADJ,
-                  r.height+wxPG_CHECKMARK_HADJ);
-    #if wxPG_CHECKMARK_DEFLATE
-        r2.Deflate(wxPG_CHECKMARK_DEFLATE);
-    #endif
-        dc.DrawCheckMark(r2);
-
-        // This would draw a simple cross check mark.
-        // dc.DrawLine(r.x,r.y,r.x+r.width-1,r.y+r.height-1);
-        // dc.DrawLine(r.x,r.y+r.height-1,r.x+r.width-1,r.y);
-    }
-
-    if ( !(state & wxSCB_STATE_BOLD) )
-    {
-        // Pen for thin rectangle.
-        dc.SetPen(useCol);
-    }
-    else
-    {
-        // Pen for bold rectangle.
-        wxPen linepen(useCol,2,wxSOLID);
-        linepen.SetJoin(wxJOIN_MITER); // This prevents round edges.
-        dc.SetPen(linepen);
-        r.x++;
-        r.y++;
-        r.width--;
-        r.height--;
-    }
-
-    dc.SetBrush(*wxTRANSPARENT_BRUSH);
-
-    dc.DrawRectangle(r);
-    dc.SetPen(*wxTRANSPARENT_PEN);
+    wxRendererNative::Get().DrawCheckBox(win, dc, r, state & wxSCB_STATE_CHECKED ? wxCONTROL_CHECKED : wxCONTROL_FOCUSED);
 }
 
 //
@@ -1552,7 +1507,7 @@
          GetFont().GetWeight() == wxBOLD )
         state |= wxSCB_STATE_BOLD;
 
-    DrawSimpleCheckBox(dc, rect, m_boxHeight, state);
+    DrawSimpleCheckBox(this, dc, rect, m_boxHeight, state);
 }
 
 void wxSimpleCheckBox::OnLeftClick( wxMouseEvent& event )
@@ -1657,7 +1612,7 @@
         state |= wxSCB_STATE_UNSPECIFIED;
     }
 
-    DrawSimpleCheckBox(dc, rect, dc.GetCharHeight(), state);
+    DrawSimpleCheckBox(property->GetGrid(), dc, rect, dc.GetCharHeight(), state);
 }
 
 void wxPGCheckBoxEditor::UpdateControl( wxPGProperty* property,
