--- D:\src\wxcvs\wxWidgets\src\msw\listctrl.cpp	Wed Jun 15 13:20:57 2005
+++ D:\src\wxWidgets-2.6.0\src\msw\listctrl.cpp	Wed Jun 22 14:26:16 2005
@@ -2810,7 +2723,10 @@
             //
             // we don't use LVCFMT_COL_HAS_IMAGES because it doesn't seem to
             // make any difference in my tests -- but maybe we should?
-            lvCol.fmt |= LVCFMT_BITMAP_ON_RIGHT | LVCFMT_IMAGE;
+            lvCol.fmt |= LVCFMT_BITMAP_ON_RIGHT;
+
+			 if ( item.m_image != -1 )
+				 lvCol.fmt |= LVCFMT_IMAGE;
 
             lvCol.iImage = item.m_image;
         }

 	  	 
