Index: src/gtk/textctrl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/gtk/textctrl.cpp,v
retrieving revision 1.188
diff -u -B -b -u -2 -r1.188 textctrl.cpp
--- src/gtk/textctrl.cpp	2004/11/21 12:21:14	1.188
+++ src/gtk/textctrl.cpp	2004/11/21 15:10:38
@@ -552,4 +552,7 @@
 #endif
         GtkTextBuffer *text_buffer = gtk_text_view_get_buffer( GTK_TEXT_VIEW(m_text) );
+        if(gtk_text_buffer_get_char_count(text_buffer) != 0)
+            IgnoreNextTextUpdate();
+
         gtk_text_buffer_set_text( text_buffer, buffer, strlen(buffer) );
 

 	  	 
