Index: src/aui/auibar.cpp
===================================================================
--- src/aui/auibar.cpp	(revision 69590)
+++ src/aui/auibar.cpp	(working copy)
@@ -2679,7 +2679,7 @@
 
     if (m_actionItem && hitItem == m_actionItem)
     {
-        if (hitItem->m_kind == wxITEM_NORMAL)
+        //if (hitItem->m_kind == wxITEM_NORMAL)
         {
             wxAuiToolBarEvent e(wxEVT_COMMAND_AUITOOLBAR_RIGHT_CLICK, m_actionItem->m_toolId);
             e.SetEventObject(this);
