--- wxWindows/samples/dnd/dnd.cpp	2003-06-09 11:13:11.000000000 +0200
+++ wxWin/samples/dnd/dnd.cpp	2003-08-21 13:54:35.000000000 +0200
@@ -1725,3 +1725,3 @@
 
-void DnDShapeFrame::OnEditShape(wxCommandEvent& event)
+void DnDShapeFrame::OnEditShape(wxCommandEvent& WXUNUSED(event))
 {
@@ -1739,3 +1739,3 @@
 
-void DnDShapeFrame::OnNewShape(wxCommandEvent& event)
+void DnDShapeFrame::OnNewShape(wxCommandEvent& WXUNUSED(event))
 {
@@ -1746,3 +1746,3 @@
 
-void DnDShapeFrame::OnClearShape(wxCommandEvent& event)
+void DnDShapeFrame::OnClearShape(wxCommandEvent& WXUNUSED(event))
 {
@@ -1751,3 +1751,3 @@
 
-void DnDShapeFrame::OnCopyShape(wxCommandEvent& event)
+void DnDShapeFrame::OnCopyShape(wxCommandEvent& WXUNUSED(event))
 {
@@ -1767,3 +1767,3 @@
 
-void DnDShapeFrame::OnPasteShape(wxCommandEvent& event)
+void DnDShapeFrame::OnPasteShape(wxCommandEvent& WXUNUSED(event))
 {

 	  	 
