diff -bu1 -r wxWindows/./include/wx/button.h wxWin/./include/wx/button.h
--- wxWindows/./include/wx/button.h	2004-01-15 14:48:12.000000000 +0100
+++ wxWin/./include/wx/button.h	2004-02-11 16:20:37.000000000 +0100
@@ -16,11 +16,7 @@
 
-#if wxUSE_BUTTON
-
 // ----------------------------------------------------------------------------
-// wxButton flags
+// wxButton flags shared with orher classes
 // ----------------------------------------------------------------------------
 
-// These two flags are obsolete
-#define wxBU_NOAUTODRAW      0x0000
-#define wxBU_AUTODRAW        0x0004
+#if wxUSE_TOGGLEBTN || wxUSE_BUTTON
 
@@ -32,2 +28,14 @@
 
+#endif
+
+#if wxUSE_BUTTON
+
+// ----------------------------------------------------------------------------
+// wxButton specific flags
+// ----------------------------------------------------------------------------
+
+// These two flags are obsolete
+#define wxBU_NOAUTODRAW      0x0000
+#define wxBU_AUTODRAW        0x0004
+
 // by default, the buttons will be created with some (system dependent)
diff -bu1 -r wxWindows/./include/wx/chkconf.h wxWin/./include/wx/chkconf.h
--- wxWindows/./include/wx/chkconf.h	2004-02-02 14:16:27.000000000 +0100
+++ wxWin/./include/wx/chkconf.h	2004-02-11 22:15:21.000000000 +0100
@@ -951,2 +951,13 @@
 
+#if wxUSE_BMPBUTTON
+#    if !wxUSE_BUTTON
+#        ifdef wxABORT_ON_CONFIG_ERROR
+#            error "wxUSE_BUTTON must be set for wxUSE_BMPBUTTON"
+#        else
+#            undef wxUSE_BUTTON
+#            define wxUSE_BUTTON 1
+#        endif
+#    endif
+#endif /* wxUSE_BMPBUTTON */
+
 #if wxUSE_NOTEBOOK || wxUSE_LISTBOOK
@@ -1151,3 +1162,3 @@
            !defined(wxUSE_TEXTCTRL) || \
-           !defined(wxUSE_BUTTON) || \
+           !defined(wxUSE_LISTCTRL) || \
            !defined(wxUSE_CHECKBOX) || \
@@ -1206,2 +1217,13 @@
 
+#if wxUSE_HELP
+#   if !wxUSE_BMPBUTTON
+#       ifdef wxABORT_ON_CONFIG_ERROR
+#           error "wxUSE_HELP require wxUSE_BMPBUTTON"
+#       else
+#           undef wxUSE_BMPBUTTON
+#           define wxUSE_BMPBUTTON 1
+#       endif
+#   endif
+#endif /* wxUSE_HELP */
+
 #if wxUSE_WXHTML_HELP
@@ -1276,2 +1298,21 @@
 
+#if !wxUSE_BUTTON
+#   if wxUSE_PROGRESSDLG || \
+       wxUSE_FONTDLG || \
+       wxUSE_FILEDLG || \
+       wxUSE_CHOICEDLG || \
+       wxUSE_NUMBERDLG || \
+       wxUSE_TEXTDLG || \
+       wxUSE_DIRDLG || \
+       wxUSE_STARTUP_TIPS || \
+       wxUSE_WIZARDDLG
+#       ifdef wxABORT_ON_CONFIG_ERROR
+#           error "Common and generic dialogs must be compiled with wxUSE_BUTTON"
+#       else
+#           undef wxUSE_BUTTON
+#           define wxUSE_BUTTON 1
+#       endif
+#   endif
+#endif /* wxUSE_PROGRESSDLG */
+
 #if !wxUSE_TOOLBAR
@@ -1342,5 +1383,5 @@
 #if wxUSE_LOG_DIALOG
-#    if !wxUSE_LISTCTRL
+#    if !wxUSE_LISTCTRL || !wxUSE_BUTTON
 #        ifdef wxABORT_ON_CONFIG_ERROR
-#            error "wxUSE_LOG_DIALOG requires wxUSE_LISTCTRL"
+#            error "wxUSE_LOG_DIALOG requires wxUSE_LISTCTRL and wxUSE_BUTTON"
 #        else
@@ -1348,2 +1389,4 @@
 #            define wxUSE_LISTCTRL 1
+#            undef wxUSE_BUTTON
+#            define wxUSE_BUTTON 1
 #        endif
@@ -1381,2 +1424,13 @@
 
+#if wxUSE_FONTMAP
+#    if !wxUSE_CHOICEDLG
+#        ifdef wxABORT_ON_CONFIG_ERROR
+#            error "wxUSE_FONTMAP require wxUSE_CHOICEDLG"
+#        else
+#            undef wxUSE_CHOICEDLG
+#            define wxUSE_CHOICEDLG 1
+#        endif
+#    endif
+#endif /* wxUSE_FONTMAP */
+
 #endif /* wxUSE_GUI */
diff -bu1 -r wxWindows/./include/wx/choicdlg.h wxWin/./include/wx/choicdlg.h
--- wxWindows/./include/wx/choicdlg.h	1999-08-11 14:53:32.000000000 +0200
+++ wxWin/./include/wx/choicdlg.h	2004-02-11 18:59:55.000000000 +0100
@@ -3,2 +3,4 @@
 
+#if wxUSE_CHOICEDLG
+
 #include "wx/generic/choicdgg.h"
@@ -6,2 +8,4 @@
 #endif
+
+#endif
     // _WX_CHOICDLG_H_BASE_
diff -bu1 -r wxWindows/./include/wx/dataobj.h wxWin/./include/wx/dataobj.h
--- wxWindows/./include/wx/dataobj.h	2003-08-09 15:37:10.000000000 +0200
+++ wxWin/./include/wx/dataobj.h	2004-02-11 21:05:57.000000000 +0100
@@ -287,3 +287,3 @@
     wxDataObjectSimple *GetObject(const wxDataFormat& format) const;
-#if defined(__WXMSW__)
+#if defined(__WXMSW__) && wxUSE_OLE
     virtual const void* GetSizeFromBuffer( const void* buffer, size_t* size,
diff -bu1 -r wxWindows/./include/wx/dnd.h wxWin/./include/wx/dnd.h
--- wxWindows/./include/wx/dnd.h	2003-08-09 15:37:11.000000000 +0200
+++ wxWin/./include/wx/dnd.h	2004-02-11 20:21:03.000000000 +0100
@@ -20,3 +20,3 @@
 
-#if wxUSE_DRAG_AND_DROP
+#if wxUSE_DRAG_AND_DROP && (wxUSE_OLE || !defined(__WXMSW__))
 
diff -bu1 -r wxWindows/./include/wx/log.h wxWin/./include/wx/log.h
--- wxWindows/./include/wx/log.h	2004-02-11 09:36:18.000000000 +0100
+++ wxWin/./include/wx/log.h	2004-02-11 21:39:25.000000000 +0100
@@ -21,2 +21,20 @@
 // ----------------------------------------------------------------------------
+// common constants for use in wxUSE_LOG/!wxUSE_LOG
+// ----------------------------------------------------------------------------
+
+// the trace masks have been superceded by symbolic trace constants, they're
+// for compatibility only andwill be removed soon - do NOT use them
+
+// meaning of different bits of the trace mask (which allows selectively
+// enable/disable some trace messages)
+#define wxTraceMemAlloc 0x0001  // trace memory allocation (new/delete)
+#define wxTraceMessages 0x0002  // trace window messages/X callbacks
+#define wxTraceResAlloc 0x0004  // trace GDI resource allocation
+#define wxTraceRefCount 0x0008  // trace various ref counting operations
+
+#ifdef  __WXMSW__
+    #define wxTraceOleCalls 0x0100  // OLE interface calls
+#endif
+
+// ----------------------------------------------------------------------------
 // types
@@ -94,16 +112,2 @@
 
-// the trace masks have been superceded by symbolic trace constants, they're
-// for compatibility only andwill be removed soon - do NOT use them
-
-// meaning of different bits of the trace mask (which allows selectively
-// enable/disable some trace messages)
-#define wxTraceMemAlloc 0x0001  // trace memory allocation (new/delete)
-#define wxTraceMessages 0x0002  // trace window messages/X callbacks
-#define wxTraceResAlloc 0x0004  // trace GDI resource allocation
-#define wxTraceRefCount 0x0008  // trace various ref counting operations
-
-#ifdef  __WXMSW__
-    #define wxTraceOleCalls 0x0100  // OLE interface calls
-#endif
-
 #include "wx/iosfwrap.h"
@@ -445,5 +449,2 @@
                                      ...) ATTRIBUTE_PRINTF_2
-#define DECLARE_LOG_FUNCTION2(level, arg)                                   \
-    DECLARE_LOG_FUNCTION2_EXP(level, arg, WXDLLIMPEXP_BASE)
-
 #else // !wxUSE_LOG
@@ -455,3 +456,3 @@
 inline void wxLog##level(const wxChar *szFormat, ...) { }
-#define DECLARE_LOG_FUNCTION2(level, arg)                                   \
+#define DECLARE_LOG_FUNCTION2_EXP(level, arg, expdecl)                      \
 inline void wxVLog##level(arg, const wxChar *szFormat,                      \
@@ -476,2 +477,5 @@
 
+#define DECLARE_LOG_FUNCTION2(level, arg)                                   \
+    DECLARE_LOG_FUNCTION2_EXP(level, arg, WXDLLIMPEXP_BASE)
+
 // a generic function for all levels (level is passes as parameter)
@@ -494,2 +498,3 @@
     // specify the frame to which the output should go
+    class wxFrame;
     DECLARE_LOG_FUNCTION2_EXP(Status, wxFrame *pFrame, WXDLLIMPEXP_CORE);
diff -bu1 -r wxWindows/./include/wx/msw/ole/dataobj.h wxWin/./include/wx/msw/ole/dataobj.h
--- wxWindows/./include/wx/msw/ole/dataobj.h	2003-01-03 00:37:50.000000000 +0100
+++ wxWin/./include/wx/msw/ole/dataobj.h	2004-02-11 21:04:51.000000000 +0100
@@ -53,2 +53,3 @@
 public:
+#if wxUSE_OLE
     virtual const void* GetSizeFromBuffer( const void* buffer, size_t* size,
@@ -58,2 +59,3 @@
     virtual size_t GetBufferOffset( const wxDataFormat& format );
+#endif
 private:
diff -bu1 -r wxWindows/./include/wx/msw/window.h wxWin/./include/wx/msw/window.h
--- wxWindows/./include/wx/msw/window.h	2004-01-15 14:48:25.000000000 +0100
+++ wxWin/./include/wx/msw/window.h	2004-02-11 20:35:06.000000000 +0100
@@ -129,3 +129,3 @@
 
-#if wxUSE_DRAG_AND_DROP
+#if wxUSE_DRAG_AND_DROP && wxUSE_OLE
     virtual void SetDropTarget( wxDropTarget *dropTarget );
diff -bu1 -r wxWindows/./include/wx/window.h wxWin/./include/wx/window.h
--- wxWindows/./include/wx/window.h	2004-01-15 14:48:12.000000000 +0100
+++ wxWin/./include/wx/window.h	2004-02-11 20:37:48.000000000 +0100
@@ -42,3 +42,3 @@
 
-#if wxUSE_ACCESSIBILITY
+#if wxUSE_ACCESSIBILITY && (wxUSE_OLE || !defined(__WXMSW__))
 #include "wx/access.h"
@@ -71,3 +71,3 @@
 
-#if wxUSE_ACCESSIBILITY
+#if wxUSE_ACCESSIBILITY && (wxUSE_OLE || !defined(__WXMSW__))
 class WXDLLEXPORT wxAccessible;
@@ -791,3 +791,3 @@
     // -------------
-#if wxUSE_DRAG_AND_DROP
+#if wxUSE_DRAG_AND_DROP && (wxUSE_OLE || !defined(__WXMSW__))
         // set/retrieve the drop target associated with this window (may be
@@ -860,3 +860,3 @@
     // ----------------------
-#if wxUSE_ACCESSIBILITY
+#if wxUSE_ACCESSIBILITY && (wxUSE_OLE || !defined(__WXMSW__))
     // Override to create a specific accessible object.
@@ -1018,3 +1018,3 @@
 
-#if wxUSE_ACCESSIBILITY
+#if wxUSE_ACCESSIBILITY && (wxUSE_OLE || !defined(__WXMSW__))
     wxAccessible*       m_accessible;
@@ -1237,3 +1237,3 @@
 
-#if wxUSE_ACCESSIBILITY
+#if wxUSE_ACCESSIBILITY && (wxUSE_OLE || !defined(__WXMSW__))
 // ----------------------------------------------------------------------------
diff -bu1 -r wxWindows/./src/common/accesscmn.cpp wxWin/./src/common/accesscmn.cpp
--- wxWindows/./src/common/accesscmn.cpp	2003-08-09 15:45:43.000000000 +0200
+++ wxWin/./src/common/accesscmn.cpp	2004-02-11 20:40:33.000000000 +0100
@@ -24,3 +24,3 @@
 
-#if wxUSE_ACCESSIBILITY
+#if wxUSE_ACCESSIBILITY && (wxUSE_OLE || !defined(__WXMSW__))
 
diff -bu1 -r wxWindows/./src/common/dobjcmn.cpp wxWin/./src/common/dobjcmn.cpp
--- wxWindows/./src/common/dobjcmn.cpp	2003-08-09 15:45:45.000000000 +0200
+++ wxWin/./src/common/dobjcmn.cpp	2004-02-11 21:04:38.000000000 +0100
@@ -145,3 +145,3 @@
 
-#if defined(__WXMSW__)
+#if defined(__WXMSW__) && wxUSE_OLE
 
@@ -390,3 +390,3 @@
 
-#if wxUSE_DRAG_AND_DROP
+#if wxUSE_DRAG_AND_DROP && wxUSE_OLE
 
diff -bu1 -r wxWindows/./src/common/strconv.cpp wxWin/./src/common/strconv.cpp
--- wxWindows/./src/common/strconv.cpp	2004-02-09 08:09:17.000000000 +0100
+++ wxWin/./src/common/strconv.cpp	2004-02-11 19:34:08.000000000 +0100
@@ -1115,3 +1115,5 @@
 extern WXDLLIMPEXP_BASE long wxCharsetToCodepage(const wxChar *charset);
+#if wxUSE_FONTMAP
 extern WXDLLIMPEXP_BASE long wxEncodingToCodepage(wxFontEncoding encoding);
+#endif
 
@@ -1130,2 +1132,3 @@
 
+#if wxUSE_FONTMAP
     wxMBConv_win32(wxFontEncoding encoding)
@@ -1134,2 +1137,3 @@
     }
+#endif
 
@@ -1485,4 +1489,8 @@
     {
+#if wxUSE_FONTMAP
         wxMBConv_win32 *conv = m_name ? new wxMBConv_win32(m_name)
                                       : new wxMBConv_win32(m_encoding);
+#else
+        wxMBConv_win32 *conv = new wxMBConv_win32(m_name);
+#endif
         if ( conv->IsOk() )
diff -bu1 -r wxWindows/./src/common/wincmn.cpp wxWin/./src/common/wincmn.cpp
--- wxWindows/./src/common/wincmn.cpp	2003-11-14 13:55:25.000000000 +0100
+++ wxWin/./src/common/wincmn.cpp	2004-02-11 20:47:37.000000000 +0100
@@ -59,3 +59,3 @@
 
-#if wxUSE_ACCESSIBILITY
+#if wxUSE_ACCESSIBILITY && (wxUSE_OLE || !defined(__WXMSW__))
     #include "wx/access.h"
@@ -183,3 +183,3 @@
 
-#if wxUSE_ACCESSIBILITY
+#if wxUSE_ACCESSIBILITY && (wxUSE_OLE || !defined(__WXMSW__))
     m_accessible = NULL;
@@ -315,3 +315,3 @@
 
-#if wxUSE_ACCESSIBILITY
+#if wxUSE_ACCESSIBILITY && (wxUSE_OLE || !defined(__WXMSW__))
     delete m_accessible;
@@ -1991,3 +1991,3 @@
 
-#if wxUSE_ACCESSIBILITY
+#if wxUSE_ACCESSIBILITY && (wxUSE_OLE || !defined(__WXMSW__))
 void wxWindowBase::SetAccessible(wxAccessible* accessible)
@@ -2207,3 +2207,3 @@
 
-#if wxUSE_ACCESSIBILITY
+#if wxUSE_ACCESSIBILITY && (wxUSE_OLE || !defined(__WXMSW__))
 // ----------------------------------------------------------------------------
@@ -2365,2 +2365,3 @@
     // control or window.
+#if wxUSE_BUTTON
     if (GetWindow()->IsKindOf(CLASSINFO(wxButton)))
@@ -2368,2 +2369,3 @@
     else
+#endif
         title = GetWindow()->GetName();
@@ -2475,2 +2477,3 @@
 
+#if wxUSE_HELP
     wxString ht(GetWindow()->GetHelpText());
@@ -2481,2 +2484,6 @@
     }
+#else
+    wxUnusedVar(description);
+#endif
+
     return wxACC_NOT_IMPLEMENTED;
@@ -2491,2 +2498,3 @@
 
+#if wxUSE_HELP
     wxString ht(GetWindow()->GetHelpText());
@@ -2497,2 +2505,6 @@
     }
+#else
+    wxUnusedVar(helpText);
+#endif
+
     return wxACC_NOT_IMPLEMENTED;
diff -bu1 -r wxWindows/./src/generic/filedlgg.cpp wxWin/./src/generic/filedlgg.cpp
--- wxWindows/./src/generic/filedlgg.cpp	2004-02-09 08:09:17.000000000 +0100
+++ wxWin/./src/generic/filedlgg.cpp	2004-02-11 11:36:34.000000000 +0100
@@ -703,3 +703,3 @@
 
-void wxFileCtrl::OnListDeleteAllItems( wxListEvent &event )
+void wxFileCtrl::OnListDeleteAllItems( wxListEvent &WXUNUSED(event) )
 {
diff -bu1 -r wxWindows/./src/generic/logg.cpp wxWin/./src/generic/logg.cpp
--- wxWindows/./src/generic/logg.cpp	2004-02-09 08:09:22.000000000 +0100
+++ wxWin/./src/generic/logg.cpp	2004-02-11 21:54:06.000000000 +0100
@@ -193,2 +193,4 @@
 
+#if wxUSE_LOG
+
 // accepts an additional argument which tells to which frame the output should
@@ -1161,2 +1163,4 @@
 
+#endif // wxUSE_LOG
+
 #if wxUSE_LOG && wxUSE_TEXTCTRL
diff -bu1 -r wxWindows/./src/html/helpdata.cpp wxWin/./src/html/helpdata.cpp
--- wxWindows/./src/html/helpdata.cpp	2003-10-12 04:46:17.000000000 +0200
+++ wxWin/./src/html/helpdata.cpp	2004-02-11 19:29:43.000000000 +0100
@@ -545,2 +545,3 @@
 
+#if wxUSE_FONTMAP
     // Convert encoding, if neccessary:
@@ -561,2 +562,7 @@
     }
+#else
+    wxUnusedVar(IndexOld);
+    wxUnusedVar(ContentsOld);
+    wxASSERT_MSG(encoding == wxFONTENCODING_SYSTEM, wxT("Encoding can't be converted"));
+#endif
 
@@ -644,4 +650,8 @@
     wxFontEncoding enc;
+#if wxUSE_FONTMAP
     if (charset == wxEmptyString) enc = wxFONTENCODING_SYSTEM;
     else enc = wxFontMapper::Get()->CharsetToEncoding(charset);
+#else
+    enc = wxFONTENCODING_SYSTEM;
+#endif
     bool rtval = AddBookParam(*fi, enc,
diff -bu1 -r wxWindows/./src/msw/clipbrd.cpp wxWin/./src/msw/clipbrd.cpp
--- wxWindows/./src/msw/clipbrd.cpp	2004-01-27 18:47:28.000000000 +0100
+++ wxWin/./src/msw/clipbrd.cpp	2004-02-11 19:53:34.000000000 +0100
@@ -664,3 +664,3 @@
             // TODO
-            wxLogError("Not implemented because wxMetafileDataObject does not contain width and height values.");
+            wxLogError(_T("Not implemented because wxMetafileDataObject does not contain width and height values."));
             return FALSE;
diff -bu1 -r wxWindows/./src/msw/display.cpp wxWin/./src/msw/display.cpp
--- wxWindows/./src/msw/display.cpp	2003-08-09 15:46:28.000000000 +0200
+++ wxWin/./src/msw/display.cpp	2004-02-11 21:33:39.000000000 +0100
@@ -242,4 +242,6 @@
 {
+#if wxUSE_LOG
     // suppress the errors if ddraw.dll is not found
     wxLog::EnableLogging(false);
+#endif
 
@@ -247,3 +249,5 @@
 
+#if wxUSE_LOG
     wxLog::EnableLogging(true);
+#endif
 
diff -bu1 -r wxWindows/./src/msw/notebook.cpp wxWin/./src/msw/notebook.cpp
--- wxWindows/./src/msw/notebook.cpp	2004-02-09 08:09:28.000000000 +0100
+++ wxWin/./src/msw/notebook.cpp	2004-02-11 16:09:11.000000000 +0100
@@ -972,3 +972,9 @@
     // switch it into ownerdraw mode
-    if (window->IsKindOf(CLASSINFO(wxButton)) && !window->IsKindOf(CLASSINFO(wxBitmapButton)))
+#if wxUSE_BUTTON
+    if (
+        window->IsKindOf(CLASSINFO(wxButton))
+#if wxUSE_BMPBUTTON
+        && !window->IsKindOf(CLASSINFO(wxBitmapButton))
+#endif
+       )
         // This is essential, otherwise you'll see dark grey
@@ -976,3 +982,5 @@
         ((wxButton*)window)->wxControl::SetBackgroundColour(colour);
-    else if (window->IsKindOf(CLASSINFO(wxStaticText)) ||
+    else
+#endif 
+         if (window->IsKindOf(CLASSINFO(wxStaticText)) ||
              window->IsKindOf(CLASSINFO(wxStaticBox)) ||
@@ -982,3 +990,5 @@
              window->IsKindOf(CLASSINFO(wxCheckBox)) ||
+#if wxUSE_BMPBUTTON
              window->IsKindOf(CLASSINFO(wxBitmapButton)) ||
+#endif 
              window->IsKindOf(CLASSINFO(wxSlider)) ||
diff -bu1 -r wxWindows/./src/msw/tglbtn.cpp wxWin/./src/msw/tglbtn.cpp
--- wxWindows/./src/msw/tglbtn.cpp	2003-05-27 01:57:51.000000000 +0200
+++ wxWin/./src/msw/tglbtn.cpp	2004-02-11 16:17:01.000000000 +0100
@@ -153,2 +153,3 @@
 
+#if wxUSE_BUTTON
    wxSize sz = wxButton::GetDefaultSize();
@@ -158,2 +159,5 @@
        sz.y = hBtn;
+#else
+   wxSize sz(wBtn, hBtn);
+#endif
 
diff -bu1 -r wxWindows/./src/msw/window.cpp wxWin/./src/msw/window.cpp
--- wxWindows/./src/msw/window.cpp	2004-02-11 09:37:11.000000000 +0100
+++ wxWin/./src/msw/window.cpp	2004-02-11 20:39:32.000000000 +0100
@@ -61,3 +61,3 @@
 
-#if wxUSE_ACCESSIBILITY
+#if wxUSE_ACCESSIBILITY && wxUSE_OLE
     #include "wx/access.h"
@@ -1347,3 +1347,3 @@
 
-#if wxUSE_DRAG_AND_DROP
+#if wxUSE_DRAG_AND_DROP && wxUSE_OLE
 void wxWindowMSW::SetDropTarget(wxDropTarget *pDropTarget)
@@ -1898,4 +1898,4 @@
                             else // no default button
-                            {
 #endif // wxUSE_BUTTON
+                            {
                                 // this is a quick and dirty test for a text
@@ -2339,2 +2339,3 @@
             {
+#if wxUSE_LISTCTRL
                 // Don't call the wx handlers in this case
@@ -2342,2 +2343,3 @@
                     break;
+#endif
 
@@ -2694,3 +2696,3 @@
 
-#if wxUSE_HOTKEY
+#if wxUSE_HOTKEY && wxUSE_ACCEL
         case WM_HOTKEY:
@@ -2816,3 +2818,3 @@
 
-#if wxUSE_ACCESSIBILITY
+#if wxUSE_ACCESSIBILITY && wxUSE_OLE
         case WM_GETOBJECT:
@@ -3357,3 +3359,3 @@
     // delete our drop target if we've got one
-#if wxUSE_DRAG_AND_DROP
+#if wxUSE_DRAG_AND_DROP && wxUSE_OLE
     if ( m_dropTarget != NULL )
@@ -5759,2 +5761,4 @@
 
+#if wxUSE_ACCEL
+
 bool wxWindowMSW::HandleHotKey(WXWPARAM wParam, WXLPARAM lParam)
@@ -5775,2 +5779,4 @@
 
+#endif
+
 #endif // wxUSE_HOTKEY
diff -bu1 -r wxWindows/./src/univ/themes/win32.cpp wxWin/./src/univ/themes/win32.cpp
--- wxWindows/./src/univ/themes/win32.cpp	2004-01-13 15:48:35.000000000 +0100
+++ wxWin/./src/univ/themes/win32.cpp	2004-02-11 12:15:38.000000000 +0100
@@ -4545,3 +4545,5 @@
     wxTopLevelWindow         *m_wnd;
+#if wxUSE_ACCEL
     wxAcceleratorTable        m_oldAccelTable;
+#endif
 };
@@ -4562,2 +4564,3 @@
     
+#if wxUSE_ACCEL
     // VS: This code relies on using generic implementation of 
@@ -4569,2 +4572,3 @@
     m_wnd->SetAcceleratorTable(table);
+#endif
 }
@@ -4575,3 +4579,5 @@
     {
+#if wxUSE_ACCEL
         m_wnd->SetAcceleratorTable(m_oldAccelTable);
+#endif
         m_wnd->RemoveEventHandler(this); 
@@ -4597,6 +4603,10 @@
 
+#if wxUSE_ACCEL
     wxAcceleratorTable table = *m_wnd->GetAcceleratorTable();
     m_wnd->SetAcceleratorTable(wxNullAcceleratorTable);
+#endif
     m_inputHnd->PopupSystemMenu(m_wnd, pt);
+#if wxUSE_ACCEL
     m_wnd->SetAcceleratorTable(table);
+#endif
 }

 	  	 
