diff --git a/src/osx/toplevel_osx.cpp b/src/osx/toplevel_osx.cpp
index 16c88a8..57146f5 100644
--- a/src/osx/toplevel_osx.cpp
+++ b/src/osx/toplevel_osx.cpp
@@ -76,7 +76,11 @@ bool wxTopLevelWindowMac::Create(wxWindow *parent,
     int w = WidthDefault(size.x);
     int h = HeightDefault(size.y);
 
-    if ( !wxNonOwnedWindow::Create(parent, id, pos, wxSize(w,h), style, name) )
+    wxRect display = wxGetClientDisplayRect() ;
+    int x = pos.x == wxDefaultPosition.x ? wxMax( display.x, (display.width - w) / 2 ) : pos.x;
+    int y = pos.y == wxDefaultPosition.y ? wxMax( display.y, (display.height - h) / 3 ) : pos.y;
+
+    if ( !wxNonOwnedWindow::Create(parent, id, wxPoint(x, y), wxSize(w,h), style, name) )
         return false;
 
     wxWindow::SetLabel( title ) ;
