Index: src/xrc/xh_text.cpp
===================================================================
--- src/xrc/xh_text.cpp	(revision 76659)
+++ src/xrc/xh_text.cpp	(working copy)
@@ -69,7 +69,9 @@
     if (HasParam(wxT("maxlength")))
         text->SetMaxLength(GetLong(wxT("maxlength")));
 
-    text->SetHint(GetText(wxS("hint")));
+    wxString hint = GetText(wxS("hint"));
+    if (!hint.empty())
+        text->SetHint(hint);
 
     return text;
 }
