Index: include/wx/generic/dataview.h
===================================================================
--- include/wx/generic/dataview.h	(revision 56208)
+++ include/wx/generic/dataview.h	(working copy)
@@ -391,6 +391,7 @@
     {
         //No sorting column at start, I think
         m_sortingColumn = NULL;
+        m_headerArea = NULL;
         Init();
     }
 
@@ -401,6 +402,7 @@
              : wxScrollHelperNative(this)
     {
         m_sortingColumn = NULL;
+        m_headerArea = NULL;
         Create(parent, id, pos, size, style, validator );
     }
 
