Index: src/msw/stackwalk.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/msw/stackwalk.cpp,v
retrieving revision 1.10
diff -u -r1.10 stackwalk.cpp
--- src/msw/stackwalk.cpp	2006/11/12 16:35:51	1.10
+++ src/msw/stackwalk.cpp	2007/06/15 11:58:59
@@ -52,11 +52,11 @@
 
     // get the name of the function for this stack frame entry
     static const size_t MAX_NAME_LEN = 1024;
-    BYTE symbolBuffer[sizeof(SYMBOL_INFO) + MAX_NAME_LEN];
+    BYTE symbolBuffer[sizeof(SYMBOL_INFO) + (MAX_NAME_LEN * sizeof(wxChar))];
     wxZeroMemory(symbolBuffer);
 
     PSYMBOL_INFO pSymbol = (PSYMBOL_INFO)symbolBuffer;
-    pSymbol->SizeOfStruct = sizeof(symbolBuffer);
+    pSymbol->SizeOfStruct = sizeof(SYMBOL_INFO);
     pSymbol->MaxNameLen = MAX_NAME_LEN;
 
     DWORD64 symDisplacement = 0;

 	  	 
