--- C:/Documents and Settings/miche/Impostazioni locali/Temp/advprops.cpp-revBASE.svn000.tmp.cpp	Wed Nov 19 10:14:15 2008
+++ D:/libs/wxTrunk/src/propgrid/advprops.cpp	Wed Nov 19 10:14:49 2008
@@ -1530,7 +1530,10 @@
                               0,
                               0,
                               NULL,
-                              DI_COMPAT | DI_DEFAULTSIZE | DI_NORMAL
+#if !defined(__WXWINCE__)
+							  DI_COMPAT | DI_DEFAULTSIZE | 
+#endif
+							  DI_NORMAL
                             );
             #endif
             }
