Index: combotreebox.py
===================================================================
--- combotreebox.py	(Revision 76928)
+++ combotreebox.py	(Arbeitskopie)
@@ -803,6 +803,8 @@
         events = super(MSWComboTreeBox, self)._eventsToBind()
         events.append((self._tree, wx.EVT_TREE_SEL_CHANGED,
             self.OnSelectionChangedInTree))
+        events.append((self._text, wx.EVT_KEY_UP,
+            self.OnKeyUp))
         return events
 
     def OnSelectionChangedInTree(self, event):
@@ -815,6 +817,18 @@
                 self.SetValue(selectedValue)
         event.Skip()
 
+    def OnKeyDown(self, keyEvent):
+        return
+ 
+    def OnKeyUp(self, keyEvent):
+        obj = keyEvent.GetEventObject()        
+        if obj._keyShouldNavigate(keyEvent):
+            obj._navigateUpOrDown(keyEvent)
+        elif obj._keyShouldPopUpTree(keyEvent):
+            obj.Popup()
+        else:
+            keyEvent.Skip()
+
     def _keyShouldPopUpTree(self, keyEvent):
         return super(MSWComboTreeBox, self)._keyShouldPopUpTree(keyEvent) or \
             (keyEvent.GetKeyCode() == wx.WXK_F4 and not keyEvent.HasModifiers()) or \
