Index: src/os2/dc.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/os2/dc.cpp,v
retrieving revision 1.85
diff -u -r1.85 dc.cpp
--- src/os2/dc.cpp	2007/04/07 12:21:14	1.85
+++ src/os2/dc.cpp	2007/04/28 09:16:21
@@ -2151,7 +2151,7 @@
 , wxCoord*                          pvY
 , wxCoord*                          pvDescent
 , wxCoord*                          pvExternalLeading
-, wxFont*                           pTheFont
+, const wxFont*                     pTheFont
 ) const
 {
     POINTL                          avPoint[TXTBOX_COUNT];
Index: include/wx/os2/dc.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/os2/dc.h,v
retrieving revision 1.29
diff -u -r1.29 dc.h
--- include/wx/os2/dc.h	2006/09/05 20:46:27	1.29
+++ include/wx/os2/dc.h	2007/04/28 09:16:22
@@ -129,7 +129,7 @@
                                     ,wxCoord*        pY
                                     ,wxCoord*        pDescent = NULL
                                     ,wxCoord*        pExternalLeading = NULL
-                                    ,wxFont*         pTheFont = NULL
+                                    ,const wxFont*   pTheFont = NULL
                                    ) const;
     virtual bool    CanDrawBitmap(void) const;
     virtual bool    CanGetTextExtent(void) const;

 	  	 
