cvs -z9 -Q diff -R -U2 include/wx/object.h 
Index: include/wx/object.h
===================================================================
RCS file: /home/wxcvs/wxWindows/include/wx/object.h,v
retrieving revision 1.71
diff -U2 -r1.71 object.h
--- include/wx/object.h	10 May 2002 19:01:47 -0000	1.71
+++ include/wx/object.h	19 Jul 2002 17:19:06 -0000
@@ -75,4 +75,20 @@
         { sm_first = this; }
 
+	~wxClassInfo() {
+		if(sm_first == this) {
+			sm_first = m_next;
+		}
+		else {
+			wxClassInfo * info = sm_first;
+			while(info) {
+				if(info->m_next == this) {
+					info->m_next = m_next;
+					break;
+				}
+				info = info->m_next;
+			}
+		}
+	}
+
     wxObject *CreateObject() { return m_objectConstructor ? (*m_objectConstructor)() : 0; }
 


 	  	 
