diff -ru -xCVS -x.#* -x*.orig -x*.rej -x*~ wxWidgets-2.6.2.orig\include\wx\list.h wxWidgets-2.6.2\include\wx\list.h
--- wxWidgets-2.6.2.orig\include\wx\list.h	Wed Jun 08 22:52:12 2005
+++ wxWidgets-2.6.2\include\wx\list.h	Tue Mar 07 15:26:55 2006
@@ -43,6 +43,7 @@
     #include <algorithm>
     #include <iterator>
     #include <list>
+    #include <deque>
     #include "wx/afterstd.h"
 #endif
 
@@ -142,19 +143,20 @@
 
 #define WX_DECLARE_LIST_XO(elT, liT, decl)                                    \
     VC6_WORKAROUND(elT, liT, decl)                                            \
-    decl liT : public std::list<elT>                                          \
+    decl liT : public std::deque<elT>                                         \
     {                                                                         \
     private:                                                                  \
         bool m_destroy;                                                       \
     private:                                                                  \
         typedef elT _WX_LIST_ITEM_TYPE_##liT;                                 \
-        static void DeleteFunction( const _WX_LIST_ITEM_TYPE_##liT X );       \
+        static void DeleteFunction( _WX_LIST_ITEM_TYPE_##liT X );             \
     public:                                                                   \
+        liT(wxKeyType WXUNUSED(keyType)) : std::deque<elT>() { }              \
         class compatibility_iterator                                          \
         {                                                                     \
         private:                                                              \
           /* Workaround for broken VC6 nested class name resolution */        \
-          typedef std::list<elT>::iterator iterator;                          \
+          typedef std::deque<elT>::iterator iterator;                         \
           friend class liT;                                                   \
         private:                                                              \
             iterator m_iter;                                                  \
@@ -205,7 +207,6 @@
         };                                                                    \
     public:                                                                   \
         liT() : m_destroy( false ) {}                                         \
-                                                                              \
         compatibility_iterator Find( const elT e ) const                      \
         {                                                                     \
           liT* _this = const_cast< liT* >( this );                            \
@@ -242,7 +243,16 @@
             { return Item( n ); }                                             \
         int IndexOf( elT e ) const                                            \
             { return Find( e ).IndexOf(); }                                   \
-                                                                              \
+        compatibility_iterator Append(long WXUNUSED(key), elT e)     \
+        {                                                                     \
+            push_back(e);                                                     \
+            return GetLast();                                                 \
+        }                                                                     \
+        compatibility_iterator Append(const wxChar* WXUNUSED(key), elT e)     \
+        {                                                                     \
+            push_back(e);                                                     \
+            return GetLast();                                                 \
+        }                                                                     \
         compatibility_iterator Append( elT e )                                \
         {                                                                     \
             push_back( e );                                                   \
@@ -294,7 +304,7 @@
         }                                                                     \
         /* Workaround for broken VC6 std::list::sort() see above */           \
         void Sort( wxSortCompareFunction compfunc )                           \
-            { sort( WX_LIST_SORTFUNCTION( elT, compfunc ) ); }                \
+            { sort( begin(), end(), WX_LIST_SORTFUNCTION( elT, compfunc ) ); }                \
         ~liT() { Clear(); }                                                   \
     }

diff -ru -xCVS -x.#* -x*.orig -x*.rej -x*~ wxWidgets-2.6.2.orig\src\common\list.cpp wxWidgets-2.6.2\src\common\list.cpp
--- wxWidgets-2.6.2.orig\src\common\list.cpp	Wed Jun 08 22:57:04 2005
+++ wxWidgets-2.6.2\src\common\list.cpp	Tue Mar 07 15:26:52 2006
@@ -756,7 +756,7 @@
     WX_DEFINE_LIST(wxObjectList);
 
 // with wxUSE_STL wxStringList contains wxString objects, not pointers
-void wxStringListBase::DeleteFunction( const wxString WXUNUSED(X) )
+void wxStringListBase::DeleteFunction( wxString WXUNUSED(X) )
 {
 }

 	  	 
