diff -uNr orig/src/aui/auibar.cpp mine/src/aui/auibar.cpp
--- orig/src/aui/auibar.cpp	2010-11-17 16:14:35.937500000 -0800
+++ mine/src/aui/auibar.cpp	2010-11-17 18:11:41.312500000 -0800
@@ -762,8 +762,9 @@
                 text = wxT(" ");
 
             wxMenuItem* m =  new wxMenuItem(&menuPopup, item.GetId(), text, item.GetShortHelp());
-
+#if wxUSE_OWNER_DRAWN
             m->SetBitmap(item.GetBitmap());
+#endif
             menuPopup.Append(m);
             items_added++;
         }
diff -uNr orig/src/aui/auibook.cpp mine/src/aui/auibook.cpp
--- orig/src/aui/auibook.cpp	2010-11-17 16:14:35.984375000 -0800
+++ mine/src/aui/auibook.cpp	2010-11-17 18:10:33.015625000 -0800
@@ -744,8 +744,10 @@
             caption = wxT(" ");
 
         wxMenuItem* item = new wxMenuItem(NULL, 1000+i, caption);
+#if wxUSE_OWNER_DRAWN
         if (page.bitmap.IsOk())
             item->SetBitmap(page.bitmap);
+#endif
         menuPopup.Append(item);
     }
 
diff -uNr orig/src/msw/menu.cpp mine/src/msw/menu.cpp
--- orig/src/msw/menu.cpp	2010-11-17 16:15:48.328125000 -0800
+++ mine/src/msw/menu.cpp	2010-11-17 17:59:08.546875000 -0800
@@ -352,13 +352,16 @@
             GetMenuBar()->RebuildAccelTable();
         }
 
+#if wxUSE_OWNER_DRAWN
         ResetMaxAccelWidth();
+#endif
     }
     //else: it is a separator, they can't have accels, nothing to do
 }
 
 #endif // wxUSE_ACCEL
 
+#if wxUSE_OWNER_DRAWN
 namespace
 {
 
@@ -404,6 +407,7 @@
 }
 
 } // anonymous namespace
+#endif // wxUSE_OWNER_DRAWN
 
 // append a new item or submenu to the menu
 bool wxMenu::DoInsertOrAppend(wxMenuItem *pItem, size_t pos)
@@ -769,7 +773,9 @@
 
         m_accels.RemoveAt(n);
 
+#if wxUSE_OWNER_DRAWN
         ResetMaxAccelWidth();
+#endif
     }
     //else: this item doesn't have an accel, nothing to do
 #endif // wxUSE_ACCEL
diff -uNr orig/src/msw/statusbar.cpp mine/src/msw/statusbar.cpp
--- orig/src/msw/statusbar.cpp	2010-11-17 16:15:48.609375000 -0800
+++ mine/src/msw/statusbar.cpp	2010-11-17 17:47:11.328125000 -0800
@@ -399,12 +399,14 @@
         // into account to make sure the text drawn by user fits inside the
         // pane. Notice that it's not the value returned by SB_GETBORDERS
         // which, at least on this Windows 2003 system, returns {0, 2, 2}
+#if wxUSE_UXTHEME
         if ( wxUxThemeEngine::GetIfActive() )
         {
             s_metrics.gripWidth = 20;
             s_metrics.textMargin = 8;
         }
         else // classic/unthemed look
+#endif
         {
             s_metrics.gripWidth = 18;
             s_metrics.textMargin = 4;
