Only in wxWidgets: .svn
diff -bru wxWidgets.orig/src/common/wincmn.cpp wxWidgets/src/common/wincmn.cpp
--- wxWidgets.orig/src/common/wincmn.cpp	2012-03-05 00:58:17.000000000 -0800
+++ wxWidgets/src/common/wincmn.cpp	2012-03-05 00:57:39.000000000 -0800
@@ -1470,8 +1470,13 @@
     // which ensures that this only happens if the user really wants it and
     // not by default which wouldn't make any sense in modern GUIs where the
     // controls don't all use the same fonts (nor colours)
+#ifdef __WXGTK__
+    if ( !IsTopLevel() && parent->m_inheritFont && !m_hasFont )
+        SetFont(parent->GetFont());
+#else
     if ( parent->m_inheritFont && !m_hasFont )
         SetFont(parent->GetFont());
+#endif
 
     // in addition, there is a possibility to explicitly forbid inheriting
     // colours at each class level by overriding ShouldInheritColours()
