diff -bru wxWidgets.orig/src/common/wincmn.cpp wxWidgets/src/common/wincmn.cpp
--- wxWidgets.orig/src/common/wincmn.cpp	2012-03-05 00:58:17.000000000 -0800
+++ wxWidgets/src/common/wincmn.cpp	2012-03-05 01:20:59.000000000 -0800
@@ -1470,6 +1470,9 @@
     // which ensures that this only happens if the user really wants it and
     // not by default which wouldn't make any sense in modern GUIs where the
     // controls don't all use the same fonts (nor colours)
+    if( IsTopLevel() )
+        SetFont( wxSystemSettings::GetFont( wxSYS_DEFAULT_GUI_FONT ) );
+    else
         if ( parent->m_inheritFont && !m_hasFont )
             SetFont(parent->GetFont());
 
