Index: interface/wx/strconv.h
===================================================================
--- interface/wx/strconv.h	(revision 70597)
+++ interface/wx/strconv.h	(working copy)
@@ -478,45 +478,3 @@
     */
     bool IsOk() const;
 };
-
-
-
-/**
-    @class wxMBConvFile
-
-    This class used to define the class instance @b wxConvFileName, but
-    nowadays @b wxConvFileName is either of type wxConvLibc (on most platforms)
-    or wxConvUTF8 (on MacOS X).
-
-    @b wxConvFileName converts filenames between filesystem multibyte encoding
-    and Unicode. @b wxConvFileName can also be set to a something else at
-    run-time which is used e.g. by wxGTK to use a class which checks the
-    environment variable @b G_FILESYSTEM_ENCODING indicating that filenames
-    should not be interpreted as UTF8 and also for converting invalid UTF8
-    characters (e.g. if there is a filename in iso8859_1) to strings with octal
-    values.
-
-    Since some platforms (such as Win32) use Unicode in the filenames,
-    and others (such as Unix) use multibyte encodings, this class should only
-    be used directly if wxMBFILES is defined to 1. A convenience macro,
-    @c wxFNCONV, is defined to @c wxConvFileName->cWX2MB in this case. You
-    could use it like this:
-
-    @code
-    wxChar *name = "rawfile.doc";
-    FILE *fil = fopen(wxFNCONV(name), "r");
-    @endcode
-
-    (although it would be better to just use wxFopen(name, "r") in this
-    particular case, you only need to use this class for functions taking file
-    names not wrapped by wxWidgets.)
-
-    @library{wxbase}
-    @category{conv}
-
-    @see @ref overview_mbconv
-*/
-class wxMBConvFile : public wxMBConv
-{
-public:
-};
