diff -rbu2 wx2.orig/include/wx/utils.h wx2/include/wx/utils.h
--- wx2.orig/include/wx/utils.h	Thu Mar  8 07:57:52 2001
+++ wx2/include/wx/utils.h	Fri Aug 24 15:12:44 2001
@@ -438,4 +438,5 @@
 WXDLLEXPORT void wxFatalError(const wxString& msg, const wxString& title = wxFatalErrorStr);
 
+char* wxRemoveLineTerminator(char* aString);
 
 #endif
diff -rbu2 wx2.orig/samples/db/dbtest.cpp wx2/samples/db/dbtest.cpp
--- wx2.orig/samples/db/dbtest.cpp	Fri Aug 24 14:36:54 2001
+++ wx2/samples/db/dbtest.cpp	Fri Aug 24 15:39:24 2001
@@ -189,17 +189,17 @@
     char buffer[1000+1];
     fgets(buffer, sizeof(params.ODBCSource), paramFile);
-    buffer[wxStrlen(buffer)-1] = '\0';
+    wxRemoveLineTerminator(buffer);
     wxStrcpy(params.ODBCSource,buffer);
 
     fgets(buffer, sizeof(params.UserName), paramFile);
-    buffer[wxStrlen(buffer)-1] = '\0';
+    wxRemoveLineTerminator(buffer);
     wxStrcpy(params.UserName,buffer);
 
     fgets(buffer, sizeof(params.Password), paramFile);
-    buffer[wxStrlen(buffer)-1] = '\0';
+    wxRemoveLineTerminator(buffer);
     wxStrcpy(params.Password,buffer);
 
     fgets(buffer, sizeof(params.DirPath), paramFile);
-    buffer[wxStrlen(buffer)-1] = '\0';
+    wxRemoveLineTerminator(buffer);
     wxStrcpy(params.DirPath,buffer);
 
diff -rbu2 wx2.orig/src/generic/helphtml.cpp wx2/src/generic/helphtml.cpp
--- wx2.orig/src/generic/helphtml.cpp	Wed Jul 19 10:30:52 2000
+++ wx2/src/generic/helphtml.cpp	Fri Aug 24 15:07:14 2001
@@ -174,7 +174,5 @@
       if(fgets(buffer,WXEXTHELP_BUFLEN,input) && *buffer != WXEXTHELP_COMMENTCHAR)
       {
-         len = strlen(buffer);
-         if(buffer[len-1] == '\n')
-            buffer[len-1] = '\0'; // cut of trailing newline
+         wxRemoveLineTerminator(buffer);
          if(sscanf(buffer,"%d", &id) != 1)
             break; // error
diff -rbu2 wx2.orig/src/msw/utils.cpp wx2/src/msw/utils.cpp
--- wx2.orig/src/msw/utils.cpp	Thu Apr 26 11:42:54 2001
+++ wx2/src/msw/utils.cpp	Fri Aug 24 15:38:34 2001
@@ -1295,2 +1295,15 @@
 
 
+/*
+ * Remove CR or CR/LF from a character string.
+ */
+char* wxRemoveLineTerminator(char* aString)
+{
+    int len = strlen(aString);
+    while (len > 0 && (aString[len-1] == '\r' || aString[len-1] == '\n')) {
+        aString[len-1] = '\0';
+        len--;
+    }
+    return aString;
+}
+

 	  	 
