Index: include/wx/xml/xml.h
===================================================================
--- include/wx/xml/xml.h	(revision 70163)
+++ include/wx/xml/xml.h	(working copy)
@@ -55,7 +55,6 @@
     wxXML_HTML_DOCUMENT_NODE = 13
 };
 
-
 // Represents node property(ies).
 // Example: in <img src="hello.gif" id="3"/> "src" is property with value
 //          "hello.gif" and "id" is prop. with value "3".
@@ -83,6 +82,9 @@
     wxXmlProperty *m_next;
 };
 
+#if wxABI_VERSION >= 20813
+#define wxXmlAttribute wxXmlProperty
+#endif
 
 
 // Represents node in XML document. Node has name and may have content
@@ -161,7 +163,7 @@
 
 #if wxABI_VERSION >= 20811
     wxString GetAttribute(const wxString& attrName,
-                         const wxString& defaultVal) const
+                         const wxString& defaultVal = wxEmptyString) const
     {
         return GetPropVal(attrName, defaultVal);
     }
@@ -180,6 +182,14 @@
 #endif // wx >= 2.8.11
 
 #if wxABI_VERSION >= 20813
+    bool HasAttribute(const wxString& propName) const
+    {
+        return HasProp(propName);
+    }
+    void SetAttributes(wxXmlAttribute* prop)
+    {
+        SetProperties(prop);
+    }
     bool DeleteAttribute(const wxString& name)
     {
         return DeleteProperty(name);
