diff -rbu2 wxCVS1/wxWindows/docs/latex/wx/grid.tex wxCVS/wxWindows/docs/latex/wx/grid.tex
--- wxCVS1/wxWindows/docs/latex/wx/grid.tex	Tue Jul  9 02:42:47 2002
+++ wxCVS/wxWindows/docs/latex/wx/grid.tex	Wed Jul 10 13:57:46 2002
@@ -648,4 +648,42 @@
 Returns the height of the specified row.
 
+\membersection{wxGrid::GetSelectionMode}\label{wxgridgetselectionmode}
+
+\constfunc{wxGrid::wxGridSelectionModes}{GetSelectionMode}{\void}
+
+Returns the current selection mode, see \helpref{wxGrid::SetSelectionMode}{wxgridsetselectionmode}.
+
+\membersection{wxGrid::GetSelectedCells}\label{wxgridgetselectedcells}
+
+\constfunc{wxGridCellCoordsArray}{GetSelectedCells}{\void}
+
+Returns an array of singly selected cells.
+
+\membersection{wxGrid::GetSelectedCols}\label{wxgridgetselectedcols}
+
+\constfunc{wxArrayInt}{GetSelectedCols}{\void}
+
+Returns an array of selected cols.
+
+\membersection{wxGrid::GetSelectedRows}\label{wxgridgetselectedrows}
+
+\constfunc{wxArrayInt}{GetSelectedRows}{\void}
+
+Returns an array of selected rows.
+
+\membersection{wxGrid::GetSelectionBlockTopLeft}\label{wxgridgetselectionblocktopleft}
+
+\constfunc{wxGridCellCoordsArray}{GetSelectionBlockTopLeft}{\void}
+
+Returns an array of the top left corners of blocks of selected cells, 
+see \helpref{wxGrid::GetSelectionBlockBottomRight}{wxgridgetselectionblockbottomright}.
+
+\membersection{wxGrid::GetSelectionBlockBottomRight}\label{wxgridgetselectionblockbottomright}
+
+\constfunc{wxGridCellCoordsArray}{GetSelectionBlockBottomRight}{\void}
+
+Returns an array of the bottom right corners of blocks of selected cells,
+see \helpref{wxGrid::GetSelectionBlockTopLeft}{wxgridgetselectionblocktopleft}.
+
 \membersection{wxGrid::GetTable}\label{wxgridgettable}
 
@@ -715,4 +753,12 @@
 controlling the editing status of grid cells.
 
+\membersection{wxGrid::IsInSelection}\label{wxgridisinselection}
+
+\constfunc{bool}{IsInSelection}{\param{int }{row}, \param{int }{col}}
+
+\constfunc{bool}{IsInSelection}{\param{const wxGridCellCoords\& }{coords}}
+
+Is this cell currently selected.
+
 \membersection{wxGrid::IsReadOnly}\label{wxgridisreadonly}
 
@@ -1250,12 +1296,4 @@
 Returns the grid row that corresponds to the logical y coordinate. Returns
 wxNOT\_FOUND if there is no row at the y position.
-
-
-
-\membersection{wxGrid::IsInSelection}\label{wxgridisinselection}
-
-\func{bool}{IsInSelection}{\param{int }{row}, \param{int }{col}}
-
-\func{bool}{IsInSelection}{\param{const wxGridCellCoords\& }{coords}}
 


diff -rbu2 wxCVS1/wxWindows/include/wx/generic/grid.h wxCVS/wxWindows/include/wx/generic/grid.h
--- wxCVS1/wxWindows/include/wx/generic/grid.h	Tue Jul  9 02:42:50 2002
+++ wxCVS/wxWindows/include/wx/generic/grid.h	Wed Jul 10 13:57:52 2002
@@ -1023,4 +1023,5 @@
 
     void SetSelectionMode(wxGrid::wxGridSelectionModes selmode);
+    wxGrid::wxGridSelectionModes GetSelectionMode() const;
 
     // ------ grid dimensions
@@ -1123,5 +1124,5 @@
     // ------ edit control functions
     //
-    bool IsEditable() { return m_editable; }
+    bool IsEditable() const { return m_editable; }
     void EnableEditing( bool edit );
 
@@ -1340,5 +1341,5 @@
     void SetReadOnly(int row, int col, bool isReadOnly = TRUE);
 
-    // ------ selections of blocks of cells
+    // ------ select blocks of cells
     //
     void SelectRow( int row, bool addToSelected = FALSE );
@@ -1359,5 +1360,5 @@
     bool IsSelection();
 
-    // ------ deselection
+    // ------ deselect blocks or cells
     //
     void DeselectRow( int row );
@@ -1367,9 +1368,15 @@
     void ClearSelection();
 
-    bool IsInSelection( int row, int col );
+    // ------ selection
+    bool IsInSelection( int row, int col ) const;
 
-    bool IsInSelection( const wxGridCellCoords& coords )
+    bool IsInSelection( const wxGridCellCoords& coords ) const
         { return IsInSelection( coords.GetRow(), coords.GetCol() ); }
 
+    wxGridCellCoordsArray GetSelectedCells() const;
+    wxGridCellCoordsArray GetSelectionBlockTopLeft() const;
+    wxGridCellCoordsArray GetSelectionBlockBottomRight() const;
+    wxArrayInt GetSelectedRows() const;
+    wxArrayInt GetSelectedCols() const;
 
     // This function returns the rectangle that encloses the block of cells


diff -rbu2 wxCVS1/wxWindows/include/wx/generic/gridsel.h wxCVS/wxWindows/include/wx/generic/gridsel.h
--- wxCVS1/wxWindows/include/wx/generic/gridsel.h	Mon Mar 13 17:11:24 2000
+++ wxCVS/wxWindows/include/wx/generic/gridsel.h	Wed Jul 10 13:57:53 2002
@@ -82,4 +82,6 @@
     wxGrid                              *m_grid;
     wxGrid::wxGridSelectionModes        m_selectionMode;
+
+    friend class wxGrid;
 };
 


diff -rbu2 wxCVS1/wxWindows/src/generic/grid.cpp wxCVS/wxWindows/src/generic/grid.cpp
--- wxCVS1/wxWindows/src/generic/grid.cpp	Tue Jul  9 02:43:05 2002
+++ wxCVS/wxWindows/src/generic/grid.cpp	Wed Jul 10 13:57:58 2002
@@ -3666,4 +3666,12 @@
 }
 
+wxGrid::wxGridSelectionModes wxGrid::GetSelectionMode() const
+{
+    wxCHECK_MSG( m_created, wxGrid::wxGridSelectCells,
+                 wxT("Called wxGrid::GetSelectionMode() before calling CreateGrid()") );
+
+    return m_selection->GetSelectionMode();
+}
+
 bool wxGrid::SetTable( wxGridTableBase *table, bool takeOwnership,
                        wxGrid::wxGridSelectionModes selmode )
@@ -8917,5 +8925,5 @@
 }
 
-bool wxGrid::IsInSelection( int row, int col )
+bool wxGrid::IsInSelection( int row, int col ) const
 {
     return ( m_selection && (m_selection->IsInSelection( row, col ) ||
@@ -8925,4 +8933,31 @@
                col <= m_selectingBottomRight.GetCol() )) );
 }
+
+wxGridCellCoordsArray wxGrid::GetSelectedCells() const
+{
+    if (!m_selection) { wxGridCellCoordsArray a; return a; }
+    return m_selection->m_cellSelection;
+}
+wxGridCellCoordsArray wxGrid::GetSelectionBlockTopLeft() const
+{
+    if (!m_selection) { wxGridCellCoordsArray a; return a; }
+    return m_selection->m_blockSelectionTopLeft;
+}
+wxGridCellCoordsArray wxGrid::GetSelectionBlockBottomRight() const
+{
+    if (!m_selection) { wxGridCellCoordsArray a; return a; }
+    return m_selection->m_blockSelectionTopLeft;
+}
+wxArrayInt wxGrid::GetSelectedRows() const
+{
+    if (!m_selection) { wxArrayInt a; return a; }
+    return m_selection->m_rowSelection;
+}
+wxArrayInt wxGrid::GetSelectedCols() const
+{
+    if (!m_selection) { wxArrayInt a; return a; }
+    return m_selection->m_colSelection;
+}
+
 
 void wxGrid::ClearSelection()


 	  	 
