--- wxWidgets-3.0.2/src/osx/cocoa/toolbar.mm	Mon Oct  6 17:33:44 2014
+++ wxWidgets-3.0.2/src/osx/cocoa/toolbar.mm	Wed Nov  5 09:24:48 2014
@@ -114,7 +114,8 @@ class wxToolBarTool : public wxToolBarToolBase
         {
             if ( !IsControl() )
             {
-                [m_controlHandle retain];
+                [m_controlHandle removeFromSuperview];
+                [m_controlHandle release];
             }
             else
             {
