Index: src/common/strconv.cpp
===================================================================
--- src/common/strconv.cpp	(revision 48677)
+++ src/common/strconv.cpp	(working copy)
@@ -391,7 +391,10 @@
     const size_t dstLen = ToWChar(NULL, 0, inBuff, inLen);
     if ( dstLen != wxCONV_FAILED )
     {
-        wxWCharBuffer wbuf(dstLen - 1);
+        // The buffer should always be terminated.  Since the
+        // conversion output may not be, allocate dstLen so the
+        // buffer will add the extra character for the terminator
+        wxWCharBuffer wbuf(dstLen);
         if ( ToWChar(wbuf.data(), dstLen, inBuff, inLen) != wxCONV_FAILED )
         {
             if ( outLen )
@@ -419,7 +422,17 @@
     {
         // special case of empty input: can't allocate 0 size buffer below as
         // wxCharBuffer insists on NUL-terminating it
-        wxCharBuffer buf(dstLen ? dstLen - 1 : 1);
+        
+        // The buffer should always be terminated.  Since the
+        // conversion output may not be, allocate dstLen and extra
+        // GetMBNulLen - 1 and the buffer will add the extra character
+        // for the entire terminator
+        
+        // wxCharBuffer buf(dstLen ? dstLen - 1 : 1);
+        
+        wxCharBuffer buf(dstLen + GetMBNulLen() - 1);
+        wxTmemset(buf.data() + dstLen, 0, GetMBNulLen());
+        
         if ( FromWChar(buf.data(), dstLen, inBuff, inLen) != wxCONV_FAILED )
         {
             if ( outLen )

 	  	 
