--- src/common/stream.cpp.old	2007-02-02 21:58:16.000000000 +0300
+++ src/common/stream.cpp	2007-02-12 18:03:39.000000000 +0300
@@ -870,6 +870,7 @@
 wxInputStream& wxInputStream::Read(wxOutputStream& stream_out)
 {
     char buf[BUF_TEMP_SIZE];
+    size_t lastcount = 0;
 
     for ( ;; )
     {
@@ -879,8 +880,12 @@
 
         if ( stream_out.Write(buf, bytes_read).LastWrite() != bytes_read )
             break;
+
+	lastcount += bytes_read;
     }
 
+    m_lastcount = lastcount;
+
     return *this;
 }
 

 	  	 
