Index: include/wx/simplebook.h
===================================================================
--- include/wx/simplebook.h	(revision 73939)
+++ include/wx/simplebook.h	(working copy)
@@ -156,8 +156,27 @@
 
     virtual wxWindow *DoRemovePage(size_t page)
     {
-        m_pageTexts.erase(m_pageTexts.begin() + page);
-        return wxBookCtrlBase::DoRemovePage(page);
+        wxWindow* win = wxBookCtrlBase::DoRemovePage(page);
+
+        if (win) {
+            m_pageTexts.erase(m_pageTexts.begin() + page);
+            
+            if (m_selection >= page) {
+                int sel;
+                
+                if (GetPageCount() == 0)
+                    sel = wxNOT_FOUND;
+                else
+                    sel = (m_selection > 0) ? m_selection - 1 : 0;
+
+                m_selection = (m_selection == page) ? wxNOT_FOUND : m_selection;
+
+                if (sel != wxNOT_FOUND && sel != m_selection)
+                    SetSelection(sel);
+            }
+        }
+
+        return win;
     }
 
     virtual void DoSize()
