Index: src/osx/cocoa/window.mm
===================================================================
--- src/osx/cocoa/window.mm	(revision 78485)
+++ src/osx/cocoa/window.mm	(working copy)
@@ -863,6 +863,15 @@
     return NO;
 }
 
+- (NSView *)hitTest:(NSPoint)aPoint;
+{
+    wxWidgetCocoaImpl* viewimpl = (wxWidgetCocoaImpl* ) wxWidgetImpl::FindFromWXWidget( self );
+    if ( viewimpl && viewimpl->GetWXPeer() && !viewimpl->GetWXPeer()->IsEnabled() )
+        return nil;
+
+    return [super hitTest:aPoint];
+}
+
 @end // wxNSView
 
 // We need to adopt NSTextInputClient protocol in order to interpretKeyEvents: to work.
@@ -1367,6 +1376,9 @@
 
 void wxWidgetCocoaImpl::keyEvent(WX_NSEvent event, WXWidget slf, void *_cmd)
 {
+    if (!m_wxPeer->IsEnabled())
+        return;
+
     if ( [event type] == NSKeyDown )
     {
         // there are key equivalents that are not command-combos and therefore not handled by cocoa automatically, 
@@ -2178,14 +2190,14 @@
         // sortSubviewsUsingFunction:context:.  Thus we don't bother checking.  Particularly because
         // that case should never occur anyway because that would imply a multi-threaded GUI call
         // which is a big no-no with Cocoa.
-		
+        
         // Subviews are ordered from back to front meaning one that is already lower will have an lower index.
         NSComparisonResult result = (firstI < secondI)
-		?   NSOrderedAscending /* -1 */
-		:   (firstI > secondI)
-		?   NSOrderedDescending /* 1 */
-		:   NSOrderedSame /* 0 */;
-		
+        ?   NSOrderedAscending /* -1 */
+        :   (firstI > secondI)
+        ?   NSOrderedDescending /* 1 */
+        :   NSOrderedSame /* 0 */;
+        
         return result;
     }
 private:
@@ -2213,15 +2225,15 @@
 
 void wxWidgetCocoaImpl::Raise()
 {
-	NSView* nsview = m_osxView;
-	
+    NSView* nsview = m_osxView;
+    
     NSView *superview = [nsview superview];
     CocoaWindowCompareContext compareContext(nsview, [superview subviews]);
-	
+    
     [superview sortSubviewsUsingFunction:
-	 CocoaRaiseWindowCompareFunction
-								 context: &compareContext];
-	
+     CocoaRaiseWindowCompareFunction
+                                 context: &compareContext];
+    
 }
 
 /* Causes Cocoa to lower the target view to the bottom of the Z-Order by telling the sort function that
@@ -2242,14 +2254,14 @@
 
 void wxWidgetCocoaImpl::Lower()
 {
-	NSView* nsview = m_osxView;
-	
+    NSView* nsview = m_osxView;
+    
     NSView *superview = [nsview superview];
     CocoaWindowCompareContext compareContext(nsview, [superview subviews]);
-	
+    
     [superview sortSubviewsUsingFunction:
-	 CocoaLowerWindowCompareFunction
-								 context: &compareContext];
+     CocoaLowerWindowCompareFunction
+                                 context: &compareContext];
 }
 
 void wxWidgetCocoaImpl::ScrollRect( const wxRect *WXUNUSED(rect), int WXUNUSED(dx), int WXUNUSED(dy) )
@@ -2616,6 +2628,9 @@
 
     if ( [targetView respondsToSelector:@selector(setEnabled:) ] )
         [targetView setEnabled:enable];
+
+    if (!enable && HasFocus())
+        m_wxPeer->Navigate();
 }
 
 void wxWidgetCocoaImpl::PulseGauge()
