diff --git a/src/qt/cursor.cpp b/src/qt/cursor.cpp
index 880a279..9271c36 100644
--- a/src/qt/cursor.cpp
+++ b/src/qt/cursor.cpp
@@ -110,7 +110,7 @@
     }
 //    case wxCURSOR_ARROW: 
     case wxCURSOR_DEFAULT:     qt_cur = Qt::ArrowCursor;    break;
-//    case wxCURSOR_RIGHT_ARROW:
+    case wxCURSOR_RIGHT_ARROW:
     case wxCURSOR_HAND:        qt_cur = Qt::OpenHandCursor; break;
     case wxCURSOR_CROSS:       qt_cur = Qt::CrossCursor;    break;
     case wxCURSOR_SIZEWE:      qt_cur = Qt::SizeHorCursor;  break;
@@ -141,7 +141,9 @@
     case wxCURSOR_BASED_ARROW_UP:   qt_cur = Qt::UpArrowCursor; break;
 */
     default:
-        wxFAIL_MSG(wxT("unsupported cursor type"));
+        wxString mystring;
+        mystring << cursorId;
+        wxFAIL_MSG(wxT("unsupported cursor type cursorId=") + mystring);
         // will use the standard one
         break;
     }
diff --git a/src/qt/settings.cpp b/src/qt/settings.cpp
index 83b3ea5..e7194b9 100644
--- a/src/qt/settings.cpp
+++ b/src/qt/settings.cpp
@@ -156,7 +156,7 @@
         default:
             // let Qt font matching algorithm to select the font
             font.SetFamily(wxFONTFAMILY_DEFAULT);
-            font.SetFaceName("");
+            font.SetFaceName("DejaVu Sans");
             break;
     }
 
