Index: datavgen.cpp
===================================================================
--- datavgen.cpp	(revision 68170)
+++ datavgen.cpp	(working copy)
@@ -4025,6 +4025,10 @@
     Layout();
 
     AdjustScrollbars();
+
+    if(!HasFlag(wxDV_NO_HEADER) && (m_headerArea) && m_headerArea->GetSize().y <= m_headerArea->GetBestSize().y){
+        m_headerArea->Refresh();
+    }
 }
 
 void wxDataViewCtrl::SetFocus()
