Index: imaggif.h
===================================================================
--- imaggif.h	(revision 77080)
+++ imaggif.h	(working copy)
@@ -24,6 +24,13 @@
 struct GifHashTableType;
 class WXDLLIMPEXP_FWD_CORE wxImageArray; // anidecod.h
 
+/**
+    @class wxGIFHandler
+
+    A @ref wxImageHandler for saving and loading GIF-animations.
+
+    @library{wxcore}
+*/
 class WXDLLIMPEXP_CORE wxGIFHandler : public wxImageHandler
 {
 public:
@@ -42,7 +49,12 @@
     virtual bool SaveFile(wxImage *image, wxOutputStream& stream,
                           bool verbose=true) wxOVERRIDE;
 
-    // Save animated gif
+    /**
+        Save the images as an animated gif.
+
+        The wxImageArray is a wxObjArray, and will delete its content
+        on destruction.
+    */
     bool SaveAnimation(const wxImageArray& images, wxOutputStream *stream,
         bool verbose = true, int delayMilliSecs = 1000);
 
