Index: src/generic/datavgen.cpp
===================================================================
--- src/generic/datavgen.cpp	(revision 68314)
+++ src/generic/datavgen.cpp	(working copy)
@@ -3374,7 +3374,7 @@
             break;
 
         case WXK_DOWN:
-            if ( m_currentRow < GetRowCount() - 1 )
+            if ( int(m_currentRow) < int(GetRowCount()) - 1 )
                 OnArrowChar( m_currentRow + 1, event );
             break;
         // Add the process for tree expanding/collapsing
