--- F:\wx230\include\wx\window.h	Thu Apr 26 22:29:04 2001
+++ C:\wx230\include\wx\window.h	Fri Nov 02 00:45:38 2001
@@ -33,4 +33,6 @@
 #include "wx/validate.h"        // for wxDefaultValidator (always include it)
 
+#include <commctrl.h>
+
 #if wxUSE_ACCEL
     #include "wx/accel.h"
@@ -567,4 +569,17 @@
     // scrollbars
     // ----------
+
+	enum wxScrollModes
+	{
+		wxFlatInvert = FSB_FLAT_MODE,
+		wxFlatRaise = FSB_ENCARTA_MODE,
+		wxStandard = FSB_REGULAR_MODE
+	};
+
+protected:
+	bool m_fsb_initialized;
+public:
+
+	virtual void SetScrollbarMode(wxScrollModes mode);
 
         // configure the window scrollbars
--- F:\wx230\src\msw\window.cpp	Sun Jun 10 13:39:16 2001
+++ C:\wx230\src\msw\window.cpp	Fri Nov 02 01:47:44 2001
@@ -595,5 +595,10 @@
     HWND hWnd = GetHwnd();
     if ( hWnd )
+	{
+		if (m_fsb_initialized)
+			::FlatSB_SetScrollInfo(hWnd, dir, &info, refresh);
+		else
         ::SetScrollInfo(hWnd, dir, &info, refresh);
+	}
 #else
     int wOrient;
@@ -605,5 +610,10 @@
     HWND hWnd = GetHwnd();
     if ( hWnd )
+	{
+		if (m_fsb_initialized)
+			::FlatSB_SetScrollRange(hWnd, wOrient, 0, range, refresh);
+		else
         ::SetScrollRange(hWnd, wOrient, 0, range, refresh);
+	}
 #endif
 }
@@ -630,5 +640,10 @@
     HWND hWnd = GetHwnd();
     if ( hWnd )
+	{
+		if (m_fsb_initialized)
+			::FlatSB_SetScrollInfo(hWnd, dir, &info, refresh);
+		else
         ::SetScrollInfo(hWnd, dir, &info, refresh);
+	}
 #else
     if ( orient == wxHORIZONTAL )
@@ -655,4 +670,7 @@
     if ( hWnd )
     {
+		if (m_fsb_initialized)
+			::FlatSB_GetScrollRange(hWnd, wOrient, &minPos, &maxPos);
+		else
         ::GetScrollRange(hWnd, wOrient, &minPos, &maxPos);
 #if defined(__WIN95__)
@@ -691,4 +709,7 @@
     if ( hWnd )
     {
+		if (m_fsb_initialized)
+			return ::FlatSB_GetScrollPos(hWnd, wOrient);
+		else
         return ::GetScrollPos(hWnd, wOrient);
     }
@@ -715,5 +736,9 @@
     if ( hWnd )
     {
+		if (m_fsb_initialized)
+			::FlatSB_GetScrollRange(hWnd, wOrient, &minPos, &maxPos);
+		else
         ::GetScrollRange(hWnd, wOrient, &minPos, &maxPos);
+
 #if defined(__WIN95__)
         // Try to adjust the range to cope with page size > 1
@@ -762,5 +787,10 @@
     HWND hWnd = GetHwnd();
     if ( hWnd )
+	{
+		if (m_fsb_initialized)
+			::FlatSB_SetScrollInfo(hWnd, dir, &info, refresh);
+	    else
         ::SetScrollInfo(hWnd, dir, &info, refresh);
+	}
 #else
     int wOrient;
@@ -772,8 +802,23 @@
     HWND hWnd = GetHwnd();
     if ( hWnd )
+	{
+		if (m_fsb_initialized)
+	        ::FlatSB_SetScrollPos(hWnd, wOrient, pos, refresh);
+		else
         ::SetScrollPos(hWnd, wOrient, pos, refresh);
+	}
 #endif
 }
 
+// Jeffrey Drake,Nov2/01
+void wxWindowBase::SetScrollbarMode(wxScrollModes mode)
+{
+	InitializeFlatSB((HWND)GetHandle());
+
+	FlatSB_SetScrollProp((HWND)GetHandle(), WSB_PROP_HSTYLE, mode, true);
+
+	m_fsb_initialized = true;
+}
+
 // New function that will replace some of the above.
 void wxWindow::SetScrollbar(int orient, int pos, int thumbVisible,
@@ -811,5 +856,10 @@
     HWND hWnd = GetHwnd();
     if ( hWnd )
+	{
+		if (m_fsb_initialized)
+	        ::FlatSB_SetScrollInfo(hWnd, dir, &info, refresh);
+		else
         ::SetScrollInfo(hWnd, dir, &info, refresh);
+	}
 #else
     int wOrient;
@@ -822,7 +872,15 @@
     if ( hWnd )
     {
+		if (m_fsb_initialized)
+		{
+			::FlatSB_SetScrollRange(hWnd, wOrient, 0, range, FALSE);
+			::FlatSB_SetScrollPos(hWnd, wOrient, pos, refresh);
+		}
+		else
+		{
         ::SetScrollRange(hWnd, wOrient, 0, range, FALSE);
         ::SetScrollPos(hWnd, wOrient, pos, refresh);
     }
+    }
 #endif
     if ( orient == wxHORIZONTAL ) {
@@ -3748,8 +3806,21 @@
             scrollInfo.fMask = SIF_TRACKPOS;
 
-            if ( !::GetScrollInfo(GetHwnd(),
+			int si;
+			if (m_fsb_initialized)
+			{
+				si = ::FlatSB_GetScrollInfo(GetHwnd(),
+                                  orientation == wxHORIZONTAL ? SB_HORZ
+                                                              : SB_VERT,
+                                  &scrollInfo );
+			}
+			else
+			{
+				si = ::GetScrollInfo(GetHwnd(),
                                   orientation == wxHORIZONTAL ? SB_HORZ
                                                               : SB_VERT,
-                                  &scrollInfo) )
+                                  &scrollInfo );
+			}
+            
+			if ( !si )
             {
                 wxLogLastError(_T("GetScrollInfo"));

 	  	 
