--- src/msw/notebook.cpp	2003/04/06 17:46:48	1.70
+++ src/msw/notebook.cpp	2003/04/25 06:19:54
@@ -439,6 +439,20 @@
 
     // succeeded: save the pointer to the page
     m_pages.Insert(pPage, nPage);
+    
+    // check if the inserted page is the only page
+    if ( m_pages.GetCount() == 1 ) {
+        // this page will again have to be resized
+        // since inserting the first page creates
+        // the 'tab' at the top of the notebook
+        
+        RECT rect;
+        rect.left = rect.top = 0;
+        GetSize( (int *)&rect.right, (int *)&rect.bottom );
+        TabCtrl_AdjustRect( m_hwnd, FALSE, &rect );
+        pPage->SetSize( rect.left, rect.top, rect.right - rect.left, 
+            rect.bottom - rect.top );
+    }
 
     // hide the page: unless it is selected, it shouldn't be shown (and if it
     // is selected it will be shown later)

 	  	 
