? filename_ce.patch
Index: src/common/filename.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/common/filename.cpp,v
retrieving revision 1.160
diff -b -u -2 -r1.160 filename.cpp
--- src/common/filename.cpp	2005/09/25 19:58:44	1.160
+++ src/common/filename.cpp	2005/12/01 07:53:44
@@ -1244,4 +1244,5 @@
     wxString sepVol;
 
+#ifndef __WXWINCE__
     if ( (GetFormat(format) == wxPATH_DOS) ||
          (GetFormat(format) == wxPATH_VMS) )
@@ -1250,4 +1251,5 @@
     }
     //else: leave empty
+#endif
 
     return sepVol;

 	  	 
