Index: src/osx/cocoa/nonownedwnd.mm
===================================================================
--- src/osx/cocoa/nonownedwnd.mm	(Revision 72426)
+++ src/osx/cocoa/nonownedwnd.mm	(Arbeitskopie)
@@ -111,7 +111,7 @@
     return (wxNonOwnedWindowCocoaImpl*) wxNonOwnedWindowImpl::FindFromWXWindow( self );
 }
 
-// TODO in cocoa everything during a drag is sent to the NSWindow the mouse down occured, 
+// TODO in cocoa everything during a drag is sent to the NSWindow the mouse down occurred, 
 // this does not conform to the wx behaviour if the window is not captured, so try to resend
 // or capture all wx mouse event handling at the tlw as we did for carbon
 
Index: src/osx/cocoa/menu.mm
===================================================================
--- src/osx/cocoa/menu.mm	(Revision 72426)
+++ src/osx/cocoa/menu.mm	(Arbeitskopie)
@@ -166,7 +166,7 @@
     virtual void InsertOrAppend(wxMenuItem *pItem, size_t pos)
     {
         NSMenuItem* nsmenuitem = (NSMenuItem*) pItem->GetPeer()->GetHMenuItem();
-        // make sure a call of SetSubMenu is also reflected (occuring after Create)
+        // make sure a call of SetSubMenu is also reflected (occurring after Create)
         // update the native menu item accordingly
         
         if ( pItem->IsSubMenu() )
Index: src/osx/cocoa/dataview.mm
===================================================================
--- src/osx/cocoa/dataview.mm	(Revision 72426)
+++ src/osx/cocoa/dataview.mm	(Arbeitskopie)
@@ -753,7 +753,7 @@
 
     // send first the event to wxWidgets that the sorting has changed so that
     // the program can do special actions before the sorting actually starts:
-    wxDataViewEvent event(wxEVT_COMMAND_DATAVIEW_COLUMN_SORTED,dvc->GetId()); // variable defintion
+    wxDataViewEvent event(wxEVT_COMMAND_DATAVIEW_COLUMN_SORTED,dvc->GetId()); // variable definition
 
     event.SetEventObject(dvc);
     if (noOfDescriptors > 0)
