diff -ru2 -xCVS -x'.#*' -x'*.orig' -x'*.rej' -x'*~' /tmp/wxWindows/wxGTK-2.4.1/src/common/dbtable.cpp /home/donald/wxWindows/wxGTK-2.4.1/src/common/dbtable.cpp
--- /tmp/wxWindows/wxGTK-2.4.1/src/common/dbtable.cpp	2003-06-08 03:13:13.000000000 -0500
+++ /home/donald/wxWindows/wxGTK-2.4.1/src/common/dbtable.cpp	2003-07-04 13:23:58.000000000 -0500
@@ -629,4 +629,7 @@
     RETCODE retcode;
 
+    // fix added by d.taylor to properly update columns on update
+    bindUpdateParams();
+
     // Execute the UPDATE statement
     retcode = SQLExecDirect(hstmtUpdate, (UCHAR FAR *) pSqlStmt.c_str(), SQL_NTS);
@@ -1237,4 +1240,5 @@
                     colValue.Printf(wxT("'%s'"), (UCHAR FAR *) colDefs[i].PtrDataObj);
                     break;
+                case SQL_C_SHORT:       // dtaylor fix missing
                 case SQL_C_SSHORT:
                     colValue.Printf(wxT("%hi"), *((SWORD *) colDefs[i].PtrDataObj));
@@ -1243,4 +1247,5 @@
                     colValue.Printf(wxT("%hu"), *((UWORD *) colDefs[i].PtrDataObj));
                     break;
+                case SQL_C_LONG:        // dtaylor fix missing
                 case SQL_C_SLONG:
                     colValue.Printf(wxT("%li"), *((SDWORD *) colDefs[i].PtrDataObj));
@@ -1255,4 +1260,13 @@
                     colValue.Printf(wxT("%.6f"), *((SDOUBLE *) colDefs[i].PtrDataObj));
                     break;
+                // added default by dtaylor to trap unknown columns types
+                default:
+                    {
+                        wxString strMsg;
+                        strMsg.Printf(wxT("wxDbTable::bindParams(): Unknown column type for colDefs %d colName %s"),
+                                    i,colDefs[i].ColName);
+                        wxFAIL_MSG(wxT(strMsg));
+                    }
+                    break;
             }
             pWhereClause += colValue;


 	  	 
