Index: include/wx/fdrepdlg.h
===================================================================
--- include/wx/fdrepdlg.h	(revision 72694)
+++ include/wx/fdrepdlg.h	(working copy)
@@ -67,8 +67,8 @@
     wxFindReplaceData(wxUint32 flags) { Init(); SetFlags(flags); }
 
     // accessors
-    const wxString& GetFindString() { return m_FindWhat; }
-    const wxString& GetReplaceString() { return m_ReplaceWith; }
+    const wxString& GetFindString() const { return m_FindWhat; }
+    const wxString& GetReplaceString() const { return m_ReplaceWith; }
 
     int GetFlags() const { return m_Flags; }
 
Index: interface/wx/fdrepdlg.h
===================================================================
--- interface/wx/fdrepdlg.h	(revision 72694)
+++ interface/wx/fdrepdlg.h	(working copy)
@@ -130,7 +130,7 @@
     /**
         Get the string to find.
     */
-    const wxString& GetFindString();
+    const wxString& GetFindString() const;
 
     /**
         Get the combination of @c wxFindReplaceFlags values.
@@ -140,7 +140,7 @@
     /**
         Get the replacement string.
     */
-    const wxString& GetReplaceString();
+    const wxString& GetReplaceString() const;
 
     /**
         Set the string to find (used as initial value by the dialog).
