Index: src/propgrid/props.cpp
===================================================================
--- src/propgrid/props.cpp	(revision 60060)
+++ src/propgrid/props.cpp	(working copy)
@@ -1246,6 +1246,8 @@
             {
                 boolProp = new wxBoolProperty( label, label, child_val );
             }
+            boolProp->SetAttribute(wxPG_BOOL_USE_CHECKBOX,GetAttributeAsLong(wxPG_BOOL_USE_CHECKBOX,0));
+            boolProp->SetAttribute(wxPG_BOOL_USE_DOUBLE_CLICK_CYCLING,GetAttributeAsLong(wxPG_BOOL_USE_DOUBLE_CLICK_CYCLING,0));
             AddPrivateChild(boolProp);
         }
 
