--- include/wx/wxcrtbase.h.orig	2020-07-22 20:20:06.000000000 +0300
+++ include/wx/wxcrtbase.h	2020-11-10 13:17:12.978404060 +0300
@@ -504,7 +504,7 @@
    ------------------------------------------------------------------------- */
 
 #define wxCRT_GetenvA           getenv
-#ifdef _tgetenv
+#if defined(wxHAVE_TCHAR_SUPPORT) && defined(_tgetenv)
     #define wxCRT_GetenvW       _wgetenv
 #endif
 
@@ -515,7 +515,7 @@
 
 #define wxCRT_SystemA               system
 /* mingw32 doesn't provide _tsystem() or _wsystem(): */
-#if defined(_tsystem)
+#if defined(wxHAVE_TCHAR_SUPPORT) && defined(_tsystem)
     #define  wxCRT_SystemW          _wsystem
 #endif
 
