Index: src/msw/app.cpp
===================================================================
RCS file: /home/wxcvs/wxWindows/src/msw/app.cpp,v
retrieving revision 1.155
diff -b -u -2 -r1.155 app.cpp
--- src/msw/app.cpp	2001/11/02 00:44:19	1.155
+++ src/msw/app.cpp	2001/11/23 13:38:45
@@ -282,4 +282,5 @@
     LOGBRUSH lb;
     lb.lbStyle = BS_PATTERN;
+    lb.lbColor = 0; // to prevent Purify from reporting an UMR in CreateBrushIndirect
     lb.lbHatch = (int)LoadBitmap( wxhInstance, wxT("wxDISABLE_BUTTON_BITMAP") );
     if ( lb.lbHatch )

 	  	 
