diff -uN wxWidgets-3.0.0-orig/configure.in wxWidgets-3.0.0-15792/configure.in
--- wxWidgets-3.0.0-orig/configure.in	2013-11-11 14:10:39.000000000 +0100
+++ wxWidgets-3.0.0-15792/configure.in	2014-01-04 17:26:59.044968600 +0100
@@ -717,6 +717,7 @@
 
 WX_ARG_ENABLE(compat26,      [  --enable-compat26       enable wxWidgets 2.6 compatibility], WXWIN_COMPATIBILITY_2_6)
 WX_ARG_DISABLE(compat28,     [  --disable-compat28      disable wxWidgets 2.8 compatibility], WXWIN_COMPATIBILITY_2_8)
+WX_ARG_DISABLE(compat30,     [  --disable-compat30      disable wxWidgets 3.0 compatibility], WXWIN_COMPATIBILITY_3_0)
 
 WX_ARG_DISABLE(rpath,        [  --disable-rpath         disable use of rpath for uninstalled builds], wxUSE_RPATH)
 
@@ -5329,6 +5330,10 @@
     AC_DEFINE(WXWIN_COMPATIBILITY_2_8)
 fi
 
+if test "x$WXWIN_COMPATIBILITY_3_0" != "xno"; then
+    AC_DEFINE(WXWIN_COMPATIBILITY_3_0)
+fi
+
 dnl ---------------------------------------------------------------------------
 dnl the library may be built without GUI classes at all
 dnl ---------------------------------------------------------------------------
@@ -8472,6 +8477,7 @@
 echo "  What level of wxWidgets compatibility should be enabled?"
 echo "                                       wxWidgets 2.6      ${WXWIN_COMPATIBILITY_2_6:-no}"
 echo "                                       wxWidgets 2.8      ${WXWIN_COMPATIBILITY_2_8:-yes}"
+echo "                                       wxWidgets 3.0      ${WXWIN_COMPATIBILITY_3_0:-yes}"
 
 echo "  Which libraries should wxWidgets use?"
 echo "                                       STL                ${wxUSE_STL}"
diff -uN wxWidgets-3.0.0-orig/setup.h.in wxWidgets-3.0.0-15792/setup.h.in
--- wxWidgets-3.0.0-orig/setup.h.in	2013-11-11 14:10:39.000000000 +0100
+++ wxWidgets-3.0.0-15792/setup.h.in	2014-01-04 17:27:43.377042400 +0100
@@ -156,6 +156,8 @@
 
 #define WXWIN_COMPATIBILITY_2_8 0
 
+#define WXWIN_COMPATIBILITY_3_0 0
+
 #define wxDIALOG_UNIT_COMPATIBILITY   0
 
 
diff -uN wxWidgets-3.0.0-orig/setup.h_vms wxWidgets-3.0.0-15792/setup.h_vms
--- wxWidgets-3.0.0-orig/setup.h_vms	2013-11-11 14:18:38.000000000 +0100
+++ wxWidgets-3.0.0-15792/setup.h_vms	2014-01-04 17:28:05.340293600 +0100
@@ -191,6 +191,8 @@
 
 #define WXWIN_COMPATIBILITY_2_8 1
 
+#define WXWIN_COMPATIBILITY_3_0 1
+
 #define wxEVENTS_COMPATIBILITY_2_8 1
 
 #define wxDIALOG_UNIT_COMPATIBILITY   1
diff -uNr wxWidgets-3.0.0-orig\include/wx/brush.h wxWidgets-3.0.0-15792\include/wx/brush.h
--- wxWidgets-3.0.0-orig\include/wx/brush.h	2013-11-11 14:18:36.000000000 +0100
+++ wxWidgets-3.0.0-15792\include/wx/brush.h	2014-01-04 17:10:28.594439300 +0100
@@ -93,7 +93,7 @@
     wxBrush *FindOrCreateBrush(const wxColour& colour,
                                wxBrushStyle style = wxBRUSHSTYLE_SOLID);
 
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
+#if WXWIN_COMPATIBILITY_3_0
     wxBrush *FindOrCreateBrush(const wxColour& colour, int style)
         { return FindOrCreateBrush(colour, (wxBrushStyle)style); }
 #endif
@@ -112,7 +112,7 @@
 //
 // to compile without warnings which it would otherwise provoke from some
 // compilers as it compares elements of different enums
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
+#if WXWIN_COMPATIBILITY_3_0
 
 // Unfortunately some compilers have ambiguity issues when enum comparisons are
 // overloaded so we have to disable the overloads in this case, see
@@ -131,6 +131,6 @@
 
 #endif // wxCOMPILER_NO_OVERLOAD_ON_ENUM
 
-#endif // FUTURE_WXWIN_COMPATIBILITY_3_0
+#endif // WXWIN_COMPATIBILITY_3_0
 
 #endif // _WX_BRUSH_H_BASE_
diff -uNr wxWidgets-3.0.0-orig\include/wx/build.h wxWidgets-3.0.0-15792\include/wx/build.h
--- wxWidgets-3.0.0-orig\include/wx/build.h	2013-11-11 14:18:36.000000000 +0100
+++ wxWidgets-3.0.0-15792\include/wx/build.h	2014-01-04 17:16:16.666546000 +0100
@@ -82,6 +82,11 @@
 #else
     #define __WX_BO_WXWIN_COMPAT_2_8
 #endif
+#if WXWIN_COMPATIBILITY_3_0
+    #define __WX_BO_WXWIN_COMPAT_3_0 ",compatible with 3.0"
+#else
+    #define __WX_BO_WXWIN_COMPAT_3_0
+#endif
 
 // deriving wxWin containers from STL ones changes them completely:
 #if wxUSE_STD_CONTAINERS
@@ -96,7 +101,7 @@
     " (" __WX_BO_UNICODE \
      __WX_BO_COMPILER \
      __WX_BO_STL \
-     __WX_BO_WXWIN_COMPAT_2_6 __WX_BO_WXWIN_COMPAT_2_8 \
+     __WX_BO_WXWIN_COMPAT_2_6 __WX_BO_WXWIN_COMPAT_2_8 __WX_BO_WXWIN_COMPAT_3_0 \
      ")"
 
 
diff -uNr wxWidgets-3.0.0-orig\include/wx/cocoa/brush.h wxWidgets-3.0.0-15792\include/wx/cocoa/brush.h
--- wxWidgets-3.0.0-orig\include/wx/cocoa/brush.h	2013-11-11 14:18:36.000000000 +0100
+++ wxWidgets-3.0.0-15792\include/wx/cocoa/brush.h	2014-01-04 17:16:48.004351600 +0100
@@ -29,8 +29,8 @@
 public:
     wxBrush();
     wxBrush(const wxColour& col, wxBrushStyle style = wxBRUSHSTYLE_SOLID);
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
-    wxDEPRECATED_FUTURE( wxBrush(const wxColour& col, int style) );
+#if WXWIN_COMPATIBILITY_3_0
+    wxDEPRECATED( wxBrush(const wxColour& col, int style) );
 #endif
     wxBrush(const wxBitmap& stipple);
     virtual ~wxBrush();
@@ -54,8 +54,8 @@
     virtual wxBrushStyle GetStyle() const;
     wxBitmap *GetStipple() const;
 
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
-    wxDEPRECATED_FUTURE( void SetStyle(int style) )
+#if WXWIN_COMPATIBILITY_3_0
+    wxDEPRECATED( void SetStyle(int style) )
         { SetStyle((wxBrushStyle)style); }
 #endif
 
diff -uNr wxWidgets-3.0.0-orig\include/wx/cocoa/font.h wxWidgets-3.0.0-15792\include/wx/cocoa/font.h
--- wxWidgets-3.0.0-orig\include/wx/cocoa/font.h	2013-11-11 14:18:36.000000000 +0100
+++ wxWidgets-3.0.0-15792\include/wx/cocoa/font.h	2014-01-04 17:16:56.239314800 +0100
@@ -53,7 +53,7 @@
 
     /*! @abstract   Platform-independent construction with individual properties
      */
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
+#if WXWIN_COMPATIBILITY_3_0
     wxFont(int size,
            int family,
            int style,
diff -uNr wxWidgets-3.0.0-orig\include/wx/cocoa/pen.h wxWidgets-3.0.0-15792\include/wx/cocoa/pen.h
--- wxWidgets-3.0.0-orig\include/wx/cocoa/pen.h	2013-11-11 14:18:36.000000000 +0100
+++ wxWidgets-3.0.0-15792\include/wx/cocoa/pen.h	2014-01-04 17:17:14.047541600 +0100
@@ -25,8 +25,8 @@
 public:
     wxPen();
     wxPen(const wxColour& col, int width = 1, wxPenStyle style = wxPENSTYLE_SOLID);
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
-    wxDEPRECATED_FUTURE( wxPen(const wxColour& col, int width, int style) );
+#if WXWIN_COMPATIBILITY_3_0
+    wxDEPRECATED( wxPen(const wxColour& col, int width, int style) );
 #endif
     wxPen(const wxBitmap& stipple, int width);
     virtual ~wxPen();
@@ -53,8 +53,8 @@
     int GetDashes(wxDash **ptr) const;
     wxBitmap *GetStipple() const;
 
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
-    wxDEPRECATED_FUTURE( void SetStyle(int style) )
+#if WXWIN_COMPATIBILITY_3_0
+    wxDEPRECATED( void SetStyle(int style) )
         { SetStyle((wxPenStyle)style); }
 #endif
 
diff -uNr wxWidgets-3.0.0-orig\include/wx/defs.h wxWidgets-3.0.0-15792\include/wx/defs.h
--- wxWidgets-3.0.0-orig\include/wx/defs.h	2013-11-11 14:18:36.000000000 +0100
+++ wxWidgets-3.0.0-15792\include/wx/defs.h	2014-01-04 17:09:58.609915800 +0100
@@ -13,11 +13,6 @@
 #ifndef _WX_DEFS_H_
 #define _WX_DEFS_H_
 
-/*
-    NOTE: this symbol will be replaced with "WXWIN_COMPATIBILITY_3_0" as soon
-          as the development branch for 3.1 is created
- */
-#define FUTURE_WXWIN_COMPATIBILITY_3_0      1
 #define wxDEPRECATED_FUTURE( x )            x
 
 /*  ---------------------------------------------------------------------------- */
@@ -2420,7 +2415,7 @@
              wxPenStyle, wxPenCap, wxPenJoin enum values instead!
 */
 
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
+#if WXWIN_COMPATIBILITY_3_0
 
 /* don't use any elements of this enum in the new code */
 enum wxDeprecatedGUIConstants
diff -uNr wxWidgets-3.0.0-orig\include/wx/dfb/brush.h wxWidgets-3.0.0-15792\include/wx/dfb/brush.h
--- wxWidgets-3.0.0-orig\include/wx/dfb/brush.h	2013-11-11 14:18:36.000000000 +0100
+++ wxWidgets-3.0.0-15792\include/wx/dfb/brush.h	2014-01-04 17:17:37.908102800 +0100
@@ -32,8 +32,8 @@
 public:
     wxBrush() {}
     wxBrush(const wxColour &colour, wxBrushStyle style = wxBRUSHSTYLE_SOLID);
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
-    wxDEPRECATED_FUTURE( wxBrush(const wxColour& col, int style) );
+#if WXWIN_COMPATIBILITY_3_0
+    wxDEPRECATED( wxBrush(const wxColour& col, int style) );
 #endif
     wxBrush(const wxBitmap &stippleBitmap);
 
@@ -49,8 +49,8 @@
     void SetStyle(wxBrushStyle style);
     void SetStipple(const wxBitmap& stipple);
 
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
-    wxDEPRECATED_FUTURE( void SetStyle(int style) )
+#if WXWIN_COMPATIBILITY_3_0
+    wxDEPRECATED( void SetStyle(int style) )
         { SetStyle((wxBrushStyle)style); }
 #endif
 
diff -uNr wxWidgets-3.0.0-orig\include/wx/dfb/font.h wxWidgets-3.0.0-15792\include/wx/dfb/font.h
--- wxWidgets-3.0.0-orig\include/wx/dfb/font.h	2013-11-11 14:18:36.000000000 +0100
+++ wxWidgets-3.0.0-15792\include/wx/dfb/font.h	2014-01-04 17:17:56.996431200 +0100
@@ -38,7 +38,7 @@
     }
 
     wxFont(const wxNativeFontInfo& info) { Create(info); }
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
+#if WXWIN_COMPATIBILITY_3_0
     wxFont(int size,
            int family,
            int style,
diff -uNr wxWidgets-3.0.0-orig\include/wx/dfb/pen.h wxWidgets-3.0.0-15792\include/wx/dfb/pen.h
--- wxWidgets-3.0.0-orig\include/wx/dfb/pen.h	2013-11-11 14:18:36.000000000 +0100
+++ wxWidgets-3.0.0-15792\include/wx/dfb/pen.h	2014-01-04 17:18:14.809057600 +0100
@@ -32,8 +32,8 @@
 public:
     wxPen() {}
     wxPen(const wxColour &colour, int width = 1, wxPenStyle style = wxPENSTYLE_SOLID);
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
-    wxDEPRECATED_FUTURE( wxPen(const wxColour& col, int width, int style) );
+#if WXWIN_COMPATIBILITY_3_0
+    wxDEPRECATED( wxPen(const wxColour& col, int width, int style) );
 #endif
 
     wxPen(const wxBitmap& stipple, int width);
@@ -60,8 +60,8 @@
     wxDash* GetDash() const;
     wxBitmap *GetStipple() const;
 
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
-    wxDEPRECATED_FUTURE( void SetStyle(int style) )
+#if WXWIN_COMPATIBILITY_3_0
+    wxDEPRECATED( void SetStyle(int style) )
         { SetStyle((wxPenStyle)style); }
 #endif
 
diff -uNr wxWidgets-3.0.0-orig\include/wx/font.h wxWidgets-3.0.0-15792\include/wx/font.h
--- wxWidgets-3.0.0-orig\include/wx/font.h	2013-11-11 14:18:36.000000000 +0100
+++ wxWidgets-3.0.0-15792\include/wx/font.h	2014-01-04 17:11:33.874998800 +0100
@@ -285,7 +285,7 @@
     virtual ~wxFontBase();
 
 
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
+#if WXWIN_COMPATIBILITY_3_0
     // from the font components
     static wxFont *New(
         int pointSize,              // size of the font in points
@@ -475,19 +475,19 @@
 WXDLLIMPEXP_CORE bool wxFromString(const wxString& str, wxFontBase* font);
 
 
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
+#if WXWIN_COMPATIBILITY_3_0
 #define wxDECLARE_FONT_COMPAT_SETTER   \
-    wxDEPRECATED_FUTURE( void SetFamily(int family) ) \
+    wxDEPRECATED( void SetFamily(int family) ) \
         { SetFamily((wxFontFamily)family); } \
-    wxDEPRECATED_FUTURE( void SetStyle(int style) ) \
+    wxDEPRECATED( void SetStyle(int style) ) \
         { SetStyle((wxFontStyle)style); } \
-    wxDEPRECATED_FUTURE( void SetWeight(int weight) ) \
+    wxDEPRECATED( void SetWeight(int weight) ) \
         { SetWeight((wxFontWeight)weight); } \
-    wxDEPRECATED_FUTURE( void SetFamily(wxDeprecatedGUIConstants family) ) \
+    wxDEPRECATED( void SetFamily(wxDeprecatedGUIConstants family) ) \
         { SetFamily((wxFontFamily)family); } \
-    wxDEPRECATED_FUTURE( void SetStyle(wxDeprecatedGUIConstants style) ) \
+    wxDEPRECATED( void SetStyle(wxDeprecatedGUIConstants style) ) \
         { SetStyle((wxFontStyle)style); } \
-    wxDEPRECATED_FUTURE( void SetWeight(wxDeprecatedGUIConstants weight) ) \
+    wxDEPRECATED( void SetWeight(wxDeprecatedGUIConstants weight) ) \
         { SetWeight((wxFontWeight)weight); }
 #else
 #define wxDECLARE_FONT_COMPAT_SETTER  /*empty*/
@@ -546,7 +546,7 @@
                              const wxString& face = wxEmptyString,
                              wxFontEncoding encoding = wxFONTENCODING_DEFAULT);
 
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
+#if WXWIN_COMPATIBILITY_3_0
      wxFont *FindOrCreateFont(int pointSize, int family, int style, int weight,
                               bool underline = false,
                               const wxString& face = wxEmptyString,
@@ -570,7 +570,7 @@
 //
 // to compile without warnings which it would otherwise provoke from some
 // compilers as it compares elements of different enums
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
+#if WXWIN_COMPATIBILITY_3_0
 
 // Unfortunately some compilers have ambiguity issues when enum comparisons are
 // overloaded so we have to disable the overloads in this case, see
@@ -592,7 +592,7 @@
 
 #endif // // wxCOMPILER_NO_OVERLOAD_ON_ENUM
 
-#endif // FUTURE_WXWIN_COMPATIBILITY_3_0
+#endif // WXWIN_COMPATIBILITY_3_0
 
 #endif
     // _WX_FONT_H_BASE_
diff -uNr wxWidgets-3.0.0-orig\include/wx/gtk/brush.h wxWidgets-3.0.0-15792\include/wx/gtk/brush.h
--- wxWidgets-3.0.0-orig\include/wx/gtk/brush.h	2013-11-11 14:18:36.000000000 +0100
+++ wxWidgets-3.0.0-15792\include/wx/gtk/brush.h	2014-01-04 17:18:35.737044400 +0100
@@ -22,8 +22,8 @@
     wxBrush() { }
 
     wxBrush( const wxColour &colour, wxBrushStyle style = wxBRUSHSTYLE_SOLID );
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
-    wxDEPRECATED_FUTURE( wxBrush(const wxColour& col, int style) );
+#if WXWIN_COMPATIBILITY_3_0
+    wxDEPRECATED( wxBrush(const wxColour& col, int style) );
 #endif
     wxBrush( const wxBitmap &stippleBitmap );
     virtual ~wxBrush();
@@ -40,8 +40,8 @@
     void SetStyle( wxBrushStyle style );
     void SetStipple( const wxBitmap& stipple );
 
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
-    wxDEPRECATED_FUTURE( void SetStyle(int style) )
+#if WXWIN_COMPATIBILITY_3_0
+    wxDEPRECATED( void SetStyle(int style) )
         { SetStyle((wxBrushStyle)style); }
 #endif
 
diff -uNr wxWidgets-3.0.0-orig\include/wx/gtk/font.h wxWidgets-3.0.0-15792\include/wx/gtk/font.h
--- wxWidgets-3.0.0-orig\include/wx/gtk/font.h	2013-11-11 14:18:36.000000000 +0100
+++ wxWidgets-3.0.0-15792\include/wx/gtk/font.h	2014-01-04 17:18:49.520780800 +0100
@@ -27,7 +27,7 @@
 
     wxFont(const wxNativeFontInfo& info);
 
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
+#if WXWIN_COMPATIBILITY_3_0
     wxFont(int size,
            int family,
            int style,
diff -uNr wxWidgets-3.0.0-orig\include/wx/gtk/pen.h wxWidgets-3.0.0-15792\include/wx/gtk/pen.h
--- wxWidgets-3.0.0-orig\include/wx/gtk/pen.h	2013-11-11 14:18:36.000000000 +0100
+++ wxWidgets-3.0.0-15792\include/wx/gtk/pen.h	2014-01-04 17:19:07.141822000 +0100
@@ -21,8 +21,8 @@
     wxPen() { }
 
     wxPen( const wxColour &colour, int width = 1, wxPenStyle style = wxPENSTYLE_SOLID );
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
-    wxDEPRECATED_FUTURE( wxPen(const wxColour& col, int width, int style) );
+#if WXWIN_COMPATIBILITY_3_0
+    wxDEPRECATED( wxPen(const wxColour& col, int width, int style) );
 #endif
 
     virtual ~wxPen();
@@ -49,8 +49,8 @@
     wxDash* GetDash() const;
     wxBitmap *GetStipple() const;
 
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
-    wxDEPRECATED_FUTURE( void SetStyle(int style) )
+#if WXWIN_COMPATIBILITY_3_0
+    wxDEPRECATED( void SetStyle(int style) )
         { SetStyle((wxPenStyle)style); }
 #endif
 
diff -uNr wxWidgets-3.0.0-orig\include/wx/gtk1/brush.h wxWidgets-3.0.0-15792\include/wx/gtk1/brush.h
--- wxWidgets-3.0.0-orig\include/wx/gtk1/brush.h	2013-11-11 14:18:36.000000000 +0100
+++ wxWidgets-3.0.0-15792\include/wx/gtk1/brush.h	2014-01-04 17:19:23.181784800 +0100
@@ -32,8 +32,8 @@
     wxBrush() { }
 
     wxBrush( const wxColour &colour, wxBrushStyle style = wxBRUSHSTYLE_SOLID );
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
-    wxDEPRECATED_FUTURE( wxBrush(const wxColour& col, int style) );
+#if WXWIN_COMPATIBILITY_3_0
+    wxDEPRECATED( wxBrush(const wxColour& col, int style) );
 #endif
     wxBrush( const wxBitmap &stippleBitmap );
     virtual ~wxBrush();
@@ -50,8 +50,8 @@
     void SetStyle( wxBrushStyle style );
     void SetStipple( const wxBitmap& stipple );
 
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
-    wxDEPRECATED_FUTURE( void SetStyle(int style) )
+#if WXWIN_COMPATIBILITY_3_0
+    wxDEPRECATED( void SetStyle(int style) )
         { SetStyle((wxBrushStyle)style); }
 #endif
 
diff -uNr wxWidgets-3.0.0-orig\include/wx/gtk1/font.h wxWidgets-3.0.0-15792\include/wx/gtk1/font.h
--- wxWidgets-3.0.0-orig\include/wx/gtk1/font.h	2013-11-11 14:18:36.000000000 +0100
+++ wxWidgets-3.0.0-15792\include/wx/gtk1/font.h	2014-01-04 17:19:38.884173600 +0100
@@ -52,7 +52,7 @@
 
     wxFont(const wxNativeFontInfo& info);
 
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
+#if WXWIN_COMPATIBILITY_3_0
     wxFont(int size,
            int family,
            int style,
diff -uNr wxWidgets-3.0.0-orig\include/wx/gtk1/pen.h wxWidgets-3.0.0-15792\include/wx/gtk1/pen.h
--- wxWidgets-3.0.0-orig\include/wx/gtk1/pen.h	2013-11-11 14:18:36.000000000 +0100
+++ wxWidgets-3.0.0-15792\include/wx/gtk1/pen.h	2014-01-04 17:19:59.296200000 +0100
@@ -37,8 +37,8 @@
     wxPen() { }
 
     wxPen( const wxColour &colour, int width = 1, wxPenStyle style = wxPENSTYLE_SOLID );
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
-    wxDEPRECATED_FUTURE( wxPen(const wxColour& col, int width, int style) );
+#if WXWIN_COMPATIBILITY_3_0
+    wxDEPRECATED( wxPen(const wxColour& col, int width, int style) );
 #endif
 
     bool operator==(const wxPen& pen) const;
@@ -63,8 +63,8 @@
     wxDash* GetDash() const;
     wxBitmap *GetStipple() const;
 
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
-    wxDEPRECATED_FUTURE( void SetStyle(int style) )
+#if WXWIN_COMPATIBILITY_3_0
+    wxDEPRECATED( void SetStyle(int style) )
         { SetStyle((wxPenStyle)style); }
 #endif
 
diff -uNr wxWidgets-3.0.0-orig\include/wx/motif/font.h wxWidgets-3.0.0-15792\include/wx/motif/font.h
--- wxWidgets-3.0.0-orig\include/wx/motif/font.h	2013-11-11 14:18:36.000000000 +0100
+++ wxWidgets-3.0.0-15792\include/wx/motif/font.h	2014-01-04 17:20:18.450323200 +0100
@@ -43,7 +43,7 @@
 
     wxFont(const wxNativeFontInfo& info);
 
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
+#if WXWIN_COMPATIBILITY_3_0
     wxFont(int size,
            int family,
            int style,
diff -uNr wxWidgets-3.0.0-orig\include/wx/msw/brush.h wxWidgets-3.0.0-15792\include/wx/msw/brush.h
--- wxWidgets-3.0.0-orig\include/wx/msw/brush.h	2013-11-11 14:18:36.000000000 +0100
+++ wxWidgets-3.0.0-15792\include/wx/msw/brush.h	2014-01-04 17:12:19.048834000 +0100
@@ -24,8 +24,8 @@
 public:
     wxBrush();
     wxBrush(const wxColour& col, wxBrushStyle style = wxBRUSHSTYLE_SOLID);
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
-    wxDEPRECATED_FUTURE( wxBrush(const wxColour& col, int style) );
+#if WXWIN_COMPATIBILITY_3_0
+    wxDEPRECATED( wxBrush(const wxColour& col, int style) );
 #endif
     wxBrush(const wxBitmap& stipple);
     virtual ~wxBrush();
@@ -42,8 +42,8 @@
     wxBrushStyle GetStyle() const;
     wxBitmap *GetStipple() const;
 
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
-    wxDEPRECATED_FUTURE( void SetStyle(int style) )
+#if WXWIN_COMPATIBILITY_3_0
+    wxDEPRECATED( void SetStyle(int style) )
         { SetStyle((wxBrushStyle)style); }
 #endif
 
diff -uNr wxWidgets-3.0.0-orig\include/wx/msw/font.h wxWidgets-3.0.0-15792\include/wx/msw/font.h
--- wxWidgets-3.0.0-orig\include/wx/msw/font.h	2013-11-11 14:18:36.000000000 +0100
+++ wxWidgets-3.0.0-15792\include/wx/msw/font.h	2014-01-04 17:12:40.216269900 +0100
@@ -25,7 +25,7 @@
 
     wxFont(const wxFontInfo& info);
 
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
+#if WXWIN_COMPATIBILITY_3_0
     wxFont(int size,
            int family,
            int style,
@@ -61,7 +61,7 @@
                         weight, underlined, face, encoding);
     }
 
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
+#if WXWIN_COMPATIBILITY_3_0
     wxFont(const wxSize& pixelSize,
            int family,
            int style,
diff -uNr wxWidgets-3.0.0-orig\include/wx/msw/pen.h wxWidgets-3.0.0-15792\include/wx/msw/pen.h
--- wxWidgets-3.0.0-orig\include/wx/msw/pen.h	2013-11-11 14:18:36.000000000 +0100
+++ wxWidgets-3.0.0-15792\include/wx/msw/pen.h	2014-01-04 17:13:02.506812200 +0100
@@ -23,8 +23,8 @@
 public:
     wxPen() { }
     wxPen(const wxColour& col, int width = 1, wxPenStyle style = wxPENSTYLE_SOLID);
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
-    wxDEPRECATED_FUTURE( wxPen(const wxColour& col, int width, int style) );
+#if WXWIN_COMPATIBILITY_3_0
+    wxDEPRECATED( wxPen(const wxColour& col, int width, int style) );
 #endif
 
     wxPen(const wxBitmap& stipple, int width);
@@ -54,8 +54,8 @@
     int GetDashCount() const;
     wxBitmap* GetStipple() const;
 
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
-    wxDEPRECATED_FUTURE( void SetStyle(int style) )
+#if WXWIN_COMPATIBILITY_3_0
+    wxDEPRECATED( void SetStyle(int style) )
         { SetStyle((wxPenStyle)style); }
 #endif
 
diff -uNr wxWidgets-3.0.0-orig\include/wx/msw/setup.h wxWidgets-3.0.0-15792\include/wx/msw/setup.h
--- wxWidgets-3.0.0-orig\include/wx/msw/setup.h	2013-11-11 14:18:36.000000000 +0100
+++ wxWidgets-3.0.0-15792\include/wx/msw/setup.h	2014-01-04 17:15:13.664957200 +0100
@@ -46,6 +46,16 @@
 // Recommended setting: 0 (please update your code)
 #define WXWIN_COMPATIBILITY_2_8 1
 
+// This setting determines the compatibility with 3.0 API: set it to 0 to
+// flag all cases of using deprecated functions.
+//
+// Default is 1 but please try building your code with 0 as the default will
+// change to 0 in the next version and the deprecated functions will disappear
+// in the version after it completely.
+//
+// Recommended setting: 0 (please update your code)
+#define WXWIN_COMPATIBILITY_3_0 1
+
 // MSW-only: Set to 0 for accurate dialog units, else 1 for old behaviour when
 // default system font is used for wxWindow::GetCharWidth/Height() instead of
 // the current font.
diff -uNr wxWidgets-3.0.0-orig\include/wx/os2/brush.h wxWidgets-3.0.0-15792\include/wx/os2/brush.h
--- wxWidgets-3.0.0-orig\include/wx/os2/brush.h	2013-11-11 14:18:36.000000000 +0100
+++ wxWidgets-3.0.0-15792\include/wx/os2/brush.h	2014-01-04 17:20:38.863549600 +0100
@@ -19,8 +19,8 @@
 public:
     wxBrush();
     wxBrush(const wxColour& rCol, wxBrushStyle nStyle = wxBRUSHSTYLE_SOLID);
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
-    wxDEPRECATED_FUTURE( wxBrush(const wxColour& col, int style) );
+#if WXWIN_COMPATIBILITY_3_0
+    wxDEPRECATED( wxBrush(const wxColour& col, int style) );
 #endif
     wxBrush(const wxBitmap& rStipple);
     virtual ~wxBrush();
@@ -39,8 +39,8 @@
     wxBitmap* GetStipple(void) const;
     int       GetPS(void) const;
 
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
-    wxDEPRECATED_FUTURE( void SetStyle(int style) )
+#if WXWIN_COMPATIBILITY_3_0
+    wxDEPRECATED( void SetStyle(int style) )
         { SetStyle((wxBrushStyle)style); }
 #endif
 
diff -uNr wxWidgets-3.0.0-orig\include/wx/os2/font.h wxWidgets-3.0.0-15792\include/wx/os2/font.h
--- wxWidgets-3.0.0-orig\include/wx/os2/font.h	2013-11-11 14:18:36.000000000 +0100
+++ wxWidgets-3.0.0-15792\include/wx/os2/font.h	2014-01-04 17:20:46.704143200 +0100
@@ -40,7 +40,7 @@
             SetPixelSize(info.GetPixelSize());
     }
 
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
+#if WXWIN_COMPATIBILITY_3_0
     wxFont(int size,
            int family,
            int style,
diff -uNr wxWidgets-3.0.0-orig\include/wx/os2/pen.h wxWidgets-3.0.0-15792\include/wx/os2/pen.h
--- wxWidgets-3.0.0-orig\include/wx/os2/pen.h	2013-11-11 14:18:36.000000000 +0100
+++ wxWidgets-3.0.0-15792\include/wx/os2/pen.h	2014-01-04 17:21:09.482787600 +0100
@@ -28,8 +28,8 @@
           ,int             nWidth = 1
           ,wxPenStyle      nStyle = wxPENSTYLE_SOLID
          );
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
-    wxDEPRECATED_FUTURE( wxPen(const wxColour& col, int width, int style) );
+#if WXWIN_COMPATIBILITY_3_0
+    wxDEPRECATED( wxPen(const wxColour& col, int width, int style) );
 #endif
 
     wxPen( const wxBitmap& rStipple
@@ -68,8 +68,8 @@
     int       GetDashCount() const;
     wxBitmap* GetStipple(void) const;
 
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
-    wxDEPRECATED_FUTURE( void SetStyle(int style) )
+#if WXWIN_COMPATIBILITY_3_0
+    wxDEPRECATED( void SetStyle(int style) )
         { SetStyle((wxPenStyle)style); }
 #endif
 
diff -uNr wxWidgets-3.0.0-orig\include/wx/osx/brush.h wxWidgets-3.0.0-15792\include/wx/osx/brush.h
--- wxWidgets-3.0.0-orig\include/wx/osx/brush.h	2013-11-11 14:18:36.000000000 +0100
+++ wxWidgets-3.0.0-15792\include/wx/osx/brush.h	2014-01-04 17:21:27.540595200 +0100
@@ -23,8 +23,8 @@
 public:
     wxBrush();
     wxBrush(const wxColour& col, wxBrushStyle style = wxBRUSHSTYLE_SOLID);
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
-    wxDEPRECATED_FUTURE( wxBrush(const wxColour& col, int style) );
+#if WXWIN_COMPATIBILITY_3_0
+    wxDEPRECATED( wxBrush(const wxColour& col, int style) );
 #endif
     wxBrush(const wxBitmap& stipple);
     virtual ~wxBrush();
@@ -41,8 +41,8 @@
     wxBrushStyle GetStyle() const ;
     wxBitmap *GetStipple() const ;
 
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
-    wxDEPRECATED_FUTURE( void SetStyle(int style) )
+#if WXWIN_COMPATIBILITY_3_0
+    wxDEPRECATED( void SetStyle(int style) )
         { SetStyle((wxBrushStyle)style); }
 #endif
 
diff -uNr wxWidgets-3.0.0-orig\include/wx/osx/font.h wxWidgets-3.0.0-15792\include/wx/osx/font.h
--- wxWidgets-3.0.0-orig\include/wx/osx/font.h	2013-11-11 14:18:36.000000000 +0100
+++ wxWidgets-3.0.0-15792\include/wx/osx/font.h	2014-01-04 17:21:42.165463600 +0100
@@ -56,7 +56,7 @@
     wxFont(WX_NSFont nsfont);
 #endif
 
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
+#if WXWIN_COMPATIBILITY_3_0
     wxFont(int size,
            int family,
            int style,
diff -uNr wxWidgets-3.0.0-orig\include/wx/osx/pen.h wxWidgets-3.0.0-15792\include/wx/osx/pen.h
--- wxWidgets-3.0.0-orig\include/wx/osx/pen.h	2013-11-11 14:18:36.000000000 +0100
+++ wxWidgets-3.0.0-15792\include/wx/osx/pen.h	2014-01-04 17:21:58.423809600 +0100
@@ -21,8 +21,8 @@
 public:
     wxPen();
     wxPen(const wxColour& col, int width = 1, wxPenStyle style = wxPENSTYLE_SOLID);
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
-    wxDEPRECATED_FUTURE( wxPen(const wxColour& col, int width, int style) );
+#if WXWIN_COMPATIBILITY_3_0
+    wxDEPRECATED( wxPen(const wxColour& col, int width, int style) );
 #endif
 
     wxPen(const wxBitmap& stipple, int width);
@@ -52,8 +52,8 @@
 
     wxBitmap *GetStipple() const ;
 
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
-    wxDEPRECATED_FUTURE( void SetStyle(int style) )
+#if WXWIN_COMPATIBILITY_3_0
+    wxDEPRECATED( void SetStyle(int style) )
         { SetStyle((wxPenStyle)style); }
 #endif
 
diff -uNr wxWidgets-3.0.0-orig\include/wx/pen.h wxWidgets-3.0.0-15792\include/wx/pen.h
--- wxWidgets-3.0.0-orig\include/wx/pen.h	2013-11-11 14:18:36.000000000 +0100
+++ wxWidgets-3.0.0-15792\include/wx/pen.h	2014-01-04 17:11:53.810137400 +0100
@@ -122,7 +122,7 @@
                            int width = 1,
                            wxPenStyle style = wxPENSTYLE_SOLID);
 
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
+#if WXWIN_COMPATIBILITY_3_0
     wxPen *FindOrCreatePen(const wxColour& colour, int width, int style)
         { return FindOrCreatePen(colour, width, (wxPenStyle)style); }
 #endif
@@ -140,7 +140,7 @@
 //
 // to compile without warnings which it would otherwise provoke from some
 // compilers as it compares elements of different enums
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
+#if WXWIN_COMPATIBILITY_3_0
 
 // Unfortunately some compilers have ambiguity issues when enum comparisons are
 // overloaded so we have to disable the overloads in this case, see
@@ -159,6 +159,6 @@
 
 #endif // wxCOMPILER_NO_OVERLOAD_ON_ENUM
 
-#endif // FUTURE_WXWIN_COMPATIBILITY_3_0
+#endif // WXWIN_COMPATIBILITY_3_0
 
 #endif // _WX_PEN_H_BASE_
diff -uNr wxWidgets-3.0.0-orig\include/wx/x11/brush.h wxWidgets-3.0.0-15792\include/wx/x11/brush.h
--- wxWidgets-3.0.0-orig\include/wx/x11/brush.h	2013-11-11 14:18:37.000000000 +0100
+++ wxWidgets-3.0.0-15792\include/wx/x11/brush.h	2014-01-04 17:22:15.306107600 +0100
@@ -31,8 +31,8 @@
     wxBrush() { }
 
     wxBrush( const wxColour &colour, wxBrushStyle style = wxBRUSHSTYLE_SOLID );
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
-    wxDEPRECATED_FUTURE( wxBrush(const wxColour& col, int style) );
+#if WXWIN_COMPATIBILITY_3_0
+    wxDEPRECATED( wxBrush(const wxColour& col, int style) );
 #endif
     wxBrush( const wxBitmap &stippleBitmap );
     virtual ~wxBrush();
@@ -49,8 +49,8 @@
     void SetStyle( wxBrushStyle style );
     void SetStipple( const wxBitmap& stipple );
 
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
-    wxDEPRECATED_FUTURE( void SetStyle(int style) )
+#if WXWIN_COMPATIBILITY_3_0
+    wxDEPRECATED( void SetStyle(int style) )
         { SetStyle((wxBrushStyle)style); }
 #endif
 
diff -uNr wxWidgets-3.0.0-orig\include/wx/x11/font.h wxWidgets-3.0.0-15792\include/wx/x11/font.h
--- wxWidgets-3.0.0-orig\include/wx/x11/font.h	2013-11-11 14:18:37.000000000 +0100
+++ wxWidgets-3.0.0-15792\include/wx/x11/font.h	2014-01-04 17:22:25.948885600 +0100
@@ -34,7 +34,7 @@
             SetPixelSize(info.GetPixelSize());
     }
 
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
+#if WXWIN_COMPATIBILITY_3_0
     wxFont(int size,
            int family,
            int style,
diff -uNr wxWidgets-3.0.0-orig\include/wx/x11/pen.h wxWidgets-3.0.0-15792\include/wx/x11/pen.h
--- wxWidgets-3.0.0-orig\include/wx/x11/pen.h	2013-11-11 14:18:37.000000000 +0100
+++ wxWidgets-3.0.0-15792\include/wx/x11/pen.h	2014-01-04 17:22:41.791695000 +0100
@@ -34,8 +34,8 @@
     wxPen() { }
 
     wxPen( const wxColour &colour, int width = 1, wxPenStyle style = wxPENSTYLE_SOLID );
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
-    wxDEPRECATED_FUTURE( wxPen(const wxColour& col, int width, int style) );
+#if WXWIN_COMPATIBILITY_3_0
+    wxDEPRECATED( wxPen(const wxColour& col, int width, int style) );
 #endif
 
     wxPen( const wxBitmap &stipple, int width );
@@ -63,8 +63,8 @@
     wxDash* GetDash() const;
     wxBitmap* GetStipple() const;
 
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
-    wxDEPRECATED_FUTURE( void SetStyle(int style) )
+#if WXWIN_COMPATIBILITY_3_0
+    wxDEPRECATED( void SetStyle(int style) )
         { SetStyle((wxPenStyle)style); }
 #endif
 
diff -uNr wxWidgets-3.0.0-orig\src/dfb/brush.cpp wxWidgets-3.0.0-15792\src/dfb/brush.cpp
--- wxWidgets-3.0.0-orig\src/dfb/brush.cpp	2013-11-11 14:18:38.000000000 +0100
+++ wxWidgets-3.0.0-15792\src/dfb/brush.cpp	2014-01-04 17:22:58.913688300 +0100
@@ -68,7 +68,7 @@
     m_refData = new wxBrushRefData(colour, style);
 }
 
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
+#if WXWIN_COMPATIBILITY_3_0
 wxBrush::wxBrush(const wxColour& col, int style)
 {
     m_refData = new wxBrushRefData(col, (wxBrushStyle)style);
diff -uNr wxWidgets-3.0.0-orig\src/dfb/pen.cpp wxWidgets-3.0.0-15792\src/dfb/pen.cpp
--- wxWidgets-3.0.0-orig\src/dfb/pen.cpp	2013-11-11 14:18:38.000000000 +0100
+++ wxWidgets-3.0.0-15792\src/dfb/pen.cpp	2014-01-04 17:23:06.801132800 +0100
@@ -67,7 +67,7 @@
     m_refData = new wxPenRefData(colour, style);
 }
 
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
+#if WXWIN_COMPATIBILITY_3_0
 wxPen::wxPen(const wxColour& col, int width, int style)
 {
     m_refData = new wxPenRefData(col, (wxPenStyle)style);
diff -uNr wxWidgets-3.0.0-orig\src/gtk/brush.cpp wxWidgets-3.0.0-15792\src/gtk/brush.cpp
--- wxWidgets-3.0.0-orig\src/gtk/brush.cpp	2013-11-11 14:18:38.000000000 +0100
+++ wxWidgets-3.0.0-15792\src/gtk/brush.cpp	2014-01-04 17:23:16.523647900 +0100
@@ -60,7 +60,7 @@
     m_refData = new wxBrushRefData(colour, style);
 }
 
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
+#if WXWIN_COMPATIBILITY_3_0
 wxBrush::wxBrush(const wxColour& col, int style)
 {
     m_refData = new wxBrushRefData(col, (wxBrushStyle)style);
diff -uNr wxWidgets-3.0.0-orig\src/gtk/pen.cpp wxWidgets-3.0.0-15792\src/gtk/pen.cpp
--- wxWidgets-3.0.0-orig\src/gtk/pen.cpp	2013-11-11 14:18:38.000000000 +0100
+++ wxWidgets-3.0.0-15792\src/gtk/pen.cpp	2014-01-04 17:23:24.545882900 +0100
@@ -95,7 +95,7 @@
     M_PENDATA->m_colour = colour;
 }
 
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
+#if WXWIN_COMPATIBILITY_3_0
 wxPen::wxPen(const wxColour& colour, int width, int style)
 {
     m_refData = new wxPenRefData();
diff -uNr wxWidgets-3.0.0-orig\src/gtk1/brush.cpp wxWidgets-3.0.0-15792\src/gtk1/brush.cpp
--- wxWidgets-3.0.0-orig\src/gtk1/brush.cpp	2013-11-11 14:18:38.000000000 +0100
+++ wxWidgets-3.0.0-15792\src/gtk1/brush.cpp	2014-01-04 17:23:36.130266700 +0100
@@ -62,7 +62,7 @@
     M_BRUSHDATA->m_colour = colour;
 }
 
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
+#if WXWIN_COMPATIBILITY_3_0
 wxBrush::wxBrush(const wxColour& col, int style)
 {
     m_refData = new wxBrushRefData;
diff -uNr wxWidgets-3.0.0-orig\src/gtk1/pen.cpp wxWidgets-3.0.0-15792\src/gtk1/pen.cpp
--- wxWidgets-3.0.0-orig\src/gtk1/pen.cpp	2013-11-11 14:18:38.000000000 +0100
+++ wxWidgets-3.0.0-15792\src/gtk1/pen.cpp	2014-01-04 17:23:44.792056600 +0100
@@ -95,7 +95,7 @@
     M_PENDATA->m_colour = colour;
 }
 
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
+#if WXWIN_COMPATIBILITY_3_0
 wxPen::wxPen(const wxColour& colour, int width, int style)
 {
     m_refData = new wxPenRefData();
diff -uNr wxWidgets-3.0.0-orig\src/msw/brush.cpp wxWidgets-3.0.0-15792\src/msw/brush.cpp
--- wxWidgets-3.0.0-orig\src/msw/brush.cpp	2013-11-11 14:18:38.000000000 +0100
+++ wxWidgets-3.0.0-15792\src/msw/brush.cpp	2014-01-04 17:13:14.981594300 +0100
@@ -232,7 +232,7 @@
     m_refData = new wxBrushRefData(col, style);
 }
 
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
+#if WXWIN_COMPATIBILITY_3_0
 wxBrush::wxBrush(const wxColour& col, int style)
 {
     m_refData = new wxBrushRefData(col, (wxBrushStyle)style);
diff -uNr wxWidgets-3.0.0-orig\src/msw/pen.cpp wxWidgets-3.0.0-15792\src/msw/pen.cpp
--- wxWidgets-3.0.0-orig\src/msw/pen.cpp	2013-11-11 14:18:38.000000000 +0100
+++ wxWidgets-3.0.0-15792\src/msw/pen.cpp	2014-01-04 17:13:28.685081500 +0100
@@ -432,7 +432,7 @@
     m_refData = new wxPenRefData(col, width, style);
 }
 
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
+#if WXWIN_COMPATIBILITY_3_0
 wxPen::wxPen(const wxColour& colour, int width, int style)
 {
     m_refData = new wxPenRefData(colour, width, (wxPenStyle)style);
diff -uNr wxWidgets-3.0.0-orig\src/os2/brush.cpp wxWidgets-3.0.0-15792\src/os2/brush.cpp
--- wxWidgets-3.0.0-orig\src/os2/brush.cpp	2013-11-11 14:18:38.000000000 +0100
+++ wxWidgets-3.0.0-15792\src/os2/brush.cpp	2014-01-04 17:23:56.224851800 +0100
@@ -116,7 +116,7 @@
     RealizeResource();
 } // end of wxBrush::wxBrush
 
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
+#if WXWIN_COMPATIBILITY_3_0
 wxBrush::wxBrush(const wxColour& col, int style)
 {
     m_refData = new wxBrushRefData(col, (wxBrushStyle)style);
diff -uNr wxWidgets-3.0.0-orig\src/os2/pen.cpp wxWidgets-3.0.0-15792\src/os2/pen.cpp
--- wxWidgets-3.0.0-orig\src/os2/pen.cpp	2013-11-11 14:18:39.000000000 +0100
+++ wxWidgets-3.0.0-15792\src/os2/pen.cpp	2014-01-04 17:24:07.589651800 +0100
@@ -148,7 +148,7 @@
     RealizeResource();
 } // end of wxPen::wxPen
 
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
+#if WXWIN_COMPATIBILITY_3_0
 wxPen::wxPen(const wxColour& colour, int width, int style)
 {
     m_refData = new wxPenRefData(colour, width, (wxPenStyle)style);
diff -uNr wxWidgets-3.0.0-orig\src/osx/brush.cpp wxWidgets-3.0.0-15792\src/osx/brush.cpp
--- wxWidgets-3.0.0-orig\src/osx/brush.cpp	2013-11-11 14:18:39.000000000 +0100
+++ wxWidgets-3.0.0-15792\src/osx/brush.cpp	2014-01-04 17:24:20.190364500 +0100
@@ -99,7 +99,7 @@
     m_refData = new wxBrushRefData( col, style );
 }
 
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
+#if WXWIN_COMPATIBILITY_3_0
 wxBrush::wxBrush(const wxColour& col, int style)
 {
     m_refData = new wxBrushRefData(col, (wxBrushStyle)style);
diff -uNr wxWidgets-3.0.0-orig\src/osx/pen.cpp wxWidgets-3.0.0-15792\src/osx/pen.cpp
--- wxWidgets-3.0.0-orig\src/osx/pen.cpp	2013-11-11 14:18:39.000000000 +0100
+++ wxWidgets-3.0.0-15792\src/osx/pen.cpp	2014-01-04 17:24:27.595243100 +0100
@@ -111,7 +111,7 @@
     RealizeResource();
 }
 
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
+#if WXWIN_COMPATIBILITY_3_0
 wxPen::wxPen(const wxColour& col, int Width, int Style)
 {
     m_refData = new wxPenRefData;
diff -uNr wxWidgets-3.0.0-orig\src/x11/brush.cpp wxWidgets-3.0.0-15792\src/x11/brush.cpp
--- wxWidgets-3.0.0-orig\src/x11/brush.cpp	2013-11-11 14:18:39.000000000 +0100
+++ wxWidgets-3.0.0-15792\src/x11/brush.cpp	2014-01-04 17:24:34.874330500 +0100
@@ -63,7 +63,7 @@
     M_BRUSHDATA->m_colour = colour;
 }
 
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
+#if WXWIN_COMPATIBILITY_3_0
 wxBrush::wxBrush(const wxColour& col, int style)
 {
     m_refData = new wxBrushRefData;
diff -uNr wxWidgets-3.0.0-orig\src/x11/pen.cpp wxWidgets-3.0.0-15792\src/x11/pen.cpp
--- wxWidgets-3.0.0-orig\src/x11/pen.cpp	2013-11-11 14:18:39.000000000 +0100
+++ wxWidgets-3.0.0-15792\src/x11/pen.cpp	2014-01-04 17:24:42.069823800 +0100
@@ -85,7 +85,7 @@
     M_PENDATA->m_colour = colour;
 }
 
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
+#if WXWIN_COMPATIBILITY_3_0
 wxPen::wxPen(const wxColour& colour, int width, int style)
 {
     m_refData = new wxPenRefData();
diff -uNr wxWidgets-3.0.0-orig\tests/font/fonttest.cpp wxWidgets-3.0.0-15792\tests/font/fonttest.cpp
--- wxWidgets-3.0.0-orig\tests/font/fonttest.cpp	2013-11-11 14:18:39.000000000 +0100
+++ wxWidgets-3.0.0-15792\tests/font/fonttest.cpp	2014-01-04 17:24:55.937245900 +0100
@@ -106,11 +106,11 @@
                                wxFONTSTYLE_NORMAL,
                                wxFONTWEIGHT_NORMAL).IsOk() );
 
-#if FUTURE_WXWIN_COMPATIBILITY_3_0
+#if WXWIN_COMPATIBILITY_3_0
     // Tests relying on the soon-to-be-deprecated ctor taking ints and not
     // wxFontXXX enum elements.
     CPPUNIT_ASSERT( wxFont(10, wxDEFAULT, wxNORMAL, wxNORMAL).IsOk() );
-#endif // FUTURE_WXWIN_COMPATIBILITY_3_0
+#endif // WXWIN_COMPATIBILITY_3_0
 }
 
 void FontTestCase::GetSet()
