diff -ru2 -xCVS -x.#* -x*.orig -x*.rej -x*~ wxWindows.orig/src/msw/toplevel.cpp wxWindows/src/msw/toplevel.cpp
--- wxWindows.orig/src/msw/toplevel.cpp	Thu Sep 18 12:53:48 2003
+++ wxWindows/src/msw/toplevel.cpp	Thu Sep 18 13:01:17 2003
@@ -236,5 +236,5 @@
             // implementation
 #ifndef __WXWINCE__
-            if ( !(style & wxFRAME_NO_TASKBAR) && GetParent() )
+            if ( !(style & wxFRAME_NO_TASKBAR) && GetParent() && !(style & wxFRAME_TOOL_WINDOW) )
             {
                 // need to force the frame to appear in the taskbar

 	  	 
