--- samples/image/image_orig.cpp	2009-11-28 18:22:02.000000000 -0800
+++ samples/image/image.cpp	2009-11-28 19:09:30.000000000 -0800
@@ -274,6 +274,7 @@
                 }
             }
         }
+#if wxUSE_LIBPNG
         else if ( extension == wxT("png") )
         {
             static const int pngvalues[] =
@@ -344,6 +345,7 @@
                 }
             }
         }
+#endif
         else if ( extension == wxT("cur") )
         {
             image.Rescale(32,32);
