Index: include/wx/propgrid/props.h
===================================================================
--- include/wx/propgrid/props.h	(révision 70939)
+++ include/wx/propgrid/props.h	(copie de travail)
@@ -564,7 +564,7 @@
     // helpers
     size_t GetItemCount() const { return m_choices.GetCount(); }
     const wxString& GetLabel( size_t ind ) const
-        { return m_choices.GetLabel(ind); }
+        { return m_choices.GetLabel((int)ind); }
 
 protected:
     // Used to detect if choices have been changed
