Index: include/wx/generic/grid.h
===================================================================
--- include/wx/generic/grid.h	(révision 70939)
+++ include/wx/generic/grid.h	(copie de travail)
@@ -811,7 +811,7 @@
 
     // these are pure virtual in wxGridTableBase
     //
-    virtual int GetNumberRows() { return m_data.size(); }
+    virtual int GetNumberRows() { return (int)m_data.size(); }
     virtual int GetNumberCols() { return m_numCols; }
     virtual wxString GetValue( int row, int col );
     virtual void SetValue( int row, int col, const wxString& s );
