diff -uNr wxWidgets-trunk\src/msw/toolbar.cpp wxWidgets-work\src/msw/toolbar.cpp
--- wxWidgets-trunk\src/msw/toolbar.cpp	2014-01-01 21:36:08.222811700 +0100
+++ wxWidgets-work\src/msw/toolbar.cpp	2014-01-14 17:43:23.841526300 +0100
@@ -789,6 +789,10 @@
 
                     // notice the last parameter: do use mask
                     dcAllButtons.DrawBitmap(bmp, x + xOffset, yOffset, true);
+                    // Handle of output bitmap can be changed in the background
+                    // during DDB <-> DIB conversion (in AlphaBlt function) if this bitmap
+                    // is a DDB and drawn bitmap has alpha channel.
+                    m_hBitmap = hBitmap = bitmap.GetHBITMAP();
                 }
                 else
                 {
