diff -uNr wxWidgets-trunk\src/msw/toolbar.cpp wxWidgets-work\src/msw/toolbar.cpp
--- wxWidgets-trunk\src/msw/toolbar.cpp	2014-01-01 21:36:08.222811700 +0100
+++ wxWidgets-work\src/msw/toolbar.cpp	2014-01-15 15:59:45.047984900 +0100
@@ -730,7 +730,8 @@
 
         // Create a bitmap and copy all the tool bitmaps into it
         wxMemoryDC dcAllButtons;
-        wxBitmap bitmap(totalBitmapWidth, totalBitmapHeight);
+        wxBitmap bitmap(totalBitmapWidth, totalBitmapHeight, 32);
+        bitmap.UseAlpha();
         dcAllButtons.SelectObject(bitmap);
 
 #ifdef wxREMAP_BUTTON_COLOURS
