Index: interface/wx/html/htmlwin.h
===================================================================
--- interface/wx/html/htmlwin.h	(revision 69318)
+++ interface/wx/html/htmlwin.h	(working copy)
@@ -12,14 +12,18 @@
     wxHtmlWindow is probably the only class you will directly use unless you want
     to do something special (like adding new tag handlers or MIME filters).
 
-    The purpose of this class is to display HTML pages (either local file or
-    downloaded via HTTP protocol) in a window.
+    The purpose of this class is to display rich content pages (either local file or
+    downloaded via HTTP protocol) in a window based on a subset of the HTML standard.
     The width of the window is constant - given in the constructor - and virtual height
     is changed dynamically depending on page size.
     Once the window is created you can set its content by calling SetPage() with raw HTML,
     LoadPage() with a wxFileSystem location or LoadFile() with a filename.
 
     @note
+    If you want complete HTML/CSS support as well as a Javascript engine, see instead
+    wxWebView.
+
+    @note
     wxHtmlWindow uses the wxImage class for displaying images, as such you need to
     initialize the handlers for any image formats you use before loading a page.
     See ::wxInitAllImageHandlers and wxImage::AddHandler.
