diff -ru wxWidgets.orig/src/msw/cursor.cpp wxWidgets/src/msw/cursor.cpp
--- wxWidgets.orig/src/msw/cursor.cpp	2005-05-23 14:51:03.000000000 +1000
+++ wxWidgets/src/msw/cursor.cpp	2005-05-24 02:35:49.051375000 +1000
@@ -202,13 +202,13 @@
     }
     else if ((w != image_w) || (h != image_h))
     {
-        hotSpotX = int(hotSpotX * double(w) / double(image_w)); 
-        hotSpotY = int(hotSpotY * double(h) / double(image_h)); 
+        hotSpotX = int(hotSpotX * double(w) / double(image_w));
+        hotSpotY = int(hotSpotY * double(h) / double(image_h));
 
         imageSized = image.Scale(w, h);
     }
 
-    HCURSOR hcursor = wxBitmapToHCURSOR( wxBitmap(imageSized), 
+    HCURSOR hcursor = wxBitmapToHCURSOR( wxBitmap(imageSized),
                                          hotSpotX, hotSpotY );
 
     if ( !hcursor )
@@ -357,7 +357,8 @@
     }
     else
     {
-        m_refData = new wxCursorRefData(hcursor);
+        m_refData = new wxCursorRefData(hcursor,
+                                        !stdCursor.isStd /* delete it later */);
     }
 }
 

 	  	 
