Index: samples/popup/popup.cpp
===================================================================
--- samples/popup/popup.cpp	(revision 56688)
+++ samples/popup/popup.cpp	(working copy)
@@ -151,7 +151,7 @@
 {
 }
 
-void SimpleTransientPopup::Popup(wxWindow *focus)
+void SimpleTransientPopup::Popup(wxWindow* WXUNUSED(focus))
 {
     wxLogMessage( wxT("0x%lx SimpleTransientPopup::Popup"), long(this) );
     wxPopupTransientWindow::Popup();
@@ -380,7 +380,7 @@
 
 // event handlers
 
-void MyFrame::OnActivate(wxActivateEvent& event)
+void MyFrame::OnActivate(wxActivateEvent& WXUNUSED(event))
 {
     wxLogMessage( wxT("In activate...") );
 }
