Index: samples/samples.dsw
===================================================================
--- samples/samples.dsw	(revision 56688)
+++ samples/samples.dsw	(working copy)
@@ -591,6 +591,18 @@
 
 ###############################################################################
 
+Project: "popup"=.\popup\popup.dsp - Package Owner=<4>
+
+Package=<5>
+{{{
+}}}
+
+Package=<4>
+{{{
+}}}
+
+###############################################################################
+
 Project: "printing"=.\printing\printing.dsp - Package Owner=<4>
 
 Package=<5>
