Index: wxWidgets/src/osx/carbon/dataview.cpp
===================================================================
--- wxWidgets/src/osx/carbon/dataview.cpp	(revision 54384)
+++ wxWidgets/src/osx/carbon/dataview.cpp	(working copy)
@@ -475,6 +475,7 @@
     wxDataViewItemArray array;
     GetOwner()->GetChildren( item, array );
     ItemsAdded( item, array );
+    this->m_dataViewControlPtr->SetScrollPosition(0, 0);
     return noFailureFlag;
   } /* Cleared() */
 
