Index: src/osx/cocoa/filedlg.mm
===================================================================
--- src/osx/cocoa/filedlg.mm	(revision 65218)
+++ src/osx/cocoa/filedlg.mm	(working copy)
@@ -290,6 +290,11 @@
         [oPanel setCanChooseFiles:YES];
         [oPanel setMessage:cf.AsNSString()];
 
+        if(HasFlag(wxFD_MULTIPLE))
+        {
+            [oPanel setAllowsMultipleSelection:YES];
+        }
+
         returnCode = [oPanel runModalForDirectory:dir.AsNSString()
                         file:file.AsNSString() types:types];
 
