From f479fb390722ca5a3753a5afb3375822d9da3bfb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Gunter=20K=C3=B6nigsmann?= <gunter@peterpall.de>
Date: Sun, 9 Jun 2019 09:59:08 +0200
Subject: [PATCH] A crash with wxFileSystem

---
 samples/minimal/minimal.cpp | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/samples/minimal/minimal.cpp b/samples/minimal/minimal.cpp
index 0d91f7fc75..cba2150459 100644
--- a/samples/minimal/minimal.cpp
+++ b/samples/minimal/minimal.cpp
@@ -29,6 +29,9 @@
     #include "wx/wx.h"
 #endif
 
+#include <wx/filesys.h>
+#include <wx/fs_mem.h>
+
 // ----------------------------------------------------------------------------
 // resources
 // ----------------------------------------------------------------------------
@@ -147,6 +150,18 @@ MyFrame::MyFrame(const wxString& title)
     // set the frame icon
     SetIcon(wxICON(sample));
 
+    // This code doesn't return a valid wxString
+    wxFileSystem *fsystem = new wxFileSystem();
+    fsystem->AddHandler(new wxMemoryFSHandler);
+    fsystem->ChangePathTo(wxT("memory:"), true);
+    wxString memFsName = fsystem->FindFirst("*", wxFILE);
+    while(memFsName != wxEmptyString)
+      {
+        wxString name = memFsName.Right(memFsName.Length()-7);
+        wxMemoryFSHandler::RemoveFile(name);
+        memFsName = fsystem->FindNext();
+      }
+
 #if wxUSE_MENUS
     // create a menu bar
     wxMenu *fileMenu = new wxMenu;
-- 
2.20.1

