Index: wxWidgets-2.8/src/common/sizer.cpp
===================================================================
--- wxWidgets-2.8/src/common/sizer.cpp	(revision 48164)
+++ wxWidgets-2.8/src/common/sizer.cpp	(working copy)
@@ -516,6 +516,9 @@
     if ( item->GetWindow() )
         item->GetWindow()->SetContainingSizer( this );
 
+    if ( item->GetSizer() )
+        item->GetSizer()->SetContainingWindow( m_containingWindow );
+
     return item;
 }
 

 	  	 
