Index: include/wx/docview.h
===================================================================
--- include/wx/docview.h	(revision 68076)
+++ include/wx/docview.h	(working copy)
@@ -143,6 +143,7 @@
     wxList& GetViews() { return m_documentViews; }
     const wxList& GetViews() const { return m_documentViews; }
     wxView *GetFirstView() const;
+    wxView *GetCurrentView() const;
 
     virtual void UpdateAllViews(wxView *sender = NULL, wxObject *hint = NULL);
     virtual void NotifyClosing();
Index: interface/wx/docview.h
===================================================================
--- interface/wx/docview.h	(revision 68076)
+++ interface/wx/docview.h	(working copy)
@@ -1277,6 +1277,16 @@
     wxView* GetFirstView() const;
 
     /**
+        A convenience function to get the current, or active, view for the document.
+        If no view is activated, the first view is returned instead.
+
+        @see GetViews(), GetFirstView()
+
+        @since 2.9.3
+    */
+    wxView* GetCurrentView() const;
+
+    /**
         Gets the title for this document. The document title is used for an
         associated frame (if any), and is usually constructed by the framework
         from the filename.
Index: src/common/docview.cpp
===================================================================
--- src/common/docview.cpp	(revision 68076)
+++ src/common/docview.cpp	(working copy)
@@ -258,6 +258,13 @@
     return static_cast<wxView *>(m_documentViews.GetFirst()->GetData());
 }
 
+wxView *wxDocument::GetCurrentView() const
+{
+    wxView* view = GetDocumentManager()->GetCurrentView();
+
+    return (view && (view->GetDocument() == this)) ? view : GetFirstView();
+}
+
 void wxDocument::Modify(bool mod)
 {
     if (mod != m_documentModified)
