--- usr/src/w/wxwidgets.trunk/samples/notebook/notebook.cpp 
+++ home/jens/wxWidgets/samples/notebook/notebook.cpp 
@@ -518,11 +518,6 @@
 
     m_bookCtrl->Hide();
 
-    if ( m_chkShowImages )
-    {
-        m_bookCtrl->SetImageList(m_imageList);
-    }
-
     if ( oldBook )
     {
 #if wxUSE_TREEBOOK
@@ -571,6 +566,17 @@
     {
         CreateInitialPages(m_bookCtrl);
     }
+
+    if ( m_chkShowImages )
+    {
+        m_bookCtrl->SetImageList(m_imageList);
+        int images = m_imageList->GetImageCount();
+        for (size_t i = 0; i < m_bookCtrl->GetPageCount(); ++i)
+        {
+            m_bookCtrl->SetPageImage(i, i % images);
+        }
+    }
+
 
     m_sizerFrame->Insert(0, m_bookCtrl, wxSizerFlags(5).Expand().Border());
 

