Index: e:/dev/wxWindows/src/univ/winuniv.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/univ/winuniv.cpp,v
retrieving revision 1.33.2.1
diff -u -2 -r1.33.2.1 winuniv.cpp
--- e:/dev/wxWindows/src/univ/winuniv.cpp	2002/09/21 20:38:05	1.33.2.1
+++ e:/dev/wxWindows/src/univ/winuniv.cpp	2002/12/03 20:42:16
@@ -285,8 +285,8 @@
 
         // Adjust DC logical origin
-        wxCoord x,y;
-        dc.GetLogicalOrigin( &x, &y );
-        x += pos.x;
-        y += pos.y;
+        wxCoord org_x, org_y, x, y;
+        dc.GetLogicalOrigin( &org_x, &org_y );
+        x = org_x + pos.x;
+        y = org_y + pos.y;
         dc.SetLogicalOrigin( x, y );
 
@@ -297,4 +297,7 @@
         // Let parent draw the background
         parent->EraseBackground( dc, rect );
+
+        // Restore DC logical origin
+        dc.SetLogicalOrigin( org_x, org_y );
     }
     else

 	  	 
