--- radiobox.cpp.Old	2006-07-19 12:40:28.000000000 +0100
+++ radiobox.cpp	2006-07-19 12:43:47.000000000 +0100
@@ -218,4 +218,6 @@
 
     m_widget = gtk_frame_new( wxGTK_CONV( title ) );
+    if ( HasFlag(wxNO_BORDER) )  // If we don't do this here, the wxNO_BORDER style is ignored in Show()
+        gtk_frame_set_shadow_type ( GTK_FRAME(m_widget), GTK_SHADOW_NONE );
 
     // majorDim may be 0 if all trailing parameters were omitted, so don't

 	  	 
