--- radioboxgtk1.cpp	2006-07-19 13:57:03.000000000 +0100
+++ radiobox.cpp	2006-07-19 14:09:34.000000000 +0100
@@ -203,4 +203,6 @@
     m_widget = gtk_frame_new(NULL);
     SetLabel(title);
+    if ( HasFlag(wxNO_BORDER) )  // If we don't do this here, the wxNO_BORDER style is ignored in Show()
+        gtk_frame_set_shadow_type ( GTK_FRAME(m_widget), GTK_SHADOW_NONE );
 
     // majorDim may be 0 if all trailing parameters were omitted, so don't

 	  	 
