Index: samples/aui/auidemo.cpp
===================================================================
--- samples/aui/auidemo.cpp	(revision 61205)
+++ samples/aui/auidemo.cpp	(working copy)
@@ -897,6 +897,12 @@
                   Name(wxT("test8")).Caption(wxT("Tree Pane")).
                   Left().Layer(1).Position(1).
                   CloseButton(true).MaximizeButton(true));
+    wxWindow* tree = m_mgr.GetPane("test8").window;
+    wxRect clientRect = tree->GetClientRect();
+    wxPoint clientCenter(clientRect.GetLeft() + clientRect.GetWidth()/2, clientRect.GetTop() + clientRect.GetHeight()/2);
+    wxPoint screenCenter(tree->ClientToScreen(clientCenter));
+    wxWindow* test = wxFindWindowAtPoint(screenCenter);
+    wxASSERT(test == tree);     // fails to find the tree ctrl!
 
     m_mgr.AddPane(CreateSizeReportCtrl(), wxAuiPaneInfo().
                   Name(wxT("test9")).Caption(wxT("Min Size 200x100")).
