Index: samples/minimal/minimal.cpp
===================================================================
--- samples/minimal/minimal.cpp	(revision 61487)
+++ samples/minimal/minimal.cpp	(working copy)
@@ -19,7 +19,7 @@
  
 // For compilers that support precompilation, includes "wx/wx.h".
 #include "wx/wxprec.h"
- 
+#include <wx/spinctrl.h>
 #ifdef __BORLANDC__
     #pragma hdrstop
 #endif
@@ -67,10 +67,12 @@
     // event handlers (these functions should _not_ be virtual)
     void OnQuit(wxCommandEvent& event);
     void OnAbout(wxCommandEvent& event);
+    void OnUpdateUI(wxUpdateUIEvent& event);
 
 private:
     // any class wishing to process wxWidgets events must use this macro
     DECLARE_EVENT_TABLE()
+    wxSpinCtrl* m_spin;
 };
 
 // ----------------------------------------------------------------------------
@@ -99,6 +101,7 @@
 BEGIN_EVENT_TABLE(MyFrame, wxFrame)
     EVT_MENU(Minimal_Quit,  MyFrame::OnQuit)
     EVT_MENU(Minimal_About, MyFrame::OnAbout)
+    EVT_UPDATE_UI(wxID_ANY, MyFrame::OnUpdateUI)
 END_EVENT_TABLE()
 
 // Create a new application object: this macro will allow wxWidgets to create
@@ -172,6 +175,9 @@
     CreateStatusBar(2);
     SetStatusText("Welcome to wxWidgets!");
 #endif // wxUSE_STATUSBAR
+
+    m_spin = new wxSpinCtrl(this, wxID_ANY);
+    SetSizer(new wxBoxSizer(wxVERTICAL)); // Just to prevent the spinctrl filling the frame
 }
 
 
@@ -198,3 +204,9 @@
                  wxOK | wxICON_INFORMATION,
                  this);
 }
+
+void MyFrame::OnUpdateUI(wxUpdateUIEvent& WXUNUSED(event))
+{
+    m_spin->GetValue();
+    //m_spin->SetRange(1,100); // This would also cause 100% cpu
+}
