Index: tests/test.bkl
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/tests/test.bkl,v
retrieving revision 1.23
diff -u -2 -r1.23 test.bkl
--- tests/test.bkl	2004/11/10 23:58:24	1.23
+++ tests/test.bkl	2004/11/17 18:32:55
@@ -8,4 +8,15 @@
     <exe id="test" template="wx_test_console"
                    template_append="wx_append_base">
+
+        <if cond="WX_DISABLE_PRECOMP_HEADERS=='0'">
+            <if cond="FORMAT!='autoconf' and TOOLKIT=='MSW'">
+                <sources>dummy.cpp</sources>
+                <precomp-headers-gen>dummy.cpp</precomp-headers-gen>
+            </if>
+            <precomp-headers-header>testprec.h</precomp-headers-header>
+            <precomp-headers>on</precomp-headers>
+            <precomp-headers-file>testprec_$(id)</precomp-headers-file>
+        </if>
+
         <sources>
             test.cpp
Index: tests/test.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/tests/test.cpp,v
retrieving revision 1.8
diff -u -2 -r1.8 test.cpp
--- tests/test.cpp	2004/11/10 21:28:54	1.8
+++ tests/test.cpp	2004/11/17 18:32:55
@@ -8,11 +8,7 @@
 ///////////////////////////////////////////////////////////////////////////////
 
-#if defined(__GNUG__) && !defined(__APPLE__)
-    #pragma implementation
-    #pragma interface
-#endif
-
-// For compilers that support precompilation, includes "wx/wx.h".
-#include "wx/wxprec.h"
+// For compilers that support precompilation, includes "wx/wx.h"
+// and "wx/cppunit.h"
+#include "testprec.h"
 
 #ifdef __BORLANDC__
@@ -26,5 +22,4 @@
 
 #include "wx/cmdline.h"
-#include "wx/cppunit.h"
 #include <iostream>
 
@@ -115,5 +110,4 @@
 }
 
-#include "wx/uri.h"
 // Run
 //
Index: tests/archive/archivetest.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/tests/archive/archivetest.cpp,v
retrieving revision 1.3
diff -u -2 -r1.3 archivetest.cpp
--- tests/archive/archivetest.cpp	2004/11/12 14:08:31	1.3
+++ tests/archive/archivetest.cpp	2004/11/17 18:32:59
@@ -8,5 +8,5 @@
 ///////////////////////////////////////////////////////////////////////////////
 
-#include "wx/wxprec.h"
+#include "testprec.h"
 
 #ifdef __BORLANDC__
@@ -26,5 +26,4 @@
 #include "wx/wfstream.h"
 #include "wx/dir.h"
-#include "wx/cppunit.h"
 #include <string>
 #include <list>
Index: tests/arrays/arrays.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/tests/arrays/arrays.cpp,v
retrieving revision 1.2
diff -u -2 -r1.2 arrays.cpp
--- tests/arrays/arrays.cpp	2004/04/06 12:14:05	1.2
+++ tests/arrays/arrays.cpp	2004/11/17 18:33:00
@@ -12,5 +12,5 @@
 // ----------------------------------------------------------------------------
 
-#include "wx/wxprec.h"
+#include "testprec.h"
 
 #ifdef __BORLANDC__
@@ -23,6 +23,4 @@
 
 #include "wx/dynarray.h"
-
-#include "wx/cppunit.h"
 
 // ----------------------------------------------------------------------------
Index: tests/datetime/datetimetest.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/tests/datetime/datetimetest.cpp,v
retrieving revision 1.2
diff -u -2 -r1.2 datetimetest.cpp
--- tests/datetime/datetimetest.cpp	2004/11/10 09:31:13	1.2
+++ tests/datetime/datetimetest.cpp	2004/11/17 18:33:02
@@ -12,5 +12,5 @@
 // ----------------------------------------------------------------------------
 
-#include "wx/wxprec.h"
+#include "testprec.h"
 
 #ifdef __BORLANDC__
@@ -22,6 +22,4 @@
 
 #include "wx/datetime.h"
-
-#include "wx/cppunit.h"
 
 // ----------------------------------------------------------------------------
Index: tests/fileconf/fileconftest.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/tests/fileconf/fileconftest.cpp,v
retrieving revision 1.1
diff -u -2 -r1.1 fileconftest.cpp
--- tests/fileconf/fileconftest.cpp	2004/10/31 04:40:40	1.1
+++ tests/fileconf/fileconftest.cpp	2004/11/17 18:33:03
@@ -12,5 +12,5 @@
 // ----------------------------------------------------------------------------
 
-#include "wx/wxprec.h"
+#include "testprec.h"
 
 #ifdef __BORLANDC__
@@ -25,6 +25,4 @@
 #include "wx/fileconf.h"
 #include "wx/sstream.h"
-
-#include "wx/cppunit.h"
 
 static const wxChar *testconfig =
Index: tests/filename/filenametest.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/tests/filename/filenametest.cpp,v
retrieving revision 1.2
diff -u -2 -r1.2 filenametest.cpp
--- tests/filename/filenametest.cpp	2004/11/07 20:11:42	1.2
+++ tests/filename/filenametest.cpp	2004/11/17 18:33:03
@@ -12,5 +12,5 @@
 // ----------------------------------------------------------------------------
 
-#include "wx/wxprec.h"
+#include "testprec.h"
 
 #ifdef __BORLANDC__
@@ -22,6 +22,4 @@
 
 #include "wx/filename.h"
-
-#include "wx/cppunit.h"
 
 // ----------------------------------------------------------------------------
Index: tests/filesys/filesystest.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/tests/filesys/filesystest.cpp,v
retrieving revision 1.1
diff -u -2 -r1.1 filesystest.cpp
--- tests/filesys/filesystest.cpp	2004/10/31 04:40:54	1.1
+++ tests/filesys/filesystest.cpp	2004/11/17 18:33:03
@@ -12,5 +12,5 @@
 // ----------------------------------------------------------------------------
 
-#include "wx/wxprec.h"
+#include "testprec.h"
 
 #ifdef __BORLANDC__
@@ -23,6 +23,4 @@
 
 #include "wx/filesys.h"
-
-#include "wx/cppunit.h"
 
 #if wxUSE_FILESYSTEM
Index: tests/fontmap/fontmaptest.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/tests/fontmap/fontmaptest.cpp,v
retrieving revision 1.1
diff -u -2 -r1.1 fontmaptest.cpp
--- tests/fontmap/fontmaptest.cpp	2004/10/31 04:41:01	1.1
+++ tests/fontmap/fontmaptest.cpp	2004/11/17 18:33:03
@@ -12,5 +12,5 @@
 // ----------------------------------------------------------------------------
 
-#include "wx/wxprec.h"
+#include "testprec.h"
 
 #ifdef __BORLANDC__
@@ -25,6 +25,4 @@
 
 #include "wx/fontmap.h"
-
-#include "wx/cppunit.h"
 
 // ----------------------------------------------------------------------------
Index: tests/formatconverter/formatconvertertest.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/tests/formatconverter/formatconvertertest.cpp,v
retrieving revision 1.2
diff -u -2 -r1.2 formatconvertertest.cpp
--- tests/formatconverter/formatconvertertest.cpp	2004/11/10 21:28:59	1.2
+++ tests/formatconverter/formatconvertertest.cpp	2004/11/17 18:33:04
@@ -37,5 +37,6 @@
 
 // For compilers that support precompilation, includes "wx/wx.h".
-#include "wx/wxprec.h"
+// and "wx/cppunit.h"
+#include "testprec.h"
 
 #ifdef __BORLANDC__
@@ -47,6 +48,4 @@
     #include "wx/wx.h"
 #endif
-
-#include "wx/cppunit.h"
 
 // wxFormatConverter can only be tested in a Unicode non-Windows debug build
Index: tests/hashes/hashes.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/tests/hashes/hashes.cpp,v
retrieving revision 1.1
diff -u -2 -r1.1 hashes.cpp
--- tests/hashes/hashes.cpp	2004/05/22 17:51:32	1.1
+++ tests/hashes/hashes.cpp	2004/11/17 18:33:05
@@ -12,5 +12,5 @@
 // ----------------------------------------------------------------------------
 
-#include "wx/wxprec.h"
+#include "testprec.h"
 
 #ifdef __BORLANDC__
@@ -25,6 +25,4 @@
 #include "wx/hashmap.h"
 #include "wx/hashset.h"
-
-#include "wx/cppunit.h"
 
 // --------------------------------------------------------------------------
Index: tests/longlong/longlongtest.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/tests/longlong/longlongtest.cpp,v
retrieving revision 1.1
diff -u -2 -r1.1 longlongtest.cpp
--- tests/longlong/longlongtest.cpp	2004/10/31 04:41:11	1.1
+++ tests/longlong/longlongtest.cpp	2004/11/17 18:33:05
@@ -12,5 +12,5 @@
 // ----------------------------------------------------------------------------
 
-#include "wx/wxprec.h"
+#include "testprec.h"
 
 #ifdef __BORLANDC__
@@ -24,6 +24,4 @@
 #include "wx/longlong.h"
 #include "wx/timer.h"
-
-#include "wx/cppunit.h"
 
 // ----------------------------------------------------------------------------
Index: tests/mbconv/mbconvtest.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/tests/mbconv/mbconvtest.cpp,v
retrieving revision 1.1
diff -u -2 -r1.1 mbconvtest.cpp
--- tests/mbconv/mbconvtest.cpp	2004/10/31 04:41:17	1.1
+++ tests/mbconv/mbconvtest.cpp	2004/11/17 18:33:06
@@ -12,5 +12,5 @@
 // ----------------------------------------------------------------------------
 
-#include "wx/wxprec.h"
+#include "testprec.h"
 
 #ifdef __BORLANDC__
@@ -24,6 +24,4 @@
 #include "wx/strconv.h"
 #include "wx/string.h"
-
-#include "wx/cppunit.h"
 
 // ----------------------------------------------------------------------------
Index: tests/regex/regextest.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/tests/regex/regextest.cpp,v
retrieving revision 1.2
diff -u -2 -r1.2 regextest.cpp
--- tests/regex/regextest.cpp	2004/11/10 21:29:03	1.2
+++ tests/regex/regextest.cpp	2004/11/17 18:33:07
@@ -31,5 +31,5 @@
 
 // For compilers that support precompilation, includes "wx/wx.h".
-#include "wx/wxprec.h"
+#include "testprec.h"
 
 #ifdef __BORLANDC__
@@ -43,5 +43,4 @@
 
 #include "wx/regex.h"
-#include "wx/cppunit.h"
 #include <string>
 #include <vector>
Index: tests/regex/wxregextest.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/tests/regex/wxregextest.cpp,v
retrieving revision 1.3
diff -u -2 -r1.3 wxregextest.cpp
--- tests/regex/wxregextest.cpp	2004/11/10 21:29:04	1.3
+++ tests/regex/wxregextest.cpp	2004/11/17 18:33:07
@@ -8,5 +8,5 @@
 ///////////////////////////////////////////////////////////////////////////////
 
-#include "wx/wxprec.h"
+#include "testprec.h"
 
 #ifdef __BORLANDC__
@@ -21,5 +21,4 @@
 
 #include "wx/regex.h"
-#include "wx/cppunit.h"
 #include "wx/tokenzr.h"
 #include <string>
Index: tests/streams/bstream.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/tests/streams/bstream.cpp,v
retrieving revision 1.4
diff -u -2 -r1.4 bstream.cpp
--- tests/streams/bstream.cpp	2004/11/10 21:29:07	1.4
+++ tests/streams/bstream.cpp	2004/11/17 18:33:07
@@ -8,11 +8,7 @@
 ///////////////////////////////////////////////////////////////////////////////
 
-#if defined(__GNUG__) && !defined(__APPLE__)
-    #pragma implementation
-    #pragma interface
-#endif
-
 // For compilers that support precompilation, includes "wx/wx.h".
-#include "wx/wxprec.h"
+// and "wx/cppunit.h"
+#include "testprec.h"
 
 #ifdef __BORLANDC__
@@ -25,5 +21,4 @@
 #endif
 
-#include "wx/cppunit.h"
 #include "bstream.h"
 
Index: tests/streams/ffilestream.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/tests/streams/ffilestream.cpp,v
retrieving revision 1.4
diff -u -2 -r1.4 ffilestream.cpp
--- tests/streams/ffilestream.cpp	2004/11/10 21:29:08	1.4
+++ tests/streams/ffilestream.cpp	2004/11/17 18:33:08
@@ -8,11 +8,7 @@
 ///////////////////////////////////////////////////////////////////////////////
 
-#if defined(__GNUG__) && !defined(__APPLE__)
-    #pragma implementation
-    #pragma interface
-#endif
-
 // For compilers that support precompilation, includes "wx/wx.h".
-#include "wx/wxprec.h"
+// and "wx/cppunit.h"
+#include "testprec.h"
 
 #ifdef __BORLANDC__
@@ -25,5 +21,4 @@
 #endif
 
-#include "wx/cppunit.h"
 #include "wx/wfstream.h"
 
Index: tests/streams/filestream.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/tests/streams/filestream.cpp,v
retrieving revision 1.3
diff -u -2 -r1.3 filestream.cpp
--- tests/streams/filestream.cpp	2004/11/10 21:29:08	1.3
+++ tests/streams/filestream.cpp	2004/11/17 18:33:08
@@ -8,11 +8,7 @@
 ///////////////////////////////////////////////////////////////////////////////
 
-#if defined(__GNUG__) && !defined(__APPLE__)
-    #pragma implementation
-    #pragma interface
-#endif
-
 // For compilers that support precompilation, includes "wx/wx.h".
-#include "wx/wxprec.h"
+// and "wx/cppunit.h"
+#include "testprec.h"
 
 #ifdef __BORLANDC__
@@ -25,5 +21,4 @@
 #endif
 
-#include "wx/cppunit.h"
 #include "wx/wfstream.h"
 
Index: tests/streams/memstream.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/tests/streams/memstream.cpp,v
retrieving revision 1.2
diff -u -2 -r1.2 memstream.cpp
--- tests/streams/memstream.cpp	2004/11/10 21:29:08	1.2
+++ tests/streams/memstream.cpp	2004/11/17 18:33:08
@@ -8,11 +8,7 @@
 ///////////////////////////////////////////////////////////////////////////////
 
-#if defined(__GNUG__) && !defined(__APPLE__)
-    #pragma implementation
-    #pragma interface
-#endif
-
 // For compilers that support precompilation, includes "wx/wx.h".
-#include "wx/wxprec.h"
+// and "wx/cppunit.h"
+#include "testprec.h"
 
 #ifdef __BORLANDC__
@@ -25,5 +21,4 @@
 #endif
 
-#include "wx/cppunit.h"
 #include "wx/mstream.h"
 
Index: tests/streams/sstream.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/tests/streams/sstream.cpp,v
retrieving revision 1.2
diff -u -2 -r1.2 sstream.cpp
--- tests/streams/sstream.cpp	2004/11/10 21:29:08	1.2
+++ tests/streams/sstream.cpp	2004/11/17 18:33:08
@@ -9,5 +9,6 @@
 
 // For compilers that support precompilation, includes "wx/wx.h".
-#include "wx/wxprec.h"
+// and "wx/cppunit.h"
+#include "testprec.h"
 
 #ifdef __BORLANDC__
@@ -19,5 +20,4 @@
 #endif
 
-#include "wx/cppunit.h"
 #include "wx/sstream.h"
 
Index: tests/streams/zlibstream.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/tests/streams/zlibstream.cpp,v
retrieving revision 1.8
diff -u -2 -r1.8 zlibstream.cpp
--- tests/streams/zlibstream.cpp	2004/11/10 21:29:08	1.8
+++ tests/streams/zlibstream.cpp	2004/11/17 18:33:10
@@ -8,11 +8,7 @@
 ///////////////////////////////////////////////////////////////////////////////
 
-#if defined(__GNUG__) && !defined(__APPLE__)
-    #pragma implementation
-    #pragma interface
-#endif
-
 // For compilers that support precompilation, includes "wx/wx.h".
-#include "wx/wxprec.h"
+// and "wx/cppunit.h"
+#include "testprec.h"
 
 #ifdef __BORLANDC__
@@ -25,5 +21,4 @@
 #endif
 
-#include "wx/cppunit.h"
 #include "wx/zstream.h"
 #include "wx/wfstream.h"
Index: tests/strings/crt.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/tests/strings/crt.cpp,v
retrieving revision 1.1
diff -u -2 -r1.1 crt.cpp
--- tests/strings/crt.cpp	2004/06/03 19:48:22	1.1
+++ tests/strings/crt.cpp	2004/11/17 18:33:10
@@ -13,5 +13,5 @@
 // ----------------------------------------------------------------------------
 
-#include "wx/wxprec.h"
+#include "testprec.h"
 
 #ifdef __BORLANDC__
@@ -24,6 +24,4 @@
 
 #include "wx/textfile.h"
-
-#include "wx/cppunit.h"
 
 // ----------------------------------------------------------------------------
Index: tests/strings/stdstrings.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/tests/strings/stdstrings.cpp,v
retrieving revision 1.3
diff -u -2 -r1.3 stdstrings.cpp
--- tests/strings/stdstrings.cpp	2004/10/07 16:50:24	1.3
+++ tests/strings/stdstrings.cpp	2004/11/17 18:33:11
@@ -12,5 +12,5 @@
 // ----------------------------------------------------------------------------
 
-#include "wx/wxprec.h"
+#include "testprec.h"
 
 #ifdef __BORLANDC__
@@ -21,6 +21,4 @@
     #include "wx/wx.h"
 #endif // WX_PRECOMP
-
-#include "wx/cppunit.h"
 
 // ----------------------------------------------------------------------------
Index: tests/strings/strings.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/tests/strings/strings.cpp,v
retrieving revision 1.17
diff -u -2 -r1.17 strings.cpp
--- tests/strings/strings.cpp	2004/10/29 22:10:26	1.17
+++ tests/strings/strings.cpp	2004/11/17 18:33:12
@@ -12,5 +12,5 @@
 // ----------------------------------------------------------------------------
 
-#include "wx/wxprec.h"
+#include "testprec.h"
 
 #ifdef __BORLANDC__
@@ -23,6 +23,4 @@
 
 #include "wx/tokenzr.h"
-
-#include "wx/cppunit.h"
 
 // ----------------------------------------------------------------------------
Index: tests/strings/unicode.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/tests/strings/unicode.cpp,v
retrieving revision 1.1
diff -u -2 -r1.1 unicode.cpp
--- tests/strings/unicode.cpp	2004/04/30 20:26:08	1.1
+++ tests/strings/unicode.cpp	2004/11/17 18:33:12
@@ -12,5 +12,5 @@
 // ----------------------------------------------------------------------------
 
-#include "wx/wxprec.h"
+#include "testprec.h"
 
 #ifdef __BORLANDC__
@@ -23,6 +23,4 @@
 
 #include "wx/textfile.h"
-
-#include "wx/cppunit.h"
 
 // ----------------------------------------------------------------------------
Index: tests/uris/uris.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/tests/uris/uris.cpp,v
retrieving revision 1.6
diff -u -2 -r1.6 uris.cpp
--- tests/uris/uris.cpp	2004/11/08 00:23:52	1.6
+++ tests/uris/uris.cpp	2004/11/17 18:33:13
@@ -12,5 +12,5 @@
 // ----------------------------------------------------------------------------
 
-#include "wx/wxprec.h"
+#include "testprec.h"
 
 #ifdef __BORLANDC__
@@ -24,6 +24,4 @@
 #include "wx/uri.h"
 #include "wx/url.h"
-
-#include "wx/cppunit.h"
 
 // Test wxURL & wxURI compat?
--- /dev/null	Fri Sep 10 16:59:46 2004
+++ tests/dummy.cpp	Mon Nov 15 12:07:04 2004
@@ -0,0 +1,23 @@
+/*
+ * File:  dummy.cc
+ * Purpose:  See below
+ * Author:  Julian Smart
+ * Created:  1993
+ * Updated:
+ * Copyright:   (c) Julian Smart
+ */
+
+/* A dummy file to include wx.h. If precompiling wx.h, I
+ * always start by compiling this and producing the PCH file.
+ * Then subsequent source files use the PCH file.
+ *
+ * If precompiling wx.h for wxWidgets and derived apps,
+ * link dummy.obj with your program (the MSC 7 linker complains otherwise).
+ *
+ * This is the only convenient way I found to use precompiled headers
+ * under MSC 7.
+ *
+ * This will produce a big PCH file.
+ */
+
+#include "testprec.h"
--- /dev/null	Fri Sep 10 16:59:46 2004
+++ tests/testprec.h	Mon Nov 15 10:32:24 2004
@@ -0,0 +1,2 @@
+#include "wx/wxprec.h"
+#include "wx/cppunit.h"

 	  	 
