--- src\generic\graphicc.cpp.orig	2011-12-14 20:42:12 +0200
+++ src\generic\graphicc.cpp	2012-04-09 17:57:12 +0300
@@ -33,9 +33,7 @@
     #include "wx/dcclient.h"
     #include "wx/dcmemory.h"
     #include "wx/dcprint.h"
-    #ifdef __WXGTK__
-        #include "wx/window.h"
-    #endif
+    #include "wx/window.h"
 #endif
 
 #include "wx/private/graphics.h"
@@ -71,9 +69,6 @@
 #endif
 
 #include <cairo.h>
-#ifdef __WXMSW__
-#include <cairo-win32.h>
-#endif
 
 #ifdef __WXGTK__
 #include <gtk/gtk.h>
@@ -83,6 +78,7 @@
 
 #ifdef __WXMSW__
 #include <cairo-win32.h>
+#include "wx/msw/winundef.h"
 #endif
 
 #ifdef __WXMAC__
@@ -363,6 +359,38 @@
     unsigned char* m_buffer;
 };
 
+
+#ifdef __WXMSW__
+class wxCairoContextWindowDCHelper
+{
+public:
+    wxCairoContextWindowDCHelper( )
+        : m_mswhwnd(NULL)
+        , m_mswhdc(NULL)
+    {
+    }
+    
+    wxCairoContextWindowDCHelper( wxWindow *window )
+    {
+        m_mswhwnd = window->GetHandle();
+        m_mswhdc = ::GetDC( m_mswhwnd ); 
+    }
+    
+    ~wxCairoContextWindowDCHelper( )
+    {
+        if ( m_mswhwnd && m_mswhdc )
+            ::ReleaseDC( m_mswhwnd, m_mswhdc );
+    }
+
+    HDC GetDC() { return m_mswhdc; }
+    
+private:
+    HWND m_mswhwnd;
+    HDC  m_mswhdc;
+};
+#endif
+
+
 class WXDLLIMPEXP_CORE wxCairoContext : public wxGraphicsContext
 {
 public:
@@ -401,6 +429,7 @@
     virtual void Clip( const wxRegion &region );
 #ifdef __WXMSW__
     cairo_surface_t* m_mswSurface;
+    wxCairoContextWindowDCHelper m_mswdchelper;
 #endif
 
     // clips drawings to the rect
@@ -1748,6 +1777,9 @@
 
 wxCairoContext::wxCairoContext( wxGraphicsRenderer* renderer, wxWindow *window)
 : wxGraphicsContext(renderer)
+#ifdef __WXMSW__
+, m_mswdchelper(window)
+#endif
 {
     m_enableOffset = true;    
 #ifdef __WXGTK__
@@ -1771,7 +1803,7 @@
 #endif
 
 #ifdef __WXMSW__
-    m_mswSurface = cairo_win32_surface_create((HDC)window->GetHandle());
+    m_mswSurface = cairo_win32_surface_create( m_mswdchelper.GetDC() );
     Init(cairo_create(m_mswSurface));
 #endif
 
