--- src/generic/graphicc.cpp.orig	2011-12-14 20:42:12.000000000 +0200
+++ src/generic/graphicc.cpp	2012-04-10 15:39:40.000000000 +0300
@@ -33,9 +33,7 @@
     #include "wx/dcclient.h"
     #include "wx/dcmemory.h"
     #include "wx/dcprint.h"
-    #ifdef __WXGTK__
-        #include "wx/window.h"
-    #endif
+    #include "wx/window.h"
 #endif
 
 #include "wx/private/graphics.h"
@@ -71,9 +69,6 @@
 #endif
 
 #include <cairo.h>
-#ifdef __WXMSW__
-#include <cairo-win32.h>
-#endif
 
 #ifdef __WXGTK__
 #include <gtk/gtk.h>
@@ -83,6 +78,8 @@
 
 #ifdef __WXMSW__
 #include <cairo-win32.h>
+#include "wx/msw/winundef.h"
+#include "wx/msw/private.h"
 #endif
 
 #ifdef __WXMAC__
@@ -401,6 +398,7 @@
     virtual void Clip( const wxRegion &region );
 #ifdef __WXMSW__
     cairo_surface_t* m_mswSurface;
+    WindowHDC m_mswwindowdc;
 #endif
 
     // clips drawings to the rect
@@ -1748,6 +1746,9 @@
 
 wxCairoContext::wxCairoContext( wxGraphicsRenderer* renderer, wxWindow *window)
 : wxGraphicsContext(renderer)
+#ifdef __WXMSW__
+, m_mswwindowdc(window->GetHandle())
+#endif
 {
     m_enableOffset = true;    
 #ifdef __WXGTK__
@@ -1771,7 +1772,7 @@
 #endif
 
 #ifdef __WXMSW__
-    m_mswSurface = cairo_win32_surface_create((HDC)window->GetHandle());
+    m_mswSurface = cairo_win32_surface_create( (HDC)m_mswwindowdc );
     Init(cairo_create(m_mswSurface));
 #endif
 
