Index: src/osx/cocoa/taskbar.mm
===================================================================
--- src/osx/cocoa/taskbar.mm	(revision 71352)
+++ src/osx/cocoa/taskbar.mm	(working copy)
@@ -338,7 +338,7 @@
 {
 }
 
-bool wxTaskBarIconCustomStatusItemImpl::SetIcon(const wxIcon& icon, const wxString& WXUNUSED(tooltip))
+bool wxTaskBarIconCustomStatusItemImpl::SetIcon(const wxIcon& icon, const wxString& tooltip)
 {
     if(!m_statusItem)
     {
@@ -368,6 +368,8 @@
     }
 
     [m_statusItem setImage:m_icon.GetNSImage()];
+    wxCFStringRef cfTooltip(tooltip);
+    [m_statusItem setToolTip:cfTooltip.AsNSString()];
     return true;
 }
