diff --git a/samples/drawing/drawing.cpp b/samples/drawing/drawing.cpp
index 354271f13b..4b094bcd18 100644
--- a/samples/drawing/drawing.cpp
+++ b/samples/drawing/drawing.cpp
@@ -18,6 +18,7 @@
 
 // For compilers that support precompilation, includes "wx/wx.h".
 #include "wx/wxprec.h"
+#include "wx/renderer.h"
 
 
 // for all others, include the necessary headers (this file is usually all you
@@ -1897,7 +1898,8 @@ void MyCanvas::Draw(wxDC& pdc)
         dc.SetClippingRegion(100, 100, 100, 100);
 
     dc.Clear();
-
+    
+    wxRendererNative::Get().DrawCheckBox(this, dc, wxRect(10, 10, 30, 30), wxCONTROL_CHECKED);
     if ( m_owner->m_textureBackground )
     {
         dc.SetPen(*wxMEDIUM_GREY_PEN);
