Index: src/msw/renderer.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/msw/renderer.cpp,v
retrieving revision 1.30
diff -u -r1.30 renderer.cpp
--- src/msw/renderer.cpp	2006/09/14 19:42:33	1.30
+++ src/msw/renderer.cpp	2006/10/07 20:18:58
@@ -223,6 +223,14 @@
         style |= DFCS_INACTIVE;
     if ( flags & wxCONTROL_PRESSED )
         style |= DFCS_PUSHED | DFCS_FLAT;
+    if ( flags & wxCONTROL_ISDEFAULT )
+    {
+        // draw a black border, and deflate the rect
+        wxDCPenChanger pen(dc, *wxBLACK_PEN);
+        wxDCBrushChanger brush(dc, *wxTRANSPARENT_BRUSH);
+        dc.DrawRectangle(rect);
+        ++r.left; ++r.top; --r.bottom; --r.right;
+    }
 
     ::DrawFrameControl(GetHdcOf(dc), &r, DFC_BUTTON, style);
 }

 	  	 
