diff -ur ../wx2/src/msw/slider95.cpp ./src/msw/slider95.cpp
--- ../wx2/src/msw/slider95.cpp	2005-08-03 19:39:49.000000000 -0700
+++ ./src/msw/slider95.cpp	2005-08-24 21:34:08.000000000 -0700
@@ -188,7 +188,7 @@
             }
     };
 
-    wxASSERT_MSG( !(style & wxSL_VERTICAL) | !(style & wxSL_HORIZONTAL),
+    wxASSERT_MSG( !(style & wxSL_VERTICAL) || !(style & wxSL_HORIZONTAL),
                     _T("incompatible slider direction and orientation") );
 
 
diff -ur ../wx2/src/mac/carbon/slider.cpp ./src/mac/carbon/slider.cpp
--- ../wx2/src/mac/carbon/slider.cpp	2005-08-03 19:39:27.000000000 -0700
+++ ./src/mac/carbon/slider.cpp	2005-08-24 19:53:53.000000000 -0700
@@ -84,7 +84,7 @@
             }
     };
 
-    wxASSERT_MSG( !(style & wxSL_VERTICAL) | !(style & wxSL_HORIZONTAL),
+    wxASSERT_MSG( !(style & wxSL_VERTICAL) || !(style & wxSL_HORIZONTAL),
         _T("incompatible slider direction and orientation") );
 
     if ( !wxControl::Create(parent, id, pos, size, style, validator, name) )

 	  	 
