Index: include/wx/cmndata.h
===================================================================
--- include/wx/cmndata.h	(revision 70611)
+++ include/wx/cmndata.h	(working copy)
@@ -280,7 +280,7 @@
     // Use paper id in wxPrintData to set this object's paper size
     void CalculatePaperSizeFromId();
 
-    wxPageSetupDialogData& operator=(const wxPageSetupData& data);
+    wxPageSetupDialogData& operator=(const wxPageSetupDialogData& data);
     wxPageSetupDialogData& operator=(const wxPrintData& data);
 
     wxPrintData& GetPrintData() { return m_printData; }
Index: include/wx/generic/prntdlgg.h
===================================================================
--- include/wx/generic/prntdlgg.h	(revision 70611)
+++ include/wx/generic/prntdlgg.h	(working copy)
@@ -34,7 +34,7 @@
 class WXDLLIMPEXP_FWD_CORE wxComboBox;
 class WXDLLIMPEXP_FWD_CORE wxStaticText;
 class WXDLLIMPEXP_FWD_CORE wxRadioBox;
-class WXDLLIMPEXP_FWD_CORE wxPageSetupData;
+class WXDLLIMPEXP_FWD_CORE wxPageSetupDialogData;
 
 // ----------------------------------------------------------------------------
 // constants
Index: include/wx/msw/printdlg.h
===================================================================
--- include/wx/msw/printdlg.h	(revision 70611)
+++ include/wx/msw/printdlg.h	(working copy)
@@ -107,7 +107,7 @@
     bool ConvertToNative( wxPageSetupDialogData &data );
     bool ConvertFromNative( wxPageSetupDialogData &data );
 
-    virtual wxPageSetupData& GetPageSetupDialogData() { return m_pageSetupData; }
+    virtual wxPageSetupDialogData& GetPageSetupDialogData() { return m_pageSetupData; }
 
 private:
     wxPageSetupDialogData   m_pageSetupData;
Index: include/wx/osx/printdlg.h
===================================================================
--- include/wx/osx/printdlg.h	(revision 70611)
+++ include/wx/osx/printdlg.h	(working copy)
@@ -58,16 +58,16 @@
 class WXDLLIMPEXP_CORE wxMacPageSetupDialog: public wxPageSetupDialogBase
 {
 public:
-    wxMacPageSetupDialog(wxWindow *parent, wxPageSetupData *data = NULL);
+    wxMacPageSetupDialog(wxWindow *parent, wxPageSetupDialogData *data = NULL);
     virtual ~wxMacPageSetupDialog();
 
-    virtual wxPageSetupData& GetPageSetupDialogData();
+    virtual wxPageSetupDialogData& GetPageSetupDialogData();
 
-    bool Create(wxWindow *parent, wxPageSetupData *data = NULL);
+    bool Create(wxWindow *parent, wxPageSetupDialogData *data = NULL);
     virtual int ShowModal();
 
 private:
-    wxPageSetupData   m_pageSetupData;
+    wxPageSetupDialogData   m_pageSetupData;
     wxWindow*               m_dialogParent;
 
 private:
@@ -85,14 +85,14 @@
 class WXDLLIMPEXP_CORE wxMacPageMarginsDialog : public wxDialog
 {
 public:
-    wxMacPageMarginsDialog(wxFrame* parent, wxPageSetupData* data);
+    wxMacPageMarginsDialog(wxFrame* parent, wxPageSetupDialogData* data);
     bool TransferToWindow();
     bool TransferDataFromWindow();
 
-    virtual wxPageSetupData& GetPageSetupDialogData() { return *m_pageSetupDialogData; }
+    virtual wxPageSetupDialogData& GetPageSetupDialogData() { return *m_pageSetupDialogData; }
 
 private:
-    wxPageSetupData* m_pageSetupDialogData;
+    wxPageSetupDialogData* m_pageSetupDialogData;
 
     wxPoint m_MinMarginTopLeft;
     wxPoint m_MinMarginBottomRight;
Index: include/wx/richtext/richtextprint.h
===================================================================
--- include/wx/richtext/richtextprint.h	(revision 70611)
+++ include/wx/richtext/richtextprint.h	(working copy)
@@ -202,7 +202,7 @@
 
     /// Set print and page setup data
     void SetPrintData(const wxPrintData& printData);
-    void SetPageSetupData(const wxPageSetupData& pageSetupData);
+    void SetPageSetupData(const wxPageSetupDialogData& pageSetupData);
 
     /// Set the rich text buffer pointer, deleting the existing object if present
     void SetRichTextBufferPreview(wxRichTextBuffer* buf);
Index: samples/stc/defsext.h
===================================================================
--- samples/stc/defsext.h	(revision 70611)
+++ samples/stc/defsext.h	(working copy)
@@ -101,7 +101,7 @@
 
 //! global print data, to remember settings during the session
 extern wxPrintData *g_printData;
-extern wxPageSetupData *g_pageSetupData;
+extern wxPageSetupDialogData *g_pageSetupData;
 
 #endif // wxUSE_PRINTING_ARCHITECTURE
 
Index: samples/stc/stctest.cpp
===================================================================
--- samples/stc/stctest.cpp	(revision 70611)
+++ samples/stc/stctest.cpp	(working copy)
@@ -79,7 +79,7 @@
 
 //! global print data, to remember settings during the session
 wxPrintData *g_printData = (wxPrintData*) NULL;
-wxPageSetupData *g_pageSetupData = (wxPageSetupData*) NULL;
+wxPageSetupDialogData *g_pageSetupData = (wxPageSetupDialogData*) NULL;
 
 #endif // wxUSE_PRINTING_ARCHITECTURE
 
Index: src/osx/core/printmac.cpp
===================================================================
--- src/osx/core/printmac.cpp	(revision 70611)
+++ src/osx/core/printmac.cpp	(working copy)
@@ -460,14 +460,14 @@
     return true ;
 }
 
-void wxOSXPrintData::TransferFrom( wxPageSetupData *WXUNUSED(data) )
+void wxOSXPrintData::TransferFrom( wxPageSetupDialogData *WXUNUSED(data) )
 {
     // should we setup the page rect here ?
     // since MacOS sometimes has two same paper rects with different
     // page rects we could make it roundtrip safe perhaps
 }
 
-void wxOSXPrintData::TransferTo( wxPageSetupData* data )
+void wxOSXPrintData::TransferTo( wxPageSetupDialogData* data )
 {
 #if wxOSX_USE_COCOA
     UpdateToPMState();
Index: src/osx/printdlg_osx.cpp
===================================================================
--- src/osx/printdlg_osx.cpp	(revision 70611)
+++ src/osx/printdlg_osx.cpp	(working copy)
@@ -80,13 +80,13 @@
 
 IMPLEMENT_CLASS(wxMacPageSetupDialog, wxPageSetupDialogBase)
 
-wxMacPageSetupDialog::wxMacPageSetupDialog( wxWindow *p, wxPageSetupData *data )
+wxMacPageSetupDialog::wxMacPageSetupDialog( wxWindow *p, wxPageSetupDialogData *data )
     : wxPageSetupDialogBase()
 {
     Create( p, data );
 }
 
-bool wxMacPageSetupDialog::Create( wxWindow *p, wxPageSetupData *data )
+bool wxMacPageSetupDialog::Create( wxWindow *p, wxPageSetupDialogData *data )
 {
     m_dialogParent = p;
 
@@ -100,14 +100,14 @@
 {
 }
 
-wxPageSetupData& wxMacPageSetupDialog::GetPageSetupDialogData()
+wxPageSetupDialogData& wxMacPageSetupDialog::GetPageSetupDialogData()
 {
     return m_pageSetupData;
 }
 
 IMPLEMENT_CLASS(wxMacPageMarginsDialog, wxDialog)
 
-wxMacPageMarginsDialog::wxMacPageMarginsDialog(wxFrame *parent, wxPageSetupData *data) :
+wxMacPageMarginsDialog::wxMacPageMarginsDialog(wxFrame *parent, wxPageSetupDialogData *data) :
   wxDialog(parent, wxID_ANY, wxString(wxT("Page Margins"))),
   m_pageSetupDialogData(data)
   {
Index: src/richtext/richtextprint.cpp
===================================================================
--- src/richtext/richtextprint.cpp	(revision 70611)
+++ src/richtext/richtextprint.cpp	(working copy)
@@ -477,7 +477,7 @@
     (*GetPrintData()) = printData;
 }
 
-void wxRichTextPrinting::SetPageSetupData(const wxPageSetupData& pageSetupData)
+void wxRichTextPrinting::SetPageSetupData(const wxPageSetupDialogData& pageSetupData)
 {
     (*GetPageSetupData()) = pageSetupData;
 }
