diff --git "a/C:\\Users\\Petr\\AppData\\Local\\Temp\\TortoiseGit\\graF1DF.tmp\\graphicsd2d-5420f96-left.cpp" "b/L:\\Dev\\Desktop\\wxWidgets-GIT\\src\\msw\\graphicsd2d.cpp"
index b509e7e..0b8b8d0 100644
--- "a/C:\\Users\\Petr\\AppData\\Local\\Temp\\TortoiseGit\\graF1DF.tmp\\graphicsd2d-5420f96-left.cpp"
+++ "b/L:\\Dev\\Desktop\\wxWidgets-GIT\\src\\msw\\graphicsd2d.cpp"
@@ -719,8 +719,7 @@ wxCOMPtr<ID2D1Geometry> wxD2DConvertRegionToGeometry(ID2D1Factory* direct2dFacto
     ID2D1Geometry** geometries = new ID2D1Geometry*[rectCount];
     regionIterator.Reset(region);
 
-    int i = 0;
-    while(regionIterator)
+    for (int i = 0; regionIterator; i++, regionIterator++)
     {
         geometries[i] = NULL;
 
@@ -731,8 +730,6 @@ wxCOMPtr<ID2D1Geometry> wxD2DConvertRegionToGeometry(ID2D1Factory* direct2dFacto
         direct2dFactory->CreateRectangleGeometry(
             wxD2DConvertRect(rect),
             (ID2D1RectangleGeometry**)(&geometries[i]));
-
-        i++; regionIterator++;
     }
 
     // Create a geometry group to hold all the rectangles
