diff -ru2 -xCVS -x.#* -x*.orig -x*.rej -x*~ wxWidgets.orig\src\generic\datectlg.cpp wxWidgets\src\generic\datectlg.cpp
--- wxWidgets.orig\src\generic\datectlg.cpp	Thu Mar 24 14:49:40 2005
+++ wxWidgets\src\generic\datectlg.cpp	Thu Mar 24 18:53:06 2005
@@ -79,7 +79,7 @@
     #define BTN_FLAGS         wxNO_BORDER
     #define CALBORDER         0
-    #define RIGHTBUTTONBORDER 3
+    #define RIGHTBUTTONBORDER 4
     #define TOPBUTTONBORDER   0
-    #define BUTTONBORDER      3
+    #define BUTTONBORDER      4
     #define TXTPOSY           1
 #else
@@ -185,6 +185,8 @@
             h = GetSize().y;
 
-        int bw = w - m_marginX - m_borderX;
-        int bh = h - m_marginY - m_borderY;
+        int borderX = m_marginX + m_borderX; 
+        int borderY = m_marginY + m_borderY;
+        int bw = w - borderX;
+        int bh = h - borderY;
         if (bh < 11) bh=11;
         if (bw < 9)  bw=9;
@@ -193,16 +195,17 @@
         dc.SelectObject(bmp);
 
-        wxRect r(0,0,bw, bh);
         wxRendererNative& renderer = wxRendererNative::Get();
 
 #ifdef __WXGTK__
+        wxRect r(-(borderX/2),-(borderY/2),w,h);
         wxColour magic(255,0,255);
         dc.SetBrush( wxBrush( magic ) );
         dc.SetPen( *wxTRANSPARENT_PEN );
         dc.DrawRectangle(0,0,bw,bh);
-        renderer.DrawComboBoxDropButton(this, dc, r);
+        renderer.DrawDropArrow(this, dc, r);
         wxMask *mask = new wxMask( bmp, magic );
         bmp.SetMask( mask );
 #else
+        wxRect r(0,0,bw,bh);
         renderer.DrawComboBoxDropButton(this, dc, r);
 #endif
@@ -216,5 +219,5 @@
         dc.SetPen( *wxTRANSPARENT_PEN );
         dc.DrawRectangle(0,0,bw,bh);
-        renderer.DrawComboBoxDropButton(this, dc, r, wxCONTROL_PRESSED);
+        renderer.DrawDropArrow(this, dc, r, wxCONTROL_PRESSED);
         mask = new wxMask( bmpSel, magic );
         bmpSel.SetMask( mask );

 	  	 
