Index: interface/wx/textdlg.h
===================================================================
--- interface/wx/textdlg.h	(revision 64087)
+++ interface/wx/textdlg.h	(working copy)
@@ -129,6 +129,8 @@
     If @c centre is @true, the message text (which may include new line
     characters) is centred; if @false, the message is left-justified.
 
+    If @c parent is not supplied, the dialog can become hidden under its parent.
+
     @header{wx/textdlg.h}
 */
 wxString wxGetTextFromUser(const wxString& message,
