Index: wxWindows/src/generic/splitter.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/generic/splitter.cpp,v
retrieving revision 1.75
diff -b -u -2 -r1.75 splitter.cpp
--- wxWindows/src/generic/splitter.cpp	2003/03/17 11:47:55	1.75
+++ wxWindows/src/generic/splitter.cpp	2003/04/02 18:53:25
@@ -359,5 +359,5 @@
         }
     }
-    else if ( event.LeftDClick() )
+    else if ( event.LeftDClick() && m_windowTwo )
     {
         OnDoubleClickSash(x, y);
@@ -893,5 +893,5 @@
     }
 
-    OnUnsplit(win);
+    win->Show(FALSE);
     DoSetSashPosition(0);
     SizeWindows();
@@ -1063,4 +1063,6 @@
 void wxSplitterWindow::OnDoubleClickSash(int x, int y)
 {
+    wxCHECK_RET(m_windowTwo, wxT("splitter: no window to remove"));
+
     // new code should handle events instead of using the virtual functions
     wxSplitterEvent event(wxEVT_COMMAND_SPLITTER_DOUBLECLICKED, this);
@@ -1071,5 +1073,9 @@
         if ( GetMinimumPaneSize() == 0 || m_permitUnsplitAlways )
         {
-            Unsplit();
+            wxWindow *win = m_windowTwo;
+            m_windowTwo = (wxWindow *) NULL;
+            OnUnsplit(win);
+            DoSetSashPosition(0);
+            SizeWindows();
         }
     }

 	  	 
