diff -uNr wxWidgets-2.9-svn-56103/include/wx/aui/auibar.h wxWidgets-Mine/include/wx/aui/auibar.h
--- wxWidgets-2.9-svn-56103/include/wx/aui/auibar.h	Mon Nov 10 09:19:07 2008
+++ wxWidgets-Mine/include/wx/aui/auibar.h	Thu Nov 13 14:08:42 2008
@@ -249,8 +249,11 @@
 
     virtual wxAuiToolBarArt* Clone() = 0;
     virtual void SetFlags(unsigned int flags) = 0;
+    virtual unsigned int GetFlags() = 0;
     virtual void SetFont(const wxFont& font) = 0;
+    virtual wxFont GetFont() = 0;
     virtual void SetTextOrientation(int orientation) = 0;
+    virtual int GetTextOrientation() = 0;
 
     virtual void DrawBackground(
                          wxDC& dc,
@@ -326,9 +329,12 @@
     virtual ~wxAuiDefaultToolBarArt();
 
     virtual wxAuiToolBarArt* Clone();
-    virtual void SetFlags(unsigned int flags);
-    virtual void SetFont(const wxFont& font);
-    virtual void SetTextOrientation(int orientation);
+    virtual void SetFlags(unsigned int flags);
+    virtual unsigned int GetFlags();
+    virtual void SetFont(const wxFont& font);
+    virtual wxFont GetFont();
+    virtual void SetTextOrientation(int orientation);
+    virtual int GetTextOrientation();
 
     virtual void DrawBackground(
                 wxDC& dc,
diff -uNr wxWidgets-2.9-svn-56103/src/aui/auibar.cpp wxWidgets-Mine/src/aui/auibar.cpp
--- wxWidgets-2.9-svn-56103/src/aui/auibar.cpp	Mon Nov 10 08:33:20 2008
+++ wxWidgets-Mine/src/aui/auibar.cpp	Thu Nov 13 14:10:22 2008
@@ -226,6 +226,21 @@
 {
     m_text_orientation = orientation;
 }
+
+unsigned int wxAuiDefaultToolBarArt::GetFlags()
+{
+    return m_flags;
+}
+
+wxFont wxAuiDefaultToolBarArt::GetFont()
+{
+    return m_font;
+}
+
+int wxAuiDefaultToolBarArt::GetTextOrientation()
+{
+    return m_text_orientation;
+}
 
 void wxAuiDefaultToolBarArt::DrawBackground(
                                     wxDC& dc,
