--- wxWindows/contrib\utils\wxrc\wxrc.cpp	2003-08-19 15:49:13.000000000 +0200
+++ wxWin/contrib\utils\wxrc\wxrc.cpp	2003-08-22 09:36:14.000000000 +0200
@@ -96,3 +96,6 @@
             return 0;
+            #if 0
+            // break is unreachable because of return earlier
             break;
+            #endif
 
@@ -106,4 +109,9 @@
             return retCode;
+            #if 0
+            // break is unreachable because of return earlier
             break;
+            #endif
 
+        #if 0
+        // default return moved outside of switch to avoid warning about lack of return in function
         default:
@@ -111,3 +119,5 @@
             break;
+        #endif
     }
+    return 1;
 }

 	  	 
