--- a/src/msw/window.cpp
+++ b/src/msw/window.cpp
@@ -1859,10 +1859,23 @@ void wxWindowMSW::DoGetPosition(int *x, int *y) const
         // children, not for the dialogs/frames
         if ( !IsTopLevel() )
         {
-            if ( wxTheApp->GetLayoutDirection() == wxLayout_RightToLeft )
+            // In RTL mode, we want the logical left x-coordinate,
+            // which would be the physical right x-coordinate.
+            // Parent window RTL mode has precedence over application mode.
+            bool useRTL = false;
+            if ( parent )
+            {
+                if ( parent->GetLayoutDirection() == wxLayout_RightToLeft )
+                {
+                    useRTL = true;
+                }
+            }
+            else if ( wxTheApp->GetLayoutDirection() == wxLayout_RightToLeft )
+            {
+                useRTL = true;
+            }
+            if( useRTL )
             {
-                // In RTL mode, we want the logical left x-coordinate,
-                // which would be the physical right x-coordinate.
                 point.x = rect.right;
             }
 
