Index: richtextformatdlg.cpp
===================================================================
--- richtextformatdlg.cpp	(revision 61967)
+++ richtextformatdlg.cpp	(working copy)
@@ -499,7 +499,7 @@
     wxControl::Create(parent, id, pos, size, style);
 
     SetColour(wxSystemSettings::GetColour(wxSYS_COLOUR_WINDOW));
-    SetBackgroundStyle(wxBG_STYLE_COLOUR);
+    SetBackgroundStyle(wxBG_STYLE_ERASE); // is it needed at all though?
 }
 
 wxRichTextColourSwatchCtrl::~wxRichTextColourSwatchCtrl()
