--- minimalOld/minimal.cpp	2006-12-29 11:31:40.000000000 +0000
+++ minimal/minimal.cpp	2006-12-29 12:42:57.000000000 +0000
@@ -30,5 +30,5 @@
     #include "wx/wx.h"
 #endif
-
+#include "wx/config.h"
 // ----------------------------------------------------------------------------
 // resources
@@ -173,4 +173,18 @@
     SetStatusText(_T("Welcome to wxWidgets!"));
 #endif // wxUSE_STATUSBAR
+
+wxConfigBase* config = wxConfigBase::Get();
+if (config==NULL)   return;
+config->Write( wxT("Group/SubGroup/SubSubGroup"), wxT("Entry") );	
+uint beforecount = config->GetNumberOfGroups( true );
+config->DeleteGroup( wxT("Group/") );
+uint betweencount = config->GetNumberOfGroups( true );
+
+config->DeleteGroup( wxT("Group") );
+uint aftercount = config->GetNumberOfGroups( true );
+
+wxMessageBox( wxString::Format(wxT("Before Deleting , there are %u groups\
+                                \nAfter DeleteGroup( Group/ ), there are %u groups\
+                                \nAfter DeleteGroup( Group ), there are %u groups"), beforecount, betweencount, aftercount ) );
 }
 

 	  	 
