Index: samples/internat/internat.cpp
===================================================================
--- samples/internat/internat.cpp	(revision 74150)
+++ samples/internat/internat.cpp	(working copy)
@@ -209,12 +209,37 @@
     return true;
 }
 
-// `Main program' equivalent, creating windows and returning main app frame
+// 'Main program' equivalent, creating windows and returning main app frame
 bool MyApp::OnInit()
 {
     if ( !wxApp::OnInit() )
         return false;
 
+#define MYTEST 1
+
+#if MYTEST
+    //if ( m_locale.Init(wxLocale::GetSystemLanguage()) ) // <- works fine
+    if ( m_locale.Init(wxLANGUAGE_DEFAULT) )              // <- not working at all (but still returning true)
+    {
+        if (m_locale.AddCatalog("internat"))
+        {
+            // these two texts are translated correctly ONLY if 
+            // calling Init(wxLocale::GetSystemLanguage()) -
+            // NOT when calling Init(wxLANGUAGE_DEFAULT)
+            wxMessageBox(wxGetTranslation("&Play a game")); // from internat.mo
+            wxMessageBox(wxGetTranslation("&Down"));        // from wxstd.mo
+        }
+        else
+            wxLogError(_("Couldn't find/load the 'internat' catalog for locale."));
+    }
+    else
+    {
+        wxLogWarning(_("This language is not supported by the system."));
+
+        // continue nevertheless
+    }
+
+#else
     if ( m_lang == wxLANGUAGE_UNKNOWN )
     {
         int lng = wxGetSingleChoiceIndex
@@ -265,7 +290,7 @@
         m_locale.AddCatalog("fileutils");
     }
 #endif
-
+#endif
     // Create the main frame window
     MyFrame *frame = new MyFrame(m_locale);
 
