diff -uNr wxWidgets-orig/src/common/sizer.cpp wxWidgets-mine/src/common/sizer.cpp
--- wxWidgets-orig/src/common/sizer.cpp	2014-10-06 23:34:04.000000000 +0200
+++ wxWidgets-mine/src/common/sizer.cpp	2015-11-27 21:28:17.107489200 +0100
@@ -734,6 +734,7 @@
             item->DetachSizer();
             delete item;
             m_children.Erase( node );
+            sizer->SetContainingWindow(NULL);
             return true;
         }
         node = node->GetNext();
