? moreifdefs.patch
Index: nativdlg.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/nativdlg.cpp,v
retrieving revision 1.19
diff -b -u -2 -r1.19 nativdlg.cpp
--- nativdlg.cpp	2003/03/17 11:21:54	1.19
+++ nativdlg.cpp	2003/05/02 07:36:41
@@ -184,4 +184,5 @@
             win = new wxRadioButton;
         }
+#if wxUSE_BMPBUTTON
 #if defined(__WIN32__) && defined(BS_BITMAP)
         else if (style & BS_BITMAP)
@@ -202,4 +203,5 @@
             win = new wxBitmapButton;
         }
+#endif
         else if ((style1 == BS_PUSHBUTTON) || (style1 == BS_DEFPUSHBUTTON))
         {
@@ -257,4 +259,5 @@
         if ((style1 == SS_LEFT) || (style1 == SS_RIGHT) || (style1 == SS_SIMPLE))
             win = new wxStaticText;
+#if wxUSE_STATBMP
 #if defined(__WIN32__) && defined(BS_BITMAP)
         else if (style1 == SS_BITMAP)
@@ -266,4 +269,5 @@
         }
 #endif
+#endif	/* wxUSE_STATBMP */
     }
     else
Index: ole/oleutils.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/ole/oleutils.cpp,v
retrieving revision 1.17
diff -b -u -2 -r1.17 oleutils.cpp
--- ole/oleutils.cpp	2003/03/17 11:22:07	1.17
+++ ole/oleutils.cpp	2003/05/02 07:36:41
@@ -61,5 +61,5 @@
 }
 
-#if wxUSE_DRAG_AND_DROP
+#if wxUSE_DATAOBJ
 
 // ----------------------------------------------------------------------------
Index: ole/uuid.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/ole/uuid.cpp,v
retrieving revision 1.14
diff -b -u -2 -r1.14 uuid.cpp
--- ole/uuid.cpp	2003/03/17 11:22:07	1.14
+++ ole/uuid.cpp	2003/05/02 07:36:42
@@ -27,5 +27,5 @@
 #include  "wx/setup.h"
 
-#if wxUSE_OLE && wxUSE_DRAG_AND_DROP
+#if wxUSE_OLE && ( wxUSE_DRAG_AND_DROP || (__WXDEBUG__ && wxUSE_DATAOBJ) )
 
 // standard headers

 	  	 
