Index: wxWindows/include/wx/thread.h
===================================================================
RCS file: /export/cvsroot/org/wxwindows/wxWindows/include/wx/thread.h,v
retrieving revision 1.1.1.5
retrieving revision 1.2
diff -b -u -2 -r1.1.1.5 -r1.2
--- wxWindows/include/wx/thread.h	2002/06/07 21:46:44	1.1.1.5
+++ wxWindows/include/wx/thread.h	2002/06/11 23:08:31	1.2
@@ -246,5 +246,6 @@
     // if CRITICAL_SECTION size changes in Windows, you'll get an assert from
     // thread.cpp and will need to increase the buffer size
-    char m_buffer[24];
+    typedef char buffer_t[24];
+    buffer_t m_buffer;
 #else
     // nothing for OS/2
Index: wxWindows/src/msw/thread.cpp
===================================================================
RCS file: /export/cvsroot/org/wxwindows/wxWindows/src/msw/thread.cpp,v
retrieving revision 1.1.1.4
retrieving revision 1.2
diff -b -u -2 -r1.1.1.4 -r1.2
--- wxWindows/src/msw/thread.cpp	2002/06/06 18:31:59	1.1.1.4
+++ wxWindows/src/msw/thread.cpp	2002/06/11 23:08:53	1.2
@@ -133,5 +133,5 @@
 wxCriticalSection::wxCriticalSection()
 {
-    wxCOMPILE_TIME_ASSERT( sizeof(CRITICAL_SECTION) <= sizeof(m_buffer),
+    wxCOMPILE_TIME_ASSERT( sizeof(CRITICAL_SECTION) <= sizeof(buffer_t),
                            wxCriticalSectionBufferTooSmall );
 

 	  	 
