--- a/samples/propgrid/sampleprops.cpp
+++ b/samples/propgrid/sampleprops.cpp
@@ -59,13 +59,15 @@ wxFontDataProperty::wxFontDataProperty( const wxString& label, const wxString& n
     wxFontData fontData(value);
 
     // Fix value.
-    fontData.SetChosenFont(value.GetInitialFont());
+    wxFont font;
+    font << m_value;  // Get font data from base object.
+    fontData.SetChosenFont(font);
     if ( !fontData.GetColour().IsOk() )
         fontData.SetColour(*wxBLACK);
 
     // Set initial value - should be done in a simpler way like this
     // (instead of calling SetValue) in derived (wxObject) properties.
-    m_value_wxFontData << value;
+    m_value_wxFontData << fontData;
 
     // Add extra children.
     AddPrivateChild( new wxColourProperty(_("Colour"), wxPG_LABEL,
@@ -180,7 +182,7 @@ wxVariant wxFontDataProperty::ChildChanged( wxVariant& thisValue,
             // Transfer from subset to superset.
             wxFont font = fontData.GetChosenFont();
             variant = WXVARIANT(font);
-            wxFontProperty::ChildChanged( variant, childIndex, childValue );
+            variant = wxFontProperty::ChildChanged( variant, childIndex, childValue );
             font << variant;
             fontData.SetChosenFont(font);
     }
