diff -urN wx2.3orig/include/wx/generic/grid.h wx2.3/include/wx/generic/grid.h
--- wx2.3orig/include/wx/generic/grid.h	Sun Apr 29 13:42:53 2001
+++ wx2.3/include/wx/generic/grid.h	Sun Apr 29 13:40:07 2001
@@ -1050,7 +1050,7 @@
                             int horizontalAlignment = wxALIGN_LEFT,
                             int verticalAlignment = wxALIGN_TOP );
  
-    void DrawTextRectangle( wxDC& dc, wxArrayString& lines, const wxRect&,
+    void DrawTextRectangle( wxDC& dc, const wxArrayString& lines, const wxRect&,
                             int horizontalAlignment = wxALIGN_LEFT,
                             int verticalAlignment = wxALIGN_TOP );
 
@@ -1061,7 +1061,7 @@
     void StringToLines( const wxString& value, wxArrayString& lines );
 
     void GetTextBoxSize( wxDC& dc,
-                         wxArrayString& lines,
+                         const wxArrayString& lines,
                          long *width, long *height );
 
 
diff -urN wx2.3orig/src/generic/grid.cpp wx2.3/src/generic/grid.cpp
--- wx2.3orig/src/generic/grid.cpp	Sun Apr 29 13:42:54 2001
+++ wx2.3/src/generic/grid.cpp	Sun Apr 29 13:40:24 2001
@@ -6242,7 +6242,7 @@
 }
 
 void wxGrid::DrawTextRectangle( wxDC& dc,
-			        wxArrayString& lines,
+			        const wxArrayString& lines,
                                 const wxRect& rect,
                                 int horizAlign,
                                 int vertAlign )
@@ -6334,7 +6334,7 @@
 
 
 void wxGrid::GetTextBoxSize( wxDC& dc,
-                             wxArrayString& lines,
+                             const wxArrayString& lines,
                              long *width, long *height )
 {
     long w = 0;

 	  	 
