diff -u -3 -P -r wxWidgets/.\include\wx\filedlg.h wxW/.\include\wx\filedlg.h
--- wxWidgets/.\include\wx\filedlg.h	Thu Jan 15 14:48:12 2004
+++ wxW/.\include\wx\filedlg.h	Thu May 13 12:28:13 2004
@@ -30,7 +30,9 @@
     wxOPEN              = 0x0001,
     wxSAVE              = 0x0002,
     wxOVERWRITE_PROMPT  = 0x0004,
+#if WXWIN_COMPATIBILITY_2_4
     wxHIDE_READONLY     = 0x0008,
+#endif
     wxFILE_MUST_EXIST   = 0x0010,
     wxMULTIPLE          = 0x0020,
     wxCHANGE_DIR        = 0x0040
diff -u -3 -P -r wxWidgets/.\src\msw\filedlg.cpp wxW/.\src\msw\filedlg.cpp
--- wxWidgets/.\src\msw\filedlg.cpp	Thu Apr 22 07:48:23 2004
+++ wxW/.\src\msw\filedlg.cpp	Thu May 13 12:32:46 2004
@@ -139,9 +139,14 @@
     *fileNameBuffer = wxT('\0');
     *titleBuffer    = wxT('\0');
 
+#if WXWIN_COMPATIBILITY_2_4
     long msw_flags = 0;
     if ( (m_dialogStyle & wxHIDE_READONLY) || (m_dialogStyle & wxSAVE) )
         msw_flags |= OFN_HIDEREADONLY;
+#else
+    long msw_flags = OFN_HIDEREADONLY;
+#endif
+
     if ( m_dialogStyle & wxFILE_MUST_EXIST )
         msw_flags |= OFN_PATHMUSTEXIST | OFN_FILEMUSTEXIST;
 
diff -u -3 -P -r wxWidgets/.\src\os2\filedlg.cpp wxW/.\src\os2\filedlg.cpp
--- wxWidgets/.\src\os2\filedlg.cpp	Mon Mar 01 07:24:18 2004
+++ wxW/.\src\os2\filedlg.cpp	Thu May 13 12:35:17 2004
@@ -126,7 +126,12 @@
     else
         lFlags = FDS_OPEN_DIALOG;
 
-    if ((m_dialogStyle & wxHIDE_READONLY) || (m_dialogStyle & wxSAVE))
+#if WXWIN_COMPATIBILITY_2_4
+    if (m_dialogStyle & wxHIDE_READONLY)
+        lFlags |= FDS_SAVEAS_DIALOG;
+#endif
+
+    if (m_dialogStyle & wxSAVE)
         lFlags |= FDS_SAVEAS_DIALOG;
     if (m_dialogStyle & wxMULTIPLE )
         lFlags |= FDS_OPEN_DIALOG | FDS_MULTIPLESEL;

 	  	 
