Index: dialog.cpp
===================================================================
--- dialog.cpp	(revision 54783)
+++ dialog.cpp	(working copy)
@@ -426,7 +426,7 @@
 
 void wxDialog::OnWindowCreate(wxWindowCreateEvent& event)
 {
-    if ( m_hGripper && IsShown() &&
+    if ( m_hGripper && 
             event.GetWindow() && event.GetWindow()->GetParent() == this )
     {
         // Put gripper below the newly created child window
