Index: minimal.cpp
===================================================================
--- minimal.cpp	(revision 69240)
+++ minimal.cpp	(working copy)
@@ -30,6 +30,8 @@
     #include "wx/wx.h"
 #endif
 
+#include "wx/dcps.h"
+
 // ----------------------------------------------------------------------------
 // resources
 // ----------------------------------------------------------------------------
@@ -130,7 +132,18 @@
     // and show it (the frames, unlike simple controls, are not shown when
     // created initially)
     frame->Show(true);
+	
+    // wxPostScriptDC test start
+    wxPrintData pd;
+    pd.SetPrintMode( wxPRINT_MODE_FILE ); 
+    pd.SetFilename( wxT("test.ps") );
 
+    wxPostScriptDC psdc(pd);
+    psdc.StartDoc(wxT("printing..."));
+    psdc.DrawRectangle(90,90, 180,180);
+    psdc.EndDoc();
+    // wxPostScriptDC test end
+
     // success: wxApp::OnRun() will be called which will enter the main message
     // loop and the application will run. If we returned false here, the
     // application would exit immediately.
