diff --git a/src/generic/treectlg.cpp b/src/generic/treectlg.cpp
index 28ac6d7..e59d4aa 100644
--- a/src/generic/treectlg.cpp
+++ b/src/generic/treectlg.cpp
@@ -95,6 +95,8 @@ class WXDLLEXPORT wxTreeTextCtrl: public wxTextCtrl
 public:
     wxTreeTextCtrl(wxGenericTreeCtrl *owner, wxGenericTreeItem *item);
 
+    ~wxTreeTextCtrl() wxNOEXCEPT {}
+
     void EndEdit( bool discardChanges );
 
     const wxGenericTreeItem* item() const { return m_itemEdited; }
diff --git a/src/gtk/textctrl.cpp b/src/gtk/textctrl.cpp
index ecb98e6..0e408c5 100644
--- a/src/gtk/textctrl.cpp
+++ b/src/gtk/textctrl.cpp
@@ -610,7 +610,7 @@ void wxTextCtrl::Init()
     m_anonymousMarkList = NULL;
 }
 
-wxTextCtrl::~wxTextCtrl()
+wxTextCtrl::~wxTextCtrl() wxNOEXCEPT
 {
     if (m_text)
         GTKDisconnect(m_text);
diff --git a/src/gtk1/textctrl.cpp b/src/gtk1/textctrl.cpp
index 7ce0d35..2254d29 100644
--- a/src/gtk1/textctrl.cpp
+++ b/src/gtk1/textctrl.cpp
@@ -226,7 +226,7 @@ void wxTextCtrl::Init()
     m_vScrollbar = NULL;
 }
 
-wxTextCtrl::~wxTextCtrl()
+wxTextCtrl::~wxTextCtrl() wxNOEXCEPT
 {
 }
 
diff --git a/src/univ/textctrl.cpp b/src/univ/textctrl.cpp
index 041ea21..8c3875c 100644
--- a/src/univ/textctrl.cpp
+++ b/src/univ/textctrl.cpp
@@ -764,7 +764,7 @@ bool wxTextCtrl::Create(wxWindow *parent,
     return true;
 }
 
-wxTextCtrl::~wxTextCtrl()
+wxTextCtrl::~wxTextCtrl() wxNOEXCEPT
 {
     delete m_cmdProcessor;
 
