Index: samples/dialogs/dialogs.cpp
===================================================================
--- samples/dialogs/dialogs.cpp	(revision 63977)
+++ samples/dialogs/dialogs.cpp	(working copy)
@@ -280,6 +280,14 @@
     wxInitAllImageHandlers();
 #endif
 
+    ShowNewFrame();
+    ShowNewFrame();
+
+    return true;
+}
+
+void MyApp::ShowNewFrame()
+{
     // Create the main frame window
     MyFrame *frame = new MyFrame(wxT("wxWidgets dialogs example"));
 
@@ -488,8 +496,6 @@
 
     frame->Centre(wxBOTH);
     frame->Show(true);
-
-    return true;
 }
 
 // My frame constructor
@@ -849,7 +855,18 @@
 
 void MyFrame::MessageBoxInfo(wxCommandEvent& WXUNUSED(event))
 {
-    ::wxInfoMessageBox(this);
+    static int depth = 0;
+    ++depth;
+
+    wxString
+        message("wxWidgets is a cross-platform framework."),
+        caption(wxString::Format("Depth %d",depth));
+    wxMessageBox(message,caption);
+
+    wxBell();
+    wxSleep(1);
+
+    --depth;
 }
 #endif // wxUSE_MSGDLG
 
Index: samples/dialogs/dialogs.h
===================================================================
--- samples/dialogs/dialogs.h	(revision 63977)
+++ samples/dialogs/dialogs.h	(working copy)
@@ -121,6 +121,9 @@
 #if wxUSE_LOG
     virtual wxAppTraits *CreateTraits() { return new MyAppTraits; }
 #endif // wxUSE_LOG
+
+private:
+    void ShowNewFrame();
 };
 
 #if USE_MODAL_PRESENTATION
