diff -bru wxWidgets17089.orig/src/gtk/hyperlink.cpp wxWidgets17089/src/gtk/hyperlink.cpp
--- wxWidgets17089.orig/src/gtk/hyperlink.cpp	2015-08-02 01:59:34.512265881 -0400
+++ wxWidgets17089/src/gtk/hyperlink.cpp	2015-08-02 10:24:08.671786804 -0400
@@ -55,22 +55,24 @@
 // ----------------------------------------------------------------------------
 
 extern "C" {
-static void gtk_hyperlink_clicked_callback( GtkWidget *WXUNUSED(widget),
+static void gtk_hyperlink_clicked_callback( GtkWidget *widget,
                                             wxHyperlinkCtrl *linkCtrl )
 {
     // send the event
+    gtk_link_button_set_visited( GTK_LINK_BUTTON( widget ), TRUE );
     linkCtrl->SendEvent();
 }
 }
 
 extern "C" {
-static bool gtk3_hyperlink_clicked_callback( GtkWidget *WXUNUSED(widget), wxHyperlinkCtrl *linkCtrl )
+static bool gtk3_hyperlink_clicked_callback( GtkWidget *widget, wxHyperlinkCtrl *linkCtrl )
 {
     bool result = FALSE;
     wxHyperlinkEvent event( linkCtrl, linkCtrl->GetId(), linkCtrl->GetURL() );
     event.SetEventObject( linkCtrl );
     if( !linkCtrl->GetEventHandler()->ProcessEvent( event ) )
         result = TRUE;
+    gtk_link_button_set_visited( GTK_LINK_BUTTON( widget ), TRUE );
     return result;
 }
 }
