diff -bru wxWidgets17089.orig/src/gtk/hyperlink.cpp wxWidgets17089/src/gtk/hyperlink.cpp
--- wxWidgets17089.orig/src/gtk/hyperlink.cpp	2015-08-02 01:59:34.512265881 -0400
+++ wxWidgets17089/src/gtk/hyperlink.cpp	2015-08-02 02:23:49.290078655 -0400
@@ -209,6 +209,19 @@
     wxColour ret;
     if ( UseNative() )
     {
+#if GTK_CHECK_VERSION( 3, 14, 0 )
+        if( gtk_check_version( 3, 14, 0 ) )
+	{
+            GdkRGBA *link_color = NULL;
+	    gtk_widget_style_get( m_widget, "link_color", &link_color, NULL );
+	    if( link_color )
+	        ret = wxColour( *link_color );
+	    gdk_rgba_free( link_color );
+	}
+	else
+#endif
+	{
+	    wxGCC_WARNING_SUPPRESS(deprecated-declarations)
         GdkColor *link_color = NULL;
 
         // convert GdkColor in wxColour
@@ -216,6 +229,8 @@
         if (link_color)
             ret = wxColour(*link_color);
         gdk_color_free (link_color);
+            wxGCC_WARNING_RESTORE()
+        }
     }
     else
         ret = wxGenericHyperlinkCtrl::GetNormalColour();
@@ -238,6 +253,19 @@
     wxColour ret;
     if ( UseNative() )
     {
+#if GTK_CHECK_VERSION( 3, 14, 0 )
+        if( gtk_check_version( 3, 14, 0 ) )
+	{
+            GdkRGBA *link_color = NULL;
+	    gtk_widget_style_get( m_widget, "visited-link-color", &link_color, NULL );
+	    if( link_color )
+	        ret = wxColour( *link_color );
+	    gdk_rgba_free( link_color );
+	}
+	else
+#endif
+	{
+            wxGCC_WARNING_SUPPRESS(deprecated-declarations)
         GdkColor *link_color = NULL;
 
         // convert GdkColor in wxColour
@@ -245,6 +273,8 @@
         if (link_color)
             ret = wxColour(*link_color);
         gdk_color_free (link_color);
+            wxGCC_WARNING_RESTORE()
+        }
     }
     else
         return wxGenericHyperlinkCtrl::GetVisitedColour();
