Index: src/aui/auibook.cpp
===================================================================
--- src/aui/auibook.cpp	(Revision 61198)
+++ src/aui/auibook.cpp	(Arbeitskopie)
@@ -4443,24 +4443,34 @@
 
     int currentSelection = GetSelection();
 
-    if (forward)
+    wxAuiTabCtrl* tabCtrl = 0;
+    int idx = -1;
+
+    if(!FindTab(GetPage(currentSelection), &tabCtrl, &idx))
+        return;
+
+    if(!tabCtrl || idx < 0)
+        return;
+
+    wxWindow* page = 0;
+    size_t maxPages = tabCtrl->GetPageCount();
+
+    forward?idx++:idx--;
+
+    if(idx < 0)
+        idx = maxPages - 1;
+
+    if ((size_t)idx < maxPages)
+        page = tabCtrl->GetPage(idx).window;
+
+    if (!page && maxPages > 0)
+        page = tabCtrl->GetPage(0).window;
+
+    if(page)
     {
-        if (currentSelection == (int) (GetPageCount() - 1))
-            return;
-        else if (currentSelection == -1)
-            currentSelection = 0;
-        else
-            currentSelection ++;
+        currentSelection = GetPageIndex(page);
+        SetSelection(currentSelection);
     }
-    else
-    {
-        if (currentSelection <= 0)
-            return;
-        else
-            currentSelection --;
-    }
-
-    SetSelection(currentSelection);
 }
 
 // Shows the window menu
