diff -ru wxWidgets.orig/include/wx/image.h wxWidgets/include/wx/image.h
--- wxWidgets.orig/include/wx/image.h	2007-08-28 09:23:56.000000000 +0200
+++ wxWidgets/include/wx/image.h	2007-08-28 22:10:52.000000000 +0200
@@ -409,16 +409,14 @@
     static HSVValue RGBtoHSV(const RGBValue& rgb);
     static RGBValue HSVtoRGB(const HSVValue& hsv);
 
-
-protected:
-    static wxList   sm_handlers;
-
     // return the index of the point with the given coordinates or -1 if the
     // image is invalid of the coordinates are out of range
-    //
     // note that index must be multiplied by 3 when using it with RGB array
     long XYToIndex(int x, int y) const;
 
+protected:
+    static wxList   sm_handlers;
+
     virtual wxObjectRefData* CreateRefData() const;
     virtual wxObjectRefData* CloneRefData(const wxObjectRefData* data) const;
 
diff -ru wxWidgets.orig/src/common/image.cpp wxWidgets/src/common/image.cpp
--- wxWidgets.orig/src/common/image.cpp	2007-08-28 09:24:00.000000000 +0200
+++ wxWidgets/src/common/image.cpp	2007-08-28 22:10:23.000000000 +0200
@@ -1459,7 +1459,7 @@
         return y*M_IMGDATA->m_width + x;
     }
 
-    return -1;
+    return wxNOT_FOUND;
 }
 
 void wxImage::SetRGB( int x, int y, unsigned char r, unsigned char g, unsigned char b )

 	  	 
