Index: src/aui/framemanager.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/aui/framemanager.cpp,v
retrieving revision 1.76
diff -u -r1.76 framemanager.cpp
--- src/aui/framemanager.cpp	2006/11/02 20:50:01	1.76
+++ src/aui/framemanager.cpp	2006/11/02 22:05:40
@@ -3503,10 +3505,10 @@
 
 void wxAuiManager::OnFloatingPaneActivated(wxWindow* wnd)
 {
-    if (GetFlags() & wxAUI_MGR_ALLOW_ACTIVE_PANE)
+    if ((GetFlags() & wxAUI_MGR_ALLOW_ACTIVE_PANE) && GetPane(wnd).IsOk())
     {
         // try to find the pane
-        wxASSERT_MSG(GetPane(wnd).IsOk(), wxT("Pane window not found"));
+        //wxASSERT_MSG(GetPane(wnd).IsOk(), wxT("Pane window not found"));
 
         SetActivePane(m_panes, wnd);
         Repaint();
@@ -3533,7 +3533,7 @@
         wxAuiDockUIPart& part = m_uiparts.Item(i);
 
         // don't draw hidden pane items
-        if (part.sizer_item && !part.sizer_item->IsShown())
+        if (part.sizer_item && ((!part.sizer_item->IsWindow() && !part.sizer_item->IsSpacer() && !part.sizer_item->IsSizer()) || !part.sizer_item->IsShown()))
             continue;
 
         switch (part.type)

 	  	 
