Index: helpview.cpp
===================================================================
--- helpview.cpp	(revision 67370)
+++ helpview.cpp	(working copy)
@@ -90,6 +90,17 @@
 
     help -> DisplayContents();
 
+// test contigute matches
+//    help->GetHelpWindow()->GetHtmlWindow()->SearchAndHighlight (wxT("o"), false, false);
+
+// test matches with different response in whole word/case sensitive
+//    help->GetHelpWindow()->GetHtmlWindow()->SearchAndHighlight (wxT("in"), false, false);
+//    help->GetHelpWindow()->GetHtmlWindow()->SearchAndHighlight (wxT("in"), true, false);
+//    help->GetHelpWindow()->GetHtmlWindow()->SearchAndHighlight (wxT("in"), false, true);
+
+// test matches other several cells
+    help->GetHelpWindow()->GetHtmlWindow()->SearchAndHighlight (wxT("to demonstrate that"), false, false);
+
     return true;
 }
 
