Index: interface/wx/bookctrl.h
===================================================================
--- interface/wx/bookctrl.h	(revision 70852)
+++ interface/wx/bookctrl.h	(working copy)
@@ -7,6 +7,18 @@
 /////////////////////////////////////////////////////////////////////////////
 
 /**
+    wxBookCtrl hit results
+ */
+enum
+{
+    wxBK_HITTEST_NOWHERE = 1,   // not on tab
+    wxBK_HITTEST_ONICON  = 2,   // on icon
+    wxBK_HITTEST_ONLABEL = 4,   // on label
+    wxBK_HITTEST_ONITEM  = wxBK_HITTEST_ONICON | wxBK_HITTEST_ONLABEL,
+    wxBK_HITTEST_ONPAGE  = 8    // not on tab control, but over the selected page
+};
+
+/**
     @class wxBookCtrlBase
 
     A book control is a convenient way of displaying multiple pages of information,
