diff -bru wxWidgets.orig/src/gtk/hyperlink.cpp wxWidgets.patched/src/gtk/hyperlink.cpp
--- wxWidgets.orig/src/gtk/hyperlink.cpp        2015-08-27 22:39:49.928688906 -0400
+++ wxWidgets.patched/src/gtk/hyperlink.cpp     2015-08-28 21:20:19.854174403 -0400
@@ -56,9 +56,10 @@
 
 #ifdef __WXGTK3__
 extern "C" {
-static gboolean activate_link(GtkWidget*, wxHyperlinkCtrl* win)
+static gboolean activate_link(GtkWidget *widget, wxHyperlinkCtrl* win)
 {
     win->SendEvent();
+    gtk_link_button_set_visited( GTK_LINK_BUTTON( widget ), TRUE );
     return true;
 }
 }
@@ -73,6 +74,7 @@
         if (win->m_widget == (GtkWidget*)button)
         {
             win->SendEvent();
+            gtk_link_button_set_visited( button, TRUE );
             return;
         }
     }
