 samples/minimal/minimal.cpp | 23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)

diff --git a/samples/minimal/minimal.cpp b/samples/minimal/minimal.cpp
index a78e462..f685b8c 100644
--- a/samples/minimal/minimal.cpp
+++ b/samples/minimal/minimal.cpp
@@ -69,6 +69,7 @@ public:
     void OnAbout(wxCommandEvent& event);
 
 private:
+    void OnPaintPanel(wxPaintEvent&);
     // any class wishing to process wxWidgets events must use this macro
     DECLARE_EVENT_TABLE()
 };
@@ -172,10 +173,32 @@ MyFrame::MyFrame(const wxString& title)
     CreateStatusBar(2);
     SetStatusText("Welcome to wxWidgets!");
 #endif // wxUSE_STATUSBAR
+
+    wxPanel* panel = new wxPanel(this);
+    panel->Bind(wxEVT_PAINT, &MyFrame::OnPaintPanel, this);
 }
 
 
 // event handlers
+void MyFrame::OnPaintPanel(wxPaintEvent& event)
+{
+    wxPaintDC dc((wxWindow*)event.GetEventObject());
+
+    wxFont fn;
+    fn.SetNativeFontInfo("0;-20;0;0;0;400;0;0;0;0;143;0;0;0;MS Shell Dlg 2");  // pixel size 20
+    dc.SetFont(fn);
+    dc.DrawText("This is a Teststring", 10, 10);
+
+    fn.SetPointSize(fn.GetPointSize());
+    dc.SetFont(fn);
+    dc.DrawText("This is a Teststring", 10, 35);
+
+    wxFont fn2;
+    fn2.SetFaceName("MS Shell Dlg 2");
+    fn2.SetPixelSize(wxSize(0, 20));
+    dc.SetFont(fn2);
+    dc.DrawText("This is a Teststring", 10, 60);
+}
 
 void MyFrame::OnQuit(wxCommandEvent& WXUNUSED(event))
 {
