--- grid_orig.cpp	2005-04-28 06:47:14.701102064 +1000
+++ grid.cpp	2005-04-28 06:48:20.752060792 +1000
@@ -7230,5 +7230,5 @@
 void wxGrid::DrawRowLabel( wxDC& dc, int row )
 {
-    if ( GetRowHeight(row) <= 0 )
+    if ( GetRowHeight(row) <= 0 || m_rowLabelWidth <= 0 )
         return;
 
@@ -7292,5 +7292,5 @@
 void wxGrid::DrawColLabel( wxDC& dc, int col )
 {
-    if ( GetColWidth(col) <= 0 )
+    if ( GetColWidth(col) <= 0 || m_colLabelHeight <= 0 )
         return;
 

 	  	 
