diff -ru2 -xCVS -x'.#*' -x'*.orig' -x'*.rej' -x'*~' wxWidgets.org/src/msw/listctrl.cpp wxWidgets/src/msw/listctrl.cpp
--- wxWidgets.org/src/msw/listctrl.cpp	2004-07-16 15:18:07.000000000 +0200
+++ wxWidgets/src/msw/listctrl.cpp	2004-08-31 07:18:15.000000000 +0200
@@ -1744,5 +1744,5 @@
                     }
 
-                    if ( !::ScreenToClient(hwndHdr, &ptClick) )
+                    if ( !::ScreenToClient(GetHwnd(), &ptClick) )
                     {
                         wxLogLastError(_T("ScreenToClient(listctrl header)"));

 	  	 
