--- listtest.cpp.orig	2004-05-23 15:01:52.000000000 +0200
+++ listtest.cpp	2004-08-31 22:44:12.000000000 +0200
@@ -434,5 +434,5 @@
     m_listCtrl->Hide();
 
-    wxStopWatch sw;
+    //wxStopWatch sw;
 
     for ( int i = 0; i < NUM_ITEMS; i++ )
@@ -441,6 +441,7 @@
     }
 
-    m_logWindow->WriteText(wxString::Format(_T("%d items inserted in %ldms\n"),
-                                            NUM_ITEMS, sw.Time()));
+    //m_logWindow->WriteText(wxString::Format(_T("%d items inserted in %ldms\n"),
+    //                                        NUM_ITEMS, sw.Time()));
+
     m_listCtrl->Show();
 
@@ -528,11 +529,11 @@
 void MyFrame::OnSort(wxCommandEvent& WXUNUSED(event))
 {
-    wxStopWatch sw;
+    //wxStopWatch sw;
 
     m_listCtrl->SortItems(MyCompareFunction, 0);
 
-    m_logWindow->WriteText(wxString::Format(_T("Sorting %d items took %ld ms\n"),
-                                            m_listCtrl->GetItemCount(),
-                                            sw.Time()));
+    //m_logWindow->WriteText(wxString::Format(_T("Sorting %d items took %ld ms\n"),
+    //                                        m_listCtrl->GetItemCount(),
+    //                                        sw.Time()));
 }
 
@@ -600,5 +601,5 @@
 void MyFrame::OnSetFgColour(wxCommandEvent& WXUNUSED(event))
 {
-    m_listCtrl->SetForegroundColour(wxGetColourFromUser(this));
+    //m_listCtrl->SetForegroundColour(wxGetColourFromUser(this));
     m_listCtrl->Refresh();
 }
@@ -606,5 +607,5 @@
 void MyFrame::OnSetBgColour(wxCommandEvent& WXUNUSED(event))
 {
-    m_listCtrl->SetBackgroundColour(wxGetColourFromUser(this));
+    //m_listCtrl->SetBackgroundColour(wxGetColourFromUser(this));
     m_listCtrl->Refresh();
 }
@@ -644,11 +645,11 @@
 void MyFrame::OnDeleteAll(wxCommandEvent& WXUNUSED(event))
 {
-    wxStopWatch sw;
+    //wxStopWatch sw;
 
     m_listCtrl->DeleteAllItems();
 
-    m_logWindow->WriteText(wxString::Format(_T("Deleting %d items took %ld ms\n"),
-                                            m_listCtrl->GetItemCount(),
-                                            sw.Time()));
+    //m_logWindow->WriteText(wxString::Format(_T("Deleting %d items took %ld ms\n"),
+    //                                        m_listCtrl->GetItemCount(),
+    //                                        sw.Time()));
 }
 
@@ -685,4 +686,8 @@
     }
 
+	// Show popupmenu at position
+	wxMenu menu("Test");
+	PopupMenu(&menu, event.GetPoint());
+
     wxLogMessage( wxT("OnColumnRightClick at %d."), event.GetColumn() );
 }

 	  	 
