*** src/xml/xml.cpp	Tue Jan 13 12:30:46 2015
--- src/xml/xml.orig	Sun Oct 26 01:01:50 2014
***************
*** 20,25 ****
--- 20,26 ----
  
  #ifndef WX_PRECOMP
      #include "wx/intl.h"
+     #include "wx/log.h"
      #include "wx/app.h"
  #endif
  
***************
*** 840,848 ****
          {
              wxString error(XML_ErrorString(XML_GetErrorCode(parser)),
                             *wxConvCurrent);
!             SetError(wxString::Format("XML parsing error: '%s' at line %d",
                         error.c_str(),
!                        (int)XML_GetCurrentLineNumber(parser)));
              ok = false;
              break;
          }
--- 841,849 ----
          {
              wxString error(XML_ErrorString(XML_GetErrorCode(parser)),
                             *wxConvCurrent);
!             wxLogError(_("XML parsing error: '%s' at line %d"),
                         error.c_str(),
!                        (int)XML_GetCurrentLineNumber(parser));
              ok = false;
              break;
          }
***************
*** 850,856 ****
  
      if (ok)
      {
-         SetError(wxEmptyString);
          if (!ctx.version.empty())
              SetVersion(ctx.version);
          if (!ctx.encoding.empty())
--- 851,856 ----
