diff --git src/univ/settingsuniv.cpp src/univ/settingsuniv.cpp
index a382d39..5bafa05 100644
--- src/univ/settingsuniv.cpp
+++ src/univ/settingsuniv.cpp
@@ -77,7 +77,7 @@ wxColour wxSystemSettings::GetColour(wxSystemColour index)
         wxColourScheme::TITLEBAR_TEXT /* wxSYS_COLOUR_GRADIENTINACTIVECAPTION */,
         wxColourScheme::MAX /* wxSYS_COLOUR_MENUHILIGHT */,
         wxColourScheme::MAX /* wxSYS_COLOUR_MENUBAR */,
-        wxColourScheme::CONTROL_TEXT /* wxSYS_COLOUR_LISTBOXTEXT */,
+        wxColourScheme::CONTROL_TEXT /* wxSYS_COLOUR_LISTBOXTEXT */
     };
 
     wxCOMPILE_TIME_ASSERT( WXSIZEOF(s_mapSysToThemeCol) == wxSYS_COLOUR_MAX,
