Index: softwareupdate.py
===================================================================
--- softwareupdate.py	(revision 73453)
+++ softwareupdate.py	(working copy)
@@ -343,9 +343,9 @@
         if sys.platform == "darwin" and hasattr(sys, 'frozen') \
            and sys.frozen == 'macosx_app' and sys.executable.endswith('MacOS/python'):
                 names = os.listdir(os.path.dirname(sys.executable))
-                assert len(names) == 2  # there should be only 2
+                assert len(names) in (2, 4)  # there should be only 2 but may be 4
                 for name in names: 
-                    if name != 'python':
+                    if not (name == 'python' or name.startswith('._')):
                         sys.executable = os.path.join(os.path.dirname(sys.executable), name)
                         break
                     
