Index: edit.cpp
===================================================================
--- edit.cpp	(revision 64204)
+++ edit.cpp	(working copy)
@@ -103,6 +103,7 @@
     EVT_STC_MARGINCLICK (wxID_ANY,     Edit::OnMarginClick)
     EVT_STC_CHARADDED (wxID_ANY,       Edit::OnCharAdded)
     EVT_STC_KEY( wxID_ANY , Edit::OnKey )
+    EVT_STC_UPDATEUI( wxID_ANY, Edit::OnUpdateUI )
 END_EVENT_TABLE()
 
 Edit::Edit (wxWindow *parent, wxWindowID id,
@@ -195,6 +196,16 @@
     wxMessageBox("OnKey");
 }
 
+void Edit::OnUpdateUI  (wxStyledTextEvent &WXUNUSED(event))
+{
+    int i = GetCurrentPos();
+    int j = BraceMatch(i);
+    if(j != -1)
+        BraceHighlight(i, j);
+    else
+        BraceBadLight(-1);
+}
+
 void Edit::OnEditCut (wxCommandEvent &WXUNUSED(event)) {
     if (GetReadOnly() || (GetSelectionEnd()-GetSelectionStart() <= 0)) return;
     Cut ();
Index: edit.h
===================================================================
--- edit.h	(revision 64204)
+++ edit.h	(working copy)
@@ -98,6 +98,7 @@
     void OnMarginClick (wxStyledTextEvent &event);
     void OnCharAdded  (wxStyledTextEvent &event);
     void OnKey  (wxStyledTextEvent &event);
+    void OnUpdateUI  (wxStyledTextEvent &event);
 
     //! language/lexer
     wxString DeterminePrefs (const wxString &filename);
