Index: include/wx/msw/iniconf.h
===================================================================
--- include/wx/msw/iniconf.h	(revision 59411)
+++ include/wx/msw/iniconf.h	(working copy)
@@ -12,6 +12,10 @@
 #ifndef   _INICONF_H
 #define   _INICONF_H
 
+#include "wx/defs.h"
+
+#if wxUSE_CONFIG
+
 // ----------------------------------------------------------------------------
 // wxIniConfig is a wxConfig implementation which uses MS Windows INI files to
 // store the data. Because INI files don't really support arbitrary nesting of
@@ -92,4 +96,6 @@
            m_strPath;     // the rest of the path (no trailing '_'!)
 };
 
-#endif  //_INICONF_H
+#endif // wxUSE_CONFIG
+
+#endif // _INICONF_H
Index: include/wx/msw/regconf.h
===================================================================
--- include/wx/msw/regconf.h	(revision 59411)
+++ include/wx/msw/regconf.h	(working copy)
@@ -12,6 +12,10 @@
 #ifndef   _REGCONF_H
 #define   _REGCONF_H
 
+#include "wx/defs.h"
+
+#if wxUSE_CONFIG
+
 #ifndef   _REGISTRY_H
   #include "wx/msw/registry.h"
 #endif
@@ -107,4 +111,6 @@
   wxString  m_strPath;
 };
 
-#endif  //_REGCONF_H
+#endif // wxUSE_CONFIG
+
+#endif // _REGCONF_H
Index: include/wx/palmos/prefconf.h
===================================================================
--- include/wx/palmos/prefconf.h	(revision 59411)
+++ include/wx/palmos/prefconf.h	(working copy)
@@ -12,6 +12,10 @@
 #ifndef _PREFCONF_H_
 #define _PREFCONF_H_
 
+#include "wx/defs.h"
+
+#if wxUSE_CONFIG
+
 // ----------------------------------------------------------------------------
 // wxPrefConfig
 // ----------------------------------------------------------------------------
@@ -86,5 +90,6 @@
   bool m_modGroup;
 };
 
+#endif // wxUSE_CONFIG
+
 #endif // _PREFCONF_H_
-
