Index: include/wx/msw/iniconf.h
===================================================================
--- include/wx/msw/iniconf.h	(revision 59411)
+++ include/wx/msw/iniconf.h	(working copy)
@@ -12,8 +12,10 @@
 #ifndef   _WX_MSW_INICONF_H_
 #define   _WX_MSW_INICONF_H_
 
-#if wxUSE_INICONF
+#include "wx/defs.h"
 
+#if wxUSE_CONFIG && wxUSE_INICONF
+
 // ----------------------------------------------------------------------------
 // wxIniConfig is a wxConfig implementation which uses MS Windows INI files to
 // store the data. Because INI files don't really support arbitrary nesting of
@@ -99,6 +101,6 @@
   DECLARE_ABSTRACT_CLASS(wxIniConfig)
 };
 
-#endif // wxUSE_INICONF
+#endif // wxUSE_CONFIG && wxUSE_INICONF
 
-#endif  //_WX_MSW_INICONF_H_
+#endif // _WX_MSW_INICONF_H_
Index: include/wx/palmos/prefconf.h
===================================================================
--- include/wx/palmos/prefconf.h	(revision 59411)
+++ include/wx/palmos/prefconf.h	(working copy)
@@ -12,6 +12,10 @@
 #ifndef _PREFCONF_H_
 #define _PREFCONF_H_
 
+#include "wx/defs.h"
+
+#if wxUSE_CONFIG
+
 // ----------------------------------------------------------------------------
 // wxPrefConfig
 // ----------------------------------------------------------------------------
@@ -87,5 +91,6 @@
   DECLARE_ABSTRACT_CLASS(wxPrefConfig)
 };
 
+#endif // wxUSE_CONFIG
+
 #endif // _PREFCONF_H_
-
