Index: controls.cpp
===================================================================
--- controls.cpp	(revision 50121)
+++ controls.cpp	(working copy)
@@ -1499,7 +1502,8 @@
             break;
 
         case ID_RADIOBOX_SEL_STR:
-            m_radio->SetStringSelection( _T("This") );
+            *m_text << "Radio selection: " << m_radio->GetSelection() << "\n";
             break;
 
         case ID_RADIOBOX_FONT:

 	  	 
