diff --git a/src/msw/dc.cpp b/src/msw/dc.cpp
index eee91b4..7013824 100644
--- a/src/msw/dc.cpp
+++ b/src/msw/dc.cpp
@@ -2494,12 +2494,29 @@ bool wxMSWDCImpl::DoStretchBlit(wxCoord xdest, wxCoord ydest,
         {
             StretchBltModeChanger stretchModeChanger(GetHdc());
 
+            /*
+            Workaround for #19190. See 6614aa496d: "For some reason in RTL
+            mode, source offset has to be -1, otherwise the right border
+            (physical) remains unpainted." [note that the offset actually
+            is applied not only to the source but also dest]
+
+            Be conservative about using the offset and only do it for
+            currently known failing cases: when xdest and xsrc are 0 and only
+            when not actually stretching. The former checks also prevent
+            the offset from being applied twice, e.g. along with the mentioned
+            patch of 6614aa496d).
+            */
+            const wxCoord ofs = (GetLayoutDirection() == wxLayout_RightToLeft
+                                 && !xdest && !xsrc
+                                 && srcWidth == dstWidth
+                                 && srcHeight == dstHeight) ? -1 : 0;
+
             if ( !::StretchBlt
                     (
                         GetHdc(),
-                        xdest, ydest, dstWidth, dstHeight,
+                        xdest + ofs, ydest, dstWidth, dstHeight,
                         hdcSrc,
-                        xsrc, ysrc, srcWidth, srcHeight,
+                        xsrc + ofs, ysrc, srcWidth, srcHeight,
                         dwRop
                     ) )
             {
