Index: src/msw/treectrl.cpp
===================================================================
--- src/msw/treectrl.cpp	(revision 63133)
+++ src/msw/treectrl.cpp	(working copy)
@@ -3173,6 +3173,9 @@
                     // unprocessed
                     MSWHandleSelectionKey(wParam);
                 }
+                
+                // Send key event to tree control
+                HandleKeyDown(wParam, lParam);
 
                 // pretend that we did process it in any case as we already
                 // generated an event for it
