Index: include/wx/generic/progdlgg.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/generic/progdlgg.h,v
retrieving revision 1.23
diff -b -u -2 -r1.23 progdlgg.h
--- include/wx/generic/progdlgg.h	2002/08/31 11:29:12	1.23
+++ include/wx/generic/progdlgg.h	2002/12/12 21:59:19
@@ -64,4 +64,6 @@
    void Resume();
 
+   bool Show( bool show = TRUE );
+
 protected:
    // callback for optional abort button
Index: src/generic/progdlgg.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/generic/progdlgg.cpp,v
retrieving revision 1.63
diff -b -u -2 -r1.63 progdlgg.cpp
--- src/generic/progdlgg.cpp	2002/12/04 14:07:31	1.63
+++ src/generic/progdlgg.cpp	2002/12/12 21:59:19
@@ -407,4 +407,15 @@
 }
 
+bool wxProgressDialog::Show( bool show )
+{
+    // reenable other windows before hiding this one because otherwise
+    // Windows wouldn't give the focus back to the window which had
+    // been previously focused because it would still be disabled
+    if(!show)
+        ReenableOtherWindows();
+
+    return wxDialog::Show(show);
+}
+
 // ----------------------------------------------------------------------------
 // event handlers

 	  	 
