diff -bu1 -r h:/wxcvshead/wxWindows/contrib/samples/plot/plot.cpp h:/wxcvshd/wxWin/contrib/samples/plot/plot.cpp
--- h:/wxcvshead/wxWindows/contrib/samples/plot/plot.cpp	2001-01-25 18:25:20.000000000 +0100
+++ h:/wxcvshd/wxWin/contrib/samples/plot/plot.cpp	2003-08-14 13:36:44.000000000 +0200
@@ -105,3 +105,3 @@
 MyFrame::MyFrame()
-       : wxFrame( (wxFrame *)NULL, -1, "wxPlotWindow sample",
+       : wxFrame( (wxFrame *)NULL, -1, _T("wxPlotWindow sample"),
                   wxPoint(20,20), wxSize(470,500) )
@@ -109,7 +109,7 @@
     wxMenu *file_menu = new wxMenu();
-    file_menu->Append( ID_ABOUT, "&About..");
-    file_menu->Append( ID_QUIT, "E&xit\tAlt-X");
+    file_menu->Append( ID_ABOUT, _T("&About.."));
+    file_menu->Append( ID_QUIT, _T("E&xit\tAlt-X"));
 
     wxMenuBar *menu_bar = new wxMenuBar();
-    menu_bar->Append(file_menu, "&File");
+    menu_bar->Append(file_menu, _T("&File"));
 
@@ -134,3 +134,3 @@
   
-    m_log = new wxTextCtrl( this, -1, "This is the log window.\n", wxPoint(0,0), wxSize(100,100), wxTE_MULTILINE );
+    m_log = new wxTextCtrl( this, -1, _T("This is the log window.\n"), wxPoint(0,0), wxSize(100,100), wxTE_MULTILINE );
     wxLog *old_log = wxLog::SetActiveTarget( new wxLogTextCtrl( m_log ) );
@@ -154,5 +154,5 @@
 {
-  (void)wxMessageBox( "wxPlotWindow Demo\n"
-                      "Robert Roebling (c) 1999,2000",
-                      "About wxPlotWindow Demo", wxICON_INFORMATION | wxOK );
+  (void)wxMessageBox( _T("wxPlotWindow Demo\n"
+                         "Robert Roebling (c) 1999,2000"),
+                      _T("About wxPlotWindow Demo"), wxICON_INFORMATION | wxOK );
 }
@@ -163,3 +163,3 @@
     double y = event.GetCurve()->GetY( event.GetPosition() );
-    wxLogMessage( "Clicked on curve at x coordinate: %f, value: %f", x, y );
+    wxLogMessage( _T("Clicked on curve at x coordinate: %f, value: %f"), x, y );
 }
@@ -170,3 +170,3 @@
     double y = event.GetCurve()->GetY( event.GetPosition() );
-    wxLogMessage( "Double clicked on curve at x coordinate: %f, value: %f", x, y );
+    wxLogMessage( _T("Double clicked on curve at x coordinate: %f, value: %f"), x, y );
 }
diff -bu1 -r h:/wxcvshead/wxWindows/contrib/src/plot/plot.cpp h:/wxcvshd/wxWin/contrib/src/plot/plot.cpp
--- h:/wxcvshead/wxWindows/contrib/src/plot/plot.cpp	2002-01-31 21:27:53.000000000 +0100
+++ h:/wxcvshd/wxWin/contrib/src/plot/plot.cpp	2003-08-14 13:33:36.000000000 +0200
@@ -138,3 +138,3 @@
 {
-    wxASSERT_MSG( on > 0, wxT("plot index < 0") );
+    wxASSERT_MSG( on > 0, _T("plot index < 0") );
     wxASSERT( on <= off );
@@ -206,3 +206,3 @@
 wxPlotArea::wxPlotArea( wxPlotWindow *parent )
-        : wxWindow( parent, -1, wxDefaultPosition, wxDefaultSize, wxSIMPLE_BORDER, "plotarea" )
+        : wxWindow( parent, -1, wxDefaultPosition, wxDefaultSize, wxSIMPLE_BORDER, _T("plotarea") )
 {
@@ -464,3 +464,3 @@
 wxPlotXAxisArea::wxPlotXAxisArea( wxPlotWindow *parent )
-        : wxWindow( parent, -1, wxDefaultPosition, wxSize(-1,40), 0, "plotxaxisarea" )
+        : wxWindow( parent, -1, wxDefaultPosition, wxSize(-1,40), 0, _T("plotxaxisarea") )
 {
@@ -545,3 +545,3 @@
     dc.SetBrush( *wxWHITE_BRUSH );
-    dc.SetPen( *wxTRANSPARENT_PEN );
+    dc.SetPen( *_TRANSPARENT_PEN );
     dc.DrawRectangle( 4, 5, client_width-14, 10 );
@@ -560,10 +560,10 @@
             {
-                label.Printf( wxT("%f"), current );
-                while (label.Last() == wxT('0')) 
+                label.Printf( _T("%f"), current );
+                while (label.Last() == _T('0')) 
                     label.RemoveLast();
-                if ((label.Last() == wxT('.')) || (label.Last() == wxT(',')))
-                    label.Append( wxT('0') );
+                if ((label.Last() == _T('.')) || (label.Last() == _T(',')))
+                    label.Append( _T('0') );
             }
             else
-                label.Printf( wxT("%d"), (int)floor(current) );
+                label.Printf( _T("%d"), (int)floor(current) );
             dc.DrawText( label, x-4, 20 );
@@ -591,3 +591,3 @@
 wxPlotYAxisArea::wxPlotYAxisArea( wxPlotWindow *parent )
-        : wxWindow( parent, -1, wxDefaultPosition, wxSize(60,-1), 0, "plotyaxisarea" )
+        : wxWindow( parent, -1, wxDefaultPosition, wxSize(60,-1), 0, _T("plotyaxisarea") )
 {
@@ -666,10 +666,10 @@
             {
-                label.Printf( wxT("%f"), current );
-                while (label.Last() == wxT('0')) 
+                label.Printf( _T("%f"), current );
+                while (label.Last() == _T('0')) 
                     label.RemoveLast();
-                if ((label.Last() == wxT('.')) || (label.Last() == wxT(',')))
-                    label.Append( wxT('0') );
+                if ((label.Last() == _T('.')) || (label.Last() == _T(',')))
+                    label.Append( _T('0') );
             }
             else
-                label.Printf( wxT("%d"), (int)floor(current) );
+                label.Printf( _T("%d"), (int)floor(current) );
             dc.DrawText( label, 5, y-7 );
@@ -715,3 +715,3 @@
 wxPlotWindow::wxPlotWindow( wxWindow *parent, wxWindowID id, const wxPoint &pos, const wxSize &size, int flag )
-        : wxScrolledWindow( parent, id, pos, size, flag, "plotcanvas" )
+        : wxScrolledWindow( parent, id, pos, size, flag, _T("plotcanvas") )
 {
@@ -1054,3 +1054,3 @@
         #if defined(__WXMSW__) || defined(__WXPM__)
-            s_bitmap = new wxBitmap("plot_enl_bmp", wxBITMAP_TYPE_RESOURCE);
+            s_bitmap = new wxBitmap(_T("plot_enl_bmp"), wxBITMAP_TYPE_RESOURCE);
         #else
@@ -1073,3 +1073,3 @@
         #if defined(__WXMSW__) || defined(__WXPM__)
-            s_bitmap = new wxBitmap("plot_shr_bmp", wxBITMAP_TYPE_RESOURCE);
+            s_bitmap = new wxBitmap(_T("plot_shr_bmp"), wxBITMAP_TYPE_RESOURCE);
         #else
@@ -1092,3 +1092,3 @@
         #if defined(__WXMSW__) || defined(__WXPM__)
-            s_bitmap = new wxBitmap("plot_zin_bmp", wxBITMAP_TYPE_RESOURCE);
+            s_bitmap = new wxBitmap(_T("plot_zin_bmp"), wxBITMAP_TYPE_RESOURCE);
         #else
@@ -1111,3 +1111,3 @@
         #if defined(__WXMSW__) || defined(__WXPM__)
-            s_bitmap = new wxBitmap("plot_zot_bmp", wxBITMAP_TYPE_RESOURCE);
+            s_bitmap = new wxBitmap(_T("plot_zot_bmp"), wxBITMAP_TYPE_RESOURCE);
         #else
@@ -1130,3 +1130,3 @@
         #if defined(__WXMSW__) || defined(__WXPM__)
-            s_bitmap = new wxBitmap("plot_up_bmp", wxBITMAP_TYPE_RESOURCE);
+            s_bitmap = new wxBitmap(_T("plot_up_bmp"), wxBITMAP_TYPE_RESOURCE);
         #else
@@ -1149,3 +1149,3 @@
         #if defined(__WXMSW__) || defined(__WXPM__)
-            s_bitmap = new wxBitmap("plot_dwn_bmp", wxBITMAP_TYPE_RESOURCE);
+            s_bitmap = new wxBitmap(_T("plot_dwn_bmp"), wxBITMAP_TYPE_RESOURCE);
         #else

 	  	 
