--- src/aui/framemanager.cpp.orig	2006-11-13 22:15:26.872252800 -0800
+++ src/aui/framemanager.cpp	2006-11-15 04:43:58.802776000 -0800
@@ -2146,9 +2151,6 @@
     wxSizer* sizer;
     int i, pane_count = m_panes.GetCount();
 
-    // delete old sizer first
-    m_frame->SetSizer(NULL);
-
     // destroy floating panes which have been
     // redocked or are becoming non-floating
     for (i = 0; i < pane_count; ++i)
@@ -2177,6 +2179,8 @@
         }
     }
 
+    // delete old sizer first
+    m_frame->SetSizer(NULL);
 
     // create a layout for all of the panes
     sizer = LayoutAll(m_panes, m_docks, m_uiparts, false);
 
 

 	  	 
