Index: include/wx/filename.h
===================================================================
--- include/wx/filename.h	(revision 73604)
+++ include/wx/filename.h	(working copy)
@@ -490,9 +490,9 @@
 
     // Dir accessors
     size_t GetDirCount() const { return m_dirs.size(); }
-    void AppendDir(const wxString& dir);
+    bool AppendDir(const wxString& dir);
     void PrependDir(const wxString& dir);
-    void InsertDir(size_t before, const wxString& dir);
+    bool InsertDir(size_t before, const wxString& dir);
     void RemoveDir(size_t pos);
     void RemoveLastDir() { RemoveDir(GetDirCount() - 1); }
 
Index: interface/wx/filename.h
===================================================================
--- interface/wx/filename.h	(revision 73604)
+++ interface/wx/filename.h	(working copy)
@@ -341,8 +341,9 @@
         single directory name level, i.e. not contain any path or volume separators nor
         should it be empty, otherwise the function does nothing (and generates an
         assert failure in debug build).
+        Returns @true if @a dir is valid.
     */
-    void AppendDir(const wxString& dir);
+    bool AppendDir(const wxString& dir);
 
     /**
         Creates the file name from another filename object.
@@ -843,8 +844,9 @@
     /**
         Inserts a directory component before the zero-based position in the directory
         list. Please see AppendDir() for important notes.
+        Returns @true if @a dir is valid.
     */
-    void InsertDir(size_t before, const wxString& dir);
+    bool InsertDir(size_t before, const wxString& dir);
 
     /**
         Returns @true if this filename is absolute.
Index: src/common/filename.cpp
===================================================================
--- src/common/filename.cpp	(revision 73604)
+++ src/common/filename.cpp	(working copy)
@@ -2034,10 +2034,12 @@
     return true;
 }
 
-void wxFileName::AppendDir( const wxString& dir )
+bool wxFileName::AppendDir( const wxString& dir )
 {
-    if ( IsValidDirComponent(dir) )
-        m_dirs.Add( dir );
+    if (!IsValidDirComponent(dir))
+        return false;
+    m_dirs.Add(dir);
+    return true;
 }
 
 void wxFileName::PrependDir( const wxString& dir )
@@ -2045,10 +2047,12 @@
     InsertDir(0, dir);
 }
 
-void wxFileName::InsertDir(size_t before, const wxString& dir)
+bool wxFileName::InsertDir(size_t before, const wxString& dir)
 {
-    if ( IsValidDirComponent(dir) )
-        m_dirs.Insert(dir, before);
+    if (!IsValidDirComponent(dir))
+        return false;
+    m_dirs.Insert(dir, before);
+    return true;
 }
 
 void wxFileName::RemoveDir(size_t pos)
