--- \include\wx\aui\auibook.h    Thu Mar 01 17:04:59 2007
+++ auibook.h    Tue Mar 06 19:15:16 2007
@@ -452,6 +452,7 @@
     void OnSize(wxSizeEvent& evt);
     void OnLeftDown(wxMouseEvent& evt);
     void OnLeftUp(wxMouseEvent& evt);
+    void OnContextMenu(wxMouseEvent& evt);
     void OnMotion(wxMouseEvent& evt);
     void OnLeaveWindow(wxMouseEvent& evt);
     void OnButton(wxAuiNotebookEvent& evt);
@@ -529,6 +530,9 @@
     size_t SetSelection(size_t new_page);
     int GetSelection() const;
 
+    void SetContextMenu(wxMenu* menu);
+    int GetContextPage() const { return m_context_page; };
+
     virtual void Split(size_t page, int direction);
 
 #if wxABI_VERSION >= 20801
@@ -563,6 +567,7 @@
     void OnTabDragMotion(wxCommandEvent& evt);
     void OnTabEndDrag(wxCommandEvent& evt);
     void OnTabButton(wxCommandEvent& evt);
+    void OnContextMenu(wxCommandEvent& evt);
     
 protected:
 
@@ -571,6 +576,9 @@
     int m_curpage;
     int m_tab_id_counter;
     wxWindow* m_dummy_wnd;
+    
+    wxMenu* m_context_menu;
+    int m_context_page;
 
     wxSize m_requested_bmp_size;
     int m_requested_tabctrl_height;

 	  	 
