Index: logg.cpp
===================================================================
--- logg.cpp	(revision 71715)
+++ logg.cpp	(working copy)
@@ -771,6 +771,9 @@
 
     wxSizerFlags flagsBtn;
     flagsBtn.Border(wxLEFT);
+#ifdef __WXMAC__
+    flagsBtn.Border(wxLEFT|wxBOTTOM);// without a bottom border on mac, the bottom edge of the buttons is cut off
+#endif // #ifdef __WXMAC__
 
 #if wxUSE_CLIPBOARD
     btnSizer->Add(new wxButton(win, wxID_COPY), flagsBtn);
