diff -ru wxGTK-2.4.0/include/wx/app.h wxGTK-2.4.0-debug/include/wx/app.h
--- wxGTK-2.4.0/include/wx/app.h	2003-01-05 12:55:00.000000000 +0100
+++ wxGTK-2.4.0-debug/include/wx/app.h	2003-02-28 15:16:04.000000000 +0100
@@ -537,10 +537,14 @@
 // be in your main program (e.g. hello.cpp). Now IMPLEMENT_APP should add this
 // code if required.
 
-#if !wxUSE_GUI || defined(__WXMOTIF__) || defined(__WXGTK__) || defined(__WXPM__) || defined(__WXMGL__)
+#if !wxUSE_GUI || defined(__WXMOTIF__) || defined(__WXPM__) || defined(__WXMGL__)
     #define IMPLEMENT_WXWIN_MAIN \
         extern int wxEntry( int argc, char **argv ); \
         int main(int argc, char **argv) { return wxEntry(argc, argv); }
+#elif defined(__WXGTK__)
+    #define IMPLEMENT_WXWIN_MAIN \
+        extern int wxEntry( int argc, char **argv, bool enterLoop = TRUE ); \
+        int main(int argc, char **argv) { return wxEntry(argc, argv); }
 #elif defined(__WXMAC__) && defined(__UNIX__)
     // wxMac seems to have a specific wxEntry prototype
     #define IMPLEMENT_WXWIN_MAIN \
diff -ru wxGTK-2.4.0/include/wx/gtk/app.h wxGTK-2.4.0-debug/include/wx/gtk/app.h
--- wxGTK-2.4.0/include/wx/gtk/app.h	2003-01-05 12:55:08.000000000 +0100
+++ wxGTK-2.4.0-debug/include/wx/gtk/app.h	2003-02-28 16:37:58.000000000 +0100
@@ -93,6 +93,6 @@
     DECLARE_EVENT_TABLE()
 };
 
-int WXDLLEXPORT wxEntry( int argc, char *argv[] );
+int WXDLLEXPORT wxEntry( int argc, char *argv[], bool enterLoop );
 
 #endif // __GTKAPPH__
diff -ru wxGTK-2.4.0/src/gtk/app.cpp wxGTK-2.4.0-debug/src/gtk/app.cpp
--- wxGTK-2.4.0/src/gtk/app.cpp	2003-01-05 12:55:09.000000000 +0100
+++ wxGTK-2.4.0-debug/src/gtk/app.cpp	2003-03-10 19:14:01.000000000 +0100
@@ -850,7 +850,7 @@
 }
 
 
-int wxEntry( int argc, char *argv[] )
+int wxEntry( int argc, char *argv[], bool enterLoop )
 {
 #if (defined(__WXDEBUG__) && wxUSE_MEMORY_TRACING) || wxUSE_DEBUG_CONTEXT
     // This seems to be necessary since there are 'rogue'
@@ -918,8 +918,11 @@
         // When is the app not initialized ?
         wxTheApp->m_initialized = TRUE;
 
-        if (wxTheApp->Initialized())
-        {
+	if (!enterLoop)
+	{
+	    return 1;
+	} else
+	{
             wxTheApp->OnRun();
 
             wxWindow *topWindow = wxTheApp->GetTopWindow();

 	  	 
