diff --git a/samples/minimal/minimal.cpp b/samples/minimal/minimal.cpp
index f765a70..08d542c 100644
--- a/samples/minimal/minimal.cpp
+++ b/samples/minimal/minimal.cpp
@@ -29,6 +29,8 @@
     #include "wx/wx.h"
 #endif
 
+#include "wx/rearrangectrl.h"
+
 // ----------------------------------------------------------------------------
 // resources
 // ----------------------------------------------------------------------------
@@ -66,8 +68,14 @@ public:
     // event handlers (these functions should _not_ be virtual)
     void OnQuit(wxCommandEvent& event);
     void OnAbout(wxCommandEvent& event);
+    
+    void OnAddItem(wxCommandEvent& event);
+    void OnDownButton(wxCommandEvent& event);
+    void OnUpButton(wxCommandEvent& event);    
 
 private:
+    wxRearrangeList *m_list;
+
     // any class wishing to process wxWidgets events must use this macro
     wxDECLARE_EVENT_TABLE();
 };
@@ -85,7 +93,10 @@ enum
     // it is important for the id corresponding to the "About" command to have
     // this standard value as otherwise it won't be handled properly under Mac
     // (where it is special and put into the "Apple" menu)
-    Minimal_About = wxID_ABOUT
+    Minimal_About = wxID_ABOUT,
+    ID_UP = wxID_HIGHEST+1,
+    ID_DOWN,
+    ID_ADD
 };
 
 // ----------------------------------------------------------------------------
@@ -98,6 +109,9 @@ enum
 wxBEGIN_EVENT_TABLE(MyFrame, wxFrame)
     EVT_MENU(Minimal_Quit,  MyFrame::OnQuit)
     EVT_MENU(Minimal_About, MyFrame::OnAbout)
+    EVT_BUTTON(ID_ADD, MyFrame::OnAddItem )
+    EVT_BUTTON(ID_UP, MyFrame::OnUpButton )
+    EVT_BUTTON(ID_DOWN, MyFrame::OnDownButton )    
 wxEND_EVENT_TABLE()
 
 // Create a new application object: this macro will allow wxWidgets to create
@@ -177,10 +191,46 @@ MyFrame::MyFrame(const wxString& title)
     CreateStatusBar(2);
     SetStatusText("Welcome to wxWidgets!");
 #endif // wxUSE_STATUSBAR
+    wxPanel *panel = new wxPanel(this, wxID_ANY);
+    wxBoxSizer *main_sizer = new wxBoxSizer(wxVERTICAL);
+
+    main_sizer->Add(new wxButton(panel, ID_ADD, "ADD"), 0, wxALL, 4 );
+    main_sizer->Add(new wxButton(panel, ID_UP, "UP"), 0, wxALL, 4 );
+    main_sizer->Add(new wxButton(panel, ID_DOWN, "DOWN"), 0, wxALL, 4 );
+
+    wxArrayInt order;
+    order.Add(1);
+    order.Add(0);
+
+    wxArrayString items;
+    items.Add("1");
+    items.Add("0");
+
+    m_list = new wxRearrangeList(panel, wxID_ANY, wxDefaultPosition, wxDefaultSize, order, items );
+    main_sizer->Add(m_list, 1, wxALL | wxEXPAND, 4);
+
+    panel->SetSizer( main_sizer );
 }
 
 
 // event handlers
+void MyFrame::OnAddItem(wxCommandEvent& event)
+{
+  wxString item;
+  item << (m_list->GetCount() + 1);
+
+  m_list->Append(item);
+}
+
+void MyFrame::OnDownButton(wxCommandEvent& event)
+{
+  m_list->MoveCurrentDown();
+}
+
+void MyFrame::OnUpButton(wxCommandEvent& event)
+{
+  m_list->MoveCurrentUp();
+}
 
 void MyFrame::OnQuit(wxCommandEvent& WXUNUSED(event))
 {
-- 
2.6.2.windows.1

