diff --git a/samples/stc/stctest.cpp b/samples/stc/stctest.cpp
index f34b476..1fbc58b 100644
--- a/samples/stc/stctest.cpp
+++ b/samples/stc/stctest.cpp
@@ -301,6 +301,7 @@ AppFrame::AppFrame (const wxString &title)
     m_edit->SetFocus();
 
     FileOpen (wxT("stctest.cpp"));
+    //wxLog::AddTraceMask(wxT("clipboard"));
 }
 
 AppFrame::~AppFrame () {
@@ -318,7 +319,16 @@ void AppFrame::OnClose (wxCloseEvent &event) {
 }
 
 void AppFrame::OnAbout (wxCommandEvent &WXUNUSED(event)) {
-    AppAbout dlg(this);
+    static unsigned int count = 1;
+    for (size_t n=0; n < 200; ++n, ++count)
+    {
+        wxDateTime starttime = wxDateTime::UNow();
+        m_edit->SelectAll(); 
+        m_edit->SelectNone();
+        wxSafeYield();
+        wxTimeSpan diff = wxDateTime::UNow() - starttime;
+        wxLogDebug("%u: %s", count, diff.GetMilliseconds().ToString());
+    }
 }
 
 void AppFrame::OnExit (wxCommandEvent &WXUNUSED(event)) {
