--- wxWindows/src/gtk/joystick.cpp.orig	Tue Aug 20 14:57:12 2002
+++ wxWindows/src/gtk/joystick.cpp	Tue Aug 20 13:59:52 2002
@@ -140,10 +140,10 @@
 int wxJoystick::GetPOVPosition(void) const
 {
-  return 0;
+  return -1;
 }
 
 int wxJoystick::GetPOVCTSPosition(void) const
 {
-  return 0;
+  return -1;
 }
 
@@ -208,5 +208,9 @@
 wxString wxJoystick::GetProductName(void) const
 {
-  return wxT("");
+  wxString dev_name;
+  // 2002-08-20 johan@linkdata.se
+  // Return the device name in lieu of a better one
+  dev_name.Printf( wxT("/dev/js%d"), (m_joystick == wxJOYSTICK1) ? 0 : 1);  // FIXME Unicode?
+  return dev_name;
 }
 

 	  	 
