--- /Users/cozmotion/Documents/wxWidgets-2.9.4/samples/wizard/wizard.cpp	2012-07-08 03:28:27.000000000 -0700
+++ wizard.cpp	2012-08-25 12:28:31.000000000 -0700
@@ -169,6 +169,22 @@
 // This is a more complicated example of validity checking: using events we may
 // allow to return to the previous page, but not to proceed. It also
 // demonstrates how to intercept [Cancel] button press.
+
+// ***COZ*** start
+#include <wx/dcclient.h>
+#include <wx/tipwin.h>
+class Buoy  : public wxObject  {
+	public:
+		wxPoint pos;		//device units
+		wxRegion region;	//device units
+		wxString name;
+};
+#define ID_PANEL 10004
+WX_DECLARE_OBJARRAY(Buoy,BuoyArray);
+#include <wx/arrimpl.cpp> // This is a magic incantation which must be done!
+WX_DEFINE_OBJARRAY(BuoyArray);
+// ***COZ*** end
+
 class wxRadioboxPage : public wxWizardPageSimple
 {
 public:
@@ -178,6 +194,7 @@
         Forward, Backward, Both, Neither
     };
 
+
     wxRadioboxPage(wxWizard *parent) : wxWizardPageSimple(parent)
     {
         // should correspond to the enum above
@@ -194,8 +211,38 @@
                                  WXSIZEOF(choices), choices,
                                  1, wxRA_SPECIFY_COLS);
         m_radio->SetSelection(Both);
+		wxBoxSizer *mainSizer = new wxBoxSizer(wxVERTICAL);
+
+		///////////////////////// ***COZ*** start
+		m_Panel = new wxPanel( this, wxID_ANY, wxDefaultPosition, wxSize(200, 200), wxNO_BORDER|wxTAB_TRAVERSAL );
+		m_Panel->SetBackgroundColour(wxColour(255, 255, 255));
+
+		int x,y;
+		m_BuoyArray.Alloc(200);
+		for (y=0;y<10;y++) {
+			for (x=0;x<10;x++) {
+				wxPoint rc_pt;
+				rc_pt.x = (x * 11) + 10;
+				rc_pt.y = (y * 11) + 10;
+				wxVector<wxPoint> points;
+				points.push_back( wxPoint(rc_pt.x-4,rc_pt.y) );
+				points.push_back( wxPoint(rc_pt.x,rc_pt.y-4) );
+				points.push_back( wxPoint(rc_pt.x+4,rc_pt.y) );
+				points.push_back( wxPoint(rc_pt.x,rc_pt.y+4) );
+				Buoy b;
+				b.pos = rc_pt;
+				b.region = wxRegion(points.size(), &points[0],wxODDEVEN_RULE);
+				b.name = "Buoy " + wxString::Format(wxT("%i"),x+1) + "," + wxString::Format(wxT("%i"),y+1);
+				m_BuoyArray.Add(b);
+			}
+		}
+		m_p_InfoPopWin = NULL;
+		// Connect events and objects
+		m_Panel->Connect(wxID_ANY, wxEVT_PAINT, wxPaintEventHandler(wxRadioboxPage::OnPaint), NULL, this);
+		m_Panel->Connect(wxID_ANY, wxEVT_MOTION, wxMouseEventHandler(wxRadioboxPage::OnMotion), NULL, this);
+		mainSizer->Add(m_Panel, 0, wxALIGN_LEFT, 0);
+		// ***COZ*** end
 
-        wxBoxSizer *mainSizer = new wxBoxSizer(wxVERTICAL);
         mainSizer->Add(
             m_radio,
             0, // No stretching
@@ -235,10 +282,54 @@
 
         event.Veto();
     }
-
+//////////////// ***COZ*** start
+	void OnPaint( wxPaintEvent& event )
+	{
+		//wxPaintDC dc(wxDynamicCast(event.GetEventObject(), wxWindow));
+		wxPaintDC dc(m_Panel);
+		dc.SetUserScale(1,1);
+		int idx;
+
+		for (size_t i = 0; i < m_BuoyArray.GetCount() ; i++) {
+			idx = i;
+			dc.SetBrush(wxColor(255,255,0));
+			dc.SetPen( wxPen( wxColor(0,0,0), 1 ) );
+
+			wxVector<wxPoint> points;
+			points.push_back( wxPoint(m_BuoyArray.Item(idx).pos.x-4,m_BuoyArray.Item(idx).pos.y) );
+			points.push_back( wxPoint(m_BuoyArray.Item(idx).pos.x,m_BuoyArray.Item(idx).pos.y-4) );
+			points.push_back( wxPoint(m_BuoyArray.Item(idx).pos.x+4,m_BuoyArray.Item(idx).pos.y) );
+			points.push_back( wxPoint(m_BuoyArray.Item(idx).pos.x,m_BuoyArray.Item(idx).pos.y+4) );
+			dc.DrawPolygon( points.size(), &points[0], 0,0);
+		}
+	}
+
+	void OnMotion( wxMouseEvent& event )
+	{
+		wxPoint position = event.GetPosition();
+		size_t count = m_BuoyArray.GetCount();
+		for (size_t i = 0; i < count; i++) {
+			if (m_BuoyArray.Item(i).region.Contains(position) == wxInRegion) {
+				if (m_p_InfoPopWin == NULL) {
+					// raise the tip window
+					Buoy &buoy = m_BuoyArray.Item(i);
+					wxRect rc = buoy.region.GetBox();
+					ClientToScreen(&rc.x,&rc.y);
+					m_p_InfoPopWin = new wxTipWindow(
+						m_Panel,
+						wxString::Format("*****************\n%s\n=====================",buoy.name),
+						200,&m_p_InfoPopWin,&rc );
+				}
+			}
+		}
+	}
+private:
+	wxPanel *m_Panel;
+	BuoyArray m_BuoyArray;
+	wxTipWindow* m_p_InfoPopWin;
+//////////////////////////////////////////////// ***COZ*** end
 private:
     wxRadioBox *m_radio;
-
     DECLARE_EVENT_TABLE()
 };
 
@@ -450,7 +541,7 @@
     menuOptions->AppendCheckItem(Wizard_ExpandBitmap, wxT("Si&ze Bitmap To Page"));
 
     wxMenu *helpMenu = new wxMenu;
-    helpMenu->Append(Wizard_About, wxT("&About\tF1"), wxT("Show about dialog"));
+    helpMenu->Append(Wizard_About, wxT("&About...\tF1"), wxT("Show about dialog"));
 
     // now append the freshly created menu to the menu bar...
     wxMenuBar *menuBar = new wxMenuBar();
