diff -rNbu2 wxW/build/bakefiles/files.bkl wxWidgets/build/bakefiles/files.bkl
--- wxW/build/bakefiles/files.bkl	Sat Oct  2 19:34:27 2004
+++ wxWidgets/build/bakefiles/files.bkl	Sun Oct  3 03:43:23 2004
@@ -1884,4 +1884,5 @@
     src/cocoa/choice.mm
     src/cocoa/clipbrd.mm
+    src/cocoa/colordlg.mm
     src/cocoa/colour.mm
     src/cocoa/combobox.mm
@@ -1931,5 +1932,4 @@
     src/generic/accel.cpp
     src/generic/caret.cpp
-    src/generic/colrdlgg.cpp
     src/generic/dirdlgg.cpp
     src/generic/fdrepdlg.cpp
@@ -1972,4 +1972,5 @@
     wx/cocoa/choice.h
     wx/cocoa/clipbrd.h
+    wx/cocoa/colordlg.h
     wx/cocoa/colour.h
     wx/cocoa/combobox.h
diff -rNbu2 wxW/include/wx/cocoa/colordlg.h wxWidgets/include/wx/cocoa/colordlg.h
--- wxW/include/wx/cocoa/colordlg.h	Wed Dec 31 16:00:00 1969
+++ wxWidgets/include/wx/cocoa/colordlg.h	Sun Oct  3 00:05:16 2004
@@ -0,0 +1,50 @@
+/////////////////////////////////////////////////////////////////////////////
+// Name:        colordlg.h
+// Purpose:     wxColourDialog class for wxCocoa.
+// Author:      Ryan Norton
+// Modified by:
+// Created:     2004-10-02
+// RCS-ID:      $Id: colordlg.h,v 1.4 2004/05/23 20:51:02 JS Exp $
+// Copyright:   (c) Ryan Norton
+// Licence:     wxWindows licence
+/////////////////////////////////////////////////////////////////////////////
+
+#ifndef _WX_COCOA_COLORDLG_H_
+#define _WX_COCOA_COLORDLG_H_
+
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
+#pragma interface "colordlg.h"
+#endif
+
+#include "wx/setup.h"
+#include "wx/dialog.h"
+#include "wx/cmndata.h"
+
+class WXDLLEXPORT wxColourDialog: public wxDialog
+{
+DECLARE_DYNAMIC_CLASS(wxColourDialog)
+public:
+    wxColourDialog();
+    wxColourDialog(wxWindow *parent, wxColourData *data = NULL);
+
+    bool Create(wxWindow *parent, wxColourData *data = NULL);
+
+    int ShowModal();
+    wxColourData& GetColourData() { return m_colourData; }
+
+    inline WX_NSColorPanel GetNSColorPanel() 
+    {	return (WX_NSColorPanel) GetNSPanel();	}
+    
+    bool Create(wxWindow *parent, wxWindowID winid,
+            const wxString& title,
+            const wxPoint& pos = wxDefaultPosition,
+            const wxSize& size = wxDefaultSize,
+            long style = wxDEFAULT_DIALOG_STYLE,
+            const wxString& name = wxDialogNameStr);
+                
+protected:
+    wxColourData        m_colourData;
+};
+
+#endif
+    // _WX_COLORDLG_H_
diff -rNbu2 wxW/include/wx/defs.h wxWidgets/include/wx/defs.h
--- wxW/include/wx/defs.h	Sun Oct  3 03:25:07 2004
+++ wxWidgets/include/wx/defs.h	Sun Oct  3 03:37:31 2004
@@ -2174,4 +2174,5 @@
 DECLARE_WXCOCOA_OBJC_CLASS(NSButton);
 DECLARE_WXCOCOA_OBJC_CLASS(NSColor);
+DECLARE_WXCOCOA_OBJC_CLASS(NSColorPanel);
 DECLARE_WXCOCOA_OBJC_CLASS(NSControl);
 DECLARE_WXCOCOA_OBJC_CLASS(NSEvent);
diff -rNbu2 wxW/include/wx/colordlg.h wxWidgets/include/wx/colordlg.h
--- wxW/include/wx/colordlg.h	Sat Oct  2 19:34:24 2004
+++ wxWidgets/include/wx/colordlg.h	Sat Oct  2 22:47:36 2004
@@ -23,4 +23,6 @@
 #elif defined(__WXGTK20__) && !defined(__WXUNIVERSAL__)
     #include "wx/gtk/colordlg.h"
+#elif defined(__WXCOCOA__) && !defined(__WXUNIVERSAL__)
+    #include "wx/cocoa/colordlg.h"
 #else
     #include "wx/generic/colrdlgg.h"
diff -rNbu2 wxW/src/cocoa/colordlg.mm wxWidgets/src/cocoa/colordlg.mm
--- wxW/src/cocoa/colordlg.mm	Wed Dec 31 16:00:00 1969
+++ wxWidgets/src/cocoa/colordlg.mm	Sun Oct  3 02:26:08 2004
@@ -0,0 +1,112 @@
+/////////////////////////////////////////////////////////////////////////////
+// Name:        src/cocoa/colordlg.mm
+// Purpose:     wxColourDialog class. NOTE: you can use the generic class
+//              if you wish, instead of using this.
+// Author:      Ryan Norton
+// Modified by:
+// Created:     2004-10-02
+// RCS-ID:      $Id: colordlg.cpp,v 1.11 2004/05/23 20:52:30 JS Exp $
+// Copyright:   (c) Ryan Norton
+// Licence:       wxWindows licence
+/////////////////////////////////////////////////////////////////////////////
+
+// For compilers that support precompilation, includes "wx.h".
+#include "wx/wxprec.h"
+
+#ifndef WX_PRECOMP
+    #include "wx/app.h"
+    #include "wx/colordlg.h"
+    #include "wx/log.h"
+#endif
+
+#include "wx/cocoa/autorelease.h"
+#include "wx/cocoa/string.h"
+
+#import <AppKit/AppKit.h>
+
+#if !USE_SHARED_LIBRARY
+IMPLEMENT_DYNAMIC_CLASS(wxColourDialog, wxDialog)
+#endif
+
+wxColourDialog::wxColourDialog()
+{
+}
+
+wxColourDialog::wxColourDialog(wxWindow *parent, wxColourData *data)
+{
+    Create(parent, data);
+}
+
+bool wxColourDialog::Create(wxWindow *parent, wxColourData *data)
+{
+    bool bOK = Create( parent, wxID_ANY, wxT(""), wxDefaultPosition, wxDefaultSize,
+                      wxDEFAULT_DIALOG_STYLE | wxRESIZE_BORDER );
+                      
+    if(data)
+        m_colourData = *data;
+        
+    return bOK;
+}
+
+bool wxColourDialog::Create(wxWindow *parent, wxWindowID winid,
+           const wxString& title,
+           const wxPoint& pos,
+           const wxSize& size,
+           long style,
+           const wxString& name)
+{
+    wxAutoNSAutoreleasePool pool;
+    wxTopLevelWindows.Append(this);
+
+    if(!CreateBase(parent,winid,pos,size,style,wxDefaultValidator,name))
+        return false;
+
+    if (parent)
+        parent->AddChild(this);
+
+    m_cocoaNSWindow = NULL;
+    
+    SetNSPanel([NSColorPanel sharedColorPanel]);
+    NSColorPanel* thePanel = GetNSPanel();
+    
+    //set the initial color the NSColorPanel
+    [thePanel setColor:m_colourData.m_dataColour.GetNSColor()];
+    
+    //allow only RGB colors
+    [thePanel setMode:NSRGBModeColorPanel];
+    
+    // NOTE: Normally, in a wxCocoa wxDialog we'd release here to get the retain count
+    // down to 1, but since the color panel is shared we should leave it 
+    // alloc'd for the rest of the app
+    wxLogTrace(wxTRACE_COCOA_RetainRelease,wxT("wxColourDialog m_cocoaNSWindow retainCount=%d"),[m_cocoaNSWindow retainCount]);
+    [m_cocoaNSWindow setTitle:wxNSStringWithWxString(title)];
+    [m_cocoaNSWindow setHidesOnDeactivate:NO];
+
+    return true;
+}
+
+int wxColourDialog::ShowModal()
+{
+    wxDialog::ShowModal();
+    
+    NSColorPanel* thePanel = GetNSPanel();
+    
+    //POSSIBLE TODO:  We could use a delegate here for the changeColor
+    //callback - but since its called every time the color changes anyway (?)
+    //it seems like unneccesary duplication (?)
+    NSColor* theColor = [[thePanel color] colorUsingColorSpaceName:NSCalibratedRGBColorSpace];
+                                    
+    m_colourData.m_dataColour.Set((unsigned char) ([theColor redComponent] * 255.0),
+                                  (unsigned char) ([theColor greenComponent] * 255.0),
+                                  (unsigned char) ([theColor blueComponent] * 255.0)) ;
+
+#ifdef __WXDEBUG__
+    wxPrintf(wxT("---Color Panel---\nColor:%i,%i,%i\n---END Color Panel---\n"),
+                    m_colourData.m_dataColour.Red(), 
+                    m_colourData.m_dataColour.Green(), 
+                    m_colourData.m_dataColour.Blue() );
+#endif
+
+    return wxID_OK;
+}
+

 	  	 
