Index: src/common/appbase.cpp
===================================================================
--- src/common/appbase.cpp	(revision 61711)
+++ src/common/appbase.cpp	(working copy)
@@ -340,6 +340,11 @@
     event.SetEventObject(this);
     ProcessEvent(event);
 
+#if wxUSE_LOG
+    // flush the logged messages if any
+    wxLog::FlushActive();
+#endif
+
     return event.MoreRequested();
 }
 
Index: src/common/appcmn.cpp
===================================================================
--- src/common/appcmn.cpp	(revision 61711)
+++ src/common/appcmn.cpp	(working copy)
@@ -355,11 +355,6 @@
         node = node->GetNext();
     }
 
-#if wxUSE_LOG
-    // flush the logged messages if any
-    wxLog::FlushActive();
-#endif
-
     wxUpdateUIEvent::ResetUpdateTime();
 
     return needMore;
