Index: interface/wx/window.h
===================================================================
--- interface/wx/window.h	(revision 72625)
+++ interface/wx/window.h	(working copy)
@@ -385,6 +385,8 @@
         Returns @true if the window (or in case of composite controls, its main
         child window) has focus.
 
+        @since 2.9.0
+
         @see FindFocus()
     */
     virtual bool HasFocus() const;
