--- src/aui/framemanager.cpp.orig	2007-05-03 13:22:53.000000000 -0700
+++ src/aui/framemanager.cpp	2007-05-03 13:27:13.000000000 -0700
@@ -1119,7 +1119,7 @@
     for (i = 0, pane_count = m_panes.GetCount(); i < pane_count; ++i)
     {
         wxAuiPaneInfo& p = m_panes.Item(i);
-        if (!p.IsToolbar())
+        if (!p.IsToolbar() && !p.IsFloating())
         {
             p.Restore();
             

 	  	 
