--- settings.cpp	2005-09-23 09:03:59.000000000 -0700
+++ /Users/davidsurovell/ChandRoot/internal/wx/src/mac/carbon/settings.cpp	2005-11-09 20:06:26.000000000 -0800
@@ -67,23 +67,31 @@
         case wxSYS_COLOUR_CAPTIONTEXT:
         case wxSYS_COLOUR_INFOTEXT:
         case wxSYS_COLOUR_INACTIVECAPTIONTEXT:
             return *wxBLACK;
             break ;
         case wxSYS_COLOUR_HIGHLIGHT:
             {
                 RGBColor hilite ;
-                GetThemeBrushAsColor( kThemeBrushPrimaryHighlightColor, 32, true, &hilite );
+                ThemeBrush colorBrushID;
+
+#if MAC_OS_X_VERSION_MAX_ALLOWED > MAC_OS_X_VERSION_10_2
+                colorBrushID = kThemeBrushAlternatePrimaryHighlightColor;
+#else
+                colorBrushID = kThemeBrushPrimaryHighlightColor;
+#endif
+
+                GetThemeBrushAsColor( colorBrushID, 32, true, &hilite );
                 return wxColor( hilite.red >> 8 , hilite.green >> 8  , hilite.blue >> 8  ) ;
             }
             break ;
         case wxSYS_COLOUR_BTNHIGHLIGHT:
         case wxSYS_COLOUR_GRAYTEXT:
             return wxColor( 0xCC , 0xCC , 0xCC ) ;
             break ;
 
         case wxSYS_COLOUR_3DDKSHADOW:
             return wxColor( 0x44 , 0x44 , 0x44 ) ;
             break ;
         case wxSYS_COLOUR_3DLIGHT:
             return wxColor( 0xCC , 0xCC , 0xCC ) ;
             break ;

 	  	 
