 samples/minimal/minimal.cpp | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/samples/minimal/minimal.cpp b/samples/minimal/minimal.cpp
index 470e765423..4b7d86bfe6 100644
--- a/samples/minimal/minimal.cpp
+++ b/samples/minimal/minimal.cpp
@@ -170,6 +170,16 @@ MyFrame::MyFrame(const wxString& title)
     SetSizer(sizer);
 #endif // wxUSE_MENUBAR/!wxUSE_MENUBAR
 
+    wxToolBar* toolBar =
+#if 1
+    new wxToolBar(this, wxID_ANY);
+#else
+    CreateToolBar();
+#endif
+    toolBar->AddTool(Minimal_About, "About", wxICON(sample));
+    toolBar->Realize();
+    SetToolBar(toolBar);
+
 #if wxUSE_STATUSBAR
     // create a status bar just for fun (by default with 1 pane only)
     CreateStatusBar(2);
