Index: src/gtk/window.cpp
===================================================================
--- src/gtk/window.cpp	(revision 49558)
+++ src/gtk/window.cpp	(working copy)
@@ -3979,7 +3979,7 @@
 
 void wxWindowGTK::DoApplyWidgetStyle(GtkRcStyle *style)
 {
-    wxSuspendStyleEvents s(this);
+    wxSuspendStyleEvents s((wxWindow *)this);
 
     if (m_wxwindow)
         gtk_widget_modify_style(m_wxwindow, style);
Index: src/gtk/toplevel.cpp
===================================================================
--- src/gtk/toplevel.cpp	(revision 49558)
+++ src/gtk/toplevel.cpp	(working copy)
@@ -49,7 +49,10 @@
 // data
 // ----------------------------------------------------------------------------
 
-extern int              g_openDialogs;
+// Don't allow window closing if there are open dialogs
+// it can't be put into gtk/dialog.cpp, that is not linked in universal build
+int g_openDialogs = 0;
+
 extern wxWindowGTK     *g_delayedFocus;
 
 // the frame that is currently active (i.e. its child has focus). It is
Index: src/gtk/dialog.cpp
===================================================================
--- src/gtk/dialog.cpp	(revision 49558)
+++ src/gtk/dialog.cpp	(working copy)
@@ -21,11 +21,10 @@
 #include <gtk/gtk.h>
 
 //-----------------------------------------------------------------------------
-// global data
+// data
 //-----------------------------------------------------------------------------
 
-// Don't allow window closing if there are open dialogs
-int g_openDialogs;
+extern int g_openDialogs;
 
 //-----------------------------------------------------------------------------
 // wxDialog
Index: include/wx/hyperlink.h
===================================================================
--- include/wx/hyperlink.h	(revision 49558)
+++ include/wx/hyperlink.h	(working copy)
@@ -143,7 +143,7 @@
 
 
 
-#if defined(__WXGTK210__)
+#if defined(__WXGTK210__) && !defined(__WXUNIVERSAL__)
     #include "wx/gtk/hyperlink.h"
 #else
     #include "wx/generic/hyperlink.h"
Index: build/bakefiles/files.bkl
===================================================================
--- build/bakefiles/files.bkl	(revision 49558)
+++ build/bakefiles/files.bkl	(working copy)
@@ -2865,12 +2865,18 @@
     src/gtk/aboutdlg.cpp
     src/gtk/animate.cpp
     src/gtk/dataview.cpp
-    src/gtk/hyperlink.cpp
     src/gtk/taskbar.cpp
     src/gtk/eggtrayicon.c
 </set>
 <set var="ADVANCED_GTK_HDR" hints="files">
     wx/gtk/animate.h
+</set>
+
+<!-- not used with wxUniv -->
+<set var="ADVANCED_GTK_NATIVE_SRC" hints="files">
+    src/gtk/hyperlink.cpp
+</set>
+<set var="ADVANCED_GTK_NATIVE_HDR" hints="files">
     wx/gtk/hyperlink.h
 </set>
 
@@ -3454,10 +3460,12 @@
     <set var="ADVANCED_PLATFORM_NATIVE_SRC" hints="files">
         <if cond="TOOLKIT=='MSW'">$(ADVANCED_MSW_NATIVE_SRC)</if>
         <if cond="TOOLKIT=='WINCE'">$(ADVANCED_MSW_NATIVE_SRC)</if>
+        <if cond="TOOLKIT=='GTK' and TOOLKIT_VERSION=='2'">$(ADVANCED_GTK_NATIVE_SRC)</if>
     </set>
     <set var="ADVANCED_PLATFORM_NATIVE_HDR" hints="files">
         <if cond="TOOLKIT=='MSW'">$(ADVANCED_MSW_NATIVE_HDR)</if>
         <if cond="TOOLKIT=='WINCE'">$(ADVANCED_MSW_NATIVE_HDR)</if>
+        <if cond="TOOLKIT=='GTK' and TOOLKIT_VERSION=='2'">$(ADVANCED_GTK_NATIVE_HDR)</if>
     </set>
 
     <set var="ADVANCED_SRC" hints="files">

 	  	 
