--- c:\helpbest.cpp.bak	Sun Aug 29 12:17:04 2004
+++ c:\helpbest.cpp	Wed May 10 09:50:36 2006
@@ -46,5 +46,6 @@
     wxLogNull dontWarnOnFailure;
 
-    if( chm->Initialize( GetValidFilename( filename ) ) )
+    wxString validName = GetValidFilename( filename );
+    if(::wxFileExists(validName) && chm->Initialize( validName ) )
     {
         m_helpController = chm;
@@ -59,5 +60,6 @@
 
     m_helpControllerType = wxUseHtmlHelp;
-    if( html->Initialize( GetValidFilename( filename ) ) )
+    validName = GetValidFilename( filename );
+    if(::wxFileExists(validName) &&  html->Initialize( GetValidFilename( filename ) ) )
     {
         m_helpController = html;

 	  	 
