Index: src/common/utilscmn.cpp
===================================================================
--- src/common/utilscmn.cpp	(revision 54331)
+++ src/common/utilscmn.cpp	(working copy)
@@ -1349,7 +1349,7 @@
 
     if ( ( style & ( wxICON_EXCLAMATION | wxICON_HAND | wxICON_INFORMATION | wxICON_QUESTION ) ) == 0 )
     {
-        decorated_style |= ( style & wxYES ) ? wxICON_QUESTION : wxICON_INFORMATION ;
+        decorated_style |= ( style & wxYES_NO ) ? wxICON_QUESTION : wxICON_INFORMATION ;
     }
 
     wxMessageDialog dialog(parent, message, caption, decorated_style);
