Index: src/osx/cocoa/nonownedwnd.mm
===================================================================
--- src/osx/cocoa/nonownedwnd.mm	(revision 60892)
+++ src/osx/cocoa/nonownedwnd.mm	(working copy)
@@ -455,7 +455,11 @@
 {
     if ( show )
     {
-        [m_macWindow makeKeyAndOrderFront:nil];
+        wxNonOwnedWindow* wxpeer = GetWXPeer();
+        if (wxpeer && !(wxpeer->GetWindowStyle() & wxFRAME_TOOL_WINDOW))
+            [m_macWindow makeKeyAndOrderFront:nil];
+        else
+            [m_macWindow orderFront:nil];
         [[m_macWindow contentView] setNeedsDisplay:YES];
     }
     else 
