Index: interface/wx/msgdlg.h
===================================================================
--- interface/wx/msgdlg.h	(revision 64087)
+++ interface/wx/msgdlg.h	(working copy)
@@ -227,7 +227,7 @@
     @param caption
         The dialog title.
     @param parent
-        Parent window.
+        Parent window.  If this is not supplied, message boxes can become hidden under their parents.
     @param style
         Combination of style flags described in wxMessageDialog documentation.
     @param x
