Index: src/common/menucmn.cpp
===================================================================
--- src/common/menucmn.cpp	(revision 76177)
+++ src/common/menucmn.cpp	(working copy)
@@ -600,14 +600,16 @@
 // window will be used.
 void wxMenuBase::UpdateUI(wxEvtHandler* source)
 {
+    bool processed;
     wxWindow * const win = GetWindow();
 
     if ( !source && win )
         source = win->GetEventHandler();
+    
+    wxEvtHandler *this_handler = GetEventHandler(); 
+
     if ( !source )
-        source = GetEventHandler();
-    if ( !source )
-        source = this;
+        source = this_handler;
 
     wxMenuItemList::compatibility_iterator  node = GetMenuItems().GetFirst();
     while ( node )
@@ -617,10 +619,18 @@
         {
             wxWindowID itemid = item->GetId();
             wxUpdateUIEvent event(itemid);
-            event.SetEventObject( this );
+            event.SetEventObject( source );
 
-            if ( source->ProcessEvent(event) )
+            processed = source->ProcessEvent(event);
+
+            if( !processed && source!=self )
             {
+                event.SetEventObject( self );
+                processed = self->ProcessEvent(event);
+            }
+
+            if ( processed )
+            {
                 // if anything changed, update the changed attribute
                 if (event.GetSetText())
                     SetLabel(itemid, event.GetText());
