Index: demo/pyTree.py
===================================================================
--- demo/pyTree.py	(revision 71603)
+++ demo/pyTree.py	(working copy)
@@ -114,7 +114,7 @@
         wxTreeItemData referencing the child object. We get this
         object using child = getattr(parent, item).
 
-        Finally, we check wether the child returns a non-empty dir()
+        Finally, we check whether the child returns a non-empty dir()
         list. If so, it is labeled as 'having children', so that it
         may be expanded. When it actually is expanded, this function
         will again figure out what the offspring is.
Index: wx/lib/masked/maskededit.py
===================================================================
--- wx/lib/masked/maskededit.py	(revision 71603)
+++ wx/lib/masked/maskededit.py	(working copy)
@@ -58,7 +58,7 @@
 ====================
 
 masked.TextCtrl:
-    is a sublassed text control that can carefully control the user's input
+    is a subclassed text control that can carefully control the user's input
     based on a mask string you provide.
 
     General usage example::
@@ -229,7 +229,7 @@
             0  integer fields get leading zeros
             D  Date[/time] field
             T  Time field
-            F  Auto-Fit: the control calulates its size from
+            F  Auto-Fit: the control calculates its size from
                the length of the template mask
             V  validate entered chars against validRegex before allowing them
                to be entered vs. being allowed by basic mask and then having
@@ -271,7 +271,7 @@
 
 shiftDecimalChar
   The default "shiftDecimalChar" (used for "backwards-tabbing" until
-  shift-tab is fixed in wxPython) is '>' (for QUERTY keyboards.) for
+  shift-tab is fixed in wxPython) is '>' (for QWERTY keyboards.) for
   other keyboards, you may want to customize this, eg '?' for shift ',' on
   AZERTY keyboards, ':' or ';' for other European keyboards, etc.
   (This is a control-level parameter.)
Index: wx/lib/plot.py
===================================================================
--- wx/lib/plot.py	(revision 71603)
+++ wx/lib/plot.py	(working copy)
@@ -1920,7 +1920,7 @@
 ##        print "DC GetSize", dc.GetSize()
 ##        print "GetPageSizePixels", self.GetPageSizePixels()
         # Note PPIScreen does not give the correct number
-        # Calulate everything for printer and then scale for preview
+        # Calculate everything for printer and then scale for preview
         PPIPrinter= self.GetPPIPrinter()        # printer dots/inch (w,h)
         #PPIScreen= self.GetPPIScreen()          # screen dots/inch (w,h)
         dcSize= dc.GetSize()                    # DC size
@@ -1932,7 +1932,7 @@
             else:
                 if self.graph._hiResEnabled:
                     dc.SetMapMode(wx.MM_TWIPS) # high precision - each logical unit is 1/20 of a point
-        pageSize= self.GetPageSizePixels() # page size in terms of pixcels
+        pageSize= self.GetPageSizePixels() # page size in terms of pixels
         clientDcSize= self.graph.GetClientSize()
         
         # find what the margins are (mm)
