Index: samples/controls/controls.cpp
===================================================================
--- samples/controls/controls.cpp	(revision 66308)
+++ samples/controls/controls.cpp	(working copy)
@@ -655,7 +678,7 @@
     wxString choices[] =
     {
         wxT("This"),
-        wxT("is"),
+        wxT("s"),
         wxT("one of my long and"),
         wxT("wonderful"),
         wxT("examples.")
