Index: src/osx/cocoa/combobox.mm
===================================================================
--- src/osx/cocoa/combobox.mm	(revision 73495)
+++ src/osx/cocoa/combobox.mm	(working copy)
@@ -36,12 +36,6 @@
 @end
 
 
-@interface wxNSComboBox : NSComboBox
-{
-}
-
-@end
-
 @implementation wxNSComboBox
 
 + (void)initialize
@@ -54,6 +48,33 @@
     }
 }
 
+- (void) dealloc
+{
+    [fieldEditor release];
+    [super dealloc];
+}
+
+// Over-riding NSComboBox onKeyDown method doesn't work for key events.
+// Ensure that we can use our own wxNSTextFieldEditor to catch key events.
+// See windowWillReturnFieldEditor in nonownedwnd.mm.
+// Key events will be caught and handled via wxNSTextFieldEditor onkey...
+// methods in textctrl.mm.
+
+- (void) setFieldEditor:(wxNSTextFieldEditor*) editor
+{
+    if ( editor != fieldEditor )
+    {
+        [editor retain];
+        [fieldEditor release];
+        fieldEditor = editor;
+    }
+}
+
+- (wxNSTextFieldEditor*) fieldEditor
+{
+    return fieldEditor;
+}
+
 - (void)controlTextDidChange:(NSNotification *)aNotification
 {
     wxUnusedVar(aNotification);
Index: src/osx/cocoa/nonownedwnd.mm
===================================================================
--- src/osx/cocoa/nonownedwnd.mm	(revision 73495)
+++ src/osx/cocoa/nonownedwnd.mm	(working copy)
@@ -128,8 +128,8 @@
         {
             if (wxTheApp)
                 wxTheApp->MacSetCurrentEvent(event, NULL);
-            ((wxWidgetCocoaImpl*)cw->GetPeer())->DoHandleMouseEvent( event);
-            handled = true;
+             handled = ((wxWidgetCocoaImpl*)cw->GetPeer())->DoHandleMouseEvent( event);
+            //handled = true;
         }
         if ( handled )
         {
@@ -467,6 +467,9 @@
     }
 }
 
+// By default a window uses one NSTextFieldEditor for all its text display
+// controls. Assign our own wxNSTextFieldEditor for event purposes.
+
 - (id)windowWillReturnFieldEditor:(NSWindow *)sender toObject:(id)anObject
 {
     wxUnusedVar(sender);
@@ -484,6 +487,19 @@
         }
         return editor;
     } 
+    else if ([anObject isKindOfClass:[wxNSComboBox class]])
+    {
+        wxNSComboBox * cb = (wxNSComboBox*) anObject;
+        wxNSTextFieldEditor* editor = [cb fieldEditor];
+        if ( editor == nil )
+        {
+            editor = [[wxNSTextFieldEditor alloc] init];
+            [editor setFieldEditor:YES];
+            [cb setFieldEditor:editor];
+            [editor release];
+        }
+        return editor;
+    }
  
     return nil;
 }
Index: include/wx/osx/cocoa/private.h
===================================================================
--- include/wx/osx/cocoa/private.h	(revision 73495)
+++ include/wx/osx/cocoa/private.h	(working copy)
@@ -339,6 +339,16 @@
 
     @end
 
+    @interface wxNSComboBox : NSComboBox
+    {
+        wxNSTextFieldEditor* fieldEditor;
+    }
+
+    - (wxNSTextFieldEditor*) fieldEditor;
+    - (void) setFieldEditor:(wxNSTextFieldEditor*) fieldEditor;
+
+    @end
+
     @interface wxNSMenu : NSMenu
     {
        wxMenuImpl* impl;
