--- C:/framemanager.cpp	Mon Nov 11 17:10:10 2013
+++ C:/wxWidgets/src/aui/framemanager.cpp	Fri Nov 15 12:30:09 2013
@@ -2254,35 +2254,22 @@
 
 
         // determine the dock's minimum size
-        bool plus_border = false;
-        bool plus_caption = false;
         int dock_min_size = 0;
         for (j = 0; j < dock_pane_count; ++j)
         {
             wxAuiPaneInfo& pane = *dock.panes.Item(j);
             if (pane.min_size != wxDefaultSize)
             {
+				int paneSize = dock.IsHorizontal() ? pane.min_size.y : pane.min_size.x;
                 if (pane.HasBorder())
-                    plus_border = true;
+					paneSize += 2 * pane_borderSize;
                 if (pane.HasCaption())
-                    plus_caption = true;
-                if (dock.IsHorizontal())
-                {
-                    if (pane.min_size.y > dock_min_size)
-                        dock_min_size = pane.min_size.y;
-                }
-                else
-                {
-                    if (pane.min_size.x > dock_min_size)
-                        dock_min_size = pane.min_size.x;
-                }
+					paneSize += caption_size;
+
+				if (paneSize > dock_min_size)
+					dock_min_size = paneSize;
             }
         }
-
-        if (plus_border)
-            dock_min_size += (pane_borderSize*2);
-        if (plus_caption && dock.IsHorizontal())
-            dock_min_size += (caption_size);
 
         dock.min_size = dock_min_size;
 
