diff -ru -U 2 -xCVS -x'.#*' -x'*.orig' -x'*.rej' -x'*~' wxWidgets-2.6.2_original/src/mac/carbon/spinctrl.cpp wxWidgets-2.6.2/src/mac/carbon/spinctrl.cpp
--- wxWidgets-2.6.2_original/src/mac/carbon/spinctrl.cpp	2005-02-07 15:58:48.000000000 -0500
+++ wxWidgets-2.6.2/src/mac/carbon/spinctrl.cpp	2005-10-04 04:01:57.000000000 -0500
@@ -58,6 +58,7 @@
             // Send event that the text was manually changed
             wxCommandEvent event(wxEVT_COMMAND_TEXT_UPDATED, m_spin->GetId());
             event.SetEventObject(m_spin);
+            event.SetString(m_spin->GetText()->GetValue());
             event.SetInt(val);
 
             m_spin->GetEventHandler()->ProcessEvent(event);

 	  	 
