Index: include/wx/sizer.h
===================================================================
--- include/wx/sizer.h	(revision 48768)
+++ include/wx/sizer.h	(working copy)
@@ -462,6 +462,7 @@
                      wxObject* userData = NULL);
     wxSizerItem* Add( wxWindow *window, const wxSizerFlags& flags);
     wxSizerItem* Add( wxSizer *sizer, const wxSizerFlags& flags);
+    wxSizerItem* Add( int width, int height, const wxSizerFlags& flags);
     wxSizerItem* Add( wxSizerItem *item);
 
     wxSizerItem* AddSpacer(int size);
@@ -492,6 +493,10 @@
     wxSizerItem* Insert(size_t index,
                         wxSizer *sizer,
                         const wxSizerFlags& flags);
+    wxSizerItem* Insert(size_t index,
+                        int width,
+                        int height,
+                        const wxSizerFlags& flags);                        
     virtual wxSizerItem* Insert( size_t index, wxSizerItem *item);
 
     wxSizerItem* InsertSpacer(size_t index, int size);
@@ -515,6 +520,7 @@
                          wxObject* userData = NULL);
     wxSizerItem* Prepend(wxWindow *window, const wxSizerFlags& flags);
     wxSizerItem* Prepend(wxSizer *sizer, const wxSizerFlags& flags);
+    wxSizerItem* Prepend(int width, int height, const wxSizerFlags& flags);
     wxSizerItem* Prepend(wxSizerItem *item);
 
     wxSizerItem* PrependSpacer(int size);
@@ -1007,6 +1013,12 @@
 }
 
 inline wxSizerItem*
+wxSizer::Add( int width, int height, const wxSizerFlags& flags )
+{
+    return Add( new wxSizerItem(width, height, flags) );
+}
+
+inline wxSizerItem*
 wxSizer::AddSpacer(int size)
 {
     return Add(size, size);
@@ -1067,6 +1079,12 @@
 }
 
 inline wxSizerItem*
+wxSizer::Prepend( int width, int height, const wxSizerFlags& flags )
+{
+    return Prepend( new wxSizerItem(width, height, flags) );
+}
+
+inline wxSizerItem*
 wxSizer::Insert( size_t index,
                  wxWindow *window,
                  int proportion,
@@ -1113,6 +1131,12 @@
 }
 
 inline wxSizerItem*
+wxSizer::Insert( size_t index, int width, int height, const wxSizerFlags& flags )
+{
+    return Insert( index, new wxSizerItem(width, height, flags) );
+}
+
+inline wxSizerItem*
 wxSizer::InsertSpacer(size_t index, int size)
 {
     return Insert(index, size, size);

 	  	 
