Index: include/wx/mac/carbon/private.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/mac/carbon/private.h,v
retrieving revision 1.50
diff -b -u -2 -r1.50 private.h
--- include/wx/mac/carbon/private.h	2006/03/28 10:32:00	1.50
+++ include/wx/mac/carbon/private.h	2006/04/06 07:35:37
@@ -221,5 +221,7 @@
 template<> inline EventParamType wxMacGetEventParamType<void*>() { return typeVoidPtr ; }
 #endif
+#if defined( __WXMAC__ ) && TARGET_API_MAC_OSX && ( MAC_OS_X_VERSION_MAX_ALLOWED > MAC_OS_X_VERSION_10_2 )
 template<> inline EventParamType wxMacGetEventParamType<CFDictionaryRef>() { return typeCFDictionaryRef ; }
+#endif
 template<> inline EventParamType wxMacGetEventParamType<Collection>() { return typeCollection ; }
 template<> inline EventParamType wxMacGetEventParamType<CGContextRef>() { return typeCGContextRef ; }
@@ -388,5 +390,5 @@
     ~wxMacCFRefHolder()
     {
-        CFRelease( m_ref ) ;
+        Release() ;
     }
 

 	  	 
