Index: include/wx/unichar.h
===================================================================
--- include/wx/unichar.h	(revision 46687)
+++ include/wx/unichar.h	(working copy)
@@ -19,7 +19,8 @@
 // ones it's a separate type and we must provide a conversion to it to allow
 // passing wxUniChar[Ref] to functions taking wint_t such as iswalnum() &c
 #if (defined(__GNUC__) && \
-        !defined(__DARWIN__) && !defined(__OS2__) && !defined(__DOS__)) || \
+        !defined(__DARWIN__) && !defined(__OS2__) && !defined(__DOS__)) && \
+        !defined(__FREEBSD__) || \
     (defined(__VISUALC__) && defined(_NATIVE_WCHAR_T_DEFINED)) || \
     wxCHECK_SUNCC_VERSION(5, 9)
     #define wxWINT_T_IS_SEPARATE_TYPE
Index: include/wx/wxcrtbase.h
===================================================================
--- include/wx/wxcrtbase.h	(revision 46687)
+++ include/wx/wxcrtbase.h	(working copy)
@@ -167,7 +167,7 @@
 #endif
 #if defined(__WINDOWS__)
     #define wxCRT_StrdupW _wcsdup
-#elif !defined(__DARWIN__) && !defined( __VMS )
+#elif !defined(__DARWIN__) && !defined( __VMS ) && !defined(__FREEBSD__)
     #define wxCRT_StrdupW wcsdup
 #endif
 

 	  	 
