diff -rbu2 ./cvs/wx2/wxwindows/src/generic/treelay.cpp ./wx2/src/generic/treelay.cpp
--- ./cvs/wx2/wxwindows/src/generic/treelay.cpp	Sat Jul 15 21:50:30 2000
+++ ./wx2/src/generic/treelay.cpp	Fri Jun  1 16:33:30 2001
@@ -15,5 +15,5 @@
 
 // For compilers that support precompilation, includes "wx.h".
-#include <wx/wxprec.h>
+#include "wx/wxprec.h"
 
 #ifdef __BORLANDC__
@@ -27,5 +27,5 @@
 #if wxUSE_TREELAYOUT
 
-#include <wx/treelay.h>
+#include "wx/treelay.h"
 
 /*

 	  	 
