Index: src/common/uri.cpp
===================================================================
--- src/common/uri.cpp	(revision 53959)
+++ src/common/uri.cpp	(working copy)
@@ -648,11 +648,11 @@
         if (bNormalize)
         {
             wxStringBufferLength theBuffer(m_path, m_path.length() + 1);
-#if wxUSE_STL || wxUSE_UNICODE_UTF8
-            // FIXME-UTF8: have some wxReadWriteStringBuffer instead?
+
+			// FIXME-UTF8: have some wxReadWriteStringBuffer instead?
             wxTmemcpy(theBuffer, m_path.c_str(), m_path.length()+1);
-#endif
-            Normalize(theBuffer, true);
+
+			Normalize(theBuffer, true);
             theBuffer.SetLength(wxStrlen(theBuffer));
         }
         //mark the path as valid
