--- wxWidgets-3.0.0/src/msw/dialog.cpp.original	2013-11-12 14:49:15.000000000 +0800
+++ wxWidgets-3.0.0/src/msw/dialog.cpp	2013-11-16 22:30:01.000000000 +0800
@@ -175,10 +175,7 @@
         // otherwise we could have some flicker
         InitDialog();
-    }
 
-    wxDialogBase::Show(show);
+        wxDialogBase::Show(show);
 
-    if ( show )
-    {
         // dialogs don't get WM_SIZE message from ::ShowWindow() for some
         // reason so generate it ourselves for consistency with frames and
