Index: src/common/stringops.cpp
===================================================================
--- src/common/stringops.cpp	(Revision 72479)
+++ src/common/stringops.cpp	(Arbeitskopie)
@@ -254,7 +254,6 @@
 {
     wxASSERT( IsValidUtf8LeadByte(*i) );
 
-    wxUniChar::value_type code = 0;
     size_t len = GetUtf8CharLength(*i);
     wxASSERT_MSG( len <= 4, wxT("invalid UTF-8 sequence length") );
 
@@ -282,7 +281,7 @@
     wxASSERT_MSG( ((unsigned char)*i & s_leadMarkerMask[len-1]) ==
                   s_leadMarkerVal[len-1],
                   wxT("invalid UTF-8 lead byte") );
-    code = (unsigned char)*i & s_leadValueMask[len-1];
+    wxUniChar::value_type code = (unsigned char)*i & s_leadValueMask[len-1];
 
     // all remaining bytes, if any, are handled in the same way regardless of
     // sequence's length:
