Index: include/wx/msw/frame.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/msw/frame.h,v
retrieving revision 1.78
diff -u -w -b -r1.78 frame.h
--- include/wx/msw/frame.h	2006/01/23 16:38:04	1.78
+++ include/wx/msw/frame.h	2006/01/24 12:26:40
@@ -57,6 +57,7 @@
                                      const wxString& name = wxToolBarNameStr);

     virtual void PositionToolBar();
+    virtual void SetToolBar(wxToolBar *toolbar);
 #endif // wxUSE_TOOLBAR

     // Status bar
Index: samples/xrc/rc/controls.xrc
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/samples/xrc/rc/controls.xrc,v
retrieving revision 1.5
diff -u -w -b -r1.5 controls.xrc
--- samples/xrc/rc/controls.xrc	2005/10/09 18:40:30	1.5
+++ samples/xrc/rc/controls.xrc	2006/01/22 22:07:31
@@ -54,8 +54,132 @@
                                     </object>
                                 </object>
                 </object>
-
-
+                <object class="notebookpage">
+                    <label>wxToolBar</label>
+                    <object class="wxPanel" name="button">
+                        <object class="wxFlexGridSizer">
+                            <cols>2</cols>
+                            <rows>0</rows>
+                            <vgap>0</vgap>
+                            <hgap>0</hgap>
+                            <growablecols>0,1</growablecols>
+                            <growablerows>0,1</growablerows>
+                            <object class="sizeritem">
+                                <flag>wxALIGN_RIGHT|wxALIGN_CENTER_VERTICAL|wxALL</flag>
+                                <border>5</border>
+                                <object class="wxStaticText" name="controls_normal_button">
+                                    <label>Transparent background:</label>
+                                </object>
+                            </object>
+                            <object class="sizeritem">
+                                <flag>wxALIGN_CENTER_VERTICAL|wxALL</flag>
+                                <border>5</border>
+                                <object class="wxToolBar">
+                                    <style>wxTB_FLAT|wxTB_NODIVIDER</style>
+                                    <margins>2,2</margins>
+                                    <object class="tool" name="non_derived_dialog_tool_or_menuitem">
+                                        <tooltip>Non-Derived Dialog Example</tooltip>
+                                        <bitmap>basicdlg.xpm</bitmap>
+                                        <longhelp>The basic instance of wxDialog loaded via XRC</longhelp>
+                                    </object>
+                                    <object class="tool" name="derived_tool_or_menuitem">
+                                        <tooltip>Derived Dialog Example</tooltip>
+                                        <bitmap>derivdlg.xpm</bitmap>
+                                        <longhelp>The most important example: load an XRC for a derived dialog</longhelp>
+                                    </object>
+                                    <object class="tool" name="controls_tool_or_menuitem">
+                                        <tooltip>Controls Example</tooltip>
+                                        <bitmap>controls.xpm</bitmap>
+                                        <longhelp>A notebook displaying all the wxWidgets controls</longhelp>
+                                    </object>
+                                    <object class="tool" name="uncentered_tool_or_menuitem">
+                                        <tooltip>Uncentered Example</tooltip>
+                                        <bitmap>uncenter.xpm</bitmap>
+                                        <longhelp>Disable autocentering of a dialog on its parent</longhelp>
+                                    </object>
+                                    <separator/>
+                                     <object class="tool" name="custom_class_tool_or_menuitem">
+                                        <tooltip>Custom Class Example</tooltip>
+                                        <bitmap>custclas.xpm</bitmap>
+                                        <longhelp>Embed your own custom classes into an XRC</longhelp>
+                                    </object>
+                                    <object class="tool" name="platform_property_tool_or_menuitem">
+                                        <tooltip>Platform Specific Example</tooltip>
+                                        <bitmap>platform.xpm</bitmap>
+                                        <longhelp>Customize parts of an interface for a desired platform</longhelp>
+                                    </object>
+                                    <object class="tool" name="art_provider_tool_or_menuitem">
+                                        <tooltip>wxArtProvider Example</tooltip>
+                                        <bitmap>artprov.xpm</bitmap>
+                                        <longhelp>Use wxArtProvider to insert desired icons at runtime</longhelp>
+                                    </object>
+                                    <object class="tool" name="variable_expansion_tool_or_menuitem">
+                                        <tooltip>Variable Expansion Example</tooltip>
+                                        <bitmap>variable.xpm</bitmap>
+                                        <longhelp>Replace variables in the XRC file at runtime</longhelp>
+                                    </object>
+                                </object>
+                            </object>
+                            <object class="sizeritem">
+                                <flag>wxALIGN_RIGHT|wxALIGN_CENTER_VERTICAL|wxALL</flag>
+                                <border>5</border>
+                                <object class="wxStaticText" name="controls_statictext2_button">
+                                    <label>Fixed color background:</label>
+                                </object>
+                            </object>
+                            <object class="sizeritem">
+                                <flag>wxALIGN_CENTER_VERTICAL|wxALL</flag>
+                                <border>5</border>
+                                <object class="wxToolBar">
+                                    <style>wxTB_FLAT|wxTB_NODIVIDER</style>
+                                    <bg>#bbbbff</bg>
+                                    <margins>2,2</margins>
+                                    <object class="tool" name="non_derived_dialog_tool_or_menuitem">
+                                        <tooltip>Non-Derived Dialog Example</tooltip>
+                                        <bitmap>basicdlg.xpm</bitmap>
+                                        <longhelp>The basic instance of wxDialog loaded via XRC</longhelp>
+                                    </object>
+                                    <object class="tool" name="derived_tool_or_menuitem">
+                                        <tooltip>Derived Dialog Example</tooltip>
+                                        <bitmap>derivdlg.xpm</bitmap>
+                                        <longhelp>The most important example: load an XRC for a derived dialog</longhelp>
+                                    </object>
+                                    <object class="tool" name="controls_tool_or_menuitem">
+                                        <tooltip>Controls Example</tooltip>
+                                        <bitmap>controls.xpm</bitmap>
+                                        <longhelp>A notebook displaying all the wxWidgets controls</longhelp>
+                                    </object>
+                                    <object class="tool" name="uncentered_tool_or_menuitem">
+                                        <tooltip>Uncentered Example</tooltip>
+                                        <bitmap>uncenter.xpm</bitmap>
+                                        <longhelp>Disable autocentering of a dialog on its parent</longhelp>
+                                    </object>
+                                    <separator/>
+                                     <object class="tool" name="custom_class_tool_or_menuitem">
+                                        <tooltip>Custom Class Example</tooltip>
+                                        <bitmap>custclas.xpm</bitmap>
+                                        <longhelp>Embed your own custom classes into an XRC</longhelp>
+                                    </object>
+                                    <object class="tool" name="platform_property_tool_or_menuitem">
+                                        <tooltip>Platform Specific Example</tooltip>
+                                        <bitmap>platform.xpm</bitmap>
+                                        <longhelp>Customize parts of an interface for a desired platform</longhelp>
+                                    </object>
+                                    <object class="tool" name="art_provider_tool_or_menuitem">
+                                        <tooltip>wxArtProvider Example</tooltip>
+                                        <bitmap>artprov.xpm</bitmap>
+                                        <longhelp>Use wxArtProvider to insert desired icons at runtime</longhelp>
+                                    </object>
+                                    <object class="tool" name="variable_expansion_tool_or_menuitem">
+                                        <tooltip>Variable Expansion Example</tooltip>
+                                        <bitmap>variable.xpm</bitmap>
+                                        <longhelp>Replace variables in the XRC file at runtime</longhelp>
+                                    </object>
+                                </object>
+                            </object>
+                        </object>
+                    </object>
+                </object>
                 <object class="notebookpage">
                     <label>wxBitmapButton</label>
                     <object class="wxPanel" name="bitmapbutton">
Index: src/msw/frame.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/msw/frame.cpp,v
retrieving revision 1.196
diff -u -w -b -r1.196 frame.cpp
--- src/msw/frame.cpp	2006/01/23 16:38:07	1.196
+++ src/msw/frame.cpp	2006/01/24 12:27:51
@@ -441,6 +441,14 @@
         m_frameStatusBar->GetEventHandler()->ProcessEvent(event2);
     }
 #endif // wxUSE_STATUSBAR
+#if wxUSE_TOOLBAR
+    if ( m_frameToolBar )
+    {
+        // Frame should force toolbar to use background once more,
+        // since it does not draw it itself.
+        m_frameToolBar->SetBackgroundColour(wxSystemSettings::GetColour(wxSYS_COLOUR_BTNFACE));
+    }
+#endif // wxUSE_TOOLBAR

     // Propagate the event to the non-top-level children
     wxWindow::OnSysColourChanged(event);
@@ -556,10 +564,20 @@
 #endif
     if ( wxFrameBase::CreateToolBar(style, id, name) )
     {
+        // Frame should force toolbar to use background, since it does not draw it itself.
+        m_frameToolBar->SetBackgroundColour(wxSystemSettings::GetColour(wxSYS_COLOUR_BTNFACE));
         PositionToolBar();
     }

     return m_frameToolBar;
+}
+
+void wxFrame::SetToolBar(wxToolBar *toolbar)
+{
+    // Frame should force toolbar to use background, since it does not draw it itself.
+    if(toolbar && !toolbar->UseBgCol())
+        toolbar->SetBackgroundColour(wxSystemSettings::GetColour(wxSYS_COLOUR_BTNFACE));
+    wxFrameBase::SetToolBar(toolbar);
 }

 void wxFrame::PositionToolBar()
Index: src/msw/tbar95.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/msw/tbar95.cpp,v
retrieving revision 1.168
diff -u -w -b -r1.168 tbar95.cpp
--- src/msw/tbar95.cpp	2006/01/21 16:47:25	1.168
+++ src/msw/tbar95.cpp	2006/01/24 13:00:34
@@ -235,8 +235,9 @@
     wxSetCCUnicodeFormat(GetHwnd());

     // set up the colors and fonts
-    SetBackgroundColour(wxSystemSettings::GetColour(wxSYS_COLOUR_BTNFACE));
     SetFont(wxSystemSettings::GetFont(wxSYS_DEFAULT_GUI_FONT));
+    // this should be set by the frame, not by the toolbar itself
+    /* SetBackgroundColour(wxSystemSettings::GetColour(wxSYS_COLOUR_BTNFACE)); */

     // workaround for flat toolbar on Windows XP classic style: we have to set
     // the style after creating the control; doing it at creation time doesn't work
@@ -1302,15 +1303,16 @@
 // Responds to colour changes, and passes event on to children.
 void wxToolBar::OnSysColourChanged(wxSysColourChangedEvent& event)
 {
-    wxRGBToColour(m_backgroundColour, ::GetSysColor(COLOR_BTNFACE));
-
     // Remap the buttons
     Realize();

-    // Relayout the toolbar
-    int nrows = m_maxRows;
-    m_maxRows = 0;      // otherwise SetRows() wouldn't do anything
-    SetRows(nrows);
+    // FIXME: Relayout the toolbar
+    // This causes toolbar to be placed incorrectly on non wxFrame windows
+    // using sizers. See XRC sample.
+
+    // int nrows = m_maxRows;
+    // m_maxRows = 0;      // otherwise SetRows() wouldn't do anything
+    // SetRows(nrows);

     Refresh();

@@ -1347,9 +1349,33 @@
 void wxToolBar::OnEraseBackground(wxEraseEvent& event)
 {
     wxColour bgCol = GetBackgroundColour();
-    if (!bgCol.Ok())
+    if (!bgCol.Ok() || !UseBgCol())
+    {
+#if !defined(__WXWINCE__) && wxUSE_UXTHEME
+        // we should draw parent background if possible on themed system
+        // for toolbar using DrawThemeParentBackground, which will make toolbar
+        // transparent for example on notebooks
+
+        wxUxThemeEngine *themeEngine = wxUxThemeEngine::GetIfActive();
+        if ( !themeEngine )
     {
         event.Skip();
+            return;
+        }
+
+        RECT rect;
+        HDC hdc = GetHdcOf((*event.GetDC()));
+
+        ::GetClientRect(GetHwnd(), &rect);
+        HRESULT hr =
+            themeEngine->DrawThemeParentBackground(GetHwnd(), hdc, &rect);
+        if ( FAILED(hr) )
+        {
+            wxLogApiError(_T("DrawThemeBackground(BP_TOOLBAR)"), hr);
+        }
+#else
+        event.Skip();
+#endif
         return;
     }

Index: src/xrc/xh_toolb.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/xrc/xh_toolb.cpp,v
retrieving revision 1.18
diff -u -w -b -r1.18 xh_toolb.cpp
--- src/xrc/xh_toolb.cpp	2005/09/23 12:56:15	1.18
+++ src/xrc/xh_toolb.cpp	2006/01/22 22:04:18
@@ -115,6 +115,8 @@
         long separation = GetLong(wxT("separation"), -1);
         if (separation != -1)
             toolbar->SetToolSeparation(separation);
+        if (HasParam(wxT("bg")))
+            toolbar->SetBackgroundColour(GetColour(wxT("bg")));

         wxXmlNode *children_node = GetParamNode(wxT("object"));
         if (!children_node)

 	  	 
