Index: src/common/memory.cpp
===================================================================
--- src/common/memory.cpp	(revision 54486)
+++ src/common/memory.cpp	(working copy)
@@ -370,7 +370,7 @@
     msg2.Printf(wxT(" at 0x%lX, size %d"), (long)GetActualData(), (int)RequestSize());
     msg += msg2;
 
-    wxDebugContext::OutputDumpLine(msg);
+    wxDebugContext::OutputDumpLine(msg.c_str());
   }
   else
   {
@@ -381,7 +381,7 @@
     wxString msg2;
     msg2.Printf(wxT("non-object data at 0x%lX, size %d"), (long)GetActualData(), (int)RequestSize() );
     msg += msg2;
-    wxDebugContext::OutputDumpLine(msg);
+    wxDebugContext::OutputDumpLine(msg.c_str());
   }
 }
 
@@ -588,13 +588,14 @@
 {
 #ifdef __WXDEBUG__
   {
-    wxChar* appName = (wxChar*) wxT("application");
+    const wxChar* appName = wxT("application");
     wxString appNameStr;
     if (wxTheApp)
     {
         appNameStr = wxTheApp->GetAppName();
-        appName = WXSTRINGCAST appNameStr;
-        OutputDumpLine(wxT("----- Memory dump of %s at %s -----"), appName, WXSTRINGCAST wxNow() );
+        appName = appNameStr.c_str();
+        OutputDumpLine(wxT("----- Memory dump of %s at %s -----"), appName,
+            wx_static_cast( const wxChar *,wxNow().c_str() ));
     }
     else
     {
@@ -644,13 +645,14 @@
 {
 #ifdef __WXDEBUG__
   {
-    wxChar* appName = (wxChar*) wxT("application");
+    const wxChar* appName = wxT("application");
     wxString appNameStr;
     if (wxTheApp)
     {
         appNameStr = wxTheApp->GetAppName();
-        appName = WXSTRINGCAST appNameStr;
-        OutputDumpLine(wxT("----- Memory statistics of %s at %s -----"), appName, WXSTRINGCAST wxNow() );
+        appName = appNameStr.c_str();
+        OutputDumpLine(wxT("----- Memory statistics of %s at %s -----"), appName,
+            wx_static_cast( const wxChar *, wxNow().c_str() ));
     }
     else
     {
@@ -738,12 +740,12 @@
 bool wxDebugContext::PrintClasses(void)
 {
   {
-    wxChar* appName = (wxChar*) wxT("application");
+    const wxChar* appName = wxT("application");
     wxString appNameStr;
     if (wxTheApp)
     {
         appNameStr = wxTheApp->GetAppName();
-        appName = WXSTRINGCAST appNameStr;
+        appName = appNameStr.c_str();
         wxLogMessage(wxT("----- Classes in %s -----"), appName);
     }
   }
