diff -uNr wxWidgets-orig/src/osx/core/timer.cpp wxWidgets-mine/src/osx/core/timer.cpp
--- wxWidgets-orig/src/osx/core/timer.cpp	2014-07-05 17:35:24.000000000 -0500
+++ wxWidgets-mine/src/osx/core/timer.cpp	2014-07-05 20:37:02.000000000 -0500
@@ -21,8 +21,9 @@
 
 struct wxOSXTimerInfo
 {
-    wxOSXTimerImpl*     m_timer;
-    CFRunLoopTimerRef   m_timerRef;
+    wxOSXTimerImpl*         m_timer;
+    CFRunLoopTimerRef       m_timerRef;
+    CFRunLoopTimerContext   m_ctx;
 };
 
 void wxProcessTimer(CFRunLoopTimerRef WXUNUSED(theTimer), void *data)
@@ -73,17 +74,16 @@
     memset(&gumilli,0,sizeof(gumilli) );
     gumilli.seconds = m_milli / 1000.0;
 
-    CFRunLoopTimerContext ctx ;
-    memset( &ctx, 0 , sizeof(ctx) );
-    ctx.version = 0;
-    ctx.info = this;
+    memset( &m_info->m_ctx, 0 , sizeof(m_info->m_ctx) );
+    m_info->m_ctx.version = 0;
+    m_info->m_ctx.info = this;
 
     m_info->m_timer = this;
     m_info->m_timerRef = CFRunLoopTimerCreate(
         kCFAllocatorDefault,
         CFAbsoluteTimeAddGregorianUnits( CFAbsoluteTimeGetCurrent() , NULL, gumilli ),
         IsOneShot() ? 0 : CFTimeInterval( m_milli / 1000.0 ) ,
-        0, 0, wxProcessTimer, &ctx);
+        0, 0, wxProcessTimer, &m_info->m_ctx);
 
     wxASSERT_MSG( m_info->m_timerRef != NULL, wxT("unable to create timer"));
 
