diff -rupN wxWidgets-Nightly-orig/include/wx/fontpicker.h wxWidgets-Nightly-new/include/wx/fontpicker.h
--- wxWidgets-Nightly-orig/include/wx/fontpicker.h	2015-09-02 15:06:52.000000000 +0200
+++ wxWidgets-Nightly-new/include/wx/fontpicker.h	2015-09-03 12:53:38.056343200 +0200
@@ -75,9 +75,11 @@ protected:
 #if defined(__WXGTK20__) && !defined(__WXUNIVERSAL__)
     #include "wx/gtk/fontpicker.h"
     #define wxFontPickerWidget      wxFontButton
+    #define SetMinMaxPointSize(min, max)
 #else
     #include "wx/generic/fontpickerg.h"
     #define wxFontPickerWidget      wxGenericFontButton
+    #define SetMinMaxPointSize(min, max)  GetPickerWidget()->GetFontData()->SetRange((min), (max))
 #endif
 
 
@@ -88,7 +90,8 @@ protected:
 #define wxFNTP_USE_TEXTCTRL       (wxPB_USE_TEXTCTRL)
 #define wxFNTP_DEFAULT_STYLE      (wxFNTP_FONTDESC_AS_LABEL|wxFNTP_USEFONT_FOR_LABEL)
 
-// not a style but rather the default value of the maximum pointsize allowed
+// not a style but rather the default value of the minimum/maximum pointsize allowed
+#define wxFNTP_MINPOINT_SIZE      0
 #define wxFNTP_MAXPOINT_SIZE      100
 
 
@@ -102,7 +105,7 @@ class WXDLLIMPEXP_CORE wxFontPickerCtrl
 {
 public:
     wxFontPickerCtrl()
-        : m_nMaxPointSize(wxFNTP_MAXPOINT_SIZE)
+        : m_nMinPointSize(wxFNTP_MINPOINT_SIZE), m_nMaxPointSize(wxFNTP_MAXPOINT_SIZE)
     {
     }
 
@@ -117,7 +120,7 @@ public:
                      long style = wxFNTP_DEFAULT_STYLE,
                      const wxValidator& validator = wxDefaultValidator,
                      const wxString& name = wxFontPickerCtrlNameStr)
-        : m_nMaxPointSize(wxFNTP_MAXPOINT_SIZE)
+        : m_nMinPointSize(wxFNTP_MINPOINT_SIZE), m_nMaxPointSize(wxFNTP_MAXPOINT_SIZE)
     {
         Create(parent, id, initial, pos, size, style, validator, name);
     }
@@ -149,9 +152,13 @@ public:         // public API
     void SetSelectedColour(const wxColour& colour)
         { GetPickerWidget()->SetSelectedColour(colour); }
 
+    // set/get the min point size
+    void SetMinPointSize(unsigned int min);
+     unsigned int GetMinPointSize() const
+        { return m_nMinPointSize; }
+        
     // set/get the max point size
-    void SetMaxPointSize(unsigned int max)
-        { m_nMaxPointSize=max; }
+    void SetMaxPointSize(unsigned int max);
     unsigned int GetMaxPointSize() const
         { return m_nMaxPointSize; }
 
@@ -173,9 +180,12 @@ protected:
     long GetPickerStyle(long style) const wxOVERRIDE
         { return (style & (wxFNTP_FONTDESC_AS_LABEL|wxFNTP_USEFONT_FOR_LABEL)); }
 
+    // the minimum pointsize allowed to the user
+    unsigned int m_nMinPointSize;
+    
     // the maximum pointsize allowed to the user
     unsigned int m_nMaxPointSize;
-
+    
 private:
     wxFontPickerWidget* GetPickerWidget() const
         { return static_cast<wxFontPickerWidget*>(m_picker); }
diff -rupN wxWidgets-Nightly-orig/include/wx/generic/fontpickerg.h wxWidgets-Nightly-new/include/wx/generic/fontpickerg.h
--- wxWidgets-Nightly-orig/include/wx/generic/fontpickerg.h	2015-09-02 15:06:52.000000000 +0200
+++ wxWidgets-Nightly-new/include/wx/generic/fontpickerg.h	2015-09-03 12:19:02.504462600 +0200
@@ -46,9 +46,6 @@ public:
 
 public:     // API extensions specific for wxGenericFontButton
 
-    // user can override this to init font data in a different way
-    virtual void InitFontData();
-
     // returns the font data shown in wxFontDialog
     wxFontData *GetFontData() { return &m_data; }
 
diff -rupN wxWidgets-Nightly-orig/src/common/fontpickercmn.cpp wxWidgets-Nightly-new/src/common/fontpickercmn.cpp
--- wxWidgets-Nightly-orig/src/common/fontpickercmn.cpp	2015-09-02 15:06:52.000000000 +0200
+++ wxWidgets-Nightly-new/src/common/fontpickercmn.cpp	2015-09-03 12:53:49.943564100 +0200
@@ -150,7 +150,17 @@ void wxFontPickerCtrl::UpdateTextCtrlFro
     m_text->ChangeValue(Font2String(GetPickerWidget()->GetSelectedFont()));
 }
 
+void wxFontPickerCtrl::SetMinPointSize(unsigned int min)
+{
+    m_nMinPointSize=min;
+    SetMinMaxPointSize(m_nMinPointSize, m_nMaxPointSize);
+}    
 
+void wxFontPickerCtrl::SetMaxPointSize(unsigned int max)
+{
+    m_nMaxPointSize=max;
+    SetMinMaxPointSize(m_nMinPointSize, m_nMaxPointSize);     
+} 
 
 // ----------------------------------------------------------------------------
 // wxFontPickerCtrl - event handlers
diff -rupN wxWidgets-Nightly-orig/src/generic/fontpickerg.cpp wxWidgets-Nightly-new/src/generic/fontpickerg.cpp
--- wxWidgets-Nightly-orig/src/generic/fontpickerg.cpp	2015-09-02 15:06:52.000000000 +0200
+++ wxWidgets-Nightly-new/src/generic/fontpickerg.cpp	2015-09-03 12:59:46.741391000 +0200
@@ -62,7 +62,9 @@ bool wxGenericFontButton::Create( wxWind
             wxCommandEventHandler(wxGenericFontButton::OnButtonClick),
             NULL, this);
 
-    InitFontData();
+    m_data.SetAllowSymbols(true);
+    m_data.SetColour(*wxBLACK);
+    m_data.EnableEffects(true);
 
     m_selectedFont = initial.IsOk() ? initial : *wxNORMAL_FONT;
     UpdateFont();
@@ -70,13 +72,6 @@ bool wxGenericFontButton::Create( wxWind
     return true;
 }
 
-void wxGenericFontButton::InitFontData()
-{
-    m_data.SetAllowSymbols(true);
-    m_data.SetColour(*wxBLACK);
-    m_data.EnableEffects(true);
-}
-
 void wxGenericFontButton::OnButtonClick(wxCommandEvent& WXUNUSED(ev))
 {
     // update the wxFontData to be shown in the dialog
