Index: include/wx/layout.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/layout.h,v
retrieving revision 1.24
diff -u -r1.24 layout.h
--- include/wx/layout.h	2005/09/23 12:48:42	1.24
+++ include/wx/layout.h	2006/03/13 09:51:11
@@ -12,6 +12,7 @@
 #ifndef _WX_LAYOUTH__
 #define _WX_LAYOUTH__
 
+#if wxUSE_CONSTRAINTS
 // ----------------------------------------------------------------------------
 // headers
 // ----------------------------------------------------------------------------
@@ -184,5 +185,6 @@
     DECLARE_DYNAMIC_CLASS(wxLayoutConstraints)
 };
 
+#endif // wxUSE_CONSTRAINTS
 #endif
     // _WX_LAYOUTH__


 	  	 
