Index: include/wx/generic/progdlgg.h
===================================================================
--- include/wx/generic/progdlgg.h	(revision 60585)
+++ include/wx/generic/progdlgg.h	(working copy)
@@ -69,6 +69,11 @@
 
    virtual bool Show( bool show = true );
 
+#if wxABI_VERSION >= 20811
+   // Refresh the dialog and test for user interaction
+   bool YieldAndRefresh(bool *skip);
+#endif
+
 protected:
    // callback for optional abort button
    void OnCancel(wxCommandEvent& event);
@@ -91,9 +96,6 @@
     // updates the label message
    void UpdateMessage(const wxString &newmsg);
 
-   // common part of Update() and Pulse(), returns true if not cancelled
-   bool DoAfterUpdate(bool *skip);
-
    // shortcuts for enabling buttons
    void EnableClose();
    void EnableSkip(bool enable=true);
Index: src/generic/progdlgg.cpp
===================================================================
--- src/generic/progdlgg.cpp	(revision 60585)
+++ src/generic/progdlgg.cpp	(working copy)
@@ -434,7 +434,7 @@
     }
     else // not at maximum yet
     {
-        return DoAfterUpdate(skip);
+        return YieldAndRefresh(skip);
     }
 
     // update the display in case yielding above didn't do it
@@ -461,10 +461,10 @@
         SetTimeLabel((unsigned long)-1, m_remaining);
     }
 
-    return DoAfterUpdate(skip);
+    return YieldAndRefresh(skip);
 }
 
-bool wxProgressDialog::DoAfterUpdate(bool *skip)
+bool wxProgressDialog::YieldAndRefresh(bool *skip)
 {
     // we have to yield because not only we want to update the display but
     // also to process the clicks on the cancel and skip buttons
