Index: src/gtk/dirdlg.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/gtk/dirdlg.cpp,v
retrieving revision 1.13
diff -u2 -r1.13 src/gtk/dirdlg.cpp
--- src/gtk/dirdlg.cpp
+++ src/gtk/dirdlg.cpp
@@ -203,5 +203,7 @@
         if (wxDirExists(dir))
         {
-            gtk_file_chooser_set_filename(GTK_FILE_CHOOSER(m_widget), wxConvFileName->cWX2MB(dir));
+            // NB: gtk_file_chooser_set_filename does not set the right path - we must use
+            //     gtk_file_chooser_set_current_folder instead
+            gtk_file_chooser_set_current_folder(GTK_FILE_CHOOSER(m_widget), wxConvFileName->cWX2MB(dir));
         }
     }

 	  	 
