Index: src/common/imagjpeg.cpp
===================================================================
RCS file: /home/wxcvs/wxWindows/src/common/imagjpeg.cpp,v
retrieving revision 1.21
diff -b -u -2 -r1.21 imagjpeg.cpp
--- src/common/imagjpeg.cpp	2001/05/02 19:48:08	1.21
+++ src/common/imagjpeg.cpp	2001/11/23 13:38:46
@@ -198,4 +198,5 @@
       if (verbose)
         wxLogError(_("JPEG: Couldn't load - file is probably corrupted."));
+      (cinfo.src->term_source) (&cinfo);;
       jpeg_destroy_decompress(&cinfo);
       if (image->Ok()) image->Destroy();


 	  	 
