Index: include/wx/hashmap.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/hashmap.h,v
retrieving revision 1.53
diff -b -u -2 -r1.53 hashmap.h
--- include/wx/hashmap.h	2006/01/18 16:45:25	1.53
+++ include/wx/hashmap.h	2006/02/07 02:48:30
@@ -35,5 +35,5 @@
 
 #define _WX_DECLARE_HASH_MAP( KEY_T, VALUE_T, HASH_T, KEY_EQ_T, CLASSNAME, CLASSEXP ) \
-    typedef WX_HASH_MAP_NAMESPACE::hash_map< KEY_T, VALUE_T, HASH_T, KEY_EQ_T > CLASSNAME;
+    typedef WX_HASH_MAP_NAMESPACE::hash_map< KEY_T, VALUE_T, HASH_T, KEY_EQ_T > CLASSNAME
 
 #else // !wxUSE_STL || !defined(HAVE_STL_HASH_MAP)
Index: include/wx/hashset.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/hashset.h,v
retrieving revision 1.8
diff -b -u -2 -r1.8 hashset.h
--- include/wx/hashset.h	2006/01/18 16:45:25	1.8
+++ include/wx/hashset.h	2006/02/07 02:48:34
@@ -24,5 +24,5 @@
 
 #define _WX_DECLARE_HASH_SET( KEY_T, HASH_T, KEY_EQ_T, CLASSNAME, CLASSEXP )\
-    typedef WX_HASH_MAP_NAMESPACE::hash_set< KEY_T, HASH_T, KEY_EQ_T > CLASSNAME;
+    typedef WX_HASH_MAP_NAMESPACE::hash_set< KEY_T, HASH_T, KEY_EQ_T > CLASSNAME
 
 #else // !wxUSE_STL || !defined(HAVE_STL_HASH_MAP)
Index: src/common/dynarray.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/common/dynarray.cpp,v
retrieving revision 1.56
diff -b -u -2 -r1.56 dynarray.cpp
--- src/common/dynarray.cpp	2006/01/18 16:45:37	1.56
+++ src/common/dynarray.cpp	2006/02/07 02:48:40
@@ -435,5 +435,5 @@
 #include "wx/afterstd.h"
 
-_WX_DEFINE_BASEARRAY(wxString, wxBaseArrayStringBase);
+_WX_DEFINE_BASEARRAY(wxString, wxBaseArrayStringBase)
 
 // some compilers (Sun CC being the only known example) distinguish between

 	  	 
