? cvsupdate
? textctrl.patch
Index: include/wx/generic/treectlg.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/generic/treectlg.h,v
retrieving revision 1.20.2.1
diff -b -u -2 -r1.20.2.1 treectlg.h
--- include/wx/generic/treectlg.h	2002/11/22 16:49:46	1.20.2.1
+++ include/wx/generic/treectlg.h	2002/12/10 05:13:47
@@ -34,4 +34,5 @@
 class WXDLLEXPORT wxTreeFindTimer;
 class WXDLLEXPORT wxTreeTextCtrl;
+class WXDLLEXPORT wxTextCtrl;
 
 // -----------------------------------------------------------------------------
@@ -316,4 +317,8 @@
     void EditLabel( const wxTreeItemId& item ) { Edit( item ); }
     void Edit( const wxTreeItemId& item );
+        // returns a pointer to the text edit control if the item is being
+        // edited, NULL otherwise (it's assumed that no more than one item may
+        // be edited simultaneously)
+    wxTextCtrl* GetEditControl() const;
 
     // sorting
@@ -386,4 +391,5 @@
     wxCursor             m_oldCursor;  // cursor is changed while dragging
     wxGenericTreeItem   *m_oldSelection;
+    wxTreeTextCtrl      *m_textCtrl;
 
     wxTimer             *m_renameTimer;
@@ -411,4 +417,7 @@
                               int image, int selectedImage,
                               wxTreeItemData *data);
+
+    // called by wxTextTreeCtrl when it marks itself for deletion
+    void ResetTextControl();
 
     // find the first item starting with the given prefix after the given item
Index: src/generic/treectlg.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/generic/treectlg.cpp,v
retrieving revision 1.85.2.2
diff -b -u -2 -r1.85.2.2 treectlg.cpp
--- src/generic/treectlg.cpp	2002/11/22 16:49:49	1.85.2.2
+++ src/generic/treectlg.cpp	2002/12/10 05:14:38
@@ -428,4 +428,6 @@
     if ( !m_finished )
     {
+        m_owner->ResetTextControl();
+
         wxPendingDelete.Append(this);
 
@@ -756,4 +758,5 @@
     m_isDragging = FALSE;
     m_dropTarget = m_oldSelection = (wxGenericTreeItem *)NULL;
+    m_textCtrl = NULL;
 
     m_renameTimer = NULL;
@@ -1266,4 +1269,10 @@
 }
 
+// called by wxTextTreeCtrl when it marks itself for deletion
+void wxGenericTreeCtrl::ResetTextControl()
+{
+  m_textCtrl = NULL;
+}
+
 // find the first item starting with the given prefix after the given item
 wxTreeItemId wxGenericTreeCtrl::FindItem(const wxTreeItemId& idParent,
@@ -2810,7 +2819,15 @@
         wxYieldIfNeeded();
 
-    wxTreeTextCtrl *text = new wxTreeTextCtrl(this, itemEdit);
+    m_textCtrl = new wxTreeTextCtrl(this, itemEdit);
 
-    text->SetFocus();
+    m_textCtrl->SetFocus();
+}
+
+// returns a pointer to the text edit control if the item is being
+// edited, NULL otherwise (it's assumed that no more than one item may
+// be edited simultaneously)
+wxTextCtrl* wxGenericTreeCtrl::GetEditControl() const
+{
+    return m_textCtrl;
 }
 

 	  	 
