diff --git a/src/msw/artmsw.cpp b/src/msw/artmsw.cpp
index eb7a6c4..fe20e1c 100644
--- a/src/msw/artmsw.cpp
+++ b/src/msw/artmsw.cpp
@@ -25,6 +25,7 @@
 #include "wx/volume.h"
 #include "wx/msw/private.h"
 #include "wx/msw/wrapwin.h"
+#include "wx/msw/wrapshl.h"
 
 #ifdef SHGSI_ICON
     #define wxHAS_SHGetStockIconInfo
@@ -82,29 +83,29 @@ MSW_SHGetStockIconInfo(SHSTOCKICONID siid,
 #endif // #ifdef wxHAS_SHGetStockIconInfo
 
 wxBitmap
-MSWGetBitmapForPath(const wxString& path, const wxSize& size, DWORD uFlags = 0)
+MSWGetBitmapForPath( const wxString &path, const wxSize &size, bool open = false, bool selected = false )
 {
     SHFILEINFO fi;
     wxZeroMemory(fi);
 
-    uFlags |= SHGFI_USEFILEATTRIBUTES | SHGFI_ICON;
-    if ( size != wxDefaultSize )
-    {
-        if ( size.x <= 16 )
-            uFlags |= SHGFI_SMALLICON;
-        else if ( size.x >= 64 )
-            uFlags |= SHGFI_LARGEICON;
-    }
+    DWORD uFlags = SHGFI_USEFILEATTRIBUTES | SHGFI_ICONLOCATION ;
+    if ( open )       uFlags |= SHGFI_OPENICON;
+    if ( selected )   uFlags |= SHGFI_SELECTED;
+
+    if ( !SHGetFileInfo( path.t_str(), FILE_ATTRIBUTE_DIRECTORY,
+                         &fi, sizeof(SHFILEINFO), uFlags) )
+       return wxNullBitmap;
 
-    if ( !SHGetFileInfo(path.t_str(), FILE_ATTRIBUTE_DIRECTORY,
-                        &fi, sizeof(SHFILEINFO), uFlags) )
+    HICON icon_large = NULL;
+    if ( !SHDefExtractIcon( fi.szDisplayName, fi.iIcon, 0,
+                            &icon_large, NULL, size.x ) )
         return wxNullBitmap;
 
     wxIcon icon;
-    icon.CreateFromHICON((WXHICON)fi.hIcon);
+    icon.CreateFromHICON( icon_large );
 
     wxBitmap bitmap(icon);
-    ::DestroyIcon(fi.hIcon);
+    ::DestroyIcon(icon_large);
 
     return bitmap;
 }
@@ -238,7 +239,7 @@ wxBitmap wxWindowsArtProvider::CreateBitmap(const wxArtID& id,
     if ( id == wxART_FOLDER )
         bitmap = MSWGetBitmapForPath("C:\\wxdummydir\\", size );
     else if ( id == wxART_FOLDER_OPEN )
-        bitmap = MSWGetBitmapForPath("C:\\wxdummydir\\", size, SHGFI_OPENICON );
+        bitmap = MSWGetBitmapForPath("C:\\wxdummydir\\", size, true /* open */  );
 
     if ( !bitmap.IsOk() )
     {
