--- wxWidgets-trunk/src/msw/dib.cpp	2013-12-29
+++ wxWidgets-work/src/msw/dib.cpp	2014-03-19
@@ -467,6 +467,21 @@
         wxLogLastError(wxT("CreateDIBitmap"));
     }
 
+    BITMAP bm;
+    if( ::GetObject(hbmp, sizeof(bm), &bm) != sizeof(bm) )
+    {
+        wxLogLastError(wxT("GetObject (ConvertToBitmap)"));
+        wxZeroMemory(bm);  //  bm.bmBitsPixel = 0
+    }
+
+    // If bitmap has different color depth 
+    // then demanded we need to reject it.
+    if( bm.bmBitsPixel != pbmih->biBitCount )
+    {
+        ::DeleteObject(hbmp);
+        hbmp = NULL;
+    }
+
     return hbmp;
 }
 
