Index: minimal.cpp
===================================================================
--- minimal.cpp	(revision 63200)
+++ minimal.cpp	(working copy)
@@ -30,6 +30,8 @@
     #include "wx/wx.h"
 #endif
 
+#include <wx/rawbmp.h>
+
 // ----------------------------------------------------------------------------
 // resources
 // ----------------------------------------------------------------------------
@@ -141,6 +143,43 @@
 // main frame
 // ----------------------------------------------------------------------------
 
+void Test()
+{
+    int w = 100;
+    int h = 100;
+    int bpp = 24;
+
+    wxBitmap dib(w, h, bpp);
+    {
+        wxNativePixelData data(dib);
+    }
+
+    {
+        wxMemoryDC mdc(dib);
+        mdc.SelectObject(wxNullBitmap);
+        {
+            // fine, dib is still a dib.
+            wxNativePixelData data(dib);
+        }
+        mdc.SelectObject(dib);
+
+    }
+
+    {
+        {
+            wxMemoryDC mdc(dib);
+            // Selected bitmap into DC twice!
+            // Would be great if this step asserted.
+            mdc.SelectObject(dib); 
+        }
+        {
+            // dib is no longer a dib!
+            // This will therefore assert.
+            wxNativePixelData data(dib);
+        }
+    }
+}
+
 // frame constructor
 MyFrame::MyFrame(const wxString& title)
        : wxFrame(NULL, wxID_ANY, title)
@@ -172,6 +211,9 @@
     CreateStatusBar(2);
     SetStatusText("Welcome to wxWidgets!");
 #endif // wxUSE_STATUSBAR
+
+
+    Test();
 }
 
 
