--- wxWidgets-trunk/src/msw/bitmap.cpp	2014-03-07
+++ wxWidgets-work/src/msw/bitmap.cpp	2014-03-24
@@ -243,6 +243,27 @@
     {
         wxDIB dib((HBITMAP)(data.m_hBitmap));
         CopyFromDIB(dib);
+        BITMAP bm;
+        if( ::GetObject(m_hBitmap, sizeof(bm), &bm) != sizeof(bm) )
+        {
+            wxLogLastError(wxT("GetObject(hBitmap@wxBitmapRefData)"));
+        }
+        else if( m_depth != bm.bmBitsPixel )
+        {
+            // We got DDB with different colour depth then demanded.
+            // We need to drop it and create DIB instead.
+            wxDIB dibDst(m_width, m_height, m_depth);
+            if( dibDst.IsOk() )
+            {
+                memcpy(dibDst.GetData(), dib.GetData(), wxDIB::GetLineSize(m_width, m_depth)*m_height);
+                AssignDIB(dibDst);
+            }
+            else
+            {
+                // Nothing else left to do...
+                m_depth = bm.bmBitsPixel;
+            }
+        }
     }
 #endif // wxUSE_WXDIB
 }
