Index: variant.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/common/variant.cpp,v
retrieving revision 1.60
diff -b -u -2 -r1.60 variant.cpp
--- variant.cpp	2004/11/05 20:45:23	1.60
+++ variant.cpp	2005/01/31 16:59:13
@@ -2075,5 +2075,5 @@
     // Fallback to string conversion
     wxString val;
-    return Convert(&val) && (value->ParseDate(val));
+    return Convert(&val) && (value->ParseDateTime(val));
 }
 #endif // wxUSE_DATETIME

 	  	 
