Index: src/gtk/renderer.cpp
===================================================================
--- src/gtk/renderer.cpp	(revisione 69740)
+++ src/gtk/renderer.cpp	(copia locale)
@@ -256,8 +256,28 @@
     if (win->GetLayoutDirection() == wxLayout_RightToLeft)
         x_diff = rect.width;
 
+    wxCoord x = dc.LogicalToDeviceX(rect.x) - x_diff;
+    wxCoord y = dc.LogicalToDeviceX(rect.y);
+
+#if GTK_CHECK_VERSION(2, 16, 0)
+    GValue expSize = {0,};
+
+    g_value_init (&expSize, G_TYPE_INT);
+
+    gtk_style_get_style_property( tree->style, GTK_TYPE_TREE_VIEW,
+                                "expander-size", &expSize );
+
+    int expOffset = g_value_get_int(&expSize) / 2 - 1;
+    x += expOffset;
+    y += expOffset;
+
+    // TODO: In GTK3, see
+    // http://developer.gnome.org/gtk3/stable/GtkStyleContext.html#gtk-style-context-get-style-property
+    // http://developer.gnome.org/gtk3/stable/GtkStyleContext.html#gtk-render-expander
+#else
     // VZ: I don't know how to get the size of the expander so as to centre it
     //     in the given rectangle, +2/3 below is just what looks good here...
+    x += 6;
+    y += 3;
+#endif
     gtk_paint_expander
     (
         gtk_widget_get_style(tree),
@@ -266,8 +286,8 @@
         NULL,
         tree,
         "treeview",
-        dc.LogicalToDeviceX(rect.x) + 6 - x_diff,
-        dc.LogicalToDeviceY(rect.y) + 3,
+        x,
+        y,
         flags & wxCONTROL_EXPANDED ? GTK_EXPANDER_EXPANDED
                                    : GTK_EXPANDER_COLLAPSED
     );