diff -ru6 wxWindows.orig/src/msw/wxuxthem.cpp wxWindows/src/msw/wxuxthem.cpp
--- wxWindows.orig/src/msw/wxuxthem.cpp	Sun Apr 13 14:14:18 2003
+++ wxWindows/src/msw/wxuxthem.cpp	Sat Apr 12 09:47:16 2003
@@ -40,12 +40,50 @@
 wxUxThemeEngine* wxUxThemeEngine::wxInitUxThemeEngine()
 {
     wxUxThemeEngine* pThemeEngine = new wxUxThemeEngine ;
     return pThemeEngine ;
 }
 
+#ifdef WXU_USE_WXMODULE
+class wxUxThemeModule : public wxModule
+{
+	DECLARE_DYNAMIC_CLASS(wxUxThemeModule)
+public:
+	virtual void OnExit()
+	{
+		if((wxCanUseInitThemeEngine()) && (g_pThemeEngine))
+		{
+			delete g_pThemeEngine ;
+			g_pThemeEngine = ((wxUxThemeEngine*)NULL) ;
+		}
+	}
+
+	virtual bool OnInit()
+	{
+		if(!wxCanUseInitThemeEngine())
+		{
+			g_pThemeEngine = ((wxUxThemeEngine*)NULL) ;
+			return TRUE ;
+		}
+		g_pThemeEngine = wxUxThemeEngine::wxInitUxThemeEngine() ;
+		if(!g_pThemeEngine)
+		{
+			return FALSE ;
+		}
+		else if(!g_pThemeEngine->IsInitialized())
+		{
+			OnExit() ;
+			return FALSE ;
+		}
+
+		return TRUE ;
+	}
+};
+
+IMPLEMENT_DYNAMIC_CLASS(wxUxThemeModule, wxModule)
+#else
 static void __cdecl wxUxThemeEngineOnExit()
 {
     if(g_pThemeEngine)
     {
         delete g_pThemeEngine ;
         g_pThemeEngine = ((wxUxThemeEngine*)NULL) ;
@@ -74,12 +112,13 @@
         wxUxThemeEngineOnExit() ;
         return FALSE ;
     }
 
     return TRUE ;
 }
+#endif
 
 #define WXU_GET_FN_PTR(hmod, type, pfunc, funcname) if(hmod != ((HINSTANCE)NULL)) { \
     pfunc = (type)::GetProcAddress(hmod, funcname) ; \
     if(pfunc == ((type)NULL)) \
     {	\
         ::FreeLibrary(hmod) ; \

 	  	 
