diff -ru2 -xCVS -x'.#*' -x'*.orig' -x'*.rej' -x'*~' wxWindows-20030501/src/generic/dcpsg.cpp wxWindows/src/generic/dcpsg.cpp
--- wxWindows-20030501/src/generic/dcpsg.cpp	2003-03-17 05:47:54.000000000 -0600
+++ wxWindows/src/generic/dcpsg.cpp	2003-05-02 09:01:29.000000000 -0500
@@ -1882,6 +1882,7 @@
         command += m_printData.GetPrinterCommand();
         command += wxT(" ");
+        command += m_printData.GetPrinterOptions();
+        command += wxT(" ");
         command += m_printData.GetFilename();
-
         wxExecute( command, TRUE );
         wxRemoveFile( m_printData.GetFilename() );

 	  	 
