diff -ru2 -xCVS -x'.#*' -x'*.orig' -x'*.rej' -x'*~' wxWindows-20031125.orig/src/msw/gsocket.c wxWindows-20031125/src/msw/gsocket.c
--- wxWindows-20031125.orig/src/msw/gsocket.c	2003-11-25 11:17:29.000000000 -0800
+++ wxWindows-20031125/src/msw/gsocket.c	2003-11-25 11:46:02.000000000 -0800
@@ -377,4 +377,6 @@
   _GSocket_Enable_Events(sck);
 
+  setsockopt(sck->m_fd, SOL_SOCKET, SO_REUSEADDR, (const char*)&arg, sizeof(u_long));
+
   /* Bind to the local address,
    * retrieve the actual address bound,
diff -ru2 -xCVS -x'.#*' -x'*.orig' -x'*.rej' -x'*~' wxWindows-20031125.orig/src/unix/gsocket.c wxWindows-20031125/src/unix/gsocket.c
--- wxWindows-20031125.orig/src/unix/gsocket.c	2003-11-25 11:17:30.000000000 -0800
+++ wxWindows-20031125/src/unix/gsocket.c	2003-11-25 11:46:00.000000000 -0800
@@ -481,4 +481,6 @@
   _GSocket_Enable_Events(sck);
 
+  setsockopt(sck->m_fd, SOL_SOCKET, SO_REUSEADDR, (const char*)&arg, sizeof(u_long));
+  
   /* Bind to the local address,
    * retrieve the actual address bound,

 	  	 
