diff -ur -x CVS ../wxWidgets/src/mac/carbon/accel.cpp ./src/mac/carbon/accel.cpp
--- ../wxWidgets/src/mac/carbon/accel.cpp	Sun Jun 20 02:02:29 2004
+++ ./src/mac/carbon/accel.cpp	Wed Jan  5 14:50:51 2005
@@ -9,11 +9,12 @@
 // Licence:     wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "accel.h"
 #endif
 
-#include "wx/setup.h"
+#include "wx/wxprec.h"
+
 #include "wx/accel.h"
 #include "wx/string.h"
 
diff -ur -x CVS ../wxWidgets/src/mac/carbon/app.cpp ./src/mac/carbon/app.cpp
--- ../wxWidgets/src/mac/carbon/app.cpp	Wed Oct 20 13:14:43 2004
+++ ./src/mac/carbon/app.cpp	Wed Jan  5 14:46:04 2005
@@ -9,11 +9,11 @@
 // Licence:     wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "app.h"
 #endif
 
-#include "wx/defs.h"
+#include "wx/wxprec.h"
 
 #include "wx/window.h"
 #include "wx/frame.h"
diff -ur -x CVS ../wxWidgets/src/mac/carbon/bitmap.cpp ./src/mac/carbon/bitmap.cpp
--- ../wxWidgets/src/mac/carbon/bitmap.cpp	Wed Jan  5 14:38:51 2005
+++ ./src/mac/carbon/bitmap.cpp	Wed Jan  5 14:46:19 2005
@@ -9,11 +9,11 @@
 // Licence:     wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "bitmap.h"
 #endif
 
-#include "wx/defs.h"
+#include "wx/wxprec.h"
 
 #include "wx/bitmap.h"
 #include "wx/icon.h"
diff -ur -x CVS ../wxWidgets/src/mac/carbon/bmpbuttn.cpp ./src/mac/carbon/bmpbuttn.cpp
--- ../wxWidgets/src/mac/carbon/bmpbuttn.cpp	Sun Jul 11 09:13:21 2004
+++ ./src/mac/carbon/bmpbuttn.cpp	Wed Jan  5 14:51:02 2005
@@ -9,9 +9,11 @@
 // Licence:       wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "bmpbuttn.h"
 #endif
+
+#include "wx/wxprec.h"
 
 #include "wx/window.h"
 #include "wx/bmpbuttn.h"
diff -ur -x CVS ../wxWidgets/src/mac/carbon/brush.cpp ./src/mac/carbon/brush.cpp
--- ../wxWidgets/src/mac/carbon/brush.cpp	Sun May 23 13:52:29 2004
+++ ./src/mac/carbon/brush.cpp	Wed Jan  5 14:51:07 2005
@@ -9,11 +9,12 @@
 // Licence:       wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "brush.h"
 #endif
 
-#include "wx/setup.h"
+#include "wx/wxprec.h"
+
 #include "wx/utils.h"
 #include "wx/brush.h"
 
@@ -226,4 +227,4 @@
 wxMacBrushKind wxBrush::MacGetBrushKind()  const 
 { 
     return (M_BRUSHDATA ? M_BRUSHDATA->m_macBrushKind : kwxMacBrushColour); 
-}
\ No newline at end of file
+}
diff -ur -x CVS ../wxWidgets/src/mac/carbon/button.cpp ./src/mac/carbon/button.cpp
--- ../wxWidgets/src/mac/carbon/button.cpp	Mon Aug 30 07:36:27 2004
+++ ./src/mac/carbon/button.cpp	Wed Jan  5 14:47:09 2005
@@ -9,11 +9,11 @@
 // Licence:       wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "button.h"
 #endif
 
-#include "wx/defs.h"
+#include "wx/wxprec.h"
 
 #include "wx/button.h"
 #include "wx/panel.h"
diff -ur -x CVS ../wxWidgets/src/mac/carbon/checkbox.cpp ./src/mac/carbon/checkbox.cpp
--- ../wxWidgets/src/mac/carbon/checkbox.cpp	Sun Jul 11 09:13:21 2004
+++ ./src/mac/carbon/checkbox.cpp	Wed Jan  5 14:47:25 2005
@@ -9,11 +9,11 @@
 // Licence:       wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "checkbox.h"
 #endif
 
-#include "wx/defs.h"
+#include "wx/wxprec.h"
 
 #include "wx/checkbox.h"
 
diff -ur -x CVS ../wxWidgets/src/mac/carbon/checklst.cpp ./src/mac/carbon/checklst.cpp
--- ../wxWidgets/src/mac/carbon/checklst.cpp	Tue Sep 28 11:36:46 2004
+++ ./src/mac/carbon/checklst.cpp	Wed Jan  5 14:49:33 2005
@@ -13,11 +13,11 @@
 // headers & declarations
 // ============================================================================
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "checklst.h"
 #endif
 
-#include "wx/defs.h"
+#include "wx/wxprec.h"
 
 #if wxUSE_CHECKLISTBOX
 
diff -ur -x CVS ../wxWidgets/src/mac/carbon/choice.cpp ./src/mac/carbon/choice.cpp
--- ../wxWidgets/src/mac/carbon/choice.cpp	Tue Aug 31 06:58:49 2004
+++ ./src/mac/carbon/choice.cpp	Wed Jan  5 14:49:46 2005
@@ -9,11 +9,11 @@
 // Licence:       wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "choice.h"
 #endif
 
-#include "wx/defs.h"
+#include "wx/wxprec.h"
 
 #if wxUSE_CHOICE
 
diff -ur -x CVS ../wxWidgets/src/mac/carbon/clipbrd.cpp ./src/mac/carbon/clipbrd.cpp
--- ../wxWidgets/src/mac/carbon/clipbrd.cpp	Sun May 23 13:52:30 2004
+++ ./src/mac/carbon/clipbrd.cpp	Wed Jan  5 14:51:21 2005
@@ -9,9 +9,11 @@
 // Licence:     wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "clipbrd.h"
 #endif
+
+#include "wx/wxprec.h"
 
 #include "wx/app.h"
 #include "wx/frame.h"
diff -ur -x CVS ../wxWidgets/src/mac/carbon/colordlg.cpp ./src/mac/carbon/colordlg.cpp
--- ../wxWidgets/src/mac/carbon/colordlg.cpp	Wed Nov 24 14:50:12 2004
+++ ./src/mac/carbon/colordlg.cpp	Wed Jan  5 14:50:39 2005
@@ -10,9 +10,11 @@
 // Licence:       wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "colordlg.h"
 #endif
+
+#include "wx/wxprec.h"
 
 #include "wx/mac/colordlg.h"
 #include "wx/fontdlg.h"
diff -ur -x CVS ../wxWidgets/src/mac/carbon/colour.cpp ./src/mac/carbon/colour.cpp
--- ../wxWidgets/src/mac/carbon/colour.cpp	Sun May 23 13:52:30 2004
+++ ./src/mac/carbon/colour.cpp	Wed Jan  5 14:51:47 2005
@@ -9,9 +9,11 @@
 // Licence:       wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "colour.h"
 #endif
+
+#include "wx/wxprec.h"
 
 #include "wx/gdicmn.h"
 #include "wx/colour.h"
diff -ur -x CVS ../wxWidgets/src/mac/carbon/combobox.cpp ./src/mac/carbon/combobox.cpp
--- ../wxWidgets/src/mac/carbon/combobox.cpp	Mon Jan  3 11:48:35 2005
+++ ./src/mac/carbon/combobox.cpp	Wed Jan  5 14:52:04 2005
@@ -9,9 +9,11 @@
 // Licence:     wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "combobox.h"
 #endif
+
+#include "wx/wxprec.h"
 
 #include "wx/combobox.h"
 #include "wx/button.h"
diff -ur -x CVS ../wxWidgets/src/mac/carbon/combobxc.cpp ./src/mac/carbon/combobxc.cpp
--- ../wxWidgets/src/mac/carbon/combobxc.cpp	Mon Jan  3 11:48:35 2005
+++ ./src/mac/carbon/combobxc.cpp	Wed Jan  5 14:52:20 2005
@@ -9,9 +9,11 @@
 // Licence:     wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "combobox.h"
 #endif
+
+#include "wx/wxprec.h"
 
 #include "wx/combobox.h"
 #include "wx/button.h"
diff -ur -x CVS ../wxWidgets/src/mac/carbon/control.cpp ./src/mac/carbon/control.cpp
--- ../wxWidgets/src/mac/carbon/control.cpp	Sun Jul 11 09:14:53 2004
+++ ./src/mac/carbon/control.cpp	Wed Jan  5 14:52:34 2005
@@ -9,11 +9,11 @@
 // Licence:     wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "control.h"
 #endif
 
-#include "wx/defs.h"
+#include "wx/wxprec.h"
 
 #include "wx/control.h"
 #include "wx/panel.h"
diff -ur -x CVS ../wxWidgets/src/mac/carbon/cursor.cpp ./src/mac/carbon/cursor.cpp
--- ../wxWidgets/src/mac/carbon/cursor.cpp	Wed Sep 22 18:37:36 2004
+++ ./src/mac/carbon/cursor.cpp	Wed Jan  5 14:52:47 2005
@@ -9,11 +9,11 @@
 // Licence:       wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "cursor.h"
 #endif
 
-#include "wx/defs.h"
+#include "wx/wxprec.h"
 
 #include "wx/app.h"
 #include "wx/cursor.h"
diff -ur -x CVS ../wxWidgets/src/mac/carbon/data.cpp ./src/mac/carbon/data.cpp
--- ../wxWidgets/src/mac/carbon/data.cpp	Sun May 23 13:52:30 2004
+++ ./src/mac/carbon/data.cpp	Wed Jan  5 14:53:04 2005
@@ -9,6 +9,8 @@
 // Licence:     wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
+#include "wx/wxprec.h"
+
 #include "wx/event.h"
 
 #if wxUSE_SHARED_LIBRARY
diff -ur -x CVS ../wxWidgets/src/mac/carbon/dataobj.cpp ./src/mac/carbon/dataobj.cpp
--- ../wxWidgets/src/mac/carbon/dataobj.cpp	Sun May 23 13:52:30 2004
+++ ./src/mac/carbon/dataobj.cpp	Wed Jan  5 14:44:38 2005
@@ -17,7 +17,7 @@
 // headers
 // ----------------------------------------------------------------------------
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
   #pragma implementation "dataobj.h"
 #endif
 
@@ -27,7 +27,6 @@
 #ifndef WX_PRECOMP
 #include "wx/intl.h"
 #endif
-#include "wx/defs.h"
 
 #include "wx/log.h"
 #include "wx/dataobj.h"
diff -ur -x CVS ../wxWidgets/src/mac/carbon/dc.cpp ./src/mac/carbon/dc.cpp
--- ../wxWidgets/src/mac/carbon/dc.cpp	Mon Jan  3 11:48:35 2005
+++ ./src/mac/carbon/dc.cpp	Wed Jan  5 14:54:23 2005
@@ -9,9 +9,11 @@
 // Licence:       wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "dc.h"
 #endif
+
+#include "wx/wxprec.h"
 
 #include "wx/dc.h"
 #include "wx/app.h"
diff -ur -x CVS ../wxWidgets/src/mac/carbon/dcclient.cpp ./src/mac/carbon/dcclient.cpp
--- ../wxWidgets/src/mac/carbon/dcclient.cpp	Mon Jan  3 11:48:35 2005
+++ ./src/mac/carbon/dcclient.cpp	Wed Jan  5 14:54:39 2005
@@ -9,9 +9,11 @@
 // Licence:       wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "dcclient.h"
 #endif
+
+#include "wx/wxprec.h"
 
 #include "wx/dcclient.h"
 #include "wx/dcmemory.h"
diff -ur -x CVS ../wxWidgets/src/mac/carbon/dcmemory.cpp ./src/mac/carbon/dcmemory.cpp
--- ../wxWidgets/src/mac/carbon/dcmemory.cpp	Sun May 23 13:52:30 2004
+++ ./src/mac/carbon/dcmemory.cpp	Wed Jan  5 14:54:55 2005
@@ -9,9 +9,11 @@
 // Licence:       wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "dcmemory.h"
 #endif
+
+#include "wx/wxprec.h"
 
 #include "wx/dcmemory.h"
 #include "wx/mac/private.h"
diff -ur -x CVS ../wxWidgets/src/mac/carbon/dcprint.cpp ./src/mac/carbon/dcprint.cpp
--- ../wxWidgets/src/mac/carbon/dcprint.cpp	Mon Jan  3 11:48:35 2005
+++ ./src/mac/carbon/dcprint.cpp	Wed Jan  5 14:44:38 2005
@@ -9,7 +9,7 @@
 // Licence:       wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "dcprint.h"
 #endif
 
diff -ur -x CVS ../wxWidgets/src/mac/carbon/dcscreen.cpp ./src/mac/carbon/dcscreen.cpp
--- ../wxWidgets/src/mac/carbon/dcscreen.cpp	Sun May 23 13:52:30 2004
+++ ./src/mac/carbon/dcscreen.cpp	Wed Jan  5 14:55:12 2005
@@ -9,9 +9,11 @@
 // Licence:       wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "dcscreen.h"
 #endif
+
+#include "wx/wxprec.h"
 
 #include "wx/dcscreen.h"
 #include "wx/mac/uma.h"
diff -ur -x CVS ../wxWidgets/src/mac/carbon/dialog.cpp ./src/mac/carbon/dialog.cpp
--- ../wxWidgets/src/mac/carbon/dialog.cpp	Mon Jan  3 11:48:35 2005
+++ ./src/mac/carbon/dialog.cpp	Wed Jan  5 14:55:36 2005
@@ -9,9 +9,11 @@
 // Licence:       wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "dialog.h"
 #endif
+
+#include "wx/wxprec.h"
 
 #include "wx/dialog.h"
 #include "wx/utils.h"
diff -ur -x CVS ../wxWidgets/src/mac/carbon/dirdlg.cpp ./src/mac/carbon/dirdlg.cpp
--- ../wxWidgets/src/mac/carbon/dirdlg.cpp	Tue Aug 24 12:38:09 2004
+++ ./src/mac/carbon/dirdlg.cpp	Wed Jan  5 14:55:49 2005
@@ -9,11 +9,12 @@
 // Licence:       wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "dirdlg.h"
 #endif
 
-#include "wx/defs.h"
+#include "wx/wxprec.h"
+
 #include "wx/utils.h"
 #include "wx/dialog.h"
 #include "wx/dirdlg.h"
diff -ur -x CVS ../wxWidgets/src/mac/carbon/dirmac.cpp ./src/mac/carbon/dirmac.cpp
--- ../wxWidgets/src/mac/carbon/dirmac.cpp	Tue Sep 28 11:36:46 2004
+++ ./src/mac/carbon/dirmac.cpp	Wed Jan  5 14:44:38 2005
@@ -17,7 +17,7 @@
 // headers
 // ----------------------------------------------------------------------------
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
     #pragma implementation "dir.h"
 #endif
 
diff -ur -x CVS ../wxWidgets/src/mac/carbon/display.cpp ./src/mac/carbon/display.cpp
--- ../wxWidgets/src/mac/carbon/display.cpp	Tue Sep 28 11:36:46 2004
+++ ./src/mac/carbon/display.cpp	Wed Jan  5 14:44:38 2005
@@ -9,7 +9,7 @@
 // Licence:     wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
     #pragma implementation "display.h"
 #endif
 
diff -ur -x CVS ../wxWidgets/src/mac/carbon/dnd.cpp ./src/mac/carbon/dnd.cpp
--- ../wxWidgets/src/mac/carbon/dnd.cpp	Tue Sep 28 11:36:46 2004
+++ ./src/mac/carbon/dnd.cpp	Wed Jan  5 14:56:08 2005
@@ -9,11 +9,11 @@
 // Licence:     wxWindows licence
 ///////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "dnd.h"
 #endif
 
-#include "wx/defs.h"
+#include "wx/wxprec.h"
 
 #if wxUSE_DRAG_AND_DROP
 
diff -ur -x CVS ../wxWidgets/src/mac/carbon/drawer.cpp ./src/mac/carbon/drawer.cpp
--- ../wxWidgets/src/mac/carbon/drawer.cpp	Sun Sep 26 10:41:26 2004
+++ ./src/mac/carbon/drawer.cpp	Wed Jan  5 14:56:20 2005
@@ -12,11 +12,12 @@
 // Licence:     wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "control.h"
 #endif
 
-#include "wx/defs.h"
+#include "wx/wxprec.h"
+
 #include "wx/mac/private.h"
 
 #if defined( __WXMAC__ ) && TARGET_API_MAC_OSX && ( MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_2 )
diff -ur -x CVS ../wxWidgets/src/mac/carbon/filedlg.cpp ./src/mac/carbon/filedlg.cpp
--- ../wxWidgets/src/mac/carbon/filedlg.cpp	Tue Sep 28 11:36:46 2004
+++ ./src/mac/carbon/filedlg.cpp	Wed Jan  5 14:56:32 2005
@@ -9,11 +9,12 @@
 // Licence:     wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "filedlg.h"
 #endif
 
-#include "wx/defs.h"
+#include "wx/wxprec.h"
+
 #include "wx/app.h"
 #include "wx/utils.h"
 #include "wx/dialog.h"
diff -ur -x CVS ../wxWidgets/src/mac/carbon/font.cpp ./src/mac/carbon/font.cpp
--- ../wxWidgets/src/mac/carbon/font.cpp	Wed Nov 24 14:50:12 2004
+++ ./src/mac/carbon/font.cpp	Wed Jan  5 14:56:43 2005
@@ -9,11 +9,12 @@
 // Licence:     wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "font.h"
 #endif
 
-#include "wx/defs.h"
+#include "wx/wxprec.h"
+
 #include "wx/string.h"
 #include "wx/font.h"
 #include "wx/fontutil.h"
diff -ur -x CVS ../wxWidgets/src/mac/carbon/fontdlg.cpp ./src/mac/carbon/fontdlg.cpp
--- ../wxWidgets/src/mac/carbon/fontdlg.cpp	Wed Nov 24 14:50:12 2004
+++ ./src/mac/carbon/fontdlg.cpp	Wed Jan  5 14:44:38 2005
@@ -17,7 +17,7 @@
 // headers
 // ---------------------------------------------------------------------------
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "fontdlg.h"
 #endif
 
@@ -93,4 +93,4 @@
     return wxID_CANCEL;
 }
 
-#endif // 10.2+
\ No newline at end of file
+#endif // 10.2+
diff -ur -x CVS ../wxWidgets/src/mac/carbon/fontenum.cpp ./src/mac/carbon/fontenum.cpp
--- ../wxWidgets/src/mac/carbon/fontenum.cpp	Sun May 23 13:52:31 2004
+++ ./src/mac/carbon/fontenum.cpp	Wed Jan  5 14:44:38 2005
@@ -17,7 +17,7 @@
 // headers
 // ----------------------------------------------------------------------------
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
     #pragma implementation "fontenum.h"
 #endif
 
diff -ur -x CVS ../wxWidgets/src/mac/carbon/fontutil.cpp ./src/mac/carbon/fontutil.cpp
--- ../wxWidgets/src/mac/carbon/fontutil.cpp	Sun May 23 13:52:31 2004
+++ ./src/mac/carbon/fontutil.cpp	Wed Jan  5 14:44:38 2005
@@ -17,7 +17,7 @@
 // headers
 // ----------------------------------------------------------------------------
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
     #pragma implementation "fontutil.h"
 #endif
 
diff -ur -x CVS ../wxWidgets/src/mac/carbon/frame.cpp ./src/mac/carbon/frame.cpp
--- ../wxWidgets/src/mac/carbon/frame.cpp	Wed Oct 20 13:14:44 2004
+++ ./src/mac/carbon/frame.cpp	Wed Jan  5 14:57:11 2005
@@ -9,9 +9,11 @@
 // Licence:     wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "frame.h"
 #endif
+
+#include "wx/wxprec.h"
 
 #include "wx/frame.h"
 #include "wx/statusbr.h"
diff -ur -x CVS ../wxWidgets/src/mac/carbon/gauge.cpp ./src/mac/carbon/gauge.cpp
--- ../wxWidgets/src/mac/carbon/gauge.cpp	Tue Nov  9 12:55:49 2004
+++ ./src/mac/carbon/gauge.cpp	Wed Jan  5 14:57:51 2005
@@ -9,13 +9,15 @@
 // Licence:       wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "gauge.h"
 #endif
 
-#include "wx/gauge.h"
+#include "wx/wxprec.h"
 
 #if wxUSE_GAUGE 
+
+#include "wx/gauge.h"
 
 #if !USE_SHARED_LIBRARY
 IMPLEMENT_DYNAMIC_CLASS(wxGauge, wxControl)
diff -ur -x CVS ../wxWidgets/src/mac/carbon/gdiobj.cpp ./src/mac/carbon/gdiobj.cpp
--- ../wxWidgets/src/mac/carbon/gdiobj.cpp	Sun May 23 13:52:31 2004
+++ ./src/mac/carbon/gdiobj.cpp	Wed Jan  5 14:58:13 2005
@@ -9,9 +9,11 @@
 // Licence:       wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "gdiobj.h"
 #endif
+
+#include "wx/wxprec.h"
 
 #include "wx/gdiobj.h"
 
diff -ur -x CVS ../wxWidgets/src/mac/carbon/glcanvas.cpp ./src/mac/carbon/glcanvas.cpp
--- ../wxWidgets/src/mac/carbon/glcanvas.cpp	Sun Jul 11 09:14:53 2004
+++ ./src/mac/carbon/glcanvas.cpp	Wed Jan  5 14:44:38 2005
@@ -9,7 +9,7 @@
 // Licence:       wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "glcanvas.h"
 #endif
 
@@ -18,8 +18,6 @@
 #if defined(__BORLANDC__)
 #pragma hdrstop
 #endif
-
-#include "wx/setup.h"
 
 #if wxUSE_GLCANVAS
 
diff -ur -x CVS ../wxWidgets/src/mac/carbon/helpxxxx.cpp ./src/mac/carbon/helpxxxx.cpp
--- ../wxWidgets/src/mac/carbon/helpxxxx.cpp	Sun May 23 13:52:31 2004
+++ ./src/mac/carbon/helpxxxx.cpp	Wed Jan  5 14:59:21 2005
@@ -9,9 +9,11 @@
 // Licence:       wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "helpxxxx.h"
 #endif
+
+#include "wx/wxprec.h"
 
 #include "wx/stubs/helpxxxx.h"
 
diff -ur -x CVS ../wxWidgets/src/mac/carbon/hid.cpp ./src/mac/carbon/hid.cpp
--- ../wxWidgets/src/mac/carbon/hid.cpp	Tue Sep 28 11:36:46 2004
+++ ./src/mac/carbon/hid.cpp	Wed Jan  5 14:44:38 2005
@@ -17,7 +17,7 @@
 // headers
 // ---------------------------------------------------------------------------
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "hid.h"
 #endif
 
@@ -27,11 +27,6 @@
 #ifdef __BORLANDC__
     #pragma hdrstop
 #endif
-
-#ifndef WX_PRECOMP
-    #include "wx/defs.h"
-#endif
-
 
 //DARWIN _ONLY_
 #ifdef __DARWIN__
diff -ur -x CVS ../wxWidgets/src/mac/carbon/icon.cpp ./src/mac/carbon/icon.cpp
--- ../wxWidgets/src/mac/carbon/icon.cpp	Wed Jan  5 14:38:51 2005
+++ ./src/mac/carbon/icon.cpp	Wed Jan  5 14:59:44 2005
@@ -9,9 +9,11 @@
 // Licence:       wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "icon.h"
 #endif
+
+#include "wx/wxprec.h"
 
 #include "wx/icon.h"
 
diff -ur -x CVS ../wxWidgets/src/mac/carbon/joystick.cpp ./src/mac/carbon/joystick.cpp
--- ../wxWidgets/src/mac/carbon/joystick.cpp	Sun May 23 13:52:31 2004
+++ ./src/mac/carbon/joystick.cpp	Wed Jan  5 15:13:30 2005
@@ -9,15 +9,15 @@
 // Licence:       wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "joystick.h"
 #endif
 
-#include "wx/setup.h"
-
-#include "wx/joystick.h"
+#include "wx/wxprec.h"
 
 #if wxUSE_JOYSTICK
+
+#include "wx/joystick.h"
 
 IMPLEMENT_DYNAMIC_CLASS(wxJoystick, wxObject)
 
diff -ur -x CVS ../wxWidgets/src/mac/carbon/listbox.cpp ./src/mac/carbon/listbox.cpp
--- ../wxWidgets/src/mac/carbon/listbox.cpp	Wed Nov 24 14:50:12 2004
+++ ./src/mac/carbon/listbox.cpp	Wed Jan  5 15:13:47 2005
@@ -9,9 +9,11 @@
 // Licence:     wxWindows licence
 ///////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "listbox.h"
 #endif
+
+#include "wx/wxprec.h"
 
 #include "wx/app.h"
 #include "wx/listbox.h"
diff -ur -x CVS ../wxWidgets/src/mac/carbon/macnotfy.cpp ./src/mac/carbon/macnotfy.cpp
--- ../wxWidgets/src/mac/carbon/macnotfy.cpp	Fri May 14 02:19:09 2004
+++ ./src/mac/carbon/macnotfy.cpp	Wed Jan  5 15:14:12 2005
@@ -7,6 +7,8 @@
  * -------------------------------------------------------------------------
  */
 
+#include "wx/wxprec.h"
+
 #include "wx/wx.h"
 
 #include "wx/mac/private.h"
diff -ur -x CVS ../wxWidgets/src/mac/carbon/mdi.cpp ./src/mac/carbon/mdi.cpp
--- ../wxWidgets/src/mac/carbon/mdi.cpp	Wed Nov 24 14:50:12 2004
+++ ./src/mac/carbon/mdi.cpp	Wed Jan  5 15:14:31 2005
@@ -9,9 +9,11 @@
 // Licence:       wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "mdi.h"
 #endif
+
+#include "wx/wxprec.h"
 
 #include "wx/mdi.h"
 #include "wx/menu.h"
diff -ur -x CVS ../wxWidgets/src/mac/carbon/menu.cpp ./src/mac/carbon/menu.cpp
--- ../wxWidgets/src/mac/carbon/menu.cpp	Wed Nov 24 14:50:12 2004
+++ ./src/mac/carbon/menu.cpp	Wed Jan  5 15:14:55 2005
@@ -9,7 +9,7 @@
 // Licence:       wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "menu.h"
 #pragma implementation "menuitem.h"
 #endif
@@ -20,6 +20,8 @@
 
 // wxWidgets headers
 // -----------------
+
+#include "wx/wxprec.h"
 
 #include "wx/app.h"
 #include "wx/menu.h"
diff -ur -x CVS ../wxWidgets/src/mac/carbon/menuitem.cpp ./src/mac/carbon/menuitem.cpp
--- ../wxWidgets/src/mac/carbon/menuitem.cpp	Wed Jan  5 14:38:51 2005
+++ ./src/mac/carbon/menuitem.cpp	Wed Jan  5 15:15:09 2005
@@ -13,6 +13,8 @@
 // headers & declarations
 // ============================================================================
 
+#include "wx/wxprec.h"
+
 #include "wx/app.h"
 #include "wx/menu.h"
 #include "wx/menuitem.h"
diff -ur -x CVS ../wxWidgets/src/mac/carbon/metafile.cpp ./src/mac/carbon/metafile.cpp
--- ../wxWidgets/src/mac/carbon/metafile.cpp	Sun May 23 13:52:31 2004
+++ ./src/mac/carbon/metafile.cpp	Wed Jan  5 14:44:38 2005
@@ -9,7 +9,7 @@
 // Licence:       wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "metafile.h"
 #endif
 
@@ -18,10 +18,6 @@
 
 #ifdef __BORLANDC__
 #pragma hdrstop
-#endif
-
-#ifndef WX_PRECOMP
-#include "wx/setup.h"
 #endif
 
 #if wxUSE_METAFILE
diff -ur -x CVS ../wxWidgets/src/mac/carbon/mimetmac.cpp ./src/mac/carbon/mimetmac.cpp
--- ../wxWidgets/src/mac/carbon/mimetmac.cpp	Sun May 23 13:52:31 2004
+++ ./src/mac/carbon/mimetmac.cpp	Wed Jan  5 14:44:38 2005
@@ -9,7 +9,7 @@
 // Licence:     wxWindows licence (part of wxExtra library)
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef    __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "mimetype.h"
 #endif
 
@@ -18,10 +18,6 @@
 
 #ifdef __BORLANDC__
   #pragma hdrstop
-#endif
-
-#ifndef WX_PRECOMP
-  #include "wx/defs.h"
 #endif
 
 #ifndef WX_PRECOMP
diff -ur -x CVS ../wxWidgets/src/mac/carbon/minifram.cpp ./src/mac/carbon/minifram.cpp
--- ../wxWidgets/src/mac/carbon/minifram.cpp	Sun May 23 13:52:31 2004
+++ ./src/mac/carbon/minifram.cpp	Wed Jan  5 15:15:34 2005
@@ -9,9 +9,11 @@
 // Licence:       wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "minifram.h"
 #endif
+
+#include "wx/wxprec.h"
 
 #include "wx/minifram.h"
 
diff -ur -x CVS ../wxWidgets/src/mac/carbon/mpthread.cpp ./src/mac/carbon/mpthread.cpp
--- ../wxWidgets/src/mac/carbon/mpthread.cpp	Mon Jan  3 11:48:35 2005
+++ ./src/mac/carbon/mpthread.cpp	Wed Jan  5 14:44:39 2005
@@ -10,7 +10,7 @@
 // Licence:	wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "thread.h"
 #endif
 
diff -ur -x CVS ../wxWidgets/src/mac/carbon/msgdlg.cpp ./src/mac/carbon/msgdlg.cpp
--- ../wxWidgets/src/mac/carbon/msgdlg.cpp	Sat Sep 11 01:25:17 2004
+++ ./src/mac/carbon/msgdlg.cpp	Wed Jan  5 15:15:58 2005
@@ -9,9 +9,11 @@
 // Licence:       wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "msgdlg.h"
 #endif
+
+#include "wx/wxprec.h"
 
 #include "wx/app.h"
 #include "wx/msgdlg.h"
diff -ur -x CVS ../wxWidgets/src/mac/carbon/notebmac.cpp ./src/mac/carbon/notebmac.cpp
--- ../wxWidgets/src/mac/carbon/notebmac.cpp	Wed Oct 20 13:14:45 2004
+++ ./src/mac/carbon/notebmac.cpp	Wed Jan  5 15:16:18 2005
@@ -9,7 +9,7 @@
 // Licence:     wxWindows licence
 ///////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "notebook.h"
 #endif
 
@@ -20,6 +20,8 @@
 // ----------------------------------------------------------------------------
 // headers
 // ----------------------------------------------------------------------------
+#include "wx/wxprec.h"
+
 #include "wx/app.h"
 #include "wx/string.h"
 #include "wx/log.h"
diff -ur -x CVS ../wxWidgets/src/mac/carbon/palette.cpp ./src/mac/carbon/palette.cpp
--- ../wxWidgets/src/mac/carbon/palette.cpp	Sun May 23 13:52:32 2004
+++ ./src/mac/carbon/palette.cpp	Wed Jan  5 15:16:30 2005
@@ -9,11 +9,11 @@
 // Licence:       wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "palette.h"
 #endif
 
-#include "wx/defs.h"
+#include "wx/wxprec.h"
 
 #if wxUSE_PALETTE
 
diff -ur -x CVS ../wxWidgets/src/mac/carbon/pen.cpp ./src/mac/carbon/pen.cpp
--- ../wxWidgets/src/mac/carbon/pen.cpp	Sun May 23 13:52:32 2004
+++ ./src/mac/carbon/pen.cpp	Wed Jan  5 15:16:45 2005
@@ -9,11 +9,12 @@
 // Licence:       wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "pen.h"
 #endif
 
-#include "wx/setup.h"
+#include "wx/wxprec.h"
+
 #include "wx/utils.h"
 #include "wx/pen.h"
 
diff -ur -x CVS ../wxWidgets/src/mac/carbon/pnghand.cpp ./src/mac/carbon/pnghand.cpp
--- ../wxWidgets/src/mac/carbon/pnghand.cpp	Fri Nov  5 10:23:42 2004
+++ ./src/mac/carbon/pnghand.cpp	Wed Jan  5 14:44:39 2005
@@ -9,7 +9,7 @@
 // Licence:     wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #  pragma implementation "pngread.h"
 #  pragma implementation "pnghand.h"
 #endif
diff -ur -x CVS ../wxWidgets/src/mac/carbon/printdlg.cpp ./src/mac/carbon/printdlg.cpp
--- ../wxWidgets/src/mac/carbon/printdlg.cpp	Wed Nov 24 14:50:12 2004
+++ ./src/mac/carbon/printdlg.cpp	Wed Jan  5 15:17:10 2005
@@ -9,9 +9,11 @@
 // Licence:       wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "printdlg.h"
 #endif
+
+#include "wx/wxprec.h"
 
 #include "wx/object.h"
 #include "wx/printdlg.h"
diff -ur -x CVS ../wxWidgets/src/mac/carbon/printmac.cpp ./src/mac/carbon/printmac.cpp
--- ../wxWidgets/src/mac/carbon/printmac.cpp	Mon Jan  3 11:48:35 2005
+++ ./src/mac/carbon/printmac.cpp	Wed Jan  5 14:44:39 2005
@@ -9,7 +9,7 @@
 // Licence:       wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "printwin.h"
 #endif
 
@@ -19,8 +19,6 @@
 #ifdef __BORLANDC__
 #pragma hdrstop
 #endif
-
-#include "wx/defs.h"
 
 #ifndef WX_PRECOMP
 #include "wx/utils.h"
diff -ur -x CVS ../wxWidgets/src/mac/carbon/radiobox.cpp ./src/mac/carbon/radiobox.cpp
--- ../wxWidgets/src/mac/carbon/radiobox.cpp	Sun Jul 11 09:17:34 2004
+++ ./src/mac/carbon/radiobox.cpp	Wed Jan  5 15:17:36 2005
@@ -9,7 +9,7 @@
 // Licence:       wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "radioboxbase.h"
 #pragma implementation "radiobox.h"
 #endif
@@ -18,7 +18,8 @@
 //         headers
 //-------------------------------------------------------------------------------------
 
-#include "wx/defs.h"
+#include "wx/wxprec.h"
+
 #include "wx/arrstr.h"
 
 #include "wx/radiobox.h"
diff -ur -x CVS ../wxWidgets/src/mac/carbon/radiobut.cpp ./src/mac/carbon/radiobut.cpp
--- ../wxWidgets/src/mac/carbon/radiobut.cpp	Sun Jul 11 09:17:35 2004
+++ ./src/mac/carbon/radiobut.cpp	Wed Jan  5 15:17:49 2005
@@ -9,11 +9,11 @@
 // Licence:       wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "radiobut.h"
 #endif
 
-#include "wx/defs.h"
+#include "wx/wxprec.h"
 
 #include "wx/radiobut.h"
 
diff -ur -x CVS ../wxWidgets/src/mac/carbon/region.cpp ./src/mac/carbon/region.cpp
--- ../wxWidgets/src/mac/carbon/region.cpp	Wed Oct 27 18:59:12 2004
+++ ./src/mac/carbon/region.cpp	Wed Jan  5 15:18:06 2005
@@ -8,9 +8,11 @@
 // Licence:   wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "region.h"
 #endif
+
+#include "wx/wxprec.h"
 
 #include "wx/region.h"
 #include "wx/gdicmn.h"
diff -ur -x CVS ../wxWidgets/src/mac/carbon/scrolbar.cpp ./src/mac/carbon/scrolbar.cpp
--- ../wxWidgets/src/mac/carbon/scrolbar.cpp	Sun Jul 11 09:17:35 2004
+++ ./src/mac/carbon/scrolbar.cpp	Wed Jan  5 15:18:36 2005
@@ -9,11 +9,11 @@
 // Licence:       wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "scrolbar.h"
 #endif
 
-#include "wx/defs.h"
+#include "wx/wxprec.h"
 
 #ifndef WX_PRECOMP
     #include "wx/intl.h"
diff -ur -x CVS ../wxWidgets/src/mac/carbon/settings.cpp ./src/mac/carbon/settings.cpp
--- ../wxWidgets/src/mac/carbon/settings.cpp	Wed Oct 20 13:14:45 2004
+++ ./src/mac/carbon/settings.cpp	Wed Jan  5 15:18:50 2005
@@ -9,9 +9,11 @@
 // Licence:     wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "settings.h"
 #endif
+
+#include "wx/wxprec.h"
 
 #include "wx/settings.h"
 #include "wx/gdicmn.h"
diff -ur -x CVS ../wxWidgets/src/mac/carbon/slider.cpp ./src/mac/carbon/slider.cpp
--- ../wxWidgets/src/mac/carbon/slider.cpp	Wed Oct 20 13:14:45 2004
+++ ./src/mac/carbon/slider.cpp	Wed Jan  5 15:19:03 2005
@@ -9,11 +9,11 @@
 // Licence:       wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "slider.h"
 #endif
 
-#include "wx/defs.h"
+#include "wx/wxprec.h"
 
 #if wxUSE_SLIDER
 
diff -ur -x CVS ../wxWidgets/src/mac/carbon/sound.cpp ./src/mac/carbon/sound.cpp
--- ../wxWidgets/src/mac/carbon/sound.cpp	Wed Oct 20 13:14:45 2004
+++ ./src/mac/carbon/sound.cpp	Wed Jan  5 15:19:16 2005
@@ -9,9 +9,11 @@
 // Licence:       wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "sound.h"
 #endif
+
+#include "wx/wxprec.h"
 
 #include "wx/object.h"
 #include "wx/string.h"
diff -ur -x CVS ../wxWidgets/src/mac/carbon/spinbutt.cpp ./src/mac/carbon/spinbutt.cpp
--- ../wxWidgets/src/mac/carbon/spinbutt.cpp	Mon Jan  3 11:48:35 2005
+++ ./src/mac/carbon/spinbutt.cpp	Wed Jan  5 15:19:28 2005
@@ -9,12 +9,12 @@
 // Licence:       wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "spinbutt.h"
 #pragma implementation "spinbuttbase.h"
 #endif
 
-#include "wx/defs.h"
+#include "wx/wxprec.h"
 
 #if wxUSE_SPINBTN
 
diff -ur -x CVS ../wxWidgets/src/mac/carbon/spinctrl.cpp ./src/mac/carbon/spinctrl.cpp
--- ../wxWidgets/src/mac/carbon/spinctrl.cpp	Mon Jan  3 11:48:35 2005
+++ ./src/mac/carbon/spinctrl.cpp	Wed Jan  5 15:19:43 2005
@@ -12,7 +12,7 @@
 #pragma implementation "spinctrl.h"
 #endif
 
-#include "wx/defs.h"
+#include "wx/wxprec.h"
 
 #if wxUSE_SPINCTRL
 
diff -ur -x CVS ../wxWidgets/src/mac/carbon/statbmp.cpp ./src/mac/carbon/statbmp.cpp
--- ../wxWidgets/src/mac/carbon/statbmp.cpp	Wed Oct 20 13:14:45 2004
+++ ./src/mac/carbon/statbmp.cpp	Wed Jan  5 15:19:57 2005
@@ -9,11 +9,11 @@
 // Licence:       wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
   #pragma implementation "statbmp.h"
 #endif
 
-#include "wx/defs.h"
+#include "wx/wxprec.h"
 
 #include "wx/statbmp.h"
 #include "wx/dcclient.h"
diff -ur -x CVS ../wxWidgets/src/mac/carbon/statbox.cpp ./src/mac/carbon/statbox.cpp
--- ../wxWidgets/src/mac/carbon/statbox.cpp	Sun Jul 11 09:18:21 2004
+++ ./src/mac/carbon/statbox.cpp	Wed Jan  5 15:20:07 2005
@@ -9,11 +9,11 @@
 // Licence:       wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "statbox.h"
 #endif
 
-#include "wx/defs.h"
+#include "wx/wxprec.h"
 
 #include "wx/statbox.h"
 #include "wx/mac/uma.h"
diff -ur -x CVS ../wxWidgets/src/mac/carbon/statbrma.cpp ./src/mac/carbon/statbrma.cpp
--- ../wxWidgets/src/mac/carbon/statbrma.cpp	Tue Jul 20 07:38:18 2004
+++ ./src/mac/carbon/statbrma.cpp	Wed Jan  5 15:20:20 2005
@@ -9,13 +9,15 @@
 // Licence:     wxWindows licence
 ///////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "statbrma.h"
 #endif
 
 // ----------------------------------------------------------------------------
 // headers
 // ----------------------------------------------------------------------------
+
+#include "wx/wxprec.h"
 
 #include "wx/statusbr.h"
 #include "wx/dc.h"
diff -ur -x CVS ../wxWidgets/src/mac/carbon/statline.cpp ./src/mac/carbon/statline.cpp
--- ../wxWidgets/src/mac/carbon/statline.cpp	Sun May 23 13:52:33 2004
+++ ./src/mac/carbon/statline.cpp	Wed Jan  5 14:44:39 2005
@@ -16,7 +16,7 @@
 // headers
 // ----------------------------------------------------------------------------
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
     #pragma implementation "statline.h"
 #endif
 
diff -ur -x CVS ../wxWidgets/src/mac/carbon/statlmac.cpp ./src/mac/carbon/statlmac.cpp
--- ../wxWidgets/src/mac/carbon/statlmac.cpp	Sun Jul 11 09:18:21 2004
+++ ./src/mac/carbon/statlmac.cpp	Wed Jan  5 14:44:39 2005
@@ -16,7 +16,7 @@
 // headers
 // ----------------------------------------------------------------------------
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
     #pragma implementation "statline.h"
 #endif
 
diff -ur -x CVS ../wxWidgets/src/mac/carbon/stattext.cpp ./src/mac/carbon/stattext.cpp
--- ../wxWidgets/src/mac/carbon/stattext.cpp	Wed Aug  4 11:20:38 2004
+++ ./src/mac/carbon/stattext.cpp	Wed Jan  5 15:20:39 2005
@@ -9,9 +9,11 @@
 // Licence:     wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "stattext.h"
 #endif
+
+#include "wx/wxprec.h"
 
 #include "wx/app.h"
 #include "wx/stattext.h"
diff -ur -x CVS ../wxWidgets/src/mac/carbon/tabctrl.cpp ./src/mac/carbon/tabctrl.cpp
--- ../wxWidgets/src/mac/carbon/tabctrl.cpp	Tue Jul 13 11:32:22 2004
+++ ./src/mac/carbon/tabctrl.cpp	Wed Jan  5 15:20:52 2005
@@ -9,11 +9,11 @@
 // Licence:       wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "tabctrl.h"
 #endif
 
-#include "wx/defs.h"
+#include "wx/wxprec.h"
 
 #include "wx/control.h"
 #include "wx/tabctrl.h"
diff -ur -x CVS ../wxWidgets/src/mac/carbon/taskbar.cpp ./src/mac/carbon/taskbar.cpp
--- ../wxWidgets/src/mac/carbon/taskbar.cpp	Wed Oct 20 13:14:45 2004
+++ ./src/mac/carbon/taskbar.cpp	Wed Jan  5 14:44:39 2005
@@ -11,8 +11,6 @@
 
 #include "wx/wxprec.h"
 
-#include "wx/defs.h"
-
 #ifdef wxHAS_TASK_BAR_ICON
 
 #include "wx/mac/private.h"
diff -ur -x CVS ../wxWidgets/src/mac/carbon/textctrl.cpp ./src/mac/carbon/textctrl.cpp
--- ../wxWidgets/src/mac/carbon/textctrl.cpp	Mon Jan  3 11:48:35 2005
+++ ./src/mac/carbon/textctrl.cpp	Wed Jan  5 15:21:11 2005
@@ -9,11 +9,11 @@
 // Licence:     wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "textctrl.h"
 #endif
 
-#include "wx/defs.h"
+#include "wx/wxprec.h"
 
 #if wxUSE_TEXTCTRL
 
diff -ur -x CVS ../wxWidgets/src/mac/carbon/tglbtn.cpp ./src/mac/carbon/tglbtn.cpp
--- ../wxWidgets/src/mac/carbon/tglbtn.cpp	Mon Jul 19 22:21:39 2004
+++ ./src/mac/carbon/tglbtn.cpp	Wed Jan  5 15:21:38 2005
@@ -18,15 +18,15 @@
 // headers
 // ----------------------------------------------------------------------------
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "button.h"
 #endif
 
-#include "wx/defs.h"
-#include "wx/tglbtn.h"
+#include "wx/wxprec.h"
 
 #if wxUSE_TOGGLEBTN
 
+#include "wx/tglbtn.h"
 #include "wx/mac/uma.h"
 // Button
 
diff -ur -x CVS ../wxWidgets/src/mac/carbon/thread.cpp ./src/mac/carbon/thread.cpp
--- ../wxWidgets/src/mac/carbon/thread.cpp	Mon Jan  3 11:48:35 2005
+++ ./src/mac/carbon/thread.cpp	Wed Jan  5 14:44:39 2005
@@ -10,7 +10,7 @@
 // Licence:	wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "thread.h"
 #endif
 
diff -ur -x CVS ../wxWidgets/src/mac/carbon/timer.cpp ./src/mac/carbon/timer.cpp
--- ../wxWidgets/src/mac/carbon/timer.cpp	Sun May 23 13:52:33 2004
+++ ./src/mac/carbon/timer.cpp	Wed Jan  5 15:21:58 2005
@@ -9,9 +9,11 @@
 // Licence:       wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "timer.h"
 #endif
+
+#include "wx/wxprec.h"
 
 #include "wx/timer.h"
 
diff -ur -x CVS ../wxWidgets/src/mac/carbon/toolbar.cpp ./src/mac/carbon/toolbar.cpp
--- ../wxWidgets/src/mac/carbon/toolbar.cpp	Wed Nov 24 14:50:12 2004
+++ ./src/mac/carbon/toolbar.cpp	Wed Jan  5 15:22:29 2005
@@ -9,14 +9,15 @@
 // Licence:     The wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "toolbar.h"
 #endif
 
-#include "wx/wx.h"
+#include "wx/wxprec.h"
 
 #if wxUSE_TOOLBAR
 
+#include "wx/wx.h"
 #include "wx/toolbar.h"
 #include "wx/notebook.h"
 #include "wx/tabctrl.h"
diff -ur -x CVS ../wxWidgets/src/mac/carbon/tooltip.cpp ./src/mac/carbon/tooltip.cpp
--- ../wxWidgets/src/mac/carbon/tooltip.cpp	Mon Jul 19 22:21:39 2004
+++ ./src/mac/carbon/tooltip.cpp	Wed Jan  5 15:22:41 2005
@@ -7,11 +7,11 @@
 // Licence:     wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
     #pragma implementation "tooltip.h"
 #endif
 
-#include "wx/defs.h"
+#include "wx/wxprec.h"
 
 #if wxUSE_TOOLTIPS
 
diff -ur -x CVS ../wxWidgets/src/mac/carbon/toplevel.cpp ./src/mac/carbon/toplevel.cpp
--- ../wxWidgets/src/mac/carbon/toplevel.cpp	Mon Jan  3 11:48:36 2005
+++ ./src/mac/carbon/toplevel.cpp	Wed Jan  5 14:44:39 2005
@@ -17,7 +17,7 @@
 // headers
 // ----------------------------------------------------------------------------
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
     #pragma implementation "toplevel.h"
 #endif
 
diff -ur -x CVS ../wxWidgets/src/mac/carbon/treectrl.cpp ./src/mac/carbon/treectrl.cpp
--- ../wxWidgets/src/mac/carbon/treectrl.cpp	Sun May 23 13:52:33 2004
+++ ./src/mac/carbon/treectrl.cpp	Wed Jan  5 15:23:06 2005
@@ -9,9 +9,11 @@
 // Licence:       wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "treectrl.h"
 #endif
+
+#include "wx/wxprec.h"
 
 #include "wx/stubs/textctrl.h"
 #include "wx/treebase.h"
diff -ur -x CVS ../wxWidgets/src/mac/carbon/uma.cpp ./src/mac/carbon/uma.cpp
--- ../wxWidgets/src/mac/carbon/uma.cpp	Sat Jun 26 03:02:36 2004
+++ ./src/mac/carbon/uma.cpp	Wed Jan  5 15:23:19 2005
@@ -9,7 +9,7 @@
 // Licence:     The wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#include "wx/defs.h"
+#include "wx/wxprec.h"
 
 #if wxUSE_GUI
 
diff -ur -x CVS ../wxWidgets/src/mac/carbon/utils.cpp ./src/mac/carbon/utils.cpp
--- ../wxWidgets/src/mac/carbon/utils.cpp	Mon Jan  3 11:48:36 2005
+++ ./src/mac/carbon/utils.cpp	Wed Jan  5 15:23:37 2005
@@ -9,12 +9,13 @@
 // Licence:       wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 // Note: this is done in utilscmn.cpp now.
 // #pragma implementation "utils.h"
 #endif
 
-#include "wx/setup.h"
+#include "wx/wxprec.h"
+
 #include "wx/utils.h"
 #include "wx/app.h"
 #include "wx/apptrait.h"
diff -ur -x CVS ../wxWidgets/src/mac/carbon/utilsexc.cpp ./src/mac/carbon/utilsexc.cpp
--- ../wxWidgets/src/mac/carbon/utilsexc.cpp	Fri Nov  5 10:23:43 2004
+++ ./src/mac/carbon/utilsexc.cpp	Wed Jan  5 14:44:39 2005
@@ -9,10 +9,6 @@
 // Licence:       wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
-//#pragma implementation
-#endif
-
 #include "wx/wxprec.h"
 #ifndef WX_PRECOMP
     #include "wx/log.h"
diff -ur -x CVS ../wxWidgets/src/mac/carbon/window.cpp ./src/mac/carbon/window.cpp
--- ../wxWidgets/src/mac/carbon/window.cpp	Mon Jan  3 11:48:36 2005
+++ ./src/mac/carbon/window.cpp	Wed Jan  5 15:24:02 2005
@@ -9,11 +9,12 @@
 // Licence:     wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "window.h"
 #endif
 
-#include "wx/setup.h"
+#include "wx/wxprec.h"
+
 #include "wx/menu.h"
 #include "wx/window.h"
 #include "wx/dc.h"

 	  	 
