Index: Makefile.in
===================================================================
RCS file: /home/wxcvs/wxWindows/Makefile.in,v
retrieving revision 1.142
diff -d -u -2 -r1.142 Makefile.in
--- Makefile.in	2001/02/08 11:19:24	1.142
+++ Makefile.in	2001/04/24 17:25:05
@@ -43,5 +43,5 @@
 VP10 = @top_srcdir@/src/zlib
 VP11 = @top_srcdir@/src/iodbc
-VP12 = @top_srcdir@/src/ole
+VP12 = @top_srcdir@/src/msw/ole
 
 FTVP01 = @top_srcdir@/src/freetype/unix
@@ -338,5 +338,13 @@
 		prepare.o \
 		result.o
-        
+
+OLEOBJS = \
+		automtn.o \
+		dataobj.o \
+		dropsrc.o \
+		droptgt.o \
+		oleutils.o \
+		uuid.o
+  
 ############################## Rules ##################################
 
Index: ./src/make.env.in
===================================================================
RCS file: /home/wxcvs/wxWindows/src/make.env.in,v
retrieving revision 1.18
diff -d -u -2 -r1.18 make.env.in
--- ./src/make.env.in	2000/07/15 19:50:24	1.18
+++ ./src/make.env.in	2001/04/24 17:25:05
@@ -71,6 +71,6 @@
 # header files get found.  (KB)
 
-CFLAGS = $(APPEXTRACFLAGS) $(APPEXTRACPPFLAGS) @TOOLKIT_DEF@ @CODE_GEN_FLAGS@ @WXDEBUG_DEFINE@ @CFLAGS@ @CPPFLAGS@
-CPPFLAGS = $(APPEXTRACXXFLAGS) $(APPEXTRACPPFLAGS) @TOOLKIT_DEF@ @CODE_GEN_FLAGS@ @WXDEBUG_DEFINE@ @CXXFLAGS@ @CPPFLAGS@
+CFLAGS = $(APPEXTRACFLAGS) $(APPEXTRACPPFLAGS) @TOOLKIT_DEF@ @CODE_GEN_FLAGS@ @WXDEBUG_DEFINE@ @CFLAGS@ @CPPFLAGS@ @CODE_GEN_FLAGS@
+CPPFLAGS = $(APPEXTRACXXFLAGS) $(APPEXTRACPPFLAGS) @TOOLKIT_DEF@ @CODE_GEN_FLAGS@ @WXDEBUG_DEFINE@ @CXXFLAGS@ @CPPFLAGS@ @CODE_GEN_FLAGS_CXX@
 PICFLAGS = @PIC_FLAG@
 

 	  	 
