Index: src/ribbon/buttonbar.cpp
===================================================================
--- src/ribbon/buttonbar.cpp	(revision 65786)
+++ src/ribbon/buttonbar.cpp	(working copy)
@@ -835,6 +835,11 @@
         }
     }
 
+    if(new_hovered == NULL && GetToolTip())
+    {
+        UnsetToolTip();
+    }
+
     if(new_hovered != m_hovered_button || (m_hovered_button != NULL &&
         new_hovered_state != m_hovered_button->base->state))
     {
@@ -846,6 +851,7 @@
         if(m_hovered_button != NULL)
         {
             m_hovered_button->base->state = new_hovered_state;
+            SetToolTip(m_hovered_button->base->help_string);
         }
         Refresh(false);
     }
Index: src/ribbon/toolbar.cpp
===================================================================
--- src/ribbon/toolbar.cpp	(revision 65651)
+++ src/ribbon/toolbar.cpp	(working copy)
@@ -581,6 +581,15 @@
         }
     }
 
+    if(new_hover)
+    {
+        SetToolTip(new_hover->help_string);
+    }
+    else if(GetToolTip())
+    {
+        UnsetToolTip();
+    }
+
     if(new_hover != m_hover_tool)
     {
         if(m_hover_tool)
