Index: wxWidgets-2.8-STABLE/src/common/filename.cpp
===================================================================
--- wxWidgets-2.8-STABLE/src/common/filename.cpp	(revision 57954)
+++ wxWidgets-2.8-STABLE/src/common/filename.cpp	(working copy)
@@ -2153,25 +2153,45 @@
                           const wxDateTime *dtCreate)
 {
 #if defined(__WIN32__)
+    FILETIME ftAccess, ftCreate, ftWrite;
+   
+    if ( dtCreate )
+        ConvertWxToFileTime(&ftCreate, *dtCreate);
+    if ( dtAccess )
+        ConvertWxToFileTime(&ftAccess, *dtAccess);
+    if ( dtMod )
+        ConvertWxToFileTime(&ftWrite, *dtMod);
+
     if ( IsDir() )
     {
-        // VZ: please let me know how to do this if you can
-        wxFAIL_MSG( _T("SetTimes() not implemented for the directories") );
+        if ( wxGetOsVersion() == wxOS_WINDOWS_9X )
+        {
+            wxLogError(_("Setting directory access times is not supported under this OS version"));
+            return false;
+        }
+        else
+        {
+            HANDLE hDir = CreateFile(GetPath(wxPATH_GET_VOLUME), GENERIC_WRITE,FILE_SHARE_WRITE, 
+            NULL, OPEN_EXISTING, FILE_FLAG_BACKUP_SEMANTICS, NULL);
+     
+            if ( hDir == INVALID_HANDLE_VALUE )
+            {
+                wxLogSysError(_("Failed to open the directory '%s' to change its access time"), GetPath(wxPATH_GET_VOLUME).c_str());
+                return false;
+            }
+            if ( SetFileTime(hDir, &ftCreate, &ftAccess, &ftWrite ) )
+            {
+                CloseHandle(hDir);
+                return true;
+            }
+            CloseHandle(hDir);
+        }
     }
     else // file
     {
         wxFileHandle fh(GetFullPath(), wxFileHandle::Write);
         if ( fh.IsOk() )
         {
-            FILETIME ftAccess, ftCreate, ftWrite;
-
-            if ( dtCreate )
-                ConvertWxToFileTime(&ftCreate, *dtCreate);
-            if ( dtAccess )
-                ConvertWxToFileTime(&ftAccess, *dtAccess);
-            if ( dtMod )
-                ConvertWxToFileTime(&ftWrite, *dtMod);
-
             if ( ::SetFileTime(fh,
                                dtCreate ? &ftCreate : NULL,
                                dtAccess ? &ftAccess : NULL,
