Index: src/mac/carbon/listctrl_mac.cpp
===================================================================
--- src/mac/carbon/listctrl_mac.cpp	(revision 65530)
+++ src/mac/carbon/listctrl_mac.cpp	(working copy)
@@ -622,7 +622,9 @@
         (hitResult & wxLIST_HITTEST_ONITEMLABEL) &&
         HasFlag(wxLC_EDIT_LABELS) )
     {
-        m_renameTimer->Start( 100, true );
+        // Generic implementation has its own rename timer
+        if (!m_genericImpl)
+            m_renameTimer->Start( 250, true );
     }
     else
     {
