Index: src/osx/carbon/listctrl_mac.cpp
===================================================================
--- src/osx/carbon/listctrl_mac.cpp	(revision 65532)
+++ src/osx/carbon/listctrl_mac.cpp	(working copy)
@@ -618,7 +618,9 @@
         (hitResult & wxLIST_HITTEST_ONITEMLABEL) &&
         HasFlag(wxLC_EDIT_LABELS) )
     {
-        m_renameTimer->Start( 100, true );
+        // Generic implementation has its own rename timer
+        if (!m_genericImpl)
+            m_renameTimer->Start( 250, true );
     }
     else
     {
