diff -uNr wxWidgets-orig/include/wx/imagpng.h wxWidgets-mine/include/wx/imagpng.h
--- wxWidgets-orig/include/wx/imagpng.h	2008-03-27 11:30:38.000000000 +0700
+++ wxWidgets-mine/include/wx/imagpng.h	2009-01-10 14:08:44.000000000 +0700
@@ -22,6 +22,11 @@
 
 #define wxIMAGE_OPTION_PNG_FORMAT    wxT("PngFormat")
 #define wxIMAGE_OPTION_PNG_BITDEPTH  wxT("PngBitDepth")
+#define wxIMAGE_OPTION_PNG_FILTER    wxT("PngF")
+#define wxIMAGE_OPTION_PNG_COMPRESSION_LEVEL       wxT("PngZL")
+#define wxIMAGE_OPTION_PNG_COMPRESSION_MEM_LEVEL   wxT("PngZM")
+#define wxIMAGE_OPTION_PNG_COMPRESSION_STRATEGY    wxT("PngZS")
+#define wxIMAGE_OPTION_PNG_COMPRESSION_BUFFER_SIZE wxT("PngZB")
 
 enum
 {
diff -uNr wxWidgets-orig/interface/wx/image.h wxWidgets-mine/interface/wx/image.h
--- wxWidgets-orig/interface/wx/image.h	2008-11-29 11:30:42.000000000 +0700
+++ wxWidgets-mine/interface/wx/image.h	2009-01-10 14:37:44.000000000 +0700
@@ -839,6 +839,20 @@
         @li wxIMAGE_OPTION_PNG_FORMAT: Format for saving a PNG file, see
             wxImagePNGType for the supported values.
         @li wxIMAGE_OPTION_PNG_BITDEPTH: Bit depth for every channel (R/G/B/A).
+        @li wxIMAGE_OPTION_PNG_FILTER: Filter for saving a PNG file, see libpng
+            for possible values.
+        @li wxIMAGE_OPTION_PNG_COMPRESSION_LEVEL: Compression level (0..9) for
+            saving a PNG file. Higher level creates smaller-but-slower PNG file.
+        @li wxIMAGE_OPTION_PNG_COMPRESSION_MEM_LEVEL: Compression memory usage
+            level (1..9) for saving a PNG file. Higher value means the saving
+            process consumes more memory, but may create smaller PNG file.
+        @li wxIMAGE_OPTION_PNG_COMPRESSION_STRATEGY: Possible values are 0 for
+            default strategy, 1 for filter, and 2 for Huffman-only. You can use
+            OptiPNG to get suitable value for your application.
+        @li wxIMAGE_OPTION_PNG_COMPRESSION_BUFFER_SIZE: Internal buffer size
+            (in bytes) for saving a PNG file. Ideally this should be as big as
+            the resulting PNG file. Use this option if your application produces
+            images with small size variation.
 
         @see SetOption(), GetOption()
 
diff -uNr wxWidgets-orig/samples/image/image.cpp wxWidgets-mine/samples/image/image.cpp
--- wxWidgets-orig/samples/image/image.cpp	2008-12-05 11:30:26.000000000 +0700
+++ wxWidgets-mine/samples/image/image.cpp	2009-01-10 14:41:42.000000000 +0700
@@ -315,6 +315,42 @@
             {
                 image.SetOption(wxIMAGE_OPTION_PNG_FORMAT, pngvalues[sel]);
                 image.SetOption(wxIMAGE_OPTION_PNG_BITDEPTH, sel % 2 ? 16 : 8);
+
+                const wxString compressionChoices[] = // these values are taken from OptiPNG with -o3 switch
+                {
+                    _T("zc = 9, zm = 8, zs = 0, f = 0"),
+                    _T("zc = 9, zm = 9, zs = 0, f = 0"),
+                    _T("zc = 9, zm = 8, zs = 1, f = 0"),
+                    _T("zc = 9, zm = 9, zs = 1, f = 0"),
+                    _T("zc = 1, zm = 8, zs = 2, f = 0"),
+                    _T("zc = 1, zm = 9, zs = 2, f = 0"),
+                    _T("zc = 9, zm = 8, zs = 0, f = 5"),
+                    _T("zc = 9, zm = 9, zs = 0, f = 5"),
+                    _T("zc = 9, zm = 8, zs = 1, f = 5"),
+                    _T("zc = 9, zm = 9, zs = 1, f = 5"),
+                    _T("zc = 1, zm = 8, zs = 2, f = 5"),
+                    _T("zc = 1, zm = 9, zs = 2, f = 5"),
+                };
+
+                int sel = wxGetSingleChoiceIndex(_T("Select compression option (Cancel to use default)\n"),
+                                                 _T("PNG Compression Options"),
+                                                 WXSIZEOF(compressionChoices),
+                                                 compressionChoices,
+                                                 this);
+                if (sel != -1)
+                {
+                    const int zc[] = {9, 9, 9, 9, 1, 1, 9, 9, 9, 9, 1, 1};
+                    const int zm[] = {8, 9, 8, 9, 8, 9, 8, 9, 8, 9, 8, 9};
+                    const int zs[] = {0, 0, 1, 1, 2, 2, 0, 0, 1, 1, 2, 2};
+                    const int f[]  = {0x08, 0x08, 0x08, 0x08, 0x08, 0x08,
+                                      0xF8, 0xF8, 0xF8, 0xF8, 0xF8, 0xF8};
+
+                    image.SetOption(wxIMAGE_OPTION_PNG_COMPRESSION_LEVEL      , zc[sel]);
+                    image.SetOption(wxIMAGE_OPTION_PNG_COMPRESSION_MEM_LEVEL  , zm[sel]);
+                    image.SetOption(wxIMAGE_OPTION_PNG_COMPRESSION_STRATEGY   , zs[sel]);
+                    image.SetOption(wxIMAGE_OPTION_PNG_FILTER                 , f[sel]);
+                    image.SetOption(wxIMAGE_OPTION_PNG_COMPRESSION_BUFFER_SIZE, 1048576); // 1 MB
+                }
             }
         }
         else if ( extension == _T("cur") )
diff -uNr wxWidgets-orig/src/common/imagpng.cpp wxWidgets-mine/src/common/imagpng.cpp
--- wxWidgets-orig/src/common/imagpng.cpp	2008-05-10 10:30:22.000000000 +0700
+++ wxWidgets-mine/src/common/imagpng.cpp	2009-01-10 14:39:22.000000000 +0700
@@ -710,6 +710,21 @@
                                   : PNG_COLOR_TYPE_GRAY;
     }
 
+    if (image->HasOption(wxIMAGE_OPTION_PNG_FILTER))
+        png_set_filter( png_ptr, PNG_FILTER_TYPE_BASE, image->GetOptionInt(wxIMAGE_OPTION_PNG_FILTER) );
+
+    if (image->HasOption(wxIMAGE_OPTION_PNG_COMPRESSION_LEVEL))
+        png_set_compression_level( png_ptr, image->GetOptionInt(wxIMAGE_OPTION_PNG_COMPRESSION_LEVEL) );
+
+    if (image->HasOption(wxIMAGE_OPTION_PNG_COMPRESSION_MEM_LEVEL))
+        png_set_compression_mem_level( png_ptr, image->GetOptionInt(wxIMAGE_OPTION_PNG_COMPRESSION_MEM_LEVEL) );
+
+    if (image->HasOption(wxIMAGE_OPTION_PNG_COMPRESSION_STRATEGY))
+        png_set_compression_strategy( png_ptr, image->GetOptionInt(wxIMAGE_OPTION_PNG_COMPRESSION_STRATEGY) );
+
+    if (image->HasOption(wxIMAGE_OPTION_PNG_COMPRESSION_BUFFER_SIZE))
+        png_set_compression_buffer_size( png_ptr, image->GetOptionInt(wxIMAGE_OPTION_PNG_COMPRESSION_BUFFER_SIZE) );
+
     png_set_IHDR( png_ptr, info_ptr, image->GetWidth(), image->GetHeight(),
                   iBitDepth, iPngColorType,
                   PNG_INTERLACE_NONE, PNG_COMPRESSION_TYPE_BASE,
