--- build/msw/wx_vc10_adv.vcxproj
+++ build/msw/wx_vc10_adv.vcxproj
@@ -74,6 +74,7 @@
     <LinkIncremental Condition="'$(Configuration)|$(Platform)'=='DLL Release|Win32'">false</LinkIncremental>
     <GenerateManifest Condition="'$(Configuration)|$(Platform)'=='DLL Release|Win32'">true</GenerateManifest>
     <TargetName Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">wxmsw29ud_adv</TargetName>
+    <TargetName Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">wxmsw29u_adv</TargetName>
     <TargetName Condition="'$(Configuration)|$(Platform)'=='DLL Debug|Win32'">wxmsw295ud_$(ProjectName)_vc_custom</TargetName>
     <TargetName Condition="'$(Configuration)|$(Platform)'=='DLL Release|Win32'">wxmsw295u_$(ProjectName)_vc_custom</TargetName>
   </PropertyGroup>
--- build/msw/wx_vc10_aui.vcxproj
+++ build/msw/wx_vc10_aui.vcxproj
@@ -74,6 +74,7 @@
     <LinkIncremental Condition="'$(Configuration)|$(Platform)'=='DLL Release|Win32'">false</LinkIncremental>
     <GenerateManifest Condition="'$(Configuration)|$(Platform)'=='DLL Release|Win32'">true</GenerateManifest>
     <TargetName Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">wxmsw29ud_aui</TargetName>
+    <TargetName Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">wxmsw29u_aui</TargetName>
     <TargetName Condition="'$(Configuration)|$(Platform)'=='DLL Debug|Win32'">wxmsw295ud_$(ProjectName)_vc_custom</TargetName>
     <TargetName Condition="'$(Configuration)|$(Platform)'=='DLL Release|Win32'">wxmsw295u_$(ProjectName)_vc_custom</TargetName>
   </PropertyGroup>
--- build/msw/wx_vc10_gl.vcxproj
+++ build/msw/wx_vc10_gl.vcxproj
@@ -74,6 +74,7 @@
     <LinkIncremental Condition="'$(Configuration)|$(Platform)'=='DLL Release|Win32'">false</LinkIncremental>
     <GenerateManifest Condition="'$(Configuration)|$(Platform)'=='DLL Release|Win32'">true</GenerateManifest>
     <TargetName Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">wxmsw29ud_gl</TargetName>
+    <TargetName Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">wxmsw29u_gl</TargetName>
     <TargetName Condition="'$(Configuration)|$(Platform)'=='DLL Debug|Win32'">wxmsw295ud_$(ProjectName)_vc_custom</TargetName>
     <TargetName Condition="'$(Configuration)|$(Platform)'=='DLL Release|Win32'">wxmsw295u_$(ProjectName)_vc_custom</TargetName>
   </PropertyGroup>
--- build/msw/wx_vc10_html.vcxproj
+++ build/msw/wx_vc10_html.vcxproj
@@ -74,6 +74,7 @@
     <LinkIncremental Condition="'$(Configuration)|$(Platform)'=='DLL Release|Win32'">false</LinkIncremental>
     <GenerateManifest Condition="'$(Configuration)|$(Platform)'=='DLL Release|Win32'">true</GenerateManifest>
     <TargetName Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">wxmsw29ud_html</TargetName>
+    <TargetName Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">wxmsw29u_html</TargetName>
     <TargetName Condition="'$(Configuration)|$(Platform)'=='DLL Debug|Win32'">wxmsw295ud_$(ProjectName)_vc_custom</TargetName>
     <TargetName Condition="'$(Configuration)|$(Platform)'=='DLL Release|Win32'">wxmsw295u_$(ProjectName)_vc_custom</TargetName>
   </PropertyGroup>
--- build/msw/wx_vc10_media.vcxproj
+++ build/msw/wx_vc10_media.vcxproj
@@ -74,6 +74,7 @@
     <LinkIncremental Condition="'$(Configuration)|$(Platform)'=='DLL Release|Win32'">false</LinkIncremental>
     <GenerateManifest Condition="'$(Configuration)|$(Platform)'=='DLL Release|Win32'">true</GenerateManifest>
     <TargetName Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">wxmsw29ud_media</TargetName>
+    <TargetName Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">wxmsw29u_media</TargetName>
     <TargetName Condition="'$(Configuration)|$(Platform)'=='DLL Debug|Win32'">wxmsw295ud_$(ProjectName)_vc_custom</TargetName>
     <TargetName Condition="'$(Configuration)|$(Platform)'=='DLL Release|Win32'">wxmsw295u_$(ProjectName)_vc_custom</TargetName>
   </PropertyGroup>
--- build/msw/wx_vc10_net.vcxproj
+++ build/msw/wx_vc10_net.vcxproj
@@ -74,6 +74,7 @@
     <LinkIncremental Condition="'$(Configuration)|$(Platform)'=='DLL Release|Win32'">false</LinkIncremental>
     <GenerateManifest Condition="'$(Configuration)|$(Platform)'=='DLL Release|Win32'">true</GenerateManifest>
     <TargetName Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">wxbase29ud_net</TargetName>
+    <TargetName Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">wxbase29u_net</TargetName>
     <TargetName Condition="'$(Configuration)|$(Platform)'=='DLL Debug|Win32'">wxbase295ud_$(ProjectName)_vc_custom</TargetName>
     <TargetName Condition="'$(Configuration)|$(Platform)'=='DLL Release|Win32'">wxbase295u_$(ProjectName)_vc_custom</TargetName>
   </PropertyGroup>
--- build/msw/wx_vc10_propgrid.vcxproj
+++ build/msw/wx_vc10_propgrid.vcxproj
@@ -74,6 +74,7 @@
     <LinkIncremental Condition="'$(Configuration)|$(Platform)'=='DLL Release|Win32'">false</LinkIncremental>
     <GenerateManifest Condition="'$(Configuration)|$(Platform)'=='DLL Release|Win32'">true</GenerateManifest>
     <TargetName Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">wxmsw29ud_propgrid</TargetName>
+    <TargetName Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">wxmsw29u_propgrid</TargetName>
     <TargetName Condition="'$(Configuration)|$(Platform)'=='DLL Debug|Win32'">wxmsw295ud_$(ProjectName)_vc_custom</TargetName>
     <TargetName Condition="'$(Configuration)|$(Platform)'=='DLL Release|Win32'">wxmsw295u_$(ProjectName)_vc_custom</TargetName>
   </PropertyGroup>
--- build/msw/wx_vc10_qa.vcxproj
+++ build/msw/wx_vc10_qa.vcxproj
@@ -74,6 +74,7 @@
     <LinkIncremental Condition="'$(Configuration)|$(Platform)'=='DLL Release|Win32'">false</LinkIncremental>
     <GenerateManifest Condition="'$(Configuration)|$(Platform)'=='DLL Release|Win32'">true</GenerateManifest>
     <TargetName Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">wxmsw29ud_qa</TargetName>
+    <TargetName Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">wxmsw29u_qa</TargetName>
     <TargetName Condition="'$(Configuration)|$(Platform)'=='DLL Debug|Win32'">wxmsw295ud_$(ProjectName)_vc_custom</TargetName>
     <TargetName Condition="'$(Configuration)|$(Platform)'=='DLL Release|Win32'">wxmsw295u_$(ProjectName)_vc_custom</TargetName>
   </PropertyGroup>
--- build/msw/wx_vc10_ribbon.vcxproj
+++ build/msw/wx_vc10_ribbon.vcxproj
@@ -74,6 +74,7 @@
     <LinkIncremental Condition="'$(Configuration)|$(Platform)'=='DLL Release|Win32'">false</LinkIncremental>
     <GenerateManifest Condition="'$(Configuration)|$(Platform)'=='DLL Release|Win32'">true</GenerateManifest>
     <TargetName Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">wxmsw29ud_ribbon</TargetName>
+    <TargetName Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">wxmsw29u_ribbon</TargetName>
     <TargetName Condition="'$(Configuration)|$(Platform)'=='DLL Debug|Win32'">wxmsw295ud_$(ProjectName)_vc_custom</TargetName>
     <TargetName Condition="'$(Configuration)|$(Platform)'=='DLL Release|Win32'">wxmsw295u_$(ProjectName)_vc_custom</TargetName>
   </PropertyGroup>
--- build/msw/wx_vc10_richtext.vcxproj
+++ build/msw/wx_vc10_richtext.vcxproj
@@ -74,6 +74,7 @@
     <LinkIncremental Condition="'$(Configuration)|$(Platform)'=='DLL Release|Win32'">false</LinkIncremental>
     <GenerateManifest Condition="'$(Configuration)|$(Platform)'=='DLL Release|Win32'">true</GenerateManifest>
     <TargetName Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">wxmsw29ud_richtext</TargetName>
+    <TargetName Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">wxmsw29u_richtext</TargetName>
     <TargetName Condition="'$(Configuration)|$(Platform)'=='DLL Debug|Win32'">wxmsw295ud_$(ProjectName)_vc_custom</TargetName>
     <TargetName Condition="'$(Configuration)|$(Platform)'=='DLL Release|Win32'">wxmsw295u_$(ProjectName)_vc_custom</TargetName>
   </PropertyGroup>
--- build/msw/wx_vc10_stc.vcxproj
+++ build/msw/wx_vc10_stc.vcxproj
@@ -74,6 +74,7 @@
     <LinkIncremental Condition="'$(Configuration)|$(Platform)'=='DLL Release|Win32'">false</LinkIncremental>
     <GenerateManifest Condition="'$(Configuration)|$(Platform)'=='DLL Release|Win32'">true</GenerateManifest>
     <TargetName Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">wxmsw29ud_stc</TargetName>
+    <TargetName Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">wxmsw29u_stc</TargetName>
     <TargetName Condition="'$(Configuration)|$(Platform)'=='DLL Debug|Win32'">wxmsw295ud_$(ProjectName)_vc_custom</TargetName>
     <TargetName Condition="'$(Configuration)|$(Platform)'=='DLL Release|Win32'">wxmsw295u_$(ProjectName)_vc_custom</TargetName>
   </PropertyGroup>
--- build/msw/wx_vc10_webview.vcxproj
+++ build/msw/wx_vc10_webview.vcxproj
@@ -74,6 +74,7 @@
     <LinkIncremental Condition="'$(Configuration)|$(Platform)'=='DLL Release|Win32'">false</LinkIncremental>
     <GenerateManifest Condition="'$(Configuration)|$(Platform)'=='DLL Release|Win32'">true</GenerateManifest>
     <TargetName Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">wxmsw29ud_webview</TargetName>
+    <TargetName Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">wxmsw29u_webview</TargetName>
     <TargetName Condition="'$(Configuration)|$(Platform)'=='DLL Debug|Win32'">wxmsw295ud_$(ProjectName)_vc_custom</TargetName>
     <TargetName Condition="'$(Configuration)|$(Platform)'=='DLL Release|Win32'">wxmsw295u_$(ProjectName)_vc_custom</TargetName>
   </PropertyGroup>
--- build/msw/wx_vc10_xml.vcxproj
+++ build/msw/wx_vc10_xml.vcxproj
@@ -74,6 +74,7 @@
     <LinkIncremental Condition="'$(Configuration)|$(Platform)'=='DLL Release|Win32'">false</LinkIncremental>
     <GenerateManifest Condition="'$(Configuration)|$(Platform)'=='DLL Release|Win32'">true</GenerateManifest>
     <TargetName Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">wxbase29ud_xml</TargetName>
+    <TargetName Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">wxbase29u_xml</TargetName>
     <TargetName Condition="'$(Configuration)|$(Platform)'=='DLL Debug|Win32'">wxbase295ud_$(ProjectName)_vc_custom</TargetName>
     <TargetName Condition="'$(Configuration)|$(Platform)'=='DLL Release|Win32'">wxbase295u_$(ProjectName)_vc_custom</TargetName>
   </PropertyGroup>
--- build/msw/wx_vc10_xrc.vcxproj
+++ build/msw/wx_vc10_xrc.vcxproj
@@ -74,6 +74,7 @@
     <LinkIncremental Condition="'$(Configuration)|$(Platform)'=='DLL Release|Win32'">false</LinkIncremental>
     <GenerateManifest Condition="'$(Configuration)|$(Platform)'=='DLL Release|Win32'">true</GenerateManifest>
     <TargetName Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">wxmsw29ud_xrc</TargetName>
+    <TargetName Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">wxmsw29u_xrc</TargetName>
     <TargetName Condition="'$(Configuration)|$(Platform)'=='DLL Debug|Win32'">wxmsw295ud_$(ProjectName)_vc_custom</TargetName>
     <TargetName Condition="'$(Configuration)|$(Platform)'=='DLL Release|Win32'">wxmsw295u_$(ProjectName)_vc_custom</TargetName>
   </PropertyGroup>
