diff -bru /mnt/winxp/wxWidgets/include/wx/event.h ./wxWidgets/include/wx/event.h
--- /mnt/winxp/wxWidgets/include/wx/event.h	2012-10-15 10:28:19.000000000 -0700
+++ ./wxWidgets/include/wx/event.h	2012-10-16 10:03:11.000000000 -0700
@@ -930,6 +930,9 @@
     int GetId() const { return m_id; }
     void SetId(int Id) { m_id = Id; }
 
+    // This function is to retrieve user specified data when the event is attached
+    wxObject *GetEventUserData() const { return m_callbackUserData; }
+
     // Can instruct event processor that we wish to ignore this event
     // (treat as if the event table entry had not been found): this must be done
     // to allow the event processing by the base classes (calling event.Skip()
diff -bru /mnt/winxp/wxWidgets/interface/wx/event.h ./wxWidgets/interface/wx/event.h
--- /mnt/winxp/wxWidgets/interface/wx/event.h	2012-10-15 10:29:16.000000000 -0700
+++ ./wxWidgets/interface/wx/event.h	2012-10-15 11:39:33.000000000 -0700
@@ -160,6 +160,13 @@
     int GetId() const;
 
     /**
+        Return the user supplied data when the event is attached
+
+        @since 2.9.5
+    */
+    wxObject *GetEventUserData();
+
+    /**
         Returns @true if the event handler should be skipped, @false otherwise.
     */
     bool GetSkipped() const;
