diff -uNr wxWidgets-orig/include/wx/helpbase.h wxWidgets-mine/include/wx/helpbase.h
--- wxWidgets-orig/include/wx/helpbase.h	2019-07-09 00:02:14.000000000 +0200
+++ wxWidgets-mine/include/wx/helpbase.h	2019-07-15 21:38:45.642629000 +0200
@@ -51,6 +51,12 @@
 
     // Displays the contents
     virtual bool DisplayContents(void) = 0;
+    
+    // Displays the index
+    virtual bool DisplayIndex(void) = 0;
+
+    // Displays the search page
+    virtual bool DisplaySearchPage(void) = 0;    
 
     // Display the given section
     virtual bool DisplaySection(int sectionNo) = 0;
diff -uNr wxWidgets-orig/include/wx/msw/helpchm.h wxWidgets-mine/include/wx/msw/helpchm.h
--- wxWidgets-orig/include/wx/msw/helpchm.h	2019-07-09 00:02:14.000000000 +0200
+++ wxWidgets-mine/include/wx/msw/helpchm.h	2019-07-15 21:39:14.731292700 +0200
@@ -27,6 +27,8 @@
     // If file is "", reloads file given in Initialize
     virtual bool LoadFile(const wxString& file = wxEmptyString) wxOVERRIDE;
     virtual bool DisplayContents() wxOVERRIDE;
+    virtual bool DisplayIndex(void) wxOVERRIDE;
+    virtual bool DisplaySearchPage(void) wxOVERRIDE;    
     virtual bool DisplaySection(int sectionNo) wxOVERRIDE;
     virtual bool DisplaySection(const wxString& section) wxOVERRIDE;
     virtual bool DisplayBlock(long blockNo) wxOVERRIDE;
diff -uNr wxWidgets-orig/src/msw/helpchm.cpp wxWidgets-mine/src/msw/helpchm.cpp
--- wxWidgets-orig/src/msw/helpchm.cpp	2019-07-09 00:02:14.000000000 +0200
+++ wxWidgets-mine/src/msw/helpchm.cpp	2019-07-15 21:39:56.593687100 +0200
@@ -113,7 +113,33 @@
     if (m_helpFile.IsEmpty())
         return false;
 
-    return CallHtmlHelp(HH_DISPLAY_TOPIC);
+    return CallHtmlHelp(HH_DISPLAY_TOC);
+}
+
+bool wxCHMHelpController::DisplayIndex()
+{
+    if (m_helpFile.IsEmpty())
+        return false;
+
+    return CallHtmlHelp(HH_DISPLAY_INDEX);
+}
+
+bool wxCHMHelpController::DisplaySearchPage()
+{
+    if (m_helpFile.IsEmpty())
+        return false;
+
+    HH_FTS_QUERY oQuery;
+    oQuery.cbStruct = sizeof(HH_FTS_QUERY);
+    oQuery.fStemmedSearch = 0;
+    oQuery.fTitleOnly = 0;
+    oQuery.fUniCodeStrings = 0;
+    oQuery.iProximity = 0;
+    oQuery.pszSearchQuery = TEXT("");
+    oQuery.pszWindow = TEXT("");
+    oQuery.fExecute = 1;
+
+    return CallHtmlHelp(HH_DISPLAY_SEARCH, &oQuery);
 }
 
 // Use topic or HTML filename
