Index: include/wx/string.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/string.h,v
retrieving revision 1.300
diff -u -t -p -r1.300 string.h
--- include/wx/string.h	2007/05/17 15:31:55	1.300
+++ include/wx/string.h	2007/05/19 19:40:43
@@ -1311,7 +1311,7 @@ public:
   // overloaded assignment
     // from another wxString
   wxString& operator=(const wxString& stringSrc)
-    { m_impl = stringSrc.m_impl; return *this; }
+    { wxASSERT_MSG(this != &stringSrc, wxT("self assignment!")); if (this == &stringSrc) return *this; m_impl = stringSrc.m_impl; return *this; }
   wxString& operator=(const wxCStrData& cstr)
     { return *this = cstr.AsString(); }
     // from a character

 	  	 
