? mypatch.patch
Index: include/wx/dbgrid.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/dbgrid.h,v
retrieving revision 1.7
diff -b -u -2 -r1.7 dbgrid.h
--- include/wx/dbgrid.h	2002/08/31 11:29:09	1.7
+++ include/wx/dbgrid.h	2002/11/07 22:12:08
@@ -79,4 +79,16 @@
     int Length() { return (m_next ? m_next->Length() +1 :  1); }
 
+    // Adds a new column info (2 step creation)
+    void AddColInfo (int colNo,
+                    wxString type,
+                    wxString title)
+    {
+        GetLast()->m_next = new wxDbGridColInfo (colNo, type, title, NULL);
+    }
+
+    // Searches last
+    wxDbGridColInfo *GetLast() { return (m_next ? m_next->GetLast() : this); }
+
+
     protected:
     wxDbGridColInfoBase  m_data;

 	  	 
