diff -uNr wxWidgets-orig/include/wx/osx/cocoa/private/textimpl.h wxWidgets-mine/include/wx/osx/cocoa/private/textimpl.h
--- wxWidgets-orig/include/wx/osx/cocoa/private/textimpl.h	2015-10-08 04:48:38.000000000 +0200
+++ wxWidgets-mine/include/wx/osx/cocoa/private/textimpl.h	2015-10-21 15:23:43.000000000 +0200
@@ -62,7 +62,7 @@
 class wxNSTextViewControl : public wxWidgetCocoaImpl, public wxTextWidgetImpl
 {
 public:
-    wxNSTextViewControl( wxTextCtrl *wxPeer, WXWidget w );
+    wxNSTextViewControl( wxTextCtrl *wxPeer, WXWidget w, long style );
     virtual ~wxNSTextViewControl();
 
     virtual void insertText(NSString* text, WXWidget slf, void *_cmd);
diff -uNr wxWidgets-orig/src/osx/cocoa/textctrl.mm wxWidgets-mine/src/osx/cocoa/textctrl.mm
--- wxWidgets-orig/src/osx/cocoa/textctrl.mm	2015-10-08 04:48:38.000000000 +0200
+++ wxWidgets-mine/src/osx/cocoa/textctrl.mm	2015-10-21 15:17:43.000000000 +0200
@@ -30,7 +30,7 @@
     #include <sys/types.h>
     #include <sys/stat.h>
 #else
-    #include <stat.h>
+    #include <stat.h>
 #endif
 
 #if wxUSE_STD_IOSTREAM
@@ -560,7 +560,7 @@
 
 // wxNSTextViewControl
 
-wxNSTextViewControl::wxNSTextViewControl( wxTextCtrl *wxPeer, WXWidget w )
+wxNSTextViewControl::wxNSTextViewControl( wxTextCtrl *wxPeer, WXWidget w, long style )
     : wxWidgetCocoaImpl(wxPeer, w),
       wxTextWidgetImpl(wxPeer)
 {
@@ -579,6 +579,15 @@
     [tv setVerticallyResizable:YES];
     [tv setHorizontallyResizable:NO];
     [tv setAutoresizingMask:NSViewWidthSizable];
+    
+    if ( style & wxTE_RIGHT)
+    {
+        [tv setAlignment:NSRightTextAlignment];
+    }
+    else if ( style & wxTE_CENTRE)
+    {
+        [tv setAlignment:NSCenterTextAlignment];
+    }
 
     if ( !wxPeer->HasFlag(wxTE_RICH | wxTE_RICH2) )
     {
@@ -1063,7 +1072,7 @@
     {
         wxNSTextScrollView* v = nil;
         v = [[wxNSTextScrollView alloc] initWithFrame:r];
-        c = new wxNSTextViewControl( wxpeer, v );
+        c = new wxNSTextViewControl( wxpeer, v, style );
         c->SetNeedsFocusRect( true );
     }
     else
