--- wxWindows-0513\include\wx\datetime.h	Thu May 15 14:17:42 2003
+++ wxWindows\include\wx\datetime.h	Fri Jun 13 15:54:26 2003
@@ -1305,6 +1305,9 @@
 class WXDLLEXPORT wxDateTimeHolidayAuthority
 {
-friend class wxDateTimeHolidaysModule;
-public:
+    friend class wxDateTimeHolidaysModule;
+ public:
+    // This allows any inherited authority modules to clean up. 
+    virtual ~wxDateTimeHolidayAuthority() {}
+
     // returns TRUE if the given date is a holiday
     static bool IsHoliday(const wxDateTime& dt);

 	  	 
