Index: src/osx/listbox_osx.cpp
===================================================================
--- src/osx/listbox_osx.cpp	(revision 61338)
+++ src/osx/listbox_osx.cpp	(working copy)
@@ -314,7 +314,10 @@
 
 int wxListBox::FindString(const wxString& s, bool bCase) const
 {
-    return IsSorted() ? m_strings.sorted->Index(s, bCase)
+    wxASSERT_MSG( !IsSorted() || bCase,
+                  "case-insensitive FindString() is not supported by sorted wxListBoxes" );
+
+    return IsSorted() ? m_strings.sorted->Index(s)
                       : m_strings.unsorted->Index(s, bCase);
 }
 
Index: include/wx/arrstr.h
===================================================================
--- include/wx/arrstr.h	(revision 61338)
+++ include/wx/arrstr.h	(working copy)
@@ -86,9 +86,14 @@
             Add(src[n]);
     }
 
-    int Index(const wxString& str, bool bCase = true, bool bFromEnd = false) const;
+    int Index(const wxString &str) const
+    {
+        return Index(str,true,false);
+    }
 
 private:
+    int Index(const wxString& str, bool bCase, bool bFromEnd) const;
+
     void Insert()
     {
         wxFAIL_MSG( "wxSortedArrayString::Insert() is not to be used" );
Index: interface/wx/arrstr.h
===================================================================
--- interface/wx/arrstr.h	(revision 61338)
+++ interface/wx/arrstr.h	(working copy)
@@ -300,8 +300,17 @@
     /**
         @copydoc wxArrayString::Index()
 
-        This function uses binary search for wxSortedArrayString, but it ignores
-        the @a bCase and @a bFromEnd parameters.
+        This function uses binary search for wxSortedArrayString.
+
+        @warning When used with a wxSortedArrayString, this function ignores
+                 the @a bCase and @a bFromEnd parameters: you should only
+                 pass a single argument explicitly (@a sz). If the @a bCase and
+                 @a bFromEnd parameters are given but do not match the default
+                 arguments, a wxASSERT_MSG is triggered when wx is in debug
+                 mode.
+
+        @warning In STL mode, this function does not take @a bCase and
+                 @a bFromEnd parameters.
     */
     int Index(const wxString& sz, bool bCase = true,
               bool bFromEnd = false) const;
