Index: samples/minimal/minimal.cpp
===================================================================
--- samples/minimal/minimal.cpp	(revision 62459)
+++ samples/minimal/minimal.cpp	(working copy)
@@ -141,6 +141,8 @@
 // main frame
 // ----------------------------------------------------------------------------
 
+#include <wx/checklst.h>
+
 // frame constructor
 MyFrame::MyFrame(const wxString& title)
        : wxFrame(NULL, wxID_ANY, title)
@@ -148,6 +150,11 @@
     // set the frame icon
     SetIcon(wxICON(sample));
 
+    wxCheckListBox* clb = new wxCheckListBox(this, wxID_ANY);
+    clb->Append("Checkable Item 1");
+    clb->Append("Checkable Item 2");
+    clb->Append("Checkable Item 3");
+
 #if wxUSE_MENUS
     // create a menu bar
     wxMenu *fileMenu = new wxMenu;
@@ -156,6 +163,12 @@
     wxMenu *helpMenu = new wxMenu;
     helpMenu->Append(Minimal_About, "&About...\tF1", "Show about dialog");
 
+    fileMenu->AppendRadioItem(wxID_ANY, "Disabled");
+    fileMenu->AppendRadioItem(wxID_ANY, "Almost Enabled");
+    fileMenu->AppendRadioItem(wxID_ANY, "Enabled");
+
+    fileMenu->AppendSeparator();
+
     fileMenu->Append(Minimal_Quit, "E&xit\tAlt-X", "Quit this program");
 
     // now append the freshly created menu to the menu bar...
