Index: wxWidgets/include/wx/msw/icon.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/msw/icon.h,v
retrieving revision 1.16
diff -b -u -2 -r1.16 icon.h
--- wxWidgets/include/wx/msw/icon.h	2004/05/23 20:51:28	1.16
+++ wxWidgets/include/wx/msw/icon.h	2004/09/11 20:06:48
@@ -97,4 +97,6 @@
     }
 
+    virtual wxObjectRefData *CloneRefData(const wxObjectRefData *data) const;
+
     // create from XPM data
     void CreateIconFromXpm(const char **data);
Index: wxWidgets/src/msw/icon.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/msw/icon.cpp,v
retrieving revision 1.32
diff -b -u -2 -r1.32 icon.cpp
--- wxWidgets/src/msw/icon.cpp	2004/08/27 18:59:37	1.32
+++ wxWidgets/src/msw/icon.cpp	2004/09/11 20:06:49
@@ -103,4 +103,23 @@
 }
 
+wxObjectRefData *wxIcon::CloneRefData(const wxObjectRefData *dataOrig) const
+{
+    const wxIconRefData *
+        data = wx_static_cast(const wxIconRefData *, dataOrig);
+    if ( !data )
+        return NULL;
+
+    wxIcon *self = wx_const_cast(wxIcon *, this);
+	self->UnRef();
+	self->m_refData = new wxIconRefData(*data);
+
+    if ( data->m_hIcon )
+    {
+		::CopyIcon(HICON(data->m_hIcon));
+    }
+
+    return m_refData;
+}
+
 void wxIcon::CopyFromBitmap(const wxBitmap& bmp)
 {

 	  	 
