Index: include/wx/msw/missing.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/msw/missing.h,v
retrieving revision 1.71
diff -u -r1.71 missing.h
--- include/wx/msw/missing.h	2006/10/29 22:46:32	1.71
+++ include/wx/msw/missing.h	2007/03/05 23:54:35
@@ -11,6 +11,8 @@
 #ifndef _WX_MISSING_H_
 #define _WX_MISSING_H_
 
+#include "wx/msw/private.h"     // Ensure <windows.h> has been included
+
 /*
  * The following are required for VC++ 6.
  */

 	  	 
