Index: WX_2_8_BRANCH/src/mac/carbon/taskbar.cpp
===================================================================
--- WX_2_8_BRANCH/src/mac/carbon/taskbar.cpp	(revision 49604)
+++ WX_2_8_BRANCH/src/mac/carbon/taskbar.cpp	(working copy)
@@ -180,8 +180,13 @@
                 taskbarMenuRef,
                 command.commandID,
                 1, &menuRef, &menuItemIndex );
-            if (err == noErr)
+            if (err != noErr)
             {
+                // Not a taskbar menu item - pass it back to wxApp
+                return eventNotHandledErr;
+            }
+            else
+            {
                 MenuCommand id = command.commandID;
                 wxMenuItem *item = NULL;
 

 	  	 
