Index: samples/docview/view.cpp
===================================================================
--- samples/docview/view.cpp	(revision 62447)
+++ samples/docview/view.cpp	(working copy)
@@ -118,7 +118,7 @@
 // Clean up windows used for displaying the view.
 bool DrawingView::OnClose(bool deleteWindow)
 {
-    if (!GetDocument()->Close())
+    if (!wxView::OnClose(deleteWindow))
         return false;
     
     // Clear the canvas in  case we're in single-window mode,
@@ -184,7 +184,7 @@
 
 bool TextEditView::OnClose(bool deleteWindow)
 {
-    if (!GetDocument()->Close())
+    if (!wxView::OnClose(deleteWindow))
         return false;
     
     Activate(false);
Index: samples/docvwmdi/view.cpp
===================================================================
--- samples/docvwmdi/view.cpp	(revision 62447)
+++ samples/docvwmdi/view.cpp	(working copy)
@@ -97,7 +97,7 @@
 // Clean up windows used for displaying the view.
 bool DrawingView::OnClose(bool deleteWindow)
 {
-  if (!GetDocument()->Close())
+  if (!wxView::OnClose(deleteWindow))
     return false;
 
   // Clear the canvas in  case we're in single-window mode,
@@ -163,7 +163,7 @@
 
 bool TextEditView::OnClose(bool deleteWindow)
 {
-  if (!GetDocument()->Close())
+  if (!wxView::OnClose(deleteWindow))
     return false;
 
   Activate(false);
