Index: src/common/fileconf.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/common/fileconf.cpp,v
retrieving revision 1.165
diff -u -r1.165 fileconf.cpp
--- src/common/fileconf.cpp	2007/03/01 20:03:50	1.165
+++ src/common/fileconf.cpp	2007/03/05 23:13:35
@@ -496,13 +496,13 @@
     // termination character
     wxString strTrans;
     {
-        wxString strTmp;
+        wxMemoryBuffer bufTmp;
 
-        char buf[1024];
         do
         {
-            inStream.Read(buf, WXSIZEOF(buf)-1);  // leave room for the NULL
-
+            inStream.Read(bufTmp.GetAppendBuf(1024), 1024);
+            bufTmp.UngetAppendBuf(inStream.LastRead());
+            
             const wxStreamError err = inStream.GetLastError();
 
             if ( err != wxSTREAM_NO_ERROR && err != wxSTREAM_EOF )
@@ -510,17 +510,12 @@
                 wxLogError(_("Error reading config options."));
                 break;
             }
-
-            // FIXME: this is broken because if we have part of multibyte
-            //        character in the buffer (and another part hasn't been
-            //        read yet) we're going to lose data because of conversion
-            //        errors
-            buf[inStream.LastRead()] = '\0';
-            strTmp += conv.cMB2WX(buf);
         }
         while ( !inStream.Eof() );
+
+        bufTmp.AppendByte(0x00);
 
-        strTrans = wxTextBuffer::Translate(strTmp);
+        strTrans = wxTextBuffer::Translate(conv.cMB2WX((char*)bufTmp.GetData()));
     }
 
     wxMemoryText memText;

 	  	 
