diff -uNr wxWidgets-orig/include/wx/aui/framemanager.h wxWidgets-changed/include/wx/aui/framemanager.h
--- wxWidgets-orig/include/wx/aui/framemanager.h	2009-03-05 16:20:17.000000000 -0500
+++ wxWidgets-changed/include/wx/aui/framemanager.h	2009-11-05 12:58:07.906250000 -0500
@@ -551,7 +551,7 @@
     void OnFloatingPaneMoved(wxWindow* window, wxDirection dir);
     void OnFloatingPaneActivated(wxWindow* window);
     void OnFloatingPaneClosed(wxWindow* window, wxCloseEvent& evt);
-    void OnFloatingPaneResized(wxWindow* window, const wxSize& size);
+    void OnFloatingPaneResized(wxWindow* window, const wxRect& rect);
     void Render(wxDC* dc);
     void Repaint(wxDC* dc = NULL);
     void ProcessMgrEvent(wxAuiManagerEvent& event);
diff -uNr wxWidgets-orig/src/aui/floatpane.cpp wxWidgets-changed/src/aui/floatpane.cpp
--- wxWidgets-orig/src/aui/floatpane.cpp	2009-04-30 23:30:21.000000000 -0400
+++ wxWidgets-changed/src/aui/floatpane.cpp	2009-11-05 12:52:12.625000000 -0500
@@ -157,11 +157,11 @@
 }
 
 
-void wxAuiFloatingFrame::OnSize(wxSizeEvent& event)
+void wxAuiFloatingFrame::OnSize(wxSizeEvent& WXUNUSED(event))
 {
     if (m_owner_mgr)
     {
-        m_owner_mgr->OnFloatingPaneResized(m_pane_window, event.GetSize());
+        m_owner_mgr->OnFloatingPaneResized(m_pane_window, GetRect());
     }
 }
 
diff -uNr wxWidgets-orig/src/aui/framemanager.cpp wxWidgets-changed/src/aui/framemanager.cpp
--- wxWidgets-orig/src/aui/framemanager.cpp	2009-10-08 10:05:57.000000000 -0400
+++ wxWidgets-changed/src/aui/framemanager.cpp	2009-11-05 12:56:32.109375000 -0500
@@ -3777,13 +3777,16 @@
     HideHint();
 }
 
-void wxAuiManager::OnFloatingPaneResized(wxWindow* wnd, const wxSize& size)
+void wxAuiManager::OnFloatingPaneResized(wxWindow* wnd, const wxRect& rect)
 {
     // try to find the pane
     wxAuiPaneInfo& pane = GetPane(wnd);
     wxASSERT_MSG(pane.IsOk(), wxT("Pane window not found"));
 
-    pane.floating_size = size;
+    pane.FloatingSize(rect.GetWidth(), rect.GetHeight());
+
+    // the top-left position may change as well as the size
+    pane.FloatingPosition(rect.x, rect.y); 
 }
 
 
