Index: src/msw/fontenum.cpp
===================================================================
--- src/msw/fontenum.cpp	(revision 66991)
+++ src/msw/fontenum.cpp	(working copy)
@@ -99,7 +99,7 @@
 
 #ifndef __WXMICROWIN__
 int CALLBACK wxFontEnumeratorProc(LPLOGFONT lplf, LPTEXTMETRIC lptm,
-                                  DWORD dwStyle, LONG lParam);
+                                  DWORD dwStyle, LPARAM lParam);
 #endif
 
 // ============================================================================
@@ -269,7 +269,7 @@
 
 #ifndef __WXMICROWIN__
 int CALLBACK wxFontEnumeratorProc(LPLOGFONT lplf, LPTEXTMETRIC lptm,
-                                  DWORD WXUNUSED(dwStyle), LONG lParam)
+                                  DWORD WXUNUSED(dwStyle), LPARAM lParam)
 {
 
     // we used to process TrueType fonts only, but there doesn't seem to be any
