Index: src/msw/mimetype.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/msw/mimetype.cpp,v
retrieving revision 1.34
diff -u -r1.34 mimetype.cpp
--- src/msw/mimetype.cpp	2005/09/23 12:55:05	1.34
+++ src/msw/mimetype.cpp	2006/03/13 08:02:44
@@ -509,7 +509,7 @@
                  _T("Associate() needs extension") );
 
     bool ok;
-    int iExtCount = 0 ;
+    size_t iExtCount = 0 ;
     wxString filetype;
     wxString extWithDot;
 


 	  	 
