--- wxrcp.cpp	2008-06-26 12:06:39.000000000 +0200
+++ wxrc.cpp	2008-06-26 12:15:23.000000000 +0200
@@ -32,6 +32,11 @@
 
 WX_DECLARE_HASH_SET(wxString, wxStringHash, wxStringEqual, StringSet);
 
+static wxString GenerateXRCIDStaticMacro(const wxString& name)
+{
+  return wxString::Format("#ifndef XRCID_%s\n#define XRCID_%s XRCID(\"%s\")\n#endif\n", name.c_str(), name.c_str(), name.c_str());
+}
+
 class XRCWidgetData
 {
 public:
@@ -158,6 +163,7 @@
                         + _T("\",")
                         + w.GetClass()
                         + _T(");\n"));
+            file.Write(GenerateXRCIDStaticMacro(w.GetName()));
         }
         file.Write(_T(" }\n"));
 
@@ -192,6 +198,8 @@
             }
             file.Write(_T("};\n"));
         }
+        //XXX Using the m_className since it's what contains the name of the object.
+        file.Write(GenerateXRCIDStaticMacro(m_className));
     }
 };
 WX_DECLARE_OBJARRAY(XRCWndClassData,ArrayOfXRCWndClassData);
