--- orig3.wxrc.cpp	2008-06-25 10:14:51.000000000 +0200
+++ wxrc.cpp	2008-06-25 11:51:41.000000000 +0200
@@ -47,6 +47,7 @@
 #include "wx/arrimpl.cpp"
 WX_DECLARE_OBJARRAY(XRCWidgetData,ArrayOfXRCWidgetData);
 WX_DEFINE_OBJARRAY(ArrayOfXRCWidgetData)
+WX_DECLARE_STRING_HASH_MAP( wxString, HashOfXrcIdMacro );
 
 class XRCWndClassData
 {
@@ -55,6 +56,7 @@
     wxString m_parentClassName;
     StringSet m_ancestorClassNames;
     ArrayOfXRCWidgetData m_wdata;
+    HashOfXrcIdMacro hXrcId;
 
     void BrowseXmlNode(wxXmlNode* node)
     {
@@ -68,6 +70,15 @@
                 && node->GetPropVal(_T("name"),&nameValue))
             {
                 m_wdata.Add(XRCWidgetData(nameValue,classValue));
+                if ( hXrcId.find(nameValue) == hXrcId.end() )
+                {
+                    wxString staticXrcid = _T("XRCID_") + nameValue;
+                    hXrcId[nameValue] = _T("#ifndef ")+ staticXrcid +_T("\n#define ") + staticXrcid + _T("() XRCID(\"") + nameValue + _T("\")\n#endif");
+                }
+                else
+                {
+                    wxLogWarning(wxT("Hope that you what you are doing when defining the same XRC_ID in your resources ! (") + nameValue + wxT(")"));
+                }
             }
             children = node->GetChildren();
             if (children)
@@ -193,6 +204,14 @@
                             _T("};\n"));
             }
         }
+        file.Write( _T("\n") );
+        HashOfXrcIdMacro::iterator ithXrcIdCurrent =  hXrcId.begin();
+        HashOfXrcIdMacro::iterator ithXrcIdEnd = hXrcId.end();
+        while(ithXrcIdCurrent != ithXrcIdEnd)
+        {
+          file.Write( ithXrcIdCurrent->second + _T("\n"));
+          ithXrcIdCurrent++;
+        }
     }
 };
 WX_DECLARE_OBJARRAY(XRCWndClassData,ArrayOfXRCWndClassData);
