--- orig3.wxrc.cpp	2008-06-25 10:14:51.000000000 +0200
+++ wxrc.cpp	2008-06-25 16:21:43.000000000 +0200
@@ -32,6 +32,11 @@
 
 WX_DECLARE_HASH_SET(wxString, wxStringHash, wxStringEqual, StringSet);
 
+static wxString FormatXRCIDString(const wxString& name)
+{
+  return wxString::Format("#ifndef XRCID_%s\n#define XRCID_%s XRCID(\"%s\")\n#endif\n", name.c_str(), name.c_str(), name.c_str());
+}
+
 class XRCWidgetData
 {
 public:
@@ -193,6 +198,13 @@
                             _T("};\n"));
             }
         }
+        file.Write( _T("\n") );
+        for(i=0;i<m_wdata.Count();++i)
+        {
+            const XRCWidgetData& w = m_wdata.Item(i);
+            file.Write(FormatXRCIDString(w.GetName()));
+        }
+        file.Write(FormatXRCIDString(m_className));
     }
 };
 WX_DECLARE_OBJARRAY(XRCWndClassData,ArrayOfXRCWndClassData);
