diff --git a/samples/minimal/minimal.cpp b/samples/minimal/minimal.cpp
index a78e462..7104da1 100644
--- a/samples/minimal/minimal.cpp
+++ b/samples/minimal/minimal.cpp
@@ -29,7 +29,8 @@
 #ifndef WX_PRECOMP
     #include "wx/wx.h"
 #endif
-
+#include "wx/aui/aui.h"
+#include "wx/stc/stc.h"
 // ----------------------------------------------------------------------------
 // resources
 // ----------------------------------------------------------------------------
@@ -172,6 +173,26 @@ MyFrame::MyFrame(const wxString& title)
     CreateStatusBar(2);
     SetStatusText("Welcome to wxWidgets!");
 #endif // wxUSE_STATUSBAR
+
+    wxAuiManager* m_auimgr11 = new wxAuiManager;
+    m_auimgr11->SetManagedWindow( this );
+    m_auimgr11->SetFlags( wxAUI_MGR_LIVE_RESIZE|wxAUI_MGR_TRANSPARENT_HINT|wxAUI_MGR_TRANSPARENT_DRAG|wxAUI_MGR_ALLOW_ACTIVE_PANE|wxAUI_MGR_ALLOW_FLOATING);
+    m_auimgr11->GetArtProvider()->SetMetric(wxAUI_DOCKART_GRADIENT_TYPE, wxAUI_GRADIENT_NONE);
+    
+    wxPanel* m_panel13 = new wxPanel(this, wxID_ANY, wxDefaultPosition, wxSize(-1,-1), wxTAB_TRAVERSAL);
+    
+    m_auimgr11->AddPane(m_panel13, wxAuiPaneInfo().Direction(wxAUI_DOCK_LEFT).Layer(0).Row(0).Position(0).BestSize(300,300).MinSize(300,300).MaxSize(300,300).CaptionVisible(true).MaximizeButton(false).CloseButton(false).MinimizeButton(false).PinButton(false));
+    m_auimgr11->Update();
+    
+    wxBoxSizer* boxSizer15 = new wxBoxSizer(wxVERTICAL);
+    m_panel13->SetSizer(boxSizer15);
+    
+    wxStyledTextCtrl* m_stc17 = new wxStyledTextCtrl(m_panel13, wxID_ANY, wxDefaultPosition, wxSize(-1,-1), 0);
+    boxSizer15->Add(m_stc17, 1, wxALL|wxEXPAND, 5);
+    
+    wxTextCtrl* m_textCtrl19 = new wxTextCtrl(m_panel13, wxID_ANY, wxT(""), wxDefaultPosition, wxSize(-1,-1), 0);
+    boxSizer15->Add(m_textCtrl19, 0, wxALL|wxEXPAND, 5);
+
 }
 
 
