Index: dataview.cpp
===================================================================
--- dataview.cpp	(revision 75104)
+++ dataview.cpp	(working copy)
@@ -563,6 +563,7 @@
                                       wxDATAVIEW_COL_RESIZABLE );
             column1->SetMinWidth(150); // this column can't be resized to be smaller
             m_ctrl[0]->AppendColumn( column1 );
+            column1->SetSortOrder(false);
 
             // column 2 of the view control:
 
