Index: src/common/tbarbase.cpp
===================================================================
--- src/common/tbarbase.cpp	(revisione 72833)
+++ src/common/tbarbase.cpp	(copia locale)
@@ -129,6 +129,9 @@
 #if wxUSE_MENUS
 void wxToolBarToolBase::SetDropdownMenu(wxMenu* menu)
 {
+    wxCHECK_RET( GetKind() == wxITEM_DROPDOWN,
+                wxT("menu can be only associated with drop down tools") );
+
     delete m_dropdownMenu;
     m_dropdownMenu = menu;
 }
@@ -736,9 +739,6 @@
     wxToolBarToolBase * const tool = FindById(toolid);
     wxCHECK_MSG( tool, false, wxT("invalid tool toolid") );
 
-    wxCHECK_MSG( tool->GetKind() == wxITEM_DROPDOWN, false,
-                    wxT("menu can be only associated with drop down tools") );
-
     tool->SetDropdownMenu(menu);
 
     return true;
