Index: include/wx/fontmap.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/fontmap.h,v
retrieving revision 1.13
diff -b -u -2 -r1.13 fontmap.h
--- include/wx/fontmap.h	2003/05/01 16:52:14	1.13
+++ include/wx/fontmap.h	2003/05/31 09:44:37
@@ -205,5 +205,5 @@
 #if wxUSE_GUI
     // wxEncodingToCodepage (utils.cpp) needs wxGetNativeFontEncoding
-    #include "wx/fontenc.h"
+    #include "wx/fontutil.h"
 #endif
 

 	  	 
