diff -uNr wxWidgets-trunk/samples/exec/exec.cpp wxWidgets-mine/samples/exec/exec.cpp
--- wxWidgets-trunk/samples/exec/exec.cpp	2013-05-15 07:23:12.309266000 -0400
+++ wxWidgets-mine/samples/exec/exec.cpp	2013-05-15 07:26:41.488921000 -0400
@@ -962,11 +962,10 @@
         wxLogStatus("Command \"%s\" terminated after %ldms; exit code %d.",
                     cmd, sw.Time(), code);
 
-        if ( code != -1 )
-        {
-            ShowOutput(cmd, output, wxT("Output"));
-            ShowOutput(cmd, errors, wxT("Errors"));
-        }
+        // It is valid for a program to do exit(-1), so do not skip
+        // trying to show the program output if return value is -1.
+        ShowOutput(cmd, output, wxT("Output"));
+        ShowOutput(cmd, errors, wxT("Errors"));
     }
     else // async exec
     {
diff -uNr wxWidgets-trunk/src/common/utilscmn.cpp wxWidgets-mine/src/common/utilscmn.cpp
--- wxWidgets-trunk/src/common/utilscmn.cpp	2013-05-14 16:40:54.098956000 -0400
+++ wxWidgets-mine/src/common/utilscmn.cpp	2013-05-15 07:28:41.016228000 -0400
@@ -671,17 +671,18 @@
     long rc = wxExecute(command, wxEXEC_SYNC | flags, process, env);
 
 #if wxUSE_STREAMS
-    if ( rc != -1 )
-    {
+    // It is valid for a program to output messages and do exit(-1),
+    // so do not skip trying to read messgaes if return value is -1.
+    // Instead check for non-null streams before trying to read them.
+    if ( process->GetInputStream() )
         if ( !ReadAll(process->GetInputStream(), output) )
             rc = -1;
 
-        if ( error )
-        {
+    if ( error )
+    {
+        if ( process->GetErrorStream() )
             if ( !ReadAll(process->GetErrorStream(), *error) )
                 rc = -1;
-        }
-
     }
 #else
     wxUnusedVar(output);
