diff --git a/include/wx/msw/setup0.h b/include/wx/msw/setup0.h
index 9ac46699d6..599e631bdb 100644
--- a/include/wx/msw/setup0.h
+++ b/include/wx/msw/setup0.h
@@ -195,6 +195,29 @@
 // wxUSE_WCHAR_T is required by wxWidgets now, don't change.
 #define wxUSE_WCHAR_T 1
 
+
+// ----------------------------------------------------------------------------
+// Compiler detection
+// ----------------------------------------------------------------------------
+
+// If using a MinGW compiler see if we are using MinGW-w64.
+
+#if ( defined( __MINGW32__ ) )
+    #include "wx/msw/winver.h"
+
+    #include <_mingw.h>
+
+    #ifdef __MINGW64_VERSION_MAJOR
+        #ifndef __MINGW64_TOOLCHAIN__
+            #define __MINGW64_TOOLCHAIN__
+        #endif
+    #else
+        #ifndef __MINGW32_TOOLCHAIN__
+            #define __MINGW32_TOOLCHAIN__
+        #endif
+    #endif
+#endif
+
 // ----------------------------------------------------------------------------
 // global features
 // ----------------------------------------------------------------------------
@@ -794,7 +817,7 @@
 
 // notice that we can't use wxCHECK_VISUALC_VERSION() here as this file is
 // included from wx/platform.h before wxCHECK_VISUALC_VERSION() is defined
-#ifdef _MSC_VER
+#if defined(_MSC_VER) || defined(__MINGW64_TOOLCHAIN__)
 #   define wxUSE_GRAPHICS_CONTEXT 1
 #else
     // Disable support for other Windows compilers, enable it if your compiler
