diff --git a/samples/collpane/collpane.cpp b/samples/collpane/collpane.cpp
index 27cccb9..31c5a6d 100644
--- a/samples/collpane/collpane.cpp
+++ b/samples/collpane/collpane.cpp
@@ -43,6 +43,7 @@
 #include "wx/filepicker.h"
 #include "wx/fontpicker.h"
 #include "wx/aboutdlg.h"
+#include "wx/odcombo.h"
 
 #ifndef wxHAS_IMAGES_IN_RESOURCES
     #include "../sample.xpm"
@@ -208,6 +209,20 @@ bool MyApp::OnInit()
     paneSubSizer->Add( new wxTextCtrl(win, PANE_TEXTCTRL, wxT("Text control"), wxDefaultPosition, wxSize(80,-1) ), 0, wxALIGN_LEFT | wxALL, 3 );
     paneSubSizer->Add( new wxButton(win, PANE_BUTTON, wxT("Press to align right") ), 0, wxALIGN_LEFT | wxALL, 3 );
 
+    wxOwnerDrawnComboBox* odc = new wxOwnerDrawnComboBox(win,wxID_ANY,wxEmptyString,
+                                    wxDefaultPosition, wxDefaultSize,
+                                    0,
+                                    NULL,
+                                    wxCB_SIMPLE
+                                   );
+    odc->Disable();
+    m_paneSizer->Add(odc, 0, wxALIGN_LEFT);
+    win->SetSizer( m_paneSizer );
+    odc->Append("Solid");
+    odc->Append("Transparent");
+    odc->Enable();
+    odc->SetSelection(0);
+    odc->SetFocus();
     win->SetSizer( m_paneSizer );
 }
 
