diff -ru -U 2 -xCVS -x'.#*' -x'*.orig' -x'*.rej' -x'*~' wxWindows.cvs/src/generic/treectlg.cpp wxWindows/src/generic/treectlg.cpp
--- wxWindows.cvs/src/generic/treectlg.cpp	2005-01-28 11:53:30.000000000 -0600
+++ wxWindows/src/generic/treectlg.cpp	2005-01-28 12:40:07.170153632 -0600
@@ -389,7 +389,13 @@
 
     if ( value == m_startValue )
     {
-        // nothing changed, always accept
+        // when an item remains unchanged, the owner
+        // needs to be notified that the user decided
+        // not to change the tree item label, and that
+        // the edit has been cancelled
+
+        m_owner->OnRenameCancelled(m_itemEdited);
+
         return true;
     }
 

 	  	 
