diff -ru wxWidgets.orig/src/msw/frame.cpp wxWidgets/src/msw/frame.cpp
--- wxWidgets.orig/src/msw/frame.cpp	2005-09-25 19:15:36.000000000 +1000
+++ wxWidgets/src/msw/frame.cpp	2005-10-11 17:21:21.823560352 +1000
@@ -518,8 +518,27 @@
         }
 #endif // wxUSE_TOOLBAR
 
-        if ((m_fsStyle & wxFULLSCREEN_NOMENUBAR) && m_hMenu)
-            ::SetMenu(GetHwnd(), (HMENU)m_hMenu);
+        if (m_fsStyle & wxFULLSCREEN_NOMENUBAR)
+        {
+            WXHMENU menu = m_hMenu;
+
+#if wxUSE_MDI_ARCHITECTURE
+            wxMDIParentFrame *frame = wxDynamicCast(this, wxMDIParentFrame);
+            if (frame)
+            {
+                wxMDIChildFrame *child = frame->GetActiveChild();
+                if (child)
+                {
+                    menu = child->GetWinMenu();
+                }
+            }
+#endif // wxUSE_MDI_ARCHITECTURE
+
+            if (menu)
+            {
+                ::SetMenu(GetHwnd(), (HMENU)menu);
+            }
+        }
 
 #if wxUSE_STATUSBAR
         wxStatusBar *theStatusBar = GetStatusBar();
@@ -954,7 +973,7 @@
             SHACTIVATEINFO* info = (SHACTIVATEINFO*) m_activateInfo;
             if (info)
                 SHHandleWMActivate(GetHwnd(), wParam, lParam, info, FALSE);
-            
+
             // This implicitly sends a wxEVT_ACTIVATE_APP event
             if (wxTheApp)
                 wxTheApp->SetActive(wParam != 0, FindFocus());

 	  	 
