Index: docs/latex/wx/htmltags.tex
===================================================================
--- docs/latex/wx/htmltags.tex	(révision 63657)
+++ docs/latex/wx/htmltags.tex	(copie de travail)
@@ -98,8 +98,8 @@
 H6
 I
 IMG             SRC=[url]
-                WIDTH=[pixels]
-                HEIGHT=[pixels]
+                WIDTH=[percent|pixels|AUTO]
+                HEIGHT=[percent|pixels|AUTO]
                 ALIGN=TEXTTOP
                 ALIGN=CENTER
                 ALIGN=ABSCENTER
Index: src/html/m_image.cpp
===================================================================
--- src/html/m_image.cpp	(révision 63657)
+++ src/html/m_image.cpp	(copie de travail)
@@ -286,7 +286,7 @@
 {
 public:
     wxHtmlImageCell(wxHtmlWindowInterface *windowIface,
-                    wxFSFile *input, int w = wxDefaultCoord, int h = wxDefaultCoord,
+                    wxFSFile *input, int w = wxDefaultCoord, bool wpc = false, int h = wxDefaultCoord, 	bool hpc = false,
                     double scale = 1.0, int align = wxHTML_ALIGN_BOTTOM,
                     const wxString& mapname = wxEmptyString);
     virtual ~wxHtmlImageCell();
@@ -303,6 +303,7 @@
 private:
     wxBitmap           *m_bitmap;
     int                 m_bmpW, m_bmpH;
+	bool                m_bmpWpc, m_bmpHpc;
     bool                m_showFrame:1;
     wxHtmlWindowInterface *m_windowIface;
 #if wxUSE_GIF && wxUSE_TIMER
@@ -343,15 +344,17 @@
 
 wxHtmlImageCell::wxHtmlImageCell(wxHtmlWindowInterface *windowIface,
                                  wxFSFile *input,
-                                 int w, int h, double scale, int align,
+                                 int w, bool wpc, int h, bool hpc, double scale, int align,
                                  const wxString& mapname) : wxHtmlCell()
 {
     m_windowIface = windowIface;
     m_scale = scale;
     m_showFrame = false;
     m_bitmap = NULL;
-    m_bmpW = w;
-    m_bmpH = h;
+    m_bmpW   = w;
+    m_bmpH   = h;
+	m_bmpWpc = wpc;
+	m_bmpHpc = hpc;
     m_imageMap = NULL;
     m_mapName = mapname;
     SetCanLiveOnPagebreak(false);
@@ -461,9 +464,9 @@
         ww = img.GetWidth();
         hh = img.GetHeight();
 
-        if ( m_bmpW == wxDefaultCoord )
+        if ( m_bmpW == wxDefaultCoord && !m_bmpWpc)
             m_bmpW = ww;
-        if ( m_bmpH == wxDefaultCoord )
+        if ( m_bmpH == wxDefaultCoord && !m_bmpHpc)
             m_bmpH = hh;
 
         // Only scale the bitmap at the rendering stage,
@@ -533,6 +536,20 @@
 
 void wxHtmlImageCell::Layout(int w)
 {
+	if (m_bmpWpc && m_bmpW >= 0)
+		m_Width = w*m_bmpW/100;
+	else 
+		m_Width = m_scale*m_bmpW;
+		
+	if (m_bmpHpc && m_bmpH < 0 && m_bitmap != NULL) {
+	
+		int orgw = m_bitmap->GetWidth();
+		int orgh = m_bitmap->GetHeight();
+		
+		m_Height = orgh*m_Width/orgw;
+		
+	}
+		
     wxHtmlCell::Layout(w);
     m_physX = m_physY = wxDefaultCoord;
 }
@@ -566,17 +583,17 @@
         // and height, so we only do the scaling once.
         double imageScaleX = 1.0;
         double imageScaleY = 1.0;
-        if (m_bmpW != m_bitmap->GetWidth())
-            imageScaleX = (double) m_bmpW / (double) m_bitmap->GetWidth();
-        if (m_bmpH != m_bitmap->GetHeight())
-            imageScaleY = (double) m_bmpH / (double) m_bitmap->GetHeight();
+        if (m_Width != m_bitmap->GetWidth())
+            imageScaleX = (double) m_Width / (double) m_bitmap->GetWidth();
+        if (m_Height != m_bitmap->GetHeight())
+            imageScaleY = (double) m_Height / (double) m_bitmap->GetHeight();
 
         double us_x, us_y;
         dc.GetUserScale(&us_x, &us_y);
-        dc.SetUserScale(us_x * m_scale * imageScaleX, us_y * m_scale * imageScaleY);
+        dc.SetUserScale(us_x * imageScaleX, us_y * imageScaleY);
 
-        dc.DrawBitmap(*m_bitmap, (int) ((x + m_PosX) / (m_scale*imageScaleX)),
-                                 (int) ((y + m_PosY) / (m_scale*imageScaleY)), true);
+        dc.DrawBitmap(*m_bitmap, (int) ((x + m_PosX) / (imageScaleX)),
+                                 (int) ((y + m_PosY) / (imageScaleY)), true);
         dc.SetUserScale(us_x, us_y);
     }
 }
@@ -627,6 +644,7 @@
             if (tag.HasParam(wxT("SRC")))
             {
                 int w = wxDefaultCoord, h = wxDefaultCoord;
+				bool wpc = false, hpc = false;
                 int al;
                 wxFSFile *str;
                 wxString tmp = tag.GetParam(wxT("SRC"));
@@ -634,10 +652,44 @@
 
                 str = m_WParser->OpenURL(wxHTML_URL_IMAGE, tmp);
 
-                if (tag.HasParam(wxT("WIDTH")))
-                    tag.GetParamAsInt(wxT("WIDTH"), &w);
-                if (tag.HasParam(wxT("HEIGHT")))
-                    tag.GetParamAsInt(wxT("HEIGHT"), &h);
+                if (tag.HasParam(wxT("WIDTH"))) 
+				{
+					wxString param = tag.GetParam(wxT("WIDTH"));
+					if (param.Upper() == wxT("AUTO")) 
+					{
+						w = -1; wpc = true;
+					} 
+					else if (param.EndsWith(wxT("%"))) 
+					{
+						wxSscanf(param.c_str(), wxT("%i"), &w);
+						wpc = true;
+					}
+					else
+					{
+						wxSscanf(param.c_str(), wxT("%i"), &w);
+					}
+						
+				}
+
+                if (tag.HasParam(wxT("HEIGHT"))) 
+				{
+					wxString param = tag.GetParam(wxT("HEIGHT"));
+					if (param.Upper() == wxT("AUTO")) 
+					{
+						h = -1; hpc = true;
+					} 
+					else if (param.EndsWith(wxT("%"))) 
+					{
+						if (wxSscanf(param.c_str(), wxT("%i"), &h) == 1)
+							hpc = true;
+					}
+					else
+					{
+						wxSscanf(param.c_str(), wxT("%i"), &h);
+					}
+						
+				}				
+				
                 al = wxHTML_ALIGN_BOTTOM;
                 if (tag.HasParam(wxT("ALIGN")))
                 {
@@ -658,7 +710,7 @@
                 }
                 wxHtmlImageCell *cel = new wxHtmlImageCell(
                                           m_WParser->GetWindowInterface(),
-                                          str, w, h,
+                                          str, w, wpc, h, hpc,
                                           m_WParser->GetPixelScale(),
                                           al, mn);
                 m_WParser->ApplyStateToCell(cel);
