Index: samples/printing/printing.cpp
===================================================================
--- samples/printing/printing.cpp	(revisione 69983)
+++ samples/printing/printing.cpp	(copia locale)
@@ -705,9 +705,9 @@
         {
             wxString word = words[i];
             word.Remove( word.Len()-1, 1 );
-            dc->GetTextExtent(word, &w, &h);
+            dc->GetTextExtent(word, &w, &h, NULL, NULL, &fnt);
             dc->DrawRectangle(x, y, w, h);
-            dc->GetTextExtent(words[i], &w, &h);
+            dc->GetTextExtent(words[i], &w, &h, NULL, NULL, &fnt);
             dc->DrawText(words[i], x, y);
             x += w;
         }
