Index: samples/printing/printing.cpp
===================================================================
--- samples/printing/printing.cpp	(revisione 69983)
+++ samples/printing/printing.cpp	(copia locale)
@@ -699,6 +699,10 @@
         long x = 200, y= 250;
         wxFont fnt(15, wxSWISS, wxNORMAL, wxNORMAL);
 
+        wxFont fnt2(12, wxSWISS, wxNORMAL, wxNORMAL);
+        dc->SetFont(fnt2);
+        dc->GetTextExtent("AAA", &w, &h, NULL, NULL, &fnt);
+
         dc->SetFont(fnt);
 
         for (int i = 0; i < 7; i++)
