Index: include/wx/generic/statusbr.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/generic/statusbr.h,v
retrieving revision 1.20
diff -u -r1.20 statusbr.h
--- include/wx/generic/statusbr.h	2004/03/07 17:28:28	1.20
+++ include/wx/generic/statusbr.h	2004/03/17 14:11:34
@@ -115,7 +115,6 @@
 
   int               m_borderX;
   int               m_borderY;
-  wxFont            m_defaultStatusBarFont;
   wxPen             m_mediumShadowPen;
   wxPen             m_hilightPen;
 
Index: src/generic/calctrl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/generic/calctrl.cpp,v
retrieving revision 1.51
diff -u -r1.51 calctrl.cpp
--- src/generic/calctrl.cpp	2003/10/16 10:27:55	1.51
+++ src/generic/calctrl.cpp	2004/03/17 14:11:41
@@ -282,6 +282,8 @@
     // needed to get the arrow keys normally used for the dialog navigation
     SetWindowStyle(style | wxWANTS_CHARS);
 
+    SetFont(parent->GetFont());
+
     m_date = date.IsValid() ? date : wxDateTime::Today();
 
     m_lowdate = wxDefaultDateTime;
@@ -322,7 +324,6 @@
     SetSize(pos.x, pos.y, sizeReal.x, sizeReal.y);
 
     SetBackgroundColour(*wxWHITE);
-    SetFont(*wxSWISS_FONT);
 
     SetHolidayAttrs();
 
Index: src/generic/listctrl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/generic/listctrl.cpp,v
retrieving revision 1.319
diff -u -r1.319 listctrl.cpp
--- src/generic/listctrl.cpp	2004/03/01 22:17:56	1.319
+++ src/generic/listctrl.cpp	2004/03/17 14:11:43
@@ -4621,6 +4621,8 @@
             m_headerWin->Show( FALSE );
         }
     }
+    
+    InheritAttributes();
 
     return TRUE;
 }
Index: src/generic/panelg.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/generic/panelg.cpp,v
retrieving revision 1.66
diff -u -r1.66 panelg.cpp
--- src/generic/panelg.cpp	2003/09/24 00:41:03	1.66
+++ src/generic/panelg.cpp	2004/03/17 14:11:43
@@ -121,7 +121,11 @@
                      long style,
                      const wxString& name)
 {
-    return wxWindow::Create(parent, id, pos, size, style, name);
+    bool rc= wxWindow::Create(parent, id, pos, size, style, name);
+
+    if (rc)
+        SetFont(parent->GetFont());
+    return rc;
 }
 
 wxPanel::~wxPanel()
Index: src/generic/splitter.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/generic/splitter.cpp,v
retrieving revision 1.91
diff -u -r1.91 splitter.cpp
--- src/generic/splitter.cpp	2004/03/04 23:48:37	1.91
+++ src/generic/splitter.cpp	2004/03/17 14:11:44
@@ -92,6 +92,8 @@
 
     m_permitUnsplitAlways = (style & wxSP_PERMIT_UNSPLIT) != 0;
 
+    SetFont(parent->GetFont());
+
     return true;
 }
 
Index: src/generic/statusbr.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/generic/statusbr.cpp,v
retrieving revision 1.53
diff -u -r1.53 statusbr.cpp
--- src/generic/statusbr.cpp	2004/03/07 15:22:28	1.53
+++ src/generic/statusbr.cpp	2004/03/17 14:11:44
@@ -91,7 +91,7 @@
 #endif
   InitColours();
 
-  SetFont(m_defaultStatusBarFont);
+  SetFont(parent->GetFont());
 
   // Set the height according to the font and the border size
   wxClientDC dc(this);
@@ -350,14 +350,12 @@
     SetBackgroundColour(vColour);
     vColour.Set(wxString("BLACK"));
     SetForegroundColour(vColour);
-    m_defaultStatusBarFont = *wxSMALL_FONT;
 #else
     m_mediumShadowPen = wxPen("GREY", 1, wxSOLID);
     m_hilightPen = wxPen("WHITE", 1, wxSOLID);
 #endif
 
 #ifndef __WXPM__
-    m_defaultStatusBarFont = wxSystemSettings::GetFont(wxSYS_DEFAULT_GUI_FONT);
     SetBackgroundColour(wxSystemSettings::GetColour(wxSYS_COLOUR_3DFACE));
 #endif
 }
Index: src/generic/treectlg.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/generic/treectlg.cpp,v
retrieving revision 1.118
diff -u -r1.118 treectlg.cpp
--- src/generic/treectlg.cpp	2004/03/07 17:28:36	1.118
+++ src/generic/treectlg.cpp	2004/03/17 14:11:45
@@ -781,6 +781,15 @@
     SetForegroundColour( wxSystemSettings::GetColour(wxSYS_COLOUR_WINDOWTEXT) );
     SetBackgroundColour( wxSystemSettings::GetColour(wxSYS_COLOUR_LISTBOX) );
 
+    m_normalFont = parent->GetFont();
+    m_boldFont = wxFont(m_normalFont.GetPointSize(),
+                        m_normalFont.GetFamily(),
+                        m_normalFont.GetStyle(),
+                        wxBOLD,
+                        m_normalFont.GetUnderlined(),
+                        m_normalFont.GetFaceName(),
+                        m_normalFont.GetEncoding());
+
 //  m_dottedPen = wxPen( "grey", 0, wxDOT );  too slow under XFree86
     m_dottedPen = wxPen( wxT("grey"), 0, 0 );
 
cvs server: Diffing src/gtk
Index: src/gtk/gauge.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/gtk/gauge.cpp,v
retrieving revision 1.22
diff -u -r1.22 gauge.cpp
--- src/gtk/gauge.cpp	2004/03/10 01:15:27	1.22
+++ src/gtk/gauge.cpp	2004/03/17 14:11:46
@@ -56,6 +56,7 @@
     m_parent->DoAddChild( this );
 
     PostCreation();
+    InheritAttributes();
     SetBestSize(size);
     
     Show( TRUE );
Index: src/gtk/statline.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/gtk/statline.cpp,v
retrieving revision 1.10
diff -u -r1.10 statline.cpp
--- src/gtk/statline.cpp	2003/08/09 12:46:10	1.10
+++ src/gtk/statline.cpp	2004/03/17 14:11:46
@@ -75,6 +75,7 @@
     m_parent->DoAddChild( this );
 
     PostCreation();
+    InheritAttributes();
 
     Show( TRUE );
 
Index: src/gtk/tbargtk.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/gtk/tbargtk.cpp,v
retrieving revision 1.90
diff -u -r1.90 tbargtk.cpp
--- src/gtk/tbargtk.cpp	2003/09/26 19:43:45	1.90
+++ src/gtk/tbargtk.cpp	2004/03/17 14:11:46
@@ -374,7 +374,8 @@
     m_parent->DoAddChild( this );
 
     PostCreation();
-
+    SetFont(parent->GetFont());
+    
     Show( TRUE );
 
     return TRUE;
Index: src/gtk/treegtk.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/gtk/treegtk.cpp,v
retrieving revision 1.5
diff -u -r1.5 treegtk.cpp
--- src/gtk/treegtk.cpp	2003/08/09 12:46:11	1.5
+++ src/gtk/treegtk.cpp	2004/03/17 14:11:46
@@ -135,6 +135,8 @@
 printf("postcreate\n");
   PostCreation();
 
+  InheritAttributes();
+  
   gtk_widget_realize(GTK_WIDGET(m_tree));
 
   Show(TRUE);

Index: src/msw/checkbox.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/msw/checkbox.cpp,v
retrieving revision 1.53
diff -u -r1.53 checkbox.cpp
--- src/msw/checkbox.cpp	2004/01/15 13:49:04	1.53
+++ src/msw/checkbox.cpp	2004/03/17 14:11:46
@@ -177,7 +177,7 @@
     if ( !s_checkSize )
     {
         wxScreenDC dc;
-        dc.SetFont(wxSystemSettings::GetFont(wxSYS_DEFAULT_GUI_FONT));
+        dc.SetFont(GetFont());
 
         s_checkSize = dc.GetCharHeight();
     }
Index: src/msw/listctrl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/msw/listctrl.cpp,v
retrieving revision 1.192
diff -u -r1.192 listctrl.cpp
--- src/msw/listctrl.cpp	2004/02/29 23:24:35	1.192
+++ src/msw/listctrl.cpp	2004/03/17 14:11:47
@@ -367,8 +367,8 @@
 
     // inherit foreground colour but our background should be the same as for
     // listboxes and other such "container" windows and not inherited
+    InheritAttributes();
     SetBackgroundColour(wxSystemSettings::GetColour(wxSYS_COLOUR_WINDOW));
-    SetForegroundColour(GetParent()->GetForegroundColour());
 
     return TRUE;
 }
Index: src/msw/notebook.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/msw/notebook.cpp,v
retrieving revision 1.111
diff -u -r1.111 notebook.cpp
--- src/msw/notebook.cpp	2004/03/07 20:00:28	1.111
+++ src/msw/notebook.cpp	2004/03/17 14:11:48
@@ -270,6 +270,7 @@
     if ( !MSWCreateControl(WC_TABCONTROL, wxEmptyString, pos, size) )
         return false;
 
+    InheritAttributes();
     SetBackgroundColour(wxColour(::GetSysColor(COLOR_BTNFACE)));
 
     return true;
Index: src/msw/radiobut.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/msw/radiobut.cpp,v
retrieving revision 1.48
diff -u -r1.48 radiobut.cpp
--- src/msw/radiobut.cpp	2004/01/15 13:49:06	1.48
+++ src/msw/radiobut.cpp	2004/03/17 14:11:48
@@ -271,7 +271,7 @@
     if ( !s_radioSize )
     {
         wxScreenDC dc;
-        dc.SetFont(wxSystemSettings::GetFont(wxSYS_DEFAULT_GUI_FONT));
+        dc.SetFont(GetFont());
 
         s_radioSize = dc.GetCharHeight();
     }
Index: src/msw/statbr95.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/msw/statbr95.cpp,v
retrieving revision 1.51
diff -u -r1.51 statbr95.cpp
--- src/msw/statbr95.cpp	2003/08/24 14:30:23	1.51
+++ src/msw/statbr95.cpp	2004/03/17 14:11:48
@@ -118,6 +118,7 @@
     SubclassWin(m_hWnd);
 
     SetBackgroundColour(wxSystemSettings::GetColour(wxSYS_COLOUR_MENUBAR));
+    SetFont(parent->GetFont());
 
     return TRUE;
 }
Index: src/msw/tabctrl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/msw/tabctrl.cpp,v
retrieving revision 1.31
diff -u -r1.31 tabctrl.cpp
--- src/msw/tabctrl.cpp	2004/02/07 14:47:48	1.31
+++ src/msw/tabctrl.cpp	2004/03/17 14:11:48
@@ -116,7 +116,7 @@
   
   SubclassWin((WXHWND) hWndTabCtrl);
 
-  SetFont(wxSystemSettings::GetFont(wxSYS_DEFAULT_GUI_FONT));
+  InheritAttributes();
 
   return TRUE;
 }
Index: src/msw/tbar95.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/msw/tbar95.cpp,v
retrieving revision 1.123
diff -u -r1.123 tbar95.cpp
--- src/msw/tbar95.cpp	2004/02/25 10:44:49	1.123
+++ src/msw/tbar95.cpp	2004/03/17 14:11:53
@@ -241,8 +241,8 @@
         return FALSE;
 
     // set up the colors and fonts
+    InheritAttributes();
     SetBackgroundColour(wxSystemSettings::GetColour(wxSYS_COLOUR_MENUBAR));
-    SetFont(wxSystemSettings::GetFont(wxSYS_DEFAULT_GUI_FONT));
 
     return TRUE;
 }

 	  	 
