 src/generic/filectrlg.cpp | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/generic/filectrlg.cpp b/src/generic/filectrlg.cpp
index 3fbb2c9..1b570dc 100644
--- a/src/generic/filectrlg.cpp
+++ b/src/generic/filectrlg.cpp
@@ -33,6 +33,7 @@
 #include "wx/dir.h"
 #include "wx/tokenzr.h"
 #include "wx/imaglist.h"
+#include "wx/numformatter.h"
 
 #ifdef __WINDOWS__
     #include "wx/msw/wrapwin.h"
@@ -338,7 +339,7 @@ wxString wxFileData::GetEntry( fileListFieldType num ) const
 
         case FileList_Size:
             if (!IsDir() && !IsLink() && !IsDrive())
-                s = wxLongLong(m_size).ToString();
+                s = wxNumberFormatter::ToString(m_size, Style_WithThousandsSep); // depends on wxLocale being initialized
             break;
 
         case FileList_Type:
@@ -457,7 +458,7 @@ void wxFileListCtrl::ChangeToReportMode()
     GetTextExtent(txt, &w, &h);
 
     InsertColumn( 0, _("Name"), wxLIST_FORMAT_LEFT, w );
-    InsertColumn( 1, _("Size"), wxLIST_FORMAT_LEFT, w/2 );
+    InsertColumn( 1, _("Size"), wxLIST_FORMAT_RIGHT, w/2 );
     InsertColumn( 2, _("Type"), wxLIST_FORMAT_LEFT, w/2 );
     InsertColumn( 3, _("Modified"), wxLIST_FORMAT_LEFT, w );
 #if defined(__UNIX__)
