Index: include/wx/fswatcher.h
===================================================================
--- include/wx/fswatcher.h	(Revision 73482)
+++ include/wx/fswatcher.h	(Arbeitskopie)
@@ -152,7 +152,7 @@
 
     virtual wxEventCategory GetEventCategory() const
     {
-        // TODO this has to be merged with "similiar" categories and changed
+        // TODO this has to be merged with "similar" categories and changed
         return wxEVT_CATEGORY_UNKNOWN;
     }
 
Index: include/wx/mediactrl.h
===================================================================
--- include/wx/mediactrl.h	(Revision 73482)
+++ include/wx/mediactrl.h	(Arbeitskopie)
@@ -242,7 +242,7 @@
 // for wxMediaCtrl.  Backends are searched alphabetically -
 // the one with the earliest letter is tried first.
 //
-// Note that this is currently not API or ABI compatable -
+// Note that this is currently not API or ABI compatible -
 // so statically link or make the client compile on-site.
 //
 // ----------------------------------------------------------------------------
@@ -392,7 +392,7 @@
 };
 
 // ----------------------------------------------------------------------------
-// End compilation gaurd
+// End compilation guard
 // ----------------------------------------------------------------------------
 #endif // wxUSE_MEDIACTRL
 
Index: include/wx/event.h
===================================================================
--- include/wx/event.h	(Revision 73482)
+++ include/wx/event.h	(Arbeitskopie)
@@ -1868,7 +1868,7 @@
             *ypos = GetY();
     }
 
-    // This version if provided only for backwards compatiblity, don't use.
+    // This version if provided only for backwards compatibility, don't use.
     void GetPosition(long *xpos, long *ypos) const
     {
         if (xpos)
Index: include/wx/stockitem.h
===================================================================
--- include/wx/stockitem.h	(Revision 73482)
+++ include/wx/stockitem.h	(Arbeitskopie)
@@ -73,7 +73,7 @@
 
 #ifdef __WXGTK20__
 
-// Translates stock ID to GTK+'s stock item string indentifier:
+// Translates stock ID to GTK+'s stock item string identifier:
 WXDLLIMPEXP_CORE const char *wxGetStockGtkID(wxWindowID id);
 
 #endif
Index: include/wx/arrstr.h
===================================================================
--- include/wx/arrstr.h	(Revision 73482)
+++ include/wx/arrstr.h	(Arbeitskopie)
@@ -425,7 +425,7 @@
 // ----------------------------------------------------------------------------
 
 // by default, these functions use the escape character to escape the
-// separators occuring inside the string to be joined, this can be disabled by
+// separators occurring inside the string to be joined, this can be disabled by
 // passing '\0' as escape
 
 WXDLLIMPEXP_BASE wxString wxJoin(const wxArrayString& arr,
Index: include/wx/dnd.h
===================================================================
--- include/wx/dnd.h	(Revision 73482)
+++ include/wx/dnd.h	(Arbeitskopie)
@@ -66,7 +66,7 @@
         { m_data = NULL; }
     virtual ~wxDropSourceBase() { }
 
-    // set the data which is transfered by drag and drop
+    // set the data which is transferred by drag and drop
     void SetData(wxDataObject& data)
       { m_data = &data; }
 
