diff -ru2 -xCVS -x.#* -x*.orig -x*.rej -x*~ wxWindows.cvs/contrib/include/wx/fl/controlbar.h wxWindows/contrib/include/wx/fl/controlbar.h
--- wxWindows.cvs/contrib/include/wx/fl/controlbar.h	Fri Aug 15 03:07:24 2003
+++ wxWindows/contrib/include/wx/fl/controlbar.h	Mon Oct 20 16:18:12 2003
@@ -45,6 +45,6 @@
 typedef cbRowInfo* RowInfoPtrT;
 
-WXFL_DEFINE_ARRAY_NO_PTR( BarInfoPtrT, BarArrayT );
-WXFL_DEFINE_ARRAY_NO_PTR( RowInfoPtrT, RowArrayT );
+WXFL_DEFINE_ARRAY_PTR( BarInfoPtrT, BarArrayT );
+WXFL_DEFINE_ARRAY_PTR( RowInfoPtrT, RowArrayT );
 
 // control bar states
@@ -1499,5 +1499,5 @@
         // these notifications to implement a more fine-grained updating strategy.
 
-    virtual void OnRowWillChange( cbRowInfo* pRow, cbDockPane* pInPane ) {}
+    virtual void OnRowWillChange( cbRowInfo* WXUNUSED(pRow), cbDockPane* WXUNUSED(pInPane) ) {}
 
         // This function receives a notification from the frame layout (in the order in which
@@ -1505,5 +1505,5 @@
         // these notifications to implement a more fine-grained updating strategy.
 
-    virtual void OnBarWillChange( cbBarInfo* pBar, cbRowInfo* pInRow, cbDockPane* pInPane ) {}
+    virtual void OnBarWillChange( cbBarInfo* WXUNUSED(pBar), cbRowInfo* WXUNUSED(pInRow), cbDockPane* WXUNUSED(pInPane) ) {}
 
         // This function receives a notification from the frame layout (in the order in which
@@ -1511,5 +1511,5 @@
         // these notifications to implement a more fine-grained updating strategy.
 
-    virtual void OnPaneMarginsWillChange( cbDockPane* pPane ) {}
+    virtual void OnPaneMarginsWillChange( cbDockPane* WXUNUSED(pPane) ) {}
 
         // This function receives a notification from the frame layout (in the order in which
@@ -1517,5 +1517,5 @@
         // these notifications to implement a more fine-grained updating strategy.
 
-    virtual void OnPaneWillChange( cbDockPane* pPane ) {}
+    virtual void OnPaneWillChange( cbDockPane* WXUNUSED(pPane) ) {}
 
         // This function receives a notification from the frame layout (in the order in which
diff -ru2 -xCVS -x.#* -x*.orig -x*.rej -x*~ wxWindows.cvs/contrib/include/wx/fl/dyntbar.h wxWindows/contrib/include/wx/fl/dyntbar.h
--- wxWindows.cvs/contrib/include/wx/fl/dyntbar.h	Fri Aug 15 03:07:24 2003
+++ wxWindows/contrib/include/wx/fl/dyntbar.h	Mon Oct 20 16:18:42 2003
@@ -39,6 +39,6 @@
 typedef wxDynToolInfo*    wxDynToolInfoPtrT;
 
-WXFL_DEFINE_ARRAY_NO_PTR( wxToolLayoutItemPtrT, wxLayoutItemArrayT  );
-WXFL_DEFINE_ARRAY_NO_PTR( wxDynToolInfoPtrT,    wxDynToolInfoArrayT );
+WXFL_DEFINE_ARRAY_PTR( wxToolLayoutItemPtrT, wxLayoutItemArrayT  );
+WXFL_DEFINE_ARRAY_PTR( wxDynToolInfoPtrT,    wxDynToolInfoArrayT );
 
 /*
diff -ru2 -xCVS -x.#* -x*.orig -x*.rej -x*~ wxWindows.cvs/contrib/include/wx/fl/fldefs.h wxWindows/contrib/include/wx/fl/fldefs.h
--- wxWindows.cvs/contrib/include/wx/fl/fldefs.h	Fri Aug 15 03:07:24 2003
+++ wxWindows/contrib/include/wx/fl/fldefs.h	Mon Oct 20 16:17:07 2003
@@ -35,7 +35,7 @@
     WX_DEFINE_ARRAY(c,l)
 
-#define WXFL_DEFINE_ARRAY_NO_PTR(c,l)\
+#define WXFL_DEFINE_ARRAY_PTR(c,l)\
     class WXDLLIMPEXP_FL l;          \
-    WX_DEFINE_ARRAY_NO_PTR(c,l)
+    WX_DEFINE_ARRAY_PTR(c,l)
 
 #define WXFL_DEFINE_ARRAY_LONG(t,l) \
diff -ru2 -xCVS -x.#* -x*.orig -x*.rej -x*~ wxWindows.cvs/contrib/include/wx/fl/frmview.h wxWindows/contrib/include/wx/fl/frmview.h
--- wxWindows.cvs/contrib/include/wx/fl/frmview.h	Fri Aug 15 03:07:24 2003
+++ wxWindows/contrib/include/wx/fl/frmview.h	Mon Oct 20 16:19:09 2003
@@ -71,5 +71,5 @@
     virtual void OnInit() {}
 
-    virtual void OnSerialize( wxObjectStorage& store ) {}
+    virtual void OnSerialize( wxObjectStorage& WXUNUSED(store) ) {}
     virtual void OnActiveate() {}
     virtual void OnDeactivate() {}
diff -ru2 -xCVS -x.#* -x*.orig -x*.rej -x*~ wxWindows.cvs/contrib/include/wx/fl/toolwnd.h wxWindows/contrib/include/wx/fl/toolwnd.h
--- wxWindows.cvs/contrib/include/wx/fl/toolwnd.h	Fri Aug 15 03:07:24 2003
+++ wxWindows/contrib/include/wx/fl/toolwnd.h	Mon Oct 20 16:18:32 2003
@@ -32,5 +32,5 @@
 typedef cbMiniButton* cbMinitButtonPtrT;
 
-WXFL_DEFINE_ARRAY_NO_PTR( cbMinitButtonPtrT, cbMiniButtonArrayT );
+WXFL_DEFINE_ARRAY_PTR( cbMinitButtonPtrT, cbMiniButtonArrayT );
 
 /*
@@ -145,8 +145,8 @@
         // Called when a mini button is clicked.
         // By default, does nothing.
-    virtual void OnMiniButtonClicked( int btnIdx ) {}
+    virtual void OnMiniButtonClicked( int WXUNUSED(btnIdx) ) {}
 
         // Handles clicking on the title. By default, does nothing.
-    virtual bool HandleTitleClick( wxMouseEvent& event ) { return FALSE; }
+    virtual bool HandleTitleClick( wxMouseEvent& WXUNUSED(event) ) { return FALSE; }
 
     DECLARE_EVENT_TABLE()


 	  	 
