Index: include/wx/generic/grid.h
===================================================================
--- include/wx/generic/grid.h	(revision 70452)
+++ include/wx/generic/grid.h	(working copy)
@@ -2125,6 +2125,10 @@
 
     friend class wxGridHeaderCtrl;
 
+
+    // Overridable for wxGRIDTABLE_NOTIFY_VALUES_INSERTED
+    virtual bool OnValuesInserted(void);
+
 private:
 
     // implement wxScrolledWindow method to return m_gridWin size
Index: src/generic/grid.cpp
===================================================================
--- src/generic/grid.cpp	(revision 70452)
+++ src/generic/grid.cpp	(working copy)
@@ -4765,6 +4765,15 @@
 {
 }
 
+
+// Overridable for wxGRIDTABLE_NOTIFY_VALUES_INSERTED
+bool wxGrid::OnValuesInserted(void)
+{
+    return true;
+}
+
+
+
 bool wxGrid::SetCurrentCell( const wxGridCellCoords& coords )
 {
     if ( SendEvent(wxEVT_GRID_SELECT_CELL, coords) == -1 )
@@ -4974,6 +4983,10 @@
         // all we need to do is repaint the grid
         //
         m_gridWin->Refresh();
+
+        // Provide interface for works after inserting data
+        OnValuesInserted();
+
         return true;
     }
 
