diff -uNrB wxWidgets-trunk\src/msw/dc.cpp wxWidgets-work\src/msw/dc.cpp
--- wxWidgets-trunk\src/msw/dc.cpp	2014-02-01 20:45:05.938430100 +0100
+++ wxWidgets-work\src/msw/dc.cpp	2014-02-11 17:49:53.683038000 +0100
@@ -1319,7 +1319,11 @@
         SelectInHDC select(hdcMem, GetHbitmapOf(bmp));
 
         if ( AlphaBlt(this, x, y, width, height, 0, 0, width, height, hdcMem, bmp) )
+        {
+            CalcBoundingBox(x, y);
+            CalcBoundingBox(x + bmp.GetWidth(), y + bmp.GetHeight());
             return;
+        }
     }
 
     SET_STRETCH_BLT_MODE(GetHdc());
@@ -1426,6 +1430,9 @@
         ::SelectObject( memdc, hOldBitmap );
         ::DeleteDC( memdc );
     }
+
+    CalcBoundingBox(x, y);
+    CalcBoundingBox(x + bmp.GetWidth(), y + bmp.GetHeight());
 }
 
 void wxMSWDCImpl::DoDrawText(const wxString& text, wxCoord x, wxCoord y)
@@ -2209,7 +2216,11 @@
     {
         if ( AlphaBlt(this, xdest, ydest, dstWidth, dstHeight,
                       xsrc, ysrc, srcWidth, srcHeight, hdcSrc, bmpSrc) )
+        {
+            CalcBoundingBox(xdest, ydest);
+            CalcBoundingBox(xdest + dstWidth, ydest + dstHeight);
             return true;
+        }
     }
 
     wxMask *mask = NULL;
@@ -2486,6 +2497,12 @@
         }
     }
 
+    if( success )
+    {
+        CalcBoundingBox(xdest, ydest);
+        CalcBoundingBox(xdest + dstWidth, ydest + dstHeight);
+    }
+
     return success;
 }
 
@@ -2911,6 +2928,8 @@
              ) )
         {
             // skip call of the base class version below
+            CalcBoundingBox(rect.GetLeft(), rect.GetBottom());
+            CalcBoundingBox(rect.GetRight(), rect.GetTop());
             return;
         }
 
