Index: include/wx/docview.h
===================================================================
--- include/wx/docview.h	(revision 64557)
+++ include/wx/docview.h	(working copy)
@@ -371,6 +371,7 @@
     void OnFileRevert(wxCommandEvent& event);
     void OnFileSave(wxCommandEvent& event);
     void OnFileSaveAs(wxCommandEvent& event);
+    void OnMRUFile(wxCommandEvent& event);
 #if wxUSE_PRINTING_ARCHITECTURE
     void OnPrint(wxCommandEvent& event);
     void OnPreview(wxCommandEvent& event);
@@ -783,12 +784,6 @@
     wxDocManager *GetDocumentManager() const { return m_docManager; }
 
 protected:
-    // Open the MRU file with the given index in our associated file history.
-    //
-    // This is called from the derived class event handler for the MRU menu
-    // items.
-    void DoOpenMRUFile(unsigned n);
-
     wxDocManager *m_docManager;
 
     wxDECLARE_NO_COPY_CLASS(wxDocParentFrameAnyBase);
@@ -830,8 +825,6 @@
 
         this->Connect(wxID_EXIT, wxEVT_COMMAND_MENU_SELECTED,
                       wxCommandEventHandler(wxDocParentFrameAny::OnExit));
-        this->Connect(wxID_FILE1, wxID_FILE9, wxEVT_COMMAND_MENU_SELECTED,
-                      wxCommandEventHandler(wxDocParentFrameAny::OnMRUFile));
         this->Connect(wxEVT_CLOSE_WINDOW,
                       wxCloseEventHandler(wxDocParentFrameAny::OnCloseWindow));
 
@@ -854,11 +847,6 @@
         this->Close();
     }
 
-    void OnMRUFile(wxCommandEvent& event)
-    {
-        DoOpenMRUFile(event.GetId() - wxID_FILE1);
-    }
-
     void OnCloseWindow(wxCloseEvent& event)
     {
         if ( m_docManager && !m_docManager->Clear(!event.CanVeto()) )
Index: src/common/docview.cpp
===================================================================
--- src/common/docview.cpp	(revision 64557)
+++ src/common/docview.cpp	(working copy)
@@ -1083,6 +1083,39 @@
     doc->SaveAs();
 }
 
+void wxDocManager::OnMRUFile(wxCommandEvent& event)
+{
+    size_t n = event.GetId() - wxID_FILE1;
+    wxString filename(GetHistoryFile(n));
+    if ( filename.empty() )
+        return;
+
+    wxString errMsg; // must contain exactly one "%s" if non-empty
+    if ( wxFile::Exists(filename) )
+    {
+        // try to open it
+        if ( CreateDocument(filename, wxDOC_SILENT) )
+            return;
+
+        errMsg = _("The file '%s' couldn't be opened.");
+    }
+    else // file doesn't exist
+    {
+        errMsg = _("The file '%s' doesn't exist and couldn't be opened.");
+    }
+
+
+    wxASSERT_MSG( !errMsg.empty(), "should have an error message" );
+
+    // remove the file which we can't open from the MRU list
+    RemoveFileFromHistory(n);
+
+    // and tell the user about it
+    wxLogError(errMsg + '\n' +
+               _("It has been removed from the most recently used files list."),
+               filename);
+}
+
 #if wxUSE_PRINTING_ARCHITECTURE
 
 void wxDocManager::OnPrint(wxCommandEvent& WXUNUSED(event))
@@ -1264,7 +1297,16 @@
 bool wxDocManager::TryBefore(wxEvent& event)
 {
     wxView * const view = GetActiveView();
-    return view && view->ProcessEventLocally(event);
+    bool processed = view && view->ProcessEventLocally(event);
+    if (   (!processed)
+        && (event.GetEventType() == wxEVT_COMMAND_MENU_SELECTED)
+        && (event.GetId() >= wxID_FILE1) 
+        && (event.GetId() < (wxID_FILE1 + m_fileHistory->GetBaseId())) )
+    {
+        OnMRUFile((wxCommandEvent&)event);
+        processed = true;
+    }
+    return processed;
 }
 
 namespace
@@ -1914,38 +1956,6 @@
 // wxDocParentFrameAnyBase
 // ----------------------------------------------------------------------------
 
-void wxDocParentFrameAnyBase::DoOpenMRUFile(unsigned n)
-{
-    wxString filename(m_docManager->GetHistoryFile(n));
-    if ( filename.empty() )
-        return;
-
-    wxString errMsg; // must contain exactly one "%s" if non-empty
-    if ( wxFile::Exists(filename) )
-    {
-        // try to open it
-        if ( m_docManager->CreateDocument(filename, wxDOC_SILENT) )
-            return;
-
-        errMsg = _("The file '%s' couldn't be opened.");
-    }
-    else // file doesn't exist
-    {
-        errMsg = _("The file '%s' doesn't exist and couldn't be opened.");
-    }
-
-
-    wxASSERT_MSG( !errMsg.empty(), "should have an error message" );
-
-    // remove the file which we can't open from the MRU list
-    m_docManager->RemoveFileFromHistory(n);
-
-    // and tell the user about it
-    wxLogError(errMsg + '\n' +
-               _("It has been removed from the most recently used files list."),
-               filename);
-}
-
 #if wxUSE_PRINTING_ARCHITECTURE
 
 wxDocPrintout::wxDocPrintout(wxView *view, const wxString& title)
