--- toplevel_orig.h	2006-01-31 19:57:13.963828648 +1100
+++ toplevel.h	2006-01-31 19:57:13.938832448 +1100
@@ -87,4 +87,5 @@
     virtual void SetRightMenu(int id = wxID_ANY, const wxString& label = wxEmptyString, wxMenu *subMenu = NULL);
     bool HandleCommand(WXWORD id, WXWORD cmd, WXHWND control);
+	bool MSWShouldPreProcessMessage(WXMSG* pMsg);
 #endif // __SMARTPHONE__ && __WXWINCE__
 

 	  	 
