Index: flatnotebook.py
===================================================================
--- flatnotebook.py	(revision 56565)
+++ flatnotebook.py	(working copy)
@@ -3785,6 +3785,7 @@
         self.Bind(wx.EVT_RIGHT_DOWN, self.OnRightDown)
         self.Bind(wx.EVT_MIDDLE_DOWN, self.OnMiddleDown)
         self.Bind(wx.EVT_MOTION, self.OnMouseMove)
+        self.Bind(wx.EVT_MOUSEWHEEL, self.OnMouseWheel)
         self.Bind(wx.EVT_ERASE_BACKGROUND, self.OnEraseBackground)
         self.Bind(wx.EVT_LEAVE_WINDOW, self.OnMouseLeave)
         self.Bind(wx.EVT_ENTER_WINDOW, self.OnMouseEnterWindow)
@@ -3908,6 +3909,24 @@
         event.Skip()
 
 
+    def OnMouseWheel(self, event):
+        """ Scroll tabs when the mouse wheel """
+        delta = event.GetWheelRotation()
+        for tab in range(abs(delta)):
+            if delta > 0:
+                before = self._nLeftButtonStatus
+                self._nLeftButtonStatus = FNB_BTN_PRESSED
+                self.RotateLeft()
+                self._nLeftButtonStatus = before
+            else:
+                before = self._nRightButtonStatus
+                self._nRightButtonStatus = FNB_BTN_PRESSED
+                self.RotateRight()
+                self._nRightButtonStatus = before
+
+        event.Skip()
+
+
     def OnRightDown(self, event):
         """ Handles the wx.EVT_RIGHT_DOWN events for L{PageContainer}. """
