Index: src/mac/carbon/slider.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/mac/carbon/slider.cpp,v
retrieving revision 1.52
diff -b -u -2 -r1.52 slider.cpp
--- src/mac/carbon/slider.cpp	2006/02/06 05:08:13	1.52
+++ src/mac/carbon/slider.cpp	2006/04/10 13:57:13
@@ -205,5 +205,14 @@
     }
 
-    SetValue( m_rangeMin );
+    // If the range is out of bounds, set it to a 
+    // value that is within bounds
+    // RN: Testing reveals OSX does its own 
+    // bounding, perhaps this isn't needed?
+    int currentValue = GetValue();
+
+    if(currentValue < m_rangeMin)
+        SetValue(m_rangeMin);
+    else if(currentValue > m_rangeMax)
+        SetValue(m_rangeMax);
 }
 
@@ -295,17 +304,12 @@
     SetValue( value );
 
-    wxEventType scrollEvent = wxEVT_NULL;
-
-    scrollEvent = wxEVT_SCROLL_THUMBTRACK;
-
-    wxScrollEvent event(scrollEvent, m_windowId);
-    event.SetPosition(value);
+    wxScrollEvent event( wxEVT_SCROLL_THUMBTRACK, m_windowId );
+    event.SetPosition( value );
     event.SetEventObject( this );
-    GetEventHandler()->ProcessEvent(event);
+    GetEventHandler()->ProcessEvent( event );
 
     wxCommandEvent cevent( wxEVT_COMMAND_SLIDER_UPDATED, m_windowId );
     cevent.SetInt( value );
     cevent.SetEventObject( this );
-
     GetEventHandler()->ProcessEvent( cevent );
 }
@@ -318,13 +322,9 @@
 
     SetValue( value ) ;
-
-    wxEventType scrollEvent = wxEVT_NULL ;
-
-    scrollEvent = wxEVT_SCROLL_THUMBRELEASE;
 
-    wxScrollEvent event(scrollEvent, m_windowId);
-    event.SetPosition(value);
+    wxScrollEvent event( wxEVT_SCROLL_THUMBRELEASE, m_windowId );
+    event.SetPosition( value );
     event.SetEventObject( this );
-    GetEventHandler()->ProcessEvent(event);
+    GetEventHandler()->ProcessEvent( event );
 
     wxCommandEvent cevent( wxEVT_COMMAND_SLIDER_UPDATED, m_windowId );

 	  	 
