diff -bru wxWidgets/src/osx/listbox_osx.cpp wxWidgets-2.9.5/src/osx/listbox_osx.cpp
--- wxWidgets/src/osx/listbox_osx.cpp	2014-01-02 00:49:23.000000000 -0800
+++ wxWidgets-2.9.5/src/osx/listbox_osx.cpp	2014-02-06 18:14:06.000000000 -0800
@@ -261,6 +262,7 @@
         wxCoord width, height ;
         dc.GetTextExtent( wxT("XX") , &width, &height);
         int cx = width ;
+        cy = height;
         lbWidth += cx;
 
         // don't make the listbox too tall (limit height to around 10 items)
@@ -370,7 +372,7 @@
     SetFirstItem(startpos);
 
     UpdateOldSelections();
-
+    InvalidateBestSize();
     return idx;
 }
 
