Index: src/unix/gsocket.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/unix/gsocket.cpp,v
retrieving revision 1.47
diff -u -r1.47 gsocket.cpp
--- src/unix/gsocket.cpp	2005/09/03 23:16:50	1.47
+++ src/unix/gsocket.cpp	2005/09/07 12:59:07
@@ -1016,7 +1016,8 @@
           result |= GSOCK_CONNECTION_FLAG;
           m_detected |= GSOCK_CONNECTION_FLAG;
         }
-        else if ((errno != EWOULDBLOCK) && (errno != EAGAIN) && (errno != EINTR))
+        else if (num == 0 ||
+                 (errno != EWOULDBLOCK && errno != EAGAIN && errno != EINTR))
         {
           m_detected = GSOCK_LOST_FLAG;
           m_establishing = false;

 	  	 
