Index: include/wx/gtk/print.h
===================================================================
--- include/wx/gtk/print.h	(revision 72625)
+++ include/wx/gtk/print.h	(working copy)
@@ -197,6 +197,7 @@
     void SetPrintConfig( GtkPrintSettings * config );
 
     GtkPrintOperation* GetPrintJob() { return m_job; }
+    void SetPrintJob(GtkPrintOperation *job) { m_job = job; }
 
     GtkPrintContext *GetPrintContext() { return m_context; }
     void SetPrintContext(GtkPrintContext *context) {m_context = context; }
Index: src/gtk/print.cpp
===================================================================
--- src/gtk/print.cpp	(revision 72625)
+++ src/gtk/print.cpp	(working copy)
@@ -234,13 +234,12 @@
 wxGtkPrintNativeData::wxGtkPrintNativeData()
 {
     m_config = gtk_print_settings_new();
-    m_job = gtk_print_operation_new();
+    m_job = NULL;
     m_context = NULL;
 }
 
 wxGtkPrintNativeData::~wxGtkPrintNativeData()
 {
-    g_object_unref(m_job);
     g_object_unref(m_config);
 }
 
@@ -904,7 +903,8 @@
     wxPrintData printdata = GetPrintDialogData().GetPrintData();
     wxGtkPrintNativeData *native = (wxGtkPrintNativeData*) printdata.GetNativeData();
 
-    GtkPrintOperation * const printOp = native->GetPrintJob();
+    GtkPrintOperation* printOp = gtk_print_operation_new();
+    native->SetPrintJob(printOp);
 
     wxPrinterToGtkData dataToSend;
     dataToSend.printer = this;
@@ -932,6 +932,8 @@
         sm_lastError = wxPRINTER_ERROR;
         wxFAIL_MSG(_("The print dialog returned an error."));
     }
+    g_object_unref(printOp);
+    native->SetPrintJob(NULL);
 
     return (sm_lastError == wxPRINTER_NO_ERROR);
 }
