--- wxrc.cpp	2008-06-24 17:00:29.000000000 +0200
+++ wxrc.cpp	2008-06-25 10:09:05.000000000 +0200
@@ -117,7 +117,8 @@
             name == _T("notebookpage") ||
             name == _T("separator") ||
             name == _T("sizeritem") ||
-            name == _T("wxMenuItem"))
+            name == _T("wxMenuItem")||
+            name.EndsWith("Sizer"))
         {
             return false;
         }
