Index: include/wx/defs.h
===================================================================
--- include/wx/defs.h	(revision 62506)
+++ include/wx/defs.h	(working copy)
@@ -978,28 +978,28 @@
 #if (defined(__VISUALC__) && defined(__WIN32__))
     #define wxLongLong_t __int64
     #define wxLongLongSuffix i64
-    #define wxLongLongFmtSpec wxT("I64")
+    #define wxLongLongFmtSpec "I64"
 #elif defined(__BORLANDC__) && defined(__WIN32__) && (__BORLANDC__ >= 0x520)
     #define wxLongLong_t __int64
     #define wxLongLongSuffix i64
-    #define wxLongLongFmtSpec wxT("L")
+    #define wxLongLongFmtSpec "L"
 #elif (defined(__WATCOMC__) && (defined(__WIN32__) || defined(__DOS__) || defined(__OS2__)))
       #define wxLongLong_t __int64
       #define wxLongLongSuffix i64
-      #define wxLongLongFmtSpec wxT("L")
+      #define wxLongLongFmtSpec "L"
 #elif defined(__DIGITALMARS__)
       #define wxLongLong_t __int64
       #define wxLongLongSuffix LL
-      #define wxLongLongFmtSpec wxT("ll")
+      #define wxLongLongFmtSpec "ll"
 #elif defined(__MINGW32__)
     #define wxLongLong_t long long
     #define wxLongLongSuffix ll
-    #define wxLongLongFmtSpec wxT("I64")
+    #define wxLongLongFmtSpec "I64"
 #elif defined(__MWERKS__)
     #if __option(longlong)
         #define wxLongLong_t long long
         #define wxLongLongSuffix ll
-        #define wxLongLongFmtSpec wxT("ll")
+        #define wxLongLongFmtSpec "ll"
     #else
         #error "The 64 bit integer support in CodeWarrior has been disabled."
         #error "See the documentation on the 'longlong' pragma."
@@ -1011,7 +1011,7 @@
         #define wxLongLong_t long long
     #endif /* __WXPALMOS6__ */
     #define wxLongLongSuffix ll
-    #define wxLongLongFmtSpec wxT("ll")
+    #define wxLongLongFmtSpec "ll"
 #elif defined(__VISAGECPP__) && __IBMCPP__ >= 400
     #define wxLongLong_t long long
 #elif (defined(SIZEOF_LONG_LONG) && SIZEOF_LONG_LONG >= 8)  || \
@@ -1021,11 +1021,11 @@
         (defined(__DJGPP__) && __DJGPP__ >= 2)
     #define wxLongLong_t long long
     #define wxLongLongSuffix ll
-    #define wxLongLongFmtSpec wxT("ll")
+    #define wxLongLongFmtSpec "ll"
 #elif defined(SIZEOF_LONG) && (SIZEOF_LONG == 8)
     #define wxLongLong_t long
     #define wxLongLongSuffix l
-    #define wxLongLongFmtSpec wxT("l")
+    #define wxLongLongFmtSpec "l"
     #define wxLongLongIsLong
 #endif
 
Index: include/wx/string.h
===================================================================
--- include/wx/string.h	(revision 62506)
+++ include/wx/string.h	(working copy)
@@ -2071,13 +2071,13 @@
       // insert a long long if they exist and aren't longs
   wxString& operator<<(wxLongLong_t ll)
     {
-      const wxChar *fmt = wxT("%") wxLongLongFmtSpec wxT("d");
+      const wxChar *fmt = wxT("%" wxLongLongFmtSpec "d");
       return (*this) << Format(fmt, ll);
     }
       // insert an unsigned long long
   wxString& operator<<(wxULongLong_t ull)
     {
-      const wxChar *fmt = wxT("%") wxLongLongFmtSpec wxT("u");
+      const wxChar *fmt = wxT("%" wxLongLongFmtSpec "u");
       return (*this) << Format(fmt , ull);
     }
 #endif // wxLongLong_t && !wxLongLongIsLong
Index: samples/console/console.cpp
===================================================================
--- samples/console/console.cpp	(revision 62506)
+++ samples/console/console.cpp	(working copy)
@@ -1915,7 +1915,7 @@
   {
       wxChar buf[200];
 
-      wxSprintf(buf, wxT("%07") wxLongLongFmtSpec wxT("o"), wxLL(040000000000));
+      wxSprintf(buf, wxT("%07" wxLongLongFmtSpec "o"), wxLL(040000000000));
       #if 0
         // for some reason below line fails under Borland
       wxPrintf (wxT("sprintf (buf, \"%%07Lo\", 040000000000ll) = %s"), buf);
Index: src/generic/timer.cpp
===================================================================
--- src/generic/timer.cpp	(revision 62506)
+++ src/generic/timer.cpp	(working copy)
@@ -74,7 +74,7 @@
     typedef wxLongLong wxTimerTick_t;
 
     #if wxUSE_LONGLONG_WX
-        #define wxTimerTickFmtSpec wxLongLongFmtSpec wxT("d")
+        #define wxTimerTickFmtSpec wxT(wxLongLongFmtSpec "d")
         #define wxTimerTickPrintfArg(tt) (tt.GetValue())
     #else // using native wxLongLong
         #define wxTimerTickFmtSpec wxT("s")
Index: src/msw/debughlp.cpp
===================================================================
--- src/msw/debughlp.cpp	(revision 62506)
+++ src/msw/debughlp.cpp	(working copy)
@@ -281,7 +281,7 @@
         }
         else // opaque 64 bit value
         {
-            s.Printf(wxT("%#" wxLongLongFmtSpec wxT("x")), *(PDWORD *)pAddress);
+            s.Printf(wxT("%#" wxLongLongFmtSpec "x"), *(PDWORD *)pAddress);
         }
     }
 
Index: src/propgrid/props.cpp
===================================================================
--- src/propgrid/props.cpp	(revision 62506)
+++ src/propgrid/props.cpp	(working copy)
@@ -359,14 +359,14 @@
 };
 
 static const wxChar* gs_uintTemplates64[wxPG_UINT_TEMPLATE_MAX] = {
-    wxT("%") wxLongLongFmtSpec wxT("x"),
-    wxT("0x%") wxLongLongFmtSpec wxT("x"),
-    wxT("$%") wxLongLongFmtSpec wxT("x"),
-    wxT("%") wxLongLongFmtSpec wxT("X"),
-    wxT("0x%") wxLongLongFmtSpec wxT("X"),
-    wxT("$%") wxLongLongFmtSpec wxT("X"),
-    wxT("%") wxLongLongFmtSpec wxT("u"),
-    wxT("%") wxLongLongFmtSpec wxT("o")
+    wxT("%" wxLongLongFmtSpec "x"),
+    wxT("0x%" wxLongLongFmtSpec "x"),
+    wxT("$%" wxLongLongFmtSpec "x"),
+    wxT("%" wxLongLongFmtSpec "X"),
+    wxT("0x%" wxLongLongFmtSpec "X"),
+    wxT("$%" wxLongLongFmtSpec "X"),
+    wxT("%" wxLongLongFmtSpec "u"),
+    wxT("%" wxLongLongFmtSpec "o")
 };
 
 WX_PG_IMPLEMENT_PROPERTY_CLASS(wxUIntProperty,wxPGProperty,
