diff -ru2 -N -xCVS -x.#* -x*.orig -x*.vcproj -x*.cmake -x*.sln -x*.suo -x*.rej -x*~ -x*.o -x*.obj -x*.a -x*.lib -x*setup.h -x*y*.c wxWidgets.orig/src/common/framecmn.cpp wxWidgets/src/common/framecmn.cpp
--- wxWidgets.orig/src/common/framecmn.cpp	Tue Dec 21 15:57:20 2004
+++ wxWidgets/src/common/framecmn.cpp	Tue Dec 21 16:03:32 2004
@@ -277,5 +277,5 @@
 {
     // do we have real status text to restore?
-    if ( m_oldStatusText.length() > 1 || m_oldStatusText[0u] )
+    if ( !m_oldStatusText.empty() )
     {
         if ( m_statusBarPane >= 0 )

 	  	 
