Index: src/gtk/listbox.cpp
===================================================================
--- src/gtk/listbox.cpp	(revisión: 65805)
+++ src/gtk/listbox.cpp	(copia de trabajo)
@@ -430,7 +430,8 @@
     else // no multi-selection flags specified
     {
         m_windowStyle |= wxLB_SINGLE;
-        mode = GTK_SELECTION_SINGLE;
+        // BROWSE means exactly one can be selected
+        mode = GTK_SELECTION_BROWSE;
     }
 
     GtkTreeSelection* selection = gtk_tree_view_get_selection( m_treeview );
