Index: wxWidgets/include/wx/gtk/colordlg.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/gtk/colordlg.h,v
retrieving revision 1.1
diff -b -u -2 -r1.1 colordlg.h
--- wxWidgets/include/wx/gtk/colordlg.h	2004/06/04 16:00:47	1.1
+++ wxWidgets/include/wx/gtk/colordlg.h	2004/10/16 19:34:06
@@ -40,8 +40,9 @@
     // implement some base class methods to do nothing to avoid asserts and
     // GTK warnings, since this is not a real wxDialog.
-    virtual void DoSetSize(int x, int y,
-                           int width, int height,
-                           int sizeFlags = wxSIZE_AUTO) {}
-    virtual void DoMoveWindow(int x, int y, int width, int height) {}
+    virtual void DoSetSize(int WXUNUSED(x), int WXUNUSED(y),
+                           int WXUNUSED(width), int WXUNUSED(height),
+                           int WXUNUSED(sizeFlags) = wxSIZE_AUTO) {}
+    virtual void DoMoveWindow(int WXUNUSED(x), int WXUNUSED(y), 
+                              int WXUNUSED(width), int WXUNUSED(height)) {}
 
     // copy data between the dialog and m_colourData:
Index: wxWidgets/include/wx/gtk/msgdlg.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/gtk/msgdlg.h,v
retrieving revision 1.8
diff -b -u -2 -r1.8 msgdlg.h
--- wxWidgets/include/wx/gtk/msgdlg.h	2004/05/23 20:50:53	1.8
+++ wxWidgets/include/wx/gtk/msgdlg.h	2004/10/16 19:34:06
@@ -34,13 +34,14 @@
 
     int ShowModal();
-    virtual bool Show( bool show = true ) { return false; };
+    virtual bool Show( bool WXUNUSED(show) = true ) { return false; };
 
 protected:
     // implement some base class methods to do nothing to avoid asserts and
     // GTK warnings, since this is not a real wxDialog.
-    virtual void DoSetSize(int x, int y,
-                           int width, int height,
-                           int sizeFlags = wxSIZE_AUTO) {}
-    virtual void DoMoveWindow(int x, int y, int width, int height) {}
+    virtual void DoSetSize(int WXUNUSED(x), int WXUNUSED(y),
+                           int WXUNUSED(width), int WXUNUSED(height),
+                           int WXUNUSED(sizeFlags) = wxSIZE_AUTO) {}
+    virtual void DoMoveWindow(int WXUNUSED(x), int WXUNUSED(y), 
+                              int WXUNUSED(width), int WXUNUSED(height)) {}
 
 private:

 	  	 
