--- a/src/generic/grid.cpp
+++ b/src/generic/grid.cpp
@@ -3730,10 +3730,14 @@ void wxGrid::ProcessColLabelMouseEvent( wxMouseEvent& event )
                     bool onNearPart;
                     const int middle = GetColLeft(colValid) +
                                             GetColWidth(colValid)/2;
-                    if ( GetLayoutDirection() == wxLayout_LeftToRight )
-                        onNearPart = (x <= middle);
-                    else // wxLayout_RightToLeft
+                    // Under MSW it is not necessary
+                    // to mirror logical coordinates in RTL mode.
+#if !defined (__WXMSW__)
+                    if ( GetLayoutDirection() == wxLayout_RightToLeft )
                         onNearPart = (x > middle);
+                    else
+#endif
+                        onNearPart = (x <= middle);
 
                     // adjust for the column being dragged itself
                     if ( pos < GetColPos(m_dragRowOrCol) )
