--- a/src/msw/dc.cpp
+++ b/src/msw/dc.cpp
@@ -1035,7 +1035,8 @@ void wxMSWDCImpl::DoDrawRectangle(wxCoord x, wxCoord y, wxCoord width, wxCoord h
     wxCoord y2 = y + height;
 
     wxCoord x2dev = XLOG2DEV(x2),
-            y2dev = YLOG2DEV(y2);
+            y2dev = YLOG2DEV(y2),
+            x1dev = XLOG2DEV(x);
 
     // Windows (but not Windows CE) draws the filled rectangles without outline
     // (i.e. drawn with a transparent pen) one pixel smaller in both directions
@@ -1043,12 +1044,23 @@ void wxMSWDCImpl::DoDrawRectangle(wxCoord x, wxCoord y, wxCoord width, wxCoord h
 #ifndef __WXWINCE__
     if ( m_pen.IsTransparent() )
     {
-        x2dev++;
+        // Right edge to be extended is "displayed right edge"
+        // and hence its device coordinates depend
+        // on layout direction and can be either x1 or x2.
+        const DWORD dwLayout = ::GetLayout(GetHdc());
+        if ( dwLayout != GDI_ERROR && (dwLayout & LAYOUT_RTL) )
+        {
+            x1dev--;
+        }
+        else
+        {
+            x2dev++;
+        }
         y2dev++;
     }
 #endif // !__WXWINCE__
 
-    (void)Rectangle(GetHdc(), XLOG2DEV(x), YLOG2DEV(y), x2dev, y2dev);
+    (void)Rectangle(GetHdc(), x1dev, YLOG2DEV(y), x2dev, y2dev);
 
     CalcBoundingBox(x, y);
     CalcBoundingBox(x2, y2);
