Index: include/wx/generic/textdlgg.h
===================================================================
--- include/wx/generic/textdlgg.h	(revision 72560)
+++ include/wx/generic/textdlgg.h	(working copy)
@@ -42,8 +42,18 @@
                       const wxString& caption = wxGetTextFromUserPromptStr,
                       const wxString& value = wxEmptyString,
                       long style = wxTextEntryDialogStyle,
-                      const wxPoint& pos = wxDefaultPosition);
+                      const wxPoint& pos = wxDefaultPosition)
+    {
+        Create(parent, message, caption, value, style, pos);
+    }
 
+    bool Create(wxWindow *parent,
+                const wxString& message,
+                const wxString& caption = wxGetTextFromUserPromptStr,
+                const wxString& value = wxEmptyString,
+                long style = wxTextEntryDialogStyle,
+                const wxPoint& pos = wxDefaultPosition);
+
     void SetValue(const wxString& val);
     wxString GetValue() const { return m_value; }
 
Index: interface/wx/textdlg.h
===================================================================
--- interface/wx/textdlg.h	(revision 72560)
+++ interface/wx/textdlg.h	(working copy)
@@ -78,8 +78,15 @@
 {
 public:
     /**
-        Constructor. Use ShowModal() to show the dialog.
+        Constructor. Use ShowModal() to show the dialog. See Create method for parameter description.
+    */
+    wxTextEntryDialog(wxWindow* parent, const wxString& message,
+                      const wxString& caption = wxGetTextFromUserPromptStr,
+                      const wxString& value = wxEmptyString,
+                      long style = wxTextEntryDialogStyle,
+                      const wxPoint& pos = wxDefaultPosition);
 
+    /**
         @param parent
             Parent window.
         @param message
@@ -95,8 +102,10 @@
             here.
         @param pos
             Dialog position.
+
+        @since 2.9.5
     */
-    wxTextEntryDialog(wxWindow* parent, const wxString& message,
+    bool Create(wxWindow* parent, const wxString& message,
                       const wxString& caption = wxGetTextFromUserPromptStr,
                       const wxString& value = wxEmptyString,
                       long style = wxTextEntryDialogStyle,
Index: src/generic/textdlgg.cpp
===================================================================
--- src/generic/textdlgg.cpp	(revision 72560)
+++ src/generic/textdlgg.cpp	(working copy)
@@ -65,17 +65,20 @@
 
 IMPLEMENT_CLASS(wxTextEntryDialog, wxDialog)
 
-wxTextEntryDialog::wxTextEntryDialog(wxWindow *parent,
+bool wxTextEntryDialog::Create(wxWindow *parent,
                                      const wxString& message,
                                      const wxString& caption,
                                      const wxString& value,
                                      long style,
                                      const wxPoint& pos)
-                 : wxDialog(GetParentForModalDialog(parent, style),
-                            wxID_ANY, caption, pos, wxDefaultSize,
-                            wxDEFAULT_DIALOG_STYLE),
-                   m_value(value)
 {
+    if (!wxDialog::Create(GetParentForModalDialog(parent, style),
+                wxID_ANY, caption, pos, wxDefaultSize,
+                wxDEFAULT_DIALOG_STYLE | wxRESIZE_BORDER))
+    {
+        return false;
+    }
+
     m_dialogStyle = style;
     m_value = value;
 
@@ -126,6 +129,7 @@
     m_textctrl->SetFocus();
 
     wxEndBusyCursor();
+    return true;
 }
 
 void wxTextEntryDialog::OnOK(wxCommandEvent& WXUNUSED(event) )
