Index: samples/dataview/dataview.cpp
===================================================================
--- samples/dataview/dataview.cpp	(revision 71081)
+++ samples/dataview/dataview.cpp	(working copy)
@@ -559,7 +559,7 @@
 
             tr = new wxDataViewTextRenderer( "string", wxDATAVIEW_CELL_EDITABLE );
             wxDataViewColumn *column1 =
-                new wxDataViewColumn( "artist", tr, 1, 150, wxALIGN_LEFT,
+                new wxDataViewColumn( "artist", tr, 1, wxCOL_WIDTH_AUTOSIZE, wxALIGN_LEFT,
                                       wxDATAVIEW_COL_SORTABLE | wxDATAVIEW_COL_REORDERABLE |
                                       wxDATAVIEW_COL_RESIZABLE );
             column1->SetMinWidth(150); // this column can't be resized to be smaller
