Index: samples/grid/griddemo.cpp
===================================================================
--- samples/grid/griddemo.cpp	(revision 64696)
+++ samples/grid/griddemo.cpp	(working copy)
@@ -157,8 +157,9 @@
     EVT_MENU( ID_TOGGLECOLMOVING, GridFrame::ToggleColMoving )
     EVT_MENU( ID_TOGGLEGRIDSIZING, GridFrame::ToggleGridSizing )
     EVT_MENU( ID_TOGGLEGRIDDRAGCELL, GridFrame::ToggleGridDragCell )
+    EVT_MENU( ID_COLDEFAULTHEADER, GridFrame::SetDefaultColHeader )
     EVT_MENU( ID_COLNATIVEHEADER, GridFrame::SetNativeColHeader )
-    EVT_MENU( ID_COLDEFAULTHEADER, GridFrame::SetDefaultColHeader )
+    EVT_MENU( ID_COLNATIVEHEADERCTRL, GridFrame::SetNativeColHeader )
     EVT_MENU( ID_COLCUSTOMHEADER, GridFrame::SetCustomColHeader )
     EVT_MENU( ID_TOGGLEGRIDLINES, GridFrame::ToggleGridLines )
     EVT_MENU( ID_AUTOSIZECOLS, GridFrame::AutoSizeCols )
@@ -283,7 +284,8 @@
                       wxT("Change style of col header") );
 
     colHeaderMenu->AppendRadioItem( ID_COLDEFAULTHEADER, wxT("&Default") );
-    colHeaderMenu->AppendRadioItem( ID_COLNATIVEHEADER, wxT("&Native") );
+    colHeaderMenu->AppendRadioItem( ID_COLNATIVEHEADER, wxT("&Native labels") );
+    colHeaderMenu->AppendRadioItem( ID_COLNATIVEHEADERCTRL, wxT("&Native headers") );
     colHeaderMenu->AppendRadioItem( ID_COLCUSTOMHEADER, wxT("&Custom") );
 
 
@@ -593,28 +595,38 @@
         GetMenuBar()->IsChecked( ID_TOGGLEGRIDDRAGCELL ) );
 }
 
-void GridFrame::SetNativeColHeader( wxCommandEvent& WXUNUSED(ev) )
+void GridFrame::SetNativeColHeader( wxCommandEvent& ev )
 {
-    CustomColumnHeadersProvider* provider =
-        static_cast<CustomColumnHeadersProvider*>(grid->GetTable()->GetAttrProvider());
-    provider->UseCustomColHeaders(false);
-    grid->SetUseNativeColLabels(true);
+    if( ev.GetId() == ID_COLNATIVEHEADERCTRL)
+    {
+        grid->SetUseNativeColLabels(false);
+        grid->UseNativeColHeader();
+    }
+    else
+    {
+        grid->UseNativeColHeader(false);
+        CustomColumnHeadersProvider* provider =
+            static_cast<CustomColumnHeadersProvider*>(grid->GetTable()->GetAttrProvider());
+        provider->UseCustomColHeaders(false);
+        grid->SetUseNativeColLabels(true);
+    }
 }
 
 void GridFrame::SetCustomColHeader( wxCommandEvent& WXUNUSED(ev) )
 {
+    grid->SetUseNativeColLabels(false);
     CustomColumnHeadersProvider* provider =
         static_cast<CustomColumnHeadersProvider*>(grid->GetTable()->GetAttrProvider());
     provider->UseCustomColHeaders(true);
-    grid->SetUseNativeColLabels(false);
 }
 
 void GridFrame::SetDefaultColHeader( wxCommandEvent& WXUNUSED(ev) )
 {
+    grid->UseNativeColHeader(false);
+    grid->SetUseNativeColLabels(false);
     CustomColumnHeadersProvider* provider =
         static_cast<CustomColumnHeadersProvider*>(grid->GetTable()->GetAttrProvider());
     provider->UseCustomColHeaders(false);
-    grid->SetUseNativeColLabels(false);
 }
 
 
Index: samples/grid/griddemo.h
===================================================================
--- samples/grid/griddemo.h	(revision 64696)
+++ samples/grid/griddemo.h	(working copy)
@@ -138,6 +138,7 @@
         ID_COLLABELVERTALIGN,
         ID_COLDEFAULTHEADER,
         ID_COLNATIVEHEADER,
+        ID_COLNATIVEHEADERCTRL,
         ID_COLCUSTOMHEADER,
         ID_GRIDLINECOLOUR,
         ID_INSERTROW,
