Index: grid.cpp
===================================================================
--- grid.cpp	(revision 70954)
+++ grid.cpp	(working copy)
@@ -6452,11 +6452,11 @@
     }
     else // don't expand selection
     {
-        ClearSelection();
-
         if ( diroper.IsAtBoundary(m_currentCellCoords) )
             return false;
 
+	ClearSelection();
+	
         wxGridCellCoords coords = m_currentCellCoords;
         diroper.Advance(coords);
 
