=== src/gtk/toplevel.cpp
==================================================================
--- src/gtk/toplevel.cpp	(revision 1143)
+++ src/gtk/toplevel.cpp	(local)
@@ -46,6 +46,7 @@
 #include "wx/gtk/win_gtk.h"
 
 #include "wx/unix/utilsx11.h"
+#include "wx/sysopt.h"
 
 // XA_CARDINAL
 #include <X11/Xatom.h>
@@ -1323,11 +1324,17 @@
     return child;
 }
 
+static const wxChar *transparentOption = wxT("gtk.transparentoverride");
+
 bool wxTopLevelWindowGTK::SetTransparent(wxByte alpha)
 {
     if (!m_widget || !m_widget->window)
         return false;
 
+    // If an override is specified, return it
+    if (wxSystemOptions::HasOption(transparentOption))
+        return (wxSystemOptions::GetOptionInt(transparentOption) != 0);
+
     Display* dpy = GDK_WINDOW_XDISPLAY (m_widget->window);
     // We need to get the X Window that has the root window as the immediate parent
     // and m_widget->window as a child. This should be the X Window that the WM manages and
@@ -1349,6 +1356,12 @@
 
 bool wxTopLevelWindowGTK::CanSetTransparent()
 {
+    if (!m_widget || !m_widget->window)
+        return false;
+
+    if (wxSystemOptions::HasOption(transparentOption) && !wxSystemOptions::GetOptionInt(transparentOption))
+        return false;
+
 #if GTK_CHECK_VERSION(2,10,0)
     if (!gtk_check_version(2,10,0))
     {

 	  	 
