Index: src/gtk/dcclient.cpp
===================================================================
--- src/gtk/dcclient.cpp	(revision 55163)
+++ src/gtk/dcclient.cpp	(working copy)
@@ -954,6 +954,14 @@
             bool originChanged;
             DrawingSetup(gc, originChanged);
 
+            // If the pen is transparent pen we increase the size
+            // for better compatibility with other platforms.
+            if (m_pen.GetStyle() == wxPENSTYLE_TRANSPARENT)
+            {
+                ++ww;
+                ++hh;
+            }
+
             gdk_draw_arc(m_gdkwindow, gc, true, xx, yy, ww, hh, 0, 360*64);
 
             if (originChanged)
@@ -961,7 +969,7 @@
         }
 
         if (m_pen.GetStyle() != wxPENSTYLE_TRANSPARENT)
-            gdk_draw_arc( m_gdkwindow, m_penGC, FALSE, xx, yy, ww, hh, 0, 360*64 );
+            gdk_draw_arc( m_gdkwindow, m_penGC, false, xx, yy, ww, hh, 0, 360*64 );
     }
 
     CalcBoundingBox( x, y );
