diff -u -3 -P -r wxWidgets/.\include\wx\filefn.h wxW/.\include\wx\filefn.h
--- wxWidgets/.\include\wx\filefn.h	Mon Dec 20 13:48:21 2004
+++ wxW/.\include\wx\filefn.h	Thu Dec 23 12:24:02 2004
@@ -169,6 +169,7 @@
     || ( defined(__MINGW32__) && !defined(__WINE__) && wxCHECK_W32API_VERSION( 0, 5 ) ) \
     || ( defined(__MWERKS__) && defined(__WXMSW__) ) \
     || ( defined(__DMC__) && defined(__WXMSW__) ) \
+    || ( defined(__BORLANDC__) && defined(__WXMSW__) ) \
     || ( defined(__WATCOMC__) && defined(__WXMSW__) )
 
     // detect compilers which have support for huge files (currently only
@@ -183,6 +184,8 @@
     #elif defined(__MWERKS__)
         #define wxHAS_HUGE_FILES 0
     #elif defined(__DMC__)
+        #define wxHAS_HUGE_FILES 0
+    #elif defined(__BORLANDC__)
         #define wxHAS_HUGE_FILES 0
     #elif ((_INTEGRAL_MAX_BITS >= 64) || defined(_LARGE_FILES))
         #define wxHAS_HUGE_FILES 1
diff -u -3 -P -r wxWidgets/.\include\wx\msw\setup0.h wxW/.\include\wx\msw\setup0.h
--- wxWidgets/.\include\wx\msw\setup0.h	Fri Nov 12 07:08:37 2004
+++ wxW/.\include\wx\msw\setup0.h	Fri Nov 12 09:10:01 2004
@@ -645,7 +645,11 @@
 // Default is 0 because it isn't yet implemented on all platforms
 //
 // Recommended setting: 1 if you need it, can be safely set to 0 otherwise
-#define wxUSE_DISPLAY       0
+#if defined(__BORLANDC__)
+    #define wxUSE_DISPLAY       1
+#else
+    #define wxUSE_DISPLAY       0
+#endif
 
 // Miscellaneous geometry code: needed for Canvas library
 #define wxUSE_GEOMETRY            1
@@ -845,7 +849,11 @@
 // Default is 0.
 //
 // Recommended setting: 1 if you intend to use OpenGL, 0 otherwise
+#ifdef __DMC__
 #define wxUSE_GLCANVAS       0
+#else
+#define wxUSE_GLCANVAS       1
+#endif
 
 // ----------------------------------------------------------------------------
 // Data transfer
@@ -938,7 +946,11 @@
 // ----------------------------------------------------------------------------
 
 // Define 1 to use ODBC classes
+#ifdef __DMC__
 #define wxUSE_ODBC          0
+#else
+#define wxUSE_ODBC          1
+#endif
 
 // For backward compatibility reasons, this parameter now only controls the
 // default scrolling method used by cursors.  This default behavior can be
diff -u -3 -P -r wxWidgets/.\include\wx\strconv.h wxW/.\include\wx\strconv.h
--- wxWidgets/.\include\wx\strconv.h	Thu Nov 18 07:35:28 2004
+++ wxW/.\include\wx\strconv.h	Thu Dec 23 12:22:33 2004
@@ -237,8 +237,11 @@
 // ----------------------------------------------------------------------------
 
 // filenames are multibyte on Unix and probably widechar on Windows?
-#if defined(__UNIX__) || defined(__BORLANDC__) || defined(__WXMAC__ )
+#if defined(__UNIX__) || defined(__WXMAC__ )
     #define wxMBFILES 1
+// file functions take wide chars on windows
+#elif defined(__WXMSW__) && wxUSE UNICODE
+    #define wxMBFILES 0
 #else
     #define wxMBFILES 0
 #endif
diff -u -3 -P -r wxWidgets/.\src\common\filefn.cpp wxW/.\src\common\filefn.cpp
--- wxWidgets/.\src\common\filefn.cpp	Wed Dec 22 08:40:47 2004
+++ wxW/.\src\common\filefn.cpp	Thu Dec 23 12:14:41 2004
@@ -951,9 +951,9 @@
   FILE *fp2 = NULL;
   FILE *fp3 = NULL;
   // Open the inputs and outputs
-  if ((fp1 = wxFopen ( file1, wxT("rb"))) == NULL ||
-      (fp2 = wxFopen ( file2, wxT("rb"))) == NULL ||
-      (fp3 = wxFopen ( outfile, wxT("wb"))) == NULL)
+  if ((fp1 = wxFopen ( file1.fn_str(), wxT("rb"))) == NULL ||
+      (fp2 = wxFopen ( file2.fn_str(), wxT("rb"))) == NULL ||
+      (fp3 = wxFopen ( outfile.fn_str(), wxT("wb"))) == NULL)
     {
       if (fp1)
         fclose (fp1);
diff -u -3 -P -r wxWidgets/.\src\common\wfstream.cpp wxW/.\src\common\wfstream.cpp
--- wxWidgets/.\src\common\wfstream.cpp	Wed Dec 22 08:40:51 2004
+++ wxW/.\src\common\wfstream.cpp	Thu Dec 23 12:15:28 2004
@@ -33,7 +33,7 @@
 wxFileInputStream::wxFileInputStream(const wxString& fileName)
   : wxInputStream()
 {
-    m_file = new wxFile(fileName, wxFile::read);
+    m_file = new wxFile(fileName.fn_str(), wxFile::read);
     m_file_destroy = true;
 }
 
@@ -109,7 +109,7 @@
 
 wxFileOutputStream::wxFileOutputStream(const wxString& fileName)
 {
-    m_file = new wxFile(fileName, wxFile::write);
+    m_file = new wxFile(fileName.fn_str(), wxFile::write);
     m_file_destroy = true;
 
     if (!m_file->IsOpened())
diff -u -3 -P -r wxWidgets/.\src\generic\dcpsg.cpp wxW/.\src\generic\dcpsg.cpp
--- wxWidgets/.\src\generic\dcpsg.cpp	Wed Dec 22 08:41:58 2004
+++ wxW/.\src\generic\dcpsg.cpp	Thu Dec 23 12:16:54 2004
@@ -1561,7 +1561,7 @@
             m_printData.SetFilename(filename);
         }
 
-        m_pstream = wxFopen( m_printData.GetFilename(), wxT("w+") );
+        m_pstream = wxFopen( m_printData.GetFilename().fn_str(), wxT("w+") );
 
         if (!m_pstream)
         {
@@ -1971,7 +1971,7 @@
         {
             afmName = data->GetFontMetricPath();
             afmName << wxFILE_SEP_PATH << name;
-            afmFile = wxFopen(afmName,wxT("r"));
+            afmFile = wxFopen(afmName.fn_str(),wxT("r"));
         }
 
 #if defined(__UNIX__) && !defined(__VMS__)
@@ -1985,7 +1985,7 @@
                    << wxT("afm") << wxFILE_SEP_PATH
 #endif
                    << name;
-           afmFile = wxFopen(afmName,wxT("r"));
+           afmFile = wxFopen(afmName.fn_str(),wxT("r"));
         }
 #endif
 
diff -u -3 -P -r wxWidgets/.\src\generic\helpext.cpp wxW/.\src\generic\helpext.cpp
--- wxWidgets/.\src\generic\helpext.cpp	Wed Dec 22 08:41:59 2004
+++ wxW/.\src\generic\helpext.cpp	Thu Dec 23 12:17:28 2004
@@ -287,7 +287,7 @@
    m_MapList = new wxList;
    m_NumOfEntries = 0;
 
-   FILE *input = wxFopen(mapFile,wxT("rt"));
+   FILE *input = wxFopen(mapFile.fn_str(),wxT("rt"));
    if(! input)
       return false;
    do

 	  	 
