Index: src/richtext/richtextbuffer.cpp
===================================================================
--- src/richtext/richtextbuffer.cpp	(revision 59699)
+++ src/richtext/richtextbuffer.cpp	(working copy)
@@ -1144,6 +1144,11 @@
             wxRichTextParagraph* firstPara = wxDynamicCast(firstParaNode->GetData(), wxRichTextParagraph);
             wxASSERT (firstPara != NULL);
 
+            // Apply the new paragraph attributes to the existing paragraph
+            wxTextAttrEx attr(para->GetAttributes());
+            wxRichTextApplyStyle(attr, firstPara->GetAttributes());
+            para->SetAttributes(attr);
+
             wxRichTextObjectList::compatibility_iterator objectNode = firstPara->GetChildren().GetFirst();
             while (objectNode)
             {
