Index: samples/widgets/static.cpp
===================================================================
--- samples/widgets/static.cpp	(revision 63959)
+++ samples/widgets/static.cpp	(working copy)
@@ -311,6 +311,8 @@
 
     CreateStatic();
 
+    m_textLabel->Bind(wxEVT_COMMAND_TEXT_UPDATED,&StaticWidgetsPage::OnButtonLabelText,this);
+
     // the 3 panes panes compose the window
     sizerTop->Add(sizerLeft, 0, wxGROW | (wxALL & ~wxLEFT), 10);
     sizerTop->Add(sizerMiddle, 0, wxGROW | wxALL, 10);
@@ -514,6 +516,7 @@
 
 void StaticWidgetsPage::OnButtonLabelText(wxCommandEvent& WXUNUSED(event))
 {
+    m_statText->SetFont(m_statText->GetFont().Italic());
     m_statText->SetLabel(m_textLabel->GetValue());
 
     // test GetLabel() and GetLabelText(); the first should return the
