diff -ru -U 2 -xCVS -x'.#*' -x'*.orig' -x'*.rej' -x'*~' wxWindows.cvs/src/generic/treectlg.cpp wxWindows/src/generic/treectlg.cpp
--- wxWindows.cvs/src/generic/treectlg.cpp	2005-01-10 16:21:21.000000000 -0600
+++ wxWindows/src/generic/treectlg.cpp	2005-01-27 09:43:16.000000000 -0600
@@ -3177,6 +3177,16 @@
 
         if ( event.RightDown() )
         {
+            // when an unselected item is right-clicked,
+            // clear the existing selection (if any)
+            // and do a single-select on the item
+            // being right-clicked
+
+            if (!IsSelected(item))
+            {
+                DoSelectItem( item, true, false );
+            }
+
             wxTreeEvent nevent(wxEVT_COMMAND_TREE_ITEM_RIGHT_CLICK, GetId());
             nevent.m_item = item;
             nevent.m_pointDrag = CalcScrolledPosition(pt);

 	  	 
