From 98c48c8510e39fff2926dbc653acbbd1615ac304 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Gunter=20K=C3=B6nigsmann?= <gunter@peterpall.de>
Date: Thu, 31 Aug 2017 00:35:14 +0200
Subject: [PATCH] An absolute file name here causes an assert.

---
 samples/config/conftest.cpp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/samples/config/conftest.cpp b/samples/config/conftest.cpp
index ae00f2841b..287a00368c 100644
--- a/samples/config/conftest.cpp
+++ b/samples/config/conftest.cpp
@@ -101,16 +101,16 @@ bool MyApp::OnInit()
     SetVendorName(wxT("wxWidgets"));
     SetAppName(wxT("conftest")); // not needed, it's the default value
 
-    wxConfigBase *pConfig = wxConfigBase::Get();
+    //    wxConfigBase *pConfig = wxConfigBase::Get();
 
     // uncomment this to force writing back of the defaults for all values
     // if they're not present in the config - this can give the user an idea
     // of all possible settings for this program
-    pConfig->SetRecordDefaults();
+    //    pConfig->SetRecordDefaults();
 
     // or you could also write something like this:
-    //  wxFileConfig *pConfig = new wxFileConfig("conftest");
-    //  wxConfigBase::Set(pConfig);
+    wxFileConfig *pConfig = new wxFileConfig("~/conftest");
+    wxConfigBase::Set(pConfig);
     // where you can also specify the file names explicitly if you wish.
     // Of course, calling Set() is optional and you only must do it if
     // you want to later retrieve this pointer with Get().
-- 
2.14.1

