Index: src/common/sizer.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/common/sizer.cpp,v
retrieving revision 1.55
diff -u -r1.55 sizer.cpp
--- src/common/sizer.cpp	2003/01/27 09:02:51	1.55
+++ src/common/sizer.cpp	2003/01/27 11:23:07
@@ -1419,8 +1419,8 @@
         *borderTop = 5;
     else
 #endif // __WXGTK__
-        *borderTop = 15;
-    (void)box;
+        *borderTop = box->GetCharHeight();
+
     *borderOther = 5;
 }
 

 	  	 
