ndex: src/generic/spinctlg.cpp
===================================================================
--- src/generic/spinctlg.cpp	(revisión: 72955)
+++ src/generic/spinctlg.cpp	(copia de trabajo)
@@ -269,6 +269,14 @@
     wxDELETE(m_spinButton);
 }
 
+wxWindowList wxSpinCtrlGenericBase::GetCompositeWindowParts() const
+{
+    wxWindowList parts;
+    parts.push_back(m_textCtrl);
+    parts.push_back(m_spinButton);
+    return parts;
+}
+
 // ----------------------------------------------------------------------------
 // geometry
 // ----------------------------------------------------------------------------
Index: include/wx/generic/spinctlg.h
===================================================================
--- include/wx/generic/spinctlg.h	(revisión: 72955)
+++ include/wx/generic/spinctlg.h	(copia de trabajo)
@@ -22,6 +22,8 @@
 
 #if wxUSE_SPINBTN
 
+#include "wx/compositewin.h"
+
 class WXDLLIMPEXP_FWD_CORE wxSpinButton;
 class WXDLLIMPEXP_FWD_CORE wxTextCtrl;
 
@@ -40,7 +42,8 @@
 // function ambiguity.
 // ----------------------------------------------------------------------------
 
-class WXDLLIMPEXP_CORE wxSpinCtrlGenericBase : public wxSpinCtrlBase
+class WXDLLIMPEXP_CORE wxSpinCtrlGenericBase
+                : public wxCompositeWindow<wxSpinCtrlBase>
 {
 public:
     wxSpinCtrlGenericBase() { Init(); }
@@ -156,6 +159,9 @@
     // common part of all ctors
     void Init();
 
+    // Implement pure virtual function inherited from wxCompositeWindow.
+    virtual wxWindowList GetCompositeWindowParts() const;
+
     DECLARE_EVENT_TABLE()
 };
 

