Index: src/msw/statbox.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/msw/statbox.cpp,v
retrieving revision 1.84
diff -u -r1.84 statbox.cpp
--- src/msw/statbox.cpp	2005/09/23 12:55:11	1.84
+++ src/msw/statbox.cpp	2006/01/10 02:37:13
@@ -151,6 +151,13 @@
             *exstyle = 0;
     }
 
+    // alignment options
+
+    if (style & wxALIGN_CENTRE)
+        styleWin |= BS_CENTER;
+    else if (style & wxALIGN_RIGHT)
+        styleWin |= BS_RIGHT;
+
     return styleWin | BS_GROUPBOX;
 }
 
@@ -385,11 +392,23 @@
         const int y = dc.GetCharHeight();
         const int x = 9;
 
+        UINT fmt = DT_SINGLELINE | DT_VCENTER;
+        if (HasFlag(wxALIGN_CENTRE))
+        {
+            fmt |= DT_CENTER;
+        }
+        else if (HasFlag(wxALIGN_RIGHT))
+        {
+            fmt |= DT_RIGHT;
+        }
+
         // TODO: RTL?
-        RECT rc = { x, 0, GetSize().x - x, y };
+        // use "+ 2" to better match win32's text positioning
+        // when using non-default alignment
+        RECT rc = { x, 0, GetSize().x - x + 2, y };
 
         const wxString label = GetLabel();
-        ::DrawText(hdc, label, label.length(), &rc, DT_SINGLELINE | DT_VCENTER);
+        ::DrawText(hdc, label, label.length(), &rc, fmt);
     }
 }
 


 	  	 
