diff -rbu2 wxGTK-2.2.9/include/wx/gtk/textctrl.h wxGTK-2.2.9.orig/include/wx/gtk/textctrl.h
--- wxGTK-2.2.9/include/wx/gtk/textctrl.h	Fri Aug  2 12:00:13 2002
+++ wxGTK-2.2.9.orig/include/wx/gtk/textctrl.h	Fri Jan  4 16:47:58 2002
@@ -143,11 +143,4 @@
     void SetModified() { m_modified = TRUE; }
 
-    // tell the control to ignore next text changed signal
-    void IgnoreNextTextUpdate();
-
-    // should we ignore the changed signal? always resets the flag
-    bool IgnoreTextUpdate();
-
-
 protected:
     virtual wxSize DoGetBestSize() const;
@@ -158,5 +151,4 @@
     GtkWidget  *m_vScrollbar;
     bool        m_vScrollbarVisible;
-    bool        m_ignoreNextUpdate;
 
     DECLARE_EVENT_TABLE()
diff -rbu2 wxGTK-2.2.9/src/gtk/textctrl.cpp wxGTK-2.2.9.orig/src/gtk/textctrl.cpp
--- wxGTK-2.2.9/src/gtk/textctrl.cpp	Fri Aug  2 16:28:37 2002
+++ wxGTK-2.2.9.orig/src/gtk/textctrl.cpp	Fri Jan  4 16:47:58 2002
@@ -50,7 +50,4 @@
 gtk_text_changed_callback( GtkWidget *WXUNUSED(widget), wxTextCtrl *win )
 {
-    if ( win->IgnoreTextUpdate() )
-        return;
-
     if (!win->m_hasVMT) return;
 
@@ -435,6 +432,4 @@
     {
         gint len = gtk_text_get_length( GTK_TEXT(m_text) );
-
-	IgnoreNextTextUpdate(); 
         gtk_editable_delete_text( GTK_EDITABLE(m_text), 0, len );
         len = 0;
@@ -751,22 +746,4 @@
     m_modified = FALSE;
 }
-
-void wxTextCtrl::IgnoreNextTextUpdate()
-{
-    m_ignoreNextUpdate = TRUE;
-}
-
-bool wxTextCtrl::IgnoreTextUpdate()
-{
-    if ( m_ignoreNextUpdate )
-    {
-        m_ignoreNextUpdate = FALSE;
-
-        return TRUE;
-    }
-
-    return FALSE;
-}
-
 
 void wxTextCtrl::SetSelection( long from, long to )

 	  	 
