diff --git a/samples/text/text.cpp b/samples/text/text.cpp
index ba36a760fa..b32118551f 100644
--- a/samples/text/text.cpp
+++ b/samples/text/text.cpp
@@ -1074,6 +1074,7 @@ MyPanel::MyPanel( wxFrame *frame, int x, int y, int w, int h )
     m_limited = new MyTextCtrl(this, wxID_ANY, "",
                               wxPoint(10, 90), wxDefaultSize);
     m_limited->SetHint("Max 8 ch");
+    m_limited->SetHint(""); // Program now segfaults if you type something into the control.
     m_limited->SetMaxLength(8);
     wxSize size2 = m_limited->GetSizeFromTextSize(m_limited->GetTextExtent("WWWWWWWW"));
     m_limited->SetSizeHints(size2, size2);
