Index: include/wx/generic/dataview.h
===================================================================
--- include/wx/generic/dataview.h	(revision 66360)
+++ include/wx/generic/dataview.h	(working copy)
@@ -223,6 +223,8 @@
     // return the index of the given column in m_cols
     int GetColumnIndex(const wxDataViewColumn *column) const;
 
+    // return index of the column based on model column
+    int GetColumnIndexFromModelColumn(int modelColumnNo) const;
     // return the column displayed at the given position in the control
     wxDataViewColumn *GetColumnAt(unsigned int pos) const;
 
@@ -237,6 +239,7 @@
     // cached column best widths or 0 if not computed, values are for
     // respective columns from m_cols and the arrays have same size
     wxVector<int>             m_colsBestWidths;
+    wxVector<int>             m_modelToDVcolumn;
     wxDataViewModelNotifier  *m_notifier;
     wxDataViewMainWindow     *m_clientArea;
     wxDataViewHeaderWindow   *m_headerArea;
Index: src/generic/datavgen.cpp
===================================================================
--- src/generic/datavgen.cpp	(revision 66360)
+++ src/generic/datavgen.cpp	(working copy)
@@ -637,7 +637,8 @@
     virtual bool ItemChanged( const wxDataViewItem & item )
         { return m_mainWindow->ItemChanged(item);  }
     virtual bool ValueChanged( const wxDataViewItem & item , unsigned int col )
-        { return m_mainWindow->ValueChanged( item, col ); }
+    {       col = m_mainWindow->GetOwner()->GetColumnIndexFromModelColumn(col);
+            return m_mainWindow->ValueChanged( item, col ); }
     virtual bool Cleared()
         { return m_mainWindow->Cleared(); }
     virtual void Resort()
@@ -4037,6 +4038,13 @@
     if (!wxDataViewCtrlBase::AppendColumn(col))
         return false;
 
+    // resize the map to fit model_column index
+    if (m_modelToDVcolumn.size() < col->GetModelColumn()+1) 
+        m_modelToDVcolumn.resize(col->GetModelColumn()+1);
+    
+    // set model_column map to dv column.
+    m_modelToDVcolumn[col->GetModelColumn()] = m_cols.size();
+
     m_cols.Append( col );
     m_colsBestWidths.push_back(0);
     OnColumnsCountChanged();
@@ -4123,6 +4131,11 @@
     return wxNOT_FOUND;
 }
 
+int wxDataViewCtrl::GetColumnIndexFromModelColumn(int idx) const {
+    if (idx >= m_modelToDVcolumn.size()) return wxNOT_FOUND;
+    return m_modelToDVcolumn[idx];
+}
+
 unsigned int wxDataViewCtrl::GetBestColumnWidth(int idx) const
 {
     if ( m_colsBestWidths[idx] != 0 )
