Index: include/wx/xti.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/xti.h,v
retrieving revision 1.45
diff -u -r1.45 xti.h
--- include/wx/xti.h	2003/10/17 19:50:46	1.45
+++ include/wx/xti.h	2003/10/18 23:07:35
@@ -1656,13 +1656,13 @@
     // gets the streaming callback from this class or any superclass
     wxObjectStreamingCallback GetStreamingCallback() const ;
 
-#ifdef WXWIN_COMPATIBILITY_2_4
+#if WXWIN_COMPATIBILITY_2_4
     // Initializes parent pointers and hash table for fast searching.
-    wxDEPRECATED( static void     InitializeClasses() );
+    wxDEPRECATED( static void InitializeClasses() );
     // Cleans up hash table used for fast searching.
-    wxDEPRECATED( static void     CleanUpClasses() );
+    wxDEPRECATED( static void CleanUpClasses() );
 #endif
-    static void     CleanUp();
+    static void CleanUp();
 
     // returns the first property
     const wxPropertyInfo* GetFirstProperty() const { return m_firstProperty ; }
Index: include/wx/wizard.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/wizard.h,v
retrieving revision 1.32
diff -u -r1.32 wizard.h
--- include/wx/wizard.h	2003/08/18 09:30:45	1.32
+++ include/wx/wizard.h	2003/10/18 23:08:17
@@ -204,7 +204,7 @@
     virtual void SetBorder(int border) = 0;
     
     // wxWizard should be created using "new wxWizard" now, not with Create()
-#ifdef WXWIN_COMPATIBILITY_2_2
+#if WXWIN_COMPATIBILITY_2_2
     static wxWizard *Create(wxWindow *parent,
                             int id = -1,
                             const wxString& title = wxEmptyString,
Index: include/wx/frame.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/frame.h,v
retrieving revision 1.41
diff -u -r1.41 frame.h
--- include/wx/frame.h	2003/10/07 22:54:59	1.41
+++ include/wx/frame.h	2003/10/18 23:08:51
@@ -170,7 +170,7 @@
     // if you are hiding the help, TRUE otherwise
     virtual void DoGiveHelp(const wxString& text, bool show);
 
-#ifdef WXWIN_COMPATIBILITY_2_2
+#if WXWIN_COMPATIBILITY_2_2
     // call this to simulate a menu command
     bool Command(int winid) { return ProcessCommand(winid); }
 #endif // WXWIN_COMPATIBILITY_2_2
Index: include/wx/object.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/object.h,v
retrieving revision 1.95
diff -u -r1.95 object.h
--- include/wx/object.h	2003/10/16 11:33:35	1.95
+++ include/wx/object.h	2003/10/18 23:09:36
@@ -107,11 +107,11 @@
                  ( m_baseInfo2 && m_baseInfo2->IsKindOf(info) ) );
     }
 
-#ifdef WXWIN_COMPATIBILITY_2_4
+#if WXWIN_COMPATIBILITY_2_4
     // Initializes parent pointers and hash table for fast searching.
-    wxDEPRECATED( static void     InitializeClasses() );
+    wxDEPRECATED( static void InitializeClasses() );
     // Cleans up hash table used for fast searching.
-    wxDEPRECATED( static void     CleanUpClasses() );
+    wxDEPRECATED( static void CleanUpClasses() );
 #endif
     static void     CleanUp();
     
@@ -149,7 +149,7 @@
 
 WXDLLIMPEXP_BASE wxObject *wxCreateDynamicObject(const wxChar *name);
 
-#ifdef WXWIN_COMPATIBILITY_2_4
+#if WXWIN_COMPATIBILITY_2_4
 inline void wxClassInfo::InitializeClasses() {}
 inline void wxClassInfo::CleanUpClasses() {}
 #endif
Index: include/wx/gtk/gauge.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/gtk/gauge.h,v
retrieving revision 1.13
diff -u -r1.13 gauge.h
--- include/wx/gtk/gauge.h	2003/09/24 20:08:25	1.13
+++ include/wx/gtk/gauge.h	2003/10/18 23:10:17
@@ -87,8 +87,8 @@
         m_gaugePos;
 
     // obsolete functions, don't use
-#ifdef WXWIN_COMPATIBILITY_2_2
-    bool GetProgressBar() const { return TRUE; }
+#if WXWIN_COMPATIBILITY_2_2
+    bool GetProgressBar() const { return true; }
 #endif // WXWIN_COMPATIBILITY_2_2
 
 protected:
Index: src/generic/wizard.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/generic/wizard.cpp,v
retrieving revision 1.49
diff -u -r1.49 wizard.cpp
--- src/generic/wizard.cpp	2003/10/16 07:22:19	1.49
+++ src/generic/wizard.cpp	2003/10/18 23:11:21
@@ -738,7 +738,7 @@
 // our public interface
 // ----------------------------------------------------------------------------
 
-#ifdef WXWIN_COMPATIBILITY_2_2
+#if WXWIN_COMPATIBILITY_2_2
 
 /* static */
 wxWizard *wxWizardBase::Create(wxWindow *parent,

 	  	 
