Index: interface/wx/image.h
===================================================================
--- interface/wx/image.h        (revisión: 63753)
+++ interface/wx/image.h        (copia de trabajo)
@@ -1397,6 +1397,13 @@
     void SetAlpha(int x, int y, unsigned char alpha);

     /**
+        Removes the alpha channel from the image.
+        This function should only be called if the image has alpha channel data,
+        use HasAlpha() to check for this.
+    */
+    void ClearAlpha();
+
+    /**
         Sets the image data without performing checks.

         The data given must have the size (width*height*3) or results will be
Index: include/wx/image.h
===================================================================
--- include/wx/image.h  (revisión: 63753)
+++ include/wx/image.h  (copia de trabajo)
@@ -451,6 +451,7 @@
     bool HasAlpha() const { return GetAlpha() != NULL; }
     void SetAlpha(unsigned char *alpha = NULL, bool static_data=false);
     void InitAlpha();
+    void ClearAlpha();

     // return true if this pixel is masked or has alpha less than specified
     // threshold
Index: src/common/image.cpp
===================================================================
--- src/common/image.cpp        (revisión: 63753)
+++ src/common/image.cpp        (copia de trabajo)
@@ -1857,6 +1857,17 @@
     }
 }

+void wxImage::ClearAlpha()
+{
+    wxCHECK_RET( HasAlpha(), wxT("image already doesn't have an alpha channel") );
+
+    if ( !M_IMGDATA->m_staticAlpha )
+        free( M_IMGDATA->m_alpha );
+
+    M_IMGDATA->m_alpha = NULL;
+}
+
+
 // ----------------------------------------------------------------------------
 // mask support
 // ----------------------------------------------------------------------------