Index: include/wx/colour.h
===================================================================
--- include/wx/colour.h	(revision 61972)
+++ include/wx/colour.h	(working copy)
@@ -90,7 +90,7 @@
              ChannelType green,
              ChannelType blue,
              ChannelType alpha = wxALPHA_OPAQUE)
-        { InitRGBA(red,green,blue, alpha); }
+        { InitRGBA(red, green, blue, alpha); }
 
     // implemented in colourcmn.cpp
     bool Set(const wxString &str)
@@ -119,6 +119,27 @@
     // implemented in colourcmn.cpp
     virtual wxString GetAsString(long flags = wxC2S_NAME | wxC2S_CSS_SYNTAX) const;
 
+    void SetRGB(wxUint32 colRGB)
+    {
+        Set((ChannelType)(0xFF & colRGB),
+            (ChannelType)(0xFF & (colRGB >> 8)),
+            (ChannelType)(0xFF & (colRGB >> 16)));
+    }
+
+    void SetRGBA(wxUint32 colRGBA)
+    {
+        Set((ChannelType)(0xFF & colRGBA),
+            (ChannelType)(0xFF & (colRGBA >> 8)),
+            (ChannelType)(0xFF & (colRGBA >> 16)),
+            (ChannelType)(0xFF & (colRGBA >> 24)));
+    }
+
+    wxUint32 GetRGB() const
+        { return Red() | (Green() << 8) | (Blue() << 16) | (wxALPHA_OPAQUE << 24); }
+
+    wxUint32 GetRGBA() const
+        { return Red() | (Green() << 8) | (Blue() << 16) | (Alpha() << 24); }
+
 #if !wxCOLOUR_IS_GDIOBJECT
     virtual bool IsOk() const= 0;
 
Index: interface/wx/colour.h
===================================================================
--- interface/wx/colour.h	(revision 61972)
+++ interface/wx/colour.h	(working copy)
@@ -112,7 +112,33 @@
     */
     virtual wxString GetAsString(long flags = wxC2S_NAME | wxC2S_CSS_SYNTAX) const;
 
+    //@{
     /**
+        Sets the RGB or RGBA colour values as a single 32 bit value
+        with the 4 least significant bytes in the form
+        0x00BBGGRR and 0xAABBGGRR respectively.
+        Note that the value may depends on endianness.
+
+        @see GetRGB
+    */
+    void SetRGB(wxUint32 colRGB);
+    void SetRGBA(wxUint32 colRGBA);
+    //@}
+
+    //@{
+    /**
+        Gets the RGB or RGBA colour values as a single 32 bit value
+        with the 4 least significant bytes in the form
+        0x00BBGGRR and 0xAABBGGRR respectively.
+        Note that the value may depends on endianness.
+
+        @see SetRGB
+    */
+    wxUint32 GetRGB() const;
+    wxUint32 GetRGBA() const;
+    //@}
+
+    /**
         Returns a pixel value which is platform-dependent.
         On Windows, a COLORREF is returned.
         On X, an allocated pixel value is returned.
