Index: utils/makegen/makegen.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/utils/makegen/makegen.cpp,v
retrieving revision 1.4
diff -u -r1.4 makegen.cpp
--- utils/makegen/makegen.cpp	2001/11/22 22:03:26	1.4
+++ utils/makegen/makegen.cpp	2003/06/03 06:29:36
@@ -83,7 +83,7 @@
 
     wxString fileOutName;
     fileOutName << m_outdir << _T('/') << filename;
-    wxFFile fileOut(fileOutName, "w");
+    wxFFile fileOut(fileOutName, wxT("w"));
     if ( !fileOut.IsOpened() )
     {
         wxLogError(_T("Makefile '%s' couldn't be generated."), filename.c_str());

 	  	 
