Index: src/common/rearrangectrl.cpp
===================================================================
--- src/common/rearrangectrl.cpp	(revision 64194)
+++ src/common/rearrangectrl.cpp	(working copy)
@@ -127,14 +127,27 @@
 {
     wxSwap(m_order[pos1], m_order[pos2]);
 
+    // swap string and checked status
     const wxString stringTmp = GetString(pos1);
     const bool checkedTmp = IsChecked(pos1);
-
     SetString(pos1, GetString(pos2));
     Check(pos1, IsChecked(pos2));
-
     SetString(pos2, stringTmp);
     Check(pos2, checkedTmp);
+
+    // now swap clientdata/object
+    if(GetClientDataType() == wxClientData_Void)
+    {
+        void* vp = GetClientData(pos1);
+        SetClientData(pos1, GetClientData(pos2));
+        SetClientData(pos2, vp);
+    }
+    else if(GetClientDataType() == wxClientData_Object)
+    {
+        wxClientData* cd = GetClientObject(pos1);
+        SetClientObject(pos1, GetClientObject(pos2));
+        SetClientObject(pos2, cd);
+    }
 }
 
 void wxRearrangeList::OnCheck(wxCommandEvent& event)
