diff --git a/include/wx/rearrangectrl.h b/include/wx/rearrangectrl.h
index c83d4ac..5d31208 100644
--- a/include/wx/rearrangectrl.h
+++ b/include/wx/rearrangectrl.h
@@ -14,6 +14,7 @@
 
 #if wxUSE_REARRANGECTRL
 
+#include "wx/ctrlsub.h"
 #include "wx/panel.h"
 #include "wx/dialog.h"
 
@@ -119,7 +120,7 @@ private:
 // wxRearrangeCtrl: composite control containing a wxRearrangeList and buttons
 // ----------------------------------------------------------------------------
 
-class WXDLLIMPEXP_CORE wxRearrangeCtrl : public wxPanel
+class WXDLLIMPEXP_CORE wxRearrangeCtrl : public wxWindowWithItems<wxPanel, wxDelegatingItemContainer>
 {
 public:
     // ctors/Create function are the same as for wxRearrangeList
@@ -176,7 +177,7 @@ private:
 // wxRearrangeDialog: dialog containing a wxRearrangeCtrl
 // ----------------------------------------------------------------------------
 
-class WXDLLIMPEXP_CORE wxRearrangeDialog : public wxDialog
+class WXDLLIMPEXP_CORE wxRearrangeDialog : public wxWindowWithItems<wxDialog, wxDelegatingItemContainer>
 {
 public:
     // default ctor, use Create() later
diff --git a/src/common/rearrangectrl.cpp b/src/common/rearrangectrl.cpp
index ebea20d..1aaa3e2 100644
--- a/src/common/rearrangectrl.cpp
+++ b/src/common/rearrangectrl.cpp
@@ -227,6 +227,8 @@ wxRearrangeCtrl::Create(wxWindow *parent,
     wxButton * const btnUp = new wxButton(this, wxID_UP);
     wxButton * const btnDown = new wxButton(this, wxID_DOWN);
 
+    wxDelegatingItemContainer::Create(m_list);
+
     // arrange them in a sizer
     wxSizer * const sizerBtns = new wxBoxSizer(wxVERTICAL);
     sizerBtns->Add(btnUp, wxSizerFlags().Centre().Border(wxBOTTOM));
@@ -292,7 +294,9 @@ bool wxRearrangeDialog::Create(wxWindow *parent,
 
     m_ctrl = new wxRearrangeCtrl(this, wxID_ANY, wxDefaultPosition, wxDefaultSize,
                                  order, items);
-
+    
+    wxDelegatingItemContainer::Create(m_ctrl);
+    
     // notice that the items in this sizer should be inserted accordingly to
     // wxRearrangeDialogSizerPositions order
     wxSizer * const sizerTop = new wxBoxSizer(wxVERTICAL);
