Index: src/generic/progdlgg.cpp
===================================================================
--- src/generic/progdlgg.cpp	(revision 69927)
+++ src/generic/progdlgg.cpp	(working copy)
@@ -126,6 +126,8 @@
 
     m_winDisabler = NULL;
     m_tempEventLoop = NULL;
+
+    SetWindowStyle(wxDEFAULT_DIALOG_STYLE);
 }
 
 wxGenericProgressDialog::wxGenericProgressDialog()
@@ -158,7 +160,7 @@
     wxWindow* const
         realParent = GetParentForModalDialog(parent, GetWindowStyle());
 
-    if (!wxDialog::Create(realParent, wxID_ANY, title))
+    if (!wxDialog::Create(realParent, wxID_ANY, title, wxDefaultPosition, wxDefaultSize, GetWindowStyle()))
         return false;
 
     SetMaximum(maximum);
