? cygwin2.diff
Index: filefn.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/filefn.h,v
retrieving revision 1.85
diff -u -r1.85 filefn.h
--- filefn.h	2004/03/28 13:09:36	1.85
+++ filefn.h	2004/04/19 00:49:55
@@ -300,15 +300,13 @@
 #define wxPATH_SEP_MAC        wxT(";")
 
 // platform independent versions
-#if defined(__UNIX__) && !defined(__CYGWIN__) && !defined(__OS2__)
+#if defined(__UNIX__) && !defined(__OS2__)
+  // CYGWIN also uses UNIX settings
   #define wxFILE_SEP_PATH     wxFILE_SEP_PATH_UNIX
   #define wxPATH_SEP          wxPATH_SEP_UNIX
 #elif defined(__MAC__)
   #define wxFILE_SEP_PATH     wxFILE_SEP_PATH_MAC
   #define wxPATH_SEP          wxPATH_SEP_MAC
-#elif defined(__CYGWIN__) // Cygwin
-  #define wxFILE_SEP_PATH     wxFILE_SEP_PATH_DOS
-  #define wxPATH_SEP          wxPATH_SEP_UNIX
 #else   // Windows and OS/2
   #define wxFILE_SEP_PATH     wxFILE_SEP_PATH_DOS
   #define wxPATH_SEP          wxPATH_SEP_DOS
Index: platform.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/platform.h,v
retrieving revision 1.44
diff -u -r1.44 platform.h
--- platform.h	2004/04/15 23:02:43	1.44
+++ platform.h	2004/04/19 00:49:58
@@ -81,7 +81,8 @@
 #   endif /* !__WIN64__ */
 #endif /* _WIN64 */
 
-#if defined(_WIN32) || defined(WIN32) || defined(__NT__) || defined(__WXWINCE__)
+#if (defined(_WIN32) || defined(WIN32) || defined(__NT__) || defined(__WXWINCE__)) \
+    && !defined(__WXMOTIF__) && !defined(__WXGTK__) && !defined(__WXX11__)
 #    ifndef __WXMSW__
 #        define __WXMSW__
 #    endif
@@ -386,7 +387,7 @@
 #if ( defined( __GNUWIN32__ ) || defined( __MINGW32__ ) || \
     defined( __CYGWIN__ ) || \
       (defined(__WATCOMC__) && __WATCOMC__ >= 1200) ) && \
-    !defined(__DOS__)
+    !defined(__DOS__) && !defined(__WXMOTIF__) && !defined(__WXGTK__) && !defined(__WXX11__)
 #    include "wx/msw/gccpriv.h"
 #else
 #    undef wxCHECK_W32API_VERSION

 	  	 
