diff -r -u -p a/interface/wx/dialog.h b/interface/wx/dialog.h
--- a/interface/wx/dialog.h	2020-04-28 08:19:02.109429100 +0200
+++ b/interface/wx/dialog.h	2020-04-28 13:11:54.673737800 +0200
@@ -140,8 +140,7 @@ enum wxDialogLayoutAdaptationMode
         The dialog is being closed by the user or programmatically (see wxWindow::Close).
         The user may generate this event clicking the close button
         (typically the 'X' on the top-right of the title bar) if it's present
-        (see the @c wxCLOSE_BOX style) or by clicking a button with the
-        @c wxID_CANCEL or @c wxID_OK ids.
+        (see the @c wxCLOSE_BOX style).
     @event{EVT_INIT_DIALOG(func)}
         Process a @c wxEVT_INIT_DIALOG event. See wxInitDialogEvent.
     @endEventTable
