Index: src/common/datetimefmt.cpp
===================================================================
--- src/common/datetimefmt.cpp	(revision 64016)
+++ src/common/datetimefmt.cpp	(working copy)
@@ -323,8 +323,13 @@
     // strftime() or if we use non standard specificators
 #ifdef wxHAS_STRFTIME
     time_t time = GetTicks();
+    bool isVOptionSupported = true;
+#ifdef __WXMSW__
+    isVOptionSupported = false;
+#endif
 
-    if ( (time != (time_t)-1) && !wxStrstr(format, wxT("%l")) )
+    if ( (time != (time_t)-1) && !wxStrstr(format, wxT("%l")) && 
+            isVOptionSupported )
     {
         // use strftime()
         struct tm tmstruct;
@@ -614,6 +619,7 @@
                     res += wxString::Format(fmt, GetWeekOfYear(Sunday_First, tz));
                     break;
 
+                case wxT('V'):       // ISO week number in the year
                 case wxT('W'):       // week number in the year (Monday 1st week day)
                     res += wxString::Format(fmt, GetWeekOfYear(Monday_First, tz));
                     break;
