diff -u -r1.4 basemsw.cpp
--- src/msw/basemsw.cpp	2004/05/23 20:52:56	1.4
+++ src/msw/basemsw.cpp	2006/06/06 12:40:16
@@ -42,8 +42,13 @@
 void wxConsoleAppTraits::AlwaysYield()
 {
     MSG msg;
+	UINT msgNumber = 0;
     while ( ::PeekMessage(&msg, NULL, 0, 0, PM_REMOVE) )
-        ;
+	{
+		if(msgNumber == WM_PAINT && msg.message == WM_PAINT) // infinite loop
+			break;
+		msgNumber = msg.message;
+	};
 }
 
 void *wxConsoleAppTraits::BeforeChildWaitLoop()


 	  	 
