--- wxWidgets/src/msw/combo.cpp	2007-03-19 15:27:50.000000000 +0200
+++ wxWidgets.patched/src/msw/combo.cpp	2007-04-30 12:18:52.687500000 +0300
@@ -164,12 +164,16 @@
     wxUxThemeEngine* theme = wxUxThemeEngine::GetIfActive();
     if ( theme )
     {
-        wxUxThemeHandle hTheme(this, L"COMBOBOX");
+        wxUxThemeHandle hTheme(this, L"EDIT");
 
         COLORREF col;
-        theme->GetThemeColor(hTheme,EP_EDITTEXT,ETS_NORMAL,TMT_FILLCOLOR,&col);
+        HRESULT hr = theme->GetThemeColor(hTheme,EP_EDITTEXT,ETS_NORMAL,TMT_FILLCOLOR,&col);
+        if( FAILED(hr) )
+            wxLogApiError(_T("GetThemeColor(EDIT, EP_EDITTEXT, ETS_NORMAL, TMT_FILLCOLOR)"), hr);
         SetBackgroundColour(wxRGBToColour(col));
-        theme->GetThemeColor(hTheme,EP_EDITTEXT,ETS_NORMAL,TMT_TEXTCOLOR,&col);
+        hr = theme->GetThemeColor(hTheme,EP_EDITTEXT,ETS_NORMAL,TMT_TEXTCOLOR,&col);
+        if( FAILED(hr) )
+            wxLogApiError(_T("GetThemeColor(EDIT, EP_EDITTEXT, ETS_NORMAL, TMT_TEXTCOLOR)"), hr);
         SetForegroundColour(wxRGBToColour(col));
     }
     else

 	  	 
