diff --git a/samples/minimal/minimal.cpp b/samples/minimal/minimal.cpp
index a78e462..c4fe06e 100644
--- a/samples/minimal/minimal.cpp
+++ b/samples/minimal/minimal.cpp
@@ -160,8 +160,18 @@ MyFrame::MyFrame(const wxString& title)
 
     // now append the freshly created menu to the menu bar...
     wxMenuBar *menuBar = new wxMenuBar();
-    menuBar->Append(fileMenu, "&File");
+    
+    fileMenu->SetTitle("&File");
+    wxString filetitle(fileMenu->GetTitle());
+    wxMessageBox(filetitle + "  " + fileMenu->GetTitle());
+    
+    menuBar->Append(fileMenu, fileMenu->GetTitle());
+    wxMessageBox(filetitle + "  " + fileMenu->GetTitle());
+    
+    helpMenu->SetTitle("&Help");
+    wxMessageBox("Before append: helpMenu->GetTitle() = " + helpMenu->GetTitle());
     menuBar->Append(helpMenu, "&Help");
+    wxMessageBox("After append: helpMenu->GetTitle() = " + helpMenu->GetTitle());
 
     // ... and attach this menu bar to the frame
     SetMenuBar(menuBar);
