Index: notebook.cpp
===================================================================
--- notebook.cpp	(revision 53465)
+++ notebook.cpp	(working copy)
@@ -59,7 +59,8 @@
 
 wxPanel *CreateUserCreatedPage(wxBookCtrlBase *parent)
 {
-    wxPanel *panel = new wxPanel(parent);
+    wxPanel *panel = new wxPanel(parent, wxID_ANY, wxDefaultPosition,
+          wxDefaultSize, wxBORDER_SUNKEN);
 
 #if wxUSE_HELP
     panel->SetHelpText( wxT( "Panel with a Button" ) );

 	  	 
