diff -Naur wxGTK-2.2.9/Makefile.in wxGTK-patched/Makefile.in
--- wxGTK-2.2.9/Makefile.in	Fri Jan  4 16:47:57 2002
+++ wxGTK-patched/Makefile.in	Sat Jan  4 13:28:08 2003
@@ -395,7 +395,7 @@
 		&& $(LN_S) @WX_LIBRARY_NAME_SHARED_GL@ @WX_LIBRARY_LINK2_GL@
 
 afminstall: preinstall
-	$(INSTALL) -d $(datadir)
+	test -d $(datadir) || $(INSTALL) -d $(datadir)
 	$(INSTALL) -d $(datadir)/wx
 	$(INSTALL) -d $(datadir)/wx/afm
 	$(INSTALL) -d $(datadir)/wx/gs_afm
@@ -409,9 +409,9 @@
 	@echo " Installing wxWindows..."
 	@echo " "
 
-	$(INSTALL) -d $(prefix)
-	$(INSTALL) -d $(bindir)
-	$(INSTALL) -d $(libdir)
+	test -d $(prefix) || $(INSTALL) -d $(prefix)
+	test -d $(bindir) || $(INSTALL) -d $(bindir)
+	test -d $(libdir) || $(INSTALL) -d $(libdir)
 
 	$(INSTALL_SCRIPT) $(top_builddir)/wx@TOOLCHAIN_NAME@-config $(bindir)/wx@TOOLCHAIN_NAME@-config
 	cd $(bindir) && rm -f wx-config && $(LN_S) wx@TOOLCHAIN_NAME@-config wx-config
@@ -441,10 +441,10 @@
 	  echo "$(INSTALL_DATA) $(top_srcdir)/include/wx/$$p $(includedir)/wx/$$p"; \
 	done
 
-	$(INSTALL) -d $(localedir)
+	test -d $(localedir) || $(INSTALL) -d $(localedir)
 	@for p in $(WX_LINGUAS); do \
-	  $(INSTALL) -d $(localedir)/$$p; \
-	  $(INSTALL) -d $(localedir)/$$p/LC_MESSAGES; \
+	  test -d $(localedir)/$$p || $(INSTALL) -d $(localedir)/$$p; \
+	  test -d $(localedir)/$$p/LC_MESSAGES || $(INSTALL) -d $(localedir)/$$p/LC_MESSAGES; \
 	  $(INSTALL_DATA) $(top_srcdir)/locale/$$p.mo $(localedir)/$$p/LC_MESSAGES/wxstd.mo; \
 	  echo "$(INSTALL_DATA) $(top_srcdir)/locale/$$p.mo $(localedir)/$$p/LC_MESSAGES/wxstd.mo"; \
 	done
@@ -453,7 +453,7 @@
 	@echo " "
 	@echo " Installing wxWindows OpenGl add-on..."
 	@echo " "
-	$(INSTALL) -d $(libdir)
+	test -d $(libdir) || $(INSTALL) -d $(libdir)
 	$(INSTALL_PROGRAM) $(top_builddir)/lib/@WX_TARGET_LIBRARY_GL@ $(libdir)/@WX_TARGET_LIBRARY_GL@
 
 install: @AFMINSTALL@ @WX_ALL_INSTALLED@

 	  	 
