Index: include/wx/defs.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/defs.h,v
retrieving revision 1.419
diff -b -u -2 -r1.419 defs.h
--- include/wx/defs.h	2004/02/10 22:46:05	1.419
+++ include/wx/defs.h	2004/02/24 18:42:57
@@ -2167,20 +2167,20 @@
 
 /*  Stand-ins for Windows types or OS/2, to avoid #including all of windows.h or os2.h */
-typedef unsigned long   WXHWND;
-typedef unsigned long   WXHANDLE;
-typedef unsigned long   WXHICON;
-typedef unsigned long   WXHFONT;
-typedef unsigned long   WXHMENU;
-typedef unsigned long   WXHPEN;
-typedef unsigned long   WXHBRUSH;
-typedef unsigned long   WXHPALETTE;
-typedef unsigned long   WXHCURSOR;
-typedef unsigned long   WXHRGN;
-typedef unsigned long   WXHACCEL;
+typedef void *          WXHWND;
+typedef void *          WXHANDLE;
+typedef void *          WXHICON;
+typedef void *          WXHFONT;
+typedef void *          WXHMENU;
+typedef void *          WXHPEN;
+typedef void *          WXHBRUSH;
+typedef void *          WXHPALETTE;
+typedef void *          WXHCURSOR;
+typedef void *          WXHRGN;
+typedef void *          WXHACCEL;
 typedef void WXFAR  *   WXHINSTANCE;
-typedef unsigned long   WXHBITMAP;
-typedef unsigned long   WXHIMAGELIST;
-typedef unsigned long   WXHGLOBAL;
-typedef unsigned long   WXHDC;
+typedef void *          WXHBITMAP;
+typedef void *          WXHIMAGELIST;
+typedef void *          WXHGLOBAL;
+typedef void *          WXHDC;
 typedef unsigned int    WXUINT;
 typedef unsigned long   WXDWORD;
@@ -2190,7 +2190,7 @@
 typedef void *          WXRGNDATA;
 typedef void *          WXMSG;
-typedef unsigned long   WXHCONV;
-typedef unsigned long   WXHKEY;
-typedef unsigned long   WXHTREEITEM;
+typedef void *          WXHCONV;
+typedef void *          WXHKEY;
+typedef void *          WXHTREEITEM;
 
 typedef void *          WXDRAWITEMSTRUCT;
@@ -2204,6 +2204,14 @@
 
 #ifdef __WXMSW__
+
+#ifdef __WIN64__
+typedef unsigned __int64    WXWPARAM;
+typedef __int64            WXLPARAM;
+typedef __int64            WXLRESULT;
+#else
 typedef unsigned int    WXWPARAM;
 typedef long            WXLPARAM;
+typedef long            WXLRESULT;
+#endif
 
 #if !defined(__WIN32__) || defined(__GNUWIN32__) || defined(__WXMICROWIN__)
Index: include/wx/generic/scrolwin.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/generic/scrolwin.h,v
retrieving revision 1.33
diff -b -u -2 -r1.33 scrolwin.h
--- include/wx/generic/scrolwin.h	2004/01/15 13:48:21	1.33
+++ include/wx/generic/scrolwin.h	2004/02/24 18:42:58
@@ -80,5 +80,5 @@
     // arrows but let the other navigation characters through
 #ifdef __WXMSW__
-    virtual long MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam);
+    virtual WXLRESULT MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam);
 #endif // __WXMSW__
 
Index: include/wx/msw/button.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/msw/button.h,v
retrieving revision 1.20
diff -b -u -2 -r1.20 button.h
--- include/wx/msw/button.h	2003/08/09 12:37:59	1.20
+++ include/wx/msw/button.h	2004/02/24 18:42:58
@@ -52,5 +52,5 @@
     // implementation from now on
     virtual void Command(wxCommandEvent& event);
-    virtual long MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam);
+    virtual WXLRESULT MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam);
     virtual bool MSWCommand(WXUINT param, WXWORD id);
 
Index: include/wx/msw/choice.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/msw/choice.h,v
retrieving revision 1.23
diff -b -u -2 -r1.23 choice.h
--- include/wx/msw/choice.h	2004/02/10 14:41:01	1.23
+++ include/wx/msw/choice.h	2004/02/24 18:42:58
@@ -84,5 +84,5 @@
     // MSW only
     virtual bool MSWCommand(WXUINT param, WXWORD id);
-    long MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam);
+    WXLRESULT MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam);
     virtual WXHBRUSH OnCtlColor(WXHDC pDC, WXHWND pWnd, WXUINT nCtlColor,
             WXUINT message, WXWPARAM wParam, WXLPARAM lParam);
Index: include/wx/msw/combobox.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/msw/combobox.h,v
retrieving revision 1.22
diff -b -u -2 -r1.22 combobox.h
--- include/wx/msw/combobox.h	2004/02/08 00:59:23	1.22
+++ include/wx/msw/combobox.h	2004/02/24 18:42:58
@@ -96,5 +96,5 @@
     virtual bool MSWCommand(WXUINT param, WXWORD id);
     bool MSWProcessEditMsg(WXUINT msg, WXWPARAM wParam, WXLPARAM lParam);
-    virtual long MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam);
+    virtual WXLRESULT MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam);
 
     virtual WXHBRUSH OnCtlColor(WXHDC pDC, WXHWND pWnd, WXUINT nCtlColor,
Index: include/wx/msw/dialog.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/msw/dialog.h,v
retrieving revision 1.38
diff -b -u -2 -r1.38 dialog.h
--- include/wx/msw/dialog.h	2004/02/05 20:51:26	1.38
+++ include/wx/msw/dialog.h	2004/02/24 18:42:58
@@ -98,5 +98,5 @@
 
     // Windows callbacks
-    long MSWWindowProc(WXUINT message, WXWPARAM wParam, WXLPARAM lParam);
+    WXLRESULT MSWWindowProc(WXUINT message, WXWPARAM wParam, WXLPARAM lParam);
 
 #if wxUSE_CTL3D
Index: include/wx/msw/frame.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/msw/frame.h,v
retrieving revision 1.66
diff -b -u -2 -r1.66 frame.h
--- include/wx/msw/frame.h	2003/08/09 12:37:59	1.66
+++ include/wx/msw/frame.h	2004/02/24 18:42:58
@@ -127,5 +127,5 @@
 
     // window proc for the frames
-    long MSWWindowProc(WXUINT message, WXWPARAM wParam, WXLPARAM lParam);
+    WXLRESULT MSWWindowProc(WXUINT message, WXWPARAM wParam, WXLPARAM lParam);
 
     // handle WM_INITMENUPOPUP message
Index: include/wx/msw/mdi.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/msw/mdi.h,v
retrieving revision 1.24
diff -b -u -2 -r1.24 mdi.h
--- include/wx/msw/mdi.h	2003/08/09 12:38:00	1.24
+++ include/wx/msw/mdi.h	2004/02/24 18:42:58
@@ -91,7 +91,7 @@
 
     // override window proc for MDI-specific message processing
-    virtual long MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam);
+    virtual WXLRESULT MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam);
 
-    virtual long MSWDefWindowProc(WXUINT, WXWPARAM, WXLPARAM);
+    virtual WXLRESULT MSWDefWindowProc(WXUINT, WXWPARAM, WXLPARAM);
     virtual bool MSWTranslateMessage(WXMSG* msg);
 
@@ -165,6 +165,6 @@
     bool HandleGetMinMaxInfo(void *mmInfo);
 
-    virtual long MSWWindowProc(WXUINT message, WXWPARAM wParam, WXLPARAM lParam);
-    virtual long MSWDefWindowProc(WXUINT message, WXWPARAM wParam, WXLPARAM lParam);
+    virtual WXLRESULT MSWWindowProc(WXUINT message, WXWPARAM wParam, WXLPARAM lParam);
+    virtual WXLRESULT MSWDefWindowProc(WXUINT message, WXWPARAM wParam, WXLPARAM lParam);
     virtual bool MSWTranslateMessage(WXMSG *msg);
 
Index: include/wx/msw/notebook.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/msw/notebook.h,v
retrieving revision 1.48
diff -b -u -2 -r1.48 notebook.h
--- include/wx/msw/notebook.h	2003/12/11 22:20:18	1.48
+++ include/wx/msw/notebook.h	2004/02/24 18:42:59
@@ -178,5 +178,5 @@
     // override WndProc.
 #if wxUSE_UXTHEME
-    virtual long MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam);
+    virtual WXLRESULT MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam);
 #endif
 
Index: include/wx/msw/radiobox.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/msw/radiobox.h,v
retrieving revision 1.26
diff -b -u -2 -r1.26 radiobox.h
--- include/wx/msw/radiobox.h	2004/01/31 18:21:10	1.26
+++ include/wx/msw/radiobox.h	2004/02/24 18:42:59
@@ -113,5 +113,5 @@
     virtual bool SetFont(const wxFont& font);
 
-    long MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam);
+    WXLRESULT MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam);
     virtual WXHBRUSH OnCtlColor(WXHDC pDC, WXHWND pWnd, WXUINT nCtlColor,
                                 WXUINT message,
Index: include/wx/msw/statbmp.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/msw/statbmp.h,v
retrieving revision 1.23
diff -b -u -2 -r1.23 statbmp.h
--- include/wx/msw/statbmp.h	2003/08/09 12:38:00	1.23
+++ include/wx/msw/statbmp.h	2004/02/24 18:42:59
@@ -78,5 +78,5 @@
     virtual bool MSWOnDraw(WXDRAWITEMSTRUCT *item);
 #endif // __WIN16__
-    virtual long MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam);
+    virtual WXLRESULT MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam);
 
 protected:
Index: include/wx/msw/statbox.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/msw/statbox.h,v
retrieving revision 1.16
diff -b -u -2 -r1.16 statbox.h
--- include/wx/msw/statbox.h	2003/08/09 12:38:01	1.16
+++ include/wx/msw/statbox.h	2004/02/24 18:42:59
@@ -43,5 +43,5 @@
     // --------------------------
 
-    virtual long MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam);
+    virtual WXLRESULT MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam);
 
     // overriden base class virtuals
Index: include/wx/msw/tbar95.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/msw/tbar95.h,v
retrieving revision 1.32
diff -b -u -2 -r1.32 tbar95.h
--- include/wx/msw/tbar95.h	2003/08/09 12:38:01	1.32
+++ include/wx/msw/tbar95.h	2004/02/24 18:42:59
@@ -102,5 +102,5 @@
 
     // override WndProc mainly to process WM_SIZE
-    virtual long MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam);
+    virtual WXLRESULT MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam);
 
     // return the appropriate size and flags for the toolbar control
Index: include/wx/msw/textctrl.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/msw/textctrl.h,v
retrieving revision 1.52
diff -b -u -2 -r1.52 textctrl.h
--- include/wx/msw/textctrl.h	2004/02/04 14:56:13	1.52
+++ include/wx/msw/textctrl.h	2004/02/24 18:42:59
@@ -203,5 +203,5 @@
 
     // intercept WM_GETDLGCODE
-    virtual long MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam);
+    virtual WXLRESULT MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam);
 
     // call this to increase the size limit (will do nothing if the current
Index: include/wx/msw/treectrl.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/msw/treectrl.h,v
retrieving revision 1.61
diff -b -u -2 -r1.61 treectrl.h
--- include/wx/msw/treectrl.h	2004/02/10 18:58:27	1.61
+++ include/wx/msw/treectrl.h	2004/02/24 18:42:59
@@ -414,5 +414,5 @@
     // --------------
 
-    virtual long MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam);
+    virtual WXLRESULT MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam);
     virtual bool MSWCommand(WXUINT param, WXWORD id);
     virtual bool MSWOnNotify(int idCtrl, WXLPARAM lParam, WXLPARAM *result);
Index: include/wx/msw/window.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/msw/window.h,v
retrieving revision 1.106
diff -b -u -2 -r1.106 window.h
--- include/wx/msw/window.h	2004/01/15 13:48:25	1.106
+++ include/wx/msw/window.h	2004/02/24 18:43:00
@@ -348,8 +348,8 @@
 
     // Window procedure
-    virtual long MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam);
+    virtual WXLRESULT MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam);
 
     // Calls an appropriate default window procedure
-    virtual long MSWDefWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam);
+    virtual WXLRESULT MSWDefWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam);
 
     // message processing helpers
Index: include/wx/msw/wx.rc
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/msw/wx.rc,v
retrieving revision 1.32
diff -b -u -2 -r1.32 wx.rc
--- include/wx/msw/wx.rc	2003/07/20 22:59:09	1.32
+++ include/wx/msw/wx.rc	2004/02/24 18:43:00
@@ -204,6 +204,8 @@
 //
 
+#if !defined(__WIN64__)
 #if !defined(wxUSE_NO_MANIFEST) || (wxUSE_NO_MANIFEST == 0)
 1 24 "wx/msw/wx.manifest"
+#endif
 #endif
 
Index: include/wx/msw/wince/tbarwce.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/msw/wince/tbarwce.h,v
retrieving revision 1.2
diff -b -u -2 -r1.2 tbarwce.h
--- include/wx/msw/wince/tbarwce.h	2003/08/09 12:38:08	1.2
+++ include/wx/msw/wince/tbarwce.h	2004/02/24 18:43:00
@@ -113,5 +113,5 @@
 
     // override WndProc mainly to process WM_SIZE
-    virtual long MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam);
+    virtual WXLRESULT MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam);
 
     // return the appropriate size and flags for the toolbar control
Index: include/wx/univ/window.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/univ/window.h,v
retrieving revision 1.25
diff -b -u -2 -r1.25 window.h
--- include/wx/univ/window.h	2004/02/12 19:45:40	1.25
+++ include/wx/univ/window.h	2004/02/24 18:43:00
@@ -262,5 +262,5 @@
 #ifdef __WXMSW__
     // override MSWWindowProc() to process WM_NCHITTEST
-    long MSWWindowProc(WXUINT message, WXWPARAM wParam, WXLPARAM lParam);
+    WXLRESULT MSWWindowProc(WXUINT message, WXWPARAM wParam, WXLPARAM lParam);
 #endif // __WXMSW__
 
Index: src/generic/scrlwing.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/generic/scrlwing.cpp,v
retrieving revision 1.50
diff -b -u -2 -r1.50 scrlwing.cpp
--- src/generic/scrlwing.cpp	2004/01/15 13:48:47	1.50
+++ src/generic/scrlwing.cpp	2004/02/24 18:43:03
@@ -1258,10 +1258,10 @@
 
 #ifdef __WXMSW__
-long
+WXLRESULT
 wxGenericScrolledWindow::MSWWindowProc(WXUINT nMsg,
                                        WXWPARAM wParam,
                                        WXLPARAM lParam)
 {
-    long rc = wxPanel::MSWWindowProc(nMsg, wParam, lParam);
+    WXLRESULT rc = wxPanel::MSWWindowProc(nMsg, wParam, lParam);
 
 #ifndef __WXWINCE__
Index: src/msw/button.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/button.cpp,v
retrieving revision 1.79
diff -b -u -2 -r1.79 button.cpp
--- src/msw/button.cpp	2004/01/27 17:47:28	1.79
+++ src/msw/button.cpp	2004/02/24 18:43:07
@@ -436,5 +436,5 @@
 }
 
-long wxButton::MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam)
+WXLRESULT wxButton::MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam)
 {
     // when we receive focus, we want to temporary become the default button in
Index: src/msw/choice.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/choice.cpp,v
retrieving revision 1.75
diff -b -u -2 -r1.75 choice.cpp
--- src/msw/choice.cpp	2004/02/10 14:41:06	1.75
+++ src/msw/choice.cpp	2004/02/24 18:43:07
@@ -234,5 +234,5 @@
 int wxChoice::DoAppend(const wxString& item)
 {
-    int n = (int)SendMessage(GetHwnd(), CB_ADDSTRING, 0, (LONG)item.c_str());
+    int n = (int)SendMessage(GetHwnd(), CB_ADDSTRING, 0, (LPARAM)item.c_str());
     if ( n == CB_ERR )
     {
@@ -254,5 +254,5 @@
     wxCHECK_MSG((pos>=0) && (pos<=GetCount()), -1, wxT("invalid index"));
 
-    int n = (int)SendMessage(GetHwnd(), CB_INSERTSTRING, pos, (LONG)item.c_str());
+    int n = (int)SendMessage(GetHwnd(), CB_INSERTSTRING, pos, (LPARAM)item.c_str());
     if ( n == CB_ERR )
     {
@@ -537,5 +537,5 @@
 }
 
-long wxChoice::MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam)
+WXLRESULT wxChoice::MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam)
 {
     if ( nMsg == WM_LBUTTONUP )
Index: src/msw/combobox.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/combobox.cpp,v
retrieving revision 1.80
diff -b -u -2 -r1.80 combobox.cpp
--- src/msw/combobox.cpp	2004/02/12 16:45:28	1.80
+++ src/msw/combobox.cpp	2004/02/24 18:43:07
@@ -239,5 +239,5 @@
 // ----------------------------------------------------------------------------
 
-long wxComboBox::MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam)
+WXLRESULT wxComboBox::MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam)
 {
     // handle WM_CTLCOLOR messages from our EDIT control to be able to set its
@@ -254,5 +254,5 @@
             UnpackCtlColor(wParam, lParam, &nCtlColor, &hdc, &hwnd);
 
-            return OnCtlColor(hdc, hwnd, nCtlColor, nMsg, wParam, lParam);
+            return (WXLRESULT)OnCtlColor(hdc, hwnd, nCtlColor, nMsg, wParam, lParam);
     }
 
Index: src/msw/dialog.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/dialog.cpp,v
retrieving revision 1.101
diff -b -u -2 -r1.101 dialog.cpp
--- src/msw/dialog.cpp	2004/02/08 16:19:57	1.101
+++ src/msw/dialog.cpp	2004/02/24 18:43:07
@@ -441,7 +441,7 @@
 // ---------------------------------------------------------------------------
 
-long wxDialog::MSWWindowProc(WXUINT message, WXWPARAM wParam, WXLPARAM lParam)
+WXLRESULT wxDialog::MSWWindowProc(WXUINT message, WXWPARAM wParam, WXLPARAM lParam)
 {
-    long rc = 0;
+    WXLRESULT rc = 0;
     bool processed = FALSE;
 
Index: src/msw/frame.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/frame.cpp,v
retrieving revision 1.161
diff -b -u -2 -r1.161 frame.cpp
--- src/msw/frame.cpp	2004/01/02 13:49:56	1.161
+++ src/msw/frame.cpp	2004/02/24 18:43:08
@@ -865,7 +865,7 @@
 // ---------------------------------------------------------------------------
 
-long wxFrame::MSWWindowProc(WXUINT message, WXWPARAM wParam, WXLPARAM lParam)
+WXLRESULT wxFrame::MSWWindowProc(WXUINT message, WXWPARAM wParam, WXLPARAM lParam)
 {
-    long rc = 0;
+    WXLRESULT rc = 0;
     bool processed = FALSE;
 
Index: src/msw/mdi.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/mdi.cpp,v
retrieving revision 1.104
diff -b -u -2 -r1.104 mdi.cpp
--- src/msw/mdi.cpp	2004/02/16 23:27:54	1.104
+++ src/msw/mdi.cpp	2004/02/24 18:43:08
@@ -383,9 +383,9 @@
 // ---------------------------------------------------------------------------
 
-long wxMDIParentFrame::MSWWindowProc(WXUINT message,
+WXLRESULT wxMDIParentFrame::MSWWindowProc(WXUINT message,
                                      WXWPARAM wParam,
                                      WXLPARAM lParam)
 {
-    long rc = 0;
+    WXLRESULT rc = 0;
     bool processed = false;
 
@@ -597,5 +597,5 @@
 }
 
-long wxMDIParentFrame::MSWDefWindowProc(WXUINT message,
+WXLRESULT wxMDIParentFrame::MSWDefWindowProc(WXUINT message,
                                         WXWPARAM wParam,
                                         WXLPARAM lParam)
@@ -860,9 +860,9 @@
 // ---------------------------------------------------------------------------
 
-long wxMDIChildFrame::MSWWindowProc(WXUINT message,
+WXLRESULT wxMDIChildFrame::MSWWindowProc(WXUINT message,
                                     WXWPARAM wParam,
                                     WXLPARAM lParam)
 {
-    long rc = 0;
+    WXLRESULT rc = 0;
     bool processed = false;
 
@@ -1079,5 +1079,5 @@
 // ---------------------------------------------------------------------------
 
-long wxMDIChildFrame::MSWDefWindowProc(WXUINT message, WXUINT wParam, WXLPARAM lParam)
+WXLRESULT wxMDIChildFrame::MSWDefWindowProc(WXUINT message, WXWPARAM wParam, WXLPARAM lParam)
 {
     return DefMDIChildProc(GetHwnd(),
Index: src/msw/notebook.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/notebook.cpp,v
retrieving revision 1.107
diff -b -u -2 -r1.107 notebook.cpp
--- src/msw/notebook.cpp	2004/02/17 04:21:41	1.107
+++ src/msw/notebook.cpp	2004/02/24 18:43:09
@@ -998,5 +998,5 @@
 
 #if wxUSE_UXTHEME
-long wxNotebook::MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam)
+WXLRESULT wxNotebook::MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam)
 {
     static bool g_TestedForTheme = false;
Index: src/msw/radiobox.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/radiobox.cpp,v
retrieving revision 1.86
diff -b -u -2 -r1.86 radiobox.cpp
--- src/msw/radiobox.cpp	2004/02/10 13:02:51	1.86
+++ src/msw/radiobox.cpp	2004/02/24 18:43:09
@@ -824,5 +824,5 @@
 // ----------------------------------------------------------------------------
 
-long wxRadioBox::MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam)
+WXLRESULT wxRadioBox::MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam)
 {
     switch ( nMsg )
@@ -840,5 +840,5 @@
                 wxBrush *brush = wxTheBrushList->FindOrCreateBrush(colBack, wxSOLID);
 
-                return (WXHBRUSH)brush->GetResourceHandle();
+                return (WXLRESULT)brush->GetResourceHandle();
             }
 #endif // Win32
Index: src/msw/spinctrl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/spinctrl.cpp,v
retrieving revision 1.38
diff -b -u -2 -r1.38 spinctrl.cpp
--- src/msw/spinctrl.cpp	2004/02/17 14:42:53	1.38
+++ src/msw/spinctrl.cpp	2004/02/24 18:43:09
@@ -159,5 +159,5 @@
             // if the focus comes from the spin control itself, don't set it
             // back to it -- we don't want to go into an infinite loop
-            if ( wParam == spin->GetHWND() )
+            if ( (WXHWND)wParam == spin->GetHWND() )
                 break;
             //else: fall through
Index: src/msw/statbmp.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/statbmp.cpp,v
retrieving revision 1.45
diff -b -u -2 -r1.45 statbmp.cpp
--- src/msw/statbmp.cpp	2003/09/24 00:41:06	1.45
+++ src/msw/statbmp.cpp	2004/02/24 18:43:10
@@ -317,5 +317,5 @@
 
 // We need this or the control can never be moved e.g. in Dialog Editor.
-long wxStaticBitmap::MSWWindowProc(WXUINT nMsg,
+WXLRESULT wxStaticBitmap::MSWWindowProc(WXUINT nMsg,
                                    WXWPARAM wParam,
                                    WXLPARAM lParam)
Index: src/msw/statbox.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/statbox.cpp,v
retrieving revision 1.42
diff -b -u -2 -r1.42 statbox.cpp
--- src/msw/statbox.cpp	2003/09/24 00:41:06	1.42
+++ src/msw/statbox.cpp	2004/02/24 18:43:10
@@ -155,5 +155,5 @@
 }
 
-long wxStaticBox::MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam)
+WXLRESULT wxStaticBox::MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam)
 {
     switch ( nMsg )
Index: src/msw/tbar95.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/tbar95.cpp,v
retrieving revision 1.122
diff -b -u -2 -r1.122 tbar95.cpp
--- src/msw/tbar95.cpp	2004/02/09 21:44:57	1.122
+++ src/msw/tbar95.cpp	2004/02/24 18:43:10
@@ -1369,5 +1369,5 @@
 }
 
-long wxToolBar::MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam)
+WXLRESULT wxToolBar::MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam)
 {
     switch ( nMsg )
Index: src/msw/textctrl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/textctrl.cpp,v
retrieving revision 1.210
diff -b -u -2 -r1.210 textctrl.cpp
--- src/msw/textctrl.cpp	2004/02/10 13:02:51	1.210
+++ src/msw/textctrl.cpp	2004/02/24 18:43:11
@@ -1729,7 +1729,7 @@
 }
 
-long wxTextCtrl::MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam)
+WXLRESULT wxTextCtrl::MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam)
 {
-    long lRc = wxTextCtrlBase::MSWWindowProc(nMsg, wParam, lParam);
+    WXLRESULT lRc = wxTextCtrlBase::MSWWindowProc(nMsg, wParam, lParam);
 
     if ( nMsg == WM_GETDLGCODE )
Index: src/msw/treectrl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/treectrl.cpp,v
retrieving revision 1.168
diff -b -u -2 -r1.168 treectrl.cpp
--- src/msw/treectrl.cpp	2004/02/12 16:33:33	1.168
+++ src/msw/treectrl.cpp	2004/02/24 18:43:13
@@ -2139,8 +2139,8 @@
 // WM_MOUSEMOVE as they're a lot of them) by catching Windows messages directly
 // instead of passing by wxWin events
-long wxTreeCtrl::MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam)
+WXLRESULT wxTreeCtrl::MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam)
 {
     bool processed = false;
-    long rc = 0;
+    WXLRESULT rc = 0;
     bool isMultiple = (GetWindowStyle() & wxTR_MULTIPLE) != 0;
 
Index: src/msw/window.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/window.cpp,v
retrieving revision 1.464
diff -b -u -2 -r1.464 window.cpp
--- src/msw/window.cpp	2004/02/17 10:06:22	1.464
+++ src/msw/window.cpp	2004/02/24 18:43:15
@@ -1764,5 +1764,5 @@
 // ===========================================================================
 
-long wxWindowMSW::MSWDefWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam)
+WXLRESULT wxWindowMSW::MSWDefWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam)
 {
     if ( m_oldWndProc )
@@ -2185,5 +2185,5 @@
 }
 
-long wxWindowMSW::MSWWindowProc(WXUINT message, WXWPARAM wParam, WXLPARAM lParam)
+WXLRESULT wxWindowMSW::MSWWindowProc(WXUINT message, WXWPARAM wParam, WXLPARAM lParam)
 {
     // did we process the message?
@@ -2194,5 +2194,5 @@
     {
         bool        allow;
-        long        result;
+        WXLRESULT   result;
         WXHICON     hIcon;
         WXHBRUSH    hBrush;
@@ -2889,5 +2889,5 @@
                 // to it if this is the case
                 wxWindowMSW *win = NULL;
-                if ( wParam != m_hWnd )
+                if ( (WXHWND)wParam != m_hWnd )
                 {
                     win = FindItemByHWND((WXHWND)wParam);
Index: src/msw/wince/tbarwce.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/wince/tbarwce.cpp,v
retrieving revision 1.8
diff -b -u -2 -r1.8 tbarwce.cpp
--- src/msw/wince/tbarwce.cpp	2003/12/28 01:07:31	1.8
+++ src/msw/wince/tbarwce.cpp	2004/02/24 18:43:16
@@ -1176,5 +1176,5 @@
 }
 
-long wxToolBar::MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam)
+WXLRESULT wxToolBar::MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam)
 {
 #if 0
Index: src/univ/winuniv.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/univ/winuniv.cpp,v
retrieving revision 1.46
diff -b -u -2 -r1.46 winuniv.cpp
--- src/univ/winuniv.cpp	2004/02/12 19:45:35	1.46
+++ src/univ/winuniv.cpp	2004/02/24 18:43:20
@@ -1340,5 +1340,5 @@
 #include "wx/msw/private.h"
 
-long wxWindow::MSWWindowProc(WXUINT message, WXWPARAM wParam, WXLPARAM lParam)
+WXLRESULT wxWindow::MSWWindowProc(WXUINT message, WXWPARAM wParam, WXLPARAM lParam)
 {
     if ( message == WM_NCHITTEST )
Index: wxPython/contrib/iewin/wxactivex.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/wxPython/contrib/iewin/wxactivex.cpp,v
retrieving revision 1.9
diff -b -u -2 -r1.9 wxactivex.cpp
--- wxPython/contrib/iewin/wxactivex.cpp	2004/01/14 01:49:21	1.9
+++ wxPython/contrib/iewin/wxactivex.cpp	2004/02/24 18:43:21
@@ -1083,5 +1083,5 @@
 }
 
-long wxActiveX::MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam)
+WXLRESULT wxActiveX::MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam)
 {
 	if (m_oleObjectHWND == NULL)
Index: wxPython/contrib/iewin/wxactivex.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/wxPython/contrib/iewin/wxactivex.h,v
retrieving revision 1.6
diff -b -u -2 -r1.6 wxactivex.h
--- wxPython/contrib/iewin/wxactivex.h	2004/02/10 22:15:13	1.6
+++ wxPython/contrib/iewin/wxactivex.h	2004/02/24 18:43:22
@@ -353,5 +353,5 @@
     MemberIdList    m_eventsIdx;
 
-    long MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam);
+    WXLRESULT MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam);
 };
 

 	  	 
