diff --git a/interface/wx/aui/auibook.h b/interface/wx/aui/auibook.h
index 747d63e..efd0f46 100644
--- a/interface/wx/aui/auibook.h
+++ b/interface/wx/aui/auibook.h
@@ -648,6 +648,22 @@ public:
     wxEvent *Clone();
 };
 
+wxEventType wxEVT_AUINOTEBOOK_PAGE_CLOSE;
+wxEventType wxEVT_AUINOTEBOOK_PAGE_CHANGED;
+wxEventType wxEVT_AUINOTEBOOK_PAGE_CHANGING;
+wxEventType wxEVT_AUINOTEBOOK_PAGE_CLOSED;
+wxEventType wxEVT_AUINOTEBOOK_BUTTON;
+wxEventType wxEVT_AUINOTEBOOK_BEGIN_DRAG;
+wxEventType wxEVT_AUINOTEBOOK_END_DRAG;
+wxEventType wxEVT_AUINOTEBOOK_DRAG_MOTION;
+wxEventType wxEVT_AUINOTEBOOK_ALLOW_DND;
+wxEventType wxEVT_AUINOTEBOOK_TAB_MIDDLE_DOWN;
+wxEventType wxEVT_AUINOTEBOOK_TAB_MIDDLE_UP;
+wxEventType wxEVT_AUINOTEBOOK_TAB_RIGHT_DOWN;
+wxEventType wxEVT_AUINOTEBOOK_TAB_RIGHT_UP;
+wxEventType wxEVT_AUINOTEBOOK_DRAG_DONE;
+wxEventType wxEVT_AUINOTEBOOK_BG_DCLICK;
+
 /**
     Default art provider for wxAuiNotebook.
 
