--- a/src/common/appbase.cpp
+++ b/src/common/appbase.cpp
@@ -45,6 +45,7 @@
 #include "wx/sysopt.h"
 #include "wx/tokenzr.h"
 #include "wx/thread.h"
+#include "wx/stdpaths.h"
 
 #if wxUSE_EXCEPTIONS
     // Do we have a C++ compiler with enough C++11 support for
@@ -206,6 +207,15 @@ wxString wxAppConsoleBase::GetAppName() const
             // the application name is, by default, the name of its executable file
             wxFileName::SplitPath(argv[0], NULL, &name, NULL);
         }
+        else
+        {
+            // Otherwise try to get 'native' executable name.
+            wxString pathExe = wxStandardPaths::Get().GetExecutablePath();
+            if (!pathExe.IsEmpty())
+            {
+                wxFileName::SplitPath(pathExe, NULL, &name, NULL);
+            }
+        }
     }
     return name;
 }
