--- strconv_old.cpp	2006-12-11 21:32:46.000000000 +0100
+++ strconv_new.cpp	2006-12-18 11:47:19.312172500 +0100
@@ -3519,4 +3519,21 @@
 }
 
+bool wxCSConv::IsOk() const
+{
+    CreateConvIfNeeded();
+	wxFontEncoding enc = m_encoding;
+
+	//Special case, no convReal created for wxFONTENCODING_ISO8859_1
+    if (enc == wxFONTENCODING_ISO8859_1)
+        return true; //It is OK, we manage it directly
+
+    //m_convReal->IsOk() is called at its own creation, so we now it
+    //returned true. If not so, m_convReal is NULL
+    if (m_convReal)
+        return true;
+
+    return false; //Failed in convReal creation
+}
+
 size_t wxCSConv::ToWChar(wchar_t *dst, size_t dstLen,
                          const char *src, size_t srcLen) const

 	  	 
