--- toplvcmn.cpp	2006-10-04 17:44:54.377875700 -0400
+++ toplvcmn_new.cpp	2006-10-04 17:44:41.626489300 -0400
@@ -194,7 +194,17 @@
         return;
 
     // we need the display rect anyhow so store it first
-    int nDisplay = wxDisplay::GetFromWindow(this);
+    int nDisplay;
+    if (!(dir & wxCENTRE_ON_SCREEN) && GetParent()) 
+    {
+        // we have a parent so limit end result to parent's display
+        nDisplay = wxDisplay::GetFromWindow(GetParent());
+    }
+    else 
+    {
+        // no parent or centering on screen so use our display
+        nDisplay = wxDisplay::GetFromWindow(this);
+    }
     wxDisplay dpy(nDisplay == wxNOT_FOUND ? 0 : nDisplay);
     const wxRect rectDisplay(dpy.GetClientArea());
 

 	  	 
