Index: listctrl.cpp
===================================================================
--- listctrl.cpp	(revision 71715)
+++ listctrl.cpp	(working copy)
@@ -4389,6 +4389,10 @@
                                   validator, name ) )
         return false;
 
+#ifdef __WXMAC__
+    style &= ~wxBORDER_MASK; // note that style isn't changed until AFTER calling wxControl::Create
+#endif // #ifdef __WXMAC__
+
 #ifdef __WXGTK__
     style &= ~wxBORDER_MASK;
     style |= wxBORDER_THEME;
