diff --git src/generic/graphicc.cpp src/generic/graphicc.cpp
index 19722f2..7a78b40 100644
--- src/generic/graphicc.cpp
+++ src/generic/graphicc.cpp
@@ -511,6 +511,11 @@ public:
 
     virtual ~wxCairoImageContext()
     {
+        Flush();
+    }
+
+    virtual void Flush() wxOVERRIDE
+    {
         m_image = m_data.ConvertToImage();
     }
 
diff --git src/msw/graphics.cpp src/msw/graphics.cpp
index a722619..badce22 100644
--- src/msw/graphics.cpp
+++ src/msw/graphics.cpp
@@ -452,6 +452,11 @@ public:
 
     virtual ~wxGDIPlusImageContext()
     {
+        Flush();
+    }
+
+    virtual void Flush() wxOVERRIDE
+    {
         m_image = m_bitmap.ConvertToImage();
     }
 
