diff -r -u -p a/samples/dataview/dataview.cpp b/samples/dataview/dataview.cpp
--- a/samples/dataview/dataview.cpp	2020-07-22 20:20:06.000000000 +0200
+++ b/samples/dataview/dataview.cpp	2020-11-19 06:59:54.413851700 +0100
@@ -158,6 +158,10 @@ private:
     void OnBeginDrag( wxDataViewEvent &event );
     void OnDropPossible( wxDataViewEvent &event );
     void OnDrop( wxDataViewEvent &event );
+
+    void OnBeginDrag_ListStore( wxDataViewEvent &event );
+    void OnDropPossible_ListStore( wxDataViewEvent &event );
+    void OnDrop_ListStore( wxDataViewEvent &event );
 #endif // wxUSE_DRAG_AND_DROP
 
     void OnDataViewChar(wxKeyEvent& event);
@@ -419,6 +423,7 @@ enum
 
     ID_MUSIC_CTRL       = 50,
     ID_ATTR_CTRL        = 51,
+    ID_LIST_STORE       = 52,
 
     ID_ADD_MOZART       = 100,
     ID_DELETE_SEL       = 101,
@@ -525,6 +530,10 @@ wxBEGIN_EVENT_TABLE(MyFrame, wxFrame)
     EVT_DATAVIEW_ITEM_BEGIN_DRAG( ID_MUSIC_CTRL, MyFrame::OnBeginDrag )
     EVT_DATAVIEW_ITEM_DROP_POSSIBLE( ID_MUSIC_CTRL, MyFrame::OnDropPossible )
     EVT_DATAVIEW_ITEM_DROP( ID_MUSIC_CTRL, MyFrame::OnDrop )
+
+    EVT_DATAVIEW_ITEM_BEGIN_DRAG( ID_LIST_STORE, MyFrame::OnBeginDrag_ListStore )
+    EVT_DATAVIEW_ITEM_DROP_POSSIBLE( ID_LIST_STORE, MyFrame::OnDropPossible_ListStore )
+    EVT_DATAVIEW_ITEM_DROP( ID_LIST_STORE, MyFrame::OnDrop_ListStore )
 #endif // wxUSE_DRAG_AND_DROP
 
     EVT_DATAVIEW_COLUMN_HEADER_CLICK(ID_ATTR_CTRL, MyFrame::OnAttrHeaderClick)
@@ -733,7 +742,7 @@ MyFrame::MyFrame(wxFrame *frame, const w
 
     m_notebook->AddPage(firstPanel, "MyMusicTreeModel");
     m_notebook->AddPage(secondPanel, "MyListModel");
-    m_notebook->AddPage(thirdPanel, "wxDataViewListCtrl");
+    m_notebook->AddPage(thirdPanel, "wxDataViewListCtrl", true);
     m_notebook->AddPage(fourthPanel, "wxDataViewTreeCtrl");
     m_notebook->AddPage(fifthPanel, "Variable line height");
     m_notebook->AddPage(sixthPanel, "MyIndexListModel");
@@ -891,10 +900,15 @@ void MyFrame::BuildDataViewCtrl(wxPanel*
     case Page_ListStore:
         {
             wxDataViewListCtrl* lc =
-                new wxDataViewListCtrl( parent, wxID_ANY, wxDefaultPosition,
+                new wxDataViewListCtrl( parent, ID_LIST_STORE, wxDefaultPosition,
                                         wxDefaultSize, style );
             m_ctrl[Page_ListStore] = lc;
 
+#if wxUSE_DRAG_AND_DROP && wxUSE_UNICODE
+            m_ctrl[Page_ListStore]->EnableDragSource( wxDataFormat("test") );
+            m_ctrl[Page_ListStore]->EnableDropTarget( wxDataFormat("test") );
+#endif // wxUSE_DRAG_AND_DROP && wxUSE_UNICODE
+
             MyListStoreDerivedModel* page2_model = new MyListStoreDerivedModel();
             lc->AssociateModel(page2_model);
             page2_model->DecRef();
@@ -1311,6 +1325,54 @@ void MyFrame::OnDrop( wxDataViewEvent &e
         wxLogMessage("Text '%s' dropped on background (proposed index = %i)", obj.GetText(), event.GetProposedDropIndex());
 }
 
+void MyFrame::OnBeginDrag_ListStore( wxDataViewEvent &event )
+{
+    if( !event.GetItem().IsOk() )
+    {
+        event.Veto();
+        return;
+    }
+
+    wxDataViewListCtrl* lc = dynamic_cast<wxDataViewListCtrl*>( m_ctrl[Page_ListStore] );
+    wxCHECK2( lc, return );
+    int drag_row = lc->ItemToRow( event.GetItem() );
+
+    wxCustomDataObject *obj = new wxCustomDataObject( wxDataFormat("test") );
+    obj->SetData(sizeof(drag_row), &drag_row);
+    event.SetDataObject(obj);
+    event.SetDragFlags(wxDrag_DefaultMove);
+
+    wxLogMessage("Starting dragging row '%d'", drag_row);
+}
+
+void MyFrame::OnDropPossible_ListStore( wxDataViewEvent &event )
+{
+    if ( !event.GetItem().IsOk() ||
+         event.GetDataFormat().GetId() != "test" )
+    {
+        event.Veto();
+    }
+}
+
+void MyFrame::OnDrop_ListStore( wxDataViewEvent &event )
+{
+    if ( !event.GetItem().IsOk() ||
+         event.GetDataFormat().GetId() != "test" )
+    {
+        event.Veto();
+        return;
+    }
+
+    int drag_row;
+    memcpy(&drag_row, event.GetDataBuffer(), sizeof(drag_row));
+
+    wxDataViewListCtrl* lc = dynamic_cast<wxDataViewListCtrl*>( m_ctrl[Page_ListStore] );
+    wxCHECK2( lc, return );
+    int drop_row = lc->ItemToRow( event.GetItem() );
+
+    wxLogMessage("Dropped row '%d' on row '%d'", drag_row, drop_row );
+}
+
 #endif // wxUSE_DRAG_AND_DROP
 
 void MyFrame::OnAddMozart( wxCommandEvent& WXUNUSED(event) )
