--- dib.cpp 2004/04/04 12:57:33 1.51
+++ dib.cpp 2004/05/06 19:48:35
@@ -176,8 +176,11 @@
         const int h = bmp.GetHeight();
         int d = bmp.GetDepth();
         if ( d <= 0 )
+        {
             d = wxDisplayDepth();
-
+            if (d == 32 && !bmp.HasAlpha())
+                d = 24;
+        }
         if ( !Create(w, h, d) || !CopyFromDDB(hbmp) )
             return false;
     }


 	  	 
