Index: src/xml/xml.cpp
===================================================================
--- src/xml/xml.cpp	(revision 56762)
+++ src/xml/xml.cpp	(working copy)
@@ -828,7 +828,7 @@
             (c == wxS('&') && str.Mid(i+1, 4) != wxS("amp;")) ||
             ((flags & XML_ESCAPE_QUOTES) && c == wxS('"')))
         {
-            if ( !OutputString(stream, str.substr(last, i), convMem, convFile) )
+            if ( !OutputString(stream, str.substr(last, i - last), convMem, convFile) )
                 return false;
 
             const char *escaped;
