--- ipcbase.head.cpp	2005-08-19 14:39:04.014375000 +0100
+++ ipcbase.cpp	2005-08-19 11:22:53.967500000 +0100
@@ -31,8 +31,8 @@
 IMPLEMENT_CLASS(wxConnectionBase, wxObject)
 
-wxConnectionBase::wxConnectionBase(wxChar *buffer, int size)
+wxConnectionBase::wxConnectionBase(wxChar *buffer, int bytes)
     : m_connected(true),
       m_buffer(buffer),
-      m_buffersize(size),
+      m_buffersize(bytes),
       m_deletebufferwhendone(false)
 {
@@ -81,5 +81,8 @@
       if ( m_buffer )
         delete m_buffer;
-      m_buffer = new wxChar[bytes];
+	  // the argument specifies **byte size**, but m_buffer is of type
+	  // wxChar. Under unicode: sizeof(wxChar) > 1, so the buffer size is
+	  // bytes / sizeof(wxChar) rounded upwards.
+      m_buffer = new wxChar[(bytes + sizeof(wxChar) - 1) / sizeof(wxChar)];
       m_buffersize = bytes;
       return m_buffer;

 	  	 
