Index: src/generic/choicbkg.cpp
===================================================================
--- src/generic/choicbkg.cpp	(revision 58190)
+++ src/generic/choicbkg.cpp	(working copy)
@@ -285,6 +285,11 @@
     return win;
 }
 
+void wxChoicebook::DoSetWindowVariant( wxWindowVariant variant )
+{
+  if (this->GetChoiceCtrl() != NULL)
+    this->GetChoiceCtrl()->SetWindowVariant(variant);
+} /* wxChoicebook::DoSetWindowVariant( wxWindowVariant variant ) */
 
 bool wxChoicebook::DeleteAllPages()
 {
Index: include/wx/choicebk.h
===================================================================
--- include/wx/choicebk.h	(revision 58190)
+++ include/wx/choicebk.h	(working copy)
@@ -88,6 +88,9 @@
 protected:
     virtual wxWindow *DoRemovePage(size_t page);
 
+    // implements the window variants
+    virtual void DoSetWindowVariant( wxWindowVariant variant ) ;
+
     // get the size which the choice control should have
     virtual wxSize GetControllerSize() const;
 
