--- wxWidgets/src/msw/window.cpp	Wed Apr 21 07:40:21 2004
+++ wxW/src/msw/window.cpp	Sun Apr 25 01:55:28 2004
@@ -3006,7 +3006,7 @@
 
     bool nonDefault = false;
 
-    if ( pos.x == -1 )
+    if ( pos.x == wxDefaultPosition.x )
     {
         // if x is set to CW_USEDEFAULT, y parameter is ignored anyhow so we
         // can just as well set it to CW_USEDEFAULT as well
@@ -3019,7 +3019,7 @@
         // neither because it is not handled as a special value by Windows then
         // and so we have to choose some default value for it
         x = pos.x;
-        y = pos.y == -1 ? DEFAULT_Y : pos.y;
+        y = pos.y == wxDefaultPosition.y ? DEFAULT_Y : pos.y;
 
         nonDefault = true;
     }
@@ -3063,16 +3063,16 @@
 	// However, on PocketPC devices, we must use the default
 	// size if possible.
 #ifdef _WIN32_WCE
-	if (size.x == -1)
+	if (size.x == wxDefaultSize.x)
 		w = CW_USEDEFAULT;
 	else
 		w = size.x;
-	if (size.y == -1)
+	if (size.y == wxDefaultSize.y)
 		h = CW_USEDEFAULT;
 	else
 		h = size.y;
 #else
-    if ( size.x == -1 || size.y == -1)
+    if ( size.x == wxDefaultSize.x || size.y == wxDefaultSize.y)
     {
         nonDefault = true;
     }

 	  	 
