Index: src/unix/taskbarx11.cpp
===================================================================
--- src/unix/taskbarx11.cpp	(revision 55938)
+++ src/unix/taskbarx11.cpp	(working copy)
@@ -251,6 +251,19 @@
 }
 
 // ----------------------------------------------------------------------------
+// wxTaskBarIconBase class:
+// ----------------------------------------------------------------------------
+bool wxTaskBarIconBase::SystemTrayAvailable()
+{
+    wxBitmap bmp( 1, 1 );
+    wxTaskBarIconArea *area = new wxTaskBarIconArea( NULL, bmp );
+    bool ret = area->IsProtocolSupported();
+    area->Destroy();
+    delete area;
+    return ret;
+}
+
+// ----------------------------------------------------------------------------
 // wxTaskBarIcon class:
 // ----------------------------------------------------------------------------
 
Index: include/wx/taskbar.h
===================================================================
--- include/wx/taskbar.h	(revision 55938)
+++ include/wx/taskbar.h	(working copy)
@@ -28,6 +28,12 @@
 {
 public:
     wxTaskBarIconBase() { }
+    
+#if defined(__WXGTK__) || defined(__WXX11__) || defined(__WXMOTIF__)
+    static bool SystemTrayAvailable();
+#else
+    static bool SystemTrayAvailable() { return true; };
+#endif
 
     // Operations:
     virtual bool SetIcon(const wxIcon& icon,
Index: interface/wx/taskbar.h
===================================================================
--- interface/wx/taskbar.h	(revision 55938)
+++ interface/wx/taskbar.h	(working copy)
@@ -72,5 +72,10 @@
         Sets the icon, and optional tooltip text.
     */
     bool SetIcon(const wxIcon& icon, const wxString& tooltip);
+
+    /**
+        Returns true if a system tray is available.
+    */
+    static bool SystemTrayAvailable();
 };
 
Index: samples/taskbar/tbtest.cpp
===================================================================
--- samples/taskbar/tbtest.cpp	(revision 55938)
+++ samples/taskbar/tbtest.cpp	(working copy)
@@ -60,6 +60,10 @@
     if ( !wxApp::OnInit() )
         return false;
 
+    if ( !wxTaskBarIcon::SystemTrayAvailable() ) {
+        wxMessageBox( wxT( "There appears to be no system tray support in your current environment. This sample may not behave as expected." ), wxT( "Warning" ), wxOK | wxICON_EXCLAMATION );
+    }
+
     // Create the main window
     gs_dialog = new MyDialog(wxT("wxTaskBarIcon Test Dialog"));
 
