Index: src/gtk/clipbrd.cpp
===================================================================
--- src/gtk/clipbrd.cpp	(Revision 64113)
+++ src/gtk/clipbrd.cpp	(Arbeitskopie)
@@ -442,9 +442,11 @@
     m_data = data;
 
     // get formats from wxDataObjects
-    wxDataFormat *array = new wxDataFormat[ m_data->GetFormatCount() ];
-    m_data->GetAllFormats( array );
+    const size_t count = m_data->GetFormatCount(wxDataObject::Set);
+ 	wxDataFormat* array = new wxDataFormat[count];
+ 	m_data->GetAllFormats(array, wxDataObject::Set);
 
+
     // primary selection or clipboard
     GdkAtom clipboard = m_usePrimary ? (GdkAtom)GDK_SELECTION_PRIMARY
                                      : g_clipboardAtom;
@@ -455,7 +457,7 @@
                               g_timestampAtom,
                               0 );
 
-    for (size_t i = 0; i < m_data->GetFormatCount(); i++)
+    for (size_t i = 0; i < count; i++)
     {
         wxLogTrace( TRACE_CLIPBOARD,
                     wxT("wxClipboard now supports atom %s"),
 
