Index: samples/layout/layout.cpp
===================================================================
--- samples/layout/layout.cpp	(revision 62677)
+++ samples/layout/layout.cpp	(working copy)
@@ -314,16 +314,18 @@
     {
         for ( int j = 0; j < 3; j++ )
         {
-            sizer->Add(new wxStaticText
+            wxStaticText *cell = new wxStaticText
                            (
                             parent,
                             wxID_ANY,
-                            wxString::Format(wxT("(%d, %d)"), i + 1, j + 1),
+                            wxString::Format("(%d, %d)", i + 1, j + 1),
                             wxDefaultPosition,
                             wxDefaultSize,
                             wxALIGN_CENTER
-                           ),
-                       0, wxEXPAND | wxALIGN_CENTER_VERTICAL | wxALL, 3);
+                           );
+            if ( i + j & 1 )
+                cell->SetBackgroundColour( *wxLIGHT_GREY );
+            sizer->Add(cell, 0, wxEXPAND | wxALIGN_CENTER_VERTICAL | wxALL, 3);
         }
     }
 }
