Index: wxWidgets-2.8/src/common/imagtiff.cpp
===================================================================
--- wxWidgets-2.8/src/common/imagtiff.cpp	(revisÆo 48595)
+++ wxWidgets-2.8/src/common/imagtiff.cpp	(c¢pia de trabalho)
@@ -268,6 +268,9 @@
     TIFFGetField( tif, TIFFTAG_IMAGELENGTH, &h );
 
     npixels = w * h;
+	uint32 xres,yres;
+	TIFFGetField(tif, TIFFTAG_XRESOLUTION,&xres);
+	TIFFGetField(tif, TIFFTAG_YRESOLUTION,&yres);
 
     raster = (uint32*) _TIFFmalloc( npixels * sizeof(uint32) );
 
@@ -292,6 +295,8 @@
 
         return false;
     }
+    image->SetOption(wxIMAGE_OPTION_RESOLUTIONX,xres);
+    image->SetOption(wxIMAGE_OPTION_RESOLUTIONY,yres);
 
     if (!TIFFReadRGBAImage( tif, w, h, raster, 0 ))
     {
Index: wxWidgets-2.8/src/common/imagjpeg.cpp
===================================================================
--- wxWidgets-2.8/src/common/imagjpeg.cpp	(revisÆo 48595)
+++ wxWidgets-2.8/src/common/imagjpeg.cpp	(c¢pia de trabalho)
@@ -278,6 +278,9 @@
         jpeg_destroy_decompress( &cinfo );
         return false;
     }
+    image->SetOption(wxIMAGE_OPTION_RESOLUTIONX,cinfo.X_density);
+    image->SetOption(wxIMAGE_OPTION_RESOLUTIONY,cinfo.Y_density);
+	image->SetOption(wxIMAGE_OPTION_RESOLUTIONUNIT,cinfo.density_unit);
     image->SetMask( false );
     ptr = image->GetData();
 
Index: wxWidgets-2.8/src/common/imagbmp.cpp
===================================================================
--- wxWidgets-2.8/src/common/imagbmp.cpp	(revisÆo 48595)
+++ wxWidgets-2.8/src/common/imagbmp.cpp	(c¢pia de trabalho)
@@ -181,7 +181,8 @@
     hdr.bpp = wxUINT16_SWAP_ON_BE(bpp);
     hdr.compression = 0; // RGB uncompressed
     hdr.size_of_bmp = wxUINT32_SWAP_ON_BE(row_width * image->GetHeight());
-    hdr.h_res = hdr.v_res = wxUINT32_SWAP_ON_BE(72);  // 72dpi is standard
+    hdr.h_res = wxUINT32_SWAP_ON_BE(image->GetOptionInt(wxIMAGE_OPTION_RESOLUTIONX)*39.37);
+    hdr.v_res = wxUINT32_SWAP_ON_BE(image->GetOptionInt(wxIMAGE_OPTION_RESOLUTIONX)*39.37);
     hdr.num_clrs = wxUINT32_SWAP_ON_BE(palette_size); // # colors in colormap
     hdr.num_signif_clrs = 0;     // all colors are significant
 
@@ -904,6 +905,8 @@
             wxLogError( _("DIB Header: Unknown encoding in file.") );
         return false;
     }
+	int resx = dbuf[2]/39.37;
+	int resy = dbuf[3]/39.37;
 
     stream.Read(dbuf, 4 * 2);
     int ncolors = wxINT32_SWAP_ON_BE( (int)dbuf[0] );
@@ -944,6 +947,8 @@
 
     }
 
+	image->SetOption(wxIMAGE_OPTION_RESOLUTIONX,resx);
+	image->SetOption(wxIMAGE_OPTION_RESOLUTIONY,resy);
     return true;
 }
 

 	  	 
