Index: src/common/docview.cpp
===================================================================
--- src/common/docview.cpp	(revision 60721)
+++ src/common/docview.cpp	(working copy)
@@ -1072,6 +1072,8 @@
 void wxDocManager::OnPreview(wxCommandEvent& WXUNUSED(event))
 {
 #if wxUSE_PRINTING_ARCHITECTURE
+    wxBusyCursor busy;
+
     wxView *view = GetActiveView();
     if (!view)
         return;
