Index: include/wx/slider.h
===================================================================
--- include/wx/slider.h	(revision 70097)
+++ include/wx/slider.h	(working copy)
@@ -105,7 +105,7 @@
     virtual int GetSelStart() const { return GetMax(); }
     virtual void SetSelection(int WXUNUSED(min), int WXUNUSED(max)) { }
 
-#ifdef WXWIN_COMPATIBILITY_2_8
+#if WXWIN_COMPATIBILITY_2_8
     wxDEPRECATED_INLINE( void SetTickFreq(int freq, int), DoSetTickFreq(freq); )
 #endif
 
Index: include/wx/font.h
===================================================================
--- include/wx/font.h	(revision 70097)
+++ include/wx/font.h	(working copy)
@@ -277,7 +277,7 @@
     static void SetDefaultEncoding(wxFontEncoding encoding);
 
     // this doesn't do anything and is kept for compatibility only
-#ifdef WXWIN_COMPATIBILITY_2_8
+#if WXWIN_COMPATIBILITY_2_8
     wxDEPRECATED_INLINE(void SetNoAntiAliasing(bool no = true), wxUnusedVar(no););
     wxDEPRECATED_INLINE(bool GetNoAntiAliasing() const, return false;)
 #endif // WXWIN_COMPATIBILITY_2_8
Index: include/wx/generic/panelg.h
===================================================================
--- include/wx/generic/panelg.h	(revision 70097)
+++ include/wx/generic/panelg.h	(working copy)
@@ -30,7 +30,7 @@
         Create(parent, winid, pos, size, style, name);
     }
 
-#ifdef WXWIN_COMPATIBILITY_2_8
+#if WXWIN_COMPATIBILITY_2_8
     wxDEPRECATED_CONSTRUCTOR(
     wxPanel(wxWindow *parent,
             int x, int y, int width, int height,
Index: include/wx/generic/grid.h
===================================================================
--- include/wx/generic/grid.h	(revision 70097)
+++ include/wx/generic/grid.h	(working copy)
@@ -1637,7 +1637,7 @@
     // unset any existing sorting column
     void UnsetSortingColumn() { SetSortingColumn(wxNOT_FOUND); }
 
-#ifdef WXWIN_COMPATIBILITY_2_8
+#if WXWIN_COMPATIBILITY_2_8
     // ------ For compatibility with previous wxGrid only...
     //
     //  ************************************************
Index: include/wx/containr.h
===================================================================
--- include/wx/containr.h	(revision 70097)
+++ include/wx/containr.h	(working copy)
@@ -258,7 +258,7 @@
 // look at them.
 // ----------------------------------------------------------------------------
 
-#ifdef WXWIN_COMPATIBILITY_2_8
+#if WXWIN_COMPATIBILITY_2_8
 
 // common part of WX_DECLARE_CONTROL_CONTAINER in the native and generic cases,
 // it should be used in the wxWindow-derived class declaration
Index: include/wx/mousestate.h
===================================================================
--- include/wx/mousestate.h	(revision 70097)
+++ include/wx/mousestate.h	(working copy)
@@ -128,7 +128,7 @@
     // version of wxMouseState but their names are confusing as wxMouseEvent
     // has methods with the same names which do something quite different so
     // don't use them any more
-#ifdef WXWIN_COMPATIBILITY_2_8
+#if WXWIN_COMPATIBILITY_2_8
     wxDEPRECATED_INLINE(bool LeftDown() const, return LeftIsDown(); )
     wxDEPRECATED_INLINE(bool MiddleDown() const, return MiddleIsDown(); )
     wxDEPRECATED_INLINE(bool RightDown() const, return RightIsDown(); )
Index: include/wx/univ/panel.h
===================================================================
--- include/wx/univ/panel.h	(revision 70097)
+++ include/wx/univ/panel.h	(working copy)
@@ -3,7 +3,7 @@
 // Purpose:     wxUniversal-specific wxPanel class.
 // Author:      Vadim Zeitlin
 // Created:     2011-03-18
-// RCS-ID:      $Id: wxhead.h,v 1.12 2010-04-22 12:44:51 zeitlin Exp $
+// RCS-ID:      $Id$
 // Copyright:   (c) 2011 Vadim Zeitlin <vadim@wxwidgets.org>
 // Licence:     wxWindows licence
 ///////////////////////////////////////////////////////////////////////////////
@@ -32,7 +32,7 @@
 
     virtual bool IsCanvasWindow() const { return true; }
 
-#ifdef WXWIN_COMPATIBILITY_2_8
+#if WXWIN_COMPATIBILITY_2_8
     wxDEPRECATED_CONSTRUCTOR(
     wxPanel(wxWindow *parent,
             int x, int y, int width, int height,
Index: include/wx/utils.h
===================================================================
--- include/wx/utils.h	(revision 70097)
+++ include/wx/utils.h	(working copy)
@@ -113,7 +113,7 @@
 // String functions (deprecated, use wxString)
 // ----------------------------------------------------------------------------
 
-#ifdef WXWIN_COMPATIBILITY_2_8
+#if WXWIN_COMPATIBILITY_2_8
 // A shorter way of using strcmp
 wxDEPRECATED_INLINE(inline bool wxStringEq(const char *s1, const char *s2),
     return wxCRT_StrcmpA(s1, s2) == 0; )
Index: include/wx/docview.h
===================================================================
--- include/wx/docview.h	(revision 70097)
+++ include/wx/docview.h	(working copy)
@@ -986,7 +986,7 @@
 
 // these flags are not used anywhere by wxWidgets and kept only for an unlikely
 // case of existing user code using them for its own purposes
-#ifdef WXWIN_COMPATIBILITY_2_8
+#if WXWIN_COMPATIBILITY_2_8
 enum
 {
     wxDOC_SDI = 1,
Index: include/wx/event.h
===================================================================
--- include/wx/event.h	(revision 70097)
+++ include/wx/event.h	(working copy)
@@ -3349,7 +3349,7 @@
     // base class implementation passes the event to wxTheApp
     virtual bool TryAfter(wxEvent& event);
 
-#ifdef WXWIN_COMPATIBILITY_2_8
+#if WXWIN_COMPATIBILITY_2_8
     // deprecated method: override TryBefore() instead of this one
     wxDEPRECATED_BUT_USED_INTERNALLY_INLINE(
         virtual bool TryValidator(wxEvent& WXUNUSED(event)), return false; )
Index: include/wx/msw/panel.h
===================================================================
--- include/wx/msw/panel.h	(revision 70097)
+++ include/wx/msw/panel.h	(working copy)
@@ -3,7 +3,7 @@
 // Purpose:     wxMSW-specific wxPanel class.
 // Author:      Vadim Zeitlin
 // Created:     2011-03-18
-// RCS-ID:      $Id: wxhead.h,v 1.12 2010-04-22 12:44:51 zeitlin Exp $
+// RCS-ID:      $Id$
 // Copyright:   (c) 2011 Vadim Zeitlin <vadim@wxwidgets.org>
 // Licence:     wxWindows licence
 ///////////////////////////////////////////////////////////////////////////////
@@ -38,7 +38,7 @@
     virtual bool HasTransparentBackground();
 
 
-#ifdef WXWIN_COMPATIBILITY_2_8
+#if WXWIN_COMPATIBILITY_2_8
     wxDEPRECATED_CONSTRUCTOR(
     wxPanel(wxWindow *parent,
             int x, int y, int width, int height,
Index: src/generic/grid.cpp
===================================================================
--- src/generic/grid.cpp	(revision 70097)
+++ src/generic/grid.cpp	(working copy)
@@ -8151,10 +8151,12 @@
     AutoSize();
 }
 
+#if WXWIN_COMPATIBILITY_2_8
 wxPen& wxGrid::GetDividerPen() const
 {
     return wxNullPen;
 }
+#endif // WXWIN_COMPATIBILITY_2_8
 
 // ----------------------------------------------------------------------------
 // cell value accessor functions
