Index: src/common/fldlgcmn.cpp
===================================================================
--- src/common/fldlgcmn.cpp	(revision 65461)
+++ src/common/fldlgcmn.cpp	(working copy)
@@ -300,7 +300,7 @@
     }
 
     return wxFileSelector(prompt, NULL, default_name, ext, wild,
-                          load ? wxFD_OPEN : wxFD_SAVE, parent);
+                          load ? (wxFD_OPEN | wxFD_FILE_MUST_EXIST) : wxFD_SAVE, parent);
 }
 
 //----------------------------------------------------------------------------
