Index: interface/wx/uri.h
===================================================================
--- interface/wx/uri.h	(revision 71073)
+++ interface/wx/uri.h	(working copy)
@@ -242,7 +242,7 @@
     /**
         Returns @true if the User component of the URI exists.
     */
-    bool HasUser() const;
+    bool HasUserInfo() const;
 
     /**
         Returns @true if a valid [absolute] URI, otherwise this URI is a URI
