Index: e:/dev/wxWindows/src/common/dynload.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/common/dynload.cpp,v
retrieving revision 1.38
diff -u -2 -r1.38 dynload.cpp
--- e:/dev/wxWindows/src/common/dynload.cpp	2004/05/23 20:52:00	1.38
+++ e:/dev/wxWindows/src/common/dynload.cpp	2004/05/29 21:21:45
@@ -121,5 +121,5 @@
                   _T("Library unloaded before all objects were destroyed") );
 
-    if ( --m_linkcount == 0 )
+    if ( m_linkcount == 0 || --m_linkcount == 0 )
     {
         delete this;
@@ -149,4 +149,8 @@
 void wxPluginLibrary::RestoreClasses()
 {
+    // Check if there is a need to restore classes.
+    if (!ms_classes)
+        return;
+
     for(wxClassInfo *info = m_after; info != m_before; info = info->m_next)
     {

 	  	 
