Index: src/common/datavcmn.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/common/datavcmn.cpp,v
retrieving revision 1.17
diff -u -r1.17 datavcmn.cpp
--- src/common/datavcmn.cpp   2006/03/27 14:01:30  1.17
+++ src/common/datavcmn.cpp   2006/04/20 04:48:13
@@ -51,7 +51,7 @@
 {
     bool ret = true;

-    wxNode *node = m_notifiers.GetFirst();
+    wxList::compatibility_iterator node = m_notifiers.GetFirst();
     while (node)
     {
         wxDataViewListModelNotifier* notifier = (wxDataViewListModelNotifier*) node->GetData();
@@ -67,7 +67,7 @@
 {
     bool ret = true;

-    wxNode *node = m_notifiers.GetFirst();
+    wxList::compatibility_iterator node = m_notifiers.GetFirst();
     while (node)
     {
         wxDataViewListModelNotifier* notifier = (wxDataViewListModelNotifier*) node->GetData();
@@ -83,7 +83,7 @@
 {
     bool ret = true;

-    wxNode *node = m_notifiers.GetFirst();
+    wxList::compatibility_iterator node = m_notifiers.GetFirst();
     while (node)
     {
         wxDataViewListModelNotifier* notifier = (wxDataViewListModelNotifier*) node->GetData();
@@ -99,7 +99,7 @@
 {
     bool ret = true;

-    wxNode *node = m_notifiers.GetFirst();
+    wxList::compatibility_iterator node = m_notifiers.GetFirst();
     while (node)
     {
         wxDataViewListModelNotifier* notifier = (wxDataViewListModelNotifier*) node->GetData();
@@ -115,7 +115,7 @@
 {
     bool ret = true;

-    wxNode *node = m_notifiers.GetFirst();
+    wxList::compatibility_iterator node = m_notifiers.GetFirst();
     while (node)
     {
         wxDataViewListModelNotifier* notifier = (wxDataViewListModelNotifier*) node->GetData();
@@ -131,7 +131,7 @@
 {
     bool ret = true;

-    wxNode *node = m_notifiers.GetFirst();
+    wxList::compatibility_iterator node = m_notifiers.GetFirst();
     while (node)
     {
         wxDataViewListModelNotifier* notifier = (wxDataViewListModelNotifier*) node->GetData();
@@ -147,7 +147,7 @@
 {
     bool ret = true;

-    wxNode *node = m_notifiers.GetFirst();
+    wxList::compatibility_iterator node = m_notifiers.GetFirst();
     while (node)
     {
         wxDataViewListModelNotifier* notifier = (wxDataViewListModelNotifier*) node->GetData();
@@ -163,7 +163,7 @@
 {
     bool ret = true;

-    wxNode *node = m_notifiers.GetFirst();
+    wxList::compatibility_iterator node = m_notifiers.GetFirst();
     while (node)
     {
         wxDataViewListModelNotifier* notifier = (wxDataViewListModelNotifier*) node->GetData();
@@ -182,7 +182,7 @@

 void wxDataViewListModel::RemoveViewingColumn( wxDataViewColumn *column )
 {
-    wxNode *node = m_viewingColumns.GetFirst();
+    wxList::compatibility_iterator node = m_viewingColumns.GetFirst();
     while (node)
     {
         wxDataViewViewingColumn* tmp = (wxDataViewViewingColumn*) node->GetData();
@@ -619,7 +619,7 @@

 wxDataViewColumn* wxDataViewCtrlBase::GetColumn( size_t pos )
 {
-    return (wxDataViewColumn*) m_cols[ pos ];
+    return (wxDataViewColumn*) m_cols.Item(pos).GetData();
 }

 #endif

 	  	 
