--- src/msw/menu-1.136.cpp	2006-08-11 12:41:22.453125000 +0200
+++ src/msw/menu.cpp	2006-08-11 12:44:35.968750000 +0200
@@ -369,6 +369,10 @@
         flags |= MF_SEPARATOR;
     }
 
+    if ( !pItem->IsEnabled() ) {
+        flags |= MF_GRAYED;
+    }
+
     // id is the numeric id for normal menu items and HMENU for submenus as
     // required by ::AppendMenu() API
     UINT id;

 	  	 
