diff --git a/include/wx/unix/glx11.h b/include/wx/unix/glx11.h
index 50305778c4..04fc5f7366 100644
--- a/include/wx/unix/glx11.h
+++ b/include/wx/unix/glx11.h
@@ -146,7 +146,7 @@ public:
     // This method is not currently used by the library itself, but remains for
     // backwards compatibility and also because wxGTK has it we could start
     // using it for the same purpose in wxX11 too some day.
-    virtual void *GetXVisualInfo()
+    virtual void *GetXVisualInfo() wxOVERRIDE
     {
         return wxGLCanvasX11::GetDefaultXVisualInfo();
     }
