diff -ur ../wxWidgets/src/msw/joystick.cpp ./src/msw/joystick.cpp
--- ../wxWidgets/src/msw/joystick.cpp	2005-10-22 09:30:09.000000000 -0700
+++ ./src/msw/joystick.cpp	2005-10-22 09:54:08.000000000 -0700
@@ -16,6 +16,9 @@
 #pragma hdrstop
 #endif
 
+#if wxUSE_JOYSTICK
+
+#include "wx/joystick.h"
 #include "wx/string.h"
 #include "wx/window.h"
 #include "wx/msw/private.h"
@@ -31,7 +34,6 @@
 
 #include "wx/window.h"
 #include "wx/msw/registry.h"
-#include "wx/msw/joystick.h"
 
 #include <regstr.h>
 
@@ -619,3 +621,4 @@
     return (res == JOYERR_NOERROR);
 }
 
+#endif // wxUSE_JOYSTICK

 	  	 
