Index: interface/wx/frame.h
===================================================================
--- interface/wx/frame.h	(revision 70628)
+++ interface/wx/frame.h	(working copy)
@@ -118,7 +118,8 @@
 
     @beginEventEmissionTable
     @event{EVT_CLOSE(func)}
-        The frame is being closed by the user or programmatically (see wxWindow::Close).
+        Process a @c wxEVT_CLOSE_WINDOW event when the frame is being
+        closed by the user or programmatically (see wxWindow::Close).
         The user may generate this event clicking the close button
         (typically the 'X' on the top-right of the title bar) if it's present
         (see the @c wxCLOSE_BOX style). See wxCloseEvent.
