Index: src/gtk/frame.cpp
===================================================================
--- src/gtk/frame.cpp	(revision 47560)
+++ src/gtk/frame.cpp	(working copy)
@@ -384,7 +384,7 @@
                 yy = GetClientSize().y;
 #if wxUSE_MENUS_NATIVE
                 yy += m_menuBarHeight;
-#endif // wxUSE_MENU_NATIVE
+#endif // wxUSE_MENUS_NATIVE
                 m_frameToolBar->m_x = xx;
                 m_frameToolBar->m_y = yy;
                 ww = m_width - 2*m_miniEdge;
Index: src/motif/font.cpp
===================================================================
--- src/motif/font.cpp	(revision 47560)
+++ src/motif/font.cpp	(working copy)
@@ -61,7 +61,7 @@
 #endif
 #if !wxMOTIF_USE_RENDER_TABLE
     WXFontList          m_fontList;     // Motif XmFontList
-#else // if wxUSE_RENDER_TABLE
+#else // if wxMOTIF_USE_RENDER_TABLE
     WXRenderTable       m_renderTable;  // Motif XmRenderTable
     WXRendition         m_rendition;    // Motif XmRendition
 #endif
@@ -147,7 +147,7 @@
     if (m_fontList)
         XmFontListFree ((XmFontList) m_fontList);
     m_fontList = NULL;
-#else // if wxUSE_RENDER_TABLE
+#else // if wxMOTIF_USE_RENDER_TABLE
     if (m_renderTable)
         XmRenderTableFree ((XmRenderTable) m_renderTable);
     m_renderTable = NULL;
Index: src/motif/checkbox.cpp
===================================================================
--- src/motif/checkbox.cpp	(revision 47560)
+++ src/motif/checkbox.cpp	(working copy)
@@ -257,4 +257,4 @@
     return true;
 }
 
-#endif // wxUSE_TOGGLEBUTTON
+#endif // wxUSE_TOGGLEBTN
Index: src/common/dpycmn.cpp
===================================================================
--- src/common/dpycmn.cpp	(revision 47560)
+++ src/common/dpycmn.cpp	(working copy)
@@ -205,7 +205,7 @@
     return m_impl->ChangeMode(mode);
 }
 
-#endif // wxUSE_DIRECTDRAW
+#endif // wxUSE_DISPLAY
 
 // ----------------------------------------------------------------------------
 // static functions implementation
Index: src/palmos/spinbutt.cpp
===================================================================
--- src/palmos/spinbutt.cpp	(revision 47560)
+++ src/palmos/spinbutt.cpp	(working copy)
@@ -147,4 +147,4 @@
 }
 
 #endif
-    // wxUSE_SPINCTN
+    // wxUSE_SPINBTN
Index: src/gtk1/app.cpp
===================================================================
--- src/gtk1/app.cpp	(revision 47560)
+++ src/gtk1/app.cpp	(working copy)
@@ -183,14 +183,14 @@
 #if wxUSE_THREADS
     if (!wxThread::IsMain())
         wxMutexGuiEnter();
-#endif // wxUSE_THREADS_
+#endif // wxUSE_THREADS
 
     wxapp_install_idle_handler();
 
 #if wxUSE_THREADS
     if (!wxThread::IsMain())
         wxMutexGuiLeave();
-#endif // wxUSE_THREADS_
+#endif // wxUSE_THREADS
 }
 
 //-----------------------------------------------------------------------------
Index: src/msw/ole/automtn.cpp
===================================================================
--- src/msw/ole/automtn.cpp	(revision 47560)
+++ src/msw/ole/automtn.cpp	(working copy)
@@ -54,7 +54,7 @@
 
 #if wxUSE_DATETIME
 #include "wx/datetime.h"
-#endif // wxUSE_TIMEDATE
+#endif // wxUSE_DATETIME
 
 static void ClearVariant(VARIANTARG *pvarg) ;
 static void ReleaseVariant(VARIANTARG *pvarg) ;
Index: include/wx/anidecod.h
===================================================================
--- include/wx/anidecod.h	(revision 47560)
+++ include/wx/anidecod.h	(working copy)
@@ -72,6 +72,6 @@
 };
 
 
-#endif  // wxUSE_STREAM && wxUSE_ICO_CUR
+#endif  // wxUSE_STREAMS && wxUSE_ICO_CUR
 
 #endif  // _WX_ANIDECOD_H
Index: include/wx/animdecod.h
===================================================================
--- include/wx/animdecod.h	(revision 47560)
+++ include/wx/animdecod.h	(working copy)
@@ -141,6 +141,6 @@
 };
 
 
-#endif  // wxUSE_STREAM
+#endif  // wxUSE_STREAMS
 #endif  // _WX_ANIMDECOD_H
 
Index: include/wx/gifdecod.h
===================================================================
--- include/wx/gifdecod.h	(revision 47560)
+++ include/wx/gifdecod.h	(working copy)
@@ -105,6 +105,6 @@
     DECLARE_NO_COPY_CLASS(wxGIFDecoder)
 };
 
-#endif // wxUSE_STREAM && wxUSE_GIF
+#endif // wxUSE_STREAMS && wxUSE_GIF
 
 #endif // _WX_GIFDECOD_H_

 	  	 
