Index: ownerdrw.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/os2/ownerdrw.cpp,v
retrieving revision 1.29
diff -u -r1.29 ownerdrw.cpp
--- ownerdrw.cpp	2006/06/19 07:12:46	1.29
+++ ownerdrw.cpp	2006/09/09 16:36:44
@@ -204,12 +204,22 @@
     //
     if (eStatus & wxODSelected)
     {
-        wxColour                        vCol2(wxT("WHITE"));
-        vColBack.Set( (unsigned char)0
-                     ,(unsigned char)0
-                     ,(unsigned char)160
-                    ); // no dark blue in color table
-        vColText = vCol2;
+        vRef = (ULONG)::WinQuerySysColor( HWND_DESKTOP
+                                         ,SYSCLR_MENUHILITEBGND
+                                         ,0L
+                                        );
+        vColBack.Set( GetRValue(vRef)
+                     ,GetGValue(vRef)
+                     ,GetBValue(vRef)
+                    );
+        vRef = (ULONG)::WinQuerySysColor( HWND_DESKTOP
+                                         ,SYSCLR_MENUHILITE
+                                         ,0L
+                                        );
+        vColText.Set( GetRValue(vRef)
+                     ,GetGValue(vRef)
+                     ,GetBValue(vRef)
+                    );
     }
     else if (eStatus & wxODDisabled)
     {
Index: defs.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/defs.h,v
retrieving revision 1.554
diff -u -r1.554 defs.h
--- defs.h	2006/09/09 13:36:36	1.554
+++ defs.h	2006/09/09 16:38:23
@@ -2732,9 +2794,9 @@
 #define OS2RGB(r,g,b) ((DWORD)((unsigned char)(b) | ((unsigned char)(g) << 8)) | ((unsigned char)(r) << 16))

 typedef unsigned long COLORREF;
-#define GetBValue(rgb) ((BYTE)((rgb) >> 16))
+#define GetRValue(rgb) ((BYTE)((rgb) >> 16))
 #define GetGValue(rgb) ((BYTE)(((WORD)(rgb)) >> 8))
-#define GetRValue(rgb) ((BYTE)(rgb))
+#define GetBValue(rgb) ((BYTE)(rgb))
 #define PALETTEINDEX(i) ((COLORREF)(0x01000000 | (DWORD)(WORD)(i)))
 #define PALETTERGB(r,g,b) (0x02000000 | OS2RGB(r,g,b))
 /*  OS2's RGB/RGB2 is backwards from this */

 	  	 
