diff -ru wxWidgets.orig/src/msw/utils.cpp wxWidgets/src/msw/utils.cpp
--- wxWidgets.orig/src/msw/utils.cpp	2007-08-28 09:24:26.000000000 +0200
+++ wxWidgets/src/msw/utils.cpp	2007-08-28 13:12:46.000000000 +0200
@@ -1187,30 +1187,35 @@
 
 wxOperatingSystemId wxGetOsVersion(int *verMaj, int *verMin)
 {
-    OSVERSIONINFO info;
-    wxZeroMemory(info);
-
-    info.dwOSVersionInfoSize = sizeof(OSVERSIONINFO);
-    if ( ::GetVersionEx(&info) )
+    static bool initialized = false;
+    static bool ok = false;
+    static OSVERSIONINFO info;
+    if (!initialized)
     {
-        if (verMaj) *verMaj = info.dwMajorVersion;
-        if (verMin) *verMin = info.dwMinorVersion;
+        wxZeroMemory(info);
+        info.dwOSVersionInfoSize = sizeof(info);
+        ok = GetVersionEx(&info) ? true : false;
+        initialized = true;
     }
-
-#if defined( __WXWINCE__ )
-    return wxOS_WINDOWS_CE;
-#elif defined( __WXMICROWIN__ )
-    return wxOS_WINDOWS_MICRO;
-#else
-    switch ( info.dwPlatformId )
+    if (ok)
     {
-    case VER_PLATFORM_WIN32_NT:
-        return wxOS_WINDOWS_NT;
+       if (verMaj) *verMaj = info.dwMajorVersion;
+       if (verMin) *verMin = info.dwMinorVersion;
 
-    case VER_PLATFORM_WIN32_WINDOWS:
-        return wxOS_WINDOWS_9X;
-    }
+    #if defined( __WXWINCE__ )
+        return wxOS_WINDOWS_CE;
+    #elif defined( __WXMICROWIN__ )
+        return wxOS_WINDOWS_MICRO;
+    #else
+        switch ( info.dwPlatformId )
+        {
+        case VER_PLATFORM_WIN32_NT:
+            return wxOS_WINDOWS_NT;
 
+        case VER_PLATFORM_WIN32_WINDOWS:
+            return wxOS_WINDOWS_9X;
+        }
+    }
     return wxOS_UNKNOWN;
 #endif
 }

 	  	 
