Index: docs/doxygen/mainpages/const_cpp.h
===================================================================
--- docs/doxygen/mainpages/const_cpp.h	(revision 74710)
+++ docs/doxygen/mainpages/const_cpp.h	(working copy)
@@ -247,6 +247,12 @@
 Notice that the base library is always included and the core is always included
 for the GUI applications (i.e. those which don't define @c wxUSE_GUI as 0).
 
+If the makefiles have been used to build the libraries from source and the @c CFG
+variable has been set to specify a different output path for that particular
+configuration of build then the @c wxCFG preprocessor symbol should be set in
+the project that uses wxWidgets to the same value as the @c CFG variable in
+order for the correct @c wx/setup.h file to automatically be included for that
+configuration.
 
 @section page_cppconst_miscellaneous Miscellaneous
 
Index: include/msvc/wx/setup.h
===================================================================
--- include/msvc/wx/setup.h	(revision 74710)
+++ include/msvc/wx/setup.h	(working copy)
@@ -79,10 +79,19 @@
     #define wxARCH_SUFFIX
 #endif
 
+// Ensure the library configuration is defined
+#ifndef wxCFG
+#define wxCFG
+#endif
+
+// Construct the path for the subdirectory under /lib/ that the included setup.h
+// will be used from
 #ifdef WXUSINGDLL
-    #define wxLIB_SUBDIR wxCONCAT3(wxCOMPILER_PREFIX, wxARCH_SUFFIX, _dll)
+#define wxLIB_SUBDIR \
+    wxCONCAT4(wxCOMPILER_PREFIX, wxARCH_SUFFIX, _dll, wxCFG)
 #else // !DLL
-    #define wxLIB_SUBDIR wxCONCAT3(wxCOMPILER_PREFIX, wxARCH_SUFFIX, _lib)
+#define wxLIB_SUBDIR \
+    wxCONCAT4(wxCOMPILER_PREFIX, wxARCH_SUFFIX, _lib, wxCFG)
 #endif // DLL/!DLL
 
 // The user can predefine a different prefix if not using the default MSW port
