Index: control.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/msw/control.cpp,v
retrieving revision 1.65
diff -u -r1.65 control.cpp
--- control.cpp	2004/06/29 02:01:11	1.65
+++ control.cpp	2004/07/01 07:35:41
@@ -168,7 +168,8 @@
 
     // set up fonts and colours
     InheritAttributes();
-    SetFont(GetDefaultAttributes().font);
+    if (!m_hasFont)
+        SetFont(GetDefaultAttributes().font);
 
     // set the size now if no initial size specified
     SetInitialBestSize(size);
Index: glcanvas.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/msw/glcanvas.cpp,v
retrieving revision 1.26
diff -u -r1.26 glcanvas.cpp
--- glcanvas.cpp	2004/05/23 20:52:59	1.26
+++ glcanvas.cpp	2004/07/01 07:35:41
@@ -168,7 +168,6 @@
   if ( ret )
   {
     SetBackgroundColour(wxSystemSettings::GetColour(wxSYS_COLOUR_3DFACE));
-    SetFont(wxSystemSettings::GetFont(wxSYS_DEFAULT_GUI_FONT));
   }
 
   m_hDC = (WXHDC) ::GetDC((HWND) GetHWND());
@@ -192,7 +191,6 @@
   if ( ret )
   {
     SetBackgroundColour(wxSystemSettings::GetColour(wxSYS_COLOUR_3DFACE));
-    SetFont(wxSystemSettings::GetFont(wxSYS_DEFAULT_GUI_FONT));
   }
 
   m_hDC = (WXHDC) ::GetDC((HWND) GetHWND());
@@ -217,7 +215,6 @@
   if ( ret )
   {
     SetBackgroundColour(wxSystemSettings::GetColour(wxSYS_COLOUR_3DFACE));
-    SetFont(wxSystemSettings::GetFont(wxSYS_DEFAULT_GUI_FONT));
   }
 
   m_hDC = (WXHDC) ::GetDC((HWND) GetHWND());
Index: listbox.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/msw/listbox.cpp,v
retrieving revision 1.107
diff -u -r1.107 listbox.cpp
--- listbox.cpp	2004/06/24 20:09:45	1.107
+++ listbox.cpp	2004/07/01 07:35:41
@@ -773,7 +773,7 @@
 
     wxDC dc;
     dc.SetHDC((WXHDC)hdc);
-    dc.SetFont(wxSystemSettings::GetFont(wxSYS_ANSI_VAR_FONT));
+    dc.SetFont(GetFont());
 
     pStruct->itemHeight = dc.GetCharHeight() + 2*OWNER_DRAWN_LISTBOX_EXTRA_SPACE;
     pStruct->itemWidth  = dc.GetCharWidth();
Index: spinctrl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/msw/spinctrl.cpp,v
retrieving revision 1.48
diff -u -r1.48 spinctrl.cpp
--- spinctrl.cpp	2004/06/29 18:02:13	1.48
+++ spinctrl.cpp	2004/07/01 07:35:41
@@ -372,7 +372,8 @@
 
     // set up fonts and colours  (This is nomally done in MSWCreateControl)
     InheritAttributes();
-    SetFont(GetDefaultAttributes().font);
+    if (!m_hasFont)
+        SetFont(GetDefaultAttributes().font);
 
     // set the size of the text window - can do it only now, because we
     // couldn't call DoGetBestSize() before as font wasn't set
Index: statbr95.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/msw/statbr95.cpp,v
retrieving revision 1.53
diff -u -r1.53 statbr95.cpp
--- statbr95.cpp	2004/05/23 20:53:01	1.53
+++ statbr95.cpp	2004/07/01 07:51:23
@@ -116,6 +116,7 @@
 
     SetFieldsCount(1);
     SubclassWin(m_hWnd);
+    InheritAttributes();
 
     SetBackgroundColour(wxSystemSettings::GetColour(wxSYS_COLOUR_MENUBAR));
 
Index: window.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/msw/window.cpp,v
retrieving revision 1.496
diff -u -r1.496 window.cpp
--- window.cpp	2004/06/30 15:47:01	1.496
+++ window.cpp	2004/07/01 07:52:25
@@ -528,7 +528,10 @@
         msflags |= WS_VISIBLE;
     }
 
-    return MSWCreate(wxCanvasClassName, NULL, pos, size, msflags, exstyle);
+    bool rc = MSWCreate(wxCanvasClassName, NULL, pos, size, msflags, exstyle);
+    if (rc)
+        InheritAttributes();
+    return rc;
 }
 
 // ---------------------------------------------------------------------------


 	  	 
