diff --git a/src/richtext/richtextbuffer.cpp b/src/richtext/richtextbuffer.cpp
index 54203e5..d1850dd 100644
--- a/src/richtext/richtextbuffer.cpp
+++ b/src/richtext/richtextbuffer.cpp
@@ -1797,13 +1797,14 @@ void wxRichTextCompositeObject::Invalidate(const wxRichTextRange& invalidRange)
         else if (child->IsTopLevel())
         {
             // Not sure why we did this, but it stops updated layout happening for floating objects.
-#if 0
-            if (wxRichTextBuffer::GetFloatingLayoutMode() && child->IsFloating() && GetBuffer()->GetFloatCollector() && GetBuffer()->GetFloatCollector()->HasFloat(child))
+            // But we must do it for floating tables, otherwise the cell layout is lost
+            if (wxRichTextBuffer::GetFloatingLayoutMode() && child->IsFloating() && wxDynamicCast(child, wxRichTextTable)
+                                                          && GetBuffer()->GetFloatCollector() && GetBuffer()->GetFloatCollector()->HasFloat(child))
             {
                 // Don't invalidate subhierarchy if we've already been laid out
             }
             else
-#endif
+
             {
                 if (invalidRange == wxRICHTEXT_NONE)
                     child->Invalidate(wxRICHTEXT_NONE);
