Index: src/gtk/msgdlg.cpp
===================================================================
--- src/gtk/msgdlg.cpp	(revision 65804)
+++ src/gtk/msgdlg.cpp	(working copy)
@@ -217,14 +217,14 @@
         {
             gtk_dialog_add_button(dlg, wxGTK_CONV(GetNoLabel()),
                                   GTK_RESPONSE_NO);
-            gtk_dialog_add_button(dlg, wxGTK_CONV(GetYesLabel()),
-                                  GTK_RESPONSE_YES);
 
             if ( m_dialogStyle & wxCANCEL )
             {
                 gtk_dialog_add_button(dlg, wxGTK_CONV(GetCancelLabel()),
                                       GTK_RESPONSE_CANCEL);
             }
+            gtk_dialog_add_button(dlg, wxGTK_CONV(GetYesLabel()),
+                                  GTK_RESPONSE_YES);
         }
 
         // it'd probably be harmless to call gtk_dialog_set_default_response()
