--- src/msw/stackwalk-prv.cpp	2008-09-04 16:02:12.000000000 +0200
+++ src/msw/stackwalk.cpp	2008-09-04 18:40:24.000000000 +0200
@@ -316,6 +316,11 @@
 
 void wxStackWalker::WalkFromException(size_t maxDepth)
 {
+    // This should probably be att file scope, since the stackwalker 
+    // is not useful when not using wxUSE_ON_FATAL_EXCEPTION.
+    // Putting it here eliminates the linker error at least.
+#if wxUSE_ON_FATAL_EXCEPTION==1
+
     extern EXCEPTION_POINTERS *wxGlobalSEInformation;
 
     wxCHECK_RET( wxGlobalSEInformation,
@@ -323,6 +328,9 @@
 
     // don't skip any frames, the first one is where we crashed
     WalkFrom(wxGlobalSEInformation, 0, maxDepth);
+#else  // wxUSE_ON_FATAL_EXCEPTION
+	maxDepth++;  // For compiler
+#endif 
 }
 
 void wxStackWalker::Walk(size_t skip, size_t WXUNUSED(maxDepth))
