--- orig/src/mac/carbon/tooltip.cpp	2008-08-21 15:18:55.000000000 +0200
+++ new/src/mac/carbon/tooltip.cpp	2008-08-21 15:20:32.000000000 +0200
@@ -181,8 +181,7 @@
     s_ToolTipWindowRef = m_window ;
     m_backpict = NULL ;
 #if wxUSE_TIMER 
-    if ( m_timer )
-        delete m_timer ;
+    delete m_timer ;
 
     m_timer = new wxMacToolTipTimer( this , s_ToolTipDelay ) ;
 #endif // wxUSE_TIMER
