Index: include/wx/longlong.h
===================================================================
--- include/wx/longlong.h	(revision 54660)
+++ include/wx/longlong.h	(working copy)
@@ -164,7 +164,7 @@
         { return wx_truncate_cast(long, m_ll >> 32); }
         // get low part
     unsigned long GetLo() const
-        { return wx_truncate_cast(unsigned long, m_ll); }
+        { return wx_truncate_cast(wxUint32, m_ll); }
 
         // get absolute value
     wxLongLongNative Abs() const { return wxLongLongNative(*this).Abs(); }
