Index: wxWidgets-2.8/src/common/variant.cpp
===================================================================
--- wxWidgets-2.8/src/common/variant.cpp	(revision 55238)
+++ wxWidgets-2.8/src/common/variant.cpp	(working copy)
@@ -1894,6 +1894,36 @@
     return * (wxVariant*) (data->GetValue().Item(idx)->GetData());
 }
 
+wxArrayString wxGetElementNamesFromVariant( const wxVariant& variant )
+{
+    wxASSERT_MSG( (variant.GetType() == wxT("list") || variant.GetType() == wxT("stringlist")), wxT("Invalid type for wxGetElementNamesFromVariant()") );
+
+    wxArrayString arr;
+    size_t count = variant.GetCount();
+    for ( size_t n = 0; n < count; n++ )
+    {
+        arr.Add(variant[n].GetName());
+    }
+
+    return arr;
+}
+
+wxVariant wxVariant::operator[] (const wxString& name) const
+{
+    wxArrayString arrElem = wxGetElementNamesFromVariant(*this);
+    size_t idx = arrElem.Index(name, FALSE);
+    wxASSERT_MSG( (idx != wxNOT_FOUND), wxT("Invalid index for array") );
+    return operator[](idx);
+}
+
+wxVariant& wxVariant::operator[] (const wxString& name)
+{
+    wxArrayString arrElem = wxGetElementNamesFromVariant(*this);
+    int idx = arrElem.Index(name, FALSE);
+    wxASSERT_MSG( (idx != wxNOT_FOUND), wxT("Invalid index for array") );
+    return operator[](idx);
+}
+
 // Return the number of elements in a list
 size_t wxVariant::GetCount() const
 {
Index: wxWidgets-2.8/include/wx/variant.h
===================================================================
--- wxWidgets-2.8/include/wx/variant.h	(revision 55238)
+++ wxWidgets-2.8/include/wx/variant.h	(working copy)
@@ -274,6 +274,8 @@
     // Treat a list variant as an array
     wxVariant operator[] (size_t idx) const;
     wxVariant& operator[] (size_t idx) ;
+    wxVariant operator[] (const wxString& name) const;
+    wxVariant& operator[] (const wxString& name);
     wxList& GetList() const ;
 
     // Return the number of elements in a list
@@ -418,6 +420,9 @@
 
 extern wxVariant WXDLLIMPEXP_BASE wxNullVariant;
 
+// returns the names of the elements in a list variant
+wxArrayString wxGetElementNamesFromVariant(const wxVariant& variant);
+
 #endif // wxUSE_VARIANT
 
 #endif // _WX_VARIANT_H_
