diff --git a/samples/aui/auidemo.cpp b/samples/aui/auidemo.cpp
index 348693cf14..816a2c5d4e 100644
--- a/samples/aui/auidemo.cpp
+++ b/samples/aui/auidemo.cpp
@@ -846,6 +846,7 @@ MyFrame::MyFrame(wxWindow* parent,
     wxBitmap tb4_bmp1 = wxArtProvider::GetBitmap(wxART_NORMAL_FILE, wxART_OTHER, FromDIP(wxSize(16,16)));
     tb4->AddTool(ID_DropDownToolbarItem, "Item 1", tb4_bmp1);
     tb4->AddTool(ID_SampleItem+23, "Item 2", tb4_bmp1);
+    tb4->SetToolSticky(ID_SampleItem+23, true);
     tb4->AddTool(ID_SampleItem+24, "Item 3", tb4_bmp1);
     tb4->AddTool(ID_SampleItem+25, "Item 4", tb4_bmp1);
     tb4->AddSeparator();
diff --git a/src/aui/barartmsw.cpp b/src/aui/barartmsw.cpp
index ee1b4b10e4..41ba1019b2 100644
--- a/src/aui/barartmsw.cpp
+++ b/src/aui/barartmsw.cpp
@@ -134,7 +134,7 @@ void wxAuiMSWToolBarArt::DrawButton(
             btnState = TS_HOTCHECKED;
         else if ( item.GetState() & wxAUI_BUTTON_STATE_CHECKED )
             btnState = TS_CHECKED;
-        else if ( item.GetState() & wxAUI_BUTTON_STATE_HOVER )
+        else if ( ( item.GetState() & wxAUI_BUTTON_STATE_HOVER ) || item.IsSticky() )
             btnState = TS_HOT;
         else
             btnState = TS_NORMAL;
@@ -259,7 +259,7 @@ void wxAuiMSWToolBarArt::DrawDropDownButton(
             btnState = TS_DISABLED;
         else if ( item.GetState() & wxAUI_BUTTON_STATE_PRESSED )
             btnState = TS_PRESSED;
-        else if ( item.GetState() & wxAUI_BUTTON_STATE_HOVER )
+        else if ( ( item.GetState() & wxAUI_BUTTON_STATE_HOVER ) || item.IsSticky() )
             btnState = TS_HOT;
         else
             btnState = TS_NORMAL;
