Index: src/generic/scrlwing.cpp
===================================================================
--- src/generic/scrlwing.cpp	(revision 78485)
+++ src/generic/scrlwing.cpp	(working copy)
@@ -39,8 +39,6 @@
 #include "wx/scrolbar.h"
 #endif
 
-#include "wx/recguard.h"
-
 #ifdef __WXMSW__
     #include <windows.h> // for DLGC_WANTARROWS
     #include "wx/msw/winundef.h"
@@ -58,6 +56,54 @@
 #endif
 #endif
 
+typedef wxVector<void *> wxInstanceRecGuardFlags;
+class wxInstanceRecGuard
+{
+public:
+    wxInstanceRecGuard(void *instance, wxInstanceRecGuardFlags& flags)
+        : m_flags(flags), m_instance(instance), m_inside(false)
+    {
+        // Determine if our instance is already inside a guard
+        for (wxInstanceRecGuardFlags::iterator iter = flags.begin();
+            iter != flags.end(); ++iter)
+        {
+            if (*iter == instance)
+            {
+                m_inside = true;
+                return;
+            }
+        }
+
+        // Not inside so add it to the back
+        flags.push_back(instance);
+    }
+
+    ~wxInstanceRecGuard()
+    {
+        if (IsInside())
+            return;
+
+        for (wxInstanceRecGuardFlags::iterator iter = m_flags.begin();
+            iter != m_flags.end(); ++iter)
+        {
+            if (*iter == m_instance)
+            {
+                m_flags.erase(iter);
+                break;
+            }
+        }
+    }
+
+    bool IsInside() const { return m_inside; }
+
+private:
+    wxInstanceRecGuardFlags& m_flags;
+    void* m_instance;
+
+    // true if the flag had been already set when we were created
+    bool m_inside;
+};
+
 /*
     TODO PROPERTIES
         style wxHSCROLL | wxVSCROLL
@@ -1303,8 +1349,8 @@
 
 void wxScrollHelper::AdjustScrollbars()
 {
-    static wxRecursionGuardFlag s_flagReentrancy;
-    wxRecursionGuard guard(s_flagReentrancy);
+    static wxInstanceRecGuardFlags s_flagReentrancy;
+    wxInstanceRecGuard guard(this, s_flagReentrancy);
     if ( guard.IsInside() )
     {
         // don't reenter AdjustScrollbars() while another call to
