diff --git a/src/osx/core/printmac.cpp b/src/osx/core/printmac.cpp
index abdea84..08b1cb3 100644
--- a/src/osx/core/printmac.cpp
+++ b/src/osx/core/printmac.cpp
@@ -198,8 +198,9 @@ void wxOSXPrintData::TransferPaperInfoFrom( const wxPrintData &data )
                     wxString id, name(wxT("Custom paper"));
                     id.Printf(wxT("wxPaperCustom%dx%d"), papersize.x, papersize.y);
 
-                    PMPaperCreateCustom(printer, wxCFStringRef( id, wxFont::GetDefaultEncoding() ), wxCFStringRef( name, wxFont::GetDefaultEncoding() ),
-                                            papersize.x, papersize.y, &margins, &paper);
+                    if( PMPaperCreateCustom(printer, wxCFStringRef( id, wxFont::GetDefaultEncoding() ), wxCFStringRef( name, wxFont::GetDefaultEncoding() ),
+                                            papersize.x, papersize.y, &margins, &paper) == noErr)
+                        bestPaper = paper;
                 }
                 if ( bestPaper != kPMNoData )
                 {
