Index: include/wx/window.h
===================================================================
--- include/wx/window.h	(revision 72304)
+++ include/wx/window.h	(working copy)
@@ -1764,8 +1764,9 @@
 
 
     // Must be called when mouse capture is lost to send
-    // wxMouseCaptureLostEvent to windows on capture stack.
-    static void NotifyCaptureLost();
+    // wxMouseCaptureLostEvent to windows on capture stack. The argument is
+    // the window gaining capture or NULL if outside of wx code or none.
+    static void NotifyCaptureLost(wxWindow *gainedCapture);
 
 private:
     // recursively call our own and our children OnEnabled() when the
Index: src/common/wincmn.cpp
===================================================================
--- src/common/wincmn.cpp	(revision 72304)
+++ src/common/wincmn.cpp	(working copy)
@@ -3199,13 +3199,17 @@
 }
 
 /* static */
-void wxWindowBase::NotifyCaptureLost()
+void wxWindowBase::NotifyCaptureLost(wxWindow *gainedCapture)
 {
     // don't do anything if capture lost was expected, i.e. resulted from
     // a wx call to ReleaseMouse or CaptureMouse:
     if ( ms_winCaptureChanging )
         return;
 
+    // don't do anything if capture was transferred from a window to itself
+    if ( gainedCapture == ms_winCaptureCurrent )
+        return;
+
     // if the capture was lost unexpectedly, notify every window that has
     // capture (on stack or current) about it and clear the stack:
 
Index: src/msw/window.cpp
===================================================================
--- src/msw/window.cpp	(revision 72304)
+++ src/msw/window.cpp	(working copy)
@@ -4552,11 +4552,12 @@
 
 bool wxWindowMSW::HandleCaptureChanged(WXHWND hWndGainedCapture)
 {
+    wxWindow *win = wxFindWinFromHandle(hWndGainedCapture);
+
     // notify windows on the capture stack about lost capture
     // (see http://sourceforge.net/tracker/index.php?func=detail&aid=1153662&group_id=9863&atid=109863):
-    wxWindowBase::NotifyCaptureLost();
+    wxWindowBase::NotifyCaptureLost(win);
 
-    wxWindow *win = wxFindWinFromHandle(hWndGainedCapture);
     wxMouseCaptureChangedEvent event(GetId(), win);
     event.SetEventObject(this);
     return HandleWindowEvent(event);
