diff --git i/include/wx/sharedptr.h w/include/wx/sharedptr.h
index 279e29b..ac13551 100644
--- i/include/wx/sharedptr.h
+++ w/include/wx/sharedptr.h
@@ -115,8 +115,7 @@ private:
     {
         if (m_ref)
         {
-            wxAtomicDec( m_ref->m_count );
-            if (m_ref->m_count == 0)
+            if (wxAtomicDec( m_ref->m_count ) == 0)
             {
                 delete m_ref->m_ptr;
                 delete m_ref;
