Index: src/stc/stc.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/contrib/src/stc/stc.cpp,v
retrieving revision 1.61
diff -u -r1.61 stc.cpp
--- src/stc/stc.cpp	2003/09/19 21:44:42	1.61
+++ src/stc/stc.cpp	2003/10/19 16:36:07
@@ -136,11 +136,24 @@
                                    const wxPoint& pos,
                                    const wxSize& size,
                                    long style,
-                                   const wxString& name) :
-    wxControl(parent, id, pos, size,
-              style | wxVSCROLL | wxHSCROLL | wxWANTS_CHARS | wxCLIP_CHILDREN,
-              wxDefaultValidator, name)
+                                   const wxString& name)
+{
+    m_swx = NULL;
+    Create(parent, id, pos, size, style, name);
+}
+
+
+void wxStyledTextCtrl::Create(wxWindow *parent,
+                                   wxWindowID id,
+                                   const wxPoint& pos,
+                                   const wxSize& size,
+                                   long style,
+                                   const wxString& name)
 {
+    wxControl::Create(parent, id, pos, size,
+              style | wxVSCROLL | wxHSCROLL | wxWANTS_CHARS | wxCLIP_CHILDREN,
+              wxDefaultValidator, name);
+
 #ifdef LINK_LEXERS
     Scintilla_LinkLexers();
 #endif
@@ -2201,8 +2214,11 @@
 }
 
 void wxStyledTextCtrl::OnSize(wxSizeEvent& WXUNUSED(evt)) {
-    wxSize sz = GetClientSize();
-    m_swx->DoSize(sz.x, sz.y);
+    if (m_swx)
+    {
+        wxSize sz = GetClientSize();
+        m_swx->DoSize(sz.x, sz.y);
+    }
 }
 
 void wxStyledTextCtrl::OnMouseLeftDown(wxMouseEvent& evt) {
Index: include/wx/stc/stc.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/contrib/include/wx/stc/stc.h,v
retrieving revision 1.45
diff -u -r1.45 stc.h
--- include/wx/stc/stc.h	2003/09/24 20:06:19	1.45
+++ include/wx/stc/stc.h	2003/10/19 16:36:09
@@ -1166,6 +1166,12 @@
 #ifndef SWIG
     ~wxStyledTextCtrl();
 #endif
+    wxStyledTextCtrl() { m_swx = NULL; }
+
+    void Create(wxWindow *parent, wxWindowID id,
+                     const wxPoint& pos = wxDefaultPosition,
+                     const wxSize& size = wxDefaultSize, long style = 0,
+                     const wxString& name = wxSTCNameStr);
 
 //----------------------------------------------------------------------
 // BEGIN generated section.  The following code is automatically generated

 	  	 
