--- hypertreelist.py	2013-08-20 15:31:57.994443693 +0100
+++ hypertreelist.fixed.py	2013-08-20 15:31:40.110443501 +0100
@@ -3169,14 +3169,15 @@
             else:
                 oldY = y_mid + h/2
             
-            for child in item.GetChildren():
+            children = item.GetChildren()
 
+            for child in children:
                 y, x_maincol = self.PaintLevel(child, dc, level+1, y, x_maincol)
 
-                # draw vertical line
-                if not self.HasAGWFlag(wx.TR_NO_LINES):
-                    Y1 = child.GetY() + child.GetHeight()/2
-                    dc.DrawLine(x, oldY, x, Y1)
+            if not self.HasAGWFlag(wx.TR_NO_LINES) and children:
+                last_child = children[-1]
+                Y1 = last_child.GetY() + last_child.GetHeight()/2
+                dc.DrawLine(x, oldY, x, Y1)
 
         return y, x_maincol        
 
