Index: src/msw/calctrl.cpp
===================================================================
--- src/msw/calctrl.cpp	(revision 72024)
+++ src/msw/calctrl.cpp	(working copy)
@@ -475,6 +475,10 @@
                 startDate.SetFromMSWSysDate(ds->stStart);
 
                 wxDateTime currentDate = m_date;
+
+                // make sure we have a valid wxDateTime to work with
+                if( !currentDate.IsValid() ) currentDate = startDate;
+
                 // Set to the start of month for comparison with startDate to
                 // work correctly.
                 currentDate.SetDay(1);
