--- src\common\sizer-o.cpp	2007-03-05 15:32:12.562500000 +0100
+++ src\common\sizer.cpp	2007-03-25 16:55:02.906250000 +0200
@@ -94,6 +94,7 @@
 {
     Init();
 
+    
     m_proportion = flags.GetProportion();
     m_flag = flags.GetFlags();
     m_border = flags.GetBorderInPixels();
@@ -106,8 +107,6 @@
     m_proportion = 0;
     m_border = 0;
     m_flag = 0;
-
-    m_kind = Item_None;
 }
 
 // window item
@@ -128,12 +127,34 @@
     SetRatio(m_minSize);
 }
 
+// window item
+void wxSizerItem::AssignWindow(wxWindow *window)
+{
+    wxCHECK_RET( window, _T("NULL window in wxSizerItem::SetWindow()") );
+
+    Free( );
+    
+    m_kind = Item_Window;
+    m_window = window;
+
+    // window doesn't become smaller than its initial size, whatever happens
+    m_minSize = window->GetSize();
+
+    if ( m_flag & wxFIXED_MINSIZE )
+        window->SetMinSize(m_minSize);
+
+    // aspect ratio calculated from initial size
+    SetRatio(m_minSize);
+}
+
 wxSizerItem::wxSizerItem(wxWindow *window,
                          int proportion,
                          int flag,
                          int border,
                          wxObject* userData)
-           : m_proportion(proportion),
+           : m_kind(Item_None),
+             m_sizer(NULL),
+             m_proportion(proportion),
              m_border(border),
              m_flag(flag),
              m_userData(userData)
@@ -148,12 +169,21 @@
     m_sizer = sizer;
 }
 
+void wxSizerItem::AssignSizer(wxSizer *sizer)
+{
+    Free( );
+    m_kind = Item_Sizer;
+    m_sizer = sizer;
+}
+
 wxSizerItem::wxSizerItem(wxSizer *sizer,
                          int proportion,
                          int flag,
                          int border,
                          wxObject* userData)
-           : m_proportion(proportion),
+           : m_kind(Item_None),
+             m_sizer(NULL),
+             m_proportion(proportion),
              m_border(border),
              m_flag(flag),
              m_ratio(0.0),
@@ -173,13 +203,30 @@
     SetRatio(size);
 }
 
+void wxSizerItem::SetSpacer(int width, int height)
+{ 
+    SetSpacer(wxSize(width, height)); 
+} 
+
+// spacer item
+void wxSizerItem::AssignSpacer(const wxSize& size)
+{
+    Free();
+    m_kind = Item_Spacer;
+    m_spacer = new wxSizerSpacer(size);
+    m_minSize = size;
+    SetRatio(size);
+}
+
 wxSizerItem::wxSizerItem(int width,
                          int height,
                          int proportion,
                          int flag,
                          int border,
                          wxObject* userData)
-           : m_minSize(width, height), // minimal size is the initial size
+           : m_kind(Item_None),
+             m_sizer(NULL),
+             m_minSize(width, height), // minimal size is the initial size
              m_proportion(proportion),
              m_border(border),
              m_flag(flag),
@@ -191,7 +238,10 @@
 wxSizerItem::~wxSizerItem()
 {
     delete m_userData;
+    Free( );
+}
 
+void wxSizerItem::Free( ){
     switch ( m_kind )
     {
         case Item_None:
@@ -213,6 +263,9 @@
         default:
             wxFAIL_MSG( _T("unexpected wxSizerItem::m_kind") );
     }
+    
+    m_kind = Item_None;// To avoid freeing them again
+    m_sizer = NULL;
 }
 
 wxSize wxSizerItem::GetSpacer() const
--- include\wx\sizer-o.h	2007-03-05 15:32:28.687500000 +0100
+++ include\wx\sizer.h	2007-03-25 16:08:33.500000000 +0200
@@ -323,21 +323,28 @@
     wxPoint GetPosition() const
         { return m_pos; }
 
+    // these functions will free/delete the old sizer/spacer. A wxWindow is not deleted
+    void AssignWindow(wxWindow *window);
+    void AssignSizer(wxSizer *sizer);
+    void AssignSpacer(const wxSize& size);
+    void AssignSpacer(int width, int height) { AssignSpacer(wxSize(width, height)); }
 
     // these functions do not free old sizer/spacer
-    void SetWindow(wxWindow *window);
-    void SetSizer(wxSizer *sizer);
-    void SetSpacer(const wxSize& size);
-    void SetSpacer(int width, int height) { SetSpacer(wxSize(width, height)); }
+    wxDEPRECATED( void SetWindow(wxWindow *window) );
+    wxDEPRECATED( void SetSizer(wxSizer *sizer) );
+    wxDEPRECATED( void SetSpacer(const wxSize& size) );
+    wxDEPRECATED( void SetSpacer(int width, int height) );
 
 protected:
     // common part of several ctors
-    void Init() { m_userData = NULL; }
+    void Init() { m_userData = NULL; m_kind = Item_None; m_sizer=NULL; }
 
     // common part of ctors taking wxSizerFlags
     void Init(const wxSizerFlags& flags);
 
-
+    // Before setting s sizeritem to something else, release previous (windows) item
+    void Free( );
+ 
     // discriminated union: depending on m_kind one of the fields is valid
     enum
     {

 	  	 
