Index: src/common/imagjpeg.cpp
===================================================================
--- src/common/imagjpeg.cpp	(revision 54843)
+++ src/common/imagjpeg.cpp	(working copy)
@@ -149,9 +149,8 @@
 // JPEG error manager:
 
 struct wx_error_mgr {
-  struct jpeg_error_mgr pub;    /* "public" fields */
-
   jmp_buf setjmp_buffer;    /* for return to caller */
+  struct jpeg_error_mgr pub;    /* "public" fields */
 };
 
 typedef struct wx_error_mgr * wx_error_ptr;
Index: src/common/utilscmn.cpp
===================================================================
--- src/common/utilscmn.cpp	(revision 54843)
+++ src/common/utilscmn.cpp	(working copy)
@@ -954,6 +954,8 @@
         if ( wxExecute(xdg_open + " " + document) )
             return true;
     }
+#else
+	wxUnusedVar(document);
 #endif
 
    return false;
Index: src/html/htmltag.cpp
===================================================================
--- src/html/htmltag.cpp	(revision 54843)
+++ src/html/htmltag.cpp	(working copy)
@@ -302,7 +302,7 @@
 
     /* Find parameters and their values: */
 
-    wxChar c;
+    wxChar c = 0;
 
     // fill-in name, params and begin pos:
     wxString::const_iterator i(pos+1);
Index: src/msw/listctrl.cpp
===================================================================
--- src/msw/listctrl.cpp	(revision 54843)
+++ src/msw/listctrl.cpp	(working copy)
@@ -2706,7 +2706,7 @@
 
     // same thing for CDIS_FOCUS (except simpler as there is only one of them)
     if ( ::GetFocus() == hwndList &&
-            ListView_GetNextItem(hwndList, (WPARAM)-1, LVNI_FOCUSED) == item )
+            ListView_GetNextItem(hwndList, -1, LVNI_FOCUSED) == item )
     {
         nmcd.uItemState |= CDIS_FOCUS;
     }
Index: src/msw/main.cpp
===================================================================
--- src/msw/main.cpp	(revision 54843)
+++ src/msw/main.cpp	(working copy)
@@ -214,17 +214,6 @@
 
 #else // !wxUSE_ON_FATAL_EXCEPTION
 
-#if defined(__VISUALC__) && !defined(__WXWINCE__)
-
-static void
-wxSETranslator(unsigned int WXUNUSED(code), EXCEPTION_POINTERS * WXUNUSED(ep))
-{
-    // see wxSETranslator() version for wxUSE_ON_FATAL_EXCEPTION above
-    throw;
-}
-
-#endif // __VISUALC__
-
 int wxEntry(int& argc, wxChar **argv)
 {
     DisableAutomaticSETranslator();
Index: src/tiff/libtiff/tif_dirinfo.c
===================================================================
--- src/tiff/libtiff/tif_dirinfo.c	(revision 54843)
+++ src/tiff/libtiff/tif_dirinfo.c	(working copy)
@@ -753,7 +753,7 @@
 
             ret = (const TIFFFieldInfo **) lfind(&pkey,
 						 tif->tif_fieldinfo, 
-						 &tif->tif_nfields,
+						 &(unsigned int)tif->tif_nfields,
 						 sizeof(TIFFFieldInfo *),
 						 tagNameCompare);
 	    return (ret) ? (*ret) : NULL;
