diff --git a/src/msw/utilsgui.cpp b/src/msw/utilsgui.cpp
index b2cce08..cc1f371 100644
--- a/src/msw/utilsgui.cpp
+++ b/src/msw/utilsgui.cpp
@@ -41,6 +41,7 @@
 // implementation
 // ============================================================================
 
+#ifdef __WXMSW__
 // Emit a beeeeeep
 void wxBell()
 {
@@ -197,6 +198,8 @@ void wxDisplaySizeMM(int *width, int *height)
 #endif
 }
 
+#endif // __WXMSW__
+
 void wxClientDisplayRect(int *x, int *y, int *width, int *height)
 {
 #if defined(__WXMICROWIN__)
@@ -215,6 +218,8 @@ void wxClientDisplayRect(int *x, int *y, int *width, int *height)
 #endif
 }
 
+#ifdef __WXMSW__
+
 // ---------------------------------------------------------------------------
 // window information functions
 // ---------------------------------------------------------------------------
@@ -370,6 +375,8 @@ extern bool wxEnableFileNameAutoComplete(HWND hwnd)
 #endif // wxUSE_DYNLIB_CLASS/!wxUSE_DYNLIB_CLASS
 }
 
+#endif // __WXMSW__
+
 // ----------------------------------------------------------------------------
 // Launch document with default app
 // ----------------------------------------------------------------------------
