diff -bu1 -r wxWindows/src/common/dlgcmn.cpp wxWin/src/common/dlgcmn.cpp
--- wxWindows/src/common/dlgcmn.cpp	2003-08-11 10:29:42.000000000 +0200
+++ wxWin/src/common/dlgcmn.cpp	2003-10-06 12:05:39.000000000 +0200
@@ -174,10 +174,5 @@
     
-    wxBoxSizer *box = NULL;
-    
     // If we have a PDA screen, put yes/no button over 
     // all other buttons, otherwise on the left side.
-    if (is_pda)
-        box = new wxBoxSizer( wxVERTICAL );
-    else
-        box = new wxBoxSizer( wxHORIZONTAL );
+    wxBoxSizer *box = (is_pda) ? new wxBoxSizer( wxVERTICAL ) : new wxBoxSizer( wxHORIZONTAL );
         
diff -bu1 -r wxWindows/src/common/docmdi.cpp wxWin/src/common/docmdi.cpp
--- wxWindows/src/common/docmdi.cpp	2003-08-11 10:29:42.000000000 +0200
+++ wxWin/src/common/docmdi.cpp	2003-10-06 12:06:28.000000000 +0200
@@ -167,7 +167,5 @@
   {
-    bool ans = FALSE;
-    if (!event.CanVeto())
-      ans = TRUE; // Must delete.
-    else
-      ans = m_childView->Close(FALSE); // FALSE means don't delete associated window
+    bool ans = (!event.CanVeto()) ?
+               TRUE : // Must delete.
+               m_childView->Close(FALSE) ; // FALSE means don't delete associated window
 
diff -bu1 -r wxWindows/src/common/docview.cpp wxWin/src/common/docview.cpp
--- wxWindows/src/common/docview.cpp	2003-08-11 10:29:44.000000000 +0200
+++ wxWin/src/common/docview.cpp	2003-10-06 15:40:29.000000000 +0200
@@ -954,4 +954,3 @@
         // Pass two printout objects: for preview, and possible printing.
-        wxPrintPreviewBase *preview = (wxPrintPreviewBase *) NULL;
-        preview = new wxPrintPreview(printout, view->OnCreatePrintout());
+        wxPrintPreviewBase *preview = new wxPrintPreview(printout, view->OnCreatePrintout());
 
@@ -1177,3 +1176,3 @@
     // Existing document
-    wxDocTemplate *temp = (wxDocTemplate *) NULL;
+    wxDocTemplate *temp;
 
@@ -1790,7 +1789,5 @@
     {
-        bool ans = FALSE;
-        if (!event.CanVeto())
-            ans = TRUE; // Must delete.
-        else
-            ans = m_childView->Close(FALSE); // FALSE means don't delete associated window
+        bool ans = (!event.CanVeto()) ?
+            TRUE : // Must delete.
+            m_childView->Close(FALSE); // FALSE means don't delete associated window
 
@@ -1908,4 +1905,10 @@
     GetPPIScreen(&ppiScreenX, &ppiScreenY);
+#ifdef __BORLANDC__
+    ppiScreenY; // prevent dummy warnings
+#endif
     int ppiPrinterX, ppiPrinterY;
     GetPPIPrinter(&ppiPrinterX, &ppiPrinterY);
+#ifdef __BORLANDC__
+    ppiPrinterY; // prevent dummy warnings
+#endif
 
@@ -1924,2 +1927,5 @@
     GetPageSizePixels(&pageWidth, &pageHeight);
+#ifdef __BORLANDC__
+    pageHeight; // prevent dummy warnings
+#endif
 
diff -bu1 -r wxWindows/src/common/dynlib.cpp wxWin/src/common/dynlib.cpp
--- wxWindows/src/common/dynlib.cpp	2003-08-25 19:00:26.000000000 +0200
+++ wxWin/src/common/dynlib.cpp	2003-10-06 15:35:10.000000000 +0200
@@ -328,2 +328,5 @@
 
+#ifdef __BORLANDC__
+    symbol; // prevent dummy warnings
+#endif
 #if defined(__WXMAC__) && !defined(__DARWIN__)
diff -bu1 -r wxWindows/src/common/encconv.cpp wxWin/src/common/encconv.cpp
--- wxWindows/src/common/encconv.cpp	2003-09-23 07:29:41.000000000 +0200
+++ wxWin/src/common/encconv.cpp	2003-10-06 11:27:37.000000000 +0200
@@ -91,3 +91,3 @@
     unsigned i;
-    wxUint16 *in_tbl = NULL, *out_tbl = NULL;
+    wxUint16 *in_tbl, *out_tbl = NULL;
 
diff -bu1 -r wxWindows/src/common/event.cpp wxWin/src/common/event.cpp
--- wxWindows/src/common/event.cpp	2003-10-06 08:25:34.000000000 +0200
+++ wxWin/src/common/event.cpp	2003-10-06 10:37:31.000000000 +0200
@@ -548,6 +548,2 @@
     {
-        default:
-            wxFAIL_MSG(wxT("invalid parameter in wxMouseEvent::ButtonDClick"));
-            // fall through
-
         case wxMOUSE_BTN_ANY:
@@ -565,2 +561,3 @@
 
+    wxFAIL_MSG(wxT("invalid parameter in wxMouseEvent::ButtonDClick"));
     return false;
@@ -573,6 +570,2 @@
     {
-        default:
-            wxFAIL_MSG(wxT("invalid parameter in wxMouseEvent::ButtonDown"));
-            // fall through
-
         case wxMOUSE_BTN_ANY:
@@ -590,2 +583,3 @@
 
+    wxFAIL_MSG(wxT("invalid parameter in wxMouseEvent::ButtonDown"));
     return false;
@@ -598,6 +592,2 @@
     {
-        default:
-            wxFAIL_MSG(wxT("invalid parameter in wxMouseEvent::ButtonUp"));
-            // fall through
-
         case wxMOUSE_BTN_ANY:
@@ -615,2 +605,3 @@
 
+    wxFAIL_MSG(wxT("invalid parameter in wxMouseEvent::ButtonUp"));
     return false;
@@ -623,6 +614,2 @@
     {
-        default:
-            wxFAIL_MSG(wxT("invalid parameter in wxMouseEvent::Button"));
-            // fall through
-
         case wxMOUSE_BTN_ANY:
@@ -642,2 +629,3 @@
 
+    wxFAIL_MSG(wxT("invalid parameter in wxMouseEvent::Button"));
     return false;
@@ -649,6 +637,2 @@
     {
-        default:
-            wxFAIL_MSG(wxT("invalid parameter in wxMouseEvent::ButtonIsDown"));
-            // fall through
-
         case wxMOUSE_BTN_ANY:
@@ -666,2 +650,3 @@
 
+    wxFAIL_MSG(wxT("invalid parameter in wxMouseEvent::ButtonIsDown"));
     return false;
diff -bu1 -r wxWindows/src/common/filefn.cpp wxWin/src/common/filefn.cpp
--- wxWindows/src/common/filefn.cpp	2003-09-14 21:13:55.000000000 +0200
+++ wxWin/src/common/filefn.cpp	2003-10-06 15:35:49.000000000 +0200
@@ -327,4 +327,3 @@
 
-  wxChar *filename = (wxChar*) NULL; /* shut up buggy egcs warning */
-  filename = wxIsAbsolutePath (buf) ? wxFileNameFromPath (buf) : (wxChar *)buf;
+  wxChar *filename = wxIsAbsolutePath (buf) ? wxFileNameFromPath (buf) : (wxChar *)buf;
 
@@ -734,5 +733,5 @@
   // Handle environment
-  const wxChar *val = (const wxChar *) NULL;
+  const wxChar *val;
 #ifndef __WXWINCE__
-  wxChar *tcp = (wxChar *) NULL;
+  wxChar *tcp;
   if (envname != WXSTRINGCAST NULL && (val = wxGetenv (WXSTRINGCAST envname)) != NULL &&
@@ -750,3 +749,3 @@
   // Handle User's home (ignore root homes!)
-  size_t len = 0;
+  size_t len;
   if ((val = wxGetUserHome (user)) != NULL &&
@@ -1146,5 +1145,5 @@
 
-  FILE *fp1 = (FILE *) NULL;
-  FILE *fp2 = (FILE *) NULL;
-  FILE *fp3 = (FILE *) NULL;
+  FILE *fp1;
+  FILE *fp2;
+  FILE *fp3;
   // Open the inputs and outputs
@@ -1456,3 +1455,3 @@
 
-    int dirFlags = 0;
+    int dirFlags;
     switch (flags)
@@ -1506,2 +1505,5 @@
     bool ok = FALSE;
+#ifdef __BORLANDC__
+    ok; // prevent dummy warnings
+#endif
 
diff -bu1 -r wxWindows/src/common/ftp.cpp wxWin/src/common/ftp.cpp
--- wxWindows/src/common/ftp.cpp	2003-08-18 08:15:04.000000000 +0200
+++ wxWin/src/common/ftp.cpp	2003-10-06 15:38:56.000000000 +0200
@@ -855,2 +855,5 @@
                                 ok = TRUE;
+                            #ifdef __BORLANDC__
+                                ok; // prevent dummy warnings
+                            #endif
                             }
@@ -871,2 +874,5 @@
                                 ok = TRUE;
+                            #ifdef __BORLANDC__
+                                ok; // prevent dummy warnings
+                            #endif
                             }
diff -bu1 -r wxWindows/src/common/imagbmp.cpp wxWin/src/common/imagbmp.cpp
--- wxWindows/src/common/imagbmp.cpp	2003-08-11 10:29:46.000000000 +0200
+++ wxWin/src/common/imagbmp.cpp	2003-10-06 14:32:20.000000000 +0200
@@ -461,3 +461,3 @@
     // allocate space for palette if needed:
-    _cmap *cmap = NULL;
+    _cmap *cmap;
 
@@ -604,4 +604,2 @@
 
-    int line = 0;
-    int column = 0;
     int linesize = ((width * bpp + 31) / 32) * 4;
@@ -609,6 +607,6 @@
     /* BMPs are stored upside down */
-    for ( line = (height - 1); line >= 0; line-- )
+    for ( int line = (height - 1); line >= 0; line-- )
     {
         int linepos = 0;
-        for ( column = 0; column < width ; )
+        for ( int column = 0; column < width ; )
         {
@@ -616,3 +614,2 @@
             {
-                int index = 0;
                 linepos++;
@@ -621,6 +618,5 @@
                 {
-                    int bit = 0;
-                    for (bit = 0; bit < 8 && column < width; bit++)
+                    for (int bit = 0; bit < 8 && column < width; bit++)
                     {
-                        index = ((aByte & (0x80 >> bit)) ? 1 : 0);
+                        int index = ((aByte & (0x80 >> bit)) ? 1 : 0);
                         ptr[poffset] = cmap[index].r;
@@ -702,6 +698,5 @@
                     {
-                        int nibble = 0;
-                        for (nibble = 0; nibble < 2 && column < width; nibble++)
+                        for (int nibble = 0; nibble < 2 && column < width; nibble++)
                         {
-                            index = ((aByte & (0xF0 >> nibble * 4)) >> (!nibble * 4));
+                            int index = ((aByte & (0xF0 >> nibble * 4)) >> (!nibble * 4));
                             if ( index >= 16 )
diff -bu1 -r wxWindows/src/common/image.cpp wxWin/src/common/image.cpp
--- wxWindows/src/common/image.cpp	2003-08-11 10:29:46.000000000 +0200
+++ wxWin/src/common/image.cpp	2003-10-06 15:55:22.000000000 +0200
@@ -1809,2 +1809,5 @@
                               (w1 + w2 + w3 + w4) );
+                    #ifdef __BORLANDC__
+                        v1;v2;v3;v4; // prevent dummy warnings
+                    #endif
                     }
@@ -1838,2 +1841,5 @@
                     *(dst++) = *(p++);
+                #ifdef __BORLANDC__
+                    p; // prevent dummy warnings
+                #endif
                 }
diff -bu1 -r wxWindows/src/common/imagjpeg.cpp wxWin/src/common/imagjpeg.cpp
--- wxWindows/src/common/imagjpeg.cpp	2003-08-11 10:29:46.000000000 +0200
+++ wxWin/src/common/imagjpeg.cpp	2003-10-06 12:09:36.000000000 +0200
@@ -180,3 +180,2 @@
             sizeof(my_source_mgr));
-        src = (my_src_ptr) cinfo->src;
     }
diff -bu1 -r wxWindows/src/common/imagpng.cpp wxWin/src/common/imagpng.cpp
--- wxWindows/src/common/imagpng.cpp	2003-08-11 10:29:46.000000000 +0200
+++ wxWin/src/common/imagpng.cpp	2003-10-06 15:41:26.000000000 +0200
@@ -469,2 +469,5 @@
                                 r++;
+                            #ifdef __BORLANDC__
+                                r; // prevent dummy warnings
+                            #endif
                             }
diff -bu1 -r wxWindows/src/common/intl.cpp wxWin/src/common/intl.cpp
--- wxWindows/src/common/intl.cpp	2003-10-06 08:25:42.000000000 +0200
+++ wxWin/src/common/intl.cpp	2003-10-06 11:32:13.000000000 +0200
@@ -1571,3 +1571,7 @@
         {
-            int codepage = -1;
+            int codepage 
+                         #ifdef SETLOCALE_FAILS_ON_UNICODE_LANGS
+                         = -1
+                         #endif
+                         ;
             wxUint32 lcid = MAKELCID(MAKELANGID(info->WinLang, info->WinSublang),
diff -bu1 -r wxWindows/src/common/prntbase.cpp wxWin/src/common/prntbase.cpp
--- wxWindows/src/common/prntbase.cpp	2003-08-14 09:55:56.000000000 +0200
+++ wxWin/src/common/prntbase.cpp	2003-10-06 12:11:56.000000000 +0200
@@ -774,8 +774,2 @@
 
-    int x = (int)((canvasWidth - actualWidth)/2.0);
-    if (x < m_leftMargin)
-        x = m_leftMargin;
-    //  int y = m_topMargin;
-
-
     if (!m_previewBitmap)
diff -bu1 -r wxWindows/src/common/quantize.cpp wxWin/src/common/quantize.cpp
--- wxWindows/src/common/quantize.cpp	2003-08-11 15:48:06.000000000 +0200
+++ wxWin/src/common/quantize.cpp	2003-10-06 13:10:42.000000000 +0200
@@ -1304,3 +1304,2 @@
   hist3d histogram = cquantize->histogram;
-  int i;
 
@@ -1316,5 +1315,2 @@
 
-    /* Make sure color count is acceptable */
-    i = cinfo->actual_number_of_colors;
-
     {
@@ -1336,3 +1332,3 @@
   if (cquantize->needs_zeroed) {
-    for (i = 0; i < HIST_C0_ELEMS; i++) {
+    for (int i = 0; i < HIST_C0_ELEMS; i++) {
       memset((void  *) histogram[i], 0,
@@ -1512,4 +1508,2 @@
     int i;
-    int w = src.GetWidth();
-    int h = src.GetHeight();
 
@@ -1531,4 +1525,5 @@
     // create rows info:
+    int h = src.GetHeight();
+    int w = src.GetWidth();
     unsigned char **rows = new unsigned char *[h];
-    h = src.GetHeight(), w = src.GetWidth();
     unsigned char *imgdt = src.GetData();
diff -bu1 -r wxWindows/src/common/sckipc.cpp wxWin/src/common/sckipc.cpp
--- wxWindows/src/common/sckipc.cpp	2003-08-11 10:29:48.000000000 +0200
+++ wxWin/src/common/sckipc.cpp	2003-10-06 11:45:34.000000000 +0200
@@ -533,3 +533,2 @@
 
-  int msg = 0;
   wxDataInputStream *codeci;
@@ -553,3 +552,3 @@
   sockstrm = connection->m_sockstrm;
-  msg = codeci->Read8();
+  int msg = codeci->Read8();
 
diff -bu1 -r wxWindows/src/common/socket.cpp wxWin/src/common/socket.cpp
--- wxWindows/src/common/socket.cpp	2003-08-18 08:15:04.000000000 +0200
+++ wxWin/src/common/socket.cpp	2003-10-06 15:39:28.000000000 +0200
@@ -306,2 +306,5 @@
   int ret = 1;
+#ifdef __BORLANDC__
+  ret; // prevent dummy warnings
+#endif
 
@@ -507,2 +510,5 @@
   int ret = 1;
+#ifdef __BORLANDC__
+  ret; // prevent dummy warnings
+#endif
 
@@ -931,3 +937,3 @@
 
-  wxSocketEventFlags flag = 0;
+  wxSocketEventFlags flag;
   switch (notification)
diff -bu1 -r wxWindows/src/common/strconv.cpp wxWin/src/common/strconv.cpp
--- wxWindows/src/common/strconv.cpp	2003-09-29 09:08:25.000000000 +0200
+++ wxWin/src/common/strconv.cpp	2003-10-06 15:36:31.000000000 +0200
@@ -114,2 +114,5 @@
         if (output) *output++ = (wxUint16) input;
+#ifdef __BORLANDC__
+        output; // prevent dummy warnings
+#endif
         return 1;
@@ -127,2 +130,5 @@
         }
+#ifdef __BORLANDC__
+        output; // prevent dummy warnings
+#endif
         return 2;
diff -bu1 -r wxWindows/src/common/stream.cpp wxWin/src/common/stream.cpp
--- wxWindows/src/common/stream.cpp	2003-08-11 10:29:50.000000000 +0200
+++ wxWin/src/common/stream.cpp	2003-10-06 11:34:36.000000000 +0200
@@ -456,3 +456,3 @@
 
-    size_t ret = 0;
+    size_t ret;
 
diff -bu1 -r wxWindows/src/common/txtstrm.cpp wxWin/src/common/txtstrm.cpp
--- wxWindows/src/common/txtstrm.cpp	2003-08-11 10:29:50.000000000 +0200
+++ wxWin/src/common/txtstrm.cpp	2003-10-06 11:35:15.000000000 +0200
@@ -99,6 +99,5 @@
 {
-    wxChar c = (wxChar) 0;
     for (;;)
     {
-        c = NextChar();
+        wxChar c = NextChar();
         if (c == wxEOT) return (wxChar) 0;
diff -bu1 -r wxWindows/src/common/unzip.c wxWin/src/common/unzip.c
--- wxWindows/src/common/unzip.c	2003-01-29 15:00:56.000000000 +0100
+++ wxWin/src/common/unzip.c	2003-10-06 14:38:35.000000000 +0200
@@ -687,6 +687,9 @@
         {
-			if (fseek(s->file,lSeek,SEEK_CUR)==0)
-				lSeek=0;
-			else
+			if (fseek(s->file,lSeek,SEEK_CUR)!=0)
 				err=UNZ_ERRNO;
+#if 0
+  // important only if lSeek is supposed to be used later
+      if(err==UNZ_ERRNO)
+				lSeek=0;
+#endif
         }
@@ -696,2 +699,7 @@
 				err=UNZ_ERRNO;
+#if 1
+  // lSeek is no more important but ...
+  }
+#else
+  // ... if you need it further, you have to increasy its value accordingly
 		lSeek+=file_info.size_file_comment - uSizeRead;
@@ -700,2 +708,3 @@
 		lSeek+=file_info.size_file_comment;
+#endif
 
@@ -744,3 +753,3 @@
 {
-	int err=UNZ_OK;
+  int err;
 	unz_s* s;
@@ -952,3 +961,2 @@
 {
-	int err=UNZ_OK;
 	int Store;
@@ -991,5 +999,2 @@
 
-	if ((s->cur_file_info.compression_method!=0) &&
-        (s->cur_file_info.compression_method!=Z_DEFLATED))
-		err=UNZ_BADZIPFILE;
 	Store = s->cur_file_info.compression_method==0;
@@ -1011,4 +1016,3 @@
 
-	  err=inflateInit2(&pfile_in_zip_read_info->stream, -MAX_WBITS);
-	  if (err == Z_OK)
+	  if (inflateInit2(&pfile_in_zip_read_info->stream, -MAX_WBITS) == Z_OK)
 	    pfile_in_zip_read_info->stream_initialised=1;
diff -bu1 -r wxWindows/src/common/url.cpp wxWin/src/common/url.cpp
--- wxWindows/src/common/url.cpp	2003-08-11 10:29:50.000000000 +0200
+++ wxWin/src/common/url.cpp	2003-10-06 11:48:18.000000000 +0200
@@ -274,4 +274,2 @@
 {
-  wxInputStream *the_i_stream = NULL;
-
   if (!m_protocol)
@@ -312,6 +310,5 @@
   // When we use a proxy, we have to pass the whole URL to it.
-  if (m_useProxy)
-    the_i_stream = m_protocol->GetInputStream(m_url);
-  else
-    the_i_stream = m_protocol->GetInputStream(m_path);
+  wxInputStream *the_i_stream = 
+       (m_useProxy) ? m_protocol->GetInputStream(m_url) :
+                      m_protocol->GetInputStream(m_path);
 
diff -bu1 -r wxWindows/src/common/zipstrm.cpp wxWin/src/common/zipstrm.cpp
--- wxWindows/src/common/zipstrm.cpp	2003-08-11 10:29:52.000000000 +0200
+++ wxWin/src/common/zipstrm.cpp	2003-10-06 11:37:17.000000000 +0200
@@ -124,3 +124,3 @@
 
-    size_t toskip = 0;
+    size_t toskip;
     if ( nextpos > m_Pos )
diff -bu1 -r wxWindows/src/generic/calctrl.cpp wxWin/src/generic/calctrl.cpp
--- wxWindows/src/generic/calctrl.cpp	2003-09-24 09:50:34.000000000 +0200
+++ wxWin/src/generic/calctrl.cpp	2003-10-06 13:31:30.000000000 +0200
@@ -670,7 +670,5 @@
 {
-    bool retval = TRUE;
     // Check if the given date is in the range specified
-    retval = ( ( ( m_lowdate.IsValid() ) ? ( date >= m_lowdate ) : TRUE )
+    return ( ( ( m_lowdate.IsValid() ) ? ( date >= m_lowdate ) : TRUE )
         && ( ( m_highdate.IsValid() ) ? ( date <= m_highdate ) : TRUE ) );
-    return retval;
 }
@@ -1373,3 +1371,3 @@
         {
-            day += ( startOnMonday ) ? 0 : 1;
+            *day += ( startOnMonday ) ? 0 : 1;
         }
diff -bu1 -r wxWindows/src/generic/colrdlgg.cpp wxWin/src/generic/colrdlgg.cpp
--- wxWindows/src/generic/colrdlgg.cpp	2003-08-11 10:29:52.000000000 +0200
+++ wxWin/src/generic/colrdlgg.cpp	2003-10-07 10:12:56.000000000 +0200
@@ -342,2 +342,5 @@
                     initColourFound = TRUE;
+                #ifdef __BORLANDC__
+                    initColourFound; // prevent dummy warnings
+                #endif
                     break;
diff -bu1 -r wxWindows/src/generic/dcpsg.cpp wxWin/src/generic/dcpsg.cpp
--- wxWindows/src/generic/dcpsg.cpp	2003-09-22 07:59:55.000000000 +0200
+++ wxWin/src/generic/dcpsg.cpp	2003-10-06 14:37:43.000000000 +0200
@@ -727,7 +727,3 @@
         // 'the proportion of the smallest X or Y dimension'
-        double smallest = 0.0;
-        if (width < height)
-            smallest = width;
-        else
-            smallest = height;
+        double smallest = (width < height) ? width : height;
         radius =  (-radius * smallest);
@@ -964,4 +960,2 @@
             name = "/ZapfChancery-MediumItalic";
-            Style  = wxNORMAL;
-            Weight = wxNORMAL;
             break;
@@ -1048,3 +1042,4 @@
 
-    const char *psdash = (char *) NULL;
+    const char *psdash;
+
     switch (m_pen.GetStyle())
@@ -1653,3 +1648,4 @@
 
-	while ((node = node->GetNext()))
+  node = node->GetNext();
+	while (node)
     {
@@ -1675,2 +1671,4 @@
         CalcBoundingBox( (wxCoord)x3, (wxCoord)y3 );
+
+        node = node->GetNext();
     }
@@ -2196,3 +2194,3 @@
 
-        const wxChar *name = NULL;
+        const wxChar *name;
 
@@ -2220,4 +2218,3 @@
                 name = wxT("Zapf.afm");
-                Style = wxNORMAL;
-                Weight = wxNORMAL;
+                break;
             }
diff -bu1 -r wxWindows/src/generic/dirctrlg.cpp wxWin/src/generic/dirctrlg.cpp
--- wxWindows/src/generic/dirctrlg.cpp	2003-09-24 09:50:34.000000000 +0200
+++ wxWin/src/generic/dirctrlg.cpp	2003-10-06 15:42:29.000000000 +0200
@@ -128,2 +128,5 @@
         int imageId = wxFileIconsTable::drive;
+#ifdef __BORLANDC__
+        imageId; // prevent dummy warnings
+#endif
         int driveType = ::GetDriveType(path);
@@ -1208,4 +1211,3 @@
     {
-        size_t i = 0;
-        for (i = 0; i < n; i++)
+        for (size_t i = 0; i < n; i++)
             Append(descriptions[i]);
diff -bu1 -r wxWindows/src/generic/filedlgg.cpp wxWin/src/generic/filedlgg.cpp
--- wxWindows/src/generic/filedlgg.cpp	2003-09-22 07:59:56.000000000 +0200
+++ wxWin/src/generic/filedlgg.cpp	2003-10-06 11:59:36.000000000 +0200
@@ -492,3 +492,2 @@
 
-    wxFileData *fd = (wxFileData *) NULL;
     wxListItem item;
@@ -506,3 +505,3 @@
             {
-            fd = new wxFileData(paths[n], names[n], wxFileData::is_drive, icons[n]);
+            wxFileData *fd = new wxFileData(paths[n], names[n], wxFileData::is_drive, icons[n]);
                 Add(fd, item);
@@ -521,3 +520,3 @@
 #endif // __UNIX__
-            fd = new wxFileData(p, wxT(".."), wxFileData::is_dir, wxFileIconsTable::folder);
+            wxFileData *fd = new wxFileData(p, wxT(".."), wxFileData::is_dir, wxFileIconsTable::folder);
             Add(fd, item);
@@ -548,3 +547,3 @@
             {
-                fd = new wxFileData(dirPrefix + f, f, wxFileData::is_dir, wxFileIconsTable::folder);
+                wxFileData *fd = new wxFileData(dirPrefix + f, f, wxFileData::is_dir, wxFileIconsTable::folder);
                 Add(fd, item);
@@ -563,3 +562,3 @@
                 {
-                    fd = new wxFileData(dirPrefix + f, f, wxFileData::is_file, wxFileIconsTable::file);
+                    wxFileData *fd = new wxFileData(dirPrefix + f, f, wxFileData::is_file, wxFileIconsTable::file);
                     Add(fd, item);
diff -bu1 -r wxWindows/src/generic/fontdlgg.cpp wxWin/src/generic/fontdlgg.cpp
--- wxWindows/src/generic/fontdlgg.cpp	2003-09-30 07:40:18.000000000 +0200
+++ wxWin/src/generic/fontdlgg.cpp	2003-10-06 13:04:19.000000000 +0200
@@ -331,8 +331,5 @@
 
-  int fontFamily = 0;  /* shut up buggy egcs warnings */
-  fontFamily = wxFontFamilyStringToInt(WXSTRINGCAST familyChoice->GetStringSelection());
-  int fontWeight = 0;
-  fontWeight = wxFontWeightStringToInt(WXSTRINGCAST weightChoice->GetStringSelection());
-  int fontStyle = 0;
-  fontStyle = wxFontStyleStringToInt(WXSTRINGCAST styleChoice->GetStringSelection());
+  int fontFamily = wxFontFamilyStringToInt(WXSTRINGCAST familyChoice->GetStringSelection());
+  int fontWeight = wxFontWeightStringToInt(WXSTRINGCAST weightChoice->GetStringSelection());
+  int fontStyle = wxFontStyleStringToInt(WXSTRINGCAST styleChoice->GetStringSelection());
   int fontSize = wxAtoi(pointSizeChoice->GetStringSelection());
@@ -344,4 +341,3 @@
   {
-    wxColour *col = (wxColour*) NULL;
-    col = wxTheColourDatabase->FindColour(colourChoice->GetStringSelection());
+    wxColour *col = wxTheColourDatabase->FindColour(colourChoice->GetStringSelection());
     if (col)
diff -bu1 -r wxWindows/src/generic/grid.cpp wxWin/src/generic/grid.cpp
--- wxWindows/src/generic/grid.cpp	2003-09-24 09:50:35.000000000 +0200
+++ wxWin/src/generic/grid.cpp	2003-10-06 14:33:41.000000000 +0200
@@ -1963,6 +1963,5 @@
         // get checkbox size
-        wxCoord checkSize = 0;
         wxCheckBox *checkbox = new wxCheckBox(&grid, -1, wxEmptyString);
         wxSize size = checkbox->GetBestSize();
-        checkSize = size.y + 2*wxGRID_CHECKMARK_MARGIN;
+        wxCoord checkSize = size.y + 2*wxGRID_CHECKMARK_MARGIN;
 
@@ -2586,9 +2585,5 @@
                     //Also check merge cache, so we don't have to re-merge every time..
-                    wxGridCellAttr *attrcell = (wxGridCellAttr *)NULL,
-                                   *attrrow = (wxGridCellAttr *)NULL,
-                                   *attrcol = (wxGridCellAttr *)NULL;
-
-                    attrcell = m_data->m_cellAttrs.GetAttr(row, col);
-                    attrcol = m_data->m_colAttrs.GetAttr(col);
-                    attrrow = m_data->m_rowAttrs.GetAttr(row);
+                    wxGridCellAttr *attrcell = m_data->m_cellAttrs.GetAttr(row, col);
+                    wxGridCellAttr *attrrow = m_data->m_rowAttrs.GetAttr(row);
+                    wxGridCellAttr *attrcol = m_data->m_colAttrs.GetAttr(col);
 
@@ -6004,3 +5999,3 @@
    bool claimed;
-   bool vetoed= FALSE;
+   bool vetoed;
 
@@ -6071,3 +6066,3 @@
    bool claimed;
-   bool vetoed= FALSE;
+   bool vetoed;
 
@@ -9094,5 +9089,3 @@
 {
-    wxGridCellAttr *attr = (wxGridCellAttr *)NULL;
-
-    attr = m_table->GetAttr(-1, col, wxGridCellAttr::Col );
+    wxGridCellAttr *attr = m_table->GetAttr(-1, col, wxGridCellAttr::Col );
     if(!attr)
diff -bu1 -r wxWindows/src/generic/gridsel.cpp wxWin/src/generic/gridsel.cpp
--- wxWindows/src/generic/gridsel.cpp	2003-08-27 11:55:18.000000000 +0200
+++ wxWin/src/generic/gridsel.cpp	2003-10-06 15:43:43.000000000 +0200
@@ -592,2 +592,5 @@
                 n--; count--;
+            #ifdef __BORLANDC__
+                n; count; // prevent dummy warnings
+            #endif
                 if ( !m_grid->GetBatchCount() )
diff -bu1 -r wxWindows/src/generic/imaglist.cpp wxWin/src/generic/imaglist.cpp
--- wxWindows/src/generic/imaglist.cpp	2003-08-11 10:29:54.000000000 +0200
+++ wxWin/src/generic/imaglist.cpp	2003-10-06 12:02:06.000000000 +0200
@@ -111,13 +111,11 @@
 
-    wxBitmap* newBitmap = NULL;
-    if (bitmap.IsKindOf(CLASSINFO(wxIcon)))
-#if defined(__VISAGECPP__)
-//just can't do this in VisualAge now, with all this new Bitmap-Icon stuff
-//so construct it from a bitmap object until I can figure this nonsense out. (DW)
-        newBitmap = new wxBitmap(bitmap) ;
-#else
-        newBitmap = new wxBitmap( (const wxIcon&) bitmap );
-#endif
-    else
-        newBitmap = new wxBitmap(bitmap) ;
+    wxBitmap* newBitmap = (bitmap.IsKindOf(CLASSINFO(wxIcon))) ?
+                             #if defined(__VISAGECPP__)
+                               //just can't do this in VisualAge now, with all this new Bitmap-Icon stuff
+                               //so construct it from a bitmap object until I can figure this nonsense out. (DW)
+                               new wxBitmap(bitmap)
+                             #else
+                               new wxBitmap( (const wxIcon&) bitmap )
+                             #endif
+                               : new wxBitmap(bitmap) ;
 
diff -bu1 -r wxWindows/src/generic/listctrl.cpp wxWin/src/generic/listctrl.cpp
--- wxWindows/src/generic/listctrl.cpp	2003-09-29 09:08:27.000000000 +0200
+++ wxWin/src/generic/listctrl.cpp	2003-10-07 10:14:05.000000000 +0200
@@ -2684,3 +2684,2 @@
 
-            int col = 0;
             wxRect firstItemRect;
@@ -2692,3 +2691,3 @@
             dc.SetBrush(* wxTRANSPARENT_BRUSH);
-            for (col = 0; col < GetColumnCount(); col++)
+            for (int col = 0; col < GetColumnCount(); col++)
             {
@@ -3215,11 +3214,3 @@
             {
-                int steps = 0;
-                if ( HasFlag(wxLC_REPORT) )
-                {
-                    steps = m_linesPerPage - 1;
-                }
-                else
-                {
-                    steps = m_current % m_linesPerPage;
-                }
+                int steps = ( HasFlag(wxLC_REPORT) ) ? m_linesPerPage - 1 : m_current % m_linesPerPage;
 
@@ -3235,11 +3226,5 @@
             {
-                int steps = 0;
-                if ( HasFlag(wxLC_REPORT) )
-                {
-                    steps = m_linesPerPage - 1;
-                }
-                else
-                {
-                    steps = m_linesPerPage - (m_current % m_linesPerPage) - 1;
-                }
+                int steps = ( HasFlag(wxLC_REPORT) ) ?
+                            m_linesPerPage - 1 :
+                            m_linesPerPage - (m_current % m_linesPerPage) - 1;
 
@@ -4042,3 +4027,2 @@
                         m_linesPerPage = 0;
-                        currentlyVisibleLines = 0;
                         break;
@@ -5349,2 +5333,5 @@
 
+#ifdef __BORLANDC__
+    item; // prevent dummy warnings
+#endif
     // no attributes by default
diff -bu1 -r wxWindows/src/generic/logg.cpp wxWin/src/generic/logg.cpp
--- wxWindows/src/generic/logg.cpp	2003-09-14 21:13:56.000000000 +0200
+++ wxWin/src/generic/logg.cpp	2003-10-06 15:42:53.000000000 +0200
@@ -1114,2 +1114,5 @@
     bool bOk = FALSE;
+#ifdef __BORLANDC__
+    bOk; // prevent dummy warnings
+#endif
     if ( wxFile::Exists(filename) ) {
diff -bu1 -r wxWindows/src/generic/printps.cpp wxWin/src/generic/printps.cpp
--- wxWindows/src/generic/printps.cpp	2003-08-18 08:15:06.000000000 +0200
+++ wxWin/src/generic/printps.cpp	2003-10-06 12:17:07.000000000 +0200
@@ -137,3 +137,3 @@
     // Create a suitable device context
-    wxDC *dc = (wxDC *) NULL;
+    wxDC *dc;
     if (prompt)
diff -bu1 -r wxWindows/src/generic/sashwin.cpp wxWin/src/generic/sashwin.cpp
--- wxWindows/src/generic/sashwin.cpp	2003-08-11 10:29:56.000000000 +0200
+++ wxWin/src/generic/sashwin.cpp	2003-10-06 14:20:49.000000000 +0200
@@ -490,7 +490,3 @@
     {
-        int sashPosition = 0;
-        if (edge == wxSASH_LEFT)
-            sashPosition = 0;
-        else
-            sashPosition = w - GetEdgeMargin(edge);
+        int sashPosition = (edge == wxSASH_LEFT) ? 0 : ( w - GetEdgeMargin(edge) );
 
@@ -520,7 +516,3 @@
     {
-        int sashPosition = 0;
-        if (edge == wxSASH_TOP)
-            sashPosition = 0;
-        else
-            sashPosition = h - GetEdgeMargin(edge);
+        int sashPosition = (edge == wxSASH_TOP) ? 0 : ( h - GetEdgeMargin(edge) );
 
diff -bu1 -r wxWindows/src/generic/splitter.cpp wxWin/src/generic/splitter.cpp
--- wxWindows/src/generic/splitter.cpp	2003-09-24 09:50:37.000000000 +0200
+++ wxWin/src/generic/splitter.cpp	2003-10-06 13:29:41.000000000 +0200
@@ -378,3 +378,3 @@
 
-    bool iconized = FALSE;
+    bool iconized;
 
@@ -722,3 +722,3 @@
 
-    wxWindow *win = NULL;
+    wxWindow *win;
     if ( toRemove == NULL || toRemove == m_windowTwo)
diff -bu1 -r wxWindows/src/generic/treectlg.cpp wxWin/src/generic/treectlg.cpp
--- wxWindows/src/generic/treectlg.cpp	2003-09-16 13:29:09.000000000 +0200
+++ wxWin/src/generic/treectlg.cpp	2003-10-06 12:19:28.000000000 +0200
@@ -1713,17 +1713,8 @@
 {
-    // item2 is not necessary after item1
-    wxGenericTreeItem *first=NULL, *last=NULL;
     m_select_me = NULL;
 
+    // item2 is not necessary after item1
     // choice first' and 'last' between item1 and item2
-    if (item1->GetY()<item2->GetY())
-    {
-        first=item1;
-        last=item2;
-    }
-    else
-    {
-        first=item2;
-        last=item1;
-    }
+    wxGenericTreeItem *first= (item1->GetY()<item2->GetY()) ? item1 : item2;
+    wxGenericTreeItem *last = (item1->GetY()<item2->GetY()) ? item2 : item1;
 
diff -bu1 -r wxWindows/src/generic/vlbox.cpp wxWin/src/generic/vlbox.cpp
--- wxWindows/src/generic/vlbox.cpp	2003-08-11 10:29:58.000000000 +0200
+++ wxWin/src/generic/vlbox.cpp	2003-10-06 15:43:22.000000000 +0200
@@ -489,2 +489,5 @@
     int current = 0; // just to silent the stupid compiler warnings
+#ifdef __BORLANDC__
+    current;         // just to silent the stupid compiler warnings
+#endif
     switch ( event.GetKeyCode() )
diff -bu1 -r wxWindows/src/generic/vscroll.cpp wxWin/src/generic/vscroll.cpp
--- wxWindows/src/generic/vscroll.cpp	2003-08-11 10:29:58.000000000 +0200
+++ wxWin/src/generic/vscroll.cpp	2003-10-06 12:21:29.000000000 +0200
@@ -94,3 +94,3 @@
         heightTotal = (wxCoord)
-                (((float)m_heightTotal / (3*NUM_LINES_TO_SAMPLE)) * m_lineMax);
+                (((float)heightTotal / (3*NUM_LINES_TO_SAMPLE)) * m_lineMax);
     }
diff -bu1 -r wxWindows/src/html/helpfrm.cpp wxWin/src/html/helpfrm.cpp
--- wxWindows/src/html/helpfrm.cpp	2003-10-06 08:25:43.000000000 +0200
+++ wxWin/src/html/helpfrm.cpp	2003-10-07 10:16:30.000000000 +0200
@@ -457,2 +457,5 @@
         m_SearchPage = notebook_page++;
+#ifdef __BORLANDC__
+        notebook_page; // prevent dummy warnings
+#endif
     }
diff -bu1 -r wxWindows/src/html/htmlcell.cpp wxWin/src/html/htmlcell.cpp
--- wxWindows/src/html/htmlcell.cpp	2003-09-19 09:27:00.000000000 +0200
+++ wxWin/src/html/htmlcell.cpp	2003-10-06 15:44:21.000000000 +0200
@@ -755,2 +755,5 @@
                 xcnt++;
+#ifdef __BORLANDC__
+                xcnt; // prevent dummy warnings
+#endif
             }
@@ -970,7 +973,5 @@
     {
-        const wxHtmlCell *r = NULL;
-
         for (wxHtmlCell *cell = m_Cells; cell; cell = cell->GetNext())
         {
-            r = cell->Find(condition, param);
+            const wxHtmlCell *r = cell->Find(condition, param);
             if (r) return r;
diff -bu1 -r wxWindows/src/html/htmltag.cpp wxWin/src/html/htmltag.cpp
--- wxWindows/src/html/htmltag.cpp	2003-08-11 10:30:02.000000000 +0200
+++ wxWin/src/html/htmltag.cpp	2003-10-06 15:44:47.000000000 +0200
@@ -152,2 +152,5 @@
                             stpos = pos;
+                        #ifdef __BORLANDC__
+                            stpos; // prevent dummy warnings
+                        #endif
                             break;
diff -bu1 -r wxWindows/src/html/htmprint.cpp wxWin/src/html/htmprint.cpp
--- wxWindows/src/html/htmprint.cpp	2003-08-11 10:30:02.000000000 +0200
+++ wxWin/src/html/htmprint.cpp	2003-10-06 15:45:09.000000000 +0200
@@ -383,4 +383,10 @@
     GetPPIPrinter(&ppiPrinterX, &ppiPrinterY);
+#ifdef __BORLANDC__
+    ppiPrinterX; // prevent dummy warnings
+#endif
     int ppiScreenX, ppiScreenY;
     GetPPIScreen(&ppiScreenX, &ppiScreenY);
+#ifdef __BORLANDC__
+    ppiScreenX; // prevent dummy warnings
+#endif
 
diff -bu1 -r wxWindows/src/html/m_pre.cpp wxWin/src/html/m_pre.cpp
--- wxWindows/src/html/m_pre.cpp	2003-08-11 10:30:04.000000000 +0200
+++ wxWin/src/html/m_pre.cpp	2003-10-06 15:46:10.000000000 +0200
@@ -37,4 +37,4 @@
     wxString out;
-    size_t i = 0, j = 0, len = str.Len();
-    for (i = 0; i < len; i++)
+    size_t len = str.Len();
+    for (size_t i = 0; i < len; i++)
     {
@@ -54,3 +54,5 @@
             case wxT('\t'):
-                for (j = 8 - i%8; j > 0; j--) out << wxT("&nbsp;");
+                {
+                    for (size_t j = 8 - i%8; j > 0; j--) out << wxT("&nbsp;");
+                }
                 break;
diff -bu1 -r wxWindows/src/msw/bitmap.cpp wxWin/src/msw/bitmap.cpp
--- wxWindows/src/msw/bitmap.cpp	2003-09-14 08:00:37.000000000 +0200
+++ wxWin/src/msw/bitmap.cpp	2003-10-06 14:31:30.000000000 +0200
@@ -1764,5 +1764,2 @@
    unsigned long   i, headerSize;
-   LPBITMAPINFO    lpDIBheader = NULL;
-   LPPALETTEENTRY  lpPe = NULL;
-
 
@@ -1770,4 +1767,4 @@
    headerSize = (sizeof(BITMAPINFOHEADER) + (256 * sizeof(PALETTEENTRY)));
-   lpDIBheader = (BITMAPINFO *) malloc(headerSize);
-   lpPe = (PALETTEENTRY *)((BYTE*)lpDIBheader + sizeof(BITMAPINFOHEADER));
+   LPBITMAPINFO lpDIBheader = (BITMAPINFO *) malloc(headerSize);
+   LPPALETTEENTRY lpPe = (PALETTEENTRY *)((BYTE*)lpDIBheader + sizeof(BITMAPINFOHEADER));
 
diff -bu1 -r wxWindows/src/msw/dc.cpp wxWin/src/msw/dc.cpp
--- wxWindows/src/msw/dc.cpp	2003-09-15 13:24:28.000000000 +0200
+++ wxWin/src/msw/dc.cpp	2003-10-06 13:21:44.000000000 +0200
@@ -880,7 +880,3 @@
     {
-        double smallest = 0.0;
-        if (width < height)
-            smallest = width;
-        else
-            smallest = height;
+        double smallest = (width < height) ? width : height;
         radius = (- radius * smallest);
@@ -1434,4 +1430,3 @@
         {
-            HBRUSH b = 0;
-            b = (HBRUSH) ::SelectObject(GetHdc(), (HBRUSH)m_brush.GetResourceHandle());
+            HBRUSH b = (HBRUSH) ::SelectObject(GetHdc(), (HBRUSH)m_brush.GetResourceHandle());
             if (!m_oldBrush)
@@ -1871,3 +1866,3 @@
 
-    DWORD dwRop = SRCCOPY;
+    DWORD dwRop;
     switch (rop)
diff -bu1 -r wxWindows/src/msw/dcclient.cpp wxWin/src/msw/dcclient.cpp
--- wxWindows/src/msw/dcclient.cpp	2003-08-11 10:30:10.000000000 +0200
+++ wxWin/src/msw/dcclient.cpp	2003-10-06 11:52:03.000000000 +0200
@@ -295,3 +295,2 @@
 {
-    wxPaintDCInfo *info = NULL;
     size_t nCache = ms_cache.GetCount();
@@ -299,3 +298,3 @@
     {
-        info = &ms_cache[n];
+        wxPaintDCInfo *info = &ms_cache[n];
         if ( info->hwnd == win->GetHWND() )
diff -bu1 -r wxWindows/src/msw/dcprint.cpp wxWin/src/msw/dcprint.cpp
--- wxWindows/src/msw/dcprint.cpp	2003-08-14 16:38:32.000000000 +0200
+++ wxWin/src/msw/dcprint.cpp	2003-10-06 13:23:30.000000000 +0200
@@ -347,3 +347,4 @@
     wxDIB dib(bmp);
-    if ( !dib.IsOk() )
+    bool ok = dib.IsOk();
+    if ( !ok )
         return FALSE;
@@ -376,3 +377,3 @@
     }
-
+    else
     return TRUE;
diff -bu1 -r wxWindows/src/msw/gdiimage.cpp wxWin/src/msw/gdiimage.cpp
--- wxWindows/src/msw/gdiimage.cpp	2003-10-01 11:15:09.000000000 +0200
+++ wxWin/src/msw/gdiimage.cpp	2003-10-06 13:23:47.000000000 +0200
@@ -374,3 +374,4 @@
 
-    return dib.IsOk() && bitmap->CopyFromDIB(dib);
+    bool ok = dib.IsOk() && bitmap->CopyFromDIB(dib);
+    return ok;
 #else
diff -bu1 -r wxWindows/src/msw/mimetype.cpp wxWin/src/msw/mimetype.cpp
--- wxWindows/src/msw/mimetype.cpp	2003-07-21 11:19:40.000000000 +0200
+++ wxWin/src/msw/mimetype.cpp	2003-10-06 14:41:51.000000000 +0200
@@ -516,3 +516,3 @@
 
-    bool ok = FALSE ;
+    bool ok;
     int iExtCount = 0 ;
@@ -553,3 +553,3 @@
 
-            ok = key.SetValue(wxEmptyString, filetype);
+            key.SetValue(wxEmptyString, filetype);
         }
@@ -562,3 +562,3 @@
                     filetype = filetypeOrig;
-                    ok = key.SetValue(wxEmptyString, filetype);
+                    key.SetValue(wxEmptyString, filetype);
                 }
@@ -566,3 +566,3 @@
                 {
-                    ok = key.QueryValue(wxEmptyString, filetype);
+                    key.QueryValue(wxEmptyString, filetype);
                 }
@@ -587,3 +587,3 @@
                     // and provide a back link to the extension
-                    ok = keyMIME.SetValue(_T("Extension"), extWithDot);
+                    keyMIME.SetValue(_T("Extension"), extWithDot);
                 }
@@ -603,4 +603,4 @@
             wxRegKey key(wxRegKey::HKCR, extWithDot);
-            if ( !key.Exists() ) ok = key.Create();
-            ok = key.SetValue(wxEmptyString, filetype);
+            if ( !key.Exists() ) key.Create();
+            key.SetValue(wxEmptyString, filetype);
 
@@ -624,3 +624,3 @@
                     // and provide a back link to the extension
-                    ok = keyMIME.SetValue(_T("Extension"), extWithDot);
+                    keyMIME.SetValue(_T("Extension"), extWithDot);
         }
@@ -635,6 +635,5 @@
     wxRegKey keyFT(wxRegKey::HKCR, filetype);
-    ok = keyFT.Create();
+    keyFT.Create();
 
-    wxFileType *ft = NULL;
-    ft = CreateFileType(filetype, extWithDot);
+    wxFileType *ft = CreateFileType(filetype, extWithDot);
 
diff -bu1 -r wxWindows/src/msw/pen.cpp wxWin/src/msw/pen.cpp
--- wxWindows/src/msw/pen.cpp	2003-08-11 10:30:12.000000000 +0200
+++ wxWin/src/msw/pen.cpp	2003-10-06 11:54:09.000000000 +0200
@@ -144,4 +144,3 @@
 
-       COLORREF ms_colour = 0;
-       ms_colour = M_PENDATA->m_colour.GetPixel();
+       COLORREF ms_colour = M_PENDATA->m_colour.GetPixel();
 
diff -bu1 -r wxWindows/src/msw/taskbar.cpp wxWin/src/msw/taskbar.cpp
--- wxWindows/src/msw/taskbar.cpp	2003-08-11 10:30:14.000000000 +0200
+++ wxWin/src/msw/taskbar.cpp	2003-10-06 12:24:30.000000000 +0200
@@ -186,3 +186,2 @@
 
-    bool        rval = false;
     wxWindow*   win;
@@ -204,3 +203,3 @@
 
-    rval = win->PopupMenu(menu, 0, 0);
+    bool rval = win->PopupMenu(menu, 0, 0);
 
diff -bu1 -r wxWindows/src/msw/thread.cpp wxWin/src/msw/thread.cpp
--- wxWindows/src/msw/thread.cpp	2003-09-22 08:00:00.000000000 +0200
+++ wxWin/src/msw/thread.cpp	2003-10-06 15:37:16.000000000 +0200
@@ -693,2 +693,5 @@
         DWORD result = 0;       // suppress warnings from broken compilers
+#ifdef __BORLANDC__
+        result;                 // suppress warnings from broken compilers
+#endif
         do
diff -bu1 -r wxWindows/src/msw/toplevel.cpp wxWin/src/msw/toplevel.cpp
--- wxWindows/src/msw/toplevel.cpp	2003-09-26 21:16:41.000000000 +0200
+++ wxWin/src/msw/toplevel.cpp	2003-10-06 15:39:50.000000000 +0200
@@ -448,2 +448,5 @@
     bool ret = false;
+#ifdef __BORLANDC__
+    ret; // prevent dummy warnings
+#endif
 
diff -bu1 -r wxWindows/src/msw/utils.cpp wxWin/src/msw/utils.cpp
--- wxWindows/src/msw/utils.cpp	2003-10-06 08:25:43.000000000 +0200
+++ wxWin/src/msw/utils.cpp	2003-10-06 11:41:18.000000000 +0200
@@ -924,3 +924,3 @@
 
-        bOK = ::ExitWindowsEx(EWX_SHUTDOWN | EWX_FORCE | EWX_REBOOT, 0) != 0;
+        bOK = ::ExitWindowsEx(flags, 0) != 0;
     }
diff -bu1 -r wxWindows/src/msw/utilsexc.cpp wxWin/src/msw/utilsexc.cpp
--- wxWindows/src/msw/utilsexc.cpp	2003-10-06 08:25:47.000000000 +0200
+++ wxWin/src/msw/utilsexc.cpp	2003-10-06 15:37:36.000000000 +0200
@@ -476,2 +476,5 @@
     bool ok = FALSE;
+#ifdef __BORLANDC__
+    ok; // prevent dummy warnings
+#endif
 
diff -bu1 -r wxWindows/src/msw/volume.cpp wxWin/src/msw/volume.cpp
--- wxWindows/src/msw/volume.cpp	2003-08-11 10:30:16.000000000 +0200
+++ wxWin/src/msw/volume.cpp	2003-10-06 15:37:54.000000000 +0200
@@ -407,2 +407,5 @@
     chars = GetLogicalDriveStrings(chars, buf);
+#ifdef __BORLANDC__
+    chars; // prevent dummy warnings
+#endif
 
diff -bu1 -r wxWindows/src/msw/window.cpp wxWin/src/msw/window.cpp
--- wxWindows/src/msw/window.cpp	2003-09-24 09:50:49.000000000 +0200
+++ wxWin/src/msw/window.cpp	2003-10-06 13:24:01.000000000 +0200
@@ -4916,3 +4916,3 @@
     *isVirtual = TRUE;
-    int keySym = 0;
+    int keySym;
     switch (id)
diff -bu1 -r wxWindows/src/univ/textctrl.cpp wxWin/src/univ/textctrl.cpp
--- wxWindows/src/univ/textctrl.cpp	2003-09-23 07:30:16.000000000 +0200
+++ wxWin/src/univ/textctrl.cpp	2003-10-06 12:03:35.000000000 +0200
@@ -2347,3 +2347,3 @@
             lines = 10;
-        h *= 10;
+        h *= lines;
     }
diff -bu1 -r wxWindows/src/univ/themes/win32.cpp wxWin/src/univ/themes/win32.cpp
--- wxWindows/src/univ/themes/win32.cpp	2003-08-04 08:18:46.000000000 +0200
+++ wxWin/src/univ/themes/win32.cpp	2003-10-07 10:14:25.000000000 +0200
@@ -3601,2 +3601,5 @@
             x -= FRAME_BUTTON_WIDTH;
+#ifdef __BORLANDC__
+            x; // prevent dummy warnings
+#endif
         }
diff -bu1 -r wxWindows/src/univ/topluniv.cpp wxWin/src/univ/topluniv.cpp
--- wxWindows/src/univ/topluniv.cpp	2003-09-18 18:23:07.000000000 +0200
+++ wxWin/src/univ/topluniv.cpp	2003-10-06 12:04:07.000000000 +0200
@@ -103,3 +103,3 @@
                    wxTHICK_FRAME);
-        style = wxSIMPLE_BORDER;
+        style |= wxSIMPLE_BORDER;
         SetExtraStyle(exstyleOrig &

 	  	 
