Index: wx/lib/splitter.py
===================================================================
--- wx/lib/splitter.py	(revision 64164)
+++ wx/lib/splitter.py	(working copy)
@@ -57,7 +57,6 @@
     def __init__(self, parent, id=-1,
                  pos = wx.DefaultPosition, size = wx.DefaultSize,
                  style = 0, name="multiSplitter"):
-        
         # always turn on tab traversal
         style |= wx.TAB_TRAVERSAL
 
@@ -111,7 +110,10 @@
         """
         return self._orient
 
-
+    def SetBackgroundColour(self,color):
+        wx.PyPanel.SetBackgroundColour(self,color)
+        self._drawSashInBackgroundColour = True
+        
     def SetMinimumPaneSize(self, minSize):
         """
         Set the smallest size that any pane will be allowed to be
@@ -588,7 +590,7 @@
         pos = 0
         for sash in self._sashes[:-1]:
             pos += sash
-            if wx.VERSION >= _RENDER_VER:
+            if wx.VERSION >= _RENDER_VER and not self._drawSashInBackgroundColour:
                 wx.RendererNative.Get().DrawSplitterSash(self, dc,
                                                          self.GetClientSize(),
                                                          pos, orient, flag)
