Index: tests/fswatcher/fswatchertest.cpp
===================================================================
--- tests/fswatcher/fswatchertest.cpp	(revision 75982)
+++ tests/fswatcher/fswatchertest.cpp	(working copy)
@@ -20,6 +20,8 @@
     #include "wx/timer.h"
 #endif
 
+#if wxUSE_FSWATCHER
+
 #include "wx/evtloop.h"
 #include "wx/filename.h"
 #include "wx/filefn.h"
@@ -1008,3 +1010,5 @@
     NoEventsAfterRemoveEventTester tester;
     tester.Run();
 }
+
+#endif // wxUSE_FSWATCHER
