Index: src/common/docview.cpp
===================================================================
--- src/common/docview.cpp	(revision 61765)
+++ src/common/docview.cpp	(working copy)
@@ -981,7 +981,7 @@
 wxString wxDocManager::GetLastDirectory() const
 {
     // if we haven't determined the last used directory yet, do it now
-    if ( m_lastDirectory.empty() )
+    if ( m_lastDirectory.empty() && GetHistoryFilesCount())
     {
         // we're going to modify m_lastDirectory in this const method, so do it
         // via non-const self pointer instead of const this one
