Index: src/gtk/statbox.cpp
===================================================================
--- src/gtk/statbox.cpp	(revisione 7300)
+++ src/gtk/statbox.cpp	(copia locale)
@@ -44,7 +44,15 @@
         gtk_widget_size_allocate(label_widget, &a);
     }
 }
+
+static gboolean expose_event(GtkWidget* widget, GdkEventExpose*, wxWindow*)
+{
+    const GtkAllocation& a = widget->allocation;
+    gtk_paint_flat_box(gtk_widget_get_style(widget), gtk_widget_get_window(widget),
+        GTK_STATE_NORMAL, GTK_SHADOW_NONE, NULL, widget, "", a.x, a.y, a.width, a.height);
+    return false;
 }
+}
 #endif
 
 //-----------------------------------------------------------------------------
@@ -122,6 +130,7 @@
         m_wxwindow = wxPizza::New();
         gtk_widget_show( m_wxwindow );
         gtk_container_add( GTK_CONTAINER (m_widget), m_wxwindow );
+        GTKApplyWidgetStyle(false);
     }
 
     wxStaticBoxBase::AddChild(child);
@@ -137,6 +146,14 @@
 void wxStaticBox::DoApplyWidgetStyle(GtkRcStyle *style)
 {
     GTKFrameApplyWidgetStyle(GTK_FRAME(m_widget), style);
+    if (m_wxwindow)
+        GTKApplyStyle(m_wxwindow, style);
+
+#ifndef __WXGTK3__
+    g_signal_handlers_disconnect_by_func(m_widget, (void*)expose_event, this);
+    if (m_backgroundColour.IsOk())
+        g_signal_connect(m_widget, "expose-event", G_CALLBACK(expose_event), this);
+#endif
 }
 
 bool wxStaticBox::GTKWidgetNeedsMnemonic() const
