Index: src/osx/cocoa/dataview.mm
===================================================================
--- src/osx/cocoa/dataview.mm	(revision 69792)
+++ src/osx/cocoa/dataview.mm	(working copy)
@@ -664,7 +664,7 @@
 
     wxCHECK_MSG( model, nil, "Valid model in data source does not exist." );
 
-    wxDataViewColumn* col(static_cast<wxDataViewColumn*>([[tableColumn identifier] pointer]));
+    wxDataViewColumn* col( (wxDataViewColumn*)[tableColumn identifier] );
     const unsigned colIdx = col->GetModelColumn();
 
     wxDataViewItem dataViewItem(wxDataViewItemFromItem(item));
@@ -687,7 +687,7 @@
 {
     wxUnusedVar(outlineView);
 
-    wxDataViewColumn* col(static_cast<wxDataViewColumn*>([[tableColumn identifier] pointer]));
+    wxDataViewColumn* col( (wxDataViewColumn*)[tableColumn identifier] );
 
     col->GetRenderer()->
         OSXOnCellChanged(object, wxDataViewItemFromItem(item), col->GetModelColumn());
@@ -1624,7 +1624,7 @@
 //
 -(void) outlineView:(NSOutlineView*)outlineView mouseDownInHeaderOfTableColumn:(NSTableColumn*)tableColumn
 {
-    wxDataViewColumn* const col(static_cast<wxDataViewColumn*>([[tableColumn identifier] pointer]));
+    wxDataViewColumn* const col( (wxDataViewColumn*)[tableColumn identifier] );
 
     wxDataViewCtrl* const dvc = implementation->GetDataViewCtrl();
 
@@ -1720,10 +1720,8 @@
     wxDataViewModel * const model = dvc->GetModel();
 
     wxDataViewColumn * const
-        dvCol(static_cast<wxDataViewColumn*>(
-                    [[tableColumn identifier] pointer]
-                    )
-             );
+        dvCol( (wxDataViewColumn*)[tableColumn identifier] );
+    
     const unsigned colIdx = dvCol->GetModelColumn();
 
     wxDataViewItem dvItem(wxDataViewItemFromItem(item));
@@ -1760,7 +1758,7 @@
 {
     int const newColumnPosition = [[[notification userInfo] objectForKey:@"NSNewColumn"] intValue];
 
-    wxDataViewColumn* const col(static_cast<wxDataViewColumn*>([[[[self tableColumns] objectAtIndex:newColumnPosition] identifier] pointer]));
+    wxDataViewColumn* const col( (wxDataViewColumn*)[[[self tableColumns] objectAtIndex:newColumnPosition] identifier] );
 
     wxDataViewCtrl* const dvc = implementation->GetDataViewCtrl();
 
@@ -1827,9 +1825,7 @@
     currentlyEditedColumn = [self editedColumn];
     currentlyEditedRow = [self editedRow];
 
-    wxDataViewColumn* const col =
-        static_cast<wxDataViewColumn*>(
-                [[[[self tableColumns] objectAtIndex:currentlyEditedColumn] identifier] pointer]);
+    wxDataViewColumn* const col = ( wxDataViewColumn*)[[[self tableColumns] objectAtIndex:currentlyEditedColumn] identifier];
 
     wxDataViewCtrl* const dvc = implementation->GetDataViewCtrl();
 
@@ -1866,8 +1862,7 @@
     if ( currentlyEditedColumn != -1 && currentlyEditedRow != -1 )
     {
         wxDataViewColumn* const col =
-            static_cast<wxDataViewColumn*>(
-                    [[[[self tableColumns] objectAtIndex:currentlyEditedColumn] identifier] pointer]);
+            ( wxDataViewColumn*)[[[self tableColumns] objectAtIndex:currentlyEditedColumn] identifier];
 
         wxDataViewCtrl* const dvc = implementation->GetDataViewCtrl();
 
@@ -1979,7 +1974,7 @@
 
 wxDataViewColumn* wxCocoaDataViewControl::GetColumn(unsigned int pos) const
 {
-    return static_cast<wxDataViewColumn*>([[[[m_OutlineView tableColumns] objectAtIndex:pos] identifier] pointer]);
+    return (wxDataViewColumn*)[[[m_OutlineView tableColumns] objectAtIndex:pos] identifier];
 }
 
 int wxCocoaDataViewControl::GetColumnPosition(const wxDataViewColumn *columnPtr) const
@@ -2328,7 +2323,7 @@
 
     for (UInt32 i=0; i<noOfColumns; ++i)
         if ([[columns objectAtIndex:i] sortDescriptorPrototype] != nil)
-            return static_cast<wxDataViewColumn*>([[[columns objectAtIndex:i] identifier] pointer]);
+            (wxDataViewColumn*)[[columns objectAtIndex:i] identifier];
     return NULL;
 }
 
@@ -2363,7 +2358,7 @@
     indexRow    = [m_OutlineView rowAtPoint:   nativePoint];
     if ((indexColumn >= 0) && (indexRow >= 0))
     {
-        columnPtr = static_cast<wxDataViewColumn*>([[[[m_OutlineView tableColumns] objectAtIndex:indexColumn] identifier] pointer]);
+        columnPtr = (wxDataViewColumn*)[[[m_OutlineView tableColumns] objectAtIndex:indexColumn] identifier];
         item      = wxDataViewItem([[m_OutlineView itemAtRow:indexRow] pointer]);
     }
     else