diff --git src/xrc/xh_sizer.cpp src/xrc/xh_sizer.cpp
index 9e3d23d..87b64ae 100644
--- src/xrc/xh_sizer.cpp
+++ src/xrc/xh_sizer.cpp
@@ -256,7 +256,16 @@ wxObject* wxSizerXmlHandler::Handle_sizer()
     m_isInside = true;
     m_isGBS = (m_class == wxT("wxGridBagSizer"));
 
-    CreateChildren(m_parent, true/*only this handler*/);
+    wxObject* parent = m_parent;
+#if wxUSE_STATBOX
+    // Since wx2.9.1 a wxStaticBoxSizer's child controls should be parented by the box
+    if (wxStaticBoxSizer* stsizer = wxDynamicCast(sizer, wxStaticBoxSizer))
+    {
+        parent = stsizer->GetStaticBox();
+    }
+#endif // wxUSE_STATBOX
+
+    CreateChildren(parent, true/*only this handler*/);
 
     // set growable rows and cols for sizers which support this
     if ( wxFlexGridSizer *flexsizer = wxDynamicCast(sizer, wxFlexGridSizer) )
