Index: samples/splitter/splitter.cpp
===================================================================
--- samples/splitter/splitter.cpp	(revision 65008)
+++ samples/splitter/splitter.cpp	(working copy)
@@ -463,7 +463,7 @@
 MySplitterWindow::MySplitterWindow(wxFrame *parent)
                 : wxSplitterWindow(parent, wxID_ANY,
                                    wxDefaultPosition, wxDefaultSize,
-                                   wxSP_3D | wxSP_LIVE_UPDATE |
+                                   wxSP_3D | wxSP_LIVE_UPDATE | wxBORDER_SUNKEN |
                                    wxCLIP_CHILDREN /* | wxSP_NO_XP_THEME */ )
 {
     m_frame = parent;
