Index: m_layout.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/html/m_layout.cpp,v
retrieving revision 1.46
diff -u -r1.46 m_layout.cpp
--- m_layout.cpp	2006/09/02 23:36:06	1.46
+++ m_layout.cpp	2007/03/25 16:23:28
@@ -352,6 +352,8 @@
                         winIface->SetHTMLBackgroundImage(image);
 #endif
                 }
+		
+		delete fileBgImage;
             }
         }
 

 	  	 
