--- wxGTK-2.4.2/src/common/url.cpp	2003-09-21 13:31:39.000000000 +0200
+++ wxGTK-2.4.2/src/common/url.cpp.patched	2003-10-01 13:33:33.000000000 +0200
@@ -413,13 +413,19 @@
 
 wxString wxURL::ConvertToValidURI(const wxString& uri, const wxChar* delims)
 {
+  //problem with uri is that it might already be
+  //a valid URI (partially or fully)
+  //so we replace all appearances of %XX before converting them again
+  //if we do not do this, the URI gets scrambled e.g. else
+  //  "%20" (encoded space) would become "%2520"
+  wxString in_str = ConvertFromURI(uri);
   wxString out_str;
   wxString hexa_code;
   size_t i;
 
-  for (i = 0; i < uri.Len(); i++)
+  for (i = 0; i < in_str.Len(); i++)
   {
-    wxChar c = uri.GetChar(i);
+    wxChar c = in_str.GetChar(i);
 
     if (c == wxT(' '))
     {

 	  	 
