Index: include/wx/calctrl.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/calctrl.h,v
retrieving revision 1.17
diff -u -r1.17 calctrl.h
--- calctrl.h	2003/08/18 09:30:45	1.17
+++ calctrl.h	2003/10/21 11:51:06
@@ -169,6 +169,8 @@
     wxCalendarEvent(wxCalendarCtrl *cal, wxEventType type);

     const wxDateTime& GetDate() const { return m_date; }
+    void SetDate(const wxDateTime &date) { m_date = date; }
+    void SetWeekDay(const wxDateTime::WeekDay wd) { m_wday = wd; }
     wxDateTime::WeekDay GetWeekDay() const { return m_wday; }
 
 protected:

 	  	 
