Index: utils.cpp
===================================================================
RCS file: /export/cvsroot/org/wxwindows/wxWindows/src/msw/utils.cpp,v
retrieving revision 1.1.1.6
retrieving revision 1.3
diff -b -u -2 -r1.1.1.6 -r1.3
--- utils.cpp	2002/05/09 22:31:44	1.1.1.6
+++ utils.cpp	2002/06/05 03:55:46	1.3
@@ -45,7 +45,12 @@
 //
 #if defined(__WIN32__) && !defined(__TWIN32__) && !defined(__WXMICROWIN__) && ! (defined(__GNUWIN32__) && !defined(__MINGW32__))
+    // when targetting new api versions, including winsock.h directly causes problems.
+    #if(_WIN32_WINNT >= 0x0400)
+        #include <windows.h>
+    #else
 extern "C" {
     #include <winsock.h>    // we use socket functions in wxGetFullHostName()
 }
+    #endif /* _WIN32_WINNT >=  0x0400 */
 #endif
 

 	  	 
