Index: src/gtk/menu.cpp
===================================================================
--- src/gtk/menu.cpp	(revision 76109)
+++ src/gtk/menu.cpp	(working copy)
@@ -86,6 +86,33 @@
     win->HandleWindowEvent( event );
 }
 
+static wxMenu* GetRootParentMenu(wxMenu* menu)
+{
+    if ( menu->GetParent() == NULL )
+        return menu;
+    return GetRootParentMenu( menu->GetParent() );
+}
+
+static void UpdateSubMenu(wxMenuItem* menu_item)
+{
+    wxMenu* menu = menu_item->GetSubMenu();
+    if ( menu != NULL )
+    {
+        wxMenuItemList::compatibility_iterator node =
+            menu->GetMenuItems().GetFirst();
+        while ( node )
+        {
+            wxMenuItem* item = (wxMenuItem*) node->GetData();
+            if ( !item->IsSeparator() )
+            {
+                item->SetGtkLabel();
+                UpdateSubMenu(item);
+            }
+            node = node->GetNext();
+        }
+    }
+}
+
 //-----------------------------------------------------------------------------
 // wxMenuBar
 //-----------------------------------------------------------------------------
@@ -678,8 +705,9 @@
     wxGetGtkAccel(this, &accel_key, &accel_mods);
     if (accel_key)
     {
+        wxMenu* root_parent_menu = GetRootParentMenu(m_parentMenu);
         gtk_widget_add_accelerator(
-            m_menuItem, "activate", m_parentMenu->m_accel,
+            m_menuItem, "activate", root_parent_menu->m_accel,
             accel_key, accel_mods, GTK_ACCEL_VISIBLE);
     }
 #endif // wxUSE_ACCEL
@@ -931,6 +959,7 @@
     if ( !mitem->IsSeparator() )
     {
         mitem->SetGtkLabel();
+        UpdateSubMenu(mitem);
         g_signal_connect (menuItem, "select",
                           G_CALLBACK(menuitem_select), mitem);
         g_signal_connect (menuItem, "deselect",
