--- wxBase-2.4.0/src/common/file.cpp	Sun Jan  5 12:53:11 2003
+++ wxBase-2.4.0/src/common/file.cpp.new	Thu Jan  9 22:26:04 2003
@@ -293,16 +293,16 @@
     wxCHECK( (pBuf != NULL) && IsOpened(), 0 );
 
 #ifdef __MWERKS__
-    int iRc = ::read(m_fd, (char*) pBuf, nCount);
+    size_t iRc = ::read(m_fd, (char*) pBuf, nCount);
 #else
-    int iRc = ::read(m_fd, pBuf, nCount);
+    size_t iRc = ::read(m_fd, pBuf, nCount);
 #endif
     if ( iRc == -1 ) {
         wxLogSysError(_("can't read from file descriptor %d"), m_fd);
         return wxInvalidOffset;
     }
     else
-        return (size_t)iRc;
+        return (off_t)iRc;
 }
 
 // write
@@ -312,12 +312,12 @@
 
 #ifdef __MWERKS__
 #if __MSL__ >= 0x6000
-    int iRc = ::write(m_fd, (void*) pBuf, nCount);
+    size_t iRc = ::write(m_fd, (void*) pBuf, nCount);
 #else
-    int iRc = ::write(m_fd, (const char*) pBuf, nCount);
+    size_t iRc = ::write(m_fd, (const char*) pBuf, nCount);
 #endif
 #else
-    int iRc = ::write(m_fd, pBuf, nCount);
+    size_t iRc = ::write(m_fd, pBuf, nCount);
 #endif
     if ( iRc == -1 ) {
         wxLogSysError(_("can't write to file descriptor %d"), m_fd);
@@ -373,13 +373,13 @@
             break;
     }
 
-    int iRc = lseek(m_fd, ofs, origin);
+    off_t iRc = lseek(m_fd, ofs, origin);
     if ( iRc == -1 ) {
         wxLogSysError(_("can't seek on file descriptor %d"), m_fd);
         return wxInvalidOffset;
     }
     else
-        return (off_t)iRc;
+        return iRc;
 }
 
 // get current off_t
@@ -387,13 +387,13 @@
 {
     wxASSERT( IsOpened() );
 
-    int iRc = wxTell(m_fd);
+    off_t iRc = wxTell(m_fd);
     if ( iRc == -1 ) {
         wxLogSysError(_("can't get seek position on file descriptor %d"), m_fd);
         return wxInvalidOffset;
     }
     else
-        return (off_t)iRc;
+        return iRc;
 }
 
 // get current file length
@@ -402,12 +402,12 @@
     wxASSERT( IsOpened() );
 
 #ifdef __VISUALC__
-    int iRc = _filelength(m_fd);
+    off_t iRc = _filelength(m_fd);
 #else // !VC++
-    int iRc = wxTell(m_fd);
+    off_t iRc = wxTell(m_fd);
     if ( iRc != -1 ) {
         // @ have to use const_cast :-(
-        int iLen = ((wxFile *)this)->SeekEnd();
+        off_t iLen = ((wxFile *)this)->SeekEnd();
         if ( iLen != -1 ) {
             // restore old position
             if ( ((wxFile *)this)->Seek(iRc) == -1 ) {
@@ -425,7 +425,7 @@
         return wxInvalidOffset;
     }
     else
-        return (off_t)iRc;
+        return iRc;
 }
 
 // is end of file reached?
@@ -433,7 +433,7 @@
 {
     wxASSERT( IsOpened() );
 
-    int iRc;
+    off_t iRc;
 
 #if defined(__DOS__) || defined(__UNIX__) || defined(__GNUWIN32__) || defined( __MWERKS__ ) || defined(__SALFORDC__)
     // @@ this doesn't work, of course, on unseekable file descriptors

 	  	 
