diff --git a/samples/sockets/baseserver.cpp b/samples/sockets/baseserver.cpp
index ac856e8..ecefd94 100644
--- a/samples/sockets/baseserver.cpp
+++ b/samples/sockets/baseserver.cpp
@@ -258,7 +258,7 @@ Server::DumpStatistics()
 
     if ((int)(m_threadWorkersDone+m_eventWorkersDone) == m_maxConnections)
     {
-        wxLogMessage("%d connection(s) served, exiting",m_maxConnections);
+        wxLogMessage("%ld connection(s) served, exiting",m_maxConnections);
         ExitMainLoop();
     }
 }
@@ -279,12 +279,12 @@ Server::OnCmdLineParsed(wxCmdLineParser& pParser)
 
     if (pParser.Found("m",&m_maxConnections))
     {
-        wxLogMessage("%d connection(s) to exit",m_maxConnections);
+        wxLogMessage("%ld connection(s) to exit",m_maxConnections);
     }
 
     if (pParser.Found("p",&m_port))
     {
-        wxLogMessage("%d connection(s) to exit",m_maxConnections);
+        wxLogMessage("%ld connection(s) to exit",m_maxConnections);
     }
 
     if (pParser.Found("t"))
@@ -331,7 +331,7 @@ bool Server::OnInit()
     m_eventWorkersFailed = 0;
     m_maxEventWorkers = 0;
 
-    wxLogMessage("Server listening at port %d, waiting for connections", m_port);
+    wxLogMessage("Server listening at port %ld, waiting for connections", m_port);
     return true;
 }
 
@@ -425,7 +425,7 @@ void  Server::OnWorkerEvent(WorkerEvent& pEvent)
     {
         if (it->GetData() == pEvent.m_sender)
         {
-            wxLogVerbose("Deleting thread worker (%d left)",
+            wxLogVerbose("Deleting thread worker (%zd left)",
                          m_threadWorkers.GetCount());
             it->GetData()->Wait();
             delete it->GetData();
@@ -441,7 +441,7 @@ void  Server::OnWorkerEvent(WorkerEvent& pEvent)
     {
         if (it2->GetData() == pEvent.m_sender)
         {
-            wxLogVerbose("Deleting event worker (%d left)",
+            wxLogVerbose("Deleting event worker (%zd left)",
                          m_eventWorkers.GetCount());
             delete it2->GetData();
             m_eventWorkers.DeleteNode(it2);
diff --git a/samples/sockets/client.cpp b/samples/sockets/client.cpp
index 30d0fa4..10d7c18 100644
--- a/samples/sockets/client.cpp
+++ b/samples/sockets/client.cpp
@@ -612,7 +612,7 @@ void MyFrame::OnTestURL(wxCommandEvent& WXUNUSED(event))
     }
 
     // Print the contents type and file size
-    wxLogMessage("Contents type: %s\nFile size: %i\nStarting to download...",
+    wxLogMessage("Contents type: %s\nFile size: %zi\nStarting to download...",
                  url.GetProtocol().GetContentType(),
                  data->GetSize());
 
diff --git a/samples/typetest/typetest.cpp b/samples/typetest/typetest.cpp
index 69e074b..d6b3bb2 100644
--- a/samples/typetest/typetest.cpp
+++ b/samples/typetest/typetest.cpp
@@ -892,10 +892,10 @@ void MyApp::DoUnicodeDemo(wxCommandEvent& WXUNUSED(event))
 
     printf( "\n\nConversion with wxConvLocal:\n" );
     wxConvCurrent = &wxConvLocal;
-    printf( (const char*) str.mbc_str() );
+    printf( "%s", (const char*) str.mbc_str() );
     printf( "\n\nConversion with wxConvLibc:\n" );
     wxConvCurrent = &wxConvLibc;
-    printf( (const char*) str.mbc_str() );
+    printf( "%s", (const char*) str.mbc_str() );
 
 }
 #endif
