diff -uNr wxWidgets-2.9.1 orig/samples/opengl/cube/cube.cpp wxWidgets-2.9.1/samples/opengl/cube/cube.cpp
--- wxWidgets-2.9.1 orig/samples/opengl/cube/cube.cpp	2010-07-18 09:39:20.000000000 -0400
+++ wxWidgets-2.9.1/samples/opengl/cube/cube.cpp	2011-04-06 16:48:06.000000000 -0400
@@ -255,13 +255,6 @@
     return true;
 }
 
-int MyApp::OnExit()
-{
-    delete m_glContext;
-
-    return wxApp::OnExit();
-}
-
 TestGLContext& MyApp::GetContext(wxGLCanvas *canvas)
 {
     if ( !m_glContext )
@@ -371,7 +364,7 @@
 MyFrame::MyFrame()
        : wxFrame(NULL, wxID_ANY, wxT("wxWidgets OpenGL Cube Sample"))
 {
-    new TestGLCanvas(this);
+    m_canvas = new TestGLCanvas(this);
 
     SetIcon(wxICON(sample));
 
@@ -396,6 +389,11 @@
                 wxGLCanvas::IsDisplaySupported(attribs) ? "is" : "not");
 }
 
+MyFrame::~MyFrame()
+{
+    delete m_canvas;
+}
+
 void MyFrame::OnClose(wxCommandEvent& WXUNUSED(event))
 {
     // true is to force the frame to close
diff -uNr wxWidgets-2.9.1 orig/samples/opengl/cube/cube.h wxWidgets-2.9.1/samples/opengl/cube/cube.h
--- wxWidgets-2.9.1 orig/samples/opengl/cube/cube.h	2010-07-18 09:39:20.000000000 -0400
+++ wxWidgets-2.9.1/samples/opengl/cube/cube.h	2011-04-06 10:46:30.000000000 -0400
@@ -47,19 +47,6 @@
     TestGLContext *m_glContext;
 };
 
-// Define a new frame type
-class MyFrame : public wxFrame
-{
-public:
-    MyFrame();
-
-private:
-    void OnClose(wxCommandEvent& event);
-    void OnNewWindow(wxCommandEvent& event);
-
-    DECLARE_EVENT_TABLE()
-};
-
 class TestGLCanvas : public wxGLCanvas
 {
 public:
@@ -76,4 +63,21 @@
     DECLARE_EVENT_TABLE()
 };
 
+// Define a new frame type
+class MyFrame : public wxFrame
+{
+public:
+    MyFrame();
+    
+    virtual ~MyFrame();
+    
+    TestGLCanvas *m_canvas;
+
+private:
+    void OnClose(wxCommandEvent& event);
+    void OnNewWindow(wxCommandEvent& event);
+
+    DECLARE_EVENT_TABLE()
+};
+
 #endif // _WX_CUBE_H_
diff -uNr wxWidgets-2.9.1 orig/src/osx/carbon/glcanvas.cpp wxWidgets-2.9.1/src/osx/carbon/glcanvas.cpp
--- wxWidgets-2.9.1 orig/src/osx/carbon/glcanvas.cpp	2010-07-18 09:39:20.000000000 -0400
+++ wxWidgets-2.9.1/src/osx/carbon/glcanvas.cpp	2011-04-06 10:54:17.000000000 -0400
@@ -345,9 +345,14 @@
     m_bufferName = gCurrentBufferName++;
     aglSetInteger (m_dummyContext, AGL_BUFFER_NAME, &m_bufferName);
 
-    AGLDrawable drawable = (AGLDrawable)GetWindowPort(MAC_WXHWND(MacGetTopLevelWindowRef()));
-    aglSetDrawable(m_dummyContext, drawable);
-
+    // TODO: this needs to be wrapped in an availablity macro to support building on 10.4...
+    if (aglSetWindowRef)
+        aglSetWindowRef(m_dummyContext, MacGetTopLevelWindowRef());
+    else
+    {
+        AGLDrawable drawable = (AGLDrawable)GetWindowPort(MAC_WXHWND(MacGetTopLevelWindowRef()));
+        aglSetDrawable(m_dummyContext, drawable);
+    }
+    
     m_macCanvasIsShown = true;
 
     return true;
