Index: src/common/imaggif.cpp
===================================================================
--- src/common/imaggif.cpp	(revision 59961)
+++ src/common/imaggif.cpp	(working copy)
@@ -33,6 +33,13 @@
 
 #if wxUSE_STREAMS
 
+int wxGIFHandler::GetImageCount( wxInputStream& stream )
+{
+    wxGIFDecoder decod;
+    decod.LoadGIF(stream);
+    return static_cast<int>( decod.GetFrameCount() );
+}
+
 bool wxGIFHandler::LoadFile(wxImage *image, wxInputStream& stream,
                             bool verbose, int index)
 {
Index: include/wx/imaggif.h
===================================================================
--- include/wx/imaggif.h	(revision 59961)
+++ include/wx/imaggif.h	(working copy)
@@ -33,6 +33,7 @@
 #if wxUSE_STREAMS
     virtual bool LoadFile( wxImage *image, wxInputStream& stream, bool verbose=true, int index=-1 );
     virtual bool SaveFile( wxImage *image, wxOutputStream& stream, bool verbose=true );
+    virtual int GetImageCount( wxInputStream& stream );
 protected:
     virtual bool DoCanRead( wxInputStream& stream );
 #endif
