diff -bu1 -r wxWindows/contrib/include/wx/fl/dyntbar.h wxWin/contrib/include/wx/fl/dyntbar.h
--- wxWindows/contrib/include/wx/fl/dyntbar.h	2003-08-18 08:14:28.000000000 +0200
+++ wxWin/contrib/include/wx/fl/dyntbar.h	2003-09-25 15:33:19.000000000 +0200
@@ -158,2 +158,7 @@
 
+        // Unhide method from parent.
+
+    virtual wxToolBarToolBase *AddTool (wxToolBarToolBase *tool)
+                              { return wxToolBarBase::AddTool(tool); };
+
     // Method from wxToolBarBase (for compatibility), only
@@ -168,3 +173,8 @@
 
-    virtual void AddSeparator( wxWindow* pSepartorWnd = NULL );
+    virtual void AddSeparator( wxWindow* pSepartorWnd );
+
+        // Unhide method from parent.
+
+    virtual wxToolBarToolBase *AddSeparator()
+                              { return wxToolBarBase::AddSeparator(); };
 
diff -bu1 -r wxWindows/contrib/include/wx/fl/newbmpbtn.h wxWin/contrib/include/wx/fl/newbmpbtn.h
--- wxWindows/contrib/include/wx/fl/newbmpbtn.h	2003-07-09 09:54:58.000000000 +0200
+++ wxWin/contrib/include/wx/fl/newbmpbtn.h	2003-09-25 11:25:07.000000000 +0200
@@ -147,2 +147,7 @@
 
+        // Unhide method from parents.
+
+    virtual void SetLabel(const wxString& label)
+                         { wxPanel::SetLabel(label); };
+
         // Sets the text alignment and margins.
diff -bu1 -r wxWindows/contrib/samples/deprecated/proplist/proplist.cpp wxWin/contrib/samples/deprecated/proplist/proplist.cpp
--- wxWindows/contrib/samples/deprecated/proplist/proplist.cpp	2003-03-31 12:23:18.000000000 +0200
+++ wxWin/contrib/samples/deprecated/proplist/proplist.cpp	2003-09-25 13:35:40.000000000 +0200
@@ -99,3 +99,3 @@
 // - must delete all frames except for the main one.
-void MyFrame::OnCloseWindow(wxCloseEvent& event)
+void MyFrame::OnCloseWindow(wxCloseEvent& WXUNUSED(event))
 {
@@ -109,3 +109,3 @@
 
-void MyFrame::OnQuit(wxCommandEvent& event)
+void MyFrame::OnQuit(wxCommandEvent& WXUNUSED(event))
 {
@@ -114,3 +114,3 @@
 
-void MyFrame::OnDialogList(wxCommandEvent& event)
+void MyFrame::OnDialogList(wxCommandEvent& WXUNUSED(event))
 {
@@ -119,3 +119,3 @@
 
-void MyFrame::OnFrameList(wxCommandEvent& event)
+void MyFrame::OnFrameList(wxCommandEvent& WXUNUSED(event))
 {
@@ -124,3 +124,3 @@
 
-void MyFrame::OnDialogForm(wxCommandEvent& event)
+void MyFrame::OnDialogForm(wxCommandEvent& WXUNUSED(event))
 {
@@ -129,3 +129,3 @@
 
-void MyFrame::OnFrameForm(wxCommandEvent& event)
+void MyFrame::OnFrameForm(wxCommandEvent& WXUNUSED(event))
 {
@@ -134,3 +134,3 @@
 
-void MyFrame::OnAbout(wxCommandEvent& event)
+void MyFrame::OnAbout(wxCommandEvent& WXUNUSED(event))
 {
diff -bu1 -r wxWindows/contrib/samples/deprecated/treelay/treelay.cpp wxWin/contrib/samples/deprecated/treelay/treelay.cpp
--- wxWindows/contrib/samples/deprecated/treelay/treelay.cpp	2003-03-31 12:23:24.000000000 +0200
+++ wxWin/contrib/samples/deprecated/treelay/treelay.cpp	2003-09-25 13:36:54.000000000 +0200
@@ -154,3 +154,3 @@
 
-void MyFrame::OnQuit(wxCommandEvent& event)
+void MyFrame::OnQuit(wxCommandEvent& WXUNUSED(event))
 {
@@ -159,3 +159,3 @@
 
-void MyFrame::OnLeftRight(wxCommandEvent& event)
+void MyFrame::OnLeftRight(wxCommandEvent& WXUNUSED(event))
 {
@@ -172,3 +172,3 @@
 
-void MyFrame::OnTopBottom(wxCommandEvent& event)
+void MyFrame::OnTopBottom(wxCommandEvent& WXUNUSED(event))
 {
@@ -185,3 +185,3 @@
 
-void MyFrame::OnAbout(wxCommandEvent& event)
+void MyFrame::OnAbout(wxCommandEvent& WXUNUSED(event))
 {
@@ -190,3 +190,3 @@
 
-void MyFrame::OnCloseWindow(wxCloseEvent& event)
+void MyFrame::OnCloseWindow(wxCloseEvent& WXUNUSED(event))
 {
@@ -207,3 +207,3 @@
 // Define the repainting behaviour
-void MyCanvas::OnPaint(wxPaintEvent& event)
+void MyCanvas::OnPaint(wxPaintEvent& WXUNUSED(event))
 {
diff -bu1 -r wxWindows/contrib/samples/fl/fl_demo1.cpp wxWin/contrib/samples/fl/fl_demo1.cpp
--- wxWindows/contrib/samples/fl/fl_demo1.cpp	2003-08-26 13:18:48.000000000 +0200
+++ wxWin/contrib/samples/fl/fl_demo1.cpp	2003-09-25 11:35:38.000000000 +0200
@@ -98,3 +98,3 @@
 
-void MyFrame::OnLoad( wxCommandEvent& event )
+void MyFrame::OnLoad( wxCommandEvent& WXUNUSED(event) )
 {						
@@ -103,3 +103,3 @@
 
-void MyFrame::OnSave( wxCommandEvent& event )
+void MyFrame::OnSave( wxCommandEvent& WXUNUSED(event) )
 {
@@ -108,3 +108,3 @@
 
-void MyFrame::OnExit( wxCommandEvent& event )
+void MyFrame::OnExit( wxCommandEvent& WXUNUSED(event) )
 {
@@ -216,3 +216,3 @@
     pToolBar->AddTool( 1001, wxString(wxT(BMP_DIR)) + wxT("new.bmp") );
-    pToolBar->AddSeparator();
+    pToolBar->AddSeparator( (wxWindow*) NULL);
     pToolBar->AddTool( 1002, wxString(wxT(BMP_DIR)) + wxT("open.bmp") );
diff -bu1 -r wxWindows/contrib/samples/fl/fl_demo2.cpp wxWin/contrib/samples/fl/fl_demo2.cpp
--- wxWindows/contrib/samples/fl/fl_demo2.cpp	2003-08-26 13:18:48.000000000 +0200
+++ wxWin/contrib/samples/fl/fl_demo2.cpp	2003-09-25 15:53:32.000000000 +0200
@@ -168,2 +168,4 @@
 {
+    int i;
+
     mpInternalFrm = (wxPanel*)this;
@@ -175,3 +177,2 @@
     
-    int i = 0;
     for( i = 0; i != MAX_LAYOUTS; ++i ) 
@@ -242,3 +243,3 @@
 
-void MyFrame::OnLoad( wxCommandEvent& event )
+void MyFrame::OnLoad( wxCommandEvent& WXUNUSED(event) )
 {
@@ -247,3 +248,3 @@
 
-void MyFrame::OnStore( wxCommandEvent& event )
+void MyFrame::OnStore( wxCommandEvent& WXUNUSED(event) )
 {
@@ -252,3 +253,3 @@
 
-void MyFrame::OnAutoSave( wxCommandEvent& event )
+void MyFrame::OnAutoSave( wxCommandEvent& WXUNUSED(event) )
 {
@@ -262,3 +263,3 @@
 
-void MyFrame::OnRemove( wxCommandEvent& event )
+void MyFrame::OnRemove( wxCommandEvent& WXUNUSED(event) )
 {
@@ -269,3 +270,3 @@
 
-void MyFrame::OnRemoveAll( wxCommandEvent& event )
+void MyFrame::OnRemoveAll( wxCommandEvent& WXUNUSED(event) )
 {
@@ -289,3 +290,3 @@
 
-void MyFrame::OnFirst( wxCommandEvent& event )
+void MyFrame::OnFirst( wxCommandEvent& WXUNUSED(event) )
 {
@@ -294,3 +295,3 @@
 
-void MyFrame::OnSecond( wxCommandEvent& event )
+void MyFrame::OnSecond( wxCommandEvent& WXUNUSED(event) )
 {
@@ -299,3 +300,3 @@
 
-void MyFrame::OnThird( wxCommandEvent& event )
+void MyFrame::OnThird( wxCommandEvent& WXUNUSED(event) )
 {
@@ -304,3 +305,3 @@
 
-void MyFrame::OnQuit( wxCommandEvent& event )
+void MyFrame::OnQuit( wxCommandEvent& WXUNUSED(event) )
 {
@@ -323,3 +324,3 @@
 
-void MyFrame::OnAbout( wxCommandEvent& event )
+void MyFrame::OnAbout( wxCommandEvent& WXUNUSED(event) )
 {
@@ -399,3 +400,3 @@
 
-void MyFrame::OnSayItsOk( wxCommandEvent& event )
+void MyFrame::OnSayItsOk( wxCommandEvent& WXUNUSED(event) )
 {
@@ -404,3 +405,3 @@
 
-void MyFrame::OnBtnYes( wxCommandEvent& event )
+void MyFrame::OnBtnYes( wxCommandEvent& WXUNUSED(event) )
 {
@@ -409,3 +410,3 @@
 
-void MyFrame::OnBtnNo( wxCommandEvent& event )
+void MyFrame::OnBtnNo( wxCommandEvent& WXUNUSED(event) )
 {
@@ -414,3 +415,3 @@
 
-void MyFrame::OnBtnEsc( wxCommandEvent& event )
+void MyFrame::OnBtnEsc( wxCommandEvent& WXUNUSED(event) )
 {
@@ -526,3 +527,3 @@
 
-void MyFrame::AddSearchToolbars( wxFrameLayout& layout, wxWindow* pParent )
+void MyFrame::AddSearchToolbars( wxFrameLayout& layout, wxWindow* WXUNUSED(pParent) )
 {
@@ -561,3 +562,3 @@
     pTBar2->AddTool( 2, wxString(wxT(BMP_DIR)) + wxT("search.bmp") );
-    //pTBar2->AddSeparator();
+    //pTBar2->AddSeparator( NULL );
     pTBar2->AddTool( 3, wxString(wxT(BMP_DIR)) + wxT("bookmarks.bmp") );
@@ -571,3 +572,3 @@
     pTBar3->AddTool( 3, wxString(wxT(BMP_DIR)) + wxT("saveall.bmp"), wxBITMAP_TYPE_BMP, wxString(_(" Save All ")) );
-    //pTBar3->AddSeparator();
+    //pTBar3->AddSeparator( NULL );
     pTBar3->AddTool( 4, wxString(wxT(BMP_DIR)) + wxT("cut.bmp"),   wxBITMAP_TYPE_BMP, wxString(_(" Open ")) );
@@ -585,3 +586,3 @@
     pTBar4->AddTool( 3, wxString(wxT(BMP_DIR)) + wxT("prevmark.bmp"),  wxBITMAP_TYPE_BMP, wxString(_("Prev bookmark ")), TRUE );
-    //pTBar4->AddSeparator();
+    //pTBar4->AddSeparator( NULL );
     pTBar4->AddTool( 4, wxString(wxT(BMP_DIR)) + wxT("search.bmp"), wxBITMAP_TYPE_BMP, wxString(_("Search ")), TRUE );
@@ -929,3 +930,3 @@
 
-void StartButton95::OnMouseDown( wxMouseEvent& event )
+void StartButton95::OnMouseDown( wxMouseEvent& WXUNUSED(event) )
 {
@@ -936,3 +937,3 @@
 
-void StartButton95::OnMouseUp( wxMouseEvent& event )
+void StartButton95::OnMouseUp( wxMouseEvent& WXUNUSED(event) )
 {
@@ -945,4 +946,3 @@
     
-    int i = 0;
-    for( i = 1; i != 6; ++i ) 
+    for( int i = 1; i != 6; ++i ) 
     { 
@@ -956,5 +956,5 @@
 
-void StartButton95::OnPaint( wxPaintEvent& event )
+void StartButton95::OnPaint( wxPaintEvent& WXUNUSED(event) )
 {
-    wxBitmap* pBmp = 0;
+    wxBitmap* pBmp;
     
diff -bu1 -r wxWindows/contrib/samples/fl/fl_sample1.cpp wxWin/contrib/samples/fl/fl_sample1.cpp
--- wxWindows/contrib/samples/fl/fl_sample1.cpp	2003-08-26 13:18:48.000000000 +0200
+++ wxWin/contrib/samples/fl/fl_sample1.cpp	2003-09-25 11:43:30.000000000 +0200
@@ -104,3 +104,3 @@
 MyFrame::MyFrame( wxWindow* parent, const wxChar *title )
-    : wxFrame( parent, -1, _("NewTest-II"), wxDefaultPosition,
+    : wxFrame( parent, -1, title, wxDefaultPosition,
           wxSize( 700, 500 ),
@@ -179,3 +179,3 @@
 
-void MyFrame::OnLoad( wxCommandEvent& event )
+void MyFrame::OnLoad( wxCommandEvent& WXUNUSED(event) )
 {
@@ -184,3 +184,3 @@
 
-void MyFrame::OnStore( wxCommandEvent& event )
+void MyFrame::OnStore( wxCommandEvent& WXUNUSED(event) )
 {
@@ -189,3 +189,3 @@
 
-void MyFrame::OnQuit( wxCommandEvent& event )
+void MyFrame::OnQuit( wxCommandEvent& WXUNUSED(event) )
 {
diff -bu1 -r wxWindows/contrib/samples/fl/fl_sample2.cpp wxWin/contrib/samples/fl/fl_sample2.cpp
--- wxWindows/contrib/samples/fl/fl_sample2.cpp	2003-08-26 13:18:48.000000000 +0200
+++ wxWin/contrib/samples/fl/fl_sample2.cpp	2003-09-25 11:44:00.000000000 +0200
@@ -105,3 +105,3 @@
 MyFrame::MyFrame( wxWindow* parent, const wxChar *title )
-    : wxFrame( parent, -1, _("NewTest-II"), wxDefaultPosition,
+    : wxFrame( parent, -1, title, wxDefaultPosition,
           wxSize( 700, 500 ),
@@ -177,3 +177,3 @@
 
-void MyFrame::OnLoad( wxCommandEvent& event )
+void MyFrame::OnLoad( wxCommandEvent& WXUNUSED(event) )
 {
@@ -182,3 +182,3 @@
 
-void MyFrame::OnStore( wxCommandEvent& event )
+void MyFrame::OnStore( wxCommandEvent& WXUNUSED(event) )
 {
@@ -187,3 +187,3 @@
 
-void MyFrame::OnQuit( wxCommandEvent& event )
+void MyFrame::OnQuit( wxCommandEvent& WXUNUSED(event) )
 {
diff -bu1 -r wxWindows/contrib/samples/fl/fl_sample3.cpp wxWin/contrib/samples/fl/fl_sample3.cpp
--- wxWindows/contrib/samples/fl/fl_sample3.cpp	2003-08-26 13:18:48.000000000 +0200
+++ wxWin/contrib/samples/fl/fl_sample3.cpp	2003-09-25 11:45:42.000000000 +0200
@@ -110,3 +110,3 @@
 MyFrame::MyFrame( wxWindow* parent, const wxChar *title )
-    : wxFrame( parent, -1, _("NewTest-II"), wxDefaultPosition,
+    : wxFrame( parent, -1, title, wxDefaultPosition,
           wxSize( 700, 500 ),
@@ -204,3 +204,3 @@
 
-void MyFrame::OnLoad( wxCommandEvent& event )
+void MyFrame::OnLoad( wxCommandEvent& WXUNUSED(event) )
 {
@@ -209,3 +209,3 @@
 
-void MyFrame::OnStore( wxCommandEvent& event )
+void MyFrame::OnStore( wxCommandEvent& WXUNUSED(event) )
 {
@@ -214,3 +214,3 @@
 
-void MyFrame::OnQuit( wxCommandEvent& event )
+void MyFrame::OnQuit( wxCommandEvent& WXUNUSED(event) )
 {
@@ -221,3 +221,3 @@
 
-void MyFrame::OnButt( wxCommandEvent& event )
+void MyFrame::OnButt( wxCommandEvent& WXUNUSED(event) )
 {
@@ -268,3 +268,3 @@
 
-void MyFrame::OnButt2( wxCommandEvent& event )
+void MyFrame::OnButt2( wxCommandEvent& WXUNUSED(event) )
 {
diff -bu1 -r wxWindows/contrib/samples/ogl/ogledit/doc.cpp wxWin/contrib/samples/ogl/ogledit/doc.cpp
--- wxWindows/contrib/samples/ogl/ogledit/doc.cpp	2003-08-21 12:26:54.000000000 +0200
+++ wxWin/contrib/samples/ogl/ogledit/doc.cpp	2003-09-25 19:28:33.000000000 +0200
@@ -224,3 +224,3 @@
     {
-      wxShape *theShape = NULL;
+      wxShape *theShape;
       if (shape)
@@ -255,3 +255,3 @@
     {
-      wxShape *theShape = NULL;
+      wxShape *theShape;
       if (shape)
@@ -426,9 +426,9 @@
 {
-  wxNode *node = shape->GetLines().First();
+  wxNode *node = shape->GetLines().GetFirst();
   while (node)
   {
-    wxLineShape *line = (wxLineShape *)node->Data();
+    wxLineShape *line = (wxLineShape *)node->GetData();
     doc->GetCommandProcessor()->Submit(new DiagramCommand(_T("Cut"), OGLEDIT_CUT, doc, NULL, 0.0, 0.0, line->Selected(), line));
     
-    node = shape->GetLines().First();
+    node = shape->GetLines().GetFirst();
   }
@@ -457,6 +457,6 @@
       bool redraw = FALSE;
-      wxNode *node = GetShape()->GetCanvas()->GetDiagram()->GetShapeList()->First();
+      wxNode *node = GetShape()->GetCanvas()->GetDiagram()->GetShapeList()->GetFirst();
       while (node)
       {
-        wxShape *eachShape = (wxShape *)node->Data();
+        wxShape *eachShape = (wxShape *)node->GetData();
         if (eachShape->GetParent() == NULL)
@@ -469,3 +469,3 @@
         }
-        node = node->Next();
+        node = node->GetNext();
       }
diff -bu1 -r wxWindows/contrib/samples/ogl/ogledit/ogledit.cpp wxWin/contrib/samples/ogl/ogledit/ogledit.cpp
--- wxWindows/contrib/samples/ogl/ogledit/ogledit.cpp	2003-08-21 12:26:54.000000000 +0200
+++ wxWin/contrib/samples/ogl/ogledit/ogledit.cpp	2003-09-25 12:31:03.000000000 +0200
@@ -74,3 +74,2 @@
   wxMenu *file_menu = new wxMenu;
-  wxMenu *edit_menu = NULL;
 
@@ -87,3 +86,3 @@
 
-  edit_menu = new wxMenu;
+  wxMenu *edit_menu = new wxMenu;
   edit_menu->Append(wxID_UNDO, _T("&Undo"));
diff -bu1 -r wxWindows/contrib/samples/ogl/ogledit/view.cpp wxWin/contrib/samples/ogl/ogledit/view.cpp
--- wxWindows/contrib/samples/ogl/ogledit/view.cpp	2003-08-21 12:26:54.000000000 +0200
+++ wxWin/contrib/samples/ogl/ogledit/view.cpp	2003-09-25 19:29:49.000000000 +0200
@@ -129,3 +129,3 @@
     /* wxCursor *old_cursor = NULL; */
-    wxNode *current = diagram_p->GetShapeList()->First();
+    wxNode *current = diagram_p->GetShapeList()->GetFirst();
 
@@ -133,3 +133,3 @@
     {
-        wxShape *object = (wxShape *)current->Data();
+        wxShape *object = (wxShape *)current->GetData();
         if (!object->GetParent())
@@ -138,3 +138,3 @@
         }
-        current = current->Next();  // Procede to the next shape in the list
+        current = current->GetNext();  // Procede to the next shape in the list
     }
@@ -179,6 +179,6 @@
   wxShape *theShape = NULL;
-  wxNode *node = doc->GetDiagram()->GetShapeList()->First();
+  wxNode *node = doc->GetDiagram()->GetShapeList()->GetFirst();
   while (node)
   {
-    wxShape *eachShape = (wxShape *)node->Data();
+    wxShape *eachShape = (wxShape *)node->GetData();
     if ((eachShape->GetParent() == NULL) && eachShape->Selected())
@@ -188,3 +188,3 @@
     }
-    else node = node->Next();
+    else node = node->GetNext();
   }
diff -bu1 -r wxWindows/contrib/samples/ogl/studio/csprint.cpp wxWin/contrib/samples/ogl/studio/csprint.cpp
--- wxWindows/contrib/samples/ogl/studio/csprint.cpp	2003-08-26 13:18:48.000000000 +0200
+++ wxWin/contrib/samples/ogl/studio/csprint.cpp	2003-09-25 19:41:36.000000000 +0200
@@ -72,6 +72,6 @@
     wxList* shapeList = diagramFrom->GetShapeList();
-    wxNode* node = shapeList->First();
+    wxNode* node = shapeList->GetFirst();
     while (node)
     {
-        wxShape* shape = (wxShape*) node->Data();
+        wxShape* shape = (wxShape*) node->GetData();
         if (((diagramFrom == this) || shape->Selected()) && !shape->IsKindOf(CLASSINFO(wxLineShape)))
@@ -92,9 +92,9 @@
         }
-        node = node->Next();
+        node = node->GetNext();
     }
 
-    node = shapeList->First();
+    node = shapeList->GetFirst();
     while (node)
     {
-        wxShape* shape = (wxShape*) node->Data();
+        wxShape* shape = (wxShape*) node->GetData();
         if (((diagramFrom == this) || shape->Selected()) && shape->IsKindOf(CLASSINFO(wxLineShape)))
@@ -124,3 +124,3 @@
         }
-        node = node->Next();
+        node = node->GetNext();
     }
@@ -128,6 +128,6 @@
     // Now make sure line ordering is correct
-    node = shapeList->First();
+    node = shapeList->GetFirst();
     while (node)
     {
-        wxShape* shape = (wxShape*) node->Data();
+        wxShape* shape = (wxShape*) node->GetData();
         if (((diagramFrom == this) || shape->Selected()) && !shape->IsKindOf(CLASSINFO(wxLineShape)))
@@ -139,6 +139,6 @@
             wxList newLines;
-            wxNode* lineNode = shape->GetLines().First();
+            wxNode* lineNode = shape->GetLines().GetFirst();
             while (lineNode)
             {
-                wxLineShape* lineShape = (wxLineShape*) lineNode->Data();
+                wxLineShape* lineShape = (wxLineShape*) lineNode->GetData();
                 if ((diagramFrom == this) || (lineShape->GetTo()->Selected() && lineShape->GetFrom()->Selected()))
@@ -152,9 +152,9 @@
 
-                lineNode = lineNode->Next();
+                lineNode = lineNode->GetNext();
             }
 
-            if (newLines.Number() > 0)
+            if (newLines.GetCount() > 0)
                 newShape->ApplyAttachmentOrdering(newLines);
         }
-        node = node->Next();
+        node = node->GetNext();
     }
@@ -286,3 +286,3 @@
     {
-        if (m_currentCmd->GetStates().Number() == 0)
+        if (m_currentCmd->GetStates().GetCount() == 0)
         {
diff -bu1 -r wxWindows/contrib/samples/ogl/studio/dialogs.cpp wxWin/contrib/samples/ogl/studio/dialogs.cpp
--- wxWindows/contrib/samples/ogl/studio/dialogs.cpp	2003-08-21 12:26:54.000000000 +0200
+++ wxWin/contrib/samples/ogl/studio/dialogs.cpp	2003-09-25 20:18:18.000000000 +0200
@@ -102,3 +102,6 @@
 
-    bool success = wxLoadFromResource(m_generalSettings, m_notebook, _T("general_settings_dialog"));
+    #ifdef  __WXDEBUG__
+    bool success = 
+    #endif
+                   wxLoadFromResource(m_generalSettings, m_notebook, _T("general_settings_dialog"));
     wxASSERT_MSG( (success), _T("Could not load general settings panel."));
@@ -108,3 +111,6 @@
 
-    success = wxLoadFromResource(m_diagramSettings, m_notebook, _T("diagram_settings_dialog"));
+    #ifdef  __WXDEBUG__
+    success = 
+    #endif
+              wxLoadFromResource(m_diagramSettings, m_notebook, _T("diagram_settings_dialog"));
     wxASSERT_MSG( (success), _T("Could not load diagram settings panel."));
@@ -201,6 +207,6 @@
     // Apply settings to all open diagram documents
-    wxNode* node = wxGetApp().GetDocManager()->GetDocuments().First();
+    wxNode* node = wxGetApp().GetDocManager()->GetDocuments().GetFirst();
     while (node)
     {
-        wxDocument* doc = (wxDocument*) node->Data();
+        wxDocument* doc = (wxDocument*) node->GetData();
         if (doc->IsKindOf(CLASSINFO(csDiagramDocument)))
@@ -230,3 +236,3 @@
         }
-        node = node->Next();
+        node = node->GetNext();
     }
@@ -264,3 +270,6 @@
     m_generalPropertiesDialog = new csGeneralShapePropertiesDialog;
-    bool success = wxLoadFromResource(m_generalPropertiesDialog, m_notebook, _T("general_shape_properties_dialog"));
+    #ifdef  __WXDEBUG__
+    bool success = 
+    #endif
+                   wxLoadFromResource(m_generalPropertiesDialog, m_notebook, _T("general_shape_properties_dialog"));
     wxASSERT_MSG( (success), _T("Could not load general properties panel."));
@@ -268,4 +277,3 @@
 
-    success = wxLoadFromResource(m_attributeDialog, m_notebook, attributeDialogName);
-    if (!success)
+    if (!wxLoadFromResource(m_attributeDialog, m_notebook, attributeDialogName))
     {
@@ -284,4 +292,3 @@
     m_alternativeAttributeDialog = new wxPanel;
-    success = wxLoadFromResource(m_alternativeAttributeDialog, m_notebook, str);
-    if (success)
+    if (wxLoadFromResource(m_alternativeAttributeDialog, m_notebook, str))
     {
@@ -360,6 +367,6 @@
 
-    wxNode* node = m_attributeDialog->GetChildren().First();
+    wxWindowListNode* node = m_attributeDialog->GetChildren().GetFirst();
     while (node)
     {
-        wxWindow* child = (wxWindow*) node->Data();
+        wxWindow* child = (wxWindow*) node->GetData();
         if (child->IsKindOf(CLASSINFO(wxChoice)))
@@ -369,3 +376,3 @@
         }
-        node = node->Next();
+        node = node->GetNext();
     }
@@ -375,6 +382,6 @@
 
-    node = m_alternativeAttributeDialog->GetChildren().First();
+    node = m_alternativeAttributeDialog->GetChildren().GetFirst();
     while (node)
     {
-        wxWindow* child = (wxWindow*) node->Data();
+        wxWindow* child = (wxWindow*) node->GetData();
         if (child->IsKindOf(CLASSINFO(wxChoice)))
@@ -384,3 +391,3 @@
         }
-        node = node->Next();
+        node = node->GetNext();
     }
diff -bu1 -r wxWindows/contrib/samples/ogl/studio/doc.cpp wxWin/contrib/samples/ogl/studio/doc.cpp
--- wxWindows/contrib/samples/ogl/studio/doc.cpp	2003-08-21 12:26:54.000000000 +0200
+++ wxWin/contrib/samples/ogl/studio/doc.cpp	2003-09-25 20:16:01.000000000 +0200
@@ -122,8 +122,8 @@
 {
-    wxNode* node = m_states.First();
+    wxNode* node = m_states.GetFirst();
     while (node)
     {
-        csCommandState* state = (csCommandState*) node->Data();
+        csCommandState* state = (csCommandState*) node->GetData();
         delete state;
-        node = node->Next();
+        node = node->GetNext();
     }
@@ -149,6 +149,6 @@
 {
-    wxNode* node = m_states.First();
+    wxNode* node = m_states.GetFirst();
     while (node)
     {
-        csCommandState* state = (csCommandState*) node->Data();
+        csCommandState* state = (csCommandState*) node->GetData();
         wxShape* shape = state->GetShapeOnCanvas();
@@ -156,6 +156,6 @@
 
-        wxNode *node1 = shape->GetLines().First();
+        wxNode *node1 = shape->GetLines().GetFirst();
         while (node1)
         {
-            wxLineShape *line = (wxLineShape *)node1->Data();
+            wxLineShape *line = (wxLineShape *)node1->GetData();
             if (!FindStateByShape(line))
@@ -166,5 +166,5 @@
 
-            node1 = node1->Next();
+            node1 = node1->GetNext();
         }
-        node = node->Next();
+        node = node->GetNext();
     }
@@ -174,9 +174,9 @@
 {
-    wxNode* node = m_states.First();
+    wxNode* node = m_states.GetFirst();
     while (node)
     {
-        csCommandState* state = (csCommandState*) node->Data();
+        csCommandState* state = (csCommandState*) node->GetData();
         if (shape == state->GetShapeOnCanvas() || shape == state->GetSavedState())
             return state;
-        node = node->Next();
+        node = node->GetNext();
     }
@@ -187,9 +187,9 @@
 {
-    wxNode* node = m_states.First();
+    wxNode* node = m_states.GetFirst();
     while (node)
     {
-        csCommandState* state = (csCommandState*) node->Data();
+        csCommandState* state = (csCommandState*) node->GetData();
         if (!state->Do())
             return FALSE;
-        node = node->Next();
+        node = node->GetNext();
     }
@@ -202,9 +202,9 @@
     // back before the lines do.
-    wxNode* node = m_states.Last();
+    wxNode* node = m_states.GetLast();
     while (node)
     {
-        csCommandState* state = (csCommandState*) node->Data();
+        csCommandState* state = (csCommandState*) node->GetData();
         if (!state->Undo())
             return FALSE;
-        node = node->Previous();
+        node = node->GetPrevious();
     }
@@ -459,3 +459,3 @@
             // Recursively redraw links if we have a composite.
-            if (m_shapeOnCanvas->GetChildren().Number() > 0)
+            if (m_shapeOnCanvas->GetChildren().GetCount() > 0)
                 m_shapeOnCanvas->DrawLinks(dc, -1, TRUE);
diff -bu1 -r wxWindows/contrib/samples/ogl/studio/project.cpp wxWin/contrib/samples/ogl/studio/project.cpp
--- wxWindows/contrib/samples/ogl/studio/project.cpp	2003-08-21 12:26:54.000000000 +0200
+++ wxWin/contrib/samples/ogl/studio/project.cpp	2003-09-25 14:56:06.000000000 +0200
@@ -33,3 +33,3 @@
 // Define my frame constructor
-csProjectTreeCtrl::csProjectTreeCtrl(wxWindow *parent, wxWindowID id, const wxPoint& pos, const wxSize& size,
+csProjectTreeCtrl::csProjectTreeCtrl(wxWindow *parent, const wxWindowID id, const wxPoint& pos, const wxSize& size,
 	long style):
diff -bu1 -r wxWindows/contrib/samples/ogl/studio/shapes.cpp wxWin/contrib/samples/ogl/studio/shapes.cpp
--- wxWindows/contrib/samples/ogl/studio/shapes.cpp	2003-08-21 12:26:56.000000000 +0200
+++ wxWin/contrib/samples/ogl/studio/shapes.cpp	2003-09-25 20:53:22.000000000 +0200
@@ -201,3 +201,3 @@
   GetShape()->GetAttachmentPositionEdge(attachment, &xp, &yp);
-  dc.DrawLine(xp, yp, x, y);
+  dc.DrawLine((wxCoord)xp, (wxCoord)yp, (wxCoord)x, (wxCoord)y);
   GetShape()->GetCanvas()->CaptureMouse();
@@ -215,3 +215,3 @@
   GetShape()->GetAttachmentPositionEdge(attachment, &xp, &yp);
-  dc.DrawLine(xp, yp, x, y);
+  dc.DrawLine((wxCoord)xp, (wxCoord)yp, (wxCoord)x, (wxCoord)y);
 }
@@ -301,6 +301,6 @@
   // Draw bounding box for other selected shapes
-  wxNode* node = GetShape()->GetCanvas()->GetDiagram()->GetShapeList()->First();
+  wxNode* node = GetShape()->GetCanvas()->GetDiagram()->GetShapeList()->GetFirst();
   while (node)
   {
-     wxShape* shape = (wxShape*) node->Data();
+     wxShape* shape = (wxShape*) node->GetData();
      if (shape->Selected() && !shape->IsKindOf(CLASSINFO(wxLineShape)) && (shape != GetShape()))
@@ -310,3 +310,3 @@
      }
-     node = node->Next();
+     node = node->GetNext();
   }
@@ -360,6 +360,6 @@
   // Draw bounding box for other selected shapes
-  wxNode* node = GetShape()->GetCanvas()->GetDiagram()->GetShapeList()->First();
+  wxNode* node = GetShape()->GetCanvas()->GetDiagram()->GetShapeList()->GetFirst();
   while (node)
   {
-     wxShape* shape = (wxShape*) node->Data();
+     wxShape* shape = (wxShape*) node->GetData();
      if (shape->Selected() && !shape->IsKindOf(CLASSINFO(wxLineShape)) && (shape != GetShape()))
@@ -369,3 +369,3 @@
      }
-     node = node->Next();
+     node = node->GetNext();
   }
@@ -415,6 +415,6 @@
   // Move line points
-  wxNode* node = GetShape()->GetCanvas()->GetDiagram()->GetShapeList()->First();
+  wxNode* node = GetShape()->GetCanvas()->GetDiagram()->GetShapeList()->GetFirst();
   while (node)
   {
-     wxShape* shape = (wxShape*) node->Data();
+     wxShape* shape = (wxShape*) node->GetData();
      // Only move the line point(s) if both ends move too
@@ -425,3 +425,3 @@
 
-        if (lineShape->GetLineControlPoints()->Number() > 2)
+        if (lineShape->GetLineControlPoints()->GetCount() > 2)
         {
@@ -429,9 +429,9 @@
 
-            wxNode *node1 = newLineShape->GetLineControlPoints()->First();
+            wxNode *node1 = newLineShape->GetLineControlPoints()->GetFirst();
             while (node1)
             {
-                wxRealPoint *point = (wxRealPoint *)node1->Data();
+                wxRealPoint *point = (wxRealPoint *)node1->GetData();
                 point->x += offsetX;
                 point->y += offsetY;
-                node1 = node1->Next();
+                node1 = node1->GetNext();
             }
@@ -441,3 +441,3 @@
      }
-     node = node->Next();
+     node = node->GetNext();
   }
@@ -445,6 +445,6 @@
   // Add other selected node shapes, if any
-  node = GetShape()->GetCanvas()->GetDiagram()->GetShapeList()->First();
+  node = GetShape()->GetCanvas()->GetDiagram()->GetShapeList()->GetFirst();
   while (node)
   {
-     wxShape* shape = (wxShape*) node->Data();
+     wxShape* shape = (wxShape*) node->GetData();
      if (shape->Selected() && !shape->IsKindOf(CLASSINFO(wxLineShape)) && (shape != GetShape()))
@@ -456,3 +456,3 @@
      }
-     node = node->Next();
+     node = node->GetNext();
   }
@@ -738,3 +738,3 @@
 
-    DrawRectangle(wxRect(- w/2, - h/2, w, h));
+    DrawRectangle(wxRect((int)(- w/2), (int)(- h/2), (int)(w), (int)(h)));
     CalculateSize();
@@ -757,3 +757,3 @@
 
-    DrawRoundedRectangle(wxRect(- w/2, - h/2, w, h), -0.3);
+    DrawRoundedRectangle(wxRect((int)(- w/2), (int)(- h/2), (int)(w), (int)(h)), -0.3);
     CalculateSize();
@@ -780,5 +780,5 @@
 
-    points[0] = wxPoint( 0 ,  - h / 2 );
-    points[1] = wxPoint( w / 2 ,  h / 2 );
-    points[2] = wxPoint( -w / 2,  h / 2 );
+    points[0] = wxPoint( 0 ,  (int)(- h / 2) );
+    points[1] = wxPoint( (int)(w / 2) ,  (int)(h / 2) );
+    points[2] = wxPoint( (int)(-w / 2),  (int)(h / 2) );
 
@@ -836,3 +836,3 @@
 
-    DrawRectangle(wxRect(-w/2.0, -h/2.0, w, h));
+    DrawRectangle(wxRect((int)(-w/2.0), (int)(-h/2.0), (int)(w), (int)(h)));
 
@@ -842,4 +842,4 @@
 
-    DrawEllipticArc(wxRect(-w/2, -h/2, w, 2*h), 0.0, 180.0);
-    DrawLine(wxPoint(-w/2, h/2), wxPoint(w/2, h/2));
+    DrawEllipticArc(wxRect((int)(-w/2), (int)(-h/2), (int)(w), (int)(2*h)), 0.0, 180.0);
+    DrawLine(wxPoint((int)(-w/2), (int)(h/2)), wxPoint((int)(w/2), (int)(h/2)));
 
@@ -857,3 +857,3 @@
 
-    DrawRectangle(wxRect(-w/2, -h/2, w, h));
+    DrawRectangle(wxRect((int)(-w/2), (int)(-h/2), (int)(w), (int)(h)));
 
@@ -862,4 +862,4 @@
 
-    DrawEllipticArc(wxRect(-w/2 - w, -h/2, 2*w, h), 270.0, 90.0);
-    DrawLine(wxPoint(-w/2, -h/2), wxPoint(-w/2, h/2));
+    DrawEllipticArc(wxRect((int)(-w/2 - w), (int)(-h/2), (int)(2*w), (int)(h)), 270.0, 90.0);
+    DrawLine(wxPoint((int)(-w/2), (int)(-h/2)), wxPoint((int)(-w/2), (int)(h/2)));
 
@@ -877,3 +877,3 @@
 
-    DrawRectangle(wxRect(-w/2, -h/2, w, h));
+    DrawRectangle(wxRect((int)(-w/2), (int)(-h/2), (int)(w), (int)(h)));
 
@@ -882,4 +882,4 @@
 
-    DrawEllipticArc(wxRect(-w/2, -h/2 - h, w, 2*h), 180.0, 0.0);
-    DrawLine(wxPoint(-w/2, -h/2), wxPoint(w/2, -h/2));
+    DrawEllipticArc(wxRect((int)(-w/2), (int)(-h/2 - h), (int)(w), (int)(2*h)), 180.0, 0.0);
+    DrawLine(wxPoint((int)(-w/2), (int)(-h/2)), wxPoint((int)(w/2), (int)(-h/2)));
 
@@ -897,3 +897,3 @@
 
-    DrawRectangle(wxRect(-w/2, -h/2, w, h));
+    DrawRectangle(wxRect((int)(-w/2), (int)(-h/2), (int)(w), (int)(h)));
 
@@ -902,4 +902,4 @@
 
-    DrawEllipticArc(wxRect(-w/2, -h/2, 2*w, h), 90.0, 270.0);
-    DrawLine(wxPoint(w/2, -h/2), wxPoint(w/2, h/2));
+    DrawEllipticArc(wxRect((int)(-w/2), (int)(-h/2), (int)(2*w), (int)(h)), 90.0, 270.0);
+    DrawLine(wxPoint((int)(w/2),(int)(-h/2)), wxPoint((int)(w/2), (int)(h/2)));
 
diff -bu1 -r wxWindows/contrib/samples/ogl/studio/studio.cpp wxWin/contrib/samples/ogl/studio/studio.cpp
--- wxWindows/contrib/samples/ogl/studio/studio.cpp	2003-09-14 07:59:41.000000000 +0200
+++ wxWin/contrib/samples/ogl/studio/studio.cpp	2003-09-25 12:25:06.000000000 +0200
@@ -246,5 +246,3 @@
 
-  wxMenu *editMenu = NULL;
-
-  editMenu = new wxMenu;
+  wxMenu* editMenu = new wxMenu;
   editMenu->Append(wxID_UNDO, _T("&Undo\tCtrl+Z"));
diff -bu1 -r wxWindows/contrib/samples/ogl/studio/symbols.cpp wxWin/contrib/samples/ogl/studio/symbols.cpp
--- wxWindows/contrib/samples/ogl/studio/symbols.cpp	2003-08-21 12:26:56.000000000 +0200
+++ wxWin/contrib/samples/ogl/studio/symbols.cpp	2003-09-25 20:09:36.000000000 +0200
@@ -76,9 +76,9 @@
 {
-    wxNode* node = m_symbols.First();
+    wxNode* node = m_symbols.GetFirst();
     while (node)
     {
-        csSymbol* symbol = (csSymbol*) node->Data();
+        csSymbol* symbol = (csSymbol*) node->GetData();
         delete symbol;
 
-        node = node->Next();
+        node = node->GetNext();
     }
@@ -89,6 +89,6 @@
 {
-    wxNode* node = m_symbols.First();
+    wxNode* node = m_symbols.GetFirst();
     while (node)
     {
-        csSymbol* symbol = (csSymbol*) node->Data();
+        csSymbol* symbol = (csSymbol*) node->GetData();
         if (symbol->GetName() == name)
@@ -96,3 +96,3 @@
 
-        node = node->Next();
+        node = node->GetNext();
     }
@@ -103,6 +103,6 @@
 {
-    wxNode* node = m_symbols.First();
+    wxNode* node = m_symbols.GetFirst();
     while (node)
     {
-        csSymbol* symbol = (csSymbol*) node->Data();
+        csSymbol* symbol = (csSymbol*) node->GetData();
         if (symbol->GetToolId() == toolId)
@@ -110,3 +110,3 @@
 
-        node = node->Next();
+        node = node->GetNext();
     }
diff -bu1 -r wxWindows/contrib/samples/ogl/studio/view.cpp wxWin/contrib/samples/ogl/studio/view.cpp
--- wxWindows/contrib/samples/ogl/studio/view.cpp	2003-08-21 12:26:56.000000000 +0200
+++ wxWin/contrib/samples/ogl/studio/view.cpp	2003-09-25 20:51:52.000000000 +0200
@@ -201,6 +201,6 @@
   wxShape *theShape = NULL;
-  wxNode *node = doc->GetDiagram()->GetShapeList()->First();
+  wxNode *node = doc->GetDiagram()->GetShapeList()->GetFirst();
   while (node)
   {
-    wxShape *eachShape = (wxShape *)node->Data();
+    wxShape *eachShape = (wxShape *)node->GetData();
     if ((eachShape->GetParent() == NULL) && !eachShape->IsKindOf(CLASSINFO(wxLabelShape)) && eachShape->Selected())
@@ -210,3 +210,3 @@
     }
-    else node = node->Next();
+    else node = node->GetNext();
   }
@@ -218,6 +218,6 @@
   csDiagramDocument *doc = (csDiagramDocument *)GetDocument();
-  wxNode *node = doc->GetDiagram()->GetShapeList()->First();
+  wxNode *node = doc->GetDiagram()->GetShapeList()->GetFirst();
   while (node)
   {
-    wxShape *eachShape = (wxShape *)node->Data();
+    wxShape *eachShape = (wxShape *)node->GetData();
     if ((eachShape->GetParent() == NULL) && !eachShape->IsKindOf(CLASSINFO(wxLabelShape)) && eachShape->Selected() && ((toFind == NULL) || (eachShape->IsKindOf(toFind))))
@@ -226,3 +226,3 @@
     }
-    node = node->Next();
+    node = node->GetNext();
   }
@@ -298,3 +298,3 @@
 {
-    event.Enable( (m_selections.Number() > 0) );
+    event.Enable( (m_selections.GetCount() > 0) );
 }
@@ -303,3 +303,3 @@
 {
-    event.Enable( (m_selections.Number() > 0) );
+    event.Enable( (m_selections.GetCount() > 0) );
 }
@@ -308,3 +308,3 @@
 {
-    event.Enable( (m_selections.Number() > 0) );
+    event.Enable( (m_selections.GetCount() > 0) );
 }
@@ -322,3 +322,3 @@
 {
-    event.Enable( (m_selections.Number() > 0) );
+    event.Enable( (m_selections.GetCount() > 0) );
 }
@@ -329,3 +329,3 @@
 
-    if (shapes.Number() > 0)
+    if (shapes.GetCount() > 0)
     {
@@ -333,6 +333,6 @@
 
-        wxNode* node = shapes.First();
+        wxNode* node = shapes.GetFirst();
         while (node)
         {
-            wxShape *theShape = (wxShape*) node->Data();
+            wxShape *theShape = (wxShape*) node->GetData();
             csCommandState* state = new csCommandState(ID_CS_CUT, NULL, theShape);
@@ -347,3 +347,3 @@
 
-            node = node->Next();
+            node = node->GetNext();
         }
@@ -361,3 +361,3 @@
 
-    if (shapes.Number() > 0)
+    if (shapes.GetCount() > 0)
     {
@@ -365,8 +365,8 @@
 
-        wxNode* node = shapes.First();
-        wxNode* node1 = oldShapes.First();
+        wxNode* node = shapes.GetFirst();
+        wxNode* node1 = oldShapes.GetFirst();
         while (node && node1)
         {
-            wxShape *theShape = (wxShape*) node->Data();
-            wxShape *oldShape = (wxShape*) node1->Data();
+            wxShape *theShape = (wxShape*) node->GetData();
+            wxShape *oldShape = (wxShape*) node1->GetData();
             csCommandState* state = new csCommandState(cmd, theShape, oldShape);
@@ -374,4 +374,4 @@
 
-            node = node->Next();
-            node1 = node1->Next();
+            node = node->GetNext();
+            node1 = node1->GetNext();
         }
@@ -388,3 +388,3 @@
 
-    if (selections.Number() > 0)
+    if (selections.GetCount() > 0)
     {
@@ -392,5 +392,5 @@
         data.SetChooseFull(TRUE);
-        if (selections.Number() == 1)
+        if (selections.GetCount() == 1)
         {
-            wxShape* firstShape = (wxShape*) selections.First()->Data();
+            wxShape* firstShape = (wxShape*) selections.GetFirst()->GetData();
             data.SetColour(firstShape->GetBrush()->GetColour());
@@ -412,6 +412,6 @@
 
-        wxNode* node = selections.First();
+        wxNode* node = selections.GetFirst();
         while (node)
         {
-            wxShape *theShape = (wxShape*) node->Data();
+            wxShape *theShape = (wxShape*) node->GetData();
             wxShape* newShape = theShape->CreateNewCopy();
@@ -422,3 +422,3 @@
 
-            node = node->Next();
+            node = node->GetNext();
         }
@@ -439,3 +439,3 @@
     FindSelectedShapes(selections);
-    event.Enable( (selections.Number() > 0) );
+    event.Enable( (selections.GetCount() > 0) );
 }
@@ -477,3 +477,3 @@
 
-    if (selections.Number() > 0)
+    if (selections.GetCount() > 0)
     {
@@ -481,6 +481,6 @@
 
-        wxNode* node = selections.First();
+        wxNode* node = selections.GetFirst();
         while (node)
         {
-            wxShape *theShape = (wxShape*) node->Data();
+            wxShape *theShape = (wxShape*) node->GetData();
             wxShape *newShape = theShape->CreateNewCopy();
@@ -500,3 +500,3 @@
 
-            node = node->Next();
+            node = node->GetNext();
         }
@@ -533,6 +533,6 @@
 
-        wxNode* node = selections.First();
+        wxNode* node = selections.GetFirst();
         while (node)
         {
-            wxShape *theShape = (wxShape*) node->Data();
+            wxShape *theShape = (wxShape*) node->GetData();
             theShape->Select(FALSE, &dc);
@@ -540,3 +540,3 @@
 
-            node = node->Next();
+            node = node->GetNext();
         }
@@ -546,6 +546,6 @@
         csDiagramDocument *doc = (csDiagramDocument *)GetDocument();
-        wxNode *node = doc->GetDiagram()->GetShapeList()->First();
+        wxNode *node = doc->GetDiagram()->GetShapeList()->GetFirst();
         while (node)
         {
-            wxShape *eachShape = (wxShape *)node->Data();
+            wxShape *eachShape = (wxShape *)node->GetData();
             if (eachShape->GetParent() == NULL &&
@@ -557,3 +557,3 @@
             }
-            node = node->Next();
+            node = node->GetNext();
         }
@@ -577,3 +577,3 @@
 
-    if (selections.Number() > 0)
+    if (selections.GetCount() > 0)
     {
@@ -581,6 +581,6 @@
 
-        wxNode* node = selections.First();
+        wxNode* node = selections.GetFirst();
         while (node)
         {
-            wxLineShape *theShape = (wxLineShape*) node->Data();
+            wxLineShape *theShape = (wxLineShape*) node->GetData();
             wxLineShape *newShape = NULL;
@@ -590,3 +590,3 @@
                 // Add arrow
-                if (theShape->GetArrows().Number() == 0)
+                if (theShape->GetArrows().GetCount() == 0)
                 {
@@ -598,3 +598,3 @@
             {
-                if (theShape->GetArrows().Number() > 0)
+                if (theShape->GetArrows().GetCount() > 0)
                 {
@@ -612,3 +612,3 @@
 
-            node = node->Next();
+            node = node->GetNext();
         }
@@ -622,3 +622,3 @@
     FindSelectedShapes(selections, CLASSINFO(wxLineShape));
-    event.Enable( (selections.Number() > 0) );
+    event.Enable( (selections.GetCount() > 0) );
 }
@@ -636,9 +636,9 @@
     bool haveArrow = FALSE;
-    wxNode *node = lineShape->GetArrows().First();
+    wxNode *node = lineShape->GetArrows().GetFirst();
     while (node)
     {
-      wxArrowHead *arrow = (wxArrowHead *)node->Data();
+      wxArrowHead *arrow = (wxArrowHead *)node->GetData();
       if (ARROW_POSITION_MIDDLE == arrow->GetArrowEnd())
         haveArrow = TRUE;
-      node = node->Next();
+      node = node->GetNext();
     }
@@ -653,6 +653,6 @@
     wxList selections;
-    wxNode* node = GetSelectionList().First();
+    wxNode* node = GetSelectionList().GetFirst();
     while (node)
     {
-        wxShape* shape = (wxShape*) node->Data();
+        wxShape* shape = (wxShape*) node->GetData();
         if ((shape->GetParent() == NULL) && (!shape->IsKindOf(CLASSINFO(wxLineShape))))
@@ -661,6 +661,6 @@
         }
-        node = node->Next();
+        node = node->GetNext();
     }
 
-    if (selections.Number() == 0)
+    if (selections.GetCount() == 0)
         return;
@@ -670,4 +670,4 @@
 
-    node = selections.First();
-    wxShape* firstShape = (wxShape*) node->Data();
+    node = selections.GetFirst();
+    wxShape* firstShape = (wxShape*) node->GetData();
 
@@ -678,6 +678,6 @@
 
-    node = selections.First();
+    node = selections.GetFirst();
     while (node)
     {
-        wxShape* shape = (wxShape*) node->Data();
+        wxShape* shape = (wxShape*) node->GetData();
         if (shape != firstShape)
@@ -736,3 +736,3 @@
         }
-        node = node->Next();
+        node = node->GetNext();
     }
@@ -745,3 +745,3 @@
     // amongst the selections.
-    event.Enable( (m_selections.Number() > 1) ) ;
+    event.Enable( (m_selections.GetCount() > 1) ) ;
 }
@@ -753,6 +753,6 @@
 
-    wxNode* node = m_selections.First();
+    wxNode* node = m_selections.GetFirst();
     while (node)
     {
-        wxShape* shape = (wxShape*) node->Data();
+        wxShape* shape = (wxShape*) node->GetData();
         if (shape->IsKindOf(CLASSINFO(wxLineShape)))
@@ -764,3 +764,3 @@
         }
-        node = node->Next();
+        node = node->GetNext();
     }
@@ -774,6 +774,6 @@
 
-    wxNode* node = m_selections.First();
+    wxNode* node = m_selections.GetFirst();
     while (node)
     {
-        wxShape* shape = (wxShape*) node->Data();
+        wxShape* shape = (wxShape*) node->GetData();
         if (shape->IsKindOf(CLASSINFO(wxLineShape)))
@@ -785,3 +785,3 @@
         }
-        node = node->Next();
+        node = node->GetNext();
     }
@@ -795,6 +795,6 @@
 
-    wxNode* node = m_selections.First();
+    wxNode* node = m_selections.GetFirst();
     while (node)
     {
-        wxShape* shape = (wxShape*) node->Data();
+        wxShape* shape = (wxShape*) node->GetData();
         if (shape->IsKindOf(CLASSINFO(wxLineShape)))
@@ -806,3 +806,3 @@
         }
-        node = node->Next();
+        node = node->GetNext();
     }
@@ -815,3 +815,3 @@
     FindSelectedShapes(selections, CLASSINFO(wxLineShape));
-    event.Enable( (selections.Number() > 0) );
+    event.Enable( (selections.GetCount() > 0) );
 }
@@ -822,3 +822,3 @@
     FindSelectedShapes(selections, CLASSINFO(wxLineShape));
-    event.Enable( (selections.Number() > 0) );
+    event.Enable( (selections.GetCount() > 0) );
 }
@@ -829,3 +829,3 @@
     FindSelectedShapes(selections, CLASSINFO(wxLineShape));
-    event.Enable( (selections.Number() > 0) );
+    event.Enable( (selections.GetCount() > 0) );
 }
@@ -1001,6 +1001,6 @@
 
-    wxNode *node = GetDiagram()->GetShapeList()->First();
+    wxNode *node = GetDiagram()->GetShapeList()->GetFirst();
     while (node)
     {
-        wxShape *shape = (wxShape *)node->Data();
+        wxShape *shape = (wxShape *)node->GetData();
         if (shape->GetParent() == NULL && !shape->IsKindOf(CLASSINFO(wxControlPoint)))
@@ -1016,3 +1016,3 @@
         }
-        node = node->Next();
+        node = node->GetNext();
     }
diff -bu1 -r wxWindows/contrib/src/deprecated/proplist.cpp wxWin/contrib/src/deprecated/proplist.cpp
--- wxWindows/contrib/src/deprecated/proplist.cpp	2003-09-11 21:07:28.000000000 +0200
+++ wxWin/contrib/src/deprecated/proplist.cpp	2003-09-25 13:09:27.000000000 +0200
@@ -1033,7 +1033,3 @@
   wxString value(view->GetValueText()->GetValue());
-  bool boolValue = FALSE;
-  if (value == wxT("True"))
-    boolValue = TRUE;
-  else
-    boolValue = FALSE;
+  bool boolValue = (value == wxT("True"));
   property->GetValue() = (bool)boolValue;
@@ -1260,3 +1256,3 @@
     {
-      wxChar *nextString = NULL;
+      wxChar *nextString;
       if (node->GetNext())
diff -bu1 -r wxWindows/contrib/src/deprecated/resource.cpp wxWin/contrib/src/deprecated/resource.cpp
--- wxWindows/contrib/src/deprecated/resource.cpp	2003-09-11 21:07:36.000000000 +0200
+++ wxWin/contrib/src/deprecated/resource.cpp	2003-09-25 19:46:00.000000000 +0200
@@ -166,9 +166,9 @@
 {
-    wxNode *node = m_children.First();
+    wxNode *node = m_children.GetFirst();
     while (node)
     {
-        wxItemResource *item = (wxItemResource *)node->Data();
+        wxItemResource *item = (wxItemResource *)node->GetData();
         delete item;
         delete node;
-        node = m_children.First();
+        node = m_children.GetFirst();
     }
@@ -217,7 +217,6 @@
         BeginFind();
-        wxNode *node = (wxNode *) NULL;
-        node = Next();
+        wxNode *node = Next();
         while (node != NULL)
         {
-            wxItemResource *parent = (wxItemResource *)node->Data();
+            wxItemResource *parent = (wxItemResource *)node->GetData();
             if (parent->GetChildren().Member(item))
@@ -316,3 +315,3 @@
         wxNode *next = Next();
-        wxItemResource *item = (wxItemResource *)node->Data();
+        wxItemResource *item = (wxItemResource *)node->GetData();
         delete item;
@@ -466,7 +465,7 @@
         int noStrings = 0;
-        if (stringList.Number() > 0)
+        if (stringList.GetCount() > 0)
         {
-            noStrings = stringList.Number();
+            noStrings = stringList.GetCount();
             strings = new wxString[noStrings];
-            wxNode *node = stringList.First();
+            wxStringListNode *node = stringList.GetFirst();
             int i = 0;
@@ -474,5 +473,5 @@
             {
-                strings[i] = (wxChar *)node->Data();
+                strings[i] = (wxChar *)node->GetData();
                 i ++;
-                node = node->Next();
+                node = node->GetNext();
             }
@@ -490,7 +489,7 @@
         int noStrings = 0;
-        if (stringList.Number() > 0)
+        if (stringList.GetCount() > 0)
         {
-            noStrings = stringList.Number();
+            noStrings = stringList.GetCount();
             strings = new wxString[noStrings];
-            wxNode *node = stringList.First();
+            wxStringListNode *node = stringList.GetFirst();
             int i = 0;
@@ -498,5 +497,5 @@
             {
-                strings[i] = (wxChar *)node->Data();
+                strings[i] = (wxChar *)node->GetData();
                 i ++;
-                node = node->Next();
+                node = node->GetNext();
             }
@@ -515,7 +514,7 @@
         int noStrings = 0;
-        if (stringList.Number() > 0)
+        if (stringList.GetCount() > 0)
         {
-            noStrings = stringList.Number();
+            noStrings = stringList.GetCount();
             strings = new wxString[noStrings];
-            wxNode *node = stringList.First();
+            wxStringListNode *node = stringList.GetFirst();
             int i = 0;
@@ -523,5 +522,5 @@
             {
-                strings[i] = (wxChar *)node->Data();
+                strings[i] = (wxChar *)node->GetData();
                 i ++;
-                node = node->Next();
+                node = node->GetNext();
             }
@@ -540,7 +539,7 @@
         int noStrings = 0;
-        if (stringList.Number() > 0)
+        if (stringList.GetCount() > 0)
         {
-            noStrings = stringList.Number();
+            noStrings = stringList.GetCount();
             strings = new wxString[noStrings];
-            wxNode *node = stringList.First();
+            wxStringListNode *node = stringList.GetFirst();
             int i = 0;
@@ -548,5 +547,5 @@
             {
-                strings[i] = (wxChar *)node->Data();
+                strings[i] = (wxChar *)node->GetData();
                 i ++;
-                node = node->Next();
+                node = node->GetNext();
             }
@@ -589,6 +588,6 @@
 {
-    wxNode *node = db.First();
+    wxNode *node = db.GetFirst();
     while (node)
     {
-        wxExpr *clause = (wxExpr *)node->Data();
+        wxExpr *clause = (wxExpr *)node->GetData();
         wxString functor(clause->Functor());
@@ -618,3 +617,3 @@
         }
-        node = node->Next();
+        node = node->GetNext();
     }
@@ -1545,3 +1544,3 @@
 {
-    int ch = 0;
+    int ch;
 
@@ -2393,6 +2392,6 @@
         // Try to find optimum bitmap for this platform/colour depth
-        wxNode *node = item->GetChildren().First();
+        wxNode *node = item->GetChildren().GetFirst();
         while (node)
         {
-            wxItemResource *child = (wxItemResource *)node->Data();
+            wxItemResource *child = (wxItemResource *)node->GetData();
             int platform = (int)child->GetValue2();
@@ -2467,3 +2466,3 @@
             }
-            node = node->Next();
+            node = node->GetNext();
         }
@@ -2547,6 +2546,6 @@
         // Try to find optimum icon for this platform/colour depth
-        wxNode *node = item->GetChildren().First();
+        wxNode *node = item->GetChildren().GetFirst();
         while (node)
         {
-            wxItemResource *child = (wxItemResource *)node->Data();
+            wxItemResource *child = (wxItemResource *)node->GetData();
             int platform = (int)child->GetValue2();
@@ -2621,3 +2620,3 @@
             }
-            node = node->Next();
+            node = node->GetNext();
         }
@@ -2688,9 +2687,9 @@
     wxMenu *menu = new wxMenu;
-    wxNode *node = item->GetChildren().First();
+    wxNode *node = item->GetChildren().GetFirst();
     while (node)
     {
-        wxItemResource *child = (wxItemResource *)node->Data();
+        wxItemResource *child = (wxItemResource *)node->GetData();
         if ((child->GetType() != wxT("")) && (child->GetType() == wxT("wxMenuSeparator")))
             menu->AppendSeparator();
-        else if (child->GetChildren().Number() > 0)
+        else if (child->GetChildren().GetCount() > 0)
         {
@@ -2704,3 +2703,3 @@
         }
-        node = node->Next();
+        node = node->GetNext();
     }
@@ -2719,6 +2718,6 @@
             menuBar = new wxMenuBar;
-        wxNode *node = menuResource->GetChildren().First();
+        wxNode *node = menuResource->GetChildren().GetFirst();
         while (node)
         {
-            wxItemResource *child = (wxItemResource *)node->Data();
+            wxItemResource *child = (wxItemResource *)node->GetData();
             wxMenu *menu = wxResourceCreateMenu(child);
@@ -2726,3 +2725,3 @@
                 menuBar->Append(menu, child->GetTitle());
-            node = node->Next();
+            node = node->GetNext();
         }
@@ -2878,3 +2877,3 @@
 {
-    int ch = 0;
+    int ch;
 
@@ -2891,3 +2890,2 @@
             {
-                int prev_ch = ch;
                 ch = getc_string(s);
@@ -2902,3 +2900,3 @@
                     // Eat C comment
-                    prev_ch = 0;
+                    int prev_ch = 0;
                     while ((ch = getc_string(s)) != EOF)
@@ -3253,6 +3251,6 @@
     // Now create children
-    wxNode *node = resource->GetChildren().First();
+    wxNode *node = resource->GetChildren().GetFirst();
     while (node)
     {
-        wxItemResource *childResource = (wxItemResource *)node->Data();
+        wxItemResource *childResource = (wxItemResource *)node->GetData();
 
@@ -3260,3 +3258,3 @@
 
-        node = node->Next();
+        node = node->GetNext();
     }
diff -bu1 -r wxWindows/contrib/src/deprecated/wxexpr.cpp wxWin/contrib/src/deprecated/wxexpr.cpp
--- wxWindows/contrib/src/deprecated/wxexpr.cpp	2003-09-11 21:07:36.000000000 +0200
+++ wxWin/contrib/src/deprecated/wxexpr.cpp	2003-09-25 13:06:25.000000000 +0200
@@ -1022,3 +1022,3 @@
       long functor_key = hash_table->MakeKey(WXSTRINGCAST functor);
-      long value_key = 0;
+      long value_key;
       if (expr && expr->Type() == wxExprString)
diff -bu1 -r wxWindows/contrib/src/fl/bardragpl.cpp wxWin/contrib/src/fl/bardragpl.cpp
--- wxWindows/contrib/src/fl/bardragpl.cpp	2003-04-02 16:11:26.000000000 +0200
+++ wxWin/contrib/src/fl/bardragpl.cpp	2003-09-25 13:05:42.000000000 +0200
@@ -467,3 +467,6 @@
 		{
-			bool success = mpLayout->RedockBar( mpDraggedBar, mHintRect, mpCurPane, FALSE );
+			#ifdef  __WXDEBUG__
+      bool success = 
+      #endif
+                     mpLayout->RedockBar( mpDraggedBar, mHintRect, mpCurPane, FALSE );
 
@@ -521,3 +524,6 @@
 
-			bool success = mpLayout->RedockBar( mpDraggedBar, mHintRect, mpCurPane, FALSE );
+			#ifdef  __WXDEBUG__
+      bool success = 
+      #endif
+                     mpLayout->RedockBar( mpDraggedBar, mHintRect, mpCurPane, FALSE );
 
diff -bu1 -r wxWindows/contrib/src/fl/cbcustom.cpp wxWin/contrib/src/fl/cbcustom.cpp
--- wxWindows/contrib/src/fl/cbcustom.cpp	2003-04-02 16:11:26.000000000 +0200
+++ wxWin/contrib/src/fl/cbcustom.cpp	2003-09-25 13:04:23.000000000 +0200
@@ -169,3 +169,3 @@
 
-		int newState = 0;
+		int newState;
 
diff -bu1 -r wxWindows/contrib/src/fl/controlbar.cpp wxWin/contrib/src/fl/controlbar.cpp
--- wxWindows/contrib/src/fl/controlbar.cpp	2003-08-19 17:50:40.000000000 +0200
+++ wxWin/contrib/src/fl/controlbar.cpp	2003-09-25 18:58:51.000000000 +0200
@@ -89,2 +89,5 @@
 
+/*
+// FIXME:: see places where _gHorizCursorImg is used
+
 static const char* _gHorizCursorImg[] = 
@@ -128,2 +131,3 @@
 };
+*/
 
@@ -448,2 +452,4 @@
 {
+    size_t i;
+
     UnhookFromFrame();
@@ -476,3 +482,2 @@
 
-    size_t i = 0;
     for ( i = 0; i != MAX_PANES; ++i )
@@ -693,3 +698,6 @@
 
-        bool success = LocateBar( pBar, &pRow, &pPane );
+        #ifdef  __WXDEBUG__
+        bool success = 
+        #endif
+                       LocateBar( pBar, &pRow, &pPane );
 
@@ -775,3 +783,3 @@
 
-    int newState = 0;
+    int newState;
 
@@ -1014,3 +1022,2 @@
     mpFrame->GetClientSize( &frmWidth, &frmHeight );
-    int paneHeight = 0;
 
@@ -1030,3 +1037,3 @@
 
-    paneHeight = pPane->GetPaneHeight();
+    int paneHeight = pPane->GetPaneHeight();
 
@@ -1364,7 +1371,2 @@
     }
-    else 
-    {
-        int avoidCompilerWarning = 0;
-        wxASSERT(avoidCompilerWarning); // DBG::
-    }
 }  // wxFrameLayout::ForwardMouseEvent()
@@ -2237,3 +2239,3 @@
 {
-    size_t i = 0;
+    size_t i;
 
@@ -2273,5 +2275,5 @@
 {
-    PaintPaneBackground( dc );
+    size_t i;
 
-    size_t i = 0;
+    PaintPaneBackground( dc );
 
@@ -2627,5 +2629,5 @@
 {
-    int totalWidth = 0;
+    size_t i;
 
-    size_t i = 0;
+    int totalWidth = 0;
 
@@ -2783,3 +2785,3 @@
 {
-    cbRowInfo* pRow = NULL;
+    cbRowInfo* pRow;
 
diff -bu1 -r wxWindows/contrib/src/fl/dyntbar.cpp wxWin/contrib/src/fl/dyntbar.cpp
--- wxWindows/contrib/src/fl/dyntbar.cpp	2003-08-19 17:50:40.000000000 +0200
+++ wxWin/contrib/src/fl/dyntbar.cpp	2003-09-25 12:58:19.000000000 +0200
@@ -376,3 +376,3 @@
                 size_t j;
-                wxDynToolInfo *pInfo = NULL;
+                wxDynToolInfo *pInfo;
                 wxDynToolInfo *pPrevInfo = NULL;
diff -bu1 -r wxWindows/contrib/src/fl/frmview.cpp wxWin/contrib/src/fl/frmview.cpp
--- wxWindows/contrib/src/fl/frmview.cpp	2003-08-19 17:50:40.000000000 +0200
+++ wxWin/contrib/src/fl/frmview.cpp	2003-09-25 12:57:04.000000000 +0200
@@ -43,5 +43,2 @@
     {
-        int o = 0;  //glt
-        ++o;
-
         // TBD::
diff -bu1 -r wxWindows/contrib/src/fl/garbagec.cpp wxWin/contrib/src/fl/garbagec.cpp
--- wxWindows/contrib/src/fl/garbagec.cpp	2003-08-19 17:50:40.000000000 +0200
+++ wxWin/contrib/src/fl/garbagec.cpp	2003-09-25 12:56:01.000000000 +0200
@@ -71,6 +71,3 @@
 
-    int avoidCompilerWarning = 0;
-    wxASSERT(avoidCompilerWarning); // DBG:: item should be present
-
-    return 0;
+    return NULL;
 }
diff -bu1 -r wxWindows/contrib/src/fl/panedrawpl.cpp wxWin/contrib/src/fl/panedrawpl.cpp
--- wxWindows/contrib/src/fl/panedrawpl.cpp	2003-08-19 17:50:40.000000000 +0200
+++ wxWin/contrib/src/fl/panedrawpl.cpp	2003-09-25 12:51:03.000000000 +0200
@@ -296,3 +296,3 @@
 
-            wxCursor* pCurs = NULL;
+            wxCursor* pCurs;
 
diff -bu1 -r wxWindows/contrib/src/fl/rowdragpl.cpp wxWin/contrib/src/fl/rowdragpl.cpp
--- wxWindows/contrib/src/fl/rowdragpl.cpp	2003-08-19 17:50:54.000000000 +0200
+++ wxWin/contrib/src/fl/rowdragpl.cpp	2003-09-25 19:00:40.000000000 +0200
@@ -228,7 +228,2 @@
         {
-            // DBG::
-            wxPoint p = event.mPos;
-            wxPoint d = mDragOrigin;
-//            int dif = event.mPos.x - mDragOrigin.x;
-
             // row is dragged up or down;
@@ -238,7 +233,2 @@
         {
-            // DBG::
-            wxPoint p = event.mPos;
-            wxPoint d = mDragOrigin;
-//            int dif = event.mPos.x - mDragOrigin.x;
-
             // row is dragged left or right
diff -bu1 -r wxWindows/contrib/src/fl/rowlayoutpl.cpp wxWin/contrib/src/fl/rowlayoutpl.cpp
--- wxWindows/contrib/src/fl/rowlayoutpl.cpp	2003-08-19 17:51:04.000000000 +0200
+++ wxWin/contrib/src/fl/rowlayoutpl.cpp	2003-09-25 15:40:49.000000000 +0200
@@ -221,4 +221,4 @@
 {
-    cbBarInfo* pFromBar = NULL;
-    cbBarInfo* pTillBar = NULL;
+    cbBarInfo* pFromBar;
+    cbBarInfo* pTillBar;
 
@@ -402,2 +402,3 @@
 {
+    size_t i;
     double pcntSum = 0;
@@ -406,3 +407,2 @@
 
-    size_t i = 0;
     for ( i = 0; i != pRow->mBars.Count(); ++i )
@@ -1111,4 +1111,2 @@
 
-    int     freeSpc        = 0;
-
     if ( forUpperHandle )
@@ -1122,4 +1120,2 @@
         {
-            freeSpc += pRow->mRowHeight - event.mpPane->GetMinimalRowHeight( pRow );
-
             pRow = pRow->mpPrev;
@@ -1136,4 +1132,2 @@
         {
-            freeSpc += pRow->mRowHeight - mpPane->GetMinimalRowHeight( pRow );
-
             pRow = pRow->mpNext;
diff -bu1 -r wxWindows/contrib/src/ogl/basic.cpp wxWin/contrib/src/ogl/basic.cpp
--- wxWindows/contrib/src/ogl/basic.cpp	2003-08-21 12:26:56.000000000 +0200
+++ wxWin/contrib/src/ogl/basic.cpp	2003-09-25 12:39:52.000000000 +0200
@@ -2066,3 +2066,3 @@
 
-  wxExpr *regionExpr = NULL;
+  wxExpr *regionExpr;
   wxExpr *textExpr = NULL;
@@ -2898,4 +2898,3 @@
 {
-    size_t i = 0;
-    for (i = 0; i < m_lines.GetCount(); i++)
+    for (size_t i = 0; i < m_lines.GetCount(); i++)
         if ((wxLineShape*) (m_lines.Item(i)->GetData()) == line)
diff -bu1 -r wxWindows/contrib/src/ogl/basic2.cpp wxWin/contrib/src/ogl/basic2.cpp
--- wxWindows/contrib/src/ogl/basic2.cpp	2003-08-21 12:26:56.000000000 +0200
+++ wxWin/contrib/src/ogl/basic2.cpp	2003-09-25 12:47:13.000000000 +0200
@@ -205,3 +205,2 @@
 
-//  char buf[300];
   for (i = 1; i < n; i++)
@@ -212,4 +211,2 @@
       isAHit = TRUE;
-//    sprintf(buf, "Line ratio = %.2f, other ratio = %.2f\n", line_ratio, other_ratio);
-//    ClipsErrorFunction(buf);
     lastx = xvec[i];
@@ -228,9 +225,4 @@
       isAHit = TRUE;
-//    sprintf(buf, "Line ratio = %.2f, other ratio = %.2f\n", line_ratio, other_ratio);
-//    ClipsErrorFunction(buf);
 
-    if (line_ratio < min_ratio)
-      min_ratio = line_ratio;
   }
-//  ClipsErrorFunction("\n");
   return isAHit;
diff -bu1 -r wxWindows/contrib/src/ogl/canvas.cpp wxWin/contrib/src/ogl/canvas.cpp
--- wxWindows/contrib/src/ogl/canvas.cpp	2003-08-21 12:26:56.000000000 +0200
+++ wxWin/contrib/src/ogl/canvas.cpp	2003-09-25 12:45:52.000000000 +0200
@@ -451,3 +451,2 @@
         {
-          nearest = dist;
           nearest_object = object;
diff -bu1 -r wxWindows/contrib/src/ogl/composit.cpp wxWin/contrib/src/ogl/composit.cpp
--- wxWindows/contrib/src/ogl/composit.cpp	2003-08-21 12:26:56.000000000 +0200
+++ wxWin/contrib/src/ogl/composit.cpp	2003-09-25 19:15:44.000000000 +0200
@@ -695,7 +695,3 @@
     }
-    int cType = 0;
-    double cXSpacing = 0.0;
-    double cYSpacing = 0.0;
     wxString cName = wxEmptyString;
-    long cId = 0;
     wxShape *m_constrainingObject = NULL;
@@ -714,7 +710,7 @@
 
-    cType = (int)typeExpr->IntegerValue();
-    cXSpacing = xExpr->RealValue();
-    cYSpacing = yExpr->RealValue();
+    int cType = (int)typeExpr->IntegerValue();
+    double cXSpacing = xExpr->RealValue();
+    double cYSpacing = yExpr->RealValue();
     cName = nameExpr->StringValue();
-    cId = idExpr->IntegerValue();
+    long cId = idExpr->IntegerValue();
 
@@ -1182,4 +1178,4 @@
   GetBoundingBoxMax(&maxX, &maxY);
-  double x, y;
-  int direction;
+  double x = 0.0 , y = 0.0;
+  int direction = 0;
 /*
@@ -1587,4 +1583,3 @@
 BEGIN_EVENT_TABLE(OGLPopupDivisionMenu, wxMenu)
-    EVT_CUSTOM_RANGE(wxEVT_COMMAND_MENU_SELECTED,
-                     DIVISION_MENU_SPLIT_HORIZONTALLY,
+    EVT_MENU_RANGE(DIVISION_MENU_SPLIT_HORIZONTALLY,
                      DIVISION_MENU_EDIT_BOTTOM_EDGE,
diff -bu1 -r wxWindows/contrib/src/ogl/divided.cpp wxWin/contrib/src/ogl/divided.cpp
--- wxWindows/contrib/src/ogl/divided.cpp	2003-08-21 12:26:56.000000000 +0200
+++ wxWin/contrib/src/ogl/divided.cpp	2003-09-25 12:44:15.000000000 +0200
@@ -245,14 +245,4 @@
   {
-    int i = 0;
-    bool isLeft = FALSE;
-    if (attachment < (n+1))
-    {
-      i = attachment-1;
-      isLeft = FALSE;
-    }
-    else
-    {
-      i = (totalNumberAttachments - attachment - 1);
-      isLeft = TRUE;
-    }
+    bool isLeft = !(attachment < (n+1));
+    int i = (isLeft) ? (totalNumberAttachments - attachment - 1) : (attachment-1);
     wxNode *node = GetRegions().Item(i);
diff -bu1 -r wxWindows/contrib/src/ogl/drawn.cpp wxWin/contrib/src/ogl/drawn.cpp
--- wxWindows/contrib/src/ogl/drawn.cpp	2003-08-21 12:26:58.000000000 +0200
+++ wxWin/contrib/src/ogl/drawn.cpp	2003-09-25 12:37:28.000000000 +0200
@@ -91,4 +91,3 @@
 
-  int i = 0;
-  for (i = 0; i < 4; i++)
+  for (int i = 0; i < 4; i++)
   {
@@ -244,4 +243,3 @@
   {
-    int i = 0;
-    for (i = 0; i < 4; i++)
+    for (int i = 0; i < 4; i++)
     {
@@ -264,4 +262,3 @@
   {
-    int i = 0;
-    for (i = 0; i < 4; i++)
+    for (int i = 0; i < 4; i++)
     {
@@ -282,4 +279,3 @@
 
-  int i = 0;
-  for (i = 0; i < 4; i++)
+  for (int i = 0; i < 4; i++)
   {
@@ -1263,4 +1259,3 @@
 
-  int i = 0;
-  for (i = 0; i < n; i++)
+  for (int i = 0; i < n; i++)
   {
diff -bu1 -r wxWindows/contrib/src/ogl/lines.cpp wxWin/contrib/src/ogl/lines.cpp
--- wxWindows/contrib/src/ogl/lines.cpp	2003-08-21 12:27:06.000000000 +0200
+++ wxWin/contrib/src/ogl/lines.cpp	2003-09-25 19:16:28.000000000 +0200
@@ -117,4 +117,3 @@
 
-  int i = 0;
-  for (i = 0; i < n; i++)
+  for (int i = 0; i < n; i++)
   {
@@ -600,3 +599,3 @@
   // Position of start point of line, at the end of which we draw the arrow.
-  double startPositionX, startPositionY;
+  double startPositionX = 0.0 , startPositionY = 0.0;
 
@@ -693,3 +692,3 @@
 
-     double theta = 0.0;
+     double theta;
      if (x3 == x1)
@@ -2038,5 +2037,5 @@
 
+  wxArrowHead *currArrow = (wxArrowHead *)currNode->GetData();
   while (refNode && currNode)
   {
-    wxArrowHead *currArrow = (wxArrowHead *)currNode->GetData();
     refArrow = (wxArrowHead *)refNode->GetData();
@@ -2320,3 +2319,3 @@
   int n = m_lineControlPoints->GetCount();
-  int nn = 0;
+  int nn;
   if (m_to == nodeObject)
diff -bu1 -r wxWindows/contrib/src/ogl/mfutils.cpp wxWin/contrib/src/ogl/mfutils.cpp
--- wxWindows/contrib/src/ogl/mfutils.cpp	2003-08-21 12:27:06.000000000 +0200
+++ wxWin/contrib/src/ogl/mfutils.cpp	2003-09-25 19:50:22.000000000 +0200
@@ -35,14 +35,2 @@
 
-static char hexArray[] = { '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'A', 'B',
-  'C', 'D', 'E', 'F' };
-
-static void DecToHex(int dec, char *buf)
-{
-  int firstDigit = (int)(dec/16.0);
-  int secondDigit = (int)(dec - (firstDigit*16.0));
-  buf[0] = hexArray[firstDigit];
-  buf[1] = hexArray[secondDigit];
-  buf[2] = 0;
-}
-
 // 16-bit unsigned integer
@@ -66,3 +54,3 @@
   unsigned long res1 = ((unsigned int) c) + (((unsigned int) c1) << 8);
-  int res = 0;
+  int res;
   if (res1 > 32767)
diff -bu1 -r wxWindows/contrib/src/ogl/ogldiag.cpp wxWin/contrib/src/ogl/ogldiag.cpp
--- wxWindows/contrib/src/ogl/ogldiag.cpp	2003-08-21 12:27:06.000000000 +0200
+++ wxWin/contrib/src/ogl/ogldiag.cpp	2003-09-25 19:17:47.000000000 +0200
@@ -233,3 +233,3 @@
     {
-      wxExpr *expr = NULL;
+      wxExpr *expr;
       if (shape->IsKindOf(CLASSINFO(wxLineShape)))
@@ -241,3 +241,3 @@
     }
-    node = node->Next();
+    node = node->GetNext();
   }
@@ -310,3 +310,3 @@
   // Scan through all clauses and register the ids
-  wxNode *node = database.First();
+  wxNode *node = database.GetFirst();
   while (node)
@@ -317,3 +317,3 @@
     wxRegisterId(id);
-    node = node->Next();
+    node = node->GetNext();
   }
@@ -536,3 +536,3 @@
   {
-    wxNode *node = shape.GetChildren().First();
+    wxNode *node = shape.GetChildren().GetFirst();
     while (node)
@@ -542,3 +542,3 @@
       OnShapeSave(db, *childShape, *childExpr);
-      node = node->Next();
+      node = node->GetNext();
     }
diff -bu1 -r wxWindows/contrib/src/ogl/oglmisc.cpp wxWin/contrib/src/ogl/oglmisc.cpp
--- wxWindows/contrib/src/ogl/oglmisc.cpp	2003-08-26 13:19:00.000000000 +0200
+++ wxWin/contrib/src/ogl/oglmisc.cpp	2003-09-25 13:16:36.000000000 +0200
@@ -362,3 +362,2 @@
   wxNode *current = text_list->GetFirst();
-  int i = 0;
   while (current)
@@ -371,3 +370,2 @@
     current = current->GetNext();
-    i ++;
   }
@@ -415,2 +413,3 @@
         new_line = TRUE; end_word = TRUE; i++;
+        break;
       }

 	  	 
