Index: docs/latex/wx/ipvaddr.tex
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/docs/latex/wx/ipvaddr.tex,v
retrieving revision 1.5
diff -u -r1.5 ipvaddr.tex
--- docs/latex/wx/ipvaddr.tex	2003/01/18 00:16:32	1.5
+++ docs/latex/wx/ipvaddr.tex	2003/02/27 02:49:46
@@ -43,6 +43,16 @@
 Returns the hostname which matches the IP address.
 
 %
+% IPAddress
+%
+
+\membersection{wxIPV4address::IPAddress}
+
+\func{wxString}{IPAddress}{\void}
+
+Returns a wxString containing the IP address in 127.0.0.1 format.
+
+%
 % Service
 %
 
Index: include/wx/sckaddr.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/sckaddr.h,v
retrieving revision 1.17
diff -u -r1.17 sckaddr.h
--- include/wx/sckaddr.h	2003/01/07 13:48:39	1.17
+++ include/wx/sckaddr.h	2003/02/27 02:49:46
@@ -69,6 +69,7 @@
   wxString Hostname();
   wxString OrigHostname() { return m_origHostname; }
   unsigned short Service();
+  wxString IPAddress();
 
   virtual int Type() { return wxSockAddress::IPV4; }
   virtual wxSockAddress *Clone() const;
Index: src/common/sckaddr.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/common/sckaddr.cpp,v
retrieving revision 1.35
diff -u -r1.35 sckaddr.cpp
--- src/common/sckaddr.cpp	2003/01/07 13:48:51	1.35
+++ src/common/sckaddr.cpp	2003/02/27 02:49:57
@@ -181,6 +181,19 @@
     return addr;
 }
 
+wxString wxIPV4address::IPAddress() {	
+	wxString ip_add;
+	unsigned long raw =  GAddress_INET_GetHostAddress(m_address);
+	ip_add.sprintf(
+		"%d.%d.%d.%d",
+		(BYTE)raw&255,
+		(BYTE)(raw>>8)&255,
+		(BYTE)(raw>>16)&255,
+		(BYTE)(raw>>24)&255
+		);
+	return ip_add;
+}
+
 #if 0
 // ---------------------------------------------------------------------------
 // wxIPV6address

 	  	 
