Index: main.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/main.cpp,v
retrieving revision 1.25
diff -u -r1.25 main.cpp
--- main.cpp	2003/07/11 21:49:59	1.25
+++ main.cpp	2003/07/16 15:52:19
@@ -178,7 +178,12 @@
     // copy all the other arguments to wxApp::argv[]
     for ( int i = 1; i < argc; i++ )
     {
-        argv[i] = wxStrdup(args[i - 1]);
+        // Not using wxStrdup here because we are not certain what it uses
+        // for allocating memory. All argv allocations must be done using
+        // new, otherwise there will be memory allocation conflicts later on
+        // when freeing argv.
+        argv[i] = new wxChar[args[i-1].Len()+1];
+        wxStrcpy(argv[i], args[i-1]);
     }
 
     // argv[] must be NULL-terminated
Index: app.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/app.cpp,v
retrieving revision 1.198
diff -u -r1.198 app.cpp
--- app.cpp	2003/07/14 08:41:06	1.198
+++ app.cpp	2003/07/16 15:52:20
@@ -554,11 +554,9 @@
 
     while ( argc )
     {
-        // m_argv elements were allocated by wxStrdup()
-        free(argv[--argc]);
+        delete argv[--argc];
     }
 
-    // but m_argv itself -- using new[]
     delete [] argv;
 }
 

 	  	 
