--- src/html/helpwnd.cpp.orig	2008-02-22 13:18:34.343750000 -0800
+++ src/html/helpwnd.cpp	2008-02-22 13:21:05.859375000 -0800
@@ -836,38 +836,41 @@
         m_SearchText->SetValue(keyword);
         m_SearchButton->Disable();
 
-        if (m_SearchChoice->GetSelection() != 0)
-            book = m_SearchChoice->GetStringSelection();
-
-        wxHtmlSearchStatus status(m_Data, keyword,
-                                  m_SearchCaseSensitive->GetValue(),
-                                  m_SearchWholeWords->GetValue(),
-                                  book);
+        // if keyword is empty, just show the search notebook
+	if (!keyword.IsEmpty()) {
+            if (m_SearchChoice->GetSelection() != 0)
+                book = m_SearchChoice->GetStringSelection();
+
+            wxHtmlSearchStatus status(m_Data, keyword,
+                                      m_SearchCaseSensitive->GetValue(),
+                                      m_SearchWholeWords->GetValue(),
+                                      book);
 
 #if wxUSE_PROGRESSDLG
-        wxProgressDialog progress(_("Searching..."),
-                                  _("No matching page found yet"),
-                                  status.GetMaxIndex(), this,
-                                  wxPD_APP_MODAL | wxPD_CAN_ABORT | wxPD_AUTO_HIDE);
+            wxProgressDialog progress(_("Searching..."),
+                                      _("No matching page found yet"),
+                                      status.GetMaxIndex(), this,
+                                      wxPD_APP_MODAL | wxPD_CAN_ABORT | wxPD_AUTO_HIDE);
 #endif
 
-        int curi;
-        while (status.IsActive())
-        {
-            curi = status.GetCurIndex();
-            if (curi % 32 == 0
+            int curi;
+            while (status.IsActive())
+            {
+                curi = status.GetCurIndex();
+                if (curi % 32 == 0
 #if wxUSE_PROGRESSDLG
-                && !progress.Update(curi)
+                    && !progress.Update(curi)
 #endif
-               )
-                break;
-            if (status.Search())
-            {
-                foundstr.Printf(_("Found %i matches"), ++foundcnt);
+                   )
+                    break;
+                if (status.Search())
+                {
+                    foundstr.Printf(_("Found %i matches"), ++foundcnt);
 #if wxUSE_PROGRESSDLG
-                progress.Update(status.GetCurIndex(), foundstr);
+                    progress.Update(status.GetCurIndex(), foundstr);
 #endif
-                m_SearchList->Append(status.GetName(), (void*)status.GetCurItem());
+                    m_SearchList->Append(status.GetName(), (void*)status.GetCurItem());
+                }
             }
         }
 

 	  	 
