--- src/common/sizer-orig.cpp	2007-02-14 16:34:35.468750000 +0100
+++ src/common/sizer.cpp	2007-02-14 16:36:27.703125000 +0100
@@ -1509,9 +1509,7 @@
             {
                 if (m_growableRows[idx] >= nrows )
                     continue;
-                if (m_rowHeights[ m_growableRows[idx] ] == -1)
-                    m_rowHeights[ m_growableRows[idx] ] = 0;
-                else
+                if (m_rowHeights[ m_growableRows[idx] ] != -1)
                 {
                     int delta = (sz.y - minsz.y);
                     if (sum_proportions == 0)
@@ -1561,9 +1559,7 @@
             {
                 if (m_growableCols[idx] >= ncols )
                     continue;
-                if (m_colWidths[ m_growableCols[idx] ] == -1)
-                    m_colWidths[ m_growableCols[idx] ] = 0;
-                else
+                if (m_colWidths[ m_growableCols[idx] ] != -1)
                 {
                     int delta = (sz.x - minsz.x);
                     if (sum_proportions == 0)

 	  	 
