Index: src/generic/filedlgg.cpp
===================================================================
--- src/generic/filedlgg.cpp	(revision 78079)
+++ src/generic/filedlgg.cpp	(working copy)
@@ -341,6 +341,24 @@
         SetPath( selectedFiles[0] );
     }
 
+    // check that the file [doesn't] exist if necessary
+    if ( HasFdFlag(wxFD_SAVE) && HasFdFlag(wxFD_OVERWRITE_PROMPT) &&
+                wxFileExists( selectedFiles[0] ) )
+    {
+        wxString msg;
+        msg.Printf( _("File '%s' already exists, do you really want to overwrite it?"), selectedFiles[0].c_str() );
+
+        if (wxMessageBox(msg, _("Confirm"), wxYES_NO) != wxYES)
+            return;
+    }
+    else if ( HasFdFlag(wxFD_OPEN) && HasFdFlag(wxFD_FILE_MUST_EXIST) &&
+                    !wxFileExists(selectedFiles[0]) )
+    {
+        wxMessageBox(_("Please choose an existing file."), _("Error"),
+                     wxOK | wxICON_ERROR );
+        return;
+    }
+
     EndModal(wxID_OK);
 }
 
