Index: src/common/prntbase.cpp
===================================================================
--- src/common/prntbase.cpp	(revision 70430)
+++ src/common/prntbase.cpp	(working copy)
@@ -1622,9 +1622,13 @@
     m_modalityKind = wxPreviewFrame_NonModal;
 
     // Give the application icon
-#ifdef __WXMSW__
     wxFrame* topFrame = wxDynamicCast(wxTheApp->GetTopWindow(), wxFrame);
     if (topFrame)
+    {
+        SetSize(topFrame->GetRect());
+    }
+#ifdef __WXMSW__
+    if (topFrame)
         SetIcons(topFrame->GetIcons());
 #endif
 }
