Index: wxWidgets-trunk/src/generic/grideditors.cpp
===================================================================
--- wxWidgets-trunk/src/generic/grideditors.cpp	(revision 65118)
+++ wxWidgets-trunk/src/generic/grideditors.cpp	(working copy)
@@ -485,6 +485,13 @@
     Text()->SetValue(startValue);
     Text()->SetInsertionPointEnd();
     Text()->SetSelection(-1, -1);
+    wxGridCellEditorEvtHandler* evtHandler = NULL;
+    if (m_control)
+        evtHandler = wxDynamicCast(m_control->GetEventHandler(), wxGridCellEditorEvtHandler);
+	
+    // Don't immediately end if we get a kill focus event within BeginEdit
+    if (evtHandler)
+        evtHandler->SetInSetFocus(true);
     Text()->SetFocus();
 }
 
