Index: wxWidgets/src/ribbon/art_msw.cpp
===================================================================
--- wxWidgets/src/ribbon/art_msw.cpp	(revision 73060)
+++ wxWidgets/src/ribbon/art_msw.cpp	(working copy)
@@ -2468,6 +2468,8 @@
                         const wxBitmap& bitmap_large,
                         const wxBitmap& bitmap_small)
 {
+    bool disabled = state & wxRIBBON_BUTTONBAR_BUTTON_DISABLED;
+
     switch(state & wxRIBBON_BUTTONBAR_BUTTON_SIZE_MASK)
     {
     case wxRIBBON_BUTTONBAR_BUTTON_LARGE:
@@ -2487,7 +2489,7 @@
                 {
                     DrawDropdownArrow(dc, rect.x + rect.width / 2,
                         ypos + (label_h * 3) / 2,
-                        m_button_bar_label_colour);
+                        disabled ? m_button_bar_label_disabled_colour : m_button_bar_label_colour);
                 }
             }
             else
@@ -2515,7 +2517,7 @@
                                 DrawDropdownArrow(dc,
                                     iX + 2 +label_w - arrow_width,
                                     ypos + label_h / 2 + 1,
-                                    m_button_bar_label_colour);
+                                    disabled ? m_button_bar_label_disabled_colour : m_button_bar_label_colour);
                             }
                             break;
                         }
@@ -2538,7 +2540,7 @@
             if(kind != wxRIBBON_BUTTON_NORMAL)
             {
                 DrawDropdownArrow(dc, x_cursor, rect.y + rect.height / 2,
-                    m_button_bar_label_colour);
+                    disabled ? m_button_bar_label_disabled_colour : m_button_bar_label_colour);
             }
             break;
         }
