Index: samples/dataview/dataview.cpp
===================================================================
--- samples/dataview/dataview.cpp	(revision 74804)
+++ samples/dataview/dataview.cpp	(working copy)
@@ -618,11 +618,18 @@
                                         MyListModel::Col_EditableText,
                                         wxDATAVIEW_CELL_EDITABLE,
                                         wxCOL_WIDTH_AUTOSIZE);
+
+/*
             m_ctrl[1]->AppendIconTextColumn("icon",
                                             MyListModel::Col_IconText,
                                             wxDATAVIEW_CELL_EDITABLE,
                                             wxCOL_WIDTH_AUTOSIZE);
+*/
 
+            m_ctrl[1]->AppendBitmapColumn("bitmap", MyListModel::Col_IconText, wxDATAVIEW_CELL_INERT, -1, wxALIGN_CENTER ); 
+            m_ctrl[1]->SetRowHeight(30);
+
+
             m_attributes =
                 new wxDataViewColumn("attributes",
                                      new wxDataViewTextRenderer,
Index: samples/dataview/mymodels.cpp
===================================================================
--- samples/dataview/mymodels.cpp	(revision 74804)
+++ samples/dataview/mymodels.cpp	(working copy)
@@ -418,6 +418,7 @@
 
         case Col_IconText:
             {
+/*
                 wxString text;
                 if ( row >= m_iconColValues.GetCount() )
                     text = "virtual icon";
@@ -425,6 +426,8 @@
                     text = m_iconColValues[row];
 
                 variant << wxDataViewIconText(text, m_icon[row % 2]);
+*/
+                variant << m_icon[row % 2];
             }
             break;
 
