Index: src/common/datavcmn.cpp
===================================================================
--- src/common/datavcmn.cpp	(revision 74804)
+++ src/common/datavcmn.cpp	(working copy)
@@ -807,7 +807,10 @@
 
     // adjust the rectangle ourselves to account for the alignment
     wxRect rectItem = rectCell;
-    const int align = GetAlignment();
+    int align = GetAlignment();
+    if ( align == wxDVR_DEFAULT_ALIGNMENT )
+        align = GetOwner()->GetAlignment();
+    
     if ( align != wxDVR_DEFAULT_ALIGNMENT )
     {
         const wxSize size = GetSize();
