Index: dlunix.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/unix/dlunix.cpp,v
retrieving revision 1.4
diff -u -r1.4 dlunix.cpp
--- dlunix.cpp	2005/01/17 02:45:32	1.4
+++ dlunix.cpp	2005/01/18 11:49:54
@@ -325,7 +325,7 @@
 
 #ifdef __LINUX__
     // examine /proc/self/maps to find out what is loaded in our address space
-    wxFFile file("/proc/self/maps");
+    wxFFile file(wxT("/proc/self/maps"));
     if ( file.IsOpened() )
     {
         // details of the module currently being parsed

 	  	 
