Index: src/aui/auibar.cpp
===================================================================
--- src/aui/auibar.cpp	(revision 71211)
+++ src/aui/auibar.cpp	(working copy)
@@ -2641,14 +2641,14 @@
 
             ReleaseMouse();
 
-            GetEventHandler()->ProcessEvent(e);
-
             // Ensure hovered item is really ok, as mouse may have moved during
             // event processing
             wxPoint cursor_pos_after_evt = ScreenToClient(wxGetMousePosition());
             SetHoverItem(FindToolByPosition(cursor_pos_after_evt.x, cursor_pos_after_evt.y));
 
-            DoIdleUpdate();
+            GetEventHandler()->ProcessEvent(e);
+            // get out now asap and do not call DoIdleUpdate() from here, 
+            // this class instance could be gone by now, on id = wxID_CLOSE or wxID_EXIT
         }
         else
             ReleaseMouse();
