diff --git a/include/wx/msw/chkconf.h b/include/wx/msw/chkconf.h
index f83d387..f3d0b0f 100644
--- a/include/wx/msw/chkconf.h
+++ b/include/wx/msw/chkconf.h
@@ -181,17 +181,6 @@
 #endif
 
 /*
-   Win64-specific checks.
- */
-#ifdef __WIN64__
-#   if wxUSE_STACKWALKER
-#       undef wxUSE_CRASHREPORT
-#       define wxUSE_CRASHREPORT 0
-#   endif
-#endif /* __WIN64__ */
-
-
-/*
    Compiler-specific checks.
  */
 
