Index: grid.cpp
===================================================================
RCS file: /home/wxcvs/wxWindows/src/generic/grid.cpp,v
retrieving revision 1.222
diff -b -u -2 -r1.222 grid.cpp
--- grid.cpp	8 Sep 2002 18:00:48 -0000	1.222
+++ grid.cpp	15 Sep 2002 08:52:37 -0000
@@ -9214,10 +9214,13 @@
     // round up the size to a multiple of scroll step - this ensures that we
     // won't get the scrollbars if we're sized exactly to this width
-    wxSize sizeFit(GetScrollX(size.x) * GRID_SCROLL_LINE_X,
-                   GetScrollY(size.y) * GRID_SCROLL_LINE_Y);
+    // CalcDimension adds m_extraWidth + 1 etc. to calculate the necessary
+    // scrollbar steps
+    wxSize sizeFit(GetScrollX(size.x + m_extraWidth + 1) * GRID_SCROLL_LINE_X,
+                   GetScrollY(size.y + m_extraHeight + 1) * GRID_SCROLL_LINE_Y);
 
     // distribute the extra space between teh columns/rows to avoid having
     // extra white space
-    wxCoord diff = sizeFit.x - size.x;
+    // Remove the extra m_extraWidth + 1 added above
+    wxCoord diff = sizeFit.x - size.x + (m_extraWidth + 1);
     if ( diff )
     {
@@ -9244,5 +9247,5 @@
 
     // same for rows
-    diff = sizeFit.y - size.y;
+    diff = sizeFit.y - size.y - (m_extraHeight + 1);
     if ( diff )
     {

 	  	 
