Index: src/common/imaggif.cpp
===================================================================
--- src/common/imaggif.cpp (revision 76952)
+++ src/common/imaggif.cpp (working copy)
@@ -726,7 +726,7 @@
     buf[0] = GIF_MARKER_EXT;    // extension marker
     buf[1] = GIF_MARKER_EXT_GRAPHICS_CONTROL;
     buf[2] = 4;     // length of block
-    buf[3] = (maskIndex != wxNOT_FOUND) ? 1 : 0;   // has transparency
+    buf[3] = (maskIndex != wxNOT_FOUND) ? 1 | 2 << 2 : 0;   // has transparency & disposal method
     buf[4] = delay & 0xff;  // delay time
     buf[5] = (delay >> 8) & 0xff;   // delay time second byte
     buf[6] = (maskIndex != wxNOT_FOUND) ? (wxUint8) maskIndex : 0;
