Index: src/mac/carbon/textctrl.cpp
===================================================================
--- src/mac/carbon/textctrl.cpp	(revision 53318)
+++ src/mac/carbon/textctrl.cpp	(working copy)
@@ -458,7 +458,7 @@
 
     if ( !forceMLTE )
     {
-        if ( m_windowStyle & wxTE_MULTILINE || ( UMAGetSystemVersion() >= 0x1050 ) )
+        if ( m_windowStyle & wxTE_MULTILINE || ( UMAGetSystemVersion() < 0x1050 && !(m_windowStyle & wxTE_PASSWORD) ) )
             m_peer = new wxMacMLTEHIViewControl( this , str , pos , size , style ) ;
     }
 
 	  	 
