Index: wxWindows/src/common/filefn.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/common/filefn.cpp,v
retrieving revision 1.174
diff -b -u -2 -r1.174 filefn.cpp
--- wxWindows/src/common/filefn.cpp	2003/05/24 20:07:19	1.174
+++ wxWindows/src/common/filefn.cpp	2003/06/04 03:36:33
@@ -1075,11 +1075,11 @@
 
 void
-wxDos2UnixFilename (char *s)
+wxDos2UnixFilename (wxChar *s)
 {
   if (s)
     while (*s)
       {
-        if (*s == '\\')
-          *s = '/';
+        if (*s == wxT('\\'))
+          *s = wxT('/');
 #ifdef __WXMSW__
         else

 	  	 
