Index: src/msw/menu.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/msw/menu.cpp,v
retrieving revision 1.147
diff -u -r1.147 menu.cpp
--- src/msw/menu.cpp	2007/03/17 10:25:50	1.147
+++ src/msw/menu.cpp	2007/03/21 19:55:27
@@ -799,9 +799,10 @@
             item->Toggle();
 
         // get the checked status of the menu item: note that menuState is the
-        // old state of the menu, so the test for MF_CHECKED must be inverted
+        // new state of the menu due to the Toggle above, so the test for
+        // MF_CHECKED is normal (NOT inverted)
         UINT menuState = ::GetMenuState(GetHmenu(), id, MF_BYCOMMAND);
-        SendEvent(id, !(menuState & MF_CHECKED));
+        SendEvent(id, (menuState & MF_CHECKED));
     }
 
     return true;

 	  	 
