Index: treectrl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/treectrl.cpp,v
retrieving revision 1.129
diff -b -u -2 -r1.129 treectrl.cpp
--- treectrl.cpp	2002/09/05 19:50:41	1.129
+++ treectrl.cpp	2002/11/01 00:01:55
@@ -2356,8 +2356,10 @@
             break;
 
-        case TVN_SELCHANGED:
+        case TVN_SELCHANGEDW:
+        case TVN_SELCHANGEDA:
             eventType = wxEVT_COMMAND_TREE_SEL_CHANGED;
             // fall through
 
-        case TVN_SELCHANGING:
+        case TVN_SELCHANGINGW:
+        case TVN_SELCHANGINGA:
             {
                 if ( eventType == wxEVT_NULL )
 	  	 
