Index: src/msw/ole/automtn.cpp
===================================================================
--- src/msw/ole/automtn.cpp	(revision 53959)
+++ src/msw/ole/automtn.cpp	(working copy)
@@ -677,7 +677,6 @@
     bool ok = true;
     if ( oleVariant.vt & VT_ARRAY )
     {
-        variant.ClearList();
 
         // Compute the total number of elements in all array dimensions
         int cElements = 1;
@@ -685,24 +684,51 @@
             cElements *= oleVariant.parray->rgsabound[cDims].cElements;
 
         // Get a pointer to the data
-        VARIANTARG* pvdata;
-        HRESULT hr = SafeArrayAccessData(oleVariant.parray, (void **)&pvdata);
+        void* pvdata;
+        HRESULT hr = SafeArrayAccessData(oleVariant.parray, &pvdata);
         if ( FAILED(hr) )
             return false;
 
-        for ( int i = 0; i < cElements; i++ )
-        {
-            VARIANTARG& oleElement = pvdata[i];
-            wxVariant vElement;
-            if ( !wxConvertOleToVariant(oleElement, vElement) )
-            {
-                ok = false;
-                variant.ClearList();
-                break;
-            }
+		switch (oleVariant.vt & VT_TYPEMASK)
+		{
+		case VT_VARIANT:
+			{
+				variant.ClearList();
+				VARIANTARG *variant_data=(VARIANTARG*)pvdata;
+				for ( int i = 0; i < cElements; i++ )
+				{
+					VARIANTARG& oleElement = variant_data[i];
+					wxVariant vElement;
+					if ( !wxConvertOleToVariant(oleElement, vElement) )
+					{
+						ok = false;
+						variant.ClearList();
+						break;
+					}
 
-            variant.Append(vElement);
-        }
+					variant.Append(vElement);
+				}
+			}
+			break;
+		case VT_BSTR:
+			{
+				wxArrayString strings;
+				BSTR *string_val=(BSTR*)pvdata;
+				for ( int i = 0; i < cElements; ++i )
+				{
+					wxString str=wxConvertStringFromOle(*string_val);
+					strings.Add(str);
+					++string_val;
+				}
+				variant=strings;
+			}
+			break;
+		default:
+			wxLogDebug(_T("unhandled VT_ARRAY type %x in wxConvertOleToVariant"), oleVariant.vt & VT_TYPEMASK);
+			variant = wxVariant();
+			ok = false;
+			break;
+		}
 
         SafeArrayUnaccessData(oleVariant.parray);
     }
