diff --git a/docs/doxygen/overviews/xrc_format.h b/docs/doxygen/overviews/xrc_format.h
index 1cf8265..187d867 100644
--- a/docs/doxygen/overviews/xrc_format.h
+++ b/docs/doxygen/overviews/xrc_format.h
@@ -2019,6 +2019,10 @@ No additional properties.
      Label to display on the button (default: empty).}
 @row3col{checked, @ref overview_xrcformat_type_bool,
      Should the button be checked/pressed initially (default: 0)?}
+@row3col{bitmap, @ref overview_xrcformat_type_bitmap,
+    Bitmap to display in the button (optional). @since 3.1.1}
+@row3col{bitmapposition, @c wxLEFT|wxRIGHT|wxTOP|wxBOTTOM,
+    Position of the bitmap in the button, see wxButton::SetBitmapPosition() (default: wxLEFT). @since 3.1.1}
 @endTable
 
 @subsubsection xrc_wxtoolbar wxToolBar
diff --git a/misc/schema/xrc_schema.rnc b/misc/schema/xrc_schema.rnc
index c890724..f9243db 100644
--- a/misc/schema/xrc_schema.rnc
+++ b/misc/schema/xrc_schema.rnc
@@ -1508,7 +1508,9 @@ wxToggleButton =
         stdObjectNodeAttributes &
         stdWindowProperties &
         [xrc:p="important"] element label {_, t_text }* &
-        [xrc:p="o"] element checked {_, t_bool }*
+        [xrc:p="o"] element checked {_, t_bool }* &
+        [xrc:p="o"] element bitmap         {_, t_bitmap }* &
+        [xrc:p="o"] element bitmapposition {_, t_direction }*         
     }
 
 
diff --git a/src/xrc/xh_sttxt.cpp b/src/xrc/xh_sttxt.cpp
index 7bd50c7..c03d67b 100644
--- a/src/xrc/xh_sttxt.cpp
+++ b/src/xrc/xh_sttxt.cpp
@@ -34,6 +34,9 @@ wxStaticTextXmlHandler::wxStaticTextXmlHandler()
     XRC_ADD_STYLE(wxALIGN_CENTRE);
     XRC_ADD_STYLE(wxALIGN_CENTER_HORIZONTAL);
     XRC_ADD_STYLE(wxALIGN_CENTRE_HORIZONTAL);
+    XRC_ADD_STYLE(wxST_ELLIPSIZE_START);
+    XRC_ADD_STYLE(wxST_ELLIPSIZE_MIDDLE);
+    XRC_ADD_STYLE(wxST_ELLIPSIZE_END);
     AddWindowStyles();
 }
 
diff --git a/src/xrc/xh_tglbtn.cpp b/src/xrc/xh_tglbtn.cpp
index 31c146b..2bac11a 100644
--- a/src/xrc/xh_tglbtn.cpp
+++ b/src/xrc/xh_tglbtn.cpp
@@ -78,6 +78,12 @@ void wxToggleButtonXmlHandler::DoCreateToggleButton(wxObject *control)
                    wxDefaultValidator,
                    GetName());
 
+    if ( GetParamNode("bitmap") )
+    {
+        button->SetBitmap(GetBitmap("bitmap", wxART_BUTTON),
+                          GetDirection("bitmapposition"));
+    }
+
     button->SetValue(GetBool( wxT("checked")));
 }
 
