Index: combocmn.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/common/combocmn.cpp,v
retrieving revision 1.54
diff -u -p -t -r1.54 combocmn.cpp
--- combocmn.cpp	2006/12/07 08:59:19	1.54
+++ combocmn.cpp	2006/12/27 19:11:24
@@ -1881,6 +1881,10 @@ void wxComboCtrlBase::ShowPopup()
 
     int rightX = scrPos.x + ctrlSz.x + m_extRight - szp.x;
     int leftX = scrPos.x - m_extLeft;
+
+    if ( wxTheApp->GetLayoutDirection() == wxLayout_RightToLeft )
+        leftX -= ctrlSz.x;
+
     int screenWidth = wxSystemSettings::GetMetric( wxSYS_SCREEN_X );
 
     // If there is not enough horizontal space, anchor on the other side.

 	  	 
