Index: db.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/common/db.cpp,v
retrieving revision 1.137
diff -u -r1.137 db.cpp
--- db.cpp	2006/11/13 14:45:44	1.137
+++ db.cpp	2006/12/13 06:18:27
@@ -2306,6 +2306,7 @@
             return false;
         }
 
+        pColInf[colNum].sqlDataType = Sqllen;
         switch (Sqllen)
         {
 #if wxUSE_UNICODE

 	  	 
