diff --git wx/lib/sized_controls.py wx/lib/sized_controls.py
index 642057b..2127f7a 100644
--- wx/lib/sized_controls.py
+++ wx/lib/sized_controls.py
@@ -774,7 +774,7 @@ class SizedPanel(wx.Panel, SizedParent):
         """
         props = self._DetachFromSizer(sizer)
         if wx.VERSION < (2,8):
-        wx.PyPanel.SetSizer(self, sizer)
+            wx.PyPanel.SetSizer(self, sizer)
         else:
             wx.Panel.SetSizer(self, sizer)
         self._AddToNewSizer(sizer, props)
