Index: include/wx/prntbase.h
===================================================================
--- include/wx/prntbase.h	(revision 67545)
+++ include/wx/prntbase.h	(working copy)
@@ -22,6 +22,7 @@
 #include "wx/scrolwin.h"
 #include "wx/dialog.h"
 #include "wx/frame.h"
+#include "wx/stattext.h"
 #include "wx/dc.h"
 
 class WXDLLIMPEXP_FWD_CORE wxDC;
@@ -453,6 +454,7 @@
     virtual ~wxPreviewControlBar();
 
     virtual void CreateButtons();
+    virtual void SetPageInfo( int minPage, int maxPage );
     virtual void SetZoomControl(int zoom);
     virtual int GetZoomControl();
     virtual wxPrintPreviewBase *GetPrintPreview() const
@@ -496,7 +498,8 @@
     wxPrintPreviewBase*   m_printPreview;
     wxButton*             m_closeButton;
     wxChoice*             m_zoomControl;
-    wxPrintPageTextCtrl* m_currentPageText;
+    wxPrintPageTextCtrl*  m_currentPageText;
+    wxStaticText*         m_maxPageText;
 
     long                  m_buttonFlags;
 
Index: src/common/prntbase.cpp
===================================================================
--- src/common/prntbase.cpp	(revision 67545)
+++ src/common/prntbase.cpp	(working copy)
@@ -36,7 +36,6 @@
     #include "wx/settings.h"
     #include "wx/dcmemory.h"
     #include "wx/dcclient.h"
-    #include "wx/stattext.h"
     #include "wx/intl.h"
     #include "wx/textdlg.h"
     #include "wx/sizer.h"
@@ -1038,6 +1037,13 @@
                 wxCommandEventHandler(wxPrintPageTextCtrl::OnTextEnter));
     }
 
+    // we need to set page number range when OnPreparePrinting is called
+    void SetPageInfo( int minPage, int maxPage )
+    {
+        m_minPage = minPage;
+        m_maxPage = maxPage;
+    }
+
     // Helpers to conveniently set or get the current page number. Return value
     // is 0 if the current controls contents is invalid.
     void SetPageNumber(int page)
@@ -1108,7 +1114,7 @@
 
     wxPreviewControlBar * const m_preview;
 
-    const int m_minPage,
+    int m_minPage,
               m_maxPage;
 
     // This is the last valid page value that we had, we revert to it if an
@@ -1155,6 +1161,7 @@
     m_closeButton = NULL;
     m_zoomControl = NULL;
     m_currentPageText = NULL;
+    m_maxPageText = NULL;
     m_buttonFlags = buttons;
 }
 
@@ -1436,18 +1443,22 @@
 
     if (m_buttonFlags & wxPREVIEW_GOTO)
     {
-        int minPage, maxPage, pageFrom, pageTo;
-        m_printPreview->GetPrintout()->GetPageInfo(&minPage, &maxPage,
-                                                   &pageFrom, &pageTo);
+        int minPage = 1, maxPage = 1;
 
         m_currentPageText = new wxPrintPageTextCtrl(this, minPage, maxPage);
         sizer.Add(m_currentPageText);
 
-        wxStaticText *
-            maxPageText = new wxStaticText(this, wxID_ANY,
-                                           wxString::Format("/ %d", maxPage));
+        // this will need to maintain its size when page totals change so
+        // set to m_currentPageText width and use wxST_NO_AUTORESIZE
+        // and wxALIGN_CENTRE so the space on either side is equal
+        wxSize size(m_currentPageText->GetSize().GetWidth(), wxDefaultCoord);
+        m_maxPageText = new wxStaticText(this, wxID_ANY,
+                                           wxString::Format("/ %d", maxPage),
+                                           wxDefaultPosition,
+                                           size,
+                                           wxST_NO_AUTORESIZE | wxALIGN_CENTRE);
 
-        sizer.Add(maxPageText);
+        sizer.Add(m_maxPageText);
     }
 
     if (m_buttonFlags & wxPREVIEW_NEXT)
@@ -1489,6 +1500,17 @@
     sizer.AddAtEnd(m_closeButton);
 }
 
+// Called when OnPreparePrinting is used to set page range 
+// members and update their display
+void wxPreviewControlBar::SetPageInfo( int minPage, int maxPage )
+{
+    if( m_currentPageText )
+        m_currentPageText->SetPageInfo( minPage, maxPage );
+
+    if( m_maxPageText )
+        m_maxPageText->SetLabel( wxString::Format("/ %d", maxPage) );
+}
+
 void wxPreviewControlBar::SetZoomControl(int zoom)
 {
     if (m_zoomControl)
@@ -1829,7 +1851,7 @@
     m_previewPrintout->SetPageSizePixels(m_pageWidth, m_pageHeight);
 
     // Need to delay OnPreparePrinting() until here, so we have enough
-    // information.
+    // information and a wxDC.
     if (!m_printingPrepared)
     {
         m_previewPrintout->OnPreparePrinting();
@@ -1896,6 +1918,16 @@
         return false;
     }
 
+    // update the wxPreviewControlBar page range display
+    if( m_previewFrame )
+    {
+        wxPreviewControlBar* controlBar = ((wxPreviewFrame*)m_previewFrame)->GetControlBar();
+        if( controlBar )
+        {
+            controlBar->SetPageInfo( pageNum, m_maxPage );
+        }
+    }
+
 #if wxUSE_STATUSBAR
     wxString status;
     if (m_maxPage != 0)
