diff -ur ../wxWidgets-2.6.2+/include/wx/gtk/textctrl.h ./include/wx/gtk/textctrl.h
--- ../wxWidgets-2.6.2+/include/wx/gtk/textctrl.h	2005-08-02 15:57:58.000000000 -0700
+++ ./include/wx/gtk/textctrl.h	2005-10-27 11:46:07.000000000 -0700
@@ -189,13 +189,6 @@
     // common part of all ctors
     void Init();
 
-    // get the vertical adjustment, if any, NULL otherwise
-    GtkAdjustment *GetVAdj() const;
-
-    // scroll the control by the given number of pixels, return true if the
-    // scroll position changed
-    bool DoScroll(GtkAdjustment *adj, int diff);
-
     // Widgets that use the style->base colour for the BG colour should
     // override this and return true.
     virtual bool UseGTKStyleBase() const { return true; }
diff -ur ../wxWidgets-2.6.2+/include/wx/gtk/window.h ./include/wx/gtk/window.h
--- ../wxWidgets-2.6.2+/include/wx/gtk/window.h	2005-08-24 04:17:05.000000000 -0700
+++ ./include/wx/gtk/window.h	2005-10-27 11:46:07.000000000 -0700
@@ -103,6 +103,10 @@
     virtual int GetScrollRange( int orient ) const;
     virtual void ScrollWindow( int dx, int dy,
                                const wxRect* rect = (wxRect *) NULL );
+#if wxABI_VERSION >= 20602
+    virtual bool ScrollLines(int lines);
+    virtual bool ScrollPages(int pages);
+#endif
 
 #if wxUSE_DRAG_AND_DROP
     virtual void SetDropTarget( wxDropTarget *dropTarget );
@@ -295,6 +299,12 @@
     // ApplyWidgetStyle -- override this, not ApplyWidgetStyle
     virtual void DoApplyWidgetStyle(GtkRcStyle *style);
 
+protected:
+#if wxABI_VERSION >= 20602
+    // GtkAdjustment to be used by Scroll{Lines,Pages}
+    void SetVScrollAdjustment(GtkAdjustment* adj);
+#endif
+
 private:
     DECLARE_DYNAMIC_CLASS(wxWindowGTK)
     DECLARE_NO_COPY_CLASS(wxWindowGTK)
diff -ur ../wxWidgets-2.6.2+/src/gtk/scrolbar.cpp ./src/gtk/scrolbar.cpp
--- ../wxWidgets-2.6.2+/src/gtk/scrolbar.cpp	2005-06-08 07:49:36.000000000 -0700
+++ ./src/gtk/scrolbar.cpp	2005-10-27 11:46:07.000000000 -0700
@@ -197,6 +197,10 @@
         m_widget = gtk_hscrollbar_new( (GtkAdjustment *) NULL );
 
     m_adjust = gtk_range_get_adjustment( GTK_RANGE(m_widget) );
+    if ((style & wxSB_VERTICAL) == wxSB_VERTICAL)
+    {
+        SetVScrollAdjustment(m_adjust);
+    }
 
     gtk_signal_connect( GTK_OBJECT(m_adjust),
                         "value_changed",
diff -ur ../wxWidgets-2.6.2+/src/gtk/textctrl.cpp ./src/gtk/textctrl.cpp
--- ../wxWidgets-2.6.2+/src/gtk/textctrl.cpp	2005-08-08 01:53:44.000000000 -0700
+++ ./src/gtk/textctrl.cpp	2005-10-27 11:46:07.000000000 -0700
@@ -736,11 +736,12 @@
     PostCreation(size);
 
     if (multi_line)
-        gtk_widget_show(m_text);
-
-#ifndef __WXGTK20__
-    if (multi_line)
     {
+        gtk_widget_show(m_text);
+#ifdef __WXGTK20__
+        SetVScrollAdjustment(gtk_scrolled_window_get_vadjustment((GtkScrolledWindow*)m_widget));
+#else
+        SetVScrollAdjustment(GTK_TEXT(m_text)->vadj);
         gtk_signal_connect(GTK_OBJECT(GTK_TEXT(m_text)->vadj), "changed",
           (GtkSignalFunc) gtk_scrollbar_changed_callback, (gpointer) this );
 
@@ -755,8 +756,8 @@
 
             draw = wxgtk_text_draw;
         }
-    }
 #endif // GTK+ 1.x
+    }
 
     if (!value.empty())
     {
@@ -2243,78 +2244,18 @@
     //event.Skip(!GetEventHandler()->ProcessEvent(url_event));
     GetEventHandler()->ProcessEvent(url_event);
 }
-#endif // gtk2
-
-// ----------------------------------------------------------------------------
-// scrolling
-// ----------------------------------------------------------------------------
-
-GtkAdjustment *wxTextCtrl::GetVAdj() const
-{
-    if ( !IsMultiLine() )
-        return NULL;
-
-#ifdef __WXGTK20__
-    return gtk_scrolled_window_get_vadjustment(GTK_SCROLLED_WINDOW(m_widget));
-#else
-    return GTK_TEXT(m_text)->vadj;
-#endif
-}
-
-bool wxTextCtrl::DoScroll(GtkAdjustment *adj, int diff)
-{
-    float value = adj->value + diff;
-
-    if ( value < 0 )
-        value = 0;
-
-    float upper = adj->upper - adj->page_size;
-    if ( value > upper )
-        value = upper;
-
-    // did we noticeably change the scroll position?
-    if ( fabs(adj->value - value) < 0.2 )
-    {
-        // well, this is what Robert does in wxScrollBar, so it must be good...
-        return false;
-    }
-
-    adj->value = value;
-
-#ifdef __WXGTK20__
-    gtk_adjustment_value_changed(GTK_ADJUSTMENT(adj));
-#else
-    gtk_signal_emit_by_name(GTK_OBJECT(adj), "value_changed");
-#endif
-
-    return true;
-}
 
 bool wxTextCtrl::ScrollLines(int lines)
 {
-    GtkAdjustment *adj = GetVAdj();
-    if ( !adj )
-        return false;
-
-#ifdef __WXGTK20__
-    int diff = (int)ceil(lines*adj->step_increment);
-#else
-    // this is hardcoded to 10 in GTK+ 1.2 (great idea)
-    int diff = 10*lines;
-#endif
-
-    return DoScroll(adj, diff);
+    return wxTextCtrlBase::ScrollLines(lines);
 }
 
 bool wxTextCtrl::ScrollPages(int pages)
 {
-    GtkAdjustment *adj = GetVAdj();
-    if ( !adj )
-        return false;
-
-    return DoScroll(adj, (int)ceil(pages*adj->page_increment));
+    return wxTextCtrlBase::ScrollPages(pages);
 }
 
+#endif // gtk2
 
 // static
 wxVisualAttributes
diff -ur ../wxWidgets-2.6.2+/src/gtk/window.cpp ./src/gtk/window.cpp
--- ../wxWidgets-2.6.2+/src/gtk/window.cpp	2005-09-26 08:17:28.000000000 -0700
+++ ./src/gtk/window.cpp	2005-10-27 11:46:07.000000000 -0700
@@ -4834,6 +4834,39 @@
     m_clipPaintRegion = false;
 }
 
+static bool wxScrollAdjust(GtkAdjustment* adj, double change)
+{
+    double value_start = adj->value;
+    double value = value_start + change;
+    double upper = adj->upper - adj->page_size;
+    if (value > upper)
+    {
+        value = upper;
+    }
+    // Lower bound will be checked by gtk_adjustment_set_value
+    gtk_adjustment_set_value(adj, value);
+    return adj->value != value_start;
+}
+
+bool wxWindowGTK::ScrollLines(int lines)
+{
+    return
+        m_vAdjust != NULL &&
+        wxScrollAdjust(m_vAdjust, lines * m_vAdjust->step_increment);
+}
+
+bool wxWindowGTK::ScrollPages(int pages)
+{
+    return
+        m_vAdjust != NULL &&
+        wxScrollAdjust(m_vAdjust, pages * m_vAdjust->page_increment);
+}
+
+void wxWindowGTK::SetVScrollAdjustment(GtkAdjustment* adj)
+{
+    wxASSERT(m_vAdjust == NULL);
+    m_vAdjust = adj;
+}
 
 // Find the wxWindow at the current mouse position, also returning the mouse
 // position.
diff -ur ../wxWidgets-2.6.2+/docs/latex/wx/window.tex ./docs/latex/wx/window.tex
--- ../wxWidgets-2.6.2+/docs/latex/wx/window.tex	2005-08-31 14:42:03.000000000 -0700
+++ ./docs/latex/wx/window.tex	2005-10-27 11:46:07.000000000 -0700
@@ -2404,9 +2404,7 @@
 
 \wxheading{Remarks}
 
-This function is currently only implemented under MSW and wxTextCtrl under
-wxGTK (it also works for wxScrolledWindow derived classes under all
-platforms).
+This function is currently only implemented under MSW and wxGTK.
 
 \wxheading{See also}
 

 	  	 
