Index: dataview.cpp
===================================================================
--- dataview.cpp	(revision 70953)
+++ dataview.cpp	(working copy)
@@ -131,6 +131,7 @@
 #endif // wxUSE_DRAG_AND_DROP
 
     void OnDataViewChar(wxKeyEvent& event);
+    void OnDataViewMouse(wxMouseEvent& event);
 
     // helper used by both OnDeleteSelected() and OnDataViewChar()
     void DeleteSelectedItems();
@@ -533,6 +534,10 @@
                                wxKeyEventHandler(MyFrame::OnDataViewChar),
                                NULL, this);
 
+            m_ctrl[0]->Connect(wxEVT_LEFT_DOWN,
+                               wxMouseEventHandler(MyFrame::OnDataViewMouse),
+                               NULL, this);
+
             m_music_model = new MyMusicTreeModel;
             m_ctrl[0]->AssociateModel( m_music_model.get() );
 
@@ -1105,6 +1110,13 @@
         event.Skip();
 }
 
+void MyFrame::OnDataViewMouse(wxMouseEvent& event){
+    wxDataViewItem item;
+    wxDataViewColumn* col = NULL;
+    m_ctrl[0]->HitTest(event.GetPosition(),item,col);
+    wxLogMessage("Left down at X:%i Y:%i Item OK:%i COL OK:%i",event.GetPosition().x,event.GetPosition().y,item.IsOk(),col != NULL);
+}
+
 // ----------------------------------------------------------------------------
 // MyFrame - event handlers for the second page
 // ----------------------------------------------------------------------------
