Index: missing.h
===================================================================
--- missing.h	(revision 50313)
+++ missing.h	(working copy)
@@ -15,6 +15,19 @@
  * The following are required for VC++ 6.
  */
 
+// Needed by window.cpp: AnimateWindow()
+#if (WINVER < 0x0500)
+#define AW_HOR_POSITIVE             0x00000001
+#define AW_HOR_NEGATIVE             0x00000002
+#define AW_VER_POSITIVE             0x00000004
+#define AW_VER_NEGATIVE             0x00000008
+#define AW_CENTER                   0x00000010
+#define AW_HIDE                     0x00010000
+#define AW_ACTIVATE                 0x00020000
+#define AW_SLIDE                    0x00040000
+#define AW_BLEND                    0x00080000
+#endif
+
 // Needed by cursor.cpp
 #ifndef IDC_HAND
     #define IDC_HAND MAKEINTRESOURCE(32649)

 	  	 
