Index: include/wx/vector.h
===================================================================
--- include/wx/vector.h	(revision 51470)
+++ include/wx/vector.h	(working copy)
@@ -102,11 +102,15 @@
 
 template<typename T>
 class wxVector
+#if !defined(__VISUALC__) || wxCHECK_VISUALC_VERSION(7)
     // this cryptic expression means "derive from wxVectorMemOpsMovable if
     // type T is movable type, otherwise derive from wxVectorMemOpsGeneric
     : private wxIf< wxIsMovable<T>::value,
                     wxPrivate::wxVectorMemOpsMovable<T>,
                     wxPrivate::wxVectorMemOpsGeneric<T> >::value
+#else
+    : private wxPrivate::wxVectorMemOpsGeneric<T>
+#endif
 {
 public:
     typedef size_t size_type;

 	  	 
