Index: src/aui/framemanager.cpp
===================================================================
--- src/aui/framemanager.cpp	(revision 81526)
+++ src/aui/framemanager.cpp	(working copy)
@@ -246,6 +246,7 @@
                 const wxSize& size = wxDefaultSize,
                 long style = wxDEFAULT_FRAME_STYLE,
                 const wxString &name = wxT("frame"))
+                    : wxFrame(parent, id, title, pos, size, style, name)
     {
          if (!CreateBase( parent, id, pos, size, style, wxDefaultValidator, name ))
             return;
