Index: src/common/effects.cpp
===================================================================
--- src/common/effects.cpp	(revision 54366)
+++ src/common/effects.cpp	(working copy)
@@ -25,6 +25,8 @@
     #include "wx/gdicmn.h"
 #endif //WX_PRECOMP
 
+#if WXWIN_COMPATIBILITY_2_8
+
 /*
  * wxEffectsImpl: various 3D effects
  */
@@ -122,3 +124,6 @@
 
     return true;
 }
+
+#endif // WXWIN_COMPATIBILITY_2_8
+
Index: include/wx/event.h
===================================================================
--- include/wx/event.h	(revision 54366)
+++ include/wx/event.h	(working copy)
@@ -1370,7 +1370,7 @@
     // return true if the window was shown, false if hidden
     bool IsShown() const { return m_show; }
 
-#ifdef WXWIN_COMPATIBILITY_2_8
+#if WXWIN_COMPATIBILITY_2_8
     wxDEPRECATED( bool GetShow() const { return IsShown(); } )
 #endif
 
@@ -1397,7 +1397,7 @@
         : wxEvent(event)
     { m_iconized = event.m_iconized; }
 
-#ifdef WXWIN_COMPATIBILITY_2_8
+#if WXWIN_COMPATIBILITY_2_8
     wxDEPRECATED( bool Iconized() const { return IsIconized(); } )
 #endif
     // return true if the frame was iconized, false if restored
Index: include/wx/effects.h
===================================================================
--- include/wx/effects.h	(revision 54366)
+++ include/wx/effects.h	(working copy)
@@ -17,7 +17,7 @@
 //
 // please use wxRenderer::DrawBorder() instead of DrawSunkenEdge(); there is no
 // replacement for TileBitmap() but it doesn't seem to be very useful anyhow
-#ifdef WXWIN_COMPATIBILITY_2_8
+#if WXWIN_COMPATIBILITY_2_8
 
 /*
  * wxEffects: various 3D effects
Index: include/wx/iconbndl.h
===================================================================
--- include/wx/iconbndl.h	(revision 54366)
+++ include/wx/iconbndl.h	(working copy)
@@ -84,7 +84,7 @@
     // check if we have any icons at all
     bool IsEmpty() const { return GetIconCount() == 0; }
 
-#ifdef WXWIN_COMPATIBILITY_2_8
+#if WXWIN_COMPATIBILITY_2_8
     wxDEPRECATED( void AddIcon(const wxString& file, long type)
         {
             AddIcon(file, (long)type);
Index: include/wx/image.h
===================================================================
--- include/wx/image.h	(revision 54366)
+++ include/wx/image.h	(working copy)
@@ -112,7 +112,7 @@
     wxBitmapType GetType() const { return m_type; }
     const wxString& GetMimeType() const { return m_mime; }
 
-#ifdef WXWIN_COMPATIBILITY_2_8
+#if WXWIN_COMPATIBILITY_2_8
     wxDEPRECATED(
         void SetType(long type) { SetType((wxBitmapType)type); }
     )
@@ -434,7 +434,7 @@
     static HSVValue RGBtoHSV(const RGBValue& rgb);
     static RGBValue HSVtoRGB(const HSVValue& hsv);
 
-#ifdef WXWIN_COMPATIBILITY_2_8
+#if WXWIN_COMPATIBILITY_2_8
     wxDEPRECATED(
         wxImage(const wxString& name, long type, int index = -1)
         {
