Index: drawing.cpp
===================================================================
--- drawing.cpp	(revision 75200)
+++ drawing.cpp	(working copy)
@@ -1322,6 +1322,31 @@
     r4.Deflate(1);
     dc.GradientFillLinear(r4, *wxBLACK, *wxGREEN, wxWEST);
 
+    wxBitmap bmpDest;
+    bmpDest.LoadFile( wxT("demo_dest.png"), wxBITMAP_TYPE_PNG );
+    
+    wxBitmap bmpSource;
+    bmpSource.LoadFile( wxT("demo_source.png"), wxBITMAP_TYPE_PNG );
+
+
+    wxMemoryDC dcDemo;
+    dcDemo.SelectObject(bmpDest);
+
+    wxGraphicsRenderer* renderer = wxGraphicsRenderer::GetCairoRenderer();
+    wxGraphicsContext *pGCDemo = renderer->CreateContext(dcDemo);
+    if (pGCDemo)
+    {
+        int w = bmpDest.GetWidth();
+        int h = bmpDest.GetHeight();
+
+        pGCDemo->SetCompositionMode(wxCOMPOSITION_OVER);
+        pGCDemo->SetInterpolationQuality(wxINTERPOLATION_BEST);
+        pGCDemo->DrawBitmap(bmpSource, 0, 0, w, h);
+    }
+    dcDemo.SelectObject(wxNullBitmap);
+
+    dc.DrawBitmap(bmpDest, 300, 10, true); 
+
 #if wxUSE_GRAPHICS_CONTEXT
     if (m_useContext)
     {
