--- src/generic/grid_orig.cpp	Fri Sep 20 15:08:02 2002
+++ src/generic/grid.cpp	Tue Oct 15 10:07:05 2002
@@ -3844,4 +3844,5 @@
     m_colLabelHorizAlign = wxALIGN_CENTRE;
     m_colLabelVertAlign  = wxALIGN_TOP;
+    m_colLabelTextOrientation = wxHORIZONTAL;
 
     m_defaultColWidth  = WXGRID_DEFAULT_COL_WIDTH;
@@ -4643,9 +4644,18 @@
     else if (event.LeftDClick() )
     {
-        if ( YToEdgeOfRow(y) < 0 )
+        int row = YToEdgeOfRow(y);
+        if ( row < 0 )
         {
             row = YToRow(y);
             SendEvent( wxEVT_GRID_LABEL_LEFT_DCLICK, row, -1, event );
         }
+        else
+        {
+            // adjust row height depending on label text
+            AutoSizeRowLabelSize( row );
+
+            ChangeCursorMode(WXGRID_CURSOR_SELECT_CELL, m_colLabelWin);
+            m_dragLastPos  = -1;
+        }
     }
 
@@ -4849,9 +4859,18 @@
     if ( event.LeftDClick() )
     {
-        if ( XToEdgeOfCol(x) < 0 )
+        int col = XToEdgeOfCol(x);
+        if ( col < 0 )
         {
             col = XToCol(x);
             SendEvent( wxEVT_GRID_LABEL_LEFT_DCLICK, -1, col, event );
         }
+        else
+        {
+            // adjust column width depending on label text
+            AutoSizeColLabelSize( col );
+
+            ChangeCursorMode(WXGRID_CURSOR_SELECT_CELL, m_colLabelWin);
+            m_dragLastPos  = -1;
+        }
     }
 
@@ -6920,6 +6939,7 @@
     dc.SetFont( GetLabelFont() );
 
-    int hAlign, vAlign;
+    int hAlign, vAlign, orient;
     GetColLabelAlignment( &hAlign, &vAlign );
+    orient = GetColLabelTextOrientation();
 
     wxRect rect;
@@ -6928,5 +6948,5 @@
     rect.SetWidth( GetColWidth(col) - 4 );
     rect.SetHeight( m_colLabelHeight - 4 );
-    DrawTextRectangle( dc, GetColLabelValue( col ), rect, hAlign, vAlign );
+    DrawTextRectangle( dc, GetColLabelValue( col ), rect, hAlign, vAlign, orient );
 }
 
@@ -6935,5 +6955,6 @@
                                 const wxRect& rect,
                                 int horizAlign,
-                                int vertAlign )
+                                int vertAlign,
+                                int textOrientation )
 {
     wxArrayString lines;
@@ -6947,5 +6968,6 @@
                         rect,
                         horizAlign,
-                        vertAlign );
+                        vertAlign,
+                        textOrientation );
 
 }
@@ -6955,5 +6977,6 @@
                                 const wxRect& rect,
                                 int horizAlign,
-                                int vertAlign )
+                                int vertAlign,
+                                int textOrientation )
 {
     long textWidth, textHeight;
@@ -6963,6 +6986,14 @@
     if ( lines.GetCount() )
     {
+        if( textOrientation == wxHORIZONTAL )
+        {
         GetTextBoxSize( dc, lines, &textWidth, &textHeight );
         dc.GetTextExtent( lines[0], &lineWidth, &lineHeight );
+        }
+        else
+        {
+            GetTextBoxSize( dc, lines, &textHeight, &textWidth );
+            dc.GetTextExtent( lines[0], &lineHeight, &lineWidth );
+        }
 
         float x, y;
@@ -6970,14 +7001,23 @@
         {
             case wxALIGN_RIGHT:
+                if( textOrientation == wxHORIZONTAL )
                 x = rect.x + (rect.width - textWidth - 1);
+                else
+                    y = rect.y + lineHeight + 1;
                 break;
 
             case wxALIGN_CENTRE:
+                if( textOrientation == wxHORIZONTAL )
                 x = rect.x + ((rect.width - textWidth)/2);
+                else
+                    y = rect.y + rect.height - ((rect.height - textHeight)/2);
                 break;
 
             case wxALIGN_LEFT:
             default:
+                if( textOrientation == wxHORIZONTAL )
                 x = rect.x + 1;
+                else
+                    y = rect.y + rect.height - 1;
                 break;
         }
@@ -6986,14 +7026,23 @@
         {
             case wxALIGN_BOTTOM:
+                if( textOrientation == wxHORIZONTAL )
                 y = rect.y + (rect.height - textHeight - 1);
+                else
+                    x = rect.x + rect.width - textWidth;
                 break;
 
             case wxALIGN_CENTRE:
+                if( textOrientation == wxHORIZONTAL )
                 y = rect.y + ((rect.height - textHeight)/2);
+                else
+                    x = rect.x + ((rect.width - textWidth)/2);
                 break;
 
             case wxALIGN_TOP:
             default:
+                if( textOrientation == wxHORIZONTAL )
                 y = rect.y + 1;
+                else
+                    x = rect.x + 1;
                 break;
         }
@@ -7001,7 +7050,15 @@
         for ( size_t i = 0;  i < lines.GetCount();  i++ )
         {
+            if( textOrientation == wxHORIZONTAL )
+            {
             dc.DrawText( lines[i], (int)x, (int)y );
             y += lineHeight;
         }
+            else
+            {
+                dc.DrawRotatedText( lines[i], (int)x, (int)y, 90.0 );
+                x += lineWidth;
+            }
+        }
     }
 
@@ -8122,4 +8179,9 @@
 }
 
+int wxGrid::GetColLabelTextOrientation()
+{
+    return m_colLabelTextOrientation;
+}
+
 wxString wxGrid::GetRowLabelValue( int row )
 {
@@ -8304,4 +8366,24 @@
 }
 
+// Note: 	under MSW, the default column label font must be changed because it
+//				does not support vertical printing
+//
+// Example: wxFont font(9, wxSWISS, wxNORMAL, wxBOLD);
+//				pGrid->SetLabelFont(font);
+//    		pGrid->SetColLabelTextOrientation(wxVERTICAL);
+//
+void wxGrid::SetColLabelTextOrientation( int textOrientation )
+{
+    if( textOrientation == wxHORIZONTAL || textOrientation == wxVERTICAL )
+    {
+        m_colLabelTextOrientation = textOrientation;
+    }
+
+    if ( !GetBatchCount() )
+    {
+        m_colLabelWin->Refresh();
+    }
+}
+
 void wxGrid::SetRowLabelValue( int row, const wxString& s )
 {
@@ -9161,5 +9243,9 @@
 
     if ( column )
+    {
         dc.GetTextExtent( GetColLabelValue(col), &w, &h );
+        if( GetColLabelTextOrientation() == wxVERTICAL )
+            w = h;
+    }
     else
         dc.GetTextExtent( GetRowLabelValue(row), &w, &h );
@@ -9340,4 +9426,37 @@
 
     SetClientSize(sizeFit);
+}
+
+void wxGrid::AutoSizeRowLabelSize( int row )
+{
+    wxArrayString lines;
+    long w, h;
+
+    // autosize row height depending on label text
+    StringToLines( GetRowLabelValue( row ), lines );
+    wxClientDC dc( m_rowLabelWin );
+    GetTextBoxSize( dc, lines, &w, &h);
+    if( h < m_defaultRowHeight )
+        h = m_defaultRowHeight;
+    SetRowSize(row, h);
+    ForceRefresh();
+}
+
+void wxGrid::AutoSizeColLabelSize( int col )
+{
+    wxArrayString lines;
+    long w, h;
+
+    // autosize column width depending on label text
+    StringToLines( GetColLabelValue( col ), lines );
+    wxClientDC dc( m_colLabelWin );
+    if( GetColLabelTextOrientation() == wxHORIZONTAL )
+        GetTextBoxSize( dc, lines, &w, &h);
+    else
+        GetTextBoxSize( dc, lines, &h, &w);
+    if( w < m_defaultColWidth )
+        w = m_defaultColWidth;
+    SetColSize(col, w);
+    ForceRefresh();
 }
 
--- include/wx/generic/grid_orig.h	Sat Sep 14 02:09:21 2002
+++ include/wx/generic/grid.h	Tue Oct 15 10:09:51 2002
@@ -1092,9 +1092,11 @@
     void DrawTextRectangle( wxDC& dc, const wxString&, const wxRect&,
                             int horizontalAlignment = wxALIGN_LEFT,
-                            int verticalAlignment = wxALIGN_TOP );
+                            int verticalAlignment = wxALIGN_TOP,
+                            int textOrientation = wxHORIZONTAL );
 
     void DrawTextRectangle( wxDC& dc, const wxArrayString& lines, const wxRect&,
                             int horizontalAlignment = wxALIGN_LEFT,
-                            int verticalAlignment = wxALIGN_TOP );
+                            int verticalAlignment = wxALIGN_TOP,
+                            int textOrientation = wxHORIZONTAL );
 
 
@@ -1208,4 +1210,5 @@
     void     GetRowLabelAlignment( int *horiz, int *vert );
     void     GetColLabelAlignment( int *horiz, int *vert );
+    int      GetColLabelTextOrientation();
     wxString GetRowLabelValue( int row );
     wxString GetColLabelValue( int col );
@@ -1222,4 +1225,5 @@
     void     SetRowLabelAlignment( int horiz, int vert );
     void     SetColLabelAlignment( int horiz, int vert );
+    void     SetColLabelTextOrientation( int textOrientation );
     void     SetRowLabelValue( int row, const wxString& );
     void     SetColLabelValue( int col, const wxString& );
@@ -1298,4 +1302,11 @@
     void     AutoSize();
 
+    // autosize row height depending on label text
+    void     AutoSizeRowLabelSize( int row );
+
+    // autosize column width depending on label text
+    void     AutoSizeColLabelSize( int col );
+
+
     // column won't be resized to be lesser width - this must be called during
     // the grid creation because it won't resize the column if it's already
@@ -1676,4 +1687,5 @@
     int        m_colLabelHorizAlign;
     int        m_colLabelVertAlign;
+    int        m_colLabelTextOrientation;
 
     bool       m_defaultRowLabelValues;

 	  	 
