Index: src/mac/carbon/dataobj.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/mac/carbon/dataobj.cpp,v
retrieving revision 1.31
diff -b -u -2 -r1.31 dataobj.cpp
--- src/mac/carbon/dataobj.cpp	2006/03/23 22:04:57	1.31
+++ src/mac/carbon/dataobj.cpp	2006/04/08 15:15:10
@@ -251,5 +251,5 @@
     filenames = wxString( (const char*)pBuf, *wxConvFileName );
 #else
-    filenames = wxString( wxConvFileName->cMB2WX( pBuf ), wxConvLocal );
+    filenames = wxString( wxConvFileName->cMB2WX( (const char*)pBuf ), wxConvLocal );
 #endif
 

 	  	 
