diff -rbu2 ./cvs/wx2/wxwindows/src/msw/mdi.cpp ./wx2/src/msw/mdi.cpp
--- ./cvs/wx2/wxwindows/src/msw/mdi.cpp	Fri Jun 15 18:08:18 2001
+++ ./wx2/src/msw/mdi.cpp	Fri Jun 15 19:38:18 2001
@@ -980,5 +980,8 @@
 
         HMENU parent_menu = (HMENU)parent->GetWinMenu();
-        if ( parent_menu )
+
+        // activate the the parent menu only when there is no other child
+        // that has been activated
+        if ( parent_menu && !hwndAct )
         {
             parent->m_parentFrameActive = TRUE;

 	  	 
