diff --git a/samples/minimal/minimal.cpp b/samples/minimal/minimal.cpp
index f765a70..7472206 100644
--- a/samples/minimal/minimal.cpp
+++ b/samples/minimal/minimal.cpp
@@ -140,6 +140,8 @@ bool MyApp::OnInit()
 // main frame
 // ----------------------------------------------------------------------------
 
+#include <wx/dataview.h>
+
 // frame constructor
 MyFrame::MyFrame(const wxString& title)
        : wxFrame(NULL, wxID_ANY, title)
@@ -147,6 +149,36 @@ MyFrame::MyFrame(const wxString& title)
     // set the frame icon
     SetIcon(wxICON(sample));
 
+    wxBoxSizer* boxSizer1 = new wxBoxSizer(wxVERTICAL);
+    SetSizer(boxSizer1);
+    wxDataViewListCtrl* m_pList = new wxDataViewListCtrl(this, wxID_ANY, wxDefaultPosition, wxSize(-1,-1), wxDV_ROW_LINES|wxDV_MULTIPLE);
+    boxSizer1->Add(m_pList, 1, wxALL|wxEXPAND, 5);
+    m_pList->AppendTextColumn(_("Column 1"), wxDATAVIEW_CELL_INERT, -2, wxALIGN_LEFT);
+    m_pList->AppendTextColumn(_("Column 2"), wxDATAVIEW_CELL_INERT, -2, wxALIGN_LEFT);
+    m_pList->GetColumn(0)->SetSortable(true);
+    m_pList->GetColumn(1)->SetSortable(true);
+
+    wxVector<wxVariant> row;
+    
+    row.push_back("1");
+    row.push_back("Hello");
+    m_pList->AppendItem(row);
+    
+    row.clear();
+    row.push_back("2");
+    row.push_back("World");
+    m_pList->AppendItem(row);
+
+    row.clear();
+    row.push_back("3");
+    row.push_back("From");
+    m_pList->AppendItem(row);
+
+    row.clear();
+    row.push_back("4");
+    row.push_back("wxWidgets");
+    m_pList->AppendItem(row);
+
 #if wxUSE_MENUS
     // create a menu bar
     wxMenu *fileMenu = new wxMenu;
