Index: src/osx/cocoa/utils.mm
===================================================================
--- src/osx/cocoa/utils.mm	(revision 57954)
+++ src/osx/cocoa/utils.mm	(working copy)
@@ -61,8 +61,10 @@
 
 bool wxApp::DoInitGui()
 {
-    [NSApplication sharedApplication];
-    [NSApp finishLaunching];
+    if (NSApp == NULL) {
+        [NSApplication sharedApplication];
+        [NSApp finishLaunching];
+    }
     return true;
 }
 
Index: src/osx/core/hid.cpp
===================================================================
--- src/osx/core/hid.cpp	(revision 57954)
+++ src/osx/core/hid.cpp	(working copy)
@@ -651,6 +651,7 @@
         {
             for(size_t i = 0; i < sm_keyboards.GetCount(); ++i)
                 delete (wxHIDKeyboard*) sm_keyboards[i];
+            sm_keyboards.Clear();
         }
 };
 
