From 8125c8dd57e9c9864f6dbd455f3dee8e5ebebd8d Mon Sep 17 00:00:00 2001
From: Artur Wieczorek <artwik@wp.pl>
Date: Sun, 4 May 2014 12:48:07 +0200
Subject: [PATCH 4/4] Handle 'key pressed' state in owner drawn wxCheckBox.

---
 src/msw/checkbox.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/msw/checkbox.cpp b/src/msw/checkbox.cpp
index dc0aa10..df335c8 100644
--- a/src/msw/checkbox.cpp
+++ b/src/msw/checkbox.cpp
@@ -396,6 +396,9 @@ bool wxCheckBox::MSWOnDraw(WXDRAWITEMSTRUCT *item)
             break;
     }
 
+    if( m_isPressed )
+        flags |= wxCONTROL_PRESSED;
+
     if ( wxFindWindowAtPoint(wxGetMousePosition()) == this )
         flags |= wxCONTROL_CURRENT;
 
-- 
1.9.2.msysgit.0

