diff --git a/src/msw/checkbox.cpp b/src/msw/checkbox.cpp
index b5601e7..8d4772d 100644
--- a/src/msw/checkbox.cpp
+++ b/src/msw/checkbox.cpp
@@ -379,8 +379,9 @@ bool wxCheckBox::MSWOnDraw(WXDRAWITEMSTRUCT *item)
          rectLabel;
     rectLabel.top = rect.top + (rect.bottom - rect.top - GetBestSize().y) / 2;
     rectLabel.bottom = rectLabel.top + GetBestSize().y;
+    // With these parameters the checkbox is most similar to the native one.
     const int MARGIN = 3;
-    const int CXMENUCHECK = ::GetSystemMetrics(SM_CXMENUCHECK);
+    const int CXMENUCHECK = ::GetSystemMetrics(SM_CXMENUCHECK)+1;
     // the space between the checkbox and the label is included in the
     // check-mark bitmap
     const int checkSize = wxMin(CXMENUCHECK - MARGIN, GetSize().y);
@@ -434,6 +435,9 @@ bool wxCheckBox::MSWOnDraw(WXDRAWITEMSTRUCT *item)
             break;
     }
 
+    if( m_isPressed )
+        flags |= wxCONTROL_PRESSED;
+
     if ( wxFindWindowAtPoint(wxGetMousePosition()) == this )
         flags |= wxCONTROL_CURRENT;
 
