--- wxWidgets.orig/src/generic/odcombo.cpp	Thu May 25 19:04:37 2006
+++ wxWidgets/src/generic/odcombo.cpp	Thu May 25 19:43:48 2006
@@ -695,5 +695,7 @@
 unsigned int wxOwnerDrawnComboBox::GetCount() const
 {
-    wxASSERT_MSG( m_popupInterface, wxT("no popup interface") );
+    if ( !m_popupInterface )
+        return m_initChs.GetCount();
+
     return m_popupInterface->GetCount();
 }
@@ -702,4 +704,8 @@
 {
     wxCHECK_MSG( IsValid(n), wxEmptyString, _T("invalid index in wxOwnerDrawnComboBox::GetString") );
+
+    if ( !m_popupInterface )
+        return m_initChs.Item(n);
+
     return m_popupInterface->GetString(n);
 }
@@ -707,5 +713,8 @@
 void wxOwnerDrawnComboBox::SetString(unsigned int n, const wxString& s)
 {
+    EnsurePopupControl();
+
     wxCHECK_RET( IsValid(n), _T("invalid index in wxOwnerDrawnComboBox::SetString") );
+
     m_popupInterface->SetString(n,s);
 }
@@ -713,5 +722,7 @@
 int wxOwnerDrawnComboBox::FindString(const wxString& s) const
 {
-    wxASSERT_MSG( m_popupInterface, wxT("no popup interface") );
+    if ( !m_popupInterface )
+        return m_initChs.Index(s);
+
     return m_popupInterface->FindString(s);
 }
@@ -719,7 +730,8 @@
 void wxOwnerDrawnComboBox::Select(int n)
 {
-    wxCHECK_RET( (n == wxNOT_FOUND) || IsValid(n), _T("invalid index in wxOwnerDrawnComboBox::Select") );
     EnsurePopupControl();
 
+    wxCHECK_RET( (n == wxNOT_FOUND) || IsValid(n), _T("invalid index in wxOwnerDrawnComboBox::Select") );
+
     m_popupInterface->SetSelection(n);
 
@@ -739,5 +751,7 @@
 int wxOwnerDrawnComboBox::GetSelection() const
 {
-    wxASSERT_MSG( m_popupInterface, wxT("no popup interface") );
+    if ( !m_popupInterface )
+        return m_initChs.Index(m_valueString);
+
     return m_popupInterface->GetSelection();
 }
@@ -769,5 +783,7 @@
 void* wxOwnerDrawnComboBox::DoGetItemClientData(unsigned int n) const
 {
-    wxASSERT_MSG( m_popupInterface, wxT("no popup interface") );
+    if ( !m_popupInterface )
+        return NULL;
+
     return m_popupInterface->GetItemClientData(n);
 }

 	  	 
