Index: include/wx/animate.h
===================================================================
--- include/wx/animate.h	(revision 53571)
+++ include/wx/animate.h	(working copy)
@@ -75,6 +75,8 @@
     // public API
     virtual bool LoadFile(const wxString& filename,
                           wxAnimationType type = wxANIMATION_TYPE_ANY) = 0;
+    virtual bool Load(wxInputStream& stream,
+                      wxAnimationType type = wxANIMATION_TYPE_ANY) = 0;
 
     virtual void SetAnimation(const wxAnimation &anim) = 0;
     virtual wxAnimation GetAnimation() const = 0;
Index: include/wx/generic/animate.h
===================================================================
--- include/wx/generic/animate.h	(revision 53571)
+++ include/wx/generic/animate.h	(working copy)
@@ -97,6 +97,7 @@
 
 public:
     virtual bool LoadFile(const wxString& filename, wxAnimationType type = wxANIMATION_TYPE_ANY);
+    virtual bool Load(wxInputStream& stream, wxAnimationType type = wxANIMATION_TYPE_ANY);
 
     virtual void Stop();
     virtual bool Play()
Index: include/wx/gtk/animate.h
===================================================================
--- include/wx/gtk/animate.h	(revision 53571)
+++ include/wx/gtk/animate.h	(working copy)
@@ -113,6 +113,7 @@
 public:     // public API
 
     virtual bool LoadFile(const wxString& filename, wxAnimationType type = wxANIMATION_TYPE_ANY);
+    virtual bool Load(wxInputStream& stream, wxAnimationType type = wxANIMATION_TYPE_ANY);
 
     virtual void SetAnimation(const wxAnimation &anim);
     virtual wxAnimation GetAnimation() const
Index: interface/animate.h
===================================================================
--- interface/animate.h	(revision 53571)
+++ interface/animate.h	(working copy)
@@ -124,6 +124,13 @@
                           wxAnimationType animType = wxANIMATION_TYPE_ANY);
 
     /**
+        Loads the animation from the given stream and calls SetAnimation().
+        See wxAnimation::Load for more info.
+    */
+    virtual bool Load(wxInputStream& file,
+                      wxAnimationType animType = wxANIMATION_TYPE_ANY);
+
+    /**
         Starts playing the animation.
 
         The animation is always played in loop mode (unless the last frame of the
Index: src/generic/animateg.cpp
===================================================================
--- src/generic/animateg.cpp	(revision 53571)
+++ src/generic/animateg.cpp	(working copy)
@@ -308,9 +308,14 @@
 
 bool wxAnimationCtrl::LoadFile(const wxString& filename, wxAnimationType type)
 {
+    wxFileInputStream fis(filename);
+    return Load(fis, type);
+}
+
+bool wxAnimationCtrl::Load(wxInputStream& stream, wxAnimationType type)
+{
     wxAnimation anim;
-    if (!anim.LoadFile(filename, type) ||
-        !anim.IsOk())
+    if ( !anim.Load(stream, type) || !anim.IsOk() )
         return false;
 
     SetAnimation(anim);
Index: src/gtk/animate.cpp
===================================================================
--- src/gtk/animate.cpp	(revision 53571)
+++ src/gtk/animate.cpp	(working copy)
@@ -233,8 +233,14 @@
 
 bool wxAnimationCtrl::LoadFile(const wxString &filename, wxAnimationType type)
 {
+    wxFileInputStream fis(filename);
+    return Load(fis, type);
+}
+
+bool wxAnimationCtrl::Load(wxInputStream& stream, wxAnimationType type)
+{
     wxAnimation anim;
-    if (!anim.LoadFile(filename, type))
+    if ( !anim.Load(stream, type) || !anim.IsOk() )
         return false;
 
     SetAnimation(anim);

 	  	 
