Index: include/wx/chkconf.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/chkconf.h,v
retrieving revision 1.128
diff -u -r1.128 chkconf.h
--- include/wx/chkconf.h	2006/09/28 22:55:00	1.128
+++ include/wx/chkconf.h	2006/10/03 12:51:10
@@ -1027,6 +1027,14 @@
 #           define wxUSE_STREAMS 1
 #       endif
 #   endif
+#   if !wxUSE_FILE && !wxUSE_FFILE
+#       ifdef wxABORT_ON_CONFIG_ERROR
+#           error "wxUSE_FILESYSTEM requires either wxUSE_FILE or wxUSE_FFILE"
+#       else
+#           undef wxUSE_FILE
+#           define wxUSE_FILE 1
+#       endif
+#   endif
 #endif /* wxUSE_FILESYSTEM */
 
 #if wxUSE_FS_INET
Index: src/common/filesys.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/common/filesys.cpp,v
retrieving revision 1.68
diff -u -r1.68 filesys.cpp
--- src/common/filesys.cpp	2006/09/06 13:49:42	1.68
+++ src/common/filesys.cpp	2006/10/02 23:18:07
@@ -214,7 +214,13 @@
 
     // we need to check whether we can really read from this file, otherwise
     // wxFSFile is not going to work
+#if wxUSE_FILE
+    wxFileInputStream *is = new wxFileInputStream(fullpath);
+#elif wxUSE_FFILE
     wxFFileInputStream *is = new wxFFileInputStream(fullpath);
+#else
+#error One of wxUSE_FILE or wxUSE_FFILE must be set to 1 for wxFSHandler to work
+#endif
     if ( !is->Ok() )
     {
         delete is;


 	  	 
