Index: samples/aui/auidemo.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/samples/aui/auidemo.cpp,v
retrieving revision 1.14
diff --unified -r1.14 auidemo.cpp
--- samples/aui/auidemo.cpp	2006/07/25 16:43:40	1.14
+++ samples/aui/auidemo.cpp	2006/08/10 01:54:15
@@ -40,8 +40,8 @@
     bool OnInit();
 };
 
-DECLARE_APP(MyApp);
-IMPLEMENT_APP(MyApp);
+DECLARE_APP(MyApp)
+IMPLEMENT_APP(MyApp)
 
 
 class wxSizeReportCtrl;
Index: samples/checklst/checklst.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/samples/checklst/checklst.cpp,v
retrieving revision 1.37
diff --unified -r1.37 checklst.cpp
--- samples/checklst/checklst.cpp	2006/03/14 19:39:43	1.37
+++ samples/checklst/checklst.cpp	2006/08/10 01:54:15
@@ -146,7 +146,7 @@
     EVT_BUTTON(Btn_Down, CheckListBoxFrame::OnButtonDown)
 END_EVENT_TABLE()
 
-IMPLEMENT_APP(CheckListBoxApp);
+IMPLEMENT_APP(CheckListBoxApp)
 
 // init our app: create windows
 bool CheckListBoxApp::OnInit(void)
Index: samples/dataview/dataview.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/samples/dataview/dataview.cpp,v
retrieving revision 1.20
diff --unified -r1.20 dataview.cpp
--- samples/dataview/dataview.cpp	2006/07/20 11:57:02	1.20
+++ samples/dataview/dataview.cpp	2006/08/10 01:54:16
@@ -34,7 +34,7 @@
 
 WX_DECLARE_LIST(wxDateTime,wxArrayDate);
 #include <wx/listimpl.cpp>
-WX_DEFINE_LIST(wxArrayDate);
+WX_DEFINE_LIST(wxArrayDate)
 
 class MyTextModel: public wxDataViewListModel
 {
Index: samples/db/listdb.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/samples/db/listdb.cpp,v
retrieving revision 1.29
diff --unified -r1.29 listdb.cpp
--- samples/db/listdb.cpp	2005/09/23 12:51:28	1.29
+++ samples/db/listdb.cpp	2006/08/10 01:54:16
@@ -513,6 +513,6 @@
 
     }
 
-};  // ClookUpDlg::OnCommand
+}  // ClookUpDlg::OnCommand
 
 // *********************************** listdb.cpp **********************************
Index: samples/dnd/dnd.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/samples/dnd/dnd.cpp,v
retrieving revision 1.105
diff --unified -r1.105 dnd.cpp
--- samples/dnd/dnd.cpp	2006/03/31 18:07:06	1.105
+++ samples/dnd/dnd.cpp	2006/08/10 01:54:18
@@ -113,7 +113,7 @@
     virtual bool OnInit();
 };
 
-IMPLEMENT_APP(DnDApp);
+IMPLEMENT_APP(DnDApp)
 
 #if wxUSE_DRAG_AND_DROP || wxUSE_CLIPBOARD
 
Index: samples/help/demo.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/samples/help/demo.cpp,v
retrieving revision 1.43
diff --unified -r1.43 demo.cpp
--- samples/help/demo.cpp	2006/06/11 22:46:07	1.43
+++ samples/help/demo.cpp	2006/08/10 01:54:19
@@ -223,7 +223,7 @@
     HelpDemo_Help_GNOME,
     HelpDemo_Help_Netscape,
     // controls start here (the numbers are, of course, arbitrary)
-    HelpDemo_Text = 1000,
+    HelpDemo_Text = 1000
 };
 
 // ----------------------------------------------------------------------------
Index: samples/html/virtual/virtual.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/samples/html/virtual/virtual.cpp,v
retrieving revision 1.16
diff --unified -r1.16 virtual.cpp
--- samples/html/virtual/virtual.cpp	2005/09/23 12:51:52	1.16
+++ samples/html/virtual/virtual.cpp	2006/08/10 01:54:19
@@ -121,7 +121,7 @@
    Minimal_Forward,
    
     // controls start here (the numbers are, of course, arbitrary)
-   Minimal_Text = 1000,
+   Minimal_Text = 1000
    };
 
 // ----------------------------------------------------------------------------
Index: samples/html/widget/widget.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/samples/html/widget/widget.cpp,v
retrieving revision 1.17
diff --unified -r1.17 widget.cpp
--- samples/html/widget/widget.cpp	2006/04/14 21:02:54	1.17
+++ samples/html/widget/widget.cpp	2006/08/10 01:54:19
@@ -122,7 +122,7 @@
     Minimal_Forward,
 
     // controls start here (the numbers are, of course, arbitrary)
-    Minimal_Text = 1000,
+    Minimal_Text = 1000
 };
 
 // ----------------------------------------------------------------------------
Index: samples/layout/layout.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/samples/layout/layout.cpp,v
retrieving revision 1.43
diff --unified -r1.43 layout.cpp
--- samples/layout/layout.cpp	2005/09/24 21:43:05	1.43
+++ samples/layout/layout.cpp	2006/08/10 01:54:20
@@ -430,7 +430,7 @@
     GBS_MOVE_BTN1,
     GBS_MOVE_BTN2,
 
-    GBS_MAX,
+    GBS_MAX
 };
 
 
Index: samples/oleauto/oleauto.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/samples/oleauto/oleauto.cpp,v
retrieving revision 1.11
diff --unified -r1.11 oleauto.cpp
--- samples/oleauto/oleauto.cpp	2006/01/26 19:06:00	1.11
+++ samples/oleauto/oleauto.cpp	2006/08/10 01:54:21
@@ -91,7 +91,7 @@
     OleAuto_Test,
 
     // controls start here (the numbers are, of course, arbitrary)
-    OleAuto_Text = 1000,
+    OleAuto_Text = 1000
 };
 
 // ----------------------------------------------------------------------------
Index: samples/ownerdrw/ownerdrw.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/samples/ownerdrw/ownerdrw.cpp,v
retrieving revision 1.19
diff --unified -r1.19 ownerdrw.cpp
--- samples/ownerdrw/ownerdrw.cpp	2005/09/24 21:43:06	1.19
+++ samples/ownerdrw/ownerdrw.cpp	2006/08/10 01:54:21
@@ -70,7 +70,7 @@
     Menu_Submenu, Menu_Sub1, Menu_Sub2, Menu_Sub3,
     Menu_Toggle, Menu_About,
     Control_First = 1000,
-    Control_Listbox, Control_Listbox2,
+    Control_Listbox, Control_Listbox2
 };
 
 BEGIN_EVENT_TABLE(OwnerDrawnFrame, wxFrame)
@@ -83,7 +83,7 @@
                 OwnerDrawnFrame::OnListboxDblClick)
 END_EVENT_TABLE()
 
-IMPLEMENT_APP(OwnerDrawnApp);
+IMPLEMENT_APP(OwnerDrawnApp)
 
 // init our app: create windows
 bool OwnerDrawnApp::OnInit(void)
Index: samples/propsize/propsize.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/samples/propsize/propsize.cpp,v
retrieving revision 1.12
diff --unified -r1.12 propsize.cpp
--- samples/propsize/propsize.cpp	2005/09/23 12:52:15	1.12
+++ samples/propsize/propsize.cpp	2006/08/10 01:54:22
@@ -81,7 +81,7 @@
 enum
 {
     // menu items
-    Minimal_Quit = 1,
+    Minimal_Quit = 1
 };
 
 // ----------------------------------------------------------------------------
Index: samples/regtest/regtest.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/samples/regtest/regtest.cpp,v
retrieving revision 1.24
diff --unified -r1.24 regtest.cpp
--- samples/regtest/regtest.cpp	2005/09/24 21:43:11	1.24
+++ samples/regtest/regtest.cpp	2006/08/10 01:54:23
@@ -233,7 +233,7 @@
     Menu_Delete,
     Menu_Info,
 
-    Ctrl_RegTree  = 200,
+    Ctrl_RegTree  = 200
 };
 
 // ----------------------------------------------------------------------------
Index: samples/rotate/rotate.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/samples/rotate/rotate.cpp,v
retrieving revision 1.17
diff --unified -r1.17 rotate.cpp
--- samples/rotate/rotate.cpp	2006/02/05 11:59:30	1.17
+++ samples/rotate/rotate.cpp	2006/08/10 01:54:23
@@ -153,7 +153,7 @@
 // ----------------------------------------------------------------------------
 
 #include "wx/arrimpl.cpp"
-WX_DEFINE_OBJARRAY(ArrayOfImages);
+WX_DEFINE_OBJARRAY(ArrayOfImages)
 
 // ----------------------------------------------------------------------------
 // custom canvas control that we can draw on
Index: samples/splash/splash.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/samples/splash/splash.cpp,v
retrieving revision 1.6
diff --unified -r1.6 splash.cpp
--- samples/splash/splash.cpp	2005/12/11 16:16:03	1.6
+++ samples/splash/splash.cpp	2006/08/10 01:54:24
@@ -90,7 +90,7 @@
 // IDs for the controls and the menu commands
 enum
 {
-    Minimal_Run = wxID_HIGHEST + 1,
+    Minimal_Run = wxID_HIGHEST + 1
 };
 
 // ----------------------------------------------------------------------------

 	  	 
