Index: wincmn.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/common/wincmn.cpp,v
retrieving revision 1.173
diff -d -p --unified=3 -r1.173 wincmn.cpp
--- wincmn.cpp	2004/04/09 21:06:31	1.173
+++ wincmn.cpp	2004/04/14 01:53:39
@@ -546,7 +546,7 @@ wxSize wxWindowBase::DoGetBestSize() con
 #endif // wxUSE_CONSTRAINTS
     else if ( !GetChildren().empty() )
     {
-        // our minimal acceptable size is such that all our windows fit inside
+        // our minimal acceptable size is such that all our visible windows fit inside
         int maxX = 0,
             maxY = 0;
 
@@ -555,7 +555,7 @@ wxSize wxWindowBase::DoGetBestSize() con
               node = node->GetNext() )
         {
             wxWindow *win = node->GetData();
-            if ( win->IsTopLevel()
+            if ( win->IsTopLevel() || ( ! win->IsShown() )
 #if wxUSE_STATUSBAR
                     || wxDynamicCast(win, wxStatusBar)
 #endif // wxUSE_STATUSBAR

 	  	 
