 samples/minimal/minimal.cpp | 46 +++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 46 insertions(+)

diff --git a/samples/minimal/minimal.cpp b/samples/minimal/minimal.cpp
index 470e765423..b0614d8563 100644
--- a/samples/minimal/minimal.cpp
+++ b/samples/minimal/minimal.cpp
@@ -26,6 +26,8 @@
     #include "wx/wx.h"
 #endif
 
+#include <windows.h>
+
 // ----------------------------------------------------------------------------
 // resources
 // ----------------------------------------------------------------------------
@@ -64,7 +66,10 @@ public:
     void OnQuit(wxCommandEvent& event);
     void OnAbout(wxCommandEvent& event);
 
+    void OnPaint(wxPaintEvent&);
+
 private:
+    HCURSOR m_cursorIBeam, m_cursorArrow;
     // any class wishing to process wxWidgets events must use this macro
     wxDECLARE_EVENT_TABLE();
 };
@@ -95,6 +100,7 @@ enum
 wxBEGIN_EVENT_TABLE(MyFrame, wxFrame)
     EVT_MENU(Minimal_Quit,  MyFrame::OnQuit)
     EVT_MENU(Minimal_About, MyFrame::OnAbout)
+    EVT_PAINT(MyFrame::OnPaint)
 wxEND_EVENT_TABLE()
 
 // Create a new application object: this macro will allow wxWidgets to create
@@ -144,6 +150,13 @@ MyFrame::MyFrame(const wxString& title)
     // set the frame icon
     SetIcon(wxICON(sample));
 
+    SetBackgroundColour(*wxRED);
+    SetClientSize(400, 400);
+    m_cursorIBeam = ::LoadCursor(NULL, MAKEINTRESOURCE(IDC_IBEAM));
+    m_cursorArrow = ::LoadCursor(NULL, MAKEINTRESOURCE(IDC_ARROW));
+    if ( !m_cursorIBeam || !m_cursorArrow )
+        wxLogError("Could not create a HCURSOR.");
+
 #if wxUSE_MENUBAR
     // create a menu bar
     wxMenu *fileMenu = new wxMenu;
@@ -201,3 +214,36 @@ void MyFrame::OnAbout(wxCommandEvent& WXUNUSED(event))
                  wxOK | wxICON_INFORMATION,
                  this);
 }
+
+void MyFrame::OnPaint(wxPaintEvent&)
+{
+    wxPaintDC dc(this);
+    wxIcon    iconBeam, iconArrow;
+    wxBitmap  bitmapBeam, bitmapArrow;
+
+    if ( !iconBeam.CreateFromHICON(WXHICON(::CopyImage(m_cursorIBeam, IMAGE_CURSOR, 0, 0, 0)))
+         || !iconArrow.CreateFromHICON(WXHICON(::CopyImage(m_cursorArrow, IMAGE_CURSOR, 0, 0, 0))) )
+    {
+        wxLogError("Could not create a wxIcon from a HCURSOR.");
+        return;
+    }
+
+    if ( !bitmapBeam.CopyFromIcon(iconBeam)
+         || !bitmapArrow.CopyFromIcon(iconArrow) )
+    {
+        wxLogError("Could not create an wxBitmap from wxIcon.");
+        return;
+    }
+
+    dc.DrawText("::DrawIconEx():", 4, 4);
+    ::DrawIconEx(dc.GetHDC(), 160, 4, m_cursorIBeam, 0, 0, 0, NULL, DI_NORMAL);
+    ::DrawIconEx(dc.GetHDC(), 200, 4, m_cursorArrow, 0, 0, 0, NULL, DI_NORMAL);
+
+    dc.DrawText("wxDC::DrawIcon():", 4, 84);
+    dc.DrawIcon(iconBeam, 160, 84);
+    dc.DrawIcon(iconArrow, 200, 84);
+
+    dc.DrawText("wxDC::DrawBitmap():", 4, 164);
+    dc.DrawBitmap(bitmapBeam, 160, 164, bitmapBeam.GetMask());
+    dc.DrawBitmap(bitmapArrow, 200, 164, bitmapArrow.GetMask());
+}
