Index: src/common/cmdline.cpp
===================================================================
RCS file: /home/wxcvs/wxWindows/src/common/cmdline.cpp,v
retrieving revision 1.27
diff -u -r1.27 cmdline.cpp
--- src/common/cmdline.cpp	2 Aug 2002 17:55:54 -0000	1.27
+++ src/common/cmdline.cpp	20 Aug 2002 14:07:11 -0000
@@ -808,9 +808,17 @@
                 }
                 else
                 {
-                    optName.Printf(_("%s (or %s)"),
-                                   opt.shortName.c_str(),
-                                   opt.longName.c_str());
+                    if ( this->AreLongOptionsEnabled() )
+                    {
+                        optName.Printf( _("%s (or %s)"),
+                                        opt.shortName.c_str(),
+                                        opt.longName.c_str() );
+                    }
+                    else
+                    {
+                        optName.Printf( wxT("%s"),
+                                        opt.shortName.c_str() );
+                    }
                 }
 
                 errorMsg << wxString::Format(_("The value for the option '%s' must be specified."),

 	  	 
