Index: src/unix/gsocket.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/unix/gsocket.cpp,v
retrieving revision 1.51
diff -b -u -2 -r1.51 gsocket.cpp
--- src/unix/gsocket.cpp	2005/09/19 15:18:27	1.51
+++ src/unix/gsocket.cpp	2005/09/24 22:21:33
@@ -887,4 +887,7 @@
     else
       m_error = GSOCK_IOERR;
+  } else if ( ret == 0 && !m_non_blocking ) {
+	m_error=GSOCK_IOERR;
+	ret=-1;
   }
 

 	  	 
