Index: src/generic/datavgen.cpp
===================================================================
--- src/generic/datavgen.cpp	(revision 72534)
+++ src/generic/datavgen.cpp	(working copy)
@@ -2338,7 +2338,7 @@
 
         // If this was the last child to be removed, it's possible the parent
         // node became a leaf. Let's ask the model about it.
-        if ( parentNode->GetChildNodes().empty() )
+        if ( !IsList() && parentNode->GetChildNodes().empty() )
         {
             bool isContainer = GetModel()->IsContainer(parent);
             parentNode->SetHasChildren(isContainer);
