Index: samples/mdi/mdi.cpp
===================================================================
--- samples/mdi/mdi.cpp	(Revision 68618)
+++ samples/mdi/mdi.cpp	(Arbeitskopie)
@@ -31,6 +31,7 @@
 #endif
 
 #include "wx/toolbar.h"
+#include "wx/busyinfo.h"
 
 #if !defined(__WXMSW__)
     #include "../sample.xpm"
@@ -260,6 +261,14 @@
 
 void MyFrame::OnNewWindow(wxCommandEvent& WXUNUSED(event) )
 {
+    // Under wxGTK, neither wxBusyCursor nor wxBusyInfo work as expected (that is, as under wxMSW):
+    //   - The hourglass cursor doesn't show until the method returns, then it becomes permanent in the menu bar.
+    //   - The info window does not appear at all.
+    wxBusyCursor BusyCursor;
+    wxBusyInfo   BusyInfo("Loading...");
+
+    wxSleep(3);     // Simulate long loading and processing of data files.
+
     // create and show another child frame
     MyChild *subframe = new MyChild(this);
     subframe->Show(true);
