Index: include/wx/dc.h
===================================================================
--- include/wx/dc.h	(revision 55043)
+++ include/wx/dc.h	(working copy)
@@ -43,68 +43,6 @@
 class WXDLLIMPEXP_FWD_CORE wxPrintData;
 
 //-----------------------------------------------------------------------------
-// wxDrawObject helper class
-//-----------------------------------------------------------------------------
-
-class WXDLLIMPEXP_CORE wxDrawObject
-{
-public:
-
-    wxDrawObject()
-        : m_isBBoxValid(false)
-        , m_minX(0), m_minY(0), m_maxX(0), m_maxY(0)
-    { }
-
-    virtual ~wxDrawObject() { }
-
-    virtual void Draw(wxDC&) const { }
-
-    virtual void CalcBoundingBox(wxCoord x, wxCoord y)
-    {
-      if ( m_isBBoxValid )
-      {
-         if ( x < m_minX ) m_minX = x;
-         if ( y < m_minY ) m_minY = y;
-         if ( x > m_maxX ) m_maxX = x;
-         if ( y > m_maxY ) m_maxY = y;
-      }
-      else
-      {
-         m_isBBoxValid = true;
-
-         m_minX = x;
-         m_minY = y;
-         m_maxX = x;
-         m_maxY = y;
-      }
-    }
-
-    void ResetBoundingBox()
-    {
-        m_isBBoxValid = false;
-
-        m_minX = m_maxX = m_minY = m_maxY = 0;
-    }
-
-    // Get the final bounding box of the PostScript or Metafile picture.
-
-    wxCoord MinX() const { return m_minX; }
-    wxCoord MaxX() const { return m_maxX; }
-    wxCoord MinY() const { return m_minY; }
-    wxCoord MaxY() const { return m_maxY; }
-
-    //to define the type of object for derived objects
-    virtual int GetType()=0;
-
-protected:
-    //for boundingbox calculation
-    bool m_isBBoxValid:1;
-    //for boundingbox calculation
-    wxCoord m_minX, m_minY, m_maxX, m_maxY;
-};
-
-
-//-----------------------------------------------------------------------------
 // wxDCFactory
 //-----------------------------------------------------------------------------
 
@@ -881,15 +819,6 @@
         { m_pimpl->SetDeviceLocalOrigin( x, y ); }
 
 
-    // draw generic object
-
-    void DrawObject(wxDrawObject* drawobject)
-    {
-        drawobject->Draw(*this);
-        CalcBoundingBox(drawobject->MinX(),drawobject->MinY());
-        CalcBoundingBox(drawobject->MaxX(),drawobject->MaxY());
-    }
-
     // -----------------------------------------------
     // the actual drawing API
 
