--- wxWidgets-2.7.0-1/src/common/dbtable.cpp	2006-08-17 11:48:11.000000000 +0200
+++ wxWidgets-2.7.0-1.orig/src/common/dbtable.cpp	2006-08-08 15:28:45.000000000 +0200
@@ -2244,20 +2244,6 @@
             pDt->second = 0;
             pDt->fraction = 0;
             break;
-        case SQL_C_DATE:
-            DATE_STRUCT *pDtd;
-            pDtd = (DATE_STRUCT *) colDefs[colNumber].PtrDataObj;
-            pDtd->year = 0;
-            pDtd->month = 0;
-            pDtd->day = 0;
-            break;
-        case SQL_C_TIME:
-            TIME_STRUCT *pDtt;
-            pDtt = (TIME_STRUCT *) colDefs[colNumber].PtrDataObj;
-            pDtt->hour = 0;
-            pDtt->minute = 0;
-            pDtt->second = 0;
-            break;
     }
 
     if (setToNull)

 	  	 
