diff -ru2 -xCVS -x'.#*' -x'*.orig' -x'*.rej' -x'*~' wxWindows.orig/include/wx/object.h wxWindows/include/wx/object.h
--- wxWindows.orig/include/wx/object.h	2003-09-20 13:23:42.000000000 +0200
+++ wxWindows/include/wx/object.h	2003-12-10 10:36:04.236353600 +0100
@@ -323,10 +323,11 @@
 
 #ifdef __WXDEBUG__
-inline void wxCheckCast(void *ptr)
+inline void* wxCheckCast(void *ptr)
 {
-    wxASSERT_MSG( ptr, _T("wxStaticCast() used incorrectly") );
+    wxASSERT_MSG( ptr, _T("wxStaticCast() used incorrectly") );
+    return ptr;
 }
 #define wxStaticCast(obj, className) \
- (wxCheckCast(wxDynamicCast(obj, className)), ((className *)(obj)))
+ ((className *)wxCheckCast(wxDynamicCast(obj, className)))
 
 #else  // !__WXDEBUG__

 	  	 
