--- samples/widgets/textctrl.cpp.old	2007-02-02 21:58:04.000000000 +0300
+++ samples/widgets/textctrl.cpp	2007-03-22 11:28:11.000000000 +0300
@@ -250,8 +250,14 @@
                     wxWindowID id,
                     const wxString& value,
                     int flags)
-        : wxTextCtrl(parent, id, value, wxDefaultPosition, wxDefaultSize, flags)
+        : wxTextCtrl(parent, id, wxEmptyString, wxDefaultPosition, wxDefaultSize, flags)
     {
+	wxTextAttr att;
+	att.SetFlags(wxTEXT_ATTR_FONT);
+	att.SetFont(wxFont(16, wxFONTFAMILY_TELETYPE, wxFONTSTYLE_NORMAL,
+		wxFONTWEIGHT_NORMAL));
+	SetDefaultStyle(att);
+	SetValue(value);
     }
 
 protected:

 	  	 
