--- listctrl.cpp.orig	2007-12-24 10:13:50.000000000 -0500
+++ listctrl.cpp	2007-12-24 12:54:25.000000000 -0500
@@ -66,6 +66,7 @@
 // static const int HEADER_HEIGHT = 23;
 
 static const int SCROLL_UNIT_X = 15;
+static const int SCROLL_UNIT_Y = 15;
 
 // the spacing between the lines (in report mode)
 static const int LINE_SPACING = 0;
@@ -3316,6 +3317,12 @@
             Scroll( (rect.x - 5) / SCROLL_UNIT_X, -1 );
         if (rect.x + rect.width - 5 > view_x + client_w)
             Scroll( (rect.x + rect.width - client_w + SCROLL_UNIT_X) / SCROLL_UNIT_X, -1 );
+
+        if (rect.y-view_y < 5)
+            Scroll( -1, (rect.y - 5) / SCROLL_UNIT_Y );
+        if (rect.y + rect.height - 5 > view_y + client_h)
+            Scroll( -1, (rect.y + rect.height - client_h + SCROLL_UNIT_Y) / SCROLL_UNIT_Y );
+
     }
 }
 

 	  	 
