diff -rbu2 wxw.orig\include\wx\msw\private.h wxw\include\wx\msw\private.h
--- wxw.orig\include\wx\msw\private.h	Thu Sep 06 08:16:52 2001
+++ wxw\include\wx\msw\private.h	Tue Jun 11 23:59:52 2002
@@ -314,5 +314,5 @@
 extern "C"
 {
-    WXDLLEXPORT HINSTANCE wxGetInstance();
+    WXDLLEXPORT HINSTANCE wxGetInstance(void);
 }

diff -rbu2 wxw.orig\src\common\dbtable.cpp wxw\src\common\dbtable.cpp
--- wxw.orig\src\common\dbtable.cpp	Fri Jul 06 12:11:04 2001
+++ wxw\src\common\dbtable.cpp	Wed Jun 12 00:01:19 2002
@@ -592,5 +592,6 @@
         return(FALSE);
     }
-
+// the code that follows is unreachable. Please check this!
+#if 0
     // Make sure the cursor is closed first
     if (! CloseCursor(hstmt))
@@ -605,5 +606,5 @@
     // Completed successfully
     return(TRUE);
-
+#endif
 }  // wxDbTable::query()
 
@@ -1768,10 +1769,10 @@
  */
     return FALSE;
-
+#if 0
     if (pDb->Dbms() == dbmsORACLE)
         return(TRUE);
     else
         return(FALSE);
-
+#endif
 }  // wxDbTable::CanUpdByROWID()
 
diff -rbu2 wxw.orig\src\common\dynlib.cpp wxw\src\common\dynlib.cpp
--- wxw.orig\src\common\dynlib.cpp	Wed Jan 17 16:45:06 2001
+++ wxw\src\common\dynlib.cpp	Wed Jun 12 00:03:41 2002
@@ -339,5 +339,5 @@
     wxClassInfo *old_sm_first;
 
-#if defined(__VISAGECPP__)
+#if defined(__VISAGECPP__) || defined(__BORLANDC__)
     node = m_loaded.Find(name.GetData());
     if (node != NULL)
diff -rbu2 wxw.orig\src\common\filefn.cpp wxw\src\common\filefn.cpp
--- wxw.orig\src\common\filefn.cpp	Wed Oct 03 17:20:24 2001
+++ wxw\src\common\filefn.cpp	Wed Jun 12 00:06:33 2002
@@ -528,6 +528,6 @@
             register int     braces = (*s == wxT('{') || *s == wxT('('));
             register wxChar  *value;
-#ifdef __VISAGECPP__
-    // VA gives assignment in logical expr warning
+#if defined(__VISAGECPP__) || defined(__BORLANDC__)
+    // VA and BCC give assignment in logical expr warning
             while (*d)
                *d++ = *s;
@@ -542,6 +542,6 @@
             value = wxGetenv(braces ? start + 1 : start);
             if (value) {
-#ifdef __VISAGECPP__
-    // VA gives assignment in logical expr warning
+#if defined(__VISAGECPP__) || defined(__BORLANDC__)
+    // VA and BCC give assignment in logical expr warning
                 for ((d = start - 1); (*d); *d++ = *value++);
 #else
@@ -599,6 +599,6 @@
     }
     s = nm;
-#ifdef __VISAGECPP__
-    // VA gives assignment in logical expr warning
+#if defined(__VISAGECPP__) || defined(__BORLANDC__)
+    // VA and BCC give assignment in logical expr warning
     while (*d)
        *d++ = *s++;
diff -rbu2 wxw.orig\src\common\resource.cpp wxw\src\common\resource.cpp
--- wxw.orig\src\common\resource.cpp	Wed May 10 17:39:36 2000
+++ wxw\src\common\resource.cpp	Wed Jun 12 00:22:57 2002
@@ -1056,5 +1056,6 @@
         wxExpr *valueList = (wxExpr *) NULL;
         
-        if ((valueList = expr->Nth(count)) && (valueList->Type() == PrologList))
+        valueList = expr->Nth(count);
+        if (valueList && (valueList->Type() == PrologList))
         {
             wxStringList stringList;
@@ -1073,5 +1074,6 @@
             controlItem->SetValue1(wxLB_SINGLE);
             */
-            if ((mult = expr->Nth(count)) && ((mult->Type() == PrologString)||(mult->Type() == PrologWord)))
+            mult = expr->Nth(count);
+            if (mult && ((mult->Type() == PrologString)||(mult->Type() == PrologWord)))
             {
             /*
@@ -1098,5 +1100,6 @@
         wxExpr *valueList = (wxExpr *) NULL;
         // Check for default value list
-        if ((valueList = expr->Nth(count)) && (valueList->Type() == PrologList))
+        valueList = expr->Nth(count);
+        if (valueList && (valueList->Type() == PrologList))
         {
             wxStringList stringList;
@@ -1133,5 +1136,6 @@
             wxExpr *valueList = (wxExpr *) NULL;
             // Check for default value list
-            if ((valueList = expr->Nth(count)) && (valueList->Type() == PrologList))
+            valueList = expr->Nth(count);
+            if (valueList && (valueList->Type() == PrologList))
             {
                 wxStringList stringList;
@@ -1162,5 +1166,6 @@
         wxExpr *valueList = (wxExpr *) NULL;
         // Check for default value list
-        if ((valueList = expr->Nth(count)) && (valueList->Type() == PrologList))
+        valueList = expr->Nth(count);
+        if (valueList && (valueList->Type() == PrologList))
         {
             wxStringList stringList;
@@ -2646,8 +2651,8 @@
 #ifdef __WXGTK__
                 wxLogWarning(_("Icon resource specification %s not found."), (const wxChar*) resource);
+                break;
 #else
                 return wxIcon(name, bitmapType);
 #endif
-                break;
             }
         }
diff -rbu2 wxw.orig\src\msw\dib.cpp wxw\src\msw\dib.cpp
--- wxw.orig\src\msw\dib.cpp	Sat Apr 01 22:19:14 2000
+++ wxw\src\msw\dib.cpp	Wed Jun 12 00:23:39 2002
@@ -643,5 +643,5 @@
     RGBQUAD far *lpRGB;
     HPALETTE hLogPal;
-    WORD i;
+    DWORD i;
 
     /* since biClrUsed field was filled during the loading of the DIB,
diff -rbu2 wxw.orig\src\msw\listctrl.cpp wxw\src\msw\listctrl.cpp
--- wxw.orig\src\msw\listctrl.cpp	Tue May 08 02:07:18 2001
+++ wxw\src\msw\listctrl.cpp	Wed Jun 12 00:24:25 2002
@@ -1750,5 +1750,11 @@
                 }
             }
+#ifdef __BORLANDC__
+#pragma option push -w-8066
+#endif
             break;
+#ifdef __BORLANDC__
+#pragma option pop
+#endif
 #endif // _WIN32_IE >= 0x300
 
diff -rbu2 wxw.orig\src\msw\treectrl.cpp wxw\src\msw\treectrl.cpp
--- wxw.orig\src\msw\treectrl.cpp	Sat Jun 09 15:57:20 2001
+++ wxw\src\msw\treectrl.cpp	Wed Jun 12 00:26:12 2002
@@ -2207,5 +2207,11 @@
                 }
             }
-            break;
+#ifdef __BORLANDC__
+#pragma option push -w-8066
+#endif
+            break;
+#ifdef __BORLANDC__
+#pragma option pop
+#endif
 #endif // _WIN32_IE >= 0x300
 
@@ -2322,5 +2328,7 @@
             // NB: so far the user can't set the image himself anyhow, so do it
             //     anyway - but this may change later
+#if 0
             if ( /* !processed && */ 1 )
+#endif
             {
                 wxTreeItemId item = event.m_item;

 	  	 
