Index: include/wx/cursor.h
===================================================================
--- include/wx/cursor.h	(revision 61599)
+++ include/wx/cursor.h	(working copy)
@@ -91,20 +91,8 @@
 class wxBusyCursorSuspender
 {
 public:
-    wxBusyCursorSuspender()
-    {
-        if( wxIsBusy() )
-        {
-            wxSetCursor( wxBusyCursor::GetStoredCursor() );
-        }
-    }
-    ~wxBusyCursorSuspender()
-    {
-        if( wxIsBusy() )
-        {
-            wxSetCursor( wxBusyCursor::GetBusyCursor() );
-        }
-    }
+    wxBusyCursorSuspender();
+    ~wxBusyCursorSuspender();
 };
 #endif
     // _WX_CURSOR_H_BASE_
Index: src/common/cursor.cpp
===================================================================
--- src/common/cursor.cpp	(revision 0)
+++ src/common/cursor.cpp	(revision 0)
@@ -0,0 +1,31 @@
+///////////////////////////////////////////////////////////////////////////////
+// Name:        src/common/cursor.cpp
+// Purpose:     implementation of wxBusyCursorSuspender class
+// Author:      Martin Perktold
+// Modified by:
+// Created:     02.06.2009
+// Licence:     wxWindows licence
+///////////////////////////////////////////////////////////////////////////////
+
+// ----------------------------------------------------------------------------
+// headers
+// ----------------------------------------------------------------------------
+
+#include "wx/wxprec.h"
+#include "wx/cursor.h"
+
+wxBusyCursorSuspender::wxBusyCursorSuspender()
+{
+    if( wxIsBusy() )
+    {
+        wxSetCursor( wxBusyCursor::GetStoredCursor() );
+    }
+}
+
+wxBusyCursorSuspender::~wxBusyCursorSuspender()
+{
+    if( wxIsBusy() )
+    {
+        wxSetCursor( wxBusyCursor::GetBusyCursor() );
+    }
+}
