Index: src/xrc/xh_listb.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/xrc/xh_listb.cpp,v
retrieving revision 1.16
diff -u -r1.16 xh_listb.cpp
--- src/xrc/xh_listb.cpp	2006/08/21 18:27:10	1.16
+++ src/xrc/xh_listb.cpp	2006/09/29 00:28:02
@@ -36,6 +36,7 @@
     XRC_ADD_STYLE(wxLB_ALWAYS_SB);
     XRC_ADD_STYLE(wxLB_NEEDED_SB);
     XRC_ADD_STYLE(wxLB_SORT);
+    XRC_ADD_STYLE(wxLB_OWNERDRAW);
     AddWindowStyles();
 }
 


 	  	 
