diff -ru wxWidgets.orig/samples/minimal/minimal.cpp wxWidgets/samples/minimal/minimal.cpp
--- wxWidgets.orig/samples/minimal/minimal.cpp	2005-02-21 03:14:03.000000000 +1100
+++ wxWidgets/samples/minimal/minimal.cpp	2005-06-18 21:41:21.411135880 +1000
@@ -30,6 +30,8 @@
     #include "wx/wx.h"
 #endif
 
+#include "wx/splitter.h"
+
 // ----------------------------------------------------------------------------
 // resources
 // ----------------------------------------------------------------------------
@@ -162,6 +164,18 @@
     SetMenuBar(menuBar);
 #endif // wxUSE_MENUS
 
+    wxSplitterWindow* sw = new wxSplitterWindow(this, -1, wxDefaultPosition, wxDefaultSize, wxSP_3D | wxSP_LIVE_UPDATE);
+    wxNotebook* nb = new wxNotebook(sw, -1);
+
+    wxPanel* p1 = new wxPanel(nb);
+    p1->SetBackgroundColour(*wxRED);
+    nb->AddPage(p1, "Red Page");
+
+    wxPanel* p2 = new wxPanel(sw);
+    p2->SetBackgroundColour(*wxBLUE);
+
+    sw->SplitVertically(nb, p2);
+
 #if wxUSE_STATUSBAR
     // create a status bar just for fun (by default with 1 pane only)
     CreateStatusBar(2);

 	  	 
