diff -ru2 -xCVS -x.#* -x*.orig -x*.rej -x*~ wxWidgets.orig\src\generic\vlbox.cpp wxWidgets\src\generic\vlbox.cpp
--- wxWidgets.orig\src\generic\vlbox.cpp	Fri Jan 13 10:48:57 2006
+++ wxWidgets\src\generic\vlbox.cpp	Tue Jan 17 18:26:57 2006
@@ -33,5 +33,7 @@
 
 #include "wx/vlbox.h"
+#include "wx/dcbuffer.h"
 #include "wx/selstore.h"
+#include "wx/bitmap.h"
 
 // ----------------------------------------------------------------------------
@@ -57,4 +59,9 @@
 // ----------------------------------------------------------------------------
 
+// due to ABI compatibility reasons, we need to declare double-buffer
+// outside the class
+static wxBitmap* gs_doubleBuffer = NULL;
+
+
 void wxVListBox::Init()
 {
@@ -83,4 +90,7 @@
     m_colBgSel = wxSystemSettings::GetColour(wxSYS_COLOUR_HIGHLIGHT);
 
+    // flicker-free drawing requires this
+    SetBackgroundStyle(wxBG_STYLE_CUSTOM);
+
     return true;
 }
@@ -89,4 +99,7 @@
 {
     delete m_selStore;
+
+    delete gs_doubleBuffer;
+    gs_doubleBuffer = NULL;
 }
 
@@ -225,8 +238,6 @@
             // it is, indeed, only partly visible, so scroll it into view to
             // make it entirely visible
-            if ( (size_t)m_current == GetLastVisibleLine() )
-            {
-                ScrollToLine(m_current);
-            }
+            while ( (size_t)m_current == GetLastVisibleLine() &&
+                    ScrollToLine(GetVisibleBegin()+1) );
 
             // but in any case refresh it as even if it was only partly visible
@@ -352,12 +363,27 @@
 void wxVListBox::OnPaint(wxPaintEvent& WXUNUSED(event))
 {
-    wxPaintDC dc(this);
+    // If size is larger, recalculate double buffer bitmap
+    wxSize clientSize = GetClientSize();
+
+    if ( !gs_doubleBuffer ||
+         clientSize.x > gs_doubleBuffer->GetWidth() ||
+         clientSize.y > gs_doubleBuffer->GetHeight() )
+    {
+        delete gs_doubleBuffer;
+        gs_doubleBuffer = new wxBitmap(clientSize.x+25,clientSize.y+25);
+    }
+
+    wxBufferedPaintDC dc(this,*gs_doubleBuffer);
 
     // the update rectangle
     wxRect rectUpdate = GetUpdateClientRect();
 
+    // Fill it with background colour
+    dc.SetBrush(GetBackgroundColour());
+    dc.Clear();
+
     // the bounding rectangle of the current line
     wxRect rectLine;
-    rectLine.width = GetClientSize().x;
+    rectLine.width = clientSize.x;
 
     // iterate over all visible lines
@@ -578,5 +604,5 @@
 {
     SetFocus();
-    
+
     int item = HitTest(event.GetPosition());
 
@@ -600,14 +626,25 @@
 }
 
-void wxVListBox::OnLeftDClick(wxMouseEvent& event)
+void wxVListBox::OnLeftDClick(wxMouseEvent& eventMouse)
 {
-    int item = HitTest(event.GetPosition());
+    int item = HitTest(eventMouse.GetPosition());
     if ( item != wxNOT_FOUND )
     {
-        wxCommandEvent event(wxEVT_COMMAND_LISTBOX_DOUBLECLICKED, GetId());
-        event.SetEventObject(this);
-        event.SetInt(item);
 
-        (void)GetEventHandler()->ProcessEvent(event);
+        // if item double-clicked was not yet selected, then treat
+        // this event as a left-click instead
+        if ( item == m_current )
+        {
+            wxCommandEvent event(wxEVT_COMMAND_LISTBOX_DOUBLECLICKED, GetId());
+            event.SetEventObject(this);
+            event.SetInt(item);
+
+            (void)GetEventHandler()->ProcessEvent(event);
+        }
+        else
+        {
+            OnLeftDown(eventMouse);
+        }
+    
     }
 }

 	  	 
