Index: tests/config/fileconf.cpp
===================================================================
--- tests/config/fileconf.cpp	(Revision 48753)
+++ tests/config/fileconf.cpp	(Arbeitskopie)
@@ -36,6 +36,13 @@
 _T("[root/group2]\n")
 ;
 
+static const wxChar *testconfig2 = 
+_T("[root/group1]\n")
+_T("subentry=subvalue\n")
+_T("subentry2=subvalue2\n")
+_T("subentry3=subvalue3\n")
+;
+
 // ----------------------------------------------------------------------------
 // local functions
 // ----------------------------------------------------------------------------
@@ -322,6 +329,26 @@
                          _T("[root/group1]\n")
                          _T("[root/group2]\n"),
                          fc );
+
+    wxStringInputStream sis2(testconfig2);
+    wxFileConfig fc2(sis2);
+
+    fc2.DeleteEntry(_T("/root/group1/subentry2"));
+    fc2.Write(_T("/root/group1/subentry2"), _T("tesvalue"));
+    fc2.DeleteEntry(wxT("/root/group2/subentry2"));
+    fc2.Write(wxT("/root/group2/subentry2"), _T("tesvalue2"));
+    fc2.DeleteEntry(_T("/root/group1/subentry2"));
+    fc2.Write(_T("/root/group1/subentry2"), _T("tesvalue"));
+    fc2.DeleteEntry(wxT("/root/group2/subentry2"));
+    fc2.Write(wxT("/root/group2/subentry2"), _T("tesvalue2"));
+
+    wxVERIFY_FILECONFIG( _T("[root/group1]\n")
+                         _T("subentry=subvalue\n")
+                         _T("subentry2=testvalue\n")
+                         _T("subentry3=subvalue3\n")
+                         _T("[root/group2]\n")
+                         _T("subentry2=testvalue2\n"),
+                         fc2 );
 }
 
 void FileConfigTestCase::DeleteGroup()

 	  	 
