Index: include/wx/aui/auibook.h
===================================================================
--- include/wx/aui/auibook.h	(revision 65142)
+++ include/wx/aui/auibook.h	(working copy)
@@ -582,10 +582,10 @@
     // we don't want focus for ourselves
     // virtual bool AcceptsFocus() const { return false; }
 
+protected:
     // Redo sizing after thawing
-    virtual void Thaw();
+    virtual void DoThaw();
 
-protected:
     // choose the default border for this window
     virtual wxBorder GetDefaultBorder() const { return wxBORDER_NONE; }
 
Index: src/aui/auibook.cpp
===================================================================
--- src/aui/auibook.cpp	(revision 65142)
+++ src/aui/auibook.cpp	(working copy)
@@ -4497,11 +4497,11 @@
         return false;
 }
 
-void wxAuiNotebook::Thaw()
+void wxAuiNotebook::DoThaw()
 {
+    wxControl::DoThaw();
+
     DoSizing();
-
-    wxControl::Thaw();
 }
 
 #endif // wxUSE_AUI
