Index: include/wx/msw/ole/dataform.h
===================================================================
--- include/wx/msw/ole/dataform.h	(révision 74972)
+++ include/wx/msw/ole/dataform.h	(copie de travail)
@@ -39,14 +39,10 @@
     // default copy ctor/assignment operators ok
 
     // comparison (must have both versions)
-    bool operator==(wxDataFormatId format) const
-        { return m_format == (NativeFormat)format; }
-    bool operator!=(wxDataFormatId format) const
-        { return m_format != (NativeFormat)format; }
-    bool operator==(const wxDataFormat& format) const
-        { return m_format == format.m_format; }
-    bool operator!=(const wxDataFormat& format) const
-        { return m_format != format.m_format; }
+    bool operator==(wxDataFormatId format) const;
+    bool operator!=(wxDataFormatId format) const;
+    bool operator==(const wxDataFormat& format) const;
+    bool operator!=(const wxDataFormat& format) const;
 
     // explicit and implicit conversions to NativeFormat which is one of
     // standard data types (implicit conversion is useful for preserving the
Index: src/msw/ole/dataobj.cpp
===================================================================
--- src/msw/ole/dataobj.cpp	(révision 74972)
+++ src/msw/ole/dataobj.cpp	(copie de travail)
@@ -77,11 +77,14 @@
     // format does not match the native constant in the way other formats do,
     // so for the format checks below to work, we must change the native
     // id to the wxDF_HTML constant.
-    wxChar s_szBuf[256];
-    if (::GetClipboardFormatName(format, s_szBuf, WXSIZEOF(s_szBuf)))
+    if (!format.IsStandard())
     {
-        if (s_szBuf == wxString("HTML Format"))
-            format = wxDF_HTML;
+        wxChar s_szBuf[256];
+        if (::GetClipboardFormatName(format, s_szBuf, WXSIZEOF(s_szBuf)))
+        {
+            if (s_szBuf == wxString("HTML Format"))
+                format = wxDF_HTML;
+        }
     }
     return format;
 }
@@ -342,6 +345,26 @@
 // wxDataFormat
 // ----------------------------------------------------------------------------
 
+bool wxDataFormat::operator==(wxDataFormatId format) const
+{
+    return m_format == (NativeFormat)format;
+}
+
+bool wxDataFormat::operator!=(wxDataFormatId format) const
+{
+    return m_format != (NativeFormat)format;
+}
+
+bool wxDataFormat::operator==(const wxDataFormat& format) const
+{
+    return HtmlFormatFixup(*this).m_format == HtmlFormatFixup(format).m_format;
+}
+
+bool wxDataFormat::operator!=(const wxDataFormat& format) const
+{
+    return HtmlFormatFixup(*this).m_format != HtmlFormatFixup(format).m_format;
+}
+
 void wxDataFormat::SetId(const wxString& format)
 {
     m_format = (wxDataFormat::NativeFormat)::RegisterClipboardFormat(format.t_str());
