Index: src/dfb/region.cpp
===================================================================
--- src/dfb/region.cpp	(Revision 72262)
+++ src/dfb/region.cpp	(Arbeitskopie)
@@ -33,7 +33,7 @@
 
     ~wxRegionRefData() {}
 
-    // default assignment and comparision operators are OK
+    // default assignment and comparison operators are OK
 
     wxRect m_rect;
 };
@@ -178,7 +178,7 @@
     }
     else if ( !M_REGION->m_rect.Intersects(rect) )
     {
-        // the rectangles are disjoint, so substracting has no effect
+        // the rectangles are disjoint, so subtracting has no effect
         return true;
     }
     else
Index: src/dfb/wrapdfb.cpp
===================================================================
--- src/dfb/wrapdfb.cpp	(Revision 72262)
+++ src/dfb/wrapdfb.cpp	(Arbeitskopie)
@@ -61,7 +61,7 @@
 
         default:
             // FIXME: should handle the errors individually
-            wxLogError(_("DirectFB error %d occured."), (int)code);
+            wxLogError(_("DirectFB error %d occurred."), (int)code);
             return false;
     }
 }
