Index: include/wx/stdpaths.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/stdpaths.h,v
retrieving revision 1.13
diff -u -r1.13 stdpaths.h
--- include/wx/stdpaths.h	2005/04/22 14:53:55	1.13
+++ include/wx/stdpaths.h	2005/06/02 19:16:33
@@ -74,7 +74,14 @@
     // Contents/Plugins app bundle subdirectory under Mac
     virtual wxString GetPluginsDir() const = 0;
 
+    // return the "Documents" directory for the current user
+    //
+    // C:\Documents and Settings\username\Documents under Windows,
+    // $HOME under Unix and
+    // ~/Documents under Mac
+    virtual wxString GetDocumentsDir() const;
 
+
     // virtual dtor for the base class
     virtual ~wxStandardPathsBase();
 
@@ -112,6 +119,7 @@
     virtual wxString GetLocalDataDir() const { return m_prefix; }
     virtual wxString GetUserDataDir() const { return m_prefix; }
     virtual wxString GetPluginsDir() const { return m_prefix; }
+    virtual wxString GetDocumentsDir() const { return m_prefix; }
 
 private:
     wxString m_prefix;
Index: include/wx/msw/stdpaths.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/msw/stdpaths.h,v
retrieving revision 1.2
diff -u -r1.2 stdpaths.h
--- include/wx/msw/stdpaths.h	2004/10/20 00:34:25	1.2
+++ include/wx/msw/stdpaths.h	2005/06/02 19:16:39
@@ -26,6 +26,7 @@
     virtual wxString GetUserDataDir() const;
     virtual wxString GetUserLocalDataDir() const;
     virtual wxString GetPluginsDir() const;
+    virtual wxString GetDocumentsDir() const;
 
 protected:
     // get the path corresponding to the given standard CSIDL_XXX constant
Index: src/common/stdpbase.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/common/stdpbase.cpp,v
retrieving revision 1.7
diff -u -r1.7 stdpbase.cpp
--- src/common/stdpbase.cpp	2005/05/31 09:19:52	1.7
+++ src/common/stdpbase.cpp	2005/06/02 19:16:42
@@ -70,6 +70,11 @@
     return GetUserDataDir();
 }
 
+wxString wxStandardPathsBase::GetDocumentsDir() const
+{
+    return wxFileName::GetHomeDir();
+}
+
 /* static */
 wxString wxStandardPathsBase::AppendAppName(const wxString& dir)
 {
Index: src/msw/stdpaths.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/msw/stdpaths.cpp,v
retrieving revision 1.8
diff -u -r1.8 stdpaths.cpp
--- src/msw/stdpaths.cpp	2005/03/20 11:52:33	1.8
+++ src/msw/stdpaths.cpp	2005/06/02 19:16:48
@@ -68,6 +68,10 @@
     #define CSIDL_PROGRAM_FILES   0x0026
 #endif
 
+#ifndef CSIDL_PERSONAL
+    #define CSIDL_PERSONAL        0x0005
+#endif
+
 #ifndef SHGFP_TYPE_CURRENT
     #define SHGFP_TYPE_CURRENT 0
 #endif
@@ -75,7 +79,6 @@
 #ifndef SHGFP_TYPE_DEFAULT
     #define SHGFP_TYPE_DEFAULT 1
 #endif
-
 // ----------------------------------------------------------------------------
 // module globals
 // ----------------------------------------------------------------------------
@@ -264,6 +267,11 @@
 wxString wxStandardPaths::GetPluginsDir() const
 {
     return wxFileName(wxGetFullModuleName()).GetPath();
+}
+
+wxString wxStandardPaths::GetDocumentsDir() const
+{
+    return DoGetDirectory(CSIDL_PERSONAL);
 }
 
 

 	  	 
