--- a/include/wx/gtk/bitmap.h
+++ b/include/wx/gtk/bitmap.h
@@ -15,6 +15,7 @@ typedef struct _cairo_surface cairo_surface_t;
 #endif
 typedef struct _GdkPixbuf GdkPixbuf;
 class WXDLLIMPEXP_FWD_CORE wxPixelDataBase;
+class WXDLLIMPEXP_FWD_CORE wxCursor;
 
 //-----------------------------------------------------------------------------
 // wxMask
@@ -81,6 +82,7 @@ public:
     wxBitmap(const wxImage& image, int depth = wxBITMAP_SCREEN_DEPTH);
 #endif // wxUSE_IMAGE
     wxBitmap(GdkPixbuf* pixbuf, int depth = 0);
+    wxBitmap(const wxCursor& cursor);
     virtual ~wxBitmap();
 
     bool Create(int width, int height, int depth = wxBITMAP_SCREEN_DEPTH) wxOVERRIDE;
--- a/src/gtk/bitmap.cpp
+++ b/src/gtk/bitmap.cpp
@@ -515,6 +515,12 @@ wxBitmap::wxBitmap(GdkPixmap* pixmap)
 }
 #endif
 
+wxBitmap::wxBitmap(const wxCursor& cursor)
+{
+    GdkPixbuf *pixbuf = gdk_cursor_get_image(cursor.GetCursor());
+    *this = wxBitmap(pixbuf);
+}
+
 wxBitmap::~wxBitmap()
 {
 }

