diff -bru wxWidgets.orig/include/wx/msw/mdi.h wxWidgets/include/wx/msw/mdi.h
--- wxWidgets.orig/include/wx/msw/mdi.h	2012-03-01 17:16:48.000000000 -0800
+++ wxWidgets/include/wx/msw/mdi.h	2012-03-05 20:53:33.000000000 -0800
@@ -215,6 +215,7 @@
     void OnIdle(wxIdleEvent& event);
 
 protected:
+    virtual void AdjustForParentClientOrigin(int& x, int& y, int sizeFlags) const;
     virtual void DoGetScreenPosition(int *x, int *y) const;
     virtual void DoGetPosition(int *x, int *y) const;
     virtual void DoSetSize(int x, int y, int width, int height, int sizeFlags);
diff -bru wxWidgets.orig/src/msw/mdi.cpp wxWidgets/src/msw/mdi.cpp
--- wxWidgets.orig/src/msw/mdi.cpp	2012-03-01 17:16:04.000000000 -0800
+++ wxWidgets/src/msw/mdi.cpp	2012-03-05 20:55:57.000000000 -0800
@@ -892,6 +892,12 @@
 }
 
 void
+wxMDIChildFrame::void AdjustForParentClientOrigin(int& WXUNUSED(x), int& WXUNUSED(y), int WXUNUSED(sizeFlags)) const
+{
+    return;
+}
+
+void
 wxMDIChildFrame::DoSetSize(int x, int y, int width, int height, int sizeFlags)
 {
     // we need to disable client area origin adjustments used for the child
