Index: src/generic/logg.cpp
===================================================================
--- src/generic/logg.cpp	(revision 62325)
+++ src/generic/logg.cpp	(working copy)
@@ -721,7 +721,7 @@
         // ensure that we won't loop here if wxGetTranslation()
         // happens to pop up a Log message while translating this :-)
         ms_details = wxTRANSLATE("&Details");
-        ms_details = wxGetTranslation(ms_details);
+        ms_details = wxString(wxGetTranslation(ms_details)); // Convert into temporary wxString to avoid self-assignment with overlapping memcpy
 #ifdef __SMARTPHONE__
         ms_details = wxStripMenuCodes(ms_details);
 #endif
