Index: include/wx/gtk/toolbar.h
===================================================================
--- include/wx/gtk/toolbar.h	(revision 78096)
+++ include/wx/gtk/toolbar.h	(working copy)
@@ -24,7 +24,7 @@
                wxWindowID id,
                const wxPoint& pos = wxDefaultPosition,
                const wxSize& size = wxDefaultSize,
-               long style = wxTB_HORIZONTAL,
+               long style = wxTB_DEFAULT_STYLE,
                const wxString& name = wxToolBarNameStr )
     {
         Init();
@@ -36,7 +36,7 @@
                  wxWindowID id,
                  const wxPoint& pos = wxDefaultPosition,
                  const wxSize& size = wxDefaultSize,
-                 long style = wxTB_HORIZONTAL,
+                 long style = wxTB_DEFAULT_STYLE,
                  const wxString& name = wxToolBarNameStr );
 
     virtual ~wxToolBar();
Index: include/wx/motif/toolbar.h
===================================================================
--- include/wx/motif/toolbar.h	(revision 78096)
+++ include/wx/motif/toolbar.h	(working copy)
@@ -21,7 +21,7 @@
         wxWindowID id,
         const wxPoint& pos = wxDefaultPosition,
         const wxSize& size = wxDefaultSize,
-        long style = wxTB_HORIZONTAL,
+        long style = wxTB_DEFAULT_STYLE,
         const wxString& name = wxToolBarNameStr)
     {
         Init();
@@ -33,7 +33,7 @@
         wxWindowID id,
         const wxPoint& pos = wxDefaultPosition,
         const wxSize& size = wxDefaultSize,
-        long style = wxTB_HORIZONTAL,
+        long style = wxTB_DEFAULT_STYLE,
         const wxString& name = wxToolBarNameStr);
 
     virtual ~wxToolBar();
Index: include/wx/msw/toolbar.h
===================================================================
--- include/wx/msw/toolbar.h	(revision 78096)
+++ include/wx/msw/toolbar.h	(working copy)
@@ -26,7 +26,7 @@
                 wxWindowID id,
                 const wxPoint& pos = wxDefaultPosition,
                 const wxSize& size = wxDefaultSize,
-                long style = wxTB_HORIZONTAL,
+                long style = wxTB_DEFAULT_STYLE,
                 const wxString& name = wxToolBarNameStr)
     {
         Init();
@@ -38,7 +38,7 @@
                 wxWindowID id,
                 const wxPoint& pos = wxDefaultPosition,
                 const wxSize& size = wxDefaultSize,
-                long style = wxTB_HORIZONTAL,
+                long style = wxTB_DEFAULT_STYLE,
                 const wxString& name = wxToolBarNameStr);
 
     virtual ~wxToolBar();
Index: include/wx/msw/wince/tbarwce.h
===================================================================
--- include/wx/msw/wince/tbarwce.h	(revision 78096)
+++ include/wx/msw/wince/tbarwce.h	(working copy)
@@ -28,7 +28,7 @@
                 wxWindowID id,
                 const wxPoint& pos = wxDefaultPosition,
                 const wxSize& size = wxDefaultSize,
-                long style = wxTB_HORIZONTAL,
+                long style = wxTB_DEFAULT_STYLE,
                 const wxString& name = wxToolBarNameStr)
     {
         Create(parent, id, pos, size, style, name);
@@ -38,7 +38,7 @@
                 wxWindowID id,
                 const wxPoint& pos = wxDefaultPosition,
                 const wxSize& size = wxDefaultSize,
-                long style = wxTB_HORIZONTAL,
+                long style = wxTB_DEFAULT_STYLE,
                 const wxString& name = wxToolBarNameStr);
 
     // override/implement base class virtuals
@@ -87,7 +87,7 @@
                 wxWindowID id,
                 const wxPoint& pos = wxDefaultPosition,
                 const wxSize& size = wxDefaultSize,
-                long style = wxTB_HORIZONTAL,
+                long style = wxTB_DEFAULT_STYLE,
                 const wxString& name = wxToolBarNameStr,
                 wxMenuBar* menuBar = NULL)
     {
@@ -100,7 +100,7 @@
                 wxWindowID id,
                 const wxPoint& pos = wxDefaultPosition,
                 const wxSize& size = wxDefaultSize,
-                long style = wxTB_HORIZONTAL,
+                long style = wxTB_DEFAULT_STYLE,
                 const wxString& name = wxToolBarNameStr,
                 wxMenuBar* menuBar = NULL);
 
Index: include/wx/osx/toolbar.h
===================================================================
--- include/wx/osx/toolbar.h	(revision 78096)
+++ include/wx/osx/toolbar.h	(working copy)
@@ -28,7 +28,7 @@
 
   inline wxToolBar(wxWindow *parent, wxWindowID id,
                    const wxPoint& pos = wxDefaultPosition, const wxSize& size = wxDefaultSize,
-                   long style = wxTB_HORIZONTAL,
+                   long style = wxTB_DEFAULT_STYLE,
                    const wxString& name = wxToolBarNameStr)
   {
     Init();
@@ -37,7 +37,7 @@
   virtual ~wxToolBar();
 
   bool Create(wxWindow *parent, wxWindowID id, const wxPoint& pos = wxDefaultPosition, const wxSize& size = wxDefaultSize,
-            long style = wxTB_HORIZONTAL,
+            long style = wxTB_DEFAULT_STYLE,
             const wxString& name = wxToolBarNameStr);
 
     virtual void SetWindowStyleFlag(long style);
Index: include/wx/qt/toolbar.h
===================================================================
--- include/wx/qt/toolbar.h	(revision 78096)
+++ include/wx/qt/toolbar.h	(working copy)
@@ -21,7 +21,7 @@
               wxWindowID id,
               const wxPoint& pos = wxDefaultPosition,
               const wxSize& size = wxDefaultSize,
-              long style = wxNO_BORDER | wxTB_HORIZONTAL,
+              long style = wxTB_DEFAULT_STYLE | wxNO_BORDER,
               const wxString& name = wxToolBarNameStr)
     {
         Init();
@@ -36,7 +36,7 @@
                 wxWindowID id,
                 const wxPoint& pos = wxDefaultPosition,
                 const wxSize& size = wxDefaultSize,
-                long style = wxNO_BORDER | wxTB_HORIZONTAL,
+                long style = wxTB_DEFAULT_STYLE | wxNO_BORDER,
                 const wxString& name = wxToolBarNameStr);
 
     virtual wxToolBarToolBase *FindToolForPosition(wxCoord x, wxCoord y) const;
