diff -uNr wxPython-src-2.9.3.1_Original/include/wx/defs.h wxPython-src-2.9.3.1_FlexGridSizerFix1/include/wx/defs.h
--- wxPython-src-2.9.3.1_Original/include/wx/defs.h	2013-08-15 14:12:56.442799654 +0100
+++ wxPython-src-2.9.3.1_FlexGridSizerFix1/include/wx/defs.h	2013-08-15 17:04:55.562910332 +0100
@@ -1534,8 +1534,11 @@
     wxSHAPED                  = 0x4000,
     wxTILE                    = wxSHAPED | wxFIXED_MINSIZE,
 
+    wxEXPAND_VERTICAL         = 0x8000,
+    wxEXPAND_HORIZONTAL       = 0x0100,
+
     /*  a mask to extract stretch from the combination of flags */
-    wxSTRETCH_MASK            = 0x7000 /* sans wxTILE */
+    wxSTRETCH_MASK            = 0xf100, /* sans wxTILE */
 };
 
 /*  border flags: the values are chosen for backwards compatibility */
diff -uNr wxPython-src-2.9.3.1_Original/src/common/sizer.cpp wxPython-src-2.9.3.1_FlexGridSizerFix1/src/common/sizer.cpp
--- wxPython-src-2.9.3.1_Original/src/common/sizer.cpp	2013-08-15 14:11:41.574798851 +0100
+++ wxPython-src-2.9.3.1_FlexGridSizerFix1/src/common/sizer.cpp	2013-08-15 17:17:25.874918380 +0100
@@ -1499,6 +1499,15 @@
     wxSize sz( item->GetMinSizeWithBorder() );
     int flag = item->GetFlag();
 
+    if (flag & wxEXPAND_HORIZONTAL)
+    {
+       sz.x = w;
+    }
+    if (flag & wxEXPAND_VERTICAL)
+    {
+       sz.y = h;
+    }
+
     if (flag & wxEXPAND || flag & wxSHAPED)
     {
        sz = wxSize(w, h);
diff -uNr wxPython-src-2.9.3.1_Original/wxPython/src/_defs.i wxPython-src-2.9.3.1_FlexGridSizerFix1/wxPython/src/_defs.i
--- wxPython-src-2.9.3.1_Original/wxPython/src/_defs.i	2013-08-15 14:12:46.502799547 +0100
+++ wxPython-src-2.9.3.1_FlexGridSizerFix1/wxPython/src/_defs.i	2013-08-15 17:30:29.694926785 +0100
@@ -1196,6 +1196,8 @@
     wxEXPAND,
     wxSHAPED,
     wxTILE,
+    wxEXPAND_HORIZONTAL,
+    wxEXPAND_VERTICAL,
     wxSTRETCH_MASK
 };
 
diff -uNr wxPython-src-2.9.3.1_Original/wxPython/src/gtk/_core.py wxPython-src-2.9.3.1_FlexGridSizerFix1/wxPython/src/gtk/_core.py
--- wxPython-src-2.9.3.1_Original/wxPython/src/gtk/_core.py	2013-08-15 14:12:46.098799543 +0100
+++ wxPython-src-2.9.3.1_FlexGridSizerFix1/wxPython/src/gtk/_core.py	2013-08-15 17:34:08.578929134 +0100
@@ -424,6 +424,8 @@
 EXPAND = _core_.EXPAND
 SHAPED = _core_.SHAPED
 TILE = _core_.TILE
+EXPAND_HORIZONTAL = _core_.EXPAND_HORIZONTAL
+EXPAND_VERTICAL = _core_.EXPAND_VERTICAL
 STRETCH_MASK = _core_.STRETCH_MASK
 BORDER_DEFAULT = _core_.BORDER_DEFAULT
 BORDER_NONE = _core_.BORDER_NONE
diff -uNr wxPython-src-2.9.3.1_Original/wxPython/src/gtk/_core_wrap.cpp wxPython-src-2.9.3.1_FlexGridSizerFix1/wxPython/src/gtk/_core_wrap.cpp
--- wxPython-src-2.9.3.1_Original/wxPython/src/gtk/_core_wrap.cpp	2013-08-15 14:12:46.342799546 +0100
+++ wxPython-src-2.9.3.1_FlexGridSizerFix1/wxPython/src/gtk/_core_wrap.cpp	2013-08-15 17:34:08.578929134 +0100
@@ -75191,6 +75191,8 @@
   SWIG_Python_SetConstant(d, "EXPAND",SWIG_From_int(static_cast< int >(wxEXPAND)));
   SWIG_Python_SetConstant(d, "SHAPED",SWIG_From_int(static_cast< int >(wxSHAPED)));
   SWIG_Python_SetConstant(d, "TILE",SWIG_From_int(static_cast< int >(wxTILE)));
+  SWIG_Python_SetConstant(d, "EXPAND_HORIZONTAL",SWIG_From_int(static_cast< int >(wxEXPAND_HORIZONTAL)));
+  SWIG_Python_SetConstant(d, "EXPAND_VERTICAL",SWIG_From_int(static_cast< int >(wxEXPAND_VERTICAL)));
   SWIG_Python_SetConstant(d, "STRETCH_MASK",SWIG_From_int(static_cast< int >(wxSTRETCH_MASK)));
   SWIG_Python_SetConstant(d, "BORDER_DEFAULT",SWIG_From_int(static_cast< int >(wxBORDER_DEFAULT)));
   SWIG_Python_SetConstant(d, "BORDER_NONE",SWIG_From_int(static_cast< int >(wxBORDER_NONE)));
diff -uNr wxPython-src-2.9.3.1_Original/wxPython/wx/_core.py wxPython-src-2.9.3.1_FlexGridSizerFix1/wxPython/wx/_core.py
--- wxPython-src-2.9.3.1_Original/wxPython/wx/_core.py	2013-08-15 14:12:36.934799444 +0100
+++ wxPython-src-2.9.3.1_FlexGridSizerFix1/wxPython/wx/_core.py	2013-08-15 17:34:08.000000000 +0100
@@ -424,6 +424,8 @@
 EXPAND = _core_.EXPAND
 SHAPED = _core_.SHAPED
 TILE = _core_.TILE
+EXPAND_HORIZONTAL = _core_.EXPAND_HORIZONTAL
+EXPAND_VERTICAL = _core_.EXPAND_VERTICAL
 STRETCH_MASK = _core_.STRETCH_MASK
 BORDER_DEFAULT = _core_.BORDER_DEFAULT
 BORDER_NONE = _core_.BORDER_NONE
