Index: src/common/platinfo.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/common/platinfo.cpp,v
retrieving revision 1.7
diff -u -r1.7 platinfo.cpp
--- src/common/platinfo.cpp	2006/10/09 15:58:56	1.7
+++ src/common/platinfo.cpp	2006/12/30 13:29:22
@@ -62,6 +62,9 @@
     _T("AIX"),
     _T("HPUX"),
 
+    _T("Other Unix"),
+    _T("Other Unix"),
+
     _T("DOS"),
     _T("OS/2")
 };

 	  	 
