Index: src/msw/gauge95.cpp
===================================================================
--- src/msw/gauge95.cpp	(revision 60585)
+++ src/msw/gauge95.cpp	(working copy)
@@ -206,6 +206,10 @@
 
 void wxGauge95::SetValue(int pos)
 {
+    // Setting the (same) position produce flicker on Vista,
+    // especially noticable if ownerdrawn
+    if (GetValue() == pos) return;
+    
     // switch to determinate mode if required
     SetDeterminateMode();
 
