diff --git a/src/osx/cocoa/srchctrl.mm b/src/osx/cocoa/srchctrl.mm
index 162408426b..555eba62a9 100644
--- a/src/osx/cocoa/srchctrl.mm
+++ b/src/osx/cocoa/srchctrl.mm
@@ -60,6 +60,19 @@ - (void)controlTextDidChange:(NSNotification *)aNotification
         impl->controlTextDidChange();
 }
 
+- (void)cancelOperation:(id)sender
+{
+	wxWidgetCocoaImpl* impl = (wxWidgetCocoaImpl* ) wxWidgetImpl::FindFromWXWidget( self );
+	if ( impl )
+	{
+		wxSearchCtrl* wxpeer = (wxSearchCtrl*) (impl->GetWXPeer());
+		if ( wxpeer )
+		{
+			wxpeer->HandleSearchFieldCancelHit();
+		}
+	}
+}
+
 - (NSArray *)control:(NSControl *)control textView:(NSTextView *)textView completions:(NSArray *)words
  forPartialWordRange:(NSRange)charRange indexOfSelectedItem:(int*)index
 {
