Index: listctrl.cpp
===================================================================
--- listctrl.cpp	(revision 54142)
+++ listctrl.cpp	(working copy)
@@ -82,6 +82,10 @@
                                   const wxListItem& item,
                                   LV_COLUMN& lvCol);
 
+// margin between the window and the items
+static const int EXTRA_BORDER_X = 2;
+static const int EXTRA_BORDER_Y = 2;
+
 // ----------------------------------------------------------------------------
 // private helper classes
 // ----------------------------------------------------------------------------
@@ -1082,9 +1086,43 @@
 
 wxRect wxListCtrl::GetViewRect() const
 {
-    wxASSERT_MSG( !HasFlag(wxLC_REPORT | wxLC_LIST),
-                    _T("wxListCtrl::GetViewRect() only works in icon mode") );
 
+    if (HasFlag(wxLC_REPORT | wxLC_LIST))
+    {
+        // we need to find the longest/tallest label
+        wxCoord xMax = 0, yMax = 0;
+        const int count = GetItemCount();
+        if ( count )
+        {
+            for ( int i = 0; i < count; i++ )
+            {
+                wxRect r;
+                GetItemRect(i, r);
+
+                wxCoord x = r.GetRight(),
+                        y = r.GetBottom();
+
+                if ( x > xMax )
+                    xMax = x;
+                if ( y > yMax )
+                    yMax = y;
+            }
+        }
+
+        // some fudge needed to make it look prettier
+        xMax += 2 * EXTRA_BORDER_X;
+        yMax += 2 * EXTRA_BORDER_Y;
+
+        // account for the scrollbars if necessary
+        const wxSize sizeAll = GetClientSize();
+        if ( xMax > sizeAll.x )
+            yMax += wxSystemSettings::GetMetric(wxSYS_HSCROLL_Y);
+        if ( yMax > sizeAll.y )
+            xMax += wxSystemSettings::GetMetric(wxSYS_VSCROLL_X);
+
+        return wxRect(0, 0, xMax, yMax);
+    }
+
     RECT rc;
     if ( !ListView_GetViewRect(GetHwnd(), &rc) )
     {
