Index: listctrl.cpp
===================================================================
--- listctrl.cpp	(revision 54142)
+++ listctrl.cpp	(working copy)
@@ -4110,8 +4110,6 @@
 
 wxRect wxListMainWindow::GetViewRect() const
 {
-    wxASSERT_MSG( !HasFlag(wxLC_REPORT | wxLC_LIST),
-                    _T("wxListCtrl::GetViewRect() only works in icon mode") );
 
     // we need to find the longest/tallest label
     wxCoord xMax = 0, yMax = 0;
