Index: samples/menu/menu.cpp
===================================================================
--- samples/menu/menu.cpp	(revision 67249)
+++ samples/menu/menu.cpp	(working copy)
@@ -72,6 +72,14 @@
 {
 public:
     bool OnInit();
+    void OnUpdateSubMenuCheck(wxUpdateUIEvent& event)
+    {
+        wxLogStatus(GetTopWindow(), __FUNCTION__);
+        event.Enable(false);
+    }
+
+protected:
+    DECLARE_EVENT_TABLE()
 };
 
 // Define a new frame
@@ -271,6 +279,10 @@
 // event tables
 // ----------------------------------------------------------------------------
 
+BEGIN_EVENT_TABLE(MyApp, wxApp)
+    EVT_UPDATE_UI(Menu_SubMenu_Check,     MyApp::OnUpdateSubMenuCheck)
+END_EVENT_TABLE()
+
 BEGIN_EVENT_TABLE(MyFrame, wxFrame)
     EVT_MENU(Menu_File_Quit,     MyFrame::OnQuit)
 #if USE_LOG_WINDOW
@@ -1146,6 +1158,7 @@
 
 void MyFrame::OnUpdateSubMenuCheck(wxUpdateUIEvent& event)
 {
+    wxLogStatus(this, __FUNCTION__);
     event.Enable(true);
 }
 
