diff --git a/interface/wx/richtext/richtextbuffer.h b/interface/wx/richtext/richtextbuffer.h
index 1d2365b..8b3e42d 100644
--- a/interface/wx/richtext/richtextbuffer.h
+++ b/interface/wx/richtext/richtextbuffer.h
@@ -1402,7 +1402,7 @@ public:
     @library{wxrichtext}
     @category{richtext}
 
-    @see wxRichTextAttr, wxTextBoxAttr, wxRichTextCtrl
+    @see wxTextAttr, wxTextBoxAttr, wxRichTextCtrl
 */
 
 class wxRichTextAttr: public wxTextAttr
