diff --git a/include/wx/wxcrtbase.h b/include/wx/wxcrtbase.h
index 4061902..315e947 100644
--- a/include/wx/wxcrtbase.h
+++ b/include/wx/wxcrtbase.h
@@ -169,7 +169,7 @@ WXDLLIMPEXP_BASE void *calloc( size_t num, size_t size );
    strdup() but MinGW does provide it under MSVC-compatible name so test for it
    before checking __WX_STRICT_ANSI_GCC__. */
 #if (defined(__VISUALC__) && __VISUALC__ >= 1400) || \
-    defined(__MINGW32__)
+    (defined(__MINGW32__) && !defined(__STRICT_ANSI__))
     #define wxCRT_StrdupA _strdup
 #elif !(defined(__WXWINCE__) || defined(__WX_STRICT_ANSI_GCC__))
     #define wxCRT_StrdupA strdup
