Index: buttonbar.cpp
===================================================================
--- buttonbar.cpp	(revision 75711)
+++ buttonbar.cpp	(working copy)
@@ -515,6 +515,8 @@
         {
             m_layouts_valid = false;
             m_buttons.RemoveAt(i);
+            m_hovered_button = 0;
+            m_active_button = 0;
             delete button;
             Realize();
             Refresh();
