 src/generic/treectlg.cpp | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/src/generic/treectlg.cpp b/src/generic/treectlg.cpp
index 9041ce101a..688a0199c6 100644
--- a/src/generic/treectlg.cpp
+++ b/src/generic/treectlg.cpp
@@ -4090,12 +4090,10 @@ void wxGenericTreeCtrl::OnGetToolTip( wxTreeEvent &event )
         // If it doesn't, show its full text in the tooltip.
         event.SetLabel(pItem->GetText());
     }
-    else
+#else
+    // veto processing the event, nixing any tooltip
+    event.Veto();
 #endif // wxUSE_TOOLTIPS
-    {
-        // veto processing the event, nixing any tooltip
-        event.Veto();
-    }
 }
 
 
