Index: notifmsgrt.cpp
===================================================================
--- notifmsgrt.cpp	(revision 128976)
+++ notifmsgrt.cpp	(working copy)
@@ -348,9 +348,9 @@
         else
         {
             // Create new shortcut
-            if ( FAILED(shellLink->SetPath(wxStandardPaths::Get().GetExecutablePath().wc_str())) )
+            if ( FAILED(shellLink->SetPath(wxStandardPaths::Get().GetExecutablePath().wx_str())) )
                 return false;
-            if ( FAILED(shellLink->SetArguments(L"")) )
+            if ( FAILED(shellLink->SetArguments(wxEmptyString)) )
                 return false;
 
             writeShortcut = true;
