Index: spinbutt.mm
===================================================================
--- spinbutt.mm	(revision 70254)
+++ spinbutt.mm	(working copy)
@@ -110,10 +110,14 @@
     [v setMinValue: minimum];
     [v setMaxValue: maximum];
     [v setIntValue: value];
-    
+    if ( style & wxSP_HORIZONTAL )
+        [v rotateByAngle:90.0];
+
+    Boolean wrap = NO;
     if ( style & wxSP_WRAP )
-        [v setValueWraps:YES];
-    
+        wrap = YES;
+    [v setValueWraps:wrap];
+
     wxWidgetCocoaImpl* c = new wxSpinButtonCocoaImpl( wxpeer, v );
     return c;
 }
