--- samples/listctrl/listtest.cpp	Wed Jan 14 16:16:59 2015
+++ samples/listctrl/listtest.cpp	Thu Jan 22 14:08:10 2015
@@ -559,7 +559,7 @@ void MyFrame::InitWithReportItems()
     m_listCtrl->SetTextColour(*wxBLUE);
 
     m_listCtrl->SetColumnWidth( 0, wxLIST_AUTOSIZE );
-    m_listCtrl->SetColumnWidth( 1, wxLIST_AUTOSIZE );
+    m_listCtrl->SetColumnWidth( 1, 0 );
     m_listCtrl->SetColumnWidth( 2, wxLIST_AUTOSIZE );
 
     // Set images in columns
