diff -uNr 2.9.5-orig/include/wx/richtext/richtextbuffer.h 2.9.5-fixed/include/wx/richtext/richtextbuffer.h
--- 2.9.5-orig/include/wx/richtext/richtextbuffer.h	2013-07-16 18:39:22.000000000 +0200
+++ 2.9.5-fixed/include/wx/richtext/richtextbuffer.h	2013-07-25 15:51:34.439933300 +0200
@@ -4842,23 +4842,41 @@
     */
     virtual void ResetAndClearCommands();
 
+#if wxUSE_FFILE && wxUSE_STREAMS
     //@{
     /**
-        Loads content from a stream or file.
+        Loads content from a file.
         Not all handlers will implement file loading.
     */
     virtual bool LoadFile(const wxString& filename, wxRichTextFileType type = wxRICHTEXT_TYPE_ANY);
-    virtual bool LoadFile(wxInputStream& stream, wxRichTextFileType type = wxRICHTEXT_TYPE_ANY);
     //@}
 
     //@{
     /**
-        Saves content to a stream or file.
+        Saves content to a file.
         Not all handlers will implement file saving.
     */
     virtual bool SaveFile(const wxString& filename, wxRichTextFileType type = wxRICHTEXT_TYPE_ANY);
+    //@}
+#endif
+
+#if wxUSE_STREAMS
+    //@{
+    /**
+        Loads content from a stream.
+        Not all handlers will implement loading from a stream.
+    */
+    virtual bool LoadFile(wxInputStream& stream, wxRichTextFileType type = wxRICHTEXT_TYPE_ANY);
+    //@}
+
+    //@{
+    /**
+        Saves content to a stream.
+        Not all handlers will implement saving to a stream.
+    */
     virtual bool SaveFile(wxOutputStream& stream, wxRichTextFileType type = wxRICHTEXT_TYPE_ANY);
     //@}
+#endif
 
     /**
         Sets the handler flags, controlling loading and saving.
diff -uNr 2.9.5-orig/include/wx/richtext/richtextctrl.h 2.9.5-fixed/include/wx/richtext/richtextctrl.h
--- 2.9.5-orig/include/wx/richtext/richtextctrl.h	2013-07-16 18:39:22.000000000 +0200
+++ 2.9.5-fixed/include/wx/richtext/richtextctrl.h	2013-07-25 16:56:53.741697500 +0200
@@ -573,6 +573,7 @@
                   int type = wxRICHTEXT_TYPE_ANY);
 #endif
 
+#if wxUSE_FFILE && wxUSE_STREAMS
     /**
         Helper function for LoadFile(). Loads content into the control's buffer using the given type.
 
@@ -582,6 +583,7 @@
         This function looks for a suitable wxRichTextFileHandler object.
     */
     virtual bool DoLoadFile(const wxString& file, int fileType);
+#endif // wxUSE_FFILE && wxUSE_STREAMS
 
 #ifdef DOXYGEN
     /**
@@ -596,6 +598,7 @@
                   int type = wxRICHTEXT_TYPE_ANY);
 #endif
 
+#if wxUSE_FFILE && wxUSE_STREAMS
     /**
         Helper function for SaveFile(). Saves the buffer content using the given type.
 
@@ -606,6 +609,7 @@
     */
     virtual bool DoSaveFile(const wxString& file = wxEmptyString,
                             int fileType = wxRICHTEXT_TYPE_ANY);
+#endif // wxUSE_FFILE && wxUSE_STREAMS
 
     /**
         Sets flags that change the behaviour of loading or saving.
diff -uNr 2.9.5-orig/include/wx/richtext/richtextprint.h 2.9.5-fixed/include/wx/richtext/richtextprint.h
--- 2.9.5-orig/include/wx/richtext/richtextprint.h	2013-07-16 18:39:22.000000000 +0200
+++ 2.9.5-fixed/include/wx/richtext/richtextprint.h	2013-07-25 17:01:58.273417900 +0200
@@ -165,11 +165,15 @@
     virtual ~wxRichTextPrinting();
 
     /// Preview the file or buffer
+#if wxUSE_FFILE && wxUSE_STREAMS
     bool PreviewFile(const wxString& richTextFile);
+#endif
     bool PreviewBuffer(const wxRichTextBuffer& buffer);
 
     /// Print the file or buffer
+#if wxUSE_FFILE && wxUSE_STREAMS
     bool PrintFile(const wxString& richTextFile, bool showPrintDialog = true);
+#endif
     bool PrintBuffer(const wxRichTextBuffer& buffer, bool showPrintDialog = true);
 
     /// Shows page setup dialog
diff -uNr 2.9.5-orig/src/common/debugrpt.cpp 2.9.5-fixed/src/common/debugrpt.cpp
--- 2.9.5-orig/src/common/debugrpt.cpp	2013-07-16 18:30:32.000000000 +0200
+++ 2.9.5-fixed/src/common/debugrpt.cpp	2013-07-25 16:43:22.325751700 +0200
@@ -33,8 +33,12 @@
 #if wxUSE_DEBUGREPORT && wxUSE_XML
 
 #include "wx/debugrpt.h"
+#if wxUSE_FFILE
+    #include "wx/ffile.h"
+#elif wxUSE_FILE
+    #include "wx/file.h"
+#endif
 
-#include "wx/ffile.h"
 #include "wx/filename.h"
 #include "wx/dir.h"
 #include "wx/dynlib.h"
@@ -288,17 +292,25 @@
                        const wxString& text,
                        const wxString& description)
 {
+#if wxUSE_FFILE || wxUSE_FILE
     wxASSERT_MSG( !wxFileName(filename).IsAbsolute(),
                   wxT("filename should be relative to debug report directory") );
 
-    wxFileName fn(GetDirectory(), filename);
-    wxFFile file(fn.GetFullPath(), wxT("w"));
-    if ( !file.IsOpened() || !file.Write(text) )
+    const wxString fullPath = wxFileName(GetDirectory(), filename).GetFullPath();
+#if wxUSE_FFILE
+    wxFFile file(fullPath, wxT("w"));
+#elif wxUSE_FILE
+    wxFile file(fullPath, wxFile::write);
+#endif
+    if ( !file.IsOpened() || !file.Write(text, wxConvAuto()) )
         return false;
 
     AddFile(filename, description);
 
     return true;
+#else // !wxUSE_FFILE && !wxUSE_FILE
+    return false;
+#endif
 }
 
 void wxDebugReport::RemoveFile(const wxString& name)
@@ -614,6 +626,8 @@
 
 bool wxDebugReportCompress::DoProcess()
 {
+#define HAS_FILE_STREAMS (wxUSE_STREAMS && (wxUSE_FILE || wxUSE_FFILE))
+#if HAS_FILE_STREAMS
     const size_t count = GetFilesCount();
     if ( !count )
         return false;
@@ -630,7 +644,14 @@
     fn.SetExt("zip");
 
     // create the streams
-    wxFFileOutputStream os(fn.GetFullPath(), wxT("wb"));
+    const wxString ofullPath = fn.GetFullPath();
+#if wxUSE_FFILE
+    wxFFileOutputStream os(ofullPath, wxT("wb"));
+#elif wxUSE_FILE
+    wxFileOutputStream os(ofullPath);
+#endif
+    if ( !os.IsOk() )
+        return false;
     wxZipOutputStream zos(os, 9);
 
     // add all files to the ZIP one
@@ -645,8 +666,12 @@
         if ( !zos.PutNextEntry(ze) )
             return false;
 
-        const wxFileName filename(GetDirectory(), name);
-        wxFFileInputStream is(filename.GetFullPath());
+        const wxString ifullPath = wxFileName(GetDirectory(), name).GetFullPath();
+#if wxUSE_FFILE
+        wxFFileInputStream is(ifullPath);
+#elif wxUSE_FILE
+        wxFileInputStream is(ifullPath);
+#endif
         if ( !is.IsOk() || !zos.Write(is).IsOk() )
             return false;
     }
@@ -654,9 +679,12 @@
     if ( !zos.Close() )
         return false;
 
-    m_zipfile = fn.GetFullPath();
+    m_zipfile = ofullPath;
 
     return true;
+#else
+    return false;
+#endif // HAS_FILE_STREAMS
 }
 
 // ----------------------------------------------------------------------------
diff -uNr 2.9.5-orig/src/common/textcmn.cpp 2.9.5-fixed/src/common/textcmn.cpp
--- 2.9.5-orig/src/common/textcmn.cpp	2013-07-16 18:30:36.000000000 +0200
+++ 2.9.5-fixed/src/common/textcmn.cpp	2013-07-25 17:12:07.707715900 +0200
@@ -915,6 +915,8 @@
             return true;
         }
     }
+#else
+    (void)filename;   // avoid compiler warning about unreferenced parameter
 #endif // wxUSE_FFILE
 
     wxLogError(_("File couldn't be loaded."));
@@ -936,6 +938,8 @@
 
         return true;
     }
+#else
+    (void)filename;   // avoid compiler warning about unreferenced parameter
 #endif // wxUSE_FFILE
 
     return false;
diff -uNr 2.9.5-orig/src/generic/dbgrptg.cpp 2.9.5-fixed/src/generic/dbgrptg.cpp
--- 2.9.5-orig/src/generic/dbgrptg.cpp	2013-07-16 18:31:28.000000000 +0200
+++ 2.9.5-fixed/src/generic/dbgrptg.cpp	2013-07-25 16:45:39.365196700 +0200
@@ -39,7 +39,11 @@
 #endif // WX_PRECOMP
 
 #include "wx/filename.h"
-#include "wx/ffile.h"
+#ifdef wxUSE_FFILE
+    #include "wx/ffile.h"
+#else
+    #include "wx/file.h"
+#endif
 #include "wx/mimetype.h"
 
 #include "wx/statline.h"
@@ -432,7 +436,12 @@
     wxFileName fn(m_dbgrpt.GetDirectory(), m_files[sel]);
     wxString str;
 
-    wxFFile file(fn.GetFullPath());
+    const wxString fullPath = fn.GetFullPath();
+#if wxUSE_FFILE
+    wxFFile file(fullPath);
+#elif wxUSE_FILE
+    wxFile file(fullPath);
+#endif
     if ( file.IsOpened() && file.ReadAll(&str) )
     {
         wxDumpPreviewDlg dlg(this, m_files[sel], str);
diff -uNr 2.9.5-orig/src/richtext/richtextbuffer.cpp 2.9.5-fixed/src/richtext/richtextbuffer.cpp
--- 2.9.5-orig/src/richtext/richtextbuffer.cpp	2013-07-16 18:30:12.000000000 +0200
+++ 2.9.5-fixed/src/richtext/richtextbuffer.cpp	2013-07-25 15:42:38.349247600 +0200
@@ -8317,6 +8317,7 @@
     return wildcard;
 }
 
+#if wxUSE_FFILE && wxUSE_STREAMS
 /// Load a file
 bool wxRichTextBuffer::LoadFile(const wxString& filename, wxRichTextFileType type)
 {
@@ -8345,7 +8346,9 @@
     else
         return false;
 }
+#endif // wxUSE_FFILE && wxUSE_STREAMS
 
+#if wxUSE_STREAMS
 /// Load from a stream
 bool wxRichTextBuffer::LoadFile(wxInputStream& stream, wxRichTextFileType type)
 {
@@ -8374,6 +8377,7 @@
     else
         return false;
 }
+#endif // wxUSE_STREAMS
 
 /// Copy the range to the clipboard
 bool wxRichTextBuffer::CopyToClipboard(const wxRichTextRange& range)
diff -uNr 2.9.5-orig/src/richtext/richtextctrl.cpp 2.9.5-fixed/src/richtext/richtextctrl.cpp
--- 2.9.5-orig/src/richtext/richtextctrl.cpp	2013-07-16 18:30:12.000000000 +0200
+++ 2.9.5-fixed/src/richtext/richtextctrl.cpp	2013-07-25 16:56:08.868640500 +0200
@@ -2665,7 +2665,7 @@
 // ----------------------------------------------------------------------------
 // file IO functions
 // ----------------------------------------------------------------------------
-
+#if wxUSE_FFILE && wxUSE_STREAMS
 bool wxRichTextCtrl::DoLoadFile(const wxString& filename, int fileType)
 {
     SetFocusObject(& GetBuffer(), true);
@@ -2707,6 +2707,7 @@
 
     return false;
 }
+#endif // wxUSE_FFILE && wxUSE_STREAMS
 
 // ----------------------------------------------------------------------------
 // wxRichTextCtrl specific functionality
diff -uNr 2.9.5-orig/src/richtext/richtextprint.cpp 2.9.5-fixed/src/richtext/richtextprint.cpp
--- 2.9.5-orig/src/richtext/richtextprint.cpp	2013-07-16 18:30:12.000000000 +0200
+++ 2.9.5-fixed/src/richtext/richtextprint.cpp	2013-07-25 17:01:50.737985900 +0200
@@ -506,6 +506,7 @@
     m_richTextBufferPreview = buf;
 }
 
+#if wxUSE_FFILE && wxUSE_STREAMS
 bool wxRichTextPrinting::PreviewFile(const wxString& richTextFile)
 {
     SetRichTextBufferPreview(new wxRichTextBuffer);
@@ -525,6 +526,7 @@
     p2->SetRichTextBuffer(m_richTextBufferPrinting);
     return DoPreview(p1, p2);
 }
+#endif // wxUSE_FFILE && wxUSE_STREAMS
 
 bool wxRichTextPrinting::PreviewBuffer(const wxRichTextBuffer& buffer)
 {
@@ -540,6 +542,7 @@
     return DoPreview(p1, p2);
 }
 
+#if wxUSE_FFILE && wxUSE_STREAMS
 bool wxRichTextPrinting::PrintFile(const wxString& richTextFile, bool showPrintDialog)
 {
     SetRichTextBufferPrinting(new wxRichTextBuffer);
@@ -557,6 +560,7 @@
     delete p;
     return ret;
 }
+#endif // wxUSE_FFILE && wxUSE_STREAMS
 
 bool wxRichTextPrinting::PrintBuffer(const wxRichTextBuffer& buffer, bool showPrintDialog)
 {
diff -uNr 2.9.5-orig/src/stc/stc.cpp 2.9.5-fixed/src/stc/stc.cpp
--- 2.9.5-orig/src/stc/stc.cpp	2013-07-16 18:30:12.000000000 +0200
+++ 2.9.5-fixed/src/stc/stc.cpp	2013-07-25 16:05:20.796172100 +0200
@@ -50,7 +50,7 @@
 #if wxUSE_FFILE
     #include "wx/ffile.h"
 #elif wxUSE_FILE
-    #include "wx/ffile.h"
+    #include "wx/file.h"
 #endif
 
 #ifdef __WXGTK__
