Index: tbarbase.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/tbarbase.h,v
retrieving revision 1.39
diff -b -u -2 -r1.39 tbarbase.h
--- tbarbase.h	2003/01/02 23:37:29	1.39
+++ tbarbase.h	2003/01/21 20:10:55
@@ -325,4 +325,7 @@
                                );
 
+    virtual wxToolBarToolBase *AddTool (wxToolBarToolBase *tool);
+    virtual wxToolBarToolBase *InsertTool (size_t pos, wxToolBarToolBase *tool);
+
     // add an arbitrary control to the toolbar, return TRUE if ok (notice that
     // the control will be deleted by the toolbar and that it will also adjust
Index: tbarbase.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/common/tbarbase.cpp,v
retrieving revision 1.46
diff -b -u -2 -r1.46 tbarbase.cpp
--- tbarbase.cpp	2002/06/02 10:57:13	1.46
+++ tbarbase.cpp	2003/01/21 20:14:23
@@ -177,4 +177,24 @@
 }
 
+wxToolBarToolBase *wxToolBarBase::AddTool(wxToolBarToolBase *tool)
+{
+    return InsertTool(GetToolsCount(), tool);
+}
+
+wxToolBarToolBase *wxToolBarBase::InsertTool(size_t pos, wxToolBarToolBase *tool)
+{
+    wxCHECK_MSG( pos <= GetToolsCount(), (wxToolBarToolBase *)NULL,
+                 _T("invalid position in wxToolBar::InsertTool()") );
+
+    if ( !tool || !DoInsertTool(pos, tool) )
+    {
+        return NULL;
+    }
+
+    m_tools.Insert(pos, tool);
+
+    return tool;
+}
+
 wxToolBarToolBase *wxToolBarBase::AddControl(wxControl *control)
 {
Index: toolbar.tex
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/docs/latex/wx/toolbar.tex,v
retrieving revision 1.26
diff -b -u -2 -r1.26 toolbar.tex
--- toolbar.tex	2003/01/18 00:16:34	1.26
+++ toolbar.tex	2003/01/21 20:23:28
@@ -187,7 +187,10 @@
 \param{wxObject* }{clientData = NULL}}
 
+\func{wxToolBarTool*}{AddTool}{\param{wxToolBarTool* }{tool}}
+
 Adds a tool to the toolbar. The first (short and most commonly used) version
 has fewer parameters than the full version at the price of not being able to
-specify some of the more rarely used button features.
+specify some of the more rarely used button features. The last version allows
+to add an existing tool.
 
 \wxheading{Parameters}
@@ -218,4 +221,6 @@
 retrieved later using \helpref{wxToolBar::GetToolClientData}{wxtoolbargettoolclientdata}.}
 
+\docparam{tool}{The tool to be added.}
+
 \wxheading{Remarks}
 
@@ -518,4 +523,7 @@
 \param{const wxBitmap\&}{ bitmap2 = wxNullBitmap}, \param{bool}{ isToggle = false},\rtfsp
 \param{wxObject* }{clientData = NULL}, \param{const wxString\& }{shortHelpString = ""}, \param{const wxString\& }{longHelpString = ""}}
+
+\func{wxToolBarTool *}{InsertTool}{\param{size\_t }{pos},\rtfsp
+\param{wxToolBarTool* }{tool}}
 
 Inserts the tool with the specified attributes into the toolbar at the given

 	  	 
