Index: src/xrc/xmlres.cpp
===================================================================
--- src/xrc/xmlres.cpp	(revision 61599)
+++ src/xrc/xmlres.cpp	(working copy)
@@ -2022,12 +2022,11 @@
 }
 
 static void AddStdXRCID_Records();
+static bool s_stdIDsAdded = false;
 
 /*static*/
 int wxXmlResource::DoGetXRCID(const char *str_id, int value_if_not_found)
 {
-    static bool s_stdIDsAdded = false;
-
     if ( !s_stdIDsAdded )
     {
         s_stdIDsAdded = true;
@@ -2070,6 +2069,7 @@
         CleanXRCID_Record(XRCID_Records[i]);
         XRCID_Records[i] = NULL;
     }
+    s_stdIDsAdded = false;
 }
 
 static void AddStdXRCID_Records()
