diff --git a/include/wx/dataview.h b/include/wx/dataview.h
index 573caf5227..c41ac8298c 100644
--- a/include/wx/dataview.h
+++ b/include/wx/dataview.h
@@ -144,21 +144,24 @@ public:
     {
         m_bold = false;
         m_italic = false;
+        m_strikethrough = false;
     }
 
     // setters
     void SetColour(const wxColour& colour) { m_colour = colour; }
     void SetBold( bool set ) { m_bold = set; }
     void SetItalic( bool set ) { m_italic = set; }
+    void SetStrikethrough( bool set ) { m_strikethrough = set; }
     void SetBackgroundColour(const wxColour& colour)  { m_bgColour = colour; }
 
     // accessors
     bool HasColour() const { return m_colour.IsOk(); }
     const wxColour& GetColour() const { return m_colour; }
 
-    bool HasFont() const { return m_bold || m_italic; }
+    bool HasFont() const { return m_bold || m_italic || m_strikethrough; }
     bool GetBold() const { return m_bold; }
     bool GetItalic() const { return m_italic; }
+    bool GetStrikethrough() const { return m_strikethrough; }
 
     bool HasBackgroundColour() const { return m_bgColour.IsOk(); }
     const wxColour& GetBackgroundColour() const { return m_bgColour; }
@@ -172,6 +175,7 @@ private:
     wxColour m_colour;
     bool     m_bold;
     bool     m_italic;
+    bool     m_strikethrough;
     wxColour m_bgColour;
 };
 
diff --git a/src/common/datavcmn.cpp b/src/common/datavcmn.cpp
index fd68880243..05a1021742 100644
--- a/src/common/datavcmn.cpp
+++ b/src/common/datavcmn.cpp
@@ -87,6 +87,8 @@ wxFont wxDataViewItemAttr::GetEffectiveFont(const wxFont& font) const
         f.MakeBold();
     if ( GetItalic() )
         f.MakeItalic();
+    if ( GetStrikethrough() )
+        f.MakeStrikethrough();
     return f;
 }
 
