--- utils.cpp_orig	2003-01-21 12:03:49.000000000 -0500
+++ utils.cpp	2003-01-21 12:18:07.000000000 -0500
@@ -561,7 +561,48 @@
               lTotalNumberOfClusters;
 
-        // FIXME: this is wrong, we should extract the root drive from path
-        //        instead, but this is the job for wxFileName...
-        if ( !::GetDiskFreeSpace(path,
+        // extract the root drive from path
+
+	char rootdir[MAX_PATH];
+	LPSTR tmp_str;    //required arg, a temp variable 
+
+	// need to find path for root directory on drive containing this file.
+	GetFullPathName(path.c_str(), sizeof(rootdir)/sizeof(rootdir[0]), rootdir, &tmp_str);
+
+	// truncate this to the name of the root directory (god how tedious)
+	if ((rootdir[0] == TEXT('\\')) && (rootdir[1] == TEXT('\\'))) 
+	{
+	    // path begins with  \\server\share\path so skip the first
+	    // three backslashes
+	    tmp_str = &rootdir[2];
+	    while (*tmp_str && (*tmp_str != TEXT('\\')))
+	    {
+                tmp_str++;
+	    }
+	    if (*tmp_str)
+	    {
+	        // advance past the third backslash
+	        tmp_str++;
+	     }
+	} 
+	else 
+	{
+	    // path must be drv:\path
+	    tmp_str = rootdir;
+        }
+
+	// find next backslash and put a null after it
+	while (*tmp_str && (*tmp_str != TEXT('\\'))) 
+	{
+            tmp_str++;
+	}
+	// found a backslash ?
+	if (*tmp_str) 
+	{
+	    // skip it and insert null
+	    tmp_str++;
+	    *tmp_str = TEXT('\0');
+	}
+
+        if ( !::GetDiskFreeSpace(rootdir,
                                  &lSectorsPerCluster,
                                  &lBytesPerSector,

 	  	 
