Index: src/common/valgen.cpp
===================================================================
--- src/common/valgen.cpp	(revision 68217)
+++ src/common/valgen.cpp	(working copy)
@@ -121,6 +121,9 @@
 #if wxUSE_DATETIME
     m_pDateTime = val.m_pDateTime;
 #endif // wxUSE_DATETIME
+    m_pFileName = val.m_pFileName;
+    m_pFloat = val.m_pFloat;
+    m_pDouble = val.m_pDouble;
 
     return true;
 }
