Index: scrolledpanel.py
===================================================================
--- scrolledpanel.py	(revision 54672)
+++ scrolledpanel.py	(working copy)
@@ -79,7 +79,10 @@
     def OnChildFocus(self, evt):
         # If the child window that gets the focus is not visible,
         # this handler will try to scroll enough to see it.
-        evt.Skip()
+
+        # Trap the event on Mac to prevent erratic jumping of the window
+        if wx.Platform != '__WXMAC__':
+            evt.Skip()
         child = evt.GetWindow()
         self.ScrollChildIntoView(child)
         
