--- C:/Documents and Settings/lazywind/Local Settings/Temp/grid.h-revBASE.svn001.tmp.h	Wed Nov 12 00:38:10 2008
+++ C:/wxWidgets-2.9.0/interface/wx/grid.h	Wed Nov 12 00:35:09 2008
@@ -315,6 +315,18 @@
      */
     virtual bool DeleteRows(size_t pos = 0, size_t numRows = 1);
 
+    /**
+        Move rows from oldPos to newPos at the table.
+
+        @param oldPos
+            The first row to be moved.
+        @param newPos
+            The new position of the rows after moved.
+        @param numRows
+            The number of rows to be moved.
+     */
+    virtual bool MoveRows( size_t oldPos, size_t newPos = 0, size_t numRows = 1 );
+
     /// Exactly the same as InsertRows() but for columns.
     virtual bool InsertCols(size_t pos = 0, size_t numCols = 1);
 
@@ -324,6 +336,18 @@
     /// Exactly the same as DeleteRows() but for columns.
     virtual bool DeleteCols(size_t pos = 0, size_t numCols = 1);
 
+    /**
+        Move columns from oldPos to newPos at the table.
+
+        @param oldPos
+            The first column to be moved.
+        @param newPos
+            The new position of the columns after moved.
+        @param numRows
+            The number of columns to be moved.
+     */
+    virtual bool MoveCols( size_t oldPos, size_t newPos = 0, size_t numCols = 1 );
+
     //@}
 
     /**
@@ -1477,6 +1501,13 @@
     bool CanDragGridSize() const;
 
     /**
+        Returns @true if rows can be moved by dragging with the mouse.
+
+        Rows can be moved by dragging on their labels.
+    */
+    bool CanDragRowMove() const;
+
+    /**
         Returns @true if rows can be resized by dragging with the mouse.
 
         Rows can be resized by dragging the edges of their labels. If grid line
@@ -1608,6 +1639,13 @@
     void DisableDragGridSize();
 
     /**
+        Disables row moving by dragging with the mouse.
+
+        Equivalent to passing @false to EnableDragRowMove().
+    */
+    void DisableDragRowMove();
+
+    /**
         Disables row sizing by dragging with the mouse.
 
         Equivalent to passing @false to EnableDragRowSize().
@@ -1639,6 +1677,11 @@
     void EnableDragGridSize(bool enable = true);
 
     /**
+        Enables or disables row moving by dragging with the mouse.
+    */
+    void EnableDragRowMove(bool enable = true);
+
+    /**
         Enables or disables row sizing by dragging with the mouse.
     */
     void EnableDragRowSize(bool enable = true);
@@ -1763,11 +1806,6 @@
     //@}
 
     /**
-        Returns the column ID of the specified column position.
-    */
-    int GetColAt(int colPos) const;
-
-    /**
         Returns the pen used for vertical grid lines.
 
         This virtual function may be overridden in derived classes in order to
@@ -1812,11 +1850,6 @@
     int GetColMinimalAcceptableWidth() const;
 
     /**
-        Returns the position of the specified column.
-    */
-    int GetColPos(int colID) const;
-
-    /**
         Returns the width of the specified column.
     */
     int GetColSize(int col) const;
@@ -2660,11 +2693,6 @@
         returned by GetColMinimalAcceptableWidth().
     */
     void SetColMinimalWidth(int col, int width);
-
-    /**
-        Sets the position of the specified column.
-    */
-    void SetColPos(int colID, int newPos);
 
     /**
         Sets the width of the specified column.
