--- wxWidgets-2.6.3\src\msw\main.cpp	Sun Aug 28 15:11:18 2005
+++ wxWidgets-2.6.3.neu\src\msw\main.cpp	Tue Mar 28 08:35:24 2006
@@ -107,4 +107,5 @@
 // flag telling us whether the application wants to handle exceptions at all
 static bool gs_handleExceptions = false;
+static LPTOP_LEVEL_EXCEPTION_FILTER gs_oldFilter = NULL;
 
 static void wxFatalExit()
@@ -175,4 +176,14 @@
     // assume this can only be called from the main thread
     gs_handleExceptions = doit;
+
+    if (doit && gs_oldFilter == NULL)
+    {
+        gs_oldFilter = SetUnhandledExceptionFilter((LPTOP_LEVEL_EXCEPTION_FILTER)wxGlobalSEHandler);
+    }
+    else if (!doit && gs_oldFilter != NULL)
+    {
+        SetUnhandledExceptionFilter(gs_oldFilter);
+        gs_oldFilter = NULL;
+    }
 
 #if wxUSE_CRASHREPORT

 	  	 
