Index: samples/text/text.cpp
===================================================================
--- samples/text/text.cpp	(revision 62553)
+++ samples/text/text.cpp	(working copy)
@@ -1050,9 +1050,9 @@
     m_password = new MyTextCtrl( this, wxID_ANY, wxT(""),
       wxPoint(10,50), wxSize(140,wxDefaultCoord), wxTE_PASSWORD );
 
-    m_readonly = new MyTextCtrl( this, wxID_ANY, wxT("Read only"),
+    m_readonly = new MyTextCtrl( this, wxID_ANY, wxT(""),
       wxPoint(10,90), wxSize(140,wxDefaultCoord), wxTE_READONLY );
-
+    m_readonly->WriteText(wxT("This text won't be written"));
     m_limited = new MyTextCtrl(this, wxID_ANY, wxT("Max 8 ch"),
                               wxPoint(10, 130), wxSize(140, wxDefaultCoord));
     m_limited->SetMaxLength(8);
