Index: src/richtext/richtextbuffer.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/richtext/richtextbuffer.cpp,v
retrieving revision 1.83
diff -u -t -p -r1.83 richtextbuffer.cpp
--- src/richtext/richtextbuffer.cpp	2007/05/18 12:30:47	1.83
+++ src/richtext/richtextbuffer.cpp	2007/05/19 19:30:25
@@ -7902,7 +7902,7 @@ wxTextAttrEx wxTextAttrEx::CombineEx(con
 
 IMPLEMENT_CLASS(wxRichTextFileHandler, wxObject)
 
-#if wxUSE_STREAMS
+#if wxUSE_FFILE
 bool wxRichTextFileHandler::LoadFile(wxRichTextBuffer *buffer, const wxString& filename)
 {
     wxFFileInputStream stream(filename);
@@ -7920,7 +7920,7 @@ bool wxRichTextFileHandler::SaveFile(wxR
 
     return false;
 }
-#endif // wxUSE_STREAMS
+#endif // wxUSE_FFILE
 
 /// Can we handle this filename (if using files)? By default, checks the extension.
 bool wxRichTextFileHandler::CanHandle(const wxString& filename) const

 	  	 
