Index: samples/docview/docview.cpp
===================================================================
--- samples/docview/docview.cpp	(revision 63095)
+++ samples/docview/docview.cpp	(working copy)
@@ -55,6 +55,7 @@
 #include "view.h"
 
 #include "wx/cmdline.h"
+#include "wx/artprov.h"
 
 #ifdef __WXMAC__
     #include "wx/filename.h"
@@ -202,6 +203,9 @@
                                      wxDefaultPosition,
                                      wxSize(500, 400));
     }
+    wxToolBar* tb = frame->CreateToolBar();
+    tb->AddTool(wxID_OPEN, _("&Open..."), wxArtProvider::GetBitmap(wxART_FILE_OPEN));
+    tb->Realize();
 
     // and its menu bar
     wxMenu *menuFile = new wxMenu;
