diff -ru wxWidgets.orig/src/msw/filedlg.cpp wxWidgets/src/msw/filedlg.cpp
--- wxWidgets.orig/src/msw/filedlg.cpp	2008-06-20 10:00:38.000000000 +0200
+++ wxWidgets/src/msw/filedlg.cpp	2008-07-03 21:03:39.000000000 +0200
@@ -438,6 +438,13 @@
     DWORD errCode;
     bool success = DoShowCommFileDialog(&of, m_windowStyle, &errCode);
 
+    if ( !success && errCode == FNERR_INVALIDFILENAME && of.lpstrFile[0] )
+    {
+        // this can happen if the default file name is invalid, try without it now
+        of.lpstrFile[0] = _T('\0');
+        success = DoShowCommFileDialog(&of, m_windowStyle, &errCode);
+    }
+
 #ifdef wxTRY_SMALLER_OPENFILENAME
     // the system might be too old to support the new version file dialog
     // boxes, try with the old size
