Index: src/msw/checkbox.cpp
===================================================================
--- src/msw/checkbox.cpp	(revision 53750)
+++ src/msw/checkbox.cpp	(working copy)
@@ -191,6 +191,8 @@
         msStyle |= BS_LEFTTEXT | BS_RIGHT;
     }
 
+	msStyle |= BS_MULTILINE;
+
     return MSWCreateControl(wxT("BUTTON"), msStyle, pos, size, label, 0);
 }
 
@@ -202,9 +204,10 @@
 {
     static int s_checkSize = 0;
 
+    wxScreenDC dc;
+
     if ( !s_checkSize )
     {
-        wxScreenDC dc;
         dc.SetFont(wxSystemSettings::GetFont(wxSYS_DEFAULT_GUI_FONT));
 
         s_checkSize = dc.GetCharHeight();
@@ -215,7 +218,8 @@
     int wCheckbox, hCheckbox;
     if ( !str.empty() )
     {
-        GetTextExtent(GetLabelText(str), &wCheckbox, &hCheckbox);
+		dc.GetMultiLineTextExtent(GetLabelText(str), &wCheckbox, &hCheckbox);
+
         wCheckbox += s_checkSize + GetCharWidth();
 
         if ( hCheckbox < s_checkSize )
