diff -b -u -2 -r1.12 filetype.tex
--- docs/latex/wx/filetype.tex	5 Aug 2002 18:49:38 -0000	1.12
+++ docs/latex/wx/filetype.tex	7 Oct 2002 06:45:20 -0000
@@ -153,8 +153,11 @@
 \membersection{wxFileType::GetIcon}\label{wxfiletypegeticon}
 
-\func{bool}{GetIcon}{\param{wxIcon*}{ icon}}
+\func{bool}{GetIcon}{\param{wxIcon*}{ icon}, \param{wxString*}{ iconFile = NULL}, \param{int*}{ iconIndex = NULL}, \param{int}{ iconSize = wxICON\_LARGE}}
 
 If the function returns {\tt TRUE}, the icon associated with this file type will be
-created and assigned to the {\it icon} parameter.
+created and assigned to the {\it icon} parameter. {\it iconFile} is assigned the file name
+that contains the icon and {\it iconIndex} is assigned the index of the icon
+(windows and unix only). A 32x32 icon is assigned if {\it iconSize} is wxICON\_LARGE
+and a 16x16 icon is assigned if {\it iconSize} is wxICON\_SMALL (windows only).
 
 {\bf Unix:} MIME manager gathers information about icons from GNOME
diff -b -u -2 -r1.22 mimetype.h
--- include/wx/mimetype.h	8 Sep 2002 00:49:48 -0000	1.22
+++ include/wx/mimetype.h	7 Oct 2002 06:45:21 -0000
@@ -47,4 +47,7 @@
 };
 
+#define wxICON_LARGE 0
+#define wxICON_SMALL 1
+
 /*
     TODO: would it be more convenient to have this class?
@@ -215,5 +218,6 @@
     bool GetIcon(wxIcon *icon,
                  wxString *iconFile = NULL,
-                 int *iconIndex = NULL) const;
+                 int *iconIndex = NULL,
+                 int iconSize = wxICON_LARGE) const;
         // get a brief file type description ("*.txt" => "text document")
     bool GetDescription(wxString *desc) const;
diff -b -u -2 -r1.7 mimetype.h
--- include/wx/msw/mimetype.h	26 Jun 2001 20:59:07 -0000	1.7
+++ include/wx/msw/mimetype.h	7 Oct 2002 06:45:22 -0000
@@ -45,5 +45,6 @@
     bool GetMimeType(wxString *mimeType) const;
     bool GetMimeTypes(wxArrayString& mimeTypes) const;
-    bool GetIcon(wxIcon *icon, wxString *sCommand = NULL, int *iIndex = NULL) const;
+    bool GetIcon(wxIcon *icon, wxString *sCommand = NULL, int *iIndex = NULL,
+                 int iconSize = wxICON_LARGE) const;
     bool GetDescription(wxString *desc) const;
     bool GetOpenCommand(wxString *openCmd,
diff -b -u -2 -r1.22 mimecmn.cpp
--- src/common/mimecmn.cpp	8 Sep 2002 00:49:48 -0000	1.22
+++ src/common/mimecmn.cpp	7 Oct 2002 06:45:24 -0000
@@ -272,5 +272,6 @@
 bool wxFileType::GetIcon(wxIcon *icon,
                          wxString *iconFile,
-                         int *iconIndex) const
+                         int *iconIndex,
+                         int iconSize) const
 {
     if ( m_info )
@@ -292,5 +293,7 @@
     }
 
-#if defined(__WXMSW__) || defined(__UNIX__)
+#if defined(__WXMSW__)
+    return m_impl->GetIcon(icon, iconFile, iconIndex, iconSize);
+#elif defined(__UNIX__)
     return m_impl->GetIcon(icon, iconFile, iconIndex);
 #else
diff -b -u -2 -r1.15 mimetype.cpp
--- src/msw/mimetype.cpp	15 Jul 2002 18:29:52 -0000	1.15
+++ src/msw/mimetype.cpp	7 Oct 2002 06:45:25 -0000
@@ -334,5 +334,6 @@
 bool wxFileTypeImpl::GetIcon(wxIcon *icon,
                              wxString *iconFile,
-                             int *iconIndex) const
+                             int *iconIndex,
+                             int iconSize) const
 {
 #if wxUSE_GUI
@@ -365,5 +366,9 @@
             int nIndex = wxAtoi(strIndex);
 
-            HICON hIcon = ExtractIcon(GetModuleHandle(NULL), strExpPath, nIndex);
+            HICON hIcon, hIconLarge, hIconSmall;
+            ExtractIconEx(strExpPath, nIndex, &hIconLarge, &hIconSmall, 1);
+
+            hIcon = (iconSize == wxICON_LARGE) ? hIconLarge : hIconSmall;
+
             switch ( (int)hIcon ) {
                 case 0: // means no icons were found

 	  	 
