Index: mdi.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/mac/carbon/mdi.cpp,v
retrieving revision 1.46
diff -u -r1.46 mdi.cpp
--- mdi.cpp	2006/10/09 05:14:38	1.46
+++ mdi.cpp	2007/04/28 00:34:40
@@ -123,6 +123,8 @@
 
 void wxMDIParentFrame::AddChild(wxWindowBase *child)
 {
+    wxFrame::AddChild(child);
+
     if ( !m_currentChild )
     {
         m_currentChild = wxDynamicCast(child, wxMDIChildFrame);
@@ -134,8 +136,6 @@
             m_shouldBeShown = true;
         }
     }
-
-    wxFrame::AddChild(child);
 }
 
 void wxMDIParentFrame::RemoveChild(wxWindowBase *child)

 	  	 
