diff -rU2 wxWidgets.orig/include/wx/msw/statbox.h wxWidgets/include/wx/msw/statbox.h
--- wxWidgets.orig/include/wx/msw/statbox.h	2005-04-13 07:01:04.000000000 +1000
+++ wxWidgets/include/wx/msw/statbox.h	2005-04-15 18:03:18.935155352 +1000
@@ -63,4 +63,6 @@
     // paint the given rectangle with our background brush/colour
     void PaintBackground(wxDC& dc, const struct tagRECT& rc);
+    // paint the foreground of the static box
+    virtual void PaintForeground(wxDC& dc, const struct tagRECT& rc);
 
     void OnPaint(wxPaintEvent& event);
diff -rU2 wxWidgets.orig/src/msw/statbox.cpp wxWidgets/src/msw/statbox.cpp
--- wxWidgets.orig/src/msw/statbox.cpp	2005-04-15 06:46:55.000000000 +1000
+++ wxWidgets/src/msw/statbox.cpp	2005-04-15 18:05:07.794606200 +1000
@@ -322,4 +322,13 @@
 }
 
+void wxStaticBox::PaintForeground(wxDC& dc, const RECT& WXUNUSED(rc))
+{
+    // NB: neither setting the text colour nor transparent background mode
+    //     doesn't change anything: the static box def window proc still
+    //     draws the label in its own colours, so if we want to have control
+    //     over this we really have to draw everything ourselves
+    MSWDefWindowProc(WM_PAINT, (WPARAM)GetHdcOf(dc), 0);
+}
+
 void wxStaticBox::OnPaint(wxPaintEvent& WXUNUSED(event))
 {
@@ -333,11 +342,5 @@
 
     PaintBackground(memdc, rc);
-
-    // NB: neither setting the text colour nor transparent background mode
-    //     doesn't change anything: the static box def window proc still
-    //     draws the label in its own colours, so if we want to have control
-    //     over this we really have to draw everything ourselves
-    MSWDefWindowProc(WM_PAINT, (WPARAM)GetHdcOf(memdc), 0);
-
+    PaintForeground(memdc, rc);
 
     // now only blit the static box border itself, not the interior, to avoid

 	  	 
