diff --git a/include/wx/dir.h b/include/wx/dir.h
index 83dab66..abe66c8 100644
--- a/include/wx/dir.h
+++ b/include/wx/dir.h
@@ -22,14 +22,16 @@ class WXDLLIMPEXP_FWD_BASE wxArrayString;
 // constants
 // ----------------------------------------------------------------------------
 
-// these flags define what kind of filenames is included in the list of files
-// enumerated by GetFirst/GetNext
+// These flags affect the behaviour of GetFirst/GetNext and Traverse.
+// They define what types are included in the list of items they produce.
+// Note that wxDIR_NO_FOLLOW is relevant only on Unix
 enum wxDirFlags
 {
     wxDIR_FILES     = 0x0001,       // include files
     wxDIR_DIRS      = 0x0002,       // include directories
     wxDIR_HIDDEN    = 0x0004,       // include hidden files
     wxDIR_DOTDOT    = 0x0008,       // include '.' and '..'
+    wxDIR_NO_FOLLOW = 0x0010,       // don't dereference any symlink
 
     // by default, enumerate everything except '.' and '..'
     wxDIR_DEFAULT   = wxDIR_FILES | wxDIR_DIRS | wxDIR_HIDDEN
diff --git a/interface/wx/dir.h b/interface/wx/dir.h
index f4b25eb..2c75322 100644
--- a/interface/wx/dir.h
+++ b/interface/wx/dir.h
@@ -94,8 +94,10 @@ public:
 
 
 /**
-    These flags define what kind of filenames are included in the list of files
-    enumerated by wxDir::GetFirst() and wxDir::GetNext().
+    These flags affect the behaviour of GetFirst/GetNext and Traverse, 
+    determining what types are included in the list of items they produce.
+    Note that wxDIR_NO_FOLLOW is relevant only on Unix, 
+    and is not included in wxDIR_DEFAULT.
 */
 enum wxDirFlags
 {
@@ -103,6 +105,7 @@ enum wxDirFlags
     wxDIR_DIRS      = 0x0002,   ///< Includes directories.
     wxDIR_HIDDEN    = 0x0004,   ///< Includes hidden files.
     wxDIR_DOTDOT    = 0x0008,   ///< Includes "." and "..".
+    wxDIR_NO_FOLLOW = 0x0010,   ///< Don't dereference any symlink
 
     //! Combination of the @c wxDIR_FILES, @c wxDIR_DIRS, @c wxDIR_HIDDEN flags
     //! defined above.
diff --git a/src/common/dircmn.cpp b/src/common/dircmn.cpp
index 1638cff..d30c323 100644
--- a/src/common/dircmn.cpp
+++ b/src/common/dircmn.cpp
@@ -115,7 +115,9 @@ size_t wxDir::Traverse(wxDirTraverser& sink,
     if ( flags & wxDIR_DIRS )
     {
         wxString dirname;
-        for ( bool cont = GetFirst(&dirname, wxEmptyString, wxDIR_DIRS | (flags & wxDIR_HIDDEN) );
+        for ( bool cont = GetFirst(&dirname, wxEmptyString,
+                                   wxDIR_DIRS | (flags & wxDIR_HIDDEN)
+                                              | (flags & wxDIR_NO_FOLLOW) );
               cont;
               cont = cont && GetNext(&dirname) )
         {
diff --git a/src/unix/dir.cpp b/src/unix/dir.cpp
index f0e6cd9..dff1521 100644
--- a/src/unix/dir.cpp
+++ b/src/unix/dir.cpp
@@ -150,10 +150,13 @@ bool wxDirData::Read(wxString *filename)
             break;
         }
 
-        // check the type now: notice that we want to check the type of this
-        // path itself and not whatever it points to in case of a symlink
+        // check the type now: notice that we may want to check the type of
+        // the path itself and not whatever it points to in case of a symlink
         wxFileName fn = wxFileName::DirName(path + de_d_name);
-        fn.DontFollowLink();
+        if ((m_flags & wxDIR_NO_FOLLOW) != 0)
+        {
+            fn.DontFollowLink();
+        }
 
         if ( !(m_flags & wxDIR_FILES) && !fn.DirExists() )
         {
