Index: control.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/msw/control.cpp,v
retrieving revision 1.110
diff -u -r1.110 control.cpp
--- control.cpp	2006/10/31 08:49:41	1.110
+++ control.cpp	2006/11/10 22:59:26
@@ -170,10 +170,17 @@
     if ( !m_hasFont )
     {
 #if wxUSE_LISTCTRL || wxUSE_TREECTRL
+        bool bMayNeedToSetFont = false;
+#if wxUSE_LISTCTRL
+        bMayNeedToSetFont = bMayNeedToSetFont || (wxDynamicCastThis(wxListCtrl) != NULL);
+#endif
+#if wxUSE_TREECTRL
+        bMayNeedToSetFont = bMayNeedToSetFont || (wxDynamicCastThis(wxTreeCtrl) != NULL);
+#endif
         // if we set a font for {list,tree}ctrls and the font size is changed in
         // the display properties then the font size for these controls doesn't
         // automatically adjust when they receive WM_SETTINGCHANGE
-        if ( wxDynamicCastThis(wxListCtrl) || wxDynamicCastThis(wxTreeCtrl) )
+        if ( bMayNeedToSetFont )
         {
             // not sure if we need to explicitly set the font here for Win95/NT4
             // but we definitely can't do it for any newer version


 	  	 
