diff --git a/samples/minimal/minimal.cpp b/samples/minimal/minimal.cpp
index 0d91f7f..53540f8 100644
--- a/samples/minimal/minimal.cpp
+++ b/samples/minimal/minimal.cpp
@@ -66,8 +66,11 @@ public:
     // event handlers (these functions should _not_ be virtual)
     void OnQuit(wxCommandEvent& event);
     void OnAbout(wxCommandEvent& event);
+    void OnClick(wxCommandEvent &event);
 
 private:
+    wxPanel *panel_1;
+    wxButton *button_1;
     // any class wishing to process wxWidgets events must use this macro
     wxDECLARE_EVENT_TABLE();
 };
@@ -178,10 +181,26 @@ MyFrame::MyFrame(const wxString& title)
     CreateStatusBar(2);
     SetStatusText("Welcome to wxWidgets!");
 #endif // wxUSE_STATUSBAR
+    panel_1 = new wxPanel(this, wxID_ANY, wxDefaultPosition, wxDefaultSize, wxBORDER_SUNKEN);
+    button_1 = new wxButton(this, wxID_ANY, wxT("button_1"));
+    wxBoxSizer* sizer_1 = new wxBoxSizer(wxVERTICAL);
+    wxBoxSizer* sizer_2 = new wxBoxSizer(wxHORIZONTAL);
+    sizer_2->Add(panel_1, 1, wxEXPAND, 0);
+    sizer_2->Add(5, 5, 0, wxEXPAND, 0);
+    sizer_2->Add(button_1, 0, 0, 0);
+    sizer_1->Add(sizer_2, 1, wxEXPAND, 0);
+    SetSizer(sizer_1);
+    Layout();
+    button_1->Bind( wxEVT_BUTTON, &MyFrame::OnClick, this );
 }
 
 
 // event handlers
+void MyFrame::OnClick(wxCommandEvent &event)
+{
+    panel_1->Destroy();
+}
+
 
 void MyFrame::OnQuit(wxCommandEvent& WXUNUSED(event))
 {
