Index: src/common/gifdecod.cpp
===================================================================
--- src/common/gifdecod.cpp	(revision 54661)
+++ src/common/gifdecod.cpp	(working copy)
@@ -709,8 +709,7 @@
                 delay = 10 * (buf[2] + 256 * buf[3]);
 
                 // read transparent colour index, if used
-                if (buf[1] & 0x01)
-                    transparent = buf[4];
+                transparent = (buf[1] & 0x01) ? buf[4] : -1;
 
                 // read disposal method
                 disposal = (wxAnimationDisposal)(((buf[1] & 0x1C) >> 2) - 1);
