--- src/generic/grid.cpp.orig	2006-02-28 20:33:26.387124800 -0800
+++ src/generic/grid.cpp	2006-02-28 21:20:55.952444800 -0800
@@ -7869,7 +7869,13 @@
         editor->DecRef();
         attr->DecRef();
 
-        m_gridWin->SetFocus();
+        bool has_focus = false;
+        for(wxWindow * focus = FindFocus(); focus && !has_focus; focus = focus->GetParent()) {
+            has_focus = (focus == this);
+        }
+        if(has_focus) {
+            m_gridWin->SetFocus();
+        }
 
         // refresh whole row to the right
         wxRect rect( CellToRect(row, col) );

 	  	 
