--- wxWindows/src/common/string.cpp	Fri Sep 28 07:34:12 2001
+++ wxWindowsnew/src/common/string.cpp	Fri Oct 12 13:44:18 2001
@@ -3,5 +3,6 @@
 // Purpose:     wxString class
 // Author:      Vadim Zeitlin
-// Modified by:
+// Modified by: Robert Cavanaugh
+//              Change default base in wxStrtol to hexadecimal
 // Created:     29/01/98
 // RCS-ID:      $Id: string.cpp,v 1.140 2001/09/28 13:34:12 VZ Exp $
@@ -1091,4 +1092,6 @@
 // ----------------------------------------------------------------------------
 
+// Modified 8/20/2001 Robert Cavanaugh
+// allow wxStrtol to default to hexadecimal
 bool wxString::ToLong(long *val) const
 {
@@ -1097,5 +1100,5 @@
     const wxChar *start = c_str();
     wxChar *end;
-    *val = wxStrtol(start, &end, 10);
+    *val = wxStrtol(start, &end, 0);
 
     // return TRUE only if scan was stopped by the terminating NUL and if the
@@ -1110,5 +1113,5 @@
     const wxChar *start = c_str();
     wxChar *end;
-    *val = wxStrtoul(start, &end, 10);
+    *val = wxStrtoul(start, &end, 0);
 
     // return TRUE only if scan was stopped by the terminating NUL and if the

 	  	 
