Index: /home/johannes/workspace/wxWidgets/src/common/combocmn.cpp
===================================================================
--- /home/johannes/workspace/wxWidgets/src/common/combocmn.cpp	(revision 53101)
+++ /home/johannes/workspace/wxWidgets/src/common/combocmn.cpp	(working copy)
@@ -872,6 +872,11 @@
         m_text->Create(this, wxID_ANY, m_valueString,
                        wxDefaultPosition, wxSize(10,-1),
                        style, validator);
+    // Set the back- and foreground on GTK explicitly
+#ifdef __WXGTK__
+        m_text->SetBackgroundColour(wxSystemSettings::GetColour(wxSYS_COLOUR_WINDOW));
+        m_text->SetForegroundColour(wxSystemSettings::GetColour(wxSYS_COLOUR_WINDOWTEXT));
+#endif
     }
 }
 
@@ -883,6 +888,14 @@
 #ifndef __WXMAC__
     SetOwnBackgroundColour(wxSystemSettings::GetColour(wxSYS_COLOUR_WINDOW));
 #endif
+    // Change the back- and foreground on GTK explicitly
+#ifdef __WXGTK__
+    if (m_text) 
+    {
+        m_text->SetBackgroundColour(wxSystemSettings::GetColour(wxSYS_COLOUR_WINDOW));
+        m_text->SetForegroundColour(wxSystemSettings::GetColour(wxSYS_COLOUR_WINDOWTEXT));
+    }
+#endif
 }
 
 wxComboCtrlBase::~wxComboCtrlBase()

 	  	 
