Index: src/common/docview.cpp
===================================================================
--- src/common/docview.cpp	(revision 58554)
+++ src/common/docview.cpp	(working copy)
@@ -394,21 +394,23 @@
 }
 
 #if wxUSE_STD_IOSTREAM
-wxSTD istream& wxDocument::LoadObject(wxSTD istream& stream)
+bool wxDocument::LoadObject(wxSTD istream& WXUNUSED(stream))
 #else
-wxInputStream& wxDocument::LoadObject(wxInputStream& stream)
+bool wxDocument::LoadObject(wxInputStream& WXUNUSED(stream))
 #endif
 {
-    return stream;
+    // return true indicates success
+    return true;
 }
 
 #if wxUSE_STD_IOSTREAM
-wxSTD ostream& wxDocument::SaveObject(wxSTD ostream& stream)
+bool wxDocument::SaveObject(wxSTD ostream& WXUNUSED(stream))
 #else
-wxOutputStream& wxDocument::SaveObject(wxOutputStream& stream)
+bool wxDocument::SaveObject(wxOutputStream& WXUNUSED(stream))
 #endif
 {
-    return stream;
+    /// return true indicates success
+    return true;
 }
 
 bool wxDocument::Revert()
@@ -616,13 +618,9 @@
         return false;
     }
 
-#if wxUSE_STD_IOSTREAM
-    LoadObject(store);
-    if ( !store )
-#else
-    int res = LoadObject(store).GetLastError();
-    if ( res != wxSTREAM_NO_ERROR && res != wxSTREAM_EOF )
-#endif
+    bool bOK;
+    bOK = LoadObject(store);
+    if ( !bOK )
     {
         wxLogError(_("Failed to read document from the file \"%s\"."), file);
         return false;
Index: include/wx/docview.h
===================================================================
--- include/wx/docview.h	(revision 58554)
+++ include/wx/docview.h	(working copy)
@@ -89,11 +89,11 @@
     virtual bool Revert();
 
 #if wxUSE_STD_IOSTREAM
-    virtual wxSTD ostream& SaveObject(wxSTD ostream& stream);
-    virtual wxSTD istream& LoadObject(wxSTD istream& stream);
+    virtual bool SaveObject(wxSTD ostream& stream);
+    virtual bool LoadObject(wxSTD istream& stream);
 #else
-    virtual wxOutputStream& SaveObject(wxOutputStream& stream);
-    virtual wxInputStream& LoadObject(wxInputStream& stream);
+    virtual bool SaveObject(wxOutputStream& stream);
+    virtual bool LoadObject(wxInputStream& stream);
 #endif
 
     // Called by wxWidgets
Index: interface/wx/docview.h
===================================================================
--- interface/wx/docview.h	(revision 58554)
+++ interface/wx/docview.h	(working copy)
@@ -1152,12 +1152,13 @@
         Override this function and call it from your own LoadObject() before
         streaming your own data. LoadObject() is called by the framework
         automatically when the document contents need to be loaded.
+        Returns true only on success.
 
         @note This version of LoadObject() may not exist depending on how
               wxWidgets was configured.
     */
-    virtual istream& LoadObject(istream& stream);
-    virtual wxInputStream& LoadObject(wxInputStream& stream);
+    virtual bool LoadObject(istream& stream);
+    virtual bool LoadObject(wxInputStream& stream);
     //@}
 
     /**
@@ -1263,12 +1264,13 @@
         Override this function and call it from your own SaveObject() before
         streaming your own data. SaveObject() is called by the framework
         automatically when the document contents need to be saved.
+        Returns true only on success.
 
         @note This version of SaveObject() may not exist depending on how
               wxWidgets was configured.
     */
-    virtual ostream& SaveObject(ostream& stream);
-    virtual wxOutputStream& SaveObject(wxOutputStream& stream);
+    virtual bool SaveObject(ostream& stream);
+    virtual bool SaveObject(wxOutputStream& stream);
     //@}
 
     /**
Index: samples/docview/doc.cpp
===================================================================
--- samples/docview/doc.cpp	(revision 58554)
+++ samples/docview/doc.cpp	(working copy)
@@ -44,7 +44,7 @@
 
 IMPLEMENT_DYNAMIC_CLASS(DrawingDocument, wxDocument)
 
-DocumentOstream& DrawingDocument::SaveObject(DocumentOstream& ostream)
+bool DrawingDocument::SaveObject(DocumentOstream& ostream)
 {
 #if wxUSE_STD_IOSTREAM
     DocumentOstream& stream = ostream;
@@ -52,8 +52,6 @@
     wxTextOutputStream stream(ostream);
 #endif
 
-    wxDocument::SaveObject(ostream);
-
     const wxInt32 count = m_doodleSegments.size();
     stream << count << '\n';
 
@@ -63,10 +61,10 @@
         stream << '\n';
     }
 
-    return ostream;
+    return (bool)ostream;
 }
 
-DocumentIstream& DrawingDocument::LoadObject(DocumentIstream& istream)
+bool DrawingDocument::LoadObject(DocumentIstream& istream)
 {
 #if wxUSE_STD_IOSTREAM
     DocumentIstream& stream = istream;
@@ -74,8 +72,6 @@
     wxTextInputStream stream(istream);
 #endif
 
-    wxDocument::LoadObject(istream);
-
     wxInt32 count = 0;
     stream >> count;
 
@@ -85,8 +81,17 @@
         segment.LoadObject(istream);
         m_doodleSegments.push_back(segment);
     }
+#if wxUSE_STD_IOSTREAM
+    if ( !stream )
+#else
+    int res = istream.GetLastError();
+    if ( res != wxSTREAM_NO_ERROR && res != wxSTREAM_EOF )
+#endif
+    {
+        return false;
+    }
 
-    return istream;
+    return true;
 }
 
 void DrawingDocument::DoUpdate()
Index: samples/docview/doc.h
===================================================================
--- samples/docview/doc.h	(revision 58554)
+++ samples/docview/doc.h	(working copy)
@@ -80,8 +80,8 @@
 public:
     DrawingDocument() : wxDocument() { }
 
-    DocumentOstream& SaveObject(DocumentOstream& stream);
-    DocumentIstream& LoadObject(DocumentIstream& stream);
+    bool SaveObject(DocumentOstream& stream);
+    bool LoadObject(DocumentIstream& stream);
 
     // add a new segment to the document
     void AddDoodleSegment(const DoodleSegment& segment);
