Index: src/osx/carbon/nonownedwnd.cpp
===================================================================
--- src/osx/carbon/nonownedwnd.cpp	(Revision 73362)
+++ src/osx/carbon/nonownedwnd.cpp	(Arbeitskopie)
@@ -68,9 +68,9 @@
 
 bool wxNonOwnedWindowCarbonImpl::Show(bool show)
 {
+#if wxUSE_SYSTEM_OPTIONS
     bool plainTransition = true;
 
-#if wxUSE_SYSTEM_OPTIONS
     if ( wxSystemOptions::HasOption(wxMAC_WINDOW_PLAIN_TRANSITION) )
         plainTransition = ( wxSystemOptions::GetOptionInt( wxMAC_WINDOW_PLAIN_TRANSITION ) == 1 ) ;
 #endif
