 src/common/datavcmn.cpp | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/src/common/datavcmn.cpp b/src/common/datavcmn.cpp
index 53252c0..2eef88d 100644
--- a/src/common/datavcmn.cpp
+++ b/src/common/datavcmn.cpp
@@ -1710,7 +1710,7 @@ wxDataViewDateRenderer::CreateEditorCtrl(wxWindow *parent, wxRect labelRect, con
                (
                    parent,
                    wxID_ANY,
-                   value.GetDateTime(),
+                   value.IsNull() ? wxInvalidDateTime : value.GetDateTime(),
                    labelRect.GetTopLeft(),
                    labelRect.GetSize()
                );
@@ -1725,7 +1725,7 @@ bool wxDataViewDateRenderer::GetValueFromEditorCtrl(wxWindow *editor, wxVariant&
 
 bool wxDataViewDateRenderer::SetValue(const wxVariant& value)
 {
-    m_date = value.GetDateTime();
+    m_date = value.IsNull() ? wxInvalidDateTime : value.GetDateTime();
     return true;
 }
 
@@ -1737,14 +1737,16 @@ bool wxDataViewDateRenderer::GetValue(wxVariant& value) const
 
 bool wxDataViewDateRenderer::Render(wxRect cell, wxDC* dc, int state)
 {
-    wxString tmp = m_date.FormatDate();
+    wxString tmp;
+    if (m_date.IsValid())
+        tmp = m_date.FormatDate();
     RenderText( tmp, 0, cell, dc, state );
     return true;
 }
 
 wxSize wxDataViewDateRenderer::GetSize() const
 {
-    return GetTextExtent(m_date.FormatDate());
+    return GetTextExtent(m_date.IsValid() ? m_date.FormatDate() : wxDateTime::Now().FormatDate());
 }
 
 #endif // (defined(wxHAS_GENERIC_DATAVIEWCTRL) || defined(__WXGTK__)) && wxUSE_DATEPICKCTRL
