--- docmdi.cpp	Fri Sep  8 20:09:34 2000
+++ ./src/common/docmdi.cpp	Sat Jul  6 11:19:04 2002
@@ -106,14 +106,26 @@
 bool wxDocMDIChildFrame::ProcessEvent(wxEvent& event)
 {
+    static wxEvent *ActiveEvent = NULL;
+
+    // Break recursion loops
+    if (ActiveEvent == &event)
+        return false;
+
+    ActiveEvent = &event;
+
+    bool ret;
 	if ( !m_childView || ! m_childView->ProcessEvent(event) )
     {
         // Only hand up to the parent if it's a menu command
         if (!event.IsKindOf(CLASSINFO(wxCommandEvent)) || !GetParent() || !GetParent()->ProcessEvent(event))
-		    return wxEvtHandler::ProcessEvent(event);
+            ret = wxEvtHandler::ProcessEvent(event);
         else
-            return TRUE;
+            ret = TRUE;
     }
 	else
-		return TRUE;
+        ret = TRUE;
+
+    ActiveEvent = NULL;
+    return ret;
 }
 

 	  	 
