Index: include/wx/wxchar.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/wxchar.h,v
retrieving revision 1.182
diff -b -u -2 -r1.182 wxchar.h
--- include/wx/wxchar.h	2005/10/07 21:33:00	1.182
+++ include/wx/wxchar.h	2005/12/05 02:47:07
@@ -142,5 +142,6 @@
 #endif /* compilers with (good) TCHAR support */
 
-#ifdef __MWERKS__
+#if defined(__MWERKS__) && \
+            !(defined(__DARWIN__) && ( MAC_OS_X_VERSION_MAX_ALLOWED <= MAC_OS_X_VERSION_10_2 ))
     #define HAVE_WPRINTF
 #endif
@@ -391,5 +392,5 @@
 #else /* !TCHAR-aware compilers */
 
-    #if !defined(__MWERKS__) && defined(__DARWIN__) && ( MAC_OS_X_VERSION_MAX_ALLOWED <= MAC_OS_X_VERSION_10_2 )
+    #if defined(__DARWIN__) && ( MAC_OS_X_VERSION_MAX_ALLOWED <= MAC_OS_X_VERSION_10_2 )
         /* even though they are defined and "implemented", they are bad and just
            stubs so we need our own - we need these even in ANSI builds!! */
@@ -402,5 +403,5 @@
 
     /* No UNICODE in the c library except wchar_t typedef on mac OSX 10.2 and less - roll our own */
-    #if !defined(__MWERKS__) && wxUSE_UNICODE && defined(__DARWIN__) && ( MAC_OS_X_VERSION_MAX_ALLOWED <= MAC_OS_X_VERSION_10_2 )
+    #if wxUSE_UNICODE && defined(__DARWIN__) && ( MAC_OS_X_VERSION_MAX_ALLOWED <= MAC_OS_X_VERSION_10_2 )
 
         /* we need everything! */
@@ -820,5 +821,7 @@
 #ifndef wxVsnprintf_
     #if wxUSE_UNICODE
-        #if defined(__MWERKS__)
+        /* 10.2 doesn't have mbsrtowcs or vswprintf et al. */ 
+        #if defined(__MWERKS__) && \
+            !(defined(__DARWIN__) && ( MAC_OS_X_VERSION_MAX_ALLOWED <= MAC_OS_X_VERSION_10_2 ))
             #define HAVE_WCSRTOMBS 1
             #define HAVE_VSWPRINTF 1

 	  	 
