Index: include/wx/docview.h
===================================================================
--- include/wx/docview.h	(revision 64576)
+++ include/wx/docview.h	(working copy)
@@ -172,6 +172,9 @@
     // dialogs. Override if necessary.
     virtual wxWindow *GetDocumentWindow() const;
 
+    virtual bool ShowDialog_Revert() const;
+    virtual int  ShowDialog_SaveModified() const;
+
 protected:
     wxList                m_documentViews;
     wxString              m_documentFile;
@@ -507,6 +510,9 @@
     // activate the first view of the given document if any
     void ActivateDocument(wxDocument *doc);
 
+    virtual bool ShowDialog_RemoveRecent(const wxString& path) const;
+    virtual void ShowDialog_OpenError() const;
+    virtual void ShowDialog_FormatError() const;
 
     int               m_defaultDocumentNameCounter;
     int               m_maxDocsOpen;
Index: src/common/docview.cpp
===================================================================
--- src/common/docview.cpp	(revision 64576)
+++ src/common/docview.cpp	(working copy)
@@ -402,13 +402,7 @@
 
 bool wxDocument::Revert()
 {
-    if ( wxMessageBox
-         (
-            _("Discard changes and reload the last saved version?"),
-            wxTheApp->GetAppDisplayName(),
-            wxYES_NO | wxCANCEL | wxICON_QUESTION,
-            GetDocumentWindow()
-          ) != wxYES )
+    if (!ShowDialog_Revert())
         return false;
 
     if ( !DoOpenDocument(GetFilename()) )
@@ -476,16 +470,7 @@
 {
     if ( IsModified() )
     {
-        switch ( wxMessageBox
-                 (
-                    wxString::Format
-                    (
-                     _("Do you want to save changes to %s?"),
-                     GetUserReadableName()
-                    ),
-                    wxTheApp->GetAppDisplayName(),
-                    wxYES_NO | wxCANCEL | wxICON_QUESTION | wxCENTRE
-                 ) )
+        switch (ShowDialog_SaveModified())
         {
             case wxNO:
                 Modify(false);
@@ -1106,33 +1091,16 @@
 void wxDocManager::DoOpenMRUFile(unsigned n)
 {
     wxString filename(GetHistoryFile(n));
-    if ( filename.empty() )
+    wxASSERT(filename.Length());
+    
+    if ( CreateDocument(filename, wxDOC_SILENT) )
         return;
 
-    wxString errMsg; // must contain exactly one "%s" if non-empty
-    if ( wxFile::Exists(filename) )
+    if (ShowDialog_RemoveRecent(filename))
     {
-        // try to open it
-        if ( CreateDocument(filename, wxDOC_SILENT) )
-            return;
-
-        errMsg = _("The file '%s' couldn't be opened.");
+        // remove the file which we can't open from the MRU list
+        RemoveFileFromHistory(n);
     }
-    else // file doesn't exist
-    {
-        errMsg = _("The file '%s' doesn't exist and couldn't be opened.");
-    }
-
-
-    wxASSERT_MSG( !errMsg.empty(), "should have an error message" );
-
-    // remove the file which we can't open from the MRU list
-    RemoveFileFromHistory(n);
-
-    // and tell the user about it
-    wxLogError(errMsg + '\n' +
-               _("It has been removed from the most recently used files list."),
-               filename);
 }
 
 #if wxUSE_PRINTING_ARCHITECTURE
@@ -1691,18 +1659,9 @@
     wxDocTemplate *theTemplate = NULL;
     if (!pathTmp.empty())
     {
-        if (!wxFileExists(pathTmp))
+        if (!wxFile::Exists(pathTmp))
         {
-            wxString msgTitle;
-            if (!wxTheApp->GetAppDisplayName().empty())
-                msgTitle = wxTheApp->GetAppDisplayName();
-            else
-                msgTitle = wxString(_("File error"));
-
-            wxMessageBox(_("Sorry, could not open this file."),
-                         msgTitle,
-                         wxOK | wxICON_EXCLAMATION | wxCENTRE);
-
+            ShowDialog_OpenError();
             path = wxEmptyString;
             return NULL;
         }
@@ -1722,9 +1681,7 @@
             // Since we do not add files with non-default extensions to the
             // file history this can only happen if the application changes the
             // allowed templates in runtime.
-            wxMessageBox(_("Sorry, the format for this file is unknown."),
-                         _("Open File"),
-                         wxOK | wxICON_EXCLAMATION | wxCENTRE);
+            ShowDialog_FormatError();
         }
     }
     else
@@ -1930,6 +1887,65 @@
 }
 
 // ----------------------------------------------------------------------------
+// Dialogs
+// ----------------------------------------------------------------------------
+
+int wxDocument::ShowDialog_SaveModified() const
+{
+    return wxMessageBox
+    (
+        wxString::Format
+        (
+          _("Do you want to save changes to %s?"),
+          GetUserReadableName()
+        ),
+        wxTheApp->GetAppDisplayName(),
+        wxYES_NO | wxCANCEL | wxICON_QUESTION | wxCENTRE
+    );
+}
+
+bool wxDocument::ShowDialog_Revert() const
+{
+    return ( wxYES == wxMessageBox
+    (
+        _("Discard changes and reload the last saved version?"),
+        wxTheApp->GetAppDisplayName(),
+        wxYES_NO | wxCANCEL | wxICON_QUESTION,
+        GetDocumentWindow()
+    ) );
+}
+
+bool wxDocManager::ShowDialog_RemoveRecent(const wxString& path) const
+{
+    return ( wxYES == wxMessageBox
+    (
+        wxString::Format(_("The file '%s' cannot be opened.\nDo you want to remove the reference from the Recent list?"),
+            path.wx_str()),
+        wxTheApp->GetAppDisplayName(),
+        wxYES_NO | wxICON_QUESTION
+    ) );
+}
+
+void wxDocManager::ShowDialog_OpenError() const
+{
+    wxString msgTitle;
+    if (!wxTheApp->GetAppDisplayName().empty())
+        msgTitle = wxTheApp->GetAppDisplayName();
+    else
+        msgTitle = wxString(_("File error"));
+    wxMessageBox(_("Sorry, could not open this file."),
+                msgTitle,
+                wxOK | wxICON_EXCLAMATION | wxCENTRE);
+}
+
+void wxDocManager::ShowDialog_FormatError() const
+{
+    wxMessageBox(_("Sorry, the format for this file is unknown."),
+             _("Open File"),
+             wxOK | wxICON_EXCLAMATION | wxCENTRE);
+}
+
+// ----------------------------------------------------------------------------
 // wxDocChildFrameAnyBase
 // ----------------------------------------------------------------------------
 
