Index: src/osx/menu_osx.cpp
===================================================================
--- src/osx/menu_osx.cpp	(revision 65218)
+++ src/osx/menu_osx.cpp	(working copy)
@@ -518,7 +518,7 @@
     m_appleMenu->Append( wxApp::s_macAboutMenuItemId, "About..." );
     m_appleMenu->AppendSeparator();
 #if !wxOSX_USE_CARBON
-    m_appleMenu->Append( wxApp::s_macPreferencesMenuItemId, "Preferences..." );
+    m_appleMenu->Append( wxApp::s_macPreferencesMenuItemId, "Preferences...\tCtrl+," );
     m_appleMenu->AppendSeparator();
     m_appleMenu->Append( wxApp::s_macExitMenuItemId, "Quit\tCtrl+Q" );
 #endif
