Index: src/msw/glcanvas.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/glcanvas.cpp,v
retrieving revision 1.18
diff -b -u -2 -r1.18 glcanvas.cpp
--- src/msw/glcanvas.cpp	2003/05/06 21:57:32	1.18
+++ src/msw/glcanvas.cpp	2003/06/16 20:20:59
@@ -33,4 +33,23 @@
 #include "wx/glcanvas.h"
 
+#ifdef _MSC_VER
+/*	
+*	The following two compiler directives are specific to the Microsoft Visual C++ family of compilers
+*	
+*	Fundementally what they do is instruct the linker to use these two libraries for the resolution
+*	of symbols. In essence, this is the equivalent of adding these two libraries to either the Makefile
+*	or project file.
+*	This is NOT a recommended technique, and certainly is unlikely to be used anywhere else in
+*	wxWindows given it is so specific to not only wxMSW, but also the VC compiler.
+*	However, in the case of opengl support, it's an applicable technique as opengl is optional in setup.h
+*	This code ( wrapped by wxUSE_GLCANVAS ), now allows opengl support to be added purely by
+*	modifying setup.h rather than by having to modify either the project or DSP fle.
+*
+*	See MSDN for further information on the exact usage of these commands.
+*/
+#	pragma comment( lib, "opengl32" )
+#	pragma comment( lib, "glu32" )
+#endif
+
 static const wxChar *wxGLCanvasClassName = wxT("wxGLCanvasClass");
 static const wxChar *wxGLCanvasClassNameNoRedraw = wxT("wxGLCanvasClassNR");

 	  	 
