Index: src/msw/mediactrl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/msw/mediactrl.cpp,v
retrieving revision 1.38
diff -b -u -2 -r1.38 mediactrl.cpp
--- src/msw/mediactrl.cpp	2005/05/31 21:12:17	1.38
+++ src/msw/mediactrl.cpp	2005/06/01 18:37:12
@@ -579,4 +579,5 @@
     IMediaEvent* m_pME;
     IMediaPosition* m_pMS;
+    bool m_bVideo;
     
     wxAMMediaThread* m_pThread;
@@ -1239,4 +1240,9 @@
     }
 
+    if(m_bestSize.x == 0 && m_bestSize.y == 0)
+        m_bVideo = false;
+    else
+        m_bVideo = true;
+
     //
     // Force the parent window of this control to recalculate
@@ -1492,5 +1498,5 @@
                             int w, int h)
 {
-    if(m_pVMC)
+    if(m_pVMC && m_bVideo)
     {
         RECT srcRect, destRect;
@@ -1633,5 +1639,5 @@
 {
     wxAMMediaBackend* pThis = (wxAMMediaBackend*) this;
-    if(pThis->m_pVMC)
+    if(pThis->m_pVMC && pThis->m_bVideo)
     {
         //TODO: Use wxClientDC?
@@ -1659,5 +1665,5 @@
     wxAMMediaBackend* pThis = (wxAMMediaBackend*) this;
     wxPaintDC dc(pThis->m_ctrl);
-    if( pThis->m_pVMC )
+    if( pThis->m_pVMC && pThis->m_bVideo)
     {
         if( pThis->m_pVMC->RepaintVideo((HWND)pThis->m_ctrl->GetHandle(), 

 	  	 
