Index: scrlwing.cpp
===================================================================
--- scrlwing.cpp	(revision 72603)
+++ scrlwing.cpp	(working copy)
@@ -544,19 +544,6 @@
         }
     }
 
-    if ( !needsRefresh )
-    {
-        // flush all pending repaints before we change m_{x,y}ScrollPosition, as
-        // otherwise invalidated area could be updated incorrectly later when
-        // ScrollWindow() makes sure they're repainted before scrolling them
-#ifdef __WXMAC__
-        // wxWindowMac is taking care of making sure the update area is correctly
-        // set up, while not forcing an immediate redraw
-#else
-        m_targetWindow->Update();
-#endif
-    }
-
     if (orient == wxHORIZONTAL)
     {
         m_xScrollPosition += nScrollInc;
@@ -575,7 +562,17 @@
     else
     {
         m_targetWindow->ScrollWindow(dx, dy, GetScrollRect());
-    }
+
+        // flush all pending repaints before we change m_{x,y}ScrollPosition, as
+        // otherwise invalidated area could be updated incorrectly later when
+        // ScrollWindow() makes sure they're repainted before scrolling them
+#ifdef __WXMAC__
+        // wxWindowMac is taking care of making sure the update area is correctly
+        // set up, while not forcing an immediate redraw
+#else
+        m_targetWindow->Update();
+#endif
+   }
 }
 
 int wxScrollHelperBase::CalcScrollInc(wxScrollWinEvent& event)
