--- dataview_orig.cpp	2016-02-28 22:33:38.000000000 +0100
+++ dataview.cpp	2016-03-10 16:47:32.832550800 +0100
@@ -703,7 +703,7 @@
             wxASSERT(!m_ctrl[2]);
             wxDataViewListCtrl* lc =
                 new wxDataViewListCtrl( parent, wxID_ANY, wxDefaultPosition,
-                                        wxDefaultSize, style );
+                                        wxSize(1000, 800), style );
             m_ctrl[2] = lc;
 
             MyListStoreDerivedModel* page2_model = new MyListStoreDerivedModel();
