Index: radiobox.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/os2/radiobox.cpp,v
retrieving revision 1.36
diff -u -r1.36 radiobox.cpp
--- radiobox.cpp	2005/12/27 11:20:46	1.36
+++ radiobox.cpp	2006/01/02 19:24:40
@@ -197,7 +197,7 @@
         // radiobox part not belonging to any radiobutton)
         //
         ::WinSetWindowPos( (HWND)m_ahRadioButtons[i]
-                          ,HWND_TOP
+                          ,HWND_BOTTOM
                           ,(LONG)nXOffset
                           ,(LONG)nYOffset
                           ,(LONG)nWidthBtn
@@ -334,7 +334,7 @@
         long                        lStyleBtn = BS_AUTORADIOBUTTON | WS_TABSTOP | WS_VISIBLE;
         int                         nNewId = NewControlId();
 
-        if (i == 0 && lStyle == 0)
+        if (i == 0)
             lStyleBtn |= WS_GROUP;
 
         HWND                        hWndBtn = (WXHWND)::WinCreateWindow ( GetHwndOf(pParent)
@@ -343,7 +343,7 @@
                                                                          ,lStyleBtn
                                                                          ,0, 0, 0, 0
                                                                          ,GetWinHwnd(pParent)
-                                                                         ,HWND_TOP
+                                                                         ,HWND_BOTTOM
                                                                          ,(HMENU)nNewId
                                                                          ,NULL
                                                                          ,NULL
@@ -376,12 +376,12 @@
     }
 
     //
-    // Create a dummy radio control to end the group.
+    // Create a dummy control to end the group.
     //
     (void)::WinCreateWindow ( GetHwndOf(pParent),
                               WC_BUTTON,
                               "",
-                              WS_GROUP | BS_AUTORADIOBUTTON,
+                              WS_GROUP,
                               0, 0, 0, 0,
                               GetWinHwnd(pParent),
                               HWND_TOP,
@@ -400,23 +400,13 @@
                       ,sizeof(LONG)
                       ,(PVOID)&lColor
                      );
-    ::WinSetPresParam( m_hWnd
-                      ,PP_BORDERDARKCOLOR
-                      ,sizeof(LONG)
-                      ,(PVOID)&lColor
-                     );
-    lColor = (LONG)m_backgroundColour.GetPixel();
 
+    lColor = (LONG)m_backgroundColour.GetPixel();
     ::WinSetPresParam( m_hWnd
                       ,PP_BACKGROUNDCOLOR
                       ,sizeof(LONG)
                       ,(PVOID)&lColor
                      );
-    ::WinSetPresParam( m_hWnd
-                      ,PP_BORDERLIGHTCOLOR
-                      ,sizeof(LONG)
-                      ,(PVOID)&lColor
-                     );
     SetXComp(0);
     SetYComp(0);
     SetSelection(0);
@@ -447,13 +437,8 @@
     int                             nHeightOld;
     int                             nXx = nX;
     int                             nYy = nY;
-#if RADIOBTN_PARENT_IS_RADIOBOX
-    int                             nXOffset = 0;
-    int                             nYOffset = 0;
-#else
     int                             nXOffset = nXx;
     int                             nYOffset = nYy;
-#endif
     int                             nCx1;
     int                             nCy1;
     wxSize                          vMaxSize = GetMaxButtonSize();
@@ -639,7 +624,7 @@
         // radiobox part not belonging to any radiobutton)
         //
         ::WinSetWindowPos( (HWND)m_ahRadioButtons[i]
-                          ,HWND_TOP
+                          ,HWND_BOTTOM
                           ,(LONG)nXOffset
                           ,(LONG)nYOffset
                           ,(LONG)nWidthBtn
@@ -784,8 +769,8 @@
     }
     if (pnX)
         *pnX = vPoint.x;
-    if (pnY)
-        *pnY = vPoint.y;
+    if (pnX)
+        *pnX = vPoint.y;
 } // end of wxRadioBox::GetPosition
 
 // Get single selection, for single choice list items
@@ -1096,11 +1081,11 @@
                         switch(uVk)
                         {
                             case VK_LEFT:
-                                eDir = wxDOWN;
+                                eDir = wxLEFT;
                                 break;
 
                             case VK_RIGHT:
-                                eDir = wxDOWN;
+                                eDir = wxRIGHT;
                                 break;
 
                             case VK_DOWN:

 	  	 
