diff -bru wxWidgets.current/samples/dialogs/dialogs.cpp /mnt/winxp/wxWidgets.current/samples/dialogs/dialogs.cpp
--- wxWidgets.current/samples/dialogs/dialogs.cpp	2009-12-13 11:59:54.000000000 -0800
+++ /mnt/winxp/wxWidgets.current/samples/dialogs/dialogs.cpp	2009-12-27 20:02:33.843750000 -0800
@@ -119,6 +119,72 @@
 END_EVENT_TABLE()
 
 
+class CSelectTableForBug : public wxDialog
+{
+public:
+	CSelectTableForBug(wxWindow* parent, int id, const wxString& title, const wxPoint& pos = wxDefaultPosition, const wxSize& size = wxDefaultSize, long style = 0);
+private:
+	wxPanel *panel_1;
+	wxListBox *m_tablesList;
+	wxCheckBox *m_systemTables;
+	wxButton *button_1, *button_2, *button_3, *button_4;
+};
+                    
+CSelectTableForBug::CSelectTableForBug(wxWindow* parent, int id, const wxString& title, const wxPoint& pos, const wxSize& size, long style):
+    wxDialog(parent, id, title, pos, size, wxDEFAULT_DIALOG_STYLE)
+{
+	style = style;
+	// begin wxGlade: CSelectTable::CSelectTable
+	panel_1 = new wxPanel( this, wxID_ANY );
+	m_tablesList = new wxListBox( panel_1, wxID_ANY, wxDefaultPosition, wxDefaultSize, 0, NULL, wxLB_MULTIPLE | wxLB_SORT );
+	m_systemTables = new wxCheckBox( panel_1, wxID_ANY, _T( "&Show system tables" ) );
+	button_1 = new wxButton( panel_1, wxID_OK, _T( "&Open" ) );
+	button_2 = new wxButton( panel_1, wxID_ANY, _T("&New..." ) );
+	button_3 = new wxButton( panel_1, wxID_CANCEL, _T("&Cancel" ) );
+	button_4 = new wxButton( panel_1, wxID_ANY, _T( "&Help" ) );
+	SetTitle(wxT("Select Table"));
+	button_1->SetDefault();
+	button_1->Enable( false );
+	for( int i = 0; i < 10; i++ )
+	{
+		wxString test;
+		test.Printf( "Test string %d", i );
+		m_tablesList->Append( test );
+	}
+	wxBoxSizer* sizer_1 = new wxBoxSizer(wxHORIZONTAL);
+	wxBoxSizer* sizer_2 = new wxBoxSizer(wxHORIZONTAL);
+	wxBoxSizer* sizer_3 = new wxBoxSizer(wxVERTICAL);
+	wxBoxSizer* sizer_4 = new wxBoxSizer(wxHORIZONTAL);
+	wxBoxSizer* sizer_6 = new wxBoxSizer(wxVERTICAL);
+	wxBoxSizer* sizer_5 = new wxBoxSizer(wxVERTICAL);
+	sizer_2->Add(5, 20, 0, wxEXPAND, 0);
+	sizer_3->Add(20, 5, 0, wxEXPAND, 0);
+	sizer_5->Add(m_tablesList, 0, wxEXPAND, 0);
+	sizer_5->Add(20, 5, 0, wxEXPAND, 0);
+	sizer_5->Add(m_systemTables, 0, wxALIGN_BOTTOM, 0);
+	sizer_4->Add(sizer_5, 0, wxEXPAND, 0);
+	sizer_4->Add(5, 20, 0, wxEXPAND | wxALIGN_BOTTOM | wxFIXED_MINSIZE, 0);
+	sizer_6->Add(button_1, 0, 0, 0);
+	sizer_6->Add(20, 5, 0, wxEXPAND, 0);
+	sizer_6->Add(button_2, 0, 0, 0);
+	sizer_6->Add(20, 5, 0, wxEXPAND, 0);
+	sizer_6->Add(button_3, 0, 0, 0);
+	sizer_6->Add(20, 5, 0, wxEXPAND, 0);
+	sizer_6->Add(button_4, 0, 0, 0);
+	sizer_4->Add(sizer_6, 0, wxEXPAND, 0);
+	sizer_3->Add(sizer_4, 0, wxEXPAND, 0);
+	sizer_3->Add(20, 5, 0, wxEXPAND, 0);
+	sizer_2->Add(sizer_3, 0, wxEXPAND, 0);
+	sizer_2->Add(5, 20, 0, wxEXPAND, 0);
+	panel_1->SetSizer(sizer_2);
+	sizer_1->Add(panel_1, 1, wxEXPAND, 0);
+	SetSizer(sizer_1);
+	sizer_1->Fit(this);
+	Layout();
+	Centre();
+	// end wxGlade
+}
+
 
 BEGIN_EVENT_TABLE(MyFrame, wxFrame)
 #if wxUSE_MSGDLG
@@ -786,7 +852,9 @@
 
 void MyFrame::MessageBoxDialog(wxCommandEvent& WXUNUSED(event))
 {
-    TestMessageBoxDialog dlg(this);
+//    TestMessageBoxDialog dlg(this);
+//    dlg.ShowModal();
+	CSelectTableForBug dlg( this, -1, _T( "Test" ) );
     dlg.ShowModal();
 }
 
Only in /mnt/winxp/wxWidgets.current/samples/dialogs: dialogs.dsw
Only in /mnt/winxp/wxWidgets.current/samples/dialogs: dialogs.ncb
Only in /mnt/winxp/wxWidgets.current/samples/dialogs: dialogs.opt
Only in /mnt/winxp/wxWidgets.current/samples/dialogs: dialogs.plg
Only in /mnt/winxp/wxWidgets.current/samples/dialogs: vc_mswud
