Index: docs/latex/wx/stdevtid.tex
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/docs/latex/wx/stdevtid.tex,v
retrieving revision 1.7
diff -bu2 -r1.7 docs/latex/wx/stdevtid.tex
--- docs/latex/wx/stdevtid.tex
+++ docs/latex/wx/stdevtid.tex
@@ -48,4 +48,7 @@
     wxID_CLOSE_ALL,
 
+    wxID_EDIT,
+    wxID_FILE,
+
     wxID_CUT = 5030,
     wxID_COPY,
Index: docs/latex/wx/stockitems.tex
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/docs/latex/wx/stockitems.tex,v
retrieving revision 1.3
diff -bu2 -r1.3 docs/latex/wx/stockitems.tex
--- docs/latex/wx/stockitems.tex
+++ docs/latex/wx/stockitems.tex
@@ -5,4 +5,5 @@
 
 \begin{twocollist}\itemsep=0pt
+\twocolitem{wxID\_ABOUT}{"\&About"}
 \twocolitem{wxID\_ADD}{"Add"}
 \twocolitem{wxID\_APPLY}{"\&Apply"}
@@ -14,5 +15,7 @@
 \twocolitem{wxID\_CUT}{"Cu\&t"}
 \twocolitem{wxID\_DELETE}{"\&Delete"}
+\twocolitem{wxID\_EDIT}{"\&Edit"}
 \twocolitem{wxID\_FIND}{"\&Find"}
+\twocolitem{wxID\_FILE}{"\&File"}
 \twocolitem{wxID\_REPLACE}{"Find and rep\&lace"}
 \twocolitem{wxID\_BACKWARD}{"\&Back"}
@@ -45,4 +48,5 @@
 \twocolitem{wxID\_SAVE}{"\&Save"}
 \twocolitem{wxID\_SAVEAS}{"Save \&As..."}
+\twocolitem{wxID\_SELECTALL}{"Select all"}
 \twocolitem{wxID\_STOP}{"\&Stop"}
 \twocolitem{wxID\_UNDELETE}{"Undelete"}
Index: src/common/stockitem.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/common/stockitem.cpp,v
retrieving revision 1.16
diff -bu2 -r1.16 src/common/stockitem.cpp
--- src/common/stockitem.cpp
+++ src/common/stockitem.cpp
@@ -36,4 +36,5 @@
     switch (id)
     {
+        case wxID_ABOUT:
         case wxID_ADD:
         case wxID_APPLY:
@@ -45,5 +46,7 @@
         case wxID_CUT:
         case wxID_DELETE:
+        case wxID_EDIT:
         case wxID_FIND:
+        case wxID_FILE:
         case wxID_REPLACE:
         case wxID_BACKWARD:
@@ -76,4 +79,5 @@
         case wxID_SAVE:
         case wxID_SAVEAS:
+        case wxID_SELECTALL:
         case wxID_STOP:
         case wxID_UNDELETE:
@@ -104,4 +108,5 @@
     switch (id)
     {
+        STOCKITEM(wxID_ABOUT,               _("&About"))
         STOCKITEM(wxID_ADD,                 _("Add"))
         STOCKITEM(wxID_APPLY,               _("&Apply"))
@@ -113,5 +118,7 @@
         STOCKITEM(wxID_CUT,                 _("Cu&t"))
         STOCKITEM(wxID_DELETE,              _("&Delete"))
+        STOCKITEM(wxID_EDIT,                _("&Edit"))
         STOCKITEM(wxID_FIND,                _("&Find"))
+        STOCKITEM(wxID_FILE,                _("&File"))
         STOCKITEM(wxID_REPLACE,             _("Rep&lace"))
         STOCKITEM(wxID_BACKWARD,            _("&Back"))
@@ -144,4 +151,5 @@
         STOCKITEM(wxID_SAVE,                _("&Save"))
         STOCKITEM(wxID_SAVEAS,              _("Save &As..."))
+        STOCKITEM(wxID_SELECTALL,           _("Select all"))
         STOCKITEM(wxID_STOP,                _("&Stop"))
         STOCKITEM(wxID_UNDELETE,            _("Undelete"))
@@ -213,6 +221,20 @@
     #endif
 
+    #if GTK_CHECK_VERSION(2,6,0)
+        #define STOCKITEM_26(wx,gtk) STOCKITEM(wx,gtk)
+    #else
+        #define STOCKITEM_26(wx,gtk) STOCKITEM_MISSING(wx)
+    #endif
+
+    #if GTK_CHECK_VERSION(2,10,0)
+        #define STOCKITEM_210(wx,gtk) STOCKITEM(wx,gtk)
+    #else
+        #define STOCKITEM_210(wx,gtk) STOCKITEM_MISSING(wx)
+    #endif
+
+
     switch (id)
     {
+        STOCKITEM_26(wxID_ABOUT,         GTK_STOCK_ABOUT)
         STOCKITEM(wxID_ADD,              GTK_STOCK_ADD)
         STOCKITEM(wxID_APPLY,            GTK_STOCK_APPLY)
@@ -224,5 +246,7 @@
         STOCKITEM(wxID_CUT,              GTK_STOCK_CUT)
         STOCKITEM(wxID_DELETE,           GTK_STOCK_DELETE)
+        STOCKITEM_26(wxID_EDIT,          GTK_STOCK_EDIT)
         STOCKITEM(wxID_FIND,             GTK_STOCK_FIND)
+        STOCKITEM_26(wxID_FILE,          GTK_STOCK_FILE)
         STOCKITEM(wxID_REPLACE,          GTK_STOCK_FIND_AND_REPLACE)
         STOCKITEM(wxID_BACKWARD,         GTK_STOCK_GO_BACK)
@@ -255,4 +279,5 @@
         STOCKITEM(wxID_SAVE,             GTK_STOCK_SAVE)
         STOCKITEM(wxID_SAVEAS,           GTK_STOCK_SAVE_AS)
+        STOCKITEM_210(wxID_SELECTALL,    GTK_STOCK_SELECT_ALL)
         STOCKITEM(wxID_STOP,             GTK_STOCK_STOP)
         STOCKITEM(wxID_UNDELETE,         GTK_STOCK_UNDELETE)

 	  	 
