diff -ru2 -xCVS -x.#* -x*.orig -x*.rej -x*~ c:/wxWidgets-2.6.2_ori/src/generic/calctrl.cpp c:/wxWidgets-2.6.2_new/src/generic/calctrl.cpp
--- c:/wxWidgets-2.6.2_ori/src/generic/calctrl.cpp	Tue May 31 13:45:10 2005
+++ c:/wxWidgets-2.6.2_new/src/generic/calctrl.cpp	Mon Sep 26 07:49:18 2005
@@ -853,5 +853,5 @@
             // 1.5 times the width gives nice margins even if the weekday
             // names are short
-            m_widthCol = width+width/2;
+	     m_widthCol = width+width/2;
         }
     }
@@ -1446,6 +1446,11 @@
 {
     RecalcGeometry();
+    // use the correct x-pos 
+    wxCoord x0 = (GetSize().x - m_widthCol*7) /2;
+    if (x0 < 0) x0 = 0;
+    wxPoint pos_corr = pos;
+    pos_corr.x -= x0;
 
-    wxCoord y = pos.y;
+    wxCoord y = pos_corr.y;
 
 ///////////////////////////////////////////////////////////////////////////////////////////////////////
@@ -1456,5 +1461,5 @@
         // we need to find out if the hit is on left arrow, on month or on right arrow
         // left arrow?
-        if ( wxRegion(m_leftArrowRect).Contains(pos) == wxInRegion )
+        if ( wxRegion(m_leftArrowRect).Contains(pos_corr) == wxInRegion )
         {
             if ( date )
@@ -1473,5 +1478,5 @@
         }
 
-        if ( wxRegion(m_rightArrowRect).Contains(pos) == wxInRegion )
+        if ( wxRegion(m_rightArrowRect).Contains(pos_corr) == wxInRegion )
         {
             if ( date )
@@ -1494,5 +1499,5 @@
 ///////////////////////////////////////////////////////////////////////////////////////////////////////
     // Header: Days
-    int wday = pos.x / m_widthCol;
+    int wday = pos_corr.x / m_widthCol;
 //    if ( y < m_heightRow )
     if ( y < (m_heightRow + m_rowOffset) )

 	  	 
