diff -uNr wxWidgets-trunk\src/msw/statbmp.cpp wxWidgets-work\src/msw/statbmp.cpp
--- wxWidgets-trunk\src/msw/statbmp.cpp	2013-12-29 11:00:43.514795200 +0100
+++ wxWidgets-work\src/msw/statbmp.cpp	2014-01-21 21:34:58.506315400 +0100
@@ -37,6 +37,7 @@
 #include "wx/msw/private.h"
 
 #include "wx/sysopt.h"
+#include "wx/msw/dib.h"
 
 #include <stdio.h>
 
@@ -265,7 +266,29 @@
     GetSize(&w, &h);
 
 #ifdef __WIN32__
-    HANDLE handle = (HANDLE)m_image->GetHandle();
+    // wxBitmap normally stores alpha in pre-multiplied format but apparently
+    // STM_SETIMAGE message handler does pre-multiplication internally so we need to undo
+    // the pre-multiplication here for a while.
+    // (Similar case to ImageList_Add API function call in wxImageList::Add)
+    HANDLE handle;
+    AutoHBITMAP hbmpTmp; // To store HBITMAP of temporary not-premultiplied bitmap.
+    if( !m_isIcon )
+    {
+        const wxBitmap& bmp = (const wxBitmap&)*image;
+        if ( bmp.HasAlpha() )
+        {   // For bitmap with alpha channel create temporary DIB with not-premultiplied alpha values.
+            handle = wxDIB(bmp.ConvertToImage(), wxDIB::PixelFormat_NotPreMultiplied).Detach();
+            hbmpTmp.Init((HBITMAP)handle);
+        }
+        else
+        {
+            handle = (HANDLE)m_image->GetHandle();
+        }
+    }
+    else
+    {
+        handle = (HANDLE)m_image->GetHandle();
+    }
     LONG style = ::GetWindowLong( (HWND)GetHWND(), GWL_STYLE ) ;
     ::SetWindowLong( (HWND)GetHWND(), GWL_STYLE, ( style & ~( SS_BITMAP|SS_ICON ) ) |
                      ( m_isIcon ? SS_ICON : SS_BITMAP ) );
@@ -278,7 +301,10 @@
         // the static control made a copy and we are responsible for deleting it
         DeleteObject((HGDIOBJ) oldHandle);
     }
-    m_currentHandle = (WXHANDLE)handle;
+    // Save bitmap handle only if it's not a temporary one.
+    if( !hbmpTmp )
+        m_currentHandle = (WXHANDLE)handle;
+
 #endif // Win32
 
     if ( ImageIsOk() )
