Index: src/common/file.cpp
===================================================================
--- src/common/file.cpp (revision 55697)
+++ src/common/file.cpp (working copy)
@@ -393,6 +393,15 @@
 {
     wxASSERT( IsOpened() );

+#ifdef __UNIX__
+    wxFileOffset iRc = wxInvalidOffset;
+    struct stat st;
+    if ( fstat(m_fd, &st) == 0 ) {
+        // There are strange files in /sys that claim to be 4096B long, but have a zero block allocation
+        iRc = st.st_blocks ? st.st_size : 0;
+    }
+
+#else
     wxFileOffset iRc = Tell();
     if ( iRc != wxInvalidOffset ) {
         // have to use const_cast :-(
@@ -407,6 +416,7 @@

         iRc = iLen;
     }
+#endif

     if ( iRc == wxInvalidOffset )
     {