Index: include/wx/splitter.h
===================================================================
--- include/wx/splitter.h	(revision 55163)
+++ include/wx/splitter.h	(working copy)
@@ -18,7 +18,11 @@
 // wxSplitterWindow flags
 // ----------------------------------------------------------------------------
 
+// Using these flags would be simpler if the default flag in ctor was 0.
+// wxSP_THIN_SASH was introduced in 2.9 to make this change possible in future.
+
 #define wxSP_NOBORDER         0x0000
+#define wxSP_THIN_SASH        0x0000
 #define wxSP_NOSASH           0x0010
 #define wxSP_PERMIT_UNSPLIT   0x0040
 #define wxSP_LIVE_UPDATE      0x0080
Index: interface/wx/splitter.h
===================================================================
--- interface/wx/splitter.h	(revision 55163)
+++ interface/wx/splitter.h	(working copy)
@@ -20,6 +20,8 @@
            Draws a 3D effect border and sash.
     @style{wxSP_3DSASH}
            Draws a 3D effect sash.
+    @style{wxSP_THIN_SASH}
+           Draws a thin border.
     @style{wxSP_3DBORDER}
            Synonym for wxSP_BORDER.
     @style{wxSP_BORDER}
