? splitter.cpp
? splitterpatch.patch
Index: splitter/splitter.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/samples/splitter/splitter.cpp,v
retrieving revision 1.14
diff -b -u -2 -r1.14 splitter.cpp
--- splitter/splitter.cpp	2002/07/12 13:23:53	1.14
+++ splitter/splitter.cpp	2003/02/18 19:47:12
@@ -38,4 +38,5 @@
 
 #include "wx/splitter.h"
+#include "wx/dc.h"
 
 // ----------------------------------------------------------------------------

 	  	 
