Index: fontdlg.cpp
===================================================================
--- fontdlg.cpp	(revision 57104)
+++ fontdlg.cpp	(working copy)
@@ -142,7 +142,7 @@
 
 void wxFontDialog::SetChosenFont(const char *fontname)
 {
-    m_fontData.SetChosenFont(wxFont( wxString::FromAscii(fontname) ));
+    m_fontData.SetChosenFont(wxFont( wxString::FromUTF8(fontname) ));
 }
 
 #endif // wxUSE_FONTDLG && !__WXGPE__
