--- wxWidgets-trunk\samples\stc\stctest.cpp	2013-12-29 11:02:29.058982400 +0100
+++ wxWidgets-work\samples\stc\stctest.cpp	2014-01-07 18:28:47.956121900 +0100
@@ -299,7 +299,9 @@
     m_edit = new Edit (this, wxID_ANY);
     m_edit->SetFocus();
 
-    FileOpen (wxT("stctest.cpp"));
+    //FileOpen (wxT("stctest.cpp"));
+    m_edit->AddText(wxT("ABCD ¹æê³ñóœŸ¿ abcd"));
+    m_edit->InsertText(6, "-123-");
 }
 
 AppFrame::~AppFrame () {
