diff -ru wxWidgets.orig/include/wx/msw/bitmap.h wxWidgets/include/wx/msw/bitmap.h
--- wxWidgets.orig/include/wx/msw/bitmap.h	2007-05-02 15:23:52.000000000 +0200
+++ wxWidgets/include/wx/msw/bitmap.h	2007-06-16 01:02:56.000000000 +0200
@@ -106,7 +106,7 @@
     wxBitmap GetSubBitmap( const wxRect& rect ) const;
 
     // copies the contents and mask of the given (colour) icon to the bitmap
-    bool CopyFromIcon(const wxIcon& icon);
+    bool CopyFromIcon(const wxIcon& icon, bool want_alpha = true);
 
     // copies the contents and mask of the given cursor to the bitmap
     bool CopyFromCursor(const wxCursor& cursor);
@@ -182,7 +182,7 @@
 
 private:
     // common part of CopyFromIcon/CopyFromCursor for Win32
-    bool CopyFromIconOrCursor(const wxGDIImage& icon);
+    bool CopyFromIconOrCursor(const wxGDIImage& icon, bool want_alpha = true);
 
 
     DECLARE_DYNAMIC_CLASS(wxBitmap)
diff -ru wxWidgets.orig/src/msw/bitmap.cpp wxWidgets/src/msw/bitmap.cpp
--- wxWidgets.orig/src/msw/bitmap.cpp	2007-05-02 15:23:28.000000000 +0200
+++ wxWidgets/src/msw/bitmap.cpp	2007-06-16 01:01:56.000000000 +0200
@@ -291,7 +291,7 @@
 
 #ifdef __WIN32__
 
-bool wxBitmap::CopyFromIconOrCursor(const wxGDIImage& icon)
+bool wxBitmap::CopyFromIconOrCursor(const wxGDIImage& icon, bool want_alpha)
 {
 #if !defined(__WXMICROWIN__) && !defined(__WXWINCE__)
     // it may be either HICON or HCURSOR
@@ -322,8 +322,9 @@
     // although there are some icons that are 32 bpp but have no alpha... So
     // convert to a DIB and manually check the 4th byte for each pixel.
     BITMAP bm;
-    if ( ::GetObject(iconInfo.hbmColor, sizeof(BITMAP), (LPVOID)&bm)
-         && bm.bmBitsPixel == 32)
+    if (   ::GetObject(iconInfo.hbmColor, sizeof(BITMAP), (LPVOID)&bm)
+        && (bm.bmBitsPixel == 32)
+        && want_alpha)
     {
         wxDIB dib(iconInfo.hbmColor);
         if (dib.IsOk())
@@ -371,14 +372,14 @@
     return CopyFromIconOrCursor(cursor);
 }
 
-bool wxBitmap::CopyFromIcon(const wxIcon& icon)
+bool wxBitmap::CopyFromIcon(const wxIcon& icon, bool want_alpha)
 {
     UnRef();
 
     if ( !icon.Ok() )
         return false;
 
-    return CopyFromIconOrCursor(icon);
+    return CopyFromIconOrCursor(icon, want_alpha);
 }
 
 #ifndef NEVER_USE_DIB

 	  	 
