--- C:/Users/John/AppData/Local/Temp/automtn-revBASE.svn004.tmp.cpp	Sun Dec 20 14:37:18 2009
+++ C:/wxWidgets-2.9/src/msw/ole/automtn.cpp	Sun Dec 20 14:37:03 2009
@@ -199,14 +199,16 @@
 
     if (FAILED(hr))
     {
-        // display the exception information if appropriate:
+        if (hr == DISP_E_EXCEPTION) // The application needs to raise an exception. In this case, the structure passed in pExcepInfo should be filled in.
+        {
+          // display the exception information if appropriate:
 //        ShowException((const char*) member, hr, &excep, uiArgErr);
+          // free exception structure information
+            SysFreeString(excep.bstrSource);
+            SysFreeString(excep.bstrDescription);
+            SysFreeString(excep.bstrHelpFile);
+        }
 
-        // free exception structure information
-        SysFreeString(excep.bstrSource);
-        SysFreeString(excep.bstrDescription);
-        SysFreeString(excep.bstrHelpFile);
-
         if (vReturnPtr)
             ReleaseVariant(vReturnPtr);
         return false;
