Index: include/wx/mac/carbon/uma.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWidgets/include/wx/mac/carbon/uma.h,v
retrieving revision 1.6
diff -u -r1.6 uma.h
--- include/wx/mac/carbon/uma.h	2006/10/16 09:29:48	1.6
+++ include/wx/mac/carbon/uma.h	2006/11/21 21:09:43
@@ -32,6 +32,7 @@
 
 MenuRef         UMANewMenu( SInt16 id , const wxString& title , wxFontEncoding encoding) ;
 void             UMASetMenuTitle( MenuRef menu , const wxString& title , wxFontEncoding encoding) ;
+OSStatus         UMAGetMenuTitle( MenuRef menu , wxString& outTitle , wxFontEncoding encoding ) ;
 UInt32             UMAMenuEvent( EventRecord *inEvent ) ;
 void             UMAEnableMenuItem( MenuRef inMenu , MenuItemIndex item , bool enable ) ;
 
Index: src/mac/carbon/app.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWidgets/src/mac/carbon/app.cpp,v
retrieving revision 1.227
diff -u -r1.227 app.cpp
--- src/mac/carbon/app.cpp	2006/10/20 06:04:12	1.227
+++ src/mac/carbon/app.cpp	2006/11/21 21:09:45
@@ -84,7 +84,18 @@
     EVT_QUERY_END_SESSION(wxApp::OnQueryEndSession)
 END_EVENT_TABLE()
 
+wxString GetMacHelpMenuTitleName() ;
 
+wxString GetMacHelpMenuTitleName()
+{
+    wxString title = wxT("&Help") ;
+    MenuHandle mh = NULL ;
+    MenuItemIndex firstUserHelpMenuItem;
+    if ( UMAGetHelpMenu( &mh , &firstUserHelpMenuItem ) == noErr )
+        UMAGetMenuTitle( mh , title , wxFont::GetDefaultEncoding() ) ;
+    return title ;
+}
+
 // platform specific static variables
 static const short kwxMacAppleMenuId = 1 ;
 
@@ -94,7 +105,7 @@
 long      wxApp::s_macAboutMenuItemId = wxID_ABOUT ;
 long      wxApp::s_macPreferencesMenuItemId = wxID_PREFERENCES ;
 long      wxApp::s_macExitMenuItemId = wxID_EXIT ;
-wxString  wxApp::s_macHelpMenuTitleName = wxT("&Help") ;
+wxString  wxApp::s_macHelpMenuTitleName = GetMacHelpMenuTitleName() ;
 
 bool      wxApp::sm_isEmbedded = false; // Normally we're not a plugin
 
Index: src/mac/carbon/menu.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWidgets/src/mac/carbon/menu.cpp,v
retrieving revision 1.96
diff -u -r1.96 menu.cpp
--- src/mac/carbon/menu.cpp	2006/10/25 07:31:46	1.96
+++ src/mac/carbon/menu.cpp	2006/11/21 21:09:45
@@ -661,7 +661,7 @@
         wxMenuItem *item;
         wxMenu* menu = menuIter->GetData() , *subMenu = NULL ;
 
-        if ( m_titles[i] == wxT("?") || m_titles[i] == wxT("&?")  || m_titles[i] == wxApp::s_macHelpMenuTitleName )
+        if ( wxStripMenuCodes( m_titles[i] ) == wxT("?") || wxStripMenuCodes( m_titles[i] ) == wxApp::s_macHelpMenuTitleName )
         {
             for (node = menu->GetMenuItems().GetFirst(); node; node = node->GetNext())
             {
Index: src/mac/carbon/uma.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWidgets/src/mac/carbon/uma.cpp,v
retrieving revision 1.80
diff -u -r1.80 uma.cpp
--- src/mac/carbon/uma.cpp	2006/10/14 15:24:58	1.80
+++ src/mac/carbon/uma.cpp	2006/11/21 21:09:45
@@ -143,6 +143,23 @@
 #endif
 }
 
+OSStatus UMAGetMenuTitle( MenuRef menu , wxString& outTitle , wxFontEncoding encoding )
+{
+    OSErr err = noErr ;
+
+#if TARGET_CARBON
+    CFStringRef titleAsCFString ;
+    err = CopyMenuTitleAsCFString( menu, &titleAsCFString ) ;
+    outTitle = wxMacCFStringHolder( titleAsCFString ).AsString( encoding ) ;
+#else
+    Str255 ptitle ;
+    GetMenuTitle( menu , ptitle ) ;
+	outTitle = wxMacMakeStringFromPascal( ptitle ) ;
+#endif
+
+    return err ;
+}
+
 void UMASetMenuItemText( MenuRef menu,  MenuItemIndex item, const wxString& title, wxFontEncoding encoding )
 {
     // we don't strip the accels here anymore, must be done before

 	  	 
