Index: minimal.cpp
===================================================================
--- minimal.cpp	(revision 72024)
+++ minimal.cpp	(working copy)
@@ -67,10 +67,14 @@
     // event handlers (these functions should _not_ be virtual)
     void OnQuit(wxCommandEvent& event);
     void OnAbout(wxCommandEvent& event);
+    void OnItem(wxCommandEvent& event);
+    void OnButton(wxCommandEvent& event);
 
 private:
     // any class wishing to process wxWidgets events must use this macro
     DECLARE_EVENT_TABLE()
+
+    wxMenu* CreateTestMenu();
 };
 
 // ----------------------------------------------------------------------------
@@ -86,7 +90,10 @@
     // it is important for the id corresponding to the "About" command to have
     // this standard value as otherwise it won't be handled properly under Mac
     // (where it is special and put into the "Apple" menu)
-    Minimal_About = wxID_ABOUT
+    Minimal_About = wxID_ABOUT,
+
+    Minimal_Item,
+    Minimal_Button,
 };
 
 // ----------------------------------------------------------------------------
@@ -99,6 +106,8 @@
 BEGIN_EVENT_TABLE(MyFrame, wxFrame)
     EVT_MENU(Minimal_Quit,  MyFrame::OnQuit)
     EVT_MENU(Minimal_About, MyFrame::OnAbout)
+    EVT_MENU(Minimal_Item, MyFrame::OnItem)
+    EVT_BUTTON(Minimal_Button, MyFrame::OnButton)
 END_EVENT_TABLE()
 
 // Create a new application object: this macro will allow wxWidgets to create
@@ -160,6 +169,7 @@
 
     // now append the freshly created menu to the menu bar...
     wxMenuBar *menuBar = new wxMenuBar();
+    menuBar->Append(CreateTestMenu(), "&Test");
     menuBar->Append(fileMenu, "&File");
     menuBar->Append(helpMenu, "&Help");
 
@@ -172,6 +182,11 @@
     CreateStatusBar(2);
     SetStatusText("Welcome to wxWidgets!");
 #endif // wxUSE_STATUSBAR
+
+    wxButton* button = new wxButton(this, Minimal_Button, wxT("Recreate Test Menu"));
+    wxBoxSizer* mainSizer = new wxBoxSizer(wxVERTICAL);
+    mainSizer->Add(button, 1, wxEXPAND);
+    SetSizer(mainSizer);
 }
 
 
@@ -198,3 +213,25 @@
                  wxOK | wxICON_INFORMATION,
                  this);
 }
+
+void MyFrame::OnItem(wxCommandEvent& WXUNUSED(event))
+{
+    wxMessageBox(wxT("Test"));
+}
+
+void MyFrame::OnButton(wxCommandEvent& WXUNUSED(event))
+{
+    const int iMax = 3; // iMax >= 3 produces the segfault
+    for (int i = 0; i < iMax; i++) {
+        wxMenu* oldTestMenu = GetMenuBar()->Replace(0, CreateTestMenu(), wxT("&Test"));
+        delete oldTestMenu;
+    }
+}
+
+wxMenu* MyFrame::CreateTestMenu()
+{
+    wxMenu* testMenu = new wxMenu;
+    testMenu->Append(Minimal_Item, "&Item\tCtrl+I"); // without the accelerator, no segfault
+    return testMenu;
+}
+
