Index: docs/latex/wx/text.tex
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/docs/latex/wx/text.tex,v
retrieving revision 1.69
diff -u -B -r1.69 text.tex
--- docs/latex/wx/text.tex	2005/02/19 02:56:38	1.69
+++ docs/latex/wx/text.tex	2005/03/04 08:31:04
@@ -307,8 +307,8 @@
 \twocolitem{\windowstyle{wxTE\_CENTRE}}{The text in the control will be centered (currently wxMSW and wxGTK2 only).}
 \twocolitem{\windowstyle{wxTE\_RIGHT}}{The text in the control will be right-justified (currently wxMSW and wxGTK2 only).}
 \twocolitem{\windowstyle{wxTE\_DONTWRAP}}{Same as {\tt wxHSCROLL} style.}
-\twocolitem{\windowstyle{wxTE\_LINEWRAP}}{Wrap the lines too long to be shown entirely at any position (wxUniv only currently).}
-\twocolitem{\windowstyle{wxTE\_WORDWRAP}}{Wrap the lines too long to be shown entirely at word boundaries only (wxUniv only currently).}
+\twocolitem{\windowstyle{wxTE\_LINEWRAP}}{Wrap the lines too long to be shown entirely at any position (wxUniv and wxGTK2 only).}
+\twocolitem{\windowstyle{wxTE\_WORDWRAP}}{Wrap the lines too long to be shown entirely at word boundaries only (wxUniv and wxGTK2 only).}
 \end{twocollist}
 
 See also \helpref{window styles overview}{windowstyles} and
Index: include/wx/textctrl.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/textctrl.h,v
retrieving revision 1.67
diff -u -B -r1.67 textctrl.h
--- include/wx/textctrl.h	2005/02/19 02:55:28	1.67
+++ include/wx/textctrl.h	2005/03/04 08:31:04
@@ -117,7 +117,9 @@
 // position and wxTE_WORDWRAP to wrap at words boundary
 #define wxTE_DONTWRAP       wxHSCROLL
 #define wxTE_LINEWRAP       0x4000
-#define wxTE_WORDWRAP       0x0000  // it's just == !wxHSCROLL
+#define wxTE_WORDWRAP       0x0000  // it's just == !wxHSCROLL on wxUniv.
+// Without any wrap styles wxGTK2 wraps only at char position when word boundary
+// is not enough (no whitespace in the whole line), as should other platforms.
 
 // force using RichEdit version 2.0 or 3.0 instead of 1.0 (default) for
 // wxTE_RICH controls - can be used together with or instead of wxTE_RICH
Index: src/gtk/textctrl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/gtk/textctrl.cpp,v
retrieving revision 1.201
diff -u -B -r1.201 textctrl.cpp
--- src/gtk/textctrl.cpp	2005/02/19 06:10:55	1.201
+++ src/gtk/textctrl.cpp	2005/03/04 08:31:05
@@ -570,11 +570,15 @@
         // Insert view into scrolled window
         gtk_container_add( GTK_CONTAINER(m_widget), m_text );
 
-        // Global settings which can be overridden by tags, I guess.
+        // Global settings which can be overridden by tags.
         if (HasFlag( wxHSCROLL ) || HasFlag( wxTE_DONTWRAP ))
             gtk_text_view_set_wrap_mode( GTK_TEXT_VIEW( m_text ), GTK_WRAP_NONE );
-        else
+        else if (HasFlag( wxTE_WORDWRAP ))
             gtk_text_view_set_wrap_mode( GTK_TEXT_VIEW( m_text ), GTK_WRAP_WORD );
+        else if (HasFlag( wxTE_LINEWRAP ))
+            gtk_text_view_set_wrap_mode( GTK_TEXT_VIEW( m_text ), GTK_WRAP_CHAR );
+        else // break lines in between words, or if that is not enough, also between graphemes
+            gtk_text_view_set_wrap_mode( GTK_TEXT_VIEW( m_text ), GTK_WRAP_WORD_CHAR );
 
         if (!HasFlag(wxNO_BORDER))
             gtk_scrolled_window_set_shadow_type( GTK_SCROLLED_WINDOW(m_widget), GTK_SHADOW_IN );

 	  	 
