? cocoa_patch
? jwapatch.patch
Index: bitmap.mm
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/cocoa/bitmap.mm,v
retrieving revision 1.14
diff -u -d -b -w -b -u -2 -r1.14 bitmap.mm
--- bitmap.mm	2004/02/09 03:27:47	1.14
+++ bitmap.mm	2004/03/01 02:01:32
@@ -160,5 +160,5 @@
         // Show before/after to prove that the bitmap itself is not changed
         // even though we just composited onto the NSImage
-        wxLogTrace(wxTRACE_COCOA,"Before: bpp=%d",[M_BITMAPDATA->m_cocoaNSBitmapImageRep bitsPerPixel]);
+        wxLogTrace(wxTRACE_COCOA, wxT("Before: bpp=%d"),[M_BITMAPDATA->m_cocoaNSBitmapImageRep bitsPerPixel]);
         NSImage *maskImage = [[NSImage alloc]
                 initWithSize:NSMakeSize(GetWidth(), GetHeight())];
@@ -168,5 +168,5 @@
         [nsimage unlockFocus];
         [maskImage release];
-        wxLogTrace(wxTRACE_COCOA,"After: bpp=%d",[M_BITMAPDATA->m_cocoaNSBitmapImageRep bitsPerPixel]);
+        wxLogTrace(wxTRACE_COCOA, wxT("After: bpp=%d"),[M_BITMAPDATA->m_cocoaNSBitmapImageRep bitsPerPixel]);
     }
     return nsimage;
Index: toplevel.mm
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/cocoa/toplevel.mm,v
retrieving revision 1.28
diff -u -d -b -w -b -u -2 -r1.28 toplevel.mm
--- toplevel.mm	2004/02/27 06:07:47	1.28
+++ toplevel.mm	2004/03/01 02:01:32
@@ -346,5 +346,5 @@
 void wxTopLevelWindowCocoa::DoMoveWindow(int x, int y, int width, int height)
 {
-    wxLogTrace(wxTRACE_COCOA_TopLevelWindow_Size,"wxTopLevelWindow=%p::DoMoveWindow(%d,%d,%d,%d)",this,x,y,width,height);
+    wxLogTrace(wxTRACE_COCOA_TopLevelWindow_Size, wxT("wxTopLevelWindow=%p::DoMoveWindow(%d,%d,%d,%d)"),this,x,y,width,height);
 
     NSRect cocoaRect = NSMakeRect(x,y,width,height);
@@ -359,5 +359,5 @@
     if(h)
         *h=(int)cocoaRect.size.height;
-    wxLogTrace(wxTRACE_COCOA_TopLevelWindow_Size,"wxTopLevelWindow=%p::DoGetSize = (%d,%d)",this,(int)cocoaRect.size.width,(int)cocoaRect.size.height);
+    wxLogTrace(wxTRACE_COCOA_TopLevelWindow_Size, wxT("wxTopLevelWindow=%p::DoGetSize = (%d,%d)"),this,(int)cocoaRect.size.width,(int)cocoaRect.size.height);
 }
 
@@ -369,5 +369,5 @@
     if(y)
         *y=(int)cocoaRect.origin.y;
-    wxLogTrace(wxTRACE_COCOA_TopLevelWindow_Size,"wxTopLevelWindow=%p::DoGetPosition = (%d,%d)",this,(int)cocoaRect.origin.x,(int)cocoaRect.origin.y);
+    wxLogTrace(wxTRACE_COCOA_TopLevelWindow_Size, wxT("wxTopLevelWindow=%p::DoGetPosition = (%d,%d)"),this,(int)cocoaRect.origin.x,(int)cocoaRect.origin.y);
 }
 
Index: window.mm
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/cocoa/window.mm,v
retrieving revision 1.41
diff -u -d -b -w -b -u -2 -r1.41 window.mm
--- window.mm	2004/02/09 05:12:38	1.41
+++ window.mm	2004/03/01 02:01:33
@@ -542,5 +542,5 @@
 void wxWindowCocoa::DoSetSize(int x, int y, int width, int height, int sizeFlags)
 {
-    wxLogTrace(wxTRACE_COCOA_Window_Size,"wxWindow=%p::DoSetSizeWindow(%d,%d,%d,%d,Auto: %s%s)",this,x,y,width,height,(sizeFlags&wxSIZE_AUTO_WIDTH)?"W":".",sizeFlags&wxSIZE_AUTO_HEIGHT?"H":".");
+    wxLogTrace(wxTRACE_COCOA_Window_Size,wxT("wxWindow=%p::DoSetSizeWindow(%d,%d,%d,%d,Auto: %s%s)"),this,x,y,width,height,(sizeFlags&wxSIZE_AUTO_WIDTH)?"W":".",sizeFlags&wxSIZE_AUTO_HEIGHT?"H":".");
     int currentX, currentY;
     int currentW, currentH;
@@ -583,5 +583,5 @@
 {
     wxAutoNSAutoreleasePool pool;
-    wxLogTrace(wxTRACE_COCOA_Window_Size,"wxWindow=%p::DoMoveWindow(%d,%d,%d,%d)",this,x,y,width,height);
+    wxLogTrace(wxTRACE_COCOA_Window_Size,wxT("wxWindow=%p::DoMoveWindow(%d,%d,%d,%d)"),this,x,y,width,height);
 
     NSView *nsview = GetNSViewForSuperview();
@@ -628,5 +628,5 @@
     if(h)
         *h=(int)cocoaRect.size.height;
-    wxLogTrace(wxTRACE_COCOA_Window_Size,"wxWindow=%p::DoGetSize = (%d,%d)",this,(int)cocoaRect.size.width,(int)cocoaRect.size.height);
+    wxLogTrace(wxTRACE_COCOA_Window_Size,wxT("wxWindow=%p::DoGetSize = (%d,%d)"),this,(int)cocoaRect.size.width,(int)cocoaRect.size.height);
 }
 
@@ -643,5 +643,5 @@
     if(y)
         *y=(int)(parentRect.size.height-(cocoaRect.origin.y+cocoaRect.size.height));
-    wxLogTrace(wxTRACE_COCOA_Window_Size,"wxWindow=%p::DoGetPosition = (%d,%d)",this,(int)cocoaRect.origin.x,(int)cocoaRect.origin.y);
+    wxLogTrace(wxTRACE_COCOA_Window_Size,wxT("wxWindow=%p::DoGetPosition = (%d,%d)"),this,(int)cocoaRect.origin.x,(int)cocoaRect.origin.y);
 }
 

 	  	 
