Index: include/wx/aboutdlg.h
===================================================================
--- include/wx/aboutdlg.h	(revision 53690)
+++ include/wx/aboutdlg.h	(working copy)
@@ -33,7 +33,7 @@
 
     // name of the program, if not used defaults to wxApp::GetAppDisplayName()
     void SetName(const wxString& name) { m_name = name; }
-    wxString GetName() const
+    const wxString& GetName() const
         { return m_name.empty() ? wxTheApp->GetAppDisplayName() : m_name; }
 
     // version of the program, in free format (but without "version" word)
Index: include/wx/app.h
===================================================================
--- include/wx/app.h	(revision 53690)
+++ include/wx/app.h	(working copy)
@@ -118,7 +118,7 @@
     //     be argv[0]
 
         // set/get the application name
-    wxString GetAppName() const
+    const wxString& GetAppName() const
     {
         return m_appName.empty() ? m_className : m_appName;
     }
@@ -129,14 +129,14 @@
         // used for paths, config, and other places the user doesn't see
         //
         // so the app name could be myapp while display name could be "My App"
-    wxString GetAppDisplayName() const
+    const wxString& GetAppDisplayName() const
     {
         return m_appDisplayName.empty() ? GetAppName() : m_appDisplayName;
     }
     void SetAppDisplayName(const wxString& name) { m_appDisplayName = name; }
 
         // set/get the app class name
-    wxString GetClassName() const { return m_className; }
+    const wxString& GetClassName() const { return m_className; }
     void SetClassName(const wxString& name) { m_className = name; }
 
         // set/get the vendor name
Index: include/wx/cmndata.h
===================================================================
--- include/wx/cmndata.h	(revision 53690)
+++ include/wx/cmndata.h	(working copy)
@@ -223,7 +223,7 @@
     void SetMedia(int media) { m_media = media; }
     void SetPrintMode(wxPrintMode printMode) { m_printMode = printMode; }
 
-    wxString GetFilename() const { return m_filename; }
+    const wxString& GetFilename() const { return m_filename; }
     void SetFilename( const wxString &filename ) { m_filename = filename; }
 
     void operator=(const wxPrintData& data);
Index: include/wx/confbase.h
===================================================================
--- include/wx/confbase.h	(revision 53690)
+++ include/wx/confbase.h	(working copy)
@@ -337,8 +337,8 @@
   wxString ExpandEnvVars(const wxString& str) const;
 
     // misc accessors
-  wxString GetAppName() const { return m_appName; }
-  wxString GetVendorName() const { return m_vendorName; }
+  const wxString& GetAppName() const { return m_appName; }
+  const wxString& GetVendorName() const { return m_vendorName; }
 
   // Used wxIniConfig to set members in constructor
   void SetAppName(const wxString& appName) { m_appName = appName; }
Index: include/wx/dataview.h
===================================================================
--- include/wx/dataview.h	(revision 53690)
+++ include/wx/dataview.h	(working copy)
@@ -426,7 +426,7 @@
     virtual bool SetValue( const wxVariant& WXUNUSED(value) ) = 0;
     virtual bool GetValue( wxVariant& WXUNUSED(value) ) const = 0;
 
-    wxString GetVariantType() const             { return m_variantType; }
+    const wxString& GetVariantType() const             { return m_variantType; }
 
     virtual void SetMode( wxDataViewCellMode mode ) = 0;
     virtual wxDataViewCellMode GetMode() const = 0;
@@ -482,7 +482,7 @@
     { m_icon = other.m_icon; m_text = other.m_text; }
 
     void SetText( const wxString &text ) { m_text = text; }
-    wxString GetText() const             { return m_text; }
+    const wxString& GetText() const             { return m_text; }
     void SetIcon( const wxIcon &icon )   { m_icon = icon; }
     const wxIcon &GetIcon() const        { return m_icon; }
 
@@ -902,7 +902,7 @@
 
     void SetText( const wxString &text )
         { m_text = text; }
-    wxString GetText() const
+    const wxString& GetText() const
         { return m_text; }
     void SetIcon( const wxIcon &icon )
         { m_icon = icon; }
Index: include/wx/docview.h
===================================================================
--- include/wx/docview.h	(revision 53690)
+++ include/wx/docview.h	(working copy)
@@ -68,13 +68,13 @@
 
     // accessors
     void SetFilename(const wxString& filename, bool notifyViews = false);
-    wxString GetFilename() const { return m_documentFile; }
+    const wxString& GetFilename() const { return m_documentFile; }
 
     void SetTitle(const wxString& title) { m_documentTitle = title; }
-    wxString GetTitle() const { return m_documentTitle; }
+    const wxString& GetTitle() const { return m_documentTitle; }
 
     void SetDocumentName(const wxString& name) { m_documentTypeName = name; }
-    wxString GetDocumentName() const { return m_documentTypeName; }
+    const wxString& GetDocumentName() const { return m_documentTypeName; }
 
     bool GetDocumentSaved() const { return m_savedYet; }
     void SetDocumentSaved(bool saved = true) { m_savedYet = saved; }
@@ -191,7 +191,7 @@
     wxDocument *GetDocument() const { return m_viewDocument; }
     virtual void SetDocument(wxDocument *doc);
 
-    wxString GetViewName() const { return m_viewTypeName; }
+    const wxString& GetViewName() const { return m_viewTypeName; }
     void SetViewName(const wxString& name) { m_viewTypeName = name; }
 
     wxWindow *GetFrame() const { return m_viewFrame ; }
@@ -272,12 +272,12 @@
     // creation
     virtual bool InitDocument(wxDocument* doc, const wxString& path, long flags = 0);
 
-    wxString GetDefaultExtension() const { return m_defaultExt; }
-    wxString GetDescription() const { return m_description; }
-    wxString GetDirectory() const { return m_directory; }
+    const wxString& GetDefaultExtension() const { return m_defaultExt; }
+    const wxString& GetDescription() const { return m_description; }
+    const wxString& GetDirectory() const { return m_directory; }
     wxDocManager *GetDocumentManager() const { return m_documentManager; }
     void SetDocumentManager(wxDocManager *manager) { m_documentManager = manager; }
-    wxString GetFileFilter() const { return m_fileFilter; }
+    const wxString& GetFileFilter() const { return m_fileFilter; }
     long GetFlags() const { return m_flags; }
     virtual wxString GetViewName() const { return m_viewTypeName; }
     virtual wxString GetDocumentName() const { return m_docTypeName; }
@@ -430,7 +430,7 @@
     virtual void FileHistoryAddFilesToMenu();
     virtual void FileHistoryAddFilesToMenu(wxMenu* menu);
 
-    wxString GetLastDirectory() const { return m_lastDirectory; }
+    const wxString& GetLastDirectory() const { return m_lastDirectory; }
     void SetLastDirectory(const wxString& dir) { m_lastDirectory = dir; }
 
     // Get the current document manager
Index: include/wx/dynlib.h
===================================================================
--- include/wx/dynlib.h	(revision 53690)
+++ include/wx/dynlib.h	(working copy)
@@ -177,10 +177,10 @@
     wxDynamicLibraryDetails() { m_address = NULL; m_length = 0; }
 
     // get the (base) name
-    wxString GetName() const { return m_name; }
+    const wxString& GetName() const { return m_name; }
 
     // get the full path of this object
-    wxString GetPath() const { return m_path; }
+    const wxString& GetPath() const { return m_path; }
 
     // get the load address and the extent, return true if this information is
     // available
@@ -198,7 +198,7 @@
     }
 
     // return the version of the DLL (may be empty if no version info)
-    wxString GetVersion() const
+    const wxString& GetVersion() const
     {
         return m_version;
     }
Index: include/wx/event.h
===================================================================
--- include/wx/event.h	(revision 53690)
+++ include/wx/event.h	(working copy)
@@ -1599,7 +1599,7 @@
     bool GetChecked() const { return m_checked; }
     bool GetEnabled() const { return m_enabled; }
     bool GetShown() const { return m_shown; }
-    wxString GetText() const { return m_text; }
+    const wxString& GetText() const { return m_text; }
     bool GetSetText() const { return m_setText; }
     bool GetSetChecked() const { return m_setChecked; }
     bool GetSetEnabled() const { return m_setEnabled; }
Index: include/wx/fdrepdlg.h
===================================================================
--- include/wx/fdrepdlg.h	(revision 53690)
+++ include/wx/fdrepdlg.h	(working copy)
@@ -144,7 +144,7 @@
         : wxCommandEvent(commandType, id) { }
 
     int GetFlags() const { return GetInt(); }
-    wxString GetFindString() const { return GetString(); }
+    const wxString& GetFindString() const { return GetString(); }
     const wxString& GetReplaceString() const { return m_strReplace; }
 
     wxFindReplaceDialog *GetDialog() const
Index: include/wx/filectrl.h
===================================================================
--- include/wx/filectrl.h	(revision 53690)
+++ include/wx/filectrl.h	(working copy)
@@ -106,7 +106,7 @@
     void SetDirectory( const wxString &directory ) { m_directory = directory; }
 
     wxArrayString GetFiles() const { return m_files; }
-    wxString GetDirectory() const { return m_directory; }
+    const wxString& GetDirectory() const { return m_directory; }
 
     wxString GetFile() const;
 
Index: include/wx/filename.h
===================================================================
--- include/wx/filename.h	(revision 53690)
+++ include/wx/filename.h	(working copy)
@@ -410,15 +410,15 @@
     void SetExt( const wxString &ext )          { m_ext = ext; m_hasExt = !m_ext.empty(); }
     void ClearExt()                             { m_ext = wxEmptyString; m_hasExt = false; }
     void SetEmptyExt()                          { m_ext = wxT(""); m_hasExt = true; }
-    wxString GetExt() const                     { return m_ext; }
+    const wxString& GetExt() const                     { return m_ext; }
     bool HasExt() const                         { return m_hasExt; }
 
     void SetName( const wxString &name )        { m_name = name; }
-    wxString GetName() const                    { return m_name; }
+    const wxString& GetName() const                    { return m_name; }
     bool HasName() const                        { return !m_name.empty(); }
 
     void SetVolume( const wxString &volume )    { m_volume = volume; }
-    wxString GetVolume() const                  { return m_volume; }
+    const wxString& GetVolume() const                  { return m_volume; }
     bool HasVolume() const                      { return !m_volume.empty(); }
 
     // full name is the file name + extension (but without the path)
Index: include/wx/filepicker.h
===================================================================
--- include/wx/filepicker.h	(revision 53690)
+++ include/wx/filepicker.h	(working copy)
@@ -44,7 +44,7 @@
     wxFileDirPickerWidgetBase() {  }
     virtual ~wxFileDirPickerWidgetBase() {  }
 
-    wxString GetPath() const { return m_path; }
+    const wxString& GetPath() const { return m_path; }
     virtual void SetPath(const wxString &str) { m_path=str; }
 
     // returns the picker widget cast to wxControl
@@ -377,7 +377,7 @@
         SetEventObject(generator);
     }
 
-    wxString GetPath() const { return m_path; }
+    const wxString& GetPath() const { return m_path; }
     void SetPath(const wxString &p) { m_path = p; }
 
     // default copy ctor, assignment operator and dtor are ok
Index: include/wx/filesys.h
===================================================================
--- include/wx/filesys.h	(revision 53690)
+++ include/wx/filesys.h	(working copy)
@@ -190,7 +190,7 @@
     // (so ChangePathTo("dir/subdir/xh.htm") sets m_Path to "dir/subdir/")
     void ChangePathTo(const wxString& location, bool is_dir = false);
 
-    wxString GetPath() const {return m_Path;}
+    const wxString& GetPath() const {return m_Path;}
 
     // opens given file and returns pointer to input stream.
     // Returns NULL if opening failed.
Index: include/wx/generic/choicdgg.h
===================================================================
--- include/wx/generic/choicdgg.h	(revision 53690)
+++ include/wx/generic/choicdgg.h	(working copy)
@@ -135,7 +135,7 @@
 
     void SetSelection(int sel);
     int GetSelection() const { return m_selection; }
-    wxString GetStringSelection() const { return m_stringSelection; }
+    const wxString& GetStringSelection() const { return m_stringSelection; }
 
     // obsolete function (NB: no need to make it return wxChar, it's untyped)
     char *GetSelectionClientData() const { return (char *)m_clientData; }
Index: include/wx/generic/filectrlg.h
===================================================================
--- include/wx/generic/filectrlg.h	(revision 53690)
+++ include/wx/generic/filectrlg.h	(working copy)
@@ -56,9 +56,9 @@
     void ReadData();
 
     // get the name of the file, dir, drive
-    wxString GetFileName() const { return m_fileName; }
+    const wxString& GetFileName() const { return m_fileName; }
     // get the full path + name of the file, dir, path
-    wxString GetFilePath() const { return m_filePath; }
+    const wxString& GetFilePath() const { return m_filePath; }
     // Set the path + name and name of the item
     void SetNewName( const wxString &filePath, const wxString &fileName );
 
@@ -71,7 +71,7 @@
     // Get the time as a formatted string
     wxString GetModificationTime() const;
     // in UNIX get rwx for file, in MSW get attributes ARHS
-    wxString GetPermissions() const { return m_permissions; }
+    const wxString& GetPermissions() const { return m_permissions; }
     // Get the id of the image used in a wxImageList
     int GetImageId() const { return m_image; }
 
@@ -154,8 +154,8 @@
     virtual void GoToHomeDir();
     virtual void GoToDir( const wxString &dir );
     virtual void SetWild( const wxString &wild );
-    wxString GetWild() const { return m_wild; }
-    wxString GetDir() const { return m_dirName; }
+    const wxString& GetWild() const { return m_wild; }
+    const wxString& GetDir() const { return m_dirName; }
 
     void OnListDeleteItem( wxListEvent &event );
     void OnListDeleteAllItems( wxListEvent &event );
Index: include/wx/generic/hyperlink.h
===================================================================
--- include/wx/generic/hyperlink.h	(revision 53690)
+++ include/wx/generic/hyperlink.h	(working copy)
@@ -54,7 +54,7 @@
     wxColour GetVisitedColour() const { return m_visitedColour; }
     void SetVisitedColour(const wxColour &colour);
 
-    wxString GetURL() const { return m_url; }
+    const wxString& GetURL() const { return m_url; }
     void SetURL (const wxString &url) { m_url=url; }
 
     void SetVisited(bool visited = true) { m_visited=visited; }
Index: include/wx/generic/tabg.h
===================================================================
--- include/wx/generic/tabg.h	(revision 53690)
+++ include/wx/generic/tabg.h	(working copy)
@@ -36,7 +36,7 @@
 
     virtual void OnDraw(wxDC& dc, bool lastInRow);
     void SetLabel(const wxString& str) { m_controlLabel = str; }
-    wxString GetLabel(void) const { return m_controlLabel; }
+    const wxString& GetLabel(void) const { return m_controlLabel; }
 
     void SetFont(const wxFont& f) { m_labelFont = f; }
     wxFont *GetFont(void) const { return (wxFont*) & m_labelFont; }
Index: include/wx/generic/textdlgg.h
===================================================================
--- include/wx/generic/textdlgg.h	(revision 53690)
+++ include/wx/generic/textdlgg.h	(working copy)
@@ -45,7 +45,7 @@
                       const wxPoint& pos = wxDefaultPosition);
 
     void SetValue(const wxString& val);
-    wxString GetValue() const { return m_value; }
+    const wxString& GetValue() const { return m_value; }
 
 #if wxUSE_VALIDATORS
     void SetTextValidator( const wxTextValidator& validator );
Index: include/wx/gtk1/menuitem.h
===================================================================
--- include/wx/gtk1/menuitem.h	(revision 53690)
+++ include/wx/gtk1/menuitem.h	(working copy)
@@ -47,7 +47,7 @@
     void SetLabelWidget(GtkWidget *labelWidget) { m_labelWidget = labelWidget; }
     wxString GetFactoryPath() const;
 
-    wxString GetHotKey() const { return m_hotKey; }
+    const wxString& GetHotKey() const { return m_hotKey; }
 
     // compatibility only, don't use in new code
     wxMenuItem(wxMenu *parentMenu,
Index: include/wx/gtk1/tooltip.h
===================================================================
--- include/wx/gtk1/tooltip.h	(revision 53690)
+++ include/wx/gtk1/tooltip.h	(working copy)
@@ -40,7 +40,7 @@
 
     // get/set the tooltip text
     void SetTip( const wxString &tip );
-    wxString GetTip() const { return m_text; }
+    const wxString& GetTip() const { return m_text; }
 
     wxWindow *GetWindow() const { return m_window; }
     bool IsOk() const { return m_window != NULL; }
Index: include/wx/gtk/dataobj2.h
===================================================================
--- include/wx/gtk/dataobj2.h	(revision 53690)
+++ include/wx/gtk/dataobj2.h	(working copy)
@@ -98,7 +98,7 @@
 public:
     wxURLDataObject(const wxString& url = wxEmptyString);
 
-    wxString GetURL() const { return m_url; }
+    const wxString& GetURL() const { return m_url; }
     void SetURL(const wxString& url) { m_url = url; }
 
     virtual size_t GetDataSize() const;
Index: include/wx/gtk/menuitem.h
===================================================================
--- include/wx/gtk/menuitem.h	(revision 53690)
+++ include/wx/gtk/menuitem.h	(working copy)
@@ -47,7 +47,7 @@
     void SetLabelWidget(GtkWidget *labelWidget) { m_labelWidget = labelWidget; }
     wxString GetFactoryPath() const;
 
-    wxString GetHotKey() const { return m_hotKey; }
+    const wxString& GetHotKey() const { return m_hotKey; }
 
     // splits given string in the label, doing & => _ translation, which is returned,
     // and in the hotkey which is used to set given pointer
Index: include/wx/gtk/tooltip.h
===================================================================
--- include/wx/gtk/tooltip.h	(revision 53690)
+++ include/wx/gtk/tooltip.h	(working copy)
@@ -40,7 +40,7 @@
 
     // get/set the tooltip text
     void SetTip( const wxString &tip );
-    wxString GetTip() const { return m_text; }
+    const wxString& GetTip() const { return m_text; }
 
     wxWindow *GetWindow() const { return m_window; }
     bool IsOk() const { return m_window != NULL; }
Index: include/wx/html/helpdata.h
===================================================================
--- include/wx/html/helpdata.h	(revision 53690)
+++ include/wx/html/helpdata.h	(working copy)
@@ -41,10 +41,10 @@
         // for debugging, give the contents index obvious default values
         m_ContentsStart = m_ContentsEnd = -1;
     }
-    wxString GetBookFile() const { return m_BookFile; }
-    wxString GetTitle() const { return m_Title; }
-    wxString GetStart() const { return m_Start; }
-    wxString GetBasePath() const { return m_BasePath; }
+    const wxString& GetBookFile() const { return m_BookFile; }
+    const wxString& GetTitle() const { return m_Title; }
+    const wxString& GetStart() const { return m_Start; }
+    const wxString& GetBasePath() const { return m_BasePath; }
     /* SetContentsRange: store in the bookrecord where in the index/contents lists the
      * book's records are stored. This to facilitate searching in a specific book.
      * This code will have to be revised when loading/removing books becomes dynamic.
Index: include/wx/html/htmlcell.h
===================================================================
--- include/wx/html/htmlcell.h	(revision 53690)
+++ include/wx/html/htmlcell.h	(working copy)
@@ -651,8 +651,8 @@
     void SetEvent(const wxMouseEvent *e) { m_Event = e; }
     void SetHtmlCell(const wxHtmlCell *e) { m_Cell = e; }
 
-    wxString GetHref() const { return m_Href; }
-    wxString GetTarget() const { return m_Target; }
+    const wxString& GetHref() const { return m_Href; }
+    const wxString& GetTarget() const { return m_Target; }
     const wxMouseEvent* GetEvent() const { return m_Event; }
     const wxHtmlCell* GetHtmlCell() const { return m_Cell; }
 
Index: include/wx/html/htmltag.h
===================================================================
--- include/wx/html/htmltag.h	(revision 53690)
+++ include/wx/html/htmltag.h	(working copy)
@@ -83,7 +83,7 @@
     wxHtmlTag *GetNextTag() const;
 
     // Returns tag's name in uppercase.
-    inline wxString GetName() const {return m_Name;}
+    inline const wxString& GetName() const {return m_Name;}
 
     // Returns true if the tag has given parameter. Parameter
     // should always be in uppercase.
Index: include/wx/html/htmlwin.h
===================================================================
--- include/wx/html/htmlwin.h	(revision 53690)
+++ include/wx/html/htmlwin.h	(working copy)
@@ -283,11 +283,11 @@
     bool LoadFile(const wxFileName& filename);
 
     // Returns full location of opened page
-    wxString GetOpenedPage() const {return m_OpenedPage;}
+    const wxString& GetOpenedPage() const {return m_OpenedPage;}
     // Returns anchor within opened page
-    wxString GetOpenedAnchor() const {return m_OpenedAnchor;}
+    const wxString& GetOpenedAnchor() const {return m_OpenedAnchor;}
     // Returns <TITLE> of opened page or empty string otherwise
-    wxString GetOpenedPageTitle() const {return m_OpenedPageTitle;}
+    const wxString& GetOpenedPageTitle() const {return m_OpenedPageTitle;}
 
     // Sets frame in which page title will  be displayed. Format is format of
     // frame title, e.g. "HtmlHelp : %s". It must contain exactly one %s
Index: include/wx/html/webkit.h
===================================================================
--- include/wx/html/webkit.h	(revision 53690)
+++ include/wx/html/webkit.h	(working copy)
@@ -60,13 +60,13 @@
     bool CanGetPageSource();
     wxString GetPageSource();
     void SetPageSource(const wxString& source, const wxString& baseUrl = wxEmptyString);
-	wxString GetPageURL(){ return m_currentURL; }
+	const wxString& GetPageURL(){ return m_currentURL; }
     void SetPageTitle(const wxString& title) { m_pageTitle = title; }
-	wxString GetPageTitle(){ return m_pageTitle; }
+	const wxString& GetPageTitle(){ return m_pageTitle; }
 
     // since these worked in 2.6, add wrappers
     void SetTitle(const wxString& title) { SetPageTitle(title); }
-    wxString GetTitle() { return GetPageTitle(); }
+    const wxString& GetTitle() { return GetPageTitle(); }
 
     wxString GetSelection();
 
@@ -140,7 +140,7 @@
 public:
     bool IsCancelled() { return m_cancelled; }
     void Cancel(bool cancel = true) { m_cancelled = cancel; }
-    wxString GetURL() { return m_url; }
+    const wxString& GetURL() { return m_url; }
     void SetURL(const wxString& url) { m_url = url; }
     void SetNavigationType(int navType) { m_navType = navType; }
     int GetNavigationType() { return m_navType; }
@@ -161,7 +161,7 @@
 public:
     int GetState() { return m_state; }
     void SetState(const int state) { m_state = state; }
-    wxString GetURL() { return m_url; }
+    const wxString& GetURL() { return m_url; }
     void SetURL(const wxString& url) { m_url = url; }
 
     wxWebKitStateChangedEvent( wxWindow* win = (wxWindow*) NULL );
Index: include/wx/html/winpars.h
===================================================================
--- include/wx/html/winpars.h	(revision 53690)
+++ include/wx/html/winpars.h	(working copy)
@@ -114,7 +114,7 @@
     void SetFontUnderlined(int x) {m_FontUnderlined = x;}
     int GetFontFixed() const {return m_FontFixed;}
     void SetFontFixed(int x) {m_FontFixed = x;}
-    wxString GetFontFace() const {return GetFontFixed() ? m_FontFaceFixed : m_FontFaceNormal;}
+    const wxString& GetFontFace() const {return GetFontFixed() ? m_FontFaceFixed : m_FontFaceNormal;}
     void SetFontFace(const wxString& face);
 
     int GetAlign() const {return m_Align;}
Index: include/wx/hyperlink.h
===================================================================
--- include/wx/hyperlink.h	(revision 53690)
+++ include/wx/hyperlink.h	(working copy)
@@ -107,7 +107,7 @@
 
     // Returns the URL associated with the hyperlink control
     // that the user clicked on.
-    wxString GetURL() const { return m_url; }
+    const wxString& GetURL() const { return m_url; }
     void SetURL(const wxString &url) { m_url=url; }
 
     // default copy ctor, assignment operator and dtor are ok
Index: include/wx/mac/carbon/helpxxxx.h
===================================================================
--- include/wx/mac/carbon/helpxxxx.h	(revision 53690)
+++ include/wx/mac/carbon/helpxxxx.h	(working copy)
@@ -39,7 +39,7 @@
   virtual bool Quit();
   virtual void OnQuit();
 
-  inline wxString GetHelpFile() const { return m_helpFile; }
+  inline const wxString& GetHelpFile() const { return m_helpFile; }
 
 protected:
   wxString m_helpFile;
Index: include/wx/mac/carbon/treectrl.h
===================================================================
--- include/wx/mac/carbon/treectrl.h	(revision 53690)
+++ include/wx/mac/carbon/treectrl.h	(working copy)
@@ -92,7 +92,7 @@
     inline long GetItemId() const { return m_itemId; }
     inline long GetState() const { return m_state; }
     inline long GetStateMask() const { return m_stateMask; }
-    inline wxString GetText() const { return m_text; }
+    inline const wxString& GetText() const { return m_text; }
     inline int GetImage() const { return m_image; }
     inline int GetSelectedImage() const { return m_selectedImage; }
     inline int GetChildren() const { return m_children; }
Index: include/wx/menu.h
===================================================================
--- include/wx/menu.h	(revision 53690)
+++ include/wx/menu.h	(working copy)
@@ -248,7 +248,7 @@
 
     // the title
     virtual void SetTitle(const wxString& title) { m_title = title; }
-    const wxString GetTitle() const { return m_title; }
+    const wxString& GetTitle() const { return m_title; }
 
     // event handler
     void SetEventHandler(wxEvtHandler *handler) { m_eventHandler = handler; }
Index: include/wx/motif/dnd.h
===================================================================
--- include/wx/motif/dnd.h	(revision 53690)
+++ include/wx/motif/dnd.h	(working copy)
@@ -92,7 +92,7 @@
     void SetId( const wxString& id )
     { m_id = id; }
 
-    wxString GetId()
+    const wxString& GetId()
     { return m_id; }
 
 private:
Index: include/wx/msw/helpchm.h
===================================================================
--- include/wx/msw/helpchm.h	(revision 53690)
+++ include/wx/msw/helpchm.h	(working copy)
@@ -37,7 +37,7 @@
                                wxHelpSearchMode mode = wxHELP_SEARCH_ALL);
     virtual bool Quit();
 
-    wxString GetHelpFile() const { return m_helpFile; }
+    const wxString& GetHelpFile() const { return m_helpFile; }
 
     // helper of DisplayTextPopup(), also used in wxSimpleHelpProvider::ShowHelp
     static bool ShowContextHelpPopup(const wxString& text,
Index: include/wx/msw/helpwin.h
===================================================================
--- include/wx/msw/helpwin.h	(revision 53690)
+++ include/wx/msw/helpwin.h	(working copy)
@@ -41,7 +41,7 @@
                                wxHelpSearchMode mode = wxHELP_SEARCH_ALL);
     virtual bool Quit();
 
-    inline wxString GetHelpFile() const { return m_helpFile; }
+    inline const wxString& GetHelpFile() const { return m_helpFile; }
 
 protected:
     // Append extension if necessary.
Index: include/wx/msw/menu.h
===================================================================
--- include/wx/msw/menu.h	(revision 53690)
+++ include/wx/msw/menu.h	(working copy)
@@ -131,7 +131,7 @@
     void Create( wxMenu *menu , const wxString &title )
     { m_menu = menu; m_title = title; }
     wxMenu* GetMenu() const { return m_menu; }
-    wxString GetTitle() const { return m_title; }
+    const wxString& GetTitle() const { return m_title; }
 private :
     wxMenu *m_menu;
     wxString m_title;
Index: include/wx/msw/notebook.h
===================================================================
--- include/wx/msw/notebook.h	(revision 53690)
+++ include/wx/msw/notebook.h	(working copy)
@@ -41,7 +41,7 @@
     }
 
     wxNotebookPage* GetPage() const { return m_page; }
-    wxString GetText() const { return m_text; }
+    const wxString& GetText() const { return m_text; }
     bool GetSelected() const { return m_selected; }
     int GetImageId() const { return m_imageId; }
 
Index: include/wx/msw/toplevel.h
===================================================================
--- include/wx/msw/toplevel.h	(revision 53690)
+++ include/wx/msw/toplevel.h	(working copy)
@@ -179,7 +179,7 @@
 
         int GetId() const {return m_id;}
         wxMenu* GetMenu() const {return m_menu;}
-        wxString GetLabel() {return m_label;}
+        const wxString& GetLabel() {return m_label;}
 
         static wxMenu *DuplicateMenu(wxMenu *menu);
 
Index: include/wx/msw/wince/helpwce.h
===================================================================
--- include/wx/msw/wince/helpwce.h	(revision 53690)
+++ include/wx/msw/wince/helpwce.h	(working copy)
@@ -37,7 +37,7 @@
                                wxHelpSearchMode mode = wxHELP_SEARCH_ALL);
     virtual bool Quit();
 
-    wxString GetHelpFile() const { return m_helpFile; }
+    const wxString& GetHelpFile() const { return m_helpFile; }
 
 protected:
     // Append extension if necessary.
Index: include/wx/os2/gdiimage.h
===================================================================
--- include/wx/os2/gdiimage.h	(revision 53690)
+++ include/wx/os2/gdiimage.h	(working copy)
@@ -101,8 +101,8 @@
     void SetExtension(const wxString& rExt) { m_sExtension = rExt; }
     void SetType(wxBitmapType lType) { m_lType = lType; }
 
-    wxString GetName() const { return m_sName; }
-    wxString GetExtension() const { return m_sExtension; }
+    const wxString& GetName() const { return m_sName; }
+    const wxString& GetExtension() const { return m_sExtension; }
     wxBitmapType GetType() const { return m_lType; }
 
     // real handler operations: to implement in derived classes
Index: include/wx/os2/helpwin.h
===================================================================
--- include/wx/os2/helpwin.h	(revision 53690)
+++ include/wx/os2/helpwin.h	(working copy)
@@ -40,7 +40,7 @@
   virtual bool Quit();
   virtual void OnQuit();
 
-  inline wxString GetHelpFile() const { return m_helpFile; }
+  inline const wxString& GetHelpFile() const { return m_helpFile; }
 
 protected:
   wxString m_helpFile;
Index: include/wx/palmos/gdiimage.h
===================================================================
--- include/wx/palmos/gdiimage.h	(revision 53690)
+++ include/wx/palmos/gdiimage.h	(working copy)
@@ -88,8 +88,8 @@
     void SetExtension(const wxString& ext) { m_extension = ext; }
     void SetType(long type) { m_type = type; }
 
-    wxString GetName() const { return m_name; }
-    wxString GetExtension() const { return m_extension; }
+    const wxString& GetName() const { return m_name; }
+    const wxString& GetExtension() const { return m_extension; }
     long GetType() const { return m_type; }
 
     // real handler operations: to implement in derived classes
Index: include/wx/palmos/help.h
===================================================================
--- include/wx/palmos/help.h	(revision 53690)
+++ include/wx/palmos/help.h	(working copy)
@@ -41,7 +41,7 @@
                                wxHelpSearchMode mode = wxHELP_SEARCH_ALL);
     virtual bool Quit();
 
-    inline wxString GetHelpFile() const { return m_helpFile; }
+    inline const wxString& GetHelpFile() const { return m_helpFile; }
 
 protected:
     // Append extension if necessary.
Index: include/wx/palmos/menu.h
===================================================================
--- include/wx/palmos/menu.h	(revision 53690)
+++ include/wx/palmos/menu.h	(working copy)
@@ -112,7 +112,7 @@
     void Create( wxMenu *menu , const wxString &title )
     { m_menu = menu ; m_title = title ; }
     wxMenu* GetMenu() const { return m_menu ; }
-    wxString GetTitle() const { return m_title ; }
+    const wxString& GetTitle() const { return m_title ; }
 private :
     wxMenu *m_menu ;
     wxString m_title ;
Index: include/wx/palmos/notebook.h
===================================================================
--- include/wx/palmos/notebook.h	(revision 53690)
+++ include/wx/palmos/notebook.h	(working copy)
@@ -33,7 +33,7 @@
     void Create( wxNotebookPage *page , const wxString &text , bool selected , int imageId )
     { m_page = page ; m_text = text ; m_selected = selected ; m_imageId = imageId ; }
     wxNotebookPage* GetPage() const { return m_page ; }
-    wxString GetText() const { return m_text ; }
+    const wxString& GetText() const { return m_text ; }
     bool GetSelected() const { return m_selected ; }
     int GetImageId() const { return m_imageId; }
 private :
Index: include/wx/palmos/stdpaths.h
===================================================================
--- include/wx/palmos/stdpaths.h	(revision 53690)
+++ include/wx/palmos/stdpaths.h	(working copy)
@@ -20,7 +20,7 @@
 {
 public:
     void SetInstallPrefix(const wxString& prefix) { m_prefix = prefix; }
-    wxString GetInstallPrefix() const { return m_prefix; }
+    const wxString& GetInstallPrefix() const { return m_prefix; }
     virtual wxString GetExecutablePath() const { return m_prefix; }
     virtual wxString GetConfigDir() const { return m_prefix; }
     virtual wxString GetUserConfigDir() const { return m_prefix; }
Index: include/wx/private/fontmgr.h
===================================================================
--- include/wx/private/fontmgr.h	(revision 53690)
+++ include/wx/private/fontmgr.h	(working copy)
@@ -222,7 +222,7 @@
     const wxNativeFontInfo *GetNativeFontInfo() const { return &m_info; }
 
     int GetPointSize() const { return m_info.pointSize; }
-    wxString GetFaceName() const { return m_info.faceName; }
+    const wxString& GetFaceName() const { return m_info.faceName; }
     int GetFamily() const { return m_info.family; }
     int GetStyle() const { return m_info.style; }
     int GetWeight() const { return m_info.weight; }
Index: include/wx/richtext/richtextbuffer.h
===================================================================
--- include/wx/richtext/richtextbuffer.h	(revision 53690)
+++ include/wx/richtext/richtextbuffer.h	(working copy)
@@ -1823,11 +1823,11 @@
 
     /// The name of the nandler
     void SetName(const wxString& name) { m_name = name; }
-    wxString GetName() const { return m_name; }
+    const wxString& GetName() const { return m_name; }
 
     /// The default extension to recognise
     void SetExtension(const wxString& ext) { m_extension = ext; }
-    wxString GetExtension() const { return m_extension; }
+    const wxString& GetExtension() const { return m_extension; }
 
     /// The handler type
     void SetType(int type) { m_type = type; }
Index: include/wx/richtext/richtextctrl.h
===================================================================
--- include/wx/richtext/richtextctrl.h	(revision 53690)
+++ include/wx/richtext/richtextctrl.h	(working copy)
@@ -134,7 +134,7 @@
     virtual wxString GetStringSelection() const;
 
     /// Get filename
-    wxString GetFilename() const { return m_filename; }
+    const wxString& GetFilename() const { return m_filename; }
 
     /// Set filename
     void SetFilename(const wxString& filename) { m_filename = filename; }
Index: include/wx/richtext/richtextsymboldlg.h
===================================================================
--- include/wx/richtext/richtextsymboldlg.h	(revision 53690)
+++ include/wx/richtext/richtextsymboldlg.h	(working copy)
@@ -125,16 +125,16 @@
 
 ////@begin wxSymbolPickerDialog member function declarations
 
-    wxString GetFontName() const { return m_fontName ; }
+    const wxString& GetFontName() const { return m_fontName ; }
     void SetFontName(wxString value) { m_fontName = value ; }
 
-    wxString GetSymbol() const { return m_symbol ; }
+    const wxString& GetSymbol() const { return m_symbol ; }
     void SetSymbol(wxString value) { m_symbol = value ; }
 
     bool GetFromUnicode() const { return m_fromUnicode ; }
     void SetFromUnicode(bool value) { m_fromUnicode = value ; }
 
-    wxString GetNormalTextFontName() const { return m_normalTextFontName ; }
+    const wxString& GetNormalTextFontName() const { return m_normalTextFontName ; }
     void SetNormalTextFontName(wxString value) { m_normalTextFontName = value ; }
 
     /// Retrieves bitmap resources
Index: include/wx/stackwalk.h
===================================================================
--- include/wx/stackwalk.h	(revision 53690)
+++ include/wx/stackwalk.h	(working copy)
@@ -49,13 +49,13 @@
 
     // return the unmangled (if possible) name of the function containing this
     // frame
-    wxString GetName() const { ConstCast()->OnGetName(); return m_name; }
+    const wxString& GetName() const { ConstCast()->OnGetName(); return m_name; }
 
     // return the instruction pointer offset from the start of the function
     size_t GetOffset() const { ConstCast()->OnGetName(); return m_offset; }
 
     // get the module this function belongs to (not always available)
-    wxString GetModule() const { ConstCast()->OnGetName(); return m_module; }
+    const wxString& GetModule() const { ConstCast()->OnGetName(); return m_module; }
 
 
     // return true if we have the filename and line number for this frame
@@ -63,7 +63,7 @@
 
     // return the name of the file containing this frame, empty if
     // unavailable (typically because debug info is missing)
-    wxString GetFileName() const
+    const wxString& GetFileName() const
         { ConstCast()->OnGetLocation(); return m_filename; }
 
     // return the line number of this frame, 0 if unavailable
Index: include/wx/stc/stc.h
===================================================================
--- include/wx/stc/stc.h	(revision 53690)
+++ include/wx/stc/stc.h	(working copy)
@@ -3716,7 +3716,7 @@
     int  GetKey()  const             { return m_key; }
     int  GetModifiers() const        { return m_modifiers; }
     int  GetModificationType() const { return m_modificationType; }
-    wxString GetText() const         { return m_text; }
+    const wxString& GetText() const         { return m_text; }
     int  GetLength() const           { return m_length; }
     int  GetLinesAdded() const       { return m_linesAdded; }
     int  GetLine() const             { return m_line; }
@@ -3729,7 +3729,7 @@
     int  GetListType() const         { return m_listType; }
     int  GetX() const                { return m_x; }
     int  GetY() const                { return m_y; }
-    wxString GetDragText()           { return m_dragText; }
+    const wxString& GetDragText()           { return m_dragText; }
     bool GetDragAllowMove()          { return m_dragAllowMove; }
 #ifdef STC_USE_DND
     wxDragResult GetDragResult()     { return m_dragResult; }
Index: include/wx/stdpaths.h
===================================================================
--- include/wx/stdpaths.h	(revision 53690)
+++ include/wx/stdpaths.h	(working copy)
@@ -195,7 +195,7 @@
 {
 public:
     void SetInstallPrefix(const wxString& prefix) { m_prefix = prefix; }
-    wxString GetInstallPrefix() const { return m_prefix; }
+    const wxString& GetInstallPrefix() const { return m_prefix; }
 
     virtual wxString GetExecutablePath() const { return m_prefix; }
     virtual wxString GetConfigDir() const { return m_prefix; }
Index: include/wx/treebase.h
===================================================================
--- include/wx/treebase.h	(revision 53690)
+++ include/wx/treebase.h	(working copy)
@@ -312,7 +312,7 @@
 
         // Set the tooltip for the item (for EVT\_TREE\_ITEM\_GETTOOLTIP events)
     void SetToolTip(const wxString& toolTip) { m_label = toolTip; }
-    wxString GetToolTip() { return m_label; }
+    const wxString& GetToolTip() { return m_label; }
 
 private:
     // not all of the members are used (or initialized) for all events
Index: include/wx/url.h
===================================================================
--- include/wx/url.h	(revision 53690)
+++ include/wx/url.h	(working copy)
@@ -56,7 +56,7 @@
 
     wxProtocol& GetProtocol()        { return *m_protocol; }
     wxURLError GetError() const      { return m_error; }
-    wxString GetURL() const          { return m_url; }
+    const wxString& GetURL() const          { return m_url; }
 
     wxURLError SetURL(const wxString &url)
         { *this = url; return m_error; }
Index: include/wx/volume.h
===================================================================
--- include/wx/volume.h	(revision 53690)
+++ include/wx/volume.h	(working copy)
@@ -88,8 +88,8 @@
 
     // get the name of the volume and the name which should be displayed to the
     // user
-    wxString GetName() const { return m_volName; }
-    wxString GetDisplayName() const { return m_dispName; }
+    const wxString& GetName() const { return m_volName; }
+    const wxString& GetDisplayName() const { return m_dispName; }
 
     // TODO: operatios (Mount(), Unmount(), Eject(), ...)?
 
Index: include/wx/x11/dnd.h
===================================================================
--- include/wx/x11/dnd.h	(revision 53690)
+++ include/wx/x11/dnd.h	(working copy)
@@ -92,7 +92,7 @@
     void SetId( const wxString& id )
     { m_id = id; }
     
-    wxString GetId()
+    const wxString& GetId()
     { return m_id; }
     
 private:
Index: include/wx/xml/xml.h
===================================================================
--- include/wx/xml/xml.h	(revision 53690)
+++ include/wx/xml/xml.h	(working copy)
@@ -69,8 +69,8 @@
             : m_name(name), m_value(value), m_next(next) {}
     virtual ~wxXmlAttribute() {}
 
-    wxString GetName() const { return m_name; }
-    wxString GetValue() const { return m_value; }
+    const wxString& GetName() const { return m_name; }
+    const wxString& GetValue() const { return m_value; }
     wxXmlAttribute *GetNext() const { return m_next; }
 
     void SetName(const wxString& name) { m_name = name; }
@@ -290,7 +290,7 @@
     // Returns encoding of in-memory representation of the document
     // (same as passed to Load or ctor, defaults to UTF-8).
     // NB: this is meaningless in Unicode build where data are stored as wchar_t*
-    wxString GetEncoding() const { return m_encoding; }
+    const wxString& GetEncoding() const { return m_encoding; }
     void SetEncoding(const wxString& enc) { m_encoding = enc; }
 #endif
 
