Only in wxWin/contrib\samples\fl: bcc_mswud
diff -bu1 -r wxWindows/contrib\samples\fl/fl_demo1.cpp wxWin/contrib\samples\fl/fl_demo1.cpp
--- wxWindows/contrib\samples\fl/fl_demo1.cpp	2003-08-03 20:29:53.000000000 +0200
+++ wxWin/contrib\samples\fl/fl_demo1.cpp	2003-08-22 13:11:54.000000000 +0200
@@ -61,5 +61,5 @@
     
-    file_menu->Append( NEW_TEST_LOAD, "&Load layouts"  );
-    file_menu->Append( NEW_TEST_SAVE, "&Store layouts" );
-    file_menu->Append( NEW_TEST_EXIT, "E&xit" );
+    file_menu->Append( NEW_TEST_LOAD, _T("&Load layouts")  );
+    file_menu->Append( NEW_TEST_SAVE, _T("&Store layouts") );
+    file_menu->Append( NEW_TEST_EXIT, _T("E&xit") );
     
@@ -67,3 +67,3 @@
     
-    menu_bar->Append(file_menu,   "&File");
+    menu_bar->Append(file_menu,   _T("&File"));
     
@@ -80,6 +80,6 @@
     
-    wxMessageBox("Hello, this demo has a bunch of yet-not-fixed-bugs and missing functionality\n" \
+    wxMessageBox(_T("Hello, this demo has a bunch of yet-not-fixed-bugs and missing functionality\n" \
                  "The ONLY purpose is to demonstrate self-layouting toolbars,\nflat-bitmapped-buttons and 2-new FL-plugins" \
                  "(cbRowDragPlugin & cbBarHintsPlugin)\n\n" \
-                 "BTW, disabled images and label-text are rendered at run-time" );
+                 "BTW, disabled images and label-text are rendered at run-time") );
     
@@ -98,13 +98,13 @@
 
-void MyFrame::OnLoad( wxCommandEvent& event )
+void MyFrame::OnLoad( wxCommandEvent& WXUNUSED(event) )
 {						
-    wxMessageBox("Hey - you found a BIG question-mark !!");
+    wxMessageBox(_T("Hey - you found a BIG question-mark !!"));
 }
 
-void MyFrame::OnSave( wxCommandEvent& event )
+void MyFrame::OnSave( wxCommandEvent& WXUNUSED(event) )
 {
-    wxMessageBox("Hey - you found another BIG question-mark !!");
+    wxMessageBox(_T("Hey - you found another BIG question-mark !!"));
 }
 
-void MyFrame::OnExit( wxCommandEvent& event )
+void MyFrame::OnExit( wxCommandEvent& WXUNUSED(event) )
 {
@@ -126,3 +126,3 @@
 MyFrame::MyFrame(wxFrame *frame)
-    : wxFrame( frame, -1, "wxWindows 2.0 wxFrameLayout Test Application", wxDefaultPosition, 
+    : wxFrame( frame, -1, _T("wxWindows 2.0 wxFrameLayout Test Application"), wxDefaultPosition, 
           wxSize( 700, 500 ), 
@@ -130,5 +130,5 @@
           wxTHICK_FRAME   | wxSYSTEM_MENU  | wxCAPTION, 
-          "freimas" )
+          _T("freimas") )
 {
-    mpClientWnd = CreateTextCtrl( "Client window" );
+    mpClientWnd = CreateTextCtrl( _T("Client window") );
     
@@ -185,3 +185,3 @@
     
-    mpLayout->AddBar( CreateTextCtrl("Hello"),  // bar window
+    mpLayout->AddBar( CreateTextCtrl(_T("Hello")),  // bar window
                       sizes0, FL_ALIGN_TOP,     // alignment ( 0-top,1-bottom, etc)
@@ -189,3 +189,3 @@
                       0,                        // offset from the start of row (in pixels)
-                      "InfoViewer1",            // name to refer in customization pop-ups
+                      _T("InfoViewer1"),        // name to refer in customization pop-ups
                       TRUE
@@ -193,3 +193,3 @@
     
-    mpLayout->AddBar( CreateTextCtrl("Bye"),    // bar window
+    mpLayout->AddBar( CreateTextCtrl(_T("Bye")),    // bar window
                       sizes0, FL_ALIGN_TOP,     // alignment ( 0-top,1-bottom, etc)
@@ -197,3 +197,3 @@
                       0,                        // offset from the start of row (in pixels)
-                      "InfoViewer2",            // name to refer in customization pop-ups
+                      _T("InfoViewer2"),        // name to refer in customization pop-ups
                       TRUE
@@ -201,3 +201,3 @@
     
-    mpLayout->AddBar( CreateTextCtrl("Fixed0"), // bar window
+    mpLayout->AddBar( CreateTextCtrl(_T("Fixed0")), // bar window
                       sizes1, FL_ALIGN_TOP,     // alignment ( 0-top,1-bottom, etc)
@@ -205,3 +205,3 @@
                       0,                        // offset from the start of row (in pixels)
-                      "ToolBar1",               // name to refer in customization pop-ups
+                      _T("ToolBar1"),               // name to refer in customization pop-ups
                       TRUE
@@ -215,10 +215,10 @@
 
-    pToolBar->AddTool( 1001, BMP_DIR "new.bmp" );
+    pToolBar->AddTool( 1001, _T(BMP_DIR "new.bmp") );
     pToolBar->AddSeparator();
-    pToolBar->AddTool( 1002, BMP_DIR "open.bmp" );
-    pToolBar->AddTool( 1003, BMP_DIR "save.bmp" );
+    pToolBar->AddTool( 1002, _T(BMP_DIR "open.bmp") );
+    pToolBar->AddTool( 1003, _T(BMP_DIR "save.bmp") );
     pToolBar->AddSeparator(new wxMySeparatorLine(pToolBar, -1));    
-    pToolBar->AddTool( 1004, BMP_DIR "cut.bmp" );
-    pToolBar->AddTool( 1005, BMP_DIR "copy.bmp" );
-    pToolBar->AddTool( 1006, BMP_DIR "paste.bmp" );
+    pToolBar->AddTool( 1004, _T(BMP_DIR "cut.bmp") );
+    pToolBar->AddTool( 1005, _T(BMP_DIR "copy.bmp") );
+    pToolBar->AddTool( 1006, _T(BMP_DIR "paste.bmp") );
     
@@ -229,3 +229,3 @@
                       0,                    // offset from the start of row (in pixels)
-                      "ToolBar2",           // name to refer in customization pop-ups
+                      _T("ToolBar2"),       // name to refer in customization pop-ups
                       FALSE
diff -bu1 -r wxWindows/contrib\samples\fl/fl_demo2.cpp wxWin/contrib\samples\fl/fl_demo2.cpp
--- wxWindows/contrib\samples\fl/fl_demo2.cpp	2003-08-03 20:29:53.000000000 +0200
+++ wxWin/contrib\samples\fl/fl_demo2.cpp	2003-08-22 13:59:37.000000000 +0200
@@ -57,3 +57,3 @@
     // Create the main frame window
-    MyFrame *frame = new MyFrame(NULL, "wxWindows 2.0 wxFrameLayout demo", 50, 50, 650, 540);
+    MyFrame *frame = new MyFrame(NULL, _T("wxWindows 2.0 wxFrameLayout demo"), 50, 50, 650, 540);
     
@@ -61,6 +61,6 @@
 #ifdef __WINDOWS__
-    frame->SetIcon(wxIcon("mondrian"));
+    frame->SetIcon(wxIcon(_T("mondrian")));
 #endif
 #ifdef __X__
-    frame->SetIcon(wxIcon("aiai.xbm"));
+    frame->SetIcon(wxIcon(_T("aiai.xbm"_));
 #endif
@@ -71,23 +71,23 @@
     
-    file_menu->Append( ID_LOAD,  "&Load layouts"  );
-    file_menu->Append( ID_STORE, "&Store layouts" );
+    file_menu->Append( ID_LOAD,  _T("&Load layouts")  );
+    file_menu->Append( ID_STORE, _T("&Store layouts") );
     file_menu->AppendSeparator();
     
-    file_menu->Append( ID_AUTOSAVE, "&Auto Save Layouts", "save layouts on exit", TRUE );
+    file_menu->Append( ID_AUTOSAVE, _T("&Auto Save Layouts"), _T("save layouts on exit"), TRUE );
     file_menu->AppendSeparator();
     
-    file_menu->Append(MINIMAL_ABOUT, "A&bout !");
-    file_menu->Append(MINIMAL_QUIT, "E&xit\tTab");
+    file_menu->Append(MINIMAL_ABOUT, _T("A&bout !"));
+    file_menu->Append(MINIMAL_QUIT, _T("E&xit\tTab"));
     
-    //active_menu->Append( ID_SETTINGS, "&Settings...\tCtrl" );
+    //active_menu->Append( ID_SETTINGS, _T("&Settings...\tCtrl") );
     //active_menu->AppendSeparator();
     
-    active_menu->Append( ID_REMOVE,    "&Remove Active" );
-    active_menu->Append( ID_REMOVEALL, "Remove &All" );
-    active_menu->Append( ID_RECREATE,  "Re&create" );
+    active_menu->Append( ID_REMOVE,    _T("&Remove Active") );
+    active_menu->Append( ID_REMOVEALL, _T("Remove &All") );
+    active_menu->Append( ID_RECREATE,  _T("Re&create") );
     active_menu->AppendSeparator();
     
-    active_menu->Append( ID_FIRST,  "Activate f&irst layout \tF1", "activate it", TRUE );
-    active_menu->Append( ID_SECOND, "Activate &second layout\tF2","activate it",  TRUE );
-    active_menu->Append( ID_THIRD,  "Activate &third layout\tF3","activate it",   TRUE );
+    active_menu->Append( ID_FIRST,  _T("Activate f&irst layout \tF1"), _T("activate it"), TRUE );
+    active_menu->Append( ID_SECOND, _T("Activate &second layout\tF2"), _T("activate it"),  TRUE );
+    active_menu->Append( ID_THIRD,  _T("Activate &third layout\tF3"), _T("activate it"),   TRUE );
     
@@ -95,4 +95,4 @@
     
-    menu_bar->Append(file_menu,   "&File");
-    menu_bar->Append(active_menu, "Active &Layout");
+    menu_bar->Append(file_menu,   _T("&File"));
+    menu_bar->Append(active_menu, _T("Active &Layout"));
     
@@ -156,3 +156,3 @@
 
-MyFrame::MyFrame(wxFrame *frame, char *title, int x, int y, int w, int h)
+MyFrame::MyFrame(wxFrame *frame, wxChar *title, int x, int y, int w, int h)
     : wxFrame(frame, -1, title, wxPoint(x, y), wxSize(w, h)),
@@ -170,3 +170,3 @@
     
-    mAboutBox.Create( this, -1,  "About box in wxWindows style...",
+    mAboutBox.Create( this, -1,  _T("About box in wxWindows style..."),
                       wxDefaultPosition,
@@ -185,7 +185,7 @@
     
-    if ( wxFileExists( BMP_DIR "folder_icon.bmp" ) )
-        bmp1.LoadFile( BMP_DIR "folder_icon.bmp", wxBITMAP_TYPE_BMP );
+    if ( wxFileExists( _T(BMP_DIR "folder_icon.bmp") ) )
+        bmp1.LoadFile( _T(BMP_DIR "folder_icon.bmp"), wxBITMAP_TYPE_BMP );
     
-    if ( wxFileExists( BMP_DIR "class_icon1.bmp" ) )
-        bmp2.LoadFile( BMP_DIR "class_icon1.bmp", wxBITMAP_TYPE_BMP );
+    if ( wxFileExists( _T(BMP_DIR "class_icon1.bmp") ) )
+        bmp2.LoadFile( _T(BMP_DIR "class_icon1.bmp"), wxBITMAP_TYPE_BMP );
     
@@ -200,3 +200,3 @@
     
-    mpClntWindow = CreateTxtCtrl("client window");
+    mpClntWindow = CreateTxtCtrl(_T("client window"));
     
@@ -242,13 +242,13 @@
 
-void MyFrame::OnLoad( wxCommandEvent& event )
+void MyFrame::OnLoad( wxCommandEvent& WXUNUSED(event) )
 {
-    wxMessageBox("Hey - you found a BIG question-mark !!");
+    wxMessageBox(_T("Hey - you found a BIG question-mark !!"));
 }
 
-void MyFrame::OnStore( wxCommandEvent& event )
+void MyFrame::OnStore( wxCommandEvent& WXUNUSED(event) )
 {
-    wxMessageBox("Hey - you found another BIG question-mark !!");
+    wxMessageBox(_T("Hey - you found another BIG question-mark !!"));
 }
 
-void MyFrame::OnAutoSave( wxCommandEvent& event )
+void MyFrame::OnAutoSave( wxCommandEvent& WXUNUSED(event) )
 {
@@ -262,3 +262,3 @@
 
-void MyFrame::OnRemove( wxCommandEvent& event )
+void MyFrame::OnRemove( wxCommandEvent& WXUNUSED(event) )
 {
@@ -269,3 +269,3 @@
 
-void MyFrame::OnRemoveAll( wxCommandEvent& event )
+void MyFrame::OnRemoveAll( wxCommandEvent& WXUNUSED(event) )
 {
@@ -289,3 +289,3 @@
 
-void MyFrame::OnFirst( wxCommandEvent& event )
+void MyFrame::OnFirst( wxCommandEvent& WXUNUSED(event) )
 {
@@ -294,3 +294,3 @@
 
-void MyFrame::OnSecond( wxCommandEvent& event )
+void MyFrame::OnSecond( wxCommandEvent& WXUNUSED(event) )
 {
@@ -299,3 +299,3 @@
 
-void MyFrame::OnThird( wxCommandEvent& event )
+void MyFrame::OnThird( wxCommandEvent& WXUNUSED(event) )
 {
@@ -304,3 +304,3 @@
 
-void MyFrame::OnQuit( wxCommandEvent& event )
+void MyFrame::OnQuit( wxCommandEvent& WXUNUSED(event) )
 {
@@ -323,3 +323,3 @@
 
-void MyFrame::OnAbout( wxCommandEvent& event )
+void MyFrame::OnAbout( wxCommandEvent& WXUNUSED(event) )
 {
@@ -327,3 +327,3 @@
 #ifdef __WXMSW__
-    font.SetFaceName("MS Sans Serif");
+    font.SetFaceName(_T("MS Sans Serif"));
 #else
@@ -368,3 +368,3 @@
                 // "AI" :-)
-                wxMessageBox("There are only 3 layouts in this demo :-(");
+                wxMessageBox(_T("There are only 3 layouts in this demo :-("));
             }
@@ -399,8 +399,8 @@
 
-void MyFrame::OnSayItsOk( wxCommandEvent& event )
+void MyFrame::OnSayItsOk( wxCommandEvent& WXUNUSED(event) )
 {
-    wxMessageBox("It's OK  :-)\n\n now click on the border around the button\n and try dragging it!" );
+    wxMessageBox(_T("It's OK  :-)\n\n now click on the border around the button\n and try dragging it!") );
 }
 
-void MyFrame::OnBtnYes( wxCommandEvent& event )
+void MyFrame::OnBtnYes( wxCommandEvent& WXUNUSED(event) )
 {
@@ -409,3 +409,3 @@
 
-void MyFrame::OnBtnNo( wxCommandEvent& event )
+void MyFrame::OnBtnNo( wxCommandEvent& WXUNUSED(event) )
 {
@@ -414,3 +414,3 @@
 
-void MyFrame::OnBtnEsc( wxCommandEvent& event )
+void MyFrame::OnBtnEsc( wxCommandEvent& WXUNUSED(event) )
 {
@@ -427,9 +427,9 @@
     
-    new wxStaticText(pArea, -1, "This is wxFrameLayout contribution demo.",
+    new wxStaticText(pArea, -1, _T("This is wxFrameLayout contribution demo."),
         wxPoint(10, 10) );
     
-    new wxStaticText(pArea, -1, "Aleksandras Gluchovas (c) 1998",
+    new wxStaticText(pArea, -1, _T("Aleksandras Gluchovas (c) 1998"),
         wxPoint(10, 30) );
     
-    new wxStaticText(pArea, -1, "<mailto:alex@soften.ktu.lt>",
+    new wxStaticText(pArea, -1, _T("<mailto:alex@soften.ktu.lt>"),
         wxPoint(10, 50) );
@@ -447,9 +447,9 @@
     
-    wxButton* pYes = CreateButton("&Yes",   &mAboutBox, ID_SAY_ITSOK );
-    wxButton* pNo  = CreateButton("&No",    &mAboutBox, ID_BTN_NO );
-    wxButton* pEsc = CreateButton("Cancel", &mAboutBox, ID_BTN_ESC );
-    
-    layout.AddBar( pEsc, sizes,  FL_ALIGN_BOTTOM, 0, 20, "cancel button");
-    layout.AddBar( pNo,  sizes,  FL_ALIGN_BOTTOM, 0, 20, "no button");
-    layout.AddBar( pYes, sizes,  FL_ALIGN_BOTTOM, 0, 20, "yes button");
+    wxButton* pYes = CreateButton(_T("&Yes"),   &mAboutBox, ID_SAY_ITSOK );
+    wxButton* pNo  = CreateButton(_T("&No"),    &mAboutBox, ID_BTN_NO );
+    wxButton* pEsc = CreateButton(_T("Cancel"), &mAboutBox, ID_BTN_ESC );
+    
+    layout.AddBar( pEsc, sizes,  FL_ALIGN_BOTTOM, 0, 20, _T("cancel button"));
+    layout.AddBar( pNo,  sizes,  FL_ALIGN_BOTTOM, 0, 20, _T("no button"));
+    layout.AddBar( pYes, sizes,  FL_ALIGN_BOTTOM, 0, 20, _T("yes button"));
     
@@ -494,4 +494,4 @@
     {
-        pTree->AppendItem(rootid, "Leaf1", 0);
-        pTree->AppendItem(rootid, "Leaf2", 0);
+        pTree->AppendItem(rootid, _T("Leaf1"), 0);
+        pTree->AppendItem(rootid, _T("Leaf2"), 0);
     }
@@ -499,4 +499,4 @@
     {
-        pTree->AppendItem(rootid, "Scully", 0);
-        pTree->AppendItem(rootid, "Mulder", 0);
+        pTree->AppendItem(rootid, _T("Scully"), 0);
+        pTree->AppendItem(rootid, _T("Mulder"), 0);
     }
@@ -511,6 +511,6 @@
     choice_strings[0] = txt;
-    choice_strings[1] = "Julian";
-    choice_strings[2] = "Hattie";
-    choice_strings[3] = "Ken";
-    choice_strings[4] = "Dick";
+    choice_strings[1] = _T("Julian");
+    choice_strings[2] = _T("Hattie");
+    choice_strings[3] = _T("Ken");
+    choice_strings[4] = _T("Dick");
     
@@ -526,3 +526,3 @@
 
-void MyFrame::AddSearchToolbars( wxFrameLayout& layout, wxWindow* pParent )
+void MyFrame::AddSearchToolbars( wxFrameLayout& layout, wxWindow* WXUNUSED(pParent) )
 {
@@ -560,7 +560,7 @@
     pTBar2->AddTool( 1, pChoice );
-    pTBar2->AddTool( 2, BMP_DIR "search.bmp" );
+    pTBar2->AddTool( 2, _T(BMP_DIR "search.bmp") );
     //pTBar2->AddSeparator();
-    pTBar2->AddTool( 3, BMP_DIR "bookmarks.bmp" );
-    pTBar2->AddTool( 4, BMP_DIR "nextmark.bmp" );
-    pTBar2->AddTool( 5, BMP_DIR "prevmark.bmp" );
+    pTBar2->AddTool( 3, _T(BMP_DIR "bookmarks.bmp") );
+    pTBar2->AddTool( 4, _T(BMP_DIR "nextmark.bmp") );
+    pTBar2->AddTool( 5, _T(BMP_DIR "prevmark.bmp") );
     
@@ -568,9 +568,9 @@
     
-    pTBar3->AddTool( 1, BMP_DIR "open.bmp", wxBITMAP_TYPE_BMP, " Open " );
-    pTBar3->AddTool( 2, BMP_DIR "save.bmp", wxBITMAP_TYPE_BMP, " Save " );
-    pTBar3->AddTool( 3, BMP_DIR "saveall.bmp", wxBITMAP_TYPE_BMP, " Save All " );
+    pTBar3->AddTool( 1, _T(BMP_DIR "open.bmp"), wxBITMAP_TYPE_BMP, _T(" Open ") );
+    pTBar3->AddTool( 2, _T(BMP_DIR "save.bmp"), wxBITMAP_TYPE_BMP, _T(" Save ") );
+    pTBar3->AddTool( 3, _T(BMP_DIR "saveall.bmp"), wxBITMAP_TYPE_BMP, _T(" Save All ") );
     //pTBar3->AddSeparator();
-    pTBar3->AddTool( 4, BMP_DIR "cut.bmp",   wxBITMAP_TYPE_BMP, " Open " );
-    pTBar3->AddTool( 5, BMP_DIR "copy.bmp",  wxBITMAP_TYPE_BMP, " Copy " );
-    pTBar3->AddTool( 6, BMP_DIR "paste.bmp", wxBITMAP_TYPE_BMP, " Paste " );
+    pTBar3->AddTool( 4, _T(BMP_DIR "cut.bmp"),   wxBITMAP_TYPE_BMP, _T(" Open ") );
+    pTBar3->AddTool( 5, _T(BMP_DIR "copy.bmp"),  wxBITMAP_TYPE_BMP, _T(" Copy ") );
+    pTBar3->AddTool( 6, _T(BMP_DIR "paste.bmp"), wxBITMAP_TYPE_BMP, _T(" Paste ") );
     
@@ -582,7 +582,7 @@
     
-    pTBar4->AddTool( 1, BMP_DIR "bookmarks.bmp", wxBITMAP_TYPE_BMP, "Bookmarks ", TRUE );
-    pTBar4->AddTool( 2, BMP_DIR "nextmark.bmp",  wxBITMAP_TYPE_BMP, "Next bookmark ", TRUE );
-    pTBar4->AddTool( 3, BMP_DIR "prevmark.bmp",  wxBITMAP_TYPE_BMP, "Prev bookmark ", TRUE );
+    pTBar4->AddTool( 1, _T(BMP_DIR "bookmarks.bmp"), wxBITMAP_TYPE_BMP, _T("Bookmarks "), TRUE );
+    pTBar4->AddTool( 2, _T(BMP_DIR "nextmark.bmp"),  wxBITMAP_TYPE_BMP, _T("Next bookmark "), TRUE );
+    pTBar4->AddTool( 3, _T(BMP_DIR "prevmark.bmp"),  wxBITMAP_TYPE_BMP, _T("Prev bookmark "), TRUE );
     //pTBar4->AddSeparator();
-    pTBar4->AddTool( 4, BMP_DIR "search.bmp", wxBITMAP_TYPE_BMP, "Search ", TRUE );
+    pTBar4->AddTool( 4, _T(BMP_DIR "search.bmp"), wxBITMAP_TYPE_BMP, _T("Search "), TRUE );
     
@@ -596,3 +596,3 @@
                    0,                
-                   "Search",       
+                   _T("Search"),       
                    TRUE
@@ -604,3 +604,3 @@
                    0,                
-                   "Titled",       
+                   _T("Titled"),       
                    TRUE
@@ -612,3 +612,3 @@
                    0,                
-                   "Bookmarks",       
+                   _T("Bookmarks"),       
                    TRUE
@@ -666,6 +666,6 @@
     
-    wxTreeItemId rootId = pClassView->AddRoot( "wxWindows 2.0 classes", 0 );
+    wxTreeItemId rootId = pClassView->AddRoot( _T("wxWindows 2.0 classes"), 0 );
     
-    pClassView->AppendItem( rootId, "wxWin Dynamic classes (grabbed at run-time)", 0 );
-    pClassView->AppendItem( rootId, "serializer-classes (grabbed at run-time)",    0 );
+    pClassView->AppendItem( rootId, _T("wxWin Dynamic classes (grabbed at run-time)"), 0 );
+    pClassView->AppendItem( rootId, _T("serializer-classes (grabbed at run-time)"),    0 );
     
@@ -681,10 +681,10 @@
     
-    pTabbedArea->AddPage( CreateTxtCtrl("build", pTabbedArea), "Build");
-    pTabbedArea->AddPage( CreateTxtCtrl("debug", pTabbedArea), "Debug");
-    pTabbedArea->AddPage( pSheet3, "is THIS recursive - or what !?");
-    pTabbedArea->AddPage( CreateTxtCtrl("profile", pTabbedArea), "Profile");
-    
-    layout.AddBar( new StartButton95(pParent), sizes5, FL_ALIGN_TOP, 0, 0, "Start..." );
-    layout.AddBar( pClassView, sizes4, FL_ALIGN_LEFT, 0, 0, "Project Workplace" );
-    layout.AddBar( pTabbedArea, sizes4, FL_ALIGN_BOTTOM, 0, 50, "Output" );
+    pTabbedArea->AddPage( CreateTxtCtrl(_T("build"), pTabbedArea), _T("Build"));
+    pTabbedArea->AddPage( CreateTxtCtrl(_T("debug"), pTabbedArea), _T("Debug"));
+    pTabbedArea->AddPage( pSheet3, _T("is THIS recursive - or what !?"));
+    pTabbedArea->AddPage( CreateTxtCtrl(_T("profile"), pTabbedArea), _T("Profile"));
+    
+    layout.AddBar( new StartButton95(pParent), sizes5, FL_ALIGN_TOP, 0, 0, _T("Start...") );
+    layout.AddBar( pClassView, sizes4, FL_ALIGN_LEFT, 0, 0, _T("Project Workplace") );
+    layout.AddBar( pTabbedArea, sizes4, FL_ALIGN_BOTTOM, 0, 50, _T("Output") );
     
@@ -752,7 +752,7 @@
         
-        layout.AddBar( pGreenOne,                      sizes,  FL_ALIGN_TOP,         0, 50, "Bar1", TRUE );
-        layout.AddBar( new MyTestPanel(mpInternalFrm), sizes,  FL_ALIGN_TOP,         2, 50, "Bar2", TRUE );
-        layout.AddBar( new MyTestPanel(mpInternalFrm), sizes,  FL_ALIGN_BOTTOM,      2, 50, "Bar3", TRUE );
-        layout.AddBar( new MyTestPanel(mpInternalFrm), sizes,  FL_ALIGN_LEFT,        2, 50, "Bar4", TRUE );
-        layout.AddBar( new MyTestPanel(mpInternalFrm), sizes1, wxCBAR_HIDDEN, 2, 50, "Super-Bar", TRUE );
+        layout.AddBar( pGreenOne,                      sizes,  FL_ALIGN_TOP,         0, 50, _T("Bar1"), TRUE );
+        layout.AddBar( new MyTestPanel(mpInternalFrm), sizes,  FL_ALIGN_TOP,         2, 50, _T("Bar2"), TRUE );
+        layout.AddBar( new MyTestPanel(mpInternalFrm), sizes,  FL_ALIGN_BOTTOM,      2, 50, _T("Bar3"), TRUE );
+        layout.AddBar( new MyTestPanel(mpInternalFrm), sizes,  FL_ALIGN_LEFT,        2, 50, _T("Bar4"), TRUE );
+        layout.AddBar( new MyTestPanel(mpInternalFrm), sizes1, wxCBAR_HIDDEN, 2, 50, _T("Super-Bar"), TRUE );
     }
@@ -764,11 +764,11 @@
             
-            layout.AddBar( CreateTxtCtrl(),           sizes,  FL_ALIGN_TOP,    0, 50,  "Fixed text Area&0"     );
-            layout.AddBar( CreateButton("OK"),        sizes,  FL_ALIGN_TOP,    0, 100, "First Button"    );
-            layout.AddBar( CreateTxtCtrl(),           sizes1, FL_ALIGN_BOTTOM, 0, 50,  "First Tree"  );
-            layout.AddBar( CreateTreeCtrl("Root"),    sizes1, FL_ALIGN_LEFT,   0, 0,   "TreeCtrl Window" );
-            layout.AddBar( CreateChoice("Choice 1"),  sizes3, FL_ALIGN_TOP,    0, 0,   "Choice 1 (buggy)", FALSE, wxCBAR_HIDDEN );
-            layout.AddBar( CreateChoice("Choice 2"),  sizes3, FL_ALIGN_TOP,    0, 0,   "Choice 2 (buggy)", FALSE, wxCBAR_HIDDEN );
-            layout.AddBar( CreateTreeCtrl("X-Files"), sizes1, FL_ALIGN_RIGHT,  0, 100, "X-Files" );
-            layout.AddBar( CreateTxtCtrl("smaller1"), sizes3, FL_ALIGN_TOP,    0, 50,  "smaller Area1"     );
-            layout.AddBar( CreateTxtCtrl("smaller2"), sizes3, FL_ALIGN_TOP,    0, 50,  "sm&ller Area2"     );
+            layout.AddBar( CreateTxtCtrl(),               sizes,  FL_ALIGN_TOP,    0, 50,  _T("Fixed text Area&0")     );
+            layout.AddBar( CreateButton(_T("OK")),        sizes,  FL_ALIGN_TOP,    0, 100, _T("First Button")    );
+            layout.AddBar( CreateTxtCtrl(),               sizes1, FL_ALIGN_BOTTOM, 0, 50,  _T("First Tree")  );
+            layout.AddBar( CreateTreeCtrl(_T("Root")),    sizes1, FL_ALIGN_LEFT,   0, 0,   _T("TreeCtrl Window") );
+            layout.AddBar( CreateChoice(_T("Choice 1")),  sizes3, FL_ALIGN_TOP,    0, 0,   _T("Choice 1 (buggy)"), FALSE, wxCBAR_HIDDEN );
+            layout.AddBar( CreateChoice(_T("Choice 2")),  sizes3, FL_ALIGN_TOP,    0, 0,   _T("Choice 2 (buggy)"), FALSE, wxCBAR_HIDDEN );
+            layout.AddBar( CreateTreeCtrl(_T("X-Files")), sizes1, FL_ALIGN_RIGHT,  0, 100, _T("X-Files") );
+            layout.AddBar( CreateTxtCtrl(_T("smaller1")), sizes3, FL_ALIGN_TOP,    0, 50,  _T("smaller Area1")     );
+            layout.AddBar( CreateTxtCtrl(_T("smaller2")), sizes3, FL_ALIGN_TOP,    0, 50,  _T("sm&ller Area2")     );
         }
@@ -785,9 +785,9 @@
                 
-                layout.AddBar( CreateTxtCtrl("Tool1"), sizes3,  FL_ALIGN_TOP,  0, 50,  "Fixed text Area1" );
-                layout.AddBar( CreateTxtCtrl("Tool2"), sizes3,  FL_ALIGN_TOP,  0, 50,  "Fixed text Area2" );
-                layout.AddBar( CreateTxtCtrl("Tool3"), sizes3,  FL_ALIGN_TOP,  0, 50,  "Fixed text Area3" );
-                layout.AddBar( CreateTxtCtrl("Tool4"), sizes3,  FL_ALIGN_TOP,  1, 50,  "Fixed text Area4" );
-                layout.AddBar( CreateTxtCtrl("Tool5"), sizes3,  FL_ALIGN_TOP,  1, 50,  "Fixed text Area5" );
-                layout.AddBar( CreateTxtCtrl("Tool6"), sizes3,  FL_ALIGN_TOP,  1, 50,  "Fixed text Area6" );
-                layout.AddBar( CreateTxtCtrl("Tool7"), sizes3,  FL_ALIGN_TOP,  2,250,  "Fixed text Area7" );
+                layout.AddBar( CreateTxtCtrl(_T("Tool1")), sizes3,  FL_ALIGN_TOP,  0, 50,  _T("Fixed text Area1") );
+                layout.AddBar( CreateTxtCtrl(_T("Tool2")), sizes3,  FL_ALIGN_TOP,  0, 50,  _T("Fixed text Area2") );
+                layout.AddBar( CreateTxtCtrl(_T("Tool3")), sizes3,  FL_ALIGN_TOP,  0, 50,  _T("Fixed text Area3") );
+                layout.AddBar( CreateTxtCtrl(_T("Tool4")), sizes3,  FL_ALIGN_TOP,  1, 50,  _T("Fixed text Area4") );
+                layout.AddBar( CreateTxtCtrl(_T("Tool5")), sizes3,  FL_ALIGN_TOP,  1, 50,  _T("Fixed text Area5") );
+                layout.AddBar( CreateTxtCtrl(_T("Tool6")), sizes3,  FL_ALIGN_TOP,  1, 50,  _T("Fixed text Area6") );
+                layout.AddBar( CreateTxtCtrl(_T("Tool7")), sizes3,  FL_ALIGN_TOP,  2,250,  _T("Fixed text Area7") );
                 
@@ -808,9 +808,9 @@
                 
-                pToolBar->AddTool( 1001, BMP_DIR "new.bmp" );
-                pToolBar->AddTool( 1002, BMP_DIR "open.bmp" );
-                pToolBar->AddTool( 1003, BMP_DIR "save.bmp" );
-                
-                pToolBar->AddTool( 1004, BMP_DIR "cut.bmp" );
-                pToolBar->AddTool( 1005, BMP_DIR "copy.bmp" );
-                pToolBar->AddTool( 1006, BMP_DIR "paste.bmp" );
+                pToolBar->AddTool( 1001, _T(BMP_DIR "new.bmp") );
+                pToolBar->AddTool( 1002, _T(BMP_DIR "open.bmp") );
+                pToolBar->AddTool( 1003, _T(BMP_DIR "save.bmp") );
+                
+                pToolBar->AddTool( 1004, _T(BMP_DIR "cut.bmp") );
+                pToolBar->AddTool( 1005, _T(BMP_DIR "copy.bmp") );
+                pToolBar->AddTool( 1006, _T(BMP_DIR "paste.bmp") );
                 
@@ -820,3 +820,3 @@
                                0,                     // offset from the start of row (in pixels)
-                               "Real-Toolbar",        // name to refere in customization pop-ups
+                               _T("Real-Toolbar"),    // name to refere in customization pop-ups
                                FALSE
@@ -833,3 +833,3 @@
                 mpNestedLayout = new wxFrameLayout( pSheet3, 
-                    CreateTxtCtrl("\"Mobils in Mobile\" --C.Nemo",pSheet3), FALSE );
+                    CreateTxtCtrl(_T("\"Mobils in Mobile\" --C.Nemo"),pSheet3), FALSE );
                 
@@ -929,3 +929,3 @@
 
-void StartButton95::OnMouseDown( wxMouseEvent& event )
+void StartButton95::OnMouseDown( wxMouseEvent& WXUNUSED(event) )
 {
@@ -936,3 +936,3 @@
 
-void StartButton95::OnMouseUp( wxMouseEvent& event )
+void StartButton95::OnMouseUp( wxMouseEvent& WXUNUSED(event) )
 {
@@ -956,3 +956,3 @@
 
-void StartButton95::OnPaint( wxPaintEvent& event )
+void StartButton95::OnPaint( wxPaintEvent& WXUNUSED(event) )
 {
@@ -962,5 +962,5 @@
     {
-        if ( !m_PBmp.Ok() && wxFileExists( BMP_DIR "start95_pr.bmp" ) )
+        if ( !m_PBmp.Ok() && wxFileExists( _T(BMP_DIR "start95_pr.bmp") ) )
             
-            m_PBmp.LoadFile( BMP_DIR "start95_pr.bmp", wxBITMAP_TYPE_BMP );
+            m_PBmp.LoadFile( _T(BMP_DIR "start95_pr.bmp"), wxBITMAP_TYPE_BMP );
         
@@ -970,5 +970,5 @@
     {
-        if ( !m_DBmp.Ok() && wxFileExists( BMP_DIR "start95_dp.bmp" ) )
+        if ( !m_DBmp.Ok() && wxFileExists( _T(BMP_DIR "start95_dp.bmp") ) )
             
-            m_DBmp.LoadFile( BMP_DIR "start95_dp.bmp", wxBITMAP_TYPE_BMP );
+            m_DBmp.LoadFile( _T(BMP_DIR "start95_dp.bmp"), wxBITMAP_TYPE_BMP );
         
diff -bu1 -r wxWindows/contrib\samples\fl/fl_demo2.h wxWin/contrib\samples\fl/fl_demo2.h
--- wxWindows/contrib\samples\fl/fl_demo2.h	2003-08-03 20:29:53.000000000 +0200
+++ wxWin/contrib\samples\fl/fl_demo2.h	2003-08-22 13:26:17.000000000 +0200
@@ -82,6 +82,6 @@
     
-    wxTextCtrl* CreateTxtCtrl ( const wxString& txt = "wxTextCtrl", wxWindow* parent = NULL  );
-    wxTreeCtrl* CreateTreeCtrl( const wxString& label = "TreeCtrl" );
-    wxChoice*   CreateChoice  ( const wxString& txt = "Choice1" );
-    wxButton*   CreateButton  ( const wxString& label = "wxButton", wxWindow* pParent = NULL, long id = ID_SAY_ITSOK );
+    wxTextCtrl* CreateTxtCtrl ( const wxString& txt = _T("wxTextCtrl"), wxWindow* parent = NULL  );
+    wxTreeCtrl* CreateTreeCtrl( const wxString& label = _T("TreeCtrl") );
+    wxChoice*   CreateChoice  ( const wxString& txt = _T("Choice1") );
+    wxButton*   CreateButton  ( const wxString& label = _T("wxButton"), wxWindow* pParent = NULL, long id = ID_SAY_ITSOK );
     
@@ -102,3 +102,3 @@
     
-    MyFrame( wxFrame *frame, char *title, int x, int y, int w, int h);
+    MyFrame( wxFrame *frame, wxChar *title, int x, int y, int w, int h);
     
diff -bu1 -r wxWindows/contrib\samples\fl/fl_sample1.cpp wxWin/contrib\samples\fl/fl_sample1.cpp
--- wxWindows/contrib\samples\fl/fl_sample1.cpp	2003-08-03 20:29:53.000000000 +0200
+++ wxWin/contrib\samples\fl/fl_sample1.cpp	2003-08-22 14:22:02.000000000 +0200
@@ -51,3 +51,3 @@
 public:
-    MyFrame( wxWindow* parent, char *title );
+    MyFrame( wxWindow* parent, wxChar *title );
     ~MyFrame();
@@ -71,3 +71,3 @@
     
-    MyFrame *frame = new MyFrame(NULL, "wxFrameLayout sample");
+    MyFrame *frame = new MyFrame(NULL, _T("wxFrameLayout sample"));
     
@@ -75,7 +75,7 @@
     
-    file_menu->Append( ID_LOAD,  "&Load layout"  );
-    file_menu->Append( ID_STORE, "&Store layout"  );
+    file_menu->Append( ID_LOAD,  _T("&Load layout")  );
+    file_menu->Append( ID_STORE, _T("&Store layout")  );
     file_menu->AppendSeparator();
     
-    file_menu->Append( ID_QUIT, "E&xit" );
+    file_menu->Append( ID_QUIT, _T("E&xit") );
     
@@ -83,3 +83,3 @@
     
-    menu_bar->Append(file_menu, "&File");
+    menu_bar->Append(file_menu, _T("&File"));
     
@@ -103,8 +103,8 @@
 
-MyFrame::MyFrame( wxWindow* parent, char *title )
-    : wxFrame( parent, -1, "NewTest-II", wxDefaultPosition,
-          wxSize( 700, 500 ),
+MyFrame::MyFrame( wxWindow* parent, wxChar *WXUNUSED(title) )
+    : wxFrame( parent, -1, _T("NewTest-II"), // should be title overwritten for sure ?
+          wxDefaultPosition, wxSize( 700, 500 ),
           wxCLIP_CHILDREN | wxMINIMIZE_BOX | wxMAXIMIZE_BOX |
           wxTHICK_FRAME   | wxSYSTEM_MENU  | wxCAPTION,
-          "freimas" )
+          _T("freimas") )
 {
@@ -112,3 +112,3 @@
     
-    mpClientWnd = CreateTextCtrl( "Client window" );
+    mpClientWnd = CreateTextCtrl( _T("Client window") );
     
@@ -144,5 +144,5 @@
     {
-        char buf[4];
-        sprintf( buf, "%d", i );
-        wxString name = wxString("Bar-");
+        wxChar buf[4];
+        wxSprintf( buf, _T("%d"), i );
+        wxString name = wxString(_T("Bar-"));
         name += buf;
@@ -151,3 +151,3 @@
         
-        if ( !sizes.mIsFixed ) name += " (flexible)";
+        if ( !sizes.mIsFixed ) name += _T(" (flexible)");
         
@@ -179,13 +179,13 @@
 
-void MyFrame::OnLoad( wxCommandEvent& event )
+void MyFrame::OnLoad( wxCommandEvent& WXUNUSED(event) )
 {
-    wxMessageBox("Hey - you found a BIG question-mark !!");
+    wxMessageBox(_T("Hey - you found a BIG question-mark !!"));
 }
 
-void MyFrame::OnStore( wxCommandEvent& event )
+void MyFrame::OnStore( wxCommandEvent& WXUNUSED(event) )
 {
-    wxMessageBox("Hey - you found another BIG question-mark !!");
+    wxMessageBox(_T("Hey - you found another BIG question-mark !!"));
 }
 
-void MyFrame::OnQuit( wxCommandEvent& event )
+void MyFrame::OnQuit( wxCommandEvent& WXUNUSED(event) )
 {
diff -bu1 -r wxWindows/contrib\samples\fl/fl_sample2.cpp wxWin/contrib\samples\fl/fl_sample2.cpp
--- wxWindows/contrib\samples\fl/fl_sample2.cpp	2003-08-03 20:29:53.000000000 +0200
+++ wxWin/contrib\samples\fl/fl_sample2.cpp	2003-08-22 14:32:00.000000000 +0200
@@ -51,3 +51,3 @@
 public:
-    MyFrame( wxWindow* parent, char *title );
+    MyFrame( wxWindow* parent, wxChar *title );
     ~MyFrame();
@@ -72,3 +72,3 @@
     
-    MyFrame *frame = new MyFrame(NULL, "wxFrameLayout sample");
+    MyFrame *frame = new MyFrame(NULL, _T("wxFrameLayout sample"));
     
@@ -76,7 +76,7 @@
     
-    file_menu->Append( ID_LOAD,  "&Load layout"  );
-    file_menu->Append( ID_STORE, "&Store layout"  );
+    file_menu->Append( ID_LOAD,  _T("&Load layout")  );
+    file_menu->Append( ID_STORE, _T("&Store layout")  );
     file_menu->AppendSeparator();
     
-    file_menu->Append( ID_QUIT, "E&xit" );
+    file_menu->Append( ID_QUIT, _T("E&xit") );
     
@@ -84,3 +84,3 @@
     
-    menu_bar->Append(file_menu, "&File");
+    menu_bar->Append(file_menu, _T("&File"));
     
@@ -104,8 +104,8 @@
 
-MyFrame::MyFrame( wxWindow* parent, char *title )
-    : wxFrame( parent, -1, "NewTest-II", wxDefaultPosition,
-          wxSize( 700, 500 ),
+MyFrame::MyFrame( wxWindow* parent, wxChar *WXUNUSED(title) )
+    : wxFrame( parent, -1, _T("NewTest-II"), // should be title overwritten for sure ?
+          wxDefaultPosition, wxSize( 700, 500 ),
           wxCLIP_CHILDREN | wxMINIMIZE_BOX | wxMAXIMIZE_BOX |
           wxTHICK_FRAME   | wxSYSTEM_MENU  | wxCAPTION,
-          "freimas" )
+          _T("freimas") )
 {
@@ -115,3 +115,3 @@
 {
-    mpClientWnd = CreateTextCtrl( "Client window" );
+    mpClientWnd = CreateTextCtrl( _T("Client window") );
     
@@ -137,5 +137,5 @@
         
-        char buf[4];
-        sprintf( buf, "%d", i );
-        wxString name = wxString("Bar-");
+        wxChar buf[4];
+        wxSprintf( buf, _T("%d"), i );
+        wxString name = wxString(_T("Bar-"));
         name += buf;
@@ -144,3 +144,3 @@
         
-        if ( !sizes.mIsFixed ) name += " (flexible)";
+        if ( !sizes.mIsFixed ) name += _T(" (flexible)");
         //      mpLayout->AddBar( CreateTextCtrl(name),// bar window
@@ -177,13 +177,13 @@
 
-void MyFrame::OnLoad( wxCommandEvent& event )
+void MyFrame::OnLoad( wxCommandEvent& WXUNUSED(event) )
 {
-    wxMessageBox("Hey - you found a BIG question-mark !!");
+    wxMessageBox(_T("Hey - you found a BIG question-mark !!"));
 }
 
-void MyFrame::OnStore( wxCommandEvent& event )
+void MyFrame::OnStore( wxCommandEvent& WXUNUSED(event) )
 {
-    wxMessageBox("Hey - you found another BIG question-mark !!");
+    wxMessageBox(_T("Hey - you found another BIG question-mark !!"));
 }
 
-void MyFrame::OnQuit( wxCommandEvent& event )
+void MyFrame::OnQuit( wxCommandEvent& WXUNUSED(event) )
 {
diff -bu1 -r wxWindows/contrib\samples\fl/fl_sample3.cpp wxWin/contrib\samples\fl/fl_sample3.cpp
--- wxWindows/contrib\samples\fl/fl_sample3.cpp	2003-08-03 20:29:53.000000000 +0200
+++ wxWin/contrib\samples\fl/fl_sample3.cpp	2003-08-22 14:55:31.000000000 +0200
@@ -54,3 +54,3 @@
 public:
-    MyFrame( wxWindow* parent, char *title );
+    MyFrame( wxWindow* parent, wxChar *title );
     ~MyFrame();
@@ -76,3 +76,3 @@
     
-    MyFrame *frame = new MyFrame(NULL, "wxFrameLayout sample");
+    MyFrame *frame = new MyFrame(NULL, _T("wxFrameLayout sample"));
     
@@ -80,7 +80,7 @@
     
-    file_menu->Append( ID_LOAD,  "&Load layout"  );
-    file_menu->Append( ID_STORE, "&Store layout"  );
+    file_menu->Append( ID_LOAD,  _T("&Load layout")  );
+    file_menu->Append( ID_STORE, _T("&Store layout")  );
     file_menu->AppendSeparator();
     
-    file_menu->Append( ID_QUIT, "E&xit" );
+    file_menu->Append( ID_QUIT, _T("E&xit") );
     
@@ -88,3 +88,3 @@
     
-    menu_bar->Append(file_menu, "&File");
+    menu_bar->Append(file_menu, _T("&File"));
     
@@ -109,8 +109,8 @@
 
-MyFrame::MyFrame( wxWindow* parent, char *title )
-    : wxFrame( parent, -1, "NewTest-II", wxDefaultPosition,
-          wxSize( 700, 500 ),
+MyFrame::MyFrame( wxWindow* parent, wxChar *WXUNUSED(title) )
+    : wxFrame( parent, -1, _T("NewTest-II"), // should be title overwritten for sure ?
+          wxDefaultPosition, wxSize( 700, 500 ),
           wxCLIP_CHILDREN | wxMINIMIZE_BOX | wxMAXIMIZE_BOX |
           wxTHICK_FRAME   | wxSYSTEM_MENU  | wxCAPTION,
-          "freimas" )
+          _T("freimas") )
 {
@@ -137,5 +137,5 @@
     {
-        char buf[4];
-        sprintf( buf, "%d", i );
-        wxString name = wxString("Bar-");
+        wxChar buf[4];
+        wxSprintf( buf, _T("%d"), i );
+        wxString name = wxString(_T("Bar-"));
         name += buf;
@@ -144,3 +144,3 @@
         
-        if ( !sizes.mIsFixed ) name += " (flexible)";
+        if ( !sizes.mIsFixed ) name += _T(" (flexible)");
         //      mpLayout->AddBar( CreateTextCtrl(name),// bar window
@@ -172,3 +172,3 @@
         } else if(i==11) {
-            mpLayout->AddBar( new wxButton(this, ID_BUTT2, name+"_2"),
+            mpLayout->AddBar( new wxButton(this, ID_BUTT2, name+_T("_2")),
                               cbDimInfo( 100,100, 200,200, 400,400, TRUE, 5, 5), 
@@ -204,13 +204,13 @@
 
-void MyFrame::OnLoad( wxCommandEvent& event )
+void MyFrame::OnLoad( wxCommandEvent& WXUNUSED(event) )
 {
-    wxMessageBox("Hey - you found a BIG question-mark !!");
+    wxMessageBox(_T("Hey - you found a BIG question-mark !!"));
 }
 
-void MyFrame::OnStore( wxCommandEvent& event )
+void MyFrame::OnStore( wxCommandEvent& WXUNUSED(event) )
 {
-    wxMessageBox("Hey - you found another BIG question-mark !!");
+    wxMessageBox(_T("Hey - you found another BIG question-mark !!"));
 }
 
-void MyFrame::OnQuit( wxCommandEvent& event )
+void MyFrame::OnQuit( wxCommandEvent& WXUNUSED(event) )
 {
@@ -221,3 +221,3 @@
 
-void MyFrame::OnButt( wxCommandEvent& event )
+void MyFrame::OnButt( wxCommandEvent& WXUNUSED(event) )
 {
@@ -230,3 +230,3 @@
         {
-            cbBarInfo* x = mpLayout->FindBarByName(wxString("Bar-1"));
+            cbBarInfo* x = mpLayout->FindBarByName(wxString(_T("Bar-1")));
             if(x) 
@@ -239,3 +239,3 @@
         {
-            cbBarInfo* x = mpLayout->FindBarByName(wxString("Bar-6"));
+            cbBarInfo* x = mpLayout->FindBarByName(wxString(_T("Bar-6")));
             if(x) 
@@ -268,3 +268,3 @@
 
-void MyFrame::OnButt2( wxCommandEvent& event )
+void MyFrame::OnButt2( wxCommandEvent& WXUNUSED(event) )
 {
@@ -277,3 +277,3 @@
         {
-            cbBarInfo* x = mpLayout->FindBarByName(wxString("Bar-1"));
+            cbBarInfo* x = mpLayout->FindBarByName(wxString(_T("Bar-1")));
             if(x) 
@@ -300,3 +300,3 @@
         {
-            cbBarInfo* x = mpLayout->FindBarByName(wxString("Bar-1"));
+            cbBarInfo* x = mpLayout->FindBarByName(wxString(_T("Bar-1")));
             if(x) 

 	  	 
