Index: src/common/valtext.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/common/valtext.cpp,v
retrieving revision 1.34
diff -u -r1.34 valtext.cpp
--- src/common/valtext.cpp	2003/05/23 18:59:05	1.34
+++ src/common/valtext.cpp	2003/06/27 18:45:18
@@ -1,4 +1,4 @@
-/////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////
 // Name:        valtext.cpp
 // Purpose:     wxTextValidator
 // Author:      Julian Smart
@@ -206,6 +206,10 @@
     if( !CheckValidator() )
         return FALSE;
 
+	if (!m_stringValue) {
+		return TRUE;
+   }
+
     wxTextCtrl *control = (wxTextCtrl *) m_validatorWindow ;
     control->SetValue(* m_stringValue) ;
 
@@ -215,8 +219,12 @@
 // Called to transfer data to the window
 bool wxTextValidator::TransferFromWindow(void)
 {
-    if( !CheckValidator() )
+   if( !CheckValidator() )
         return FALSE;
+
+   if (!m_stringValue) {	//silent failure if you didn't specify a variable
+	   return TRUE;
+   }
 
     wxTextCtrl *control = (wxTextCtrl *) m_validatorWindow ;
     * m_stringValue = control->GetValue() ;

 	  	 
