--- G:\cyberboard\cb4\wxWidgets\samples\drawing\drawing.orig.cpp	Sun Feb 15 20:28:53 2009
+++ G:\cyberboard\cb4\wxWidgets\samples\drawing\drawing.cpp	Mon Feb 16 00:53:13 2009
@@ -846,8 +846,25 @@
         memDC.SelectObject(*gs_bmpWithColMask);
         if ( mode == Draw_Stretch )
         {
+#if 0
             dc.StretchBlit(x, y, cx, cy, &memDC, 0, 0, cx/2, cy/2,
                            rasterOperations[n].rop, true);
+#else
+            /* the problem is with stretchblit when source and dest have
+                different sizes, and the source has no mask, and is a dib,
+                so create a dib-based bitmap that is the top left quadrant
+                of gs_bmpNoMask to demo the problem */
+            wxBitmap trouble(gs_bmpNoMask->GetWidth()/2, gs_bmpNoMask->GetWidth()/2);
+            wxMemoryDC source;
+            source.SelectObject(*gs_bmpNoMask);
+            wxMemoryDC dest;
+            dest.SelectObject(trouble);
+            dest.Blit(0, 0, trouble.GetWidth(), trouble.GetHeight(), &source, 0, 0);
+            trouble = wxBitmap(trouble.ConvertToImage());
+            memDC.SelectObject(trouble);
+            dc.StretchBlit(x, y, cx, cy, &memDC, 0, 0, trouble.GetWidth(), trouble.GetHeight(),
+                           rasterOperations[n].rop, false);
+#endif
         }
         else
         {
