Index: wxWidgets/include/wx/msw/checkbox.h
===================================================================
--- wxWidgets/include/wx/msw/checkbox.h	(revision 72524)
+++ wxWidgets/include/wx/msw/checkbox.h	(working copy)
@@ -52,15 +52,15 @@
     // returns true if the platform should explicitly apply a theme border
     virtual bool CanApplyThemeBorder() const { return false; }
 
+    // make the checkbox owner drawn or reset it to normal style
+    void MSWMakeOwnerDrawn(bool ownerDrawn);
+
 protected:
     virtual wxSize DoGetBestSize() const;
 
     virtual void DoSet3StateValue(wxCheckBoxState value);
     virtual wxCheckBoxState DoGet3StateValue() const;
 
-    // make the checkbox owner drawn or reset it to normal style
-    void MakeOwnerDrawn(bool ownerDrawn);
-
     // return true if this checkbox is owner drawn
     bool IsOwnerDrawn() const;
 
Index: wxWidgets/src/msw/checkbox.cpp
===================================================================
--- wxWidgets/src/msw/checkbox.cpp	(revision 72524)
+++ wxWidgets/src/msw/checkbox.cpp	(working copy)
@@ -273,7 +273,7 @@
     // the only way to change the checkbox foreground colour under Windows XP
     // is to owner draw it
     if ( wxUxThemeEngine::GetIfActive() )
-        MakeOwnerDrawn(colour.IsOk());
+        MSWMakeOwnerDrawn(colour.IsOk());
 
     return true;
 }
@@ -284,7 +284,7 @@
         (::GetWindowLong(GetHwnd(), GWL_STYLE) & BS_OWNERDRAW) == BS_OWNERDRAW;
 }
 
-void wxCheckBox::MakeOwnerDrawn(bool ownerDrawn)
+void wxCheckBox::MSWMakeOwnerDrawn(bool ownerDrawn)
 {
     long style = ::GetWindowLong(GetHwnd(), GWL_STYLE);
 
