? msw_thread.diff
Index: src/msw/thread.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/msw/thread.cpp,v
retrieving revision 1.108
diff -u -r1.108 thread.cpp
--- src/msw/thread.cpp	2007/03/16 00:56:26	1.108
+++ src/msw/thread.cpp	2007/06/06 12:01:33
@@ -996,7 +996,7 @@
     // set it: we can't link to SetProcessAffinityMask() because it doesn't
     // exist in Win9x, use RT binding instead
 
-    typedef BOOL (*SETPROCESSAFFINITYMASK)(HANDLE, DWORD);
+    typedef BOOL ( WINAPI *  SETPROCESSAFFINITYMASK)(HANDLE, DWORD_PTR);
 
     // can use static var because we're always in the main thread here
     static SETPROCESSAFFINITYMASK pfnSetProcessAffinityMask = NULL;

 	  	 
