Index: include/wx/os2/menuitem.h
===================================================================
--- include/wx/os2/menuitem.h	(revision 49825)
+++ include/wx/os2/menuitem.h	(working copy)
@@ -17,6 +17,7 @@
 // ----------------------------------------------------------------------------
 
 #include "wx/defs.h"
+#include "wx/os2/private.h"     // for MENUITEM
 
 // an exception to the general rule that a normal header doesn't include other
 // headers - only because ownerdrw.h is not always included and I don't want

 	  	 
