Index: minimal.cpp
===================================================================
--- minimal.cpp	(revision 65277)
+++ minimal.cpp	(working copy)
@@ -30,6 +30,8 @@
     #include "wx/wx.h"
 #endif
 
+#include "wx/radiobut.h"
+
 // ----------------------------------------------------------------------------
 // resources
 // ----------------------------------------------------------------------------
@@ -71,6 +73,8 @@
 private:
     // any class wishing to process wxWidgets events must use this macro
     DECLARE_EVENT_TABLE()
+
+    wxRadioButton* m_radio;
 };
 
 // ----------------------------------------------------------------------------
@@ -172,6 +176,9 @@
     CreateStatusBar(2);
     SetStatusText("Welcome to wxWidgets!");
 #endif // wxUSE_STATUSBAR
+
+    m_radio = new wxRadioButton(wxTheApp->GetTopWindow(), wxID_ANY, 
+                                "wxRadioButton");
 }
 
 
@@ -185,16 +192,5 @@
 
 void MyFrame::OnAbout(wxCommandEvent& WXUNUSED(event))
 {
-    wxMessageBox(wxString::Format
-                 (
-                    "Welcome to %s!\n"
-                    "\n"
-                    "This is the minimal wxWidgets sample\n"
-                    "running under %s.",
-                    wxVERSION_STRING,
-                    wxGetOsDescription()
-                 ),
-                 "About wxWidgets minimal sample",
-                 wxOK | wxICON_INFORMATION,
-                 this);
+    m_radio->SetValue(false);
 }
