--- floatspin.py.original	2011-10-19 16:37:22.000000000 -0400
+++ floatspin.py	2012-07-17 16:16:53.000000000 -0400
@@ -403,15 +403,6 @@
         # end Philip Semanchuk addition
 
         self.SetLabel(name)
-        # start Philip Semanchuk change
-        # Set the background color to match that of the parent. wxMac supports
-        # transparency, and that gives the best results on that platform.
-        # Other platforms ignore wx.ALPHA_TRANSPARENT.
-        colour = parent.GetBackgroundColour()
-        colour.Set(colour.Red(), colour.Green(), colour.Blue(), 
-                   wx.ALPHA_TRANSPARENT)
-        self.SetBackgroundColour(colour)
-        # end Philip Semanchuk change
         self.SetForegroundColour(parent.GetForegroundColour())
 
         width = size[0]
