diff -uNrb wxWidgets-trunk\src/msw/graphics.cpp wxWidgets-work\src/msw/graphics.cpp
--- wxWidgets-trunk\src/msw/graphics.cpp	2014-01-07
+++ wxWidgets-work\src/msw/graphics.cpp	2014-02-25
@@ -946,7 +946,9 @@
     if ( font.GetWeight() == wxFONTWEIGHT_BOLD )
         style |= FontStyleBold;
 
-    Init(font.GetFaceName(), font.GetPointSize(), style, col, UnitPoint);
+    // Create font which size is measured in logical units 
+    // and let the system rescale it according to the target resolution.
+    Init(font.GetFaceName(), font.GetPixelSize().GetHeight(), style, col, UnitPixel);
 }
 
 wxGDIPlusFontData::wxGDIPlusFontData(wxGraphicsRenderer* renderer,
