diff -uNrBb -x setup.h wxWidgets-trunk\include/wx/graphics.h wxWidgets-work\include/wx/graphics.h
--- wxWidgets-trunk\include/wx/graphics.h	2013-12-29
+++ wxWidgets-work\include/wx/graphics.h	2014-02-21
@@ -738,6 +738,9 @@
                                          wxDouble angle,
                                          const wxGraphicsBrush& backgroundBrush);
 
+private:
+    virtual wxDouble GetFontScale() const { return 1.0; }
+
     wxDECLARE_NO_COPY_CLASS(wxGraphicsContext);
     DECLARE_ABSTRACT_CLASS(wxGraphicsContext)
 };
diff -uNrb wxWidgets-trunk\src/common/graphcmn.cpp wxWidgets-work\src/common/graphcmn.cpp
--- wxWidgets-trunk\src/common/graphcmn.cpp	2013-12-29
+++ wxWidgets-work\src/common/graphcmn.cpp	2014-02-21
@@ -587,8 +587,11 @@
 
 void wxGraphicsContext::GetDPI( wxDouble* dpiX, wxDouble* dpiY)
 {
-    *dpiX = 72.0;
-    *dpiY = 72.0;
+    // DPI-Y is used for counting font scale in GDI+ text operations.
+    // Prinout is most similar to the screen view
+    // if reported DPI is the same as screen DPI.
+    *dpiX = 96.0;
+    *dpiY = 96.0;
 }
 
 // sets the pen
@@ -628,7 +631,17 @@
 void wxGraphicsContext::SetFont( const wxFont& font, const wxColour& colour )
 {
     if ( font.IsOk() )
+    {
+        if ( GetFontScale() != 1.0 )
+        {
+            wxFont fontTmp = font.Scaled(GetFontScale());
+            SetFont( CreateFont( fontTmp, colour ) );
+        }
+        else
+        {
         SetFont( CreateFont( font, colour ) );
+        }
+    }
     else
         SetFont( wxNullGraphicsFont );
 }
diff -uNrb wxWidgets-trunk\src/msw/graphics.cpp wxWidgets-work\src/msw/graphics.cpp
--- wxWidgets-trunk\src/msw/graphics.cpp	2014-01-07
+++ wxWidgets-work\src/msw/graphics.cpp	2014-02-21
@@ -429,6 +429,7 @@
 
 private:
     virtual void DoDrawText(const wxString& str, wxDouble x, wxDouble y);
+    virtual wxDouble GetFontScale() const { return 1.0 / m_fontScaleRatio; }
 
     Graphics* m_context;
     wxStack<GraphicsState> m_stateStack;
@@ -1954,7 +1955,9 @@
 
     // We use this modifier when measuring fonts. It is needed because the
     // page scale is modified above.
-    m_fontScaleRatio = context->GetDpiY() / 72.0;
+    double dpiX, dpiY;
+    GetDPI(&dpiX, &dpiY);
+    m_fontScaleRatio = context->GetDpiY() / dpiY;
 }
 
 //-----------------------------------------------------------------------------
