
C:\Code\wxWindows>cvs -z9 diff -u src\common\filefn.cpp 
Index: src/common/filefn.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/common/filefn.cpp,v
retrieving revision 1.160
diff -u -r1.160 filefn.cpp
--- src/common/filefn.cpp	2002/12/04 14:06:30	1.160
+++ src/common/filefn.cpp	2002/12/10 11:24:44
@@ -428,7 +428,11 @@
             if (p[1] == wxT('.') && p[2] == wxT('.') && (p[3] == SEP || p[3] == wxT('\0')))
               {
                 wxChar *q;
-                for (q = p - 1; q >= path && *q != SEP; q--);
+                for (q = p - 1; q >= path && *q != SEP; q--)
+                {
+                    // Empty
+                }
+
                 if (q[0] == SEP && (q[1] != wxT('.') || q[2] != wxT('.') || q[3] != SEP)
                     && (q - 1 <= path || q[-1] != SEP))
                   {
@@ -592,7 +596,11 @@
             *--d = 0;
             value = wxGetenv(braces ? start + 1 : start);
             if (value) {
-                for ((d = start - 1); (*d++ = *value++) != 0;);
+                for ((d = start - 1); (*d++ = *value++) != 0;)
+                {
+                    // Empty
+                }
+
                 d--;
                 if (braces && *s)
                     s++;
@@ -616,7 +624,10 @@
         {                /* ~user/filename */
             register wxChar  *nnm;
             register wxChar  *home;
-            for (s = nm; *s && *s != SEP; s++);
+            for (s = nm; *s && *s != SEP; s++)
+            {
+                // Empty
+            }
             int was_sep; /* MATTHEW: Was there a separator, or NULL? */
             was_sep = (*s == SEP);
             nnm = *s ? s + 1 : s;
@@ -643,7 +654,10 @@
           *(d - 1) = SEP;
     }
     s = nm;
-    while ((*d++ = *s++) != 0);
+    while ((*d++ = *s++) != 0)
+    {
+        // Empty
+    }
     delete[] nm_tmp; // clean up alloc
     /* Now clean up the buffer */
     return wxRealPath(buf);

 	  	 
