Index: src/msw/listctrl.cpp
===================================================================
--- src/msw/listctrl.cpp	(revision 64368)
+++ src/msw/listctrl.cpp	(working copy)
@@ -1564,8 +1564,15 @@
     // We shouldn't destroy the control ourselves according to MSDN, which
     // proposes WM_CANCELMODE to do this, but it doesn't seem to work so
     // emulate the corresponding user action instead.
-    ::SendMessage(hwnd, WM_KEYDOWN, cancel ? VK_ESCAPE : VK_RETURN, 0);
-
+    if (::IsWindowVisible(hwnd))
+    {
+        ::SendMessage(hwnd, WM_KEYDOWN, cancel ? VK_ESCAPE : VK_RETURN, 0);
+    }
+    else
+    {
+        // not ready for WM_KEYDOWN (in EVT_LIST_BEGIN_LABEL_EDIT)
+        ::SendMessage(hwnd, WM_CLOSE, 0, 0);
+    }
     return true;
 }
 
