Index: interface/wx/listctrl.h
===================================================================
--- interface/wx/listctrl.h	(revision 64368)
+++ interface/wx/listctrl.h	(working copy)
@@ -253,6 +253,8 @@
         This method allows to programmatically end editing a list control item
         in place. Usually it will only be called when editing is in progress,
         i.e. if GetEditControl() returns non-NULL.
+        Note: Calling EndEditLabel() from a EVT_LIST_BEGIN_LABEL_EDIT() handler
+        won't work, use event.Veto() instead.
 
         Currently only implemented in wxMSW.
 
Index: src/msw/listctrl.cpp
===================================================================
--- src/msw/listctrl.cpp	(revision 64368)
+++ src/msw/listctrl.cpp	(working copy)
@@ -1561,9 +1561,19 @@
     if ( !hwnd )
         return false;
 
+#ifdef ListView_CancelEditLabel
+    if (cancel && (wxApp::GetComCtl32Version() >= 600))
+    {
+        ListView_CancelEditLabel(GetHwnd());
+        return true;
+    }
+#endif
     // We shouldn't destroy the control ourselves according to MSDN, which
     // proposes WM_CANCELMODE to do this, but it doesn't seem to work so
     // emulate the corresponding user action instead.
+    // Note: This won't work from a EVT_LIST_BEGIN_LABEL_EDIT() handler
+    // because the control is not yet ready to receive WM_KEYDOWN:
+    // use event.Veto() instead
     ::SendMessage(hwnd, WM_KEYDOWN, cancel ? VK_ESCAPE : VK_RETURN, 0);
 
     return true;
