Index: samples/listctrl/listtest.cpp
===================================================================
--- samples/listctrl/listtest.cpp	(revision 64353)
+++ samples/listctrl/listtest.cpp	(working copy)
@@ -840,6 +840,14 @@
 
 void MyFrame::OnEdit(wxCommandEvent& WXUNUSED(event))
 {
+#ifdef __WXMSW__
+    if (m_listCtrl->GetEditControl())
+    {
+        bool cancel = true;
+        m_listCtrl->EndEditLabel(cancel); // not implemented on all platforms
+        return;
+    }
+#endif
     long itemCur = m_listCtrl->GetNextItem(-1, wxLIST_NEXT_ALL,
                                            wxLIST_STATE_FOCUSED);
 
