Index: E:/Dev/test/wxWindows/src/common/init.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/common/init.cpp,v
retrieving revision 1.43
diff -u -2 -r1.43 init.cpp
--- E:/Dev/test/wxWindows/src/common/init.cpp	2004/02/28 23:57:51	1.43
+++ E:/Dev/test/wxWindows/src/common/init.cpp	2004/04/03 08:55:38
@@ -182,4 +182,5 @@
     }
 
+    gs_initData.argc = argc;
     gs_initData.argv[argc] = NULL;
 }
@@ -196,4 +197,5 @@
         delete [] gs_initData.argv;
         gs_initData.argv = NULL;
+        gs_initData.argc = 0;
     }
 }

 	  	 
