--- configure.in	2010-08-16 21:34:10 +1000
+++ configure.in_orig	2010-07-18 23:39:20 +1000
@@ -2838,15 +2838,6 @@
             dnl the default in wx/msw/rcdefs.h)
             WINDRES_CPU_DEFINE="--define WX_CPU_AMD64"
         ;;
-        i686-*-mingw32* )
-            dnl --- For mingw-w64 lctl3d32's name has changed
-            LIBS="$LIBS -lwinspool -lwinmm -lshell32 -lcomctl32 -lcomdlg32 -lwctl3d32 -ladvapi32 -lwsock32 -lgdi32"
-
-            dnl we need to define this to embed the manifest for correct
-            dnl platform from wx/msw/wx.rc (this is not needed for x86 which is
-            dnl the default in wx/msw/rcdefs.h)
-            WINDRES_CPU_DEFINE="--define WX_CPU_AMD64"
-        ;;
         * )
             LIBS="$LIBS -lwinspool -lwinmm -lshell32 -lcomctl32 -lcomdlg32 -lctl3d32 -ladvapi32 -lwsock32 -lgdi32"
         ;;
@@ -2862,13 +2853,13 @@
     esac
 
     dnl This one is still used by some sample makefiles.
-    RESFLAGS="--include-dir \$(top_srcdir)/include --include-dir \$(top_srcdir)/\$(program_dir) --define __WIN32__ --define __WIN95__ --define __GNUWIN32__ $WINDRES_CPU_DEFINE"
+    RESFLAGS="--include-dir \$(top_srcdir)/include --include-dir \$(top_srcdir)/\$(program_dir) --define __WIN32__ --define __WIN95__ --define __GNUWIN32__"
     RESPROGRAMOBJ="\$(PROGRAM)_resources.o"
 
     dnl This lot we export to wx-config.  It must add the relevant
     dnl include directories at the point when they can be known.
     dnl (but are these (still) required anyway?)
-    WXCONFIG_RESFLAGS="--define __WIN32__ --define __WIN95__ --define __GNUWIN32__ $WINDRES_CPU_DEFINE"
+    WXCONFIG_RESFLAGS="--define __WIN32__ --define __WIN95__ --define __GNUWIN32__"
 
     dnl install Win32-specific files in "make install"
     WIN32INSTALL=win32install
@@ -8058,7 +8049,7 @@
         AC_MSG_ERROR([Required windres program not found])
     fi
 
-    RESCOMP="$WINDRES"
+    RESCOMP="$WINDRES $WINDRES_CPU_DEFINE"
 fi
 
 if test "$wxUSE_MAC" = 1 -o "$wxUSE_OLD_COCOA" = 1; then
