diff -bru wxWidgets/samples/docview/docview.cpp /home/igor/docview/docview.cpp
--- wxWidgets/samples/docview/docview.cpp	2018-02-18 14:30:31.000000000 -0600
+++ /home/igor/docview/docview.cpp	2019-10-27 22:59:39.247025065 -0500
@@ -48,7 +48,9 @@
 
 #include "wx/docview.h"
 #include "wx/docmdi.h"
-
+// Added this for the second toolbar
+#include "chart.xpm"
+// ---------------------------------
 #include "docview.h"
 #include "doc.h"
 #include "view.h"
@@ -206,6 +208,10 @@
                                         GetAppDisplayName(),
                                         wxDefaultPosition,
                                         wxSize(500, 400));
+// Add main toolbar
+        m_tb1 = frame->CreateToolBar( wxNO_BORDER | wxTB_FLAT | wxTB_HORIZONTAL );
+        m_tb1->Realize();
+// -----------------------------------------------
     }
     else
 #endif // wxUSE_MDI_ARCHITECTURE
@@ -256,7 +262,6 @@
         for ( size_t i = 0; i != m_filesFromCmdLine.size(); ++i )
             docManager->CreateDocument(m_filesFromCmdLine[i], wxDOC_SILENT);
     }
-
     return true;
 }
 
@@ -317,7 +322,9 @@
 wxFrame *MyApp::CreateChildFrame(wxView *view, bool isCanvas)
 {
     // create a child frame of appropriate class for the current mode
-    wxFrame *subframe;
+// This becomes a member of the class ------
+//    wxFrame *subframe
+// ---------------------------------------
     wxDocument *doc = view->GetDocument();
 #if wxUSE_MDI_ARCHITECTURE
     if ( GetMode() == Mode_MDI )
@@ -332,6 +339,18 @@
                             wxDefaultPosition,
                             wxSize(300, 300)
                        );
+// Added the secondary toolbar with the child repositioning
+        wxMDIClientWindow *frame = (wxMDIClientWindow *) wxStaticCast(GetTopWindow(), wxDocMDIParentFrame)->GetClientWindow();
+        wxSize cientSize = wxStaticCast(GetTopWindow(), wxDocMDIParentFrame)->GetClientSize();
+        m_tb2 = new wxToolBar( wxStaticCast(GetTopWindow(), wxDocMDIParentFrame), wxID_ANY, wxDefaultPosition, wxDefaultSize, wxTB_HORIZONTAL | wxTB_FLAT | wxNO_BORDER );
+        m_tb2->AddTool( wxID_ANY, "Test", wxBitmap( chrt_xpm ) );
+        m_tb2->Realize();
+        int offset = m_tb1->GetSize().y;
+        m_tb2->SetSize( 0, offset, cientSize.x, wxDefaultCoord );
+        offset += m_tb2->GetSize().y;
+        frame->SetSize( 0, offset, cientSize.x, cientSize.y - offset );
+        subframe->SetSize( 0, 0, cientSize.x, cientSize.y - offset - 2 );
+// --------------------------------------------------------------
     }
     else
 #endif // wxUSE_MDI_ARCHITECTURE
@@ -387,8 +406,17 @@
     switch ( m_mode )
     {
 #if wxUSE_MDI_ARCHITECTURE
+// Hide the second toolbar and reposition the child
         case Mode_MDI:
+        {
+            m_tb2->Show( false );
+            int heightToolbar = m_tb2->GetSize().y;
+            int framePosition = subframe->GetPosition().y;
+            if( framePosition == 0 )
+                subframe->SetPosition( wxPoint( wxDefaultCoord, -heightToolbar ) );
             modeName = "MDI";
+        }
+// ------------------------------------------------------
             break;
 #endif // wxUSE_MDI_ARCHITECTURE
 
diff -bru wxWidgets/samples/docview/docview.h /home/igor/docview/docview.h
--- wxWidgets/samples/docview/docview.h	2018-02-18 14:30:31.000000000 -0600
+++ /home/igor/docview/docview.h	2019-10-27 22:59:53.743025403 -0500
@@ -82,7 +82,10 @@
     // only used if m_mode == Mode_Single
     MyCanvas *m_canvas;
     wxMenu *m_menuEdit;
-
+// Added those members-----------------
+    wxToolBar *m_tb1, *m_tb2;
+    wxFrame *subframe;
+// ------------------------------------
     wxDECLARE_EVENT_TABLE();
     wxDECLARE_NO_COPY_CLASS(MyApp);
 };
