diff --git a/samples/docview/docview.cpp b/samples/docview/docview.cpp
index 74409d313d..51341a0728 100644
--- a/samples/docview/docview.cpp
+++ b/samples/docview/docview.cpp
@@ -46,6 +46,7 @@
     #error You must set wxUSE_DOC_VIEW_ARCHITECTURE to 1 in setup.h!
 #endif
 
+#include "../mdi/chart.xpm"
 #include "wx/docview.h"
 #include "wx/docmdi.h"
 
@@ -230,6 +231,9 @@ bool MyApp::OnInit()
                                             GetAppDisplayName(),
                                             wxDefaultPosition,
                                             wxSize(500, 400));
+            m_tb1 = frame->CreateToolBar( wxNO_BORDER | wxTB_FLAT | wxTB_HORIZONTAL );
+            m_tb1->AddTool( wxID_ANY, "Test", wxBitmap( chart_xpm ) );
+            m_tb1->Realize();
             break;
 #endif // wxUSE_MDI_ARCHITECTURE
 
@@ -358,12 +362,12 @@ void MyApp::CreateMenuBarForFrame(wxFrame *frame, wxMenu *file, wxMenu *edit)
 wxFrame *MyApp::CreateChildFrame(wxView *view, bool isCanvas)
 {
     // create a child frame of appropriate class for the current mode
-    wxFrame *subframe = NULL;
     wxDocument *doc = view->GetDocument();
     switch ( GetMode() )
 #if wxUSE_MDI_ARCHITECTURE
     {
         case Mode_MDI:
+        {
             subframe = new wxDocMDIChildFrame
                            (
                                 doc,
@@ -374,6 +378,17 @@ wxFrame *MyApp::CreateChildFrame(wxView *view, bool isCanvas)
                                 wxDefaultPosition,
                                 wxSize(300, 300)
                            );
+            wxMDIClientWindow *frame = (wxMDIClientWindow *) wxStaticCast(GetTopWindow(), wxDocMDIParentFrame)->GetClientWindow();
+            wxSize cientSize = wxStaticCast(GetTopWindow(), wxDocMDIParentFrame)->GetClientSize();
+            m_tb2 = new wxToolBar( wxStaticCast(GetTopWindow(), wxDocMDIParentFrame), wxID_ANY, wxDefaultPosition, wxDefaultSize, wxTB_HORIZONTAL | wxTB_FLAT | wxNO_BORDER );
+            m_tb2->AddTool( wxID_ANY, "Test", wxBitmap( chart_xpm ) );
+            m_tb2->Realize();
+            int offset = m_tb1->GetSize().y;
+            m_tb2->SetSize( 0, offset, cientSize.x, wxDefaultCoord );
+            offset += m_tb2->GetSize().y;
+            frame->SetSize( 0, offset, cientSize.x, cientSize.y - offset );
+            subframe->SetSize( 0, 0, cientSize.x, cientSize.y - offset - 2 );
+        }
             break;
 #endif // wxUSE_MDI_ARCHITECTURE
 
@@ -447,7 +462,14 @@ void MyApp::OnAbout(wxCommandEvent& WXUNUSED(event))
     {
 #if wxUSE_MDI_ARCHITECTURE
         case Mode_MDI:
+        {
+            m_tb2->Show( false );
+            int heightToolbar = m_tb2->GetSize().y;
+            int framePosition = subframe->GetPosition().y;
+            if( framePosition == 0 )
+                subframe->SetPosition( wxPoint( wxDefaultCoord, -heightToolbar ) );
             modeName = "MDI";
+        }
             break;
 #endif // wxUSE_MDI_ARCHITECTURE
 
diff --git a/samples/docview/docview.h b/samples/docview/docview.h
index 27d5175f1f..2fe6fb510a 100644
--- a/samples/docview/docview.h
+++ b/samples/docview/docview.h
@@ -85,7 +85,8 @@ private:
     // only used if m_mode == Mode_Single
     MyCanvas *m_canvas;
     wxMenu *m_menuEdit;
-
+    wxToolBar *m_tb1, *m_tb2;
+    wxFrame *subframe;
     wxDECLARE_EVENT_TABLE();
     wxDECLARE_NO_COPY_CLASS(MyApp);
 };
