Index: src/common/datetimefmt.cpp
===================================================================
--- src/common/datetimefmt.cpp	(revision 65116)
+++ src/common/datetimefmt.cpp	(working copy)
@@ -1625,9 +1625,10 @@
     while ( p != pEnd )
     {
         // skip white space and date delimiters
-        while ( wxStrchr(".,/-\t\r\n ", *p) )
+        if( wxStrchr(".,/-\t\r\n ", *p) )
         {
             ++p;
+            continue;
         }
 
         // modify copy of the iterator as we're not sure if the next token is
