diff --git a/src/common/debugrpt.cpp b/src/common/debugrpt.cpp
index 4200249..6068f04 100644
--- a/src/common/debugrpt.cpp
+++ b/src/common/debugrpt.cpp
@@ -234,7 +234,7 @@ wxDebugReport::~wxDebugReport()
 
     if ( !m_dir.empty() )
     {
-        if ( wxRmDir(m_dir.fn_str()) != 0 )
+        if ( wxRmDir(m_dir) != 0 )
         {
             wxLogSysError(_("Failed to clean up debug report directory \"%s\""),
                           m_dir.c_str());
diff --git a/src/common/filefn.cpp b/src/common/filefn.cpp
index 188f954..d27a753 100644
--- a/src/common/filefn.cpp
+++ b/src/common/filefn.cpp
@@ -1190,7 +1190,7 @@ bool wxRmdir(const wxString& dir, int WXUNUSED(flags))
 #if defined(__VMS__)
     return false; //to be changed since rmdir exists in VMS7.x
 #else
-    if ( wxRmDir(dir.fn_str()) != 0 )
+    if ( wxRmDir(dir) != 0 )
     {
         wxLogSysError(_("Directory '%s' couldn't be deleted"), dir);
         return false;
