Index: wx/lib/masked/maskededit.py
===================================================================
--- wx/lib/masked/maskededit.py	(revision 72360)
+++ wx/lib/masked/maskededit.py	(working copy)
@@ -5490,8 +5490,7 @@
                 if timeStr:
 ##                    dbg('timeStr: "%s"' % timeStr)
                     try:
-                        checkTime    = dateHandler.ParseTime(timeStr)
-                        valid = checkTime == len(timeStr)
+                        valid    = dateHandler.ParseTime(timeStr)
                     except:
                         valid = False
                     if not valid:
@@ -5516,9 +5515,8 @@
 
         dateHandler = wx.DateTime_Today()
         try:
-            checkTime    = dateHandler.ParseTime(value)
-##            dbg('checkTime:', checkTime, 'len(value)', len(value))
-            valid = checkTime == len(value)
+            valid    = dateHandler.ParseTime(value)
+##            dbg('valid:', valid, 'len(value)', len(value))
         except:
             valid = False
 
