Index: wxWidgets-2.9/src/msw/treectrl.cpp
===================================================================
--- wxWidgets-2.9/src/msw/treectrl.cpp	(revision 61643)
+++ wxWidgets-2.9/src/msw/treectrl.cpp	(working copy)
@@ -2717,9 +2717,12 @@
 
                 if ( !(tvht.flags & TVHT_ONITEM) )
                 {
-                    if ( !HandleMouseEvent(nMsg, x, y, wParam) )
+                    if ( tvht.flags & TVHT_ONITEMBUTTON )
                     {
-                        if ( tvht.flags & TVHT_ONITEMBUTTON )
+                        processed = true;
+                        SetFocus();
+
+                        if ( !HandleMouseEvent(nMsg, x, y, wParam) )
                         {
                             if ( !IsExpanded(htItem) )
                             {
@@ -2730,8 +2733,6 @@
                                 Collapse(htItem);
                             }
                         }
-
-                        processed = true;
                     }
 
                     m_focusLost = false;
