diff --git a/samples/minimal/minimal.cpp b/samples/minimal/minimal.cpp
index 135595c..cf7e0aa 100644
--- a/samples/minimal/minimal.cpp
+++ b/samples/minimal/minimal.cpp
@@ -29,6 +29,8 @@
 #ifndef WX_PRECOMP
     #include "wx/wx.h"
 #endif
+#include "wx/listctrl.h"
+#include "wx/choicdlg.h"
 
 // ----------------------------------------------------------------------------
 // resources
@@ -48,6 +50,8 @@
 class MyApp : public wxApp
 {
 public:
+    void onClick(wxCommandEvent&);
+    wxFrame* m_frame;
     // override base class virtuals
     // ----------------------------
 
@@ -126,6 +130,19 @@ bool MyApp::OnInit()
 
     // create the main application window
     MyFrame *frame = new MyFrame("Minimal wxWidgets App");
+    m_frame = frame;
+    wxPanel* panel = new wxPanel(frame);
+    wxSizer* s1 = new wxBoxSizer(wxVERTICAL);
+    panel->SetSizer(s1);
+    wxButton* button = new wxButton(panel, -1, "Go");
+    button->Bind(wxEVT_BUTTON, &MyApp::onClick, this);
+    s1->Add(button);
+    wxListCtrl* list = new wxListCtrl(panel, -1, wxDefaultPosition, wxDefaultSize, wxLC_REPORT);
+    list->AppendColumn("column");
+    list->InsertItem(0, "Aaaaaa");
+    list->InsertItem(1, "Bbbbbb");
+    list->InsertItem(2, "Cccccc");
+    s1->Add(list, 0, wxEXPAND);
 
     // and show it (the frames, unlike simple controls, are not shown when
     // created initially)
@@ -137,6 +154,12 @@ bool MyApp::OnInit()
     return true;
 }
 
+void MyApp::onClick(wxCommandEvent&)
+{
+    wxString a[] = { wxString("One"), wxString("Two") };
+    wxSingleChoiceDialog d(m_frame, "Pick one", "Dialog", 2, a);
+    d.ShowModal();
+}
 // ----------------------------------------------------------------------------
 // main frame
 // ----------------------------------------------------------------------------
