Index: grid.cpp
===================================================================
--- grid.cpp	(r‚vision 64459)
+++ grid.cpp	(copie de travail)
@@ -8,7 +8,6 @@
 // Copyright:   (c) Michael Bedward (mbedward@ozemail.com.au)
 // Licence:     wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
-
 /*
     TODO:
 
@@ -4876,8 +4875,13 @@
     }
 
     m_selectedBlockCorner = wxGridCellCoords(bottomRow, rightCol);
-    MakeCellVisible(m_selectedBlockCorner);
-
+   if (m_selection) {
+       if (m_selection->GetSelectionMode() == wxGridSelectCells) {
+           MakeCellVisible(m_selectedBlockCorner);
+       }
+   } else {
+       MakeCellVisible(m_selectedBlockCorner);
+   }
     EnsureFirstLessThanSecond(topRow, bottomRow);
     EnsureFirstLessThanSecond(leftCol, rightCol);
 