diff -r -u -p a/samples/propgrid/propgrid.cpp b/samples/propgrid/propgrid.cpp
--- a/samples/propgrid/propgrid.cpp	2018-12-09 15:40:08.000000000 +0100
+++ b/samples/propgrid/propgrid.cpp	2019-04-19 08:03:41.805301300 +0200
@@ -435,6 +435,7 @@ enum
 #endif
     ID_ONEXTENDEDKEYNAV,
     ID_SHOWPOPUP,
+    ID_SHOWBUG,
     ID_POPUPGRID
 };
 
@@ -563,6 +564,7 @@ wxBEGIN_EVENT_TABLE(FormMain, wxFrame)
 
     EVT_CONTEXT_MENU( FormMain::OnContextMenu )
     EVT_BUTTON(ID_SHOWPOPUP, FormMain::OnShowPopup)
+    EVT_BUTTON(ID_SHOWBUG, FormMain::OnShowBug)
 wxEND_EVENT_TABLE()
 
 // -----------------------------------------------------------------------
@@ -1170,6 +1172,14 @@ void FormMain::PopulateWithStandardItems
         pg->SetPropertyCell( p, 3, "Cell 3", bmp );
         pg->SetPropertyCell( p, 4, "Cell 4", wxNullBitmap, *wxWHITE, *wxBLACK );
     }
+
+    m_propTest1 = new wxStringProperty("Test 1");
+    pg->Append( m_propTest1 );                   // The main parameter
+    m_propTest2 = new wxStringProperty("Test 2");
+    m_propTest1->AppendChild( m_propTest2 );     // An associated sub-parameter
+
+    m_TestToogleStatus = false;
+    SetTestPropValues();
 }
 
 // -----------------------------------------------------------------------
@@ -1855,6 +1865,9 @@ void FormMain::FinalizePanel( bool wasCr
     m_topSizer->Add( new wxButton(m_panel, ID_SHOWPOPUP,
                      "Show Popup"),
                      wxSizerFlags(0).Expand());
+    m_topSizer->Add( new wxButton(m_panel, ID_SHOWBUG,
+                     "Toogle test values to see the problem"),
+                     wxSizerFlags(0).Expand());
 
     m_panel->SetSizer( m_topSizer );
     m_topSizer->SetSizeHints( m_panel );
@@ -3335,3 +3348,21 @@ void FormMain::OnShowPopup(wxCommandEven
     popup->Position(pt, wxSize(0, 0));
     popup->Show();
 }
+
+void FormMain::OnShowBug(wxCommandEvent& WXUNUSED(event))
+{
+    m_TestToogleStatus = !m_TestToogleStatus;
+    SetTestPropValues();
+}
+
+void FormMain::SetTestPropValues()
+{
+    wxString strValue = m_TestToogleStatus ? "2"      : "1";
+    wxString strUnit  = m_TestToogleStatus ? "PIXELS" : "Pixel";
+
+    m_propTest1->SetValue( strValue );
+    m_propTest1->SetAttribute( wxPG_ATTR_UNITS, strUnit );
+
+    m_propTest2->SetValue( strValue );
+    m_propTest2->SetAttribute( wxPG_ATTR_UNITS, strUnit );
+}
diff -r -u -p a/samples/propgrid/propgrid.h b/samples/propgrid/propgrid.h
--- a/samples/propgrid/propgrid.h	2018-12-09 15:40:08.000000000 +0100
+++ b/samples/propgrid/propgrid.h	2019-04-19 08:05:01.035003400 +0200
@@ -134,6 +134,10 @@ public:
     wxPropertyGridManager*  m_pPropGridManager;
     wxPropertyGrid*     m_propGrid;
 
+    wxStringProperty* m_propTest1;
+    wxStringProperty* m_propTest2;
+    bool m_TestToogleStatus;
+
     wxTextCtrl*     m_tcPropLabel;
     wxWindow*       m_panel;
     wxBoxSizer*     m_topSizer;
@@ -259,6 +263,9 @@ public:
     void OnIdle( wxIdleEvent& event );
     void OnShowPopup( wxCommandEvent& event );
 
+    void OnShowBug( wxCommandEvent& event );
+    void SetTestPropValues();
+
     void AddTestProperties( wxPropertyGridPage* pg );
 
     bool RunTests( bool fullTest, bool interactive = false );
