--- HTML2_WebView.py	2013-09-10 11:28:46.000000000 -0400
+++ HTML2_WebView.py	2013-09-10 11:28:49.000000000 -0400
@@ -17,8 +17,8 @@
         sizer = wx.BoxSizer(wx.VERTICAL)
         btnSizer = wx.BoxSizer(wx.HORIZONTAL)
         self.wv = webview.WebView.New(self)
-        self.Bind(webview.EVT_WEB_VIEW_NAVIGATING, self.OnWebViewNavigating, self.wv)
-        self.Bind(webview.EVT_WEB_VIEW_LOADED, self.OnWebViewLoaded, self.wv)
+        self.Bind(webview.EVT_WEBVIEW_NAVIGATING, self.OnWebViewNavigating, self.wv)
+        self.Bind(webview.EVT_WEBVIEW_LOADED, self.OnWebViewLoaded, self.wv)
         
                   
         btn = wx.Button(self, -1, "Open", style=wx.BU_EXACTFIT)
