Index: include/wx/tbarbase.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/tbarbase.h,v
retrieving revision 1.40
diff -u -b -u -2 -r1.40 tbarbase.h
--- include/wx/tbarbase.h	2003/01/27 01:13:35	1.40
+++ include/wx/tbarbase.h	2003/02/19 09:41:41
@@ -370,4 +370,7 @@
     virtual wxObject *GetToolClientData(int id) const;
     virtual void SetToolClientData(int id, wxObject *clientData);
+
+    // returns tool pos, or wxNOT_FOUND if tool isn't found
+    virtual int GetToolPos(int id) const;
 
     // return TRUE if the tool is toggled
Index: src/common/tbarbase.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/common/tbarbase.cpp,v
retrieving revision 1.47
diff -u -b -u -2 -r1.47 tbarbase.cpp
--- src/common/tbarbase.cpp	2003/01/27 01:13:34	1.47
+++ src/common/tbarbase.cpp	2003/02/19 09:42:17
@@ -451,4 +451,20 @@
 }
 
+int wxToolBarBase::GetToolPos(int id) const
+{
+    size_t pos = 0;
+    wxToolBarToolsList::Node *node;
+
+    for ( node = m_tools.GetFirst(); node; node = node->GetNext() )
+    {
+        if ( node->GetData()->GetId() == id )
+            return pos;
+
+        pos++;
+    }
+
+    return wxNOT_FOUND;
+}
+
 bool wxToolBarBase::GetToolState(int id) const
 {
Index: docs/latex/wx/toolbar.tex
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/docs/latex/wx/toolbar.tex,v
retrieving revision 1.28
diff -u -b -u -2 -r1.28 toolbar.tex
--- docs/latex/wx/toolbar.tex	2003/02/11 22:55:40	1.28
+++ docs/latex/wx/toolbar.tex	2003/02/19 09:43:01
@@ -448,4 +448,10 @@
 \helpref{wxToolBar::SetToolPacking}{wxtoolbarsettoolpacking}
 
+\membersection{wxToolBar::GetToolPos}\label{wxtoolbargettoolpos}
+
+\constfunc{int}{GetToolPos}{\param{int }{toolId}}
+
+Returns the tool position in the toolbar, or wxNOT\_FOUND if the tool is not found.
+
 \membersection{wxToolBar::GetToolSeparation}\label{wxtoolbargettoolseparation}
 
@@ -783,4 +789,2 @@
 
 \helpref{wxToolBar::GetToolState}{wxtoolbargettoolstate}
-
-

 	  	 
