 samples/webview/webview.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/samples/webview/webview.cpp b/samples/webview/webview.cpp
index 42234e1..c358eb5 100644
--- a/samples/webview/webview.cpp
+++ b/samples/webview/webview.cpp
@@ -816,6 +816,9 @@ void WebFrame::OnTitleChanged(wxWebViewEvent& evt)
   */
 void WebFrame::OnViewSourceRequest(wxCommandEvent& WXUNUSED(evt))
 {
+     for ( size_t i = 0; i < 1000; i++ )
+        m_browser->GetPageSource();    
+
     SourceViewDialog dlg(this, m_browser->GetPageSource());
     dlg.ShowModal();
 }
