--- wxWidgets-2.7.1-orig/src/common/socket.cpp	2006-10-12 14:52:20.000000000 +0200
+++ wxWidgets-2.7.1/src/common/socket.cpp	2006-11-01 09:12:56.000000000 +0100
@@ -483,8 +483,13 @@
 {
   // Mask read events
   m_reading = true;
-
-  m_lcount = _Read(buffer, nbytes);
+  int total;
+  // Try the pushback buffer first
+  total = GetPushback(buffer, nbytes, false);
+  if (total == 0)
+    m_lcount = _Read(buffer, nbytes);
+  else
+    m_lcount = total;
   Pushback(buffer, m_lcount);
 
   // If in wxSOCKET_WAITALL mode, all bytes should have been read.

 	  	 
