Index: src/msw/statusbar.cpp
===================================================================
--- src/msw/statusbar.cpp	(revision 67838)
+++ src/msw/statusbar.cpp	(working copy)
@@ -502,9 +502,6 @@
                        );
     }
 
-    // adjust fields widths to the new size
-    MSWUpdateFieldsWidths();
-
     // we have to trigger wxSizeEvent if there are children window in status
     // bar because GetFieldRect returned incorrect (not updated) values up to
     // here, which almost certainly resulted in incorrectly redrawn statusbar
@@ -595,17 +592,23 @@
     }
 #endif
 
-    bool needsEllipsization = HasFlag(wxSTB_ELLIPSIZE_START) ||
-                              HasFlag(wxSTB_ELLIPSIZE_MIDDLE) ||
-                              HasFlag(wxSTB_ELLIPSIZE_END);
-    if ( nMsg == WM_SIZE && needsEllipsization )
+    if ( nMsg == WM_SIZE )
     {
-        for (int i=0; i<GetFieldsCount(); i++)
-            DoUpdateStatusText(i);
-            // re-set the field text, in case we need to ellipsize
-            // (or de-ellipsize) some parts of it
+        MSWUpdateFieldsWidths();
+
+        bool needsEllipsization = HasFlag(wxSTB_ELLIPSIZE_START) ||
+                                  HasFlag(wxSTB_ELLIPSIZE_MIDDLE) ||
+                                  HasFlag(wxSTB_ELLIPSIZE_END);
+
+        if ( needsEllipsization )
+        {
+            for (int i=0; i<GetFieldsCount(); i++)
+                DoUpdateStatusText(i);
+                // re-set the field text, in case we need to ellipsize
+                // (or de-ellipsize) some parts of it
+        }
     }
-
+    
     return wxStatusBarBase::MSWWindowProc(nMsg, wParam, lParam);
 }
 
