diff --git samples/minimal/minimal.cpp samples/minimal/minimal.cpp
index 80bf766..b72cbd4 100644
--- samples/minimal/minimal.cpp
+++ samples/minimal/minimal.cpp
@@ -172,6 +172,10 @@ MyFrame::MyFrame(const wxString& title)
     CreateStatusBar(2);
     SetStatusText("Welcome to wxWidgets!");
 #endif // wxUSE_STATUSBAR
+
+wxChar c1 = 1, c2 = 2; wxString::Format("%c", c1 + c2);
+
+size_t s = 123; wxString::Format("%zu", s);
 }
 
 
