Index: src/generic/listctrl.cpp
===================================================================
--- src/generic/listctrl.cpp	(revision 51968)
+++ src/generic/listctrl.cpp	(working copy)
@@ -1576,6 +1576,10 @@
         int width = m_owner->GetColumnWidth(col);
         int xOld = x;
         x += width;
+           
+        // Prevents the drawing of images into the
+        // next collumn, in case of small widths.
+        wxDCClipper clipper(*dc, xOld, rect.y, width - 8, rect.height);
 
         if ( item->HasImage() )
         {

 	  	 
