--- text.cpp.orig	Mon Mar 17 05:20:52 2003
+++ text.cpp	Mon Mar 17 17:48:58 2003
@@ -108,6 +108,7 @@
 #endif // wxUSE_CLIPBOARD
 
     void DoRemoveText();
+    void DoReplaceText();
     void DoMoveToEndOfText();
     void DoMoveToEndOfEntry();
 
@@ -162,6 +163,8 @@
         { DoAddText(false); }
     void OnRemoveText( wxCommandEvent& event )
         { m_panel->DoRemoveText(); }
+    void OnReplaceText( wxCommandEvent& event )
+        { m_panel->DoReplaceText(); }
 
     void OnMoveToEndOfText( wxCommandEvent &event )
         { m_panel->DoMoveToEndOfText(); }
@@ -296,6 +299,7 @@
     TEXT_PAGE_DOWN,
     TEXT_PAGE_UP,
     TEXT_REMOVE,
+    TEXT_REPLACE,
     TEXT_SET,
 
     // log menu
@@ -350,7 +354,8 @@
     wxMenu *menuText = new wxMenu;
     menuText->Append(TEXT_ADD_SOME, _T("&Append some text\tCtrl-A"));
     menuText->Append(TEXT_ADD_FREEZE, _T("&Append text with freeze/thaw\tShift-Ctrl-A"));
-    menuText->Append(TEXT_REMOVE, _T("&Remove first 10 characters\tCtrl-X"));
+    menuText->Append(TEXT_REMOVE, _T("&Remove first 10 characters\tCtrl-Y"));
+    menuText->Append(TEXT_REPLACE, _T("&Replace characters 4 to 8 with ABC\tCtrl-R"));
     menuText->Append(TEXT_SET, _T("&Set the first text zone value\tCtrl-E"));
     menuText->AppendSeparator();
     menuText->Append(TEXT_MOVE_ENDTEXT, _T("Move cursor to the end of &text"));
@@ -1033,6 +1038,11 @@
     GetFocusedText(m_multitext)->Remove(0, 10);
 }
 
+void MyPanel::DoReplaceText()
+{
+    GetFocusedText(m_multitext)->Replace(3, 8, _T("ABC"));
+}
+
 //----------------------------------------------------------------------
 // MyFrame
 //----------------------------------------------------------------------
@@ -1061,6 +1071,7 @@
 #endif // wxUSE_CLIPBOARD
 
     EVT_MENU(TEXT_REMOVE,             MyFrame::OnRemoveText)
+    EVT_MENU(TEXT_REPLACE,            MyFrame::OnReplaceText)
     EVT_MENU(TEXT_ADD_SOME,           MyFrame::OnAddText)
     EVT_MENU(TEXT_ADD_FREEZE,         MyFrame::OnAddTextFreeze)
     EVT_MENU(TEXT_MOVE_ENDTEXT,       MyFrame::OnMoveToEndOfText)

 	  	 
