? monolithic.patch
Index: src/common/selectdispatcher.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/common/selectdispatcher.cpp,v
retrieving revision 1.3
diff -u -r1.3 selectdispatcher.cpp
--- src/common/selectdispatcher.cpp	2007/05/14 23:42:57	1.3
+++ src/common/selectdispatcher.cpp	2007/05/16 09:21:33
@@ -20,13 +20,14 @@
 #include "wx/wxprec.h"
 
 #include "wx/private/selectdispatcher.h"
-#include "wx/module.h"
 #include "wx/timer.h"
 #include "wx/unix/private.h"
-#include "wx/log.h"
 
 #ifndef WX_PRECOMP
     #include "wx/hash.h"
+    #include "wx/intl.h"
+    #include "wx/module.h"
+    #include "wx/log.h"
 #endif
 
 #ifdef HAVE_SYS_SELECT_H
Index: src/unix/appunix.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/unix/appunix.cpp,v
retrieving revision 1.1
diff -u -r1.1 appunix.cpp
--- src/unix/appunix.cpp	2007/05/14 23:43:35	1.1
+++ src/unix/appunix.cpp	2007/05/16 09:21:33
@@ -12,6 +12,10 @@
 #include "wx/log.h"
 #include "wx/evtloop.h"
 
+//this code should not be compiled when GUI is defined
+//(monolithic build issue)
+#if !wxUSE_GUI
+
 #include <signal.h>
 #include <unistd.h>
 
@@ -75,3 +79,5 @@
 
     return true;
 }
+
+#endif // !wxUSE_GUI
Index: src/unix/epolldispatcher.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/unix/epolldispatcher.cpp,v
retrieving revision 1.1
diff -u -r1.1 epolldispatcher.cpp
--- src/unix/epolldispatcher.cpp	2007/05/14 23:43:36	1.1
+++ src/unix/epolldispatcher.cpp	2007/05/16 09:21:33
@@ -17,14 +17,19 @@
 // ----------------------------------------------------------------------------
 
 // for compilers that support precompilation, includes "wx.h".
-#include "wx/wxprec.h"
 
 #ifdef HAVE_SYS_EPOLL_H
 
+#include "wx/wxprec.h"
+
+#ifndef WX_PRECOMP
+    #include "wx/intl.h"
+    #include "wx/log.h"
+    #include "wx/module.h"
+#endif
+
 #include "wx/unix/private/epolldispatcher.h"
 #include "wx/unix/private.h"
-#include "wx/log.h"
-#include "wx/module.h"
 
 #include <sys/epoll.h>
 #include <errno.h>
Index: src/unix/evtloopunix.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/unix/evtloopunix.cpp,v
retrieving revision 1.1
diff -u -r1.1 evtloopunix.cpp
--- src/unix/evtloopunix.cpp	2007/05/14 23:43:36	1.1
+++ src/unix/evtloopunix.cpp	2007/05/16 09:21:33
@@ -35,6 +35,10 @@
 
 #define TRACE_EVENTS _T("events")
 
+//this code should not be compiled when GUI is defined
+//(monolithic build issue)
+#if !wxUSE_GUI
+
 // ===========================================================================
 // wxEventLoop::PipeIOHandler implementation
 // ===========================================================================
@@ -190,3 +194,4 @@
     wxTheApp->CheckSignal();
 }
 
+#endif // !wxUSE_GUI

 	  	 
