--- textctrl.orig.cpp	2008-04-21 14:02:46.000000000 +0100
+++ textctrl.cpp	2008-05-29 08:28:26.531250000 +0100
@@ -1310,5 +1310,9 @@
             if ( controlPart != kControlFocusNoPart )
             {
-                // about to gain focus -> set selection from field
+				// to mimic default Mac behaviour: select when focussed (e.g. by tabbing in dialog)
+				// perhaps this should not be used if the selection	was set explicitly
+				focus->SetSelection(-1, -1);
+
+				// about to gain focus -> set selection from field
                 focus->SetData<ControlEditTextSelectionRec>( 0, kControlEditTextSelectionTag, &focus->m_selection );
             }
