Index: src/common/filefn.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWidgets/src/common/filefn.cpp,v
retrieving revision 1.249.2.5
diff -u -r1.249.2.5 filefn.cpp
--- src/common/filefn.cpp	2006/02/27 15:16:37	1.249.2.5
+++ src/common/filefn.cpp	2006/06/06 15:53:20
@@ -873,13 +873,21 @@
     }
     CFStringRef cfString = CFURLCopyFileSystemPath(fullURLRef, kDefaultPathStyle);
     CFRelease( fullURLRef ) ;
-    return wxMacCFStringHolder(cfString).AsString(wxLocale::GetSystemEncoding());
+	wxFontEncoding encoding = wxFONTENCODING_DEFAULT;
+	#if wxUSE_INTL
+       encoding = wxLocale::GetSystemEncoding() ;
+	#endif
+    return wxMacCFStringHolder(cfString).AsString(encoding);
 }
 
 OSStatus wxMacPathToFSRef( const wxString&path , FSRef *fsRef )
 {
     OSStatus err = noErr ;
-    CFURLRef url = CFURLCreateWithFileSystemPath(kCFAllocatorDefault, wxMacCFStringHolder(path ,wxLocale::GetSystemEncoding() ) , kDefaultPathStyle, false);
+	wxFontEncoding encoding = wxFONTENCODING_DEFAULT;
+	#if wxUSE_INTL
+       encoding = wxLocale::GetSystemEncoding() ;
+	#endif
+    CFURLRef url = CFURLCreateWithFileSystemPath(kCFAllocatorDefault, wxMacCFStringHolder(path ,encoding ) , kDefaultPathStyle, false);
     if ( NULL != url )
     {
         if ( CFURLGetFSRef(url, fsRef) == false )



Index: src/mac/corefoundation/cfstring.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWidgets/src/mac/corefoundation/cfstring.cpp,v
retrieving revision 1.4
diff -u -r1.4 cfstring.cpp
--- src/mac/corefoundation/cfstring.cpp	2005/06/08 23:31:13	1.4
+++ src/mac/corefoundation/cfstring.cpp	2006/06/06 15:53:21
@@ -104,7 +104,11 @@
 #if wxUSE_GUI
         encoding = wxFont::GetDefaultEncoding() ;
 #else
-        encoding = wxLocale::GetSystemEncoding() ;
+	#if wxUSE_INTL
+       encoding = wxLocale::GetSystemEncoding() ;
+	#else
+	   encoding = wxFONTENCODING_SYSTEM;
+	#endif
 #endif
     }
     if ( encoding == wxFONTENCODING_SYSTEM )
Index: src/mac/corefoundation/utilsexc_base.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWidgets/src/mac/corefoundation/utilsexc_base.cpp,v
retrieving revision 1.1.2.1
diff -u -r1.1.2.1 utilsexc_base.cpp
--- src/mac/corefoundation/utilsexc_base.cpp	2005/12/19 08:49:24	1.1.2.1
+++ src/mac/corefoundation/utilsexc_base.cpp	2006/06/06 15:53:21
@@ -87,11 +87,15 @@
 
     // Create a CFURL for the application path
     // Created this way because we are opening a bundle which is a directory
+	wxFontEncoding encoding = wxFONTENCODING_DEFAULT;
+	#if wxUSE_INTL
+       encoding = wxLocale::GetSystemEncoding() ;
+	#endif
     CFURLRef cfurlApp = 
         CFURLCreateWithFileSystemPath(
             kCFAllocatorDefault,
             wxMacCFStringHolder(path, 
-                                wxLocale::GetSystemEncoding()),
+                                encoding),
             kDefaultPathStyle, 
             true); //false == not a directory
 
@@ -160,7 +164,7 @@
             cfurlCurrentFile = CFURLCreateWithFileSystemPath(
                                 kCFAllocatorDefault,
                                 wxMacCFStringHolder(*argv, 
-                                wxLocale::GetSystemEncoding()),
+                               encoding),
                                 kDefaultPathStyle, 
                                 true); //true == directory
         }
@@ -171,7 +175,7 @@
             cfurlCurrentFile = CFURLCreateWithFileSystemPath(
                                 kCFAllocatorDefault,
                                 wxMacCFStringHolder(*argv, 
-                                wxLocale::GetSystemEncoding()),
+                                encoding),
                                 kDefaultPathStyle, 
                                 false); //false == regular file
         }    
@@ -182,7 +186,7 @@
             // so try creating it through CFURLCreateWithString
             cfurlCurrentFile = CFURLCreateWithString(
                     kCFAllocatorDefault,
-                wxMacCFStringHolder(*argv, wxLocale::GetSystemEncoding()),
+					wxMacCFStringHolder(*argv,encoding),
                     NULL);
         }
         

 	  	 
