Index: include/wx/cocoa/slider.h
===================================================================
--- include/wx/cocoa/slider.h	(revision 66794)
+++ include/wx/cocoa/slider.h	(working copy)
@@ -94,9 +94,9 @@
     virtual int GetThumbLength() const;
 
     // copied from (wxSliderCocoa.h)
-    virtual void SetTickFreq(int n, int pos);
+    virtual void SetTickFreq(int n);
     virtual int GetTickFreq() const;
-    virtual void ClearTicks() { SetTickFreq(0, 0); }
+    virtual void ClearTicks() { SetTickFreq(0); }
 
     virtual void SetTickPos(int pos);
 
Index: include/wx/msw/slider.h
===================================================================
--- include/wx/msw/slider.h	(revision 66794)
+++ include/wx/msw/slider.h	(working copy)
@@ -59,7 +59,7 @@
     int GetMax() const { return m_rangeMax; }
 
     // Win32-specific slider methods
-    void SetTickFreq(int n, int pos);
+    void SetTickFreq(int n);
     int GetTickFreq() const { return m_tickFreq; }
     void SetPageSize(int pageSize);
     int GetPageSize() const;
Index: include/wx/os2/slider.h
===================================================================
--- include/wx/os2/slider.h	(revision 66794)
+++ include/wx/os2/slider.h	(working copy)
@@ -94,9 +94,7 @@
                                   );
                  void SetThumbLength(int nLen) ;
                  void SetTick(int ntickPos) ;
-                 void SetTickFreq( int n
-                                  ,int nPos
-                                 );
+                 void SetTickFreq(int n) ;
 
     //
     // IMPLEMENTATION
Index: include/wx/osx/slider.h
===================================================================
--- include/wx/osx/slider.h	(revision 66794)
+++ include/wx/osx/slider.h	(working copy)
@@ -59,7 +59,7 @@
     void SetMax(int maxValue) { SetRange(m_rangeMin, maxValue); }
 
     // For trackbars only
-    void SetTickFreq(int n, int pos);
+    void SetTickFreq(int n);
     inline int GetTickFreq() const { return m_tickFreq; }
     void SetPageSize(int pageSize);
     int GetPageSize() const ;
Index: include/wx/palmos/slider.h
===================================================================
--- include/wx/palmos/slider.h	(revision 66794)
+++ include/wx/palmos/slider.h	(working copy)
@@ -52,7 +52,7 @@
     int GetMax() const;
 
     // For trackbars only
-    void SetTickFreq(int n, int pos);
+    void SetTickFreq(int n);
     int GetTickFreq() const;
     void SetPageSize(int pageSize);
     int GetPageSize() const;
Index: include/wx/slider.h
===================================================================
--- include/wx/slider.h	(revision 66794)
+++ include/wx/slider.h	(working copy)
@@ -95,7 +95,7 @@
     // warning: most of subsequent methods are currently only implemented in
     //          wxMSW under Win95 and are silently ignored on other platforms
 
-    virtual void SetTickFreq(int WXUNUSED(n), int WXUNUSED(pos)) { }
+    virtual void SetTickFreq(int WXUNUSED(n)) { }
     virtual int GetTickFreq() const { return 0; }
     virtual void ClearTicks() { }
     virtual void SetTick(int WXUNUSED(tickPos)) { }
Index: include/wx/univ/slider.h
===================================================================
--- include/wx/univ/slider.h	(revision 66794)
+++ include/wx/univ/slider.h	(working copy)
@@ -77,7 +77,7 @@
     virtual void SetThumbLength(int lenPixels);
     virtual int GetThumbLength() const;
 
-    virtual void SetTickFreq(int n, int WXUNUSED(dummy) = 0);
+    virtual void SetTickFreq(int n);
     virtual int GetTickFreq() const { return m_tickFreq; }
 
     // wxUniv-specific methods
Index: interface/wx/slider.h
===================================================================
--- interface/wx/slider.h	(revision 66794)
+++ interface/wx/slider.h	(working copy)
@@ -337,14 +337,12 @@
         @param n
             Frequency. For example, if the frequency is set to two, a tick mark is
             displayed for every other increment in the slider's range.
-        @param pos
-            Position. Must be greater than zero. @todo: what is this for?
 
         @onlyfor{wxmsw}
 
         @see GetTickFreq()
     */
-    virtual void SetTickFreq(int n, int pos);
+    virtual void SetTickFreq(int n);
 
     /**
         Sets the slider position.
Index: samples/controls/controls.cpp
===================================================================
--- samples/controls/controls.cpp	(revision 66794)
+++ samples/controls/controls.cpp	(working copy)
@@ -855,7 +855,7 @@
     m_slider = new wxSlider( panel, ID_SLIDER, 0, 0, 200,
                              wxDefaultPosition, wxSize(155,wxDefaultCoord),
                              wxSL_AUTOTICKS | wxSL_LABELS);
-    m_slider->SetTickFreq(40, 0);
+    m_slider->SetTickFreq(40);
     sz->Add( m_slider, 0, wxALL, 10 );
 
     m_gaugeVert = new wxGauge( panel, wxID_ANY, 100,
Index: samples/widgets/slider.cpp
===================================================================
--- samples/widgets/slider.cpp	(revision 66794)
+++ samples/widgets/slider.cpp	(working copy)
@@ -531,7 +531,7 @@
         return;
     }
 
-    m_slider->SetTickFreq(freq, 0 /* unused */);
+    m_slider->SetTickFreq(freq);
 }
 
 void SliderWidgetsPage::DoSetThumbLen()
Index: src/cocoa/slider.mm
===================================================================
--- src/cocoa/slider.mm	(revision 66794)
+++ src/cocoa/slider.mm	(working copy)
@@ -87,8 +87,7 @@
     // minValue > maxValue not handled, tickMarks set to 0
     if ( style & wxSL_AUTOTICKS )
         tickMarks = ((maxValue - minValue >= 0) ? (maxValue - minValue) : 0);
-    // arg2 needed a value, doesnt do anything
-    SetTickFreq(tickMarks,1);
+    SetTickFreq(tickMarks);
 
     return true;
 }
@@ -206,7 +205,7 @@
     return [GetNSSlider() maxValue];
 }
 
-void wxSlider::SetTickFreq(int n, int pos)
+void wxSlider::SetTickFreq(int n)
 {
     const int numTicks = (n > 0) ? ((GetMax() - GetMin()) / n) + 1 : 0;
     [GetNSSlider() setNumberOfTickMarks:numTicks];
Index: src/msw/slider.cpp
===================================================================
--- src/msw/slider.cpp	(revision 66794)
+++ src/msw/slider.cpp	(working copy)
@@ -666,10 +666,10 @@
     }
 }
 
-void wxSlider::SetTickFreq(int n, int pos)
+void wxSlider::SetTickFreq(int n)
 {
     m_tickFreq = n;
-    ::SendMessage( GetHwnd(), TBM_SETTICFREQ, (WPARAM) n, (LPARAM) pos );
+    ::SendMessage( GetHwnd(), TBM_SETTICFREQ, (WPARAM) n, (LPARAM) 0 );
 }
 
 void wxSlider::SetPageSize(int pageSize)
Index: src/os2/slider.cpp
===================================================================
--- src/os2/slider.cpp	(revision 66794)
+++ src/os2/slider.cpp	(working copy)
@@ -1074,7 +1074,7 @@
 } // end of wxSlider::SetTick
 
 // For trackbars only
-void wxSlider::SetTickFreq( int n, int WXUNUSED(nPos) )
+void wxSlider::SetTickFreq( int n )
 {
     SLDCDATA  vSlData;
     WNDPARAMS vWndParams;
Index: src/osx/slider_osx.cpp
===================================================================
--- src/osx/slider_osx.cpp	(revision 66794)
+++ src/osx/slider_osx.cpp	(working copy)
@@ -206,7 +206,7 @@
 }
 
 // For trackbars only
-void wxSlider::SetTickFreq(int n, int WXUNUSED(pos))
+void wxSlider::SetTickFreq(int n)
 {
     // TODO
     m_tickFreq = n;
Index: src/palmos/slider.cpp
===================================================================
--- src/palmos/slider.cpp	(revision 66794)
+++ src/palmos/slider.cpp	(working copy)
@@ -168,7 +168,7 @@
     // unsupported feature
 }
 
-void wxSlider::SetTickFreq(int WXUNUSED(n), int WXUNUSED(pos))
+void wxSlider::SetTickFreq(int WXUNUSED(n))
 {
     // unsupported feature
 }
Index: src/univ/slider.cpp
===================================================================
--- src/univ/slider.cpp	(revision 66794)
+++ src/univ/slider.cpp	(working copy)
@@ -352,7 +352,7 @@
 // wxSlider ticks
 // ----------------------------------------------------------------------------
 
-void wxSlider::SetTickFreq(int n, int WXUNUSED(dummy))
+void wxSlider::SetTickFreq(int n)
 {
     wxCHECK_RET (n > 0, wxT("invalid slider tick frequency"));
 
Index: src/xrc/xh_slidr.cpp
===================================================================
--- src/xrc/xh_slidr.cpp	(revision 66794)
+++ src/xrc/xh_slidr.cpp	(working copy)
@@ -63,7 +63,7 @@
 
     if( HasParam(wxT("tickfreq")))
     {
-        control->SetTickFreq(GetLong(wxT("tickfreq")), 0);
+        control->SetTickFreq(GetLong(wxT("tickfreq")));
     }
     if( HasParam(wxT("pagesize")))
     {
