Index: include/wx/stc/stc.h
===================================================================
--- include/wx/stc/stc.h	(revision 74045)
+++ include/wx/stc/stc.h	(working copy)
@@ -4577,7 +4577,10 @@
 
     */
 
-    virtual void SetInsertionPoint(long pos) { SetCurrentPos(pos); }
+    virtual void SetInsertionPoint(long pos)
+    {
+        SetCurrentPos( (pos == -1) ? GetLastPosition() : pos );
+    }
     virtual long GetInsertionPoint() const { return GetCurrentPos(); }
     virtual long GetLastPosition() const { return GetTextLength(); }
 
Index: src/stc/stc.h.in
===================================================================
--- src/stc/stc.h.in	(revision 74045)
+++ src/stc/stc.h.in	(working copy)
@@ -319,7 +319,10 @@
 
     */
 
-    virtual void SetInsertionPoint(long pos) { SetCurrentPos(pos); }
+    virtual void SetInsertionPoint(long pos)
+    {
+        SetCurrentPos( (pos == -1) ? GetLastPosition() : pos );
+    }
     virtual long GetInsertionPoint() const { return GetCurrentPos(); }
     virtual long GetLastPosition() const { return GetTextLength(); }
 
