Index: include/wx/rawbmp.h
===================================================================
--- include/wx/rawbmp.h	(revision 53920)
+++ include/wx/rawbmp.h	(working copy)
@@ -680,7 +680,7 @@
 
 #endif //wxUSE_GUI
 
-template <class Image, class PixelFormat = typename wxPixelFormatFor<Image>::Format >
+template <class Image, class PixelFormat = wxPixelFormatFor<Image>::Format >
 class wxPixelData :
     public wxPixelDataOut<Image>::template wxPixelDataIn<PixelFormat>
 {
