Index: wxWidgets-2.8/src/mac/carbon/dialog.cpp
===================================================================
--- wxWidgets-2.8/src/mac/carbon/dialog.cpp	(revision 55921)
+++ wxWidgets-2.8/src/mac/carbon/dialog.cpp	(working copy)
@@ -153,9 +153,12 @@
     if ( GetParent() == NULL )
     {
         windowGroup = GetWindowGroup(windowRef) ;
-        formerParentGroup = GetWindowGroupParent( windowGroup );
-        SetWindowGroupParent( windowGroup, GetWindowGroupOfClass( kMovableModalWindowClass ) );
-        resetGroupParent = true;
+        if ( windowGroup != GetWindowGroupOfClass( kMovableModalWindowClass ) )
+        {
+            formerParentGroup = GetWindowGroupParent( windowGroup );
+            SetWindowGroupParent( windowGroup, GetWindowGroupOfClass( kMovableModalWindowClass ) );
+            resetGroupParent = true;
+        }
     }
     BeginAppModalStateForWindow(windowRef) ;
 
