Index: src/common/serbase.cpp
===================================================================
RCS file: /home/wxcvs/wxWindows/src/common/serbase.cpp,v
retrieving revision 1.9
diff -u -r1.9 serbase.cpp
--- src/common/serbase.cpp	2000/01/18 12:28:20	1.9
+++ src/common/serbase.cpp	2001/06/27 23:30:38
@@ -76,7 +76,7 @@
   wxList *list = (wxList *)Object();
   int number, i;
 
-  list->DeleteContents( data_s.Read8() );
+  list->DeleteContents( data_s.Read8() ? true : false);
   list->SetKeyType( (wxKeyType) data_s.Read8() );
   number = data_s.Read32();
 

 	  	 
