Index: include/wx/intl.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/intl.h,v
retrieving revision 1.49
diff -u -r1.49 intl.h
--- include/wx/intl.h	2003/11/26 19:32:02	1.49
+++ include/wx/intl.h	2003/12/09 10:34:32
@@ -269,6 +269,7 @@
     wxLANGUAGE_SWEDISH,
     wxLANGUAGE_SWEDISH_FINLAND,
     wxLANGUAGE_TAGALOG,
+    wxLANGUAGE_TAGALOG_PHILIPPINES,
     wxLANGUAGE_TAJIK,
     wxLANGUAGE_TAMIL,
     wxLANGUAGE_TATAR,
Index: src/common/intl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/common/intl.cpp,v
retrieving revision 1.110
diff -u -r1.110 intl.cpp
--- src/common/intl.cpp	2003/11/26 20:03:17	1.110
+++ src/common/intl.cpp	2003/12/09 10:34:42
@@ -2783,6 +2783,9 @@
 #ifndef LANG_SWEDISH
 #define LANG_SWEDISH (0)
 #endif
+#ifndef LANG_TAGALOG
+#define LANG_TAGALOG (0)
+#endif
 #ifndef LANG_TAMIL
 #define LANG_TAMIL (0)
 #endif
@@ -3065,6 +3068,9 @@
 #ifndef SUBLANG_SWEDISH_FINLAND
 #define SUBLANG_SWEDISH_FINLAND SUBLANG_DEFAULT
 #endif
+#ifndef SUBLANG_TAGALOG_PHILIPPINES
+#define SUBLANG_TAGALOG_PHILIPPINES SUBLANG_DEFAULT
+#endif
 #ifndef SUBLANG_URDU_INDIA
 #define SUBLANG_URDU_INDIA SUBLANG_DEFAULT
 #endif
@@ -3291,7 +3297,8 @@
    LNG(wxLANGUAGE_SWAHILI,                    "sw_KE", LANG_SWAHILI   , SUBLANG_DEFAULT                   , "Swahili")
    LNG(wxLANGUAGE_SWEDISH,                    "sv_SE", LANG_SWEDISH   , SUBLANG_SWEDISH                   , "Swedish")
    LNG(wxLANGUAGE_SWEDISH_FINLAND,            "sv_FI", LANG_SWEDISH   , SUBLANG_SWEDISH_FINLAND           , "Swedish (Finland)")
-   LNG(wxLANGUAGE_TAGALOG,                    "tl"   , 0              , 0                                 , "Tagalog")
+   LNG(wxLANGUAGE_TAGALOG,                    "tl"   , LANG_TAGALOG   , SUBLANG_DEFAULT                   , "Tagalog")
+   LNG(wxLANGUAGE_TAGALOG_PHILIPPINES,        "tl_PH", LANG_TAGALOG   , SUBLANG_TAGALOG_PHILIPPINES       , "Tagalog (Philippines)")
    LNG(wxLANGUAGE_TAJIK,                      "tg"   , 0              , 0                                 , "Tajik")
    LNG(wxLANGUAGE_TAMIL,                      "ta"   , LANG_TAMIL     , SUBLANG_DEFAULT                   , "Tamil")
    LNG(wxLANGUAGE_TATAR,                      "tt"   , LANG_TATAR     , SUBLANG_DEFAULT                   , "Tatar")

 	  	 
