Index: include/wx/defs.h
===================================================================
--- include/wx/defs.h	(Revision 72498)
+++ include/wx/defs.h	(Arbeitskopie)
@@ -474,7 +474,7 @@
     #define wxSTDCALL
 #endif /*  platform */
 
-/*  LINKAGEMODE mode is empty for everyting except OS/2 */
+/*  LINKAGEMODE mode is empty for everything except OS/2 */
 #ifndef LINKAGEMODE
     #define LINKAGEMODE
 #endif /*  LINKAGEMODE */
@@ -2790,7 +2790,7 @@
 
 /*  --------------------------------------------------------------------------- */
 /*  macros that enable wxWidgets apps to be compiled in absence of the */
-/*  sytem headers, although some platform specific types are used in the */
+/*  system headers, although some platform specific types are used in the */
 /*  platform specific (implementation) parts of the headers */
 /*  --------------------------------------------------------------------------- */
 
Index: include/wx/textbuf.h
===================================================================
--- include/wx/textbuf.h	(Revision 72498)
+++ include/wx/textbuf.h	(Arbeitskopie)
@@ -20,7 +20,7 @@
 // constants
 // ----------------------------------------------------------------------------
 
-// the line termination type (kept wxTextFileType name for compability)
+// the line termination type (kept wxTextFileType name for compatibility)
 enum wxTextFileType
 {
     wxTextFileType_None,  // incomplete (the last line of the file only)
Index: include/wx/encconv.h
===================================================================
--- include/wx/encconv.h	(Revision 72498)
+++ include/wx/encconv.h	(Arbeitskopie)
@@ -76,7 +76,7 @@
             //              try some (lossy) substitutions - e.g. replace
             //              unconvertable latin capitals with acute by ordinary
             //              capitals, replace en-dash or em-dash by '-' etc.
-            //     both modes gurantee that output string will have same length
+            //     both modes guarantee that output string will have same length
             //     as input string
             //
             // Returns false if given conversion is impossible, true otherwise
@@ -109,7 +109,7 @@
             //
             // Equivalence is defined in terms of convertibility:
             // 2 encodings are equivalent if you can convert text between
-            // then without loosing information (it may - and will - happen
+            // then without losing information (it may - and will - happen
             // that you loose special chars like quotation marks or em-dashes
             // but you shouldn't loose any diacritics and language-specific
             // characters when converting between equivalent encodings).
Index: include/wx/event.h
===================================================================
--- include/wx/event.h	(Revision 72498)
+++ include/wx/event.h	(Arbeitskopie)
@@ -3006,7 +3006,7 @@
 protected:
     // Init the hash table with the entries of the static event table.
     void InitHashTable();
-    // Helper funtion of InitHashTable() to insert 1 entry into the hash table.
+    // Helper function of InitHashTable() to insert 1 entry into the hash table.
     void AddEntry(const wxEventTableEntry &entry);
     // Allocate and init with null pointers the base hash table.
     void AllocEventTypeTable(size_t size);
