--- wxrc.cpp	Thu Nov 14 09:51:02 2002
+++ contrib/utils/wxrc/wxrc.cpp	Sat Mar 15 11:55:48 2003
@@ -409,13 +409,13 @@
     
     file.Write(_T("")
-_T("void " + parFuncname + "()\n")
+_T("void ") + parFuncname + wxT("()\n")
 _T("{\n")
 _T("\n")
 _T("    // Check for memory FS. If not present, load the handler:\n")
 _T("    {\n")
-_T("        wxMemoryFSHandler::AddFile(\"XRC_resource/dummy_file\", \"dummy one\");\n")
+_T("        wxMemoryFSHandler::AddFile(wxT(\"XRC_resource/dummy_file\"), wxT(\"dummy one\"));\n")
 _T("        wxFileSystem fsys;\n")
-_T("        wxFSFile *f = fsys.OpenFile(\"memory:XRC_resource/dummy_file\");\n")
-_T("        wxMemoryFSHandler::RemoveFile(\"XRC_resource/dummy_file\");\n")
+_T("        wxFSFile *f = fsys.OpenFile(wxT(\"memory:XRC_resource/dummy_file\"));\n")
+_T("        wxMemoryFSHandler::RemoveFile(wxT(\"XRC_resource/dummy_file\"));\n")
 _T("        if (f) delete f;\n")
 _T("        else wxFileSystem::AddHandler(new wxMemoryFSHandler);\n")
@@ -426,6 +426,6 @@
     {
         wxString s;
-        s.Printf(_T("    wxMemoryFSHandler::AddFile(\"XRC_resource/") + flist[i] +
-                 _T("\", xml_res_file_%i, xml_res_size_%i);\n"), i, i);
+        s.Printf(_T("    wxMemoryFSHandler::AddFile(wxT(\"XRC_resource/") + flist[i] +
+                 _T("\"), xml_res_file_%i, xml_res_size_%i);\n"), i, i);
         file.Write(s);
     }
@@ -433,6 +433,6 @@
     for (i = 0; i < parFiles.Count(); i++)
     {
-        file.Write(_T("    wxXmlResource::Get()->Load(\"memory:XRC_resource/") + 
-                   GetInternalFileName(parFiles[i], flist) + _T("\");\n"));
+        file.Write(_T("    wxXmlResource::Get()->Load(wxT(\"memory:XRC_resource/") +
+                   GetInternalFileName(parFiles[i], flist) + _T("\"));\n"));
     }
     

 	  	 
