+++ E:/wxWidgets-master/include/wx/generic/grid.h	Tue May 26 22:15:19 2015
@@ -942,6 +942,13 @@
         wxGridSelectColumns       = 2,  // allow selecting only entire columns
         wxGridSelectRowsOrColumns = wxGridSelectRows | wxGridSelectColumns
     };
+    
+    enum wxGridMoveModes
+    {
+        wxGridMoveNone            = 0,
+        wxGridMoveDown            = 1,
+        wxGridMoveRight           = 2
+    };
 
     // Different behaviour of the TAB key when the end (or the beginning, for
     // Shift-TAB) of the current row is reached:
@@ -1000,6 +1007,9 @@
 
     void SetSelectionMode(wxGridSelectionModes selmode);
     wxGridSelectionModes GetSelectionMode() const;
+    
+    void SetAfterEditMoveMode(wxGridMoveModes movemode);
+    wxGridMoveModes GetAfterEditMoveMode() const;
 
     // ------ grid dimensions
     //
@@ -1873,6 +1883,7 @@
     wxGridWindow             *m_gridWin;
     wxGridCornerLabelWindow  *m_cornerLabelWin;
     wxGridRowLabelWindow     *m_rowLabelWin;
+    wxGridMoveModes           m_moveMode;
 
     // the real type of the column window depends on m_useNativeHeader value:
     // if it is true, its dynamic type is wxHeaderCtrl, otherwise it is
