--- E:/Downloads/wxWidgets-master/samples/grid/griddemo.h	Tue May 26 22:15:19 2015
+++ E:/Work/.build/wxWidgets/samples/grid/griddemo.h	Mon Jun 01 10:14:21 2015
@@ -42,6 +42,7 @@
     void SetCustomColHeader( wxCommandEvent& );
     void SetDefaultColHeader( wxCommandEvent& );
     void SetTabBehaviour( wxCommandEvent& );
+    void SetCursorMode( wxCommandEvent& );
     void SetTabCustomHandler( wxCommandEvent& );
     void ToggleGridLines( wxCommandEvent& );
     void AutoSizeCols( wxCommandEvent& );
@@ -206,6 +207,10 @@
         ID_RENDER_ZOOM,
         ID_RENDER_MARGIN,
         ID_RENDER_DEFAULT_SIZE,
+        
+        ID_MOVE_NONE, 
+        ID_MOVE_DOWN, 
+        ID_MOVE_RIGHT,
     };
 
 #if wxUSE_LOG
