--- wxWidgets.orig/include/wx/odcombo.h	Mon Jul 17 17:28:00 2006
+++ wxWidgets/include/wx/odcombo.h	Tue Sep  5 22:58:25 2006
@@ -45,5 +45,12 @@
     // when set, we are painting the selected item in control,
     // not in the popup
-    wxODCB_PAINTING_CONTROL         = 0x0001
+    wxODCB_PAINTING_CONTROL         = 0x0001,
+
+
+    // when set, we are painting an item which should have
+    // focus rectangle painted in the background. Text colour
+    // and clipping region are then appropriately set in
+    // the default OnDrawBackground implementation.
+    wxODCB_PAINTING_SELECTED        = 0x0002
 };
 
--- wxWidgets.orig/samples/combo/combo.cpp	Thu Jul 20 18:31:01 2006
+++ wxWidgets/samples/combo/combo.cpp	Tue Sep  5 22:17:58 2006
@@ -231,6 +231,5 @@
         // If item is selected or even, or we are painting the
         // combo control itself, use the default rendering.
-        if ( GetVListBoxComboPopup()->IsCurrent((size_t)item) ||
-             (flags & wxODCB_PAINTING_CONTROL) ||
+        if ( (flags & (wxODCB_PAINTING_CONTROL|wxODCB_PAINTING_SELECTED)) ||
              (item & 1) == 0 )
         {
--- wxWidgets.orig/samples/widgets/odcombobox.cpp	Mon Jul 31 13:48:51 2006
+++ wxWidgets/samples/widgets/odcombobox.cpp	Tue Sep  5 22:16:44 2006
@@ -265,6 +265,5 @@
         // If item is selected or even, or we are painting the
         // combo control itself, use the default rendering.
-        if ( GetVListBoxComboPopup()->IsCurrent((size_t)item) ||
-             (flags & wxODCB_PAINTING_CONTROL) ||
+        if ( (flags & (wxODCB_PAINTING_CONTROL|wxODCB_PAINTING_SELECTED)) ||
              (item & 1) == 0 )
         {
--- wxWidgets.orig/src/generic/odcombo.cpp	Tue Sep  5 21:32:27 2006
+++ wxWidgets/src/generic/odcombo.cpp	Tue Sep  5 23:06:15 2006
@@ -108,5 +108,11 @@
     if ( !(m_combo->GetWindowStyle() & wxODCB_STD_CONTROL_PAINT) )
     {
-        OnDrawBg(dc,rect,m_value,wxODCB_PAINTING_CONTROL);
+        int flags = wxODCB_PAINTING_CONTROL;
+
+        if ( m_combo->ShouldDrawFocus() )
+            flags |= wxODCB_PAINTING_SELECTED;
+
+        OnDrawBg(dc, rect, m_value, flags);
+
         if ( m_value >= 0 )
         {
@@ -166,4 +172,7 @@
                   wxT("you must subclass wxVListBoxComboPopup for drawing and measuring methods") );
 
+    if ( IsCurrent((size_t)item) && !(flags & wxODCB_PAINTING_CONTROL) )
+        flags |= wxODCB_PAINTING_SELECTED;
+
     combo->OnDrawBackground(dc,rect,item,flags);
 }
@@ -1054,21 +1063,23 @@
 }
 
-void wxOwnerDrawnComboBox::OnDrawBackground(wxDC& dc, const wxRect& rect, int item, int flags) const
-{
-    // we need to render selected and current items differently
-    if ( GetVListBoxComboPopup()->IsCurrent((size_t)item) ||
-         (flags & wxODCB_PAINTING_CONTROL) )
+void wxOwnerDrawnComboBox::OnDrawBackground(wxDC& dc,
+                                            const wxRect& rect,
+                                            int WXUNUSED(item),
+                                            int flags) const
+{
+    // We need only to explicitly draw background for items
+    // that should have selected background. Also, call PrepareBackground
+    // always when painting the control so that clipping is done properly.
+
+    if ( (flags & wxODCB_PAINTING_SELECTED) ||
+         ((flags & wxODCB_PAINTING_CONTROL) && HasFlag(wxCB_READONLY)) )
     {
         int bgFlags = wxCONTROL_SELECTED;
 
-        if ( (flags & wxODCB_PAINTING_CONTROL) != wxODCB_PAINTING_CONTROL )
-        {
+        if ( !(flags & wxODCB_PAINTING_CONTROL) )
             bgFlags |= wxCONTROL_ISSUBMENU;
-            PrepareBackground(dc, rect, bgFlags);
-        }
-        else if ( HasFlag(wxCB_READONLY) )
-            PrepareBackground(dc, rect, bgFlags);
+
+        PrepareBackground(dc, rect, bgFlags);
     }
-    //else: do nothing for the normal items
 }
 

 	  	 
