--- ./cvs/wx2/wxwindows/src/msw/mdi.cpp	Sat Apr 28 20:17:10 2001
+++ ./wx2/src/msw/mdi.cpp	Sat May 12 12:31:16 2001
@@ -1147,5 +1147,5 @@
     ccs.idFirstChild = wxFIRST_MDI_CHILD;
 
-    DWORD msStyle = /* MDIS_ALLCHILDSTYLES | */ WS_VISIBLE | WS_CHILD | WS_CLIPCHILDREN;
+    DWORD msStyle = MDIS_ALLCHILDSTYLES | WS_VISIBLE | WS_CHILD | WS_CLIPCHILDREN;
     if ( style & wxHSCROLL )
         msStyle |= WS_HSCROLL;

 	  	 
