Index: src/msw/treectrl.cpp
===================================================================
--- src/msw/treectrl.cpp	(revision 49091)
+++ src/msw/treectrl.cpp	(working copy)
@@ -2151,6 +2151,33 @@
                 }
                 break;
 
+            case WM_RBUTTONDOWN:
+                // only multiple select mode needs to be handled specially;
+                if ( isMultiple )
+                {
+                    if (!IsItemSelected(GetHwnd(), htItem))
+                    {
+                        UnselectAll();
+                        SelectItem(htItem);
+                        ::SetFocus(GetHwnd(), htItem);
+                    }
+
+                    // fire EVT_RIGHT_DOWN
+                    HandleMouseEvent( nMsg, x, y, wParam );
+                    
+                    // send NM_RCLICK
+                    NMHDR nmhdr;
+                    nmhdr.hwndFrom = GetHwnd();
+                    nmhdr.idFrom = ::GetWindowLong( GetHwnd(), GWL_ID );
+                    nmhdr.code = NM_RCLICK;
+                    ::SendMessage( ::GetParent( GetHwnd() ), WM_NOTIFY, nmhdr.idFrom, (LPARAM)&nmhdr );
+
+                    // prevent tree control default processing,
+                    // as we've already done everything                   
+                    processed = true;
+                }
+                break;
+
             case WM_MOUSEMOVE:
 #ifndef __WXWINCE__
                 if ( m_htClickedItem )

 	  	 
