diff -ur ../wxWidgets-20040708/src/os2/spinctrl.cpp ./src/os2/spinctrl.cpp
--- ../wxWidgets-20040708/src/os2/spinctrl.cpp	2004-05-23 13:53:15.000000000 -0700
+++ ./src/os2/spinctrl.cpp	2004-07-15 20:58:50.000000000 -0700
@@ -31,7 +31,7 @@
     #include "wx/wx.h"
 #endif
 
-#if wxUSE_SPINBTN
+#if wxUSE_SPINCTRL
 
 #include "wx/spinctrl.h"
 #include "wx/os2/private.h"
@@ -516,4 +516,4 @@
     ::WinSendMsg(m_hWnd, EM_SETSEL, MPFROM2SHORT((USHORT)lFrom, (USHORT)lTo), (MPARAM)0);
 } // end of wxSpinCtrl::SetSelection
 
-#endif //wxUSE_SPINBTN
+#endif // wxUSE_SPINCTRL

 	  	 
