diff -ur ../wxWidgets-20040708/src/mac/carbon/slider.cpp ./src/mac/carbon/slider.cpp
--- ../wxWidgets-20040708/src/mac/carbon/slider.cpp	2004-06-26 02:31:55.000000000 -0700
+++ ./src/mac/carbon/slider.cpp	2004-07-15 21:48:42.000000000 -0700
@@ -14,6 +14,9 @@
 #endif
 
 #include "wx/slider.h"
+
+#if wxUSE_SLIDER
+
 #include "wx/mac/uma.h"
 
 #if !USE_SHARED_LIBRARY
@@ -424,3 +427,5 @@
 {
     wxControl::DoMoveWindow(x,y,width,height) ;
 }
+
+#endif // wxUSE_SLIDER
diff -ur ../wxWidgets-20040708/src/mac/classic/slider.cpp ./src/mac/classic/slider.cpp
--- ../wxWidgets-20040708/src/mac/classic/slider.cpp	2004-05-23 13:52:41.000000000 -0700
+++ ./src/mac/classic/slider.cpp	2004-07-15 21:49:10.000000000 -0700
@@ -14,6 +14,9 @@
 #endif
 
 #include "wx/slider.h"
+
+#if wxUSE_SLIDER
+
 #include "wx/mac/uma.h"
 
 #if !USE_SHARED_LIBRARY
@@ -431,3 +434,5 @@
 {
     wxControl::DoMoveWindow(x,y,width,height) ;
 }
+
+#endif // wxUSE_SLIDER
diff -ur ../wxWidgets-20040708/src/motif/slider.cpp ./src/motif/slider.cpp
--- ../wxWidgets-20040708/src/motif/slider.cpp	2004-05-23 13:52:49.000000000 -0700
+++ ./src/motif/slider.cpp	2004-07-15 21:49:53.000000000 -0700
@@ -16,6 +16,8 @@
 // For compilers that support precompilation, includes "wx.h".
 #include "wx/wxprec.h"
 
+#if wxUSE_SLIDER
+
 #include "wx/slider.h"
 #include "wx/utils.h"
 
@@ -242,3 +244,4 @@
     slider->GetEventHandler()->ProcessEvent(event2);
 }
 
+#endif // wxUSE_SLIDER
diff -ur ../wxWidgets-20040708/src/os2/slider.cpp ./src/os2/slider.cpp
--- ../wxWidgets-20040708/src/os2/slider.cpp	2004-05-23 13:53:14.000000000 -0700
+++ ./src/os2/slider.cpp	2004-07-15 21:50:37.000000000 -0700
@@ -16,6 +16,8 @@
 #pragma hdrstop
 #endif
 
+#if wxUSE_SLIDER
+
 #ifndef WX_PRECOMP
 #include <stdio.h>
 #include <wx/utils.h>
@@ -1171,3 +1173,4 @@
     return TRUE;
 } // end of wxSlider::Show
 
+#endif // wxUSE_SLIDER

 	  	 
