Index: C:/dev-fix/2.3.x/dev/sdk/wxWidgets-2.8.10/include/wx/mac/carbon/listctrl.h
===================================================================
--- C:/dev-fix/2.3.x/dev/sdk/wxWidgets-2.8.10/include/wx/mac/carbon/listctrl.h	(revision 722)
+++ C:/dev-fix/2.3.x/dev/sdk/wxWidgets-2.8.10/include/wx/mac/carbon/listctrl.h	(revision 723)
@@ -325,6 +325,9 @@
   virtual bool SetForegroundColour(const wxColour& colour);
   virtual bool SetBackgroundColour(const wxColour& colour);
   virtual wxColour GetBackgroundColour() const;
+  virtual void Freeze ();
+  virtual void Thaw ();
+  virtual void Update ();
   
   // functions for editing/timer
   void OnRenameTimer();
Index: C:/dev-fix/2.3.x/dev/sdk/wxWidgets-2.8.10/src/mac/carbon/listctrl_mac.cpp
===================================================================
--- C:/dev-fix/2.3.x/dev/sdk/wxWidgets-2.8.10/src/mac/carbon/listctrl_mac.cpp	(revision 722)
+++ C:/dev-fix/2.3.x/dev/sdk/wxWidgets-2.8.10/src/mac/carbon/listctrl_mac.cpp	(revision 723)
@@ -913,6 +913,27 @@
     return wxNullColour;
 }
 
+void wxListCtrl::Freeze ()
+{
+    if (m_genericImpl)
+        m_genericImpl->Freeze();
+    wxControl::Freeze();
+}
+
+void wxListCtrl::Thaw ()
+{
+    if (m_genericImpl)
+        m_genericImpl->Thaw();
+    wxControl::Thaw();
+}
+
+void wxListCtrl::Update ()
+{
+    if (m_genericImpl)
+        m_genericImpl->Update();
+    wxControl::Update();
+}
+
 // ----------------------------------------------------------------------------
 // accessors
 // ----------------------------------------------------------------------------
