? samples/dialogs/Makefile
Index: samples/dialogs/dialogs.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/samples/dialogs/dialogs.cpp,v
retrieving revision 1.117
diff -u -r1.117 dialogs.cpp
--- samples/dialogs/dialogs.cpp	2004/10/04 20:22:58	1.117
+++ samples/dialogs/dialogs.cpp	2004/10/15 08:51:05
@@ -942,6 +942,8 @@
 {
     wxLogStatus(_T("Sleeping for 3 seconds to allow you to switch to another window"));
 
+    ::wxSafeYield(); /* required at least under GTK2, probably because we're in an event handler and thus wouldn't be able to repaint the status bar text above once we're stuck in this handler due to calling wxSleep below */
+
     wxSleep(3);
 
     RequestUserAttention(wxUSER_ATTENTION_ERROR);
@@ -1023,7 +1025,7 @@
         }
         else if ( i == max / 2 )
         {
-            cont = dialog.Update(i, _T("Only a half left (very long message)!"));
+            cont = dialog.Update(i, _T("Only one half left (very long message)!"));
         }
         else
         {

 	  	 
