From fb2e565bb242b79faebcf630908360c15f2b561e Mon Sep 17 00:00:00 2001
From: abc <a@b.com>
Date: Mon, 7 Oct 2019 09:07:48 +0800
Subject: [PATCH] Fix bug: space leaves when hiding a column header image

---
 src/msw/listctrl.cpp | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/src/msw/listctrl.cpp b/src/msw/listctrl.cpp
index aa5afe781c..9ab59a6b4a 100644
--- a/src/msw/listctrl.cpp
+++ b/src/msw/listctrl.cpp
@@ -3667,6 +3667,22 @@ static void wxConvertToMSWListCol(HWND hwndList,
 
             lvCol.fmt |= LVCFMT_BITMAP_ON_RIGHT | LVCFMT_IMAGE;
         }
+        else
+        { // remove the image
+            LV_COLUMN lvColOld;
+            wxZeroMemory(lvColOld);
+            lvColOld.mask = LVCF_FMT;
+            if (ListView_GetColumn(hwndList, col, &lvColOld))
+            {
+                lvCol.fmt = lvColOld.fmt;
+            }
+
+            lvCol.mask |= LVCF_FMT;
+
+            // both works, consider replace LVCFMT_IMAGE with LVCFMT_COL_HAS_IMAGES in the `if` above?
+            lvCol.fmt &= ~(LVCFMT_COL_HAS_IMAGES | LVCFMT_BITMAP_ON_RIGHT);
+            //lvCol.fmt &= ~(LVCFMT_IMAGE | LVCFMT_BITMAP_ON_RIGHT);
+        }
 
         lvCol.iImage = item.m_image;
     }
-- 
2.17.1.windows.2

