 src/common/textentrycmn.cpp | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/common/textentrycmn.cpp b/src/common/textentrycmn.cpp
index f6c7db5..eec7853 100644
--- a/src/common/textentrycmn.cpp
+++ b/src/common/textentrycmn.cpp
@@ -96,8 +96,11 @@ private:
 
         // Save the old text colour and set a more inconspicuous one for the
         // hint.
-        m_colFg = m_win->GetForegroundColour();
-        m_win->SetForegroundColour(*wxLIGHT_GREY);
+        if (!m_colFg.IsOk())
+        {
+            m_colFg = m_win->GetForegroundColour();
+            m_win->SetForegroundColour(*wxLIGHT_GREY);
+        }
 
         m_entry->DoSetValue(m_hint, wxTextEntryBase::SetValue_NoEvent);
     }
