Index: include/wx/mac/carbon/colour.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/mac/carbon/colour.h,v
retrieving revision 1.8
diff -b -u -2 -r1.8 colour.h
--- include/wx/mac/carbon/colour.h	2006/02/09 00:48:46	1.8
+++ include/wx/mac/carbon/colour.h	2006/04/25 12:21:20
@@ -17,5 +17,5 @@
 
 // Colour
-class WXDLLEXPORT wxColour: public wxObject
+class WXDLLEXPORT wxColour: public wxColourBase
 {
 public:
@@ -25,32 +25,9 @@
     // default
     wxColour() { Init(); }
+    DEFINE_STD_WXCOLOUR_CONSTRUCTORS
 
-    // from separate RGB
-    wxColour( unsigned char red, unsigned char green, unsigned char blue )
-        { Set(red, green, blue); }
-
-    // from packed RGB
-    wxColour( unsigned long colRGB )
-        { Set(colRGB); }
-
-    // implicit conversion from the colour name
-    wxColour( const wxString &colourName )
-        { InitFromName(colourName); }
-    wxColour( const wxChar *colourName )
-        { InitFromName(colourName); }
-
     // dtor
     ~wxColour();
 
-    // Set() functions
-    void Set( unsigned char red, unsigned char green, unsigned char blue );
-    void Set( unsigned long colRGB )
-    {
-        // we don't need to know sizeof(long) here because we assume that the three
-        // least significant bytes contain the R, G and B values
-        Set((unsigned char)colRGB,
-            (unsigned char)(colRGB >> 8),
-            (unsigned char)(colRGB >> 16));
-    }
 
     // accessors
@@ -73,10 +50,10 @@
     const WXCOLORREF& GetPixel() const { return m_pixel; };
 
-    void InitFromName(const wxString& col);
-
 protected :
 
     // Helper function
     void Init();
+
+    void InitWith( unsigned char red, unsigned char green, unsigned char blue );
 
 private:
Index: include/wx/mac/classic/colour.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/mac/classic/colour.h,v
retrieving revision 1.5
diff -b -u -2 -r1.5 colour.h
--- include/wx/mac/classic/colour.h	2005/10/05 16:22:29	1.5
+++ include/wx/mac/classic/colour.h	2006/04/25 12:21:20
@@ -17,5 +17,5 @@
 
 // Colour
-class WXDLLEXPORT wxColour: public wxObject
+class WXDLLEXPORT wxColour: public wxColourBase
 {
 public:
@@ -25,19 +25,6 @@
     // default
     wxColour() { Init(); }
+    DEFINE_STD_WXCOLOUR_CONSTRUCTORS
 
-    // from separate RGB
-    wxColour( unsigned char red, unsigned char green, unsigned char blue )
-        { Set(red, green, blue); }
-
-    // from packed RGB
-    wxColour( unsigned long colRGB )
-        { Set(colRGB); }
-
-    // implicit conversion from the colour name
-    wxColour( const wxString &colourName )
-        { InitFromName(colourName); }
-    wxColour( const wxChar *colourName )
-        { InitFromName(colourName); }
-
     // copy ctors and assignment operators
     wxColour( const wxColour& col );
@@ -48,14 +35,4 @@
     ~wxColour();
 
-    // Set() functions
-    void Set( unsigned char red, unsigned char green, unsigned char blue );
-    void Set( unsigned long colRGB )
-    {
-        // we don't need to know sizeof(long) here because we assume that the three
-        // least significant bytes contain the R, G and B values
-        Set((unsigned char)colRGB,
-            (unsigned char)(colRGB >> 8),
-            (unsigned char)(colRGB >> 16));
-    }
 
     // accessors
@@ -78,10 +55,10 @@
     const WXCOLORREF& GetPixel() const { return m_pixel; };
 
-    void InitFromName(const wxString& col);
+protected:
 
-protected :
-
     // Helper function
     void Init();
+
+    void InitWith( unsigned char red, unsigned char green, unsigned char blue );
 
 private:

 	  	 
