Index: docs/latex/wx/accel.tex
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/docs/latex/wx/accel.tex,v
retrieving revision 1.12
diff -bu2 -r1.12 docs/latex/wx/accel.tex
--- docs/latex/wx/accel.tex
+++ docs/latex/wx/accel.tex
@@ -86,4 +86,10 @@
 <wx/accel.h>
 
+\wxheading{Predefined objects}
+
+Objects:
+
+{\bf wxNullAcceleratorTable}
+
 \wxheading{Example}
 
@@ -151,4 +157,5 @@
 
 Destroys the wxAcceleratorTable object.
+See \helpref{reference-counted object destruction}{refcountdestruct} for more info.
 
 \membersection{wxAcceleratorTable::IsOk}\label{wxacceleratortableisok}
@@ -171,34 +178,2 @@
 
 Returns reference to this object.
-
-\membersection{wxAcceleratorTable::operator $==$}\label{wxacceleratortableequal}
-
-\func{bool}{operator $==$}{\param{const wxAcceleratorTable\& }{accel}}
-
-Equality operator. This operator tests whether the internal data pointers are
-equal (a fast test).
-
-\wxheading{Parameters}
-
-\docparam{accel}{Accelerator table to compare with}
-
-\wxheading{Return value}
-
-Returns true if the accelerator tables were effectively equal, false otherwise.
-
-\membersection{wxAcceleratorTable::operator $!=$}\label{wxacceleratortablenotequal}
-
-\func{bool}{operator $!=$}{\param{const wxAcceleratorTable\& }{accel}}
-
-Inequality operator. This operator tests whether the internal data pointers are
-unequal (a fast test).
-
-\wxheading{Parameters}
-
-\docparam{accel}{Accelerator table to compare with}
-
-\wxheading{Return value}
-
-Returns true if the accelerator tables were unequal, false otherwise.
-
-
Index: include/wx/generic/accel.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/generic/accel.h,v
retrieving revision 1.13
diff -bu2 -r1.13 include/wx/generic/accel.h
--- include/wx/generic/accel.h
+++ include/wx/generic/accel.h
@@ -24,11 +24,4 @@
     virtual ~wxAcceleratorTable();
 
-#if WXWIN_COMPATIBILITY_2_4
-    bool operator==(const wxAcceleratorTable& accel) const
-        { return m_refData == accel.m_refData; }
-    bool operator!=(const wxAcceleratorTable& accel) const
-        { return !(*this == accel); }
-#endif
-
     bool Ok() const { return IsOk(); }
     bool IsOk() const;
Index: include/wx/mac/carbon/accel.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/mac/carbon/accel.h,v
retrieving revision 1.9
diff -bu2 -r1.9 include/wx/mac/carbon/accel.h
--- include/wx/mac/carbon/accel.h
+++ include/wx/mac/carbon/accel.h
@@ -25,9 +25,4 @@
     virtual ~wxAcceleratorTable();
 
-    bool operator == (const wxAcceleratorTable& accel) const
-    { return m_refData == accel.m_refData; }
-    bool operator != (const wxAcceleratorTable& accel) const
-    { return m_refData != accel.m_refData; }
-
     bool Ok() const { return IsOk(); }
     bool IsOk() const;
Index: include/wx/motif/accel.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/motif/accel.h,v
retrieving revision 1.16
diff -bu2 -r1.16 include/wx/motif/accel.h
--- include/wx/motif/accel.h
+++ include/wx/motif/accel.h
@@ -27,7 +27,4 @@
     virtual ~wxAcceleratorTable();
 
-    bool operator == (const wxAcceleratorTable& accel) const { return m_refData == accel.m_refData; }
-    bool operator != (const wxAcceleratorTable& accel) const { return m_refData != accel.m_refData; }
-
     bool Ok() const { return IsOk(); }
     bool IsOk() const;
Index: include/wx/msw/accel.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/msw/accel.h,v
retrieving revision 1.18
diff -bu2 -r1.18 include/wx/msw/accel.h
--- include/wx/msw/accel.h
+++ include/wx/msw/accel.h
@@ -31,8 +31,4 @@
     wxAcceleratorTable(int n, const wxAcceleratorEntry entries[]);
 
-    bool operator==(const wxAcceleratorTable& accel) const;
-    bool operator!=(const wxAcceleratorTable& accel) const
-        { return !(*this == accel); }
-
     bool Ok() const { return IsOk(); }
     bool IsOk() const;
Index: include/wx/os2/accel.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/os2/accel.h,v
retrieving revision 1.14
diff -bu2 -r1.14 include/wx/os2/accel.h
--- include/wx/os2/accel.h
+++ include/wx/os2/accel.h
@@ -41,9 +41,4 @@
     virtual ~wxAcceleratorTable();
 
-    inline bool operator== (const wxAcceleratorTable& rAccel) const
-    { return m_refData == rAccel.m_refData; };
-    inline bool operator!= (const wxAcceleratorTable& rAccel) const
-    { return m_refData != rAccel.m_refData; };
-
     bool Ok() const { return IsOk(); }
     bool IsOk() const;
Index: include/wx/palmos/accel.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/palmos/accel.h,v
retrieving revision 1.6
diff -bu2 -r1.6 include/wx/palmos/accel.h
--- include/wx/palmos/accel.h
+++ include/wx/palmos/accel.h
@@ -31,11 +31,4 @@
     virtual ~wxAcceleratorTable();
 
-#if WXWIN_COMPATIBILITY_2_4
-    bool operator==(const wxAcceleratorTable& accel) const
-        { return m_refData == accel.m_refData; }
-    bool operator!=(const wxAcceleratorTable& accel) const
-        { return !(*this == accel); }
-#endif
-
     bool Ok() const { return IsOk(); }
     bool IsOk() const;
Index: src/generic/accel.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/generic/accel.cpp,v
retrieving revision 1.18
diff -bu2 -r1.18 src/generic/accel.cpp
Index: src/mac/carbon/accel.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/mac/carbon/accel.cpp,v
retrieving revision 1.21
diff -bu2 -r1.21 src/mac/carbon/accel.cpp
Index: src/motif/accel.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/motif/accel.cpp,v
retrieving revision 1.19
diff -bu2 -r1.19 src/motif/accel.cpp
Index: src/msw/accel.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/msw/accel.cpp,v
retrieving revision 1.34
diff -bu2 -r1.34 src/msw/accel.cpp
--- src/msw/accel.cpp
+++ src/msw/accel.cpp
@@ -131,14 +131,4 @@
 }
 
-bool wxAcceleratorTable::operator==(const wxAcceleratorTable& accel) const
-{
-    const wxAcceleratorRefData *
-        accelData = (wxAcceleratorRefData *)accel.m_refData;
-
-    return m_refData ? (accelData &&
-                           M_ACCELDATA->m_hAccel == accelData->m_hAccel)
-                     : !accelData;
-}
-
 bool wxAcceleratorTable::IsOk() const
 {
Index: src/os2/accel.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/os2/accel.cpp,v
retrieving revision 1.24
diff -bu2 -r1.24 src/os2/accel.cpp
--- src/os2/accel.cpp
+++ src/os2/accel.cpp
@@ -203,4 +203,5 @@
 } // end of wxAcceleratorTable::Translate
 
+
 // ---------------------------------------------------------------------------
 // function for translating labels
Index: src/palmos/accel.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/palmos/accel.cpp,v
retrieving revision 1.5
diff -bu2 -r1.5 src/palmos/accel.cpp

 	  	 
