Index: docs/latex/wx/metafile.tex
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/docs/latex/wx/metafile.tex,v
retrieving revision 1.8
diff -bu2 -r1.8 docs/latex/wx/metafile.tex
--- docs/latex/wx/metafile.tex
+++ docs/latex/wx/metafile.tex
@@ -34,4 +34,5 @@
 
 Destructor.
+See \helpref{reference-counted object destruction}{refcountdestruct} for more info.
 
 \membersection{wxMetafile::IsOk}\label{wxmetafileisok}
Index: include/wx/mac/carbon/metafile.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/mac/carbon/metafile.h,v
retrieving revision 1.11
diff -bu2 -r1.11 include/wx/mac/carbon/metafile.h
--- include/wx/mac/carbon/metafile.h
+++ include/wx/mac/carbon/metafile.h
@@ -58,10 +58,4 @@
     WXHMETAFILE GetHMETAFILE() const ;
     void SetHMETAFILE(WXHMETAFILE mf) ;
-
-    // Operators
-    inline bool operator == (const wxMetafile& metafile) const { return m_refData == metafile.m_refData; }
-    inline bool operator != (const wxMetafile& metafile) const { return m_refData != metafile.m_refData; }
-
-protected:
 };
 
Index: include/wx/msw/metafile.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/msw/metafile.h,v
retrieving revision 1.19
diff -bu2 -r1.19 include/wx/msw/metafile.h
--- include/wx/msw/metafile.h
+++ include/wx/msw/metafile.h
@@ -69,10 +69,4 @@
     void SetWindowsMappingMode(int mm);
 
-    // Operators
-    bool operator==(const wxMetafile& metafile) const
-        { return m_refData == metafile.m_refData; }
-    bool operator!=(const wxMetafile& metafile) const
-        { return m_refData != metafile.m_refData; }
-
 private:
     DECLARE_DYNAMIC_CLASS(wxMetafile)
Index: include/wx/os2/metafile.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/os2/metafile.h,v
retrieving revision 1.13
diff -bu2 -r1.13 include/wx/os2/metafile.h
--- include/wx/os2/metafile.h
+++ include/wx/os2/metafile.h
@@ -67,10 +67,4 @@
     inline int GetWindowsMappingMode(void) { return M_METAFILEDATA->m_windowsMappingMode; }
     void SetWindowsMappingMode(int mm);
-
-    // Operators
-    inline bool operator== (const wxMetafile& metafile) const
-        { return m_refData == metafile.m_refData; }
-    inline bool operator!= (const wxMetafile& metafile) const
-        { return m_refData != metafile.m_refData; }
 };
 
Index: include/wx/palmos/metafile.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/palmos/metafile.h,v
retrieving revision 1.7
diff -bu2 -r1.7 include/wx/palmos/metafile.h
--- include/wx/palmos/metafile.h
+++ include/wx/palmos/metafile.h
@@ -69,10 +69,4 @@
     void SetWindowsMappingMode(int mm);
 
-    // Operators
-    bool operator==(const wxMetafile& metafile) const
-        { return m_refData == metafile.m_refData; }
-    bool operator!=(const wxMetafile& metafile) const
-        { return m_refData != metafile.m_refData; }
-
 private:
     DECLARE_DYNAMIC_CLASS(wxMetafile)
Index: src/palmos/metafile.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/palmos/metafile.cpp,v
retrieving revision 1.5
diff -bu2 -r1.5 src/palmos/metafile.cpp

 	  	 
