Index: docs/latex/wx/palette.tex
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/docs/latex/wx/palette.tex,v
retrieving revision 1.16
diff -bu2 -r1.16 docs/latex/wx/palette.tex
--- docs/latex/wx/palette.tex
+++ docs/latex/wx/palette.tex
@@ -87,4 +87,5 @@
 
 Destructor.
+See \helpref{reference-counted object destruction}{refcountdestruct} for more info.
 
 \membersection{wxPalette::Create}\label{wxpalettecreate}
@@ -183,18 +184,2 @@
 Assignment operator, using \helpref{reference counting}{trefcount}.
 
-\membersection{wxPalette::operator $==$}\label{wxpaletteequals}
-
-\func{bool}{operator $==$}{\param{const wxPalette\& }{palette}}
-
-Equality operator. Two palettes are equal if they contain pointers
-to the same underlying palette data. It does not compare each attribute,
-so two independently-created palettes using the same parameters will
-fail the test.
-
-\membersection{wxPalette::operator $!=$}\label{wxpalettenotequals}
-
-\func{bool}{operator $!=$}{\param{const wxPalette\& }{palette}}
-
-Inequality operator. Two palettes are not equal if they contain pointers
-to different underlying palette data. It does not compare each attribute.
-
Index: include/wx/generic/paletteg.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/generic/paletteg.h,v
retrieving revision 1.16
diff -bu2 -r1.16 include/wx/generic/paletteg.h
--- include/wx/generic/paletteg.h
+++ include/wx/generic/paletteg.h
@@ -34,6 +34,4 @@
     wxPalette( int n, const unsigned char *red, const unsigned char *green, const unsigned char *blue );
     virtual ~wxPalette();
-    bool operator == ( const wxPalette& palette ) const;
-    bool operator != ( const wxPalette& palette ) const;
     virtual bool Ok() const { return IsOk(); }
     virtual bool IsOk() const;
Index: include/wx/mac/carbon/palette.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/mac/carbon/palette.h,v
retrieving revision 1.12
diff -bu2 -r1.12 include/wx/mac/carbon/palette.h
--- include/wx/mac/carbon/palette.h
+++ include/wx/mac/carbon/palette.h
@@ -48,7 +48,4 @@
     virtual bool IsOk() const { return (m_refData != NULL) ; }
 
-    inline bool operator == (const wxPalette& palette) const { return m_refData == palette.m_refData; }
-    inline bool operator != (const wxPalette& palette) const { return m_refData != palette.m_refData; }
-
     virtual int GetColoursCount() const;
     
Index: include/wx/mgl/palette.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/mgl/palette.h,v
retrieving revision 1.13
diff -bu2 -r1.13 include/wx/mgl/palette.h
--- include/wx/mgl/palette.h
+++ include/wx/mgl/palette.h
@@ -36,6 +36,4 @@
     wxPalette(int n, const unsigned char *red, const unsigned char *green, const unsigned char *blue);
     virtual ~wxPalette();
-    bool operator == (const wxPalette& palette) const;
-    bool operator != (const wxPalette& palette) const;
     virtual bool Ok() const { return IsOk(); }
     virtual bool IsOk() const;
Index: include/wx/motif/palette.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/motif/palette.h,v
retrieving revision 1.17
diff -bu2 -r1.17 include/wx/motif/palette.h
--- include/wx/motif/palette.h
+++ include/wx/motif/palette.h
@@ -62,7 +62,4 @@
     virtual bool IsOk() const { return (m_refData != NULL) ; }
 
-    bool operator == (const wxPalette& palette) const { return m_refData == palette.m_refData; }
-    bool operator != (const wxPalette& palette) const { return m_refData != palette.m_refData; }
-
     // Motif-specific
     WXColormap GetXColormap(WXDisplay* display = NULL) const;
Index: include/wx/msw/palette.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/msw/palette.h,v
retrieving revision 1.17
diff -bu2 -r1.17 include/wx/msw/palette.h
--- include/wx/msw/palette.h
+++ include/wx/msw/palette.h
@@ -45,7 +45,4 @@
     virtual bool IsOk(void) const { return (m_refData != NULL) ; }
 
-    inline bool operator == (const wxPalette& palette) const { return m_refData == palette.m_refData; }
-    inline bool operator != (const wxPalette& palette) const { return m_refData != palette.m_refData; }
-
     virtual bool FreeResource(bool force = false);
 
Index: include/wx/os2/palette.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/os2/palette.h,v
retrieving revision 1.14
diff -bu2 -r1.14 include/wx/os2/palette.h
--- include/wx/os2/palette.h
+++ include/wx/os2/palette.h
@@ -63,9 +63,4 @@
     virtual bool IsOk(void) const { return (m_refData != NULL) ; }
 
-    inline bool       operator == (const wxPalette& rPalette) const
-        { return m_refData == rPalette.m_refData; }
-    inline bool       operator != (const wxPalette& rPalette) const
-        { return m_refData != rPalette.m_refData; }
-
     virtual bool FreeResource(bool bForce = false);
 
Index: include/wx/palmos/palette.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/palmos/palette.h,v
retrieving revision 1.8
diff -bu2 -r1.8 include/wx/palmos/palette.h
--- include/wx/palmos/palette.h
+++ include/wx/palmos/palette.h
@@ -45,7 +45,4 @@
   virtual bool IsOk(void) const { return (m_refData != NULL) ; }
 
-  inline bool operator == (const wxPalette& palette) const { return m_refData == palette.m_refData; }
-  inline bool operator != (const wxPalette& palette) const { return m_refData != palette.m_refData; }
-
   virtual bool FreeResource(bool force = false);
 
Index: include/wx/x11/palette.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/x11/palette.h,v
retrieving revision 1.12
diff -bu2 -r1.12 include/wx/x11/palette.h
--- include/wx/x11/palette.h
+++ include/wx/x11/palette.h
@@ -62,7 +62,4 @@
     virtual bool IsOk() const { return (m_refData != NULL) ; }
 
-    bool operator == (const wxPalette& palette) const { return m_refData == palette.m_refData; }
-    bool operator != (const wxPalette& palette) const { return m_refData != palette.m_refData; }
-
     // X-specific
     WXColormap GetXColormap(WXDisplay* display = NULL) const;
Index: src/generic/paletteg.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/generic/paletteg.cpp,v
retrieving revision 1.16
diff -bu2 -r1.16 src/generic/paletteg.cpp
--- src/generic/paletteg.cpp
+++ src/generic/paletteg.cpp
@@ -71,14 +71,4 @@
 }
 
-bool wxPalette::operator == (const wxPalette& palette) const
-{
-    return m_refData == palette.m_refData;
-}
-
-bool wxPalette::operator != (const wxPalette& palette) const
-{
-    return m_refData != palette.m_refData;
-}
-
 bool wxPalette::IsOk() const
 {
Index: src/mgl/palette.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/mgl/palette.cpp,v
retrieving revision 1.12
diff -bu2 -r1.12 src/mgl/palette.cpp
--- src/mgl/palette.cpp
+++ src/mgl/palette.cpp
@@ -64,14 +64,4 @@
 }
 
-bool wxPalette::operator == (const wxPalette& palette) const
-{
-    return m_refData == palette.m_refData;
-}
-
-bool wxPalette::operator != (const wxPalette& palette) const
-{
-    return m_refData != palette.m_refData;
-}
-
 bool wxPalette::IsOk(void) const
 {

 	  	 
