Index: docs/latex/wx/brush.tex
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/docs/latex/wx/brush.tex,v
retrieving revision 1.22
diff -bu2 -r1.22 docs/latex/wx/brush.tex
--- docs/latex/wx/brush.tex
+++ docs/latex/wx/brush.tex
@@ -136,10 +136,8 @@
 
 Destructor.
+See \helpref{reference-counted object destruction}{refcountdestruct} for more info.
 
 \wxheading{Remarks}
 
-The destructor may not delete the underlying brush object of the native windowing
-system, since wxBrush uses a reference counting system for efficiency.
-
 Although all remaining brushes are deleted when the application exits,
 the application should try to clean up all brushes itself. This is because
@@ -301,8 +299,6 @@
 \func{bool}{operator $==$}{\param{const wxBrush\& }{brush}}
 
-Equality operator. Two brushes are equal if they contain pointers
-to the same underlying brush data. It does not compare each attribute,
-so two independently-created brushes using the same parameters will
-fail the test.
+Equality operator.
+See \helpref{reference-counted object comparison}{refcountequality} for more info.
 
 
@@ -311,6 +307,7 @@
 \func{bool}{operator $!=$}{\param{const wxBrush\& }{brush}}
 
-Inequality operator. Two brushes are not equal if they contain pointers
-to different underlying brush data. It does not compare each attribute.
+Inequality operator.
+See \helpref{reference-counted object comparison}{refcountequality} for more info.
+
 
 \section{\class{wxBrushList}}\label{wxbrushlist}
Index: include/wx/mac/carbon/brush.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/mac/carbon/brush.h,v
retrieving revision 1.13
diff -bu2 -r1.13 include/wx/mac/carbon/brush.h
--- include/wx/mac/carbon/brush.h
+++ include/wx/mac/carbon/brush.h
@@ -45,8 +45,7 @@
     virtual void MacSetThemeBackground(unsigned long macThemeBackground ,  WXRECTPTR extent) ;
 
-    bool operator == (const wxBrush& brush) const
-    { return m_refData == brush.m_refData; }
+    bool operator == (const wxBrush& brush) const;
     bool operator != (const wxBrush& brush) const
-    { return m_refData != brush.m_refData; }
+        { return !(*this == brush); }
 
     wxMacBrushKind MacGetBrushKind()  const ;
Index: include/wx/os2/brush.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/os2/brush.h,v
retrieving revision 1.17
diff -bu2 -r1.17 include/wx/os2/brush.h
--- include/wx/os2/brush.h
+++ include/wx/os2/brush.h
@@ -48,6 +48,6 @@
     virtual ~wxBrush();
 
-    inline bool operator == (const wxBrush& rBrush) const { return m_refData == rBrush.m_refData; }
-    inline bool operator != (const wxBrush& rBrush) const { return m_refData != rBrush.m_refData; }
+    inline bool operator == (const wxBrush& rBrush) const;
+    inline bool operator != (const wxBrush& rBrush) const { return !(*this == rBrush); }
 
     virtual void SetColour(const wxColour& rColour);
Index: src/gtk/brush.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/gtk/brush.cpp,v
retrieving revision 1.29
diff -bu2 -r1.29 src/gtk/brush.cpp
--- src/gtk/brush.cpp
+++ src/gtk/brush.cpp
@@ -41,5 +41,5 @@
     {
         return (m_style == data.m_style &&
-                m_stipple == data.m_stipple &&
+                m_stipple.IsRefTo(&data.m_stipple) &&
                 m_colour == data.m_colour);
     }
Index: src/gtk1/brush.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/gtk1/brush.cpp,v
retrieving revision 1.29
diff -bu2 -r1.29 src/gtk1/brush.cpp
--- src/gtk1/brush.cpp
+++ src/gtk1/brush.cpp
@@ -42,5 +42,5 @@
     {
         return (m_style == data.m_style &&
-                m_stipple == data.m_stipple &&
+                m_stipple.IsRefTo(&data.m_stipple) &&
                 m_colour == data.m_colour);
     }
Index: src/mac/carbon/brush.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/mac/carbon/brush.cpp,v
retrieving revision 1.25
diff -bu2 -r1.25 src/mac/carbon/brush.cpp
--- src/mac/carbon/brush.cpp
+++ src/mac/carbon/brush.cpp
@@ -31,4 +31,16 @@
     virtual ~wxBrushRefData();
 
+    bool operator == ( const wxBrushRefData& brush ) const
+    {
+        return m_style == brush.m_style &&
+                m_stipple.IsRefTo(&data.m_stipple) &&
+                m_colour == brush.m_colour &&
+                m_macBrushKind == brush.m_macBrushKind &&
+                m_macThemeBrush == brush.m_macThemeBrush &&
+                m_macThemeBackground == brush.m_macThemeBackground &&
+                m_macThemeBackgroundExtent == brush.m_macThemeBackgroundExtent;
+    }
+
+
 protected:
     wxMacBrushKind m_macBrushKind ;
@@ -234,2 +246,11 @@
     return (M_BRUSHDATA ? M_BRUSHDATA->m_macBrushKind : kwxMacBrushColour);
 }
+
+bool wxBrush::operator == ( const wxBrush& brush ) const
+{
+    if (m_refData == brush.m_refData) return true;
+
+    if (!m_refData || !brush.m_refData) return false;
+
+    return ( *(wxBrushRefData*)m_refData == *(wxBrushRefData*)brush.m_refData );
+}
Index: src/mgl/brush.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/mgl/brush.cpp,v
retrieving revision 1.16
diff -bu2 -r1.16 src/mgl/brush.cpp
--- src/mgl/brush.cpp
+++ src/mgl/brush.cpp
@@ -77,4 +77,11 @@
     wxBrushRefData(const wxBrushRefData& data);
 
+    bool operator == (const wxBrushRefData& data) const
+    {
+        return (m_style == data.m_style &&
+                m_stipple.IsRefTo(&data.m_stipple) &&
+                m_colour == data.m_colour);
+    }
+
     int            m_style;
     wxColour       m_colour;
@@ -146,5 +153,9 @@
 bool wxBrush::operator == (const wxBrush& brush) const
 {
-    return m_refData == brush.m_refData;
+    if (m_refData == brush.m_refData) return true;
+
+    if (!m_refData || !brush.m_refData) return false;
+
+    return ( *(wxBrushRefData*)m_refData == *(wxBrushRefData*)brush.m_refData );
 }
 
Index: src/msw/brush.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/msw/brush.cpp,v
retrieving revision 1.26
diff -bu2 -r1.26 src/msw/brush.cpp
--- src/msw/brush.cpp
+++ src/msw/brush.cpp
@@ -125,5 +125,5 @@
     return m_style == data.m_style &&
            m_colour == data.m_colour &&
-           m_stipple == data.m_stipple;
+           m_stipple.IsRefTo(&data.m_stipple);
 }
 
Index: src/x11/brush.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/x11/brush.cpp,v
retrieving revision 1.15
diff -bu2 -r1.15 src/x11/brush.cpp
--- src/x11/brush.cpp
+++ src/x11/brush.cpp
@@ -43,5 +43,5 @@
     {
         return (m_style == data.m_style &&
-                m_stipple == data.m_stipple &&
+                m_stipple.IsRefTo(&data.m_stipple) &&
                 m_colour == data.m_colour);
     }

 	  	 
