diff -u -3 -P -r wxWidgets/contrib\samples\fl\fl_demo2.cpp wxW/contrib\samples\fl\fl_demo2.cpp
--- wxWidgets/contrib\samples\fl\fl_demo2.cpp	Sun Nov 09 22:40:23 2003
+++ wxW/contrib\samples\fl\fl_demo2.cpp	Tue May 11 15:18:01 2004
@@ -489,7 +489,7 @@
 {
     wxTreeCtrl* pTree = new wxTreeCtrl( mpInternalFrm, -1 );
     
-    int rootid = pTree->AppendItem( (long)0, label, 0);
+    const wxTreeItemId rootid = pTree->AppendItem( (long)0, label, 0);
     
     if ( label[0] != 'X' )
     {
diff -u -3 -P -r wxWidgets/contrib\samples\gizmos\led\led.cpp wxW/contrib\samples\gizmos\led\led.cpp
--- wxWidgets/contrib\samples\gizmos\led\led.cpp	Thu Jan 08 09:35:27 2004
+++ wxW/contrib\samples\gizmos\led\led.cpp	Tue May 11 15:07:28 2004
@@ -33,6 +33,7 @@
 #include "../../../include/wx/gizmos/ledctrl.h"
 #include "wx/sizer.h"
 #include "wx/panel.h"
+#include "wx/numdlg.h"
 
 // ----------------------------------------------------------------------------
 // private classes
diff -u -3 -P -r wxWidgets/contrib\src\deprecated\prop.cpp wxW/contrib\src\deprecated\prop.cpp
--- wxWidgets/contrib\src\deprecated\prop.cpp	Thu Sep 11 18:41:14 2003
+++ wxW/contrib\src\deprecated\prop.cpp	Tue May 11 14:54:17 2004
@@ -1080,7 +1080,7 @@
 void wxPropertyValidatorRegistry::ClearRegistry(void)
 {
   BeginFind();
-  wxNode *node;
+  wxHashTable::Node *node;
   while ((node = Next()) != NULL)
   {
     delete (wxPropertyValidator *)node->GetData();
diff -u -3 -P -r wxWidgets/contrib\src\deprecated\resource.cpp wxW/contrib\src\deprecated\resource.cpp
--- wxWidgets/contrib\src\deprecated\resource.cpp	Thu Apr 29 10:27:39 2004
+++ wxW/contrib\src\deprecated\resource.cpp	Tue May 11 15:35:12 2004
@@ -215,7 +215,7 @@
         // See if any resource has this as its child; if so, delete from
         // parent's child list.
         BeginFind();
-        wxNode *node = Next();
+        wxHashTable::Node *node = Next();
         while (node != NULL)
         {
             wxItemResource *parent = (wxItemResource *)node->GetData();
@@ -309,10 +309,10 @@
 void wxResourceTable::ClearTable()
 {
     BeginFind();
-    wxNode *node = Next();
+    wxHashTable::Node *node = Next();
     while (node)
     {
-        wxNode *next = Next();
+        wxHashTable::Node *next = Next();
         wxItemResource *item = (wxItemResource *)node->GetData();
         delete item;
         delete node;
diff -u -3 -P -r wxWidgets/contrib\utils\convertrc\wxr2xml.cpp wxW/contrib\utils\convertrc\wxr2xml.cpp
--- wxWidgets/contrib\utils\convertrc\wxr2xml.cpp	Thu Dec 11 12:27:53 2003
+++ wxW/contrib\utils\convertrc\wxr2xml.cpp	Tue May 11 15:10:04 2004
@@ -68,7 +68,7 @@
 bool wxr2xml::ParseResources()
 {
     m_table.BeginFind();
-    wxNode *node;
+    wxHashTable::Node *node;
 
     node = m_table.Next();
     while (node) 

 	  	 
