--- include/wx/filename.h (revision 70796)
+++ include/wx/filename.h (working copy)
@@ -500,6 +500,9 @@
     // Is this a file or directory (not necessarily an existing one)
     bool IsDir() const { return m_name.empty() && m_ext.empty(); }
 
+    // Is this a symbolic link (returns false on platforms without such links)
+    bool IsLink() const;
+
     // various helpers
 
         // get the canonical path format for this platform
