--- scroll.cpp	2011-02-11 16:41:20.322012448 +0000
+++ scrollbak.cpp	2011-02-11 16:46:15.888012524 +0000
@@ -138,7 +138,6 @@
     MyCanvas( wxWindow *parent, wxWindowID, const wxPoint &pos, const wxSize &size );
     ~MyCanvas(){};
     void OnPaint( wxPaintEvent &event );
-    void Show(wxDC& dc);    
     void OnQueryPosition( wxCommandEvent &event );
     void OnAddButton( wxCommandEvent &event );
     void OnDeleteButton( wxCommandEvent &event );
@@ -349,7 +348,7 @@
     : wxScrolledWindow( parent, id, pos, size, wxSUNKEN_BORDER | wxTAB_TRAVERSAL, _T("test canvas") )
 {
     SetScrollRate( 10, 10 );
-    SetVirtualSize( 500, 5000 );
+    SetVirtualSize( 500, 1000 );
 
     (void) new wxButton( this, ID_ADDBUTTON,  _T("add button"), wxPoint(10,10) );
     (void) new wxButton( this, ID_DELBUTTON,  _T("del button"), wxPoint(10,40) );
@@ -426,21 +425,10 @@
 
 void MyCanvas::OnMouseRightDown( wxMouseEvent &event )
 {
-    wxInitAllImageHandlers();
-   
-    wxSize csize =GetVirtualSize();
-    wxBitmap bitmap = wxBitmap( csize.x,csize.y, -1 );
-    wxMemoryDC  memory( bitmap );
-  
-    Show(memory);
- 
-    bitmap.SaveFile(wxT("testfile.png"), wxBITMAP_TYPE_PNG );
-    memory.SelectObject( wxNullBitmap);
-// *** the old version  
     wxPoint pt( event.GetPosition() );
     int x,y;
     CalcUnscrolledPosition( pt.x, pt.y, &x, &y );
-    wxLogMessage( wxT("Saved to testfile.png.... also you clicked : %d %d, scrolled: %d %d"), pt.x, pt.y, x, y );  
+    wxLogMessage( wxT("Mouse down event at: %d %d, scrolled: %d %d"), pt.x, pt.y, x, y );
 }
 
 void MyCanvas::OnMouseWheel( wxMouseEvent &event )
@@ -456,27 +444,16 @@
     event.Skip();
 }
 
-void MyCanvas::Show( wxDC& dc )
-{  
-    dc.DrawRectangle( 0, 0, 500, 5000 );
+void MyCanvas::OnPaint( wxPaintEvent &WXUNUSED(event) )
+{
+    wxPaintDC dc( this );
+    PrepareDC( dc );
+
     dc.DrawText( _T("Press mouse button to test calculations!"), 160, 50 );
 
     dc.DrawText( _T("Some text"), 140, 140 );
 
     dc.DrawRectangle( 100, 160, 200, 200 );
-
-    wxString text = wxT("Example text");
-    for(int c1=0;c1<500;c1++){
-	dc.DrawRectangle(100,10*c1,30,5);
-	dc.DrawText(text,0,11*c1);
-    }
-}
-
-void MyCanvas::OnPaint( wxPaintEvent &WXUNUSED(event) )
-{
-    wxPaintDC dc( this );
-    PrepareDC( dc );
-    Show(dc);
 }
 
 void MyCanvas::OnQueryPosition( wxCommandEvent &WXUNUSED(event) )
