--- wxWidgets-3.0.0-orig\samples\minimal\minimal.cpp	2013-11-11 14:18:37.000000000 +0100
+++ wxWidgets-3.0.0-work\samples\minimal\minimal.cpp	2014-01-06 20:30:53.772795500 +0100
@@ -167,6 +167,12 @@
     SetMenuBar(menuBar);
 #endif // wxUSE_MENUS
 
+    wxButton *b1 = new wxButton(this, wxID_OK, "OK 1", wxPoint(10, 10));
+    wxButton *b2 = new wxButton(this, wxID_OK, "OK 2", wxPoint(10, 50));
+    wxButton *b3 = new wxButton(this, wxID_OK, "OK 3", wxPoint(10, 90));
+    b1->SetLabelMarkup("OK 1!!!");
+    b2->SetLabelMarkup("OK 2!!!");
+    b3->SetLabelMarkup("OK 3!!!");
 #if wxUSE_STATUSBAR
     // create a status bar just for fun (by default with 1 pane only)
     CreateStatusBar(2);
