Index: radiobut.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/radiobut.cpp,v
retrieving revision 1.33
diff -u -2 -r1.33 radiobut.cpp
--- radiobut.cpp	2003/07/10 19:55:08	1.33
+++ radiobut.cpp	2003/07/25 12:26:40
@@ -207,4 +207,8 @@
         // see above: we want to ignore this event
         m_focusJustSet = FALSE;
+
+        // we cannot completely ignore the event
+        // focused radio buttons must be selected (standard msw behavior)
+        SetValue(TRUE);
     }
     else // a real clicked event

 	  	 
