Index: src/richtext/richtextformatdlg.cpp
===================================================================
--- src/richtext/richtextformatdlg.cpp	(Revision 70985)
+++ src/richtext/richtextformatdlg.cpp	(Arbeitskopie)
@@ -144,11 +144,15 @@
 }
 
 // Apply attributes to the object being edited, if any
-bool wxRichTextFormattingDialog::ApplyStyle(wxRichTextCtrl* ctrl, int WXUNUSED(flags))
+//bool wxRichTextFormattingDialog::ApplyStyle(wxRichTextCtrl* ctrl, int WXUNUSED(flags))
+bool wxRichTextFormattingDialog::ApplyStyle(wxRichTextCtrl* WXUNUSED(ctrl), int flags)
 {
     if (GetObject())
     {
-        ctrl->SetStyle(GetObject(), m_attributes);
+        //ctrl->SetStyle(GetObject(), m_attributes);
+        wxRichTextParagraphLayoutBox* parentContainer = GetObject()->GetParentContainer();
+        if (parentContainer)
+            parentContainer->SetStyle(GetObject(), m_attributes, flags);
         return true;
     }
     else
