Index: src/univ/topluniv.cpp
===================================================================
--- src/univ/topluniv.cpp	(Revision 71344)
+++ src/univ/topluniv.cpp	(Arbeitskopie)
@@ -113,7 +113,7 @@
         styleOrig = style;
         exstyleOrig = GetExtraStyle();
         style &= ~(wxCAPTION | wxMINIMIZE_BOX | wxMAXIMIZE_BOX |
-                   wxSYSTEM_MENU | wxRESIZE_BORDER | wxFRAME_TOOL_WINDOW |
+                   wxSYSTEM_MENU | wxFRAME_TOOL_WINDOW |
                    wxRESIZE_BORDER);
         style |= wxBORDER_NONE;
         SetExtraStyle(exstyleOrig & ~wxWS_EX_CONTEXTHELP);
@@ -203,7 +203,7 @@
     }
     if ( (m_windowStyle & (wxSIMPLE_BORDER | wxNO_BORDER)) == 0 )
         style |= wxTOPLEVEL_BORDER;
-    if ( m_windowStyle & (wxRESIZE_BORDER | wxRESIZE_BORDER) )
+    if ( m_windowStyle & wxRESIZE_BORDER )
         style |= wxTOPLEVEL_RESIZEABLE;
 
     if ( IsMaximized() )
