--- orig.wxrc.cpp	2008-06-24 11:43:29.000000000 +0200
+++ wxrc.cpp	2008-06-24 16:13:27.000000000 +0200
@@ -683,7 +683,7 @@
 void XmlResApp::GenCPPHeader()
 {
     wxString fileSpec = ((parOutput.BeforeLast('.')).AfterLast('/')).AfterLast('\\');
-    wxString heaFileName = fileSpec + _T(".h");
+    wxString heaFileName = wxPathOnly(parOutput) + wxFILE_SEP_PATH + fileSpec + _T(".h");
 
     wxFFile file(heaFileName, wxT("wt"));
     file.Write(
