Index: samples/minimal/minimal.cpp
===================================================================
--- samples/minimal/minimal.cpp	(revision 73815)
+++ samples/minimal/minimal.cpp	(working copy)
@@ -67,8 +67,11 @@
     // event handlers (these functions should _not_ be virtual)
     void OnQuit(wxCommandEvent& event);
     void OnAbout(wxCommandEvent& event);
+    void OnButton(wxCommandEvent& event);
 
 private:
+	wxPanel *m_panel;
+
     // any class wishing to process wxWidgets events must use this macro
     DECLARE_EVENT_TABLE()
 };
@@ -99,6 +102,7 @@
 BEGIN_EVENT_TABLE(MyFrame, wxFrame)
     EVT_MENU(Minimal_Quit,  MyFrame::OnQuit)
     EVT_MENU(Minimal_About, MyFrame::OnAbout)
+    EVT_BUTTON(wxID_ANY, MyFrame::OnButton)
 END_EVENT_TABLE()
 
 // Create a new application object: this macro will allow wxWidgets to create
@@ -147,7 +151,7 @@
 {
     // set the frame icon
     SetIcon(wxICON(sample));
-
+  
 #if wxUSE_MENUS
     // create a menu bar
     wxMenu *fileMenu = new wxMenu;
@@ -172,10 +176,32 @@
     CreateStatusBar(2);
     SetStatusText("Welcome to wxWidgets!");
 #endif // wxUSE_STATUSBAR
+
+	m_panel = new wxPanel(this);
+	m_panel->SetBackgroundColour(*wxGREEN);
+
+	wxStaticBox *sb = new wxStaticBox(m_panel, wxID_ANY, wxT("TEST"), wxPoint(0,0), wxSize(240,300));
+	sb->SetBackgroundColour(*wxRED);
+	//sb->SetBackgroundStyle(wxBG_STYLE_CUSTOM);
+
+    wxCoord y=20;
+	wxButton *button; 
+	button = new wxButton(m_panel, wxID_ANY, wxT("BUTTON"), wxPoint(20,y)); y+=30;
+	button = new wxButton(m_panel, wxID_ANY, wxT("BUTTON"), wxPoint(20,y)); y+=30;
+	button = new wxButton(m_panel, wxID_ANY, wxT("BUTTON"), wxPoint(20,y)); y+=30;
+	button = new wxButton(m_panel, wxID_ANY, wxT("BUTTON"), wxPoint(20,y)); y+=30;
+	button = new wxButton(m_panel, wxID_ANY, wxT("BUTTON"), wxPoint(20,y)); y+=30;
 }
 
 
 // event handlers
+void MyFrame::OnButton(wxCommandEvent& WXUNUSED(event))
+{
+	for(int i=0; i<50; i++) {
+		m_panel->Refresh();
+		m_panel->Update();
+	}
+}
 
 void MyFrame::OnQuit(wxCommandEvent& WXUNUSED(event))
 {
