Index: src/regex/regerror.c
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/regex/regerror.c,v
retrieving revision 1.10
diff -u -r1.10 regerror.c
--- src/regex/regerror.c	2004/02/19 16:47:02	1.10
+++ src/regex/regerror.c	2004/06/29 22:25:31
@@ -49,12 +49,17 @@
  - regerror - the interface to error numbers
  */
 /* ARGSUSED */
+#if 0   
 size_t				/* actual space needed (including NUL) */
 regerror(errcode, preg, errbuf, errbuf_size)
 int errcode;			/* error code, or REG_ATOI or REG_ITOA */
 CONST regex_t *preg;		/* associated regex_t (unused at present) */
 char *errbuf;			/* result buffer (unless errbuf_size==0) */
 size_t errbuf_size;		/* available space in errbuf, can be 0 */
+#else
+/* New-style declaration avoids glitch with VS 2005 and "typedef int errcode" */
+size_t regerror(int errcode, CONST regex_t *preg, char *errbuf, size_t errbuf_size)
+#endif
 {
 	struct rerr *r;
 	char *msg;

 	  	 
