diff -ru -U 2 -xCVS -x'.#*' -x'*.orig' -x'*.rej' -x'*~' wxWindows.orig/configure.in wxWindows/configure.in
--- wxWindows.orig/configure.in	2003-12-24 07:42:46.000000000 -0500
+++ wxWindows/configure.in	2004-01-18 19:23:37.000000000 -0500
@@ -1803,7 +1803,7 @@
                               \
     /usr/openwin/share/include"
 
-SEARCH_LIB="`echo "$SEARCH_INCLUDE" | sed s/include/lib/g` "
+SEARCH_LIB="`echo "$SEARCH_INCLUDE" | sed s/include/lib/g` /usr/lib64 /usr/X11R6/lib64"
 
 dnl ------------------------------------------------------------------------
 dnl Check for libraries

 	  	 
