Index: include/wx/defs.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/defs.h,v
retrieving revision 1.327
diff -u -u -2 -r1.327 defs.h
--- include/wx/defs.h	2003/01/04 13:29:36	1.327
+++ include/wx/defs.h	2003/01/07 17:46:31
@@ -255,5 +255,5 @@
 
 // stdcall is used for all functions called by Windows under Windows
-#if defined(__WINDOWS__) && !defined(__WXWINE__)
+#if defined(__WINDOWS__)
     #if defined(__GNUWIN32__)
         #define wxSTDCALL __attribute__((stdcall))
@@ -1932,5 +1932,5 @@
 typedef long            WXLPARAM;
 
-#if !defined(__WIN32__) || defined(__GNUWIN32__) || defined(__WXWINE__) || defined(__WXMICROWIN__)
+#if !defined(__WIN32__) || defined(__GNUWIN32__) || defined(__WXMICROWIN__)
 typedef int             (*WXFARPROC)();
 #else
Index: include/wx/platform.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/platform.h,v
retrieving revision 1.10
diff -u -u -2 -r1.10 platform.h
--- include/wx/platform.h	2003/01/07 13:48:39	1.10
+++ include/wx/platform.h	2003/01/07 17:46:31
@@ -40,13 +40,7 @@
 #endif
 
-#ifdef __WXWINE__
-    #ifndef __WIN32__
-        #define __WIN32__
-    #endif
+#ifdef __WINE__
     #ifndef __WIN95__
         #define __WIN95__
-    #endif
-    #ifndef STRICT
-        #define STRICT
     #endif
 #endif /* WINE */
Index: include/wx/msw/missing.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/msw/missing.h,v
retrieving revision 1.6
diff -u -u -2 -r1.6 missing.h
--- include/wx/msw/missing.h	2002/12/04 13:59:06	1.6
+++ include/wx/msw/missing.h	2003/01/07 17:46:32
@@ -244,18 +244,4 @@
 #endif
 
-#ifdef __WXWINE__
-#define LV_ITEMA LVITEMA
-#define LV_ITEMW LVITEMW
-#define LV_DISPINFOA NMLVDISPINFOA
-#define LV_DISPINFOW NMLVDISPINFOW
-
-#if wxUSE_UNICODE
-#define LV_FINDINFO LVFINDINFOW
-#else
-#define LV_FINDINFO LVFINDINFOA
-#endif
-
-#endif
-
 #endif
     // _WX_MISSING_H_
Index: include/wx/msw/winundef.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/msw/winundef.h,v
retrieving revision 1.25
diff -u -u -2 -r1.25 winundef.h
--- include/wx/msw/winundef.h	2003/01/06 22:46:53	1.25
+++ include/wx/msw/winundef.h	2003/01/07 17:46:33
@@ -342,5 +342,5 @@
 // For WINE
 
-#if defined(GetWindowStyle) || defined(__WXWINE__)
+#if defined(GetWindowStyle)
   #undef GetWindowStyle
 #endif
Index: src/common/cmndata.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/common/cmndata.cpp,v
retrieving revision 1.67
diff -u -u -2 -r1.67 cmndata.cpp
--- src/common/cmndata.cpp	2002/12/15 22:30:46	1.67
+++ src/common/cmndata.cpp	2003/01/07 17:46:40
@@ -57,9 +57,4 @@
     #endif // Win16
 
-    #ifdef __WXWINE__
-        #include <cderr.h>
-        #include <commdlg.h>
-    #endif
-
     #if defined(__WATCOMC__) || defined(__SC__) || defined(__SALFORDC__)
         #include <windowsx.h>
@@ -486,8 +481,5 @@
         //// Orientation
 
-#ifndef __WXWINE__
         devMode->dmOrientation = m_printOrientation;
-#endif
-        devMode->dmFields = DM_ORIENTATION;
 
         //// Collation
@@ -521,5 +513,4 @@
         devMode->dmFields |= DM_COLOR;
 
-#ifndef __WXWINE__
         //// Paper size
 
@@ -545,5 +536,4 @@
             }
         }
-#endif
 
         //// Duplex
@@ -614,10 +604,8 @@
         LPDEVMODE devMode = (LPDEVMODE)GlobalLock(hDevMode);
 
-#ifndef __WXWINE__
         //// Orientation
 
         if (devMode->dmFields & DM_ORIENTATION)
             m_printOrientation = devMode->dmOrientation;
-#endif
 
         //// Collation
@@ -659,5 +647,4 @@
             m_colour = TRUE;
 
-#ifndef __WXWINE__
         //// Paper size
 
@@ -713,5 +700,4 @@
             m_paperId = wxPAPER_NONE;
         }
-#endif
 
         //// Duplex
Index: src/common/file.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/common/file.cpp,v
retrieving revision 1.66
diff -u -u -2 -r1.66 file.cpp
--- src/common/file.cpp	2002/09/12 19:32:29	1.66
+++ src/common/file.cpp	2003/01/07 17:46:42
@@ -29,5 +29,5 @@
 
 // standard
-#if defined(__WXMSW__) && !defined(__GNUWIN32__) && !defined(__WXWINE__) && !defined(__WXMICROWIN__)
+#if defined(__WXMSW__) && !defined(__GNUWIN32__) && !defined(__WXMICROWIN__)
   #include  <io.h>
 
Index: src/common/filefn.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/common/filefn.cpp,v
retrieving revision 1.163
diff -u -u -2 -r1.163 filefn.cpp
--- src/common/filefn.cpp	2003/01/07 13:48:51	1.163
+++ src/common/filefn.cpp	2003/01/07 17:46:44
@@ -76,5 +76,5 @@
     #include "wx/os2/private.h"
 #endif
-#if defined(__WINDOWS__) && !defined(__WXMICROWIN__) && !defined(__WXWINE__)
+#if defined(__WINDOWS__) && !defined(__WXMICROWIN__)
 #if !defined( __GNUWIN32__ ) && !defined( __MWERKS__ ) && !defined(__SALFORDC__)
     #include <direct.h>
@@ -1263,5 +1263,5 @@
     // assume mkdir() has 2 args on non Windows-OS/2 platforms and on Windows too
     // for the GNU compiler
-#if (!(defined(__WXMSW__) || defined(__WXPM__) || defined(__DOS__))) || (defined(__GNUWIN32__) && !defined(__MINGW32__)) || defined(__WXWINE__) || defined(__WINE__) || defined(__WXMICROWIN__)
+#if (!(defined(__WXMSW__) || defined(__WXPM__) || defined(__DOS__))) || (defined(__GNUWIN32__) && !defined(__MINGW32__)) || defined(__WINE__) || defined(__WXMICROWIN__)
     if ( mkdir(wxFNCONV(dirname), perm) != 0 )
 #elif defined(__WXPM__)
Index: src/common/mimecmn.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/common/mimecmn.cpp,v
retrieving revision 1.24
diff -u -u -2 -r1.24 mimecmn.cpp
--- src/common/mimecmn.cpp	2003/01/07 13:48:51	1.24
+++ src/common/mimecmn.cpp	2003/01/07 17:46:47
@@ -487,5 +487,5 @@
 bool wxMimeTypesManager::Unassociate(wxFileType *ft)
 {
-#if defined(__UNIX__) && !defined(__WXPM__) && !defined(__CYGWIN__) && !defined(__WXWINE__) && !defined(__WINE__)
+#if defined(__UNIX__) && !defined(__WXPM__) && !defined(__CYGWIN__) && !defined(__WINE__)
     return m_impl->Unassociate(ft);
 #else
@@ -597,5 +597,5 @@
                                     const wxString& sExtraDir)
 {
-#if defined(__UNIX__) && !defined(__WXPM__) && !defined(__CYGWIN__) && !defined(__WXWINE__) && !defined(__WINE__)
+#if defined(__UNIX__) && !defined(__WXPM__) && !defined(__CYGWIN__) && !defined(__WINE__)
     EnsureImpl();
 
@@ -610,5 +610,5 @@
 void wxMimeTypesManager::ClearData()
 {
-#if defined(__UNIX__) && !defined(__WXPM__) && !defined(__CYGWIN__) && !defined(__WXWINE__) && !defined(__WINE__)
+#if defined(__UNIX__) && !defined(__WXPM__) && !defined(__CYGWIN__) && !defined(__WINE__)
     EnsureImpl();
 
Index: src/common/strconv.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/common/strconv.cpp,v
retrieving revision 1.70
diff -u -u -2 -r1.70 strconv.cpp
--- src/common/strconv.cpp	2002/12/27 14:58:12	1.70
+++ src/common/strconv.cpp	2003/01/07 17:46:47
@@ -785,10 +785,4 @@
 #if defined(__WIN32__) && !defined(__WXMICROWIN__) && !defined(__WXUNIVERSAL__)
 
-#ifdef __WXWINE__
-    #define WINE_CAST (WCHAR *)
-#else
-    #define WINE_CAST
-#endif
-
 extern long wxCharsetToCodepage(const wxChar *charset); // from utils.cpp
 
@@ -810,5 +804,5 @@
                                 psz,            // input string
                                 -1,             // its length (NUL-terminated)
-                                WINE_CAST buf,  // output string
+                                buf,            // output string
                                 buf ? n : 0     // size of output buffer
                              );
@@ -825,5 +819,5 @@
                                 m_CodePage,     // code page
                                 0,              // flags (none)
-                                WINE_CAST psz,  // input string
+                                psz,            // input string
                                 -1,             // it is (wide) NUL-terminated
                                 buf,            // output buffer
Index: src/generic/dirctrlg.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/generic/dirctrlg.cpp,v
retrieving revision 1.51
diff -u -u -2 -r1.51 dirctrlg.cpp
--- src/generic/dirctrlg.cpp	2003/01/01 17:53:00	1.51
+++ src/generic/dirctrlg.cpp	2003/01/07 17:46:51
@@ -58,6 +58,5 @@
 //         older releases don't, but it should be verified and the checks modified
 //         accordingly.
-#if !defined(__WXWINE__) && (!defined(__GNUWIN32__) || \
-    (defined(__MINGW32_MAJOR_VERSION) && __MINGW32_MAJOR_VERSION >= 1))
+#if !defined(__GNUWIN32__) || (defined(__MINGW32_MAJOR_VERSION) && __MINGW32_MAJOR_VERSION >= 1)
   #include <direct.h>
   #include <stdlib.h>
@@ -382,6 +381,5 @@
     {
         wxString dirNameLower(dirName.Lower());
-#if defined(__WXWINE__) || (defined(__GNUWIN32__) && \
-    !(defined(__MINGW32_MAJOR_VERSION) && __MINGW32_MAJOR_VERSION >= 1))
+#if defined(__GNUWIN32__) && !(defined(__MINGW32_MAJOR_VERSION) && __MINGW32_MAJOR_VERSION >= 1)
         success = wxPathExists(dirNameLower);
 #else
Index: src/msw/accel.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/accel.cpp,v
retrieving revision 1.19
diff -u -u -2 -r1.19 accel.cpp
--- src/msw/accel.cpp	2003/01/02 23:38:07	1.19
+++ src/msw/accel.cpp	2003/01/07 17:46:53
@@ -97,5 +97,5 @@
 
 // Create from an array
-#if !defined(__WIN16__) && !defined(__TWIN32__) && !defined(__WXWINE__)
+#if !defined(__WIN16__) && !defined(__TWIN32__)
 wxAcceleratorTable::wxAcceleratorTable(int n, const wxAcceleratorEntry entries[])
 {
Index: src/msw/colordlg.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/colordlg.cpp,v
retrieving revision 1.12
diff -u -u -2 -r1.12 colordlg.cpp
--- src/msw/colordlg.cpp	2002/05/09 22:31:44	1.12
+++ src/msw/colordlg.cpp	2003/01/07 17:46:53
@@ -45,5 +45,5 @@
 #include <windows.h>
 
-#if !defined(__WIN32__) || defined(__SALFORDC__) || defined(__WXWINE__)
+#if !defined(__WIN32__) || defined(__SALFORDC__)
     #include <commdlg.h>
 #endif
Index: src/msw/curico.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/curico.cpp,v
retrieving revision 1.18
diff -u -u -2 -r1.18 curico.cpp
--- src/msw/curico.cpp	2002/08/30 20:34:25	1.18
+++ src/msw/curico.cpp	2003/01/07 17:46:54
@@ -17,12 +17,5 @@
 #endif
 
-#ifndef __WXWINE__
 #include <io.h>
-#endif
-
-#ifdef __WXWINE__
-#define NPSTR LPVOID
-#endif
-
 #include <windows.h>
 
Index: src/msw/dcprint.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/dcprint.cpp,v
retrieving revision 1.33
diff -u -u -2 -r1.33 dcprint.cpp
--- src/msw/dcprint.cpp	2003/01/02 23:28:56	1.33
+++ src/msw/dcprint.cpp	2003/01/07 17:46:55
@@ -42,5 +42,5 @@
 #include "math.h"
 
-#if wxUSE_COMMON_DIALOGS || defined(__WXWINE__)
+#if wxUSE_COMMON_DIALOGS
     #include <commdlg.h>
 #endif
Index: src/msw/dde.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/dde.cpp,v
retrieving revision 1.27
diff -u -u -2 -r1.27 dde.cpp
--- src/msw/dde.cpp	2002/09/03 11:22:56	1.27
+++ src/msw/dde.cpp	2003/01/07 17:46:57
@@ -47,8 +47,4 @@
 #include <ddeml.h>
 
-#ifdef __WXWINE__
-#define PCONVCONTEXT CONVCONTEXT*
-#endif
-
 #if defined(__TWIN32__) || defined(__GNUWIN32_OLD__)
     #include "wx/msw/gnuwin32/extra.h"
Index: src/msw/dialup.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/dialup.cpp,v
retrieving revision 1.35
diff -u -u -2 -r1.35 dialup.cpp
--- src/msw/dialup.cpp	2003/01/07 13:49:00	1.35
+++ src/msw/dialup.cpp	2003/01/07 17:46:59
@@ -55,5 +55,4 @@
     !defined(__GNUWIN32_OLD__) && \
     !defined(__WATCOMC__) && \
-    !defined(__WXWINE__) && \
     !defined(__WINE__) && \
     (!defined(__VISUALC__) || (__VISUALC__ >= 1020))
Index: src/msw/dib.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/dib.cpp,v
retrieving revision 1.32
diff -u -u -2 -r1.32 dib.cpp
--- src/msw/dib.cpp	2002/04/16 08:57:36	1.32
+++ src/msw/dib.cpp	2003/01/07 17:47:00
@@ -354,5 +354,5 @@
 {
         DWORD     ulT = ul;
-#if defined(WINNT) || defined(__WIN32__) || defined(__WIN32__) || defined(__WXWINE__)
+#if defined(WINNT) || defined(__WIN32__) || defined(__WIN32__)
         BYTE *hp = (BYTE *) pv;
 #else
@@ -384,5 +384,5 @@
 {
         DWORD     ulT = ul;
-#if defined(WINNT) || defined(__WIN32__) || defined(__WIN32__) || defined(__WXWINE__)
+#if defined(WINNT) || defined(__WIN32__) || defined(__WIN32__)
         BYTE *hp = (BYTE *) pv;
 #else
@@ -640,7 +640,4 @@
 HPALETTE wxMakeDIBPalette(LPBITMAPINFOHEADER lpInfo)
 {
-#ifdef __WXWINE__
-        return (FALSE);
-#else
     LPLOGPALETTE npPal;
     RGBQUAD far *lpRGB;
@@ -690,6 +687,4 @@
     else
         return((HPALETTE) GetStockObject(DEFAULT_PALETTE));
-#endif
-
 }
 
Index: src/msw/dibutils.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/dibutils.cpp,v
retrieving revision 1.21
diff -u -u -2 -r1.21 dibutils.cpp
--- src/msw/dibutils.cpp	2002/08/30 20:34:25	1.21
+++ src/msw/dibutils.cpp	2003/01/07 17:47:01
@@ -33,10 +33,4 @@
 #include "wx/msw/dibutils.h"
 
-#ifdef __WXWINE__
-/* Why module.h? No longer finds this header.
-   #include <module.h>
-*/
-#endif
-
 #if defined(__WIN32__)
 #if !defined(__MWERKS__) && !defined(__SALFORDC__)
@@ -116,9 +110,5 @@
 
   // *** Create, select, realize, deselect, and delete the palette
-#ifdef __WXWINE__
-  ScreenDC = GetDC((HWND)NULL);
-#else
   ScreenDC = GetDC(NULL);
-#endif
   ScreenPalette = CreatePalette((LOGPALETTE *)&Palette);
 
@@ -131,9 +121,5 @@
   }
 
-#ifdef __WXWINE__
-  nOK = ReleaseDC((HWND)NULL, ScreenDC);
-#else
   nOK = ReleaseDC(NULL, ScreenDC);
-#endif
 
   return;
Index: src/msw/enhmeta.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/enhmeta.cpp,v
retrieving revision 1.8
diff -u -u -2 -r1.8 enhmeta.cpp
--- src/msw/enhmeta.cpp	2002/08/30 20:34:25	1.8
+++ src/msw/enhmeta.cpp	2003/01/07 17:47:01
@@ -194,12 +194,5 @@
 
         // CreateEnhMetaFile() wants them in HIMETRIC
-#ifdef __WXWINE__
-        LONG r, b;
-        PixelToHIMETRIC(&r, &b);
-        rect.right = r;
-        rect.bottom = b;
-#else
         PixelToHIMETRIC(&rect.right, &rect.bottom);
-#endif
         
         pRect = &rect;
@@ -302,8 +295,4 @@
     else
     {
-#ifdef __WXWINE__
-        wxFAIL_MSG( _T("unsupported format") );
-        return FALSE;
-#else
         wxASSERT_MSG( format == wxDF_METAFILE, _T("unsupported format") );
 
@@ -347,5 +336,4 @@
 
         PixelToHIMETRIC(&mfpict->xExt, &mfpict->yExt);
-#endif
     }
 
Index: src/msw/fdrepdlg.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/fdrepdlg.cpp,v
retrieving revision 1.8
diff -u -u -2 -r1.8 fdrepdlg.cpp
--- src/msw/fdrepdlg.cpp	2003/01/02 23:38:07	1.8
+++ src/msw/fdrepdlg.cpp	2003/01/07 17:47:02
@@ -38,5 +38,5 @@
 #include "wx/msw/private.h"
 
-#if !defined(__WIN32__) || defined(__SALFORDC__) || defined(__WXWINE__)
+#if !defined(__WIN32__) || defined(__SALFORDC__)
     #include <commdlg.h>
 #endif
Index: src/msw/filedlg.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/filedlg.cpp,v
retrieving revision 1.47
diff -u -u -2 -r1.47 filedlg.cpp
--- src/msw/filedlg.cpp	2002/12/04 14:09:12	1.47
+++ src/msw/filedlg.cpp	2003/01/07 17:47:03
@@ -44,5 +44,5 @@
 #include "wx/msw/private.h"
 
-#if !defined(__WIN32__) || defined(__SALFORDC__) || defined(__WXWINE__)
+#if !defined(__WIN32__) || defined(__SALFORDC__)
     #include <commdlg.h>
 #endif
Index: src/msw/fontdlg.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/fontdlg.cpp,v
retrieving revision 1.15
diff -u -u -2 -r1.15 fontdlg.cpp
--- src/msw/fontdlg.cpp	2002/08/30 20:34:25	1.15
+++ src/msw/fontdlg.cpp	2003/01/07 17:47:03
@@ -39,5 +39,5 @@
 #include "wx/fontdlg.h"
 
-#if !defined(__WIN32__) || defined(__SALFORDC__) || defined(__WXWINE__)
+#if !defined(__WIN32__) || defined(__SALFORDC__)
 #include <windows.h>
 #include <commdlg.h>
Index: src/msw/fontenum.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/fontenum.cpp,v
retrieving revision 1.31
diff -u -u -2 -r1.31 fontenum.cpp
--- src/msw/fontenum.cpp	2003/01/07 13:49:00	1.31
+++ src/msw/fontenum.cpp	2003/01/07 17:47:04
@@ -149,12 +149,6 @@
 }
 
-#if defined(__WXWINE__)
-    #define wxFONTENUMPROC FONTENUMPROCEX
-#elif (defined(__GNUWIN32__) && !defined(__CYGWIN10__) && !wxCHECK_W32API_VERSION( 1, 1 ))
-    #if wxUSE_NORLANDER_HEADERS
-        #define wxFONTENUMPROC FONTENUMPROC
-    #else
-        #define wxFONTENUMPROC int(*)(ENUMLOGFONTEX *, NEWTEXTMETRICEX*, int, LPARAM)
-    #endif
+#if defined(__GNUWIN32__) && !defined(__CYGWIN10__) && !wxCHECK_W32API_VERSION( 1, 1 ) && !wxUSE_NORLANDER_HEADERS
+    #define wxFONTENUMPROC int(*)(ENUMLOGFONTEX *, NEWTEXTMETRICEX*, int, LPARAM)
 #else
     #define wxFONTENUMPROC FONTENUMPROC
Index: src/msw/listbox.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/listbox.cpp,v
retrieving revision 1.70
diff -u -u -2 -r1.70 listbox.cpp
--- src/msw/listbox.cpp	2002/12/19 23:12:28	1.70
+++ src/msw/listbox.cpp	2003/01/07 17:47:05
@@ -37,10 +37,4 @@
 #include <windowsx.h>
 
-#ifdef __WXWINE__
-  #if defined(GetWindowStyle)
-    #undef GetWindowStyle
-  #endif
-#endif
-
 #include "wx/dynarray.h"
 #include "wx/log.h"
@@ -54,31 +48,4 @@
         #include "wx/msw/gnuwin32/extra.h"
     #endif
-#endif
-
-#ifdef __WXWINE__
-  #ifndef ListBox_SetItemData
-    #define ListBox_SetItemData(hwndCtl, index, data) \
-      ((int)(DWORD)SendMessage((hwndCtl), LB_SETITEMDATA, (WPARAM)(int)(index), (LPARAM)(data)))
-  #endif
-  #ifndef ListBox_GetHorizontalExtent
-    #define ListBox_GetHorizontalExtent(hwndCtl) \
-      ((int)(DWORD)SendMessage((hwndCtl), LB_GETHORIZONTALEXTENT, 0L, 0L))
-  #endif
-  #ifndef ListBox_GetSelCount
-    #define ListBox_GetSelCount(hwndCtl) \
-      ((int)(DWORD)SendMessage((hwndCtl), LB_GETSELCOUNT, 0L, 0L))
-  #endif
-  #ifndef ListBox_GetSelItems
-    #define ListBox_GetSelItems(hwndCtl, cItems, lpItems) \
-      ((int)(DWORD)SendMessage((hwndCtl), LB_GETSELITEMS, (WPARAM)(int)(cItems), (LPARAM)(int *)(lpItems)))
-  #endif
-  #ifndef ListBox_GetTextLen
-    #define ListBox_GetTextLen(hwndCtl, index) \
-      ((int)(DWORD)SendMessage((hwndCtl), LB_GETTEXTLEN, (WPARAM)(int)(index), 0L))
-  #endif
-  #ifndef ListBox_GetText
-    #define ListBox_GetText(hwndCtl, index, lpszBuffer) \
-      ((int)(DWORD)SendMessage((hwndCtl), LB_GETTEXT, (WPARAM)(int)(index), (LPARAM)(LPCTSTR)(lpszBuffer)))
-  #endif
 #endif
 
Index: src/msw/listctrl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/listctrl.cpp,v
retrieving revision 1.144
diff -u -u -2 -r1.144 listctrl.cpp
--- src/msw/listctrl.cpp	2003/01/02 23:38:07	1.144
+++ src/msw/listctrl.cpp	2003/01/07 17:47:08
@@ -120,10 +120,5 @@
         if ( (item.mask & LVIF_TEXT) && item.pszText )
         {
-#ifdef __WXWINE__
-            // FIXME
-            m_buf = new wxWC2WXbuf(wxConvLocal.cWC2WX((const __wchar_t* ) item.pszText));
-#else
             m_buf = new wxWC2WXbuf(wxConvLocal.cWC2WX(item.pszText));
-#endif
             m_item->pszText = (wxChar*)m_buf->data();
         }
Index: src/msw/main.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/main.cpp,v
retrieving revision 1.13
diff -u -u -2 -r1.13 main.cpp
--- src/msw/main.cpp	2002/12/17 10:52:05	1.13
+++ src/msw/main.cpp	2003/01/07 17:47:08
@@ -73,13 +73,9 @@
 #if !defined(_WINDLL)
 
-#if defined(__WXWINE__)
-    #define HINSTANCE HINSTANCE__*
-
-    extern "C"
-#elif defined(__TWIN32__) || defined(__WXMICROWIN__)
+#if defined(__TWIN32__) || defined(__WXMICROWIN__)
     #define HINSTANCE HANDLE
 
     extern "C"
-#endif // WINE
+#endif
 
 int PASCAL WinMain(HINSTANCE hInstance,
Index: src/msw/palette.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/palette.cpp,v
retrieving revision 1.7
diff -u -u -2 -r1.7 palette.cpp
--- src/msw/palette.cpp	2001/09/30 22:06:39	1.7
+++ src/msw/palette.cpp	2003/01/07 17:47:08
@@ -74,5 +74,5 @@
   UnRef();
 
-#if defined(__WXWINE__) || defined(__WXMICROWIN__)
+#if defined(__WXMICROWIN__)
 
   return (FALSE);
Index: src/msw/textctrl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/textctrl.cpp,v
retrieving revision 1.157
diff -u -u -2 -r1.157 textctrl.cpp
--- src/msw/textctrl.cpp	2003/01/07 01:09:08	1.157
+++ src/msw/textctrl.cpp	2003/01/07 17:47:11
@@ -596,13 +596,4 @@
 extern long wxEncodingToCodepage(wxFontEncoding encoding); // from utils.cpp
 
-#ifdef __WXWINE__
-bool wxTextCtrl::StreamIn(const wxString& value,
-                          wxFontEncoding WXUNUSED(encoding),
-                          bool selectionOnly)
-{
-    return FALSE;
-}
-#else // !__WXWINE__
-
 #if wxUSE_UNICODE_MSLU
 bool wxTextCtrl::StreamIn(const wxString& value,
@@ -677,6 +668,4 @@
 }
 
-#endif // __WXWINE__/!__WXWINE__
- 
 #endif // wxUSE_RICHEDIT
 
@@ -716,5 +705,5 @@
 #endif // wxUSE_UNICODE_MSLU
 
-#if !wxUSE_UNICODE && !defined(__WXWINE__)
+#if !wxUSE_UNICODE
         // next check if the text we're inserting must be shown in a non
         // default charset -- this only works for RichEdit > 1.0
@@ -1759,5 +1748,4 @@
 bool wxTextCtrl::MSWOnNotify(int idCtrl, WXLPARAM lParam, WXLPARAM *result)
 {
-#ifndef __WXWINE__
     NMHDR *hdr = (NMHDR* )lParam;
     switch ( hdr->code )
@@ -1846,5 +1834,4 @@
             return TRUE;
     }
-#endif
     
     // not processed, leave it to the base class
@@ -1904,7 +1891,4 @@
 bool wxTextCtrl::SetStyle(long start, long end, const wxTextAttr& style)
 {
-#ifdef __WXWINE__
-    return FALSE;
-#else
     if ( !IsRich() )
     {
@@ -2036,5 +2020,4 @@
 
     return ok;
-#endif
 }
 
Index: src/msw/utils.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/utils.cpp,v
retrieving revision 1.103
diff -u -u -2 -r1.103 utils.cpp
--- src/msw/utils.cpp	2003/01/02 23:38:08	1.103
+++ src/msw/utils.cpp	2003/01/07 17:47:13
@@ -49,5 +49,5 @@
 #include "wx/timer.h"
 
-#if !defined(__GNUWIN32__) && !defined(__WXWINE__) && !defined(__SALFORDC__) && !defined(__WXMICROWIN__)
+#if !defined(__GNUWIN32__) && !defined(__SALFORDC__) && !defined(__WXMICROWIN__)
     #include <direct.h>
 
@@ -79,5 +79,5 @@
 #endif // USE_NET_API
 
-#if defined(__WIN32__) && !defined(__WXWINE__) && !defined(__WXMICROWIN__)
+#if defined(__WIN32__) && !defined(__WXMICROWIN__)
     #include <io.h>
 
@@ -888,5 +888,4 @@
                                  TOKEN_ADJUST_PRIVILEGES | TOKEN_QUERY,
                                  &hToken) != 0;
-#ifndef __WXWINE__
         if ( bOK )
         {
@@ -907,5 +906,4 @@
             bOK = ::GetLastError() == ERROR_SUCCESS;
         }
-#endif
     }
 
Index: src/msw/utilsexc.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/utilsexc.cpp,v
retrieving revision 1.55
diff -u -u -2 -r1.55 utilsexc.cpp
--- src/msw/utilsexc.cpp	2003/01/02 23:38:09	1.55
+++ src/msw/utilsexc.cpp	2003/01/07 17:47:14
@@ -48,5 +48,5 @@
 #include <ctype.h>
 
-#if !defined(__GNUWIN32__) && !defined(__WXWINE__) && !defined(__SALFORDC__) && !defined(__WXMICROWIN__)
+#if !defined(__GNUWIN32__) && !defined(__SALFORDC__) && !defined(__WXMICROWIN__)
     #include <direct.h>
 #ifndef __MWERKS__
@@ -60,5 +60,5 @@
 #endif
 
-#if defined(__WIN32__) && !defined(__WXWINE__) && !defined(__WXMICROWIN__)
+#if defined(__WIN32__) && !defined(__WXMICROWIN__)
 #include <io.h>
 
@@ -348,8 +348,4 @@
 bool wxPipeInputStream::CanRead() const
 {
-    // FIXME
-#ifdef __WXWINE__
-    return FALSE;
-#else // !Wine
     if ( !IsOpened() )
         return FALSE;
@@ -388,5 +384,4 @@
 
     return nAvailable != 0;
-#endif // Wine/!Wine
 }
 
Index: src/msw/window.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/window.cpp,v
retrieving revision 1.352
diff -u -u -2 -r1.352 window.cpp
--- src/msw/window.cpp	2003/01/03 12:14:59	1.352
+++ src/msw/window.cpp	2003/01/07 17:47:22
@@ -1464,12 +1464,8 @@
 void wxWindowMSW::Update()
 {
-#ifdef __WXWINE__
-    ::UpdateWindow(GetHwnd());
-#else
     if ( !::UpdateWindow(GetHwnd()) )
     {
         wxLogLastError(_T("UpdateWindow"));
     }
-#endif
 
 #if defined(__WIN32__) && !defined(__WXMICROWIN__)
Index: src/msw/ole/automtn.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/ole/automtn.cpp,v
retrieving revision 1.27
diff -u -u -2 -r1.27 automtn.cpp
--- src/msw/ole/automtn.cpp	2003/01/07 13:49:05	1.27
+++ src/msw/ole/automtn.cpp	2003/01/07 17:47:23
@@ -25,5 +25,5 @@
 // Watcom C++ gives a linker error if this is compiled in.
 // With Borland C++, all samples crash if this is compiled in.
-#if wxUSE_OLE &&!defined(__WATCOMC__) && !(defined(__BORLANDC__) && (__BORLANDC__ < 0x520)) && !defined(__CYGWIN10__) && !defined(__WXWINE__)
+#if wxUSE_OLE &&!defined(__WATCOMC__) && !(defined(__BORLANDC__) && (__BORLANDC__ < 0x520)) && !defined(__CYGWIN10__)
 
 #define _FORCENAMELESSUNION
Index: src/msw/ole/dataobj.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/ole/dataobj.cpp,v
retrieving revision 1.59
diff -u -u -2 -r1.59 dataobj.cpp
--- src/msw/ole/dataobj.cpp	2003/01/02 23:38:11	1.59
+++ src/msw/ole/dataobj.cpp	2003/01/07 17:47:25
@@ -56,8 +56,4 @@
 #include "wx/msw/dib.h"
 
-#ifdef __WXWINE__
-#define LPDROPFILES DROPFILES*
-#endif
-
 #ifndef CFSTR_SHELLURL
 #define CFSTR_SHELLURL _T("UniformResourceLocator")
Index: src/msw/ole/uuid.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/ole/uuid.cpp,v
retrieving revision 1.12
diff -u -u -2 -r1.12 uuid.cpp
--- src/msw/ole/uuid.cpp	2002/08/30 20:34:27	1.12
+++ src/msw/ole/uuid.cpp	2003/01/07 17:47:25
@@ -137,8 +137,4 @@
 bool Uuid::Set(const wxChar *pc)
 {
-#ifdef __WXWINE__
-    wxFAIL_MSG(_T("UUid::Set not implemented"));
-    return FALSE;
-#else
   // get UUID from string
 #ifdef _UNICODE
@@ -161,5 +157,4 @@
 
   return TRUE;
-#endif
 }
 

 	  	 
