Index: src/xrc/xh_hyperlink.cpp
===================================================================
--- src/xrc/xh_hyperlink.cpp	(revision 54463)
+++ src/xrc/xh_hyperlink.cpp	(working copy)
@@ -64,11 +64,13 @@
 {
     XRC_MAKE_INSTANCE(control, wxHyperlinkCtrl)
 
-    control->Create(m_parentAsWindow, GetID(),
-        GetParamValue(wxT("label")), GetParamValue(wxT("url")),
-        GetPosition(), GetSize(),
-        GetStyle(wxT("style"), wxHL_DEFAULT_STYLE),
-        GetName());
+    control->Create(m_parentAsWindow,
+                    GetID(),
+                    GetText(wxT("label")),
+                    GetText(wxT("url")),
+                    GetPosition(), GetSize(),
+                    GetStyle(wxT("style"), wxHL_DEFAULT_STYLE),
+                    GetName());
 
     SetupWindow(control);
 
