Index: auidemo.cpp
===================================================================
--- auidemo.cpp	(revision 67750)
+++ auidemo.cpp	(working copy)
@@ -59,6 +59,7 @@
     enum
     {
         ID_CreateTree = wxID_HIGHEST+1,
+		ID_Qq,
         ID_CreateGrid,
         ID_CreateText,
         ID_CreateHTML,
@@ -126,6 +127,7 @@
     wxTextCtrl* CreateTextCtrl(const wxString& text = wxEmptyString);
     wxGrid* CreateGrid();
     wxTreeCtrl* CreateTreeCtrl();
+    wxWindow* qq;
     wxSizeReportCtrl* CreateSizeReportCtrl(int width = 80, int height = 80);
     wxPoint GetStartPosition();
     wxHtmlWindow* CreateHTMLCtrl(wxWindow* parent = NULL);
@@ -138,6 +140,7 @@
     void OnEraseBackground(wxEraseEvent& evt);
     void OnSize(wxSizeEvent& evt);
 
+    void OnQq(wxCommandEvent& evt);
     void OnCreateTree(wxCommandEvent& evt);
     void OnCreateGrid(wxCommandEvent& evt);
     void OnCreateHTML(wxCommandEvent& evt);
@@ -579,6 +582,7 @@
     EVT_ERASE_BACKGROUND(MyFrame::OnEraseBackground)
     EVT_SIZE(MyFrame::OnSize)
     EVT_MENU(MyFrame::ID_CreateTree, MyFrame::OnCreateTree)
+    EVT_MENU(MyFrame::ID_Qq, MyFrame::OnQq)
     EVT_MENU(MyFrame::ID_CreateGrid, MyFrame::OnCreateGrid)
     EVT_MENU(MyFrame::ID_CreateText, MyFrame::OnCreateText)
     EVT_MENU(MyFrame::ID_CreateHTML, MyFrame::OnCreateHTML)
@@ -684,6 +688,7 @@
     wxMenu* view_menu = new wxMenu;
     view_menu->Append(ID_CreateText, _("Create Text Control"));
     view_menu->Append(ID_CreateHTML, _("Create HTML Control"));
+    view_menu->Append(ID_Qq, _("Hide/unhide qq"));
     view_menu->Append(ID_CreateTree, _("Create Tree"));
     view_menu->Append(ID_CreateGrid, _("Create Grid"));
     view_menu->Append(ID_CreateNotebook, _("Create Notebook"));
@@ -1005,6 +1010,8 @@
     m_perspectives.Add(perspective_default);
     m_perspectives.Add(perspective_all);
 
+	m_mgr.AddPane(qq=new wxTextCtrl(this,wxID_ANY,"qq"), wxAuiPaneInfo().Name("qq").Caption(_("Qq")).CloseButton(true).Bottom());
+
     // "commit" all changes made to wxAuiManager
     m_mgr.Update();
 }
@@ -1400,6 +1407,12 @@
     return wxPoint(pt.x + x, pt.y + x);
 }
 
+void MyFrame::OnQq(wxCommandEvent& WXUNUSED(event))
+{
+	m_mgr.GetPane("qq").Show(!qq->IsShown());
+    m_mgr.Update();
+}
+
 void MyFrame::OnCreateTree(wxCommandEvent& WXUNUSED(event))
 {
     m_mgr.AddPane(CreateTreeCtrl(), wxAuiPaneInfo().
