diff -ur wxWidgets.orig/docs/latex/wx/xmlresh.tex wxWidgets/docs/latex/wx/xmlresh.tex
--- wxWidgets.orig/docs/latex/wx/xmlresh.tex	Fri May  5 10:27:26 2006
+++ wxWidgets/docs/latex/wx/xmlresh.tex	Thu Jul 13 19:15:20 2006
@@ -142,6 +142,12 @@
 
 Returns an icon.
 
+\membersection{wxXmlResourceHandler::GetImage}\label{wxxmlresourcehandlergetimage}
+
+\func{wxImage}{GetImage}{\param{const wxString\& }{param = wxT("image")}, \param{wxSize\& }{size = wxDefaultSize}}
+
+Returns a wxImage.
+
 \membersection{wxXmlResourceHandler::GetLong}\label{wxxmlresourcehandlergetlong}
 
 \func{long}{GetLong}{\param{const wxString\& }{param}, \param{long }{defaultv = 0}}
diff -ur wxWidgets.orig/include/wx/xrc/xh_bmp.h wxWidgets/include/wx/xrc/xh_bmp.h
--- wxWidgets.orig/include/wx/xrc/xh_bmp.h	Sun Jul  9 22:07:19 2006
+++ wxWidgets/include/wx/xrc/xh_bmp.h	Sun Jul  9 22:46:00 2006
@@ -32,5 +32,13 @@
     virtual bool CanHandle(wxXmlNode *node);
 };
 
+class WXDLLIMPEXP_XRC wxImageXmlHandler : public wxXmlResourceHandler
+{
+DECLARE_DYNAMIC_CLASS(wxImageXmlHandler)
+public:
+    wxImageXmlHandler();
+    virtual wxObject *DoCreateResource();
+    virtual bool CanHandle(wxXmlNode *node);
+};
 
 #endif // _WX_XH_BMP_H_
diff -ur wxWidgets.orig/include/wx/xrc/xmlres.h wxWidgets/include/wx/xrc/xmlres.h
--- wxWidgets.orig/include/wx/xrc/xmlres.h	Sun Jul  9 22:07:21 2006
+++ wxWidgets/include/wx/xrc/xmlres.h	Sun Jul  9 22:46:00 2006
@@ -22,6 +22,7 @@
 #include "wx/gdicmn.h"
 #include "wx/filesys.h"
 #include "wx/bitmap.h"
+#include "wx/image.h"
 #include "wx/icon.h"
 #include "wx/artprov.h"
 
@@ -203,6 +204,9 @@
     // Loads a bitmap resource from a file.
     wxBitmap LoadBitmap(const wxString& name);
 
+    // Loads an image resource from a file.
+    wxImage LoadImage(const wxString& name);
+    
     // Loads an icon resource from a file.
     wxIcon LoadIcon(const wxString& name);
 
@@ -433,6 +437,10 @@
                        const wxArtClient& defaultArtClient = wxART_OTHER,
                        wxSize size = wxDefaultSize);
 
+    // Gets an image.
+    wxImage GetImage(const wxString& param = wxT("image"),
+		     wxSize size = wxDefaultSize);
+    
     // Gets an icon.
     wxIcon GetIcon(const wxString& param = wxT("icon"),
                    const wxArtClient& defaultArtClient = wxART_OTHER,
diff -ur wxWidgets.orig/src/xrc/xh_bmp.cpp wxWidgets/src/xrc/xh_bmp.cpp
--- wxWidgets.orig/src/xrc/xh_bmp.cpp	Sun Jul  9 22:11:10 2006
+++ wxWidgets/src/xrc/xh_bmp.cpp	Sun Jul  9 22:46:00 2006
@@ -61,4 +61,23 @@
     return IsOfClass(node, wxT("wxIcon"));
 }
 
+IMPLEMENT_DYNAMIC_CLASS(wxImageXmlHandler, wxXmlResourceHandler)
+
+wxImageXmlHandler::wxImageXmlHandler()
+: wxXmlResourceHandler()
+{
+}
+
+wxObject *wxImageXmlHandler::DoCreateResource()
+{
+    // NB: empty parameter name means "take directly from this node's next
+    //     instead of from subnode with given name"
+    return new wxImage(GetImage(wxEmptyString));
+}
+
+bool wxImageXmlHandler::CanHandle(wxXmlNode *node)
+{
+    return IsOfClass(node, wxT("wxImage"));
+}
+
 #endif // wxUSE_XRC
diff -ur wxWidgets.orig/src/xrc/xmlres.cpp wxWidgets/src/xrc/xmlres.cpp
--- wxWidgets.orig/src/xrc/xmlres.cpp	Sun Jul  9 22:11:17 2006
+++ wxWidgets/src/xrc/xmlres.cpp	Sun Jul  9 22:46:00 2006
@@ -297,6 +297,16 @@
     return rt;
 }
 
+wxImage wxXmlResource::LoadImage(const wxString& name)
+{
+    wxImage *img = (wxImage*)CreateResFromNode(
+                             FindResource(name, wxT("wxImage")), NULL, NULL);
+    wxImage rt;
+
+    if (img) { rt = *img; delete img; }
+    return rt;
+}
+
 wxIcon wxXmlResource::LoadIcon(const wxString& name)
 {
     wxIcon *icon = (wxIcon*)CreateResFromNode(
@@ -1118,6 +1128,35 @@
     return wxBitmap(img);
 }
 
+wxImage wxXmlResourceHandler::GetImage(const wxString& param,
+                                       wxSize size)
+{
+    /* Load the image from file: */
+    wxString name = GetParamValue(param);
+    if (name.empty()) return wxNullImage;
+#if wxUSE_FILESYSTEM
+    wxFSFile *fsfile = GetCurFileSystem().OpenFile(name);
+    if (fsfile == NULL)
+    {
+        wxLogError(_("XRC resource: Cannot create image from '%s'."),
+                   name.c_str());
+        return wxNullImage;
+    }
+    wxImage img(*(fsfile->GetStream()));
+    delete fsfile;
+#else
+    wxImage img(name);
+#endif
+
+    if (!img.Ok())
+    {
+        wxLogError(_("XRC resource: Cannot create image from '%s'."),
+                   name.c_str());
+        return wxNullImage;
+    }
+    if (!(size == wxDefaultSize)) img.Rescale(size.x, size.y);
+    return img;
+}
 
 
 wxIcon wxXmlResourceHandler::GetIcon(const wxString& param,
diff -ur wxWidgets.orig/src/xrc/xmlrsall.cpp wxWidgets/src/xrc/xmlrsall.cpp
--- wxWidgets.orig/src/xrc/xmlrsall.cpp	Sun Jul  9 22:11:17 2006
+++ wxWidgets/src/xrc/xmlrsall.cpp	Sun Jul  9 22:46:00 2006
@@ -23,6 +23,7 @@
 void wxXmlResource::InitAllHandlers()
 {
     AddHandler(new wxBitmapXmlHandler);
+    AddHandler(new wxImageXmlHandler);
     AddHandler(new wxIconXmlHandler);
     AddHandler(new wxMenuXmlHandler);
     AddHandler(new wxMenuBarXmlHandler);

 	  	 
