Index: src/msw/dc.cpp
===================================================================
--- src/msw/dc.cpp	(revision 51611)
+++ src/msw/dc.cpp	(working copy)
@@ -1782,7 +1782,12 @@
         wxLogLastError(_T("GetTextExtentPoint32()"));
     }
 
-#if !defined(_WIN32_WCE) || (_WIN32_WCE >= 400)
+#if !defined(_WIN32_WCE)
+    // ::GetCharABCWidths( GetHdc(), 'I', 'I', &width ) inexplicably hangs some WinCE devices
+    // Samsung i730 Intel PXA272 4.21.1088 (Build 15045.2.6.0), GetCharABCWidths works great
+    // HP iPaq hx4705 Intel PXA270 4.21.1088 (Build 15045.2.6.0), GetCharABCWidths hangs the device
+    // Dell Axim X50V 4.21.1088 (Build ???), GetCharABCWidths hangs the device
+
     // the result computed by GetTextExtentPoint32() may be too small as it
     // accounts for under/overhang of the first/last character while we want
     // just the bounding rect for this string so adjust the width as needed
@@ -1808,7 +1813,7 @@
         }
         //else: GetCharABCWidths() failed, not a TrueType font?
     }
-#endif // !defined(_WIN32_WCE) || (_WIN32_WCE >= 400)
+#endif // !defined(_WIN32_WCE)
 
     TEXTMETRIC tm;
     ::GetTextMetrics(GetHdc(), &tm);

 	  	 
