Index: htmllbox.h
===================================================================
--- htmllbox.h	(revision 60071)
+++ htmllbox.h	(working copy)
@@ -257,7 +257,7 @@
     // accessing strings
     // -----------------
 
-    virtual unsigned int GetCount() const
+    virtual size_t GetCount() const
         { return m_items.GetCount(); }
 
     virtual wxString GetString(unsigned int n) const;
