From ef4319a1eb10b09e9ff0d3e88a6e674fe6816aba Mon Sep 17 00:00:00 2001
From: Scott Talbert <swt@techie.net>
Date: Tue, 3 Mar 2015 23:33:00 -0500
Subject: [PATCH] Ensure the PangoFontFace type is loaded

---
 src/gtk/fontdlg.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/gtk/fontdlg.cpp b/src/gtk/fontdlg.cpp
index bdf2dd4..09122f7 100644
--- a/src/gtk/fontdlg.cpp
+++ b/src/gtk/fontdlg.cpp
@@ -78,6 +78,7 @@ bool wxFontDialog::DoCreate(wxWindow *parent)
         gtk_parent = GTK_WINDOW(parent->m_widget);
 
 #if GTK_CHECK_VERSION(3,2,0)
+    g_type_ensure(PANGO_TYPE_FONT_FACE);
     if (gtk_check_version(3,2,0) == NULL)
         m_widget = gtk_font_chooser_dialog_new(wxGTK_CONV(message), gtk_parent);
     else
-- 
2.1.0

