diff -up wxPython-src-3.0.2.0/wxPython/src/__init__.py.dlopenflags wxPython-src-3.0.2.0/wxPython/src/__init__.py
--- wxPython-src-3.0.2.0/wxPython/src/__init__.py.dlopenflags	2013-02-27 15:14:01.000000000 -0500
+++ wxPython-src-3.0.2.0/wxPython/src/__init__.py	2015-02-28 20:14:07.653903689 -0500
@@ -41,10 +41,31 @@ __all__ = [
     'stc',
     ]
 
+# GTK3 expects all of its library symbols to be loaded into the global
+# namespace.  Temporarily change the dlopen flags to include RTLD_GLOBAL before
+# importing the core module so it will pull its libraries into the global NS. 
+import sys
+_RTLD_GLOBAL = 0
+try:
+    from dl import RTLD_GLOBAL as _RTLD_GLOBAL
+except ImportError:
+    try:
+        from DLFCN import RTLD_GLOBAL as _RTLD_GLOBAL
+    except ImportError:
+        pass
+
+if _RTLD_GLOBAL != 0:
+    _dlopenflags = sys.getdlopenflags()
+    sys.setdlopenflags(_dlopenflags|_RTLD_GLOBAL)
+
 # Load the package namespace with the core classes and such
 from wx._core import *
 del wx
 
+# If we changed the dlopen flags earlier, change them back.
+if _RTLD_GLOBAL != 0:
+    sys.setdlopenflags(_dlopenflags)
+
 if 'wxMSW' in PlatformInfo:
     __all__ += ['activex']
 
