Index: src/common/graphcmn.cpp
===================================================================
--- src/common/graphcmn.cpp	(revision 67311)
+++ src/common/graphcmn.cpp	(working copy)
@@ -881,9 +881,9 @@
 #endif
 #endif
 
-wxGraphicsContext* wxGraphicsContext::CreateFromNative( void * context )
+wxGraphicsContext* wxGraphicsContext::CreateFromNative( void * context, bool isForPrinting )
 {
-    return wxGraphicsRenderer::GetDefaultRenderer()->CreateContextFromNativeContext(context);
+    return wxGraphicsRenderer::GetDefaultRenderer()->CreateContextFromNativeContext(context, isForPrinting);
 }
 
 wxGraphicsContext* wxGraphicsContext::CreateFromNativeWindow( void * window )
Index: src/msw/graphics.cpp
===================================================================
--- src/msw/graphics.cpp	(revision 67311)
+++ src/msw/graphics.cpp	(working copy)
@@ -456,7 +456,7 @@
     virtual wxGraphicsContext * CreateContext( const wxEnhMetaFileDC& dc);
 #endif
 
-    virtual wxGraphicsContext * CreateContextFromNativeContext( void * context );
+    virtual wxGraphicsContext * CreateContextFromNativeContext( void * context, bool isForPrinting );
 
     virtual wxGraphicsContext * CreateContextFromNativeWindow( void * window );
 
@@ -1906,7 +1906,7 @@
     return new wxGDIPlusMeasuringContext(this);
 }
 
-wxGraphicsContext * wxGDIPlusRenderer::CreateContextFromNativeContext( void * context )
+wxGraphicsContext * wxGDIPlusRenderer::CreateContextFromNativeContext( void * context, bool isForPrinting )
 {
     ENSURE_LOADED_OR_RETURN(NULL);
     return new wxGDIPlusContext(this,(Graphics*) context);
Index: src/osx/cocoa/overlay.mm
===================================================================
--- src/osx/cocoa/overlay.mm	(revision 67311)
+++ src/osx/cocoa/overlay.mm	(working copy)
@@ -151,7 +151,7 @@
     wxGCDCImpl *win_impl = wxDynamicCast(impl,wxGCDCImpl);
     if (win_impl)
     {
-        win_impl->SetGraphicsContext( wxGraphicsContext::CreateFromNative( m_overlayContext ) );
+        win_impl->SetGraphicsContext( wxGraphicsContext::CreateFromNative( m_overlayContext, false ) );
         dc->SetClippingRegion( m_x , m_y , m_width , m_height ) ;
     }
 }
Index: src/osx/core/dcmemory.cpp
===================================================================
--- src/osx/core/dcmemory.cpp	(revision 67311)
+++ src/osx/core/dcmemory.cpp	(working copy)
@@ -85,7 +85,7 @@
         {
             CGContextSetFillColorSpace( bmCtx, genericColorSpace );
             CGContextSetStrokeColorSpace( bmCtx, genericColorSpace );
-            SetGraphicsContext( wxGraphicsContext::CreateFromNative( bmCtx ) );
+            SetGraphicsContext( wxGraphicsContext::CreateFromNative( bmCtx, false ) );
         }
         m_ok = (m_graphicContext != NULL) ;
     }
Index: src/osx/carbon/dcprint.cpp
===================================================================
--- src/osx/carbon/dcprint.cpp	(revision 67311)
+++ src/osx/carbon/dcprint.cpp	(working copy)
@@ -241,7 +241,7 @@
         // have to manually scale them later.
         CGContextScaleCTM( pageContext, 72.0 / (double)m_ppi.x, -72.0 / (double)m_ppi.y);
 
-        impl->SetGraphicsContext( wxGraphicsContext::CreateFromNative( pageContext ) );
+        impl->SetGraphicsContext( wxGraphicsContext::CreateFromNative( pageContext, true ) );
     }
 }
 
Index: src/osx/carbon/graphics.cpp
===================================================================
--- src/osx/carbon/graphics.cpp	(revision 67311)
+++ src/osx/carbon/graphics.cpp	(working copy)
@@ -1374,15 +1374,15 @@
 class WXDLLEXPORT wxMacCoreGraphicsContext : public wxGraphicsContext
 {
 public:
-    wxMacCoreGraphicsContext( wxGraphicsRenderer* renderer, CGContextRef cgcontext, wxDouble width = 0, wxDouble height = 0 );
+    wxMacCoreGraphicsContext( wxGraphicsRenderer* renderer, bool isForPrinting, CGContextRef cgcontext, wxDouble width = 0, wxDouble height = 0 );
 
 #if wxOSX_USE_CARBON
-    wxMacCoreGraphicsContext( wxGraphicsRenderer* renderer, WindowRef window );
+    wxMacCoreGraphicsContext( wxGraphicsRenderer* renderer, bool isForPrinting, WindowRef window );
 #endif
 
-    wxMacCoreGraphicsContext( wxGraphicsRenderer* renderer, wxWindow* window );
+    wxMacCoreGraphicsContext( wxGraphicsRenderer* renderer, bool isForPrinting, wxWindow* window );
 
-    wxMacCoreGraphicsContext( wxGraphicsRenderer* renderer);
+    wxMacCoreGraphicsContext( wxGraphicsRenderer* renderer );
 
     wxMacCoreGraphicsContext();
 
@@ -1460,6 +1460,8 @@
 
     virtual bool ShouldOffset() const
     {
+        if (!m_should_offset) return false;
+        
         int penwidth = 0 ;
         if ( !m_pen.IsNull() )
         {
@@ -1493,6 +1496,9 @@
     DECLARE_DYNAMIC_CLASS_NO_COPY(wxMacCoreGraphicsContext)
 
 private:
+    
+    bool m_should_offset;
+    
     bool EnsureIsValid();
 
     virtual void DoDrawText( const wxString &str, wxDouble x, wxDouble y );
@@ -1566,6 +1572,7 @@
 {
     m_cgContext = NULL;
     m_contextSynthesized = false;
+    m_should_offset = false;
     m_width = 0;
     m_height = 0;
 #if wxOSX_USE_CARBON
@@ -1577,26 +1584,29 @@
     m_invisible = false;
 }
 
-wxMacCoreGraphicsContext::wxMacCoreGraphicsContext( wxGraphicsRenderer* renderer, CGContextRef cgcontext, wxDouble width, wxDouble height ) : wxGraphicsContext(renderer)
+wxMacCoreGraphicsContext::wxMacCoreGraphicsContext( wxGraphicsRenderer* renderer, bool isForPrinting, CGContextRef cgcontext, wxDouble width, wxDouble height ) : wxGraphicsContext(renderer)
 {
     Init();
     SetNativeContext(cgcontext);
     m_width = width;
     m_height = height;
+    m_should_offset = !isForPrinting;
 }
 
 #if wxOSX_USE_CARBON
-wxMacCoreGraphicsContext::wxMacCoreGraphicsContext( wxGraphicsRenderer* renderer, WindowRef window ): wxGraphicsContext(renderer)
+wxMacCoreGraphicsContext::wxMacCoreGraphicsContext( wxGraphicsRenderer* renderer, bool isForPrinting, WindowRef window ): wxGraphicsContext(renderer)
 {
     Init();
     m_windowRef = window;
+    m_should_offset = !isForPrinting;
 }
 #endif
 
-wxMacCoreGraphicsContext::wxMacCoreGraphicsContext( wxGraphicsRenderer* renderer, wxWindow* window ): wxGraphicsContext(renderer)
+wxMacCoreGraphicsContext::wxMacCoreGraphicsContext( wxGraphicsRenderer* renderer, bool isForPrinting, wxWindow* window ): wxGraphicsContext(renderer)
 {
     Init();
 
+    m_should_offset = !isForPrinting;
     wxSize sz = window->GetSize();
     m_width = sz.x;
     m_height = sz.y;
@@ -2676,7 +2686,7 @@
     virtual wxGraphicsContext * CreateContext( const wxPrinterDC& dc);
 #endif
 
-    virtual wxGraphicsContext * CreateContextFromNativeContext( void * context );
+    virtual wxGraphicsContext * CreateContextFromNativeContext( void * context, bool forprinting );
 
     virtual wxGraphicsContext * CreateContextFromNativeWindow( void * window );
 
@@ -2753,7 +2763,7 @@
 
         // having a cgctx being NULL is fine (will be created on demand)
         // this is the case for all wxWindowDCs except wxPaintDC
-        return new wxMacCoreGraphicsContext( this, cgctx, (wxDouble) w, (wxDouble) h );
+        return new wxMacCoreGraphicsContext( this, cgctx, false, (wxDouble) w, (wxDouble) h );
     }
     return NULL;
 }
@@ -2768,7 +2778,7 @@
         int w, h;
         mem_impl->GetSize( &w, &h );
         return new wxMacCoreGraphicsContext( this,
-            (CGContextRef)(mem_impl->GetGraphicsContext()->GetNativeContext()), (wxDouble) w, (wxDouble) h );
+            (CGContextRef)(mem_impl->GetGraphicsContext()->GetNativeContext()), false, (wxDouble) w, (wxDouble) h );
     }
 #endif
     return NULL;
@@ -2784,7 +2794,7 @@
     {
         int w, h;
         print_impl->GetSize( &w, &h );
-        return new wxMacCoreGraphicsContext( this,
+        return new wxMacCoreGraphicsContext( this, true,
             (CGContextRef)(print_impl->GetGraphicsContext()->GetNativeContext()), (wxDouble) w, (wxDouble) h );
     }
 #endif
@@ -2792,9 +2802,9 @@
 }
 #endif
 
-wxGraphicsContext * wxMacCoreGraphicsRenderer::CreateContextFromNativeContext( void * context )
+wxGraphicsContext * wxMacCoreGraphicsRenderer::CreateContextFromNativeContext( void * context, bool forprinting )
 {
-    return new wxMacCoreGraphicsContext(this,(CGContextRef)context);
+    return new wxMacCoreGraphicsContext(this, forprinting, (CGContextRef)context);
 }
 
 wxGraphicsContext * wxMacCoreGraphicsRenderer::CreateContextFromNativeWindow( void * window )
@@ -2809,7 +2819,7 @@
 
 wxGraphicsContext * wxMacCoreGraphicsRenderer::CreateContext( wxWindow* window )
 {
-    return new wxMacCoreGraphicsContext(this, window );
+    return new wxMacCoreGraphicsContext(this, false, window );
 }
 
 wxGraphicsContext * wxMacCoreGraphicsRenderer::CreateMeasuringContext()
Index: src/osx/carbon/dcclient.cpp
===================================================================
--- src/osx/carbon/dcclient.cpp	(revision 67311)
+++ src/osx/carbon/dcclient.cpp	(working copy)
@@ -68,7 +68,7 @@
         if ( window->MacGetLeftBorderSize() != 0 || window->MacGetTopBorderSize() != 0 )
             CGContextTranslateCTM( cg , -window->MacGetLeftBorderSize() , -window->MacGetTopBorderSize() );
 
-        SetGraphicsContext( wxGraphicsContext::CreateFromNative( cg ) );
+        SetGraphicsContext( wxGraphicsContext::CreateFromNative( cg, false ) );
     }
     DoSetClippingRegion( 0 , 0 , m_width , m_height ) ;
 
Index: src/osx/carbon/metafile.cpp
===================================================================
--- src/osx/carbon/metafile.cpp	(revision 67311)
+++ src/osx/carbon/metafile.cpp	(working copy)
@@ -303,7 +303,7 @@
     m_metaFile->UnRef();
     m_metaFile->SetRefData( metafiledata );
 
-    SetGraphicsContext( wxGraphicsContext::CreateFromNative(metafiledata->GetContext()));
+    SetGraphicsContext( wxGraphicsContext::CreateFromNative(metafiledata->GetContext(), true));
     m_ok = (m_graphicContext != NULL) ;
 
     SetMapMode( wxMM_TEXT );
Index: src/osx/carbon/font.cpp
===================================================================
--- src/osx/carbon/font.cpp	(revision 67311)
+++ src/osx/carbon/font.cpp	(working copy)
@@ -719,7 +719,7 @@
 {
 #if wxUSE_GRAPHICS_CONTEXT
     // TODO: consider caching the value
-    wxGraphicsContext* dc = wxGraphicsContext::CreateFromNative((CGContextRef) NULL);
+    wxGraphicsContext* dc = wxGraphicsContext::CreateFromNative((CGContextRef) NULL, false);
     dc->SetFont(*(wxFont *)this,*wxBLACK);
     wxDouble width, height = 0;
     dc->GetTextExtent( wxT("g"), &width, &height, NULL, NULL);
Index: src/generic/graphicc.cpp
===================================================================
--- src/generic/graphicc.cpp	(revision 67311)
+++ src/generic/graphicc.cpp	(working copy)
@@ -1749,7 +1749,7 @@
     virtual wxGraphicsContext * CreateContext( const wxMemoryDC& dc);
     virtual wxGraphicsContext * CreateContext( const wxPrinterDC& dc);
 
-    virtual wxGraphicsContext * CreateContextFromNativeContext( void * context );
+    virtual wxGraphicsContext * CreateContextFromNativeContext( void * context, bool isForPrinting );
 
     virtual wxGraphicsContext * CreateContextFromNativeWindow( void * window );
 
@@ -1831,7 +1831,7 @@
        return NULL;
 }
 
-wxGraphicsContext * wxCairoRenderer::CreateContextFromNativeContext( void * context )
+wxGraphicsContext * wxCairoRenderer::CreateContextFromNativeContext( void * context, bool isForPrinting )
 {
 #ifdef __WXMSW__
     return new wxCairoContext(this,(HDC)context);
Index: include/wx/graphics.h
===================================================================
--- include/wx/graphics.h	(revision 67311)
+++ include/wx/graphics.h	(working copy)
@@ -404,7 +404,7 @@
 #endif
 #endif
 
-    static wxGraphicsContext* CreateFromNative( void * context );
+    static wxGraphicsContext* CreateFromNative( void * context, bool isForPrinting );
 
     static wxGraphicsContext* CreateFromNativeWindow( void * window );
 
@@ -727,7 +727,7 @@
 #endif
 #endif
 
-    virtual wxGraphicsContext * CreateContextFromNativeContext( void * context ) = 0;
+    virtual wxGraphicsContext * CreateContextFromNativeContext( void * context, bool isForPrinting ) = 0;
 
     virtual wxGraphicsContext * CreateContextFromNativeWindow( void * window ) = 0;
Index: interface/wx/graphics.h
===================================================================
--- interface/wx/graphics.h	(revision 67311)
+++ interface/wx/graphics.h	(working copy)
@@ -351,7 +351,7 @@
 
         @see wxGraphicsRenderer::CreateContextFromNativeContext()
     */
-    static wxGraphicsContext* CreateFromNative(void* context);
+    static wxGraphicsContext* CreateFromNative(void* context, bool isForPrinting);
 
     /**
         Creates a wxGraphicsContext from a native window.
@@ -835,7 +835,7 @@
         must be a CGContextRef for Core Graphics, a Graphics pointer for
         GDIPlus, or a cairo_t pointer for cairo.
     */
-    virtual wxGraphicsContext* CreateContextFromNativeContext(void* context) = 0;
+    virtual wxGraphicsContext* CreateContextFromNativeContext(void* context, bool isForPrinting) = 0;
 
     /**
         Creates a wxGraphicsContext from a native window.