Index: hypertreelist.py
===================================================================
--- hypertreelist.py	(Revision 74315)
+++ hypertreelist.py	(Arbeitskopie)
@@ -330,6 +330,8 @@
 """ Use this style if you wish the user to be able to edit labels in the tree control. """
 TR_ROW_LINES = wx.TR_ROW_LINES                                 # put border around items
 """ Use this style to draw a contrasting border between displayed rows. """
+TR_COLUMN_LINES = wx.TR_COLUMN_LINES                           # put border between columns
+""" Use this style to draw a contrasting border between displayed columns. """
 TR_HIDE_ROOT = wx.TR_HIDE_ROOT                                 # don't display root node
 """ Use this style to suppress the display of the root node, effectively causing the first-level nodes to appear as a series of root nodes. """
 TR_FULL_ROW_HIGHLIGHT = wx.TR_FULL_ROW_HIGHLIGHT               # highlight full horz space
