Index: src/common/event.cpp
===================================================================
--- src/common/event.cpp	(revision 56695)
+++ src/common/event.cpp	(working copy)
@@ -1142,7 +1142,6 @@
 
     m_pendingEvents->Append(eventCopy);
 
-    wxLEAVE_CRIT_SECT( Lock() );
 
     // 2) Add this event handler to list of event handlers that
     //    have pending events.
@@ -1155,6 +1154,7 @@
 
     wxLEAVE_CRIT_SECT(*wxPendingEventsLocker);
 
+    wxLEAVE_CRIT_SECT( Lock() );
     // 3) Inform the system that new pending events are somewhere,
     //    and that these should be processed in idle time.
     wxWakeUpIdle();
