 src/common/treebase.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/common/treebase.cpp b/src/common/treebase.cpp
index d1493de28d..f3b11d3b11 100644
--- a/src/common/treebase.cpp
+++ b/src/common/treebase.cpp
@@ -357,6 +357,7 @@ void wxTreeCtrlBase::OnCharHook(wxKeyEvent& event)
                 wxFALLTHROUGH;
 
             case WXK_RETURN:
+            case WXK_NUMPAD_ENTER:
                 EndEditLabel(GetFocusedItem(), discardChanges);
 
                 // Do not call Skip() below.
