Index: src/msw/mdi.cpp
===================================================================
--- src/msw/mdi.cpp	(revision 61998)
+++ src/msw/mdi.cpp	(working copy)
@@ -699,8 +699,15 @@
     if ( event.GetEventType() == wxEVT_COMMAND_MENU_SELECTED )
     {
         wxMDIChildFrame * const child = GetActiveChild();
-        if ( child && child->ProcessEventHere(event) )
-            return true;
+
+        if (child)
+        {
+            for (wxEvtHandler* handler=child->GetEventHandler(); handler; handler=handler->GetNextHandler())
+            {
+                if (handler->ProcessEventHere(event))
+                    return true;
+            }
+        }
     }
 
     return wxMDIParentFrameBase::TryBefore(event);
