diff -ru2 wxWindows-cvs-orig/src/generic/logg.cpp wxWindows-cvs-patch/src/generic/logg.cpp
--- wxWindows-cvs-orig/src/generic/logg.cpp	Sat Feb 07 23:14:06 2004
+++ wxWindows-cvs-patch/src/generic/logg.cpp	Sat Feb 07 22:19:30 2004
@@ -172,5 +172,5 @@
 // FALSE if file couldn't be opened/created and -1 if the file selection
 // dialog was cancelled
-static int OpenLogFile(wxFile& file, wxString *filename = NULL);
+static int OpenLogFile(wxFile& file, wxString *filename = NULL, wxWindow *parent = NULL);
 
 #endif // wxUSE_FILE
@@ -539,5 +539,5 @@
     wxString filename;
     wxFile file;
-    int rc = OpenLogFile(file, &filename);
+    int rc = OpenLogFile(file, &filename, this);
     if ( rc == -1 )
     {
@@ -960,5 +960,5 @@
 #if wxUSE_FILEDLG
     wxFile file;
-    int rc = OpenLogFile(file);
+    int rc = OpenLogFile(file, NULL, this);
     if ( rc == -1 )
     {
@@ -1100,9 +1100,9 @@
 // FALSE if file couldn't be opened/created and -1 if the file selection
 // dialog was cancelled
-static int OpenLogFile(wxFile& file, wxString *pFilename)
+static int OpenLogFile(wxFile& file, wxString *pFilename, wxWindow *parent)
 {
     // get the file name
     // -----------------
-    wxString filename = wxSaveFileSelector(wxT("log"), wxT("txt"), wxT("log.txt"));
+    wxString filename = wxSaveFileSelector(wxT("log"), wxT("txt"), wxT("log.txt"), parent);
     if ( !filename ) {
         // cancelled

 	  	 
