--- include/wx/aui/tabdocmdi.h	(revision 0)
+++ include/wx/aui/tabdocmdi.h	(working copy)
@@ -29,6 +29,8 @@
 typedef
   wxDocParentFrameAny<wxAuiMDIParentFrame> wxAuiDocMDIParentFrameBase;
 
+template class WXDLLIMPEXP_AUI wxDocParentFrameAny<wxAuiMDIParentFrame>;
+
 class WXDLLIMPEXP_AUI wxAuiDocMDIParentFrame : public wxAuiDocMDIParentFrameBase
 {
 public:
@@ -78,6 +80,8 @@
 
 typedef
   wxDocChildFrameAny<wxAuiMDIChildFrame, wxAuiMDIParentFrame> wxAuiDocMDIChildFrameBase;
+
+template class WXDLLIMPEXP_AUI wxDocChildFrameAny<wxAuiMDIChildFrame, wxAuiMDIParentFrame>;
 
 class WXDLLIMPEXP_AUI wxAuiDocMDIChildFrame : public wxAuiDocMDIChildFrameBase
 {
