diff -bu1 -r wxWindows/demos/bombs/bombs.cpp wxWin/demos/bombs/bombs.cpp
--- wxWindows/demos/bombs/bombs.cpp	2003-12-19 07:40:45.000000000 +0100
+++ wxWin/demos/bombs/bombs.cpp	2003-12-22 17:48:06.000000000 +0100
@@ -55,3 +55,3 @@
 
-    m_frame->NewGame(bombsID_EASY);
+    m_frame->NewGame(bombsID_EASY, false);
 
@@ -123,4 +123,14 @@
 
-void BombsFrame::NewGame(int level)
+void BombsFrame::NewGame(int level, bool query)
 {
+    if(query)
+    {
+       int ok = wxMessageBox(
+                  wxT("Start new game regardless previous board?"),
+                  wxT("Confirm"),
+                  wxYES_NO | wxICON_QUESTION,
+                  this
+                );
+       if(ok!=wxYES)return;
+    }
 
@@ -164,3 +174,3 @@
 {
-    NewGame(bombsID_EASY);
+    NewGame(bombsID_EASY, true);
 }
@@ -169,3 +179,3 @@
 {
-    NewGame(bombsID_MEDIUM);
+    NewGame(bombsID_MEDIUM, true);
 }
@@ -174,3 +184,3 @@
 {
-    NewGame(bombsID_HARD);
+    NewGame(bombsID_HARD, true);
 }
diff -bu1 -r wxWindows/demos/bombs/bombs.h wxWin/demos/bombs/bombs.h
--- wxWindows/demos/bombs/bombs.h	2003-12-19 07:40:45.000000000 +0100
+++ wxWin/demos/bombs/bombs.h	2003-12-22 17:01:37.000000000 +0100
@@ -43,3 +43,3 @@
 
-    void NewGame(int level);
+    void NewGame(int level, bool query);
 
diff -bu1 -r wxWindows/demos/bombs/bombs1.cpp wxWin/demos/bombs/bombs1.cpp
--- wxWindows/demos/bombs/bombs1.cpp	2003-12-19 07:40:45.000000000 +0100
+++ wxWin/demos/bombs/bombs1.cpp	2003-12-22 16:58:35.000000000 +0100
@@ -342,2 +342,6 @@
     {
+        // cause focused field to be visible after first key hit after launching new game
+        if( m_game->m_gridFocusX < 0 ) m_game->m_gridFocusX = 0;
+        if( m_game->m_gridFocusY < 0 ) m_game->m_gridFocusY = 0;
+
         // refresh previous field and focused field

 	  	 
