Index: wx/src/msw/bmpbuttn.cpp
===================================================================
--- wx/src/msw/bmpbuttn.cpp	(revision 4455)
+++ wx/src/msw/bmpbuttn.cpp	(working copy)
@@ -411,6 +411,32 @@
         wxDCTemp dst((WXHDC)hDC);
         dst.DrawBitmap(*bitmap, x1, y1, true);
 
+		if (state & ODS_DISABLED)
+		{
+// PJD 6/14/2010 11:01AM
+			if ( !m_brushDisabled.Ok() )
+			{
+    			// draw a bitmap with two black and two background colour pixels
+    			wxBitmap bmp(2, 2);
+    			wxMemoryDC dc;
+    			dc.SelectObject(bmp);
+    			dc.SetPen(*wxBLACK_PEN);
+    			dc.DrawPoint(0, 0);
+    			dc.DrawPoint(1, 1);
+    			dc.SetPen(GetBackgroundColour());
+    			dc.DrawPoint(0, 1);
+    			dc.DrawPoint(1, 0);
+
+    			m_brushDisabled = wxBrush(bmp);
+			}
+
+			SelectInHDC selectBrush( hDC, GetHbrushOf(m_brushDisabled));
+			static const DWORD PATTERNPAINT = 0xFA0089UL;
+
+			::PatBlt( hDC, x1, y1, bitmap->GetWidth(), bitmap->GetHeight(), PATTERNPAINT);
+		}
+// PJD 6/14/2010 11:01AM
+
         return true;
     }
 #endif // wxUSE_UXTHEME
