Index: include/wx/datetime.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/datetime.h,v
retrieving revision 1.34
diff -b -u -2 -r1.34 datetime.h
--- include/wx/datetime.h	2002/08/31 11:29:09	1.34
+++ include/wx/datetime.h	2002/10/29 19:33:33
@@ -1162,4 +1162,9 @@
 // days added will be 7*weeks + days! See also GetTotalDays() function.
 //
+// Equality operators are defined for wxDateSpans. Two datespans are equal if they both
+// give the same target date when added to *every* source date. wxDateSpan::Months(1)
+// is not equal to wxDateSpan::Days(30), becouse they not give the same date when added to 
+// 1 Feb. But wxDateSpan::Days(14) is equal to wxDateSpan::Weeks(2)
+//
 // Finally, notice that for adding hours, minutes &c you don't need this
 // class: wxTimeSpan will do the job because there are no subtleties
@@ -1507,4 +1512,24 @@
 // wxDateSpan
 // ----------------------------------------------------------------------------
+
+// comparsion (equality)
+// ---------------------
+
+// equal when for every wxDateTime t: (t + ds1 == t + ds2)
+inline WXDLLEXPORT bool operator==(const wxDateSpan& ds1,
+                                         const wxDateSpan& ds2)
+{
+    if (ds1.GetYears() == ds2.GetYears() &&
+        ds1.GetMonths() == ds2.GetMonths() &&
+        ds1.GetTotalDays() == ds2.GetTotalDays())
+          return true;
+    return false;
+}                                       
+
+inline WXDLLEXPORT bool operator!=(const wxDateSpan& ds1,
+                                         const wxDateSpan& ds2)
+{
+  return (!(ds1 == ds2));
+}                                         
 
 // arithmetics

 	  	 
