Index: src/gtk/combobox.cpp
===================================================================
--- src/gtk/combobox.cpp	(revision 52386)
+++ src/gtk/combobox.cpp	(working copy)
@@ -512,7 +512,7 @@
 
 void wxComboBox::OnSelectAll(wxCommandEvent& WXUNUSED(event))
 {
-    SetSelection(-1, -1);
+    SelectAll();
 }
 
 void wxComboBox::OnUpdateCut(wxUpdateUIEvent& event)
@@ -547,7 +547,7 @@
 
 void wxComboBox::OnUpdateSelectAll(wxUpdateUIEvent& event)
 {
-    event.Enable(GetLastPosition() > 0);
+    event.Enable(!wxTextEntry::IsEmpty());
 }
 
 #endif // wxUSE_COMBOBOX
Index: src/msw/combobox.cpp
===================================================================
--- src/msw/combobox.cpp	(revision 52386)
+++ src/msw/combobox.cpp	(working copy)
@@ -589,7 +589,7 @@
 
 void wxComboBox::OnSelectAll(wxCommandEvent& WXUNUSED(event))
 {
-    SetSelection(-1, -1);
+    SelectAll();
 }
 
 void wxComboBox::OnUpdateCut(wxUpdateUIEvent& event)
@@ -624,7 +624,7 @@
 
 void wxComboBox::OnUpdateSelectAll(wxUpdateUIEvent& event)
 {
-    event.Enable(IsEditable() && GetLastPosition() > 0);
+    event.Enable(IsEditable() && !wxTextEntry::IsEmpty());
 }
 
 #if wxUSE_TOOLTIPS

 	  	 
