Index: image.cpp
===================================================================
--- image.cpp	(Revision 66568)
+++ image.cpp	(Arbeitskopie)
@@ -448,10 +448,9 @@
     {
         case wxIMAGE_QUALITY_BICUBIC:
         case wxIMAGE_QUALITY_BILINEAR:
-            // both of these algorithms should be used for up-sampling the
-            // image only, when down-sampling always use box averaging for best
-            // results
-            if ( width < old_width && height < old_height )
+            // bicubic or bilinear scaling should only be used if the
+			// destination image is _not_ smaller than half the source image
+            if ( width < old_width/2 && height < old_height/2 )
                 image = ResampleBox(width, height);
             else if ( quality == wxIMAGE_QUALITY_BILINEAR )
                 image = ResampleBilinear(width, height);
