Index: src/gtk/menu.cpp
===================================================================
--- src/gtk/menu.cpp	(revision 54812)
+++ src/gtk/menu.cpp	(working copy)
@@ -1189,6 +1189,12 @@
         #define STOCKITEM_26(wx,gtk)
     #endif
 
+    #if GTK_CHECK_VERSION(2,8,0)
+        #define STOCKITEM_28(wx,gtk) STOCKITEM(wx,gtk)
+    #else
+        #define STOCKITEM_28(wx,gtk)
+    #endif
+
     #if GTK_CHECK_VERSION(2,10,0)
         #define STOCKITEM_210(wx,gtk) STOCKITEM(wx,gtk)
     #else
@@ -1201,52 +1207,70 @@
         STOCKITEM_26(wxID_ABOUT,         GTK_STOCK_ABOUT)
         STOCKITEM(wxID_ADD,              GTK_STOCK_ADD)
         STOCKITEM(wxID_APPLY,            GTK_STOCK_APPLY)
+        STOCKITEM(wxID_BACKWARD,         GTK_STOCK_GO_BACK)
         STOCKITEM(wxID_BOLD,             GTK_STOCK_BOLD)
+        STOCKITEM(wxID_BOTTOM,           GTK_STOCK_GOTO_BOTTOM)
         STOCKITEM(wxID_CANCEL,           GTK_STOCK_CANCEL)
+        STOCKITEM(wxID_CDROM,            GTK_STOCK_CDROM)
         STOCKITEM(wxID_CLEAR,            GTK_STOCK_CLEAR)
         STOCKITEM(wxID_CLOSE,            GTK_STOCK_CLOSE)
+        STOCKITEM(wxID_CONVERT,          GTK_STOCK_CONVERT)
         STOCKITEM(wxID_COPY,             GTK_STOCK_COPY)
         STOCKITEM(wxID_CUT,              GTK_STOCK_CUT)
         STOCKITEM(wxID_DELETE,           GTK_STOCK_DELETE)
+        STOCKITEM(wxID_DOWN,             GTK_STOCK_GO_DOWN)
         STOCKITEM_26(wxID_EDIT,          GTK_STOCK_EDIT)
+        STOCKITEM(wxID_EXECUTE,          GTK_STOCK_EXECUTE)
+        STOCKITEM(wxID_EXIT,             GTK_STOCK_QUIT)
+        STOCKITEM_26(wxID_FILE,          GTK_STOCK_FILE)
         STOCKITEM(wxID_FIND,             GTK_STOCK_FIND)
-        STOCKITEM_26(wxID_FILE,          GTK_STOCK_FILE)
-        STOCKITEM(wxID_REPLACE,          GTK_STOCK_FIND_AND_REPLACE)
-        STOCKITEM(wxID_BACKWARD,         GTK_STOCK_GO_BACK)
-        STOCKITEM(wxID_DOWN,             GTK_STOCK_GO_DOWN)
+        STOCKITEM(wxID_FIRST,            GTK_STOCK_GOTO_FIRST)
+        STOCKITEM(wxID_FLOPPY,           GTK_STOCK_FLOPPY)
         STOCKITEM(wxID_FORWARD,          GTK_STOCK_GO_FORWARD)
-        STOCKITEM(wxID_UP,               GTK_STOCK_GO_UP)
+        STOCKITEM(wxID_HARDDISK,         GTK_STOCK_HARDDISK)
         STOCKITEM(wxID_HELP,             GTK_STOCK_HELP)
         STOCKITEM(wxID_HOME,             GTK_STOCK_HOME)
         STOCKITEM(wxID_INDENT,           GTK_STOCK_INDENT)
         STOCKITEM(wxID_INDEX,            GTK_STOCK_INDEX)
+        STOCKITEM_28(wxID_INFO,           GTK_STOCK_INFO)
         STOCKITEM(wxID_ITALIC,           GTK_STOCK_ITALIC)
+        STOCKITEM(wxID_JUMP_TO,          GTK_STOCK_JUMP_TO)
         STOCKITEM(wxID_JUSTIFY_CENTER,   GTK_STOCK_JUSTIFY_CENTER)
         STOCKITEM(wxID_JUSTIFY_FILL,     GTK_STOCK_JUSTIFY_FILL)
         STOCKITEM(wxID_JUSTIFY_LEFT,     GTK_STOCK_JUSTIFY_LEFT)
         STOCKITEM(wxID_JUSTIFY_RIGHT,    GTK_STOCK_JUSTIFY_RIGHT)
+        STOCKITEM(wxID_LAST,             GTK_STOCK_GOTO_LAST)
+        STOCKITEM(wxID_NETWORK,          GTK_STOCK_NETWORK)
         STOCKITEM(wxID_NEW,              GTK_STOCK_NEW)
         STOCKITEM(wxID_NO,               GTK_STOCK_NO)
         STOCKITEM(wxID_OK,               GTK_STOCK_OK)
         STOCKITEM(wxID_OPEN,             GTK_STOCK_OPEN)
         STOCKITEM(wxID_PASTE,            GTK_STOCK_PASTE)
         STOCKITEM(wxID_PREFERENCES,      GTK_STOCK_PREFERENCES)
+        STOCKITEM(wxID_PREVIEW,          GTK_STOCK_PRINT_PREVIEW)
         STOCKITEM(wxID_PRINT,            GTK_STOCK_PRINT)
-        STOCKITEM(wxID_PREVIEW,          GTK_STOCK_PRINT_PREVIEW)
         STOCKITEM(wxID_PROPERTIES,       GTK_STOCK_PROPERTIES)
-        STOCKITEM(wxID_EXIT,             GTK_STOCK_QUIT)
         STOCKITEM(wxID_REDO,             GTK_STOCK_REDO)
         STOCKITEM(wxID_REFRESH,          GTK_STOCK_REFRESH)
         STOCKITEM(wxID_REMOVE,           GTK_STOCK_REMOVE)
+        STOCKITEM(wxID_REPLACE,          GTK_STOCK_FIND_AND_REPLACE)
         STOCKITEM(wxID_REVERT_TO_SAVED,  GTK_STOCK_REVERT_TO_SAVED)
         STOCKITEM(wxID_SAVE,             GTK_STOCK_SAVE)
         STOCKITEM(wxID_SAVEAS,           GTK_STOCK_SAVE_AS)
         STOCKITEM_210(wxID_SELECTALL,    GTK_STOCK_SELECT_ALL)
+        STOCKITEM(wxID_SELECT_COLOR,     GTK_STOCK_SELECT_COLOR)
+        STOCKITEM(wxID_SELECT_FONT,      GTK_STOCK_SELECT_FONT)
+        STOCKITEM(wxID_SORT_ASCENDING,   GTK_STOCK_SORT_ASCENDING)
+        STOCKITEM(wxID_SORT_DESCENDING,  GTK_STOCK_SORT_DESCENDING)
+        STOCKITEM(wxID_SPELL_CHECK,      GTK_STOCK_SPELL_CHECK)
         STOCKITEM(wxID_STOP,             GTK_STOCK_STOP)
+        STOCKITEM(wxID_STRIKETHROUGH,    GTK_STOCK_STRIKETHROUGH)
+        STOCKITEM(wxID_TOP,              GTK_STOCK_GOTO_TOP)
         STOCKITEM(wxID_UNDELETE,         GTK_STOCK_UNDELETE)
         STOCKITEM(wxID_UNDERLINE,        GTK_STOCK_UNDERLINE)
         STOCKITEM(wxID_UNDO,             GTK_STOCK_UNDO)
         STOCKITEM(wxID_UNINDENT,         GTK_STOCK_UNINDENT)
+        STOCKITEM(wxID_UP,               GTK_STOCK_GO_UP)
         STOCKITEM(wxID_YES,              GTK_STOCK_YES)
         STOCKITEM(wxID_ZOOM_100,         GTK_STOCK_ZOOM_100)
         STOCKITEM(wxID_ZOOM_FIT,         GTK_STOCK_ZOOM_FIT)
Index: src/xrc/xmlres.cpp
===================================================================
--- src/xrc/xmlres.cpp	(revision 54812)
+++ src/xrc/xmlres.cpp	(working copy)
@@ -1784,6 +1784,24 @@
     stdID(wxID_MAXIMIZE_FRAME);
     stdID(wxID_ICONIZE_FRAME);
     stdID(wxID_RESTORE_FRAME);
+    stdID(wxID_CDROM);
+    stdID(wxID_CONVERT);
+    stdID(wxID_EXECUTE);
+    stdID(wxID_FLOPPY);
+    stdID(wxID_HARDDISK);
+    stdID(wxID_BOTTOM);
+    stdID(wxID_FIRST);
+    stdID(wxID_LAST);
+    stdID(wxID_TOP);
+    stdID(wxID_INFO);
+    stdID(wxID_JUMP_TO);
+    stdID(wxID_NETWORK);
+    stdID(wxID_SELECT_COLOR);
+    stdID(wxID_SELECT_FONT);
+    stdID(wxID_SORT_ASCENDING);
+    stdID(wxID_SORT_DESCENDING);
+    stdID(wxID_SPELL_CHECK);
+    stdID(wxID_STRIKETHROUGH);
 
 #undef stdID
 }
Index: src/common/stockitem.cpp
===================================================================
--- src/common/stockitem.cpp	(revision 54812)
+++ src/common/stockitem.cpp	(working copy)
@@ -38,52 +38,70 @@
         case wxID_ABOUT:
         case wxID_ADD:
         case wxID_APPLY:
+        case wxID_BACKWARD:
         case wxID_BOLD:
+        case wxID_BOTTOM:
         case wxID_CANCEL:
+        case wxID_CDROM:
         case wxID_CLEAR:
         case wxID_CLOSE:
+        case wxID_CONVERT:
         case wxID_COPY:
         case wxID_CUT:
         case wxID_DELETE:
+        case wxID_DOWN:
         case wxID_EDIT:
+        case wxID_EXECUTE:
+        case wxID_EXIT:
+        case wxID_FILE:
         case wxID_FIND:
-        case wxID_FILE:
-        case wxID_REPLACE:
-        case wxID_BACKWARD:
-        case wxID_DOWN:
+        case wxID_FIRST:
+        case wxID_FLOPPY:
         case wxID_FORWARD:
-        case wxID_UP:
+        case wxID_HARDDISK:
         case wxID_HELP:
         case wxID_HOME:
         case wxID_INDENT:
         case wxID_INDEX:
+        case wxID_INFO:
         case wxID_ITALIC:
+        case wxID_JUMP_TO:
         case wxID_JUSTIFY_CENTER:
         case wxID_JUSTIFY_FILL:
         case wxID_JUSTIFY_LEFT:
         case wxID_JUSTIFY_RIGHT:
+        case wxID_LAST:
+        case wxID_NETWORK:
         case wxID_NEW:
         case wxID_NO:
         case wxID_OK:
         case wxID_OPEN:
         case wxID_PASTE:
         case wxID_PREFERENCES:
+        case wxID_PREVIEW:
         case wxID_PRINT:
-        case wxID_PREVIEW:
         case wxID_PROPERTIES:
-        case wxID_EXIT:
         case wxID_REDO:
         case wxID_REFRESH:
         case wxID_REMOVE:
+        case wxID_REPLACE:
         case wxID_REVERT_TO_SAVED:
         case wxID_SAVE:
         case wxID_SAVEAS:
         case wxID_SELECTALL:
+        case wxID_SELECT_COLOR:
+        case wxID_SELECT_FONT:
+        case wxID_SORT_ASCENDING:
+        case wxID_SORT_DESCENDING:
+        case wxID_SPELL_CHECK:
         case wxID_STOP:
+        case wxID_STRIKETHROUGH:
+        case wxID_TOP:
         case wxID_UNDELETE:
         case wxID_UNDERLINE:
         case wxID_UNDO:
         case wxID_UNINDENT:
+        case wxID_UP:
         case wxID_YES:
         case wxID_ZOOM_100:
         case wxID_ZOOM_FIT:
@@ -110,52 +128,70 @@
         STOCKITEM(wxID_ABOUT,               _("&About..."))
         STOCKITEM(wxID_ADD,                 _("Add"))
         STOCKITEM(wxID_APPLY,               _("&Apply"))
+        STOCKITEM(wxID_BACKWARD,            _("&Back"))
         STOCKITEM(wxID_BOLD,                _("&Bold"))
+        STOCKITEM(wxID_BOTTOM,              _("&Bottom"))
         STOCKITEM(wxID_CANCEL,              _("&Cancel"))
+        STOCKITEM(wxID_CDROM,               _("&CD-Rom"))
         STOCKITEM(wxID_CLEAR,               _("&Clear"))
         STOCKITEM(wxID_CLOSE,               _("&Close"))
+        STOCKITEM(wxID_CONVERT,             _("&Convert"))
         STOCKITEM(wxID_COPY,                _("&Copy"))
         STOCKITEM(wxID_CUT,                 _("Cu&t"))
         STOCKITEM(wxID_DELETE,              _("&Delete"))
+        STOCKITEM(wxID_DOWN,                _("&Down"))
         STOCKITEM(wxID_EDIT,                _("&Edit"))
+        STOCKITEM(wxID_EXECUTE,             _("&Execute"))
+        STOCKITEM(wxID_EXIT,                _("&Quit"))
+        STOCKITEM(wxID_FILE,                _("&File"))
         STOCKITEM(wxID_FIND,                _("&Find"))
-        STOCKITEM(wxID_FILE,                _("&File"))
-        STOCKITEM(wxID_REPLACE,             _("Rep&lace"))
-        STOCKITEM(wxID_BACKWARD,            _("&Back"))
-        STOCKITEM(wxID_DOWN,                _("&Down"))
+        STOCKITEM(wxID_FIRST,               _("&First"))
+        STOCKITEM(wxID_FLOPPY,              _("&Floppy"))
         STOCKITEM(wxID_FORWARD,             _("&Forward"))
-        STOCKITEM(wxID_UP,                  _("&Up"))
+        STOCKITEM(wxID_HARDDISK,            _("&Harddisk"))
         STOCKITEM(wxID_HELP,                _("&Help"))
         STOCKITEM(wxID_HOME,                _("&Home"))
         STOCKITEM(wxID_INDENT,              _("Indent"))
         STOCKITEM(wxID_INDEX,               _("&Index"))
+        STOCKITEM(wxID_INFO,                _("&Info"))
         STOCKITEM(wxID_ITALIC,              _("&Italic"))
+        STOCKITEM(wxID_JUMP_TO,             _("&Jump to"))
         STOCKITEM(wxID_JUSTIFY_CENTER,      _("Centered"))
         STOCKITEM(wxID_JUSTIFY_FILL,        _("Justified"))
         STOCKITEM(wxID_JUSTIFY_LEFT,        _("Align Left"))
         STOCKITEM(wxID_JUSTIFY_RIGHT,       _("Align Right"))
+        STOCKITEM(wxID_LAST,                _("&Last"))
+        STOCKITEM(wxID_NETWORK,             _("&Network"))
         STOCKITEM(wxID_NEW,                 _("&New"))
         STOCKITEM(wxID_NO,                  _("&No"))
         STOCKITEM(wxID_OK,                  _("&OK"))
         STOCKITEM(wxID_OPEN,                _("&Open..."))
         STOCKITEM(wxID_PASTE,               _("&Paste"))
         STOCKITEM(wxID_PREFERENCES,         _("&Preferences"))
+        STOCKITEM(wxID_PREVIEW,             _("Print previe&w"))
         STOCKITEM(wxID_PRINT,               _("&Print..."))
-        STOCKITEM(wxID_PREVIEW,             _("Print previe&w"))
         STOCKITEM(wxID_PROPERTIES,          _("&Properties"))
-        STOCKITEM(wxID_EXIT,                _("&Quit"))
         STOCKITEM(wxID_REDO,                _("&Redo"))
         STOCKITEM(wxID_REFRESH,             _("Refresh"))
         STOCKITEM(wxID_REMOVE,              _("Remove"))
+        STOCKITEM(wxID_REPLACE,             _("Rep&lace"))
         STOCKITEM(wxID_REVERT_TO_SAVED,     _("Revert to Saved"))
         STOCKITEM(wxID_SAVE,                _("&Save"))
         STOCKITEM(wxID_SAVEAS,              _("Save &As..."))
         STOCKITEM(wxID_SELECTALL,           _("Select &All"))
+        STOCKITEM(wxID_SELECT_COLOR,        _("&Color"))
+        STOCKITEM(wxID_SELECT_FONT,         _("&Font"))
+        STOCKITEM(wxID_SORT_ASCENDING,      _("&Ascending"))
+        STOCKITEM(wxID_SORT_DESCENDING,     _("&Descending"))
+        STOCKITEM(wxID_SPELL_CHECK,         _("&Spell Check"))
         STOCKITEM(wxID_STOP,                _("&Stop"))
+        STOCKITEM(wxID_STRIKETHROUGH,       _("&Strikethrough"))
+        STOCKITEM(wxID_TOP,                 _("&Top"))
         STOCKITEM(wxID_UNDELETE,            _("Undelete"))
         STOCKITEM(wxID_UNDERLINE,           _("&Underline"))
         STOCKITEM(wxID_UNDO,                _("&Undo"))
         STOCKITEM(wxID_UNINDENT,            _("&Unindent"))
+        STOCKITEM(wxID_UP,                  _("&Up"))
         STOCKITEM(wxID_YES,                 _("&Yes"))
         STOCKITEM(wxID_ZOOM_100,            _("&Actual Size"))
         STOCKITEM(wxID_ZOOM_FIT,            _("Zoom to &Fit"))
Index: docs/doxygen/mainpages/const_stockitems.h
===================================================================
--- docs/doxygen/mainpages/const_stockitems.h	(revision 54812)
+++ docs/doxygen/mainpages/const_stockitems.h	(working copy)
@@ -16,55 +16,73 @@
 
 @beginDefList
 @itemdef{<b>Stock ID</b>, <b>Stock label</b>}
-@itemdef{wxID_ABOUT, "&About"}
-@itemdef{wxID_ADD, "Add" and}
+@itemdef{wxID_ABOUT, "&About..."}
+@itemdef{wxID_ADD, "Add"}
 @itemdef{wxID_APPLY, "&Apply"}
+@itemdef{wxID_BACKWARD, "&Back"}
 @itemdef{wxID_BOLD, "&Bold"}
+@itemdef{wxID_BOTTOM, "&Bottom"}
 @itemdef{wxID_CANCEL, "&Cancel"}
+@itemdef{wxID_CDROM, "&CD-Rom"}
 @itemdef{wxID_CLEAR, "&Clear"}
 @itemdef{wxID_CLOSE, "&Close"}
+@itemdef{wxID_CONVERT, "&Convert"}
 @itemdef{wxID_COPY, "&Copy"}
 @itemdef{wxID_CUT, "Cu&t"}
 @itemdef{wxID_DELETE, "&Delete"}
+@itemdef{wxID_DOWN, "&Down"}
 @itemdef{wxID_EDIT, "&Edit"}
+@itemdef{wxID_EXECUTE, "&Execute"}
+@itemdef{wxID_EXIT, "&Quit"}
+@itemdef{wxID_FILE, "&File"}
 @itemdef{wxID_FIND, "&Find"}
-@itemdef{wxID_FILE, "&File"}
-@itemdef{wxID_REPLACE, "Find and rep&lace"}
-@itemdef{wxID_BACKWARD, "&Back"}
-@itemdef{wxID_DOWN, "&Down"}
+@itemdef{wxID_FIRST, "&First"}
+@itemdef{wxID_FLOPPY, "&Floppy"}
 @itemdef{wxID_FORWARD, "&Forward"}
-@itemdef{wxID_UP, "&Up"}
+@itemdef{wxID_HARDDISK, "&Harddisk"}
 @itemdef{wxID_HELP, "&Help"}
 @itemdef{wxID_HOME, "&Home"}
 @itemdef{wxID_INDENT, "Indent"}
 @itemdef{wxID_INDEX, "&Index"}
+@itemdef{wxID_INFO, "&Info"}
 @itemdef{wxID_ITALIC, "&Italic"}
+@itemdef{wxID_JUMP_TO, "&Jump to"}
 @itemdef{wxID_JUSTIFY_CENTER, "Centered"}
 @itemdef{wxID_JUSTIFY_FILL, "Justified"}
 @itemdef{wxID_JUSTIFY_LEFT, "Align Left"}
 @itemdef{wxID_JUSTIFY_RIGHT, "Align Right"}
+@itemdef{wxID_LAST, "&Last"}
+@itemdef{wxID_NETWORK, "&Network"}
 @itemdef{wxID_NEW, "&New"}
 @itemdef{wxID_NO, "&No"}
 @itemdef{wxID_OK, "&OK"}
-@itemdef{wxID_OPEN, "&Open"}
+@itemdef{wxID_OPEN, "&Open..."}
 @itemdef{wxID_PASTE, "&Paste"}
 @itemdef{wxID_PREFERENCES, "&Preferences"}
-@itemdef{wxID_PRINT, "&Print"}
 @itemdef{wxID_PREVIEW, "Print previe&w"}
+@itemdef{wxID_PRINT, "&Print..."}
 @itemdef{wxID_PROPERTIES, "&Properties"}
-@itemdef{wxID_EXIT, "&Quit"}
 @itemdef{wxID_REDO, "&Redo"}
 @itemdef{wxID_REFRESH, "Refresh"}
 @itemdef{wxID_REMOVE, "Remove"}
+@itemdef{wxID_REPLACE, "Rep&lace"}
 @itemdef{wxID_REVERT_TO_SAVED, "Revert to Saved"}
 @itemdef{wxID_SAVE, "&Save"}
 @itemdef{wxID_SAVEAS, "Save &As..."}
-@itemdef{wxID_SELECTALL, "Select all"}
+@itemdef{wxID_SELECTALL, "Select &All"}
+@itemdef{wxID_SELECT_COLOR, "&Color"}
+@itemdef{wxID_SELECT_FONT, "&Font"}
+@itemdef{wxID_SORT_ASCENDING, "&Ascending"}
+@itemdef{wxID_SORT_DESCENDING, "&Descending"}
+@itemdef{wxID_SPELL_CHECK, "&Spell Check"}
 @itemdef{wxID_STOP, "&Stop"}
+@itemdef{wxID_STRIKETHROUGH, "&Strikethrough"}
+@itemdef{wxID_TOP, "&Top"}
 @itemdef{wxID_UNDELETE, "Undelete"}
 @itemdef{wxID_UNDERLINE, "&Underline"}
 @itemdef{wxID_UNDO, "&Undo"}
 @itemdef{wxID_UNINDENT, "&Unindent"}
+@itemdef{wxID_UP, "&Up"}
 @itemdef{wxID_YES, "&Yes"}
 @itemdef{wxID_ZOOM_100, "&Actual Size"}
 @itemdef{wxID_ZOOM_FIT, "Zoom to &Fit"}
Index: include/wx/defs.h
===================================================================
--- include/wx/defs.h	(revision 54812)
+++ include/wx/defs.h	(working copy)
@@ -2026,6 +2026,24 @@
     wxID_ZOOM_OUT,
     wxID_UNDELETE,
     wxID_REVERT_TO_SAVED,
+    wxID_CDROM,
+    wxID_CONVERT,
+    wxID_EXECUTE,
+    wxID_FLOPPY,
+    wxID_HARDDISK,
+    wxID_BOTTOM,
+    wxID_FIRST,
+    wxID_LAST,
+    wxID_TOP,
+    wxID_INFO,
+    wxID_JUMP_TO,
+    wxID_NETWORK,
+    wxID_SELECT_COLOR,
+    wxID_SELECT_FONT,
+    wxID_SORT_ASCENDING,
+    wxID_SORT_DESCENDING,
+    wxID_SPELL_CHECK,
+    wxID_STRIKETHROUGH,
 
     /*  System menu IDs (used by wxUniv): */
     wxID_SYSTEM_MENU = 5200,
Index: samples/menu/menu.cpp
===================================================================
--- samples/menu/menu.cpp	(revision 54812)
+++ samples/menu/menu.cpp	(working copy)
@@ -388,49 +388,67 @@
     wxMenu *stockSubMenu = new wxMenu;
     stockSubMenu->Append(wxID_ADD);
     stockSubMenu->Append(wxID_APPLY);
+    stockSubMenu->Append(wxID_BACKWARD);
     stockSubMenu->Append(wxID_BOLD);
+    stockSubMenu->Append(wxID_BOTTOM);
     stockSubMenu->Append(wxID_CANCEL);
+    stockSubMenu->Append(wxID_CDROM);
     stockSubMenu->Append(wxID_CLEAR);
     stockSubMenu->Append(wxID_CLOSE);
+    stockSubMenu->Append(wxID_CONVERT);
     stockSubMenu->Append(wxID_COPY);
     stockSubMenu->Append(wxID_CUT);
     stockSubMenu->Append(wxID_DELETE);
+    stockSubMenu->Append(wxID_DOWN);
+    stockSubMenu->Append(wxID_EXECUTE);
+    stockSubMenu->Append(wxID_EXIT);
     stockSubMenu->Append(wxID_FIND);
-    stockSubMenu->Append(wxID_REPLACE);
-    stockSubMenu->Append(wxID_BACKWARD);
-    stockSubMenu->Append(wxID_DOWN);
+    stockSubMenu->Append(wxID_FIRST);
+    stockSubMenu->Append(wxID_FLOPPY);
     stockSubMenu->Append(wxID_FORWARD);
-    stockSubMenu->Append(wxID_UP);
+    stockSubMenu->Append(wxID_HARDDISK);
     stockSubMenu->Append(wxID_HELP);
     stockSubMenu->Append(wxID_HOME);
     stockSubMenu->Append(wxID_INDENT);
     stockSubMenu->Append(wxID_INDEX);
+    stockSubMenu->Append(wxID_INFO);
     stockSubMenu->Append(wxID_ITALIC);
+    stockSubMenu->Append(wxID_JUMP_TO);
     stockSubMenu->Append(wxID_JUSTIFY_CENTER);
     stockSubMenu->Append(wxID_JUSTIFY_FILL);
     stockSubMenu->Append(wxID_JUSTIFY_LEFT);
     stockSubMenu->Append(wxID_JUSTIFY_RIGHT);
+    stockSubMenu->Append(wxID_LAST);
+    stockSubMenu->Append(wxID_NETWORK);
     stockSubMenu->Append(wxID_NEW);
     stockSubMenu->Append(wxID_NO);
     stockSubMenu->Append(wxID_OK);
     stockSubMenu->Append(wxID_OPEN);
     stockSubMenu->Append(wxID_PASTE);
     stockSubMenu->Append(wxID_PREFERENCES);
+    stockSubMenu->Append(wxID_PREVIEW);
     stockSubMenu->Append(wxID_PRINT);
-    stockSubMenu->Append(wxID_PREVIEW);
     stockSubMenu->Append(wxID_PROPERTIES);
-    stockSubMenu->Append(wxID_EXIT);
     stockSubMenu->Append(wxID_REDO);
     stockSubMenu->Append(wxID_REFRESH);
     stockSubMenu->Append(wxID_REMOVE);
+    stockSubMenu->Append(wxID_REPLACE);
     stockSubMenu->Append(wxID_REVERT_TO_SAVED);
     stockSubMenu->Append(wxID_SAVE);
     stockSubMenu->Append(wxID_SAVEAS);
+    stockSubMenu->Append(wxID_SELECT_COLOR);
+    stockSubMenu->Append(wxID_SELECT_FONT);
+    stockSubMenu->Append(wxID_SORT_ASCENDING);
+    stockSubMenu->Append(wxID_SORT_DESCENDING);
+    stockSubMenu->Append(wxID_SPELL_CHECK);
     stockSubMenu->Append(wxID_STOP);
+    stockSubMenu->Append(wxID_STRIKETHROUGH);
+    stockSubMenu->Append(wxID_TOP);
     stockSubMenu->Append(wxID_UNDELETE);
     stockSubMenu->Append(wxID_UNDERLINE);
     stockSubMenu->Append(wxID_UNDO);
     stockSubMenu->Append(wxID_UNINDENT);
+    stockSubMenu->Append(wxID_UP);
     stockSubMenu->Append(wxID_YES);
     stockSubMenu->Append(wxID_ZOOM_100);
     stockSubMenu->Append(wxID_ZOOM_FIT);
