--- orig/wxWidgets-3.0.2/src/msw/window.cpp	2014-10-06 16:33:44.000000000 -0500
+++ wxWidgets-3.0.2/src/msw/window.cpp	2015-07-06 14:50:59.448830900 -0500
@@ -3386,7 +3386,8 @@
 
                 if (dwObjId == (LPARAM)OBJID_CLIENT && GetOrCreateAccessible())
                 {
-                    return LresultFromObject(IID_IAccessible, wParam, (IUnknown*) GetAccessible()->GetIAccessible());
+                    processed = true;
+                    rc.result = LresultFromObject(IID_IAccessible, wParam, (IUnknown*) GetAccessible()->GetIAccessible());
                 }
                 break;
             }
