diff --git samples/xrc/myframe.cpp samples/xrc/myframe.cpp
index 0e8e1f6..4d42afe 100644
--- samples/xrc/myframe.cpp
+++ samples/xrc/myframe.cpp
@@ -76,8 +76,10 @@
 // ID name to help new users emphasize this point which is often overlooked
 // when starting out with wxWidgets.
 BEGIN_EVENT_TABLE(MyFrame, wxFrame)
-    EVT_MENU(XRCID("unload_resource_menuitem"), MyFrame::OnUnloadResourceMenuCommand)
-    EVT_MENU(XRCID("reload_resource_menuitem"), MyFrame::OnReloadResourceMenuCommand)
+/*    EVT_UPDATE_UI(XRCID("unload_resource_menuitem"), MyFrame::DevNull) // Uncomment these 2 lines to make the event-range fail
+    EVT_UPDATE_UI(XRCID("reload_resource_menuitem"), MyFrame::DevNull)*/
+    EVT_MENU_RANGE(XRCID("unload_resource_menuitem"), XRCID("reload_resource_menuitem"), MyFrame::OnUnloadResourceMenuCommand)
+    //EVT_MENU(XRCID("reload_resource_menuitem"), MyFrame::OnReloadResourceMenuCommand)
     EVT_MENU(wxID_EXIT,  MyFrame::OnExitToolOrMenuCommand)
     EVT_MENU(XRCID("non_derived_dialog_tool_or_menuitem"), MyFrame::OnNonDerivedDialogToolOrMenuCommand)
     EVT_MENU(XRCID("derived_tool_or_menuitem"), MyFrame::OnDerivedDialogToolOrMenuCommand)
@@ -134,17 +136,9 @@ MyFrame::MyFrame(wxWindow* parent)
 // Private methods
 //-----------------------------------------------------------------------------
 
-void MyFrame::OnUnloadResourceMenuCommand(wxCommandEvent& WXUNUSED(event))
+void MyFrame::OnUnloadResourceMenuCommand(wxCommandEvent& event)
 {
-    if ( wxXmlResource::Get()->Unload(wxT("rc/basicdlg.xrc")) )
-    {
-        wxLogMessage(wxT("Basic dialog resource has now been unloaded, you ")
-                     wxT("won't be able to use it before loading it again"));
-    }
-    else
-    {
-        wxLogWarning(wxT("Failed to unload basic dialog resource"));
-    }
+    wxLogMessage(wxT("Reached OnUnloadResourceMenuCommand using EVT_MENU_RANGE. ID = %i"), event.GetId());
 }
 
 void MyFrame::OnReloadResourceMenuCommand(wxCommandEvent& WXUNUSED(event))
diff --git samples/xrc/myframe.h samples/xrc/myframe.h
index 9701213..21234f2 100644
--- samples/xrc/myframe.h
+++ samples/xrc/myframe.h
@@ -50,6 +50,7 @@ private:
     void OnVariableExpansionToolOrMenuCommand(wxCommandEvent& event);
     void OnRecursiveLoad(wxCommandEvent& event);
     void OnAnimationCtrlPlay(wxCommandEvent& event);
+    void DevNull(wxUpdateUIEvent& event){wxLogDebug("ID = %i",event.GetId());}
 
     // Any class wishing to process wxWidgets events must use this macro
     DECLARE_EVENT_TABLE()
