diff --git include/wx/event.h include/wx/event.h
index 6be32bc..4bc9308 100644
--- include/wx/event.h
+++ include/wx/event.h
@@ -2760,7 +2760,18 @@ struct WXDLLIMPEXP_BASE wxEventTableEntryBase
           m_lastId(idLast),
           m_fn(fn),
           m_callbackUserData(data)
-    { }
+    {
+        // Ensure that event ranges are sane
+        if (m_lastId != wxID_ANY)
+        {
+            if ( ((m_id >= 0) && (m_id > m_lastId)) ||
+                 ((m_id < 0) && (m_id < m_lastId)) )
+            {
+                m_lastId = winid;
+                m_id = idLast;
+            }
+        }
+    }
 
     wxEventTableEntryBase( const wxEventTableEntryBase &entry )
         : m_id( entry.m_id ),
diff --git src/common/event.cpp src/common/event.cpp
index 4dd9eae..142662f 100644
--- src/common/event.cpp
+++ src/common/event.cpp
@@ -1274,7 +1274,9 @@ bool wxEvtHandler::ProcessEventIfMatchesId(const wxEventTableEntryBase& entry,
     if ((tableId1 == wxID_ANY) ||
         (tableId2 == wxID_ANY && tableId1 == event.GetId()) ||
         (tableId2 != wxID_ANY &&
-         (event.GetId() >= tableId1 && event.GetId() <= tableId2)))
+         ((event.GetId() >= tableId1 && event.GetId() <= tableId2) ||
+        // Cater for XRCID ranges, which are negative
+        (event.GetId() < 0 && event.GetId() <= tableId1 && event.GetId() >= tableId2))) )
     {
         event.Skip(false);
         event.m_callbackUserData = entry.m_callbackUserData;
