diff --git samples/xrc/myframe.cpp samples/xrc/myframe.cpp
index 0e8e1f6..d8bdcb8 100644
--- samples/xrc/myframe.cpp
+++ samples/xrc/myframe.cpp
@@ -56,6 +56,19 @@
 //-----------------------------------------------------------------------------
 // Regular resources (the non-XRC kind).
 //-----------------------------------------------------------------------------
+enum
+{
+    // menu items
+    Minimal_Quit = wxID_EXIT,
+
+    // it is important for the id corresponding to the "About" command to have
+    // this standard value as otherwise it won't be handled properly under Mac
+    // (where it is special and put into the "Apple" menu)
+    Minimal_About = wxID_ABOUT,
+    default_range_maxno = 4,
+    default_range_first = 6000,
+    default_range_last = default_range_first + default_range_maxno -1
+};
 
 // The application icon
 // All non-MSW platforms use an xpm. MSW uses an .ico file
@@ -63,6 +76,9 @@
     #include "rc/appicon.xpm"
 #endif
 
+#define XRCID_WITH_DEFAULT(str_id, default_id) \
+    wxXmlResource::GetXRCID(str_id, default_id)
+
 //-----------------------------------------------------------------------------
 // Event table: connect the events to the handler functions to process them
 //-----------------------------------------------------------------------------
@@ -89,6 +105,8 @@ BEGIN_EVENT_TABLE(MyFrame, wxFrame)
     EVT_MENU(XRCID("variable_expansion_tool_or_menuitem"), MyFrame::OnVariableExpansionToolOrMenuCommand)
     EVT_MENU(XRCID("recursive_load"), MyFrame::OnRecursiveLoad)
     EVT_MENU(wxID_ABOUT, MyFrame::OnAboutToolOrMenuCommand)
+    EVT_MENU_RANGE(XRCID_WITH_DEFAULT("default_range_first",default_range_first),XRCID_WITH_DEFAULT("default_range_last",default_range_last),MyFrame::OnDefaultIDRange)
+    EVT_MENU_RANGE(XRCID("7000"),XRCID("7003"),MyFrame::OnStringIDRange)
 END_EVENT_TABLE()
 
 //-----------------------------------------------------------------------------
@@ -111,14 +129,37 @@ MyFrame::MyFrame(wxWindow* parent)
     // Set the icon for the frame.
     SetIcon(wxICON(appicon));
 
-    // Load the menubar from XRC and set this frame's menubar to it.
-    SetMenuBar(wxXmlResource::Get()->LoadMenuBar(wxT("main_menu")));
-    // Load the toolbar from XRC and set this frame's toolbar to it.
-    // NOTE: For toolbars you currently should do it exactly like this.
-    // With toolbars, you currently can't create one, and set it later. It
-    // needs to be all in one step.
-    wxSystemOptions::SetOption ( wxT("msw.remap"), 0 );
-    SetToolBar(wxXmlResource::Get()->LoadToolBar(this, wxT("main_toolbar")));
+    wxMenu *fileMenu = new wxMenu;
+
+    // the "About" item should be in the help menu
+    wxMenu *helpMenu = new wxMenu;
+    helpMenu->Append(Minimal_About, "&About...\tF1", "Show about dialog");
+
+    fileMenu->Append(Minimal_Quit, "E&xit\tAlt-X", "Quit this program");
+
+    wxMenu *XRCID_WITH_DEFAULTMenu = new wxMenu;
+    // 6000 and 6003 work because they've already been given defaults in the event table
+    XRCID_WITH_DEFAULTMenu->Append(XRCID("default_range_first"), _T("default value of 6000\tF2"));
+    // 6001 won't work with a plain XRCID() as this string hasn't yet been seen
+    XRCID_WITH_DEFAULTMenu->Append(XRCID("default_range_first+1"), _T("default value of 6001\tF3"));
+     // 6002 works as it's given a default value here which is inside the range   
+    XRCID_WITH_DEFAULTMenu->Append(XRCID_WITH_DEFAULT("default_range_first+2",6002), _T("default value of 6002\tF4"));
+    XRCID_WITH_DEFAULTMenu->Append(XRCID("default_range_last"), _T("default value of 6003\tF5"));
+
+    wxMenu *XRCIDMenu = new wxMenu;
+    XRCIDMenu->Append(XRCID("7000"), _T("string id holds 7000\tF6"));
+    XRCIDMenu->Append(XRCID("7001"), _T("string id holds 7001\tF7"));
+    XRCIDMenu->Append(XRCID("7002"), _T("string id holds 7002\tF8"));
+    XRCIDMenu->Append(XRCID("7003"), _T("string id holds 7003\tF9"));
+
+    // now append the freshly created menu to the menu bar...
+    wxMenuBar *menuBar = new wxMenuBar();
+    menuBar->Append(fileMenu, "&File");
+    menuBar->Append(XRCID_WITH_DEFAULTMenu, "Testing default numbers");
+    menuBar->Append(XRCIDMenu, "Testing numbers from strings");
+
+    // ... and attach this menu bar to the frame
+    SetMenuBar(menuBar);
 
 #if wxUSE_STATUSBAR
     // Give the frame an optional statusbar. The '1' just means one field.
@@ -130,6 +171,17 @@ MyFrame::MyFrame(wxWindow* parent)
 #endif // wxUSE_STATUSBAR
 }
 
+void MyFrame::OnDefaultIDRange(wxCommandEvent& event)
+{
+    wxMessageBox(wxString::Format("OnDefaultIDRange caught an event with an id of %i", event.GetId()),
+                 "OnDefaultIDRange", wxOK | wxICON_INFORMATION, this);
+}
+
+void MyFrame::OnStringIDRange(wxCommandEvent& event)
+{
+    wxMessageBox(wxString::Format("OnStringIDRange caught an event with an id of %i", event.GetId()),
+                 "OnStringIDRange", wxOK | wxICON_INFORMATION, this);
+}
 //-----------------------------------------------------------------------------
 // Private methods
 //-----------------------------------------------------------------------------
diff --git samples/xrc/myframe.h samples/xrc/myframe.h
index 9701213..bcd267f 100644
--- samples/xrc/myframe.h
+++ samples/xrc/myframe.h
@@ -34,6 +34,8 @@ public:
     MyFrame( wxWindow* parent=(wxWindow *)NULL);
 
 private:
+    void OnDefaultIDRange(wxCommandEvent& event);
+    void OnStringIDRange(wxCommandEvent& event);
 
     // Event handlers (these functions should _not_ be virtual)
     void OnUnloadResourceMenuCommand(wxCommandEvent& event);
