diff -u -3 -P -r wxWidgets/contrib\include\wx\plot\plot.h wxW/contrib\include\wx\plot\plot.h
--- wxWidgets/contrib\include\wx\plot\plot.h	Tue Feb 10 23:15:04 2004
+++ wxW/contrib\include\wx\plot\plot.h	Fri Apr 23 12:28:36 2004
@@ -293,8 +293,8 @@
     size_t GetCount();
     wxPlotCurve *GetAt( size_t n );
 
-    void SetCurrent( wxPlotCurve* current );
-    wxPlotCurve *GetCurrent();
+    void SetCurrentCurve( wxPlotCurve* current );
+    wxPlotCurve *GetCurrentCurve();
 
     // mark list accessors
     // -------------------
diff -u -3 -P -r wxWidgets/contrib\src\deprecated\doslex.c wxW/contrib\src\deprecated\doslex.c
--- wxWidgets/contrib\src\deprecated\doslex.c	Sun Jan 11 12:49:40 2004
+++ wxW/contrib\src\deprecated\doslex.c	Fri Apr 23 17:37:57 2004
@@ -680,10 +680,10 @@
         {
           loop:
             #ifdef __cplusplus
-                          while (yyinput() != '*');
+                          while (yyinput() != '*') ;
                           switch (yyinput())
             #else
-                          while (input() != '*');
+                          while (input() != '*') ;
                           switch (input())
             #endif
               {
diff -u -3 -P -r wxWidgets/contrib\src\plot\plot.cpp wxW/contrib\src\plot\plot.cpp
--- wxWidgets/contrib\src\plot\plot.cpp	Sun Feb 29 16:15:59 2004
+++ wxW/contrib\src\plot\plot.cpp	Fri Apr 23 12:29:22 2004
@@ -251,7 +251,7 @@
             event1.SetPosition( (int)floor(x/m_owner->GetZoom()) );
             m_owner->GetEventHandler()->ProcessEvent( event1 );
             
-            if (curve != m_owner->GetCurrent())
+            if (curve != m_owner->GetCurrentCurve())
             {
                 wxPlotEvent event2( wxEVT_PLOT_SEL_CHANGING, m_owner->GetId() );
                 event2.SetEventObject( m_owner );
@@ -259,7 +259,7 @@
                 event2.SetCurve( curve );
                 if (!m_owner->GetEventHandler()->ProcessEvent( event2 ) || event2.IsAllowed())
                 {
-                    m_owner->SetCurrent( curve );
+                    m_owner->SetCurrentCurve( curve );
                 }
             }
             return;
@@ -418,7 +418,7 @@
         {
             wxPlotCurve *curve = (wxPlotCurve*) node->GetData();
             
-            if (curve == m_owner->GetCurrent())
+            if (curve == m_owner->GetCurrentCurve())
                 dc.SetPen( *wxBLACK_PEN );
             else
                 dc.SetPen( *wxGREY_PEN );
@@ -606,7 +606,7 @@
 {
     wxPaintDC dc( this );
     
-    wxPlotCurve *curve = m_owner->GetCurrent();
+    wxPlotCurve *curve = m_owner->GetCurrentCurve();
     
     if (!curve) return;
     
@@ -817,7 +817,7 @@
     return (wxPlotCurve*) node->GetData();
 }
 
-void wxPlotWindow::SetCurrent( wxPlotCurve* current )
+void wxPlotWindow::SetCurrentCurve( wxPlotCurve* current )
 {
     m_current = current;
     m_area->Refresh( FALSE );
@@ -844,7 +844,7 @@
     if (curve == m_current) m_current = (wxPlotCurve *) NULL;
 }
 
-wxPlotCurve *wxPlotWindow::GetCurrent()
+wxPlotCurve *wxPlotWindow::GetCurrentCurve()
 {
     return m_current;
 }

 	  	 
