Index: src/gtk/aboutdlg.cpp
===================================================================
--- src/gtk/aboutdlg.cpp	(revision 49751)
+++ src/gtk/aboutdlg.cpp	(working copy)
@@ -85,10 +85,14 @@
 // implementation
 // ============================================================================
 
+static GtkAboutDialog *dlg;
+
 extern "C" void
 wxGtkAboutDialogOnClose(GtkAboutDialog *about)
 {
     gtk_widget_destroy(GTK_WIDGET(about));
+    if (dlg == about)
+        dlg = 0;
 }
 
 extern "C" void
@@ -103,7 +107,8 @@
 {
     if ( !gtk_check_version(2,6,0) )
     {
-        GtkAboutDialog * const dlg = GTK_ABOUT_DIALOG(gtk_about_dialog_new());
+        if (!dlg)
+            dlg = GTK_ABOUT_DIALOG(gtk_about_dialog_new());
         gtk_about_dialog_set_name(dlg, wxGTK_CONV_SYS(info.GetName()));
         if ( info.HasVersion() )
             gtk_about_dialog_set_version(dlg, wxGTK_CONV_SYS(info.GetVersion()));
@@ -172,6 +177,7 @@
                             G_CALLBACK(wxGtkAboutDialogOnClose), NULL);
 
         gtk_widget_show(GTK_WIDGET(dlg));
+        gtk_window_present(GTK_WINDOW(dlg));
         return;
     }
 

 	  	 
