diff -ru wxWidgets.orig/src/common/wincmn.cpp wxWidgets/src/common/wincmn.cpp
--- wxWidgets.orig/src/common/wincmn.cpp	2008-09-15 11:28:18.000000000 +0200
+++ wxWidgets/src/common/wincmn.cpp	2008-12-31 13:26:06.515625000 +0100
@@ -2363,7 +2363,7 @@
         msg.Printf(_T("wxWidgets Library (%s port)\n")
                    _T("Version %d.%d.%d%s%s, compiled at %s %s\n")
                    _T("Runtime version of toolkit used is %d.%d.%s\n")
-                   _T("Copyright (c) 1995-2008 wxWidgets team"),
+                   _T("Copyright (c) 1995-2009 wxWidgets team"),
                    wxPlatformInfo::Get().GetPortIdName().c_str(),
                    wxMAJOR_VERSION,
                    wxMINOR_VERSION,
