--- /cygdrive/e/downloads/flatnotebook.py	2009-11-04 16:26:58.765625000 +0000
+++ /cygdrive/e/downloads/whyteboard/lib/flatnotebook.py	2009-11-04 16:49:16.625000000 +0000
@@ -104,6 +104,7 @@
 ``EVT_FLATNOTEBOOK_PAGE_CLOSED``       Notify client objects when a page in `FlatNotebook` has been closed.
 ``EVT_FLATNOTEBOOK_PAGE_CLOSING``      Notify client objects when a page in `FlatNotebook` is closing.
 ``EVT_FLATNOTEBOOK_PAGE_CONTEXT_MENU`` Notify client objects when a pop-up menu should appear next to a tab.
+``EVT_FLATNOTEBOOK_PAGE_DROPPED``      Notify client objects when a tab has been dropped and re-arranged (on the *same* notebook)
 ====================================== ==================================================
 
 
@@ -262,12 +263,14 @@
 # In Your Event Handler;
 # wxEVT_FLATNOTEBOOK_PAGE_CLOSED: Event Fired When A Page Is Closed.
 # wxEVT_FLATNOTEBOOK_PAGE_CONTEXT_MENU: Event Fired When A Menu Pops-up In A Tab.
+# wxEVT_FLATNOTEBOOK_PAGE_DROPPED: Event Fired When A Tab Is Dropped On The Same Notebook
 
 wxEVT_FLATNOTEBOOK_PAGE_CHANGED = wx.wxEVT_COMMAND_NOTEBOOK_PAGE_CHANGED
 wxEVT_FLATNOTEBOOK_PAGE_CHANGING = wx.wxEVT_COMMAND_NOTEBOOK_PAGE_CHANGING
 wxEVT_FLATNOTEBOOK_PAGE_CLOSING = wx.NewEventType()
 wxEVT_FLATNOTEBOOK_PAGE_CLOSED = wx.NewEventType()
 wxEVT_FLATNOTEBOOK_PAGE_CONTEXT_MENU = wx.NewEventType()
+wxEVT_FLATNOTEBOOK_PAGE_DROPPED = wx.NewEventType()
 
 #-----------------------------------#
 #        FlatNotebookEvent
@@ -283,7 +286,8 @@
 """ Notify client objects when a page in `FlatNotebook` has been closed."""
 EVT_FLATNOTEBOOK_PAGE_CONTEXT_MENU = wx.PyEventBinder(wxEVT_FLATNOTEBOOK_PAGE_CONTEXT_MENU, 1)
 """ Notify client objects when a pop-up menu should appear next to a tab."""
-
+EVT_FLATNOTEBOOK_PAGE_DROPPED = wx.PyEventBinder(wxEVT_FLATNOTEBOOK_PAGE_DROPPED, 1)
+""" Notify client objects when a tab has been dropped and re-arranged (on the *same* notebook)."""
 
 # Some icons in XPM format
 
@@ -3032,7 +3036,7 @@
         self._windows = []
         self._popupWin = None
         self._naviIcon = None
-
+        
         wx.PyPanel.__init__(self, parent, id, pos, size, style)
         
         self._pages = PageContainer(self, wx.ID_ANY, wx.DefaultPosition, wx.DefaultSize, style)
@@ -3352,8 +3356,8 @@
 
         self._pages._iActivePage = page
         self._pages.DoSetSelection(page)
-
-
+                
+                
     def DeletePage(self, page):
         """
         Deletes the specified page, and the associated window.
@@ -3483,7 +3487,7 @@
 
     def OnNavigationKey(self, event):
         """ Handles the wx.EVT_NAVIGATION_KEY event for L{FlatNotebook}. """
-
+        
         if event.IsWindowChange():
             if len(self._windows) == 0:
                 return
@@ -4254,7 +4258,7 @@
     
             if da_page != None:
                 da_page.SetFocus()
-        
+
         if not self.IsTabVisible(page):
             # Try to remove one tab from start and try again
             
@@ -4610,7 +4614,7 @@
 
     def OnDropTarget(self, x, y, nTabPage, wnd_oldContainer):
         """ Handles the drop action from a DND operation. """
-
+        
         # Disable drag'n'drop for disabled tab
         if len(wnd_oldContainer._pagesInfoVec) > nTabPage and \
            not wnd_oldContainer._pagesInfoVec[nTabPage].GetEnabled():
@@ -4631,7 +4635,13 @@
             
                 if where == FNB_TAB:
                     self.MoveTabPage(nTabPage, nIndex)
-                           
+                    
+                event = FlatNotebookEvent(wxEVT_FLATNOTEBOOK_PAGE_DROPPED, self.GetParent().GetId())
+                event.SetSelection(nIndex)
+                event.SetOldSelection(nTabPage)
+                event.SetEventObject(self.GetParent())
+                self.GetParent().GetEventHandler().ProcessEvent(event)
+                                           
         elif self.GetParent().GetWindowStyleFlag() & FNB_ALLOW_FOREIGN_DND:
         
             if wx.Platform in ["__WXMSW__", "__WXGTK__", "__WXMAC__"]:
