Index: samples/text/text.cpp
===================================================================
--- samples/text/text.cpp	(revision 52253)
+++ samples/text/text.cpp	(working copy)
@@ -1144,7 +1144,6 @@
     topSizer->Add( m_log, 1, wxALL | wxEXPAND, 10 );
 #endif
 
-    SetAutoLayout( true );
     SetSizer(topSizer);
 }
 
Index: samples/except/except.cpp
===================================================================
--- samples/except/except.cpp	(revision 52253)
+++ samples/except/except.cpp	(working copy)
@@ -475,8 +475,7 @@
     sizerTop->Add(new wxButton(this, wxID_CANCEL, _T("&Cancel")),
                   0, wxCENTRE | wxALL, 5);
 
-    SetSizer(sizerTop);
-    sizerTop->Fit(this);
+    SetSizerAndFit(sizerTop);
 }
 
 void MyDialog::OnThrowInt(wxCommandEvent& WXUNUSED(event))
Index: samples/controls/controls.cpp
===================================================================
--- samples/controls/controls.cpp	(revision 52253)
+++ samples/controls/controls.cpp	(working copy)
@@ -1015,7 +1015,6 @@
 
     // sizer
     panel = new wxPanel(m_book);
-    panel->SetAutoLayout( true );
 
     wxBoxSizer *sizer = new wxBoxSizer( wxVERTICAL );
 
Index: samples/wizard/wizard.cpp
===================================================================
--- samples/wizard/wizard.cpp	(revision 52253)
+++ samples/wizard/wizard.cpp	(working copy)
@@ -146,8 +146,7 @@
             wxALL,
             5 // Border
         );
-        SetSizer(mainSizer);
-        mainSizer->Fit(this);
+        SetSizerAndFit(mainSizer);
     }
 
     virtual bool TransferDataFromWindow()
@@ -204,8 +203,7 @@
             5 // Border
         );
 
-        SetSizer(mainSizer);
-        mainSizer->Fit(this);
+        SetSizerAndFit(mainSizer);
     }
 
     // wizard event handlers
@@ -314,8 +312,7 @@
         wxTextCtrl* textCtrl = new wxTextCtrl(this, wxID_ANY, wxEmptyString, wxDefaultPosition, textSize, wxTE_MULTILINE);
         mainSizer->Add(textCtrl, 0, wxALL|wxEXPAND, 5);
 
-        SetSizer(mainSizer);
-        mainSizer->Fit(this);
+        SetSizerAndFit(mainSizer);
     }
 
     // implement wxWizardPage functions
Index: samples/display/display.cpp
===================================================================
--- samples/display/display.cpp	(revision 52253)
+++ samples/display/display.cpp	(working copy)
@@ -313,7 +313,6 @@
     wxBoxSizer *sizer = new wxBoxSizer(wxHORIZONTAL);
     sizer->Add(m_book, 1, wxEXPAND);
     panel->SetSizer(sizer);
-    sizer->Fit(this);
     sizer->SetSizeHints(this);
 }
 
Index: samples/widgets/widgets.cpp
===================================================================
--- samples/widgets/widgets.cpp	(revision 52253)
+++ samples/widgets/widgets.cpp	(working copy)
@@ -448,7 +448,6 @@
 
     m_panel->SetSizer(sizerTop);
 
-    sizerTop->Fit(this);
     sizerTop->SetSizeHints(this);
 
 #if USE_LOG && !defined(__WXCOCOA__)
Index: samples/scrollsub/scrollsub.cpp
===================================================================
--- samples/scrollsub/scrollsub.cpp	(revision 52253)
+++ samples/scrollsub/scrollsub.cpp	(working copy)
@@ -174,7 +174,6 @@
 
     mainsizer->Add( middlesizer, 1, wxEXPAND );
 
-    SetAutoLayout( true );
     SetSizer( mainsizer );
 }
 
@@ -421,7 +420,6 @@
     topsizer->Add( m_log, 0, wxEXPAND );
 #endif // wxUSE_LOG
 
-    SetAutoLayout( true );
     SetSizer( topsizer );
 }
 
Index: samples/editlbox/editlbox.cpp
===================================================================
--- samples/editlbox/editlbox.cpp	(revision 52253)
+++ samples/editlbox/editlbox.cpp	(working copy)
@@ -48,8 +48,7 @@
     sizer->Add(5,5);
 
     sizer->Add(new wxButton(&dlg, wxID_OK, _("OK")), 0, wxALIGN_RIGHT | wxALL, 10);
-    dlg.SetSizer(sizer);
-    sizer->Fit(&dlg);
+    dlg.SetSizerAndFit(sizer);
     dlg.Centre();
 
     dlg.ShowModal();
Index: samples/notebook/notebook.cpp
===================================================================
--- samples/notebook/notebook.cpp	(revision 52253)
+++ samples/notebook/notebook.cpp	(working copy)
@@ -332,7 +332,6 @@
 
     m_panel->SetSizer(m_sizerFrame);
 
-    m_sizerFrame->Fit(this);
     m_sizerFrame->SetSizeHints(this);
 
     Centre(wxBOTH);
Index: samples/mediaplayer/mediaplayer.cpp
===================================================================
--- samples/mediaplayer/mediaplayer.cpp	(revision 52253)
+++ samples/mediaplayer/mediaplayer.cpp	(working copy)
@@ -1534,7 +1534,6 @@
     //
     wxFlexGridSizer* sizer = new wxFlexGridSizer(2, 1, 0, 0);
     this->SetSizer(sizer);
-    this->SetAutoLayout(true);
     sizer->AddGrowableRow(0);
     sizer->AddGrowableCol(0);
 
Index: samples/statbar/statbar.cpp
===================================================================
--- samples/statbar/statbar.cpp	(revision 52253)
+++ samples/statbar/statbar.cpp	(working copy)
@@ -576,10 +576,7 @@
     sizerTop->Add(-1, 10, 1, wxGROW);
     sizerTop->Add(statbarBottom, 0, wxGROW);
 
-    SetSizer(sizerTop);
-
-    sizerTop->Fit(this);
-    sizerTop->SetSizeHints(this);
+    SetSizerAndFit(sizerTop);
 }
 
 // ----------------------------------------------------------------------------
Index: samples/layout/layout.cpp
===================================================================
--- samples/layout/layout.cpp	(revision 52253)
+++ samples/layout/layout.cpp	(working copy)
@@ -426,12 +426,10 @@
     wxButton *button2 = new wxButton( panel, wxID_ANY, _T("Hallo") );
     panelsizer->Add( button2, 0, wxALIGN_RIGHT | wxLEFT|wxRIGHT|wxBOTTOM, 30 );
 
-    panel->SetAutoLayout( true );
     panel->SetSizer( panelsizer );
 
     // Tell dialog to use sizer
-    SetSizer( topsizer );
-    topsizer->SetSizeHints( this );
+    SetSizerAndFit( topsizer );
 }
 
 // ----------------------------------------------------------------------------
@@ -662,11 +660,7 @@
     m_target = new wxTextCtrl( this, wxID_ANY, wxEmptyString, wxDefaultPosition, wxSize( 80, wxDefaultCoord ) );
     main_sizer->Add( m_target, 1, wxALL|wxGROW, 5 );
 
-    SetSizer( main_sizer);
-
-    Layout();
-    GetSizer()->Fit( this );
-    GetSizer()->SetSizeHints( this );
+    SetSizerAndFit( main_sizer);
 }
 
 
@@ -708,8 +702,6 @@
     root->Add( ps_bottom, 1, wxEXPAND | wxALL, 5 );
         
     // Set sizer for window
-    SetSizer( root );
-    root->Fit( this );
-    root->SetSizeHints( this );
+    SetSizerAndFit( root );
 }
 
Index: samples/grid/griddemo.cpp
===================================================================
--- samples/grid/griddemo.cpp	(revision 52253)
+++ samples/grid/griddemo.cpp	(working copy)
@@ -371,11 +371,8 @@
                    wxEXPAND );
 #endif // wxUSE_LOG
 
-    SetAutoLayout(true);
-    SetSizer( topSizer );
+    SetSizerAndFit( topSizer );
 
-    topSizer->Fit( this );
-
     Centre();
     SetDefaults();
 }
Index: samples/help/demo.cpp
===================================================================
--- samples/help/demo.cpp	(revision 52253)
+++ samples/help/demo.cpp	(working copy)
@@ -711,11 +711,8 @@
     sizerTop->Add(text, 0, wxEXPAND|wxALIGN_CENTER_VERTICAL|wxALL, 5 );
     sizerTop->Add(sizerRow, 0, wxALIGN_RIGHT|wxALIGN_CENTER_VERTICAL|wxALL, 5 );
 
-    SetSizer(sizerTop);
+    SetSizerAndFit(sizerTop);
 
-    sizerTop->SetSizeHints(this);
-    sizerTop->Fit(this);
-
     btnOK->SetFocus();
     btnOK->SetDefault();
 }
Index: samples/artprov/artbrows.cpp
===================================================================
--- samples/artprov/artbrows.cpp	(revision 52253)
+++ samples/artprov/artbrows.cpp	(working copy)
@@ -162,8 +162,7 @@
     ok->SetDefault();
     sizer->Add(ok, 0, wxALIGN_RIGHT | wxALL, 10);
 
-    SetSizer(sizer);
-    sizer->Fit(this);
+    SetSizerAndFit(sizer);
 
     choice->SetSelection(6/*wxART_MESSAGE_BOX*/);
     SetArtClient(wxART_MESSAGE_BOX);
Index: samples/popup/popup.cpp
===================================================================
--- samples/popup/popup.cpp	(revision 52253)
+++ samples/popup/popup.cpp	(working copy)
@@ -142,7 +142,6 @@
     topSizer->Add( m_spinCtrl, 0, wxALL, 5 );
     topSizer->Add( m_mouseText, 0, wxCENTRE|wxALL, 5 );
 
-    m_panel->SetAutoLayout( true );
     m_panel->SetSizer( topSizer );
     topSizer->Fit(m_panel);
     topSizer->Fit(this);
@@ -365,7 +364,6 @@
     topSizer->Add( button2, 0, wxALL, 5 );
     topSizer->Add( m_logWin, 1, wxEXPAND|wxALL, 5 );
 
-    panel->SetAutoLayout( true );
     panel->SetSizer( topSizer );
 
 }
@@ -452,9 +450,7 @@
     topSizer->AddSpacer(40);
     topSizer->Add( okButton, 0, wxALL, 5 );
 
-    panel->SetAutoLayout( true );
-    panel->SetSizer( topSizer );
-    topSizer->Fit(this);
+    panel->SetSizerAndFit( topSizer );
 }
 
 void MyDialog::OnStartSimplePopup(wxCommandEvent& event)
Index: samples/dnd/dnd.cpp
===================================================================
--- samples/dnd/dnd.cpp	(revision 52253)
+++ samples/dnd/dnd.cpp	(working copy)
@@ -1624,8 +1624,7 @@
     buttonSizer->Add( bt, 0, wxALL, 2 );
     topSizer->Add( buttonSizer, 0, wxALL|wxALIGN_RIGHT, 2 );
 
-    SetSizer( topSizer );
-    topSizer->Fit( this );
+    SetSizerAndFit( topSizer );
 }
 
 DnDShape *DnDShapeDialog::GetShape() const

 	  	 
