diff -ru2 -xCVS -x.#* -x*.orig -x*.rej -x*~ wxWidgets-2.5.3/include/wx/msw/statbr95.h wxWidgets/include/wx/msw/statbr95.h
--- wxWidgets-2.5.3/include/wx/msw/statbr95.h	Tue Sep 14 14:28:54 2004
+++ wxWidgets/include/wx/msw/statbr95.h	Mon Nov 15 20:22:48 2004
@@ -62,4 +62,6 @@
     virtual int GetBorderY() const;
 
+    void OnSysColourChanged(wxSysColourChangedEvent& event);
+
 protected:
     void CopyFieldsWidth(const int widths[]);
@@ -71,4 +73,5 @@
 private:
     DECLARE_DYNAMIC_CLASS_NO_COPY(wxStatusBar95)
+    DECLARE_EVENT_TABLE()
 };
 
diff -ru2 -xCVS -x.#* -x*.orig -x*.rej -x*~ wxWidgets-2.5.3/src/msw/statbr95.cpp wxWidgets/src/msw/statbr95.cpp
--- wxWidgets-2.5.3/src/msw/statbr95.cpp	Fri Oct  1 10:27:44 2004
+++ wxWidgets/src/msw/statbr95.cpp	Tue Nov 16 00:26:50 2004
@@ -51,4 +51,9 @@
 #define StatusBar_GetText(h, n, s)  LOWORD(SendMessage(h, SB_GETTEXT, (WPARAM)n, (LPARAM)(LPTSTR)s))
 
+BEGIN_EVENT_TABLE(wxStatusBar95, wxWindow)
+  EVT_SYS_COLOUR_CHANGED(wxStatusBar95::OnSysColourChanged)
+END_EVENT_TABLE()
+
+
 // ============================================================================
 // implementation
@@ -119,5 +124,5 @@
     InheritAttributes();
 
-    SetBackgroundColour(wxSystemSettings::GetColour(wxSYS_COLOUR_MENUBAR));
+    SetBackgroundColour(wxSystemSettings::GetColour(wxSYS_COLOUR_BTNFACE));
     
     // we must refresh the frame size when the statusbar is created, because
@@ -339,4 +344,23 @@
         }
     }
+}
+
+// ----------------------------------------------------------------------------
+// event handlers
+// ----------------------------------------------------------------------------
+
+// Responds to colour changes, and passes event on to children.
+void wxStatusBar95::OnSysColourChanged(wxSysColourChangedEvent& WXUNUSED(event))
+{
+    wxColour backgroundColour = wxSystemSettings::GetColour(wxSYS_COLOUR_BTNFACE);
+    COLORREF ref = PALETTERGB(backgroundColour.Red(),
+                              backgroundColour.Green(),
+                              backgroundColour.Blue());
+    
+    SetBackgroundColour(backgroundColour);
+    SendMessage(GetHwnd(), SB_SETBKCOLOR, 0, (LPARAM)(COLORREF)ref);
+
+    // let the event propagate further
+    event.Skip();
 }
 
diff -ru2 -xCVS -x.#* -x*.orig -x*.rej -x*~ wxWidgets-2.5.3/src/msw/tbar95.cpp wxWidgets/src/msw/tbar95.cpp
--- wxWidgets-2.5.3/src/msw/tbar95.cpp	Sat Oct  9 11:01:30 2004
+++ wxWidgets/src/msw/tbar95.cpp	Mon Nov 15 20:19:12 2004
@@ -248,5 +248,5 @@
 
     // set up the colors and fonts
-    SetBackgroundColour(wxSystemSettings::GetColour(wxSYS_COLOUR_MENUBAR));
+    SetBackgroundColour(wxSystemSettings::GetColour(wxSYS_COLOUR_BTNFACE));
     SetFont(wxSystemSettings::GetFont(wxSYS_DEFAULT_GUI_FONT));
 

 	  	 
