Index: include/wx/protocol/log.h
===================================================================
--- include/wx/protocol/log.h	(revision 0)
+++ include/wx/protocol/log.h	(revision 0)
@@ -0,0 +1,55 @@
+///////////////////////////////////////////////////////////////////////////////
+// Name:        wx/protocol/log.h
+// Purpose:     wxProtocolLog class for logging network exchanges
+// Author:      Troelsk, Vadim Zeitlin
+// Created:     2009-03-06
+// RCS-ID:      $Id$
+// Copyright:   (c) 2009 Vadim Zeitlin <vadim@wxwidgets.org>
+// Licence:     wxWindows licence
+///////////////////////////////////////////////////////////////////////////////
+
+#ifndef _WX_PROTOCOL_LOG_H_
+#define _WX_PROTOCOL_LOG_H_
+
+#include "wx/string.h"
+
+// ----------------------------------------------------------------------------
+// wxProtocolLog: simple class for logging network requests and responses
+// ----------------------------------------------------------------------------
+
+class WXDLLIMPEXP_NET wxProtocolLog
+{
+public:
+    // Create object doing the logging using wxLogTrace() with the specified
+    // trace mask.
+    wxProtocolLog(const wxString& traceMask)
+        : m_traceMask(traceMask)
+    {
+    }
+
+    // Virtual dtor for the base class
+    virtual ~wxProtocolLog() { }
+
+    // Called by wxProtocol-derived classes to actually log something
+    virtual void LogRequest(const wxString& str)
+    {
+        DoLogString(wxT("==> ") + str);
+    }
+
+    virtual void LogResponse(const wxString& str)
+    {
+        DoLogString(wxT("<== ") + str);
+    }
+
+protected:
+    // Can be overridden by the derived classes.
+    virtual void DoLogString(const wxString& str);
+
+private:
+    const wxString m_traceMask;
+
+    DECLARE_NO_COPY_CLASS(wxProtocolLog)
+};
+
+#endif // _WX_PROTOCOL_LOG_H_
+
Index: include/wx/protocol/protocol.h
===================================================================
--- include/wx/protocol/protocol.h	(revision 59471)
+++ include/wx/protocol/protocol.h	(working copy)
@@ -24,6 +24,8 @@
     #include "wx/socket.h"
 #endif
 
+class WXDLLIMPEXP_FWD_NET wxProtocolLog;
+
 // ----------------------------------------------------------------------------
 // constants
 // ----------------------------------------------------------------------------
@@ -77,7 +79,33 @@
     virtual void SetUser(const wxString& WXUNUSED(user)) {}
     virtual void SetPassword(const wxString& WXUNUSED(passwd) ) {}
 
+    // logging support: each wxProtocol object may have the associated logger
+    // (by default there is none) which is used to log network requests and
+    // responses
+
+    // set the logger, deleting the old one and taking ownership of this one
+    void SetLog(wxProtocolLog *log);
+
+    // return the current logger, may be NULL
+    wxProtocolLog *GetLog() const { return m_log; }
+
+    // detach the existing logger without deleting it, the caller is
+    // responsible for deleting the returned pointer if it's non-NULL
+    wxProtocolLog *DetachLog()
+    {
+        wxProtocolLog * const log = m_log;
+        m_log = NULL;
+        return log;
+    }
+
+    // these functions forward to the same functions with the same names in
+    // wxProtocolLog if we have a valid logger and do nothing otherwise
+    void LogRequest(const wxString& str);
+    void LogResponse(const wxString& str);
+
 private:
+    wxProtocolLog *m_log;
+
     DECLARE_DYNAMIC_CLASS_NO_COPY(wxProtocol)
 };
 
Index: src/common/ftp.cpp
===================================================================
--- src/common/ftp.cpp	(revision 59471)
+++ src/common/ftp.cpp	(working copy)
@@ -220,7 +220,7 @@
         cmd = command;
     }
 
-    wxLogTrace(FTP_TRACE_MASK, _T("==> %s"), cmd.c_str());
+    LogRequest(cmd);
 #endif // __WXDEBUG__
 
     return GetResult();
@@ -263,6 +263,8 @@
             return 0;
         }
 
+        LogResponse(line);
+
         if ( !m_lastResult.empty() )
         {
             // separate from last line
Index: src/common/protocol.cpp
===================================================================
--- src/common/protocol.cpp	(revision 59471)
+++ src/common/protocol.cpp	(working copy)
@@ -19,6 +19,7 @@
 #if wxUSE_PROTOCOL
 
 #include "wx/protocol/protocol.h"
+#include "wx/protocol/log.h"
 
 #ifndef WX_PRECOMP
     #include "wx/module.h"
@@ -68,6 +69,7 @@
  : wxSocketClient()
 #endif
 {
+    m_log = NULL;
 }
 
 #if wxUSE_SOCKETS
@@ -212,4 +214,31 @@
 }
 #endif // wxUSE_SOCKETS
 
+// ----------------------------------------------------------------------------
+// logging
+// ----------------------------------------------------------------------------
+
+void wxProtocol::SetLog(wxProtocolLog *log)
+{
+    delete m_log;
+    m_log = log;
+}
+
+void wxProtocol::LogRequest(const wxString& str)
+{
+    if ( m_log )
+        m_log->LogRequest(str);
+}
+
+void wxProtocol::LogResponse(const wxString& str)
+{
+    if ( m_log )
+        m_log->LogResponse(str);
+}
+
+void wxProtocolLog::DoLogString(const wxString& str)
+{
+    wxLogTrace(m_traceMask, wxT("%s"), str);
+}
+
 #endif // wxUSE_PROTOCOL
