diff -bru wxWidgets/samples/splitter/splitter.cpp /mnt/winxp/wxWidgets/samples/splitter/splitter.cpp
--- wxWidgets/samples/splitter/splitter.cpp	2013-04-05 14:51:45.000000000 -0700
+++ /mnt/winxp/wxWidgets/samples/splitter/splitter.cpp	2013-04-10 18:23:31.062500000 -0700
@@ -36,6 +36,8 @@
     #include "wx/textdlg.h"       // for wxGetTextFromUser
 #endif
 
+#include "wx/sizer.h"
+#include "wx/button.h"
 #include "wx/splitter.h"
 #include "wx/dcmirror.h"
 
@@ -138,13 +140,14 @@
     wxDECLARE_NO_COPY_CLASS(MySplitterWindow);
 };
 
-class MyCanvas: public wxScrolledWindow
+//class MyCanvas: public wxScrolledWindow
+class MyCanvas : public wxWindow
 {
 public:
     MyCanvas(wxWindow* parent, bool mirror);
     virtual ~MyCanvas(){};
 
-    virtual void OnDraw(wxDC& dc);
+//    virtual void OnDraw(wxDC& dc);
 
 private:
     bool m_mirror;
@@ -152,6 +155,20 @@
     wxDECLARE_NO_COPY_CLASS(MyCanvas);
 };
 
+class Canvas: public wxScrolledWindow
+{
+public:
+    Canvas(wxWindow* parent, bool mirror);
+    virtual ~Canvas(){};
+
+    virtual void OnDraw(wxDC& dc);
+
+private:
+    bool m_mirror;
+
+    wxDECLARE_NO_COPY_CLASS(Canvas);
+};
+
 // ============================================================================
 // implementation
 // ============================================================================
@@ -278,7 +295,7 @@
     m_left->SetBackgroundColour(*wxRED);
     m_left->SetCursor(wxCursor(wxCURSOR_MAGNIFIER));
 
-    m_right = new MyCanvas(m_splitter, false);
+    m_right = new Canvas(m_splitter, false);
     m_right->SetBackgroundColour(*wxCYAN);
 #else // for testing kbd navigation inside the splitter
     m_left = new wxTextCtrl(m_splitter, wxID_ANY, wxT("first text"));
@@ -531,6 +548,34 @@
 // ----------------------------------------------------------------------------
 
 MyCanvas::MyCanvas(wxWindow* parent, bool mirror)
+//        : wxScrolledWindow(parent, wxID_ANY, wxDefaultPosition, wxDefaultSize,
+//                           wxHSCROLL | wxVSCROLL | wxNO_FULL_REPAINT_ON_RESIZE)
+          : wxWindow( parent, wxID_ANY, wxDefaultPosition, wxDefaultSize )
+{
+//    m_mirror = mirror;
+//    SetScrollbars(20, 20, 5, 5);
+	wxBoxSizer *vertical = new wxBoxSizer( wxVERTICAL );
+	vertical->Add( new wxButton( this, wxID_ANY, "Hello" ) );
+	vertical->Add( new wxButton( this, wxID_ANY, "Good-Bye" ) );
+	SetSizerAndFit( vertical );
+}
+/*
+void MyCanvas::OnDraw(wxDC& dcOrig)
+{
+    wxMirrorDC dc(dcOrig, m_mirror);
+
+    dc.SetPen(*wxBLACK_PEN);
+    dc.DrawLine(0, 0, 100, 200);
+
+    dc.SetBackgroundMode(wxBRUSHSTYLE_TRANSPARENT);
+    dc.DrawText(wxT("Testing"), 50, 50);
+
+    dc.SetPen(*wxRED_PEN);
+    dc.SetBrush(*wxGREEN_BRUSH);
+    dc.DrawRectangle(120, 120, 100, 80);
+}
+*/
+Canvas::Canvas(wxWindow* parent, bool mirror)
         : wxScrolledWindow(parent, wxID_ANY, wxDefaultPosition, wxDefaultSize,
                            wxHSCROLL | wxVSCROLL | wxNO_FULL_REPAINT_ON_RESIZE)
 {
@@ -538,7 +583,7 @@
     SetScrollbars(20, 20, 5, 5);
 }
 
-void MyCanvas::OnDraw(wxDC& dcOrig)
+void Canvas::OnDraw(wxDC& dcOrig)
 {
     wxMirrorDC dc(dcOrig, m_mirror);
 

