--- include/wx/mstream.h	2005-02-25 20:37:46.000000000 +0300
+++ include/wx/mstream.h	2005-03-25 10:45:16.000000000 +0300
@@ -20,6 +20,7 @@
 {
 public:
     wxMemoryInputStream(const void *data, size_t length);
+    wxMemoryInputStream(wxMemoryInputStream& stream);
     virtual ~wxMemoryInputStream();
     virtual wxFileOffset GetLength() const { return m_length; }
     virtual bool Eof() const;
@@ -57,6 +58,7 @@
     size_t CopyTo(void *buffer, size_t len) const;
 
     wxStreamBuffer *GetOutputStreamBuffer() const { return m_o_streambuf; }
+    wxMemoryInputStream *GetInputStream();
 
     // deprecated, compatibility only
     wxStreamBuffer *OutputStreamBuffer() const { return m_o_streambuf; }
--- src/common/mstream.cpp	2005-01-20 21:48:13.000000000 +0300
+++ src/common/mstream.cpp	2005-03-25 11:27:41.000000000 +0300
@@ -52,6 +52,23 @@
     m_length = len;
 }
 
+wxMemoryInputStream::wxMemoryInputStream(wxMemoryInputStream& stream)
+{
+    int len = stream.GetLength();
+    if (len == wxInvalidOffset) {
+        m_lasterror = wxSTREAM_EOF;
+        m_i_streambuf = NULL;
+        return;
+    }
+    m_i_streambuf = new wxStreamBuffer(wxStreamBuffer::read);
+    m_i_streambuf->SetBufferIO(len); // create buffer
+    memcpy(m_i_streambuf->GetBufferStart(),
+	    stream.GetInputStreamBuffer()->GetBufferStart(), len);
+    m_i_streambuf->SetIntPosition(0); // seek to start pos
+    m_i_streambuf->Fixed(true);
+    m_length = len;
+}
+
 wxMemoryInputStream::~wxMemoryInputStream()
 {
     delete m_i_streambuf;
@@ -120,6 +137,11 @@
     delete m_o_streambuf;
 }
 
+wxMemoryInputStream *wxMemoryOutputStream::GetInputStream()
+{
+    return new wxMemoryInputStream(m_o_streambuf->GetBufferStart(), GetSize());
+}
+
 size_t wxMemoryOutputStream::OnSysWrite(const void *buffer, size_t nbytes)
 {
     size_t oldpos = m_o_streambuf->GetIntPosition();

 	  	 
