--- old/src/generic/listctrl.cpp	2006-01-26 07:04:13.000000000 +0300
+++ new/src/generic/listctrl.cpp	2006-03-23 14:43:00.837394526 +0300
@@ -2830,6 +2830,7 @@
 
 void wxListMainWindow::OnMouse( wxMouseEvent &event )
 {
+
 #ifdef __WXMAC__
     // On wxMac we can't depend on the EVT_KILL_FOCUS event to properly
     // shutdown the edit control when the mouse is clicked elsewhere on the
@@ -2850,11 +2851,19 @@
         return;
     }
 
-    if ( !HasCurrent() || IsEmpty() )
+    if ( !HasCurrent() || IsEmpty() ){
+	    // ticket 10855 (fixed: context menu on empty listspace)
+		if (event.RightDown()){
+			SendNotify( -1, wxEVT_COMMAND_LIST_ITEM_RIGHT_CLICK, event.GetPosition() );
+			// Allow generation of context menu event
+			event.Skip();
+		}
         return;
+	}
 
-    if (m_dirty)
+    if (m_dirty){
         return;
+	}
 
     if ( !(event.Dragging() || event.ButtonDown() || event.LeftUp() ||
         event.ButtonDClick()) )
@@ -2919,11 +2928,16 @@
         m_dragCount = 0;
     }
 
-    if ( !hitResult )
-    {
-        // outside of any item
-        return;
-    }
+	if ( !hitResult ) {
+	    // ticket 10855 (fixed:  context menu on empty listspace)
+		if (event.RightDown()){
+			SendNotify( -1, wxEVT_COMMAND_LIST_ITEM_RIGHT_CLICK, event.GetPosition() );
+			// Allow generation of context menu event
+			event.Skip();
+		}
+
+		return;
+	}
 
     bool forceClick = false;
     if (event.ButtonDClick())
@@ -3235,6 +3249,9 @@
 
         case WXK_PRIOR:
             {
+// ticket 10393 (fixed: pressing PG_UP/PG_DOWN => floating point exception)
+		if(!m_linesPerPage)
+				m_linesPerPage = 1;
                 int steps = InReportView() ? m_linesPerPage - 1 : m_current % m_linesPerPage;
 
                 int index = m_current - steps;
@@ -3247,6 +3264,9 @@
 
         case WXK_NEXT:
             {
+// ticket 10393 (fixed: pressing PG_UP/PG_DOWN => floating point exception)
+		if(!m_linesPerPage)
+				m_linesPerPage = 1;
                 int steps = InReportView()
                                ? m_linesPerPage - 1
                                : m_linesPerPage - (m_current % m_linesPerPage) - 1;
@@ -4559,7 +4579,13 @@
 
 void wxListMainWindow::OnScroll(wxScrollWinEvent& event)
 {
-    // update our idea of which lines are shown when we redraw the window the
+	int cw, ch, vw, vh;
+	GetVirtualSize(&vw, &vh);  
+	GetClientSize(&cw, &ch);
+
+	if( event.GetOrientation() == wxVERTICAL && ch >= vh)
+		return;
+	// update our idea of which lines are shown when we redraw the window the
     // next time
     ResetVisibleLinesRange();
 

 	  	 
