Index: include/wx/msw/missing.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/msw/missing.h,v
retrieving revision 1.61
diff -b -u -2 -r1.61 missing.h
--- include/wx/msw/missing.h	2006/03/07 19:53:43	1.61
+++ include/wx/msw/missing.h	2006/03/11 05:15:26
@@ -118,4 +118,8 @@
 #endif
 
+#ifndef TPM_RECURSE
+#define TPM_RECURSE 1
+#endif
+
 // ----------------------------------------------------------------------------
 // ListView common control

 	  	 
