Index: samples/minimal/minimal.cpp
===================================================================
--- samples/minimal/minimal.cpp	(revision 71268)
+++ samples/minimal/minimal.cpp	(working copy)
@@ -28,6 +28,7 @@
 // need because it includes almost all "standard" wxWidgets headers)
 #ifndef WX_PRECOMP
     #include "wx/wx.h"
+    #include <wx/fileconf.h>
 #endif
 
 // ----------------------------------------------------------------------------
@@ -185,16 +186,10 @@
 
 void MyFrame::OnAbout(wxCommandEvent& WXUNUSED(event))
 {
-    wxMessageBox(wxString::Format
-                 (
-                    "Welcome to %s!\n"
-                    "\n"
-                    "This is the minimal wxWidgets sample\n"
-                    "running under %s.",
-                    wxVERSION_STRING,
-                    wxGetOsDescription()
-                 ),
-                 "About wxWidgets minimal sample",
-                 wxOK | wxICON_INFORMATION,
-                 this);
+    wxFileConfig config(wxEmptyString,wxEmptyString,::wxGetCwd()+wxFILE_SEP_PATH+"config.ini",wxEmptyString,wxCONFIG_USE_LOCAL_FILE);
+    //we write the key and then delete it, the write call is not necessary if key is already present in config file.
+    config.Write("key","hello");
+    config.DeleteEntry("key");
+    //this line below will crash the application.
+    config.Write("key","hello");
 }
