diff -uNrb wxWidgets-trunk\src/msw/bitmap.cpp wxWidgets-work\src/msw/bitmap.cpp
--- wxWidgets-trunk\src/msw/bitmap.cpp	2014-01-20
+++ wxWidgets-work\src/msw/bitmap.cpp	2014-02-24
@@ -380,6 +380,40 @@
     return false;
 }
 
+static bool CheckIfPremuliplied(HBITMAP hbmp, HBITMAP* hdib = NULL)
+{
+    // Check if 32-bit bitmap realy has premultiplied RGB data
+    // and premuliply it if necessary.
+
+    BITMAP bm;
+    if ( !::GetObject(hbmp, sizeof(bm), &bm) || (bm.bmBitsPixel != 32) )
+        return false;
+
+    wxDIB dib(hbmp);
+    if ( !dib.IsOk() )
+        return false;
+
+    unsigned char* pixels = dib.GetData();
+    unsigned char* const end = pixels + 4*dib.GetWidth()*dib.GetHeight();
+    for ( ; pixels < end; pixels += 4 )
+    {
+        unsigned char a = pixels[3];
+        if ( a > 0 && (pixels[0] > a || pixels[1] > a || pixels[2] > a) )
+        {
+            // RGB data for sure not premuliplied.
+            if ( hdib )
+            {
+                PremultiplyPixels(dib.GetData(), end);
+
+                *hdib = dib.Detach();
+            }
+            return true;
+        }
+    }
+
+    return false;
+}
+
 bool wxBitmap::CopyFromIconOrCursor(const wxGDIImage& icon,
                                     wxBitmapTransparency transp)
 {
@@ -1103,7 +1137,19 @@
     {
         m_refData = new wxBitmapRefData;
 
-        return handler->LoadFile(this, filename, type, -1, -1);
+        bool ok = handler->LoadFile(this, filename, type, -1, -1);
+        if( ok )
+        {
+            // Check if 32-bit bitmap really contains premultiplied RGB data.
+            HBITMAP hdib = 0;
+            if ( CheckIfPremuliplied(GetHbitmap(), &hdib) )
+            {
+                reinterpret_cast<wxBitmapRefData*>(m_refData)->m_hasAlpha = true;
+                ::DeleteObject(reinterpret_cast<wxBitmapRefData*>(m_refData)->m_hBitmap);
+                reinterpret_cast<wxBitmapRefData*>(m_refData)->m_hBitmap = hdib;
+            }
+        }
+        return ok;
     }
 #if wxUSE_IMAGE && wxUSE_WXDIB
     else // no bitmap handler found
