diff -ru2 wxWidgets-2.5.2-orig/samples/dialogs/dialogs.cpp wxWidgets-2.5.2-patch/samples/dialogs/dialogs.cpp
--- wxWidgets-2.5.2-orig/samples/dialogs/dialogs.cpp	Sat May 29 12:52:26 2004
+++ wxWidgets-2.5.2-patch/samples/dialogs/dialogs.cpp	Thu Jul 29 20:29:20 2004
@@ -42,4 +42,8 @@
 #include "wx/image.h"
 
+#if wxUSE_STOPWATCH && wxUSE_LONGLONG
+    #include <wx/datetime.h>      // wxDateTime
+#endif
+
 #define wxTEST_GENERIC_DIALOGS_IN_MSW 0
 
@@ -688,5 +692,31 @@
 void MyFrame::ShowProgress( wxCommandEvent& WXUNUSED(event) )
 {
+    #if wxUSE_STOPWATCH && wxUSE_LONGLONG
+    // check the performance
+    int countrandomnumbers = 0, count = 0;
+    wxTimeSpan tsTest(0,0,0,250);
+    wxDateTime DT2, DT1 = wxDateTime::UNow();
+    srand(0);
+    while(1)
+    {
+        rand();
+        ++countrandomnumbers;
+        if ( countrandomnumbers == 1000 )
+        {
+            srand(0);
+            countrandomnumbers = 0;
+            ++count;
+            DT2 = wxDateTime::UNow();
+            wxTimeSpan ts = DT2.Subtract( DT1 );
+            if ( ts.IsLongerThan( tsTest ) )
+            {
+                break;
+            }
+        }
+    }
+    const int max = 40 * count;
+    #else
     static const int max = 10;
+    #endif // wxUSE_STOPWATCH && wxUSE_LONGLONG
 
     wxProgressDialog dialog(_T("Progress dialog example"),
@@ -701,8 +731,26 @@
                             wxPD_REMAINING_TIME);
 
-    bool cont = TRUE;
+    bool cont = true;
     for ( int i = 0; i <= max; i++ )
     {
+        #if wxUSE_STOPWATCH && wxUSE_LONGLONG
+        // do (almost) the same operations as we did for the performance test
+        srand(0);
+        for ( int j = 0; j < 1000; j++ )
+        {
+            rand();
+            if ( j == 999 )
+            {
+                DT2 = wxDateTime::UNow();
+                wxTimeSpan ts = DT2.Subtract( DT1 );
+                if ( ts.IsLongerThan( tsTest ) )
+                {
+                    // nothing to do
+                }
+            }
+        }
+        #else
         wxSleep(1);
+        #endif
         if ( i == max )
         {
@@ -715,7 +763,13 @@
         else
         {
+            #if wxUSE_STOPWATCH && wxUSE_LONGLONG
+            if ( (i % (max/100)) == 0 ) // // only 100 updates, this makes it much faster
+            {
+                cont = dialog.Update(i);
+            }
+            #else
             cont = dialog.Update(i);
+            #endif
         }
-
         if ( !cont )
         {
@@ -725,4 +779,5 @@
                 break;
 
+            cont = true;
             dialog.Resume();
         }

 	  	 
