diff -ru2 wxWidgets-2.5.2-orig/samples/controls/controls.cpp wxWidgets-2.5.2-patch/samples/controls/controls.cpp
--- wxWidgets-2.5.2-orig/samples/controls/controls.cpp	Sat May 29 12:52:26 2004
+++ wxWidgets-2.5.2-patch/samples/controls/controls.cpp	Sat Jul 17 12:42:42 2004
@@ -50,4 +50,8 @@
 #endif
 
+#if wxUSE_STOPWATCH && wxUSE_LONGLONG
+    #include <wx/datetime.h>      // wxDateTime
+#endif
+
 #include "wx/progdlg.h"
 
@@ -1411,6 +1415,32 @@
 void MyPanel::OnShowProgress( wxCommandEvent& WXUNUSED(event) )
 {
+    #if wxUSE_STOPWATCH && wxUSE_LONGLONG
+    // check the performance
+    int countrandomnumbers = 0, count = 0;
+    wxTimeSpan tsTest(0,0,0,250);
+    wxDateTime DT2, DT1 = wxDateTime::UNow();
+    srand(0);
+    while(1)
+    {
+        rand();
+        ++countrandomnumbers;
+        if ( countrandomnumbers == 1000 )
+        {
+            srand(0);
+            countrandomnumbers = 0;
+            ++count;
+            DT2 = wxDateTime::UNow();
+            wxTimeSpan ts = DT2.Subtract( DT1 );
+            if ( ts.IsLongerThan( tsTest ) )
+            {
+                break;
+            }
+        }
+    }
+    int max = m_spinbutton->GetValue() * 4 * count;
+    #else
     int max = m_spinbutton->GetValue();
-    
+    #endif // wxUSE_STOPWATCH && wxUSE_LONGLONG
+
     if ( max <= 0 )
     {
@@ -1418,5 +1448,5 @@
         return;
     }
-    
+
     wxProgressDialog dialog(_T("Progress dialog example"),
                             _T("An informative message"),
@@ -1428,5 +1458,6 @@
                             wxPD_ELAPSED_TIME |
                             wxPD_ESTIMATED_TIME |
-                            wxPD_REMAINING_TIME);
+                            wxPD_REMAINING_TIME,
+                            0); // <<< change to 5 to slow down updates
 
 
@@ -1434,5 +1465,23 @@
     for ( int i = 0; i <= max && cont; i++ )
     {
+        #if wxUSE_STOPWATCH && wxUSE_LONGLONG
+        // do (almost) the same operations as we did for the performance test
+        srand(0);
+        for ( int j = 0; j < 1000; j++ )
+        {
+            rand();
+            if ( j == 999 )
+            {
+                DT2 = wxDateTime::UNow();
+                wxTimeSpan ts = DT2.Subtract( DT1 );
+                if ( ts.IsLongerThan( tsTest ) )
+                {
+                    // nothing to do
+                }
+            }
+        }
+        #else
         wxSleep(1);
+        #endif
         if ( i == max )
         {
@@ -1445,5 +1494,12 @@
         else
         {
+            #if wxUSE_STOPWATCH && wxUSE_LONGLONG
+            if ( (i % (max/100)) == 0 ) // // only 100 updates, this makes it much faster
+            {
+                cont = dialog.Update(i);
+            }
+            #else
             cont = dialog.Update(i);
+            #endif
         }
     }

 	  	 
