--- viewer.py	2011-09-22 08:15:22.114000000 -0500
+++ viewer_print.py	2011-10-08 17:27:28.794200000 -0500
@@ -9,6 +9,10 @@
 
 # History:      Created 17 Jun 2009
 #
+#               08 Oct 2011, Michael Hipp    michael@redmule.com
+#               Added prompt, printer_name, orientation options to
+#               pdfViewer.Print()
+#
 #----------------------------------------------------------------------------
 
 import sys, os, time
@@ -171,16 +175,31 @@
             shutil.copy(self.pdfpathname, pathname)
         dlg.Destroy()    
 
-    def Print(self):
-        "Print pdf"
+    def Print(self, prompt=True, printer_name=None, orientation=None):
+        """
+        Print the pdf.
+        
+        :param boolean `prompt`: show the print dialog to the user (True/False). If
+         False, the print dialog will not be shown and the pdf will be printed
+         immediately. Default: True.
+        :param string `printer_name`: the name of the printer that is to
+         receive the printout. Default: as set by the O/S.
+        :param `orientation`: select the orientation (wx.PORTRAIT or
+         wx.LANDSCAPE) for the printout. Default: as set by the O/S.
+        """
         pdd = wx.PrintDialogData()
         pdd.SetMinPage(1)
         pdd.SetFromPage(1)
         pdd.SetMaxPage(self.numpages)
         pdd.SetToPage(self.numpages)
+        pdata = pdd.GetPrintData()
+        if printer_name:
+            pdata.SetPrinterName(printer_name)
+        if orientation:
+            pdata.SetOrientation(orientation)
         printer = wx.Printer(pdd)
         printout = pdfPrintout('', self)
-        if (not printer.Print(self, printout, prompt=True) and
+        if (not printer.Print(self, printout, prompt=prompt) and
                        printer.GetLastError() == wx.PRINTER_ERROR):
             dlg = wx.MessageDialog(self, 'Unable to perform printing',
                               'Printer' , wx.OK | wx.ICON_INFORMATION)
