diff -ru2 -xCVS -x.#* -x*.orig -x*.rej -x*~ wxWindows.orig\include\wx\vscroll.h wxWindows\include\wx\vscroll.h
--- wxWindows.orig\include\wx\vscroll.h	Thu Aug 12 17:37:54 2004
+++ wxWindows\include\wx\vscroll.h	Thu Feb 03 11:09:30 2005
@@ -164,4 +164,7 @@
     void OnSize(wxSizeEvent& event);
     void OnScroll(wxScrollWinEvent& event);
+#if wxUSE_MOUSEWHEEL
+    void OnMouseWheel(wxMouseEvent& event);
+#endif

     // find the index of the line we need to show at the top of the window such
@@ -194,4 +197,8 @@
     size_t m_nVisible;

+    // accumulated mouse wheel rotation
+#if wxUSE_MOUSEWHEEL
+    int m_sumWheelRotation;
+#endif

     DECLARE_EVENT_TABLE()
diff -ru2 -xCVS -x.#* -x*.orig -x*.rej -x*~ wxWindows.orig\src\generic\vscroll.cpp wxWindows\src\generic\vscroll.cpp
--- wxWindows.orig\src\generic\vscroll.cpp	Thu Jun 17 19:22:36 2004
+++ wxWindows\src\generic\vscroll.cpp	Thu Feb 03 11:54:28 2005
@@ -38,4 +38,7 @@
     EVT_SIZE(wxVScrolledWindow::OnSize)
     EVT_SCROLLWIN(wxVScrolledWindow::OnScroll)
+#if wxUSE_MOUSEWHEEL
+    EVT_MOUSEWHEEL(wxVScrolledWindow::OnMouseWheel)
+#endif
 END_EVENT_TABLE()

@@ -61,4 +64,8 @@

     m_heightTotal = 0;
+
+#if wxUSE_MOUSEWHEEL
+    m_sumWheelRotation = 0;
+#endif
 }

@@ -422,2 +429,24 @@
 }

+#if wxUSE_MOUSEWHEEL
+
+void wxVScrolledWindow::OnMouseWheel(wxMouseEvent& event)
+{
+    m_sumWheelRotation += event.GetWheelRotation();
+    int delta = event.GetWheelDelta();
+
+    // how much to scroll this time
+    int units_to_scroll = -(m_sumWheelRotation/delta);
+    if ( !units_to_scroll )
+        return;
+
+    m_sumWheelRotation += units_to_scroll*delta;
+
+    if ( !event.IsPageScroll() )
+        ScrollLines( units_to_scroll*event.GetLinesPerAction() );
+    else
+        // scroll pages instead of lines
+        ScrollPages( units_to_scroll );
+}
+
+#endif

 	  	 
