Index: docs/latex/wx/grid.tex
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/docs/latex/wx/grid.tex,v
retrieving revision 1.30
diff -u -r1.30 grid.tex
--- docs/latex/wx/grid.tex	2004/12/06 17:22:48	1.30
+++ docs/latex/wx/grid.tex	2005/02/23 01:47:28
@@ -93,7 +93,9 @@
 \helpref{wxGrid::SelectAll}{wxgridselectall}\\
 \helpref{wxGrid::SelectBlock}{wxgridselectblock}\\
 \helpref{wxGrid::SelectCol}{wxgridselectcol}\\
-\helpref{wxGrid::SelectRow}{wxgridselectrow}
+\helpref{wxGrid::SelectRow}{wxgridselectrow}\\
+\helpref{wxGrid::SetAllowMultipleSelect} {wxgridallowmultipleselect}\\
+\helpref{wxGrid::GetAllowMultipleSelect} {wxgridallowmultipleselect}
 
 %%%%%%%%% MEMBER FUNCTIONS %%%%%%%%%%
 \helponly{\insertatlevel{2}{
@@ -1089,6 +1091,16 @@
 Selects the specified row. If addToSelected is false then any existing selection will be
 deselected; if true the row will be added to the existing selection.
 
+\membersection{wxGrid::SetAllowMultipleSelect}\label{wxgridallowmultipleselect}
+\func{void}{SetAllowMultipleSelect}{\param{bool }{allow}}
+\func{bool}{GetAllowMultipleSelect}{\param{void}}
+Get/Sets the multiple selection behavior. If true(the default), then dragging the mouse while clicking will
+select multiple cells, shift-clicking will select blocks, and ctrl-clicking will add or remove individual cells from the selection.
+(read cells as rows or columns if that selection mode is set with SetSelectionMode())
+
+If false, then you cannot select multiple cells by dragging, ctrl- or shift-click. Programmatic selection
+will still allow multiple selections. Clicking on the corner label will still select all, you can override this behavior by
+defining an event handler for wxEVT_GRID_LABEL_LEFT_CLICK and checking for a column and row of -1.
 \membersection{wxGrid::SetCellAlignment}\label{wxgridsetcellalignment}
 
 \func{void}{SetCellAlignment}{\param{int }{row}, \param{int }{col}, \param{int }{horiz}, \param{int }{vert}}
Index: include/wx/generic/grid.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/generic/grid.h,v
retrieving revision 1.143
diff -u -r1.143 grid.h
--- include/wx/generic/grid.h	2005/02/14 23:53:36	1.143
+++ include/wx/generic/grid.h	2005/02/23 01:29:26
@@ -1100,6 +1100,8 @@
 
     void SetSelectionMode(wxGrid::wxGridSelectionModes selmode);
     wxGrid::wxGridSelectionModes GetSelectionMode() const;
+    void SetAllowMultipleSelect(bool allow) {this->m_allowmultiple = allow;}
+    bool GetAllowMultipleSelect() {return this->m_allowmultiple;}
 
     // ------ grid dimensions
     //
@@ -1785,7 +1787,7 @@
 
     int GetRowTop(int row) const;
     int GetRowBottom(int row) const;
-
+    bool m_allowmultiple;
     int m_rowLabelWidth;
     int m_colLabelHeight;
 
Index: samples/grid/griddemo.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/samples/grid/griddemo.cpp,v
retrieving revision 1.10
diff -u -r1.10 griddemo.cpp
--- samples/grid/griddemo.cpp	2004/10/06 20:53:19	1.10
+++ samples/grid/griddemo.cpp	2005/02/23 01:39:28
@@ -129,6 +129,7 @@
 
     EVT_GRID_EDITOR_SHOWN( GridFrame::OnEditorShown )
     EVT_GRID_EDITOR_HIDDEN( GridFrame::OnEditorHidden )
+    EVT_MENU(ID_ALLOW_MULTIPLE, GridFrame::ToggleMultiple)
 END_EVENT_TABLE()
 
 
@@ -205,6 +206,8 @@
     selectMenu->Append( ID_DESELECT_COL, _T("Deselect col 2"));
     selectMenu->Append( ID_DESELECT_CELL, _T("Deselect cell (3, 1)"));
     wxMenu *selectionMenu = new wxMenu;
+    selectMenu->Append( ID_ALLOW_MULTIPLE, _T("Allow multiple cell selection"), _T(""), TRUE);
+    selectMenu->Check(ID_ALLOW_MULTIPLE, true);
     selectMenu->Append( ID_CHANGESEL, _T("Change &selection mode"),
                       selectionMenu,
                       _T("Change selection mode") );
@@ -963,7 +966,9 @@
     grid->CreateGrid(3,3);
     frame->Show(true);
 }
-
+void GridFrame::ToggleMultiple(wxCommandEvent &event) {
+    this->grid->SetAllowMultipleSelect(event.IsChecked());
+}
 void GridFrame::OnVTable(wxCommandEvent& )
 {
     static long s_sizeGrid = 10000;
Index: samples/grid/griddemo.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/samples/grid/griddemo.h,v
retrieving revision 1.7
diff -u -r1.7 griddemo.h
--- samples/grid/griddemo.h	2004/10/06 20:53:19	1.7
+++ samples/grid/griddemo.h	2005/02/23 01:35:36
@@ -86,6 +86,7 @@
     void OnEditorShown(wxGridEvent&);
     void OnEditorHidden(wxGridEvent&);
 
+    void ToggleMultiple(wxCommandEvent &event);
     void OnSetHighlightWidth(wxCommandEvent&);
     void OnSetROHighlightWidth(wxCommandEvent&);
 
@@ -149,6 +150,7 @@
 
         ID_SET_HIGHLIGHT_WIDTH,
         ID_SET_RO_HIGHLIGHT_WIDTH,
+        ID_ALLOW_MULTIPLE,
 
         ID_TESTFUNC
     };
Index: src/generic/grid.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/generic/grid.cpp,v
retrieving revision 1.323
diff -u -r1.323 grid.cpp
--- src/generic/grid.cpp	2005/02/16 20:34:20	1.323
+++ src/generic/grid.cpp	2005/02/23 01:44:54
@@ -3886,7 +3886,8 @@
                  const wxString& name )
   : wxScrolledWindow( parent, id, pos, size, (style | wxWANTS_CHARS), name ),
     m_colMinWidths(GRID_HASH_SIZE),
-    m_rowMinHeights(GRID_HASH_SIZE)
+    m_rowMinHeights(GRID_HASH_SIZE),
+    m_allowmultiple(true)	//support for toggling of multiple cell selection
 {
     Create();
     SetBestFittingSize(size);
@@ -4873,7 +4874,7 @@
                 case WXGRID_CURSOR_SELECT_ROW:
                     if ( (row = YToRow( y )) >= 0 )
                     {
-                        if ( m_selection )
+                        if ( m_selection && m_allowmultiple)
                         {
                             m_selection->SelectRow( row,
                                                     event.ControlDown(),
@@ -4927,7 +4928,10 @@
                     ClearSelection();
                 if ( m_selection )
                 {
-                    if ( event.ShiftDown() )
+                    if (m_selection->IsSelection() && !m_allowmultiple) {
+                        return;
+                    }
+                    if ( event.ShiftDown() && m_allowmultiple)
                     {
                         m_selection->SelectBlock( m_currentCellCoords.GetRow(),
                                                   0,
@@ -5095,7 +5099,7 @@
                 case WXGRID_CURSOR_SELECT_COL:
                     if ( (col = XToCol( x )) >= 0 )
                     {
-                        if ( m_selection )
+                        if ( m_selection && m_allowmultiple)
                         {
                             m_selection->SelectCol( col,
                                                     event.ControlDown(),
@@ -5149,6 +5153,9 @@
                     ClearSelection();
                 if ( m_selection )
                 {
+                    if (m_selection->IsSelection() && !m_allowmultiple) {
+                        return;
+                    }
                     if ( event.ShiftDown() )
                     {
                         m_selection->SelectBlock( 0,
@@ -5415,6 +5422,9 @@
         {
             // Hide the edit control, so it
             // won't interfer with drag-shrinking.
+            if (!m_allowmultiple && !CanDragCell())
+                return; //skip all this if we don't do anything with the drags
+
             if ( IsCellEditControlShown() )
             {
                 HideCellEditControl();
@@ -5537,7 +5547,7 @@
                 ClearSelection();
             if ( event.ShiftDown() )
             {
-                if ( m_selection )
+                if ( m_selection && m_allowmultiple )
                 {
                     m_selection->SelectBlock( m_currentCellCoords.GetRow(),
                                               m_currentCellCoords.GetCol(),
@@ -5557,7 +5567,7 @@
 
                 if ( event.ControlDown() )
                 {
-                    if ( m_selection )
+                    if ( m_selection && m_allowmultiple)
                     {
                         m_selection->ToggleCellSelection( coords.GetRow(),
                                                           coords.GetCol(),

 	  	 
