Index: include/wx/chkconf.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/chkconf.h,v
retrieving revision 1.34
diff -u -r1.34 chkconf.h
--- include/wx/chkconf.h	2003/01/12 22:20:37	1.34
+++ include/wx/chkconf.h	2003/01/21 22:53:41
@@ -12,10 +12,10 @@
 /* THIS IS A C FILE, DON'T USE C++ FEATURES (IN PARTICULAR COMMENTS) IN IT */
 
 /*
-   this global setting determintes what should we do if the setting FOO
+   this global setting determines what should we do if the setting FOO
    requires BAR and BAR is not set: we can either silently define BAR
    (default, recommended) or give an error and abort (mainly useful for
-   developpers only)
+   developers only)
  */
 #define wxABORT_ON_CONFIG_ERROR
 
@@ -1061,7 +1061,7 @@
 #endif /* wxUSE_RADIOBTN */
 
 #if wxUSE_WXHTML_HELP
-#   if !wxUSE_HELP || !wxUSE_HTML || !wxUSE_COMBOBOX
+#   if !wxUSE_HELP || !wxUSE_HTML || !wxUSE_COMBOBOX || !wxUSE_NOTEBOOK
 #       ifdef wxABORT_ON_CONFIG_ERROR
 #           error "Built in help controller can't be compiled"
 #       else
@@ -1071,6 +1071,8 @@
 #           define wxUSE_HTML 1
 #           undef wxUSE_COMBOBOX
 #           define wxUSE_COMBOBOX 1
+#           undef wxUSE_NOTEBOOK
+#           define wxUSE_NOTEBOOK 1
 #       endif
 #   endif
 #endif /* wxUSE_WXHTML_HELP */

 	  	 
