cvs server: Diffing .
Index: makefile.b32
===================================================================
RCS file: /home/wxcvs/wxWindows/samples/makefile.b32,v
retrieving revision 1.6
diff -u -2 -r1.6 makefile.b32
--- makefile.b32	2001/04/26 10:17:58	1.6
+++ makefile.b32	2001/11/29 14:14:28
@@ -28,133 +28,158 @@
 # Need PNG
 PNGSETUP=
+# Need wxUSE_ODBC; define this to compile db sample
+ODBCSETUP=
 
+
 #
 # Please keep samples list up-to-date, in alphabetical order
-#
+# not alphabetically organised ; 29 Nov 01
 all:
         cd $(WXDIR)\samples\calendar
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
         cd $(WXDIR)\samples\caret
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
         cd $(WXDIR)\samples\checklst
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
         cd $(WXDIR)\samples\config
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
         cd $(WXDIR)\samples\controls
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
+!if "$(ODBCSETUP)"=="1"
         cd $(WXDIR)\samples\db
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
+!endif
         cd $(WXDIR)\samples\ipc
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
         cd $(WXDIR)\samples\dialogs
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
         cd $(WXDIR)\samples\dnd
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
         cd $(WXDIR)\samples\docview
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
         cd $(WXDIR)\samples\docvwmdi
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
         cd $(WXDIR)\samples\drawing
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
         cd $(WXDIR)\samples\dynamic
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
         cd $(WXDIR)\samples\font
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
         cd $(WXDIR)\samples\grid
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
         cd $(WXDIR)\samples\newgrid
-        make -f makefile.b32 FINAL=$(FINAL)
-!if "$(COMPIL)"==""
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
+
         cd $(WXDIR)\samples\help
-        make -f makefile.b32 FINAL=$(FINAL)
-!endif
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
+
 !if "$(PNGSETUP)"==""
         cd $(WXDIR)\samples\html
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
 !endif
         cd $(WXDIR)\samples\image
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
         cd $(WXDIR)\samples\internat
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
         cd $(WXDIR)\samples\joytest
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
         cd $(WXDIR)\samples\layout
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
         cd $(WXDIR)\samples\listctrl
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
         cd $(WXDIR)\samples\mdi
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
         cd $(WXDIR)\samples\menu
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
 !if "$(FINAL)" == "0"
         cd $(WXDIR)\samples\memcheck
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
 !endif
 !if "$(COMPIL)"==""
         cd $(WXDIR)\samples\mfc
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
 !endif
         cd $(WXDIR)\samples\minifram
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
         cd $(WXDIR)\samples\minimal
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
         cd $(WXDIR)\samples\nativdlg
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
 !if "$(COMPIL)"==""
         cd $(WXDIR)\samples\dialup
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
 !endif
         cd $(WXDIR)\samples\notebook
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
         cd $(WXDIR)\samples\oleauto
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
         cd $(WXDIR)\samples\ownerdrw
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
         cd $(WXDIR)\samples\png
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
         cd $(WXDIR)\samples\printing
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
         cd $(WXDIR)\samples\proplist
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
         cd $(WXDIR)\samples\regtest
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
         cd $(WXDIR)\samples\resource
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
 !if "$(COMPIL)"==""
         cd $(WXDIR)\samples\richedit
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
 !endif
         cd $(WXDIR)\samples\rotate
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
         cd $(WXDIR)\samples\sashtest
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
         cd $(WXDIR)\samples\scroll
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
         cd $(WXDIR)\samples\scrollsub
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
         cd $(WXDIR)\samples\splitter
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
 !if "$(COMPIL)"==""
         cd $(WXDIR)\samples\tab
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
 !endif
         cd $(WXDIR)\samples\taskbar
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
         cd $(WXDIR)\samples\text
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
         cd $(WXDIR)\samples\thread
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
         cd $(WXDIR)\samples\toolbar
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
         cd $(WXDIR)\samples\treectrl
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
         cd $(WXDIR)\samples\typetest
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
         cd $(WXDIR)\samples\validate
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
         cd $(WXDIR)\samples\wizard
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
+        cd $(WXDIR)\samples\dragimag
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
+        cd $(WXDIR)\samples\exec
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
+        cd $(WXDIR)\samples\richedit
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
+        cd $(WXDIR)\samples\statbar
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
         cd $(WXDIR)\samples\sockets
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
+        cd $(WXDIR)\samples\widgets
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
+        cd $(WXDIR)\samples\tab
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
+        cd $(WXDIR)\samples\treelay
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
+        cd $(WXDIR)\samples\erase
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
+        cd $(WXDIR)\samples\event
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
+        cd $(WXDIR)\samples\propsize
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
 
 clean:
@@ -271,5 +296,28 @@
         make -f makefile.b32 clean
 !endif
+        cd $(WXDIR)\samples\dragimag
+        make -f makefile.b32 clean
+        cd $(WXDIR)\samples\erase
+        make -f makefile.b32 clean
+        cd $(WXDIR)\samples\exec
+        make -f makefile.b32 clean
+        cd $(WXDIR)\samples\event
+        make -f makefile.b32 clean
+        cd $(WXDIR)\samples\statbar
+        make -f makefile.b32 clean
+        cd $(WXDIR)\samples\richedit
+        make -f makefile.b32 clean
+        cd $(WXDIR)\samples\propsize
+        make -f makefile.b32 clean
+
         cd $(WXDIR)\samples\sockets
+        make -f makefile.b32 clean
+        cd $(WXDIR)\samples\treelay
+        make -f makefile.b32 clean
+        cd $(WXDIR)\samples\treectrl
+        make -f makefile.b32 clean
+        cd $(WXDIR)\samples\tab
+        make -f makefile.b32 clean
+        cd $(WXDIR)\samples\widgets
         make -f makefile.b32 clean
 
cvs server: Diffing html
Index: html/makefile.b32
===================================================================
RCS file: /home/wxcvs/wxWindows/samples/html/makefile.b32,v
retrieving revision 1.1
diff -u -2 -r1.1 makefile.b32
--- html/makefile.b32	2000/03/14 11:51:28	1.1
+++ html/makefile.b32	2001/11/29 14:15:18
@@ -19,19 +19,19 @@
 all:
         cd $(WXDIR)\samples\html\about
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
         cd $(WXDIR)\samples\html\help
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
         cd $(WXDIR)\samples\html\helpview
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
         cd $(WXDIR)\samples\html\printing
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
         cd $(WXDIR)\samples\html\test
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
         cd $(WXDIR)\samples\html\virtual
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
         cd $(WXDIR)\samples\html\widget
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
         cd $(WXDIR)\samples\html\zip
-        make -f makefile.b32 FINAL=$(FINAL)
+        make -f makefile.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL)
 
 clean:
Index: ipc/makefile.b32
===================================================================
RCS file: /home/wxcvs/wxWindows/samples/ipc/makefile.b32,v
retrieving revision 1.2
diff -u -2 -r1.2 makefile.b32
--- ipc/makefile.b32	2000/03/03 19:24:20	1.2
+++ ipc/makefile.b32	2001/11/29 14:15:56
@@ -9,6 +9,6 @@
 
 all:
-    make -f server.b32 all
-    make -f client.b32 all
+    make -f server.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL) all
+    make -f client.b32 FINAL=$(FINAL) WXUSINGDLL=$(WXUSINGDLL) all
 
 clean:
Index: proplist/proplist.cpp
===================================================================
RCS file: /home/wxcvs/wxWindows/samples/proplist/proplist.cpp,v
retrieving revision 1.5
diff -u -2 -r1.5 proplist.cpp
--- proplist/proplist.cpp	2001/11/22 12:58:44	1.5
+++ proplist/proplist.cpp	2001/11/29 14:16:39
@@ -238,6 +238,14 @@
     m_childWindow = propFrame;
   }
-  
-  wxWindow *panel = propDialog ? propDialog : propFrame->GetPropertyPanel();
+  wxWindow *panel ;
+  if (  propDialog )
+  {
+    panel = propDialog ;
+  }
+  else
+  {
+    panel = propFrame->GetPropertyPanel() ;
+  }
+    
   wxLayoutConstraints* c;
 
Index: tab/tab.cpp
===================================================================
RCS file: /home/wxcvs/wxWindows/samples/tab/tab.cpp,v
retrieving revision 1.1
diff -u -2 -r1.1 tab.cpp
--- tab/tab.cpp	2000/03/14 18:58:21	1.1
+++ tab/tab.cpp	2001/11/29 14:17:09
@@ -24,4 +24,8 @@
 #include "tab.h"
 
+#if wxUSE_TAB_DIALOG == 0
+#error "wxUSE_TAB_DIALOG must be defined as 1 in setup.h."
+#endif
+
 // If 1, use a dialog. Otherwise use a frame.
 #define USE_TABBED_DIALOG 1
@@ -167,5 +171,5 @@
   // Note, omit the wxTAB_STYLE_COLOUR_INTERIOR, so we will guarantee a match
   // with the panel background, and save a bit of time.
-  wxPanelTabView *view = new wxPanelTabView(this, wxTAB_STYLE_DRAW_BOX);
+  wxPanelTabView *view = new wxPanelTabView((wxPanel *)this, wxTAB_STYLE_DRAW_BOX);
 
   wxGetApp().InitTabView(view, this);




 	  	 
