Index: src/generic/treectlg.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/generic/treectlg.cpp,v
retrieving revision 1.173.2.7
diff -u -r1.173.2.7 treectlg.cpp
--- src/generic/treectlg.cpp	2006/07/25 15:11:09	1.173.2.7
+++ src/generic/treectlg.cpp	2006/08/07 16:28:13
@@ -2019,7 +2019,7 @@
 #if defined( __WXMSW__ ) || defined(__WXMAC__)
         Update();
 #else
-        wxYieldIfNeeded();
+        DoDirtyProcessing();
 #endif
     wxGenericTreeItem *gitem = (wxGenericTreeItem*) item.m_pItem;
 
@@ -2988,7 +2988,7 @@
 #if defined( __WXMSW__ ) || defined(__WXMAC__)
         Update();
 #else
-        wxYieldIfNeeded();
+        DoDirtyProcessing();
 #endif
 
     m_textCtrl = new wxTreeTextCtrl(this, itemEdit);
@@ -3386,18 +3386,11 @@
         else if (GetRootItem().IsOk())
             SelectItem(GetRootItem());
     }
-
-    /* after all changes have been done to the tree control,
-     * we actually redraw the tree when everything is over */
-
-    if (!m_dirty) return;
-    if (m_freezeCount) return;
 
-    m_dirty = false;
-
-    CalculatePositions();
-    Refresh();
-    AdjustMyScrollbars();
+    // after all changes have been done to the tree control,
+    // actually redraw the tree when everything is over
+    if (m_dirty)
+        DoDirtyProcessing();
 }
 
 void wxGenericTreeCtrl::CalculateSize( wxGenericTreeItem *item, wxDC &dc )
@@ -3665,5 +3658,17 @@
 }
 
 #endif  // WXWIN_COMPATIBILITY_2_2
+
+void wxGenericTreeCtrl::DoDirtyProcessing()
+{
+    if (m_freezeCount)
+        return;
+
+    m_dirty = false;
+
+    CalculatePositions();
+    Refresh();
+    AdjustMyScrollbars();
+}
 
 #endif // wxUSE_TREECTRL
Index: include/wx/generic/treectlg.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/generic/treectlg.h,v
retrieving revision 1.52
diff -u -r1.52 treectlg.h
--- include/wx/generic/treectlg.h	2005/01/30 12:44:46	1.52
+++ include/wx/generic/treectlg.h	2006/08/07 16:32:37
@@ -510,6 +510,8 @@
     void UnselectAllChildren( wxGenericTreeItem *item );
     void ChildrenClosing(wxGenericTreeItem* item);
 
+    void DoDirtyProcessing();
+
 private:
     DECLARE_EVENT_TABLE()
     DECLARE_DYNAMIC_CLASS(wxGenericTreeCtrl)

 	  	 
