diff -bru wxWidgets/src/msw/window.cpp /mnt/winxp/wxWidgets/src/msw/window.cpp
--- wxWidgets/src/msw/window.cpp	2012-02-15 13:16:34.000000000 -0800
+++ /mnt/winxp/wxWidgets/src/msw/window.cpp	2012-02-17 00:21:36.531250000 -0800
@@ -2250,7 +2250,10 @@
         flags |= TPM_RECURSE;
     }
 #endif // __WXWINCE__/!__WXWINCE__
-
+    BOOL accelActive = TRUE;
+	BOOL displayAcceleratorsInPopup;
+	SystemParametersInfo( SPI_GETKEYBOARDCUES, 0, &displayAcceleratorsInPopup, 0 );
+	SystemParametersInfo( SPI_SETKEYBOARDCUES, 0, &accelActive, 0 );
     ::TrackPopupMenu(GetHmenuOf(menu), flags, pt.x, pt.y, 0, GetHwnd(), NULL);
 
     // we need to do it right now as otherwise the events are never going to be
@@ -2261,7 +2264,7 @@
     // destroyed as soon as we return (it can be a local variable in the caller
     // for example) and so we do need to process the event immediately
     wxYieldForCommandsOnly();
-
+	SystemParametersInfo( SPI_SETKEYBOARDCUES, 0, &displayAcceleratorsInPopup, 0 );
     return true;
 }
 
