Index: include/wx/msw/private/fswatcher.h
===================================================================
--- include/wx/msw/private/fswatcher.h	(revision 76021)
+++ include/wx/msw/private/fswatcher.h	(working copy)
@@ -273,8 +273,8 @@
     struct wxEventProcessingData
     {
         wxEventProcessingData(const FILE_NOTIFY_INFORMATION* ne,
-                              const wxFSWatchEntryMSW* watch) :
-            nativeEvent(ne), watch(watch)
+                              const wxFSWatchEntryMSW* watch_) :
+            nativeEvent(ne), watch(watch_)
         {}
 
         const FILE_NOTIFY_INFORMATION* nativeEvent;
Index: src/generic/datavgen.cpp
===================================================================
--- src/generic/datavgen.cpp	(revision 76021)
+++ src/generic/datavgen.cpp	(working copy)
@@ -3074,9 +3074,9 @@
 class RowToTreeNodeJob: public DoJob
 {
 public:
-    RowToTreeNodeJob( unsigned int row , int current, wxDataViewTreeNode * node )
+    RowToTreeNodeJob( unsigned int row_ , int current, wxDataViewTreeNode * node )
     {
-        this->row = row;
+        this->row = row_;
         this->current = current;
         ret = NULL;
         parent = node;
Index: src/msw/ole/dataobj.cpp
===================================================================
--- src/msw/ole/dataobj.cpp	(revision 76021)
+++ src/msw/ole/dataobj.cpp	(working copy)
@@ -233,19 +233,19 @@
     {
     public:
         // Ctor takes ownership of the pointers.
-        SystemDataEntry(FORMATETC *pformatetc, STGMEDIUM *pmedium)
-            : pformatetc(pformatetc), pmedium(pmedium)
+        SystemDataEntry(FORMATETC *pformatetc_, STGMEDIUM *pmedium_)
+            : pformatetc(pformatetc_), pmedium(pmedium_)
         {
         }
 
         ~SystemDataEntry()
         {
-            delete pformatetc;
-            delete pmedium;
+            delete pformatetc_;
+            delete pmedium_;
         }
 
-        FORMATETC *pformatetc;
-        STGMEDIUM *pmedium;
+        FORMATETC *pformatetc_;
+        STGMEDIUM *pmedium_;
     };
     typedef wxVector<SystemDataEntry*> SystemData;
 
