Index: src/generic/datectlg.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/generic/datectlg.cpp,v
retrieving revision 1.33
diff -u -r1.33 datectlg.cpp
--- src/generic/datectlg.cpp	2005/04/16 11:07:30	1.33
+++ src/generic/datectlg.cpp	2005/04/26 18:13:06
@@ -204,7 +204,7 @@
 
     wxBitmap chkBmp(15,15);  // arbitrary
     if ( !wxBitmapButton::Create(parent, id, chkBmp,
-                                 pos, wxDefaultSize,
+                                 pos, size,
                                  style | (m_useDropArrow ? wxBU_AUTODRAW : wxNO_BORDER),
                                  validator) )
         return false;
@@ -421,7 +421,10 @@
 
     InheritAttributes();
 
-    m_txt = new wxTextCtrl(this, CTRLID_TXT, wxEmptyString, wxDefaultPosition, wxDefaultSize, TXTCTRL_FLAGS);
+    m_btn = new wxDropdownButton(this, CTRLID_BTN, wxDefaultPosition, wxSize(-1, size.y), wxBU_COMBO);
+    wxSize bs=m_btn->GetSize();
+    m_btn->SetSize(size.x-bs.x, 0, bs.x, bs.y);
+    m_txt = new wxTextCtrl(this, CTRLID_TXT, wxEmptyString, wxPoint(0,0), wxSize(size.x-bs.x, bs.y), TXTCTRL_FLAGS);
 
     m_txt->Connect(wxEVT_KEY_DOWN,
                    wxKeyEventHandler(wxDatePickerCtrlGeneric::OnEditKey),
@@ -430,7 +433,6 @@
                    wxFocusEventHandler(wxDatePickerCtrlGeneric::OnKillFocus),
                    NULL, this);
 
-    m_btn = new wxDropdownButton(this, CTRLID_BTN, wxDefaultPosition, wxDefaultSize, wxBU_COMBO);
 
     m_popup = new wxDatePopupInternal(this);
     m_popup->SetFont(GetFont());
@@ -504,8 +506,6 @@
 
     SetBestFittingSize(size);
 
-    SetBackgroundColour(m_txt->GetBackgroundColour());
-
     return true;
 }
 
@@ -766,11 +766,9 @@
     if ( m_btn )
     {
         wxSize sz = GetClientSize();
-
         wxSize bs=m_btn->GetSize();
-        int eh=m_txt->GetBestSize().y;
 
-        m_txt->SetSize(0, TXTPOSY, sz.x-bs.x, sz.y > eh ? eh-TXTPOSY : sz.y-TXTPOSY);
+        m_txt->SetSize(0, TXTPOSY, sz.x-bs.x, sz.y-2*TXTPOSY);
         m_btn->SetSize(sz.x - bs.x, 0, bs.x, sz.y);
     }
 

 	  	 
