Index: wxactivex.cpp
===================================================================
--- wxactivex.cpp	(revision 65537)
+++ wxactivex.cpp	(working copy)
@@ -1806,7 +1806,7 @@
 
 bool wxActiveX::MSWTranslateMessage(WXMSG *msg){
 	
-    if (msg->message == WM_KEYDOWN) {
+    if (msg->message == WM_KEYDOWN || msg->message == WM_KEYUP) {
         if ( m_oleInPlaceActiveObject.Ok() )
         {
             HRESULT result = m_oleInPlaceActiveObject->TranslateAccelerator(msg);
