--- wxWidgets-3.0.2.backup\src\msw\slider.cpp	Mon Oct 06 22:34:04 2014
+++ wxWidgets-3.0.2\src\msw\slider.cpp	Tue Feb 16 16:45:02 2016
@@ -471,6 +471,7 @@
             (longestLabelWidth / 2);
 
         int ySlider = y;
+        int i_vgap = 0 ;
 
         if ( HasFlag(wxSL_VALUE_LABEL) )
         {
@@ -496,13 +497,14 @@
                 x + width - maxLabelWidth,
                 yLabelMinMax,
                 maxLabelWidth, labelHeight);
+            i_vgap = VGAP ;
         }
 
         // position the slider itself along the top/bottom edge
         wxSliderBase::DoMoveWindow(
-            x + minLabelWidth + VGAP,
+            x + minLabelWidth + i_vgap,
             ySlider,
-            width  - (minLabelWidth + maxLabelWidth  + (VGAP*2)),
+            width  - (minLabelWidth + maxLabelWidth  + (i_vgap*2)),
             THUMB + tickOffset);
     }
 }
