diff -ur --exclude='*.o' --exclude='config*' --exclude=Makefile wxWidgets-2.6.1/src/xrc/xh_menu.cpp wxWidgets-2.6.1.mine/src/xrc/xh_menu.cpp
--- wxWidgets-2.6.1/src/xrc/xh_menu.cpp	2005-02-07 18:59:27.000000000 -0200
+++ wxWidgets-2.6.1.mine/src/xrc/xh_menu.cpp	2005-07-15 13:04:19.000000000 -0300
@@ -37,7 +37,12 @@
 {
     if (m_class == wxT("wxMenu"))
     {
-        wxMenu *menu = new wxMenu(GetStyle());
+	wxMenu *menu = NULL;
+	if(m_instance)
+	    menu = wxStaticCast(m_instance, wxMenu);
+	if(!menu)
+	    menu = new wxMenu(GetStyle());
+
         wxString title = GetText(wxT("label"));
         wxString help = GetText(wxT("help"));
 
diff -ur --exclude='*.o' --exclude='config*' --exclude=Makefile wxWidgets-2.6.1/utils/wxrc/wxrc.cpp wxWidgets-2.6.1.mine/utils/wxrc/wxrc.cpp
--- wxWidgets-2.6.1/utils/wxrc/wxrc.cpp	2005-05-31 17:50:28.000000000 -0300
+++ wxWidgets-2.6.1.mine/utils/wxrc/wxrc.cpp	2005-07-15 13:35:37.000000000 -0300
@@ -32,7 +32,9 @@
 #include "wx/filename.h"
 #include "wx/wfstream.h"
 #include "wx/utils.h"
+#include "wx/hashset.h"
 
+WX_DECLARE_HASH_SET(wxString, wxStringHash, wxStringEqual, StringSet);
 
 class XRCWidgetData
 {
@@ -55,6 +57,7 @@
 private:
     wxString m_className;
     wxString m_parentClassName;
+    StringSet m_ancestorClassNames;
     ArrayOfXRCWidgetData m_wdata;
 
     void BrowseXmlNode(wxXmlNode* node)
@@ -78,9 +81,26 @@
     }
 
 public:
-    XRCWndClassData(const wxString& className,const wxString& parentClassName, const wxXmlNode* node) :
+    XRCWndClassData(const wxString& className,const wxString& parentClassName, 
+	const wxXmlNode* node) :
         m_className(className) , m_parentClassName(parentClassName)
     {
+	if(className == "wxMenu")
+	{
+	    m_ancestorClassNames.insert("wxMenu");
+	    m_ancestorClassNames.insert("wxMenuBar");
+	}
+	else if(className == "wxMDIChildFrame")
+	    m_ancestorClassNames.insert("wxMDIParentFrame");
+	else if(className == "wxMenuBar"
+	    || className == "wxStatusBar" 
+	    || className == "wxToolBar")
+	{
+	    m_ancestorClassNames.insert("wxFrame");
+	}
+	else
+	    m_ancestorClassNames.insert("wxWindow");
+		
         BrowseXmlNode(node->GetChildren());
     }
 
@@ -118,8 +138,8 @@
                 _T(" ") + w.GetClass() + _T("* ") + w.GetName()
                 + _T(";\n"));
         }
-        file.Write(_T("\nprivate:\n void InitWidgetsFromXRC(){\n")
-                   _T("  wxXmlResource::Get()->LoadObject(this,NULL,_T(\"")
+        file.Write(_T("\nprivate:\n void InitWidgetsFromXRC(wxWindow *parent){\n")
+                   _T("  wxXmlResource::Get()->LoadObject(this,parent,_T(\"")
                    +  m_className
                    +  _T("\"), _T(\"")
                    +  m_parentClassName
@@ -135,22 +155,40 @@
                         + w.GetName()
                         + _T("\",")
                         + w.GetClass()
-                        + _T(");\n")
-                );
+                        + _T(");\n"));
         }
         file.Write(_T(" }\n"));
 
-        file.Write(
-            _T("public:\n")
-            + m_className
-            + _T("::")
-            + m_className
+        file.Write( _T("public:\n"));
+
+	if(m_ancestorClassNames.size() == 1)
+	{
+	    file.Write(m_className
+            + _T("(")+*m_ancestorClassNames.begin()+_T(" *parent=NULL){\n")
+            + _T("  InitWidgetsFromXRC((wxWindow *)parent);\n")
+              _T(" }\n")
+              _T("};\n"));
+	}
+	else
+	{
+	    file.Write(m_className
             + _T("(){\n")
-            + _T("  InitWidgetsFromXRC();\n")
+            + _T("  InitWidgetsFromXRC(NULL);\n")
               _T(" }\n")
-              _T("};\n")
-        );
-    };
+              _T("};\n"));
+	   
+	    for(StringSet::const_iterator it=m_ancestorClassNames.begin();
+		it!=m_ancestorClassNames.end(); ++it)
+	    {
+	    	file.Write(m_className
+		     + _T("(")+*it+_T(" *parent){\n")
+		    + _T("  InitWidgetsFromXRC((wxWindow *)parent);\n")
+		      _T(" }\n")
+		      _T("};\n")
+		);
+	    }
+	}
+    }
 };
 WX_DECLARE_OBJARRAY(XRCWndClassData,ArrayOfXRCWndClassData);
 WX_DEFINE_OBJARRAY(ArrayOfXRCWndClassData);

 	  	 
