Index: anybutton.cpp
===================================================================
--- anybutton.cpp	(revision 68871)
+++ anybutton.cpp	(working copy)
@@ -3,7 +3,7 @@
 // Purpose:     wxAnyButton
 // Author:      Julian Smart
 // Created:     1998-01-04 (extracted from button.cpp)
-// RCS-ID:      $Id: anybutton.cpp 67384 2011-04-03 20:31:32Z DS $
+// RCS-ID:      $Id$
 // Copyright:   (c) Julian Smart
 // Licence:     wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
@@ -690,8 +690,8 @@
 #endif // wxUSE_UXTHEME
         {
             m_imageData = new wxODButtonImageData(this, bitmap);
-            MakeOwnerDrawn();
         }
+        MakeOwnerDrawn();
     }
     else
     {
