diff -ru2 -N -xCVS -x.#* -x*.orig -x*.vcproj -x*.cmake -x*.sln -x*.suo -x*.rej -x*~ -x*.o -x*.obj -x*.a -x*.lib -x*setup.h -x*y*.c wxWidgets.orig/include/wx/list.h wxWidgets/include/wx/list.h
--- wxWidgets.orig/include/wx/list.h	Mon Mar  7 09:07:32 2005
+++ wxWidgets/include/wx/list.h	Mon Mar  7 09:41:58 2005
@@ -99,29 +99,39 @@
     WX_DECLARE_LIST_XO(elT*, liT, decl)
 
-#define WX_DECLARE_LIST_XO(elT, liT, decl)                                    \
-    decl liT : public std::list<elT>                                          \
-    {                                                                         \
-    private:                                                                  \
-        bool m_destroy;                                                       \
-    private:                                                                  \
-        class SortCompareFunction                                             \
-        {                                                                     \
-        private:                                                              \
-            wxSortCompareFunction m_CompFunc;                                 \
-        public:                                                               \
-            SortCompareFunction( wxSortCompareFunction compfunc )             \
-                : m_CompFunc( compfunc ) {}                                   \
-            bool operator()( const elT X, const elT Y ) const                 \
-                { return ( m_CompFunc( X, Y ) < 0 ); }                        \
-        };                                                                    \
-                                                                              \
-        typedef elT _WX_LIST_ITEM_TYPE_##liT;                                 \
-        static void DeleteFunction( const _WX_LIST_ITEM_TYPE_##liT X );       \
-    public:                                                                   \
-        class compatibility_iterator                                          \
-        {                                                                     \
-        private:                                                              \
-            typedef liT::iterator iterator;                                   \
-            friend class liT;                                                 \
+#define WX_DECLARE_LIST_XO(elT, liT, decl)                                    \
+    decl liT;                                                                 \
+                                                                              \
+    /* Workaround for broken VC6 STL incorrectly requires a std::greater<> */ \
+    /* to be passed into std::list::sort() */                                 \
+    template <>                                                               \
+    struct std::greater<elT>                                                  \
+    {                                                                         \
+        private:                                                              \
+            wxSortCompareFunction m_CompFunc;                                 \
+        public:                                                               \
+            greater( wxSortCompareFunction compfunc = NULL )                  \
+                : m_CompFunc( compfunc ) {}                                   \
+            bool operator()(const elT X, const elT Y) const                   \
+                {                                                             \
+                    return m_CompFunc ?                                       \
+                        ( m_CompFunc( X, Y ) < 0 ) :                          \
+                        ( X > Y );                                            \
+                }                                                             \
+    };                                                                        \
+                                                                              \
+    decl liT : public std::list<elT>                                          \
+    {                                                                         \
+    private:                                                                  \
+        bool m_destroy;                                                       \
+    private:                                                                  \
+        typedef elT _WX_LIST_ITEM_TYPE_##liT;                                 \
+        static void DeleteFunction( const _WX_LIST_ITEM_TYPE_##liT X );       \
+    public:                                                                   \
+        class compatibility_iterator                                          \
+        {                                                                     \
+        private:                                                              \
+          /* Workaround for broken VC6 nested class name resolution */        \
+          typedef std::list<elT>::iterator iterator;                          \
+          friend class liT;                                                   \
         private:                                                              \
             iterator m_iter;                                                  \
@@ -261,6 +271,7 @@
         }                                                                     \
                                                                               \
-        void Sort( wxSortCompareFunction compfunc )                           \
-            { sort( SortCompareFunction( compfunc ) ); }                      \
+        /* Workaround for broken VC6 std::list::sort() see above */           \
+        void Sort( wxSortCompareFunction compfunc )                           \
+            { sort( std::greater<elT>( compfunc ) ); }                        \
                                                                               \
         ~liT() { Clear(); }                                                   \

 	  	 
