Index: wx-config.in
===================================================================
--- wx-config.in	(revision 63002)
+++ wx-config.in	(working copy)
@@ -353,7 +353,7 @@
 
     # use 2.8 or 2.9 version of the mask: the difference is the presence of
     # debug type in pre-2.9
-    if [ $is29orlater == 0 ]; then
+    if [ $is29orlater = 0 ]; then
         eval echo "\${$1_host:+\$$1_host-}\${$1_toolkit}\${$1_widgetset}-\${$1_chartype}-\${$1_debugtype}\${$1_linkage:+-\$$1_linkage}-\${$1_version}\${$1_flavour}"
     else
         eval echo "\${$1_host:+\$$1_host-}\${$1_toolkit}\${$1_widgetset}-\${$1_chartype}\${$1_linkage:+-\$$1_linkage}-\${$1_version}\${$1_flavour}"
