Index: src/common/utilscmn.cpp
===================================================================
--- src/common/utilscmn.cpp	(revision 51611)
+++ src/common/utilscmn.cpp	(working copy)
@@ -1031,6 +1031,15 @@
     sei.lpVerb = _T("open");
     sei.nShow = SW_SHOWNORMAL;
 
+#ifdef __WXWINCE__
+    // sei.hInstApp is only set if the function fails
+    if ( ::ShellExecuteEx(&sei) )
+    {
+        return true;
+    }
+
+    wxLogDebug(wxT("ShellExecuteEx failed %i"), (int) sei.hInstApp );
+#else
     ::ShellExecuteEx(&sei);
 
     const int nResult = (int) sei.hInstApp;
@@ -1046,6 +1055,8 @@
 #endif // __WXDEBUG__
         return true;
     }
+#endif
+
 #elif defined(__WXCOCOA__)
     // NOTE: We need to call the real implementation from src/cocoa/utils.mm
     // because the code must use Objective-C features.

 	  	 
