Index: wxWindows/src/gtk/pen.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/gtk/pen.cpp,v
retrieving revision 1.23
diff -b -u -2 -r1.23 pen.cpp
--- wxWindows/src/gtk/pen.cpp	2003/04/11 14:01:29	1.23
+++ wxWindows/src/gtk/pen.cpp	2003/05/24 17:02:27
@@ -31,6 +31,6 @@
         m_joinStyle = wxJOIN_ROUND;
         m_capStyle = wxCAP_ROUND;
-        m_dash = (wxGTKDash*) NULL;
         m_countDashes = 0;
+        m_dash = (wxGTKDash*) NULL;
     }
     
@@ -44,18 +44,26 @@
         m_colour = data.m_colour;
         m_countDashes = data.m_countDashes;
-/*
-        if (data.m_dash)  TODO
-            m_dash = new
-*/
-        m_dash = data.m_dash;
+        m_dash = (wxGTKDash*) NULL;
+
+        if (data.m_dash)
+        {
+            m_dash = (wxGTKDash*)malloc(m_countDashes*sizeof(wxGTKDash));
+            memcpy(m_dash, data.m_dash, m_countDashes*sizeof(wxGTKDash));
+        }
+    }
+
+    ~wxPenRefData()
+    {
+        if (m_dash) free(m_dash);
     }
 
     bool operator == (const wxPenRefData& data) const
     {
-        // It is impossible to tell if the dashes have changed
-        // so the only thing to do is assume they have
-        if (m_countDashes != 0 || data.m_countDashes != 0)
-            return false;
+        if (m_countDashes != data.m_countDashes)
+            return FALSE;
 
+        if (m_dash && memcmp(m_dash, data.m_dash, m_countDashes*sizeof(wxGTKDash)))
+            return FALSE;
+
         return (m_style == data.m_style &&
                 m_width == data.m_width &&
@@ -123,6 +131,20 @@
     AllocExclusive();
     
+    wxCHECK_RET((number_of_dashes >= 0) && ((number_of_dashes > 0) == (dash != NULL)),
+                wxT("invalid pen dashes"));
+
     M_PENDATA->m_countDashes = number_of_dashes;
-    M_PENDATA->m_dash = (wxGTKDash *)dash; /* TODO */
+
+    if (M_PENDATA->m_dash)
+    {
+        free(M_PENDATA->m_dash);
+        M_PENDATA->m_dash = (wxGTKDash*)NULL;
+    }
+
+    if (dash)
+    {
+        M_PENDATA->m_dash = (wxGTKDash*)malloc(number_of_dashes*sizeof(wxGTKDash));
+        memcpy(M_PENDATA->m_dash, dash, number_of_dashes*sizeof(wxGTKDash));
+    }
 }
 

 	  	 
