diff -bu1 -r wxWidgets/build/bakefiles/Bakefiles.bkgen wxW/build/bakefiles/Bakefiles.bkgen
--- wxWidgets/build/bakefiles/Bakefiles.bkgen	2004-04-06 00:04:09.000000000 +0200
+++ wxW/build/bakefiles/Bakefiles.bkgen	2004-04-07 11:05:11.000000000 +0200
@@ -78,3 +78,2 @@
     <del-formats files="../../contrib/utils/wxrc/*">msevc4prj</del-formats>
-    <del-formats files="../../tests/*">msevc4prj</del-formats>
 
diff -bu1 -r wxWidgets/build/bakefiles/common_samples.bkl wxW/build/bakefiles/common_samples.bkl
--- wxWidgets/build/bakefiles/common_samples.bkl	2004-03-03 22:56:16.000000000 +0100
+++ wxW/build/bakefiles/common_samples.bkl	2004-04-06 09:51:53.000000000 +0200
@@ -64,2 +64,6 @@
 
+    <template id="wx_test" template="wx_sample">
+        <cppflags>$(CPPUNIT_CFLAGS)</cppflags>
+        <ldflags>$(CPPUNIT_LIBS)</ldflags>
+    </template>
     <template id="wx_test_console" template="wx_sample_console">
diff -bu1 -r wxWidgets/include/wx/cppunit.h wxW/include/wx/cppunit.h
--- wxWidgets/include/wx/cppunit.h	2004-04-01 09:17:40.000000000 +0200
+++ wxW/include/wx/cppunit.h	2004-04-07 12:09:13.000000000 +0200
@@ -34,2 +34,6 @@
 #include <cppunit/extensions/HelperMacros.h>
+#include <cppunit/Outputter.h>
+#include <cppunit/TestFailure.h>
+#include <cppunit/TestResultCollector.h>
+#include <cppunit/TestResult.h>
 #include "wx/afterstd.h"
diff -bu1 -r wxWidgets/tests/longlong/longlong.cpp wxW/tests/longlong/longlong.cpp
--- wxWidgets/tests/longlong/longlong.cpp	2004-04-06 14:15:09.000000000 +0200
+++ wxW/tests/longlong/longlong.cpp	2004-04-07 16:03:59.000000000 +0200
@@ -239,2 +239,5 @@
     {
+        if ( ( n % 100 ) == 0 )
+            wxSafeYield();
+
         wxLongLong a = RAND_LL();
diff -bu1 -r wxWidgets/tests/test.bkl wxW/tests/test.bkl
--- wxWidgets/tests/test.bkl	2004-04-07 09:16:31.000000000 +0200
+++ wxW/tests/test.bkl	2004-04-07 10:57:56.000000000 +0200
@@ -7,6 +7,3 @@
 
-    <exe id="test" template="wx_test_console"
-                   template_append="wx_append_base">
-        <sources>
-            test.cpp
+    <set var="TEST_BASE">
             mbconv/main.cpp
@@ -17,4 +14,35 @@
             longlong/longlong.cpp
+    </set>
+
+    <set var="TEST_CORE">
+    </set>
+
+    <set var="TEST_NET">
+    </set>
+
+    <exe id="test" template="wx_test_console"
+                   template_append="wx_append_base"
+                   cond="FORMAT!='msevc4prj'">
+        <sources>
+            $(TEST_BASE)
+            $(TEST_NET)
+            test.cpp
+        </sources>
+        <wx-lib>base</wx-lib>
+        <wx-lib>net</wx-lib>
+    </exe>
+
+    <exe id="test_gui" template="wx_test"
+                       template_append="wx_append_base"
+                       cond="USE_GUI=='1'">
+        <sources>
+            $(TEST_BASE)
+            $(TEST_NET)
+            $(TEST_CORE)
+            test_gui.cpp
         </sources>
         <wx-lib>base</wx-lib>
+        <wx-lib>net</wx-lib>
+        <wx-lib>core</wx-lib>
+        <win32-res>test.rc</win32-res>
     </exe>
Only in wxW/tests: test.ico
Only in wxW/tests: test.rc
Only in wxW/tests: test_gui.cpp

 	  	 
