Index: src/generic/grideditors.cpp
===================================================================
--- src/generic/grideditors.cpp	(revision 65218)
+++ src/generic/grideditors.cpp	(working copy)
@@ -545,9 +552,11 @@
     long pos;
 
 #if wxUSE_UNICODE
-    ch = event.GetUnicodeKey();
-    if (ch <= 127)
-        ch = (wxChar)event.GetKeyCode();
+    ch = (wxChar)event.GetKeyCode();
+    if( (ch != (wxChar)WXK_DELETE) && (ch != (wxChar)WXK_BACK) )
+    {
+        ch = event.GetUnicodeKey();
+    }
 #else
     ch = (wxChar)event.GetKeyCode();
 #endif
