--- src/common/memory.cpp.original	2013-07-30 08:04:26.000000000 +0800
+++ src/common/memory.cpp	2013-07-30 08:05:36.000000000 +0800
@@ -895,10 +895,10 @@
 }
 
-void * operator new (size_t size)
+void * operator new (size_t size) throw (std::bad_alloc)
 {
     return wxDebugAlloc(size, NULL, 0, false);
 }
 
-void operator delete (void * buf)
+void operator delete (void * buf) throw ()
 {
     wxDebugFree(buf, false);
@@ -906,5 +906,5 @@
 
 #if wxUSE_ARRAY_MEMORY_OPERATORS
-void * operator new[] (size_t size)
+void * operator new[] (size_t size) throw (std::bad_alloc)
 {
     return wxDebugAlloc(size, NULL, 0, false, true);
@@ -916,5 +916,5 @@
 }
 
-void operator delete[] (void * buf)
+void operator delete[] (void * buf) throw ()
 {
   wxDebugFree(buf, true);
