Index: docs/changes.txt
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/docs/changes.txt,v
retrieving revision 1.583
diff -u -B -r1.583 changes.txt
--- docs/changes.txt	2004/12/08 17:51:48	1.583
+++ docs/changes.txt	2004/12/10 03:47:17
@@ -7,6 +7,9 @@
 
 All:
 
+- The wxTextCtrl EVT_TEXT command event does not carry the controls whole string
+  anymore. Use wxTextCtrl::GetValue() yourself on the wxTextCtrl retrieved with
+  event.GetEventObject() instead of event.GetString() if necessary (Mart Raudsepp)
 - new classes for reading and writing ZIP files (M.J.Wetherell)
 - Norwegian (Bokmål) translation added (Hans F. Nordhaug)
 - wxDynamicLibrary::HasSymbol() added
@@ -44,6 +47,7 @@
 
 - fixed wxFileDialog::SetWildcard()
 - native file dialog is now used if available (Zbigniew Zagorski)
+- wxTextCtrl text insertion efficiency fixed. (Mart Raudsepp)
 - implemented wxTextCtrl::Freeze() and Thaw(). The GtkTextBuffer
   is not a valid one during frozen state. Get a pointer to it right
   after wxTextCtrl creation if you really need to. (Mart Raudsepp)
Index: docs/latex/wx/tchanges.tex
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/docs/latex/wx/tchanges.tex,v
retrieving revision 1.7
diff -u -B -r1.7 tchanges.tex
--- docs/latex/wx/tchanges.tex	2004/12/08 17:00:29	1.7
+++ docs/latex/wx/tchanges.tex	2004/12/10 03:47:19
@@ -139,6 +139,8 @@
 
 - \texttt{wxTheApp} can't be assigned to any longer, use \helpref{wxApp::SetInstance()}{wxappsetinstance} instead
 
+- The wxTextCtrl EVT_TEXT command event does not carry the controls whole string anymore. Use wxTextCtrl::GetValue() yourself on the wxTextCtrl retrieved with event.GetEventObject() instead of event.GetString() if necessary.
+
 - wxFileType::GetIcon() returns wxIconLocation, not wxIcon
 
 - wxColourDatabase is not a wxList any more, use AddColour to add new colours
Index: src/gtk/textctrl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/gtk/textctrl.cpp,v
retrieving revision 1.194
diff -u -B -r1.194 textctrl.cpp
--- src/gtk/textctrl.cpp	2004/12/05 23:03:09	1.194
+++ src/gtk/textctrl.cpp	2004/12/10 03:47:20
@@ -202,7 +202,6 @@
 
     wxCommandEvent event( wxEVT_COMMAND_TEXT_UPDATED, win->GetId() );
     event.SetEventObject( win );
-    event.SetString( win->GetValue() );
     win->GetEventHandler()->ProcessEvent( event );
 }
 
Index: src/mac/carbon/textctrl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/mac/carbon/textctrl.cpp,v
retrieving revision 1.123
diff -u -B -r1.123 textctrl.cpp
--- src/mac/carbon/textctrl.cpp	2004/12/02 21:35:40	1.123
+++ src/mac/carbon/textctrl.cpp	2004/12/10 03:47:21
@@ -478,7 +478,6 @@
         GetPeer()->Cut() ;    
 
         wxCommandEvent event(wxEVT_COMMAND_TEXT_UPDATED, m_windowId);
-        event.SetString( GetValue() ) ;
         event.SetEventObject( this );
         GetEventHandler()->ProcessEvent(event);
       }
@@ -492,7 +491,6 @@
         // eventually we should add setting the default style again
 
         wxCommandEvent event(wxEVT_COMMAND_TEXT_UPDATED, m_windowId);
-        event.SetString( GetValue() ) ;
         event.SetEventObject( this );
         GetEventHandler()->ProcessEvent(event);
     }
@@ -901,7 +899,6 @@
          key == WXK_BACK)
     {
         wxCommandEvent event1(wxEVT_COMMAND_TEXT_UPDATED, m_windowId);
-        event1.SetString( GetValue() ) ;
         event1.SetEventObject( this );
         wxPostEvent(GetEventHandler(),event1);
     }
Index: src/mac/classic/textctrl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/mac/classic/textctrl.cpp,v
retrieving revision 1.7
diff -u -B -r1.7 textctrl.cpp
--- src/mac/classic/textctrl.cpp	2004/08/18 15:37:00	1.7
+++ src/mac/classic/textctrl.cpp	2004/12/10 03:47:22
@@ -1039,7 +1039,6 @@
             TXNConvertToPublicScrap();
         }
         wxCommandEvent event(wxEVT_COMMAND_TEXT_UPDATED, m_windowId);
-        event.SetString( GetValue() ) ;
         event.SetEventObject( this );
         GetEventHandler()->ProcessEvent(event);
       }
@@ -1062,7 +1061,6 @@
             SetStyle( kTXNUseCurrentSelection , kTXNUseCurrentSelection , GetDefaultStyle() ) ;
         }
         wxCommandEvent event(wxEVT_COMMAND_TEXT_UPDATED, m_windowId);
-        event.SetString( GetValue() ) ;
         event.SetEventObject( this );
         GetEventHandler()->ProcessEvent(event);
     }
@@ -1704,7 +1702,6 @@
          key == WXK_BACK)
     {
         wxCommandEvent event1(wxEVT_COMMAND_TEXT_UPDATED, m_windowId);
-        event1.SetString( GetValue() ) ;
         event1.SetEventObject( this );
         wxPostEvent(GetEventHandler(),event1);
     }
Index: src/msw/textctrl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/msw/textctrl.cpp,v
retrieving revision 1.235
diff -u -B -r1.235 textctrl.cpp
--- src/msw/textctrl.cpp	2004/12/05 14:01:29	1.235
+++ src/msw/textctrl.cpp	2004/12/10 03:47:22
@@ -1835,7 +1835,6 @@
 
     wxCommandEvent event(wxEVT_COMMAND_TEXT_UPDATED, GetId());
     InitCommandEvent(event);
-    event.SetString(GetValue());
 
     return ProcessCommand(event);
 }
Index: src/os2/textctrl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/os2/textctrl.cpp,v
retrieving revision 1.42
diff -u -B -r1.42 textctrl.cpp
--- src/os2/textctrl.cpp	2004/05/23 20:53:15	1.42
+++ src/os2/textctrl.cpp	2004/12/10 03:47:23
@@ -1103,7 +1103,6 @@
                                           );
 
                 InitCommandEvent(vEvent);
-                vEvent.SetString((char*)GetValue().c_str());
                 ProcessCommand(vEvent);
             }
             break;
Index: src/univ/textctrl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/univ/textctrl.cpp,v
retrieving revision 1.31
diff -u -B -r1.31 textctrl.cpp
--- src/univ/textctrl.cpp	2004/11/05 21:04:46	1.31
+++ src/univ/textctrl.cpp	2004/12/10 03:47:24
@@ -4680,7 +4680,6 @@
 
         wxCommandEvent event(wxEVT_COMMAND_TEXT_UPDATED, GetId());
         InitCommandEvent(event);
-        event.SetString(GetValue());
         GetEventHandler()->ProcessEvent(event);
 
         // as the text changed...

 	  	 
