Index: include/wx/dynarray.h
===================================================================
--- include/wx/dynarray.h	(revision 73426)
+++ include/wx/dynarray.h	(working copy)
@@ -828,14 +828,13 @@
 // Some commonly used predefined base arrays
 // ----------------------------------------------------------------------------
 
-WX_DECLARE_USER_EXPORTED_BASEARRAY(const void *, wxBaseArrayPtrVoid,
-                                   WXDLLIMPEXP_BASE);
-WX_DECLARE_USER_EXPORTED_BASEARRAY(char, wxBaseArrayChar, WXDLLIMPEXP_BASE);
-WX_DECLARE_USER_EXPORTED_BASEARRAY(short, wxBaseArrayShort, WXDLLIMPEXP_BASE);
-WX_DECLARE_USER_EXPORTED_BASEARRAY(int, wxBaseArrayInt, WXDLLIMPEXP_BASE);
-WX_DECLARE_USER_EXPORTED_BASEARRAY(long, wxBaseArrayLong, WXDLLIMPEXP_BASE);
-WX_DECLARE_USER_EXPORTED_BASEARRAY(size_t, wxBaseArraySizeT, WXDLLIMPEXP_BASE);
-WX_DECLARE_USER_EXPORTED_BASEARRAY(double, wxBaseArrayDouble, WXDLLIMPEXP_BASE);
+WX_DECLARE_BASEARRAY(const void *, wxBaseArrayPtrVoid);
+WX_DECLARE_BASEARRAY(char, wxBaseArrayChar);
+WX_DECLARE_BASEARRAY(short, wxBaseArrayShort);
+WX_DECLARE_BASEARRAY(int, wxBaseArrayInt);
+WX_DECLARE_BASEARRAY(long, wxBaseArrayLong);
+WX_DECLARE_BASEARRAY(size_t, wxBaseArraySizeT);
+WX_DECLARE_BASEARRAY(double, wxBaseArrayDouble);
 
 // ----------------------------------------------------------------------------
 // Convenience macros to define arrays from base arrays
