Index: calctrl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/generic/calctrl.cpp,v
retrieving revision 1.77
diff -u -r1.77 calctrl.cpp
--- calctrl.cpp	2006/04/16 22:18:03	1.77
+++ calctrl.cpp	2006/05/27 08:12:41
@@ -788,7 +789,7 @@
 {
     int yDiff;

-    if ( !HasFlag(wxCAL_SEQUENTIAL_MONTH_SELECTION) )
+    if ( !HasFlag(wxCAL_SEQUENTIAL_MONTH_SELECTION) && m_staticMonth)
     {
         wxSize sizeCombo = m_comboMonth->GetSize();
         wxSize sizeStatic = m_staticMonth->GetSize();
@@ -819,7 +820,7 @@
     if ( !(GetWindowStyle() & wxCAL_SEQUENTIAL_MONTH_SELECTION) )
     {
         // our real top corner is not in this position
-        if ( y )
+        if ( y && GetMonthControl())
         {
             *y -= GetMonthControl()->GetSize().y + VERT_MARGIN;
         }

 	  	 
