diff -uNr wxWidgets-trunk\src/msw/toolbar.cpp wxWidgets-work\src/msw/toolbar.cpp
--- wxWidgets-trunk\src/msw/toolbar.cpp	2014-01-20 13:25:52.110105200 +0100
+++ wxWidgets-work\src/msw/toolbar.cpp	2014-02-02 16:58:12.882630400 +0100
@@ -659,11 +659,13 @@
             if ( bmpDisabled.IsOk() )
             {
                 const wxSize sizeBitmap = bmpDisabled.GetSize();
+                // For Common Controls prior to v6.0 image list with disbled bitmaps doesn't support transparency.
+                // In this case we need to enforce to use the list of bitmaps with bitmap masks.
                 m_disabledImgList = new wxImageList
                                         (
                                             sizeBitmap.x,
                                             sizeBitmap.y,
-                                            bmpDisabled.GetMask() != NULL,
+                                            wxApp::GetComCtl32Version() < 600 || bmpDisabled.GetMask() != NULL,
                                             GetToolsCount()
                                         );
                 break;
@@ -837,7 +839,6 @@
                             }
                         }
 #endif // wxREMAP_BUTTON_COLOURS
-
                         bmpDisabled = wxBitmap(imgGreyed);
                     }
 #endif // wxUSE_IMAGE
@@ -846,6 +847,12 @@
                     if ( remapValue == Remap_Buttons )
                         MapBitmap(bmpDisabled.GetHBITMAP(), w, h);
 #endif // wxREMAP_BUTTON_COLOURS
+                    // For Common Controls prior to v6.0 image list with
+                    // disbled bitmaps doesn't support transparency.
+                    // In this case we need to disable alpha channel
+                    // and corresponding bitmap mask will be created in wxImageList::Add().
+                    if( wxApp::GetComCtl32Version() < 600 )
+                        bmpDisabled.ResetAlpha();
 
                     m_disabledImgList->Add(bmpDisabled);
                 }
