
C:\Shortcut\wxWindows>cvs -z9 diff -u src\msw\treectrl.cpp 
Index: src/msw/treectrl.cpp
===================================================================
RCS file: /home/wxcvs/wxWindows/src/msw/treectrl.cpp,v
retrieving revision 1.124
diff -u -r1.124 treectrl.cpp
--- src/msw/treectrl.cpp	17 May 2002 23:54:42 -0000	1.124
+++ src/msw/treectrl.cpp	6 Jun 2002 12:38:15 -0000
@@ -2573,7 +2573,7 @@
             // comctl32.dll or our code?
             {
                 NM_TREEVIEW* tv = (NM_TREEVIEW*)lParam;
-                if ( tv->action == TVE_EXPAND )
+                if ( tv->action == TVE_EXPAND || tv->action == TVE_COLLAPSE)
                 {
                     wxTreeItemId id = (WXHTREEITEM)tv->itemNew.hItem;
 

 	  	 
