--- dc.cpp.orig	Fri Feb 27 14:01:45 2004
+++ dc.cpp	Fri Feb 27 17:54:13 2004
@@ -1256,6 +1256,7 @@
 	m_macFontInstalled = false ;
   return TRUE;
 }
+#if 0
 inline Fixed	IntToFixed( int inInt )
 	{
 		return (((SInt32) inInt) << 16);
@@ -1264,6 +1265,7 @@
 	{
 		return (((SInt32) inFixed) >> 16);
 	}
+#endif
 
 void  wxDC::DoDrawRotatedText(const wxString& str, wxCoord x, wxCoord y,
                               double angle)

 	  	 
