Index: src/msw/button.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/button.cpp,v
retrieving revision 1.75
diff -b -u -2 -r1.75 button.cpp
--- src/msw/button.cpp	2003/09/24 00:41:05	1.75
+++ src/msw/button.cpp	2004/01/10 16:44:14
@@ -184,4 +184,7 @@
     if ( style & wxBU_BOTTOM )
         msStyle |= BS_BOTTOM;
+    // flat 2d buttons
+    if ( style & wxNO_BORDER )
+        msStyle |= BS_FLAT;
 #endif // __WIN32__
 

 	  	 
