Index: /wxWidgets/trunk/include/wx/event.h
===================================================================
--- /wxWidgets/trunk/include/wx/event.h	(revision 71667)
+++ /wxWidgets/trunk/include/wx/event.h	(working copy)
@@ -1244,7 +1244,7 @@
     {
         // make sure our string member (which uses COW, aka refcounting) is not
         // shared by other wxString instances:
-        SetString(GetString().c_str());
+        SetString(GetString().Clone());
     }
 
     virtual wxEvent *Clone() const
