Index: utilscmn.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/common/utilscmn.cpp,v
retrieving revision 1.140
diff -b -u -2 -r1.140 utilscmn.cpp
--- utilscmn.cpp	2005/05/13 12:07:08	1.140
+++ utilscmn.cpp	2005/05/15 18:21:56
@@ -584,7 +584,10 @@
     if ( command.empty() || wxExecute(command) == -1)
     {
+        int nResult; //HINSTANCE error code
+
+#if !defined(__WXWINCE__)
         // CYGWIN and MINGW may have problems - so load ShellExecute
         // dynamically
-        typedef HINSTANCE (*LPShellExecute)(HWND hwnd, const wxChar* lpOperation,
+        typedef HINSTANCE (WINAPI *LPShellExecute)(HWND hwnd, const wxChar* lpOperation,
                                             const wxChar* lpFile,
                                             const wxChar* lpParameters,
@@ -616,18 +619,44 @@
         // Windows sometimes doesn't open the browser correctly when using mime
         // types, so do ShellExecute - i.e. start <url> (from James Carroll)
-        unsigned int nResult = (int) (*lpShellExecute)(NULL, NULL, finalurl.c_str(),
+        nResult = (int) (*lpShellExecute)(NULL, NULL, finalurl.c_str(),
                                                        NULL, wxT(""), SW_SHOWNORMAL);
-
         // Unload Shell32.dll
         ::FreeLibrary(hShellDll);
+#else
+        //Windows CE does not have normal ShellExecute - but it has
+        //ShellExecuteEx all the way back to version 1.0
+
+
+        //Set up the SHELLEXECUTEINFO structure to pass to ShellExecuteEx
+        SHELLEXECUTEINFO sei;
+        sei.cbSize = sizeof(SHELLEXECUTEINFO);
+        sei.dwHotKey = 0;
+        sei.fMask = 0;
+        sei.hIcon = NULL;
+        sei.hInstApp = NULL;
+        sei.hkeyClass = NULL;
+        sei.hMonitor = NULL;
+        sei.hProcess = NULL;
+        sei.hwnd = NULL;
+        sei.lpClass = NULL;
+        sei.lpDirectory = NULL;
+        sei.lpFile = finalurl.c_str();
+        sei.lpIDList = NULL;
+        sei.lpParameters = NULL;
+        sei.lpVerb = TEXT("open");
+        sei.nShow = SW_SHOWNORMAL;
 
-        // HINSTANCE_ERROR not defined on WinCE
-#ifndef __WXWINCE__
+        //Call ShellExecuteEx
+        ShellExecuteEx(&sei);
+
+        //Get error code
+        nResult = (int) sei.hInstApp;
+#endif
+
         // Hack for Firefox (returns file not found for some reason)
         // from Angelo Mandato's wxHyperlinksCtrl
-        // HINSTANCE_ERROR == 32
-        if (nResult <= HINSTANCE_ERROR && nResult != SE_ERR_FNF)
+        // HINSTANCE_ERROR == 32 (HINSTANCE_ERROR does not exist on Windows CE)
+        if (nResult <= 32 && nResult != SE_ERR_FNF)
             return false;
-#endif
 
 #ifdef __WXDEBUG__

 	  	 
