Index: src/mac/carbon/icon.cpp
===================================================================
--- src/mac/carbon/icon.cpp	(revision 49671)
+++ src/mac/carbon/icon.cpp	(working copy)
@@ -150,6 +150,54 @@
         {
             theId = kGenericDocumentIcon ;
         }
+        else if ( filename == wxT("wxICON_CDROM") )
+        {
+            theId = kGenericCDROMIcon ;
+        }
+        else if ( filename == wxT("wxICON_FLOPPY") )
+        {
+            theId = kGenericFloppyIcon ;
+        }
+        else if ( filename == wxT("wxICON_HARDDISK") )
+        {
+            theId = kGenericHardDiskIcon ;
+        }
+        else if ( filename == wxT("wxICON_REMOVABLE") )
+        {
+            theId = kGenericRemovableMediaIcon ;
+        }
+        else if ( filename == wxT("wxICON_DELETE") )
+        {
+            theId = kTrashIcon ;
+        }
+        else if ( filename == wxT("wxICON_GO_BACK") )
+        {
+            theId = kBackwardArrowIcon ;
+        }
+        else if ( filename == wxT("wxICON_GO_FORWARD") )
+        {
+            theId = kForwardArrowIcon ;
+        }
+        else if ( filename == wxT("wxICON_GO_HOME") )
+        {
+            theId = kUserFolderIcon ;
+        }
+        else if ( filename == wxT("wxICON_HELP_FOLDER") )
+        {
+            theId = kHelpFolderIcon ;
+        }
+        else if ( filename == wxT("wxICON_HELP_SETTINGS") )
+        {
+            theId = kGenericFontIcon ;
+        }
+        else if ( filename == wxT("wxICON_HELP_PAGE") )
+        {
+            theId = kGenericDocumentIcon ;
+        }
+        else if ( filename == wxT("wxICON_PRINT") )
+        {
+            theId = kPrintMonitorFolderIcon ;
+        }
         else
         {
 #if 0
Index: src/mac/carbon/artmac.cpp
===================================================================
--- src/mac/carbon/artmac.cpp	(revision 49671)
+++ src/mac/carbon/artmac.cpp	(working copy)
@@ -72,6 +72,23 @@
     ART_MSGBOX(wxART_FOLDER_OPEN, wxICON_FOLDER_OPEN, folder_open)
     ART_MSGBOX(wxART_NORMAL_FILE, wxICON_NORMAL_FILE, deffile)
 
+    ART_MSGBOX(wxART_CDROM,       wxICON_CDROM,       cdrom)
+    ART_MSGBOX(wxART_FLOPPY,      wxICON_FLOPPY,      floppy)
+    ART_MSGBOX(wxART_HARDDISK,    wxICON_HARDDISK,    harddisk)
+    ART_MSGBOX(wxART_REMOVABLE,   wxICON_REMOVABLE,   removable)
+
+    ART_MSGBOX(wxART_DELETE,      wxICON_DELETE,      delete)
+
+    ART_MSGBOX(wxART_GO_BACK,     wxICON_GO_BACK,     back)
+    ART_MSGBOX(wxART_GO_FORWARD,  wxICON_GO_FORWARD,  forward)
+    ART_MSGBOX(wxART_GO_HOME,     wxICON_GO_HOME,     home)
+
+    ART_MSGBOX(wxART_HELP_SETTINGS, wxICON_HELP_SETTINGS, htmoptns)
+    ART_MSGBOX(wxART_HELP_FOLDER, wxICON_HELP_FOLDER, htmfoldr)
+    ART_MSGBOX(wxART_HELP_PAGE,   wxICON_HELP_PAGE,   htmpage)
+
+    ART_MSGBOX(wxART_PRINT,       wxICON_PRINT,       print)
+
     return wxNullIconBundle;
 }
 

 	  	 
