From efc8efee90826efaad55b988237b8bab39a7e344 Mon Sep 17 00:00:00 2001
From: Carsten Fuchs <carsten.fuchs@cafu.de>
Date: Fri, 15 Aug 2014 09:39:46 +0200
Subject: [PATCH 7/8] In wxDataViewTreeNode::AlignTree(), (re-)sort only when
 necessary.

---
 src/generic/datavgen.cpp | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/src/generic/datavgen.cpp b/src/generic/datavgen.cpp
index 503ffcd..47732ee 100644
--- a/src/generic/datavgen.cpp
+++ b/src/generic/datavgen.cpp
@@ -421,17 +421,22 @@ public:
         model->GetChildren(m_item, itemChildren);
 
         wxDataViewTreeNodes& nodeChildren = m_children;
+        bool needResort = false;
 
         // If the item has fewer children than the node, remove node children accordingly.
         while (itemChildren.size() < nodeChildren.size())
         {
             delete nodeChildren[nodeChildren.size() - 1];
             nodeChildren.RemoveAt(nodeChildren.size() - 1);
+            // needResort = true;   // Not needed here, assuming they were sorted before.
         }
 
         // If the item has more children than the node, add node children accordingly.
         for (size_t i = nodeChildren.size(); i < itemChildren.size(); i++)
+        {
             nodeChildren.Add(new wxDataViewTreeNode(this, itemChildren[i]));
+            needResort = true;
+        }
 
         wxASSERT(itemChildren.size() == nodeChildren.size());
 
@@ -442,12 +447,13 @@ public:
             {
                 delete nodeChildren[i];
                 nodeChildren[i] = new wxDataViewTreeNode(this, itemChildren[i]);
+                needResort = true;
             }
 
             nodeChildren[i]->AlignTree(model);
         }
 
-        if (g_column >= -1)
+        if (needResort && g_column >= -1)
             nodeChildren.Sort(&wxGenericTreeModelNodeCmp);
     }
 
-- 
1.8.4.msysgit.0

