diff --git a/build/bakefiles/common.bkl b/build/bakefiles/common.bkl
index 240c344..995f78d 100644
--- a/build/bakefiles/common.bkl
+++ b/build/bakefiles/common.bkl
@@ -626,6 +643,8 @@ $(TAB)$(VC_COMPILER) /EP /nologo "$(DOLLAR)(InputPath)" > "$(SETUPHDIR)\wx\msw\r
             <precomp-headers-file>wxprec_$(id)</precomp-headers-file>
             <precomp-headers-exclude>
                 src/common/extended.c
+                src/gtk/eggtrayicon.c
+                src/gtk/treeentry_gtk.c
             </precomp-headers-exclude>
         </if>
 
diff --git a/src/gtk/assertdlg_gtk.cpp b/src/gtk/assertdlg_gtk.cpp
index 2a80149..1ec690b 100644
--- a/src/gtk/assertdlg_gtk.cpp
+++ b/src/gtk/assertdlg_gtk.cpp
@@ -7,6 +7,8 @@
 // Licence:     wxWindows licence
 /////////////////////////////////////////////////////////////////////////// */
 
+#include "wx/wxprec.h"
+
 #include "wx/platform.h"
 #include <gtk/gtk.h>
 #include "wx/gtk/assertdlg_gtk.h"
diff --git a/src/gtk/eggtrayicon.c b/src/gtk/eggtrayicon.c
index bd048ae..f23d9a9 100644
--- a/src/gtk/eggtrayicon.c
+++ b/src/gtk/eggtrayicon.c
@@ -66,21 +66,21 @@ References: <200312091142.54542.vaclav.slavik@matfyz.cz>
 Content-Type: text/plain
 Message-Id: <1070971316.30989.0.camel@carbon.csbnet.se>
 Mime-Version: 1.0
-X-Mailer: Ximian Evolution 1.5 
+X-Mailer: Ximian Evolution 1.5
 Date: Tue, 09 Dec 2003 13:01:56 +0100
 Content-Transfer-Encoding: 7bit
 X-Scanned-By: CLAM (openantivirus DB) antivirus scanner at mail.csbnet.se
 X-Virus-Scanned: by amavisd-new-20030616-p5 (Debian) at armitage
 X-Spam-Status: No, hits=-4.9 tagged_above=-999.0 required=6.3 tests=BAYES_00
-X-Spam-Level: 
-Status: R 
+X-Spam-Level:
+Status: R
 X-Status: N
-X-KMail-EncryptionState:  
-X-KMail-SignatureState:  
+X-KMail-EncryptionState:
+X-KMail-SignatureState:
 
 On tis, 2003-12-09 at 11:42 +0100, Vaclav Slavik wrote:
 > Hi,
-> 
+>
 > I'm working on the wxWindows cross-platform GUI toolkit
 > (http://www.wxwindows.org) which uses GTK+ and it would save me a lot
 > of time if I could use your eggtrayicon code to implement tray icons
@@ -104,7 +104,6 @@ Anders
 --------
 */
 
-
 #include "wx/platform.h"
 
 #if wxUSE_TASKBARICON
@@ -178,7 +177,7 @@ egg_tray_icon_get_type (void)
 	(GInstanceInitFunc) egg_tray_icon_init
       };
 
-      our_type = g_type_register_static (GTK_TYPE_PLUG, "EggTrayIcon", &our_info, 0);
+      our_type = g_type_register_static (GTK_TYPE_PLUG, "EggTrayIcon", &our_info, (GTypeFlags)0);
     }
 
   return our_type;
@@ -200,7 +199,7 @@ egg_tray_icon_class_init (EggTrayIconClass *klass)
   GtkWidgetClass *widget_class = (GtkWidgetClass *)klass;
   GtkContainerClass *container_class = (GtkContainerClass *)klass;
 
-  parent_class = g_type_class_peek_parent (klass);
+  parent_class = (GtkPlugClass *)g_type_class_peek_parent (klass);
 
   gobject_class->get_property = egg_tray_icon_get_property;
 
@@ -295,7 +294,7 @@ egg_tray_icon_get_orientation_property (EggTrayIcon *icon)
 static GdkFilterReturn
 egg_tray_icon_manager_filter (GdkXEvent *xevent, GdkEvent *event, gpointer user_data)
 {
-  EggTrayIcon *icon = user_data;
+  EggTrayIcon *icon = (EggTrayIcon *)user_data;
   XEvent *xev = (XEvent *)xevent;
 
   if (xev->xany.type == ClientMessage &&
@@ -538,13 +537,13 @@ egg_tray_icon_new_for_screen (GdkScreen *screen, const char *name)
 {
   g_return_val_if_fail (GDK_IS_SCREEN (screen), NULL);
 
-  return g_object_new (EGG_TYPE_TRAY_ICON, "screen", screen, "title", name, NULL);
+  return (EggTrayIcon*)g_object_new (EGG_TYPE_TRAY_ICON, "screen", screen, "title", name, NULL);
 }
 
 EggTrayIcon*
 egg_tray_icon_new (const gchar *name)
 {
-  return g_object_new (EGG_TYPE_TRAY_ICON, "title", name, NULL);
+  return (EggTrayIcon*)g_object_new (EGG_TYPE_TRAY_ICON, "title", name, NULL);
 }
 
 guint
diff --git a/src/gtk/treeentry_gtk.c b/src/gtk/treeentry_gtk.c
index e652f83..6dd61f4 100644
--- a/src/gtk/treeentry_gtk.c
+++ b/src/gtk/treeentry_gtk.c
@@ -10,7 +10,7 @@
 #ifdef __VMS
 #include <types.h>
 typedef pid_t GPid;
-#define G_GNUC_INTERNAL 
+#define G_GNUC_INTERNAL
 #define GSEAL(x) x
 #endif
 
@@ -95,8 +95,8 @@ gpointer   gtk_tree_entry_get_userdata  (GtkTreeEntry* entry)
 
 void     gtk_tree_entry_set_label       (GtkTreeEntry* entry, const gchar* label)
 {
-    g_assert(GTK_IS_TREE_ENTRY(entry));
     gchar *temp;
+    g_assert(GTK_IS_TREE_ENTRY(entry));
 
     /* free previous if it exists */
     if(entry->label)
diff --git a/src/gtk/win_gtk.cpp b/src/gtk/win_gtk.cpp
index 256e94f..d5c4a35 100644
--- a/src/gtk/win_gtk.cpp
+++ b/src/gtk/win_gtk.cpp
@@ -7,6 +7,8 @@
 // Licence:     wxWindows licence
 ///////////////////////////////////////////////////////////////////////////////
 
+#include "wx/wxprec.h"
+
 #include "wx/defs.h"
 #include "wx/gtk/private.h"
 #include "wx/gtk/private/win_gtk.h"
