--- wxWidgets-trunk/src/common/imagbmp.cpp	2014-03-13
+++ wxWidgets-work/src/common/imagbmp.cpp	2014-03-13
@@ -1019,15 +1019,30 @@
         if ( !stream.ReadAll(dbuf, 4) )
             return false;
 
-        offset = wxInvalidOffset; // not used in reading ICO/CUR DIBs
+        offset = wxInvalidOffset; // not used in loading ICO/CUR DIBs
         hdrSize = wxINT32_SWAP_ON_BE(dbuf[0]);
     }
 
+    int width;
+    int height;
+    if( hdrSize == 12 )
+    {
+        // we have BITMAPCOREHEADER
+         wxInt16 buf[2];
+        if ( !stream.ReadAll(buf, 2 * 2) )
+            return false;
+        
+        width = wxINT16_SWAP_ON_BE((short)buf[0]);
+        height = wxINT16_SWAP_ON_BE((short)buf[1]);
+    }
+    else
+    {
     if ( !stream.ReadAll(dbuf, 4 * 2) )
         return false;
 
-    int width = wxINT32_SWAP_ON_BE((int)dbuf[0]);
-    int height = wxINT32_SWAP_ON_BE((int)dbuf[1]);
+        width = wxINT32_SWAP_ON_BE((int)dbuf[0]);
+        height = wxINT32_SWAP_ON_BE((int)dbuf[1]);
+    }
     if ( !IsBmp)height = height  / 2; // for icons divide by 2
 
     if ( width > 32767 )
@@ -1067,10 +1082,22 @@
         return false;
     }
 
+    int comp;
+    int ncolors;
+    int resX, resY;
+    if ( hdrSize == 12 )
+    {
+        // BITMAPCOREHEADER
+        comp = BI_RGB;
+        ncolors = 0;
+        resX = resY = 0;
+    }
+    else
+    {
     if ( !stream.ReadAll(dbuf, 4 * 4) )
         return false;
 
-    int comp = wxINT32_SWAP_ON_BE((int)dbuf[0]);
+        comp = wxINT32_SWAP_ON_BE((int)dbuf[0]);
     if ( comp != BI_RGB && comp != BI_RLE4 && comp != BI_RLE8 &&
          comp != BI_BITFIELDS )
     {
@@ -1084,7 +1111,10 @@
     if ( !stream.ReadAll(dbuf, 4 * 2) )
         return false;
 
-    int ncolors = wxINT32_SWAP_ON_BE( (int)dbuf[0] );
+        ncolors = wxINT32_SWAP_ON_BE( (int)dbuf[0] );
+        resX = dbuf[2]/100;
+        resY = dbuf[3]/100;
+    }
     if (ncolors == 0)
         ncolors = 1 << bpp;
     /* some more sanity checks */
@@ -1101,7 +1131,7 @@
 
     //read DIB; this is the BMP image or the XOR part of an icon image
     if ( !DoLoadDib(image, width, height, bpp, ncolors, comp, offset, stream,
-                    verbose, IsBmp, true) )
+                    verbose, IsBmp, true, (hdrSize == 12)? 3: 4) )
     {
         if (verbose)
         {
@@ -1130,8 +1160,8 @@
 
     // the resolution in the bitmap header is in meters, convert to centimeters
     image->SetOption(wxIMAGE_OPTION_RESOLUTIONUNIT, wxIMAGE_RESOLUTION_CM);
-    image->SetOption(wxIMAGE_OPTION_RESOLUTIONX, dbuf[2]/100);
-    image->SetOption(wxIMAGE_OPTION_RESOLUTIONY, dbuf[3]/100);
+    image->SetOption(wxIMAGE_OPTION_RESOLUTIONX, resX);
+    image->SetOption(wxIMAGE_OPTION_RESOLUTIONY, resY);
 
     return true;
 }
