--- wxWidgets-trunk/src/common/imagbmp.cpp	2014-03-13
+++ wxWidgets-work/src/common/imagbmp.cpp	2014-03-13
@@ -997,22 +997,31 @@
     wxInt32         dbuf[4];
     wxInt8          bbuf[4];
 
+    // offset to bitmap data
+    wxFileOffset offset;
+    // DIB header size (used to distinguish different versions of DIB header)
+    wxInt32 hdrSize;
     if ( IsBmp )
     {
         // read the header off the .BMP format file
         if ( !stream.ReadAll(bbuf, 2) ||
              !stream.ReadAll(dbuf, 16) )
             return false;
+
+        #if 0 // unused
+            wxInt32 size = wxINT32_SWAP_ON_BE(dbuf[0]);
+        #endif
+        offset = wxINT32_SWAP_ON_BE(dbuf[2]);
+        hdrSize = wxINT32_SWAP_ON_BE(dbuf[3]);
     }
     else
     {
         if ( !stream.ReadAll(dbuf, 4) )
             return false;
+
+        offset = wxInvalidOffset; // not used in loading ICO/CUR DIBs
+        hdrSize = wxINT32_SWAP_ON_BE(dbuf[0]);
     }
-    #if 0 // unused
-        wxInt32 size = wxINT32_SWAP_ON_BE(dbuf[0]);
-    #endif
-    wxFileOffset offset = wxINT32_SWAP_ON_BE(dbuf[2]);
 
     if ( !stream.ReadAll(dbuf, 4 * 2) )
         return false;
