--- wxWidgets-trunk/include/wx/imagbmp.h	2013-12-29
+++ wxWidgets-work/include/wx/imagbmp.h	2014-03-13
@@ -60,7 +60,7 @@
                  bool IsBmp, bool IsMask);
     bool DoLoadDib(wxImage *image, int width, int height, int bpp, int ncolors,
                    int comp, wxFileOffset bmpOffset, wxInputStream& stream,
-                   bool verbose, bool IsBmp, bool hasPalette);
+                   bool verbose, bool IsBmp, bool hasPalette, int colEntrySize = 4);
     bool LoadDib(wxImage *image, wxInputStream& stream, bool verbose, bool IsBmp);
 #endif // wxUSE_STREAMS
 
--- wxWidgets-trunk/src/common/imagbmp.cpp	2014-03-13
+++ wxWidgets-work/src/common/imagbmp.cpp	2014-03-13
@@ -499,7 +499,7 @@
 bool wxBMPHandler::DoLoadDib(wxImage * image, int width, int height,
                              int bpp, int ncolors, int comp,
                              wxFileOffset bmpOffset, wxInputStream& stream,
-                             bool verbose, bool IsBmp, bool hasPalette)
+                             bool verbose, bool IsBmp, bool hasPalette, int colEntrySize)
 {
     wxInt32         aDword, rmask = 0, gmask = 0, bmask = 0, amask = 0;
     int             rshift = 0, gshift = 0, bshift = 0, ashift = 0;
@@ -585,7 +585,7 @@
         {
             if (hasPalette)
             {
-                if ( !stream.ReadAll(bbuf, 4) )
+                if ( !stream.ReadAll(bbuf, colEntrySize) )
                     return false;
 
                 cmap[j].b = bbuf[0];
