Index: include/wx/log.h
===================================================================
--- include/wx/log.h	(revision 64332)
+++ include/wx/log.h	(working copy)
@@ -1572,7 +1572,7 @@
 
 // macro which disables debug logging in release builds: this is done by
 // default by IMPLEMENT_APP() so usually it doesn't need to be used explicitly
-#ifdef NDEBUG
+#if defined(NDEBUG) && wxUSE_LOG_DEBUG
     #define wxDISABLE_DEBUG_LOGGING_IN_RELEASE_BUILD() \
         wxLog::SetLogLevel(wxLOG_Info)
 #else // !NDEBUG
