Index: tbtest.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/samples/taskbar/tbtest.cpp,v
retrieving revision 1.29
diff -u -r1.29 tbtest.cpp
--- tbtest.cpp  2004/10/16 02:15:48     1.29
+++ tbtest.cpp  2005/01/07 09:33:57
@@ -109,6 +109,7 @@
     PU_RESTORE = 10001,
     PU_NEW_ICON,
     PU_EXIT,
+       PU_CHECKMARK
 };


@@ -116,6 +117,8 @@
     EVT_MENU(PU_RESTORE, MyTaskBarIcon::OnMenuRestore)
     EVT_MENU(PU_EXIT,    MyTaskBarIcon::OnMenuExit)
     EVT_MENU(PU_NEW_ICON,MyTaskBarIcon::OnMenuSetNewIcon)
+       EVT_MENU(PU_CHECKMARK,MyTaskBarIcon::OnMenuCheckmark)
+       EVT_UPDATE_UI(PU_CHECKMARK,MyTaskBarIcon::OnMenuUICheckmark)
     EVT_TASKBAR_LEFT_DCLICK  (MyTaskBarIcon::OnLeftButtonDClick)
 END_EVENT_TABLE()

@@ -129,6 +132,17 @@
     dialog->Close(true);
 }

+static bool check = true;
+
+void MyTaskBarIcon::OnMenuCheckmark(wxCommandEvent& )
+{
+       check =!check;
+}
+void MyTaskBarIcon::OnMenuUICheckmark(wxUpdateUIEvent &event)
+{
+       event.Check( check );
+}
+
 void MyTaskBarIcon::OnMenuSetNewIcon(wxCommandEvent&)
 {
     wxIcon icon(smile_xpm);
@@ -144,6 +158,7 @@

     menu->Append(PU_RESTORE, _T("&Restore TBTest"));
     menu->Append(PU_NEW_ICON,_T("&Set New Icon"));
+    menu->Append(PU_CHECKMARK,  _T("Checkmark"),wxT( "" ), wxITEM_CHECK );
     menu->Append(PU_EXIT,    _T("E&xit"));

     return menu;
Index: tbtest.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/samples/taskbar/tbtest.h,v
retrieving revision 1.15
diff -u -r1.15 tbtest.h
--- tbtest.h    2004/10/16 02:15:48     1.15
+++ tbtest.h    2005/01/07 09:33:57
@@ -24,7 +24,8 @@
     void OnMenuRestore(wxCommandEvent&);
     void OnMenuExit(wxCommandEvent&);
     void OnMenuSetNewIcon(wxCommandEvent&);
-
+       void OnMenuCheckmark(wxCommandEvent&);
+       void OnMenuUICheckmark(wxUpdateUIEvent&);
     virtual wxMenu *CreatePopupMenu();

 DECLARE_EVENT_TABLE()

 	  	 
