Index: wxWidgets-trunk/src/osx/menuitem_osx.cpp
===================================================================
--- wxWidgets-trunk/src/osx/menuitem_osx.cpp	(revision 67269)
+++ wxWidgets-trunk/src/osx/menuitem_osx.cpp	(working copy)
@@ -41,13 +41,14 @@
 
     // In other languages there is no difference in naming the Exit/Quit menu item between MacOS and Windows guidelines
     // therefore these item must not be translated
-    if ( wxStripMenuCodes(m_text).Upper() == wxT("EXIT") )
-        m_text = wxT("Quit\tCtrl+Q") ;
+	if (pParentMenu != NULL && !pParentMenu->GetNoEventsMode())
+        if ( wxStripMenuCodes(m_text).Upper() == wxT("EXIT") )
+            m_text = wxT("Quit\tCtrl+Q") ;
 
     m_radioGroup.start = -1;
     m_isRadioGroupStart = false;
 
-    wxString text = wxStripMenuCodes(m_text);
+    wxString text = wxStripMenuCodes(m_text, (pParentMenu != NULL && pParentMenu->GetNoEventsMode()) ? wxStrip_Accel : wxStrip_All);
     if (text.IsEmpty() && !IsSeparator())
     {
         wxASSERT_MSG(wxIsStockID(GetId()), wxT("A non-stock menu item with an empty label?"));
@@ -202,7 +203,7 @@
     if ( !m_parentMenu )
         return ;
 
-    wxString text = wxStripMenuCodes(m_text);
+    wxString text = wxStripMenuCodes(m_text, m_parentMenu != NULL && m_parentMenu->GetNoEventsMode() ? wxStrip_Accel : wxStrip_All);
     if (text.IsEmpty() && !IsSeparator())
     {
         wxASSERT_MSG(wxIsStockID(GetId()), wxT("A non-stock menu item with an empty label?"));
