Index: src/mac/dc.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/mac/dc.cpp,v
retrieving revision 1.84
diff -u -r1.84 dc.cpp
--- dc.cpp	2003/09/25 23:10:35	1.84
+++ dc.cpp	2003/10/01 20:11:30
@@ -1679,7 +1679,7 @@
     if ( UMAGetSystemVersion() < 0x1000 || ((wxFont*)&m_font)->GetNoAntiAliasing() )
         useGetThemeText = false ;
 #endif
-    char text[] = "H" ;
+    char text[] = "g" ;
 #if TARGET_CARBON
     if ( useGetThemeText )
     {
Index: window.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/gtk/window.cpp,v
retrieving revision 1.441
diff -u -r1.441 window.cpp
--- window.cpp	2003/09/24 00:26:24	1.441
+++ window.cpp	2003/10/01 20:14:03
@@ -3374,7 +3374,7 @@
     PangoFontDescription *desc = m_font.GetNativeFontInfo()->description;
     PangoLayout *layout = pango_layout_new(context);
     pango_layout_set_font_description(layout, desc);
-    pango_layout_set_text(layout, "H", 1);
+    pango_layout_set_text(layout, "g", 1);
     PangoLayoutLine *line = (PangoLayoutLine *)pango_layout_get_lines(layout)->data;
 
     PangoRectangle rect;
@@ -3386,7 +3386,7 @@
 #else
     GdkFont *font = m_font.GetInternalFont( 1.0 );
 
-    return gdk_string_width( font, "H" );
+    return gdk_string_width( font, "g" );
 #endif
 }
 	  	 
