Index: src/msw/window.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/window.cpp,v
retrieving revision 1.727
diff -u -r1.727 window.cpp
--- src/msw/window.cpp	2006/10/21 15:37:55	1.727
+++ src/msw/window.cpp	2006/11/17 16:08:16
@@ -1322,6 +1322,9 @@
     // don't support overlapping windows and it only makes sense for them and,
     // presumably, gives the system some extra work (to manage more clipping
     // regions), so avoid it alltogether
+		if ( !wxSystemOptions::GetOptionInt(wxT("msw.window.no-clip-siblings"))
+			|| (flags & wxCLIP_SIBLINGS) )
+			style |= WS_CLIPSIBLINGS;
 
 
     if ( flags & wxVSCROLL )

 	  	 
