*** wxOGL.py	Thu May 29 17:36:30 2003
--- D:\dubner\python\energy\wxOGL.py	Thu Aug 21 01:25:31 2003
***************
*** 195,200 ****
--- 195,201 ----
  
          dc = wxClientDC(self)
          self.PrepareDC(dc)
+         self.arrow_mf = self.CreateArrow()
          for x in range(len(self.shapes)):
              fromShape = self.shapes[x]
              if x+1 == len(self.shapes):
***************
*** 205,211 ****
              line.SetCanvas(self)
              line.SetPen(wxBLACK_PEN)
              line.SetBrush(wxBLACK_BRUSH)
!             line.AddArrow(ARROW_ARROW)
              line.MakeLineControlPoints(2)
              fromShape.AddLine(line, toShape)
              self.diagram.AddShape(line)
--- 206,212 ----
              line.SetCanvas(self)
              line.SetPen(wxBLACK_PEN)
              line.SetBrush(wxBLACK_BRUSH)
!             line.AddArrow(ARROW_METAFILE, ARROW_POSITION_END, 10, 0, "", self.arrow_mf)
              line.MakeLineControlPoints(2)
              fromShape.AddLine(line, toShape)
              self.diagram.AddShape(line)
***************
*** 217,222 ****
--- 218,234 ----
          EVT_WINDOW_DESTROY(self, self.OnDestroy)
  
  
+     def CreateArrow(self):
+         mf = wxPseudoMetaFile()
+         mf.SetPen(wxRED_PEN)
+         h = w = 10
+         mf.DrawLine((-w,-h/2),(-w/2,0))
+         mf.DrawLine((-w/2,0),(-w,h/2))
+         mf.DrawLine((-w/2,-h/2),(0,0))
+         mf.DrawLine((0,0),(-w/2,h/2))
+         return mf
+ 
+ 
      def MyAddShape(self, shape, x, y, pen, brush, text):
          shape.SetDraggable(True, True)
          shape.SetCanvas(self)

 	  	 
