--- D:/Downloads/wxWidgets-3.0.0/src/msw/window.cpp	Mo Nov 11 14:18:38 2013
+++ D:/Dev/wxWidgets-3.0.0/src/msw/window.cpp	So Jan  5 20:58:39 2014
@@ -5607,8 +5607,10 @@
     // forwarded up to the parent by DefWindowProc()) and not in the client
     // ones as all the other messages, translate them to the client coords for
     // consistency
-    const wxPoint
-        pt = ScreenToClient(wxPoint(GET_X_LPARAM(lParam), GET_Y_LPARAM(lParam)));
+
+    POINT pt = {GET_X_LPARAM(lParam), GET_Y_LPARAM(lParam)};
+    ::ScreenToClient(GetHwnd(), &pt);
+
     wxMouseEvent event(wxEVT_MOUSEWHEEL);
     InitMouseEvent(event, pt.x, pt.y, LOWORD(wParam));
     event.m_wheelRotation = (short)HIWORD(wParam);
