Index: src/unix/displayx11.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/unix/displayx11.cpp,v
retrieving revision 1.2
diff -b -u -2 -r1.2 displayx11.cpp
--- src/unix/displayx11.cpp	2003/08/09 12:46:47	1.2
+++ src/unix/displayx11.cpp	2003/09/27 13:27:59
@@ -93,5 +93,14 @@
   else
   {
+    wxSize size = wxGetDisplaySize();
+    if (p.x >= 0 &&
+        p.x <= size.GetWidth() && 
+        p.y > 0 &&
+        p.y <= size.GetHeight())
+    {
     return 0;
+    }
+
+    return -1
   }
 }

 	  	 
