Index: samples/xrc/rc/objref.xrc
===================================================================
--- samples/xrc/rc/objref.xrc	(revision 74730)
+++ samples/xrc/rc/objref.xrc	(working copy)
@@ -6,27 +6,13 @@
         <object class="wxNotebook" name="objref_notebook">
           <object class="notebookpage">
             <object class="wxPanel" name="page1">
-              <object class="wxFlexGridSizer">
+              <style>wxBORDER_DOUBLE</style>
+              <object class="wxStaticBoxSizer" name="wxID_ANY">
                 <object class="sizeritem">
-                  <object class="wxTextCtrl" name="description_text">
-                    <value>This dialog demonstrates the use of object references and ID arrays.\n\nAs you turn the pages of the notebook, you will notice that each has an identical section at the bottom. Instead of writing that section's xml several times, this is done just once, as a wxPanel named 'bottom__panel'. The panel is then added to each page's sizer by the single line:   &lt;object__ref ref=&quot;bottom__panel&quot;/&gt;</value>
-                    <style>wxTE_MULTILINE|wxTE_READONLY</style>
+                  <object class="wxButton">
+                    <label>MyButton</label>
                   </object>
-                  <option>1</option>
-                  <flag>wxALL|wxEXPAND</flag>
-                  <border>5</border>
                 </object>
-                <object class="sizeritem">
-                  <object_ref ref="bottom_panel"/>
-                  <flag>wxEXPAND</flag>
-                </object>
-                <cols>1</cols>
-                <rows>0</rows>
-                <vgap>0</vgap>
-                <hgap>10</hgap>
-                <growablecols>0</growablecols>
-                <growablerows>0</growablerows>
-                <growablerows>1</growablerows>
               </object>
             </object>
             <label>Page 1</label>
