diff -rbu2 d:\wx21/include/wx/msw/textctrl.h d:\wx2/include/wx/msw/textctrl.h
--- d:\wx21/include/wx/msw/textctrl.h	Thu Jul 06 19:34:08 2000
+++ d:\wx2/include/wx/msw/textctrl.h	Sat Apr 06 14:11:46 2002
@@ -133,4 +133,9 @@
     virtual bool SetBackgroundColour(const wxColour& colour);
     virtual bool SetForegroundColour(const wxColour& colour);
+    bool SetSelectionForegroundColour(const wxColour& colour);
+    bool SetSelectionBold();
+    bool SetSelectionItalic();
+    bool SetSelectionStrikeout();
+    bool SetSelectionUnderline();
 #endif // wxUSE_RICHEDIT
 
diff -rbu2 d:\wx21/src/msw/textctrl.cpp d:\wx2/src/msw/textctrl.cpp
--- d:\wx21/src/msw/textctrl.cpp	Tue Nov 21 16:00:38 2000
+++ d:\wx2/src/msw/textctrl.cpp	Sat Apr 06 14:11:21 2002
@@ -1192,4 +1192,98 @@
 }
 
+bool wxTextCtrl::SetSelectionForegroundColour(const wxColour& colour)
+{
+//    if ( !wxTextCtrlBase::SetForegroundColour(colour) )
+//    {
+//        // colour didn't really change
+//        return FALSE;
+//    }
+
+    if ( IsRich() )
+    {
+        // change the colour of current selection
+        CHARFORMAT cf;
+        wxZeroMemory(cf);
+        cf.cbSize = sizeof(cf);
+        cf.dwMask = CFM_COLOR;
+        cf.crTextColor = wxColourToRGB(colour);
+        ::SendMessage(GetHwnd(), EM_SETCHARFORMAT, SCF_SELECTION, (LPARAM)&cf);
+        return TRUE;
+    }
+    return FALSE;
+}
+
+bool wxTextCtrl::SetSelectionBold()
+{
+
+    if ( IsRich() )
+    {
+        // change the format of current selection
+        CHARFORMAT cf;
+        wxZeroMemory(cf);
+        cf.cbSize = sizeof(cf);
+        cf.dwMask = CFM_BOLD;
+        cf.dwEffects = CFE_BOLD;
+        ::SendMessage(GetHwnd(), EM_SETCHARFORMAT, SCF_SELECTION, (LPARAM)&cf);
+        return TRUE;
+    }
+    return FALSE;
+}
+
+bool wxTextCtrl::SetSelectionItalic()
+{
+
+    if ( IsRich() )
+    {
+        // change the format of current selection
+        CHARFORMAT cf;
+        wxZeroMemory(cf);
+        cf.cbSize = sizeof(cf);
+        cf.dwMask = CFM_ITALIC;
+        cf.dwEffects = CFE_ITALIC;
+        ::SendMessage(GetHwnd(), EM_SETCHARFORMAT, SCF_SELECTION, (LPARAM)&cf);
+        return TRUE;
+    }
+    return FALSE;
+}
+
+bool wxTextCtrl::SetSelectionStrikeout()
+{
+
+    if ( IsRich() )
+    {
+        // change the format of current selection
+        CHARFORMAT cf;
+        wxZeroMemory(cf);
+        cf.cbSize = sizeof(cf);
+        cf.dwMask = CFM_STRIKEOUT;
+        cf.dwEffects = CFE_STRIKEOUT;
+        ::SendMessage(GetHwnd(), EM_SETCHARFORMAT, SCF_SELECTION, (LPARAM)&cf);
+        return TRUE;
+    }
+    return FALSE;
+}
+
+bool wxTextCtrl::SetSelectionUnderline()
+{
+
+    if ( IsRich() )
+    {
+        // change the format of current selection
+        CHARFORMAT cf;
+        wxZeroMemory(cf);
+        cf.cbSize = sizeof(cf);
+        cf.dwMask = CFM_UNDERLINE;
+        cf.dwEffects = CFE_UNDERLINE;
+        ::SendMessage(GetHwnd(), EM_SETCHARFORMAT, SCF_SELECTION, (LPARAM)&cf);
+        return TRUE;
+    }
+    return FALSE;
+    
+}
+
+
+
+
 #endif // wxUSE_RICHEDIT
 

diff -rbu2N d:\wx21\samples/richedit2/makefile.b32 d:\wx2\samples/richedit2/makefile.b32
--- d:\wx21\samples/richedit2/makefile.b32	Thu Jan 01 01:00:00 1970
+++ d:\wx2\samples/richedit2/makefile.b32	Sat Apr 06 15:39:43 2002
@@ -0,0 +1,16 @@
+#
+# File:		makefile.b32
+# Author:	Julian Smart
+# Created:	1999
+# Updated:	
+# Copyright:
+#
+# Makefile : Builds sample for 32-bit BC++
+
+WXDIR = $(WXWIN)
+
+TARGET=richdemo
+OBJECTS = $(TARGET).obj
+
+!include $(WXDIR)\src\makeprog.b32
+
diff -rbu2N d:\wx21\samples/richedit2/richdemo.cpp d:\wx2\samples/richedit2/richdemo.cpp
--- d:\wx21\samples/richedit2/richdemo.cpp	Thu Jan 01 01:00:00 1970
+++ d:\wx2\samples/richedit2/richdemo.cpp	Sat Apr 06 15:40:52 2002
@@ -0,0 +1,226 @@
+///////////////////////////////////////////////////////////////////////////////
+// Name:        richdemo.cpp
+// Purpose:     demo of extended wxTextCtrl 
+// Author:      Marton Papp
+// Modified by:
+// Created:     29.03.2002
+// Copyright:   (c) 2002 Marton Papp <anteusz@freemail.hu>
+// Licence:     wxWindows license
+///////////////////////////////////////////////////////////////////////////////
+
+// ============================================================================
+// declarations
+// ============================================================================
+
+// ----------------------------------------------------------------------------
+// headers
+// ----------------------------------------------------------------------------
+#include "wx/wxprec.h"
+
+#ifndef  WX_PRECOMP
+  #include "wx/wx.h"
+#endif //precompiled headers
+
+
+#include "wx/colour.h"
+
+// ----------------------------------------------------------------------------
+// classes
+// ----------------------------------------------------------------------------
+class MyApp: public wxApp
+{
+public:
+  // implement base class virtuals
+  virtual bool OnInit();
+  virtual int OnExit();
+};
+
+class MyFrame: public wxFrame
+{
+public:
+  MyFrame();
+
+  // callbacks
+  void OnQuit(wxCommandEvent& event);
+  void OnAbout(wxCommandEvent& event);
+  void OnBold(wxCommandEvent& event);
+  void OnUnderline(wxCommandEvent& event);
+  void OnStrikeout(wxCommandEvent& event);
+  void OnItalic(wxCommandEvent& event);
+  void OnGreen(wxCommandEvent& event);
+  void OnYellow(wxCommandEvent& event);
+  void OnBlue(wxCommandEvent& event);
+  void OnCloseWindow(wxCloseEvent& event);
+
+private:
+  wxTextCtrl *m_text;
+
+  DECLARE_EVENT_TABLE()
+};
+
+enum
+{
+  Minimal_Quit,
+  Minimal_About,
+  Minimal_Bold,
+Minimal_Underline,
+Minimal_Strikeout,
+Minimal_Italic,
+Minimal_Green,
+Minimal_Yellow,
+Minimal_Blue
+};
+
+// ----------------------------------------------------------------------------
+// event tables
+// ----------------------------------------------------------------------------
+BEGIN_EVENT_TABLE(MyFrame, wxFrame)
+  EVT_MENU(Minimal_Quit, MyFrame::OnQuit)
+  EVT_MENU(Minimal_About, MyFrame::OnAbout)
+  EVT_MENU(Minimal_Bold,MyFrame::OnBold)
+  EVT_MENU(Minimal_Underline,MyFrame::OnUnderline)
+  EVT_MENU(Minimal_Strikeout,MyFrame::OnStrikeout)
+  EVT_MENU(Minimal_Italic,MyFrame::OnItalic) 
+  EVT_MENU(Minimal_Green,MyFrame::OnGreen)
+  EVT_MENU(Minimal_Yellow,MyFrame::OnYellow)
+  EVT_MENU(Minimal_Blue,MyFrame::OnBlue)
+
+  EVT_CLOSE(MyFrame::OnCloseWindow)
+END_EVENT_TABLE()
+  
+// ============================================================================
+// implementation
+// ============================================================================
+  
+// ----------------------------------------------------------------------------
+// application
+// ----------------------------------------------------------------------------
+IMPLEMENT_APP(MyApp)
+
+// `Main program' equivalent, creating windows and returning main app frame
+bool MyApp::OnInit()
+{
+  
+  SetVendorName("wxWindows");
+  SetAppName("richtext"); // not needed, it's the default value
+
+  
+  // create the main program window
+  MyFrame *frame = new MyFrame;
+  frame->Show(TRUE);
+  SetTopWindow(frame);
+
+  
+  return TRUE;
+}
+
+int MyApp::OnExit()
+{
+  
+
+  return 0;
+}
+
+// ----------------------------------------------------------------------------
+// frame
+// ----------------------------------------------------------------------------
+
+// main frame ctor
+MyFrame::MyFrame()
+       : wxFrame((wxFrame *) NULL, -1, "wxTextCtrl Richtext Demo",wxDefaultPosition,wxSize(350, 300))
+{
+ 
+
+  // menu
+  wxMenu *file_menu = new wxMenu;
+
+  file_menu->Append(Minimal_Bold, "Bold", "Make selection bold");
+  file_menu->Append(Minimal_Underline, "Underline", "Make selection underline");
+  file_menu->Append(Minimal_Strikeout, "Strikeout", "Make selection strikeout");
+  file_menu->Append(Minimal_Italic, "Italic", "Make selection italic");
+  file_menu->AppendSeparator();
+  file_menu->Append(Minimal_Green, "Green", "Make selection green");
+  file_menu->Append(Minimal_Yellow, "Yellow", "Make selection yellow");
+  file_menu->Append(Minimal_Blue, "Blue", "Make selection blue");
+  file_menu->AppendSeparator();
+  file_menu->Append(Minimal_About, "&About", "About this sample");
+  file_menu->AppendSeparator();
+  file_menu->Append(Minimal_Quit, "E&xit", "Exit the program");
+  wxMenuBar *menu_bar = new wxMenuBar;
+  menu_bar->Append(file_menu, "&Styles");
+  SetMenuBar(menu_bar);
+
+  //CreateStatusBar();
+
+  // child controls
+  wxPanel *panel = new wxPanel(this);
+ 
+  m_text = new wxTextCtrl(panel, -1, "", wxPoint(0, 0), wxSize(300, 200),wxTE_RICH|wxTE_MULTILINE);
+  
+
+ 
+
+  m_text->SetValue("Write some text, make selections and set the style from the menu!\n");
+  m_text->SetSelection(0,15);
+  m_text->SetSelectionBold();
+  m_text->SetSelectionForegroundColour(wxColour(128, 0, 0));
+  m_text->SetSelection(17,20);
+  m_text->SetSelectionUnderline();
+ m_text->SetSelection(25,30);
+  m_text->SetSelectionStrikeout();
+  m_text->SetSelection(30,35);
+  m_text->SetSelectionItalic();
+}
+
+void MyFrame::OnCloseWindow(wxCloseEvent& event)
+{
+    this->Destroy();
+}
+
+void MyFrame::OnQuit(wxCommandEvent&)
+{
+  Close(TRUE);
+}
+
+void MyFrame::OnAbout(wxCommandEvent&)
+{
+  wxMessageBox(_T("wxTextCtrl richtext demo\n© Marton Papp 2002"), _T("About"),
+               wxICON_INFORMATION | wxOK);
+}
+
+
+void MyFrame::OnBold(wxCommandEvent&)
+{
+   m_text->SetSelectionBold(); 
+}
+
+void MyFrame::OnUnderline(wxCommandEvent&)
+{
+   m_text->SetSelectionUnderline(); 
+}
+
+void MyFrame::OnStrikeout(wxCommandEvent&)
+{
+   m_text->SetSelectionStrikeout(); 
+}
+
+void MyFrame::OnItalic(wxCommandEvent&)
+{
+   m_text->SetSelectionItalic(); 
+}
+
+void MyFrame::OnGreen(wxCommandEvent&)
+{
+   m_text->SetSelectionForegroundColour(wxColour("GREEN")); 
+}
+
+void MyFrame::OnBlue(wxCommandEvent&)
+{
+   m_text->SetSelectionForegroundColour(wxColour("BLUE")); 
+}
+void MyFrame::OnYellow(wxCommandEvent&)
+{
+   m_text->SetSelectionForegroundColour(wxColour("YELLOW")); 
+}
+
+
diff -rbu2N d:\wx21\samples/richedit2/richdemo.rc d:\wx2\samples/richedit2/richdemo.rc
--- d:\wx21\samples/richedit2/richdemo.rc	Thu Jan 01 01:00:00 1970
+++ d:\wx2\samples/richedit2/richdemo.rc	Sat Aug 22 18:07:12 1998
@@ -0,0 +1,2 @@
+#include "wx/msw/wx.rc"
+

 	  	 
