Index: samples/combo/combo.cpp
===================================================================
--- samples/combo/combo.cpp	(revision 152)
+++ samples/combo/combo.cpp	(working copy)
@@ -703,7 +703,7 @@
 
     odc->SetSelection(0);
 
-    rowSizer->Add( odc, 1, wxALIGN_CENTER_VERTICAL|wxALL, 4 );
+    rowSizer->Add( odc, 1, wxALIGN_CENTER_VERTICAL|wxALL, 5 );
     rowSizer->AddStretchSpacer(1);
     colSizer->Add( rowSizer, 0, wxEXPAND|wxALL, 5 );
 
@@ -742,10 +742,14 @@
                            2 // horizontal spacing
                           );
 
-    rowSizer->Add( odc, 1, wxALIGN_CENTER_VERTICAL|wxALL, 4 );
+    rowSizer->Add( odc, 1, wxALIGN_CENTER_VERTICAL|wxALL, 5 );
     rowSizer->AddStretchSpacer(1);
     colSizer->Add( rowSizer, 0, wxEXPAND|wxALL, 5 );
 
+    wxTextCtrl* tc = new wxTextCtrl( panel, wxID_ANY, "wxTextCtrl left margin reference" );
+    rowSizer = new wxBoxSizer( wxHORIZONTAL );
+    rowSizer->Add(tc, 1, wxALIGN_CENTER_VERTICAL|wxALL, 5 );
+    colSizer->Add( rowSizer, 0, wxEXPAND|wxALL, 5 );
 
     //
     // List View wxComboCtrl
@@ -828,6 +832,11 @@
     rowSizer->Add( gcc, 1, wxALIGN_CENTER_VERTICAL|wxALL, 5 );
 
     colSizer->Add( rowSizer, 0, wxEXPAND|wxALL, 5 );
+    
+    wxComboBox* cb = new wxComboBox( panel, wxID_ANY, "wxComboBox left margin reference" );
+    rowSizer = new wxBoxSizer( wxHORIZONTAL );
+    rowSizer->Add(cb, 1, wxALIGN_CENTER_VERTICAL|wxALL, 5 );
+    colSizer->Add( rowSizer, 0, wxEXPAND|wxALL, 5 );
 
 #if wxUSE_IMAGE
     wxInitAllImageHandlers();
