Index: include/wx/object.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWidgets/include/wx/object.h,v
retrieving revision 1.122
diff -u -r1.122 object.h
--- include/wx/object.h	2006/02/09 00:48:33	1.122
+++ include/wx/object.h	2006/03/06 11:30:32
@@ -394,7 +394,7 @@
 
     int GetRefCount() const { return m_count; }
 
-private:
+protected:
     int m_count;
 };
 

 	  	 
