Index: docs/latex/wx/stattext.tex
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/docs/latex/wx/stattext.tex,v
retrieving revision 1.10
diff -u -r1.10 stattext.tex
--- docs/latex/wx/stattext.tex	2003/06/22 11:04:13	1.10
+++ docs/latex/wx/stattext.tex	2004/10/13 02:48:10
@@ -26,6 +26,14 @@
 given, the control will not change its size (this style is especially useful
 with controls which also have wxALIGN\_RIGHT or CENTER style because otherwise
 they won't make sense any longer after a call to SetLabel)}
+\twocolitem{\windowstyle{wxST\_DOTS\_MIDDLE}}{If \windowstyle{wxST\_NO\_AUTORESIZE}
+is given, the control will not change its size, and will insert an ellipsis in
+the middle of the label if the string provided to SetLabel() is too large for
+the given size of the control. Availiable on wxMac(Carbon)}
+\twocolitem{\windowstyle{wxST\_DOTS\_END}}{If \windowstyle{wxST\_NO\_AUTORESIZE}
+is given, the control will not change its size, and will insert an ellipsis at
+the end of the label if the string provided to SetLabel() is too large for
+the given size of the control. Availiable on wxMac(Carbon)}
 \end{twocollist}
 
 See also \helpref{window styles overview}{windowstyles}.
Index: include/wx/defs.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/defs.h,v
retrieving revision 1.451
diff -u -r1.451 defs.h
--- include/wx/defs.h	2004/10/12 16:58:44	1.451
+++ include/wx/defs.h	2004/10/13 02:48:13
@@ -1451,6 +1451,8 @@
  * wxStaticText flags
  */
 #define wxST_NO_AUTORESIZE    0x0001
+#define wxST_DOTS_MIDDLE      0x0002
+#define wxST_DOTS_END         0x0004
 
 /*
  * wxStaticBitmap flags
Index: src/mac/carbon/stattext.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/mac/carbon/stattext.cpp,v
retrieving revision 1.52
diff -u -r1.52 stattext.cpp
--- src/mac/carbon/stattext.cpp	2004/08/04 18:20:38	1.52
+++ src/mac/carbon/stattext.cpp	2004/10/13 02:48:20
@@ -98,6 +98,40 @@
         InvalidateBestSize();
         SetSize( GetBestSize() ) ;
     }
+	
+	if (  ( GetWindowStyle() & wxST_DOTS_END ) || ( GetWindowStyle() & wxST_DOTS_MIDDLE )  )    //coded added by Ryan Wilcox October 12, 2004
+    {
+        wxSize targetSize = GetSize();  //trust that the user has set an appropriate size using SetSize() (or the constructor)
+        OSStatus theErr;
+        Boolean didTrunc;
+        CFMutableStringRef inString = CFStringCreateMutableCopy(NULL , 0 , ref) ;
+        TruncCode tcode;
+        wxMacCFStringHolder* cfToWxString = NULL;
+
+        wxString stringFromCF(wxEmptyString);
+        if ( GetWindowStyle() & wxST_DOTS_MIDDLE )
+            tcode = truncMiddle;
+        else
+            tcode = truncEnd;
+
+        theErr = TruncateThemeText( inString , kThemeCurrentPortFont ,
+            kThemeStateActive , targetSize.GetWidth() , tcode, &didTrunc );
+        
+        wxASSERT_MSG( theErr == noErr , _T("Could not truncate the text!") );
+        
+        cfToWxString = new wxMacCFStringHolder( inString , false );
+        
+        stringFromCF = cfToWxString->AsString();
+        verify_noerr( m_peer->SetData<CFStringRef>(kControlEntireControl , kControlStaticTextCFStringTag, inString ) ) ;
+        m_label = wxStripMenuCodes(stringFromCF);
+        
+        CFRelease( inString );
+        delete cfToWxString;
+		
+        InvalidateBestSize();
+        SetSize( GetBestSize() );   //set the size to the size of the truncated text we created
+    }
+	
     Refresh() ;
     Update() ;
 }

 	  	 
