? src/gtk/window.cpp.cvs_broken
Index: docs/latex/wx/thread.tex
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/docs/latex/wx/thread.tex,v
retrieving revision 1.30
diff -u -r1.30 thread.tex
--- docs/latex/wx/thread.tex	2004/06/30 23:21:49	1.30
+++ docs/latex/wx/thread.tex	2004/07/01 13:56:01
@@ -4,7 +4,7 @@
 sometimes called {\it light-weight processes}, but the fundamental difference
 between threads and processes is that memory spaces of different processes are
 separated while all threads share the same address space. While it makes it
-much easier to share common data between several threads, it also makes much
+much easier to share common data between several threads, it also makes it much
 easier to shoot oneself in the foot, so careful use of synchronization objects
 such as \helpref{mutexes}{wxmutex} and/or \helpref{critical sections}{wxcriticalsection} is recommended.
 
Index: docs/latex/wx/tsamples.tex
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/docs/latex/wx/tsamples.tex,v
retrieving revision 1.40
diff -u -r1.40 tsamples.tex
--- docs/latex/wx/tsamples.tex	2004/06/24 11:59:38	1.40
+++ docs/latex/wx/tsamples.tex	2004/07/01 13:56:01
@@ -41,8 +41,8 @@
 \end{twocollist}
 
 Finally, it might be helpful to do a search in the entire sample directory if
-you can't find the sample you showing the control you are interested in by
-name. Most of wxWidgets classes, occur in at least one of the samples.
+you can't find the sample showing the control you are interested in by
+name. Most classes contained in wxWidgets occur in at least one of the samples.
 
 
 \subsection{Minimal sample}\label{sampleminimal}
@@ -73,7 +73,7 @@
 This sample demonstrates the use of the \helpref{wxCheckListBox}{wxchecklistbox}
 class intercepting check, select and double click events. It also tests the
 use of various methods modifying the control, such as by deleting items
-from it or inserting new once (these functions are actually implemented in
+from it or inserting new ones (these functions are actually implemented in
 the parent class \helpref{wxListBox}{wxlistbox} so the sample tests that class
 as well). The layout of the dialog is created using a \helpref{wxBoxSizer}{wxboxsizer}
 demonstrating a simple dynamic layout.
@@ -94,7 +94,7 @@
 The controls sample is the main test program for most simple controls used in
 wxWidgets. The sample tests their basic functionality, events, placement,
 modification in terms of colour and font as well as the possibility to change
-the controls programmatically, such as adding item to a list box etc. Apart
+the controls programmatically, such as adding an item to a list box etc. Apart
 from that, the sample uses a \helpref{wxNotebook}{wxnotebook} and tests most
 features of this special control (using bitmap in the tabs, using 
 \helpref{wxSizers}{wxsizer} and \helpref{constraints}{wxlayoutconstraints} within
@@ -132,16 +132,16 @@
 \subsection{Dialogs sample}\label{sampledialogs}
 
 This sample shows how to use the common dialogs available from wxWidgets. These
-dialogs are described in details in the \helpref{Common dialogs overview}{commondialogsoverview}.
+dialogs are described in detail in the \helpref{Common dialogs overview}{commondialogsoverview}.
 
 
 \subsection{Dialup sample}\label{sampledialup}
 
-This sample shows \helpref{wxDialUpManager}{wxdialupmanager} 
+This sample shows the \helpref{wxDialUpManager}{wxdialupmanager} 
 class. It displays in the status bar the information gathered through its
 interface: in particular, the current connection status (online or offline) and
 whether the connection is permanent (in which case a string `LAN' appears in
-the third status bar field - but note that you may have be on a LAN not
+the third status bar field - but note that you may be on a LAN not
 connected to the Internet, in which case you will not see this) or not.
 
 Using the menu entries, you may also dial or hang up the line if you have a
@@ -168,7 +168,7 @@
 Finally, the last part is used for dragging text from it to either one of the
 listboxes (only one will accept it) or another application. The last
 functionality available from the main frame is to paste a bitmap from the
-clipboard (or, in the case of Windows version, also a metafile) - it will be
+clipboard (or, in the case of the Windows version, also a metafile) - it will be
 shown in a new frame.
 
 So far, everything we mentioned was implemented with minimal amount of code
@@ -192,11 +192,11 @@
 
 This sample is a very small sample that demonstrates the use of the
 \helpref{wxEvtHandler::Connect}{wxevthandlerconnect} method. This method
-should be used whenever it is not known at compile time, which control
+should be used whenever it is not known at compile time which control
 will receive which event or which controls are actually going to be in
 a dialog or frame. This is most typically the case for any scripting
 language that would work as a wrapper for wxWidgets or programs where
-forms or similar datagrams can be created by the uses.
+forms or similar datagrams can be created by the users.
 
 See also the \helpref{event sample}{sampleevent}
 
@@ -220,7 +220,7 @@
 understand what goes on.
 
 You need to build the library with \texttt{wxUSE\_EXCEPTIONS} being set to $1$
-and compile your code with the C++ exceptions support to be able to build this
+and compile your code with C++ exceptions support to be able to build this
 sample.
 
 
@@ -254,20 +254,20 @@
 \subsection{HTML samples}\label{samplehtml}
 
 Eight HTML samples (you can find them in directory {\tt samples/html})
-cover all features of HTML sub-library.
+cover all features of the HTML sub-library.
 
 {\bf Test} demonstrates how to create \helpref{wxHtmlWindow}{wxhtmlwindow} 
-and also shows most of supported HTML tags.
+and also shows most supported HTML tags.
 
-{\bf Widget} shows how you can embed ordinary controls or windows within
+{\bf Widget} shows how you can embed ordinary controls or windows within an
 HTML page. It also nicely explains how to write new tag handlers and extend
 the library to work with unsupported tags.
 
-{\bf About} may give you an idea how to write good-looking about boxes.
+{\bf About} may give you an idea how to write good-looking About boxes.
 
 {\bf Zip} demonstrates use of virtual file systems in wxHTML. The zip archives
 handler (ships with wxWidgets) allows you to access HTML pages stored 
-in compressed archive as if they were ordinary files.
+in a compressed archive as if they were ordinary files.
 
 {\bf Virtual} is yet another virtual file systems demo. This one generates pages at run-time.
 You may find it useful if you need to display some reports in your application.
@@ -279,7 +279,7 @@
 {\bf Help} and {\bf Helpview} are variations on displaying HTML help 
 (compatible with MS HTML Help Workshop). {\it Help} shows how to embed 
 \helpref{wxHtmlHelpController}{wxhtmlhelpcontroller} in your application
-while {\it Helpview} is simple tool that only pops up help window and
+while {\it Helpview} is a simple tool that only pops up the help window and
 displays help books given at command line.
 
 
@@ -287,13 +287,13 @@
 
 The image sample demonstrates the use of the \helpref{wxImage}{wximage} class
 and shows how to download images in a variety of formats, currently PNG, GIF,
-TIFF, JPEG, BMP, PNM and PCX. The top of the sample shows to rectangles, one
+TIFF, JPEG, BMP, PNM and PCX. The top of the sample shows two rectangles, one
 of which is drawn directly in the window, the other one is drawn into a 
 \helpref{wxBitmap}{wxbitmap}, converted to a wxImage, saved as a PNG image
 and then reloaded from the PNG file again so that conversions between wxImage
-and wxBitmap as well as loading and save PNG files are tested.
+and wxBitmap as well as loading and saving PNG files are tested.
 
-At the bottom of the main frame is a test for using a monochrome bitmap by
+At the bottom of the main frame there is a test for using a monochrome bitmap by
 drawing into a \helpref{wxMemoryDC}{wxmemorydc}. The bitmap is then drawn
 specifying the foreground and background colours with 
 \helpref{wxDC::SetTextForeground}{wxdcsettextforeground} and 
@@ -305,7 +305,7 @@
 \subsection{Internat(ionalization) sample}\label{sampleinternat}
 
 The not very clearly named internat sample demonstrates the wxWidgets
-internatationalization (i18n for short from now on) features. To be more
+internationalization (i18n for short from now on) features. To be more
 precise, it only shows localization support, i.e. support for translating the
 program messages in another language while true i18n would also involve
 changing the other aspects of the programs behaviour.
@@ -333,7 +333,7 @@
 
 \subsection{Listctrl sample}\label{samplelistctrl}
 
-This sample shows \helpref{wxListCtrl}{wxlistctrl} control. Different modes
+This sample shows the \helpref{wxListCtrl}{wxlistctrl} control. Different modes
 supported by the control (list, icons, small icons, report) may be chosen from
 the menu.
 
@@ -392,7 +392,7 @@
 
 The server binds to the local address, using TCP port number 3000,
 sets up an event handler to be notified of incoming connection requests
-({\bf wxSOCKET\_CONNECTION} events), and stands there, waiting for clients
+({\bf wxSOCKET\_CONNECTION} events), and sits there, waiting for clients
 ({\it listening} in the socket parlance). For each accepted connection,
 a new \helpref{wxSocketBase}{wxsocketbase} object is created. These
 socket objects are independent from the server that created them, so
@@ -422,7 +422,7 @@
 to catch {\bf wxSOCKET\_LOST} events, each one is immediately notified
 if the other end closes the connection.
 
-There is also an URL test which shows how to use
+There is also a URL test which shows how to use
 the \helpref{wxURL}{wxurl} class to fetch data from a given URL.
 
 The sockets sample is work in progress. Some things to do:
@@ -514,13 +514,13 @@
 \end{itemize}
 
 Some buttons in the main toolbar are check buttons, i.e. they stay checked when
-pressed. On the platforms which support it, the sample also add a combobox
+pressed. On the platforms which support it, the sample also adds a combobox
 to the toolbar showing how you can use arbitrary controls and not only buttons
 in it.
 
 If you toggle another toolbar in the sample (using {\tt Ctrl-A}) you will also
 see the radio toolbar buttons in action: the first three buttons form a radio
-group, that is checking any of them automatically unchecks the previously
+group, i.e. checking any of them automatically unchecks the previously
 checked one.
 
 
@@ -552,7 +552,7 @@
 continuing).
 \item Using more elaborated techniques to allow returning to the previous
 page, but not continuing to the next one or vice versa (in wxRadioboxPage)
-\item This (wxRadioboxPage) page also shows how the page may process {\tt
+\item This (wxRadioboxPage) page also shows how the page may process the {\tt
 Cancel} button itself instead of relying on the wizard parent to do it.
 \item Normally, the order of the pages in the wizard is known at compile-time,
 but sometimes it depends on the user choices: wxCheckboxPage shows how to
Index: docs/latex/wx/window.tex
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/docs/latex/wx/window.tex,v
retrieving revision 1.154
diff -u -r1.154 window.tex
--- docs/latex/wx/window.tex	2004/07/01 11:44:22	1.154
+++ docs/latex/wx/window.tex	2004/07/01 13:56:02
@@ -1,6 +1,6 @@
 \section{\class{wxWindow}}\label{wxwindow}
 
-wxWindow is the base class for all windows and represents any visible objecto n
+wxWindow is the base class for all windows and represents any visible object on
 screen. All controls, top level windows and so on are windows. Sizers and
 device contexts are not, however, as they don't appear on screen themselves.
 
Index: include/wx/app.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/app.h,v
retrieving revision 1.119
diff -u -r1.119 app.h
--- include/wx/app.h	2004/06/23 14:49:42	1.119
+++ include/wx/app.h	2004/07/01 13:56:02
@@ -92,7 +92,7 @@
     // class OnInit() to do it.
     virtual bool OnInit();
 
-    // this is here only temproary hopefully (FIXME)
+    // this is here only temporary hopefully (FIXME)
     virtual bool OnInitGui() { return true; }
 
     // This is the replacement for the normal main(): all program work should
Index: include/wx/event.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/event.h,v
retrieving revision 1.205
diff -u -r1.205 event.h
--- include/wx/event.h	2004/07/01 00:04:42	1.205
+++ include/wx/event.h	2004/07/01 13:56:03
@@ -358,7 +358,7 @@
     // for them wouldn't work (it needs to do a copy of the event)
     virtual wxEvent *Clone() const = 0;
 
-    // Implementation only: this test is explicitlty anti OO and this functions
+    // Implementation only: this test is explicitly anti OO and this function
     // exists only for optimization purposes.
     bool IsCommandEvent() const { return m_isCommandEvent; }
 
Index: include/wx/file.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/file.h,v
retrieving revision 1.35
diff -u -r1.35 file.h
--- include/wx/file.h	2004/06/12 23:43:43	1.35
+++ include/wx/file.h	2004/07/01 13:56:03
@@ -66,7 +66,7 @@
   // ----------------
     // check whether a regular file by this name exists
   static bool Exists(const wxChar *name);
-    // check whetther we can access the given file in given mode
+    // check whether we can access the given file in given mode
     // (only read and write make sense here)
   static bool Access(const wxChar *name, OpenMode mode);
 
Index: include/wx/thread.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/thread.h,v
retrieving revision 1.69
diff -u -r1.69 thread.h
--- include/wx/thread.h	2004/06/08 14:48:10	1.69
+++ include/wx/thread.h	2004/07/01 13:56:03
@@ -281,7 +281,7 @@
 #undef wxCRITSECT_IS_MUTEX
 
 // wxCriticalSectionLocker is the same to critical sections as wxMutexLocker is
-// to th mutexes
+// to mutexes
 class WXDLLIMPEXP_BASE wxCriticalSectionLocker
 {
 public:
@@ -405,7 +405,7 @@
 };
 
 // ----------------------------------------------------------------------------
-// wxThread: class encpasulating a thread of execution
+// wxThread: class encapsulating a thread of execution
 // ----------------------------------------------------------------------------
 
 // there are two different kinds of threads: joinable and detached (default)
Index: include/wx/window.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/window.h,v
retrieving revision 1.155
diff -u -r1.155 window.h
--- include/wx/window.h	2004/07/01 11:44:29	1.155
+++ include/wx/window.h	2004/07/01 13:56:04
@@ -541,7 +541,7 @@
         // is this window a top level one?
     virtual bool IsTopLevel() const;
 
-        // it doesn't really change parent, use ReParent() instead
+        // it doesn't really change parent, use Reparent() instead
     void SetParent( wxWindowBase *parent ) { m_parent = (wxWindow *)parent; }
         // change the real parent of this window, return true if the parent
         // was changed, false otherwise (error or newParent == oldParent)
Index: locale/de.po
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/locale/de.po,v
retrieving revision 1.35
diff -u -r1.35 de.po
--- locale/de.po	2004/06/10 22:17:50	1.35
+++ locale/de.po	2004/07/01 13:56:05
@@ -404,7 +404,7 @@
 
 #: ../src/html/helpfrm.cpp:358
 msgid "Add current page to bookmarks"
-msgstr "Aktuelle HTLM-Seite den Lesezeichen hinzufügen"
+msgstr "Aktuelle HTML-Seite den Lesezeichen hinzufügen"
 
 #: ../src/generic/colrdlgg.cpp:287
 msgid "Add to custom colours"
@@ -467,23 +467,23 @@
 
 #: ../src/common/paper.cpp:146
 msgid "B4 Envelope, 250 x 353 mm"
-msgstr "B4 Umschlag, 250 x 353 mm"
+msgstr "B4-Umschlag, 250 x 353 mm"
 
 #: ../src/common/paper.cpp:128
 msgid "B4 sheet, 250 x 354 mm"
-msgstr "B4 Blatt, 250 x 354 mm"
+msgstr "B4-Blatt, 250 x 354 mm"
 
 #: ../src/common/paper.cpp:147
 msgid "B5 Envelope, 176 x 250 mm"
-msgstr "B5 Umschlag, 176 x 250 mm"
+msgstr "B5-Umschlag, 176 x 250 mm"
 
 #: ../src/common/paper.cpp:129
 msgid "B5 sheet, 182 x 257 millimeter"
-msgstr "B5 Blatt, 182 x 257 mm"
+msgstr "B5-Blatt, 182 x 257 mm"
 
 #: ../src/common/paper.cpp:148
 msgid "B6 Envelope, 176 x 125 mm"
-msgstr "B6 Umschlag, 176 x 125 mm"
+msgstr "B6-Umschlag, 176 x 125 mm"
 
 #: ../src/common/imagbmp.cpp:476
 #: ../src/common/imagbmp.cpp:492
@@ -544,23 +544,23 @@
 
 #: ../src/common/paper.cpp:142
 msgid "C3 Envelope, 324 x 458 mm"
-msgstr "C3 Umschlag, 324 x 458 mm"
+msgstr "C3-Umschlag, 324 x 458 mm"
 
 #: ../src/common/paper.cpp:143
 msgid "C4 Envelope, 229 x 324 mm"
-msgstr "C4 Umschlag, 229 x 324 mm"
+msgstr "C4-Umschlag, 229 x 324 mm"
 
 #: ../src/common/paper.cpp:141
 msgid "C5 Envelope, 162 x 229 mm"
-msgstr "C5 Umschlag, 162 x 229 mm"
+msgstr "C5-Umschlag, 162 x 229 mm"
 
 #: ../src/common/paper.cpp:144
 msgid "C6 Envelope, 114 x 162 mm"
-msgstr "C6 Umschlag, 114 x 162 mm"
+msgstr "C6-Umschlag, 114 x 162 mm"
 
 #: ../src/common/paper.cpp:145
 msgid "C65 Envelope, 114 x 229 mm"
-msgstr "C65 Umschlag, 114 x 229 mm"
+msgstr "C65-Umschlag, 114 x 229 mm"
 
 #: ../src/html/chm.cpp:783
 #: ../src/html/chm.cpp:842
@@ -692,7 +692,7 @@
 #: ../src/common/image.cpp:1048
 #, c-format
 msgid "Can't load image from file '%s': file does not exist."
-msgstr "Kann Bild aus Datei '%s' nicht laden : Datei ist nicht vorhanden."
+msgstr "Kann Bild aus Datei '%s' nicht laden: Datei ist nicht vorhanden."
 
 #: ../src/msw/registry.cpp:385
 #, c-format
@@ -812,7 +812,7 @@
 
 #: ../src/mgl/app.cpp:292
 msgid "Cannot initialize SciTech MGL!"
-msgstr "Kann SciTech MGL nicht initialisieren !"
+msgstr "Kann SciTech MGL nicht initialisieren!"
 
 #: ../src/mgl/window.cpp:546
 msgid "Cannot initialize display."
@@ -826,7 +826,7 @@
 #: ../contrib/src/xrc/xmlres.cpp:382
 #, c-format
 msgid "Cannot load resources from file '%s'."
-msgstr "Kann die Resourcen nicht aus der Datei '%s'  laden."
+msgstr "Kann die Resourcen nicht aus der Datei '%s' laden."
 
 #: ../src/html/htmlfilt.cpp:164
 #, c-format
@@ -851,7 +851,7 @@
 #: ../contrib/src/xrc/xmlres.cpp:353
 #, c-format
 msgid "Cannot open file '%s'."
-msgstr "Konnte Datei \"%s\" nicht öffnen"
+msgstr "Konnte Datei \"%s\" nicht öffnen."
 
 #: ../src/generic/dcpsg.cpp:1798
 msgid "Cannot open file for PostScript printing!"
@@ -906,11 +906,11 @@
 
 #: ../src/common/fmapbase.cpp:104
 msgid "Central European (ISO-8859-2)"
-msgstr "Zentral Europäisch (ISO-8859-2)"
+msgstr "Zentral-Europäisch (ISO-8859-2)"
 
 #: ../src/msw/dialup.cpp:750
 msgid "Choose ISP to dial"
-msgstr "Wähle ISP um anzurufen"
+msgstr "Wähle anzurufenden ISP"
 
 #: ../src/gtk/fontdlg.cpp:127
 msgid "Choose font"
Index: src/common/file.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/common/file.cpp,v
retrieving revision 1.80
diff -u -r1.80 file.cpp
--- src/common/file.cpp	2004/05/23 20:52:00	1.80
+++ src/common/file.cpp	2004/07/01 13:56:06
@@ -414,10 +414,10 @@
     wxCHECK( (pBuf != NULL) && IsOpened(), 0 );
 
 #ifdef __WXWINCE__
-    DWORD bytesRead = 0;
+    DWORD bytesWritten = 0;
     int iRc = 0;
-    if (WriteFile((HANDLE) m_fd, pBuf, (DWORD) nCount, & bytesRead, NULL))
-        iRc = bytesRead;
+    if (WriteFile((HANDLE) m_fd, pBuf, (DWORD) nCount, & bytesWritten, NULL))
+        iRc = bytesWritten;
     else
         iRc = -1;
 #elif defined(__MWERKS__)
@@ -554,6 +554,7 @@
     wxASSERT( IsOpened() );
 
 #ifdef __WXWINCE__
+    /* huh? Why not simply use GetFileSize()? */
     DWORD off0 = SetFilePointer((HANDLE) m_fd, 0, 0, FILE_CURRENT);
     DWORD off1 = SetFilePointer((HANDLE) m_fd, 0, 0, FILE_END);
     off_t len = off1;
Index: src/msw/button.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/msw/button.cpp,v
retrieving revision 1.84
diff -u -r1.84 button.cpp
--- src/msw/button.cpp	2004/06/24 14:25:47	1.84
+++ src/msw/button.cpp	2004/07/01 13:56:07
@@ -418,7 +418,7 @@
     {
         // NOTE: Apparently older versions (NT 4?) of the common controls send
         //       BN_DOUBLECLICKED but not a second BN_CLICKED for owner-drawn
-        //       buttons, so in order to send two EVET_BUTTON events we should
+        //       buttons, so in order to send two EVT_BUTTON events we should
         //       catch both types.  Currently (Feb 2003) up-to-date versions of
         //       win98, win2k and winXP all send two BN_CLICKED messages for
         //       all button types, so we don't catch BN_DOUBLECLICKED anymore
@@ -437,7 +437,7 @@
 
 WXLRESULT wxButton::MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam)
 {
-    // when we receive focus, we want to temporary become the default button in
+    // when we receive focus, we want to temporarily become the default button in
     // our parent panel so that pressing "Enter" would activate us -- and when
     // losing it we should restore the previous default button as well
     if ( nMsg == WM_SETFOCUS )
Index: src/msw/glcanvas.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/msw/glcanvas.cpp,v
retrieving revision 1.27
diff -u -r1.27 glcanvas.cpp
--- src/msw/glcanvas.cpp	2004/07/01 10:46:38	1.27
+++ src/msw/glcanvas.cpp	2004/07/01 13:56:07
@@ -77,7 +77,7 @@
   m_hDC = win->GetHDC();
 
   m_glContext = wglCreateContext((HDC) m_hDC);
-  wxCHECK_RET( m_glContext, wxT("Couldn't create OpenGl context") );
+  wxCHECK_RET( m_glContext, wxT("Couldn't create OpenGL context") );
 
   wglMakeCurrent((HDC) m_hDC, m_glContext);
 }
@@ -93,7 +93,7 @@
   m_hDC = win->GetHDC();
 
   m_glContext = wglCreateContext((HDC) m_hDC);
-  wxCHECK_RET( m_glContext, wxT("Couldn't create OpenGl context") );
+  wxCHECK_RET( m_glContext, wxT("Couldn't create OpenGL context") );
 
   if( other != 0 )
     wglShareLists( other->m_glContext, m_glContext );
Index: src/msw/pen.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/msw/pen.cpp,v
retrieving revision 1.29
diff -u -r1.29 pen.cpp
--- src/msw/pen.cpp	2004/05/28 11:08:45	1.29
+++ src/msw/pen.cpp	2004/07/01 13:56:07
@@ -219,7 +219,7 @@
                default:
                    logb.lbStyle = BS_SOLID;
 #ifdef __WXDEBUG__
-                   // this should be unnecessary (it's unused) but suppresses the Purigy
+                   // this should be unnecessary (it's unused) but suppresses the Purify
                    // messages about uninitialized memory read
                    logb.lbHatch = 0;
 #endif

 	  	 
