Index: samples/dataview/dataview.cpp
===================================================================
--- samples/dataview/dataview.cpp	(révision 76283)
+++ samples/dataview/dataview.cpp	(copie de travail)
@@ -862,8 +862,8 @@
 {
     wxDataViewItem item( event.GetItem() );
 
-    // only allow drags for item, not containers
-    if (m_music_model->IsContainer( item ) )
+    // only allow drags for item or background, not containers
+    if ( item.IsOk() && m_music_model->IsContainer( item ) )
         event.Veto();
 
     if (event.GetDataFormat() != wxDF_UNICODETEXT)
@@ -875,7 +875,7 @@
     wxDataViewItem item( event.GetItem() );
 
     // only allow drops for item, not containers
-    if (m_music_model->IsContainer( item ) )
+    if ( item.IsOk() && m_music_model->IsContainer( item ) )
     {
         event.Veto();
         return;
@@ -890,7 +890,10 @@
     wxTextDataObject obj;
     obj.SetData( wxDF_UNICODETEXT, event.GetDataSize(), event.GetDataBuffer() );
 
-    wxLogMessage( "Text dropped: %s", obj.GetText() );
+    if ( item.IsOk() )
+	wxLogMessage( "Text dropped on item %s: %s", m_music_model->GetTitle( item ), obj.GetText() );
+    else
+	wxLogMessage( "Text dropped on background: %s", obj.GetText() );
 }
 
 #endif // wxUSE_DRAG_AND_DROP
