Index: src/gtk/dataview.cpp
===================================================================
--- src/gtk/dataview.cpp	(révision 76283)
+++ src/gtk/dataview.cpp	(copie de travail)
@@ -3562,8 +3562,6 @@
                                            GtkSelectionData *selection_data)
 {
     wxDataViewItem item(GetOwner()->GTKPathToItem(path));
-    if ( !item )
-        return FALSE;
 
     wxDataViewEvent event( wxEVT_DATAVIEW_ITEM_DROP, m_owner->GetId() );
     event.SetEventObject( m_owner );
@@ -3587,8 +3585,6 @@
                                           GtkSelectionData *selection_data)
 {
     wxDataViewItem item(GetOwner()->GTKPathToItem(path));
-    if ( !item )
-        return FALSE;
 
     wxDataViewEvent event( wxEVT_DATAVIEW_ITEM_DROP_POSSIBLE, m_owner->GetId() );
     event.SetEventObject( m_owner );
