diff -bru wxWidgets.orig/src/osx/cocoa/textctrl.mm wxWidgets.patched/src/osx/cocoa/textctrl.mm
--- wxWidgets.orig/src/osx/cocoa/textctrl.mm	2015-09-06 01:59:19.000000000 -0400
+++ wxWidgets.patched/src/osx/cocoa/textctrl.mm	2015-09-13 17:19:39.000000000 -0400
@@ -355,6 +355,22 @@
         impl->controlTextDidChange();
 }
 
+- (void)textViewDidChangeSelection:(NSNotification *)aNotification
+{
+    wxUnusedVar( aNotification );
+    wxWidgetCocoaImpl *impl = (wxWidgetCocoaImpl *) wxWidgetImpl::FindFromWXWidget( self );
+    if( impl )
+    {
+        wxWindow* wxpeer = (wxWindow*) impl->GetWXPeer();
+        if ( wxpeer && wxpeer->HasFlag( wxTE_GENERATE_SEL_EVENT ) )
+        {
+            wxCommandEvent event( wxEVT_TEXT_SEL_CARET, wxpeer->GetId() );
+            event.SetEventObject( wxpeer );
+            wxpeer->HandleWindowEvent( event );
+        }
+    }
+}
+
 - (void) setEnabled:(BOOL) flag
 {
     // from Technical Q&A QA1461
diff -bru wxWidgets.orig/src/osx/textctrl_osx.cpp wxWidgets.patched/src/osx/textctrl_osx.cpp
--- wxWidgets.orig/src/osx/textctrl_osx.cpp	2015-09-06 01:59:19.000000000 -0400
+++ wxWidgets.patched/src/osx/textctrl_osx.cpp	2015-09-13 14:05:34.000000000 -0400
@@ -94,6 +94,7 @@
     const wxValidator& validator,
     const wxString& name )
 {
+    wxASSERT_MSG( !( HasFlag( wxTE_GENERATE_SEL_EVENT ) && !( HasFlag( wxTE_RICH2 ) ) ), "generating selection event requires rich edit control" );
     DontCreatePeer();
     m_editable = true ;
 
