Index: docs/latex/wx/listctrl.tex
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/docs/latex/wx/listctrl.tex,v
retrieving revision 1.69
diff -u -r1.69 listctrl.tex
--- docs/latex/wx/listctrl.tex	2006/01/25 23:29:02	1.69
+++ docs/latex/wx/listctrl.tex	2006/02/02 10:10:09
@@ -920,13 +920,22 @@
 \func{bool}{SetItemImage}{\param{long }{item}, \param{int }{image}}
 
 Sets the image associated with the item. The image is an index into the
-image list associated with the list control.
+image list associated with the list control. In report view, this only sets
+the image for the first column.
 
 \func{bool}{SetItemImage}{\param{long }{item}, \param{int }{image}, \param{int }{selImage}}
 
 Sets the unselected and selected images associated with the item. The images are indices into the
 image list associated with the list control. This form is deprecated: {\it selImage} is not
 used.
+
+
+\membersection{wxListCtrl::SetItemColumnImage}\label{wxlistctrlsetitemcolumnimage}
+
+\func{bool}{SetItemImage}{\param{long }{item}, \param{long }{column}\param{int }{image}}
+
+Sets the image associated with the item. In report view, you can specify the column.
+The image is an index into the image list associated with the list control.
 
 
 \membersection{wxListCtrl::SetItemPosition}\label{wxlistctrlsetitemposition}
Index: include/wx/mac/carbon/listctrl.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/mac/carbon/listctrl.h,v
retrieving revision 1.6
diff -u -r1.6 listctrl.h
--- include/wx/mac/carbon/listctrl.h	2005/09/23 12:49:34	1.6
+++ include/wx/mac/carbon/listctrl.h	2006/02/02 09:57:24
@@ -223,6 +223,7 @@
 
   // Sets the item image
   bool SetItemImage(long item, int image, int selImage) ;
+  bool SetItemColumnImage(long item, long column, int image);
 
   // Gets the item text
   wxString GetItemText(long item) const ;
Index: include/wx/mac/classic/listctrl.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/mac/classic/listctrl.h,v
retrieving revision 1.5
diff -u -r1.5 listctrl.h
--- include/wx/mac/classic/listctrl.h	2005/09/23 12:49:43	1.5
+++ include/wx/mac/classic/listctrl.h	2006/02/02 09:57:24
@@ -223,6 +223,7 @@
 
   // Sets the item image
   bool SetItemImage(long item, int image, int selImage) ;
+  bool SetItemColumnImage(long item, long column, int image);
 
   // Gets the item text
   wxString GetItemText(long item) const ;
Index: include/wx/msw/listctrl.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/msw/listctrl.h,v
retrieving revision 1.59
diff -u -r1.59 listctrl.h
--- include/wx/msw/listctrl.h	2006/01/25 23:28:57	1.59
+++ include/wx/msw/listctrl.h	2006/02/02 09:57:23
@@ -157,6 +157,7 @@
 
     // Sets the item image
     bool SetItemImage(long item, int image, int selImage = -1) ;
+    bool SetItemColumnImage(long item, long column, int image);
 
     // Gets the item text
     wxString GetItemText(long item) const ;
Index: include/wx/os2/listctrl.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/os2/listctrl.h,v
retrieving revision 1.10
diff -u -r1.10 listctrl.h
--- include/wx/os2/listctrl.h	2006/01/25 23:28:54	1.10
+++ include/wx/os2/listctrl.h	2006/02/02 10:06:42
@@ -133,6 +133,10 @@
                       ,int  nImage
                       ,int  lSelImage
                      );
+    bool SetItemColumnImage( long lItem
+                            ,long lColumn
+                            ,int  nImage
+                           );
 
     //
     // Item text
Index: include/wx/palmos/listctrl.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/palmos/listctrl.h,v
retrieving revision 1.7
diff -u -r1.7 listctrl.h
--- include/wx/palmos/listctrl.h	2006/01/25 23:28:51	1.7
+++ include/wx/palmos/listctrl.h	2006/02/02 10:00:54
@@ -154,6 +154,7 @@
 
     // Sets the item image
     bool SetItemImage(long item, int image, int selImage) ;
+    bool SetItemColumnImage(long item, long column, int image);
 
     // Gets the item text
     wxString GetItemText(long item) const ;
Index: samples/listctrl/listtest.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/samples/listctrl/listtest.cpp,v
retrieving revision 1.84
diff -u -r1.84 listtest.cpp
--- samples/listctrl/listtest.cpp	2005/09/23 12:51:56	1.84
+++ samples/listctrl/listtest.cpp	2006/02/02 13:43:31
@@ -492,6 +492,15 @@
     m_listCtrl->SetColumnWidth( 1, wxLIST_AUTOSIZE );
     m_listCtrl->SetColumnWidth( 2, wxLIST_AUTOSIZE );
 
+    // Set images in columns
+    m_listCtrl->SetItemColumnImage(1, 1, 0);
+
+    wxListItem info;
+    info.SetImage(0);
+    info.SetId(3);
+    info.SetColumn(2);
+    m_listCtrl->SetItem(info);
+
     // test SetItemFont too
     m_listCtrl->SetItemFont(0, *wxITALIC_FONT);
 }
@@ -1013,9 +1022,15 @@
     }
 }
 
-int MyListCtrl::OnGetItemImage(long WXUNUSED(item)) const
+int MyListCtrl::OnGetItemColumnImage(long item, long column) const
 {
-    return 0;
+    if (!column)
+        return 0;
+
+    if (!(item %3) && column == 1)
+        return 0;
+
+    return -1;
 }
 
 wxListItemAttr *MyListCtrl::OnGetItemAttr(long item) const
Index: samples/listctrl/listtest.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/samples/listctrl/listtest.h,v
retrieving revision 1.42
diff -u -r1.42 listtest.h
--- samples/listctrl/listtest.h	2005/05/22 21:31:42	1.42
+++ samples/listctrl/listtest.h	2006/02/02 13:39:46
@@ -68,7 +68,7 @@
     void LogColEvent(const wxListEvent& event, const wxChar *eventName);
 
     virtual wxString OnGetItemText(long item, long column) const;
-    virtual int OnGetItemImage(long item) const;
+    virtual int OnGetItemColumnImage(long item, long column) const;
     virtual wxListItemAttr *OnGetItemAttr(long item) const;
 
     wxListItemAttr m_attr;
Index: src/generic/listctrl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/generic/listctrl.cpp,v
retrieving revision 1.387
diff -u -r1.387 listctrl.cpp
--- src/generic/listctrl.cpp	2006/01/31 02:37:23	1.387
+++ src/generic/listctrl.cpp	2006/02/02 10:06:39
@@ -4905,10 +4905,17 @@
 bool
 wxGenericListCtrl::SetItemImage( long item, int image, int WXUNUSED(selImage) )
 {
+    return SetItemColumnImage(item, 0, image);
+}
+
+bool
+wxGenericListCtrl::SetItemColumnImage( long item, long column, int image )
+{
     wxListItem info;
     info.m_image = image;
     info.m_mask = wxLIST_MASK_IMAGE;
     info.m_itemId = item;
+    info.m_col = column;
     m_mainWin->SetItem( info );
     return true;
 }
Index: src/msw/listctrl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/msw/listctrl.cpp,v
retrieving revision 1.243
diff -u -r1.243 listctrl.cpp
--- src/msw/listctrl.cpp	2006/01/25 23:28:45	1.243
+++ src/msw/listctrl.cpp	2006/02/02 10:06:40
@@ -917,11 +917,18 @@
 // Sets the item image
 bool wxListCtrl::SetItemImage(long item, int image, int WXUNUSED(selImage))
 {
+    return SetItemColumnImage(item, 0, image);
+}
+
+// Sets the item image
+bool wxListCtrl::SetItemColumnImage(long item, long column, int image)
+{
     wxListItem info;
 
     info.m_mask = wxLIST_MASK_IMAGE;
     info.m_image = image;
     info.m_itemId = item;
+    info.m_col = column;
 
     return SetItem(info);
 }
Index: src/os2/listctrl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/os2/listctrl.cpp,v
retrieving revision 1.15
diff -u -r1.15 listctrl.cpp
--- src/os2/listctrl.cpp	2006/01/25 23:28:43	1.15
+++ src/os2/listctrl.cpp	2006/02/02 10:06:41
@@ -1544,13 +1544,23 @@
 , int                               nImage
 , int                               WXUNUSED(nSelImage))
 {
+    return SetItemColumnInfo(lItem, 0, nImage);
+} // end of wxListCtrl::SetItemImage
+
+// Sets the item image
+bool wxListCtrl::SetItemColumnImage (
+  long                              lItem
+, long                              lColumn
+, int                               nImage
+{
     wxListItem                      vInfo;
 
     vInfo.m_mask   = wxLIST_MASK_IMAGE;
     vInfo.m_image  = nImage;
     vInfo.m_itemId = lItem;
+    vInfo.m_col    = lColumn;
     return SetItem(vInfo);
-} // end of wxListCtrl::SetItemImage
+} // end of wxListCtrl::SetItemColumnImage
 
 // Gets the item text
 wxString wxListCtrl::GetItemText (
Index: src/palmos/listctrl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/palmos/listctrl.cpp,v
retrieving revision 1.7
diff -u -r1.7 listctrl.cpp
--- src/palmos/listctrl.cpp	2006/01/25 23:28:40	1.7
+++ src/palmos/listctrl.cpp	2006/02/02 10:06:41
@@ -284,6 +284,12 @@
     return false;
 }
 
+// Sets the item image
+bool wxListCtrl::SetItemColumnImage(long item, long column, int image)
+{
+    return false;
+}
+
 // Gets the item text
 wxString wxListCtrl::GetItemText(long item) const
 {
Index: include/wx/generic/listctrl.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/generic/listctrl.h,v
retrieving revision 1.103
diff -u -r1.103 listctrl.h
--- include/wx/generic/listctrl.h	2006/01/25 23:28:59	1.103
+++ include/wx/generic/listctrl.h	2006/02/05 17:34:01
@@ -99,6 +99,7 @@
     int  GetItemState( long item, long stateMask ) const;
     bool SetItemState( long item, long state, long stateMask);
     bool SetItemImage( long item, int image, int selImage = -1 );
+    bool SetItemColumnImage( long item, long column, int image );
     wxString GetItemText( long item ) const;
     void SetItemText( long item, const wxString& str );
     wxUIntPtr GetItemData( long item ) const;

 	  	 
