diff --git a/samples/propgrid/propgrid.cpp b/samples/propgrid/propgrid.cpp
index 5112f5a7e3..57f7a54474 100644
--- a/samples/propgrid/propgrid.cpp
+++ b/samples/propgrid/propgrid.cpp
@@ -1473,14 +1473,7 @@ void FormMain::PopulateWithExamples ()
     pg->Append( new wxDateProperty(wxT("DateProperty"), wxPG_LABEL, wxDateTime::Now() ) );
 
 #if wxUSE_DATEPICKCTRL
-    pg->SetPropertyAttribute( wxT("DateProperty"), wxPG_DATE_PICKER_STYLE,
-                              (long)(wxDP_DROPDOWN |
-                                     wxDP_SHOWCENTURY |
-                                     wxDP_ALLOWNONE) );
-
-    pg->SetPropertyHelpString( wxT("DateProperty"),
-        wxT("Attribute wxPG_DATE_PICKER_STYLE has been set to (long)")
-        wxT("(wxDP_DROPDOWN | wxDP_SHOWCENTURY | wxDP_ALLOWNONE).") );
+    pg->SetPropertyAttribute( wxT( "DateProperty" ), wxPG_DATE_PICKER_STYLE, (long)wxDP_SPIN );
 #endif
 
 #endif
