Index: src/generic/sashwin.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/generic/sashwin.cpp,v
retrieving revision 1.48
diff -u -r1.48 sashwin.cpp
--- src/generic/sashwin.cpp	2005/09/26 00:29:38	1.48
+++ src/generic/sashwin.cpp	2005/12/28 22:41:48
@@ -33,6 +33,7 @@
 #include "wx/dcscreen.h"
 #include "wx/sashwin.h"
 #include "wx/laywin.h"
+#include "wx/dcclient.h"
 
 DEFINE_EVENT_TYPE(wxEVT_SASH_DRAGGED)
 
Index: src/msw/button.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/msw/button.cpp,v
retrieving revision 1.94
diff -u -r1.94 button.cpp
--- src/msw/button.cpp	2005/09/23 12:54:52	1.94
+++ src/msw/button.cpp	2005/12/28 22:19:36
@@ -34,6 +34,7 @@
     #include "wx/bmpbuttn.h"
     #include "wx/settings.h"
     #include "wx/dcscreen.h"
+    #include "wx/dcclient.h"
 #endif
 
 #include "wx/stockitem.h"

 	  	 
