Index: src/aui/auibar.cpp
===================================================================
--- src/aui/auibar.cpp	(revision 76959)
+++ src/aui/auibar.cpp	(working copy)
@@ -233,7 +233,7 @@
                                     const wxRect& rect)
 {
     dc.SetFont(m_font);
-    dc.SetTextForeground(*wxBLACK);
+    dc.SetTextForeground(wxSystemSettings::GetColour(wxSYS_COLOUR_WINDOWTEXT));
 
     // we only care about the text height here since the text
     // will get cropped based on the width of the item
@@ -343,9 +343,10 @@
         dc.DrawBitmap(bmp, bmpX, bmpY, true);
 
     // set the item's text color based on if it is disabled
-    dc.SetTextForeground(*wxBLACK);
     if (item.GetState() & wxAUI_BUTTON_STATE_DISABLED)
         dc.SetTextForeground(DISABLED_TEXT_COLOR);
+    else
+        dc.SetTextForeground(wxSystemSettings::GetColour(wxSYS_COLOUR_WINDOWTEXT));
 
     if ( (m_flags & wxAUI_TB_TEXT) && !item.GetLabel().empty() )
     {
@@ -470,10 +471,14 @@
     dc.DrawBitmap(dropbmp, dropBmpX, dropBmpY, true);
 
     // set the item's text color based on if it is disabled
-    dc.SetTextForeground(*wxBLACK);
-    if (item.GetState() & wxAUI_BUTTON_STATE_DISABLED)
+    if (item.GetState() & wxAUI_BUTTON_STATE_DISABLED) {
         dc.SetTextForeground(DISABLED_TEXT_COLOR);
+    }
+    else {
+        dc.SetTextForeground(wxSystemSettings::GetColour(wxSYS_COLOUR_WINDOWTEXT));
+    }
 
+
     if ( (m_flags & wxAUI_TB_TEXT) && !item.GetLabel().empty() )
     {
         dc.DrawText(item.GetLabel(), textX, textY);
@@ -511,7 +516,7 @@
         return;
 
     // set the label's text color
-    dc.SetTextForeground(*wxBLACK);
+    dc.SetTextForeground(wxSystemSettings::GetColour(wxSYS_COLOUR_WINDOWTEXT));
 
     textX = rect.x + (rect.width/2) - (textWidth/2) + 1;
     textY = rect.y + rect.height - textHeight - 1;
