Index: build/bakefiles/files.bkl
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/build/bakefiles/files.bkl,v
retrieving revision 1.326
diff -bu2 -r1.326 build/bakefiles/files.bkl
--- build/bakefiles/files.bkl
+++ build/bakefiles/files.bkl
@@ -574,4 +574,5 @@
     src/common/fldlgcmn.cpp
     src/common/fontcmn.cpp
+    src/common/fontenumcmn.cpp
     src/common/fontmap.cpp
     src/common/framecmn.cpp
Index: docs/latex/wx/fontenum.tex
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/docs/latex/wx/fontenum.tex,v
retrieving revision 1.10
diff -bu2 -r1.10 docs/latex/wx/fontenum.tex
--- docs/latex/wx/fontenum.tex
+++ docs/latex/wx/fontenum.tex
@@ -75,19 +75,17 @@
 \membersection{wxFontEnumerator::GetEncodings}\label{wxfontenumeratorgetencodings}
 
-\func{wxArrayString*}{GetEncodings}{\void}
+\func{static wxArrayString}{GetEncodings}{\void}
 
 Return array of strings containing all encodings found by 
-\helpref{EnumerateEncodings}{wxfontenumeratorenumerateencodings}. This is convenience function. It is 
-based on default implementation of \helpref{OnFontEncoding}{wxfontenumeratoronfontencoding} so don't expect
-it to work if you overwrite that method.
+\helpref{EnumerateEncodings}{wxfontenumeratorenumerateencodings}.
 
 \membersection{wxFontEnumerator::GetFacenames}\label{wxfontenumeratorgetfacenames}
 
-\func{wxArrayString*}{GetFacenames}{\void}
+\func{static wxArrayString}{GetFacenames}{
+    \param{wxFontEncoding }{encoding = wxFONTENCODING\_SYSTEM},
+    \param{bool }{fixedWidthOnly = false}}
 
 Return array of strings containing all facenames found by 
-\helpref{EnumerateFacenames}{wxfontenumeratorenumeratefacenames}. This is convenience function. It is 
-based on default implementation of \helpref{OnFacename}{wxfontenumeratoronfacename} so don't expect
-it to work if you overwrite that method.
+\helpref{EnumerateFacenames}{wxfontenumeratorenumeratefacenames}.
 
 
Index: include/wx/fontenum.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/fontenum.h,v
retrieving revision 1.19
diff -bu2 -r1.19 include/wx/fontenum.h
--- include/wx/fontenum.h
+++ include/wx/fontenum.h
@@ -25,5 +25,8 @@
 {
 public:
-    wxFontEnumerator() : m_Facenames(NULL), m_Encodings(NULL) { }
+    wxFontEnumerator() {}
+
+    // virtual dtor for the base class
+    virtual ~wxFontEnumerator() {}
 
     // start enumerating font facenames (either all of them or those which
@@ -47,40 +50,24 @@
 
     // called by EnumerateFacenames
-    virtual bool OnFacename(const wxString& facename)
-        {
-            if (m_Facenames == NULL) m_Facenames = new wxArrayString;
-            m_Facenames -> Add(facename);
-            return true;
-        }
+    virtual bool OnFacename(const wxString& WXUNUSED(facename))
+        { return true; }
 
     // called by EnumerateEncodings
     virtual bool OnFontEncoding(const wxString& WXUNUSED(facename),
-                                const wxString& encoding)
-        {
-            if (m_Encodings == NULL) m_Encodings = new wxArrayString;
-            m_Encodings -> Add(encoding);
-            return true;
-        }
-
-    // convenience function that returns array of facenames. Cannot be called
-    // before EnumerateFacenames.
-    wxArrayString *GetFacenames()
-        { return m_Facenames; }
+                                const wxString& WXUNUSED(encoding))
+        { return true; }
+
+
+
+    // convenience function that returns array of facenames.
+    static wxArrayString GetFacenames(
+                    wxFontEncoding encoding = wxFONTENCODING_SYSTEM, // all
+                    bool fixedWidthOnly = false);
 
     // convenience function that returns array of encodings.
-    // Cannot be called before EnumerateEncodings.
-    wxArrayString *GetEncodings()
-        { return m_Encodings; }
+    static wxArrayString GetEncodings();
 
-    // virtual dtor for the base class
-    virtual ~wxFontEnumerator()
-        {
-            if (m_Facenames) delete m_Facenames;
-            if (m_Encodings) delete m_Encodings;
-        }
 
 private:
-    wxArrayString *m_Facenames, *m_Encodings;
-
     DECLARE_NO_COPY_CLASS(wxFontEnumerator)
 };
Index: src/common/fontenumcmn.cpp
--- /dev/null
+++ src/common/fontenumcmn.cpp
@@ -0,0 +1,77 @@
+/////////////////////////////////////////////////////////////////////////////
+// Name:        src/common/fontenumcmn.cpp
+// Purpose:     wxFontEnumerator class
+// Author:      Vadim Zeitlin
+// Modified by:
+// Created:     7/5/2006
+// RCS-ID:      $Id: fontmap.cpp,v 1.75 2006/04/11 01:37:36 VZ Exp $
+// Copyright:   (c) 1999-2003 Vadim Zeitlin <vadim@wxwindows.org>
+// Licence:     wxWindows licence
+/////////////////////////////////////////////////////////////////////////////
+
+// ============================================================================
+// declarations
+// ============================================================================
+
+// ----------------------------------------------------------------------------
+// headers
+// ----------------------------------------------------------------------------
+
+// For compilers that support precompilation, includes "wx.h".
+#include "wx/wxprec.h"
+
+#ifdef __BORLANDC__
+    #pragma hdrstop
+#endif
+
+#include "wx/fontenum.h"
+
+// ============================================================================
+// implementation
+// ============================================================================
+
+
+// A simple wxFontEnumerator which do not perform any filtering and
+// just returns all facenames and encodings found in the system
+class WXDLLEXPORT wxSimpleFontEnumerator : public wxFontEnumerator
+{
+public:
+    wxSimpleFontEnumerator() {}
+
+
+    // called by EnumerateFacenames
+    virtual bool OnFacename(const wxString& facename)
+        {
+            m_arrFacenames.Add(facename);
+            return true;
+        }
+
+    // called by EnumerateEncodings
+    virtual bool OnFontEncoding(const wxString& WXUNUSED(facename),
+                                const wxString& encoding)
+        {
+            m_arrEncodings.Add(encoding);
+            return true;
+        }
+
+
+public:
+    wxArrayString m_arrFacenames, m_arrEncodings;
+};
+
+
+/* static */
+wxArrayString wxFontEnumerator::GetFacenames(wxFontEncoding encoding, bool fixedWidthOnly)
+{
+    wxSimpleFontEnumerator temp;
+    temp.EnumerateFacenames(encoding, fixedWidthOnly);
+    return temp.m_arrFacenames;
+}
+
+/* static */
+wxArrayString wxFontEnumerator::GetEncodings()
+{
+    wxSimpleFontEnumerator temp;
+    temp.EnumerateEncodings();
+    return temp.m_arrEncodings;
+}
Index: src/html/helpwnd.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/html/helpwnd.cpp,v
retrieving revision 1.8
diff -bu2 -r1.8 src/html/helpwnd.cpp
--- src/html/helpwnd.cpp
+++ src/html/helpwnd.cpp
@@ -1285,16 +1285,12 @@
     if (m_NormalFonts == NULL)
     {
-        wxFontEnumerator enu;
-        enu.EnumerateFacenames();
-        m_NormalFonts = new wxArrayString;
-        *m_NormalFonts = *enu.GetFacenames();
+        m_NormalFonts = new wxArrayString(wxFontEnumerator::GetFacenames());
         m_NormalFonts->Sort(); // ascending sort
     }
     if (m_FixedFonts == NULL)
     {
-        wxFontEnumerator enu;
-        enu.EnumerateFacenames(wxFONTENCODING_SYSTEM, true /*enum fixed width only*/);
-        m_FixedFonts = new wxArrayString;
-        *m_FixedFonts = *enu.GetFacenames();
+        m_FixedFonts = new wxArrayString(
+                    wxFontEnumerator::GetFacenames(wxFONTENCODING_SYSTEM,
+                    true /*enum fixed width only*/));
         m_FixedFonts->Sort(); // ascending sort
     }
Index: src/html/m_fonts.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/html/m_fonts.cpp,v
retrieving revision 1.28
diff -bu2 -r1.28 src/html/m_fonts.cpp
--- src/html/m_fonts.cpp
+++ src/html/m_fonts.cpp
@@ -68,11 +68,6 @@
         {
             if (m_Faces.GetCount() == 0)
-            {
-                wxFontEnumerator enu;
-                enu.EnumerateFacenames();
-                const wxArrayString *faces = enu.GetFacenames();
-                if ( faces )
-                    m_Faces = *faces;
-            }
+                m_Faces = wxFontEnumerator::GetFacenames();
+
             wxStringTokenizer tk(tag.GetParam(wxT("FACE")), wxT(","));
             int index;
Index: src/xrc/xmlres.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/xrc/xmlres.cpp,v
retrieving revision 1.88
diff -b -u -2 -r1.88 xmlres.cpp
--- src/xrc/xmlres.cpp  2006/05/06 16:59:59     1.88
+++ src/xrc/xmlres.cpp  2006/05/11 14:27:18
@@ -1352,13 +1352,12 @@
     {
         wxString faces = GetParamValue(wxT("face"));
-        wxFontEnumerator enu;
-        enu.EnumerateFacenames();
+        wxArrayString facenames(wxFontEnumerator::GetFacenames());
         wxStringTokenizer tk(faces, wxT(","));
         while (tk.HasMoreTokens())
         {
-            int index = enu.GetFacenames()->Index(tk.GetNextToken(), false);
+            int index = facenames.Index(tk.GetNextToken(), false);
             if (index != wxNOT_FOUND)
             {
-                facename = (*enu.GetFacenames())[index];
+                facename = facenames[index];
                 break;
             }

 	  	 
