diff -ru wxWidgets-2008-02-11/docs/latex/wx/app.tex MYwxWidgets-2008-02-11/docs/latex/wx/app.tex
--- wxWidgets-2008-02-11/docs/latex/wx/app.tex	2008-02-11 05:48:31.000000000 +0100
+++ MYwxWidgets-2008-02-11/docs/latex/wx/app.tex	2008-02-21 13:46:50.000000000 +0100
@@ -734,6 +734,17 @@
 \helpref{wxApp::GetInstance}{wxappgetinstance}
 
 
+\membersection{wxApp::SetNativeTheme}\label{wxappsetnativetheme}
+
+\func{void}{SetNativeTheme}{\param{const}{wxString\&}{theme_name}}
+
+Allows runtime switching of GTK-Themes.
+
+\wxheading{Parameters}
+
+\docparam{theme_name}{The name of the new theme or an absolute path to a gtkrc-theme-file}
+
+
 \membersection{wxApp::SetTopWindow}\label{wxappsettopwindow}
 
 \func{void}{SetTopWindow}{\param{wxWindow* }{window}}
diff -ru wxWidgets-2008-02-11/include/wx/app.h MYwxWidgets-2008-02-11/include/wx/app.h
--- wxWidgets-2008-02-11/include/wx/app.h	2008-02-11 05:47:14.000000000 +0100
+++ MYwxWidgets-2008-02-11/include/wx/app.h	2008-02-21 13:41:54.000000000 +0100
@@ -399,6 +399,7 @@
 
     // the virtual functions which may/must be overridden in the derived class
     // -----------------------------------------------------------------------
+    virtual void SetNativeTheme( const wxString& ) {}
 
         // very first initialization function
         //
diff -ru wxWidgets-2008-02-11/include/wx/gtk/app.h MYwxWidgets-2008-02-11/include/wx/gtk/app.h
--- wxWidgets-2008-02-11/include/wx/gtk/app.h	2008-02-11 05:48:19.000000000 +0100
+++ MYwxWidgets-2008-02-11/include/wx/gtk/app.h	2008-02-21 13:43:10.000000000 +0100
@@ -36,6 +36,7 @@
      * (palette/visual/colorcube). under wxMSW, OnInitGui() does nothing by
      * default. when overriding this method, the code in it is likely to be
      * platform dependent, otherwise use OnInit(). */
+    virtual void SetNativeTheme( const wxString& theme_name );
     virtual bool OnInitGui();
 
     // override base class (pure) virtuals
diff -ru wxWidgets-2008-02-11/src/gtk/app.cpp MYwxWidgets-2008-02-11/src/gtk/app.cpp
--- wxWidgets-2008-02-11/src/gtk/app.cpp	2008-02-11 05:48:19.000000000 +0100
+++ MYwxWidgets-2008-02-11/src/gtk/app.cpp	2008-02-21 13:45:22.000000000 +0100
@@ -251,6 +251,24 @@
 {
 }
 
+void wxApp::SetNativeTheme( const wxString& theme_name )
+{
+    wxString path;
+    path = gtk_rc_get_theme_dir();
+    path += "/";
+    path += theme_name.utf8_str();
+    path += "/gtk-2.0/gtkrc";
+
+    if (wxFileExists(path.utf8_str()))
+        gtk_rc_add_default_file (path.utf8_str());
+    else if (wxFileExists(theme_name.utf8_str()))
+        gtk_rc_add_default_file (theme_name.utf8_str());
+    else
+        wxLogWarning("ERROR: Theme not found");
+
+    gtk_rc_reparse_all_for_settings(gtk_settings_get_default(), TRUE);
+}
+
 bool wxApp::OnInitGui()
 {
     if ( !wxAppBase::OnInitGui() )

 	  	 
