--- wxWidgets/src/common/sckstrm.cpp	Mon Apr 24 15:26:01 2006
+++ sckstrm.cpp	Wed Apr 26 14:38:10 2006
@@ -40,11 +40,28 @@
 
 size_t wxSocketOutputStream::OnSysWrite(const void *buffer, size_t size)
 {
-    size_t ret = m_o_socket->Write((const char *)buffer, size).LastCount();
+    const char *buf = (const char *)buffer;
+    size_t count = 0;
 
-    m_lasterror = m_o_socket->Error() ? wxSTREAM_WRITE_ERROR : wxSTREAM_NO_ERROR;
+    while ( count < size && m_o_socket->WaitForWrite() )
+    {
+        const size_t ret = m_o_socket->Write(buf, size - count).LastCount();
+
+        buf += ret;
+        count += ret;
+
+        if ( m_o_socket->Error() )
+        {
+            if (m_o_socket->LastError() != wxSOCKET_WOULDBLOCK)
+            {
+                m_lasterror = wxSTREAM_WRITE_ERROR;
+                return count;
+            }
+        }
+    }
 
-    return ret;
+    m_lasterror = wxSTREAM_NO_ERROR;
+    return count;
 }
 
 // ---------------------------------------------------------------------------
@@ -62,11 +79,28 @@
 
 size_t wxSocketInputStream::OnSysRead(void *buffer, size_t size)
 {
-    size_t ret = m_i_socket->Read((char *)buffer, size).LastCount();
+    char *buf = (char *)buffer;
+    size_t count = 0;
 
-    m_lasterror = m_i_socket->Error() ? wxSTREAM_READ_ERROR : wxSTREAM_NO_ERROR;
+    while ( count < size && m_i_socket->WaitForRead() )
+    {
+        const size_t ret = m_i_socket->Read(buf, size - count).LastCount();
+
+        buf += ret;
+        count += ret;
+
+        if ( m_i_socket->Error() )
+        {
+            if (m_i_socket->LastError() != wxSOCKET_WOULDBLOCK)
+            {
+                m_lasterror = wxSTREAM_READ_ERROR;
+                return count;
+            }
+        }
+    }
 
-    return ret;
+    m_lasterror = wxSTREAM_NO_ERROR;
+    return count;
 }
 
 // ---------------------------------------------------------------------------

 	  	 
