Index: samples/listctrl/listtest.cpp
===================================================================
--- samples/listctrl/listtest.cpp	(revision 64194)
+++ samples/listctrl/listtest.cpp	(working copy)
@@ -503,7 +503,6 @@
     // items with images initially even if we specify dummy image id
     wxListItem itemCol;
     itemCol.SetText(wxT("Column 1"));
-    itemCol.SetImage(-1);
     m_listCtrl->InsertColumn(0, itemCol);
 
     itemCol.SetText(wxT("Column 2"));
@@ -549,11 +548,7 @@
     m_listCtrl->SetColumnWidth( 1, wxLIST_AUTOSIZE );
     m_listCtrl->SetColumnWidth( 2, wxLIST_AUTOSIZE );
 
-    // Set images in columns
-    m_listCtrl->SetItemColumnImage(1, 1, 0);
-
     wxListItem info;
-    info.SetImage(0);
     info.SetId(3);
     info.SetColumn(2);
     m_listCtrl->SetItem(info);
