--- progdlg.h.old	2017-11-03 09:03:02.170345000 +0100
+++ progdlg.h	2017-11-03 09:39:27.614097000 +0100
@@ -49,6 +49,9 @@
             }
         }
     @endcode
+    This becomes even more important if the dialog is instanced during the
+    program initialization, e.g. from wxApp::OnInit(): the dialog must be
+    destroyed before the main event loop is started.
 
     @beginStyleTable
     @style{wxPD_APP_MODAL}
