Index: src/common/dpycmn.cpp
===================================================================
--- src/common/dpycmn.cpp	(revision 65199)
+++ src/common/dpycmn.cpp	(working copy)
@@ -234,8 +234,11 @@
 int wxDisplayFactory::GetFromWindow(const wxWindow *window)
 {
     // consider that the window belongs to the display containing its centre
-    const wxRect r(window->GetRect());
-    return GetFromPoint(wxPoint(r.x + r.width/2, r.y + r.height/2));
+    const wxRect r(window->GetScreenRect());
+
+    wxPoint center(r.x + r.width/2, r.y + r.height/2);
+
+    return GetFromPoint(center);
 }
 
 // ============================================================================
