Index: src/generic/grid.cpp
===================================================================
--- src/generic/grid.cpp	(r‚vision 54636)
+++ src/generic/grid.cpp	(copie de travail)
@@ -10194,7 +10194,7 @@
                       wxT("wxGrid::SetCellSize setting cell size to < 1"));
 
         // if this was already a multicell then "turn off" the other cells first
-        if ((cell_rows > 1) || (cell_rows > 1))
+        if ((cell_rows > 1) || (cell_cols > 1))
         {
             int i, j;
             for (j=row; j < row + cell_rows; j++)
