From e981221f0d7151c203c9581e48d090f059b91580 Mon Sep 17 00:00:00 2001
From: Pierre Lejeune <superheron@gmail.com>
Date: Tue, 25 Aug 2015 22:05:06 +0200
Subject: [PATCH] Get pkg-config path from environment

---
 config.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/config.py b/config.py
index b082631..5f908d3 100644
--- a/config.py
+++ b/config.py
@@ -245,7 +245,7 @@ CLEANUP = []
 force = '--force' in sys.argv or '-f' in sys.argv
 debug = '--debug' in sys.argv or '-g' in sys.argv
 cleaning = 'clean' in sys.argv
-
+pkg_config = os.getenv('PKG_CONFIG') or 'pkg-config'
 
 # change the PORT default for wxMac
 if sys.platform[:6] == "darwin":
@@ -1102,11 +1102,11 @@ elif os.name == 'posix' or COMPILER == 'mingw32':
         elif WXPORT == 'gtk2':
             WXPLAT = '__WXGTK__'
             GENDIR = 'gtk' # no code differences so use the same generated sources
-            portcfg = os.popen('pkg-config gtk+-2.0 --cflags', 'r').read()[:-1]
+            portcfg = os.popen(pkg_config + ' gtk+-2.0 --cflags', 'r').read()[:-1]
         elif WXPORT == 'gtk3':
             WXPLAT = '__WXGTK__'
             GENDIR = 'gtk' # no code differences so use the same generated sources
-            portcfg = os.popen('pkg-config gtk+-3.0 --cflags', 'r').read()[:-1]
+            portcfg = os.popen(pkg_config + ' gtk+-3.0 --cflags', 'r').read()[:-1]
         elif WXPORT == 'x11':
             WXPLAT = '__WXX11__'
             portcfg = ''
-- 
2.5.0

