--- wxWindows/include/wx/wxchar.h	2003-12-31 07:39:10.000000000 +0100
+++ wxWin/include/wx/wxchar.h	2004-01-06 16:30:00.000000000 +0100
@@ -73,3 +73,3 @@
 /*  Actually MinGW has tchar.h, but it does not include wchar.h */
-#if defined(__MWERKS__) || defined(__VISAGECPP__) || defined(__MINGW32__)
+#if defined(__MWERKS__) || defined(__VISAGECPP__) || defined(__MINGW32__) || defined(__WATCOMC__)
     #ifndef HAVE_WCHAR_H
@@ -703,2 +703,6 @@
         #endif
+        #if defined(__WATCOMC__)
+            #define wxVsnprintf_    _vsnwprintf
+            #define wxSnprintf_     _snwprintf
+        #endif
         #if defined(HAVE__VSNWPRINTF)
@@ -712,3 +716,3 @@
         /*  vsnprintf() */
-        #if defined(HAVE_VSNPRINTF) || defined(__MWERKS__)
+        #if defined(HAVE_VSNPRINTF) || defined(__MWERKS__) || defined(__WATCOMC__)
             /*  assume we have snprintf() too if we have vsnprintf() */

 	  	 
