Index: src/msw/dc.cpp
===================================================================
RCS file: /usr/local/cvsroot/iml/FiveL/Release/libs/wxWindows/src/msw/dc.cpp,v
retrieving revision 1.1.1.4
diff -u -r1.1.1.4 dc.cpp
--- src/msw/dc.cpp	2 May 2003 17:22:48 -0000	1.1.1.4
+++ src/msw/dc.cpp	7 May 2003 21:12:08 -0000
@@ -2021,10 +2021,16 @@
             {
                 StretchBltModeChanger changeMode(GetHdc(), COLORONCOLOR);
 
+                // Figure out what co-ordinate system we're supposed to
+                // specify ysrc in.
+                LONG dib_height = ds.dsBmih.biHeight;
+                bool bottom_up = (dib_height > 0);
+                int fixed_ysrc = bottom_up ? dib_height-(ysrc+height) : ysrc;
+
                 if ( ::StretchDIBits(GetHdc(),
                                      xdest, ydest,
                                      width, height,
-                                     0, 0,
+                                     xsrc, fixed_ysrc,
                                      width, height,
                                      ds.dsBm.bmBits,
                                      (LPBITMAPINFO)&ds.dsBmih,

 	  	 
