Index: src/mac/carbon/toplevel.cpp
===================================================================
--- src/mac/carbon/toplevel.cpp	(revision 53252)
+++ src/mac/carbon/toplevel.cpp	(working copy)
@@ -177,7 +177,7 @@
 {
     if ( show )
     {
-        FullScreenData *data = (FullScreenData *)m_macFullScreenData ;
+        FullScreenData *data = (FullScreenData *) m_macFullScreenData ;
         delete data ;
         data = new FullScreenData() ;
 
@@ -230,15 +230,18 @@
     }
     else
     {
-        ShowMenuBar() ;
-        FullScreenData *data = (FullScreenData *) m_macFullScreenData ;
-        if ( data->m_wasResizable )
-            MacChangeWindowAttributes( kWindowResizableAttribute ,  kWindowNoAttributes ) ;
-        SetPosition( data->m_position ) ;
-        SetSize( data->m_size ) ;
+        if ( m_macFullScreenData != NULL )
+        {
+            ShowMenuBar() ;
+            FullScreenData *data = (FullScreenData *) m_macFullScreenData ;
+            if ( data->m_wasResizable )
+                MacChangeWindowAttributes( kWindowResizableAttribute ,  kWindowNoAttributes ) ;
+            SetPosition( data->m_position ) ;
+            SetSize( data->m_size ) ;
 
-        delete data ;
-        m_macFullScreenData = NULL ;
+            delete data ;
+            m_macFullScreenData = NULL ;
+        }
     }
 
     return false;

 	  	 
