diff -bu1 -r wxWindows/samples/vscroll/vstest.cpp wxWin/samples/vscroll/vstest.cpp
--- wxWindows/samples/vscroll/vstest.cpp	2003-05-30 23:08:30.000000000 +0200
+++ wxWin/samples/vscroll/vstest.cpp	2003-08-27 17:23:56.000000000 +0200
@@ -75,3 +75,3 @@
             wxSize sz = GetClientSize();
-            SetStatusText(wxString::Format("%dx%d", sz.x, sz.y), 1);
+            SetStatusText(wxString::Format(_T("%dx%d"), sz.x, sz.y), 1);
         }
@@ -103,3 +103,3 @@
                                (
-                                    "Page size = %d, pos = %d, max = %d",
+                                    _T("Page size = %d, pos = %d, max = %d"),
                                     GetScrollThumb(wxVERTICAL),

 	  	 
