diff --git a/samples/widgets/widgets.cpp b/samples/widgets/widgets.cpp
index 0d26fc05e8..99e9fb972f 100644
--- a/samples/widgets/widgets.cpp
+++ b/samples/widgets/widgets.cpp
@@ -809,6 +809,7 @@ wxColour GetColourFromUser(wxWindow *parent, const wxColour& colDefault)
 
 void WidgetsFrame::OnSetFgCol(wxCommandEvent& WXUNUSED(event))
 {
+    wxMessageBox( wxGetCwd() );
     // allow for debugging the default colour the first time this is called
     WidgetsPage *page = CurrentPage();
 
