--- toolbar.cpp.old	2002-07-23 14:02:40.000000000 +0200
+++ toolbar.cpp	2002-07-23 16:22:08.000000000 +0200
@@ -286,5 +286,8 @@
                 wxSize sz = control->GetSize();
                 wxPoint pos = control->GetPosition();
-                control->Move(currentX, pos.y);
+                // Allow a control to specify a y-offset by setting its initial position,
+                // but still don't allow it to position itself above the top margin.
+                int controlY = (pos.y > 0) ? currentY + pos.y : currentY;
+                control->Move(currentX, controlY);
                 currentX += sz.x + packing;
 

 	  	 
