--- D:/wxWidgets/2.9.3-org/include/wx/msw/dialog.h	Wed Dec 14 19:44:00 2011
+++ D:/wxWidgets/2.9.3/include/wx/msw/dialog.h	Wed Jan 11 12:23:31 2012
@@ -94,7 +94,9 @@
 
     virtual void Raise();
 
-    virtual void SetWindowStyleFlag(long style);
+#if wxUSE_DIALOG_SIZEGRIP
+	virtual void SetWindowStyleFlag(long style);
+#endif
 
 #ifdef __POCKETPC__
     // Responds to the OK button in a PocketPC titlebar. This
