Index: src/generic/datectlg.cpp
===================================================================
--- src/generic/datectlg.cpp	(revision 54278)
+++ src/generic/datectlg.cpp	(working copy)
@@ -76,7 +76,7 @@
                               | wxCAL_SHOW_HOLIDAYS | wxBORDER_SUNKEN) )
             return false;
 
-        SetFormat("%x");
+        SetFormat(GetLocaleDateFormat());
 
         m_useSize  = wxCalendarCtrl::GetBestSize();
 
@@ -197,45 +197,72 @@
         return m_combo->GetParent()->HasFlag(flag);
     }
 
-    bool SetFormat(const wxString& fmt)
+    // it expands "%x" format and changes %y to %Y if wxDP_SHOWCENTURY flag 
+    // is given. If the locale format can't be easily analyzed (e.g. when 
+    // the month is given as a name, not number), "%x" is returned
+    wxString GetLocaleDateFormat()
     {
-        m_format.clear();
+        wxString x_format = "%x";
+        wxString fmt;
+        int year_cnt = 0, month_cnt = 0, day_cnt = 0;
 
         wxDateTime dt;
-        dt.ParseFormat(wxT("2003-10-13"), wxT("%Y-%m-%d"));
-        wxString str(dt.Format(fmt));
+        dt.ParseFormat(wxT("2003-10-17"), wxT("%Y-%m-%d"));
+        wxString str(dt.Format(x_format));
 
         const wxChar *p = str.c_str();
         while ( *p )
         {
-            int n=wxAtoi(p);
-            if (n == dt.GetDay())
+            if (wxIsdigit(*p)) 
             {
-                m_format.Append(wxT("%d"));
-                p += 2;
+                int n=wxAtoi(p);
+                if (n == dt.GetDay())
+                {
+                    fmt.Append(wxT("%d"));
+                    day_cnt++;
+                    p += 2;
+                }
+                else if (n == (int)dt.GetMonth()+1)
+                {
+                    fmt.Append(wxT("%m"));
+                    month_cnt++;
+                    p += 2;
+                }
+                else if (n == dt.GetYear())
+                {
+                    fmt.Append(wxT("%Y"));
+                    year_cnt++;
+                    p += 4;
+                }
+                else if (n == (dt.GetYear() % 100))
+                {
+                    if ( HasDPFlag(wxDP_SHOWCENTURY) )
+                        fmt.Append(wxT("%Y"));
+                    else
+                        fmt.Append(wxT("%y"));
+                    year_cnt++;
+                    p += 2;
+                }
+                else
+                    // this shouldn't happen
+                    return x_format;
             }
-            else if (n == (int)dt.GetMonth()+1)
-            {
-                m_format.Append(wxT("%m"));
-                p += 2;
+            else {
+                fmt.Append(*p);
+                p++;
             }
-            else if (n == dt.GetYear())
-            {
-                m_format.Append(wxT("%Y"));
-                p += 4;
-            }
-            else if (n == (dt.GetYear() % 100))
-            {
-                if ( HasDPFlag(wxDP_SHOWCENTURY) )
-                    m_format.Append(wxT("%Y"));
-                else
-                    m_format.Append(wxT("%y"));
-                p += 2;
-            }
-            else
-                m_format.Append(*p++);
         }
 
+        if (year_cnt == 1 && month_cnt == 1 && day_cnt == 1)
+            return fmt;
+        else
+            return x_format;
+    }
+
+    bool SetFormat(const wxString& fmt)
+    {
+        m_format = fmt;
+
         if ( m_combo )
         {
             wxArrayString allowedChars;
