Index: src/generic/datectlg.cpp
===================================================================
--- src/generic/datectlg.cpp	(revision 53876)
+++ src/generic/datectlg.cpp	(working copy)
@@ -248,6 +248,14 @@
 
     bool SetFormat(const wxString& fmt)
     {
+        m_format = fmt;
+#if 0
+        // The commented code here is the old way to set m_format.
+        // It seems to me (MW) that the only it wasn't written in one line
+        // was to support wxDP_SHOWCENTURY flag. The drawback was that it
+        // didn't work well with locale settings that use textual month in %x,
+        // see #4743.
+        
         m_format.clear();
 
         wxDateTime dt;
@@ -284,6 +292,7 @@
             else
                 m_format.Append(*p++);
         }
+#endif
 
         if ( m_combo )
         {
