Index: dcprint.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/dcprint.cpp,v
retrieving revision 1.31.2.1
diff -b -u -2 -r1.31.2.1 dcprint.cpp
--- dcprint.cpp	2002/10/22 21:34:22	1.31.2.1
+++ dcprint.cpp	2003/01/20 19:50:27
@@ -531,5 +531,9 @@
 
             BITMAPINFO *info = (BITMAPINFO *) malloc( sizeof( BITMAPINFOHEADER ) + 256 * sizeof(RGBQUAD ) );
-            int iBitsSize = ((width + 3 ) & ~3 ) * height;
+#if wxUSE_DRAWBITMAP_24BITS
+            int iBitsSize = (((width * 3) + 3 ) & ~3 ) * height;
+#else
+            int iBitsSize = ((width + 3 ) & ~3 ) * height ;
+#endif
 
             void* bits = malloc( iBitsSize );
@@ -541,5 +545,9 @@
             info->bmiHeader.biHeight = height;
             info->bmiHeader.biPlanes = 1;
+#if wxUSE_DRAWBITMAP_24BITS
+            info->bmiHeader.biBitCount = 24;
+#else
             info->bmiHeader.biBitCount = 8;
+#endif
             info->bmiHeader.biCompression = BI_RGB;
 

 	  	 
