--- /home/donald/wxWindows/wxGTK-2.4.1/configure.in	2003-08-08 18:10:58.000000000 -0500
+++ /home/donald/wxWindows/wxGTK-2.4.1a/configure.in	2003-08-10 16:30:47.000000000 -0500
@@ -708,5 +708,5 @@
 WX_ARG_SYS_WITH(regex,     [  --with-regex            enable support for wxRegEx class], wxUSE_REGEX)
 WX_ARG_SYS_WITH(zlib,      [  --with-zlib             use zlib for LZW compression], wxUSE_ZLIB)
-WX_ARG_WITH(odbc,          [  --with-odbc             use the IODBC and wxODBC classes], wxUSE_ODBC)
+WX_ARG_ODBC(odbc,          [  --with-odbc             yes, unixodbc, or iodbc], wxUSE_ODBC)
 
 dnl ---------------------------------------------------------------------------
@@ -2664,6 +2664,63 @@
 
 dnl ODBC objects are Unix only
-if test "$TOOLKIT" != "MSW" -a "$wxUSE_ODBC" = "yes" ; then
-    ALL_OBJECTS="${ALL_OBJECTS} \$(IODBCOBJS)"
+if test "$TOOLKIT" != "MSW" ; then
+    if test "$wxUSE_ODBC" = "yes" ; then
+        ALL_OBJECTS="${ALL_OBJECTS} \$(IODBCOBJS)"
+    elif test "$wxUSE_ODBC" = "unixodbc" ; then
+        AC_MSG_CHECKING(for unixODBC headers)
+        WX_PATH_FIND_INCLUDES("",[sql.h])
+        if test "$ac_find_includes" != "" ; then
+            AC_MSG_RESULT(found in $ac_find_includes)
+        else
+            WX_PATH_FIND_INCLUDES($SEARCH_INCLUDE,[sql.h]) 
+            if test "$ac_find_includes" != "" ; then
+                AC_MSG_RESULT(found in $ac_find_includes)
+                WX_INCLUDE_PATH_EXIST($ac_find_includes, $TOOLKIT_INCLUDE)
+                TOOLKIT_INCLUDE="$TOOLKIT_INCLUDE$ac_path_to_include"    
+            else
+                AC_MSG_RESULT([not found!])
+                AC_MSG_ERROR([Aborting...])
+            fi
+        fi
+
+        AC_MSG_CHECKING([for unixODBC Shared Library])
+        WX_PATH_FIND_LIBRARIES([$SEARCH_LIB],[odbc])
+        if test "$ac_find_libraries" != "" ; then
+            WX_LINK_PATH_EXIST([$ac_find_libraries],[$LDFLAGS])
+            if test "$ac_path_to_link" != " -L/usr/lib" ; then
+                LDFLAGS="$LDFLAGS$ac_path_to_link"
+            fi
+            LIBS=" -lodbc$LIBS"
+            AC_MSG_RESULT([yes])
+        else
+            AC_MSG_RESULT([no])
+            AC_MSG_ERROR([Cannot locate unixODBC shared library (libodbc)])
+        fi
+    elif test "$wxUSE_ODBC" = "iodbc" ; then
+        AC_MSG_CHECKING(for iODBC headers)
+        WX_PATH_FIND_INCLUDES($SEARCH_INCLUDE,[sql.h]) 
+        if test "$ac_find_includes" != "" ; then
+            AC_MSG_RESULT(found in $ac_find_includes)
+            WX_INCLUDE_PATH_EXIST($ac_find_includes, $TOOLKIT_INCLUDE)
+            TOOLKIT_INCLUDE="$TOOLKIT_INCLUDE$ac_path_to_include"    
+        else
+            AC_MSG_RESULT([not found!])
+            AC_MSG_ERROR([Aborting...])
+        fi
+
+        AC_MSG_CHECKING([for iODBC Shared Library])
+        WX_PATH_FIND_LIBRARIES([$SEARCH_LIB],[iodbc])
+        if test "$ac_find_libraries" != "" ; then
+            WX_LINK_PATH_EXIST([$ac_find_libraries],[$LDFLAGS])
+            if test "$ac_path_to_link" != " -L/usr/lib" ; then
+                LDFLAGS="$LDFLAGS$ac_path_to_link"
+            fi
+            LIBS=" -liodbc$LIBS"
+            AC_MSG_RESULT([yes])
+        else
+            AC_MSG_RESULT([no])
+            AC_MSG_ERROR([Cannot locate iODBC shared library (libiodbc)])
+        fi
+    fi
 fi
 
@@ -4483,5 +4540,5 @@
 
 IODBC_C_SRC=""
-if test "$wxUSE_ODBC" = "yes" ; then
+if test "$wxUSE_ODBC" = "yes" -o "$wxUSE_ODBC" = "unixodbc" -o "$wxUSE_ODBC" = "iodbc"; then
     AC_DEFINE(wxUSE_ODBC)
     WXODBCFLAG="-D_IODBC_"
@@ -5581,6 +5638,5 @@
 echo "                                       tiff               ${wxUSE_LIBTIFF-none}"
 echo "                                       zlib               ${wxUSE_ZLIB}"
-
-echo ""
+echo "                                       odbc               ${wxUSE_ODBC-no}"
 
 dnl vi: set et sts=4 sw=4 com=\:dnl:

 	  	 
