Index: src/common/wincmn.cpp
===================================================================
--- src/common/wincmn.cpp	(revision 62774)
+++ src/common/wincmn.cpp	(working copy)
@@ -319,6 +319,20 @@
 
     wxASSERT_MSG( GetChildren().GetCount() == 0, wxT("children not destroyed") );
 
+    // Remove us from chain of event handlers.
+    if ( GetEventHandler () != this ) {
+
+        wxEvtHandler* pLast = GetEventHandler ();
+        wxEvtHandler* pNext = pLast->GetNextHandler();
+        while ( pNext && pNext != this ) {
+
+            pLast = pNext;
+            pNext = pLast->GetNextHandler();
+        }
+        wxASSERT_MSG ( pNext == this , wxT("Window not in wxEvtHandler chain.") );
+        pLast->SetNextHandler ( NULL );
+    }
+
     // notify the parent about this window destruction
     if ( m_parent )
         m_parent->RemoveChild(this);
