Index: interface/wx/help.h
===================================================================
--- interface/wx/help.h	(revision 70852)
+++ interface/wx/help.h	(working copy)
@@ -7,6 +7,15 @@
 /////////////////////////////////////////////////////////////////////////////
 
 /**
+    Help search modes for wxHelpController.
+*/
+enum wxHelpSearchMode
+{
+    wxHELP_SEARCH_INDEX,
+    wxHELP_SEARCH_ALL
+};
+
+/**
     @class wxHelpController
 
     This is a family of classes by which applications may invoke a help viewer
