Index: src/gtk/scrolwin.cpp
===================================================================
RCS file: /home/cvssoftware/build/wxPython-3.0/src/gtk/scrolwin.cpp,v
retrieving revision 1.1
diff -u -r1.1 scrolwin.cpp
--- src/gtk/scrolwin.cpp	29 Jan 2014 15:19:39 -0000	1.1
+++ src/gtk/scrolwin.cpp	16 Sep 2015 11:28:17 -0000
@@ -97,6 +97,12 @@
         return;
     }
 
+    m_targetWindow->GetClientSize(&w, &h);
+	if ( w == 0 || h == 0 )
+	{
+		return;
+	}
+
     m_targetWindow->GetClientSize(&w, NULL);
     DoAdjustHScrollbar(w, vw);
 
