Index: src/osx/cocoa/nonownedwnd.mm
===================================================================
--- src/osx/cocoa/nonownedwnd.mm	(revision 70293)
+++ src/osx/cocoa/nonownedwnd.mm	(working copy)
@@ -990,6 +990,11 @@
     return [m_macWindow isDocumentEdited];
 }
 
+void wxNonOwnedWindowCocoaImpl::SetRepresentedFilename(const wxString& filename)
+{
+    [m_macWindow setRepresentedFilename:wxCFStringRef(filename).AsNSString()];
+}
+
 void wxNonOwnedWindowCocoaImpl::RestoreWindowLevel()
 {
     if ( [m_macWindow level] != m_macWindowLevel )
Index: src/osx/toplevel_osx.cpp
===================================================================
--- src/osx/toplevel_osx.cpp	(revision 70293)
+++ src/osx/toplevel_osx.cpp	(working copy)
@@ -213,3 +213,8 @@
 {
     return m_nowpeer->IsModified();
 }
+
+void wxTopLevelWindowMac::SetRepresentedFilename(const wxString& filename)
+{
+    m_nowpeer->SetRepresentedFilename(filename);
+}
Index: include/wx/toplevel.h
===================================================================
--- include/wx/toplevel.h	(revision 70293)
+++ include/wx/toplevel.h	(working copy)
@@ -301,6 +301,8 @@
     virtual void OSXSetModified(bool modified) { m_modified = modified; }
     virtual bool OSXIsModified() const { return m_modified; }
 
+    virtual void SetRepresentedFilename(const wxString& WXUNUSED(filename)) { }
+
 protected:
     // the frame client to screen translation should take account of the
     // toolbar which may shift the origin of the client area
Index: include/wx/osx/toplevel.h
===================================================================
--- include/wx/osx/toplevel.h	(revision 70293)
+++ include/wx/osx/toplevel.h	(working copy)
@@ -77,6 +77,8 @@
     virtual void OSXSetModified(bool modified);
     virtual bool OSXIsModified() const;
 
+    virtual void SetRepresentedFilename(const wxString& filename);
+
 protected:
     // common part of all ctors
     void Init();
Index: include/wx/osx/cocoa/private.h
===================================================================
--- include/wx/osx/cocoa/private.h	(revision 70293)
+++ include/wx/osx/cocoa/private.h	(working copy)
@@ -239,6 +239,8 @@
     virtual void SetModified(bool modified);
     virtual bool IsModified() const;
 
+    virtual void SetRepresentedFilename(const wxString& filename);
+
     wxNonOwnedWindow*   GetWXPeer() { return m_wxPeer; }
     
     CGWindowLevel   GetWindowLevel() const { return m_macWindowLevel; }
Index: include/wx/osx/core/private.h
===================================================================
--- include/wx/osx/core/private.h	(revision 70293)
+++ include/wx/osx/core/private.h	(working copy)
@@ -863,6 +863,8 @@
     virtual void SetModified(bool WXUNUSED(modified)) { }
     virtual bool IsModified() const { return false; }
 
+    virtual void SetRepresentedFilename(const wxString& WXUNUSED(filename)) { }
+
 #if wxOSX_USE_IPHONE
     virtual CGFloat GetWindowLevel() const { return 0.0; }
 #else
Index: interface/wx/toplevel.h
===================================================================
--- interface/wx/toplevel.h	(revision 70293)
+++ interface/wx/toplevel.h	(working copy)
@@ -504,6 +504,13 @@
     virtual bool OSXIsModified() const;
 
     /**
+        This function sets the wxTopLevelWindow's represented filename. This
+        function currently only sets the represented filename and proxy icon
+        on OS X and does nothing on other ports.
+    */
+    virtual void SetRepresentedFilename(const wxString& filename);
+
+    /**
         Depending on the value of @a show parameter the window is either shown
         full screen or restored to its normal state. @a style is a bit list
         containing some or all of the following values, which indicate what
