Index: src/msw/treectrl.cpp
===================================================================
--- src/msw/treectrl.cpp	(revision 63133)
+++ src/msw/treectrl.cpp	(working copy)
@@ -1667,6 +1667,11 @@
             return;
         }
 
+        if ( item == m_htSelStart )
+        {
+            m_htSelStart.Unset();
+        }
+
         if ( next.IsOk() )
         {
             wxTreeEvent changingEvent(wxEVT_COMMAND_TREE_SEL_CHANGING, this, next);
@@ -1734,6 +1739,8 @@
     {
         wxLogLastError(wxT("TreeView_DeleteAllItems"));
     }
+
+    m_htSelStart.Unset();
 }
 
 void wxTreeCtrl::DoExpand(const wxTreeItemId& item, int flag)
