Index: build/bakefiles/wxpresets/presets/wx.bkl
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/build/bakefiles/wxpresets/presets/wx.bkl,v
retrieving revision 1.1
diff -b -u -2 -r1.1 wx.bkl
--- build/bakefiles/wxpresets/presets/wx.bkl	2005/01/27 22:47:16	1.1
+++ build/bakefiles/wxpresets/presets/wx.bkl	2005/08/08 21:00:53
@@ -11,4 +11,7 @@
 <makefile>
     
+	<if cond="not isdefined('_WX_INC')">
+		<set var="_WX_INC">1</set>
+		
     <!-- this is just a wrapper that includes the real implementation: -->
 
@@ -28,4 +31,5 @@
         <error>This format is not (yet) supported by wx preset.</error>
     </if>
+	</if>
 
 </makefile>
Index: build/bakefiles/wxpresets/presets/wx_unix.bkl
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/build/bakefiles/wxpresets/presets/wx_unix.bkl,v
retrieving revision 1.2
diff -b -u -2 -r1.2 wx_unix.bkl
--- build/bakefiles/wxpresets/presets/wx_unix.bkl	2005/01/28 22:25:08	1.2
+++ build/bakefiles/wxpresets/presets/wx_unix.bkl	2005/08/14 10:26:25
@@ -8,11 +8,13 @@
 Usage:
     Options WX_CPPFLAGS, WX_CFLAGS, WX_CXXFLAGS, WX_LIBS are defined.
+    For compatibility with wx_win32.bkl and to allow conditions of the
+    wxWidgets build settings, also the WX_SHARED, WX_UNICODE, 
+    WX_DEBUG, WX_VERSION, WX_PORT options are defined.	
     
-
 Format-specific notes:
 
 * autoconf:
-        Beware that you have to use AM_OPTIONS_WXCONFIG and
-        AM_PATH_WXCONFIG in your configure.in!
+        Beware that you have to use AM_WXPRESETS in your configure.in to initialize
+        all the values of the options listed above !
 
 -->
@@ -31,4 +33,16 @@
         <option name="WX_CPPFLAGS"/>
         <option name="WX_LIBS"/>
+		
+		<!-- anyway we still provide the following for port-specific settings;
+		        they can be set in the configure.in/ac file using the AM_WXPRESETS macro -->
+		<option name="WX_SHARED"/>
+		<option name="WX_UNICODE"/>
+		<option name="WX_DEBUG"/>
+		<option name="WX_VERSION"/>
+		<option name="WX_PORT"/>
+		
+		<!-- we just avoid to define WX_DIR since it has no sense with an installed version of wx in a unix-like system 
+		        (includes are in $PREFIX/include, libraries in $PREFIX/lib... etc); in fact, user bakefiles should completely
+				avoid the use of WX_DIR variable both under win32 and under other platforms. -->
     </if>
     
Index: build/bakefiles/wxpresets/presets/wx_win32.bkl
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/build/bakefiles/wxpresets/presets/wx_win32.bkl,v
retrieving revision 1.16
diff -b -u -2 -r1.16 wx_win32.bkl
--- build/bakefiles/wxpresets/presets/wx_win32.bkl	2005/06/16 18:40:20	1.16
+++ build/bakefiles/wxpresets/presets/wx_win32.bkl	2005/08/14 10:29:41
@@ -114,4 +114,12 @@
     </if>
 
+	<!-- on win32 this is a fixed variable; on other platforms (like unix, where exist multiple ports) this is an option -->
+    <set var="WX_PORT">
+        <if cond="FORMAT=='msevc4prj'">wince</if>
+        <if cond="FORMAT!='msevc4prj'">msw</if>
+    </set>
+	
+	
+	
     <!--                     HELPER VARIABLES                       -->
     <!--                                                            -->
@@ -153,9 +161,5 @@
     </set>
 
-    <set var="WXPORT">
-        <if cond="FORMAT=='msevc4prj'">wince</if>
-        <if cond="FORMAT!='msevc4prj'">msw</if>
-    </set>
-    <set var="WXLIBINCLUDE">$(WXLIBPATH)$(DIRSEP)$(WXPORT)$(WXLIBPOSTFIX)</set>
+    <set var="WXLIBINCLUDE">$(WXLIBPATH)$(DIRSEP)$(WX_PORT)$(WXLIBPOSTFIX)</set>
 
 
@@ -189,5 +193,5 @@
         </if>
         <if cond="value not in ['base','net','xml']">
-            <sys-lib>wx$(WXPORT)$(WX_VERSION)$(WXLIBPOSTFIX)_$(value)</sys-lib>
+            <sys-lib>wx$(WX_PORT)$(WX_VERSION)$(WXLIBPOSTFIX)_$(value)</sys-lib>
         </if>
     </define-tag>

 	  	 
