Index: include/wx/docview.h
===================================================================
--- include/wx/docview.h	(revision 64441)
+++ include/wx/docview.h	(working copy)
@@ -929,7 +929,7 @@
 class WXDLLIMPEXP_CORE wxDocPrintout : public wxPrintout
 {
 public:
-    wxDocPrintout(wxView *view = NULL, const wxString& title = wxT("Printout"));
+    wxDocPrintout(wxView *view = NULL, const wxString& title = _("Printout"));
 
     // implement wxPrintout methods
     virtual bool OnPrintPage(int page);
Index: include/wx/prntbase.h
===================================================================
--- include/wx/prntbase.h	(revision 64441)
+++ include/wx/prntbase.h	(working copy)
@@ -244,7 +244,7 @@
 class WXDLLIMPEXP_CORE wxPrintout: public wxObject
 {
 public:
-    wxPrintout(const wxString& title = wxT("Printout"));
+    wxPrintout(const wxString& title = _("Printout"));
     virtual ~wxPrintout();
 
     virtual bool OnBeginDocument(int startPage, int endPage);
@@ -376,7 +376,7 @@
 public:
     wxPreviewFrame(wxPrintPreviewBase *preview,
                    wxWindow *parent,
-                   const wxString& title = wxT("Print Preview"),
+                   const wxString& title = _("Print Preview"),
                    const wxPoint& pos = wxDefaultPosition,
                    const wxSize& size = wxDefaultSize,
                    long style = wxDEFAULT_FRAME_STYLE | wxFRAME_FLOAT_ON_PARENT,
