Index: D:/wxWidgets-2.8.8/src/msw/ole/droptgt.cpp
===================================================================
--- D:/wxWidgets-2.8.8/src/msw/ole/droptgt.cpp	(revision 38)
+++ D:/wxWidgets-2.8.8/src/msw/ole/droptgt.cpp	(working copy)
@@ -245,9 +245,12 @@
         wxLogLastError(wxT("ScreenToClient"));
     }
 
-    *pdwEffect = ConvertDragResultToEffect(
-                    m_pTarget->OnDragOver(pt.x, pt.y, result)
-                 );
+    if (result == wxDragNone)
+        *pdwEffect = DROPEFFECT_NONE;
+    else
+        *pdwEffect = ConvertDragResultToEffect(
+                        m_pTarget->OnDragOver(pt.x, pt.y, result)
+                     );
 
     return S_OK;
 }
