Index: text.cpp
===================================================================
--- text.cpp	(revision 76890)
+++ text.cpp	(working copy)
@@ -351,7 +351,10 @@
     }
 
     MyPanel *m_panel;
+    wxTimer m_timer;
 
+    void OnTimer(wxTimerEvent& ev);
+
     wxDECLARE_EVENT_TABLE();
 };
 
@@ -1190,11 +1193,24 @@
     wxBoxSizer *row1 = new wxBoxSizer(wxHORIZONTAL);
     row1->Add( column1, 0, wxALL | wxEXPAND, 10 );
     row1->Add( column2, 1, wxALL | wxEXPAND, 10 );
-    row1->Add( m_textrich, 1, wxALL | wxEXPAND, 10 );
 
+    wxBoxSizer *foo = new wxBoxSizer(wxVERTICAL);
+    wxButton* b = new wxButton( this, wxID_OK, wxS("OK"), wxDefaultPosition, wxDefaultSize);
+    b->SetDefault();
+    foo->Add(b);
+
+    // BUG: If the changing text control is very close to a
+    // default button, the program will crash after a while.
+    // Change border to e.g. 10 and it no longer crashes.
+    int border = 0;
+    foo->Add(m_textrich, 1, wxALL | wxEXPAND, border );
+
+    row1->Add( foo );
+
     wxBoxSizer *topSizer = new wxBoxSizer(wxVERTICAL);
     topSizer->Add( row1, 2, wxALL | wxEXPAND, 10 );
 
+
 #if wxUSE_LOG
     topSizer->Add( m_log, 1, wxALL | wxEXPAND, 10 );
 #endif
@@ -1369,6 +1385,7 @@
 //----------------------------------------------------------------------
 
 wxBEGIN_EVENT_TABLE(MyFrame, wxFrame)
+    EVT_TIMER(wxID_ANY, MyFrame::OnTimer)
     EVT_MENU(TEXT_QUIT,   MyFrame::OnQuit)
     EVT_MENU(TEXT_ABOUT,  MyFrame::OnAbout)
     EVT_MENU(TEXT_SAVE,   MyFrame::OnFileSave)
@@ -1435,6 +1452,9 @@
 #endif // wxUSE_STATUSBAR
 
     m_panel = new MyPanel( this, 10, 10, 300, 100 );
+
+    m_timer.SetOwner(this);
+    m_timer.Start(1);
 }
 
 void MyFrame::OnQuit (wxCommandEvent& WXUNUSED(event) )
@@ -1941,3 +1961,10 @@
         m_currentPosition = insertionPoint;
     }
 }
+
+void MyFrame::OnTimer(wxTimerEvent& event)
+{
+    m_panel->m_textrich->Remove(0, 10);
+    m_panel->m_textrich->SetInsertionPointEnd();
+    m_panel->m_textrich->WriteText(wxS("123456789\n"));
+}
