Index: src/mac/carbon/menu.cpp
===================================================================
--- src/mac/carbon/menu.cpp	(revision 49603)
+++ src/mac/carbon/menu.cpp	(working copy)
@@ -184,8 +184,16 @@
 wxMenu::~wxMenu()
 {
     wxRemoveMacMenuAssociation( this ) ;
-    if (MAC_WXHMENU(m_hMenu))
-        ::DisposeMenu(MAC_WXHMENU(m_hMenu));
+
+    // Note: the native window handle isn't released if it's the
+    // Window menu's native handle because that handle is reused in
+    // multiple menu bars. This means that handle is never ever
+    // released, but the alternative would be reference counting which
+    // appears to be overkill given that the handle should be released
+    // automatically when the application exits anyway.
+    if (MAC_WXHMENU(m_hMenu) && m_hMenu != wxMenuBar::MacGetWindowMenuHMenu()) {
+       ::DisposeMenu(MAC_WXHMENU(m_hMenu));
+    }
 }
 
 void wxMenu::Break()
@@ -402,40 +410,42 @@
 // helper functions returning the mac menu position for a certain item, note that this is
 // mac-wise 1 - based, i.e. the first item has index 1 whereas on MSWin it has pos 0
 
+
+// XXX If there are multiple items with the same command ID (e.g. for doing
+// dynamic menu items), this will only find the first one.
 int wxMenu::MacGetIndexFromId( int id )
 {
-    size_t pos;
-    wxMenuItemList::compatibility_iterator node = GetMenuItems().GetFirst();
-    for ( pos = 0; node; pos++ )
-    {
-        if ( node->GetData()->GetId() == id )
-            break;
-
-        node = node->GetNext();
+    // Use the Mac menu to find the right item.
+    UInt32 command = wxIdToMacCommand(id);
+    MenuRef menuref = (MenuRef) GetHMenu();
+    UInt16 item_count = CountMenuItems(menuref);
+    for(UInt16 idx = 1; idx <= item_count; ++ idx) {
+        UInt32 other_command = 0;
+        GetMenuItemCommandID(menuref, idx, &other_command);
+        if(command == other_command) {
+            return idx;
+        }
     }
 
-    if (!node)
-        return 0;
-
-    return pos + 1 ;
+    // Didn't find it.
+    return 0;
 }
 
 int wxMenu::MacGetIndexFromItem( wxMenuItem *pItem )
 {
-    size_t pos;
-    wxMenuItemList::compatibility_iterator node = GetMenuItems().GetFirst();
-    for ( pos = 0; node; pos++ )
-    {
-        if ( node->GetData() == pItem )
-            break;
-
-        node = node->GetNext();
+    // Use the Mac menu to find the right item.
+    MenuRef menuref = (MenuRef) GetHMenu();
+    UInt16 item_count = CountMenuItems(menuref);
+    for(UInt16 idx = 1; idx <= item_count; ++ idx) {
+      UInt32 refcon = 0;
+      GetMenuItemRefCon(menuref, idx, &refcon);
+      if((wxMenuItem*) refcon == (wxMenuItem*) pItem) {
+          return idx;
+      }
     }
 
-    if (!node)
-        return 0;
-
-    return pos + 1 ;
+    // Didn't find it.
+    return 0;
 }
 
 void wxMenu::MacEnableMenu( bool bDoEnable )
@@ -795,6 +805,16 @@
             }
 
             wxInsertMenuItemsInMenu(menu, wm, winListStart);
+
+	    // Re-associate the wxMenu with the window menu, otherwise the
+	    // menu item indexes will be wrong.  This is safe since the Mac
+	    // menu items were copied above:
+            if( (MenuRef) menu->GetHMenu() != wm ) {
+                ::DisposeMenu(MAC_WXHMENU(menu->GetHMenu()));
+                menu->SetHMenu((WXHMENU) wm);
+                wxRemoveMacMenuAssociation( menu );
+                wxAssociateMenuWithMacMenu( wm , menu ) ;
+            }
         }
         else
         {

 	  	 
