Index: src/osx/choice_osx.cpp
===================================================================
--- src/osx/choice_osx.cpp	(revision 57954)
+++ src/osx/choice_osx.cpp	(working copy)
@@ -169,7 +169,7 @@
 // ----------------------------------------------------------------------------
 int wxChoice::GetSelection() const
 {
-    return m_peer->GetValue() -1;
+    return m_peer->GetValue();
 }
 
 void wxChoice::SetSelection( int n )
