Index: src/univ/combobox.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/univ/combobox.cpp,v
retrieving revision 1.59
diff -u -2 -r1.59 combobox.cpp
--- src/univ/combobox.cpp	2007/04/26 02:21:03	1.59
+++ src/univ/combobox.cpp	2007/05/10 19:51:06
@@ -268,4 +268,15 @@
 }
 
+void wxComboBox::OnSize( wxSizeEvent& event )
+{
+    // Without this, only the last letter of the item will initially be seen
+    if ( GetTextCtrl() ) GetTextCtrl()->ShowHorzPosition(0);
+    event.Skip();
+}
+
+BEGIN_EVENT_TABLE(wxComboBox, wxComboCtrl)
+    EVT_SIZE(wxComboBox::OnSize)
+END_EVENT_TABLE()
+
 // ----------------------------------------------------------------------------
 // wxComboBox methods forwarded to wxTextCtrl
Index: include/wx/univ/combobox.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/univ/combobox.h,v
retrieving revision 1.38
diff -u -2 -r1.38 combobox.h
--- include/wx/univ/combobox.h	2006/09/14 19:36:38	1.38
+++ include/wx/univ/combobox.h	2007/05/10 19:51:06
@@ -156,4 +156,5 @@
     // get the associated listbox
     wxListBox *GetLBox() const { return m_lbox; }
+    void OnSize( wxSizeEvent& event );
 
 private:
@@ -161,5 +162,5 @@
     wxListBox *m_lbox;
 
-    //DECLARE_EVENT_TABLE()
+    DECLARE_EVENT_TABLE()
     DECLARE_DYNAMIC_CLASS(wxComboBox)
 };

 	  	 
