Index: framemanager.cpp
===================================================================
--- framemanager.cpp	(révision 65666)
+++ framemanager.cpp	(copie de travail)
@@ -1571,6 +1571,12 @@
         p.SafeSet(pane);
     }
 
+	// ---- Restore the maximized state as it should be
+	m_has_maximized = false;
+    for (size_t pane_i = 0; pane_i < m_panes.GetCount(); ++pane_i)
+		if (m_panes.Item(pane_i).IsMaximized())
+			{ m_has_maximized = true; break; }
+
     if (update)
         Update();
 
