Index: src/msw/dialog.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/dialog.cpp,v
retrieving revision 1.96
diff -u -r1.96 dialog.cpp
--- src/msw/dialog.cpp	2003/09/24 00:26:31	1.96
+++ src/msw/dialog.cpp	2003/10/01 17:41:39
@@ -198,8 +198,8 @@
 
     if ( !wxTopLevelWindow::Create(parent, id, title, pos, size, style, name) )
         return FALSE;
-
-    SetFont(wxSystemSettings::GetFont(wxSYS_DEFAULT_GUI_FONT));
+    if (!m_hasFont)
+        SetFont(wxSystemSettings::GetFont(wxSYS_DEFAULT_GUI_FONT));
 
     return TRUE;
 }

 	  	 
