Index: src/msw/radiobut.cpp
===================================================================
--- src/msw/radiobut.cpp	(revision 51611)
+++ src/msw/radiobut.cpp	(working copy)
@@ -309,6 +309,11 @@
         hRadio = s_radioSize;
     }
 
+#ifdef __WXWINCE__
+    // The height calculated above clips the bottom pixel of the radio button
+    hRadio += 1;
+#endif
+
     wxSize best(wRadio, hRadio);
     CacheBestSize(best);
     return best;

 	  	 
