Index: src/common/filename.cpp
===================================================================
--- src/common/filename.cpp	(revision 62459)
+++ src/common/filename.cpp	(working copy)
@@ -158,28 +158,28 @@
 public:
     enum OpenMode
     {
-        Read,
-        Write
+        ReadAttr,
+        WriteAttr
     };
 
     wxFileHandle(const wxString& filename, OpenMode mode, int flags = 0)
     {
         m_hFile = ::CreateFile
                     (
-                     filename.fn_str(),             // name
-                     mode == Read ? GENERIC_READ    // access mask
-                                  : GENERIC_WRITE,
-                     FILE_SHARE_READ |              // sharing mode
-                     FILE_SHARE_WRITE,              // (allow everything)
-                     NULL,                          // no secutity attr
-                     OPEN_EXISTING,                 // creation disposition
-                     flags,                         // flags
-                     NULL                           // no template file
+                     filename.fn_str(),                         // name
+                     mode == ReadAttr ? FILE_READ_ATTRIBUTES    // access mask
+                                      : FILE_WRITE_ATTRIBUTES,
+                     FILE_SHARE_READ |                          // sharing mode
+                     FILE_SHARE_WRITE,                          // (allow everything)
+                     NULL,                                      // no secutity attr
+                     OPEN_EXISTING,                             // creation disposition
+                     flags,                                     // flags
+                     NULL                                       // no template file
                     );
 
         if ( m_hFile == INVALID_HANDLE_VALUE )
         {
-            if ( mode == Read )
+            if ( mode == ReadAttr )
             {
                 wxLogSysError(_("Failed to open '%s' for reading"),
                               filename.c_str());
@@ -2308,7 +2308,7 @@
         flags = 0;
     }
 
-    wxFileHandle fh(path, wxFileHandle::Write, flags);
+    wxFileHandle fh(path, wxFileHandle::WriteAttr, flags);
     if ( fh.IsOk() )
     {
         if ( ::SetFileTime(fh,
@@ -2392,7 +2392,7 @@
     }
     else // file
     {
-        wxFileHandle fh(GetFullPath(), wxFileHandle::Read);
+        wxFileHandle fh(GetFullPath(), wxFileHandle::ReadAttr);
         if ( fh.IsOk() )
         {
             ok = ::GetFileTime(fh,
@@ -2462,7 +2462,7 @@
     // TODO
     return wxInvalidSize;
 #elif defined(__WIN32__)
-    wxFileHandle f(filename, wxFileHandle::Read);
+    wxFileHandle f(filename, wxFileHandle::ReadAttr);
     if (!f.IsOk())
         return wxInvalidSize;
 
