--- grid.cpp	2008-06-12 15:12:44.000000000 -0700
+++ grid_after.cpp	2008-06-12 15:23:35.000000000 -0700
@@ -2592,11 +2592,20 @@
 
 void wxGridCellAttrData::SetAttr(wxGridCellAttr *attr, int row, int col)
 {
+    // If you pass in a NULL for attr, then:
+    //        when there was a prior attr for this cell,
+    //            then the prior attr is REMOVED, DecRef'ed, and not replaced.
+    //        when there was no prior attr, then we do nothing,
+    //            (meaning that this cell STILL has no attr).
+
     int n = FindIndex(row, col);
     if ( n == wxNOT_FOUND )
     {
-        // add the attribute
-        m_attrs.Add(new wxGridCellWithAttr(row, col, attr));
+        if ( attr )
+        {
+            // add the attribute
+            m_attrs.Add(new wxGridCellWithAttr(row, col, attr));
+        }
     }
     else
     {
