diff -ru2 -xCVS -x.#* -x*.orig -x*.rej -x*~ --ignore-matching-lines=RCS-ID wxWidgets-2.5.3.orig/samples/minimal/minimal.cpp wxWidgets-2.5.3/samples/minimal/minimal.cpp
--- wxWidgets-2.5.3.orig/samples/minimal/minimal.cpp	Thu Sep 30 03:33:11 2004
+++ wxWidgets-2.5.3/samples/minimal/minimal.cpp	Sun May 22 23:20:02 2005
@@ -29,4 +29,5 @@
 #ifndef WX_PRECOMP
     #include "wx/wx.h"
+	#include "wx/glcanvas.h"
 #endif
 
@@ -113,4 +114,35 @@
 // ============================================================================
 
+class MyCanvas : public wxGLCanvas
+{
+public:
+	void OnKey(wxKeyEvent& WXUNUSED(event));
+	void OnDraw(wxDC& dc);
+	MyCanvas(wxWindow* pParent) : wxGLCanvas(pParent, -1) {};
+private:
+DECLARE_EVENT_TABLE()
+};
+
+BEGIN_EVENT_TABLE(MyCanvas, wxGLCanvas)
+	EVT_CHAR(MyCanvas::OnKey)
+END_EVENT_TABLE()
+
+void MyCanvas::OnKey(wxKeyEvent& WXUNUSED(event))
+{
+	wxClientDC dc(this);
+	OnDraw(dc);
+}
+
+void MyCanvas::OnDraw(wxDC& dc)
+{
+	dc.SetBrush(*wxWHITE_BRUSH);
+	dc.SetPen(*wxBLACK_PEN);
+	// Draw an X
+	dc.DrawLine(0,0,99,99);
+	dc.DrawLine(99,0,0,99);
+	// draw a square
+	dc.DrawRectangle(33,33,33,33);
+}
+
 // ----------------------------------------------------------------------------
 // the application class
@@ -123,8 +155,10 @@
     MyFrame *frame = new MyFrame(_T("Minimal wxWidgets App"));
 
+	MyCanvas *canvas = new MyCanvas(frame);
+
     // and show it (the frames, unlike simple controls, are not shown when
     // created initially)
     frame->Show(true);
-
+    
     // success: wxApp::OnRun() will be called which will enter the main message
     // loop and the application will run. If we returned false here, the

 	  	 
