Index: src/common/utilscmn.cpp
===================================================================
--- src/common/utilscmn.cpp	(revision 3371)
+++ src/common/utilscmn.cpp	(working copy)
@@ -551,6 +551,9 @@
     wxCHECK_MSG( map, false, wxS("output pointer can't be NULL") );
 
 #if defined(__VISUALC__)
+    if (!_tenviron)
+        _tgetenv(_T("TEMP")); // forces _tenviron to be initialized
+
     wxChar **env = _tenviron;
 #elif defined(__VMS)
    // Now this routine wil give false for OpenVMS
