Index: samples/event/event.cpp
===================================================================
--- samples/event/event.cpp	(revision 58686)
+++ samples/event/event.cpp	(working copy)
@@ -302,8 +302,10 @@
 {
     if ( event.IsChecked() )
     {
-        Connect(Event_Dynamic, wxEVT_COMMAND_MENU_SELECTED,
-                wxCommandEventHandler(MyFrame::OnDynamic));
+        //Connect(Event_Dynamic, wxEVT_COMMAND_MENU_SELECTED,
+        //        wxCommandEventHandler(MyFrame::OnDynamic));
+        Connect(wxID_ANY, Event_Dynamic, (wxObjectEventFunction)(wxEventFunction)
+                &MyFrame::OnDynamic);
 
 #if wxUSE_STATUSBAR
         SetStatusText(_T("You can now use \"Dynamic\" item in the menu"));
