Index: radiobox.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/radiobox.cpp,v
retrieving revision 1.69
diff -b -u -2 -r1.69 radiobox.cpp
--- radiobox.cpp	2003/03/17 11:21:54	1.69
+++ radiobox.cpp	2003/06/23 20:28:06
@@ -349,5 +349,4 @@
 
     ::SendMessage((HWND)m_radioButtons[N], BM_SETCHECK, 1, 0L);
-    ::SetFocus((HWND)m_radioButtons[N]);
 
     m_selectedButton = N;
@@ -718,4 +717,5 @@
 {
     SetSelection (event.m_commandInt);
+    SetFocus();
     ProcessCommand (event);
 }
@@ -955,4 +955,5 @@
                     {
                         radiobox->SetSelection(selNew);
+                        radiobox->SetFocus();
 
                         // emulate the button click

 	  	 
