Index: grid.cpp
===================================================================
--- grid.cpp	(revision 53959)
+++ grid.cpp	(working copy)
@@ -9881,9 +9880,16 @@
 {
     if ( m_attrCache.row != -1 )
     {
-        wxSafeDecRef(m_attrCache.attr);
+        wxGridCellAttr *oldAttr = m_attrCache.attr;
+        // mark m_attrCache as invalid _before_
+        // wxSafeDecRef(...)
         m_attrCache.attr = NULL;
         m_attrCache.row = -1;
+        // wxSafeDecRec(...) might cause event processing that
+        // accesses the cached attribute, if one exists.
+        // Therefore it is important to invalidate the cache
+        // before calling wxSafeDecRef
+        wxSafeDecRef(oldAttr);
     }
 }
 
