Index: src/msw/ole/dataobj.cpp
===================================================================
--- src/msw/ole/dataobj.cpp	(revision 48457)
+++ src/msw/ole/dataobj.cpp	(working copy)
@@ -1139,7 +1139,7 @@
 #endif // wxUSE_UNICODE_MSLU
         {
             len = m_filenames[i].length();
-            memcpy(pbuf, m_filenames[i].c_str(), len*sizeOfChar);
+            memcpy(pbuf, m_filenames[i].wc_str(), len*sizeOfChar);
         }
 
         pbuf += len*sizeOfChar;

 	  	 
