--- src/aui/framemanager.cpp.orig	2007-05-17 14:16:42.000000000 -0700
+++ src/aui/framemanager.cpp	2007-05-17 14:17:38.000000000 -0700
@@ -3551,6 +3551,11 @@
 
 void wxAuiManager::OnRender(wxAuiManagerEvent& evt)
 {
+    // if the frame is about to be deleted, don't bother
+    if(!m_frame || wxPendingDelete.Member(m_frame)) {
+	return;
+    }
+    
     wxDC* dc = evt.GetDC();
 
 #ifdef __WXMAC__

 	  	 
