Index: samples/minimal/minimal.cpp
===================================================================
--- samples/minimal/minimal.cpp	(revision 77517)
+++ samples/minimal/minimal.cpp	(working copy)
@@ -19,6 +19,9 @@
 
 // For compilers that support precompilation, includes "wx/wx.h".
 #include "wx/wxprec.h"
+// patched <
+#include "wx/aui/aui.h"
+// > patched
 
 #ifdef __BORLANDC__
     #pragma hdrstop
@@ -63,6 +66,9 @@
 public:
     // ctor(s)
     MyFrame(const wxString& title);
+    // patched <
+    ~MyFrame();
+    // > patched
 
     // event handlers (these functions should _not_ be virtual)
     void OnQuit(wxCommandEvent& event);
@@ -69,10 +75,37 @@
     void OnAbout(wxCommandEvent& event);
 
 private:
+    // patched <
+    wxAuiManager m_Mgr;
+    // > patched
+
     // any class wishing to process wxWidgets events must use this macro
     wxDECLARE_EVENT_TABLE();
 };
 
+// patched <
+class Test: public wxControl {
+public:
+    Test(wxWindow *parent, wxWindowID id, const wxPoint& pos = wxDefaultPosition, const wxSize& size = wxDefaultSize): wxControl(parent, id, pos, size, wxBORDER_THEME, wxDefaultValidator, wxControlNameStr) {}
+    ~Test() {}
+
+private:
+    void OnCreate(wxWindowCreateEvent& event)
+    {
+        wxClientDC DC(this);
+        DC.DrawRectangle(wxPoint(0, 0), GetClientSize());
+    }
+
+    DECLARE_EVENT_TABLE()
+};
+
+BEGIN_EVENT_TABLE(Test, wxControl)
+
+    EVT_WINDOW_CREATE(Test::OnCreate)
+
+END_EVENT_TABLE()
+// > patched
+
 // ----------------------------------------------------------------------------
 // constants
 // ----------------------------------------------------------------------------
@@ -145,6 +178,11 @@
 MyFrame::MyFrame(const wxString& title)
        : wxFrame(NULL, wxID_ANY, title)
 {
+    // patched <
+    m_Mgr.SetManagedWindow(this);
+    m_Mgr.SetFlags(wxAUI_MGR_DEFAULT);
+    // > patched
+
     // set the frame icon
     SetIcon(wxICON(sample));
 
@@ -172,9 +210,21 @@
     CreateStatusBar(2);
     SetStatusText("Welcome to wxWidgets!");
 #endif // wxUSE_STATUSBAR
+
+    // patched <
+    Test* pTest = new Test(this, wxID_ANY, wxPoint(0, 0), wxSize(100, 100));
+    m_Mgr.AddPane(pTest, wxAuiPaneInfo().CaptionVisible(false).Row(0).Position(0).Layer(0).Top().Floatable(false));
+    m_Mgr.Update();
+    // > patched
 }
 
+// patched <
+MyFrame::~MyFrame() {
+    m_Mgr.UnInit();
+}
+// > patched
 
+
 // event handlers
 
 void MyFrame::OnQuit(wxCommandEvent& WXUNUSED(event))
