diff --git a/samples/controls/controls.cpp b/samples/controls/controls.cpp
index 24482f3..f4dd380 100644
--- a/samples/controls/controls.cpp
+++ b/samples/controls/controls.cpp
@@ -1266,7 +1266,7 @@ void MyPanel::OnListBox( wxCommandEvent &event )
                                                      : m_listboxSorted;
 
     bool deselect = false;
-    if (listbox->HasFlag(wxLB_MULTIPLE) || listbox->HasFlag(wxLB_EXTENDED))
+    if (listbox->HasMultipleSelection())
     {
         deselect = !event.IsSelection();
         if (deselect)
@@ -1279,7 +1279,7 @@ void MyPanel::OnListBox( wxCommandEvent &event )
 
     // can't use GetStringSelection() with multiple selections, there could be
     // more than one of them
-    if ( !listbox->HasFlag(wxLB_MULTIPLE) && !listbox->HasFlag(wxLB_EXTENDED) )
+    if (!listbox->HasMultipleSelection())
     {
         m_text->AppendText( wxT("ListBox control selection string is: '") );
         m_text->AppendText( listbox->GetStringSelection() );
@@ -1364,9 +1364,16 @@ void MyPanel::OnListBoxButtons( wxCommandEvent &event )
         case ID_LISTBOX_DELETE:
             {
                 int idx;
-                idx = m_listbox->GetSelection();
-                if ( idx != wxNOT_FOUND )
-                    m_listbox->Delete( idx );
+                
+                //GetSelection does not work, use GetSelections
+                wxArrayInt selects;
+                m_listbox->GetSelections(selects);
+                
+                for(wxArrayInt::reverse_iterator it = selects.rbegin(); it != selects.rend(); ++it )
+                {
+                    m_listbox->Delete( *it );
+                }
+                
                 idx = m_listboxSorted->GetSelection();
                 if ( idx != wxNOT_FOUND )
                     m_listboxSorted->Delete( idx );
@@ -1397,8 +1404,11 @@ void MyPanel::OnChoice( wxCommandEvent &event )
 
     const int sel = choice->GetSelection();
 
-    wxClientData *dataEvt = event.GetClientObject(),
-                 *dataCtrl = choice->GetClientObject(sel);
+    wxClientData* dataEvt = event.GetClientObject();
+    wxClientData* dataCtrl = NULL;
+
+    if(choice->HasClientData())
+        dataCtrl = choice->GetClientObject(sel);
 
     wxLogMessage(wxT("EVT_CHOICE: item %d/%d (event/control), ")
                  wxT("string \"%s\"/\"%s\", ")
