Index: src/osx/cocoa/nonownedwnd.mm
===================================================================
--- src/osx/cocoa/nonownedwnd.mm	(revision 63402)
+++ src/osx/cocoa/nonownedwnd.mm	(working copy)
@@ -148,6 +148,7 @@
     wxNonOwnedWindowCocoaImpl* impl;
 }
 
+- (NSRect)constrainFrameRect:(NSRect)frameRect toScreen:(NSScreen *)screen;
 - (void)setImplementation: (wxNonOwnedWindowCocoaImpl *) theImplementation;
 - (wxNonOwnedWindowCocoaImpl*) implementation;
 - (void)noResponderFor: (SEL) selector;
@@ -155,6 +156,14 @@
 
 @implementation wxNSPanel
 
+- (NSRect)constrainFrameRect:(NSRect)frameRect toScreen:(NSScreen *)screen
+{
+    if (impl->IsFullScreen())
+        return frameRect;
+    else
+        return [super constrainFrameRect:frameRect toScreen:screen];
+}
+
 - (void)setImplementation: (wxNonOwnedWindowCocoaImpl *) theImplementation
 {
     impl = theImplementation;
@@ -677,7 +686,14 @@
         data->m_formerFrame = [m_macWindow frame];
         CGDisplayCapture( kCGDirectMainDisplay );
         [m_macWindow setLevel:CGShieldingWindowLevel()];
-        [m_macWindow setFrame:[[NSScreen mainScreen] frame] display:YES];
+        NSRect screenframe = [[NSScreen mainScreen] frame];
+        NSRect frame = NSMakeRect (0, 0, 100, 100);
+        NSRect contentRect;
+        contentRect = [NSWindow contentRectForFrameRect: frame
+                                styleMask: NSTitledWindowMask];
+        screenframe.origin.y += (frame.origin.y - contentRect.origin.y);
+        screenframe.size.height += (frame.size.height - contentRect.size.height);
+        [m_macWindow setFrame:screenframe display:YES];
     }
     else if ( m_macFullScreenData != NULL )
     {
