diff -bu1 -r wxWindows/samples/drawing/drawing.cpp wxWin/samples/drawing/drawing.cpp
--- wxWindows/samples/drawing/drawing.cpp	2003-06-07 14:19:28.000000000 +0200
+++ wxWin/samples/drawing/drawing.cpp	2003-11-21 17:37:20.000000000 +0100
@@ -152,3 +152,3 @@
 
-    void Show(ScreenToShow show) { m_show = show; Refresh(); }
+    void ToShow(ScreenToShow show) { m_show = show; Refresh(); }
 
@@ -1143,3 +1143,3 @@
 {
-    m_canvas->Show((ScreenToShow)(event.GetId() - MenuShow_First));
+    m_canvas->ToShow((ScreenToShow)(event.GetId() - MenuShow_First));
 }
diff -bu1 -r wxWindows/samples/internat/internat.cpp wxWin/samples/internat/internat.cpp
--- wxWindows/samples/internat/internat.cpp	2003-11-20 08:16:59.000000000 +0100
+++ wxWin/samples/internat/internat.cpp	2003-11-21 18:08:02.000000000 +0100
@@ -339,4 +339,3 @@
     wxTextEntryDialog d(this,
-	    _("Please enter range for plural forms of \"n files deleted\""
-		"phrase"),
+	    _("Please enter range for plural forms of \"n files deleted\" phrase"),
 	    title, _T("0-10"));
@@ -351,3 +350,3 @@
         	    s << n << _T(" ") << 
-                     wxGetTranslation("file deleted", "files deleted", n) << _T("\n");
+                     wxGetTranslation(_T("file deleted"), _T("files deleted"), n) << _T("\n");
     	}
diff -bu1 -r wxWindows/samples/keyboard/keyboard.cpp wxWin/samples/keyboard/keyboard.cpp
--- wxWindows/samples/keyboard/keyboard.cpp	2003-09-09 23:23:54.000000000 +0200
+++ wxWin/samples/keyboard/keyboard.cpp	2003-11-21 18:23:21.000000000 +0100
@@ -354,3 +354,3 @@
     wxString key;
-    long keycode = event.KeyCode();
+    long keycode = event.GetKeyCode();
     switch ( keycode )
diff -bu1 -r wxWindows/samples/listbox/lboxtest.cpp wxWin/samples/listbox/lboxtest.cpp
--- wxWindows/samples/listbox/lboxtest.cpp	2003-09-09 23:23:57.000000000 +0200
+++ wxWin/samples/listbox/lboxtest.cpp	2003-11-21 17:24:18.000000000 +0100
@@ -40,2 +40,3 @@
 #ifndef WX_PRECOMP
+    #include "wx/wx.h"
     #include "wx/app.h"
@@ -259,7 +260,2 @@
 
-#ifdef __WXUNIVERSAL__
-    WX_USE_THEME(win32);
-    WX_USE_THEME(gtk);
-#endif // __WXUNIVERSAL__
-
 // ----------------------------------------------------------------------------
@@ -672,3 +668,3 @@
     int sel = event.GetInt();
-    m_textDelete->SetValue(wxString::Format(_T("%ld"), sel));
+    m_textDelete->SetValue(wxString::Format(_T("%d"), sel));
 
@@ -679,3 +675,4 @@
 {
-    wxLogMessage(_T("Listbox item %d double clicked"), event.GetInt());
+    int sel = event.GetInt();
+    wxLogMessage(_T("Listbox item %d double clicked"), sel);
 }
diff -bu1 -r wxWindows/samples/mobile/wxedit/wxedit.cpp wxWin/samples/mobile/wxedit/wxedit.cpp
--- wxWindows/samples/mobile/wxedit/wxedit.cpp	2003-08-14 20:35:49.000000000 +0200
+++ wxWin/samples/mobile/wxedit/wxedit.cpp	2003-11-21 12:15:39.000000000 +0100
@@ -154,3 +154,3 @@
 
-void MyFrame::OnCopy( wxCommandEvent &event )
+void MyFrame::OnCopy( wxCommandEvent& WXUNUSED(event) )
 {
@@ -158,3 +158,3 @@
 
-void MyFrame::OnCut( wxCommandEvent &event )
+void MyFrame::OnCut( wxCommandEvent& WXUNUSED(event) )
 {
@@ -162,3 +162,3 @@
 
-void MyFrame::OnPaste( wxCommandEvent &event )
+void MyFrame::OnPaste( wxCommandEvent& WXUNUSED(event) )
 {
@@ -166,3 +166,3 @@
 
-void MyFrame::OnDelete( wxCommandEvent &event )
+void MyFrame::OnDelete( wxCommandEvent& WXUNUSED(event) )
 {
@@ -179,4 +179,4 @@
     
-    wxASSERT( index < m_history.GetCount() );
-    
+    if( index < m_history.GetCount() )
+    {
     m_filename = m_history[index];
@@ -187,5 +187,15 @@
     SetStatusText( m_filename );
+    }
+    else
+    {
+        wxMessageBox(
+            _T("This entry is empty. It should be filled once you will start opening."),
+            _T("Empty entry"),
+            wxOK | wxICON_INFORMATION,
+            this
+        );
+    }
 }
 
-void MyFrame::OnNew( wxCommandEvent &event )
+void MyFrame::OnNew( wxCommandEvent& WXUNUSED(event) )
 {
@@ -203,3 +213,3 @@
 
-void MyFrame::OnOpen( wxCommandEvent &event )
+void MyFrame::OnOpen( wxCommandEvent& WXUNUSED(event) )
 {
@@ -248,3 +258,3 @@
 
-void MyFrame::OnSave( wxCommandEvent &event )
+void MyFrame::OnSave( wxCommandEvent& WXUNUSED(event) )
 {
@@ -253,3 +263,3 @@
 
-void MyFrame::OnSaveAs( wxCommandEvent &event )
+void MyFrame::OnSaveAs( wxCommandEvent& WXUNUSED(event) )
 {
@@ -267,3 +277,3 @@
 
-void MyFrame::OnAbout( wxCommandEvent &event )
+void MyFrame::OnAbout( wxCommandEvent& WXUNUSED(event) )
 {
@@ -274,3 +284,3 @@
 
-void MyFrame::OnQuit( wxCommandEvent &event )
+void MyFrame::OnQuit( wxCommandEvent& WXUNUSED(event) )
 {
@@ -342,3 +352,3 @@
 
-void MyFrame::OnCloseWindow( wxCloseEvent &event )
+void MyFrame::OnCloseWindow( wxCloseEvent& WXUNUSED(event) )
 {
diff -bu1 -r wxWindows/samples/notebook/notebook.cpp wxWin/samples/notebook/notebook.cpp
--- wxWindows/samples/notebook/notebook.cpp	2003-08-22 02:12:41.000000000 +0200
+++ wxWin/samples/notebook/notebook.cpp	2003-11-21 17:47:20.000000000 +0100
@@ -163,7 +163,5 @@
 {
-    wxPanel *panel = (wxPanel *) NULL;
-
     // Create and add some panels to the notebook
 
-    panel = CreateRadioButtonsPage();
+    wxPanel *panel = CreateRadioButtonsPage();
     AddPage( panel, RADIOBUTTONS_PAGE_NAME, FALSE, GetIconIndex() );
diff -bu1 -r wxWindows/samples/treectrl/treetest.cpp wxWin/samples/treectrl/treetest.cpp
--- wxWindows/samples/treectrl/treetest.cpp	2003-10-24 23:08:12.000000000 +0200
+++ wxWin/samples/treectrl/treetest.cpp	2003-11-21 18:31:20.000000000 +0100
@@ -48,3 +48,6 @@
 #include "icon5.xpm"
+
+#if defined(__WXGTK__) || defined(__WXX11__) || defined(__WXMOTIF__) || defined(__WXMAC__) || defined(__WXMGL__)
 #include "mondrian.xpm"
+#endif
 
diff -bu1 -r wxWindows/samples/typetest/typetest.cpp wxWin/samples/typetest/typetest.cpp
--- wxWindows/samples/typetest/typetest.cpp	2003-08-14 23:27:05.000000000 +0200
+++ wxWin/samples/typetest/typetest.cpp	2003-11-21 19:51:26.000000000 +0100
@@ -729,3 +729,2 @@
     char ch = 0;
-    size_t pos = 0;
     wxString str;
@@ -745,3 +744,3 @@
     ch = file_input.GetC();
-    pos = file_input.TellI();
+    size_t pos = file_input.TellI();
     str.Printf( wxT("Read char: %d. Now at position %d\n\n"), (int) ch, (int) pos );
@@ -809,3 +808,2 @@
     char ch = 0;
-    size_t pos = 0;
     wxString str;
@@ -826,3 +824,3 @@
     ch = buf_input.GetC();
-    pos = buf_input.TellI();
+    size_t pos = buf_input.TellI();
     str.Printf( wxT("Read char: %d. Now at position %d\n\n"), (int) ch, (int) pos );
@@ -978,6 +976,7 @@
 
-    if (wxBYTE_ORDER == wxLITTLE_ENDIAN)
+    #if wxBYTE_ORDER == wxLITTLE_ENDIAN
         textCtrl << _T("This is a little endian system.\n\n");
-    else
+    #else
         textCtrl << _T("This is a big endian system.\n\n");
+    #endif
 
Only in wxWin/samples/typetest: typetest.err
diff -bu1 -r wxWindows/samples/vscroll/vstest.cpp wxWin/samples/vscroll/vstest.cpp
--- wxWindows/samples/vscroll/vstest.cpp	2003-09-09 23:24:17.000000000 +0200
+++ wxWin/samples/vscroll/vstest.cpp	2003-11-21 18:37:39.000000000 +0100
@@ -29,2 +29,3 @@
 #ifndef WX_PRECOMP
+    #include "wx/wx.h"
     #include "wx/app.h"
diff -bu1 -r wxWindows/samples/widgets/checkbox.cpp wxWin/samples/widgets/checkbox.cpp
--- wxWindows/samples/widgets/checkbox.cpp	2003-09-28 16:53:07.000000000 +0200
+++ wxWin/samples/widgets/checkbox.cpp	2003-11-21 16:25:03.000000000 +0100
@@ -150,4 +150,4 @@
 
-    wxCHECK_RET( (state >= 0) && (state < WXSIZEOF(stateNames)),
-        "event.GetInt() returned an invalid wxCheckBoxState" );
+    wxCHECK_RET( (state >= (wxCheckBoxState)0) && (state < (wxCheckBoxState)WXSIZEOF(stateNames)),
+        _T("event.GetInt() returned an invalid wxCheckBoxState") );
 
diff -bu1 -r wxWindows/samples/widgets/radiobox.cpp wxWin/samples/widgets/radiobox.cpp
--- wxWindows/samples/widgets/radiobox.cpp	2003-07-31 16:59:28.000000000 +0200
+++ wxWin/samples/widgets/radiobox.cpp	2003-11-21 10:58:14.000000000 +0100
@@ -387,2 +387,4 @@
     int sel = m_radio->GetSelection();
+    int event_sel = event.GetSelection();
+    wxUnusedVar(event_sel);
 
@@ -390,3 +392,3 @@
 
-    wxASSERT_MSG( sel == event.GetSelection(),
+    wxASSERT_MSG( sel == event_sel,
                   _T("selection should be the same in event and radiobox") );
diff -bu1 -r wxWindows/samples/wxtest/test_wdr.h wxWin/samples/wxtest/test_wdr.h
--- wxWindows/samples/wxtest/test_wdr.h	2002-04-26 22:38:48.000000000 +0200
+++ wxWin/samples/wxtest/test_wdr.h	2003-11-21 19:09:07.000000000 +0100
@@ -26,3 +26,2 @@
 #include <wx/notebook.h>
-#include <wx/grid.h>
 

 	  	 
