--- sizer-o.cpp	2007-03-05 15:32:12.562500000 +0100
+++ sizer.cpp	2007-03-05 15:24:20.906250000 +0100
@@ -110,11 +110,17 @@
     m_kind = Item_None;
 }
 
+void wxSizerItem::Release( ){ 
+    if( m_kind==Item_Window && m_window ) 
+        m_window->SetContainingSizer(NULL); 
+} 
+
 // window item
 void wxSizerItem::SetWindow(wxWindow *window)
 {
     wxCHECK_RET( window, _T("NULL window in wxSizerItem::SetWindow()") );
 
+    Release( );
     m_kind = Item_Window;
     m_window = window;
 
@@ -144,6 +150,7 @@
 // sizer item
 void wxSizerItem::SetSizer(wxSizer *sizer)
 {
+    Release( );
     m_kind = Item_Sizer;
     m_sizer = sizer;
 }
@@ -167,6 +174,7 @@
 // spacer item
 void wxSizerItem::SetSpacer(const wxSize& size)
 {
+    Release( );
     m_kind = Item_Spacer;
     m_spacer = new wxSizerSpacer(size);
     m_minSize = size;
 
--- sizer-o.h	2007-03-05 15:32:28.687500000 +0100
+++ sizer.h	2007-03-05 14:53:11.953125000 +0100
@@ -331,6 +331,10 @@
     void SetSpacer(int width, int height) { SetSpacer(wxSize(width, height)); }
 
 protected:
+	
+	// Before setting s sizeritem to something else, release previous (windows) item
+    void Release( );
+    
     // common part of several ctors
     void Init() { m_userData = NULL; }
      

 	  	 
