Index: slider.cpp
===================================================================
RCS file: /home/wxcvs/wxWindows/src/gtk/slider.cpp,v
retrieving revision 1.45
diff -b -u -2 -r1.45 slider.cpp
--- slider.cpp	9 Jul 2002 21:01:04 -0000	1.45
+++ slider.cpp	2 Aug 2002 09:38:28 -0000
@@ -132,8 +132,8 @@
     m_adjust = gtk_range_get_adjustment( GTK_RANGE(m_widget) );
 
+    GtkEnableEvents();
+
     SetRange( minValue, maxValue );
     SetValue( value );
-
-    GtkEnableEvents();
 
     m_parent->DoAddChild( this );

 	  	 
