--- wxWidgets-2.6.2/src/msw/clipbrd.cpp	Mon Mar 21 11:20:06 2005
+++ wx_2.6.2/src/msw/clipbrd.cpp	Thu Feb 23 13:28:31 2006
@@ -529,6 +529,8 @@
 
 IMPLEMENT_DYNAMIC_CLASS(wxClipboard, wxObject)
 
+static IDataObject *gLastDataObject = NULL;
+
 wxClipboard::wxClipboard()
 {
 #if wxUSE_OLE_CLIPBOARD
@@ -554,10 +556,17 @@
 void wxClipboard::Clear()
 {
 #if wxUSE_OLE_CLIPBOARD
-    HRESULT hr = OleSetClipboard(NULL);
-    if ( FAILED(hr) )
+    if (gLastDataObject)
     {
-        wxLogApiError(wxT("OleSetClipboard(NULL)"), hr);
+        if (S_OK == OleIsCurrentClipboard(gLastDataObject))
+        {
+            HRESULT hr = OleSetClipboard(NULL);
+            if ( FAILED(hr) )
+            {
+                wxLogApiError(wxT("OleSetClipboard(NULL)"), hr);
+            }
+        }
+        gLastDataObject = NULL;
     }
 #endif // wxUSE_OLE_CLIPBOARD
 }
@@ -565,19 +574,18 @@
 bool wxClipboard::Flush()
 {
 #if wxUSE_OLE_CLIPBOARD
-    HRESULT hr = OleFlushClipboard();
-    if ( FAILED(hr) )
+    if (gLastDataObject && S_OK == OleIsCurrentClipboard(gLastDataObject))
     {
-        wxLogApiError(wxT("OleFlushClipboard"), hr);
+        HRESULT hr = OleFlushClipboard();
+        if ( FAILED(hr) )
+        {
+            wxLogApiError(wxT("OleFlushClipboard"), hr);
 
-        return false;
-    }
-    else
-    {
-        m_clearOnExit = false;
-
-        return true;
+            return false;
+        }
     }
+    gLastDataObject = NULL;
+    return true;
 #else // !wxUSE_OLE_CLIPBOARD
     return false;
 #endif // wxUSE_OLE_CLIPBOARD/!wxUSE_OLE_CLIPBOARD
@@ -629,6 +637,11 @@
 
         return false;
     }
+
+    // used to track whether the data in the clipboard is still this data
+    // otherwise don't call either OleSetClipboard(NULL) nor 
+    // OleFlushClipboard() as it will interfere with other applications
+    gLastDataObject = data->GetInterface();
 
     // we have a problem here because we should delete wxDataObject, but we
     // can't do it because IDataObject which we just gave to the clipboard

 	  	 
