Index: samples/controls/controls.cpp
===================================================================
--- samples/controls/controls.cpp	(revisione 67291)
+++ samples/controls/controls.cpp	(copia locale)
@@ -901,6 +901,11 @@
                           );
     non_wrapping_sizer->Add( m_nonWrappingText );
 
+    wxRadioButton *rb1 = new wxRadioButton( panel, wxID_ANY, _("Radio 1"), wxDefaultPosition, wxDefaultSize, 0 );
+    non_wrapping_sizer->Add( rb1 );
+    wxRadioButton *rb2 = new wxRadioButton( panel, wxID_ANY, _("Radio 2"), wxDefaultPosition, wxDefaultSize, 0 );
+    non_wrapping_sizer->Add( rb2 );
+
     (void)new wxButton( panel, ID_BTNNEWTEXT, wxT("New text"), wxPoint(450, 160) );
 
     int initialSpinValue = -5;
