Index: src/msw/listctrl.cpp
===================================================================
--- src/msw/listctrl.cpp	(revision 51611)
+++ src/msw/listctrl.cpp	(working copy)
@@ -346,14 +346,18 @@
 
 void wxListCtrl::MSWSetExListStyles()
 {
+#if !defined(__WXWINCE__) // Windows CE 2.0 and later support these styles
     // for comctl32.dll v 4.70+ we want to have some non default extended
     // styles because it's prettier (and also because wxGTK does it like this)
     if ( wxApp::GetComCtl32Version() >= 470 )
+#endif
     {
         ::SendMessage
         (
             GetHwnd(), LVM_SETEXTENDEDLISTVIEWSTYLE, 0,
+#if !defined(__WXWINCE__) // LVS_EX_LABELTIP is not in the supported styles list
             LVS_EX_LABELTIP |
+#endif
             LVS_EX_FULLROWSELECT |
             LVS_EX_SUBITEMIMAGES |
             // normally this should be governed by a style as it's probably not
@@ -424,13 +428,14 @@
 #if !( defined(__GNUWIN32__) && !wxCHECK_W32API_VERSION( 1, 0 ) )
     if ( style & wxLC_VIRTUAL )
     {
+#if !defined(__WXWINCE__) // Windows CE 2.0 and later support virtual list views
         int ver = wxApp::GetComCtl32Version();
         if ( ver < 470 )
         {
             wxLogWarning(_("Please install a newer version of comctl32.dll\n(at least version 4.70 is required but you have %d.%02d)\nor this program won't operate correctly."),
                         ver / 100, ver % 100);
         }
-
+#endif
         wstyle |= LVS_OWNERDATA;
     }
 #endif // ancient cygwin
@@ -1569,8 +1574,16 @@
     hitTestInfo.pt.y = (int) point.y;
 
     long item;
+
+#if defined(__WXWINCE__)
+    // Windows CE 2.0 and later have ListView_SubItemHitTest
+    bool hasListView_SubItemHitTest = true;
+#else
+    bool hasListView_SubItemHitTest = wxApp::GetComCtl32Version() >= 470;
+#endif
+
 #ifdef LVM_SUBITEMHITTEST
-    if ( ptrSubItem && wxApp::GetComCtl32Version() >= 470 )
+    if ( ptrSubItem && hasListView_SubItemHitTest )
     {
         item = ListView_SubItemHitTest(GetHwnd(), &hitTestInfo);
         *ptrSubItem = hitTestInfo.iSubItem;
@@ -3190,7 +3203,9 @@
 #ifdef NM_CUSTOMDRAW // _WIN32_IE >= 0x0300
     if ( item.m_mask & wxLIST_MASK_IMAGE )
     {
+#if !defined(__WXWINCE__)
         if ( wxApp::GetComCtl32Version() >= 470 )
+#endif
         {
             lvCol.mask |= LVCF_IMAGE;
 

 	  	 
