Index: samples/controls/controls.cpp
===================================================================
--- samples/controls/controls.cpp	(revisione 60243)
+++ samples/controls/controls.cpp	(copia locale)
@@ -903,7 +903,7 @@
                             wxSize(240, wxDefaultCoord)
                           );
 #endif
-    wrapping_sizer->Add( m_wrappingText );
+    wrapping_sizer->Add( m_wrappingText, 1 );
 
     wxStaticBoxSizer *non_wrapping_sizer = new wxStaticBoxSizer( wxVERTICAL, panel, wxT("Non-wrapping") );
     main_sizer->Add( non_wrapping_sizer, 0, wxALL, 5 );
@@ -944,9 +944,12 @@
 
     panel = new wxPanel(m_book);
 
+    wxStaticBoxSizer *szTemp = new wxStaticBoxSizer(
+            new wxStaticBox (panel, wxID_ANY, _T("temp")), wxHORIZONTAL );
+
 #if !defined(__WXMOTIF__) // wxStaticBitmap not working under Motif yet.
     wxIcon icon = wxArtProvider::GetIcon(wxART_INFORMATION);
-    (void) new wxStaticBitmap( panel, wxID_ANY, icon, wxPoint(10, 10) );
+    szTemp->Add( new wxStaticBitmap( panel, wxID_ANY, icon, wxPoint(10, 10) ) );
 
     // VZ: don't leak memory
     // bmpStatic = new wxStaticBitmap(panel, wxID_ANY, wxNullIcon, wxPoint(50, 10));
@@ -963,9 +966,12 @@
     dc.DrawText(_T("Bitmap"), 30, 40);
     dc.SelectObject( wxNullBitmap );
 
-    (void)new wxBitmapButton(panel, ID_BITMAP_BTN, bitmap, wxPoint(100, 20));
-    (void)new wxToggleButton(panel, ID_BITMAP_BTN_ENABLE,
-                             _T("Enable/disable &bitmap"), wxPoint(100, 140));
+    wxBitmapButton *p = new wxBitmapButton(panel, ID_BITMAP_BTN, bitmap, wxPoint(100, 20));
+    p->SetToolTip("casdfasfdsafdsa");
+    szTemp->Add(p);
+    
+    szTemp->Add(new wxToggleButton(panel, ID_BITMAP_BTN_ENABLE,
+                             _T("Enable/disable &bitmap"), wxPoint(100, 140)));
 
 #if defined(__WXMSW__) || defined(__WXMOTIF__)
     // test for masked bitmap display
@@ -990,14 +996,20 @@
 
     bmpBtn->SetBitmapSelected(bmp2);
     bmpBtn->SetBitmapFocus(bmp3);
+    bmpBtn->SetToolTip( "ciao" );
+    bmpBtn->SetToolTip( "ciao" );
+    szTemp->Add(bmpBtn);
 
-    (void)new wxToggleButton(panel, ID_BUTTON_LABEL,
-                             _T("&Toggle label"), wxPoint(250, 20));
+    szTemp->Add(new wxToggleButton(panel, ID_BUTTON_LABEL,
+                             _T("&Toggle label"), wxPoint(250, 20)));
 
     m_label = new wxStaticText(panel, wxID_ANY, _T("Label with some long text"),
                                wxPoint(250, 60), wxDefaultSize,
                                wxALIGN_RIGHT /*| wxST_NO_AUTORESIZE*/);
     m_label->SetForegroundColour( *wxBLUE );
+    szTemp->Add(m_label);
+    
+    panel->SetSizerAndFit(szTemp);
 
     m_book->AddPage(panel, _T("wxBitmapXXX"));
 
