Index: src/osx/cocoa/dataview.mm
===================================================================
--- src/osx/cocoa/dataview.mm	(revision 76432)
+++ src/osx/cocoa/dataview.mm	(working copy)
@@ -2751,7 +2751,9 @@
       m_editorCtrlPtr(NULL),
       m_DCPtr(NULL)
 {
-    SetNativeData(new wxDataViewRendererNativeData([[wxCustomCell alloc] init]));
+    wxCustomCell* cell = [[wxCustomCell alloc] init];
+    SetNativeData(new wxDataViewRendererNativeData(cell));
+    [cell release];
 }
 
 bool wxDataViewCustomRenderer::MacRender()
