Index: src/common/docview.cpp
===================================================================
--- src/common/docview.cpp	(revision 70562)
+++ src/common/docview.cpp	(working copy)
@@ -1316,26 +1316,36 @@
 
 void wxDocManager::OnUpdateUndo(wxUpdateUIEvent& event)
 {
+    if (!GetCurrentDocument())
+    {
+        event.Enable(false);
+        return;
+    }
     wxCommandProcessor * const cmdproc = GetCurrentCommandProcessor();
     if ( !cmdproc )
     {
-        event.Enable(false);
+        // doc has no wxCommandProcessor, but may handle wxID_UNDO itself
+        event.Skip();
         return;
     }
-
     event.Enable(cmdproc->CanUndo());
     cmdproc->SetMenuStrings();
 }
 
 void wxDocManager::OnUpdateRedo(wxUpdateUIEvent& event)
 {
+    if (!GetCurrentDocument())
+    {
+        event.Enable(false);
+        return;
+    }
     wxCommandProcessor * const cmdproc = GetCurrentCommandProcessor();
     if ( !cmdproc )
     {
-        event.Enable(false);
+        // doc has no wxCommandProcessor, but may handle wxID_REDO itself
+        event.Skip();
         return;
     }
-
     event.Enable(cmdproc->CanRedo());
     cmdproc->SetMenuStrings();
 }
