Index: interface/wx/choicdlg.h
===================================================================
--- interface/wx/choicdlg.h	(revision 70705)
+++ interface/wx/choicdlg.h	(working copy)
@@ -314,11 +314,11 @@
                            const wxString& caption,
                            const wxArrayString& aChoices,
                            wxWindow* parent = NULL,
-                           int x = -1,
-                           int y = -1,
+                           int x = wxDefaultCoord,
+                           int y = wxDefaultCoord,
                            bool centre = true,
-                           int width = 150,
-                           int height = 200,
+                           int width = wxCHOICE_WIDTH,
+                           int height = wxCHOICE_HEIGHT,
                            int initialSelection = 0);
 
 int wxGetSingleChoiceIndex(const wxString& message,
@@ -326,11 +326,11 @@
                            int n,
                            const wxString& choices[],
                            wxWindow* parent = NULL,
-                           int x = -1,
-                           int y = -1,
+                           int x = wxDefaultCoord,
+                           int y = wxDefaultCoord,
                            bool centre = true,
-                           int width = 150,
-                           int height = 200,
+                           int width = wxCHOICE_WIDTH,
+                           int height = wxCHOICE_HEIGHT,
                            int initialSelection = 0);
 
 
@@ -374,22 +374,22 @@
                            const wxString& caption,
                            const wxArrayString& aChoices,
                            wxWindow* parent = NULL,
-                           int x = -1,
-                           int y = -1,
+                           int x = wxDefaultCoord,
+                           int y = wxDefaultCoord,
                            bool centre = true,
-                           int width = 150,
-                           int height = 200
+                           int width = wxCHOICE_WIDTH,
+                           int height = wxCHOICE_HEIGHT,
                            int initialSelection = 0);
 wxString wxGetSingleChoice(const wxString& message,
                            const wxString& caption,
                            int n,
                            const wxString& choices[],
                            wxWindow* parent = NULL,
-                           int x = -1,
-                           int y = -1,
+                           int x = wxDefaultCoord,
+                           int y = wxDefaultCoord,
                            bool centre = true,
-                           int width = 150,
-                           int height = 200
+                           int width = wxCHOICE_WIDTH,
+                           int height = wxCHOICE_HEIGHT,
                            int initialSelection = 0);
 
 
@@ -427,11 +427,11 @@
                                const wxArrayString& aChoices,
                                const wxString& client_data[],
                                wxWindow* parent = NULL,
-                               int x = -1,
-                               int y = -1,
+                               int x = wxDefaultCoord,
+                               int y = wxDefaultCoord,
                                bool centre = true,
-                               int width = 150,
-                               int height = 200
+                               int width = wxCHOICE_WIDTH,
+                               int height = wxCHOICE_HEIGHT,
                                int initialSelection = 0);
 
 wxString wxGetSingleChoiceData(const wxString& message,
@@ -440,11 +440,11 @@
                                const wxString& choices[],
                                const wxString& client_data[],
                                wxWindow* parent = NULL,
-                               int x = -1,
-                               int y = -1,
+                               int x = wxDefaultCoord,
+                               int y = wxDefaultCoord,
                                bool centre = true,
-                               int width = 150,
-                               int height = 200
+                               int width = wxCHOICE_WIDTH,
+                               int height = wxCHOICE_HEIGHT,
                                int initialSelection = 0);
 
 void* wxGetSingleChoiceData(const wxString& message,
@@ -494,11 +494,11 @@
                             const wxString& caption,
                             const wxArrayString& aChoices,
                             wxWindow* parent = NULL,
-                            int x = -1,
-                            int y = -1,
+                            int x = wxDefaultCoord,
+                            int y = wxDefaultCoord,
                             bool centre = true,
-                            int width = 150,
-                            int height = 200);
+                            int width = wxCHOICE_WIDTH,
+                            int height = wxCHOICE_HEIGHT);
 
 int wxGetSelectedChoices(wxArrayInt& selections,
                             const wxString& message,
@@ -506,11 +506,11 @@
                             int n,
                             const wxString& choices[],
                             wxWindow* parent = NULL,
-                            int x = -1,
-                            int y = -1,
+                            int x = wxDefaultCoord,
+                            int y = wxDefaultCoord,
                             bool centre = true,
-                            int width = 150,
-                            int height = 200);
+                            int width = wxCHOICE_WIDTH,
+                            int height = wxCHOICE_HEIGHT);
 
 //@}
 
Index: interface/wx/defs.h
===================================================================
--- interface/wx/defs.h	(revision 70705)
+++ interface/wx/defs.h	(working copy)
@@ -1074,11 +1074,11 @@
     wxPAPER_10X14,              ///<  10-by-14-inch sheet
     wxPAPER_11X17,              ///<  11-by-17-inch sheet
     wxPAPER_NOTE,               ///<  Note, 8 1/2 by 11 inches
-    wxPAPER_ENV_9,              ///<  #9 Envelope, 3 7/8 by 8 7/8 inches
-    wxPAPER_ENV_10,             ///<  #10 Envelope, 4 1/8 by 9 1/2 inches
-    wxPAPER_ENV_11,             ///<  #11 Envelope, 4 1/2 by 10 3/8 inches
-    wxPAPER_ENV_12,             ///<  #12 Envelope, 4 3/4 by 11 inches
-    wxPAPER_ENV_14,             ///<  #14 Envelope, 5 by 11 1/2 inches
+    wxPAPER_ENV_9,              ///<  \#9 Envelope, 3 7/8 by 8 7/8 inches
+    wxPAPER_ENV_10,             ///<  \#10 Envelope, 4 1/8 by 9 1/2 inches
+    wxPAPER_ENV_11,             ///<  \#11 Envelope, 4 1/2 by 10 3/8 inches
+    wxPAPER_ENV_12,             ///<  \#12 Envelope, 4 3/4 by 11 inches
+    wxPAPER_ENV_14,             ///<  \#14 Envelope, 5 by 11 1/2 inches
     wxPAPER_ENV_DL,             ///<  DL Envelope, 110 by 220 millimeters
     wxPAPER_ENV_C5,             ///<  C5 Envelope, 162 by 229 millimeters
     wxPAPER_ENV_C3,             ///<  C3 Envelope, 324 by 458 millimeters
@@ -1691,3 +1691,54 @@
 //@}
 
 
+/**
+    Default width of the choice dialog.
+*/
+#define wxCHOICE_WIDTH 150
+
+/**
+    Default height of the choice dialog.
+*/
+#define wxCHOICE_HEIGHT 200
+
+/**
+    Default style of the choice dialog.
+    @remarks For WinCE the value is only (wxDEFAULT_DIALOG_STYLE | wxOK | wxCANCEL | wxCENTRE).
+*/
+#define wxCHOICEDLG_STYLE (wxDEFAULT_DIALOG_STYLE | wxOK | wxCANCEL | wxCENTRE | wxRESIZE_BORDER)
+
+/**
+    Initial folder for generic directory dialog.
+*/
+const char wxDirDialogDefaultFolderStr[] = "/";
+
+/**
+    Default message for directory selector dialog.
+*/
+const char wxDirSelectorPromptStr[] = "Select a directory";
+
+/**
+    Default name for directory selector dialog.
+*/
+const char wxDirDialogNameStr[] = "wxDirCtrl";
+
+/**
+    It is defined as "*.*" unde MSW and OS/2 and "*" everywhere else.
+*/
+const char wxFileSelectorDefaultWildcardStr[];
+
+/**
+    Default message box caption string.
+*/
+const char wxMessageBoxCaptionStr[] = "Message";
+
+/**
+    Default text dialog style.
+*/
+#define wxTextEntryDialogStyle (wxOK | wxCANCEL | wxCENTRE | wxWS_EX_VALIDATE_RECURSIVELY)
+
+const char wxGetTextFromUserPromptStr[] = "Input Text";    ///< Default text dialog caption.
+const char wxGetPasswordFromUserPromptStr[] = "Enter Password";    ///< Default password dialog caption.
+
+#define wxDEFAULT_DELIMITERS wxT(" \t\r\n")   ///< Default delimiters are the usual white space characters.
+
Index: interface/wx/filedlg.h
===================================================================
--- interface/wx/filedlg.h	(revision 70705)
+++ interface/wx/filedlg.h	(working copy)
@@ -360,11 +360,40 @@
                         const wxString& default_path = wxEmptyString,
                         const wxString& default_filename = wxEmptyString,
                         const wxString& default_extension = wxEmptyString,
-                        const wxString& wildcard = ".",
+                        const wxString& wildcard = wxFileSelectorDefaultWildcardStr,
                         int flags = 0,
                         wxWindow* parent = NULL,
-                        int x = -1,
-                        int y = -1);
+                        int x = wxDefaultCoord,
+                        int y = wxDefaultCoord);
 
+/**
+    An extended version of wxFileSelector
+*/
+wxString wxFileSelectorEx(const wxString& message = wxFileSelectorPromptStr,
+                          const wxString& default_path = wxEmptyString,
+                          const wxString& default_filename = wxEmptyString,
+                          int *indexDefaultExtension = NULL,
+                          const wxString& wildcard = wxFileSelectorDefaultWildcardStr,
+                          int flags = 0,
+                          wxWindow *parent = NULL,
+                          int x = wxDefaultCoord,
+                          int y = wxDefaultCoord);
+
+/**
+    Ask for filename to load
+*/
+wxString wxLoadFileSelector(const wxString& what,
+                            const wxString& extension,
+                            const wxString& default_name = wxEmptyString,
+                            wxWindow *parent = NULL);
+
+/**
+    Ask for filename to save
+*/
+wxString wxSaveFileSelector(const wxString& what,
+                            const wxString& extension,
+                            const wxString& default_name = wxEmptyString,
+                            wxWindow *parent = NULL);
+
 //@}
 
Index: interface/wx/textdlg.h
===================================================================
--- interface/wx/textdlg.h	(revision 70705)
+++ interface/wx/textdlg.h	(working copy)
@@ -45,7 +45,7 @@
     wxPasswordEntryDialog(wxWindow* parent, const wxString& message,
                           const wxString& caption = wxGetPasswordFromUserPromptStr,
                           const wxString& defaultValue = wxEmptyString,
-                          long style = wxOK | wxCANCEL | wxCENTRE,
+                          long style = wxTextEntryDialogStyle,
                           const wxPoint& pos = wxDefaultPosition);
 };
 
@@ -87,7 +87,7 @@
     wxTextEntryDialog(wxWindow* parent, const wxString& message,
                       const wxString& caption = wxGetTextFromUserPromptStr,
                       const wxString& value = wxEmptyString,
-                      long style = wxOK | wxCANCEL | wxCENTRE,
+                      long style = wxTextEntryDialogStyle,
                       const wxPoint& pos = wxDefaultPosition);
 
     /**
@@ -139,7 +139,7 @@
     @header{wx/textdlg.h}
 */
 wxString wxGetTextFromUser(const wxString& message,
-                           const wxString& caption = "Input text",
+                           const wxString& caption = wxGetTextFromUserPromptStr,
                            const wxString& default_value = wxEmptyString,
                            wxWindow* parent = NULL,
                            int x = wxDefaultCoord,
@@ -154,7 +154,7 @@
     @header{wx/textdlg.h}
 */
 wxString wxGetPasswordFromUser(const wxString& message,
-                               const wxString& caption = "Input text",
+                               const wxString& caption = wxGetPasswordFromUserPromptStr,
                                const wxString& default_value = wxEmptyString,
                                wxWindow* parent = NULL,
                                int x = wxDefaultCoord,
Index: interface/wx/tokenzr.h
===================================================================
--- interface/wx/tokenzr.h	(revision 70705)
+++ interface/wx/tokenzr.h	(working copy)
@@ -106,7 +106,7 @@
         @see SetString()
    */
     wxStringTokenizer(const wxString& str,
-                      const wxString& delims = " \t\r\n",
+                      const wxString& delims = wxDEFAULT_DELIMITERS,
                       wxStringTokenizerMode mode = wxTOKEN_DEFAULT);
 
     /**
@@ -154,8 +154,8 @@
         containing delimiters, and the @a mode specifying how the string
         should be tokenized.
     */
-    void SetString(const wxString& to_tokenize,
-                   const wxString& delims = " \t\r\n",
+    void SetString(const wxString& str,
+                   const wxString& delims = wxDEFAULT_DELIMITERS,
                    wxStringTokenizerMode mode = wxTOKEN_DEFAULT);
 };
 
@@ -178,5 +178,5 @@
 wxStringTokenize(const wxString& str,
                  const wxString& delims = wxDEFAULT_DELIMITERS,
                  wxStringTokenizerMode mode = wxTOKEN_DEFAULT);
-                 
+
 //@}
