--- orig1.wxrc.cpp	2008-06-24 16:46:41.000000000 +0200
+++ wxrc.cpp	2008-06-24 16:56:54.000000000 +0200
@@ -136,7 +136,7 @@
             if( !IsRealClass(w.GetClass()) ) continue;
             if( w.GetName().Length() == 0 ) continue;
             file.Write(
-                _T(" ") + w.GetClass() + _T("* ") + w.GetName()
+                _T(" ") + w.GetClass() + _T("* m_") + w.GetName()
                 + _T(";\n"));
         }
         file.Write(_T("\nprivate:\n void InitWidgetsFromXRC(wxWindow *parent){\n")
@@ -150,7 +150,7 @@
             const XRCWidgetData& w = m_wdata.Item(i);
             if( !IsRealClass(w.GetClass()) ) continue;
             if( w.GetName().Length() == 0 ) continue;
-            file.Write( _T("  ")
+            file.Write( _T("  m_")
                         + w.GetName()
                         + _T(" = XRCCTRL(*this,\"")
                         + w.GetName()
