Index: src/common/imagtiff.cpp
===================================================================
--- src/common/imagtiff.cpp	(revision 66259)
+++ src/common/imagtiff.cpp	(working copy)
@@ -635,7 +635,7 @@
         micro;
 
     const wxString ver(::TIFFGetVersion());
-    if ( wxSscanf(ver, "LIBTIFF, Version %d.%d.%d", &major, &micro, &micro) != 3 )
+    if ( wxSscanf(ver, "LIBTIFF, Version %d.%d.%d", &major, &minor, &micro) != 3 )
     {
         wxLogDebug("Unrecognized libtiff version string \"%s\"", ver);
 
Index: src/stc/stc.cpp
===================================================================
--- src/stc/stc.cpp	(revision 66259)
+++ src/stc/stc.cpp	(working copy)
@@ -4620,7 +4620,7 @@
 
 /*static*/ wxVersionInfo wxStyledTextCtrl::GetLibraryVersionInfo()
 {
-    return wxVersionInfo("Scintilla", 2, 3);
+    return wxVersionInfo("Scintilla", 2, 3, 0, "Scintilla 2.03");
 }
 
 #endif // wxUSE_STC
