Index: src/xml/xml.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/xml/xml.cpp,v
retrieving revision 1.25
diff -u -r1.25 xml.cpp
--- src/xml/xml.cpp	2006/06/08 08:09:27	1.25
+++ src/xml/xml.cpp	2006/07/03 16:12:19
@@ -348,12 +348,16 @@
 bool wxXmlDocument::Load(const wxString& filename, const wxString& encoding)
 {
     wxFileInputStream stream(filename);
+    if (!stream.Ok())
+        return false;
     return Load(stream, encoding);
 }
 
 bool wxXmlDocument::Save(const wxString& filename) const
 {
     wxFileOutputStream stream(filename);
+    if (!stream.Ok())
+        return false;
     return Save(stream);
 }
 

 	  	 
