Index: include/wx/aui/auibook.h
===================================================================
--- include/wx/aui/auibook.h	(revision 62459)
+++ include/wx/aui/auibook.h	(working copy)
@@ -611,6 +611,7 @@
     void OnTabBeginDrag(wxAuiNotebookEvent& evt);
     void OnTabDragMotion(wxAuiNotebookEvent& evt);
     void OnTabEndDrag(wxAuiNotebookEvent& evt);
+    void OnTabCancelDrag(wxAuiNotebookEvent& evt); 
     void OnTabButton(wxAuiNotebookEvent& evt);
     void OnTabMiddleDown(wxAuiNotebookEvent& evt);
     void OnTabMiddleUp(wxAuiNotebookEvent& evt);
Index: src/aui/auibook.cpp
===================================================================
--- src/aui/auibook.cpp	(revision 62459)
+++ src/aui/auibook.cpp	(working copy)
@@ -48,6 +48,7 @@
 wxDEFINE_EVENT(wxEVT_COMMAND_AUINOTEBOOK_BUTTON, wxAuiNotebookEvent);
 wxDEFINE_EVENT(wxEVT_COMMAND_AUINOTEBOOK_BEGIN_DRAG, wxAuiNotebookEvent);
 wxDEFINE_EVENT(wxEVT_COMMAND_AUINOTEBOOK_END_DRAG, wxAuiNotebookEvent);
+wxDEFINE_EVENT(wxEVT_COMMAND_AUINOTEBOOK_CANCEL_DRAG, wxAuiNotebookEvent);
 wxDEFINE_EVENT(wxEVT_COMMAND_AUINOTEBOOK_DRAG_MOTION, wxAuiNotebookEvent);
 wxDEFINE_EVENT(wxEVT_COMMAND_AUINOTEBOOK_ALLOW_DND, wxAuiNotebookEvent);
 wxDEFINE_EVENT(wxEVT_COMMAND_AUINOTEBOOK_BG_DCLICK, wxAuiNotebookEvent);
@@ -2276,6 +2277,16 @@
 
 void wxAuiTabCtrl::OnCaptureLost(wxMouseCaptureLostEvent& WXUNUSED(event))
 {
+    if (m_is_dragging) 
+    { 
+        m_is_dragging = false; 
+
+        wxAuiNotebookEvent evt(wxEVT_COMMAND_AUINOTEBOOK_CANCEL_DRAG, m_windowId); 
+        evt.SetSelection(GetIdxFromWindow(m_click_tab)); 
+        evt.SetOldSelection(evt.GetSelection()); 
+        evt.SetEventObject(this); 
+        GetEventHandler()->ProcessEvent(evt); 
+    }
 }
 
 void wxAuiTabCtrl::OnLeftUp(wxMouseEvent& evt)
@@ -2791,6 +2802,9 @@
                       wxEVT_COMMAND_AUINOTEBOOK_END_DRAG,
                       wxAuiNotebook::OnTabEndDrag)
     EVT_AUI_RANGE(wxAuiBaseTabCtrlId, wxAuiBaseTabCtrlId+500,
+                      wxEVT_COMMAND_AUINOTEBOOK_CANCEL_DRAG,
+                      wxAuiNotebook::OnTabCancelDrag)
+    EVT_AUI_RANGE(wxAuiBaseTabCtrlId, wxAuiBaseTabCtrlId+500,
                       wxEVT_COMMAND_AUINOTEBOOK_DRAG_MOTION,
                       wxAuiNotebook::OnTabDragMotion)
     EVT_AUI_RANGE(wxAuiBaseTabCtrlId, wxAuiBaseTabCtrlId+500,
@@ -4063,6 +4077,18 @@
 
 
 
+void wxAuiNotebook::OnTabCancelDrag(wxAuiNotebookEvent& command_evt) 
+{
+    wxAuiNotebookEvent& evt = (wxAuiNotebookEvent&)command_evt; 
+
+    m_mgr.HideHint(); 
+
+    wxAuiTabCtrl* src_tabs = (wxAuiTabCtrl*)evt.GetEventObject(); 
+    wxCHECK_RET( src_tabs, _T("no source object?") ); 
+
+    src_tabs->SetCursor(wxCursor(wxCURSOR_ARROW)); 
+} 
+
 wxAuiTabCtrl* wxAuiNotebook::GetTabCtrlFromPoint(const wxPoint& pt)
 {
     // if we've just removed the last tab from the source
