Index: src/mac/carbon/toplevel.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/mac/carbon/toplevel.cpp,v
retrieving revision 1.145
diff -u -p -r1.145 toplevel.cpp
--- src/mac/carbon/toplevel.cpp	2005/04/15 07:22:33	1.145
+++ src/mac/carbon/toplevel.cpp	2005/05/18 16:32:58
@@ -978,6 +978,15 @@ void wxTopLevelWindowMac::Maximize(bool 
 {
     wxMacPortStateHelper help( (GrafPtr) GetWindowPort( (WindowRef) m_macWindow) ) ;
     wxMacWindowClipper clip (this);
+
+    if ( !IsWindowInStandardState( (WindowRef)m_macWindow, NULL, NULL) ) 
+    {
+        Rect rect;
+        GetWindowBounds((WindowRef)m_macWindow, kWindowGlobalPortRgn, &rect);
+        SetWindowIdealUserState((WindowRef)m_macWindow, &rect);
+        SetWindowUserState((WindowRef)m_macWindow, &rect);
+    }
+    
     ZoomWindow( (WindowRef)m_macWindow , maximize ? inZoomOut : inZoomIn , false ) ;
 }
 

 	  	 
