Index: src/msw/listctrl.cpp
===================================================================
--- src/msw/listctrl.cpp	(revision 54333)
+++ src/msw/listctrl.cpp	(working copy)
@@ -2142,9 +2142,11 @@
                     // <Enter> or <Space> activate the selected item if any (but
                     // not with Shift and/or Ctrl as then they have a predefined
                     // meaning for the list view)
+					// Don't activate on Alt + Space either since that opens the 
+					// system menu of the currently active TLW
                     if ( lItem != -1 &&
                          (wVKey == VK_RETURN || wVKey == VK_SPACE) &&
-                         !(wxIsShiftDown() || wxIsCtrlDown()) )
+                         !(wxIsShiftDown() || wxIsCtrlDown() || (wVKey == VK_SPACE && GetKeyState(VK_MENU) < 0)) )
                     {
                         eventType = wxEVT_COMMAND_LIST_ITEM_ACTIVATED;
                     }
