Index: samples/minimal/minimal.cpp
===================================================================
--- samples/minimal/minimal.cpp	(revision 65008)
+++ samples/minimal/minimal.cpp	(working copy)
@@ -16,10 +16,10 @@
 // ----------------------------------------------------------------------------
 // headers
 // ----------------------------------------------------------------------------
- 
+
 // For compilers that support precompilation, includes "wx/wx.h".
 #include "wx/wxprec.h"
- 
+
 #ifdef __BORLANDC__
     #pragma hdrstop
 #endif
@@ -28,6 +28,7 @@
 // need because it includes almost all "standard" wxWidgets headers)
 #ifndef WX_PRECOMP
     #include "wx/wx.h"
+    #include <wx/tipwin.h>
 #endif
 
 // ----------------------------------------------------------------------------
@@ -172,6 +173,12 @@
     CreateStatusBar(2);
     SetStatusText("Welcome to wxWidgets!");
 #endif // wxUSE_STATUSBAR
+
+     wxTipWindow* tip1 = new wxTipWindow(this, "First tip.");
+     tip1->Move(GetPosition());
+
+     wxTipWindow* tip2 = new wxTipWindow(this, "Second tip.");
+     tip2->Move(GetPosition() + tip1->GetSize());
 }
 
 
