 samples/mediaplayer/mediaplayer.cpp | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/samples/mediaplayer/mediaplayer.cpp b/samples/mediaplayer/mediaplayer.cpp
index e9e779b0b9..a5be8bc6d4 100644
--- a/samples/mediaplayer/mediaplayer.cpp
+++ b/samples/mediaplayer/mediaplayer.cpp
@@ -990,6 +990,8 @@ void wxMediaPlayerFrame::DoPlayFile(const wxString& path)
 // ----------------------------------------------------------------------------
 void wxMediaPlayerFrame::OnMediaLoaded(wxMediaEvent& WXUNUSED(evt))
 {
+    wxLogMessage("OnMediaLoaded");
+
     wxMediaPlayerNotebookPage* currentpage =
         (wxMediaPlayerNotebookPage*) m_notebook->GetCurrentPage();
 
@@ -1490,13 +1492,21 @@ wxMediaPlayerNotebookPage::wxMediaPlayerNotebookPage(wxMediaPlayerFrame* parentF
     //
     m_mediactrl = new wxMediaCtrl();
 
+    wxArrayString backends;
+    wxString backend;
+
+    backends.push_back(szBackend);
+    backends.push_back(wxMEDIABACKEND_DIRECTSHOW);
+    backends.push_back(wxMEDIABACKEND_WMP10);
+    backend = wxGetSingleChoice("empty string means default backend", "Select backend", backends);
+
     //  Make sure creation was successful
     bool bOK = m_mediactrl->Create(this, wxID_MEDIACTRL, wxEmptyString,
                                     wxDefaultPosition, wxDefaultSize, 0,
 // you could specify a macro backend here like
 //  wxMEDIABACKEND_WMP10);
 //        "wxPDFMediaBackend");
-                                   szBackend);
+                                   backend);
 // you could change the cursor here like
 //    m_mediactrl->SetCursor(wxCURSOR_BLANK);
 // note that this may not effect it if SetPlayerControls
@@ -1766,6 +1776,7 @@ void wxMediaPlayerNotebookPage::OnMediaPause(wxMediaEvent& WXUNUSED(event))
 // ----------------------------------------------------------------------------
 void wxMediaPlayerNotebookPage::OnMediaStop(wxMediaEvent& WXUNUSED(event))
 {
+    wxLogMessage("OnMediaStop");
     m_playlist->SetItem(m_nLastFileId, 0, "[]");
 }
 
