Index: src/aui/tabartgtk.cpp
===================================================================
--- src/aui/tabartgtk.cpp	(Revision 71002)
+++ src/aui/tabartgtk.cpp	(Arbeitskopie)
@@ -213,7 +213,7 @@
             tab_rect.y -= 2 * GTK_NOTEBOOK (wxGTKPrivate::GetNotebookWidget())->tab_hborder;
             if (!page.active)
                 tab_rect.y += 2 * GTK_NOTEBOOK (wxGTKPrivate::GetNotebookWidget())->tab_hborder;
-            gap_y = tab_rect.y + tab_rect.height;
+            gap_y = tab_rect.y + tab_rect.height  - GTK_NOTEBOOK (wxGTKPrivate::GetNotebookWidget())->tab_hborder / 2;
             // fall through
         case wxAUI_NB_BOTTOM:
             gap_x = tab_rect.x - GTK_NOTEBOOK (wxGTKPrivate::GetNotebookWidget())->tab_vborder / 2;
@@ -247,7 +247,7 @@
             gtk_paint_box_gap(style_notebook, window, GTK_STATE_NORMAL, GTK_SHADOW_OUT,
                               NULL, widget,
                               const_cast<char*>("notebook"),
-                              window_rect.x, gap_y,
+                              1, gap_y,
                               window_rect.width, gap_height,
                               GTK_POS_BOTTOM, gap_x , gap_width);
         }
@@ -266,7 +266,7 @@
             gtk_paint_box_gap(style_notebook, window, GTK_STATE_NORMAL, GTK_SHADOW_OUT,
                               NULL, widget,
                               const_cast<char*>("notebook"),
-                              window_rect.x, gap_y,
+                              1, gap_y,
                               window_rect.width, gap_height,
                               GTK_POS_TOP, gap_x , gap_width);
         }
