diff -ru2 wxWindows-cvs-orig/include/univ/menuitem.h wxWindows-cvs-patch/include/univ/menuitem.h
--- wxWindows-cvs-orig/include/univ/menuitem.h	Sat Feb 07 19:40:32 2004
+++ wxWindows-cvs-patch/include/univ/menuitem.h	Sat Feb 07 20:36:36 2004
@@ -49,4 +49,9 @@
       { return checked ? m_bmpChecked : m_bmpUnchecked; }
 
+    void SetDisabledBitmap( const wxBitmap& bmpDisabled )
+      { m_bmpDisabled = bmpDisabled; }
+    const wxBitmap& GetDisabledBitmap() const
+      { return m_bmpDisabled; }
+
     // mark item as belonging to the given radio group
     void SetAsRadioGroupStart();
@@ -93,5 +98,6 @@
     // the bitmaps (may be invalid, then they're not used)
     wxBitmap m_bmpChecked,
-             m_bmpUnchecked;
+             m_bmpUnchecked,
+             m_bmpDisabled; 
 
     // the positions of the first and last items of the radio group this item
diff -ru2 wxWindows-cvs-orig/src/univ/menu.cpp wxWindows-cvs-patch/src/univ/menu.cpp
--- wxWindows-cvs-orig/src/univ/menu.cpp	Sat Feb 07 19:41:54 2004
+++ wxWindows-cvs-patch/src/univ/menu.cpp	Sat Feb 07 20:39:16 2004
@@ -586,4 +586,19 @@
                 flags |= wxCONTROL_SELECTED;
 
+            wxBitmap bmp;
+
+            if ( !item->IsEnabled() )
+            {
+                bmp = item->GetDisabledBitmap();
+            }
+
+            if ( !bmp.Ok() )
+            {
+                // strangely enough, for unchecked item we use the
+                // "checked" bitmap because this is the default one - this
+                // explains this strange boolean expression
+                bmp = item->GetBitmap(!item->IsCheckable() || item->IsChecked());
+            }
+
             rend->DrawMenuItem
                   (
@@ -593,8 +608,5 @@
                      item->GetLabel(),
                      item->GetAccelString(),
-                     // strangely enough, for unchecked item we use the
-                     // "checked" bitmap because this is the default one - this
-                     // explains this strange boolean expression
-                     item->GetBitmap(!item->IsCheckable() || item->IsChecked()),
+                     bmp,
                      flags,
                      item->GetAccelIndex()
@@ -1487,4 +1499,6 @@
     m_radioGroup.start = -1;
     m_isRadioGroupStart = FALSE;
+
+    m_bmpDisabled = wxNullBitmap;
 
     UpdateAccelInfo();

 	  	 
