Index: src/os2/frame.cpp
===================================================================
--- src/os2/frame.cpp	(Revision 72610)
+++ src/os2/frame.cpp	(Arbeitskopie)
@@ -1071,9 +1071,8 @@
 {
     MRESULT  rc = (MRESULT)0;
     bool     bProcessed = false;
-    wxFrame* pWnd  = NULL;
 
-    pWnd = (wxFrame*) wxFindWinFromHandle((WXHWND) hWnd);
+    wxFrame* pWnd = (wxFrame*) wxFindWinFromHandle((WXHWND) hWnd);
     switch (ulMsg)
     {
         case WM_QUERYFRAMECTLCOUNT:
