Index: include/wx/richtext/richtextformatdlg.h
===================================================================
--- include/wx/richtext/richtextformatdlg.h	(revision 59810)
+++ include/wx/richtext/richtextformatdlg.h	(working copy)
@@ -230,12 +230,15 @@
 
         SetBackgroundColour(*wxWHITE);
         m_textEffects = 0;
+        m_textAttrs.SetFlags(wxTEXT_ATTR_FONT);
     }
 
     void SetTextEffects(int effects) { m_textEffects = effects; }
     int GetTextEffects() const { return m_textEffects; }
+    void SetUserTextAttrs(wxRichTextAttr &textAttrs) { m_textAttrs = m_textAttrs.Combine( textAttrs ); }
 
 private:
+    wxRichTextAttr m_textAttrs;
     int m_textEffects;
 
     void OnPaint(wxPaintEvent& event);
Index: src/richtext/richtextfontpage.cpp
===================================================================
--- src/richtext/richtextfontpage.cpp	(revision 59810)
+++ src/richtext/richtextfontpage.cpp	(working copy)
@@ -557,7 +557,7 @@
 /// Updates the font preview
 void wxRichTextFontPage::UpdatePreview()
 {
-    wxFont font(*wxNORMAL_FONT);
+    wxRichTextAttr textAttrs;
 
     if (m_colourPresent)
         m_previewCtrl->SetForegroundColour(m_colourCtrl->GetBackgroundColour());
@@ -565,7 +565,7 @@
     if (m_faceListBox->GetSelection() != wxNOT_FOUND)
     {
         wxString faceName = m_faceListBox->GetFaceName(m_faceListBox->GetSelection());
-        font.SetFaceName(faceName);
+        textAttrs.SetFontFaceName(faceName);
     }
 
     wxString strSize = m_sizeTextCtrl->GetValue();
@@ -573,7 +573,7 @@
     {
         int sz = wxAtoi(strSize);
         if (sz > 0)
-            font.SetPointSize(sz);
+            textAttrs.SetFontSize(sz);
     }
 
     if (m_styleCtrl->GetSelection() != wxNOT_FOUND)
@@ -584,7 +584,7 @@
         else
             style = wxNORMAL;
 
-        font.SetStyle(style);
+        textAttrs.SetFontStyle(style);
     }
 
     if (m_weightCtrl->GetSelection() != wxNOT_FOUND)
@@ -595,7 +595,7 @@
         else
             weight = wxNORMAL;
 
-        font.SetWeight(weight);
+        textAttrs.SetFontWeight(weight);
     }
 
     if (m_underliningCtrl->GetSelection() != wxNOT_FOUND)
@@ -606,7 +606,7 @@
         else
             underlined = false;
 
-        font.SetUnderlined(underlined);
+        textAttrs.SetFontUnderlined(underlined);
     }
 
     int textEffects = 0;
@@ -626,7 +626,7 @@
     else if ( m_subscriptCtrl->Get3StateValue() == wxCHK_CHECKED )
         textEffects |= wxTEXT_ATTR_EFFECT_SUBSCRIPT;
 
-    m_previewCtrl->SetFont(font);
+    m_previewCtrl->SetUserTextAttrs(textAttrs);
     m_previewCtrl->SetTextEffects(textEffects);
     m_previewCtrl->Refresh();
 }
Index: src/richtext/richtextformatdlg.cpp
===================================================================
--- src/richtext/richtextformatdlg.cpp	(revision 59810)
+++ src/richtext/richtextformatdlg.cpp	(working copy)
@@ -403,7 +403,7 @@
     wxPaintDC dc(this);
 
     wxSize size = GetSize();
-    wxFont font = GetFont();
+    wxFont font = m_textAttrs.CreateFont();
 
     if ((GetTextEffects() & wxTEXT_ATTR_EFFECT_SUPERSCRIPT) || (GetTextEffects() & wxTEXT_ATTR_EFFECT_SUBSCRIPT))
     {
