Index: src/aui/framemanager.cpp
===================================================================
--- src/aui/framemanager.cpp	(revision 65052)
+++ src/aui/framemanager.cpp	(working copy)
@@ -1004,7 +1004,7 @@
     if (pinfo.name.empty() || already_exists)
     {
         pinfo.name.Printf(wxT("%08lx%08x%08x%08lx"),
-             wxPtrToUInt(pinfo.window) & 0xffffffff,
+             (unsigned long)(wxPtrToUInt(pinfo.window) & 0xffffffff),
              (unsigned int)time(NULL),
 #ifdef __WXWINCE__
              (unsigned int)GetTickCount(),
Index: utils/wxrc/wxrc.cpp
===================================================================
--- utils/wxrc/wxrc.cpp	(revision 65052)
+++ utils/wxrc/wxrc.cpp	(working copy)
@@ -586,7 +586,11 @@
                   wxT("Huge file not supported") );
 
     snum.Printf(wxT("%i"), num);
-    output.Printf(wxT("static size_t xml_res_size_") + snum + wxT(" = %zu;\n"), lng);
+#ifdef __WXMSW__
+	output.Printf(wxT("static size_t xml_res_size_") + snum + wxT(" = %Iu;\n"), lng);
+#else
+	output.Printf(wxT("static size_t xml_res_size_") + snum + wxT(" = %zu;\n"), lng);
+#endif
     output += wxT("static unsigned char xml_res_file_") + snum + wxT("[] = {\n");
     // we cannot use string literals because MSVC is dumb wannabe compiler
     // with arbitrary limitation to 2048 strings :(
