diff -ru wxWidgets.orig/src/xrc/xmlres.cpp wxWidgets/src/xrc/xmlres.cpp
--- wxWidgets.orig/src/xrc/xmlres.cpp	2007-08-28 09:24:16.000000000 +0200
+++ wxWidgets/src/xrc/xmlres.cpp	2007-08-28 21:31:48.000000000 +0200
@@ -971,17 +971,12 @@
     double value;
     wxString str1 = GetParamValue(param);
 
-#ifndef __WXWINCE__
-    const char *prevlocale = setlocale(LC_NUMERIC, "C");
-#endif
+    const wxString dp = wxLocale::GetInfo(wxLOCALE_DECIMAL_POINT, wxLOCALE_CAT_NUMBER);
+    str1.Replace(wxT("."), dp);
 
     if (!str1.ToDouble(&value))
         value = defaultv;
 
-#ifndef __WXWINCE__
-    setlocale(LC_NUMERIC, prevlocale);
-#endif
-
     return wx_truncate_cast(float, value);
 }
 

 	  	 
