--- wxWindows/src/common/iconbndl.cpp	Thu Mar 28 20:51:56 2002
+++ iconbndl.cpp	Tue Apr  2 20:22:44 2002
@@ -80,7 +80,11 @@
     wxCoord sysX = wxSystemSettings::GetMetric( wxSYS_ICON_X ),
             sysY = wxSystemSettings::GetMetric( wxSYS_ICON_Y );
-    wxIcon* sysIcon = 0;
 
-    for( i = 0; i < max; ++i )
+    wxIcon *sysIcon = 0;
+    // temp. variable needed to fix Borland C++ 5.5.1 problem
+    // with passing a return value through two functions
+    wxIcon *tmp;
+
+    for( i = 0; i < max; i++ )
     {
         if( !m_icons[i].Ok() )
@@ -89,5 +93,8 @@
         // requested size
         if( sx == size.x && sy == size.y )
-            return m_icons[i];
+        {
+            tmp = &m_icons[i];
+            return *tmp;
+        }
         // keep track if there is a system-size icon
         if( sx == sysX && sy == sysY )
@@ -96,7 +103,12 @@
 
     // return the system-sized icon if we've got one
-    if( sysIcon ) return *sysIcon;
+    if( sysIcon )
+        return *sysIcon;
     // return the first icon, if we have one
-    return max > 0 ? m_icons[0] : wxNullIcon;
+    if ( max > 0 )
+        tmp = &m_icons[0];
+    else
+        tmp = &wxNullIcon;
+    return *tmp;
 }
 

 	  	 
