--- wxWindows/samples/menu/menu.cpp	2003-07-11 07:44:09.000000000 +0200
+++ wxWin/samples/menu/menu.cpp	2003-09-11 12:16:08.000000000 +0200
@@ -878,3 +878,3 @@
 
-void MyFrame::OnTestNormal(wxCommandEvent& event)
+void MyFrame::OnTestNormal(wxCommandEvent& WXUNUSED(event))
 {
@@ -921,3 +921,3 @@
 
-void MyFrame::OnSize(wxSizeEvent& event)
+void MyFrame::OnSize(wxSizeEvent& WXUNUSED(event))
 {

 	  	 
