--- src/mac/carbon/toplevel.cpp	2007-04-26 13:27:18.000000000 -0400
+++ src/mac/carbon/toplevel.cpp	2007-04-20 12:23:15.000000000 -0400
@@ -249,6 +249,7 @@
 wxWindow* g_MacLastWindow = NULL ;
 
 EventMouseButton g_lastButton = 0 ;
+bool g_lastButtonWasFakeRight = false ;
 
 void SetupMouseEvent( wxMouseEvent &wxevent , wxMacCarbonEvent &cEvent )
 {
@@ -270,8 +271,12 @@
     wxevent.SetTimestamp( cEvent.GetTicks() ) ;
 
    // a control click is interpreted as a right click
+    bool thisButtonIsFakeRight = false ;
     if ( button == kEventMouseButtonPrimary && (modifiers & controlKey) )
+    {
         button = kEventMouseButtonSecondary ;
+        thisButtonIsFakeRight = true ;
+    }
 
     // otherwise we report double clicks by connecting a left click with a ctrl-left click
     if ( clickCount > 1 && button != g_lastButton )
@@ -281,11 +286,17 @@
     // mouse down, moved and mouse up, and does not deliver a right down and left up
 
     if ( cEvent.GetKind() == kEventMouseDown )
+    {
         g_lastButton = button ;
+        g_lastButtonWasFakeRight = thisButtonIsFakeRight ;
+    }
 
     if ( button == 0 )
+    {
         g_lastButton = 0 ;
-    else if ( g_lastButton )
+        g_lastButtonWasFakeRight = false ;
+    }
+    else if ( g_lastButton == kEventMouseButtonSecondary && g_lastButtonWasFakeRight )
         button = g_lastButton ;
 
     // determine the correct down state, wx does not want a 'down' for a mouseUp event,

 	  	 
