Index: src/mac/carbon/app.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/mac/carbon/app.cpp,v
retrieving revision 1.189
diff -b -u -2 -r1.189 app.cpp
--- src/mac/carbon/app.cpp	2005/10/03 09:28:01	1.189
+++ src/mac/carbon/app.cpp	2005/11/01 03:29:20
@@ -458,6 +458,12 @@
         GetMenuItemRefCon( command.menu.menuRef , command.menu.menuItemIndex , &refCon ) ;
         // make sure it is one of our own menus, otherwise don't touch
+        // Note that we do -NOT- do this for predefined menu ids since that would
+        // break the functionality of them
         itsMenu = wxFindMenuFromMacMenu( command.menu.menuRef ) ;
-        if ( itsMenu != NULL )
+        if ( itsMenu != NULL || id == (MenuCommand) wxApp::s_macAboutMenuItemId ||
+                ( UMAGetSystemVersion() >= 0x1000 && (
+                    id == (MenuCommand) wxApp::s_macPreferencesMenuItemId ||
+                    id == (MenuCommand) wxApp::s_macExitMenuItemId ) )
+           )
         {
             item = (wxMenuItem*) refCon ;

 	  	 
