Index: src/msw/progdlg.cpp
===================================================================
--- src/msw/progdlg.cpp	(revision 65528)
+++ src/msw/progdlg.cpp	(working copy)
@@ -402,8 +402,11 @@
         // We only get here when we need to wait for the dialog to terminate so
         // do just this by running a custom event loop until the dialog is
         // dismissed.
-        wxProgressDialogModalLoop loop(*m_sharedData);
-        loop.Run();
+        if( !HasPDFlag(wxPD_AUTO_HIDE))
+        {
+            wxProgressDialogModalLoop loop(*m_sharedData);
+            loop.Run();
+        }
         return true;
     }
 #endif // wxHAS_MSW_TASKDIALOG
