--- wxWidgets-2.5.4-orig/src/msw/tglbtn.cpp.orig	2004-11-22 15:55:00 +0200
+++ wxWidgets-2.5.4/src/msw/tglbtn.cpp	2005-03-17 08:36:32 +0200
@@ -126,11 +126,14 @@
    int hBtn = BUTTON_HEIGHT_FROM_CHAR_HEIGHT(hChar);
 
 #if wxUSE_BUTTON
-   wxSize sz = wxButton::GetDefaultSize();
-   if (wBtn > sz.x)
+   wxSize sz(wBtn, hBtn);
+   if (!HasFlag(wxBU_EXACTFIT)) {
+       sz = wxButton::GetDefaultSize();
+       if (wBtn > sz.x)
        sz.x = wBtn;
-   if (hBtn > sz.y)
-       sz.y = hBtn;
+       if (hBtn > sz.y)
+           sz.y = hBtn;
+   }
 #else
    wxSize sz(wBtn, hBtn);
 #endif

 	  	 
