--- src/generic/listctrl-bu.cpp	2008-05-19 21:48:47.000000000 +0200
+++ src/generic/listctrl.cpp	2008-08-08 19:57:19.000000000 +0200
@@ -1896,6 +1896,19 @@
 
         x += wCol;
     }
+    
+    // Fill in what's missing to the right of the columns, otherwise we will 
+    // leave an unpainted area when columns are removed (and it looks better)
+    if( x<w )
+    {
+        wxRendererNative::Get().DrawHeaderButton
+                                (
+                                    this,
+                                    dc,
+                                    wxRect(x, HEADER_OFFSET_Y, w-x, h),
+                                    0
+                                );
+    }
 }
 
 void wxListHeaderWindow::DrawCurrent()
