From fdcc51691e2f315f377e84ccae4e11d07c888524 Mon Sep 17 00:00:00 2001
From: Sebastian Walderich <sebastian@walderich.de>
Date: Wed, 7 Nov 2012 16:39:59 +0100
Subject: [PATCH] Made some virtual items bold

---
 samples/listctrl/listtest.cpp |    9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/samples/listctrl/listtest.cpp b/samples/listctrl/listtest.cpp
index c7fbde5..b12b63e 100644
--- a/samples/listctrl/listtest.cpp
+++ b/samples/listctrl/listtest.cpp
@@ -1321,6 +1321,14 @@ int MyListCtrl::OnGetItemColumnImage(long item, long column) const
 
 wxListItemAttr *MyListCtrl::OnGetItemAttr(long item) const
 {
+#if 1
+        static wxListItemAttr s_attrHighlight(*wxRED, wxNullColour, wxNullFont);
+        wxFont fnt(*wxNORMAL_FONT);
+        fnt.MakeBold();
+        s_attrHighlight.SetFont(fnt);
+
+    return item % 2 ? NULL : &s_attrHighlight;
+#else
     // test to check that RefreshItem() works correctly: when m_updated is
     // set to some item and it is refreshed, we highlight the item
     if ( item == m_updated )
@@ -1330,6 +1338,7 @@ wxListItemAttr *MyListCtrl::OnGetItemAttr(long item) const
     }
 
     return item % 2 ? NULL : (wxListItemAttr *)&m_attr;
+#endif
 }
 
 void MyListCtrl::InsertItemInReportView(int i)
-- 
1.7.10.msysgit.1

