Index: src/common/fileconf.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/common/fileconf.cpp,v
retrieving revision 1.144
diff -b -u -2 -r1.144 fileconf.cpp
--- src/common/fileconf.cpp	2006/01/20 00:55:17	1.144
+++ src/common/fileconf.cpp	2006/02/08 15:21:41
@@ -1161,7 +1161,4 @@
           return false;
       }
-
-      m_strLocalFile =
-      m_strGlobalFile = wxEmptyString;
   }
 

 	  	 
