--- listtest_orig.cpp	2016-02-28 22:33:38.000000000 +0100
+++ listtest.cpp	2016-03-10 17:12:51.603459400 +0100
@@ -285,6 +285,9 @@
     SetMenuBar(menubar);
 
     m_panel = new wxPanel(this, wxID_ANY);
+    
+    wxBoxSizer *vbox = new wxBoxSizer(wxVERTICAL);
+    
     m_logWindow = new wxTextCtrl(m_panel, wxID_ANY, wxEmptyString,
                                  wxDefaultPosition, wxDefaultSize,
                                  wxTE_READONLY | wxTE_MULTILINE | wxSUNKEN_BORDER);
@@ -293,6 +296,12 @@
 
     RecreateList(wxLC_REPORT | wxLC_SINGLE_SEL);
 
+    vbox->Add(m_listCtrl, 1, wxEXPAND|wxALL, 16);
+    vbox->Add(m_logWindow, 1, wxEXPAND|wxALL, 16);  
+
+    m_panel->SetSizer(vbox);		
+    SetClientSize(m_panel->GetBestSize()); 
+            
 #ifdef __WXMSW__
     // this is useful to know specially when debugging :)
     wxLogMessage("Your version of comctl32.dll is: %d",
@@ -314,7 +323,7 @@
 
 void MyFrame::OnSize(wxSizeEvent& event)
 {
-    DoSize();
+//    DoSize();
 
     event.Skip();
 }
@@ -459,7 +468,7 @@
         delete m_listCtrl;
 
         m_listCtrl = new MyListCtrl(m_panel, LIST_CTRL,
-                                    wxDefaultPosition, wxDefaultSize,
+                                    wxDefaultPosition, wxSize(1000, 400),
                                     flags |
                                     wxBORDER_THEME | wxLC_EDIT_LABELS);
 
@@ -493,7 +502,7 @@
             m_listCtrl->EnableBellOnNoMatch(mb->IsChecked(LIST_TOGGLE_BELL));
     }
 
-    DoSize();
+//    DoSize();
 
     GetMenuBar()->Check(LIST_ROW_LINES, false);
 
