Index: src/msw/dc.cpp
===================================================================
--- src/msw/dc.cpp	(révision 70954)
+++ src/msw/dc.cpp	(copie de travail)
@@ -2725,6 +2725,40 @@
 
     static AlphaBlend_t
         pfnAlphaBlend = (AlphaBlend_t)wxMSIMG32DLL.GetSymbol(wxT("AlphaBlend"));
+           
+    // We have to pre multiply the source bitmap red,green and blue components by their alpha
+    // component before feeding the bitmap to the AlphaBlend() function.
+    // We create and intermediate bitmap as a buffer to do this
+    wxBitmap bmpSrc(srcWidth, srcHeight, 32);
+    MemoryHDC hdcMem;
+    SelectInHDC select(hdcMem, GetHbitmapOf(bmpSrc));
+
+    if ( !::BitBlt(hdcMem, 0, 0, srcWidth, srcHeight, hdcSrc, srcX, srcY, SRCCOPY) )
+    {
+        wxLogLastError(wxT("BitBlt"));
+    } else {
+
+        wxAlphaPixelData dataSrc(bmpSrc);
+        wxAlphaPixelData::Iterator pSrc(dataSrc);
+
+        for ( int y = 0; y < srcHeight; y++ )
+        {
+            wxAlphaPixelData::Iterator pSrcRowStart = pSrc;
+
+            for ( int x = 0; x < srcWidth; x++ )
+            {
+                int alpha = pSrc.Alpha();
+                pSrc.Red()   = (pSrc.Red()*alpha)/255;
+                pSrc.Green() = (pSrc.Green()*alpha)/255;
+                pSrc.Blue()  = (pSrc.Blue()*alpha)/255;
+                ++pSrc;
+            }
+
+            pSrc = pSrcRowStart;
+            pSrc.OffsetY(dataSrc, 1);
+        }       
+    }
+                    
     if ( pfnAlphaBlend )
     {
         BLENDFUNCTION bf;
@@ -2734,7 +2768,7 @@
         bf.AlphaFormat = AC_SRC_ALPHA;
 
         if ( pfnAlphaBlend(hdcDst, x, y, dstWidth, dstHeight,
-                           hdcSrc, srcX, srcY, srcWidth, srcHeight,
+                           hdcMem, 0, 0, srcWidth, srcHeight,
                            bf) )
         {
             // skip wxAlphaBlend() call below
@@ -2750,7 +2784,7 @@
     // AlphaBlend() unavailable of failed: use our own (probably much slower)
     // implementation
 #ifdef wxHAS_RAW_BITMAP
-    wxAlphaBlend(hdcDst, x, y, dstWidth, dstHeight, srcX, srcY, srcWidth, srcHeight, bmp);
+    wxAlphaBlend(hdcDst, x, y, dstWidth, dstHeight, 0, 0, srcWidth, srcHeight, bmpSrc);
 
     return true;
 #else // !wxHAS_RAW_BITMAP
@@ -2810,6 +2844,7 @@
             pDst.Red() = pSrc.Red() + (beta * pDst.Red() + 127) / 255;
             pDst.Blue() = pSrc.Blue() + (beta * pDst.Blue() + 127) / 255;
             pDst.Green() = pSrc.Green() + (beta * pDst.Green() + 127) / 255;
+            pDst.Alpha() = pSrc.Alpha() + (beta * pDst.Alpha() + 127) / 255;
 
             ++pDst;
         }
