Index: src/msw/gdiimage.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/msw/gdiimage.cpp,v
retrieving revision 1.54
diff --unified -r1.54 gdiimage.cpp
--- src/msw/gdiimage.cpp	2006/06/14 10:02:07	1.54
+++ src/msw/gdiimage.cpp	2006/08/17 11:33:57
@@ -572,7 +572,7 @@
     wxSize size(::GetSystemMetrics(SM_CXICON), ::GetSystemMetrics(SM_CYICON));
 
 #ifndef __WXWINCE__
-    if ( hicon && wxGetOsVersion() != wxWIN32S )
+    if ( hicon && wxGetOsVersion() != wxOS_WINDOWS_9X )
     {
         ICONINFO info;
         if ( !::GetIconInfo(hicon, &info) )
Index: src/msw/utils.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/msw/utils.cpp,v
retrieving revision 1.192
diff --unified -r1.192 utils.cpp
--- src/msw/utils.cpp	2006/08/17 03:02:15	1.192
+++ src/msw/utils.cpp	2006/08/17 11:33:59
@@ -1265,7 +1265,11 @@
                     return wxWinVersion_NT6;
             }
             break;
-
+            
+        // To silence GCC's warnings about the remaining
+        // enumeration values not handled in the switch
+        default:
+            break;
     }
 
     return wxWinVersion_Unknown;

 	  	 
