Index: setup.h.in
===================================================================
--- setup.h.in	(revision 48645)
+++ setup.h.in	(working copy)
@@ -563,79 +563,111 @@
 /*
  * Supports bool type
  */
+#ifndef HAVE_BOOL
 #undef HAVE_BOOL
+#endif
 
 /*
  * Define if your compiler supports the explicit keyword
  */
+#ifndef HAVE_EXPLICIT
 #undef HAVE_EXPLICIT
+#endif
 
 /*
  * Define if your compiler supports const_cast<>()
  */
+#ifndef HAVE_CONST_CAST
 #undef HAVE_CONST_CAST
+#endif
 
 /*
  * Define if your compiler supports reinterpret_cast<>()
  */
+#ifndef HAVE_REINTERPRET_CAST
 #undef HAVE_REINTERPRET_CAST
+#endif
 
 /*
  * Define if your compiler supports static_cast<>()
  */
+#ifndef HAVE_STATIC_CAST
 #undef HAVE_STATIC_CAST
+#endif
 
 /*
  * Define if your compiler has C99 va_copy
  */
+#ifndef HAVE_VA_COPY
 #undef HAVE_VA_COPY
+#endif
 
 /*
  * Define if va_list type is an array
  */
+#ifndef VA_LIST_IS_ARRAY
 #undef VA_LIST_IS_ARRAY
+#endif
 
 /*
  * Define if the compiler supports variadic macros
  */
+#ifndef HAVE_VARIADIC_MACROS
 #undef HAVE_VARIADIC_MACROS
+#endif
 
 /*
  * Define if your compiler has std::wstring
  */
+#ifndef HAVE_STD_WSTRING
 #undef HAVE_STD_WSTRING
+#endif
 /*
  * Define if your compiler has compliant std::string::compare
  */
+#ifndef HAVE_STD_STRING_COMPARE
 #undef HAVE_STD_STRING_COMPARE
+#endif
 /*
  * Define if your compiler has <hash_map>
  */
+#ifndef HAVE_HASH_MAP
 #undef HAVE_HASH_MAP
+#endif
 /*
  * Define if your compiler has <ext/hash_map>
  */
+#ifndef HAVE_EXT_HASH_MAP
 #undef HAVE_EXT_HASH_MAP
+#endif
 /*
  * Define if your compiler has std::hash_map/hash_set
  */
+#ifndef HAVE_STD_HASH_MAP
 #undef HAVE_STD_HASH_MAP
+#endif
 /*
  * Define if your compiler has __gnu_cxx::hash_map/hash_set
  */
+#ifndef HAVE_GNU_CXX_HASH_MAP
 #undef HAVE_GNU_CXX_HASH_MAP
+#endif
 
 /*
  * The built-in regex supports advanced REs in additional to POSIX's basic
  * and extended. Your system regex probably won't support this, and in this
  * case WX_NO_REGEX_ADVANCED should be defined.
  */
+#ifndef WX_NO_REGEX_ADVANCED
 #undef WX_NO_REGEX_ADVANCED
+#endif
 /*
  * On GNU systems use re_search instead of regexec, since the latter does a
  * strlen on the search text affecting the performance of some operations.
  */
+#ifndef HAVE_RE_SEARCH
 #undef HAVE_RE_SEARCH
+#endif
 /*
  * Use SDL for audio (Unix)
  */
@@ -781,7 +813,9 @@
 #undef WX_SOCKLEN_T
 
 /* The type of 5th argument to getsockopt() - usually size_t or int */
+#ifndef SOCKOPTLEN_T
 #undef SOCKOPTLEN_T
+#endif
 
 /* The type of statvfs(2) argument */
 #undef WX_STATFS_T
@@ -797,148 +831,242 @@
 #undef WX_GMTOFF_IN_TM
 
 /* Define if you have poll(2) function */
+#ifndef HAVE_POLL
 #undef HAVE_POLL
+#endif
 
 /* Define if you have pw_gecos field in struct passwd */
+#ifndef HAVE_PW_GECOS
 #undef HAVE_PW_GECOS
+#endif
 
 /* Define if you have __cxa_demangle() in <cxxabi.h> */
+#ifndef HAVE_CXA_DEMANGLE
 #undef HAVE_CXA_DEMANGLE
+#endif
 
 /* Define if you have dlopen() */
+#ifndef HAVE_DLOPEN
 #undef HAVE_DLOPEN
+#endif
 
 /* Define if you have gettimeofday() */
+#ifndef HAVE_GETTIMEOFDAY
 #undef HAVE_GETTIMEOFDAY
+#endif
 
 /* Define if fsync() is available */
+#ifndef HAVE_FSYNC
 #undef HAVE_FSYNC
+#endif
 
 /* Define if you have ftime() */
+#ifndef HAVE_FTIME
 #undef HAVE_FTIME
+#endif
 
 /* Define if round() is available */
+#ifndef HAVE_ROUND
 #undef HAVE_ROUND
+#endif
 
 /* Define if you have support for large (64 bit size) files */
+#ifndef HAVE_LARGEFILE_SUPPORT
 #undef HAVE_LARGEFILE_SUPPORT
+#endif
 
 /* Define if you have localtime() */
+#ifndef HAVE_LOCALTIME
 #undef HAVE_LOCALTIME
+#endif
 
 /* Define if you have nanosleep() */
+#ifndef HAVE_NANOSLEEP
 #undef HAVE_NANOSLEEP
+#endif
 
 /* Define if you have sched_yield */
+#ifndef HAVE_SCHED_YIELD
 #undef HAVE_SCHED_YIELD
+#endif
 
 /* Define if you have pthread_mutexattr_t and functions to work with it */
+#ifndef HAVE_PTHREAD_MUTEXATTR_T
 #undef HAVE_PTHREAD_MUTEXATTR_T
+#endif
 
 /* Define if you have pthread_mutexattr_settype() declaration */
+#ifndef HAVE_PTHREAD_MUTEXATTR_SETTYPE_DECL
 #undef HAVE_PTHREAD_MUTEXATTR_SETTYPE_DECL
+#endif
 
 /* Define if you have PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP */
+#ifndef HAVE_PTHREAD_RECURSIVE_MUTEX_INITIALIZER
 #undef HAVE_PTHREAD_RECURSIVE_MUTEX_INITIALIZER
+#endif
 
 /* Define if you have pthread_cancel */
+#ifndef HAVE_PTHREAD_CANCEL
 #undef HAVE_PTHREAD_CANCEL
+#endif
 
 /* Define if you have pthread_attr_setstacksize */
+#ifndef HAVE_PTHREAD_ATTR_SETSTACKSIZE
 #undef HAVE_PTHREAD_ATTR_SETSTACKSIZE
+#endif
 
 /* Define if you have sigaction() */
+#ifndef HAVE_SIGACTION
 #undef HAVE_SIGACTION
+#endif
 
 /* Define if you have sigemptyset() */
+#ifndef HAVE_SIGEMPTYSET
 #undef HAVE_SIGEMPTYSET
+#endif
 
 /* Define if you have shl_load() */
+#ifndef HAVE_SHL_LOAD
 #undef HAVE_SHL_LOAD
+#endif
 
 /* Define if you have snprintf() */
+#ifndef HAVE_SNPRINTF
 #undef HAVE_SNPRINTF
+#endif
 
 /* Define if you have snprintf() declaration in the header */
+#ifndef HAVE_SNPRINTF_DECL
 #undef HAVE_SNPRINTF_DECL
+#endif
 
 /* Define if you have a _broken_ snprintf() declaration in the header,
  * with 'char*' for the 3rd parameter instead of 'const char*' */
+#ifndef HAVE_BROKEN_SNPRINTF_DECL
 #undef HAVE_BROKEN_SNPRINTF_DECL
+#endif
 
 /* Define if you have a snprintf() which supports positional arguments
    (defined in the unix98 standard) */
+#ifndef HAVE_UNIX98_PRINTF
 #undef HAVE_UNIX98_PRINTF
+#endif
 
 /* define if you have statfs function */
+#ifndef HAVE_STATFS
 #undef HAVE_STATFS
+#endif
 
 /* define if you have statfs prototype */
+#ifndef HAVE_STATFS_DECL
 #undef HAVE_STATFS_DECL
+#endif
 
 /* define if you have statvfs function */
+#ifndef HAVE_STATVFS
 #undef HAVE_STATVFS
+#endif
 
 /* Define if you have strptime() */
+#ifndef HAVE_STRPTIME
 #undef HAVE_STRPTIME
+#endif
 
 /* Define if strptime() is declared in headers */
+#ifndef HAVE_STRPTIME_DECL
 #undef HAVE_STRPTIME_DECL
+#endif
 
 /* Define if you have strtoull() and strtoll() */
+#ifndef HAVE_STRTOULL
 #undef HAVE_STRTOULL
+#endif
 
 /* Define if you have all functions to set thread priority */
+#ifndef HAVE_THREAD_PRIORITY_FUNCTIONS
 #undef HAVE_THREAD_PRIORITY_FUNCTIONS
+#endif
 
 /* Define if you have timegm() function */
+#ifndef HAVE_TIMEGM
 #undef HAVE_TIMEGM
+#endif
 
 /* Define if you have vsnprintf() */
+#ifndef HAVE_VSNPRINTF
 #undef HAVE_VSNPRINTF
+#endif
 
 /* Define if you have vsnprintf() declaration in the header */
+#ifndef HAVE_VSNPRINTF_DECL
 #undef HAVE_VSNPRINTF_DECL
+#endif
 
 /* Define if you have a _broken_ vsnprintf() declaration in the header,
  * with 'char*' for the 3rd parameter instead of 'const char*' */
+#ifndef HAVE_BROKEN_VSNPRINTF_DECL
 #undef HAVE_BROKEN_VSNPRINTF_DECL
+#endif
 
 /* Define if you have usleep() */
+#ifndef HAVE_USLEEP
 #undef HAVE_USLEEP
+#endif
 
 /* Define if you have wcslen function  */
+#ifndef HAVE_WCSLEN
 #undef HAVE_WCSLEN
+#endif
 
 /* Define if you have wcstoull() and wcstoll() */
+#ifndef HAVE_WCSTOULL
 #undef HAVE_WCSTOULL
+#endif
 
 /* define if you have vsscanf function */
+#ifndef HAVE_VSSCANF
 #undef HAVE_VSSCANF
+#endif
 
 /* The number of bytes in a char.  */
+#ifndef SIZEOF_CHAR
 #undef SIZEOF_CHAR
+#endif
 
 /* The number of bytes in a wchar_t.  */
+#ifndef SIZEOF_WCHAR_T
 #undef SIZEOF_WCHAR_T
+#endif
 
 /* The number of bytes in a int.  */
+#ifndef SIZEOF_INT
 #undef SIZEOF_INT
+#endif
 
 /* The number of bytes in a pointer.  */
+#ifndef SIZEOF_VOID_P
 #undef SIZEOF_VOID_P
+#endif
 
 /* The number of bytes in a long.  */
+#ifndef SIZEOF_LONG
 #undef SIZEOF_LONG
+#endif
 
 /* The number of bytes in a long long.  */
+#ifndef SIZEOF_LONG_LONG
 #undef SIZEOF_LONG_LONG
+#endif
 
 /* The number of bytes in a short.  */
+#ifndef SIZEOF_SHORT
 #undef SIZEOF_SHORT
+#endif
 
 /* The number of bytes in a size_t.  */
+#ifndef SIZEOF_SIZE_T
 #undef SIZEOF_SIZE_T
+#endif
 
 /* Define if size_t on your machine is the same type as unsigned int. */
 #undef wxSIZE_T_IS_UINT
@@ -947,218 +1075,358 @@
 #undef wxSIZE_T_IS_ULONG
 
 /* Define if you have the dlopen function.  */
+#ifndef HAVE_DLOPEN
 #undef HAVE_DLOPEN
+#endif
 
 /* Define if you have the dlerror function.  */
+#ifndef HAVE_DLERROR
 #undef HAVE_DLERROR
+#endif
 
 /* Define if you have Posix fnctl() function. */
+#ifndef HAVE_FCNTL
 #undef HAVE_FCNTL
+#endif
 
 /* Define if you have BSD flock() function. */
+#ifndef HAVE_FLOCK
 #undef HAVE_FLOCK
+#endif
 
 /* Define if you have getaddrinfo function. */
+#ifndef HAVE_GETADDRINFO
 #undef HAVE_GETADDRINFO
+#endif
 
 /* Define if you have a gethostbyname_r function taking 6 arguments. */
+#ifndef HAVE_FUNC_GETHOSTBYNAME_R_6
 #undef HAVE_FUNC_GETHOSTBYNAME_R_6
+#endif
 
 /* Define if you have a gethostbyname_r function taking 5 arguments. */
+#ifndef HAVE_FUNC_GETHOSTBYNAME_R_5
 #undef HAVE_FUNC_GETHOSTBYNAME_R_5
+#endif
 
 /* Define if you have a gethostbyname_r function taking 3 arguments. */
+#ifndef HAVE_FUNC_GETHOSTBYNAME_R_3
 #undef HAVE_FUNC_GETHOSTBYNAME_R_3
+#endif
 
 /* Define if you only have a gethostbyname function */
+#ifndef HAVE_GETHOSTBYNAME
 #undef HAVE_GETHOSTBYNAME
+#endif
 
 /* Define if you have the gethostname function.  */
+#ifndef HAVE_GETHOSTNAME
 #undef HAVE_GETHOSTNAME
+#endif
 
 /* Define if you have a getservbyname_r function taking 6 arguments. */
+#ifndef HAVE_FUNC_GETSERVBYNAME_R_6
 #undef HAVE_FUNC_GETSERVBYNAME_R_6
+#endif
 
 /* Define if you have a getservbyname_r function taking 5 arguments. */
+#ifndef HAVE_FUNC_GETSERVBYNAME_R_5
 #undef HAVE_FUNC_GETSERVBYNAME_R_5
+#endif
 
 /* Define if you have a getservbyname_r function taking 4 arguments. */
+#ifndef HAVE_FUNC_GETSERVBYNAME_R_4
 #undef HAVE_FUNC_GETSERVBYNAME_R_4
+#endif
 
 /* Define if you only have a getservbyname function */
+#ifndef HAVE_GETSERVBYNAME
 #undef HAVE_GETSERVBYNAME
+#endif
 
 /* Define if you have the gmtime_r function.  */
+#ifndef HAVE_GMTIME_R
 #undef HAVE_GMTIME_R
+#endif
 
 /* Define if you have the inet_addr function.  */
+#ifndef HAVE_INET_ADDR
 #undef HAVE_INET_ADDR
+#endif
 
 /* Define if you have the inet_aton function.  */
+#ifndef HAVE_INET_ATON
 #undef HAVE_INET_ATON
+#endif
 
 /* Define if you have the localtime_r function.  */
+#ifndef HAVE_LOCALTIME_R
 #undef HAVE_LOCALTIME_R
+#endif
 
 /* Define if you have the mktemp function.  */
+#ifndef HAVE_MKTEMP
 #undef HAVE_MKTEMP
+#endif
 
 /* Define if you have the mkstemp function.  */
+#ifndef HAVE_MKSTEMP
 #undef HAVE_MKSTEMP
+#endif
 
 /* Define if you have the nanosleep function.  */
+#ifndef HAVE_NANOSLEEP
 #undef HAVE_NANOSLEEP
+#endif
 
 /* Define if you have the putenv function.  */
+#ifndef HAVE_PUTENV
 #undef HAVE_PUTENV
+#endif
 
 /* Define if you have the readdir_r function.  */
+#ifndef HAVE_READDIR_R
 #undef HAVE_READDIR_R
+#endif
 
 /* Define if you have the setenv function.  */
+#ifndef HAVE_SETENV
 #undef HAVE_SETENV
+#endif
 
 /* Define if you have the shl_load function.  */
+#ifndef HAVE_SHL_LOAD
 #undef HAVE_SHL_LOAD
+#endif
 
 /* Define if you have strtok_r function. */
+#ifndef HAVE_STRTOK_R
 #undef HAVE_STRTOK_R
+#endif
 
 /* Define if you have thr_setconcurrency function */
+#ifndef HAVE_THR_SETCONCURRENCY
 #undef HAVE_THR_SETCONCURRENCY
+#endif
 
 /* Define if you have the uname function.  */
+#ifndef HAVE_UNAME
 #undef HAVE_UNAME
+#endif
 
 /* Define if you have the unsetenv function.  */
 #undef HAVE_UNSETENV
 
 /* Define if you have the usleep function.  */
+#ifndef HAVE_USLEEP
 #undef HAVE_USLEEP
+#endif
 
 /* Define if you have the <X11/XKBlib.h> header file.  */
+#ifndef HAVE_X11_XKBLIB_H
 #undef HAVE_X11_XKBLIB_H
+#endif
 
 /* Define if you have the <X11/extensions/xf86vmode.h> header file.  */
+#ifndef HAVE_X11_EXTENSIONS_XF86VMODE_H
 #undef HAVE_X11_EXTENSIONS_XF86VMODE_H
+#endif
 
 /* Define if you have the fnmatch() function */
+#ifndef HAVE_FNMATCH
 #undef HAVE_FNMATCH
+#endif
 
 /* Define if you have the <fnmatch.h> header file.  */
+#ifndef HAVE_FNMATCH_H
 #undef HAVE_FNMATCH_H
+#endif
 
 /* Define if you have the <iostream> header file.  */
+#ifndef HAVE_IOSTREAM
 #undef HAVE_IOSTREAM
+#endif
 
 /* Define if you have the <linux/joystick.h> header file.  */
+#ifndef HAVE_LINUX_JOYSTICK_H
 #undef HAVE_LINUX_JOYSTICK_H
+#endif
 
 /* Define if you have the <sched.h> header file.  */
+#ifndef HAVE_SCHED_H
 #undef HAVE_SCHED_H
+#endif
 
 /* Define if you have the <unistd.h> header file.  */
+#ifndef HAVE_UNISTD_H
 #undef HAVE_UNISTD_H
+#endif
 
 /* Define if you have the <fcntl.h> header file.  */
+#ifndef HAVE_FCNTL_H
 #undef HAVE_FCNTL_H
+#endif
 
 /* Define if you have the <wchar.h> header file.  */
+#ifndef HAVE_WCHAR_H
 #undef HAVE_WCHAR_H
+#endif
 
 /* Define if you have the <wcstr.h> header file.  */
+#ifndef HAVE_WCSTR_H
 #undef HAVE_WCSTR_H
+#endif
 
 /* Define if you have the <wctype.h> header file.  */
+#ifndef HAVE_WCTYPE_H
 #undef HAVE_WCTYPE_H
+#endif
 
 /* Define if you have <widec.h> (Solaris only) */
+#ifndef HAVE_WIDEC_H
 #undef HAVE_WIDEC_H
+#endif
 
 /* Define if you have the <iconv.h> header file and iconv() symbol.  */
+#ifndef HAVE_ICONV
 #undef HAVE_ICONV
+#endif
 
 /* Define as "const" if the declaration of iconv() needs const.  */
+#ifndef ICONV_CONST
 #undef ICONV_CONST
+#endif
 
 /* Define if you have the <langinfo.h> header file.  */
+#ifndef HAVE_LANGINFO_H
 #undef HAVE_LANGINFO_H
+#endif
 
 /* Define if you have the <w32api.h> header file (mingw,cygwin).  */
+#ifndef HAVE_W32API_H
 #undef HAVE_W32API_H
+#endif
 
 /* Define if you have the <esd.h> header file.  */
+#ifndef HAVE_ESD_H
 #undef HAVE_ESD_H
+#endif
 
 /* Define if you have the <sys/soundcard.h> header file. */
+#ifndef HAVE_SYS_SOUNDCARD_H
 #undef HAVE_SYS_SOUNDCARD_H
+#endif
 
 /* Define if you have a CD-ROM interface known to mmedia.  */
+#ifndef HAVE_KNOWN_CDROM_INTERFACE
 #undef HAVE_KNOWN_CDROM_INTERFACE
+#endif
 
 /* Define if you have wcsrtombs() function */
+#ifndef HAVE_WCSRTOMBS
 #undef HAVE_WCSRTOMBS
+#endif
 
 /* Define this if you have wputc() */
+#ifndef HAVE_WPUTC
 #undef HAVE_WPUTC
+#endif
 
 /* Define this if you have wputchar() */
+#ifndef HAVE_WPUTCHAR
 #undef HAVE_WPUTCHAR
+#endif
 
 /* Define this if you have putws() */
+#ifndef HAVE_PUTWS
 #undef HAVE_PUTWS
+#endif
 
 /* Define this if you have fputws() */
+#ifndef HAVE_FPUTWS
 #undef HAVE_FPUTWS
+#endif
 
 /* Define this if you have strcasecmp() function in <string.h> */
+#ifndef HAVE_STRCASECMP_IN_STRING_H
 #undef HAVE_STRCASECMP_IN_STRING_H
+#endif
 
 /* Define this if you have strcasecmp() function in <strings.h> */
+#ifndef HAVE_STRCASECMP_IN_STRINGS_H
 #undef HAVE_STRCASECMP_IN_STRINGS_H
+#endif
 
 /* Define this if you have wprintf() and related functions */
+#ifndef HAVE_WPRINTF
 #undef HAVE_WPRINTF
+#endif
 
 /* Define this if you have swprintf() */
+#ifndef HAVE_SWPRINTF
 #undef HAVE_SWPRINTF
+#endif
 
 /* Define if you have a _broken_ swprintf() declaration in the header,
  * without a 'size_t max' parameter as second argument */
+#ifndef HAVE_BROKEN_SWPRINTF_DECL
 #undef HAVE_BROKEN_SWPRINTF_DECL
+#endif
 
 /* Define this if you have vswprintf() and related functions */
+#ifndef HAVE_VSWPRINTF
 #undef HAVE_VSWPRINTF
+#endif
 
 /* Define this if you have _vsnwprintf */
+#ifndef HAVE__VSNWPRINTF
 #undef HAVE__VSNWPRINTF
+#endif
 
 /* Define if fseeko and ftello are available.  */
+#ifndef HAVE_FSEEKO
 #undef HAVE_FSEEKO
+#endif
 
 /* Define this if you are using gtk and gdk contains support for X11R6 XIM */
+#ifndef HAVE_XIM
 #undef HAVE_XIM
+#endif
 
 /* Define this if you have X11/extensions/shape.h */
+#ifndef HAVE_XSHAPE
 #undef HAVE_XSHAPE
+#endif
 
 /* Define this if you have type SPBCDATA */
+#ifndef HAVE_SPBCDATA
 #undef HAVE_SPBCDATA
+#endif
 
 /* Define if you have pango_font_family_is_monospace() (Pango >= 1.3.3) */
+#ifndef HAVE_PANGO_FONT_FAMILY_IS_MONOSPACE
 #undef HAVE_PANGO_FONT_FAMILY_IS_MONOSPACE
+#endif
 
 /* Define if gtk_icon_size_lookup is missing from the GTK+ headers */
+#ifndef NEED_GTK_ICON_SIZE_LOOKUP
 #undef NEED_GTK_ICON_SIZE_LOOKUP
+#endif
 
 /* Define if you have Pango xft support */
+#ifndef HAVE_PANGO_XFT
 #undef HAVE_PANGO_XFT
+#endif
 
 /* Define if you have the <sys/select.h> header file.  */
+#ifndef HAVE_SYS_SELECT_H
 #undef HAVE_SYS_SELECT_H
+#endif
 
 /* Define if fdopen is available.  */
+#ifndef HAVE_FDOPEN
 #undef HAVE_FDOPEN
+#endif
 
 /* -------------------------------------------------------------------------
    Win32 adjustments section

 	  	 
