Index: gsockmsw.cpp
===================================================================
--- gsockmsw.cpp	(revision 50876)
+++ gsockmsw.cpp	(working copy)
@@ -335,8 +335,20 @@
 {
   /* Remove the socket from the list */
   EnterCriticalSection(&critical);
+
   if ( socket->IsOk() )
-      socketList[(socket->m_msgnumber - WM_USER)] = NULL;
+  {
+    // Remove pending window messages.
+    // Otherwise a new socket that reuses the same ID could end up
+    // receiving messages from the old one.
+    MSG msg;
+    while (PeekMessage(&msg, hWin, socket->m_msgnumber, socket->m_msgnumber, PM_REMOVE))
+    {
+      // Discard message
+    }
+
+    socketList[(socket->m_msgnumber - WM_USER)] = NULL;
+  }
   LeaveCriticalSection(&critical);
 }
 

 	  	 
