Index: src/mac/carbon/window.cpp
===================================================================
--- src/mac/carbon/window.cpp	(revision 1347)
+++ src/mac/carbon/window.cpp	(working copy)
@@ -910,6 +910,7 @@
     m_vScrollBar = NULL ;
     m_hScrollBarAlwaysShown = false;
     m_vScrollBarAlwaysShown = false;
+    m_growBox = NULL ;
 
     m_macIsUserPane = true;
     m_clipChildren = false ;
@@ -1056,6 +1057,8 @@
         m_vScrollBar->Raise() ;
     if ( m_hScrollBar )
         m_hScrollBar->Raise() ;
+    if ( m_growBox )
+        m_growBox->Raise() ;
 }
 
 void wxWindowMac::MacPostControlCreate(const wxPoint& WXUNUSED(pos), const wxSize& size)
@@ -2263,6 +2266,7 @@
 
     if ( MacHasScrollBarCorner() )
     {
+#if 0
         Rect rect ;
 
         CGContextRef cgContext = (CGContextRef) MacGetCGContextRef() ;
@@ -2285,6 +2289,14 @@
         }
         CGContextFillRect( cgContext, cgrect );
         CGContextRestoreGState( cgContext );
+#else
+        if (m_growBox) {
+             if ( m_backgroundColour.Ok() )
+                 m_growBox->SetBackgroundColour(m_backgroundColour);
+             else
+                 m_growBox->SetBackgroundColour(*wxWHITE);
+        }
+#endif
     }
 }
 
@@ -2357,6 +2369,8 @@
         m_hScrollBar = NULL ;
     if ( child == m_vScrollBar )
         m_vScrollBar = NULL ;
+    if ( child == m_growBox )
+        m_growBox = NULL ;
 
     wxWindowBase::RemoveChild( child ) ;
 }
@@ -2460,6 +2474,8 @@
             continue;
         if (child == m_hScrollBar)
             continue;
+        if (child == m_growBox)
+            continue;
         if (child->IsTopLevel())
             continue;
 
@@ -2806,6 +2822,8 @@
                 continue;
             if (child == m_hScrollBar)
                 continue;
+            if (child == m_growBox)
+                continue;
             if (child->IsTopLevel())
                 continue;
             if (!child->IsShown())
@@ -2956,6 +2974,10 @@
             m_hScrollBar = new wxScrollBar((wxWindow*)this, wxID_ANY, hPoint, hSize , wxHORIZONTAL);
             m_hScrollBar->SetMinSize( wxDefaultSize );
         }
+
+        wxPoint gPoint(width - scrlsize, height - scrlsize);
+        wxSize gSize(scrlsize, scrlsize);
+        m_growBox = new wxPanel((wxWindow *)this, wxID_ANY, gPoint, gSize, 0);
     }
 
     // because the create does not take into account the client area origin
@@ -2965,7 +2987,7 @@
 
 bool wxWindowMac::MacIsChildOfClientArea( const wxWindow* child ) const
 {
-    bool result = ((child == NULL) || ((child != m_hScrollBar) && (child != m_vScrollBar)));
+    bool result = ((child == NULL) || ((child != m_hScrollBar) && (child != m_vScrollBar) && (child != m_growBox)));
 
     return result ;
 }
@@ -3038,6 +3060,17 @@
         m_vScrollBar->SetSize( vPoint.x , vPoint.y, vSize.x, vSize.y , wxSIZE_ALLOW_MINUS_ONE );
     if ( m_hScrollBar )
         m_hScrollBar->SetSize( hPoint.x , hPoint.y, hSize.x, hSize.y, wxSIZE_ALLOW_MINUS_ONE );
+    if ( m_growBox ) {
+        if ( MacHasScrollBarCorner() ) {
+            m_growBox->SetSize( width - scrlsize, height - scrlsize, wxDefaultCoord, wxDefaultCoord, wxSIZE_USE_EXISTING );
+            if ( !m_growBox->IsShown() )
+                m_growBox->Show();
+        }
+        else {
+            if ( m_growBox->IsShown() )
+                m_growBox->Hide();
+        }
+    }
 }
 
 bool wxWindowMac::AcceptsFocus() const
@@ -3215,6 +3248,7 @@
     wxASSERT_MSG( container != NULL , wxT("No valid mac container control") ) ;
 
     ::EmbedControl( m_peer->GetControlRef() , container ) ;
+    GetParent()->MacChildAdded() ;
 
     return true;
 }
Index: include/wx/mac/carbon/window.h
===================================================================
--- include/wx/mac/carbon/window.h	(revision 1347)
+++ include/wx/mac/carbon/window.h	(working copy)
@@ -17,6 +17,7 @@
 
 class WXDLLIMPEXP_FWD_CORE wxButton;
 class WXDLLIMPEXP_FWD_CORE wxScrollBar;
+class WXDLLIMPEXP_FWD_CORE wxPanel;
 class WXDLLIMPEXP_FWD_CORE wxNonOwnedWindow;
 
 class WXDLLIMPEXP_FWD_CORE wxMacControl ;
@@ -291,6 +292,7 @@
     wxScrollBar*        m_vScrollBar ;
     bool                m_hScrollBarAlwaysShown;
     bool                m_vScrollBarAlwaysShown;
+    wxPanel*            m_growBox ;
     wxString            m_label ;
 
     // set to true if we do a sharp clip at the content area of this window

 	  	 
