Index: wxWidgets-2.8/src/mac/carbon/window.cpp
===================================================================
--- wxWidgets-2.8/src/mac/carbon/window.cpp	(revision 52420)
+++ wxWidgets-2.8/src/mac/carbon/window.cpp	(working copy)
@@ -1031,6 +1031,7 @@
 
     m_hScrollBar = NULL ;
     m_vScrollBar = NULL ;
+    m_growBox = NULL ;
     m_macBackgroundBrush = wxNullBrush ;
 
     m_macIsUserPane = true;
@@ -1194,6 +1195,8 @@
 
 void wxWindowMac::MacChildAdded()
 {
+    if ( m_growBox )
+        m_growBox->Raise() ;
     if ( m_vScrollBar )
         m_vScrollBar->Raise() ;
     if ( m_hScrollBar )
@@ -2585,6 +2588,7 @@
     if ( IsTopLevel() )
         return ;
 
+#if 0
 #if wxMAC_USE_CORE_GRAPHICS
     if ( m_peer != NULL && MacHasScrollBarCorner() )
     {
@@ -2613,6 +2617,14 @@
         CGContextRestoreGState( cgContext );
     }
 #endif
+#else
+    if (m_growBox) {
+         if ( m_macBackgroundBrush.Ok() && m_macBackgroundBrush.GetStyle() != wxTRANSPARENT )
+             m_growBox->SetBackgroundColour(m_macBackgroundBrush.GetColour());
+         else
+             m_growBox->SetBackgroundColour(*wxWHITE);
+    }
+#endif
 }
 
 
@@ -2712,6 +2724,8 @@
         m_hScrollBar = NULL ;
     if ( child == m_vScrollBar )
         m_vScrollBar = NULL ;
+    if ( child == m_growBox )
+        m_growBox = NULL ;
 
     wxWindowBase::RemoveChild( child ) ;
 }
@@ -2820,6 +2834,8 @@
             continue;
         if (child == m_hScrollBar)
             continue;
+        if (child == m_growBox)
+            continue;
         if (child->IsTopLevel())
             continue;
 
@@ -3231,6 +3247,8 @@
                 continue;
             if (child == m_hScrollBar)
                 continue;
+            if (child == m_growBox)
+                continue;
             if (child->IsTopLevel())
                 continue;
             if (!child->IsShown())
@@ -3390,6 +3408,10 @@
             m_hScrollBar = new wxScrollBar((wxWindow*)this, wxID_ANY, hPoint, hSize , wxHORIZONTAL);
             m_hScrollBar->SetMinSize( wxDefaultSize );
         }
+        
+        wxPoint gPoint(width - scrlsize, height - scrlsize);
+        wxSize gSize(scrlsize, scrlsize);
+        m_growBox = new wxPanel((wxWindow *)this, wxID_ANY, gPoint, gSize, 0);
     }
 
     // because the create does not take into account the client area origin
@@ -3399,7 +3421,7 @@
 
 bool wxWindowMac::MacIsChildOfClientArea( const wxWindow* child ) const
 {
-    bool result = ((child == NULL) || ((child != m_hScrollBar) && (child != m_vScrollBar)));
+    bool result = ((child == NULL) || ((child != m_hScrollBar) && (child != m_vScrollBar) && (child != m_growBox)));
 
     return result ;
 }
@@ -3472,6 +3494,17 @@
         m_vScrollBar->SetSize( vPoint.x , vPoint.y, vSize.x, vSize.y , wxSIZE_ALLOW_MINUS_ONE );
     if ( m_hScrollBar )
         m_hScrollBar->SetSize( hPoint.x , hPoint.y, hSize.x, hSize.y, wxSIZE_ALLOW_MINUS_ONE );
+    if ( m_growBox ) {
+        if ( MacHasScrollBarCorner() ) {
+            m_growBox->SetSize( width - scrlsize, height - scrlsize, wxDefaultCoord, wxDefaultCoord, wxSIZE_USE_EXISTING );
+            if ( !m_growBox->IsShown() )
+                m_growBox->Show();
+        }
+        else {
+            if ( m_growBox->IsShown() )
+                m_growBox->Hide();
+        }
+    }
 }
 
 bool wxWindowMac::AcceptsFocus() const
@@ -3644,6 +3677,8 @@
         wxASSERT_MSG( container != NULL , wxT("No valid mac container control") ) ;
 
         ::EmbedControl( m_peer->GetControlRef() , container ) ;
+
+        GetParent()->MacChildAdded() ;
     }
     return true;
 }
Index: wxWidgets-2.8/include/wx/mac/carbon/window.h
===================================================================
--- wxWidgets-2.8/include/wx/mac/carbon/window.h	(revision 52420)
+++ wxWidgets-2.8/include/wx/mac/carbon/window.h	(working copy)
@@ -17,6 +17,7 @@
 
 class WXDLLEXPORT wxButton;
 class WXDLLEXPORT wxScrollBar;
+class WXDLLEXPORT wxPanel;
 class WXDLLEXPORT wxTopLevelWindowMac;
 
 class wxMacControl ;
@@ -303,6 +304,7 @@
     wxScrollBar*        m_hScrollBar ;
     wxScrollBar*        m_vScrollBar ;
     wxString            m_label ;
+    wxPanel*            m_growBox ;
 
     // set to true if we do a sharp clip at the content area of this window
     // must be dynamic as eg a panel normally is not clipping precisely, but if

 	  	 
