Index: e:/dev/wxWindows/contrib/include/wx/fl/antiflickpl.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/contrib/include/wx/fl/antiflickpl.h,v
retrieving revision 1.4
diff -u -2 -r1.4 antiflickpl.h
--- e:/dev/wxWindows/contrib/include/wx/fl/antiflickpl.h	2002/09/07 12:10:19	1.4
+++ e:/dev/wxWindows/contrib/include/wx/fl/antiflickpl.h	2002/10/14 19:53:15
@@ -27,5 +27,5 @@
 */
 
-class cbAntiflickerPlugin : public cbPluginBase
+class WXFL_DECLSPEC cbAntiflickerPlugin : public cbPluginBase
 {
     DECLARE_DYNAMIC_CLASS( cbAntiflickerPlugin )
Index: e:/dev/wxWindows/contrib/include/wx/fl/bardragpl.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/contrib/include/wx/fl/bardragpl.h,v
retrieving revision 1.4
diff -u -2 -r1.4 bardragpl.h
--- e:/dev/wxWindows/contrib/include/wx/fl/bardragpl.h	2002/09/07 12:10:19	1.4
+++ e:/dev/wxWindows/contrib/include/wx/fl/bardragpl.h	2002/10/14 19:53:15
@@ -24,5 +24,5 @@
 */
 
-class cbBarDragPlugin : public cbPluginBase
+class WXFL_DECLSPEC cbBarDragPlugin : public cbPluginBase
 {
     DECLARE_DYNAMIC_CLASS( cbBarDragPlugin )
Index: e:/dev/wxWindows/contrib/include/wx/fl/barhintspl.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/contrib/include/wx/fl/barhintspl.h,v
retrieving revision 1.4
diff -u -2 -r1.4 barhintspl.h
--- e:/dev/wxWindows/contrib/include/wx/fl/barhintspl.h	2002/09/07 12:10:19	1.4
+++ e:/dev/wxWindows/contrib/include/wx/fl/barhintspl.h	2002/10/14 19:53:15
@@ -25,5 +25,5 @@
 */
 
-class cbBarHintsPlugin : public cbPluginBase
+class WXFL_DECLSPEC cbBarHintsPlugin : public cbPluginBase
 {
     DECLARE_DYNAMIC_CLASS( cbBarHintsPlugin )
Index: e:/dev/wxWindows/contrib/include/wx/fl/cbcustom.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/contrib/include/wx/fl/cbcustom.h,v
retrieving revision 1.4
diff -u -2 -r1.4 cbcustom.h
--- e:/dev/wxWindows/contrib/include/wx/fl/cbcustom.h	2002/09/07 12:10:19	1.4
+++ e:/dev/wxWindows/contrib/include/wx/fl/cbcustom.h	2002/10/14 19:53:15
@@ -25,5 +25,5 @@
 */
 
-class cbSimpleCustomizationPlugin : public cbPluginBase
+class WXFL_DECLSPEC cbSimpleCustomizationPlugin : public cbPluginBase
 {
 public:
Index: e:/dev/wxWindows/contrib/include/wx/fl/controlbar.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/contrib/include/wx/fl/controlbar.h,v
retrieving revision 1.10
diff -u -2 -r1.10 controlbar.h
--- e:/dev/wxWindows/contrib/include/wx/fl/controlbar.h	2002/09/07 12:10:19	1.10
+++ e:/dev/wxWindows/contrib/include/wx/fl/controlbar.h	2002/10/14 19:53:17
@@ -22,4 +22,5 @@
 #include "wx/window.h"
 #include "wx/dynarray.h"
+#include "wx/fl/fldefs.h"
 
 #define WXCONTROLBAR_VERSION      1.3
@@ -27,23 +28,23 @@
 // forward declarations
 
-class  wxFrameLayout;
+class  WXFL_DECLSPEC wxFrameLayout;
 
-class  cbDockPane;
-class  cbUpdatesManagerBase;
-class  cbBarDimHandlerBase;
-class  cbPluginBase;
-class  cbPluginEvent;
-class  cbPaneDrawPlugin;
-
-class cbBarInfo;
-class cbRowInfo;
-class cbDimInfo;
-class cbCommonPaneProperties;
+class  WXFL_DECLSPEC cbDockPane;
+class  WXFL_DECLSPEC cbUpdatesManagerBase;
+class  WXFL_DECLSPEC cbBarDimHandlerBase;
+class  WXFL_DECLSPEC cbPluginBase;
+class  WXFL_DECLSPEC cbPluginEvent;
+class  WXFL_DECLSPEC cbPaneDrawPlugin;
+
+class WXFL_DECLSPEC cbBarInfo;
+class WXFL_DECLSPEC cbRowInfo;
+class WXFL_DECLSPEC cbDimInfo;
+class WXFL_DECLSPEC cbCommonPaneProperties;
 
 typedef cbBarInfo* BarInfoPtrT;
 typedef cbRowInfo* RowInfoPtrT;
 
-WX_DEFINE_ARRAY( BarInfoPtrT, BarArrayT );
-WX_DEFINE_ARRAY( RowInfoPtrT, RowArrayT );
+WXFL_DEFINE_ARRAY( BarInfoPtrT, BarArrayT );
+WXFL_DEFINE_ARRAY( RowInfoPtrT, RowArrayT );
 
 // control bar states
@@ -98,5 +99,5 @@
 */
 
-class cbBarSpy : public wxEvtHandler
+class WXFL_DECLSPEC cbBarSpy : public wxEvtHandler
 {
 public:
@@ -130,5 +131,5 @@
 */
 
-class wxFrameLayout : public wxEvtHandler
+class WXFL_DECLSPEC wxFrameLayout : public wxEvtHandler
 {
 public:
@@ -648,5 +649,5 @@
 */
 
-class cbUpdateMgrData : public wxObject
+class WXFL_DECLSPEC cbUpdateMgrData : public wxObject
 {
     DECLARE_DYNAMIC_CLASS( cbUpdateMgrData )
@@ -688,5 +689,5 @@
 */
 
-class cbBarDimHandlerBase : public wxObject
+class WXFL_DECLSPEC cbBarDimHandlerBase : public wxObject
 {
     DECLARE_ABSTRACT_CLASS( cbBarDimHandlerBase )
@@ -726,5 +727,5 @@
 */
 
-class cbDimInfo : public wxObject
+class WXFL_DECLSPEC cbDimInfo : public wxObject
 {
     DECLARE_DYNAMIC_CLASS( cbDimInfo )
@@ -825,5 +826,5 @@
 
 // FIXME: this array definition compiles but probably doesn't do what was intended (GD)
-WX_DEFINE_ARRAY_LONG(float, cbArrayFloat);
+WXFL_DEFINE_ARRAY_LONG(float, cbArrayFloat);
 
 /*
Index: e:/dev/wxWindows/contrib/include/wx/fl/dyntbar.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/contrib/include/wx/fl/dyntbar.h,v
retrieving revision 1.6
diff -u -2 -r1.6 dyntbar.h
--- e:/dev/wxWindows/contrib/include/wx/fl/dyntbar.h	2002/09/07 12:10:19	1.6
+++ e:/dev/wxWindows/contrib/include/wx/fl/dyntbar.h	2002/10/14 19:53:17
@@ -19,4 +19,5 @@
 #include "wx/tbarbase.h"
 #include "wx/dynarray.h"
+#include "wx/fl/fldefs.h"
 
 /*
@@ -24,5 +25,5 @@
 */
 
-class wxToolLayoutItem : public wxObject
+class WXFL_DECLSPEC wxToolLayoutItem : public wxObject
 {
     DECLARE_DYNAMIC_CLASS(wxToolLayoutItem)
@@ -33,11 +34,11 @@
 };
 
-class wxDynToolInfo;
+class WXFL_DECLSPEC wxDynToolInfo;
 
 typedef wxToolLayoutItem* wxToolLayoutItemPtrT;
 typedef wxDynToolInfo*    wxDynToolInfoPtrT;
 
-WX_DEFINE_ARRAY( wxToolLayoutItemPtrT, wxLayoutItemArrayT  );
-WX_DEFINE_ARRAY( wxDynToolInfoPtrT,    wxDynToolInfoArrayT );
+WXFL_DEFINE_ARRAY( wxToolLayoutItemPtrT, wxLayoutItemArrayT  );
+WXFL_DEFINE_ARRAY( wxDynToolInfoPtrT,    wxDynToolInfoArrayT );
 
 /*
@@ -45,5 +46,5 @@
 */
 
-class LayoutManagerBase
+class WXFL_DECLSPEC LayoutManagerBase
 {
 public:
@@ -64,5 +65,5 @@
 */
 
-class BagLayout : public LayoutManagerBase
+class WXFL_DECLSPEC BagLayout : public LayoutManagerBase
 {
 public:
@@ -79,5 +80,5 @@
 */
 
-class wxDynToolInfo : public wxToolLayoutItem
+class WXFL_DECLSPEC wxDynToolInfo : public wxToolLayoutItem
 {
     DECLARE_DYNAMIC_CLASS(wxDynToolInfo)
@@ -99,5 +100,5 @@
 */
 
-class wxDynamicToolBar : public wxToolBarBase
+class WXFL_DECLSPEC wxDynamicToolBar : public wxToolBarBase
 {
 protected:
Index: e:/dev/wxWindows/contrib/include/wx/fl/dyntbarhnd.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/contrib/include/wx/fl/dyntbarhnd.h,v
retrieving revision 1.3
diff -u -2 -r1.3 dyntbarhnd.h
--- e:/dev/wxWindows/contrib/include/wx/fl/dyntbarhnd.h	2002/09/07 12:10:19	1.3
+++ e:/dev/wxWindows/contrib/include/wx/fl/dyntbarhnd.h	2002/10/14 19:53:17
@@ -24,5 +24,5 @@
 */
 
-class cbDynToolBarDimHandler : public cbBarDimHandlerBase
+class WXFL_DECLSPEC cbDynToolBarDimHandler : public cbBarDimHandlerBase
 {
     DECLARE_DYNAMIC_CLASS( cbDynToolBarDimHandler )
Index: e:/dev/wxWindows/contrib/include/wx/fl/frmview.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/contrib/include/wx/fl/frmview.h,v
retrieving revision 1.3
diff -u -2 -r1.3 frmview.h
--- e:/dev/wxWindows/contrib/include/wx/fl/frmview.h	2002/09/07 12:10:19	1.3
+++ e:/dev/wxWindows/contrib/include/wx/fl/frmview.h	2002/10/14 19:53:17
@@ -33,5 +33,5 @@
 */
 
-class wxFrameView : public wxEvtHandler
+class WXFL_DECLSPEC wxFrameView : public wxEvtHandler
 {
 protected:
@@ -88,5 +88,5 @@
 */
 
-class wxFrameManager : wxObject
+class WXFL_DECLSPEC wxFrameManager : wxObject
 {
 protected:
Index: e:/dev/wxWindows/contrib/include/wx/fl/garbagec.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/contrib/include/wx/fl/garbagec.h,v
retrieving revision 1.3
diff -u -2 -r1.3 garbagec.h
--- e:/dev/wxWindows/contrib/include/wx/fl/garbagec.h	2002/09/07 12:10:19	1.3
+++ e:/dev/wxWindows/contrib/include/wx/fl/garbagec.h	2002/10/14 19:53:17
@@ -18,4 +18,5 @@
 
 #include "wx/list.h"
+#include "wx/fl/fldefs.h"
 
 struct GCItem
@@ -34,5 +35,5 @@
 */
 
-class GarbageCollector
+class WXFL_DECLSPEC GarbageCollector
 {
 protected:
Index: e:/dev/wxWindows/contrib/include/wx/fl/gcupdatesmgr.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/contrib/include/wx/fl/gcupdatesmgr.h,v
retrieving revision 1.3
diff -u -2 -r1.3 gcupdatesmgr.h
--- e:/dev/wxWindows/contrib/include/wx/fl/gcupdatesmgr.h	2002/09/07 12:10:19	1.3
+++ e:/dev/wxWindows/contrib/include/wx/fl/gcupdatesmgr.h	2002/10/14 19:53:17
@@ -94,5 +94,5 @@
 */
 
-class cbGCUpdatesMgr : public cbSimpleUpdatesMgr
+class WXFL_DECLSPEC cbGCUpdatesMgr : public cbSimpleUpdatesMgr
 {
     DECLARE_DYNAMIC_CLASS( cbGCUpdatesMgr )
Index: e:/dev/wxWindows/contrib/include/wx/fl/hintanimpl.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/contrib/include/wx/fl/hintanimpl.h,v
retrieving revision 1.3
diff -u -2 -r1.3 hintanimpl.h
--- e:/dev/wxWindows/contrib/include/wx/fl/hintanimpl.h	2002/09/07 12:10:19	1.3
+++ e:/dev/wxWindows/contrib/include/wx/fl/hintanimpl.h	2002/10/14 19:53:17
@@ -21,5 +21,5 @@
 #include "wx/timer.h"
 
-class cbHintAnimTimer;
+class WXFL_DECLSPEC cbHintAnimTimer;
 
 /*
@@ -27,5 +27,5 @@
 */
 
-class cbHintAnimationPlugin : public cbPluginBase
+class WXFL_DECLSPEC cbHintAnimationPlugin : public cbPluginBase
 {
     DECLARE_DYNAMIC_CLASS( cbHintAnimationPlugin )
@@ -102,5 +102,5 @@
 */
 
-struct MorphInfoT
+struct WXFL_DECLSPEC MorphInfoT
 {
     wxPoint mFrom;
@@ -112,5 +112,5 @@
 */
 
-class cbHintAnimTimer : public wxTimer 
+class WXFL_DECLSPEC cbHintAnimTimer : public wxTimer 
 {
 protected:
Index: e:/dev/wxWindows/contrib/include/wx/fl/newbmpbtn.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/contrib/include/wx/fl/newbmpbtn.h,v
retrieving revision 1.7
diff -u -2 -r1.7 newbmpbtn.h
--- e:/dev/wxWindows/contrib/include/wx/fl/newbmpbtn.h	2002/09/07 12:10:19	1.7
+++ e:/dev/wxWindows/contrib/include/wx/fl/newbmpbtn.h	2002/10/14 19:53:18
@@ -19,4 +19,5 @@
 #include "wx/button.h"
 #include "wx/string.h"
+#include "wx/fl/fldefs.h"
 
 // defaults
@@ -32,6 +33,6 @@
 // classes declared in this header file
 
-class wxNewBitmapButton;
-class wxBorderLessBitmapButton;
+class WXFL_DECLSPEC wxNewBitmapButton;
+class WXFL_DECLSPEC wxBorderLessBitmapButton;
 
 /*
@@ -194,9 +195,4 @@
         // Responds to a kill focus event.
     void OnKillFocus( wxFocusEvent& event );
-
-        // Maps bitmap to current system colours on Windows
-#ifdef __WXMSW__
-    WXHBITMAP MapBitmap(WXHBITMAP bitmap, int width, int height);
-#endif
 
     DECLARE_EVENT_TABLE()
Index: e:/dev/wxWindows/contrib/include/wx/fl/panedrawpl.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/contrib/include/wx/fl/panedrawpl.h,v
retrieving revision 1.3
diff -u -2 -r1.3 panedrawpl.h
--- e:/dev/wxWindows/contrib/include/wx/fl/panedrawpl.h	2002/09/07 12:10:19	1.3
+++ e:/dev/wxWindows/contrib/include/wx/fl/panedrawpl.h	2002/10/14 19:53:18
@@ -28,5 +28,5 @@
 */
 
-class cbPaneDrawPlugin : public cbPluginBase
+class WXFL_DECLSPEC cbPaneDrawPlugin : public cbPluginBase
 {
 public:
Index: e:/dev/wxWindows/contrib/include/wx/fl/rowdragpl.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/contrib/include/wx/fl/rowdragpl.h,v
retrieving revision 1.3
diff -u -2 -r1.3 rowdragpl.h
--- e:/dev/wxWindows/contrib/include/wx/fl/rowdragpl.h	2002/09/07 12:10:19	1.3
+++ e:/dev/wxWindows/contrib/include/wx/fl/rowdragpl.h	2002/10/14 19:53:18
@@ -26,5 +26,5 @@
 */
 
-class cbRowDragPlugin : public cbPluginBase
+class WXFL_DECLSPEC cbRowDragPlugin : public cbPluginBase
 {
     DECLARE_DYNAMIC_CLASS( cbRowDragPlugin )
@@ -218,5 +218,5 @@
 */
 
-class cbHiddenBarInfo : public wxObject
+class WXFL_DECLSPEC cbHiddenBarInfo : public wxObject
 {
     DECLARE_DYNAMIC_CLASS( cbHiddenBarInfo )
Index: e:/dev/wxWindows/contrib/include/wx/fl/rowlayoutpl.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/contrib/include/wx/fl/rowlayoutpl.h,v
retrieving revision 1.3
diff -u -2 -r1.3 rowlayoutpl.h
--- e:/dev/wxWindows/contrib/include/wx/fl/rowlayoutpl.h	2002/09/07 12:10:19	1.3
+++ e:/dev/wxWindows/contrib/include/wx/fl/rowlayoutpl.h	2002/10/14 19:53:18
@@ -24,5 +24,5 @@
 */
 
-class cbRowLayoutPlugin : public cbPluginBase
+class WXFL_DECLSPEC cbRowLayoutPlugin : public cbPluginBase
 {
     DECLARE_DYNAMIC_CLASS( cbRowLayoutPlugin )
Index: e:/dev/wxWindows/contrib/include/wx/fl/toolwnd.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/contrib/include/wx/fl/toolwnd.h,v
retrieving revision 1.3
diff -u -2 -r1.3 toolwnd.h
--- e:/dev/wxWindows/contrib/include/wx/fl/toolwnd.h	2002/09/07 12:10:19	1.3
+++ e:/dev/wxWindows/contrib/include/wx/fl/toolwnd.h	2002/10/14 19:53:18
@@ -20,4 +20,5 @@
 #include "wx/frame.h"
 #include "wx/dynarray.h"
+#include "wx/fl/fldefs.h"
 
 // fixed settings
@@ -27,9 +28,9 @@
 #define BTN_X_WIEGHT         2
 
-class cbMiniButton;
+class WXFL_DECLSPEC cbMiniButton;
 
 typedef cbMiniButton* cbMinitButtonPtrT;
 
-WX_DEFINE_ARRAY( cbMinitButtonPtrT, cbMiniButtonArrayT );
+WXFL_DEFINE_ARRAY( cbMinitButtonPtrT, cbMiniButtonArrayT );
 
 /*
@@ -38,5 +39,5 @@
 */
 
-class wxToolWindow : public wxFrame
+class WXFL_DECLSPEC wxToolWindow : public wxFrame
 {
     DECLARE_DYNAMIC_CLASS( wxToolWindow )
@@ -222,5 +223,5 @@
 */
 
-class cbCloseBox : public cbMiniButton
+class WXFL_DECLSPEC cbCloseBox : public cbMiniButton
 {
 public:
@@ -233,5 +234,5 @@
 */
 
-class cbCollapseBox  : public cbMiniButton
+class WXFL_DECLSPEC cbCollapseBox  : public cbMiniButton
 {
 public:
@@ -246,5 +247,5 @@
 */
 
-class cbDockBox : public cbMiniButton
+class WXFL_DECLSPEC cbDockBox : public cbMiniButton
 {
 public:
@@ -258,5 +259,5 @@
 */
 
-class cbFloatedBarWindow : public wxToolWindow
+class WXFL_DECLSPEC cbFloatedBarWindow : public wxToolWindow
 {
     DECLARE_DYNAMIC_CLASS( cbFloatedBarWindow )
Index: e:/dev/wxWindows/contrib/include/wx/fl/updatesmgr.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/contrib/include/wx/fl/updatesmgr.h,v
retrieving revision 1.3
diff -u -2 -r1.3 updatesmgr.h
--- e:/dev/wxWindows/contrib/include/wx/fl/updatesmgr.h	2002/09/07 12:10:19	1.3
+++ e:/dev/wxWindows/contrib/include/wx/fl/updatesmgr.h	2002/10/14 19:53:18
@@ -24,5 +24,5 @@
 */
 
-class cbSimpleUpdatesMgr : public cbUpdatesManagerBase
+class WXFL_DECLSPEC cbSimpleUpdatesMgr : public cbUpdatesManagerBase
 {
     DECLARE_DYNAMIC_CLASS( cbSimpleUpdatesMgr )
Index: e:/dev/wxWindows/contrib/samples/fl/fl_demo1/fl_demo.dsp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/contrib/samples/fl/fl_demo1/fl_demo.dsp,v
retrieving revision 1.4
diff -u -2 -r1.4 fl_demo.dsp
--- e:/dev/wxWindows/contrib/samples/fl/fl_demo1/fl_demo.dsp	2001/11/19 16:05:21	1.4
+++ e:/dev/wxWindows/contrib/samples/fl/fl_demo1/fl_demo.dsp	2002/10/14 19:53:36
@@ -5,5 +5,5 @@
 # TARGTYPE "Win32 (x86) Application" 0x0101
 
-CFG=fl_demo1 - Win32 Debug
+CFG=fl_demo1 - Win32 Debug DLL
 !MESSAGE This is not a valid makefile. To build this project using NMAKE,
 !MESSAGE use the Export Makefile command and run
@@ -14,5 +14,5 @@
 !MESSAGE by defining the macro CFG on the command line. For example:
 !MESSAGE 
-!MESSAGE NMAKE /f "fl_demo.mak" CFG="fl_demo1 - Win32 Debug"
+!MESSAGE NMAKE /f "fl_demo.mak" CFG="fl_demo1 - Win32 Debug DLL"
 !MESSAGE 
 !MESSAGE Possible choices for configuration are:
@@ -20,4 +20,6 @@
 !MESSAGE "fl_demo1 - Win32 Release" (based on "Win32 (x86) Application")
 !MESSAGE "fl_demo1 - Win32 Debug" (based on "Win32 (x86) Application")
+!MESSAGE "fl_demo1 - Win32 Debug DLL" (based on "Win32 (x86) Application")
+!MESSAGE "fl_demo1 - Win32 Release DLL" (based on "Win32 (x86) Application")
 !MESSAGE 
 
@@ -82,4 +84,58 @@
 # ADD LINK32 kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib comctl32.lib rpcrt4.lib wsock32.lib winmm.lib wxmswd.lib pngd.lib zlibd.lib jpegd.lib tiffd.lib regexd.lib fld.lib /nologo /subsystem:windows /debug /machine:I386 /nodefaultlib:"libcd.lib" /nodefaultlib:"libcid.lib" /nodefaultlib:"msvcrt.lib" /pdbtype:sept /libpath:"../../../../lib"
 
+!ELSEIF  "$(CFG)" == "fl_demo1 - Win32 Debug DLL"
+
+# PROP BASE Use_MFC 0
+# PROP BASE Use_Debug_Libraries 1
+# PROP BASE Output_Dir "Debug_DLL"
+# PROP BASE Intermediate_Dir "Debug_DLL"
+# PROP BASE Ignore_Export_Lib 0
+# PROP BASE Target_Dir ""
+# PROP Use_MFC 0
+# PROP Use_Debug_Libraries 1
+# PROP Output_Dir "DebugDll"
+# PROP Intermediate_Dir "DebugDll"
+# PROP Ignore_Export_Lib 0
+# PROP Target_Dir ""
+# ADD BASE CPP /nologo /MDd /W3 /Gm /GX /ZI /Od /I "../../../../include" /I "../../../../contrib/include" /I "../../../../lib/mswd" /D "_DEBUG" /D DEBUG=1 /D "__WXDEBUG__" /D "WIN32" /D "_WINDOWS" /D "__WINDOWS__" /D "__WXMSW__" /D "__WIN95__" /D "__WIN32__" /D WINVER=0x0400 /D "STRICT" /Yu"wx/wxprec.h" /FD /D BMP_DIR=\"../bitmaps/\" /c
+# ADD CPP /nologo /MDd /W3 /Gm /GX /ZI /Od /I "../../../../include" /I "../../../../contrib/include" /I "../../../../lib/mswdlld" /D "_DEBUG" /D DEBUG=1 /D "__WXDEBUG__" /D "WIN32" /D "_WINDOWS" /D "__WINDOWS__" /D "__WXMSW__" /D "__WIN95__" /D "__WIN32__" /D WINVER=0x0400 /D "STRICT" /D "WXUSINGDLL" /D "WXUSING_FL_DLL" /Yu"wx/wxprec.h" /FD /D BMP_DIR=\"../bitmaps/\" /c
+# ADD BASE MTL /nologo /D "_DEBUG" /mktyplib203 /o "NUL" /win32
+# ADD MTL /nologo /D "_DEBUG" /mktyplib203 /o "NUL" /win32
+# ADD BASE RSC /l 0x809 /d "_DEBUG"
+# ADD RSC /l 0x809 /d "_DEBUG"
+BSC32=bscmake.exe
+# ADD BASE BSC32 /nologo
+# ADD BSC32 /nologo
+LINK32=link.exe
+# ADD BASE LINK32 kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib comctl32.lib rpcrt4.lib wsock32.lib winmm.lib wxmswd.lib pngd.lib zlibd.lib jpegd.lib tiffd.lib regexd.lib fld.lib /nologo /subsystem:windows /debug /machine:I386 /nodefaultlib:"libcd.lib" /nodefaultlib:"libcid.lib" /nodefaultlib:"msvcrt.lib" /pdbtype:sept /libpath:"../../../../lib"
+# ADD LINK32 kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib comctl32.lib rpcrt4.lib wsock32.lib winmm.lib wxmsw233d.lib pngd.lib zlibd.lib jpegd.lib tiffd.lib regexd.lib fldlld.lib /nologo /subsystem:windows /debug /machine:I386 /nodefaultlib:"libcd.lib" /nodefaultlib:"libcid.lib" /nodefaultlib:"msvcrt.lib" /pdbtype:sept /libpath:"../../../../lib"
+
+!ELSEIF  "$(CFG)" == "fl_demo1 - Win32 Release DLL"
+
+# PROP BASE Use_MFC 0
+# PROP BASE Use_Debug_Libraries 0
+# PROP BASE Output_Dir "Release_DLL"
+# PROP BASE Intermediate_Dir "Release_DLL"
+# PROP BASE Ignore_Export_Lib 0
+# PROP BASE Target_Dir ""
+# PROP Use_MFC 0
+# PROP Use_Debug_Libraries 0
+# PROP Output_Dir "ReleaseDll"
+# PROP Intermediate_Dir "ReleaseDll"
+# PROP Ignore_Export_Lib 0
+# PROP Target_Dir ""
+# ADD BASE CPP /nologo /MD /W3 /GX /O1 /Ob2 /I "../../../../include" /I "../../../../contrib/include" /I "../../../../lib/msw" /D "NDEBUG" /D "WIN32" /D "_WINDOWS" /D "__WINDOWS__" /D "__WXMSW__" /D "__WIN95__" /D "__WIN32__" /D WINVER=0x0400 /D "STRICT" /FD /D BMP_DIR=\"../bitmaps/\" /c
+# ADD CPP /nologo /MD /W3 /GX /O2 /Ob2 /I "../../../../include" /I "../../../../contrib/include" /I "../../../../lib/mswdll" /D "NDEBUG" /D "WIN32" /D "_WINDOWS" /D "__WINDOWS__" /D "__WXMSW__" /D "__WIN95__" /D "__WIN32__" /D WINVER=0x0400 /D "STRICT" /D "WXUSINGDLL" /D "WXUSING_FL_DLL" /YX /FD /D BMP_DIR=\"../bitmaps/\" /c
+# ADD BASE MTL /nologo /D "NDEBUG" /mktyplib203 /o "NUL" /win32
+# ADD MTL /nologo /D "NDEBUG" /mktyplib203 /o "NUL" /win32
+# ADD BASE RSC /l 0x809 /d "NDEBUG"
+# ADD RSC /l 0x809 /d "NDEBUG"
+BSC32=bscmake.exe
+# ADD BASE BSC32 /nologo
+# ADD BSC32 /nologo
+LINK32=link.exe
+# ADD BASE LINK32 kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib comctl32.lib rpcrt4.lib wsock32.lib winmm.lib wxmsw.lib png.lib zlib.lib jpeg.lib tiff.lib regex.lib fl.lib /nologo /subsystem:windows /machine:I386 /nodefaultlib:"libc.lib" /nodefaultlib:"libci.lib" /nodefaultlib:"msvcrtd.lib" /libpath:"../../../../lib"
+# ADD LINK32 kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib comctl32.lib rpcrt4.lib wsock32.lib winmm.lib wxmsw233.lib png.lib zlib.lib jpeg.lib tiff.lib regex.lib fldll.lib /nologo /subsystem:windows /machine:I386 /nodefaultlib:"libc.lib" /nodefaultlib:"libci.lib" /nodefaultlib:"msvcrtd.lib" /libpath:"../../../../lib"
+
 !ENDIF 
 
@@ -88,4 +144,6 @@
 # Name "fl_demo1 - Win32 Release"
 # Name "fl_demo1 - Win32 Debug"
+# Name "fl_demo1 - Win32 Debug DLL"
+# Name "fl_demo1 - Win32 Release DLL"
 # Begin Source File
 
@@ -97,4 +155,11 @@
 
 # ADD CPP /Yc"wx/wxprec.h"
+
+!ELSEIF  "$(CFG)" == "fl_demo1 - Win32 Debug DLL"
+
+# ADD BASE CPP /Yc"wx/wxprec.h"
+# ADD CPP /Yc"wx/wxprec.h"
+
+!ELSEIF  "$(CFG)" == "fl_demo1 - Win32 Release DLL"
 
 !ENDIF 
Index: e:/dev/wxWindows/contrib/samples/fl/fl_demo1/makefile.vc
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/contrib/samples/fl/fl_demo1/makefile.vc,v
retrieving revision 1.1
diff -u -2 -r1.1 makefile.vc
--- e:/dev/wxWindows/contrib/samples/fl/fl_demo1/makefile.vc	2001/07/24 15:27:11	1.1
+++ e:/dev/wxWindows/contrib/samples/fl/fl_demo1/makefile.vc	2002/10/14 19:53:37
@@ -13,8 +13,22 @@
 
 EXTRAINC = -DBMP_DIR=\"../bitmaps/\"
+
+!if "$(WXUSING_FL_DLL)" == "1"
+# Define WXUSING_FL_DLL so the compiler knows it is set.
+EXTRAINC = $(EXTRAINC) -DWXUSING_FL_DLL
+# Set to use wxWin in DLL format...
+WXUSINGDLL = 1
+
 !if "$(FINAL)" == "1"
+EXTRALIBS = $(WXDIR)\lib\fldll.lib
+!else
+EXTRALIBS = $(WXDIR)\lib\fldlld.lib
+!endif
+!else
+!if "$(FINAL)" == "1"
 EXTRALIBS = $(WXDIR)\lib\fl.lib
 !else
 EXTRALIBS = $(WXDIR)\lib\fld.lib
+!endif
 !endif
 
Index: e:/dev/wxWindows/contrib/samples/fl/fl_demo2/fl_demo.dsp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/contrib/samples/fl/fl_demo2/fl_demo.dsp,v
retrieving revision 1.4
diff -u -2 -r1.4 fl_demo.dsp
--- e:/dev/wxWindows/contrib/samples/fl/fl_demo2/fl_demo.dsp	2001/11/19 16:05:22	1.4
+++ e:/dev/wxWindows/contrib/samples/fl/fl_demo2/fl_demo.dsp	2002/10/14 19:53:37
@@ -5,5 +5,5 @@
 # TARGTYPE "Win32 (x86) Application" 0x0101
 
-CFG=fl_demo2 - Win32 Debug
+CFG=fl_demo2 - Win32 Debug DLL
 !MESSAGE This is not a valid makefile. To build this project using NMAKE,
 !MESSAGE use the Export Makefile command and run
@@ -14,5 +14,5 @@
 !MESSAGE by defining the macro CFG on the command line. For example:
 !MESSAGE 
-!MESSAGE NMAKE /f "fl_demo.mak" CFG="fl_demo2 - Win32 Debug"
+!MESSAGE NMAKE /f "fl_demo.mak" CFG="fl_demo2 - Win32 Debug DLL"
 !MESSAGE 
 !MESSAGE Possible choices for configuration are:
@@ -20,4 +20,6 @@
 !MESSAGE "fl_demo2 - Win32 Release" (based on "Win32 (x86) Application")
 !MESSAGE "fl_demo2 - Win32 Debug" (based on "Win32 (x86) Application")
+!MESSAGE "fl_demo2 - Win32 Debug DLL" (based on "Win32 (x86) Application")
+!MESSAGE "fl_demo2 - Win32 Release DLL" (based on "Win32 (x86) Application")
 !MESSAGE 
 
@@ -82,4 +84,58 @@
 # ADD LINK32 kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib comctl32.lib rpcrt4.lib wsock32.lib winmm.lib wxmswd.lib pngd.lib zlibd.lib jpegd.lib tiffd.lib regexd.lib fld.lib /nologo /subsystem:windows /debug /machine:I386 /nodefaultlib:"libcd.lib" /nodefaultlib:"libcid.lib" /nodefaultlib:"msvcrt.lib" /pdbtype:sept /libpath:"../../../../lib"
 
+!ELSEIF  "$(CFG)" == "fl_demo2 - Win32 Debug DLL"
+
+# PROP BASE Use_MFC 0
+# PROP BASE Use_Debug_Libraries 1
+# PROP BASE Output_Dir "Debug_DLL"
+# PROP BASE Intermediate_Dir "Debug_DLL"
+# PROP BASE Ignore_Export_Lib 0
+# PROP BASE Target_Dir ""
+# PROP Use_MFC 0
+# PROP Use_Debug_Libraries 1
+# PROP Output_Dir "DebugDll"
+# PROP Intermediate_Dir "DebugDll"
+# PROP Ignore_Export_Lib 0
+# PROP Target_Dir ""
+# ADD BASE CPP /nologo /MDd /W3 /Gm /GX /ZI /Od /I "../../../../include" /I "../../../../contrib/include" /I "../../../../lib/mswd" /D "_DEBUG" /D DEBUG=1 /D "__WXDEBUG__" /D "WIN32" /D "_WINDOWS" /D "__WINDOWS__" /D "__WXMSW__" /D "__WIN95__" /D "__WIN32__" /D WINVER=0x0400 /D "STRICT" /Yu"wx/wxprec.h" /FD /D BMP_DIR=\"../bitmaps/\" /c
+# ADD CPP /nologo /MDd /W3 /Gm /GX /ZI /Od /I "../../../../include" /I "../../../../contrib/include" /I "../../../../lib/mswdlld" /D "_DEBUG" /D DEBUG=1 /D "__WXDEBUG__" /D "WIN32" /D "_WINDOWS" /D "__WINDOWS__" /D "__WXMSW__" /D "__WIN95__" /D "__WIN32__" /D WINVER=0x0400 /D "STRICT" /D "WXUSINGDLL" /D "WXUSING_FL_DLL" /Yu"wx/wxprec.h" /FD /D BMP_DIR=\"../bitmaps/\" /c
+# ADD BASE MTL /nologo /D "_DEBUG" /mktyplib203 /o "NUL" /win32
+# ADD MTL /nologo /D "_DEBUG" /mktyplib203 /o "NUL" /win32
+# ADD BASE RSC /l 0x809 /d "_DEBUG"
+# ADD RSC /l 0x809 /d "_DEBUG"
+BSC32=bscmake.exe
+# ADD BASE BSC32 /nologo
+# ADD BSC32 /nologo
+LINK32=link.exe
+# ADD BASE LINK32 kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib comctl32.lib rpcrt4.lib wsock32.lib winmm.lib wxmswd.lib pngd.lib zlibd.lib jpegd.lib tiffd.lib regexd.lib fld.lib /nologo /subsystem:windows /debug /machine:I386 /nodefaultlib:"libcd.lib" /nodefaultlib:"libcid.lib" /nodefaultlib:"msvcrt.lib" /pdbtype:sept /libpath:"../../../../lib"
+# ADD LINK32 kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib comctl32.lib rpcrt4.lib wsock32.lib winmm.lib wxmsw233d.lib pngd.lib zlibd.lib jpegd.lib tiffd.lib regexd.lib fldlld.lib /nologo /subsystem:windows /debug /machine:I386 /nodefaultlib:"libcd.lib" /nodefaultlib:"libcid.lib" /nodefaultlib:"msvcrt.lib" /pdbtype:sept /libpath:"../../../../lib"
+
+!ELSEIF  "$(CFG)" == "fl_demo2 - Win32 Release DLL"
+
+# PROP BASE Use_MFC 0
+# PROP BASE Use_Debug_Libraries 0
+# PROP BASE Output_Dir "Release_DLL"
+# PROP BASE Intermediate_Dir "Release_DLL"
+# PROP BASE Ignore_Export_Lib 0
+# PROP BASE Target_Dir ""
+# PROP Use_MFC 0
+# PROP Use_Debug_Libraries 0
+# PROP Output_Dir "ReleaseDll"
+# PROP Intermediate_Dir "ReleaseDll"
+# PROP Ignore_Export_Lib 0
+# PROP Target_Dir ""
+# ADD BASE CPP /nologo /MD /W3 /GX /O1 /Ob2 /I "../../../../include" /I "../../../../contrib/include" /I "../../../../lib/msw" /D "NDEBUG" /D "WIN32" /D "_WINDOWS" /D "__WINDOWS__" /D "__WXMSW__" /D "__WIN95__" /D "__WIN32__" /D WINVER=0x0400 /D "STRICT" /FD /D BMP_DIR=\"../bitmaps/\" /c
+# ADD CPP /nologo /MD /W3 /GX /O2 /Ob2 /I "../../../../include" /I "../../../../contrib/include" /I "../../../../lib/mswdll" /D "NDEBUG" /D "WIN32" /D "_WINDOWS" /D "__WINDOWS__" /D "__WXMSW__" /D "__WIN95__" /D "__WIN32__" /D WINVER=0x0400 /D "STRICT" /D "WXUSINGDLL" /D "WXUSING_FL_DLL" /YX /FD /D BMP_DIR=\"../bitmaps/\" /c
+# ADD BASE MTL /nologo /D "NDEBUG" /mktyplib203 /o "NUL" /win32
+# ADD MTL /nologo /D "NDEBUG" /mktyplib203 /o "NUL" /win32
+# ADD BASE RSC /l 0x809 /d "NDEBUG"
+# ADD RSC /l 0x809 /d "NDEBUG"
+BSC32=bscmake.exe
+# ADD BASE BSC32 /nologo
+# ADD BSC32 /nologo
+LINK32=link.exe
+# ADD BASE LINK32 kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib comctl32.lib rpcrt4.lib wsock32.lib winmm.lib wxmsw.lib png.lib zlib.lib jpeg.lib tiff.lib regex.lib fl.lib /nologo /subsystem:windows /machine:I386 /nodefaultlib:"libc.lib" /nodefaultlib:"libci.lib" /nodefaultlib:"msvcrtd.lib" /libpath:"../../../../lib"
+# ADD LINK32 kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib comctl32.lib rpcrt4.lib wsock32.lib winmm.lib wxmsw233.lib png.lib zlib.lib jpeg.lib tiff.lib regex.lib fldll.lib /nologo /subsystem:windows /machine:I386 /nodefaultlib:"libc.lib" /nodefaultlib:"libci.lib" /nodefaultlib:"msvcrtd.lib" /libpath:"../../../../lib"
+
 !ENDIF 
 
@@ -88,4 +144,6 @@
 # Name "fl_demo2 - Win32 Release"
 # Name "fl_demo2 - Win32 Debug"
+# Name "fl_demo2 - Win32 Debug DLL"
+# Name "fl_demo2 - Win32 Release DLL"
 # Begin Source File
 
@@ -97,4 +155,11 @@
 
 # ADD CPP /Yc"wx/wxprec.h"
+
+!ELSEIF  "$(CFG)" == "fl_demo2 - Win32 Debug DLL"
+
+# ADD BASE CPP /Yc"wx/wxprec.h"
+# ADD CPP /Yc"wx/wxprec.h"
+
+!ELSEIF  "$(CFG)" == "fl_demo2 - Win32 Release DLL"
 
 !ENDIF 
Index: e:/dev/wxWindows/contrib/samples/fl/fl_demo2/makefile.vc
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/contrib/samples/fl/fl_demo2/makefile.vc,v
retrieving revision 1.1
diff -u -2 -r1.1 makefile.vc
--- e:/dev/wxWindows/contrib/samples/fl/fl_demo2/makefile.vc	2001/07/24 15:27:11	1.1
+++ e:/dev/wxWindows/contrib/samples/fl/fl_demo2/makefile.vc	2002/10/14 19:53:37
@@ -13,8 +13,22 @@
 
 EXTRAINC = -DBMP_DIR=\"../bitmaps/\"
+
+!if "$(WXUSING_FL_DLL)" == "1"
+# Define WXUSING_FL_DLL so the compiler knows it is set.
+EXTRAINC = $(EXTRAINC) -DWXUSING_FL_DLL
+# Set to use wxWin in DLL format...
+WXUSINGDLL = 1
+
 !if "$(FINAL)" == "1"
+EXTRALIBS = $(WXDIR)\lib\fldll.lib
+!else
+EXTRALIBS = $(WXDIR)\lib\fldlld.lib
+!endif
+!else
+!if "$(FINAL)" == "1"
 EXTRALIBS = $(WXDIR)\lib\fl.lib
 !else
 EXTRALIBS = $(WXDIR)\lib\fld.lib
+!endif
 !endif
 
Index: e:/dev/wxWindows/contrib/samples/fl/fl_sample1/fl_demo.dsp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/contrib/samples/fl/fl_sample1/fl_demo.dsp,v
retrieving revision 1.4
diff -u -2 -r1.4 fl_demo.dsp
--- e:/dev/wxWindows/contrib/samples/fl/fl_sample1/fl_demo.dsp	2001/11/19 16:05:22	1.4
+++ e:/dev/wxWindows/contrib/samples/fl/fl_sample1/fl_demo.dsp	2002/10/14 19:53:37
@@ -5,5 +5,5 @@
 # TARGTYPE "Win32 (x86) Application" 0x0101
 
-CFG=fl_sample1 - Win32 Debug
+CFG=fl_sample1 - Win32 Debug DLL
 !MESSAGE This is not a valid makefile. To build this project using NMAKE,
 !MESSAGE use the Export Makefile command and run
@@ -14,5 +14,5 @@
 !MESSAGE by defining the macro CFG on the command line. For example:
 !MESSAGE 
-!MESSAGE NMAKE /f "fl_demo.mak" CFG="fl_sample1 - Win32 Debug"
+!MESSAGE NMAKE /f "fl_demo.mak" CFG="fl_sample1 - Win32 Debug DLL"
 !MESSAGE 
 !MESSAGE Possible choices for configuration are:
@@ -20,4 +20,6 @@
 !MESSAGE "fl_sample1 - Win32 Release" (based on "Win32 (x86) Application")
 !MESSAGE "fl_sample1 - Win32 Debug" (based on "Win32 (x86) Application")
+!MESSAGE "fl_sample1 - Win32 Debug DLL" (based on "Win32 (x86) Application")
+!MESSAGE "fl_sample1 - Win32 Release DLL" (based on "Win32 (x86) Application")
 !MESSAGE 
 
@@ -82,4 +84,58 @@
 # ADD LINK32 kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib comctl32.lib rpcrt4.lib wsock32.lib winmm.lib wxmswd.lib pngd.lib zlibd.lib jpegd.lib tiffd.lib regexd.lib fld.lib /nologo /subsystem:windows /debug /machine:I386 /nodefaultlib:"libcd.lib" /nodefaultlib:"libcid.lib" /nodefaultlib:"msvcrt.lib" /pdbtype:sept /libpath:"../../../../lib"
 
+!ELSEIF  "$(CFG)" == "fl_sample1 - Win32 Debug DLL"
+
+# PROP BASE Use_MFC 0
+# PROP BASE Use_Debug_Libraries 1
+# PROP BASE Output_Dir "Debug_DLL"
+# PROP BASE Intermediate_Dir "Debug_DLL"
+# PROP BASE Ignore_Export_Lib 0
+# PROP BASE Target_Dir ""
+# PROP Use_MFC 0
+# PROP Use_Debug_Libraries 1
+# PROP Output_Dir "DebugDll"
+# PROP Intermediate_Dir "DebugDll"
+# PROP Ignore_Export_Lib 0
+# PROP Target_Dir ""
+# ADD BASE CPP /nologo /MDd /W3 /Gm /GX /ZI /Od /I "../../../../include" /I "../../../../contrib/include" /I "../../../../lib/mswd" /D "_DEBUG" /D DEBUG=1 /D "__WXDEBUG__" /D "WIN32" /D "_WINDOWS" /D "__WINDOWS__" /D "__WXMSW__" /D "__WIN95__" /D "__WIN32__" /D WINVER=0x0400 /D "STRICT" /Yu"wx/wxprec.h" /FD /D BMP_DIR=\"../bitmaps/\" /c
+# ADD CPP /nologo /MDd /W3 /Gm /GX /ZI /Od /I "../../../../include" /I "../../../../contrib/include" /I "../../../../lib/mswdlld" /D "_DEBUG" /D DEBUG=1 /D "__WXDEBUG__" /D "WIN32" /D "_WINDOWS" /D "__WINDOWS__" /D "__WXMSW__" /D "__WIN95__" /D "__WIN32__" /D WINVER=0x0400 /D "STRICT" /D "WXUSINGDLL" /D "WXUSING_FL_DLL" /Yu"wx/wxprec.h" /FD /D BMP_DIR=\"../bitmaps/\" /c
+# ADD BASE MTL /nologo /D "_DEBUG" /mktyplib203 /o "NUL" /win32
+# ADD MTL /nologo /D "_DEBUG" /mktyplib203 /o "NUL" /win32
+# ADD BASE RSC /l 0x809 /d "_DEBUG"
+# ADD RSC /l 0x809 /d "_DEBUG"
+BSC32=bscmake.exe
+# ADD BASE BSC32 /nologo
+# ADD BSC32 /nologo
+LINK32=link.exe
+# ADD BASE LINK32 kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib comctl32.lib rpcrt4.lib wsock32.lib winmm.lib wxmswd.lib pngd.lib zlibd.lib jpegd.lib tiffd.lib regexd.lib fld.lib /nologo /subsystem:windows /debug /machine:I386 /nodefaultlib:"libcd.lib" /nodefaultlib:"libcid.lib" /nodefaultlib:"msvcrt.lib" /pdbtype:sept /libpath:"../../../../lib"
+# ADD LINK32 kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib comctl32.lib rpcrt4.lib wsock32.lib winmm.lib wxmsw233d.lib pngd.lib zlibd.lib jpegd.lib tiffd.lib regexd.lib fldlld.lib /nologo /subsystem:windows /debug /machine:I386 /nodefaultlib:"libcd.lib" /nodefaultlib:"libcid.lib" /nodefaultlib:"msvcrt.lib" /pdbtype:sept /libpath:"../../../../lib"
+
+!ELSEIF  "$(CFG)" == "fl_sample1 - Win32 Release DLL"
+
+# PROP BASE Use_MFC 0
+# PROP BASE Use_Debug_Libraries 0
+# PROP BASE Output_Dir "Release_DLL"
+# PROP BASE Intermediate_Dir "Release_DLL"
+# PROP BASE Ignore_Export_Lib 0
+# PROP BASE Target_Dir ""
+# PROP Use_MFC 0
+# PROP Use_Debug_Libraries 0
+# PROP Output_Dir "ReleaseDll"
+# PROP Intermediate_Dir "ReleaseDll"
+# PROP Ignore_Export_Lib 0
+# PROP Target_Dir ""
+# ADD BASE CPP /nologo /MD /W3 /GX /O1 /Ob2 /I "../../../../include" /I "../../../../contrib/include" /I "../../../../lib/msw" /D "NDEBUG" /D "WIN32" /D "_WINDOWS" /D "__WINDOWS__" /D "__WXMSW__" /D "__WIN95__" /D "__WIN32__" /D WINVER=0x0400 /D "STRICT" /FD /D BMP_DIR=\"../bitmaps/\" /c
+# ADD CPP /nologo /MD /W3 /GX /O2 /Ob2 /I "../../../../include" /I "../../../../contrib/include" /I "../../../../lib/mswdll" /D "NDEBUG" /D "WIN32" /D "_WINDOWS" /D "__WINDOWS__" /D "__WXMSW__" /D "__WIN95__" /D "__WIN32__" /D WINVER=0x0400 /D "STRICT" /D "WXUSINGDLL" /D "WXUSING_FL_DLL" /YX /FD /D BMP_DIR=\"../bitmaps/\" /c
+# ADD BASE MTL /nologo /D "NDEBUG" /mktyplib203 /o "NUL" /win32
+# ADD MTL /nologo /D "NDEBUG" /mktyplib203 /o "NUL" /win32
+# ADD BASE RSC /l 0x809 /d "NDEBUG"
+# ADD RSC /l 0x809 /d "NDEBUG"
+BSC32=bscmake.exe
+# ADD BASE BSC32 /nologo
+# ADD BSC32 /nologo
+LINK32=link.exe
+# ADD BASE LINK32 kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib comctl32.lib rpcrt4.lib wsock32.lib winmm.lib wxmsw.lib png.lib zlib.lib jpeg.lib tiff.lib regex.lib fl.lib /nologo /subsystem:windows /machine:I386 /nodefaultlib:"libc.lib" /nodefaultlib:"libci.lib" /nodefaultlib:"msvcrtd.lib" /libpath:"../../../../lib"
+# ADD LINK32 kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib comctl32.lib rpcrt4.lib wsock32.lib winmm.lib wxmsw233.lib png.lib zlib.lib jpeg.lib tiff.lib regex.lib fldll.lib /nologo /subsystem:windows /machine:I386 /nodefaultlib:"libc.lib" /nodefaultlib:"libci.lib" /nodefaultlib:"msvcrtd.lib" /libpath:"../../../../lib"
+
 !ENDIF 
 
@@ -88,4 +144,6 @@
 # Name "fl_sample1 - Win32 Release"
 # Name "fl_sample1 - Win32 Debug"
+# Name "fl_sample1 - Win32 Debug DLL"
+# Name "fl_sample1 - Win32 Release DLL"
 # Begin Source File
 
@@ -97,4 +155,11 @@
 
 # ADD CPP /Yc"wx/wxprec.h"
+
+!ELSEIF  "$(CFG)" == "fl_sample1 - Win32 Debug DLL"
+
+# ADD BASE CPP /Yc"wx/wxprec.h"
+# ADD CPP /Yc"wx/wxprec.h"
+
+!ELSEIF  "$(CFG)" == "fl_sample1 - Win32 Release DLL"
 
 !ENDIF 
Index: e:/dev/wxWindows/contrib/samples/fl/fl_sample1/makefile.vc
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/contrib/samples/fl/fl_sample1/makefile.vc,v
retrieving revision 1.1
diff -u -2 -r1.1 makefile.vc
--- e:/dev/wxWindows/contrib/samples/fl/fl_sample1/makefile.vc	2001/07/24 15:27:11	1.1
+++ e:/dev/wxWindows/contrib/samples/fl/fl_sample1/makefile.vc	2002/10/14 19:53:37
@@ -13,8 +13,22 @@
 
 EXTRAINC = -DBMP_DIR=\"../bitmaps/\"
+
+!if "$(WXUSING_FL_DLL)" == "1"
+# Define WXUSING_FL_DLL so the compiler knows it is set.
+EXTRAINC = $(EXTRAINC) -DWXUSING_FL_DLL
+# Set to use wxWin in DLL format...
+WXUSINGDLL = 1
+
 !if "$(FINAL)" == "1"
+EXTRALIBS = $(WXDIR)\lib\fldll.lib
+!else
+EXTRALIBS = $(WXDIR)\lib\fldlld.lib
+!endif
+!else
+!if "$(FINAL)" == "1"
 EXTRALIBS = $(WXDIR)\lib\fl.lib
 !else
 EXTRALIBS = $(WXDIR)\lib\fld.lib
+!endif
 !endif
 
Index: e:/dev/wxWindows/contrib/samples/fl/fl_sample2/fl_demo.dsp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/contrib/samples/fl/fl_sample2/fl_demo.dsp,v
retrieving revision 1.4
diff -u -2 -r1.4 fl_demo.dsp
--- e:/dev/wxWindows/contrib/samples/fl/fl_sample2/fl_demo.dsp	2001/11/19 16:05:23	1.4
+++ e:/dev/wxWindows/contrib/samples/fl/fl_sample2/fl_demo.dsp	2002/10/14 19:53:37
@@ -5,5 +5,5 @@
 # TARGTYPE "Win32 (x86) Application" 0x0101
 
-CFG=fl_sample2 - Win32 Debug
+CFG=fl_sample2 - Win32 Debug DLL
 !MESSAGE This is not a valid makefile. To build this project using NMAKE,
 !MESSAGE use the Export Makefile command and run
@@ -14,5 +14,5 @@
 !MESSAGE by defining the macro CFG on the command line. For example:
 !MESSAGE 
-!MESSAGE NMAKE /f "fl_demo.mak" CFG="fl_sample2 - Win32 Debug"
+!MESSAGE NMAKE /f "fl_demo.mak" CFG="fl_sample2 - Win32 Debug DLL"
 !MESSAGE 
 !MESSAGE Possible choices for configuration are:
@@ -20,4 +20,6 @@
 !MESSAGE "fl_sample2 - Win32 Release" (based on "Win32 (x86) Application")
 !MESSAGE "fl_sample2 - Win32 Debug" (based on "Win32 (x86) Application")
+!MESSAGE "fl_sample2 - Win32 Debug DLL" (based on "Win32 (x86) Application")
+!MESSAGE "fl_sample2 - Win32 Release DLL" (based on "Win32 (x86) Application")
 !MESSAGE 
 
@@ -82,4 +84,58 @@
 # ADD LINK32 kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib comctl32.lib rpcrt4.lib wsock32.lib winmm.lib wxmswd.lib pngd.lib zlibd.lib jpegd.lib tiffd.lib regexd.lib fld.lib /nologo /subsystem:windows /debug /machine:I386 /nodefaultlib:"libcd.lib" /nodefaultlib:"libcid.lib" /nodefaultlib:"msvcrt.lib" /pdbtype:sept /libpath:"../../../../lib"
 
+!ELSEIF  "$(CFG)" == "fl_sample2 - Win32 Debug DLL"
+
+# PROP BASE Use_MFC 0
+# PROP BASE Use_Debug_Libraries 1
+# PROP BASE Output_Dir "Debug_DLL"
+# PROP BASE Intermediate_Dir "Debug_DLL"
+# PROP BASE Ignore_Export_Lib 0
+# PROP BASE Target_Dir ""
+# PROP Use_MFC 0
+# PROP Use_Debug_Libraries 1
+# PROP Output_Dir "DebugDll"
+# PROP Intermediate_Dir "DebugDll"
+# PROP Ignore_Export_Lib 0
+# PROP Target_Dir ""
+# ADD BASE CPP /nologo /MDd /W3 /Gm /GX /ZI /Od /I "../../../../include" /I "../../../../contrib/include" /I "../../../../lib/mswd" /D "_DEBUG" /D DEBUG=1 /D "__WXDEBUG__" /D "WIN32" /D "_WINDOWS" /D "__WINDOWS__" /D "__WXMSW__" /D "__WIN95__" /D "__WIN32__" /D WINVER=0x0400 /D "STRICT" /Yu"wx/wxprec.h" /FD /D BMP_DIR=\"../bitmaps/\" /c
+# ADD CPP /nologo /MDd /W3 /Gm /GX /ZI /Od /I "../../../../include" /I "../../../../contrib/include" /I "../../../../lib/mswdlld" /D "_DEBUG" /D DEBUG=1 /D "__WXDEBUG__" /D "WIN32" /D "_WINDOWS" /D "__WINDOWS__" /D "__WXMSW__" /D "__WIN95__" /D "__WIN32__" /D WINVER=0x0400 /D "STRICT" /D "WXUSINGDLL" /D "WXUSING_FL_DLL" /Yu"wx/wxprec.h" /FD /D BMP_DIR=\"../bitmaps/\" /c
+# ADD BASE MTL /nologo /D "_DEBUG" /mktyplib203 /o "NUL" /win32
+# ADD MTL /nologo /D "_DEBUG" /mktyplib203 /o "NUL" /win32
+# ADD BASE RSC /l 0x809 /d "_DEBUG"
+# ADD RSC /l 0x809 /d "_DEBUG"
+BSC32=bscmake.exe
+# ADD BASE BSC32 /nologo
+# ADD BSC32 /nologo
+LINK32=link.exe
+# ADD BASE LINK32 kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib comctl32.lib rpcrt4.lib wsock32.lib winmm.lib wxmswd.lib pngd.lib zlibd.lib jpegd.lib tiffd.lib regexd.lib fld.lib /nologo /subsystem:windows /debug /machine:I386 /nodefaultlib:"libcd.lib" /nodefaultlib:"libcid.lib" /nodefaultlib:"msvcrt.lib" /pdbtype:sept /libpath:"../../../../lib"
+# ADD LINK32 kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib comctl32.lib rpcrt4.lib wsock32.lib winmm.lib wxmsw233d.lib pngd.lib zlibd.lib jpegd.lib tiffd.lib regexd.lib fldlld.lib /nologo /subsystem:windows /debug /machine:I386 /nodefaultlib:"libcd.lib" /nodefaultlib:"libcid.lib" /nodefaultlib:"msvcrt.lib" /pdbtype:sept /libpath:"../../../../lib"
+
+!ELSEIF  "$(CFG)" == "fl_sample2 - Win32 Release DLL"
+
+# PROP BASE Use_MFC 0
+# PROP BASE Use_Debug_Libraries 0
+# PROP BASE Output_Dir "Release_DLL"
+# PROP BASE Intermediate_Dir "Release_DLL"
+# PROP BASE Ignore_Export_Lib 0
+# PROP BASE Target_Dir ""
+# PROP Use_MFC 0
+# PROP Use_Debug_Libraries 0
+# PROP Output_Dir "ReleaseDll"
+# PROP Intermediate_Dir "ReleaseDll"
+# PROP Ignore_Export_Lib 0
+# PROP Target_Dir ""
+# ADD BASE CPP /nologo /MD /W3 /GX /O1 /Ob2 /I "../../../../include" /I "../../../../contrib/include" /I "../../../../lib/msw" /D "NDEBUG" /D "WIN32" /D "_WINDOWS" /D "__WINDOWS__" /D "__WXMSW__" /D "__WIN95__" /D "__WIN32__" /D WINVER=0x0400 /D "STRICT" /FD /D BMP_DIR=\"../bitmaps/\" /c
+# ADD CPP /nologo /MD /W3 /GX /O2 /Ob2 /I "../../../../include" /I "../../../../contrib/include" /I "../../../../lib/mswdll" /D "NDEBUG" /D "WIN32" /D "_WINDOWS" /D "__WINDOWS__" /D "__WXMSW__" /D "__WIN95__" /D "__WIN32__" /D WINVER=0x0400 /D "STRICT" /D "WXUSINGDLL" /D "WXUSING_FL_DLL" /YX /FD /D BMP_DIR=\"../bitmaps/\" /c
+# ADD BASE MTL /nologo /D "NDEBUG" /mktyplib203 /o "NUL" /win32
+# ADD MTL /nologo /D "NDEBUG" /mktyplib203 /o "NUL" /win32
+# ADD BASE RSC /l 0x809 /d "NDEBUG"
+# ADD RSC /l 0x809 /d "NDEBUG"
+BSC32=bscmake.exe
+# ADD BASE BSC32 /nologo
+# ADD BSC32 /nologo
+LINK32=link.exe
+# ADD BASE LINK32 kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib comctl32.lib rpcrt4.lib wsock32.lib winmm.lib wxmsw.lib png.lib zlib.lib jpeg.lib tiff.lib regex.lib fl.lib /nologo /subsystem:windows /machine:I386 /nodefaultlib:"libc.lib" /nodefaultlib:"libci.lib" /nodefaultlib:"msvcrtd.lib" /libpath:"../../../../lib"
+# ADD LINK32 kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib comctl32.lib rpcrt4.lib wsock32.lib winmm.lib wxmsw233.lib png.lib zlib.lib jpeg.lib tiff.lib regex.lib fldll.lib /nologo /subsystem:windows /machine:I386 /nodefaultlib:"libc.lib" /nodefaultlib:"libci.lib" /nodefaultlib:"msvcrtd.lib" /libpath:"../../../../lib"
+
 !ENDIF 
 
@@ -88,4 +144,6 @@
 # Name "fl_sample2 - Win32 Release"
 # Name "fl_sample2 - Win32 Debug"
+# Name "fl_sample2 - Win32 Debug DLL"
+# Name "fl_sample2 - Win32 Release DLL"
 # Begin Source File
 
@@ -97,4 +155,11 @@
 
 # ADD CPP /Yc"wx/wxprec.h"
+
+!ELSEIF  "$(CFG)" == "fl_sample2 - Win32 Debug DLL"
+
+# ADD BASE CPP /Yc"wx/wxprec.h"
+# ADD CPP /Yc"wx/wxprec.h"
+
+!ELSEIF  "$(CFG)" == "fl_sample2 - Win32 Release DLL"
 
 !ENDIF 
Index: e:/dev/wxWindows/contrib/samples/fl/fl_sample2/makefile.vc
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/contrib/samples/fl/fl_sample2/makefile.vc,v
retrieving revision 1.1
diff -u -2 -r1.1 makefile.vc
--- e:/dev/wxWindows/contrib/samples/fl/fl_sample2/makefile.vc	2001/07/24 15:27:12	1.1
+++ e:/dev/wxWindows/contrib/samples/fl/fl_sample2/makefile.vc	2002/10/14 19:53:37
@@ -13,8 +13,22 @@
 
 EXTRAINC = -DBMP_DIR=\"../bitmaps/\"
+
+!if "$(WXUSING_FL_DLL)" == "1"
+# Define WXUSING_FL_DLL so the compiler knows it is set.
+EXTRAINC = $(EXTRAINC) -DWXUSING_FL_DLL
+# Set to use wxWin in DLL format...
+WXUSINGDLL = 1
+
 !if "$(FINAL)" == "1"
+EXTRALIBS = $(WXDIR)\lib\fldll.lib
+!else
+EXTRALIBS = $(WXDIR)\lib\fldlld.lib
+!endif
+!else
+!if "$(FINAL)" == "1"
 EXTRALIBS = $(WXDIR)\lib\fl.lib
 !else
 EXTRALIBS = $(WXDIR)\lib\fld.lib
+!endif
 !endif
 
Index: e:/dev/wxWindows/contrib/samples/fl/fl_sample3/fl_demo.dsp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/contrib/samples/fl/fl_sample3/fl_demo.dsp,v
retrieving revision 1.4
diff -u -2 -r1.4 fl_demo.dsp
--- e:/dev/wxWindows/contrib/samples/fl/fl_sample3/fl_demo.dsp	2001/11/19 16:05:24	1.4
+++ e:/dev/wxWindows/contrib/samples/fl/fl_sample3/fl_demo.dsp	2002/10/14 19:53:37
@@ -5,5 +5,5 @@
 # TARGTYPE "Win32 (x86) Application" 0x0101
 
-CFG=fl_sample3 - Win32 Debug
+CFG=fl_sample3 - Win32 Debug DLL
 !MESSAGE This is not a valid makefile. To build this project using NMAKE,
 !MESSAGE use the Export Makefile command and run
@@ -14,5 +14,5 @@
 !MESSAGE by defining the macro CFG on the command line. For example:
 !MESSAGE 
-!MESSAGE NMAKE /f "fl_demo.mak" CFG="fl_sample3 - Win32 Debug"
+!MESSAGE NMAKE /f "fl_demo.mak" CFG="fl_sample3 - Win32 Debug DLL"
 !MESSAGE 
 !MESSAGE Possible choices for configuration are:
@@ -20,4 +20,6 @@
 !MESSAGE "fl_sample3 - Win32 Release" (based on "Win32 (x86) Application")
 !MESSAGE "fl_sample3 - Win32 Debug" (based on "Win32 (x86) Application")
+!MESSAGE "fl_sample3 - Win32 Debug DLL" (based on "Win32 (x86) Application")
+!MESSAGE "fl_sample3 - Win32 Release DLL" (based on "Win32 (x86) Application")
 !MESSAGE 
 
@@ -82,4 +84,58 @@
 # ADD LINK32 kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib comctl32.lib rpcrt4.lib wsock32.lib winmm.lib wxmswd.lib pngd.lib zlibd.lib jpegd.lib tiffd.lib regexd.lib fld.lib /nologo /subsystem:windows /debug /machine:I386 /nodefaultlib:"libcd.lib" /nodefaultlib:"libcid.lib" /nodefaultlib:"msvcrt.lib" /pdbtype:sept /libpath:"../../../../lib"
 
+!ELSEIF  "$(CFG)" == "fl_sample3 - Win32 Debug DLL"
+
+# PROP BASE Use_MFC 0
+# PROP BASE Use_Debug_Libraries 1
+# PROP BASE Output_Dir "Debug_DLL"
+# PROP BASE Intermediate_Dir "Debug_DLL"
+# PROP BASE Ignore_Export_Lib 0
+# PROP BASE Target_Dir ""
+# PROP Use_MFC 0
+# PROP Use_Debug_Libraries 1
+# PROP Output_Dir "DebugDll"
+# PROP Intermediate_Dir "DebugDll"
+# PROP Ignore_Export_Lib 0
+# PROP Target_Dir ""
+# ADD BASE CPP /nologo /MDd /W3 /Gm /GX /ZI /Od /I "../../../../include" /I "../../../../contrib/include" /I "../../../../lib/mswd" /D "_DEBUG" /D DEBUG=1 /D "__WXDEBUG__" /D "WIN32" /D "_WINDOWS" /D "__WINDOWS__" /D "__WXMSW__" /D "__WIN95__" /D "__WIN32__" /D WINVER=0x0400 /D "STRICT" /Yu"wx/wxprec.h" /FD /D BMP_DIR=\"../bitmaps/\" /c
+# ADD CPP /nologo /MDd /W3 /Gm /GX /ZI /Od /I "../../../../include" /I "../../../../contrib/include" /I "../../../../lib/mswdlld" /D "_DEBUG" /D DEBUG=1 /D "__WXDEBUG__" /D "WIN32" /D "_WINDOWS" /D "__WINDOWS__" /D "__WXMSW__" /D "__WIN95__" /D "__WIN32__" /D WINVER=0x0400 /D "STRICT" /D "WXUSINGDLL" /D "WXUSING_FL_DLL" /Yu"wx/wxprec.h" /FD /D BMP_DIR=\"../bitmaps/\" /c
+# ADD BASE MTL /nologo /D "_DEBUG" /mktyplib203 /o "NUL" /win32
+# ADD MTL /nologo /D "_DEBUG" /mktyplib203 /o "NUL" /win32
+# ADD BASE RSC /l 0x809 /d "_DEBUG"
+# ADD RSC /l 0x809 /d "_DEBUG"
+BSC32=bscmake.exe
+# ADD BASE BSC32 /nologo
+# ADD BSC32 /nologo
+LINK32=link.exe
+# ADD BASE LINK32 kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib comctl32.lib rpcrt4.lib wsock32.lib winmm.lib wxmswd.lib pngd.lib zlibd.lib jpegd.lib tiffd.lib regexd.lib fld.lib /nologo /subsystem:windows /debug /machine:I386 /nodefaultlib:"libcd.lib" /nodefaultlib:"libcid.lib" /nodefaultlib:"msvcrt.lib" /pdbtype:sept /libpath:"../../../../lib"
+# ADD LINK32 kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib comctl32.lib rpcrt4.lib wsock32.lib winmm.lib wxmsw233d.lib pngd.lib zlibd.lib jpegd.lib tiffd.lib regexd.lib fldlld.lib /nologo /subsystem:windows /debug /machine:I386 /nodefaultlib:"libcd.lib" /nodefaultlib:"libcid.lib" /nodefaultlib:"msvcrt.lib" /pdbtype:sept /libpath:"../../../../lib"
+
+!ELSEIF  "$(CFG)" == "fl_sample3 - Win32 Release DLL"
+
+# PROP BASE Use_MFC 0
+# PROP BASE Use_Debug_Libraries 0
+# PROP BASE Output_Dir "Release_DLL"
+# PROP BASE Intermediate_Dir "Release_DLL"
+# PROP BASE Ignore_Export_Lib 0
+# PROP BASE Target_Dir ""
+# PROP Use_MFC 0
+# PROP Use_Debug_Libraries 0
+# PROP Output_Dir "ReleaseDll"
+# PROP Intermediate_Dir "ReleaseDll"
+# PROP Ignore_Export_Lib 0
+# PROP Target_Dir ""
+# ADD BASE CPP /nologo /MD /W3 /GX /O1 /Ob2 /I "../../../../include" /I "../../../../contrib/include" /I "../../../../lib/msw" /D "NDEBUG" /D "WIN32" /D "_WINDOWS" /D "__WINDOWS__" /D "__WXMSW__" /D "__WIN95__" /D "__WIN32__" /D WINVER=0x0400 /D "STRICT" /FD /D BMP_DIR=\"../bitmaps/\" /c
+# ADD CPP /nologo /MD /W3 /GX /O2 /Ob2 /I "../../../../include" /I "../../../../contrib/include" /I "../../../../lib/mswdll" /D "NDEBUG" /D "WIN32" /D "_WINDOWS" /D "__WINDOWS__" /D "__WXMSW__" /D "__WIN95__" /D "__WIN32__" /D WINVER=0x0400 /D "STRICT" /D "WXUSINGDLL" /D "WXUSING_FL_DLL" /YX /FD /D BMP_DIR=\"../bitmaps/\" /c
+# ADD BASE MTL /nologo /D "NDEBUG" /mktyplib203 /o "NUL" /win32
+# ADD MTL /nologo /D "NDEBUG" /mktyplib203 /o "NUL" /win32
+# ADD BASE RSC /l 0x809 /d "NDEBUG"
+# ADD RSC /l 0x809 /d "NDEBUG"
+BSC32=bscmake.exe
+# ADD BASE BSC32 /nologo
+# ADD BSC32 /nologo
+LINK32=link.exe
+# ADD BASE LINK32 kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib comctl32.lib rpcrt4.lib wsock32.lib winmm.lib wxmsw.lib png.lib zlib.lib jpeg.lib tiff.lib regex.lib fl.lib /nologo /subsystem:windows /machine:I386 /nodefaultlib:"libc.lib" /nodefaultlib:"libci.lib" /nodefaultlib:"msvcrtd.lib" /libpath:"../../../../lib"
+# ADD LINK32 kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib comctl32.lib rpcrt4.lib wsock32.lib winmm.lib wxmsw233.lib png.lib zlib.lib jpeg.lib tiff.lib regex.lib fldll.lib /nologo /subsystem:windows /machine:I386 /nodefaultlib:"libc.lib" /nodefaultlib:"libci.lib" /nodefaultlib:"msvcrtd.lib" /libpath:"../../../../lib"
+
 !ENDIF 
 
@@ -88,4 +144,6 @@
 # Name "fl_sample3 - Win32 Release"
 # Name "fl_sample3 - Win32 Debug"
+# Name "fl_sample3 - Win32 Debug DLL"
+# Name "fl_sample3 - Win32 Release DLL"
 # Begin Source File
 
@@ -97,4 +155,11 @@
 
 # ADD CPP /Yc"wx/wxprec.h"
+
+!ELSEIF  "$(CFG)" == "fl_sample3 - Win32 Debug DLL"
+
+# ADD BASE CPP /Yc"wx/wxprec.h"
+# ADD CPP /Yc"wx/wxprec.h"
+
+!ELSEIF  "$(CFG)" == "fl_sample3 - Win32 Release DLL"
 
 !ENDIF 
Index: e:/dev/wxWindows/contrib/samples/fl/fl_sample3/makefile.vc
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/contrib/samples/fl/fl_sample3/makefile.vc,v
retrieving revision 1.1
diff -u -2 -r1.1 makefile.vc
--- e:/dev/wxWindows/contrib/samples/fl/fl_sample3/makefile.vc	2001/07/24 15:27:12	1.1
+++ e:/dev/wxWindows/contrib/samples/fl/fl_sample3/makefile.vc	2002/10/14 19:53:37
@@ -13,8 +13,22 @@
 
 EXTRAINC = -DBMP_DIR=\"../bitmaps/\"
+
+!if "$(WXUSING_FL_DLL)" == "1"
+# Define WXUSING_FL_DLL so the compiler knows it is set.
+EXTRAINC = $(EXTRAINC) -DWXUSING_FL_DLL
+# Set to use wxWin in DLL format...
+WXUSINGDLL = 1
+
 !if "$(FINAL)" == "1"
+EXTRALIBS = $(WXDIR)\lib\fldll.lib
+!else
+EXTRALIBS = $(WXDIR)\lib\fldlld.lib
+!endif
+!else
+!if "$(FINAL)" == "1"
 EXTRALIBS = $(WXDIR)\lib\fl.lib
 !else
 EXTRALIBS = $(WXDIR)\lib\fld.lib
+!endif
 !endif
 
Index: e:/dev/wxWindows/contrib/src/fl/flVC.dsp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/contrib/src/fl/flVC.dsp,v
retrieving revision 1.3
diff -u -2 -r1.3 flVC.dsp
--- e:/dev/wxWindows/contrib/src/fl/flVC.dsp	2002/01/21 22:34:41	1.3
+++ e:/dev/wxWindows/contrib/src/fl/flVC.dsp	2002/10/14 19:53:53
@@ -3,7 +3,8 @@
 # ** DO NOT EDIT **
 
+# TARGTYPE "Win32 (x86) Dynamic-Link Library" 0x0102
 # TARGTYPE "Win32 (x86) Static Library" 0x0104
 
-CFG=flVC - Win32 Debug
+CFG=flVC - Win32 Release DLL
 !MESSAGE This is not a valid makefile. To build this project using NMAKE,
 !MESSAGE use the Export Makefile command and run
@@ -14,5 +15,5 @@
 !MESSAGE by defining the macro CFG on the command line. For example:
 !MESSAGE 
-!MESSAGE NMAKE /f "flVC.mak" CFG="flVC - Win32 Debug"
+!MESSAGE NMAKE /f "flVC.mak" CFG="flVC - Win32 Release DLL"
 !MESSAGE 
 !MESSAGE Possible choices for configuration are:
@@ -20,4 +21,6 @@
 !MESSAGE "flVC - Win32 Release" (based on "Win32 (x86) Static Library")
 !MESSAGE "flVC - Win32 Debug" (based on "Win32 (x86) Static Library")
+!MESSAGE "flVC - Win32 Debug DLL" (based on "Win32 (x86) Dynamic-Link Library")
+!MESSAGE "flVC - Win32 Release DLL" (based on "Win32 (x86) Dynamic-Link Library")
 !MESSAGE 
 
@@ -26,6 +29,4 @@
 # PROP Scc_ProjName ""
 # PROP Scc_LocalPath ""
-CPP=cl.exe
-RSC=rc.exe
 
 !IF  "$(CFG)" == "flVC - Win32 Release"
@@ -41,7 +42,9 @@
 # PROP Intermediate_Dir "Release"
 # PROP Target_Dir ""
+LINK32=link.exe
+CPP=cl.exe
 # ADD BASE CPP /nologo /W3 /GX /O2 /D "WIN32" /D "NDEBUG" /D "_WINDOWS" /YX /FD /c
-# ADD CPP /nologo /MD /W3 /GX /O1 /Ob2 /I "../../../include" /I "../../include" /I "../../../lib/msw" /D "WIN32" /D "_WINDOWS" /D "__WINDOWS__" /D "__WXMSW__" /D "__HACK_MY_MSDEV40__" /D "__WIN95__" /D "__WIN32__" /D WINVER=0x0400 /D "STRICT" /FD /c
-# SUBTRACT CPP /YX
+# ADD CPP /nologo /MD /W3 /GX /O1 /Ob2 /I "../../../include" /I "../../include" /I "../../../lib/msw" /D "WIN32" /D "_WINDOWS" /D "__WINDOWS__" /D "__WXMSW__" /D "__HACK_MY_MSDEV40__" /D "__WIN95__" /D "__WIN32__" /D WINVER=0x0400 /D "STRICT" /YX /FD /c
+RSC=rc.exe
 # ADD BASE RSC /l 0x809
 # ADD RSC /l 0x809
@@ -65,7 +68,9 @@
 # PROP Intermediate_Dir "Debug"
 # PROP Target_Dir ""
+LINK32=link.exe
+CPP=cl.exe
 # ADD BASE CPP /nologo /W3 /GX /Z7 /Od /D "WIN32" /D "_DEBUG" /D "_WINDOWS" /YX /FD /c
-# ADD CPP /nologo /MDd /W3 /GX /Zi /Od /I "../../../include" /I "../../include" /I "../../../lib/mswd" /D "WIN32" /D "_DEBUG" /D "_WINDOWS" /D "__WINDOWS__" /D "__WXMSW__" /D DEBUG=1 /D "__WXDEBUG__" /D "__WIN95__" /D "__WIN32__" /D WINVER=0x0400 /D "STRICT" /FD /c
-# SUBTRACT CPP /YX
+# ADD CPP /nologo /MDd /W3 /GX /Zi /Od /I "../../../include" /I "../../include" /I "../../../lib/mswd" /D "WIN32" /D "_DEBUG" /D "_WINDOWS" /D "__WINDOWS__" /D "__WXMSW__" /D DEBUG=1 /D "__WXDEBUG__" /D "__WIN95__" /D "__WIN32__" /D WINVER=0x0400 /D "STRICT" /YX /FD /c
+RSC=rc.exe
 # ADD BASE RSC /l 0x809
 # ADD RSC /l 0x809
@@ -77,4 +82,72 @@
 # ADD LIB32 /nologo /out:"..\..\..\lib\fld.lib"
 
+!ELSEIF  "$(CFG)" == "flVC - Win32 Debug DLL"
+
+# PROP BASE Use_MFC 0
+# PROP BASE Use_Debug_Libraries 1
+# PROP BASE Output_Dir "Debug_DLL"
+# PROP BASE Intermediate_Dir "Debug_DLL"
+# PROP BASE Target_Dir ""
+# PROP Use_MFC 0
+# PROP Use_Debug_Libraries 1
+# PROP Output_Dir "..\..\..\lib"
+# PROP Intermediate_Dir "Debug_DLL"
+# PROP Ignore_Export_Lib 0
+# PROP Target_Dir ""
+LIB32=link.exe -lib
+# ADD BASE LIB32 /nologo /out:"..\..\..\lib\fld.lib"
+# ADD LIB32 /nologo /out:"..\..\..\lib\fld.lib"
+CPP=cl.exe
+# ADD BASE CPP /nologo /MDd /W3 /GX /Zi /Od /I "../../../include" /I "../../include" /I "../../../lib/mswd" /D "WIN32" /D "_DEBUG" /D "_WINDOWS" /D "__WINDOWS__" /D "__WXMSW__" /D DEBUG=1 /D "__WXDEBUG__" /D "__WIN95__" /D "__WIN32__" /D WINVER=0x0400 /D "STRICT" /FD /c
+# SUBTRACT BASE CPP /YX
+# ADD CPP /nologo /MDd /W3 /GX /Zi /Od /I "../../../include" /I "../../include" /I "../../../lib/mswdlld" /D "WIN32" /D "_DEBUG" /D "_WINDOWS" /D "__WINDOWS__" /D "__WXMSW__" /D DEBUG=1 /D "__WXDEBUG__" /D "__WIN95__" /D "__WIN32__" /D WINVER=0x0400 /D "STRICT" /D "WXUSINGDLL" /D "WXMAKING_FL_DLL" /YX /FD /c
+# SUBTRACT CPP /u
+MTL=midl.exe
+RSC=rc.exe
+# ADD BASE RSC /l 0x809
+# ADD RSC /l 0x809
+BSC32=bscmake.exe
+# ADD BASE BSC32 /nologo
+# ADD BSC32 /nologo
+LINK32=link.exe
+# ADD BASE LINK32 /machine:IX86
+# ADD LINK32 kernel32.lib user32.lib gdi32.lib advapi32.lib comdlg32.lib shell32.lib ole32.lib oleaut32.lib odbc32.lib uuid.lib rpcrt4.lib comctl32.lib wsock32.lib winmm.lib wxmsw233d.lib /nologo /version:1.0 /dll /debug /machine:IX86 /out:"../../../lib/fldlld.dll" /libpath:"../../../lib"
+# SUBTRACT LINK32 /pdb:none
+
+!ELSEIF  "$(CFG)" == "flVC - Win32 Release DLL"
+
+# PROP BASE Use_MFC 0
+# PROP BASE Use_Debug_Libraries 1
+# PROP BASE Output_Dir "flVC___Win32_Release_DLL"
+# PROP BASE Intermediate_Dir "flVC___Win32_Release_DLL"
+# PROP BASE Ignore_Export_Lib 0
+# PROP BASE Target_Dir ""
+# PROP Use_MFC 0
+# PROP Use_Debug_Libraries 1
+# PROP Output_Dir "..\..\..\lib"
+# PROP Intermediate_Dir "Release_DLL"
+# PROP Ignore_Export_Lib 0
+# PROP Target_Dir ""
+LIB32=link.exe -lib
+# ADD BASE LIB32 /nologo /out:"..\..\..\lib\fld.lib"
+# ADD LIB32 /nologo /out:"..\..\..\lib\fld.lib"
+CPP=cl.exe
+# ADD BASE CPP /nologo /MDd /W3 /GX /Zi /Od /I "../../../include" /I "../../include" /I "../../../lib/mswdlld" /D "WIN32" /D "_DEBUG" /D "_WINDOWS" /D "__WINDOWS__" /D "__WXMSW__" /D DEBUG=1 /D "__WXDEBUG__" /D "__WIN95__" /D "__WIN32__" /D WINVER=0x0400 /D "STRICT" /D WXUSINGDLL=1 /D "WXBUILD_FL_DLL" /YX /FD /c
+# SUBTRACT BASE CPP /u
+# ADD CPP /nologo /MD /W3 /GX /Zi /O2 /Ob2 /I "../../../include" /I "../../include" /I "../../../lib/mswdlld" /D "WIN32" /D "NDEBUG" /D "_WINDOWS" /D "__WINDOWS__" /D "__WXMSW__" /D "__WIN95__" /D "__WIN32__" /D WINVER=0x0400 /D "STRICT" /D "WXUSINGDLL" /D "WXMAKING_FL_DLL" /YX /FD /c
+# SUBTRACT CPP /u
+MTL=midl.exe
+RSC=rc.exe
+# ADD BASE RSC /l 0x809
+# ADD RSC /l 0x809
+BSC32=bscmake.exe
+# ADD BASE BSC32 /nologo
+# ADD BSC32 /nologo
+LINK32=link.exe
+# ADD BASE LINK32 kernel32.lib user32.lib gdi32.lib advapi32.lib comdlg32.lib shell32.lib ole32.lib oleaut32.lib odbc32.lib uuid.lib rpcrt4.lib comctl32.lib wsock32.lib winmm.lib wxmsw233d.lib /nologo /version:1.0 /dll /debug /machine:IX86 /out:"../../../lib/fldlld.dll" /libpath:"../../../lib"
+# SUBTRACT BASE LINK32 /pdb:none
+# ADD LINK32 kernel32.lib user32.lib gdi32.lib advapi32.lib comdlg32.lib shell32.lib ole32.lib oleaut32.lib odbc32.lib uuid.lib rpcrt4.lib comctl32.lib wsock32.lib winmm.lib wxmsw233.lib /nologo /version:1.0 /dll /machine:IX86 /out:"../../../lib/fldll.dll" /libpath:"../../../lib"
+# SUBTRACT LINK32 /pdb:none /incremental:no /debug
+
 !ENDIF 
 
@@ -83,4 +156,6 @@
 # Name "flVC - Win32 Release"
 # Name "flVC - Win32 Debug"
+# Name "flVC - Win32 Debug DLL"
+# Name "flVC - Win32 Release DLL"
 # Begin Group "Headers"
 
@@ -117,4 +192,8 @@
 
 SOURCE=..\..\include\wx\fl\dyntbarhnd.h
+# End Source File
+# Begin Source File
+
+SOURCE=..\..\include\wx\fl\fldefs.h
 # End Source File
 # Begin Source File
Index: e:/dev/wxWindows/contrib/src/fl/newbmpbtn.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/contrib/src/fl/newbmpbtn.cpp,v
retrieving revision 1.8
diff -u -2 -r1.8 newbmpbtn.cpp
--- e:/dev/wxWindows/contrib/src/fl/newbmpbtn.cpp	2002/06/23 13:38:50	1.8
+++ e:/dev/wxWindows/contrib/src/fl/newbmpbtn.cpp	2002/10/14 19:53:53
@@ -526,5 +526,5 @@
 #ifdef __WXMSW__
     // Map to system colours
-    (void) MapBitmap(destBmp->GetHBITMAP(), destBmp->GetWidth(), destBmp->GetHeight());
+    (void) wxToolBar::MapBitmap(destBmp->GetHBITMAP(), destBmp->GetWidth(), destBmp->GetHeight());
 #endif    
 }
@@ -800,48 +800,2 @@
 }
 
-#ifdef __WXMSW__
-WXHBITMAP wxNewBitmapButton::MapBitmap(WXHBITMAP bitmap, int width, int height)
-{
-    MemoryHDC hdcMem;
-
-    if ( !hdcMem )
-    {
-        wxLogLastError(_T("CreateCompatibleDC"));
-
-        return bitmap;
-    }
-
-    SelectInHDC bmpInHDC(hdcMem, (HBITMAP)bitmap);
-
-    if ( !bmpInHDC )
-    {
-        wxLogLastError(_T("SelectObject"));
-
-        return bitmap;
-    }
-
-    wxCOLORMAP *cmap = wxGetStdColourMap();
-
-    for ( int i = 0; i < width; i++ )
-    {
-        for ( int j = 0; j < height; j++ )
-        {
-            COLORREF pixel = ::GetPixel(hdcMem, i, j);
-
-            for ( size_t k = 0; k < wxSTD_COL_MAX; k++ )
-            {
-                COLORREF col = cmap[k].from;
-                if ( abs(GetRValue(pixel) - GetRValue(col)) < 10 &&
-                     abs(GetGValue(pixel) - GetGValue(col)) < 10 &&
-                     abs(GetBValue(pixel) - GetBValue(col)) < 10 )
-                {
-                    ::SetPixel(hdcMem, i, j, cmap[k].to);
-                    break;
-                }
-            }
-        }
-    }
-
-    return bitmap;
-}
-#endif

 	  	 
