Index: interface/wx/gauge.h
===================================================================
--- interface/wx/gauge.h	(revision 56547)
+++ interface/wx/gauge.h	(working copy)
@@ -56,8 +56,7 @@
         @param id
             Window identifier.
         @param range
-            Integer range (maximum value) of the gauge. It is ignored when the
-            gauge is used in indeterminate mode.
+            Integer range (maximum value) of the gauge.
         @param pos
             Window position.
         @param size
