diff -bru wxWidgets/include/wx/filedlg.h wxWidgets.patched/include/wx/filedlg.h
--- wxWidgets/include/wx/filedlg.h	2014-09-14 16:31:22.000000000 -0700
+++ wxWidgets.patched/include/wx/filedlg.h	2014-09-17 01:08:10.000000000 -0700
@@ -47,7 +47,8 @@
     wxFD_FILE_MUST_EXIST   = 0x0010,
     wxFD_MULTIPLE          = 0x0020,
     wxFD_CHANGE_DIR        = 0x0080,
-    wxFD_PREVIEW           = 0x0100
+    wxFD_PREVIEW           = 0x0100,
+    wxFD_SHOWHIDDEN        = 0x0200
 };
 
 #define wxFD_DEFAULT_STYLE      wxFD_OPEN
diff -bru wxWidgets/interface/wx/filedlg.h wxWidgets.patched/interface/wx/filedlg.h
--- wxWidgets/interface/wx/filedlg.h	2014-09-14 16:31:31.000000000 -0700
+++ wxWidgets.patched/interface/wx/filedlg.h	2014-09-17 01:08:26.000000000 -0700
@@ -14,7 +14,8 @@
     wxFD_FILE_MUST_EXIST   = 0x0010,
     wxFD_MULTIPLE          = 0x0020,
     wxFD_CHANGE_DIR        = 0x0080,
-    wxFD_PREVIEW           = 0x0100
+    wxFD_PREVIEW           = 0x0100,
+    wxFD_SHOWHIDDEN        = 0x0200
 };
 
 #define wxFD_DEFAULT_STYLE      wxFD_OPEN
@@ -141,6 +142,8 @@
     @style{wxFD_PREVIEW}
            Show the preview of the selected files (currently only supported by
            wxGTK).
+    @style{wxFD_SHOWHIDDEN}
+           Show hidden files.
     @endStyleTable
 
     @library{wxcore}
diff -bru wxWidgets/samples/dialogs/dialogs.cpp wxWidgets.patched/samples/dialogs/dialogs.cpp
--- wxWidgets/samples/dialogs/dialogs.cpp	2014-09-14 16:31:34.000000000 -0700
+++ wxWidgets.patched/samples/dialogs/dialogs.cpp	2014-09-17 01:12:49.000000000 -0700
@@ -1444,7 +1444,7 @@
                                         wxFileSelectorDefaultWildcardStr,
                                         wxFileSelectorDefaultWildcardStr
                                     ),
-                                    wxFD_OPEN|wxFD_CHANGE_DIR|wxFD_PREVIEW|wxFD_NO_FOLLOW,
+                                    wxFD_OPEN|wxFD_CHANGE_DIR|wxFD_PREVIEW|wxFD_NO_FOLLOW|wxFD_SHOWHIDDEN,
                                     this
                                    );
 
diff -bru wxWidgets/src/gtk/filedlg.cpp wxWidgets.patched/src/gtk/filedlg.cpp
--- wxWidgets/src/gtk/filedlg.cpp	2014-09-14 16:30:48.000000000 -0700
+++ wxWidgets.patched/src/gtk/filedlg.cpp	2014-09-19 01:36:54.000000000 -0700
@@ -146,6 +146,19 @@
 
 } // extern "C"
 
+extern "C" {
+static gboolean gtk_frame_map_callback( GtkFileChooser *file_chooser,
+                        GdkEvent * WXUNUSED(event),
+                        wxFileDialog *dlg )
+{
+    if( dlg->GetWindowStyle() & wxFD_SHOWHIDDEN )
+        gtk_file_chooser_set_show_hidden( file_chooser, TRUE );
+    else
+        gtk_file_chooser_set_show_hidden( file_chooser, FALSE );
+    return FALSE;
+}
+}
+
 void wxFileDialog::AddChildGTK(wxWindowGTK* child)
 {
     // allow dialog to be resized smaller horizontally
@@ -236,7 +249,6 @@
 
     if ( style & wxFD_MULTIPLE )
         gtk_file_chooser_set_select_multiple(file_chooser, true);
-
     // local-only property could be set to false to allow non-local files to be
     // loaded. In that case get/set_uri(s) should be used instead of
     // get/set_filename(s) everywhere and the GtkFileChooserDialog should
@@ -251,6 +263,9 @@
     g_signal_connect (m_widget, "selection-changed",
         G_CALLBACK (gtk_filedialog_selchanged_callback), this);
 
+    g_signal_connect (m_widget, "map_event",
+                      G_CALLBACK (gtk_frame_map_callback), this);
+
     // deal with extensions/filters
     SetWildcard(wildCard);
 
diff -bru wxWidgets/src/msw/filedlg.cpp wxWidgets.patched/src/msw/filedlg.cpp
--- wxWidgets/src/msw/filedlg.cpp	2014-09-14 16:31:01.000000000 -0700
+++ wxWidgets.patched/src/msw/filedlg.cpp	2014-09-17 01:13:54.000000000 -0700
@@ -502,6 +502,8 @@
 
     if ( HasFdFlag(wxFD_FILE_MUST_EXIST) )
         msw_flags |= OFN_PATHMUSTEXIST | OFN_FILEMUSTEXIST;
+    if( HasFdFlag( wxFD_SHOWHIDDEN ) )
+        msw_flags |= OFN_FORCESHOWHIDDEN;
     /*
         If the window has been moved the programmer is probably
         trying to center or position it.  Thus we set the callback

