Index: include/wx/mac/carbon/app.h
===================================================================
--- include/wx/mac/carbon/app.h	(revision 53318)
+++ include/wx/mac/carbon/app.h	(working copy)
@@ -95,6 +95,7 @@
     static long           s_macPreferencesMenuItemId ;
     static long           s_macExitMenuItemId ;
     static wxString       s_macHelpMenuTitleName ;
+    static wxString       s_macWindowMenuTitleName ;
 
     WXEVENTREF            MacGetCurrentEvent() { return m_macCurrentEvent ; }
     void                  MacHandleOneEvent( WXEVENTREF ev ) ;
Index: src/mac/carbon/app.cpp
===================================================================
--- src/mac/carbon/app.cpp	(revision 53318)
+++ src/mac/carbon/app.cpp	(working copy)
@@ -81,6 +81,7 @@
 long      wxApp::s_macPreferencesMenuItemId = wxID_PREFERENCES ;
 long      wxApp::s_macExitMenuItemId = wxID_EXIT ;
 wxString  wxApp::s_macHelpMenuTitleName = wxT("&Help") ;
+wxString  wxApp::s_macWindowMenuTitleName = wxT("&Window") ;
 
 bool      wxApp::sm_isEmbedded = false; // Normally we're not a plugin
 
Index: src/mac/carbon/menu.cpp
===================================================================
--- src/mac/carbon/menu.cpp	(revision 53318)
+++ src/mac/carbon/menu.cpp	(working copy)
@@ -831,8 +831,8 @@
                 }
             }
         }
-
-        else if ( ( m_titles[i] == wxT("Window") || m_titles[i] == wxT("&Window") )
+        
+        else if ( ( m_titles[i] == wxT("Window") || m_titles[i] == wxT("&Window") || m_titles[i] == wxApp::s_macWindowMenuTitleName ) 
                 && GetAutoWindowMenu() )
         {
             if ( MacGetWindowMenuHMenu() == NULL )

 	  	 
