--- src\aui\framemanager.cpp    Mon Dec 25 01:21:38 2006
+++ framemanager.cpp    Mon Apr 16 18:23:53 2007
@@ -1308,6 +1308,19 @@
     pane_part.Replace(wxT("\a"), wxT("|"));
     pane_part.Replace(wxT("\b"), wxT(";"));
 
+#ifdef __WXMSW__
+    // if users change resolution or go from multi- to single screen,
+    // the floating position can end up outside screen where it is unreachable
+    // wxWidgets does not currently offer metrics for multi-screen setup
+    // so we have to use native api (SM_CXVIRTUALSCREEN=78 && SM_CYVIRTUALSCREEN=79)
+    const wxRect screen(0, 0, ::GetSystemMetrics(78), ::GetSystemMetrics(79));
+    if (!screen.Contains(pane.floating_pos.x, pane.floating_pos.y))
+    {
+        pane.floating_pos.x = 0;
+        pane.floating_pos.y = 0;
+    }
+#endif //__WXMSW__
+
     return;
 }
 

 	  	 
