diff --git a/src/osx/carbon/dataobj.cpp b/src/osx/carbon/dataobj.cpp
index b48ce93..753af08 100644
--- a/src/osx/carbon/dataobj.cpp
+++ b/src/osx/carbon/dataobj.cpp
@@ -629,6 +629,11 @@ void wxDataObject::AddSupportedTypes( void* cfarray)
         {
             CFArrayAppendValue((CFMutableArrayRef)cfarray, kUTTypePDF);
         }
+        else if (dataFormat.GetType() == wxDF_PRIVATE )
+        {
+            CFArrayAppendValue((CFMutableArrayRef)cfarray,
+                               (CFStringRef)dataFormat.GetFormatId());
+        }
     }
     delete[] array;
 }
