Index: src/generic/listctrl.cpp
===================================================================
--- src/generic/listctrl.cpp	(Revision 66737)
+++ src/generic/listctrl.cpp	(Arbeitskopie)
@@ -4024,6 +4024,8 @@
     wxListLineData *line = new wxListLineData(this);
 
     line->SetItem( item.m_col, item );
+    if ( item.m_mask & wxLIST_MASK_IMAGE )
+        m_lineHeight=0; // reset the buffered height because the new image might have a different one
 
     m_lines.Insert( line, id );
 
