Index: auidemo.cpp
===================================================================
--- auidemo.cpp	(revision 65792)
+++ auidemo.cpp	(working copy)
@@ -1617,6 +1617,10 @@
                                    wxDefaultPosition,
                                    wxSize(400,300));
     ctrl->SetPage(GetIntroText());
+    wxFont m_font = *wxNORMAL_FONT;
+    ctrl->SetStandardFonts(m_font.GetPointSize(),
+        m_font.GetFaceName(),
+        m_font.GetFaceName());
     return ctrl;
 }
 
@@ -1684,6 +1688,7 @@
 {
     const char* text =
         "<html><body>"
+        "<center><i>some years</i></center>"
         "<h3>Welcome to wxAUI</h3>"
         "<br/><b>Overview</b><br/>"
         "<p>wxAUI is an Advanced User Interface library for the wxWidgets toolkit "
