diff -ru2 -xCVS -x.#* -x*.orig -x*.rej -x*~ wxWindows.orig/include/wx/generic/grid.h wxWindows/include/wx/generic/grid.h
--- wxWindows.orig/include/wx/generic/grid.h	Fri Nov 22 15:03:42 2002
+++ wxWindows/include/wx/generic/grid.h	Mon Nov 25 11:01:21 2002
@@ -358,4 +358,8 @@
     virtual wxGridCellEditor *Clone() const = 0;
 
+    // DJC MAPTEK
+    // added GetValue so we can get the value which is in the control
+    virtual wxString GetValue() const = 0;
+
 protected:
     // the dtor is private because only DecRef() can delete us
@@ -406,4 +410,7 @@
         { return new wxGridCellTextEditor; }
 
+    // DJC MAPTEK
+    // added GetValue so we can get the value which is in the control
+    virtual wxString GetValue() const;
 protected:
     wxTextCtrl *Text() const { return (wxTextCtrl *)m_control; }
@@ -442,4 +449,7 @@
     virtual wxGridCellEditor *Clone() const
         { return new wxGridCellNumberEditor(m_min, m_max); }
+    // DJC MAPTEK
+    // added GetValue so we can get the value which is in the control
+    virtual wxString GetValue() const;
 
 protected:
@@ -517,4 +527,7 @@
     virtual wxGridCellEditor *Clone() const
         { return new wxGridCellBoolEditor; }
+    // DJC MAPTEK
+    // added GetValue so we can get the value which is in the control
+    virtual wxString GetValue() const;
 
 protected:
@@ -553,9 +566,14 @@
 
     virtual wxGridCellEditor *Clone() const;
+    // DJC MAPTEK
+    // added GetValue so we can get the value which is in the control
+    virtual wxString GetValue() const;
 
 protected:
     wxComboBox *Combo() const { return (wxComboBox *)m_control; }
 
-private:
+// DJC - (MAPTEK) you at least need access to m_choices if you
+//                wish to override this class
+protected:
     wxString        m_startValue;
     wxArrayString   m_choices;
@@ -1417,10 +1435,11 @@
                           wxGridCellRenderer* renderer,
                           wxGridCellEditor* editor);
-    wxGridCellEditor* GetDefaultEditorForCell(int row, int col) const;
+    // DJC MAPTEK
+    virtual wxGridCellEditor* GetDefaultEditorForCell(int row, int col) const;
     wxGridCellEditor* GetDefaultEditorForCell(const wxGridCellCoords& c) const
         { return GetDefaultEditorForCell(c.GetRow(), c.GetCol()); }
-    wxGridCellRenderer* GetDefaultRendererForCell(int row, int col) const;
-    wxGridCellEditor* GetDefaultEditorForType(const wxString& typeName) const;
-    wxGridCellRenderer* GetDefaultRendererForType(const wxString& typeName) const;
+    virtual wxGridCellRenderer* GetDefaultRendererForCell(int row, int col) const;
+    virtual wxGridCellEditor* GetDefaultEditorForType(const wxString& typeName) const;
+    virtual wxGridCellRenderer* GetDefaultRendererForType(const wxString& typeName) const;
 
     // grid may occupy more space than needed for its rows/columns, this
diff -ru2 -xCVS -x.#* -x*.orig -x*.rej -x*~ wxWindows.orig/src/generic/grid.cpp wxWindows/src/generic/grid.cpp
--- wxWindows.orig/src/generic/grid.cpp	Fri Nov 22 15:03:54 2002
+++ wxWindows/src/generic/grid.cpp	Mon Nov 25 11:02:50 2002
@@ -754,4 +754,11 @@
 }
 
+// DJC MAPTEK
+// return the value in the text control
+wxString wxGridCellTextEditor::GetValue() const
+{
+  return Text()->GetValue();
+}
+
 // ----------------------------------------------------------------------------
 // wxGridCellNumberEditor
@@ -955,4 +962,22 @@
 }
 
+// DJC MAPTEK
+// return the value in the spin control if it is there (the text control otherwise)
+wxString wxGridCellNumberEditor::GetValue() const
+{
+  wxString s;
+
+  if( HasRange() )
+  {
+    int value = Spin()->GetValue();
+    s.Printf(wxT("%ld"), value);
+  }
+  else
+  {
+    s = Text()->GetValue();
+  }
+  return s;
+}
+
 // ----------------------------------------------------------------------------
 // wxGridCellFloatEditor
@@ -1281,4 +1306,11 @@
     return FALSE;
 }
+// DJC MAPTEK
+// return the value as "1" for true and the empty string for false
+wxString wxGridCellBoolEditor::GetValue() const
+{
+  bool bSet = CBox()->GetValue();
+  return bSet ? "1" : wxEmptyString;
+}
 
 #endif // wxUSE_CHECKBOX
@@ -1406,4 +1438,10 @@
 }
 
+// DJC MAPTEK
+// return the value in the text control
+wxString wxGridCellChoiceEditor::GetValue() const
+{
+  return Combo()->GetValue();
+}
 #endif // wxUSE_COMBOBOX

 	  	 
