Index: wxWindows/include/wx/imaglist.h
===================================================================
RCS file: /export/cvsroot/org/wxwindows/wxWindows/include/wx/imaglist.h,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -b -u -2 -r1.1.1.1 -r1.2
--- wxWindows/include/wx/imaglist.h	2002/02/14 14:23:35	1.1.1.1
+++ wxWindows/include/wx/imaglist.h	2002/07/19 15:53:21	1.2
@@ -4,6 +4,9 @@
 #if defined(__WIN32__)
     #include "wx/msw/imaglist.h"
-#else
-    #include "wx/generic/imaglist.h"
+#endif
+#include "wx/generic/imaglist.h"
+#if !defined(__WIN32__)
+    #define wxImageList wxGenericImageList
+    #define sm_classwxImageList sm_classwxGenericImageList
 #endif
 
Index: wxWindows/include/wx/listctrl.h
===================================================================
RCS file: /export/cvsroot/org/wxwindows/wxWindows/include/wx/listctrl.h,v
retrieving revision 1.1.1.3
retrieving revision 1.2
diff -b -u -2 -r1.1.1.3 -r1.2
--- wxWindows/include/wx/listctrl.h	2002/06/10 17:58:08	1.1.1.3
+++ wxWindows/include/wx/listctrl.h	2002/07/19 15:53:22	1.2
@@ -316,6 +316,9 @@
 #if defined(__WIN32__) && !defined(__WXUNIVERSAL__)
     #include "wx/msw/listctrl.h"
-#else
-    #include "wx/generic/listctrl.h"
+#endif
+#include "wx/generic/listctrl.h"
+#if !defined(__WIN32__) || defined(__WXUNIVERSAL__)
+    #define wxListCtrl wxGenericListCtrl
+    #define sm_classwxListCtrl sm_classwxGenericListCtrl
 #endif
 
Index: wxWindows/include/wx/notebook.h
===================================================================
RCS file: /export/cvsroot/org/wxwindows/wxWindows/include/wx/notebook.h,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -b -u -2 -r1.1.1.1 -r1.2
--- wxWindows/include/wx/notebook.h	2001/08/25 14:52:26	1.1.1.1
+++ wxWindows/include/wx/notebook.h	2002/07/19 15:53:22	1.2
@@ -27,6 +27,5 @@
 #include "wx/control.h"
 #include "wx/dynarray.h"
-
-class WXDLLEXPORT wxImageList;
+#include "wx/imaglist.h"
 
 // ----------------------------------------------------------------------------
Index: wxWindows/include/wx/generic/imaglist.h
===================================================================
RCS file: /export/cvsroot/org/wxwindows/wxWindows/include/wx/generic/imaglist.h,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -b -u -2 -r1.1.1.1 -r1.2
--- wxWindows/include/wx/generic/imaglist.h	2002/01/28 21:29:40	1.1.1.1
+++ wxWindows/include/wx/generic/imaglist.h	2002/07/19 15:53:33	1.2
@@ -35,4 +35,7 @@
  */
 
+#if !defined(wxIMAGELIST_DRAW_NORMAL) || !defined(wxIMAGELIST_DRAW_NORMAL) ||  \
+    !defined(wxIMAGELIST_DRAW_NORMAL) ||  !defined(wxIMAGELIST_DRAW_NORMAL)
+
 // Flags for Draw
 #define wxIMAGELIST_DRAW_NORMAL         0x0001
@@ -47,16 +50,18 @@
     wxIMAGE_LIST_STATE   // State icons: unimplemented (see WIN32 documentation)
 };
+
+#endif
 
-class WXDLLEXPORT wxImageList: public wxObject
+class WXDLLEXPORT wxGenericImageList: public wxObject
 {
 public:
-    wxImageList() { }
-    wxImageList( int width, int height, bool mask = TRUE, int initialCount = 1 );
-    ~wxImageList();
+    wxGenericImageList() { }
+    wxGenericImageList( int width, int height, bool mask = TRUE, int initialCount = 1 );
+    ~wxGenericImageList();
     bool Create( int width, int height, bool mask = TRUE, int initialCount = 1 );
     bool Create();
 
-    int GetImageCount() const;
-    bool GetSize( int index, int &width, int &height ) const;
+    virtual int GetImageCount() const;
+    virtual bool GetSize( int index, int &width, int &height ) const;
 
     int Add( const wxBitmap& bitmap );
@@ -68,5 +73,5 @@
     bool RemoveAll();
 
-    bool Draw(int index, wxDC& dc, int x, int y,
+    virtual bool Draw(int index, wxDC& dc, int x, int y,
               int flags = wxIMAGELIST_DRAW_NORMAL,
               bool solidBackground = FALSE);
@@ -78,5 +83,5 @@
     int     m_height;
 
-    DECLARE_DYNAMIC_CLASS(wxImageList)
+    DECLARE_DYNAMIC_CLASS(wxGenericImageList)
 };
 
Index: wxWindows/include/wx/generic/listctrl.h
===================================================================
RCS file: /export/cvsroot/org/wxwindows/wxWindows/include/wx/generic/listctrl.h,v
retrieving revision 1.1.1.2
retrieving revision 1.2
diff -b -u -2 -r1.1.1.2 -r1.2
--- wxWindows/include/wx/generic/listctrl.h	2002/04/24 20:31:45	1.1.1.2
+++ wxWindows/include/wx/generic/listctrl.h	2002/07/19 15:53:33	1.2
@@ -55,9 +55,9 @@
 //-----------------------------------------------------------------------------
 
-class WXDLLEXPORT wxListCtrl: public wxControl
+class WXDLLEXPORT wxGenericListCtrl: public wxControl
 {
 public:
-    wxListCtrl();
-    wxListCtrl( wxWindow *parent,
+    wxGenericListCtrl();
+    wxGenericListCtrl( wxWindow *parent,
                 wxWindowID id = -1,
                 const wxPoint &pos = wxDefaultPosition,
@@ -69,5 +69,5 @@
         Create(parent, id, pos, size, style, validator, name);
     }
-    ~wxListCtrl();
+    ~wxGenericListCtrl();
 
     bool Create( wxWindow *parent,
@@ -115,7 +115,7 @@
     void RecreateWindow() {}
     long GetNextItem( long item, int geometry = wxLIST_NEXT_ALL, int state = wxLIST_STATE_DONTCARE ) const;
-    wxImageList *GetImageList( int which ) const;
-    void SetImageList( wxImageList *imageList, int which );
-    void AssignImageList( wxImageList *imageList, int which );
+    wxGenericImageList *GetImageList( int which ) const;
+    void SetImageList( wxGenericImageList *imageList, int which );
+    void AssignImageList( wxGenericImageList *imageList, int which );
     bool Arrange( int flag = wxLIST_ALIGN_DEFAULT ); // always wxLIST_ALIGN_LEFT in wxGLC
 
@@ -184,7 +184,7 @@
     // --------------
 
-    wxImageList         *m_imageListNormal;
-    wxImageList         *m_imageListSmall;
-    wxImageList         *m_imageListState;  // what's that ?
+    wxGenericImageList         *m_imageListNormal;
+    wxGenericImageList         *m_imageListSmall;
+    wxGenericImageList         *m_imageListState;  // what's that ?
     bool                 m_ownsImageListNormal,
                          m_ownsImageListSmall,
@@ -218,5 +218,5 @@
 
     DECLARE_EVENT_TABLE()
-    DECLARE_DYNAMIC_CLASS(wxListCtrl);
+    DECLARE_DYNAMIC_CLASS(wxGenericListCtrl);
 };
 
Index: wxWindows/include/wx/generic/treectlg.h
===================================================================
RCS file: /export/cvsroot/org/wxwindows/wxWindows/include/wx/generic/treectlg.h,v
retrieving revision 1.1.1.3
retrieving revision 1.2
diff -b -u -2 -r1.1.1.3 -r1.2
--- wxWindows/include/wx/generic/treectlg.h	2002/07/08 16:38:46	1.1.1.3
+++ wxWindows/include/wx/generic/treectlg.h	2002/07/19 15:53:33	1.2
@@ -21,4 +21,5 @@
 #include "wx/scrolwin.h"
 #include "wx/pen.h"
+#include "wx/imaglist.h"
 
 // -----------------------------------------------------------------------------
@@ -26,5 +27,4 @@
 // -----------------------------------------------------------------------------
 
-class WXDLLEXPORT wxImageList;
 class WXDLLEXPORT wxGenericTreeItem;
 
Index: wxWindows/src/generic/imaglist.cpp
===================================================================
RCS file: /export/cvsroot/org/wxwindows/wxWindows/src/generic/imaglist.cpp,v
retrieving revision 1.1.1.2
retrieving revision 1.2
diff -b -u -2 -r1.1.1.2 -r1.2
--- wxWindows/src/generic/imaglist.cpp	2002/03/24 00:19:45	1.1.1.2
+++ wxWindows/src/generic/imaglist.cpp	2002/07/19 15:53:52	1.2
@@ -27,21 +27,21 @@
 //-----------------------------------------------------------------------------
 
-IMPLEMENT_DYNAMIC_CLASS(wxImageList, wxObject)
+IMPLEMENT_DYNAMIC_CLASS(wxGenericImageList, wxObject)
 
-wxImageList::wxImageList( int width, int height, bool mask, int initialCount )
+wxGenericImageList::wxGenericImageList( int width, int height, bool mask, int initialCount )
 {
     (void)Create(width, height, mask, initialCount);
 }
 
-wxImageList::~wxImageList()
+wxGenericImageList::~wxGenericImageList()
 {
 }
 
-int wxImageList::GetImageCount() const
+int wxGenericImageList::GetImageCount() const
 {
     return m_images.Number();
 }
 
-bool wxImageList::Create( int width, int height, bool WXUNUSED(mask), int WXUNUSED(initialCount) )
+bool wxGenericImageList::Create( int width, int height, bool WXUNUSED(mask), int WXUNUSED(initialCount) )
 {
     m_width = width;
@@ -51,5 +51,5 @@
 }
 
-bool wxImageList::Create()
+bool wxGenericImageList::Create()
 {
     m_images.DeleteContents( TRUE );
@@ -57,5 +57,5 @@
 }
 
-int wxImageList::Add( const wxBitmap &bitmap )
+int wxGenericImageList::Add( const wxBitmap &bitmap )
 {
     if (bitmap.IsKindOf(CLASSINFO(wxIcon)))
@@ -66,5 +66,5 @@
 }
 
-int wxImageList::Add( const wxBitmap& bitmap, const wxBitmap& mask )
+int wxGenericImageList::Add( const wxBitmap& bitmap, const wxBitmap& mask )
 {
     wxBitmap bmp(bitmap);
@@ -74,5 +74,5 @@
 }
 
-int wxImageList::Add( const wxBitmap& bitmap, const wxColour& maskColour )
+int wxGenericImageList::Add( const wxBitmap& bitmap, const wxColour& maskColour )
 {
     wxImage img = bitmap.ConvertToImage();
@@ -81,5 +81,5 @@
 }
 
-const wxBitmap *wxImageList::GetBitmap( int index ) const
+const wxBitmap *wxGenericImageList::GetBitmap( int index ) const
 {
     wxNode *node = m_images.Nth( index );
@@ -90,5 +90,5 @@
 }
 
-bool wxImageList::Replace( int index, const wxBitmap &bitmap )
+bool wxGenericImageList::Replace( int index, const wxBitmap &bitmap )
 {
     wxNode *node = m_images.Nth( index );
@@ -123,5 +123,5 @@
 }
 
-bool wxImageList::Remove( int index )
+bool wxGenericImageList::Remove( int index )
 {
     wxNode *node = m_images.Nth( index );
@@ -134,5 +134,5 @@
 }
 
-bool wxImageList::RemoveAll()
+bool wxGenericImageList::RemoveAll()
 {
     m_images.Clear();
@@ -141,5 +141,5 @@
 }
 
-bool wxImageList::GetSize( int index, int &width, int &height ) const
+bool wxGenericImageList::GetSize( int index, int &width, int &height ) const
 {
     width = 0;
@@ -157,5 +157,5 @@
 }
 
-bool wxImageList::Draw( int index, wxDC &dc, int x, int y,
+bool wxGenericImageList::Draw( int index, wxDC &dc, int x, int y,
                         int flags, bool WXUNUSED(solidBackground) )
 {
Index: wxWindows/src/generic/listctrl.cpp
===================================================================
RCS file: /export/cvsroot/org/wxwindows/wxWindows/src/generic/listctrl.cpp,v
retrieving revision 1.1.1.14
retrieving revision 1.3
diff -b -u -2 -r1.1.1.14 -r1.3
--- wxWindows/src/generic/listctrl.cpp	2002/07/12 16:05:46	1.1.1.14
+++ wxWindows/src/generic/listctrl.cpp	2002/07/19 15:53:52	1.3
@@ -647,5 +647,5 @@
     int GetTextLength( const wxString &s ) const;
 
-    void SetImageList( wxImageList *imageList, int which );
+    void SetImageList( wxGenericImageList *imageList, int which );
     void SetItemSpacing( int spacing, bool isSmall = FALSE );
     int GetItemSpacing( bool isSmall = FALSE );
@@ -734,7 +734,7 @@
 
     // get the backpointer to the list ctrl
-    wxListCtrl *GetListCtrl() const
+    wxGenericListCtrl *GetListCtrl() const
     {
-        return wxStaticCast(GetParent(), wxListCtrl);
+        return wxStaticCast(GetParent(), wxGenericListCtrl);
     }
 
@@ -776,6 +776,6 @@
     int                  m_xScroll,
                          m_yScroll;
-    wxImageList         *m_small_image_list;
-    wxImageList         *m_normal_image_list;
+    wxGenericImageList         *m_small_image_list;
+    wxGenericImageList         *m_normal_image_list;
     int                  m_small_spacing;
     int                  m_normal_spacing;
@@ -1923,5 +1923,5 @@
         if ( image != -1 )
         {
-            wxImageList *imageList = m_owner->m_small_image_list;
+            wxGenericImageList *imageList = m_owner->m_small_image_list;
             if ( imageList )
             {
@@ -2281,6 +2281,6 @@
     m_lineHeight = 0;
 
-    m_small_image_list = (wxImageList *) NULL;
-    m_normal_image_list = (wxImageList *) NULL;
+    m_small_image_list = (wxGenericImageList *) NULL;
+    m_normal_image_list = (wxGenericImageList *) NULL;
 
     m_small_spacing = 30;
@@ -2376,5 +2376,5 @@
 void wxListMainWindow::CacheLineData(size_t line)
 {
-    wxListCtrl *listctrl = GetListCtrl();
+    wxGenericListCtrl *listctrl = GetListCtrl();
 
     wxListLineData *ld = GetDummyLine();
@@ -2934,5 +2934,5 @@
 {
     wxCHECK_RET( (item >= 0) && ((size_t)item < GetItemCount()),
-                 wxT("wrong index in wxListCtrl::EditLabel()") );
+                 wxT("wrong index in wxGenericListCtrl::EditLabel()") );
 
     size_t itemEdit = (size_t)item;
@@ -3530,5 +3530,5 @@
 }
 
-void wxListMainWindow::SetImageList( wxImageList *imageList, int which )
+void wxListMainWindow::SetImageList( wxGenericImageList *imageList, int which )
 {
     m_dirty = TRUE;
@@ -4440,5 +4440,5 @@
     if ( event.GetOrientation() == wxHORIZONTAL && HasHeader() )
     {
-        wxListCtrl* lc = GetListCtrl();
+        wxGenericListCtrl* lc = GetListCtrl();
         wxCHECK_RET( lc, _T("no listctrl window?") );
 
@@ -4501,25 +4501,30 @@
 // -------------------------------------------------------------------------------------
 
+#if !defined(__WIN32__)
 IMPLEMENT_DYNAMIC_CLASS(wxListItem, wxObject)
+#endif
 
 // -------------------------------------------------------------------------------------
-// wxListCtrl
+// wxGenericListCtrl
 // -------------------------------------------------------------------------------------
+
+IMPLEMENT_DYNAMIC_CLASS(wxGenericListCtrl, wxControl)
 
-IMPLEMENT_DYNAMIC_CLASS(wxListCtrl, wxControl)
-IMPLEMENT_DYNAMIC_CLASS(wxListView, wxListCtrl)
+#if !defined(__WIN32__)
+IMPLEMENT_DYNAMIC_CLASS(wxListView, wxGenericListCtrl)
 
 IMPLEMENT_DYNAMIC_CLASS(wxListEvent, wxNotifyEvent)
+#endif
 
-BEGIN_EVENT_TABLE(wxListCtrl,wxControl)
-  EVT_SIZE(wxListCtrl::OnSize)
-  EVT_IDLE(wxListCtrl::OnIdle)
+BEGIN_EVENT_TABLE(wxGenericListCtrl,wxControl)
+  EVT_SIZE(wxGenericListCtrl::OnSize)
+  EVT_IDLE(wxGenericListCtrl::OnIdle)
 END_EVENT_TABLE()
 
-wxListCtrl::wxListCtrl()
+wxGenericListCtrl::wxGenericListCtrl()
 {
-    m_imageListNormal = (wxImageList *) NULL;
-    m_imageListSmall = (wxImageList *) NULL;
-    m_imageListState = (wxImageList *) NULL;
+    m_imageListNormal = (wxGenericImageList *) NULL;
+    m_imageListSmall = (wxGenericImageList *) NULL;
+    m_imageListState = (wxGenericImageList *) NULL;
 
     m_ownsImageListNormal =
@@ -4531,5 +4536,5 @@
 }
 
-wxListCtrl::~wxListCtrl()
+wxGenericListCtrl::~wxGenericListCtrl()
 {
     if (m_ownsImageListNormal)
@@ -4541,5 +4546,5 @@
 }
 
-void wxListCtrl::CreateHeaderWindow()
+void wxGenericListCtrl::CreateHeaderWindow()
 {
     m_headerWin = new wxListHeaderWindow
@@ -4552,5 +4557,5 @@
 }
 
-bool wxListCtrl::Create(wxWindow *parent,
+bool wxGenericListCtrl::Create(wxWindow *parent,
                         wxWindowID id,
                         const wxPoint &pos,
@@ -4562,5 +4567,5 @@
     m_imageListNormal =
     m_imageListSmall =
-    m_imageListState = (wxImageList *) NULL;
+    m_imageListState = (wxGenericImageList *) NULL;
     m_ownsImageListNormal =
     m_ownsImageListSmall =
@@ -4597,5 +4602,5 @@
 }
 
-void wxListCtrl::SetSingleStyle( long style, bool add )
+void wxGenericListCtrl::SetSingleStyle( long style, bool add )
 {
     wxASSERT_MSG( !(style & wxLC_VIRTUAL),
@@ -4626,5 +4631,5 @@
 }
 
-void wxListCtrl::SetWindowStyleFlag( long flag )
+void wxGenericListCtrl::SetWindowStyleFlag( long flag )
 {
     if (m_mainWin)
@@ -4667,5 +4672,5 @@
 }
 
-bool wxListCtrl::GetColumn(int col, wxListItem &item) const
+bool wxGenericListCtrl::GetColumn(int col, wxListItem &item) const
 {
     m_mainWin->GetColumn( col, item );
@@ -4673,5 +4678,5 @@
 }
 
-bool wxListCtrl::SetColumn( int col, wxListItem& item )
+bool wxGenericListCtrl::SetColumn( int col, wxListItem& item )
 {
     m_mainWin->SetColumn( col, item );
@@ -4679,10 +4684,10 @@
 }
 
-int wxListCtrl::GetColumnWidth( int col ) const
+int wxGenericListCtrl::GetColumnWidth( int col ) const
 {
     return m_mainWin->GetColumnWidth( col );
 }
 
-bool wxListCtrl::SetColumnWidth( int col, int width )
+bool wxGenericListCtrl::SetColumnWidth( int col, int width )
 {
     m_mainWin->SetColumnWidth( col, width );
@@ -4690,10 +4695,10 @@
 }
 
-int wxListCtrl::GetCountPerPage() const
+int wxGenericListCtrl::GetCountPerPage() const
 {
   return m_mainWin->GetCountPerPage();  // different from Windows ?
 }
 
-bool wxListCtrl::GetItem( wxListItem &info ) const
+bool wxGenericListCtrl::GetItem( wxListItem &info ) const
 {
     m_mainWin->GetItem( info );
@@ -4701,5 +4706,5 @@
 }
 
-bool wxListCtrl::SetItem( wxListItem &info )
+bool wxGenericListCtrl::SetItem( wxListItem &info )
 {
     m_mainWin->SetItem( info );
@@ -4707,5 +4712,5 @@
 }
 
-long wxListCtrl::SetItem( long index, int col, const wxString& label, int imageId )
+long wxGenericListCtrl::SetItem( long index, int col, const wxString& label, int imageId )
 {
     wxListItem info;
@@ -4723,10 +4728,10 @@
 }
 
-int wxListCtrl::GetItemState( long item, long stateMask ) const
+int wxGenericListCtrl::GetItemState( long item, long stateMask ) const
 {
     return m_mainWin->GetItemState( item, stateMask );
 }
 
-bool wxListCtrl::SetItemState( long item, long state, long stateMask )
+bool wxGenericListCtrl::SetItemState( long item, long state, long stateMask )
 {
     m_mainWin->SetItemState( item, state, stateMask );
@@ -4734,5 +4739,5 @@
 }
 
-bool wxListCtrl::SetItemImage( long item, int image, int WXUNUSED(selImage) )
+bool wxGenericListCtrl::SetItemImage( long item, int image, int WXUNUSED(selImage) )
 {
     wxListItem info;
@@ -4744,15 +4749,15 @@
 }
 
-wxString wxListCtrl::GetItemText( long item ) const
+wxString wxGenericListCtrl::GetItemText( long item ) const
 {
     return m_mainWin->GetItemText(item);
 }
 
-void wxListCtrl::SetItemText( long item, const wxString& str )
+void wxGenericListCtrl::SetItemText( long item, const wxString& str )
 {
     m_mainWin->SetItemText(item, str);
 }
 
-long wxListCtrl::GetItemData( long item ) const
+long wxGenericListCtrl::GetItemData( long item ) const
 {
     wxListItem info;
@@ -4762,5 +4767,5 @@
 }
 
-bool wxListCtrl::SetItemData( long item, long data )
+bool wxGenericListCtrl::SetItemData( long item, long data )
 {
     wxListItem info;
@@ -4772,5 +4777,5 @@
 }
 
-bool wxListCtrl::GetItemRect( long item, wxRect &rect,  int WXUNUSED(code) ) const
+bool wxGenericListCtrl::GetItemRect( long item, wxRect &rect,  int WXUNUSED(code) ) const
 {
     m_mainWin->GetItemRect( item, rect );
@@ -4778,5 +4783,5 @@
 }
 
-bool wxListCtrl::GetItemPosition( long item, wxPoint& pos ) const
+bool wxGenericListCtrl::GetItemPosition( long item, wxPoint& pos ) const
 {
     m_mainWin->GetItemPosition( item, pos );
@@ -4784,30 +4789,30 @@
 }
 
-bool wxListCtrl::SetItemPosition( long WXUNUSED(item), const wxPoint& WXUNUSED(pos) )
+bool wxGenericListCtrl::SetItemPosition( long WXUNUSED(item), const wxPoint& WXUNUSED(pos) )
 {
     return 0;
 }
 
-int wxListCtrl::GetItemCount() const
+int wxGenericListCtrl::GetItemCount() const
 {
     return m_mainWin->GetItemCount();
 }
 
-int wxListCtrl::GetColumnCount() const
+int wxGenericListCtrl::GetColumnCount() const
 {
     return m_mainWin->GetColumnCount();
 }
 
-void wxListCtrl::SetItemSpacing( int spacing, bool isSmall )
+void wxGenericListCtrl::SetItemSpacing( int spacing, bool isSmall )
 {
     m_mainWin->SetItemSpacing( spacing, isSmall );
 }
 
-int wxListCtrl::GetItemSpacing( bool isSmall ) const
+int wxGenericListCtrl::GetItemSpacing( bool isSmall ) const
 {
     return m_mainWin->GetItemSpacing( isSmall );
 }
 
-void wxListCtrl::SetItemTextColour( long item, const wxColour &col )
+void wxGenericListCtrl::SetItemTextColour( long item, const wxColour &col )
 {
     wxListItem info;
@@ -4817,5 +4822,5 @@
 }
 
-wxColour wxListCtrl::GetItemTextColour( long item ) const
+wxColour wxGenericListCtrl::GetItemTextColour( long item ) const
 {
     wxListItem info;
@@ -4825,5 +4830,5 @@
 }
 
-void wxListCtrl::SetItemBackgroundColour( long item, const wxColour &col )
+void wxGenericListCtrl::SetItemBackgroundColour( long item, const wxColour &col )
 {
     wxListItem info;
@@ -4833,5 +4838,5 @@
 }
 
-wxColour wxListCtrl::GetItemBackgroundColour( long item ) const
+wxColour wxGenericListCtrl::GetItemBackgroundColour( long item ) const
 {
     wxListItem info;
@@ -4841,30 +4846,30 @@
 }
 
-int wxListCtrl::GetSelectedItemCount() const
+int wxGenericListCtrl::GetSelectedItemCount() const
 {
     return m_mainWin->GetSelectedItemCount();
 }
 
-wxColour wxListCtrl::GetTextColour() const
+wxColour wxGenericListCtrl::GetTextColour() const
 {
     return GetForegroundColour();
 }
 
-void wxListCtrl::SetTextColour(const wxColour& col)
+void wxGenericListCtrl::SetTextColour(const wxColour& col)
 {
     SetForegroundColour(col);
 }
 
-long wxListCtrl::GetTopItem() const
+long wxGenericListCtrl::GetTopItem() const
 {
     return 0;
 }
 
-long wxListCtrl::GetNextItem( long item, int geom, int state ) const
+long wxGenericListCtrl::GetNextItem( long item, int geom, int state ) const
 {
     return m_mainWin->GetNextItem( item, geom, state );
 }
 
-wxImageList *wxListCtrl::GetImageList(int which) const
+wxGenericImageList *wxGenericListCtrl::GetImageList(int which) const
 {
     if (which == wxIMAGE_LIST_NORMAL)
@@ -4880,8 +4885,8 @@
         return m_imageListState;
     }
-    return (wxImageList *) NULL;
+    return (wxGenericImageList *) NULL;
 }
 
-void wxListCtrl::SetImageList( wxImageList *imageList, int which )
+void wxGenericListCtrl::SetImageList( wxGenericImageList *imageList, int which )
 {
     if ( which == wxIMAGE_LIST_NORMAL )
@@ -4907,5 +4912,5 @@
 }
 
-void wxListCtrl::AssignImageList(wxImageList *imageList, int which)
+void wxGenericListCtrl::AssignImageList(wxGenericImageList *imageList, int which)
 {
     SetImageList(imageList, which);
@@ -4918,10 +4923,10 @@
 }
 
-bool wxListCtrl::Arrange( int WXUNUSED(flag) )
+bool wxGenericListCtrl::Arrange( int WXUNUSED(flag) )
 {
     return 0;
 }
 
-bool wxListCtrl::DeleteItem( long item )
+bool wxGenericListCtrl::DeleteItem( long item )
 {
     m_mainWin->DeleteItem( item );
@@ -4929,5 +4934,5 @@
 }
 
-bool wxListCtrl::DeleteAllItems()
+bool wxGenericListCtrl::DeleteAllItems()
 {
     m_mainWin->DeleteAllItems();
@@ -4935,5 +4940,5 @@
 }
 
-bool wxListCtrl::DeleteAllColumns()
+bool wxGenericListCtrl::DeleteAllColumns()
 {
     size_t count = m_mainWin->m_columns.GetCount();
@@ -4944,10 +4949,10 @@
 }
 
-void wxListCtrl::ClearAll()
+void wxGenericListCtrl::ClearAll()
 {
     m_mainWin->DeleteEverything();
 }
 
-bool wxListCtrl::DeleteColumn( int col )
+bool wxGenericListCtrl::DeleteColumn( int col )
 {
     m_mainWin->DeleteColumn( col );
@@ -4962,10 +4967,10 @@
 }
 
-void wxListCtrl::Edit( long item )
+void wxGenericListCtrl::Edit( long item )
 {
     m_mainWin->EditLabel( item );
 }
 
-bool wxListCtrl::EnsureVisible( long item )
+bool wxGenericListCtrl::EnsureVisible( long item )
 {
     m_mainWin->EnsureVisible( item );
@@ -4973,15 +4978,15 @@
 }
 
-long wxListCtrl::FindItem( long start, const wxString& str,  bool partial )
+long wxGenericListCtrl::FindItem( long start, const wxString& str,  bool partial )
 {
     return m_mainWin->FindItem( start, str, partial );
 }
 
-long wxListCtrl::FindItem( long start, long data )
+long wxGenericListCtrl::FindItem( long start, long data )
 {
     return m_mainWin->FindItem( start, data );
 }
 
-long wxListCtrl::FindItem( long WXUNUSED(start), const wxPoint& WXUNUSED(pt),
+long wxGenericListCtrl::FindItem( long WXUNUSED(start), const wxPoint& WXUNUSED(pt),
                            int WXUNUSED(direction))
 {
@@ -4989,10 +4994,10 @@
 }
 
-long wxListCtrl::HitTest( const wxPoint &point, int &flags )
+long wxGenericListCtrl::HitTest( const wxPoint &point, int &flags )
 {
     return m_mainWin->HitTest( (int)point.x, (int)point.y, flags );
 }
 
-long wxListCtrl::InsertItem( wxListItem& info )
+long wxGenericListCtrl::InsertItem( wxListItem& info )
 {
     m_mainWin->InsertItem( info );
@@ -5000,5 +5005,5 @@
 }
 
-long wxListCtrl::InsertItem( long index, const wxString &label )
+long wxGenericListCtrl::InsertItem( long index, const wxString &label )
 {
     wxListItem info;
@@ -5009,5 +5014,5 @@
 }
 
-long wxListCtrl::InsertItem( long index, int imageIndex )
+long wxGenericListCtrl::InsertItem( long index, int imageIndex )
 {
     wxListItem info;
@@ -5018,5 +5023,5 @@
 }
 
-long wxListCtrl::InsertItem( long index, const wxString &label, int imageIndex )
+long wxGenericListCtrl::InsertItem( long index, const wxString &label, int imageIndex )
 {
     wxListItem info;
@@ -5028,5 +5033,5 @@
 }
 
-long wxListCtrl::InsertColumn( long col, wxListItem &item )
+long wxGenericListCtrl::InsertColumn( long col, wxListItem &item )
 {
     wxCHECK_MSG( m_headerWin, -1, _T("can't add column in non report mode") );
@@ -5046,5 +5051,5 @@
 }
 
-long wxListCtrl::InsertColumn( long col, const wxString &heading,
+long wxGenericListCtrl::InsertColumn( long col, const wxString &heading,
                                int format, int width )
 {
@@ -5062,5 +5067,5 @@
 }
 
-bool wxListCtrl::ScrollList( int WXUNUSED(dx), int WXUNUSED(dy) )
+bool wxGenericListCtrl::ScrollList( int WXUNUSED(dx), int WXUNUSED(dy) )
 {
     return 0;
@@ -5077,5 +5082,5 @@
 // data is arbitrary data to be passed to the sort function.
 
-bool wxListCtrl::SortItems( wxListCtrlCompare fn, long data )
+bool wxGenericListCtrl::SortItems( wxListCtrlCompare fn, long data )
 {
     m_mainWin->SortItems( fn, data );
@@ -5087,5 +5092,5 @@
 // ----------------------------------------------------------------------------
 
-void wxListCtrl::OnSize(wxSizeEvent& WXUNUSED(event))
+void wxGenericListCtrl::OnSize(wxSizeEvent& WXUNUSED(event))
 {
     if ( !m_mainWin )
@@ -5097,5 +5102,5 @@
 }
 
-void wxListCtrl::ResizeReportView(bool showHeader)
+void wxGenericListCtrl::ResizeReportView(bool showHeader)
 {
     int cw, ch;
@@ -5113,5 +5118,5 @@
 }
 
-void wxListCtrl::OnIdle( wxIdleEvent & event )
+void wxGenericListCtrl::OnIdle( wxIdleEvent & event )
 {
     event.Skip();
@@ -5128,5 +5133,5 @@
 // ----------------------------------------------------------------------------
 
-bool wxListCtrl::SetBackgroundColour( const wxColour &colour )
+bool wxGenericListCtrl::SetBackgroundColour( const wxColour &colour )
 {
     if (m_mainWin)
@@ -5139,5 +5144,5 @@
 }
 
-bool wxListCtrl::SetForegroundColour( const wxColour &colour )
+bool wxGenericListCtrl::SetForegroundColour( const wxColour &colour )
 {
     if ( !wxWindow::SetForegroundColour( colour ) )
@@ -5158,5 +5163,5 @@
 }
 
-bool wxListCtrl::SetFont( const wxFont &font )
+bool wxGenericListCtrl::SetFont( const wxFont &font )
 {
     if ( !wxWindow::SetFont( font ) )
@@ -5183,10 +5188,10 @@
 #if wxUSE_DRAG_AND_DROP
 
-void wxListCtrl::SetDropTarget( wxDropTarget *dropTarget )
+void wxGenericListCtrl::SetDropTarget( wxDropTarget *dropTarget )
 {
     m_mainWin->SetDropTarget( dropTarget );
 }
 
-wxDropTarget *wxListCtrl::GetDropTarget() const
+wxDropTarget *wxGenericListCtrl::GetDropTarget() const
 {
     return m_mainWin->GetDropTarget();
@@ -5195,20 +5200,20 @@
 #endif // wxUSE_DRAG_AND_DROP
 
-bool wxListCtrl::SetCursor( const wxCursor &cursor )
+bool wxGenericListCtrl::SetCursor( const wxCursor &cursor )
 {
     return m_mainWin ? m_mainWin->wxWindow::SetCursor(cursor) : FALSE;
 }
 
-wxColour wxListCtrl::GetBackgroundColour() const
+wxColour wxGenericListCtrl::GetBackgroundColour() const
 {
     return m_mainWin ? m_mainWin->GetBackgroundColour() : wxColour();
 }
 
-wxColour wxListCtrl::GetForegroundColour() const
+wxColour wxGenericListCtrl::GetForegroundColour() const
 {
     return m_mainWin ? m_mainWin->GetForegroundColour() : wxColour();
 }
 
-bool wxListCtrl::DoPopupMenu( wxMenu *menu, int x, int y )
+bool wxGenericListCtrl::DoPopupMenu( wxMenu *menu, int x, int y )
 {
 #if wxUSE_MENUS
@@ -5219,5 +5224,5 @@
 }
 
-void wxListCtrl::SetFocus()
+void wxGenericListCtrl::SetFocus()
 {
     /* The test in window.cpp fails as we are a composite
@@ -5231,22 +5236,22 @@
 // ----------------------------------------------------------------------------
 
-wxString wxListCtrl::OnGetItemText(long WXUNUSED(item), long WXUNUSED(col)) const
+wxString wxGenericListCtrl::OnGetItemText(long WXUNUSED(item), long WXUNUSED(col)) const
 {
     // this is a pure virtual function, in fact - which is not really pure
     // because the controls which are not virtual don't need to implement it
-    wxFAIL_MSG( _T("wxListCtrl::OnGetItemText not supposed to be called") );
+    wxFAIL_MSG( _T("wxGenericListCtrl::OnGetItemText not supposed to be called") );
 
     return wxEmptyString;
 }
 
-int wxListCtrl::OnGetItemImage(long WXUNUSED(item)) const
+int wxGenericListCtrl::OnGetItemImage(long WXUNUSED(item)) const
 {
     // same as above
-    wxFAIL_MSG( _T("wxListCtrl::OnGetItemImage not supposed to be called") );
+    wxFAIL_MSG( _T("wxGenericListCtrl::OnGetItemImage not supposed to be called") );
 
     return -1;
 }
 
-wxListItemAttr *wxListCtrl::OnGetItemAttr(long item) const
+wxListItemAttr *wxGenericListCtrl::OnGetItemAttr(long item) const
 {
     wxASSERT_MSG( item >= 0 && item < GetItemCount(),
@@ -5257,5 +5262,5 @@
 }
 
-void wxListCtrl::SetItemCount(long count)
+void wxGenericListCtrl::SetItemCount(long count)
 {
     wxASSERT_MSG( IsVirtual(), _T("this is for virtual controls only") );
@@ -5264,20 +5269,20 @@
 }
 
-void wxListCtrl::RefreshItem(long item)
+void wxGenericListCtrl::RefreshItem(long item)
 {
     m_mainWin->RefreshLine(item);
 }
 
-void wxListCtrl::RefreshItems(long itemFrom, long itemTo)
+void wxGenericListCtrl::RefreshItems(long itemFrom, long itemTo)
 {
     m_mainWin->RefreshLines(itemFrom, itemTo);
 }
 
-void wxListCtrl::Freeze()
+void wxGenericListCtrl::Freeze()
 {
     m_mainWin->Freeze();
 }
 
-void wxListCtrl::Thaw()
+void wxGenericListCtrl::Thaw()
 {
     m_mainWin->Thaw();

 	  	 
