Index: src/msw/spinctrl.cpp
===================================================================
--- src/msw/spinctrl.cpp	(revision 1135)
+++ src/msw/spinctrl.cpp	(working copy)
@@ -525,6 +525,52 @@
     return true;
 }
 
+bool wxSpinCtrl::Reparent(wxWindowBase *newParent)
+{
+    long style = GetWindowStyle(), id = GetId();
+    wxPoint btnPos;
+    wxSize btnSize, size = GetSize();
+    wxString name = GetName();
+    int value = GetValue();
+    RECT rect;
+    
+    /* reparenting both the updown control and its buddy does not seem to work: they continue to
+     * be connected somehow, but visually there is no feedback on the buddy edit control. To avoid
+     * this problem, we normally reparent the buddy window, but we recreate the updown control and
+     * reassign its buddy. This solves the problem.
+     */
+    
+    if ( !wxWindowBase::Reparent(newParent) )
+        return false;
+    for ( wxWindowListNode *node = newParent->GetChildren().GetFirst(); node; node = node->GetNext() ) {
+        if ( (wxWindow *)node->GetData() == this ) {
+            newParent->GetChildren().DeleteNode(node);
+            break;
+        }
+    }
+    
+    ::GetWindowRect(GetHwnd(), &rect);
+    btnPos = wxPoint(rect.left, rect.top);
+    btnSize = wxSize(rect.right - rect.left, rect.bottom - rect.top);
+    
+    UnsubclassWin();
+    if ( !::DestroyWindow(GetHwnd()) )
+        wxLogLastError(wxT("DestroyWindow"));
+    
+    if ( !wxSpinButton::Create(GetParent(), id, btnPos, btnSize, style, name) )
+        return false;
+
+    SetValue(value);
+    SetRange(m_min, m_max);
+    SetInitialSize(size);
+   
+    HWND hWndParent = GetParent() ? GetWinHwnd(GetParent()) : (HWND)0;
+    ::SetParent(GetBuddyHwnd(), hWndParent);
+    (void)::SendMessage(GetHwnd(), UDM_SETBUDDY, (WPARAM)GetBuddyHwnd(), 0);
+    
+    return true;
+}
+
 bool wxSpinCtrl::Enable(bool enable)
 {
     if ( !wxControl::Enable(enable) )
Index: src/generic/spinctlg.cpp
===================================================================
--- src/generic/spinctlg.cpp	(revision 1135)
+++ src/generic/spinctlg.cpp	(working copy)
@@ -265,6 +265,17 @@
     return true;
 }
 
+bool wxSpinCtrl::Reparent(wxWindow *newParent)
+{
+    if ( m_btn )
+    {
+        m_btn->Reparent(newParent);
+        m_text->Reparent(newParent);
+    }
+    
+    return true;
+}
+
 // ----------------------------------------------------------------------------
 // value and range access
 // ----------------------------------------------------------------------------
Index: include/wx/msw/spinctrl.h
===================================================================
--- include/wx/msw/spinctrl.h	(revision 1135)
+++ include/wx/msw/spinctrl.h	(working copy)
@@ -72,6 +72,8 @@
 
     virtual bool Enable(bool enable = true);
     virtual bool Show(bool show = true);
+    
+    virtual bool Reparent(wxWindowBase *newParent);
 
     // wxSpinButton doesn't accept focus, but we do
     virtual bool AcceptsFocus() const { return wxWindow::AcceptsFocus(); }
Index: include/wx/generic/spinctlg.h
===================================================================
--- include/wx/generic/spinctlg.h	(revision 1135)
+++ include/wx/generic/spinctlg.h	(working copy)
@@ -74,6 +74,7 @@
     // forward these functions to all subcontrols
     virtual bool Enable(bool enable = true);
     virtual bool Show(bool show = true);
+    virtual bool Reparent(wxWindow *newParent);
 
     // get the subcontrols
     wxTextCtrl *GetText() const { return m_text; }

 	  	 
