--- src\common\dcgraph.cpp	2014-10-06 17:34:04.000000000 -0400
+++ src\common\dcgraph.cpp	2015-02-03 10:59:47.207019800 -0500
@@ -664,7 +664,7 @@
     }
     else
     {
-        wxGraphicsPath path = m_graphicContext->CreatePath();
+        path = m_graphicContext->CreatePath();
         path.AddArc( 0, 0, h/2.0 , DegToRad(-sa) , DegToRad(-ea), sa > ea );
         m_graphicContext->DrawPath( path );
     }
