Index: samples/minimal/minimal.cpp
===================================================================
--- samples/minimal/minimal.cpp	(revision 54857)
+++ samples/minimal/minimal.cpp	(working copy)
@@ -129,7 +129,8 @@
 
     // and show it (the frames, unlike simple controls, are not shown when
     // created initially)
-    frame->Show(true);
+	//frame->Show();
+    frame->ShowWithEffect(wxSHOW_EFFECT_BLEND);
 
     // success: wxApp::OnRun() will be called which will enter the main message
     // loop and the application will run. If we returned false here, the
@@ -172,6 +173,9 @@
     CreateStatusBar(2);
     SetStatusText("Welcome to wxWidgets!");
 #endif // wxUSE_STATUSBAR
+
+	wxPanel* panel = new wxPanel(this, wxID_ANY, wxDefaultPosition, wxDefaultSize, wxBORDER_SUNKEN);
+	panel->SetBackgroundColour(wxColour( 255, 0, 0 ));  // only for contrast reasons, does not affect the issue at hand
 }
 
 
Index: samples/toolbar/toolbar.cpp
===================================================================
--- samples/toolbar/toolbar.cpp	(revision 54857)
+++ samples/toolbar/toolbar.cpp	(working copy)
@@ -307,7 +307,8 @@
                                  _T("wxToolBar Sample"),
                                   wxPoint(100, 100), wxSize(550, 300));
 
-    frame->Show(true);
+	//frame->Show();
+    frame->ShowWithEffect(wxSHOW_EFFECT_BLEND);
 
 #if wxUSE_STATUSBAR
     frame->SetStatusText(_T("Hello, wxWidgets"));
