--- ../wxWidgets/wx-config.in	Wed Oct 20 13:13:04 2004
+++ ./wx-config.in	Wed Oct 27 15:34:08 2004
@@ -17,6 +17,21 @@
 # Licence:     wxWindows licence
 ############################################################################
 
+# Bourne shell will not run this script.
+# If this is a Bourne shell, try to re-run with bash or ksh.
+if [ ~ = '~' ]
+then
+    if (bash -c echo) >/dev/null 2>&1
+    then
+        exec bash "$0" "$@"
+    fi
+    if (ksh -c echo) >/dev/null 2>&1
+    then
+        exec ksh "$0" "$@"
+    fi
+    echo "$0: this script requires bash or ksh"
+    exit 1
+fi
 
 # Start with some basic stuff, like the ability to die gracefully,
 # and to tell people what we are about.

 	  	 
