diff -uNr wxWidgets-orig/src/msw/dcclient.cpp wxWidgets-mine/src/msw/dcclient.cpp
--- wxWidgets-orig/src/msw/dcclient.cpp	2019-10-27 18:41:42.000000000 +0100
+++ wxWidgets-mine/src/msw/dcclient.cpp	2019-11-10 17:11:12.663059900 +0100
@@ -33,6 +33,8 @@
     #include "wx/window.h"
 #endif
 
+#include "wx/stack.h"
+
 #include "wx/msw/private.h"
 
 // ----------------------------------------------------------------------------
@@ -277,9 +279,9 @@
 #endif // wxHAS_PAINT_DEBUG
 
     // see comments in src/msw/window.cpp where this is defined
-    extern bool wxDidCreatePaintDC;
+    extern wxStack<bool> wxDidCreatePaintDC;
 
-    wxDidCreatePaintDC = true;
+    wxDidCreatePaintDC.top() = true;
 
 
     m_window = window;
diff -uNr wxWidgets-orig/src/msw/window.cpp wxWidgets-mine/src/msw/window.cpp
--- wxWidgets-orig/src/msw/window.cpp	2019-10-27 18:41:42.000000000 +0100
+++ wxWidgets-mine/src/msw/window.cpp	2019-11-10 17:12:40.648016200 +0100
@@ -63,6 +63,7 @@
 #include "wx/popupwin.h"
 #include "wx/power.h"
 #include "wx/scopeguard.h"
+#include "wx/stack.h"
 #include "wx/sysopt.h"
 
 #if wxUSE_DRAG_AND_DROP
@@ -5230,8 +5231,8 @@
 // endless stream of WM_PAINT messages for this window resulting in a lot of
 // difficult to debug problems (e.g. impossibility to repaint other windows,
 // lack of timer and idle events and so on)
-extern bool wxDidCreatePaintDC;
-bool wxDidCreatePaintDC = false;
+extern wxStack<bool> wxDidCreatePaintDC;
+wxStack<bool> wxDidCreatePaintDC;
 
 bool wxWindowMSW::HandlePaint()
 {
@@ -5247,14 +5248,14 @@
 
     m_updateRegion = wxRegion((WXHRGN) hRegion);
 
-    wxDidCreatePaintDC = false;
+    wxDidCreatePaintDC.push(false);
 
     wxPaintEvent event(m_windowId);
     event.SetEventObject(this);
 
     bool processed = HandleWindowEvent(event);
 
-    if ( wxDidCreatePaintDC && !processed )
+    if ( wxDidCreatePaintDC.top() && !processed )
     {
         // Event handler did paint something as wxPaintDC object was created
         // but then it must have skipped the event to indicate that default
@@ -5285,7 +5286,9 @@
     // was processed, we must still call MSWDefWindowProc() to ensure that the
     // window is validated, i.e. to avoid the problem described in the comment
     // before wxDidCreatePaintDC definition above.
-    return wxDidCreatePaintDC;
+    bool ret = wxDidCreatePaintDC.top();    
+    wxDidCreatePaintDC.pop();
+    return ret;  
 }
 
 // Can be called from an application's OnPaint handler
