Index: stctest.cpp
===================================================================
--- stctest.cpp	(revision 74946)
+++ stctest.cpp	(working copy)
@@ -126,6 +126,7 @@
 
     //! event handlers
     //! common
+	void OnText (wxCommandEvent &event);
     void OnClose (wxCloseEvent &event);
     void OnAbout (wxCommandEvent &event);
     void OnExit (wxCommandEvent &event);
@@ -150,6 +151,8 @@
 private:
     // edit object
     Edit *m_edit;
+	wxTextCtrl *m_text;
+	wxBoxSizer *m_boxSizer;
     void FileOpen (wxString fname);
 
     //! creates the application menu bar
@@ -271,6 +274,7 @@
     EVT_MENU (wxID_REDO,             AppFrame::OnEdit)
     EVT_MENU (wxID_UNDO,             AppFrame::OnEdit)
     EVT_MENU (wxID_FIND,             AppFrame::OnEdit)
+	EVT_TEXT (wxID_ANY,				 AppFrame::OnText)
     // And all our edit-related menu commands.
     EVT_MENU_RANGE (myID_EDIT_FIRST, myID_EDIT_LAST,
                                      AppFrame::OnEdit)
@@ -299,6 +303,14 @@
     m_edit = new Edit (this, wxID_ANY);
     m_edit->SetFocus();
 
+	m_text = new wxTextCtrl(this, wxID_ANY);
+
+	m_boxSizer = new wxBoxSizer(wxVERTICAL);
+	m_boxSizer->Add(m_text, 0, wxEXPAND, 0);
+	m_boxSizer->Add(m_edit, 1, wxEXPAND, 0);
+
+	this->SetSizer(m_boxSizer);
+
     FileOpen (wxT("stctest.cpp"));
 }
 
@@ -306,6 +318,15 @@
 }
 
 // common event handlers
+void AppFrame::OnText(wxCommandEvent& event)
+{
+	//once m_edit is destroyed, TextCtrl's caret disappears (but you can continue typing into it normally) 
+	if (this->m_edit->IsShown()) {
+		this->m_edit->Destroy();
+	}
+	return;
+}
+
 void AppFrame::OnClose (wxCloseEvent &event) {
     wxCommandEvent evt;
     OnFileClose (evt);
