Index: src/common/event.cpp
===================================================================
--- src/common/event.cpp	(revision 67241)
+++ src/common/event.cpp	(working copy)
@@ -861,6 +861,21 @@
 {
     AllocEventTypeTable(EVENT_TYPE_TABLE_INIT_SIZE);
 
+#if defined(__WXDEBUG__)
+    // Gal Shalif: avoid a cycle that can be created by a bad automatic GUI builders - it will cause an indefinite loop at the initialization time. Verify that "this" is not already within the linked-list
+    for (wxEventHashTable *i = sm_first; 0 != i; i = i->m_next)
+    {
+        if (i == this)
+        {
+            // a cyclic is detected - "this" is already within the list
+            wxString msg;
+            msg.Printf(wxT("table %p is already within the linked list - and will not be added again to avoid creating a cycle"), (void *)this);
+            wxLogError(msg.c_str());
+            return; // Gal Shalif: this is probably a fatal error. Anyway, avoid a cycle - do not add "this" again
+        }
+    }
+#endif /* __WXDEBUG__ */
+
     m_next = sm_first;
     if (m_next)
         m_next->m_previous = this;
