--- C:/Users/JOOSTN~1/AppData/Local/Temp/app.cpp-revBASE.svn000.tmp.cpp	di okt 26 15:06:51 2010
+++ N:/wxwtrunk/src/msw/app.cpp	di okt 26 15:04:17 2010
@@ -242,14 +242,18 @@
     if ( !wxEventLoop::GetActive() )
         return DoSimpleWaitForThread(hThread);
 
+    DWORD dwWakeMask=0;
+    if(wxThread::IsMain())
+    {
+        dwWakeMask=QS_ALLINPUT | QS_ALLPOSTMESSAGE;
+    }
     return ::MsgWaitForMultipleObjects
              (
                1,                   // number of objects to wait for
                (HANDLE *)&hThread,  // the objects
                false,               // wait for any objects, not all
                INFINITE,            // no timeout
-               QS_ALLINPUT |        // return as soon as there are any events
-               QS_ALLPOSTMESSAGE
+               dwWakeMask
              );
 }
 
