diff --git a/src/common/dcbufcmn.cpp b/src/common/dcbufcmn.cpp
--- a/src/common/dcbufcmn.cpp
+++ b/src/common/dcbufcmn.cpp
@@ -140,23 +140,9 @@ void wxBufferedDC::UnMask()
     if ( m_style & wxBUFFER_CLIENT_AREA )
         GetDeviceOrigin(&x, &y);
 
-    // It's possible that the buffer may be bigger than the area that needs to
-    // be drawn (the client size of the window is smaller than the bitmap, or
-    // a shared bitmap has been reused for a smaller area, etc.) so avoid
-    // blitting too much if possible, but only use the real DC size if the
-    // wxBUFFER_VIRTUAL_AREA style is not set.
     int width = m_area.GetWidth(),
         height = m_area.GetHeight();
 
-    if (!(m_style & wxBUFFER_VIRTUAL_AREA))
-    {
-        int widthDC,
-            heightDC;
-        m_dc->GetSize(&widthDC, &heightDC);
-        width = wxMin(width, widthDC);
-        height = wxMin(height, heightDC);
-    }
-
     const wxPoint origin = GetLogicalOrigin();
     m_dc->Blit(-origin.x, -origin.y, width, height, this, -x, -y);
     m_dc = NULL;
