Index: samples/image/canvas.cpp
===================================================================
--- samples/image/canvas.cpp	(revision 63095)
+++ samples/image/canvas.cpp	(working copy)
@@ -27,6 +27,7 @@
 #include "wx/wfstream.h"
 #include "wx/quantize.h"
 #include "wx/stopwatch.h"
+#include "wx/dcbuffer.h"
 
 #if wxUSE_CLIPBOARD
     #include "wx/dataobj.h"
@@ -387,7 +388,23 @@
 {
     wxPaintDC dc( this );
     PrepareDC( dc );
+    if (0)
+    {
+        // looks fine, in trunk and 2.8 branch
+        OnPaint(dc);
+    }
+    else
+    {
+        // clipped beyond the window client area, in trunk; ok in 2.8 branch
+        wxBufferedDC buf_dc(&dc, GetVirtualSize());
+        //PrepareDC( buf_dc);
+        buf_dc.Clear();
+        OnPaint(buf_dc);
+    }
+}
 
+void MyCanvas::OnPaint(wxDC& dc)
+{
     dc.DrawText( wxT("Loaded image"), 30, 10 );
     if (my_square.Ok())
         dc.DrawBitmap( my_square, 30, 30 );
Index: samples/image/canvas.h
===================================================================
--- samples/image/canvas.h	(revision 63095)
+++ samples/image/canvas.h	(working copy)
@@ -22,6 +22,7 @@
     ~MyCanvas();
 
     void OnPaint( wxPaintEvent &event );
+    void OnPaint(wxDC&);
     void CreateAntiAliasedBitmap();
 
     wxBitmap  my_horse_png;
