Index: configure
===================================================================
--- configure	(revision 67297)
+++ configure	(working copy)
@@ -30954,7 +30954,7 @@
 
                 TOOLKIT_INCLUDE="$TOOLKIT_INCLUDE -DG_DISABLE_CAST_CHECKS"
 
-                GTK_CHECKS_FLAGS="-DGTK_DISABLE_SINGLE_INCLUDES"
+                # GTK_CHECKS_FLAGS="-DGTK_DISABLE_SINGLE_INCLUDES"
         TOOLKIT_INCLUDE="$TOOLKIT_INCLUDE $GTK_CHECKS_FLAGS"
 
         AFMINSTALL=afminstall
