Index: grid.h
===================================================================
--- grid.h	(revision 64105)
+++ grid.h	(working copy)
@@ -1622,10 +1622,13 @@
     The default table class is called wxGridStringTable and holds an array of
     strings. An instance of such a class is created by CreateGrid().
 
-    wxGridCellRenderer is the abstract base class for rendereing contents in a
+    wxGridCellRenderer is the abstract base class for rendering contents in a
     cell. The following renderers are predefined:
 
+    - wxGridCellAutoWrapStringRenderer
     - wxGridCellBoolRenderer
+    - wxGridCellDateTimeRenderer
+    - wxGridCellEnumRenderer
     - wxGridCellFloatRenderer
     - wxGridCellNumberRenderer
     - wxGridCellStringRenderer
@@ -1636,8 +1639,10 @@
     wxGridCellEditor is the abstract base class for editing the value of a
     cell. The following editors are predefined:
 
+    - wxGridCellAutoWrapStringEditor
     - wxGridCellBoolEditor
     - wxGridCellChoiceEditor
+    - wxGridCellEnumEditor
     - wxGridCellFloatEditor
     - wxGridCellNumberEditor
     - wxGridCellTextEditor
