Index: src/common/popupcmn.cpp
===================================================================
--- src/common/popupcmn.cpp	(revision 56862)
+++ src/common/popupcmn.cpp	(working copy)
@@ -260,7 +260,20 @@
 
     m_child->PushEventHandler(m_handlerPopup);
 
-    m_focus = winFocus ? winFocus : this;
+#if defined(__WXMSW__)
+    // Focusing on child of popup window does not work on MSW unless WS_POPUP
+    // style is set. We do not even want to try to set the focus, as it may
+    // provoke errors on some Windows versions (Vista and later).
+    if ( !(::GetWindowLong(GetHwnd(), GWL_STYLE) & WS_POPUP) )
+    {
+        m_focus = this;
+    }
+    else
+#endif
+    {
+        m_focus = winFocus ? winFocus : this;
+    }
+
     m_focus->SetFocus();
 
 #if defined( __WXMSW__ ) || defined( __WXMAC__)
