diff -rbu2 ORIGINAL.wxGTK-2.3.2/include/wx/gtk/toplevel.h wxGTK-2.3.2/include/wx/gtk/toplevel.h
--- ORIGINAL.wxGTK-2.3.2/include/wx/gtk/toplevel.h	2001-12-11 03:07:06.000000000 +0800
+++ wxGTK-2.3.2/include/wx/gtk/toplevel.h	2002-06-19 11:23:51.449999000 +0800
@@ -61,4 +61,5 @@
 
     virtual bool Show(bool show = TRUE);
+    virtual bool Destroy();
 
     virtual void SetTitle( const wxString &title );
diff -rbu2 ORIGINAL.wxGTK-2.3.2/include/wx/gtk/window.h wxGTK-2.3.2/include/wx/gtk/window.h
--- ORIGINAL.wxGTK-2.3.2/include/wx/gtk/window.h	2001-12-11 03:07:06.000000000 +0800
+++ wxGTK-2.3.2/include/wx/gtk/window.h	2002-06-19 12:03:31.280004000 +0800
@@ -51,4 +51,5 @@
 
     virtual bool Destroy();
+    virtual void NotifyDestroy();
 
     virtual void Raise();
Only in wxGTK-2.3.2/samples/calendar: Makefile
Only in wxGTK-2.3.2/samples/caret: Makefile
Only in wxGTK-2.3.2/samples/checklst: Makefile
Only in wxGTK-2.3.2/samples/config: Makefile
Only in wxGTK-2.3.2/samples: config.cache
Only in wxGTK-2.3.2/samples: config.log
Only in wxGTK-2.3.2/samples: config.status
Only in wxGTK-2.3.2/samples/console: Makefile
Only in wxGTK-2.3.2/samples/controls: Makefile
Only in wxGTK-2.3.2/samples/db: Makefile
Only in wxGTK-2.3.2/samples/dialogs: Makefile
Only in wxGTK-2.3.2/samples/dialup: Makefile
Only in wxGTK-2.3.2/samples/dnd: Makefile
Only in wxGTK-2.3.2/samples/docview: Makefile
Only in wxGTK-2.3.2/samples/docvwmdi: Makefile
Only in wxGTK-2.3.2/samples/dragimag: Makefile
Only in wxGTK-2.3.2/samples/drawing: Makefile
Only in wxGTK-2.3.2/samples/dynamic: Makefile
Only in wxGTK-2.3.2/samples/erase: Makefile
Only in wxGTK-2.3.2/samples/event: Makefile
Only in wxGTK-2.3.2/samples/exec: Makefile
Only in wxGTK-2.3.2/samples/font: Makefile
Only in wxGTK-2.3.2/samples/grid: Makefile
Only in wxGTK-2.3.2/samples/help: Makefile
Only in wxGTK-2.3.2/samples/html/about: Makefile
Only in wxGTK-2.3.2/samples/html/help: Makefile
Only in wxGTK-2.3.2/samples/html/helpview: Makefile
Only in wxGTK-2.3.2/samples/html: Makefile
Only in wxGTK-2.3.2/samples/html/printing: Makefile
Only in wxGTK-2.3.2/samples/html/test: Makefile
Only in wxGTK-2.3.2/samples/html/virtual: Makefile
Only in wxGTK-2.3.2/samples/html/widget: Makefile
Only in wxGTK-2.3.2/samples/html/zip: Makefile
Only in wxGTK-2.3.2/samples/image: Makefile
Only in wxGTK-2.3.2/samples/internat: Makefile
Only in wxGTK-2.3.2/samples/ipc: Makefile
Only in wxGTK-2.3.2/samples/joytest: Makefile
Only in wxGTK-2.3.2/samples/layout: Makefile
Only in wxGTK-2.3.2/samples/listctrl: Makefile
Only in wxGTK-2.3.2/samples: Makefile
Only in wxGTK-2.3.2/samples/mdi: Makefile
Only in wxGTK-2.3.2/samples/memcheck: Makefile
Only in wxGTK-2.3.2/samples/menu: Makefile
Only in wxGTK-2.3.2/samples/minifram: Makefile
Only in wxGTK-2.3.2/samples/minimal: Makefile
diff -rbu2 ORIGINAL.wxGTK-2.3.2/samples/minimal/minimal.cpp wxGTK-2.3.2/samples/minimal/minimal.cpp
--- ORIGINAL.wxGTK-2.3.2/samples/minimal/minimal.cpp	2001-12-11 03:07:13.000000000 +0800
+++ wxGTK-2.3.2/samples/minimal/minimal.cpp	2002-06-20 17:04:35.853123000 +0800
@@ -67,4 +67,5 @@
     void OnQuit(wxCommandEvent& event);
     void OnAbout(wxCommandEvent& event);
+    void OnWindowDestroy( wxWindowDestroyEvent& event );
 
 private:
@@ -99,4 +100,5 @@
     EVT_MENU(Minimal_Quit,  MyFrame::OnQuit)
     EVT_MENU(Minimal_About, MyFrame::OnAbout)
+    EVT_WINDOW_DESTROY( MyFrame::OnWindowDestroy )
 END_EVENT_TABLE()
 
@@ -187,2 +189,7 @@
     wxMessageBox(msg, _T("About Minimal"), wxOK | wxICON_INFORMATION, this);
 }
+
+void MyFrame::OnWindowDestroy( wxWindowDestroyEvent& event )
+{
+    wxMessageBox(_T("The frame has been destroyed"), _T("Notify"), wxOK | wxICON_INFORMATION, this);
+}
Only in wxGTK-2.3.2/samples/newgrid: Makefile
Only in wxGTK-2.3.2/samples/notebook: Makefile
Only in wxGTK-2.3.2/samples/opengl/cube: Makefile
Only in wxGTK-2.3.2/samples/opengl/isosurf: Makefile
Only in wxGTK-2.3.2/samples/opengl: Makefile
Only in wxGTK-2.3.2/samples/opengl/penguin: Makefile
Only in wxGTK-2.3.2/samples/png: Makefile
Only in wxGTK-2.3.2/samples/printing: Makefile
Only in wxGTK-2.3.2/samples/proplist: Makefile
Only in wxGTK-2.3.2/samples/propsize: Makefile
Only in wxGTK-2.3.2/samples/resource: Makefile
Only in wxGTK-2.3.2/samples/richedit: Makefile
Only in wxGTK-2.3.2/samples/rotate: Makefile
Only in wxGTK-2.3.2/samples/sashtest: Makefile
Only in wxGTK-2.3.2/samples/scroll: Makefile
Only in wxGTK-2.3.2/samples/scrollsub: Makefile
Only in wxGTK-2.3.2/samples/sockets: Makefile
Only in wxGTK-2.3.2/samples/splitter: Makefile
Only in wxGTK-2.3.2/samples/statbar: Makefile
Only in wxGTK-2.3.2/samples/text: Makefile
Only in wxGTK-2.3.2/samples/thread: Makefile
Only in wxGTK-2.3.2/samples/toolbar: Makefile
Only in wxGTK-2.3.2/samples/treectrl: Makefile
Only in wxGTK-2.3.2/samples/treelay: Makefile
Only in wxGTK-2.3.2/samples/typetest: Makefile
Only in wxGTK-2.3.2/samples/validate: Makefile
Only in wxGTK-2.3.2/samples/widgets: Makefile
Only in wxGTK-2.3.2/samples/wizard: Makefile
diff -rbu2 ORIGINAL.wxGTK-2.3.2/src/gtk/toplevel.cpp wxGTK-2.3.2/src/gtk/toplevel.cpp
--- ORIGINAL.wxGTK-2.3.2/src/gtk/toplevel.cpp	2001-12-11 03:07:06.000000000 +0800
+++ wxGTK-2.3.2/src/gtk/toplevel.cpp	2002-06-21 10:35:27.530001000 +0800
@@ -579,4 +579,12 @@
 }
 
+bool wxTopLevelWindowGTK::Destroy( void )
+{
+    wxASSERT_MSG( (m_widget != NULL), wxT("invalid frame") );
+    this->NotifyDestroy( );
+
+    return wxTopLevelWindowBase::Destroy( );
+}
+
 void wxTopLevelWindowGTK::DoMoveWindow(int WXUNUSED(x), int WXUNUSED(y), int WXUNUSED(width), int WXUNUSED(height) )
 {
diff -rbu2 ORIGINAL.wxGTK-2.3.2/src/gtk/window.cpp wxGTK-2.3.2/src/gtk/window.cpp
--- ORIGINAL.wxGTK-2.3.2/src/gtk/window.cpp	2001-12-11 03:07:06.000000000 +0800
+++ wxGTK-2.3.2/src/gtk/window.cpp	2002-06-21 11:34:07.080019000 +0800
@@ -2748,4 +2748,22 @@
 }
 
+void wxWindowGTK::NotifyDestroy( void )
+{
+    wxWindowDestroyEvent event(this);
+    (void)this->GetEventHandler()->ProcessEvent(event);
+	
+    for ( wxWindowList::Node *node = this->GetChildren().GetFirst();
+          node;
+          node = node->GetNext() )
+	{
+        wxWindow *child = (wxWindow*)node->GetData();
+
+        wxASSERT_MSG( child, wxT("children list contains empty nodes") );
+
+        child->NotifyDestroy( );
+	}
+}
+
+
 void wxWindowGTK::DoMoveWindow(int x, int y, int width, int height)
 {

 	  	 
