diff --git a/include/wx/datetime.h b/include/wx/datetime.h
index 04285dc..6baa050 100644
--- a/include/wx/datetime.h
+++ b/include/wx/datetime.h
@@ -840,13 +840,11 @@ public:
 
     inline bool operator==(const wxDateTime& dt) const
     {
-        wxASSERT_MSG( IsValid() && dt.IsValid(), wxT("invalid wxDateTime") );
         return m_time == dt.m_time;
     }
 
     inline bool operator!=(const wxDateTime& dt) const
     {
-        wxASSERT_MSG( IsValid() && dt.IsValid(), wxT("invalid wxDateTime") );
         return m_time != dt.m_time;
     }
 
