diff -r -u -p a/samples/widgets/button.cpp b/samples/widgets/button.cpp
--- a/samples/widgets/button.cpp	2020-11-17 07:09:36.385097100 +0100
+++ b/samples/widgets/button.cpp	2020-11-17 07:06:46.838222100 +0100
@@ -231,6 +231,13 @@ void ButtonWidgetsPage::CreateContent()
 {
     wxSizer *sizerTop = new wxBoxSizer(wxHORIZONTAL);
 
+    wxBitmapButton* btn1 = new wxBitmapButton( this, wxID_ANY, wxBitmap(button_xpm), wxDefaultPosition, wxDefaultSize, wxBU_AUTODRAW );
+    sizerTop->Add(btn1, 0, wxALIGN_TOP|wxALL, 5);
+
+    wxBitmapButton* btn2 = new wxBitmapButton( this, wxID_ANY, wxBitmap(button_xpm), wxDefaultPosition, wxDefaultSize, wxBU_AUTODRAW );
+    sizerTop->Add(btn2, 0, wxALIGN_TOP|wxALL, 5);
+    btn2->Disable();
+
     // left pane
     wxStaticBox *box = new wxStaticBox(this, wxID_ANY, "&Set style");
 
