 samples/minimal/minimal.cpp | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/samples/minimal/minimal.cpp b/samples/minimal/minimal.cpp
index f765a70dac..b8eebd3144 100644
--- a/samples/minimal/minimal.cpp
+++ b/samples/minimal/minimal.cpp
@@ -177,6 +177,13 @@ MyFrame::MyFrame(const wxString& title)
     CreateStatusBar(2);
     SetStatusText("Welcome to wxWidgets!");
 #endif // wxUSE_STATUSBAR
+
+    wxScrolledWindow* sw = new wxScrolledWindow(this);
+
+    wxSizer* s = new wxBoxSizer(wxVERTICAL);
+    s->Add(new wxPanel(sw), 1, wxEXPAND); // a StretchSpacer would not break the following button
+    s->Add(new wxButton(sw, wxID_ANY, "b"), 0, wxALIGN_RIGHT);
+    sw->SetSizer(s);
 }
 
 
