Index: docs/changes.txt
===================================================================
--- docs/changes.txt	(revision 62578)
+++ docs/changes.txt	(working copy)
@@ -142,10 +142,12 @@
   wxBORDER_THEME in a container window.
 - Suppressed spurious character event for decimal key in numeric keypad.
 - Prevented wxCheckListBox from corrupting menu items.
+- Status bar help texts missing when mdi child active (troelsk).
 
 wxGTK:
 
 - Don't access a possibly destroyed timer object from GTK callback (Tim Kosse).
+- Status bar help texts missing when mdi child active (troelsk).
 
 wxMac:
 
Index: src/gtk/mdi.cpp
===================================================================
--- src/gtk/mdi.cpp	(revision 62578)
+++ src/gtk/mdi.cpp	(working copy)
@@ -417,12 +417,7 @@
 {
 #if wxUSE_STATUSBAR
     wxMDIParentFrame *mdi_frame = (wxMDIParentFrame*)m_parent->GetParent();
-    if ( !ShowMenuHelp(mdi_frame->GetStatusBar(), event.GetMenuId()) )
-    {
-        // we don't have any help text for this item, but may be the MDI frame
-        // does?
-        mdi_frame->OnMenuHighlight(event);
-    }
+    mdi_frame->OnMenuHighlight(event);
 #endif // wxUSE_STATUSBAR
 }
 
Index: src/msw/mdi.cpp
===================================================================
--- src/msw/mdi.cpp	(revision 62578)
+++ src/msw/mdi.cpp	(working copy)
@@ -489,6 +489,8 @@
                 {
                     processed = m_currentChild->
                         HandleMenuSelect(item, flags, hmenu);
+                    // HandleMenuSelect above yields no status bar help text, try to obtain a text like this
+                    ShowMenuHelp(GetStatusBar(), item);
                 }
             }
             break;
