 samples/minimal/minimal.cpp | 53 +++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 53 insertions(+)

diff --git a/samples/minimal/minimal.cpp b/samples/minimal/minimal.cpp
index f765a70dac..9b44848090 100644
--- a/samples/minimal/minimal.cpp
+++ b/samples/minimal/minimal.cpp
@@ -29,6 +29,8 @@
     #include "wx/wx.h"
 #endif
 
+#include <wx/notebook.h>
+
 // ----------------------------------------------------------------------------
 // resources
 // ----------------------------------------------------------------------------
@@ -172,6 +174,57 @@ MyFrame::MyFrame(const wxString& title)
     sizer->Add(aboutBtn, wxSizerFlags().Center());
 #endif // wxUSE_MENUS/!wxUSE_MENUS
 
+    wxPanel *panel = new wxPanel(this);
+    wxNotebook *notebook = new wxNotebook(panel, wxID_ANY);
+    panel->SetSizer(new wxBoxSizer(wxHORIZONTAL));
+    panel->GetSizer()->Add(notebook, 1, wxALIGN_CENTER, 5);
+    wxPanel *tab = new wxPanel(notebook);
+    notebook->AddPage(tab, "Tab", true);
+
+    wxBoxSizer *sizer1 = new wxBoxSizer(wxVERTICAL);
+    wxBoxSizer *sizer2 = new wxBoxSizer(wxHORIZONTAL);
+    tab->SetSizer(sizer1);
+
+    struct TestFunc1
+    {
+        void operator()(wxCommandEvent &event)
+        {
+            event.Skip();
+            wxDynamicCast(event.GetEventUserData(), wxWindow)->Layout();
+        }
+    };
+    struct TestFunc2
+    {
+        void operator()(wxCommandEvent &event)
+        {
+            event.Skip();
+            wxStaticText *text = wxDynamicCast(event.GetEventUserData(), wxStaticText);
+            text->SetLabel(text->GetLabel());
+        }
+    };
+
+    wxStaticText *static1 = new wxStaticText(tab, wxID_ANY, "Static Text 1");
+    wxStaticText *static2 = new wxStaticText(tab, wxID_ANY, "Static Text 2");
+    sizer2->Add(static1);
+    sizer2->AddStretchSpacer();
+    sizer2->Add(static2);
+    sizer1->Add(sizer2, 1, wxEXPAND);
+    wxButton *button = new wxButton(tab, wxID_ANY, "Test Me");
+    sizer1->Add(button, 0, wxALIGN_CENTER);
+    button->Bind(wxEVT_BUTTON, TestFunc1(), wxID_ANY, wxID_ANY, tab);
+    button->Bind(wxEVT_BUTTON, TestFunc2(), wxID_ANY, wxID_ANY, static2);
+
+    // Add a new tab with several more labels to try to force the app to do more work to make flickering worse
+    // Not entirely sure if this actually helps, but whatever
+    wxPanel *tab2 = new wxPanel(notebook);
+    wxBoxSizer *tab2Sizer = new wxBoxSizer(wxVERTICAL);
+    tab2->SetSizer(tab2Sizer);
+    notebook->AddPage(tab2, "Dummy Controls");
+    for (int i = 0; i < 1000; i++)
+        tab2Sizer->Add(new wxStaticText(tab2, wxID_ANY, wxString::Format("Label%d", i)));
+
+
+
 #if wxUSE_STATUSBAR
     // create a status bar just for fun (by default with 1 pane only)
     CreateStatusBar(2);
