Index: src/common/filefn.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/common/filefn.cpp,v
retrieving revision 1.242
diff -u -r1.242 filefn.cpp
--- src/common/filefn.cpp	2005/06/08 14:32:16	1.242
+++ src/common/filefn.cpp	2005/06/15 09:02:14
@@ -302,7 +302,11 @@
 #elif defined(__WIN32__) && !defined(__WXMICROWIN__)
     // we must use GetFileAttributes() instead of the ANSI C functions because
     // it can cope with network (UNC) paths unlike them
+#if wxUSE_UNICODE
+    DWORD ret = ::GetFileAttributesW(filename);
+#else // !wxUSE_UNICODE
     DWORD ret = ::GetFileAttributes(filename);
+#endif // !wxUSE_UNICODE
 
     return (ret != (DWORD)-1) && !(ret & FILE_ATTRIBUTE_DIRECTORY);
 #else // !__WIN32__

 	  	 
