diff -uNr wxWidgets.orig/include/wx/osx/listbox.h wxWidgets/include/wx/osx/listbox.h
--- wxWidgets.orig/include/wx/osx/listbox.h	2018-06-29 09:37:34.000000000 +0300
+++ wxWidgets/include/wx/osx/listbox.h	2018-06-29 09:56:14.000000000 +0300
@@ -122,6 +122,12 @@
     bool MacGetBlockEvents() const { return m_blockEvents; }
 
     virtual void HandleLineEvent( unsigned int n, bool doubleClick );
+
+	// These are called by wxNSTableView
+	using wxListBoxBase::DoChangeSingleSelection;
+	using wxListBoxBase::CalcAndSendEvent;
+	int GetOldSelection() const { return m_oldSelections.empty() ? wxNOT_FOUND : m_oldSelections[0]; }
+
 protected:
     // callback for derived classes which may have to insert additional data
     // at a certain line - which cannot be predetermined for sorted list data
diff -uNr wxWidgets.orig/src/osx/cocoa/listbox.mm wxWidgets/src/osx/cocoa/listbox.mm
--- wxWidgets.orig/src/osx/cocoa/listbox.mm	2018-06-29 09:36:44.000000000 +0300
+++ wxWidgets/src/osx/cocoa/listbox.mm	2018-06-29 09:57:54.000000000 +0300
@@ -291,24 +291,34 @@
     wxUnusedVar(notification);
     
     int row = [self selectedRow];
-    
-    if (row == -1) 
-    {
-        // no row selected
-    } 
-    else 
-    {
-        wxWidgetCocoaImpl* impl = (wxWidgetCocoaImpl* ) wxWidgetImpl::FindFromWXWidget( self );
-        wxListBox *list = static_cast<wxListBox*> ( impl->GetWXPeer());
-        wxCHECK_RET( list != NULL , wxT("Listbox expected"));
-        
-        if ((row < 0) || (row > (int) list->GetCount()))  // OS X can select an item below the last item
-            return;
-        
-        if ( !list->MacGetBlockEvents() )
-            list->HandleLineEvent( row, false );
-    }
-    
+	
+	wxWidgetCocoaImpl* impl = (wxWidgetCocoaImpl* ) wxWidgetImpl::FindFromWXWidget( self );
+	wxListBox *list = static_cast<wxListBox*> ( impl->GetWXPeer());
+	wxCHECK_RET( list != NULL , wxT("Listbox expected"));
+	
+	// Correct notification events for multiselection list, like in Carbon version
+	if (list->HasMultipleSelection() && !list->MacGetBlockEvents())
+	{
+		list->CalcAndSendEvent();
+		return;
+	}
+
+	if ( !list->MacGetBlockEvents() )
+	{
+		// OS X can select an item below the last item. In that case keep the old selection because
+		// in wxWidgets API there is no notification event for removing the selection from a single-selection list box.
+		// Otherwise call DoChangeSingleSelection so GetOldSelection() will return the correct value if row < 0 later.
+		if ((row < 0) || (row > (int) list->GetCount()))
+		{
+			int oldsel = list->GetOldSelection();
+			if (oldsel >= 0)
+				list->SetSelection(oldsel);
+			return;
+		}
+		if ( !list->DoChangeSingleSelection(row) )
+			return ;
+		list->HandleLineEvent( row, false );
+	}
 } 
 
 - (void)setFont:(NSFont *)aFont
