diff --git samples/minimal/minimal.cpp samples/minimal/minimal.cpp
index 80bf766..4b3af40 100644
--- samples/minimal/minimal.cpp
+++ samples/minimal/minimal.cpp
@@ -69,6 +69,7 @@ public:
     void OnAbout(wxCommandEvent& event);
 
 private:
+wxWindow* m_win;
     // any class wishing to process wxWidgets events must use this macro
     DECLARE_EVENT_TABLE()
 };
@@ -172,6 +173,16 @@ MyFrame::MyFrame(const wxString& title)
     CreateStatusBar(2);
     SetStatusText("Welcome to wxWidgets!");
 #endif // wxUSE_STATUSBAR
+
+wxBoxSizer* sizer = new wxBoxSizer(wxVERTICAL);
+
+m_win = new wxWindow(this, wxID_ANY, wxDefaultPosition, wxSize(100,100), /*wxBORDER_RAISED*//*wxBORDER_SUNKEN*/wxBORDER_SIMPLE|wxTAB_TRAVERSAL);
+m_win->SetBackgroundColour(*wxRED);
+
+sizer->Add(m_win, 0, wxALL|wxALIGN_CENTRE, 20);
+
+SetSizer(sizer);
+sizer->Layout();
 }
 
 
@@ -185,16 +196,12 @@ void MyFrame::OnQuit(wxCommandEvent& WXUNUSED(event))
 
 void MyFrame::OnAbout(wxCommandEvent& WXUNUSED(event))
 {
-    wxMessageBox(wxString::Format
-                 (
-                    "Welcome to %s!\n"
-                    "\n"
-                    "This is the minimal wxWidgets sample\n"
-                    "running under %s.",
-                    wxVERSION_STRING,
-                    wxGetOsDescription()
-                 ),
-                 "About wxWidgets minimal sample",
-                 wxOK | wxICON_INFORMATION,
-                 this);
+if (m_win->IsShown())
+  {
+	  m_win->Hide();
+  }
+  else
+  {
+	  m_win->Show();
+  }
 }
