diff -bu1 -r wxWindows/demos/dbbrowse/browsedb.cpp wxWin/demos/dbbrowse/browsedb.cpp
--- wxWindows/demos/dbbrowse/browsedb.cpp	2004-01-12 10:19:54.000000000 +0100
+++ wxWin/demos/dbbrowse/browsedb.cpp	2004-01-19 12:42:14.000000000 +0100
@@ -203,3 +203,3 @@
 //----------------------------------------------------------------------------------------
-bool BrowserDB::OnGetNext(int Cols,int Quiet)
+bool BrowserDB::OnGetNext(int Cols,int WXUNUSED(Quiet))
 {
@@ -217,3 +217,3 @@
     {
-        return FALSE;
+#ifdef __WXDEBUG__
         Temp0.Printf(_("\n-E-> BrowserDB::OnGetNext - ODBC-Error with GetNext \n-E-> "));
@@ -222,2 +222,4 @@
         wxMessageBox(Temp0);
+#endif
+        return FALSE;
     }
@@ -346,3 +348,4 @@
     //---------------------------------------------------------------------------------------
-    SQLStmt.sprintf(_T("SELECT * FROM %s"),db_BrowserDB->SQLTableName(tb_Name.c_str()));
+    wxString tablename = db_BrowserDB->SQLTableName(tb_Name.c_str());
+    SQLStmt.sprintf(_T("SELECT * FROM %s"),tablename.c_str());
     if (!db_BrowserDB->ExecSql((wxChar *)(SQLStmt.GetData())))
@@ -400,3 +403,3 @@
 //----------------------------------------------------------------------------------------
-wxDbInf* BrowserDB::OnGetCatalog(int Quiet)
+wxDbInf* BrowserDB::OnGetCatalog(int WXUNUSED(Quiet))
 {
@@ -409,3 +412,3 @@
 //----------------------------------------------------------------------------------------
-wxDbColInf* BrowserDB::OnGetColumns(wxChar *tableName, UWORD numCols, int Quiet)
+wxDbColInf* BrowserDB::OnGetColumns(wxChar *tableName, UWORD numCols, int WXUNUSED(Quiet))
 {
diff -bu1 -r wxWindows/demos/dbbrowse/dbgrid.cpp wxWin/demos/dbbrowse/dbgrid.cpp
--- wxWindows/demos/dbbrowse/dbgrid.cpp	2004-01-12 10:19:55.000000000 +0100
+++ wxWin/demos/dbbrowse/dbgrid.cpp	2004-01-19 13:39:50.000000000 +0100
@@ -87,3 +87,3 @@
     //---------------------------------------------------------------------------------------
-    int  i=0,x,y,z, ValidTable=0;
+    int  x,y,z;
     wxString Temp0;
@@ -100,3 +100,3 @@
             {    // Yes, the Data of this Table shall be put into the Grid
-                ValidTable = x;                                                  // Save the Tablenumber
+                int ValidTable = x;                                              // Save the Tablenumber
                 (db_Br+i_Which)->OnSelect(Table,FALSE);                          // Select * from "table"
@@ -106,3 +106,3 @@
                 {   // Pointer is Valid, Column Informationen sind Vorhanden
-                    i = (db_Br+i_Which)->i_Records;                                 // How many Records are there
+                    int i = (db_Br+i_Which)->i_Records;                             // How many Records are there
                     (db_Br+i_Which)->i_Which = ValidTable;                          // Still used ???? mj10777
@@ -156,3 +156,3 @@
 //----------------------------------------------------------------------------------------
-void DBGrid::OnModusEdit(wxCommandEvent& event)
+void DBGrid::OnModusEdit(wxCommandEvent& WXUNUSED(event))
 {
@@ -165,3 +165,3 @@
 //----------------------------------------------------------------------------------------
-void DBGrid::OnModusBrowse(wxCommandEvent& event)
+void DBGrid::OnModusBrowse(wxCommandEvent& WXUNUSED(event))
 {
diff -bu1 -r wxWindows/demos/dbbrowse/dbtree.cpp wxWin/demos/dbbrowse/dbtree.cpp
--- wxWindows/demos/dbbrowse/dbtree.cpp	2004-01-12 10:19:55.000000000 +0100
+++ wxWin/demos/dbbrowse/dbtree.cpp	2004-01-19 12:43:02.000000000 +0100
@@ -357,3 +357,3 @@
 //----------------------------------------------------------------------------------------
-void DBTree::OnDBGrid(wxCommandEvent& event)
+void DBTree::OnDBGrid(wxCommandEvent& WXUNUSED(event))
 {
@@ -380,3 +380,3 @@
 //----------------------------------------------------------------------------------------
-void DBTree::OnDBClass(wxCommandEvent& event)
+void DBTree::OnDBClass(wxCommandEvent& WXUNUSED(event))
 {
@@ -394,3 +394,3 @@
 //----------------------------------------------------------------------------------------
-void DBTree::OnTableClass(wxCommandEvent& event)
+void DBTree::OnTableClass(wxCommandEvent& WXUNUSED(event))
 {
@@ -416,3 +416,3 @@
 //----------------------------------------------------------------------------------------
-void DBTree::OnTableClassAll(wxCommandEvent& event)
+void DBTree::OnTableClassAll(wxCommandEvent& WXUNUSED(event))
 {
diff -bu1 -r wxWindows/demos/dbbrowse/doc.cpp wxWin/demos/dbbrowse/doc.cpp
--- wxWindows/demos/dbbrowse/doc.cpp	2004-01-12 10:19:55.000000000 +0100
+++ wxWin/demos/dbbrowse/doc.cpp	2004-01-19 13:29:55.000000000 +0100
@@ -116,2 +116,3 @@
     p_FolderClose = NULL;
+    wxUnusedVar(p_FolderClose);
     //---------------------------------------------------------------------------------------
@@ -158,3 +159,3 @@
     i_DSN = 0;       // Counter
-    int i = 0;
+    int i;
     //---------------------------------------------------------------------------------------
@@ -202,3 +203,2 @@
     }
-    i = 0;
     //---------------------------------------------------------------------------------------
diff -bu1 -r wxWindows/demos/dbbrowse/tabpgwin.cpp wxWin/demos/dbbrowse/tabpgwin.cpp
--- wxWindows/demos/dbbrowse/tabpgwin.cpp	2004-01-12 10:19:55.000000000 +0100
+++ wxWin/demos/dbbrowse/tabpgwin.cpp	2004-01-19 12:40:56.000000000 +0100
@@ -159,3 +159,3 @@
 {
-    wxNode* pTab = mTabs.First();
+    wxNode* pTab = mTabs.GetFirst();
     
@@ -163,4 +163,4 @@
     {
-        delete ((twTabInfo*)pTab->Data());
-        pTab = pTab->Next();
+        delete ((twTabInfo*)pTab->GetData());
+        pTab = pTab->GetNext();
     }
@@ -169,6 +169,6 @@
 //---------------------------------------------------------------------------
-void wxTabbedWindow::SizeTabs(int x,int y, int width, int height, bool repant)
+void wxTabbedWindow::SizeTabs(int x,int y, int width, int height, bool WXUNUSED(repant))
 {
-    wxNode* pTabNode = mTabs.First();
-    int n = 0;
+    wxNode* pTabNode = mTabs.GetFirst();
+    size_t n = 0;
     
@@ -176,3 +176,3 @@
     {
-        twTabInfo& info = *((twTabInfo*)pTabNode->Data());
+        twTabInfo& info = *((twTabInfo*)pTabNode->GetData());
         
@@ -193,3 +193,3 @@
         
-        pTabNode = pTabNode->Next();
+        pTabNode = pTabNode->GetNext();
         ++n;
@@ -251,9 +251,9 @@
 {
-    twTabInfo* pTab = ((twTabInfo*)(mTabs.Nth( tabNo )->Data()));
+    twTabInfo* pTab = ((twTabInfo*)(mTabs.Item( tabNo )->GetData()));
     pTab->mpContent->Destroy();
     delete pTab;
-    mTabs.DeleteNode( mTabs.Nth( tabNo ) );
-    // if ( mActiveTab >= mTabs.Number() );
-    if ( mActiveTab >= mTabs.Number() )
-        mActiveTab = mTabs.Number() - 1;
+    mTabs.DeleteNode( mTabs.Item( tabNo ) );
+    // if ( mActiveTab >= mTabs.GetCount() );
+    if ( mActiveTab >= mTabs.GetCount() )
+        mActiveTab = mTabs.GetCount() - 1;
     SetActiveTab( mActiveTab );
@@ -264,3 +264,3 @@
 {
-    return mTabs.Number();
+    return mTabs.GetCount();
 }
@@ -270,3 +270,3 @@
 {
-    return ((twTabInfo*)(mTabs.Nth( tabNo )->Data()))->mpContent;
+    return ((twTabInfo*)(mTabs.Item( tabNo )->GetData()))->mpContent;
 }
@@ -365,4 +365,4 @@
     
-    int tabNo = 0;
-    wxNode* pNode = mTabs.First();
+    size_t tabNo = 0;
+    wxNode* pNode = mTabs.GetFirst();
     
@@ -372,3 +372,3 @@
         
-        twTabInfo& tab = *((twTabInfo*)(pNode->Data()));
+        twTabInfo& tab = *((twTabInfo*)(pNode->GetData()));
         
@@ -405,3 +405,3 @@
         
-        pNode = pNode->Next();
+        pNode = pNode->GetNext();
         ++tabNo;
@@ -455,3 +455,3 @@
     int     tabNo = 0;
-    wxNode* pNode = mTabs.First();
+    wxNode* pNode = mTabs.GetFirst();
     
@@ -459,7 +459,4 @@
     {
-        twTabInfo& tab = *((twTabInfo*)(pNode->Data()));
+        twTabInfo& tab = *((twTabInfo*)(pNode->GetData()));
         
-        int w,h;
-        w = tab.mDims.x;
-        h = tab.mDims.y;
         // hit test rectangle of the currnet tab title bar
@@ -474,3 +471,3 @@
         
-        pNode = pNode->Next();
+        pNode = pNode->GetNext();
         ++tabNo;
@@ -487,4 +484,4 @@
     
-    wxNode* pNode = mTabs.First();
-    int     tabNo = 0;
+    wxNode* pNode = mTabs.GetFirst();
+    size_t  tabNo = 0;
     
@@ -494,3 +491,3 @@
         {
-            twTabInfo& tab = *((twTabInfo*)(pNode->Data()));
+            twTabInfo& tab = *((twTabInfo*)(pNode->GetData()));
             tab.mpContent->Show(FALSE);
@@ -498,3 +495,3 @@
         
-        pNode = pNode->Next();
+        pNode = pNode->GetNext();
         ++tabNo;
@@ -546,3 +543,3 @@
     
-    wxNode* pNode = mTabs.First();
+    wxNode* pNode = mTabs.GetFirst();
     
@@ -553,3 +550,3 @@
     {
-        twTabInfo& tab = *((twTabInfo*)(pNode->Data()));
+        twTabInfo& tab = *((twTabInfo*)(pNode->GetData()));
         
@@ -572,3 +569,3 @@
         
-        pNode = pNode->Next();
+        pNode = pNode->GetNext();
     }
@@ -579,3 +576,3 @@
     
-    pNode = mTabs.First();
+    pNode = mTabs.GetFirst();
     
@@ -583,4 +580,4 @@
     {
-        ((twTabInfo*)(pNode->Data()))->mDims.y = mTitleHeight;;
-        pNode = pNode->Next();
+        ((twTabInfo*)(pNode->GetData()))->mDims.y = mTitleHeight;;
+        pNode = pNode->GetNext();
     }
@@ -595,3 +592,3 @@
     
-    pNode = mTabs.First();
+    pNode = mTabs.GetFirst();
     
@@ -599,3 +596,3 @@
     
-    int denomiator = mTabs.Number();
+    int denomiator = mTabs.GetCount();
     if ( denomiator == 0 )
@@ -607,3 +604,3 @@
     {
-        twTabInfo& tab = *((twTabInfo*)(pNode->Data()));
+        twTabInfo& tab = *((twTabInfo*)(pNode->GetData()));
         
@@ -622,3 +619,3 @@
         
-        pNode = pNode->Next();
+        pNode = pNode->GetNext();
     }
@@ -635,3 +632,3 @@
     
-    pNode = mTabs.First();
+    pNode = mTabs.GetFirst();
     
@@ -639,3 +636,3 @@
     {
-        twTabInfo& tab = *((twTabInfo*)(pNode->Data()));
+        twTabInfo& tab = *((twTabInfo*)(pNode->GetData()));
         
@@ -644,3 +641,3 @@
         
-        pNode = pNode->Next();
+        pNode = pNode->GetNext();
     }
@@ -651,3 +648,3 @@
 //---------------------------------------------------------------------------
-void wxTabbedWindow::OnPaint( wxPaintEvent& event )
+void wxTabbedWindow::OnPaint( wxPaintEvent& WXUNUSED(event) )
 {
@@ -658,3 +655,3 @@
 //---------------------------------------------------------------------------
-void wxTabbedWindow::OnSize ( wxSizeEvent& event )
+void wxTabbedWindow::OnSize ( wxSizeEvent& WXUNUSED(event) )
 {
@@ -665,3 +662,3 @@
 //---------------------------------------------------------------------------
-void wxTabbedWindow::OnBkErase( wxEraseEvent& event )
+void wxTabbedWindow::OnBkErase( wxEraseEvent& WXUNUSED(event) )
 {
@@ -751,3 +748,3 @@
 //---------------------------------------------------------------------------
-void wxPagedWindow::OnTabAdded( twTabInfo* pInfo )
+void wxPagedWindow::OnTabAdded( twTabInfo* WXUNUSED(pInfo) )
 {
@@ -773,3 +770,3 @@
 {
-    wxNode* pNode = mTabs.First();
+    wxNode* pNode = mTabs.GetFirst();
     
@@ -779,6 +776,6 @@
     {
-        twTabInfo& tab = *((twTabInfo*)(pNode->Data()));
+        twTabInfo& tab = *((twTabInfo*)(pNode->GetData()));
         
         len += tab.mDims.x;
-        pNode = pNode->Next();
+        pNode = pNode->GetNext();
     }
@@ -866,4 +863,4 @@
 
-    wxNode* pNode = mTabs.First();
-    int     tabNo = 0;
+    wxNode* pNode = mTabs.GetFirst();
+    size_t  tabNo = 0;
 
@@ -894,3 +891,3 @@
     {
-        twTabInfo& tab = *((twTabInfo*)(pNode->Data()));
+        twTabInfo& tab = *((twTabInfo*)(pNode->GetData()));
 
@@ -901,3 +898,3 @@
 
-        pNode = pNode->Next();
+        pNode = pNode->GetNext();
         ++tabNo;
@@ -907,3 +904,3 @@
     
-    pNode = mTabs.First();
+    pNode = mTabs.GetFirst();
     tabNo = 0;
@@ -914,3 +911,3 @@
     {
-        twTabInfo& tab = *((twTabInfo*)(pNode->Data()));
+        twTabInfo& tab = *((twTabInfo*)(pNode->GetData()));
         
@@ -928,3 +925,3 @@
         
-        pNode = pNode->Next();
+        pNode = pNode->GetNext();
         ++tabNo;
@@ -1038,3 +1035,3 @@
     
-    wxNode* pNode = mTabs.First();
+    wxNode* pNode = mTabs.GetFirst();
     
@@ -1042,3 +1039,3 @@
     {
-        twTabInfo& tab = *((twTabInfo*)(pNode->Data()));
+        twTabInfo& tab = *((twTabInfo*)(pNode->GetData()));
         
@@ -1057,3 +1054,3 @@
         
-        pNode = pNode->Next();
+        pNode = pNode->GetNext();
     }
@@ -1068,3 +1065,3 @@
 //---------------------------------------------------------------------------
-void wxPagedWindow::OnPaint( wxPaintEvent& event )
+void wxPagedWindow::OnPaint( wxPaintEvent& WXUNUSED(event) )
 {
@@ -1099,3 +1096,3 @@
 //---------------------------------------------------------------------------
-void wxPagedWindow::OnLButtonUp( wxMouseEvent& event )
+void wxPagedWindow::OnLButtonUp( wxMouseEvent& WXUNUSED(event) )
 {
diff -bu1 -r wxWindows/demos/dbbrowse/tabpgwin.h wxWin/demos/dbbrowse/tabpgwin.h
--- wxWindows/demos/dbbrowse/tabpgwin.h	2004-01-12 10:19:55.000000000 +0100
+++ wxWin/demos/dbbrowse/tabpgwin.h	2004-01-19 10:41:13.000000000 +0100
@@ -85,3 +85,3 @@
 
- int    mActiveTab;
+ size_t mActiveTab;
  int    mTitleHeight;

 	  	 
