diff -r -u -p a/interface/wx/propgrid/property.h b/interface/wx/propgrid/property.h
--- a/interface/wx/propgrid/property.h	2019-12-05 14:32:57.549737500 +0100
+++ b/interface/wx/propgrid/property.h	2019-12-05 14:51:35.137086000 +0100
@@ -2009,6 +2009,8 @@ public:
         @remarks
         Unlike wxPropertyGridInterface::SetPropertyColoursToDefault(),
         this does not automatically update the display.
+
+        @since 3.1.0
     */
     void SetDefaultColours(int flags = wxPG_RECURSE);
 
@@ -2164,6 +2166,8 @@ protected:
 
         @param recursively
             If @true, apply this operation recursively in child properties.
+
+        @since 3.1.0
     */
     void ClearCells(FlagType ignoreWithFlags, bool recursively);
 
