Index: include/wx/generic/dataview.h
===================================================================
--- include/wx/generic/dataview.h	(revision 73825)
+++ include/wx/generic/dataview.h	(working copy)
@@ -70,7 +70,7 @@
 
     virtual bool IsSortKey() const { return m_sort; }
 
-    virtual void UnsetAsSortKey() { m_sort = false; UpdateDisplay(); }
+    virtual void UnsetAsSortKey();
 
     virtual void SetSortOrder(bool ascending);
 
Index: src/generic/datavgen.cpp
===================================================================
--- src/generic/datavgen.cpp	(revision 73825)
+++ src/generic/datavgen.cpp	(working copy)
@@ -161,6 +161,22 @@
     }
 }
 
+void wxDataViewColumn::UnsetAsSortKey()
+{
+    m_sort = false;
+
+    if ( m_owner )
+    {
+        m_owner->SetSortingColumnIndex(wxNOT_FOUND);
+        const int idx = m_owner->GetColumnIndex(this);
+        m_owner->OnColumnChange(idx);
+    }
+    else
+    {
+        UpdateDisplay();
+    }
+}
+
 void wxDataViewColumn::SetSortOrder(bool ascending)
 {
     if ( !m_owner )
@@ -613,26 +629,32 @@
     {
         g_model = GetModel();
 
-        // Avoid sorting while the window is frozen, this allows to quickly add
-        // many items without resorting after each addition and only resort
-        // them all at once when the window is finally thawed, see above.
-        if ( IsFrozen() )
-        {
-            g_column = SortColumn_OnThaw;
-            return;
-        }
-
         wxDataViewColumn* col = GetOwner()->GetSortingColumn();
         if( !col )
         {
             if (g_model->HasDefaultCompare())
-                g_column = SortColumn_Default;
+            {
+                if ( IsFrozen() )
+                    g_column = SortColumn_OnThaw;
+                else
+                    g_column = SortColumn_Default;
+            }
             else
                 g_column = SortColumn_None;
 
             g_asending = true;
             return;
         }
+
+        // Avoid sorting while the window is frozen, this allows to quickly add
+        // many items without resorting after each addition and only resort
+        // them all at once when the window is finally thawed, see above.
+        if ( IsFrozen() )
+        {
+            g_column = SortColumn_OnThaw;
+            return;
+        }
+
         g_column = col->GetModelColumn();
         g_asending = col->IsSortOrderAscending();
     }
