diff -uNr wxWidgets/src/xrc/xh_menu.cpp wxWidgets-2.9.2/src/xrc/xh_menu.cpp
--- wxWidgets/src/xrc/xh_menu.cpp	2010-06-28 05:15:30.000000000 -0400
+++ wxWidgets-2.9.2/src/xrc/xh_menu.cpp	2010-12-02 15:11:46.937635200 -0500
@@ -111,8 +111,8 @@
                 // and unchecked bitmaps for menu items
 #ifdef __WXMSW__
                 if (HasParam(wxT("bitmap2")))
-                    mitem->SetBitmaps(GetBitmap(wxT("bitmap2"), wxART_MENU),
-                                      GetBitmap(wxT("bitmap"), wxART_MENU));
+                    mitem->SetBitmaps(GetBitmap(wxT("bitmap"), wxART_MENU),
+                                      GetBitmap(wxT("bitmap2"), wxART_MENU));
                 else
 #endif // __WXMSW__
                     mitem->SetBitmap(GetBitmap(wxT("bitmap"), wxART_MENU));
