diff -bru wxWidgets.orig/src/osx/cocoa/toolbar.mm wxWidgets/src/osx/cocoa/toolbar.mm
--- wxWidgets.orig/src/osx/cocoa/toolbar.mm	2015-05-09 21:17:00.000000000 -0400
+++ wxWidgets/src/osx/cocoa/toolbar.mm	2015-05-10 12:58:04.000000000 -0400
@@ -202,6 +202,8 @@
         wxToolBar *tbar = (wxToolBar*) GetToolBar();
         if (CanBeToggled())
         {
+            if( IsToggled() )
+                return;
             bool    shouldToggle;
 
             shouldToggle = !IsToggled();
