Index: include/wx/window.h
===================================================================
--- include/wx/window.h	(revision 56079)
+++ include/wx/window.h	(working copy)
@@ -1108,9 +1108,9 @@
 
     // simply return the id of the selected item or wxID_NONE without
     // generating any events
-    int GetPopupMenuSelectionFromUser(wxMenu& menu, const wxPoint& pos)
+    int GetPopupMenuSelectionFromUser(wxMenu* menu, const wxPoint& pos)
         { return DoGetPopupMenuSelectionFromUser(menu, pos.x, pos.y); }
-    int GetPopupMenuSelectionFromUser(wxMenu& menu, int x, int y)
+    int GetPopupMenuSelectionFromUser(wxMenu* menu, int x, int y)
         { return DoGetPopupMenuSelectionFromUser(menu, x, y); }
 #endif // wxUSE_MENUS
 
@@ -1645,7 +1645,7 @@
     void InternalOnPopupMenu(wxCommandEvent& event);
 
     // implementation of the public GetPopupMenuSelectionFromUser() method
-    int DoGetPopupMenuSelectionFromUser(wxMenu& menu, int x, int y);
+    int DoGetPopupMenuSelectionFromUser(wxMenu* menu, int x, int y);
 #endif // wxUSE_MENUS
 
     // the stack of windows which have captured the mouse
Index: interface/wx/window.h
===================================================================
--- interface/wx/window.h	(revision 56079)
+++ interface/wx/window.h	(working copy)
@@ -1079,14 +1079,14 @@
         @return The selected menu item id or wxID_NONE if none selected or an
                  error occurred.
     */
-    int GetPopupMenuSelectionFromUser(wxMenu& menu, const wxPoint& pos);
+    int GetPopupMenuSelectionFromUser(wxMenu* menu, const wxPoint& pos);
 
     /**
         See the GetPopupMenuSelectionFromUser(wxMenu&, const wxPoint&) overload.
         This overload differs only because it takes two integers for the
         menu position instead of a wxPoint.
     */
-    int GetPopupMenuSelectionFromUser(wxMenu& menu, int x, int y);
+    int GetPopupMenuSelectionFromUser(wxMenu* menu, int x, int y);
 
     /**
         This gets the position of the window in pixels, relative to the parent window
Index: src/common/wincmn.cpp
===================================================================
--- src/common/wincmn.cpp	(revision 56079)
+++ src/common/wincmn.cpp	(working copy)
@@ -2407,7 +2407,7 @@
 }
 
 int
-wxWindowBase::DoGetPopupMenuSelectionFromUser(wxMenu& menu, int x, int y)
+wxWindowBase::DoGetPopupMenuSelectionFromUser(wxMenu* menu, int x, int y)
 {
     gs_popupMenuSelection = wxID_NONE;
 
@@ -2416,7 +2416,7 @@
             NULL,
             this);
 
-    PopupMenu(&menu, x, y);
+    PopupMenu(menu, x, y);
 
     Disconnect(wxEVT_COMMAND_MENU_SELECTED,
                wxCommandEventHandler(wxWindowBase::InternalOnPopupMenu),
