 samples/minimal/minimal.cpp | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/samples/minimal/minimal.cpp b/samples/minimal/minimal.cpp
index a78e462..db01d11 100644
--- a/samples/minimal/minimal.cpp
+++ b/samples/minimal/minimal.cpp
@@ -172,6 +172,16 @@ MyFrame::MyFrame(const wxString& title)
     CreateStatusBar(2);
     SetStatusText("Welcome to wxWidgets!");
 #endif // wxUSE_STATUSBAR
+
+    wxButton* btn1 = new wxButton(this, -1, "LeftAligned text, default colours", wxDefaultPosition, wxDefaultSize, wxBU_LEFT);
+    wxButton* btn2 = new wxButton(this, -1, "LeftAligned text, custom colours", wxDefaultPosition, wxDefaultSize, wxBU_LEFT);
+    btn2->SetBackgroundColour(*wxRED);
+    btn2->SetForegroundColour(*wxGREEN);
+
+    wxSizer* sz = new wxBoxSizer(wxVERTICAL);
+    sz->Add(btn1, 1, wxEXPAND);
+    sz->Add(btn2, 1, wxEXPAND);
+    SetSizer(sz);
 }
 
 
