Index: include/wx/docview.h
===================================================================
--- include/wx/docview.h	(revision 56351)
+++ include/wx/docview.h	(working copy)
@@ -100,6 +100,10 @@
     // modified to false)
     virtual bool OnSaveModified();
 
+    // if you override, remember to call the default 
+    // implementation (wxDocument::OnChangeFilename)
+    virtual void OnChangeFilename(bool notifyViews);
+
     // Called by framework if created automatically by the default document
     // manager: gives document a chance to initialise and (usually) create a
     // view
Index: interface/wx/docview.h
===================================================================
--- interface/wx/docview.h	(revision 56351)
+++ interface/wx/docview.h	(working copy)
@@ -1246,10 +1246,18 @@
     /**
         Sets the filename for this document. Usually called by the framework.
 
+        Calls OnChangeFilename which in turn calls wxView::OnChangeFilename() for
+        all views if @a notifyViews is @true, 
+    */
+    void SetFilename(const wxString& filename, bool notifyViews = false);
+
+    /**
         If @a notifyViews is @true, wxView::OnChangeFilename() is called for
         all views.
+        
+        @since 2.9.0
     */
-    void SetFilename(const wxString& filename, bool notifyViews = false);
+    virtual void OnChangeFilename(bool notifyViews);
 
     /**
         Sets the title for this document. The document title is used for an
Index: src/common/docview.cpp
===================================================================
--- src/common/docview.cpp	(revision 56351)
+++ src/common/docview.cpp	(working copy)
@@ -565,6 +565,11 @@
 void wxDocument::SetFilename(const wxString& filename, bool notifyViews)
 {
     m_documentFile = filename;
+    OnChangeFilename(notifyViews);
+}
+
+void wxDocument::OnChangeFilename(bool notifyViews)
+{
     if ( notifyViews )
     {
         // Notify the views that the filename has changed
