diff -ru wxWidgets.orig/src/msw/notebook.cpp wxWidgets/src/msw/notebook.cpp
--- wxWidgets.orig/src/msw/notebook.cpp	2007-05-23 15:46:59.000000000 +0200
+++ wxWidgets/src/msw/notebook.cpp	2007-05-23 16:23:03.443761200 +0200
@@ -287,7 +287,7 @@
     // work around it later (after control creation), but if we don't have uxtheme, we have to clear
     // those styles
     const int verComCtl32 = wxApp::GetComCtl32Version();
-    if ( verComCtl32 == 600 )
+    if ( verComCtl32 >= 600 )
     {
         style &= ~(wxBK_BOTTOM | wxBK_LEFT | wxBK_RIGHT);
     }
@@ -363,7 +363,7 @@
     // It's probably not possible to have UXTHEME without ComCtl32 6 or better, but lets
     // check it anyway.
     const int verComCtl32 = wxApp::GetComCtl32Version();
-    if ( verComCtl32 == 600 )
+    if ( verComCtl32 >= 600 )
     {
         // check if we use themes at all -- if we don't, we're still okay
         if ( wxUxThemeEngine::GetIfActive() && (style & (wxBK_BOTTOM|wxBK_LEFT|wxBK_RIGHT)))

 	  	 
