Index: samples/image/image.cpp
===================================================================
--- samples/image/image.cpp	(revision 60078)
+++ samples/image/image.cpp	(working copy)
@@ -181,12 +181,25 @@
         // do nothing here to be able to see how transparent images are shown
     }
 
+    void ClearBackground(wxDC* dc)
+    {
+       // base class implementation clears everything including non-invalidated area, looks bad:
+       //wxWindowBase::ClearBackground(); return;
+      
+       // clears only the invalidated area, looks good:
+#ifndef __WXGTK__
+       wxBrush brush(GetBackgroundColour(), wxSOLID);
+       dc->SetBackground(brush);
+       dc->Clear();
+   #endif // __WXGTK__
+    }
+
     void OnPaint(wxPaintEvent& WXUNUSED(event))
     {
         wxPaintDC dc(this);
 
         if ( GetMenuBar()->IsChecked(ID_PAINT_BG) )
-            ClearBackground();
+            ClearBackground(&dc);
 
         dc.SetUserScale(m_zoom, m_zoom);
 
