Index: samples/combo/combo.cpp
===================================================================
--- samples/combo/combo.cpp	(revision 53959)
+++ samples/combo/combo.cpp	(working copy)
@@ -872,6 +872,21 @@
 
     rowSizer->Add( cc, 1, wxALIGN_CENTER_VERTICAL|wxALL, 5 );
 
+	// -----------------------------------------------------------
+	// disable the cc's parent
+	Disable();
+	// disable the cc
+	cc->Disable();
+	// re-enable the cc's parent
+	// this also enables the cc
+	// 1. I think this is wrong. cc should remain disabled
+	//    and wx 2.8 and earlier did indeed _not_ enable cc.
+	//    The new behaviour at least breaks existing code.
+	// 2. The cc visual cues are still set to disabled.
+	//	  The text control and the button are grayed out,
+	//	  but they accept input.
+	Enable();
+	// -----------------------------------------------------------
 
     //
     // Tree Ctrl wxComboCtrl
