diff -rbu2 wxWidgets-2007-01-07/src/gtk/clipbrd.cpp wxWidgets-2007-01-07-patch/src/gtk/clipbrd.cpp
--- wxWidgets-2007-01-07/src/gtk/clipbrd.cpp	Sun Jan  7 05:58:44 2007
+++ wxWidgets-2007-01-07-patch/src/gtk/clipbrd.cpp	Sun Jan  7 20:55:19 2007
@@ -631,6 +631,19 @@
         while (m_waiting) gtk_main_iteration();
 
-        /* this is a true error as we checked for the presence of such data before */
+        /* Normally this is a true error as we checked for the presence of such data before, but
+           there are applications that may return an empty string (e.g. Gnumeric-1.6.1 on Linux
+           if an empty cell is copied) which would produce a false error message here, so we check
+           for the size of the string first.
+           In ansi, GetDataSize returns an extra value (for the closing null?),
+           with unicode, the exact number of tokens is given (that is more than 1 for
+           special characters) - tested with Gnumeric-1.6.1 and OpenOffice.org-2.0.2 */
+#if wxUSE_UNICODE
+        if ( format != wxDF_UNICODETEXT || data.GetDataSize(format) > 0 )
+#else // !UNICODE
+        if ( format != wxDF_TEXT || data.GetDataSize(format) > 1 )
+#endif // UNICODE / !UNICODE
+        {
         wxCHECK_MSG( m_formatSupported, false, wxT("error retrieving data from clipboard") );
+        }
 
         /* return success */
diff -rbu2 wxWidgets-2007-01-07/src/gtk1/clipbrd.cpp wxWidgets-2007-01-07-patch/src/gtk1/clipbrd.cpp
--- wxWidgets-2007-01-07/src/gtk1/clipbrd.cpp	Sun Jan  7 05:58:52 2007
+++ wxWidgets-2007-01-07-patch/src/gtk1/clipbrd.cpp	Sun Jan  7 20:58:37 2007
@@ -620,6 +620,19 @@
         while (m_waiting) gtk_main_iteration();
 
-        /* this is a true error as we checked for the presence of such data before */
+        /* Normally this is a true error as we checked for the presence of such data before, but
+           there are applications that may return an empty string (e.g. Gnumeric-1.6.1 on Linux
+           if an empty cell is copied) which would produce a false error message here, so we check
+           for the size of the string first.
+           In ansi, GetDataSize returns an extra value (for the closing null?),
+           with unicode, the exact number of tokens is given (that is more than 1 for
+           special characters) - tested with Gnumeric-1.6.1 and OpenOffice.org-2.0.2 */
+#if wxUSE_UNICODE
+        if ( format != wxDF_UNICODETEXT || data.GetDataSize(format) > 0 )
+#else // !UNICODE
+        if ( format != wxDF_TEXT || data.GetDataSize(format) > 1 )
+#endif // UNICODE / !UNICODE
+        {
         wxCHECK_MSG( m_formatSupported, false, wxT("error retrieving data from clipboard") );
+        }
 
         /* return success */

 	  	 
