From 9fdc8e410f203e9b3db6c957e77c9a00a5625b3d Mon Sep 17 00:00:00 2001
From: Marian 'VooDooMan' Meravy <vdm113@gmail.com>
Date: Thu, 3 Oct 2013 13:29:09 +0200
Subject: [PATCH] include Intel compiler (ICC) into IgnoreAppBuildSubDirs()
 function

Signed-off-by: Marian 'VooDooMan' Meravy <vdm113@gmail.com>
---
 src/msw/stdpaths.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/msw/stdpaths.cpp b/src/msw/stdpaths.cpp
index f8b0586..82c4ad0 100644
--- a/src/msw/stdpaths.cpp
+++ b/src/msw/stdpaths.cpp
@@ -306,6 +306,8 @@ void wxStandardPaths::IgnoreAppBuildSubDirs()
     compilerPrefix = "dmc";
 #elif defined(__WATCOMC__)
     compilerPrefix = "wat";
+#elif defined(__INTELC__)
+    compilerPrefix = "icc";
 #else
     return;
 #endif
-- 
1.8.3.msysgit.0

