Index: src/gtk/app.cpp
===================================================================
--- src/gtk/app.cpp	(revision 52444)
+++ src/gtk/app.cpp	(working copy)
@@ -224,15 +224,32 @@
 
     if (!moreIdles)
     {
+        // If pending events have been added meanwhile, set moreIdles to
+        // true to prevent idle processing and as such pending event
+        // processing from stopping.
 #if wxUSE_THREADS
         wxMutexLocker lock(gs_idleTagsMutex);
 #endif
-        // Indicate that we are now in idle mode and event handlers
-        // will have to reinstall the idle handler again.
-        g_isIdle = true;
-        wxTheApp->m_idleTag = 0;
 
-        wxAddEmissionHook();
+#if wxUSE_THREADS
+        wxPendingEventsLocker->Enter();
+#endif
+        const bool pending = wxPendingEvents && wxPendingEvents->GetFirst();
+#if wxUSE_THREADS
+        wxPendingEventsLocker->Leave();
+#endif
+
+        if (pending)
+            moreIdles = true;
+        else
+        {
+            // Indicate that we are now in idle mode and event handlers
+            // will have to reinstall the idle handler again.
+            g_isIdle = true;
+            wxTheApp->m_idleTag = 0;
+
+            wxAddEmissionHook();
+        }
     }
 
     // Return FALSE if no more idle events are to be sent

 	  	 
