Index: src/mac/carbon/renderer.cpp
===================================================================
--- src/mac/carbon/renderer.cpp	(revision 49211)
+++ src/mac/carbon/renderer.cpp	(working copy)
@@ -57,6 +57,11 @@
         wxOrientation orient,
         int flags = 0 );
 
+    virtual void DrawCheckBox(wxWindow *win,
+                              wxDC& dc,
+                              const wxRect& rect,
+                              int flags = 0);
+
     virtual void DrawComboBoxDropButton(wxWindow *win,
                                         wxDC& dc,
                                         const wxRect& rect,
@@ -459,6 +464,8 @@
             drawInfo.kind = kind;
             drawInfo.state = (flags & wxCONTROL_DISABLED) ? kThemeStateInactive : kThemeStateActive;
             drawInfo.value = (flags & wxCONTROL_SELECTED) ? kThemeButtonOn : kThemeButtonOff;
+            if (flags & wxCONTROL_UNDETERMINED)
+                drawInfo.value = kThemeButtonMixed;
             drawInfo.adornment = adornment;
 
             HIThemeDrawButton( &headerRect, &drawInfo, cgContext, kHIThemeOrientationNormal, &labelRect );
@@ -471,7 +478,19 @@
     }
 }
 
+void
+wxRendererMac::DrawCheckBox(wxWindow *win,
+                            wxDC& dc,
+                            const wxRect& rect,
+                            int flags)
+{
+    if (flags & wxCONTROL_CHECKED)
+        flags |= wxCONTROL_SELECTED;
 
+    DrawMacThemeButton(win, dc, rect, flags,
+                       kThemeCheckBox, kThemeAdornmentNone);
+}
+
 void
 wxRendererMac::DrawComboBoxDropButton(wxWindow *win,
                               wxDC& dc,
Index: include/wx/mac/carbon/icon.h
===================================================================
--- include/wx/mac/carbon/icon.h	(revision 49211)
+++ include/wx/mac/carbon/icon.h	(working copy)
@@ -29,6 +29,7 @@
     {
       LoadFile(loc.GetFileName(), wxBITMAP_TYPE_ICON);
     }
+    wxIcon(WXHICON icon, const wxSize& size);
     virtual ~wxIcon();
 
     bool LoadFile(const wxString& name, wxBitmapType flags /* = wxBITMAP_TYPE_ICON_RESOURCE */ ,
Index: samples/render/render.cpp
===================================================================
--- samples/render/render.cpp	(revision 49211)
+++ samples/render/render.cpp	(working copy)
@@ -135,6 +135,20 @@
 
         wxRendererNative::Get().DrawHeaderButton(this, dc,
                                                  wxRect(20, 70, 100, 60));
+
+        // Draw some check boxes in various states
+        dc.SetBrush(*wxBLACK_BRUSH);
+        dc.SetTextForeground(*wxBLACK);
+        dc.DrawText(_T("Checkbox Drawn with native renderer"), 10, 150);
+        // Checked
+        wxRendererNative::Get().DrawCheckBox(this, dc, wxRect(20, 170, 16, 16), wxCONTROL_CHECKED);
+        // Undetermined
+        wxRendererNative::Get().DrawCheckBox(this, dc, wxRect(40, 170, 16, 16), wxCONTROL_CHECKABLE);
+        // Unchecked
+        wxRendererNative::Get().DrawCheckBox(this, dc, wxRect(60, 170, 16, 16), 0);
+        // Checked and Disabled
+        wxRendererNative::Get().DrawCheckBox(this, dc, wxRect(80, 170, 16, 16), wxCONTROL_CHECKED | wxCONTROL_DISABLED);
+
     }
 
     DECLARE_EVENT_TABLE()

 	  	 
