Index: framemanager.cpp
===================================================================
--- framemanager.cpp	(revision 62709)
+++ framemanager.cpp	(working copy)
@@ -706,7 +706,6 @@
     EVT_MOUSE_CAPTURE_LOST(wxAuiManager::OnCaptureLost)
     EVT_CHILD_FOCUS(wxAuiManager::OnChildFocus)
     EVT_AUI_FIND_MANAGER(wxAuiManager::OnFindManager)
-    EVT_TIMER(101, wxAuiManager::OnHintFadeTimer)
 END_EVENT_TABLE()
 
 
@@ -3353,9 +3352,11 @@
 
         if (m_hint_fadeamt != m_hint_fademax) //  Only fade if we need to
         {
-            // start fade in timer
-            m_hint_fadetimer.SetOwner(this, 101);
+			if (m_hint_fadetimer.GetId() != wxID_ANY)
+				Disconnect(m_hint_fadetimer.GetId());
+            m_hint_fadetimer.SetOwner(this, wxID_ANY);
             m_hint_fadetimer.Start(5);
+			Connect(m_hint_fadetimer.GetId(), wxEVT_TIMER, wxTimerEventHandler(wxAuiManager::OnHintFadeTimer));
         }
     }
     else  // Not using a transparent hint window...
