Index: src/msw/main.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/msw/main.cpp,v
retrieving revision 1.65
diff -u -r1.65 main.cpp
--- src/msw/main.cpp	2006/12/09 14:38:41	1.65
+++ src/msw/main.cpp	2007/03/06 16:31:50
@@ -27,6 +27,7 @@
 #ifndef WX_PRECOMP
     #include "wx/event.h"
     #include "wx/app.h"
+    #include "wx/utils.h"
 #endif //WX_PRECOMP
 
 #include "wx/cmdline.h"

 	  	 
