Index: interface/wx/accel.h
===================================================================
--- interface/wx/accel.h	(revision 71201)
+++ interface/wx/accel.h	(working copy)
@@ -115,6 +115,15 @@
     wxString ToString() const;
 
     /**
+        Returns a textual representation of this accelerator which is
+        appropriate for saving in configuration files. The string can then be
+        parsed with FromString() when reloading.
+
+        @since 2.9.4
+    */
+    wxString ToRawString() const;
+
+    /**
         Parses the given string and sets the accelerator accordingly.
 
         @param str
