Index: samples/sockets/baseclient.cpp
===================================================================
--- samples/sockets/baseclient.cpp	(revision 56067)
+++ samples/sockets/baseclient.cpp	(working copy)
@@ -254,13 +254,14 @@
 int
 Client::OnRun()
 {
+    int i;
     switch(m_sendType)
     {
         case STRESS_TEST:
             switch(m_workMode)
             {
                 case THREADS:
-                    for (int i = 0; i < m_stressWorkers; i++) {
+                    for (i = 0; i < m_stressWorkers; i++) {
                         if (m_message.empty())
                             StartWorker(THREADS);
                         else
@@ -268,7 +269,7 @@
                     }
                     break;
                 case EVENTS:
-                    for (int i = 0; i < m_stressWorkers; i++) {
+                    for (i = 0; i < m_stressWorkers; i++) {
                         if (m_message.empty())
                             StartWorker(EVENTS);
                         else
@@ -276,7 +277,7 @@
                     }
                     break;
                 default:
-                    for (int i = 0; i < m_stressWorkers; i++) {
+                    for (i = 0; i < m_stressWorkers; i++) {
                         if (m_message.empty())
                             StartWorker(i % 5 == 0 ? THREADS : EVENTS);
                         else
@@ -450,11 +451,11 @@
                 break;
             }
         }
-        for(EList::compatibility_iterator it = m_eventWorkers.GetFirst(); it ; it = it->GetNext())
+        for(EList::compatibility_iterator it2 = m_eventWorkers.GetFirst(); it2 ; it2 = it2->GetNext())
         {
-            if (it->GetData() == pEvent.m_sender) {
-                delete it->GetData();
-                m_eventWorkers.DeleteNode(it);
+            if (it2->GetData() == pEvent.m_sender) {
+                delete it2->GetData();
+                m_eventWorkers.DeleteNode(it2);
                 break;
             }
         }
Index: samples/sockets/baseserver.cpp
===================================================================
--- samples/sockets/baseserver.cpp	(revision 56067)
+++ samples/sockets/baseserver.cpp	(working copy)
@@ -342,8 +342,8 @@
         delete it->GetData();
     }
 
-    for(EList::compatibility_iterator it = m_eventWorkers.GetFirst(); it ; it->GetNext()) {
-        delete it->GetData();
+    for(EList::compatibility_iterator it2 = m_eventWorkers.GetFirst(); it2 ; it2->GetNext()) {
+        delete it2->GetData();
     }
 
     m_threadWorkers.Clear();
@@ -429,13 +429,13 @@
             break;
         }
     }
-    for(EList::compatibility_iterator it = m_eventWorkers.GetFirst(); it ; it = it->GetNext())
+    for(EList::compatibility_iterator it2 = m_eventWorkers.GetFirst(); it2 ; it2 = it2->GetNext())
     {
-        if (it->GetData() == pEvent.m_sender)
+        if (it2->GetData() == pEvent.m_sender)
         {
             wxLogVerbose(wxT("Deleting event worker (%d left)"),m_eventWorkers.GetCount());
-            delete it->GetData();
-            m_eventWorkers.DeleteNode(it);
+            delete it2->GetData();
+            m_eventWorkers.DeleteNode(it2);
             if (!pEvent.m_workerFailed)
                 m_eventWorkersDone++;
             else
