Index: include/wx/app.h
===================================================================
--- include/wx/app.h	(révision 7388)
+++ include/wx/app.h	(révision 7389)
@@ -810,7 +810,7 @@
 // almost always used and there is no way to retrieve the Unicode command line
 // anyhow).
 #if wxUSE_UNICODE && defined(__WINDOWS__)
-    #ifdef __VISUALC__
+    #if defined(__VISUALC__) || defined( __MINGW32__) 
         #define wxIMPLEMENT_WXWIN_MAIN_CONSOLE                                \
             int wmain(int argc, wchar_t **argv)                               \
             {                                                                 \
