 samples/minimal/minimal.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/samples/minimal/minimal.cpp b/samples/minimal/minimal.cpp
index a78e462..f79d02c 100644
--- a/samples/minimal/minimal.cpp
+++ b/samples/minimal/minimal.cpp
@@ -172,6 +172,8 @@ MyFrame::MyFrame(const wxString& title)
     CreateStatusBar(2);
     SetStatusText("Welcome to wxWidgets!");
 #endif // wxUSE_STATUSBAR
+
+    SetToolTip("Very loooooooooooooooooooooooooooooooooooooong tttttttttttttttttooooooooooooooooooooooooooooooooollllllllllllttttiiiiiiiiiiiiiiiipppppppppppppppppppp");
 }
 
 
