Index: samples/minimal/minimal.cpp
===================================================================
--- samples/minimal/minimal.cpp	(revision 63733)
+++ samples/minimal/minimal.cpp	(working copy)
@@ -16,10 +16,10 @@
 // ----------------------------------------------------------------------------
 // headers
 // ----------------------------------------------------------------------------
- 
+
 // For compilers that support precompilation, includes "wx/wx.h".
 #include "wx/wxprec.h"
- 
+
 #ifdef __BORLANDC__
     #pragma hdrstop
 #endif
@@ -116,9 +116,26 @@
 // the application class
 // ----------------------------------------------------------------------------
 
+class Foo;
+
+class Bar
+{
+    public:
+    wxWeakRef<Foo> wr;
+};
+
+class Foo : public wxPanel
+{
+    public:
+    Bar obj;
+};
+
 // 'Main program' equivalent: the program execution "starts" here
 bool MyApp::OnInit()
 {
+    Foo fooVar;
+    fooVar.obj.wr = new Foo();  // fails to compile
+
     // call the base class initialization method, currently it only parses a
     // few common command-line options but it could be do more in the future
     if ( !wxApp::OnInit() )
