Index: src/gtk/dnd.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/gtk/dnd.cpp,v
retrieving revision 1.89
diff -b -u -2 -r1.89 dnd.cpp
--- src/gtk/dnd.cpp	2004/05/23 20:52:20	1.89
+++ src/gtk/dnd.cpp	2005/01/11 17:52:23
@@ -221,5 +221,7 @@
     {
         GdkDragAction action;
-        if (result == wxDragCopy)
+        if ((result == wxDragCopy) && (context->actions & GDK_ACTION_COPY) ||
+            (result == wxDragMove) && !(context->actions & GDK_ACTION_MOVE) ||
+            (result == wxDragLink) && !(context->actions & GDK_ACTION_LINK))
             action = GDK_ACTION_COPY;
         else if (result == wxDragLink)
@@ -372,5 +374,5 @@
     drop_target->SetDragData( data );
 
-    wxDragResult result = ConvertFromGTK(context->suggested_action);
+    wxDragResult result = ConvertFromGTK(context->action);
 
     if ( wxIsDragResultOk( drop_target->OnData( x, y, result ) ) )

 	  	 
