Index: tests/graphics/dc.cpp
===================================================================
--- tests/graphics/dc.cpp	(revision 0)
+++ tests/graphics/dc.cpp	(revision 0)
@@ -0,0 +1,107 @@
+///////////////////////////////////////////////////////////////////////////////
+// Name:        tests/graphics/dc.cpp
+// Purpose:     wxDC unit test
+// Author:      Vadim Zeitlin / Maarten Spoek
+// Created:     2011-01-36
+// RCS-ID:      $Id$
+// Copyright:   (c) 2011 Vadim Zeitlin <vadim@wxwidgets.org>
+///////////////////////////////////////////////////////////////////////////////
+
+// ----------------------------------------------------------------------------
+// headers
+// ----------------------------------------------------------------------------
+
+#include "testprec.h"
+
+#ifdef __BORLANDC__
+    #pragma hdrstop
+#endif
+
+#include "wx/bitmap.h"
+#include "wx/dcmemory.h"
+
+// ----------------------------------------------------------------------------
+// test class
+// ----------------------------------------------------------------------------
+
+class DCTestCase : public CppUnit::TestCase
+{
+public:
+    DCTestCase() { }
+
+    virtual void setUp();
+    virtual void tearDown();
+
+private:
+    CPPUNIT_TEST_SUITE( DCTestCase );
+        CPPUNIT_TEST( BoundingBox );
+    CPPUNIT_TEST_SUITE_END();
+
+    void BoundingBox();
+    void GetPixel();
+
+    wxBitmap m_bmp;
+	wxMemoryDC m_dc;
+
+    DECLARE_NO_COPY_CLASS(DCTestCase)
+};
+
+// register in the unnamed registry so that these tests are run by default
+CPPUNIT_TEST_SUITE_REGISTRATION( DCTestCase );
+
+// also include in it's own registry so that these tests can be run alone
+CPPUNIT_TEST_SUITE_NAMED_REGISTRATION( DCTestCase, "DCTestCase" );
+
+const int DIM = 100;
+
+void DCTestCase::setUp()
+{
+    m_bmp.Create(DIM, DIM);
+
+    m_dc.SelectObject(m_bmp);;
+    m_dc.SetBackground(*wxWHITE);
+    m_dc.Clear();
+}
+
+void DCTestCase::tearDown()
+{
+    m_dc.SelectObject(wxNullBitmap);
+    m_bmp = wxNullBitmap;
+}
+
+void DCTestCase::BoundingBox()
+{
+    m_dc.SetFont(wxFont(18, wxFONTFAMILY_DEFAULT, wxFONTSTYLE_NORMAL, wxFONTWEIGHT_NORMAL));
+    m_dc.SetTextBackground(* wxWHITE);
+    m_dc.SetTextForeground(* wxBLACK);
+    m_dc.DrawText("H", 0, 0);
+    int height = m_dc.MaxY(); 
+    // Should be greater than 0
+    CPPUNIT_ASSERT(height > 0);
+}
+
+void DCTestCase::GetPixel()
+{
+    BoundingBox(); // Draw a character.
+    
+    bool pixfound = false;
+    
+    for (int y = 0; (y < DIM) && !pixfound; y++) {
+        for (int x = 0; x < DIM; x++) {
+            wxColour testclr; 
+ 
+            if (m_dc.GetPixel(x, y, &testclr)) {
+                if (testclr != *wxWHITE) {
+                    // Got a drawn pixel.
+                    pixfound = true;
+                    break;
+                }
+            } else {
+                CPPUNIT_ASSERT("GetPixel failed");
+            }
+        }
+    }
+    
+    CPPUNIT_ASSERT(pixfound); // A pixel other than white should been found.
+}
+
