--- ../wxWidgets/src/common/filename.cpp	2005-12-11 09:10:46.000000000 -0800
+++ ./src/common/filename.cpp	2005-12-13 23:04:07.000000000 -0800
@@ -654,10 +654,14 @@
 #if defined(__WXMAC__) && !defined(__DARWIN__)
         dir = wxMacFindFolder(  (short) kOnSystemDisk, kTemporaryFolderType, kCreateFolder ) ;
 #else // !Mac
-        dir = wxGetenv(_T("TMP"));
+        dir = wxGetenv(_T("TMPDIR"));
         if ( dir.empty() )
         {
-            dir = wxGetenv(_T("TEMP"));
+            dir = wxGetenv(_T("TMP"));
+            if (dir.empty())
+            {
+                dir = wxGetenv(_T("TEMP"));
+            }
         }
 
         if ( dir.empty() )

 	  	 
