Index: activex.cpp
===================================================================
--- activex.cpp	(revision 69753)
+++ activex.cpp	(working copy)
@@ -1177,8 +1177,9 @@
     posRect.top = 0;
     posRect.right = w;
     posRect.bottom = h;
-
-    if (w <= 0 && h <= 0)
+    //if width OR height are <= 0 then there is no need
+    //to actually execute the code below since the control is not visible anyway.
+    if (w <= 0 || h <= 0)
         return;
 
     // extents are in HIMETRIC units
