diff -uNrb wxWidgets-trunk\src/common/filefn.cpp wxWidgets-work\src/common/filefn.cpp
--- wxWidgets-trunk\src/common/filefn.cpp	2014-01-04
+++ wxWidgets-work\src/common/filefn.cpp	2014-02-22
@@ -38,6 +38,7 @@
 #include "wx/dir.h"
 
 #include "wx/tokenzr.h"
+#include "wx/sharedptr.h"
 
 // there are just too many of those...
 #ifdef __VISUALC__
@@ -1309,7 +1310,7 @@
 
 // Get first file name matching given wild card.
 
-static wxDir *gs_dir = NULL;
+static wxSharedPtr<wxDir> gs_dir;
 static wxString gs_dirPath;
 
 wxString wxFindFirstFile(const wxString& spec, int flags)
@@ -1320,7 +1321,6 @@
     if ( !wxEndsWithPathSeparator(gs_dirPath ) )
         gs_dirPath << wxFILE_SEP_PATH;
 
-    delete gs_dir; // can be NULL, this is ok
     gs_dir = new wxDir(gs_dirPath);
 
     if ( !gs_dir->IsOpened() )
@@ -1341,7 +1341,7 @@
     gs_dir->GetFirst(&result, wxFileNameFromPath(spec), dirFlags);
     if ( result.empty() )
     {
-        wxDELETE(gs_dir);
+        gs_dir.reset();
         return result;
     }
 
@@ -1355,7 +1355,7 @@
     wxString result;
     if ( !gs_dir->GetNext(&result) || result.empty() )
     {
-        wxDELETE(gs_dir);
+        gs_dir.reset();
         return result;
     }
 
