fix compilation error introduced by r54926
---
Index: wxWidgets/src/common/utilscmn.cpp
===================================================================
--- wxWidgets.orig/src/common/utilscmn.cpp	2008-08-13 12:05:01.000000000 +0200
+++ wxWidgets/src/common/utilscmn.cpp	2008-08-13 13:03:33.000000000 +0200
@@ -964,7 +964,7 @@
                                       (
                                         NULL,           // parent window
                                         _T("open"),
-                                        document,
+                                        document.c_str(),
                                         NULL,           // parameters
                                         NULL,           // working directory
                                         SW_SHOWDEFAULT
