Index: src/osx/carbon/nonownedwnd.cpp
===================================================================
--- src/osx/carbon/nonownedwnd.cpp	(revision 55520)
+++ src/osx/carbon/nonownedwnd.cpp	(working copy)
@@ -837,19 +837,22 @@
             if ( (attributes & kWindowBoundsChangeSizeChanged) || (attributes & kWindowBoundsChangeOriginChanged) )
             {
                 // all (Mac) rects are in content area coordinates, all wxRects in structure coordinates
-                int left , top , right , bottom ;
+                int left , top , width , height, dwidth, dheight ;
 
-                toplevelWindow->GetNonOwnedPeer()->GetContentArea(left, top, right, bottom);
+                toplevelWindow->GetNonOwnedPeer()->GetContentArea(left, top, width, height);
+                toplevelWindow->GetNonOwnedPeer()->GetSize(dwidth, dheight);
+                dwidth -= width;
+                dheight -= height;
 
                 wxRect adjustR(
-                    newRect.left - left,
-                    newRect.top - top,
-                    newRect.right - newRect.left + left + right,
-                    newRect.bottom - newRect.top + top + bottom ) ;
+                    newRect.left,
+                    newRect.top,
+                    newRect.right - newRect.left + dwidth,
+                    newRect.bottom - newRect.top + dheight);
 
                 toplevelWindow->HandleResizing( cEvent.GetTicks(), &adjustR );
                 
-                const Rect adjustedRect = { adjustR.y + top  , adjustR.x + left , adjustR.y + adjustR.height - bottom , adjustR.x + adjustR.width - right } ;
+                const Rect adjustedRect = { adjustR.y, adjustR.x, adjustR.y + adjustR.height - dheight, adjustR.x + adjustR.width - dwidth } ;
                 if ( !EqualRect( &newRect , &adjustedRect ) )
                     cEvent.SetParameter<Rect>( kEventParamCurrentBounds , &adjustedRect ) ;
             }
