Index: docs/wine/install.txt
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/docs/wine/install.txt,v
retrieving revision 1.6
diff -u -r1.6 install.txt
--- docs/wine/install.txt	2002/08/31 12:08:01	1.6
+++ docs/wine/install.txt	2003/11/26 15:41:43
@@ -1,33 +1,4 @@
 
-  !!! When sending bug reports tell us what version of wxWindows you are 
-      using (including the beta) and what compiler on what system. One 
-      example: wxWINE 2.1 snapshot 6, egcs 1.1.1, Redhat 5.0 !!!
-
-* Preparing WINE
-----------------
-
-Most C++ compilers cannot compile the WINE sources yet (this includes
-all versions of g++ and egcs) so you have to make a minimal change
-and recompile all of WINE in order to get anywhere.
-
-(Note: this information may be out of date now. See notes at
-the bottom of this file.)
-
-This change has do be applied to the /include/windef.h file, line 59
-in the section "Calling convention definitions", where the sources
-reads:
-
-#if __i386__
-
-which has to be changed into
-
-#if 0
-
-After this change, you'll have to recompile all of WINE without
-forgetting to install it. The direct consequence of this change
-is that the Win32 binary emulator won't work anymore, as the
-change disables the Windows native calling convention.
-
 * The most simple case
 -----------------------
 
Index: docs/wine/todo.txt
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/docs/wine/todo.txt,v
retrieving revision 1.1
diff -u -r1.1 todo.txt
--- docs/wine/todo.txt	1999/06/16 16:34:29	1.1
+++ docs/wine/todo.txt	2003/11/26 15:41:43
@@ -1,10 +1,11 @@
 
 -------------------- High priority ---------------------
 
-Finish WINE.
 
 -------------------- Medium priority ---------------------
 
+Fix the configure script to work with Wine.
   
 -------------------- Low priority ---------------------
     
Index: include/wx/msw/chkconf.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/msw/chkconf.h,v
retrieving revision 1.3
diff -u -r1.3 chkconf.h
--- include/wx/msw/chkconf.h	2003/08/15 10:17:02	1.3
+++ include/wx/msw/chkconf.h	2003/11/26 15:41:45
@@ -26,7 +26,7 @@
 #endif
 
 #ifndef wxUSE_NORLANDER_HEADERS
-#if (defined(__WATCOMC__) && (__WATCOMC__ >= 1200)) || ((defined(__MINGW32__) || defined(__CYGWIN__)) && ((__GNUC__>2) ||((__GNUC__==2) && (__GNUC_MINOR__>=95))))
+#if (defined(__WATCOMC__) && (__WATCOMC__ >= 1200)) || defined(__WINE__) || ((defined(__MINGW32__) || defined(__CYGWIN__)) && ((__GNUC__>2) ||((__GNUC__==2) && (__GNUC_MINOR__>=95))))
 #   define wxUSE_NORLANDER_HEADERS 1
 #else
 #   define wxUSE_NORLANDER_HEADERS 0
Index: src/common/imagjpeg.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/common/imagjpeg.cpp,v
retrieving revision 1.38
diff -u -r1.38 imagjpeg.cpp
--- src/common/imagjpeg.cpp	2003/10/31 15:58:58	1.38
+++ src/common/imagjpeg.cpp	2003/11/26 15:41:59
@@ -40,6 +40,9 @@
 
 extern "C"
 {
+    #if defined(__WXMSW__)
+        #define XMD_H
+    #endif
     #include "jpeglib.h"
 }
 
Index: src/common/stopwatch.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/common/stopwatch.cpp,v
retrieving revision 1.5
diff -u -r1.5 stopwatch.cpp
--- src/common/stopwatch.cpp	2003/08/14 18:15:24	1.5
+++ src/common/stopwatch.cpp	2003/11/26 15:41:59
@@ -250,7 +250,7 @@
     // If possible, use a function which avoids conversions from
     // broken-up time structures to milliseconds
 
-#if defined(__WXMSW__) && defined(__MWERKS__)
+#if defined(__WXMSW__) && (defined(__WINE__) || defined(__MWERKS__))
     // This should probably be the way all WXMSW compilers should do it
     // Go direct to the OS for time
 

 	  	 
