? mypatch.patch
? mypatch1.patch
Index: src/mac/carbon/toolbar.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/mac/carbon/toolbar.cpp,v
retrieving revision 1.118
diff -u -r1.118 toolbar.cpp
--- src/mac/carbon/toolbar.cpp	2007/02/06 21:12:53	1.118
+++ src/mac/carbon/toolbar.cpp	2007/02/16 01:20:29
@@ -1390,7 +1390,7 @@
         tool->UpdateToggleImage( toggle );
 }
 
-bool wxToolBar::DoInsertTool(size_t WXUNUSED(pos), wxToolBarToolBase *toolBase)
+bool wxToolBar::DoInsertTool(size_t pos, wxToolBarToolBase *toolBase)
 {
     wxToolBarTool *tool = wx_static_cast( wxToolBarTool*, toolBase );
     if (tool == NULL)
@@ -1503,6 +1503,10 @@
                 if (err  == noErr)
                 {
                     tool->SetToolbarItemRef( item );
+                    err = HIToolbarInsertItemAtIndex(
+                        (HIToolbarRef) m_macHIToolbarRef,
+                        item,
+                        pos);
                 }
                 CFRelease( data ) ;
            }

 	  	 
