diff --git "a/D:\\DOCUME~1\\bandre\\LOCALS~1\\Temp\\gbsF158.tmp\\gbsizer-432e9d0-left.cpp" "b/D:\\DOCUME~1\\bandre\\LOCALS~1\\Temp\\gbsF157.tmp\\gbsizer-858478b-right.cpp"
index 773c0f0..143251e 100644
--- "a/D:\\DOCUME~1\\bandre\\LOCALS~1\\Temp\\gbsF158.tmp\\gbsizer-432e9d0-left.cpp"
+++ "b/D:\\DOCUME~1\\bandre\\LOCALS~1\\Temp\\gbsF157.tmp\\gbsizer-858478b-right.cpp"
@@ -519,6 +519,19 @@ void wxGridBagSizer::RecalcSizes()
     m_cols = m_colWidths.GetCount();
     int idx, width, height;
 
+    /*
+     * Note that when we perform a layout with all elements hidden,
+     * this function is called with m_rowHeights and m_colWidths empty,
+     * which generates a crash.
+     * This test identifies those cases and stops the execution of
+     * RecalcSizes as if sizer was empty.
+     */
+    if ((m_rows == 0) ||
+        (m_cols == 0))
+    {
+        return;
+    }
+
     AdjustForGrowables(sz);
 
     // Find the start positions on the window of the rows and columns
