Index: samples/listctrl/listtest.cpp
===================================================================
--- samples/listctrl/listtest.cpp	(revision 62459)
+++ samples/listctrl/listtest.cpp	(working copy)
@@ -1135,6 +1135,16 @@
                 wxLogMessage(wxT("Bounding rect of item %ld column %d is (%d, %d)-(%d, %d)"),
                              item, subItem + 1,
                              r.x, r.y, r.x + r.width, r.y + r.height);
+
+                GetSubItemRect(item, subItem, r, wxLIST_RECT_ICON);     // No error checks for quick bug demo.  ;-)
+                wxLogMessage(wxT("Bounding rect of ICON of item %ld column %d is (%d, %d)-(%d, %d)"),
+                             item, subItem + 1,
+                             r.x, r.y, r.x + r.width, r.y + r.height);
+
+                GetSubItemRect(item, subItem, r, wxLIST_RECT_LABEL);
+                wxLogMessage(wxT("Bounding rect of LABEL of item %ld column %d is (%d, %d)-(%d, %d)"),
+                             item, subItem + 1,
+                             r.x, r.y, r.x + r.width, r.y + r.height);
             }
             break;
 
