Index: src/msw/listctrl.cpp
===================================================================
--- src/msw/listctrl.cpp	(revision 62459)
+++ src/msw/listctrl.cpp	(working copy)
@@ -1183,7 +1183,7 @@
     // there is no way to retrieve the first sub item bounding rectangle using
     // wxGetListCtrlSubItemRect() as 0 means the whole item, so we need to
     // truncate it at first column ourselves
-    if ( subItem == 0 )
+    if ( subItem == 0 && code == wxLIST_RECT_BOUNDS)
         rect.width = GetColumnWidth(0);
 
     return true;
