Index: src/mac/app.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/mac/app.cpp,v
retrieving revision 1.72.2.7
diff -u -r1.72.2.7 app.cpp
--- src/mac/app.cpp	2002/11/04 22:48:27	1.72.2.7
+++ src/mac/app.cpp	2002/11/15 21:29:24
@@ -2331,7 +2331,12 @@
             item->Check( !item->IsChecked() ) ;
         }
 
-        wxWindow* frontwindow = wxFindWinFromMacWindow( ::FrontWindow() )  ;
+        wxWindow* frontwindow = NULL;
+        if(wxMenuBar::MacGetInstalledMenuBar())
+            frontwindow = wxMenuBar::MacGetInstalledMenuBar()->GetFrame();
+        if(!frontwindow) // menu bar is not installed or not attached to frame
+            frontwindow = wxFindWinFromMacWindow( ::FrontWindow() )  ;
+        wxASSERT(frontwindow);
         wxCommandEvent event(wxEVT_COMMAND_MENU_SELECTED, id );
         event.m_timeStamp =  ((EventRecord*) MacGetCurrentEvent())->when ;
         event.SetEventObject(menu);

 	  	 
