Index: include/wx/richtext/richtextctrl.h
===================================================================
--- include/wx/richtext/richtextctrl.h	(revision 72543)
+++ include/wx/richtext/richtextctrl.h	(working copy)
@@ -1007,13 +1007,7 @@
     void SetSelection(const wxRichTextSelection& sel) { m_selection = sel; }
     //@}
 
-
     /**
-        Selects all the text in the buffer.
-    */
-    virtual void SelectAll();
-
-    /**
         Makes the control editable, or not.
     */
     virtual void SetEditable(bool editable);
Index: src/common/combocmn.cpp
===================================================================
--- src/common/combocmn.cpp	(revision 72543)
+++ src/common/combocmn.cpp	(working copy)
@@ -765,7 +765,7 @@
         if ( m_combo->GetTextCtrl() )
             m_combo->GetTextCtrl()->SelectAll();
         else
-            m_combo->SetSelection(-1,-1);
+            m_combo->SelectAll();
     }
 
     // Send focus indication to parent.
Index: src/generic/grideditors.cpp
===================================================================
--- src/generic/grideditors.cpp	(revision 72543)
+++ src/generic/grideditors.cpp	(working copy)
@@ -496,7 +496,7 @@
 {
     Text()->SetValue(startValue);
     Text()->SetInsertionPointEnd();
-    Text()->SetSelection(-1, -1);
+    Text()->SelectAll();
     Text()->SetFocus();
 }
 
Index: src/generic/textdlgg.cpp
===================================================================
--- src/generic/textdlgg.cpp	(revision 72543)
+++ src/generic/textdlgg.cpp	(working copy)
@@ -122,7 +122,7 @@
     if ( style & wxCENTRE )
         Centre( wxBOTH );
 
-    m_textctrl->SetSelection(-1, -1);
+    m_textctrl->SelectAll();
     m_textctrl->SetFocus();
 
     wxEndBusyCursor();
Index: src/generic/treectlg.cpp
===================================================================
--- src/generic/treectlg.cpp	(revision 72543)
+++ src/generic/treectlg.cpp	(working copy)
@@ -454,7 +454,7 @@
     (void)Create(m_owner, wxID_ANY, m_startValue,
                  rect.GetPosition(), rect.GetSize());
 
-    SetSelection(-1, -1);
+    SelectAll();
 }
 
 void wxTreeTextCtrl::EndEdit(bool discardChanges)
Index: src/propgrid/editors.cpp
===================================================================
--- src/propgrid/editors.cpp	(revision 72543)
+++ src/propgrid/editors.cpp	(working copy)
@@ -497,7 +497,7 @@
         tc->SetValue(correctText);
     }
 
-    tc->SetSelection(-1,-1);
+    tc->SelectAll();
 }
 
 void wxPGTextCtrlEditor::OnFocus( wxPGProperty* property,
Index: src/richtext/richtextctrl.cpp
===================================================================
--- src/richtext/richtextctrl.cpp	(revision 72543)
+++ src/richtext/richtextctrl.cpp	(working copy)
@@ -2709,11 +2709,6 @@
 // selection and ranges
 // ----------------------------------------------------------------------------
 
-void wxRichTextCtrl::SelectAll()
-{
-    SetSelection(-1, -1);
-}
-
 /// Select none
 void wxRichTextCtrl::SelectNone()
 {
