? monolib
? wxmozsupport.patch
? src/mac/.DS_Store
Index: src/mac/toplevel.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/mac/toplevel.cpp,v
retrieving revision 1.66
diff -b -u -2 -r1.66 toplevel.cpp
--- src/mac/toplevel.cpp	2003/09/11 20:33:02	1.66
+++ src/mac/toplevel.cpp	2003/09/22 02:38:54
@@ -344,5 +344,4 @@
                     newContentRect.right - newContentRect.left ,
                     newContentRect.bottom - newContentRect.top, wxSIZE_USE_EXISTING);
-
                 result = noErr;
             }
@@ -539,4 +538,6 @@
 void wxTopLevelWindowMac::Maximize(bool maximize)
 {
+    wxMacPortStateHelper help( (GrafPtr) GetWindowPort( (WindowRef) m_macWindow) ) ;
+    wxMacWindowClipper clip (this);
     ZoomWindow( (WindowRef)m_macWindow , maximize ? inZoomOut : inZoomIn , false ) ;
 
@@ -682,4 +683,5 @@
     }
 
+    //attr |= kWindowLiveResizeAttribute; //turn on live resizing
 #if TARGET_CARBON
 #if 0 //  having problems right now with that
@@ -1036,4 +1038,6 @@
 void wxTopLevelWindowMac::DoMoveWindow(int x, int y, int width, int height)
 {
+    wxMacPortStateHelper help( (GrafPtr) GetWindowPort( (WindowRef) m_macWindow) ) ;
+    wxMacWindowClipper clip (this);
     int former_x = m_x ;
     int former_y = m_y ;
Index: src/mac/uma.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/mac/uma.cpp,v
retrieving revision 1.46
diff -b -u -2 -r1.46 uma.cpp
--- src/mac/uma.cpp	2003/09/13 00:03:09	1.46
+++ src/mac/uma.cpp	2003/09/22 02:38:55
@@ -538,4 +538,5 @@
 {
     WindowRef theWindow = GetControlOwner(inControl) ;
+    wxMacPortStateHelper help( (GrafPtr) GetWindowPort(theWindow) ) ;
     RgnHandle updateRgn = NewRgn() ;
     GetWindowUpdateRgn( theWindow , updateRgn ) ;
@@ -629,4 +630,5 @@
 void UMAUpdateControls( WindowPtr inWindow , RgnHandle inRgn )
 {
+    wxMacPortStateHelper help( (GrafPtr) GetWindowPort( (WindowRef) inWindow) ) ;
     RgnHandle updateRgn = NewRgn() ;
     GetWindowUpdateRgn( inWindow , updateRgn ) ;
@@ -729,4 +731,5 @@
     GetPort( &m_oldPort ) ;
     SetPort( newport ) ;
+    SetOrigin(0,0);
     wxASSERT_MSG( m_clip == NULL , wxT("Cannot call setup twice") ) ;
     m_clip = NewRgn() ;
Index: src/mac/window.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/mac/window.cpp,v
retrieving revision 1.114
diff -b -u -2 -r1.114 window.cpp
--- src/mac/window.cpp	2003/09/09 15:37:19	1.114
+++ src/mac/window.cpp	2003/09/22 02:38:56
@@ -1463,4 +1463,8 @@
 bool wxWindowMac::MacDispatchMouseEvent(wxMouseEvent& event)
 {
+    //In case a third-party component changed the port...
+    wxMacPortStateHelper help( (GrafPtr) GetWindowPort( (WindowRef) MacGetRootWindow()) ) ;
+    wxMacWindowClipper clip (this);
+    
     if ((event.m_x < m_x) || (event.m_y < m_y) ||
         (event.m_x > (m_x + m_width)) || (event.m_y > (m_y + m_height)))

 	  	 
