diff -uNr wxWidgets-org/include/wx/fontenc.h wxWidgets-revised/include/wx/fontenc.h
--- wxWidgets-org/include/wx/fontenc.h	2010-07-24 08:30:14 +0800
+++ wxWidgets-revised/include/wx/fontenc.h	2012-05-31 09:08:12 +0800
@@ -64,7 +64,11 @@
     wxFONTENCODING_CP1254,          // WinTurkish
     wxFONTENCODING_CP1255,          // WinHebrew
     wxFONTENCODING_CP1256,          // WinArabic
-    wxFONTENCODING_CP1257,          // WinBaltic (same as Latin 7)
+    wxFONTENCODING_CP1257,          // WinBaltic (same as Latin 7)
+   ///addition begin
+    wxFONTENCODING_CP1258,///Vietnamese
+    wxFONTENCODING_CP1361,///Also spelled "Johap". Specifies the Johab Korean character set.
+    ///addition end
     wxFONTENCODING_CP12_MAX,
 
     wxFONTENCODING_UTF7,            // UTF-7 Unicode encoding
@@ -151,7 +155,11 @@
     wxFONTENCODING_SHIFT_JIS = wxFONTENCODING_CP932, // Shift JIS
 
         // Korean (CP 949 not actually the same but close enough)
-    wxFONTENCODING_EUC_KR = wxFONTENCODING_CP949
+    wxFONTENCODING_EUC_KR = wxFONTENCODING_CP949,
+   ///addition begin
+    wxFONTENCODING_JOHAB=wxFONTENCODING_CP1361,///Also spelled "Johap". Specifies the Johab Korean character set.
+    wxFONTENCODING_VIETNAMESE =wxFONTENCODING_CP1258,///Vietnamese
+    ///addition end
 };
 
 #endif // _WX_FONTENC_H_
diff -uNr wxWidgets-org/src/common/fmapbase.cpp wxWidgets-revised/src/common/fmapbase.cpp
--- wxWidgets-org/src/common/fmapbase.cpp	2012-03-04 09:30:16 +0800
+++ wxWidgets-revised/src/common/fmapbase.cpp	2012-05-31 09:16:18 +0800
@@ -87,7 +87,11 @@
     wxFONTENCODING_CP1254,
     wxFONTENCODING_CP1255,
     wxFONTENCODING_CP1256,
-    wxFONTENCODING_CP1257,
+    wxFONTENCODING_CP1257,
+   ///addition begin
+    wxFONTENCODING_CP1258,///Vietnamese
+    wxFONTENCODING_CP1361,///Also spelled "Johap". Specifies the Johab Korean character set.
+    ///addition end
     wxFONTENCODING_CP437,
     wxFONTENCODING_UTF7,
     wxFONTENCODING_UTF8,
@@ -174,7 +178,11 @@
     wxTRANSLATE( "Windows Turkish (CP 1254)" ),
     wxTRANSLATE( "Windows Hebrew (CP 1255)" ),
     wxTRANSLATE( "Windows Arabic (CP 1256)" ),
-    wxTRANSLATE( "Windows Baltic (CP 1257)" ),
+    wxTRANSLATE( "Windows Baltic (CP 1257)" ),
+    ///addition begin
+    wxTRANSLATE( "Windows Vietnamese (CP 1258)" ),
+    wxTRANSLATE( "Windows Johab (CP 1361)" ),
+    ///addition end
     wxTRANSLATE( "Windows/DOS OEM (CP 437)" ),
     wxTRANSLATE( "Unicode 7 bit (UTF-7)" ),
     wxTRANSLATE( "Unicode 8 bit (UTF-8)" ),
@@ -266,7 +274,10 @@
 
     { wxT( "WINDOWS-874" ), wxT( "CP874" ), wxT( "MS874" ), wxT( "IBM-874" ), NULL },
     { wxT( "WINDOWS-932" ), wxT( "CP932" ), wxT( "MS932" ), wxT( "IBM-932" ), wxT( "SJIS" ), wxT( "SHIFT-JIS" ), wxT( "SHIFT_JIS" ), NULL },
-    { wxT( "WINDOWS-936" ), wxT( "CP936" ), wxT( "MS936" ), wxT( "IBM-936" ), wxT( "GB2312" ), NULL },
+   /// { wxT( "WINDOWS-936" ), wxT( "CP936" ), wxT( "MS936" ), wxT( "IBM-936" ), wxT( "GB2312" ), NULL },
+    ///modification begin -gbk in Chinese web page and database is widely used
+    { wxT( "WINDOWS-936" ), wxT( "CP936" ), wxT( "MS936" ), wxT( "IBM-936" ), wxT( "GB2312" ),wxT( "gbk" ),wxT( "GBK" ), NULL },
+    ///modification end
     { wxT( "WINDOWS-949" ), wxT( "CP949" ), wxT( "MS949" ), wxT( "IBM-949" ), wxT( "EUC-KR" ), wxT( "eucKR" ), wxT( "euc_kr" ), NULL },
     { wxT( "WINDOWS-950" ), wxT( "CP950" ), wxT( "MS950" ), wxT( "IBM-950" ), wxT( "BIG5" ), wxT( "BIG-5" ), wxT( "BIG-FIVE" ), NULL },
     { wxT( "WINDOWS-1250" ),wxT( "CP1250" ),wxT( "MS1250" ),wxT( "IBM-1250" ),NULL },
@@ -276,7 +287,11 @@
     { wxT( "WINDOWS-1254" ),wxT( "CP1254" ),wxT( "MS1254" ),wxT( "IBM-1254" ),NULL },
     { wxT( "WINDOWS-1255" ),wxT( "CP1255" ),wxT( "MS1255" ),wxT( "IBM-1255" ),NULL },
     { wxT( "WINDOWS-1256" ),wxT( "CP1256" ),wxT( "MS1256" ),wxT( "IBM-1256" ),NULL },
-    { wxT( "WINDOWS-1257" ),wxT( "CP1257" ),wxT( "MS1257" ),wxT( "IBM-1257" ),NULL },
+    { wxT( "WINDOWS-1257" ),wxT( "CP1257" ),wxT( "MS1257" ),wxT( "IBM-1257" ),NULL },
+    ///addition begin
+    { wxT( "WINDOWS-1258" ), wxT( "CP1258" ),wxT( "MS1258" ),wxT( "IBM-1258" ), NULL },
+    { wxT( "WINDOWS-1361" ), wxT( "CP1361" ),wxT( "MS1361" ), wxT( "JOHAB" ), NULL },
+    ///addition end
     { wxT( "WINDOWS-437" ), wxT( "CP437" ), wxT( "MS437" ), wxT( "IBM-437" ), NULL },
 
     { wxT( "UTF-7" ), wxT("UTF7"), NULL },
@@ -341,7 +356,7 @@
     { wxT( "MacRomanian" ), NULL },
     { wxT( "MacCeltic" ), NULL },
     { wxT( "MacGaelic" ), NULL },
-    { wxT( "MacKeyboardGlyphs" ), NULL }
+    { wxT( "MacKeyboardGlyphs" ), NULL }
 };
 
 wxCOMPILE_TIME_ASSERT( WXSIZEOF(gs_encodingDescs) == WXSIZEOF(gs_encodings), EncodingsArraysNotInSync );
@@ -748,8 +763,17 @@
                             break;
 
                         case 950:
-                            encoding = wxFONTENCODING_CP950;
-                            break;
+                            encoding = wxFONTENCODING_CP950;
+                            break;
+  ///addition begin
+                        case 1258:
+                            encoding = wxFONTENCODING_CP1258;
+                            break;
+
+                        case 1361:
+                            encoding = wxFONTENCODING_CP1361;
+                            break;
+  ///addition end
                     }
                 }
             }
diff -uNr wxWidgets-org/src/common/fontmap.cpp wxWidgets-revised/src/common/fontmap.cpp
--- wxWidgets-org/src/common/fontmap.cpp	2012-03-21 09:30:18 +0800
+++ wxWidgets-revised/src/common/fontmap.cpp	2012-05-31 09:15:11 +0800
@@ -98,7 +98,11 @@
 wxENUM_MEMBER( wxFONTENCODING_CP1254 )
 wxENUM_MEMBER( wxFONTENCODING_CP1255 )
 wxENUM_MEMBER( wxFONTENCODING_CP1256 )
-wxENUM_MEMBER( wxFONTENCODING_CP1257 )
+wxENUM_MEMBER( wxFONTENCODING_CP1257 )
+///addition begin
+wxENUM_MEMBER( wxFONTENCODING_CP1258 )
+wxENUM_MEMBER( wxFONTENCODING_CP1361 )
+///addition end
 wxENUM_MEMBER( wxFONTENCODING_CP12_MAX )
 wxENUM_MEMBER( wxFONTENCODING_UTF7 )
 wxENUM_MEMBER( wxFONTENCODING_UTF8 )
@@ -106,7 +110,7 @@
 wxENUM_MEMBER( wxFONTENCODING_BIG5 )
 wxENUM_MEMBER( wxFONTENCODING_SHIFT_JIS )
 wxENUM_MEMBER( wxFONTENCODING_EUC_JP )
-wxENUM_MEMBER( wxFONTENCODING_UNICODE )
+wxENUM_MEMBER( wxFONTENCODING_UNICODE )
 wxEND_ENUM( wxFontEncoding )
 
 // ----------------------------------------------------------------------------
diff -uNr wxWidgets-org/src/msw/fontutil.cpp wxWidgets-revised/src/msw/fontutil.cpp
--- wxWidgets-org/src/msw/fontutil.cpp	2011-03-20 09:30:38 +0800
+++ wxWidgets-revised/src/msw/fontutil.cpp	2012-05-31 09:09:15 +0800
@@ -249,7 +249,16 @@
 
         case CHINESEBIG5_CHARSET:
             fontEncoding = wxFONTENCODING_CP950;
-            break;
+            break;
+///addition begin
+        case VIETNAMESE_CHARSET:
+            fontEncoding = wxFONTENCODING_CP1258;
+            break;
+
+        case JOHAB_CHARSET:
+            fontEncoding = wxFONTENCODING_CP1361;
+            break;
+///addition end
 
 #endif // Win32
 
