Index: src/cocoa/app.mm
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/cocoa/app.mm,v
retrieving revision 1.20
diff -p -u -r1.20 app.mm
--- src/cocoa/app.mm	2003/07/14 13:19:50	1.20
+++ src/cocoa/app.mm	2003/10/20 09:58:03
@@ -140,10 +140,11 @@ bool wxApp::Initialize(int& argc, wxChar
     if ( argc > 1 )
     {
         static const wxChar *ARG_PSN = _T("-psn_");
-        if ( wxStrncmp(argv[1], ARG_PSN, sizeof(ARG_PSN) - 1) == 0 )
+        if ( wxStrncmp(argv[1], ARG_PSN, strlen(ARG_PSN)) == 0 )
         {
             // remove this argument
-            memmove(argv, argv + 1, argc--);
+            --argc;
+            memmove(argv + 1, argv + 2, argc * sizeof(char *));
         }
     }
 
Index: src/mac/app.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/mac/app.cpp,v
retrieving revision 1.108
diff -p -u -r1.108 app.cpp
--- src/mac/app.cpp	2003/07/14 08:41:03	1.108
+++ src/mac/app.cpp	2003/10/20 09:58:03
@@ -571,10 +571,11 @@ bool wxApp::Initialize(int& argc, wxChar
     if ( argc > 1 )
     {
         static const wxChar *ARG_PSN = _T("-psn_");
-        if ( wxStrncmp(argv[1], ARG_PSN, sizeof(ARG_PSN) - 1) == 0 )
+        if ( wxStrncmp(argv[1], ARG_PSN, strlen(ARG_PSN)) == 0 )
         {
             // remove this argument
-            memmove(argv, argv + 1, argc--);
+            --argc;
+            memmove(argv + 1, argv + 2, argc * sizeof(char *));
         }
     }
 

 	  	 
