--- wxWindows-20040223\src\msw\tbar95.cpp	Tue Feb 24 19:56:14 2004
+++ wxWindows\src\msw\tbar95.cpp	Tue Feb 24 19:33:16 2004
@@ -919,4 +919,13 @@
             SetRows(m_nButtons);
         }
+        else
+        {
+            // In XP the earlier deleting of the tools clears the number of rows
+            // from the control. We must tell that we are still vertical.
+            RECT rect;
+            ::SendMessage(GetHwnd(), TB_SETROWS,
+                          MAKEWPARAM(m_maxRows, !(GetWindowStyle() & wxTB_VERTICAL)),
+                          (LPARAM) &rect);
+        }
     }
 

 	  	 
