diff -ru2 -xCVS -x.#* -x*.orig -x*.rej -x*~ wxWindows.orig/src/generic/grid.cpp wxWindows/src/generic/grid.cpp
--- wxWindows.orig/src/generic/grid.cpp	Fri Nov 22 15:03:54 2002
+++ wxWindows/src/generic/grid.cpp	Mon Nov 25 11:21:40 2002
@@ -3489,11 +3489,13 @@
     GetClientSize( &client_width, &client_height );
 
-    dc.SetPen( *wxBLACK_PEN );
+    dc.SetPen( wxPen(wxSystemSettings::GetColour(wxSYS_COLOUR_3DDKSHADOW),1, wxSOLID) );
     dc.DrawLine( client_width-1, client_height-1, client_width-1, 0 );
     dc.DrawLine( client_width-1, client_height-1, 0, client_height-1 );
-
-    dc.SetPen( *wxWHITE_PEN );
     dc.DrawLine( 0, 0, client_width, 0 );
     dc.DrawLine( 0, 0, 0, client_height );
+
+    dc.SetPen( *wxWHITE_PEN );
+    dc.DrawLine( 1, 1, client_width-1, 1 );
+    dc.DrawLine( 1, 1, 1, client_height-1 );
 }
 
@@ -3837,11 +3839,12 @@
     //
     m_labelFont = this->GetFont();
-    m_labelFont.SetWeight( m_labelFont.GetWeight() + 2 );
+//    m_labelFont = wxSystemSettings::GetFont(wxSYS_DEFAULT_GUI_FONT);
+//    m_labelFont.SetWeight( m_labelFont.GetWeight() + 2 );
 
-    m_rowLabelHorizAlign = wxALIGN_LEFT;
+    m_rowLabelHorizAlign = wxALIGN_CENTRE;
     m_rowLabelVertAlign  = wxALIGN_CENTRE;
 
     m_colLabelHorizAlign = wxALIGN_CENTRE;
-    m_colLabelVertAlign  = wxALIGN_TOP;
+    m_colLabelVertAlign  = wxALIGN_CENTRE;
 
     m_defaultColWidth  = WXGRID_DEFAULT_COL_WIDTH;
@@ -3854,7 +3857,7 @@
 #endif
 
-    m_gridLineColour = wxColour( 128, 128, 255 );
+    m_gridLineColour = wxColour( 192,192,192 );
     m_gridLinesEnabled = TRUE;
-    m_cellHighlightColour = m_gridLineColour;
+    m_cellHighlightColour = *wxBLACK;
     m_cellHighlightPenWidth = 2;
     m_cellHighlightROPenWidth = 1;
@@ -3879,6 +3882,8 @@
     m_selectingTopLeft = wxGridNoCellCoords;
     m_selectingBottomRight = wxGridNoCellCoords;
-    m_selectionBackground = wxSystemSettings::GetColour(wxSYS_COLOUR_HIGHLIGHT);
-    m_selectionForeground = wxSystemSettings::GetColour(wxSYS_COLOUR_HIGHLIGHTTEXT);
+//  m_selectionBackground = wxSystemSettings::GetColour(wxSYS_COLOUR_HIGHLIGHT);
+//  m_selectionForeground = wxSystemSettings::GetColour(wxSYS_COLOUR_HIGHLIGHTTEXT);
+    m_selectionBackground = *wxBLACK; 
+    m_selectionForeground = *wxWHITE; 
 
     m_editable = TRUE;  // default for whole grid
@@ -6625,5 +6630,7 @@
 
         // Now draw the rectangle
-        dc.SetPen(wxPen(m_cellHighlightColour, penWidth, wxSOLID));
+        // use the cellHighlightColour if the cell is inside a selection, this
+        // will ensure the cell is always visible.
+        dc.SetPen(wxPen(IsInSelection(row,col)?m_selectionForeground:m_cellHighlightColour, penWidth, wxSOLID));
         dc.SetBrush(*wxTRANSPARENT_BRUSH);
         dc.DrawRectangle(rect);
@@ -6853,13 +6860,15 @@
         rowBottom = GetRowBottom(row) - 1;
 
-    dc.SetPen( *wxBLACK_PEN );
+    dc.SetPen( wxPen(wxSystemSettings::GetColour(wxSYS_COLOUR_3DDKSHADOW),1, wxSOLID) );
     dc.DrawLine( m_rowLabelWidth-1, rowTop,
                  m_rowLabelWidth-1, rowBottom );
 
-    dc.DrawLine( 0, rowBottom, m_rowLabelWidth-1, rowBottom );
+    dc.DrawLine( 0, rowTop, 0, rowBottom );
+
+    dc.DrawLine( 0, rowBottom, m_rowLabelWidth, rowBottom );
 
     dc.SetPen( *wxWHITE_PEN );
-    dc.DrawLine( 0, rowTop, 0, rowBottom );
-    dc.DrawLine( 0, rowTop, m_rowLabelWidth-1, rowTop );
+    dc.DrawLine( 1, rowTop, 1, rowBottom );
+    dc.DrawLine( 1, rowTop, m_rowLabelWidth-1, rowTop );
 
     dc.SetBackgroundMode( wxTRANSPARENT );
@@ -6901,14 +6910,16 @@
         colRight = GetColRight(col) - 1;
 
-    dc.SetPen( *wxBLACK_PEN );
+    dc.SetPen( wxPen(wxSystemSettings::GetColour(wxSYS_COLOUR_3DDKSHADOW),1, wxSOLID) );
     dc.DrawLine( colRight, 0,
                  colRight, m_colLabelHeight-1 );
 
+    dc.DrawLine( colLeft, 0, colRight, 0 );
+
     dc.DrawLine( colLeft, m_colLabelHeight-1,
-                 colRight, m_colLabelHeight-1 );
+                 colRight+1, m_colLabelHeight-1 );
 
     dc.SetPen( *wxWHITE_PEN );
-    dc.DrawLine( colLeft, 0, colLeft, m_colLabelHeight-1 );
-    dc.DrawLine( colLeft, 0, colRight, 0 );
+    dc.DrawLine( colLeft, 1, colLeft, m_colLabelHeight-1 );
+    dc.DrawLine( colLeft, 1, colRight, 1 );
 
     dc.SetBackgroundMode( wxTRANSPARENT );

 	  	 
