Index: src/aui/framemanager.cpp
===================================================================
--- src/aui/framemanager.cpp	(revision 55735)
+++ src/aui/framemanager.cpp	(working copy)
@@ -3319,6 +3319,16 @@
     m_action_window = pane_window;
     m_action_offset = offset;
     m_frame->CaptureMouse();
+
+    if (pane.frame)
+    {
+        wxRect window_rect = pane.frame->GetRect();
+        wxRect client_rect = pane.frame->GetClientRect();
+        wxPoint client_pt = pane.frame->ClientToScreen(client_rect.GetTopLeft());
+        wxPoint origin_pt = client_pt - window_rect.GetTopLeft();
+        m_action_offset += origin_pt;
+    }
 }
 
 
