Index: src/msw/mediactrl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/msw/mediactrl.cpp,v
retrieving revision 1.31
diff -b -u -2 -r1.31 mediactrl.cpp
--- src/msw/mediactrl.cpp	2005/05/23 11:10:19	1.31
+++ src/msw/mediactrl.cpp	2005/05/23 16:56:10
@@ -898,5 +898,5 @@
 
 //Class that utilizes Robert Roeblings Dynamic Library Macros
-class wxQuickTimeLibrary 
+class WXDLLIMPEXP_MEDIA wxQuickTimeLibrary 
 {
 public:

 	  	 
