diff --git a/src/aui/auibar.cpp b/src/aui/auibar.cpp
index 6812786..3d9f346 100644
--- a/src/aui/auibar.cpp
+++ b/src/aui/auibar.cpp
@@ -2497,12 +2497,11 @@ void wxAuiToolBar::OnLeftDown(wxMouseEvent& evt)
         }
     }
 
-    if (m_overflowSizerItem)
+    if (m_overflowSizerItem && m_art)
     {
         wxRect overflow_rect = GetOverflowRect();
 
-        if (m_art &&
-            m_overflowVisible &&
+        if (m_overflowVisible &&
             overflow_rect.Contains(evt.m_x, evt.m_y))
         {
             wxAuiToolBarEvent e(wxEVT_AUITOOLBAR_OVERFLOW_CLICK, -1);
