--- wxWindows-20031223.orig/samples/opengl/cube/cube.cpp	Fri Nov 14 23:21:03 2003
+++ wxWindows-20031223/samples/opengl/cube/cube.cpp	Tue Dec 23 23:36:44 2003
@@ -391,5 +391,5 @@
         Action( m_Key, m_LastTime-m_StartTime, currTime-m_StartTime );
 
-#if defined(__WXMAC__) && !defined(__DARWIN__)
+#if defined(__WXMAC__)
         m_LastRedraw = currTime;    // wxStopWatch() doesn't work on Mac...
 #else

 	  	 
