Index: src/generic/wizard.cpp
===================================================================
--- src/generic/wizard.cpp	(revision 77543)
+++ src/generic/wizard.cpp	(working copy)
@@ -271,6 +271,7 @@
 {
     m_posWizard = wxDefaultPosition;
     m_page = NULL;
+    m_firstpage = NULL;
     m_btnPrev = m_btnNext = NULL;
     m_statbmp = NULL;
     m_sizerBmpAndPage = NULL;
@@ -620,7 +621,7 @@
 
 
     // and update the buttons state
-    m_btnPrev->Enable(HasPrevPage(m_page));
+    m_btnPrev->Enable(m_page != m_firstpage);
 
     const bool hasNext = HasNextPage(m_page);
     const wxString label = hasNext ? _("&Next >") : _("&Finish");
@@ -687,6 +688,8 @@
 {
     wxCHECK_MSG( firstPage, false, wxT("can't run empty wizard") );
 
+    m_firstpage = firstPage;
+
     // can't return false here because there is no old page
     (void)ShowPage(firstPage, true /* forward */);
 
Index: include/wx/generic/wizard.h
===================================================================
--- include/wx/generic/wizard.h	(revision 77543)
+++ include/wx/generic/wizard.h	(working copy)
@@ -124,6 +124,7 @@
 
     // wizard state
     wxWizardPage *m_page;       // the current page or NULL
+    wxWizardPage *m_firstpage;  // the page RunWizard started on or NULL
     wxBitmap      m_bitmap;     // the default bitmap to show
 
     // wizard controls
