diff -ur wxWidgets-2.6.1/src/unix/stackwalk.cpp wxWidgets-2.6.1.mine/src/unix/stackwalk.cpp
--- wxWidgets-2.6.1/src/unix/stackwalk.cpp	2005-03-31 13:26:39.000000000 -0300
+++ wxWidgets-2.6.1.mine/src/unix/stackwalk.cpp	2005-06-27 17:53:01.000000000 -0300
@@ -129,9 +129,10 @@
                     m_offset = ofs;
             }
         }
+	m_module.assign(syminfo, posOpen);
     }
-
-    m_module.assign(syminfo, posOpen);
+    else
+	m_module = syminfo;
 }
 
 void wxStackFrame::OnGetLocation()

 	  	 
