Index: auibook.py
===================================================================
--- auibook.py	(revision 67011)
+++ auibook.py	(working copy)
@@ -5378,6 +5378,10 @@
         self.SetSelectedFont(selectedFont)
         self.SetMeasuringFont(selectedFont)
 
+        # Recalculate tab container size based on new font
+        self.UpdateTabCtrlHeight(force=False)
+        self.DoSizing()
+
         return True
 
 
