Index: create-archive.py
===================================================================
--- create-archive.py	(revision 61956)
+++ create-archive.py	(working copy)
@@ -17,8 +17,9 @@
 rootDir = os.path.abspath(os.path.join(scriptDir, "..", ".."))
 contribDir = os.path.join("contrib", "src")
 
-dirsToCopy = ["art", "build", "debian", "demos", "docs", "include", "interface", "lib",
-                "locale", "samples", "src", "tests", "utils"]
+dirsToCopy = ["art", "build", "debian", "demos", "distrib/mac", "docs",
+                "include", "interface", "lib", "locale", "samples", "src",
+                "src", "tests", "utils"]
 
 dirsToIgnore = [".svn", "CVS"]
 excludeExtensions = [".rej", ".orig", ".mine", ".tmp"]
