diff -uNrb /Users/chris/Downloads/wxPython_Phoenix-2.9.5.81-r74279-darwin-py3.3/wx/lib/agw/aui/aui_utilities.py wx/lib/agw/aui/aui_utilities.py
--- /Users/chris/Downloads/wxPython_Phoenix-2.9.5.81-r74279-darwin-py3.3/wx/lib/agw/aui/aui_utilities.py	2013-06-24 03:12:34.000000000 -0400
+++ wx/lib/agw/aui/aui_utilities.py	2013-06-25 13:55:23.000000000 -0400
@@ -14,7 +14,12 @@
 
 
 if wx.Platform == "__WXMAC__":
+    try:
     import Carbon.Appearance
+    except ImportError:
+        CARBON = False
+    else:
+        CARBON = True
     
     
 def BlendColour(fg, bg, alpha):
@@ -166,7 +171,7 @@
     """
 
     if wx.Platform == "__WXMAC__":
-
+        if CARBON:
         if hasattr(wx, 'MacThemeColour'):
             base_colour = wx.MacThemeColour(Carbon.Appearance.kThemeBrushToolbarBackground)
         else:
diff -uNrb /Users/chris/Downloads/wxPython_Phoenix-2.9.5.81-r74279-darwin-py3.3/wx/lib/agw/aui/tabart.py wx/lib/agw/aui/tabart.py
--- /Users/chris/Downloads/wxPython_Phoenix-2.9.5.81-r74279-darwin-py3.3/wx/lib/agw/aui/tabart.py	2013-06-24 03:12:34.000000000 -0400
+++ wx/lib/agw/aui/tabart.py	2013-06-25 13:57:34.000000000 -0400
@@ -18,7 +18,12 @@
 import wx
 
 if wx.Platform == '__WXMAC__':
+    try:
     import Carbon.Appearance
+    except ImportError:
+        CARBON = False
+    else:
+        CARBON = True
 
 from .aui_utilities import BitmapFromBits, StepColour, IndentPressedBitmap, ChopText
 from .aui_utilities import GetBaseColour, DrawMACCloseButton, LightColour, TakeScreenShot
@@ -139,6 +144,7 @@
         self._disabled_windowlist_bmp = BitmapFromBits(nb_list_bits, 16, 16, wx.Colour(128, 128, 128))
 
         if wx.Platform == "__WXMAC__":
+            if CARBON:
             # Get proper highlight colour for focus rectangle from the
             # current Mac theme.  kThemeBrushFocusHighlight is
             # available on Mac OS 8.5 and higher
diff -uNrb /Users/chris/Downloads/wxPython_Phoenix-2.9.5.81-r74279-darwin-py3.3/wx/lib/agw/flatnotebook.py wx/lib/agw/flatnotebook.py
--- /Users/chris/Downloads/wxPython_Phoenix-2.9.5.81-r74279-darwin-py3.3/wx/lib/agw/flatnotebook.py	2013-06-24 03:12:35.000000000 -0400
+++ wx/lib/agw/flatnotebook.py	2013-06-25 13:52:02.000000000 -0400
@@ -196,7 +196,12 @@
 
 # Used on OSX to get access to carbon api constants
 if wx.Platform == '__WXMAC__':
+    try:
     import Carbon.Appearance
+    except ImportError:
+        CARBON = False
+    else:
+        CARBON = True
 
 # Check for the new method in 2.7 (not present in 2.6.3.3)
 if wx.VERSION_STRING < "2.7":
@@ -1711,6 +1716,7 @@
         self._tabHeight = None
 
         if wx.Platform == "__WXMAC__":
+            if CARBON:
             # Get proper highlight colour for focus rectangle from the
             # current Mac theme.  kThemeBrushFocusHighlight is
             # available on Mac OS 8.5 and higher
diff -uNrb /Users/chris/Downloads/wxPython_Phoenix-2.9.5.81-r74279-darwin-py3.3/wx/lib/agw/ribbon/art_aui.py wx/lib/agw/ribbon/art_aui.py
--- /Users/chris/Downloads/wxPython_Phoenix-2.9.5.81-r74279-darwin-py3.3/wx/lib/agw/ribbon/art_aui.py	2013-06-24 03:12:34.000000000 -0400
+++ wx/lib/agw/ribbon/art_aui.py	2013-06-25 13:54:03.000000000 -0400
@@ -41,7 +41,12 @@
 from .art import *
 
 if wx.Platform == "__WXMAC__":
+    try:
     import Carbon.Appearance
+    except ImportError:
+        CARBON = False
+    else:
+        CARBON = True
 
 
 def FontFromFont(original):
@@ -60,7 +65,7 @@
         RibbonMSWArtProvider.__init__(self)
         
         if wx.Platform == "__WXMAC__":
-
+            if CARBON:
             if hasattr(wx, 'MacThemeColour'):
                 base_colour = wx.MacThemeColour(Carbon.Appearance.kThemeBrushToolbarBackground)
             else:
