diff --git samples/propgrid/propgrid.cpp samples/propgrid/propgrid.cpp
index afd52d9..6a37ba7 100644
--- samples/propgrid/propgrid.cpp
+++ samples/propgrid/propgrid.cpp
@@ -1389,7 +1389,7 @@ void FormMain::PopulateWithExamples ()
     pg->Append( new wxStringProperty( wxT("StringPropertyWithBitmap"),
                 wxPG_LABEL,
                 wxT("Test Text")) );
-    wxBitmap myTestBitmap(60, 15, 32);
+    wxBitmap myTestBitmap(60, 15);
     wxMemoryDC mdc;
     mdc.SelectObject(myTestBitmap);
     mdc.Clear();
diff --git src/propgrid/property.cpp src/propgrid/property.cpp
index 1e4f0cd..7d8af54 100644
--- src/propgrid/property.cpp
+++ src/propgrid/property.cpp
@@ -2107,7 +2107,7 @@ void wxPGProperty::SetValueImage( wxBitmap& bmp )
             img.Rescale(wxRound(bmp.GetWidth()*scaleY),
                         wxRound(bmp.GetHeight()*scaleY),
                         wxIMAGE_QUALITY_HIGH);
-            wxBitmap* bmpNew = new wxBitmap(img, 32);
+            wxBitmap* bmpNew = new wxBitmap(img);
         #else
             // This is the old, deprecated method of scaling the image
             wxBitmap* bmpNew = new wxBitmap(maxSz.x,maxSz.y,bmp.GetDepth());
