Index: menu.h
===================================================================
--- menu.h	(revision 57069)
+++ menu.h	(working copy)
@@ -678,6 +678,21 @@
     */
     void Enable(int id, bool enable);
 
+    /**
+        Finds the menu item object associated with the given menu item identifier and,
+        optionally, the position of the item in the menu if pos != NULL.
+        Unlike FindItem(), this function doesn't recurse but only looks through
+        our direct children.
+        @param id
+            Menu item identifier.
+        @param pos
+            If the pointer is not @c NULL, it will be filled with the item's
+            position. If the item was not found the pointer is filled with @c wxNOT_FOUND
+
+        @return Menu item object or @c NULL if none is found.
+    */
+    wxMenuItem *FindChildItem(int id, size_t *pos = NULL) const;
+
     /**
         Finds the menu id for a menu item string.
 
