diff -rbu2 wxMSW-2.2.7-original/include/wx/msw/wx.rc wxMSW-2.2.7-modified/include/wx/msw/wx.rc
--- wxMSW-2.2.7-original/include/wx/msw/wx.rc	Tue Jan 23 15:54:40 2001
+++ wxMSW-2.2.7-modified/include/wx/msw/wx.rc	Mon Jul 23 11:35:58 2001
@@ -117,8 +117,4 @@
 
 // Standard icons
-wxICON_QUESTION		ICON "wx/msw/question.ico"
-wxICON_WARNING		ICON "wx/msw/warning.ico"
-wxICON_ERROR		ICON "wx/msw/error.ico"
-wxICON_INFO		ICON "wx/msw/info.ico"
 wxICON_TIP		ICON "wx/msw/tip.ico"
 
diff -rbu2 wxMSW-2.2.7-original/src/msw/gdiimage.cpp wxMSW-2.2.7-modified/src/msw/gdiimage.cpp
--- wxMSW-2.2.7-original/src/msw/gdiimage.cpp	Fri Mar 24 13:02:20 2000
+++ wxMSW-2.2.7-modified/src/msw/gdiimage.cpp	Mon Jul 23 11:13:08 2001
@@ -458,4 +458,50 @@
     }
 
+    if ( hicon == (HICON)NULL )
+    {
+        LPTSTR lpstrIconName = (LPTSTR)NULL ;
+        if (name == wxT("wxICON_QUESTION"))
+        {
+            lpstrIconName = (LPTSTR)IDI_QUESTION ;
+        }
+        else
+        {
+            if (name == wxT("wxICON_WARNING"))
+            {
+                lpstrIconName = (LPTSTR)IDI_EXCLAMATION ;
+            }
+            else
+            {
+                if(name == wxT("wxICON_ERROR"))
+                {
+                    lpstrIconName = (LPTSTR)IDI_HAND ;
+                }
+                else
+                {
+                    if(name == wxT("wxICON_INFO"))
+                    {
+                        lpstrIconName = (LPTSTR)IDI_ASTERISK ;
+                    }
+                }
+            }
+        }
+
+        if (lpstrIconName != NULL)
+        {
+#if defined(__WIN32__) && !defined(__SC__)
+            if ( desiredWidth != -1 && desiredHeight != -1 )
+            {
+                hicon = (HICON)::LoadImage((HINSTANCE)NULL, lpstrIconName, IMAGE_ICON,
+                                            desiredWidth, desiredHeight,
+                                            LR_DEFAULTCOLOR);
+            }
+            else
+#endif // Win32
+            {
+                hicon = ::LoadIcon((HINSTANCE)NULL, lpstrIconName);
+            }
+        }
+    }
+
     wxSize size = GetHiconSize(hicon);
     icon->SetSize(size.x, size.y);

 	  	 
