Index: src/generic/toolbkg.cpp
===================================================================
--- src/generic/toolbkg.cpp	(revision 63617)
+++ src/generic/toolbkg.cpp	(working copy)
@@ -132,20 +132,19 @@
 wxSize wxToolbook::GetControllerSize() const
 {
     const wxSize sizeClient = GetClientSize(),
-                 sizeBorder = m_bookctrl->GetSize() - m_bookctrl->GetClientSize(),
-                 sizeToolBar = GetToolBar()->GetSize() + sizeBorder;
+                 sizeToolBar = GetToolBar()->GetBestSize();
 
     wxSize size;
 
     if ( IsVertical() )
     {
-        size.x = sizeClient.x;
+        size.x = wxMax(sizeClient.x, sizeToolBar.x);
         size.y = sizeToolBar.y;
     }
     else // left/right aligned
     {
         size.x = sizeToolBar.x;
-        size.y = sizeClient.y;
+        size.y = wxMax(sizeClient.y, sizeToolBar.y);
     }
 
     return size;
@@ -168,10 +167,12 @@
     if ( IsVertical() )
     {
         size.y += sizeToolBar.y + GetInternalBorder();
+        size.x = wxMax(size.x, sizeToolBar.x);
     }
     else // left/right aligned
     {
         size.x += sizeToolBar.x + GetInternalBorder();
+        size.y = wxMax(size.x, sizeToolBar.y);
     }
 
     return size;
