Index: src/common/textentrycmn.cpp
===================================================================
--- src/common/textentrycmn.cpp	(revision 71399)
+++ src/common/textentrycmn.cpp	(working copy)
@@ -233,9 +233,12 @@
     }
     else // Same value, no need to do anything.
     {
-        // Except that we still need to generate the event for consistency with
-        // the normal case when the text does change.
-        SendTextUpdatedEvent(GetEditableWindow());
+        if ( (flags & SetValue_SendEvent) != 0)
+        {
+            // Except that we still need to generate the event for consistency with
+            // the normal case when the text does change.
+            SendTextUpdatedEvent(GetEditableWindow());
+        }
     }
 
     SetInsertionPoint(0);
