Index: src/msw/graphics.cpp
===================================================================
--- src/msw/graphics.cpp	(revision 53373)
+++ src/msw/graphics.cpp	(working copy)
@@ -323,6 +323,8 @@
     virtual void GetPartialTextExtents(const wxString& text, wxArrayDouble& widths) const;
     virtual bool ShouldOffset() const;
 
+    void SetBaseDC(wxDC* dc) { m_baseDC = dc; }
+    
 private:
     void    Init();
     void    SetDefaults();
@@ -331,7 +333,8 @@
     vector<GraphicsState> m_stateStack;
     GraphicsState m_state1;
     GraphicsState m_state2;
-
+    wxDC* m_baseDC;
+    
     DECLARE_DYNAMIC_CLASS_NO_COPY(wxGDIPlusContext)
 };
 
@@ -355,6 +358,12 @@
     DECLARE_DYNAMIC_CLASS_NO_COPY(wxGDIPlusMeasuringContext)
 } ;
 
+
+wxDC* wxGraphicsContext::GetBaseDC() const
+{
+    return ((wxGDIPlusContext*)this)->m_baseDC;
+}
+
 //-----------------------------------------------------------------------------
 // wxGDIPlusPen implementation
 //-----------------------------------------------------------------------------
@@ -1446,13 +1455,17 @@
 wxGraphicsContext * wxGDIPlusRenderer::CreateContext( const wxWindowDC& dc)
 {
     EnsureIsLoaded();
-    return new wxGDIPlusContext(this,(HDC) dc.GetHDC());
+    wxGDIPlusContext* ctx = new wxGDIPlusContext(this,(HDC) dc.GetHDC());
+    ctx->SetBaseDC(dc);
+    return ctx
 }
 
 wxGraphicsContext * wxGDIPlusRenderer::CreateContext( const wxMemoryDC& dc)
 {
     EnsureIsLoaded();
-    return new wxGDIPlusContext(this,(HDC) dc.GetHDC());
+    wxGDIPlusContext* ctx = new wxGDIPlusContext(this,(HDC) dc.GetHDC());
+    ctx->SetBaseDC(dc);
+    return ctx;
 }
 
 wxGraphicsContext * wxGDIPlusRenderer::CreateMeasuringContext()
Index: include/wx/dcgraph.h
===================================================================
--- include/wx/dcgraph.h	(revision 53373)
+++ include/wx/dcgraph.h	(working copy)
@@ -93,6 +93,8 @@
 
     wxGraphicsContext* GetGraphicsContext() { return m_graphicContext; }
     virtual void SetGraphicsContext( wxGraphicsContext* ctx );
+
+    wxDC* GetBaseDC() const { return m_graphicContext->GetBaseDC(); }
     
 protected:
     // the true implementations
Index: include/wx/graphics.h
===================================================================
--- include/wx/graphics.h	(revision 53373)
+++ include/wx/graphics.h	(working copy)
@@ -562,6 +562,8 @@
     // helper to determine if a 0.5 offset should be applied for the drawing operation
     virtual bool ShouldOffset() const { return false; }
 
+    wxDC* GetBaseDC() const;
+    
 protected :
 
     wxGraphicsPen m_pen;
