diff -ru2 -xCVS -x.#* -x*.orig -x*.rej -x*~ wxWindows.orig/include/wx/wizard.h wxWindows/include/wx/wizard.h
--- wxWindows.orig/include/wx/wizard.h	Fri Nov 22 15:03:46 2002
+++ wxWindows/include/wx/wizard.h	Fri Nov 22 15:38:01 2002
@@ -245,4 +245,5 @@
     DECLARE_EVENT_TYPE(wxEVT_WIZARD_CANCEL, 902)
     DECLARE_EVENT_TYPE(wxEVT_WIZARD_HELP, 903)
+    DECLARE_EVENT_TYPE(wxEVT_WIZARD_FINISHED, 903)
 END_DECLARE_EVENT_TYPES()
 
@@ -259,4 +260,7 @@
 // unless the event handler vetoes the event
 #define EVT_WIZARD_CANCEL(id, fn) DECLARE_EVENT_TABLE_ENTRY(wxEVT_WIZARD_CANCEL, id, -1, (wxObjectEventFunction) (wxEventFunction) (wxWizardEventFunction) & fn, (wxObject *)NULL),
+
+// the user pressed "Finish" button and the wizard is going to be dismissed -
+#define EVT_WIZARD_FINISHED(id, fn) DECLARE_EVENT_TABLE_ENTRY(wxEVT_WIZARD_FINISHED, id, -1, (wxObjectEventFunction) (wxEventFunction) (wxWizardEventFunction) & fn, (wxObject *)NULL),
 
 // the user pressed "Help" button 
diff -ru2 -xCVS -x.#* -x*.orig -x*.rej -x*~ wxWindows.orig/src/generic/wizard.cpp wxWindows/src/generic/wizard.cpp
--- wxWindows.orig/src/generic/wizard.cpp	Fri Nov 22 15:03:54 2002
+++ wxWindows/src/generic/wizard.cpp	Fri Nov 22 15:38:46 2002
@@ -58,4 +58,5 @@
 DEFINE_EVENT_TYPE(wxEVT_WIZARD_PAGE_CHANGING)
 DEFINE_EVENT_TYPE(wxEVT_WIZARD_CANCEL)
+DEFINE_EVENT_TYPE(wxEVT_WIZARD_FINISHED)
 DEFINE_EVENT_TYPE(wxEVT_WIZARD_HELP)
 
@@ -69,4 +70,5 @@
     EVT_WIZARD_PAGE_CHANGING(-1, wxWizard::OnWizEvent)
     EVT_WIZARD_CANCEL(-1, wxWizard::OnWizEvent)
+    EVT_WIZARD_FINISHED(-1, wxWizard::OnWizEvent)
     EVT_WIZARD_HELP(-1, wxWizard::OnWizEvent)
 END_EVENT_TABLE()
@@ -366,4 +368,9 @@
         // terminate successfully
         EndModal(wxID_OK);
+        if ( !IsModal() )
+         {
+           wxWizardEvent event(wxEVT_WIZARD_FINISHED, GetId(),FALSE, 0);
+           (void)GetEventHandler()->ProcessEvent(event);
+         }
         return TRUE;
     }

 	  	 
