--- string.h	2006-11-10 11:47:34.000000000 -0500
+++ string_new.h	2007-03-19 10:16:40.281250000 -0500
@@ -703,6 +703,7 @@
            const wxMBConv& WXUNUSED(conv),
            size_t nLength = npos)
       : wxStringBase(psz, nLength == npos ? wxStrlen(psz) : nLength) { }
+  virtual ~wxString() {Clear();};
 
   // even if we're not built with wxUSE_STL == 1 it is very convenient to allow
   // implicit conversions from std::string to wxString as this allows to use
@@ -1597,3 +1598,4 @@
 #endif  // wxSTD_STRING_COMPATIBILITY
 
 #endif  // _WX_WXSTRINGH__
+

 	  	 
