Index: auidemo.cpp
===================================================================
--- auidemo.cpp	(revision 77091)
+++ auidemo.cpp	(working copy)
@@ -855,7 +855,8 @@
     wxChoice* choice = new wxChoice(tb4, ID_SampleItem+35);
     choice->AppendString(wxT("One choice"));
     choice->AppendString(wxT("Another choice"));
-    tb4->AddControl(choice);
+    wxAuiToolBarItem* myitem = tb4->AddControl(choice);
+    tb4->DeleteTool(myitem->GetId());
     tb4->Realize();
 
 
