Index: samples/dialogs/dialogs.cpp
===================================================================
--- samples/dialogs/dialogs.cpp	(revision 65451)
+++ samples/dialogs/dialogs.cpp	(working copy)
@@ -1765,7 +1765,7 @@
                             this,   // parent
                             wxPD_CAN_ABORT |
                             wxPD_CAN_SKIP |
-                            wxPD_APP_MODAL |
+                            //wxPD_APP_MODAL |
                             //wxPD_AUTO_HIDE | // -- try this as well
                             wxPD_ELAPSED_TIME |
                             wxPD_ESTIMATED_TIME |
