Index: include/wx/wxchar.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/wxchar.h,v
retrieving revision 1.205
diff -bu2 -r1.205 include/wx/wxchar.h
--- include/wx/wxchar.h
+++ include/wx/wxchar.h
@@ -2,5 +2,5 @@
  * Name:        wx/wxchar.h
  * Purpose:     Declarations common to wx char/wchar_t usage (wide chars)
- * Author:      Joel Farley, Ove Kåven
+ * Author:      Joel Farley, Ove Kï¿½en
  * Modified by: Vadim Zeitlin, Robert Roebling, Ron Lee
  * Created:     1998/06/12
@@ -1021,14 +1021,14 @@
     #include <stdio.h>  /* for FILE */
 
-    int wxScanf( const wxChar *format, ... ) ATTRIBUTE_PRINTF_1;
-    int wxSscanf( const wxChar *str, const wxChar *format, ... ) ATTRIBUTE_PRINTF_2;
-    int wxFscanf( FILE *stream, const wxChar *format, ... ) ATTRIBUTE_PRINTF_2;
-    int wxVsscanf( const wxChar *str, const wxChar *format, va_list ap );
-    int wxPrintf( const wxChar *format, ... ) ATTRIBUTE_PRINTF_1;
-    int wxSprintf( wxChar *str, const wxChar *format, ... ) ATTRIBUTE_PRINTF_2;
-    int wxFprintf( FILE *stream, const wxChar *format, ... ) ATTRIBUTE_PRINTF_2;
-    int wxVfprintf( FILE *stream, const wxChar *format, va_list ap );
-    int wxVprintf( const wxChar *format, va_list ap );
-    int wxVsprintf( wxChar *str, const wxChar *format, va_list ap );
+    WXDLLIMPEXP_BASE int wxScanf( const wxChar *format, ... ) ATTRIBUTE_PRINTF_1;
+    WXDLLIMPEXP_BASE int wxSscanf( const wxChar *str, const wxChar *format, ... ) ATTRIBUTE_PRINTF_2;
+    WXDLLIMPEXP_BASE int wxFscanf( FILE *stream, const wxChar *format, ... ) ATTRIBUTE_PRINTF_2;
+    WXDLLIMPEXP_BASE int wxVsscanf( const wxChar *str, const wxChar *format, va_list ap );
+    WXDLLIMPEXP_BASE int wxPrintf( const wxChar *format, ... ) ATTRIBUTE_PRINTF_1;
+    WXDLLIMPEXP_BASE int wxSprintf( wxChar *str, const wxChar *format, ... ) ATTRIBUTE_PRINTF_2;
+    WXDLLIMPEXP_BASE int wxFprintf( FILE *stream, const wxChar *format, ... ) ATTRIBUTE_PRINTF_2;
+    WXDLLIMPEXP_BASE int wxVfprintf( FILE *stream, const wxChar *format, va_list ap );
+    WXDLLIMPEXP_BASE int wxVprintf( const wxChar *format, va_list ap );
+    WXDLLIMPEXP_BASE int wxVsprintf( wxChar *str, const wxChar *format, va_list ap );
 #endif /* wxNEED_PRINTF_CONVERSION */
 
@@ -1040,6 +1040,6 @@
 */
 #if defined(wxNEED_PRINTF_CONVERSION) && defined(wxVsnprintf_)
-    int wxSnprintf( wxChar *str, size_t size, const wxChar *format, ... ) ATTRIBUTE_PRINTF_3;
-    int wxVsnprintf( wxChar *str, size_t size, const wxChar *format, va_list ap );
+    WXDLLIMPEXP_BASE int wxSnprintf( wxChar *str, size_t size, const wxChar *format, ... ) ATTRIBUTE_PRINTF_3;
+    WXDLLIMPEXP_BASE int wxVsnprintf( wxChar *str, size_t size, const wxChar *format, va_list ap );
 #else
     #define wxSnprintf wxSnprintf_
@@ -1093,5 +1093,5 @@
 
 #ifdef wxNEED_ISASCII
-    inline int isascii(int c) { return (unsigned)c < 0x80; }
+    WXDLLIMPEXP_BASE inline int isascii(int c) { return (unsigned)c < 0x80; }
 #endif
 
@@ -1158,11 +1158,11 @@
     /* inlined versions */
     #ifdef __cplusplus
-        inline wxChar * wxStrchr(wxChar *s, wxChar c)
+        WXDLLIMPEXP_BASE inline wxChar * wxStrchr(wxChar *s, wxChar c)
             { return (wxChar *)wxStrchr((const wxChar *)s, c); }
-        inline wxChar * wxStrpbrk(wxChar *s, const wxChar *accept)
+        WXDLLIMPEXP_BASE inline wxChar * wxStrpbrk(wxChar *s, const wxChar *accept)
             { return (wxChar *)wxStrpbrk((const wxChar *)s, accept); }
-        inline wxChar * wxStrrchr(wxChar *s, wxChar c)
+        WXDLLIMPEXP_BASE inline wxChar * wxStrrchr(wxChar *s, wxChar c)
             { return (wxChar *)wxStrrchr((const wxChar *)s, c); }
-        inline wxChar *wxStrstr(wxChar *haystack, const wxChar *needle)
+        WXDLLIMPEXP_BASE inline wxChar *wxStrstr(wxChar *haystack, const wxChar *needle)
             { return (wxChar *)wxStrstr((const wxChar *)haystack, needle); }
     #endif
@@ -1296,5 +1296,5 @@
 
         //implement our own wmem variants
-        inline wxChar* wxTmemchr(const wxChar* s, wxChar c, size_t l)
+        WXDLLIMPEXP_BASE inline wxChar* wxTmemchr(const wxChar* s, wxChar c, size_t l)
         {
             for(;l && *s != c;--l, ++s) {}
@@ -1305,5 +1305,5 @@
         }
 
-        inline int wxTmemcmp(const wxChar* sz1, const wxChar* sz2, size_t len)
+        WXDLLIMPEXP_BASE inline int wxTmemcmp(const wxChar* sz1, const wxChar* sz2, size_t len)
         {
             for(; *sz1 == *sz2 && len; --len, ++sz1, ++sz2) {}
@@ -1315,15 +1315,15 @@
         }
 
-        inline wxChar* wxTmemcpy(wxChar* szOut, const wxChar* szIn, size_t len)
+        WXDLLIMPEXP_BASE inline wxChar* wxTmemcpy(wxChar* szOut, const wxChar* szIn, size_t len)
         {
             return (wxChar*) memcpy(szOut, szIn, len * sizeof(wxChar));
         }
 
-        inline wxChar* wxTmemmove(wxChar* szOut, const wxChar* szIn, size_t len)
+        WXDLLIMPEXP_BASE inline wxChar* wxTmemmove(wxChar* szOut, const wxChar* szIn, size_t len)
         {
             return (wxChar*) memmove(szOut, szIn, len * sizeof(wxChar));
         }
 
-        inline wxChar* wxTmemset(wxChar* szOut, const wxChar cIn, size_t len)
+        WXDLLIMPEXP_BASE inline wxChar* wxTmemset(wxChar* szOut, const wxChar cIn, size_t len)
         {
             wxChar* szRet = szOut;

 	  	 
