Index: src/common/filename.cpp
===================================================================
--- src/common/filename.cpp	(revision 61008)
+++ src/common/filename.cpp	(working copy)
@@ -592,7 +592,7 @@
     return cwd;
 }
 
-bool wxFileName::SetCwd()
+bool wxFileName::SetCwd() const
 {
     return wxFileName::SetCwd( GetPath() );
 }
@@ -1433,7 +1433,7 @@
 
 bool wxFileName::GetShortcutTarget(const wxString& shortcutPath,
                                    wxString& targetFilename,
-                                   wxString* arguments)
+                                   wxString* arguments) const
 {
     wxString path, file, ext;
     wxFileName::SplitPath(shortcutPath, & path, & file, & ext);
@@ -2277,7 +2295,7 @@
 
 bool wxFileName::SetTimes(const wxDateTime *dtAccess,
                           const wxDateTime *dtMod,
-                          const wxDateTime *dtCreate)
+                          const wxDateTime *dtCreate) const
 {
 #if defined(__WIN32__)
     FILETIME ftAccess, ftCreate, ftWrite;
@@ -2350,7 +2368,7 @@
     return false;
 }
 
-bool wxFileName::Touch()
+bool wxFileName::Touch() const
 {
 #if defined(__UNIX_LIKE__)
     // under Unix touching file is simple: just pass NULL to utime()
@@ -2608,7 +2626,7 @@
     return false ;
 }
 
-bool wxFileName::MacGetTypeAndCreator( wxUint32 *type , wxUint32 *creator )
+bool wxFileName::MacGetTypeAndCreator( wxUint32 *type , wxUint32 *creator ) const
 {
     FSRef fsRef ;
     FSCatalogInfo catInfo;
Index: include/wx/filename.h
===================================================================
--- include/wx/filename.h	(revision 61008)
+++ include/wx/filename.h	(working copy)
@@ -243,10 +243,10 @@
         // (any of the pointers may be NULL)
     bool SetTimes(const wxDateTime *dtAccess,
                   const wxDateTime *dtMod,
-                  const wxDateTime *dtCreate);
+                  const wxDateTime *dtCreate) const;
 
         // set the access and modification times to the current moment
-    bool Touch();
+    bool Touch() const;
 
         // return the last access, last modification and create times
         // (any of the pointers may be NULL)
@@ -265,7 +265,7 @@
 
 #if defined( __WXOSX_MAC__ ) && wxOSX_USE_CARBON
     bool MacSetTypeAndCreator( wxUint32 type , wxUint32 creator ) ;
-    bool MacGetTypeAndCreator( wxUint32 *type , wxUint32 *creator ) ;
+    bool MacGetTypeAndCreator( wxUint32 *type , wxUint32 *creator ) const;
     // gets the 'common' type and creator for a certain extension
     static bool MacFindDefaultTypeAndCreator( const wxString& ext , wxUint32 *type , wxUint32 *creator ) ;
     // registers application defined extensions and their default type and creator
@@ -281,7 +281,7 @@
     static wxString GetCwd(const wxString& volume = wxEmptyString);
 
         // change the current working directory
-    bool SetCwd();
+    bool SetCwd() const;
     static bool SetCwd( const wxString &cwd );
 
         // get the value of user home (Unix only mainly)
@@ -358,7 +358,7 @@
         // the arguments
     bool GetShortcutTarget(const wxString& shortcutPath,
                            wxString& targetFilename,
-                           wxString* arguments = NULL);
+                           wxString* arguments = NULL) const;
 #endif
 
 #ifndef __WXWINCE__
Index: interface/wx/filename.h
===================================================================
--- interface/wx/filename.h	(revision 61008)
+++ interface/wx/filename.h	(working copy)
@@ -1035,7 +1035,7 @@
     /**
         Changes the current working directory.
     */
-    bool SetCwd();
+    bool SetCwd() const;
 
     /**
         Changes the current working directory.
@@ -1080,7 +1080,7 @@
     */
     bool SetTimes(const wxDateTime* dtAccess,
                   const wxDateTime* dtMod,
-                  const wxDateTime* dtCreate);
+                  const wxDateTime* dtCreate) const;
 
     /**
         Sets the volume specifier.
@@ -1162,7 +1162,7 @@
     /**
         Sets the access and modification times to the current moment.
     */
-    bool Touch();
+    bool Touch() const;
 
     /**
         Returns @true if the filenames are different. The string @e filenames
