Index: xh_sizer.cpp
===================================================================
RCS file: /home/wxcvs/wxWindows/contrib/src/xrc/xh_sizer.cpp,v
retrieving revision 1.6
diff -b -u -2 -r1.6 xh_sizer.cpp
--- xh_sizer.cpp	26 Jul 2002 22:09:04 -0000	1.6
+++ xh_sizer.cpp	7 Aug 2002 11:51:31 -0000
@@ -58,4 +58,5 @@
     XRC_ADD_STYLE(wxSHAPED);
     XRC_ADD_STYLE(wxSTRETCH_NOT);
+	XRC_ADD_STYLE(wxADJUST_MINSIZE);
 
     XRC_ADD_STYLE(wxALIGN_CENTER);

 	  	 
