Index: include/wx/propgrid/propgrid.h
===================================================================
--- include/wx/propgrid/propgrid.h	(revision 61671)
+++ include/wx/propgrid/propgrid.h	(working copy)
@@ -1916,7 +1916,10 @@
 
 inline void wxPGProperty::SetName( const wxString& newName )
 {
-    GetGrid()->SetPropertyName(this, newName);
+    if(GetGrid())
+        GetGrid()->SetPropertyName(this, newName);
+    else
+        DoSetName(newName);
 }
 
 inline bool wxPGProperty::Hide( bool hide, int flags )
