--- gsocket_orig.cpp	2006-02-10 09:51:31.155982776 +1100
+++ gsocket.cpp	2006-02-10 09:51:31.120988096 +1100
@@ -835,29 +835,26 @@
     }
 
+    /* Check for exceptions and errors (is this useful in Unices?) */
+    if (FD_ISSET(m_fd, &exceptfds))
+    {
+      m_establishing = false;
+      m_detected = GSOCK_LOST_FLAG;
+
+      /* LOST event: Abort any further processing */
+      return (GSOCK_LOST_FLAG & flags);
+    }
+
     /* Check for readability */
     if (FD_ISSET(m_fd, &readfds))
     {
-      char c;
-
-      if (!m_stream || recv(m_fd, &c, 1, MSG_PEEK) > 0)
-      {
-        result |= GSOCK_INPUT_FLAG;
-      }
-      else
-      {
         if (m_server && m_stream)
         {
-          result |= GSOCK_CONNECTION_FLAG;
-          m_detected |= GSOCK_CONNECTION_FLAG;
+            result |= GSOCK_CONNECTION_FLAG;
+            m_detected |= GSOCK_CONNECTION_FLAG;
         }
         else
         {
-          m_detected = GSOCK_LOST_FLAG;
-          m_establishing = false;
-
-          /* LOST event: Abort any further processing */
-          return (GSOCK_LOST_FLAG & flags);
+            result |= GSOCK_INPUT_FLAG;
         }
-      }
     }
 
@@ -893,14 +890,4 @@
     }
 
-    /* Check for exceptions and errors (is this useful in Unices?) */
-    if (FD_ISSET(m_fd, &exceptfds))
-    {
-      m_establishing = false;
-      m_detected = GSOCK_LOST_FLAG;
-
-      /* LOST event: Abort any further processing */
-      return (GSOCK_LOST_FLAG & flags);
-    }
-
     return (result & flags);
   }

 	  	 
