--- a/src/msw/dc.cpp
+++ b/src/msw/dc.cpp
@@ -2797,11 +2797,12 @@ static bool AlphaBlt(wxMSWDCImpl* dcDst,
             if ( data )
             {
                 wxAlphaPixelData::Iterator p(data);
-                for ( int y = 0; y < data.GetHeight(); y++ )
+                p.Offset(data, x, y);
+                for ( int _y = 0; _y < dstHeight; _y++ )
                 {
                     wxAlphaPixelData::Iterator rowStart = p;
 
-                    for ( int x = 0; x < data.GetWidth(); x++ )
+                    for ( int _x = 0; _x < dstWidth; _x++ )
                     {
                         // We choose to use wxALPHA_TRANSPARENT instead
                         // of perhaps more logical wxALPHA_OPAQUE here
