Index: samples/dialogs/dialogs.cpp
===================================================================
--- samples/dialogs/dialogs.cpp	(revision 69871)
+++ samples/dialogs/dialogs.cpp	(working copy)
@@ -2100,7 +2100,9 @@
 {
     static const int max = 100;
 
-    wxProgressDialog dialog("Progress dialog example",
+    wxProgressDialog dialog(
+    //wxGenericProgressDialog dialog(
+                            "Progress dialog example",
                             // "Reserve" enough space for the multiline
                             // messages below, we'll change it anyhow
                             // immediately in the loop below
@@ -2116,6 +2118,7 @@
                             wxPD_REMAINING_TIME |
                             wxPD_SMOOTH // - makes indeterminate mode bar on WinXP very small
                             );
+    wxASSERT(dialog.GetParent() == this);
 
     bool cont = true;
     for ( int i = 0; i <= max; i++ )
Index: src/generic/progdlgg.cpp
===================================================================
--- src/generic/progdlgg.cpp	(revision 69871)
+++ src/generic/progdlgg.cpp	(working copy)
@@ -153,10 +153,10 @@
                                       wxWindow *parent,
                                       int style )
 {
-    wxDialog::Create(GetParentForModalDialog(parent, style), wxID_ANY, title);
+    wxDialog::Create(GetParentForModalDialog(parent, GetWindowStyle()), wxID_ANY, title);
 
-    SetParent( GetParentForModalDialog(parent, style) );
-    SetTitle( title );
+    wxASSERT(GetParent() == GetParentForModalDialog(parent, GetWindowStyle()));
+    wxASSERT(GetTitle() == title);
 
     SetMaximum(maximum);
 
Index: src/msw/progdlg.cpp
===================================================================
--- src/msw/progdlg.cpp	(revision 69871)
+++ src/msw/progdlg.cpp	(working copy)
@@ -316,6 +316,7 @@
 #ifdef wxHAS_MSW_TASKDIALOG
     if ( HasNativeTaskDialog() )
     {
+        SetParent(GetParentForModalDialog(parent, GetWindowStyle()));
         SetMaximum(maximum);
 
         Show();
