Index: src/xrc/xh_treebk.cpp
===================================================================
--- src/xrc/xh_treebk.cpp	(revisione 68310)
+++ src/xrc/xh_treebk.cpp	(copia locale)
@@ -76,6 +76,25 @@
 
         CreateChildren(m_tbk, true/*only this handler*/);
 
+        wxXmlNode *node = GetParamNode("object");
+        int pageIndex = 0;
+        for (unsigned int i = 0; i < m_tbk->GetPageCount(); i++)
+        {
+            wxWindow *page = wxDynamicCast(m_tbk->GetPage(i), wxWindow);
+            if (page)
+            {
+                wxXmlNode *child = node->GetChildren();
+                while (child)
+                {
+                    if (child->GetName() == "expanded" && child->GetNodeContent() == "1")
+                        m_tbk->ExpandNode(pageIndex, true);
+
+                    child = child->GetNext();
+                }
+                pageIndex++;
+            }
+        }
+
         m_treeContext = old_treeContext;
         m_isInside = old_ins;
         m_tbk = old_par;
Index: docs/doxygen/overviews/xrc_format.h
===================================================================
--- docs/doxygen/overviews/xrc_format.h	(revisione 68310)
+++ docs/doxygen/overviews/xrc_format.h	(copia locale)
@@ -1730,6 +1730,8 @@
     into the image list.}
 @row3col{selected, @ref overview_xrcformat_type_bool,
      Is the page selected initially (only one page can be selected; default: 0)?}
+@row3col{expanded, @ref overview_xrcformat_type_bool,
+     Expands the current treebookpage node.}
 @endTable
 
 Each @c treebookpage has exactly one non-toplevel window as its child.
Index: samples/xrc/rc/controls.xrc
===================================================================
--- samples/xrc/rc/controls.xrc	(revisione 68310)
+++ samples/xrc/rc/controls.xrc	(copia locale)
@@ -1147,9 +1147,18 @@
                                 <object class="wxTreebook" name="controls_treebook">
                                     <size>350,280</size>
                                     <style>wxSUNKEN_BORDER</style>
+                                    <imagelist>
+                                        <size>16,16</size>
+                                        <bitmap stock_id="wxART_HELP_BOOK"/>
+                                        <bitmap stock_id="wxART_QUESTION"/>
+                                        <bitmap stock_id="wxART_INFORMATION"/>
+                                        <bitmap stock_id="wxART_GO_HOME"/>
+                                    </imagelist>
                                     <object class="treebookpage">
                                         <label>Page 1</label>
                                         <depth>0</depth>
+                                        <image>0</image>
+                                        <expanded>1</expanded>
                                         <object class="wxButton" name="controls_treebook_button1">
                                             <size>200,180</size>
                                             <label>Button N1</label>
@@ -1158,10 +1167,13 @@
                                     <object class="treebookpage">
                                         <label>Empty Page 2</label>
                                         <depth>1</depth>
+                                        <image>1</image>
+                                        <expanded>1</expanded>
                                     </object>
                                     <object class="treebookpage">
                                         <label>Page 3</label>
                                         <depth>2</depth>
+                                        <image>2</image>
                                         <object class="wxButton" name="controls_treebook_button3">
                                             <size>200,180</size>
                                             <label>Button N3</label>
@@ -1170,6 +1182,7 @@
                                     <object class="treebookpage">
                                         <label>Page 4</label>
                                         <depth>1</depth>
+                                        <image>3</image>
                                         <object class="wxButton" name="controls_treebook_button4">
                                             <size>200,180</size>
                                             <label>Button N4</label>
