Index: src/common/strconv.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/common/strconv.cpp,v
retrieving revision 1.208
diff --unified -r1.208 strconv.cpp
--- src/common/strconv.cpp	2006/05/06 18:53:43	1.208
+++ src/common/strconv.cpp	2006/05/22 02:46:18
@@ -1997,6 +1997,7 @@
     }
 
     wxMBConv_win32(const wxMBConv_win32& conv)
+        : wxMBConv()
     {
         m_CodePage = conv.m_CodePage;
         m_minMBCharWidth = conv.m_minMBCharWidth;

 	  	 
