Index: include/wx/frame.h
===================================================================
--- include/wx/frame.h	(revision 72864)
+++ include/wx/frame.h	(working copy)
@@ -206,6 +206,8 @@
     virtual void AttachMenuBar(wxMenuBar *menubar);
 
     wxMenuBar *m_frameMenuBar;
+
+    static bool ms_idleMenuUpdates;
 #endif // wxUSE_MENUS
 
 #if wxUSE_STATUSBAR && (wxUSE_MENUS || wxUSE_TOOLBAR)
Index: src/common/framecmn.cpp
===================================================================
--- src/common/framecmn.cpp	(revision 72864)
+++ src/common/framecmn.cpp	(working copy)
@@ -40,16 +40,18 @@
 // event table
 // ----------------------------------------------------------------------------
 
-#if wxUSE_MENUS && wxUSE_STATUSBAR
-
+#if wxUSE_MENUS
+#if wxUSE_STATUSBAR
 BEGIN_EVENT_TABLE(wxFrameBase, wxTopLevelWindow)
     EVT_MENU_OPEN(wxFrameBase::OnMenuOpen)
     EVT_MENU_CLOSE(wxFrameBase::OnMenuClose)
 
     EVT_MENU_HIGHLIGHT_ALL(wxFrameBase::OnMenuHighlight)
 END_EVENT_TABLE()
+#endif // wxUSE_STATUSBAR
 
-#endif // wxUSE_MENUS && wxUSE_STATUSBAR
+bool wxFrameBase::ms_idleMenuUpdates = wxUSE_IDLEMENUUPDATES != 0;
+#endif // wxUSE_MENUS
 
 // ============================================================================
 // implementation
@@ -300,9 +302,7 @@
         // If coming from an idle event, we only want to update the menus if
         // we're in the wxUSE_IDLEMENUUPDATES configuration, otherwise they
         // will be update when the menu is opened later
-#if !wxUSE_IDLEMENUUPDATES
-        if ( !(flags & wxUPDATE_UI_FROMIDLE) )
-#endif // wxUSE_IDLEMENUUPDATES
+        if (!(flags & wxUPDATE_UI_FROMIDLE) || ms_idleMenuUpdates)
             DoMenuUpdates();
     }
 #endif // wxUSE_MENUS
@@ -323,12 +323,11 @@
 
 void wxFrameBase::OnMenuOpen(wxMenuEvent& event)
 {
-#if wxUSE_IDLEMENUUPDATES
-    wxUnusedVar(event);
-#else // !wxUSE_IDLEMENUUPDATES
-    // as we didn't update the menus from idle time, do it now
-    DoMenuUpdates(event.GetMenu());
-#endif // wxUSE_IDLEMENUUPDATES/!wxUSE_IDLEMENUUPDATES
+    if (!ms_idleMenuUpdates)
+    {
+        // as we didn't update the menus from idle time, do it now
+        DoMenuUpdates(event.GetMenu());
+    }
 }
 
 void wxFrameBase::OnMenuClose(wxMenuEvent& WXUNUSED(event))
@@ -343,8 +342,8 @@
 {
     wxTopLevelWindow::OnInternalIdle();
 
-#if wxUSE_MENUS && wxUSE_IDLEMENUUPDATES
-    if (wxUpdateUIEvent::CanUpdate(this))
+#if wxUSE_MENUS
+    if (ms_idleMenuUpdates && wxUpdateUIEvent::CanUpdate(this))
         DoMenuUpdates();
 #endif
 }
Index: src/gtk/frame.cpp
===================================================================
--- src/gtk/frame.cpp	(revision 72864)
+++ src/gtk/frame.cpp	(working copy)
@@ -19,6 +19,9 @@
 #endif // WX_PRECOMP
 
 #include <gtk/gtk.h>
+#ifdef GDK_WINDOWING_X11
+    #include <gdk/gdkx.h>
+#endif
 
 #if wxUSE_LIBHILDON
     #include <hildon-widgets/hildon-window.h>
@@ -53,7 +56,27 @@
                       long style,
                       const wxString &name )
 {
-    return wxFrameBase::Create(parent, id, title, pos, sizeOrig, style, name);
+    if (!wxFrameBase::Create(parent, id, title, pos, sizeOrig, style, name))
+        return false;
+
+#if wxUSE_MENUS && defined(GDK_WINDOWING_X11)
+    static bool once;
+    if (!once)
+    {
+        once = true;
+        GdkScreen* screen = gtk_window_get_screen(GTK_WINDOW(m_widget));
+#ifdef __WXGTK3__
+        if (GDK_IS_X11_SCREEN(screen))
+#endif
+        {
+            const char* wmname = gdk_x11_screen_get_window_manager_name(screen);
+            if (strcmp(wmname, "Unity") == 0)
+                ms_idleMenuUpdates = true;
+        }
+    }
+#endif // wxUSE_MENUS && defined(GDK_WINDOWING_X11)
+
+    return true;
 }
 
 wxFrame::~wxFrame()
