Index: src/gtk/dataview.cpp
===================================================================
--- src/gtk/dataview.cpp	(revision 68647)
+++ src/gtk/dataview.cpp	(working copy)
@@ -3805,7 +3805,7 @@
 
         iter->stamp = m_gtk_model->stamp;
         // user_data is just the index +1
-        iter->user_data = (gpointer) (i+1);
+        iter->user_data = (gpointer) long(i+1);
 
         return TRUE;
     }
@@ -3908,7 +3908,7 @@
         }
 
         // user_data is just the index +1 (+2 because we need the next)
-        iter->user_data = (gpointer) (n+2);
+        iter->user_data = (gpointer) long(n+2);
     }
     else
     {
@@ -4062,7 +4062,7 @@
 
         iter->stamp = m_gtk_model->stamp;
         // user_data is just the index +1
-        iter->user_data = (gpointer) (n+1);
+        iter->user_data = (gpointer) long(n+1);
 
         return TRUE;
     }
