commit 91854aa1ca84e76d2539e2188b71ec73cfa34533
Author: Jens Lody <jens@codeblocks.org>
Date:   Tue Nov 6 10:39:49 2012 +0100

    * windows build fixes

diff --git a/include/wx/aui/auibook.h b/include/wx/aui/auibook.h
index ee641e3..d02a84d 100644
--- a/include/wx/aui/auibook.h
+++ b/include/wx/aui/auibook.h
@@ -26,6 +26,7 @@
 #if wxUSE_AUI
 
 #include "wx/bookctrl.h"
+#include "wx/containr.h"
 #include "wx/aui/framemanager.h"
 
 
diff --git a/src/aui/framemanager.cpp b/src/aui/framemanager.cpp
index 51cfff1..72a0b4c 100644
--- a/src/aui/framemanager.cpp
+++ b/src/aui/framemanager.cpp
@@ -40,6 +40,7 @@
     #include "wx/app.h"
     #include "wx/dcclient.h"
     #include "wx/dcscreen.h"
+    #include "wx/dcmemory.h"
     #include "wx/toolbar.h"
     #include "wx/image.h"
     #include "wx/statusbr.h"
