diff --git a/src/aui/framemanager.cpp b/src/aui/framemanager.cpp
index 23deb19..a4ea2d6 100644
--- a/src/aui/framemanager.cpp
+++ b/src/aui/framemanager.cpp
@@ -749,13 +749,12 @@ int wxAuiManager::SetActivePane(wxWindow* activePane)
                 activePaneInfo = &pane;
                 wxAuiTabContainer* ctrl;
                 int ctrlIndex;
-                FindTab(pane.GetWindow(), &ctrl, &ctrlIndex);
-                ctrl->SetActivePage(pane.GetWindow());
-            }
-            //else
-            {
-                pane.SetFlag(wxAuiPaneInfo::optionActive,true);
+                if(FindTab(pane.GetWindow(), &ctrl, &ctrlIndex))
+                {
+                    ctrl->SetActivePage(pane.GetWindow());
+                }
             }
+            pane.SetFlag(wxAuiPaneInfo::optionActive,true);
         }
     }
 
@@ -5669,20 +5668,15 @@ void wxAuiManager::OnLeftUp(wxMouseEvent& evt)
             int buttonid=0;
             if(m_actionPart->type == wxAuiDockUIPart::typePaneTab)
             {
-                passHitTest = (m_actionPart->m_tab_container->ButtonHitTest(evt.m_x,evt.m_y,&m_hoverButton2));
+                wxAuiTabContainerButton* hitbutton;
+                if(m_actionPart->m_tab_container->ButtonHitTest(evt.m_x,evt.m_y,&hitbutton))
+                {
+                    passHitTest=( (hitbutton==m_hoverButton2) && (m_hoverButton2->curState==wxAUI_BUTTON_STATE_PRESSED) );
+                }
+
                 buttonid=m_hoverButton2->id;
                 m_hoverButton2->curState = wxAUI_BUTTON_STATE_NORMAL;
                 m_hoverButton2=NULL;
-
-                // If we are a wxAuiNotebook then we must fire off a wxEVT_COMMAND_AUINOTEBOOK_BUTTON event to notify user of change.
-                if(wxDynamicCast(GetManagedWindow(),wxAuiNotebook))
-                {
-                    wxAuiNotebookEvent e(wxEVT_COMMAND_AUINOTEBOOK_BUTTON, GetManagedWindow()->GetId());
-                    e.SetSelection(GetAllPanes().Index(pane));
-                    e.SetInt(buttonid);
-                    e.SetEventObject(GetManagedWindow());
-                    GetManagedWindow()->GetEventHandler()->ProcessEvent(e);
-                }
             }
             else
             {
@@ -5692,6 +5686,16 @@ void wxAuiManager::OnLeftUp(wxMouseEvent& evt)
             }
             if (passHitTest)
             {
+                // If we are a wxAuiNotebook then we must fire off a wxEVT_COMMAND_AUINOTEBOOK_BUTTON event to notify user of change.
+                if(wxDynamicCast(GetManagedWindow(),wxAuiNotebook))
+                {
+                    wxAuiNotebookEvent e(wxEVT_COMMAND_AUINOTEBOOK_BUTTON, GetManagedWindow()->GetId());
+                    e.SetSelection(GetAllPanes().Index(pane));
+                    e.SetInt(buttonid);
+                    e.SetEventObject(GetManagedWindow());
+                    GetManagedWindow()->GetEventHandler()->ProcessEvent(e);
+                }
+                
                 // fire button-click event
                 wxAuiManagerEvent e(wxEVT_AUI_PANE_BUTTON);
                 e.SetManager(this);
@@ -6417,7 +6421,6 @@ void wxAuiManager::OnMotion(wxMouseEvent& evt)
                 m_hoverButton2->curState = wxAUI_BUTTON_STATE_NORMAL;
             }
             Repaint();
-            m_hoverButton2 = NULL;
         }
         else
         {
