commit ca47200743638bd2ba04b8b35a2c90a141658052
Author: Jens Lody <jens@codeblocks.org>
Date:   Sat Oct 27 03:18:29 2012 +0200

    * fix for fixed width/height notebook-tabs

diff --git a/src/aui/framemanager.cpp b/src/aui/framemanager.cpp
index e137cfe..0b4869a 100644
--- a/src/aui/framemanager.cpp
+++ b/src/aui/framemanager.cpp
@@ -2379,26 +2379,21 @@ int wxAuiManager::GetNotebookFlags()
 void wxAuiManager::LayoutAddNotebook(wxAuiTabArt* tabArt,wxAuiTabContainer* notebookContainer,wxSizer* notebookSizer,wxAuiDockUIPart& part,wxAuiDockInfo& dock,wxAuiDockUIPartArray& uiparts,wxAuiTabContainerPointerArray& tabContainerRecalcList,wxAuiSizerItemPointerArray& tabContainerRecalcSizers,wxAuiPaneInfo* pane,int orient)
 {
     wxSizerItem* sizerItem;
+
+    wxSize tabSize = tabArt->GetBestTabSize(m_frame, notebookContainer->GetPages(), wxSize(16,16));
+
     if(orient==wxHORIZONTAL)
     {
-        //fixme: (MJM) This should be set via the art provider not hardcoded, temporary hardcoding while we iron out some issues with the tab art providers.
-        sizerItem = notebookSizer->Add(m_art->GetMetric(wxAUI_DOCKART_SASH_SIZE), /*tabArt->m_tabCtrlHeight*/notebookTabHeight, 0, wxEXPAND);
-        if(!HasFlag(wxAUI_MGR_NB_TAB_FIXED_HEIGHT))
-        {
-            tabContainerRecalcList.Add(notebookContainer);
-            tabContainerRecalcSizers.Add(sizerItem);
-        }
+        sizerItem = notebookSizer->Add(m_art->GetMetric(wxAUI_DOCKART_SASH_SIZE), tabSize.y, 0, wxEXPAND);
     }
     else
     {
-        sizerItem = notebookSizer->Add(tabArt->m_fixedTabSize, m_art->GetMetric(wxAUI_DOCKART_SASH_SIZE), 0, wxEXPAND);
-        if(!HasFlag(wxAUI_MGR_NB_TAB_FIXED_WIDTH))
-        {
-            tabContainerRecalcList.Add(notebookContainer);
-            tabContainerRecalcSizers.Add(sizerItem);
-        }
+        sizerItem = notebookSizer->Add(tabSize.x, m_art->GetMetric(wxAUI_DOCKART_SASH_SIZE), 0, wxEXPAND);
     }
 
+    tabContainerRecalcList.Add(notebookContainer);
+    tabContainerRecalcSizers.Add(sizerItem);
+
     part.type = wxAuiDockUIPart::typePaneTab;
     part.dock = &dock;
     part.pane = pane;
diff --git a/src/aui/tabart.cpp b/src/aui/tabart.cpp
index 1f08abf..c66d115 100644
--- a/src/aui/tabart.cpp
+++ b/src/aui/tabart.cpp
@@ -262,10 +262,14 @@ void wxAuiGenericTabArt::SetSizingInfo(const wxSize& tabCtrlSize, size_t tabCoun
 
         int tot_height = (int)tabCtrlSize.y - GetIndentSize() - 4;
 
+        int btn_height = 0;
+
         if (m_flags & wxAUI_NB_CLOSE_BUTTON)
-            tot_height -= m_activeCloseBmp.GetHeight();
+            btn_height = m_activeCloseBmp.GetHeight();
         if (m_flags & wxAUI_NB_WINDOWLIST_BUTTON)
-            tot_height -= m_activeWindowListBmp.GetHeight();
+            btn_height = wxMax(btn_height,m_activeWindowListBmp.GetHeight());
+
+        tot_height -= btn_height;
 
         if (tabCount > 0)
         {
@@ -958,6 +962,8 @@ wxSize wxAuiGenericTabArt::GetBestTabSize(wxWindow* wnd, const wxAuiPaneInfoPtrA
     wxClientDC dc(wnd);
     dc.SetFont(m_measuringFont);
 
+    SetSizingInfo(wnd->GetSize(), pages.GetCount());
+
     // sometimes a standard bitmap size needs to be enforced, especially
     // if some tabs have bitmaps and others don't.  This is important because
     // it prevents the tab control from resizing when tabs are added.
@@ -1119,10 +1125,14 @@ void wxAuiSimpleTabArt::SetSizingInfo(const wxSize& tabCtrlSize, size_t tabCount
 
         int tot_height = (int)tabCtrlSize.y - GetIndentSize() - 4;
 
+        int btn_height = 0;
+
         if (m_flags & wxAUI_NB_CLOSE_BUTTON)
-            tot_height -= m_activeCloseBmp.GetHeight();
+            btn_height = m_activeCloseBmp.GetHeight();
         if (m_flags & wxAUI_NB_WINDOWLIST_BUTTON)
-            tot_height -= m_activeWindowListBmp.GetHeight();
+            btn_height = wxMax(btn_height,m_activeWindowListBmp.GetHeight());
+
+        tot_height -= btn_height;
 
         if (tabCount > 0)
         {
@@ -1428,14 +1438,14 @@ wxSize wxAuiSimpleTabArt::GetTabSize(wxDC& dc, wxWindow* WXUNUSED(wnd), const wx
     {
         if (HasFlag(wxAUI_MGR_NB_TAB_FIXED_WIDTH))
         {
-            tabWidth = m_fixedTabSize;
+            tabWidth = m_fixedTabSize + tabHeight/2 ;
         }
         *extent = tabWidth - (tabHeight/2) - 1;
     }
     else
     {
         if (HasFlag(wxAUI_NB_TAB_FIXED_HEIGHT))
-             tabWidth = m_fixedTabSize;
+             tabHeight = m_fixedTabSize - 2;
         tabWidth += 16;
         *extent = tabHeight + 2;
     }
@@ -1573,6 +1583,8 @@ wxSize wxAuiSimpleTabArt::GetBestTabSize(wxWindow* wnd, const wxAuiPaneInfoPtrAr
     wxClientDC dc(wnd);
     dc.SetFont(m_measuringFont);
 
+    SetSizingInfo(wnd->GetSize(), pages.GetCount());
+
     wxSize maxSize;
     size_t i, pageCount = pages.GetCount();
     for (i = 0; i < pageCount; ++i)
