diff -rbu2 ORIGINAL.wxGTK-2.3.2/include/wx/app.h wxGTK-2.3.2/include/wx/app.h
--- ORIGINAL.wxGTK-2.3.2/include/wx/app.h	2001-12-11 03:07:01.000000000 +0800
+++ wxGTK-2.3.2/include/wx/app.h	2002-06-21 10:31:25.560004000 +0800
@@ -422,14 +422,6 @@
 // Post a message to the given eventhandler which will be processed during the
 // next event loop iteration
-inline void wxPostEvent(wxEvtHandler *dest, wxEvent& event)
-{
-    wxCHECK_RET( dest, wxT("need an object to post event to in wxPostEvent") );
+extern void WXDLLEXPORT wxPostEvent(wxEvtHandler *dest, wxEvent& event);
 
-#if wxUSE_GUI
-    dest->AddPendingEvent(event);
-#else
-    dest->ProcessEvent(event);
-#endif // wxUSE_GUI
-}
 
 // console applications may avoid using DECLARE_APP and IMPLEMENT_APP macros
diff -rbu2 ORIGINAL.wxGTK-2.3.2/src/common/appcmn.cpp wxGTK-2.3.2/src/common/appcmn.cpp
--- ORIGINAL.wxGTK-2.3.2/src/common/appcmn.cpp	2001-12-11 03:07:03.000000000 +0800
+++ wxGTK-2.3.2/src/common/appcmn.cpp	2002-06-21 10:28:55.480006000 +0800
@@ -55,4 +55,19 @@
 
 // ----------------------------------------------------------------------------
+// Post a message to the given eventhandler which will be processed during the
+// next event loop iteration
+// ----------------------------------------------------------------------------
+void wxPostEvent(wxEvtHandler *dest, wxEvent& event)
+{
+    wxCHECK_RET( dest, wxT("need an object to post event to in wxPostEvent") );
+
+#if wxUSE_GUI
+    dest->AddPendingEvent(event);
+#else
+    dest->ProcessEvent(event);
+#endif // wxUSE_GUI
+}
+
+// ----------------------------------------------------------------------------
 // initialization and termination
 // ----------------------------------------------------------------------------

 	  	 
