Index: wxWidgets-2.8/src/generic/hyperlink.cpp
===================================================================
--- wxWidgets-2.8/src/generic/hyperlink.cpp	(revision 61457)
+++ wxWidgets-2.8/src/generic/hyperlink.cpp	(working copy)
@@ -108,6 +108,7 @@
     // colours
     m_normalColour = *wxBLUE;
     m_hoverColour = *wxRED;
+    m_visitedColour = *wxLIGHT_GREY;
     SetForegroundColour(m_normalColour);
 
     // by default the font of an hyperlink control is underlined
