--- ../wxWidgets/include/wx/mac/carbon/cursor.h	2004-05-23 13:51:02.000000000 -0700
+++ ./include/wx/mac/carbon/cursor.h	2004-11-24 15:01:52.376920976 -0800
@@ -68,8 +68,8 @@
   virtual bool Ok() const { return (m_refData != NULL && ( M_CURSORDATA->m_hCursor != NULL || M_CURSORDATA->m_themeCursor != -1 ) ) ; }
 
   inline wxCursor& operator = (const wxCursor& cursor) { if (*this == cursor) return (*this); Ref(cursor); return *this; }
-  inline bool operator == (const wxCursor& cursor) { return m_refData == cursor.m_refData; }
-  inline bool operator != (const wxCursor& cursor) { return m_refData != cursor.m_refData; }
+  inline bool operator == (const wxCursor& cursor) const { return m_refData == cursor.m_refData; }
+  inline bool operator != (const wxCursor& cursor) const { return m_refData != cursor.m_refData; }
 
     void MacInstall() const ;
 

 	  	 
