 samples/minimal/minimal.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/samples/minimal/minimal.cpp b/samples/minimal/minimal.cpp
index a78e462..71db62e 100644
--- a/samples/minimal/minimal.cpp
+++ b/samples/minimal/minimal.cpp
@@ -167,6 +167,11 @@ MyFrame::MyFrame(const wxString& title)
     SetMenuBar(menuBar);
 #endif // wxUSE_MENUS
 
+    wxToolBar* tbar = CreateToolBar(wxTB_VERTICAL);
+    tbar->AddControl(new wxStaticText(tbar, wxID_HIGHEST + 1, _T("Txt 1")));
+    tbar->AddControl(new wxStaticText(tbar, wxID_HIGHEST + 2, _T("Txt 2")));
+    tbar->Realize();
+
 #if wxUSE_STATUSBAR
     // create a status bar just for fun (by default with 1 pane only)
     CreateStatusBar(2);
