diff -bu1 -r wxWindows/./include/wx/log.h wxWin/./include/wx/log.h
--- wxWindows/./include/wx/log.h	2003-12-01 07:53:22.000000000 +0100
+++ wxWin/./include/wx/log.h	2004-01-14 16:12:43.000000000 +0100
@@ -540,4 +540,9 @@
 #else   //!debug
+    #ifdef __BORLANDC__
+        #define wxLogApiError(api, rc){}
+        #define wxLogLastError(api){}
+    #else // !Borland
     inline void wxLogApiError(const wxChar *, long) { }
     inline void wxLogLastError(const wxChar *) { }
+    #endif
 #endif  //debug/!debug
diff -bu1 -r wxWindows/./src/common/dlgcmn.cpp wxWin/./src/common/dlgcmn.cpp
--- wxWindows/./src/common/dlgcmn.cpp	2004-01-12 10:20:35.000000000 +0100
+++ wxWin/./src/common/dlgcmn.cpp	2004-01-14 16:12:40.000000000 +0100
@@ -203,3 +203,2 @@
     wxButton *ok = (wxButton *) NULL;
-    wxButton *cancel = (wxButton *) NULL;
     wxButton *yes = (wxButton *) NULL;
@@ -245,3 +244,3 @@
     {
-        cancel = new wxButton( this, wxID_CANCEL, _("Cancel"),wxDefaultPosition,wxDefaultSize,wxCLIP_SIBLINGS );
+        wxButton *cancel = new wxButton( this, wxID_CANCEL, _("Cancel"),wxDefaultPosition,wxDefaultSize,wxCLIP_SIBLINGS );
         inner_rest->Add( cancel, 0, wxLEFT|wxRIGHT, margin );
diff -bu1 -r wxWindows/./src/common/framecmn.cpp wxWin/./src/common/framecmn.cpp
--- wxWindows/./src/common/framecmn.cpp	2003-12-29 07:51:16.000000000 +0100
+++ wxWin/./src/common/framecmn.cpp	2004-01-14 16:13:29.000000000 +0100
@@ -255,3 +255,7 @@
 
+#if !wxUSE_IDLEMENUUPDATES
 void wxFrameBase::OnMenuOpen(wxMenuEvent& event)
+#else
+void wxFrameBase::OnMenuOpen(wxMenuEvent& WXUNUSED(event))
+#endif
 {
diff -bu1 -r wxWindows/./src/msw/evtloop.cpp wxWin/./src/msw/evtloop.cpp
--- wxWindows/./src/msw/evtloop.cpp	2003-12-23 17:43:36.000000000 +0100
+++ wxWin/./src/msw/evtloop.cpp	2004-01-14 17:17:31.000000000 +0100
@@ -335,2 +335,3 @@
                     throw;  
+                default:
                     break;

 	  	 
