--- wxGTK-2.8.12.orig/src/gtk/stattext.cpp	2011-03-22 13:19:00.000000000 +0100
+++ wxGTK-2.8.12.new/src/gtk/stattext.cpp	2011-11-30 13:59:16.000000000 +0100
@@ -87,7 +87,8 @@ bool wxStaticText::Create(wxWindow *pare
     static const float labelAlignments[] = { 0.0, 1.0, 0.5 };
     gtk_misc_set_alignment(GTK_MISC(m_widget), labelAlignments[justify], 0.0);
 
-    gtk_label_set_line_wrap( GTK_LABEL(m_widget), TRUE );
+    if ( justify == GTK_JUSTIFY_LEFT )
+        gtk_label_set_line_wrap( GTK_LABEL(m_widget), TRUE );
 
     m_parent->DoAddChild( this );
 
@@ -159,6 +160,7 @@ wxSize wxStaticText::DoGetBestSize() con
     // gtk_label_set_line_wrap() from here is a bad idea as it queues another
     // size request by calling gtk_widget_queue_resize() and we end up in
     // infinite loop sometimes (notably when the control is in a toolbar)
+    bool wrap = GTK_LABEL(m_widget)->wrap;
     GTK_LABEL(m_widget)->wrap = FALSE;
 
     GtkRequisition req;
@@ -167,7 +169,7 @@ wxSize wxStaticText::DoGetBestSize() con
     (* GTK_WIDGET_CLASS( GTK_OBJECT_GET_CLASS(m_widget) )->size_request )
         (m_widget, &req );
 
-    GTK_LABEL(m_widget)->wrap = TRUE; // restore old value
+    GTK_LABEL(m_widget)->wrap = wrap; // restore old value
 
     // Adding 1 to width to workaround GTK sometimes wrapping the text needlessly
     return wxSize (req.width+1, req.height);
