Index: gsocket.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/unix/gsocket.cpp,v
retrieving revision 1.33
diff -u -4 -r1.33 gsocket.cpp
--- gsocket.cpp	2005/02/04 05:52:32	1.33
+++ gsocket.cpp	2005/03/04 02:44:36
@@ -966,9 +966,11 @@
     if (FD_ISSET(m_fd, &readfds))
     {
       char c;
 
-      if (recv(m_fd, &c, 1, MSG_PEEK) > 0)
+      int num = recv(m_fd, &c, 1, MSG_PEEK | MSG_NOSIGNAL);
+
+      if (num > 0)
       {
         result |= GSOCK_INPUT_FLAG;
       }
       else
@@ -977,9 +979,10 @@
         {
           result |= GSOCK_CONNECTION_FLAG;
           m_detected |= GSOCK_CONNECTION_FLAG;
         }
-        else
+        /* Only close if we really aren't connected */
+        else if (num == 0 || errno == ENOTCONN)
         {
           m_detected = GSOCK_LOST_FLAG;
           m_establishing = false;
 
@@ -1278,9 +1281,9 @@
 }
 
 int GSocket::Recv_Stream(char *buffer, int size)
 {
-  return recv(m_fd, buffer, size, 0);
+  return recv(m_fd, buffer, size,  MSG_NOSIGNAL);
 }
 
 int GSocket::Recv_Dgram(char *buffer, int size)
 {
@@ -1323,12 +1326,12 @@
   int ret;
 
 #ifndef __VISAGECPP__
   MASK_SIGNAL();
-  ret = send(m_fd, buffer, size, 0);
+  ret = send(m_fd, buffer, size,  MSG_NOSIGNAL);
   UNMASK_SIGNAL();
 #else
-  ret = send(m_fd, (char *)buffer, size, 0);
+  ret = send(m_fd, (char *)buffer, size, MSG_NOSIGNAL);
 #endif
 
   return ret;
 }
@@ -1369,8 +1372,14 @@
 void GSocket::Detected_Read()
 {
   char c;
 
+  /* Safeguard against straggling call to Detected_Read */
+  if (m_fd == INVALID_SOCKET)
+  {
+    return;
+  }
+
   /* If we have already detected a LOST event, then don't try
    * to do any further processing.
    */
   if ((m_detected & GSOCK_LOST_FLAG) != 0)
@@ -1380,10 +1389,12 @@
     CALL_CALLBACK(this, GSOCK_LOST);
     Shutdown();
     return;
   }
+
+  int num =  recv(m_fd, &c, 1, MSG_PEEK | MSG_NOSIGNAL);
 
-  if (recv(m_fd, &c, 1, MSG_PEEK) > 0)
+  if (num > 0)
   {
     CALL_CALLBACK(this, GSOCK_INPUT);
   }
   else
@@ -1391,9 +1402,10 @@
     if (m_server && m_stream)
     {
       CALL_CALLBACK(this, GSOCK_CONNECTION);
     }
-    else
+    /* Only close if we really aren't connected */
+    else if (num == 0 || errno == ENOTCONN)
     {
       CALL_CALLBACK(this, GSOCK_LOST);
       Shutdown();
     }

 	  	 
