diff -rbu2P /wx232/contrib/include/wx/xrc/xh_all.h /Programme/wx2/contrib/include/wx/xrc/xh_all.h
--- /wx232/contrib/include/wx/xrc/xh_all.h	Sat Jun  9 23:13:34 2001
+++ /Programme/wx2/contrib/include/wx/xrc/xh_all.h	Sun Jan 20 16:04:24 2002
@@ -45,4 +45,5 @@
 #include "wx/xrc/xh_bmp.h"
 #include "wx/xrc/xh_unkwn.h"
+#include "wx/xrc/xh_gdctl.h"
 #include "wx/xrc/xh_frame.h"
 
diff -rbu2P /wx232/contrib/include/wx/xrc/xh_gdctl.h /Programme/wx2/contrib/include/wx/xrc/xh_gdctl.h
--- /wx232/contrib/include/wx/xrc/xh_gdctl.h	Thu Jan  1 01:00:00 1970
+++ /Programme/wx2/contrib/include/wx/xrc/xh_gdctl.h	Sun Jan 20 15:06:56 2002
@@ -0,0 +1,29 @@
+/////////////////////////////////////////////////////////////////////////////
+// Name:        xh_gdctl.h
+// Purpose:     XML resource handler for wxGenericDirCtrl
+// Author:      Markus Greither
+// Created:     2002/01/20
+// RCS-ID:
+// Copyright:   (c) 2002 Markus Greither
+// Licence:     wxWindows licence
+/////////////////////////////////////////////////////////////////////////////
+
+#ifndef _WX_XH_GDCTL_H_
+#define _WX_XH_GDCTL_H_
+
+#ifdef __GNUG__
+#pragma interface "xh_gdctl.h"
+#endif
+
+#include "wx/xrc/xmlres.h"
+
+class WXXMLDLLEXPORT wxGenericDirCtrlXmlHandler : public wxXmlResourceHandler
+{
+public:
+    wxGenericDirCtrlXmlHandler();
+    virtual wxObject *DoCreateResource();
+    virtual bool CanHandle(wxXmlNode *node);
+};
+
+
+#endif // _WX_XH_GDCTL_H_
diff -rbu2P /wx232/contrib/src/xrc/makefile.b32 /Programme/wx2/contrib/src/xrc/makefile.b32
--- /wx232/contrib/src/xrc/makefile.b32	Sat Jun  9 23:42:50 2001
+++ /Programme/wx2/contrib/src/xrc/makefile.b32	Sun Jan 20 19:22:54 2002
@@ -28,5 +28,5 @@
         xh_text.obj xh_listb.obj xh_toolb.obj xh_stlin.obj xh_bmp.obj \
         xh_bmpbt.obj xh_cald.obj xh_listc.obj xh_scrol.obj xh_stbox.obj \
-        xh_tree.obj xh_unkwn.obj xh_frame.obj
+        xh_tree.obj xh_unkwn.obj xh_frame.obj xh_gdctl.obj
 
 !include $(WXDIR)\src\makelib.b32
diff -rbu2P /wx232/contrib/src/xrc/makefile.g95 /Programme/wx2/contrib/src/xrc/makefile.g95
--- /wx232/contrib/src/xrc/makefile.g95	Sat Jun  9 23:42:50 2001
+++ /Programme/wx2/contrib/src/xrc/makefile.g95	Sun Jan 20 19:14:34 2002
@@ -29,5 +29,5 @@
         xh_text.o xh_listb.o xh_toolb.o xh_stlin.o xh_bmp.o xh_unkwn.o \
         xh_bmpbt.o xh_cald.o xh_listc.o xh_scrol.o xh_stbox.o xh_tree.o \
-        xh_frame.o
+        xh_frame.o xh_gdctl.o
 
 include $(WXDIR)/src/makelib.g95
diff -rbu2P /wx232/contrib/src/xrc/xh_gdctl.cpp /Programme/wx2/contrib/src/xrc/xh_gdctl.cpp
--- /wx232/contrib/src/xrc/xh_gdctl.cpp	Thu Jan  1 01:00:00 1970
+++ /Programme/wx2/contrib/src/xrc/xh_gdctl.cpp	Sun Jan 20 16:11:58 2002
@@ -0,0 +1,60 @@
+/////////////////////////////////////////////////////////////////////////////
+// Name:        xh_gdctl.cpp
+// Purpose:     XRC resource for wxGenericDirCtrl
+// Author:      Markus Greither
+// Created:     2002/01/20
+// RCS-ID:
+// Copyright:   (c) 2002 Markus Greither
+// Licence:     wxWindows licence
+/////////////////////////////////////////////////////////////////////////////
+
+#ifdef __GNUG__
+#pragma implementation "xh_gdctl.h"
+#endif
+
+// For compilers that support precompilation, includes "wx.h".
+#include "wx/wxprec.h"
+
+#ifdef __BORLANDC__
+    #pragma hdrstop
+#endif
+
+#include "wx/textctrl.h"
+#include "wx/xrc/xh_gdctl.h"
+#include "wx/dirctrl.h"
+
+
+wxGenericDirCtrlXmlHandler::wxGenericDirCtrlXmlHandler()
+: wxXmlResourceHandler()
+{
+    ADD_STYLE(wxDIRCTRL_DIR_ONLY);
+    ADD_STYLE(wxDIRCTRL_3D_INTERNAL);
+    ADD_STYLE(wxDIRCTRL_SELECT_FIRST);
+    ADD_STYLE(wxDIRCTRL_SHOW_FILTERS);
+    AddWindowStyles();
+}
+
+
+wxObject *wxGenericDirCtrlXmlHandler::DoCreateResource()
+{
+    wxGenericDirCtrl *list = new wxGenericDirCtrl(m_parentAsWindow,
+                                    GetID(),
+                                    GetText(wxT("defaultfolder")),
+                                    GetPosition(), GetSize(),
+                                    GetStyle(),
+                                    GetText(wxT("filter")),
+                                    (int)GetLong(wxT("defaultfilter")),
+                                    GetName());
+    /* TODO: columns definition */
+
+    SetupWindow(list);
+
+    return list;
+}
+
+
+
+bool wxGenericDirCtrlXmlHandler::CanHandle(wxXmlNode *node)
+{
+    return IsOfClass(node, wxT("wxGenericDirCtrl"));
+}
diff -rbu2P /wx232/contrib/src/xrc/xmlrsall.cpp /Programme/wx2/contrib/src/xrc/xmlrsall.cpp
--- /wx232/contrib/src/xrc/xmlrsall.cpp	Sat Jun  9 23:42:50 2001
+++ /Programme/wx2/contrib/src/xrc/xmlrsall.cpp	Sun Jan 20 19:27:26 2002
@@ -93,4 +93,5 @@
 #endif
     AddHandler(new wxUnknownWidgetXmlHandler);
+    AddHandler(new wxGenericDirCtrlXmlHandler);
 
     AddHandler(new wxFrameXmlHandler);
diff -rbu2P /wx232/contrib/utils/wxrcedit/df/wxGenericDirCtrl.df /Programme/wx2/contrib/utils/wxrcedit/df/wxGenericDirCtrl.df
--- /wx232/contrib/utils/wxrcedit/df/wxGenericDirCtrl.df	Thu Jan  1 01:00:00 1970
+++ /Programme/wx2/contrib/utils/wxrcedit/df/wxGenericDirCtrl.df	Sun Jan 20 15:54:56 2002
@@ -0,0 +1,6 @@
+node wxGenericDirCtrl
+var defaultfolder of text
+var filter of text
+var defaultfilter of integer
+var style of flags wxDIRCTRL_DIR_ONLY,wxDIRCTRL_3D_INTERNAL,wxDIRCTRL_SELECT_FIRST,wxDIRCTRL_SHOW_FILTERS
+derived from control

 	  	 
