diff -uNr wxWidgets-snapshot/src/common/imagpng.cpp wxWidgets-mine/src/common/imagpng.cpp
--- wxWidgets-snapshot/src/common/imagpng.cpp	2009-03-06 04:19:58.000000000 +0700
+++ wxWidgets-mine/src/common/imagpng.cpp	2009-05-26 16:02:59.875000000 +0700
@@ -691,9 +691,6 @@
                             ? image->GetOptionInt(wxIMAGE_OPTION_PNG_BITDEPTH)
                             : 8;
 
-    wxASSERT_MSG( iBitDepth == 8 || iBitDepth == 16,
-                    _T("PNG bit depth must be 8 or 16") );
-
     bool bHasAlpha = image->HasAlpha();
     bool bHasMask = image->HasMask();
     bool bUseAlpha = bHasAlpha || bHasMask;
