Index: src/common/dcbufcmn.cpp
===================================================================
--- src/common/dcbufcmn.cpp	(revision 51997)
+++ src/common/dcbufcmn.cpp	(working copy)
@@ -53,7 +53,10 @@
                     h > ms_buffer->GetHeight() )
         {
             delete ms_buffer;
-            ms_buffer = new wxBitmap(w, h);
+
+	    // We must ensure that we always return a valid
+	    // bitmap. Hence the minimum width and height.
+            ms_buffer = new wxBitmap(wxMax(1, w), wxMax(1, h));
         }
         return ms_buffer;
     }

 	  	 
