Index: src/common/dobjcmn.cpp
===================================================================
--- src/common/dobjcmn.cpp	(révision 74270)
+++ src/common/dobjcmn.cpp	(copie de travail)
@@ -448,7 +448,7 @@
 
 size_t wxHTMLDataObject::GetDataSize() const
 {
-    const wxScopedCharBuffer buffer(GetHTML().utf8_str());
+    const wxScopedCharBuffer buffer(m_html.utf8_str());
 
     size_t size = buffer.length();
 
@@ -467,7 +467,7 @@
         return false;
 
     // Windows and Mac always use UTF-8, and docs suggest GTK does as well.
-    const wxScopedCharBuffer html(GetHTML().utf8_str());
+    const wxScopedCharBuffer html(m_html.utf8_str());
     if ( !html )
         return false;
 
