diff -bu1 -r wxWindows/contrib/include/wx/stc/stc.h wxWin/contrib/include/wx/stc/stc.h
--- wxWindows/contrib/include/wx/stc/stc.h	2003-08-25 09:10:08.000000000 +0200
+++ wxWin/contrib/include/wx/stc/stc.h	2003-09-14 10:06:17.000000000 +0200
@@ -2017,2 +2017,7 @@
     void SetCursor(int cursorType);
+    // prevent hiding of virtual function given by wxControl (wxWindow)
+    virtual bool SetCursor( const wxCursor &cursor )
+    {
+        return wxControl::SetCursor( cursor );
+    }
 
diff -bu1 -r wxWindows/contrib/samples/stc/stctest.cpp wxWin/contrib/samples/stc/stctest.cpp
--- wxWindows/contrib/samples/stc/stctest.cpp	2003-08-26 13:18:48.000000000 +0200
+++ wxWin/contrib/samples/stc/stctest.cpp	2003-09-14 16:04:59.000000000 +0200
@@ -213,3 +213,4 @@
     wxString st;
-    wxFileInputStream stream(wxT("stctest.cpp"));
+    wxString filename(wxT("stctest.cpp"));
+    wxFileInputStream stream(filename);
     size_t sz = stream.GetSize();
diff -bu1 -r wxWindows/contrib/src/stc/PlatWX.cpp wxWin/contrib/src/stc/PlatWX.cpp
--- wxWindows/contrib/src/stc/PlatWX.cpp	2003-08-06 10:28:44.000000000 +0200
+++ wxWin/contrib/src/stc/PlatWX.cpp	2003-09-14 09:42:56.000000000 +0200
@@ -290,3 +290,3 @@
 
-void SurfaceImpl::InitPixMap(int width, int height, Surface *surface_, WindowID) {
+void SurfaceImpl::InitPixMap(int width, int height, Surface *WXUNUSED(surface_), WindowID) {
     Release();
@@ -544,3 +544,3 @@
 
-int SurfaceImpl::InternalLeading(Font &font) {
+int SurfaceImpl::InternalLeading(Font &WXUNUSED(font)) {
     return 0;
@@ -565,3 +565,3 @@
 
-int SurfaceImpl::SetPalette(Palette *pal, bool inBackGround) {
+int SurfaceImpl::SetPalette(Palette *WXUNUSED(pal), bool WXUNUSED(inBackGround)) {
     return 0;
@@ -580,3 +580,3 @@
 
-void SurfaceImpl::SetDBCSMode(int codePage) {
+void SurfaceImpl::SetDBCSMode(int WXUNUSED(codePage)) {
     // dbcsMode = codePage == SC_CP_DBCS;
@@ -971,3 +971,3 @@
 
-int ListBoxImpl::Find(const char *prefix) {
+int ListBoxImpl::Find(const char *WXUNUSED(prefix)) {
     // No longer used
@@ -1065,3 +1065,3 @@
 
-DynamicLibrary *DynamicLibrary::Load(const char *modulePath) {
+DynamicLibrary *DynamicLibrary::Load(const char *WXUNUSED(modulePath)) {
     wxFAIL_MSG(wxT("Dynamic lexer loading not implemented yet"));
@@ -1105,3 +1105,3 @@
 
-bool Platform::IsKeyDown(int key) {
+bool Platform::IsKeyDown(int WXUNUSED(key)) {
     return false;  // I don't think we'll need this.
@@ -1198,3 +1198,3 @@
 
-bool Platform::IsDBCSLeadByte(int codePage, char ch) {
+bool Platform::IsDBCSLeadByte(int WXUNUSED(codePage), char WXUNUSED(ch)) {
     return false;
@@ -1202,3 +1202,3 @@
 
-int Platform::DBCSCharLength(int codePage, const char *s) {
+int Platform::DBCSCharLength(int WXUNUSED(codePage), const char *WXUNUSED(s)) {
     return 1;
diff -bu1 -r wxWindows/contrib/src/stc/ScintillaWX.cpp wxWin/contrib/src/stc/ScintillaWX.cpp
--- wxWindows/contrib/src/stc/ScintillaWX.cpp	2003-07-18 09:16:14.000000000 +0200
+++ wxWin/contrib/src/stc/ScintillaWX.cpp	2003-09-14 09:48:31.000000000 +0200
@@ -385,3 +385,4 @@
     if (Editor::CanPaste()) {
-        if ( (didOpen = !wxTheClipboard->IsOpened()) )
+        didOpen = !wxTheClipboard->IsOpened();
+        if ( (didOpen) )
             wxTheClipboard->Open();
@@ -599,3 +600,3 @@
 
-void ScintillaWX::DoSize(int width, int height) {
+void ScintillaWX::DoSize(int WXUNUSED(width), int WXUNUSED(height)) {
 //      PRectangle rcClient(0,0,width,height);
@@ -630,4 +631,4 @@
 
-void ScintillaWX::DoMiddleButtonUp(Point pt) {
 #ifdef __WXGTK__
+void ScintillaWX::DoMiddleButtonUp(Point pt) {
     // Set the current position to the mouse click point and
@@ -658,2 +659,4 @@
     EnsureCaretVisible();
+#else
+void ScintillaWX::DoMiddleButtonUp(Point WXUNUSED(pt)) {
 #endif
@@ -762,3 +765,3 @@
 
-wxDragResult ScintillaWX::DoDragEnter(wxCoord x, wxCoord y, wxDragResult def) {
+wxDragResult ScintillaWX::DoDragEnter(wxCoord WXUNUSED(x), wxCoord WXUNUSED(y), wxDragResult def) {
     dragResult = def;
@@ -819,4 +822,4 @@
 
-void ScintillaWX::ClipChildren(wxDC& dc, PRectangle rect) {
 #ifdef __WXGTK__
+void ScintillaWX::ClipChildren(wxDC& dc, PRectangle rect) {
     wxRegion rgn(wxRectFromPRectangle(rect));
@@ -832,2 +835,4 @@
     dc.SetClippingRegion(rgn);
+#else
+void ScintillaWX::ClipChildren(wxDC& WXUNUSED(dc), PRectangle WXUNUSED(rect)) {
 #endif
diff -bu1 -r wxWindows/contrib/src/stc/stc.cpp wxWin/contrib/src/stc/stc.cpp
--- wxWindows/contrib/src/stc/stc.cpp	2003-08-06 10:29:04.000000000 +0200
+++ wxWin/contrib/src/stc/stc.cpp	2003-09-14 09:51:54.000000000 +0200
@@ -2154,3 +2154,3 @@
 
-void wxStyledTextCtrl::OnPaint(wxPaintEvent& evt) {
+void wxStyledTextCtrl::OnPaint(wxPaintEvent& WXUNUSED(evt)) {
     wxPaintDC dc(this);
@@ -2176,3 +2176,3 @@
 
-void wxStyledTextCtrl::OnSize(wxSizeEvent& evt) {
+void wxStyledTextCtrl::OnSize(wxSizeEvent& WXUNUSED(evt)) {
     wxSize sz = GetClientSize();
@@ -2266,3 +2266,3 @@
 
-void wxStyledTextCtrl::OnLoseFocus(wxFocusEvent& evt) {
+void wxStyledTextCtrl::OnLoseFocus(wxFocusEvent& WXUNUSED(evt)) {
     m_swx->DoLoseFocus();
@@ -2271,3 +2271,3 @@
 
-void wxStyledTextCtrl::OnGainFocus(wxFocusEvent& evt) {
+void wxStyledTextCtrl::OnGainFocus(wxFocusEvent& WXUNUSED(evt)) {
     m_swx->DoGainFocus();
@@ -2276,3 +2276,3 @@
 
-void wxStyledTextCtrl::OnSysColourChanged(wxSysColourChangedEvent& evt) {
+void wxStyledTextCtrl::OnSysColourChanged(wxSysColourChangedEvent& WXUNUSED(evt)) {
     m_swx->DoSysColourChange();
@@ -2281,3 +2281,3 @@
 
-void wxStyledTextCtrl::OnEraseBackground(wxEraseEvent& evt) {
+void wxStyledTextCtrl::OnEraseBackground(wxEraseEvent& WXUNUSED(evt)) {
     // do nothing to help avoid flashing
@@ -2292,3 +2292,3 @@
 
-void wxStyledTextCtrl::OnListBox(wxCommandEvent& evt) {
+void wxStyledTextCtrl::OnListBox(wxCommandEvent& WXUNUSED(evt)) {
     m_swx->DoOnListBox();

 	  	 
