diff --git a/src/msw/volume.cpp b/src/msw/volume.cpp
index 24a774b..41aa6fe 100644
--- a/src/msw/volume.cpp
+++ b/src/msw/volume.cpp
@@ -613,7 +613,11 @@ wxIcon wxFSVolume::GetIcon(wxFSIconType type) const
 
         SHFILEINFO fi;
         long rc = SHGetFileInfo(m_volName.t_str(), 0, &fi, sizeof(fi), flags);
+#if defined(__WXMSW__) || defined(__WXPM__)
         m_icons[type].SetHICON((WXHICON)fi.hIcon);
+#else
+        wxFAIL_MSG(wxS("Can't convert HICON to wxIcon"));
+#endif
         if (!rc || !fi.hIcon)
         {
             wxLogError(_("Cannot load icon from '%s'."), m_volName.c_str());
