diff -ru -U 2 -xCVS -x'.#*' -x'*.orig' -x'*.rej' -x'*~' wxWindows.cvs/include/wx/generic/treectlg.h wxWindows/include/wx/generic/treectlg.h
--- wxWindows.cvs/include/wx/generic/treectlg.h	2005-01-10 16:21:16.000000000 -0600
+++ wxWindows/include/wx/generic/treectlg.h	2005-01-27 16:33:55.000000000 -0600
@@ -150,6 +150,9 @@
         // the item will be shown in bold
     void SetItemBold(const wxTreeItemId& item, bool bold = true);
 
+        // the item will be shown with a drop highlight
+    void SetItemDropHighlight(const wxTreeItemId& item, bool highlight = true);
+
         // set the item's text colour
     void SetItemTextColour(const wxTreeItemId& item, const wxColour& col);
 
diff -ru -U 2 -xCVS -x'.#*' -x'*.orig' -x'*.rej' -x'*~' wxWindows.cvs/src/generic/treectlg.cpp wxWindows/src/generic/treectlg.cpp
--- wxWindows.cvs/src/generic/treectlg.cpp	2005-01-10 16:21:21.000000000 -0600
+++ wxWindows/src/generic/treectlg.cpp	2005-01-27 16:37:35.000000000 -0600
@@ -1009,6 +1009,25 @@
     }
 }
 
+void wxGenericTreeCtrl::SetItemDropHighlight(const wxTreeItemId& item,
+                                             bool highlight)
+{
+    wxCHECK_RET( item.IsOk(), wxT("invalid tree item") );
+
+    wxColour fg, bg;
+
+    if (highlight)
+    {
+        bg = wxSystemSettings::GetColour(wxSYS_COLOUR_HIGHLIGHT);
+        fg = wxSystemSettings::GetColour(wxSYS_COLOUR_HIGHLIGHTTEXT);
+    }
+
+    wxGenericTreeItem *pItem = (wxGenericTreeItem*) item.m_pItem;
+    pItem->Attr().SetTextColour(fg);
+    pItem->Attr().SetBackgroundColour(bg);
+    RefreshLine(pItem);
+}
+
 void wxGenericTreeCtrl::SetItemTextColour(const wxTreeItemId& item,
                                    const wxColour& col)
 {

 	  	 
