Index: app.cpp
===================================================================
--- app.cpp	(revision 52195)
+++ app.cpp	(working copy)
@@ -542,10 +542,18 @@
     wxWindow *topWindow = wxTheApp->GetTopWindow();
     if ( topWindow )
     {
-        if ( !::PostMessage(GetHwndOf(topWindow), WM_NULL, 0, 0) )
+        MSG m;
+        // Do not post WM_NULL if there's already a pending WM_NULL.
+        // Due to a limitation of PeekMessage, we have to check the range from 
+        // 0-1 instead.
+        if (!::PeekMessage(&m, GetHwndOf(topWindow), 0, 1, PM_NOREMOVE) ||
+            ::PeekMessage(&m, GetHwndOf(topWindow), 1, 1, PM_NOREMOVE))
         {
-            // should never happen
-            wxLogLastError(wxT("PostMessage(WM_NULL)"));
+            if ( !::PostMessage(GetHwndOf(topWindow), WM_NULL, 0, 0) )
+            {
+                // should never happen
+                wxLogLastError(wxT("PostMessage(WM_NULL)"));
+            }
         }
     }
 }

 	  	 
