Index: event.cpp
===================================================================
--- event.cpp	(revision 52195)
+++ event.cpp	(working copy)
@@ -380,6 +380,16 @@
 
 void MyFrame::OnProcessCustom(wxCommandEvent& WXUNUSED(event))
 {
-    wxLogMessage(_T("Got a custom event!"));
+    // Post same event twice on every even event
+    static bool x = true;
+    if (x)
+    {
+        wxCommandEvent eventCustom(wxEVT_MY_CUSTOM_COMMAND);
+        wxPostEvent(this, eventCustom);
+        wxPostEvent(this, eventCustom);
+        x = false;
+    }
+    else
+        x = true;
 }
 

 	  	 
