Index: tests/controls/spinctrltest.cpp
===================================================================
--- tests/controls/spinctrltest.cpp	(revision 73366)
+++ tests/controls/spinctrltest.cpp	(working copy)
@@ -103,7 +103,7 @@
 
     m_spin->Create(parent, wxID_ANY, "",
                    wxDefaultPosition, wxDefaultSize, 0,
-                   0, 100, 17);
+                   0, 255, 255);
 
     CPPUNIT_ASSERT_EQUAL(0, updated.GetCount());
 }
