Index: include/wx/stc/stc.h
===================================================================
--- include/wx/stc/stc.h	(revision 73148)
+++ include/wx/stc/stc.h	(working copy)
@@ -4628,7 +4628,7 @@
     // implement wxTextAreaBase pure virtual methods
     // ---------------------------------------------
 
-    virtual int GetLineLength(long n) const { return GetLine(n).length(); }
+    virtual int GetLineLength(long lineNo) const { return static_cast<int>(GetLineText(lineNo).length()); }
     virtual wxString GetLineText(long lineNo) const
     {
         wxString text = GetLine(static_cast<int>(lineNo));
Index: src/stc/stc.h.in
===================================================================
--- src/stc/stc.h.in	(revision 73148)
+++ src/stc/stc.h.in	(working copy)
@@ -370,7 +370,7 @@
     // implement wxTextAreaBase pure virtual methods
     // ---------------------------------------------
 
-    virtual int GetLineLength(long n) const { return GetLine(n).length(); }
+    virtual int GetLineLength(long lineNo) const { return static_cast<int>(GetLineText(lineNo).length()); }
     virtual wxString GetLineText(long lineNo) const
     {
         wxString text = GetLine(static_cast<int>(lineNo));
