Index: interface/wx/event.h
===================================================================
--- interface/wx/event.h	(revision 70611)
+++ interface/wx/event.h	(working copy)
@@ -1154,25 +1154,6 @@
     virtual bool TryBefore(wxEvent& event);
 
     /**
-        Try to process the event in this event handler.
-
-        This method is called from ProcessEventLocally() and thus, indirectly,
-        from ProcessEvent(), please see the detailed description of the event
-        processing logic there.
-
-        It is currently @em not virtual and so may not be overridden.
-
-        @since 2.9.1
-
-        @param event
-            Event to process.
-        @return
-            @true if this object itself defines a handler for this event and
-            the handler didn't skip the event.
-     */
-    bool TryThis(wxEvent& event);
-
-    /**
         Method called by ProcessEvent() as last resort.
 
         This method can be overridden to implement post-processing for the
