Index: src/msw/choice.cpp
===================================================================
--- src/msw/choice.cpp	(revision 70015)
+++ src/msw/choice.cpp	(working copy)
@@ -350,6 +350,11 @@
     // we have to delete and add back the string as there is no way to change a
     // string in place
 
+    // if we change the string for the current selection,
+    // deleting/inserting will lose the selection
+    // so we have to restore it later
+    int oldSelection = GetSelection();
+
     // we need to preserve the client data manually
     void *oldData = NULL;
     wxClientData *oldObjData = NULL;
@@ -367,6 +372,9 @@
     else if ( oldObjData )
         SetClientObject(n, oldObjData);
 
+    // restore selection only if it's the same index we just changed
+    if ( oldSelection == n ) SetSelection(n);
+
     InvalidateBestSize();
 }
 
