Index: src/xrc/xh_filepicker.cpp
===================================================================
--- src/xrc/xh_filepicker.cpp	(revision 46775)
+++ src/xrc/xh_filepicker.cpp	(working copy)
@@ -30,6 +30,7 @@
     XRC_ADD_STYLE(wxFLP_FILE_MUST_EXIST);
     XRC_ADD_STYLE(wxFLP_CHANGE_DIR);
     XRC_ADD_STYLE(wxFLP_DEFAULT_STYLE);
+    XRC_ADD_STYLE(wxFLP_USE_TEXTCTRL);
     AddWindowStyles();
 }
 

 	  	 
