Index: include/wx/mac/bitmap.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/mac/bitmap.h,v
retrieving revision 1.25
diff -u -B -b -r1.25 bitmap.h
--- include/wx/mac/bitmap.h	2003/08/09 12:37:44	1.25
+++ include/wx/mac/bitmap.h	2004/02/18 10:31:59
@@ -86,7 +86,9 @@
   int           m_depth;
   bool          m_ok;
   int           m_numColors;
+#if wxUSE_PALETTE
   wxPalette     m_bitmapPalette;
+#endif // wxUSE_PALETTE
   int           m_quality;
 
   int            m_bitmapType ;
@@ -186,8 +188,10 @@
   void SetQuality(int q);
   void SetOk(bool isOk);
 
+#if wxUSE_PALETTE
   wxPalette* GetPalette() const;
   void SetPalette(const wxPalette& palette);
+#endif // wxUSE_PALETTE
 
   wxMask *GetMask() const;
   void SetMask(wxMask *mask) ;
Index: src/mac/bitmap.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/mac/bitmap.cpp,v
retrieving revision 1.44
diff -u -B -b -r1.44 bitmap.cpp
--- src/mac/bitmap.cpp	2003/09/20 15:38:43	1.44
+++ src/mac/bitmap.cpp	2004/02/18 10:32:01
@@ -551,7 +551,9 @@
    wxBitmapRefData *ref = (wxBitmapRefData *)ret.GetRefData();
 
    ref->m_numColors     = M_BITMAPDATA->m_numColors;
+#if wxUSE_PALETTE
    ref->m_bitmapPalette = M_BITMAPDATA->m_bitmapPalette;
+#endif // wxUSE_PALETTE
    ref->m_bitmapType    = M_BITMAPDATA->m_bitmapType;
 
    // Copy sub region of this bitmap
@@ -1031,6 +1033,7 @@
     M_BITMAPDATA->m_ok = isOk;
 }
 
+#if wxUSE_PALETTE
 wxPalette *wxBitmap::GetPalette() const
 {
    wxCHECK_MSG( Ok(), NULL, wxT("Invalid bitmap  GetPalette()") );
@@ -1045,6 +1048,7 @@
 
     M_BITMAPDATA->m_bitmapPalette = palette ;
 }
+#endif // wxUSE_PALETTE
 
 void wxBitmap::SetMask(wxMask *mask)
 {

 	  	 
