diff --git samples/stc/stctest.cpp samples/stc/stctest.cpp
index 9efc2f8..25782b5 100644
--- samples/stc/stctest.cpp
+++ samples/stc/stctest.cpp
@@ -318,7 +318,9 @@ AppFrame::AppFrame (const wxString &title)
     CreateMenu ();
 
     // open first page
-    m_edit = new Edit (this, wxID_ANY);
+    wxNotebook* nb = new wxNotebook(this, wxID_ANY);
+    m_edit = new Edit (nb, wxID_ANY);
+    nb->AddPage(m_edit, wxT("Editor"));
     m_edit->SetFocus();
 
     FileOpen (_T("stctest.cpp"));
diff --git src/stc/ScintillaWX.cpp src/stc/ScintillaWX.cpp
index 76af113..7a0f499 100644
--- src/stc/ScintillaWX.cpp
+++ src/stc/ScintillaWX.cpp
@@ -306,7 +306,7 @@ void ScintillaWX::StartDrag() {
     // We defer the starting of the DnD, otherwise the LeftUp of a normal
     // click could be lost and the STC will think it is doing a DnD when the
     // user just wanted a normal click.
-    startDragTimer->Start(200, true);
+    DoStartDrag();
 #endif // wxUSE_DRAG_AND_DROP
 }
 
