Index: region.cpp
===================================================================
RCS file: /home/wxcvs/wxWindows/src/gtk/region.cpp,v
retrieving revision 1.35
diff -b -u -2 -r1.35 region.cpp
--- region.cpp	13 Aug 2002 20:59:05 -0000	1.35
+++ region.cpp	15 Aug 2002 10:16:27 -0000
@@ -99,5 +99,12 @@
 {
     m_refData = new wxRegionRefData();
+#ifdef __WXGTK20__
     M_REGIONDATA->m_region = gdk_region_copy( region );
+#else
+    M_REGIONDATA->m_region = gdk_region_new();
+    GdkRegion *regCopy = gdk_regions_union(M_REGIONDATA->m_region, region);
+    gdk_region_destroy(M_REGIONDATA->m_region);
+    M_REGIONDATA->m_region = regCopy;
+#endif
 }
 

 	  	 
