Index: src/osx/cocoa/window.mm
===================================================================
--- src/osx/cocoa/window.mm	(revision 60892)
+++ src/osx/cocoa/window.mm	(working copy)
@@ -417,7 +417,7 @@
             wxevent.m_wheelDelta = 10;
             wxevent.m_linesPerAction = 1;
             NSLog(@"deltaX %f, deltaY %f",[nsEvent deltaX], [nsEvent deltaY]);
-            if ( abs([nsEvent deltaX]) > abs([nsEvent deltaY]) )
+            if ( fabs([nsEvent deltaX]) > fabs([nsEvent deltaY]) )
             {
                 wxevent.m_wheelAxis = 1;
                 wxevent.m_wheelRotation = [nsEvent deltaX] * 10.0;