Index: src/aui/tabmdi.cpp
===================================================================
--- src/aui/tabmdi.cpp	(revision 72577)
+++ src/aui/tabmdi.cpp	(working copy)
@@ -244,7 +244,15 @@
 
 wxAuiMDIChildFrame *wxAuiMDIParentFrame::GetActiveChild() const
 {
-    return GetClientWindow()->GetActiveChild();
+    wxAuiMDIClientWindow* pClient = GetClientWindow();
+    if (pClient)
+    {
+        return pClient->GetActiveChild();
+    }
+    else
+    {
+        return NULL;
+    }
 }
 
 void wxAuiMDIParentFrame::SetActiveChild(wxAuiMDIChildFrame* pChildFrame)
