Index: src/gtk/tooltip.cpp
===================================================================
--- src/gtk/tooltip.cpp	(revision 47911)
+++ src/gtk/tooltip.cpp	(working copy)
@@ -95,4 +95,12 @@
     gtk_tooltips_set_delay( gs_tooltips, (int)msecs );
 }
 
+void wxToolTip::SetAutoPop( long WXUNUSED(msecs) )
+{
+}
+
+void wxToolTip::SetReshow( long WXUNUSED(msecs) )
+{
+}
+
 #endif // wxUSE_TOOLTIPS
Index: src/mac/carbon/tooltip.cpp
===================================================================
--- src/mac/carbon/tooltip.cpp	(revision 47911)
+++ src/mac/carbon/tooltip.cpp	(working copy)
@@ -146,6 +146,14 @@
     s_ToolTipDelay = msecs ;
 }
 
+void wxToolTip::SetAutoPop( long WXUNUSED(msecs) )
+{
+}
+
+void wxToolTip::SetReshow( long WXUNUSED(msecs) )
+{
+}
+
 void wxToolTip::RelayEvent( wxWindow *win , wxMouseEvent &event )
 {
     if ( s_ShowToolTips )
Index: src/palmos/tooltip.cpp
===================================================================
--- src/palmos/tooltip.cpp	(revision 47911)
+++ src/palmos/tooltip.cpp	(working copy)
@@ -99,6 +99,14 @@
 {
 }
 
+void wxToolTip::SetAutoPop( long WXUNUSED(milliseconds) )
+{
+}
+
+void wxToolTip::SetReshow( long WXUNUSED(milliseconds) )
+{
+}
+
 // ---------------------------------------------------------------------------
 // implementation helpers
 // ---------------------------------------------------------------------------
Index: src/gtk1/tooltip.cpp
===================================================================
--- src/gtk1/tooltip.cpp	(revision 47911)
+++ src/gtk1/tooltip.cpp	(working copy)
@@ -79,4 +79,12 @@
     gtk_tooltips_set_delay( ss_tooltips, (int)msecs );
 }
 
+void wxToolTip::SetAutoPop( long WXUNUSED(msecs) )
+{
+}
+
+void wxToolTip::SetReshow( long WXUNUSED(msecs) )
+{
+}
+
 #endif
Index: src/msw/tooltip.cpp
===================================================================
--- src/msw/tooltip.cpp	(revision 47911)
+++ src/msw/tooltip.cpp	(working copy)
@@ -191,6 +191,18 @@
                             TTDT_INITIAL, milliseconds);
 }
 
+void wxToolTip::SetAutoPop(long milliseconds)
+{
+    SendTooltipMessageToAll(ms_hwndTT, TTM_SETDELAYTIME,
+                            TTDT_AUTOPOP, milliseconds);
+}
+
+void wxToolTip::SetReshow(long milliseconds)
+{
+    SendTooltipMessageToAll(ms_hwndTT, TTM_SETDELAYTIME,
+                            TTDT_RESHOW, milliseconds);
+}
+
 // ---------------------------------------------------------------------------
 // implementation helpers
 // ---------------------------------------------------------------------------
Index: docs/latex/wx/tooltip.tex
===================================================================
--- docs/latex/wx/tooltip.tex	(revision 47911)
+++ docs/latex/wx/tooltip.tex	(working copy)
@@ -3,8 +3,10 @@
 This class holds information about a tooltip associated with a window
 (see \helpref{wxWindow::SetToolTip}{wxwindowsettooltip}).
 
-The two static methods, \helpref{wxToolTip::Enable}{wxtooltipenable} and
-\helpref{wxToolTip::SetDelay}{wxtooltipsetdelay} can be used to globally
+The four static methods, \helpref{wxToolTip::Enable}{wxtooltipenable},
+\helpref{wxToolTip::SetDelay}{wxtooltipsetdelay} 
+\helpref(wxToolTip::SetAutoPop)(wxtooltipsetautopop) and 
+\helpref(wxToolTip::SetReshow)(wxtooltipsetreshow) can be used to globally
 alter tooltips behaviour.
 
 \wxheading{Derived from}
@@ -39,6 +41,21 @@
 
 May not be supported on all platforms (eg. wxCocoa).
 
+\membersection{wxToolTip::SetAutoPop}\label{wxtooltipsetautopop}
+
+\func{static void}{SetAutoPop}{\param{long }{msecs}}
+
+Set the delay after which the tooltip disappears or how long a
+tooltip remains visible.
+May not be supported on all platforms (eg. wxCocoa, GTK, Palmos).
+
+\membersection{wxToolTip::SetReshow}\label{wxtooltipsetreshow}
+
+\func{static void}{SetReshow}{\param{long }{msecs}}
+
+Set the delay between subsequent tooltips to appear.
+May not be supported on all platforms (eg. wxCocoa, GTK, Palmos).
+
 \membersection{wxToolTip::wxToolTip}\label{wxtooltipwxtooltip}
 
 \func{}{wxToolTip}{\param{const wxString\& }{tip}}
Index: include/wx/gtk1/tooltip.h
===================================================================
--- include/wx/gtk1/tooltip.h	(revision 47911)
+++ include/wx/gtk1/tooltip.h	(working copy)
@@ -31,6 +31,10 @@
     // globally change the tooltip parameters
     static void Enable( bool flag );
     static void SetDelay( long msecs );
+        // set the delay after which the tooltip disappears or how long the tooltip remains visible
+    static void SetAutoPop(long msecs);
+        // set the delay between subsequent tooltips to appear
+    static void SetReshow(long msecs);
 
     wxToolTip( const wxString &tip );
 
Index: include/wx/mac/carbon/tooltip.h
===================================================================
--- include/wx/mac/carbon/tooltip.h	(revision 47911)
+++ include/wx/mac/carbon/tooltip.h	(working copy)
@@ -36,6 +36,10 @@
     static void Enable(bool flag);
         // set the delay after which the tooltip appears
     static void SetDelay(long milliseconds);
+        // set the delay after which the tooltip disappears or how long the tooltip remains visible
+    static void SetAutoPop(long milliseconds);
+        // set the delay between subsequent tooltips to appear
+    static void SetReshow(long milliseconds);
     static void NotifyWindowDelete( WXHWND win ) ;
 
     // implementation only from now on
Index: include/wx/palmos/tooltip.h
===================================================================
--- include/wx/palmos/tooltip.h	(revision 47911)
+++ include/wx/palmos/tooltip.h	(working copy)
@@ -33,6 +33,10 @@
     static void Enable(bool flag);
         // set the delay after which the tooltip appears
     static void SetDelay(long milliseconds);
+        // set the delay after which the tooltip disappears or how long the tooltip remains visible
+    static void SetAutoPop(long milliseconds);
+        // set the delay between subsequent tooltips to appear
+    static void SetReshow(long milliseconds);
 
     // implementation only from now on
     // -------------------------------
Index: include/wx/msw/tooltip.h
===================================================================
--- include/wx/msw/tooltip.h	(revision 47911)
+++ include/wx/msw/tooltip.h	(working copy)
@@ -37,6 +37,10 @@
     static void Enable(bool flag);
         // set the delay after which the tooltip appears
     static void SetDelay(long milliseconds);
+        // set the delay after which the tooltip disappears or how long the tooltip remains visible
+    static void SetAutoPop(long milliseconds);
+        // set the delay between subsequent tooltips to appear
+    static void SetReshow(long milliseconds);
 
     // implementation only from now on
     // -------------------------------
Index: include/wx/os2/tooltip.h
===================================================================
--- include/wx/os2/tooltip.h	(revision 47911)
+++ include/wx/os2/tooltip.h	(working copy)
@@ -33,6 +33,10 @@
     static void Enable(bool WXUNUSED(flag)) {}
         // set the delay after which the tooltip appears
     static void SetDelay(long WXUNUSED(milliseconds)) {}
+        // set the delay after which the tooltip disappears or how long the tooltip remains visible
+    static void SetAutoPop(long WXUNUSED(milliseconds)) {}
+        // set the delay between subsequent tooltips to appear
+    static void SetReshow(long WXUNUSED(milliseconds)) {}
 
     //
     // Implementation
Index: include/wx/gtk/tooltip.h
===================================================================
--- include/wx/gtk/tooltip.h	(revision 47911)
+++ include/wx/gtk/tooltip.h	(working copy)
@@ -33,6 +33,10 @@
     // globally change the tooltip parameters
     static void Enable( bool flag );
     static void SetDelay( long msecs );
+        // set the delay after which the tooltip disappears or how long the tooltip remains visible
+    static void SetAutoPop(long msecs);
+        // set the delay between subsequent tooltips to appear
+    static void SetReshow(long msecs);
 
     // get/set the tooltip text
     void SetTip( const wxString &tip );
Index: include/wx/cocoa/tooltip.h
===================================================================
--- include/wx/cocoa/tooltip.h	(revision 47911)
+++ include/wx/cocoa/tooltip.h	(working copy)
@@ -36,6 +36,10 @@
     static void Enable(bool flag);
         // set the delay after which the tooltip appears
     static void SetDelay(long milliseconds);
+        // set the delay after which the tooltip disappears or how long the tooltip remains visible
+    static void SetAutoPop(long milliseconds);
+        // set the delay between subsequent tooltips to appear
+    static void SetReshow(long milliseconds);
 
 private:
     void SetWindow(wxWindow* window);

 	  	 
