Index: src/osx/cocoa/stattext.mm
===================================================================
--- src/osx/cocoa/stattext.mm	(revision 58236)
+++ src/osx/cocoa/stattext.mm	(working copy)
@@ -49,12 +49,14 @@
         wxClientDC dc(const_cast<wxStaticText*>(this));
         wxCoord width, height ;
         dc.GetTextExtent( m_label , &width, &height);
-        // Some labels seem to have their last characters
-        // stripped out.  Adding 4 pixels seems to be enough to fix this.
         bounds.h = width+4;
         bounds.v = height;
     }
 
+    // Some labels seem to have their last characters
+    // stripped out.  Adding 4 pixels seems to be enough to fix this.
+    bounds.h += 4;
+
     if ( m_label.empty() )
         bounds.h = 0;
 
