Index: samples/layout/layout.cpp
===================================================================
--- samples/layout/layout.cpp	(revision 59502)
+++ samples/layout/layout.cpp	(working copy)
@@ -124,13 +124,13 @@
     // 1) top: create wxStaticText with minimum size equal to its default size
     topsizer->Add(
         new wxStaticText( p, wxID_ANY, _T("An explanation (wxALIGN_RIGHT).") ),
-        wxSizerFlags().Align(wxALIGN_RIGHT).Border(wxALL & ~wxBOTTOM, 5));
+        wxSizerFlags().Align(wxALIGN_RIGHT));
     topsizer->Add(
-        new wxStaticText( p, wxID_ANY, _T("An explanation (wxALIGN_LEFT).") ),
-        wxSizerFlags().Align(wxALIGN_LEFT).Border(wxALL & ~wxBOTTOM, 5));
+        new wxStaticText( p, wxID_ANY, _T("Quality") ),
+        wxSizerFlags().Align(wxALIGN_LEFT));
     topsizer->Add(
         new wxStaticText( p, wxID_ANY, _T("An explanation (wxALIGN_CENTRE_HORIZONTAL).") ),
-        wxSizerFlags().Align(wxALIGN_CENTRE_HORIZONTAL).Border(wxALL & ~wxBOTTOM, 5));
+        wxSizerFlags().Align(wxALIGN_CENTRE_HORIZONTAL));
 
     // 2) top: create wxTextCtrl with minimum size (100x60)
     topsizer->Add(
