Index: wxWidgets-2.9/src/common/datetimefmt.cpp
===================================================================
--- wxWidgets-2.9/src/common/datetimefmt.cpp	(revision 59800)
+++ wxWidgets-2.9/src/common/datetimefmt.cpp	(working copy)
@@ -1525,8 +1525,6 @@
                     // try to parse what follows as "%H:%M:%S" and, if this
                     // fails, as "%I:%M:%S %p" - this should catch the most
                     // common cases
-                    wxDateTime dt;
-
                     const wxDateTime
                         dt = ParseFormatAt(input, date.end(), "%T", "%r");
                     if ( !dt.IsValid() )
