diff -rbu2 ./cvs/wx2/wxwindows/contrib/utils/wxrcedit/editor.cpp ./wx2/contrib/utils/wxrcedit/editor.cpp
--- ./cvs/wx2/wxwindows/contrib/utils/wxrcedit/editor.cpp	Wed May 30 06:13:46 2001
+++ ./wx2/contrib/utils/wxrcedit/editor.cpp	Sun Jun  3 13:47:44 2001
@@ -355,5 +355,5 @@
     if (m_Modified) s << _T("* ");
     s << _("wxrcedit");
-    if (!m_FileName)
+    if (m_FileName != "")
         s << _T(" - ") << wxFileNameFromPath(m_FileName);
     SetTitle(s);

 	  	 
