--- drawing.cpp	Fri Apr  3 06:48:47 2009
+++ test2.cpp	Fri Apr  3 06:21:43 2009
@@ -33,6 +33,7 @@
 #include "wx/colordlg.h"
 #include "wx/image.h"
 #include "wx/artprov.h"
+#include "wx/dcbuffer.h" //MODIFIED (ADDED)
 
 #define wxTEST_GRAPHICS 1
 
@@ -408,6 +409,7 @@
 END_EVENT_TABLE()
 
 #include "smile.xpm"
+#include "bg.xpm"//MODIFIED (ADDED)
 
 MyCanvas::MyCanvas(MyFrame *parent)
         : wxScrolledWindow(parent, wxID_ANY, wxDefaultPosition, wxDefaultSize,
@@ -551,7 +553,8 @@
 {
     // mark the origin
     dc.DrawCircle(0, 0, 10);
-
+    wxBitmap mybmp(bg_xpm);//MODIFIED (ADDED)
+    dc.DrawBitmap(mybmp,0,0,true);//MODIFIED (ADDED)
 #if !defined(wxMAC_USE_CORE_GRAPHICS) || !wxMAC_USE_CORE_GRAPHICS
     // GetPixel and FloodFill not supported by Mac OS X CoreGraphics
     // (FloodFill uses Blit from a non-wxMemoryDC)
@@ -588,7 +591,7 @@
         dc.DrawBitmap(m_smile_bmp, x + rectSize - 20, rectSize - 10, true);
 
     dc.SetBrush( *wxBLACK_BRUSH );
-    dc.DrawRectangle( 0, 160, 1000, 300 );
+    //dc.DrawRectangle( 0, 160, 1000, 300 );//MODIFIED (COMMENTED)
 
     // draw lines
     wxBitmap bitmap(20,70);
@@ -1198,7 +1201,7 @@
 
 void MyCanvas::OnPaint(wxPaintEvent &WXUNUSED(event))
 {
-    wxPaintDC pdc(this);
+    wxAutoBufferedPaintDC pdc(this);//MODIFIED (--wxPaintDc ++wxAutoBufferedPaintDC)
 
 #if wxUSE_GRAPHICS_CONTEXT
      wxGCDC gdc( pdc ) ;
@@ -1297,6 +1300,7 @@
         default:
             break;
     }
+    dc.SetUserScale( 1/m_owner->m_xUserScale, 1/m_owner->m_yUserScale );//MODIFIED (ADDED)
 }
 
 void MyCanvas::OnMouseMove(wxMouseEvent &event)
@@ -1581,7 +1585,7 @@
 {
     dc.SetLogicalOrigin( m_xLogicalOrigin, m_yLogicalOrigin );
     dc.SetAxisOrientation( !m_xAxisReversed, m_yAxisReversed );
-    dc.SetUserScale( m_xUserScale, m_yUserScale );
+    //dc.SetUserScale( m_xUserScale, m_yUserScale );//MODIFIED (COMMENTED)
     dc.SetMapMode( m_mapMode );
 }
 
