Index: src/gtk/combobox.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/gtk/combobox.cpp,v
retrieving revision 1.127
diff -b -u -2 -r1.127 combobox.cpp
--- src/gtk/combobox.cpp	2005/02/15 23:19:32	1.127
+++ src/gtk/combobox.cpp	2005/02/28 14:22:19
@@ -39,5 +39,5 @@
 
 extern bool   g_blockEventsOnDrag;
-static int    g_SelectionBeforePopup = -1;
+static int    g_SelectionBeforePopup = -2; // -2 <=> the popup is hidden
 //-----------------------------------------------------------------------------
 //  "changed" - typing and list item matches get changed, select-child
@@ -84,6 +84,6 @@
     }
 
-    // reset the selection flag to an identifiable value
-    g_SelectionBeforePopup = -1;
+    // reset the selection flag to an identifiable value (-2 = hidden)
+    g_SelectionBeforePopup = -2;
 }
 
@@ -92,4 +92,5 @@
 {
     // store the combobox selection value before the popup is shown
+  // if there is no selection, combo->GetSelection() returns -1
     g_SelectionBeforePopup = combo->GetSelection();
 }
@@ -126,10 +127,10 @@
       GTK_SIGNAL_FUNC(gtk_text_changed_callback), (gpointer)combo );
 
-    // throw a SELECTED event only if the combobox popup is hidden
+    // throw a SELECTED event only if the combobox popup is hidden (-2)
     // because when combobox popup is shown, gtk_combo_select_child_callback is
     // called each times the mouse is over an item with a pressed button so a lot
     // of SELECTED event could be generated if the user keep the mouse button down
     // and select other items ...
-    if (g_SelectionBeforePopup == -1)
+    if (g_SelectionBeforePopup == -2)
     {
         wxCommandEvent event( wxEVT_COMMAND_COMBOBOX_SELECTED, combo->GetId() );

 	  	 
