Index: button.cpp
===================================================================
--- button.cpp	(revision 61504)
+++ button.cpp	(working copy)
@@ -959,16 +959,15 @@
             m_imageData = new wxODButtonImageData(this, bitmap);
             MakeOwnerDrawn();
         }
-
-        // if a bitmap was assigned to the bitmap, its best size must be
-        // changed to account for it
-        InvalidateBestSize();
     }
     else
     {
         m_imageData->SetBitmap(bitmap, which);
     }
 
+    if ( which == State_Normal )
+        InvalidateBestSize();
+
     Refresh();
 }
 
