diff -r -u -p a/docs/doxygen/mainpages/cat_classes.h b/docs/doxygen/mainpages/cat_classes.h
--- a/docs/doxygen/mainpages/cat_classes.h	2020-02-20 06:40:02.856514600 +0100
+++ b/docs/doxygen/mainpages/cat_classes.h	2020-02-26 07:03:12.000333500 +0100
@@ -667,6 +667,7 @@ Related overview: @ref overview_file
 @li wxFile: Low-level file input/output class.
 @li wxFFile: Another low-level file input/output class.
 @li wxTempFile: Class to safely replace an existing file
+@li wxTempFFile: Another class to safely replace an existing file
 @li wxTextFile: Class for working with text files as with arrays of lines
 @li wxStandardPaths: Paths for standard directories
 @li wxPathList: A class to help search multiple paths
@@ -702,6 +703,7 @@ Related overview: @ref overview_stream
 @li wxFFileInputStream: Another file input stream class
 @li wxFFileOutputStream: Another file output stream class
 @li wxTempFileOutputStream: Stream to safely replace an existing file
+@li wxTempFFileOutputStream: Another stream to safely replace an existing file
 @li wxStringInputStream: String input stream class
 @li wxStringOutputStream: String output stream class
 @li wxLZMAInputStream: LZMA decompression stream class
