Index: src/msw/treectrl.cpp
===================================================================
--- src/msw/treectrl.cpp	(revision 63080)
+++ src/msw/treectrl.cpp	(working copy)
@@ -2708,15 +2708,20 @@
         HTREEITEM htOldItem = TreeView_GetSelection(GetHwnd());
         HTREEITEM htItem = TreeView_HitTest(GetHwnd(), &tvht);
 
+
         switch ( nMsg )
         {
             case WM_LBUTTONDOWN:
-                if ( !isMultiple )
+			{
+				if ( !isMultiple )
                     break;
 
+
                 m_htClickedItem.Unset();
+				bool fullrow = (GetWindowStyle() & wxTR_FULL_ROW_HIGHLIGHT) > 0;
 
-                if ( !(tvht.flags & TVHT_ONITEM) )
+				if ( (!fullrow && !(tvht.flags & TVHT_ONITEM) ) 
+					|| (fullrow && !(tvht.flags & (TVHT_ONITEM | TVHT_ONITEMRIGHT | TVHT_ONITEMINDENT) ) ) )
                 {
                     if ( tvht.flags & TVHT_ONITEMBUTTON )
                     {
@@ -2921,6 +2926,7 @@
                     }
                 }
                 break;
+			}
 
             case WM_RBUTTONDOWN:
                 if ( !isMultiple )
