Index: src/common/glcmn.cpp
===================================================================
--- src/common/glcmn.cpp	(revision 61244)
+++ src/common/glcmn.cpp	(working copy)
@@ -120,25 +120,21 @@
 /* static */
 bool wxGLCanvasBase::IsExtensionInList(const char *list, const char *extension)
 {
-    if ( !list )
-        return false;
+    size_t extension_len = strlen(extension);
+    const char *p = list;
 
-    for ( const char *p = list; *p; p++ )
+    while ((p = wxStrstr(p, extension)) != NULL)
     {
-        // advance up to the next possible match
-        p = wxStrstr(p, extension);
-        if ( !p )
-            break;
+        const char *p_after_ext = p + extension_len;
 
         // check that the extension appears at the beginning/ending of the list
         // or is preceded/followed by a space to avoid mistakenly finding
         // "glExtension" in a list containing some "glFunkyglExtension"
-        if ( (p == list || p[-1] == ' ') )
-        {
-            char c = p[strlen(extension)];
-            if ( c == '\0' || c == ' ' )
+        if ( p == list || p[-1] == ' ' )
+            if ( *p_after_ext == '\0' || *p_after_ext == ' ' )
                 return true;
-        }
+
+        p = p_after_ext;
     }
 
     return false;
Index: include/wx/glcanvas.h
===================================================================
--- include/wx/glcanvas.h	(revision 61244)
+++ include/wx/glcanvas.h	(working copy)
@@ -158,6 +158,9 @@
     // check if the given extension name is present in the space-separated list
     // of extensions supported by the current implementation such as returned
     // by glXQueryExtensionsString() or glGetString(GL_EXTENSIONS)
+    //
+    // list is a  non-NULL pointer to a null-terminated string
+    // extension is a non-NULL pointer to a non-empty null-terminated string
     static bool IsExtensionInList(const char *list, const char *extension);
 
     wxPalette m_palette;
