Index: src/gtk/anybutton.cpp
===================================================================
--- src/gtk/anybutton.cpp	(revision 78355)
+++ src/gtk/anybutton.cpp	(working copy)
@@ -162,7 +162,7 @@
         // valid bitmap
         State state = GTKGetCurrentState();
 
-        GTKDoShowBitmap(m_bitmaps[state]);
+        GTKDoShowBitmap(DoGetBitmap(state));
     }
 }
 
Index: src/gtk/tglbtn.cpp
===================================================================
--- src/gtk/tglbtn.cpp	(revision 78355)
+++ src/gtk/tglbtn.cpp	(working copy)
@@ -247,6 +247,30 @@
     return ret;
 }
 
+// Get the bitmap for this control.
+wxBitmap wxToggleButton::DoGetBitmap(wxAnyButton::State state) const
+{
+    // If button is in the "on" state and we'd otherwise return the
+    // "normal" bitmap, return the "pressed" bitmap instead...
+    wxAnyButton::State returnState = state;
+
+    if ( state == wxAnyButton::State_Normal && GetValue() )
+    {
+        returnState = wxAnyButton::State_Pressed;
+    }
+
+    wxBitmap result = wxAnyButton::DoGetBitmap(returnState);
+
+    // ... unless the pressed bitmap isn't valid, in which case return
+    // the "normal" bitmap.
+    if ( !result.IsOk() && state != returnState )
+    {
+        result = wxAnyButton::DoGetBitmap(wxAnyButton::State_Normal);
+    }
+
+    return result;
+}
+
 // static
 wxVisualAttributes
 wxToggleButton::GetClassDefaultAttributes(wxWindowVariant WXUNUSED(variant))
Index: include/wx/gtk/tglbtn.h
===================================================================
--- include/wx/gtk/tglbtn.h	(revision 78355)
+++ include/wx/gtk/tglbtn.h	(working copy)
@@ -60,6 +60,7 @@
 
     virtual wxSize DoGetBestSize() const wxOVERRIDE;
     virtual void DoApplyWidgetStyle(GtkRcStyle *style) wxOVERRIDE;
+    virtual wxBitmap DoGetBitmap(State which) const wxOVERRIDE;
 
 #if wxUSE_MARKUP
     virtual bool DoSetLabelMarkup(const wxString& markup) wxOVERRIDE;
