diff --git a/../../../wxWidgets-3.1.0/samples/minimal/minimal.cpp b/minimal.cpp
index f765a70..1391b79 100644
--- a/../../../wxWidgets-3.1.0/samples/minimal/minimal.cpp
+++ b/minimal.cpp
@@ -72,6 +72,19 @@ private:
     wxDECLARE_EVENT_TABLE();
 };
 
+class SubFrame : public wxFrame
+{
+public:
+    SubFrame(wxFrame * parent, const wxString& title);
+    void OnButton(wxCommandEvent& event);
+};
+
+class Dialog : public wxDialog
+{
+public:
+    Dialog(wxFrame * parent, const wxString& msg);
+};
+
 // ----------------------------------------------------------------------------
 // constants
 // ----------------------------------------------------------------------------
@@ -177,6 +190,10 @@ MyFrame::MyFrame(const wxString& title)
     CreateStatusBar(2);
     SetStatusText("Welcome to wxWidgets!");
 #endif // wxUSE_STATUSBAR
+    
+    // create sub frame
+    SubFrame * subframe = new SubFrame(this, "SubFrame");
+    subframe->Show(true);
 }
 
 
@@ -203,3 +220,27 @@ void MyFrame::OnAbout(wxCommandEvent& WXUNUSED(event))
                  wxOK | wxICON_INFORMATION,
                  this);
 }
+
+SubFrame::SubFrame(wxFrame * parent, const wxString& title)
+: wxFrame(parent, wxID_ANY, title, wxDefaultPosition, wxDefaultSize, wxDEFAULT_FRAME_STYLE | wxFRAME_FLOAT_ON_PARENT)
+{
+    wxSizer* sizer = new wxBoxSizer(wxHORIZONTAL);
+    wxButton* btn = new wxButton(this, wxID_ANY, "Show model...");
+    btn->Bind(wxEVT_BUTTON, &SubFrame::OnButton, this);
+    sizer->Add(btn, wxSizerFlags().Center());
+}
+
+void SubFrame::OnButton(wxCommandEvent& event)
+{
+    Dialog dlg(this, "Now swap applications back and forth.\nThis dialog will disappear behind the sub frame.\nForcing you to quit the app.");
+    dlg.ShowModal();
+}
+
+Dialog::Dialog(wxFrame * parent, const wxString& msg)
+: wxDialog(parent, wxID_ANY, "Dialog")
+{
+    wxSizer* sizer = new wxBoxSizer(wxHORIZONTAL);
+    wxStaticText* txt = new wxStaticText(this, wxID_ANY, msg);
+    sizer->Add(txt, wxSizerFlags().Center());
+}
+
