diff -bu1 -r wxWindows/contrib/include/wx/mmedia/vidwin.h wxWin/contrib/include/wx/mmedia/vidwin.h
--- wxWindows/contrib/include/wx/mmedia/vidwin.h	2003-09-11 12:55:17.000000000 +0200
+++ wxWin/contrib/include/wx/mmedia/vidwin.h	2004-01-12 19:32:03.000000000 +0100
@@ -17,2 +17,6 @@
 
+#if defined(__WINDOWS__) && !defined(__MINGW32__) && !defined(__WATCOMC__)
+// versions of Open Watcom and MinGW tested against this source does not
+// deliver "digitalv.h" required in this feature
+
 // ----------------------------------------------------------------------------
@@ -102 +106,3 @@
 #endif
+
+#endif
diff -bu1 -r wxWindows/contrib/samples/mmedia/mmbman.cpp wxWin/contrib/samples/mmedia/mmbman.cpp
--- wxWindows/contrib/samples/mmedia/mmbman.cpp	2003-09-11 12:55:19.000000000 +0200
+++ wxWin/contrib/samples/mmedia/mmbman.cpp	2004-01-12 19:33:01.000000000 +0100
@@ -359,3 +359,5 @@
     m_video_driver = new wxVideoXANIM(filename);
-#elif defined(__WIN32__)
+#elif defined(__WINDOWS__) && !defined(__MINGW32__) && !defined(__WATCOMC__)
+    // versions of Open Watcom and MinGW tested against this source does not
+    // deliver "digitalv.h" required in this feature
     m_video_driver = new wxVideoWindows(filename);
diff -bu1 -r wxWindows/contrib/src/mmedia/cdwin.cpp wxWin/contrib/src/mmedia/cdwin.cpp
--- wxWindows/contrib/src/mmedia/cdwin.cpp	2003-08-22 13:00:25.000000000 +0200
+++ wxWin/contrib/src/mmedia/cdwin.cpp	2004-01-12 19:59:05.000000000 +0100
@@ -48,3 +48,3 @@
 wxCDAudioWin::wxCDAudioWin(void)
-  : wxCDAudio(), m_trksize(NULL), m_trkpos(NULL), m_ok(TRUE), m_toc(NULL)
+  : wxCDAudio(), m_trksize(NULL), m_trkpos(NULL), m_toc(NULL), m_ok(TRUE)
 {
@@ -52,3 +52,2 @@
   MCI_SET_PARMS set_struct;
-  DWORD ret;
 
@@ -56,3 +55,3 @@
   open_struct.lpstrDeviceType = _T("cdaudio");
-  ret = mciSendCommand(NULL, MCI_OPEN, MCI_OPEN_TYPE,
+  DWORD ret = mciSendCommand((MCIDEVICEID)NULL, MCI_OPEN, MCI_OPEN_TYPE,
                        (DWORD)&open_struct);
@@ -65,3 +64,3 @@
   set_struct.dwTimeFormat = MCI_FORMAT_MSF;
-  ret = mciSendCommand(m_internal->dev_id, MCI_SET, MCI_SET_TIME_FORMAT,
+  mciSendCommand(m_internal->dev_id, MCI_SET, MCI_SET_TIME_FORMAT,
                        (DWORD)(LPVOID)&set_struct);
@@ -71,3 +70,3 @@
   set_struct.dwTimeFormat = MCI_FORMAT_TMSF;
-  ret = mciSendCommand(m_internal->dev_id, MCI_SET, MCI_SET_TIME_FORMAT,
+  mciSendCommand(m_internal->dev_id, MCI_SET, MCI_SET_TIME_FORMAT,
                        (DWORD)(LPVOID)&set_struct);
@@ -78,3 +77,3 @@
   if (m_ok) {
-    mciSendCommand(m_internal->dev_id, MCI_CLOSE, 0, NULL);
+    mciSendCommand(m_internal->dev_id, MCI_CLOSE, 0,(DWORD_PTR)NULL);
     delete m_toc;
@@ -204,3 +203,3 @@
   MCI_STATUS_PARMS status_struct;
-  wxCDtime cd_time = {-1, -1, -1, -1};
+  wxCDtime cd_time = {0,0,0,0};
 
diff -bu1 -r wxWindows/contrib/src/mmedia/sndaiff.cpp wxWin/contrib/src/mmedia/sndaiff.cpp
--- wxWindows/contrib/src/mmedia/sndaiff.cpp	2003-08-22 13:00:25.000000000 +0200
+++ wxWin/contrib/src/mmedia/sndaiff.cpp	2004-01-12 12:38:09.000000000 +0100
@@ -119,2 +119,3 @@
     len = data.Read32(); 
+    wxUnusedVar(len);
     FAIL_WITH(m_input->LastRead() != 4, wxSOUND_INVSTRM);
diff -bu1 -r wxWindows/contrib/src/mmedia/sndbase.cpp wxWin/contrib/src/mmedia/sndbase.cpp
--- wxWindows/contrib/src/mmedia/sndbase.cpp	2001-10-30 14:28:45.000000000 +0100
+++ wxWin/contrib/src/mmedia/sndbase.cpp	2004-01-12 17:09:16.000000000 +0100
@@ -36,4 +36,4 @@
 {
-    wxLogFatalError(wxT("In wxSoundFormatBase::Clone() but I should"
-                        " not be there"));
+    wxLogFatalError(wxT("In wxSoundFormatBase::Clone() but I should")
+                    wxT(" not be there"));
     return NULL;
diff -bu1 -r wxWindows/contrib/src/mmedia/sndcpcm.cpp wxWin/contrib/src/mmedia/sndcpcm.cpp
--- wxWindows/contrib/src/mmedia/sndcpcm.cpp	2003-08-03 20:11:48.000000000 +0200
+++ wxWin/contrib/src/mmedia/sndcpcm.cpp	2004-01-12 12:38:53.000000000 +0100
@@ -398,2 +398,3 @@
     wxASSERT( SetSoundFormatReturn );
+    wxUnusedVar( SetSoundFormatReturn );
     
diff -bu1 -r wxWindows/contrib/src/mmedia/sndwav.cpp wxWin/contrib/src/mmedia/sndwav.cpp
--- wxWindows/contrib/src/mmedia/sndwav.cpp	2003-08-22 13:00:25.000000000 +0200
+++ wxWin/contrib/src/mmedia/sndwav.cpp	2004-01-12 12:39:28.000000000 +0100
@@ -198,2 +198,3 @@
     len = data.Read32();
+    wxUnusedVar(len);
     FAIL_WITH(m_input->LastRead() != 4, wxSOUND_INVSTRM);
diff -bu1 -r wxWindows/contrib/src/mmedia/sndwin.cpp wxWin/contrib/src/mmedia/sndwin.cpp
--- wxWindows/contrib/src/mmedia/sndwin.cpp	2003-09-11 12:55:21.000000000 +0200
+++ wxWin/contrib/src/mmedia/sndwin.cpp	2004-01-12 19:30:39.000000000 +0100
@@ -55,3 +55,3 @@
     return NULL;
-  return (wxSoundStreamWin *)node->Data();
+  return (wxSoundStreamWin *)node->GetData();
 }
diff -bu1 -r wxWindows/contrib/src/mmedia/vidwin.cpp wxWin/contrib/src/mmedia/vidwin.cpp
--- wxWindows/contrib/src/mmedia/vidwin.cpp	2003-09-11 12:55:21.000000000 +0200
+++ wxWin/contrib/src/mmedia/vidwin.cpp	2004-01-12 19:32:40.000000000 +0100
@@ -20,3 +20,5 @@
 
-#if defined(__WINDOWS__) && !defined(__GNUWIN32__) && !defined(__WATCOMC__)
+#if defined(__WINDOWS__) && !defined(__MINGW32__) && !defined(__WATCOMC__)
+// versions of Open Watcom and MinGW tested against this source does not
+// deliver "digitalv.h" required in this feature
 
@@ -88,3 +90,3 @@
     openStruct.lpstrDeviceType = _T("avivideo");
-    openStruct.lpstrElementName = (wxChar *)m_filename.mb_str().data();
+    openStruct.lpstrElementName = (wxChar *)m_filename.c_str();
     openStruct.hWndParent = 0;

 	  	 
