diff --git a/src/osx/core/bitmap.cpp b/src/osx/core/bitmap.cpp
index e2d640e..8432ea9 100644
--- a/src/osx/core/bitmap.cpp
+++ b/src/osx/core/bitmap.cpp
@@ -1241,10 +1241,10 @@ wxBitmap::wxBitmap(const wxImage& image, int depth, double scale)
         if ( destinationstart != NULL && data != NULL )
         {
             const unsigned char *alpha = hasAlpha ? image.GetAlpha() : NULL ;
-            for (int y = 0; y < height; destinationstart += M_BITMAPDATA->GetBytesPerRow(), y++)
+            for (int y = 0; y < bitmapRefData->GetHeight(); destinationstart += M_BITMAPDATA->GetBytesPerRow(), y++)
             {
                 unsigned char * destination = destinationstart;
-                for (int x = 0; x < width; x++)
+                for (int x = 0; x < bitmapRefData->GetWidth(); x++)
                 {
                     if ( hasAlpha )
                     {
