Index: configure
===================================================================
--- configure	(revision 78054)
+++ configure	(working copy)
@@ -3334,6 +3334,52 @@
   as_fn_set_status $ac_retval
 
 } # ac_fn_cxx_try_run
+
+# ac_fn_c_check_decl LINENO SYMBOL VAR INCLUDES
+# ---------------------------------------------
+# Tests whether SYMBOL is declared in INCLUDES, setting cache variable VAR
+# accordingly.
+ac_fn_c_check_decl ()
+{
+  as_lineno=${as_lineno-"$1"} as_lineno_stack=as_lineno_stack=$as_lineno_stack
+  as_decl_name=`echo $2|sed 's/ *(.*//'`
+  as_decl_use=`echo $2|sed -e 's/(/((/' -e 's/)/) 0&/' -e 's/,/) 0& (/g'`
+  { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether $as_decl_name is declared" >&5
+$as_echo_n "checking whether $as_decl_name is declared... " >&6; }
+if eval \${$3+:} false; then :
+  $as_echo_n "(cached) " >&6
+else
+  cat confdefs.h - <<_ACEOF >conftest.$ac_ext
+/* end confdefs.h.  */
+$4
+int
+main ()
+{
+#ifndef $as_decl_name
+#ifdef __cplusplus
+  (void) $as_decl_use;
+#else
+  (void) $as_decl_name;
+#endif
+#endif
+
+  ;
+  return 0;
+}
+_ACEOF
+if ac_fn_c_try_compile "$LINENO"; then :
+  eval "$3=yes"
+else
+  eval "$3=no"
+fi
+rm -f core conftest.err conftest.$ac_objext conftest.$ac_ext
+fi
+eval ac_res=\$$3
+	       { $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_res" >&5
+$as_echo "$ac_res" >&6; }
+  eval $as_lineno_stack; ${as_lineno_stack:+:} unset as_lineno
+
+} # ac_fn_c_check_decl
 cat >config.log <<_ACEOF
 This file contains any messages produced by compilers while
 running configure, to aid debugging if configure makes a mistake.
@@ -34596,8 +34642,104 @@
     fi
 
         SAMPLES_SUBDIRS="$SAMPLES_SUBDIRS keyboard"
+
+        ac_fn_c_check_decl "$LINENO" "strerror_r" "ac_cv_have_decl_strerror_r" "$ac_includes_default"
+if test "x$ac_cv_have_decl_strerror_r" = xyes; then :
+  ac_have_decl=1
+else
+  ac_have_decl=0
 fi
 
+cat >>confdefs.h <<_ACEOF
+#define HAVE_DECL_STRERROR_R $ac_have_decl
+_ACEOF
+
+for ac_func in strerror_r
+do :
+  ac_fn_c_check_func "$LINENO" "strerror_r" "ac_cv_func_strerror_r"
+if test "x$ac_cv_func_strerror_r" = xyes; then :
+  cat >>confdefs.h <<_ACEOF
+#define HAVE_STRERROR_R 1
+_ACEOF
+
+fi
+done
+
+{ $as_echo "$as_me:${as_lineno-$LINENO}: checking whether strerror_r returns char *" >&5
+$as_echo_n "checking whether strerror_r returns char *... " >&6; }
+if ${ac_cv_func_strerror_r_char_p+:} false; then :
+  $as_echo_n "(cached) " >&6
+else
+
+    ac_cv_func_strerror_r_char_p=no
+    if test $ac_cv_have_decl_strerror_r = yes; then
+      cat confdefs.h - <<_ACEOF >conftest.$ac_ext
+/* end confdefs.h.  */
+$ac_includes_default
+int
+main ()
+{
+
+	  char buf[100];
+	  char x = *strerror_r (0, buf, sizeof buf);
+	  char *p = strerror_r (0, buf, sizeof buf);
+	  return !p || x;
+
+  ;
+  return 0;
+}
+_ACEOF
+if ac_fn_c_try_compile "$LINENO"; then :
+  ac_cv_func_strerror_r_char_p=yes
+fi
+rm -f core conftest.err conftest.$ac_objext conftest.$ac_ext
+    else
+      # strerror_r is not declared.  Choose between
+      # systems that have relatively inaccessible declarations for the
+      # function.  BeOS and DEC UNIX 4.0 fall in this category, but the
+      # former has a strerror_r that returns char*, while the latter
+      # has a strerror_r that returns `int'.
+      # This test should segfault on the DEC system.
+      if test "$cross_compiling" = yes; then :
+  :
+else
+  cat confdefs.h - <<_ACEOF >conftest.$ac_ext
+/* end confdefs.h.  */
+$ac_includes_default
+	extern char *strerror_r ();
+int
+main ()
+{
+char buf[100];
+	  char x = *strerror_r (0, buf, sizeof buf);
+	  return ! isalpha (x);
+  ;
+  return 0;
+}
+_ACEOF
+if ac_fn_c_try_run "$LINENO"; then :
+  ac_cv_func_strerror_r_char_p=yes
+fi
+rm -f core *.core core.conftest.* gmon.out bb.out conftest$ac_exeext \
+  conftest.$ac_objext conftest.beam conftest.$ac_ext
+fi
+
+    fi
+
+fi
+{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_func_strerror_r_char_p" >&5
+$as_echo "$ac_cv_func_strerror_r_char_p" >&6; }
+if test $ac_cv_func_strerror_r_char_p = yes; then
+
+$as_echo "#define STRERROR_R_CHAR_P 1" >>confdefs.h
+
+fi
+
+    if test "$ac_cv_func_strerror_r" != "yes"; then
+      as_fn_error $? "strerror_r is required" "$LINENO" 5
+    fi
+fi
+
 if test "$wxUSE_LONGLONG" = "yes"; then
   $as_echo "#define wxUSE_LONGLONG 1" >>confdefs.h
 
Index: setup.h.in
===================================================================
--- setup.h.in	(revision 78054)
+++ setup.h.in	(working copy)
@@ -1249,6 +1249,12 @@
 /* Define if you have kqueu_xxx() functions. */
 #undef wxHAS_KQUEUE
 
+/* Define if streror_r is available */
+#undef HAVE_STRERROR_R
+
+/* Define if strerror_r is available and returns char* */
+#undef STRERROR_R_CHAR_P
+
 /* -------------------------------------------------------------------------
    Win32 adjustments section
    ------------------------------------------------------------------------- */
Index: configure.in
===================================================================
--- configure.in	(revision 78054)
+++ configure.in	(working copy)
@@ -5726,6 +5726,12 @@
 
     dnl the keyboard sample requires wxUSE_LOG
     SAMPLES_SUBDIRS="$SAMPLES_SUBDIRS keyboard"
+
+    dnl wxSysErrorMsg uses strerror_r
+    AC_FUNC_STRERROR_R
+    if test "$ac_cv_func_strerror_r" != "yes"; then
+      AC_MSG_ERROR([strerror_r is required])
+    fi
 fi
 
 if test "$wxUSE_LONGLONG" = "yes"; then
Index: docs/changes.txt
===================================================================
--- docs/changes.txt	(revision 78054)
+++ docs/changes.txt	(working copy)
@@ -1,4 +1,4 @@
--------------------------------------------------------------------------------
+﻿-------------------------------------------------------------------------------
                              wxWidgets Change Log
 -------------------------------------------------------------------------------
 
@@ -22,8 +22,8 @@
 
 - "webview" library is not included in `wx-config --libs` output any more, you
   need to request it explicitly, e.g. `wx-config --libs std,webview`.
+- wxSysErrorMsg now returns wxString instead of wxChar*
 
-
 3.1.0: (released 2014-xx-xx)
 ----------------------------
 
Index: include/wx/log.h
===================================================================
--- include/wx/log.h	(revision 78054)
+++ include/wx/log.h	(working copy)
@@ -1188,7 +1188,7 @@
 WXDLLIMPEXP_BASE unsigned long wxSysErrorCode();
 
 // return the error message for given (or last if 0) error code
-WXDLLIMPEXP_BASE const wxChar* wxSysErrorMsg(unsigned long nErrCode = 0);
+WXDLLIMPEXP_BASE wxString wxSysErrorMsg(unsigned long nErrCode = 0);
 
 // ----------------------------------------------------------------------------
 // define wxLog<level>() functions which can be used by application instead of
@@ -1395,7 +1395,7 @@
 
 // Dummy macros to replace some functions.
 #define wxSysErrorCode() (unsigned long)0
-#define wxSysErrorMsg( X ) (const wxChar*)NULL
+#define wxSysErrorMsg( X ) wxEmptyString
 
 // Fake symbolic trace masks... for those that are used frequently
 #define wxTRACE_OleCalls wxEmptyString // OLE interface calls
Index: interface/wx/log.h
===================================================================
--- interface/wx/log.h	(revision 78054)
+++ interface/wx/log.h	(working copy)
@@ -1213,6 +1213,8 @@
     Returns the error code from the last system call. This function uses
     @c errno on Unix platforms and @c GetLastError under Win32.
 
+    This function is thread-safe, each thread keeps its own error code.
+
     @see wxSysErrorMsg(), wxLogSysError()
 
     @header{wx/log.h}
@@ -1224,11 +1226,13 @@
     @a errCode is 0 (default), the last error code (as returned by
     wxSysErrorCode()) is used.
 
+    This function is thread-safe.
+
     @see wxSysErrorCode(), wxLogSysError()
 
     @header{wx/log.h}
 */
-const wxChar* wxSysErrorMsg(unsigned long errCode = 0);
+wxString wxSysErrorMsg(unsigned long errCode = 0);
 
 //@}
 
Index: src/common/log.cpp
===================================================================
--- src/common/log.cpp	(revision 78054)
+++ src/common/log.cpp	(working copy)
@@ -1061,14 +1061,35 @@
 #endif  //Win/Unix
 }
 
+#if !defined(__WINDOWS__) || defined(__WXMICROWIN__)
+namespace {
+// Wrap around strerror_r which exists in two different versions
+bool wxStrerror_r(int errnum, char *buf, size_t buflen)
+{
+    #ifdef STRERROR_R_CHAR_P
+        // strerror_r returns char*, null indicates failure.
+        bool ret = strerror_r(errnum, buf, buflen) != 0;
+    #else
+        // strerror_r returns int, 0 indicates success.
+        bool ret = strerror_r(errnum, buf, buflen) == 0;
+    #endif
+    if (!ret)
+    {
+        *buf = wxS('\0');
+    }
+    return ret;
+}
+}
+#endif
+
 // get error message from system
-const wxChar *wxSysErrorMsg(unsigned long nErrCode)
+wxString wxSysErrorMsg(unsigned long nErrCode)
 {
     if ( nErrCode == 0 )
         nErrCode = wxSysErrorCode();
 
 #if defined(__WINDOWS__) && !defined(__WXMICROWIN__)
-    static wxChar s_szBuf[1024];
+    wxChar szBuf[1024];
 
     // get error message from system
     LPVOID lpMsgBuf;
@@ -1085,8 +1106,8 @@
     {
         // if this happens, something is seriously wrong, so don't use _() here
         // for safety
-        wxSprintf(s_szBuf, wxS("unknown error %lx"), nErrCode);
-        return s_szBuf;
+        wxSprintf(szBuf, wxS("unknown error %lx"), nErrCode);
+        return szBuf;
     }
 
 
@@ -1095,35 +1116,37 @@
 #if !defined(__SMARTPHONE__) /* of WinCE */
     if( lpMsgBuf != 0 )
     {
-        wxStrlcpy(s_szBuf, (const wxChar *)lpMsgBuf, WXSIZEOF(s_szBuf));
+        wxStrlcpy(szBuf, (const wxChar *)lpMsgBuf, WXSIZEOF(szBuf));
 
         LocalFree(lpMsgBuf);
 
         // returned string is capitalized and ended with '\r\n' - bad
-        s_szBuf[0] = (wxChar)wxTolower(s_szBuf[0]);
-        size_t len = wxStrlen(s_szBuf);
+        szBuf[0] = (wxChar)wxTolower(szBuf[0]);
+        size_t len = wxStrlen(szBuf);
         if ( len > 0 ) {
             // truncate string
-            if ( s_szBuf[len - 2] == wxS('\r') )
-                s_szBuf[len - 2] = wxS('\0');
+            if ( szBuf[len - 2] == wxS('\r') )
+                szBuf[len - 2] = wxS('\0');
         }
     }
     else
 #endif // !__SMARTPHONE__
     {
-        s_szBuf[0] = wxS('\0');
+        szBuf[0] = wxS('\0');
     }
 
-    return s_szBuf;
+    return szBuf;
 #else // !__WINDOWS__
+    wxChar szBuf[1024];
     #if wxUSE_UNICODE
-        static wchar_t s_wzBuf[1024];
-        wxConvCurrent->MB2WC(s_wzBuf, strerror((int)nErrCode),
-                             WXSIZEOF(s_wzBuf) - 1);
-        return s_wzBuf;
-    #else
-        return strerror((int)nErrCode);
-    #endif
+        char mbstr[1024];
+        wxStrerror_r((int)nErrCode, mbstr, WXSIZEOF(mbstr));
+        wxConvCurrent->MB2WC(szBuf, mbstr,
+                             WXSIZEOF(szBuf) - 1);
+    #else
+        wxStrerror_r((int)nErrCode, szBuf, WXSIZEOF(szBuf));
+    #endif
+    return szBuf;
 #endif  // __WINDOWS__/!__WINDOWS__
 }
 
