Index: src/mac/carbon/textctrl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/mac/carbon/textctrl.cpp,v
retrieving revision 1.179
diff -b -u -2 -r1.179 textctrl.cpp
--- src/mac/carbon/textctrl.cpp	2006/03/28 11:02:32	1.179
+++ src/mac/carbon/textctrl.cpp	2006/04/07 14:51:40
@@ -877,4 +877,11 @@
     bool eat_key = false ;
 
+    if ( key == 'a' && event.MetaDown() )
+    {
+        SelectAll() ;     
+
+        return ;
+    }
+    
     if ( key == 'c' && event.MetaDown() )
     {

 	  	 
