Index: docs/latex/wx/process.tex
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/docs/latex/wx/process.tex,v
retrieving revision 1.26
diff -u -r1.26 process.tex
--- docs/latex/wx/process.tex	2005/02/06 02:47:30	1.26
+++ docs/latex/wx/process.tex	2006/10/30 04:34:53
@@ -260,6 +260,12 @@
 
 \helpref{wxExecute}{wxexecute}
 
+\membersection{wxProcess::GetPid}\label{wxprocessgetpid}
+
+\constfunc{long}{GetPid}{\void}
+
+Returns the process ID of the process launched by \helpref{Open}{wxprocessopen}.
+
 \membersection{wxProcess::Redirect}\label{wxprocessredirect}
 
 \func{void}{Redirect}{\void}
Index: include/wx/process.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/process.h,v
retrieving revision 1.39
diff -u -r1.39 process.h
--- include/wx/process.h	2005/10/09 15:48:09	1.39
+++ include/wx/process.h	2006/10/30 04:34:53
@@ -62,6 +62,9 @@
 
     virtual ~wxProcess();
 
+    // get the process ID of the process executed by Open()
+    long GetPid() const { return m_pid; }
+
     // may be overridden to be notified about process termination
     virtual void OnTerminate(int pid, int status);
 
@@ -103,8 +106,10 @@
 
 protected:
     void Init(wxEvtHandler *parent, int id, int flags);
+    void SetPid(long pid) { m_pid = pid; }
 
     int m_id;
+    long m_pid;
 
 #if wxUSE_STREAMS
     // these streams are connected to stdout, stderr and stdin of the child
Index: samples/exec/exec.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/samples/exec/exec.cpp,v
retrieving revision 1.38
diff -u -r1.38 exec.cpp
--- samples/exec/exec.cpp	2005/09/23 12:51:43	1.38
+++ samples/exec/exec.cpp	2006/10/30 04:34:54
@@ -774,6 +774,8 @@
         return;
     }
 
+    wxLogVerbose(_T("PID of the new process: %ld"), process->GetPid());
+
     wxOutputStream *out = process->GetOutputStream();
     if ( !out )
     {
Index: src/common/process.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/common/process.cpp,v
retrieving revision 1.26
diff -u -r1.26 process.cpp
--- src/common/process.cpp	2005/10/09 15:48:29	1.26
+++ src/common/process.cpp	2006/10/30 04:34:54
@@ -49,6 +49,7 @@
         SetNextHandler(parent);
 
     m_id         = id;
+    m_pid        = 0;
     m_redirect   = (flags & wxPROCESS_REDIRECT) != 0;
 
 #if wxUSE_STREAMS
@@ -63,12 +64,15 @@
 {
     wxASSERT_MSG( !(flags & wxEXEC_SYNC), wxT("wxEXEC_SYNC should not be used." ));
     wxProcess *process = new wxProcess(wxPROCESS_REDIRECT);
-    if ( !wxExecute(cmd, flags, process) )
+    long pid = wxExecute(cmd, flags, process);
+    if( !pid )
     {
         // couldn't launch the process
         delete process;
         return NULL;
     }
+
+    process->SetPid(pid);
 
     return process;
 }

 	  	 
