diff -ur ../wxWidgets/src/mac/carbon/cfsocket.cpp ./src/mac/carbon/cfsocket.cpp
--- ../wxWidgets/src/mac/carbon/cfsocket.cpp	Sun May 23 13:52:29 2004
+++ ./src/mac/carbon/cfsocket.cpp	Tue Nov 30 17:59:50 2004
@@ -28,7 +28,6 @@
 
 #include "wx/app.h"
 #include "wx/apptrait.h"
-#include "wx/defs.h"
 #include "wx/object.h"
 #include "wx/string.h"
 #include "wx/timer.h"
diff -ur ../wxWidgets/src/mac/carbon/dataobj.cpp ./src/mac/carbon/dataobj.cpp
--- ../wxWidgets/src/mac/carbon/dataobj.cpp	Sun May 23 13:52:30 2004
+++ ./src/mac/carbon/dataobj.cpp	Tue Nov 30 18:00:02 2004
@@ -17,7 +17,7 @@
 // headers
 // ----------------------------------------------------------------------------
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
   #pragma implementation "dataobj.h"
 #endif
 
@@ -27,7 +27,6 @@
 #ifndef WX_PRECOMP
 #include "wx/intl.h"
 #endif
-#include "wx/defs.h"
 
 #include "wx/log.h"
 #include "wx/dataobj.h"
diff -ur ../wxWidgets/src/mac/carbon/dcprint.cpp ./src/mac/carbon/dcprint.cpp
--- ../wxWidgets/src/mac/carbon/dcprint.cpp	Sun May 23 13:52:30 2004
+++ ./src/mac/carbon/dcprint.cpp	Tue Nov 30 18:00:17 2004
@@ -9,7 +9,7 @@
 // Licence:       wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "dcprint.h"
 #endif
 
diff -ur ../wxWidgets/src/mac/carbon/dirmac.cpp ./src/mac/carbon/dirmac.cpp
--- ../wxWidgets/src/mac/carbon/dirmac.cpp	Tue Sep 28 11:36:46 2004
+++ ./src/mac/carbon/dirmac.cpp	Tue Nov 30 18:00:26 2004
@@ -17,7 +17,7 @@
 // headers
 // ----------------------------------------------------------------------------
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
     #pragma implementation "dir.h"
 #endif
 
diff -ur ../wxWidgets/src/mac/carbon/display.cpp ./src/mac/carbon/display.cpp
--- ../wxWidgets/src/mac/carbon/display.cpp	Tue Sep 28 11:36:46 2004
+++ ./src/mac/carbon/display.cpp	Tue Nov 30 18:00:34 2004
@@ -9,7 +9,7 @@
 // Licence:     wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
     #pragma implementation "display.h"
 #endif
 
diff -ur ../wxWidgets/src/mac/carbon/fontdlg.cpp ./src/mac/carbon/fontdlg.cpp
--- ../wxWidgets/src/mac/carbon/fontdlg.cpp	Wed Nov 24 14:50:12 2004
+++ ./src/mac/carbon/fontdlg.cpp	Tue Nov 30 18:00:41 2004
@@ -17,7 +17,7 @@
 // headers
 // ---------------------------------------------------------------------------
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "fontdlg.h"
 #endif
 
@@ -93,4 +93,4 @@
     return wxID_CANCEL;
 }
 
-#endif // 10.2+
\ No newline at end of file
+#endif // 10.2+
diff -ur ../wxWidgets/src/mac/carbon/fontenum.cpp ./src/mac/carbon/fontenum.cpp
--- ../wxWidgets/src/mac/carbon/fontenum.cpp	Sun May 23 13:52:31 2004
+++ ./src/mac/carbon/fontenum.cpp	Tue Nov 30 18:00:54 2004
@@ -17,7 +17,7 @@
 // headers
 // ----------------------------------------------------------------------------
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
     #pragma implementation "fontenum.h"
 #endif
 
diff -ur ../wxWidgets/src/mac/carbon/fontutil.cpp ./src/mac/carbon/fontutil.cpp
--- ../wxWidgets/src/mac/carbon/fontutil.cpp	Sun May 23 13:52:31 2004
+++ ./src/mac/carbon/fontutil.cpp	Tue Nov 30 18:01:01 2004
@@ -17,7 +17,7 @@
 // headers
 // ----------------------------------------------------------------------------
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
     #pragma implementation "fontutil.h"
 #endif
 
diff -ur ../wxWidgets/src/mac/carbon/glcanvas.cpp ./src/mac/carbon/glcanvas.cpp
--- ../wxWidgets/src/mac/carbon/glcanvas.cpp	Sun Jul 11 09:14:53 2004
+++ ./src/mac/carbon/glcanvas.cpp	Tue Nov 30 18:01:16 2004
@@ -9,7 +9,7 @@
 // Licence:       wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "glcanvas.h"
 #endif
 
@@ -18,8 +18,6 @@
 #if defined(__BORLANDC__)
 #pragma hdrstop
 #endif
-
-#include "wx/setup.h"
 
 #if wxUSE_GLCANVAS
 
diff -ur ../wxWidgets/src/mac/carbon/hid.cpp ./src/mac/carbon/hid.cpp
--- ../wxWidgets/src/mac/carbon/hid.cpp	Tue Sep 28 11:36:46 2004
+++ ./src/mac/carbon/hid.cpp	Tue Nov 30 18:01:34 2004
@@ -17,7 +17,7 @@
 // headers
 // ---------------------------------------------------------------------------
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "hid.h"
 #endif
 
@@ -27,11 +27,6 @@
 #ifdef __BORLANDC__
     #pragma hdrstop
 #endif
-
-#ifndef WX_PRECOMP
-    #include "wx/defs.h"
-#endif
-
 
 //DARWIN _ONLY_
 #ifdef __DARWIN__
diff -ur ../wxWidgets/src/mac/carbon/metafile.cpp ./src/mac/carbon/metafile.cpp
--- ../wxWidgets/src/mac/carbon/metafile.cpp	Sun May 23 13:52:31 2004
+++ ./src/mac/carbon/metafile.cpp	Tue Nov 30 18:02:00 2004
@@ -9,7 +9,7 @@
 // Licence:       wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "metafile.h"
 #endif
 
@@ -18,10 +18,6 @@
 
 #ifdef __BORLANDC__
 #pragma hdrstop
-#endif
-
-#ifndef WX_PRECOMP
-#include "wx/setup.h"
 #endif
 
 #if wxUSE_METAFILE
diff -ur ../wxWidgets/src/mac/carbon/mimetmac.cpp ./src/mac/carbon/mimetmac.cpp
--- ../wxWidgets/src/mac/carbon/mimetmac.cpp	Sun May 23 13:52:31 2004
+++ ./src/mac/carbon/mimetmac.cpp	Tue Nov 30 18:02:15 2004
@@ -9,7 +9,7 @@
 // Licence:     wxWindows licence (part of wxExtra library)
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef    __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "mimetype.h"
 #endif
 
@@ -18,10 +18,6 @@
 
 #ifdef __BORLANDC__
   #pragma hdrstop
-#endif
-
-#ifndef WX_PRECOMP
-  #include "wx/defs.h"
 #endif
 
 #ifndef WX_PRECOMP
diff -ur ../wxWidgets/src/mac/carbon/mpthread.cpp ./src/mac/carbon/mpthread.cpp
--- ../wxWidgets/src/mac/carbon/mpthread.cpp	Sun May 23 13:52:31 2004
+++ ./src/mac/carbon/mpthread.cpp	Tue Nov 30 18:02:26 2004
@@ -10,7 +10,7 @@
 // Licence:	wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "thread.h"
 #endif
 
diff -ur ../wxWidgets/src/mac/carbon/pnghand.cpp ./src/mac/carbon/pnghand.cpp
--- ../wxWidgets/src/mac/carbon/pnghand.cpp	Fri Nov  5 10:23:42 2004
+++ ./src/mac/carbon/pnghand.cpp	Tue Nov 30 18:02:38 2004
@@ -9,7 +9,7 @@
 // Licence:     wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #  pragma implementation "pngread.h"
 #  pragma implementation "pnghand.h"
 #endif
diff -ur ../wxWidgets/src/mac/carbon/printmac.cpp ./src/mac/carbon/printmac.cpp
--- ../wxWidgets/src/mac/carbon/printmac.cpp	Wed Nov 24 14:50:12 2004
+++ ./src/mac/carbon/printmac.cpp	Tue Nov 30 18:02:50 2004
@@ -9,7 +9,7 @@
 // Licence:       wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "printwin.h"
 #endif
 
@@ -19,8 +19,6 @@
 #ifdef __BORLANDC__
 #pragma hdrstop
 #endif
-
-#include "wx/defs.h"
 
 #ifndef WX_PRECOMP
 #include "wx/utils.h"
diff -ur ../wxWidgets/src/mac/carbon/statline.cpp ./src/mac/carbon/statline.cpp
--- ../wxWidgets/src/mac/carbon/statline.cpp	Sun May 23 13:52:33 2004
+++ ./src/mac/carbon/statline.cpp	Tue Nov 30 18:03:05 2004
@@ -16,7 +16,7 @@
 // headers
 // ----------------------------------------------------------------------------
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
     #pragma implementation "statline.h"
 #endif
 
diff -ur ../wxWidgets/src/mac/carbon/statlmac.cpp ./src/mac/carbon/statlmac.cpp
--- ../wxWidgets/src/mac/carbon/statlmac.cpp	Sun Jul 11 09:18:21 2004
+++ ./src/mac/carbon/statlmac.cpp	Tue Nov 30 18:03:13 2004
@@ -16,7 +16,7 @@
 // headers
 // ----------------------------------------------------------------------------
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
     #pragma implementation "statline.h"
 #endif
 
diff -ur ../wxWidgets/src/mac/carbon/taskbar.cpp ./src/mac/carbon/taskbar.cpp
--- ../wxWidgets/src/mac/carbon/taskbar.cpp	Wed Oct 20 13:14:45 2004
+++ ./src/mac/carbon/taskbar.cpp	Tue Nov 30 18:03:24 2004
@@ -11,8 +11,6 @@
 
 #include "wx/wxprec.h"
 
-#include "wx/defs.h"
-
 #ifdef wxHAS_TASK_BAR_ICON
 
 #include "wx/mac/private.h"
diff -ur ../wxWidgets/src/mac/carbon/thread.cpp ./src/mac/carbon/thread.cpp
--- ../wxWidgets/src/mac/carbon/thread.cpp	Tue Jun  8 07:47:17 2004
+++ ./src/mac/carbon/thread.cpp	Tue Nov 30 18:03:38 2004
@@ -10,7 +10,7 @@
 // Licence:	wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
 #pragma implementation "thread.h"
 #endif
 
diff -ur ../wxWidgets/src/mac/carbon/toplevel.cpp ./src/mac/carbon/toplevel.cpp
--- ../wxWidgets/src/mac/carbon/toplevel.cpp	Wed Nov 24 14:50:24 2004
+++ ./src/mac/carbon/toplevel.cpp	Tue Nov 30 18:03:52 2004
@@ -17,7 +17,7 @@
 // headers
 // ----------------------------------------------------------------------------
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(NO_GCC_PRAGMA)
     #pragma implementation "toplevel.h"
 #endif
 
diff -ur ../wxWidgets/src/mac/carbon/utilsexc.cpp ./src/mac/carbon/utilsexc.cpp
--- ../wxWidgets/src/mac/carbon/utilsexc.cpp	Fri Nov  5 10:23:43 2004
+++ ./src/mac/carbon/utilsexc.cpp	Tue Nov 30 18:04:06 2004
@@ -9,10 +9,6 @@
 // Licence:       wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
-#ifdef __GNUG__
-//#pragma implementation
-#endif
-
 #include "wx/wxprec.h"
 #ifndef WX_PRECOMP
     #include "wx/log.h"

 	  	 
