? wxWindows/wxgtkd-2.5-config
? wxWindows/version-script
? wxWindows/lib/wx
Index: wxWindows/src/generic/listctrl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/generic/listctrl.cpp,v
retrieving revision 1.278
diff -b -u -2 -r1.278 listctrl.cpp
--- wxWindows/src/generic/listctrl.cpp	2003/04/08 22:21:43	1.278
+++ wxWindows/src/generic/listctrl.cpp	2003/04/23 11:55:05
@@ -3204,4 +3204,5 @@
         wxListEvent le( command, GetParent()->GetId() );
         le.SetEventObject( GetParent() );
+	le.m_itemIndex = current;
         le.m_pointDrag = m_dragStart;
         GetParent()->GetEventHandler()->ProcessEvent( le );

 	  	 
