diff --git a/src/common/imagbmp.cpp b/src/common/imagbmp.cpp
index ce8a1336585753cb7720c3bb80a2e2b84167075b..efaf8963163a7c9e3917219bc0f0c94164c9ad84 100644
--- a/src/common/imagbmp.cpp
+++ b/src/common/imagbmp.cpp
@@ -1171,6 +1171,12 @@ bool wxBMPHandler::LoadDib(wxImage *image, wxInputStream& stream,
         return false;
     }
 
+    if( !usesV1 )
+    {
+      // forward stream position to after the bitmap header
+      stream.SeekI(hdrSize - sizeof(BITMAPINFOHEADER), wxFromCurrent);
+    }
+
     //read DIB; this is the BMP image or the XOR part of an icon image
     if ( !DoLoadDib(image, width, height, bpp, ncolors, comp, offset, stream,
                     verbose, IsBmp, true,
