index 2be681b..879da06 100644
--- a/src/osx/cocoa/settings.mm
+++ b/src/osx/cocoa/settings.mm
@@ -45,7 +45,6 @@ wxColour wxSystemSettingsNative::GetColour(wxSystemColour index)
         // fall through, window background is reasonable
     case wxSYS_COLOUR_MENU:
     case wxSYS_COLOUR_MENUBAR:
-    case wxSYS_COLOUR_WINDOW:
     case wxSYS_COLOUR_WINDOWFRAME:
     case wxSYS_COLOUR_ACTIVEBORDER:
     case wxSYS_COLOUR_INACTIVEBORDER:
@@ -53,6 +52,9 @@ wxColour wxSystemSettingsNative::GetColour(wxSystemColour index)
     case wxSYS_COLOUR_GRADIENTINACTIVECAPTION:
         sysColor = [NSColor windowFrameColor];
         break;
+    case wxSYS_COLOUR_WINDOW:
+        sysColor = [NSColor textBackgroundColor];
+        break;
     case wxSYS_COLOUR_BTNFACE:
         sysColor = [NSColor controlColor];
         break;
