Index: src/common/image.cpp
===================================================================
--- src/common/image.cpp	(revision 75741)
+++ src/common/image.cpp	(working copy)
@@ -117,10 +117,16 @@
 
 wxImageRefData::~wxImageRefData()
 {
-    if ( !m_static )
+    if ( !m_static && m_data )
+    {
         free( m_data );
-    if ( !m_staticAlpha )
+        m_data = NULL;
+    }
+    if ( !m_staticAlpha && m_alpha )
+    {
         free( m_alpha );
+        m_alpha = NULL;
+    }
 }
 
 
@@ -1986,8 +1992,11 @@
         alpha = (unsigned char *)malloc(M_IMGDATA->m_width*M_IMGDATA->m_height);
     }
 
-    if( !M_IMGDATA->m_staticAlpha )
+    if( !M_IMGDATA->m_staticAlpha && M_IMGDATA->m_alpha )
+    {
         free(M_IMGDATA->m_alpha);
+        M_IMGDATA->m_alpha = NULL;
+    }
 
     M_IMGDATA->m_alpha = alpha;
     M_IMGDATA->m_staticAlpha = static_data;
Index: src/common/strconv.cpp
===================================================================
--- src/common/strconv.cpp	(revision 75741)
+++ src/common/strconv.cpp	(working copy)
@@ -3076,7 +3076,10 @@
 
 void wxCSConv::Clear()
 {
-    free(m_name);
+    if ( m_name )
+    {
+        free(m_name);
+    }
     m_name = NULL;
 
     wxDELETE(m_convReal);
Index: src/common/stream.cpp
===================================================================
--- src/common/stream.cpp	(revision 75741)
+++ src/common/stream.cpp	(working copy)
@@ -114,7 +114,7 @@
 
 void wxStreamBuffer::FreeBuffer()
 {
-    if ( m_destroybuf )
+    if ( m_destroybuf && m_buffer_start )
     {
         free(m_buffer_start);
         m_buffer_start = NULL;
@@ -725,7 +725,10 @@
 
 wxInputStream::~wxInputStream()
 {
-    free(m_wback);
+    if ( m_wback )
+    {
+        free(m_wback);
+    }
 }
 
 bool wxInputStream::CanRead() const
