Index: valtext.h
===================================================================
--- valtext.h	(revision 58675)
+++ valtext.h	(working copy)
@@ -121,6 +121,21 @@
     long GetStyle() const;
 
     /**
+        Check if the given string matches with the values present into
+        the include list (the list of valid values).
+        @param val
+            The wxString value to compare with the include list.
+    */
+    bool IsInCharIncludes(const wxString& val);
+    /**
+        Check if the given string does not match with the values present
+        into the exclude list (the list of invalid values).
+        @param val
+            The wxString value to compare with the include list.
+    */
+    bool IsNotInCharExcludes(const wxString& val);
+
+    /**
         Receives character input from the window and filters it according to
         the current validator style.
     */
