Index: wxWidgets/src/msw/window.cpp
===================================================================
--- wxWidgets/src/msw/window.cpp	(revision 57064)
+++ wxWidgets/src/msw/window.cpp	(working copy)
@@ -4088,7 +4088,6 @@
         ::DragQueryFile(hFilesInfo, wIndex,
                         wxStringBuffer(files[wIndex], len), len);
     }
-    DragFinish (hFilesInfo);
 
     wxDropFilesEvent event(wxEVT_DROP_FILES, gwFilesDropped, files);
     event.SetEventObject(this);
@@ -4098,6 +4097,8 @@
     event.m_pos.x = dropPoint.x;
     event.m_pos.y = dropPoint.y;
 
+    DragFinish (hFilesInfo);
+
     return HandleWindowEvent(event);
 #endif
 }
