Index: include/wx/filepicker.h
===================================================================
--- include/wx/filepicker.h	(revision 51470)
+++ include/wx/filepicker.h	(working copy)
@@ -144,8 +144,7 @@
 
 protected:
     // creates the picker control
-    virtual
-    wxFileDirPickerWidgetBase *CreatePicker(wxWindow *parent,
+    virtual wxFileDirPickerWidgetBase *CreatePicker(wxWindow *parent,
                                             const wxString& path,
                                             const wxString& message,
                                             const wxString& wildcard) = 0;
@@ -237,17 +236,10 @@
         { return wxEVT_COMMAND_FILEPICKER_CHANGED; }
 
 protected:
-    wxFileDirPickerWidgetBase *CreatePicker(wxWindow *parent,
+    virtual wxFileDirPickerWidgetBase *CreatePicker(wxWindow *parent,
                                             const wxString& path,
                                             const wxString& message,
-                                            const wxString& wildcard)
-    {
-        return new wxFilePickerWidget(parent, wxID_ANY,
-                                      wxGetTranslation(wxFilePickerWidgetLabel),
-                                      path, message, wildcard,
-                                      wxDefaultPosition, wxDefaultSize,
-                                      GetPickerStyle(GetWindowStyle()));
-    }
+                                            const wxString& wildcard);
 
     // extracts the style for our picker from wxFileDirPickerCtrlBase's style
     long GetPickerStyle(long style) const
@@ -332,17 +324,10 @@
         { return wxEVT_COMMAND_DIRPICKER_CHANGED; }
 
 protected:
-    wxFileDirPickerWidgetBase *CreatePicker(wxWindow *parent,
+    virtual wxFileDirPickerWidgetBase* CreatePicker(wxWindow *parent,
                                             const wxString& path,
                                             const wxString& message,
-                                            const wxString& WXUNUSED(wildcard))
-    {
-        return new wxDirPickerWidget(parent, wxID_ANY,
-                                     wxGetTranslation(wxDirPickerWidgetLabel),
-                                     path, message,
-                                     wxDefaultPosition, wxDefaultSize,
-                                     GetPickerStyle(GetWindowStyle()));
-    }
+                                            const wxString& wildcard);
 
     // extracts the style for our picker from wxFileDirPickerCtrlBase's style
     long GetPickerStyle(long style) const
Index: src/common/filepickercmn.cpp
===================================================================
--- src/common/filepickercmn.cpp	(revision 51470)
+++ src/common/filepickercmn.cpp	(working copy)
@@ -200,6 +200,18 @@
     return wxFileName(m_text->GetValue()).GetFullPath();
 }
 
+wxFileDirPickerWidgetBase* wxFilePickerCtrl::CreatePicker(wxWindow *parent,
+                                                          const wxString& path,
+                                                          const wxString& message,
+                                                          const wxString& wildcard)
+{
+    return new wxFilePickerWidget(parent, wxID_ANY,
+                                  wxGetTranslation(wxFilePickerWidgetLabel),
+                                  path, message, wildcard,
+                                  wxDefaultPosition, wxDefaultSize,
+                                  GetPickerStyle(GetWindowStyle()));
+}
+
 #endif // wxUSE_FILEPICKERCTRL
 
 // ----------------------------------------------------------------------------
@@ -221,4 +233,16 @@
     return wxFileName::DirName(m_text->GetValue()).GetPath();
 }
 
+wxFileDirPickerWidgetBase* wxDirPickerCtrl::CreatePicker(wxWindow *parent,
+                                                         const wxString& path,
+                                                         const wxString& message,
+                                                         const wxString& WXUNUSED(wildcard))
+{
+    return new wxDirPickerWidget(parent, wxID_ANY,
+                                 wxGetTranslation(wxDirPickerWidgetLabel),
+                                 path, message,
+                                 wxDefaultPosition, wxDefaultSize,
+                                 GetPickerStyle(GetWindowStyle()));
+}
+
 #endif // wxUSE_DIRPICKERCTRL

 	  	 
