Index: src/msw/notebook.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/notebook.cpp,v
retrieving revision 1.149
diff -u -r1.149 notebook.cpp
--- src/msw/notebook.cpp	2005/05/14 16:57:51	1.149
+++ src/msw/notebook.cpp	2005/05/25 10:05:15
@@ -892,24 +892,10 @@
       {
         wxNotebookPage *pPage = m_pages[sel];
         pPage->Show(true);
-        pPage->SetFocus();
-
-        // If the newly focused window is not a child of the new page,
-        // SetFocus was not successful and the notebook itself should be
-        // focused
-        wxWindow *currentFocus = FindFocus();
-        wxWindow *startFocus = currentFocus;
-        while ( currentFocus && currentFocus != pPage && currentFocus != this )
-            currentFocus = currentFocus->GetParent();
-
-        if ( startFocus == pPage || currentFocus != pPage )
-            SetFocus();
-
       }
-      else // no pages in the notebook, give the focus to itself
-      {
-          SetFocus();
-      }
+
+      //always give ourself focus
+      SetFocus();
 
       m_nSelection = sel;
   }

 	  	 
