Index: samples/docview/docview.cpp
===================================================================
--- samples/docview/docview.cpp	(revision 70398)
+++ samples/docview/docview.cpp	(working copy)
@@ -296,7 +296,7 @@
     frame->SetMenuBar(menubar);
 }
 
-wxFrame *MyApp::CreateChildFrame(wxView *view, bool isCanvas)
+wxFrame *MyApp::CreateChildFrame(wxView *view, bool isCanvas, bool supportsPrinting)
 {
     // create a child frame of appropriate class for the current mode
     wxFrame *subframe;
@@ -336,7 +336,7 @@
 
     menuFile->Append(wxID_NEW);
     menuFile->Append(wxID_OPEN);
-    AppendDocumentFileCommands(menuFile, isCanvas);
+    AppendDocumentFileCommands(menuFile, supportsPrinting);
     menuFile->AppendSeparator();
     menuFile->Append(wxID_EXIT);
 
Index: samples/docview/docview.h
===================================================================
--- samples/docview/docview.h	(revision 70398)
+++ samples/docview/docview.h	(working copy)
@@ -44,7 +44,7 @@
 
     // our specific methods
     Mode GetMode() const { return m_mode; }
-    wxFrame *CreateChildFrame(wxView *view, bool isCanvas);
+    wxFrame *CreateChildFrame(wxView *view, bool isCanvas, bool supportsPrinting);
 
     // these accessors should only be called in single document mode, otherwise
     // the pointers are NULL and an assert is triggered
Index: samples/docview/view.cpp
===================================================================
--- samples/docview/view.cpp	(revision 70398)
+++ samples/docview/view.cpp	(working copy)
@@ -50,7 +50,7 @@
     if ( app.GetMode() != MyApp::Mode_Single )
     {
         // create a new window and canvas inside it
-        wxFrame* frame = app.CreateChildFrame(this, true);
+        wxFrame* frame = app.CreateChildFrame(this, true, true);
         wxASSERT(frame == GetFrame());
         m_canvas = new MyCanvas(this);
         frame->Show();
@@ -158,7 +158,7 @@
     if ( !wxView::OnCreate(doc, flags) )
         return false;
 
-    wxFrame* frame = wxGetApp().CreateChildFrame(this, false);
+    wxFrame* frame = wxGetApp().CreateChildFrame(this, false, false);
     wxASSERT(frame == GetFrame());
     m_text = new wxTextCtrl(frame, wxID_ANY, "",
                             wxDefaultPosition, wxDefaultSize,
@@ -313,7 +313,7 @@
     if ( !wxView::OnCreate(doc, flags) )
         return false;
 
-    wxFrame* frame = wxGetApp().CreateChildFrame(this, false);
+    wxFrame* frame = wxGetApp().CreateChildFrame(this, false, true);
     wxASSERT(frame == GetFrame());
     m_canvas = new ImageCanvas(this);
     frame->Show();
@@ -371,7 +371,7 @@
 {
     SetDocument(doc);
 
-    m_frame = wxGetApp().CreateChildFrame(this, false);
+    m_frame = wxGetApp().CreateChildFrame(this, false, false);
     m_frame->SetTitle("Image Details");
 
     wxPanel * const panel = new wxPanel(m_frame);
