Index: src/generic/treectlg.cpp
===================================================================
--- src/generic/treectlg.cpp	(revision 58409)
+++ src/generic/treectlg.cpp	(working copy)
@@ -436,19 +436,20 @@
 #ifdef __WXMSW__
     rect.x -= 5;
     rect.width += 10;
-#elif __WXGTK__
+#elif defined(__WXGTK__)
     rect.x -= 5;
     rect.y -= 2;
     rect.width  += 8;
     rect.height += 4;
-#elif __WXMAC__
+#elif defined(__WXMAC__)
     wxSize bs = DoGetBestSize();
     // edit control height
     if ( rect.height > bs.y - 8 )
+    {
         int diff = rect.height - ( bs.y - 8 );
         rect.height -= diff;
         rect.y += diff / 2;
-     }
+    }
 #endif
 
     (void)Create(m_owner, wxID_ANY, m_startValue,
