Index: include/wx/event.h
===================================================================
--- include/wx/event.h	(revision 74912)
+++ include/wx/event.h	(working copy)
@@ -2275,19 +2275,35 @@
 class WXDLLIMPEXP_CORE wxActivateEvent : public wxEvent
 {
 public:
-    wxActivateEvent(wxEventType type = wxEVT_NULL, bool active = true, int Id = 0)
-        : wxEvent(Id, type)
-        { m_active = active; }
+    // Type of activation. For now we can only detect if it was by mouse or by some
+    // other method (wxMSW only).
+    enum Reason
+    {
+        Reason_Mouse,
+        Reason_Unknown
+    };
+
+    wxActivateEvent(wxEventType type = wxEVT_NULL, bool active = true, int Id = 0, Reason ActivationReason = Reason_Unknown)
+        : wxEvent(Id, type),
+        m_activationReason(ActivationReason)
+    {
+        m_active = active;
+    }
     wxActivateEvent(const wxActivateEvent& event)
         : wxEvent(event)
-    { m_active = event.m_active; }
+    {
+        m_active = event.m_active;
+        m_activationReason = event.m_activationReason;
+    }
 
     bool GetActive() const { return m_active; }
+    Reason GetActivationReason() const { return m_activationReason;}
 
     virtual wxEvent *Clone() const { return new wxActivateEvent(*this); }
 
 private:
     bool m_active;
+    Reason m_activationReason;
 
 private:
     DECLARE_DYNAMIC_CLASS_NO_ASSIGN(wxActivateEvent)
Index: interface/wx/event.h
===================================================================
--- interface/wx/event.h	(revision 74912)
+++ interface/wx/event.h	(working copy)
@@ -2977,15 +2977,37 @@
 {
 public:
     /**
+        Enum for specifying the activation reason, i.e. if the window was activated
+        by mouse click or by some other method.
+
+        @since 3.0
+    */
+    enum Reason
+    {
+        /// Window activated by mouse click.
+        Reason_Mouse,
+        /// Window was activated with some other method than mouse click.
+        Reason_Unknown
+    };
+
+    /**
         Constructor.
     */
     wxActivateEvent(wxEventType eventType = wxEVT_NULL, bool active = true,
-                    int id = 0);
+                    int id = 0, Reason ActivationReason = Reason_Unknown);
 
     /**
         Returns @true if the application or window is being activated, @false otherwise.
     */
     bool GetActive() const;
+
+    /**
+        Returns @c Reason_Mouse if window was activated with mouse click.
+        Otherwise returns @c Reason_Unknown if window was activated by some
+        other method as keyboard, calling activation function directly, etc.
+        (wxMSW only)
+    */
+    Reason GetActivationReason() const;
 };
 
 
Index: src/msw/window.cpp
===================================================================
--- src/msw/window.cpp	(revision 74912)
+++ src/msw/window.cpp	(working copy)
@@ -4049,7 +4049,8 @@
 {
     wxActivateEvent event(wxEVT_ACTIVATE,
                           (state == WA_ACTIVE) || (state == WA_CLICKACTIVE),
-                          m_windowId);
+                          m_windowId,
+                          state == WA_CLICKACTIVE ? wxActivateEvent::Reason_Mouse : wxActivateEvent::Reason_Unknown);
     event.SetEventObject(this);
 
     return HandleWindowEvent(event);
