Index: include/wx/event.h
===================================================================
--- include/wx/event.h	(revision 74574)
+++ include/wx/event.h	(working copy)
@@ -2276,19 +2276,27 @@
 class WXDLLIMPEXP_CORE wxActivateEvent : public wxEvent
 {
 public:
-    wxActivateEvent(wxEventType type = wxEVT_NULL, bool active = true, int Id = 0)
+    wxActivateEvent(wxEventType type = wxEVT_NULL, bool active = true, int Id = 0, bool MouseClick = false)
         : wxEvent(Id, type)
-        { m_active = active; }
+    {
+        m_active = active;
+        m_MouseClick = MouseClick;
+    }
     wxActivateEvent(const wxActivateEvent& event)
         : wxEvent(event)
-    { m_active = event.m_active; }
+    {
+        m_active = event.m_active;
+	    m_MouseClick = event.m_MouseClick;   
+    }
 
     bool GetActive() const { return m_active; }
+	bool GetActiveWithMouseClick() const { return m_MouseClick;}
 
     virtual wxEvent *Clone() const { return new wxActivateEvent(*this); }
 
 private:
     bool m_active;
+	bool m_MouseClick;
 
 private:
     DECLARE_DYNAMIC_CLASS_NO_ASSIGN(wxActivateEvent)
Index: interface/wx/event.h
===================================================================
--- interface/wx/event.h	(revision 74574)
+++ interface/wx/event.h	(working copy)
@@ -2989,6 +2989,11 @@
         Returns @true if the application or window is being activated, @false otherwise.
     */
     bool GetActive() const;
+
+	/**
+        Returns @true if the application or window is being activated by mouse click. @false otherwise.
+    */
+	bool GetActiveWithMouseClick() const;
 };
 
 
Index: src/msw/window.cpp
===================================================================
--- src/msw/window.cpp	(revision 74574)
+++ src/msw/window.cpp	(working copy)
@@ -4029,7 +4029,8 @@
 {
     wxActivateEvent event(wxEVT_ACTIVATE,
                           (state == WA_ACTIVE) || (state == WA_CLICKACTIVE),
-                          m_windowId);
+						  m_windowId,
+						  state == WA_CLICKACTIVE);
     event.SetEventObject(this);
 
     return HandleWindowEvent(event);
