--- wxWidgets-trunk/src/common/imagbmp.cpp	2014-03-11
+++ wxWidgets-work/src/common/imagbmp.cpp	2014-03-11
@@ -1188,8 +1188,8 @@
                             bool verbose)
 
 {
-    //sanity check; icon must be less than 127 pixels high and 255 wide
-    if ( image->GetHeight () > 127 )
+    //sanity check; icon must be no larger than 256x256
+    if ( image->GetHeight () > 256 )
     {
         if ( verbose )
         {
@@ -1197,7 +1197,7 @@
         }
         return false;
     }
-    if ( image->GetWidth () > 255 )
+    if ( image->GetWidth () > 256 )
     {
         if ( verbose )
         {
@@ -1284,7 +1284,28 @@
 
         //calculate size and offset of image and mask
         wxCountingOutputStream cStream;
-        bool bResult = SaveDib(image, cStream, verbose, IsBmp, IsMask);
+        bool bResult;
+#if wxUSE_LIBPNG
+        bool saveAsPNG = false;
+        // Typically, icons larger then 128x128 are saved as PNG images.
+        if( image->GetHeight () > 128 || image->GetWidth () > 128 )
+        {
+            wxPNGHandler handlerPNG;
+            bResult = handlerPNG.SaveFile(image, cStream, verbose);
+            if ( !bResult )
+            {
+                if ( verbose )
+                {
+                    wxLogError(_("ICO: Error writing the image file!"));
+                }
+                return false;
+            }
+            saveAsPNG = true;
+        }
+        if( !saveAsPNG )
+#endif
+        {
+            bResult = SaveDib(image, cStream, verbose, IsBmp, IsMask);
         if ( !bResult )
         {
             if ( verbose )
@@ -1304,6 +1325,7 @@
             }
             return false;
         }
+        }
         wxUint32 Size = cStream.GetSize();
 
         // wxCountingOutputStream::IsOk() always returns true for now and this
@@ -1321,8 +1343,8 @@
 
         offset = offset + sizeof(ICONDIRENTRY);
 
-        icondirentry.bWidth = (wxUint8)image->GetWidth();
-        icondirentry.bHeight = (wxUint8)(2 * image->GetHeight());
+        icondirentry.bWidth = (image->GetWidth() == 256)? 0: (wxUint8)image->GetWidth();
+        icondirentry.bHeight = (image->GetHeight() == 256)? 0: (wxUint8)image->GetHeight();
         icondirentry.bColorCount = 0;
         icondirentry.bReserved = 0;
         icondirentry.wPlanes = wxUINT16_SWAP_ON_BE(1);
@@ -1364,6 +1386,23 @@
         }
 
         // actually save it:
+#if wxUSE_LIBPNG
+        if( saveAsPNG )
+        {
+            wxPNGHandler handlerPNG;
+            bResult = handlerPNG.SaveFile(image, stream, verbose);
+            if ( !bResult )
+            {
+                if ( verbose )
+                {
+                    wxLogError(_("ICO: Error writing the image file!"));
+                }
+                return false;
+            }
+        }
+        else
+#endif
+        {
         IsMask = false;
         bResult = SaveDib(image, stream, verbose, IsBmp, IsMask);
         if ( !bResult )
@@ -1385,6 +1424,7 @@
             }
             return false;
         }
+        }
 
     } // end of for loop
 
