diff -uNrB wxWidgets-trunk\src/msw/icon.cpp wxWidgets-work\src/msw/icon.cpp
--- wxWidgets-trunk\src/msw/icon.cpp	2013-12-29
+++ wxWidgets-work\src/msw/icon.cpp	2014-03-10
@@ -123,6 +123,7 @@
     {
         SetHICON((WXHICON)hicon);
         SetSize(bmp.GetWidth(), bmp.GetHeight());
+        SetDepth((bmp.GetDepth() == 32 && !bmp.HasAlpha())? 24: bmp.GetDepth());
     }
 #endif // __WXMICROWIN__
 }
