 src/msw/dir.cpp | 35 +++++++++++++++++++++++++----------
 1 file changed, 25 insertions(+), 10 deletions(-)

diff --git a/src/msw/dir.cpp b/src/msw/dir.cpp
index 2eea8fe..9ab6ba3 100644
--- a/src/msw/dir.cpp
+++ b/src/msw/dir.cpp
@@ -64,20 +64,35 @@ inline void FreeFindData(FIND_DATA fd)
     }
 }
 
-inline FIND_DATA FindFirst(const wxString& spec,
-                           FIND_STRUCT *finddata)
+const wxChar *GetNameFromFindData(FIND_STRUCT *finddata)
 {
-    return ::FindFirstFile(spec.t_str(), finddata);
+    return finddata->cFileName;
 }
 
-inline bool FindNext(FIND_DATA fd, FIND_STRUCT *finddata)
+inline bool FindNext(FIND_DATA fd, const wxString& spec, const wxString& filter, FIND_STRUCT *finddata)
 {
-    return ::FindNextFile(fd, finddata) != 0;
+    if ( ! ::FindNextFile(fd, finddata) )
+        return false;
+    else if ( wxString(GetNameFromFindData(finddata)).Matches(filter) )
+        return true;
+    else
+        return FindNext(fd, spec, filter, finddata);
 }
 
-const wxChar *GetNameFromFindData(FIND_STRUCT *finddata)
+inline FIND_DATA FindFirst(const wxString& spec, const wxString& filter,
+                           FIND_STRUCT *finddata)
 {
-    return finddata->cFileName;
+    FIND_DATA fd = ::FindFirstFile(spec.t_str(), finddata);
+
+    // filter can be empty when called from wxGetDirectoryTimes()
+    if ( fd && !filter.empty() &&
+         !wxString(GetNameFromFindData(finddata)).Matches(filter) &&
+         !FindNext(fd, spec, filter, finddata) )
+    {
+        return INVALID_HANDLE_VALUE;
+    }
+
+    return fd;
 }
 
 inline FIND_ATTR GetAttrFromFindData(FIND_STRUCT *finddata)
@@ -196,7 +211,7 @@ bool wxDirData::Read(wxString *filename)
         else
             filespec += m_filespec;
 
-        m_finddata = FindFirst(filespec, PTR_TO_FINDDATA);
+        m_finddata = FindFirst(filespec, m_filespec, PTR_TO_FINDDATA);
 
         first = true;
     }
@@ -228,7 +243,7 @@ bool wxDirData::Read(wxString *filename)
         }
         else
         {
-            if ( !FindNext(m_finddata, PTR_TO_FINDDATA) )
+            if ( !FindNext(m_finddata, m_filespec, m_filespec, PTR_TO_FINDDATA) )
             {
 #ifdef __WIN32__
                 DWORD err = ::GetLastError();
@@ -400,7 +415,7 @@ wxGetDirectoryTimes(const wxString& dirname,
 #endif
 
     FIND_STRUCT fs;
-    FIND_DATA fd = FindFirst(dirname, &fs);
+    FIND_DATA fd = FindFirst(dirname, wxEmptyString, &fs);
     if ( !IsFindDataOk(fd) )
     {
         return false;
