Index: src/osx/cocoa/toolbar.mm
===================================================================
--- src/osx/cocoa/toolbar.mm	(revision 65218)
+++ src/osx/cocoa/toolbar.mm	(working copy)
@@ -506,7 +513,7 @@
         dc.SelectObject( m_alternateBitmap );
         dc.SetPen( wxPen(*wxBLACK) );
         dc.SetBrush( wxBrush( *wxLIGHT_GREY ));
-        dc.DrawRectangle( 0, 0, w, h );
+        dc.DrawRoundedRectangle(0, 0, w, h, 2);
         dc.DrawBitmap( m_bmpNormal, 0, 0, true );
         dc.SelectObject( wxNullBitmap );
 
