diff --git a/include/wx/filefn.h b/include/wx/filefn.h
index c75d309e36..9572aba302 100644
--- a/include/wx/filefn.h
+++ b/include/wx/filefn.h
@@ -229,6 +229,9 @@ enum wxPosixPermissions
     // complications
     #define   wxClose      wxPOSIX_IDENT(close)
 
+    // Windows uses 'closesocket', not 'close' to close a socket.
+    #define wxCloseSocket  closesocket
+
     #define wxRead         wxPOSIX_IDENT(read)
     #define wxWrite        wxPOSIX_IDENT(write)
 
@@ -384,6 +387,7 @@ enum wxPosixPermissions
     #endif
     // functions
     #define   wxClose      close
+    #define   wxCloseSocket wxClose
     #define   wxRead       ::read
     #define   wxWrite      ::write
     #define   wxLseek      lseek
diff --git a/src/common/socket.cpp b/src/common/socket.cpp
index 207b3c9751..eb971f952f 100644
--- a/src/common/socket.cpp
+++ b/src/common/socket.cpp
@@ -529,7 +529,7 @@ wxSocketImpl *wxSocketImpl::Accept(wxSocketBase& wxsocket)
     WX_SOCKLEN_T fromlen = sizeof(from);
     const wxSOCKET_T fd = accept(m_fd, &from.addr, &fromlen);
 
-    wxScopeGuard closeSocket = wxMakeGuard(wxClose, fd);
+    wxScopeGuard closeSocket = wxMakeGuard(wxCloseSocket, fd);
 
     // accepting is similar to reading in the sense that it resets "ready for
     // read" flag on the socket
