Index: src/common/txtstrm.cpp
===================================================================
--- src/common/txtstrm.cpp	(revision 76171)
+++ src/common/txtstrm.cpp	(working copy)
@@ -398,10 +398,8 @@
 
 void wxTextOutputStream::Write64(wxUint64 i)
 {
-    wxString str;
-    str.Printf(wxT("%" wxLongLongFmtSpec "u"), (unsigned long long)i);
-
-    WriteString(str);
+    WriteString(wxString::Format("%" wxLongLongFmtSpec "u",
+                                 (unsigned long long)i));
 }
 
 void wxTextOutputStream::Write32(wxUint32 i)
