Index: dcgraph.h
===================================================================
--- interface/wx/dcgraph.h	(revision 71197)
+++ interface/wx/dcgraph.h	(working copy)
@@ -36,7 +36,10 @@
     wxGCDC( const wxPrinterDC& printerDC );
 
     /**
-       Construct a wxGCDC from an existing grtaphics context.
+       Construct a wxGCDC from an existing graphics context.
+       @note After creating a wxGCDC from a wxGraphicsContext, all future
+             drawing operations should occur on the wxGCDC (and the
+             wxGraphicsContext should no more be directly drawn to)
     */
     wxGCDC(wxGraphicsContext* context);
 
