Index: dib.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/msw/dib.cpp,v
retrieving revision 1.66
diff -u -r1.66 dib.cpp
--- dib.cpp	2006/07/03 19:02:38	1.66
+++ dib.cpp	2006/11/15 14:48:14
@@ -775,14 +775,22 @@
             dst[1] = *src++;
             dst[0] = *src++;
 
-            dst += 3;
-
             if ( is32bit )
             {
                 if ( alpha )
-                    *alpha++ = *src;
+                {
+                    const unsigned char a = *alpha++ = *src;
+                    if (a>0)
+                    {
+                        dst[0] = (dst[0] * 255 - 127) / a;
+                        dst[1] = (dst[1] * 255 - 127) / a;
+                        dst[2] = (dst[2] * 255 - 127) / a;
+                    }
+                }
                 src++;
             }
+
+            dst += 3;
         }
 
         // pass to the previous line in the image


 	  	 
