Index: listctrl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWidgets/src/generic/listctrl.cpp,v
retrieving revision 1.354
diff -b -u -2 -r1.354 listctrl.cpp
--- listctrl.cpp	2005/01/19 13:26:04	1.354
+++ listctrl.cpp	2005/01/30 15:35:43
@@ -770,5 +771,6 @@
     // for double click logic
     size_t m_lineLastClicked,
-           m_lineBeforeLastClicked;
+           m_lineBeforeLastClicked,
+           m_lineSelectSingleOnUp;
 
 protected:
@@ -2177,4 +2179,5 @@
     m_current =
     m_lineLastClicked =
+	m_lineSelectSingleOnUp =
     m_lineBeforeLastClicked = (size_t)-1;
 
@@ -2979,6 +2982,14 @@
     }
 
-    if (event.LeftUp() && m_lastOnSame)
+    if (event.LeftUp())
     {
+        if(m_lineSelectSingleOnUp != (size_t) -1)
+        {
+            // select single line
+            HighlightAll( false );
+            ReverseHighlight(m_lineSelectSingleOnUp);
+        }
+        else if (m_lastOnSame)
+        {
         if ((current == m_current) &&
             (hitResult == wxLIST_HITTEST_ONITEMLABEL) &&
@@ -2987,16 +2998,20 @@
             m_renameTimer->Start( 100, true );
         }
+        }
         m_lastOnSame = false;
+        m_lineSelectSingleOnUp = (size_t) -1;
     }
-    else if (event.RightDown())
+    else
     {
-        SendNotify( current, wxEVT_COMMAND_LIST_ITEM_RIGHT_CLICK,
-                    event.GetPosition() );
+        m_lineSelectSingleOnUp = (size_t) -1; // this is neccessary , because after 
+        //a dnd operation in from and to ourself, the up event is swallowed by the dnd code.
+        // so on next non-up event( now!) m_lineSelectSingleOnUp should be reset
     }
-    else if (event.MiddleDown())
+
+    if (event.MiddleDown())
     {
         SendNotify( current, wxEVT_COMMAND_LIST_ITEM_MIDDLE_CLICK );
     }
-    else if ( event.LeftDown() || forceClick )
+    else if ( event.LeftDown() || event.RightDown() || forceClick )
     {
         m_lineBeforeLastClicked = m_lineLastClicked;
@@ -3005,5 +3020,7 @@
         size_t oldCurrent = m_current;
         bool cmdModifierDown = event.CmdDown();
-        if ( IsSingleSel() || !(cmdModifierDown || event.ShiftDown()) )
+        if ( !(cmdModifierDown || event.ShiftDown()) )
+        {
+            if( IsSingleSel() || !IsHighlighted(current) )
         {
             HighlightAll( false );
@@ -3013,4 +3030,10 @@
             ReverseHighlight(m_current);
         }
+			else // multi sel & current is highlighted & no mod keys
+			{
+				m_lineSelectSingleOnUp = current;	
+				ChangeCurrent(current); // change focus
+			}
+        }
         else // multi sel & either ctrl or shift is down
         {
@@ -3051,4 +3074,10 @@
         m_lastOnSame = !forceClick && (m_current == oldCurrent);
     }
+    if (event.RightDown())
+    {
+        SendNotify( current, wxEVT_COMMAND_LIST_ITEM_RIGHT_CLICK,
+                    event.GetPosition() );
+    }
+
 }
 

 	  	 
