Index: listctrl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWidgets/src/generic/listctrl.cpp,v
retrieving revision 1.356
diff -b -u -2 -r1.356 listctrl.cpp
--- listctrl.cpp	2005/01/29 23:52:55	1.356
+++ listctrl.cpp	2005/01/30 20:21:36
@@ -770,5 +771,6 @@
     // for double click logic
     size_t m_lineLastClicked,
-           m_lineBeforeLastClicked;
+           m_lineBeforeLastClicked,
+           m_lineSelectSingleOnUp;
 
 protected:
@@ -2176,4 +2178,5 @@
     m_current =
     m_lineLastClicked =
+    m_lineSelectSingleOnUp =
     m_lineBeforeLastClicked = (size_t)-1;
 
@@ -2939,5 +2942,5 @@
         wxListEvent le( command, GetParent()->GetId() );
         le.SetEventObject( GetParent() );
-        le.m_itemIndex = current;
+        le.m_itemIndex = m_lineLastClicked;
         le.m_pointDrag = m_dragStart;
         GetParent()->GetEventHandler()->ProcessEvent( le );
@@ -2976,6 +2979,14 @@
     }
 
-    if (event.LeftUp() && m_lastOnSame)
+    if (event.LeftUp())
     {
+        if(m_lineSelectSingleOnUp != (size_t) -1)
+        {
+            // select single line
+            HighlightAll( false );
+            ReverseHighlight(m_lineSelectSingleOnUp);
+        }
+        else if (m_lastOnSame)
+        {
         if ((current == m_current) &&
             (hitResult == wxLIST_HITTEST_ONITEMLABEL) &&
@@ -2984,8 +2995,18 @@
             m_renameTimer->Start( 100, true );
         }
+        }
         m_lastOnSame = false;
+        m_lineSelectSingleOnUp = (size_t) -1;
     }
-    else if (event.RightDown())
+    else
+    {
+        m_lineSelectSingleOnUp = (size_t) -1; // this is neccessary , because after 
+        //a dnd operation in from and to ourself, the up event is swallowed by the dnd code.
+        // so on next non-up event( now!) m_lineSelectSingleOnUp should be reset
+    }
+    if (event.RightDown())
     {
+        m_lineBeforeLastClicked = m_lineLastClicked;
+        m_lineLastClicked = current;
         // If the item is already selected, do not update the selection.
         // Multi-selections should not be cleared if a selected item is clicked.
@@ -2996,5 +3017,4 @@
             ReverseHighlight(m_current);
         }
-
         SendNotify( current, wxEVT_COMMAND_LIST_ITEM_RIGHT_CLICK,
                     event.GetPosition() );
@@ -3011,5 +3031,7 @@
         size_t oldCurrent = m_current;
         bool cmdModifierDown = event.CmdDown();
-        if ( IsSingleSel() || !(cmdModifierDown || event.ShiftDown()) )
+        if ( !(cmdModifierDown || event.ShiftDown()) )
+        {
+            if( IsSingleSel() || !IsHighlighted(current) )
         {
             HighlightAll( false );
@@ -3019,4 +3041,10 @@
             ReverseHighlight(m_current);
         }
+	     else // multi sel & current is highlighted & no mod keys
+            {
+                m_lineSelectSingleOnUp = current;	
+                ChangeCurrent(current); // change focus
+            }
+        }
         else // multi sel & either ctrl or shift is down
         {

 	  	 
