Index: src/osx/cocoa/dataview.mm
===================================================================
--- src/osx/cocoa/dataview.mm	(revision 78024)
+++ src/osx/cocoa/dataview.mm	(working copy)
@@ -1634,6 +1634,29 @@
     dvc->GetEventHandler()->ProcessEvent(event);
 }
 
+// Default enter key behaviour is to begin cell editing. Subclass keyDown to 
+// provide a keyboard wxEVT_DATAVIEW_ITEM_ACTIVATED event and allow the NSEvent
+// to pass if the wxEvent is not processed.
+- (void)keyDown:(NSEvent *)event
+{
+    if( [[event charactersIgnoringModifiers]
+         characterAtIndex: 0] == NSCarriageReturnCharacter )
+    {
+        wxDataViewCtrl* const dvc = implementation->GetDataViewCtrl();
+        
+        wxDataViewEvent eventDV( wxEVT_DATAVIEW_ITEM_ACTIVATED, dvc->GetId() );
+        eventDV.SetEventObject(dvc);
+        eventDV.SetItem( wxDataViewItem( [[self itemAtRow:[self selectedRow]] pointer]) );
+        eventDV.SetModel( dvc->GetModel() );
+        
+        if ( !dvc->GetEventHandler()->ProcessEvent(eventDV) )
+            [super keyDown:event];
+    }
+    else
+    {
+        [super keyDown:event];  // all other keys
+    }
+}
 
 //
 // contextual menus
