Index: treectrl.cpp
===================================================================
--- treectrl.cpp	(revision 54708)
+++ treectrl.cpp	(working copy)
@@ -2862,7 +2862,7 @@
             // the return code of this event handler as the return value for
             // NM_DBLCLK - otherwise, double clicking the item to toggle its
             // expanded status would never work
-            *result = false;
+            *result = !event.IsAllowed();
             break;
 
         case NM_RCLICK:
