diff -r -u -p a/src/msw/window.cpp b/src/msw/window.cpp
--- a/src/msw/window.cpp	2021-04-13 00:23:58.000000000 +0200
+++ b/src/msw/window.cpp	2021-07-05 12:29:14.012159300 +0200
@@ -663,7 +663,18 @@ bool wxWindowMSW::Show(bool show)
     // should work without errors
     if ( hWnd )
     {
-        ::ShowWindow(hWnd, show ? SW_SHOW : SW_HIDE);
+        BOOL ret = ::ShowWindow(hWnd, show ? SW_SHOW : SW_HIDE);
+
+        // Windows does not generate its WM_SHOW notification for frozen items.
+        // In such a case we have to generate the wxEVT_SHOW event ourselves
+        // to be consistent with other platforms.
+        bool changed = (ret != 0) != show;
+        if ( !changed && IsFrozen() )
+        {
+            wxShowEvent eventShow(GetId(), show);
+            eventShow.SetEventObject(this);
+            HandleWindowEvent(eventShow);
+        }
     }
 
     if ( IsFrozen() )
