diff -ur --exclude=CVS ../wxWidgets/include/wx/event.h ./include/wx/event.h
--- ../wxWidgets/include/wx/event.h	2005-05-31 19:43:19.000000000 -0700
+++ ./include/wx/event.h	2005-06-06 20:18:14.000000000 -0700
@@ -2236,7 +2236,6 @@
 // ----------------------------------------------------------------------------
 
 WX_DEFINE_ARRAY_PTR(const wxEventTableEntry*, wxEventTableEntryPointerArray);
-class WXDLLIMPEXP_BASE wxEvtHandler;
 
 class WXDLLIMPEXP_BASE wxEventHashTable
 {
diff -ur --exclude=CVS ../wxWidgets/include/wx/list.h ./include/wx/list.h
--- ../wxWidgets/include/wx/list.h	2005-04-07 18:32:38.000000000 -0700
+++ ./include/wx/list.h	2005-06-06 20:18:14.000000000 -0700
@@ -328,8 +328,6 @@
 extern WXDLLIMPEXP_BASE wxChar* copystring(const wxChar *s);
 #endif
 
-class WXDLLEXPORT wxObjectListNode;
-
 // undef it to get rid of old, deprecated functions
 #define wxLIST_COMPATIBILITY
 
diff -ur --exclude=CVS ../wxWidgets/include/wx/string.h ./include/wx/string.h
--- ../wxWidgets/include/wx/string.h	2005-04-10 10:04:56.000000000 -0700
+++ ./include/wx/string.h	2005-06-06 20:18:14.000000000 -0700
@@ -1463,12 +1463,6 @@
     { return (s2.Cmp((const char *)s1) != 0); }
 #endif // wxUSE_UNICODE/!wxUSE_UNICODE
 
-wxString WXDLLIMPEXP_BASE operator+(const wxString& string1,  const wxString& string2);
-wxString WXDLLIMPEXP_BASE operator+(const wxString& string, wxChar ch);
-wxString WXDLLIMPEXP_BASE operator+(wxChar ch, const wxString& string);
-wxString WXDLLIMPEXP_BASE operator+(const wxString& string, const wxChar *psz);
-wxString WXDLLIMPEXP_BASE operator+(const wxChar *psz, const wxString& string);
-
 #if wxUSE_UNICODE
 inline wxString operator+(const wxString& string, const wxWCharBuffer& buf)
     { return string + (const wchar_t *)buf; }
diff -ur --exclude=CVS ../wxWidgets/src/common/cmndata.cpp ./src/common/cmndata.cpp
--- ../wxWidgets/src/common/cmndata.cpp	2005-04-01 19:50:33.000000000 -0800
+++ ./src/common/cmndata.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -67,10 +67,6 @@
     IMPLEMENT_DYNAMIC_CLASS(wxFontData, wxObject)
     IMPLEMENT_DYNAMIC_CLASS(wxColourData, wxObject)
 
-#ifndef DMPAPER_USER
-     #define DMPAPER_USER                256
-#endif
-
 // ============================================================================
 // implementation
 // ============================================================================
diff -ur --exclude=CVS ../wxWidgets/src/common/dynarray.cpp ./src/common/dynarray.cpp
--- ../wxWidgets/src/common/dynarray.cpp	2005-02-04 16:57:45.000000000 -0800
+++ ./src/common/dynarray.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -29,10 +29,6 @@
 #include <stdlib.h>
 #include <string.h> // for memmove
 
-#ifndef max
-  #define max(a, b)   (((a) > (b)) ? (a) : (b))
-#endif
-
 // we cast the value to long from which we cast it to void * in IndexForInsert:
 // this can't work if the pointers are not big enough
 wxCOMPILE_TIME_ASSERT( sizeof(wxUIntPtr) <= sizeof(void *),
diff -ur --exclude=CVS ../wxWidgets/src/common/filefn.cpp ./src/common/filefn.cpp
--- ../wxWidgets/src/common/filefn.cpp	2005-05-31 19:43:58.000000000 -0700
+++ ./src/common/filefn.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -56,11 +56,6 @@
 
 #include "wx/log.h"
 
-// No, Cygwin doesn't appear to have fnmatch.h after all.
-#if defined(HAVE_FNMATCH_H)
-    #include "fnmatch.h"
-#endif
-
 #ifdef __WINDOWS__
     #include "wx/msw/private.h"
     #include "wx/msw/mslu.h"
diff -ur --exclude=CVS ../wxWidgets/src/common/fontmap.cpp ./src/common/fontmap.cpp
--- ../wxWidgets/src/common/fontmap.cpp	2005-03-26 20:27:18.000000000 -0800
+++ ./src/common/fontmap.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -443,7 +443,6 @@
             if ( dialog.ShowModal() == wxID_OK )
             {
                 wxFontData retData = dialog.GetFontData();
-                wxFont font = retData.GetChosenFont();
 
                 *info = retData.EncodingInfo();
                 info->encoding = retData.GetEncoding();
diff -ur --exclude=CVS ../wxWidgets/src/common/http.cpp ./src/common/http.cpp
--- ../wxWidgets/src/common/http.cpp	2004-10-20 13:14:28.000000000 -0700
+++ ./src/common/http.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -40,8 +40,6 @@
 IMPLEMENT_DYNAMIC_CLASS(wxHTTP, wxProtocol)
 IMPLEMENT_PROTOCOL(wxHTTP, wxT("http"), wxT("80"), true)
 
-#define HTTP_BSIZE 2048
-
 wxHTTP::wxHTTP()
   : wxProtocol()
 {
diff -ur --exclude=CVS ../wxWidgets/src/common/imagtiff.cpp ./src/common/imagtiff.cpp
--- ../wxWidgets/src/common/imagtiff.cpp	2005-03-01 13:36:52.000000000 -0800
+++ ./src/common/imagtiff.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -394,8 +394,6 @@
 
     TIFFSetField(tif, TIFFTAG_ROWSPERSTRIP,TIFFDefaultStripSize(tif, (uint32) -1));
 
-    uint8 bitmask;
-
     unsigned char *ptr = image->GetData();
     for ( int row = 0; row < image->GetHeight(); row++ )
     {
@@ -411,7 +409,6 @@
                 for ( int column = 0; column < linebytes; column++ )
                 {
                     uint8 reverse = 0;
-                    bitmask = 1;
                     for ( int bp = 0; bp < 8; bp++ )
                     {
                         if ( ptr[column*24 + bp*3] > 0 )
@@ -419,8 +416,6 @@
                             // check only red as this is sufficient
                             reverse = reverse | 128 >> bp;
                         }
-
-                        bitmask <<= 1;
                     }
 
                     buf[column] = reverse;
diff -ur --exclude=CVS ../wxWidgets/src/common/intl.cpp ./src/common/intl.cpp
--- ../wxWidgets/src/common/intl.cpp	2005-05-31 19:43:58.000000000 -0700
+++ ./src/common/intl.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -496,7 +496,6 @@
     ~wxPluralFormsCalculator() {}
 
     void  init(wxPluralFormsToken::Number nplurals, wxPluralFormsNode* plural);
-    wxString getString() const;
 
 private:
     wxPluralFormsToken::Number m_nplurals;
diff -ur --exclude=CVS ../wxWidgets/src/common/list.cpp ./src/common/list.cpp
--- ../wxWidgets/src/common/list.cpp	2005-03-26 20:27:19.000000000 -0800
+++ ./src/common/list.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -594,12 +594,6 @@
 // wxStringList
 // ----------------------------------------------------------------------------
 
-static inline wxChar* MYcopystring(const wxString& s)
-{
-    wxChar* copy = new wxChar[s.length() + 1];
-    return wxStrcpy(copy, s.c_str());
-}
-
 static inline wxChar* MYcopystring(const wxChar* s)
 {
     wxChar* copy = new wxChar[wxStrlen(s) + 1];
diff -ur --exclude=CVS ../wxWidgets/src/generic/dirdlgg.cpp ./src/generic/dirdlgg.cpp
--- ../wxWidgets/src/generic/dirdlgg.cpp	2005-05-31 19:43:58.000000000 -0700
+++ ./src/generic/dirdlgg.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -47,8 +47,6 @@
 
 static const int ID_DIRCTRL = 1000;
 static const int ID_TEXTCTRL = 1001;
-static const int ID_OK = 1002;
-static const int ID_CANCEL = 1003;
 static const int ID_NEW = 1004;
 static const int ID_SHOW_HIDDEN = 1005;
 static const int ID_GO_HOME = 1006;
diff -ur --exclude=CVS ../wxWidgets/src/generic/filedlgg.cpp ./src/generic/filedlgg.cpp
--- ../wxWidgets/src/generic/filedlgg.cpp	2005-05-21 09:50:29.000000000 -0700
+++ ./src/generic/filedlgg.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -143,10 +143,6 @@
 #define IsTopMostDir(dir)   (dir.empty())
 #endif
 
-#if defined(__DOS__) || defined(__WINDOWS__) || defined(__OS2__)
-// defined in src/generic/dirctrlg.cpp
-extern bool wxIsDriveAvailable(const wxString& dirName);
-#endif
 
 // defined in src/generic/dirctrlg.cpp
 extern size_t wxGetAvailableDrives(wxArrayString &paths, wxArrayString &names, wxArrayInt &icon_ids);
@@ -855,7 +851,6 @@
 #define  ID_CHOICE        (wxID_FILEDLGG + 8)
 #define  ID_TEXT          (wxID_FILEDLGG + 9)
 #define  ID_LIST_CTRL     (wxID_FILEDLGG + 10)
-#define  ID_ACTIVATED     (wxID_FILEDLGG + 11)
 #define  ID_CHECK         (wxID_FILEDLGG + 12)
 
 IMPLEMENT_DYNAMIC_CLASS(wxGenericFileDialog, wxFileDialogBase)
diff -ur --exclude=CVS ../wxWidgets/src/generic/fontdlgg.cpp ./src/generic/fontdlgg.cpp
--- ../wxWidgets/src/generic/fontdlgg.cpp	2005-04-02 21:12:11.000000000 -0800
+++ ./src/generic/fontdlgg.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -307,10 +307,6 @@
     {
         wxStaticText* itemStaticText15 = new wxStaticText( this, wxID_STATIC, _("C&olour:"), wxDefaultPosition, wxDefaultSize, 0 );
         itemBoxSizer14->Add(itemStaticText15, 0, wxALIGN_LEFT|wxLEFT|wxRIGHT|wxTOP|wxADJUST_MINSIZE, 5);
-        
-        wxSize colourSize = wxDefaultSize;
-        if (is_pda)
-            colourSize.x = 100;
 
         wxChoice* itemChoice16 = new wxChoice( this, wxID_FONT_COLOUR, wxDefaultPosition, wxDefaultSize, NUM_COLS, wxColourDialogNames, 0 );
         itemChoice16->SetHelpText(_("The font colour."));
diff -ur --exclude=CVS ../wxWidgets/src/generic/listctrl.cpp ./src/generic/listctrl.cpp
--- ../wxWidgets/src/generic/listctrl.cpp	2005-05-31 19:43:58.000000000 -0700
+++ ./src/generic/listctrl.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -2243,9 +2243,6 @@
                                        wxSOLID
                                     );
 
-    wxSize sz = size;
-    sz.y = 25;
-
     SetScrollbars( 0, 0, 0, 0, 0, 0 );
 
     wxVisualAttributes attr = wxGenericListCtrl::GetClassDefaultAttributes();
diff -ur --exclude=CVS ../wxWidgets/src/generic/numdlgg.cpp ./src/generic/numdlgg.cpp
--- ../wxWidgets/src/generic/numdlgg.cpp	2005-04-02 21:12:11.000000000 -0800
+++ ./src/generic/numdlgg.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -62,16 +62,6 @@
 // macros
 // ---------------------------------------------------------------------------
 
-/* Macro for avoiding #ifdefs when value have to be different depending on size of
-   device we display on - take it from something like wxDesktopPolicy in the future
- */
-
-#if defined(__SMARTPHONE__)
-    #define wxLARGESMALL(large,small) small
-#else
-    #define wxLARGESMALL(large,small) large
-#endif
-
 // ============================================================================
 // implementation
 // ============================================================================
diff -ur --exclude=CVS ../wxWidgets/src/generic/statusbr.cpp ./src/generic/statusbr.cpp
--- ../wxWidgets/src/generic/statusbr.cpp	2005-05-16 20:33:03.000000000 -0700
+++ ./src/generic/statusbr.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -56,7 +56,6 @@
 
 // Default status border dimensions
 #define         wxTHICK_LINE_BORDER 2
-#define         wxTHICK_LINE_WIDTH  1
 
 void wxStatusBarGeneric::Init()
 {
diff -ur --exclude=CVS ../wxWidgets/src/gtk/dcclient.cpp ./src/gtk/dcclient.cpp
--- ../wxWidgets/src/gtk/dcclient.cpp	2005-05-21 09:50:29.000000000 -0700
+++ ./src/gtk/dcclient.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -143,8 +143,6 @@
 // Implement Pool of Graphic contexts. Creating them takes too much time.
 //-----------------------------------------------------------------------------
 
-#define GC_POOL_SIZE 200
-
 enum wxPoolGCType
 {
    wxGC_ERROR = 0,
diff -ur --exclude=CVS ../wxWidgets/src/gtk/dialog.cpp ./src/gtk/dialog.cpp
--- ../wxWidgets/src/gtk/dialog.cpp	2004-05-25 05:44:56.000000000 -0700
+++ ./src/gtk/dialog.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -25,12 +25,6 @@
 
 #include "wx/gtk/win_gtk.h"
 
-//-----------------------------------------------------------------------------
-// idle system
-//-----------------------------------------------------------------------------
-
-extern void wxapp_install_idle_handler();
-extern bool g_isIdle;
 extern int g_openDialogs;
 
 //-----------------------------------------------------------------------------
diff -ur --exclude=CVS ../wxWidgets/src/gtk/filedlg.cpp ./src/gtk/filedlg.cpp
--- ../wxWidgets/src/gtk/filedlg.cpp	2005-05-16 20:33:03.000000000 -0700
+++ ./src/gtk/filedlg.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -39,7 +39,6 @@
 //-----------------------------------------------------------------------------
 
 extern void wxapp_install_idle_handler();
-extern bool g_isIdle;
 
 //-----------------------------------------------------------------------------
 // "clicked" for OK-button
diff -ur --exclude=CVS ../wxWidgets/src/gtk/frame.cpp ./src/gtk/frame.cpp
--- ../wxWidgets/src/gtk/frame.cpp	2005-03-26 20:28:01.000000000 -0800
+++ ./src/gtk/frame.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -58,7 +58,6 @@
 
 extern void wxapp_install_idle_handler();
 extern bool g_isIdle;
-extern int g_openDialogs;
 
 // ----------------------------------------------------------------------------
 // event tables
diff -ur --exclude=CVS ../wxWidgets/src/gtk/renderer.cpp ./src/gtk/renderer.cpp
--- ../wxWidgets/src/gtk/renderer.cpp	2005-04-16 10:08:21.000000000 -0700
+++ ./src/gtk/renderer.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -289,7 +289,9 @@
     const bool isVert = orient == wxVERTICAL;
 
     GdkRectangle rect;
+#if 0
     GdkRectangle erase_rect;
+#endif
     if ( isVert )
     {
         int h = win->GetClientSize().GetHeight();
@@ -298,11 +300,12 @@
         rect.y = 0;
         rect.width = full_size;
         rect.height = h;
-
+#if 0
         erase_rect.x = position;
         erase_rect.y = 0;
         erase_rect.width = full_size;
         erase_rect.height = h;
+#endif
     }
     else // horz
     {
@@ -312,11 +315,12 @@
         rect.y = position;
         rect.height = full_size;
         rect.width = w;
-
+#if 0
         erase_rect.y = position;
         erase_rect.x = 0;
         erase_rect.height = full_size;
         erase_rect.width = w;
+#endif
     }
 
 #if 0
diff -ur --exclude=CVS ../wxWidgets/src/gtk/scrolbar.cpp ./src/gtk/scrolbar.cpp
--- ../wxWidgets/src/gtk/scrolbar.cpp	2005-03-26 20:28:01.000000000 -0800
+++ ./src/gtk/scrolbar.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -35,7 +35,6 @@
 //-----------------------------------------------------------------------------
 
 extern bool   g_blockEventsOnDrag;
-extern bool   g_blockEventsOnScroll;
 static wxEventType g_currentUpDownEvent = wxEVT_NULL;
 
 static const float sensitivity = 0.02;
diff -ur --exclude=CVS ../wxWidgets/src/gtk/tbargtk.cpp ./src/gtk/tbargtk.cpp
--- ../wxWidgets/src/gtk/tbargtk.cpp	2005-03-26 20:28:01.000000000 -0800
+++ ./src/gtk/tbargtk.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -32,8 +32,6 @@
 #include <glib.h>
 #include "wx/gtk/private.h"
 
-extern GdkFont *GtkGetDefaultGuiFont();
-
 // ----------------------------------------------------------------------------
 // globals
 // ----------------------------------------------------------------------------
diff -ur --exclude=CVS ../wxWidgets/src/gtk/textctrl.cpp ./src/gtk/textctrl.cpp
--- ../wxWidgets/src/gtk/textctrl.cpp	2005-06-06 19:26:01.000000000 -0700
+++ ./src/gtk/textctrl.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -42,7 +42,6 @@
 // data
 //-----------------------------------------------------------------------------
 
-extern bool       g_blockEventsOnDrag;
 extern wxCursor   g_globalCursor;
 extern wxWindowGTK *g_delayedFocus;
 
diff -ur --exclude=CVS ../wxWidgets/src/gtk/tglbtn.cpp ./src/gtk/tglbtn.cpp
--- ../wxWidgets/src/gtk/tglbtn.cpp	2005-03-26 20:28:01.000000000 -0800
+++ ./src/gtk/tglbtn.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -24,7 +24,6 @@
 extern bool g_isIdle;
 extern bool      g_blockEventsOnDrag;
 extern wxCursor   g_globalCursor;
-extern wxWindowGTK *g_delayedFocus;
 
 extern "C" {
 static void gtk_togglebutton_clicked_callback(GtkWidget *WXUNUSED(widget), wxToggleButton *cb)
diff -ur --exclude=CVS ../wxWidgets/src/mac/carbon/app.cpp ./src/mac/carbon/app.cpp
--- ../wxWidgets/src/mac/carbon/app.cpp	2005-06-06 19:26:01.000000000 -0700
+++ ./src/mac/carbon/app.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -79,15 +79,10 @@
 
 #define wxMAC_USE_RAEL 0
 
-#if wxUSE_THREADS
-extern size_t g_numberOfThreads;
-#endif // wxUSE_THREADS
-
 // statics for implementation
 
 static bool s_inYield = false;
 
-static bool s_inReceiveEvent = false ;
 static EventTime sleepTime = kEventDurationNoWait ;
 
 IMPLEMENT_DYNAMIC_CLASS(wxApp, wxEvtHandler)
@@ -97,13 +92,8 @@
     EVT_QUERY_END_SESSION(wxApp::OnQueryEndSession)
 END_EVENT_TABLE()
 
-
-const short    kMacMinHeap = (29 * 1024) ;
 // platform specific static variables
 
-const short kwxMacMenuBarResource = 1 ;
-const short kwxMacAppleMenuId = 1 ;
-
 WXHRGN    wxApp::s_macCursorRgn = NULL;
 wxWindow* wxApp::s_captureWindow = NULL ;
 int       wxApp::s_lastMouseDown = 0 ;
@@ -829,7 +819,6 @@
 bool wxMacConvertEventToRecord( EventRef event , EventRecord *rec)
 {
     bool converted = ConvertEventRefToEventRecord( event,rec) ;
-    OSStatus err = noErr ;
     if ( !converted )
     {
         switch( GetEventClass( event ) )
@@ -862,9 +851,9 @@
                     UInt32 modifiers ;
                     GetMouse( &rec->where) ;
 
-                    err = GetEventParameter(event, kEventParamKeyModifiers, typeUInt32, NULL, 4, NULL, &modifiers);
-                    err = GetEventParameter(event, kEventParamKeyCode, typeUInt32, NULL, 4, NULL, &keyCode);
-                    err = GetEventParameter(event, kEventParamKeyMacCharCodes, typeChar, NULL, 1, NULL, &charCode);
+                    GetEventParameter(event, kEventParamKeyModifiers, typeUInt32, NULL, 4, NULL, &modifiers);
+                    GetEventParameter(event, kEventParamKeyCode, typeUInt32, NULL, 4, NULL, &keyCode);
+                    GetEventParameter(event, kEventParamKeyMacCharCodes, typeChar, NULL, 1, NULL, &charCode);
                     rec->modifiers = modifiers ;
                     rec->message = (keyCode << 8 ) + charCode ;
                 }
@@ -877,7 +866,7 @@
                     case kEventTextInputUnicodeForKeyEvent :
                         {
                             EventRef rawEvent ;
-                            err = GetEventParameter( event , kEventParamTextInputSendKeyboardEvent ,typeEventRef,NULL,sizeof(rawEvent),NULL,&rawEvent ) ;
+                            GetEventParameter( event , kEventParamTextInputSendKeyboardEvent ,typeEventRef,NULL,sizeof(rawEvent),NULL,&rawEvent ) ;
                             converted = true ;
                             {
                                 UInt32 keyCode ;
@@ -885,9 +874,9 @@
                                 UInt32 modifiers ;
                                 GetMouse( &rec->where) ;
                                 rec->what = keyDown ;
-                                err = GetEventParameter(rawEvent, kEventParamKeyModifiers, typeUInt32, NULL, 4, NULL, &modifiers);
-                                err = GetEventParameter(rawEvent, kEventParamKeyCode, typeUInt32, NULL, 4, NULL, &keyCode);
-                                err = GetEventParameter(rawEvent, kEventParamKeyMacCharCodes, typeChar, NULL, 1, NULL, &charCode);
+                                GetEventParameter(rawEvent, kEventParamKeyModifiers, typeUInt32, NULL, 4, NULL, &modifiers);
+                                GetEventParameter(rawEvent, kEventParamKeyCode, typeUInt32, NULL, 4, NULL, &keyCode);
+                                GetEventParameter(rawEvent, kEventParamKeyMacCharCodes, typeChar, NULL, 1, NULL, &charCode);
                                 rec->modifiers = modifiers ;
                                 rec->message = (keyCode << 8 ) + charCode ;
                             }
@@ -1030,9 +1019,7 @@
     OSStatus status = noErr ;
     do
     {
-        s_inReceiveEvent = true ;
         status = ReceiveNextEvent(0, NULL,kEventDurationNoWait,true,&theEvent) ;
-        s_inReceiveEvent = false ;
 
         if ( status == eventLoopTimedOutErr )
         {
@@ -1061,9 +1048,7 @@
 {
     EventRef theEvent;
 
-    s_inReceiveEvent = true ;
     OSStatus status = ReceiveNextEvent(0, NULL,sleepTime,true,&theEvent) ;
-    s_inReceiveEvent = false ;
     if ( status == eventLoopTimedOutErr )
     {
         if ( wxTheApp->ProcessIdle() )
diff -ur --exclude=CVS ../wxWidgets/src/mac/carbon/bitmap.cpp ./src/mac/carbon/bitmap.cpp
--- ../wxWidgets/src/mac/carbon/bitmap.cpp	2005-06-06 19:26:02.000000000 -0700
+++ ./src/mac/carbon/bitmap.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -676,10 +676,10 @@
         }
 
 
-        OSStatus err = ( IconRefToIconFamily( MAC_WXHICON(icon.GetHICON()) , selector , &iconFamily ) ) ;
+        IconRefToIconFamily( MAC_WXHICON(icon.GetHICON()) , selector , &iconFamily ) ;
 
-        err =( GetIconFamilyData( iconFamily , dataType , imagehandle ) ) ;
-        err =( GetIconFamilyData( iconFamily , maskType , maskhandle ) ) ;
+        GetIconFamilyData( iconFamily , dataType , imagehandle ) ;
+        GetIconFamilyData( iconFamily , maskType , maskhandle ) ;
         size_t imagehandlesize = GetHandleSize( imagehandle ) ;
         size_t maskhandlesize = GetHandleSize( maskhandle ) ;
 
diff -ur --exclude=CVS ../wxWidgets/src/mac/carbon/combobox.cpp ./src/mac/carbon/combobox.cpp
--- ../wxWidgets/src/mac/carbon/combobox.cpp	2005-05-16 20:33:04.000000000 -0700
+++ ./src/mac/carbon/combobox.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -51,7 +51,6 @@
 static const wxCoord MARGIN = 2;
 static const int    TEXTFOCUSBORDER = 0 ;
 #endif
-static const int    POPUPHEIGHT = 23;
 
 
 // ----------------------------------------------------------------------------
diff -ur --exclude=CVS ../wxWidgets/src/mac/carbon/combobxc.cpp ./src/mac/carbon/combobxc.cpp
--- ../wxWidgets/src/mac/carbon/combobxc.cpp	2005-02-25 22:58:30.000000000 -0800
+++ ./src/mac/carbon/combobxc.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -368,7 +368,6 @@
         return false;
     }
 #if USE_HICOMBOBOX
-    Rect bounds = wxMacGetBoundsForControl( this , pos , size ) ;
     HIRect hiRect;
 
     hiRect.origin.x = 20; //bounds.left;
@@ -377,6 +376,7 @@
     hiRect.size.height = 24;
 
     //For some reason, this code causes the combo box not to be displayed at all.
+    //Rect bounds = wxMacGetBoundsForControl( this , pos , size ) ;
     //hiRect.origin.x = bounds.left;
     //hiRect.origin.y = bounds.top;
     //hiRect.size.width = bounds.right - bounds.left;
diff -ur --exclude=CVS ../wxWidgets/src/mac/carbon/cursor.cpp ./src/mac/carbon/cursor.cpp
--- ../wxWidgets/src/mac/carbon/cursor.cpp	2005-03-05 09:11:23.000000000 -0800
+++ ./src/mac/carbon/cursor.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -46,7 +46,6 @@
 };
 
 #define M_CURSORDATA ((wxCursorRefData *)m_refData)
-#define M_CURSORHANDLERDATA ((wxCursorRefData *)bitmap->m_refData)
 
 const short kwxCursorBullseye = 0 ;
 const short kwxCursorBlank = 1 ;
diff -ur --exclude=CVS ../wxWidgets/src/mac/carbon/dc.cpp ./src/mac/carbon/dc.cpp
--- ../wxWidgets/src/mac/carbon/dc.cpp	2005-05-16 20:33:04.000000000 -0700
+++ ./src/mac/carbon/dc.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -46,8 +46,6 @@
 const short kEmulatedMode = -1 ;
 const short kUnsupportedMode = -2 ;
 
-extern TECObjectRef s_TECNativeCToUnicode ;
-
 // set to 0 if problems arise
 #define wxMAC_EXPERIMENTAL_DC 1
 
@@ -143,13 +141,6 @@
 }
 
 //-----------------------------------------------------------------------------
-// Local functions
-//-----------------------------------------------------------------------------
-static inline double dmin(double a, double b) { return a < b ? a : b; }
-static inline double dmax(double a, double b) { return a > b ? a : b; }
-static inline double DegToRad(double deg) { return (deg * M_PI) / 180.0; }
-
-//-----------------------------------------------------------------------------
 // wxDC
 //-----------------------------------------------------------------------------
 // this function emulates all wx colour manipulations, used to verify the implementation
diff -ur --exclude=CVS ../wxWidgets/src/mac/carbon/dcclient.cpp ./src/mac/carbon/dcclient.cpp
--- ../wxWidgets/src/mac/carbon/dcclient.cpp	2005-05-16 20:33:04.000000000 -0700
+++ ./src/mac/carbon/dcclient.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -29,8 +29,6 @@
 // constants
 //-----------------------------------------------------------------------------
 
-#define RAD2DEG 57.2957795131
-
 //-----------------------------------------------------------------------------
 // wxPaintDC
 //-----------------------------------------------------------------------------
@@ -114,7 +112,6 @@
     WindowRef windowref = (WindowRef) rootwindow->MacGetWindowRef() ;
     int x , y ;
     x = y = 0 ;
-    wxSize size = window->GetSize() ;
     window->MacWindowToRootWindow( &x , &y ) ;
     m_macPort = UMAGetWindowPort( windowref ) ;
     
diff -ur --exclude=CVS ../wxWidgets/src/mac/carbon/dirmac.cpp ./src/mac/carbon/dirmac.cpp
--- ../wxWidgets/src/mac/carbon/dirmac.cpp	2005-04-05 20:18:16.000000000 -0700
+++ ./src/mac/carbon/dirmac.cpp	2005-06-06 20:19:05.000000000 -0700
@@ -46,14 +46,6 @@
 #include "MoreFilesX.h"
 
 // ----------------------------------------------------------------------------
-// constants
-// ----------------------------------------------------------------------------
-
-#ifndef MAX_PATH
-    #define MAX_PATH 260        // from VC++ headers
-#endif
-
-// ----------------------------------------------------------------------------
 // macros
 // ----------------------------------------------------------------------------
 
diff -ur --exclude=CVS ../wxWidgets/src/mac/carbon/dnd.cpp ./src/mac/carbon/dnd.cpp
--- ../wxWidgets/src/mac/carbon/dnd.cpp	2005-03-01 13:36:53.000000000 -0800
+++ ./src/mac/carbon/dnd.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -108,9 +108,8 @@
     if ( !supported )
     {
         UInt16 items ;
-        OSErr result;
         CountDragItems((DragReference)m_currentDrag, &items);
-        for (UInt16 index = 1; index <= items && supported == false ; ++index) 
+        for (UInt16 index = 1; index <= items && !supported ; ++index) 
         {
             ItemReference theItem;
             FlavorType theType ;
@@ -119,7 +118,7 @@
             CountDragItemFlavors( (DragReference)m_currentDrag, theItem , &flavors ) ;
             for ( UInt16 flavor = 1 ; flavor <= flavors ; ++flavor )
             {
-                result = GetFlavorType((DragReference)m_currentDrag, theItem, flavor , &theType);
+                GetFlavorType((DragReference)m_currentDrag, theItem, flavor , &theType);
                 if ( m_dataObject->IsSupportedFormat( wxDataFormat( theType ) ) )
                 {
                     supported = true ;
@@ -297,10 +296,9 @@
     if (m_data->GetFormatCount() == 0)
         return (wxDragResult) wxDragNone;
     
-    OSErr result;
     DragReference theDrag;
     RgnHandle dragRegion;
-    if ((result = NewDrag(&theDrag)))
+    if (NewDrag(&theDrag))
     {
         return wxDragNone ;
     }
@@ -391,7 +389,7 @@
     // was successfully completed
     
     gTrackingGlobals.m_currentSource = this ;
-    result = TrackDrag(theDrag, ev , dragRegion);
+    TrackDrag(theDrag, ev , dragRegion);
     DisposeRgn(dragRegion);
     DisposeDrag(theDrag);
     gTrackingGlobals.m_currentSource = NULL ;
diff -ur --exclude=CVS ../wxWidgets/src/mac/carbon/filedlg.cpp ./src/mac/carbon/filedlg.cpp
--- ../wxWidgets/src/mac/carbon/filedlg.cpp	2005-05-16 20:33:04.000000000 -0700
+++ ./src/mac/carbon/filedlg.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -37,8 +37,6 @@
 
 #include "MoreFilesX.h"
 
-extern bool gUseNavServices ;
-
 // the data we need to pass to our standard file hook routine
 // includes a pointer to the dialog, a pointer to the standard
 // file reply record (so we can inspect the current selection)
diff -ur --exclude=CVS ../wxWidgets/src/mac/carbon/gsocket.cpp ./src/mac/carbon/gsocket.cpp
--- ../wxWidgets/src/mac/carbon/gsocket.cpp	2005-06-06 19:26:02.000000000 -0700
+++ ./src/mac/carbon/gsocket.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -78,7 +78,6 @@
     #define  htons(x) (x)
 #endif
 
-void wxCYield() ;
 #ifdef __WXDEBUG__
 #define qDebug 1
 #define qDebug2 1
@@ -1051,13 +1050,6 @@
   }
 }
 
-
-#define CALL_CALLBACK(socket, event) {                                  \
-  _GSocket_Disable(socket, event);                                      \
-  if (socket->m_cbacks[event])                                          \
-    socket->m_cbacks[event](socket, event, socket->m_data[event]);      \
-}
-
 int GSocket::Recv_Stream(char *buffer, int size)
 {
 	OTFlags flags ;
diff -ur --exclude=CVS ../wxWidgets/src/mac/carbon/mdi.cpp ./src/mac/carbon/mdi.cpp
--- ../wxWidgets/src/mac/carbon/mdi.cpp	2005-05-16 20:33:04.000000000 -0700
+++ ./src/mac/carbon/mdi.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -42,20 +42,11 @@
 
 static const wxChar *TRACE_MDI = _T("mdi");
 
-static const int IDM_WINDOWTILE  = 4001;
 static const int IDM_WINDOWTILEHOR  = 4001;
 static const int IDM_WINDOWCASCADE = 4002;
 static const int IDM_WINDOWICONS = 4003;
 static const int IDM_WINDOWNEXT = 4004;
 static const int IDM_WINDOWTILEVERT = 4005;
-static const int IDM_WINDOWPREV = 4006;
-
-// This range gives a maximum of 500 MDI children. Should be enough :-)
-static const int wxFIRST_MDI_CHILD = 4100;
-static const int wxLAST_MDI_CHILD = 4600;
-
-// Status border dimensions
-static const int wxTHICK_LINE_BORDER = 3;
 
 // Parent frame
 
diff -ur --exclude=CVS ../wxWidgets/src/mac/carbon/menu.cpp ./src/mac/carbon/menu.cpp
--- ../wxWidgets/src/mac/carbon/menu.cpp	2005-06-06 19:26:02.000000000 -0700
+++ ./src/mac/carbon/menu.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -43,7 +43,6 @@
 // the (popup) menu title has this special id
 static const int idMenuTitle = -3;
 
-const short kwxMacMenuBarResource = 1 ;
 const short kwxMacAppleMenuId = 1 ;
 
 
@@ -527,11 +526,9 @@
     if ( isSubMenu )
         ::DeleteMenu(MacGetMenuId());
 
-    wxMenuItem* previousItem = NULL ;
-    int pos ;
     wxMenuItemList::compatibility_iterator node;
     wxMenuItem *item;
-    for (pos = 0, node = GetMenuItems().GetFirst(); node; node = node->GetNext(), pos++)
+    for (node = GetMenuItems().GetFirst(); node; node = node->GetNext())
     {
         item = (wxMenuItem *)node->GetData();
         wxMenu* subMenu = item->GetSubMenu() ;
@@ -543,7 +540,6 @@
         {
             // no need to undo hidings
         }
-        previousItem = item ;
     }
 }
 
@@ -699,12 +695,11 @@
     {
         wxMenuItemList::compatibility_iterator node;
         wxMenuItem *item;
-        int pos ;
         wxMenu* menu = menuIter->GetData() , *subMenu = NULL ;
 
         if( m_titles[i] == wxT("?") || m_titles[i] == wxT("&?")  || m_titles[i] == wxApp::s_macHelpMenuTitleName )
         {
-            for (pos = 0 , node = menu->GetMenuItems().GetFirst(); node; node = node->GetNext(), pos++)
+            for (node = menu->GetMenuItems().GetFirst(); node; node = node->GetNext())
             {
                 item = (wxMenuItem *)node->GetData();
                 subMenu = item->GetSubMenu() ;
diff -ur --exclude=CVS ../wxWidgets/src/mac/carbon/metafile.cpp ./src/mac/carbon/metafile.cpp
--- ../wxWidgets/src/mac/carbon/metafile.cpp	2005-06-06 19:26:02.000000000 -0700
+++ ./src/mac/carbon/metafile.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -35,8 +35,6 @@
 #include <stdio.h>
 #include <string.h>
 
-extern bool wxClipboardIsOpen;
-
 IMPLEMENT_DYNAMIC_CLASS(wxMetafile, wxObject)
 IMPLEMENT_ABSTRACT_CLASS(wxMetafileDC, wxDC)
 
diff -ur --exclude=CVS ../wxWidgets/src/mac/carbon/mimetmac.cpp ./src/mac/carbon/mimetmac.cpp
--- ../wxWidgets/src/mac/carbon/mimetmac.cpp	2005-06-01 09:20:55.000000000 -0700
+++ ./src/mac/carbon/mimetmac.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -1272,7 +1272,6 @@
 // ----------------------------------------------------------------------------
 
 #define wxCFDATA_RELEASEBUFFER  1
-#define wxCFDATA_RETAINBUFFER   0
 
 class wxCFData
 {
@@ -2006,4 +2005,4 @@
                             CFWriteStreamClose(cfwsInfo);
 
 */
-#endif //wxUSE_MIMETYPE
\ No newline at end of file
+#endif //wxUSE_MIMETYPE
diff -ur --exclude=CVS ../wxWidgets/src/mac/carbon/radiobox.cpp ./src/mac/carbon/radiobox.cpp
--- ../wxWidgets/src/mac/carbon/radiobox.cpp	2005-05-16 20:33:05.000000000 -0700
+++ ./src/mac/carbon/radiobox.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -383,13 +383,10 @@
 
 void wxRadioBox::SetFocus()
 {
-    int i;
     wxRadioButton *current;
 
-    i=0;
     current=m_radioButtonCycle;
     while (!current->GetValue()) {
-        i++;
         current=current->NextInCycle();
     }
     current->SetFocus();
diff -ur --exclude=CVS ../wxWidgets/src/mac/carbon/radiobut.cpp ./src/mac/carbon/radiobut.cpp
--- ../wxWidgets/src/mac/carbon/radiobut.cpp	2005-05-16 20:33:05.000000000 -0700
+++ ./src/mac/carbon/radiobut.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -139,7 +139,7 @@
 
 wxRadioButton *wxRadioButton::AddInCycle(wxRadioButton *cycle)
 {
-    wxRadioButton *next,*current;
+    wxRadioButton *current;
         
     if (cycle==NULL) {
         m_cycle=this;
@@ -147,7 +147,7 @@
         }
     else {
         current=cycle;
-          while ((next=current->m_cycle)!=cycle) 
+          while (current->m_cycle != cycle) 
             current=current->m_cycle;
           m_cycle=cycle;
           current->m_cycle=this;
diff -ur --exclude=CVS ../wxWidgets/src/mac/carbon/slider.cpp ./src/mac/carbon/slider.cpp
--- ../wxWidgets/src/mac/carbon/slider.cpp	2005-05-31 19:44:02.000000000 -0700
+++ ./src/mac/carbon/slider.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -26,7 +26,6 @@
 END_EVENT_TABLE()
 
  // The dimensions of the different styles of sliders (From Aqua document)
-#define wxSLIDER_DIMENSIONACROSS 15
 #define wxSLIDER_DIMENSIONACROSS_WITHTICKMARKS 24
 #define wxSLIDER_DIMENSIONACROSS_ARROW 18
 
diff -ur --exclude=CVS ../wxWidgets/src/mac/carbon/sound.cpp ./src/mac/carbon/sound.cpp
--- ../wxWidgets/src/mac/carbon/sound.cpp	2005-05-31 19:44:02.000000000 -0700
+++ ./src/mac/carbon/sound.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -302,14 +302,13 @@
             Track targetTrack = nil;
             TimeValue addedDuration = 0;
             long outFlags = 0;
-            OSErr err;
             ComponentResult result;
 
             myHandle = NewHandleClear((Size)m_waveLength);
 
             BlockMove(m_hSnd, *myHandle, m_waveLength);
 
-            err = PtrToHand(&myHandle, &dataRef, sizeof(Handle));
+            PtrToHand(&myHandle, &dataRef, sizeof(Handle));
 
             if (memcmp(&m_hSnd[8], "WAVE", 4) == 0)
                 miComponent = OpenDefaultComponent(MovieImportType, kQTFileTypeWave);
diff -ur --exclude=CVS ../wxWidgets/src/mac/carbon/statline.cpp ./src/mac/carbon/statline.cpp
--- ../wxWidgets/src/mac/carbon/statline.cpp	2005-02-04 16:57:52.000000000 -0800
+++ ./src/mac/carbon/statline.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -53,8 +53,7 @@
     // ok, this is ugly but it's better than nothing: use a thin static box to
     // emulate static line
 
-    wxSize sizeReal = AdjustSize(size);
-
+//    wxSize sizeReal = AdjustSize(size);
 //    m_statbox = new wxStaticBox(parent, id, wxT(""), pos, sizeReal, style, name);
 
     return TRUE;
diff -ur --exclude=CVS ../wxWidgets/src/mac/carbon/textctrl.cpp ./src/mac/carbon/textctrl.cpp
--- ../wxWidgets/src/mac/carbon/textctrl.cpp	2005-05-31 19:44:02.000000000 -0700
+++ ./src/mac/carbon/textctrl.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -2035,7 +2035,6 @@
 
 pascal void wxMacMLTEClassicControl::TXNScrollActionProc( ControlRef controlRef , ControlPartCode partCode )
 {
-    OSStatus err ;
     wxMacMLTEClassicControl* mlte = (wxMacMLTEClassicControl*) GetControlReference( controlRef ) ;
     if ( mlte == NULL )
         return ;
@@ -2088,7 +2087,7 @@
         SInt32 verticalDelta = isHorizontal ? 0 : delta ;
         SInt32 horizontalDelta = isHorizontal ? delta : 0 ;
         
-        err = TXNScroll( mlte->m_txn , kTXNScrollUnitsInPixels, kTXNScrollUnitsInPixels,
+        TXNScroll( mlte->m_txn , kTXNScrollUnitsInPixels, kTXNScrollUnitsInPixels,
                                     &verticalDelta , &horizontalDelta  );
                                     
         if ( isHorizontal )
diff -ur --exclude=CVS ../wxWidgets/src/mac/carbon/tglbtn.cpp ./src/mac/carbon/tglbtn.cpp
--- ../wxWidgets/src/mac/carbon/tglbtn.cpp	2005-02-25 22:58:37.000000000 -0800
+++ ./src/mac/carbon/tglbtn.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -30,9 +30,6 @@
 #include "wx/mac/uma.h"
 // Button
 
-static const int kMacOSXHorizontalBorder = 2 ;
-static const int kMacOSXVerticalBorder = 4 ;
-
 // ----------------------------------------------------------------------------
 // macros
 // ----------------------------------------------------------------------------
diff -ur --exclude=CVS ../wxWidgets/src/mac/carbon/timer.cpp ./src/mac/carbon/timer.cpp
--- ../wxWidgets/src/mac/carbon/timer.cpp	2005-05-16 20:33:05.000000000 -0700
+++ ./src/mac/carbon/timer.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -32,7 +32,7 @@
 
 #if wxMAC_USE_CARBON_TIMER
 
-typedef struct MacTimerInfo
+struct MacTimerInfo
 {
     wxTimer* m_timer ;
     EventLoopTimerUPP m_proc ;
diff -ur --exclude=CVS ../wxWidgets/src/mac/carbon/toplevel.cpp ./src/mac/carbon/toplevel.cpp
--- ../wxWidgets/src/mac/carbon/toplevel.cpp	2005-06-06 19:26:03.000000000 -0700
+++ ./src/mac/carbon/toplevel.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -75,8 +75,6 @@
 // Carbon Events
 // ---------------------------------------------------------------------------
 
-extern long wxMacTranslateKey(unsigned char key, unsigned char code) ;
-
 static const EventTypeSpec eventList[] =
 {
     // TODO remove control related event like key and mouse (except for WindowLeave events)
@@ -1481,18 +1479,15 @@
 
             EventRef currentEvent = (EventRef) wxTheApp->MacGetCurrentEvent() ;
             UInt32 currentEventClass = 0 ;
-            UInt32 currentEventKind = 0 ;
             if ( currentEvent != NULL )
             {
                 currentEventClass = ::GetEventClass( currentEvent ) ;
-                currentEventKind = ::GetEventKind( currentEvent ) ;
             }
             if ( currentEventClass != kEventClassMenu )
             {
                 // when tracking a menu, strange redraw errors occur if we flush now, so leave..
                 EventRef theEvent;
-                OSStatus status = noErr ;
-                status = ReceiveNextEvent( 0 , NULL , kEventDurationNoWait , false , &theEvent ) ;
+                ReceiveNextEvent( 0 , NULL , kEventDurationNoWait , false , &theEvent ) ;
             }
         }
     }
diff -ur --exclude=CVS ../wxWidgets/src/mac/carbon/uma.cpp ./src/mac/carbon/uma.cpp
--- ../wxWidgets/src/mac/carbon/uma.cpp	2005-05-31 19:44:02.000000000 -0700
+++ ./src/mac/carbon/uma.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -50,7 +50,6 @@
 
 static bool sUMAHasInittedAppearance = false;
 
-extern int gAGABackgroundColor ;
 bool UMAHasAppearance() { return sUMAHasAppearance ; }
 long UMAGetAppearanceVersion() { return sUMAAppearanceVersion ; }
 long UMAGetSystemVersion() { return sUMASystemVersion ; }
@@ -465,13 +464,9 @@
 
 void UMAShowWatchCursor()
 {
-    OSErr err = noErr;
-
     CursHandle watchFob = GetCursor (watchCursor);
 
-    if (!watchFob)
-        err = nilHandleErr;
-    else
+    if (watchFob)
     {
     #if TARGET_CARBON
 //        Cursor preservedArrow;
diff -ur --exclude=CVS ../wxWidgets/src/mac/carbon/window.cpp ./src/mac/carbon/window.cpp
--- ../wxWidgets/src/mac/carbon/window.cpp	2005-06-06 19:26:03.000000000 -0700
+++ ./src/mac/carbon/window.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -92,8 +92,6 @@
 #define wxMAC_DEBUG_REDRAW 0
 #endif
 
-#define wxMAC_USE_THEME_BORDER 1
-
 // ---------------------------------------------------------------------------
 // Utility Routines to move between different coordinate systems
 // ---------------------------------------------------------------------------
@@ -146,9 +144,6 @@
 // Carbon Events
 // ---------------------------------------------------------------------------
 
-extern long wxMacTranslateKey(unsigned char key, unsigned char code) ;
-pascal OSStatus wxMacSetupControlBackground( ControlRef iControl , SInt16 iMessage , SInt16 iDepth , Boolean iIsColor ) ;
-
 #if TARGET_API_MAC_OSX
 
 #if MAC_OS_X_VERSION_MAX_ALLOWED < MAC_OS_X_VERSION_10_3
@@ -2976,11 +2971,9 @@
     {
         bool hasBoth = ( style & wxVSCROLL ) && ( style & wxHSCROLL ) ;
         int scrlsize = MAC_SCROLLBAR_SIZE ;
-        wxWindowVariant variant = wxWINDOW_VARIANT_NORMAL ;
         if ( GetWindowVariant() == wxWINDOW_VARIANT_SMALL || GetWindowVariant() == wxWINDOW_VARIANT_MINI )
         {
             scrlsize = MAC_SMALL_SCROLLBAR_SIZE ;
-            variant = wxWINDOW_VARIANT_SMALL ;
         }
 
         int adjust = hasBoth ? scrlsize - 1: 0 ;
diff -ur --exclude=CVS ../wxWidgets/src/msw/bmpbuttn.cpp ./src/msw/bmpbuttn.cpp
--- ../wxWidgets/src/msw/bmpbuttn.cpp	2005-05-17 20:52:15.000000000 -0700
+++ ./src/msw/bmpbuttn.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -103,8 +103,6 @@
 bitmap "disabled" ,
 */
 
-#define BUTTON_HEIGHT_FACTOR (EDIT_CONTROL_FACTOR * 1.1)
-
 bool wxBitmapButton::Create(wxWindow *parent, wxWindowID id,
     const wxBitmap& bitmap,
     const wxPoint& pos,
diff -ur --exclude=CVS ../wxWidgets/src/msw/mdi.cpp ./src/msw/mdi.cpp
--- ../wxWidgets/src/msw/mdi.cpp	2005-05-22 09:37:04.000000000 -0700
+++ ./src/msw/mdi.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -76,7 +76,6 @@
 // constants
 // ---------------------------------------------------------------------------
 
-static const int IDM_WINDOWTILE  = 4001;
 static const int IDM_WINDOWTILEHOR  = 4001;
 static const int IDM_WINDOWCASCADE = 4002;
 static const int IDM_WINDOWICONS = 4003;
@@ -88,10 +87,6 @@
 static const int wxFIRST_MDI_CHILD = 4100;
 static const int wxLAST_MDI_CHILD = 4600;
 
-// Status border dimensions
-static const int wxTHICK_LINE_BORDER = 3;
-static const int wxTHICK_LINE_WIDTH  = 1;
-
 // ---------------------------------------------------------------------------
 // private functions
 // ---------------------------------------------------------------------------
diff -ur --exclude=CVS ../wxWidgets/src/msw/menu.cpp ./src/msw/menu.cpp
--- ../wxWidgets/src/msw/menu.cpp	2005-05-31 19:44:18.000000000 -0700
+++ ./src/msw/menu.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -66,8 +66,6 @@
 // global variables
 // ----------------------------------------------------------------------------
 
-extern wxMenu *wxCurrentPopupMenu;
-
 // ----------------------------------------------------------------------------
 // constants
 // ----------------------------------------------------------------------------
diff -ur --exclude=CVS ../wxWidgets/src/msw/mimetype.cpp ./src/msw/mimetype.cpp
--- ../wxWidgets/src/msw/mimetype.cpp	2004-11-05 10:23:47.000000000 -0800
+++ ./src/msw/mimetype.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -787,7 +787,6 @@
     wxCHECK_MSG( !m_ext.empty() && !verb.empty(), false,
                  _T("RemoveCommand() needs an extension and a verb") );
 
-    wxString  sKey = m_strFileType;
     wxRegKey rkey(wxRegKey::HKCR, GetVerbPath(verb));
 
     // if the key already doesn't exist, it's a success
diff -ur --exclude=CVS ../wxWidgets/src/msw/tbar95.cpp ./src/msw/tbar95.cpp
--- ../wxWidgets/src/msw/tbar95.cpp	2005-05-29 09:33:18.000000000 -0700
+++ ./src/msw/tbar95.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -93,9 +93,6 @@
 // these values correspond to those used by comctl32.dll
 #define DEFAULTBITMAPX   16
 #define DEFAULTBITMAPY   15
-#define DEFAULTBUTTONX   24
-#define DEFAULTBUTTONY   24
-#define DEFAULTBARHEIGHT 27
 
 // ----------------------------------------------------------------------------
 // wxWin macros
diff -ur --exclude=CVS ../wxWidgets/src/unix/sound.cpp ./src/unix/sound.cpp
--- ../wxWidgets/src/unix/sound.cpp	2004-11-24 14:50:29.000000000 -0800
+++ ./src/unix/sound.cpp	2005-06-06 20:18:14.000000000 -0700
@@ -620,7 +620,6 @@
     wxUint16      uiBitsPerSample;
 } WAVEFORMAT;
 
-#define MONO             1  // and stereo is 2 by wav format
 #define WAVE_FORMAT_PCM  1
 #define WAVE_INDEX       8
 #define FMT_INDEX       12

 	  	 
