Index: msw/slider.cpp
===================================================================
--- msw/slider.cpp	(revision 68233)
+++ msw/slider.cpp	(working copy)
@@ -490,7 +490,7 @@
         int xLabelValue =
             x + minLabelWidth +
             ((width  - (minLabelWidth + maxLabelWidth)) / 2) -
-            (maxLabelWidth / 2);
+            (longestLabelWidth / 2);
 
         int ySlider = y;
 
