Index: src/common/filename.cpp
===================================================================
--- src/common/filename.cpp	(revision 64586)
+++ src/common/filename.cpp	(working copy)
@@ -297,13 +297,12 @@
 
     if ( !volume.empty() )
     {
-        format = wxFileName::GetFormat(format);
-
+    #ifdef __WXMSW__
         // Special Windows UNC paths hack, part 2: undo what we did in
         // SplitPath() and make an UNC path if we have a drive which is not a
         // single letter (hopefully the network shares can't be one letter only
         // although I didn't find any authoritative docs on this)
-        if ( format == wxPATH_DOS && volume.length() > 1 )
+        if ( volume.length() > 1 )
         {
             // We also have to check for Windows unique volume names here and
             // return it with '\\?\' prepended to it
@@ -315,16 +314,15 @@
             else
             {
                 // it must be a UNC path
-                path << wxFILE_SEP_PATH_DOS << wxFILE_SEP_PATH_DOS << volume;
+                path << wxFileName::GetPathSeparator(format) << wxFileName::GetPathSeparator(format) << volume;
             }
         }
-        else if  ( format == wxPATH_DOS || format == wxPATH_VMS )
+        else
         {
-            path << volume << wxFileName::GetVolumeSeparator(format);
+            path << volume << wxFileName::GetVolumeSeparator(wxPATH_DOS);
         }
-        // else ignore
+    #endif
     }
-
     return path;
 }
 
