--- choice.orig	2008-09-22 16:13:04.000000000 +0200
+++ choice.cpp	2008-09-22 16:18:30.000000000 +0200
@@ -751,8 +751,12 @@
             if ( m_lastAcceptedSelection != wxID_NONE )
             {
                 SetSelection(m_lastAcceptedSelection);
-                m_lastAcceptedSelection = wxID_NONE;
             }
+            else
+            {
+                SetSelection(-1);
+            }		
+            m_lastAcceptedSelection = wxID_NONE;
             break;
 
         case CBN_SELENDOK:
