Index: include/wx/anidecod.h
===================================================================
--- include/wx/anidecod.h	(revision 54603)
+++ include/wx/anidecod.h	(working copy)
@@ -18,8 +18,8 @@
 #include "wx/image.h"
 #include "wx/animdecod.h"
 #include "wx/dynarray.h"
+#include "wx/imagbmp.h"
 
-
 class /*WXDLLIMPEXP_CORE*/ wxANIFrameInfo;      // private implementation detail
 
 WX_DECLARE_EXPORTED_OBJARRAY(wxANIFrameInfo, wxANIFrameInfoArray);
Index: include/wx/image.h
===================================================================
--- include/wx/image.h	(revision 54603)
+++ include/wx/image.h	(working copy)
@@ -531,17 +531,6 @@
 // wxImage handlers
 //-----------------------------------------------------------------------------
 
-#include "wx/imagbmp.h"
-#include "wx/imagpng.h"
-#include "wx/imaggif.h"
-#include "wx/imagpcx.h"
-#include "wx/imagjpeg.h"
-#include "wx/imagtga.h"
-#include "wx/imagtiff.h"
-#include "wx/imagpnm.h"
-#include "wx/imagxpm.h"
-#include "wx/imagiff.h"
-
 #endif // wxUSE_IMAGE
 
 #endif
Index: samples/image/image.cpp
===================================================================
--- samples/image/image.cpp	(revision 54603)
+++ samples/image/image.cpp	(working copy)
@@ -26,6 +26,8 @@
 #include "wx/mstream.h"
 #include "wx/wfstream.h"
 #include "wx/quantize.h"
+#include "wx/imagbmp.h"
+#include "wx/imagpng.h"
 
 #if wxUSE_CLIPBOARD
     #include "wx/dataobj.h"
Index: src/common/dcsvg.cpp
===================================================================
--- src/common/dcsvg.cpp	(revision 54603)
+++ src/common/dcsvg.cpp	(working copy)
@@ -26,6 +26,7 @@
 
 #include "wx/dcsvg.h"
 #include "wx/wfstream.h"
+#include "wx/imagpng.h"
 
 #define wxSVG_DEBUG FALSE
 // or TRUE to see the calls being executed
Index: src/common/imagall.cpp
===================================================================
--- src/common/imagall.cpp	(revision 54603)
+++ src/common/imagall.cpp	(working copy)
@@ -20,6 +20,17 @@
     #include "wx/image.h"
 #endif
 
+#include "wx/imagbmp.h"
+#include "wx/imagpng.h"
+#include "wx/imaggif.h"
+#include "wx/imagpcx.h"
+#include "wx/imagjpeg.h"
+#include "wx/imagtga.h"
+#include "wx/imagtiff.h"
+#include "wx/imagpnm.h"
+#include "wx/imagxpm.h"
+#include "wx/imagiff.h"
+
 //-----------------------------------------------------------------------------
 // This function allows dynamic access to all image handlers compile within
 // the library. This function should be in a separate file as some compilers
Index: src/common/image.cpp
===================================================================
--- src/common/image.cpp	(revision 54603)
+++ src/common/image.cpp	(working copy)
@@ -31,6 +31,7 @@
 #include "wx/filefn.h"
 #include "wx/wfstream.h"
 #include "wx/xpmdecod.h"
+#include "wx/imagbmp.h"
 
 // For memcpy
 #include <string.h>
Index: src/msw/cursor.cpp
===================================================================
--- src/msw/cursor.cpp	(revision 54603)
+++ src/msw/cursor.cpp	(working copy)
@@ -39,6 +39,7 @@
 
 #include "wx/msw/private.h"
 #include "wx/msw/missing.h" // IDC_HAND
+#include "wx/imagbmp.h"
 
 // define functions missing in MicroWin
 #ifdef __WXMICROWIN__
