Index: include/wx/image.h
===================================================================
--- include/wx/image.h	(revision 54660)
+++ include/wx/image.h	(working copy)
@@ -424,6 +424,8 @@
     static wxImageHandler *FindHandler( const wxString& extension, wxBitmapType imageType );
     static wxImageHandler *FindHandler( wxBitmapType imageType );
 
+    static void InitHandlers(const wxBitmapType* array = NULL, size_t array_count = 0);
+
     static wxImageHandler *FindHandlerMime( const wxString& mimetype );
 
     static wxString GetImageExtWildcard();
@@ -523,7 +525,10 @@
 };
 
 
-extern void WXDLLIMPEXP_CORE wxInitAllImageHandlers();
+inline void wxInitAllImageHandlers()
+{
+   wxImage::InitHandlers(NULL, 0);
+}
 
 extern WXDLLIMPEXP_DATA_CORE(wxImage)    wxNullImage;
 
Index: interface/wx/image.h
===================================================================
--- interface/wx/image.h	(revision 54660)
+++ interface/wx/image.h	(working copy)
@@ -680,6 +680,22 @@
     void InitAlpha();
 
     /**
+        Adds a set of image format handlers. 
+
+        @param array
+            Array of bitmap handlers to add.
+            If NULL, add all handlers, equivalent to wxInitAllImageHandlers().
+
+        @param array_count
+            Elements in array.
+        
+        @since 2.9.0
+
+        @see wxInitAllImageHandlers()
+    */
+    static void InitHandlers(const wxBitmapType* array = NULL, size_t array_count = 0);
+
+    /**
         Internal use only. Adds standard image format handlers. It only install BMP
         for the time being, which is used by wxBitmap.
         This function is called by wxWidgets on startup, and shouldn't be called by
@@ -1060,6 +1076,7 @@
 /**
     Initializes all available image handlers. For a list of available handlers,
     see wxImage.
+    Same as calling wxImage::InitHandlers() without arguments
 
     @see wxImage, wxImageHandler
 
Index: samples/html/printing/printing.cpp
===================================================================
--- samples/html/printing/printing.cpp	(revision 54660)
+++ samples/html/printing/printing.cpp	(working copy)
@@ -131,15 +131,10 @@
     if ( !wxApp::OnInit() )
         return false;
 
-#if wxUSE_LIBPNG
-    wxImage::AddHandler(new wxPNGHandler);
-#endif
-#if wxUSE_LIBJPEG
-    wxImage::AddHandler(new wxJPEGHandler);
-#endif
-#if wxUSE_GIF
-    wxImage::AddHandler(new wxGIFHandler);
-#endif
+    const wxBitmapType handlers[] =
+    { wxBITMAP_TYPE_PNG, wxBITMAP_TYPE_JPEG, wxBITMAP_TYPE_GIF };
+    
+    wxImage::InitHandlers(handlers, WXSIZEOF(handlers));
 
     MyFrame *frame = new MyFrame(_("Printing test"),
         wxDefaultPosition, wxSize(640, 480));
Index: src/common/imagall.cpp
===================================================================
--- src/common/imagall.cpp	(revision 54660)
+++ src/common/imagall.cpp	(working copy)
@@ -25,40 +25,114 @@
 // the library. This function should be in a separate file as some compilers
 // link against the whole object file as long as just one of is function is called!
 
-void wxInitAllImageHandlers()
+/*static*/ void wxImage::InitHandlers(const wxBitmapType* array, size_t array_count)
 {
-#if wxUSE_LIBPNG
-  wxImage::AddHandler( new wxPNGHandler );
-#endif
-#if wxUSE_LIBJPEG
-  wxImage::AddHandler( new wxJPEGHandler );
-#endif
-#if wxUSE_LIBTIFF
-  wxImage::AddHandler( new wxTIFFHandler );
-#endif
-#if wxUSE_GIF
-  wxImage::AddHandler( new wxGIFHandler );
-#endif
-#if wxUSE_PNM
-  wxImage::AddHandler( new wxPNMHandler );
-#endif
-#if wxUSE_PCX
-  wxImage::AddHandler( new wxPCXHandler );
-#endif
-#if wxUSE_IFF
-  wxImage::AddHandler( new wxIFFHandler );
-#endif
-#if wxUSE_ICO_CUR
-  wxImage::AddHandler( new wxICOHandler );
-  wxImage::AddHandler( new wxCURHandler );
-  wxImage::AddHandler( new wxANIHandler );
-#endif
-#if wxUSE_TGA
-  wxImage::AddHandler( new wxTGAHandler );
-#endif
-#if wxUSE_XPM
-  wxImage::AddHandler( new wxXPMHandler );
-#endif
+    if (array)
+    {
+        for (size_t i = 0; i < array_count; i++)
+        {
+            switch (array[i])
+            {
+                case wxBITMAP_TYPE_PNG:
+                #if wxUSE_LIBPNG
+                    wxImage::AddHandler( new wxPNGHandler );
+                #endif
+                    break;
+                case wxBITMAP_TYPE_JPEG:
+                #if wxUSE_LIBJPEG
+                    wxImage::AddHandler( new wxJPEGHandler );
+                #endif
+                    break;
+                case wxBITMAP_TYPE_TIF:
+                #if wxUSE_LIBTIFF
+                    wxImage::AddHandler( new wxTIFFHandler );
+                #endif
+                    break;
+                case wxBITMAP_TYPE_GIF:
+                #if wxUSE_GIF
+                    wxImage::AddHandler( new wxGIFHandler );
+                #endif
+                    break;
+                case wxBITMAP_TYPE_PNM:
+                #if wxUSE_PNM
+                    wxImage::AddHandler( new wxPNMHandler );
+                #endif
+                    break;
+                case wxBITMAP_TYPE_PCX:
+                #if wxUSE_PCX
+                    wxImage::AddHandler( new wxPCXHandler );
+                #endif
+                    break;
+                case wxBITMAP_TYPE_IFF:
+                #if wxUSE_IFF
+                    wxImage::AddHandler( new wxIFFHandler );
+                #endif
+                    break;
+                case wxBITMAP_TYPE_ICO:
+                #if wxUSE_ICO_CUR
+                    wxImage::AddHandler( new wxICOHandler );
+                #endif
+                    break;
+                case wxBITMAP_TYPE_CUR:
+                #if wxUSE_ICO_CUR
+                    wxImage::AddHandler( new wxCURHandler );
+                #endif
+                    break;
+                case wxBITMAP_TYPE_ANI:
+                #if wxUSE_ICO_CUR
+                    wxImage::AddHandler( new wxANIHandler );
+                #endif
+                    break;
+                case wxBITMAP_TYPE_TGA:
+                #if wxUSE_TGA
+                    wxImage::AddHandler( new wxTGAHandler );
+                #endif
+                    break;
+                case wxBITMAP_TYPE_XPM:
+                #if wxUSE_XPM
+                    wxImage::AddHandler( new wxXPMHandler );
+                #endif
+                    break;
+                default:
+                    break;
+            }
+        }
+    }
+    else
+    {
+    #if wxUSE_LIBPNG
+        wxImage::AddHandler( new wxPNGHandler );
+    #endif
+    #if wxUSE_LIBJPEG
+        wxImage::AddHandler( new wxJPEGHandler );
+    #endif
+    #if wxUSE_LIBTIFF
+        wxImage::AddHandler( new wxTIFFHandler );
+    #endif
+    #if wxUSE_GIF
+        wxImage::AddHandler( new wxGIFHandler );
+    #endif
+    #if wxUSE_PNM
+        wxImage::AddHandler( new wxPNMHandler );
+    #endif
+    #if wxUSE_PCX
+        wxImage::AddHandler( new wxPCXHandler );
+    #endif
+    #if wxUSE_IFF
+        wxImage::AddHandler( new wxIFFHandler );
+    #endif
+    #if wxUSE_ICO_CUR
+        wxImage::AddHandler( new wxICOHandler );
+        wxImage::AddHandler( new wxCURHandler );
+        wxImage::AddHandler( new wxANIHandler );
+    #endif
+    #if wxUSE_TGA
+        wxImage::AddHandler( new wxTGAHandler );
+    #endif
+    #if wxUSE_XPM
+        wxImage::AddHandler( new wxXPMHandler );
+    #endif
+    }
 }
 
 #endif // wxUSE_IMAGE
