diff -uNr wxWidgets-orig/src/osx/cocoa/window.mm wxWidgets-mine/src/osx/cocoa/window.mm
--- wxWidgets-orig/src/osx/cocoa/window.mm	2021-04-13 00:23:58.000000000 +0200
+++ wxWidgets-mine/src/osx/cocoa/window.mm	2021-04-28 14:09:45.000000000 +0200
@@ -2602,11 +2602,9 @@
     [m_osxView setHidden:(visible ? NO:YES)];
     
     // trigger redraw upon shown for layer-backed views
-#if __MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_14
     if ( WX_IS_MACOS_AVAILABLE(10, 14 ) )
         if( !m_osxView.isHiddenOrHasHiddenAncestor )
             SetNeedsDisplay(NULL);
-#endif
 }
 
 double wxWidgetCocoaImpl::GetContentScaleFactor() const
@@ -3060,7 +3058,6 @@
     }
 }
 
-#if __MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_14
 namespace
 {
     
@@ -3094,8 +3091,6 @@
 }
 
 }
-#endif
-
 
 void wxWidgetCocoaImpl::SetNeedsDisplay( const wxRect* where )
 {
@@ -3109,7 +3104,6 @@
     // Layer-backed views (which are all in Mojave's Dark Mode) may not have
     // their children implicitly redrawn with the parent. For compatibility,
     // do it manually here:
-#if __MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_14
     if ( WX_IS_MACOS_AVAILABLE(10, 14 ) )
     {
         if ( where )
@@ -3117,7 +3111,6 @@
         else
             SetSubviewsNeedDisplay(m_osxView);
     }
-#endif
 }
 
 bool wxWidgetCocoaImpl::GetNeedsDisplay() const
