diff -ur ../wxWidgets/src/gtk/bitmap.cpp ./src/gtk/bitmap.cpp
--- ../wxWidgets/src/gtk/bitmap.cpp	2004-08-22 11:32:55.000000000 -0700
+++ ./src/gtk/bitmap.cpp	2004-10-31 23:55:58.000000000 -0800
@@ -16,8 +16,8 @@
 
 #include "wx/defs.h"
 
-#include "wx/palette.h"
 #include "wx/bitmap.h"
+#include "wx/palette.h"
 #include "wx/icon.h"
 #include "wx/filefn.h"
 #include "wx/image.h"
diff -ur ../wxWidgets/src/gtk/colour.cpp ./src/gtk/colour.cpp
--- ../wxWidgets/src/gtk/colour.cpp	2004-06-04 08:57:04.000000000 -0700
+++ ./src/gtk/colour.cpp	2004-10-31 23:22:36.000000000 -0800
@@ -15,8 +15,8 @@
 // For compilers that support precompilation, includes "wx.h".
 #include "wx/wxprec.h"
 
-#include "wx/gdicmn.h"
 #include "wx/colour.h"
+#include "wx/gdicmn.h"
 #include "wx/gtk/private.h"
 
 #include <gdk/gdk.h>
diff -ur ../wxWidgets/src/gtk/evtloop.cpp ./src/gtk/evtloop.cpp
--- ../wxWidgets/src/gtk/evtloop.cpp	2004-09-22 18:37:36.000000000 -0700
+++ ./src/gtk/evtloop.cpp	2004-10-31 23:41:29.000000000 -0800
@@ -28,11 +28,8 @@
     #pragma hdrstop
 #endif
 
-#ifndef WX_PRECOMP
-#endif //WX_PRECOMP
-
-#include "wx/app.h"
 #include "wx/evtloop.h"
+#include "wx/app.h"
 
 #include <gtk/gtk.h>
 
diff -ur ../wxWidgets/src/gtk/fontdlg.cpp ./src/gtk/fontdlg.cpp
--- ../wxWidgets/src/gtk/fontdlg.cpp	2004-06-12 05:36:26.000000000 -0700
+++ ./src/gtk/fontdlg.cpp	2004-10-31 23:25:32.000000000 -0800
@@ -20,8 +20,8 @@
 
 #ifndef __WXGPE__
 
-#include "wx/fontutil.h"
 #include "wx/fontdlg.h"
+#include "wx/fontutil.h"
 #include "wx/utils.h"
 #include "wx/intl.h"
 #include "wx/debug.h"
diff -ur ../wxWidgets/src/gtk/frame.cpp ./src/gtk/frame.cpp
--- ../wxWidgets/src/gtk/frame.cpp	2004-09-21 04:34:38.000000000 -0700
+++ ./src/gtk/frame.cpp	2004-10-31 23:49:44.000000000 -0800
@@ -24,6 +24,7 @@
 
 #include "wx/defs.h"
 
+#include "wx/frame.h"
 #include "wx/dialog.h"
 #include "wx/control.h"
 #include "wx/app.h"
diff -ur ../wxWidgets/src/gtk/menu.cpp ./src/gtk/menu.cpp
--- ../wxWidgets/src/gtk/menu.cpp	2004-10-20 13:14:38.000000000 -0700
+++ ./src/gtk/menu.cpp	2004-10-31 23:27:40.000000000 -0800
@@ -15,11 +15,11 @@
 // For compilers that support precompilation, includes "wx.h".
 #include "wx/wxprec.h"
 
+#include "wx/menu.h"
 #include "wx/log.h"
 #include "wx/intl.h"
 #include "wx/app.h"
 #include "wx/bitmap.h"
-#include "wx/menu.h"
 
 #if wxUSE_ACCEL
     #include "wx/accel.h"
diff -ur ../wxWidgets/src/gtk/msgdlg.cpp ./src/gtk/msgdlg.cpp
--- ../wxWidgets/src/gtk/msgdlg.cpp	2004-06-20 10:19:59.000000000 -0700
+++ ./src/gtk/msgdlg.cpp	2004-10-31 23:28:06.000000000 -0800
@@ -22,10 +22,10 @@
 
 #if wxUSE_MSGDLG && defined(__WXGTK20__) && !defined(__WXGPE__)
 
+#include "wx/msgdlg.h"
 #include "wx/gtk/private.h"
 #include <gtk/gtk.h>
 
-#include "wx/msgdlg.h"
 #include "wx/intl.h"
 
 IMPLEMENT_CLASS(wxMessageDialog, wxDialog)
diff -ur ../wxWidgets/src/gtk/region.cpp ./src/gtk/region.cpp
--- ../wxWidgets/src/gtk/region.cpp	2004-05-23 13:52:21.000000000 -0700
+++ ./src/gtk/region.cpp	2004-10-31 23:29:05.000000000 -0800
@@ -23,10 +23,8 @@
 // For compilers that support precompilation, includes "wx.h".
 #include "wx/wxprec.h"
 
-#include "wx/log.h"
-
 #include "wx/region.h"
-
+#include "wx/log.h"
 #include "wx/gtk/private.h"
 
 #ifndef __WXGTK20__
diff -ur ../wxWidgets/src/gtk/renderer.cpp ./src/gtk/renderer.cpp
--- ../wxWidgets/src/gtk/renderer.cpp	2004-08-30 09:42:36.000000000 -0700
+++ ./src/gtk/renderer.cpp	2004-10-31 23:30:42.000000000 -0800
@@ -24,15 +24,12 @@
     #pragma hdrstop
 #endif
 
-#ifndef WX_PRECOMP
-#endif // WX_PRECOMP
-
+#include "wx/renderer.h"
 #include <gtk/gtk.h>
 #include "wx/gtk/win_gtk.h"
 
 #include "wx/window.h"
 #include "wx/dc.h"
-#include "wx/renderer.h"
 
 #ifdef __WXGTK20__
     #include "wx/settings.h"
diff -ur ../wxWidgets/src/gtk/scrolwin.cpp ./src/gtk/scrolwin.cpp
--- ../wxWidgets/src/gtk/scrolwin.cpp	2004-07-20 09:30:22.000000000 -0700
+++ ./src/gtk/scrolwin.cpp	2004-10-31 23:31:12.000000000 -0800
@@ -28,10 +28,9 @@
     #pragma hdrstop
 #endif
 
+#include "wx/scrolwin.h"
 #include "wx/utils.h"
 #include "wx/dcclient.h"
-
-#include "wx/scrolwin.h"
 #include "wx/panel.h"
 #include "wx/sizer.h"
 
diff -ur ../wxWidgets/src/gtk/taskbar.cpp ./src/gtk/taskbar.cpp
--- ../wxWidgets/src/gtk/taskbar.cpp	2004-06-02 15:38:13.000000000 -0700
+++ ./src/gtk/taskbar.cpp	2004-10-31 23:32:21.000000000 -0800
@@ -16,6 +16,7 @@
 // For compilers that support precompilation, includes "wx.h".
 #include "wx/wxprec.h"
 
+#include "wx/gtk/taskbarpriv.h"
 #include "wx/log.h"
 #include "wx/frame.h"
 
@@ -25,7 +26,6 @@
 #include <gtk/gtkversion.h>
 #if GTK_CHECK_VERSION(2, 1, 0)
 
-#include "wx/gtk/taskbarpriv.h"
 #include "eggtrayicon.h"
 
 wxTaskBarIconAreaBase::wxTaskBarIconAreaBase()
diff -ur ../wxWidgets/src/gtk/threadno.cpp ./src/gtk/threadno.cpp
--- ../wxWidgets/src/gtk/threadno.cpp	2004-05-23 13:52:21.000000000 -0700
+++ ./src/gtk/threadno.cpp	2004-10-31 23:32:59.000000000 -0800
@@ -16,9 +16,9 @@
 // For compilers that support precompilation, includes "wx.h".
 #include "wx/wxprec.h"
 
+#include "wx/thread.h"
 #include "wx/wx.h"
 #include "wx/module.h"
-#include "wx/thread.h"
 #include "wx/log.h"
 
 wxMutex::wxMutex()
diff -ur ../wxWidgets/src/gtk/threadsgi.cpp ./src/gtk/threadsgi.cpp
--- ../wxWidgets/src/gtk/threadsgi.cpp	2004-05-23 13:52:21.000000000 -0700
+++ ./src/gtk/threadsgi.cpp	2004-10-31 23:33:24.000000000 -0800
@@ -15,13 +15,13 @@
 // For compilers that support precompilation, includes "wx.h".
 #include "wx/wxprec.h"
 
+#include "wx/thread.h"
 #include <stdio.h>
 #include <unistd.h>
 
 #include <signal.h>
 #include <sys/wait.h>
 #include <sys/prctl.h>
-#include "wx/thread.h"
 #include "wx/module.h"
 #include "wx/utils.h"
 #include "wx/log.h"
diff -ur ../wxWidgets/src/gtk/tooltip.cpp ./src/gtk/tooltip.cpp
--- ../wxWidgets/src/gtk/tooltip.cpp	2004-06-03 11:40:13.000000000 -0700
+++ ./src/gtk/tooltip.cpp	2004-10-31 23:33:54.000000000 -0800
@@ -18,8 +18,8 @@
 
 #if wxUSE_TOOLTIPS
 
-#include "wx/window.h"
 #include "wx/tooltip.h"
+#include "wx/window.h"
 
 #include "wx/gtk/private.h"
 
diff -ur ../wxWidgets/src/gtk/toplevel.cpp ./src/gtk/toplevel.cpp
--- ../wxWidgets/src/gtk/toplevel.cpp	2004-10-27 18:59:08.000000000 -0700
+++ ./src/gtk/toplevel.cpp	2004-10-31 23:51:34.000000000 -0800
@@ -28,6 +28,7 @@
 
 #include "wx/defs.h"
 
+#include "wx/toplevel.h"
 #include "wx/log.h"
 #include "wx/dialog.h"
 #include "wx/control.h"

 	  	 
