--- wxWidgets-3.0.0/utils/helpview/src/helpview.cpp.original	2013-11-11 21:10:39.000000000 +0800
+++ wxWidgets-3.0.0/utils/helpview/src/helpview.cpp	2014-01-16 16:27:28.000000000 +0800
@@ -192,6 +192,7 @@
     wxConfig::Get(); // create an instance
 
     m_helpController = new wxHtmlHelpController( istyle );
+    m_helpController->SetShouldPreventAppExit(true);
 
     if ( !hasWindowName )
     {
