--- C:/wx2/wxPython-src-2.8.10.1/src/html/helpwnd.cpp	Tue May 11 15:50:56 2010
+++ C:/wx2/wxPython-src-2.8.10.1/src/html/helpwnd.fixed.cpp	Tue May 11 15:51:11 2010
@@ -119,9 +119,7 @@
     {
         const wxMouseEvent *e = ev.GetLinkInfo().GetEvent();
         if (e == NULL || e->LeftUp())
-        // skip the event so that normal processing (i.e. following the link)
-        // is done:
-        ev.Skip();
+			LoadPage(ev.GetLinkInfo().GetHref());
 		m_Window->NotifyPageChanged();
     }
 
