diff --git a/samples/minimal/minimal.cpp b/samples/minimal/minimal.cpp
index 65da9f35b6..de39b941ec 100644
--- a/samples/minimal/minimal.cpp
+++ b/samples/minimal/minimal.cpp
@@ -29,6 +29,8 @@
     #include "wx/wx.h"
 #endif
 
+#include "wx/grid.h"
+
 // ----------------------------------------------------------------------------
 // resources
 // ----------------------------------------------------------------------------
@@ -66,8 +68,11 @@ public:
     // event handlers (these functions should _not_ be virtual)
     void OnQuit(wxCommandEvent& event);
     void OnAbout(wxCommandEvent& event);
+    void OnGrid(wxCommandEvent& event);
 
 private:
+    wxGrid* m_grid;
+
     // any class wishing to process wxWidgets events must use this macro
     wxDECLARE_EVENT_TABLE();
 };
@@ -82,6 +87,8 @@ enum
     // menu items
     Minimal_Quit = wxID_EXIT,
 
+    Minimal_Grid,
+
     // it is important for the id corresponding to the "About" command to have
     // this standard value as otherwise it won't be handled properly under Mac
     // (where it is special and put into the "Apple" menu)
@@ -98,6 +105,7 @@ enum
 wxBEGIN_EVENT_TABLE(MyFrame, wxFrame)
     EVT_MENU(Minimal_Quit,  MyFrame::OnQuit)
     EVT_MENU(Minimal_About, MyFrame::OnAbout)
+    EVT_MENU(Minimal_Grid,  MyFrame::OnGrid)
 wxEND_EVENT_TABLE()
 
 // Create a new application object: this macro will allow wxWidgets to create
@@ -155,6 +163,7 @@ MyFrame::MyFrame(const wxString& title)
     wxMenu *helpMenu = new wxMenu;
     helpMenu->Append(Minimal_About, "&About\tF1", "Show about dialog");
 
+    fileMenu->Append(Minimal_Grid, "Grid bug\tCtrl-N", "Trigger grid bug");
     fileMenu->Append(Minimal_Quit, "E&xit\tAlt-X", "Quit this program");
 
     // now append the freshly created menu to the menu bar...
@@ -164,6 +173,10 @@ MyFrame::MyFrame(const wxString& title)
 
     // ... and attach this menu bar to the frame
     SetMenuBar(menuBar);
+
+    m_grid = new wxGrid(this, wxID_ANY);
+    m_grid->CreateGrid(10, 12);
+
 #else // !wxUSE_MENUBAR
     // If menus are not available add a button to access the about box
     wxSizer* sizer = new wxBoxSizer(wxHORIZONTAL);
@@ -204,3 +217,21 @@ void MyFrame::OnAbout(wxCommandEvent& WXUNUSED(event))
                  wxOK | wxICON_INFORMATION,
                  this);
 }
+
+void MyFrame::OnGrid(wxCommandEvent& WXUNUSED(event))
+{
+    static bool show = false;
+
+    if ( show )
+    {
+        m_grid->ShowRow(9);
+        m_grid->ShowCol(11);
+    }
+    else // Hide the last row and column.
+    {
+        m_grid->HideRow(9);
+        m_grid->HideCol(11);
+    }
+
+    show = !show;
+}
