--- scrollsub.cpp	2012-03-28 19:42:52.000000000 +0400
+++ new_scrollsub.cpp	2012-03-29 16:26:46.000000000 +0400
@@ -30,6 +30,7 @@
 class MyScrolledWindow;
 class MyCanvas;
 class MyApp;
+class DerivedMyScrolledWindow;
 
 // MyScrolledWindow
 
@@ -48,6 +49,29 @@
     DECLARE_EVENT_TABLE()
 };
 
+// DerivedMyScrolledWindow
+
+class DerivedMyScrolledWindow: public MyScrolledWindow
+{
+public:
+    DerivedMyScrolledWindow(){};
+    DerivedMyScrolledWindow( wxWindow *parent, wxWindowID id, const wxPoint &pos, const wxSize &size );
+    ~DerivedMyScrolledWindow(){};
+    void OnPaint( wxPaintEvent &event );
+    void OnSize( wxSizeEvent &event );
+    void OnScroll(wxScrollWinEvent &event);
+    void SetGroup();
+private:
+
+    wxWindow *m_parentOwner;
+    wxWindow *m_wxWindow;
+    wxButton *m_wxButton;
+    wxSize sWinSize;
+
+    DECLARE_DYNAMIC_CLASS(DerivedMyScrolledWindow)
+    DECLARE_EVENT_TABLE()
+};
+
 // MyTopLabels
 
 class MyTopLabels: public wxWindow
@@ -113,8 +137,9 @@
     void OnAbout( wxCommandEvent &event );
     void OnQuit( wxCommandEvent &event );
     void OnFullScreen( wxCommandEvent &event );
+    void OnButtonGroup( wxCommandEvent& event );
 
-    wxScrolledWindow   *m_scrolled;
+    DerivedMyScrolledWindow  *m_scrolled;
 #if wxUSE_LOG
     wxTextCtrl         *m_log;
 #endif // wxUSE_LOG
@@ -206,6 +231,62 @@
 */
 }
 
+// DerivedMyScrolledWindow
+
+IMPLEMENT_DYNAMIC_CLASS(DerivedMyScrolledWindow, MyScrolledWindow)
+
+BEGIN_EVENT_TABLE(DerivedMyScrolledWindow, MyScrolledWindow)
+  EVT_PAINT(      DerivedMyScrolledWindow::OnPaint)
+  EVT_SIZE(       DerivedMyScrolledWindow::OnSize)
+  EVT_SCROLLWIN(DerivedMyScrolledWindow::OnScroll)
+END_EVENT_TABLE()
+
+DerivedMyScrolledWindow::DerivedMyScrolledWindow( wxWindow *parent, wxWindowID id, const wxPoint &pos, const wxSize &size )
+        : MyScrolledWindow( parent, id, pos, size )
+{
+
+    m_parentOwner = parent;
+
+    sWinSize = m_parentOwner->GetClientSize();
+
+    m_wxWindow = new wxWindow( this, wxID_ANY, wxPoint(0, 25), wxSize(60, sWinSize.GetHeight()), wxCLIP_CHILDREN|wxNO_BORDER );
+
+    m_wxWindow->SetClientSize(60, sWinSize.GetHeight());
+
+    m_wxWindow->SetVirtualSize(60, this->GetScrollRange(wxVERTICAL));
+
+}
+
+void DerivedMyScrolledWindow::OnSize( wxSizeEvent &WXUNUSED(event) ) //wxScrolledWindow
+{
+    Layout();
+
+    AdjustScrollbars();
+}
+
+void DerivedMyScrolledWindow::OnPaint( wxPaintEvent &WXUNUSED(event) ) //wxScrolledWindow
+{
+   wxPaintDC dc( m_wxWindow );
+}
+
+void DerivedMyScrolledWindow::OnScroll(wxScrollWinEvent& event)
+{
+    int scrlpos_new = this->GetScrollPos(wxVERTICAL);
+
+    int scrlpos_old =  m_wxWindow->GetScrollPos(wxVERTICAL);
+
+    int delta = scrlpos_old - scrlpos_new ;
+
+    m_wxWindow->ScrollWindow(0, delta);
+
+    event.Skip();
+}
+
+void DerivedMyScrolledWindow::SetGroup()
+{
+     m_wxButton = new wxButton(m_wxWindow, wxID_ANY, wxT("Group"), wxPoint(5, 40), wxSize(50, 25));
+}
+
 // MyTopLabels
 
 IMPLEMENT_DYNAMIC_CLASS(MyTopLabels,wxWindow)
@@ -379,6 +460,7 @@
 const int ID_QUIT  = wxID_EXIT;
 const int ID_FULL  = 109;
 const int ID_ABOUT = wxID_ABOUT;
+const int ID_BUTTON_GROUP = 110;
 
 IMPLEMENT_DYNAMIC_CLASS( MyFrame, wxFrame )
 
@@ -386,6 +468,7 @@
   EVT_MENU    (ID_ABOUT, MyFrame::OnAbout)
   EVT_MENU    (ID_QUIT,  MyFrame::OnQuit)
   EVT_MENU    (ID_FULL,  MyFrame::OnFullScreen)
+  EVT_BUTTON  (ID_BUTTON_GROUP,   MyFrame::OnButtonGroup)
 END_EVENT_TABLE()
 
 MyFrame::MyFrame()
@@ -408,9 +491,18 @@
     SetStatusWidths( 2, widths );
 #endif // wxUSE_STATUSBAR
 
-    m_scrolled = new MyScrolledWindow( this, wxID_ANY, wxDefaultPosition, wxSize(100,100) );
+    m_scrolled = new DerivedMyScrolledWindow( this, wxID_ANY, wxDefaultPosition, wxSize(100,100) );
     m_scrolled->SetScrollbars( 10, 10, 50, 50 );
 
+    wxToolBar* m_toolBar1 = this->CreateToolBar( wxTB_HORIZONTAL, wxID_ANY );
+	wxButton* m_button1 = new wxButton( m_toolBar1, ID_BUTTON_GROUP, wxT("Set Group"), wxDefaultPosition, wxDefaultSize, 0 );
+	m_toolBar1->AddControl( m_button1 );
+
+	wxStaticText* m_staticText2 = new wxStaticText( m_toolBar1, wxID_ANY, wxT("  <-- Click this button before scrolling"), wxDefaultPosition, wxDefaultSize, 0 );
+	m_staticText2->Wrap( -1 );
+	m_toolBar1->AddControl( m_staticText2 );
+	m_toolBar1->Realize();
+
 #if wxUSE_LOG
     m_log = new wxTextCtrl( this, wxID_ANY, _T("This is the log window.\n"), wxPoint(0,0), wxSize(100,100), wxTE_MULTILINE );
     wxLog *old_log = wxLog::SetActiveTarget( new wxLogTextCtrl( m_log ) );
@@ -444,6 +536,12 @@
                       _T("About wxScroll II Demo"), wxICON_INFORMATION | wxOK );
 }
 
+void MyFrame::OnButtonGroup( wxCommandEvent& event )
+{
+    m_scrolled->SetGroup();
+    event.Skip();
+}
+
 //-----------------------------------------------------------------------------
 // MyApp
 //-----------------------------------------------------------------------------
@@ -455,4 +553,3 @@
 
     return true;
 }
-
