Index: src/msw/thread.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/thread.cpp,v
retrieving revision 1.63
diff -b -u -2 -r1.63 thread.cpp
--- src/msw/thread.cpp	2003/05/03 15:40:41	1.63
+++ src/msw/thread.cpp	2003/05/07 00:40:39
@@ -675,6 +675,4 @@
     }
 
-    m_state = STATE_NEW;
-
     return TRUE;
 }
Index: src/unix/threadpsx.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/unix/threadpsx.cpp,v
retrieving revision 1.66
diff -b -u -2 -r1.66 threadpsx.cpp
--- src/unix/threadpsx.cpp	2003/05/06 08:52:14	1.66
+++ src/unix/threadpsx.cpp	2003/05/07 00:40:45
@@ -1184,6 +1184,4 @@
     }
 
-    m_internal->SetState( STATE_NEW );
-
     return wxTHREAD_NO_ERROR;
 }

 	  	 
