diff --git a/include/wx/propgrid/manager.h b/include/wx/propgrid/manager.h
index 833254b..ce4aca0 100644
--- a/include/wx/propgrid/manager.h
+++ b/include/wx/propgrid/manager.h
@@ -484,12 +484,6 @@ public:
     bool IsAnyModified() const;
 
     /**
-        Returns true if updating is frozen (ie Freeze() called but not yet
-        Thaw() ).
-     */
-    bool IsFrozen() const { return m_pPropGrid->m_frozen > 0; }
-
-    /**
         Returns true if any property on given page has been modified by the
         user.
      */
@@ -635,9 +629,6 @@ public:
     //
 
     void SetId( wxWindowID winid );
-
-    virtual void Freeze();
-    virtual void Thaw();
     virtual void SetExtraStyle ( long exStyle );
     virtual bool SetFont ( const wxFont& font );
     virtual void SetWindowStyleFlag ( long style );
@@ -646,6 +637,9 @@ public:
 protected:
     virtual wxSize DoGetBestSize() const;
 
+    virtual void DoFreeze();
+    virtual void DoThaw();
+
     //
     // Event handlers
     //
diff --git a/src/propgrid/manager.cpp b/src/propgrid/manager.cpp
index ed0f510..0a9fcd1 100644
--- a/src/propgrid/manager.cpp
+++ b/src/propgrid/manager.cpp
@@ -679,17 +679,17 @@ void wxPropertyGridManager::SetExtraStyle( long exStyle )
 
 // -----------------------------------------------------------------------
 
-void wxPropertyGridManager::Freeze()
+void wxPropertyGridManager::DoFreeze()
 {
     m_pPropGrid->Freeze();
-    wxWindow::Freeze();
+    wxWindow::DoFreeze();
 }
 
 // -----------------------------------------------------------------------
 
-void wxPropertyGridManager::Thaw()
+void wxPropertyGridManager::DoThaw()
 {
-    wxWindow::Thaw();
+    wxWindow::DoThaw();
     m_pPropGrid->Thaw();
 }
 
