Index: include/wx/aui/auibook.h
===================================================================
--- include/wx/aui/auibook.h	(revision 70614)
+++ include/wx/aui/auibook.h	(working copy)
@@ -584,7 +584,7 @@
     // virtual bool AcceptsFocus() const { return false; }
 
     // Redo sizing after thawing
-    virtual void Thaw();
+    virtual void DoThaw();
 
     //wxBookCtrlBase functions
 
@@ -598,7 +598,7 @@
 
     virtual int ChangeSelection(size_t n);
 
-    virtual bool AddPage(wxWindow *page, const wxString &text, bool select, 
+    virtual bool AddPage(wxWindow *page, const wxString &text, bool select,
                          int imageId);
     virtual bool DeleteAllPages();
     virtual bool InsertPage(size_t index, wxWindow *page, const wxString &text,
Index: src/aui/auibook.cpp
===================================================================
--- src/aui/auibook.cpp	(revision 71073)
+++ src/aui/auibook.cpp	(working copy)
@@ -3238,11 +3238,11 @@
         return false;
 }
 
-void wxAuiNotebook::Thaw()
+void wxAuiNotebook::DoThaw()
 {
     DoSizing();
 
-    wxControl::Thaw();
+    wxControl::DoThaw();
 }
 
 void wxAuiNotebook::SetPageSize (const wxSize& WXUNUSED(size))
@@ -3279,10 +3279,10 @@
     return DoModifySelection(n, false);
 }
 
-bool wxAuiNotebook::AddPage(wxWindow *page, const wxString &text, bool select, 
+bool wxAuiNotebook::AddPage(wxWindow *page, const wxString &text, bool select,
                             int imageId)
 {
-    if(HasImageList()) 
+    if(HasImageList())
     {
         return AddPage(page, text, select, GetImageList()->GetBitmap(imageId));
     }
@@ -3302,13 +3302,13 @@
     return true;
 }
 
-bool wxAuiNotebook::InsertPage(size_t index, wxWindow *page, 
-                               const wxString &text, bool select, 
+bool wxAuiNotebook::InsertPage(size_t index, wxWindow *page,
+                               const wxString &text, bool select,
                                int imageId)
 {
     if(HasImageList())
     {
-        return InsertPage(index, page, text, select, 
+        return InsertPage(index, page, text, select,
                           GetImageList()->GetBitmap(imageId));
     }
     else
