Index: src/msw/timer.cpp
===================================================================
--- src/msw/timer.cpp	(wersja 62322)
+++ src/msw/timer.cpp	(kopia robocza)
@@ -107,7 +107,7 @@
 
     m_id = ::SetTimer(
         wxTimerHiddenWindowModule::GetHWND(),  // window to send the messages to
-        GetId(),                               // timer ID
+        GetId() ? GetId() : -1,                // timer ID, can't be equal 0
         (UINT)m_milli,                         // delay
         NULL                                   // timer proc.  Not used since we pass hwnd
         );
