diff -rbu2 wxGTK-2.4.1-original/filedlg.cpp wxGTK-2.4.1-patch/filedlg.cpp
--- wxGTK-2.4.1-original/filedlg.cpp	Mon Oct 09 01:35:10 2000
+++ wxGTK-2.4.1-patch/filedlg.cpp	Wed Aug 06 00:09:38 2003
@@ -90,5 +90,5 @@
         wxSplitPath(filename, &cwd, NULL, NULL);
 
-        if ( cwd != wxGetWorkingDirectory() )
+        if ( cwd != wxGetCwd() )
         {
             wxSetWorkingDirectory(cwd);
diff -rbu2 wxGTK-2.4.1-original/filedlgg.cpp wxGTK-2.4.1-patch/filedlgg.cpp
--- wxGTK-2.4.1-original/filedlgg.cpp	Sun Apr 06 20:02:46 2003
+++ wxGTK-2.4.1-patch/filedlgg.cpp	Wed Aug 06 00:09:18 2003
@@ -1450,5 +1450,5 @@
         wxSplitPath(filename, &cwd, NULL, NULL);
 
-        if ( cwd != wxGetWorkingDirectory() )
+        if ( cwd != wxGetCwd() )
         {
             wxSetWorkingDirectory(cwd);

 	  	 
