--- wxWidgets-2.7.1_orig/src/gtk/textctrl.cpp	2006-10-12 14:53:01.000000000 +0200
+++ wxWidgets-2.7.1/src/gtk/textctrl.cpp	2006-10-26 12:42:12.000000000 +0200
@@ -926,7 +926,7 @@
 bool wxTextCtrl::IsEmpty() const
 {
     if ( IsMultiLine() )
-        return gtk_text_buffer_get_char_count(m_buffer) != 0;
+        return gtk_text_buffer_get_char_count(m_buffer) == 0;
 
     return wxTextCtrlBase::IsEmpty();
 }

 	  	 
