Index: wxWidgets/src/common/strconv.cpp
===================================================================
--- wxWidgets/src/common/strconv.cpp	(revision 62567)
+++ wxWidgets/src/common/strconv.cpp	(working copy)
@@ -3410,7 +3413,6 @@
 
 #ifdef __DARWIN__
 // The xnu kernel always communicates file paths in decomposed UTF-8.
-// WARNING: Are we sure that CFString's conversion will cause decomposition?
 static wxMBConv_cf wxConvMacUTF8DObj(wxFONTENCODING_UTF8);
 #endif
 
Index: wxWidgets/src/osx/core/strconv_cf.cpp
===================================================================
--- wxWidgets/src/osx/core/strconv_cf.cpp	(revision 62567)
+++ wxWidgets/src/osx/core/strconv_cf.cpp	(working copy)
@@ -90,6 +90,11 @@
         if ( theString == NULL )
             return wxCONV_FAILED;
 
+        // Decompose the string
+        wxCFRef<CFMutableStringRef> cfMutableString( CFStringCreateMutableCopy(NULL, 0, theString) );
+        CFStringNormalize(cfMutableString, kCFStringNormalizationFormC);
+        theString = cfMutableString;
+                
         /* NOTE: The string content includes the NULL element if the source string did
          * That means we have to do nothing special because the destination will have
          * the NULL element iff the source did and the NULL element will be included
