Index: src/html/winpars.cpp
===================================================================
--- src/html/winpars.cpp	(revision 62760)
+++ src/html/winpars.cpp	(working copy)
@@ -213,6 +213,7 @@
     m_UseLink = false;
     m_Link = wxHtmlLinkInfo( wxEmptyString );
     m_LinkColor.Set(0, 0, 0xFF);
+    m_LinkUnderlined = true;
     m_ActualColor.Set(0, 0, 0);
     m_Align = wxHTML_ALIGN_LEFT;
     m_ScriptMode = wxHTML_SCRIPT_NORMAL;
Index: src/html/m_links.cpp
===================================================================
--- src/html/m_links.cpp	(revision 62760)
+++ src/html/m_links.cpp	(working copy)
@@ -69,22 +69,36 @@
         if (tag.HasParam( wxT("HREF") ))
         {
             wxHtmlLinkInfo oldlnk = m_WParser->GetLink();
-            wxColour oldclr = m_WParser->GetActualColor();
-            int oldund = m_WParser->GetFontUnderlined();
-            wxString name(tag.GetParam( wxT("HREF") )), target;
+            wxColour       oldclr = m_WParser->GetActualColor();
+            int            oldund = m_WParser->GetFontUnderlined(),
+                           newund = m_WParser->GetLinkUnderlined();
+            wxString       name(tag.GetParam( "HREF" )),
+                           target;
 
-            if (tag.HasParam( wxT("TARGET") )) target = tag.GetParam( wxT("TARGET") );
+            if (tag.HasParam( "TARGET" )) target = tag.GetParam( "TARGET" );
+
+            if (tag.HasParam( "STYLE" ) && tag.GetParam( "STYLE" ).Lower() == "text-decoration:none")
+            {
+                newund = false;
+            }
+
             m_WParser->SetActualColor(m_WParser->GetLinkColor());
             m_WParser->GetContainer()->InsertCell(new wxHtmlColourCell(m_WParser->GetLinkColor()));
-            m_WParser->SetFontUnderlined(true);
-            m_WParser->GetContainer()->InsertCell(new wxHtmlFontCell(m_WParser->CreateCurrentFont()));
+            if ( newund != oldund )
+            {
+                m_WParser->SetFontUnderlined( newund );
+                m_WParser->GetContainer()->InsertCell(new wxHtmlFontCell(m_WParser->CreateCurrentFont()));
+            }
             m_WParser->SetLink(wxHtmlLinkInfo(name, target));
 
             ParseInner(tag);
 
             m_WParser->SetLink(oldlnk);
-            m_WParser->SetFontUnderlined(oldund);
-            m_WParser->GetContainer()->InsertCell(new wxHtmlFontCell(m_WParser->CreateCurrentFont()));
+            if ( newund != oldund )
+            {
+                m_WParser->SetFontUnderlined( oldund );
+                m_WParser->GetContainer()->InsertCell(new wxHtmlFontCell(m_WParser->CreateCurrentFont()));
+            }
             m_WParser->SetActualColor(oldclr);
             m_WParser->GetContainer()->InsertCell(new wxHtmlColourCell(oldclr));
 
Index: include/wx/html/winpars.h
===================================================================
--- include/wx/html/winpars.h	(revision 62760)
+++ include/wx/html/winpars.h	(working copy)
@@ -104,34 +104,37 @@
     // container
     wxHtmlContainerCell *CloseContainer();
 
-    int GetFontSize() const {return m_FontSize;}
+    int GetFontSize() const                   { return m_FontSize; }
     void SetFontSize(int s);
-    int GetFontBold() const {return m_FontBold;}
-    void SetFontBold(int x) {m_FontBold = x;}
-    int GetFontItalic() const {return m_FontItalic;}
-    void SetFontItalic(int x) {m_FontItalic = x;}
-    int GetFontUnderlined() const {return m_FontUnderlined;}
-    void SetFontUnderlined(int x) {m_FontUnderlined = x;}
-    int GetFontFixed() const {return m_FontFixed;}
-    void SetFontFixed(int x) {m_FontFixed = x;}
-    wxString GetFontFace() const {return GetFontFixed() ? m_FontFaceFixed : m_FontFaceNormal;}
+    int GetFontBold() const                   { return m_FontBold; }
+    void SetFontBold(int x)                   { m_FontBold = x; }
+    int GetFontItalic() const                 { return m_FontItalic; }
+    void SetFontItalic(int x)                 { m_FontItalic = x; }
+    int GetFontUnderlined() const             { return m_FontUnderlined; }
+    void SetFontUnderlined(int x)             { m_FontUnderlined = x; }
+    int GetFontFixed() const                  { return m_FontFixed; }
+    void SetFontFixed(int x)                  { m_FontFixed = x; }
+    wxString GetFontFace() const              { return GetFontFixed() ? m_FontFaceFixed : m_FontFaceNormal; }
     void SetFontFace(const wxString& face);
 
-    int GetAlign() const {return m_Align;}
-    void SetAlign(int a) {m_Align = a;}
+    int GetAlign() const                      { return m_Align; }
+    void SetAlign(int a)                      { m_Align = a; }
 
-    wxHtmlScriptMode GetScriptMode() const { return m_ScriptMode; }
+    wxHtmlScriptMode GetScriptMode() const    { return m_ScriptMode; }
     void SetScriptMode(wxHtmlScriptMode mode) { m_ScriptMode = mode; }
-    long GetScriptBaseline() const { return m_ScriptBaseline; }
-    void SetScriptBaseline(long base) { m_ScriptBaseline = base; }
+    long GetScriptBaseline() const            { return m_ScriptBaseline; }
+    void SetScriptBaseline(long base)         { m_ScriptBaseline = base; }
 
-    const wxColour& GetLinkColor() const { return m_LinkColor; }
-    void SetLinkColor(const wxColour& clr) { m_LinkColor = clr; }
-    const wxColour& GetActualColor() const { return m_ActualColor; }
-    void SetActualColor(const wxColour& clr) { m_ActualColor = clr ;}
-    const wxHtmlLinkInfo& GetLink() const { return m_Link; }
+    const wxColour& GetLinkColor() const      { return m_LinkColor; }
+    void SetLinkColor(const wxColour& clr)    { m_LinkColor = clr; }
+    bool GetLinkUnderlined() const            { return m_LinkUnderlined; }
+    void SetLinkUnderlined(bool x)            { m_LinkUnderlined = x; }
+    const wxHtmlLinkInfo& GetLink() const     { return m_Link; }
     void SetLink(const wxHtmlLinkInfo& link);
 
+    const wxColour& GetActualColor() const    { return m_ActualColor; }
+    void SetActualColor(const wxColour& clr)  { m_ActualColor = clr ;}
+
     // applies current parser state (link, sub/supscript, ...) to given cell
     void ApplyStateToCell(wxHtmlCell *cell);
 
@@ -184,6 +187,7 @@
     int m_FontBold, m_FontItalic, m_FontUnderlined, m_FontFixed; // this is not true,false but 1,0, we need it for indexing
     int m_FontSize; /* -2 to +4,  0 is default */
     wxColour m_LinkColor;
+    bool m_LinkUnderlined;
     wxColour m_ActualColor;
             // basic font parameters.
     wxHtmlLinkInfo m_Link;
