diff --git i/include/wx/weakref.h w/include/wx/weakref.h
index a17c36d..e612db8 100644
--- i/include/wx/weakref.h
+++ w/include/wx/weakref.h
@@ -227,14 +227,14 @@ public:
     template <class TDerived>
     wxWeakRef(TDerived* pobj)
     {
-        Assign(pobj);
+        this->Assign(pobj);
     }
 
     // We need this copy ctor, since otherwise a default compiler (binary) copy
     // happens (if embedded as an object member).
     wxWeakRef(const wxWeakRef<T>& wr)
     {
-        Assign(wr.get());
+        this->Assign(wr.get());
     }
 
     wxWeakRef<T>& operator=(const wxWeakRef<T>& wr)
