Index: image.h
===================================================================
--- image.h	(revision 61550)
+++ image.h	(working copy)
@@ -718,8 +718,15 @@
         Can be one of the following:
         - wxIMAGE_QUALITY_NORMAL: Uses the normal default scaling method of pixel
                                   replication
+        - wxIMAGE_QUALITY_NEAR: Uses nearest neighbour resampling method
+                                (same as wxIMAGE_QUALITY_NORMAL)
+        - wxIMAGE_QUALITY_BILINEAR: Uses bilinear and box averaging resampling methods
+                                    for upsampling and downsampling respectively
+        - wxIMAGE_QUALITY_BICUBIC: Uses bicubic and box averaging resampling methods
+                                   for upsampling and downsampling respectively
         - wxIMAGE_QUALITY_HIGH: Uses bicubic and box averaging resampling methods
                                 for upsampling and downsampling respectively
+                                (same as wxIMAGE_QUALITY_BICUBIC)
 
         It should be noted that although using @c wxIMAGE_QUALITY_HIGH produces much nicer
         looking results it is a slower method. Downsampling will use the box averaging
