Index: minimal.cpp
===================================================================
--- minimal.cpp	(revision 76519)
+++ minimal.cpp	(working copy)
@@ -63,10 +63,12 @@
 public:
     // ctor(s)
     MyFrame(const wxString& title);
+	~MyFrame();
 
     // event handlers (these functions should _not_ be virtual)
     void OnQuit(wxCommandEvent& event);
     void OnAbout(wxCommandEvent& event);
+	void OnSetCursor(wxSetCursorEvent& event);
 
 private:
     // any class wishing to process wxWidgets events must use this macro
@@ -99,6 +101,7 @@
 wxBEGIN_EVENT_TABLE(MyFrame, wxFrame)
     EVT_MENU(Minimal_Quit,  MyFrame::OnQuit)
     EVT_MENU(Minimal_About, MyFrame::OnAbout)
+	EVT_SET_CURSOR(MyFrame::OnSetCursor)
 wxEND_EVENT_TABLE()
 
 // Create a new application object: this macro will allow wxWidgets to create
@@ -147,6 +150,7 @@
 {
     // set the frame icon
     SetIcon(wxICON(sample));
+	SetCursor(*wxCROSS_CURSOR);
 
 #if wxUSE_MENUS
     // create a menu bar
@@ -174,7 +178,16 @@
 #endif // wxUSE_STATUSBAR
 }
 
+MyFrame::~MyFrame()
+{
+	SetCursor(wxNullCursor);
+}
 
+void MyFrame::OnSetCursor(wxSetCursorEvent& event)
+{
+	event.Skip();
+}
+
 // event handlers
 
 void MyFrame::OnQuit(wxCommandEvent& WXUNUSED(event))
