Index: src/msw/headerctrl.cpp
===================================================================
--- src/msw/headerctrl.cpp	(revision 69240)
+++ src/msw/headerctrl.cpp	(working copy)
@@ -588,6 +588,7 @@
             break;
 
         case HDN_ITEMCHANGING:
+        case HDN_TRACK:
             if ( nmhdr->pitem && (nmhdr->pitem->mask & HDI_WIDTH) )
             {
                 // prevent the column from being shrunk beneath its min width
