Index: include/wx/dataview.h
===================================================================
--- include/wx/dataview.h	(revision 68170)
+++ include/wx/dataview.h	(working copy)
@@ -767,7 +767,8 @@
 #if wxUSE_DRAG_AND_DROP
         , m_dataObject(NULL),
         m_dataBuffer(NULL),
-        m_dataSize(0)
+        m_dataSize(0),
+        m_drag_effect(-1)
 #endif
         { }
 
@@ -785,7 +786,8 @@
         , m_dataObject(event.m_dataObject),
         m_dataFormat(event.m_dataFormat),
         m_dataBuffer(event.m_dataBuffer),
-        m_dataSize(event.m_dataSize)
+        m_dataSize(event.m_dataSize),
+        m_drag_effect(event.m_drag_effect)
 #endif
         { }
 
@@ -816,6 +818,10 @@
 
 
 #if wxUSE_DRAG_AND_DROP
+    // For both drag and drop operations
+    void SetDragEffect(int effect) { m_drag_effect = effect; }
+    int GetDragEffect() const { return m_drag_effect; }
+
     // For drag operations
     void SetDataObject( wxDataObject *obj ) { m_dataObject = obj; }
     wxDataObject *GetDataObject() const { return m_dataObject; }
@@ -847,6 +853,7 @@
     wxDataFormat        m_dataFormat;
     void*               m_dataBuffer;
     size_t              m_dataSize;
+    int                 m_drag_effect;
 #endif // wxUSE_DRAG_AND_DROP
 
 private:
Index: src/generic/datavgen.cpp
===================================================================
--- src/generic/datavgen.cpp	(revision 68170)
+++ src/generic/datavgen.cpp	(working copy)
@@ -596,6 +596,7 @@
     wxDataFormat                m_dropFormat;
     bool                        m_dropHint;
     unsigned int                m_dropHintLine;
+    int                         m_last_drag_effect;
 #endif // wxUSE_DRAG_AND_DROP
 
     // for double click logic
@@ -1420,6 +1421,8 @@
     event.SetItem( item );
     event.SetModel( model );
     event.SetDataFormat( format );
+    event.SetDragEffect( def );
+    m_last_drag_effect = def;
     if (!m_owner->HandleWindowEvent( event ))
     {
         RemoveDropHint();
@@ -1439,7 +1442,7 @@
     m_dropHintLine = row;
     RefreshRow( row );
 
-    return def;
+    return (wxDragResult)event.GetDragEffect();
 }
 
 bool wxDataViewMainWindow::OnDrop( wxDataFormat format, wxCoord x, wxCoord y )
@@ -1463,6 +1466,7 @@
     event.SetItem( item );
     event.SetModel( model );
     event.SetDataFormat( format );
+    event.SetDragEffect( m_last_drag_effect );
     if (!m_owner->HandleWindowEvent( event ))
         return false;
 
@@ -1496,6 +1500,7 @@
     event.SetDataFormat( format );
     event.SetDataSize( obj->GetSize() );
     event.SetDataBuffer( obj->GetData() );
+    event.SetDragEffect( m_last_drag_effect );
     if (!m_owner->HandleWindowEvent( event ))
         return wxDragNone;
 
@@ -3614,7 +3619,7 @@
 
             wxDataViewDropSource drag( this, drag_item_row );
             drag.SetData( *obj );
-            /* wxDragResult res = */ drag.DoDragDrop();
+            /* wxDragResult res = */ drag.DoDragDrop((event.GetDragEffect() == -1 ? wxDrag_DefaultMove : event.GetDragEffect()));
             delete obj;
         }
         return;
