 src/generic/progdlgg.cpp | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/src/generic/progdlgg.cpp b/src/generic/progdlgg.cpp
index e61586b..80a415e 100644
--- a/src/generic/progdlgg.cpp
+++ b/src/generic/progdlgg.cpp
@@ -144,13 +144,9 @@ bool wxGenericProgressDialog::Create( const wxString& title,
 {
     SetTopParent(parent);
 
-    m_parentTop = wxGetTopLevelParent(parent);
     m_pdStyle = style;
 
-    wxWindow* const
-        realParent = GetParentForModalDialog(parent, GetWindowStyle());
-
-    if (!wxDialog::Create(realParent, wxID_ANY, title, wxDefaultPosition, wxDefaultSize, GetWindowStyle()))
+    if (!wxDialog::Create(m_parentTop, wxID_ANY, title, wxDefaultPosition, wxDefaultSize, GetWindowStyle()))
         return false;
 
     SetMaximum(maximum);
