diff -ru2 -xCVS -x.#* -x*.orig -x*.rej -x*~ c:\wxWindows\src\common\cmndata.cpp c:\supp\wxWindows\src\common\cmndata.cpp
--- c:\wxWindows\src\common\cmndata.cpp	Sat Aug 17 11:29:05 2002
+++ c:\supp\wxWindows\src\common\cmndata.cpp	Sun Sep 01 23:53:31 2002
@@ -158,6 +158,5 @@
 wxFontData::wxFontData()
 {
-    // Intialize colour to black.
-    fontColour.Set(0, 0, 0);
+    fontColour = wxNullColour;
 
     showHelp = FALSE;

 	  	 
