*** dialogs.cpp	2013-11-11 18:40:39.000000000 +0530
--- dialogs.cpp.new	2014-02-28 17:02:34.258050239 +0530
*************** MyModalDialog::MyModalDialog(wxWindow *p
*** 2598,2603 ****
--- 2598,2604 ----
  
  void MyModalDialog::OnButton(wxCommandEvent& event)
  {
+     m_btnModal->MoveAfterInTabOrder(m_btnDelete);
      if ( event.GetEventObject() == m_btnDelete )
      {
          wxDELETE(m_btnModal);
