diff -ru2 -xCVS -x.#* -x*.orig -x*.rej -x*~ c:\wxWindows\src\msw\treectrl.cpp c:\supp\wxWindows\src\msw\treectrl.cpp
--- c:\wxWindows\src\msw\treectrl.cpp	Sun Jul 21 09:29:13 2002
+++ c:\supp\wxWindows\src\msw\treectrl.cpp	Thu Aug 22 19:48:05 2002
@@ -1991,4 +1991,19 @@
         switch ( nMsg )
         {
+            case WM_RBUTTONDOWN:
+                // if the item we are about to right click on
+                // is not already select, remove the entire
+                // previous selection
+                if (!::IsItemSelected(GetHwnd(), htItem))
+                {
+                    UnselectAll();
+                }
+
+                // select item and set the focus to the
+                // newly selected item
+                ::SelectItem(GetHwnd(), htItem);
+                ::SetFocus(GetHwnd(), htItem);
+                break;
+
 #if !wxUSE_CHECKBOXES_IN_MULTI_SEL_TREE
             case WM_LBUTTONDOWN:

 	  	 
