Index: wxWindows/contrib/src/stc/stc.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/contrib/src/stc/stc.cpp,v
retrieving revision 1.69
diff -b -u -2 -r1.69 stc.cpp
--- wxWindows/contrib/src/stc/stc.cpp	2004/03/06 00:09:04	1.69
+++ wxWindows/contrib/src/stc/stc.cpp	2004/03/17 05:49:23
@@ -2550,5 +2550,6 @@
             wxMemoryBuffer buffer(len);
             success = (file.Read(buffer.GetData(), len) == len);
-            contents = wxString(buffer, *wxConvCurrent);
+            if (success)
+	        contents = stc2wx(buffer, len);
 #else
             wxString buffer;

 	  	 
