Index: src/osx/cocoa/window.mm
===================================================================
--- src/osx/cocoa/window.mm	(revision 58236)
+++ src/osx/cocoa/window.mm	(working copy)
@@ -471,8 +474,10 @@
         x -= pt.x;
         y -= pt.y;
     }
+    [[m_osxView superview] setNeedsDisplayInRect:[m_osxView frame]];
     NSRect r = wxToNSRect( [m_osxView superview], wxRect(x,y,width, height) );
     [m_osxView setFrame:r];
+    [[m_osxView superview] setNeedsDisplayInRect:r];
 }
 
 void wxWidgetCocoaImpl::GetPosition( int &x, int &y ) const
