diff -bru wxWidgets/include/wx/generic/grid.h /mnt/winxp/wxWidgets/include/wx/generic/grid.h
--- wxWidgets/include/wx/generic/grid.h	2013-03-26 14:33:20.000000000 -0700
+++ /mnt/winxp/wxWidgets/include/wx/generic/grid.h	2013-04-01 16:20:42.234375000 -0700
@@ -92,7 +92,7 @@
 class WXDLLIMPEXP_FWD_ADV wxGridWindow;
 class WXDLLIMPEXP_FWD_ADV wxGridTypeRegistry;
 class WXDLLIMPEXP_FWD_ADV wxGridSelection;
-
+class WXDLLIMPEXP_FWD_ADV wxGridSizeEvent;
 class WXDLLIMPEXP_FWD_CORE wxHeaderCtrl;
 class WXDLLIMPEXP_FWD_CORE wxCheckBox;
 class WXDLLIMPEXP_FWD_CORE wxComboBox;
@@ -2122,6 +2122,7 @@
                            const wxMouseEvent& mouseEv);
 
     void OnPaint( wxPaintEvent& );
+    void OnAutoSizeColumn( wxGridSizeEvent &event );
     void OnSize( wxSizeEvent& );
     void OnKeyDown( wxKeyEvent& );
     void OnKeyUp( wxKeyEvent& );
@@ -2605,6 +2606,7 @@
 wxDECLARE_EXPORTED_EVENT( WXDLLIMPEXP_ADV, wxEVT_GRID_LABEL_RIGHT_DCLICK, wxGridEvent );
 wxDECLARE_EXPORTED_EVENT( WXDLLIMPEXP_ADV, wxEVT_GRID_ROW_SIZE, wxGridSizeEvent );
 wxDECLARE_EXPORTED_EVENT( WXDLLIMPEXP_ADV, wxEVT_GRID_COL_SIZE, wxGridSizeEvent );
+wxDECLARE_EXPORTED_EVENT( WXDLLIMPEXP_ADV, wxEVT_GRID_AUTO_COL_SIZE, wxGridSizeEvent );
 wxDECLARE_EXPORTED_EVENT( WXDLLIMPEXP_ADV, wxEVT_GRID_RANGE_SELECT, wxGridRangeSelectEvent );
 wxDECLARE_EXPORTED_EVENT( WXDLLIMPEXP_ADV, wxEVT_GRID_CELL_CHANGING, wxGridEvent );
 wxDECLARE_EXPORTED_EVENT( WXDLLIMPEXP_ADV, wxEVT_GRID_CELL_CHANGED, wxGridEvent );
@@ -2656,6 +2658,7 @@
 #define EVT_GRID_CMD_LABEL_RIGHT_DCLICK(id, fn)  wx__DECLARE_GRIDEVT(LABEL_RIGHT_DCLICK, id, fn)
 #define EVT_GRID_CMD_ROW_SIZE(id, fn)            wx__DECLARE_GRIDSIZEEVT(ROW_SIZE, id, fn)
 #define EVT_GRID_CMD_COL_SIZE(id, fn)            wx__DECLARE_GRIDSIZEEVT(COL_SIZE, id, fn)
+#define EVT_GRID_CMD_AUTO_COL_SIZE(id, fn)       wx__DECLARE_GRIDSIZEEVT(AUTO_COL_SIZE, id, fn)
 #define EVT_GRID_CMD_COL_MOVE(id, fn)            wx__DECLARE_GRIDEVT(COL_MOVE, id, fn)
 #define EVT_GRID_CMD_COL_SORT(id, fn)            wx__DECLARE_GRIDEVT(COL_SORT, id, fn)
 #define EVT_GRID_CMD_RANGE_SELECT(id, fn)        wx__DECLARE_GRIDRANGESELEVT(RANGE_SELECT, id, fn)
@@ -2680,6 +2683,7 @@
 #define EVT_GRID_LABEL_RIGHT_DCLICK(fn)  EVT_GRID_CMD_LABEL_RIGHT_DCLICK(wxID_ANY, fn)
 #define EVT_GRID_ROW_SIZE(fn)            EVT_GRID_CMD_ROW_SIZE(wxID_ANY, fn)
 #define EVT_GRID_COL_SIZE(fn)            EVT_GRID_CMD_COL_SIZE(wxID_ANY, fn)
+#define EVT_GRID_AUTO_COL_SIZE(fn)       EVT_GRID_CMD_AUTO_COL_SIZE(wxID_ANY, fn)
 #define EVT_GRID_COL_MOVE(fn)            EVT_GRID_CMD_COL_MOVE(wxID_ANY, fn)
 #define EVT_GRID_COL_SORT(fn)            EVT_GRID_CMD_COL_SORT(wxID_ANY, fn)
 #define EVT_GRID_RANGE_SELECT(fn)        EVT_GRID_CMD_RANGE_SELECT(wxID_ANY, fn)
diff -bru wxWidgets/interface/wx/grid.h /mnt/winxp/wxWidgets/interface/wx/grid.h
--- wxWidgets/interface/wx/grid.h	2013-03-26 14:34:33.000000000 -0700
+++ /mnt/winxp/wxWidgets/interface/wx/grid.h	2013-04-01 16:51:53.187500000 -0700
@@ -4707,6 +4707,10 @@
         type.
     @event{EVT_GRID_COL_SIZE(func)}
         Same as EVT_GRID_CMD_COL_SIZE() but uses `wxID_ANY` id.
+    @event{EVT_GRID_AUTO_COL_SIZE(func)}
+        This event is sent when the user double clicks the row divider. Default implementation  
+        sizes the column to fit the column label. Corresponds to @c wxEVT_GRID_AUTO_COL_SIZE event
+        This event is available since wx2.9.5
     @event{EVT_GRID_ROW_SIZE(func)}
         Same as EVT_GRID_CMD_ROW_SIZE() but uses `wxID_ANY` id.
     @endEventTable
@@ -4920,6 +4924,7 @@
 wxEventType wxEVT_GRID_LABEL_RIGHT_DCLICK;
 wxEventType wxEVT_GRID_ROW_SIZE;
 wxEventType wxEVT_GRID_COL_SIZE;
+wxEventType wxEVT_GRID_AUTO_COL_SIZE;
 wxEventType wxEVT_GRID_RANGE_SELECT;
 wxEventType wxEVT_GRID_CELL_CHANGING;
 wxEventType wxEVT_GRID_CELL_CHANGED;
diff -bru wxWidgets/samples/grid/griddemo.cpp /mnt/winxp/wxWidgets/samples/grid/griddemo.cpp
--- wxWidgets/samples/grid/griddemo.cpp	2013-03-26 14:35:56.000000000 -0700
+++ /mnt/winxp/wxWidgets/samples/grid/griddemo.cpp	2013-04-05 13:53:00.906250000 -0700
@@ -226,6 +226,7 @@
     EVT_GRID_CELL_LEFT_CLICK( GridFrame::OnCellLeftClick )
     EVT_GRID_ROW_SIZE( GridFrame::OnRowSize )
     EVT_GRID_COL_SIZE( GridFrame::OnColSize )
+    EVT_GRID_AUTO_COL_SIZE( GridFrame::OnColAutoSize )
     EVT_GRID_SELECT_CELL( GridFrame::OnSelectCell )
     EVT_GRID_RANGE_SELECT( GridFrame::OnRangeSelected )
     EVT_GRID_CELL_CHANGING( GridFrame::OnCellValueChanging )
@@ -1174,6 +1175,10 @@
     ev.Skip();
 }
 
+void GridFrame::OnColAutoSize( wxGridSizeEvent &event )
+{
+    grid->AutoSizeColumn( event.GetRowOrCol() );
+}
 
 void GridFrame::OnSelectCell( wxGridEvent& ev )
 {
diff -bru wxWidgets/samples/grid/griddemo.h /mnt/winxp/wxWidgets/samples/grid/griddemo.h
--- wxWidgets/samples/grid/griddemo.h	2013-03-26 14:35:56.000000000 -0700
+++ /mnt/winxp/wxWidgets/samples/grid/griddemo.h	2013-04-05 13:53:00.921875000 -0700
@@ -98,6 +98,7 @@
     void OnCellLeftClick( wxGridEvent& );
     void OnRowSize( wxGridSizeEvent& );
     void OnColSize( wxGridSizeEvent& );
+    void OnColAutoSize( wxGridSizeEvent& );
     void OnSelectCell( wxGridEvent& );
     void OnRangeSelected( wxGridRangeSelectEvent& );
     void OnCellValueChanging( wxGridEvent& );
diff -bru wxWidgets/src/generic/grid.cpp /mnt/winxp/wxWidgets/src/generic/grid.cpp
--- wxWidgets/src/generic/grid.cpp	2013-03-26 14:30:31.000000000 -0700
+++ /mnt/winxp/wxWidgets/src/generic/grid.cpp	2013-04-05 13:24:38.625000000 -0700
@@ -146,6 +146,7 @@
 wxDEFINE_EVENT( wxEVT_GRID_LABEL_RIGHT_DCLICK, wxGridEvent );
 wxDEFINE_EVENT( wxEVT_GRID_ROW_SIZE, wxGridSizeEvent );
 wxDEFINE_EVENT( wxEVT_GRID_COL_SIZE, wxGridSizeEvent );
+wxDEFINE_EVENT( wxEVT_GRID_AUTO_COL_SIZE, wxGridSizeEvent );
 wxDEFINE_EVENT( wxEVT_GRID_COL_MOVE, wxGridEvent );
 wxDEFINE_EVENT( wxEVT_GRID_COL_SORT, wxGridEvent );
 wxDEFINE_EVENT( wxEVT_GRID_RANGE_SELECT, wxGridRangeSelectEvent );
@@ -2152,6 +2153,7 @@
 /////////////////////////////////////////////////////////////////////
 
 BEGIN_EVENT_TABLE( wxGrid, wxScrolledWindow )
+    EVT_GRID_AUTO_COL_SIZE( wxGrid::OnAutoSizeColumn )
     EVT_PAINT( wxGrid::OnPaint )
     EVT_SIZE( wxGrid::OnSize )
     EVT_KEY_DOWN( wxGrid::OnKeyDown )
@@ -3720,7 +3722,8 @@
             // adjust column width depending on label text
             //
             // TODO: generate RESIZING event, see #10754
-            AutoSizeColLabelSize( colEdge );
+            SendGridSizeEvent(wxEVT_GRID_AUTO_COL_SIZE, -1, colEdge, event);
+//            AutoSizeColLabelSize( colEdge );
 
             SendGridSizeEvent(wxEVT_GRID_COL_SIZE, -1, colEdge, event);
 
@@ -4754,6 +4757,11 @@
     return claimed ? 1 : 0;
 }
 
+void wxGrid::OnAutoSizeColumn(wxGridSizeEvent &event)
+{
+    AutoSizeColumn( event.GetRowOrCol() );
+}
+
 void wxGrid::OnPaint( wxPaintEvent& WXUNUSED(event) )
 {
     // needed to prevent zillions of paint events on MSW
