diff -aburN ../wxWidgets_original/include/wx/gtk/print.h ./include/wx/gtk/print.h
--- ../wxWidgets_original/include/wx/gtk/print.h	2007-12-15 16:32:11.000000000 +0100
+++ ./include/wx/gtk/print.h	2007-12-15 16:37:15.000000000 +0100
@@ -334,6 +334,8 @@
 private:
     void Init(wxPrintout *printout, wxPrintout *printoutForPrinting);
 
+    int m_resolution;
+
     DECLARE_CLASS(wxGtkPrintPreview)
 };
 
diff -aburN ../wxWidgets_original/src/gtk/print.cpp ./src/gtk/print.cpp
--- ../wxWidgets_original/src/gtk/print.cpp	2007-12-15 16:28:01.000000000 +0100
+++ ./src/gtk/print.cpp	2007-12-15 16:38:14.000000000 +0100
@@ -2261,6 +2261,16 @@
                                                    wxPrintDialogData *data)
                         : wxPrintPreviewBase(printout, printoutForPrinting, data)
 {
+    // Match print quality to resolution (high = 1200dpi)
+    if (data)
+    {
+        m_resolution = data->GetPrintData().GetQuality();
+        if (m_resolution < 0)
+            m_resolution = (1 << (m_resolution+4)) *150;
+    }
+    else
+        m_resolution = 600;
+
     Init(printout, printoutForPrinting);
 }
 
@@ -2269,6 +2279,16 @@
                                                    wxPrintData *data)
                         : wxPrintPreviewBase(printout, printoutForPrinting, data)
 {
+    // Match print quality to resolution (high = 1200dpi)
+    if (data)
+    {
+        m_resolution = data->GetQuality();
+        if (m_resolution < 0)
+            m_resolution = (1 << (m_resolution+4)) *150;
+    }
+    else
+        m_resolution = 600;
+
     Init(printout, printoutForPrinting);
 }
 
@@ -2301,15 +2321,13 @@
         m_previewPrintout->SetPPIScreen( (int) ((ScreenPixels.GetWidth() * 25.4) / ScreenMM.GetWidth()),
                                          (int) ((ScreenPixels.GetHeight() * 25.4) / ScreenMM.GetHeight()) );
 
-        // TODO !!!!!!!!!!!!!!!      
-        int resolution = 600;
-        m_previewPrintout->SetPPIPrinter( resolution, resolution );
+        m_previewPrintout->SetPPIPrinter( m_resolution, m_resolution );
 
         // Get width and height in points (1/72th of an inch)
         wxSize sizeDevUnits(paper->GetSizeDeviceUnits());
 
-        sizeDevUnits.x = wxRound((double)sizeDevUnits.x * (double)resolution / 72.0);
-        sizeDevUnits.y = wxRound((double)sizeDevUnits.y * (double)resolution / 72.0);
+        sizeDevUnits.x = wxRound((double)sizeDevUnits.x * (double)m_resolution / 72.0);
+        sizeDevUnits.y = wxRound((double)sizeDevUnits.y * (double)m_resolution / 72.0);
         wxSize sizeTenthsMM(paper->GetSize());
         wxSize sizeMM(sizeTenthsMM.x / 10, sizeTenthsMM.y / 10);
 
@@ -2330,7 +2348,7 @@
         m_previewPrintout->SetPaperRectPixels(wxRect(0, 0, m_pageWidth, m_pageHeight));
 
         // At 100%, the page should look about page-size on the screen.
-        m_previewScaleX = 0.8 * 72.0 / (double)resolution;
+        m_previewScaleX = 0.8 * 72.0 / (double)m_resolution;
         m_previewScaleY = m_previewScaleX;
     }
 }

 	  	 
