diff -rbu2 ./cvs/wx2/wxwindows/contrib/src/xml/xh_listc.cpp ./wx2/contrib/src/xml/xh_listc.cpp
--- ./cvs/wx2/wxwindows/contrib/src/xml/xh_listc.cpp	Sun Jan 28 23:21:20 2001
+++ ./wx2/contrib/src/xml/xh_listc.cpp	Tue Apr 17 18:04:08 2001
@@ -20,4 +20,5 @@
 #endif
 
+#include "wx/textctrl.h"
 #include "wx/xml/xh_listc.h"
 #include "wx/listctrl.h"
diff -rbu2 ./cvs/wx2/wxwindows/contrib/src/xml/xmlres.cpp ./wx2/contrib/src/xml/xmlres.cpp
--- ./cvs/wx2/wxwindows/contrib/src/xml/xmlres.cpp	Wed Feb 21 23:51:40 2001
+++ ./wx2/contrib/src/xml/xmlres.cpp	Sun Apr  8 06:07:06 2001
@@ -562,5 +562,5 @@
     unsigned long tmp = 0; 
     
-    if (v.Length() != 7 || v[0] != wxT('#') ||
+    if (v.Length() != 7 || v.c_str()[0] != wxT('#') ||
         wxSscanf(v.c_str(), wxT("#%lX"), &tmp) != 1)
     {

 	  	 
