Index: src/msw/notebook.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/msw/notebook.cpp,v
retrieving revision 1.189
diff -u -r1.189 notebook.cpp
--- src/msw/notebook.cpp	2006/10/09 13:22:11	1.189
+++ src/msw/notebook.cpp	2006/10/10 19:29:09
@@ -911,6 +911,8 @@
     wxBitmap bmp(rc.right, rc.bottom);
     memdc.SelectObject(bmp);
 
+    memdc.SetLayoutDirection(dc.GetLayoutDirection());
+
     // if there is no special brush just use the solid background colour
 #if wxUSE_UXTHEME
     HBRUSH hbr = (HBRUSH)m_hbrBackground;
@@ -928,7 +930,10 @@
 
     MSWDefWindowProc(WM_PAINT, (WPARAM)memdc.GetHDC(), 0);
 
-    dc.Blit(0, 0, rc.right, rc.bottom, &memdc, 0, 0);
+    // For some reason in RTL mode, source offset has to be -1, otherwise the 
+    // right border (physical) remains unpainted.
+    const bool rtl = wxTheApp->GetLayoutDirection() == wxLayout_RightToLeft;
+    dc.Blit(rtl ? -1 : 0, 0, rc.right + 0, rc.bottom, &memdc, (rtl ? -1 : 0), 0);
 }
 
 #endif // USE_NOTEBOOK_ANTIFLICKER

 	  	 
