Index: include/wx/msw/mdi.h
===================================================================
--- include/wx/msw/mdi.h	(revision 71211)
+++ include/wx/msw/mdi.h	(working copy)
@@ -196,6 +196,15 @@
 
     virtual bool Show(bool show = true);
 
+    virtual void Raise()
+    {
+        if (IsIconized())
+        {
+            Restore();
+        }
+        wxMDIChildFrameBase::Raise();
+    }
+
     // Implementation only from now on
     // -------------------------------
 
