Index: include/wx/mdi.h
===================================================================
--- include/wx/mdi.h	(revision 71055)
+++ include/wx/mdi.h	(working copy)
@@ -184,7 +184,14 @@
     // Raising any frame is supposed to show it but wxFrame Raise()
     // implementation doesn't work for MDI child frames in most forms so
     // forward this to Activate() which serves the same purpose by default.
-    virtual void Raise() { Activate(); }
+    virtual void Raise()
+    {
+        if (IsIconized())
+        {
+            Restore();
+        }
+        Activate();
+    }
 
 protected:
     wxMDIParentFrame *m_mdiParent;
