--- wxWidgets-trunk/include/wx/msw/statbmp.h	2013-12-29
+++ wxWidgets-work/include/wx/msw/statbmp.h	2014-03-14
@@ -60,7 +60,7 @@
     virtual wxSize DoGetBestClientSize() const;
 
     // ctor/dtor helpers
-    void Init() { m_isIcon = true; m_image = NULL; m_currentHandle = 0; }
+    void Init() { m_isIcon = true; m_image = NULL; m_currentHandle = 0; m_isTempHandle = false; }
     void Free();
 
     // true if icon/bitmap is valid
@@ -83,6 +83,8 @@
 
     // handle used in last call to STM_SETIMAGE
     WXHANDLE m_currentHandle;
+    // flag to tell if this is a handle to the temporary bitmap created due to the de-premultiplication.
+    bool m_isTempHandle;
 
 private:
     DECLARE_DYNAMIC_CLASS(wxStaticBitmap)
--- wxWidgets-trunk/src/msw/statbmp.cpp	2014-03-14
+++ wxWidgets-work/src/msw/statbmp.cpp	2014-03-14
@@ -199,6 +199,9 @@
         // the static control made a copy and we are responsible for deleting it
         ::DeleteObject(oldHandle);
     }
+    // If bitmap is a temporary one  we are responsible for deleting it.
+    if ( m_isTempHandle )
+        ::DeleteObject(m_currentHandle);
 
     wxDELETE(m_image);
 }
@@ -278,7 +281,6 @@
 #ifdef __WIN32__
     HANDLE handle = (HANDLE)m_image->GetHandle();
 
-    AutoHBITMAP hbmpRelease;
     if ( !m_isIcon )
     {
         // wxBitmap normally stores alpha in pre-multiplied format but
@@ -292,9 +294,6 @@
             // not-premultiplied alpha values.
             handle = wxDIB(bmp.ConvertToImage(),
                            wxDIB::PixelFormat_NotPreMultiplied).Detach();
-
-            // Ensure that this temporary HBITMAP will be destroyed.
-            hbmpRelease.Init((HBITMAP)handle);
         }
     }
 
@@ -311,11 +310,13 @@
         // the static control made a copy and we are responsible for deleting it
         ::DeleteObject((HGDIOBJ) oldHandle);
     }
+    // If prevous bitmap was a temporary one we are responsible for deleting it.
+    if ( m_isTempHandle )
+        ::DeleteObject(m_currentHandle);
 
-    // Save bitmap handle only if it's not a temporary one, otherwise it's
-    // going to be destroyed right now anyhow.
-    if ( !hbmpRelease )
+    // Save bitmap handle and its flag
         m_currentHandle = (WXHANDLE)handle;
+    m_isTempHandle = handle != (HANDLE)m_image->GetHandle();
 
 #endif // Win32
 
