diff --git a/src/stc/stc.cpp b/src/stc/stc.cpp
index 79b83ca..7787856 100644
--- a/src/stc/stc.cpp
+++ b/src/stc/stc.cpp
@@ -5169,7 +5169,11 @@ void wxStyledTextCtrl::StartStyling(int start, int unused)
 // Event handlers
 
 void wxStyledTextCtrl::OnPaint(wxPaintEvent& WXUNUSED(evt)) {
+#ifdef __WXGTK3__
+    wxBufferedPaintDC dc(this);
+#else
     wxPaintDC dc(this);
+#endif
     m_swx->DoPaint(&dc, GetUpdateRegion().GetBox());
 }
 
