Index: include/wx/image.h
===================================================================
--- include/wx/image.h	(revision 69693)
+++ include/wx/image.h	(working copy)
@@ -40,6 +40,9 @@
 #define wxIMAGE_OPTION_MAX_WIDTH             wxString(wxT("MaxWidth"))
 #define wxIMAGE_OPTION_MAX_HEIGHT            wxString(wxT("MaxHeight"))
 
+#define wxIMAGE_OPTION_ORIGINAL_SIZE_WIDTH   wxString(wxT("OriginalSizeWidth"))
+#define wxIMAGE_OPTION_ORIGINAL_SIZE_HEIGHT  wxString(wxT("OriginalSizeHeight"))
+
 // constants used with wxIMAGE_OPTION_RESOLUTIONUNIT
 //
 // NB: don't change these values, they correspond to libjpeg constants
Index: interface/wx/image.h
===================================================================
--- interface/wx/image.h	(revision 69693)
+++ interface/wx/image.h	(working copy)
@@ -86,6 +86,8 @@
 #define wxIMAGE_OPTION_RESOLUTIONUNIT                   wxString(wxT("ResolutionUnit"))
 #define wxIMAGE_OPTION_MAX_WIDTH                        wxString(wxT("MaxWidth"))
 #define wxIMAGE_OPTION_MAX_HEIGHT                       wxString(wxT("MaxHeight"))
+#define wxIMAGE_OPTION_ORIGINAL_SIZE_WIDTH              wxString(wxT("OriginalSizeWidth"))
+#define wxIMAGE_OPTION_ORIGINAL_SIZE_HEIGHT             wxString(wxT("OriginalSizeHeight"))
 
 #define wxIMAGE_OPTION_BMP_FORMAT                       wxString(wxT("wxBMP_FORMAT"))
 #define wxIMAGE_OPTION_CUR_HOTSPOT_X                    wxString(wxT("HotSpotX"))
@@ -1182,6 +1184,12 @@
             handler, this is still what happens however). These options must be
             set before calling LoadFile() to have any effect.
 
+        @li @c wxIMAGE_OPTION_ORIGINAL_SIZE_WIDTH and @c wxIMAGE_OPTION_ORIGINAL_SIZE_HEIGHT:
+            These options will return the original size of the image if either
+            @c wxIMAGE_OPTION_MAX_WIDTH or @c wxIMAGE_OPTION_MAX_HEIGHT is
+            specified.
+            @since 2.9.3
+
         @li @c wxIMAGE_OPTION_QUALITY: JPEG quality used when saving. This is an
             integer in 0..100 range with 0 meaning very poor and 100 excellent
             (but very badly compressed). This option is currently ignored for
Index: samples/image/image.cpp
===================================================================
--- samples/image/image.cpp	(revision 69693)
+++ samples/image/image.cpp	(working copy)
@@ -944,10 +944,14 @@
         return;
     }
 
+    int origWidth = image.GetOptionInt( wxIMAGE_OPTION_ORIGINAL_SIZE_WIDTH );
+    int origHeight = image.GetOptionInt( wxIMAGE_OPTION_ORIGINAL_SIZE_HEIGHT );
+
     const long loadTime = sw.Time();
 
     MyImageFrame * const frame = new MyImageFrame(this, filename, image);
-    wxLogStatus(frame, "Loaded \"%s\" in %ldms", filename, loadTime);
+    wxLogStatus(frame, "Loaded \"%s\" in %ldms; original size was (%d, %d)",
+                                            filename, loadTime, origWidth, origHeight);
 #else
     wxLogError( wxT("Couldn't create file selector dialog") );
     return;
Index: src/common/image.cpp
===================================================================
--- src/common/image.cpp	(revision 69693)
+++ src/common/image.cpp	(working copy)
@@ -2450,7 +2450,17 @@
         }
 
         if ( width != widthOrig || height != heightOrig )
+        {
+            // get the original size if it was set by the image handler
+            // but also in order to restore it after Rescale
+            int widthOrigOption = GetOptionInt(wxIMAGE_OPTION_ORIGINAL_SIZE_WIDTH),
+                heightOrigOption = GetOptionInt(wxIMAGE_OPTION_ORIGINAL_SIZE_HEIGHT);
+
             Rescale(width, height, wxIMAGE_QUALITY_HIGH);
+
+            SetOption(wxIMAGE_OPTION_ORIGINAL_SIZE_WIDTH, widthOrigOption ? widthOrigOption : widthOrig);
+            SetOption(wxIMAGE_OPTION_ORIGINAL_SIZE_HEIGHT, heightOrigOption ? heightOrigOption : heightOrig);
+        }
     }
 
     // Set this after Rescale, which currently does not preserve it
Index: src/common/imagjpeg.cpp
===================================================================
--- src/common/imagjpeg.cpp	(revision 69693)
+++ src/common/imagjpeg.cpp	(working copy)
@@ -334,6 +334,13 @@
         image->SetOption(wxIMAGE_OPTION_RESOLUTIONUNIT, cinfo.density_unit);
     }
 
+    if ( cinfo.image_width != cinfo.output_width || cinfo.image_height != cinfo.output_height )
+    {
+        // save the original image size
+        image->SetOption(wxIMAGE_OPTION_ORIGINAL_SIZE_WIDTH, cinfo.image_width);
+        image->SetOption(wxIMAGE_OPTION_ORIGINAL_SIZE_HEIGHT, cinfo.image_height);
+    }
+
     jpeg_finish_decompress( &cinfo );
     jpeg_destroy_decompress( &cinfo );
     return true;
