diff --git a/samples/propgrid/propgrid.cpp b/samples/propgrid/propgrid.cpp
index 69d9c23..424c480 100644
--- a/samples/propgrid/propgrid.cpp
+++ b/samples/propgrid/propgrid.cpp
@@ -1038,6 +1038,10 @@ void FormMain::PopulateWithStandardItems ()
     // Append is ideal way to add items to wxPropertyGrid.
     pg->Append( new wxPropertyCategory(wxT("Appearance"),wxPG_LABEL) );
 
+    wxPGProperty* prop = new wxBoolProperty("Property with checkbox", wxPG_LABEL);
+    pg->Append(prop);
+    prop->SetAttribute(wxS("UseCheckbox"), true);
+
     pg->Append( new wxStringProperty(wxT("Label"),wxPG_LABEL,GetTitle()) );
     pg->Append( new wxFontProperty(wxT("Font"),wxPG_LABEL) );
     pg->SetPropertyHelpString ( wxT("Font"), wxT("Editing this will change font used in the property grid.") );
