Index: splash.h
===================================================================
--- splash.h	(revision 58645)
+++ splash.h	(working copy)
@@ -29,6 +29,24 @@
       wxYield();
     @endcode
 
+    @beginStyleTable
+    @style{wxSPLASH_CENTRE_ON_PARENT}
+           Position the splash screen in the center of the parent window.
+    @style{wxSPLASH_CENTRE_ON_SCREEN}
+           Position the splash screen in the center of the screen.
+    @style{wxSPLASH_NO_CENTRE}
+           Position the splash screen at the top to the left of the screen.
+    @style{wxSPLASH_TIMEOUT}
+           Activate the automatic closure of the splash screen after the
+           defined timeout.
+    @style{wxSPLASH_NO_TIMEOUT}
+           Do not use any timeout.
+    @style{wxSPLASH_USE_BITMAP_MASK}
+           Apply the mask associated with the bitmap (if there is one)
+           before drawing it on the screen. Useful to manage the transparent
+           areas of the bitmap.
+    @endStyleTable
+
     @library{wxadv}
     @category{managedwnd}
 */
@@ -45,6 +63,7 @@
         - wxSPLASH_NO_CENTRE
         - wxSPLASH_TIMEOUT
         - wxSPLASH_NO_TIMEOUT
+        - wxSPLASH_USE_BITMAP_MASK
 
         @a milliseconds is the timeout in milliseconds.
     */
