Index: include/wx/generic/dirctrlg.h
===================================================================
--- include/wx/generic/dirctrlg.h	(revision 72815)
+++ include/wx/generic/dirctrlg.h	(working copy)
@@ -161,6 +161,8 @@
     // If the path string has been used (we're at the leaf), done is set to true
     virtual wxTreeItemId FindChild(wxTreeItemId parentId, const wxString& path, bool& done);
 
+    wxString GetPath(wxTreeItemId itemId) const;
+
     // Resize the components of the control
     virtual void DoResize();
 
Index: interface/wx/dirctrl.h
===================================================================
--- interface/wx/dirctrl.h	(revision 72815)
+++ interface/wx/dirctrl.h	(working copy)
@@ -163,6 +163,13 @@
     virtual wxString GetPath() const;
 
     /**
+        Gets directory or filename from the tree control.
+
+        @since 2.9.5
+    */
+    wxString GetPath(wxTreeItemId itemId) const;
+
+    /**
         Fills the array @a paths with the selected directories and filenames.
     */
     virtual void GetPaths(wxArrayString& paths) const;
Index: src/generic/dirctrlg.cpp
===================================================================
--- src/generic/dirctrlg.cpp	(revision 72815)
+++ src/generic/dirctrlg.cpp	(working copy)
@@ -1038,7 +1038,13 @@
     return true;
 }
 
+wxString wxGenericDirCtrl::GetPath(wxTreeItemId itemId) const
+{
+    const wxDirItemData* data = (const wxDirItemData*) m_treeCtrl->GetItemData(itemId);
 
+    return data->m_path;
+}
+
 wxString wxGenericDirCtrl::GetPath() const
 {
     // Allow calling GetPath() in multiple selection from OnSelFilter
@@ -1050,8 +1056,7 @@
         {
             // return first string only
             wxTreeItemId treeid = items[0];
-            wxDirItemData* data = (wxDirItemData*) m_treeCtrl->GetItemData(treeid);
-            return data->m_path;
+            return GetPath(treeid);
         }
 
         return wxEmptyString;
@@ -1060,8 +1065,7 @@
     wxTreeItemId treeid = m_treeCtrl->GetSelection();
     if (treeid)
     {
-        wxDirItemData* data = (wxDirItemData*) m_treeCtrl->GetItemData(treeid);
-        return data->m_path;
+        return GetPath(treeid);
     }
     else
         return wxEmptyString;
@@ -1076,8 +1080,7 @@
     for ( unsigned n = 0; n < items.size(); n++ )
     {
         wxTreeItemId treeid = items[n];
-        wxDirItemData* data = (wxDirItemData*) m_treeCtrl->GetItemData(treeid);
-        paths.Add(data->m_path);
+        paths.push_back(GetPath(treeid));
     }
 }
 
