Index: include/wx/generic/dirctrlg.h
===================================================================
--- include/wx/generic/dirctrlg.h	(revision 72815)
+++ include/wx/generic/dirctrlg.h	(working copy)
@@ -161,6 +161,9 @@
     // If the path string has been used (we're at the leaf), done is set to true
     virtual wxTreeItemId FindChild(wxTreeItemId parentId, const wxString& path, bool& done);
 
+    // Get item data from tree control
+    const wxDirItemData* GetItemData(wxTreeItemId itemId) const;
+
     // Resize the components of the control
     virtual void DoResize();
 
@@ -190,6 +193,7 @@
 
 private:
     void PopulateNode(wxTreeItemId node);
+    wxDirItemData* GetItemData(wxTreeItemId itemId);
 
     bool            m_showHidden;
     wxTreeItemId    m_rootId;
Index: interface/wx/dirctrl.h
===================================================================
--- interface/wx/dirctrl.h	(revision 72815)
+++ interface/wx/dirctrl.h	(working copy)
@@ -178,6 +178,13 @@
     virtual wxTreeCtrl* GetTreeCtrl() const;
 
     /**
+        Get item data from tree control.
+
+        @since 2.9.5
+    */
+    const wxDirItemData* GetItemData(wxTreeItemId itemId) const;
+
+    /**
         Initializes variables.
     */
     virtual void Init();
Index: src/generic/dirctrlg.cpp
===================================================================
--- src/generic/dirctrlg.cpp	(revision 72815)
+++ src/generic/dirctrlg.cpp	(working copy)
@@ -630,6 +630,16 @@
         m_treeCtrl->SetFocus();
 }
 
+const wxDirItemData* wxGenericDirCtrl::GetItemData(wxTreeItemId itemId) const
+{
+    return (const wxDirItemData*)m_treeCtrl->GetItemData(itemId);
+}
+
+wxDirItemData* wxGenericDirCtrl::GetItemData(wxTreeItemId itemId)
+{
+    return (wxDirItemData*)m_treeCtrl->GetItemData(itemId);
+}
+
 void wxGenericDirCtrl::OnBeginEditItem(wxTreeEvent &event)
 {
     // don't rename the main entry "Sections"
@@ -666,7 +676,7 @@
     }
 
     wxTreeItemId treeid = event.GetItem();
-    wxDirItemData *data = (wxDirItemData*)m_treeCtrl->GetItemData( treeid );
+    wxDirItemData *data = GetItemData( treeid );
     wxASSERT( data );
 
     wxString new_name( wxPathOnly( data->m_path ) );
@@ -716,7 +726,7 @@
 {
     wxTreeItemId child;
 
-    wxDirItemData *data = (wxDirItemData *) m_treeCtrl->GetItemData(parentId);
+    wxDirItemData *data = GetItemData(parentId);
     if (!data->m_isExpanded)
         return;
 
@@ -731,7 +741,7 @@
 
 void wxGenericDirCtrl::PopulateNode(wxTreeItemId parentId)
 {
-    wxDirItemData *data = (wxDirItemData *) m_treeCtrl->GetItemData(parentId);
+    wxDirItemData *data = GetItemData(parentId);
 
     if (data->m_isExpanded)
         return;
@@ -922,7 +932,7 @@
     wxTreeItemId childId = m_treeCtrl->GetFirstChild(parentId, cookie);
     while (childId.IsOk())
     {
-        wxDirItemData* data = (wxDirItemData*) m_treeCtrl->GetItemData(childId);
+        wxDirItemData* data = GetItemData(childId);
 
         if (data && !data->m_path.empty())
         {
@@ -973,7 +983,7 @@
     if (!lastId.IsOk())
         return false;
 
-    wxDirItemData *data = (wxDirItemData *) m_treeCtrl->GetItemData(lastId);
+    wxDirItemData *data = GetItemData(lastId);
     if (data->m_isDir)
     {
         m_treeCtrl->Expand(lastId);
@@ -986,7 +996,7 @@
         bool selectedChild = false;
         while (childId.IsOk())
         {
-            data = (wxDirItemData*) m_treeCtrl->GetItemData(childId);
+            data = GetItemData(childId);
 
             if (data && data->m_path != wxEmptyString && !data->m_isDir)
             {
@@ -1050,7 +1060,7 @@
         {
             // return first string only
             wxTreeItemId treeid = items[0];
-            wxDirItemData* data = (wxDirItemData*) m_treeCtrl->GetItemData(treeid);
+            const wxDirItemData* data = GetItemData(treeid);
             return data->m_path;
         }
 
@@ -1060,7 +1070,7 @@
     wxTreeItemId treeid = m_treeCtrl->GetSelection();
     if (treeid)
     {
-        wxDirItemData* data = (wxDirItemData*) m_treeCtrl->GetItemData(treeid);
+        const wxDirItemData* data = GetItemData(treeid);
         return data->m_path;
     }
     else
@@ -1076,7 +1086,7 @@
     for ( unsigned n = 0; n < items.size(); n++ )
     {
         wxTreeItemId treeid = items[n];
-        wxDirItemData* data = (wxDirItemData*) m_treeCtrl->GetItemData(treeid);
+        const wxDirItemData* data = GetItemData(treeid);
         paths.Add(data->m_path);
     }
 }
@@ -1086,7 +1096,7 @@
     wxTreeItemId treeid = m_treeCtrl->GetSelection();
     if (treeid)
     {
-        wxDirItemData* data = (wxDirItemData*) m_treeCtrl->GetItemData(treeid);
+        const wxDirItemData* data = GetItemData(treeid);
         if (data->m_isDir)
             return wxEmptyString;
         else
@@ -1105,7 +1115,7 @@
     for ( unsigned n = 0; n < items.size(); n++ )
     {
         wxTreeItemId treeid = items[n];
-        wxDirItemData* data = (wxDirItemData*) m_treeCtrl->GetItemData(treeid);
+        const wxDirItemData* data = GetItemData(treeid);
         if ( !data->m_isDir )
             paths.Add(data->m_path);
     }
@@ -1159,7 +1169,7 @@
 #if 0
 void wxGenericDirCtrl::FindChildFiles(wxTreeItemId treeid, int dirFlags, wxArrayString& filenames)
 {
-    wxDirItemData *data = (wxDirItemData *) m_treeCtrl->GetItemData(treeid);
+    wxDirItemData *data = GetItemData(treeid);
 
     // This may take a longish time. Go to busy cursor
     wxBusyCursor busy;
