--- toolbar.cpp.orig	2012-01-21 18:59:58.000000000 +0200
+++ toolbar.cpp	2012-01-21 18:57:27.000000000 +0200
@@ -327,7 +327,7 @@
     wxToolBar* toolBar = CreateToolBar();
 #else
     // delete and recreate the toolbar
-    wxToolBarBase *toolBar = GetToolBar();
+    wxToolBar *toolBar = GetToolBar();
     long style = toolBar ? toolBar->GetWindowStyle() : TOOLBAR_STYLE;
 
     if (toolBar && m_searchTool && m_searchTool->GetToolBar() == NULL)
@@ -366,10 +366,14 @@
     if ( style & wxTB_TEXT && !(style & wxTB_NOICONS) && m_horzText )
         style |= wxTB_HORZ_LAYOUT;
 
-    toolBar = CreateToolBar(style, ID_TOOLBAR);
+    toolBar = new wxToolBar(this,ID_TOOLBAR,wxDefaultPosition,wxDefaultSize,style, wxT(""));
 #endif
 
+    //Uncomment this one to see working behaviour
+    //SetToolBar(toolBar);
     PopulateToolbar(toolBar);
+    //Uncomment this one to see broken behaviour
+    SetToolBar(toolBar);
 }
 
