--- c:\string.cpp.bak	Sat Apr 16 18:13:14 2005
+++ c:\string.cpp	Wed Apr 27 23:55:30 2005
@@ -1727,4 +1727,34 @@
 }
 
+// find a sub-string (like stristr)
+long wxString::FindNoCase(const wxChar *pszSub) const
+{
+  size_type idx = wxNOT_FOUND;
+
+  size_t substringLen = ::wxStrlen_(pszSub);
+  if (Length() < substringLen)
+    { return wxNOT_FOUND; }
+
+  for (size_type i = 0; i < Length()-(substringLen-1); ++i)
+    {
+    //compare the characters one at a time
+    for (size_t j = 0; j < substringLen; ++j)
+      {
+	  if (::wxTolower(GetChar(i+j)) != ::wxTolower(pszSub[j]) )
+		{
+		break;
+		}
+      }
+    //if the substring loop completed then the substring was found
+    if (j == substringLen)
+      {
+	  idx = i;
+      break;
+      }
+    }
+
+  return (idx == wxNOT_FOUND) ? wxNOT_FOUND : (long)idx;
+}
+
 // ----------------------------------------------------------------------------
 // conversion to numbers

 	  	 
