Index: src/generic/grid.cpp
===================================================================
--- src/generic/grid.cpp	(Revision 49598)
+++ src/generic/grid.cpp	(Arbeitskopie)
@@ -1560,6 +1560,7 @@
     if (m_allowOthers)
     {
         Combo()->SetValue(m_startValue);
+        Combo()->SetInsertionPointEnd();
     }
     else
     {
@@ -1570,7 +1571,6 @@
         Combo()->SetSelection(pos);
     }
 
-    Combo()->SetInsertionPointEnd();
     Combo()->SetFocus();
 
     if (evtHandler)

 	  	 
