? lib/vc_dll
Index: include/wx/docview.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/docview.h,v
retrieving revision 1.61
diff -u -r1.61 docview.h
--- include/wx/docview.h	2004/06/03 19:04:16	1.61
+++ include/wx/docview.h	2004/07/01 17:05:40
@@ -295,6 +295,12 @@
     wxClassInfo*      m_docClassInfo;
     wxClassInfo*      m_viewClassInfo;
 
+    // Called by CreateDocument and CreateView to create the actual document/view object.
+    // By default uses the ClassInfo provided to the constructor. Override these functions
+    // to provide a different method of creation.
+    virtual wxDocument *DoCreateDocument();
+    virtual wxView *DoCreateView();
+
 private:
     DECLARE_CLASS(wxDocTemplate)
     DECLARE_NO_COPY_CLASS(wxDocTemplate)
Index: src/common/docview.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/common/docview.cpp,v
retrieving revision 1.122
diff -u -r1.122 docview.cpp
--- src/common/docview.cpp	2004/06/05 20:47:41	1.122
+++ src/common/docview.cpp	2004/07/01 17:31:53
@@ -719,9 +719,9 @@
 // Tries to dynamically construct an object of the right class.
 wxDocument *wxDocTemplate::CreateDocument(const wxString& path, long flags)
 {
-    if (!m_docClassInfo)
+    wxDocument *doc = DoCreateDocument();
+    if ( doc == NULL )
         return (wxDocument *) NULL;
-    wxDocument *doc = (wxDocument *)m_docClassInfo->CreateObject();
     
     if (InitDocument(doc, path, flags))
     {
@@ -752,9 +752,10 @@
 
 wxView *wxDocTemplate::CreateView(wxDocument *doc, long flags)
 {
-    if (!m_viewClassInfo)
+    wxView *view = DoCreateView();
+    if ( view == NULL )
         return (wxView *) NULL;
-    wxView *view = (wxView *)m_viewClassInfo->CreateObject();
+
     view->SetDocument(doc);
     if (view->OnCreate(doc, flags))
     {
@@ -772,6 +773,22 @@
 bool wxDocTemplate::FileMatchesTemplate(const wxString& path)
 {
     return GetDefaultExtension().IsSameAs(FindExtension(path));
+}
+
+wxDocument *wxDocTemplate::DoCreateDocument()
+{
+    if (!m_docClassInfo)
+        return (wxDocument *) NULL;
+
+    return (wxDocument *)m_docClassInfo->CreateObject();
+}
+
+wxView *wxDocTemplate::DoCreateView()
+{
+    if (!m_viewClassInfo)
+        return (wxView *) NULL;
+
+    return (wxView *)m_viewClassInfo->CreateObject();
 }
 
 // ----------------------------------------------------------------------------

 	  	 
