--- ..\wxPython-src-2.6.1.0\wxPython\src\_dataobj.i	2005-06-04 07:49:16.000000000 +0900
+++ .\_dataobj.i	2005-07-16 23:55:02.280500000 +0900
@@ -248,8 +248,7 @@
             for (size_t i=0; i<count; i++) {
                 wxDataFormat* format = new wxDataFormat(formats[i]);
                 PyObject* obj = wxPyConstructObject((void*)format, wxT("wxDataFormat"), true);
-                PyList_Append(list, obj);
-                Py_DECREF(obj);
+                PyList_SET_ITEM(list, i, obj); // PyList_SET_ITEM steals a refference
             }            
             wxPyEndBlockThreads(blocked);
             delete [] formats;

 	  	 
