Index: src/msw/mdi.cpp
===================================================================
--- src/msw/mdi.cpp	(revision 71211)
+++ src/msw/mdi.cpp	(working copy)
@@ -160,6 +160,7 @@
   {
       // normal case: we have the window menu, so construct it
       m_windowMenu = new wxMenu;
+      m_windowMenu->SetInvokingWindow(this);
 
       m_windowMenu->Append(wxID_MDI_WINDOW_CASCADE, _("&Cascade"));
       m_windowMenu->Append(wxID_MDI_WINDOW_TILE_HORZ, _("Tile &Horizontally"));
