Index: e:/dev/wxWindows/src/common/xti.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/common/xti.cpp,v
retrieving revision 1.33
diff -u -2 -r1.33 xti.cpp
--- e:/dev/wxWindows/src/common/xti.cpp	2003/10/17 19:53:19	1.33
+++ e:/dev/wxWindows/src/common/xti.cpp	2004/02/13 18:01:37
@@ -243,15 +243,30 @@
 //
 
-wxBuiltInTypeInfo s_typeInfovoid( wxT_VOID , NULL , NULL , typeid(void).name() ) ;
-wxBuiltInTypeInfo s_typeInfobool( wxT_BOOL , &wxToStringConverter<bool> , &wxFromStringConverter<bool>, typeid(bool).name()) ;
-wxBuiltInTypeInfo s_typeInfochar( wxT_CHAR , &wxToStringConverter<char> , &wxFromStringConverter<char>, typeid(char).name()) ;
-wxBuiltInTypeInfo s_typeInfounsignedchar( wxT_UCHAR , &wxToStringConverter< unsigned char > , &wxFromStringConverter<unsigned char>, typeid(unsigned char).name()) ;
-wxBuiltInTypeInfo s_typeInfoint( wxT_INT , &wxToStringConverter<int> , &wxFromStringConverter<int>, typeid(int).name()) ;
-wxBuiltInTypeInfo s_typeInfounsignedint( wxT_UINT , &wxToStringConverter<unsigned int> , &wxFromStringConverter<unsigned int>, typeid(unsigned int).name()) ;
-wxBuiltInTypeInfo s_typeInfolong( wxT_LONG , &wxToStringConverter<long> , &wxFromStringConverter<long>, typeid(long).name()) ;
-wxBuiltInTypeInfo s_typeInfounsignedlong( wxT_ULONG , &wxToStringConverter<unsigned long> , &wxFromStringConverter<unsigned long>, typeid(unsigned long).name()) ;
-wxBuiltInTypeInfo s_typeInfofloat( wxT_FLOAT , &wxToStringConverter<float> , &wxFromStringConverter<float>, typeid(float).name()) ;
-wxBuiltInTypeInfo s_typeInfodouble( wxT_DOUBLE , &wxToStringConverter<double> , &wxFromStringConverter<double>, typeid(double).name()) ;
-wxBuiltInTypeInfo s_typeInfowxString( wxT_STRING , &wxToStringConverter<wxString> , &wxFromStringConverter<wxString>, typeid(wxString).name()) ;
+#if wxUSE_FUNC_TEMPLATE_POINTER
+#define wxBUILTIN_TYPE_INFO( element , type ) \
+    wxBuiltInTypeInfo s_typeInfo##type(element , &wxToStringConverter<type> , &wxFromStringConverter<type> , typeid(type).name()) ;
+#else
+#define wxBUILTIN_TYPE_INFO( element , type ) \
+    void _toString##element( const wxxVariant& data , wxString &result ) { wxToStringConverter<type>(data, result); } \
+    void _fromString##element( const wxString& data , wxxVariant &result ) { wxFromStringConverter<type>(data, result); } \
+    wxBuiltInTypeInfo s_typeInfo##type(element , &_toString##element , &_fromString##element , typeid(type).name()) ;
+#endif
+
+typedef unsigned char unsigned_char;
+typedef unsigned int unsigned_int;
+typedef unsigned long unsigned_long;
+
+wxBuiltInTypeInfo s_typeInfovoid( wxT_VOID , NULL , NULL , typeid(void).name());
+wxBUILTIN_TYPE_INFO( wxT_BOOL ,  bool);
+wxBUILTIN_TYPE_INFO( wxT_CHAR ,  char);
+wxBUILTIN_TYPE_INFO( wxT_UCHAR , unsigned_char);
+wxBUILTIN_TYPE_INFO( wxT_INT , int);
+wxBUILTIN_TYPE_INFO( wxT_UINT , unsigned_int);
+wxBUILTIN_TYPE_INFO( wxT_LONG , long);
+wxBUILTIN_TYPE_INFO( wxT_ULONG , unsigned_long);
+wxBUILTIN_TYPE_INFO( wxT_FLOAT , float);
+wxBUILTIN_TYPE_INFO( wxT_DOUBLE , double);
+wxBUILTIN_TYPE_INFO( wxT_STRING , wxString);
+
 
 // this are compiler induced specialization which are never used anywhere
@@ -509,11 +524,14 @@
 }
 
-void wxClassInfo::GetProperties( wxPropertyInfoMap &map ) const
+// void wxClassInfo::GetProperties( wxPropertyInfoMap &map ) const 
+// The map parameter (the name map that is) seems something special
+// to MSVC and so we use a other name.
+void wxClassInfo::GetProperties( wxPropertyInfoMap &infomap ) const 
 {
     const wxPropertyInfo *pi = GetFirstProperty() ;
     while( pi ) 
     {
-        if ( map.find( pi->GetName() ) == map.end() )
-            map[pi->GetName()] = (wxPropertyInfo*) pi ;
+        if ( infomap.find( pi->GetName() ) == infomap.end() )
+            infomap[pi->GetName()] = (wxPropertyInfo*) pi ;
 
         pi = pi->GetNext() ;
@@ -523,5 +541,5 @@
     for ( int i = 0 ; parents[i] ; ++ i )
     {
-        parents[i]->GetProperties( map ) ;
+        parents[i]->GetProperties( infomap ) ;
     }
 }
Index: e:/dev/wxWindows/src/common/xtistrm.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/common/xtistrm.cpp,v
retrieving revision 1.21
diff -u -2 -r1.21 xtistrm.cpp
--- e:/dev/wxWindows/src/common/xtistrm.cpp	2003/10/19 13:41:33	1.21
+++ e:/dev/wxWindows/src/common/xtistrm.cpp	2004/02/13 18:01:39
@@ -163,20 +163,22 @@
         map.erase( name ) ;
     }
-
-    for( wxPropertyInfoMap::iterator iter = map.begin() ; iter != map.end() ; ++iter )
-    {
-        const wxPropertyInfo* prop = iter->second ;
-        if ( prop->GetFlags() & wxPROP_OBJECT_GRAPH )
+    { // Extra block for broken compilers
+        for( wxPropertyInfoMap::iterator iter = map.begin() ; iter != map.end() ; ++iter )
         {
-            WriteOneProperty( obj , prop->GetDeclaringClass() , prop , persister , data ) ;
+            const wxPropertyInfo* prop = iter->second ;
+            if ( prop->GetFlags() & wxPROP_OBJECT_GRAPH )
+            {
+                WriteOneProperty( obj , prop->GetDeclaringClass() , prop , persister , data ) ;
+            }
         }
     }
-
-    for( wxPropertyInfoMap::iterator iter = map.begin() ; iter != map.end() ; ++iter )
-    {
-        const wxPropertyInfo* prop = iter->second ;
-        if ( !(prop->GetFlags() & wxPROP_OBJECT_GRAPH) )
+    { // Extra block for broken compilers
+        for( wxPropertyInfoMap::iterator iter = map.begin() ; iter != map.end() ; ++iter )
         {
-            WriteOneProperty( obj , prop->GetDeclaringClass() , prop , persister , data ) ;
+            const wxPropertyInfo* prop = iter->second ;
+            if ( !(prop->GetFlags() & wxPROP_OBJECT_GRAPH) )
+            {
+                WriteOneProperty( obj , prop->GetDeclaringClass() , prop , persister , data ) ;
+            }
         }
     }
Index: e:/dev/wxWindows/src/generic/dirctrlg.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/generic/dirctrlg.cpp,v
retrieving revision 1.87
diff -u -2 -r1.87 dirctrlg.cpp
--- e:/dev/wxWindows/src/generic/dirctrlg.cpp	2004/02/08 16:13:12	1.87
+++ e:/dev/wxWindows/src/generic/dirctrlg.cpp	2004/02/13 18:01:46
@@ -510,8 +510,8 @@
 wxBEGIN_PROPERTIES_TABLE(wxGenericDirCtrl)
     wxHIDE_PROPERTY( Children )
-	wxPROPERTY( DefaultPath , wxString , SetDefaultPath , GetDefaultPath  , , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
-	wxPROPERTY( Filter , wxString , SetFilter , GetFilter  ,, 0 /*flags*/ , wxT("Helpstring") , wxT("group") )
-	wxPROPERTY( DefaultFilter , int , SetFilterIndex, GetFilterIndex,, 0 /*flags*/ , wxT("Helpstring") , wxT("group") )
-    wxPROPERTY_FLAGS( WindowStyle, wxGenericDirCtrlStyle, long, SetWindowStyleFlag, GetWindowStyleFlag, , 0, wxT("Helpstring"), wxT("group") )
+	wxPROPERTY( DefaultPath , wxString , SetDefaultPath , GetDefaultPath  , EMPTY_MACROVALUE , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
+	wxPROPERTY( Filter , wxString , SetFilter , GetFilter  , EMPTY_MACROVALUE , 0 /*flags*/ , wxT("Helpstring") , wxT("group") )
+	wxPROPERTY( DefaultFilter , int , SetFilterIndex, GetFilterIndex, EMPTY_MACROVALUE , 0 /*flags*/ , wxT("Helpstring") , wxT("group") )
+    wxPROPERTY_FLAGS( WindowStyle, wxGenericDirCtrlStyle, long, SetWindowStyleFlag, GetWindowStyleFlag, EMPTY_MACROVALUE , 0, wxT("Helpstring"), wxT("group") )
 wxEND_PROPERTIES_TABLE()
 
Index: e:/dev/wxWindows/src/generic/grid.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/generic/grid.cpp,v
retrieving revision 1.288
diff -u -2 -r1.288 grid.cpp
--- e:/dev/wxWindows/src/generic/grid.cpp	2004/02/10 22:16:55	1.288
+++ e:/dev/wxWindows/src/generic/grid.cpp	2004/02/13 18:02:05
@@ -3813,5 +3813,5 @@
 wxBEGIN_PROPERTIES_TABLE(wxGrid)
     wxHIDE_PROPERTY( Children )
-    wxPROPERTY_FLAGS( WindowStyle , wxGridStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
+    wxPROPERTY_FLAGS( WindowStyle , wxGridStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , EMPTY_MACROVALUE, 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
 wxEND_PROPERTIES_TABLE()
 
Index: e:/dev/wxWindows/src/generic/panelg.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/generic/panelg.cpp,v
retrieving revision 1.66
diff -u -2 -r1.66 panelg.cpp
--- e:/dev/wxWindows/src/generic/panelg.cpp	2003/09/24 00:41:03	1.66
+++ e:/dev/wxWindows/src/generic/panelg.cpp	2004/02/13 18:02:05
@@ -80,5 +80,5 @@
 
 wxBEGIN_PROPERTIES_TABLE(wxPanel)
-    wxPROPERTY_FLAGS( WindowStyle , wxPanelStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
+    wxPROPERTY_FLAGS( WindowStyle , wxPanelStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , EMPTY_MACROVALUE, 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
 // style wxTAB_TRAVERSAL
 wxEND_PROPERTIES_TABLE()
Index: e:/dev/wxWindows/src/msw/bmpbuttn.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/bmpbuttn.cpp,v
retrieving revision 1.44
diff -u -2 -r1.44 bmpbuttn.cpp
--- e:/dev/wxWindows/src/msw/bmpbuttn.cpp	2004/01/24 21:16:59	1.44
+++ e:/dev/wxWindows/src/msw/bmpbuttn.cpp	2004/02/13 18:02:08
@@ -77,5 +77,5 @@
 
 wxBEGIN_PROPERTIES_TABLE(wxBitmapButton)
-    wxPROPERTY_FLAGS( WindowStyle , wxBitmapButtonStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
+    wxPROPERTY_FLAGS( WindowStyle , wxBitmapButtonStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , EMPTY_MACROVALUE, 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
 wxEND_PROPERTIES_TABLE()
 
Index: e:/dev/wxWindows/src/msw/button.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/button.cpp,v
retrieving revision 1.79
diff -u -2 -r1.79 button.cpp
--- e:/dev/wxWindows/src/msw/button.cpp	2004/01/27 17:47:28	1.79
+++ e:/dev/wxWindows/src/msw/button.cpp	2004/02/13 18:02:09
@@ -91,8 +91,8 @@
 	wxEVENT_PROPERTY( Click , wxEVT_COMMAND_BUTTON_CLICKED , wxCommandEvent)
 
-	wxPROPERTY( Font , wxFont , SetFont , GetFont  , , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
+	wxPROPERTY( Font , wxFont , SetFont , GetFont  , EMPTY_MACROVALUE, 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
 	wxPROPERTY( Label, wxString , SetLabel, GetLabel, wxString(), 0 /*flags*/ , wxT("Helpstring") , wxT("group") )
 
-    wxPROPERTY_FLAGS( WindowStyle , wxButtonStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
+    wxPROPERTY_FLAGS( WindowStyle , wxButtonStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , EMPTY_MACROVALUE , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
 
 wxEND_PROPERTIES_TABLE()
Index: e:/dev/wxWindows/src/msw/checkbox.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/checkbox.cpp,v
retrieving revision 1.53
diff -u -2 -r1.53 checkbox.cpp
--- e:/dev/wxWindows/src/msw/checkbox.cpp	2004/01/15 13:49:04	1.53
+++ e:/dev/wxWindows/src/msw/checkbox.cpp	2004/02/13 18:02:10
@@ -94,8 +94,8 @@
 	wxEVENT_PROPERTY( Click , wxEVT_COMMAND_CHECKBOX_CLICKED , wxCommandEvent )
 
-	wxPROPERTY( Font , wxFont , SetFont , GetFont  , , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
+	wxPROPERTY( Font , wxFont , SetFont , GetFont , EMPTY_MACROVALUE , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
 	wxPROPERTY( Label,wxString, SetLabel, GetLabel, wxString() , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
-	wxPROPERTY( Value ,bool, SetValue, GetValue, , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
-    wxPROPERTY_FLAGS( WindowStyle , wxCheckBoxStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
+	wxPROPERTY( Value ,bool, SetValue, GetValue, EMPTY_MACROVALUE, 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
+    wxPROPERTY_FLAGS( WindowStyle , wxCheckBoxStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , EMPTY_MACROVALUE, 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
 wxEND_PROPERTIES_TABLE()
 
Index: e:/dev/wxWindows/src/msw/checklst.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/checklst.cpp,v
retrieving revision 1.53
diff -u -2 -r1.53 checklst.cpp
--- e:/dev/wxWindows/src/msw/checklst.cpp	2004/01/31 18:21:38	1.53
+++ e:/dev/wxWindows/src/msw/checklst.cpp	2004/02/13 18:02:11
@@ -115,5 +115,5 @@
 wxBEGIN_PROPERTIES_TABLE(wxCheckListBox)
 	wxEVENT_PROPERTY( Toggle , wxEVT_COMMAND_CHECKLISTBOX_TOGGLED , wxCommandEvent )
-    wxPROPERTY_FLAGS( WindowStyle , wxCheckListBoxStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , , wxLB_OWNERDRAW /*flags*/ , wxT("Helpstring") , wxT("group")) // style
+    wxPROPERTY_FLAGS( WindowStyle , wxCheckListBoxStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , EMPTY_MACROVALUE , wxLB_OWNERDRAW /*flags*/ , wxT("Helpstring") , wxT("group")) // style
 wxEND_PROPERTIES_TABLE()
 
Index: e:/dev/wxWindows/src/msw/choice.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/choice.cpp,v
retrieving revision 1.75
diff -u -2 -r1.75 choice.cpp
--- e:/dev/wxWindows/src/msw/choice.cpp	2004/02/10 14:41:06	1.75
+++ e:/dev/wxWindows/src/msw/choice.cpp	2004/02/13 18:02:12
@@ -79,8 +79,8 @@
 	wxEVENT_PROPERTY( Select , wxEVT_COMMAND_CHOICE_SELECTED , wxCommandEvent )
 
-    wxPROPERTY( Font , wxFont , SetFont , GetFont  , , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
+    wxPROPERTY( Font , wxFont , SetFont , GetFont  , EMPTY_MACROVALUE , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
     wxPROPERTY_COLLECTION( Choices , wxArrayString , wxString , AppendString , GetStrings , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
-	wxPROPERTY( Selection ,int, SetSelection, GetSelection, , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
-    wxPROPERTY_FLAGS( WindowStyle , wxChoiceStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
+	wxPROPERTY( Selection ,int, SetSelection, GetSelection, EMPTY_MACROVALUE , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
+    wxPROPERTY_FLAGS( WindowStyle , wxChoiceStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , EMPTY_MACROVALUE , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
 wxEND_PROPERTIES_TABLE()
 
Index: e:/dev/wxWindows/src/msw/colour.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/colour.cpp,v
retrieving revision 1.16
diff -u -2 -r1.16 colour.cpp
--- e:/dev/wxWindows/src/msw/colour.cpp	2003/12/11 10:10:31	1.16
+++ e:/dev/wxWindows/src/msw/colour.cpp	2004/02/13 18:02:12
@@ -54,10 +54,13 @@
 }
 
-IMPLEMENT_DYNAMIC_CLASS_WITH_COPY_AND_STREAMERS_XTI( wxColour , wxObject , "wx/colour.h" ,  &wxToStringConverter<wxColour> , &wxFromStringConverter<wxColour>)
+wxTO_STRING_IMP( wxColour )
+wxFROM_STRING_IMP( wxColour )
+  
+IMPLEMENT_DYNAMIC_CLASS_WITH_COPY_AND_STREAMERS_XTI( wxColour , wxObject , "wx/colour.h" ,  &wxTO_STRING( wxColour ) , &wxFROM_STRING( wxColour ))
 
 wxBEGIN_PROPERTIES_TABLE(wxColour)
-    wxREADONLY_PROPERTY( Red, unsigned char, Red,  , 0 /*flags*/, wxT("Helpstring"), wxT("group"))
-    wxREADONLY_PROPERTY( Green, unsigned char, Green,  , 0 /*flags*/, wxT("Helpstring"), wxT("group"))
-    wxREADONLY_PROPERTY( Blue, unsigned char, Blue,  , 0 /*flags*/, wxT("Helpstring"), wxT("group"))
+    wxREADONLY_PROPERTY( Red, unsigned char, Red, EMPTY_MACROVALUE , 0 /*flags*/, wxT("Helpstring"), wxT("group"))
+    wxREADONLY_PROPERTY( Green, unsigned char, Green, EMPTY_MACROVALUE , 0 /*flags*/, wxT("Helpstring"), wxT("group"))
+    wxREADONLY_PROPERTY( Blue, unsigned char, Blue, EMPTY_MACROVALUE , 0 /*flags*/, wxT("Helpstring"), wxT("group"))
 wxEND_PROPERTIES_TABLE()
 
Index: e:/dev/wxWindows/src/msw/combobox.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/combobox.cpp,v
retrieving revision 1.80
diff -u -2 -r1.80 combobox.cpp
--- e:/dev/wxWindows/src/msw/combobox.cpp	2004/02/12 16:45:28	1.80
+++ e:/dev/wxWindows/src/msw/combobox.cpp	2004/02/13 18:02:14
@@ -99,9 +99,9 @@
 
     // TODO DELEGATES
-	wxPROPERTY( Font , wxFont , SetFont , GetFont  , , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
+	wxPROPERTY( Font , wxFont , SetFont , GetFont  , EMPTY_MACROVALUE , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
     wxPROPERTY_COLLECTION( Choices , wxArrayString , wxString , AppendString , GetStrings , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
-	wxPROPERTY( Value ,wxString, SetValue, GetValue, , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
-	wxPROPERTY( Selection ,int, SetSelection, GetSelection, , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
-    wxPROPERTY_FLAGS( WindowStyle , wxComboBoxStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
+	wxPROPERTY( Value ,wxString, SetValue, GetValue, EMPTY_MACROVALUE , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
+	wxPROPERTY( Selection ,int, SetSelection, GetSelection, EMPTY_MACROVALUE , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
+    wxPROPERTY_FLAGS( WindowStyle , wxComboBoxStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , EMPTY_MACROVALUE , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
 wxEND_PROPERTIES_TABLE()
 
Index: e:/dev/wxWindows/src/msw/dialog.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/dialog.cpp,v
retrieving revision 1.101
diff -u -2 -r1.101 dialog.cpp
--- e:/dev/wxWindows/src/msw/dialog.cpp	2004/02/08 16:19:57	1.101
+++ e:/dev/wxWindows/src/msw/dialog.cpp	2004/02/13 18:02:15
@@ -96,6 +96,6 @@
 
 wxBEGIN_PROPERTIES_TABLE(wxDialog)
-    wxPROPERTY( Title,wxString, SetTitle, GetTitle, wxString() , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
-    wxPROPERTY_FLAGS( WindowStyle , wxDialogStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
+    wxPROPERTY( Title, wxString, SetTitle, GetTitle, wxString() , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
+    wxPROPERTY_FLAGS( WindowStyle , wxDialogStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , EMPTY_MACROVALUE , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
 wxEND_PROPERTIES_TABLE()
 
Index: e:/dev/wxWindows/src/msw/font.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/font.cpp,v
retrieving revision 1.59
diff -u -2 -r1.59 font.cpp
--- e:/dev/wxWindows/src/msw/font.cpp	2003/09/24 01:23:12	1.59
+++ e:/dev/wxWindows/src/msw/font.cpp	2004/02/13 18:02:16
@@ -78,5 +78,5 @@
 	wxPROPERTY( Weight, int , SetWeight, GetWeight, (int)wxNORMAL , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // wxFontWeight
 	wxPROPERTY( Underlined, bool , SetUnderlined, GetUnderlined, false , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
-	wxPROPERTY( Face, wxString , SetFaceName, GetFaceName, , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
+	wxPROPERTY( Face, wxString , SetFaceName, GetFaceName, EMPTY_MACROVALUE , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
 	wxPROPERTY( Encoding, wxFontEncoding , SetEncoding, GetEncoding, wxFONTENCODING_DEFAULT , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
 wxEND_PROPERTIES_TABLE()
Index: e:/dev/wxWindows/src/msw/frame.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/frame.cpp,v
retrieving revision 1.161
diff -u -2 -r1.161 frame.cpp
--- e:/dev/wxWindows/src/msw/frame.cpp	2004/01/02 13:49:56	1.161
+++ e:/dev/wxWindows/src/msw/frame.cpp	2004/02/13 18:02:18
@@ -135,6 +135,6 @@
 
     wxPROPERTY( Title,wxString, SetTitle, GetTitle, wxString() , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
-    wxPROPERTY_FLAGS( WindowStyle , wxFrameStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
-    wxPROPERTY( MenuBar , wxMenuBar * , SetMenuBar , GetMenuBar , , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
+    wxPROPERTY_FLAGS( WindowStyle , wxFrameStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , EMPTY_MACROVALUE , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
+    wxPROPERTY( MenuBar , wxMenuBar * , SetMenuBar , GetMenuBar , EMPTY_MACROVALUE , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
 wxEND_PROPERTIES_TABLE()
 
Index: e:/dev/wxWindows/src/msw/gauge95.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/gauge95.cpp,v
retrieving revision 1.40
diff -u -2 -r1.40 gauge95.cpp
--- e:/dev/wxWindows/src/msw/gauge95.cpp	2003/09/28 13:57:52	1.40
+++ e:/dev/wxWindows/src/msw/gauge95.cpp	2004/02/13 18:02:18
@@ -112,5 +112,5 @@
     wxPROPERTY( ShadowWidth , int , SetShadowWidth, GetShadowWidth, 0 , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
     wxPROPERTY( BezelFace , int , SetBezelFace, GetBezelFace, 0 , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
-    wxPROPERTY_FLAGS( WindowStyle , wxGaugeStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
+    wxPROPERTY_FLAGS( WindowStyle , wxGaugeStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , EMPTY_MACROVALUE , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
 wxEND_PROPERTIES_TABLE()
 
Index: e:/dev/wxWindows/src/msw/listbox.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/listbox.cpp,v
retrieving revision 1.96
diff -u -2 -r1.96 listbox.cpp
--- e:/dev/wxWindows/src/msw/listbox.cpp	2004/02/07 19:24:16	1.96
+++ e:/dev/wxWindows/src/msw/listbox.cpp	2004/02/13 18:02:20
@@ -95,8 +95,8 @@
 	wxEVENT_PROPERTY( DoubleClick , wxEVT_COMMAND_LISTBOX_DOUBLECLICKED , wxCommandEvent )
 
-    wxPROPERTY( Font , wxFont , SetFont , GetFont  , , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
+    wxPROPERTY( Font , wxFont , SetFont , GetFont  , EMPTY_MACROVALUE, 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
     wxPROPERTY_COLLECTION( Choices , wxArrayString , wxString , AppendString , GetStrings, 0 /*flags*/ , wxT("Helpstring") , wxT("group") )
-	wxPROPERTY( Selection ,int, SetSelection, GetSelection,, 0 /*flags*/ , wxT("Helpstring") , wxT("group") )
-    wxPROPERTY_FLAGS( WindowStyle , wxListBoxStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
+	wxPROPERTY( Selection ,int, SetSelection, GetSelection, EMPTY_MACROVALUE , 0 /*flags*/ , wxT("Helpstring") , wxT("group") )
+    wxPROPERTY_FLAGS( WindowStyle , wxListBoxStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , EMPTY_MACROVALUE , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
 wxEND_PROPERTIES_TABLE()
 
Index: e:/dev/wxWindows/src/msw/listctrl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/listctrl.cpp,v
retrieving revision 1.191
diff -u -2 -r1.191 listctrl.cpp
--- e:/dev/wxWindows/src/msw/listctrl.cpp	2004/02/08 15:02:22	1.191
+++ e:/dev/wxWindows/src/msw/listctrl.cpp	2004/02/13 18:02:25
@@ -292,5 +292,5 @@
     wxEVENT_PROPERTY( TextUpdated , wxEVT_COMMAND_TEXT_UPDATED , wxCommandEvent ) 
 
-    wxPROPERTY_FLAGS( WindowStyle , wxListCtrlStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
+    wxPROPERTY_FLAGS( WindowStyle , wxListCtrlStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , EMPTY_MACROVALUE , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
 wxEND_PROPERTIES_TABLE()
 
Index: e:/dev/wxWindows/src/msw/menu.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/menu.cpp,v
retrieving revision 1.97
diff -u -2 -r1.97 menu.cpp
--- e:/dev/wxWindows/src/msw/menu.cpp	2004/02/08 15:29:41	1.97
+++ e:/dev/wxWindows/src/msw/menu.cpp	2004/02/13 18:02:31
@@ -138,5 +138,5 @@
 	wxEVENT_PROPERTY( Select , wxEVT_COMMAND_MENU_SELECTED , wxCommandEvent)
     wxPROPERTY( Title, wxString , SetTitle, GetTitle, wxString(), 0 /*flags*/ , wxT("Helpstring") , wxT("group") )
-    wxREADONLY_PROPERTY_FLAGS( MenuStyle , wxMenuStyle , long , GetStyle , , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
+    wxREADONLY_PROPERTY_FLAGS( MenuStyle , wxMenuStyle , long , GetStyle , EMPTY_MACROVALUE , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
     wxPROPERTY_COLLECTION( MenuItems , wxMenuItemList , wxMenuItem* , Append , GetMenuItems , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
 wxEND_PROPERTIES_TABLE()
@@ -164,5 +164,5 @@
 
 wxBEGIN_PROPERTIES_TABLE(wxMenuInfo)
-    wxREADONLY_PROPERTY( Menu , wxMenu* , GetMenu , , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
+    wxREADONLY_PROPERTY( Menu , wxMenu* , GetMenu , EMPTY_MACROVALUE , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
     wxREADONLY_PROPERTY( Title , wxString , GetTitle , wxString() , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
 wxEND_PROPERTIES_TABLE()
Index: e:/dev/wxWindows/src/msw/menuitem.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/menuitem.cpp,v
retrieving revision 1.52
diff -u -2 -r1.52 menuitem.cpp
--- e:/dev/wxWindows/src/msw/menuitem.cpp	2004/01/17 17:04:11	1.52
+++ e:/dev/wxWindows/src/msw/menuitem.cpp	2004/02/13 18:02:32
@@ -102,10 +102,10 @@
 
 wxBEGIN_PROPERTIES_TABLE(wxMenuItem)
-	wxPROPERTY( Parent,wxMenu*, SetMenu, GetMenu, , 0 /*flags*/ , wxT("Helpstring") , wxT("group") )
-	wxPROPERTY( Id,int, SetId, GetId, , 0 /*flags*/ , wxT("Helpstring") , wxT("group") )
+	wxPROPERTY( Parent,wxMenu*, SetMenu, GetMenu, EMPTY_MACROVALUE , 0 /*flags*/ , wxT("Helpstring") , wxT("group") )
+	wxPROPERTY( Id,int, SetId, GetId, EMPTY_MACROVALUE , 0 /*flags*/ , wxT("Helpstring") , wxT("group") )
     wxPROPERTY( Text, wxString , SetText, GetText, wxString(), 0 /*flags*/ , wxT("Helpstring") , wxT("group") )
     wxPROPERTY( Help, wxString , SetHelp, GetHelp, wxString(), 0 /*flags*/ , wxT("Helpstring") , wxT("group") )
-    wxREADONLY_PROPERTY( Kind, wxItemKind , GetKind , , 0 /*flags*/ , wxT("Helpstring") , wxT("group") )
-	wxPROPERTY( SubMenu,wxMenu*, SetSubMenu, GetSubMenu, , 0 /*flags*/ , wxT("Helpstring") , wxT("group") )
+    wxREADONLY_PROPERTY( Kind, wxItemKind , GetKind , EMPTY_MACROVALUE , 0 /*flags*/ , wxT("Helpstring") , wxT("group") )
+	wxPROPERTY( SubMenu,wxMenu*, SetSubMenu, GetSubMenu, EMPTY_MACROVALUE , 0 /*flags*/ , wxT("Helpstring") , wxT("group") )
 	wxPROPERTY( Enabled , bool , Enable , IsEnabled , wxxVariant((bool)true) , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
 	wxPROPERTY( Checked , bool , Check , IsChecked , wxxVariant((bool)false) , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
Index: e:/dev/wxWindows/src/msw/notebook.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/notebook.cpp,v
retrieving revision 1.106
diff -u -2 -r1.106 notebook.cpp
--- e:/dev/wxWindows/src/msw/notebook.cpp	2004/02/07 20:48:15	1.106
+++ e:/dev/wxWindows/src/msw/notebook.cpp	2004/02/13 18:02:34
@@ -168,5 +168,5 @@
 
     wxPROPERTY_COLLECTION( PageInfos , wxNotebookPageInfoList , wxNotebookPageInfo* , AddPageInfo , GetPageInfos , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
-    wxPROPERTY_FLAGS( WindowStyle , wxNotebookStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
+    wxPROPERTY_FLAGS( WindowStyle , wxNotebookStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , EMPTY_MACROVALUE , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
 wxEND_PROPERTIES_TABLE()
 
@@ -178,5 +178,5 @@
 
 wxBEGIN_PROPERTIES_TABLE(wxNotebookPageInfo)
-    wxREADONLY_PROPERTY( Page , wxNotebookPage* , GetPage , , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
+    wxREADONLY_PROPERTY( Page , wxNotebookPage* , GetPage , EMPTY_MACROVALUE , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
     wxREADONLY_PROPERTY( Text , wxString , GetText , wxString() , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
     wxREADONLY_PROPERTY( Selected , bool , GetSelected , false, 0 /*flags*/ , wxT("Helpstring") , wxT("group") )
Index: e:/dev/wxWindows/src/msw/radiobut.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/radiobut.cpp,v
retrieving revision 1.48
diff -u -2 -r1.48 radiobut.cpp
--- e:/dev/wxWindows/src/msw/radiobut.cpp	2004/01/15 13:49:06	1.48
+++ e:/dev/wxWindows/src/msw/radiobut.cpp	2004/02/13 18:02:34
@@ -87,8 +87,8 @@
 wxBEGIN_PROPERTIES_TABLE(wxRadioButton)
     wxEVENT_PROPERTY( Click , wxEVT_COMMAND_RADIOBUTTON_SELECTED , wxCommandEvent )
-    wxPROPERTY( Font , wxFont , SetFont , GetFont  , , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
+    wxPROPERTY( Font , wxFont , SetFont , GetFont  , EMPTY_MACROVALUE , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
     wxPROPERTY( Label,wxString, SetLabel, GetLabel, wxString(), 0 /*flags*/ , wxT("Helpstring") , wxT("group") )
-    wxPROPERTY( Value ,bool, SetValue, GetValue,, 0 /*flags*/ , wxT("Helpstring") , wxT("group") )
-    wxPROPERTY_FLAGS( WindowStyle , wxRadioButtonStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
+    wxPROPERTY( Value ,bool, SetValue, GetValue, EMPTY_MACROVALUE , 0 /*flags*/ , wxT("Helpstring") , wxT("group") )
+    wxPROPERTY_FLAGS( WindowStyle , wxRadioButtonStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , EMPTY_MACROVALUE , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
 wxEND_PROPERTIES_TABLE()
 
Index: e:/dev/wxWindows/src/msw/scrolbar.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/scrolbar.cpp,v
retrieving revision 1.41
diff -u -2 -r1.41 scrolbar.cpp
--- e:/dev/wxWindows/src/msw/scrolbar.cpp	2003/09/24 00:41:06	1.41
+++ e:/dev/wxWindows/src/msw/scrolbar.cpp	2004/02/13 18:02:35
@@ -75,5 +75,5 @@
 	wxPROPERTY( ThumbSize , int , SetThumbSize, GetThumbSize, 0 , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
 	wxPROPERTY( PageSize , int , SetPageSize, GetPageSize, 0 , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
-    wxPROPERTY_FLAGS( WindowStyle , wxScrollBarStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
+    wxPROPERTY_FLAGS( WindowStyle , wxScrollBarStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , EMPTY_MACROVALUE , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
 wxEND_PROPERTIES_TABLE()
 
Index: e:/dev/wxWindows/src/msw/slider95.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/slider95.cpp,v
retrieving revision 1.54
diff -u -2 -r1.54 slider95.cpp
--- e:/dev/wxWindows/src/msw/slider95.cpp	2004/02/12 18:19:12	1.54
+++ e:/dev/wxWindows/src/msw/slider95.cpp	2004/02/13 18:02:36
@@ -94,5 +94,5 @@
 	wxPROPERTY( LineSize , int , SetLineSize, GetLineSize, 1 , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
 	wxPROPERTY( ThumbLength , int , SetThumbLength, GetThumbLength, 1 , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
-    wxPROPERTY_FLAGS( WindowStyle , wxSliderStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
+    wxPROPERTY_FLAGS( WindowStyle , wxSliderStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , EMPTY_MACROVALUE , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
 wxEND_PROPERTIES_TABLE()
 
Index: e:/dev/wxWindows/src/msw/spinbutt.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/spinbutt.cpp,v
retrieving revision 1.53
diff -u -2 -r1.53 spinbutt.cpp
--- e:/dev/wxWindows/src/msw/spinbutt.cpp	2003/10/30 23:08:30	1.53
+++ e:/dev/wxWindows/src/msw/spinbutt.cpp	2004/02/13 18:02:37
@@ -103,5 +103,5 @@
 	wxPROPERTY( Min , int , SetMin, GetMin, 0 , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
 	wxPROPERTY( Max , int , SetMax, GetMax, 0 , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
-    wxPROPERTY_FLAGS( WindowStyle , wxSpinButtonStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
+    wxPROPERTY_FLAGS( WindowStyle , wxSpinButtonStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , EMPTY_MACROVALUE , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
 wxEND_PROPERTIES_TABLE()
 
Index: e:/dev/wxWindows/src/msw/spinctrl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/spinctrl.cpp,v
retrieving revision 1.37
diff -u -2 -r1.37 spinctrl.cpp
--- e:/dev/wxWindows/src/msw/spinctrl.cpp	2004/02/10 13:02:51	1.37
+++ e:/dev/wxWindows/src/msw/spinctrl.cpp	2004/02/13 18:02:38
@@ -97,9 +97,9 @@
     wxEVENT_PROPERTY( TextEnter , wxEVT_COMMAND_TEXT_ENTER , wxCommandEvent )
 
-    wxPROPERTY( ValueString , wxString , SetValue , GetValue , , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) ;
+    wxPROPERTY( ValueString , wxString , SetValue , GetValue , EMPTY_MACROVALUE , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) ;
 	wxPROPERTY( Value , int , SetValue, GetValue, 0 , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
 	wxPROPERTY( Min , int , SetMin, GetMin, 0, 0 /*flags*/ , wxT("Helpstring") , wxT("group") )
 	wxPROPERTY( Max , int , SetMax, GetMax, 0 , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
-    wxPROPERTY_FLAGS( WindowStyle , wxSpinCtrlStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
+    wxPROPERTY_FLAGS( WindowStyle , wxSpinCtrlStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , EMPTY_MACROVALUE , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
 /*
 	TODO PROPERTIES
Index: e:/dev/wxWindows/src/msw/statbmp.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/statbmp.cpp,v
retrieving revision 1.45
diff -u -2 -r1.45 statbmp.cpp
--- e:/dev/wxWindows/src/msw/statbmp.cpp	2003/09/24 00:41:06	1.45
+++ e:/dev/wxWindows/src/msw/statbmp.cpp	2004/02/13 18:02:39
@@ -81,5 +81,5 @@
 
 wxBEGIN_PROPERTIES_TABLE(wxStaticBitmap)
-    wxPROPERTY_FLAGS( WindowStyle , wxStaticBitmapStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
+    wxPROPERTY_FLAGS( WindowStyle , wxStaticBitmapStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , EMPTY_MACROVALUE, 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
 wxEND_PROPERTIES_TABLE()
 
Index: e:/dev/wxWindows/src/msw/statbox.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/statbox.cpp,v
retrieving revision 1.42
diff -u -2 -r1.42 statbox.cpp
--- e:/dev/wxWindows/src/msw/statbox.cpp	2003/09/24 00:41:06	1.42
+++ e:/dev/wxWindows/src/msw/statbox.cpp	2004/02/13 18:02:39
@@ -81,5 +81,5 @@
 wxBEGIN_PROPERTIES_TABLE(wxStaticBox)
 	wxPROPERTY( Label,wxString, SetLabel, GetLabel, wxString() , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
-    wxPROPERTY_FLAGS( WindowStyle , wxStaticBoxStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
+    wxPROPERTY_FLAGS( WindowStyle , wxStaticBoxStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , EMPTY_MACROVALUE, 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
 /*
 	TODO PROPERTIES :
Index: e:/dev/wxWindows/src/msw/statline.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/statline.cpp,v
retrieving revision 1.22
diff -u -2 -r1.22 statline.cpp
--- e:/dev/wxWindows/src/msw/statline.cpp	2003/09/24 00:41:06	1.22
+++ e:/dev/wxWindows/src/msw/statline.cpp	2004/02/13 18:02:40
@@ -86,5 +86,5 @@
 
 wxBEGIN_PROPERTIES_TABLE(wxStaticLine)
-    wxPROPERTY_FLAGS( WindowStyle , wxStaticLineStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
+    wxPROPERTY_FLAGS( WindowStyle , wxStaticLineStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , EMPTY_MACROVALUE, 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
 wxEND_PROPERTIES_TABLE()
 
Index: e:/dev/wxWindows/src/msw/stattext.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/stattext.cpp,v
retrieving revision 1.49
diff -u -2 -r1.49 stattext.cpp
--- e:/dev/wxWindows/src/msw/stattext.cpp	2003/12/02 08:56:47	1.49
+++ e:/dev/wxWindows/src/msw/stattext.cpp	2004/02/13 18:02:40
@@ -75,5 +75,5 @@
 wxBEGIN_PROPERTIES_TABLE(wxStaticText)
 	wxPROPERTY( Label,wxString, SetLabel, GetLabel, wxString() , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
-    wxPROPERTY_FLAGS( WindowStyle , wxStaticTextStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
+    wxPROPERTY_FLAGS( WindowStyle , wxStaticTextStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , EMPTY_MACROVALUE, 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
 wxEND_PROPERTIES_TABLE()
 
Index: e:/dev/wxWindows/src/msw/textctrl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/textctrl.cpp,v
retrieving revision 1.210
diff -u -2 -r1.210 textctrl.cpp
--- e:/dev/wxWindows/src/msw/textctrl.cpp	2004/02/10 13:02:51	1.210
+++ e:/dev/wxWindows/src/msw/textctrl.cpp	2004/02/13 18:02:45
@@ -163,7 +163,7 @@
     wxEVENT_PROPERTY( TextEnter , wxEVT_COMMAND_TEXT_ENTER , wxCommandEvent )
 
-    wxPROPERTY( Font , wxFont , SetFont , GetFont  ,, 0 /*flags*/ , wxT("Helpstring") , wxT("group") )
+    wxPROPERTY( Font , wxFont , SetFont , GetFont  , EMPTY_MACROVALUE, 0 /*flags*/ , wxT("Helpstring") , wxT("group") )
 	wxPROPERTY( Value , wxString , SetValue, GetValue, wxString() , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
-    wxPROPERTY_FLAGS( WindowStyle , wxTextCtrlStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
+    wxPROPERTY_FLAGS( WindowStyle , wxTextCtrlStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , EMPTY_MACROVALUE , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
 wxEND_PROPERTIES_TABLE()
 
Index: e:/dev/wxWindows/src/msw/treectrl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/treectrl.cpp,v
retrieving revision 1.168
diff -u -2 -r1.168 treectrl.cpp
--- e:/dev/wxWindows/src/msw/treectrl.cpp	2004/02/12 16:33:33	1.168
+++ e:/dev/wxWindows/src/msw/treectrl.cpp	2004/02/13 18:02:51
@@ -527,5 +527,5 @@
     wxEVENT_PROPERTY( TextUpdated , wxEVT_COMMAND_TEXT_UPDATED , wxCommandEvent )
     wxEVENT_RANGE_PROPERTY( TreeEvent , wxEVT_COMMAND_TREE_BEGIN_DRAG , wxEVT_COMMAND_TREE_STATE_IMAGE_CLICK , wxTreeEvent )
-    wxPROPERTY_FLAGS( WindowStyle , wxTreeCtrlStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
+    wxPROPERTY_FLAGS( WindowStyle , wxTreeCtrlStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , EMPTY_MACROVALUE , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
 wxEND_PROPERTIES_TABLE()
 
Index: e:/dev/wxWindows/src/msw/window.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/window.cpp,v
retrieving revision 1.461
diff -u -2 -r1.461 window.cpp
--- e:/dev/wxWindows/src/msw/window.cpp	2004/02/12 09:29:19	1.461
+++ e:/dev/wxWindows/src/msw/window.cpp	2004/02/13 18:03:03
@@ -286,9 +286,9 @@
     // Always constructor Properties first
 
-    wxREADONLY_PROPERTY( Parent,wxWindow*, GetParent,  , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
+    wxREADONLY_PROPERTY( Parent,wxWindow*, GetParent, EMPTY_MACROVALUE , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
 	wxPROPERTY( Id,wxWindowID, SetId, GetId, -1, 0 /*flags*/ , wxT("Helpstring") , wxT("group") )
 	wxPROPERTY( Position,wxPoint, SetPosition , GetPosition, wxPoint(-1,-1) , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // pos
 	wxPROPERTY( Size,wxSize, SetSize, GetSize, wxSize(-1,-1) , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // size
-    wxPROPERTY( WindowStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
+    wxPROPERTY( WindowStyle , long , SetWindowStyleFlag , GetWindowStyleFlag , EMPTY_MACROVALUE , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // style
 
     // Then all relations of the object graph
@@ -298,7 +298,7 @@
    // and finally all other properties
 
-	wxPROPERTY( ExtraStyle , long , SetExtraStyle , GetExtraStyle , , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // extstyle
-	wxPROPERTY( BackgroundColour , wxColour , SetBackgroundColour , GetBackgroundColour , , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // bg
-	wxPROPERTY( ForegroundColour , wxColour , SetForegroundColour , GetForegroundColour , , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // fg
+	wxPROPERTY( ExtraStyle , long , SetExtraStyle , GetExtraStyle , EMPTY_MACROVALUE , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // extstyle
+	wxPROPERTY( BackgroundColour , wxColour , SetBackgroundColour , GetBackgroundColour , EMPTY_MACROVALUE , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // bg
+	wxPROPERTY( ForegroundColour , wxColour , SetForegroundColour , GetForegroundColour , EMPTY_MACROVALUE , 0 /*flags*/ , wxT("Helpstring") , wxT("group")) // fg
 	wxPROPERTY( Enabled , bool , Enable , IsEnabled , wxxVariant((bool)true) , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
 	wxPROPERTY( Shown , bool , Show , IsShown , wxxVariant((bool)true) , 0 /*flags*/ , wxT("Helpstring") , wxT("group"))
Index: e:/dev/wxWindows/include/wx/xti.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/xti.h,v
retrieving revision 1.47
diff -u -2 -r1.47 xti.h
--- e:/dev/wxWindows/include/wx/xti.h	2003/10/19 13:41:04	1.47
+++ e:/dev/wxWindows/include/wx/xti.h	2004/02/13 18:03:51
@@ -50,26 +50,39 @@
 // we will move this later to defs.h
 
-#if !wxCHECK_GCC_VERSION( 3 , 4 )
+#if defined(__GNUC__) && !wxCHECK_GCC_VERSION( 3 , 4 )
 #  define wxUSE_MEMBER_TEMPLATES 0
 #endif
 
-#ifdef _MSC_VER
-#  if _MSC_VER <= 1200
-#    define wxUSE_MEMBER_TEMPLATES 0
-#  endif
+#if defined(_MSC_VER) && _MSC_VER <= 1200
+#  define wxUSE_MEMBER_TEMPLATES 0
+#  define wxUSE_FUNC_TEMPLATE_POINTER 0
 #endif
 
 #ifndef wxUSE_MEMBER_TEMPLATES
-#define wxUSE_MEMBER_TEMPLATES 1
+#  define wxUSE_MEMBER_TEMPLATES 1
+#endif
+
+#ifndef wxUSE_FUNC_TEMPLATE_POINTER
+#  define wxUSE_FUNC_TEMPLATE_POINTER 1
 #endif
 
 #if wxUSE_MEMBER_TEMPLATES
-#define wxTEMPLATED_MEMBER_CALL( method , type ) method<type>()
-#define wxTEMPLATED_MEMBER_FIX( type )
+#  define wxTEMPLATED_MEMBER_CALL( method , type ) method<type>()
+#  define wxTEMPLATED_MEMBER_FIX( type )
 #else
-#define wxTEMPLATED_MEMBER_CALL( method , type ) method((type*)NULL)
-#define wxTEMPLATED_MEMBER_FIX( type ) type* =NULL
+#  define wxTEMPLATED_MEMBER_CALL( method , type ) method((type*)NULL)
+#  define wxTEMPLATED_MEMBER_FIX( type ) type* =NULL
 #endif
 
+#if defined(_MSC_VER) && _MSC_VER <= 1200
+#  define wxTEMPLATED_FUNCTION_FIX( type ) , wxTEMPLATED_MEMBER_FIX(type)
+#  define wxINFUNC_CLASS_TYPE_FIX( type ) typedef type type;
+#else
+#  define wxTEMPLATED_FUNCTION_FIX( type )
+#  define wxINFUNC_CLASS_TYPE_FIX( type )
+#endif
+
+#define EMPTY_MACROVALUE /**/
+
 class WXDLLIMPEXP_BASE wxObject;
 class WXDLLIMPEXP_BASE wxClassInfo;
@@ -81,4 +94,16 @@
 typedef void (wxObject::*wxObjectEventFunction)(wxEvent&);
 
+#if wxUSE_FUNC_TEMPLATE_POINTER
+#  define wxTO_STRING(type) wxToStringConverter<type>  
+#  define wxTO_STRING_IMP(type)
+#  define wxFROM_STRING(type) wxFromStringConverter<type>
+#  define wxFROM_STRING_IMP(type)
+#else
+#  define wxTO_STRING(type) ToString##type
+#  define wxTO_STRING_IMP(type) inline void ToString##type( const wxxVariant& data , wxString &result ) { wxToStringConverter<type>(data, result); }
+#  define wxFROM_STRING(type) FromString##type
+#  define wxFROM_STRING_IMP(type) inline void FromString##type( const wxString& data , wxxVariant &result ) { wxFromStringConverter<type>(data, result); }
+#endif
+
 // ----------------------------------------------------------------------------
 // Enum Support
@@ -150,5 +175,7 @@
     void FromLong##e( long data , wxxVariant& result ) { result = wxxVariant((e)data) ;} \
     void ToLong##e( const wxxVariant& data , long &result ) { result = (long) data.wxTEMPLATED_MEMBER_CALL(Get , e) ;} \
-    wxEnumTypeInfo s_typeInfo##e(wxT_ENUM , &s_enumData##e , &wxToStringConverter<e> , &wxFromStringConverter<e> , &ToLong##e , &FromLong##e , typeid(e).name() ) ;
+    wxTO_STRING_IMP( e ) \
+    wxFROM_STRING_IMP( e ) \
+    wxEnumTypeInfo s_typeInfo##e(wxT_ENUM , &s_enumData##e , &wxTO_STRING( e ) , &wxFROM_STRING( e ) , &ToLong##e , &FromLong##e , typeid(e).name() ) ;
 
 // ----------------------------------------------------------------------------
@@ -235,5 +262,7 @@
     void FromLong##SetName( long data , wxxVariant& result ) { result = wxxVariant(SetName((unsigned long)data)) ;} \
     void ToLong##SetName( const wxxVariant& data , long &result ) { result = (long) data.wxTEMPLATED_MEMBER_CALL(Get , SetName).to_ulong() ;} \
-    wxEnumTypeInfo s_typeInfo##SetName(wxT_SET , &s_enumData##e , &wxToStringConverter<SetName> , &wxFromStringConverter<SetName> , &ToLong##SetName , &FromLong##SetName, typeid(SetName).name() ) ;  \
+    wxTO_STRING_IMP( SetName ) \
+    wxFROM_STRING_IMP( SetName ) \
+    wxEnumTypeInfo s_typeInfo##SetName(wxT_SET , &s_enumData##e , &wxTO_STRING( SetName ) , &wxFROM_STRING( SetName ) , &ToLong##SetName , &FromLong##SetName, typeid(SetName).name() ) ;  \
 }
 
@@ -300,5 +329,7 @@
     void FromLong##e( long data , wxxVariant& result ) { result = wxxVariant(e(data)) ;} \
     void ToLong##e( const wxxVariant& data , long &result ) { result = (long) data.wxTEMPLATED_MEMBER_CALL(Get , e).m_data ;} \
-    wxEnumTypeInfo s_typeInfo##e(wxT_SET , &s_enumData##e , &wxToStringConverter<e> , &wxFromStringConverter<e> , &ToLong##e , &FromLong##e, typeid(e).name()  ) ;
+    wxTO_STRING_IMP( e ) \
+    wxFROM_STRING_IMP( e ) \
+    wxEnumTypeInfo s_typeInfo##e(wxT_SET , &s_enumData##e , &wxTO_STRING( e ) , &wxFROM_STRING( e ) , &ToLong##e , &FromLong##e, typeid(e).name()  ) ;
 // ----------------------------------------------------------------------------
 // Type Information
@@ -533,6 +564,13 @@
 // this macro is for usage with custom, non-object derived classes and structs, wxPoint is such a custom type
 
+#if wxUSE_FUNC_TEMPLATE_POINTER
 #define wxCUSTOM_TYPE_INFO( e , toString , fromString ) \
     wxCustomTypeInfo s_typeInfo##e(typeid(e).name() , &toString , &fromString) ;
+#else
+#define wxCUSTOM_TYPE_INFO( e , toString , fromString ) \
+    void ToString##e( const wxxVariant& data , wxString &result ) { toString(data, result); } \
+    void FromString##e( const wxString& data , wxxVariant &result ) { fromString(data, result); } \
+    wxCustomTypeInfo s_typeInfo##e(typeid(e).name() , &ToString##e , &FromString##e) ;
+#endif
 
 #define wxCOLLECTION_TYPE_INFO( element , collection ) \
@@ -601,5 +639,5 @@
     {
         wxxVariantDataT<T> *dataptr = dynamic_cast<wxxVariantDataT<T>*> (m_data) ;
-        wxASSERT_MSG( dataptr , wxT("Cast not possible") ) ;
+        wxASSERT_MSG( dataptr , wxString::Format(wxT("Cast to %s not possible"), typeid(T).name()) ) ;
         return dataptr->Get() ;
     }
@@ -609,5 +647,5 @@
     {
         const wxxVariantDataT<T> *dataptr = dynamic_cast<const wxxVariantDataT<T>*> (m_data) ;
-        wxASSERT_MSG( dataptr , wxT("Cast not possible") ) ;
+        wxASSERT_MSG( dataptr , wxString::Format(wxT("Cast to %s not possible"), typeid(T).name()) ) ;
         return dataptr->Get() ;
     }
@@ -615,5 +653,5 @@
     bool IsEmpty() const { return m_data == NULL ; }
 
-    template<typename T> bool HasData() const
+    template<typename T> bool HasData(wxTEMPLATED_MEMBER_FIX(T)) const
     {
         const wxxVariantDataT<T> *dataptr = dynamic_cast<const wxxVariantDataT<T>*> (m_data) ;
@@ -667,8 +705,8 @@
 
 template<typename T>
-void wxToStringConverter( const wxxVariant &v, wxString &s) { wxStringWriteValue( s , v.wxTEMPLATED_MEMBER_CALL(Get , T) ) ; }
+void wxToStringConverter( const wxxVariant &v, wxString &s wxTEMPLATED_FUNCTION_FIX(T)) { wxStringWriteValue( s , v.wxTEMPLATED_MEMBER_CALL(Get , T) ) ; }
 
 template<typename T>
-void wxFromStringConverter( const wxString &s, wxxVariant &v) { T d ; wxStringReadValue( s , d ) ; v = wxxVariant(d) ; } \
+void wxFromStringConverter( const wxString &s, wxxVariant &v wxTEMPLATED_FUNCTION_FIX(T)) { T d ; wxStringReadValue( s , d ) ; v = wxxVariant(d) ; } 
 
 // ----------------------------------------------------------------------------
@@ -727,9 +765,9 @@
 
 
-
 #define wxSETTER( property, Klass, valueType, setterMethod ) \
 class wxSetter##property : public wxSetter \
 { \
 public: \
+    wxINFUNC_CLASS_TYPE_FIX(Klass) \
     wxSetter##property() : wxSetter( wxT(#setterMethod) ) {} \
     ~wxSetter##property() {} \
@@ -737,5 +775,5 @@
 { \
     Klass *obj = dynamic_cast<Klass*>(object) ;  \
-    if ( variantValue.HasData<valueType>() ) \
+    if ( variantValue.wxTEMPLATED_MEMBER_CALL(HasData, valueType) ) \
     obj->setterMethod(variantValue.wxTEMPLATED_MEMBER_CALL(Get , valueType)) ; \
             else \
@@ -748,4 +786,5 @@
 { \
 public : \
+    wxINFUNC_CLASS_TYPE_FIX(Klass) \
     wxGetter##property() : wxGetter( wxT(#gettermethod) ) {} \
     ~wxGetter##property() {} \
@@ -761,4 +800,5 @@
 { \
 public: \
+    wxINFUNC_CLASS_TYPE_FIX(Klass) \
     wxAdder##property() : wxAdder( wxT(#addermethod) ) {} \
     ~wxAdder##property() {} \
@@ -766,5 +806,5 @@
 { \
     Klass *obj = dynamic_cast<Klass*>(object) ;  \
-    if ( variantValue.HasData<valueType>() ) \
+    if ( variantValue.wxTEMPLATED_MEMBER_CALL(HasData, valueType) ) \
     obj->addermethod(variantValue.wxTEMPLATED_MEMBER_CALL(Get , valueType)) ; \
             else \
@@ -777,4 +817,5 @@
 { \
 public : \
+    wxINFUNC_CLASS_TYPE_FIX(Klass) \
     wxCollectionGetter##property() : wxCollectionGetter( wxT(#gettermethod) ) {} \
     ~wxCollectionGetter##property() {} \
@@ -1087,5 +1128,5 @@
     static wxPropertyInfo _propertyInfo##pname( first , class_t::GetClassInfoStatic() , wxT(#pname) , typeid(void).name() ,NULL , wxxVariant() , wxPROP_DONT_STREAM , wxEmptyString , wxEmptyString ) ;
 
-#define wxPROPERTY( pname , type , setter , getter ,defaultValue , flags , help , group) \
+#define wxPROPERTY( pname , type , setter , getter , defaultValue , flags , help , group) \
     wxSETTER( pname , class_t , type , setter ) \
     static wxSetter##pname _setter##pname ; \

 	  	 
