Index: src/generic/statusbr.cpp
===================================================================
--- src/generic/statusbr.cpp	(révision 51420)
+++ src/generic/statusbr.cpp	(copie de travail)
@@ -385,8 +385,11 @@
     SetBackgroundColour(*wxLIGHT_GREY);
     SetForegroundColour(*wxBLACK);
 #else
-    m_mediumShadowPen = *wxGREY_PEN;
-    m_hilightPen = *wxWHITE_PEN;
+    wxColour mediumShadowColour(wxSystemSettings::GetColour(wxSYS_COLOUR_3DSHADOW));
+    m_mediumShadowPen = wxPen(mediumShadowColour, 1, wxSOLID);
+
+    wxColour hilightColour(wxSystemSettings::GetColour(wxSYS_COLOUR_3DLIGHT));
+    m_hilightPen = wxPen(hilightColour, 1, wxSOLID);
 #endif
 }
 

 	  	 
