Index: src/generic/wizard.cpp
===================================================================
--- src/generic/wizard.cpp	(revision 65651)
+++ src/generic/wizard.cpp	(working copy)
@@ -563,11 +563,8 @@
             m_sizerBmpAndPage->Detach(m_page);
     }
 
-    // set the new page
-    m_page = page;
-
     // is this the end?
-    if ( !m_page )
+    if ( !page )
     {
         // terminate successfully
         if ( IsModal() )
@@ -582,12 +579,18 @@
 
         // and notify the user code (this is especially useful for modeless
         // wizards)
-        wxWizardEvent event(wxEVT_WIZARD_FINISHED, GetId(), false, 0);
+        wxWizardEvent event(wxEVT_WIZARD_FINISHED, GetId(), false, m_page);
         (void)GetEventHandler()->ProcessEvent(event);
 
+        m_page = NULL;
+
         return true;
     }
 
+    // set the new page here so that wxEVT_WIZARD_FINISHED event above
+    // returns the current page in GetPage()
+    m_page = page;
+
     // position and show the new page
     (void)m_page->TransferDataToWindow();
 
