Index: src/stc/stc.cpp
===================================================================
--- src/stc/stc.cpp	(revision 64412)
+++ src/stc/stc.cpp	(working copy)
@@ -273,6 +273,11 @@
     SendMsg(2003, pos, (sptr_t)(const char*)wx2stc(text));
 }
 
+// Add a character at the current position and advance the cursor
+void wxStyledTextCtrl::AddChar(int key) {
+    m_swx->DoAddChar(key);
+}
+
 // Delete all text in the document.
 void wxStyledTextCtrl::ClearAll()
 {
Index: include/wx/stc/stc.h
===================================================================
--- include/wx/stc/stc.h	(revision 64412)
+++ include/wx/stc/stc.h	(working copy)
@@ -2245,6 +2245,9 @@
     // Insert string at a position.
     void InsertText(int pos, const wxString& text);
 
+    // Add a character at the current position and advance the cursor
+    void AddChar(int key);  
+
     // Delete all text in the document.
     void ClearAll();
 
