 src/generic/textdlgg.cpp | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/generic/textdlgg.cpp b/src/generic/textdlgg.cpp
index ce24c6f..6a342be 100644
--- a/src/generic/textdlgg.cpp
+++ b/src/generic/textdlgg.cpp
@@ -71,10 +71,11 @@ bool wxTextEntryDialog::Create(wxWindow *parent,
                                      long style,
                                      const wxPoint& pos)
 {
-    if ( !wxDialog::Create(GetParentForModalDialog(parent, style),
+    long dialog_style = wxDEFAULT_DIALOG_STYLE | wxRESIZE_BORDER;
+    if ( !wxDialog::Create(GetParentForModalDialog(parent, dialog_style),
                            wxID_ANY, caption,
                            pos, wxDefaultSize,
-                           wxDEFAULT_DIALOG_STYLE | wxRESIZE_BORDER) )
+                           dialog_style) )
     {
         return false;
     }
