diff -rupN wxGTK-2.8.12.orig/src/gtk/dcscreen.cpp wxGTK-2.8.12/src/gtk/dcscreen.cpp
--- wxGTK-2.8.12.orig/src/gtk/dcscreen.cpp	2011-03-22 13:18:57.000000000 +0100
+++ wxGTK-2.8.12/src/gtk/dcscreen.cpp	2011-05-31 15:12:55.000000000 +0200
@@ -59,6 +59,9 @@ wxScreenDC::wxScreenDC()
 
 wxScreenDC::~wxScreenDC()
 {
+    if (m_context)
+        g_object_unref (m_context);
+    
     gdk_gc_set_subwindow( m_penGC, GDK_CLIP_BY_CHILDREN );
     gdk_gc_set_subwindow( m_brushGC, GDK_CLIP_BY_CHILDREN );
     gdk_gc_set_subwindow( m_textGC, GDK_CLIP_BY_CHILDREN );
 
