diff --git a/src/osx/glcanvas_osx.cpp b/src/osx/glcanvas_osx.cpp
index 7ec7831..ccea8f4 100755
--- a/src/osx/glcanvas_osx.cpp
+++ b/src/osx/glcanvas_osx.cpp
@@ -391,7 +391,7 @@ wxGLContext::wxGLContext(wxGLCanvas *win,
             m_isOK = true;
         }
 
-        WXGLDestroyPixelFormat(pf)
+        WXGLDestroyPixelFormat(pf);
     }
     if ( !m_isOK )
         wxLogMessage(_("Couldn't create OpenGL context"));
@@ -466,6 +466,8 @@ bool wxGLCanvas::Create(wxWindow *parent,
                         const wxString& name,
                         const wxPalette& WXUNUSED(palette))
 {
+    m_glFormat = NULL;
+
     // Don't allow an empty list
     if ( !dispAttrs.GetGLAttrs() )
     {
@@ -547,7 +549,7 @@ bool wxGLCanvas::IsAGLMultiSampleAvailable()
 /* static */
 bool wxGLCanvasBase::IsDisplaySupported(const wxGLAttributes& dispAttrs)
 {
-    WXGLPixelFormat testFormat = WXGLChoosePixelFormat(dispAttrs.GetGLAttrs(), dispAttrs.GetSize()) )
+    WXGLPixelFormat testFormat = WXGLChoosePixelFormat(dispAttrs.GetGLAttrs(), dispAttrs.GetSize());
     if ( testFormat )
     {
         WXGLDestroyPixelFormat(testFormat);
