diff -u src/msw/version.rc src/msw/version.rc
--- src/msw/version.rc	Fri Mar 06 12:10:00 2009
+++ src/msw/version.rc	Sun Jul 05 21:41:48 2009
@@ -27,8 +27,12 @@
 
 #ifdef _UNICODE
     #define LANG "04090000"
+    #define LANG2 0x0409
+    #define LANG3 0
 #else
     #define LANG "040904b0"
+    #define LANG2 0x0409
+    #define LANG3 1200
 #endif
 
 1 VERSIONINFO
@@ -58,5 +62,9 @@
             VALUE "ProductVersion", wxVERSION_NUM_DOT_STRING "\0"
             VALUE "SpecialBuild", "\0"
         END
+    END
+    BLOCK "VarFileInfo"
+    BEGIN
+            VALUE "Translation", LANG2, LANG3
     END
 END
