Index: src/osx/carbon/region.cpp
===================================================================
--- src/osx/carbon/region.cpp	(revision 68277)
+++ src/osx/carbon/region.cpp	(working copy)
@@ -191,6 +191,13 @@
         // nothing to do
         return true;
 
+    if (m_refData->GetRefCount() > 1)
+    {
+        wxRegionRefData* ref = (wxRegionRefData*)m_refData;
+        UnRef();
+        m_refData = new wxRegionRefData(*ref);
+    }
+
     verify_noerr( HIShapeOffset( M_REGION , x , y ) ) ;
 
     return true ;
