Index: sizer.cpp
===================================================================
--- sizer.cpp	(revision 52356)
+++ sizer.cpp	(working copy)
@@ -847,7 +847,7 @@
 
 wxSize wxSizer::ComputeFittingClientSize(wxWindow *window)
 {
-    wxCHECK_MSG( window, wxDefaultSize, "window can't be NULL" );
+    wxCHECK_MSG( window, wxDefaultSize, wxT("window can't be NULL") );
 
     return window->WindowToClientSize(ComputeFittingWindowSize(window));
 }

 	  	 
