diff -bru wxWidgets.orig/include/wx/window.h /mnt/winxp/wxWidgets/include/wx/window.h
--- wxWidgets.orig/include/wx/window.h	2012-03-06 11:14:15.000000000 -0800
+++ /mnt/winxp/wxWidgets/include/wx/window.h	2012-03-07 00:48:22.140625000 -0800
@@ -623,9 +623,10 @@
 
     bool HasExtraStyle(int exFlag) const { return (m_exStyle & exFlag) != 0; }
 
+#if WXWIN_COMPATIBILITY_2_8
         // make the window modal (all other windows unresponsive)
-    virtual void MakeModal(bool modal = true);
-
+    wxDEPRECATED( virtual void MakeModal(bool modal = true) );
+#endif
 
     // (primitive) theming support
     // ---------------------------
diff -bru wxWidgets.orig/interface/wx/window.h /mnt/winxp/wxWidgets/interface/wx/window.h
--- wxWidgets.orig/interface/wx/window.h	2012-03-06 11:14:31.000000000 -0800
+++ /mnt/winxp/wxWidgets/interface/wx/window.h	2012-03-07 01:55:32.453125000 -0800
@@ -3172,17 +3172,6 @@
     */
     virtual bool IsTopLevel() const;
 
-    /**
-        Disables all other windows in the application so that
-        the user can only interact with this window.
-
-        @param modal
-            If @true, this call disables all other windows in the application so that
-            the user can only interact with this window. If @false, the effect is
-            reversed.
-    */
-    virtual void MakeModal(bool modal = true);
-
     
     /**
         This virtual function is normally only used internally, but
diff -bru wxWidgets.orig/src/common/wincmn.cpp /mnt/winxp/wxWidgets/src/common/wincmn.cpp
--- wxWidgets.orig/src/common/wincmn.cpp	2012-03-06 11:13:06.000000000 -0800
+++ /mnt/winxp/wxWidgets/src/common/wincmn.cpp	2012-03-07 00:48:22.359375000 -0800
@@ -1915,7 +1915,7 @@
 // ----------------------------------------------------------------------------
 // dialog oriented functions
 // ----------------------------------------------------------------------------
-
+#if WXWIN_COMPATIBILITY_2_8
 void wxWindowBase::MakeModal(bool modal)
 {
     // Disable all other windows
@@ -1932,6 +1932,7 @@
         }
     }
 }
+#endif
 
 bool wxWindowBase::Validate()
 {
