? mypatch.patch
Index: maskededit.py
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/wxPython/wx/lib/masked/maskededit.py,v
retrieving revision 1.12
diff -b -u -2 -r1.12 maskededit.py
--- maskededit.py	2004/10/22 18:51:03	1.12
+++ maskededit.py	2004/11/16 20:41:33
@@ -5254,5 +5254,5 @@
 ##                        dbg('cannot convert string to valid time')
                         pass
-        if valid: dbg('valid date')
+##        if valid: dbg('valid date')
 ##        dbg(indent=0)
         return valid

 	  	 
