Index: include/wx/afterstd.h
===================================================================
--- include/wx/afterstd.h	(revision 61508)
+++ include/wx/afterstd.h	(working copy)
@@ -46,3 +46,14 @@
     #endif
 #endif
 
+#if defined(__SUNPRO_CC) || defined(__SUNPRO_C)
+#ifndef _T
+#if !wxUSE_UNICODE
+#define _T(x) x
+#else /* Unicode */
+/* use wxCONCAT_HELPER so that x could be expanded if it's a macro */
+#define _T(x) wxCONCAT_HELPER(L, x)
+#endif /* ASCII/Unicode */
+#endif /* !defined(_T) */
+
+#endif /* defined(__SUNPRO_CC) || defined(__SUNPRO_C) */
Index: include/wx/beforestd.h
===================================================================
--- include/wx/beforestd.h	(revision 61508)
+++ include/wx/beforestd.h	(working copy)
@@ -63,3 +63,7 @@
     #pragma warning(disable:4786)
 #endif // VC++ < 7
 
+#if defined(__SUNPRO_CC) || defined(__SUNPRO_C)
+#undef _T
+#endif // SUNCC
+
