Index: src/msw/gauge.cpp
===================================================================
--- src/msw/gauge.cpp	(revision 62459)
+++ src/msw/gauge.cpp	(working copy)
@@ -243,7 +243,11 @@
     // add the PBS_MARQUEE style to the progress bar
     LONG style = ::GetWindowLong(GetHwnd(), GWL_STYLE);
     if ((style & PBS_MARQUEE) == 0)
+    {
         ::SetWindowLong(GetHwnd(), GWL_STYLE, style|PBS_MARQUEE);
+        SendMessage(GetHwnd(), (UINT) PBM_SETSTEP, (WPARAM)10, (LPARAM)0);
+        SendMessage(GetHwnd(), (UINT) PBM_SETMARQUEE, (WPARAM)1, (LPARAM)60000);
+    }
 
     // now the control can only run in indeterminate mode
 }
@@ -268,7 +272,7 @@
         // NOTE: when in indeterminate mode, the PBM_SETPOS message will just make
         //       the bar's blocks move a bit and the WPARAM value is just ignored
         //       so that we can safely use zero
-        SendMessage(GetHwnd(), (UINT) PBM_SETPOS, (WPARAM)0, (LPARAM)0);
+        SendMessage(GetHwnd(), (UINT) PBM_STEPIT, (WPARAM)0, (LPARAM)0);
     }
     else
     {
