Index: configure
===================================================================
--- configure	(revision 65537)
+++ configure	(working copy)
@@ -35059,7 +35059,7 @@
             DYLIB_RPATH_POSTLINK="${HOST_PREFIX}install_name_tool -id \$@ \$@"
             cat <<EOF >change-install-names
 #!/bin/sh
-libnames=\`cd \${1} ; ls -1 | grep '\.[0-9][0-9]*\.dylib\$'\`
+libnames=\`cd \${2} ; ls -1 | grep '\.[0-9][0-9]*\.dylib\$'\`
 for i in \${libnames} ; do
     ${HOST_PREFIX}install_name_tool -id \${1}/\${i} \${1}/\${i}
     for dep in \${libnames} ; do
Index: configure.in
===================================================================
--- configure.in	(revision 65537)
+++ configure.in	(working copy)
@@ -3985,7 +3985,7 @@
             DYLIB_RPATH_POSTLINK="${HOST_PREFIX}install_name_tool -id \$@ \$@"
             cat <<EOF >change-install-names
 #!/bin/sh
-libnames=\`cd \${1} ; ls -1 | grep '\.[[0-9]][[0-9]]*\.dylib\$'\`
+libnames=\`cd \${2} ; ls -1 | grep '\.[[0-9]][[0-9]]*\.dylib\$'\`
 for i in \${libnames} ; do
     ${HOST_PREFIX}install_name_tool -id \${1}/\${i} \${1}/\${i}
     for dep in \${libnames} ; do
