Index: src/gtk/window.cpp
===================================================================
--- src/gtk/window.cpp	(revision 902)
+++ src/gtk/window.cpp	(working copy)
@@ -2931,6 +2931,27 @@
         dx = pizza->m_scroll_x;
         dy = pizza->m_scroll_y;
     }
+    if (m_x == -1 && m_y == -1)
+    {
+        GdkWindow *source = (GdkWindow *) NULL;
+        //if (m_wxwindow)
+            //source = WX_PIZZA(m_wxwindow)->bin_window;
+        //else
+            source = m_widget->window;
+
+        if (source)
+        {
+            int org_x = 0;
+            int org_y = 0;
+            gdk_window_get_origin( source, &org_x, &org_y );
+
+            if (GetParent())
+                GetParent()->ScreenToClient(&org_x, &org_y);
+
+            wx_const_cast(wxWindowGTK*, this)->m_x = org_x;
+            wx_const_cast(wxWindowGTK*, this)->m_y = org_y;
+        }
+    }
     if (x) (*x) = m_x - dx;
     if (y) (*y) = m_y - dy;
 }
