Index: include/wx/msw/notifmsg.h
===================================================================
--- include/wx/msw/notifmsg.h	(revisión: 71276)
+++ include/wx/msw/notifmsg.h	(copia de trabajo)
@@ -23,8 +23,9 @@
     wxNotificationMessage() { Init(); }
     wxNotificationMessage(const wxString& title,
                           const wxString& message = wxString(),
-                          wxWindow *parent = NULL)
-        : wxNotificationMessageBase(title, message, parent)
+                          wxWindow *parent = NULL,
+						  int flags = wxICON_INFORMATION)
+        : wxNotificationMessageBase(title, message, parent, flags)
     {
         Init();
     }
