diff --git a/src/msw/dirdlg.cpp b/src/msw/dirdlg.cpp
index b6a3360..8eb595a 100644
--- a/src/msw/dirdlg.cpp
+++ b/src/msw/dirdlg.cpp
@@ -388,15 +388,20 @@ int wxDirDialog::ShowIFileDialog(WXHWND owner)
                                                              &folder));
         if ( FAILED(hr) )
         {
-            wxLogApiError(wxS("SHCreateItemFromParsingName"), hr);
-            return wxID_NONE;
+            if ( hr != HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND) )
+            {
+                wxLogApiError(wxS("SHCreateItemFromParsingName"), hr);
+                return wxID_NONE;
+            }
         }
-
-        hr = fileDialog->SetFolder(folder);
-        if ( FAILED(hr) )
+        else
         {
-            wxLogApiError(wxS("IFileDialog::SetFolder"), hr);
-            return wxID_NONE;
+            hr = fileDialog->SetFolder(folder);
+            if ( FAILED(hr) )
+            {
+                wxLogApiError(wxS("IFileDialog::SetFolder"), hr);
+                return wxID_NONE;
+            }
         }
     }
 
-- 
1.8.3.msysgit.0

