cvs diff -u datectlg.cpp (in directory C:\wxWidgets-2.6\src\generic\)
Index: datectlg.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/generic/datectlg.cpp,v
retrieving revision 1.33
diff -u -r1.33 datectlg.cpp
--- datectlg.cpp	2005/04/16 11:07:30	1.33
+++ datectlg.cpp	2005/04/26 12:26:06
@@ -888,18 +888,7 @@
     wxString txt = m_txt->GetValue();
     wxDateTime dt;
     if (!txt.empty())
-    {
         dt.ParseFormat(txt, m_format);
-        if (!dt.IsValid())
-            return;
-    }
-
-    wxCalendarEvent cev(m_cal, wxEVT_CALENDAR_SEL_CHANGED);
-    cev.SetEventObject(this);
-    cev.SetId(GetId());
-    cev.SetDate(dt);
-
-    GetParent()->ProcessEvent(cev);
 
     wxDateEvent dev(this, dt, wxEVT_DATE_CHANGED);
     GetParent()->ProcessEvent(dev);


 	  	 
