Index: samples/minimal/minimal.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/samples/minimal/minimal.cpp,v
retrieving revision 1.76
diff -u -r1.76 minimal.cpp
--- samples/minimal/minimal.cpp	2006/12/11 09:33:34	1.76
+++ samples/minimal/minimal.cpp	2007/03/21 15:41:42
@@ -141,11 +141,32 @@
 // main frame
 // ----------------------------------------------------------------------------
 
+#include <wx/fileconf.h>
+
 // frame constructor
 MyFrame::MyFrame(const wxString& title)
        : wxFrame(NULL, wxID_ANY, title)
 {
-    // set the frame icon
+  wxFileConfig ini( wxT("INI"), wxT("INI"), wxT("/INI.ini"), wxEmptyString, wxCONFIG_USE_LOCAL_FILE, wxConvUTF8 );
+  ini.Write( wxT("Config/A"), wxT("1") );
+  for ( int i = 1; i < 5; i++ )
+    ini.Write( wxString::Format( wxT("User/U%d"), i ), wxT("test-value") );
+  ini.Flush();
+  wxMessageBox( wxT("Check the INI file now. (1)") );
+
+  ini.DeleteGroup( wxT("User") );
+  ini.Flush();
+  wxMessageBox( wxT("Check the INI file now. (2)") );
+
+  ini.Write( wxT("User/U0"), wxT("test-value2") );
+  ini.Flush();
+  wxMessageBox( wxT("Check the INI file now. (3)") );
+
+  // set the frame icon
     SetIcon(wxICON(sample));
 
 #if wxUSE_MENUS


 	  	 
