Index: wxWidgets-2.8/src/html/m_layout.cpp
===================================================================
--- wxWidgets-2.8/src/html/m_layout.cpp	(revision 55439)
+++ wxWidgets-2.8/src/html/m_layout.cpp	(working copy)
@@ -108,8 +108,18 @@
     // m_PosY is only the vertical offset from the parent. The pagebreak
     // required here is the total page offset, so m_PosY must be added
     // to the parent's offset and height.
-    int total_height = m_PosY + GetParent()->GetPosY() + GetParent()->GetHeight();
+    wxHtmlCell *parent = GetParent();
 
+    int total_height = m_PosY;
+
+    do
+    {
+      total_height += parent->GetPosY();
+      parent = parent->GetParent();
+    }
+    while (parent != NULL);
+
+
     // Search the array of pagebreaks to see whether we've already set
     // a pagebreak here. The standard bsearch() function is appropriate
     // because the array of pagebreaks through known_pagebreaks[number_of_pages]
