--- src/aui/auibar.cpp	Mon Nov 11 15:18:38 2013
+++ src/aui/auibar.cpp	Thu Aug 04 12:58:32 2016
@@ -2546,7 +2546,7 @@
                 {
                     wxCommandEvent event(wxEVT_MENU, res);
                     event.SetEventObject(this);
-                    GetParent()->GetEventHandler()->ProcessEvent(event);
+                    GetEventHandler()->ProcessEvent(event);
                 }
             }
 
