Index: htmlwin.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/html/htmlwin.h,v
retrieving revision 1.48
diff -b -u -2 -r1.48 htmlwin.h
--- htmlwin.h	2003/08/09 12:37:42	1.48
+++ htmlwin.h	2003/08/31 12:00:39
@@ -163,7 +163,4 @@
     static void AddGlobalProcessor(wxHtmlProcessor *processor);
 
-    // what would we do with it?
-    virtual bool AcceptsFocusFromKeyboard() const { return FALSE; }
-
     // -- Callbacks --
 

 	  	 
