diff --git a/src/aui/framemanager.cpp b/src/aui/framemanager.cpp
index 5266991..ba3d132 100644
--- a/src/aui/framemanager.cpp
+++ b/src/aui/framemanager.cpp
@@ -265,12 +265,24 @@ public:
 
         g_signal_connect( m_widget, "realize",
                       G_CALLBACK (gtk_pseudo_window_realized_callback), this );
-
-        GdkColor col;
-        col.red = 128 * 256;
-        col.green = 192 * 256;
-        col.blue = 255 * 256;
-        gtk_widget_modify_bg( m_widget, GTK_STATE_NORMAL, &col );
+#if GTK_CHECK_VERSION( 3, 0, 0 ) && !GTK_CHECK_VERSION( 3, 16, 0 )
+        if( gtk_check_version( 3, 0, 0 ) && !gtk_check_version( 3, 16, 0 ) )
+       {
+           GdkRGBA col;
+           col.red = 128 * 256; col.green = 192.256; col.blue = 255 * 256; col.alpha = 1;
+           gtk_widget_override_background_color( m_widget, GTK_STATE_FLAG_NORMAL, &col );
+       }
+#endif
+        if( !gtk_check_version( 3, 0, 0 ) )
+       {
+           wxGCC_WARNING_SUPPRESS(deprecated-declarations)
+            GdkColor col;
+            col.red = 128 * 256;
+            col.green = 192 * 256;
+            col.blue = 255 * 256;
+            gtk_widget_modify_bg( m_widget, GTK_STATE_NORMAL, &col );
+           wxGCC_WARNING_RESTORE()
+       }
     }
 
     bool SetTransparent(wxByte WXUNUSED(alpha)) wxOVERRIDE
