Index: src/msw/fdrepdlg.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/msw/fdrepdlg.cpp,v
retrieving revision 1.22
diff --unified -r1.22 fdrepdlg.cpp
--- src/msw/fdrepdlg.cpp	2005/12/22 23:31:51	1.22
+++ src/msw/fdrepdlg.cpp	2006/02/10 02:24:24
@@ -190,7 +190,7 @@
 
     // check that we don't subclass the parent twice: this would be a bad idea
     // as then we'd have infinite recursion in wxFindReplaceWindowProc
-    wxCHECK_RET( wxGetWindowProc(hwnd) != wxFindReplaceWindowProc,
+    wxCHECK_RET( wxGetWindowProc(hwnd) != reinterpret_cast<void*> (wxFindReplaceWindowProc),
                  _T("can't have more than one find dialog currently") );
 
     // set the new one and save the old as user data to allow access to it

 	  	 
