Index: src/osx/cocoa/menuitem.mm
===================================================================
--- src/osx/cocoa/menuitem.mm	(revision 66181)
+++ src/osx/cocoa/menuitem.mm	(working copy)
@@ -89,8 +89,50 @@
         {
             switch ( key )
             {
+                // (joostn) I've tested the left and right arrow shortcuts, they work:
+                case WXK_LEFT :
+                    modifiers |= NSFunctionKeyMask ;
+                    shortcut = NSLeftArrowFunctionKey ;
+                    break ;
+
+                case WXK_RIGHT :
+                    modifiers |= NSFunctionKeyMask ;
+                    shortcut = NSRightArrowFunctionKey ;
+                    break ;
+                    
+                // TODO (joostn) The following were taken from NSEvent.h but I haven't tested them:
+                case WXK_UP :
+                    modifiers |= NSFunctionKeyMask ;
+                    shortcut = NSUpArrowFunctionKey ;
+                    break ;
+
+                case WXK_DOWN :
+                    modifiers |= NSFunctionKeyMask ;
+                    shortcut = NSDownArrowFunctionKey ;
+                    break ;
+
+                case WXK_PAGEUP :
+                    modifiers |= NSFunctionKeyMask ;
+                    shortcut = NSPageUpFunctionKey ;
+                    break ;
+                    
+                case WXK_PAGEDOWN :
+                    modifiers |= NSFunctionKeyMask ;
+                    shortcut = NSPageDownFunctionKey ;
+                    break ;
+
+                case WXK_HOME :
+                    modifiers |= NSFunctionKeyMask ;
+                    shortcut = NSHomeFunctionKey ;
+                    break ;
+                    
+                case WXK_END :
+                    modifiers |= NSFunctionKeyMask ;
+                    shortcut = NSEndFunctionKey ;
+                    break ;
+
 /*
-                // standard function keys from here
+                // TODO implement these:
                 case WXK_TAB :
                     modifiers |= NSFunctionKeyMask ;
                     shortcut = NSTabCharacter ;
@@ -115,45 +157,11 @@
                     shortcut = ' ' ;
                     break ;
 
-
                 case WXK_CLEAR :
                     cocoaKey = kClearCharCode ;
                     break ;
 
-                case WXK_PAGEUP :
-                    cocoaKey = kPageUpCharCode ;
-                    break ;
-
-                case WXK_PAGEDOWN :
-                    cocoaKey = kPageDownCharCode ;
-                    break ;
-
-                case WXK_LEFT :
-                    cocoaKey = kLeftArrowCharCode ;
-                    break ;
-
-                case WXK_UP :
-                    cocoaKey = kUpArrowCharCode ;
-                    break ;
-
-                case WXK_RIGHT :
-                    cocoaKey = kRightArrowCharCode ;
-                    break ;
-
-                case WXK_DOWN :
-                    cocoaKey = kDownArrowCharCode ;
-                    break ;
-
-                case WXK_HOME :
-                    cocoaKey = kHomeCharCode ;
-                    break ;
-
-                case WXK_END :
-                    cocoaKey = kEndCharCode ;
-                    break ;
 */
-                // TODO Test all above with their function key equiv.
-                // from NSEvent.h
                 default :
                     if(entry->GetFlags() & wxACCEL_SHIFT)
                         shortcut = toupper(key);
