Index: wxWidgets-2.8/src/msw/mdi.cpp
===================================================================
--- wxWidgets-2.8/src/msw/mdi.cpp	(revision 53750)
+++ wxWidgets-2.8/src/msw/mdi.cpp	(working copy)
@@ -740,7 +740,9 @@
   else
       mcs.cy = CW_USEDEFAULT;
 
-  DWORD msflags = WS_OVERLAPPED | WS_CLIPCHILDREN;
+  DWORD msflags = WS_OVERLAPPED;
+  if (style & wxCLIP_CHILDREN)
+    msflags |= WS_CLIPCHILDREN;
   if (style & wxMINIMIZE_BOX)
     msflags |= WS_MINIMIZEBOX;
   if (style & wxMAXIMIZE_BOX)
