Index: minimal.cpp
===================================================================
--- minimal.cpp	(revision 63970)
+++ minimal.cpp	(working copy)
@@ -30,6 +30,9 @@
     #include "wx/wx.h"
 #endif
 
+#include "wx/colordlg.h"
+#include "wx/fontdlg.h"
+
 // ----------------------------------------------------------------------------
 // resources
 // ----------------------------------------------------------------------------
@@ -131,6 +134,13 @@
     // created initially)
     frame->Show(true);
 
+    wxColourDialog color_dlg(frame);
+    color_dlg.ShowModal();
+
+    wxFontData font_data;
+    wxFontDialog font_dlg(frame, font_data);
+    font_dlg.ShowModal();
+
     // success: wxApp::OnRun() will be called which will enter the main message
     // loop and the application will run. If we returned false here, the
     // application would exit immediately.
