Index: include/wx/version.h
===================================================================
--- include/wx/version.h	(revision 65591)
+++ include/wx/version.h	(working copy)
@@ -28,9 +28,9 @@
 /*  NB: this file is parsed by automatic tools so don't change its format! */
 #define wxMAJOR_VERSION      2
 #define wxMINOR_VERSION      8
-#define wxRELEASE_NUMBER     11
+#define wxRELEASE_NUMBER     12
 #define wxSUBRELEASE_NUMBER  0
-#define wxVERSION_STRING   wxT("wxWidgets 2.8.11")
+#define wxVERSION_STRING   wxT("wxWidgets 2.8.12")
 
 /*  nothing to update below this line when updating the version */
 /*  ---------------------------------------------------------------------------- */
