Index: e:/dev/wxWindows/src/generic/renderg.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/generic/renderg.cpp,v
retrieving revision 1.18
diff -u -2 -r1.18 renderg.cpp
--- e:/dev/wxWindows/src/generic/renderg.cpp	2003/08/11 12:49:15	1.18
+++ e:/dev/wxWindows/src/generic/renderg.cpp	2003/10/23 17:28:09
@@ -242,14 +242,15 @@
             border;
 
-    if ( win->HasFlag(wxSP_3D) )
-    {
+    if ( win->HasFlag(wxSP_3DSASH) )
         sashWidth = 7;
-        border = 2;
-    }
+    else if ( win->HasFlag(wxSP_NOSASH) )
+        sashWidth = 0;
     else // no 3D effect
-    {
         sashWidth = 3;
+
+    if ( win->HasFlag(wxSP_3DBORDER) )
+        border = 2;
+    else // no 3D effect
         border = 0;
-    }
 
     return wxSplitterRenderParams(sashWidth, border, false);
@@ -262,5 +263,5 @@
                                       int WXUNUSED(falgs))
 {
-    if ( win->HasFlag(wxSP_3D) )
+    if ( win->HasFlag(wxSP_3DBORDER) )
     {
         wxRect rect = rectOrig;
@@ -305,32 +306,34 @@
     wxCoord offset = 0;
     
-    // If we're not drawing the border, droppings will
-    // be left unless we make the sash shorter
-    if ( !win->HasFlag(wxSP_3DBORDER) )
+    // If we're drawing the border, draw the sash 3d lines shorter
+    if ( win->HasFlag(wxSP_3DBORDER) )
     {
-        offset = 3;
+        offset = 1;
     }
 
-    // from left to right
-    if ( win->HasFlag(wxSP_3D) )
+    dc.SetPen(*wxTRANSPARENT_PEN);
+    dc.SetBrush(wxBrush(wxSystemSettings::GetColour(wxSYS_COLOUR_3DFACE)));    
+    
+    if ( win->HasFlag(wxSP_3DSASH) )
     {
+        // Draw the 3D sash
+        dc.DrawRectangle(position + 2, 0, 3, h);
+
         dc.SetPen(m_penLightGrey);
-        dc.DrawLine(position, 1 + offset, position, h - 1 - offset);
+        dc.DrawLine(position, offset, position, h - offset);
 
         dc.SetPen(m_penHighlight);
-        dc.DrawLine(position + 1, offset, position + 1, h - offset);
-    }
-
-    dc.SetPen(*wxTRANSPARENT_PEN);
-    dc.SetBrush(wxBrush(wxSystemSettings::GetColour(wxSYS_COLOUR_3DFACE)));
-    dc.DrawRectangle(position + 2, offset, 3, h - 2*offset);
+        dc.DrawLine(position + 1, 0, position + 1, h);
 
-    if ( win->HasFlag(wxSP_3D) )
-    {
         dc.SetPen(m_penDarkGrey);
-        dc.DrawLine(position + 5, offset, position + 5, h - offset);
+        dc.DrawLine(position + 5, 0, position + 5, h);
 
         dc.SetPen(m_penBlack);
-        dc.DrawLine(position + 6, offset, position + 6, h - 1 - offset);
+        dc.DrawLine(position + 6, offset, position + 6, h - offset);
+    }
+    else
+    {
+        // Draw a flat sash
+        dc.DrawRectangle(position, 0, 3, h);
     }
 }

 	  	 
