Index: dialog.cpp
===================================================================
--- dialog.cpp	(revision 60472)
+++ dialog.cpp	(working copy)
@@ -327,9 +327,11 @@
                                     wxT(""),
                                     WS_CHILD |
                                     WS_CLIPSIBLINGS |
+#if !defined (__WXWINCE__)
                                     SBS_SIZEGRIP |
-                                    SBS_SIZEBOX |
-                                    SBS_SIZEBOXBOTTOMRIGHTALIGN,
+                                    SBS_SIZEBOXBOTTOMRIGHTALIGN |
+#endif
+                                    SBS_SIZEBOX ,
                                     0, 0, 0, 0,
                                     GetHwnd(),
                                     0,
