Index: include/wx/generic/dirctrlg.h
===================================================================
--- include/wx/generic/dirctrlg.h	(revision 72820)
+++ include/wx/generic/dirctrlg.h	(working copy)
@@ -111,6 +111,7 @@
     void OnCollapseItem(wxTreeEvent &event );
     void OnBeginEditItem(wxTreeEvent &event );
     void OnEndEditItem(wxTreeEvent &event );
+    void OnTreeSelChange(wxTreeEvent &event);
     void OnSize(wxSizeEvent &event );
 
     // Try to expand as much of the given path as possible.
@@ -209,6 +210,13 @@
     wxDECLARE_NO_COPY_CLASS(wxGenericDirCtrl);
 };
 
+wxDECLARE_EXPORTED_EVENT( WXDLLIMPEXP_CORE, wxEVT_COMMAND_DIRCTRL_CHANGED, wxTreeEvent );
+
+#define wx__DECLARE_DIRCTRL_EVT(evt, id, fn) \
+    wx__DECLARE_EVT1(wxEVT_COMMAND_DIRCTRL_ ## evt, id, wxTreeEventHandler(fn))
+
+#define EVT_DIRCTRL_CHANGED(id, fn) wx__DECLARE_DIRCTRL_EVT(CHANGED, id, fn)
+
 //-----------------------------------------------------------------------------
 // wxDirFilterListCtrl
 //-----------------------------------------------------------------------------
Index: interface/wx/dirctrl.h
===================================================================
--- interface/wx/dirctrl.h	(revision 72820)
+++ interface/wx/dirctrl.h	(working copy)
@@ -47,6 +47,9 @@
     @library{wxcore}
     @category{ctrl}
     @appearance{genericdirctrl.png}
+    @event{EVT_DIRCTRL_CHANGED(id, func)}
+          Selection has changed.
+          Processes a @c wxEVT_COMMAND_DIRCTRL_CHANGED event type.
 */
 class wxGenericDirCtrl : public wxControl
 {
Index: samples/widgets/dirctrl.cpp
===================================================================
--- samples/widgets/dirctrl.cpp	(revision 72820)
+++ samples/widgets/dirctrl.cpp	(working copy)
@@ -112,6 +112,7 @@
     void OnStdPath(wxCommandEvent& event);
     void OnCheckBox(wxCommandEvent& event);
     void OnRadioBox(wxCommandEvent& event);
+    void OnSelChanged(wxTreeEvent& event);
 
     // reset the control parameters
     void Reset();
@@ -154,6 +155,7 @@
     EVT_BUTTON(DirCtrlPage_SetPath, DirCtrlWidgetsPage::OnButtonSetPath)
     EVT_CHECKBOX(wxID_ANY, DirCtrlWidgetsPage::OnCheckBox)
     EVT_RADIOBOX(wxID_ANY, DirCtrlWidgetsPage::OnRadioBox)
+    EVT_DIRCTRL_CHANGED(DirCtrlPage_Ctrl, DirCtrlWidgetsPage::OnSelChanged)
 END_EVENT_TABLE()
 
 // ============================================================================
@@ -360,4 +362,11 @@
     }
 }
 
+void DirCtrlWidgetsPage::OnSelChanged(wxTreeEvent& event)
+{
+    wxLogMessage("Selection -> \"%s\"",
+                 m_dirCtrl->GetPath(event.GetItem()).wx_str());
+    event.Skip();
+}
+
 #endif // wxUSE_DIRDLG
Index: src/generic/dirctrlg.cpp
===================================================================
--- src/generic/dirctrlg.cpp	(revision 72820)
+++ src/generic/dirctrlg.cpp	(working copy)
@@ -103,6 +103,12 @@
 bool wxIsDriveAvailable(const wxString& dirName);
 
 // ----------------------------------------------------------------------------
+// events
+// ----------------------------------------------------------------------------
+
+wxDEFINE_EVENT( wxEVT_COMMAND_DIRCTRL_CHANGED, wxTreeEvent );
+
+// ----------------------------------------------------------------------------
 // wxGetAvailableDrives, for WINDOWS, DOS, OS2, MAC, UNIX (returns "/")
 // ----------------------------------------------------------------------------
 
@@ -441,6 +447,7 @@
   EVT_TREE_ITEM_COLLAPSED     (wxID_TREECTRL, wxGenericDirCtrl::OnCollapseItem)
   EVT_TREE_BEGIN_LABEL_EDIT   (wxID_TREECTRL, wxGenericDirCtrl::OnBeginEditItem)
   EVT_TREE_END_LABEL_EDIT     (wxID_TREECTRL, wxGenericDirCtrl::OnEndEditItem)
+  EVT_TREE_SEL_CHANGED        (wxID_TREECTRL, wxGenericDirCtrl::OnTreeSelChange)
   EVT_SIZE                    (wxGenericDirCtrl::OnSize)
 END_EVENT_TABLE()
 
@@ -694,6 +701,17 @@
     }
 }
 
+void wxGenericDirCtrl::OnTreeSelChange(wxTreeEvent &event)
+{
+    wxTreeEvent changedEvent(wxEVT_COMMAND_DIRCTRL_CHANGED, GetId());
+
+    changedEvent.SetItem(event.GetItem());
+    changedEvent.SetClientObject(m_treeCtrl->GetItemData(event.GetItem()));
+
+    GetEventHandler()->SafelyProcessEvent(changedEvent);
+    event.Skip();
+}
+
 void wxGenericDirCtrl::OnExpandItem(wxTreeEvent &event)
 {
     wxTreeItemId parentId = event.GetItem();
