Index: src/gtk/app.cpp
===================================================================
RCS file: /home/wxcvs/wxWindows/src/gtk/app.cpp,v
retrieving revision 1.158
diff -c -r1.158 app.cpp
*** src/gtk/app.cpp	7 Feb 2002 14:41:47 -0000	1.158
--- src/gtk/app.cpp	12 Feb 2002 16:55:59 -0000
***************
*** 882,884 ****
--- 882,908 ----
  
  #endif // __WXDEBUG__
  
+ 
+ 
+ // Functions to make wxGTK compatible with console programs
+ 
+ bool wxInitialize() {
+ 
+     // wxApp::SetInitializerFunction( wxConsoleAppInitializerFunction );
+ 
+     static char *dummy_argv[] = { "", NULL };
+     int argc = 1;
+ 
+     int result = wxEntryStart( argc, dummy_argv );
+     if ( result < 0 ) {
+         return false;
+     }
+ 
+     wxTheApp = new wxApp();
+ 
+     return true;
+ }
+ 
+ void wxUninitialize() {
+     wxEntryCleanup();
+ }
Index: include/wx/app.h
===================================================================
RCS file: /home/wxcvs/wxWindows/include/wx/app.h,v
retrieving revision 1.57
diff -c -r1.57 app.h
*** include/wx/app.h	5 Feb 2002 16:34:28 -0000	1.57
--- include/wx/app.h	12 Feb 2002 16:56:00 -0000
***************
*** 434,440 ****
  // and call these functions instead at the program startup and termination
  // -------------------------------------------------------------------------
  
! #if !wxUSE_GUI
  
  // initialize the library (may be called as many times as needed, but each
  // call to wxInitialize() must be matched by wxUninitialize())
--- 434,440 ----
  // and call these functions instead at the program startup and termination
  // -------------------------------------------------------------------------
  
! #if !wxUSE_GUI || defined(__WXGTK__)
  
  // initialize the library (may be called as many times as needed, but each
  // call to wxInitialize() must be matched by wxUninitialize())
***************
*** 465,471 ****
      bool m_ok;
  };
  
! #endif // !wxUSE_GUI
  
  // ----------------------------------------------------------------------------
  // macros for dynamic creation of the application object
--- 465,471 ----
      bool m_ok;
  };
  
! #endif // !wxUSE_GUI || defined(__WXGTK__)
  
  // ----------------------------------------------------------------------------
  // macros for dynamic creation of the application object

 	  	 
