--- include/wx/gtk/dcmemory.h	2002/09/07 12:28:46	1.12
+++ include/wx/gtk/dcmemory.h	2003/04/15 05:20:51
@@ -43,6 +43,7 @@
     // and everything else as drawing 1.
     virtual void SetPen( const wxPen &pen );
     virtual void SetBrush( const wxBrush &brush );
+    virtual void SetBackground( const wxBrush &brush );
     virtual void SetTextForeground( const wxColour &col );
     virtual void SetTextBackground( const wxColour &col );
 
--- src/gtk/dcmemory.cpp	2003/01/26 12:40:44	1.20
+++ src/gtk/dcmemory.cpp	2003/04/15 05:20:58
@@ -83,10 +83,16 @@
 {
     if (m_selected.Ok() && m_selected.GetBitmap() && (*wxTRANSPARENT_PEN != pen))
     {
-        if (*wxWHITE_PEN == pen)
-            wxWindowDC::SetPen( *wxBLACK_PEN );
-        else
-            wxWindowDC::SetPen( *wxWHITE_PEN );
+        wxPen copiedPen( pen );
+        if ( pen.GetColour() == *wxWHITE ) 
+        {
+            copiedPen.SetColour( *wxBLACK );
+        }
+        else 
+        {
+            copiedPen.SetColour( *wxWHITE );
+        }
+        wxWindowDC::SetPen( copiedPen );
     }
     else
     {
@@ -98,14 +104,41 @@
 {
     if (m_selected.Ok() && m_selected.GetBitmap() && (*wxTRANSPARENT_BRUSH != brush))
     {
-        if (*wxWHITE_BRUSH == brush)
-            wxWindowDC::SetBrush( *wxBLACK_BRUSH );
+        wxBrush copiedBrush( brush );
+        if ( copiedBrush.GetColour() == *wxWHITE )
+        {
+            copiedBrush.SetColour( *wxBLACK );
+        }
         else
-            wxWindowDC::SetBrush( *wxWHITE_BRUSH );
+        {
+            copiedBrush.SetColour( *wxWHITE );
+        }
+        wxWindowDC::SetBrush( copiedBrush );
     }
     else
     {
         wxWindowDC::SetBrush( brush );
+    }
+}
+
+void wxMemoryDC::SetBackground( const wxBrush &brush ) 
+{
+    if (m_selected.Ok() && m_selected.GetBitmap() && (*wxTRANSPARENT_BRUSH != brush))
+    {
+        wxBrush copiedBrush( brush );
+        if ( copiedBrush.GetColour() == *wxWHITE )
+        {
+            copiedBrush.SetColour( *wxBLACK );
+        }
+        else
+        {
+            copiedBrush.SetColour( *wxWHITE );
+        }
+        wxWindowDC::SetBackground( copiedBrush );
+    }
+    else
+    {
+        wxWindowDC::SetBackground( brush );
     }
 }
 

 	  	 
