diff --git a/src/unix/fswatcher_inotify.cpp b/src/unix/fswatcher_inotify.cpp
index e4d72b0..01e4421 100644
--- a/src/unix/fswatcher_inotify.cpp
+++ b/src/unix/fswatcher_inotify.cpp
@@ -350,7 +350,7 @@ protected:
                         wxString::Format("%u", inevt.mask);
         return wxString::Format("Event: wd=%d, mask=%s, cookie=%u, len=%u, "
                                 "name=%s", inevt.wd, mask, inevt.cookie,
-                                inevt.len, inevt.name);
+                                inevt.len, wxString(inevt.name, inevt.len));
     }
 
     static wxFileName GetEventPath(const wxFSWatchEntry& watch,
@@ -360,7 +360,7 @@ protected:
         wxFileName path = watch.GetPath();
         if (path.IsDir())
         {
-            path = wxFileName(path.GetPath(), inevt.name);
+            path = wxFileName(path.GetPath(), wxString(inevt.name, inevt.len));
         }
         return path;
     }
