Index: include/wx/html/htmlcell.h
===================================================================
--- include/wx/html/htmlcell.h	(revision 74502)
+++ include/wx/html/htmlcell.h	(working copy)
@@ -210,6 +210,10 @@
     // Returns cursor to be used when mouse is over the cell:
     virtual wxCursor GetMouseCursor(wxHtmlWindowInterface *window) const;
 
+    // returns cursor to be used when mouse is over the cell
+    virtual wxCursor GetMouseCursorRelative(wxHtmlWindowInterface *window,
+                                    const wxPoint& relMousePos) const;
+
 #if WXWIN_COMPATIBILITY_2_6
     // this was replaced by GetMouseCursor, don't use in new code!
     virtual wxCursor GetCursor() const;
Index: src/html/htmlcell.cpp
===================================================================
--- src/html/htmlcell.cpp	(revision 74502)
+++ src/html/htmlcell.cpp	(working copy)
@@ -196,8 +196,17 @@
 }
 #endif // WXWIN_COMPATIBILITY_2_6
 
-wxCursor wxHtmlCell::GetMouseCursor(wxHtmlWindowInterface *window) const
+wxCursor wxHtmlCell::GetMouseCursor(wxHtmlWindowInterface* WXUNUSED(window)) const
 {
+  // NB: GetMouseCursor is deprecated in favor of GetMouseCursorRelative
+  //    Similarly to GetCursor(), we return a wxNullCursor which is checked
+  //    by GetMouseCursorRelative, so that user code overriding GetMouseCursor
+  //    continues to work.
+  return wxNullCursor;
+}
+
+wxCursor wxHtmlCell::GetMouseCursorRelative(wxHtmlWindowInterface *window, const wxPoint& relMousePos) const
+{
 #if WXWIN_COMPATIBILITY_2_6
     // NB: Older versions of wx used GetCursor() virtual method in place of
     //     GetMouseCursor(interface). This code ensures that user code that
@@ -209,7 +218,11 @@
         return cur;
 #endif // WXWIN_COMPATIBILITY_2_6
 
-    if ( GetLink() )
+    wxCursor cur2 = GetMouseCursor(window);
+    if (cur2.IsOk())
+      return cur2;
+
+    if ( GetLink(relMousePos.x, relMousePos.y) )
     {
         return window->GetHTMLCursor(wxHtmlWindowInterface::HTMLCursor_Link);
     }
Index: src/html/htmlwin.cpp
===================================================================
--- src/html/htmlwin.cpp	(revision 74502)
+++ src/html/htmlwin.cpp	(working copy)
@@ -206,7 +206,7 @@
 
         wxCursor cur;
         if (cell)
-            cur = cell->GetMouseCursor(m_interface);
+            cur = cell->GetMouseCursorRelative(m_interface, pos);
         else
             cur = m_interface->GetHTMLCursor(
                         wxHtmlWindowInterface::HTMLCursor_Default);
@@ -229,6 +229,11 @@
     {
         if ( cell )
         {
+            // A single cell can have different cursors for different positions,
+            // so update cursor for this case as well.
+            wxCursor cur = cell->GetMouseCursorRelative(m_interface, pos);
+            m_interface->GetHTMLWindow()->SetCursor(cur);
+
             OnCellMouseHover(cell, pos.x, pos.y);
         }
     }
