Index: grid.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/generic/grid.cpp,v
retrieving revision 1.248
diff -b -u -2 -r1.248 grid.cpp
--- grid.cpp	2003/04/03 21:14:07	1.248
+++ grid.cpp	2003/04/23 16:30:55
@@ -8031,5 +8031,5 @@
 
     int row = m_currentCellCoords.GetRow();
-    if ( row < m_numRows )
+    if ( row + 1 < m_numRows )
     {
         int cw, ch;

 	  	 
