Index: src/os2/app.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/os2/app.cpp,v
retrieving revision 1.80
diff -u -r1.80 app.cpp
--- src/os2/app.cpp	2004/05/23 20:53:10	1.80
+++ src/os2/app.cpp	2004/08/08 10:29:15
@@ -115,34 +115,9 @@
     void* gsock;
 };
 
-// These defines and wrapper functions are used here and in gsockpm.c
+// These defines are used here and in gsockpm.c
 #define wxSockReadMask  0x01
 #define wxSockWriteMask 0x02
-
-#ifdef __EMX__
-extern "C"
-int wxAppAddSocketHandler(int handle, int mask,
-                           void (*callback)(void*), void * gsock)
-{
-    return wxTheApp->AddSocketHandler(handle, mask, callback, gsock);
-}
-extern "C"
-void wxAppRemoveSocketHandler(int handle)
-{
-    wxTheApp->RemoveSocketHandler(handle);
-}
-#else
-//  Linkage mode problems using callbacks with extern C in a .cpp module
-int wxAppAddSocketHandler(int handle, int mask,
-                           void (*callback)(void*), void * gsock)
-{
-    return wxTheApp->AddSocketHandler(handle, mask, callback, gsock);
-}
-void wxAppRemoveSocketHandler(int handle)
-{
-    wxTheApp->RemoveSocketHandler(handle);
-}
-#endif
 
 void wxApp::HandleSockets()
 {
Index: src/os2/gsockpm.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/os2/gsockpm.cpp,v
retrieving revision 1.5
diff -u -r1.5 gsockpm.cpp
--- src/os2/gsockpm.cpp	2004/08/08 05:30:33	1.5
+++ src/os2/gsockpm.cpp	2004/08/08 10:29:21
@@ -13,12 +13,9 @@
 #include <stdlib.h>
 #include "wx/unix/gsockunx.h"
 #include "wx/gsocket.h"
+#include "wx/app.h"
 
-extern int wxAppAddSocketHandler(int handle, int mask,
-				 void (*callback)(void*), void * gsock);
 
-extern void wxAppRemoveSocketHandler(int handle);
-
 #define wxSockReadMask  0x01
 #define wxSockWriteMask 0x02
 
@@ -80,16 +77,16 @@
     }
 
     if (m_id[c] != -1)
-        wxAppRemoveSocketHandler(m_id[c]);
+        wxTheApp->RemoveSocketHandler(m_id[c]);
 
     if (c == 0)
     {
-	m_id[0] = wxAppAddSocketHandler(socket->m_fd, wxSockReadMask,
+	m_id[0] = wxTheApp->AddSocketHandler(socket->m_fd, wxSockReadMask,
 					_GSocket_PM_Input, (void *)socket);
     }
     else
     {
-	m_id[1] = wxAppAddSocketHandler(socket->m_fd, wxSockWriteMask,
+	m_id[1] = wxTheApp->AddSocketHandler(socket->m_fd, wxSockWriteMask,
 					_GSocket_PM_Output, (void *)socket);
     }
 }
@@ -107,7 +104,7 @@
 	default: return;
     }
     if (m_id[c] != -1)
-        wxAppRemoveSocketHandler(m_id[c]);
+        wxTheApp->RemoveSocketHandler(m_id[c]);
 
     m_id[c] = -1;
 }
Index: src/unix/gsocket.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/unix/gsocket.cpp,v
retrieving revision 1.17
diff -u -r1.17 gsocket.cpp
--- src/unix/gsocket.cpp	2004/08/08 05:30:34	1.17
+++ src/unix/gsocket.cpp	2004/08/08 10:29:51
@@ -87,6 +87,13 @@
 int _System soclose(int);
 #  endif
 #endif
+
+#ifdef __EMX__
+#include <sys/socket.h>
+#include <sys/ioctl.h>
+#include <sys/select.h>
+#endif
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <stddef.h>

 	  	 
