--- window.fixed.cpp	2013-11-05 13:43:30.420515134 +0000
+++ window.cpp	2012-10-16 11:02:41.303505144 +0100
@@ -670,7 +670,7 @@
                                wxShowEffect effect,
                                unsigned timeout)
 {
-    if ( !GetParent()->IsShownOnScreen() || effect == wxSHOW_EFFECT_NONE )
+    if ( effect == wxSHOW_EFFECT_NONE )
         return Show(show);
 
     if ( !wxWindowBase::Show(show) )
