diff --git a/samples/minimal/minimal.cpp b/samples/minimal/minimal.cpp
index 64d5356..95e558e 100644
--- a/samples/minimal/minimal.cpp
+++ b/samples/minimal/minimal.cpp
@@ -126,7 +126,14 @@ bool MyApp::OnInit()
 
     // create the main application window
     MyFrame *frame = new MyFrame("Minimal wxWidgets App");
-
+    
+    wxBoxSizer* sizer = new wxBoxSizer(wxVERTICAL);
+    sizer->Add(new wxStaticText(frame, wxID_ANY, "wxALIGN_LEFT", wxDefaultPosition, wxDefaultSize, wxALIGN_LEFT), 0, wxEXPAND);
+    sizer->Add(new wxStaticText(frame, wxID_ANY, "wxALIGN_CENTRE", wxDefaultPosition, wxDefaultSize, wxALIGN_CENTRE), 0, wxEXPAND);
+    sizer->Add(new wxStaticText(frame, wxID_ANY, "wxALIGN_CENTRE_HORIZONTAL", wxDefaultPosition, wxDefaultSize, wxALIGN_CENTRE_HORIZONTAL), 0, wxEXPAND);
+    sizer->Add(new wxStaticText(frame, wxID_ANY, "wxALIGN_RIGHT", wxDefaultPosition, wxDefaultSize, wxALIGN_RIGHT), 0, wxEXPAND);
+    frame->SetSizer(sizer);
+    
     // and show it (the frames, unlike simple controls, are not shown when
     // created initially)
     frame->Show(true);
