diff -rbu2 ./cvs/wx2/wxwindows/include/wx/msw/window.h ./wx2/include/wx/msw/window.h
--- ./cvs/wx2/wxwindows/include/wx/msw/window.h	Wed Aug  1 19:47:26 2001
+++ ./wx2/include/wx/msw/window.h	Sun Aug  5 16:56:42 2001
@@ -444,5 +444,5 @@
 
     DECLARE_DYNAMIC_CLASS(wxWindowMSW);
-    DECLARE_NO_COPY_CLASS(wxWindowMSW);
+    DECLARE_NO_COPY_CLASS(wxWindowMSW)
     DECLARE_EVENT_TABLE()
 };
diff -rbu2 ./cvs/wx2/wxwindows/src/makeb32.env ./wx2/src/makeb32.env
--- ./cvs/wx2/wxwindows/src/makeb32.env	Wed Aug  1 19:52:46 2001
+++ ./wx2/src/makeb32.env	Sun Aug  5 16:57:52 2001
@@ -7,5 +7,5 @@
 WXLIBDIR = $(WXDIR)\lib
 WXLIBDIR2 = $(WXDIR)\contrib\lib
-WXINC = $(WXDIR)\include;$(WXDIR)\contrib\include;$(WXDIR)\src\iodbc;
+WXINC = $(WXDIR)\include;$(WXDIR)\contrib\include;$(WXDIR)\src\iodbc;$(WXDIR)\src\regex;
 WIN95FLAG = -D__WIN95__ -D__WINDOWS__
 RFLAGS = -i$(WXDIR)\include;$(WXDIR)\src
diff -rbu2P ./cvs/wx2/wxwindows/src/regex/makefile.g95 ./wx2/src/regex/makefile.g95
--- ./cvs/wx2/wxwindows/src/regex/makefile.g95	Thu Jan  1 01:00:00 1970
+++ ./wx2/src/regex/makefile.g95	Sun Aug  5 17:03:52 2001
@@ -0,0 +1,12 @@
+
+LIBTARGET=$(WXWIN)/lib/libregex.a
+
+OBJECTS= \
+		regcomp.o \
+		regexec.o \
+		regerror.o \
+		regfree.o
+
+# Pull in standard variable definitions
+
+include $(WXWIN)/src/makelib.g95

 	  	 
