--- src/common/valtext.cpp	2005-09-23 13:53:11.000000000 +0100
+++ src/common/valtext.cpp	2006-01-25 19:46:30.000000000 +0000
@@ -184,11 +184,11 @@
 // Called to transfer data to the window
 bool wxTextValidator::TransferToWindow(void)
 {
-    if( !CheckValidator() )
-        return false;
-
     if ( m_stringValue )
     {
+        if( !CheckValidator() )
+            return false;
+
         wxTextCtrl *control = (wxTextCtrl *) m_validatorWindow;
         control->SetValue(* m_stringValue);
     }
@@ -199,11 +199,11 @@
 // Called to transfer data to the window
 bool wxTextValidator::TransferFromWindow(void)
 {
-    if( !CheckValidator() )
-        return false;
-
     if ( m_stringValue )
     {
+        if( !CheckValidator() )
+            return false;
+
         wxTextCtrl *control = (wxTextCtrl *) m_validatorWindow;
         *m_stringValue = control->GetValue();
     }

 	  	 
