From: Anders Larsen <al@alarsen.net>
Date: Tue, 26 Jan 2010 15:15:28 +0100
Subject: [PATCH] Fix the console event-loop on MSW to emit wxIdleEvent

Signed-off-by: Anders Larsen <al@alarsen.net>
---
 src/msw/evtloop.cpp |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/src/msw/evtloop.cpp b/src/msw/evtloop.cpp
index 6d8ecdd..362e04f 100644
--- a/src/msw/evtloop.cpp
+++ b/src/msw/evtloop.cpp
@@ -565,7 +565,10 @@ void wxConsoleEventLoop::WakeUp()
 
 void wxConsoleEventLoop::ProcessMessage(WXMSG *msg)
 {
-    ::DispatchMessage(msg);
+    if (msg->message != WM_NULL)
+        ::DispatchMessage(msg);
+    else
+        ProcessIdle();
 }
 
 bool wxConsoleEventLoop::Dispatch()
-- 
1.5.4.3

