Index: src/mac/carbon/statbox.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/mac/carbon/statbox.cpp,v
retrieving revision 1.26
diff -b -u -2 -r1.26 statbox.cpp
--- src/mac/carbon/statbox.cpp	2006/01/09 00:46:43	1.26
+++ src/mac/carbon/statbox.cpp	2006/04/06 01:50:19
@@ -28,7 +28,5 @@
     const wxString& name )
 {
-    m_macIsUserPane = false;
-
-    if ( !wxControl::Create( parent, id, pos, size, style, wxDefaultValidator, name ) )
+    if ( !CreateBase( parent, id, pos, size, style, wxDefaultValidator, name ) )
         return false;
 

 	  	 
