Index: include/wx/defs.h
===================================================================
--- include/wx/defs.h	(revision 64217)
+++ include/wx/defs.h	(working copy)
@@ -827,11 +827,11 @@
         typedef unsigned int wxUint32;
 
         /* Assume that if SIZEOF_INT is defined that all the other ones except
-           SIZEOF_SIZE_T, are too.  See next #if below.  */
+           SIZEOF_SIZE_T and SIZEOF_WCHAR_T, are too.  See next #if below.
+        */
         #ifndef SIZEOF_INT
             #define SIZEOF_INT 4
             #define SIZEOF_LONG 4
-            #define SIZEOF_WCHAR_T 2
 
             /*
                under Win64 sizeof(size_t) == 8 and so it is neither unsigned
@@ -857,8 +857,9 @@
 
         /*
           If Python.h was included first, it defines all of the SIZEOF's above
-          except for SIZEOF_SIZE_T, so we need to do it here to avoid
-          triggering the #error in the ssize_t typedefs below...
+          except for SIZEOF_SIZE_T and SIZEOF_WCHAR_T, so we need to do them
+          here to avoid triggering the #error in the ssize_t
+          typedefs below...
         */
         #ifndef SIZEOF_SIZE_T
             #ifdef __WIN64__
@@ -867,6 +868,10 @@
                 #define SIZEOF_SIZE_T 4
             #endif
         #endif
+
+        #ifndef SIZEOF_WCHAR_T
+            #define SIZEOF_WCHAR_T 2
+        #endif
     #else
         #error "Unsupported Windows version"
     #endif
