Index: menu.cpp
===================================================================
--- menu.cpp	(revision 65567)
+++ menu.cpp	(working copy)
@@ -129,6 +129,8 @@
     void OnUpdateSubMenuCheck(wxUpdateUIEvent& event);
     void OnUpdateSubMenuRadio(wxUpdateUIEvent& event);
 
+    void OnMouseRightDown(wxMouseEvent& ME);
+    void OnMouseRightUp(wxMouseEvent& ME);
 #if USE_CONTEXT_MENU
     void OnContextMenu(wxContextMenuEvent& event);
 #else
@@ -320,6 +322,8 @@
 
     EVT_UPDATE_UI(Menu_Menu_Check, MyFrame::OnUpdateCheckMenuItemUI)
 
+    EVT_RIGHT_DOWN(MyFrame::OnMouseRightDown)
+    EVT_RIGHT_UP  (MyFrame::OnMouseRightUp)
 #if USE_CONTEXT_MENU
     EVT_CONTEXT_MENU(MyFrame::OnContextMenu)
 #else
@@ -1127,6 +1131,22 @@
         event.Check(false);
 }
 
+void MyFrame::OnMouseRightDown(wxMouseEvent& ME)
+{
+    const bool Shift=wxGetKeyState(WXK_SHIFT);
+
+    if (Shift) ME.Skip();
+    wxLogMessage("RMB down (%s)", Shift ? "skipped" : "consumed");
+}
+
+void MyFrame::OnMouseRightUp(wxMouseEvent& ME)
+{
+    const bool Shift=wxGetKeyState(WXK_SHIFT);
+
+    if (Shift) ME.Skip();
+    wxLogMessage("RMB up (%s)", Shift ? "skipped" : "consumed");
+}
+
 #if USE_CONTEXT_MENU
 void MyFrame::OnContextMenu(wxContextMenuEvent& event)
 {
