Index: src/stc/PlatWX.cpp
===================================================================
diff --git a/trunk/src/stc/PlatWX.cpp b/trunk/src/stc/PlatWX.cpp
--- a/trunk/src/stc/PlatWX.cpp	(revision 126947)
+++ b/trunk/src/stc/PlatWX.cpp	(working copy)
@@ -95,7 +95,7 @@
     wxFontEncodingArray ea = wxEncodingConverter::GetPlatformEquivalents(encoding);
     if (ea.GetCount())
         encoding = ea[0];
-
+    /*
     wxFontWeight weight;
     if (fp.weight <= 300)
         weight = wxFONTWEIGHT_LIGHT;
@@ -111,14 +111,30 @@
                               false,
                               stc2wx(fp.faceName),
                               encoding);
-    fid = font;
+    fid = font;
+    */
+    // make things to be simplified
+    wxFontWeight weight = (fp.weight <= 300) ? wxFONTWEIGHT_LIGHT :
+                            ((fp.weight >= 700) ? wxFONTWEIGHT_BOLD : wxFONTWEIGHT_NORMAL);
+    //  search for a previously created font of the desired type and create it if not already found
+    //  When an application finishes, all fonts will be deleted and their resources freed
+    fid = wxTheFontList->FindOrCreateFont(fp.size,
+                              wxFONTFAMILY_DEFAULT,
+                              style,
+                              weight,
+                              false,
+                              face,
+                              encoding);
 }
 
 
-void Font::Release() {
+void Font::Release() {
+    /*
     if (fid)
         delete (wxFont*)fid;
-    fid = 0;
+    fid = 0;
+    */
+    fid = NULL;
 }
 
 //----------------------------------------------------------------------
