Index: src/stc/PlatWX.cpp
===================================================================
--- src/stc/PlatWX.cpp	(revision 126947)
+++ src/stc/PlatWX.cpp	(working copy)
@@ -95,30 +95,23 @@
     wxFontEncodingArray ea = wxEncodingConverter::GetPlatformEquivalents(encoding);
     if (ea.GetCount())
         encoding = ea[0];
-
-    wxFontWeight weight;
-    if (fp.weight <= 300)
-        weight = wxFONTWEIGHT_LIGHT;
-    else if (fp.weight >= 700)
-        weight = wxFONTWEIGHT_BOLD;
-    else
-        weight = wxFONTWEIGHT_NORMAL;
-
-    wxFont* font = new wxFont(fp.size,
+    // make things to be simplified
+    wxFontWeight weight = (fp.weight <= 300) ? wxFONTWEIGHT_LIGHT :
+                            ((fp.weight >= 700) ? wxFONTWEIGHT_BOLD : wxFONTWEIGHT_NORMAL);
+    //  search for a previously created font of the desired type and create it if not already found
+    //  When an application finishes, all fonts will be deleted and their resources freed
+    fid = wxTheFontList->FindOrCreateFont(fp.size,
                               wxFONTFAMILY_DEFAULT,
-                              fp.italic ? wxFONTSTYLE_ITALIC :  wxFONTSTYLE_NORMAL,
+                              style,
                               weight,
                               false,
                               stc2wx(fp.faceName),
                               encoding);
-    fid = font;
 }
 
 
 void Font::Release() {
-    if (fid)
-        delete (wxFont*)fid;
-    fid = 0;
+    fid = NULL;
 }
 
 //----------------------------------------------------------------------
