Index: include/wx/gtk/listbox.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/gtk/listbox.h,v
retrieving revision 1.60
diff -b -u -2 -r1.60 listbox.h
--- include/wx/gtk/listbox.h	2006/03/23 22:04:27	1.60
+++ include/wx/gtk/listbox.h	2006/06/15 21:54:07
@@ -101,4 +101,5 @@
     void GtkInsertItems(const wxArrayString& items,
                         void** clientData, unsigned int pos);
+    void GtkDeselectAll(const bool blockEvent);
     void GtkSetSelection(int n, const bool select, const bool blockEvent);
 
Index: samples/listbox/lboxtest.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/samples/listbox/lboxtest.cpp,v
retrieving revision 1.14
diff -b -u -2 -r1.14 lboxtest.cpp
--- samples/listbox/lboxtest.cpp	2006/03/10 21:37:20	1.14
+++ samples/listbox/lboxtest.cpp	2006/06/15 21:54:07
@@ -78,4 +78,5 @@
     LboxTest_DeleteText,
     LboxTest_DeleteSel,
+    LboxTest_DeselectAll,
     LboxTest_Listbox,
     LboxTest_Quit
@@ -114,5 +115,7 @@
     void OnButtonDelete(wxCommandEvent& event);
     void OnButtonDeleteSel(wxCommandEvent& event);
+    void OnButtonDeselectAll(wxCommandEvent& event);
     void OnButtonClear(wxCommandEvent& event);
+
 #if wxUSE_LOG
     void OnButtonClearLog(wxCommandEvent& event);
@@ -134,4 +137,5 @@
     void OnUpdateUIDeleteButton(wxUpdateUIEvent& event);
     void OnUpdateUIDeleteSelButton(wxUpdateUIEvent& event);
+    void OnUpdateUIDeselectAllButton(wxUpdateUIEvent& event);
 
     // reset the listbox parameters
@@ -278,4 +282,5 @@
     EVT_BUTTON(LboxTest_Delete, LboxTestFrame::OnButtonDelete)
     EVT_BUTTON(LboxTest_DeleteSel, LboxTestFrame::OnButtonDeleteSel)
+    EVT_BUTTON(LboxTest_DeselectAll, LboxTestFrame::OnButtonDeselectAll)
     EVT_BUTTON(LboxTest_Clear, LboxTestFrame::OnButtonClear)
 #if wxUSE_LOG
@@ -300,4 +305,5 @@
     EVT_UPDATE_UI(LboxTest_ChangeText, LboxTestFrame::OnUpdateUIDeleteSelButton)
     EVT_UPDATE_UI(LboxTest_DeleteSel, LboxTestFrame::OnUpdateUIDeleteSelButton)
+    EVT_UPDATE_UI(LboxTest_DeselectAll, LboxTestFrame::OnUpdateUIDeselectAllButton)
 
     EVT_LISTBOX(LboxTest_Listbox, LboxTestFrame::OnListbox)
@@ -433,4 +439,7 @@
     sizerMiddle->Add(btn, 0, wxALL | wxGROW, 5);
 
+    btn = new wxButton(m_panel, LboxTest_DeselectAll, _T("Deselect All"));
+    sizerMiddle->Add(btn, 0, wxALL | wxGROW, 5);
+
     btn = new wxButton(m_panel, LboxTest_Clear, _T("&Clear"));
     sizerMiddle->Add(btn, 0, wxALL | wxGROW, 5);
@@ -620,4 +629,9 @@
 }
 
+void LboxTestFrame::OnButtonDeselectAll(wxCommandEvent& WXUNUSED(event))
+{
+    m_lbox->SetSelection(-1); 
+}
+
 void LboxTestFrame::OnButtonClear(wxCommandEvent& WXUNUSED(event))
 {
@@ -677,4 +691,10 @@
 
 void LboxTestFrame::OnUpdateUIDeleteSelButton(wxUpdateUIEvent& event)
+{
+    wxArrayInt selections;
+    event.Enable(m_lbox->GetSelections(selections) != 0);
+}
+
+void LboxTestFrame::OnUpdateUIDeselectAllButton(wxUpdateUIEvent& event)
 {
     wxArrayInt selections;
Index: src/gtk/listbox.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/gtk/listbox.cpp,v
retrieving revision 1.186
diff -b -u -2 -r1.186 listbox.cpp
--- src/gtk/listbox.cpp	2006/06/13 11:30:24	1.186
+++ src/gtk/listbox.cpp	2006/06/15 21:54:08
@@ -945,5 +945,25 @@
 void wxListBox::DoSetSelection( int n, bool select )
 {
+    wxCHECK_RET( n == wxNOT_FOUND || IsValid(n),
+        wxT("invalid index in wxListBox::SetSelection") );
+
+    //passing -1 here deselects all
+    if(n == wxNOT_FOUND) 
+        return GtkDeselectAll(true); //docs say no events here
+    else
     return GtkSetSelection(n, select, true); //docs say no events here
+}
+
+void wxListBox::GtkDeselectAll(const bool blockEvent)
+{
+    wxCHECK_RET( m_treeview != NULL, wxT("invalid listbox") );
+
+    GtkTreeSelection* selection = gtk_tree_view_get_selection(m_treeview);
+
+    m_blockEvent = blockEvent;
+
+    gtk_tree_selection_unselect_all(selection);
+
+    m_blockEvent = false;
 }
 

 	  	 
