--- src/gtk/button.cpp
+++ src/gtk/button.cpp
@@ -80,7 +80,6 @@ gtk_button_style_set_callback( GtkWidget
             right_border += default_border->right;
             top_border += default_border->top;
             bottom_border += default_border->bottom;
-            g_free( default_border );
         }
         win->MoveWindow(
             win->m_x - top_border,
--- src/gtk/window.cpp
+++ src/gtk/window.cpp
@@ -2706,7 +2706,6 @@ void wxWindowGTK::DoSetSize( int x, int 
                 right_border += default_border->right;
                 top_border += default_border->top;
                 bottom_border += default_border->bottom;
-                g_free( default_border );
             }
         }
 

 	  	 
