--- wxWidgets.orig/src/generic/odcombo.cpp	Wed Apr 11 17:09:30 2007
+++ wxWidgets/src/generic/odcombo.cpp	Tue May 15 18:04:22 2007
@@ -117,5 +117,5 @@
         if ( m_value >= 0 )
         {
-            OnDrawItem(dc,rect,m_value,wxODCB_PAINTING_CONTROL);
+            OnDrawItem(dc,rect,m_value,flags);
             return;
         }
@@ -130,11 +130,18 @@
     dc.SetFont(m_useFont);
 
+    int flags = 0;
+
     // Set correct text colour for selected items
     if ( wxVListBox::GetSelection() == (int) n )
+    {
         dc.SetTextForeground( wxSystemSettings::GetColour(wxSYS_COLOUR_HIGHLIGHTTEXT) );
+        flags |= wxODCB_PAINTING_SELECTED;
+    }
     else
+    {
         dc.SetTextForeground( wxSystemSettings::GetColour(wxSYS_COLOUR_WINDOWTEXT) );
+    }
 
-    OnDrawItem(dc,rect,(int)n,0);
+    OnDrawItem(dc,rect,(int)n,flags);
 }
 

 	  	 
