diff -ru2 -N -xCVS -x.#* -x*.orig -x*.vcproj -x*.cmake -x*.sln -x*.suo -x*.rej -x*~ -x*.o -x*.obj -x*.a -x*.lib -x*setup.h -x*y*.c wxWindows.orig/include/wx/list.h wxWindows/include/wx/list.h
--- wxWindows.orig/include/wx/list.h	Fri Dec  3 11:42:44 2004
+++ wxWindows/include/wx/list.h	Fri Dec  3 11:45:31 2004
@@ -403,4 +403,5 @@
     public:                                                                 \
         typedef nodetype Node;                                              \
+        typedef Node* compatibility_iterator;                               \
                                                                             \
         name(wxKeyType keyType = wxKEY_NONE) : wxListBase(keyType)          \

 	  	 
