Index: src/common/zipstrm.cpp
===================================================================
--- src/common/zipstrm.cpp	(revision 76519)
+++ src/common/zipstrm.cpp	(working copy)
@@ -809,6 +809,14 @@
     return fn.GetFullPath(format);
 }
 
+namespace
+{
+    inline bool IsDOSPathSep(wxUniChar ch)
+    {
+        return ch == wxFILE_SEP_PATH_DOS || ch == wxFILE_SEP_PATH_UNIX;
+    }
+} // anonymous namespace
+
 // Static - Internally tars and zips use forward slashes for the path
 // separator, absolute paths aren't allowed, and directory names have a
 // trailing slash.  This function converts a path into this internal format,
@@ -825,15 +833,15 @@
     else
         internal = name;
 
-    bool isDir = !internal.empty() && internal.Last() == '/';
+    bool isDir = !internal.empty() && IsDOSPathSep(internal.Last());
     if (pIsDir)
         *pIsDir = isDir;
     if (isDir)
         internal.erase(internal.length() - 1);
 
-    while (!internal.empty() && *internal.begin() == '/')
+    while (!internal.empty() && IsDOSPathSep(*internal.begin()))
         internal.erase(0, 1);
-    while (!internal.empty() && internal.compare(0, 2, wxT("./")) == 0)
+    while (!internal.empty() && ( (internal.compare(0, 2, wxT("./")) == 0) || (internal.compare(0, 2, wxT(".\\")) == 0) ) )
         internal.erase(0, 2);
     if (internal == wxT(".") || internal == wxT(".."))
         internal = wxEmptyString;
