diff --git a/samples/render/render.cpp b/samples/render/render.cpp
index 132d172..3e60373 100644
--- a/samples/render/render.cpp
+++ b/samples/render/render.cpp
@@ -274,8 +274,6 @@ private:
         dc.DrawText("DrawGauge()", x1, y);
         wxRendererNative::GetDefault().DrawGauge(this, dc,
             wxRect(x2, y, widthGauge, heightGauge), 25, 100, m_flags);
-
-        y += lineHeight + heightGauge;
     }
 
     int m_flags;
