Index: wxWidgets-trunk/include/wx/osx/filedlg.h
===================================================================
--- wxWidgets-trunk/include/wx/osx/filedlg.h	(revision 66957)
+++ wxWidgets-trunk/include/wx/osx/filedlg.h	(working copy)
@@ -12,6 +12,8 @@
 #ifndef _WX_FILEDLG_H_
 #define _WX_FILEDLG_H_
 
+class wxChoice;
+
 //-------------------------------------------------------------------------
 // wxFileDialog
 //-------------------------------------------------------------------------
