? mypatch.patch
Index: utilsx11.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/unix/utilsx11.cpp,v
retrieving revision 1.38
diff -u -r1.38 utilsx11.cpp
--- utilsx11.cpp	2006/10/19 14:39:49	1.38
+++ utilsx11.cpp	2007/03/18 16:11:48
@@ -811,13 +811,15 @@
     Window       wDummy1, wDummy2;
     int          iDummy3, iDummy4, iDummy5, iDummy6;
     unsigned int iMask;
-    XModifierKeymap* map = XGetModifierMapping(pDisplay);
+
     KeyCode keyCode = XKeysymToKeycode(pDisplay,iKey);
     if (keyCode == NoSymbol)
         return false;
 
     if ( IsModifierKey(iKey) )  // If iKey is a modifier key, use a different method
     {
+        XModifierKeymap* map = XGetModifierMapping(pDisplay);
+
         for (int i = 0; i < 8; ++i)
         {
             if ( map->modifiermap[map->max_keypermod * i] == keyCode)

 	  	 
