Index: notebook.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/notebook.cpp,v
retrieving revision 1.102
diff -b -u -2 -r1.102 notebook.cpp
--- notebook.cpp	2003/09/26 21:36:27	1.102
+++ notebook.cpp	2003/09/27 11:52:09
@@ -728,5 +728,18 @@
         wxNotebookPage *pPage = m_pages[sel];
         pPage->Show(true);
-      }
+            pPage->SetFocus();
+            // If the newly focused window is not a child
+            // of the new page, SetFocus was not
+            // successful and the notebook itself
+            // should be focused
+            wxWindow *currentFocus=FindFocus();
+            wxWindow *startFocus=currentFocus;
+            while (currentFocus && currentFocus!=pPage && currentFocus!=this)
+                currentFocus=currentFocus->GetParent();
+            if (startFocus==pPage || currentFocus!=pPage)
+                SetFocus();
+
+        } else
+            SetFocus();
 
       m_nSelection = sel;
@@ -741,10 +754,13 @@
     const MSG * const msg = (MSG *)wxmsg;
 
-    // we want to process (simple, i.e. without Shift or Ctrl) TAB to pass it
-    // to our page for keyboard navigation
+    // intercept TAB, CTRL+TAB and CTRL+SHIFT+TAB
+    // for processing by wxNotebook. TAB will be passed
+    // to the currently selected page, CTRL+TAB and CTRL+SHIFT+TAB
+    // will be processed by the notebook itself.
+    // do not intercept SHIFT+TAB. This goes to the parent of the
+    // notebook which will process it.
     if ( msg->message == WM_KEYDOWN && msg->wParam == VK_TAB &&
             msg->hwnd == m_hwnd &&
-                !wxIsCtrlDown() && !wxIsShiftDown() &&
-                    m_nSelection != -1 )
+                (wxIsCtrlDown() || !wxIsShiftDown()))
     {
         return MSWProcessMessage(wxmsg);

 	  	 
