Index: include/wx/confbase.h
===================================================================
--- include/wx/confbase.h	(revision 50462)
+++ include/wx/confbase.h	(working copy)
@@ -2,11 +2,11 @@
 // Name:        confbase.h
 // Purpose:     declaration of the base class of all config implementations
 //              (see also: fileconf.h and msw/regconf.h and iniconf.h)
-// Author:      Karsten Ballüder & Vadim Zeitlin
+// Author:      Karsten Ballueder & Vadim Zeitlin
 // Modified by:
 // Created:     07.04.98 (adapted from appconf.h)
 // RCS-ID:      $Id$
-// Copyright:   (c) 1997 Karsten Ballüder   Ballueder@usa.net
+// Copyright:   (c) 1997 Karsten Ballueder   Ballueder@usa.net
 //                       Vadim Zeitlin      <zeitlin@dptmaths.ens-cachan.fr>
 // Licence:     wxWindows licence
 ///////////////////////////////////////////////////////////////////////////////
Index: include/wx/fileconf.h
===================================================================
--- include/wx/fileconf.h	(revision 50462)
+++ include/wx/fileconf.h	(working copy)
@@ -5,7 +5,7 @@
 // Modified by:
 // Created:     07.04.98 (adapted from appconf.cpp)
 // RCS-ID:      $Id$
-// Copyright:   (c) 1997 Karsten Ballüder   &  Vadim Zeitlin
+// Copyright:   (c) 1997 Karsten Ballueder   &  Vadim Zeitlin
 //                       Ballueder@usa.net     <zeitlin@dptmaths.ens-cachan.fr>
 // Licence:     wxWindows licence
 ///////////////////////////////////////////////////////////////////////////////
Index: include/wx/generic/helpext.h
===================================================================
--- include/wx/generic/helpext.h	(revision 50462)
+++ include/wx/generic/helpext.h	(working copy)
@@ -1,7 +1,7 @@
 /*-*- c++ -*-********************************************************
  * helpext.h - an external help controller for wxWidgets            *
  *                                                                  *
- * (C) 1998 by Karsten Ballüder (Ballueder@usa.net)                 *
+ * (C) 1998 by Karsten Ballueder (Ballueder@usa.net)                *
  * License: wxWindows licence                                       *
  *                                                                  *
  * $Id$
Index: include/wx/generic/progdlgg.h
===================================================================
--- include/wx/generic/progdlgg.h	(revision 50462)
+++ include/wx/generic/progdlgg.h	(working copy)
@@ -1,11 +1,11 @@
 ////////////////////////////////////////////////////
 // Name:        progdlgg.h
 // Purpose:     wxProgressDialog class
-// Author:      Karsten Ballüder
+// Author:      Karsten Ballueder
 // Modified by:
 // Created:     09.05.1999
 // RCS-ID:      $Id$
-// Copyright:   (c) Karsten Ballüder
+// Copyright:   (c) Karsten Ballueder
 // Licence:     wxWindows licence
 ////////////////////////////////////////////////////
 
Index: include/wx/regex.h
===================================================================
--- include/wx/regex.h	(revision 50462)
+++ include/wx/regex.h	(working copy)
@@ -1,11 +1,11 @@
 ///////////////////////////////////////////////////////////////////////////////
 // Name:        wx/regex.h
 // Purpose:     regular expression matching
-// Author:      Karsten Ballüder
+// Author:      Karsten Ballueder
 // Modified by: VZ at 13.07.01 (integrated to wxWin)
 // Created:     05.02.2000
 // RCS-ID:      $Id$
-// Copyright:   (c) 2000 Karsten Ballüder <ballueder@gmx.net>
+// Copyright:   (c) 2000 Karsten Ballueder <ballueder@gmx.net>
 // Licence:     wxWindows licence
 ///////////////////////////////////////////////////////////////////////////////
 
Index: src/common/config.cpp
===================================================================
--- src/common/config.cpp	(revision 50462)
+++ src/common/config.cpp	(working copy)
@@ -5,7 +5,7 @@
 // Modified by:
 // Created:     07.04.98
 // RCS-ID:      $Id$
-// Copyright:   (c) 1997 Karsten Ballüder   Ballueder@usa.net
+// Copyright:   (c) 1997 Karsten Ballueder   Ballueder@usa.net
 //                       Vadim Zeitlin      <zeitlin@dptmaths.ens-cachan.fr>
 // Licence:     wxWindows licence
 ///////////////////////////////////////////////////////////////////////////////
Index: src/common/fileconf.cpp
===================================================================
--- src/common/fileconf.cpp	(revision 50462)
+++ src/common/fileconf.cpp	(working copy)
@@ -5,7 +5,7 @@
 // Modified by:
 // Created:     07.04.98 (adapted from appconf.cpp)
 // RCS-ID:      $Id$
-// Copyright:   (c) 1997 Karsten Ballüder   &  Vadim Zeitlin
+// Copyright:   (c) 1997 Karsten Ballueder   &  Vadim Zeitlin
 //                       Ballueder@usa.net     <zeitlin@dptmaths.ens-cachan.fr>
 // Licence:     wxWindows licence
 ///////////////////////////////////////////////////////////////////////////////
Index: src/common/regex.cpp
===================================================================
--- src/common/regex.cpp	(revision 50462)
+++ src/common/regex.cpp	(working copy)
@@ -1,11 +1,11 @@
 ///////////////////////////////////////////////////////////////////////////////
 // Name:        src/common/regex.cpp
 // Purpose:     regular expression matching
-// Author:      Karsten Ballüder and Vadim Zeitlin
+// Author:      Karsten Ballueder and Vadim Zeitlin
 // Modified by:
 // Created:     13.07.01
 // RCS-ID:      $Id$
-// Copyright:   (c) 2000 Karsten Ballüder <ballueder@gmx.net>
+// Copyright:   (c) 2000 Karsten Ballueder <ballueder@gmx.net>
 //                  2001 Vadim Zeitlin <vadim@wxwindows.org>
 // Licence:     wxWindows licence
 ///////////////////////////////////////////////////////////////////////////////
Index: src/common/string.cpp
===================================================================
--- src/common/string.cpp	(revision 50462)
+++ src/common/string.cpp	(working copy)
@@ -1589,9 +1589,9 @@
 // ---------------------------------------------------------------------------
 
 // some compilers (VC++ 6.0 not to name them) return true for a call to
-// isspace('ê') in the C locale which seems to be broken to me, but we have to
-// live with this by checking that the character is a 7 bit one - even if this
-// may fail to detect some spaces (I don't know if Unicode doesn't have
+// isspace('\xEA') in the C locale which seems to be broken to me, but we have
+// to live with this by checking that the character is a 7 bit one - even if 
+// this may fail to detect some spaces (I don't know if Unicode doesn't have
 // space-like symbols somewhere except in the first 128 chars), it is arguably
 // still better than trimming away accented letters
 inline int wxSafeIsspace(wxChar ch) { return (ch < 127) && wxIsspace(ch); }
Index: src/generic/dcpsg.cpp
===================================================================
--- src/generic/dcpsg.cpp	(revision 50462)
+++ src/generic/dcpsg.cpp	(working copy)
@@ -2123,13 +2123,13 @@
            /  these values from AFM files, too. Maybe later ... */
 
         // NB: casts to int are needed to suppress gcc 3.3 warnings
-        lastWidths[196] = lastWidths[(int)'A'];  // Ä
-        lastWidths[228] = lastWidths[(int)'a'];  // ä
-        lastWidths[214] = lastWidths[(int)'O'];  // Ö
-        lastWidths[246] = lastWidths[(int)'o'];  // ö
-        lastWidths[220] = lastWidths[(int)'U'];  // Ü
-        lastWidths[252] = lastWidths[(int)'u'];  // ü
-        lastWidths[223] = lastWidths[(int)251];  // ß
+        lastWidths[196] = lastWidths[(int)'A'];  // U+00C4 A Umlaute
+        lastWidths[228] = lastWidths[(int)'a'];  // U+00E4 a Umlaute
+        lastWidths[214] = lastWidths[(int)'O'];  // U+00D6 O Umlaute
+        lastWidths[246] = lastWidths[(int)'o'];  // U+00F6 o Umlaute
+        lastWidths[220] = lastWidths[(int)'U'];  // U+00DC U Umlaute
+        lastWidths[252] = lastWidths[(int)'u'];  // U+00FC u Umlaute
+        lastWidths[223] = lastWidths[(int)251];  // U+00DF eszett (scharfes s)
 
         /* JC: calculate UnderlineThickness/UnderlinePosition */
 
Index: src/generic/progdlgg.cpp
===================================================================
--- src/generic/progdlgg.cpp	(revision 50462)
+++ src/generic/progdlgg.cpp	(working copy)
@@ -1,11 +1,11 @@
 /////////////////////////////////////////////////////////////////////////////
 // Name:        src/generic/progdlgg.cpp
 // Purpose:     wxProgressDialog class
-// Author:      Karsten Ballüder
+// Author:      Karsten Ballueder
 // Modified by:
 // Created:     09.05.1999
 // RCS-ID:      $Id$
-// Copyright:   (c) Karsten Ballüder
+// Copyright:   (c) Karsten Ballueder
 // Licence:     wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
Index: src/msw/listctrl.cpp
===================================================================
--- src/msw/listctrl.cpp	(revision 50462)
+++ src/msw/listctrl.cpp	(working copy)
@@ -279,7 +279,7 @@
 wxCONSTRUCTOR_5( wxListCtrl , wxWindow* , Parent , wxWindowID , Id , wxPoint , Position , wxSize , Size , long , WindowStyle )
 
 /*
- TODO : Expose more information of a list's layout etc. via appropriate objects (à la NotebookPageInfo)
+ TODO : Expose more information of a list's layout etc. via appropriate objects (a la NotebookPageInfo)
 */
 #else
 IMPLEMENT_DYNAMIC_CLASS(wxListCtrl, wxControl)

 	  	 
