? filename.cpp.patch
Index: filename.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/common/filename.cpp,v
retrieving revision 1.182
diff -u -r1.182 filename.cpp
--- filename.cpp	2006/12/17 14:52:10	1.182
+++ filename.cpp	2007/03/14 18:05:31
@@ -2328,11 +2328,11 @@
 
     DWORD lpFileSizeHigh;
     DWORD ret = GetFileSize(f, &lpFileSizeHigh);
-    if (ret == INVALID_FILE_SIZE)
+    if (ret == INVALID_FILE_SIZE && ::GetLastError() != NO_ERROR)
         return wxInvalidSize;
 
     // compose the low-order and high-order byte sizes
-    return wxULongLong(ret | (lpFileSizeHigh << sizeof(WORD)*2));
+    return wxULongLong(static_cast<unsigned long>(lpFileSizeHigh), static_cast<unsigned long>(ret));
 
 #else           // ! __WIN32__
 

 	  	 
