 samples/mdi/mdi.cpp | 36 ++++++++++++++----------------------
 1 file changed, 14 insertions(+), 22 deletions(-)

diff --git a/samples/mdi/mdi.cpp b/samples/mdi/mdi.cpp
index 657ac0b..1d39317 100644
--- a/samples/mdi/mdi.cpp
+++ b/samples/mdi/mdi.cpp
@@ -30,6 +30,7 @@
 #endif
 
 #include "wx/toolbar.h"
+#include "wx/artprov.h"
 
 #ifndef wxHAS_IMAGES_IN_RESOURCES
     #include "../sample.xpm"
@@ -329,28 +330,19 @@ void MyFrame::OnSize(wxSizeEvent& event)
 #if wxUSE_TOOLBAR
 void MyFrame::InitToolBar(wxToolBar* toolBar)
 {
-    wxBitmap bitmaps[8];
-
-    bitmaps[0] = wxBitmap( new_xpm );
-    bitmaps[1] = wxBitmap( open_xpm );
-    bitmaps[2] = wxBitmap( save_xpm );
-    bitmaps[3] = wxBitmap( copy_xpm );
-    bitmaps[4] = wxBitmap( cut_xpm );
-    bitmaps[5] = wxBitmap( paste_xpm );
-    bitmaps[6] = wxBitmap( print_xpm );
-    bitmaps[7] = wxBitmap( help_xpm );
-
-    toolBar->AddTool(wxID_NEW, "New", bitmaps[0], "New file");
-    toolBar->AddTool(1, "Open", bitmaps[1], "Open file");
-    toolBar->AddTool(2, "Save", bitmaps[2], "Save file");
-    toolBar->AddSeparator();
-    toolBar->AddTool(3, "Copy", bitmaps[3], "Copy");
-    toolBar->AddTool(4, "Cut", bitmaps[4], "Cut");
-    toolBar->AddTool(5, "Paste", bitmaps[5], "Paste");
-    toolBar->AddSeparator();
-    toolBar->AddTool(6, "Print", bitmaps[6], "Print");
-    toolBar->AddSeparator();
-    toolBar->AddTool(wxID_ABOUT, "About", bitmaps[7], "Help");
+    // alpha bitmap breaks transparency of subsequent mask bitmaps
+
+    wxBitmap bmp;
+
+    // with alpha (from tango)
+    bmp = wxArtProvider::GetBitmap(wxART_NEW , wxART_TOOLBAR);
+    wxASSERT(bmp.HasAlpha());
+    toolBar->AddTool(wxID_NEW , "New", bmp); // -> transparent bg - good
+
+    // with mask (from xpm)
+    bmp = wxArtProvider::GetBitmap(wxART_QUIT, wxART_TOOLBAR);
+    wxASSERT(bmp.GetMask());
+    toolBar->AddTool(wxID_EXIT, "Exit", bmp); // -> solid black bg - bad
 
     toolBar->Realize();
 }
