--- debian/rules.orig	2010-07-18 20:39:20.000000000 +0700
+++ debian/rules	2010-10-29 21:21:03.000000000 +0700
@@ -832,8 +832,8 @@
 	mkdir -p debian/$(package_headers)/usr/include/wx-$(release)/wx/wxPython/i_files
 	cp -p wxPython/include/wx/wxPython/*.h \
 	    debian/$(package_headers)/usr/include/wx-$(release)/wx/wxPython/
-    cp -p wxPython/include/wx/wxPython/src/*.i wxPython/include/wx/wxPython/src/_*.py \
-        wxPython/include/wx/wxPython/src/*.swg \
+	cp -p wxPython/include/wx/wxPython/src/*.i wxPython/include/wx/wxPython/src/_*.py \
+	    wxPython/include/wx/wxPython/src/*.swg \
 	    debian/$(package_headers)/usr/include/wx-$(release)/wx/wxPython/i_files
 
 install-i18n: DH_OPTIONS=-p$(package_i18n)
