Index: configure.in
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/configure.in,v
retrieving revision 1.884
diff -u -2 -r1.884 configure.in
--- configure.in	2004/11/08 17:17:04	1.884
+++ configure.in	2004/11/11 00:28:22
@@ -3446,4 +3446,6 @@
 dnl   defines size_t if not already defined
 AC_TYPE_SIZE_T
+dnl   sets HAVE_SSIZE_T if ssize_t is defined
+AC_CHECK_TYPES(ssize_t)
 dnl   defines uid_t and gid_t if not already defined
 AC_TYPE_UID_T
Index: setup.h.in
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/setup.h.in,v
retrieving revision 1.152
diff -u -2 -r1.152 setup.h.in
--- setup.h.in	2004/11/01 18:40:43	1.152
+++ setup.h.in	2004/11/11 00:28:25
@@ -40,4 +40,7 @@
 #undef size_t
 
+/* Define if the compiler has the ssize_t type (signed size_t) */
+#undef HAVE_SSIZE_T
+
 /* Define if you have the ANSI C header files.  */
 #undef STDC_HEADERS
Index: setup.h_vms
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/setup.h_vms,v
retrieving revision 1.80
diff -u -2 -r1.80 setup.h_vms
--- setup.h_vms	2004/11/05 08:04:38	1.80
+++ setup.h_vms	2004/11/11 00:28:26
@@ -54,4 +54,7 @@
 #undef size_t
 
+/* Define if the compiler has the ssize_t type (signed size_t) */
+#undef HAVE_SSIZE_T
+
 /* Define if you have the ANSI C header files.  */
 #define STDC_HEADERS
Index: include/wx/defs.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/defs.h,v
retrieving revision 1.458
diff -u -2 -r1.458 defs.h
--- include/wx/defs.h	2004/11/09 03:22:56	1.458
+++ include/wx/defs.h	2004/11/11 00:29:07
@@ -886,4 +886,15 @@
 #endif
 
+/* Make sure ssize_t is defined (a signed type the same size as size_t) */
+/* HAVE_SSIZE_T should be defined for compiliers that already have it */
+#ifndef HAVE_SSIZE_T
+    #if SIZEOF_SIZE_T == 4
+        typedef wxInt32 ssize_t;
+    #elif SIZEOF_SIZE_T == 8
+        typedef wxInt64 ssize_t
+    #else
+        #error "error defining ssize_t, size_t is not 4 or 8 bytes"
+    #endif
+#endif
 
 /*  base floating point types */
Index: src/common/file.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/common/file.cpp,v
retrieving revision 1.91
diff -u -2 -r1.91 file.cpp
--- src/common/file.cpp	2004/11/10 21:02:20	1.91
+++ src/common/file.cpp	2004/11/11 00:29:18
@@ -145,13 +145,4 @@
 #endif
 
-#if !defined __UNIX__ && !defined __DJGPP__
-    #ifdef __WXWINCE__
-        typedef int ssize_t;
-    #else
-        typedef ptrdiff_t ssize_t;
-    #endif
-#endif
-wxCOMPILE_TIME_ASSERT(sizeof(ssize_t) == sizeof(size_t), ssize_t_wrong_size);
-
 // ============================================================================
 // implementation of wxFile

 	  	 
