Index: interface/wx/sizer.h
===================================================================
--- interface/wx/sizer.h	(revision 61629)
+++ interface/wx/sizer.h	(working copy)
@@ -789,31 +790,17 @@
 
         @see wxSizerItem::SetMinSize()
     */
-    bool SetItemMinSize(wxWindow* window, int width, int height);
+    bool SetItemMinSize( wxWindow* window, int width, int height );
+    bool SetItemMinSize( wxWindow* window, const wxSize& size );
 
-    /**
-        Set an item's minimum size by window, sizer, or position.
+    bool SetItemMinSize( wxSizer* sizer, int width, int height );
+    bool SetItemMinSize( wxSizer* sizer, const wxSize& size );
 
-        The item will be found recursively in the sizer's descendants.
-        This function enables an application to set the size of an item after
-        initial creation.
+    bool SetItemMinSize( size_t index, int width, int height );
+    bool SetItemMinSize( size_t index, const wxSize& size );
+    //@}
 
-        @see wxSizerItem::SetMinSize()
-    */
-    bool SetItemMinSize(wxSizer* sizer, int width, int height);
-
     /**
-        Set an item's minimum size by window, sizer, or position.
-
-        The item will be found recursively in the sizer's descendants.
-        This function enables an application to set the size of an item after
-        initial creation.
-
-        @see wxSizerItem::SetMinSize()
-    */
-    bool SetItemMinSize(size_t index, int width, int height);
-
-    /**
         Call this to give the sizer a minimal size.
 
         Normally, the sizer will calculate its minimal size based purely on how
