Index: src/common/sizer.cpp
===================================================================
--- src/common/sizer.cpp	(Revision 73196)
+++ src/common/sizer.cpp	(Arbeitskopie)
@@ -2102,11 +2102,9 @@
         minMajorSize += GetSizeInMajorDir(item->GetMinSizeWithBorder());
     }
 
-    // update our min size and delta which may have changed
+    // update our min size which may have changed
     SizeInMajorDir(m_minSize) = minMajorSize;
-    delta = totalMajorSize - minMajorSize;
 
-
     // space and sum of proportions for the remaining items, both may change
     // below
     wxCoord remaining = totalMajorSize;
