Index: src/msw/listctrl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/listctrl.cpp,v
retrieving revision 1.142.2.26
diff -b -u -2 -r1.142.2.26 listctrl.cpp
--- src/msw/listctrl.cpp	2003/08/13 10:06:02	1.142.2.26
+++ src/msw/listctrl.cpp	2003/10/30 18:23:36
@@ -1462,4 +1462,9 @@
             // take copy of attributes
             data->attr = new wxListItemAttr(*info.GetAttributes());
+
+						m_hasAnyAttr = TRUE;
+
+						// we need this to make the change visible right now
+						RefreshItem(item.iItem);
     }
     };
 	  	 
