 src/generic/treectlg.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/generic/treectlg.cpp b/src/generic/treectlg.cpp
index 0109bf016d..18f594b207 100644
--- a/src/generic/treectlg.cpp
+++ b/src/generic/treectlg.cpp
@@ -2854,8 +2854,8 @@ wxGenericTreeCtrl::PaintLevel(wxGenericTreeItem *item,
             }
             else // no custom buttons
             {
-                static const int wImage = 9;
-                static const int hImage = 9;
+                int wImage = FromDIP(9);
+                int hImage = FromDIP(9);
 
                 int flag = 0;
                 if (item->IsExpanded())
