diff --git a/samples/minimal/minimal.cpp b/samples/minimal/minimal.cpp
index a78e462..ec46eb3 100644
--- a/samples/minimal/minimal.cpp
+++ b/samples/minimal/minimal.cpp
@@ -172,6 +172,14 @@ MyFrame::MyFrame(const wxString& title)
     CreateStatusBar(2);
     SetStatusText("Welcome to wxWidgets!");
 #endif // wxUSE_STATUSBAR
+
+wxBoxSizer* mainsizer = new wxBoxSizer(wxVERTICAL);
+mainsizer->Add(new wxButton(this, wxID_ANY, "Button at top"), 0, wxEXPAND);
+wxStaticBoxSizer* sz = new wxStaticBoxSizer(wxVERTICAL, this, "This StaticBox should appear below the button");
+//sz->Add(new wxStaticText(sz->GetStaticBox(), wxID_ANY, "After this child is added to the staticboxsizer, it works correctly"), 0, wxTOP, 20);
+mainsizer->Add(sz, 1, wxEXPAND|wxTOP, 10);
+SetSizer(mainsizer);
+Layout();
 }
 
 
