Index: samples/minimal/minimal.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/samples/minimal/minimal.cpp,v
retrieving revision 1.70
diff -u -r1.70 minimal.cpp
--- samples/minimal/minimal.cpp	2005/10/08 21:55:33	1.70
+++ samples/minimal/minimal.cpp	2005/11/12 06:19:55
@@ -68,6 +68,8 @@
     void OnQuit(wxCommandEvent& event);
     void OnAbout(wxCommandEvent& event);
 
+    void OnButton1Click(wxCommandEvent&);
+
 private:
     // any class wishing to process wxWidgets events must use this macro
     DECLARE_EVENT_TABLE()
@@ -167,10 +169,30 @@
     CreateStatusBar(2);
     SetStatusText(_T("Welcome to wxWidgets!"));
 #endif // wxUSE_STATUSBAR
+
+    // default-button bug demonstration
+    wxPanel *panel = new wxPanel(this);
+    wxButton *button1 = new wxButton(panel, -1);
+    wxButton *button2 = new wxButton(panel, -1);
+
+    wxSizer *sizer = new wxBoxSizer(wxVERTICAL);
+    sizer->Add(button1);
+    sizer->Add(button2);
+    panel->SetSizer(sizer);
+
+    button2->SetDefault();
+    button1->Connect(wxEVT_COMMAND_BUTTON_CLICKED,
+                     wxCommandEventHandler(MyFrame::OnButton1Click),
+                     NULL, this);
 }
 
 
 // event handlers
+
+void MyFrame::OnButton1Click(wxCommandEvent &event)
+{
+    wxDynamicCast(event.GetEventObject(), wxButton)->Destroy();
+}
 
 void MyFrame::OnQuit(wxCommandEvent& WXUNUSED(event))
 {

 	  	 
