diff -ru2 -xCVS -x.#* -x*.orig -x*.rej -x*~ wxWidgets.orig/valgen.cpp wxWidgets/valgen.cpp
--- wxWidgets.orig/valgen.cpp	Wed Aug 24 10:35:43 2005
+++ wxWidgets/valgen.cpp	Wed Aug 24 11:13:49 2005
@@ -45,7 +45,7 @@
 #endif
 
-
+#if wxUSE_SPINCTRL
   #include "wx/spinctrl.h"
-
+#endif
 #if wxUSE_SPINBTN
   #include "wx/spinbutt.h"
@@ -54,4 +54,7 @@
   #include "wx/checklst.h"
 #endif
+#if wxUSE_LISTCTRL
+  #include "wx/listctrl.h"
+#endif
 
 #include "wx/valgen.h"
@@ -328,5 +331,27 @@
     } else
 #endif
-        ;   // to match the last 'else' above
+#if wxUSE_LISTCTRL
+    if (m_validatorWindow->IsKindOf(CLASSINFO(wxListCtrl)) )
+    {
+        wxListCtrl* pControl = (wxListCtrl*) m_validatorWindow;
+        long index = -1;
+        if (m_pInt)
+        {
+            index = *m_pInt;
+        }
+        else if (m_pString)
+        {
+            index = pControl->FindItem(-1, *m_pString);
+        }
+        if (index != -1)
+        {
+            pControl->SetItemState(index, wxLIST_STATE_SELECTED | wxLIST_STATE_FOCUSED, 
+                                          wxLIST_STATE_SELECTED | wxLIST_STATE_FOCUSED);
+            pControl->EnsureVisible(index);
+            return true;
+        }
+    } else
+#endif
+        ;   // to match the last 'else' above
 
   // unrecognized control, or bad pointer
@@ -555,5 +580,21 @@
     } else
 #endif
-
+#if wxUSE_LISTCTRL
+    if (m_validatorWindow->IsKindOf(CLASSINFO(wxListCtrl)) )
+    {
+        wxListCtrl* pControl = (wxListCtrl*) m_validatorWindow;
+        const int index = pControl->GetNextItem(-1, wxLIST_NEXT_ALL, wxLIST_STATE_SELECTED);
+        if (m_pInt)
+        {
+            *m_pInt = index;
+            return true;
+        }
+        else if (m_pString && (index != -1))
+        {
+            *m_pString = pControl->GetItemText(index);
+            return true;
+        }
+    } else
+#endif
     // unrecognized control, or bad pointer
         return false;

 	  	 
