Index: samples/minimal/minimal.cpp
===================================================================
--- samples/minimal/minimal.cpp	(revision 61567)
+++ samples/minimal/minimal.cpp	(working copy)
@@ -73,6 +73,12 @@
     DECLARE_EVENT_TABLE()
 };
 
+class MyPanel : public wxPanel
+{
+public:
+    MyPanel(wxWindow *aWindow):wxPanel(aWindow){}
+};
+
 // ----------------------------------------------------------------------------
 // constants
 // ----------------------------------------------------------------------------
@@ -127,6 +133,8 @@
     // create the main application window
     MyFrame *frame = new MyFrame("Minimal wxWidgets App");
 
+    frame->Maximize();
+
     // and show it (the frames, unlike simple controls, are not shown when
     // created initially)
     frame->Show(true);
@@ -169,9 +177,11 @@
 
 #if wxUSE_STATUSBAR
     // create a status bar just for fun (by default with 1 pane only)
-    CreateStatusBar(2);
-    SetStatusText("Welcome to wxWidgets!");
+    //CreateStatusBar(2);
+    //SetStatusText("Welcome to wxWidgets!");
 #endif // wxUSE_STATUSBAR
+
+    new MyPanel(this);
 }