Index: build/msw/wx_vc10_core.vcxproj
===================================================================
--- build/msw/wx_vc10_core.vcxproj	(revision 75741)
+++ build/msw/wx_vc10_core.vcxproj	(working copy)
@@ -1642,6 +1642,8 @@
     <ClInclude Include="..\..\include\wx\wxhtml.h" />
     <ClInclude Include="..\..\include\wx\xrc\xh_all.h" />
     <ClInclude Include="..\..\include\wx\xrc\xh_animatctrl.h" />
+    <ClInclude Include="..\..\include\wx\xrc\xh_auinotbk.h" />
+    <ClInclude Include="..\..\include\wx\xrc\xh_auitoolb.h" />
     <ClInclude Include="..\..\include\wx\xrc\xh_bannerwindow.h" />
     <ClInclude Include="..\..\include\wx\xrc\xh_bmp.h" />
     <ClInclude Include="..\..\include\wx\xrc\xh_bmpbt.h" />
@@ -1682,6 +1684,8 @@
     <ClInclude Include="..\..\include\wx\xrc\xh_propdlg.h" />
     <ClInclude Include="..\..\include\wx\xrc\xh_radbt.h" />
     <ClInclude Include="..\..\include\wx\xrc\xh_radbx.h" />
+    <ClInclude Include="..\..\include\wx\xrc\xh_ribbon.h" />
+    <ClInclude Include="..\..\include\wx\xrc\xh_richtext.h" />
     <ClInclude Include="..\..\include\wx\xrc\xh_scrol.h" />
     <ClInclude Include="..\..\include\wx\xrc\xh_scwin.h" />
     <ClInclude Include="..\..\include\wx\xrc\xh_sizer.h" />
Index: build/msw/wx_vc10_core.vcxproj.filters
===================================================================
--- build/msw/wx_vc10_core.vcxproj.filters	(revision 75741)
+++ build/msw/wx_vc10_core.vcxproj.filters	(working copy)
@@ -2332,6 +2332,12 @@
     <ClInclude Include="..\..\include\wx\xrc\xh_animatctrl.h">
       <Filter>Common Headers</Filter>
     </ClInclude>
+    <ClInclude Include="..\..\include\wx\xrc\xh_auinotbk.h">
+      <Filter>Common Headers</Filter>
+    </ClInclude>
+    <ClInclude Include="..\..\include\wx\xrc\xh_auitoolb.h">
+      <Filter>Common Headers</Filter>
+    </ClInclude>
     <ClInclude Include="..\..\include\wx\xrc\xh_bannerwindow.h">
       <Filter>Common Headers</Filter>
     </ClInclude>
@@ -2452,6 +2458,12 @@
     <ClInclude Include="..\..\include\wx\xrc\xh_radbx.h">
       <Filter>Common Headers</Filter>
     </ClInclude>
+    <ClInclude Include="..\..\include\wx\xrc\xh_ribbon.h">
+      <Filter>Common Headers</Filter>
+    </ClInclude>
+    <ClInclude Include="..\..\include\wx\xrc\xh_richtext.h">
+      <Filter>Common Headers</Filter>
+    </ClInclude>
     <ClInclude Include="..\..\include\wx\xrc\xh_scrol.h">
       <Filter>Common Headers</Filter>
     </ClInclude>
Index: build/msw/wx_vc10_ribbon.vcxproj
===================================================================
--- build/msw/wx_vc10_ribbon.vcxproj	(revision 75741)
+++ build/msw/wx_vc10_ribbon.vcxproj	(working copy)
@@ -481,6 +481,7 @@
     <ClCompile Include="..\..\src\ribbon\page.cpp" />
     <ClCompile Include="..\..\src\ribbon\panel.cpp" />
     <ClCompile Include="..\..\src\ribbon\toolbar.cpp" />
+    <ClCompile Include="..\..\src\xrc\xh_ribbon.cpp" />
   </ItemGroup>
   <ItemGroup>
     <ResourceCompile Include="..\..\src\msw\version.rc">
@@ -560,6 +561,7 @@
     <ClInclude Include="..\..\include\wx\ribbon\page.h" />
     <ClInclude Include="..\..\include\wx\ribbon\panel.h" />
     <ClInclude Include="..\..\include\wx\ribbon\toolbar.h" />
+    <ClInclude Include="..\..\include\wx\xrc\xh_ribbon.h" />
   </ItemGroup>
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.targets" />
   <ImportGroup Label="ExtensionTargets">
Index: build/msw/wx_vc10_ribbon.vcxproj.filters
===================================================================
--- build/msw/wx_vc10_ribbon.vcxproj.filters	(revision 75741)
+++ build/msw/wx_vc10_ribbon.vcxproj.filters	(working copy)
@@ -55,6 +55,9 @@
     <ClCompile Include="..\..\src\ribbon\toolbar.cpp">
       <Filter>Source Files</Filter>
     </ClCompile>
+    <ClCompile Include="..\..\src\xrc\xh_ribbon.cpp">
+      <Filter>Source Files</Filter>
+    </ClCompile>
   </ItemGroup>
   <ItemGroup>
     <ResourceCompile Include="..\..\src\msw\version.rc">
@@ -89,6 +92,9 @@
     <ClInclude Include="..\..\include\wx\ribbon\toolbar.h">
       <Filter>Common Headers</Filter>
     </ClInclude>
+    <ClInclude Include="..\..\include\wx\xrc\xh_ribbon.h">
+      <Filter>Source Files</Filter>
+    </ClCompile>
   </ItemGroup>
   <ItemGroup>
     <CustomBuild Include="..\..\include\wx\msw\setup.h">
