Index: src/common/appbase.cpp
===================================================================
--- src/common/appbase.cpp	(revision 143)
+++ src/common/appbase.cpp	(working copy)
@@ -370,6 +370,13 @@
     wxLog::FlushActive();
 #endif
 
+    // When user idle event handling code
+    //   - schedules an object for deletion, and
+    //   - continuously requests more idle events,
+    // then without this, the objects are collected in ProcessPendingEvents()
+    // only when the next *external* event occurs.
+    DeletePendingObjects();
+
     return event.MoreRequested();
 }
 
Index: src/common/evtloopcmn.cpp
===================================================================
--- src/common/evtloopcmn.cpp	(revision 143)
+++ src/common/evtloopcmn.cpp	(working copy)
@@ -139,7 +139,7 @@
 
                 // generate and process idle events for as long as we don't
                 // have anything else to do
-                while ( !Pending() && ProcessIdle() )
+                while ( !Pending() && ProcessIdle() && !m_shouldExit )
                     ;
 
                 // if the "should exit" flag is set, the loop should terminate
