Index: thread.h
===================================================================
--- thread.h	(revision 52103)
+++ thread.h	(working copy)
@@ -553,6 +553,12 @@
     // NB: this function will not be called if the thread is Kill()ed
     virtual void OnExit() { }
 
+    // called when the thread is deleted - in the context of this thread
+	virtual void OnDelete(ExitCode *pRc) {}
+
+    // called when the thread is killed - in the context of this thread
+	virtual void OnKill() {}
+
     // Returns true if the thread was asked to terminate: this function should
     // be called by the thread from time to time, otherwise the main thread
     // will be left forever in Delete()!

 	  	 
