Index: E:/Dev/wxWindows/samples/wizard/wizard.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/samples/wizard/wizard.cpp,v
retrieving revision 1.8
diff -u -2 -r1.8 wizard.cpp
--- E:/Dev/wxWindows/samples/wizard/wizard.cpp	2002/12/13 21:33:14	1.8
+++ E:/Dev/wxWindows/samples/wizard/wizard.cpp	2003/01/14 21:24:24
@@ -121,4 +121,32 @@
 };
 
+class wxMyPanel : public wxPanel
+{
+public:
+    wxMyPanel(wxWindow *parent,
+            int x, int y, int width, int height,
+            long style = wxTAB_TRAVERSAL | wxNO_BORDER)
+            :wxPanel(parent, x, y, width, height, style)
+    {}
+#if 0   
+    void OnPaint(wxPaintEvent& event)
+    {
+        wxPanel::OnPaint(event);        
+    }
+    
+    void OnErase(wxEraseEvent& event)
+    {
+        wxPanel::OnErase(event);
+    }
+#endif
+    DECLARE_EVENT_TABLE()
+};
+
+
+BEGIN_EVENT_TABLE(wxMyPanel, wxPanel)
+    //EVT_PAINT(wxMyPanel::OnPaint)
+    //EVT_ERASE_BACKGROUND(wxMyPanel::OnErase)
+END_EVENT_TABLE()
+
 // This is a more complicated example of validity checking: using events we may
 // allow to return to the previous page, but not to proceed. It also
@@ -133,5 +161,6 @@
     };
 
-    wxRadioboxPage(wxWizard *parent) : wxWizardPageSimple(parent)
+    wxRadioboxPage(wxWizard *parent) 
+        : wxWizardPageSimple(parent)
     {
         // should correspond to the enum above
@@ -144,10 +173,35 @@
         choices[3] = _T("neither");
 
-        m_radio = new wxRadioBox(this, -1, _T("Allow to proceed:"),
+        pPanel = new wxMyPanel(this, 0, 0, 200, 200);
+
+        m_radio = new wxRadioBox(pPanel, -1, _T("Allow to proceed:"),
                                  wxPoint(5, 5), wxDefaultSize,
                                  WXSIZEOF(choices), choices,
                                  1, wxRA_SPECIFY_COLS);
         m_radio->SetSelection(Both);
+
+        new wxButton(pPanel, -1, "test", wxPoint(50, 50), wxDefaultSize);
+
+        wxPanel::SetBackgroundColour(*wxRED);
+        pPanel->SetBackgroundColour(*wxGREEN);
+
+        m_dbg = FALSE;
+    }
+
+#if 0
+    void OnPaint(wxPaintEvent& event)
+    {
+        if (m_dbg)
+            int test = 1 + 5;
+        wxWindow::OnPaint(event); 
+    }
+    
+    void OnErase(wxEraseEvent& event)
+    {
+        if (m_dbg)
+            int test = 1 + 5;
+        wxWindow::OnErase(event);
     }
+#endif
 
     // wizard event handlers
@@ -175,7 +229,11 @@
             return;
 
+        m_dbg = TRUE;
+
         wxMessageBox(_T("You can't go there"), _T("Not allowed"),
                      wxICON_WARNING | wxOK, this);
 
+        m_dbg = FALSE;
+
         event.Veto();
     }
@@ -183,4 +241,6 @@
 private:
     wxRadioBox *m_radio;
+    bool m_dbg;
+    wxPanel *pPanel;
 
     DECLARE_EVENT_TABLE()
@@ -239,4 +299,6 @@
     EVT_WIZARD_PAGE_CHANGING(-1, wxRadioboxPage::OnWizardPageChanging)
     EVT_WIZARD_CANCEL(-1, wxRadioboxPage::OnWizardCancel)
+
+    //EVT_PAINT(wxRadioboxPage::OnPaint)
 END_EVENT_TABLE()
 
Index: E:/Dev/wxWindows/samples/wizard/wizard.dsp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/samples/wizard/wizard.dsp,v
retrieving revision 1.4
diff -u -2 -r1.4 wizard.dsp
--- E:/Dev/wxWindows/samples/wizard/wizard.dsp	2002/12/05 11:07:22	1.4
+++ E:/Dev/wxWindows/samples/wizard/wizard.dsp	2003/01/14 21:24:24
@@ -5,5 +5,5 @@
 # TARGTYPE "Win32 (x86) Application" 0x0101
 
-CFG=wizard - Win32 Debug
+CFG=wizard - Win32 wxUniv Debug
 !MESSAGE This is not a valid makefile. To build this project using NMAKE,
 !MESSAGE use the Export Makefile command and run
@@ -14,5 +14,5 @@
 !MESSAGE by defining the macro CFG on the command line. For example:
 !MESSAGE 
-!MESSAGE NMAKE /f "wizard.mak" CFG="wizard - Win32 Debug"
+!MESSAGE NMAKE /f "wizard.mak" CFG="wizard - Win32 wxUniv Debug"
 !MESSAGE 
 !MESSAGE Possible choices for configuration are:
@@ -22,7 +22,9 @@
 !MESSAGE "wizard - Win32 Release" (based on "Win32 (x86) Application")
 !MESSAGE "wizard - Win32 Debug" (based on "Win32 (x86) Application")
+!MESSAGE "wizard - Win32 wxUniv Debug" (based on "Win32 (x86) Application")
 !MESSAGE 
 
 # Begin Project
+# PROP AllowPerConfigDependencies 0
 # PROP Scc_ProjName ""
 # PROP Scc_LocalPath ""
@@ -31,5 +33,5 @@
 RSC=rc.exe
 
-!ELSEIF "$(CFG)" == "wizard - Win32 Release DLL"
+!IF  "$(CFG)" == "wizard - Win32 Release DLL"
 
 # PROP BASE Use_MFC 0
@@ -45,9 +47,9 @@
 # PROP Target_Dir ""
 # ADD BASE CPP /nologo /W4 /O2 /D "NDEBUG" /D "WIN32" /D "_WINDOWS" /D WINVER=0x400 /YX /FD /c
-# ADD CPP /nologo /W4 /O2 /I "../../include" /D "NDEBUG" /D "WIN32" /D "_WINDOWS" /D WINVER=0x400 /D WIN32 /D WINVER=0x400 /D _MT /D wxUSE_GUI=1 /YX /FD /c /MD  /D WXUSINGDLL  /I..\..\lib\mswdll 
-# ADD BASE MTL /nologo /D "NDEBUG" /mktyplib203 /o NUL /win32
-# ADD MTL /nologo /D "NDEBUG" /mktyplib203 /o NUL /win32
-# ADD BASE RSC /l 0x409 /d "NDEBUG" /I "../../include"
-# ADD RSC /l 0x409 /d "NDEBUG" /I "../../include"
+# ADD CPP /nologo /MD /W4 /O2 /I "../../include" /I "..\..\lib\mswdll" /D "NDEBUG" /D "WIN32" /D "_WINDOWS" /D WINVER=0x400 /D "_MT" /D wxUSE_GUI=1 /D "WXUSINGDLL" /YX /FD /c
+# ADD BASE MTL /nologo /D "NDEBUG" /mktyplib203 /o "NUL" /win32
+# ADD MTL /nologo /D "NDEBUG" /mktyplib203 /o "NUL" /win32
+# ADD BASE RSC /l 0x409 /i "../../include" /d "NDEBUG"
+# ADD RSC /l 0x409 /i "../../include" /d "NDEBUG"
 BSC32=bscmake.exe
 # ADD BASE BSC32 /nologo
@@ -55,7 +57,7 @@
 LINK32=link.exe
 # ADD BASE LINK32 kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib comctl32.lib rpcrt4.lib wsock32.lib /nologo /subsystem:windows /machine:I386
-# ADD LINK32 kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib comctl32.lib rpcrt4.lib wsock32.lib /nologo /subsystem:windows /machine:I386 ..\..\lib\wxmsw250.lib
+# ADD LINK32 kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib comctl32.lib rpcrt4.lib wsock32.lib ..\..\lib\wxmsw250.lib /nologo /subsystem:windows /machine:I386
 
-!ELSEIF "$(CFG)" == "wizard - Win32 Debug DLL"
+!ELSEIF  "$(CFG)" == "wizard - Win32 Debug DLL"
 
 # PROP BASE Use_MFC 0
@@ -71,9 +73,9 @@
 # PROP Target_Dir ""
 # ADD BASE CPP /nologo /W4 /Zi /Od /D "_DEBUG" /D "WIN32" /D "_WINDOWS" /D WINVER=0x400 /YX /FD /c
-# ADD CPP /nologo /W4 /Zi /Od /I "../../include" /D "_DEBUG" /D "WIN32" /D "_WINDOWS" /D WINVER=0x400 /D WIN32 /D WINVER=0x400 /D _MT /D wxUSE_GUI=1 /YX /FD /c /MDd /D "__WXDEBUG__" /D "WXDEBUG=1"  /D WXUSINGDLL  /I..\..\lib\mswdlld 
-# ADD BASE MTL /nologo /D "_DEBUG" /mktyplib203 /o NUL /win32
-# ADD MTL /nologo /D "_DEBUG" /mktyplib203 /o NUL /win32
-# ADD BASE RSC /l 0x409 /d "_DEBUG" /I "../../include"
-# ADD RSC /l 0x409 /d "_DEBUG" /I "../../include"
+# ADD CPP /nologo /MDd /W4 /Zi /Od /I "../../include" /I "..\..\lib\mswdlld" /D "_DEBUG" /D "WIN32" /D "_WINDOWS" /D WINVER=0x400 /D "_MT" /D wxUSE_GUI=1 /D "__WXDEBUG__" /D WXDEBUG=1 /D "WXUSINGDLL" /YX /FD /c
+# ADD BASE MTL /nologo /D "_DEBUG" /mktyplib203 /o "NUL" /win32
+# ADD MTL /nologo /D "_DEBUG" /mktyplib203 /o "NUL" /win32
+# ADD BASE RSC /l 0x409 /i "../../include" /d "_DEBUG"
+# ADD RSC /l 0x409 /i "../../include" /d "_DEBUG"
 BSC32=bscmake.exe
 # ADD BASE BSC32 /nologo
@@ -81,7 +83,7 @@
 LINK32=link.exe
 # ADD BASE LINK32 kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib comctl32.lib rpcrt4.lib wsock32.lib /nologo /subsystem:windows /debug /machine:I386 /pdbtype:sept
-# ADD LINK32 kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib comctl32.lib rpcrt4.lib wsock32.lib /nologo /subsystem:windows /debug /machine:I386 /pdbtype:sept ..\..\lib\wxmsw250d.lib
+# ADD LINK32 kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib comctl32.lib rpcrt4.lib wsock32.lib ..\..\lib\wxmsw250d.lib /nologo /subsystem:windows /debug /machine:I386 /pdbtype:sept
 
-!ELSEIF "$(CFG)" == "wizard - Win32 Release"
+!ELSEIF  "$(CFG)" == "wizard - Win32 Release"
 
 # PROP BASE Use_MFC 0
@@ -97,9 +99,9 @@
 # PROP Target_Dir ""
 # ADD BASE CPP /nologo /W4 /O2 /D "NDEBUG" /D "WIN32" /D "_WINDOWS" /D WINVER=0x400 /YX /FD /c
-# ADD CPP /nologo /W4 /O2 /I "../../include" /D "NDEBUG" /D "WIN32" /D "_WINDOWS" /D WINVER=0x400 /D WIN32 /D WINVER=0x400 /D _MT /D wxUSE_GUI=1 /YX /FD /c /MD  /I..\..\lib\msw 
-# ADD BASE MTL /nologo /D "NDEBUG" /mktyplib203 /o NUL /win32
-# ADD MTL /nologo /D "NDEBUG" /mktyplib203 /o NUL /win32
-# ADD BASE RSC /l 0x409 /d "NDEBUG" /I "../../include"
-# ADD RSC /l 0x409 /d "NDEBUG" /I "../../include"
+# ADD CPP /nologo /MD /W4 /O2 /I "../../include" /I "..\..\lib\msw" /D "NDEBUG" /D "WIN32" /D "_WINDOWS" /D WINVER=0x400 /D "_MT" /D wxUSE_GUI=1 /YX /FD /c
+# ADD BASE MTL /nologo /D "NDEBUG" /mktyplib203 /o "NUL" /win32
+# ADD MTL /nologo /D "NDEBUG" /mktyplib203 /o "NUL" /win32
+# ADD BASE RSC /l 0x409 /i "../../include" /d "NDEBUG"
+# ADD RSC /l 0x409 /i "../../include" /d "NDEBUG"
 BSC32=bscmake.exe
 # ADD BASE BSC32 /nologo
@@ -107,7 +109,7 @@
 LINK32=link.exe
 # ADD BASE LINK32 kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib comctl32.lib rpcrt4.lib wsock32.lib /nologo /subsystem:windows /machine:I386
-# ADD LINK32 kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib comctl32.lib rpcrt4.lib wsock32.lib /nologo /subsystem:windows /machine:I386 ..\..\lib\zlib.lib ..\..\lib\regex.lib ..\..\lib\png.lib ..\..\lib\jpeg.lib ..\..\lib\tiff.lib ..\..\lib\wxmsw.lib
+# ADD LINK32 kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib comctl32.lib rpcrt4.lib wsock32.lib ..\..\lib\zlib.lib ..\..\lib\regex.lib ..\..\lib\png.lib ..\..\lib\jpeg.lib ..\..\lib\tiff.lib ..\..\lib\wxmsw.lib /nologo /subsystem:windows /machine:I386
 
-!ELSEIF "$(CFG)" == "wizard - Win32 Debug"
+!ELSEIF  "$(CFG)" == "wizard - Win32 Debug"
 
 # PROP BASE Use_MFC 0
@@ -123,9 +125,9 @@
 # PROP Target_Dir ""
 # ADD BASE CPP /nologo /W4 /Zi /Od /D "_DEBUG" /D "WIN32" /D "_WINDOWS" /D WINVER=0x400 /YX /FD /c
-# ADD CPP /nologo /W4 /Zi /Od /I "../../include" /D "_DEBUG" /D "WIN32" /D "_WINDOWS" /D WINVER=0x400 /D WIN32 /D WINVER=0x400 /D _MT /D wxUSE_GUI=1 /YX /FD /c /MDd /D "__WXDEBUG__" /D "WXDEBUG=1"  /I..\..\lib\mswd 
-# ADD BASE MTL /nologo /D "_DEBUG" /mktyplib203 /o NUL /win32
-# ADD MTL /nologo /D "_DEBUG" /mktyplib203 /o NUL /win32
-# ADD BASE RSC /l 0x409 /d "_DEBUG" /I "../../include"
-# ADD RSC /l 0x409 /d "_DEBUG" /I "../../include"
+# ADD CPP /nologo /MDd /W4 /Zi /Od /I "../../include" /I "..\..\lib\mswd" /D "_DEBUG" /D "WIN32" /D "_WINDOWS" /D WINVER=0x400 /D "_MT" /D wxUSE_GUI=1 /D "__WXDEBUG__" /D WXDEBUG=1 /YX /FD /c
+# ADD BASE MTL /nologo /D "_DEBUG" /mktyplib203 /o "NUL" /win32
+# ADD MTL /nologo /D "_DEBUG" /mktyplib203 /o "NUL" /win32
+# ADD BASE RSC /l 0x409 /i "../../include" /d "_DEBUG"
+# ADD RSC /l 0x409 /i "../../include" /d "_DEBUG"
 BSC32=bscmake.exe
 # ADD BASE BSC32 /nologo
@@ -133,6 +135,33 @@
 LINK32=link.exe
 # ADD BASE LINK32 kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib comctl32.lib rpcrt4.lib wsock32.lib /nologo /subsystem:windows /debug /machine:I386 /pdbtype:sept
-# ADD LINK32 kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib comctl32.lib rpcrt4.lib wsock32.lib /nologo /subsystem:windows /debug /machine:I386 /pdbtype:sept ..\..\lib\zlibd.lib ..\..\lib\regexd.lib ..\..\lib\pngd.lib ..\..\lib\jpegd.lib ..\..\lib\tiffd.lib ..\..\lib\wxmswd.lib
+# ADD LINK32 kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib comctl32.lib rpcrt4.lib wsock32.lib ..\..\lib\zlibd.lib ..\..\lib\regexd.lib ..\..\lib\pngd.lib ..\..\lib\jpegd.lib ..\..\lib\tiffd.lib ..\..\lib\wxmswd.lib /nologo /subsystem:windows /debug /machine:I386 /pdbtype:sept
 
+!ELSEIF  "$(CFG)" == "wizard - Win32 wxUniv Debug"
+
+# PROP BASE Use_MFC 0
+# PROP BASE Use_Debug_Libraries 1
+# PROP BASE Output_Dir "wizard___Win32_wxUniv_Debug0"
+# PROP BASE Intermediate_Dir "wizard___Win32_wxUniv_Debug0"
+# PROP BASE Ignore_Export_Lib 0
+# PROP BASE Target_Dir ""
+# PROP Use_MFC 0
+# PROP Use_Debug_Libraries 1
+# PROP Output_Dir "wizard___Win32_wxUniv_Debug0"
+# PROP Intermediate_Dir "wizard___Win32_wxUniv_Debug0"
+# PROP Ignore_Export_Lib 0
+# PROP Target_Dir ""
+# ADD BASE CPP /nologo /MDd /W4 /Zi /Od /I "../../include" /I "..\..\lib\mswd" /D "_DEBUG" /D "WIN32" /D "_WINDOWS" /D WINVER=0x400 /D "_MT" /D wxUSE_GUI=1 /D "__WXDEBUG__" /D WXDEBUG=1 /YX /FD /c
+# ADD CPP /nologo /MDd /W4 /Zi /Od /I "../../include" /I "..\..\lib\univd" /D "_DEBUG" /D "WIN32" /D "_WINDOWS" /D WINVER=0x400 /D "_MT" /D wxUSE_GUI=1 /D "__WXDEBUG__" /D WXDEBUG=1 /D "__WXUNIVERSAL__" /YX /FD /c
+# ADD BASE MTL /nologo /D "_DEBUG" /mktyplib203 /o "NUL" /win32
+# ADD MTL /nologo /D "_DEBUG" /mktyplib203 /o "NUL" /win32
+# ADD BASE RSC /l 0x409 /i "../../include" /d "_DEBUG"
+# ADD RSC /l 0x409 /i "../../include" /d "_DEBUG"
+BSC32=bscmake.exe
+# ADD BASE BSC32 /nologo
+# ADD BSC32 /nologo
+LINK32=link.exe
+# ADD BASE LINK32 kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib comctl32.lib rpcrt4.lib wsock32.lib ..\..\lib\zlibd.lib ..\..\lib\regexd.lib ..\..\lib\pngd.lib ..\..\lib\jpegd.lib ..\..\lib\tiffd.lib ..\..\lib\wxmswd.lib /nologo /subsystem:windows /debug /machine:I386 /pdbtype:sept
+# ADD LINK32 kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib comctl32.lib rpcrt4.lib wsock32.lib ..\..\lib\zlibd.lib ..\..\lib\regexd.lib ..\..\lib\pngd.lib ..\..\lib\jpegd.lib ..\..\lib\tiffd.lib ..\..\lib\wxunivd.lib /nologo /subsystem:windows /debug /machine:I386 /pdbtype:sept
+
 !ENDIF 
 
@@ -143,4 +172,5 @@
 # Name "wizard - Win32 Release"
 # Name "wizard - Win32 Debug"
+# Name "wizard - Win32 wxUniv Debug"
 # Begin Source File
 

 	  	 
