diff -bru wxWidgets.orig/include/wx/font.h wxWidgets/include/wx/font.h
--- wxWidgets.orig/include/wx/font.h	2015-05-17 00:10:20.000000000 -0400
+++ wxWidgets/include/wx/font.h	2015-06-08 08:08:41.000000000 -0400
@@ -62,6 +62,12 @@
     wxFONTWEIGHT_NORMAL = wxNORMAL,
     wxFONTWEIGHT_LIGHT = wxLIGHT,
     wxFONTWEIGHT_BOLD = wxBOLD,
+    wxFONTWEIGHT_THIN,
+    wxFONTWEIGHT_EXTRALIGHT,
+    wxFONTWEIGHT_MEDIUM,
+    wxFONTWEIGHT_SEMIBOLD,
+    wxFONTWEIGHT_EXTRABOLD,
+    wxFONTWEIGHT_HEAVY,
     wxFONTWEIGHT_MAX
 };
 
diff -bru wxWidgets.orig/interface/wx/font.h wxWidgets/interface/wx/font.h
--- wxWidgets.orig/interface/wx/font.h	2015-05-17 00:10:21.000000000 -0400
+++ wxWidgets/interface/wx/font.h	2015-06-11 23:28:10.000000000 -0400
@@ -64,9 +64,15 @@
 */
 enum wxFontWeight
 {
-    wxFONTWEIGHT_NORMAL = wxNORMAL,  //!< Normal font.
+    wxFONTWEIGHT_LIGHT,
+    wxFONTWEIGHT_EXTRALIGHT,
     wxFONTWEIGHT_LIGHT = wxLIGHT,    //!< Light font.
+    wxFONTWEIGHT_NORMAL = wxNORMAL,  //!< Normal font.
+    wxFONTWEIGHT_MEDIUM,
+    wxFONTWEIGHT_SEMIBOLD,
     wxFONTWEIGHT_BOLD = wxBOLD,      //!< Bold font.
+    wxFONTWEIGHT_EXTRABOLD,
+    wxFONTWEIGHT_HEAVY,
     wxFONTWEIGHT_MAX
 };
 
diff -bru wxWidgets.orig/samples/font/font.cpp wxWidgets/samples/font/font.cpp
--- wxWidgets.orig/samples/font/font.cpp	2015-05-17 00:10:26.000000000 -0400
+++ wxWidgets/samples/font/font.cpp	2015-06-11 22:37:58.000000000 -0400
@@ -709,9 +709,46 @@
 
 void MyFrame::OnBold(wxCommandEvent& event)
 {
+    wxFontWeight weight;
+    const wxString choices[] = { "THIN", "EXTRALIGHT", "LIGHT", "NORMAL", "MEDIUM", "SEMIBOLD", "BOLD", "EXTRABOLD", "HEAVY" };
     wxFont font = m_canvas->GetTextFont();
 
-    font.SetWeight(event.IsChecked() ? wxFONTWEIGHT_BOLD : wxFONTWEIGHT_NORMAL);
+    wxSingleChoiceDialog dlg( this, "", "Please select boldness type", WXSIZEOF( choices ), choices );
+    dlg.SetSelection( 3 );
+    if( dlg.ShowModal() == wxID_OK )
+    {
+        switch( dlg.GetSelection() )
+        {
+        case 0:
+             weight = wxFONTWEIGHT_THIN;
+             break;
+        case 1:
+             weight = wxFONTWEIGHT_EXTRALIGHT;
+             break;
+        case 2:
+             weight = wxFONTWEIGHT_LIGHT;
+             break;
+        case 3:
+             weight = wxFONTWEIGHT_NORMAL;
+             break;
+        case 4:
+             weight = wxFONTWEIGHT_MEDIUM;
+             break;
+        case 5:
+             weight = wxFONTWEIGHT_SEMIBOLD;
+             break;
+        case 6:
+             weight = wxFONTWEIGHT_BOLD;
+             break;
+        case 7:
+             weight = wxFONTWEIGHT_EXTRABOLD;
+             break;
+        case 8:
+             weight = wxFONTWEIGHT_HEAVY;
+             break;
+        }
+    }
+    font.SetWeight( weight );
     DoChangeFont(font);
 }
 
diff -bru wxWidgets.orig/src/common/fontcmn.cpp wxWidgets/src/common/fontcmn.cpp
--- wxWidgets.orig/src/common/fontcmn.cpp	2015-05-30 19:44:15.000000000 -0400
+++ wxWidgets/src/common/fontcmn.cpp	2015-06-11 23:14:09.000000000 -0400
@@ -453,9 +453,15 @@
 
     switch ( GetWeight() )
     {
+        case wxFONTWEIGHT_THIN:     return "wxFONTWEIGHT_THIN";
+        case wxFONTWEIGHT_EXTRALIGHT: return "wxFONTWEIGHT_EXTRALIGHT";
+        case wxFONTWEIGHT_LIGHT:    return "wxFONTWEIGHT_LIGHT";
         case wxFONTWEIGHT_NORMAL:   return "wxFONTWEIGHT_NORMAL";
+        case wxFONTWEIGHT_MEDIUM:   return "wxFONTWEIGHT_MEDIUM";
+        case wxFONTWEIGHT_SEMIBOLD: return "wxFONTWEIGHT_SEMIBOLD";
         case wxFONTWEIGHT_BOLD:     return "wxFONTWEIGHT_BOLD";
-        case wxFONTWEIGHT_LIGHT:    return "wxFONTWEIGHT_LIGHT";
+        case wxFONTWEIGHT_EXTRABOLD: return "wxFONTWEIGHT_EXTRABOLD";
+        case wxFONTWEIGHT_HEAVY:    return "wxFONTWEIGHT_HEAVY";
         default:                    return "wxFONTWEIGHT_DEFAULT";
     }
 }
@@ -807,9 +813,13 @@
 
     switch ( GetWeight() )
     {
-        default:
-            wxFAIL_MSG( wxT("unknown font weight") );
-            wxFALLTHROUGH;
+        case wxFONTWEIGHT_THIN:
+            desc << _( "thin" );
+            break;
+
+        case wxFONTWEIGHT_EXTRALIGHT:
+            desc << _( "extralight" );
+            break;
 
         case wxFONTWEIGHT_NORMAL:
             break;
@@ -818,9 +828,29 @@
             desc << _(" light");
             break;
 
+        case wxFONTWEIGHT_MEDIUM:
+            desc << _( "medium" );
+            break;
+
+        case wxFONTWEIGHT_SEMIBOLD:
+            desc << _( "semi-bold" );
+            break;
+            
         case wxFONTWEIGHT_BOLD:
             desc << _(" bold");
             break;
+
+        case wxFONTWEIGHT_EXTRABOLD:
+            desc << _( "extra-bold" );
+            break;
+
+        case wxFONTWEIGHT_HEAVY:
+            desc << _( "heavy" );
+            break;
+
+        default:
+            wxFAIL_MSG( wxT("unknown font weight") );
+            wxFALLTHROUGH;
     }
 
     switch ( GetStyle() )
@@ -981,16 +1011,56 @@
         {
             SetStrikethrough(true);
         }
+        else if ( token == wxT("thin") || token == _("thin") )
+        {
+                SetWeight(wxFONTWEIGHT_THIN);
+                weightfound = true;
+        }
+        else if ( token == wxT("extralight") || token == _("extralight") )
+        {
+                SetWeight(wxFONTWEIGHT_EXTRALIGHT);
+                weightfound = true;
+        }
         else if ( token == wxT("light") || token == _("light") )
         {
             SetWeight(wxFONTWEIGHT_LIGHT);
             weightfound = true;
         }
+        else if ( token == wxT("normal") || token == _("normal") )
+        {
+                SetWeight(wxFONTWEIGHT_NORMAL);
+                weightfound = true;
+        }
+        else if ( token == wxT("medium") || token == _("medium") )
+        {
+                SetWeight(wxFONTWEIGHT_MEDIUM);
+                weightfound = true;
+        }
+        else if ( token == wxT("semi-bold") || token == _("semi-bold") )
+        {
+                SetWeight(wxFONTWEIGHT_SEMIBOLD);
+                weightfound = true;
+        }
         else if ( token == wxT("bold") || token == _("bold") )
         {
             SetWeight(wxFONTWEIGHT_BOLD);
             weightfound = true;
         }
+        else if ( token == wxT("extra-bold") || token == _("extra-bold") )
+        {
+                SetWeight(wxFONTWEIGHT_EXTRABOLD);
+                weightfound = true;
+        }
+        else if ( token == wxT("semi-bold") || token == _("semi-bold") )
+        {
+                SetWeight(wxFONTWEIGHT_SEMIBOLD);
+                weightfound = true;
+        }
+        else if ( token == wxT("heavy") || token == _("heavy") )
+        {
+                SetWeight(wxFONTWEIGHT_HEAVY);
+                weightfound = true;
+        }
         else if ( token == wxT("italic") || token == _("italic") )
         {
             SetStyle(wxFONTSTYLE_ITALIC);
diff -bru wxWidgets.orig/src/msw/font.cpp wxWidgets/src/msw/font.cpp
--- wxWidgets.orig/src/msw/font.cpp	2015-05-17 00:10:33.000000000 -0400
+++ wxWidgets/src/msw/font.cpp	2015-06-10 00:09:26.000000000 -0400
@@ -465,13 +465,36 @@
 
 wxFontWeight wxNativeFontInfo::GetWeight() const
 {
-    if ( lf.lfWeight <= 300 )
-        return wxFONTWEIGHT_LIGHT;
+    wxFontWeight weight;
+    if( lf.lfWeight <= 100 )
+         weight = wxFONTWEIGHT_THIN;
 
-    if ( lf.lfWeight >= 600 )
-        return wxFONTWEIGHT_BOLD;
+    if( lf.lfWeight > 100 && lf.lfWeight <= 200 )
+         weight = wxFONTWEIGHT_EXTRALIGHT;
 
-    return wxFONTWEIGHT_NORMAL;
+    -    return wxFONTWEIGHT_NORMAL;
+    if( lf.lfWeight > 200 && lf.lfWeight <= 300 )
+         weight = wxFONTWEIGHT_LIGHT;
+    
+    if( lf.lfWeight > 300 && lf.lfWeight <= 400 )
+         weight = wxFONTWEIGHT_NORMAL;
+    
+    if( lf.lfWeight > 400 && lf.lfWeight <= 500 )
+         weight = wxFONTWEIGHT_MEDIUM;
+
+    if( lf.lfWeight > 500 && lf.lfWeight <= 600 )
+         weight = wxFONTWEIGHT_SEMIBOLD;
+    
+    if( lf.lfWeight > 600 && lf.lfWeight <= 700 )
+         weight = wxFONTWEIGHT_BOLD;
+    
+    if( lf.lfWeight > 700 && lf.lfWeight <= 800 )
+        weight = wxFONTWEIGHT_EXTRABOLD;
+    
+    if( lf.lfWeight > 800 && lf.lfWeight <= 900 )
+        weight = wxFONTWEIGHT_HEAVY;
+    
+    return weight;
 }
 
 bool wxNativeFontInfo::GetUnderlined() const
@@ -585,6 +608,14 @@
             wxFAIL_MSG( "unknown font weight" );
             // fall through
 
+        case wxFONTWEIGHT_THIN:
+            lf.lfWeight = FW_LIGHT;
+            break;
+            
+        case wxFONTWEIGHT_EXTRALIGHT:
+            lf.lfWeight = FW_EXTRALIGHT;
+            break;
+            
         case wxFONTWEIGHT_NORMAL:
             lf.lfWeight = FW_NORMAL;
             break;
@@ -593,9 +624,25 @@
             lf.lfWeight = FW_LIGHT;
             break;
 
+        case wxFONTWEIGHT_MEDIUM:
+            lf.lfWeight = FW_MEDIUM;
+            break;
+
+        case wxFONTWEIGHT_SEMIBOLD:
+            lf.lfWeight = FW_SEMIBOLD;
+            break;
+
         case wxFONTWEIGHT_BOLD:
             lf.lfWeight = FW_BOLD;
             break;
+
+        case wxFONTWEIGHT_EXTRABOLD:
+            lf.lfWeight = FW_EXTRABOLD;
+            break;
+
+        case wxFONTWEIGHT_HEAVY:
+            lf.lfWeight = FW_HEAVY;
+            break;
     }
 }
 
diff -bru wxWidgets.orig/src/osx/carbon/utilscocoa.mm wxWidgets/src/osx/carbon/utilscocoa.mm
--- wxWidgets.orig/src/osx/carbon/utilscocoa.mm	2015-05-17 00:10:34.000000000 -0400
+++ wxWidgets/src/osx/carbon/utilscocoa.mm	2015-06-11 22:23:10.000000000 -0400
@@ -134,12 +134,24 @@
 
         int size = (int) ([theFont pointSize]+0.5);
  
-        if ( theFontWeight >= 9 )
-            fontweight = wxFONTWEIGHT_BOLD ;
-        else if ( theFontWeight < 5 )
+        if( theFontWeight == 1 || theFontWeight == 2 )
+            fontweight = wxFONTWEIGHT_THIN;
+        if( theFontWeight == 3 )
+            fontweight = wxFONTWEIGHT_EXTRALIGHT;
+        if( theFontWeight == 4 )
             fontweight = wxFONTWEIGHT_LIGHT;
-        else
-            fontweight = wxFONTWEIGHT_NORMAL ;
+        if( theFontWeight == 5 )
+            fontweight = wxFONTWEIGHT_NORMAL;
+        if( theFontWeight == 6 )
+            fontweight = wxFONTWEIGHT_MEDIUM;
+        if( theFontWeight == 7 || theFontWeight == 8 )
+            fontweight = wxFONTWEIGHT_SEMIBOLD;
+        if( theFontWeight == 9 )
+            fontweight = wxFONTWEIGHT_BOLD;
+        if( theFontWeight == 10 )
+            fontweight = wxFONTWEIGHT_EXTRABOLD;
+        if( theFontWeight > 11 )
+            fontweight = wxFONTWEIGHT_HEAVY;
             
         if ( theTraits & NSItalicFontMask )
             fontstyle = wxFONTSTYLE_ITALIC ;
@@ -201,6 +213,31 @@
         traits |= NSBoldFontMask;
         weight = 9;
     }
+    else if( info->m_weight == wxFONTWEIGHT_THIN )
+    {
+        weight = 2;
+    }
+    else if( info->m_weight == wxFONTWEIGHT_EXTRALIGHT )
+    {
+        weight = 3;
+    }
+    else if( info->m_weight == wxFONTWEIGHT_MEDIUM )
+        weight = 6;
+    else if( info->m_weight == wxFONTWEIGHT_SEMIBOLD )
+    {
+        traits |= NSBoldFontMask;
+        weight = 8;
+    }
+    else if( info->m_weight == wxFONTWEIGHT_EXTRABOLD )
+    {
+        traits |= NSBoldFontMask;
+        weight = 10;
+    }
+    else if( info->m_weight == wxFONTWEIGHT_HEAVY )
+    {
+        traits |= NSBoldFontMask;
+        weight = 11;
+    }
     else if (info->m_weight == wxFONTWEIGHT_LIGHT)
         weight = 3;
 
