diff -ur xrc.old/xh_unkwn.cpp xrc/xh_unkwn.cpp
--- xrc.old/xh_unkwn.cpp	2007-01-26 00:33:11.000000000 -0200
+++ xrc/xh_unkwn.cpp	2007-01-26 00:34:00.000000000 -0200
@@ -68,8 +68,7 @@
 
     wxSizer *sizer = new wxBoxSizer(wxHORIZONTAL);
     sizer->Add((wxWindow*)child, 1, wxEXPAND);
-    SetSizer(sizer);
-    Layout();
+    SetSizerAndFit(sizer);
 }
 
 void wxUnknownControlContainer::RemoveChild(wxWindowBase *child)

 	  	 
