 samples/treelist/treelist.cpp | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/samples/treelist/treelist.cpp b/samples/treelist/treelist.cpp
index f3a1e07..4474c47 100644
--- a/samples/treelist/treelist.cpp
+++ b/samples/treelist/treelist.cpp
@@ -35,6 +35,7 @@
     #include "wx/textctrl.h"
 #endif
 
+#include "wx/button.h"
 #include "wx/treelist.h"
 
 #include "wx/aboutdlg.h"
@@ -330,6 +331,9 @@ MyFrame::MyFrame()
     // Construct the image list with the standard images.
     InitImageList();
 
+    wxButton* b = new wxButton();
+    b->Disable();
+    b->Create(this, wxID_ANY, "Should be disabled");
 
     // Create and layout child controls.
     m_treelist = CreateTreeListCtrl(wxTL_DEFAULT_STYLE);
@@ -340,6 +344,7 @@ MyFrame::MyFrame()
     m_oldLogTarget = wxLog::SetActiveTarget(new wxLogTextCtrl(textLog));
 
     wxSizer* sizer = new wxBoxSizer(wxVERTICAL);
+    sizer->Add(b);
     sizer->Add(m_treelist, wxSizerFlags(2).Expand());
     sizer->Add(textLog, wxSizerFlags(1).Expand());
     SetSizer(sizer);
@@ -384,7 +389,9 @@ void MyFrame::InitImageList()
 wxTreeListCtrl* MyFrame::CreateTreeListCtrl(long style)
 {
     wxTreeListCtrl* const
-        tree = new wxTreeListCtrl(this, wxID_ANY,
+        tree = new wxTreeListCtrl();
+    tree->Disable();
+    tree->Create(this, wxID_ANY,
                                   wxDefaultPosition, wxDefaultSize,
                                   style);
     tree->SetImageList(m_imageList);
