--- ../wxWidgets/src/common/uri.cpp	2005-03-01 13:36:52.000000000 -0800
+++ ./src/common/uri.cpp	2005-03-06 22:27:26.000000000 -0800
@@ -118,7 +118,7 @@
     wxASSERT_MSG(IsHex(*s) && IsHex(*(s+1)), wxT("Invalid escape!"));
 
     //<<4 == 16
-    return (wxChar)( CharToHex(*s) << 4 ) | CharToHex(*++s);
+    return (wxChar)( CharToHex(s[0]) << 4 ) | CharToHex(s[1]);
 }
 
 wxString wxURI::Unescape(const wxString& uri)

 	  	 
