Index: samples/dialogs/dialogs.cpp
===================================================================
--- samples/dialogs/dialogs.cpp	(revision 74315)
+++ samples/dialogs/dialogs.cpp	(working copy)
@@ -1848,7 +1848,7 @@
                                     wxDefaultPosition, wxDefaultSize,
                                     wxTE_MULTILINE);
         wxButton* btnShowText = new wxButton(this, wxID_ANY, "Show for &text");
-        wxButton* btnShowBtn = new wxButton(this, wxID_ANY, "Show for &button");
+        wxButton* btnShowBtn = new wxButton(this, 15555, "Show for &button");
 
         const wxString icons[] =
         {
@@ -1968,7 +1968,7 @@
 
     void OnShowTipForBtn(wxCommandEvent& WXUNUSED(event))
     {
-        DoShowTip(FindWindow(wxID_OK));
+        DoShowTip(FindWindow(15555));
     }
 
     void DoShowTip(wxWindow* win)
