diff --git a/interface/wx/mediactrl.h b/interface/wx/mediactrl.h
index b2e7042..29d3b29 100644
--- a/interface/wx/mediactrl.h
+++ b/interface/wx/mediactrl.h
@@ -5,6 +5,17 @@
 // Licence:     wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
+enum wxMediaState
+{
+    /** No media is being currently played. */
+    wxMEDIASTATE_STOPPED,
+    
+    /** Current media is paused. */
+    wxMEDIASTATE_PAUSED,
+    
+    /** There is media currently playing. */
+    wxMEDIASTATE_PLAYING
+};
 
 enum wxMediaCtrlPlayerControls
 {
