Index: include/wx/xrc/xmlres.h
===================================================================
--- include/wx/xrc/xmlres.h	(Revision 72577)
+++ include/wx/xrc/xmlres.h	(Arbeitskopie)
@@ -137,7 +137,7 @@
     // code for all controls used within the resource.
     void AddHandler(wxXmlResourceHandler *handler);
 
-    // Add a new handler at the begining of the handler list
+    // Add a new handler at the beginning of the handler list
     void InsertHandler(wxXmlResourceHandler *handler);
 
     // Removes all handlers
Index: include/wx/xrc/xh_unkwn.h
===================================================================
--- include/wx/xrc/xh_unkwn.h	(Revision 72577)
+++ include/wx/xrc/xh_unkwn.h	(Arbeitskopie)
@@ -1,6 +1,6 @@
 /////////////////////////////////////////////////////////////////////////////
 // Name:        wx/xrc/xh_unkwn.h
-// Purpose:     XML resource handler for unkown widget
+// Purpose:     XML resource handler for unknown widget
 // Author:      Vaclav Slavik
 // Created:     2000/03/05
 // RCS-ID:      $Id$
