? wxWidgets/buildgtk
Index: wxWidgets/contrib/include/wx/plot/plot.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/contrib/include/wx/plot/plot.h,v
retrieving revision 1.10
diff -b -u -2 -r1.10 plot.h
--- wxWidgets/contrib/include/wx/plot/plot.h	2005/05/04 15:22:02	1.10
+++ wxWidgets/contrib/include/wx/plot/plot.h	2005/07/21 12:17:10
@@ -129,13 +129,13 @@
     void SetStartY( double startY )
         { m_startY = startY; }
-    double GetStartY()
+    double GetStartY() const
         { return m_startY; }
     void SetEndY( double endY )
         { m_endY = endY; }
-    double GetEndY()
+    double GetEndY() const
         { return m_endY; }
     void SetOffsetY( int offsetY )
        { m_offsetY = offsetY; }
-    int GetOffsetY()
+    int GetOffsetY() const
        { return m_offsetY; }
 
@@ -145,4 +145,9 @@
        { m_penSelected = pen; }
 
+    const wxPen& GetPenNormal() const
+       { return m_penNormal; }
+    const wxPen& GetPenSelected() const
+       { return m_penSelected; }
+
 private:
     int     m_offsetY;
@@ -359,4 +364,7 @@
     void ResetScrollbar();
 
+    void AddChartTitle(const wxString&, wxFont = *wxNORMAL_FONT, 
+		       wxColour = *wxBLACK);
+
 private:
     friend class wxPlotArea;
@@ -377,4 +385,12 @@
     bool               m_scrollOnThumbRelease;
     bool               m_enlargeAroundWindowCentre;
+
+    wxString           m_title;
+    wxFont	       m_titleFont;
+    wxColour	       m_titleColour;
+    wxStaticText*      m_titleStaticText;
+    wxBoxSizer*        m_plotAndTitleSizer;
+
+    void DrawChartTitle();
 
     DECLARE_CLASS(wxPlotWindow)
Index: wxWidgets/contrib/samples/plot/plot.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/contrib/samples/plot/plot.cpp,v
retrieving revision 1.8
diff -b -u -2 -r1.8 plot.cpp
--- wxWidgets/contrib/samples/plot/plot.cpp	2005/05/04 15:22:06	1.8
+++ wxWidgets/contrib/samples/plot/plot.cpp	2005/07/21 12:17:10
@@ -130,4 +130,11 @@
 //    m_plot->SetScrollOnThumbRelease( true );
 
+    //Add a blue, 16pt chart title
+    wxString titleText( _T("The Chart Title") );
+    wxFont titleFont( *wxNORMAL_FONT );
+    titleFont.SetPointSize(16);
+    wxColour titleColour( *wxBLUE );
+    m_plot->AddChartTitle(titleText, titleFont, titleColour);
+
     m_plot->Add( new MyPlotCurve( 0,  -1.5, 1.5 ) );
     m_plot->Add( new MyPlotCurve( 50, -1.5, 1.5 ) );
Index: wxWidgets/contrib/src/plot/plot.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/contrib/src/plot/plot.cpp,v
retrieving revision 1.11
diff -b -u -2 -r1.11 plot.cpp
--- wxWidgets/contrib/src/plot/plot.cpp	2005/05/04 15:22:09	1.11
+++ wxWidgets/contrib/src/plot/plot.cpp	2005/07/21 12:17:12
@@ -113,4 +113,5 @@
 
 wxPlotCurve::wxPlotCurve( int offsetY, double startY, double endY )
+: m_penNormal(*wxGREY_PEN), m_penSelected(*wxBLACK_PEN)
 {
     m_offsetY = offsetY;
@@ -416,7 +417,7 @@
 
             if (curve == m_owner->GetCurrentCurve())
-                dc.SetPen( *wxBLACK_PEN );
+	        dc.SetPen( curve->GetPenSelected() );
             else
-                dc.SetPen( *wxGREY_PEN );
+	        dc.SetPen( curve->GetPenNormal() );
 
             DrawCurve( &dc, curve, update_x-1, update_x+update_width+2 );
@@ -710,5 +711,6 @@
 
 wxPlotWindow::wxPlotWindow( wxWindow *parent, wxWindowID id, const wxPoint &pos, const wxSize &size, int flag )
-        : wxScrolledWindow( parent, id, pos, size, flag, _T("plotcanvas") )
+        : wxScrolledWindow( parent, id, pos, size, flag, _T("plotcanvas") ),
+	  m_titleStaticText(0)
 {
     m_xUnitsPerValue = 1.0;
@@ -746,4 +748,9 @@
     wxBoxSizer *plotsizer = new wxBoxSizer( wxHORIZONTAL );
 
+    //Add sizer to hold the title and plot.
+    //Title to be added later.
+    m_plotAndTitleSizer = new wxBoxSizer( wxVERTICAL );
+    m_plotAndTitleSizer->Add( plotsizer, 1, wxEXPAND|wxTOP, 10 );
+
     if ((GetWindowStyleFlag() & wxPLOT_Y_AXIS) != 0)
     {
@@ -776,5 +783,5 @@
     }
 
-    mainsizer->Add( plotsizer, 1, wxEXPAND );
+    mainsizer->Add( m_plotAndTitleSizer, 1, wxEXPAND );
 
     SetAutoLayout( true );
@@ -982,8 +989,43 @@
 }
 
+void wxPlotWindow::AddChartTitle(const wxString& title, wxFont font, 
+				 wxColour colour)
+{
+    m_title = title;
+    m_titleFont = font;
+    m_titleColour = colour;	
+    DrawChartTitle();
+}
+
+void wxPlotWindow::DrawChartTitle()
+{
+    if(m_title.size() != 0)
+    {
+	//If it is already added, remove child and delete
+	if(m_titleStaticText)
+	{
+	    this->RemoveChild(m_titleStaticText);
+	    m_titleStaticText->Destroy();
+	}
+
+	//Create the text control and set the font, colour
+	m_titleStaticText = new wxStaticText(this, -1, m_title);
+	m_titleStaticText->SetFont(m_titleFont);
+	m_titleStaticText->SetForegroundColour(m_titleColour);
+
+	//Create a sizer for the title. Prepend it to the Plot + Title sizer.
+	wxBoxSizer* titleSizer = new wxBoxSizer(wxHORIZONTAL);
+	titleSizer->Add(m_titleStaticText, 0, wxALIGN_CENTER | wxALL, 10);
+	m_plotAndTitleSizer->Prepend(titleSizer, 0, wxALIGN_CENTER_HORIZONTAL);
+
+	//Finally, force layout
+	m_plotAndTitleSizer->Layout();
+    }
+}
+
 void wxPlotWindow::RedrawXAxis()
 {
     if (m_xaxis)
-        m_xaxis->Refresh( false );
+        m_xaxis->Refresh( true );
 }
 
@@ -1001,4 +1043,6 @@
         m_yaxis->Refresh( true );
     m_area->Refresh( true );
+
+    DrawChartTitle();
 }
 

 	  	 
