--- src/msw/utils.cpp.orig	2005-04-14 16:54:50.156250000 +0800
+++ src/msw/utils.cpp	2005-05-07 14:56:44.296875000 +0800
@@ -1381,5 +1381,5 @@
         case wxFONTENCODING_CP1256:         ret = 1256; break;
         case wxFONTENCODING_CP1257:         ret = 1257; break;
-        case wxFONTENCODING_EUC_JP:         ret = 51932; break;
+        case wxFONTENCODING_EUC_JP:         ret = 20932; break;
         case wxFONTENCODING_MACROMAN:       ret = 10000; break;
         case wxFONTENCODING_MACJAPANESE:    ret = 10001; break;

 	  	 
