Index: docs/doxygen/overviews/xrc_format.h
===================================================================
--- docs/doxygen/overviews/xrc_format.h	(revisione 70362)
+++ docs/doxygen/overviews/xrc_format.h	(copia locale)
@@ -607,7 +607,7 @@
 @endcode
 
 
-@subsubsection xrc_wxtogglebuttton wxBitmapToggleButton
+@subsubsection xrc_wxbitmaptogglebutton wxBitmapToggleButton
 
 @beginTable
 @hdr3col{property, type, description}
@@ -1594,7 +1594,7 @@
 No additional properties.
 
 
-@subsubsection xrc_wxtogglebuttton wxToggleButton
+@subsubsection xrc_wxtogglebutton wxToggleButton
 
 @beginTable
 @hdr3col{property, type, description}
