diff --git a/src/generic/aboutdlgg.cpp b/src/generic/aboutdlgg.cpp
index 9a81ffa..c5ff3a9 100644
--- a/src/generic/aboutdlgg.cpp
+++ b/src/generic/aboutdlgg.cpp
@@ -309,7 +309,8 @@ void wxGenericAboutBox(const wxAboutDialogInfo& info, wxWindow* parent)
 
 // currently wxAboutBox is implemented natively only under these platforms, for
 // the others we provide a generic fallback here
-#if !defined(__WXMSW__) && !defined(__WXMAC__) && !defined(__WXGTK26__)
+#if !defined(__WXMSW__) && !defined(__WXMAC__) && !defined(__WXGTK26__) \
+    || defined(__WXGTK__) && defined(__WXUNIVERSAL__)
 
 void wxAboutBox(const wxAboutDialogInfo& info, wxWindow* parent)
 {
