diff -bru wxWidgets/samples/listctrl/listtest.cpp /mnt/winxp/wxWidgets/samples/listctrl/listtest.cpp
--- wxWidgets/samples/listctrl/listtest.cpp	2014-08-11 13:00:54.000000000 -0700
+++ /mnt/winxp/wxWidgets/samples/listctrl/listtest.cpp	2014-08-29 02:29:20.859375000 -0700
@@ -338,6 +338,7 @@
 
 void MyFrame::OnAbout(wxCommandEvent& WXUNUSED(event))
 {
+	m_listCtrl->SetImageList( m_oldList, wxIMAGE_LIST_SMALL );
     wxMessageDialog dialog(this, wxT("List test sample\nJulian Smart (c) 1997"),
             wxT("About list test"));
 
@@ -510,6 +511,7 @@
 
 void MyFrame::InitWithReportItems()
 {
+    m_oldList = m_listCtrl->GetImageList( wxIMAGE_LIST_SMALL );
     m_listCtrl->SetImageList(m_imageListSmall, wxIMAGE_LIST_SMALL);
 
     // note that under MSW for SetColumnWidth() to work we need to create the
diff -bru wxWidgets/samples/listctrl/listtest.h /mnt/winxp/wxWidgets/samples/listctrl/listtest.h
--- wxWidgets/samples/listctrl/listtest.h	2014-08-11 13:00:54.000000000 -0700
+++ /mnt/winxp/wxWidgets/samples/listctrl/listtest.h	2014-08-29 02:12:09.640625000 -0700
@@ -157,6 +157,7 @@
 
     wxImageList *m_imageListNormal;
     wxImageList *m_imageListSmall;
+    wxImageList *m_oldList;
 
     wxPanel *m_panel;
     MyListCtrl *m_listCtrl;

