diff -bru wxWidgets.orig/samples/console/console.cpp wxWidgets/samples/console/console.cpp
--- wxWidgets.orig/samples/console/console.cpp	2014-07-10 02:00:57.000000000 -0700
+++ wxWidgets/samples/console/console.cpp	2014-07-10 16:38:41.000000000 -0700
@@ -30,6 +30,8 @@
 #endif
 
 #include <wx/app.h>
+#include <wx/protocol/ftp.h>
+#include <wx/wfstream.h>
 #include <wx/cmdline.h>
 
 // ============================================================================
@@ -56,7 +58,50 @@
         fprintf(stderr, "Failed to initialize the wxWidgets library, aborting.");
         return -1;
     }
-
+    wxFTP ftp;
+    if( !ftp.Connect( "ftp.wxwidgets.org" ) )
+    {
+        fprintf( stderr, "Couldn't connect." );
+        return -1;
+    }
+    bool res;
+    printf( "Changing the directory.\n\r" );
+    res = ftp.ChDir( "/pub/2.8.9" );
+    if( !res )
+    {
+        printf( "Can't change the directory" );
+        return -1;
+    }
+    ftp.SetBinary();
+    const char *filename = "wxWidgets-2.8.9.tar.bz2";
+    int size = ftp.GetFileSize( filename );
+    if( size == -1 )
+    {
+        fprintf( stderr, "Couldn't get the file size for \"%s\"", filename );
+        return -1;
+    }
+    wxInputStream *in = ftp.GetInputStream( filename );
+    if( !in )
+    {
+        fprintf( stderr, "Couldn't get the file" );
+        return -1;
+    }
+    else
+    {
+        printf( "Starting the download.\n\r" );
+        wxOutputStream *out = new wxFileOutputStream( "temp" );
+        out->Write( *in );
+        delete out;
+        delete in;
+    }
+    printf( "Transfer is done. Leaving the directory.\n\r" );
+    res = ftp.ChDir( "/pub" );
+    if( !res )
+    {
+        printf( "Can't change the directory" );
+        return -1;
+    }
+    ftp.Close();
     wxCmdLineParser parser(cmdLineDesc, argc, argv);
     switch ( parser.Parse() )
     {
