--- access.cpp.orig	2007-11-21 11:38:58.000000000 +0000
+++ access.cpp	2008-06-01 09:23:48.078125000 +0100
@@ -188,6 +188,9 @@
 public:
     wxIAccessible(wxAccessible *pAccessible);
 
+    // Called to indicate object should prepare to be deleted.
+    void Quiesce();
+
     DECLARE_IUNKNOWN_METHODS;
 
 // IAccessible
@@ -335,6 +338,7 @@
 
 private:
     wxAccessible *m_pAccessible;      // pointer to C++ class we belong to
+    bool m_bQuiescing;                // Object is to be deleted
 
     DECLARE_NO_COPY_CLASS(wxIAccessible)
 };
@@ -359,6 +363,15 @@
     wxASSERT( pAccessible != NULL );
 
     m_pAccessible = pAccessible;
+    m_bQuiescing = false;
+}
+
+// Called to indicate object should prepare to be deleted.
+
+void wxIAccessible::Quiesce()
+{
+    m_bQuiescing = true;
+    m_pAccessible = NULL;
 }
 
 // Retrieves the child element or child object at a given point on the screen.
@@ -367,7 +380,7 @@
 STDMETHODIMP wxIAccessible::accHitTest(long xLeft, long yLeft, VARIANT* pVarID)
 {
     wxLogTrace(wxT("access"), wxT("accHitTest"));
-    wxASSERT (m_pAccessible != NULL);
+    wxASSERT( ( m_pAccessible != NULL ) || ( m_bQuiescing == true ) );
     if (!m_pAccessible)
         return E_FAIL;
 
@@ -436,7 +449,7 @@
 STDMETHODIMP wxIAccessible::accLocation ( long* pxLeft, long* pyTop, long* pcxWidth, long* pcyHeight, VARIANT varID)
 {
     wxLogTrace(wxT("access"), wxT("accLocation"));
-    wxASSERT (m_pAccessible != NULL);
+    wxASSERT( ( m_pAccessible != NULL ) || ( m_bQuiescing == true ) );
     if (!m_pAccessible)
         return E_FAIL;
 
@@ -482,7 +495,7 @@
 
 STDMETHODIMP wxIAccessible::accNavigate ( long navDir, VARIANT varStart, VARIANT* pVarEnd)
 {
-    wxASSERT (m_pAccessible != NULL);
+    wxASSERT( ( m_pAccessible != NULL ) || ( m_bQuiescing == true ) );
     if (!m_pAccessible)
         return E_FAIL;
     wxLogTrace(wxT("access"), wxString(wxT("accNavigate for ")) + m_pAccessible->GetWindow()->GetClassInfo()->GetClassName());
@@ -639,7 +652,7 @@
 STDMETHODIMP wxIAccessible::get_accChild ( VARIANT varChildID, IDispatch** ppDispChild)
 {
     wxLogTrace(wxT("access"), wxT("get_accChild"));
-    wxASSERT (m_pAccessible != NULL);
+    wxASSERT( ( m_pAccessible != NULL ) || ( m_bQuiescing == true ) );
     if (!m_pAccessible)
         return E_FAIL;
 
@@ -713,7 +726,7 @@
 STDMETHODIMP wxIAccessible::get_accChildCount ( long* pCountChildren)
 {
     wxLogTrace(wxT("access"), wxT("get_accChildCount"));
-    wxASSERT (m_pAccessible != NULL);
+    wxASSERT( ( m_pAccessible != NULL ) || ( m_bQuiescing == true ) );
     if (!m_pAccessible)
         return E_FAIL;
 
@@ -757,7 +770,7 @@
 STDMETHODIMP wxIAccessible::get_accParent ( IDispatch** ppDispParent)
 {
     wxLogTrace(wxT("access"), wxT("get_accParent"));
-    wxASSERT (m_pAccessible != NULL);
+    wxASSERT( ( m_pAccessible != NULL ) || ( m_bQuiescing == true ) );
     if (!m_pAccessible)
         return E_FAIL;
 
@@ -821,7 +834,7 @@
 STDMETHODIMP wxIAccessible::accDoDefaultAction(VARIANT varID)
 {
     wxLogTrace(wxT("access"), wxT("accDoDefaultAction"));
-    wxASSERT (m_pAccessible != NULL);
+    wxASSERT( ( m_pAccessible != NULL ) || ( m_bQuiescing == true ) );
     if (!m_pAccessible)
         return E_FAIL;
 
@@ -866,7 +879,7 @@
 STDMETHODIMP wxIAccessible::get_accDefaultAction ( VARIANT varID, BSTR* pszDefaultAction)
 {
     wxLogTrace(wxT("access"), wxT("get_accDefaultAction"));
-    wxASSERT (m_pAccessible != NULL);
+    wxASSERT( ( m_pAccessible != NULL ) || ( m_bQuiescing == true ) );
     if (!m_pAccessible)
         return E_FAIL;
 
@@ -926,7 +939,7 @@
 STDMETHODIMP wxIAccessible::get_accDescription ( VARIANT varID, BSTR* pszDescription)
 {
     wxLogTrace(wxT("access"), wxT("get_accDescription"));
-    wxASSERT (m_pAccessible != NULL);
+    wxASSERT( ( m_pAccessible != NULL ) || ( m_bQuiescing == true ) );
     if (!m_pAccessible)
         return E_FAIL;
 
@@ -983,7 +996,7 @@
 STDMETHODIMP wxIAccessible::get_accHelp ( VARIANT varID, BSTR* pszHelp)
 {
     wxLogTrace(wxT("access"), wxT("get_accHelp"));
-    wxASSERT (m_pAccessible != NULL);
+    wxASSERT( ( m_pAccessible != NULL ) || ( m_bQuiescing == true ) );
     if (!m_pAccessible)
         return E_FAIL;
 
@@ -1043,7 +1056,7 @@
 STDMETHODIMP wxIAccessible::get_accHelpTopic ( BSTR* pszHelpFile, VARIANT varChild, long* pidTopic)
 {
     wxLogTrace(wxT("access"), wxT("get_accHelpTopic"));
-    wxASSERT (m_pAccessible != NULL);
+    wxASSERT( ( m_pAccessible != NULL ) || ( m_bQuiescing == true ) );
     if (!m_pAccessible)
         return E_FAIL;
 
@@ -1088,7 +1101,7 @@
     wxLogTrace(wxT("access"), wxT("get_accKeyboardShortcut"));
     *pszKeyboardShortcut = NULL;
 
-    wxASSERT (m_pAccessible != NULL);
+    wxASSERT( ( m_pAccessible != NULL ) || ( m_bQuiescing == true ) );
     if (!m_pAccessible)
         return E_FAIL;
 
@@ -1147,7 +1160,7 @@
     wxLogTrace(wxT("access"), wxT("get_accName"));
     *pszName = NULL;
 
-    wxASSERT (m_pAccessible != NULL);
+    wxASSERT( ( m_pAccessible != NULL ) || ( m_bQuiescing == true ) );
     if (!m_pAccessible)
         return E_FAIL;
 
@@ -1198,7 +1211,7 @@
 STDMETHODIMP wxIAccessible::get_accRole ( VARIANT varID, VARIANT* pVarRole)
 {
     wxLogTrace(wxT("access"), wxT("get_accRole"));
-    wxASSERT (m_pAccessible != NULL);
+    wxASSERT( ( m_pAccessible != NULL ) || ( m_bQuiescing == true ) );
     if (!m_pAccessible)
         return E_FAIL;
 
@@ -1258,7 +1271,7 @@
 STDMETHODIMP wxIAccessible::get_accState ( VARIANT varID, VARIANT* pVarState)
 {
     wxLogTrace(wxT("access"), wxT("get_accState"));
-    wxASSERT (m_pAccessible != NULL);
+    wxASSERT( ( m_pAccessible != NULL ) || ( m_bQuiescing == true ) );
     if (!m_pAccessible)
         return E_FAIL;
 
@@ -1309,7 +1322,7 @@
 STDMETHODIMP wxIAccessible::get_accValue ( VARIANT varID, BSTR* pszValue)
 {
     wxLogTrace(wxT("access"), wxT("get_accValue"));
-    wxASSERT (m_pAccessible != NULL);
+    wxASSERT( ( m_pAccessible != NULL ) || ( m_bQuiescing == true ) );
     if (!m_pAccessible)
         return E_FAIL;
 
@@ -1361,7 +1374,7 @@
 STDMETHODIMP wxIAccessible::accSelect ( long flagsSelect, VARIANT varID )
 {
     wxLogTrace(wxT("access"), wxT("get_accSelect"));
-    wxASSERT (m_pAccessible != NULL);
+    wxASSERT( ( m_pAccessible != NULL ) || ( m_bQuiescing == true ) );
     if (!m_pAccessible)
         return E_FAIL;
 
@@ -1408,7 +1421,7 @@
 STDMETHODIMP wxIAccessible::get_accFocus ( VARIANT* pVarID)
 {
     wxLogTrace(wxT("access"), wxT("get_accFocus"));
-    wxASSERT (m_pAccessible != NULL);
+    wxASSERT( ( m_pAccessible != NULL ) || ( m_bQuiescing == true ) );
     if (!m_pAccessible)
         return E_FAIL;
 
@@ -1475,7 +1488,7 @@
 {
 #if wxUSE_VARIANT
     wxLogTrace(wxT("access"), wxT("get_accSelection"));
-    wxASSERT (m_pAccessible != NULL);
+    wxASSERT( ( m_pAccessible != NULL ) || ( m_bQuiescing == true ) );
     if (!m_pAccessible)
         return E_FAIL;
 
@@ -1707,6 +1720,7 @@
 
 wxAccessible::~wxAccessible()
 {
+    m_pIAccessible->Quiesce();
     m_pIAccessible->Release();
     if (m_pIAccessibleStd)
         ((IAccessible*)m_pIAccessibleStd)->Release();
