Index: wxWindows/src/gtk/textctrl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/gtk/textctrl.cpp,v
retrieving revision 1.155
diff -b -u -2 -r1.155 textctrl.cpp
--- wxWindows/src/gtk/textctrl.cpp	2003/04/02 17:55:42	1.155
+++ wxWindows/src/gtk/textctrl.cpp	2003/05/10 05:24:04
@@ -1188,5 +1188,7 @@
     gint from, to;
 #ifdef __WXGTK20__
-    if ( !gtk_editable_get_selection_bounds(GTK_EDITABLE(m_text), &from, &to) )
+    GtkTextBuffer *buffer = gtk_text_view_get_buffer (GTK_TEXT_VIEW (m_text));
+    GtkTextIter ifrom, ito;
+    if (!gtk_text_buffer_get_selection_bounds(buffer, &ifrom, &ito))
 #else
     if ( !(GTK_EDITABLE(m_text)->has_selection) )
@@ -1198,5 +1200,8 @@
     else // got selection
     {
-#ifndef __WXGTK20__
+#ifdef __WXGTK20__
+        from = gtk_text_iter_get_offset(&ifrom);
+        to = gtk_text_iter_get_offset(&ito);
+#else
         from = (long) GTK_EDITABLE(m_text)->selection_start_pos;
         to = (long) GTK_EDITABLE(m_text)->selection_end_pos;

 	  	 
