Index: wxWindows/src/generic/grid.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/generic/grid.cpp,v
retrieving revision 1.234
diff -b -u -2 -r1.234 grid.cpp
--- wxWindows/src/generic/grid.cpp	2002/12/09 11:14:04	1.234
+++ wxWindows/src/generic/grid.cpp	2002/12/14 00:51:07
@@ -971,5 +971,5 @@
   {
     int value = Spin()->GetValue();
-    s.Printf(wxT("%ld"), value);
+    s.Printf(wxT("%d"), value);
   }
   else
@@ -2169,15 +2169,15 @@
 
 // GetRenderer and GetEditor use a slightly different decision path about
-// which attribute to use.  If a non-default attr object has one then it is
+// which attribute to use.  If an attr object has one then it is
 // used, otherwise the default editor or renderer is fetched from the grid and
 // used.  It should be the default for the data type of the cell.  If it is
 // NULL (because the table has a type that the grid does not have in its
-// registry,) then the grid's default editor or renderer is used.
+// registry,) then a wxGridCellStringRenderer is used.
 
 wxGridCellRenderer* wxGridCellAttr::GetRenderer(wxGrid* grid, int row, int col) const
 {
-    wxGridCellRenderer *renderer;
+    wxGridCellRenderer *renderer = NULL;
 
-    if ( m_renderer && this != m_defGridAttr )
+    if ( m_renderer )
     {
         // use the cells renderer if it has one
@@ -2187,30 +2187,15 @@
     else // no non default cell renderer
     {
-        // get default renderer for the data type
-        if ( grid )
-        {
-            // GetDefaultRendererForCell() will do IncRef() for us
-            renderer = grid->GetDefaultRendererForCell(row, col);
-        }
-        else
-        {
-            renderer = NULL;
-        }
-
-        if ( !renderer )
-        {
-            if (m_defGridAttr &&  this != m_defGridAttr )
-            {
                 // if we still don't have one then use the grid default
-                // (no need for IncRef() here neither)
-                renderer = m_defGridAttr->GetRenderer(NULL, 0, 0);
-            }
-            else // default grid attr
+        if ( m_defGridAttr && m_defGridAttr->m_renderer )
             {
-                // use m_renderer which we had decided not to use initially
-                renderer = m_renderer;
-                if ( renderer )
+            renderer = m_defGridAttr->m_renderer;
                     renderer->IncRef();
             }
+        // get default renderer for the data type
+        if ( grid && !renderer )
+        {
+            // GetDefaultRendererForCell() will do IncRef() for us
+            renderer = grid->GetDefaultRendererForCell(row, col);
         }
     }
@@ -2225,7 +2210,7 @@
 wxGridCellEditor* wxGridCellAttr::GetEditor(wxGrid* grid, int row, int col) const
 {
-    wxGridCellEditor *editor;
+    wxGridCellEditor *editor = NULL;
 
-    if ( m_editor && this != m_defGridAttr )
+    if ( m_editor )
     {
         // use the cells editor if it has one
@@ -2235,30 +2220,15 @@
     else // no non default cell editor
     {
-        // get default editor for the data type
-        if ( grid )
-        {
-            // GetDefaultEditorForCell() will do IncRef() for us
-            editor = grid->GetDefaultEditorForCell(row, col);
-        }
-        else
-        {
-            editor = NULL;
-        }
-
-        if ( !editor )
-        {
-            if ( m_defGridAttr && this != m_defGridAttr )
-            {
                 // if we still don't have one then use the grid default
-                // (no need for IncRef() here neither)
-                editor = m_defGridAttr->GetEditor(NULL, 0, 0);
-            }
-            else // default grid attr
+        if ( m_defGridAttr && m_defGridAttr->m_editor )
             {
-                // use m_editor which we had decided not to use initially
-                editor = m_editor;
-                if ( editor )
+            editor = m_defGridAttr->m_editor;
                     editor->IncRef();
             }
+        // get default editor for the data type
+        if ( grid && !editor)
+        {
+            // GetDefaultEditorForCell() will do IncRef() for us
+            editor = grid->GetDefaultEditorForCell(row, col);
         }
     }
@@ -3761,6 +3731,4 @@
     m_defaultCellAttr->SetBackgroundColour(
         wxSystemSettings::GetColour(wxSYS_COLOUR_WINDOW));
-    m_defaultCellAttr->SetRenderer(new wxGridCellStringRenderer);
-    m_defaultCellAttr->SetEditor(new wxGridCellTextEditor);
 
 

 	  	 
