Index: except.h
===================================================================
--- except.h	(revision 64969)
+++ except.h	(working copy)
@@ -24,8 +24,10 @@
 // compilation errors in e.g. wx/scopeguard.h with at least g++ 4
 #if !wxUSE_EXCEPTIONS || \
         (defined(__GNUG__) && !defined(__EXCEPTIONS))
+#ifndef wxNO_EXCEPTIONS
     #define wxNO_EXCEPTIONS
 #endif
+#endif
 
 #ifdef wxNO_EXCEPTIONS
     #define wxTRY
