Index: src/common/docview.cpp
===================================================================
--- src/common/docview.cpp	(revision 56297)
+++ src/common/docview.cpp	(working copy)
@@ -480,7 +480,7 @@
                  (
                     wxString::Format
                     (
-                     _("Do you want to save changes to document %s?"),
+                     _("Do you want to save changes to %s?"),
                      GetUserReadableName()
                     ),
                     wxTheApp->GetAppDisplayName(),
