Index: src/common/docview.cpp
===================================================================
--- src/common/docview.cpp	(revision 64441)
+++ src/common/docview.cpp	(working copy)
@@ -901,7 +901,6 @@
 
     EVT_UPDATE_UI(wxID_PRINT, wxDocManager::OnUpdateDisableIfNoDoc)
     EVT_UPDATE_UI(wxID_PREVIEW, wxDocManager::OnUpdateDisableIfNoDoc)
-    EVT_UPDATE_UI(wxID_PRINT_SETUP, wxDocManager::OnUpdateDisableIfNoDoc)
 #endif // wxUSE_PRINTING_ARCHITECTURE
 END_EVENT_TABLE()
 
