Index: src/common/fontcmn.cpp
===================================================================
--- src/common/fontcmn.cpp	(revision 47692)
+++ src/common/fontcmn.cpp	(working copy)
@@ -799,3 +799,26 @@
 }
 
 #endif // generic or wxMSW or wxOS2
+
+
+// wxFont <-> wxString utilities, used by wxConfig
+wxString wxToString(const wxFontBase& font)
+{
+    return font.IsOk() ? font.GetNativeFontInfoDesc()
+                       : wxString();
+}
+
+bool wxFromString(const wxString& str, wxFontBase *font)
+{
+    wxCHECK_MSG( font, false, _T("NULL output parameter") );
+
+    if ( str.empty() )
+    {
+        *font = wxNullFont;
+        return true;
+    }
+
+    return font->SetNativeFontInfo(str);
+}
+
+
Index: docs/latex/wx/manual.tex
===================================================================
--- docs/latex/wx/manual.tex	(revision 47692)
+++ docs/latex/wx/manual.tex	(working copy)
@@ -32,7 +32,7 @@
 }}
 \winhelpignore{\author{Julian Smart, Robert Roebling, Vadim Zeitlin,
 Robin Dunn, et al}
-\date{January, 2007}
+\date{July, 2007}
 }
 \makeindex
 \begin{document}
Index: docs/latex/wx/function.tex
===================================================================
--- docs/latex/wx/function.tex	(revision 47692)
+++ docs/latex/wx/function.tex	(working copy)
@@ -3266,6 +3266,9 @@
 \func{bool}{wxFromString}{\param{const wxString\& }{str},
                            \param{wxColourBase* }{col}}
 
+\func{bool}{wxFromString}{\param{const wxString\& }{str},
+                           \param{wxFontBase* }{col}}
+
 Converts string to the type of the second argument. Returns \true on success.
 See also: \helpref{wxToString}{wxtostring}.
 
@@ -3589,6 +3592,8 @@
 
 \func{wxString}{wxToString}{\param{const wxColourBase\& }{col}}
 
+\func{wxString}{wxToString}{\param{const wxFontBase\& }{col}}
+
 Converts its argument to string.
 See also: \helpref{wxFromString}{wxfromstring}.
 
Index: include/wx/font.h
===================================================================
--- include/wx/font.h	(revision 47692)
+++ include/wx/font.h	(working copy)
@@ -222,6 +222,12 @@
     static wxFontEncoding ms_encodingDefault;
 };
 
+// wxFontBase <-> wxString utilities, used by wxConfig
+WXDLLIMPEXP_CORE wxString wxToString(const wxFontBase& font);
+WXDLLIMPEXP_CORE bool wxFromString(const wxString& str, wxFontBase* font);
+
+
+
 // include the real class declaration
 #if defined(__WXPALMOS__)
     #include "wx/palmos/font.h"

 	  	 
