--- ../wxWidgets/include/wx/filefn.h	2011-10-11 20:48:50.296000253 +0200
+++ filefn-patched.h	2011-10-11 21:26:35.836000253 +0200
@@ -365,8 +365,28 @@
     // first the ANSI versions
     #define   wxCRT_OpenA       wxPOSIX_IDENT(open)
     #define   wxCRT_AccessA     wxPOSIX_IDENT(access)
-    #define   wxCRT_MkDirA      wxPOSIX_IDENT(mkdir)
-    #define   wxCRT_RmDirA      wxPOSIX_IDENT(rmdir)
+    
+    #if defined(__MINGW32__) || defined(__MINGW64__)
+        #include <_mingw.h>
+        #ifdef __MINGW64_VERSION_MAJOR
+        // The C interface to msvcrt.dll of mingw-w64 
+        // defines the mkdir() and rmdir() functions.
+            #define   wxCRT_MkDirA      mkdir
+            #define   wxCRT_RmDirA      rmdir
+        #else
+        //  MinGW defines both mkdir() and _mkdir(),
+        //  and rmdir() as well as _rmdir(). The versions
+        //  without leading underscore are deprecated by
+        //  MS as of Visual C++ 2005. Using the preferred
+        //  versions:
+            #define   wxCRT_MkDirA      wxPOSIX_IDENT(mkdir)
+            #define   wxCRT_RmDirA      wxPOSIX_IDENT(rmdir)
+        #endif
+    #else
+        #define   wxCRT_MkDirA      wxPOSIX_IDENT(mkdir)
+        #define   wxCRT_RmDirA      wxPOSIX_IDENT(rmdir)
+    #endif
+
     #ifdef wxHAS_HUGE_FILES
         #define   wxCRT_StatA       wxPOSIX_IDENT(stati64)
     #else
