diff --git a/include/wx/stc/stc.h b/include/wx/stc/stc.h
index e9154eb..909f60a 100644
--- a/include/wx/stc/stc.h
+++ b/include/wx/stc/stc.h
@@ -5543,7 +5543,8 @@ wxDECLARE_EXPORTED_EVENT( WXDLLIMPEXP_STC, wxEVT_STC_CHARADDED, wxStyledTextEven
 wxDECLARE_EXPORTED_EVENT( WXDLLIMPEXP_STC, wxEVT_STC_SAVEPOINTREACHED, wxStyledTextEvent );
 wxDECLARE_EXPORTED_EVENT( WXDLLIMPEXP_STC, wxEVT_STC_SAVEPOINTLEFT, wxStyledTextEvent );
 wxDECLARE_EXPORTED_EVENT( WXDLLIMPEXP_STC, wxEVT_STC_ROMODIFYATTEMPT, wxStyledTextEvent );
-wxDECLARE_EXPORTED_EVENT( WXDLLIMPEXP_STC, wxEVT_STC_KEY, wxStyledTextEvent );
+wxDEPRECATED_MSG( "Don't handle wxEVT_STC_KEY. It's never generated." ) \
+    wxDECLARE_EXPORTED_EVENT( WXDLLIMPEXP_STC, wxEVT_STC_KEY, wxStyledTextEvent );
 wxDECLARE_EXPORTED_EVENT( WXDLLIMPEXP_STC, wxEVT_STC_DOUBLECLICK, wxStyledTextEvent );
 wxDECLARE_EXPORTED_EVENT( WXDLLIMPEXP_STC, wxEVT_STC_UPDATEUI, wxStyledTextEvent );
 wxDECLARE_EXPORTED_EVENT( WXDLLIMPEXP_STC, wxEVT_STC_MODIFIED, wxStyledTextEvent );
@@ -5552,7 +5553,8 @@ wxDECLARE_EXPORTED_EVENT( WXDLLIMPEXP_STC, wxEVT_STC_MARGINCLICK, wxStyledTextEv
 wxDECLARE_EXPORTED_EVENT( WXDLLIMPEXP_STC, wxEVT_STC_NEEDSHOWN, wxStyledTextEvent );
 wxDECLARE_EXPORTED_EVENT( WXDLLIMPEXP_STC, wxEVT_STC_PAINTED, wxStyledTextEvent );
 wxDECLARE_EXPORTED_EVENT( WXDLLIMPEXP_STC, wxEVT_STC_USERLISTSELECTION, wxStyledTextEvent );
-wxDECLARE_EXPORTED_EVENT( WXDLLIMPEXP_STC, wxEVT_STC_URIDROPPED, wxStyledTextEvent );
+wxDEPRECATED_MSG( "Don't handle wxEVT_STC_URIDROPPED. It's never generated." ) \
+    wxDECLARE_EXPORTED_EVENT( WXDLLIMPEXP_STC, wxEVT_STC_URIDROPPED, wxStyledTextEvent );
 wxDECLARE_EXPORTED_EVENT( WXDLLIMPEXP_STC, wxEVT_STC_DWELLSTART, wxStyledTextEvent );
 wxDECLARE_EXPORTED_EVENT( WXDLLIMPEXP_STC, wxEVT_STC_DWELLEND, wxStyledTextEvent );
 wxDECLARE_EXPORTED_EVENT( WXDLLIMPEXP_STC, wxEVT_STC_START_DRAG, wxStyledTextEvent );
diff --git a/interface/wx/stc/stc.h b/interface/wx/stc/stc.h
index ebe222c..5f44d96 100644
--- a/interface/wx/stc/stc.h
+++ b/interface/wx/stc/stc.h
@@ -2780,8 +2780,6 @@
         TOWRITE
     @event{EVT_STC_ROMODIFYATTEMPT(id, fn)}
         TOWRITE
-    @event{EVT_STC_KEY(id, fn)}
-        TOWRITE
     @event{EVT_STC_DOUBLECLICK(id, fn)}
         TOWRITE
     @event{EVT_STC_UPDATEUI(id, fn)}
@@ -2798,8 +2796,6 @@
         TOWRITE
     @event{EVT_STC_USERLISTSELECTION(id, fn)}
         TOWRITE
-    @event{EVT_STC_URIDROPPED(id, fn)}
-        TOWRITE
     @event{EVT_STC_DWELLSTART(id, fn)}
         TOWRITE
     @event{EVT_STC_DWELLEND(id, fn)}
@@ -2847,6 +2843,16 @@
         @since 3.1.1
 
     @endEventTable
+    
+    Most of the scintilla notifications are mapped to a similarly named 
+    wxStyledTextEvent.  However a few of the notifications would only offer 
+    information available in other wxWidgets event types, and in those cases a 
+    corresponding wxStyledTextEvent is not defined. Currently, the 
+    @c wxEVT_KEY_DOWN event is used instead of the SCN_KEY notification.  The 
+    regular wxWidgets drag and drop functionality can be used instead of the 
+    SCN_URIDROPPED notification. The @c wxEVT_SET_FOCUS event is used instead 
+    of the SCN_FOCUSIN notification.  And the @c wxEVT_KILL_FOCUS event is used 
+    instead of the SCN_FOCUSOUT notification.
 
     @library{wxstc}
     @category{stc}
diff --git a/samples/stc/edit.cpp b/samples/stc/edit.cpp
index 0b3e198..05aeba9 100644
--- a/samples/stc/edit.cpp
+++ b/samples/stc/edit.cpp
@@ -114,7 +114,6 @@ wxBEGIN_EVENT_TABLE (Edit, wxStyledTextCtrl)
     // stc
     EVT_STC_MARGINCLICK (wxID_ANY,     Edit::OnMarginClick)
     EVT_STC_CHARADDED (wxID_ANY,       Edit::OnCharAdded)
-    EVT_STC_KEY( wxID_ANY , Edit::OnKey )
 
     EVT_KEY_DOWN( Edit::OnKeyDown )
 wxEND_EVENT_TABLE()
@@ -210,11 +209,6 @@ void Edit::OnEditClear (wxCommandEvent &WXUNUSED(event)) {
     Clear ();
 }
 
-void Edit::OnKey (wxStyledTextEvent &WXUNUSED(event))
-{
-    wxMessageBox("OnKey");
-}
-
 void Edit::OnKeyDown (wxKeyEvent &event)
 {
     if (CallTipActive())
diff --git a/samples/stc/edit.h b/samples/stc/edit.h
index 1349688..5613acc 100644
--- a/samples/stc/edit.h
+++ b/samples/stc/edit.h
@@ -104,7 +104,6 @@ public:
     // stc
     void OnMarginClick (wxStyledTextEvent &event);
     void OnCharAdded  (wxStyledTextEvent &event);
-    void OnKey  (wxStyledTextEvent &event);
 
     void OnKeyDown(wxKeyEvent &event);
 
diff --git a/src/stc/stc.cpp b/src/stc/stc.cpp
index 197c5f3..16b84a7 100644
--- a/src/stc/stc.cpp
+++ b/src/stc/stc.cpp
@@ -5476,10 +5476,6 @@ void wxStyledTextCtrl::NotifyParent(SCNotification* _scn) {
         evt.SetEventType(wxEVT_STC_ROMODIFYATTEMPT);
         break;
 
-    case SCN_KEY:
-        evt.SetEventType(wxEVT_STC_KEY);
-        break;
-
     case SCN_DOUBLECLICK:
         evt.SetEventType(wxEVT_STC_DOUBLECLICK);
         evt.SetLine(scn.line);
@@ -5532,11 +5528,6 @@ void wxStyledTextCtrl::NotifyParent(SCNotification* _scn) {
         evt.SetListCompletionMethod(scn.listCompletionMethod);
         break;
 
-    case SCN_URIDROPPED:
-        evt.SetEventType(wxEVT_STC_URIDROPPED);
-        SetEventText(evt, scn.text, strlen(scn.text));
-        break;
-
     case SCN_DWELLSTART:
         evt.SetEventType(wxEVT_STC_DWELLSTART);
         evt.SetX(scn.x);
diff --git a/src/stc/stc.cpp.in b/src/stc/stc.cpp.in
index a3de94c..d1413d1 100644
--- a/src/stc/stc.cpp.in
+++ b/src/stc/stc.cpp.in
@@ -1048,10 +1048,6 @@ void wxStyledTextCtrl::NotifyParent(SCNotification* _scn) {
         evt.SetEventType(wxEVT_STC_ROMODIFYATTEMPT);
         break;
 
-    case SCN_KEY:
-        evt.SetEventType(wxEVT_STC_KEY);
-        break;
-
     case SCN_DOUBLECLICK:
         evt.SetEventType(wxEVT_STC_DOUBLECLICK);
         evt.SetLine(scn.line);
@@ -1104,11 +1100,6 @@ void wxStyledTextCtrl::NotifyParent(SCNotification* _scn) {
         evt.SetListCompletionMethod(scn.listCompletionMethod);
         break;
 
-    case SCN_URIDROPPED:
-        evt.SetEventType(wxEVT_STC_URIDROPPED);
-        SetEventText(evt, scn.text, strlen(scn.text));
-        break;
-
     case SCN_DWELLSTART:
         evt.SetEventType(wxEVT_STC_DWELLSTART);
         evt.SetX(scn.x);
diff --git a/src/stc/stc.h.in b/src/stc/stc.h.in
index c9bb095..6e4d4c3 100644
--- a/src/stc/stc.h.in
+++ b/src/stc/stc.h.in
@@ -665,7 +665,8 @@ wxDECLARE_EXPORTED_EVENT( WXDLLIMPEXP_STC, wxEVT_STC_CHARADDED, wxStyledTextEven
 wxDECLARE_EXPORTED_EVENT( WXDLLIMPEXP_STC, wxEVT_STC_SAVEPOINTREACHED, wxStyledTextEvent );
 wxDECLARE_EXPORTED_EVENT( WXDLLIMPEXP_STC, wxEVT_STC_SAVEPOINTLEFT, wxStyledTextEvent );
 wxDECLARE_EXPORTED_EVENT( WXDLLIMPEXP_STC, wxEVT_STC_ROMODIFYATTEMPT, wxStyledTextEvent );
-wxDECLARE_EXPORTED_EVENT( WXDLLIMPEXP_STC, wxEVT_STC_KEY, wxStyledTextEvent );
+wxDEPRECATED_MSG( "Don't handle wxEVT_STC_KEY. It's never generated." ) \
+    wxDECLARE_EXPORTED_EVENT( WXDLLIMPEXP_STC, wxEVT_STC_KEY, wxStyledTextEvent );
 wxDECLARE_EXPORTED_EVENT( WXDLLIMPEXP_STC, wxEVT_STC_DOUBLECLICK, wxStyledTextEvent );
 wxDECLARE_EXPORTED_EVENT( WXDLLIMPEXP_STC, wxEVT_STC_UPDATEUI, wxStyledTextEvent );
 wxDECLARE_EXPORTED_EVENT( WXDLLIMPEXP_STC, wxEVT_STC_MODIFIED, wxStyledTextEvent );
@@ -674,7 +675,8 @@ wxDECLARE_EXPORTED_EVENT( WXDLLIMPEXP_STC, wxEVT_STC_MARGINCLICK, wxStyledTextEv
 wxDECLARE_EXPORTED_EVENT( WXDLLIMPEXP_STC, wxEVT_STC_NEEDSHOWN, wxStyledTextEvent );
 wxDECLARE_EXPORTED_EVENT( WXDLLIMPEXP_STC, wxEVT_STC_PAINTED, wxStyledTextEvent );
 wxDECLARE_EXPORTED_EVENT( WXDLLIMPEXP_STC, wxEVT_STC_USERLISTSELECTION, wxStyledTextEvent );
-wxDECLARE_EXPORTED_EVENT( WXDLLIMPEXP_STC, wxEVT_STC_URIDROPPED, wxStyledTextEvent );
+wxDEPRECATED_MSG( "Don't handle wxEVT_STC_URIDROPPED. It's never generated." ) \
+    wxDECLARE_EXPORTED_EVENT( WXDLLIMPEXP_STC, wxEVT_STC_URIDROPPED, wxStyledTextEvent );
 wxDECLARE_EXPORTED_EVENT( WXDLLIMPEXP_STC, wxEVT_STC_DWELLSTART, wxStyledTextEvent );
 wxDECLARE_EXPORTED_EVENT( WXDLLIMPEXP_STC, wxEVT_STC_DWELLEND, wxStyledTextEvent );
 wxDECLARE_EXPORTED_EVENT( WXDLLIMPEXP_STC, wxEVT_STC_START_DRAG, wxStyledTextEvent );
diff --git a/src/stc/stc.interface.h.in b/src/stc/stc.interface.h.in
index c82c4a3..dcd44fd 100644
--- a/src/stc/stc.interface.h.in
+++ b/src/stc/stc.interface.h.in
@@ -58,8 +58,6 @@
         TOWRITE
     @event{EVT_STC_ROMODIFYATTEMPT(id, fn)}
         TOWRITE
-    @event{EVT_STC_KEY(id, fn)}
-        TOWRITE
     @event{EVT_STC_DOUBLECLICK(id, fn)}
         TOWRITE
     @event{EVT_STC_UPDATEUI(id, fn)}
@@ -76,8 +74,6 @@
         TOWRITE
     @event{EVT_STC_USERLISTSELECTION(id, fn)}
         TOWRITE
-    @event{EVT_STC_URIDROPPED(id, fn)}
-        TOWRITE
     @event{EVT_STC_DWELLSTART(id, fn)}
         TOWRITE
     @event{EVT_STC_DWELLEND(id, fn)}
@@ -125,6 +121,16 @@
         @since 3.1.1
 
     @endEventTable
+    
+    Most of the scintilla notifications are mapped to a similarly named 
+    wxStyledTextEvent.  However a few of the notifications would only offer 
+    information available in other wxWidgets event types, and in those cases a 
+    corresponding wxStyledTextEvent is not defined. Currently, the 
+    @c wxEVT_KEY_DOWN event is used instead of the SCN_KEY notification.  The 
+    regular wxWidgets drag and drop functionality can be used instead of the 
+    SCN_URIDROPPED notification. The @c wxEVT_SET_FOCUS event is used instead 
+    of the SCN_FOCUSIN notification.  And the @c wxEVT_KILL_FOCUS event is used 
+    instead of the SCN_FOCUSOUT notification.
 
     @library{wxstc}
     @category{stc}
