Index: include/wx/msw/stattext.h
===================================================================
--- include/wx/msw/stattext.h	(revision 275)
+++ include/wx/msw/stattext.h	(working copy)
@@ -47,6 +47,7 @@
     virtual void DoSetSize(int x, int y, int w, int h,
                            int sizeFlags = wxSIZE_AUTO);
     virtual wxSize DoGetBestClientSize() const;
+    virtual wxSize DoGetBestSize() const;
 
     virtual wxString DoGetLabel() const;
     virtual void DoSetLabel(const wxString& str);
Index: src/msw/stattext.cpp
===================================================================
--- src/msw/stattext.cpp	(revision 275)
+++ src/msw/stattext.cpp	(working copy)
@@ -162,6 +162,11 @@
     return wxSize(widthTextMax, heightTextTotal);
 }
 
+wxSize wxStaticText::DoGetBestSize() const
+{
+    return GetBestSize();
+}
+
 void wxStaticText::DoSetSize(int x, int y, int w, int h, int sizeFlags)
 {
     // note: we first need to set the size and _then_ call UpdateLabel
