Index: src/common/image.cpp
===================================================================
--- src/common/image.cpp	(revision 66686)
+++ src/common/image.cpp	(working copy)
@@ -458,6 +458,12 @@
             else if ( quality == wxIMAGE_QUALITY_BICUBIC )
                 image = ResampleBicubic(width, height);
             break;
+        case wxIMAGE_QUALITY_BILINEAR_ALWAYS:
+            image = ResampleBilinear(width, height);
+            break;
+        case wxIMAGE_QUALITY_BICUBIC_ALWAYS:
+            image = ResampleBicubic(width,height);
+            break;
 
         case wxIMAGE_QUALITY_NEAREST:
             if ( old_width % width == 0 && old_width >= width &&
Index: include/wx/image.h
===================================================================
--- include/wx/image.h	(revision 66686)
+++ include/wx/image.h	(working copy)
@@ -62,6 +62,8 @@
     wxIMAGE_QUALITY_NEAREST = 0,
     wxIMAGE_QUALITY_BILINEAR = 1,
     wxIMAGE_QUALITY_BICUBIC = 2,
+	wxIMAGE_QUALITY_BILINEAR_ALWAYS = 3,
+	wxIMAGE_QUALITY_BICUBIC_ALWAYS = 4,
 
     // default quality is low (but fast)
     wxIMAGE_QUALITY_NORMAL = wxIMAGE_QUALITY_NEAREST,
