Index: wxWidgets-2.8/src/mac/carbon/font.cpp
===================================================================
--- wxWidgets-2.8/src/mac/carbon/font.cpp	(revision 47863)
+++ wxWidgets-2.8/src/mac/carbon/font.cpp	(working copy)
@@ -330,7 +330,12 @@
 
     status = ::ATSUSetAttributes( (ATSUStyle)m_macATSUStyle, attributeCount, atsuTags, atsuSizes, atsuValues);
 
-    wxASSERT_MSG( status == noErr , wxT("couldn't modify ATSU style") );
+    if (status != noErr)
+    {
+        wxString msg;
+        msg.Printf(wxT("couldn't modify ATSU style (status: %d%s)"), (int) status, (m_pointSize <= 0) ? wxT(", point size <= 0!") : wxT(""));
+        wxASSERT_MSG( status == noErr , msg.c_str() );
+    }
 #else
     if ( m_macThemeFontID != kThemeCurrentPortFont )
     {
@@ -484,7 +489,12 @@
         sizeof(atsuTags) / sizeof(ATSUAttributeTag) ,
         atsuTags, atsuSizes, atsuValues);
 
-    wxASSERT_MSG( status == noErr , wxT("couldn't modify ATSU style") );
+    if (status != noErr)
+    {
+        wxString msg;
+        msg.Printf(wxT("couldn't modify ATSU style (status: %d%s)"), (int) status, (m_pointSize <= 0) ? wxT(", point size <= 0!") : wxT(""));
+        wxASSERT_MSG( status == noErr , msg.c_str() );
+    }
 #endif
 }
 

 	  	 
