Index: src/generic/grid.cpp
===================================================================
--- src/generic/grid.cpp	(r‚vision 64459)
+++ src/generic/grid.cpp	(copie de travail)
@@ -8,7 +8,6 @@
 // Copyright:   (c) Michael Bedward (mbedward@ozemail.com.au)
 // Licence:     wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
-
 /*
     TODO:
 
@@ -6261,7 +6260,9 @@
         else if ( line > 0 &&
                     pos - oper.GetLineStartPos(this,
                                                line) < WXGRID_LABEL_EDGE_ZONE )
-            return line - 1;
+		{
+			return oper.GetPreviousLineAt(this, line);
+		}
     }
 
     return -1;
Index: include/wx/generic/private/grid.h
===================================================================
--- include/wx/generic/private/grid.h	(r‚vision 64459)
+++ include/wx/generic/private/grid.h	(copie de travail)
@@ -517,7 +517,12 @@
     //     implemented for the lines
     virtual int GetLineAt(const wxGrid *grid, int line) const = 0;
 
+	// Return the index of the line just before the given position
+	virtual int GetPreviousLineAt(const wxGrid* grid, int line) const = 0;
 
+	// Return the index of the line just after the given position
+	virtual int GetNextLineAt(const wxGrid* grid, int line) const = 0;
+
     // Get the row or column label window
     virtual wxWindow *GetHeaderWindow(wxGrid *grid) const = 0;
 
@@ -585,6 +590,12 @@
     virtual int GetLineAt(const wxGrid * WXUNUSED(grid), int line) const
         { return line; } // TODO: implement row reordering
 
+	virtual int GetPreviousLineAt(const wxGrid* WXUNUSED(grid), int line) const 
+		{ return (line > 0 ? line - 1 : line); }
+
+	virtual int GetNextLineAt(const wxGrid* grid, int line) const
+		{ return (line < (grid->m_numRows - 1)? line + 1 : line); }
+
     virtual wxWindow *GetHeaderWindow(wxGrid *grid) const
         { return grid->GetGridRowLabelWindow(); }
     virtual int GetHeaderWindowSize(wxGrid *grid) const
@@ -645,7 +656,13 @@
 
     virtual int GetLineAt(const wxGrid *grid, int line) const
         { return grid->GetColAt(line); }
+	
+	virtual int GetPreviousLineAt(const wxGrid* grid, int line) const 
+		{ return grid->GetPreviousColAt(line); }
 
+	virtual int GetNextLineAt(const wxGrid* grid, int line) const
+		{ return grid->GetNextColAt(line); }
+
     virtual wxWindow *GetHeaderWindow(wxGrid *grid) const
         { return grid->GetGridColLabelWindow(); }
     virtual int GetHeaderWindowSize(wxGrid *grid) const
Index: wx/generic/grid.h
===================================================================
--- wx/generic/grid.h	(r‚vision 64459)
+++ wx/generic/grid.h	(copie de travail)
@@ -1372,6 +1372,56 @@
         return m_colAt.empty() ? pos : m_colAt[pos];
     }
 
+	// return the column index corresponding to the column at left of the current 
+	// given (valid) position
+	int GetPreviousColAt(int pos) const
+	{
+		if (m_colAt.empty() == true) {
+			if ( ( pos - 1) >= 0 )
+				return (pos - 1);
+			else
+				return pos;
+		}
+
+		for ( int i = 0; i < m_numCols; i++ ) 
+		{
+			if (m_colAt[i] == pos) {
+				// If current column is the first one, return current column
+				if (i == 0) return pos;
+
+				// If not, return the column at left (Previous)
+				return m_colAt[i-1];
+			}
+		}
+
+		return pos;
+	}
+	
+	// return the column index corresponding to the column at right of the current 
+	// given (valid) position
+	int GetNextColAt(int pos) const
+	{
+		if (m_colAt.empty() == true) {
+			if ( (pos + 1) < ( m_numCols - 1 ) )
+				return ( pos + 1 );
+			else
+				return pos;
+		}
+
+		for ( int i = 0; i < m_numCols; i++ ) 
+		{
+			if (m_colAt[i] == pos) {
+				// If current column is the last one, return current column
+				if (i == (m_numCols-1)) return pos;
+
+				// If not, return the column at left (Previous)
+				return m_colAt[i-1];
+			}
+		}
+
+		return pos;
+	}
+
     // reorder the columns so that the column with the given index is now shown
     // as the position pos
     void SetColPos(int idx, int pos);
Index: interface/wx/grid.h
===================================================================
--- interface/wx/grid.h	(r‚vision 64459)
+++ interface/wx/grid.h	(copie de travail)
@@ -3071,6 +3071,19 @@
     */
     int GetColAt(int colPos) const;
 
+	/**
+        Returns the column ID of the column position just before 
+		the specified column position.
+    */
+    int GetPreviousColAt(int colPos) const;
+
+	/**
+        Returns the column ID of the column position just after
+		the specified column position.
+    */
+    int GetNextColAt(int colPos) const;
+
+
     /**
         Returns the position of the specified column.
     */
