Index: include/wx/wfstream.h
===================================================================
--- include/wx/wfstream.h	(revision 55744)
+++ include/wx/wfstream.h	(working copy)
@@ -93,8 +93,8 @@
     virtual ~wxTempFileOutputStream();
 
     bool Close() { return Commit(); }
-    WXEXPORT virtual bool Commit() { return m_file->Commit(); }
-    WXEXPORT virtual void Discard() { m_file->Discard(); }
+    virtual bool Commit();
+    virtual void Discard();
 
     wxFileOffset GetLength() const { return m_file->Length(); }
     bool IsSeekable() const { return true; }
Index: src/common/wfstream.cpp
===================================================================
--- src/common/wfstream.cpp	(revision 55744)
+++ src/common/wfstream.cpp	(working copy)
@@ -215,6 +215,16 @@
     return 0;
 }
 
+bool wxTempFileOutputStream::Commit()
+{ 
+    return m_file->Commit();
+}
+
+void wxTempFileOutputStream::Discard()
+{ 
+    m_file->Discard();
+}
+
 // ----------------------------------------------------------------------------
 // wxFileStream
 // ----------------------------------------------------------------------------
