Index: notebook.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/os2/notebook.cpp,v
retrieving revision 1.32
diff -u -r1.32 notebook.cpp
--- notebook.cpp	2007/04/06 20:06:41	1.32
+++ notebook.cpp	2007/04/28 13:23:51
@@ -266,7 +266,7 @@
     return (bool)::WinSendMsg( m_hWnd
                               ,BKM_SETTABTEXT
                               ,MPFROMLONG((ULONG)m_alPageId[nPage])
-                              ,MPFROMP(rsStrText.c_str())
+                              ,MPFROMP((const char*)rsStrText.c_str())
                              );
 } // end of wxNotebook::SetPageText
 

 	  	 
