Index: src/osx/carbon/window.cpp
===================================================================
--- src/osx/carbon/window.cpp	(revision 60881)
+++ src/osx/carbon/window.cpp	(working copy)
@@ -184,7 +184,6 @@
 #endif
 
                 {
-                    bool created = false ;
                     CGContextRef cgContext = NULL ;
                     OSStatus err = cEvent.GetParameter<CGContextRef>(kEventParamCGContextRef, &cgContext) ;
                     if ( err != noErr )
@@ -208,7 +207,7 @@
                                     iter = iter->GetParent() ;
                             }
                         }
-                        CGContextSetAlpha( cgContext , alpha ) ;
+                        CGContextSetAlpha( cgContext, alpha ) ;
 
                         if ( thisWindow->GetBackgroundStyle() == wxBG_STYLE_TRANSPARENT )
                         {
@@ -229,16 +228,18 @@
                                     ( thisWindow->IsTopLevel() && thisWindow->GetBackgroundStyle() == wxBG_STYLE_SYSTEM ) )
                                 {
                                     if ( thisWindow->GetBackgroundStyle() != wxBG_STYLE_TRANSPARENT )
-                                        CallNextEventHandler( handler ,event ) ;
+                                        CallNextEventHandler( handler, event ) ;
                                 }
                             }
+                            else
+                            {
+                                // signal that paint was handled
+                                result = noErr;
+                            }
                             thisWindow->MacPaintChildrenBorders();
                         }
                         thisWindow->MacSetCGContextRef( NULL ) ;
                     }
-
-                    if ( created )
-                        CGContextRelease( cgContext ) ;
                 }
 
                 if ( allocatedRgn )
