 samples/toolbar/toolbar.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/samples/toolbar/toolbar.cpp b/samples/toolbar/toolbar.cpp
index 90e1a3d4c8..01c0f74ed4 100644
--- a/samples/toolbar/toolbar.cpp
+++ b/samples/toolbar/toolbar.cpp
@@ -362,7 +362,7 @@ void MyFrame::RecreateToolbar()
     if ( style & wxTB_TEXT && !(style & wxTB_NOICONS) && m_horzText )
         style |= wxTB_HORZ_LAYOUT;
 
-    toolBar = CreateToolBar(style, ID_TOOLBAR);
+    toolBar = CreateToolBar(wxTB_HORIZONTAL | wxTB_BOTTOM, ID_TOOLBAR);
 
     PopulateToolbar(toolBar);
 }
@@ -652,7 +652,7 @@ MyFrame::MyFrame(wxFrame* parent,
         sizer->Add(m_extraToolBar, 0, wxEXPAND, 0);
 #endif
     sizer->Add(m_textWindow, 1, wxEXPAND, 0);
-
+    m_textWindow->SetBackgroundColour(*wxRED);
     SetInitialSize(FromDIP(wxSize(650, 350)));
 }
 
