Index: src/mac/carbon/msgdlg.cpp
===================================================================
--- src/mac/carbon/msgdlg.cpp	(revision 49529)
+++ src/mac/carbon/msgdlg.cpp	(working copy)
@@ -39,30 +39,35 @@
 
 bool wxMessageDialog::SetYesNoLabels(const wxString& yes,const wxString& no)
 {
-    m_yes = yes;
-    m_no = no;
-    return true;
+	m_yes = yes;
+	m_yes.Replace(_("&"),_(""));
+	m_no = no;
+	m_no.Replace(_("&"),_(""));
+	return true;
 }
-
 bool wxMessageDialog::SetYesNoCancelLabels(const wxString& yes, const wxString& no, const wxString& cancel)
 {
-    m_yes = yes;
-    m_no = no;
-    m_cancel = cancel;
-    return true;
+	m_yes = yes;
+	m_yes.Replace(_("&"),_(""));
+	m_no = no;
+	m_no.Replace(_("&"),_(""));
+	m_cancel = cancel;
+	m_cancel.Replace(_("&"),_(""));
+	return true;
 }
-
 bool wxMessageDialog::SetOKLabel(const wxString& ok)
 {
-    m_ok = ok;
-    return true;
+	m_ok = ok;
+	m_ok.Replace(_("&"),_(""));
+	return true;
 }
-
 bool wxMessageDialog::SetOKCancelLabels(const wxString& ok, const wxString& cancel)
 {
-    m_ok = ok;
-    m_cancel = cancel;
-    return true;
+	m_ok = ok;
+	m_ok.Replace(_("&"),_(""));
+	m_cancel = cancel;
+	m_cancel.Replace(_("&"),_(""));
+	return true;
 }
 
 int wxMessageDialog::ShowModal()

 	  	 
