? busyinfo.cpp.patch
Index: busyinfo.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/generic/busyinfo.cpp,v
retrieving revision 1.23
diff -u -r1.23 busyinfo.cpp
--- busyinfo.cpp	2007/03/17 16:20:33	1.23
+++ busyinfo.cpp	2007/03/18 19:54:49
@@ -115,6 +115,13 @@
 wxBusyInfo::wxBusyInfo(const wxString& message, wxWindow *parent)
 {
     m_InfoFrame = new wxInfoFrame( parent, message);
+
+    // need to show busyinfo in front of an topstaying parent frame
+    if (parent != NULL && parent->GetWindowStyleFlag() & wxSTAY_ON_TOP)
+    {
+        m_InfoFrame->SetWindowStyleFlag(wxSTAY_ON_TOP);
+    }
+
     m_InfoFrame->Show(true);
     m_InfoFrame->Refresh();
     m_InfoFrame->Update();
@@ -134,4 +141,4 @@
     m_InfoFrame->Close();
 }
 
-#endif // wxUSE_BUSYINFO
+#endif // wxUSE_BUSYINFO
\ No newline at end of file

 	  	 
