Index: include/wx/buffer.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/buffer.h,v
retrieving revision 1.34
diff -u -r1.34 buffer.h
--- include/wx/buffer.h	2005/04/16 04:08:35	1.34
+++ include/wx/buffer.h	2005/09/19 05:39:12
@@ -251,7 +251,7 @@
         m_bufdata->m_len += 1;
     }
 
-    void  AppendData(void* data, size_t len)
+    void  AppendData(const void* data, size_t len)
     {
         memcpy(GetAppendBuf(len), data, len);
         UngetAppendBuf(len);


 	  	 
