diff --git a/src/generic/grid.cpp b/src/generic/grid.cpp
index 594aea1..2312755 100644
--- a/src/generic/grid.cpp
+++ b/src/generic/grid.cpp
@@ -5013,7 +5013,7 @@ void wxGrid::OnKeyUp( wxKeyEvent& event )
 void wxGrid::OnChar( wxKeyEvent& event )
 {
     // is it possible to edit the current cell at all?
-    if ( !IsCellEditControlEnabled() && CanEnableCellControl() )
+    if ( !IsCellEditControlEnabled() && CanEnableCellControl() && m_currentCellCoords.GetCol()>0 && m_colWidths.size()>size_t(m_currentCellCoords.GetCol()) && m_colWidths[m_currentCellCoords.GetCol()]>0 )
     {
         // yes, now check whether the cells editor accepts the key
         int row = m_currentCellCoords.GetRow();
