diff -rbu2 wxWindows.orig/src/common/gdicmn.cpp wxWindows/src/common/gdicmn.cpp
--- wxWindows.orig/src/common/gdicmn.cpp	Fri Oct 26 09:39:03 2001
+++ wxWindows/src/common/gdicmn.cpp	Fri Oct 26 10:13:51 2001
@@ -608,4 +608,5 @@
     // (dangling pointer).
     pen->SetVisible(TRUE);
+    AddPen(pen); 
 
     return pen;
@@ -654,4 +655,5 @@
     }
 
+    AddBrush(brush);
     brush->SetVisible(TRUE);
 
@@ -760,4 +762,5 @@
                           underline, facename, encoding);
 
+        AddFont(font);
         // and mark it as being cacheable
         font->SetVisible(TRUE);
diff -rbu2 wxWindows.orig/src/msw/bitmap.cpp wxWindows/src/msw/bitmap.cpp
--- wxWindows.orig/src/msw/bitmap.cpp	Fri Oct 26 09:39:39 2001
+++ wxWindows/src/msw/bitmap.cpp	Fri Oct 26 10:05:09 2001
@@ -108,6 +108,4 @@
     // m_refData = NULL; done in the base class ctor
 
-    if ( wxTheBitmapList )
-        wxTheBitmapList->AddBitmap(this);
 }
 
@@ -223,6 +221,4 @@
 wxBitmap::~wxBitmap()
 {
-    if (wxTheBitmapList)
-        wxTheBitmapList->DeleteObject(this);
 }
 
@@ -616,6 +612,4 @@
     GetBitmapData()->SetOk();
 #endif // WXWIN_COMPATIBILITY_2
-
-    if (wxTheBitmapList) wxTheBitmapList->AddBitmap(this);
 
     return TRUE;
diff -rbu2 wxWindows.orig/src/msw/brush.cpp wxWindows/src/msw/brush.cpp
--- wxWindows.orig/src/msw/brush.cpp	Fri Oct 26 09:39:39 2001
+++ wxWindows/src/msw/brush.cpp	Fri Oct 26 10:09:19 2001
@@ -59,12 +59,8 @@
 wxBrush::wxBrush(void)
 {
-  if ( wxTheBrushList )
-    wxTheBrushList->AddBrush(this);
 }
 
 wxBrush::~wxBrush()
 {
-    if (wxTheBrushList)
-        wxTheBrushList->RemoveBrush(this);
 }
 
@@ -79,6 +75,4 @@
   RealizeResource();
 
-  if ( wxTheBrushList )
-    wxTheBrushList->AddBrush(this);
 }
 
@@ -97,6 +91,4 @@
   RealizeResource();
 
-  if ( wxTheBrushList )
-    wxTheBrushList->AddBrush(this);
 }
 
diff -rbu2 wxWindows.orig/src/msw/font.cpp wxWindows/src/msw/font.cpp
--- wxWindows.orig/src/msw/font.cpp	Fri Oct 26 09:39:41 2001
+++ wxWindows/src/msw/font.cpp	Fri Oct 26 10:07:32 2001
@@ -371,6 +371,4 @@
 void wxFont::Init()
 {
-    if ( wxTheFontList )
-        wxTheFontList->Append(this);
 }
 
@@ -421,6 +419,4 @@
 wxFont::~wxFont()
 {
-    if ( wxTheFontList )
-        wxTheFontList->DeleteObject(this);
 }
 
diff -rbu2 wxWindows.orig/src/msw/pen.cpp wxWindows/src/msw/pen.cpp
--- wxWindows.orig/src/msw/pen.cpp	Fri Oct 26 09:39:44 2001
+++ wxWindows/src/msw/pen.cpp	Fri Oct 26 10:08:35 2001
@@ -68,12 +68,8 @@
 wxPen::wxPen()
 {
-  if (wxThePenList)
-    wxThePenList->AddPen(this);
 }
 
 wxPen::~wxPen()
 {
-    if (wxThePenList)
-        wxThePenList->RemovePen(this);
 }
 
@@ -118,6 +114,4 @@
   RealizeResource();
 
-  if ( wxThePenList )
-    wxThePenList->AddPen(this);
 }
 
@@ -138,6 +132,4 @@
     RealizeResource();
 
-    if (wxThePenList)
-       wxThePenList->AddPen(this);
 }
 

 	  	 
