Index: src/mac/carbon/filedlg.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWidgets/src/mac/carbon/filedlg.cpp,v
retrieving revision 1.59
diff -u -r1.59 filedlg.cpp
--- src/mac/carbon/filedlg.cpp	2006/01/15 07:20:59	1.59
+++ src/mac/carbon/filedlg.cpp	2006/03/06 11:30:39
@@ -171,13 +171,13 @@
         {
             wxUint32 fileType, creator;
             wxString extension = myData->extensions[i];
-
+			
             // Remove leading '*'
-            if (extension.GetChar(0) == '*')
+            if (extension.length() && (extension.GetChar(0) == '*'))
                 extension = extension.Mid( 1 );
 
             // Remove leading '.'
-            if (extension.GetChar(0) == '.')
+            if (extension.length() && (extension.GetChar(0) == '*'))
                 extension = extension.Mid( 1 );
        
             if (wxFileName::MacFindDefaultTypeAndCreator( extension, &fileType, &creator ))

 	  	 
