Index: src/mac/thread.cpp
===================================================================
RCS file: /home/wxcvs/wxWindows/src/mac/thread.cpp,v
retrieving revision 1.18
diff -u -r1.18 thread.cpp
--- src/mac/thread.cpp	23 Jun 2002 17:18:11 -0000	1.18
+++ src/mac/thread.cpp	19 Jul 2002 22:34:53 -0000
@@ -523,11 +523,11 @@
 
 void wxThread::Sleep(unsigned long milliseconds)
 {
-        clock_t start = clock() ;
-        do
-        {
-            YieldToAnyThread() ;
-        } while( clock() - start < milliseconds / CLOCKS_PER_SEC ) ;
+    clock_t start = clock();
+    do
+    {
+        YieldToAnyThread();
+    } while( clock() - start < (milliseconds * CLOCKS_PER_SEC) / 1000 ) ;
 }
 
 int wxThread::GetCPUCount()

 	  	 
