Index: include/wx/xrc/xmlres.h
===================================================================
--- include/wx/xrc/xmlres.h	(revision 75725)
+++ include/wx/xrc/xmlres.h	(working copy)
@@ -481,6 +481,8 @@
     // e.g. <object class="wxDialog">.
     bool IsOfClass(wxXmlNode *node, const wxString& classname) const;
 
+    bool IsElementNode(const wxXmlNode *node) const;
+    const wxString& GetNodeName(const wxXmlNode *node) const;
     // Gets node content from wxXML_ENTITY_NODE
     // The problem is, <tag>content<tag> is represented as
     // wxXML_ENTITY_NODE name="tag", content=""
@@ -488,6 +490,10 @@
     //        wxXML_CDATA_SECTION_NODE name="" content="content"
     wxString GetNodeContent(const wxXmlNode *node);
 
+    wxXmlNode *GetNodeParent(const wxXmlNode *node) const;
+    wxXmlNode *GetNodeNext(const wxXmlNode *node) const;
+    wxXmlNode *GetNodeChildren(const wxXmlNode *node) const;
+
     // Check to see if a parameter exists.
     bool HasParam(const wxString& param);
 
Index: include/wx/xrc/xmlreshandler.h
===================================================================
--- include/wx/xrc/xmlreshandler.h	(revision 75725)
+++ include/wx/xrc/xmlreshandler.h	(working copy)
@@ -52,7 +52,12 @@
     virtual wxObject *CreateResource(wxXmlNode *node, wxObject *parent,
                                      wxObject *instance) = 0;
     virtual bool IsOfClass(wxXmlNode *node, const wxString& classname) const = 0;
+    virtual bool IsElementNode(const wxXmlNode *node) const = 0;
+    virtual const wxString& GetNodeName(const wxXmlNode *node) const = 0;
     virtual wxString GetNodeContent(const wxXmlNode *node) = 0;
+    virtual wxXmlNode *GetNodeParent(const wxXmlNode *node) const = 0;
+    virtual wxXmlNode *GetNodeNext(const wxXmlNode *node) const = 0;
+    virtual wxXmlNode *GetNodeChildren(const wxXmlNode *node) const = 0;
     virtual bool HasParam(const wxString& param) = 0;
     virtual wxXmlNode *GetParamNode(const wxString& param) = 0;
     virtual wxString GetParamValue(const wxString& param) = 0;
@@ -206,10 +211,33 @@
     {
         return GetImpl()->IsOfClass(node, classname);
     }
+    
+    bool IsElementNode(const wxXmlNode *node) const
+    {
+        return GetImpl()->IsElementNode(node);
+    }
+    const wxString& GetNodeName(const wxXmlNode *node) const
+    {
+        return GetImpl()->GetNodeName(node);
+    }
     wxString GetNodeContent(const wxXmlNode *node)
     {
         return GetImpl()->GetNodeContent(node);
     }
+    
+    wxXmlNode *GetNodeParent(const wxXmlNode *node) const
+    {
+        return GetImpl()->GetNodeParent(node);
+    }
+    wxXmlNode *GetNodeNext(const wxXmlNode *node) const
+    {
+        return GetImpl()->GetNodeNext(node);
+    }
+    wxXmlNode *GetNodeChildren(const wxXmlNode *node) const
+    {
+        return GetImpl()->GetNodeChildren(node);
+    }
+
     bool HasParam(const wxString& param)
     {
         return GetImpl()->HasParam(param);
Index: interface/wx/xrc/xmlres.h
===================================================================
--- interface/wx/xrc/xmlres.h	(revision 75725)
+++ interface/wx/xrc/xmlres.h	(working copy)
@@ -652,11 +652,33 @@
     wxString GetName();
 
     /**
+        Checks if the given node is of type wxXML_ELEMENT_NODE.
+    */
+    bool IsElementNode(const wxXmlNode *node) const;
+    /**
+        Gets the name of the node given or returns wxEmptyString.
+    */
+    const wxString& GetNodeName(const wxXmlNode *node) const;
+    /**
         Gets node content from wxXML_ENTITY_NODE.
     */
     wxString GetNodeContent(wxXmlNode* node);
 
     /**
+        Gets the parent of the node given or returns @NULL.
+    */
+    wxXmlNode *GetNodeParent(const wxXmlNode *node) const;
+    /**
+        Gets the next sibling node related to the node given or returns @NULL.
+    */
+    wxXmlNode *GetNodeNext(const wxXmlNode *node) const;
+    /**
+        Gets the children of the node given or returns @NULL.
+    */
+    wxXmlNode *GetNodeChildren(const wxXmlNode *node) const;
+
+
+    /**
         Finds the node or returns @NULL.
     */
     wxXmlNode* GetParamNode(const wxString& param);
Index: src/xrc/xh_auitoolb.cpp
===================================================================
--- src/xrc/xh_auitoolb.cpp	(revision 75725)
+++ src/xrc/xh_auitoolb.cpp	(working copy)
@@ -84,7 +84,7 @@
             // also check for the menu specified inside dropdown (it is
             // optional and may be absent for e.g. dynamically-created
             // menus)
-            wxXmlNode * const nodeMenu = nodeDropdown->GetChildren();
+            wxXmlNode * const nodeMenu = GetNodeChildren(nodeDropdown);
             if ( nodeMenu )
             {
                 wxObject *res = CreateResFromNode(nodeMenu, NULL);
@@ -98,11 +98,11 @@
                     );
                 }
 
-                if ( nodeMenu->GetNext() )
+                if ( GetNodeNext(nodeMenu) )
                 {
                     ReportError
                     (
-                        nodeMenu->GetNext(),
+                        GetNodeNext(nodeMenu),
                         "unexpected extra contents under drop-down tool"
                     );
                 }
@@ -195,8 +195,8 @@
 
         while (n)
         {
-            if ((n->GetType() == wxXML_ELEMENT_NODE) &&
-                (n->GetName() == wxS("object") || n->GetName() == wxS("object_ref")))
+            if (IsElementNode(n) &&
+                (GetNodeName(n) == wxS("object") || GetNodeName(n) == wxS("object_ref")))
             {
                 wxObject *created = CreateResFromNode(n, toolbar, NULL);
                 wxControl *control = wxDynamicCast(created, wxControl);
@@ -206,7 +206,7 @@
                     control != NULL)
                     toolbar->AddControl(control);
             }
-            n = n->GetNext();
+            n = GetNodeNext(n);
         }
 
         m_isInside = false;
Index: src/xrc/xmlres.cpp
===================================================================
--- src/xrc/xmlres.cpp	(revision 75725)
+++ src/xrc/xmlres.cpp	(working copy)
@@ -2007,7 +2007,22 @@
 }
 
 
+bool wxXmlResourceHandlerImpl::IsElementNode(const wxXmlNode *node) const
+{
+    if (node == NULL) return false;
+    return (node->GetType() == wxXML_ELEMENT_NODE);
+}
 
+const wxString& wxXmlResourceHandlerImpl::GetNodeName(const wxXmlNode *node) const
+{
+    if (node == NULL)
+    {
+        static wxString sEmptyString = wxEmptyString;
+        return sEmptyString;
+    }
+    return node->GetContent();
+}
+
 wxString wxXmlResourceHandlerImpl::GetNodeContent(const wxXmlNode *node)
 {
     const wxXmlNode *n = node;
@@ -2024,8 +2039,26 @@
     return wxEmptyString;
 }
 
+wxXmlNode *wxXmlResourceHandlerImpl::GetNodeParent(const wxXmlNode *node) const
+{
+    if (node == NULL) return NULL;
+    return node->GetParent();
+}
 
+wxXmlNode *wxXmlResourceHandlerImpl::GetNodeNext(const wxXmlNode *node) const
+{
+    if (node == NULL) return NULL;
+    return node->GetNext();
+}
 
+wxXmlNode *wxXmlResourceHandlerImpl::GetNodeChildren(const wxXmlNode *node) const
+{
+    if (node == NULL) return NULL;
+    return node->GetChildren();
+}
+
+
+
 wxString wxXmlResourceHandlerImpl::GetParamValue(const wxString& param)
 {
     if (param.empty())
