Index: include/wx/app.h
===================================================================
--- include/wx/app.h	(revision 61819)
+++ include/wx/app.h	(working copy)
@@ -136,20 +136,18 @@
     //     be argv[0]
 
         // set/get the application name
-    wxString GetAppName() const
-    {
-        return m_appName.empty() ? m_className : m_appName;
-    }
+    wxString GetAppName() const;
     void SetAppName(const wxString& name) { m_appName = name; }
 
         // set/get the application display name: the display name is the name
         // shown to the user in titles, reports, etc while the app name is
         // used for paths, config, and other places the user doesn't see
         //
-        // so the app name could be myapp while display name could be "My App"
+        // Capitalize, so the app name could be myapp while display name could be "Myapp",
+        // but only if SetAppName() not already called
     wxString GetAppDisplayName() const
     {
-        return m_appDisplayName.empty() ? GetAppName().Capitalize()
+        return m_appDisplayName.empty() ? (m_appName.empty() ? GetAppName().Capitalize() : GetAppName())
                                         : m_appDisplayName;
     }
 
Index: src/common/appbase.cpp
===================================================================
--- src/common/appbase.cpp	(revision 61819)
+++ src/common/appbase.cpp	(working copy)
@@ -161,21 +161,30 @@
 // initialization/cleanup
 // ----------------------------------------------------------------------------
 
-bool wxAppConsoleBase::Initialize(int& WXUNUSED(argc), wxChar **argv)
+bool wxAppConsoleBase::Initialize(int& WXUNUSED(argc), wxChar **WXUNUSED(argv))
 {
 #if wxUSE_INTL
     GetTraits()->SetLocale();
 #endif // wxUSE_INTL
 
+    return true;
+}
+
+wxString wxAppConsoleBase::GetAppName() const
+{
+    wxString name = m_appName;
 #ifndef __WXPALMOS__
-    if ( m_appName.empty() && argv && argv[0] )
+    if ( name.empty() )
     {
-        // the application name is, by default, the name of its executable file
-        wxFileName::SplitPath(argv[0], NULL, &m_appName, NULL);
+        wxChar** _argv = argv;
+        if (_argv && _argv[0])
+        {
+            // the application name is, by default, the name of its executable file
+            wxFileName::SplitPath(_argv[0], NULL, &name, NULL);
+        }
     }
 #endif // !__WXPALMOS__
-
-    return true;
+    return name;
 }
 
 wxEventLoopBase *wxAppConsoleBase::CreateMainLoop()
