Index: dcclient.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/mac/carbon/dcclient.cpp,v
retrieving revision 1.45
diff -u -r1.45 dcclient.cpp
--- dcclient.cpp	2006/05/08 07:58:21	1.45
+++ dcclient.cpp	2006/05/14 17:03:33
@@ -75,7 +75,8 @@
                 break ;
             }
 
-            if ( parent->IsKindOf( CLASSINFO( wxNotebook ) ) || parent->IsKindOf( CLASSINFO( wxTabCtrl ) ) )
+            //if ( parent->IsKindOf( CLASSINFO( wxNotebook ) ) || parent->IsKindOf( CLASSINFO( wxTabCtrl ) ) )
+            if ( parent->IsKindOf( CLASSINFO( wxNotebook ) ) )
             {
                 Rect extent = { 0 , 0 , 0 , 0 } ;
                 int x , y ;

 	  	 
