Index: graphics.cpp
===================================================================
--- graphics.cpp	(revision 71808)
+++ graphics.cpp	(working copy)
@@ -1286,12 +1286,13 @@
 
     REAL factorY = m_context->GetDpiY() / 72.0 ;
 
-    REAL rDescent = ffamily.GetCellDescent(FontStyleRegular) *
-        f->GetSize() / ffamily.GetEmHeight(FontStyleRegular);
-    REAL rAscent = ffamily.GetCellAscent(FontStyleRegular) *
-        f->GetSize() / ffamily.GetEmHeight(FontStyleRegular);
-    REAL rHeight = ffamily.GetLineSpacing(FontStyleRegular) *
-        f->GetSize() / ffamily.GetEmHeight(FontStyleRegular);
+    INT style = f->GetStyle();
+    REAL rDescent = ffamily.GetCellDescent(style) *
+        f->GetSize() / ffamily.GetEmHeight(style);
+    REAL rAscent = ffamily.GetCellAscent(style) *
+        f->GetSize() / ffamily.GetEmHeight(style);
+    REAL rHeight = ffamily.GetLineSpacing(style) *
+        f->GetSize() / ffamily.GetEmHeight(style);
 
     if ( height )
         *height = rHeight * factorY;
