diff -bu1 -r wxWindows/samples/nativdlg/nativdlg.cpp wxWin/samples/nativdlg/nativdlg.cpp
--- wxWindows/samples/nativdlg/nativdlg.cpp	2003-06-05 13:56:24.000000000 +0200
+++ wxWin/samples/nativdlg/nativdlg.cpp	2004-01-13 13:40:08.000000000 +0100
@@ -34,5 +34,2 @@
 
-// Declare two frames
-MyFrame   *frame = NULL;
-
 IMPLEMENT_APP(MyApp)
@@ -47,3 +44,3 @@
   // Create the main frame window
-  frame = new MyFrame(NULL, -1, _T("wxWindows Native Dialog Sample"), wxPoint(0, 0), wxSize(300, 250));
+  MyFrame   *frame = new MyFrame(NULL, wxID_ANY, _T("wxWindows Native Dialog Sample"), wxPoint(0, 0), wxSize(300, 250));
 
@@ -66,4 +63,4 @@
   // Make a panel
-  frame->panel = new wxWindow(frame, -1, wxPoint(0, 0), wxSize(400, 400), 0, _T("MyMainFrame"));
-  frame->Show(TRUE);
+  frame->panel = new wxWindow(frame, wxID_ANY, wxPoint(0, 0), wxSize(400, 400), 0, _T("MyMainFrame"));
+  frame->Show(true);
 
@@ -72,3 +69,3 @@
 
-  return TRUE;
+  return true;
 }
@@ -87,9 +84,10 @@
 
-void MyFrame::OnQuit(wxCommandEvent& event)
+void MyFrame::OnQuit(wxCommandEvent& WXUNUSED(event))
 {
-    Close(TRUE);
+  Close(true);
 }
 
-void MyFrame::OnTest1(wxCommandEvent& event)
+void MyFrame::OnTest1(wxCommandEvent& WXUNUSED(event))
 {
+#if ( defined(__WXOS2__) || defined(__WXMSW__) ) && !defined(__WXUNIVERSAL__)
       MyDialog *dialog = new MyDialog;
@@ -97,6 +95,9 @@
       {
-        dialog->SetModal(TRUE);
+    dialog->SetModal(true);
         dialog->ShowModal();
       }
-      dialog->Close(TRUE);
+  dialog->Close(true);
+#else
+  wxMessageBox(_T("No native dialog support"),_T("Platform limitation"));
+#endif
 }
@@ -109,3 +110,3 @@
 
-void MyDialog::OnOk(wxCommandEvent& event)
+void MyDialog::OnOk(wxCommandEvent& WXUNUSED(event))
 {
@@ -114,3 +115,3 @@
 
-void MyDialog::OnCancel(wxCommandEvent& event)
+void MyDialog::OnCancel(wxCommandEvent& WXUNUSED(event))
 {

 	  	 
