diff -bu1 -r wxWindows/samples/richedit/wxLayout.cpp wxWin/samples/richedit/wxLayout.cpp
--- wxWindows/samples/richedit/wxLayout.cpp	2002-03-27 18:44:50.000000000 +0100
+++ wxWin/samples/richedit/wxLayout.cpp	2003-08-26 18:46:50.000000000 +0200
@@ -72,3 +72,3 @@
 MyFrame::MyFrame(void) :
-   wxFrame( (wxFrame *) NULL, -1, "wxLayout",
+   wxFrame( (wxFrame *) NULL, -1, _T("wxLayout"),
              wxPoint(880,100), wxSize(256,256) )
@@ -77,3 +77,3 @@
 
-   SetStatusText( "wxLayout by Karsten Ballüder." );
+   SetStatusText( _T("wxLayout by Karsten Ballüder.") );
 
@@ -82,39 +82,39 @@
    wxMenu *file_menu = new wxMenu;
-   file_menu->Append(ID_PRINT, "&Print...", "Print");
-   file_menu->Append(ID_PRINT_SETUP, "Print &Setup...","Setup printer properties");
-   file_menu->Append(ID_PAGE_SETUP, "Page Set&up...", "Page setup");
-   file_menu->Append(ID_PREVIEW, "Print Pre&view", "Preview");
+   file_menu->Append(ID_PRINT, _T("&Print..."), _T("Print"));
+   file_menu->Append(ID_PRINT_SETUP, _T("Print &Setup..."),_T("Setup printer properties"));
+   file_menu->Append(ID_PAGE_SETUP, _T("Page Set&up..."), _T("Page setup"));
+   file_menu->Append(ID_PREVIEW, _T("Print Pre&view"), _T("Preview"));
 #ifdef __WXMSW__
    file_menu->AppendSeparator();
-   file_menu->Append(ID_PRINT_PS, "Print PostScript...", "Print (PostScript)");
-   file_menu->Append(ID_PRINT_SETUP_PS, "Print Setup PostScript...", "Setup printer properties (PostScript)");
-   file_menu->Append(ID_PAGE_SETUP_PS, "Page Setup PostScript...", "Page setup (PostScript)");
-   file_menu->Append(ID_PREVIEW_PS, "Print Preview PostScript", "Preview (PostScript)");
+   file_menu->Append(ID_PRINT_PS, _T("Print PostScript..."), _T("Print (PostScript)"));
+   file_menu->Append(ID_PRINT_SETUP_PS, _T("Print Setup PostScript..."), _T("Setup printer properties (PostScript)"));
+   file_menu->Append(ID_PAGE_SETUP_PS, _T("Page Setup PostScript..."), _T("Page setup (PostScript)"));
+   file_menu->Append(ID_PREVIEW_PS, _T("Print Preview PostScript"), _T("Preview (PostScript)"));
 #endif
    file_menu->AppendSeparator();
-   file_menu->Append( ID_TEXT, "Export &Text");
-   file_menu->Append( ID_HTML, "Export &HTML");
-   file_menu->Append( ID_QUIT, "E&xit");
-   menu_bar->Append(file_menu, "&File" );
+   file_menu->Append( ID_TEXT, _T("Export &Text"));
+   file_menu->Append( ID_HTML, _T("Export &HTML"));
+   file_menu->Append( ID_QUIT, _T("E&xit"));
+   menu_bar->Append(file_menu, _T("&File"));
 
    wxMenu *edit_menu = new wxMenu;
-   edit_menu->Append( ID_CLEAR, "C&lear");
-   edit_menu->Append( ID_ADD_SAMPLE, "&Example");
-   edit_menu->Append( ID_LONG_TEST, "Add &many lines");
+   edit_menu->Append( ID_CLEAR, _T("C&lear"));
+   edit_menu->Append( ID_ADD_SAMPLE, _T("&Example"));
+   edit_menu->Append( ID_LONG_TEST, _T("Add &many lines"));
    edit_menu->AppendSeparator();
-   edit_menu->Append( ID_LINEBREAKS_TEST, "Add &several lines");
-   edit_menu->Append( ID_URL_TEST, "Insert an &URL");
+   edit_menu->Append( ID_LINEBREAKS_TEST, _T("Add &several lines"));
+   edit_menu->Append( ID_URL_TEST, _T("Insert an &URL"));
    edit_menu->AppendSeparator();
-   edit_menu->Append(ID_WRAP, "&Wrap mode", "Activate wrapping at pixel 200.");
-   edit_menu->Append(ID_NOWRAP, "&No-wrap mode", "Deactivate wrapping.");
+   edit_menu->Append(ID_WRAP, _T("&Wrap mode"), _T("Activate wrapping at pixel 200."));
+   edit_menu->Append(ID_NOWRAP, _T("&No-wrap mode"), _T("Deactivate wrapping."));
    edit_menu->AppendSeparator();
-   edit_menu->Append(ID_COPY, "&Copy", "Copy text to clipboard.");
-   edit_menu->Append(ID_CUT, "Cu&t", "Cut text to clipboard.");
-   edit_menu->Append(ID_PASTE,"&Paste", "Paste text from clipboard.");
+   edit_menu->Append(ID_COPY, _T("&Copy"), _T("Copy text to clipboard."));
+   edit_menu->Append(ID_CUT, _T("Cu&t"), _T("Cut text to clipboard."));
+   edit_menu->Append(ID_PASTE,_T("&Paste"), _T("Paste text from clipboard."));
 #ifdef __WXGTK__
-   edit_menu->Append(ID_COPY_PRIMARY, "C&opy primary", "Copy text to primary selecton.");
-   edit_menu->Append(ID_PASTE_PRIMARY,"&Paste primary", "Paste text from primary selection.");
+   edit_menu->Append(ID_COPY_PRIMARY, _T("C&opy primary"), _T("Copy text to primary selecton."));
+   edit_menu->Append(ID_PASTE_PRIMARY, _T("&Paste primary"), _T("Paste text from primary selection."));
 #endif
-   edit_menu->Append(ID_FIND, "&Find", "Find text.");
-   menu_bar->Append(edit_menu, "&Edit" );
+   edit_menu->Append(ID_FIND, _T("&Find"), _T("Find text."));
+   menu_bar->Append(edit_menu, _T("&Edit") );
 
@@ -163,6 +163,6 @@
    llist->Clear(wxSWISS,16,wxNORMAL,wxNORMAL, false);
-   llist->SetFont(-1,-1,-1,-1,-1,"blue");
-   llist->Insert("blue");
-   llist->SetFont(-1,-1,-1,-1,-1,"black");
-   llist->Insert("The quick brown fox jumps over the lazy dog.");
+   llist->SetFont(-1,-1,-1,-1,-1,_T("blue"));
+   llist->Insert(_T("blue"));
+   llist->SetFont(-1,-1,-1,-1,-1,_T("black"));
+   llist->Insert(_T("The quick brown fox jumps over the lazy dog."));
    llist->LineBreak();
@@ -170,3 +170,3 @@
    llist->SetFont(wxROMAN,16,wxNORMAL,wxNORMAL, false);
-   llist->Insert("--");
+   llist->Insert(_T("--"));
    llist->LineBreak();
@@ -174,5 +174,5 @@
    llist->SetFont(wxROMAN);
-   llist->Insert("The quick brown fox jumps over the lazy dog.");
+   llist->Insert(_T("The quick brown fox jumps over the lazy dog."));
    llist->LineBreak();
-   llist->Insert("Hello ");
+   llist->Insert(_T("Hello "));
 #if wxICON_IS_BITMAP
@@ -183,33 +183,33 @@
    llist->SetFontWeight(wxBOLD);
-   llist->Insert("World! ");
+   llist->Insert(_T("World! "));
    llist->SetFontWeight(wxNORMAL);
-   llist->Insert("The quick brown fox jumps...");
+   llist->Insert(_T("The quick brown fox jumps..."));
    llist->LineBreak();
-   llist->Insert("over the lazy dog.");
+   llist->Insert(_T("over the lazy dog."));
    llist->SetFont(-1,-1,-1,-1,true);
-   llist->Insert("underlined");
+   llist->Insert(_T("underlined"));
    llist->SetFont(-1,-1,-1,-1,false);
    llist->SetFont(wxROMAN);
-   llist->Insert("This is ");
-   llist->SetFont(-1,-1,-1,wxBOLD);  llist->Insert("BOLD ");  llist->SetFont(-1,-1,-1,wxNORMAL);
-   llist->Insert("and ");
+   llist->Insert(_T("This is "));
+   llist->SetFont(-1,-1,-1,wxBOLD);  llist->Insert(_T("BOLD "));  llist->SetFont(-1,-1,-1,wxNORMAL);
+   llist->Insert(_T("and "));
    llist->SetFont(-1,-1,wxITALIC);
-   llist->Insert("italics ");
+   llist->Insert(_T("italics "));
    llist->SetFont(-1,-1,wxNORMAL);
    llist->LineBreak();
-   llist->Insert("and ");
+   llist->Insert(_T("and "));
    llist->SetFont(-1,-1,wxSLANT);
-   llist->Insert("slanted");
+   llist->Insert(_T("slanted"));
    llist->SetFont(-1,-1,wxNORMAL);
-   llist->Insert(" text.");
+   llist->Insert(_T(" text."));
    llist->LineBreak();
-   llist->Insert("and ");
-   llist->SetFont(-1,-1,-1,-1,-1,"blue");
-   llist->Insert("blue");
-   llist->SetFont(-1,-1,-1,-1,-1,"black");
-   llist->Insert(" and ");
-   llist->SetFont(-1,-1,-1,-1,-1,"green","black");
-   llist->Insert("green on black");
-   llist->SetFont(-1,-1,-1,-1,-1,"black","white");
-   llist->Insert(" text.");
+   llist->Insert(_T("and "));
+   llist->SetFont(-1,-1,-1,-1,-1,_T("blue"));
+   llist->Insert(_T("blue"));
+   llist->SetFont(-1,-1,-1,-1,-1,_T("black"));
+   llist->Insert(_T(" and "));
+   llist->SetFont(-1,-1,-1,-1,-1,_T("green"),_T("black"));
+   llist->Insert(_T("green on black"));
+   llist->SetFont(-1,-1,-1,-1,-1,_T("black"),_T("white"));
+   llist->Insert(_T(" text."));
    llist->LineBreak();
@@ -217,9 +217,9 @@
    llist->SetFont(-1,-1,wxSLANT);
-   llist->Insert("Slanted");
+   llist->Insert(_T("Slanted"));
    llist->SetFont(-1,-1,wxNORMAL);
-   llist->Insert(" and normal text and ");
+   llist->Insert(_T(" and normal text and "));
    llist->SetFont(-1,-1,wxSLANT);
-   llist->Insert("slanted");
+   llist->Insert(_T("slanted"));
    llist->SetFont(-1,-1,wxNORMAL);
-   llist->Insert(" again.");
+   llist->Insert(_T(" again."));
    llist->LineBreak();
@@ -227,22 +227,21 @@
    // add some more text for testing:
-   llist->Insert("And here the source for the test program:");
+   llist->Insert(_T("And here the source for the test program:"));
    llist->LineBreak();
    llist->SetFont(wxTELETYPE,16);
-   llist->Insert("And here the source for the test program:");
+   llist->Insert(_T("And here the source for the test program:"));
    llist->LineBreak();
-   llist->Insert("And here the source for the test program:");
+   llist->Insert(_T("And here the source for the test program:"));
    llist->LineBreak();
-   llist->Insert("And here the source for the test program:");
+   llist->Insert(_T("And here the source for the test program:"));
 
-   char buffer[1024];
-   FILE *in = fopen("wxLayout.cpp","r");
-   if(in)
+   wxFile file( _T("wxLayout.cpp") );
+   if ( file.IsOpened() )
    {
-      for(;;)
+      off_t len = file.Length();
+      wxChar *data = (wxChar *)malloc(2*len);
+      if ( file.Read(data, len) == len )
       {
-         fgets(buffer,1024,in);
-         if(feof(in))
-            break;
-         wxLayoutImportText(llist, buffer);
+        wxLayoutImportText(llist, data);
       }
+      free(data);
    }
@@ -402,3 +401,3 @@
         //     (instead of doing it manually like M does now)
-        m_lwin->GetLayoutList()->Insert("http://www.wxwindows.org/");
+        m_lwin->GetLayoutList()->Insert(_T("http://www.wxwindows.org/"));
     }
@@ -414,7 +413,7 @@
    wxPrinter printer;
-   wxLayoutPrintout printout( m_lwin->GetLayoutList(),"Printout from wxLayout");
+   wxLayoutPrintout printout( m_lwin->GetLayoutList(), _T("Printout from wxLayout"));
    if (! printer.Print(this, &printout, TRUE))
       wxMessageBox(
-         "There was a problem printing.\nPerhaps your current printer is not set correctly?",
-         "Printing", wxOK);
+         _T("There was a problem printing.\nPerhaps your current printer is not set correctly?"),
+         _T("Printing"), wxOK);
 }
@@ -448,3 +447,3 @@
       delete preview;
-      wxMessageBox("There was a problem previewing.\nPerhaps your current printer is not set correctly?", "Previewing", wxOK);
+      wxMessageBox(_T("There was a problem previewing.\nPerhaps your current printer is not set correctly?"), _T("Previewing"), wxOK);
       return;
@@ -452,3 +451,3 @@
 
-   wxPreviewFrame *frame = new wxPreviewFrame(preview, this, "Demo Print Preview", wxPoint(100, 100), wxSize(600, 650));
+   wxPreviewFrame *frame = new wxPreviewFrame(preview, this, _T("Demo Print Preview"), wxPoint(100, 100), wxSize(600, 650));
    frame->Centre(wxBOTH);
@@ -466,3 +465,3 @@
    wxPrintPreview *preview = new wxPrintPreview(new wxLayoutPrintout( m_lwin->GetLayoutList()), new wxLayoutPrintout( m_lwin->GetLayoutList()), & printData);
-   wxPreviewFrame *frame = new wxPreviewFrame(preview, this, "Demo Print Preview", wxPoint(100, 100), wxSize(600, 650));
+   wxPreviewFrame *frame = new wxPreviewFrame(preview, this, _T("Demo Print Preview"), wxPoint(100, 100), wxSize(600, 650));
    frame->Centre(wxBOTH);
diff -bu1 -r wxWindows/samples/richedit/wxllist.cpp wxWin/samples/richedit/wxllist.cpp
--- wxWindows/samples/richedit/wxllist.cpp	2002-03-27 18:44:50.000000000 +0100
+++ wxWin/samples/richedit/wxllist.cpp	2003-08-26 19:24:17.000000000 +0200
@@ -75,5 +75,5 @@
 
-   static const char *g_aTypeStrings[] =
+   static const wxChar *g_aTypeStrings[] =
    {
-      "invalid", "text", "cmd", "icon"
+      _T("invalid"), _T("text"), _T("cmd"), _T("icon")
    };
@@ -335,3 +335,3 @@
 void
-wxLayoutObjectText::Layout(wxDC &dc, class wxLayoutList *llist)
+wxLayoutObjectText::Layout(wxDC &dc, class wxLayoutList *WXUNUSED(llist))
 {
@@ -348,4 +348,4 @@
    CoordType a,b,c,d,e,f;
-   dc.GetTextExtent("test ", &a, &b, &c);
-   dc.GetTextExtent("test", &d, &e, &f);
+   dc.GetTextExtent(_T("test "), &a, &b, &c);
+   dc.GetTextExtent(_T("test"), &d, &e, &f);
    wxASSERT(a != d);
@@ -353,3 +353,3 @@
    wxASSERT(c == f);
-   dc.GetTextExtent(" ", &d, &e, &f);
+   dc.GetTextExtent(_T(" "), &d, &e, &f);
    wxASSERT(a > 0);
@@ -430,3 +430,3 @@
 
-   wxString file = wxGetTempFileName("wxloexport");
+   wxString file = wxGetTempFileName(_T("wxloexport"));
 
@@ -474,4 +474,4 @@
 wxLayoutObjectIcon::Draw(wxDC &dc, wxPoint const &coords,
-                         wxLayoutList *wxllist,
-                         CoordType begin, CoordType /* len */)
+                         wxLayoutList *WXUNUSED(wxllist),
+                         CoordType WXUNUSED(begin), CoordType WXUNUSED(len) )
 {
@@ -683,5 +683,5 @@
 void
-wxLayoutObjectCmd::Draw(wxDC &dc, wxPoint const & /* coords */,
+wxLayoutObjectCmd::Draw(wxDC &dc, wxPoint const & WXUNUSED(coords),
                         wxLayoutList *wxllist,
-                        CoordType begin, CoordType /* len */)
+                        CoordType WXUNUSED(begin), CoordType WXUNUSED(len))
 {
@@ -1063,3 +1063,5 @@
 
+    #if 0
     wxFAIL_MSG(wxT("unreachable"));
+    #endif
 }
@@ -1104,3 +1106,3 @@
 
-   CoordType from, to, tempto;
+   CoordType from, to;
 
@@ -1118,3 +1120,3 @@
          // parts of the line need highlighting
-         tempto = xpos+(**i).GetLength();
+         xpos+(**i).GetLength();
          (**i).Draw(dc, pos, llist, from-xpos, to-xpos);
@@ -1140,3 +1142,3 @@
                      int cx,
-                     bool suppressSIupdate)
+                     bool WXUNUSED(suppressSIupdate))
 {
@@ -1201,3 +1203,3 @@
                else
-                  str = WXLO_CURSORCHAR;
+                  str = _T(WXLO_CURSORCHAR);
                dc.GetTextExtent(str, &width, &height, &descent);
@@ -1274,3 +1276,3 @@
       CoordType width, height, descent;
-      dc.GetTextExtent(WXLO_CURSORCHAR, &width, &height, &descent);
+      dc.GetTextExtent(_T(WXLO_CURSORCHAR), &width, &height, &descent);
       m_Height = height;
@@ -1293,3 +1295,3 @@
          CoordType width, height, descent;
-         dc.GetTextExtent(WXLO_CURSORCHAR, &width, &height, &descent);
+         dc.GetTextExtent(_T(WXLO_CURSORCHAR), &width, &height, &descent);
          cursorSize->x = width;
@@ -1370,3 +1372,3 @@
    // next line later on, remember it here:
-   wxString prependText = "";
+   wxString prependText = _T("");
    // we might need to adjust the cursor position later, so remember it
@@ -1472,3 +1474,6 @@
                                 ((xpos > breakpos) ? 1 : 0 ));
+      #if 0
+      // this assert is useless when xpos has unsigned type
       wxASSERT(xpos >= 0);
+      #endif
       llist->MoveCursorTo( wxPoint( xpos, m_Next->GetLineNumber()) );
@@ -1815,3 +1820,3 @@
 wxLayoutList::SetFont(int family, int size, int style, int weight,
-                      int underline, char const *fg, char const *bg)
+                      int underline, wxChar const *fg, wxChar const *bg)
 
@@ -2686,3 +2691,15 @@
 void
-wxLayoutList::DrawCursor(wxDC &dc, bool active, wxPoint const &translate)
+wxLayoutList::DrawCursor(wxDC &
+                               #ifdef WXLAYOUT_USE_CARET
+                               WXUNUSED(dc)
+                               #else
+                               dc
+                               #endif
+                               , bool 
+                                      #ifdef WXLAYOUT_USE_CARET
+                                      WXUNUSED(active)
+                                      #else
+                                      active
+                                      #endif
+                                      , wxPoint const &translate)
 {
@@ -3246,3 +3263,3 @@
 #if defined(__WXMSW__)
-   wxPrinterDC *psdc = new wxPrinterDC("","",WXLLIST_TEMPFILE,false);
+   wxPrinterDC *psdc = new wxPrinterDC(wxEmptyString,wxEmptyString,_T(WXLLIST_TEMPFILE),false);
 #else
@@ -3277,3 +3294,3 @@
    delete psdc;
-   wxRemoveFile(WXLLIST_TEMPFILE);
+   wxRemoveFile(_T(WXLLIST_TEMPFILE));
 }
diff -bu1 -r wxWindows/samples/richedit/wxllist.h wxWin/samples/richedit/wxllist.h
--- wxWindows/samples/richedit/wxllist.h	2002-03-27 18:44:50.000000000 +0100
+++ wxWin/samples/richedit/wxllist.h	2003-08-26 17:52:17.000000000 +0200
@@ -263,3 +263,3 @@
 public:
-   wxLayoutObjectText(const wxString &txt = "");
+   wxLayoutObjectText(const wxString &txt = wxEmptyString);
 
@@ -950,4 +950,4 @@
                 int weight=-1, int underline = -1,
-                char const *fg = NULL,
-                char const *bg = NULL);
+                wxChar const *fg = NULL,
+                wxChar const *bg = NULL);
    /// changes to the next larger font size
@@ -970,3 +970,3 @@
    /// set font colours by name
-   inline void SetFontColour(char const *fg, char const *bg = NULL)
+   inline void SetFontColour(wxChar const *fg, wxChar const *bg = NULL)
       { SetFont(-1,-1,-1,-1,-1,fg,bg); }
@@ -1281,3 +1281,3 @@
       { SetData(text.length() + 1, text.c_str()); }
-   const char *GetLayoutData() const { return (const char *)GetData(); }
+   const wxChar *GetLayoutData() const { return (const wxChar *)GetData(); }
 };
@@ -1302,3 +1302,3 @@
                     wxString const & title =
-                    "wxLayout Printout");
+                    _T("wxLayout Printout"));
    /// Destructor.
diff -bu1 -r wxWindows/samples/richedit/wxlparser.cpp wxWin/samples/richedit/wxlparser.cpp
--- wxWindows/samples/richedit/wxlparser.cpp	2001-10-30 13:32:30.000000000 +0100
+++ wxWin/samples/richedit/wxlparser.cpp	2003-08-26 19:53:46.000000000 +0200
@@ -30,3 +30,3 @@
 
-inline static bool IsEndOfLine(const char *p)
+inline static bool IsEndOfLine(const wxChar *p)
 {
@@ -45,5 +45,5 @@
    // VZ: I still don't like it... the string data may be shared...
-   char * cptr = (char *)str.c_str(); // const_cast
-   const char * begin = cptr;
-   char  backup;
+   wxChar * cptr = (wxChar *)str.c_str(); // const_cast
+   const wxChar * begin = cptr;
+   wxChar  backup;
 
@@ -78,3 +78,3 @@
 {
-   static char buffer[20];
+   static wxChar buffer[20];
    wxString html;
@@ -85,3 +85,3 @@
 
-   html += "<font ";
+   html += _T("<font ");
 
@@ -89,4 +89,4 @@
    {
-      html +="color=";
-      sprintf(buffer,"\"#%02X%02X%02X\"", si->m_fg.Red(),si->m_fg.Green(),si->m_fg.Blue());
+      html += _T("color=");
+      wxSprintf(buffer,_T("\"#%02X%02X%02X\""), si->m_fg.Red(),si->m_fg.Green(),si->m_fg.Blue());
       html += buffer;
@@ -96,4 +96,4 @@
    {
-      html += " bgcolor=";
-      sprintf(buffer,"\"#%02X%02X%02X\"", si->m_bg.Red(),si->m_bg.Green(),si->m_bg.Blue());
+      html += _T(" bgcolor=");
+      wxSprintf(buffer,_T("\"#%02X%02X%02X\""), si->m_bg.Red(),si->m_bg.Green(),si->m_bg.Blue());
       html += buffer;
@@ -105,7 +105,7 @@
    case wxMODERN:
-      html += " face=\"Arial,Helvetica\""; break;
+      html += _T(" face=\"Arial,Helvetica\""); break;
    case wxROMAN:
-      html += " face=\"Times New Roman, Times\""; break;
+      html += _T(" face=\"Times New Roman, Times\""); break;
    case wxTELETYPE:
-      html += " face=\"Courier New, Courier\""; break;
+      html += _T(" face=\"Courier New, Courier\""); break;
    default:
@@ -125,16 +125,16 @@
    }
-   html += "size=";
-   sprintf(buffer,"%+1d", sizecount);
+   html += _T("size=");
+   wxSprintf(buffer,_T("%+1d"), sizecount);
    html += buffer;
 
-   html +=">";
+   html += _T(">");
 
    if(styleInfo != NULL && ! firstTime)
-      html ="</font>"+html; // terminate any previous font command
+      html = _T("</font>")+html; // terminate any previous font command
 
    if((si->weight == wxBOLD) && ( (!styleInfo) || (styleInfo->weight != wxBOLD)))
-      html += "<b>";
+      html += _T("<b>");
    else
       if(si->weight != wxBOLD && ( styleInfo && (styleInfo->weight == wxBOLD)))
-         html += "</b>";
+         html += _T("</b>");
 
@@ -144,11 +144,11 @@
    if((si->style == wxITALIC) && ( (!styleInfo) || (styleInfo->style != wxITALIC)))
-      html += "<i>";
+      html += _T("<i>");
    else
       if(si->style != wxITALIC && ( styleInfo && (styleInfo->style == wxITALIC)))
-         html += "</i>";
+         html += _T("</i>");
 
    if(si->underline && ( (!styleInfo) || ! styleInfo->underline))
-      html += "<u>";
+      html += _T("<u>");
    else if(si->underline == false && ( styleInfo && styleInfo->underline))
-      html += "</u>";
+      html += _T("</u>");
 
@@ -225,5 +225,5 @@
          if(mode & WXLO_EXPORT_AS_HTML)
-            *str += "<br>";
+            *str += _T("<br>");
          if(flags & WXLO_EXPORT_WITH_CRLF)
-            *str += "\r\n";
+            *str += _T("\r\n");
          else
diff -bu1 -r wxWindows/samples/richedit/wxlwindow.cpp wxWin/samples/richedit/wxlwindow.cpp
--- wxWindows/samples/richedit/wxlwindow.cpp	2002-03-27 18:44:50.000000000 +0100
+++ wxWin/samples/richedit/wxlwindow.cpp	2003-08-26 20:09:53.000000000 +0200
@@ -216,3 +216,3 @@
    SetBackgroundColour(GetLayoutList()->GetDefaultStyleInfo().GetBGColour());
-   wxScrolledWindow::Clear();
+   wxScrolledWindow::ClearBackground();
    ResizeScrollbars(true);
@@ -336,3 +336,3 @@
          {
-            m_StatusBar->SetStatusText("", m_StatusFieldLabel);
+            m_StatusBar->SetStatusText(wxEmptyString, m_StatusFieldLabel);
          }
@@ -578,3 +578,3 @@
          case 's': // search
-            Find("");
+            Find(wxEmptyString);
             break;
@@ -626,3 +626,3 @@
             case 's': // search
-               Find("");
+               Find(wxEmptyString);
                break;
@@ -848,3 +848,9 @@
 void
-wxLayoutWindow::RequestUpdate(const wxRect *updateRect)
+wxLayoutWindow::RequestUpdate(const wxRect *
+                                            #ifdef __WXGTK__
+                                            updateRect
+                                            #else
+                                            WXUNUSED(updateRect)
+                                            #endif
+                                            )
 {
@@ -1265,5 +1271,23 @@
 bool
-wxLayoutWindow::Find(const wxString &needle,
-                     wxPoint * fromWhere,
-                     const wxString &configPath)
+wxLayoutWindow::Find(const wxString &
+                                     #ifdef M_BASEDIR
+                                     needle
+                                     #else
+                                     WXUNUSED(needle)
+                                     #endif
+                                     ,
+                     wxPoint * 
+                                     #ifdef M_BASEDIR
+                                     fromWhere
+                                     #else
+                                     WXUNUSED(fromWhere)
+                                     #endif
+                                     ,
+                     const wxString &
+                                     #ifdef M_BASEDIR
+                                     configPath
+                                     #else
+                                     WXUNUSED(configPath)
+                                     #endif
+                                     )
 {
diff -bu1 -r wxWindows/samples/richedit/wxlwindow.h wxWin/samples/richedit/wxlwindow.h
--- wxWindows/samples/richedit/wxlwindow.h	2000-07-15 21:50:21.000000000 +0200
+++ wxWin/samples/richedit/wxlwindow.h	2003-08-26 17:29:51.000000000 +0200
@@ -109,3 +109,3 @@
              wxPoint * fromWhere = NULL,
-             const wxString &configPath = "MsgViewFindString");
+             const wxString &configPath = _T("MsgViewFindString"));
    /// find the same string again

 	  	 
