diff --git a/samples/minimal/minimal.cpp b/samples/minimal/minimal.cpp
index 076d88b..da986ee 100644
--- a/samples/minimal/minimal.cpp
+++ b/samples/minimal/minimal.cpp
@@ -66,6 +66,7 @@ public:
     // event handlers (these functions should _not_ be virtual)
     void OnQuit(wxCommandEvent& event);
     void OnAbout(wxCommandEvent& event);
+    void OnPaint(wxPaintEvent &event);
 
 private:
     // any class wishing to process wxWidgets events must use this macro
@@ -98,6 +99,7 @@ enum
 wxBEGIN_EVENT_TABLE(MyFrame, wxFrame)
     EVT_MENU(Minimal_Quit,  MyFrame::OnQuit)
     EVT_MENU(Minimal_About, MyFrame::OnAbout)
+    EVT_PAINT(MyFrame::OnPaint)
 wxEND_EVENT_TABLE()
 
 // Create a new application object: this macro will allow wxWidgets to create
@@ -197,3 +199,18 @@ void MyFrame::OnAbout(wxCommandEvent& WXUNUSED(event))
                  wxOK | wxICON_INFORMATION,
                  this);
 }
+
+void MyFrame::OnPaint(wxPaintEvent &event)
+{
+    wxPaintDC dc( this );
+    wxSize size = GetClientSize();
+    dc.SetAxisOrientation( true, true );
+    dc.SetDeviceOrigin( 0, size.y - 1 );
+    dc.SetPen( wxPen( wxColour( 255, 0, 0 ) ) );
+    dc.DrawRectangle( 0, 0, size.x, size.y );
+    dc.SetPen( wxPen( wxColour( 0, 255, 0 ) ) );
+    dc.DrawLine( 0, 0, 0, size.y );
+    dc.DrawLine( size.x - 1, 0, size.x - 1, size.y );
+    dc.DrawLine( 0, 0, size.x, 0 );
+    dc.DrawLine( 0, size.y - 1, size.x, size.y - 1 );
+}
