diff --git a/auidemo_wxMsw315.cpp b/auidemo.cpp
index 666e8db..f0a5930 100644
--- a/auidemo_wxMsw315.cpp
+++ b/auidemo.cpp
@@ -82,6 +82,8 @@ class MyFrame : public wxFrame
         ID_HorizontalGradient,
         ID_LiveUpdate,
         ID_AllowToolbarResizing,
+        ID_ShowHideToolbarLabels,
+        ID_Toolbar2,
         ID_Settings,
         ID_CustomizeToolbar,
         ID_DropDownToolbarItem,
@@ -155,6 +157,7 @@ private:
 
     void OnGradient(wxCommandEvent& evt);
     void OnToolbarResizing(wxCommandEvent& evt);
+    void OnToolbarLabels(wxCommandEvent& evt);
     void OnManagerFlag(wxCommandEvent& evt);
     void OnNotebookFlag(wxCommandEvent& evt);
     void OnUpdateUI(wxUpdateUIEvent& evt);
@@ -610,6 +613,7 @@ wxBEGIN_EVENT_TABLE(MyFrame, wxFrame)
     EVT_MENU(ID_VerticalGradient, MyFrame::OnGradient)
     EVT_MENU(ID_HorizontalGradient, MyFrame::OnGradient)
     EVT_MENU(ID_AllowToolbarResizing, MyFrame::OnToolbarResizing)
+    EVT_MENU(ID_ShowHideToolbarLabels, MyFrame::OnToolbarLabels)
     EVT_MENU(ID_Settings, MyFrame::OnSettings)
     EVT_MENU(ID_CustomizeToolbar, MyFrame::OnCustomizeToolbar)
     EVT_MENU(ID_GridContent, MyFrame::OnChangeContentPane)
@@ -713,6 +717,7 @@ MyFrame::MyFrame(wxWindow* parent,
     options_menu->AppendRadioItem(ID_HorizontalGradient, _("Horizontal Caption Gradient"));
     options_menu->AppendSeparator();
     options_menu->AppendCheckItem(ID_AllowToolbarResizing, _("Allow Toolbar Resizing"));
+    options_menu->AppendCheckItem(ID_ShowHideToolbarLabels, _("Show/Hide Toolbar Labels"));
     options_menu->AppendSeparator();
     options_menu->Append(ID_Settings, _("Settings Pane"));
 
@@ -792,7 +797,7 @@ MyFrame::MyFrame(wxWindow* parent,
     tb1->Realize();
 
 
-    wxAuiToolBar* tb2 = new wxAuiToolBar(this, wxID_ANY, wxDefaultPosition, wxDefaultSize,
+    wxAuiToolBar* tb2 = new wxAuiToolBar(this, ID_Toolbar2, wxDefaultPosition, wxDefaultSize,
                                          wxAUI_TB_DEFAULT_STYLE | wxAUI_TB_OVERFLOW | wxAUI_TB_HORIZONTAL);
     tb2->SetToolBitmapSize(FromDIP(wxSize(16,16)));
 
@@ -1078,6 +1083,31 @@ void MyFrame::OnToolbarResizing(wxCommandEvent& WXUNUSED(evt))
     m_mgr.Update();
 }
 
+void MyFrame::OnToolbarLabels(wxCommandEvent& evt)
+{
+    auto window = wxWindow::FindWindowById(ID_Toolbar2);
+
+    if (window)
+    {
+        auto toolbar = wxDynamicCast(window, wxAuiToolBar);
+
+        if (toolbar)
+        {
+            if (evt.IsChecked())
+            {
+                toolbar->SetWindowStyleFlag(toolbar->GetWindowStyleFlag() | wxAUI_TB_TEXT);
+            }
+            else
+            {
+                toolbar->SetWindowStyleFlag(toolbar->GetWindowStyleFlag() & ~wxAUI_TB_TEXT);
+            }
+
+            toolbar->Realize();
+            m_mgr.Update();
+        }
+    }
+}
+
 void MyFrame::OnManagerFlag(wxCommandEvent& event)
 {
     unsigned int flag = 0;
