diff --git "a/C:\\Users\\Petr\\AppData\\Local\\Temp\\TortoiseGit\\wra56FB.tmp\\wrapgdip-5420f96-left.h" "b/L:\\Dev\\Desktop\\wxWidgets-GIT\\include\\wx\\msw\\wrapgdip.h"
index 04d4009..e78e97e 100644
--- "a/C:\\Users\\Petr\\AppData\\Local\\Temp\\TortoiseGit\\wra56FB.tmp\\wrapgdip-5420f96-left.h"
+++ "b/L:\\Dev\\Desktop\\wxWidgets-GIT\\include\\wx\\msw\\wrapgdip.h"
@@ -24,7 +24,17 @@
     #define min(a,b)            (((a) < (b)) ? (a) : (b))
 #endif
 
+#ifdef __VISUALC__
+    #pragma warning(push)
+    #pragma warning(disable:4458) // declaration of 'xxx' hides class member
+#endif
+
 #include <gdiplus.h>
+
+#ifdef __VISUALC__
+    #pragma warning(pop)
+#endif
+
 using namespace Gdiplus;
 
 #endif // _WX_MSW_WRAPGDIP_H_
