--- minimal-orig.cpp	2008-12-10 16:16:10.000000000 -0500
+++ minimal.cpp	2009-09-16 12:59:02.494888500 -0400
@@ -41,9 +41,6 @@
     #include "../sample.xpm"
 #endif
 
-// the aero glass include
-#include <dwmapi.h>
-
 // ----------------------------------------------------------------------------
 // private classes
 // ----------------------------------------------------------------------------
@@ -67,8 +64,6 @@
 public:
     // ctor(s)
     MyFrame(const wxString& title);
-	void OnErase(wxEraseEvent& event);
-
 };
 
 // ----------------------------------------------------------------------------
@@ -98,6 +93,8 @@
     if ( !wxApp::OnInit() )
         return false;
 
+	wxInitAllImageHandlers();
+
     // create the main application window
     MyFrame *frame = new MyFrame(_T("Minimal wxWidgets App"));
 
@@ -122,40 +119,37 @@
     // set the frame icon
     SetIcon(wxICON(sample));
 
-	MARGINS margins;
-	margins.cyTopHeight = -1;
-	margins.cyBottomHeight = -1;
-	margins.cxLeftWidth = -1;
-	margins.cxRightWidth = -1;
-		
-		// Expand Glass Area
-	HRESULT hr = ::DwmExtendFrameIntoClientArea((HWND)GetHWND(), &margins);
-
-	Connect(wxEVT_ERASE_BACKGROUND, wxEraseEventHandler(MyFrame::OnErase));
-}
-
-
-// event handlers
-
-void MyFrame::OnErase(wxEraseEvent& event)
-{
-	wxDC* dc = event.GetDC();
-	wxBufferedDC bdc(dc);
-	wxGCDC gdc(bdc);
-
-	gdc.SetBackground(wxBrush(wxColour(0, 255, 0, wxALPHA_OPAQUE)));
-	gdc.Clear();
-
-	gdc.SetBrush(wxBrush(wxColour(0, 255, 0, wxALPHA_OPAQUE)));
-	gdc.DrawRectangle(0, 0, 100, 100);
-
-	wxFont font(11, wxFONTFAMILY_SWISS, wxFONTSTYLE_NORMAL, wxFONTWEIGHT_NORMAL, false, _T("Segoe UI"));
-	gdc.SetFont(font);
-
-	// uncomment this to get an opaque text
-	//wxFont font2(11, wxFONTFAMILY_SWISS, wxFONTSTYLE_NORMAL, wxFONTWEIGHT_BOLD, false, _T("Chiller"));
-	//gdc.SetFont(font2);
-
-	gdc.SetTextForeground(wxColour(0, 0, 0, wxALPHA_OPAQUE));
-	gdc.DrawText(_("wxWidgets"), 120, 20);
+	// Sample code to demonstrate the problem
+	wxFont font = wxSystemSettings::GetFont(wxSYS_DEFAULT_GUI_FONT);
+	wxBitmap bitmap(200, 120, 32);
+	bitmap.UseAlpha();
+
+	wxMemoryDC memDC(bitmap);
+	wxGCDC dc(memDC); 	
+
+	// Clear background
+	dc.SetBackground(*wxTRANSPARENT_BRUSH); 
+	dc.Clear();
+
+	// Draw some text
+	font.SetPointSize(12);
+	font.SetWeight(wxBOLD);
+	dc.SetFont(font);
+	dc.SetTextForeground(*wxBLACK);
+
+	// Uncomment the following line to make the text in the image draw correctly,
+	// but with an opaque background
+	// dc.SetBackgroundMode( wxSOLID );
+
+	dc.DrawText(_("wxWidgets"), 2, 2);
+	dc.DrawText(_("is"), 2, 20);
+	dc.DrawText(_("cool"), 2, 35);
+
+	memDC.SelectObject( wxNullBitmap ); 
+	wxFileDialog *fd = new wxFileDialog(this, _("Where to save the image?"), wxEmptyString, _("test.png"), _("PNG files (*.png)|*.png"), wxSAVE | wxOVERWRITE_PROMPT);
+	if (fd->ShowModal() == wxID_OK)
+	{
+		bitmap.SaveFile(fd->GetPath(), wxBITMAP_TYPE_PNG);
+	}
+	delete fd;
 }
