Index: src/msw/toplevel.cpp
===================================================================
--- src/msw/toplevel.cpp	(revision 62506)
+++ src/msw/toplevel.cpp	(working copy)
@@ -661,10 +661,12 @@
         frame->GetMenuBar()->AddAdornments(GetWindowStyleFlag());
 #endif
 
+#if wxUSE_DEFERRED_SIZING
     // we only set pending size if we're maximized before being shown, now that
     // we're shown we don't need it any more (it is reset in size event handler
     // for child windows but we have to do it ourselves for this parent window)
     m_pendingSize = wxDefaultSize;
+#endif // wxUSE_DEFERRED_SIZING
 
     return true;
 }
@@ -686,6 +688,7 @@
         // so just remember that we should do it later in this case
         m_maximizeOnShow = maximize;
 
+#if wxUSE_DEFERRED_SIZING
         // after calling Maximize() the client code expects to get the frame
         // "real" size and doesn't want to know that, because of implementation
         // details, the frame isn't really maximized yet but will be only once
@@ -703,6 +706,7 @@
             m_pendingSize = wxGetClientDisplayRect().GetSize();
         }
         //else: can't do anything in this case, we don't have the old size
+#endif // wxUSE_DEFERRED_SIZING
     }
 }
 
Index: src/msw/window.cpp
===================================================================
--- src/msw/window.cpp	(revision 62506)
+++ src/msw/window.cpp	(working copy)
@@ -1831,11 +1831,13 @@
     wxWindow * const parent = GetParent();
 
     wxPoint pos;
+#if wxUSE_DEFERRED_SIZING
     if ( m_pendingPosition != wxDefaultPosition )
     {
         pos = m_pendingPosition;
     }
     else // use current position
+#endif // wxUSE_DEFERRED_SIZING
     {
         RECT rect = wxGetWindowRect(GetHwnd());
 
