cvs server: Diffing demos/bombs
Index: demos/bombs/bombs.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/demos/bombs/bombs.cpp,v
retrieving revision 1.9
diff -u -r1.9 bombs.cpp
--- demos/bombs/bombs.cpp	2005/01/31 18:02:36	1.9
+++ demos/bombs/bombs.cpp	2005/04/27 00:29:49
@@ -61,6 +61,7 @@
 }
 
 BEGIN_EVENT_TABLE(BombsFrame, wxFrame)
+    EVT_MENU(bombsID_NEWGAME, BombsFrame::OnNewSameGame)
     EVT_MENU(bombsID_EASY, BombsFrame::OnNewEasyGame)
     EVT_MENU(bombsID_MEDIUM, BombsFrame::OnNewMediumGame)
     EVT_MENU(bombsID_HARD, BombsFrame::OnNewHardGame)
@@ -84,6 +85,7 @@
     wxMenuBar *menuBar = new wxMenuBar;
     wxMenu *menuFile = new wxMenu;
     wxMenu *menuLevel = new wxMenu;
+    menuLevel->AppendRadioItem(bombsID_NEWGAME, wxT("&Same level\tCtrl-N"));
     menuLevel->AppendRadioItem(bombsID_EASY, wxT("&Easy (10x10)\tCtrl-1"));
     menuLevel->AppendRadioItem(bombsID_MEDIUM, wxT("&Medium (15x15)\tCtrl-2"));
     menuLevel->AppendRadioItem(bombsID_HARD, wxT("&Hard (25x20)\tCtrl-3"));
@@ -125,8 +127,16 @@
 
 void BombsFrame::NewGame(int level, bool query)
 {
-    if(query)
+    static int last_level = bombsID_EASY;
+  
+    if (bombsID_NEWGAME == level) 
     {
+       level = last_level;
+    }
+    
+
+    if(query && level != last_level)
+    {
        int ok = wxMessageBox(
                   wxT("Start new game regardless previous board?"),
                   wxT("Confirm"),
@@ -170,6 +180,11 @@
     wxMessageBox(
         wxT("wxBombs (c) 1996 by P. Foggia\n<foggia@amalfi.dis.unina.it>"),
         wxT("About wxBombs") );
+}
+
+void BombsFrame::OnNewSameGame(wxCommandEvent& WXUNUSED(event))
+{
+    NewGame(bombsID_NEWGAME, true);
 }
 
 void BombsFrame::OnNewEasyGame(wxCommandEvent& WXUNUSED(event))
Index: demos/bombs/bombs.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/demos/bombs/bombs.h,v
retrieving revision 1.5
diff -u -r1.5 bombs.h
--- demos/bombs/bombs.h	2003/12/23 10:45:28	1.5
+++ demos/bombs/bombs.h	2005/04/27 00:29:49
@@ -45,6 +45,7 @@
 
 private:
 
+    void OnNewSameGame(wxCommandEvent& event);
     void OnNewEasyGame(wxCommandEvent& event);
     void OnNewMediumGame(wxCommandEvent& event);
     void OnNewHardGame(wxCommandEvent& event);
Index: demos/bombs/game.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/demos/bombs/game.cpp,v
retrieving revision 1.2
diff -u -r1.2 game.cpp
--- demos/bombs/game.cpp	2003/12/19 01:34:40	1.2
+++ demos/bombs/game.cpp	2005/04/27 00:29:49
@@ -75,6 +75,12 @@
         }
     }
 
+    /* Force (0,0) not to have a bomb for those that don't want to have
+       to guess on the first move. Better would be to for the MS rule that
+       whatever is picked first isn't a bomb.
+     */
+    m_field[0] = BG_HIDDEN;
+
     m_numBombCells = 0;
     for(x=0; x<m_width; x++)
         for(y=0; y<m_height; y++)

 	  	 
