--- src\generic\grid.cpp	2008-09-15 11:09:56.000000000 +0200
+++ src\generic\new	2009-10-24 19:14:48.468750000 +0200
@@ -5526,8 +5526,8 @@
         }
         else
         {
-            // adjust row height depending on label text
-            AutoSizeRowLabelSize( row );
+            // adjust row height depending on largest row
+            AutoSizeRow(row, false);
 
             ChangeCursorMode(WXGRID_CURSOR_SELECT_CELL, m_colLabelWin);
             m_dragLastPos = -1;
@@ -5820,8 +5820,8 @@
         }
         else
         {
-            // adjust column width depending on label text
-            AutoSizeColLabelSize( col );
+            // adjust column width depending on widest column
+            AutoSizeColumn(col, false);
 
             ChangeCursorMode(WXGRID_CURSOR_SELECT_CELL, m_colLabelWin);
             m_dragLastPos = -1;
