diff -u -r d:/wx/wxWidgets/src/msw/menu.cpp d:/wx/wx_qsp/src/msw/menu.cpp
--- d:/wx/wxWidgets/src/msw/menu.cpp	2009-01-27 21:27:51 +0000
+++ d:/wx/wx_qsp/src/msw/menu.cpp	2009-01-30 07:14:58 +0000
@@ -449,10 +449,19 @@
     if ( pItem->IsOwnerDrawn() )
     {
         // is the item owner-drawn just because of the [checked] bitmap?
+        // HzD_Byte's hack
+        /*
         if ( (pItem->GetBitmap(false).Ok() || pItem->GetBitmap(true).Ok()) &&
                 !pItem->GetTextColour().Ok() &&
                     !pItem->GetBackgroundColour().Ok() &&
                         !pItem->GetFont().Ok() )
+        */
+        const wxBitmap& bmp1 = pItem->GetBitmap(false);
+        const wxBitmap& bmp2 = pItem->GetBitmap(true);
+        if (((bmp1.Ok() && bmp1.GetWidth() == 16 && bmp1.GetHeight() == 16) ||
+             (bmp2.Ok() && bmp2.GetWidth() == 16 && bmp2.GetHeight() == 16)) &&
+             !pItem->GetTextColour().Ok() && !pItem->GetBackgroundColour().Ok() &&
+             !pItem->GetFont().Ok())
         {
             // try to use InsertMenuItem() as it's guaranteed to look correct
             // while our owner-drawn code is not
diff -u -r d:/wx/wxWidgets/src/msw/menuitem.cpp d:/wx/wx_qsp/src/msw/menuitem.cpp
--- d:/wx/wxWidgets/src/msw/menuitem.cpp	2008-11-03 13:30:53 +0000
+++ d:/wx/wx_qsp/src/msw/menuitem.cpp	2009-01-30 07:33:38 +0000
@@ -374,12 +374,15 @@
     }
     else
     {
+        // HzD_Byte's hack
+        /*
         if ( IsSubMenu() )
         {
             // high byte contains the number of items in a submenu for submenus
             flagsOld &= 0xFF;
             flagsOld |= MF_POPUP;
         }
+        */
 
         LPCTSTR data;
 
@@ -396,6 +399,8 @@
             data = (wxChar*) m_text.wx_str();
         }
 
+        // HzD_Byte's hack
+        /*
 #ifdef __WXWINCE__
         // FIXME: complete this, applying the old
         // flags.
@@ -421,6 +426,26 @@
             wxLogLastError(wxT("ModifyMenu"));
         }
 #endif
+        */
+        MENUITEMINFO info;
+        wxZeroMemory(info);
+        info.cbSize = sizeof(info);
+        info.fMask = MIIM_BITMAP | MIIM_STATE | MIIM_ID | MIIM_SUBMENU |
+                     MIIM_CHECKMARKS | MIIM_DATA | MIIM_FTYPE;
+        if ( !::GetMenuItemInfo(hMenu, id, FALSE, &info) )
+        {
+            wxLogLastError(wxT("GetMenuItemInfo"));
+        }
+        else
+        {
+            info.fMask |= MIIM_STRING;
+            info.cch = m_text.length();
+            info.dwTypeData = (LPTSTR)data;
+            if ( !::SetMenuItemInfo(hMenu, id, FALSE, &info) )
+            {
+                wxLogLastError(wxT("SetMenuItemInfo"));
+            }
+        }
     }
 }
 
diff -u -r d:/wx/wxWidgets/src/msw/ownerdrw.cpp d:/wx/wx_qsp/src/msw/ownerdrw.cpp
--- d:/wx/wxWidgets/src/msw/ownerdrw.cpp	2009-01-27 21:27:51 +0000
+++ d:/wx/wx_qsp/src/msw/ownerdrw.cpp	2009-01-30 06:46:58 +0000
@@ -243,7 +243,8 @@
 
         // add space at the end of the menu for the submenu expansion arrow
         // this will also allow offsetting the accel string from the right edge
-        *pwidth += GetMarginWidth() + 16;
+        // HzD_Byte's hack
+        // *pwidth += GetMarginWidth() + 16;
     }
     else // don't draw the text, just the bitmap (if any)
     {
@@ -274,6 +275,9 @@
 
     // add a 4-pixel separator, otherwise menus look cluttered
     *pwidth += 4;
+    // HzD_Byte's hack
+    *pheight += 4;
+    if (IsOwnerDrawn()) *pwidth += GetMarginWidth() + 16;
 
     // make sure that this item is at least as tall as the system menu height
     const size_t heightStd = wxMSWSystemMenuFontModule::GetSystemMenuHeight();