Index: include/wx/msw/tooltip.h
===================================================================
--- include/wx/msw/tooltip.h	(revision 76535)
+++ include/wx/msw/tooltip.h	(working copy)
@@ -69,6 +69,7 @@
     // makes sense to use it for tooltips associated with a single window only.
     void SetRect(const wxRect& rc);
 
+    static void HideToolTipCtrl();
 private:
     // Adds a window other than our main m_window to this tooltip.
     void DoAddHWND(WXHWND hWnd);
Index: src/msw/tooltip.cpp
===================================================================
--- src/msw/tooltip.cpp	(revision 76535)
+++ src/msw/tooltip.cpp	(working copy)
@@ -299,6 +299,13 @@
 }
 
 /* static */
+void wxToolTip::HideToolTipCtrl()
+{
+    if (ms_hwndTT && ::IsWindowVisible(ms_hwndTT))
+        ::ShowWindow(ms_hwndTT, SW_HIDE); // Hide tooltip
+}
+
+/* static */
 void wxToolTip::RelayEvent(WXMSG *msg)
 {
     (void)SendTooltipMessage(GetToolTipCtrl(), TTM_RELAYEVENT, msg);
Index: src/msw/toplevel.cpp
===================================================================
--- src/msw/toplevel.cpp	(revision 76535)
+++ src/msw/toplevel.cpp	(working copy)
@@ -708,6 +708,9 @@
 
 void wxTopLevelWindowMSW::DoShowWindow(int nShowCmd)
 {
+    if (nShowCmd == SW_HIDE)
+        wxToolTip::HideToolTipCtrl();
+
     ::ShowWindow(GetHwnd(), nShowCmd);
 
     // Hiding the window doesn't change its iconized state.
