Index: minimal.cpp
===================================================================
--- minimal.cpp	(revision 61425)
+++ minimal.cpp	(working copy)
@@ -16,10 +16,10 @@
 // ----------------------------------------------------------------------------
 // headers
 // ----------------------------------------------------------------------------
- 
+
 // For compilers that support precompilation, includes "wx/wx.h".
 #include "wx/wxprec.h"
- 
+
 #ifdef __BORLANDC__
     #pragma hdrstop
 #endif
@@ -29,6 +29,9 @@
 #ifndef WX_PRECOMP
     #include "wx/wx.h"
 #endif
+#include <commctrl.h>
+#include <wx/imaglist.h>
+#include <wx/msw/private.h>
 
 // ----------------------------------------------------------------------------
 // resources
@@ -129,8 +132,29 @@
 
     // and show it (the frames, unlike simple controls, are not shown when
     // created initially)
+    frame->SetBackgroundColour(*wxWHITE);
     frame->Show(true);
-
+    wxImage::AddHandler(new wxPNGHandler);
+    wxImage Image("alpha.png", wxBITMAP_TYPE_PNG);
+    wxBitmap Bitmap(Image);
+    wxClientDC DC(frame);
+    // draw directly
+    DC.DrawBitmap(Bitmap, 0, 0);
+    wxImageList List(32, 32);
+    // draw from image list
+    List.Add(Bitmap);
+    List.Draw(0, DC, 0, 32);
+    // draw from Windows image list (using modified wxImageList code)
+    int dd = wxDisplayDepth(), flags = ILC_COLORDDB;
+    if (dd <= 4)       flags |= ILC_COLOR;   // 16 color
+    else if (dd <= 8)  flags |= ILC_COLOR8;  // 256 color
+    else if (dd <= 16) flags |= ILC_COLOR16; // 64k hi-color
+    else if (dd <= 24) flags |= ILC_COLOR24; // 16m truecolor
+    else if (dd <= 32) flags |= ILC_COLOR32; // 16m truecolor
+    HIMAGELIST Handle = ImageList_Create(32, 32, flags, 1, 1);
+    ImageList_Add(Handle, GetHbitmapOf(Bitmap), GetHbitmapOf(wxNullBitmap));
+    ImageList_Draw(Handle, 0, GetHdcOf(DC), 0, 64, ILD_NORMAL);
+    ImageList_Destroy(Handle);
     // success: wxApp::OnRun() will be called which will enter the main message
     // loop and the application will run. If we returned false here, the
     // application would exit immediately.
