diff --git a/samples/minimal/minimal.cpp b/samples/minimal/minimal.cpp
index f765a70..074c3b7 100644
--- a/samples/minimal/minimal.cpp
+++ b/samples/minimal/minimal.cpp
@@ -67,7 +67,10 @@ public:
     void OnQuit(wxCommandEvent& event);
     void OnAbout(wxCommandEvent& event);
 
+    void OnCheckBox(wxCommandEvent& event);
+
 private:
+    wxRadioBox* m_radBox;
     // any class wishing to process wxWidgets events must use this macro
     wxDECLARE_EVENT_TABLE();
 };
@@ -164,6 +167,49 @@ MyFrame::MyFrame(const wxString& title)
 
     // ... and attach this menu bar to the frame
     SetMenuBar(menuBar);
+
+    wxPanel* panel = new wxPanel(this, wxID_ANY);
+    wxSizer* mainSizer = new wxBoxSizer(wxVERTICAL);
+    panel->SetSizer(mainSizer);
+
+    wxCheckBox* chkBox = new wxCheckBox(panel, wxID_ANY, "Show/hide radio buttons");
+    chkBox->SetValue(false); // if true, wxRadioBox will allocate enough space for its buttons
+                             // regardless of the selected font.
+                             // if false, wxRadioBox seems to use the default font to calculate
+                             // its initial size.
+    chkBox->Bind(wxEVT_CHECKBOX, &MyFrame::OnCheckBox, this);
+    mainSizer->Add(chkBox, wxSizerFlags().Center());
+    mainSizer->AddSpacer(10);
+
+    wxArrayString options;
+    options.Add("Option 1");
+    options.Add("Option 2");
+    options.Add("Option 3");
+    options.Add("Option 4");
+    options.Add("Option 5");
+
+    m_radBox = new wxRadioBox(panel, wxID_ANY, "Options:",
+                              wxDefaultPosition, wxDefaultSize,
+                              options, 1, wxRA_SPECIFY_COLS);
+
+    // it seems that wxRadioBox does not use the selected font to calculate its initial size!
+    // try commenting this out and everything will be alright.
+    m_radBox->SetFont( wxFontInfo(12).FaceName("Calibri").Family(wxFONTFAMILY_SWISS) );
+
+    m_radBox->Show(3, chkBox->IsChecked()); // might not be shown at all later (gtk3) or might be shown outside the radiobox (gtk2)
+    m_radBox->Show(4, chkBox->IsChecked()); // might not be shown at all later (gtk3) or might be shown outside the radiobox (gtk2)
+
+    mainSizer->Add(m_radBox, wxSizerFlags().Center());
+
+    mainSizer->Layout();
+
+    mainSizer->AddStretchSpacer();
+
+    wxSizer* topSizer = new wxBoxSizer(wxVERTICAL);
+    topSizer->Add(panel, wxSizerFlags(1).Expand());
+    SetSizer(topSizer);
+    Layout();
+
 #else // !wxUSE_MENUS
     // If menus are not available add a button to access the about box
     wxSizer* sizer = new wxBoxSizer(wxHORIZONTAL);
@@ -203,3 +249,13 @@ void MyFrame::OnAbout(wxCommandEvent& WXUNUSED(event))
                  wxOK | wxICON_INFORMATION,
                  this);
 }
+
+void MyFrame::OnCheckBox(wxCommandEvent& event)
+{
+    const bool show = event.IsChecked();
+
+    m_radBox->Show(3, show);
+    m_radBox->Show(4, show);
+
+    GetSizer()->Layout();
+}
