Index: src/common/datetime.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/common/datetime.cpp,v
retrieving revision 1.111
diff -u -r1.111 datetime.cpp
--- src/common/datetime.cpp	2005/01/17 21:29:21	1.111
+++ src/common/datetime.cpp	2005/01/23 12:22:36
@@ -1204,16 +1204,21 @@
                       _T("Invalid time in wxDateTime::Set()") );
 
     // get the current date from system
-    struct tm *tm = GetTmNow();
+    struct tm tm = *GetTmNow();
 
-    wxDATETIME_CHECK( tm, _T("localtime() failed") );
+    //wxDATETIME_CHECK( tm, _T("localtime() failed") );
 
     // adjust the time
-    tm->tm_hour = hour;
-    tm->tm_min = minute;
-    tm->tm_sec = second;
+    tm.tm_hour = hour;
+    tm.tm_min = minute;
+    tm.tm_sec = second;
 
-    (void)Set(*tm);
+		struct tm tm2(tm);
+		mktime(&tm2);
+		if(tm2.tm_isdst != tm.tm_isdst)
+			tm.tm_isdst = tm2.tm_isdst;
+		
+		(void)Set(tm);
 
     // and finally adjust milliseconds
     return SetMillisecond(millisec);

 	  	 
