Index: include/wx/defs.h
===================================================================
--- include/wx/defs.h	(revision 76178)
+++ include/wx/defs.h	(working copy)
@@ -2878,12 +2878,14 @@
 /* Print quality.
  */
 
-#define wxPRINT_QUALITY_HIGH    -1
-#define wxPRINT_QUALITY_MEDIUM  -2
-#define wxPRINT_QUALITY_LOW     -3
-#define wxPRINT_QUALITY_DRAFT   -4
+enum wxPrintQuality
+{
+    wxPRINT_QUALITY_HIGH = -1,
+    wxPRINT_QUALITY_MEDIUM = -2,
+    wxPRINT_QUALITY_LOW = -3,
+    wxPRINT_QUALITY_DRAFT = -4
+};
 
-typedef int wxPrintQuality;
 
 /* Print mode (currently PostScript only)
  */
  
Index: src/msw/printdlg.cpp
===================================================================
--- src/msw/printdlg.cpp	(revision 76178)
+++ src/msw/printdlg.cpp	(working copy)
@@ -348,7 +348,7 @@
                 // TODO: if the printer fills in the resolution in DPI, how
                 // will the application know if it's high, low, draft etc.??
                 //                    wxFAIL_MSG("Warning: DM_PRINTQUALITY was not one of the standard values.");
-                data.SetQuality( devMode->dmPrintQuality );
+				data.SetQuality(static_cast<wxPrintQuality>(devMode->dmPrintQuality));
                 break;
 
             }
