--- old/window.cpp	Mon Oct 14 10:11:08 2002
+++ new/window.cpp	Mon Oct 14 10:11:17 2002
@@ -1303,6 +1303,12 @@
     event.m_leftDown = (gdk_event->state & GDK_BUTTON1_MASK);                 \
     event.m_middleDown = (gdk_event->state & GDK_BUTTON2_MASK);               \
     event.m_rightDown = (gdk_event->state & GDK_BUTTON3_MASK);                \
+    if (event.GetEventType()==wxEVT_MOUSEWHEEL) {                             \
+      if (((GdkEventButton *)gdk_event)->button==4)                           \
+       event.m_wheelRotation=120;                                             \
+      else if (((GdkEventButton *)gdk_event)->button==5)                      \
+       event.m_wheelRotation=-120;                                            \
+    }                                                                         \
                                                                               \
     wxPoint pt = win->GetClientAreaOrigin();                                  \
     event.m_x = (wxCoord)gdk_event->x - pt.x;                                 \
@@ -1476,6 +1482,22 @@
             default:  break;
         }
     }
+    else if (gdk_event->button == 4)
+    {
+        switch (gdk_event->type)
+        {
+            case GDK_BUTTON_PRESS: event_type = wxEVT_MOUSEWHEEL; break;
+            default:  break;
+        }
+    }
+    else if (gdk_event->button == 5)
+    {
+        switch (gdk_event->type)
+        {
+            case GDK_BUTTON_PRESS: event_type = wxEVT_MOUSEWHEEL; break;
+            default:  break;
+        }
+    }
 
     if ( event_type == wxEVT_NULL )
     {

 	  	 
