Index: include/wx/stdstream.h
===================================================================
--- include/wx/stdstream.h	(revision 70045)
+++ include/wx/stdstream.h	(working copy)
@@ -43,6 +43,13 @@
                                        std::ios_base::out);
     virtual std::streamsize showmanyc();
     virtual std::streamsize xsgetn(char *s, std::streamsize n);
+#if defined(__VISUALC8__) || defined(__VISUALC9__)
+    virtual std::streamsize __CLR_OR_THIS_CALL _Xsgetn_s(char *s,
+                                    size_t WXUNUSED(size), std::streamsize n)
+    {
+        return xsgetn(s, n);
+    }
+#endif
     virtual int underflow();
     virtual int uflow();
     virtual int pbackfail(int c = EOF);
