Index: include/wx/tbarbase.h
===================================================================
--- include/wx/tbarbase.h	(revision 70579)
+++ include/wx/tbarbase.h	(working copy)
@@ -118,6 +118,10 @@
 
     wxToolBarBase *GetToolBar() const { return m_tbar; }
 
+    // Flag of displaying of a label on this tool button
+    bool ShowLabel() const { return m_showLabel; }
+    void ShowLabel(bool value) { m_showLabel = value; }
+
     // style/kind
     bool IsStretchable() const { return m_stretchable; }
     bool IsButton() const { return m_toolStyle == wxTOOL_STYLE_BUTTON; }
@@ -227,7 +231,7 @@
 #if wxUSE_MENUS
         m_dropdownMenu = NULL;
 #endif
-
+        m_showLabel = true;
     }
 
     wxToolBarBase *m_tbar;  // the toolbar to which we belong (may be NULL)
@@ -244,6 +248,9 @@
         wxControl        *m_control;
     };
 
+    // true if label should be displayed 
+    bool m_showLabel;
+
     // true if this tool is stretchable: currently is only value for separators
     bool m_stretchable;
 
@@ -407,6 +414,9 @@
     virtual void EnableTool(int toolid, bool enable);
     virtual void ToggleTool(int toolid, bool toggle);
 
+    // Setting a flag of displaying of a label on tool button
+    virtual void ShowToolLabel(int toolid, bool show);
+
     // Set this to be togglable (or not)
     virtual void SetToggle(int toolid, bool toggle);
 
Index: include/wx/toolbar.h
===================================================================
--- include/wx/toolbar.h	(revision 70579)
+++ include/wx/toolbar.h	(working copy)
@@ -60,7 +60,10 @@
     wxTB_BOTTOM       = 0x2000,
 
     // lay out toolbar at the right edge of the window
-    wxTB_RIGHT        = 0x4000
+    wxTB_RIGHT        = 0x4000,
+
+    // Allow mixed buttons on the toolbar (Win32 only)
+    wxTB_MIXED_BUTTONS = 0x8000
 };
 
 #if wxUSE_TOOLBAR
Index: interface/wx/toolbar.h
===================================================================
--- interface/wx/toolbar.h	(revision 70579)
+++ interface/wx/toolbar.h	(working copy)
@@ -79,6 +79,11 @@
         Align the toolbar at the bottom of parent window.
     @style{wxTB_RIGHT}
         Align the toolbar at the right side of parent window.
+    @style{wxTB_MIXED_BUTTONS}
+        This style allows you to set label for all buttons, but only display it 
+        for those buttons for which was set the appropriate flag, by means of a 
+        method wxToolBar::ShowToolLabel. By default, the label is displayed for 
+        all buttons. (Windows only)
     @endStyleTable
 
     See also @ref overview_windowstyles. Note that the wxMSW native toolbar
@@ -386,6 +391,20 @@
     virtual void EnableTool(int toolId, bool enable);
 
     /**
+        Setting a flag of displaying of a label on the tool button, 
+        if style wxTB_MIXED_BUTTINS was set on the toolbar.
+
+        @param toolId
+            ID of the tool button, as passed to AddTool()
+        @param show
+            @true if label should be displayed, @false otherwise.
+
+        @remarks
+            By default, label is displayed for all buttons.
+    */
+    virtual void ShowToolLabel(int toolId, bool show);
+
+    /**
         Returns a pointer to the tool identified by @a id or @NULL if no
         corresponding tool is found.
     */
Index: src/common/tbarbase.cpp
===================================================================
--- src/common/tbarbase.cpp	(revision 70579)
+++ src/common/tbarbase.cpp	(working copy)
@@ -509,6 +509,15 @@
     }
 }
 
+void wxToolBarBase::ShowToolLabel(int toolid, bool show)
+{
+    wxToolBarToolBase *tool = FindById(toolid);
+    if ( tool )
+    {
+        tool->ShowLabel(show);
+    }
+}
+
 void wxToolBarBase::SetToggle(int toolid, bool toggle)
 {
     wxToolBarToolBase *tool = FindById(toolid);
Index: src/msw/toolbar.cpp
===================================================================
--- src/msw/toolbar.cpp	(revision 70579)
+++ src/msw/toolbar.cpp	(working copy)
@@ -356,6 +356,9 @@
 #ifdef TB_SETEXTENDEDSTYLE
     if ( wxApp::GetComCtl32Version() >= 471 )
         ::SendMessage(GetHwnd(), TB_SETEXTENDEDSTYLE, 0, TBSTYLE_EX_DRAWDDARROWS);
+
+    if ( wxApp::GetComCtl32Version() >= 581 && HasFlag(wxTB_MIXED_BUTTONS) )
+        ::SendMessage(GetHwnd(), TB_SETEXTENDEDSTYLE, 0, TBSTYLE_EX_MIXEDBUTTONS);
 #endif
 
     return true;
@@ -494,7 +497,7 @@
     if ( style & wxTB_FLAT && wxApp::GetComCtl32Version() > 400 )
         msStyle |= TBSTYLE_FLAT;
 
-    if ( style & wxTB_HORZ_LAYOUT && wxApp::GetComCtl32Version() >= 470 )
+    if ( (style & wxTB_HORZ_LAYOUT || style & wxTB_MIXED_BUTTONS) && wxApp::GetComCtl32Version() >= 470 )
         msStyle |= TBSTYLE_LIST;
 
     if ( style & wxTB_NODIVIDER )
@@ -942,7 +945,7 @@
                 if ( !HasFlag(wxTB_NOICONS) )
                     button.iBitmap = bitmapId;
 
-                if ( HasFlag(wxTB_TEXT) )
+                if ( HasFlag(wxTB_TEXT) || (HasFlag(wxTB_MIXED_BUTTONS) && tool->ShowLabel()) )
                 {
                     const wxString& label = tool->GetLabel();
                     if ( !label.empty() )
@@ -1014,6 +1017,9 @@
                         break;
                 }
 
+                if ( HasFlag(wxTB_MIXED_BUTTONS) && tool->ShowLabel() )
+                    button.fsStyle |= BTNS_SHOWTEXT;
+
                 bitmapId++;
                 break;
         }
