diff -bu1 -r wxWindows/contrib/include/wx/gizmos/dynamicsash.h wxWin/contrib/include/wx/gizmos/dynamicsash.h
--- wxWindows/contrib/include/wx/gizmos/dynamicsash.h	2003-07-26 03:18:57.000000000 +0200
+++ wxWin/contrib/include/wx/gizmos/dynamicsash.h	2003-12-29 14:42:25.000000000 +0100
@@ -134,3 +134,3 @@
 typedef void (wxEvtHandler::*wxDynamicSashSplitEventFunction)(wxDynamicSashSplitEvent&);
-typedef void (wxEvtHandler::*wxDynamicSashUnifyEventFunction)(wxDynamicSashSplitEvent&);
+typedef void (wxEvtHandler::*wxDynamicSashUnifyEventFunction)(wxDynamicSashUnifyEvent&);
 
diff -bu1 -r wxWindows/contrib/include/wx/gizmos/multicell.h wxWin/contrib/include/wx/gizmos/multicell.h
--- wxWindows/contrib/include/wx/gizmos/multicell.h	2003-07-19 15:03:15.000000000 +0200
+++ wxWin/contrib/include/wx/gizmos/multicell.h	2003-12-29 12:08:23.000000000 +0100
@@ -156,2 +156,12 @@
 
+  /* These are to hide Add() method of parents and to avoid Borland warning about hiding virtual functions */
+  void Add( wxWindow *window, int proportion = 0, int flag = 0, int border = 0, wxObject* userData = NULL )
+       { wxFlexGridSizer::Add( window, proportion, flag, border, userData); }
+  void Add( wxSizer *sizer, int proportion = 0, int flag = 0, int border = 0, wxObject* userData = NULL )
+       { wxFlexGridSizer::Add( sizer, proportion, flag, border, userData); }
+  void Add( int width, int height, int proportion = 0, int flag = 0, int border = 0, wxObject* userData = NULL )
+       { wxFlexGridSizer::Add( width, height, proportion, flag, border, userData); }
+  void Add( wxSizerItem *item )
+       { wxFlexGridSizer::Add( item); }
+
 private:
diff -bu1 -r wxWindows/contrib/samples/gizmos/dynsash/dynsash.cpp wxWin/contrib/samples/gizmos/dynsash/dynsash.cpp
--- wxWindows/contrib/samples/gizmos/dynsash/dynsash.cpp	2003-07-10 14:44:42.000000000 +0200
+++ wxWin/contrib/samples/gizmos/dynsash/dynsash.cpp	2003-12-29 16:00:18.000000000 +0100
@@ -54,8 +54,8 @@
 wxChar *HTML_content =
-wxT("<P><H1>wxDynamicSashWindow demo</H1>"
-"<P>Here is an example of how you can use <TT>wxDynamicSashWindow</TT> to allow your users to "
-"dynamically split and unify the views of your windows.  Try dragging out a few splits "
-"and then reunifying the window."
-"<P>Also, see the <TT>dynsash_switch</TT> sample for an example of an application which "
-"manages the scrollbars provided by <TT>wxDynamicSashWindow</TT> itself.");
+wxT("<P><H1>wxDynamicSashWindow demo</H1>")
+wxT("<P>Here is an example of how you can use <TT>wxDynamicSashWindow</TT> to allow your users to ")
+wxT("dynamically split and unify the views of your windows.  Try dragging out a few splits ")
+wxT("and then reunifying the window.")
+wxT("<P>Also, see the <TT>dynsash_switch</TT> sample for an example of an application which ")
+wxT("manages the scrollbars provided by <TT>wxDynamicSashWindow</TT> itself.");
 
@@ -80,4 +80,5 @@
                                     wxHtmlWindow(parent, id, pos, size, style, name) {
-    Connect(-1, wxEVT_DYNAMIC_SASH_SPLIT,
-        (wxObjectEventFunction)(wxCommandEventFunction)(wxDynamicSashSplitEventFunction) &SashHtmlWindow::OnSplit);
+    Connect(-1, wxEVT_DYNAMIC_SASH_SPLIT, (wxObjectEventFunction)
+                                          (wxEventFunction)
+                                          (wxDynamicSashSplitEventFunction)&SashHtmlWindow::OnSplit);
 
@@ -97,3 +98,3 @@
 
-void SashHtmlWindow::OnSplit(wxDynamicSashSplitEvent& event) {
+void SashHtmlWindow::OnSplit(wxDynamicSashSplitEvent& WXUNUSED(event)) {
     wxHtmlWindow *html = new SashHtmlWindow(m_dyn_sash, -1);
diff -bu1 -r wxWindows/contrib/samples/gizmos/dynsash_switch/dynsash_switch.cpp wxWin/contrib/samples/gizmos/dynsash_switch/dynsash_switch.cpp
--- wxWindows/contrib/samples/gizmos/dynsash_switch/dynsash_switch.cpp	2003-07-10 14:44:45.000000000 +0200
+++ wxWin/contrib/samples/gizmos/dynsash_switch/dynsash_switch.cpp	2003-12-29 14:26:37.000000000 +0100
@@ -111,19 +111,49 @@
 
-    Connect(GetId(), wxEVT_SIZE, (wxObjectEventFunction)&SwitchView::OnSize);
-    Connect(m_choice->GetId(), wxEVT_COMMAND_CHOICE_SELECTED, (wxObjectEventFunction)&SwitchView::OnChoice);
-    Connect(m_view->GetId(), wxEVT_PAINT, (wxObjectEventFunction)&SwitchView::OnPaint);
-
-    Connect(-1, wxEVT_SET_FOCUS, (wxObjectEventFunction)&SwitchView::OnFocus);
-    Connect(-1, wxEVT_SCROLL_TOP, (wxObjectEventFunction)&SwitchView::OnScroll);
-    Connect(-1, wxEVT_SCROLL_BOTTOM, (wxObjectEventFunction)&SwitchView::OnScroll);
-    Connect(-1, wxEVT_SCROLL_LINEUP, (wxObjectEventFunction)&SwitchView::OnScroll);
-    Connect(-1, wxEVT_SCROLL_LINEDOWN, (wxObjectEventFunction)&SwitchView::OnScroll);
-    Connect(-1, wxEVT_SCROLL_PAGEUP, (wxObjectEventFunction)&SwitchView::OnScroll);
-    Connect(-1, wxEVT_SCROLL_PAGEDOWN, (wxObjectEventFunction)&SwitchView::OnScroll);
-    Connect(-1, wxEVT_SCROLL_THUMBTRACK, (wxObjectEventFunction)&SwitchView::OnScroll);
-    Connect(-1, wxEVT_SCROLL_THUMBRELEASE, (wxObjectEventFunction)&SwitchView::OnScroll);
-    Connect(-1, wxEVT_ERASE_BACKGROUND, (wxObjectEventFunction)&SwitchView::OnErase);
-
-    Connect(-1, wxEVT_DYNAMIC_SASH_SPLIT, (wxObjectEventFunction)&SwitchView::OnSplit);
-    Connect(-1, wxEVT_DYNAMIC_SASH_UNIFY, (wxObjectEventFunction)&SwitchView::OnUnify);
+    Connect(GetId(), wxEVT_SIZE, (wxObjectEventFunction)
+                                 (wxEventFunction)
+                                 (wxSizeEventFunction)&SwitchView::OnSize);
+    Connect(m_choice->GetId(), wxEVT_COMMAND_CHOICE_SELECTED, (wxObjectEventFunction)
+                                                              (wxEventFunction)
+                                                              (wxCommandEventFunction)&SwitchView::OnChoice);
+    Connect(m_view->GetId(), wxEVT_PAINT, (wxObjectEventFunction)
+                                          (wxEventFunction)
+                                          (wxPaintEventFunction)&SwitchView::OnPaint);
+
+    Connect(-1, wxEVT_SET_FOCUS, (wxObjectEventFunction)
+                                 (wxEventFunction)
+                                 (wxFocusEventFunction)&SwitchView::OnFocus);
+    Connect(-1, wxEVT_SCROLL_TOP, (wxObjectEventFunction)
+                                  (wxEventFunction)
+                                  (wxScrollEventFunction)&SwitchView::OnScroll);
+    Connect(-1, wxEVT_SCROLL_BOTTOM, (wxObjectEventFunction)
+                                     (wxEventFunction)
+                                     (wxScrollEventFunction)&SwitchView::OnScroll);
+    Connect(-1, wxEVT_SCROLL_LINEUP, (wxObjectEventFunction)
+                                     (wxEventFunction)
+                                     (wxScrollEventFunction)&SwitchView::OnScroll);
+    Connect(-1, wxEVT_SCROLL_LINEDOWN, (wxObjectEventFunction)
+                                       (wxEventFunction)
+                                       (wxScrollEventFunction)&SwitchView::OnScroll);
+    Connect(-1, wxEVT_SCROLL_PAGEUP, (wxObjectEventFunction)
+                                     (wxEventFunction)
+                                     (wxScrollEventFunction)&SwitchView::OnScroll);
+    Connect(-1, wxEVT_SCROLL_PAGEDOWN, (wxObjectEventFunction)
+                                       (wxEventFunction)
+                                       (wxScrollEventFunction)&SwitchView::OnScroll);
+    Connect(-1, wxEVT_SCROLL_THUMBTRACK, (wxObjectEventFunction)
+                                         (wxEventFunction)
+                                         (wxScrollEventFunction)&SwitchView::OnScroll);
+    Connect(-1, wxEVT_SCROLL_THUMBRELEASE, (wxObjectEventFunction)
+                                           (wxEventFunction)
+                                           (wxScrollEventFunction)&SwitchView::OnScroll);
+    Connect(-1, wxEVT_ERASE_BACKGROUND, (wxObjectEventFunction)
+                                        (wxEventFunction)
+                                        (wxEraseEventFunction)&SwitchView::OnErase);
+
+    Connect(-1, wxEVT_DYNAMIC_SASH_SPLIT, (wxObjectEventFunction)
+                                          (wxEventFunction)
+                                          (wxDynamicSashSplitEventFunction)&SwitchView::OnSplit);
+    Connect(-1, wxEVT_DYNAMIC_SASH_UNIFY, (wxObjectEventFunction)
+                                          (wxEventFunction)
+                                          (wxDynamicSashUnifyEventFunction)&SwitchView::OnUnify);
 }
@@ -134,3 +164,3 @@
 
-void SwitchView::OnSize(wxSizeEvent& event) {
+void SwitchView::OnSize(wxSizeEvent& WXUNUSED(event)) {
     Layout();
@@ -155,3 +185,3 @@
 
-void SwitchView::OnPaint(wxPaintEvent& event) {
+void SwitchView::OnPaint(wxPaintEvent& WXUNUSED(event)) {
     wxPaintDC dc(m_view);
@@ -176,3 +206,3 @@
 
-void SwitchView::OnErase(wxEraseEvent& event) {
+void SwitchView::OnErase(wxEraseEvent& WXUNUSED(event)) {
     // Do nothing
@@ -180,3 +210,3 @@
 
-void SwitchView::OnSplit(wxDynamicSashSplitEvent& event) {
+void SwitchView::OnSplit(wxDynamicSashSplitEvent& WXUNUSED(event)) {
     SwitchView *view = new SwitchView(m_dyn_sash);
@@ -191,3 +221,3 @@
 
-void SwitchView::OnUnify(wxDynamicSashUnifyEvent& event) {
+void SwitchView::OnUnify(wxDynamicSashUnifyEvent& WXUNUSED(event)) {
     wxScrollBar *hscroll = m_dyn_sash->GetHScrollBar(this);
@@ -199,3 +229,3 @@
 
-void SwitchView::OnChoice(wxCommandEvent& event) {
+void SwitchView::OnChoice(wxCommandEvent& WXUNUSED(event)) {
     m_view->Refresh();
@@ -203,3 +233,3 @@
 
-void SwitchView::OnScroll(wxScrollEvent& event) {
+void SwitchView::OnScroll(wxScrollEvent& WXUNUSED(event)) {
     m_view->Refresh();
diff -bu1 -r wxWindows/contrib/samples/gizmos/led/led.cpp wxWin/contrib/samples/gizmos/led/led.cpp
--- wxWindows/contrib/samples/gizmos/led/led.cpp	2002-12-04 14:53:58.000000000 +0100
+++ wxWin/contrib/samples/gizmos/led/led.cpp	2003-12-29 11:02:26.000000000 +0100
@@ -227,3 +227,3 @@
 
-void MyFrame::OnIncrement(wxCommandEvent& event)
+void MyFrame::OnIncrement(wxCommandEvent& WXUNUSED(event))
 {
@@ -232,3 +232,3 @@
 
-void MyFrame::OnDecrement(wxCommandEvent& event)
+void MyFrame::OnDecrement(wxCommandEvent& WXUNUSED(event))
 {
@@ -237,3 +237,3 @@
 
-void MyFrame::OnSetValue(wxCommandEvent& event)
+void MyFrame::OnSetValue(wxCommandEvent& WXUNUSED(event))
 {
@@ -242,3 +242,3 @@
 
-void MyFrame::OnAlignLeft(wxCommandEvent& event)
+void MyFrame::OnAlignLeft(wxCommandEvent& WXUNUSED(event))
 {
@@ -247,3 +247,3 @@
 
-void MyFrame::OnAlignCenter(wxCommandEvent& event)
+void MyFrame::OnAlignCenter(wxCommandEvent& WXUNUSED(event))
 {
@@ -252,3 +252,3 @@
 
-void MyFrame::OnAlignRight(wxCommandEvent& event)
+void MyFrame::OnAlignRight(wxCommandEvent& WXUNUSED(event))
 {
@@ -257,3 +257,3 @@
 
-void MyFrame::OnDrawFaded(wxCommandEvent& event)
+void MyFrame::OnDrawFaded(wxCommandEvent& WXUNUSED(event))
 {
diff -bu1 -r wxWindows/contrib/samples/gizmos/multicell/mtest.cpp wxWin/contrib/samples/gizmos/multicell/mtest.cpp
--- wxWindows/contrib/samples/gizmos/multicell/mtest.cpp	2003-07-10 14:44:47.000000000 +0200
+++ wxWin/contrib/samples/gizmos/multicell/mtest.cpp	2003-12-29 12:17:38.000000000 +0100
@@ -210,3 +210,3 @@
 
-void MyFrame::OnCloseWindow(wxCloseEvent& event)
+void MyFrame::OnCloseWindow(wxCloseEvent& WXUNUSED(event))
 {
diff -bu1 -r wxWindows/contrib/src/gizmos/dynamicsash.cpp wxWin/contrib/src/gizmos/dynamicsash.cpp
--- wxWindows/contrib/src/gizmos/dynamicsash.cpp	2003-09-11 12:50:15.000000000 +0200
+++ wxWin/contrib/src/gizmos/dynamicsash.cpp	2003-12-29 14:36:00.000000000 +0100
@@ -315,9 +315,23 @@
 
-    Connect(-1, wxEVT_SIZE, (wxObjectEventFunction)&wxDynamicSashWindowImpl::OnSize);
-    Connect(-1, wxEVT_PAINT, (wxObjectEventFunction)&wxDynamicSashWindowImpl::OnPaint);
-    Connect(-1, wxEVT_MOTION, (wxObjectEventFunction)&wxDynamicSashWindowImpl::OnMouseMove);
-    Connect(-1, wxEVT_ENTER_WINDOW, (wxObjectEventFunction)&wxDynamicSashWindowImpl::OnMouseMove);
-    Connect(-1, wxEVT_LEAVE_WINDOW, (wxObjectEventFunction)&wxDynamicSashWindowImpl::OnLeave);
-    Connect(-1, wxEVT_LEFT_DOWN, (wxObjectEventFunction)&wxDynamicSashWindowImpl::OnPress);
-    Connect(-1, wxEVT_LEFT_UP, (wxObjectEventFunction)&wxDynamicSashWindowImpl::OnRelease);
+    Connect(-1, wxEVT_SIZE, (wxObjectEventFunction)
+                            (wxEventFunction)
+                            (wxSizeEventFunction)&wxDynamicSashWindowImpl::OnSize);
+    Connect(-1, wxEVT_PAINT, (wxObjectEventFunction)
+                             (wxEventFunction)
+                             (wxPaintEventFunction)&wxDynamicSashWindowImpl::OnPaint);
+    Connect(-1, wxEVT_MOTION, (wxObjectEventFunction)
+                              (wxEventFunction)
+                              (wxMouseEventFunction)&wxDynamicSashWindowImpl::OnMouseMove);
+    Connect(-1, wxEVT_ENTER_WINDOW, (wxObjectEventFunction)
+                                    (wxEventFunction)
+                                    (wxMouseEventFunction)&wxDynamicSashWindowImpl::OnMouseMove);
+    Connect(-1, wxEVT_LEAVE_WINDOW, (wxObjectEventFunction)
+                                    (wxEventFunction)
+                                    (wxMouseEventFunction)&wxDynamicSashWindowImpl::OnLeave);
+    Connect(-1, wxEVT_LEFT_DOWN, (wxObjectEventFunction)
+                                 (wxEventFunction)
+                                 (wxMouseEventFunction)&wxDynamicSashWindowImpl::OnPress);
+    Connect(-1, wxEVT_LEFT_UP, (wxObjectEventFunction)
+                               (wxEventFunction)
+                               (wxMouseEventFunction)&wxDynamicSashWindowImpl::OnRelease);
 
@@ -904,11 +918,29 @@
 
-        Connect(-1, wxEVT_SET_FOCUS, (wxObjectEventFunction)&wxDynamicSashWindowLeaf::OnFocus);
-        Connect(-1, wxEVT_SCROLL_TOP, (wxObjectEventFunction)&wxDynamicSashWindowLeaf::OnScroll);
-        Connect(-1, wxEVT_SCROLL_BOTTOM, (wxObjectEventFunction)&wxDynamicSashWindowLeaf::OnScroll);
-        Connect(-1, wxEVT_SCROLL_LINEUP, (wxObjectEventFunction)&wxDynamicSashWindowLeaf::OnScroll);
-        Connect(-1, wxEVT_SCROLL_LINEDOWN, (wxObjectEventFunction)&wxDynamicSashWindowLeaf::OnScroll);
-        Connect(-1, wxEVT_SCROLL_PAGEUP, (wxObjectEventFunction)&wxDynamicSashWindowLeaf::OnScroll);
-        Connect(-1, wxEVT_SCROLL_PAGEDOWN, (wxObjectEventFunction)&wxDynamicSashWindowLeaf::OnScroll);
-        Connect(-1, wxEVT_SCROLL_THUMBTRACK, (wxObjectEventFunction)&wxDynamicSashWindowLeaf::OnScroll);
-        Connect(-1, wxEVT_SCROLL_THUMBRELEASE, (wxObjectEventFunction)&wxDynamicSashWindowLeaf::OnScroll);
+        Connect(-1, wxEVT_SET_FOCUS, (wxObjectEventFunction)
+                                     (wxEventFunction)
+                                     (wxFocusEventFunction)&wxDynamicSashWindowLeaf::OnFocus);
+        Connect(-1, wxEVT_SCROLL_TOP, (wxObjectEventFunction)
+                                      (wxEventFunction)
+                                      (wxScrollEventFunction)&wxDynamicSashWindowLeaf::OnScroll);
+        Connect(-1, wxEVT_SCROLL_BOTTOM, (wxObjectEventFunction)
+                                         (wxEventFunction)
+                                         (wxScrollEventFunction)&wxDynamicSashWindowLeaf::OnScroll);
+        Connect(-1, wxEVT_SCROLL_LINEUP, (wxObjectEventFunction)
+                                         (wxEventFunction)
+                                         (wxScrollEventFunction)&wxDynamicSashWindowLeaf::OnScroll);
+        Connect(-1, wxEVT_SCROLL_LINEDOWN, (wxObjectEventFunction)
+                                           (wxEventFunction)
+                                           (wxScrollEventFunction)&wxDynamicSashWindowLeaf::OnScroll);
+        Connect(-1, wxEVT_SCROLL_PAGEUP, (wxObjectEventFunction)
+                                         (wxEventFunction)
+                                         (wxScrollEventFunction)&wxDynamicSashWindowLeaf::OnScroll);
+        Connect(-1, wxEVT_SCROLL_PAGEDOWN, (wxObjectEventFunction)
+                                           (wxEventFunction)
+                                           (wxScrollEventFunction)&wxDynamicSashWindowLeaf::OnScroll);
+        Connect(-1, wxEVT_SCROLL_THUMBTRACK, (wxObjectEventFunction)
+                                             (wxEventFunction)
+                                             (wxScrollEventFunction)&wxDynamicSashWindowLeaf::OnScroll);
+        Connect(-1, wxEVT_SCROLL_THUMBRELEASE, (wxObjectEventFunction)
+                                               (wxEventFunction)
+                                               (wxScrollEventFunction)&wxDynamicSashWindowLeaf::OnScroll);
     }
diff -bu1 -r wxWindows/contrib/src/gizmos/multicell.cpp wxWin/contrib/src/gizmos/multicell.cpp
--- wxWindows/contrib/src/gizmos/multicell.cpp	2003-09-11 12:50:15.000000000 +0200
+++ wxWin/contrib/src/gizmos/multicell.cpp	2003-12-29 13:03:38.000000000 +0100
@@ -382,8 +382,5 @@
 
-	int m_minWidth = 0;
-	int m_minHeight = 0;
-
 	GetMinimums();
-	m_minWidth = Sum(m_maxWidth, m_cell_count.GetWidth());
-	m_minHeight = Sum(m_maxHeight, m_cell_count.GetHeight());
+	int m_minWidth = Sum(m_maxWidth, m_cell_count.GetWidth());
+	int m_minHeight = Sum(m_maxHeight, m_cell_count.GetHeight());
 	return wxSize( m_minWidth, m_minHeight );
diff -bu1 -r wxWindows/contrib/src/gizmos/splittree.cpp wxWin/contrib/src/gizmos/splittree.cpp
--- wxWindows/contrib/src/gizmos/splittree.cpp	2003-09-11 12:50:15.000000000 +0200
+++ wxWin/contrib/src/gizmos/splittree.cpp	2003-12-29 13:05:34.000000000 +0100
@@ -300,3 +300,2 @@
     wxRect itemRect;
-    int cy=0;
     wxTreeItemId h, lastH;
@@ -306,3 +305,3 @@
         {
-            cy = itemRect.GetTop();
+            int cy = itemRect.GetTop();
             dc.DrawLine(0, cy, clientSize.x, cy);
@@ -313,3 +312,3 @@
     {
-        cy = itemRect.GetBottom();
+        int cy = itemRect.GetBottom();
         dc.DrawLine(0, cy, clientSize.x, cy);
@@ -510,3 +509,2 @@
     wxRect itemRect;
-    int cy=0;
     wxTreeItemId h, lastH;
@@ -516,3 +514,3 @@
         {
-            cy = itemRect.GetTop();
+            int cy = itemRect.GetTop();
             wxRect drawItemRect(0, cy, clientSize.x, itemRect.GetHeight());
@@ -528,3 +526,3 @@
     {
-        cy = itemRect.GetBottom();
+        int cy = itemRect.GetBottom();
         dc.DrawLine(0, cy, clientSize.x, cy);

 	  	 
