diff -uNr org\include\wx\docview.h new\include\wx\docview.h
--- org\include\wx\docview.h	Tue Nov 05 09:14:41 2013
+++ new\include\wx\docview.h	Wed Oct 09 10:02:44 2013
@@ -1020,12 +1020,13 @@
 wxTransferFileToStream(const wxString& filename, wxSTD ostream& stream);
 bool WXDLLIMPEXP_CORE
 wxTransferStreamToFile(wxSTD istream& stream, const wxString& filename);
-#else
+#endif // wxUSE_STD_IOSTREAM
+#if wxUSE_STREAMS
 bool WXDLLIMPEXP_CORE
 wxTransferFileToStream(const wxString& filename, wxOutputStream& stream);
 bool WXDLLIMPEXP_CORE
 wxTransferStreamToFile(wxInputStream& stream, const wxString& filename);
-#endif // wxUSE_STD_IOSTREAM
+#endif // wxUSE_STREAMS
 
 
 // these flags are not used anywhere by wxWidgets and kept only for an unlikely
diff -uNr org\src\common\docview.cpp new\src\common\docview.cpp
--- org\src\common\docview.cpp	Mon Nov 04 16:54:08 2013
+++ new\src\common\docview.cpp	Wed Oct 09 10:02:44 2013
@@ -2248,8 +2248,9 @@
     return true;
 }
 
-#else // !wxUSE_STD_IOSTREAM
+#endif // !wxUSE_STD_IOSTREAM
 
+#if wxUSE_STREAMS
 bool wxTransferFileToStream(const wxString& filename, wxOutputStream& stream)
 {
 #if wxUSE_FFILE
@@ -2309,6 +2310,6 @@
     return true;
 }
 
-#endif // wxUSE_STD_IOSTREAM/!wxUSE_STD_IOSTREAM
+#endif // wxUSE_STREAM
 
 #endif // wxUSE_DOC_VIEW_ARCHITECTURE
