Index: interface/wx/dataview.h
===================================================================
--- interface/wx/dataview.h	(revision 70045)
+++ interface/wx/dataview.h	(working copy)
@@ -581,6 +581,12 @@
     void SetColour(const wxColour& colour);
 
     /**
+        Call this to indicate that the item shall be displayed with that background colour.
+        @since 2.9.4
+    */
+    void SetBackgroundColour(const wxColour& colour);
+
+    /**
         Call this to indicate that the item shall be displayed in italic text.
     */
     void SetItalic(bool set);
