diff --git a/src/gtk/window.cpp b/src/gtk/window.cpp
index a27263a..9eafc87 100644
--- a/src/gtk/window.cpp
+++ b/src/gtk/window.cpp
@@ -1199,7 +1199,7 @@ wxWindowGTK *FindWindowForMouseEvent(wxWindowGTK *win, wxCoord& x, wxCoord& y)
         else
         {
             if ((child->m_wxwindow == NULL) &&
-                win->IsClientAreaChild(child) &&
+                win->IsClientAreaChild(static_cast<wxWindow*>(child)) &&
                 (child->m_x <= xx) &&
                 (child->m_y <= yy) &&
                 (child->m_x+child->m_width  >= xx) &&
