diff --git a/src/common/datavcmn.cpp b/src/common/datavcmn.cpp
--- a/src/common/datavcmn.cpp
+++ b/src/common/datavcmn.cpp
@@ -1259,7 +1259,7 @@ wxDataViewCtrlBase::PrependTextColumn( const wxString &label, unsigned int model
                             wxDataViewCellMode mode, int width, wxAlignment align, int flags )
 {
     wxDataViewColumn *ret = new wxDataViewColumn( label,
-        new wxDataViewTextRenderer( wxT("string"), mode ),
+        new wxDataViewTextRenderer( wxT("string"), mode, align ),
         model_column, width, align, flags );
     PrependColumn( ret );
     return ret;
@@ -1270,7 +1270,7 @@ wxDataViewCtrlBase::PrependIconTextColumn( const wxString &label, unsigned int m
                             wxDataViewCellMode mode, int width, wxAlignment align, int flags )
 {
     wxDataViewColumn *ret = new wxDataViewColumn( label,
-        new wxDataViewIconTextRenderer( wxT("wxDataViewIconText"), mode ),
+        new wxDataViewIconTextRenderer( wxT("wxDataViewIconText"), mode, align ),
         model_column, width, align, flags );
     PrependColumn( ret );
     return ret;
@@ -1282,7 +1282,7 @@ wxDataViewCtrlBase::PrependToggleColumn( const wxString &label, unsigned int mod
 {
 
     wxDataViewColumn *ret = new wxDataViewColumn( label,
-        new wxDataViewToggleRenderer( wxT("bool"), mode ),
+        new wxDataViewToggleRenderer( wxT("bool"), mode, align ),
         model_column, width, align, flags );
     PrependColumn( ret );
     return ret;
@@ -1293,7 +1293,7 @@ wxDataViewCtrlBase::PrependProgressColumn( const wxString &label, unsigned int m
                             wxDataViewCellMode mode, int width, wxAlignment align, int flags )
 {
     wxDataViewColumn *ret = new wxDataViewColumn( label,
-        new wxDataViewProgressRenderer( wxEmptyString, wxT("long"), mode ),
+        new wxDataViewProgressRenderer( wxEmptyString, wxT("long"), mode, align ),
         model_column, width, align, flags );
     PrependColumn( ret );
     return ret;
@@ -1304,7 +1304,7 @@ wxDataViewCtrlBase::PrependDateColumn( const wxString &label, unsigned int model
                             wxDataViewCellMode mode, int width, wxAlignment align, int flags )
 {
     wxDataViewColumn *ret = new wxDataViewColumn( label,
-        new wxDataViewDateRenderer( wxT("datetime"), mode ),
+        new wxDataViewDateRenderer( wxT("datetime"), mode, align ),
         model_column, width, align, flags );
     PrependColumn( ret );
     return ret;
@@ -1315,7 +1315,7 @@ wxDataViewCtrlBase::PrependBitmapColumn( const wxString &label, unsigned int mod
                             wxDataViewCellMode mode, int width, wxAlignment align, int flags )
 {
     wxDataViewColumn *ret = new wxDataViewColumn( label,
-        new wxDataViewBitmapRenderer( wxT("wxBitmap"), mode ),
+        new wxDataViewBitmapRenderer( wxT("wxBitmap"), mode, align ),
         model_column, width, align, flags );
     PrependColumn( ret );
     return ret;
@@ -1326,7 +1326,7 @@ wxDataViewCtrlBase::PrependTextColumn( const wxBitmap &label, unsigned int model
                             wxDataViewCellMode mode, int width, wxAlignment align, int flags )
 {
     wxDataViewColumn *ret = new wxDataViewColumn( label,
-        new wxDataViewTextRenderer( wxT("string"), mode ),
+        new wxDataViewTextRenderer( wxT("string"), mode, align ),
         model_column, width, align, flags );
     PrependColumn( ret );
     return ret;
@@ -1337,7 +1337,7 @@ wxDataViewCtrlBase::PrependIconTextColumn( const wxBitmap &label, unsigned int m
                             wxDataViewCellMode mode, int width, wxAlignment align, int flags )
 {
     wxDataViewColumn *ret = new wxDataViewColumn( label,
-        new wxDataViewIconTextRenderer( wxT("wxDataViewIconText"), mode ),
+        new wxDataViewIconTextRenderer( wxT("wxDataViewIconText"), mode, align ),
         model_column, width, align, flags );
     PrependColumn( ret );
     return ret;
@@ -1348,7 +1348,7 @@ wxDataViewCtrlBase::PrependToggleColumn( const wxBitmap &label, unsigned int mod
                             wxDataViewCellMode mode, int width, wxAlignment align, int flags )
 {
     wxDataViewColumn *ret = new wxDataViewColumn( label,
-        new wxDataViewToggleRenderer( wxT("bool"), mode ),
+        new wxDataViewToggleRenderer( wxT("bool"), mode, align ),
         model_column, width, align, flags );
     PrependColumn( ret );
     return ret;
@@ -1359,7 +1359,7 @@ wxDataViewCtrlBase::PrependProgressColumn( const wxBitmap &label, unsigned int m
                             wxDataViewCellMode mode, int width, wxAlignment align, int flags )
 {
     wxDataViewColumn *ret = new wxDataViewColumn( label,
-        new wxDataViewProgressRenderer( wxEmptyString, wxT("long"), mode ),
+        new wxDataViewProgressRenderer( wxEmptyString, wxT("long"), mode, align ),
         model_column, width, align, flags );
     PrependColumn( ret );
     return ret;
@@ -1370,7 +1370,7 @@ wxDataViewCtrlBase::PrependDateColumn( const wxBitmap &label, unsigned int model
                             wxDataViewCellMode mode, int width, wxAlignment align, int flags )
 {
     wxDataViewColumn *ret = new wxDataViewColumn( label,
-        new wxDataViewDateRenderer( wxT("datetime"), mode ),
+        new wxDataViewDateRenderer( wxT("datetime"), mode, align ),
         model_column, width, align, flags );
     PrependColumn( ret );
     return ret;
@@ -1381,7 +1381,7 @@ wxDataViewCtrlBase::PrependBitmapColumn( const wxBitmap &label, unsigned int mod
                             wxDataViewCellMode mode, int width, wxAlignment align, int flags )
 {
     wxDataViewColumn *ret = new wxDataViewColumn( label,
-        new wxDataViewBitmapRenderer( wxT("wxBitmap"), mode ),
+        new wxDataViewBitmapRenderer( wxT("wxBitmap"), mode, align ),
         model_column, width, align, flags );
     PrependColumn( ret );
     return ret;
