--- ./wx-config.in-	2004-10-29 18:52:37.000000000 -0700
+++ ./wx-config.in	2004-10-30 15:05:09.000000000 -0700
@@ -578,9 +578,13 @@
 # Returns a list of configs installed by wx2.4 releases.
 find_legacy_configs()
 {
-    cd "$prefix/bin" 2> /dev/null &&                        \
-        ls wx*-2.4-config 2> /dev/null | grep -v ^wxbase;   \
-        ls wx*-2.4-config 2> /dev/null | grep ^wxbase
+  (
+    cd "$prefix/bin" &&
+      {
+        ls wx*-2.4-config | grep -v ^wxbase
+        ls wx*-2.4-config | grep ^wxbase
+      }
+  ) 2> /dev/null
 }
 
 # find_best_legacy_config

 	  	 
