Index: src/osx/dialog_osx.cpp
===================================================================
--- src/osx/dialog_osx.cpp	(revision 60729)
+++ src/osx/dialog_osx.cpp	(working copy)
@@ -32,6 +32,9 @@
 
 void wxDialog::Init()
 {
+#if wxOSX_USE_CARBON
+    m_modalLoop = NULL;
+#endif
     m_isModalStyle = false;
 }
 
@@ -118,6 +121,12 @@
             // this will cause IsModalShowing() return false and our local
             // message loop will terminate
             wxModalDialogs.DeleteObject(this);
+#if wxOSX_USE_CARBON
+            // Ensure Exit() is only called once. The dialog's event loop may be terminated
+            // externally due to an uncaught exception.
+            if (m_modalLoop && m_modalLoop->IsRunning())
+                m_modalLoop->Exit();
+#endif
         }
     }
 
Index: src/osx/carbon/dialog.cpp
===================================================================
--- src/osx/carbon/dialog.cpp	(revision 60729)
+++ src/osx/carbon/dialog.cpp	(working copy)
@@ -22,9 +22,12 @@
 
 #include "wx/osx/private.h"
 #include "wx/evtloop.h"
+#include "wx/scopedptr.h"
 
 extern wxList wxModalDialogs;
 
+wxDEFINE_TIED_SCOPED_PTR_TYPE(wxGUIEventLoop)
+
 void wxDialog::DoShowModal()
 {
     wxCHECK_RET( !IsModal(), wxT("DoShowModal() called twice") );
@@ -47,16 +50,10 @@
     }
     BeginAppModalStateForWindow(windowRef) ;
 
-    wxEventLoop * const
-        loop = static_cast<wxEventLoop *>(wxEventLoop::GetActive());
-        
-    
-    wxASSERT_MSG( loop , wxT("No Event Loop in place") );
-        
-    if ( loop )
+    // Run modal dialog event loop.
     {
-        while ( IsModal() )
-            loop->Dispatch();
+        wxGUIEventLoopTiedPtr modal(&m_modalLoop, new wxGUIEventLoop());
+        m_modalLoop->Run();
     }
 
     EndAppModalStateForWindow(windowRef) ;
@@ -64,4 +61,4 @@
     {
         SetWindowGroupParent( windowGroup , formerParentGroup );
     }
-}
\ No newline at end of file
+}
Index: include/wx/osx/dialog.h
===================================================================
--- include/wx/osx/dialog.h	(revision 60729)
+++ include/wx/osx/dialog.h	(working copy)
@@ -16,6 +16,7 @@
 
 WXDLLIMPEXP_DATA_CORE(extern const char) wxDialogNameStr[];
 
+class WXDLLIMPEXP_FWD_CORE wxGUIEventLoop;
 class WXDLLIMPEXP_FWD_CORE wxMacToolTip ;
 
 // Dialog boxes
@@ -72,6 +73,9 @@
 private:
     void Init();
 
+#if wxOSX_USE_CARBON
+    wxGUIEventLoop *m_modalLoop;
+#endif
     bool m_isModalStyle;
 };
 
