diff -bu1 -r wxWindows/utils/configtool/src/configtooldoc.cpp wxWin/utils/configtool/src/configtooldoc.cpp
--- wxWindows/utils/configtool/src/configtooldoc.cpp	2003-12-12 10:09:17.000000000 +0100
+++ wxWin/utils/configtool/src/configtooldoc.cpp	2004-01-10 22:27:34.000000000 +0100
@@ -103,9 +103,7 @@
 {
-    bool ret = FALSE;
-
     if (!IsModified() && m_savedYet) return TRUE;
-    if (m_documentFile == wxT("") || !m_savedYet)
-        ret = SaveAs();
-    else
-        ret = OnSaveDocument(m_documentFile);
+
+    bool ret = (m_documentFile == wxT("") || !m_savedYet) ?
+                 SaveAs() :
+                 OnSaveDocument(m_documentFile);
     if ( ret )
diff -bu1 -r wxWindows/utils/configtool/src/mainframe.cpp wxWin/utils/configtool/src/mainframe.cpp
--- wxWindows/utils/configtool/src/mainframe.cpp	2003-10-01 14:51:42.000000000 +0200
+++ wxWin/utils/configtool/src/mainframe.cpp	2004-01-12 17:16:17.000000000 +0100
@@ -100,3 +100,5 @@
     m_setupPage = NULL;
+#ifdef USE_CONFIG_BROWSER_PAGE
     m_configBrowserPage = NULL;
+#endif
     m_mainNotebook = NULL;
@@ -120,3 +122,3 @@
 
-#if 0
+#ifdef USE_CONFIG_BROWSER_PAGE
     m_configBrowserPage = new ctConfigurationBrowserWindow(m_mainNotebook, -1, wxDefaultPosition, wxSize(300, 200),
@@ -126,3 +128,3 @@
     m_mainNotebook->AddPage(m_propertyEditor, _T("Properties"));
-#if 0
+#ifdef USE_CONFIG_BROWSER_PAGE
     m_mainNotebook->AddPage(m_configBrowserPage, _T("Configuration Browser"));
diff -bu1 -r wxWindows/utils/configtool/src/mainframe.h wxWin/utils/configtool/src/mainframe.h
--- wxWindows/utils/configtool/src/mainframe.h	2003-10-01 14:51:42.000000000 +0200
+++ wxWin/utils/configtool/src/mainframe.h	2004-01-12 17:16:42.000000000 +0100
@@ -21,2 +21,4 @@
 
+// #define USE_CONFIG_BROWSER_PAGE
+
 class WXDLLEXPORT wxHtmlWindow;
@@ -29,3 +31,5 @@
 class ctFindReplaceDialog;
+#ifdef USE_CONFIG_BROWSER_PAGE
 class ctConfigurationBrowserWindow;
+#endif
 
@@ -158,3 +162,5 @@
     // configurations.
+#ifdef USE_CONFIG_BROWSER_PAGE
     ctConfigurationBrowserWindow*   m_configBrowserPage;
+#endif
 
diff -bu1 -r wxWindows/utils/configtool/src/utils.cpp wxWin/utils/configtool/src/utils.cpp
--- wxWindows/utils/configtool/src/utils.cpp	2003-09-11 12:29:25.000000000 +0200
+++ wxWin/utils/configtool/src/utils.cpp	2004-01-10 22:31:04.000000000 +0100
@@ -79,8 +79,5 @@
 {
-    unsigned int r = 0;
-    unsigned int g = 0;
-    unsigned int b = 0;
-    r = wxHexToDec(hex.Mid(0, 2));
-    g = wxHexToDec(hex.Mid(2, 2));
-    b = wxHexToDec(hex.Mid(4, 2));
+    unsigned int r = wxHexToDec(hex.Mid(0, 2));
+    unsigned int g = wxHexToDec(hex.Mid(2, 2));
+    unsigned int b = wxHexToDec(hex.Mid(4, 2));
 
@@ -260,9 +257,6 @@
     wxString cmd;
-    bool ok = ft->GetOpenCommand(&cmd,
-                                 wxFileType::MessageParameters(filename, _T("")));
+    ft->GetOpenCommand(&cmd, wxFileType::MessageParameters(filename, _T("")));
     delete ft;
 
-    ok = (wxExecute(cmd, FALSE) != 0);
-
-    return ok;
+    return (wxExecute(cmd, FALSE) != 0);
 }
@@ -523,3 +517,2 @@
     bool success = FALSE;
-    int pos = 0;
     int matchTextLen = (int) matchText.Length();
@@ -527,3 +520,3 @@
     {
-        pos = left.Find(matchText);
+        int pos = left.Find(matchText);
         if (pos == -1)

 	  	 
