--- src\msw\app-prv2.cpp	2008-06-15 13:36:37.000000000 +0200
+++ src\msw\app.cpp	2008-06-15 12:24:53.000000000 +0200
@@ -598,8 +598,16 @@
 
 void wxApp::OnEndSession(wxCloseEvent& WXUNUSED(event))
 {
-    if (GetTopWindow())
+    // Close all TLW windows. Make sure message queues are emptied and that
+    // the windows are fully destroyed before calling wxApp::OnExit below
+    // (otherwise the windows exist after wxApp::OnExit).
+    // Note: TLW:s are removed from list of TLW only after they destroyed, so 
+    // we need ProcessPendingEvents / DeletePendingObjects inside the loop.
+    while(GetTopWindow()){
         GetTopWindow()->Close(true);
+        ProcessPendingEvents();
+        DeletePendingObjects();
+    }
         
     const int rc = OnExit();
     wxEntryCleanup();
