Index: src/common/stockitem.cpp
===================================================================
--- src/common/stockitem.cpp	(revision 70472)
+++ src/common/stockitem.cpp	(working copy)
@@ -180,7 +180,7 @@
         STOCKITEM(wxID_OPEN,                _("&Open..."),            _("Open..."));
         STOCKITEM(wxID_PASTE,               _("&Paste"),              _("Paste"));
         STOCKITEM(wxID_PREFERENCES,         _("&Preferences"),        _("Preferences"));
-        STOCKITEM(wxID_PREVIEW,             _("&Preview..."),         _("Preview..."));
+        STOCKITEM(wxID_PREVIEW,             _("Print previe&w"),      _("Print preview"));
         STOCKITEM(wxID_PRINT,               _("&Print..."),           _("Print..."));
         STOCKITEM(wxID_PROPERTIES,          _("&Properties"),         _("Properties"));
         STOCKITEM(wxID_REDO,                _("&Redo"),               _("Redo"));
