--- wxWidgets-trunk/src/generic/grid.cpp	2014-04-09
+++ wxWidgets-work/src/generic/grid.cpp	2014-04-09
@@ -3724,10 +3724,7 @@
                     // get the position of the column we're over
                     int pos = XToPos(x);
 
-                    // we may need to adjust the drop position but don't bother
-                    // checking for it if we can't anyhow
-                    if ( pos > 1 )
-                    {
+                    // we may need to adjust the drop position
                         // also find the index of the column we're over: notice
                         // that the existing "col" variable may be invalid but
                         // we need a valid one here
@@ -3744,9 +3741,14 @@
                         else // wxLayout_RightToLeft
                             onNearPart = (x > middle);
 
+                    // We need to adjust target column index
+                    // to account also dragged column itself.
+                    int dragPos = GetColPos(m_dragRowOrCol);
+                    if( pos < dragPos )
+                        pos++;
+
                         if ( onNearPart )
                             pos--;
-                    }
 
                     DoEndMoveCol(pos);
                 }
