Index: src/x11/dcclient.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/x11/dcclient.cpp,v
retrieving revision 1.43
diff -b -u -2 -r1.43 dcclient.cpp
--- src/x11/dcclient.cpp	2006/01/17 15:53:56	1.43
+++ src/x11/dcclient.cpp	2006/01/19 15:08:45
@@ -1186,5 +1186,5 @@
             Window xroot = RootWindow( xdisplay, xscreen );
 
-            new_pixmap = XCreatePixmap( xdisplay, xroot, w, h, 1 );
+            new_pixmap = XCreatePixmap( xdisplay, xroot, ww, hh, 1 );
             GC gc = XCreateGC( xdisplay, new_pixmap, 0, NULL );
 
@@ -1196,5 +1196,5 @@
         
             XSetFillStyle( xdisplay, gc, FillSolid );
-            XFillRectangle( xdisplay, new_pixmap, gc, 0, 0, w, h );
+            XFillRectangle( xdisplay, new_pixmap, gc, 0, 0, ww, hh );
 
             if (bpp == 8)
@@ -1243,8 +1243,8 @@
     if (is_mono)
         XCopyPlane( (Display*) m_display, (Pixmap) use_bitmap.GetBitmap(), (Window) m_window,
-            (GC) m_textGC, 0, 0, w, h, xx, yy, 1 );
+            (GC) m_textGC, 0, 0, ww, hh, xx, yy, 1 );
     else
         XCopyArea( (Display*) m_display, (Pixmap) use_bitmap.GetPixmap(), (Window) m_window,
-            (GC) m_penGC, 0, 0, w, h, xx, yy );
+            (GC) m_penGC, 0, 0, ww, hh, xx, yy );
 
     // remove mask again if any

 	  	 
