--- wxWidgets/src/msw/choice.cpp	2007-01-08 16:17:30.000000000 +0200
+++ wxWidgets.fixed/src/msw/choice.cpp	2007-03-20 12:27:08.755842100 +0200
@@ -720,6 +720,8 @@
             // need to reset the selection back to it if it's eventually
             // cancelled by user
             m_lastAcceptedSelection = GetCurrentSelection();
+            if(m_lastAcceptedSelection == -1)
+                m_lastAcceptedSelection = wxID_NONE;
             break;
 
         case CBN_CLOSEUP:

 	  	 
