Index: interface/wx/button.h
===================================================================
--- interface/wx/button.h	(revision 66893)
+++ interface/wx/button.h	(working copy)
@@ -78,7 +78,10 @@
         get reasonably good behaviour on all platforms.
 
     All of the bitmaps must be of the same size and the normal bitmap must be
-    set first (to a valid bitmap), before setting any other ones.
+    set first (to a valid bitmap), before setting any other ones. Since 2.9.2
+    setting a bitmap of different size is possible by re-setting the normal 
+    bitmap using SetBitmap(). However, doing so invalidates any previously set
+    bitmap images and position data.
 
     The position of the image inside the button be configured using
     SetBitmapPosition(). By default the image is on the left of the text.
Index: src/msw/button.cpp
===================================================================
--- src/msw/button.cpp	(revision 66893)
+++ src/msw/button.cpp	(working copy)
@@ -947,6 +947,32 @@
 
 void wxButton::DoSetBitmap(const wxBitmap& bitmap, State which)
 {
+    // #ifndef wxUSE_UXTHEME then wxODButtonImageData m_imageData is used
+    // and has wxBitmap m_bitmaps[wxButton::State_Max] which should hold
+    // varying bitmap sizes
+#if wxUSE_UXTHEME
+    if ( m_imageData )
+    {
+        wxASSERT_MSG( bitmap.GetSize() == 
+                        m_imageData->GetBitmap( State_Normal ).GetSize() ||
+                        which == State_Normal, 
+                        "New bitmap size must use SetBitmap normal." );
+
+        // Check if we are using wxXPButtonImageData and bitmap size is 
+        // different than the wxImagelist bitmap size already set
+        if ( ShowsLabel() && wxUxThemeEngine::GetIfActive() && 
+                             bitmap.GetSize() != 
+                             m_imageData->GetBitmap( State_Normal ).GetSize() )
+        {
+            // We can't change the wxImageList bitmap size without 
+            // invalidating its MSW Handle so delete its container m_imageData 
+            // to allow a re-create for the new size
+            delete m_imageData;
+            m_imageData = NULL;
+        }
+    }
+#endif // wxUSE_UXTHEME
+
     // allocate the image data when the first bitmap is set
     if ( !m_imageData )
     {
