Index: src/osx/cocoa/window.mm
===================================================================
--- src/osx/cocoa/window.mm	(revision 78290)
+++ src/osx/cocoa/window.mm	(working copy)
@@ -232,6 +232,10 @@
             {
                 switch ( [s characterAtIndex:0] )
                 {
+                    // numpad enter key End-of-text character ETX U+0003
+                    case 3:
+                        retval = WXK_NUMPAD_ENTER;
+                        break;
                     // backspace key
                     case 0x7F :
                     case 8 :
@@ -345,9 +349,6 @@
             case 69: // +
                 retval = WXK_NUMPAD_ADD;
                 break;
-            case 76: // Enter
-                retval = WXK_NUMPAD_ENTER;
-                break;
             case 65: // .
                 retval = WXK_NUMPAD_DECIMAL;
                 break;
