Index: include/wx/defs.h
===================================================================
--- include/wx/defs.h	(revision 61421)
+++ include/wx/defs.h	(working copy)
@@ -1712,6 +1712,7 @@
 #define wxLB_OWNERDRAW      0x0100
 #define wxLB_NEEDED_SB      0x0200
 #define wxLB_ALWAYS_SB      0x0400
+#define wxLB_NO_SB          0x0800
 #define wxLB_HSCROLL        wxHSCROLL
 /*  always show an entire number of rows */
 #define wxLB_INT_HEIGHT     0x0800
Index: include/wx/msw/listbox.h
===================================================================
--- include/wx/msw/listbox.h	(revision 61421)
+++ include/wx/msw/listbox.h	(working copy)
@@ -43,7 +43,7 @@
             const wxPoint& pos = wxDefaultPosition,
             const wxSize& size = wxDefaultSize,
             int n = 0, const wxString choices[] = NULL,
-            long style = 0,
+            long style = wxLB_NEEDED_SB,
             const wxValidator& validator = wxDefaultValidator,
             const wxString& name = wxListBoxNameStr)
     {
@@ -53,7 +53,7 @@
             const wxPoint& pos,
             const wxSize& size,
             const wxArrayString& choices,
-            long style = 0,
+            long style = wxLB_NEEDED_SB,
             const wxValidator& validator = wxDefaultValidator,
             const wxString& name = wxListBoxNameStr)
     {
@@ -64,14 +64,14 @@
                 const wxPoint& pos = wxDefaultPosition,
                 const wxSize& size = wxDefaultSize,
                 int n = 0, const wxString choices[] = NULL,
-                long style = 0,
+                long style = wxLB_NEEDED_SB,
                 const wxValidator& validator = wxDefaultValidator,
                 const wxString& name = wxListBoxNameStr);
     bool Create(wxWindow *parent, wxWindowID id,
                 const wxPoint& pos,
                 const wxSize& size,
                 const wxArrayString& choices,
-                long style = 0,
+                long style = wxLB_NEEDED_SB,
                 const wxValidator& validator = wxDefaultValidator,
                 const wxString& name = wxListBoxNameStr);
 
Index: interface/wx/listbox.h
===================================================================
--- interface/wx/listbox.h	(revision 61421)
+++ interface/wx/listbox.h	(working copy)
@@ -39,6 +39,8 @@
         Always show a vertical scrollbar.
     @style{wxLB_NEEDED_SB}
         Only create a vertical scrollbar if needed.
+    @style{wxLB_NO_SB}
+        Don't create vertical scrollbar.
     @style{wxLB_SORT}
         The listbox contents are sorted in alphabetical order.
     @endStyleTable
@@ -100,7 +102,7 @@
               const wxSize& size = wxDefaultSize,
               int n = 0,
               const wxString choices[] = NULL,
-              long style = 0,
+              long style = wxLB_NEEDED_SB,
               const wxValidator& validator = wxDefaultValidator,
               const wxString& name = wxListBoxNameStr);
 
@@ -116,7 +118,7 @@
               const wxPoint& pos,
               const wxSize& size,
               const wxArrayString& choices,
-              long style = 0,
+              long style = wxLB_NEEDED_SB,
               const wxValidator& validator = wxDefaultValidator,
               const wxString& name = wxListBoxNameStr);
 
@@ -134,14 +136,14 @@
                 const wxPoint& pos = wxDefaultPosition,
                 const wxSize& size = wxDefaultSize,
                 int n = 0, const wxString choices[] = NULL,
-                long style = 0,
+                long style = wxLB_NEEDED_SB,
                 const wxValidator& validator = wxDefaultValidator,
                 const wxString& name = wxListBoxNameStr);
     bool Create(wxWindow *parent, wxWindowID id,
                 const wxPoint& pos,
                 const wxSize& size,
                 const wxArrayString& choices,
-                long style = 0,
+                long style = wxLB_NEEDED_SB,
                 const wxValidator& validator = wxDefaultValidator,
                 const wxString& name = wxListBoxNameStr);
     //@}
Index: src/msw/listbox.cpp
===================================================================
--- src/msw/listbox.cpp	(revision 61421)
+++ src/msw/listbox.cpp	(working copy)
@@ -77,6 +77,7 @@
     wxFLAGS_MEMBER(wxLB_HSCROLL)
     wxFLAGS_MEMBER(wxLB_ALWAYS_SB)
     wxFLAGS_MEMBER(wxLB_NEEDED_SB)
+    wxFLAGS_MEMBER(wxLB_NO_SB)
     wxFLAGS_MEMBER(wxLB_SORT)
 
 wxEND_FLAGS( wxListBoxStyle )
@@ -207,10 +208,6 @@
 {
     WXDWORD msStyle = wxControl::MSWGetStyle(style, exstyle);
 
-    // always show the vertical scrollbar if necessary -- otherwise it is
-    // impossible to use the control with the mouse
-    msStyle |= WS_VSCROLL;
-
     // we always want to get the notifications
     msStyle |= LBS_NOTIFY;
 
@@ -226,8 +223,19 @@
     else if ( style & wxLB_EXTENDED )
         msStyle |= LBS_EXTENDEDSEL;
 
-    if ( m_windowStyle & wxLB_ALWAYS_SB )
-        msStyle |= LBS_DISABLENOSCROLL;
+    wxASSERT_MSG
+    (
+        !(style & wxLB_ALWAYS_SB) && !(style & wxLB_NEEDED_SB) || !(style & wxLB_NO_SB),
+        _T( "wxLB_NO_SB style conflicts with one of wxLB_ALWAYS_SB or wxLB_NEEDED_SB stlyes." )
+    );
+
+    if ( !( style & wxLB_NO_SB ) )
+    {
+        msStyle |= WS_VSCROLL;
+        if ( style & wxLB_ALWAYS_SB )
+            msStyle |= LBS_DISABLENOSCROLL;
+    }
+
     if ( m_windowStyle & wxLB_HSCROLL )
         msStyle |= WS_HSCROLL;
     if ( m_windowStyle & wxLB_SORT )
