Index: interface/wx/docview.h
===================================================================
--- interface/wx/docview.h	(revision 73722)
+++ interface/wx/docview.h	(working copy)
@@ -1242,7 +1242,18 @@
 
     /**
         Activate the first view of the document if any.
+        This function is calling the Raise() method of the frame (of the first view).
+        You may need to override the Raise() method to get the desired effect, for instance
+        if your document is living inside its own notebook page you could implement
+        Raise() like this:
 
+        @code
+        void MyNotebookPage::Raise()
+        {
+            wxStaticCast(GetParent(), wxNotebook)->SetSelection(this);
+        }
+        @endcode
+
         @see GetFirstView()
 
         @since 2.9.5
