Index: src/xrc/xmlres.cpp
===================================================================
--- src/xrc/xmlres.cpp	(revision 61771)
+++ src/xrc/xmlres.cpp	(working copy)
@@ -1793,6 +1793,7 @@
     // encoding
     wxFontEncoding enc = wxFONTENCODING_DEFAULT;
     bool hasEncoding = HasParam(wxT("encoding"));
+#if wxUSE_FONTMAP
     if (hasEncoding)
     {
         wxString encoding = GetParamValue(wxT("encoding"));
@@ -1802,6 +1803,7 @@
         if (enc == wxFONTENCODING_SYSTEM)
             enc = wxFONTENCODING_DEFAULT;
     }
+#endif
 
     // is this font based on a system font?
     wxFont font = GetSystemFont(GetParamValue(wxT("sysfont")));
