--- wxWidgets-2.9.4-orig\src\msw\treectrl.cpp	2012-07-09 15:03:07.000000000 +0200
+++ wxWidgets-2.9.4\src\msw\treectrl.cpp	2013-02-08 17:22:26.097884400 +0100
@@ -2768,6 +2768,7 @@
                     if ( HandleMouseEvent(nMsg, x, y, wParam) )
                     {
                         m_htClickedItem.Unset();
+                        m_focusLost = false;
                         break;
                     }
 
@@ -2790,12 +2791,14 @@
                         changedEvent.m_itemOld = htOldItem;
                         (void)HandleTreeEvent(changedEvent);
                     }
+                    m_focusLost = false;
                 }
                 else if ( wParam & MK_SHIFT )
                 {
                     if ( HandleMouseEvent(nMsg, x, y, wParam) )
                     {
                         m_htClickedItem.Unset();
+                        m_focusLost = false;
                         break;
                     }
 
@@ -2846,6 +2849,7 @@
                             (void)HandleTreeEvent(changedEvent);
                         }
                     }
+                    m_focusLost = false;
                 }
                 else // normal click
                 {
@@ -2862,6 +2866,7 @@
                         if ( HandleMouseEvent(nMsg, x, y, wParam) )
                         {
                             m_htClickedItem.Unset();
+                            m_focusLost = false;
                             break;
                         }
 
@@ -2892,6 +2897,7 @@
                             SetFocusedItem(wxTreeItemId(htItem));
                             m_mouseUpDeselect = true;
                         }
+                        m_focusLost = false;
                     }
                     else // click on a single selected item
                     {
@@ -2900,27 +2906,13 @@
                         // proc will start the drag itself if we let have
                         // WM_LBUTTONDOWN
                         m_htClickedItem.Unset();
-
-                        // prevent in-place editing from starting if focus lost
-                        // since previous click
-                        if ( m_focusLost )
-                        {
-                            ClearFocusedItem();
-                            DoSelectItem(wxTreeItemId(htItem));
-                            SetFocusedItem(wxTreeItemId(htItem));
-                        }
-                        else
-                        {
-                            processed = false;
-                        }
+                        processed = false;
                     }
 
                     // reset on any click without Shift
                     m_htSelStart.Unset();
                 }
 
-                m_focusLost = false;
-
                 // we consumed the event so we need to trigger state image
                 // click if needed
                 if ( processed )
@@ -3276,6 +3268,15 @@
 
         case TVN_BEGINLABELEDIT:
             {
+                // prevent in-place editing from starting if focus lost
+                // since previous click
+                if ( m_focusLost )
+                {
+                    m_focusLost = false;
+                    *result = true;
+                    return true;
+                }
+                
                 eventType = wxEVT_COMMAND_TREE_BEGIN_LABEL_EDIT;
                 TV_DISPINFO *info = (TV_DISPINFO *)lParam;
 
