Index: src/msw/button.cpp
===================================================================
--- src/msw/button.cpp	(revision 67159)
+++ src/msw/button.cpp	(working copy)
@@ -1100,15 +1100,22 @@
 void DrawButtonText(HDC hdc,
                     RECT *pRect,
                     const wxString& text,
+                    wxButton *btn,
+                    COLORREF col,
                     int flags)
 {
-    // center text horizontally in any case
-    flags |= DT_CENTER;
+    wxTextColoursChanger changeFg(hdc, col, CLR_INVALID);
+    wxBkModeChanger changeBkMode(hdc, wxBRUSHSTYLE_TRANSPARENT);
 
+    const wxString text = btn->GetLabel();
+
     if ( text.find(wxT('\n')) != wxString::npos )
     {
         // draw multiline label
 
+        // center text horizontally in any case
+        flags |= DT_CENTER;
+
         // first we need to compute its bounding rect
         RECT rc;
         ::CopyRect(&rc, pRect);
@@ -1127,10 +1134,34 @@
     }
     else // single line label
     {
-        // centre text vertically too (notice that we must have DT_SINGLELINE
-        // for DT_VCENTER to work)
+        // no else if () because there may be wxBU_xxx combinations
+        if ( btn->HasFlag( wxBU_LEFT ))
+        {
+            flags |= DT_LEFT;
+        }
+        if ( btn->HasFlag( wxBU_RIGHT ))
+        {
+            flags |= DT_RIGHT;
+        }
+        if ( btn->HasFlag( wxBU_TOP ))
+        {
+            flags |= DT_TOP;
+        }
+        if ( btn->HasFlag( wxBU_BOTTOM ))
+        {
+            flags |= DT_BOTTOM;
+        }
+
+        if ( !btn->HasFlag( wxBU_TOP ) && !btn->HasFlag( wxBU_BOTTOM ))
+            flags |= DT_VCENTER;
+        if ( !btn->HasFlag( wxBU_LEFT ) && !btn->HasFlag( wxBU_RIGHT ))
+            flags |= DT_CENTER;
+
+        // notice that we must have DT_SINGLELINE
+        // for DT_TOP, DT_BOTTOM and DT_VCENTER to work
         ::DrawText(hdc, text.wx_str(), text.length(), pRect,
                    flags | DT_SINGLELINE | DT_VCENTER);
+                   flags | DT_SINGLELINE );
     }
 }
 
@@ -1475,7 +1506,7 @@
             // notice that DT_HIDEPREFIX doesn't work on old (pre-Windows 2000)
             // systems but by happy coincidence ODS_NOACCEL is not used under
             // them neither so DT_HIDEPREFIX should never be used there
-            DrawButtonText(hdc, &rectBtn, GetLabel(),
+            DrawButtonText(hdc, &rectBtn, this,
                            state & ODS_NOACCEL ? DT_HIDEPREFIX : 0);
         }
     }
