Index: samples/console/console.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/samples/console/console.cpp,v
retrieving revision 1.138.2.7
diff -b -u -2 -r1.138.2.7 console.cpp
--- samples/console/console.cpp	2003/06/12 22:55:16	1.138.2.7
+++ samples/console/console.cpp	2003/09/14 14:15:21
@@ -50,5 +50,5 @@
 
 // what to test (in alphabetic order)? uncomment the line below to do all tests
-#define TEST_ALL
+//#define TEST_ALL
 #ifdef TEST_ALL
     #define TEST_ARRAYS
@@ -96,4 +96,5 @@
 #else
     #define TEST_DATETIME
+    #define TEST_INTERACTIVE
 
     static const bool TEST_ALL = FALSE;
@@ -4925,5 +4926,5 @@
 
         wxDateTime dt;
-        const wxChar *p = dt.ParseDate(buf);
+        const wxChar *p = dt.ParseDateTime(buf);
         if ( !p )
         {
@@ -4937,10 +4938,11 @@
         }
 
-        wxPrintf(_T("%s: day %u, week of month %u/%u, week of year %u\n"),
+        wxPrintf(_T("%s: day %u, week of month %u/%u, week of year %u, Full datetime: %s\n"),
                  dt.Format(_T("%b %d, %Y")).c_str(),
                  dt.GetDayOfYear(),
                  dt.GetWeekOfMonth(wxDateTime::Monday_First),
                  dt.GetWeekOfMonth(wxDateTime::Sunday_First),
-                 dt.GetWeekOfYear(wxDateTime::Monday_First));
+                 dt.GetWeekOfYear(wxDateTime::Monday_First),
+                 dt.Format(_T("%Y-%m-%d %H:%M:%S.%l")).c_str());
     }
 
Index: src/common/datetime.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/common/datetime.cpp,v
retrieving revision 1.70.2.4
diff -b -u -2 -r1.70.2.4 datetime.cpp
--- src/common/datetime.cpp	2003/04/06 15:39:36	1.70.2.4
+++ src/common/datetime.cpp	2003/09/14 14:15:26
@@ -3168,9 +3168,41 @@
     wxCHECK_MSG( date, (wxChar *)NULL, _T("NULL pointer in wxDateTime::Parse") );
 
-    // there is a public domain version of getdate.y, but it only works for
-    // English...
-    wxFAIL_MSG(_T("TODO"));
+    // Set Current day and hour, so strings like '14:00' becomes today at 14, not some other random date
+    wxDateTime datDate = wxDateTime::Today();
+    wxDateTime datTime = wxDateTime::Today();
+    //datTime.SetHour(wxDateTime::Now().GetHour());
 
-    return (wxChar *)NULL;
+    const wxChar* pchTime;
+
+    // Try to parse the beginning of the string as a date
+    const wxChar* pchDate = datDate.ParseDate(date);
+
+    // We got a date in the beginning, see if there is a time specified after the date
+    if(pchDate) {
+
+       // Skip spaces, as the ParseTime() function fails on spaces
+        while(*pchDate == wxT(' ') && *pchDate != 0)
+            pchDate++;
+
+        pchTime = datTime.ParseTime(pchDate);
+    }
+
+    // If we didnt have a date in the beginning, check and see if we have a time followed by a date
+    else {
+        pchTime = datTime.ParseTime(date);
+        if(pchTime) {
+            while(*pchTime == wxT(' ') && *pchTime != 0)
+                pchTime++;
+
+            pchDate = datDate.ParseDate(pchTime);
+        }
+    }
+
+    // If we have a date specified, set our own data to the same date
+    if(pchDate || pchTime)
+        Set(datDate.GetDay(), datDate.GetMonth(), datDate.GetYear(), datTime.GetHour(), datTime.GetMinute(), datTime.GetSecond(), datTime.GetMillisecond());
+
+    // Return endpoint of scan
+    return pchDate > pchTime ? pchDate : pchTime;
 }
 

 	  	 
