Index: xh_dlg.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/xrc/xh_dlg.cpp,v
retrieving revision 1.18
diff -b -u -2 -r1.18 xh_dlg.cpp
--- xh_dlg.cpp	2005/01/07 21:33:13	1.18
+++ xh_dlg.cpp	2005/02/28 20:04:14
@@ -48,4 +48,5 @@
     XRC_ADD_STYLE(wxMAXIMIZE_BOX);
     XRC_ADD_STYLE(wxMINIMIZE_BOX);
+    XRC_ADD_STYLE(wxFRAME_SHAPED);
 
     AddWindowStyles();

 	  	 
