diff -ru wxWidgets.orig/include/wx/window.h wxWidgets/include/wx/window.h
--- wxWidgets.orig/include/wx/window.h	2007-11-08 04:53:54.000000000 +0100
+++ wxWidgets/include/wx/window.h	2007-11-08 22:07:42.000000000 +0100
@@ -719,6 +719,7 @@
     virtual bool Validate();
 
         // transfer data between internal and GUI representations
+    virtual bool TransferData(bool save);
     virtual bool TransferDataToWindow();
     virtual bool TransferDataFromWindow();
 
diff -ru wxWidgets.orig/samples/validate/validate.cpp wxWidgets/samples/validate/validate.cpp
--- wxWidgets.orig/samples/validate/validate.cpp	2007-11-08 04:54:22.000000000 +0100
+++ wxWidgets/samples/validate/validate.cpp	2007-11-08 22:32:41.000000000 +0100
@@ -234,11 +234,31 @@
 
 bool MyDialog::TransferDataToWindow()
 {
-    bool r = wxDialog::TransferDataToWindow();
-    // These function calls have to be made here, after the
-    // dialog has been created.
-    text->SetFocus();
-    combobox->SetSelection(0);
-    return r;
+    bool ok = wxDialog::TransferDataToWindow();
+    if (ok)
+    {
+        // These function calls have to be made here (or in TransferData below), 
+        // after the dialog has been created.
+        text->SetFocus();
+        combobox->SetSelection(0);
+    }
+    return ok;
+}
+
+bool MyDialog::TransferData(bool save)
+{
+    bool ok = wxDialog::TransferData(save);
+    if (ok)
+    {
+        if (save)
+        {
+        }
+        else
+        {
+            //text->SetFocus();
+            //combobox->SetSelection(0);
+        }
+    }
+    return ok;
 }
 
diff -ru wxWidgets.orig/samples/validate/validate.h wxWidgets/samples/validate/validate.h
--- wxWidgets.orig/samples/validate/validate.h	2007-11-08 04:54:22.000000000 +0100
+++ wxWidgets/samples/validate/validate.h	2007-11-08 22:12:00.000000000 +0100
@@ -30,9 +30,9 @@
 public:
     MyFrame(wxFrame *frame, const wxString&title, int x, int y, int w, int h);
 
-    void OnQuit(wxCommandEvent& event);
-    void OnTestDialog(wxCommandEvent& event);
-    void OnToggleBell(wxCommandEvent& event);
+    void OnQuit(wxCommandEvent&);
+    void OnTestDialog(wxCommandEvent&);
+    void OnToggleBell(wxCommandEvent&);
 
 private:
     wxListBox *m_listbox;
@@ -48,7 +48,8 @@
             const wxPoint& pos = wxDefaultPosition,
             const wxSize& size = wxDefaultSize,
             const long style = wxDEFAULT_DIALOG_STYLE);
-    bool TransferDataToWindow();
+    virtual bool TransferData(bool save);
+    virtual bool TransferDataToWindow();
     wxTextCtrl *text;
     wxComboBox *combobox;
 };
diff -ru wxWidgets.orig/src/common/wincmn.cpp wxWidgets/src/common/wincmn.cpp
--- wxWidgets.orig/src/common/wincmn.cpp	2007-11-08 04:54:00.000000000 +0100
+++ wxWidgets/src/common/wincmn.cpp	2007-11-08 22:31:42.000000000 +0100
@@ -1513,8 +1513,14 @@
     return true;
 }
 
+bool wxWindowBase::TransferData(bool WXUNUSED(save))
+{
+    return true;
+}
+
 bool wxWindowBase::TransferDataToWindow()
 {
+    bool ok = true;
 #if wxUSE_VALIDATORS
     bool recurse = (GetExtraStyle() & wxWS_EX_VALIDATE_RECURSIVELY) != 0;
 
@@ -1530,7 +1536,8 @@
             wxLog::FlushActive();
 #endif // wxUSE_LOG
 
-            return false;
+            ok = false;
+            break;
         }
 
         if ( recurse )
@@ -1538,17 +1545,19 @@
             if ( !child->TransferDataToWindow() )
             {
                 // warning already given
-                return false;
+                ok = false;
+                break;
             }
         }
     }
 #endif // wxUSE_VALIDATORS
-
-    return true;
+    if (ok) ok = TransferData(false);
+    return ok;
 }
 
 bool wxWindowBase::TransferDataFromWindow()
 {
+    bool ok = true;
 #if wxUSE_VALIDATORS
     bool recurse = (GetExtraStyle() & wxWS_EX_VALIDATE_RECURSIVELY) != 0;
 
@@ -1562,7 +1571,8 @@
             // nop warning here because the application is supposed to give
             // one itself - we don't know here what might have gone wrongly
 
-            return false;
+            ok = false;
+            break;
         }
 
         if ( recurse )
@@ -1570,13 +1580,14 @@
             if ( !child->TransferDataFromWindow() )
             {
                 // warning already given
-                return false;
+                ok = false;
+                break;
             }
         }
     }
 #endif // wxUSE_VALIDATORS
-
-    return true;
+    if (ok) ok = TransferData(true);
+    return ok;
 }
 
 void wxWindowBase::InitDialog()

 	  	 
