diff -bru wxWidgets.orig/samples/text/text.cpp wxWidgets/samples/text/text.cpp
--- wxWidgets.orig/samples/text/text.cpp	2015-01-01 12:41:23.000000000 -0500
+++ wxWidgets/samples/text/text.cpp	2015-01-01 17:10:11.000000000 -0500
@@ -1163,7 +1163,9 @@
     m_textrich->SetStyle(10, 20, *wxBLUE);
     m_textrich->SetStyle(30, 40,
                          wxTextAttr(*wxGREEN, wxNullColour, *wxITALIC_FONT));
-    m_textrich->SetDefaultStyle(wxTextAttr());
+    wxTextAttr attr;
+    attr.SetLeftIndent( 100 );
+    m_textrich->SetDefaultStyle(attr);
     m_textrich->AppendText(wxT("\n\nFirst 10 characters should be in red\n"));
     m_textrich->AppendText(wxT("Next 10 characters should be in blue\n"));
     m_textrich->AppendText(wxT("Next 10 characters should be normal\n"));
diff -bru wxWidgets.orig/src/osx/cocoa/textctrl.mm wxWidgets/src/osx/cocoa/textctrl.mm
--- wxWidgets.orig/src/osx/cocoa/textctrl.mm	2015-01-01 12:42:41.000000000 -0500
+++ wxWidgets/src/osx/cocoa/textctrl.mm	2015-01-03 21:47:05.000000000 -0500
@@ -740,11 +740,14 @@
                                 long end,
                                 const wxTextAttr& style)
 {
+    double mm2pts = (double) 72 / 25.4;
+    NSMutableParagraphStyle *paragraphStyle;
     if ( !m_textView )
         return;
 
     if ( start == -1 && end == -1 )
     {
+//        NSMutableAttributedString *attrString;
         NSMutableDictionary* const
             attrs = [NSMutableDictionary dictionaryWithCapacity:3];
         if ( style.HasFont() )
@@ -753,8 +756,16 @@
             [attrs setValue:style.GetBackgroundColour().OSXGetNSColor() forKey:NSBackgroundColorAttributeName];
         if ( style.HasTextColour() )
             [attrs setValue:style.GetTextColour().OSXGetNSColor() forKey:NSForegroundColorAttributeName];
-
+        if( style.HasLeftIndent() )
+        {
+            paragraphStyle = [[NSMutableParagraphStyle alloc] init];
+            [paragraphStyle setHeadIndent: style.GetLeftIndent() * mm2pts / 10.0];
+            [paragraphStyle setFirstLineHeadIndent: style.GetLeftIndent() * mm2pts / 10.0];
+            [m_textView setDefaultParagraphStyle:paragraphStyle];
+        }
         [m_textView setTypingAttributes:attrs];
+        if( style.HasLeftIndent() )
+            [paragraphStyle release];
     }
     else // Set the attributes just for this range.
     {
@@ -769,6 +780,14 @@
 
         if ( style.HasTextColour() )
             [storage addAttribute:NSForegroundColorAttributeName value:style.GetTextColour().OSXGetNSColor() range:range];
+        if( style.HasLeftIndent() )
+        {
+             paragraphStyle = [[NSMutableParagraphStyle alloc] init];
+             [paragraphStyle setFirstLineHeadIndent: style.GetLeftIndent() * mm2pts / 10.0];
+             [paragraphStyle setHeadIndent: style.GetLeftIndent() * mm2pts / 10.0];
+             [storage addAttribute: NSParagraphStyleAttributeName value: paragraphStyle range: range];
+             [paragraphStyle release];
+        }
     }
 }
 
