diff -uNr wxWidgets-orig/src/msw/mimetype.cpp wxWidgets-new/src/msw/mimetype.cpp
--- wxWidgets-orig/src/msw/mimetype.cpp	2013-07-26 17:30:44.000000000 -0700
+++ wxWidgets-new/src/msw/mimetype.cpp	2013-08-05 14:19:28.000000000 -0700
@@ -282,7 +282,7 @@
         // it's the default value of the key
         if ( key.QueryValue(wxEmptyString, command) ) {
             bool foundFilename = CanonicalizeParams(command);
-
+            
 #if wxUSE_IPC
             // look whether we must issue some DDE requests to the application
             // (and not just launch it)
@@ -291,23 +291,26 @@
             if ( keyDDE.Open(wxRegKey::Read) ) {
                 wxString ddeCommand, ddeServer, ddeTopic;
                 keyDDE.QueryValue(wxEmptyString, ddeCommand);
-                ddeCommand.Replace(wxT("%1"), wxT("%s"));
-
-                wxRegKey keyServer(wxRegKey::HKCR, strKey + wxT("\\Application"));
-                keyServer.QueryValue(wxEmptyString, ddeServer);
-                wxRegKey keyTopic(wxRegKey::HKCR, strKey + wxT("\\Topic"));
-                keyTopic.QueryValue(wxEmptyString, ddeTopic);
-
-                if (ddeTopic.empty())
-                    ddeTopic = wxT("System");
-
-                // HACK: we use a special feature of wxExecute which exists
-                //       just because we need it here: it will establish DDE
-                //       conversation with the program it just launched
-                command.Prepend(wxT("WX_DDE#"));
-                command << wxT('#') << ddeServer
-                        << wxT('#') << ddeTopic
-                        << wxT('#') << ddeCommand;
+                // make sure a DDE command exists for this key
+                if (!ddeCommand.IsEmpty()) {
+                    ddeCommand.Replace(wxT("%1"), wxT("%s"));
+                    
+                    wxRegKey keyServer(wxRegKey::HKCR, strKey + wxT("\\Application"));
+                    keyServer.QueryValue(wxEmptyString, ddeServer);
+                    wxRegKey keyTopic(wxRegKey::HKCR, strKey + wxT("\\Topic"));
+                    keyTopic.QueryValue(wxEmptyString, ddeTopic);
+                    
+                    if (ddeTopic.empty())
+                        ddeTopic = wxT("System");
+                    
+                    // HACK: we use a special feature of wxExecute which exists
+                    //       just because we need it here: it will establish DDE
+                    //       conversation with the program it just launched
+                    command.Prepend(wxT("WX_DDE#"));
+                    command << wxT('#') << ddeServer
+                    << wxT('#') << ddeTopic
+                    << wxT('#') << ddeCommand;
+                }
             }
             else
 #endif // wxUSE_IPC
