Index: include/wx/msw/listctrl.h
===================================================================
--- include/wx/msw/listctrl.h	(Revision 58809)
+++ include/wx/msw/listctrl.h	(Arbeitskopie)
@@ -402,7 +402,7 @@
 
     // get the item attribute, either by quering it for virtual control, or by
     // returning the one previously set using setter methods for a normal one
-    wxListItemAttr *DoGetItemColumnAttr(long item, long column) const;
+    const wxListItemAttr *DoGetItemColumnAttr(long item, long column) const;
 
 
     wxTextCtrl*       m_textCtrl;        // The control used for editing a label
@@ -440,10 +440,10 @@
     virtual int OnGetItemColumnImage(long item, long column) const;
 
     // return the attribute for the item (may return NULL if none)
-    virtual wxListItemAttr *OnGetItemAttr(long item) const;
+    virtual const wxListItemAttr *OnGetItemAttr(long item) const;
 
     // return the attribute for the given item and column (may return NULL if none)
-    virtual wxListItemAttr *OnGetItemColumnAttr(long item, long WXUNUSED(column)) const
+    virtual const wxListItemAttr *OnGetItemColumnAttr(long item, long WXUNUSED(column)) const
     {
         return OnGetItemAttr(item);
     }
Index: src/msw/listctrl.cpp
===================================================================
--- src/msw/listctrl.cpp	(Revision 58809)
+++ src/msw/listctrl.cpp	(Arbeitskopie)
@@ -241,7 +241,7 @@
 // Get the internal data structure
 static wxListItemInternalData *wxGetInternalData(HWND hwnd, long itemId);
 static wxListItemInternalData *wxGetInternalData(const wxListCtrl *ctl, long itemId);
-static wxListItemAttr *wxGetInternalDataAttr(const wxListCtrl *ctl, long itemId);
+static const wxListItemAttr *wxGetInternalDataAttr(const wxListCtrl *ctl, long itemId);
 static void wxDeleteInternalData(wxListCtrl* ctl, long itemId);
 
 
@@ -2765,7 +2765,7 @@
 
 static WXLPARAM HandleItemPrepaint(wxListCtrl *listctrl,
                                    LPNMLVCUSTOMDRAW pLVCD,
-                                   wxListItemAttr *attr)
+                                   const wxListItemAttr *attr)
 {
     if ( !attr )
     {
@@ -2989,7 +2989,7 @@
     return -1;
 }
 
-wxListItemAttr *wxListCtrl::OnGetItemAttr(long WXUNUSED_UNLESS_DEBUG(item)) const
+const wxListItemAttr *wxListCtrl::OnGetItemAttr(long WXUNUSED_UNLESS_DEBUG(item)) const
 {
     wxASSERT_MSG( item >= 0 && item < GetItemCount(),
                   _T("invalid item index in OnGetItemAttr()") );
@@ -2998,7 +2998,7 @@
     return NULL;
 }
 
-wxListItemAttr *wxListCtrl::DoGetItemColumnAttr(long item, long column) const
+const wxListItemAttr *wxListCtrl::DoGetItemColumnAttr(long item, long column) const
 {
     return IsVirtual() ? OnGetItemColumnAttr(item, column)
                        : wxGetInternalDataAttr(this, item);
@@ -3051,7 +3051,7 @@
 }
 
 static
-wxListItemAttr *wxGetInternalDataAttr(const wxListCtrl *ctl, long itemId)
+const wxListItemAttr *wxGetInternalDataAttr(const wxListCtrl *ctl, long itemId)
 {
     wxListItemInternalData *data = wxGetInternalData(ctl, itemId);
 
