*** customtreectrl_old.py	2010-11-27 17:56:43.000000000 -0600
--- customtreectrl.py	2010-11-27 17:54:58.000000000 -0600
***************
*** 1162,1168 ****
          keycode = event.GetKeyCode()
          shiftDown = event.ShiftDown()
  
!         if keycode == wx.WXK_RETURN:
              if shiftDown:
                  event.Skip()
              else:
--- 1162,1168 ----
          keycode = event.GetKeyCode()
          shiftDown = event.ShiftDown()
  
!         if keycode in [wx.WXK_RETURN, wx.WXK_NUMPAD_ENTER]:
              if shiftDown:
                  event.Skip()
              else:
***************
*** 6266,6272 ****
              event.SetEventObject(self)
              self.GetEventHandler().ProcessEvent(event)
                  
!         elif keyCode in [wx.WXK_RETURN, wx.WXK_SPACE]:
  
              if not self.IsItemEnabled(self._current):
                  event.Skip()
--- 6266,6272 ----
              event.SetEventObject(self)
              self.GetEventHandler().ProcessEvent(event)
                  
!         elif keyCode in [wx.WXK_RETURN, wx.WXK_NUMPAD_ENTER, wx.WXK_SPACE]:
  
              if not self.IsItemEnabled(self._current):
                  event.Skip()
