--- src/unix/fswatcher_kqueue.cpp.orig	2011-04-22 02:00:42.000000000 +0700
+++ src/unix/fswatcher_kqueue.cpp	2011-04-22 02:03:44.000000000 +0700
@@ -312,16 +312,8 @@
                 wxArrayInt::iterator changeType = changedFlags.begin();
                 for ( ; it != changedFiles.end(); ++it, ++changeType )
                 {
-                    wxFileName path;
-                    if ( wxDirExists(*it) )
-                    {
-                        path = wxFileName::DirName(w.GetPath() + *it);
-                    }
-                    else
-                    {
-                        path = wxFileName::FileName(w.GetPath() + *it);
-                    }
-
+                    wxString p = w.GetPath() + wxFileName::GetPathSeparator() + *it;
+                    wxFileName path = wxDirExists(p) ? wxFileName::DirName(p) : wxFileName::FileName(p);
                     wxFileSystemWatcherEvent event(*changeType, path, path);
                     SendEvent(event);
                 }
