diff --git a/src/common/dcbase.cpp b/src/common/dcbase.cpp
index 48d719c..8ac893f 100644
--- a/src/common/dcbase.cpp
+++ b/src/common/dcbase.cpp
@@ -584,12 +584,15 @@ wxDCImpl::DoStretchBlit(wxCoord xdest, wxCoord ydest,
     double xscaleOld, yscaleOld;
     GetUserScale(&xscaleOld, &yscaleOld);
     SetUserScale(xscaleOld/xscale, yscaleOld/yscale);
+    wxCoord xoriginOld, yoriginOld;
+    DoGetDeviceOrigin(&xoriginOld, &yoriginOld);
+    SetDeviceOrigin(xoriginOld + xdest, yoriginOld + ydest);
 
-    bool rc = DoBlit(wxCoord(xdest*xscale), wxCoord(ydest*yscale),
-                     wxCoord(dstWidth*xscale), wxCoord(dstHeight*yscale),
+    bool rc = DoBlit(0, 0, srcWidth, srcHeight,
                      source,
                      xsrc, ysrc, rop, useMask, xsrcMask, ysrcMask);
 
+    SetDeviceOrigin(xoriginOld, yoriginOld);
     SetUserScale(xscaleOld, yscaleOld);
 
     return rc;
