diff --git a/src/common/dcbase.cpp b/src/common/dcbase.cpp
index 48d719cbda..15eb7fd093 100644
--- a/src/common/dcbase.cpp
+++ b/src/common/dcbase.cpp
@@ -580,16 +580,23 @@ wxDCImpl::DoStretchBlit(wxCoord xdest, wxCoord ydest,
     // emulate the stretching by modifying the DC scale
     double xscale = (double)srcWidth/dstWidth,
            yscale = (double)srcHeight/dstHeight;
+    wxCoord xdestDev = LogicalToDeviceX(xdest),
+            ydestDev = LogicalToDeviceY(ydest),
+            xdestScaled = wxRound(xdest * xscale),
+            ydestScaled = wxRound(ydest * yscale);
 
     double xscaleOld, yscaleOld;
     GetUserScale(&xscaleOld, &yscaleOld);
     SetUserScale(xscaleOld/xscale, yscaleOld/yscale);
+    wxCoord xoriginOld, yoriginOld;
+    DoGetDeviceOrigin(&xoriginOld, &yoriginOld);
+    SetDeviceOrigin(xdestDev - LogicalToDeviceX(xdestScaled),
+                    ydestDev - LogicalToDeviceY(ydestScaled));
 
-    bool rc = DoBlit(wxCoord(xdest*xscale), wxCoord(ydest*yscale),
-                     wxCoord(dstWidth*xscale), wxCoord(dstHeight*yscale),
-                     source,
-                     xsrc, ysrc, rop, useMask, xsrcMask, ysrcMask);
+    bool rc = DoBlit(xdestScaled, ydestScaled, srcWidth, srcHeight,
+                     source, xsrc, ysrc, rop, useMask, xsrcMask, ysrcMask);
 
+    SetDeviceOrigin(xoriginOld, yoriginOld);
     SetUserScale(xscaleOld, yscaleOld);
 
     return rc;
