Index: include/wx/bookctrl.h
===================================================================
--- include/wx/bookctrl.h	(revision 65931)
+++ include/wx/bookctrl.h	(working copy)
@@ -252,6 +252,16 @@
     // choose the default border for this window
     virtual wxBorder GetDefaultBorder() const { return wxBORDER_NONE; }
 
+    // After the insertion of the page in the method InsertPage, calling this
+    // method set the selection to the given page or the first one if there
+    // is still no selection. The event is send only if bSelect is true, so
+    // if bSelect is false, no unwanted event is send, even with the selection
+    // of the first inserted wxListbook page.
+    void DoSetSelectionAfterInsertion(size_t n,
+                                      wxWindow *page,
+                                      bool bSelect,
+                                      bool bHideUnselectedPage = true);
+
     // set the selection to the given page, sending the events (which can
     // possibly prevent the page change from taking place) if SendEvent flag is
     // included
Index: src/common/bookctrl.cpp
===================================================================
--- src/common/bookctrl.cpp	(revision 65931)
+++ src/common/bookctrl.cpp	(working copy)
@@ -471,6 +471,20 @@
     return nPage;
 }
 
+void wxBookCtrlBase::DoSetSelectionAfterInsertion(size_t n,
+                                                  wxWindow *page,
+                                                  bool bSelect,
+                                                  bool bHideUnselectedPage /*= true*/)
+{
+    if ( bSelect )
+        SetSelection(n);
+    else if ( m_selection == wxNOT_FOUND )
+        ChangeSelection(0);
+
+    if (bHideUnselectedPage && ((int)n != m_selection) )
+        page->Hide();
+}
+
 int wxBookCtrlBase::DoSetSelection(size_t n, int flags)
 {
     wxCHECK_MSG( n < GetPageCount(), wxNOT_FOUND,
Index: src/generic/choicbkg.cpp
===================================================================
--- src/generic/choicbkg.cpp	(revision 65931)
+++ src/generic/choicbkg.cpp	(working copy)
@@ -197,18 +197,8 @@
 
     // some page should be selected: either this one or the first one if there
     // is still no selection
-    int selNew = wxNOT_FOUND;
-    if ( bSelect )
-        selNew = n;
-    else if ( m_selection == wxNOT_FOUND )
-        selNew = 0;
+    DoSetSelectionAfterInsertion(n, page, bSelect);
 
-    if ( selNew != m_selection )
-        page->Hide();
-
-    if ( selNew != wxNOT_FOUND )
-        SetSelection(selNew);
-
     return true;
 }
 
Index: src/generic/listbkg.cpp
===================================================================
--- src/generic/listbkg.cpp	(revision 65931)
+++ src/generic/listbkg.cpp	(working copy)
@@ -345,18 +345,8 @@
 
     // some page should be selected: either this one or the first one if there
     // is still no selection
-    int selNew = wxNOT_FOUND;
-    if ( bSelect )
-        selNew = n;
-    else if ( m_selection == wxNOT_FOUND )
-        selNew = 0;
+    DoSetSelectionAfterInsertion(n, page, bSelect);
 
-    if ( selNew != m_selection )
-        page->Hide();
-
-    if ( selNew != wxNOT_FOUND )
-        SetSelection(selNew);
-
     UpdateSize();
 
     return true;
Index: src/msw/notebook.cpp
===================================================================
--- src/msw/notebook.cpp	(revision 65931)
+++ src/msw/notebook.cpp	(working copy)
@@ -844,15 +844,8 @@
 
     // some page should be selected: either this one or the first one if there
     // is still no selection
-    int selNew = wxNOT_FOUND;
-    if ( bSelect )
-        selNew = nPage;
-    else if ( m_selection == wxNOT_FOUND )
-        selNew = 0;
+    DoSetSelectionAfterInsertion(nPage, pPage, bSelect, false);
 
-    if ( selNew != wxNOT_FOUND )
-        SetSelection(selNew);
-
     InvalidateBestSize();
 
     return true;
Index: src/os2/notebook.cpp
===================================================================
--- src/os2/notebook.cpp	(revision 65931)
+++ src/os2/notebook.cpp	(working copy)
@@ -651,15 +651,8 @@
     //
     // Some page should be selected: either this one or the first one if there is
     // still no selection
     //
-    int nSelNew = wxNOT_FOUND;
 
-    if (bSelect)
-        nSelNew = nPage;
-    else if ( m_selection == wxNOT_FOUND )
-        nSelNew = 0;
+    DoSetSelectionAfterInsertion(nPage, pPage, bSelect, false);
 
-    if (nSelNew != wxNOT_FOUND)
-        SetSelection(nSelNew);
-
     InvalidateBestSize();
 
     return true;
Index: src/osx/notebook.cpp
===================================================================
--- src/osx/notebook_osx.cpp	(revision 65931)
+++ src/osx/notebook_osx.cpp	(working copy)
@@ -242,15 +242,8 @@
 
     // some page should be selected: either this one or the first one if there
     // is still no selection
-    int selNew = wxNOT_FOUND;
-    if ( bSelect )
-        selNew = nPage;
-    else if ( m_selection == wxNOT_FOUND )
-        selNew = 0;
+    DoSetSelectionAfterInsertion(nPage, pPage, bSelect, false);
 
-    if ( selNew != wxNOT_FOUND )
-        SetSelection( selNew );
-
     InvalidateBestSize();
 
     return true;