Index: include/wx/msw/ole/automtn.h
===================================================================
--- include/wx/msw/ole/automtn.h	(revision 66893)
+++ include/wx/msw/ole/automtn.h	(working copy)
@@ -33,7 +33,11 @@
     wxAutomationInstance_UseExistingOnly = 0,
 
     // Create a new instance if there are no existing ones.
-    wxAutomationInstance_CreateIfNeeded = 1
+    wxAutomationInstance_CreateIfNeeded = 1,
+
+    // Do not display wxSysLogError if there are no existing instances
+    // and HR == MK_E_UNAVAILABLE.
+    wxAutomationInstance_GetExisting_FailSilently = 2
 };
 
 /*
Index: interface/wx/msw/ole/automtn.h
===================================================================
--- interface/wx/msw/ole/automtn.h	(revision 66893)
+++ interface/wx/msw/ole/automtn.h	(working copy)
@@ -31,7 +31,13 @@
         instance failed, we should call wxAutomationObject::CreateInstance() to
         create a new one.
      */
-    wxAutomationInstance_CreateIfNeeded = 1
+    wxAutomationInstance_CreateIfNeeded = 1,
+
+    /**
+        Do not show an error message if no existing instance is currently
+        running. All other errors will be reported.
+     */
+    wxAutomationInstance_GetExisting_FailSilently = 2
 };
 
 
Index: src/msw/ole/automtn.cpp
===================================================================
--- src/msw/ole/automtn.cpp	(revision 66893)
+++ src/msw/ole/automtn.cpp	(working copy)
@@ -543,6 +543,11 @@
             if ( m_dispatchPtr )
                 return true;
         }
+        else if ( hr == MK_E_UNAVAILABLE && 
+                flags & wxAutomationInstance_GetExisting_FailSilently )
+        {
+            return false;
+        }
         else
         {
             wxLogSysError(hr,
