diff --git a/include/wx/ffile.h b/include/wx/ffile.h
index 91a7c28ed4..74d092df25 100644
--- a/include/wx/ffile.h
+++ b/include/wx/ffile.h
@@ -76,13 +76,13 @@ public:
   wxFileOffset Length() const;
 
   // simple accessors: note that Eof() and Error() may only be called if
-  // IsOpened()!
+  // IsOpened() otherwise they return false
     // is file opened?
   bool IsOpened() const { return m_fp != NULL; }
     // is end of file reached?
-  bool Eof() const { return feof(m_fp) != 0; }
+  bool Eof() const;
     // has an error occurred?
-  bool Error() const { return ferror(m_fp) != 0; }
+  bool Error() const;
     // get the file name
   const wxString& GetName() const { return m_name; }
     // type such as disk or pipe
diff --git a/src/common/ffile.cpp b/src/common/ffile.cpp
index 1f729c2cc2..ccd5362c5c 100644
--- a/src/common/ffile.cpp
+++ b/src/common/ffile.cpp
@@ -281,4 +281,18 @@ wxFileOffset wxFFile::Length() const
     return wxInvalidOffset;
 }
 
+bool wxFFile::Eof() const
+{
+    wxCHECK_MSG( IsOpened(), false,
+                 wxT("wxFFile::Eof(): file is closed!") );
+    return IsOpened() && feof(m_fp) != 0;
+}
+
+bool wxFFile::Error() const
+{
+    wxCHECK_MSG( IsOpened(), false,
+                 wxT("wxFFile::Error(): file is closed!") );
+    return IsOpened() && ferror(m_fp) != 0;
+}
+
 #endif // wxUSE_FFILE
