Index: src/generic/treectlg.cpp
===================================================================
--- src/generic/treectlg.cpp	(revision 64305)
+++ src/generic/treectlg.cpp	(working copy)
@@ -1626,8 +1626,8 @@
         // into children
         m_anchor->SetHasPlus();
         m_anchor->Expand();
-        CalculatePositions();
     }
+    CalculatePositions();
 
     if (!HasFlag(wxTR_MULTIPLE))
     {
@@ -3618,9 +3618,9 @@
     dc.SetFont( m_normalFont );
 
     dc.SetPen( m_dottedPen );
-    //if(GetImageList() == NULL)
-    // m_lineHeight = (int)(dc.GetCharHeight() + 4);
 
+    m_anchor->CalculateSize(this, dc);
+
     int y = 2;
     CalculateLevel( m_anchor, dc, 0, y ); // start recursion
 }
