Index: include/wx/string.h
===================================================================
--- include/wx/string.h	(revision 53749)
+++ include/wx/string.h	(working copy)
@@ -1677,9 +1677,19 @@
 
       // replace first (or all of bReplaceAll) occurences of substring with
       // another string, returns the number of replacements made
-  size_t Replace(const wxString& strOld,
-                 const wxString& strNew,
+  size_t Replace(const wxString& from,
+                 const wxString& to,
                  bool bReplaceAll = true);
+      // single character versions
+  size_t Replace(const wxUniChar& from, const wxUniChar& to, bool bReplaceAll = true);
+  size_t Replace(char from, char to, bool bReplaceAll = true)
+  {
+      return Replace(wxUniChar(from), wxUniChar(to), bReplaceAll);
+  }
+  size_t Replace(wchar_t from, wchar_t to, bool bReplaceAll = true)
+  {
+      return Replace(wxUniChar(from), wxUniChar(to), bReplaceAll);
+  }
 
     // check if the string contents matches a mask containing '*' and '?'
   bool Matches(const wxString& mask) const;
Index: interface/string.h
===================================================================
--- interface/string.h	(revision 53749)
+++ interface/string.h	(working copy)
@@ -817,13 +817,34 @@
 
     /**
         Replace first (or all) occurrences of substring with another one.
-        @e replaceAll: global replace (default), or only the first occurrence.
-        Returns the number of replacements made.
+
+        @param from
+            Substring to replace.
+        @param to
+            Substring to replace with.
+        @param replaceAll
+            Global replace (default), or only the first occurrence.        
+        @return Returns the number of replacements made.
     */
-    size_t Replace(const wxString& strOld, const wxString& strNew,
+    size_t Replace(const wxString& from, const wxString& to,
                    bool replaceAll = true);
 
     /**
+        Replace first (or all) occurrences of character with another one.
+        
+        @param from
+            Character to replace.
+        @param to
+            Character to replace with.
+        @param replaceAll
+            Global replace (default), or only the first occurrence.        
+        @return Returns the number of replacements made.
+        @since 2.9.0
+    */
+    size_t Replace(const wxUniChar& from, const wxUniChar& to,
+                   bool replaceAll = true);
+
+    /**
         Returns the last @a count characters.
     */
     wxString Right(size_t count) const;
Index: src/common/string.cpp
===================================================================
--- src/common/string.cpp	(revision 53749)
+++ src/common/string.cpp	(working copy)
@@ -1265,6 +1265,31 @@
     return uiCount;
 }
 
+// replace first (or all) occurences of some character with another one
+size_t wxString::Replace(const wxUniChar& from,
+                         const wxUniChar& to, bool bReplaceAll)
+{
+    size_t uiCount = 0;   // count of replacements made
+    for (iterator i = begin(); i != end(); i++)
+    {
+        if (from == *i)
+        {
+            // replace this occurance of the old char with the new one
+            *i = to;
+
+            // increase replace count
+            ++uiCount;
+
+            // stop after the first one?
+            if ( !bReplaceAll )
+            {
+                break;
+            }
+        }
+    }
+    return uiCount;
+}
+
 bool wxString::IsAscii() const
 {
     for ( const_iterator i = begin(); i != end(); ++i )
Index: src/html/winpars.cpp
===================================================================
--- src/html/winpars.cpp	(revision 53749)
+++ src/html/winpars.cpp	(working copy)
@@ -426,7 +426,7 @@
             // we need to substitute spaces for &nbsp; here just like we
             // did in the Whitespace_Normal branch above
             wxString txt2(txt);
-            txt2.Replace(CUR_NBSP_VALUE, ' ');
+            txt2.Replace(CUR_NBSP_VALUE, wxT(' '));
             AddPreBlock(txt2);
         }
         else
Index: tests/strings/strings.cpp
===================================================================
--- tests/strings/strings.cpp	(revision 53749)
+++ tests/strings/strings.cpp	(working copy)
@@ -1,4 +1,4 @@
-///////////////////////////////////////////////////////////////////////////////
+      ///////////////////////////////////////////////////////////////////////////////
 // Name:        tests/strings/strings.cpp
 // Purpose:     wxString unit test
 // Author:      Vadim Zeitlin, Wlodzimierz ABX Skiba
@@ -59,6 +59,7 @@
         CPPUNIT_TEST( CStrDataOperators );
         CPPUNIT_TEST( CStrDataImplicitConversion );
         CPPUNIT_TEST( ExplicitConversion );
+        CPPUNIT_TEST( Replace );
     CPPUNIT_TEST_SUITE_END();
 
     void String();
@@ -89,6 +90,7 @@
     void CStrDataOperators();
     void CStrDataImplicitConversion();
     void ExplicitConversion();
+    void Replace();
 
     DECLARE_NO_COPY_CLASS(StringTestCase)
 };
@@ -809,3 +811,13 @@
     CPPUNIT_ASSERT( CheckStrConstWChar(s, s.wc_str()) );
     CPPUNIT_ASSERT( CheckStrWChar(s, s.wchar_str()) );
 }
+
+void StringTestCase::Replace()
+{
+    wxString s("foo");
+    CPPUNIT_ASSERT( 1 == s.Replace( 'f', 'c') );
+    CPPUNIT_ASSERT( 2 == s.Replace(L'o',L'p') );
+    CPPUNIT_ASSERT( s[0] == 'c' );
+    CPPUNIT_ASSERT( s[1] == 'p' );
+    CPPUNIT_ASSERT( s[2] == 'p' );
+}
