Index: include/wx/generic/dirctrlg.h
===================================================================
--- include/wx/generic/dirctrlg.h	(revision 66313)
+++ include/wx/generic/dirctrlg.h	(working copy)
@@ -189,6 +189,8 @@
     bool ExtractWildcard(const wxString& filterStr, int n, wxString& filter, wxString& description);
 
 private:
+    void PopulateNode(wxTreeItemId node);
+
     bool            m_showHidden;
     wxTreeItemId    m_rootId;
     wxString        m_defaultPath; // Starting path
Index: src/generic/dirctrlg.cpp
===================================================================
--- src/generic/dirctrlg.cpp	(revision 66313)
+++ src/generic/dirctrlg.cpp	(working copy)
@@ -788,7 +788,7 @@
     m_treeCtrl->Thaw();
 }
 
-void wxGenericDirCtrl::ExpandDir(wxTreeItemId parentId)
+void wxGenericDirCtrl::PopulateNode(wxTreeItemId parentId)
 {
     wxDirItemData *data = (wxDirItemData *) m_treeCtrl->GetItemData(parentId);
 
@@ -935,6 +935,12 @@
     }
 }
 
+void wxGenericDirCtrl::ExpandDir(wxTreeItemId parentId)
+{
+    // ExpandDir() will not actually expand the tree node, just populate it
+    PopulateNode(parentId);
+}
+
 void wxGenericDirCtrl::ReCreateTree()
 {
     CollapseDir(m_treeCtrl->GetRootItem());
