diff -uNr wxWidgets-orig/src/osx/cocoa/statbox.mm wxWidgets-mine/src/osx/cocoa/statbox.mm
--- wxWidgets-orig/src/osx/cocoa/statbox.mm	2016-02-28 22:33:38.000000000 +0100
+++ wxWidgets-mine/src/osx/cocoa/statbox.mm	2016-03-15 17:44:05.000000000 +0100
@@ -63,7 +63,7 @@
 wxWidgetImplType* wxWidgetImpl::CreateGroupBox( wxWindowMac* wxpeer,
                                     wxWindowMac* WXUNUSED(parent),
                                     wxWindowID WXUNUSED(id),
-                                    const wxString& WXUNUSED(label),
+                                    const wxString& label,
                                     const wxPoint& pos,
                                     const wxSize& size,
                                     long WXUNUSED(style),
@@ -75,6 +75,7 @@
     [v setContentViewMargins: margin];
     [v sizeToFit];
     wxStaticBoxCocoaImpl* c = new wxStaticBoxCocoaImpl( wxpeer, v );
+    if(label.empty()) [v setTitlePosition:NSNoTitle];
 #if !wxOSX_USE_NATIVE_FLIPPED
     c->SetFlipped(false);
 #endif
