Index: samples/splitter/splitter.cpp
===================================================================
--- samples/splitter/splitter.cpp	(revision 63921)
+++ samples/splitter/splitter.cpp	(working copy)
@@ -38,6 +38,8 @@
 
 #include "wx/splitter.h"
 #include "wx/dcmirror.h"
+#include "wx/sizer.h"
+#include "wx/stattext.h"
 
 #ifndef __WXMSW__
     #include "../sample.xpm"
@@ -108,7 +110,7 @@
     void OnUpdateUIUnsplit(wxUpdateUIEvent& event);
 
 private:
-    wxScrolledWindow *m_left, *m_right;
+    wxWindow *m_left,*m_right;
 
     wxSplitterWindow* m_splitter;
     wxWindow *m_replacewindow;
@@ -143,6 +145,8 @@
 
     virtual void OnDraw(wxDC& dc);
 
+    virtual bool AcceptsFocusFromKeyboard() const { return false; }
+
 private:
     bool m_mirror;
 
@@ -255,24 +259,33 @@
     SetMenuBar(menuBar);
 
     menuBar->Check(SPLIT_LIVE, true);
+
     m_splitter = new MySplitterWindow(this);
 
     m_splitter->SetSashGravity(1.0);
 
-#if 1
-    m_left = new MyCanvas(m_splitter, true);
-    m_left->SetBackgroundColour(*wxRED);
-    m_left->SetScrollbars(20, 20, 5, 5);
-    m_left->SetCursor(wxCursor(wxCURSOR_MAGNIFIER));
+    m_left = new wxPanel(m_splitter);
 
-    m_right = new MyCanvas(m_splitter, false);
-    m_right->SetBackgroundColour(*wxCYAN);
-    m_right->SetScrollbars(20, 20, 5, 5);
-#else // for testing kbd navigation inside the splitter
-    m_left = new wxTextCtrl(m_splitter, wxID_ANY, wxT("first text"));
-    m_right = new wxTextCtrl(m_splitter, wxID_ANY, wxT("second text"));
-#endif
+        wxBoxSizer *leftSizer = new wxBoxSizer(wxVERTICAL);
 
+            leftSizer->Add(new MyCanvas(m_left, true), 1, wxEXPAND);
+
+//            leftSizer->Add(new wxStaticText(m_left, wxID_ANY, "The code poet"), 1, wxALL | wxEXPAND, 5);
+//            leftSizer->Add(new wxStaticText(m_left, wxID_ANY, "is IN"), 1, wxALL | wxEXPAND, 5);
+//            leftSizer->Add(new wxStaticText(m_left, wxID_ANY, "5¢"), 1, wxALL | wxEXPAND, 5);
+
+    m_left->SetSizer(leftSizer);
+
+    m_right = new wxPanel(m_splitter);
+
+        wxBoxSizer *rightSizer = new wxBoxSizer(wxVERTICAL);
+
+            rightSizer->Add(new wxTextCtrl(m_right, wxID_ANY, "The code poet"), 1, wxALL | wxEXPAND, 5);
+            rightSizer->Add(new wxTextCtrl(m_right, wxID_ANY, "is OUT"), 1, wxALL | wxEXPAND, 5);
+            rightSizer->Add(new wxTextCtrl(m_right, wxID_ANY, "5¢"), 1, wxALL | wxEXPAND, 5);
+
+    m_right->SetSizer(rightSizer);
+
     // you can also do this to start with a single window
 #if 0
     m_right->Show(false);
