diff --git a/samples/popup/popup.cpp b/samples/popup/popup.cpp
index 09a80b9df2..a920b617c9 100644
--- a/samples/popup/popup.cpp
+++ b/samples/popup/popup.cpp
@@ -376,6 +376,7 @@ MyFrame::MyFrame(const wxString& title)
     wxButton *button1 = new wxButton( panel, Minimal_StartSimplePopup, "Show simple popup", wxPoint(20,20) );
     wxButton *button2 = new wxButton( panel, Minimal_StartScrolledPopup, "Show scrolled popup", wxPoint(20,70) );
 
+
     m_logWin = new wxTextCtrl( panel, wxID_ANY, wxEmptyString, wxDefaultPosition,
                                wxDefaultSize, wxTE_MULTILINE );
     m_logWin->SetEditable(false);
@@ -385,6 +386,8 @@ MyFrame::MyFrame(const wxString& title)
 
     wxBoxSizer *topSizer = new wxBoxSizer( wxVERTICAL );
     topSizer->Add( button1, 0, wxALL, 5 );
+	wxTextCtrl * txt = new wxTextCtrl(panel, wxID_ANY);
+    topSizer->Add( txt, 0, wxALL, 5 );
     topSizer->Add( button2, 0, wxALL, 5 );
     topSizer->Add( m_logWin, 1, wxEXPAND|wxALL, 5 );
 
