diff --git a/samples/propgrid/propgrid_minimal.cpp b/samples/propgrid/propgrid_minimal.cpp
--- a/samples/propgrid/propgrid_minimal.cpp
+++ b/samples/propgrid/propgrid_minimal.cpp
@@ -20,9 +20,11 @@ public:
     void OnAction(wxCommandEvent& event);
     void OnPropertyGridChange(wxPropertyGridEvent& event);
     void OnPropertyGridChanging(wxPropertyGridEvent& event);
+    void Construct();
 
 private:
     wxPropertyGrid* m_pg;
+    wxPGProperty* m_prop;
     wxDECLARE_EVENT_TABLE();
 };
 
@@ -45,14 +47,48 @@ MyFrame::MyFrame(wxWindow* parent)
                         wxPG_SPLITTER_AUTO_CENTER |
                         wxPG_BOLD_MODIFIED );
     m_pg = pg;
+    m_prop = NULL;
 
-    pg->Append( new wxStringProperty("String Property", wxPG_LABEL) );
-    pg->Append( new wxIntProperty("Int Property", wxPG_LABEL) );
-    pg->Append( new wxBoolProperty("Bool Property", wxPG_LABEL) );
+    Construct();
 
     SetSize(400, 600);
 }
 
+void MyFrame::Construct()
+{
+//    m_pg->Clear();
+    if (m_prop)
+    {
+        m_pg->DeleteProperty(m_prop);
+        m_prop = NULL;
+    }
+
+/*    for (int n=0; n < 10; ++n)
+    {
+        wxSafeYield();
+    }*/
+
+/*    m_pg->Append( new wxStringProperty("String Property", wxPG_LABEL) );
+    m_pg->Append( new wxIntProperty("Int Property", wxPG_LABEL) );
+    m_pg->Append( new wxBoolProperty("Bool Property", wxPG_LABEL) );
+
+   static const wxChar* enum_prop_labels[] = { wxT("One Item"),
+        wxT("Another Item"), wxT("One More"), wxT("This Is Last"), NULL };    static long enum_prop_values[] = { 40, 80, 120, 160 };
+
+    m_pg->Append( new wxEnumProperty(wxT("EnumProperty"),wxPG_LABEL,
+        enum_prop_labels, enum_prop_values, 80 ) );*/
+
+    wxPGChoices choices;
+    for (int n=0; n < 5; ++n)
+    {
+        choices.Add(wxString::Format("Item %i", n+1), n);
+    }
+    wxFlagsProperty* pflags;
+    pflags = (wxFlagsProperty*)m_pg->Append( new wxFlagsProperty(_("Flags"), wxPG_LABEL, choices, 0) );
+    wxPrintf("MyFrame::Construct:  pflags %p parent %p grid %p\n", pflags, pflags->GetParent(), m_pg);
+    m_prop = pflags;
+}
+
 void MyFrame::OnPropertyGridChange(wxPropertyGridEvent &event)
 {
     wxPGProperty* p = event.GetProperty();
@@ -66,6 +102,9 @@ void MyFrame::OnPropertyGridChange(wxPropertyGridEvent &event)
     {
         wxLogVerbose("OnPropertyGridChange(NULL)");
     }
+
+    Construct();
+    CallAfter(&MyFrame::Construct);
 }
 
 void MyFrame::OnPropertyGridChanging(wxPropertyGridEvent &event)
@@ -73,6 +112,8 @@ void MyFrame::OnPropertyGridChanging(wxPropertyGridEvent &event)
     wxPGProperty* p = event.GetProperty();
 
     wxLogVerbose("OnPropertyGridChanging(%s)", p->GetName().c_str());
+
+    //Construct();
 }
 
 void MyFrame::OnAction(wxCommandEvent &)
