diff --git a/samples/propgrid/propgrid_minimal.cpp b/samples/propgrid/propgrid_minimal.cpp
index 8b0ac77..2a0f009 100644
--- a/samples/propgrid/propgrid_minimal.cpp
+++ b/samples/propgrid/propgrid_minimal.cpp
@@ -20,6 +20,7 @@ public:
     void OnAction(wxCommandEvent& event);
     void OnPropertyGridChange(wxPropertyGridEvent& event);
     void OnPropertyGridChanging(wxPropertyGridEvent& event);
+    void Construct();
 
 private:
     wxPropertyGrid* m_pg;
@@ -46,13 +47,20 @@ MyFrame::MyFrame(wxWindow* parent)
                         wxPG_BOLD_MODIFIED );
     m_pg = pg;
 
-    pg->Append( new wxStringProperty("String Property", wxPG_LABEL) );
-    pg->Append( new wxIntProperty("Int Property", wxPG_LABEL) );
-    pg->Append( new wxBoolProperty("Bool Property", wxPG_LABEL) );
+    Construct();
 
     SetSize(400, 600);
 }
 
+void MyFrame::Construct()
+{    
+    m_pg->Clear();
+
+    m_pg->Append( new wxStringProperty("String Property", wxPG_LABEL) );
+    m_pg->Append( new wxIntProperty("Int Property", wxPG_LABEL) );
+    m_pg->Append( new wxBoolProperty("Bool Property", wxPG_LABEL) );
+}
+
 void MyFrame::OnPropertyGridChange(wxPropertyGridEvent &event)
 {
     wxPGProperty* p = event.GetProperty();
@@ -66,6 +74,8 @@ void MyFrame::OnPropertyGridChange(wxPropertyGridEvent &event)
     {
         wxLogVerbose("OnPropertyGridChange(NULL)");
     }
+
+    //Construct();
 }
 
 void MyFrame::OnPropertyGridChanging(wxPropertyGridEvent &event)
@@ -73,6 +83,8 @@ void MyFrame::OnPropertyGridChanging(wxPropertyGridEvent &event)
     wxPGProperty* p = event.GetProperty();
 
     wxLogVerbose("OnPropertyGridChanging(%s)", p->GetName().c_str());
+
+    Construct();
 }
 
 void MyFrame::OnAction(wxCommandEvent &)
