Index: src/msw/textctrl.cpp
===================================================================
--- src/msw/textctrl.cpp	(revision 66867)
+++ src/msw/textctrl.cpp	(working copy)
@@ -225,6 +225,7 @@
     m_privateContextMenu = NULL;
     m_updatesCount = -1;
     m_isNativeCaretShown = true;
+    m_isIMEComposition = false;
 }
 
 wxTextCtrl::~wxTextCtrl()
@@ -1859,6 +1860,14 @@
             }
             break;
 
+        case WM_IME_STARTCOMPOSITION:
+            m_isIMEComposition = true;
+            break;
+
+        case WM_IME_ENDCOMPOSITION:
+            m_isIMEComposition = false;
+            break;
+
 #if wxUSE_MENUS
         case WM_SETCURSOR:
             // rich text controls seem to have a bug and don't change the
Index: include/wx/msw/textctrl.h
===================================================================
--- include/wx/msw/textctrl.h	(revision 66867)
+++ include/wx/msw/textctrl.h	(working copy)
@@ -279,6 +279,8 @@
 
     bool m_isNativeCaretShown;
 
+    bool m_isIMEComposition;
+
 #if wxUSE_INKEDIT && wxUSE_RICHEDIT
     int  m_isInkEdit;
 #endif
