Index: dircmn.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWidgets/src/common/dircmn.cpp,v
retrieving revision 1.16
diff -u -2 -r1.16 dircmn.cpp
--- dircmn.cpp	2004/05/25 16:59:35	1.16
+++ dircmn.cpp	2004/08/13 16:23:57
@@ -100,5 +100,5 @@
     {
         wxString dirname;
-        for ( bool cont = GetFirst(&dirname, _T(""), wxDIR_DIRS | wxDIR_HIDDEN);
+        for ( bool cont = GetFirst(&dirname, _T(""), wxDIR_DIRS | (flags & wxDIR_HIDDEN) );
               cont;
               cont = cont && GetNext(&dirname) )

 	  	 
