Index: src/generic/grid.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/generic/grid.cpp,v
retrieving revision 1.279
diff -u -r1.279 grid.cpp
--- src/generic/grid.cpp	2003/10/16 10:27:55	1.279
+++ src/generic/grid.cpp	2003/11/07 03:29:14
@@ -7478,7 +7478,7 @@
 
 bool wxGrid::CanEnableCellControl() const
 {
-    return m_editable && !IsCurrentCellReadOnly();
+    return (m_numCols > 0) && (m_numRows > 0) && m_editable && !IsCurrentCellReadOnly();
 }
 
 bool wxGrid::IsCellEditControlEnabled() const

 	  	 
