diff -Naur minimal/minimal.cpp maxsizeproblem/minimal.cpp
--- minimal/minimal.cpp	2009-09-04 07:55:44.000000000 +0200
+++ maxsizeproblem/minimal.cpp	2009-11-29 16:41:57.062500000 +0100
@@ -16,10 +16,10 @@
 // ----------------------------------------------------------------------------
 // headers
 // ----------------------------------------------------------------------------
- 
+
 // For compilers that support precompilation, includes "wx/wx.h".
 #include "wx/wxprec.h"
- 
+
 #ifdef __BORLANDC__
     #pragma hdrstop
 #endif
@@ -172,6 +172,15 @@
     CreateStatusBar(2);
     SetStatusText("Welcome to wxWidgets!");
 #endif // wxUSE_STATUSBAR
+
+	wxButton *TheButton = new wxButton(this, wxID_HIGHEST + 1, _("filler text filler text"));
+	TheButton->SetMinSize(TheButton->GetBestSize() * 2);
+	TheButton->SetMaxSize(TheButton->GetBestSize() * 4); // <-- this seems to make no difference
+	wxBoxSizer *TheSizer = new wxBoxSizer(wxVERTICAL);
+	TheSizer->Add(TheButton, 1, wxEXPAND);
+	//TheSizer->AddStretchSpacer(); // <-- doesn't help
+	SetSizer(TheSizer);
+	TheSizer->SetSizeHints(this);
 }
 
 
