Index: src/common/docmdi.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/common/docmdi.cpp,v
retrieving revision 1.16
diff -b -u -2 -r1.16 docmdi.cpp
--- src/common/docmdi.cpp	2002/07/22 14:46:41	1.16
+++ src/common/docmdi.cpp	2003/01/03 14:53:26
@@ -88,12 +88,34 @@
 END_EVENT_TABLE()
 
+void wxDocMDIChildFrame::Init()
+{
+  m_childDocument = (wxDocument*)  NULL;
+  m_childView = (wxView*) NULL;
+}
+
+wxDocMDIChildFrame::wxDocMDIChildFrame()
+{
+    Init();
+}
+
 wxDocMDIChildFrame::wxDocMDIChildFrame(wxDocument *doc, wxView *view, wxMDIParentFrame *frame, wxWindowID  id,
-  const wxString& title, const wxPoint& pos, const wxSize& size, long style, const wxString& name):
-    wxMDIChildFrame(frame, id, title, pos, size, style, name)
+  const wxString& title, const wxPoint& pos, const wxSize& size, long style, const wxString& name)
+{
+  Init();
+  Create(doc, view, frame, id, title, pos, size, style, name);
+}
+
+bool wxDocMDIChildFrame::Create(wxDocument *doc, wxView *view, wxMDIParentFrame *frame, wxWindowID  id,
+  const wxString& title, const wxPoint& pos, const wxSize& size, long style, const wxString& name)
 {
   m_childDocument = doc;
   m_childView = view;
+    if (wxMDIChildFrame::Create(frame, id, title, pos, size, style, name)) {
   if (view)
     view->SetFrame(this);
+        return TRUE;
+    }
+
+    return FALSE;
 }
 
Index: include/wx/docmdi.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/docmdi.h,v
retrieving revision 1.11
diff -b -u -2 -r1.11 docmdi.h
--- include/wx/docmdi.h	2003/01/02 23:37:26	1.11
+++ include/wx/docmdi.h	2003/01/03 14:53:26
@@ -60,4 +60,5 @@
 {
 public:
+    wxDocMDIChildFrame();
     wxDocMDIChildFrame(wxDocument *doc, wxView *view, wxMDIParentFrame *frame, wxWindowID id,
         const wxString& title, const wxPoint& pos = wxDefaultPosition, const wxSize& size = wxDefaultSize,
@@ -65,4 +66,14 @@
     ~wxDocMDIChildFrame();
 
+    bool Create(wxDocument *doc,
+                wxView *view,
+                wxMDIParentFrame *frame,
+                wxWindowID id,
+                const wxString& title,
+                const wxPoint& pos = wxDefaultPosition,
+                const wxSize& size = wxDefaultSize,
+                long type = wxDEFAULT_FRAME_STYLE,
+                const wxString& name = wxFrameNameStr);
+
     // Extend event processing to search the view's event table
     virtual bool ProcessEvent(wxEvent& event);
@@ -78,4 +89,5 @@
     
 protected:
+    void Init();
     wxDocument*       m_childDocument;
     wxView*           m_childView;

 	  	 
