 tests/controls/textentrytest.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/controls/textentrytest.cpp b/tests/controls/textentrytest.cpp
index 052cdb5..f5bfa48 100644
--- a/tests/controls/textentrytest.cpp
+++ b/tests/controls/textentrytest.cpp
@@ -194,7 +194,7 @@ void TextEntryTestCase::Editable()
     {
         return;
     }
-#endif // __WGTKK__
+#endif // __WXGTK__
 
     wxTextEntry * const entry = GetTestEntry();
     wxWindow * const window = GetTestWindow();
