diff --git a/src/html/helpctrl.cpp b/src/html/helpctrl.cpp
index 1d19b0d..0669eba 100644
--- a/src/html/helpctrl.cpp
+++ b/src/html/helpctrl.cpp
@@ -113,6 +113,11 @@ void wxHtmlHelpController::OnCloseFrame(wxCloseEvent& evt)
     if ( m_helpWindow )
         m_helpWindow->SetController(NULL);
     m_helpWindow = NULL;
+
+    // If we are setting the m_helpDialog to NULL then we should destroy it.
+    if ( m_helpDialog && !(m_FrameStyle & wxHF_EMBEDDED) )
+        m_helpDialog->Destroy();
+
     m_helpDialog = NULL;
     m_helpFrame = NULL;
 }

diff --git a/src/html/helpctrl.cpp b/src/html/helpctrl.cpp
index 0669eba..6ad41d3 100644
--- a/src/html/helpctrl.cpp
+++ b/src/html/helpctrl.cpp
@@ -406,6 +406,7 @@ void wxHtmlHelpController::MakeModalIfNeeded()
         else if (dialog && (m_FrameStyle & wxHF_MODAL))
         {
             dialog->ShowModal();
+            dialog->Close();
         }
     }
 }