--- a/src/propgrid/propgrid.cpp
+++ b/src/propgrid/propgrid.cpp
@@ -5798,23 +5798,14 @@ void wxPropertyGrid::OnIdle( wxIdleEvent& WXUNUSED(event) )
 
     //
     // Resolve pending property removals
-    if ( m_deletedProperties.size() > 0 )
+    // (Note that arrays are changed at every operation.)
+    while ( !m_deletedProperties.empty() )
     {
-        wxArrayPGProperty& arr = m_deletedProperties;
-        for ( unsigned int i=0; i<arr.size(); i++ )
-        {
-            DeleteProperty(arr[i]);
-        }
-        arr.clear();
+        DeleteProperty(m_deletedProperties[0]);
     }
-    if ( m_removedProperties.size() > 0 )
+    while ( !m_removedProperties.empty() )
     {
-        wxArrayPGProperty& arr = m_removedProperties;
-        for ( unsigned int i=0; i<arr.size(); i++ )
-        {
-            RemoveProperty(arr[i]);
-        }
-        arr.clear();
+        RemoveProperty(m_removedProperties[0]);
     }
 }
 
--- a/src/propgrid/propgridpagestate.cpp
+++ b/src/propgrid/propgridpagestate.cpp
@@ -286,18 +286,32 @@ void wxPropertyGridPageState::DoClear()
         m_selection.clear();
     }
 
-    m_regularArray.Empty();
-    if ( m_abcArray )
-        m_abcArray->Empty();
+    if ( m_pPropGrid && m_pPropGrid->m_processedEvent )
+    {
+        wxPropertyGridIterator it;
+        for ( it = m_pPropGrid->GetIterator(wxPG_ITERATE_ALL);
+              !it.AtEnd();
+              it++ )
+        {
+            wxPGProperty* p = *it;
+            DoDelete(p, true);
+        }
+    }
+    else
+    {
+        m_regularArray.Empty();
+        if ( m_abcArray )
+            m_abcArray->Empty();
 
-    m_dictName.clear();
+        m_dictName.clear();
 
-    m_currentCategory = NULL;
-    m_lastCaptionBottomnest = 1;
-    m_itemsAdded = 0;
+        m_currentCategory = NULL;
+        m_lastCaptionBottomnest = 1;
+        m_itemsAdded = 0;
 
-    m_virtualHeight = 0;
-    m_vhCalcPending = 0;
+        m_virtualHeight = 0;
+        m_vhCalcPending = 0;
+    }
 }
 
 // -----------------------------------------------------------------------
@@ -1952,10 +1966,31 @@ void wxPropertyGridPageState::DoDelete( wxPGProperty* item, bool doDelete )
     item->m_parent = NULL;
 
     // We can actually delete it now
+    // Item is also removed from the respective list of pending operations.
     if ( doDelete )
+    {
+        wxArrayPGProperty::iterator it =
+            std::find(pg->m_deletedProperties.begin(), pg->m_deletedProperties.end(), item);
+        while ( it != pg->m_deletedProperties.end() )
+        {
+            pg->m_deletedProperties.erase(it);
+            it = std::find(pg->m_deletedProperties.begin(), pg->m_deletedProperties.end(), item);
+        }
+
         delete item;
+    }
     else
+    {
+        wxArrayPGProperty::iterator it =
+            std::find(pg->m_removedProperties.begin(), pg->m_removedProperties.end(), item);
+        while ( it != pg->m_removedProperties.end() )
+        {
+            pg->m_removedProperties.erase(it);
+            it = std::find(pg->m_removedProperties.begin(), pg->m_removedProperties.end(), item);
+        }
+
         item->OnDetached(this, pg);
+    }
 
     m_itemsAdded = 1; // Not a logical assignment (but required nonetheless).
 
