Index: include/wx/msw/tooltip.h
===================================================================
--- include/wx/msw/tooltip.h	(revision 3387)
+++ include/wx/msw/tooltip.h	(working copy)
@@ -60,7 +60,7 @@
     static void RelayEvent(WXMSG *msg);
 
     // add a window to the tooltip control
-    void Add(WXHWND hwnd);
+    void AddOtherWindow(WXHWND hwnd);
 
     // remove any tooltip from the window
     static void Remove(WXHWND hwnd, unsigned int id, const wxRect& rc);
@@ -72,7 +72,7 @@
 
 private:
     // Adds a window other than our main m_window to this tooltip.
-    void DoAddOtherWindow(WXHWND hWnd);
+    void DoAddHWND(WXHWND hWnd);
 
     // Perform the specified operation for the given window only.
     void DoSetTip(WXHWND hWnd);
Index: src/msw/combobox.cpp
===================================================================
--- src/msw/combobox.cpp	(revision 3388)
+++ src/msw/combobox.cpp	(working copy)
@@ -665,7 +665,7 @@
     wxChoice::DoSetToolTip(tip);
 
     if ( tip && !HasFlag(wxCB_READONLY) )
-        tip->Add(GetEditHWND());
+        tip->AddOtherWindow(GetEditHWND());
 }
 
 #endif // wxUSE_TOOLTIPS
Index: src/msw/radiobox.cpp
===================================================================
--- src/msw/radiobox.cpp	(revision 3388)
+++ src/msw/radiobox.cpp	(working copy)
@@ -498,7 +498,7 @@
     // we have already checked for the item to be valid in wxRadioBoxBase
     const HWND hwndRbtn = (*m_radioButtons)[item];
     if ( tooltip != NULL )
-        tooltip->Add(hwndRbtn);
+        tooltip->AddOtherWindow(hwndRbtn);
     else // unset the tooltip
         wxToolTip::Remove(hwndRbtn, 0, wxRect(0,0,0,0));
         // the second parameter can be zero since it's ignored by Remove()
Index: src/msw/spinctrl.cpp
===================================================================
--- src/msw/spinctrl.cpp	(revision 3388)
+++ src/msw/spinctrl.cpp	(working copy)
@@ -604,7 +604,7 @@
     wxSpinButton::DoSetToolTip(tip);
 
     if ( tip )
-        tip->Add(m_hwndBuddy);
+        tip->AddOtherWindow(m_hwndBuddy);
 }
 
 #endif // wxUSE_TOOLTIPS
Index: src/msw/tooltip.cpp
===================================================================
--- src/msw/tooltip.cpp	(revision 3379)
+++ src/msw/tooltip.cpp	(working copy)
@@ -374,17 +374,17 @@
     DoForAllWindows(&wxToolTip::DoRemove);
 }
 
-void wxToolTip::Add(WXHWND hWnd)
+void wxToolTip::AddOtherWindow(WXHWND hWnd)
 {
     if ( !m_others )
         m_others = new wxToolTipOtherWindows;
 
     m_others->push_back(hWnd);
 
-    DoAddOtherWindow(hWnd);
+    DoAddHWND(hWnd);
 }
 
-void wxToolTip::DoAddOtherWindow(WXHWND hWnd)
+void wxToolTip::DoAddHWND(WXHWND hWnd)
 {
     HWND hwnd = (HWND)hWnd;
 
@@ -502,7 +502,7 @@
     // add the window itself
     if ( m_window )
     {
-        Add(m_window->GetHWND());
+        DoAddHWND(m_window->GetHWND());
     }
 #if !defined(__WXUNIVERSAL__)
     // and all of its subcontrols (e.g. radio buttons in a radiobox) as well
@@ -526,7 +526,7 @@
             // must have it by now!
             wxASSERT_MSG( hwnd, wxT("no hwnd for subcontrol?") );
 
-            Add((WXHWND)hwnd);
+            AddOtherWindow((WXHWND)hwnd);
         }
     }
 #endif // !defined(__WXUNIVERSAL__)
