dIndex: src/osx/carbon/toolbar.cpp
===================================================================
--- src/osx/carbon/toolbar.cpp	(revision 63591)
+++ src/osx/carbon/toolbar.cpp	(working copy)
@@ -926,17 +926,22 @@
 
 wxToolBar::~wxToolBar()
 {
-    CFIndex count = CFGetRetainCount( m_macToolbar ) ;
-    // Leopard seems to have one refcount more, so we cannot check reliably at the moment
-    if ( UMAGetSystemVersion() < 0x1050 )
+#if wxOSX_USE_NATIVE_TOOLBAR
+    if (m_macToolbar)
     {
-        if ( count != 1 )
+        CFIndex count = CFGetRetainCount( m_macToolbar ) ;
+        // Leopard seems to have one refcount more, so we cannot check reliably at the moment
+        if ( UMAGetSystemVersion() < 0x1050 )
         {
-            wxFAIL_MSG("Reference count of native control was not 1 in wxToolBar destructor");
+            if ( count != 1 )
+            {
+                wxFAIL_MSG("Reference count of native control was not 1 in wxToolBar destructor");
+            }
         }
+        CFRelease( (HIToolbarRef)m_macToolbar );
+        m_macToolbar = NULL;
     }
-    CFRelease( (HIToolbarRef)m_macToolbar );
-    m_macToolbar = NULL;
+#endif // wxOSX_USE_NATIVE_TOOLBAR
 }
 
 bool wxToolBar::Show( bool show )
