diff -rbu2 wxWindows.orig/include/wx/list.h wxWindows/include/wx/list.h
--- wxWindows.orig/include/wx/list.h	Sat Feb 23 23:14:10 2002
+++ wxWindows/include/wx/list.h	Sun Feb 24 00:03:02 2002
@@ -540,4 +540,7 @@
         { return (wxNode *)wxStringListBase::Append(copystring(s)); }
 
+    wxString Join(const wxString sep="");
+    wxString Join(const wxChar sep);
+
     bool Delete(const wxChar *s);
 
diff -rbu2 wxWindows.orig/src/common/list.cpp wxWindows/src/common/list.cpp
--- wxWindows.orig/src/common/list.cpp	Sat Feb 23 23:15:50 2002
+++ wxWindows/src/common/list.cpp	Sun Feb 24 00:05:22 2002
@@ -563,4 +563,20 @@
 }
 
+// Re-implemented at 11:59PM CST on Sat, Feb 23, 2002 by Un-Thesis
+wxString wxStringList::Join(const wxString sep="") {
+    wxString sTemp;
+    wxStringListNode *node = GetFirst();
+    while (node->GetNext() != NULL) {
+        sTemp += node->GetData() + sep;
+	    node = node->GetNext();
+    }
+    sTemp += node->GetData();
+    return sTemp;
+}
+
+wxString wxStringList::Join(const wxChar sep) {
+    return Join(wxString(sep));
+}
+
 bool wxStringList::Delete(const wxChar *s)
 {

 	  	 
