diff -rbu2 wxWindows.orig\include\wx\defs.h wxWindows\include\wx\defs.h
--- wxWindows.orig\include\wx\defs.h	Wed Apr 17 06:47:58 2002
+++ wxWindows\include\wx\defs.h	Tue Apr 23 15:07:35 2002
@@ -178,8 +178,10 @@
     // define boolean constants: don't use true/false here as not all compilers
     // support them
-    #undef TRUE
-    #undef FALSE
+#ifndef TRUE
     #define TRUE  ((bool)1)
+#endif
+#ifndef FALSE
     #define FALSE ((bool)0)
+#endif
 #else // !__cplusplus
     // the definitions above don't work for C sources

 	  	 
