Index: window.mm
===================================================================
--- window.mm	(revision 64681)
+++ window.mm	(working copy)
@@ -553,8 +553,23 @@
                 
             if ( isMouseScrollEvent )
             {
-                deltaX = [nsEvent deviceDeltaX];
-                deltaY = [nsEvent deviceDeltaY];
+                SInt32 scroll_delta_x = 0, scroll_delta_y = 0;
+                OSErr err = noErr;
+                err = GetEventParameter( cEvent, kEventParamMouseWheelSmoothHorizontalDelta, 
+                                        typeSInt32, NULL, sizeof(scroll_delta_x), NULL, 
+                                        &scroll_delta_x);
+                if (err == noErr) 
+                {
+                    deltaX = scroll_delta_x;
+                }
+                err = noErr;
+                err = GetEventParameter( cEvent, kEventParamMouseWheelSmoothVerticalDelta, 
+                                        typeSInt32, NULL, sizeof(scroll_delta_y), NULL, 
+                                        &scroll_delta_y);
+                if (err == noErr) 
+                {
+                    deltaY = scroll_delta_y;
+                }
             }
             else
             {
