Index: src/generic/treectlg.cpp
===================================================================
--- src/generic/treectlg.cpp	(revision 1893)
+++ src/generic/treectlg.cpp	(working copy)
@@ -3626,7 +3626,8 @@
            event.RightDown() ||
            event.LeftDClick() ||
            event.Dragging() ||
-           ((event.Moving() || event.RightUp()) && m_isDragging)) )
+           event.RightUp() ||
+           (event.Moving() && m_isDragging)) )
     {
         event.Skip();
 
@@ -3744,7 +3745,17 @@
         }
         else
         {
-            event.Skip();
+            if(event.RightUp() && item == NULL)
+            {
+                wxContextMenuEvent evtCtx(
+                                          wxEVT_CONTEXT_MENU,
+                                          GetId(),
+                                          ClientToScreen(event.GetPosition()));
+                evtCtx.SetEventObject(this);
+                GetEventHandler()->ProcessEvent(evtCtx);
+            }
+            else
+                event.Skip();
         }
     }
     else
@@ -3780,13 +3791,13 @@
             wxTreeEvent
                 nevent(wxEVT_TREE_ITEM_RIGHT_CLICK,  this, item);
             nevent.m_pointDrag = CalcScrolledPosition(pt);
-            event.Skip(!GetEventHandler()->ProcessEvent(nevent));
+            bool handled = GetEventHandler()->ProcessEvent(nevent);
 
             // Consistent with MSW (for now), send the ITEM_MENU *after*
             // the RIGHT_CLICK event. TODO: This behaviour may change.
             wxTreeEvent nevent2(wxEVT_TREE_ITEM_MENU,  this, item);
             nevent2.m_pointDrag = CalcScrolledPosition(pt);
-            GetEventHandler()->ProcessEvent(nevent2);
+            event.Skip(!(handled || GetEventHandler()->ProcessEvent(nevent2)));
         }
         else if ( event.MiddleDown() )
         {
