Index: htmlcell.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/html/htmlcell.cpp,v
retrieving revision 1.80
diff -c -r1.80 htmlcell.cpp
*** htmlcell.cpp	2003/09/18 22:44:16	1.80
--- htmlcell.cpp	2003/12/04 16:22:39
***************
*** 366,372 ****
          dc.SetBackgroundMode(wxTRANSPARENT);
          dc.SetTextForeground(fg);
          dc.SetTextBackground(bg);
!         dc.SetBackground(wxBrush(bg, wxSOLID));
      }
  }
  
--- 366,373 ----
          dc.SetBackgroundMode(wxTRANSPARENT);
          dc.SetTextForeground(fg);
          dc.SetTextBackground(bg);
! 		if (bg.Ok())
! 			dc.SetBackground(wxBrush(bg, wxSOLID));
      }
  }
  

 	  	 
