Index: filename.cpp
===================================================================
RCS file: /home/wxcvs/wxWindows/src/common/filename.cpp,v
retrieving revision 1.99
diff -b -u -2 -r1.99 filename.cpp
--- filename.cpp    25 Aug 2002 18:26:31 -0000  1.99
+++ filename.cpp    28 Aug 2002 00:38:15 -0000
@@ -972,6 +1115,6 @@
     // get cwd only once - small time saving
     wxString cwd = wxGetCwd();
-    Normalize(wxPATH_NORM_ALL, cwd, format);
-    fnBase.Normalize(wxPATH_NORM_ALL, cwd, format);
+    Normalize(wxPATH_NORM_ALL & ~wxPATH_NORM_CASE, cwd, format);
+    fnBase.Normalize(wxPATH_NORM_ALL & ~wxPATH_NORM_CASE, cwd, format);

     bool withCase = IsCaseSensitive(format);
@@ -1030,8 +1173,9 @@
     // get cwd only once - small time saving
     wxString cwd = wxGetCwd();
-    fn1.Normalize(wxPATH_NORM_ALL, cwd, format);
-    fn2.Normalize(wxPATH_NORM_ALL, cwd, format);
+    fn1.Normalize(wxPATH_NORM_ALL & ~wxPATH_NORM_CASE, cwd, format);
+    fn2.Normalize(wxPATH_NORM_ALL & ~wxPATH_NORM_CASE, cwd, format);

@@ -1363,4 +1507,10 @@
         WIN32_FIND_DATA findFileData;
         HANDLE hFind;
+
+        if ( HasVolume() )
+          pathOut = GetVolume() +
+                    GetVolumeSeparator(wxPATH_DOS) +
+                    GetPathSeparator(wxPATH_DOS);
+        else
         pathOut = wxEmptyString;

@@ -1381,5 +1531,6 @@
                 continue;

-            if ( tmpPath.Last() == wxT(':') )
+            // can't see this being necessary? MF
+            if ( tmpPath.Last() == GetVolumeSeparator(wxPATH_DOS) )
             {
                 // Can't pass a drive and root dir to FindFirstFile,
@@ -1393,6 +1544,10 @@
             if (hFind == INVALID_HANDLE_VALUE)
             {
-                // Error: return immediately with the original path
-                return path;
+                // Error: most likely reason is that path doesn't exist, so
+                // append any unprocessed parts and return
+                for ( i += 1; i < count; i++ )
+                  tmpPath += wxFILE_SEP_PATH + dirs[i];
+
+                return tmpPath;
             }

 	  	 
