Index: src/generic/grid.cpp
===================================================================
RCS file: /home/wxcvs/wxWindows/src/generic/grid.cpp,v
retrieving revision 1.207
diff -c -r1.207 grid.cpp
*** src/generic/grid.cpp	24 Aug 2002 21:40:43 -0000	1.207
--- src/generic/grid.cpp	29 Aug 2002 16:09:19 -0000
***************
*** 900,906 ****
      if ( !HasRange() )
      {
          int keycode = (int) event.KeyCode();
!         if ( isdigit(keycode) || keycode == '+' || keycode == '-' )
          {
              wxGridCellTextEditor::StartingKey(event);
  
--- 900,920 ----
      if ( !HasRange() )
      {
          int keycode = (int) event.KeyCode();
!         if ( isdigit(keycode) || keycode == '+' || keycode == '-'
! 	     || keycode ==  WXK_NUMPAD0
! 	     || keycode ==  WXK_NUMPAD1
! 	     || keycode ==  WXK_NUMPAD2
! 	     || keycode ==  WXK_NUMPAD3
! 	     || keycode ==  WXK_NUMPAD4
! 	     || keycode ==  WXK_NUMPAD5
! 	     || keycode ==  WXK_NUMPAD6
! 	     || keycode ==  WXK_NUMPAD7
! 	     || keycode ==  WXK_NUMPAD8
! 	     || keycode ==  WXK_NUMPAD9
! 	     || keycode ==  WXK_ADD
! 	     || keycode ==  WXK_NUMPAD_ADD
! 	     || keycode ==  WXK_SUBTRACT
! 	     || keycode ==  WXK_NUMPAD_SUBTRACT)
          {
              wxGridCellTextEditor::StartingKey(event);
  
***************
*** 1013,1019 ****
  {
      int keycode = (int)event.KeyCode();
      if ( isdigit(keycode) ||
!          keycode == '+' || keycode == '-' || keycode == '.' )
      {
          wxGridCellTextEditor::StartingKey(event);
  
--- 1027,1047 ----
  {
      int keycode = (int)event.KeyCode();
      if ( isdigit(keycode) ||
!          keycode == '+' || keycode == '-' || keycode == '.'
! 	 || keycode ==  WXK_NUMPAD0
! 	 || keycode ==  WXK_NUMPAD1
! 	 || keycode ==  WXK_NUMPAD2
! 	 || keycode ==  WXK_NUMPAD3
! 	 || keycode ==  WXK_NUMPAD4
! 	 || keycode ==  WXK_NUMPAD5
! 	 || keycode ==  WXK_NUMPAD6
! 	 || keycode ==  WXK_NUMPAD7
! 	 || keycode ==  WXK_NUMPAD8
! 	 || keycode ==  WXK_NUMPAD9
! 	 || keycode ==  WXK_ADD
! 	 || keycode ==  WXK_NUMPAD_ADD
! 	 || keycode ==  WXK_SUBTRACT
! 	 || keycode ==  WXK_NUMPAD_SUBTRACT)
      {
          wxGridCellTextEditor::StartingKey(event);
  

 	  	 
