Index: minimal.cpp
===================================================================
--- minimal.cpp	(revision 71965)
+++ minimal.cpp	(working copy)
@@ -67,6 +67,7 @@
     // event handlers (these functions should _not_ be virtual)
     void OnQuit(wxCommandEvent& event);
     void OnAbout(wxCommandEvent& event);
+    void OnChangeTitle(wxCommandEvent& event);
 
 private:
     // any class wishing to process wxWidgets events must use this macro
@@ -86,7 +87,9 @@
     // it is important for the id corresponding to the "About" command to have
     // this standard value as otherwise it won't be handled properly under Mac
     // (where it is special and put into the "Apple" menu)
-    Minimal_About = wxID_ABOUT
+    Minimal_About = wxID_ABOUT,
+
+    Minimal_ChangeTitle,
 };
 
 // ----------------------------------------------------------------------------
@@ -99,6 +102,7 @@
 BEGIN_EVENT_TABLE(MyFrame, wxFrame)
     EVT_MENU(Minimal_Quit,  MyFrame::OnQuit)
     EVT_MENU(Minimal_About, MyFrame::OnAbout)
+    EVT_BUTTON(Minimal_ChangeTitle, MyFrame::OnChangeTitle)
 END_EVENT_TABLE()
 
 // Create a new application object: this macro will allow wxWidgets to create
@@ -172,6 +176,11 @@
     CreateStatusBar(2);
     SetStatusText("Welcome to wxWidgets!");
 #endif // wxUSE_STATUSBAR
+
+    wxButton* button = new wxButton(this, Minimal_ChangeTitle, wxT("Change Title"));
+    wxBoxSizer* mainSizer = new wxBoxSizer(wxVERTICAL);
+    mainSizer->Add(button, 1, wxEXPAND);
+    SetSizer(mainSizer);
 }
 
 
@@ -183,6 +192,11 @@
     Close(true);
 }
 
+void MyFrame::OnChangeTitle(wxCommandEvent& WXUNUSED(event))
+{
+    SetTitle(wxT("New Title"));
+}
+
 void MyFrame::OnAbout(wxCommandEvent& WXUNUSED(event))
 {
     wxMessageBox(wxString::Format
