Index: include/wx/aui/auibook.h
===================================================================
--- include/wx/aui/auibook.h	(revision 3410)
+++ include/wx/aui/auibook.h	(working copy)
@@ -95,6 +95,7 @@
 public:
     wxWindow* window;     // page's associated window
     wxString caption;     // caption displayed on the tab
+    wxString tooltip;     // tooltip displayed when hovering tab title
     wxBitmap bitmap;      // tab's bitmap
     wxRect rect;          // tab's hit rectangle
     bool active;          // true if the page is currently active
@@ -295,6 +296,9 @@
     bool SetPageText(size_t page, const wxString& text);
     wxString GetPageText(size_t pageIdx) const;
 
+    bool SetPageToolTip(size_t page, const wxString& text);
+    wxString GetPageToolTip(size_t pageIdx) const;
+
     bool SetPageBitmap(size_t page, const wxBitmap& bitmap);
     wxBitmap GetPageBitmap(size_t pageIdx) const;
 
Index: interface/wx/aui/auibook.h
===================================================================
--- interface/wx/aui/auibook.h	(revision 3427)
+++ interface/wx/aui/auibook.h	(working copy)
@@ -220,6 +220,12 @@
     wxString GetPageText(size_t page) const;
 
     /**
+        Returns the tooltip for the tab label of the page.
+        @since 2.9.4
+    */
+    wxString GetPageToolTip(size_t pageIdx) const;
+
+    /**
         Returns the currently selected page.
     */
     int GetSelection() const;
@@ -308,7 +314,14 @@
     */
     bool SetPageText(size_t page, const wxString& text);
 
+
     /**
+        Sets the tooltip displayed when hovering the tab label of the page.
+        @since 2.9.4
+    */
+    bool SetPageToolTip(size_t page, const wxString& text);
+
+    /**
         Sets the font for drawing selected tab labels.
     */
     void SetSelectedFont(const wxFont& font);
Index: samples/aui/auidemo.cpp
===================================================================
--- samples/aui/auidemo.cpp	(revision 3427)
+++ samples/aui/auidemo.cpp	(working copy)
@@ -1644,6 +1644,7 @@
    wxBitmap page_bmp = wxArtProvider::GetBitmap(wxART_NORMAL_FILE, wxART_OTHER, wxSize(16,16));
 
    ctrl->AddPage(CreateHTMLCtrl(ctrl), wxT("Welcome to wxAUI") , false, page_bmp);
+   ctrl->SetPageToolTip (0, wxT("Welcome to wxAUI (tooltip to be displayed on page title)"));
 
    wxPanel *panel = new wxPanel( ctrl, wxID_ANY );
    wxFlexGridSizer *flex = new wxFlexGridSizer( 4, 2, 0, 0 );
@@ -1682,6 +1683,7 @@
 
    ctrl->AddPage( new wxTextCtrl( ctrl, wxID_ANY, wxT("Some more text"),
                 wxDefaultPosition, wxDefaultSize, wxTE_MULTILINE|wxNO_BORDER) , wxT("wxTextCtrl 7 (longer title)") );
+   ctrl->SetPageToolTip (ctrl->GetPageCount()-1, wxT("wxTextCtrl 7: and the tooltip message can be even longer!"));
 
    ctrl->AddPage( new wxTextCtrl( ctrl, wxID_ANY, wxT("Some more text"),
                 wxDefaultPosition, wxDefaultSize, wxTE_MULTILINE|wxNO_BORDER) , wxT("wxTextCtrl 8") );
Index: src/aui/auibook.cpp
===================================================================
--- src/aui/auibook.cpp	(revision 3410)
+++ src/aui/auibook.cpp	(working copy)
@@ -1237,7 +1237,19 @@
         }
     }
 
+    wxWindow* wnd = NULL;
+    if (evt.Moving() && TabHitTest(evt.m_x, evt.m_y, &wnd))
+    {
+        wxString tooltip_text(m_pages[GetIdxFromWindow(wnd)].tooltip);
 
+        // If the text changes, set it else, keep old, to avoid
+        // 'moving tooltip' effect
+        if (GetToolTipText() != tooltip_text)
+            SetToolTip (tooltip_text);
+    }
+    else
+        UnsetToolTip();
+
     if (!evt.LeftIsDown() || m_clickPt == wxDefaultPosition)
         return;
 
@@ -2126,6 +2138,37 @@
     return page_info.caption;
 }
 
+bool wxAuiNotebook::SetPageToolTip(size_t page_idx, const wxString& text)
+{
+    if (page_idx >= m_tabs.GetPageCount())
+        return false;
+
+    // update our own tab catalog
+    wxAuiNotebookPage& page_info = m_tabs.GetPage(page_idx);
+    page_info.tooltip = text;
+
+    wxAuiTabCtrl* ctrl;
+    int ctrl_idx;
+    if (FindTab(page_info.window, &ctrl, &ctrl_idx))
+    {
+        wxAuiNotebookPage& info = ctrl->GetPage(ctrl_idx);
+        info.tooltip = text;
+    }
+
+    // NB: we don't update the tooltip if it is already being displayed, it
+    //  typically never happens, no need to code that
+    return true;
+}
+
+wxString wxAuiNotebook::GetPageToolTip(size_t page_idx) const
+{
+    if (page_idx >= m_tabs.GetPageCount())
+        return wxEmptyString;
+
+    const wxAuiNotebookPage& page_info = m_tabs.GetPage(page_idx);
+    return page_info.tooltip;
+}
+
 bool wxAuiNotebook::SetPageBitmap(size_t page_idx, const wxBitmap& bitmap)
 {
     if (page_idx >= m_tabs.GetPageCount())
