 samples/minimal/minimal.cpp | 61 +++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 61 insertions(+)

diff --git a/samples/minimal/minimal.cpp b/samples/minimal/minimal.cpp
index a78e462..8cf7011 100644
--- a/samples/minimal/minimal.cpp
+++ b/samples/minimal/minimal.cpp
@@ -64,6 +64,13 @@ public:
     // ctor(s)
     MyFrame(const wxString& title);
 
+    void OnContextMenuF(wxContextMenuEvent&);
+    void OnContextMenuP(wxContextMenuEvent&);
+    void OnMenuOpenF(wxMenuEvent& event);
+    void OnMenuOpenP(wxMenuEvent& event);
+    void OnMenuHighlightF(wxMenuEvent& event);
+    void OnMenuHighlightP(wxMenuEvent& event);
+
     // event handlers (these functions should _not_ be virtual)
     void OnQuit(wxCommandEvent& event);
     void OnAbout(wxCommandEvent& event);
@@ -172,6 +179,60 @@ MyFrame::MyFrame(const wxString& title)
     CreateStatusBar(2);
     SetStatusText("Welcome to wxWidgets!");
 #endif // wxUSE_STATUSBAR
+
+    wxPanel* panel = new wxPanel(this);
+    Connect(wxEVT_CONTEXT_MENU, wxContextMenuEventHandler(MyFrame::OnContextMenuF));
+    panel->Connect(wxEVT_CONTEXT_MENU, wxContextMenuEventHandler(MyFrame::OnContextMenuP), NULL, this);
+    Connect(wxEVT_MENU_OPEN, wxMenuEventHandler(MyFrame::OnMenuOpenF));
+    panel->Connect(wxEVT_MENU_OPEN, wxMenuEventHandler(MyFrame::OnMenuOpenP), NULL, this);
+    Connect(wxEVT_MENU_HIGHLIGHT, wxMenuEventHandler(MyFrame::OnMenuHighlightF));
+    panel->Connect(wxEVT_MENU_HIGHLIGHT, wxMenuEventHandler(MyFrame::OnMenuHighlightP), NULL, this);
+}
+
+
+void MyFrame::OnContextMenuF(wxContextMenuEvent& event)
+{
+    wxLogMessage("wxFrame handler for wxEVT_CONTEXT_MENU");
+    event.Skip();
+
+    wxMenu m;
+    m.Append(-1, "test");
+    PopupMenu(&m);
+}
+
+
+void MyFrame::OnContextMenuP(wxContextMenuEvent& event)
+{
+    wxLogMessage("wxPanel handler for wxEVT_CONTEXT_MENU");
+    event.Skip();
+}
+
+
+void MyFrame::OnMenuOpenF(wxMenuEvent& event)
+{
+    wxLogMessage("wxFrame handler for wxEVT_MENU_OPEN");
+    event.Skip();
+}
+
+
+void MyFrame::OnMenuOpenP(wxMenuEvent& event)
+{
+    wxLogMessage("wxPanel handler for wxEVT_MENU_OPEN");
+    event.Skip();
+}
+
+
+void MyFrame::OnMenuHighlightF(wxMenuEvent& event)
+{
+    wxLogMessage("wxFrame handler for wxEVT_MENU_HIGHLIGHT, %04x", event.GetId());
+    event.Skip();
+}
+
+
+void MyFrame::OnMenuHighlightP(wxMenuEvent& event)
+{
+    wxLogMessage("wxPanel handler for wxEVT_MENU_HIGHLIGHT");
+    event.Skip();
 }
 
 
