diff --git a/include/wx/aui/auibar.h b/include/wx/aui/auibar.h
index 52e2525..ca54fe4 100644
--- a/include/wx/aui/auibar.h
+++ b/include/wx/aui/auibar.h
@@ -473,7 +473,7 @@ public:
                 wxWindowID id = wxID_ANY,
                 const wxPoint& pos = wxDefaultPosition,
                 const wxSize& size = wxDefaultSize,
-                long style = wxAUI_TB_DEFAULT_STYLE);
+                long style = -1);
 
     virtual void SetWindowStyleFlag(long style);
 
diff --git a/src/aui/auibar.cpp b/src/aui/auibar.cpp
index f2af2eb..c04f9a0 100644
--- a/src/aui/auibar.cpp
+++ b/src/aui/auibar.cpp
@@ -830,6 +830,7 @@ void wxAuiToolBar::Init()
     m_overflowVisible = false;
     m_overflowState = 0;
     m_orientation = wxHORIZONTAL;
+    m_windowStyle = -1;
 }
 
 bool wxAuiToolBar::Create(wxWindow* parent,
@@ -838,7 +839,15 @@ bool wxAuiToolBar::Create(wxWindow* parent,
                            const wxSize& size,
                            long style)
 {
-    style = style|wxBORDER_NONE;
+    if (style == -1) {
+        if (m_windowStyle == -1) {
+            style = wxAUI_TB_DEFAULT_STYLE | wxBORDER_NONE;
+        } else {
+            style = m_windowStyle | wxBORDER_NONE;
+        }
+    } else if (style > -1) {
+        style = style | wxBORDER_NONE;
+    }
 
     if (!wxControl::Create(parent, id, pos, size, style))
         return false;
