Index: configure.in
===================================================================
RCS file: /home/wxcvs/wxWindows/configure.in,v
retrieving revision 1.576
diff -b -u -2 -r1.576 configure.in
--- configure.in	7 Jun 2002 00:34:01 -0000	1.576
+++ configure.in	8 Jun 2002 14:42:42 -0000
@@ -4545,4 +4545,8 @@
 fi
 
+if test "$TOOLKIT" = "MSW"; then
+    SAMPLES_SUBDIRS="$SAMPLES_SUBDIRS taskbar"
+fi
+
 if test "$wxUSE_PROTOCOL" = "yes"; then
     if test "$wxUSE_SOCKETS" != "yes"; then
Index: samples/configure.in
===================================================================
RCS file: /home/wxcvs/wxWindows/samples/configure.in,v
retrieving revision 1.24
diff -b -u -2 -r1.24 configure.in
--- samples/configure.in	18 May 2002 23:36:55 -0000	1.24
+++ samples/configure.in	8 Jun 2002 14:42:43 -0000
@@ -79,4 +79,5 @@
             splitter/Makefile
             statbar/Makefile
+	    taskbar/Makefile
             text/Makefile
             thread/Makefile

 	  	 
