Index: wxWindows/src/generic/listctrl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/generic/listctrl.cpp,v
retrieving revision 1.305
diff -b -u -2 -r1.305 listctrl.cpp
--- wxWindows/src/generic/listctrl.cpp	2003/09/26 23:47:28	1.305
+++ wxWindows/src/generic/listctrl.cpp	2003/10/06 23:18:28
@@ -4646,4 +4646,6 @@
     m_headerWin = (wxListHeaderWindow*) NULL;
 
+    m_headerHeight = 0;
+
     if ( !(style & wxLC_MASK_TYPE) )
     {
Index: wxWindows/src/generic/imaglist.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/generic/imaglist.cpp,v
retrieving revision 1.32
diff -b -u -2 -r1.32 imaglist.cpp
--- wxWindows/src/generic/imaglist.cpp	2003/08/09 12:46:01	1.32
+++ wxWindows/src/generic/imaglist.cpp	2003/10/06 23:18:28
@@ -46,4 +46,5 @@
 wxGenericImageList::~wxGenericImageList()
 {
+    (void)RemoveAll();
 }
 
Index: wxWindows/src/generic/dirctrlg.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/generic/dirctrlg.cpp,v
retrieving revision 1.75
diff -b -u -2 -r1.75 dirctrlg.cpp
--- wxWindows/src/generic/dirctrlg.cpp	2003/09/24 00:41:02	1.75
+++ wxWindows/src/generic/dirctrlg.cpp	2003/10/06 23:18:29
@@ -1610,14 +1610,15 @@
     }
 
-    wxBitmap tmpBmp;
-    tmpBmp.CopyFromIcon(ic);
-    wxImage img = tmpBmp.ConvertToImage();
+    wxBitmap bmp;
+    bmp.CopyFromIcon(ic);
 
     int id = m_smallImageList->GetImageCount();
-    if (img.GetWidth() == 16 && img.GetHeight() == 16)
-        m_smallImageList->Add(wxBitmap(img));
+    if ((bmp.GetWidth() == 16) && (bmp.GetHeight() == 16))
+        m_smallImageList->Add(bmp);
     else
     {
-        if (img.GetWidth() != 32 || img.GetHeight() != 32)
+        wxImage img = bmp.ConvertToImage();
+
+        if ((img.GetWidth() != 32) || (img.GetHeight() != 32))
             m_smallImageList->Add(CreateAntialiasedBitmap(CutEmptyBorders(img).Rescale(32, 32)));
         else
Index: wxWindows/src/gtk/clipbrd.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/gtk/clipbrd.cpp,v
retrieving revision 1.53
diff -b -u -2 -r1.53 clipbrd.cpp
--- wxWindows/src/gtk/clipbrd.cpp	2003/08/09 12:46:08	1.53
+++ wxWindows/src/gtk/clipbrd.cpp	2003/10/06 23:18:29
@@ -274,4 +274,5 @@
 {
     m_open = FALSE;
+    m_waiting = FALSE;
 
     m_ownsClipboard = FALSE;

 	  	 
