diff --git a/include/wx/log.h b/include/wx/log.h
index be0a245..663dc6a 100644
--- a/include/wx/log.h
+++ b/include/wx/log.h
@@ -1487,11 +1487,6 @@ WXDLLIMPEXP_BASE const wxChar* wxSysErrorMsg(unsigned long nErrCode = 0);
 
 #define wxDEFINE_EMPTY_LOG_FUNCTION2(level, argclass)                       \
     WX_DEFINE_VARARG_FUNC_NOP(wxLog##level, 2, (argclass, const wxFormatString&)) \
-    WX_WATCOM_OR_MINGW_ONLY_CODE(                                           \
-        WX_DEFINE_VARARG_FUNC_NOP(wxLog##level, 2, (argclass, const char*)) \
-        WX_DEFINE_VARARG_FUNC_NOP(wxLog##level, 2, (argclass, const wchar_t*)) \
-        WX_DEFINE_VARARG_FUNC_NOP(wxLog##level, 2, (argclass, const wxCStrData&)) \
-    )                                                                       \
     inline void wxVLog##level(argclass WXUNUSED(arg),                       \
                               const wxFormatString& WXUNUSED(format),       \
                               va_list WXUNUSED(argptr)) {}
