diff -ur ../wxWidgets/src/mac/carbon/dnd.cpp ./src/mac/carbon/dnd.cpp
--- ../wxWidgets/src/mac/carbon/dnd.cpp	2006-03-15 19:20:50.000000000 -0800
+++ ./src/mac/carbon/dnd.cpp	2006-03-15 19:30:51.000000000 -0800
@@ -428,11 +428,9 @@
 #if !TARGET_CARBON // TODO
     ev = (EventRecord*) wxTheApp->MacGetCurrentEvent();
 #else
-    {
-        EventRecord rec;
-        ev = &rec;
-        wxMacConvertEventToRecord( (EventRef) wxTheApp->MacGetCurrentEvent(), &rec );
-    }
+    EventRecord rec;
+    ev = &rec;
+    wxMacConvertEventToRecord( (EventRef) wxTheApp->MacGetCurrentEvent(), &rec );
 #endif
 
     const short dragRegionOuterBoundary = 10;

 	  	 
