Index: src/common/gifdecod.cpp
===================================================================
--- src/common/gifdecod.cpp	(revision 59961)
+++ src/common/gifdecod.cpp	(working copy)
@@ -757,9 +757,20 @@
             pimg->w = buf[4] + 256 * buf[5];
             pimg->h = buf[6] + 256 * buf[7];
 
-            if ( anim && ((pimg->w == 0) || (pimg->w > (unsigned int)m_szAnimation.GetWidth()) ||
-                  (pimg->h == 0) || (pimg->h > (unsigned int)m_szAnimation.GetHeight())) )
-                return wxGIF_INVFORMAT;
+            if (anim)
+            {
+                // For the first frame, we may still repair dimensions
+                if (m_nFrames == 0)
+                {
+                    if (pimg->left + pimg->w > (unsigned int)m_szAnimation.GetWidth())
+                        m_szAnimation.SetWidth(pimg->left + pimg->w);
+                    if (pimg->top + pimg->h > (unsigned int)m_szAnimation.GetHeight())
+                        m_szAnimation.SetHeight(pimg->top + pimg->h);
+                }
+                if ((pimg->w == 0) || (pimg->w > (unsigned int)m_szAnimation.GetWidth()) || 
+                    (pimg->h == 0) || (pimg->h > (unsigned int)m_szAnimation.GetHeight()))
+                    return wxGIF_INVFORMAT;
+            }
 
             interl = ((buf[8] & 0x40)? 1 : 0);
             size = pimg->w * pimg->h;
