--- wxWindows-2.4.1/contrib/src/ogl/bmpshape.cpp	2002-12-18 17:40:10.000000000 +0100
+++ 2.40/contrib/src/ogl/bmpshape.cpp	2003-11-19 15:47:35.000000000 +0100
@@ -53,12 +53,10 @@
   if (!m_bitmap.Ok())
     return;
 
-  wxMemoryDC tempDC;
-  tempDC.SelectObject(m_bitmap);
   double x, y;
   x = WXROUND(m_xpos - m_bitmap.GetWidth() / 2.0);
   y = WXROUND(m_ypos - m_bitmap.GetHeight() / 2.0);
-  dc.Blit((long) x, (long) y, m_bitmap.GetWidth(), m_bitmap.GetHeight(), &tempDC, 0, 0);
+  dc.DrawBitmap(m_bitmap, x, y, true);
 }
 
 void wxBitmapShape::SetSize(double w, double h, bool recursive)

 	  	 
