--- wxWidgets.orig/samples/combo/combo.cpp	Mon Jul 17 17:28:15 2006
+++ wxWidgets/samples/combo/combo.cpp	Tue Jul 18 19:48:03 2006
@@ -229,6 +229,8 @@
     {
 
-        // If item is selected or even, use the default rendering.
+        // If item is selected or even, or we are painting the
+        // combo control itself, use the default rendering.
         if ( GetVListBoxComboPopup()->IsCurrent((size_t)item) ||
+             (flags & wxODCB_PAINTING_CONTROL) ||
              (item & 1) == 0 )
         {
--- wxWidgets.orig/src/generic/odcombo.cpp	Mon Jul 17 17:28:38 2006
+++ wxWidgets/src/generic/odcombo.cpp	Tue Jul 18 19:43:42 2006
@@ -1042,5 +1042,6 @@
 {
     // we need to render selected and current items differently
-    if ( GetVListBoxComboPopup()->IsCurrent((size_t)item) )
+    if ( GetVListBoxComboPopup()->IsCurrent((size_t)item) ||
+         (flags & wxODCB_PAINTING_CONTROL) )
     {
         DrawFocusBackground(dc,

 	  	 
