Index: src/msw/notebook.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/msw/notebook.cpp,v
retrieving revision 1.159.2.3
diff -u -r1.159.2.3 notebook.cpp
--- src/msw/notebook.cpp	2006/01/21 16:46:46	1.159.2.3
+++ src/msw/notebook.cpp	2006/01/22 18:11:54
@@ -277,19 +277,16 @@
         style |= wxBORDER_SUNKEN;
 #endif
 
-    // comctl32.dll 6.0 doesn't support non-top tabs with visual styles (the
-    // control is simply not rendered correctly), so disable them in this case
+#if !wxUSE_UXTHEME 
+    //ComCtl32 notebook tabs simply don't work unless they're on top if we have uxtheme, we can
+    //work around it later (after control creation), but if we don't have have to clear those styles
     const int verComCtl32 = wxApp::GetComCtl32Version();
     if ( verComCtl32 == 600 )
     {
-        // check if we use themes at all -- if we don't, we're still ok
-#if wxUSE_UXTHEME
-        if ( wxUxThemeEngine::GetIfActive() )
-#endif
-        {
-            style &= ~(wxNB_BOTTOM | wxNB_LEFT | wxNB_RIGHT);
-        }
-    }
+        style &= ~(wxNB_BOTTOM | wxNB_LEFT | wxNB_RIGHT);
+    }   
+        
+#endif //wxUSE_UXTHEME
 
     LPCTSTR className = WC_TABCONTROL;
 
@@ -349,6 +346,22 @@
     {
         // create backing store
         UpdateBgBrush();
+    }
+    
+    // comctl32.dll 6.0 doesn't support non-top tabs with visual styles (the
+    // control is simply not rendered correctly), so we disable themes
+    // if possible, otherwise we simply clear the styles
+    // it's probably not possible to have UXTHEME without ComCtl32 6 or better, but lets
+    // check it anyway
+    const int verComCtl32 = wxApp::GetComCtl32Version();
+    if ( verComCtl32 == 600 ) 
+    {
+        // check if we use themes at all -- if we don't, we're still okay
+        if ( wxUxThemeEngine::GetIfActive() && (style && (wxNB_BOTTOM|wxNB_LEFT|wxNB_RIGHT)))
+        {
+            wxUxThemeEngine::GetIfActive()->SetWindowTheme((HWND)this->GetHandle(), L"", L"");
+            SetBackgroundColour(GetThemeBackgroundColour());    //correct the background color for the new non-themed control
+        }
     }
 #endif // wxUSE_UXTHEME
 

 	  	 
