Index: include/wx/dataview.h
===================================================================
--- include/wx/dataview.h	(revision 71546)
+++ include/wx/dataview.h	(working copy)
@@ -939,24 +939,24 @@
 class WXDLLIMPEXP_ADV wxDataViewListStoreLine
 {
 public:
-    wxDataViewListStoreLine( wxClientData *data = NULL )
+    wxDataViewListStoreLine( wxUIntPtr data = NULL )
     {
         m_data = data;
     }
     virtual ~wxDataViewListStoreLine()
     {
-        delete m_data;
+
     }
 
-    void SetData( wxClientData *data )
-        { if (m_data) delete m_data; m_data = data; }
-    wxClientData *GetData() const
+    void SetData( wxUIntPtr data )
+        { m_data = data; }
+    wxUIntPtr GetData() const
         { return m_data; }
 
     wxVector<wxVariant>  m_values;
 
 private:
-    wxClientData    *m_data;
+    wxUIntPtr m_data;
 };
 
 
@@ -970,12 +970,17 @@
     void InsertColumn( unsigned int pos, const wxString &varianttype );
     void AppendColumn( const wxString &varianttype );
 
-    void AppendItem( const wxVector<wxVariant> &values, wxClientData *data = NULL );
-    void PrependItem( const wxVector<wxVariant> &values, wxClientData *data = NULL );
-    void InsertItem(  unsigned int row, const wxVector<wxVariant> &values, wxClientData *data = NULL );
+    void AppendItem( const wxVector<wxVariant> &values, wxUIntPtr data = NULL );
+    void PrependItem( const wxVector<wxVariant> &values, wxUIntPtr data = NULL );
+    void InsertItem(  unsigned int row, const wxVector<wxVariant> &values, wxUIntPtr data = NULL );
     void DeleteItem( unsigned int pos );
     void DeleteAllItems();
 
+	unsigned int GetItemCount() const;
+
+    void SetItemData( const wxDataViewItem& item, wxUIntPtr data );
+    wxUIntPtr GetItemData( const wxDataViewItem& item ) const;
+
     // override base virtuals
 
     virtual unsigned int GetColumnCount() const;
@@ -1052,11 +1057,11 @@
           wxDataViewCellMode mode = wxDATAVIEW_CELL_INERT,
           int width = -1, wxAlignment align = wxALIGN_LEFT, int flags = wxDATAVIEW_COL_RESIZABLE );
 
-    void AppendItem( const wxVector<wxVariant> &values, wxClientData *data = NULL )
+    void AppendItem( const wxVector<wxVariant> &values, wxUIntPtr data = NULL )
         { GetStore()->AppendItem( values, data ); }
-    void PrependItem( const wxVector<wxVariant> &values, wxClientData *data = NULL )
+    void PrependItem( const wxVector<wxVariant> &values, wxUIntPtr data = NULL )
         { GetStore()->PrependItem( values, data ); }
-    void InsertItem(  unsigned int row, const wxVector<wxVariant> &values, wxClientData *data = NULL )
+    void InsertItem(  unsigned int row, const wxVector<wxVariant> &values, wxUIntPtr data = NULL )
         { GetStore()->InsertItem( row, values, data ); }
     void DeleteItem( unsigned row )
         { GetStore()->DeleteItem( row ); }
@@ -1081,6 +1086,14 @@
     bool GetToggleValue( unsigned int row, unsigned int col ) const
         { wxVariant value; GetStore()->GetValueByRow( value, row, col ); return value.GetBool(); }
 
+    void SetItemData( const wxDataViewItem& item, wxUIntPtr data )
+        { GetStore()->SetItemData( item, data ); }
+    wxUIntPtr GetItemData( const wxDataViewItem& item ) const
+        { return GetStore()->GetItemData( item ); }
+
+    int GetItemCount() const
+        { return GetStore()->GetItemCount(); }
+
     void OnSize( wxSizeEvent &event );
 
 private:
Index: interface/wx/dataview.h
===================================================================
--- interface/wx/dataview.h	(revision 71546)
+++ interface/wx/dataview.h	(working copy)
@@ -2315,17 +2315,17 @@
     /**
         Appends an item (=row) to the control and store.
     */
-    void AppendItem( const wxVector<wxVariant> &values, wxClientData *data = NULL );
+    void AppendItem( const wxVector<wxVariant> &values, wxUIntPtr data = NULL );
 
     /**
         Prepends an item (=row) to the control and store.
     */
-    void PrependItem( const wxVector<wxVariant> &values, wxClientData *data = NULL );
+    void PrependItem( const wxVector<wxVariant> &values, wxUIntPtr data = NULL );
 
     /**
         Inserts an item (=row) to the control and store.
     */
-    void InsertItem( unsigned int row, const wxVector<wxVariant> &values, wxClientData *data = NULL );
+    void InsertItem( unsigned int row, const wxVector<wxVariant> &values, wxUIntPtr data = NULL );
 
     /**
         Delete the row at position @a row.
@@ -2338,6 +2338,20 @@
     void DeleteAllItems();
 
     /**
+        Calls the identical method from wxDataViewListStore.
+
+        @since 2.9.4
+    */
+    unsigned int GetItemCount() const;
+
+	/**
+        Calls the identical method from wxDataViewListStore.
+
+        @since 2.9.4
+    */
+    wxUIntPtr GetItemData(const wxDataViewItem& item) const;
+
+    /**
          Sets the value in the store and update the control.
     */
     void SetValue( const wxVariant &value, unsigned int row, unsigned int col );
@@ -2379,6 +2393,13 @@
     */
     bool GetToggleValue( unsigned int row, unsigned int col ) const;
 
+    /**
+        Calls the identical method from wxDataViewListStore.
+
+        @since 2.9.4
+    */
+    void SetItemData(const wxDataViewItem& item, wxUIntPtr data);
+
     //@}
 };
 
@@ -2658,7 +2679,7 @@
         in number and type. No (default) values are filled in
         automatically.
     */
-    void AppendItem( const wxVector<wxVariant> &values, wxClientData *data = NULL );
+    void AppendItem( const wxVector<wxVariant> &values, wxUIntPtr data = NULL );
 
     /**
         Prepends an item (=row) and fills it with @a values.
@@ -2667,7 +2688,7 @@
         in number and type. No (default) values are filled in
         automatically.
     */
-    void PrependItem( const wxVector<wxVariant> &values, wxClientData *data = NULL );
+    void PrependItem( const wxVector<wxVariant> &values, wxUIntPtr data = NULL );
 
     /**
         Inserts an item (=row) and fills it with @a values.
@@ -2676,7 +2697,7 @@
         in number and type. No (default) values are filled in
         automatically.
     */
-    void InsertItem(  unsigned int row, const wxVector<wxVariant> &values, wxClientData *data = NULL );
+    void InsertItem(  unsigned int row, const wxVector<wxVariant> &values, wxUIntPtr data = NULL );
 
     /**
         Delete the item (=row) at position @a pos.
@@ -2689,6 +2710,20 @@
     void DeleteAllItems();
 
     /**
+        Returns the number of items (=rows) in the control
+
+        @since 2.9.4
+    */
+    unsigned int GetItemCount() const;
+
+    /**
+        Returns the client data associated with the item.
+
+        @since 2.9.4
+    */
+    wxUIntPtr GetItemData(const wxDataViewItem& item) const;
+
+    /**
         Overridden from wxDataViewModel
     */
     virtual unsigned int GetColumnCount() const;
@@ -2699,6 +2734,13 @@
     virtual wxString GetColumnType( unsigned int col ) const;
 
     /**
+        Sets the client data associated with the item.
+
+        @since 2.9.4
+    */
+    void SetItemData(const wxDataViewItem& item, wxUIntPtr data);
+
+    /**
         Overridden from wxDataViewIndexListModel
     */
     virtual void GetValueByRow( wxVariant &value,
Index: src/common/datavcmn.cpp
===================================================================
--- src/common/datavcmn.cpp	(revision 71546)
+++ src/common/datavcmn.cpp	(working copy)
@@ -1705,12 +1705,16 @@
     return m_cols.GetCount();
 }
 
+unsigned int wxDataViewListStore::GetItemCount() const
+{
+	return m_data.size();
+}
 wxString wxDataViewListStore::GetColumnType( unsigned int pos ) const
 {
     return m_cols[pos];
 }
 
-void wxDataViewListStore::AppendItem( const wxVector<wxVariant> &values, wxClientData *data )
+void wxDataViewListStore::AppendItem( const wxVector<wxVariant> &values, wxUIntPtr data )
 {
     wxDataViewListStoreLine *line = new wxDataViewListStoreLine( data );
     line->m_values = values;
@@ -1719,7 +1723,7 @@
     RowAppended();
 }
 
-void wxDataViewListStore::PrependItem( const wxVector<wxVariant> &values, wxClientData *data )
+void wxDataViewListStore::PrependItem( const wxVector<wxVariant> &values, wxUIntPtr data )
 {
     wxDataViewListStoreLine *line = new wxDataViewListStoreLine( data );
     line->m_values = values;
@@ -1729,7 +1733,7 @@
 }
 
 void wxDataViewListStore::InsertItem(  unsigned int row, const wxVector<wxVariant> &values,
-                                       wxClientData *data )
+                                       wxUIntPtr data )
 {
     wxDataViewListStoreLine *line = new wxDataViewListStoreLine( data );
     line->m_values = values;
@@ -1761,6 +1765,22 @@
     Reset( 0 );
 }
 
+void wxDataViewListStore::SetItemData( const wxDataViewItem& item, wxUIntPtr data )
+{
+    wxDataViewListStoreLine* line = m_data[wxPtrToUInt( item.GetID() ) - 1]; 
+    if (!line) return;
+
+    line->SetData( data );
+}
+
+wxUIntPtr wxDataViewListStore::GetItemData( const wxDataViewItem& item ) const
+{
+    wxDataViewListStoreLine* line = m_data[wxPtrToUInt( item.GetID() ) - 1]; 
+    if (!line) return NULL;
+
+    return line->GetData();
+}
+
 void wxDataViewListStore::GetValueByRow( wxVariant &value, unsigned int row, unsigned int col ) const
 {
     wxDataViewListStoreLine *line = m_data[row];
