Index: include/wx/dataview.h
===================================================================
--- include/wx/dataview.h	(revision 62504)
+++ include/wx/dataview.h	(working copy)
@@ -1123,7 +1123,8 @@
     virtual bool SetValueByRow( const wxVariant &value,
                            unsigned int row, unsigned int col );
 
-
+void SetItemData( const wxDataViewItem& item, wxClientData *data );
+wxClientData *GetItemData( const wxDataViewItem& item ) const;
 public:
     wxVector<wxDataViewListStoreLine*> m_data;
     wxArrayString                      m_cols;
Index: src/common/datavcmn.cpp
===================================================================
--- src/common/datavcmn.cpp	(revision 62504)
+++ src/common/datavcmn.cpp	(working copy)
@@ -1403,6 +1403,23 @@
     return true;
 }
 
+void wxDataViewListStore::SetItemData( const wxDataViewItem& item, wxClientData *data )
+{
+    wxDataViewListStoreLine* line = m_data[wxPtrToUInt( item.GetID() ) - 1];
+    if ( !line )
+	      return;
+    line->SetData( data );
+}
+
+wxClientData *wxDataViewListStore::GetItemData( const wxDataViewItem& item ) const
+{
+    wxDataViewListStoreLine* line = m_data[wxPtrToUInt( item.GetID() ) - 1];
+    if ( !line )
+	      return NULL;
+
+    return line->GetData();
+}
+
 //-----------------------------------------------------------------------------
 // wxDataViewListCtrl
 //-----------------------------------------------------------------------------
