diff --git a/samples/minimal/minimal.cpp b/samples/minimal/minimal.cpp
index a78e462..d142a6a 100644
--- a/samples/minimal/minimal.cpp
+++ b/samples/minimal/minimal.cpp
@@ -30,6 +30,11 @@
     #include "wx/wx.h"
 #endif
 
+#include "wx/combobox.h"
+#include "wx/dialog.h"
+#include "wx/button.h"
+#include "wx/sizer.h"
+
 // ----------------------------------------------------------------------------
 // resources
 // ----------------------------------------------------------------------------
@@ -185,16 +190,12 @@ void MyFrame::OnQuit(wxCommandEvent& WXUNUSED(event))
 
 void MyFrame::OnAbout(wxCommandEvent& WXUNUSED(event))
 {
-    wxMessageBox(wxString::Format
-                 (
-                    "Welcome to %s!\n"
-                    "\n"
-                    "This is the minimal wxWidgets sample\n"
-                    "running under %s.",
-                    wxVERSION_STRING,
-                    wxGetOsDescription()
-                 ),
-                 "About wxWidgets minimal sample",
-                 wxOK | wxICON_INFORMATION,
-                 this);
+    wxDialog dlg(this, wxID_ANY, "test", wxDefaultPosition, wxSize(200,200));
+
+    wxBoxSizer *sizer = new wxBoxSizer(wxVERTICAL);
+    wxComboBox *cb = new wxComboBox(&dlg, wxID_ANY, "", wxPoint(10,10), wxSize(100,-1));
+    sizer->Add(cb);
+    sizer->Add(dlg.CreateButtonSizer(wxOK|wxCANCEL));
+    dlg.SetSizer(sizer);
+    dlg.ShowModal();
 }
