Index: samples/minimal/minimal.cpp
===================================================================
--- samples/minimal/minimal.cpp	(revision 53061)
+++ samples/minimal/minimal.cpp	(working copy)
@@ -30,6 +30,8 @@
     #include "wx/wx.h"
 #endif
 
+#include "wx/treectrl.h"
+
 // ----------------------------------------------------------------------------
 // resources
 // ----------------------------------------------------------------------------
@@ -68,7 +70,12 @@
     void OnQuit(wxCommandEvent& event);
     void OnAbout(wxCommandEvent& event);
 
+    void OnAddChild(wxCommandEvent& event);
+    void OnDeleteChildren(wxCommandEvent& event);
+
 private:
+    wxTreeCtrl* tree;
+
     // any class wishing to process wxWidgets events must use this macro
     DECLARE_EVENT_TABLE()
 };
@@ -83,6 +90,9 @@
     // menu items
     Minimal_Quit = wxID_EXIT,
 
+    Minimal_AddChild = wxID_HIGHEST + 1,
+    Minimal_DeleteChildren,
+
     // it is important for the id corresponding to the "About" command to have
     // this standard value as otherwise it won't be handled properly under Mac
     // (where it is special and put into the "Apple" menu)
@@ -99,6 +109,9 @@
 BEGIN_EVENT_TABLE(MyFrame, wxFrame)
     EVT_MENU(Minimal_Quit,  MyFrame::OnQuit)
     EVT_MENU(Minimal_About, MyFrame::OnAbout)
+
+    EVT_MENU(Minimal_AddChild,  MyFrame::OnAddChild)
+    EVT_MENU(Minimal_DeleteChildren,  MyFrame::OnDeleteChildren)
 END_EVENT_TABLE()
 
 // Create a new application object: this macro will allow wxWidgets to create
@@ -148,6 +161,9 @@
     // set the frame icon
     SetIcon(wxICON(sample));
 
+    tree = new wxTreeCtrl(this);
+    tree->AddRoot(wxT("Root node"));
+
 #if wxUSE_MENUS
     // create a menu bar
     wxMenu *fileMenu = new wxMenu;
@@ -156,6 +172,10 @@
     wxMenu *helpMenu = new wxMenu;
     helpMenu->Append(Minimal_About, _T("&About...\tF1"), _T("Show about dialog"));
 
+    fileMenu->Append(Minimal_AddChild, _T("Add child\tCtrl+A"));
+    fileMenu->Append(Minimal_DeleteChildren, _T("Delete children\tCtrl+D"));
+    fileMenu->AppendSeparator();
+
     fileMenu->Append(Minimal_Quit, _T("E&xit\tAlt-X"), _T("Quit this program"));
 
     // now append the freshly created menu to the menu bar...
@@ -197,3 +217,13 @@
                  wxOK | wxICON_INFORMATION,
                  this);
 }
+
+void MyFrame::OnAddChild(wxCommandEvent& WXUNUSED(event))
+{
+    tree->AppendItem(tree->GetRootItem(), wxT("Child node"));
+}
+
+void MyFrame::OnDeleteChildren(wxCommandEvent& WXUNUSED(event))
+{
+    tree->DeleteChildren(tree->GetRootItem());
+}

 	  	 
