diff --git a/include/wx/stc/stc.h b/include/wx/stc/stc.h
index 3c390b1b08..03a621b952 100644
--- a/include/wx/stc/stc.h
+++ b/include/wx/stc/stc.h
@@ -5195,6 +5195,15 @@ public:
     // In that case the character data will be utf-8 encoded since that is
     // what is used internally by Scintilla in unicode builds.
 
+    // Replace the selected text with the argument text.
+    void ReplaceSelectionRaw(const char* text);
+
+    // Replace the target text with the argument text.
+    int ReplaceTargetRaw(const char* text, int length=-1);
+
+    // Replace the target text with the argument text after \\d processing.
+    int ReplaceTargetRERaw(const char* text, int length=-1);
+
     // Add text to the document at current position.
     void AddTextRaw(const char* text, int length=-1);
 
diff --git a/src/stc/stc.cpp b/src/stc/stc.cpp
index 65a7a74684..45cf18efd2 100644
--- a/src/stc/stc.cpp
+++ b/src/stc/stc.cpp
@@ -5049,8 +5049,22 @@ void wxStyledTextCtrl::RegisterImage(int type, const wxBitmap& bmp)
     m_swx->DoRegisterImage(type, bmp);
 }
 
+void wxStyledTextCtrl::ReplaceSelectionRaw(const char* text)
+{
+    SendMsg(SCI_REPLACESEL, 0, (sptr_t)text);
+}
 
+int wxStyledTextCtrl::ReplaceTargetRaw(const char* text, int length) {
+    if (length == -1)
+        length = strlen(text);
+    return SendMsg(SCI_REPLACETARGET, length, (sptr_t)text);
+}
 
+int wxStyledTextCtrl::ReplaceTargetRERaw(const char* text, int length) {
+    if (length == -1)
+        length = strlen(text);
+    return SendMsg(SCI_REPLACETARGETRE, length, (sptr_t)text);
+}
 
 void wxStyledTextCtrl::AddTextRaw(const char* text, int length)
 {
diff --git a/src/stc/stc.h.in b/src/stc/stc.h.in
index 679c0e16e0..dabb81f4df 100644
--- a/src/stc/stc.h.in
+++ b/src/stc/stc.h.in
@@ -312,6 +312,15 @@ public:
     // In that case the character data will be utf-8 encoded since that is
     // what is used internally by Scintilla in unicode builds.
 
+    // Replace the selected text with the argument text.
+    void ReplaceSelectionRaw(const char* text);
+
+    // Replace the target text with the argument text.
+    int ReplaceTargetRaw(const char* text, int length=-1);
+
+    // Replace the target text with the argument text after \\d processing.
+    int ReplaceTargetRERaw(const char* text, int length=-1);
+
     // Add text to the document at current position.
     void AddTextRaw(const char* text, int length=-1);
 
