Index: src/osx/cocoa/listbox.mm
===================================================================
--- src/osx/cocoa/listbox.mm	(revision 59381)
+++ src/osx/cocoa/listbox.mm	(working copy)
@@ -53,12 +53,25 @@
 
 @end
 
-@interface wxNSTableView : NSTableView
+@interface wxNSTableViewKeyFix : NSTableView
 {
+    BOOL _fakeAction;
 }
 
+- (id)initWithFrame:(NSRect)frame;
+- (NSInteger)clickedColumn;
+- (NSInteger)clickedRow;
+- (void)keyDown:(NSEvent *)event;
+- (void)tableViewSelectionDidChange:(NSNotification *)aNotification;
+
 @end
 
+@interface wxNSTableView : wxNSTableViewKeyFix
+{
+}
+
+@end
+
 //
 // table column
 //
@@ -270,6 +283,42 @@
 
 @end
 
+@implementation wxNSTableViewKeyFix
+
+- (id)initWithFrame:(NSRect)frame
+{
+    [super initWithFrame:frame];
+    _fakeAction = NO;
+    return self;
+}
+
+- (NSInteger)clickedColumn
+{
+    return (_fakeAction) ? [self selectedColumn] : [super clickedColumn];
+}
+
+- (NSInteger)clickedRow
+{
+    return (_fakeAction) ? [self selectedRow] : [super clickedRow];
+}
+
+- (void)keyDown:(NSEvent *)event
+{
+    _fakeAction = YES;
+    id savedDelegate = [self delegate];
+    [self setDelegate: self];
+    [super keyDown:event];
+    [self setDelegate: savedDelegate];
+    _fakeAction = NO;
+}
+
+- (void)tableViewSelectionDidChange:(NSNotification *)aNotification
+{
+    [[self target] performSelector:[self action] withObject:self];
+}
+
+@end
+
 @implementation wxNSTableView
 
 + (void)initialize
