Index: src/generic/datavgen.cpp
===================================================================
--- src/generic/datavgen.cpp	(revision 74824)
+++ src/generic/datavgen.cpp	(working copy)
@@ -1021,7 +1021,7 @@
 bool wxDataViewBitmapRenderer::Render( wxRect cell, wxDC *dc, int WXUNUSED(state) )
 {
     if (m_bitmap.IsOk())
-        dc->DrawBitmap( m_bitmap, cell.x, cell.y );
+        dc->DrawBitmap( m_bitmap, cell.x, cell.y, true );
     else if (m_icon.IsOk())
         dc->DrawIcon( m_icon, cell.x, cell.y );
 
