Index: samples/menu/menu.cpp
===================================================================
--- samples/menu/menu.cpp       (revision 54766)
+++ samples/menu/menu.cpp       (working copy)
@@ -1003,10 +1003,27 @@
 }
 #endif

+#if 1
+#include "wx/checklst.h"
 void MyFrame::ShowContextMenu(const wxPoint& pos)
 {
     wxMenu menu;
+    wxMenuItem *item = new wxMenuItem(&menu, Menu_Help_About, _T("&About"));
+    item->SetBitmap(copy_xpm);
+    menu.Append(item);
+    menu.Append(Menu_Popup_Submenu, _T("&Submenu"), CreateDummyMenu(NULL));
+    PopupMenu(&menu, pos.x, pos.y);

+    wxCheckListBox lb;
+    lb.Create(this, wxID_ANY);
+    lb.Append("THIS SCREWS UP MENU MARGINS. THE BITMAP ABOVE BECOMES MISALIGNED NEXT TIME");
+
+}
+#else
+void MyFrame::ShowContextMenu(const wxPoint& pos)
+{
+    wxMenu menu;
+
     if ( wxGetKeyState(WXK_SHIFT) )
     {
         // when Shift is pressed, demonstrate the use of a simple function
@@ -1052,6 +1069,7 @@
 #endif // 0
     }
 }
+#endif

 void MyFrame::OnTestNormal(wxCommandEvent& WXUNUSED(event))
 {

