Index: include/wx/arrimpl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/arrimpl.cpp,v
retrieving revision 1.18
diff -u -r1.18 arrimpl.cpp
--- include/wx/arrimpl.cpp	2004/05/25 17:30:57	1.18
+++ include/wx/arrimpl.cpp	2005/05/17 18:36:50
@@ -98,7 +98,7 @@
       size_t ui = size() - 1;                                                 \
       do {                                                                    \
         if ( (T*)base_array::operator[](ui) == &Item )                        \
-          return ui;                                                          \
+          return (int)ui;                                                          \
         ui--;                                                                 \
       }                                                                       \
       while ( ui != 0 );                                                      \
@@ -107,7 +107,7 @@
   else {                                                                      \
     for( size_t ui = 0; ui < size(); ui++ ) {                                 \
       if( (T*)base_array::operator[](ui) == &Item )                           \
-        return ui;                                                            \
+        return (int)ui;                                                            \
     }                                                                         \
   }                                                                           \
                                                                               \

 	  	 
