diff -u -3 -P -r wxWidgets/src/html/chm.cpp wxW/src/html/chm.cpp
--- wxWidgets/src/html/chm.cpp	Tue Oct 07 22:14:00 2003
+++ wxW/src/html/chm.cpp	Fri May 07 18:46:38 2004
@@ -557,11 +557,11 @@
     const char *tmp;
 
     // Try to open the #SYSTEM-File and create the HHP File out of it
-    // see http://bonedaddy.net/pabs3/chmspec/0.1/ch05s03.html#SYSTEM
-    if ( ! m_chm->Contains(_("/#SYSTEM")) )
+    // see http://bonedaddy.net/pabs3/chmspec/0.1.2/Internal.html#SYSTEM
+    if ( ! m_chm->Contains(_T("/#SYSTEM")) )
     {
 #ifdef DEBUG
-        wxLogDebug("Archive doesnt contain #SYSTEM file");
+        wxLogDebug(_("Archive doesnt contain #SYSTEM file"));
 #endif
         return;
     }

 	  	 
