--- wxBase.spec.orig	2005-03-24 12:00:36.000000000 +0800
+++ wxBase.spec	2005-03-30 06:19:51.177030048 +0800
@@ -1,6 +1,6 @@
 %define pref /usr
-%define ver 2.5.5
+%define ver  2.5.5
 %define ver2 2.5
-%define rel 1
+%define rel  1
 
 # Configurable settings (use --with(out) unicode on rpmbuild command line):
@@ -10,19 +10,15 @@
 
 %if %{unicode}
-%define wxconfigname base-unicode-release-%{ver2}
-%define wxconfiglinkname wxbaseu-%{ver2}-config
+    %define name         wx-base-unicode
+    %define wxconfig     base-unicode-release-%{ver2}
+    %define wxconfiglink wxbaseu-%{ver2}-config
 %else
-%define wxconfigname base-ansi-release-%{ver2}
-%define wxconfiglinkname wxbase-%{ver2}-config
+    %define name         wx-base
+    %define wxconfig     base-ansi-release-%{ver2}
+    %define wxconfiglink wxbase-%{ver2}-config
 %endif
 
-%if %{unicode}
-%define name wx-base-unicode
-%else 
-%define name wx-base
-%endif
-
-Summary: wxBase library - non-GUI support classes of wxWindows toolkit
-Name:   %{name}
+Summary: wxBase library - non-GUI support classes of wxWidgets toolkit
+Name: %{name}
 Version: %{ver}
 Release: %{rel}
@@ -30,8 +26,8 @@
 Group: Development/Libraries
 Source: wxBase-%{ver}.tar.bz2
-URL: http://www.wxwindows.org
+URL: http://www.wxwidgets.org
 Packager: Vadim Zeitlin <vadim@wxwindows.org>
 Prefix: %{pref}
-BuildRoot: /var/tmp/%{name}-root
+BuildRoot: %{_tmppath}/%{name}-root
 Provides: wxBase
 
@@ -57,4 +53,5 @@
 Summary: wxBase static libraries
 Group: Development/Libraries
+Requires: %{name}-devel = %{ver}
 
 %description static
@@ -66,7 +63,7 @@
 %build
 if [ "$SMP" != "" ]; then
-  export MAKE="make -j$SMP"
+    export MAKE="make -j$SMP"
 else
-  export MAKE="make"
+    export MAKE="make"
 fi
 
@@ -75,7 +72,7 @@
 ../configure --prefix=%{pref} --disable-gui \
 %if %{unicode}
-             --enable-unicode
+                              --enable-unicode
 %else
-             --with-odbc
+                              --with-odbc
 %endif
 $MAKE
@@ -84,9 +81,9 @@
 mkdir obj-static
 cd obj-static
-../configure --prefix=%{pref} --disable-shared --disable-gui \
+../configure --prefix=%{pref} --disable-gui --disable-shared \
 %if %{unicode}
-             --enable-unicode
+                              --enable-unicode
 %else
-             --with-odbc
+                              --with-odbc
 %endif
 $MAKE
@@ -111,29 +108,32 @@
 %post devel
 # link wx-config when you install RPM.
-ln -sf %{_libdir}/wx/config/%{wxconfigname} %{_bindir}/wx-config
+ln -sf %{_libdir}/wx/config/%{wxconfig} %{_bindir}/wx-config
 # link wx-config with explicit name.
-ln -sf %{_libdir}/wx/config/%{wxconfigname} %{_bindir}/%{wxconfiglinkname}
-                                                                                   
+ln -sf %{_libdir}/wx/config/%{wxconfig} %{_bindir}/%{wxconfiglink}
+/sbin/ldconfig
+
+%postun devel
+/sbin/ldconfig
+
 %preun devel
 if test -f %{_bindir}/wx-config -a -f /usr/bin/md5sum ; then
-  SUM1=`md5sum %{_libdir}/wx/config/%{wxconfigname} | cut -c 0-32`
-  SUM2=`md5sum %{_bindir}/wx-config | cut -c 0-32`
-  if test "x$SUM1" = "x$SUM2" ; then
-    rm -f %{_bindir}/wx-config
-  fi
+    SUM1=`md5sum %{_libdir}/wx/config/%{wxconfig} | cut -c 0-32`
+    SUM2=`md5sum %{_bindir}/wx-config | cut -c 0-32`  
+    if test "x$SUM1" = "x$SUM2" ; then
+        rm -f %{_bindir}/wx-config
+    fi
 fi
-                                                                                   
-rm -f %{_bindir}/%{wxconfiglinkname}
+rm -f %{_bindir}/%{wxconfiglink}
 
 %files -f wxstd.lang
 %defattr (-,root,root)
 %doc COPYING.LIB *.txt
-%{_libdir}/libwx_base*so.*
+%{_libdir}/libwx_base*-%{ver2}.so.*
 
 %files devel
 %defattr (-,root,root)
-%dir %{_includedir}/wx-*
-%{_includedir}/wx-*/*
-%{_libdir}/libwx_base*.so
+%dir %{_includedir}/wx-%{ver2}
+%{_includedir}/wx-%{ver2}/*
+%{_libdir}/libwx_base*-%{ver2}.so
 %dir %{_libdir}/wx
 %{_libdir}/wx/*

 	  	 
