Index: interface/wx/stream.h
===================================================================
--- interface/wx/stream.h	(revision 71073)
+++ interface/wx/stream.h	(working copy)
@@ -150,6 +150,13 @@
 class wxStreamBuffer
 {
 public:
+    /** BufMode flags */
+    enum BufMode
+    {
+        read,
+        write,
+        read_write
+    };
 
     /**
         Constructor, creates a new stream buffer using @a stream as a parent stream
@@ -241,7 +248,7 @@
         Destructor.
         It finalizes all IO calls and frees all internal buffers if necessary.
     */
-    wxStreamBuffer();
+    ~wxStreamBuffer();
 
     /**
         Fill the IO buffer.
