Index: src/osx/carbon/region.cpp
===================================================================
--- src/osx/carbon/region.cpp	(revision 61507)
+++ src/osx/carbon/region.cpp	(working copy)
@@ -301,7 +301,7 @@
     if (!m_refData)
         return wxOutRegion;
 
-    CGPoint p = { y , x } ;
+    CGPoint p = CGPointMake(x, y);
     if (HIShapeContainsPoint( M_REGION , &p ) )
         return wxInRegion;
 
