Index: src/generic/grid.cpp
===================================================================
--- src/generic/grid.cpp	(revision 76109)
+++ src/generic/grid.cpp	(working copy)
@@ -6240,9 +6240,9 @@
 {
     if ( edit != m_editable )
     {
+        m_editable = edit;
         if (!edit)
             EnableCellEditControl(edit);
-        m_editable = edit;
     }
 }
 
@@ -6291,15 +6291,16 @@
 
 bool wxGrid::CanEnableCellControl() const
 {
-    return m_editable && (m_currentCellCoords != wxGridNoCellCoords) &&
-        !IsCurrentCellReadOnly();
+    return m_editable && (m_currentCellCoords != wxGridNoCellCoords);
 }
 
 bool wxGrid::IsCellEditControlEnabled() const
 {
     // the cell edit control might be disable for all cells or just for the
     // current one if it's read only
-    return m_cellEditCtrlEnabled ? !IsCurrentCellReadOnly() : false;
+    // BUT: if we want to copy-paste, read only condition mustn't be here,
+    // and it's also deleted in CanEnableCellControl() above 
+    return m_cellEditCtrlEnabled;// ? !IsCurrentCellReadOnly() : false;
 }
 
 bool wxGrid::IsCellEditControlShown() const
@@ -7619,7 +7620,6 @@
     wxGridCellAttr* attr = GetCellAttr(row, col);
     bool isReadOnly = attr->IsReadOnly();
     attr->DecRef();
-
     return isReadOnly;
 }
 
Index: src/generic/grideditors.cpp
===================================================================
--- src/generic/grideditors.cpp	(revision 76109)
+++ src/generic/grideditors.cpp	(working copy)
@@ -490,7 +490,10 @@
     wxASSERT_MSG(m_control, wxT("The wxGridCellEditor must be created first!"));
 
     m_value = grid->GetTable()->GetValue(row, col);
-
+    if ( grid->IsReadOnly(row, col) )
+    {
+        Text()->SetEditable(false);
+    }
     DoBeginEdit(m_value);
 }
 
@@ -714,7 +717,12 @@
             return;
         }
     }
-
+    
+    
+    if ( grid->IsReadOnly(row, col) )
+    {
+        Text()->SetEditable(false);
+    }
 #if wxUSE_SPINCTRL
     if ( HasRange() )
     {
@@ -935,6 +943,11 @@
             }
         }
     }
+    
+    if ( grid->IsReadOnly(row, col) )
+    {
+        Text()->SetEditable(false);
+    }
 
     DoBeginEdit(GetString());
 }
@@ -1640,7 +1653,7 @@
             m_index = -1;
         }
     }
-
+    
     Combo()->SetSelection(m_index);
     Combo()->SetFocus();
 
