Index: src/common/any.cpp
===================================================================
--- src/common/any.cpp	(revision 66520)
+++ src/common/any.cpp	(working copy)
@@ -471,7 +471,9 @@
 WX_IMPLEMENT_ANY_VALUE_TYPE(wxAnyValueTypeImplConstCharPtr)
 WX_IMPLEMENT_ANY_VALUE_TYPE(wxAnyValueTypeImplConstWchar_tPtr)
 
+#if wxUSE_DATETIME
 WX_IMPLEMENT_ANY_VALUE_TYPE(wxAnyValueTypeImpl<wxDateTime>)
+#endif // wxUSE_DATETIME
 //WX_IMPLEMENT_ANY_VALUE_TYPE(wxAnyValueTypeImpl<wxObject*>)
 //WX_IMPLEMENT_ANY_VALUE_TYPE(wxAnyValueTypeImpl<wxArrayString>)
 
Index: include/wx/chkconf.h
===================================================================
--- include/wx/chkconf.h	(revision 66520)
+++ include/wx/chkconf.h	(working copy)
@@ -1630,6 +1630,15 @@
 #   endif
 #endif /* wxUSE_FILEDLG */
 
+#if wxUSE_FILECTRL && !wxUSE_DATETIME
+#   ifdef wxABORT_ON_CONFIG_ERROR
+#       error "wxFileCtrl requires wxDateTime"
+#   else
+#       undef wxUSE_DATETIME
+#       define wxUSE_DATETIME 1
+#   endif
+#endif /* wxUSE_FILECTRL */
+
 /* common dependencies */
 #if wxUSE_ARTPROVIDER_TANGO
 #   if !(wxUSE_STREAMS && wxUSE_IMAGE && wxUSE_LIBPNG)
