--- ./include/wx/gtk/win_gtk.h-	2004-09-15 18:05:39.168964064 -0700
+++ ./include/wx/gtk/win_gtk.h	2004-09-15 19:09:37.618431048 -0700
@@ -82,7 +82,7 @@
 				     GtkAdjustment  *vadjustment);
 };
 
-guint      gtk_pizza_get_type        (void);
+GtkType    gtk_pizza_get_type        (void);
 GtkWidget* gtk_pizza_new             (void);
 
 void       gtk_pizza_set_shadow_type (GtkPizza          *pizza,
--- ./src/gtk/win_gtk.c-	2004-09-15 19:10:51.643177576 -0700
+++ ./src/gtk/win_gtk.c	2004-09-15 19:10:58.587121936 -0700
@@ -104,10 +104,10 @@
 
 static gboolean gravity_works;
 
-guint
+GtkType
 gtk_pizza_get_type ()
 {
-    static guint pizza_type = 0;
+    static GtkType pizza_type = 0;
 
     if (!pizza_type)
     {

 	  	 
