diff --git a/samples/console/console.cpp b/samples/console/console.cpp
index 2d2d435..8d3333b 100644
--- a/samples/console/console.cpp
+++ b/samples/console/console.cpp
@@ -28,7 +28,7 @@
 #ifndef WX_PRECOMP
     #include "wx/wx.h"
 #endif
-
+#include "wx/textfile.h"
 #include <wx/app.h>
 #include <wx/cmdline.h>
 
@@ -58,6 +58,13 @@ int main(int argc, char **argv)
     }
 
     wxCmdLineParser parser(cmdLineDesc, argc, argv);
+    wxTextFile f;
+    f.Open( "text" );
+    for( wxString str = f.GetFirstLine(); !f.Eof(); str = f.GetNextLine() )
+    {
+        if( str.IsEmpty() )
+            f.RemoveLine( f.GetCurrentLine() );
+    }
     switch ( parser.Parse() )
     {
         case -1:
