diff --git a/include/wx/textbuf.h b/include/wx/textbuf.h
index a2511b4..3b1134c 100644
--- a/include/wx/textbuf.h
+++ b/include/wx/textbuf.h
@@ -120,12 +120,15 @@ public:
     wxString& GetFirstLine() /* const */
         { return m_aLines.empty() ? ms_eof : m_aLines[m_nCurLine = 0]; }
     wxString& GetNextLine()  /* const */
-        { return ++m_nCurLine == m_aLines.size() ? ms_eof
-                                                 : m_aLines[m_nCurLine]; }
+        { return m_nCurLine < m_aLines.size() 
+                  ? (++m_nCurLine == m_aLines.size()
+                        ? ms_eof
+                        : m_aLines[m_nCurLine])
+                  : ms_eof; }
     wxString& GetPrevLine()  /* const */
         { wxASSERT(m_nCurLine > 0); return m_aLines[--m_nCurLine]; }
     wxString& GetLastLine() /* const */
-        { m_nCurLine = m_aLines.size() - 1; return m_aLines.Last(); }
+        { return m_aLines.empty() ? ms_eof : m_aLines[m_nCurLine = m_aLines.size() - 1]; }
 
     // get the type of the line (see also GetEOL)
     wxTextFileType GetLineType(size_t n) const { return m_aTypes[n]; }
@@ -148,7 +151,7 @@ public:
                   wxTextFileType type = typeDefault)
     {
         m_aLines.insert(m_aLines.begin() + n, str);
-        m_aTypes.insert(m_aTypes.begin()+n, type);
+        m_aTypes.insert(m_aTypes.begin() + n, type);
     }
 
     // delete one line
@@ -156,6 +159,10 @@ public:
     {
         m_aLines.erase(m_aLines.begin() + n);
         m_aTypes.erase(m_aTypes.begin() + n);
+        if(m_nCurLine > m_aLines.size())
+        {
+            m_nCurLine = m_aLines.size();
+        }
     }
 
     // remove all lines
