Index: src/unix/gsocket.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/unix/gsocket.cpp,v
retrieving revision 1.51
diff -b -u -2 -r1.51 gsocket.cpp
--- src/unix/gsocket.cpp	2005/09/19 15:18:27	1.51
+++ src/unix/gsocket.cpp	2005/09/24 22:06:06
@@ -870,8 +870,9 @@
 
   /* If the socket is blocking, wait for data (with a timeout) */
-  if (Input_Timeout() == GSOCK_TIMEDOUT)
+  if (Input_Timeout() == GSOCK_TIMEDOUT) {
     /* We no longer return here immediately, otherwise socket events would not be re-enabled! */
+    m_error=GSOCK_TIMEDOUT;
     ret = -1;
-  else {
+  } else {
     /* Read the data */
     if (m_stream)
@@ -881,5 +882,5 @@
   }
 
-  if (ret == -1)
+  if (ret == -1 && m_error != GSOCK_TIMEDOUT)
   {
     if ((errno == EWOULDBLOCK) || (errno == EAGAIN))

 	  	 
