Index: layout.cpp
===================================================================
RCS file: /home/wxcvs/wxWindows/samples/layout/layout.cpp,v
retrieving revision 1.26
diff -b -u -2 -r1.26 layout.cpp
--- layout.cpp	17 Mar 2002 14:15:41 -0000	1.26
+++ layout.cpp	30 Sep 2002 22:25:51 -0000
@@ -213,5 +213,6 @@
     wxBoxSizer *topsizer = new wxBoxSizer( wxVERTICAL );
 
-    wxNotebook *notebook = new wxNotebook( &dialog, -1 );
+    wxNotebook *notebook = new wxNotebook( &dialog, -1 , wxDefaultPosition,
+                                           wxDefaultSize, wxNB_MULTILINE);
     wxNotebookSizer *nbs = new wxNotebookSizer( notebook );
     topsizer->Add( nbs, 1, wxGROW );
@@ -222,9 +223,9 @@
     // First page: one big text ctrl
     wxTextCtrl *multi = new wxTextCtrl( notebook, -1, "TextCtrl.", wxDefaultPosition, wxDefaultSize, wxTE_MULTILINE );
-    notebook->AddPage( multi, "Page One" );
+    notebook->AddPage( multi, "Page One of Two" );
 
     // Second page: a text ctrl and a button
     wxPanel *panel = new wxPanel( notebook, -1 );
-    notebook->AddPage( panel, "Page Two" );
+    notebook->AddPage( panel, "Page Two of Two" );
 
     wxSizer *panelsizer = new wxBoxSizer( wxVERTICAL );

 	  	 
