Index: helpdata.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/html/helpdata.cpp,v
retrieving revision 1.76
diff -u -r1.76 helpdata.cpp
--- helpdata.cpp	2004/09/29 17:53:43	1.76
+++ helpdata.cpp	2004/10/03 15:52:19
@@ -78,6 +78,11 @@
     wxHtmlHelpDataItem *ia = *a;
     wxHtmlHelpDataItem *ib = *b;
 
+    if (ia == 0)
+            return -1;
+    if (ib == 0)
+            return 1;
+    
     if (ia->parent == ib->parent)
     {
         return ia->name.CmpNoCase(ib->name);



 	  	 
