Index: wx/wxcrt.h
===================================================================
--- wx/wxcrt.h	(revision 63066)
+++ wx/wxcrt.h	(working copy)
@@ -477,7 +477,7 @@
 // forward-declare the template and implement it below WX_STRCMP_FUNC. OTOH,
 // this fails to compile with VC6, so don't do it for VC. It also causes
 // problems with GCC visibility in newer GCC versions.
-#if !(defined(__VISUALC__) || wxCHECK_GCC_VERSION(3,4))
+#if !(defined(__VISUALC__) || wxCHECK_GCC_VERSION(3,5))
 template<typename T>
 inline int wxStrcoll_String(const wxString& s1, const T& s2);
 WX_STRCMP_FUNC(wxStrcoll, wxCRT_StrcollA, wxCRT_StrcollW, wxStrcoll_String)
@@ -497,7 +497,7 @@
 #endif
 }
 
-#if defined(__VISUALC__) || wxCHECK_GCC_VERSION(3,4)
+#if defined(__VISUALC__) || wxCHECK_GCC_VERSION(3,5)
 // this is exactly the same WX_STRCMP_FUNC line as above wxStrcoll_String<>
 WX_STRCMP_FUNC(wxStrcoll, wxCRT_StrcollA, wxCRT_StrcollW, wxStrcoll_String)
 #endif
