Index: build/msw/wx.dsw
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/build/msw/wx.dsw,v
retrieving revision 1.10
diff -u -r1.10 wx.dsw
--- build/msw/wx.dsw	2005/03/21 00:10:51	1.10
+++ build/msw/wx.dsw	2005/06/09 10:03:12
@@ -2,19 +2,26 @@
 # WARNING: DO NOT EDIT OR DELETE THIS WORKSPACE FILE!
 
 ###############################################################################
-Project: "wxregex"=wx_wxregex.dsp - Package Owner=<4>
 
+Project: "adv"=.\wx_adv.dsp - Package Owner=<4>
+
 Package=<5>
 {{{
 }}}
 
 Package=<4>
 {{{
+    Begin Project Dependency
+    Project_Dep_Name core
+    End Project Dependency
+    Begin Project Dependency
+    Project_Dep_Name base
+    End Project Dependency
 }}}
 
 ###############################################################################
 
-Project: "wxzlib"=wx_wxzlib.dsp - Package Owner=<4>
+Project: "base"=.\wx_base.dsp - Package Owner=<4>
 
 Package=<5>
 {{{
@@ -22,11 +29,29 @@
 
 Package=<4>
 {{{
+    Begin Project Dependency
+    Project_Dep_Name wxexpat
+    End Project Dependency
+    Begin Project Dependency
+    Project_Dep_Name wxjpeg
+    End Project Dependency
+    Begin Project Dependency
+    Project_Dep_Name wxpng
+    End Project Dependency
+    Begin Project Dependency
+    Project_Dep_Name wxregex
+    End Project Dependency
+    Begin Project Dependency
+    Project_Dep_Name wxtiff
+    End Project Dependency
+    Begin Project Dependency
+    Project_Dep_Name wxzlib
+    End Project Dependency
 }}}
 
 ###############################################################################
 
-Project: "wxpng"=wx_wxpng.dsp - Package Owner=<4>
+Project: "core"=.\wx_core.dsp - Package Owner=<4>
 
 Package=<5>
 {{{
@@ -34,11 +59,14 @@
 
 Package=<4>
 {{{
+    Begin Project Dependency
+    Project_Dep_Name base
+    End Project Dependency
 }}}
 
 ###############################################################################
 
-Project: "wxjpeg"=wx_wxjpeg.dsp - Package Owner=<4>
+Project: "dbgrid"=.\wx_dbgrid.dsp - Package Owner=<4>
 
 Package=<5>
 {{{
@@ -50,7 +78,7 @@
 
 ###############################################################################
 
-Project: "wxtiff"=wx_wxtiff.dsp - Package Owner=<4>
+Project: "gl"=.\wx_gl.dsp - Package Owner=<4>
 
 Package=<5>
 {{{
@@ -58,11 +86,17 @@
 
 Package=<4>
 {{{
+    Begin Project Dependency
+    Project_Dep_Name base
+    End Project Dependency
+    Begin Project Dependency
+    Project_Dep_Name core
+    End Project Dependency
 }}}
 
 ###############################################################################
 
-Project: "wxexpat"=wx_wxexpat.dsp - Package Owner=<4>
+Project: "html"=.\wx_html.dsp - Package Owner=<4>
 
 Package=<5>
 {{{
@@ -70,11 +104,17 @@
 
 Package=<4>
 {{{
+    Begin Project Dependency
+    Project_Dep_Name base
+    End Project Dependency
+    Begin Project Dependency
+    Project_Dep_Name core
+    End Project Dependency
 }}}
 
 ###############################################################################
 
-Project: "base"=wx_base.dsp - Package Owner=<4>
+Project: "media"=.\wx_media.dsp - Package Owner=<4>
 
 Package=<5>
 {{{
@@ -82,11 +122,17 @@
 
 Package=<4>
 {{{
+    Begin Project Dependency
+    Project_Dep_Name base
+    End Project Dependency
+    Begin Project Dependency
+    Project_Dep_Name core
+    End Project Dependency
 }}}
 
 ###############################################################################
 
-Project: "net"=wx_net.dsp - Package Owner=<4>
+Project: "net"=.\wx_net.dsp - Package Owner=<4>
 
 Package=<5>
 {{{
@@ -94,11 +140,14 @@
 
 Package=<4>
 {{{
+    Begin Project Dependency
+    Project_Dep_Name base
+    End Project Dependency
 }}}
 
 ###############################################################################
 
-Project: "core"=wx_core.dsp - Package Owner=<4>
+Project: "odbc"=.\wx_odbc.dsp - Package Owner=<4>
 
 Package=<5>
 {{{
@@ -106,11 +155,20 @@
 
 Package=<4>
 {{{
+    Begin Project Dependency
+    Project_Dep_Name base
+    End Project Dependency
+    Begin Project Dependency
+    Project_Dep_Name core
+    End Project Dependency
+    Begin Project Dependency
+    Project_Dep_Name xml
+    End Project Dependency
 }}}
 
 ###############################################################################
 
-Project: "adv"=wx_adv.dsp - Package Owner=<4>
+Project: "qa"=.\wx_qa.dsp - Package Owner=<4>
 
 Package=<5>
 {{{
@@ -118,11 +176,20 @@
 
 Package=<4>
 {{{
+    Begin Project Dependency
+    Project_Dep_Name base
+    End Project Dependency
+    Begin Project Dependency
+    Project_Dep_Name core
+    End Project Dependency
+    Begin Project Dependency
+    Project_Dep_Name xml
+    End Project Dependency
 }}}
 
 ###############################################################################
 
-Project: "media"=wx_media.dsp - Package Owner=<4>
+Project: "wxexpat"=.\wx_wxexpat.dsp - Package Owner=<4>
 
 Package=<5>
 {{{
@@ -134,7 +201,7 @@
 
 ###############################################################################
 
-Project: "odbc"=wx_odbc.dsp - Package Owner=<4>
+Project: "wxjpeg"=.\wx_wxjpeg.dsp - Package Owner=<4>
 
 Package=<5>
 {{{
@@ -146,7 +213,7 @@
 
 ###############################################################################
 
-Project: "dbgrid"=wx_dbgrid.dsp - Package Owner=<4>
+Project: "wxpng"=.\wx_wxpng.dsp - Package Owner=<4>
 
 Package=<5>
 {{{
@@ -158,7 +225,7 @@
 
 ###############################################################################
 
-Project: "html"=wx_html.dsp - Package Owner=<4>
+Project: "wxregex"=.\wx_wxregex.dsp - Package Owner=<4>
 
 Package=<5>
 {{{
@@ -170,7 +237,7 @@
 
 ###############################################################################
 
-Project: "qa"=wx_qa.dsp - Package Owner=<4>
+Project: "wxtiff"=.\wx_wxtiff.dsp - Package Owner=<4>
 
 Package=<5>
 {{{
@@ -182,7 +249,7 @@
 
 ###############################################################################
 
-Project: "xml"=wx_xml.dsp - Package Owner=<4>
+Project: "wxzlib"=.\wx_wxzlib.dsp - Package Owner=<4>
 
 Package=<5>
 {{{
@@ -194,7 +261,7 @@
 
 ###############################################################################
 
-Project: "xrc"=wx_xrc.dsp - Package Owner=<4>
+Project: "xml"=.\wx_xml.dsp - Package Owner=<4>
 
 Package=<5>
 {{{
@@ -202,17 +269,47 @@
 
 Package=<4>
 {{{
+    Begin Project Dependency
+    Project_Dep_Name base
+    End Project Dependency
 }}}
 
 ###############################################################################
 
-Project: "gl"=wx_gl.dsp - Package Owner=<4>
+Project: "xrc"=.\wx_xrc.dsp - Package Owner=<4>
 
 Package=<5>
 {{{
 }}}
 
 Package=<4>
+{{{
+    Begin Project Dependency
+    Project_Dep_Name adv
+    End Project Dependency
+    Begin Project Dependency
+    Project_Dep_Name base
+    End Project Dependency
+    Begin Project Dependency
+    Project_Dep_Name core
+    End Project Dependency
+    Begin Project Dependency
+    Project_Dep_Name xml
+    End Project Dependency
+    Begin Project Dependency
+    Project_Dep_Name html
+    End Project Dependency
+}}}
+
+###############################################################################
+
+Global:
+
+Package=<5>
+{{{
+}}}
+
+Package=<3>
 {{{
 }}}
 
Index: build/msw/wx_dll.dsw
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/build/msw/wx_dll.dsw,v
retrieving revision 1.5
diff -u -r1.5 wx_dll.dsw
--- build/msw/wx_dll.dsw	2005/05/29 11:30:44	1.5
+++ build/msw/wx_dll.dsw	2005/06/09 10:01:26
@@ -56,6 +56,15 @@
     Begin Project Dependency
     Project_Dep_Name wxzlib
     End Project Dependency
+    Begin Project Dependency
+    Project_Dep_Name wxtiff
+    End Project Dependency
+    Begin Project Dependency
+    Project_Dep_Name wxjpeg
+    End Project Dependency
+    Begin Project Dependency
+    Project_Dep_Name wxpng
+    End Project Dependency
 }}}

 	  	 
