Index: samples/statbar/statbar.cpp
===================================================================
--- samples/statbar/statbar.cpp	(revisione 71294)
+++ samples/statbar/statbar.cpp	(copia locale)
@@ -126,6 +126,8 @@
 #endif
     wxStaticBitmap *m_statbmp;
 
+    wxGauge* m_ProgressBarCtrl;
+
     DECLARE_EVENT_TABLE()
 };
 
@@ -916,6 +918,8 @@
 
     m_statbmp = new wxStaticBitmap(this, wxID_ANY, wxIcon(green_xpm));
 
+    m_ProgressBarCtrl = new wxGauge( this, wxID_ANY, 100, wxDefaultPosition, wxDefaultSize, wxGA_HORIZONTAL );
+
 #if wxUSE_TIMER
     m_timer.Start(1000);
 #endif
@@ -949,7 +953,7 @@
 
     // TEMPORARY HACK: TODO find a more general solution
 #ifdef wxStatusBarGeneric
-    wxStatusBar::OnSize(event);
+//    wxStatusBar::OnSize(event);
 #endif
 
     wxRect rect;
@@ -971,6 +975,21 @@
     m_statbmp->Move(rect.x + (rect.width - size.x) / 2,
                     rect.y + (rect.height - size.y) / 2);
 
+    {
+	wxRect tmpRectFirst, tmpRectLast;
+	
+	if( GetFieldsCount() > 0 ) {
+		// determine the area to be used by the gauge control
+		GetFieldRect(0, tmpRectFirst);
+		tmpRectFirst.x += 3;	// move a bit to the right
+		tmpRectFirst.width -= 3;
+		GetFieldRect(GetFieldsCount() - 1, tmpRectLast);
+		wxRect tmpRect = tmpRectFirst.Union( tmpRectLast );
+	
+		m_ProgressBarCtrl->SetSize( tmpRect );
+	}
+    }
+
     event.Skip();
 }
 
