Index: include/wx/listbase.h
===================================================================
--- include/wx/listbase.h	(revision 74251)
+++ include/wx/listbase.h	(working copy)
@@ -415,7 +415,7 @@
     //
     // Returns the index of the newly inserted column or -1 on error.
     long AppendColumn(const wxString& heading,
-                      int format = wxLIST_FORMAT_LEFT,
+                      wxListColumnFormat format = wxLIST_FORMAT_LEFT,
                       int width = -1);
 
     // Add a new column to the control at the position "col".
Index: interface/wx/listctrl.h
===================================================================
--- interface/wx/listctrl.h	(revision 74251)
+++ interface/wx/listctrl.h	(working copy)
@@ -324,7 +324,7 @@
         @since 2.9.4
      */
     long AppendColumn(const wxString& heading,
-                      int format = wxLIST_FORMAT_LEFT,
+                      wxListColumnFormat format = wxLIST_FORMAT_LEFT,
                       int width = -1);
 
     /**
Index: src/common/listctrlcmn.cpp
===================================================================
--- src/common/listctrlcmn.cpp	(revision 74251)
+++ src/common/listctrlcmn.cpp	(working copy)
@@ -138,7 +138,7 @@
 
 long
 wxListCtrlBase::AppendColumn(const wxString& heading,
-                             int format,
+                             wxListColumnFormat format,
                              int width)
 {
     return InsertColumn(GetColumnCount(), heading, format, width);
