diff -ru wxWidgets.orig/src/msw/renderer.cpp wxWidgets/src/msw/renderer.cpp
--- wxWidgets.orig/src/msw/renderer.cpp	2005-09-18 06:03:22.000000000 +1000
+++ wxWidgets/src/msw/renderer.cpp	2005-09-18 06:26:57.000000000 +1000
@@ -244,36 +244,18 @@
 {
     if ( !win->HasFlag(wxSP_NO_XP_THEME) )
     {
-        wxUxThemeHandle hTheme(win, L"REBAR");
-        if ( hTheme )
+        dc.SetPen(*wxTRANSPARENT_PEN);
+        dc.SetBrush(wxBrush(wxSystemSettings::GetColour(wxSYS_COLOUR_BTNFACE)));
+        if ( orient == wxVERTICAL )
         {
-            RECT rect;
-            if ( orient == wxVERTICAL )
-            {
-                rect.left = position;
-                rect.right = position + SASH_WIDTH;
-                rect.top = 0;
-                rect.bottom = size.y;
-            }
-            else // wxHORIZONTAL
-            {
-                rect.left = 0;
-                rect.right = size.x;
-                rect.top = position;
-                rect.bottom = position + SASH_WIDTH;
-            }
-
-            wxUxThemeEngine::Get()->DrawThemeBackground
-                                    (
-                                        hTheme,
-                                        (HDC) dc.GetHDC(),
-                                        29, // WP_DIALOG: dlg background
-                                        0, // no particular state
-                                        &rect,
-                                        NULL
-                                    );
-            return;
+            dc.DrawRectangle(position, 0, SASH_WIDTH, size.y);
+        }
+        else // wxHORIZONTAL
+        {
+            dc.DrawRectangle(0, position, size.x, SASH_WIDTH);
         }
+
+        return;
     }
 
     m_rendererNative.DrawSplitterSash(win, dc, size, position, orient, flags);

 	  	 
