Index: richtextctrl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/richtext/richtextctrl.cpp,v
retrieving revision 1.62.2.4
diff -u -r1.62.2.4 richtextctrl.cpp
--- richtextctrl.cpp	2007/03/05 16:13:22	1.62.2.4
+++ richtextctrl.cpp	2007/03/11 23:18:18
@@ -781,7 +781,11 @@
                     GetId());
                 cmdEvent.SetEventObject(this);
                 cmdEvent.SetFlags(flags);
+#if wxUSE_UNICODE
+                cmdEvent.SetCharacter(event.GetUnicodeKey());
+#else
                 cmdEvent.SetCharacter((wxChar) keycode);
+#endif
                 cmdEvent.SetPosition(m_caretPosition+1);
 
                 if (keycode == wxT('\t'))
@@ -813,7 +817,11 @@
                 long newPos = m_caretPosition;
                 DeleteSelectedContent(& newPos);
 
+#if wxUSE_UNICODE
+                wxString str = event.GetUnicodeKey();
+#else
                 wxString str = (wxChar) event.GetKeyCode();
+#endif
                 GetBuffer().InsertTextWithUndo(newPos+1, str, this, 0);
 
                 EndBatchUndo();

 	  	 
