Index: include/wx/gtk/listbox.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/gtk/listbox.h,v
retrieving revision 1.56
diff -b -u -2 -r1.56 listbox.h
--- include/wx/gtk/listbox.h	2006/03/09 12:46:22	1.56
+++ include/wx/gtk/listbox.h	2006/03/11 01:11:32
@@ -3,5 +3,5 @@
 // Purpose:     wxListBox class declaration
 // Author:      Robert Roebling
-// Id:          $Id: listbox.h,v 1.56 2006/03/09 12:46:22 VZ Exp $
+// Id:          $Id: listbox.h,v 1.55 2006/03/02 09:55:07 RR Exp $
 // Copyright:   (c) 1998 Robert Roebling
 // Licence:     wxWindows licence
@@ -19,5 +19,8 @@
 public:
     // ctors and such
-    wxListBox();
+    wxListBox()
+    {
+        Init();
+    }
     wxListBox( wxWindow *parent, wxWindowID id,
             const wxPoint& pos = wxDefaultPosition,
@@ -28,7 +31,5 @@
             const wxString& name = wxListBoxNameStr )
     {
-#if wxUSE_CHECKLISTBOX
-        m_hasCheckBoxes = FALSE;
-#endif // wxUSE_CHECKLISTBOX
+        Init();
         Create(parent, id, pos, size, n, choices, style, validator, name);
     }
@@ -41,10 +42,10 @@
             const wxString& name = wxListBoxNameStr )
     {
-#if wxUSE_CHECKLISTBOX
-        m_hasCheckBoxes = FALSE;
-#endif // wxUSE_CHECKLISTBOX
+        Init();
         Create(parent, id, pos, size, choices, style, validator, name);
     }
     virtual ~wxListBox();
+
+    void Init(); //common construction
 
     bool Create(wxWindow *parent, wxWindowID id,
Index: src/gtk/listbox.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/gtk/listbox.cpp,v
retrieving revision 1.171
diff -b -u -2 -r1.171 listbox.cpp
--- src/gtk/listbox.cpp	2006/03/09 13:47:25	1.171
+++ src/gtk/listbox.cpp	2006/03/11 01:11:32
@@ -420,5 +420,5 @@
 // ----------------------------------------------------------------------------
 
-wxListBox::wxListBox()
+void wxListBox::Init()
 {
     m_treeview = (GtkTreeView*) NULL;

 	  	 
