Index: include/wx/generic/fontdlgg.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/generic/fontdlgg.h,v
retrieving revision 1.18.2.1
diff -u -r1.18.2.1 fontdlgg.h
--- include/wx/generic/fontdlgg.h	2006/03/02 12:57:18	1.18.2.1
+++ include/wx/generic/fontdlgg.h	2006/05/16 15:30:42
@@ -52,6 +52,8 @@
 {
 public:
     wxGenericFontDialog() { Init(); }
+    wxGenericFontDialog(wxWindow *parent)
+        : wxFontDialogBase(parent) { Init(); }
     wxGenericFontDialog(wxWindow *parent, const wxFontData& data)
         : wxFontDialogBase(parent, data) { Init(); }
     virtual ~wxGenericFontDialog();

 	  	 
