--- include/wx/checklst.h
+++ include/wx/checklst.h
@@ -37,6 +37,8 @@ public:
     virtual bool IsChecked(unsigned int item) const = 0;
     virtual void Check(unsigned int item, bool check = true) = 0;
 
+    virtual unsigned int GetCheckedItems(wxArrayInt& checkedItems) const;
+
     wxDECLARE_NO_COPY_CLASS(wxCheckListBoxBase);
 };
 
--- src/common/checklstcmn.cpp
+++ src/common/checklstcmn.cpp
@@ -99,4 +99,19 @@ wxCONSTRUCTOR_4( wxCheckListBox, wxWindow*, Parent, wxWindowID, Id, \
                  wxPoint, Position, wxSize, Size )
 
 
+// ============================================================================
+// implementation
+// ============================================================================
+unsigned int wxCheckListBoxBase::GetCheckedItems(wxArrayInt& checkedItems) const
+{
+  unsigned int const numberOfItems = GetCount();
+
+
+  checkedItems.Empty();
+  for (unsigned int i=0; i<numberOfItems; ++i)
+    if (IsChecked(i))
+      checkedItems.Add(i);
+  return checkedItems.GetCount();
+}
+
 #endif
