Index: src/osx/carbon/region.cpp
===================================================================
--- src/osx/carbon/region.cpp	(revision 68277)
+++ src/osx/carbon/region.cpp	(working copy)
@@ -250,11 +257,15 @@
 //# Information on region
 //-----------------------------------------------------------------------------
 
-bool wxRegion::DoIsEqual(const wxRegion& WXUNUSED(region)) const
+bool wxRegion::DoIsEqual(const wxRegion& region) const
 {
-    wxFAIL_MSG( wxT("not implemented") );
+    wxRegion r = *this;
+    wxRegion r2 = region;
 
-    return false;
+    r.Subtract(region);
+    r2.Subtract(*this);
+
+    return r.IsEmpty() && r2.IsEmpty();
 }
 
 // Outer bounds of region
