Index: src/gtk/dcclient.cpp
===================================================================
--- src/gtk/dcclient.cpp	(revision 65839)
+++ src/gtk/dcclient.cpp	(working copy)
@@ -74,22 +74,22 @@
         switch (style)
         {
         case wxBRUSHSTYLE_BDIAGONAL_HATCH:
-            hatches[i] = gdk_bitmap_create_from_data(NULL, bdiag_bits, bdiag_width, bdiag_height);
+            hatches[i] = gdk_bitmap_create_from_data(NULL, reinterpret_cast<const char*>(bdiag_bits), bdiag_width, bdiag_height);
             break;
         case wxBRUSHSTYLE_CROSSDIAG_HATCH:
-            hatches[i] = gdk_bitmap_create_from_data(NULL, cdiag_bits, cdiag_width, cdiag_height);
+            hatches[i] = gdk_bitmap_create_from_data(NULL, reinterpret_cast<const char*>(cdiag_bits), cdiag_width, cdiag_height);
             break;
         case wxBRUSHSTYLE_CROSS_HATCH:
-            hatches[i] = gdk_bitmap_create_from_data(NULL, cross_bits, cross_width, cross_height);
+            hatches[i] = gdk_bitmap_create_from_data(NULL, reinterpret_cast<const char*>(cross_bits), cross_width, cross_height);
             break;
         case wxBRUSHSTYLE_FDIAGONAL_HATCH:
-            hatches[i] = gdk_bitmap_create_from_data(NULL, fdiag_bits, fdiag_width, fdiag_height);
+            hatches[i] = gdk_bitmap_create_from_data(NULL, reinterpret_cast<const char*>(fdiag_bits), fdiag_width, fdiag_height);
             break;
         case wxBRUSHSTYLE_HORIZONTAL_HATCH:
-            hatches[i] = gdk_bitmap_create_from_data(NULL, horiz_bits, horiz_width, horiz_height);
+            hatches[i] = gdk_bitmap_create_from_data(NULL, reinterpret_cast<const char*>(horiz_bits), horiz_width, horiz_height);
             break;
         case wxBRUSHSTYLE_VERTICAL_HATCH:
-            hatches[i] = gdk_bitmap_create_from_data(NULL, verti_bits, verti_width, verti_height);
+            hatches[i] = gdk_bitmap_create_from_data(NULL, reinterpret_cast<const char*>(verti_bits), verti_width, verti_height);
             break;
         }
     }
Index: src/gtk/bdiag.xbm
===================================================================
--- src/gtk/bdiag.xbm	(revision 65839)
+++ src/gtk/bdiag.xbm	(working copy)
@@ -1,6 +1,6 @@
 #define bdiag_width 16
 #define bdiag_height 16
-static char bdiag_bits[] = {
+static unsigned char bdiag_bits[] = {
    0x80, 0x80, 0x40, 0x40, 0x20, 0x20, 0x10, 0x10, 0x08, 0x08, 0x04, 0x04,
    0x02, 0x02, 0x01, 0x01, 0x80, 0x80, 0x40, 0x40, 0x20, 0x20, 0x10, 0x10,
    0x08, 0x08, 0x04, 0x04, 0x02, 0x02, 0x01, 0x01};
Index: src/gtk/cdiag.xbm
===================================================================
--- src/gtk/cdiag.xbm	(revision 65839)
+++ src/gtk/cdiag.xbm	(working copy)
@@ -1,6 +1,6 @@
 #define cdiag_width 16
 #define cdiag_height 16
-static char cdiag_bits[] = {
+static unsigned char cdiag_bits[] = {
    0x81, 0x81, 0x42, 0x42, 0x24, 0x24, 0x18, 0x18, 0x18, 0x18, 0x24, 0x24,
    0x42, 0x42, 0x81, 0x81, 0x81, 0x81, 0x42, 0x42, 0x24, 0x24, 0x18, 0x18,
    0x18, 0x18, 0x24, 0x24, 0x42, 0x42, 0x81, 0x81};
Index: src/gtk/fdiag.xbm
===================================================================
--- src/gtk/fdiag.xbm	(revision 65839)
+++ src/gtk/fdiag.xbm	(working copy)
@@ -1,6 +1,6 @@
 #define fdiag_width 16
 #define fdiag_height 16
-static char fdiag_bits[] = {
+static unsigned char fdiag_bits[] = {
    0x01, 0x01, 0x02, 0x02, 0x04, 0x04, 0x08, 0x08, 0x10, 0x10, 0x20, 0x20,
    0x40, 0x40, 0x80, 0x80, 0x01, 0x01, 0x02, 0x02, 0x04, 0x04, 0x08, 0x08,
    0x10, 0x10, 0x20, 0x20, 0x40, 0x40, 0x80, 0x80};
Index: src/gtk/horiz.xbm
===================================================================
--- src/gtk/horiz.xbm	(revision 65839)
+++ src/gtk/horiz.xbm	(working copy)
@@ -1,6 +1,6 @@
 #define horiz_width 15
 #define horiz_height 15
-static char horiz_bits[] = {
+static unsigned char horiz_bits[] = {
    0x00, 0x00, 0x00, 0x00, 0xff, 0x7f, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
    0x00, 0x00, 0xff, 0x7f, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
    0xff, 0x7f, 0x00, 0x00, 0x00, 0x00};
Index: src/gtk/cross.xbm
===================================================================
--- src/gtk/cross.xbm	(revision 65839)
+++ src/gtk/cross.xbm	(working copy)
@@ -1,6 +1,6 @@
 #define cross_width 15
 #define cross_height 15
-static char cross_bits[] = {
+static unsigned char cross_bits[] = {
    0x84, 0x10, 0x84, 0x10, 0xff, 0x7f, 0x84, 0x10, 0x84, 0x10, 0x84, 0x10,
    0x84, 0x10, 0xff, 0x7f, 0x84, 0x10, 0x84, 0x10, 0x84, 0x10, 0x84, 0x10,
    0xff, 0x7f, 0x84, 0x10, 0x84, 0x10};
Index: src/gtk/verti.xbm
===================================================================
--- src/gtk/verti.xbm	(revision 65839)
+++ src/gtk/verti.xbm	(working copy)
@@ -1,6 +1,6 @@
 #define verti_width 15
 #define verti_height 15
-static char verti_bits[] = {
+static unsigned char verti_bits[] = {
    0x84, 0x10, 0x84, 0x10, 0x84, 0x10, 0x84, 0x10, 0x84, 0x10, 0x84, 0x10,
    0x84, 0x10, 0x84, 0x10, 0x84, 0x10, 0x84, 0x10, 0x84, 0x10, 0x84, 0x10,
    0x84, 0x10, 0x84, 0x10, 0x84, 0x10};
